== Why is this an issue?

Well-named functions can allow the users of your code to understand at a glance what to expect from the function - even before reading the documentation. Toward that end, methods returning a boolean property should have names that start with "is" or "has" rather than with "get".


Note that this rule will only apply to functions that are documented to return a boolean.


=== Noncompliant code example

[source,php]
----
/**
 * @return boolean
 */
public function getFoo() // Noncompliant
{
  return foo;
}
----


=== Compliant solution

[source,php]
----
/**
 * @return boolean
 */
public function isFoo()
{
  return true;
}
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
