== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,vbnet]
----
Module Module1
    Sub Main()
        Dim foo() As String ' Noncompliant
    End Sub
End Module
----

=== Compliant solution

[source,vbnet]
----
Module Module1
    Sub Main()
        Dim foo As String() ' Compliant
    End Sub
End Module
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
