The use of a ``++pass++`` statement where it's not required by the syntax is pure cruft and should be removed.

== Noncompliant Code Example

----
    def __init__(self, log="", who="", date=0, files=[]):
        self.log = log
        self.files = files
        self.who = who
        self.date = date
        pass    # Noncompliant

    def lookup():
        pass    # Compliant; method can't be empty
----

== Compliant Solution

----
    def __init__(self, log="", who="", date=0, files=[]):
        self.log = log
        self.files = files
        self.who = who
        self.date = date

    def lookup():
        pass
----
