== Why is this an issue?

Nested control flow statements ``++if++``, ``++for++``, ``++for in++``, ``++while++``, ``++do while++`` and ``++switch++``  are often key ingredients in creating
what's known as "Spaghetti code". This code smell can make your program difficult to understand and maintain.

When numerous control structures are placed inside one another, the code becomes a tangled, complex web.
This significantly reduces the code's readability and maintainability, and it also complicates the testing process.

== How to fix it

=== Code examples

The following example demonstrates the behavior of the rule with the default threshold of 3 levels of nesting:

==== Noncompliant code example

[source,swift]
----
  if condition1 {                  // Compliant - depth = 1
    /* ... */
    if condition2 {                // Compliant - depth = 2
      /* ... */
      for var i = 0; i < 10; i++ {  // Compliant - depth = 3, not exceeding the limit
        /* ... */
        if condition4 {            // Non-Compliant - depth = 4
          if condition5 {          // Depth = 5, exceeding the limit, but issues are only reported on depth = 4
            /* ... */
          }
        }
      }
    }
  }
----

include::../resources.adoc[]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Refactor this code to not nest more than {} "if", "for", "for in", "while", "do while" and "switch" statements.


include::../parameters.adoc[]

include::../highlighting.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
