== Why is this an issue?

The ``++asm++`` declaration is available to all {cpp} implementations, allowing a consistent mechanism to be used.


However, the parameters to ``++asm++`` are still implementation-defined.


=== Noncompliant code example

[source,cpp]
----
void Delay ( void )
{
#pragma asm
  "NOP" // Noncompliant
#pragma endasm
}
----


=== Compliant solution

[source,cpp]
----
void Delay ( void )
{
  asm ( "NOP" ); // Compliant
}
----


== Resources

* MISRA {cpp} 2008, 7-4-2


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
