== Why is this an issue?

The ``++asm++`` declaration is available to all {cpp} implementations, allowing a consistent mechanism to be used.


However, the parameters to ``++asm++`` are still implementation-defined.


=== Noncompliant code example

[source,cpp]
----
void Delay ( void )
{
#pragma asm
  "NOP" // Noncompliant
#pragma endasm
}
----


=== Compliant solution

[source,cpp]
----
void Delay ( void )
{
  asm ( "NOP" ); // Compliant
}
----


== Resources

* MISRA {cpp} 2008, 7-4-2


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Refactor this code using the "asm" statement.


'''
== Comments And Links
(visible only on this page)

=== is related to: S784

=== on 16 Oct 2014, 14:56:27 Ann Campbell wrote:
\[~samuel.mercier] see also MISRA C:2012, Dir 4.3

In general, the 2012 version always has much better descriptions


This rule is also defined in MISRA C:2004, 2.1

endif::env-github,rspecator-view[]
