This rule reports circular dependencies between classes across different packages.

== Why is this an issue?

Circular dependencies occur when two classes reference each other, either directly or indirectly.
This means that the classes' dependency structure does not represent an intuitive hierarchy, which makes it harder to understand and maintain.

Compared to S7027, this issue is even more significant when a cycle crosses package boundaries because:

1. It disrupts the dependency hierarchy or layer structure even between packages.
2. It spreads the problem across more distant locations in the source code.

=== What is the potential impact?

Circular dependencies increase the architectural complexity of the code, reducing its readability, extensibility, and maintainability.
As the project grows, circular dependencies often lead to more circular dependencies, further complication of the architecture and increasing technical debt.
Over time, untangling these dependencies becomes increasingly difficult.

== How to fix it

1. **Extract an interface**: If two classes share similar functionality, move that functionality to an interface that both can implement. This allows each class to depend on the interface rather than on each other.

2. **Use dependency injection**: Replace direct references between classes with dependency injection. By passing necessary objects as parameters through constructors or setters, you break the circular reference, making your code more flexible and easier to test.

3. **Split responsibilities**: Evaluate whether each class is handling too many responsibilities. If so, break them down into smaller, more focused classes. This eliminates circular dependencies and makes sure your code has fewer reasons to change, making it easier to manage and extend.

4. **Rethink the package structure**

=== Code examples

==== Noncompliant code example

The following code contains two cycles: Order &rarr; Customer &rarr; Order, and Order &rarr; Product &rarr; Order. Furthermore, both cycles are connected to each other via class Order. This is called _a tangle consisting of two cycles_.

[source,java,diff-id=1,diff-type=noncompliant]
----
class Order {
    public Customer customer;
    public List<Product> products;
}
class Customer {
    public List<Order> orders;
}
class Product {
    public List<Order> orders;
}
----

==== Compliant solution

The issue can be resolved by changing the structure. Two service functions can replace the dependencies Customer &rarr; Order and Product &rarr; Order.

[source,java,diff-id=1,diff-type=compliant]
----
class Order {
    public Customer customer;
    public List<Product> products;
}
class Customer { }
class Product { }
interface OrderService {
  List<Order> getOrdersByCustomer(Customer customer);
  List<Order> getOrdersByProduct(Product product);
}
----

== Resources

- Wikipedia - https://en.wikipedia.org/wiki/Acyclic_dependencies_principle[Acyclic dependencies principle]
- STAN - https://stan4j.com/advanced/adp/[Acyclic dependencies principle]
- RSPEC - https://sonarsource.github.io/rspec/#/rspec/S7027/java[S7027: Circular dependencies between classes in the same package should be resolved]
