include::../rule.adoc[]

=== Noncompliant code example

[source,csharp]
----
using System;

public class MyAttribute: Attribute // Noncompliant
{
    public string Name { get; }

    public MyAttribute(string name) =>
        Name = name;
}
----

=== Compliant solution

[source,csharp]
----
using System;

public sealed class MyAttribute : Attribute
{
    public string Name { get; }

    public MyAttribute(string name) =>
        Name = name;
}
----

include::../rspecator.adoc[]
