include::../rule.adoc[]

== Noncompliant Code Example

[source,csharp]
----
using System;

public class MyAttribute: Attribute // Noncompliant
{
    public string Name { get; }

    public MyAttribute(string name) =>
        Name = name;
}
----

== Compliant Solution

[source,csharp]
----
using System;

public sealed class MyAttribute : Attribute
{
    public string Name { get; }

    public MyAttribute(string name) =>
        Name = name;
}
----

include::../rspecator.adoc[]