== Why is this an issue?

The use of operators pairs ( ``++=+++`` or ``++=-++``) where the reversed, single operator was meant (``+++=++`` or ``++-=++``) will run fine, but not produce the expected results.


This rule raises an issue when ``++=+++`` or ``++=-++`` is used without any spacing between the two operators and when there is at least one whitespace character after.

=== Noncompliant code example

[source,python]
----
target = -5
num = 3

target =- num  # Noncompliant; target = -3. Is that really what's meant?
target =+ num # Noncompliant; target = 3
----

=== Compliant solution

[source,python]
----
target = -5
num = 3

target = -num  # Compliant; intent to assign inverse value of num is clear
target += num
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
