<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>The system field <code>SY-SUBRC</code> must be tested immediately after any statement setting this variable. Reading this variable informs on previous operation success or errors. Such errors should be handled properly so that the program continues in a consistent state.</p>
</div>
<div class="paragraph">
<p>This rule raises an issue when the field <code>SY-SUBRC</code> is not checked just after performing one of the following operations:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>Calling a function or method which can throw exceptions.</p>
</li>
<li>
<p>Calling one of the file access operation <code>OPEN DATASET</code>, <code>READ DATASET</code> or <code>DELETE DATASET</code>.</p>
</li>
</ul>
</div>
<div class="paragraph">
<p><code>SY-SUBRC</code> check must be done either with the <code>CASE</code>, <code>IF</code> or <code>CHECK</code> statement.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="paragraph">
<p>In the following case nothing happens if the exceptions <code>NOT_FOUND</code> or <code>OTHERS</code> are raised:</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-abap" data-lang="abap">CALL FUNCTION 'STRING_SPLIT'
  EXPORTING
    DELIMITER = ':'
    STRING = FELD
  IMPORTING
    HEAD =   HEAD
    TAIL = TAIL
  EXCEPTIONS
    NOT_FOUND = 1
    OTHERS = 2.</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-abap" data-lang="abap">CALL FUNCTION 'STRING_SPLIT'
  EXPORTING
    DELIMITER = ':'
    STRING = FELD
  IMPORTING
    HEAD =   HEAD
    TAIL = TAIL
  EXCEPTIONS
    NOT_FOUND = 1
    OTHERS = 2.
CASE SY-SUBRC.
  WHEN 1. ...
  WHEN 2. ...
  WHEN OTHER.
ENDCASE.</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_exceptions">Exceptions</h3>
<div class="paragraph">
<p>No issue will be raised in the following cases:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>One or more <code>WRITE</code> operation are performed between the statement setting <code>SY-SUBRC</code> and its check. An exception will be however raised if the <code>WRITE</code> operation is a <code>WRITE ... TO</code>, as this will set <code>SY-SUBRC</code> too.</p>
</li>
<li>
<p><code>SY-SUBRC</code>'s value is assigned to a variable. We then assume that it will be checked later.</p>
</li>
</ul>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-abap" data-lang="abap">OPEN DATASET my_dataset FOR INPUT IN TEXT MODE ENCODING DEFAULT. " Compliant
WRITE 'Test'. " WRITE is accepted before checking SY-SUBRC
IF SY-SUBRC &lt;&gt; 0.
    EXIT.
ENDIF.

OPEN DATASET my_dataset FOR INPUT IN TEXT MODE ENCODING DEFAULT. " Compliant
Tmp = SY-SUBRC. " Assigning SY-SUBRC value to a variable. We assume that it will be checked later.
IF Tmp &lt;&gt; 0.
    EXIT.
ENDIF.</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="ulist">
<ul>
<li>
<p>SY-SUBRC must be tested after each call to "XXX"</p>
</li>
<li>
<p>SY-SUBRC must be tested after calling (OPEN|READ|DELETE) DATASET</p>
</li>
</ul>
</div>
</div>
</div>
</div>