== How to fix it in dart:io library

=== Code examples

include::../../common/fix/code-rationale.adoc[]

:cert_method_name: badCertificateCallback

include::../../common/fix/code-rationale-override.adoc[]

==== Noncompliant code example

[source,dart]
----
class MyHttpOverrides extends HttpOverrides {
  @override
  HttpClient createHttpClient(context) {
    return super.createHttpClient(context)
      ..badCertificateCallback = (cert, String host, int port) => true; // Noncompliant
  }
}
----

=== How does this work?

include::../../common/fix/validation.adoc[]
