== How to fix it in dart:io library

=== Code examples

include::../../common/fix/code-rationale.adoc[]

The certificate validation gets disabled by implementing a `badCertificateCallback` that always return `true`. It is highly recommended to use the original implementation and to avoid any bypass.

==== Noncompliant code example

[source,dart]
----
class MyHttpOverrides extends HttpOverrides {
  @override
  HttpClient createHttpClient(context) {
    return super.createHttpClient(context)
      ..badCertificateCallback = (cert, String host, int port) => true; // Noncompliant
  }
}
----

=== How does this work?

include::../../common/fix/validation.adoc[]
