This rule raises an issue when a TimeUnit conversion method is called to convert to the same unit as the source. The rule applies to all TimeUnit conversion methods like `toSeconds()`, `toMinutes()`, `toHours()`, etc., when the source and target units are identical (e.g., `TimeUnit.SECONDS.toSeconds()`).

== Why is this an issue?

Converting a time value from one unit to the same unit is redundant and serves no purpose. This typically indicates a copy-paste error, leftover code from refactoring, or a misunderstanding of TimeUnit conversions.

These redundant conversions make code harder to read and may mislead other developers about the intended behavior.

=== What is the potential impact?

This reduces code clarity and may indicate a logical error where a different unit conversion was intended.

== How to fix it

Remove the redundant conversion and use the value directly.

=== Code examples

==== Noncompliant code example

[source,java,diff-id=1,diff-type=noncompliant]
----
long durationInSeconds = TimeUnit.SECONDS.toSeconds(duration); // Noncompliant
----

==== Compliant solution

[source,java,diff-id=1,diff-type=compliant]
----
long durationInSeconds = duration;
----

== Resources

=== Documentation

 * TimeUnit (Java SE 17 & JDK 17) - https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/util/concurrent/TimeUnit.html[Official Java documentation for the TimeUnit enum and its conversion methods]

=== Related rules

 * RSPEC-1905 - https://rules.sonarsource.com/java/RSPEC-1905/[Redundant casts should not be used - similar pattern of removing unnecessary operations]
