This is an issue when code attempts to directly convert between `LocalDate` and `Instant` without specifying timezone information, which will cause a `DateTimeException` at runtime.

== Why is this an issue?

`LocalDate` represents a date without time or timezone (like "2023-12-25"), while `Instant` represents a specific point in time on the UTC timeline.

Direct conversions fail because converting `LocalDate` to `Instant` requires time and timezone information, while converting `Instant` to `LocalDate` requires timezone context. Java throws a `DateTimeException` because it cannot assume missing temporal information.

=== What is the potential impact?

This causes runtime failures with `DateTimeException`, leading to application crashes. It's particularly problematic in production where timezone bugs may only surface in different geographical regions.

== How to fix it

Convert LocalDate to Instant via LocalDateTime and specify timezone.

=== Code examples

==== Noncompliant code example

[source,java,diff-id=1,diff-type=noncompliant]
----
LocalDate date = LocalDate.of(2023, 12, 25);
Instant instant = Instant.from(date); // Noncompliant - throws DateTimeException
----

==== Compliant solution

[source,java,diff-id=1,diff-type=compliant]
----
LocalDate date = LocalDate.of(2023, 12, 25);
Instant instant = date.atStartOfDay(ZoneId.systemDefault()).toInstant();
// or with explicit timezone
Instant instant2 = date.atStartOfDay().toInstant(ZoneOffset.UTC);
----

== Resources

=== Documentation

 * LocalDate JavaDoc - https://docs.oracle.com/javase/8/docs/api/java/time/LocalDate.html[Official documentation for LocalDate class]

 * Instant JavaDoc - https://docs.oracle.com/javase/8/docs/api/java/time/Instant.html[Official documentation for Instant class]
