<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>The catch-all handler, written <code>catch(...)</code> in C&#43;&#43;, or <code>@catch(...)</code> in Objective-C, catches every type of exception. If there is another catch statement for a specific exception after the catch-all handler, it will not be executed because the catch-all handler will already have handled the exception.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="paragraph">
<p>The following C&#43;&#43; example is similar in Objective-C: the <code>try</code> and <code>catch</code> equivalents are <code>@try</code> and <code>@catch</code>.</p>
</div>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">void f() {
  try {
    // ...
  } catch (...) {
    // Handle all exception types
  } catch (std::exception const &amp;e) { // Noncompliant: it will never be called
  }
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">void f() {
  try {
    // ...
  } catch (std::exception const &amp;e) {
    // Handle standard exceptions
  } catch (...) { // Compliant: handle all other exception types
  }
}</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>C&#43;&#43; reference - <a href="https://en.cppreference.com/w/cpp/language/try_catch">try-block</a></p>
</li>
<li>
<p>Apple Developer Documentation - <a href="https://developer.apple.com/library/archive/documentation/Cocoa/Conceptual/Exceptions/Tasks/HandlingExceptions.html">Objective-C: Handling Exceptions</a></p>
</li>
</ul>
</div>
</div>
<div class="sect2">
<h3 id="_external_coding_guidelines">External coding guidelines</h3>
<div class="ulist">
<ul>
<li>
<p>MISRA C&#43;&#43;:2008, 15-3-7 - Where multiple handlers are provided in a single try-catch statement or function-try-block, any ellipsis (catch-all) handler shall occur last.</p>
</li>
</ul>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_is_related_to_s1045">is related to: <a data-rspec-id="S1045" class="rspec-auto-link">S1045</a></h3>

</div>
<div class="sect2">
<h3 id="_on_30_jan_2020_162541_nicolas_harraudeau_wrote">on 30 Jan 2020, 16:25:41 Nicolas Harraudeau wrote:</h3>
<div class="paragraph">
<p>This rule has little value for python as Python interpreter will raise a <code>SyntaxError</code>.</p>
</div>
</div>
</div>
</div>