== Why is this an issue?

Erasing operations (e.g., any operation involving multiplying by zero, dividing zero by a value, or bitwise AND with zero) always result in zero regardless of the operands, which makes the expression unnecessary and can be replaced directly with zero. This simplifies the code and avoids potential confusion about the intent of the expression.

=== Code examples

==== Noncompliant code example

[source,rust,diff-id=1,diff-type=noncompliant]
----
let x = 1;
let result = 0 * x; // Noncompliant: Result is always zero.
----

==== Compliant solution

[source,rust,diff-id=1,diff-type=compliant]
----
let x = 1;
let result = 0; // Compliant: Simplified expression.
----

== Resources
=== Documentation

* Clippy Lints - https://rust-lang.github.io/rust-clippy/master/index.html#erasing_op
