== Why is this an issue?

The fact that an enumeration type is actually an enumeration should not be duplicated in its name.


=== Noncompliant code example

[source,text]
----
Enum FooEnum ' Noncompliant
    Foo
End Enum
----


=== Compliant solution

[source,text]
----
Enum Foo
    Foo
End Enum
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Rename "xxx" to remove the "Enum" suffix.


endif::env-github,rspecator-view[]
