The fact that an enumeration type is actually an enumeration should not be duplicated in its name.


== Noncompliant Code Example

[source,text]
----
Enum FooEnum ' Noncompliant
    Foo
End Enum
----


== Compliant Solution

[source,text]
----
Enum Foo
    Foo
End Enum
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

endif::env-github,rspecator-view[]
