include::../description.adoc[]

== Noncompliant Code Example

With default provided regular expression ``++^([A-Z0-9_]*|[a-z0-9_]*)$++``:

[source,abap]
----
METHOD MyMethod
...
ENDMETHOD.
----

== Compliant Solution

[source,abap]
----
METHOD MY_METHOD
...
ENDMETHOD.
----
