This rule raises an issue when a class constant (a variable declared with ``++static final++`` modifiers) uses a naming convention other than SNAKE_CASE with all uppercase letters.

== Why is this an issue?

Consistent naming conventions are essential for code maintainability and team collaboration. In Apex, the established convention for constants is to use SNAKE_CASE with all uppercase letters, where words are separated by underscores.

When constants don't follow this convention, it creates several problems:

* **Reduced readability**: Mixed naming styles make it harder to quickly identify constants in the code
* **Team confusion**: Developers expect constants to follow the established pattern, and deviations can cause misunderstandings
* **Maintenance overhead**: Inconsistent naming makes code reviews more difficult and increases the cognitive load when working with the codebase

The SNAKE_CASE convention for constants is widely adopted across many programming languages and is specifically recommended in Salesforce development best practices. This convention makes constants easily distinguishable from variables and methods at a glance.

=== What is the potential impact?

While this issue doesn't affect functionality, it reduces code maintainability and can slow down development. Inconsistent naming conventions make the codebase harder to understand and maintain, especially for new team members or when working with large codebases.

== How to fix it

Rename the constant to use SNAKE_CASE with all uppercase letters. Replace any lowercase letters with uppercase and separate words with underscores.

=== Code examples

==== Noncompliant code example

[source,apex,diff-id=1,diff-type=noncompliant]
----
private static final String accountLimit = '10'; // Noncompliant
----

==== Compliant solution

[source,apex,diff-id=1,diff-type=compliant]
----
private static final String ACCOUNT_LIMIT = '10';
----

== Resources

=== Documentation

 * Salesforce Naming Conventions Best Practices - https://www.apexhours.com/salesforce-naming-conventions-best-practices[Comprehensive guide to Salesforce naming conventions including constants]

=== Related rules

 * S115
