:image_type: Dockerfile

include::../common/description.adoc[]

== How to fix it

include::../common/how-to-fix.adoc[]

=== Code examples

==== Noncompliant code example

[source,docker,diff-id=1,diff-type=noncompliant]
----
name: Example

on:
  pull_request:
jobs:
  alpine-nopin:
    container: alpine  # Noncompliant

    steps:
      - run: |
          cat /etc/os-release

  alpine-latest:
    container:
      image: alpine:latest  # Noncompliant

    steps:
      - run: |
          cat /etc/os-release
----

==== Compliant solution

[source,docker,diff-id=1,diff-type=compliant]
----
name: Example

on:
  pull_request:
jobs:
  alpine-pin-version:
    container: alpine:3.21

    steps:
      - run: |
          cat /etc/os-release

  alpine-pin-hash:
    container:
      image: alpine@sha256:4bcff63911fcb4448bd4fdacec207030997caf25e9bea4045fa6c8c44de311d1

    steps:
      - run: |
          cat /etc/os-release
----

include::../common/how-does-this-work.adoc[]

include::../common/extra-mile.adoc[]

== Resources
=== Documentation

* GitHub Docs - https://docs.github.com/en/actions/how-tos/writing-workflows/choosing-where-your-workflow-runs/running-jobs-in-a-container[Running jobs in a container]
* Docker Docs - https://docs.docker.com/engine/reference/commandline/image_pull/#pull-an-image-by-digest-immutable-identifier[Pull an image by digest (immutable identifier)]

include::../common/implementation-specific.adoc[]
