<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>In Node.js, it&#8217;s possible to import modules by specifying an absolute path, such as <code>/lib/foo/bar.js</code>. However, this approach can limit the portability of your code, as it becomes tied to your computer&#8217;s file system. This could potentially lead to problems when the code is distributed, for instance, via NPM packages. Therefore, it&#8217;s advisable to use relative paths or module names for importing modules to enhance the portability and compatibility of your code across different systems.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Replace the absolute path with one that is relative to your current file.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-js" data-lang="js">import { foo } from '/home/project/api/bar.js';</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-js" data-lang="js">import { foo } from '../api/bar.js';</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p><a href="https://github.com/import-js/eslint-plugin-import">eslint-plugin-import</a> - Rule <a href="https://github.com/import-js/eslint-plugin-import/blob/HEAD/docs/rules/no-absolute-path.md">no-absolute-path</a></p>
</li>
<li>
<p>MDN web docs - <a href="https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/import">import</a></p>
</li>
<li>
<p>Node.js docs - <a href="https://nodejs.org/api/esm.html">ECMAScript modules</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>