include::../description.adoc[]

=== Noncompliant code example

[source,rust,diff-id=1,diff-type=noncompliant]
----
Regex::new("(")
----


=== Compliant solution

[source,rust,diff-id=1,diff-type=compliant]
----
Regex::new("\(")
----

== Resources

=== Documentation

* Regex crate documentation - https://crates.io/crates/regex[regex]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Invalid regular expression syntax.

=== Highlighting

The error inside the regular expression

endif::env-github,rspecator-view[]