<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Jumping back to a previous statement using <code>GOTO</code> is a way to reimplement loops, which PL/SQL already provides in much more readable forms.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-sql" data-lang="sql">SET SERVEROUTPUT ON

DECLARE
  result PLS_INTEGER := 0;
  counter PLS_INTEGER := 1;
BEGIN
  &lt;&lt;loop&gt;&gt;
  result := result + counter;
  counter := counter + 1;

  IF counter &lt;= 9 THEN
    GOTO loop;                    -- Noncompliant
  END IF;

  DBMS_OUTPUT.PUT_LINE('Sum from 1 to 9 is ' || result); -- Displays 1 + 2 + ... + 8 + 9 = 45
END;
/</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-sql" data-lang="sql">SET SERVEROUTPUT ON

DECLARE
  result PLS_INTEGER := 0;
BEGIN
  FOR counter IN 1 .. 9 LOOP
    result := result + counter;
  END LOOP;

  DBMS_OUTPUT.PUT_LINE('Sum from 1 to 9 is ' || result); -- Displays 1 + 2 + ... + 8 + 9 = 45
END;
/</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Replace this backwards GOTO by a loop.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_22_may_2013_093952_fabrice_bellingard_wrote">on 22 May 2013, 09:39:52 Fabrice Bellingard wrote:</h3>
<div class="paragraph">
<p>Implementation: http://jira.sonarsource.com/browse/PLSQL-222</p>
</div>
</div>
<div class="sect2">
<h3 id="_relates_to_s1064">relates to: <a data-rspec-id="S1064" class="rspec-auto-link">S1064</a></h3>

</div>
<div class="sect2">
<h3 id="_on_6_apr_2015_141414_evgeny_mandrikov_wrote">on 6 Apr 2015, 14:14:14 Evgeny Mandrikov wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] I&#8217;m wondering if SQALE is missing on purpose or not?</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_13_apr_2015_193157_evgeny_mandrikov_wrote">on 13 Apr 2015, 19:31:57 Evgeny Mandrikov wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] I&#8217;m wondering why blocker, but not active by default? Note that in implementation currently major and active.</p>
</div>
</div>
</div>
</div>