<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Both the "&#43;" operator and the "&amp;" operator can be used to concatenate strings. However, there is a complicated set of rules around the actual  behavior of the "+" operator, and whether you will get a concatenation operation, addition, a compiler error, or a <code>Type mismatch</code> error. On the other hand, the "&amp;" operator can only perform concatenation, and is therefore preferred.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-vb6" data-lang="vb6">Dim y As String = "Con" + "caten" + "ation"</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-vb6" data-lang="vb6">Dim y As String = "Con" &amp; "caten" &amp; "ation"</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Switch this use of the "+" operator to the "&amp;".</p>
</div>
</div>
</div>
</div>