== Why is this an issue?

The function ``++strerror++`` returns a buffer that is only valid until the next call to ``++strerror++``. In a multithreaded environment, you don't know when this next call will happen, which makes this function dangerous to call. You should use thread-safe alternatives, such as https://en.cppreference.com/w/c/string/byte/strerror[``++strerror_s++``] or ``++strerror_r++``.


Note that ``++strerror_s++`` is defined in annex K of C11, so to have access to it, you need a standard library that supports it (this can be tested with the macro ``++__STDC_LIB_EXT1__++``), and you need to enable it by defining the macro ``++__STDC_WANT_LIB_EXT1__++`` before including ``++<string.h>++``.


=== Noncompliant code example

[source,cpp]
----
char *msg = strerror(errno);{code}
----

=== Compliant solution

[source,cpp]
----
size_t size = strerrorlen_s(errno);
char *msg = malloc(size);
strerror_s(msg, size);{code}
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

endif::env-github,rspecator-view[]
