Starting from {cpp}20, mathematical constants are defined in the header ``++<numbers>++``.

You should prefer using them instead of hardcoding your own constants.


== Noncompliant Code Example

----
const double pi = 3.14159265358979323846; // Noncompliant

auto computeCirclePerimeter(double radius) {
    return 2 * pi * radius;
}
----


== Compliant Solution

----
#include <numbers>

auto computeCirclePerimeter(double radius) {
    return 2 * std::numbers::pi * radius;
}
----


ifdef::env-github,rspecator-view[]
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
