<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>It can be extremely confusing when a <code>for</code> condition tests a variable which is not updated inside the <code>for</code> post statement.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-go" data-lang="go">for i := 1; i &lt;= 10; j++ { // Noncompliant
	// ...
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-go" data-lang="go">for i := 1; i &lt;= 10; i++ {
	// ...
}</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>This loop&#8217;s stop condition tests variable which is not incremented in update clause.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="ulist">
<ul>
<li>
<p>primary: <code>(initializers; condition; updates)</code></p>
</li>
<li>
<p>secondary: where the counter is actually incremented</p>
<div class="ulist">
<ul>
<li>
<p>message: 'Move this update of "x".'</p>
</li>
</ul>
</div>
</li>
</ul>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_is_related_to_s886">is related to: <a data-rspec-id="S886" class="rspec-auto-link">S886</a></h3>

</div>
<div class="sect2">
<h3 id="_on_15_feb_2018_144443_alban_auzeill_wrote">on 15 Feb 2018, 14:44:43 Alban Auzeill wrote:</h3>
<div class="paragraph">
<p>Irrelevant for C, C&#43;&#43;, Objective-C, because this case is handled by [<a data-rspec-id="S886" class="rspec-auto-link">RSPEC-886</a>]</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_16_jul_2019_105331_quentin_jaquier_wrote">on 16 Jul 2019, 10:53:31 Quentin Jaquier wrote:</h3>
<div class="paragraph">
<p>Irrelevant for Scala, because the <a href="https://docs.scala-lang.org/tour/for-comprehensions.html">syntax</a> does not allow it.</p>
</div>
</div>
</div>
</div>