== Why is this an issue?

include::description.adoc[]

=== Noncompliant code example

[source,text]
----
for (i = 0; i < 10; j++) { // Noncompliant 
  // ... 
  i++; 
} 
----


=== Compliant solution

[source,text]
----
for (i = 0; i < 10; i++, j++) { 
  // ... 
} 
----
Or 

[source,text]
----
for (i = 0; i < 10; i++) { 
  // ... 
  j++; 
} 
----

