This rule raises an issue when the dimensions of the input arrays for the ``++numpy.dot++`` function are not compatible for matrix multiplication.

== Why is this an issue?

The ``++numpy.dot()++`` function (or equivalently, the @ operator for matrix multiplication) performs matrix multiplication between two arrays.
For matrix multiplication to be valid, the number of columns in the first matrix must be equal to the number of rows in the second matrix.
You can use ``++numpy.shape()++`` function to check the dimensions before performing the operation.

=== Code examples

==== Noncompliant code example

[source,python,diff-id=1,diff-type=noncompliant]
----
import numpy as np

a = [[1, 2, 3], [4, 5, 6]]
b = [1, 2, 3, 4]
c = np.dot(a, b) # Noncompliant: ValueError: shapes (2,3) and (4,) not aligned: 3 (dim 1) != 4 (dim 0)
    ...
----


==== Compliant solution

[source,python,diff-id=1,diff-type=compliant]
----
import numpy as np

a = [[1, 2, 3], [4, 5, 6]]
b = [1, 2, 3]
c = np.dot(a, b)
----

== Resources
=== Documentation
* NumPy API Reference - https://numpy.org/doc/stable/reference/generated/numpy.dot.html[numpy.dot]
* PSU - Matrix algebra review - https://online.stat.psu.edu/statprogram/reviews/matrix-algebra/arithmetic#paragraph--2761[Matrix Multiplication]


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Use 'numpy.isnan()' function instead of direct comparison.


'''
== Comments And Links
(visible only on this page)

endif::env-github,rspecator-view[]
