include::../description.adoc[]

== Noncompliant Code Example

----
class Child extends Parent {

  public function func($n,$m) {
    parent::func($n$m);  // Noncompliant
  }
}

class Parent {
  public function func($n, $m) {
    // do something
  }
}
----

== Compliant Solution

----
class Child extends Parent {

  public function func($n,$m) {
    parent::func($n$m);  
    // do additional things...
  }
}

class Parent {
  public function func($n, $m) {
    // do something
  }
}
----
or

----
class Child extends Parent {
  // function eliminated
}

class Parent {
  public function func($n, $m) {
    // do something
  }
}
----

ifdef::env-github,rspecator-view[]
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
