== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,php]
----
class Child extends Parent {

  public function func($n,$m) {
    parent::func($n$m);  // Noncompliant
  }
}

class Parent {
  public function func($n, $m) {
    // do something
  }
}
----

=== Compliant solution

[source,php]
----
class Child extends Parent {

  public function func($n,$m) {
    parent::func($n$m);  
    // do additional things...
  }
}

class Parent {
  public function func($n, $m) {
    // do something
  }
}
----
or

[source,php]
----
class Child extends Parent {
  // function eliminated
}

class Parent {
  public function func($n, $m) {
    // do something
  }
}
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
