<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Since Oracle 10g, <code>DBMS_UTILITY.FORMAT_ERROR_BACKTRACE</code> is available to get an exception&#8217;s stack trace, i.e. files and lines that lead up to the exception. When combined with <code>DBMS_UTILITY.FORMAT_ERROR_STACK</code>, which contains the exception error code and message, developers are able quickly identify defects.</p>
</div>
<div class="paragraph">
<p>This rule verifies that whenever either is used in an exception handler, the other is used as well.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-sql" data-lang="sql">BEGIN
  RAISE_APPLICATION_ERROR(-20000, 'This is an error example');
EXCEPTION
  WHEN OTHERS THEN  -- Noncompliant; only FORMAT_ERROR_STACK is used
    DBMS_OUTPUT.PUT(DBMS_UTILITY.FORMAT_ERROR_STACK);           -- "ORA-20000: This is an error example"
    DBMS_OUTPUT.PUT_LINE('');
END;
/</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-sql" data-lang="sql">BEGIN
  RAISE_APPLICATION_ERROR(-20000, 'This is an error example');
EXCEPTION
  WHEN OTHERS THEN
    DBMS_OUTPUT.PUT(DBMS_UTILITY.FORMAT_ERROR_STACK);           -- "ORA-20000: This is an error example"
    DBMS_OUTPUT.PUT(DBMS_UTILITY.FORMAT_ERROR_BACKTRACE);       -- "ORA-06512: at line 2"
    DBMS_OUTPUT.PUT_LINE('');
END;
/</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Use both "FORMAT_ERROR_STACK" and "FORMAT_ERROR_BACKTRACE" in this exception handler.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_23_may_2013_140521_dinesh_bolkensteyn_wrote">on 23 May 2013, 14:05:21 Dinesh Bolkensteyn wrote:</h3>
<div class="paragraph">
<p><a href="http://rajeshwaranbtech.blogspot.fr/2010/05/dbmsutilityformaterrorbacktrace-10g-new.html" class="bare">http://rajeshwaranbtech.blogspot.fr/2010/05/dbmsutilityformaterrorbacktrace-10g-new.html</a></p>
</div>
</div>
</div>
</div>