<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>In software development, logs serve as a record of events within an application, providing crucial insights for debugging.
When logging, it is essential to ensure that the logs are:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>easily accessible</p>
</li>
<li>
<p>uniformly formatted for readability</p>
</li>
<li>
<p>properly recorded</p>
</li>
<li>
<p>securely logged when dealing with sensitive data</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>Those requirements are not met if a program directly writes to the standard outputs (e.g., print!, println!).
That is why defining and using a dedicated logger is highly recommended.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-rust" data-lang="rust">fn do_something() {
    println!("my message");  // Noncompliant, output directly to stdout without a logger
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-rust" data-lang="rust">use log::{info, LevelFilter};
use simple_logger::SimpleLogger;

fn do_something() {
    SimpleLogger::new().with_level(LevelFilter::Info).init().unwrap();
    // ...
    info!("my message");  // Compliant, output via logger
    // ...
}</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Clippy Lints - <a href="https://rust-lang.github.io/rust-clippy/master/index.html#print_stdout" class="bare">https://rust-lang.github.io/rust-clippy/master/index.html#print_stdout</a></p>
</li>
<li>
<p>OWASP - <a href="https://owasp.org/Top10/A09_2021-Security_Logging_and_Monitoring_Failures/">Top 10 2021 Category A9 - Security Logging and Monitoring Failures</a></p>
</li>
<li>
<p>OWASP - <a href="https://owasp.org/www-project-top-ten/2017/A3_2017-Sensitive_Data_Exposure">Top 10 2017 Category A3 - Sensitive Data Exposure</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>