<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p><code>Double.longBitsToDouble</code> converts the bit pattern into its corresponding floating-point representation.
The method expects a 64-bit long argument to interpret the bits as a double value correctly.</p>
</div>
<div class="paragraph">
<p>When the argument is a smaller data type, the cast to <code>long</code> may lead to a different value than expected due to
the interpretation of the most significant bit, which, in turn, results in <code>Double.longBitsToDouble</code> returning an incorrect value.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">int i = 0x80003800;
Double.longBitsToDouble(i);   // Noncompliant - NaN</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">long i = 0x80003800L;
Double.longBitsToDouble(i);   // Compliant - 1.0610049784E-314</code></pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p><a href="https://docs.oracle.com/en/java/javase/20/docs/api/java.base/java/lang/Double.html#doubleToLongBits(double)">Oracle Java SE - Double.doubleToLongBits</a></p>
</li>
</ul>
</div>
</div>
<div class="sect2">
<h3 id="_articles_blog_posts">Articles &amp; blog posts</h3>
<div class="ulist">
<ul>
<li>
<p><a href="https://en.wikipedia.org/wiki/Double-precision_floating-point_format">Wikipedia - Double Precision floating point format</a></p>
</li>
<li>
<p><a href="https://en.wikipedia.org/wiki/Single-precision_floating-point_format">Wikipedia - Single Precision floating point format</a></p>
</li>
</ul>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove this "Double.longBitsToDouble" call.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_10_oct_2014_140737_freddy_mallet_wrote">on 10 Oct 2014, 14:07:37 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>Perfect !</p>
</div>
</div>
</div>
</div>