<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Primitives can be read and written to atomically. Except for <code>long</code> and <code>double</code>, that is. These 64-bit primitives must be marked <code>volatile</code> in multi-threaded environments, or swapped out for their atomic counterparts: <code>AtomicLong</code>, and <code>AtomicDouble</code> to guarantee that their updates are always visible to other threads.</p>
</div>
<div class="paragraph">
<p>Similarly, to ensure that updates to 32-bit primitives are visible to all threads, they should also be marked <code>volatile</code>.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">long m = 0;  // Noncompliant

public void increment() {
  m++;
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">volatile long m = 0;

public void increment() {
  m++;
}</code></pre>
</div>
</div>
<div class="paragraph">
<p>or</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">AtomicLong m = new AtomicLong(0);

public void increment() {
  m.incrementAndGet();
}</code></pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p><a href="https://wiki.sei.cmu.edu/confluence/x/QTdGBQ">CERT, VNA00-J.</a> - Ensure visibility when accessing shared primitive variables</p>
</li>
<li>
<p><a href="https://wiki.sei.cmu.edu/confluence/x/zzdGBQ">CERT, VNA05-J.</a> - Ensure atomicity when reading and writing 64-bit values</p>
</li>
</ul>
</div>
<hr>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Mark this "xxx" "volatile" or use an "AtomicXxx" instead.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p><code>long|double xxx</code></p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_relates_to_s3687">relates to: <a data-rspec-id="S3687" class="rspec-auto-link">S3687</a></h3>

</div>
</div>
</div>