<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>The <a href="https://learn.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.components.supplyparameterfromqueryattribute">SupplyParameterFromQuery</a> attribute can be used to specify that a component parameter, of a routable component, comes from the query string.</p>
</div>
<div class="paragraph">
<p>Component parameters supplied from the query string support the following types:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>bool, DateTime, decimal, double, float, Guid, int, long, string.</p>
</li>
<li>
<p>Nullable variants of the preceding types.</p>
</li>
<li>
<p>Arrays of the preceding types, whether they&#8217;re nullable or not nullable.</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>Query parameters should have one of the supported types. Otherwise, an unhandled exception will be raised at runtime.</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-text" data-lang="text">Unhandled exception rendering component: Querystring values cannot be parsed as type '&lt;type&gt;'.
System.NotSupportedException: Querystring values cannot be parsed as type '&lt;type&gt;'
...</code></pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Change the parameter type to one of the following ones:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>bool, DateTime, decimal, double, float, Guid, int, long, string.</p>
</li>
<li>
<p>Nullable variants of the preceding types.</p>
</li>
<li>
<p>Arrays of the preceding types, whether they&#8217;re nullable or not nullable.</p>
</li>
</ul>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-csharp" data-lang="csharp">@page "/print"
&lt;p&gt; Parameter value is: @Value &lt;/p&gt;
@code {
    [Parameter]
    [SupplyParameterFromQuery()]
    public TimeSpan Value { get; set; }     // Noncompliant
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-csharp" data-lang="csharp">@page "/print"
&lt;p&gt; Parameter value is: @Value &lt;/p&gt;
@code {
    [Parameter]
    [SupplyParameterFromQuery()]
    public long Value { get; set; }         // Compliant
}</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Microsoft Learn - <a href="https://learn.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.components.supplyparameterfromqueryattribute">SupplyParameterFromQueryAttribute</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>