== How to fix it in ASP.NET

=== Code examples

include::../../common/fix/code-rationale.adoc[]

==== Noncompliant code example

[source,vbnet,diff-id=1,diff-type=noncompliant]
----
Imports System.Web
Imports System.Web.Mvc

Public Class ExampleController
    Inherits Controller

    Private Shared ReadOnly _logger As log4net.ILog = log4net.LogManager.GetLogger(System.Reflection.MethodBase.GetCurrentMethod().DeclaringType)
    
    <HttpGet>
    Public Sub Log(data As String)
        If data IsNot Nothing Then
            _logger.Info("Log: " + data) ' Noncompliant
        End If
    End Sub
End Class
----

==== Compliant solution

[source,vbnet,diff-id=1,diff-type=compliant]
----
Imports System.Web
Imports System.Web.Mvc

Public Class ExampleController
    Inherits Controller

    Private Shared ReadOnly _logger As log4net.ILog = log4net.LogManager.GetLogger(System.Reflection.MethodBase.GetCurrentMethod().DeclaringType)

    <HttpGet>
    Public Sub Log(data As String)
        If data IsNot Nothing Then
            data = data.Replace(vbLf, "_"c).Replace(vbCr, "_"c)
            _logger.Info("Log: " + data)
        End If
    End Sub
End Class
----

include::../../common/fix/how-does-this-work.adoc[]
