== How to fix it in ASP.NET

=== Code examples

include::../../common/fix/code-rationale.adoc[]

==== Noncompliant code example

[source,vbnet,diff-id=1,diff-type=noncompliant]
----
Public Class ExampleController
    Inherits Controller

    Private Shared ReadOnly Logger As log4net.ILog = log4net.LogManager.GetLogger(System.Reflection.MethodBase.GetCurrentMethod().DeclaringType)

    <HttpGet>
    Public Sub Log(Data As String)
        If Data IsNot Nothing Then
            Logger.Info("Log: " & Data) ' Noncompliant
        End If
    End Sub
End Class
----

==== Compliant solution

[source,vbnet,diff-id=1,diff-type=compliant]
----
Public Class ExampleController
    Inherits Controller

    Private Shared ReadOnly Logger As log4net.ILog = log4net.LogManager.GetLogger(System.Reflection.MethodBase.GetCurrentMethod().DeclaringType)

    <HttpGet>
    Public Sub Log(Data As String)
        If Data IsNot Nothing Then
            Data = Data.Replace(vbLf, "_"c).Replace(vbCr, "_"c)
            Logger.Info("Log: " & Data)
        End If
    End Sub
End Class
----

include::../../common/fix/how-does-this-work.adoc[]
