<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Generic types, such as <code>list</code> or <code>dict</code> accept type arguments to specify the type of elements contained in the list or the keys and values in the dictionary.</p>
</div>
<div class="paragraph">
<p>If a generic type is used without a type argument, the type arguments will implicitly assumed to be <code>Any</code>. This makes the type hint less informative and makes the contract of the function or variable annotated with the type hint more difficult to understand.</p>
</div>
<div class="paragraph">
<p>Furthermore, incomplete type hints can hinder IDE autocompletion and code insight capabilities of static analyis tools.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Add the missing type parameters to this generic type.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">def print_list(numbers: list) -&gt; None:
    for n in numbers:
        print(n)</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">def print_list(numbers: list[int]) -&gt; None:
    for n in numbers:
        print(n)</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p><a href="https://docs.python.org/3/library/typing.html#generic-concrete-collections">Python documentation on generic collections</a></p>
</li>
<li>
<p><a href="https://mypy.readthedocs.io/en/stable/builtin_types.html#generic-types">Built-in generic types</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>