<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Asserts should only be used in test code. Salesforce provide better ways to handle errors:</p>
</div>
<div class="ulist">
<ul>
<li>
<p><a href="https://developer.salesforce.com/docs/atlas.en-us.apexcode.meta/apexcode/apex_triggers_exceptions.htm">Trigger Exceptions</a></p>
</li>
<li>
<p><a href="https://developer.salesforce.com/blogs/2017/09/error-handling-best-practices-lightning-apex.html">Error Handling Best Practices for Lightning and Apex</a></p>
</li>
</ul>
</div>
<div class="paragraph">
<p>Classes containing test code, including test helper methods, should always have the <code>@isTest</code> annotation. There is a <a href="https://help.salesforce.com/articleView?id=000314162&amp;language=en_US&amp;type=1&amp;mode=1">limit to the amount of code</a> an organization can upload into Salesforce. Test code does not count towards this limit, and test code is recognized via the <code>@isTest</code> annotation.</p>
</div>
<div class="paragraph">
<p>This rule raises an issue when methods <code>System.assert</code>, <code>System.assertEquals</code> or <code>System.assertNotEquals</code> are called in a class which is not annotated with <code>@isTest</code>.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-apex" data-lang="apex">class ApexTests {
    private static void testHelper(Task task) {
        System.assert("expected subject", task.subject, "task has not the expected subject");  // Noncompliant
    }
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-apex" data-lang="apex">@isTest
class ApexTests {
    private static void testHelper(Task task) {
        System.assert("expected subject", task.subject, "task has not the expected subject");
    }
}</code></pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p><a href="https://help.salesforce.com/articleView?id=000314162&amp;language=en_US&amp;type=1&amp;mode=1">Increase Apex code character limit</a></p>
</li>
<li>
<p><a href="https://developer.salesforce.com/docs/atlas.en-us.apexcode.meta/apexcode/apex_testing_utility_classes.htm">Common Test Utility Classes for Test Data Creation</a></p>
</li>
</ul>
</div>
<hr>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Either annotate the class with "@isTest" or remove this assert statement.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p><code>System.assert</code>, <code>System.assertEquals</code> and <code>System.assertNotEquals</code> statements</p>
</div>
</div>
</div>
</div>