== Why is this an issue?

``++StringBuilder++`` is more efficient than string concatenation, especially when the operator is repeated over and over as in loops.

=== Noncompliant code example

[source,vbnet]
----
Module Module1
    Sub Main()
        Dim foo = ""
        foo &= "Result: "       ' Compliant - outside of loop

        For i = 1 To 9
            foo &= i            ' Noncompliant
        Next
    End Sub
End Module
----

=== Compliant solution

[source,vbnet]
----
Module Module1
    Sub Main()
        Dim foo = New System.Text.StringBuilder
        foo.Append("Result: ")  ' Compliant

        For i = 1 To 9
            foo.Append(i)       ' Compliant
        Next
    End Sub
End Module
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
