<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Transmuting between collections of different types can lead to undefined behavior due to differences in ABI, size, or alignment. This can result in serious issues like out-of-bounds memory access.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-rust" data-lang="rust">let vec = vec![2_u16];
unsafe {
    std::mem::transmute::&lt;_, Vec&lt;u32&gt;&gt;(vec); // Noncompliant: Different size, likely out-of-bounds memory access.
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-rust" data-lang="rust">let vec = vec![2_u16];
let vec_u32: Vec&lt;u32&gt; = vec.into_iter().map(u32::from).collect(); // Compliant: Iterates, maps, and collects the values safely.</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Clippy Lints - <a href="https://rust-lang.github.io/rust-clippy/master/index.html#unsound_collection_transmute" class="bare">https://rust-lang.github.io/rust-clippy/master/index.html#unsound_collection_transmute</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>