== Why is this an issue?

include::../why.adoc[]


== How to fix it

include::../how.adoc[]

=== Code examples

==== Noncompliant code example

[source,csharp,diff-id=1,diff-type=noncompliant]
----
int Abs(int n) // Noncompliant: cognitive complexity = 5
{
    if (n >= 0)   // +1
    {
        return n; 
    }
    else          // +2, due to nesting
    { 
        if (n == int.MinValue)  // +1 
        { 
            throw new ArgumentException("The absolute value of int.MinValue is outside of int boundaries");
        }
        else                    // +1
        {
            return -n;
        }
    }             
}
----

==== Compliant solution

[source,csharp,diff-id=1,diff-type=compliant]
----
int Abs(int n) // Compliant: cognitive complexity = 3
{
    if (n == int.MinValue)  // +1 
    { 
        throw new ArgumentException("The absolute value of int.MinValue is outside of int boundaries");
    } 
    else if (n >= 0)        // +1
    {
        return n; 
    }
    else                    // +1
    { 
        return -n;
    }             
}
----

include::../resources.adoc[]

include::../rspecator-dotnet.adoc[]