<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>In Dart instances of a class could be treated as functions. To achieve this, the class should implement a <code>call</code> method. This example illustrates the usage of this language feature:</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-dart" data-lang="dart">class F {
  void call() {}
}
void g(void Function() f) {
  f();
}

g(F()); // Here an instance of F is passed where the Function is expected</code></pre>
</div>
</div>
<div class="paragraph">
<p>While this implicit tear off looks very concise, it has a serious drawback. This implicit conversion might happen accidentally, if you just happen to have a <code>call</code> method and by mistake passed an instance as a <code>Function</code>. Moreover, such code is much harder to read and understand as you don&#8217;t see any difference, between being passed as an object or as a function. Taking this all into account, this behavior could be removed in the next versions of the language. Thus, to avoid confusion and potential compatibility problem, this is highly recommended to make the passing of a <code>call()</code> method explicit.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Pass <code>instance.call</code> instead of <code>instance</code>, where the <code>Function</code> is expected.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-dart" data-lang="dart">class F {
  void call() {}
}
void g(void Function() f) {
  f();
}

g(F());</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-dart" data-lang="dart">class F {
  void call() {}
}
void g(void Function() f) {
  f();
}

g(F().call);</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Dart Docs - <a href="https://dart.dev/tools/linter-rules/implicit_call_tearoffs">Dart Linter rule - implicit_call_tearoffs</a></p>
</li>
<li>
<p>Dart Docs - <a href="https://dart.dev/language/callable-objects">Callable objects</a></p>
</li>
</ul>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="ulist">
<ul>
<li>
<p>Implicit tear-off of the 'call' method.</p>
</li>
</ul>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>Cascade expression</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
</div>
</div>