<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>88-level variables, also known as "condition name" variables, represent possible values of the "conditional variables" they&#8217;re tied to. Because a condition name can be used to test the value of its conditional variable without any other contextual references to the conditional variable being tested, it makes the code easier to understand if the name of the 88-level variable references its conditional variable.</p>
</div>
<div class="paragraph">
<p>This rule raises an issue when the name of an 88-level variable does not start with the first characters of the name of its conditional variable.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="paragraph">
<p>With the default minimum of 3:</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cobol" data-lang="cobol">01 COLOR PIC X.
  88 YELLOW VALUE 'Y'. *&gt; Noncompliant
  88 GREEN VALUE 'G'. *&gt; Noncompliant
  88 RED VALUE 'R'. *&gt; Noncompliant

* ...
IF GREEN  *&gt; What does this mean?
* ...
END-IF</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cobol" data-lang="cobol">01 COLOR PIC X.
  88 COL-YELLOW VALUE 'Y'.
  88 COL-GREEN VALUE 'G'.
  88 COL-RED VALUE 'R'.

* ...
IF COL-GREEN
* ...
END-IF</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Rename variable "yyy" so that the first n chars are "xxx".</p>
</div>
</div>
<div class="sect2">
<h3 id="_parameters">Parameters</h3>
<div class="sidebarblock">
<div class="content">
<div class="title">minimumChars</div>
<div class="listingblock">
<div class="content">
<pre>3</pre>
</div>
</div>
<div class="paragraph">
<p>The number of characters from the beginning of the conditional variable name that the name of the condition name should start with.</p>
</div>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="ulist">
<ul>
<li>
<p>primary: name in condition name declaration</p>
</li>
<li>
<p>secondary: name in conditional variable declaration</p>
</li>
</ul>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_21_dec_2016_153250_robert_shawver_wrote">on 21 Dec 2016, 15:32:50 Robert Shawver wrote:</h3>
<div class="paragraph">
<p>shouldn&#8217;t the last line of the description read "This rule raises an issue when the <strong>condition name</strong> does not start with the first characters of the name of its conditional variable."? also, will this error be raised if the conditional variable is FILLER, which it can legitimately be?</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_3_jan_2017_161836_ann_campbell_wrote">on 3 Jan 2017, 16:18:36 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~rshawver] I&#8217;ve corrected the last sentence. For the <code>FILLER</code> question, please open a new thread on the <a href="https://groups.google.com/forum/?pli=1#!forum/sonarqube">Google Group</a> because I&#8217;ve got questions, but this is not the right venue for an extended conversation.</p>
</div>
<div class="paragraph">
<p>And thanks for the correction!</p>
</div>
</div>
</div>
</div>