This rule raises an issue when multiple controller files define classes with identical names in different directories within a Rails application.

== Why is this an issue?

Rails uses an autoloading mechanism that maps file paths to class names. When multiple controller files define classes with the same name in different directories, Rails cannot determine which file to load when the class is referenced.

This creates several problems:

* *Circular dependency errors*: Rails may attempt to load both files simultaneously, creating a circular reference
* *Unpredictable behavior*: The application may load different controller classes depending on the order of file loading
* *Runtime failures*: The application may crash with autoloading errors, especially in production environments

Rails expects each class to have a unique fully-qualified name. When controllers are organized in subdirectories, they should use proper namespacing to reflect their directory structure. This ensures that each controller has a distinct identity and can be loaded reliably.

=== What is the potential impact?

The application may experience runtime errors, unpredictable behavior, or complete failure to start. In production environments, this can lead to service outages and difficult-to-debug issues.

== How to fix it

Use proper namespacing for controllers in subdirectories. The namespace should match the directory structure to create unique class names.

=== Code examples

==== Noncompliant code example

[source,ruby,diff-id=1,diff-type=noncompliant]
----
# app/controllers/logs_controller.rb
class LogsController < ApplicationController
  def index
    # Controller logic here
  end
end

# app/controllers/api/logs_controller.rb
class LogsController < ApplicationController # Noncompliant
  def index
    # This creates a circular dependency issue
  end
end
----

==== Compliant solution

[source,ruby,diff-id=1,diff-type=compliant]
----
# app/controllers/logs_controller.rb
class LogsController < ApplicationController
  def index
    # Controller logic here
  end
end

# app/controllers/api/logs_controller.rb
class Api::LogsController < ApplicationController
  def index
    # API-specific controller logic here
  end
end
----

== Resources

=== Documentation

 * Rails Autoloading and Reloading Constants - https://guides.rubyonrails.org/autoloading_and_reloading_constants.html[Official Rails guide explaining how autoloading works and naming conventions]

 * Rails Routing from the Outside In - https://guides.rubyonrails.org/routing.html#controller-namespaces-and-routing[Rails guide covering controller namespaces and routing]
