<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Triple slash (<code>///</code>) should be used for documentation comments, instead of multi-line documentation comments (<code>/** ... */</code>).</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Dart supports two types of documentation comments:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>single-line (<code>/// &#8230;&#8203;</code>)</p>
</li>
<li>
<p>and multi-line (<code>/** ... */</code>)</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>Single-line comments have been chosen as the standard for documentation comments in Dart, since they are generally easier to vertically align and clearer to read.</p>
</div>
<div class="paragraph">
<p>They are generally easier to align because following lines in multi-line comments are typically indented by the <code>*</code> character, to make it clear that they are part of the same comment block.</p>
</div>
<div class="paragraph">
<p>However, this makes each of the following line of the multi-line comment differently indented than the first line, resulting in a higher future maintenance cost to keep it correctly aligned.</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-dart" data-lang="dart">/** First comment line
 *  Following comment line, 1 char-indentation
 *  ...
 */</code></pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Replace each line of the documentation comment with a single-line comment starting with <code>\\\</code>. Remove the <code>*</code> character and related spacing that you may have at the beginning of each of the following line.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-dart" data-lang="dart">/** First comment line
 *  Following comment line, 1 char-indentation
 */</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-dart" data-lang="dart">/// First comment line
/// Following comment line, 1 char-indentation</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p>Dart Docs - <a href="https://dart.dev/tools/linter-rules/slash_for_doc_comments">Dart Linter rule - slash_for_doc_comments</a></p>
</li>
<li>
<p>Dart Docs - <a href="https://dart.dev/language/comments#documentation-comments">Language - Documentation comments</a></p>
</li>
</ul>
</div>
<hr>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Use the end-of-line form ('///') for doc comments.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>The entire comment, including the <code>/**</code> and <code>*/</code> delimiters.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
</div>
</div>