== Noncompliant Code Example

----
class Base {
  virtual f();
}

class Derived : public Base {
  virtual f(int); // Noncompliant; hides "Base::f"
}
----



ifdef::env-github,rspecator-view[]
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
