== Why is this an issue?

=== Noncompliant code example

[source,cpp]
----
class Base {
  virtual f();
}

class Derived : public Base {
  virtual f(int); // Noncompliant; hides "Base::f"
}
----



ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Highlighting

Entire method signature


'''
== Comments And Links
(visible only on this page)

=== on 6 Sep 2016, 09:07:12 Ann Campbell wrote:
\[~evgeny.mandrikov] there is no OOP03-CPP. ...?

=== on 6 Sep 2016, 09:28:36 Evgeny Mandrikov wrote:
\[~ann.campbell.2] \https://www.securecoding.cert.org/confluence/display/cplusplus/OOP03-CPP.+Prefer+not+to+overload+virtual+functions

endif::env-github,rspecator-view[]
