== Why is this an issue?

Shared naming conventions allow teams to collaborate effectively. This rule raises an issue when the a pom's ``++groupId++`` does not match the provided regular expression.


=== Noncompliant code example

With the default regular expression: ``++(com|org)(\.[a-z][a-z-0-9]*)+++``

[source,xml]
----
<project ...>
  <groupId>myCo</groupId>  <!-- Noncompliant -->

  <!-- ... -->
</project>
----


=== Compliant solution

[source,xml]
----
<project ...>
  <groupId>com.myco</groupId>

  <!-- ... -->
</project>
----



ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Update this "groupId" to match the provided regular expression: xxx.


=== Parameters

.regex
****

----
(com|org)(\.[a-z][a-z-0-9]*)+
----

The regular expression the "groupId" should match
****


=== Highlighting

groupId value


'''
== Comments And Links
(visible only on this page)

=== on 30 Nov 2015, 14:06:29 Michael Gumowski wrote:
LGTM [~ann.campbell.2]!

Note that I'm not sure about your regex. You can have more that one group, separated by dots, after the first ``++com++``/``++org++``.

=== on 30 Nov 2015, 16:00:30 Ann Campbell wrote:
Uhm... yeah [~michael.gumowski]. That's what I intended.

endif::env-github,rspecator-view[]
