== Why is this an issue?

Shared naming conventions allow teams to collaborate effectively. This rule raises an issue when the a pom's ``++groupId++`` does not match the provided regular expression.


=== Noncompliant code example

With the default regular expression: ``++(com|org)(\.[a-z][a-z-0-9]*)+++``

[source,xml]
----
<project ...>
  <groupId>myCo</groupId>  <!-- Noncompliant -->

  <!-- ... -->
</project>
----


=== Compliant solution

[source,xml]
----
<project ...>
  <groupId>com.myco</groupId>

  <!-- ... -->
</project>
----



ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

include::parameters.adoc[]

include::highlighting.adoc[]

'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
