== Why is this an issue?

Chained assignments are confusing and hard to read, and should be avoided.


=== Noncompliant code example

[source,text]
----
x = y = 0; // Noncompliant
----


=== Compliant solution

[source,text]
----
x = 0;
y = 0; // or y = x;
----

