Chained assignments are confusing and hard to read, and should be avoided.


== Noncompliant Code Example

[source,text]
----
x = y = 0; // Noncompliant
----


== Compliant Solution

----
x = 0;
y = 0; // or y = x;
----

