== How to fix it in Faraday

=== Code examples

include::../../common/fix/code-rationale.adoc[]

Hostname validation is disabled if ``++verify_hostname++`` is set to `false`, the default value is `true`.

==== Noncompliant code example

[source,ruby,diff-id=11,diff-type=noncompliant]
----
ssl_options = {
    verify_hostname: false, # Noncompliant
}
conn = Faraday.new(url: "https://example.com", ssl: ssl_options)
response = conn.get("/")
----

==== Compliant solution

[source,ruby,diff-id=11,diff-type=compliant]
----
conn = Faraday.new(url: "https://example.com")
response = conn.get("/")
----

=== How does this work?

include::../../common/fix/validation.adoc[]
