== How to fix it in Ruby Standard Library

=== Code examples

include::../../common/fix/code-rationale.adoc[]

Hostname validation is disabled if ``++verify_hostname++`` is set to `false`, the default value is `true`.

==== Noncompliant code example

[source,ruby,diff-id=1,diff-type=noncompliant]
----
http = Net::HTTP.new("https://example.com")
http.use_ssl = true
http.verify_hostname = false # Noncompliant
response = http.get("/")
----

==== Compliant solution

[source,ruby,diff-id=1,diff-type=compliant]
----
http = Net::HTTP.new("https://example.com")
http.use_ssl = true
response = http.get("/")
----

=== How does this work?

include::../../common/fix/validation.adoc[]
