In-stream procedures should always have a name.

== Why is this an issue?

In JCL, it is expected for a PROC statement inside of a job stream to have a name.

== How to fix it

=== Code examples

==== Noncompliant code example

[source,jcl,diff-id=1,diff-type=noncompliant]
----
//MYJOB   JOB DXXX
//        PROC
//STEP1   EXEC PGM=MYPGM
//        PEND
----

==== Compliant solution

[source,jcl,diff-id=1,diff-type=compliant]
----
//MYJOB   JOB DXXX
//TRPOC   PROC
//STEP1   EXEC PGM=MYPGM
//        PEND
----

== Resources

=== Documentation

* https://www.ibm.com/docs/en/zos/3.1.0?topic=d-name-field-8[IBM reference - PROC statement - Name field]
