== Why is this an issue?

In JCL, some statements require to have a label or name defined, otherwise it will trigger a JCL error.
The statements that require a name/label are:
* CNTL
* JOB
* NOTIFY
* OUTPUT
* in-stream PROC

== How to fix it

=== Code examples

==== Noncompliant code example

[source,jcl,diff-id=1,diff-type=noncompliant]
----
//MYJOB   JOB DXXX
//        PROC
//STEP1   EXEC PGM=MYPGM
//        PEND
----

==== Compliant solution

[source,jcl,diff-id=1,diff-type=compliant]
----
//MYJOB   JOB DXXX
//TRPOC   PROC
//STEP1   EXEC PGM=MYPGM
//        PEND
----

== Resources

=== Documentation

* https://www.ibm.com/docs/en/zos/3.1.0?topic=description-label-field[IBM Reference - CNTL Label field]
* https://www.ibm.com/docs/en/zos/3.1.0?topic=d-name-field-5[IBM Reference - JOB Name field]
* https://www.ibm.com/docs/en/zos/3.1.0?topic=statement-label-field[IBM Reference - NOTIFY Label field]
* https://www.ibm.com/docs/en/zos/3.1.0?topic=d-name-field-6[IBM Reference - OUTPUT Name field]
* https://www.ibm.com/docs/en/zos/3.1.0?topic=d-name-field-8[IBM reference - PROC statement - Name field]
