A PROC in a job stream should always have a name, also unnamed DD statement should be preceded by a named DD statement.

== Why is this an issue?

In JCL, it is expected for a PROC statement inside of a job stream to have a name.
Unnamed DD statement should be preceded by another named DD statement.

== How to fix it

=== Code examples

==== Noncompliant code example

[source,jcl,diff-id=1,diff-type=noncompliant]
----
//MYJOB   JOB DXXX
//        PROC
//STEP1   EXEC PGM=MYPGM
//        PEND
----

[source,jcl,diff-id=2,diff-type=noncompliant]
----
//CATALOG EXEC PROC=MY_PROC
//        DD DSN=NAME
----

==== Compliant solution

[source,jcl,diff-id=1,diff-type=compliant]
----
//MYJOB   JOB DXXX
//TRPOC   PROC
//STEP1   EXEC PGM=MYPGM
//        PEND
----

[source,jcl,diff-id=2,diff-type=compliant]
----
//CATALOG EXEC PROC=MY_PROC
//SYSIN   DD DSN=VAL
//        DD DSN=NAME
----

== Resources

=== Documentation

* https://www.ibm.com/docs/en/zos/3.1.0?topic=d-name-field-8[IBM reference - PROC statement - Name field]
