== Why is this an issue?

Loggers should be:

* ``++private++``: not accessible outside of their parent classes. If another class needs to log something, it should instantiate its own logger.
* ``++static++``: not dependent on an instance of a class (an object). When logging something, contextual information can of course be provided in the messages but the logger should be created at class level to prevent creating a logger along with each object.
* ``++const++``: created once and only once per class.

=== Noncompliant code example

With the default regular expression ``++LOG(?:GER)?++``:

[source,flex]
----
public const logger:ILogger = LogUtil.getLogger(MyClass);
----

=== Compliant solution

[source,flex]
----
private static const LOG:ILogger = LogUtil.getLogger(MyClass);
----

include::../rspecator.adoc[]