`MutableStateFlow` and `MutableSharedFlow` are very convenient for storing and adding updates of some data structures in event-driven paradigm. This is widely used in Android Views for handling updates. While it's extremely useful to manage such objects inside some class, it's not recommended to expose them outside of the class. 

If other class will have a write access to the variales of `MutableStateFlow` or `MutableSharedFlow`, the data can be accidentally updated with wrong results. So it is generally recommended to make only one class responsible for the updates of these structures, otherwise it might bring incosistency, problems with maintainability and be error-prone.

To restrict the read/write access `StateFlow` or `SharedFlow` should be used together with private `MutableStateFlow` or `MutableSharedFlow` fields.

This rule raises an issue when there is a non-private variable of MutableStateFlow or MutableSharedFlow type.

== Noncompliant Code Example

----
class MyView : ViewModel() {

    val state = MutableStateFlow(State.New)
   
}
----

== Compliant Solution

----
class MyView : ViewModel() {

    private val _state = MutableStateFlow(State.New)
    val state: StateFlow<LatestNewsUiState> = _uiState
   
}
----

== See

* https://developer.android.com/kotlin/coroutines/coroutines-best-practices#mutable-types[Android Coroutines Best Practices]
