`MutableStateFlow` and `MutableSharedFlow` are very convenient for storing and adding the events data. While it's extremely useful to manage such objects inside some class, it's not recommended to expose the visibility for the outside. To make read access secure `StateFlow` or `SharedFlow` should be used together with private `MutableStateFlow` or `MutableSharedFlow` fields.

The issue should be reported if there is a non-private variable of `MutableStateFlow` or `MutableSharedFlow`type.

== Noncompliant Code Example

----
class MyView : ViewModel() {

    val state = MutableStateFlow(State.New)
   
}
----

== Compliant Solution

----
class MyView : ViewModel() {

    private val _state = MutableStateFlow(State.New)
    val state: StateFlow<LatestNewsUiState> = _uiState
   
}
----

== See

* https://developer.android.com/kotlin/coroutines/coroutines-best-practices#mutable-types[Android Coroutines Best Practices]
