== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,vb6]
----
Rem FIXME: Add documentation - Noncompliant
Function Divide(ByVal Numerator As Integer, ByVal Denominator As Integer) As Integer
    Divide = Numerator / Denominator ' FIXME: Denominator value might be 0 - Noncompliant
End Function
----

include::../see.adoc[]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
