<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>By implementing class interfaces with member functions the implementation retains more control over how the object state can be modified and helps to allow a class to be maintained without affecting clients. Returning a handle to class-data allows for clients to modify the state of the object without using any interfaces.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">class C
{
public:
  int32_t &amp; getA () // Noncompliant, non-const handle to class-data returned.
  {
    return a;
  }
private:
  int32_t a;
};

void b ( C &amp; c )
{
  int32_t &amp; a_ref = c.getA ();
  a_ref = 10; // External modification of private C::a
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">class C
{
public:
  int32_t const &amp; getA () // Compliant
  {
    return a;
  }
  void setA(int32_t a)
  {
    this.a = a;
  }
private:
  int32_t a;
};

void b ( C &amp; c )
{
  int32_t const &amp; a_ref = c.getA ();
  a_ref = 10; // Not allowed, use setter setA instead
}</code></pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p>MISRA C&#43;&#43;:2008, 9-3-2</p>
</li>
</ul>
</div>
<hr>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Mark the return type with the "const" qualifier.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_15_oct_2014_205148_ann_campbell_wrote">on 15 Oct 2014, 20:51:48 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~samuel.mercier] please:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>fill in the appropriate reference field(s).</p>
</li>
<li>
<p>provide a See section.</p>
</li>
</ul>
</div>
</div>
<div class="sect2">
<h3 id="_on_17_oct_2014_153233_ann_campbell_wrote">on 17 Oct 2014, 15:32:33 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>FYI [~samuel.mercier] I&#8217;ve moved this to Data-related reliability.</p>
</div>
</div>
</div>
</div>