This rule raises an issue when backticks (\`) are used for command substitution instead of the modern `$()` syntax.

== Why is this an issue?

Backticks (\`) are an older way to perform command substitution in shell scripts. While they still work, they have several drawbacks compared to the modern `$()` syntax.

The main problems with backticks are:

* **Poor readability**: Backticks can be hard to distinguish from regular quotes, especially in small fonts or when printed
* **Nesting difficulties**: You cannot easily nest backticks without complex escaping
* **Editor limitations**: Many text editors don't highlight backtick syntax as well as `$()`

The `$()` syntax was introduced as part of POSIX standards to address these issues. It provides the same functionality but with better visual clarity and more flexible usage patterns.

=== What is the potential impact?

Using backticks instead of `$()` reduces code readability and maintainability. Complex command substitutions become harder to understand and modify, especially when nesting is required. This can lead to bugs during script maintenance and makes code reviews more difficult.

== How to fix it

Replace backticks with the modern `$()` syntax for command substitution. This improves readability and allows for easier nesting of commands.

=== Code examples

==== Noncompliant code example

[source,shell,diff-id=1,diff-type=noncompliant]
----
result=`date`                               # Noncompliant
files=`find /path -name "*.txt" | head -5`  # Noncompliant
count=`echo \`ls *.txt | wc -l\``           # Noncompliant
----

==== Compliant code example

[source,shell,diff-id=1,diff-type=compliant]
----
result=$(date)
files=$(find /path -name "*.txt" | head -5)
count=$(echo $(ls *.txt | wc -l))
----

== Resources
=== Documentation

* Bash Manual - Command Substitution - https://www.gnu.org/software/bash/manual/html_node/Command-Substitution.html[Official Bash documentation explaining command substitution syntax]
* POSIX Shell Command Language - https://pubs.opengroup.org/onlinepubs/9699919799/utilities/V3_chap02.html#tag_18_06_03[POSIX standard specification for command substitution]

=== Standards

* POSIX.1-2017 Shell Command Language - https://pubs.opengroup.org/onlinepubs/9699919799/utilities/V3_chap02.html[POSIX standard that defines modern shell syntax including $() command substitution]

