<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Global variables provide a convenient way of accessing the same value or set of values from multiple places in the code without the need to pass them through the interface and possibly the entire call hierarchy. As a consequence, they can be used to avoid repetition and guarantee that the same value is used in multiple places.</p>
</div>
<div class="paragraph">
<p>However, modification of such a global variable may potentially change the behavior of any program component which makes it difficult to reason locally about the program&#8217;s correctness. Additionally, the modification of the same global variable in a multi-threaded program may lead to data races.</p>
</div>
<div class="paragraph">
<p>This rule raises an issue on the modification of publicly accessible non-atomic global variables.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">std::size_t operationCounter = 0;

int add(int x, int y) {
  ++operationCounter; // Noncompliant
  return x + y;
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">class StatsCollector
{
public:
  static void registerOp() {
      ++operationCounter;
  }

private:
  static std::atomic&lt;std::size_t&gt; operationCounter{0};
};

int add(int x, int y) {
  StatsCollector::registerOp();
  return x + y;
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_exceptions">Exceptions</h3>
<div class="paragraph">
<p>An issue will not be raised on modifications of a private static member of a class, as the impact of such modification is limited to its members and friends. This makes additional states visible from the class declaration and makes it possible to protect variables from multi-threaded access if necessary.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p><a data-rspec-id="S5421" class="rspec-auto-link">S5421</a> - marking global variables <code>const</code> to prevent modification</p>
</li>
</ul>
</div>
<hr>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_relates_to_s5421">relates to: <a data-rspec-id="S5421" class="rspec-auto-link">S5421</a></h3>

</div>
</div>
</div>