<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when a goroutine is manually created with <code>time.Sleep()</code> to implement timeouts, instead of using the idiomatic <code>time.After()</code> function.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Go provides the <code>time.After()</code> function as the standard, idiomatic way to create timeout channels. This function returns a channel that receives a value after a specified duration.</p>
</div>
<div class="paragraph">
<p>Manually creating timeout goroutines with <code>time.Sleep()</code> has several drawbacks:</p>
</div>
<div class="ulist">
<ul>
<li>
<p><strong>More verbose</strong>: Requires more boilerplate code to achieve the same result</p>
</li>
<li>
<p><strong>Resource management</strong>: You need to ensure proper cleanup and handle edge cases</p>
</li>
<li>
<p><strong>Less readable</strong>: The intent is less clear compared to the explicit <code>time.After()</code> call</p>
</li>
<li>
<p><strong>Error-prone</strong>: Manual implementations can introduce subtle bugs in goroutine lifecycle management</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>The <code>time.After()</code> function handles all the complexity internally and provides a clean, expressive API that clearly communicates the timeout intent. It automatically manages the underlying goroutine and ensures proper resource cleanup.</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="paragraph">
<p>Using manual timeout goroutines instead of <code>time.After()</code> leads to more complex, harder-to-maintain code. While not a security issue, it can make the codebase less readable and more prone to bugs in timeout handling logic.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Replace the manual timeout goroutine with <code>time.After()</code>. This eliminates the need for explicit goroutine creation and channel management.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-go" data-lang="go">timeout := make(chan bool, 1)
go func() {
    time.Sleep(1 * time.Second)
    timeout &lt;- true
}()

select {
case &lt;-ch:
    // handle data
case &lt;-timeout: // Noncompliant
    // handle timeout
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-go" data-lang="go">select {
case &lt;-ch:
    // handle data
case &lt;-time.After(1 * time.Second):
    // handle timeout
}</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Go time.After documentation - <a href="https://pkg.go.dev/time#After">Official documentation for the time.After function</a></p>
</li>
<li>
<p>Go Concurrency Patterns: Timing out - <a href="https://go.dev/blog/go-concurrency-patterns-timing-out-and">Official Go blog post explaining timeout patterns and recommending time.After</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>