include::../description.adoc[]

== Noncompliant Code Example

[source,javascript]
----
function doSomething(a, b) { // "a" is unused
  return compute(b);
}
----

== Compliant Solution

[source,javascript]
----
function doSomething(b) {
  return compute(b);
}
----


== Exceptions

When ``++arguments++`` is used in the function body, no parameter is reported as unused.


[source,javascript]
----
function doSomething(a, b, c) {
  compute(arguments);
}
----


Also, parameters whose name starts with an underscore are ignored to match the behaviour of TypeScript compiler, that is, such parameters are considered unused.


[source,javascript]
----
function doSomething(_a, b) {
  return compute(b);
}
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
