== Why is this an issue?

In 1978, Brian Kernighan and Dennis Ritchie published the first edition of The C Programming Language. This book, known to C programmers as "K&R", served for many years as an informal specification of the language. The version of C that it describes is commonly referred to as K&R C. 


The K&R function definition syntax introduced in the book was later deprecated in the ANSI C and ISO C standards. Even though the K&R syntax is still supported in the ISO C11 standard, it's not in ISO {cpp} standard versions and is not considered readable by most C/{cpp} developers today. 


=== Noncompliant code example

[source,cpp]
----
int foo(a, b)   // Noncompliant K&R C syntax
  int a;
  char* b;
{
}
----


=== Compliant solution

[source,cpp]
----
int foo(int a, char* b) { // Compliant
}
----



ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
