<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>If an attempt is made to use an identifier in a preprocessor directive, and that identifier has not been defined, the preprocessor will assume the value zero. <code>#ifdef</code>, <code>#ifndef</code> and <code>defined()</code> are provided to test the existence of a macro, and are therefore excluded.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">#if x &lt; 0 // Noncompliant, x assumed to be zero as it is not defined
// ...
#endif</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">#ifdef x
#if x &lt; 0 // Compliant, x is guaranteed to be defined.
// ...
#endif
#endif

// ...

#define Y 12

// ...

#if Y &lt; 12 // Compliant, y is defined.
#endif</code></pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p>MISRA C&#43;&#43;:2008, 16-0-7</p>
</li>
</ul>
</div>
<hr>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>"xxx" must be defined prior expansion.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_relates_to_s966">relates to: <a data-rspec-id="S966" class="rspec-auto-link">S966</a></h3>

</div>
<div class="sect2">
<h3 id="_on_21_oct_2014_191829_ann_campbell_wrote">on 21 Oct 2014, 19:18:29 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~samuel.mercier] please incorporate the text under the Compliant heading either into the description or the code sample in the form of comments.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_21_oct_2014_192147_ann_campbell_wrote">on 21 Oct 2014, 19:21:47 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~samuel.mercier] [~massimo.paladin] [~evgeny.mandrikov] this rule says it&#8217;s okay to define preprocessing identifiers either in code or compile-time, but in code is preferred. Would a rule like this someday be feasible and/or desirable?:</p>
</div>
<div class="paragraph">
<p>Title: Preprocessing identifiers should be defined in the code</p>
</div>
<div class="paragraph">
<p>Message: "xxx" was (defined|overridden) at compile time.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_24_oct_2014_073428_samuel_mercier_wrote">on 24 Oct 2014, 07:34:28 Samuel Mercier wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] IMO the two ways serve different purposes.</p>
</div>
<div class="ulist">
<ul>
<li>
<p>You define a macro directly in code when you want to unconditionally enable or disable some features found in a library. A typical example is the LEAN_AND_MEAN macro on Windows.</p>
</li>
<li>
<p>You define a macro on the command line to conditionally change the behaviour of the compilation (e.g. target 32bits vs 64bits, or windows vs unix-like).
Detecting if a macro was defined in code or on the command line is feasible. Understanding why it was done this way is not.</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>We could however write a rule to check if a macro defined on the command line is redefined in code, because in this case the former definition is lost.</p>
</div>
</div>
</div>
</div>