If an attempt is made to use an identifier in a preprocessor directive, and that identifier has not been defined, the preprocessor will assume the value zero. `+#ifdef+`, `+#ifndef+` and `+defined()+` are provided to test the existence of a macro, and are therefore excluded.


== Noncompliant Code Example

----
#if x < 0 // Noncompliant, x assumed to be zero as it is not defined
// ...
#endif
----


== Compliant Solution

----
#ifdef x
#if x < 0 // Compliant, x is guaranteed to be defined.
// ...
#endif
#endif

// ...

#define Y 12

// ...

#if Y < 12 // Compliant, y is defined.
#endif
----


== See

* MISRA C++:2008, 16-0-7

