This rule raises an issue when an empty list is provided to PySpark `DataFrame.dropDuplicates`.

== Why is this an issue?

A list of columns can be provided to the `subset` argument of PySpark's `DataFrame.dropDuplicates` method.
Providing such a parameter allows for removal of duplicates by comparing rows only using the columns present in the argument.
It is also possible to use all columns of the `DataFrame` by passing `None` to the `subset` argument or leaving it empty (as `None` is the default value).
However when an empty list is provided to the `subset` argument, `dropDuplicates` does not perform any deduplication but instead removes all row except one, 
which can lead to unexpected results and potentially incorrect data analysis. 
This rule ensures that `DataFrame.dropDuplicates` is used correctly by specifying at least one column, or not specifying a column at all.

This rule will raise issues as well on `DataFrame.drop_duplicates` and `DataFrame.dropDuplicatesWithinWatermark`.

== How to fix it

To fix this issue remove the empty list or provide at least one column to the subset argument.

=== Code examples

==== Noncompliant code example

[source,python,diff-id=1,diff-type=noncompliant]
----
from pyspark.sql import SparkSession

spark = SparkSession.builder.master("local").appName("Example").getOrCreate()

data = [("Alice", 1), ("Bob", 2), ("Alice", 1)]
df = spark.createDataFrame(data, ["name", "id"])

df.dropDuplicates([]) # Noncompliant: an empty list is provided as parameter
----

==== Compliant solution

[source,python,diff-id=1,diff-type=compliant]
----
from pyspark.sql import SparkSession

spark = SparkSession.builder.master("local").appName("Example").getOrCreate()

data = [("Alice", 1), ("Bob", 2), ("Alice", 1)]
df = spark.createDataFrame(data, ["name", "id"])

df.dropDuplicates(["name"]) # Compliant
----

== Resources
=== Documentation

* PySpark Documentation - https://spark.apache.org/docs/latest/api/python/reference/pyspark.sql/api/pyspark.sql.DataFrame.dropDuplicates.html[DataFrame.dropDuplicates] 
* PySpark Documentation - https://spark.apache.org/docs/latest/api/python/reference/pyspark.sql/api/pyspark.sql.DataFrame.dropDuplicatesWithinWatermark.html[DataFrame.dropDuplicates] 

ifdef::env-github,rspecator-view[]
=== Implementation Specification

=== Message

Provide an adequate parameter.

=== Highlighting

The main location is the method `dropDuplicates` and the secondary location is the empty list literal.

=== Quickfix

We can simply remove the empty list literal if it is directly provided to the method or change it to `None` (this could be done also when a variable is provided as argument.
Quick fix message: `Remove this empty list.`, or `Replace this empty list with None.`

endif::env-github,rspecator-view[]
