<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p><code>NotImplemented</code> is a constant which is intended to be used only by comparison methods such as <code>__lt__</code>. Use it instead of  <code>NotImplementedError</code>, which is an exception, and callers will have a hard time using your code.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">class MyClass:
    def do_something(self):
        raise NotImplemented("Haven't gotten this far yet.")  #Noncompliant</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">class MyClass:
    def do_something(self):
        raise NotImplementedError("Haven't gotten this far yet.")</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_28_apr_2015_100400_ann_campbell_wrote">on 28 Apr 2015, 10:04:00 Ann Campbell wrote:</h3>
<div class="paragraph">
<p><a href="http://mouadino.appspot.com/notimpelementederror-vs-notimplemented-in-python/" class="bare">http://mouadino.appspot.com/notimpelementederror-vs-notimplemented-in-python/</a></p>
</div>
</div>
<div class="sect2">
<h3 id="_on_31_jan_2020_154843_nicolas_harraudeau_wrote">on 31 Jan 2020, 15:48:43 Nicolas Harraudeau wrote:</h3>
<div class="paragraph">
<p>Closing in favor of <a data-rspec-id="S5632" class="rspec-auto-link">RSPEC-5632</a>.</p>
</div>
<div class="paragraph">
<p><code>NotImplemented</code> does not derive from BaseException so it will fail as any other non-exception.</p>
</div>
</div>
</div>
</div>