=== Code examples

==== Noncompliant code example

[source,vbnet,diff-id=1,diff-type=noncompliant]
----
Function IsAMatch(inputString As String) As Boolean
    Dim regex As New Regex("^[a-zA-Z]$")
    Return regex.IsMatch(inputString)
End Function
----

==== Compliant solution

[source,vbnet,diff-id=1,diff-type=compliant]
----
Private ReadOnly Ipv6Regex As New Regex("^[a-zA-Z]$")

Function IsAMatch(inputString As String) As Boolean
    Return Ipv6Regex.IsMatch(inputString)
End Function
----

