=== Code examples

==== Noncompliant code example

[source,vbnet,diff-id=1,diff-type=noncompliant]
----
Function Matches(inputString As String) As Boolean
    Dim myRegex As New Regex("^[a-zA-Z]$")
    Return myRegex.IsMatch(inputString)
End Function
----

==== Compliant solution

[source,vbnet,diff-id=1,diff-type=compliant]
----
Private ReadOnly MyRegex As New Regex("^[a-zA-Z]$")

Function Matches(inputString As String) As Boolean
    Return MyRegex.IsMatch(inputString)
End Function
----

