== Why is this an issue?

In Blazor, the https://learn.microsoft.com/en-us/aspnet/core/blazor/fundamentals/routing#catch-all-route-parameters[catch-all route parameters] allows to capture the rest of the URL path.

The catch-all route parameters should be at the end of the URL. Otherwise, the route template will be considered invalid and a runtime error will be raised:

[source,text]
----
Invalid template 'route/{*catchAllRoute}/another/segment'. A catch-all parameter can only appear as the last segment of the route template.
...
----


== How to fix it

Ensure the catch-all route parameter is at the end of the URL.

=== Code examples

==== Noncompliant code example

[source,csharp,diff-id=1,diff-type=noncompliant]
----
@page "/route/{*catchAllRoute}/another/segment" // Noncompliant

@code {
    [Parameter]
    public string CatchAllRoute { get; set; }
}
----

==== Compliant solution

[source,csharp,diff-id=1,diff-type=compliant]
----
@page "/route/another/segment/{*catchAllRoute}" // Compliant

@code {
    [Parameter]
    public string CatchAllRoute { get; set; }
}
----

== Resources

=== Documentation

* Microsoft Learn - https://learn.microsoft.com/en-us/aspnet/core/blazor/fundamentals/routing#catch-all-route-parameters[ASP.NET Core Blazor routing and navigation]
