<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>A <code>std::optional&lt;T&gt;</code> can contain a value of type <code>T</code>, or can be empty. So in order to avoid crashes or misbehaviors of code, it is better to check the presence of a value in an optional before accessing it. This can be done:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>By using the <code>has_value</code> member function</p>
</li>
<li>
<p>By converting the optional to a boolean</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>Alternatively, the member function <code>value_or</code> can be used to return the contained value or a specific value if the optional is empty.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">using namespace std;

auto create(bool b) {
    return b ? optional&lt;string&gt;{"myString"} : nullopt;
}

void f(bool b) {
    auto str = create(b);
    cout &lt;&lt; *str &lt;&lt; endl; // Noncompliant, optional could be empty
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">using namespace std;

auto create(bool b) {
    return b ? optional&lt;string&gt;{"myString"} : nullopt;
}

void f1(bool b) {
    auto str = create(b);
    if (str.hasValue()) {
        cout &lt;&lt; *str &lt;&lt; endl;
    }
}
void f2(bool b) {
    if (auto str = create(b)) {
        cout &lt;&lt; *str &lt;&lt; endl;
    }
}
void f3(bool b) {
    auto str = create(b);
    cout &lt;&lt; str.value_or("&lt;empty&gt;") &lt;&lt; endl;
}</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Check this optional contains a value before accessing it.</p>
</div>
</div>
</div>
</div>