== Why is this an issue?

Shared coding conventions allow teams to collaborate efficiently. This rule checks that all enumeration member names match a provided regular expression.

=== Noncompliant code example

With default provided regular expression: ``++^[a-z][a-zA-Z0-9]*$++``

[source,swift]
----
enum SomeEnumeration {
    case SomeMember  // Non-Compliant
}
----

=== Compliant solution

[source,swift]
----
enum SomeEnumeration {
    case someMember
}
----
ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::parameters.adoc[]

endif::env-github,rspecator-view[]
