Shared coding conventions allow teams to collaborate efficiently. This rule checks that all enumeration member names match a provided regular expression.

== Noncompliant Code Example

With default provided regular expression: ``++^[a-z][a-zA-Z0-9]*$++``

----
enum SomeEnumeration {
    case SomeMember  // Non-Compliant
}
----

== Compliant Solution

----
enum SomeEnumeration {
    case someMember
}
----
ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::parameters.adoc[]

endif::env-github,rspecator-view[]
