<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when an equality check is made against <code>numpy.nan</code>.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>The <code>numpy.nan</code> is a floating point representation of Not a Number (NaN) used as a placeholder for undefined or missing values in numerical computations.</p>
</div>
<div class="paragraph">
<p>Equality checks of variables against <code>numpy.nan</code> in NumPy will always be <code>False</code> due to the special nature of <code>numpy.nan</code>. This can lead to unexpected and incorrect results.</p>
</div>
<div class="paragraph">
<p>Instead of standard comparison the <code>numpy.isnan()</code> function should be used.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">import numpy as np

x = np.nan

if x == np.nan: # Noncompliant: always False
    ...</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">import numpy as np

x = np.nan

if np.isnan(x):
   ...</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>NumPy API Reference - <a href="https://numpy.org/doc/stable/reference/constants.html#numpy.nan">numpy.nan</a></p>
</li>
<li>
<p>NumPy API Reference - <a href="https://numpy.org/doc/stable/reference/generated/numpy.isnan.html">numpy.isnan()</a></p>
</li>
</ul>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Use 'numpy.isnan()' function instead of direct comparison.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
</div>
</div>