include::../description.adoc[]

== Noncompliant Code Example

[source,csharp]
----
public class Empty // Noncompliant
{
}
----

== Compliant Solution

[source,csharp]
----
public interface IEmpty
{
}
----

== Exceptions

Subclasses of System.Exception will be ignored, as even an empty Exception class can provide useful information by its type name alone.
Subclasses of certain framework types - like the PageModel class used in ASP.NET Core Razor Pages - will also be ignored.

----
using Microsoft.AspNetCore.Mvc.RazorPages;

public class EmptyPageModel: PageModel // Compliant - an empty PageModel can be fully functional, the C# code can be in the cshtml file
{ 
}
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]