== How to fix it in ASP.NET

=== Code examples

The following noncompliant code contains a hard-coded secret that can be exposed unintentionally.

==== Noncompliant code example

[source,csharp,diff-id=2,diff-type=noncompliant]
----
public class ConstLoginController : ApiController
{
    private const string key = "SecretSecretSecretSecretSecretSecretSecretSecret";

    public IHttpActionResult Post([FromBody] Login login)
    {
        // Code to validate user omitted

        var securityKey = new SymmetricSecurityKey(Encoding.UTF8.GetBytes(key)); // Noncompliant: hard-coded key in code
        var credentials = new SigningCredentials(securityKey, SecurityAlgorithms.HmacSha256);

        var secToken = new JwtSecurityToken(
            "sonarsource.com",
            "sonarsource.com",
            null,
            expires: DateTime.Now.AddMinutes(128),
            signingCredentials: credentials
        );

        var token = new JwtSecurityTokenHandler().WriteToken(secToken);

        return Ok(token);
    }
}
----

==== Compliant solution

[source,csharp,diff-id=2,diff-type=compliant]
----
public class ConstLoginController : ApiController
{
    private const string key = "SecretSecretSecretSecretSecretSecretSecretSecret";

    public IHttpActionResult Post([FromBody] Login login)
    {
        // Code to validate user omitted

        var key = Environment.GetEnvironmentVariable("JWT_KEY");
        var credentials = new SigningCredentials(securityKey, SecurityAlgorithms.HmacSha256);

        var secToken = new JwtSecurityToken(
            "sonarsource.com",
            "sonarsource.com",
            null,
            expires: DateTime.Now.AddMinutes(128),
            signingCredentials: credentials
        );

        var token = new JwtSecurityTokenHandler().WriteToken(secToken);

        return Ok(token);
    }
}
----

=== How does this work?

Here, the compliant solution uses an environement variable for the secret. 

