<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Using <code>0 as *const T</code> or <code>0 as *mut T</code> to represent a null pointer is error-prone and less readable. It can lead to confusion and potential bugs, as it is not immediately clear that <code>0</code> is intended to represent a null pointer. Additionally, using <code>0</code> for null pointers is not idiomatic Rust, and it can make the code harder to understand and maintain.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Use <code>std::ptr::null</code> or <code>std::ptr::null_mut</code> to represent null pointers. These functions are explicitly designed for this purpose and make the intent of the code clear.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-rust" data-lang="rust">let ptr = 0 as *const i32;
let mut_ptr = 0 as *mut i32;</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-rust" data-lang="rust">let ptr = std::ptr::null::&lt;i32&gt;();
let mut_ptr = std::ptr::null_mut::&lt;i32&gt;();</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Clippy Lints - <a href="https://rust-lang.github.io/rust-clippy/master/index.html#zero_ptr" class="bare">https://rust-lang.github.io/rust-clippy/master/index.html#zero_ptr</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>