== Why is this an issue?

If a private variable is declared but not used in the program, it can be considered dead code and should therefore be removed. This will improve maintainability because developers will not wonder what the variable is used for.

=== Noncompliant code example

[source,vb6]
----
Private Foo as Integer

Function Compute(A As Integer)
   Compute = A * 42
End Function
----

=== Compliant solution

[source,vb6]
----
Function Compute(A As Integer)
   Compute = A * 42
End Function
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Remove this unused "xxx" private variable. It is set n times but never read.


'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
