== Why is this an issue?

PHP's "variable variables" feature (dynamically-named variables) is temptingly powerful, but can lead to unmaintainable code.

==== Noncompliant code example

[source,php,diff-id=1,diff-type=noncompliant]
----
$var = 'foo';
$$var = 'bar';      //Noncompliant
$$$var = 'hello';  //Noncompliant

echo $foo; //will display 'bar'
echo $bar; //will display 'hello'
----

== Resources

=== Documentation

* PHP Manual - https://www.php.net/manual/en/language.variables.variable.php[Variable variables]

ifdef::env-github,rspecator-view[]

'''

== Implementation Specification

(visible only on this page)

include::message.adoc[]

'''

== Comments And Links

(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
