include::../introduction.adoc[]

== Why is this an issue?

include::../why-is-this-an-issue.adoc[]

include::../rule-behaviour.adoc[]

=== What is the potential impact?

include::../what-is-the-potential-impact.adoc[]

== How to fix it

include::../how-to-fix-it.adoc[]

=== Code examples

==== Noncompliant code example

The following example assumes that constant names should match the default
regular expression ``++[a-zA-Z]([a-zA-Z0-9_]*[a-zA-Z0-9])?++``:

[source,sql,diff-id=1,diff-type=noncompliant]
----
DECLARE
  constant_ CONSTANT PLS_INTEGER := 42; -- Noncompliant
BEGIN
  NULL;
END;
/
----

==== Compliant solution

[source,sql,diff-id=1,diff-type=noncompliant]
----
DECLARE
  constant CONSTANT PLS_INTEGER := 42; -- Compliant
BEGIN
  NULL;
END;
/
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

=== Parameters

.regexp
****

----
[a-zA-Z]([a-zA-Z0-9_]*[a-zA-Z0-9])?
----

The regular expression the name should match.
****


'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
