== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

With the default regular expression ``++[a-zA-Z]([a-zA-Z0-9_]*[a-zA-Z0-9])?++``:

[source,sql]
----
DECLARE
  constant_ CONSTANT PLS_INTEGER := 42; -- Noncompliant
BEGIN
  NULL;
END;
/
----

=== Compliant solution

[source,sql]
----
DECLARE
  constant CONSTANT PLS_INTEGER := 42; -- Compliant
BEGIN
  NULL;
END;
/
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::parameters.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
