== Why is this an issue?

include::../whyAnIssue.adoc[]

=== What is the potential impact?

include::../impact.adoc[]

== How to fix it

include::../howToFIxIt.adoc[]

=== Code examples

==== Noncompliant code example

[source,vbnet]
----
Private Sub CreateNewDateAndTime()
    Dim date = New DateTime(1994, 7, 5)
    Dim time = New DateTime(1, 1, 1, 16, 23, 42)
End Sub
----

==== Compliant solution

[source,vbnet]
----
Private Sub CreateNewDateAndTime()
    Dim date = New DateOnly(1994, 7, 5)
    Dim time = New TimeOnly(16, 23, 42)
End Sub
----

== Resources
=== Documentation

include::../documentation.adoc[]
