include::../why-dotnet.adoc[]

include::../impact-dotnet.adoc[]

include::../how-dotnet.adoc[]

=== Code examples

==== Noncompliant code example

[source,vbnet,diff-id=1,diff-type=noncompliant]
----
Private Sub CreateNewDateAndTime()
    Dim date = New DateTime(1994, 7, 5)
    Dim time = New DateTime(1, 1, 1, 16, 23, 42)
End Sub
----

==== Compliant solution

[source,vbnet,diff-id=1,diff-type=compliant]
----
Private Sub CreateNewDateAndTime()
    Dim date = New DateOnly(1994, 7, 5)
    Dim time = New TimeOnly(16, 23, 42)
End Sub
----

include::../resources-dotnet.adoc[]
