== How to fix it in Azure Storage Accounts

=== Code examples

==== Noncompliant code example

[source,terraform,diff-id=51,diff-type=noncompliant]
----
resource "azurerm_storage_account" "example" {
  min_tls_version = "TLS1_0" # Noncompliant
}
----

==== Compliant solution

Starting from AzureRM Provider 3.0, `min_tls_version` also defaults to `TLS1_2`.

[source,terraform,diff-id=51,diff-type=compliant]
----
resource "azurerm_storage_account" "example" {
  min_tls_version = "TLS1_2"
}
----

=== How does this work?

include::../../common/fix/fix.adoc[]
