== How to fix it in Azure Database for PostgreSQL

=== Code examples

==== Noncompliant code example

[source,terraform,diff-id=1,diff-type=noncompliant]
----
resource "azurerm_postgresql_server" "example" {
  ssl_minimal_tls_version_enforced = "TLS1_0" # Noncompliant
}
----

==== Compliant solution

Starting from AzureRM Provider 3.0, `ssl_minimal_tls_version_enforced` also defaults to `TLS1_2`.

[source,terraform,diff-id=1,diff-type=compliant]
----
resource "azurerm_postgresql_server" "example" {
  ssl_minimal_tls_version_enforced = "TLS1_2"
}
----

=== How does this work?

include::../../common/fix/fix.adoc[]
