== How to fix it in Azure MSSQL

=== Code examples

==== Noncompliant code example

[source,terraform,diff-id=1,diff-type=noncompliant]
----
resource "azurerm_mssql_server" "example" {
  name = "example"
  version = "12.0"

  minimum_tls_version = "1.0" # Noncompliant
}
----

==== Compliant solution

[source,terraform,diff-id=1,diff-type=compliant]
----
resource "azurerm_mssql_server" "example" {
  name = "example"
  version = "12.0"

  minimum_tls_version = "1.2"
}
----

=== How does this work?

include::../../common/fix/fix.adoc[]
