== How to fix it in GCP Load Balancers

=== Code examples

==== Noncompliant code example

[source,terraform,diff-id=41,diff-type=noncompliant]
----
resource "google_compute_ssl_policy" "example" {
  name            = "example"
  min_tls_version = "TLS_1_0" # Noncompliant
}
----

==== Compliant solution

[source,terraform,diff-id=41,diff-type=compliant]
----
resource "google_compute_ssl_policy" "example" {
  name            = "example"
  min_tls_version = "TLS_1_2"
}
----

=== How does this work?

include::../../common/fix/fix.adoc[]
