<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Historically <code>/COPY</code> procedures contained all of the specifications required for each procedure, so for example <code>xxxxF /COPY</code> would be used to copy the F lines from another <code>F</code> spec. However, this practice results in code that is difficult to understand and maintain. Instead, the use of <code>/COPY</code> statements should be replaced with explicit declarations.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="paragraph">
<p>Given the default list of specs, H,F,I,D,C,O:</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-rpg" data-lang="rpg">F/COPY HRSILERPG,RSX36</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-rpg" data-lang="rpg">FSSINRTP   IF   E           K DISK    INFSR(*PSSR)
F*
F*  Interest Rates File
F*</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Replace this <code>/COPY</code> directive with explicitly coded specifications</p>
</div>
</div>
<div class="sect2">
<h3 id="_parameters">Parameters</h3>
<div class="sidebarblock">
<div class="content">
<div class="title">specs</div>
<div class="listingblock">
<div class="content">
<pre>H,F,I,D,C,O</pre>
</div>
</div>
<div class="paragraph">
<p>Comma-delimited list of specs in which "/COPY" should not be used</p>
</div>
</div>
</div>
</div>
</div>
</div>