This rule raises an issue when `TemporalAccessor.get(ChronoField)` is called with a `ChronoField` value that is not supported by the specific temporal type.

== Why is this an issue?

Java's time API includes various temporal types like `DayOfWeek`, `Month`, `LocalDate`, and `LocalTime`. Each type only supports specific `ChronoField` values through the `get()` method. For example, `DayOfWeek` only supports `ChronoField.DAY_OF_WEEK`, while `LocalTime` supports time-related fields like `ChronoField.HOUR_OF_DAY` and `ChronoField.MINUTE_OF_HOUR`.

This rule affects the following general temporal classes:

* `DayOfWeek`
* `Instant`
* `IsoEra`
* `LocalDate`
* `LocalDateTime`
* `LocalTime`
* `Month`
* `MonthDay`
* `OffsetDateTime`
* `OffsetTime`
* `Year`
* `YearMonth`
* `ZonedDateTime`
* `ZoneOffset`

When you call `get()` with an unsupported field, the method throws an `UnsupportedTemporalTypeException` at runtime. This exception can crash your application and is entirely preventable through proper field selection.

The problem often occurs when developers assume all temporal types support the same fields, or when refactoring code without considering the specific capabilities of each temporal type.

=== What is the potential impact?

Using unsupported `ChronoField` values will cause your application to throw `UnsupportedTemporalTypeException` at runtime, potentially crashing the application.

== How to fix it

Use only supported ChronoField values for the specific TemporalAccessor type.

=== Code examples

==== Noncompliant code example

[source,java,diff-id=1,diff-type=noncompliant]
----
DayOfWeek dayOfWeek = DayOfWeek.MONDAY;
int hour = dayOfWeek.get(ChronoField.HOUR_OF_DAY); // Noncompliant
----

==== Compliant solution

[source,java,diff-id=1,diff-type=compliant]
----
DayOfWeek dayOfWeek = DayOfWeek.MONDAY;
int dayOfWeekValue = dayOfWeek.get(ChronoField.DAY_OF_WEEK);
----

== Resources

=== Documentation

 * TemporalAccessor (Java SE 17 & JDK 17) - https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/time/temporal/TemporalAccessor.html[Official documentation for TemporalAccessor interface and its get() method]

 * ChronoField (Java SE 17 & JDK 17) - https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/time/temporal/ChronoField.html[Complete list of ChronoField constants and their descriptions]
