== Why is this an issue?

From the official Oracle Javadoc:

____
NOTE++:++ The functionality of this Enumeration interface is duplicated by the Iterator interface. In addition, Iterator adds an optional remove operation, and has shorter method names. New implementations should consider using Iterator in preference to Enumeration.

____


=== Noncompliant code example

[source,java]
----
public class MyClass implements Enumeration {  // Non-Compliant
  /* ... */
}
----


=== Compliant solution

[source,java]
----
public class MyClass implements Iterator {     // Compliant
  /* ... */
}
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
