<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Unwrapping the result of <code>option_env!</code> will panic at runtime if the environment variable doesn&#8217;t exist, whereas <code>env!</code> catches it at compile-time, ensuring safer code execution.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-rust" data-lang="rust">---_
let _ = option_env!("HOME").unwrap(); // Noncompliant: Can panic at runtime.
---_</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-rust" data-lang="rust">---_
let _ = env!("HOME"); // Compliant: Ensures compile-time checking.
---_</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Clippy Lints - <a href="https://rust-lang.github.io/rust-clippy/master/index.html#option_env_unwrap" class="bare">https://rust-lang.github.io/rust-clippy/master/index.html#option_env_unwrap</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>