This rule raises an issue when a method name does not follow camelCase conventions, such as starting with an uppercase letter or using underscores.

== Why is this an issue?

Consistent method naming conventions are essential for code readability and maintainability. The camelCase convention, where method names start with a lowercase letter followed by mixed case letters and numbers, is widely adopted in many programming languages including Groovy.

When method names don't follow this convention, several problems arise:

* **Reduced readability**: Inconsistent naming makes code harder to scan and understand quickly
* **Team confusion**: Different naming styles within the same codebase create cognitive overhead for developers
* **Professional appearance**: Following established conventions makes code look more polished and professional
* **Tool compatibility**: Many IDEs and tools expect camelCase naming for proper code completion and refactoring

Method names that start with uppercase letters can be particularly confusing because they resemble class names, potentially misleading developers about the code's structure. Similarly, underscore-separated names (snake_case) are more common in other languages and can make Groovy code feel inconsistent with language conventions.

=== What is the potential impact?

While this issue doesn't directly affect application functionality, it impacts code maintainability and team productivity. Inconsistent naming conventions make code reviews more difficult, reduce code readability, and can slow down development as team members need extra time to understand non-standard naming patterns.

== How to fix it

Rename the method to follow camelCase convention, starting with a lowercase letter and using mixed case for subsequent words.

=== Code examples

==== Noncompliant code example

[source,groovy,diff-id=1,diff-type=noncompliant]
----
def MyMethod() { // Noncompliant
    return "result"
}
----

==== Compliant solution

[source,groovy,diff-id=1,diff-type=compliant]
----
def myMethod() {
    return "result"
}
----

== Resources

=== Documentation

 * Groovy Style Guide - https://groovy-lang.org/style-guide.html[Official Groovy style guide with naming conventions]

 * CodeNarc Naming Rules - https://codenarc.org/codenarc-rules-naming.html[Documentation for CodeNarc naming convention rules]

=== Related rules

 * RSPEC-100 - https://rules.sonarsource.com/java/RSPEC-100[Method names should comply with a naming convention (Java)]

 * RSPEC-101 - https://rules.sonarsource.com/csharp/RSPEC-101[Class names should comply with a naming convention]
