A ``++org.assertj.core.configuration.Configuration++`` will be effective only once you call ``++Configuration.apply()++`` or ``++Configuration.applyAndDisplay()++``.


This rule raises an issue when configurations are set without the appropriate call to apply them.

== Noncompliant Code Example

----
Configuration configuration = new Configuration(); // Noncompliant, this configuration will not be applied.
configuration.setComparingPrivateFields(true);{code}
----

== Compliant Solution

----
Configuration configuration = new Configuration();
configuration.setComparingPrivateFields(true);
configuration.applyAndDisplay();
// Alternatively: configuration.apply();
----

== See

* https://assertj.github.io/doc/#assertj-configuration[AssertJ configuration documentation]
