<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>When a variable in a nested scope has the same name as one in an outer scope, it hides the outer scope&#8217;s variable, which can no longer be accessed in the inner scope. Thus all uses of that variable name in the inner scope will reference the variable declared in the inner scope. However, that may not be clear to everyone. particularly in large code blocks. Therefore identifiers in inner scopes should never hide those from outer scopes.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-text" data-lang="text">void fn1 ( void )
{
  int16_t i;
  {
    int8_t i; /* Non-compliant - hides previous "i " */
    i = 3;       /* Could be confusing as to which "i" this is */
  }
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-text" data-lang="text">void fn1 ( void )
{
  int16_t i;
  {
    int8_t j;
    j = 3;
  }
}</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_duplicates_s1524">duplicates: <a data-rspec-id="S1524" class="rspec-auto-link">S1524</a></h3>

</div>
<div class="sect2">
<h3 id="_relates_to_s1524">relates to: <a data-rspec-id="S1524" class="rspec-auto-link">S1524</a></h3>

</div>
<div class="sect2">
<h3 id="_is_related_to_s1117">is related to: <a data-rspec-id="S1117" class="rspec-auto-link">S1117</a></h3>

</div>
</div>
</div>