When a variable in a nested scope has the same name as one in an outer scope, it hides the outer scope's variable, which can no longer be accessed in the inner scope. Thus all uses of that variable name in the inner scope will reference the variable declared in the inner scope. However, that may not be clear to everyone. particularly in large code blocks. Therefore identifiers in inner scopes should never hide those from outer scopes.


== Noncompliant Code Example

----
void fn1 ( void )
{
  int16_t i;
  {
    int8_t i; /* Non-compliant - hides previous "i " */
    i = 3;       /* Could be confusing as to which "i" this is */
  }
}
----


== Compliant Solution

----
void fn1 ( void )
{
  int16_t i;
  {
    int8_t j;
    j = 3;
  }
}
----


== See

* MISRA C:2004, 5.2 - Identifiers in an inner scope shall not use the same name as an identifier in an outer scope, and therefore hide that identifier
* MISRA {cpp}:2008, 2-10-2 - Identifiers declared in an inner scope shall not hide an identifier declared in an outer scope
* MISRA C:2012, 5.3 - An identifier declared in an inner scope shall not hide an identifier declared in an outer scope
* https://www.securecoding.cert.org/confluence/x/VwE[CERT, DCL01-C] - Do not reuse variable names in subscopes
* https://www.securecoding.cert.org/confluence/x/cwAhAQ[CERT, DCL01-CPP] - Do not reuse variable names in subscopes

