== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,vbnet]
----
For index As Integer = 6 To 5  // Noncompliant 
    // ...
Next
----

=== Compliant solution

[source,vbnet]
----
For index As Integer = 0 To 5
    // ...
Next
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
