=== Noncompliant code example

[source,text]
----
public void evaluate(int operator) {
  switch (operator) {
    /* ... */
    case ADD: { // Noncompliant - nested code block '{' ... '}'
        int a = stack.pop();
        int b = stack.pop();
        int result = a + b;
        stack.push(result);
        break;
      }
    /* ... */
  }
}
----
