== Why is this an issue?

include::description.adoc[]

== How to fix it

The superfluous code blocks should be extracted into separate methods

=== Code examples

==== Noncompliant code example

[source,text,diff-id=1,diff-type=noncompliant]
----
public void evaluate(int operator) {
  switch (operator) {
    /* ... */
    case ADD: { // Noncompliant - nested code block '{' ... '}'
        int a = stack.pop();
        int b = stack.pop();
        int result = a + b;
        stack.push(result);
        break;
      }
    /* ... */
  }
}
----

==== Compliant solution

To fix your code remove redundant blocks.

[source,text,diff-id=1,diff-type=compliant]
----
public void evaluate(int operator) {
  switch (operator) {
    /* ... */
    case ADD: // Compliant
      evaluateAdd();
      break;
    /* ... */
  }
}

private void evaluateAdd() {
  int a = stack.pop();
  int b = stack.pop();
  int result = a + b;
  stack.push(result);
}
----

== Resources

=== Documentation
* https://en.wikipedia.org/wiki/Single-responsibility_principle[Single Responsibility Principle | Wikipedia]