<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when a class constant (a variable declared with <code>static final</code> modifiers) uses a naming convention other than SNAKE_CASE with all uppercase letters.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Consistent naming conventions are essential for code maintainability and team collaboration. In Apex, the established convention for constants is to use SNAKE_CASE with all uppercase letters, where words are separated by underscores.</p>
</div>
<div class="paragraph">
<p>When constants don&#8217;t follow this convention, it creates several problems:</p>
</div>
<div class="ulist">
<ul>
<li>
<p><strong>Reduced readability</strong>: Mixed naming styles make it harder to quickly identify constants in the code</p>
</li>
<li>
<p><strong>Team confusion</strong>: Developers expect constants to follow the established pattern, and deviations can cause misunderstandings</p>
</li>
<li>
<p><strong>Maintenance overhead</strong>: Inconsistent naming makes code reviews more difficult and increases the cognitive load when working with the codebase</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>The SNAKE_CASE convention for constants is widely adopted across many programming languages and is specifically recommended in Salesforce development best practices. This convention makes constants easily distinguishable from variables and methods at a glance.</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="paragraph">
<p>While this issue doesn&#8217;t affect functionality, it reduces code maintainability and can slow down development. Inconsistent naming conventions make the codebase harder to understand and maintain, especially for new team members or when working with large codebases.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Rename the constant to use SNAKE_CASE with all uppercase letters. Replace any lowercase letters with uppercase and separate words with underscores.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-apex" data-lang="apex">private static final String accountLimit = '10'; // Noncompliant</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-apex" data-lang="apex">private static final String ACCOUNT_LIMIT = '10';</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Salesforce Naming Conventions Best Practices - <a href="https://www.apexhours.com/salesforce-naming-conventions-best-practices">Comprehensive guide to Salesforce naming conventions including constants</a></p>
</li>
</ul>
</div>
</div>
<div class="sect2">
<h3 id="_related_rules">Related rules</h3>
<div class="ulist">
<ul>
<li>
<p><a data-rspec-id="S115" class="rspec-auto-link">S115</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>