This rule raises an issue when `config.assets.compile` is set to `true` in Rails production environment configuration files.

== Why is this an issue?

Enabling asset compilation in production environments creates significant performance problems for Rails applications.

When `config.assets.compile = true` is set in production, the Rails asset pipeline compiles assets on-demand during runtime. This means every request for a CSS, JavaScript, or image file must be processed by Sprockets, the Rails asset compilation engine.

This live compilation approach has several serious drawbacks:

* *Performance overhead*: Each asset request requires compilation processing, adding latency to page loads
* *Server resource consumption*: The compilation process uses CPU and memory resources that should be dedicated to serving user requests
* *Missing optimizations*: Live compilation bypasses important production optimizations like asset minification, compression, and fingerprinting
* *Caching issues*: Dynamically compiled assets are harder to cache effectively, both on the server and in browsers
* *Deployment complexity*: Production servers need development dependencies like JavaScript runtimes for compilation

The Rails framework defaults to `config.assets.compile = false` in production for good reason. Assets should be precompiled during the deployment process using `rake assets:precompile`, which creates optimized, compressed, and fingerprinted versions that can be served efficiently by web servers.

=== What is the potential impact?

Applications with asset compilation enabled in production will experience slower page load times, increased server resource usage, and reduced scalability. Users will face longer wait times when loading pages, especially those with many CSS and JavaScript files. The server will consume more CPU and memory for asset processing instead of handling user requests, potentially leading to performance bottlenecks under load.

== How to fix it

Disable asset compilation in production and ensure assets are precompiled during deployment. Set up proper asset serving configuration for optimal performance.

=== Code examples

==== Noncompliant code example

[source,ruby,diff-id=1,diff-type=noncompliant]
----
# config/environments/production.rb
Rails.application.configure do
  # Avoid: Enabling live compilation in production causes performance issues
  config.assets.compile = true # Noncompliant
end
----

==== Compliant solution

[source,ruby,diff-id=1,diff-type=compliant]
----
# config/environments/production.rb
Rails.application.configure do
  # Disable asset compilation in production for better performance
  config.assets.compile = false
  
  # Enable serving of images, stylesheets, and JavaScripts from an asset server
  config.assets.digest = true
  
  # Precompile additional assets during deployment
  config.assets.precompile += %w( admin.js admin.css )
end
----

== Resources

=== Documentation

 * Rails Asset Pipeline Guide - https://guides.rubyonrails.org/asset_pipeline.html[Official Rails documentation on the asset pipeline and production configuration]

 * Rails Configuration Guide - Assets - https://guides.rubyonrails.org/configuring.html#configuring-assets[Detailed documentation on Rails asset configuration options]

 * Sprockets Documentation - https://github.com/rails/sprockets[Documentation for the Rails asset compilation engine]
