include::../description.adoc[]

== Noncompliant Code Example

----
for (int i = 10; i < 10; i++)  // Noncompliant 
{
    // ...
}
----

== Compliant Solution

----
for (int i = 0; i < 10; i++)
{
    // ...
}
----

ifdef::rspecator-view[]
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::rspecator-view[]
