== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,csharp]
----
for (int i = 10; i < 10; i++)  // Noncompliant 
{
    // ...
}
----

=== Compliant solution

[source,csharp]
----
for (int i = 0; i < 10; i++)
{
    // ...
}
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
