== Why is this an issue?

include::../rationale.adoc[]

include::../impact.adoc[]

== How to fix it

=== Code examples

include::../common/fix/code-rationale.adoc[]

==== Noncompliant code example

[source,cobol,diff-id=1,diff-type=noncompliant]
----
SOURCE-COMPUTER. IBM-370 WITH DEBUGGING MODE.
----

==== Compliant solution

[source,cobol,diff-id=1,diff-type=compliant]
----
SOURCE-COMPUTER. IBM-370.
----

=== How does this work?

``++WITH DEBUGGING MODE++`` activates all debug lines, meaning the ones with `D`
or `d` in the indicator area. This clause should not be used in production.

== Resources

include::../common/resources/standards.adoc[]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Make sure this debug clause is removed before delivering the code in production.

endif::env-github,rspecator-view[]
