<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This is an issue when you include an APIRouter in a FastAPI application before adding child routers to it. The child router&#8217;s path operations will not be registered with the application.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>FastAPI registers path operations at the moment you call <code>include_router()</code>. When you include a parent router in your FastAPI app, it captures only the path operations that exist in that router at that specific moment.</p>
</div>
<div class="paragraph">
<p>If you later add a child router to the parent router using <code>router.include_router(child_router)</code>, those new path operations are added to the parent router object. However, the FastAPI app has already completed its registration process and won&#8217;t automatically pick up these new routes.</p>
</div>
<div class="paragraph">
<p>This creates a situation where:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>The child router&#8217;s endpoints exist in the parent router&#8217;s internal structure</p>
</li>
<li>
<p>The FastAPI app doesn&#8217;t know about these endpoints</p>
</li>
<li>
<p>Requests to these endpoints will return 404 errors</p>
</li>
<li>
<p>The endpoints won&#8217;t appear in the automatic API documentation</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>The issue is particularly subtle because no error or warning is raised. The code runs without complaints, but the endpoints simply don&#8217;t work. This can lead to confusion during development and testing, as the routes appear to be properly defined in the code but are mysteriously unavailable at runtime.</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="paragraph">
<p>When child router endpoints are not properly registered:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>API endpoints return 404 errors even though they are defined in the code</p>
</li>
<li>
<p>The automatic API documentation (Swagger UI/ReDoc) will not show the missing endpoints</p>
</li>
<li>
<p>Integration tests may fail unexpectedly</p>
</li>
<li>
<p>Client applications will be unable to access the affected endpoints</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>This can cause significant debugging time as the code structure appears correct, but the runtime behavior doesn&#8217;t match expectations. In production, this could mean critical API functionality is unavailable to users.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it_in_fastapi">How to fix it in FastAPI</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Include all child routers in their parent routers before including the parent router in the FastAPI application. This ensures all path operations are registered when the parent router is added to the app.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">from fastapi import FastAPI, APIRouter

app = FastAPI()
parent_router = APIRouter()
child_router = APIRouter()

@child_router.get("/items")
def get_items():
    return {"items": []}

app.include_router(parent_router, prefix="/api")
parent_router.include_router(child_router)  # Noncompliant</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">from fastapi import FastAPI, APIRouter

app = FastAPI()
parent_router = APIRouter()
child_router = APIRouter()

@child_router.get("/items")
def get_items():
    return {"items": []}

parent_router.include_router(child_router)
app.include_router(parent_router, prefix="/api")</code></pre>
</div>
</div>
<div class="paragraph">
<p>When organizing routers across multiple files, ensure the module-level includes happen in the correct order. Import and include child routers before including the parent router in the main application.</p>
</div>
</div>
<div class="sect3">
<h4 id="_noncompliant_code_example_2">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python"># main.py
from fastapi import FastAPI
from .routers import parent, child

app = FastAPI()
app.include_router(parent.router)
parent.router.include_router(child.router)  # Noncompliant</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution_2">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python"># main.py
from fastapi import FastAPI
from .routers import parent, child

app = FastAPI()
parent.router.include_router(child.router)
app.include_router(parent.router)</code></pre>
</div>
</div>
<div class="paragraph">
<p>For complex applications with multiple levels of nesting, build the router hierarchy from the bottom up. Include the deepest child routers first, then work your way up to the top-level routers.</p>
</div>
</div>
<div class="sect3">
<h4 id="_noncompliant_code_example_3">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">from fastapi import FastAPI, APIRouter

app = FastAPI()
api_router = APIRouter()
v1_router = APIRouter()
users_router = APIRouter()

app.include_router(api_router)
api_router.include_router(v1_router, prefix="/v1")  # Noncompliant
v1_router.include_router(users_router, prefix="/users")  # Noncompliant</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution_3">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">from fastapi import FastAPI, APIRouter

app = FastAPI()
api_router = APIRouter()
v1_router = APIRouter()
users_router = APIRouter()

v1_router.include_router(users_router, prefix="/users")
api_router.include_router(v1_router, prefix="/v1")
app.include_router(api_router)</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>FastAPI - Bigger Applications - <a href="https://fastapi.tiangolo.com/tutorial/bigger-applications/">Official FastAPI documentation on structuring larger applications with multiple routers</a></p>
</li>
<li>
<p>FastAPI APIRouter Reference - <a href="https://fastapi.tiangolo.com/reference/apirouter/">API reference for the APIRouter class and its methods</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>