<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Automatic variables are those that are allocated automatically when program flow enters the object&#8217;s scope, and deallocated automatically when program flow leaves. Therefore returning the address of an automatic variable or object is an error because by the time the calling code attempts to use the value at the returned address, it no longer exists.</p>
</div>
<div class="paragraph">
<p>Unfortunately, this type of error is not always immediately evident. While the relevant memory has been deallocated, it may not have been overwritten by the time the object is dereferenced, thus leading to unpredictable behavior because sometimes the dereference is fully successful, sometimes it is partially successful (only parts of the object have been overwritten) and other times the dereference is a complete failure.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">int * divide (int numerator, int denominator) {
  int result = numerator / denominator;
  return (&amp;result);
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">int divide (int numerator, int denominator) {
  int result = numerator / denominator;
  return result;
}</code></pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p>MISRA C&#43;&#43;:2008, 7-5-1 - A function shall not return a reference or a pointer to an automatic variable (including parameters), defined within the function.</p>
</li>
<li>
<p><a href="https://wiki.sei.cmu.edu/confluence/x/UtcxBQ">CERT, DCL30-C.</a> - Declare objects with appropriate storage durations</p>
</li>
</ul>
</div>
<hr>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>XXX will be deallocated automatically when program flow leaves this function.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_relates_to_s2997">relates to: <a data-rspec-id="S2997" class="rspec-auto-link">S2997</a></h3>

</div>
<div class="sect2">
<h3 id="_relates_to_s946">relates to: <a data-rspec-id="S946" class="rspec-auto-link">S946</a></h3>

</div>
<div class="sect2">
<h3 id="_on_28_sep_2014_210543_evgeny_mandrikov_wrote">on 28 Sep 2014, 21:05:43 Evgeny Mandrikov wrote:</h3>
<div class="paragraph">
<p>What about MISRA C:2004 17.6 and MISRA C:2012 18.6, i.e. <a data-rspec-id="S946" class="rspec-auto-link">RSPEC-946</a>?</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_2_oct_2014_203231_ann_campbell_wrote">on 2 Oct 2014, 20:32:31 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~evgeny.mandrikov] I&#8217;d say the C rules are related but have a larger scope.</p>
</div>
<div class="listingblock">
<div class="content">
<pre>typedef struct {
  int *a;
} my_struct;

void doBad(my_struct *ms) {
  int i;
  my-&gt;a = &amp;i;  // Violates C rule but not C++ rule
}</pre>
</div>
</div>
<div class="paragraph">
<p>Would you like to cover the larger scope instead?</p>
</div>
</div>
</div>
</div>