== Why is this an issue?

Using element type in class selectors is slower than using only the class selector.


=== Noncompliant code example

[source,javascript]
----
var $products = $("div.products");    // Noncompliant - slow
----


=== Compliant solution

[source,javascript]
----
var $products = $(".products");    // Compliant - fast
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Remove "XXX" in this selector.


'''
== Comments And Links
(visible only on this page)

=== on 17 Mar 2015, 10:25:51 Linda Martin wrote:
\[~ann.campbell.2] assigning it to you for completion and review.

=== on 16 Jan 2020, 14:03:29 Elena Vilchik wrote:
Deprecating rule related to jquery (in the process of migration to new JS frontend we decided to deprecate Jquery rules, see \https://github.com/SonarSource/SonarJS/issues/1698)

endif::env-github,rspecator-view[]
