This rule raises an issue when `getNano()` is called on `LocalDateTime` or `LocalTime` objects without a corresponding `getSecond()` call in the same scope.

== Why is this an issue?

The `getNano()` method only returns the nanoseconds-of-second field (0 to 999,999,999), which is just the fractional part of a second, not a complete timestamp.

Many developers assume `getNano()` provides complete nanosecond timing information. However, without the seconds component from `getSecond()`, the nanoseconds value alone is rarely meaningful.

For example, times like `14:30:25.123456789` and `14:30:26.123456789` both return the same nanoseconds value (123456789) from `getNano()`, even though they are one second apart. This can lead to incorrect time calculations and faulty performance measurements.

=== What is the potential impact?

Using `getNano()` without `getSecond()` can result in incorrect time calculations and unreliable performance measurements. Applications that depend on precise timing may produce wrong results or exhibit unexpected behavior.

== How to fix it

Use both `getSecond()` and `getNano()` together to get complete sub-second precision. Store both values when you need accurate timing information.

=== Code examples

==== Noncompliant code example

[source,java,diff-id=1,diff-type=noncompliant]
----
LocalDateTime now = LocalDateTime.now();
long nanos = now.getNano(); // Noncompliant: only getting nanoseconds without seconds

// Later usage might be incorrect
long elapsed = nanos - previousNanos;
----

==== Compliant solution

[source,java,diff-id=1,diff-type=compliant]
----
LocalDateTime now = LocalDateTime.now();
int seconds = now.getSecond();
int nanos = now.getNano(); // Getting both components

// Or use a more appropriate method for timing
long totalNanos = now.toLocalTime().toNanoOfDay();
----

== Resources

=== Documentation

 * LocalDateTime.getNano() JavaDoc - https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/time/LocalDateTime.html#getNano()[Official documentation explaining that getNano() returns the nano-of-second field]
