Any extensible class might have subclasses located in a different package. When that happens, the use of ``++this.getClass().getResource++`` with a relative path would mean that the resource isn't found for the child class.


Instead, use an absolute path or make the class ``++final++``.


== Noncompliant Code Example

----
public class MyClass {

private URL url = null;

  public MyClass(){
    this.url = this.getClass().getResource("file.txt");  // Noncompliant
  }
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
