<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>C&#43;&#43; allows you to append a macro value onto the end of a string literal. Prior to C&#43;&#43;11, it was possible to do this either with or without a space between the two. But with the introduction of user-defined literals in C&#43;&#43;11, the preprocessing of string suffixes changed. To get the same string + macro behavior under C&#43;&#43; 11, you must separate the string literal and the macro with a space. Without the space, you&#8217;ll get a compile error.</p>
</div>
<div class="paragraph">
<p>For the purpose of preparing for migration to C&#43;&#43;11, this rule raises an issue when there&#8217;s no space between a string literal and a macro.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">#define _Hrs " hours"
static const char* OPENING = "7"_Hrs; // Noncompliant</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">#define _Hrs " hours"
static const char* OPENING = "7" _Hrs; // there's one space after "7"</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Add a space after this string literal.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>string literal</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_26_jul_2016_160720_ann_campbell_wrote">on 26 Jul 2016, 16:07:20 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>I&#8217;ve made a few more changes [~alban.auzeill]</p>
</div>
</div>
</div>
</div>