<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when template directives do not have a whitespace after the opening braces and before the closing braces.
It&#8217;s based on best practices in Helm and emphasizes the importance of maintaining this whitespace for better readability and understanding of the code.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>The absence of whitespace in template directives in Helm can lead to several issues:</p>
</div>
<div class="paragraph">
<p><strong>Readability</strong>: Code is read more often than it is written.
When template directives are tightly packed without whitespaces, it can be challenging to distinguish between the braces and the content inside.
This can slow down the process of understanding the code, especially for those who are new to the codebase.</p>
</div>
<div class="paragraph">
<p><strong>Maintainability</strong>: Hard-to-read code is also hard to maintain.
If a developer struggles to understand what a piece of code does due to poor formatting, they may introduce bugs when they try to modify it.</p>
</div>
<div class="paragraph">
<p>In summary, not having a whitespace after the opening brace and before the closing brace in template directives can make the code harder to read, understand, and maintain.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it_in_helm">How to fix it in Helm</h2>
<div class="sectionbody">
<div class="paragraph">
<p>To fix this issue, add a whitespace after the opening braces <code>{{</code> and before the closing braces <code>}}</code> in template directives.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-yaml" data-lang="yaml">apiVersion: v1
kind: Pod
metadata:
  name: {{.Values.podName}} # Noncompliant
spec:
  containers:
  - name: containerName</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-yaml" data-lang="yaml">apiVersion: v1
kind: Pod
metadata:
  name: {{ .Values.podName }}
spec:
  containers:
  - name: containerName</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Helm Best Practices - <a href="https://helm.sh/docs/chart_best_practices/templates/#formatting-templates">Formatting Templates</a></p>
</li>
<li>
<p>Go Documentation - <a href="https://pkg.go.dev/text/template?utm_source=godoc#hdr-Text_and_spaces">Text and spaces</a></p>
</li>
</ul>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>If possible try to display a specific message tailored to the issue.</p>
</div>
<div class="ulist">
<ul>
<li>
<p>Add a whitespace after <code>{{</code> in the template directive.</p>
</li>
<li>
<p>Add a whitespace before <code>}}</code> in the template directive.</p>
</li>
</ul>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="ulist">
<ul>
<li>
<p>Highlight the template directive that does not have a whitespace after opening braces <code>{{</code> or before closing braces <code>}}</code>.</p>
</li>
</ul>
</div>
</div>
</div>
</div>