<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Using include guards, wrapping around the entire content of a header file, is a best practice ensuring that no matter how many times the header is included in a translation unit, its content will only be seen once.</p>
</div>
<div class="paragraph">
<p>The include guard pattern is made up of four parts:</p>
</div>
<div class="ulist">
<ul>
<li>
<p><code>#ifndef</code> at the top of the file, with a unique macro name (usually, the name relates to the file&#8217;s name to ensure uniqueness).</p>
</li>
<li>
<p><code>#define</code>Â with the same macro name.</p>
</li>
<li>
<p>The content of the file</p>
</li>
<li>
<p><code>#endif</code> at the end of the file</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>The rule raises an issue when the name in the second part differs from the first (usually because of a typo or a copy/paste issue).</p>
</div>
<div class="paragraph">
<p>Because the include guard pattern is cumbersome, virtually every compiler provides a non-standard alternative: <code>#pragma once</code>. This directive prevents multiple inclusions of the file that contains it without needing to invent a unique macro name for each file. Note that it relies on the notion of <em>same file</em>, which can be tricky to determine. Additionally, it will not work with build systems that copy headers in different places.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">#ifndef MYFILE_H
#define MY_FILE_H // Noncompliant
//...
#endif</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="paragraph">
<p>The most straightforward way is to make both macro names match:</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">#ifndef MYFILE_H
#define MYFILE_H
//...
#endif</code></pre>
</div>
</div>
<div class="paragraph">
<p>An alternative is to use <code>#pragma once</code> instead:</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">#pragma once
//...</code></pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_external_coding_guidelines">External coding guidelines</h3>
<div class="ulist">
<ul>
<li>
<p>MISRA C&#43;&#43;:2023, 19.2.1 - Precautions shall be taken in order to prevent the contents of a header file being included more than once</p>
</li>
<li>
<p>MISRA C&#43;&#43;:2008, 16.2.3 - Include guards shall be provided</p>
</li>
</ul>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Replace this macro definition with header guard macro.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>macro definition</p>
</div>
</div>
</div>
</div>