=== On 2014-10-14T18:11:13Z Ann Campbell Wrote:
\[~samuel.mercier] please correct the following

* Code should be quoted in the message. Also, I don't quite understand the use of "::" here.
* Add a See section to the description listing the appropriate MISRA number
* Fill in Applicability.
* Make sure the appropriate MISRA C and MISRA {cpp} fields on the references tab are filled in
* Please use the standard headings, e.g. Noncompliant Code Example instead of just Example

In addition, please consider clarifying the following:

* "final type" during construction vs the type of "the completely constructed object" 
* the Noncompliant Code Example tells us in a comment that B2 is derived, but B2 doesn't actually seem to extend anything. 

=== On 2014-10-17T09:00:51Z Samuel Mercier Wrote:
\[~ann.campbell.2] updated.

* "final type" during construction vs the type of "the completely constructed object" : this is a copy-paste of the MISRA document, so I suspect it is meaningful and correct in the {cpp} context.
* ... but B2 doesn't actually seem to extend anything : that's correct. This example says that, if B2 is extended, then during the call of constructor B2 the dynamic type is not clearly defined.

