== How to fix it

=== Code examples

==== Noncompliant code example

[source,terraform]
----
resource "azurerm_role_definition" "example" { # Sensitive
  name        = "example"
  scope       = data.azurerm_subscription.primary.id

  permissions {
    actions     = ["*"]
    not_actions = []
  }

  assignable_scopes = [ 
    data.azurerm_subscription.primary.id 
  ]
}
----

==== Compliant solution

[source,terraform]
----
resource "azurerm_role_definition" "example" {
  name        = "example"
  scope       = data.azurerm_subscription.primary.id

  permissions {
    actions     = ["Microsoft.Compute/*"]
    not_actions = []
  }

  assignable_scopes = [ 
    data.azurerm_subscription.primary.id 
  ]
}
----