<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Calling <code>unittest</code> methods <code>assertEqual</code>, <code>assertNotEqual</code>, <code>assertIs</code> or <code>assertIsNot</code> on objects of incompatible types will always fail or always succeed.</p>
</div>
<div class="paragraph">
<p>For methods <code>assertEqual</code> and <code>assertNotEqual</code>, arguments' types are incompatible if:</p>
</div>
<div class="literalblock">
<div class="content">
<pre>they are unrelated builtin types such as string and integer.</pre>
</div>
</div>
<div class="literalblock">
<div class="content">
<pre>they are instances of unrelated classes which do not implement ``++__eq__++`` or ``++__ne__++`` (if a class implements one of these methods it could compare to any other type it wants).</pre>
</div>
</div>
<div class="paragraph">
<p>As for methods <code>assertIs</code> and <code>assertIsNot</code>, if arguments' types are different it is not possible for them to point to the same object, thus <code>assertIs</code> will always fail and <code>assertIsNot</code> will always succeed.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">import unittest
class A(): ...
class MyTest(unittest.TestCase):
  def test_something(self):
    a = A()
    mydict = {"x": a}
    self.assertEqual(a, "x")  # Noncompliant
    self.assertIs(a, "x")  # Noncompliant</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">import unittest
class A(): ...
class MyTest(unittest.TestCase):
  def test_something(self):
    a = A()
    mydict = {"x": a}
    self.assertEqual(a, mydict["x"])  # OK
    self.assertIs(a, mydict["x"])  # OK</code></pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p><a data-rspec-id="S2159" class="rspec-auto-link">S2159</a> Silly equality checks should not be made</p>
</li>
<li>
<p><a data-rspec-id="S3403" class="rspec-auto-link">S3403</a> Identity operators should not be used with dissimilar types</p>
</li>
</ul>
</div>
<hr>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Primary: Change this assertion to not compare dissimilar types ("XXX" and "YYY").</p>
</div>
<div class="paragraph">
<p>Secondary: Last assignment of "ZZZ".</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>Primary: the arguments of the assertXXX call.</p>
</div>
<div class="paragraph">
<p>Secondaries (if args are variables): the last assignment of the variable.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_relates_to_s2159">relates to: <a data-rspec-id="S2159" class="rspec-auto-link">S2159</a></h3>

</div>
</div>
</div>