Code snippets are useful for adding code examples to javadoc comments. This will help developers to better understand the usage of documented method. However, large code snippets might be confusing and can decrease the 
readability. So the general rule is to keep snippets as small as possible.

Sometimes there are cases where this is not possible to have a small snippet. For these cases we recommend to extract the snippet into a separate file and make it either external or a hybrid one.  

== Noncompliant Code Example

[source,java]
----
/**
 * The code prints "Hello, World!"
 * {@snippet :                                        // Noncompliant
 *  public static void main(String... args) {
 *      System.out.println("Hello, World!0");
 *      System.out.println("Hello, World!1");
 *      System.out.println("Hello, World!2");
 *      System.out.println("Hello, World!3");
 *      System.out.println("Hello, World!4");
 *      System.out.println("Hello, World!5");
 *      System.out.println("Hello, World!6");
 *      System.out.println("Hello, World!7");
 *      System.out.println("Hello, World!8");
 *      System.out.println("Hello, World!9");
 *      System.out.println("Hello, World!10");
 *      System.out.println("Hello, World!11");
 *      System.out.println("Hello, World!12");
 *      System.out.println("Hello, World!13");
 *      System.out.println("Hello, World!14");
 *      System.out.println("Hello, World!15");
 *      System.out.println("Hello, World!16");
 *      System.out.println("Hello, World!17");
 *      System.out.println("Hello, World!18");
 *      System.out.println("Hello, World!19");
 *      System.out.println("Hello, World!20");
 *  }
 * }
 */
----

== Compliant Solution

[source,java]
----
/**
 * The code prints "Hello, World!"
 * {@snippet :
 *  public static void main(String... args) {
 *      System.out.println("Hello, World!");
 *  }
 * }
 */
----

Or

[source,java]
----
{@snippet file=ExternalSnippet.java }
----

== See

Oracleâ€™s Programmer's Guide to Snippets: https://docs.oracle.com/en/java/javase/18/code-snippet/index.html[https://docs.oracle.com/en/java/javase/18/code-snippet/index.html]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

include::highlighting.adoc[]

include::parameters.adoc[]

endif::env-github,rspecator-view[]

