== How to fix it in Spring Data Cassandra

=== Code examples

include::../../common/fix/code-rationale.adoc[]

==== Noncompliant code example

[source,java,diff-id=21,diff-type=noncompliant]
----
@RestController
public class ApiController
{
    @Autowired
    CqlSession cqlSession;

    @GetMapping(value = "/read")
    @ResponseBody
    public List<String> readPublicOnly(
            @RequestParam("user") String user,
            @RequestParam("date") String date)
    {
        String query = "SELECT content FROM app.message WHERE user = '" + user + "' AND private = false AND date = '"+ date +"'";

        CqlOperations template = new CqlTemplate(cqlSession);
        return template.queryForList(query, String.class);
    }
}
----

==== Compliant solution

[source,java,diff-id=21,diff-type=compliant]
----
@RestController
public class ApiController
{
    @Autowired
    CqlSession cqlSession;

    @GetMapping(value = "/read")
    @ResponseBody
    public List<String> readPublicOnly(
            @RequestParam("user") String user,
            @RequestParam("date") String date)
    {
        String query = "SELECT content FROM app.message WHERE user = ? AND private = false AND date = ?";

        CqlOperations template = new CqlTemplate(cqlSession);
        LocalDate filterDate = LocalDate.parse(date);
        return template.queryForList(query, String.class, user, filterDate);
    }
}
----

=== How does this work?

include::../../common/fix/prepared-statements.adoc[]

