== How to fix it in Android

=== Code examples

==== Noncompliant code example

[source,kotlin,diff-id=11,diff-type=noncompliant]
----
public class ExampleActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        WebView aWebView = new WebView(this);
        aWebView.getSettings().setJavaScriptEnabled(true);
        setContentView(aWebView);

        String name = getIntent().getStringExtra("name");
        if (name == null) {
            name = "Guest";
        }

        aWebView.evaluateJavascript("greeting('" + name + "')", null); // Noncompliant
    }
}

----

==== Compliant solution

[source,kotlin,diff-id=11,diff-type=compliant]
----
public class ExampleActivity extends AppCompatActivity {

    private static final Pattern NAME_REGEX = Pattern.compile("^[a-zA-Z ]+$");

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        WebView aWebView = new WebView(this);
        aWebView.getSettings().setJavaScriptEnabled(true);
        setContentView(aWebView);

        String name = getIntent().getStringExtra("name");
        if (name == null) {
            uname = "Guest";
        }

        if (NAME_REGEX.matcher(untrustedName).matches()) {
            aWebView.evaluateJavascript("greeting('" + name + "')";, null); // Compliant
        } 
    }
}
----
