include::../description.adoc[]

== Noncompliant Code Example

----
If compare(myPoint.x, myPoint.x) <> 0 Then ' Noncompliant
    '...
End If

If compare(getNextValue(), getNextValue()) <> 0 Then ' Noncompliant
    '...
End If
----

== Compliant Solution

----
If compare(myPoint.x, myPoint.y) <> 0 Then
    '...
End If

Dim v1 As Integer = getNextValue()
Dim v2 As Integer = getNextValue()
If compare(v1, v2) <> 0 Then
    '...
End If
----
ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../highlighting.adoc[]

endif::env-github,rspecator-view[]
