Having all branches in a `+CASE+`, `+IF+` or `+IIF+` chain with the same implementation is an error. Either a copy-paste error was made and something different should be executed, or there shouldn't be a `+CASE+`/`+IF+`/`+IIF+` chain at all.

== Noncompliant Code Example

----
IF @x < 25 -- Noncompliant
  PRINT 'A'
ELSE IF @x < 10
  PRINT 'A'
ELSE
  PRINT 'A'

SELECT
  CASE col1 -- Noncompliant
    WHEN 1 THEN 'A'
    WHEN 2 THEN 'A'
    ELSE        'A'
  END,
  IIF(col1 < 25, 'A', 'A') -- Noncompliant
 FROM table1
----

== Exceptions

This rule does not apply to `+IF+`/`+CASE+` chains without `+ELSE+` clauses.

----
IF @x < 25 -- no issue, this could have been done on purpose to make the code more readable
  PRINT 'A'
ELSE IF @x > 10
  PRINT 'A'
----
