All reserved words should be written using the same case to ensure consistency in the code.

This rule checks that reserved words are all in lower case.


== Noncompliant Code Example

----
begin
  null;
  NULL; -- Noncompliant
end;
/
----

