Float data types, such as ``++FLOAT++``, ``++DOUBLE PRECISION++``, and ``++REAL++`` cannot have a scale constraint. Trying to specify a scale results in the exception ``++PLS-00510: Float cannot have scale being raised.++``


== Noncompliant Code Example

[source,sql]
----
DECLARE
  foo FLOAT(10, 3); -- Noncompliant - raises PLS-00510
BEGIN
  NULL;
END;
/
----


== Compliant Solution

[source,sql]
----
DECLARE
  foo FLOAT(10); -- Compliant
BEGIN
  NULL;
END;
/
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

endif::env-github,rspecator-view[]
