<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This is an issue when constructor functions return structs containing map fields that are not explicitly initialized with <code>make()</code>.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>In Go, maps have a zero value of <code>nil</code>. When a struct contains map fields and these maps are not initialized in the constructor function, they remain <code>nil</code>.</p>
</div>
<div class="paragraph">
<p>Attempting to write to a <code>nil</code> map causes a runtime panic with the message "assignment to entry in nil map". This can crash your application unexpectedly, especially when the map assignment happens later in the code, making the root cause harder to trace.</p>
</div>
<div class="paragraph">
<p>Constructor functions are meant to return fully usable instances of structs. When map fields are left uninitialized, the returned struct is in an incomplete state that can lead to runtime failures.</p>
</div>
<div class="paragraph">
<p>The <code>make()</code> function creates an initialized, empty map that is ready for use. By explicitly initializing map fields in constructors, you ensure that the struct is fully functional immediately after creation.</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="paragraph">
<p>The application will crash with a runtime panic when code attempts to write to the uninitialized map. This can lead to service disruptions and poor user experience, especially if the panic occurs in production environments.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Initialize map fields explicitly using <code>make()</code> in the struct literal returned by the constructor function.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-go" data-lang="go">func NewGraph() *Graph {
    return &amp;Graph{} // Noncompliant: vertices map is nil
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-go" data-lang="go">func NewGraph() *Graph {
    return &amp;Graph{
        vertices: make(map[int][]int),
    }
}</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Go Maps - <a href="https://go.dev/blog/maps">Official Go blog post explaining maps and their initialization</a></p>
</li>
<li>
<p>Effective Go - Maps - <a href="https://go.dev/doc/effective_go#maps">Best practices for using maps in Go</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>