In java 7 to 9, ``++FileInputStream++`` and ``++FileOutputStream++`` rely on finalization to perform final closes if the stream is not already closed. Whether or not the stream is already closed, the finalizer will be called, resulting in extra work for the garbage collector. This can easily be avoided using the ``++Files++`` API.


== Noncompliant Code Example

----
try(FileInputStream fis = new FileInputStream(...)) {  // Noncompliant
} finally {
}
----


== Compliant Solution

----
try(InputStream is = Files.newInputStream(...)) {
} finally {
}
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
