There are two types of sequence operations: intermediate operations, which return another sequence, and terminal operations, which return something other than a sequence. Intermediate operations are lazy, meaning they aren't actually executed until and unless a terminal sequence operation is performed on their results. Consequently, if the result of an intermediate sequence operation is not fed to a terminal operation, it serves no purpose, which is almost certainly an error.


== Noncompliant Code Example

----
widgets.asSequence()
    .filter { it.color == RED } // Noncompliant
----


== Compliant Solution

----
val sum = widgets.asSequence()
    .filter { it.color == RED }
    .map { it.weight }
    .sum()
----
----
val pipeline = widgets.asSequence()
    .filter { it.color == RED }
    .map { it.weight }

val sum = pipeline.sum()
----


== See

* https://kotlinlang.org/docs/sequences.html#sequence-operations[Sequence operations]

