== How to fix it in Core PHP

=== Code examples

include::../../common/fix/code-rationale.adoc[]

==== Noncompliant code example

[source,php,diff-id=1,diff-type=noncompliant]
----
function apply($effectName)
{
    try {
        $result = call_user_func($effectName, "applyFilter");
    } catch (\Throwable $e) {
        return "Filter Failure";
    }
    
    if ( $result == TRUE) {
        return "Filter Success";
    }
    else {
        return "Filter Failure";
    }
}

apply($_GET["filter"]);
----

==== Compliant solution

[source,php,diff-id=1,diff-type=compliant]
----
$EFFECT_ALLOW_LIST = array(
    "SepiaEffect",
    "BlackAndWhiteEffect",
    "WaterColorEffect",
    "OilPaintingEffect"
);

function apply($effectName)
{
    global $EFFECT_ALLOW_LIST;
    if (!in_array($effectName, $EFFECT_ALLOW_LIST)) {
        return "Filter Failure";
    }

    try {
        $result = call_user_func($effectName, "applyFilter");
    } catch (\Throwable $e) {
        return "Filter Failure";
    }
    
    if ( $result == TRUE) {
        return "Filter Success";
    }
    else {
        return "Filter Failure";
    }
}

apply($_GET["filter"]);
----

=== How does this work?

include::../../common/fix/prepared-statements.adoc[]

