include::../description.adoc[]

== Noncompliant Code Example

----
public class Foo
{
    public const double Version = 1.0;           // Noncompliant
}
----

== Compliant Solution

----
public class Foo
{
    public static double Version 
    {
      get { return 1.0; }
    }
}
----
ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

endif::env-github,rspecator-view[]
