== Why is this an issue?

Comparisons of dissimilar types will always return false. The comparison and all its dependent code can simply be removed. This includes:

* comparing an object with null
* comparing an object with an unrelated primitive (E.G. a string with an int)
* comparing unrelated types

=== Noncompliant code example

[source,dart]
----
void f() {
  var a = "Hello, World!";
  if (a == 42) {
    print("BOOM!");
  }
}
----

== Resources

* https://dart.dev/tools/linter-rules/unrelated_type_equality_checks[Dart Lint rule]