<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This is an issue when fuzzing functions using <code>*testing.F</code> are defined without appropriate build constraints to ensure compatibility with Go versions.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Go&#8217;s native fuzzing feature was introduced in Go 1.18. When you write fuzzing functions that use <code>*testing.F</code>, <code>f.Add()</code>, or <code>f.Fuzz()</code> without proper build constraints, the code will fail to compile on Go versions prior to 1.18.</p>
</div>
<div class="paragraph">
<p>This creates compatibility issues in projects that need to support multiple Go versions. Build failures can occur when:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>Team members use different Go versions</p>
</li>
<li>
<p>CI/CD pipelines run on older Go versions</p>
</li>
<li>
<p>The project needs to maintain backward compatibility</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>Build constraints (also called build tags) tell the Go compiler to include or exclude files based on conditions like Go version, operating system, or architecture. For fuzzing code, you need to specify that the file should only be compiled when Go 1.18 or later is available.</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="paragraph">
<p>Without proper build constraints, projects may experience compilation failures when built with Go versions prior to 1.18. This can break CI/CD pipelines, prevent team collaboration across different Go versions, and create deployment issues in environments with older Go installations.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Add build constraints at the top of files containing fuzzing functions. Use both the new <code>//go:build</code> syntax and the legacy <code>// +build</code> syntax for maximum compatibility.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-go" data-lang="go">package fuzz

import (
    "testing"
)

func FuzzExample(f *testing.F) { // Noncompliant
    f.Add("test")
    f.Fuzz(func(t *testing.T, s string) {
        // fuzzing logic
    })
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-go" data-lang="go">//go:build go1.18
// +build go1.18

package fuzz

import (
    "testing"
)

func FuzzExample(f *testing.F) {
    f.Add("test")
    f.Fuzz(func(t *testing.T, s string) {
        // fuzzing logic
    })
}</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Go Fuzzing Documentation - <a href="https://go.dev/doc/tutorial/fuzz">Official Go tutorial on fuzzing with examples and best practices</a></p>
</li>
<li>
<p>Build Constraints Documentation - <a href="https://pkg.go.dev/go/build#hdr-Build_Constraints">Official documentation on Go build constraints and build tags</a></p>
</li>
<li>
<p>Go 1.18 Fuzzing Blog Post - <a href="https://go.dev/blog/fuzz-beta">Announcement and introduction to Go&#8217;s native fuzzing feature</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>