<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Taking the address of a library function is not something robust: The library might make changes to a function that are compatible with a normal use of a function, but not with taking its address (for instance, adding a parameter with a default value, or adding an overload to an overload set). More specifically, the standard library has stated that there would be no barrier against such changes, and that for stability users should not take the address of standard library functions.</p>
</div>
<div class="paragraph">
<p>Starting with C&#43;&#43;20, it&#8217;s no longer allowed to take the address of a standard library function (with some exceptions with functions for formatting streams).</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">int main() {
  std::unique_ptr&lt;std::FILE, int(*)(std::FILE*)&gt; fp(
    std::fopen("test.txt", "r"),
    std::fclose); // Noncompliant, address of fclose is implicitly taken
  // Work with fp
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">int main() {
  std::unique_ptr&lt;std::FILE, int(*)(std::FILE*)&gt; fp(
    std::fopen("test.txt", "r"),
    [](std::FILE*file){return std::fclose(file);});
  // Work with fp
}</code></pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p><a href="https://wg21.link/P0921">Document of the C&#43;&#43; standardization committee about change management in the standard library.</a></p>
</li>
</ul>
</div>
<hr>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Don&#8217;t take the address of XXX, call it from a lambda instead</p>
</div>
</div>
</div>
</div>