include::../why.adoc[]

=== Code examples

==== Noncompliant code example

[source,flex,diff-id=1,diff-type=noncompliant]
----
public class Foo
{
  private function Foo() {}   //Compliant, private empty constructor intentionally used to prevent any direct instantiation of a class.
  public static function doSomething(): void 
  {
    var foo:Foo = new Foo();
    ...
  }
  
  private function unusedPrivateFunction(): void {...}
}
----

==== Compliant solution

[source,flex]
----
public class Foo
{
  private function Foo() {}   //Compliant, private empty constructor intentionally used to prevent any direct instantiation of a class.
  public static function doSomething():void 
  {
    var foo:Foo = new Foo();
    ...
  }
}
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
