:operationName: function
:visibility: private

include::../why.adoc[]

=== Code examples

==== Noncompliant code example

[source,flex,diff-id=1,diff-type=noncompliant]
----
public class Foo
{
  public static function doSomething(): void  // Compliant - public function
  {
    var foo:Foo = new Foo();
    ...
  }
  
  private function unusedPrivateFunction(): void {...}  // Noncompliant
}
----

==== Compliant solution

[source,flex, diff-id=1,diff-type=compliant]
----
public class Foo
{
  public static function doSomething(): void  // Compliant - public function
  {
    var foo:Foo = new Foo();
    ...
  }
}
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
