== Why is this an issue?

Passing an empty finisher to `Gatherer.of` or to `Gatherer.ofSequential` provides no additional value and removing the finisher clearly communicates that no finisher is applied.

== How to fix it

Call the overload of `Gatherer.of` or `Gatherer.ofSequential` that does not take a finisher.

=== Code examples

==== Noncompliant code example

[source,java,diff-id=1,diff-type=noncompliant]
----
Gatherer<Integer, AtomicInteger, Integer> gatherer = Gatherer.ofSequential(
  () -> new AtomicInteger(-1),
  (state, number, downstream) -> {
    if (state.get() < 0) {
      state.set(number);
      return true;
    }
    return downstream.push(number - state.get());
  },
  Gatherer.defaultFinisher()); // Noncompliant: useless finisher
----

==== Compliant solution

[source,java,diff-id=1,diff-type=compliant]
----
Gatherer<Integer, AtomicInteger, Integer> gatherer = Gatherer.ofSequential(
  () -> new AtomicInteger(-1),
  (state, number, downstream) -> {
    if (state.get() < 0) {
      state.set(number);
      return true;
    }
    return downstream.push(number - state.get());
  }); // Compliant
----

== Resources

* Oracle Documentation - https://docs.oracle.com/en/java/javase/24/docs/api/java.base/java/util/stream/Gatherer.html[Gatherer API]
