== Why is this an issue?

Since Oracle 11.2, ``++RELIES_ON++`` has been deprecated because the dependencies of result cache-enabled functions are automatically computed.


=== Noncompliant code example

[source,sql]
----
CREATE OR REPLACE FUNCTION foo RETURN PLS_INTEGER RESULT_CACHE RELIES_ON(DUAL) AS -- Noncompliant
BEGIN
  RETURN 0;
END;
/

DROP FUNCTION foo;
----


=== Compliant solution

[source,sql]
----
CREATE OR REPLACE FUNCTION foo RETURN PLS_INTEGER RESULT_CACHE AS
BEGIN
  RETURN 0;
END;
/

DROP FUNCTION foo;
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Remove this deprecated "RELIES_ON" clause which is now automatically computed.


endif::env-github,rspecator-view[]
