include::../description.adoc[]

== Noncompliant Code Example

----
Rem TODO Add documentation
Sub DoSomething()
    ' TODO
End Sub
----

include::../see.adoc[]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
