== Why is this an issue?

include::../description.adoc[]

==== Noncompliant code example

[source,yaml]
----
name: GitHub Actions Demo
run-name: ${{ github.actor }} is testing out GitHub Actions
on: [push]
jobs:
  Explore-GitHub-Actions:
    runs-on: ubuntu-latest
    steps:
      # TODO: Add more steps in this Github Actions
      - run: echo "The job was automatically triggered by a ${{ github.event_name }} event."
----

include::../see.adoc[]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
