== Why is this an issue?

The use of escape sequences is mostly unnecessary in text blocks.


=== Noncompliant code example

``++\n++`` can be replaced by simply introducing the newline, ``++\"\"\"++`` it is sufficient to escape only the first qoute.

[source,java]
----
String textBlock = """
        \"\"\" this \nis
        text  block!
        !!!!
      """;
----


=== Compliant solution

[source,java]
----
String textBlock = """
        \""" this
        is
        text  block!
        !!!!
      """;
----


== Resources

* https://openjdk.java.net/jeps/378[JEP 378: Text Blocks]
* https://openjdk.org/projects/amber/guides/text-blocks-guide[Programmer's Guide To Text Blocks], by Jim Laskey and Stuart Marks

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Remove this unnecessary escape sequence


endif::env-github,rspecator-view[]
