The use of escape sequences is mostly unnecessary in text blocks.


== Noncompliant Code Example

``++\n++`` can be replaced by simply introducing the newline, ``++\"\"\"++`` it is sufficient to escape only the first qoute.

----
String textBlock = """
        \"\"\" this \nis
        text  block!
        !!!!
      """;
----


== Compliant Solution

----
String textBlock = """
        \""" this 
        is
        text  block!
        !!!!
      """;
----


== See

* https://openjdk.java.net/jeps/378[JEP 378: Text Blocks]
* https://cr.openjdk.java.net/~jlaskey/Strings/TextBlocksGuide_v9.html[Programmer's Guide To Text Blocks], by Jim Laskey and Stuart Marks

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

endif::env-github,rspecator-view[]
