include::../description.adoc[]

== Why is this an issue?

include::../impact.adoc[]

== How to fix it in Android

include::../common/fix/android.adoc[]

=== Code examples

==== Noncompliant code example

[source,java,diff-id=11,diff-type=noncompliant]
----
public class LoginActivity extends Activity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        String username = ((EditText) findViewById(R.id.username)).getText().toString();
        String password = ((EditText) findViewById(R.id.password)).getText().toString();

        Log.i("LoginAttempt", "Login attempt: " + username + " with password: " + password); // Noncompliant
    }
}
----

==== Compliant solution

The simplest way to fix this is to not log sensitive data.

[source,java,diff-id=11,diff-type=compliant]
----
public class LoginActivity extends Activity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        String username = ((EditText) findViewById(R.id.username)).getText().toString();
        String password = ((EditText) findViewById(R.id.password)).getText().toString();

        Log.i("LoginAttempt", "Login attempt: " + username);
    }
}
----

include::../common/fix/proguard.adoc[]

include::../common/resources.adoc[]
