<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>The class <code>java.util.zip.GZIPInputStream</code> is already buffering its input while reading. Thus passing a <code>java.io.BufferedInputStream</code> to a <code>java.util.zip.GZIPInputStream</code> is redundant. It is more efficient to directly pass the original input stream to <code>java.util.zip.GZIPInputStream</code>.</p>
</div>
<div class="paragraph">
<p>Note that the default buffer size of <code>GZIPInputStream</code> is not the same as the one in <code>BufferedInputStream</code>. Configure it if need be.</p>
</div>
<div class="paragraph">
<p>This rule raises an issue when a <code>java.util.zip.GZIPInputStream</code> reads from a <code>java.io.BufferedInputStream</code>.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">import java.io.*;
import java.util.zip.GZIPInputStream;

public class Noncompliant {

    void deflateFile(final File file) throws IOException {
        try (
            FileInputStream fileStream = new FileInputStream(file);
            BufferedInputStream bufferedStream = new BufferedInputStream(fileStream);
            InputStream input = new GZIPInputStream(bufferedStream); // Noncompliant
        ) {
            // process the input
        }
    }
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">import java.io.*;
import java.util.zip.GZIPInputStream;
public class Compliant {

    void deflateFile(final File file) throws IOException {
        try (
            FileInputStream fileStream = new FileInputStream(file);
            InputStream input = new GZIPInputStream(fileStream);
        ) {
            // process the input
        }
    }
}</code></pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p>Stackoverflow question about usage of GZIPInputStream and the BufferedInputStream - <a href="https://stackoverflow.com/questions/4438085/seeking-out-the-optimum-size-for-bufferedinputstream-in-java/4438217#4438217">GZIPInputStream and the BufferedInputStream</a></p>
</li>
</ul>
</div>
<hr>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove the intermediate BufferedInputStream</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>Primary location: the GZIPInputStream instantiation</p>
</div>
<div class="paragraph">
<p>Secondary location: the BufferedInputStream instantiation.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_9_jul_2020_093733_nicolas_harraudeau_wrote">on 9 Jul 2020, 09:37:33 Nicolas Harraudeau wrote:</h3>
<div class="paragraph">
<p>\[~pierre-loup.tristant] There are a few references in other articles but this one explains it the best so I would prefer to keep it if possible. Otherwise I&#8217;ll replace it with <a href="https://stackoverflow.com/a/4438217">this stack overflow link</a></p>
</div>
</div>
</div>
</div>