<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when <code>getNano()</code> is called on <code>LocalDateTime</code> or <code>LocalTime</code> objects without a corresponding <code>getSecond()</code> call in the same scope.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>The <code>getNano()</code> method only returns the nanoseconds-of-second field (0 to 999,999,999), which is just the fractional part of a second, not a complete timestamp.</p>
</div>
<div class="paragraph">
<p>Many developers assume <code>getNano()</code> provides complete nanosecond timing information. However, without the seconds component from <code>getSecond()</code>, the nanoseconds value alone is rarely meaningful.</p>
</div>
<div class="paragraph">
<p>For example, times like <code>14:30:25.123456789</code> and <code>14:30:26.123456789</code> both return the same nanoseconds value (123456789) from <code>getNano()</code>, even though they are one second apart. This can lead to incorrect time calculations and faulty performance measurements.</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="paragraph">
<p>Using <code>getNano()</code> without <code>getSecond()</code> can result in incorrect time calculations and unreliable performance measurements. Applications that depend on precise timing may produce wrong results or exhibit unexpected behavior.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Use both <code>getSecond()</code> and <code>getNano()</code> together to get complete sub-second precision. Store both values when you need accurate timing information.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">LocalDateTime now = LocalDateTime.now();
long nanos = now.getNano(); // Noncompliant: only getting nanoseconds without seconds

// Later usage might be incorrect
long elapsed = nanos - previousNanos;</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">LocalDateTime now = LocalDateTime.now();
int seconds = now.getSecond();
int nanos = now.getNano(); // Getting both components

// Or use a more appropriate method for timing
long totalNanos = now.toLocalTime().toNanoOfDay();</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>LocalDateTime.getNano() JavaDoc - <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/time/LocalDateTime.html#getNano()">Official documentation explaining that getNano() returns the nano-of-second field</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>