Jump statements (``++return++``, ``++break++``, ``++continue++``, and ``++raise++``) move control flow out of the current code block. So any statements that come after a jump are dead code.

== Noncompliant Code Example

----
def fun(a):
  i = 10
  return i + a       # Noncompliant 
  i += 1             # this is never executed
----

== Compliant Solution

----
def fun(a):
  i = 10
  return i + a
----

include::../see.adoc[]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::highlighting.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
