Having two clauses in a ``++when++`` statement or two branches in an ``++if++`` chain with the same implementation is at best duplicate code, and at worst a coding error. If the same logic is truly needed for both instances, then in an ``++if++`` chain they should be combined, or for a ``++when++``, duplicates should be refactored. 

== Noncompliant Code Example

----
fun s1871(x: Int) {
    when (x) {
        1 -> {
            val y = x / 2
            print(y)
        }
        2 -> {
            val y = x / 2
            print(y)
        }
    }
}
----

== Exceptions

Blocks in an ``++if++`` chain that contain a single line of code are ignored, as are blocks in a ``++when++`` statement that contain a single line of code with or without a following ``++break++``.


----
if (a == 1) {
    doSomething()  //no issue, usually this is done on purpose to increase the readability
} else if (a == 2) {
    doSomethingElse()
} else {
    doSomething()
}
----

But this exception does not apply to ``++if++`` chains without ``++else++``-s, or to ``++when++``-es without ``++else++`` clauses when all branches have the same single line of code. In case of ``++if++`` chains with ``++else++``-s, or of ``++when++``-es with default clauses, rule S3923 raises a bug. 

----
if (a == 1) {
  doSomething()  //Noncompliant, this might have been done on purpose but probably not
} else if (a == 2) {
  doSomething()
}
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../highlighting.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
