<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Using certain features of regular expressions, it is possible to create regular expressions that can never match or contain subpatterns that can never match. Since a pattern or sub-pattern that can never match any input is pointless, this is a sign that the pattern does not work as intended and needs to be fixed.</p>
</div>
<div class="paragraph">
<p>This rule finds some such regular expressions and subpatterns, specifically ones that meet one of the following conditions:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>Beginning- and end-of-line/input boundaries appearing in a position where they can never match (e.g. an end-of-input marker being followed by other characters)</p>
</li>
<li>
<p>A back reference refers to a capturing group that will never be matched before the back reference</p>
</li>
</ul>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_boundaries">Boundaries</h3>
<div class="listingblock">
<div class="content">
<pre>$[a-z]*^</pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_backreference">Backreference</h3>
<div class="listingblock">
<div class="content">
<pre>\1(.)</pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_boundaries_2">Boundaries</h3>
<div class="listingblock">
<div class="content">
<pre>^[a-z]*$</pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_backreference_2">Backreference</h3>
<div class="listingblock">
<div class="content">
<pre>(.)\1</pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove this sub-pattern or rewrite the regex.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>The failing sub-pattern</p>
</div>
</div>
</div>
</div>