<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Test files in JavaScript and TypeScript are meant to contain test cases. These test cases are used to verify the functionality of your code and ensure that it behaves as expected. If a test file doesn&#8217;t contain any test cases, it&#8217;s not serving its purpose.</p>
</div>
<div class="paragraph">
<p>A test file without test cases might indicate:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>An incomplete test suite: Perhaps the developer started writing tests but didn&#8217;t finish.</p>
</li>
<li>
<p>A mistake: The developer might have accidentally deleted the test cases or moved them to another file.</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>This rule flags any file that has <code>.test</code> or <code>.spec</code> as part of its suffix but does not contain any test cases defined using the different forms of the <code>it</code> and <code>test</code> functions from Jasmine, Jest, Mocha, or Node.js testing API.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Add test cases to the file or delete it if it isn&#8217;t needed anymore.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-javascript" data-lang="javascript">// eval.test.js

/* no test cases */</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-javascript" data-lang="javascript">// eval.test.js

it('1 + 2 should give 3', () =&gt; {
    expect(1 + 2).toBe(3);
});</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Jasmine docs - <a href="https://jasmine.github.io/api/edge/global">API</a></p>
</li>
<li>
<p>Jest docs - <a href="https://jestjs.io/docs/api">API</a></p>
</li>
<li>
<p>Mocha docs - <a href="https://mochajs.org/#getting-started">API</a></p>
</li>
<li>
<p>Node.js docs - <a href="https://nodejs.org/api/test.html">API</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>