<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>There&#8217;s no point in having a PHPUnit test case without any test methods. Similarly, you shouldn&#8217;t have a file in the tests directory which extends PHPUnit\Framework\TestCase but no tests in the file. Doing either of these things may lead someone to think that uncovered classes have been tested. Add some test method or make the class abstract if it is used by a real test case class.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-php" data-lang="php">use PHPUnit\Framework\TestCase;

class MyTest extends TestCase {

  protected function setUp() {
    doSomethind();
  }


  private function doSomethind() {
    //...
  }
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-php" data-lang="php">use PHPUnit\Framework\TestCase;

class MyTest extends TestCase {

  public function testBehaviour() {
   //...
  }

  //...
}

// or

abstract class MyAbstractTest extends TestCase {

 protected function setUp() {
   doSomethind();
 }


 private function doSomethind() {
   //...
 }
}</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Add some tests to this class.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_replaces_s2700">replaces: <a data-rspec-id="S2700" class="rspec-auto-link">S2700</a></h3>

</div>
<div class="sect2">
<h3 id="_on_24_nov_2014_192835_nicolas_peru_wrote">on 24 Nov 2014, 19:28:35 Nicolas Peru wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2]What does "without any tests" means ? is that for classes without any test methods ? or is it tests without asserts ? This rule seems to lack some example.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_1_dec_2014_172444_ann_campbell_wrote">on 1 Dec 2014, 17:24:44 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~nicolas.peru] I meant without tests. WDYT, should this rule require both?</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_4_dec_2014_183816_ann_campbell_wrote">on 4 Dec 2014, 18:38:16 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~nicolas.peru] I just realized I answered your question by repeating what you asked about. Sorry.</p>
</div>
<div class="paragraph">
<p>I meant test methods.</p>
</div>
</div>
</div>
</div>