A loop with at most one iteration is equivalent to the use of an ``++if++`` statement to conditionally execute one piece of code. No developer expects to find such a use of a loop statement. If the initial intention of the author was really to conditionally execute one piece of code, an ``++if++`` statement should be used instead.


At worst that was not the initial intention of the author and so the body of the loop should be fixed to use the nested ``++return++``, ``++break++`` or ``++raise++`` statements in a more appropriate way.

== Noncompliant Code Example

----
while node is not None:
  node = node.parent()
  print(node)
  break
----

== Compliant Solution

----
while node is not None:
  node = node.parent()
  print(node)
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../highlighting.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
