<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Browsers best support the <code>window</code> mode for the <code>wmode</code> parameter, also in terms of accessibility.</p>
</div>
<div class="paragraph">
<p>As it is the default mode, it is acceptable to either not specify a <code>wmode</code> parameter altogether, or to set it explicitly to <code>window</code>.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-html" data-lang="html">&lt;object classid="clsid:d27cdb6e-ae6d-11cf-96b8-444553540000" width="550" height="400"&gt;
  &lt;param name="movie" value="movie_name.swf" /&gt;
  &lt;param name="wmode" value="direct" /&gt;                              &lt;!-- Non-Compliant --&gt;
&lt;/object&gt;

&lt;embed src="movie_name.swf"
       width="550"
       height="400"
       wmode="direct"                                                                 &lt;!-- Non-Compliant --&gt;
       type="application/x-shockwave-flash"
       pluginspage="http://www.macromedia.com/go/getflashplayer" /&gt;</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-html" data-lang="html">&lt;object classid="clsid:d27cdb6e-ae6d-11cf-96b8-444553540000" width="550" height="400"&gt;
  &lt;param name="movie" value="movie_name.swf" /&gt;
&lt;/object&gt;

&lt;embed src="movie_name.swf"
       width="550"
       height="400"
       type="application/x-shockwave-flash"
       pluginspage="http://www.macromedia.com/go/getflashplayer" /&gt;</code></pre>
</div>
</div>
<div class="paragraph">
<p>or</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-html" data-lang="html">&lt;object classid="clsid:d27cdb6e-ae6d-11cf-96b8-444553540000" width="550" height="400"&gt;
  &lt;param name="movie" value="movie_name.swf" /&gt;
  &lt;param name="wmode" value="window" /&gt;
&lt;/object&gt;

&lt;embed src="movie_name.swf"
       width="550"
       height="400"
       wmode="window"
       type="application/x-shockwave-flash"
       pluginspage="http://www.macromedia.com/go/getflashplayer" /&gt;</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Set the value of the 'wmode' parameter to 'window'.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_24_jun_2013_085207_dinesh_bolkensteyn_wrote">on 24 Jun 2013, 08:52:07 Dinesh Bolkensteyn wrote:</h3>
<div class="paragraph">
<p><a href="http://helpx.adobe.com/flash/kb/flash-object-embed-tag-attributes.html#main_Browser_support_for_Window_Mode" class="bare">http://helpx.adobe.com/flash/kb/flash-object-embed-tag-attributes.html#main_Browser_support_for_Window_Mode</a><em>wmode</em>values</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_8_jul_2013_182405_freddy_mallet_wrote">on 8 Jul 2013, 18:24:05 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>Is implemented by http://jira.codehaus.org/browse/SONARPLUGINS-2988</p>
</div>
</div>
</div>
</div>