== Why is this an issue?

The W3C specifications define the valid CSS properties. Only the official and browser-specific properties should be used to get the expected impact in the final rendering.


This rule ignores:

* ``++$sass++``, ``++@less++``, and ``++var(--custom-property)++`` variable syntaxes.
* vendor-prefixed properties (e.g., ``++-moz-align-self++``, ``++-webkit-align-self++``).


=== Noncompliant code example

[source,css]
----
a {
  colour: blue; /* Noncompliant; colour is not part of the specifications */
}
----


=== Compliant solution

[source,css]
----
a {
  color: blue;
}
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Replace this property by a valid one


endif::env-github,rspecator-view[]
