The W3C specifications define the valid CSS properties. Only the official and browser-specific properties should be used to get the expected impact in the final rendering.


This rule ignores:

* ``++$sass++``, ``++@less++``, and ``++var(--custom-property)++`` variable syntaxes.
* vendor-prefixed properties (e.g., ``++-moz-align-self++``, ``++-webkit-align-self++``).


== Noncompliant Code Example

----
a {
  colour: blue; /* Noncompliant; colour is not part of the specifications */
}
----


== Compliant Solution

----
a {
  color: blue;
}
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

endif::env-github,rspecator-view[]
