Code snippets are useful for adding code examples to javadoc comments. This will help developers to better understand the usage of documented methods. However, sometimes you can have the same code example duplicated in different inline snippets. In this case you can benefit from having an external or a hybrid snippet by extracting the code example into a separate file.


== Noncompliant Code Example

With the default threshold of 3:

[source,java]
----
/**
 * {@snippet name="snippet1" :            // Noncompliant, this snippet is duplicated
 *public static void main(String... args) {
 *    System.out.println("Hello World!");
 *}}
 */
public void f1() {}

/**
 * {@snippet name="snippet2" :
 *public static void main(String... args) {
 *    System.out.println("Hello World!");  // @highlight substring="out"
 *}}
 */
public void f2() {}


/**
 * {@snippet name="snippet3" :
 *public static void main(String... args) {
 *    System.out.println("Hello World!");  // @highlight substring="System"
 *}}
 */
public void f3() {}
----


== Compliant Solution

[source,java]
----
/**
 * {@snippet name="snippet1" file="HelloWorld.java"}
 */
public void f1() {}

/**
 * {@snippet name="snippet2" file="HelloWorld.java":
 * public static void main(String... args) {
 *     System.out.println("Hello World!");  // @highlight substring="out"
 * }}
 */
public void f2() {}

/**
 * {@snippet name="snippet3" file="HelloWorld.java":
 * public static void main(String... args) {
 *     System.out.println("Hello World!");  // @highlight substring="System"
 * }}
 */
public void f3() {}
----


== See

Oracleâ€™s Programmer's Guide to Snippets: https://docs.oracle.com/en/java/javase/18/code-snippet/index.html[https://docs.oracle.com/en/java/javase/18/code-snippet/index.html]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)


=== Message

Extract the content of the @nippet to the separate file instead of duplicating it {number} times.


=== Highlighting

* Primary: @snippet tag of the first occurrence
* Secondary: @snippet tag of the following occurrences
** message: 'Duplication.'


=== Parameters

.threshold
****

----
3
----

Number of times a literal must be duplicated to trigger an issue
****

endif::env-github,rspecator-view[]
