Code snippets are useful for adding code examples to javadoc comments. This will help developers to better understand the usage of documented methods. However, sometimes you can have the same code example duplicated in different inline snippets. In this case you can benefit from having an external or a hybrid snippet by extracting the code example into a separate file.

If you just want to reuse the content from the file you can simply do it with external snippets. However, if you want to use some additional tags you need to use the hybrid ones.


== Noncompliant Code Example

With the default threshold of 3 (the identical snippets should appear at least 3 times to trigger this rule) :

[source,java]
----
/**
 * {@snippet name="snippet1" :            // Noncompliant, this snippet is duplicated
 *public static void main(String... args) {
 *    System.out.println("Hello World!");
 *}}
 */
public void f1() {}

/**
 * {@snippet name="snippet2" :
 *public static void main(String... args) {
 *    System.out.println("Hello World!");  // @highlight substring="out"
 *}}
 */
public void f2() {}


/**
 * {@snippet name="snippet3" :
 *public static void main(String... args) {
 *    System.out.println("Hello World!");  // @highlight substring="System"
 *}}
 */
public void f3() {}
----


== Compliant Solution

This snippet is external 

[source,java]
----
/**
 * {@snippet name="snippet1" file="HelloWorld.java"}
 */
public void f1() {}
----

The following 2 snippets are hybrid so @highlight can be used here

[source,java]
----
/**
 * {@snippet name="snippet2" file="HelloWorld.java":
 * public static void main(String... args) {
 *     System.out.println("Hello World!");  // @highlight substring="out"
 * }}
 */
public void f2() {}

/**
 * {@snippet name="snippet3" file="HelloWorld.java":
 * public static void main(String... args) {
 *     System.out.println("Hello World!");  // @highlight substring="System"
 * }}
 */
public void f3() {}
----

And the content of the HelloWorld.java is:

[source, java]
----
public static void main(String... args) {
    System.out.println("Hello World!");
}
----


== See

Oracleâ€™s Programmer's Guide to Snippets: https://docs.oracle.com/en/java/javase/18/code-snippet/index.html[https://docs.oracle.com/en/java/javase/18/code-snippet/index.html]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)


=== Message

Extract the content of the @snippet to a separate file instead of duplicating it {number} times.


=== Highlighting

* Primary: @snippet tag of the first occurrence
* Secondary: @snippet tag of the following occurrences
** message: 'Duplication.'


=== Parameters

.threshold
****

----
3
----

Number of times a literal must be duplicated to trigger an issue
****

endif::env-github,rspecator-view[]
