:identifier_capital_plural: Local variables and parameters
:identifier: local variable and parameter
:identifier_plural: local variables and parameters
:identifier_or: local variable or parameter
:regex: ^[a-z][a-z0-9]*([A-Z]{1,3}[a-z0-9]+)*([A-Z]{2})?$

include::../rule.adoc[]

=== Code examples

==== Noncompliant code example

With the default regular expression ``{regex}``, bringing the following constraints:

* Camel casing, starting with a lowercase character, for example backColor
* Short abbreviations of 2 letters can be capitalized only when not at the beginning, for example id, productID
* Longer abbreviations need to be lowercased, for example html

[source,vbnet,diff-id=1,diff-type=noncompliant]
----
Module Module1
    Sub Main()
        Dim Foo = 0 ' Noncompliant
    End Sub
End Module
----

==== Compliant solution

[source,vbnet,diff-id=1,diff-type=compliant]
----
Module Module1
    Sub Main()
        Dim foo = 0 ' Compliant
    End Sub
End Module
----

== Resources

=== Documentation

* Microsoft Learn - https://learn.microsoft.com/en-us/dotnet/visual-basic/programming-guide/program-structure/naming-conventions[Visual Basic Naming Conventions]
* Wikipedia - https://en.wikipedia.org/wiki/Naming_convention_(programming)[Naming Convention (programming)]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../parameters.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
