:identifier_capital_plural: Local variables and parameters
:identifier: local variable and parameter
:identifier_plural: local variables and parameters
:identifier_or: local variable or parameter
:regex: ^[a-z][a-z0-9]*([A-Z]{1,3}[a-z0-9]+)*([A-Z]{2})?$

include::../rule.adoc[]
include::../parameters.adoc[]
// TODO REMOVE

=== Code examples

==== Noncompliant code example

With the default regular expression ``{regex}``, bringing the following constraints:

* Camel casing, starting with a lowercase character, for example backColor
* Short abbreviations of 2 letters can be capitalized only when not at the beginning, for example id, productID
* Longer abbreviations need to be lowercased, for example html


[source,vbnet]
----
Module Module1
    Sub Main()
        Dim Foo = 0 ' Noncompliant
    End Sub
End Module
----

=== Compliant solution

[source,vbnet]
----
Module Module1
    Sub Main()
        Dim foo = 0 ' Compliant
    End Sub
End Module
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../parameters.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
