== Why is this an issue?

include::../description.adoc[]

Code like

[source,python]
----
if condition1:
    if condition2:             # Noncompliant
        # ...
----

Will be more readable as

[source,python]
----
if condition1 and condition2:  # Compliant
    # ...
----

== How to fix it

If merging the conditions seems to result in a more complex code, extracting the condition or part of it in a named function or variable is a better approach to fix readability.

=== Code examples

==== Noncompliant code example

[source,python]
----
if file.isValid():
  if file.isfile() or file.isdir():     # Noncompliant
    # ...
----

==== Compliant solution

[source,python]
----
def isFileOrDirectory(File file):
  return file.isFile() or file.isDirectory()

if file.isValid() and isFileOrDirectory(file): # Compliant
  # ...
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../highlighting.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
