Yandex Cloud is a complete platform that provides services such as virtual machines, cloud storage, API gateways, and private networks, to name a few.
In Yandex Cloud, users are authenticated using secret keys and tokens. If one of these secret is compromised, attackers will be able to perform any action on behalf of the account or user associated with this secret.

include::../../../shared_content/secrets/description.adoc[]

== Why is this an issue?

include::../../../shared_content/secrets/rationale.adoc[]

If an attacker gains access to a Yandex token or key, they might be able to compromise your Yandex Cloud environment. This includes control over any applications or services that are running, as well as data that are managed by the account.

=== What is the potential impact?

If an attacker manages to gain access to the Yandex Cloud environment, there exist several ways that they could seriously harm your organization. Any data that is stored in the environment could be leaked, and the environment itself could even be tampered with.

include::../../../shared_content/secrets/impact/data_compromise.adoc[]

:service_name: Yandex Cloud

include::../../../shared_content/secrets/impact/infrastructure_takeover.adoc[]

== How to fix it

include::../../../shared_content/secrets/fix/revoke.adoc[]

include::../../../shared_content/secrets/fix/vault.adoc[]

=== Code examples

==== Noncompliant code example

[source,javascript,diff-id=1,diff-type=noncompliant]
----
import { Session, cloudApi, serviceClients } from '@yandex-cloud/nodejs-sdk';

const { resourcemanager: { cloud_service: { ListCloudsRequest } } } = cloudApi;

const session = new Session({ iamToken: 't1.7euelSbPyceKx87JqpuRl1qZiY-Ryi3rnpWaksrKaZqUppnLncmDnpeajZvl8_dZNAFl-e8ENXMH_t3z9xljfmT57wQ1cwf-.-LErty1vRh4S__VEp-aDnM5huB5MEfm_Iu1u2IzNgyrn0emiWDYA6rSQXDvzjE0O3HBbUlqoDeCmXYYInzZ6Cg' }); // Noncompliant
const cloudService = session.client(serviceClients.CloudServiceClient);

const response = await cloudService.list(ListCloudsRequest.fromPartial({
    pageSize: 100,
}));
----

==== Compliant solution

[source,javascript,diff-id=1,diff-type=compliant]
----
import { Session, cloudApi, serviceClients } from '@yandex-cloud/nodejs-sdk';

const { resourcemanager: { cloud_service: { ListCloudsRequest } } } = cloudApi;

const session = new Session({ iamToken: process.env.YANDEX_TOKEN });
const cloudService = session.client(serviceClients.CloudServiceClient);

const response = await cloudService.list(ListCloudsRequest.fromPartial({
    pageSize: 100,
}));
----

//=== How does this work?

//=== Pitfalls

//=== Going the extra mile

== Resources

=== Documentation

https://cloud.yandex.com/en/docs/iam/concepts/authorization/oauth-token[Yandex OAuth token]

https://cloud.yandex.com/en/docs/iam/concepts/authorization/iam-token[Yandex IAM token]

https://cloud.yandex.com/en/docs/iam/concepts/authorization/api-key[Yandex API key]

include::../../../shared_content/secrets/resources/standards.adoc[]
