include::../common/why.adoc[]

include::../common/how.adoc[]

=== Code examples

==== Noncompliant code example

[source,html,diff-id=1,diff-type=noncompliant]
----
<div>
    <span tabIndex="5">foo</span> <!-- Noncompliant -->
    <span tabIndex="3">bar</span> <!-- Noncompliant -->
    <span tabIndex="1">baz</span> <!-- Noncompliant -->
    <span tabIndex="2">qux</span> <!-- Noncompliant -->
</div>
----

==== Compliant solution

[source,html,diff-id=1,diff-type=compliant]
----
<div>
    <span tabIndex="0">foo</span>
    <span tabIndex="-1">bar</span>
    <span>qux</span>
</div>
----

include::../common/resources.adoc[]
