<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Nested <code>switch</code> structures are difficult to understand because you can easily confuse the cases of an inner <code>switch</code> as belonging to an outer statement. Therefore nested <code>switch</code> statements should be avoided.</p>
</div>
<div class="paragraph">
<p>Specifically, you should structure your code to avoid the need for nested <code>switch</code> statements, but if you cannot, then consider moving the inner <code>switch</code> to another function.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">void func(int n, int m) {

  switch (n) {
    case 1:
      // ...
    case 2:
      // ...
    case 3:
      switch (m) {  // Noncompliant
    case 4:  // Bad indentation makes this particularly hard to read properly
      // ...
    case 5:
      // ...
    case 6:
      // ...
    }
    case 4:
      // ...
    default:
      // ...
  }
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">void func(int n, int m) {

  switch (n) {
    case 1:
      // ...
    case 2:
      // ...
    case 3:
      int m2 = handle_m(m);
    case 4:
      // ...
    default:
      // ...
  }
}</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Refactor the code to eliminate this nested "switch".</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_2_oct_2014_140136_ann_campbell_wrote">on 2 Oct 2014, 14:01:36 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~evgeny.mandrikov] why duplicate the description here? The purpose of these sub-tasks is to hold variations.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_2_oct_2014_141559_evgeny_mandrikov_wrote">on 2 Oct 2014, 14:15:59 Evgeny Mandrikov wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] there is variation - "[subroutine|method|function]" replaced by "function".</p>
</div>
</div>
<div class="sect2">
<h3 id="_relates_to_s134">relates to: <a data-rspec-id="S134" class="rspec-auto-link">S134</a></h3>

</div>
<div class="sect2">
<h3 id="_on_23_sep_2014_081049_ann_campbell_wrote">on 23 Sep 2014, 08:10:49 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>Add C-Family code sample</p>
</div>
</div>
</div>
</div>