<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Nested <code>case</code> structures are difficult to understand because you can easily confuse the cases of an inner <code>case</code> as belonging to an outer statement. Therefore nested <code>case</code> statements should be avoided.</p>
</div>
<div class="paragraph">
<p>Specifically, you should structure your code to avoid the need for nested <code>case</code> statements, but if you cannot, then consider moving the inner <code>case</code> to another function.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-ruby" data-lang="ruby">def foo(n, m)
  case n
  when 0
      case m  # Noncompliant; nested case
      when 0 then puts "0"
        # ...
      end
  when 1 then puts "1"
  else puts "2"
  end
end</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-ruby" data-lang="ruby">def foo(n, m)
  case n
  when 0
    bar(m)
  when 1 then puts "1"
  else puts "2"
  end
end

def bar(m)
  case m
  when 0 then puts "0"
    # ...
  end
end</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Refactor the code to eliminate this nested "switch".</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_relates_to_s134">relates to: <a data-rspec-id="S134" class="rspec-auto-link">S134</a></h3>

</div>
<div class="sect2">
<h3 id="_on_23_sep_2014_081049_ann_campbell_wrote">on 23 Sep 2014, 08:10:49 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>Add C-Family code sample</p>
</div>
</div>
</div>
</div>