== Why is this an issue?

According to the PSR2 coding standard:

____
The keyword ``++elseif++`` SHOULD be used instead of ``++else if++`` so that all control keywords look like single words.

____


=== Noncompliant code example

[source,php]
----
if ($expr1) {
  ...
} else if ($expr2) {
  ...
} else {...}
----


=== Compliant solution

[source,php]
----
if ($expr1) {
  ...
} elseif ($expr2) {
  ...
} else {...}
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

endif::env-github,rspecator-view[]
