== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

With a maximum number of 4 parameters:

[source,php]
----
function doSomething($param1, $param2, $param3, $param4, $param5) { 
...
}
----

=== Compliant solution

[source,php]
----
function doSomething($param1, $param2, $param3, $param4) { 
...
}
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../parameters.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
