<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>In C&#43;&#43;, the literal 0 is both an integer type and the null-pointer-constant. To meet developer expectations, <code>NULL</code> should be used as the null-pointer-constant, and 0 for the integer zero.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">#include &lt;cstddef&gt;

void f1 ( int32_t );
void f2 ( int32_t * );
void f3 ( )
{
  f1 ( 0 ); // Compliant, integer expected
  f2 ( 0 ); // Noncompliant, 0 used as the null pointer constant
  f2 ( NULL ); // Compliant, pointer expected
}</code></pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p>MISRA C&#43;&#43;:2008, 4-10-2 - Literal zero (0) shall not be used as the null-pointer-constant.</p>
</li>
</ul>
</div>
<hr>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Replace this use of 0 with "NULL".</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_is_related_to_s948">is related to: <a data-rspec-id="S948" class="rspec-auto-link">S948</a></h3>

</div>
<div class="sect2">
<h3 id="_on_16_oct_2014_131353_ann_campbell_wrote">on 16 Oct 2014, 13:13:53 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~samuel.mercier] please:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>fill in the appropriate reference field(s).</p>
</li>
<li>
<p>provide a See section.</p>
</li>
<li>
<p>use double curly braces around code in the description, e.g. NULL.</p>
</li>
<li>
<p>use the standard section titles: Noncompliant Code Example</p>
</li>
</ul>
</div>
</div>
<div class="sect2">
<h3 id="_on_17_oct_2014_092122_samuel_mercier_wrote">on 17 Oct 2014, 09:21:22 Samuel Mercier wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] about the standard section titles for code, I wonder if it is appropriate.</p>
</div>
<div class="paragraph">
<p>The first line is compliant, and we want to show it. Since it is compliant it shouldn&#8217;t be in the noncompliant part. But since it is not a solution, it shouldn&#8217;t be in the Solution either. Also the current example is a copy-paste of the MISRA document, so we currently stick better to the spec.</p>
</div>
</div>
</div>
</div>