<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p><code>OCCURS DEPENDING ON</code> clauses are complicated to use correctly and do not provide any benefits with regard to memory consumption. It is best to avoid them.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cobol" data-lang="cobol">01  MYTABLEACCOUNT PIC S9(4) BINARY.
01  MYTABLE.
  05  MYITEM OCCURS 1 to 1000 DEPENDING ON MYTABLEACCOUNT.
    10  MYFIELD1 PIC X(8).
    10  MYFIELD2 PIC S9(4) BINARY.</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cobol" data-lang="cobol">01  MYTABLE.
  05  MYITEM OCCURS 1000.
    10  MYFIELD1 X(8).
    10  MYFIELD2 PIC S9(4) BINARY.</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Replace this clause with a simple "OCCURS" clause with no "DEPENDING [ON]".</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_24_mar_2017_172403_pierre_yves_nicolas_wrote">on 24 Mar 2017, 17:24:03 Pierre-Yves Nicolas wrote:</h3>
<div class="paragraph">
<p>On the web:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>"OCCURS DEPENDING ON is best avoided&#8230;&#8203;" (http://computer-programming-forum.com/48-cobol/58d65724a4bdc95b.htm)</p>
</li>
<li>
<p>"The main purpose of ODO is to increase the efficiency of the Binary Search." (ODO=Occurs Depending On) (http://ibmmainframes.com/about47790.html)</p>
</li>
</ul>
</div>
</div>
</div>
</div>