<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Step names should be unique within a job. Duplicate step names can lead to unpredictable behavior when using step references and overrides.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-text" data-lang="text">//MYJOB JOB TEST1
//STEP  EXEC PGM=PROG1   &lt;-- Noncompliant
//STEP  EXEC PGM=PROG2   &lt;-- Noncompliant</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-text" data-lang="text">//MYJOB JOB TEST1
//STEP1 EXEC PGM=PROG1
//STEP2 EXEC PGM=PROG2</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p><a href="https://www.ibm.com/docs/en/zos/3.1.0?topic=d-name-field-1">IBM Reference - EXEC Name field</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>