<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Restricting the number of exits from a loop is done in the interests of good structured programming. One <code>break</code> or <code>goto</code> statement is acceptable in a loop since this allows, for example, for dual-outcome loops or optimal coding.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="paragraph">
<p>With the default threshold of 1:</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">for (int i = 0; i &lt; 10; i++) {
  if (...) {
    break;      //  Compliant
  }
  else if (...) {
    break;      //  Non-compliant - second jump from loop
  }
  else {
    ...
  }
}
while (...) {
  if (...) {
    break;      // Compliant
  }
  if (...) {
    break;      // Non-compliant - second jump from loop
  }
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">for (int i = 0; i &lt; 10; i++) {
  if (...) {
    break;      //  Compliant
  }
}
while (...) {
  if (...) {
    break;    // Compliant
  }
}</code></pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p>MISRA C:2004, 14.6 - For any iteration statement there shall be at most one break statement used for loop termination.</p>
</li>
<li>
<p>MISRA C&#43;&#43;:2008, 6-6-4 - For any iteration statement there shall be no more than one break or goto statement used for loop termination.</p>
</li>
<li>
<p>MISRA C:2012, 15.4 - There should be no more than one break or goto statement used to terminate any iteration statement</p>
</li>
</ul>
</div>
<hr>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Reduce the number of nested break and goto statements from X to Y authorized.</p>
</div>
</div>
<div class="sect2">
<h3 id="_parameters">Parameters</h3>
<div class="sidebarblock">
<div class="content">
<div class="title">maxNumberOfTerminationStatements</div>
<div class="listingblock">
<div class="content">
<pre>1</pre>
</div>
</div>
<div class="paragraph">
<p>Maximum authorized number of break/goto statements</p>
</div>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_duplicates_s135">duplicates: <a data-rspec-id="S135" class="rspec-auto-link">S135</a></h3>

</div>
<div class="sect2">
<h3 id="_relates_to_s135">relates to: <a data-rspec-id="S135" class="rspec-auto-link">S135</a></h3>

</div>
<div class="sect2">
<h3 id="_relates_to_s1227">relates to: <a data-rspec-id="S1227" class="rspec-auto-link">S1227</a></h3>

</div>
</div>
</div>