<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Having too many steps in a single job or procedure makes it harder to understand and maintain.
It can also mean that it is aggregating too many responsibilities and should be split.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="paragraph">
<p>With a maximum number of step configured to 3:</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-jcl" data-lang="jcl">//MYJOB  JOB
//STEP01 EXEC PGM=DOTHING
//STEP02 EXEC PGM=DOTHING
//STEP03 EXEC PGM=DOTHING
//STEP04 EXEC PGM=DOTHING   &lt;-- Noncompliant</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-jcl" data-lang="jcl">//MYJOB  JOB
//STEP01 EXEC PGM=DOTHING
//STEP02 EXEC PGM=DOTHING
//STEP03 EXEC PGM=DOTHING</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_parameters">Parameters</h3>
<div class="sidebarblock">
<div class="content">
<div class="title">Max</div>
<div class="listingblock">
<div class="content">
<pre>50</pre>
</div>
</div>
<div class="paragraph">
<p>Maximum authorized steps in a job.</p>
</div>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p><a href="https://www.ibm.com/docs/en/zos/3.1.0?topic=tasks-entering-jobs">IBM reference - Entering Jobs</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>