== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,text]
----
function foo() {
  let x = bar(1);
  if (x > 0) {
      let x = bar(2); // Noncompliant
      console.log(x);
  } else {
     console.log("Wrong Value");
  }
}
----


=== Compliant solution

[source,text]
----
function foo() {
  let x = bar(1);
  if (x > 0) {
      let y = bar(2);
      console.log(y);
  } else {
     console.log("Wrong Value");
  }
}
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
