Using https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/operators/member-access-operators#null-conditional-operators--and-[conditional access] (`?.`) in C# is a common practice to safely handle potentially null objects and avoid NullReferenceException. However, if you are certain that an object is not null, you should consider not using conditional access and opt for the https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/operators/null-forgiving[null-forgiving operator] (`!`) instead if needed under nullable context.

== Why is this an issue?

Not using conditional access when you know an object is not `null` makes the code intent clearer and easier to maintain. It communicates to others reading your code that nullability is not a concern in that particular context. Also, while the performance impact is minimal, conditional access does introduce a slight overhead as it introduces an extra check.
Finally, since the conditional access is actually an if statement, it introduces a condition that cannot be fully tested, impacting the code coverage.

== How to fix it

If the https://learn.microsoft.com/en-us/dotnet/csharp/nullable-references[nullable context] is enabled and compiler raises a warning, you can replace the ? with !, otherwise you should just remove it.

=== Code examples

==== Noncompliant code example

[source,csharp,diff-id=1,diff-type=noncompliant]
----
string s = "NotNull";
_ = s?.ToString(); // Noncompliant
----

==== Compliant solution

[source,csharp,diff-id=1,diff-type=compliant]
----
string s = "NotNull";
_ = s.ToString(); // Compliant
----

== Resources

=== Documentation
* Microsoft Learn - https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/operators/member-access-operators#null-conditional-operators--and-[Conditional access]
* Microsoft Learn - https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/operators/null-forgiving[null-forgiving operator]
* Microsoft Learn - https://learn.microsoft.com/en-us/dotnet/csharp/nullable-references[nullable context]

include::../rspecator.adoc[]