Shared coding conventions allow teams to collaborate efficiently. This rule checks that data item levels are incremented according to the configured values.


Using the default configuration:


== Noncompliant Code Example

----
01 StudentDetails.
   05 StudentName. 
      12 FirstName     Pic x(10).   *> Noncompliant, level 10 is expected
      10 MiddleInitial Pic x.
      10 Surname       Pic x(15).  
   03 StudentId        Pic 9(7).      *> Noncompliant, only 05 level is expected
                 *> We should not mix two types of incrementation in the same data structure
...
01  Customer-Record.
    03  Customer-Name.
        10 Last-Name         Pic x(17).   *> Noncompliant, level 05 is expected
        05 Filler            Pic x.
        05 Initials          Pic xx.
    03  Part-Order.
        05 Part-Name         Pic x(15).
        05 Part-Color        Pic x(10).
            10 Part-Color-Code        Pic x.   *> Noncompliant, level 07 is expected
----


== Compliant Solution

----
01 StudentDetails.
   05 StudentName. 
      10 FirstName     PIC X(10).
      10 MiddleInitial PIC X.
      10 Surname       PIC X(15).  
   05 StudentId        PIC 9(7).      
...
01  Customer-Record.
    03  Customer-Name.
        05 Last-Name         Pic x(17).
        05 Filler            Pic x.
        05 Initials          Pic xx.
    03  Part-Order.
        05 Part-Name         Pic x(15).
        05 Part-Color        Pic x(10).
            07 Part-Color-Code        Pic x.
----


ifdef::env-github,rspecator-view[]
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
