== Why is this an issue?

This rule raises an issue when an HTML element has a ``++lang++`` and/or ``++xml:lang++`` attribute, contains text content within its subtree, and the value of the attribute does not begin with a valid two‑letter ISO 639‑1 language code.

The `lang` attribute is used by assistive technologies, translation tools, and search engines to correctly interpret and process text content.
If the value starts with an invalid language code — for example, a misspelled code or one not defined in ISO 639‑1 — these tools may mispronounce text, fail to translate it correctly, or misclassify the content’s language.

By ensuring that `lang` and `xml:lang` attributes start with a valid two‑letter language code, you improve accessibility, usability, and interoperability.

=== Noncompliant code examples

[source,html]
----
<html lang="engl">
  <head>
    <title>Example</title>
    <meta charset="utf-8" />
  </head>
  <body>
    <p>Welcome to our website</p> <!-- Noncompliant: "engl" does not start with a valid 2-letter code -->
  </body>
</html>
----

[source,html]
----
<html lang="en">
  <div lang="xx">
    Bienvenido a nuestro sitio web <!-- Noncompliant: "xx" is not a valid ISO 639-1 code -->
  </div>
</html>
----
=== Compliant solutions

[source,html]
----
<html lang="en">
  <head>
    <title>Example</title>
    <meta charset="utf-8" />
  </head>
  <body>
    <p>Welcome to our website</p> <!-- Valid: "en" is a valid ISO 639-1 code -->
  </body>
</html>
----

[source,html]
----
<html lang="en">
  <div lang="es">
    Bienvenido a nuestro sitio web <!-- Valid: "es" is a valid ISO 639-1 code -->
  </div>
</html>
----

== Resources

* HTML Standard – The `lang` attribute – https://html.spec.whatwg.org/multipage/dom.html#attr-lang
* MDN – The `lang` global attribute – https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/lang
* ISO 639‑1 – Codes for the representation of names of languages – https://en.wikipedia.org/wiki/List_of_ISO_639-1_codes
* WCAG 2.1 – Success Criterion 3.1.1: Language of Page – https://www.w3.org/WAI/WCAG21/Understanding/language-of-page.html
* WCAG 2.1 – Success Criterion 3.1.2: Language of Parts – https://www.w3.org/WAI/WCAG21/Understanding/language-of-parts.html

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Add "lang" and/or "xml:lang" attributes to this "<html>" element
or
Text is missing a valid lang attribute in its ancestor elements

=== Highlighting

Highlight the <html> element
or
Highlight the text content, which has an invalid lang attribute

endif::env-github,rspecator-view[]
