<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>By default, Hibernate <code>session</code> flushing is set to <code>FlushMode.AUTO</code>, and is called from <code>Transaction.commit</code>, <code>Session.flush</code> and before some queries are executed. Setting it to <code>FlushMode.COMMIT</code>, <code>FlushMode.NEVER</code>, or <code>FlushMode.MANUAL</code> could mean that parts of your application get stale data, so you should be sure of what you&#8217;re doing before you use any of these modes.</p>
</div>
<div class="paragraph">
<p>This rule raises an issue when flush mode is explicitly set to any of these modes.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">Session session = sessionFactory.openSession();
session.setFlushMode(FlushMode.NEVER);  // Noncompliant</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Make sure you really don&#8217;t want "AUTO" flushing.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="ulist">
<ul>
<li>
<p>primary: the FlushMode</p>
</li>
</ul>
</div>
</div>
</div>
</div>