<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>With the introduction of the <code>@snippet</code> tag in Java 18, it is possible to include code snippets inside a javadoc. Code inside these snippets should use spaces only or tabs only (i.e., not a mix of both). Not doing so will lead to the inconsistent indentation in the javadoc output.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">{@snippet :
class HelloWorld {                               // Noncompliant - spaces and tabs are mixed
     public static void main(String... args) {
&lt;TAB&gt;&lt;TAB&gt;System.out.println("Hello World!");
     }
}
}</code></pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_use_spaces_only">Use spaces only</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">{@snippet :
class HelloWorld {
     public static void main(String... args) {
        System.out.println("Hello World!");
     }
}
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_use_tabulations_only">Use tabulations only</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">{@snippet :
class HelloWorld {
&lt;TAB&gt;public static void main(String... args) {
&lt;TAB&gt;&lt;TAB&gt;System.out.println("Hello World!");
&lt;TAB&gt;}
}
}</code></pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_see">See</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Oracleâ€™s Programmer&#8217;s Guide to Snippets: <a href="https://docs.oracle.com/en/java/javase/18/code-snippet/index.html">https://docs.oracle.com/en/java/javase/18/code-snippet/index.html</a></p>
</div>
<hr>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Use spaces only or tabs only for indentation.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>The full code section inside the @snippet tag</p>
</div>
</div>
</div>
</div>