<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>The <code>empty()</code> function in PHP is specifically designed to check if a variable is empty, and it returns true if the variable is empty or evaluates to false. It handles various types of values, such as empty strings, zero, null, and arrays with no elements. This makes the code more readable because it clearly expresses the intention of checking for emptiness.</p>
</div>
<div class="paragraph">
<p>On the other hand, the <code>count()</code> function in PHP is primarily used to count the number of elements in an array or the properties of an object. It’s less explicit and may be less intuitive for someone reading the code.</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="paragraph">
<p>In terms of performance, <code>empty()</code> is generally faster than <code>count()</code> because it doesn’t need to iterate over all the elements in an array to count them. Instead, it directly checks if the variable is empty or evaluates to false. This can be advantageous when dealing with large arrays or performance-sensitive code.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-php" data-lang="php">if (count($a) === 0) { // Noncompliant
 echo $a[0];
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-php" data-lang="php">if (empty($a)) {
  echo $a[0];
}</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p><a href="https://www.php.net/manual/en/function.empty">PHP Manual - empty</a></p>
</li>
<li>
<p><a href="https://www.php.net/manual/en/function.count.php">PHP Manual - count</a></p>
</li>
</ul>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Use isEmpty() to check whether the collection is empty or not.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_is_related_to_s3981">is related to: <a data-rspec-id="S3981" class="rspec-auto-link">S3981</a></h3>

</div>
<div class="sect2">
<h3 id="_is_related_to_s2250">is related to: <a data-rspec-id="S2250" class="rspec-auto-link">S2250</a></h3>

</div>
<div class="sect2">
<h3 id="_on_30_jul_2013_170625_dinesh_bolkensteyn_wrote">on 30 Jul 2013, 17:06:25 Dinesh Bolkensteyn wrote:</h3>
<div class="paragraph">
<p>Implemented by https://jira.codehaus.org/browse/SONARJAVA-243</p>
</div>
</div>
</div>
</div>