<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Using <code>[Int]().count</code> to test for emptiness works, but using <code>[Int]().isEmpty</code> makes the code more readable and can be more performant. The time complexity of any <code>isEmpty</code> implementation should be <code>O(1)</code> whereas some implementations of <code>count()</code> can be <code>O(n)</code>.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-swift" data-lang="swift">if (arr.count == 0) {  // Noncompliant
  /* ... */
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-swift" data-lang="swift">if (arr.isEmpty) {
  /* ... */
}</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Use isEmpty() to check whether the collection is empty or not.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_8_jul_2016_204054_ann_campbell_wrote">on 8 Jul 2016, 20:40:54 Ann Campbell wrote:</h3>
<div class="paragraph">
<p><a href="https://github.com/realm/SwiftLint/blob/master/Source/SwiftLintFramework/Rules/EmptyCountRule.swift" class="bare">https://github.com/realm/SwiftLint/blob/master/Source/SwiftLintFramework/Rules/EmptyCountRule.swift</a></p>
</div>
</div>
<div class="sect2">
<h3 id="_is_related_to_s3981">is related to: <a data-rspec-id="S3981" class="rspec-auto-link">S3981</a></h3>

</div>
<div class="sect2">
<h3 id="_is_related_to_s2250">is related to: <a data-rspec-id="S2250" class="rspec-auto-link">S2250</a></h3>

</div>
<div class="sect2">
<h3 id="_on_30_jul_2013_170625_dinesh_bolkensteyn_wrote">on 30 Jul 2013, 17:06:25 Dinesh Bolkensteyn wrote:</h3>
<div class="paragraph">
<p>Implemented by https://jira.codehaus.org/browse/SONARJAVA-243</p>
</div>
</div>
</div>
</div>