<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Using <code>.Count()</code> to test for emptiness works, but using <code>.Any()</code> makes the intent clearer, and the code more readable. However, there are some cases where special attention should be paid:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>if the collection is an <code>EntityFramework</code> or other ORM query, calling <code>.Count()</code> will cause executing a potentially massive SQL query and could put a large overhead on the application database. Calling <code>.Any()</code> will also connect to the database, but will generate much more efficient SQL.</p>
</li>
<li>
<p>if the collection is part of a LINQ query that contains <code>.Select()</code> statements that create objects, a large amount of memory could be unnecessarily allocated. Calling <code>.Any()</code> will be much more efficient because it will execute fewer iterations of the enumerable.</p>
</li>
</ul>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-vbnet" data-lang="vbnet">Private Function HasContent(Strings As IEnumerable(Of String)) As Boolean
    Return Strings.Count() &gt; 0      ' Noncompliant
End Function

Private Function HasContent2(Strings As IEnumerable(Of String)) As Boolean
    Return Strings.Count() &gt;= 1     ' Noncompliant
End Function

Private Function IsEmpty(Strings As IEnumerable(Of String)) As Boolean
    Return Strings.Count() = 0      ' Noncompliant
End Function</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-vbnet" data-lang="vbnet">Private Function HasContent(Strings As IEnumerable(Of String)) As Boolean
    Return Strings.Any
End Function

Private Function HasContent2(Strings As IEnumerable(Of String)) As Boolean
    Return Strings.Any
End Function

Private Function IsEmpty(Strings As IEnumerable(Of String)) As Boolean
    Return Not Strings.Any
End Function</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Use ".Any()" to test whether this "IEnumerable(Of XXX)" is empty or not.</p>
</div>
</div>
</div>
</div>