<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when a <code>tensorflow.function</code> depends on a Python object.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>A Python object here is refers to a plain Python object where its attributes are not <code>tensorflow.Variable</code>s.
When using such an object as an argument of a <code>tensorflow.function</code>, no new <code>tracing</code> is created when the Python object is modified.
This can lead to unpredictable outcomes.</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">import tensorflow as tf

class SimpleModel(tf.Module):
  def __init__(self):
    self.bias = 0.
    self.weight = 2.

@tf.function
def evaluate(model, x):
  return model.weight * x + model.bias

x = tf.constant(10.)

simple_model = SimpleModel()
evaluate(simple_model, x)  # returns 20 as expected

simple_model.bias += 5.0
evaluate(simple_model, x) # still returns 20 instead of 25</code></pre>
</div>
</div>
<div class="paragraph">
<p>In the example above only the first call to evaluate will create a <code>ConcreteFunction</code> during <code>tracing</code>.
This <code>ConcreteFunction</code> will be reused on further calls of <code>evaluate</code>.</p>
</div>
<div class="paragraph">
<p>For more information about tracing rules see: <a href="https://www.tensorflow.org/guide/function#rules_of_tracing">Rules of tracing</a></p>
</div>
<div class="paragraph">
<p>For this reason it a good practice not to use Python objects as arguments to <code>tensorflow.function</code>s.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>To fix this issue use a model with <code>tensorflow.Variable</code>s as attributes.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">import tensorflow as tf

class BiasModel(tf.Module):
  def __init__(self):
    self.bias = 0.
    self.weight = 2.

@tf.function
def evaluate(model, x):
  return model.weight * x + model.bias

bias_model = BiasModel()
evaluate(bias_model, x)  # Noncompliant: the model used is a Python object

bias_model.bias += 5.0
evaluate(bias_model, x)</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">import tensorflow as tf

class BiasModel(tf.Module):
  def __init__(self):
    self.bias = tf.Variable(0.)
    self.weight = tf.Variable(2.)

@tf.function
def evaluate(model, x):
  return model.weight * x + model.bias

bias_model = BiasModel()
evaluate(bias_model, x)  # Compliant: mutations will be taken into account

bias_model.bias.assign_add(5.0)
evaluate(bias_model, x)</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>TensorFlow Documentation - <a href="https://www.tensorflow.org/guide/function#depending_on_python_objects">Depending on Python objects</a></p>
</li>
<li>
<p>TensorFlow Documentation - <a href="https://www.tensorflow.org/guide/function#rules_of_tracing">Rules of tracing</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>