== How to fix it in ASP.NET

=== Code examples

include::../../common/fix/code-rationale.adoc[]

==== Noncompliant code example

[source,vbnet,diff-id=1,diff-type=noncompliant]
----
Imports System.Web
Imports System.Web.Mvc

Public Class ExampleController
    Inherits Controller

    <HttpGet>
    Public Function CheckCookie(cookie As String) As IActionResult
        If Request.Cookies("ASP.NET_SessionId") Is Nothing Then
            Response.Cookies.Append("ASP.NET_SessionId", cookie)
        End If

        Return View("Welcome")
    End Function
End Class
----

==== Compliant solution

[source,vbnet,diff-id=1,diff-type=compliant]
----
Imports System.Web
Imports System.Web.Mvc

Public Class ExampleController
    Inherits Controller

    <HttpGet>
    Public Function CheckCookie() As IActionResult
        If Request.Cookies("ASP.NET_SessionId") Is Nothing Then
            Return View("GetCookie")
        End If

        Return View("Welcome")
    End Function
End Class
----

include::../../common/fix/how-does-this-work.adoc[]
