<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Full outer joins aren&#8217;t in common use, and as a result many developers don&#8217;t really understand them. Therefore, each use of this language feature should be reviewed.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-sql" data-lang="sql">BEGIN
  SELECT *
  BULK COLLECT INTO result
  FROM DUAL d1
  FULL OUTER JOIN DUAL d2 ON d1.dummy != d2.dummy; -- Noncompliant
END;
/</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Review this use of "FULL OUTER JOIN".</p>
</div>
</div>
</div>
</div>