<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Because statically-called programs must be relinked before they reflect changes in the code, it makes sense to prefer dynamic calls instead. Further, since statically-called programs are included in the caller&#8217;s load module, those modules could require more main storage than if the calls were dynamic, and the called programs could reside in memory multiple times - one for each caller.</p>
</div>
<div class="paragraph">
<p>While static calls are faster, their other disadvantages make dynamic calls the preferred method. Thus, this rule raises an issue when the program to <code>CALL</code> is hard-coded, rather than specified in a variable.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cobol" data-lang="cobol">CALL 'MYPRGM01' USING PARAM1.  *&gt; Noncompliant</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cobol" data-lang="cobol">77 PRGM-NAME      PIC X(8) VALUE 'MYPRGM01'.
[...]
CALL PRGM-NAME USING PARAM1.</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Use a variable to specify the program to "CALL".</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p><code>CALL 'xxxx'</code></p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_22_mar_2016_091531_pierre_yves_nicolas_wrote">on 22 Mar 2016, 09:15:31 Pierre-Yves Nicolas wrote:</h3>
<div class="paragraph">
<p>I think the title should not mention "arguments". The rule is only about the program which is called, not the arguments (defined in the <code>USING</code> clause).</p>
</div>
<div class="paragraph">
<p>See the <a href="http://www.ibm.com/support/knowledgecenter/SS6SG3_5.1.0/com.ibm.entcobol.doc_5.1/PGandLR/ref/rlpscall.html">IBM reference for CALL</a>.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_22_mar_2016_134103_ann_campbell_wrote">on 22 Mar 2016, 13:41:03 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>How&#8217;s this [~pierre-yves.nicolas]?</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_22_mar_2016_134902_pierre_yves_nicolas_wrote">on 22 Mar 2016, 13:49:02 Pierre-Yves Nicolas wrote:</h3>
<div class="paragraph">
<p>Looks good [~ann.campbell.2].</p>
</div>
</div>
</div>
</div>