== Why is this an issue?

Checker to find use of double quote where single quote is preferred.


=== Noncompliant code example

[source,html]
----
&lt;div id="header"&gt;&lt;/div&gt;
----


=== Compliant solution

[source,html]
----
&lt;div id='header'&gt;&lt;/div&gt;
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Use single quotes instead of double ones.


'''
== Comments And Links
(visible only on this page)

=== relates to: S1457

endif::env-github,rspecator-view[]
