Checker to find use of double quote where single quote is preferred.

== Noncompliant Code Example

----
&lt;div id="header"&gt;&lt;/div&gt;
----

== Compliant Solution

----
&lt;div id='header'&gt;&lt;/div&gt;
----
