Checker to find use of double quote where single quote is preferred.


== Noncompliant Code Example

[source,html]
----
&lt;div id="header"&gt;&lt;/div&gt;
----


== Compliant Solution

[source,html]
----
&lt;div id='header'&gt;&lt;/div&gt;
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
