== Why is this an issue?

Because the ``++equals++`` method takes a generic ``++Object++`` as a parameter, any type of object may be passed to it. The method should not assume it will only be used to test objects of its class type. It must instead check the parameter's type.


=== Noncompliant code example

[source,java]
----
public boolean equals(Object obj) { // Noncompliant
  MyClass mc = (MyClass)obj;
  // ...
}
----


=== Compliant solution

[source,java]
----
public boolean equals(Object obj) {
  if (obj == null)
    return false;

  if (this.getClass() != obj.getClass())
    return false;

  MyClass mc = (MyClass)obj;
  // ...
}
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Add a type test to this method.


'''
== Comments And Links
(visible only on this page)

=== on 3 Oct 2014, 20:09:10 Ann Campbell wrote:
\[~nicolas.peru] this will be covered by RSPEC-1944 if it includes downcasts.

=== on 12 Oct 2014, 19:00:05 Freddy Mallet wrote:
This rule should be linked to Findbugs rule BC_EQUALS_METHOD_SHOULD_WORK_FOR_ALL_OBJECTS

endif::env-github,rspecator-view[]
