Parameters that are declared as "INONLY" or "NONASSIGNABLE" are flagged as non-assignable by the compiler.


This rule raises an issue when a non-assignable parameter is written to.


== Noncompliant Code Example

----
sum: PROC(a, b);
   dcl a fixed dec(15,2) NONASSIGNABLE;
   dcl b fixed dec(15,2) NONASSIGNABLE;
   a = a + b; /* Noncompliant */
   return (a);
end;
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

include::highlighting.adoc[]

endif::env-github,rspecator-view[]
