include::../description.adoc[]

== Noncompliant Code Example

----
For Index As Integer = 1 To 42 ' Empty on purpose or missing piece of code?
Next
----

== Compliant Solution

----
For Index As Integer = 1 To 42
    Process(Index)
Next
----

include::../exceptions.adoc[]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
