Classes can register themselves to receive notifications using ``++NotificationCenter.add++``. Having done so, it seems suspicious that a class would opt to stop receiving notifications before de-initialization. For that reason, this rule raises an issue when ``++NotificationCenter.default.removeObserver(self)++`` is called anywhere but in ``++deinit++``


== Noncompliant Code Example

----
class MyClass {
  func doTheThing() {
    //...
    NotificationCenter.default.removeObserver(self)  // Noncompliant
  }
}
----


== Compliant Solution

----
class MyClass {
  func doTheThing() {
    //...
  }

  func deinit() {
    NotificationCenter.default.removeObserver(self)
  }
}
----


