include::../description.adoc[]

include::../ask-yourself.adoc[]

include::../recommended.adoc[]

== Noncompliant Code Example
In this example, an install script is downloaded using curl and then executed.

Several options are set to ensure secure execution of the HTTP request using curl. However, as the option `-L` is set, `\https://might-redirect.example.com/install.sh` might redirect curl to a location that uses HTTP. If this is the case, an attacker can change the install script that will be executed on the host system.

[source,docker]
----
FROM ubuntu:22.04

# Sensitive
RUN curl --tlsv1.2 -sSf -L https://might-redirect.example.com/install.sh | sh
----

== Compliant Solution
By adding the option `--proto "=https"` to the HTTP request, curl will only execute requests made using HTTPS.
If a redirect goes to an insecure (HTTP) location, the request will not be executed. This ensures that no tampering can occur.

[source,docker]
----
FROM ubuntu:22.04

RUN curl --proto "=https" --tlsv1.2 -sSf -L https://might-redirect.example.com/install.sh | sh
----

include::../see.adoc[]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::highlighting.adoc[]
