This rule raises an issue when S3 operations are performed without verifying bucket ownership using the ExpectedBucketOwner parameter.

== Why is this an issue?

Verifying S3 bucket ownership is a critical security practice in AWS applications. The ExpectedBucketOwner parameter confirms you're accessing the correct bucket owned by the intended AWS account.

Applications can accidentally interact with wrong buckets when ownership isn't verified. Configuration mistakes, naming overlaps, or security gaps all contribute to this risk.

Multi-account AWS environments amplify these dangers. Similar bucket names across accounts or dynamically generated names increase the chance of errors. The ExpectedBucketOwner parameter serves as a vital verification step.

=== What is the potential impact?

Failing to verify bucket ownership exposes systems to security threats. Applications may process data in unintended locations, including test environments or malicious buckets.

Data integrity suffers when operations target wrong buckets. Sensitive information could be exposed or corrupted without proper verification.

== How to fix it in boto3

Add the ExpectedBucketOwner parameter to your S3 operations to verify the bucket owner before performing any operations. This parameter should contain the AWS account ID that owns the bucket you expect to access.

=== Code examples

==== Noncompliant code example
[source,python,diff-id=1,diff-type=noncompliant]
----
import boto3

def lambda_handler(event, context):
    s3_client = boto3.client('s3')
    bucket_name = 'my-production-bucket'

    response = s3_client.get_object(  # Noncompliant
        Bucket=bucket_name,
        Key='data.json'
    )
----

==== Compliant solution
[source,python,diff-id=1,diff-type=compliant]
----
import boto3

def lambda_handler(event, context):
    s3_client = boto3.client('s3')
    bucket_name = 'my-production-bucket'
    expected_owner = '123456789012'

    response = s3_client.get_object(
        Bucket=bucket_name,
        Key='data.json',
        ExpectedBucketOwner=expected_owner  # Compliant
    )
----

== How to fix it in aiobotocore

When using aiobotocore, include the ExpectedBucketOwner parameter in your S3 operations to ensure bucket ownership verification.

=== Code examples

==== Noncompliant code example
[source,python,diff-id=2,diff-type=noncompliant]
----
import aiobotocore.session

async def lambda_handler(event, context):
    session = aiobotocore.session.get_session()
    async with session.create_client('s3') as s3_client:
        bucket_name = 'my-production-bucket'

        response = await s3_client.get_object(  # Noncompliant
            Bucket=bucket_name,
            Key='data.json'
        )
----

==== Compliant solution
[source,python,diff-id=2,diff-type=compliant]
----
import aiobotocore.session

async def lambda_handler(event, context):
    session = aiobotocore.session.get_session()
    async with session.create_client('s3') as s3_client:
        bucket_name = 'my-production-bucket'
        expected_owner = '123456789012'

        response = await s3_client.get_object(
            Bucket=bucket_name,
            Key='data.json',
            ExpectedBucketOwner=expected_owner  # Compliant
        )
----

== Resources

=== Documentation
* https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/s3.html#S3.Client.get_object[boto3 S3 get_object documentation with ExpectedBucketOwner parameter]
* https://docs.aws.amazon.com/AmazonS3/latest/userguide/bucket-owner-condition.html[AWS S3 bucket owner condition documentation]

ifdef::env-github,rspecator-view[]

== Implementation Specification
(visible only on this page)

=== Message

Add 'ExpectedBucketOwner' parameter to verify S3 bucket ownership.

=== Highlighting

* Primary location: the S3 client method call without ExpectedBucketOwner parameter

endif::env-github,rspecator-view[]