== Why is this an issue?

According to the {cpp} standard, ``++this++`` can never be null, so comparisons of the two are pointless at best. At worst, because of compiler optimizations, such comparisons could lead to null pointer dereferences or obscure, difficult-to-diagnose errors in production.


This rule raises an issue when ``++this++`` is compared to ``++nullptr++`` or 0 or anything ``++#defined++`` as ``++nullptr++`` or 0, such as ``++NULL++`` in most environments.


=== Noncompliant code example

[source,cpp]
----
class MyClass {
  string name;

  string GetName() {
    if (this != 0) {  // Noncompliant
      return name;
    }
    return 0;
  }
}
----


=== Compliant solution

[source,cpp]
----
class MyClass {
  string name;

  string GetName() {
    return name;
  }
}
----
ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Remove comparison of "this" with null.


'''
== Comments And Links
(visible only on this page)

=== on 8 Apr 2015, 17:03:24 Evgeny Mandrikov wrote:
\[~ann.campbell.2] in {cpp} compared to Java, notion of "null" exists in a bit different form(s) - see \http://en.cppreference.com/w/cpp/types/NULL

So actually wonder - whether we should use term (styled as code) "null", or "nullptr" or "0"? WDYT?

=== on 8 Apr 2015, 19:10:45 Ann Campbell wrote:
I've removed the code formatting on "null" but don't want to use 0 instead here because \"``++this++`` can never be 0" doesn't read as smoothly (i.e. takes a little more thought to understand IMO) than \"``++this++`` can never be null".

=== on 28 May 2015, 12:22:38 Evgeny Mandrikov wrote:
\[~ann.campbell.2] I changed title and message according to your last comment, added literal "nullptr" into description and did simplification of examples. Could you please check?

=== on 29 May 2015, 12:34:58 Ann Campbell wrote:
looks good [~evgeny.mandrikov]


endif::env-github,rspecator-view[]
