include::../description.adoc[]

== Noncompliant Code Example

[source,php]
----
preg_match("/(?=a)b/", $str); // Noncompliant, the same character can't be equal to 'a' and 'b' at the same time
----


== Compliant Solution

[source,php]
----
preg_match("/(?<=a)b/", $str);
preg_match("/a(?=b)/", $str);
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../highlighting.adoc[]

endif::env-github,rspecator-view[]
