On the principle that clearer code is better code, you should explicitly ``++import++`` the things you want to use in a module. Using ``++import *++`` imports everything in the module, and runs the risk of confusing maintainers. Similarly, ``++export * from "module";++`` imports and then re-exports everything in the module, and runs the risk of confusing not just maintainers but also users of the module.

== Noncompliant Code Example

----
import * as Imported from "aModule";  // Noncompliant
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
