<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when using system calls like <code>system "cd directory"</code> to change the working directory.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Directory changes made through system calls like <code>system "cd directory"</code> do not affect the current Ruby process. Each system call runs in a separate subprocess with its own environment. When the subprocess exits, any directory changes are lost, and the parent Ruby process remains in its original directory.</p>
</div>
<div class="paragraph">
<p>This leads to unexpected behavior where subsequent system calls or file operations don&#8217;t run in the intended directory. For example, if you run <code>system "cd my_app"</code> followed by <code>system "rails server"</code>, the Rails server will start in the original directory, not in the <code>my_app</code> directory.</p>
</div>
<div class="paragraph">
<p>Ruby provides the <code>Dir.chdir</code> method specifically for changing the working directory within the current process. This method actually changes the directory for the Ruby process itself, making it the correct approach for directory navigation in Ruby scripts.</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="paragraph">
<p>Using system calls for directory changes can cause scripts to fail silently or behave unexpectedly. Commands may run in the wrong directory, leading to file not found errors, incorrect file operations, or applications starting in unexpected locations. This can result in deployment failures, data corruption, or security issues if files are created or modified in unintended directories.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Replace system calls with <code>Dir.chdir</code> using the block form for scoped directory changes. The block form automatically reverts to the original directory after execution.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-ruby" data-lang="ruby">system "rails new my_app"
system "cd my_app"  # Noncompliant: has no effect
system "rails server &amp;"  # Still runs in parent directory</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-ruby" data-lang="ruby">system "rails new my_app"
Dir.chdir("my_app") do
  system "rails server &amp;"
  system "rake db:migrate"
end
# Directory automatically reverted after block</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Ruby Dir.chdir documentation - <a href="https://ruby-doc.org/core/Dir.html#method-c-chdir">Official Ruby documentation for the Dir.chdir method</a></p>
</li>
<li>
<p>Ruby system method documentation - <a href="https://ruby-doc.org/core/Kernel.html#method-i-system">Official Ruby documentation explaining how system calls work in subprocesses</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>