== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example
[source,bicep]
----
resource Microsoft_ApiManagement_service_apis_Raise_issue_because_protocols_contains_http 'Microsoft.ApiManagement/service/apis@2022-08-01' = {
  properties: {
    protocols: [
      // TODO change to secure protocol
      'http'
    ]
  }
}
----

include::../see.adoc[]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
