<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p><code>UNION</code> is a convenient syntax to combine the results of two or more SQL statements because it helps you cut a complex problem into multiple simple SQL statements. But when it comes to execution, using <code>UNION</code> is debatable.</p>
</div>
<div class="paragraph">
<p>First, it may be possible to fuse two simple SQL statements into a bigger one that will run faster. Second, <code>UNION</code> is significantly less performant compared to <code>UNION ALL</code> because it removes duplicated entries and runS an internal <code>DISTINCT</code> to achieve this.</p>
</div>
<div class="paragraph">
<p><code>UNION ALL</code> does not remove duplicates and returns all the results from the queries. It performs faster in most cases compared to <code>UNION</code>. Nevertheless, the quantity of data returned by <code>UNION ALL</code> can be significantly larger than with <code>UNION</code>. On a slow network, the performance gain of using <code>UNION ALL</code> instead of <code>UNION</code> can be negated by the time lost in the larger data transfer.</p>
</div>
<div class="paragraph">
<p>This rule raises an issue on each <code>UNION</code>. It&#8217;s up to the developer to challenge its use and see if there is a better way to rewrite without <code>UNION</code>.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-sql" data-lang="sql">-- case #1
SELECT EMAIL FROM EMPLOYEES WHERE COUNTRY = 'FR'
UNION                           -- Noncompliant
SELECT EMAIL FROM EMPLOYEES WHERE COUNTRY = 'CH'

-- case #2
-- if you care about not having duplicated entries, then UNION is the good choice
SELECT EMAIL FROM EMPLOYEES
UNION                           -- Noncompliant
SELECT EMAIL FROM CUSTOMERS</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-sql" data-lang="sql">-- case #1
SELECT EMAIL FROM EMPLOYEES WHERE COUNTRY = 'FR' OR COUNTRY = 'CH'

-- case #2
-- if you don't care about duplicated entries in the results of this UNION, then UNION ALL should be preferred
SELECT EMAIL FROM EMPLOYEES
UNION ALL
SELECT EMAIL FROM CUSTOMERS</code></pre>
</div>
</div>
</div>
</div>
</div>