There are valid cases for passing a variable multiple times into the same function or procedure call, but usually doing so is a mistake, and something else was intended for one of the arguments.

== Noncompliant Code Example

----
SET @result = dbo.MyAdd(@val1, @val1) -- Noncompliant
----

== Compliant Solution

----
SET @result = dbo.MyAdd(@val1, @val2)
----
ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../highlighting.adoc[]

endif::env-github,rspecator-view[]
