== Why is this an issue?

Sharing naming conventions allows teams to collaborate efficiently. This rule checks that type names match a provided regular expression.


=== Noncompliant code example

With the default regular expression:

[source,xml]
----
<xs:complexType name="fruit">    <!-- Noncompliant -->
  ...
</xs:complexType>
----


=== Compliant solution

[source,xml]
----
<xs:complexType name="fruitType">
  ...
</xs:complexType>
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Rename this type to match the regular expression: ${format}.


=== Parameters

.format
****

----
^[a-z]+([A-Z][a-z0-9]+)*Type$
----

Regular expression used to check types against
****


=== Highlighting

name value


endif::env-github,rspecator-view[]
