<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This is an issue when Active Record persistence methods like <code>save</code>, <code>create</code>, <code>update</code>, or <code>destroy</code> are called without using the exception-raising bang variants or without handling their return values.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Active Record persistence methods return boolean values to indicate success or failure. When these return values are ignored, validation errors and other persistence failures can occur silently, leading to data integrity issues and difficult-to-debug problems.</p>
</div>
<div class="paragraph">
<p>For example, if <code>user.save</code> fails due to validation errors, the method returns <code>false</code>, but the failure goes unnoticed if the return value isn&#8217;t checked. This can result in:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>Data not being persisted when expected</p>
</li>
<li>
<p>Silent failures that are hard to track down</p>
</li>
<li>
<p>Inconsistent application state</p>
</li>
<li>
<p>Poor user experience when operations appear successful but actually fail</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>The bang methods (<code>save!</code>, <code>create!</code>, etc.) raise exceptions when persistence fails, making errors immediately visible. Alternatively, explicitly checking return values allows for proper error handling and user feedback.</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="paragraph">
<p>Silent persistence failures can lead to data loss, inconsistent application state, and poor user experience. Users may believe their data was saved when it wasn&#8217;t, leading to frustration and potential business impact. These issues are often difficult to debug because the failures occur silently without any indication of what went wrong.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it_in_rails">How to fix it in Rails</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Use the exception-raising bang method instead of the regular method. This will raise an exception if the operation fails, making the error immediately visible.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-ruby" data-lang="ruby">user = User.new(name: 'John')
user.save # Noncompliant</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-ruby" data-lang="ruby">user = User.new(name: 'John')
user.save!</code></pre>
</div>
</div>
<div class="paragraph">
<p>Handle the return value of the persistence method explicitly. Check if the operation succeeded and handle failures appropriately.</p>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_code_examples_2">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example_2">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-ruby" data-lang="ruby">user = User.new(name: 'John')
user.save # Noncompliant</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution_2">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-ruby" data-lang="ruby">user = User.new(name: 'John')
if user.save
  # Handle success
  redirect_to user_path(user)
else
  # Handle failure
  render :new
end</code></pre>
</div>
</div>
<div class="paragraph">
<p>The same principle applies to other persistence methods like <code>create</code>, <code>update</code>, and <code>destroy</code>. Use bang methods or handle return values.</p>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_code_examples_3">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example_3">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-ruby" data-lang="ruby">User.create(name: 'John') # Noncompliant
user.update(email: 'new@example.com') # Noncompliant</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution_3">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-ruby" data-lang="ruby">User.create!(name: 'John')
# or
user = User.create(name: 'John')
if user.persisted?
  # Handle success
else
  # Handle failure
end</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Rails Style Guide - save! - <a href="https://rails.rubystyle.guide/#save-bang">Official Rails Style Guide recommendation for using bang methods</a></p>
</li>
<li>
<p>Active Record Validations and Callbacks - <a href="https://guides.rubyonrails.org/active_record_validations.html">Rails guide explaining validation and persistence behavior</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>