This rule raises an issue when hardcoded platform-specific temporary directory paths like `/tmp/` are used instead of cross-platform alternatives.

== Why is this an issue?

Hardcoding platform-specific paths like `/tmp/` creates portability issues because these paths don't exist on all operating systems. The `/tmp/` directory is common on Unix-like systems (Linux, macOS) but doesn't exist on Windows, where temporary files are typically stored in locations like `C:\Users\Username\AppData\Local\Temp\`.

When code with hardcoded Unix paths runs on Windows, it will fail at runtime because the operating system cannot find the specified directory. This leads to application crashes or unexpected behavior when the same codebase is deployed across different platforms.

Go provides built-in functions specifically designed to handle cross-platform path operations. The `os.TempDir()` function returns the appropriate temporary directory for the current operating system, while `filepath.Join()` ensures proper path separators are used regardless of the platform.

=== What is the potential impact?

Applications will fail at runtime when deployed on operating systems that don't have the hardcoded directory structure. This can cause crashes, prevent proper functionality, and create deployment issues across different environments.

== How to fix it

Replace hardcoded temporary directory paths with Go's cross-platform functions. Use `os.TempDir()` to get the system's temporary directory and `filepath.Join()` to construct paths properly.

=== Code examples

==== Noncompliant code example

[source,go,diff-id=1,diff-type=noncompliant]
----
portFile := fmt.Sprintf("/tmp/testnextnethttp_port_%d", pid) // Noncompliant
----

==== Compliant solution

[source,go,diff-id=1,diff-type=compliant]
----
portFile := filepath.Join(os.TempDir(), fmt.Sprintf("testnextnethttp_port_%d", pid))
----

== Resources

=== Documentation

 * Go os.TempDir() documentation - https://pkg.go.dev/os#TempDir[Official documentation for the os.TempDir() function]

 * Go filepath.Join() documentation - https://pkg.go.dev/path/filepath#Join[Official documentation for the filepath.Join() function]
