include::../description.adoc[]

include::../ask-yourself.adoc[]

include::../recommended.adoc[]

== Noncompliant Code Example

A customer managed policy for AWS that grants all permissions by using the wildcard (*) in the ``++Action++`` property:

----
resource "aws_iam_policy" "noncompliantpolicy" {
  name        = "noncompliantpolicy"

  policy = jsonencode({
    Version = "2012-10-17"
    Statement = [
      {
        Action = [
          "*" # Sensitive
        ]
        Effect   = "Allow"
        Resource = [
          aws_s3_bucket.mybucket.arn
        ]
      }
    ]
  })
}
----

A customer managed policy for GCP that grants all permissions by using the actions admin role ``++role++`` property:

----
resource "google_project_iam_binding" "noncompliantpolicy" {
  project = "example"
  role    = "roles/actions.Admin" # Sensitive

  members = [
    "user:jane@example.com",
  ]
}
----

== Compliant Solution

A customer managed policy for AWS that grants only the required permissions:

----
resource "aws_iam_policy" "compliantpolicy" {
  name        = "compliantpolicy"

  policy = jsonencode({
    Version = "2012-10-17"
    Statement = [
      {
        Action = [
          "s3:GetObject" 
        ]
        Effect   = "Allow"
        Resource = [
          aws_s3_bucket.mybucket.arn
        ]
      }
    ]
  })
}
----

A customer managed policy for GCP that grants restricted permissions by using the actions admin role ``++role++`` property:

----
resource "google_project_iam_binding" "noncompliantpolicy" {
  project = "example"
  role    = "roles/actions.Viewer"

  members = [
    "user:jane@example.com",
  ]
}
----

include::../see.adoc[]
