<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Unnecessary braces in string interpolation should be removed.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Dart <a href="https://dart.dev/language/built-in-types#strings">strings</a> support interpolation, which allows you to embed expressions within a string.</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-dart" data-lang="dart">String expression = 'words';
print('A string containing ${expression}');</code></pre>
</div>
</div>
<div class="paragraph">
<p>When the expression is a simple identifier, and the identifier is followed in the string interpolation by a character which is not allowed in an identifier (e.g. whitespace, <code>:</code>, <code>-</code> but not a letter, digit, <code>_</code>, &#8230;&#8203;), you can omit the curly braces.</p>
</div>
<div class="paragraph">
<p>This makes the code more readable and concise.</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-dart" data-lang="dart">String expression = 'words';
print('A string containing $expression');</code></pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Remove the curly braces.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-dart" data-lang="dart">String expression = 'words';
print('A string containing ${expression}');</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-dart" data-lang="dart">String expression = 'words';
print('A string containing $expression');</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Dart Docs - <a href="https://dart.dev/tools/linter-rules/unnecessary_brace_in_string_interps">Dart Linter rule - unnecessary_brace_in_string_interps</a></p>
</li>
<li>
<p>Dart Docs - <a href="https://dart.dev/language/built-in-types#strings">Language - Strings</a></p>
</li>
</ul>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Unnecessary braces in a string interpolation.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>The interpolation expression, including the <code>$</code> sign and the braces: e.g. <code>${expression}</code> in <code>'A string containing ${expression}'</code>.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
</div>
</div>