<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Complex chains of IF, ELSIF and ELSE statements should be replaced by the more readable CASE one. A complex IF statement has either several ELSIF clauses, or both an ELSIF and an ELSE clause.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-sql" data-lang="sql">DECLARE
  x PLS_INTEGER := 0;
BEGIN
  IF x = 0 THEN                     -- Noncompliant
    DBMS_OUTPUT.PUT_LINE('x = 0');
  ELSIF x = 1 THEN
    DBMS_OUTPUT.PUT_LINE('x = 1');
  ELSIF x = 2 THEN
    DBMS_OUTPUT.PUT_LINE('x = 2');
  END IF;
END;
/

DECLARE
  x PLS_INTEGER := 0;
  y PLS_INTEGER := 0;
BEGIN
  IF x = 0 THEN                     -- Noncompliant
    DBMS_OUTPUT.PUT_LINE('x = 0, y = ?');
  ELSIF y = 1 THEN
    DBMS_OUTPUT.PUT_LINE('x != 0, y = 1');
  ELSE
    DBMS_OUTPUT.PUT_LINE('x != 0, y != 1');
  END IF;
END;
/</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-sql" data-lang="sql">DECLARE
  x PLS_INTEGER := 0;
BEGIN
  CASE x
    WHEN 0 THEN
      DBMS_OUTPUT.PUT_LINE('x = 0');
    WHEN 1 THEN
      DBMS_OUTPUT.PUT_LINE('x = 1');
    WHEN 2 THEN
      DBMS_OUTPUT.PUT_LINE('x = 2');
    ELSE
      -- Do not forget the ELSE to prevent ORA-06592
      NULL;
  END CASE;
END;
/

DECLARE
  x PLS_INTEGER := 0;
  y PLS_INTEGER := 0;
BEGIN
  CASE                              -- Compliant
    WHEN x = 0 THEN
      DBMS_OUTPUT.PUT_LINE('x = 0, y = ?');
    WHEN y = 1 THEN
      DBMS_OUTPUT.PUT_LINE('x != 0, y = 1');
    ELSE
      DBMS_OUTPUT.PUT_LINE('x != 0, y != 1');
  END CASE;
END;
/</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_duplicates_s2145">duplicates: <a data-rspec-id="S2145" class="rspec-auto-link">S2145</a></h3>

</div>
<div class="sect2">
<h3 id="_on_30_may_2013_134823_fabrice_bellingard_wrote">on 30 May 2013, 13:48:23 Fabrice Bellingard wrote:</h3>
<div class="paragraph">
<p>This is originally a TOAD rule.</p>
</div>
<div class="paragraph">
<p>It could be improve to check only the first case which involves a single variable (because the gain in readability of the 2nd case is not really clear&#8230;&#8203;).</p>
</div>
</div>
</div>
</div>