== Why is this an issue?

``++switch++`` can contain a ``++default++`` clause for various reasons: to handle unexpected values, to show that all the cases were properly considered...

For readability purposes, to help a developer quickly spot the default behavior of a ``++switch++`` statement, it is recommended to put the ``++default++`` clause at the end of the ``++switch++`` statement. This rule raises an issue if the ``++default++`` clause is not the last one of the ``++switch++``'s cases.

=== Noncompliant code example

[source,java,diff-id=1,diff-type=noncompliant]
----
switch (param) {
  case 0:
    doSomething();
    break;
  default: // default clause should be the last one
    error();
    break;
  case 1:
    doSomethingElse();
    break;
}
----

=== Compliant solution

[source,java,diff-id=1,diff-type=compliant]
----
switch (param) {
  case 0:
    doSomething();
    break;
  case 1:
    doSomethingElse();
    break;
  default:
    error();
    break;
}
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
