Invoking other Lambdas from a Lambda is a scalability anti-pattern. As the runtime of your function is bounded, waiting for another Lambda to finish executing could cause a timeout.


Alternative solutions include:

* Generating events that can be consumed asynchronously by other Lambdas
* Making the Lambda invocation asynchronous


== Noncompliant Code Example

With AWS SDKv1

----
InvokeRequest invokeRequest = new InvokeRequest()
         .withFunctionName("myFunction");

AWSLambda awsLambda = AWSLambdaClientBuilder.standard()
		.withCredentials(new ProfileCredentialsProvider())
		.withRegion(Regions.US_WEST_2).build();

awsLambda.invoke(invokeRequest); // Noncompliant
----


== Compliant solution

----
InvokeRequest invokeRequest = new InvokeRequest()
         .withFunctionName("myFunction");

AWSLambdaAsync client = AWSLambdaAsyncClientBuilder.defaultClient();

client.invokeAsync(request); // Compliant
----


== See

* https://docs.aws.amazon.com/lambda/latest/dg/best-practices.html[Best practices for working with AWS Lambda functions]

