This rule raises an issue when the `addError()` method is called with generic, non-descriptive error messages like "Error", "Invalid", or "Failed".

== Why is this an issue?

Generic error messages provide no actionable information to users when validation fails. When a trigger or validation rule blocks a record save with a message like "Error", users are left confused about what went wrong and how to fix it.

Descriptive error messages are essential for good user experience. They should explain what validation failed and ideally provide guidance on how to resolve the issue. This reduces user frustration, decreases support tickets, and helps users complete their tasks successfully.

In Salesforce, error messages are often the primary way users understand why their data entry was rejected. A well-crafted error message can turn a frustrating experience into a learning opportunity that guides users toward the correct action.

=== What is the potential impact?

Generic error messages lead to poor user experience and increased support burden. Users may abandon their tasks, enter incorrect data repeatedly, or contact support for help understanding validation failures. This reduces productivity and increases operational costs.

== How to fix it

Replace generic error messages with specific, descriptive messages that explain what went wrong and how to resolve the issue. Include relevant field values or context to help users understand the problem.

=== Code examples

==== Noncompliant code example

[source,apex,diff-id=1,diff-type=noncompliant]
----
if (existing.containsKey(accountName)) {
    a.addError('Error'); // Noncompliant
}
----

==== Compliant solution

[source,apex,diff-id=1,diff-type=compliant]
----
if (existing.containsKey(accountName)) {
    a.addError('Duplicate Account found: ' + a.Name + '. Please check for existing records before creating new ones.');
}
----

== Resources

=== Documentation

 * Apex Trigger Exceptions - https://developer.salesforce.com/docs/atlas.en-us.apexcode.meta/apexcode/apex_triggers_exceptions.htm[Official Salesforce documentation on handling exceptions in Apex triggers]

 * SObject addError Method - https://developer.salesforce.com/docs/atlas.en-us.apexref.meta/apexref/apex_methods_system_sobject.htm#apex_System_SObject_addError[Reference documentation for the addError method in Apex]
