== Why is this an issue?

The ``++__exit__++`` method is invoked with four arguments: self, type, value and traceback. Leave one of these out of the method declaration and the result will be a ``++TypeError++`` at runtime.


=== Noncompliant code example

[source,python]
----
class MyClass:
   def __enter__(self):
       pass
   def __exit__(self, exc_type, exc_val):  # Noncompliant
       pass
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

*Add the missing argument.

*Remove the unnecessary argument.


'''
== Comments And Links
(visible only on this page)

=== on 18 Mar 2015, 17:47:56 Ann Campbell wrote:
Pylint E0235

endif::env-github,rspecator-view[]
