== Why is this an issue?

Return of boolean literal statements wrapped into ``++If-Then-Else++`` ones should be simplified.

=== Noncompliant code example

[source,vb6]
----
Function MyFunction
  If condition Then
    MyFunction = True
  Else
    MyFunction = False
  End If
End Function
----

=== Compliant solution

[source,vb6]
----
Function MyFunction
  MyFunction = condition
End Function
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
