== Why is this an issue?

``++DELETE FROM dbtab++`` without a ``++WHERE++`` condition deletes all the entries of the table. Check whether dataset to be deleted can be limited by a suitable WHERE condition.

=== Noncompliant code example

[source,abap]
----
DELETE FROM COUNTRIES.
----

=== Compliant solution

[source,abap]
----
DELETE FROM COUNTRIES WHERE CODE = country_code.
----
ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

endif::env-github,rspecator-view[]
