== Why is this an issue?

The main reason for using chained statements is to increase readability, but when used with operational statements, chaining can have the opposite effect. Even worse, it can lead to unexpected program behavior.


=== Noncompliant code example

[source,abap]
----
TRY. 
     ... 
  CATCH: cx_1, cx_2, cx_3.  " only cx_3 gets the following CATCH block
     "exception handling 
      ... 
ENDTRY.
----


=== Compliant solution

[source,abap]
----
TRY. 
     ... 
  CATCH cx_1.
     "exception handling 
  CATCH cx_2.
     "exception handling 
  CATCH cx_3. 
     "exception handling 
      ... 
ENDTRY.
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Write each of these n statements separately


'''
== Comments And Links
(visible only on this page)

=== on 2 Apr 2014, 20:28:34 Ann Campbell wrote:
@Freddy, I don't really understand the distinction between logic-related and instruction-related reliability

endif::env-github,rspecator-view[]
