== Why is this an issue?

The main reason for using chained statements is to increase readability, but when used with operational statements, chaining can have the opposite effect. Even worse, it can lead to unexpected program behavior.


=== Noncompliant code example

[source,abap]
----
TRY. 
     ... 
  CATCH: cx_1, cx_2, cx_3.  " only cx_3 gets the following CATCH block
     "exception handling 
      ... 
ENDTRY.
----


=== Compliant solution

[source,abap]
----
TRY. 
     ... 
  CATCH cx_1.
     "exception handling 
  CATCH cx_2.
     "exception handling 
  CATCH cx_3. 
     "exception handling 
      ... 
ENDTRY.
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
