<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when a semicolon is found at the end of a statement in Groovy code.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Groovy is designed to be more concise and readable than Java. One of its key features is that semicolons are optional at the end of statements.</p>
</div>
<div class="paragraph">
<p>Using semicolons in Groovy code goes against the language&#8217;s conventions and style guidelines. The official Groovy style guide explicitly recommends omitting semicolons to make code more idiomatic and cleaner.</p>
</div>
<div class="paragraph">
<p>When developers come from Java or other C-style languages, they often carry over the habit of using semicolons everywhere. However, in Groovy, this creates unnecessary visual noise and makes the code less idiomatic.</p>
</div>
<div class="paragraph">
<p>Removing semicolons helps achieve:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>More readable and cleaner code</p>
</li>
<li>
<p>Better adherence to Groovy conventions</p>
</li>
<li>
<p>Consistency with idiomatic Groovy style</p>
</li>
<li>
<p>Reduced visual clutter</p>
</li>
</ul>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="paragraph">
<p>The impact is primarily on code maintainability and readability. While semicolons don&#8217;t cause functional issues, they make the code less idiomatic and can indicate that the developer is not following Groovy best practices.</p>
</div>
<div class="paragraph">
<p>Code with unnecessary semicolons may:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>Appear less professional to Groovy developers</p>
</li>
<li>
<p>Create inconsistency in codebases where some files follow Groovy conventions and others don&#8217;t</p>
</li>
<li>
<p>Make the code harder to read due to visual noise</p>
</li>
</ul>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Remove the semicolon at the end of variable declarations and assignments.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-groovy" data-lang="groovy">int count = 5; // Noncompliant</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-groovy" data-lang="groovy">int count = 5</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Groovy Style Guide - <a href="https://groovy-lang.org/style-guide.html#_no_semicolons">No semicolons</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>