include::../common/description.adoc[]

include::../common/impact.adoc[]

include::../common/fix.adoc[]

=== Code examples

==== Noncompliant code example

[source,javascript,diff-id=1,diff-type=noncompliant]
----
const crypto = require("crypto");

const hash = crypto.createHash('sha1');  // Noncompliant 
----

==== Compliant solution

[source,javascript,diff-id=1,diff-type=compliant]
----
const crypto = require("crypto");

const hash = crypto.createHash('sha512');
----

include::../common/fix-pitfalls.adoc[]

include::../common/resources.adoc[]


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../common/message.adoc[]

include::../common/highlighting.adoc[]

'''
== Comments And Links
(visible only on this page)

=== on 6 Dec 2018, 21:32:23 Lars Svensson wrote:
https://nodejs.org/api/crypto.html

https://developer.mozilla.org/en-US/docs/Web/API/SubtleCrypto/digest



include::../common/comments-and-links.adoc[]

endif::env-github,rspecator-view[]
