<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when a variable is declared with an initial value and then immediately reassigned without the original value being used.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>When a variable is initialized and then immediately reassigned without using the original value, the first initialization becomes redundant code.</p>
</div>
<div class="paragraph">
<p>This pattern reduces code readability and can confuse other developers who might wonder why the variable was initialized twice. It also suggests that the code may have been refactored without proper cleanup, or that the developer was uncertain about the correct initial value.</p>
</div>
<div class="paragraph">
<p>In Go, this commonly happens with context variables and HTTP clients where developers first assign a default value and then immediately replace it with a more specific configuration.</p>
</div>
<div class="paragraph">
<p>Removing the redundant initialization makes the code more direct and easier to understand. It also eliminates any confusion about which value the variable actually holds.</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="paragraph">
<p>This issue affects code maintainability by making the code less readable and potentially confusing to other developers. While it doesn&#8217;t cause runtime errors, it can slow down code reviews and make the codebase harder to maintain.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Remove the redundant initialization and use only the final assignment. If the variable needs to be declared in a specific scope, use the short variable declaration syntax with the final value directly.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-go" data-lang="go">ctx := context.Background()
ctx, cancel := context.WithTimeout(context.Background(), 60*time.Second) // Noncompliant</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-go" data-lang="go">ctx, cancel := context.WithTimeout(context.Background(), 60*time.Second)</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Go Language Specification - Variable declarations - <a href="https://golang.org/ref/spec#Variable_declarations">Official Go documentation on variable declarations and initialization</a></p>
</li>
<li>
<p>Effective Go - Variables - <a href="https://golang.org/doc/effective_go#variables">Best practices for variable usage in Go</a></p>
</li>
</ul>
</div>
</div>
<div class="sect2">
<h3 id="_related_rules">Related rules</h3>
<div class="ulist">
<ul>
<li>
<p><a data-rspec-id="S1854" class="rspec-auto-link">RSPEC-1854</a> - <a href="https://rules.sonarsource.com/javascript/RSPEC-1854/">Unused assignments should be removed (JavaScript)</a></p>
</li>
<li>
<p><a data-rspec-id="S1854" class="rspec-auto-link">RSPEC-1854</a> - <a href="https://rules.sonarsource.com/python/RSPEC-1854/">Unused assignments should be removed (Python)</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>