Don't test if the variable is null before a switch


== Why is this an issue?

Since Java 21, can accept null values. Thus, it not necessary anymore to check if a variable is null before matching it and should even be avoided, as it lead to less readable code.


== How to fix it

Move the null check inside the switch statement.

=== Code examples

==== Noncompliant code example

[source,java,diff-id=1,diff-type=noncompliant]
----
static void f(String s) {
  if (s == null) { // Noncompliant, we can move the null check in the switch statement.
    System.out.println("Oops!");
    return;
  }
  switch (s) {
    case "Foo", "Bar" -> System.out.println("Great");
    default           -> System.out.println("Ok");
   }
}
----

==== Compliant solution

[source,java,diff-id=1,diff-type=compliant]
----
static void f(String s) {
  switch (s) {
    case null         -> System.out.println("Oops!");
    case "Foo", "Bar" -> System.out.println("Great");
    default           -> System.out.println("Ok");
   }
}
----



== Resources
* https://openjdk.org/jeps/441[Pattern Matching for Switch]
