include::../description.adoc[]


== Noncompliant Code Example

----
val fos = FileOutputStream(fileName, true) // fos opened in append mode
val out = ObjectOutputStream(fos) // Noncompliant
----


== Compliant Solution

----
val fos = FileOutputStream(fileName)
val out = ObjectOutputStream(fos)
----


