== Why is this an issue?

include::description.adoc[]

=== Noncompliant code example

[source,text]
----
<<foo>> -- Noncompliant
BEGIN
  DBMS_OUTPUT.PUT_LINE('Hello, world!');
END;
----


=== Compliant solution

[source,text]
----
-- Compliant
BEGIN
  DBMS_OUTPUT.PUT_LINE('Hello, world!');
END;
----


