<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>If a label is declared but not used in the program, it can be considered as dead code and should therefore be removed.</p>
</div>
<div class="paragraph">
<p>This will improve maintainability as developers will not wonder what this label is used for.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-sql" data-lang="sql">label: -- Noncompliant
  PRINT 'hello world';
GO</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-sql" data-lang="sql">PRINT 'hello world';
GO</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove this unused label.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_22_may_2013_102652_fabrice_bellingard_wrote">on 22 May 2013, 10:26:52 Fabrice Bellingard wrote:</h3>
<div class="paragraph">
<p>Implementation: http://jira.sonarsource.com/browse/PLSQL-346</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_4_jun_2013_161138_dinesh_bolkensteyn_wrote">on 4 Jun 2013, 16:11:38 Dinesh Bolkensteyn wrote:</h3>
<div class="paragraph">
<p>Crappy &lt; and &gt; which needs to be enscaped in the HTML doc.</p>
</div>
<div class="paragraph">
<p>This totally sucks!</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_17_nov_2014_002203_evgeny_mandrikov_wrote">on 17 Nov 2014, 00:22:03 Evgeny Mandrikov wrote:</h3>
<div class="paragraph">
<p>Implementation: http://jira.codehaus.org/browse/SONARJAVA-765</p>
</div>
</div>
</div>
</div>