== Why is this an issue?

include::../description_without_reference_to_s5998.adoc[]

=== Noncompliant code example

[source,php]
----
preg_match("/a|b|c/", $str); // Noncompliant
----

=== Compliant solution

[source,php]
----
preg_match("/[abc]/", $str);
// or
preg_match("/[a-c]/", $str);
----
