include::../description_without_reference_to_s5998.adoc[]

== Noncompliant Code Example

[source,php]
----
preg_match("/a|b|c/", $str); // Noncompliant
----

== Compliant Solution

[source,php]
----
preg_match("/[abc]/", $str);
// or
preg_match("/[a-c]/", $str);
----
