include::../description_without_reference_to_s5998.adoc[]

== Noncompliant Code Example

----
preg_match("/a|b|c/", $str); // Noncompliant
----

== Compliant Solution

----
preg_match("/[abc]/", $str);
// or
preg_match("/[a-c]/", $str);
----
