== Why is this an issue?

A test case without assertions ensures only that no exceptions are thrown. Beyond basic runnability, it ensures nothing about the behavior of the code under test.


This rule raised an issue when no assertions are found within a PHPUnit test method. 

=== Noncompliant code example

[source,php]
----
public function testDoSomething() {  // Compliant
  $myClass = new MyClass();
  $myClass->getSomething();
}
----

=== Compliant solution

[source,php]
----
public function testDoSomething() {  // Noncompliant
  $myClass = new MyClass();
  $this->assertEquals("foo", $myClass->getSomething());
}
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
