<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>While Dockerfile instructions are not case-sensitive, adhering to uppercase conventions for instructions helps enhance clarity and collaboration within development teams.
This ensures that instructions are more easily distinguishable from arguments and contributes to effective collaboration.</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="paragraph">
<p>In the world of programming, even small details can have significant repercussions.
The inconsistent use of uppercase letters in Dockerfile instructions may seem inconsequential at first glance, but it can lead to confusion, misinterpretation, and hinder smooth teamwork.</p>
</div>
<div class="paragraph">
<p>When Dockerfile instructions are written in a mix of lowercase and uppercase, it becomes challenging to quickly identify and differentiate instructions from arguments.
This lack of visual uniformity can make it harder for developers to understand the structure and purpose of the Dockerfile.
Teams may spend unnecessary time deciphering the code, which could be better utilized in productive tasks.</p>
</div>
<div class="paragraph">
<p>Inconsistencies in letter casing can lead to errors that are difficult to diagnose.
Imagine spending hours trying to locate a bug only to realize that it resulted from a misinterpretation of an instruction due to inconsistent casing.
Such situations can be frustrating and can slow down the development process.</p>
</div>
<div class="paragraph">
<p>Additionally, modern IDEs provide syntax highlighting to help programmers navigate and comprehend code.
However, IDEs can sometimes fail to highlight lowercase or mixed-case Dockerfile keywords properly, as they typically expect instructions to be in uppercase.</p>
</div>
<div class="paragraph">
<p>In essence, the inconsistent use of uppercase letters in Dockerfile instructions might seem like a minor concern, but its impact reaches beyond aesthetics.
It affects comprehension, teamwork, error detection, and the overall development lifecycle.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Ensure that all Dockerfile instructions are written in uppercase letters.
For instance, use FROM instead of from, RUN instead of run, and so on.
By adopting this convention, you create a visual distinction between instructions and arguments.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-dockerfile" data-lang="dockerfile">from ubuntu:22.04 aS jammy</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-dockerfile" data-lang="dockerfile">FROM ubuntu:22.04 AS jammy</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Dockerfile reference <a href="https://docs.docker.com/engine/reference/builder/#format">Format</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>