In Dockerfiles, when using multi-line `RUN` instructions, it's important to ensure that the arguments are sorted alphabetically. This practice enhances the readability and maintainability of the code. It allows for easier tracking of modifications and can help prevent potential errors.

== Why is this an issue?

When arguments in `RUN` instructions are not sorted alphabetically, it can make the Dockerfile harder to read and maintain. However, when arguments are sorted, it is easier to track changes in version control systems and to locate specific arguments.

== How to fix it

=== Code examples

==== Noncompliant code example

[source,docker,diff-id=1,diff-type=noncompliant]
----
FROM ubuntu:20.04

RUN apt-get update && apt-get install -y \
    unzip \
    wget \
    curl \
    git \
    zip
----

==== Compliant solution

[source,docker,diff-id=1,diff-type=compliant]
----
FROM ubuntu:20.04

RUN apt-get update && apt-get install -y \
    curl \
    git \
    unzip \
    wget \
    zip
----

== Resources

=== Documentation

* Docker Docs - https://docs.docker.com/build/building/best-practices/#sort-multi-line-arguments[Building best practices]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Sort these package names alphanumerically.

=== Highlighting

Highlight the part of the RUN command that contains the list of packages.

'''
== Comments And Links
(visible only on this page)

endif::env-github,rspecator-view[]
