== Why is this an issue?

Using the value of a pointer to a ``++FILE++`` object after the associated file is closed is undefined behavior.


=== Noncompliant code example

[source,cpp]
----
void fun() {
  FILE * pFile;
  pFile = fopen(fileName, "w");

  if (condition) {
    fclose(pFile);
    // ...
  }

  fclose(pFile); // Noncompliant, the file has already been closed
}
----


=== Compliant solution

[source,cpp]
----
void fun() {
  FILE * pFile;
  pFile = fopen(fileName, "w");

  if (condition) {
    // ...
  }

  fclose(pFile);
}
----


== Resources

* https://wiki.sei.cmu.edu/confluence/x/QdUxBQ[CERT, FIO46-C.] - Do not access a closed file


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

File "xxx" has already been closed.


=== Highlighting

primary: ``++FILE++`` access

secondary: where the ``++FILE++`` was closed


'''
== Comments And Links
(visible only on this page)

=== relates to: S5485

endif::env-github,rspecator-view[]
