This rule raises an issue when AWS resource names or regions are hard-coded as string literals in `boto3` or `aiobotocore` client calls.

== Why is this an issue?

Hard-coding AWS resource identifiers like S3 bucket names, DynamoDB table names, or regions directly into Lambda function code creates tight coupling between your code and specific deployment environments.
This practice makes your functions brittle and difficult to manage across different environments.

Lambda functions are designed to be portable and scalable across multiple environments, e.g: development, staging, production, and AWS regions.
When resource names are hard-coded, the same function cannot be easily deployed to different environments without code modifications.

=== What is the potential impact?

Hard-coded resource names lead to deployment failures when moving between environments, increase maintenance overhead requiring code changes for each deployment, and make CI/CD automation complex.
They also expose internal infrastructure details and make it harder to implement least privilege access controls when resource names cannot be dynamically determined.

== How to fix it

Replace hard-coded resource names and regions with environment variables or configuration parameters.
Use `os.environ` to retrieve these values at runtime, allowing the same code to work across different environments without modification.

=== Code examples

==== Noncompliant code example

[source,python,diff-id=1,diff-type=noncompliant]
----
import boto3

s3 = boto3.client("s3", region_name="us-east-1")

def lambda_handler(event, context):
    return s3.get_object(Bucket="my-dev-bucket", Key="file.txt")  # Noncompliant
----

==== Compliant solution

[source,python,diff-id=1,diff-type=compliant]
----
import os
import boto3

region = os.environ.get("AWS_REGION", "us-east-1")
bucket_name = os.environ["MY_BUCKET_NAME"]
s3 = boto3.client("s3", region_name=region)

def lambda_handler(event, context):
    return s3.get_object(Bucket=bucket_name, Key="file.txt")
----

== Resources

=== Documentation

* AWS Documentation - https://docs.aws.amazon.com/lambda/latest/dg/configuration-envvars.html[AWS Lambda environment variables documentation]
* AWS Documentation - https://docs.aws.amazon.com/wellarchitected/latest/operational-excellence-pillar/welcome.html[AWS Well-Architected Operational Excellence Pillar]
* AWS Documentation - https://docs.aws.amazon.com/systems-manager/latest/userguide/systems-manager-parameter-store.html[AWS Systems Manager Parameter Store]


ifdef::env-github,rspecator-view[]

== Implementation Specification

(visible only on this page)

=== Message

Replace this hard-coded resource identifier with an environment variable or configuration parameter.

=== Highlighting

* Primary location: the string literal containing the hard-coded resource name or region
* Secondary location: the entire method call containing the hard-coded parameter

endif::env-github,rspecator-view[]