<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>There are several reasons that a class might have a method that throws an <code>UnsupportedOperationException</code>. The method may be required by an interface or an abstract superclass but not actually needed in the class. Or it may be that the class itself is intended as a superclass, and the method may optionally be implemented by subclasses but not invoked on the superclass. Finally, it could be that the method has been stubbed into the code but not implemented yet.</p>
</div>
<div class="paragraph">
<p>Whatever the reason, methods that throw <code>UnsupportedOperationException</code> should not be called.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">public class Callee {

  public int doTheThing() {
    throw new UnsupportedOperationException("Not implemented");
  }
}

public class Caller {
  public void accomplishStuff() {
    //...

    Callee callee = new Callee();
    callee.doTheThing();  // Noncompliant</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove this call to "xxx"; it will return an "UnsupportedOperationException".</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_22_oct_2014_191157_nicolas_peru_wrote">on 22 Oct 2014, 19:11:57 Nicolas Peru wrote:</h3>
<div class="paragraph">
<p>Scope of the rule is really important here:</p>
</div>
<div class="paragraph">
<p>We don&#8217;t have interprocedural analysis so, we can&#8217;t actually analyze what is the implementation of a remote method call.</p>
</div>
<div class="paragraph">
<p>So what&#8217;s left : implementation of call to methods declared within the same class, but then you would raise issue exactly because you are calling a method that is supposed to be overidden in inherited classes.</p>
</div>
<div class="paragraph">
<p>So I don&#8217;t see how we can actually deal with this rule.</p>
</div>
</div>
</div>
</div>