<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Nullability of types is an important part of Dart&#8217;s type system. In order to simplify the work with nullable types in Dart language there are many null-aware operators (<code>??</code>, <code>??=</code>, <code>?.</code>, <code>!</code>, <code>?..</code>, <code>?[]</code>, <code>&#8230;&#8203;?</code>). Using those operators, will make your code more clear and concise.</p>
</div>
<div class="paragraph">
<p>Null-aware assignment (<code>??=</code>), will check if the value on the left side is <code>null</code>, and, if yes, will perform the assignment. This is an easy way to set default values instead of dealing with <code>null</code>. Thus, this operator doesn&#8217;t make sense if used with <code>null</code> on the right side. For example, <code>x ??= null</code> will be reported as an issue.</p>
</div>
<div class="sect2">
<h3 id="_exceptions">Exceptions</h3>
<div class="paragraph">
<p>The rule doesn&#8217;t apply if property access is used. For example:</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-dart" data-lang="dart">class Person {
  String? name;
  int? age;

  void g() {
    name ??= null; // Ok, field
    age ??= null; // Ok, field
  }
}</code></pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Remove the assignment or change the new value to not be <code>null</code>.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-dart" data-lang="dart">void f() {
  var x;
  x ??= null; // Noncompliant
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-dart" data-lang="dart">void f() {
  var x;
  x ??= 0;
}</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Dart Docs - <a href="https://dart.dev/tools/linter-rules/unnecessary_null_aware_assignments">Dart Linter rule - unnecessary_null_aware_assignments</a></p>
</li>
</ul>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Unnecessary assignment of 'null'.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>The whole assignment statement</p>
</div>
</div>
</div>
</div>