<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>An iframe, or inline frame, is an HTML document embedded inside another HTML document on a website. The iframe HTML element is often used to insert content from another source, such as an advertisement, into a web page.</p>
</div>
<div class="paragraph">
<p>In the context of web accessibility, <code>&lt;iframe&gt;</code>'s should have a <code>title</code> attribute. This is because screen readers for the visually impaired use this title to help users understand the content of the iframe.</p>
</div>
<div class="paragraph">
<p>Without a title, it can be difficult for these users to understand the context or purpose of the iframe&#8217;s content.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>To fix missing iframe titles, you simply need to add a <code>title</code> attribute to your <code>&lt;iframe&gt;</code> element. The value of this attribute should be a brief description of the iframe&#8217;s content.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-javascript" data-lang="javascript">function iframe() {
    return (
        &lt;iframe src="https://openweathermap.org"&gt;&lt;/iframe&gt; // Noncompliant
    );
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-javascript" data-lang="javascript">function iframe() {
    return (
        &lt;iframe src="https://openweathermap.org" title="Weather forecasts, nowcasts and history"&gt;&lt;/iframe&gt;
    );
}</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p><a href="https://github.com/jsx-eslint/eslint-plugin-jsx-a11y">eslint-plugin-jsx-a11y</a> - Rule <a href="https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/blob/HEAD/docs/rules/iframe-has-title.md">iframe-has-title</a></p>
</li>
<li>
<p>MDN web docs - <a href="https://developer.mozilla.org/en-US/docs/Web/HTML/Element/iframe">iframe element</a></p>
</li>
<li>
<p>WCAG - <a href="https://www.w3.org/WAI/WCAG21/Understanding/bypass-blocks">Bypass Blocks</a></p>
</li>
<li>
<p>WCAG - <a href="https://www.w3.org/WAI/WCAG21/Understanding/name-role-value">Name, Role, Value</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>