== Why is this an issue?

The ``++ACCEPT++`` keyword does no editing or error checking of the data it stores, therefore its use can be dangerous. For this reason, ``++ACCEPT++`` should be avoided. 


=== Noncompliant code example

[source,cobol]
----
 01 USER-INPUT PIC X(4). 

  GET-USER-INPUT. 
       MOVE 'N' TO WS-NUMERIC. 
       PERFORM UNTIL WS-NUMERIC = 'Y' 
           DISPLAY 'ENTER YOUR 4 DIGIT RECORD NUMBER: ' NO ADVANCING 
           ACCEPT USER-RECORD *> Noncompliant 
----


=== Exceptions

This rule ignores uses of ``++ACCEPT FROM++`` with date/time-related inputs.


== Resources

* https://owasp.org/Top10/A03_2021-Injection/[OWASP Top 10 2021 Category A3] - Injection
* https://owasp.org/www-project-top-ten/2017/A1_2017-Injection[OWASP Top 10 2017 Category A1] - Injection


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

include::highlighting.adoc[]

'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
