== Why is this an issue?

The ``++ACCEPT++`` keyword does no editing or error checking of the data it stores, therefore its use can be dangerous. For this reason, ``++ACCEPT++`` should be avoided. 


=== Noncompliant code example

[source,cobol]
----
 01 USER-INPUT PIC X(4). 

  GET-USER-INPUT. 
       MOVE 'N' TO WS-NUMERIC. 
       PERFORM UNTIL WS-NUMERIC = 'Y' 
           DISPLAY 'ENTER YOUR 4 DIGIT RECORD NUMBER: ' NO ADVANCING 
           ACCEPT USER-RECORD *> Noncompliant 
----


=== Exceptions

This rule ignores uses of ``++ACCEPT FROM++`` with date/time-related inputs.


== Resources

* https://owasp.org/Top10/A03_2021-Injection/[OWASP Top 10 2021 Category A3] - Injection
* https://owasp.org/www-project-top-ten/2017/A1_2017-Injection[OWASP Top 10 2017 Category A1] - Injection


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Remove this use of "ACCEPT".


=== Highlighting

the ``++ACCEPT++`` statement - so both ``++ACCEPT++`` and what the input is being accepted into.


'''
== Comments And Links
(visible only on this page)

=== on 3 Nov 2015, 20:23:13 Ann Campbell wrote:
\[~pierre-yves.nicolas], I wrote the exception based on http://supportline.microfocus.com/Documentation/AcucorpProducts/docs/v6_online_doc/gtman3/gt3678.htm[this] but suspect it should be expanded to include more cases.

=== on 5 Nov 2015, 12:05:30 Pierre-Yves Nicolas wrote:
\[~ann.campbell.2] The documentation you found is specific to AcuCobol which adds many extensions on top of "standard" COBOL. I suggest to keep only the exception about date/time and to not activate that rule by default.

=== on 5 Nov 2015, 14:28:13 Ann Campbell wrote:
Okay [~pierre-yves.nicolas]. Done

=== on 21 Mar 2017, 17:54:00 Pierre-Yves Nicolas wrote:
\[~ann.campbell.2] The scope of this rule changed completely: it used to be "ACCEPT should not be used" and is now "Track uses of forbidden statements". I don't see how we can keep the exception for ``++ACCEPT FROM++`` with the new scope.

=== on 22 Mar 2017, 09:04:25 Ann Campbell wrote:
\[~pierre-yves.nicolas] I've reverted this rule & created RSPEC-3938

endif::env-github,rspecator-view[]
