This rule raises an issue when `DateTimeFormatter.ofPattern()` is called repeatedly instead of storing the formatter in a static constant.

== Why is this an issue?

Creating `DateTimeFormatter` instances repeatedly using `DateTimeFormatter.ofPattern()` is inefficient. Each call to `ofPattern()` involves parsing and compiling the pattern string, which has computational overhead.

Since `DateTimeFormatter` instances are immutable and thread-safe, they can be safely reused. Store them in `static final` constants to avoid repeated creation and improve performance.

=== What is the potential impact?

Repeated calls to `DateTimeFormatter.ofPattern()` can degrade application performance, especially in high-throughput scenarios. The pattern parsing and compilation overhead accumulates over time, leading to increased CPU usage and memory pressure.

== How to fix it

Move the DateTimeFormatter creation to a static final constant. This ensures the formatter is created only once when the class is loaded and can be reused safely across all instances and threads.

=== Code examples

==== Noncompliant code example

[source,java,diff-id=1,diff-type=noncompliant]
----
public class DateService {
    public String formatDate(LocalDateTime dateTime) {
        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss"); // Noncompliant
        return formatter.format(dateTime);
    }
}
----

==== Compliant solution

[source,java,diff-id=1,diff-type=compliant]
----
public class DateService {
    private static final DateTimeFormatter FORMATTER = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss");
    
    public String formatDate(LocalDateTime dateTime) {
        return FORMATTER.format(dateTime);
    }
}
----

== Resources

=== Documentation

 * DateTimeFormatter (Java Platform SE 8) - https://docs.oracle.com/javase/8/docs/api/java/time/format/DateTimeFormatter.html[Official Java documentation for DateTimeFormatter, explaining its immutable and thread-safe nature]

=== Standards

 * Java Performance Best Practices - https://www.oracle.com/java/technologies/javase/performance-best-practices.html[Oracle's guidelines for Java performance optimization, including object reuse patterns]
