React expects a unique identifier for performance optimizations. An array index is not a stable identifier most of the time. This results in unnecessary renders when the array items change index following some mutation. When components have state, this might also provoke bugs that are hard to diagnose.
We recommend using an explicit identifier to avoid misuse and accidental re-renders. If there is no unique attribute available, consider concatenating properties to create one. Using generated values like `Math.random()` or `Date.now()` is discouraged as it would force a re-render every time and might also provoke bugs if values collide.

React expects a unique identifier `key` in each component to optimize rendering. Missing keys will negatively impact performance and can bring your application to a halt when there are enough elements.

Avoid Array indexes since they are not stable. Instead, use a unique attribute like an id.

== Noncompliant Code Example

[source,javascript]
----
function generateButtons(buttons) {
  return buttons.map((button, index) => {
    <Button key={index}>button.number</Button>
  });
}
----

== Compliant Solution

[source,javascript]
----
function generateButtons(buttons) {
  return buttons.map((button, index) => {
    <Button key={button.number}>button.number</Button>
  });
}
----

== See

* https://reactjs.org/docs/reconciliation.html#recursing-on-children - React API reference
