<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when the region configuration of the AWS client is hardcoded.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Hard-coding configuration values directly into your function code is an anti-pattern. This practice couples your code tightly to specific deployment environments and makes it brittle, difficult to manage, and prone to errors.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>The solution involves externalizing these configurations using environment variables.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">import boto3

s3 = boto3.client("s3", region_name = "us-east-1") # Noncompliant region_name is hardcoded</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">import os
import boto3

region = os.environ.get("AWS_REGION", "us-east-1")
s3 = boto3.client("s3", region_name = region) # Compliant region_name is set through an environment variable</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>AWS docs : <a href="https://docs.aws.amazon.com/wellarchitected/latest/operational-excellence-pillar/welcome.html">Operational Excellence Pillar</a></p>
</li>
<li>
<p>Stackoverflow : <a href="https://stackoverflow.com/questions/63538531/aws-lambda-how-to-get-environment-variables">Environment variables in AWS</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>