include::../rule.adoc[]

== Noncompliant Code Example

For the https://docs.mongodb.com/drivers/java/sync/current/[MongoDB Java Driver]:

----
public void exampleFunction(@RequestParam String input)
{
    try
    {
        MongoClient mongoClient = new MongoClient();
        DB database             = mongoClient.getDB("ExampleDatabase");
        DBCollection collection = database.getCollection("exampleCollection");
    }

    BasicDBObject query = new BasicDBObject();

    query.put("$where", "this.field == \"" + input + "\""); // Non-compliant
    collection.find(query);
}
----

== Compliant Solution

For the https://docs.mongodb.com/drivers/java/sync/current/[MongoDB Java Driver]:

----
public void exampleFunction(@RequestParam String input)
{
    try
    {
        MongoClient mongoClient = new MongoClient();
        DB database             = mongoClient.getDB("ExampleDatabase");
        DBCollection collection = database.getCollection("exampleCollection");
    }

    BasicDBObject query = new BasicDBObject();
    query.put("field", input);

    DBCursor cursor = collection.find(query);
    while(cursor.hasNext())
    {
        doSomething();
    }
}
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../highlighting.adoc[]

endif::env-github,rspecator-view[]
