<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when the built-in <code>input()</code> function is called in an asynchronous function.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>In Python&#8217;s asynchronous programming (using <code>asyncio</code>, <code>Trio</code>, or <code>AnyIO</code>), an event loop manages concurrent tasks by having them yield control during time-consuming operations, enabling other tasks to run.</p>
</div>
<div class="paragraph">
<p>However, the synchronous <code>input()</code> function blocks the current thread until user input is received, and when called from a coroutine, it blocks the entire event loop, preventing other tasks from executing and making the application unresponsive - effectively defeating the purpose of asynchronous programming for applications requiring concurrent operations or user interaction.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it_in_asyncio">How to fix it in Asyncio</h2>
<div class="sectionbody">
<div class="paragraph">
<p>You can use <code>asyncio.to_thread()</code> to run the <code>input()</code> function in a separate thread.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">import asyncio

async def get_name():
    print("Please enter your name:")
    name = input() # Noncompliant
    return name</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">import asyncio

async def get_name():
    print("Please enter your name:")
    name = await asyncio.to_thread(input) # Compliant
    return name</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it_in_trio">How to fix it in Trio</h2>
<div class="sectionbody">
<div class="paragraph">
<p>You can use <code>trio.to_thread.run_sync()</code> to run the <code>input()</code> function in a separate thread.</p>
</div>
<div class="sect2">
<h3 id="_code_examples_2">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example_2">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">import trio

async def get_name():
    print("Please enter your name:")
    name = input() # Noncompliant
    return name</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution_2">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">import trio

async def get_name():
    print("Please enter your name:")
    name = await trio.to_thread.run_sync(input) # Compliant
    return name</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it_in_anyio">How to fix it in AnyIO</h2>
<div class="sectionbody">
<div class="paragraph">
<p>You can use <code>anyio.to_thread.run_sync()</code> to run the <code>input()</code> function in a separate thread.</p>
</div>
<div class="sect2">
<h3 id="_code_examples_3">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example_3">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">import anyio

async def get_name():
    print("Please enter your name:")
    name = input() # Noncompliant
    return name</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution_3">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">import anyio

async def get_name():
    print("Please enter your name:")
    name = await anyio.to_thread.run_sync(input) # Compliant
    return name</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Python asyncio - <a href="https://docs.python.org/3/library/asyncio-task.html#asyncio.to_thread">Running in threads</a></p>
</li>
<li>
<p>Trio - <a href="https://trio.readthedocs.io/en/stable/reference-core.html#trio.to_thread.run_sync">Putting blocking I/O into worker threads</a></p>
</li>
<li>
<p>AnyIO - <a href="https://anyio.readthedocs.io/en/stable/threads.html#calling-synchronous-code-from-a-worker-thread">Calling synchronous code from a worker thread</a></p>
</li>
</ul>
</div>
</div>
<div class="sect2">
<h3 id="_articles_blog_posts">Articles &amp; blog posts</h3>
<div class="ulist">
<ul>
<li>
<p>Python - <a href="https://realpython.com/python-concurrency/">Concurrency and Parallelism in Python</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Wrap this call to <code>input()</code> with {<code>await asyncio.to_thread(input)</code>, <code>await trio.to_thread.run_sync(input)</code>, <code>await anyio.to_thread.run_sync(input)</code>}.</p>
</div>
<div class="paragraph">
<p>If we can&#8217;t detect the asynchronous library: Wrap call to <code>input()</code> with the appropriate function from the asynchronous library.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>The <code>input()</code> function call.
Secondary: the <code>async</code> keyword of the function definition.</p>
</div>
</div>
</div>
</div>