<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when a loop iteration through the dataframe is used instead of vectorized operations.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Pandas is optimized for vectorized operations, which means it&#8217;s much faster to apply operations to entire columns or slices of data than to loop through rows. Using a <code>for</code> loop can be significantly slower, especially on large datasets.</p>
</div>
<div class="paragraph">
<p>Many Pandas methods and functions are designed to work with entire columns or slices of data. When you iterate through rows, you miss out on the full power of Pandas.</p>
</div>
<div class="paragraph">
<p>Instead of using a <code>for</code> loop to iterate through a DataFrame, it&#8217;s generally recommended to use vectorized operations and Pandas functions to perform operations on columns or subsets of data. This approach is not only more efficient but also leads to cleaner and more maintainable code.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">import pandas as pd

df = pd.DataFrame({'X': ['a', 'c', 'c'],
                   'Y': [1, 2, 3]})

for index, row in df.iterrows(): # Noncompliant: iteration over dataframe with loop is slow
  print(f"X: {row['X']}, Y: {row['Y']}")</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">import pandas as pd

df = pd.DataFrame({'X': ['a', 'c', 'c'],
                   'Y': [1, 2, 3]})

df.apply(lambda row: print(f"X: {row['X']}, Y: {row['Y']}"), axis = 1) # Compliant</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Pandas Documentation - <a href="https://pandas.pydata.org/pandas-docs/stable/user_guide/basics.html#iteration">Iteration</a></p>
</li>
<li>
<p>Pandas Documentation - <a href="https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.apply.html">pandas.DataFrame.apply</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>