The ``++LONG++`` and ``++LONG RAW++`` datatypes are deprecated and Oracle recommends to migrate them to the LOB datatypes ``++CLOB++``, ``++NCLOB++`` or ``++BLOB++``.


== Noncompliant Code Example

[source,sql]
----
CREATE TABLE images(
  data LONG RAW
);
----


== Compliant Solution

[source,sql]
----
CREATE TABLE images(
  data BLOB
);
----



ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
