== Why is this an issue?

The ``++LONG++`` and ``++LONG RAW++`` datatypes are deprecated and Oracle recommends to migrate them to the LOB datatypes ``++CLOB++``, ``++NCLOB++`` or ``++BLOB++``.


=== Noncompliant code example

[source,sql]
----
CREATE TABLE images(
  data LONG RAW
);
----


=== Compliant solution

[source,sql]
----
CREATE TABLE images(
  data BLOB
);
----



ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Migrate this "[LONG|LONG RAW]" datatype to a ["CLOB" or "NCLOB"|"BLOB"].


'''
== Comments And Links
(visible only on this page)

=== on 14 Mar 2014, 11:17:36 Dinesh Bolkensteyn wrote:
see


http://www.oracle-developer.net/display.php?id=430


http://docs.oracle.com/cd/B28359_01/server.111/b28318/datatype.htm#CNCPT613 and \http://docs.oracle.com/cd/B28359_01/server.111/b28318/datatype.htm#CNCPT1831


http://stackoverflow.com/questions/12722636/what-is-the-difference-between-long-and-long-raw-data-types-in-oracle

endif::env-github,rspecator-view[]
