include::../rule.adoc[]

=== Noncompliant code example

[source,vbnet]
----
Public Class MyAttribute    ' Noncompliant
    Inherits Attribute

    Public ReadOnly Property Name As String

    Public Sub New(Name As String)
        Me.Name = Name
    End Sub

End Class
----

=== Compliant solution

[source,vbnet]
----
Public NotInheritable Class MyAttribute
    Inherits Attribute

    Public ReadOnly Property Name As String

    Public Sub New(Name As String)
        Me.Name = Name
    End Sub

End Class
----

include::../rspecator.adoc[]