include::../rule.adoc[]

== Noncompliant Code Example

[source,vbnet]
----
Public Class MyAttribute    ' Noncompliant
    Inherits Attribute

    Public ReadOnly Property Name As String

    Public Sub New(Name As String)
        Me.Name = Name
    End Sub

End Class
----

== Compliant Solution

[source,vbnet]
----
Public NotInheritable Class MyAttribute    ' Noncompliant
    Inherits Attribute

    Public ReadOnly Property Name As String

    Public Sub New(Name As String)
        Me.Name = Name
    End Sub

End Class
----

include::../rspectator.adoc[]