== Why is this an issue?

A lambda consisting only of a method call with exactly the same parameters passed to the lambda itself is unneeded. Simply call the enclosed method directly, instead.


=== Noncompliant code example

[source,text]
----
def add(x, y):
    return x + y

add_lambda = lambda x, y: add(x, y)  # Noncompliant

print(add_lambda(1, 2))
----


=== Compliant solution

[source,text]
----
def add(x, y):
    return x + y

print(add(1, 2))
----


