<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Spring Data Repository supports paging for queries, allowing you to return results in small, manageable chunks rather than retrieving an entire large result set.</p>
</div>
<div class="paragraph">
<p>The conventional approach to paginating data in Spring is to use the <code>Pageable</code> interface to control pagination and to store the query results into a <code>Page</code> or <code>Slice</code>.
If a query declaration in a <code>Spring Data Repository</code> returns a <code>Page</code> or <code>Slice</code> without taking a <code>Pageable</code> as an input, it raises a runtime exception.</p>
</div>
<div class="paragraph">
<p>This rule raises an issue on queries in a <code>Repository</code> that return a <code>Page</code> or <code>Slice</code> without taking a <code>Pageable</code> as an input.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Ensure that query methods returning a <code>Page</code> or <code>Slice</code> include a <code>Pageable</code> parameter in their method signature.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">interface ItemRepository extends JpaRepository&lt;Item, Long&gt; {
    Page&lt;Item&gt; findItems(); //non compliant, no Pageable parameter
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">interface ItemRepository extends JpaRepository&lt;Item, Long&gt; {
    Page&lt;Item&gt; findItems(Pageable pageable);
}</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Spring - <a href="https://docs.spring.io/spring-data/jpa/reference/jpa/query-methods.html">JPA Query Methods</a></p>
</li>
<li>
<p>Spring - <a href="https://docs.spring.io/spring-data/jpa/reference/repositories/query-methods-details.html#repositories.paging-and-sorting">Defining Query Methods</a></p>
</li>
</ul>
</div>
</div>
<div class="sect2">
<h3 id="_articles_blog_posts">Articles &amp; blog posts</h3>
<div class="ulist">
<ul>
<li>
<p>Spring Guides - <a href="https://reflectoring.io/spring-boot-paging/">Paging with Spring Boot</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>