== Why is this an issue?

The forms ``++<>++`` and ``++!=++`` are equivalent. But in Python 2.7.3 the ``++<>++`` form is considered obsolete.


=== Noncompliant code example

[source,python]
----
return a <> b # Noncompliant
----


=== Compliant solution

[source,python]
----
return a != b
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Use "!=" instead.


endif::env-github,rspecator-view[]
