This rule raises an issue when the inequality operator `<>` is used.
== Why is this an issue?

The operators ``++<>++`` and ``++!=++`` are equivalent, but since Python 2.7 the ``++<>++`` form is considered obsolete and is removed in newer version of Python.

It is recommended to use `!=` instead.

=== Code examples

==== Noncompliant code example

[source,python]
----
return a <> b # Noncompliant: the operator "<>" is deprecated.
----


==== Compliant solution

[source,python]
----
return a != b
----

ifdef::env-github,rspecator-view[]

== Resources

=== Documentation

* Python Documentation: https://docs.python.org/2.7/reference/lexical_analysis.html#operators[Python 2.7 - Operators]

'''
== Implementation Specification
(visible only on this page)

=== Message

Use "!=" instead.


endif::env-github,rspecator-view[]
