The forms ``++<>++`` and ``++!=++`` are equivalent. But in Python 2.7.3 the ``++<>++`` form is considered obsolete.


== Noncompliant Code Example

[source,python]
----
return a <> b # Noncompliant
----


== Compliant Solution

[source,python]
----
return a != b
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

endif::env-github,rspecator-view[]
