==== Noncompliant code example

[source,csharp,diff-id=2,diff-type=noncompliant]
----
if ((DateTime.Now - lastRefresh).TotalMilliseconds > MinRefreshInterval)
{
    lastRefresh = DateTime.Now;
    // Refresh
}
----

==== Compliant solution

[source,csharp,diff-id=2,diff-type=compliant]
----
if ((DateTime.UtcNow - lastRefresh).TotalMilliseconds > MinRefreshInterval)
{
    lastRefresh = DateTime.UtcNow;
    // Refresh
}
----