== Why is this an issue?

include::../common/rationale.adoc[]

include::../common/impact.adoc[]

== How to fix it

include::../common/fix.adoc[]

=== Code examples

==== Noncompliant code example

[source,vbnet,diff-id=1,diff-type=noncompliant]
----
Imports System.Security.Cryptography

Sub ComputeHash()
    ' Review all instantiations of classes that inherit from HashAlgorithm, for example:
    Dim hashAlgo As HashAlgorithm = HashAlgorithm.Create()  ' Noncompliant
    Dim hashAlgo2 As HashAlgorithm = HashAlgorithm.Create("SHA1")  ' Noncompliant
    Dim sha As SHA1 = New SHA1CryptoServiceProvider()  ' Noncompliant
    Dim md5 As MD5 = New MD5CryptoServiceProvider()  ' Noncompliant

    ' ...
End Sub

Class MyHashAlgorithm
    Inherits HashAlgorithm  ' Noncompliant

    ' ...
End Class
----

==== Compliant solution

[source,vbnet,diff-id=1,diff-type=compliant]
----
Imports System.Security.Cryptography

Sub ComputeHash()
    Dim sha256 = New SHA256CryptoServiceProvider()
    Dim sha384 = New SHA384CryptoServiceProvider()
    Dim sha512 = New SHA512CryptoServiceProvider()

    ' ...
End Sub
----

include::../common/fix-pitfalls.adoc[]

== Resources

include::../common/resources/standards.adoc[]


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../common/message.adoc[]

include::../common/highlighting.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../common/comments-and-links.adoc[]

endif::env-github,rspecator-view[]
