include::../kubernetes/description.adoc[]

== How to fix it

=== Code examples

==== Noncompliant code example

[source,yaml,diff-id=1,diff-type=noncompliant]
----
- name: Noncompliant example task
  kubernetes.core.k8s:
    state: present
    definition:
      apiVersion: rbac.authorization.k8s.io/v1
      kind: Role
      metadata:
        namespace: default
        name: example-role
      rules:
        - apiGroups: [""]
          resources: ["*"] # Noncompliant
          verbs: ["get", "list"]
----

==== Compliant solution

[source,yaml,diff-id=1,diff-type=compliant]
----
- name: Compliant example task
  kubernetes.core.k8s:
    state: present
    definition:
      apiVersion: rbac.authorization.k8s.io/v1
      kind: Role
      metadata:
        namespace: default
        name: example-role
      rules:
        - apiGroups: [""]
          resources: ["pods"]
          verbs: ["get", "list"]
----

=== How does this work?

When defining RBAC permissions, it is important to follow the principle of least privilege. By explicitly specifying the verbs and resources a user should have access to instead of using wildcards, it can be ensured that users have only the permissions they need to perform their job function.

//=== Pitfalls

//=== Going the extra mile


include::../kubernetes/resources.adoc[]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../kubernetes/message.adoc[]


include::../kubernetes/highlighting.adoc[]

endif::env-github,rspecator-view[]
