== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

With the default threshold value of 3:

[source,java]
----
if (((condition1 && condition2) || (condition3 && condition4)) && condition5) { ... } 
----

=== Compliant solution

[source,java]
----
if ( (myFirstCondition() || mySecondCondition()) && myLastCondition()) { ... }     
----

=== Exceptions

No issue is reported inside ``++equals++`` methods, because it is common to compare all the fields of a class for equality inside this kind of method.

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../parameters.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
