== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,sql]
----
IF expression THEN
  RETURN TRUE;
ELSE
  RETURN FALSE;
END IF;
----

=== Compliant solution

[source,sql]
----
RETURN expression;
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
