== Why is this an issue?

include::../description.adoc[]

== How to fix it

include::../how_to_fix.adoc[]

=== Code examples

==== Noncompliant code example

[source,javascript,diff-id=1,diff-type=noncompliant]
----
/**
 * @deprecated Use newFunction instead.
 */
function oldFunction() {
  console.log("This is the old function.");
}

function newFunction() {
  console.log("This is the new function.");
}
oldFunction(); // Noncompliant: "oldFunction is deprecated"
----

==== Compliant solution

[source,javascript,diff-id=1,diff-type=compliant]
----
/**
 * @deprecated Use newFunction instead.
 */
function oldFunction() {
  console.log("This is the old function.");
}

function newFunction() {
  console.log("This is the new function.");
}
newFunction();
----

include::../see.adoc[]
