== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,javascript,diff-id=1,diff-type=noncompliant]
----
export interface LanguageService {
  /**
  * @deprecated Use getEncodedSyntacticClassifications instead.
  */
  getSyntacticClassifications(fileName: string, span: TextSpan): ClassifiedSpan[];
}

const syntacticClassifications = getLanguageService().getSyntacticClassifications(file, span); //Noncompliant
----

include::../see.adoc[]
