== Why is this an issue?

Although Variants are convenient, they are inefficient because they must be converted to the appropriate type before operations are performed on them. 


=== Noncompliant code example

[source,text]
----
Dim I
Dim A as Variant
----


=== Compliant solution

[source,text]
----
Dim I as Integer
Dim A as String
----


