Although Variants are convenient, they are inefficient because they must be converted to the appropriate type before operations are performed on them. 


== Noncompliant Code Example

----
Dim I
Dim A as Variant
----


== Compliant Solution

----
Dim I as Integer
Dim A as String
----


