<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Using <code>.skip(0)</code> on an iterator does nothing, which is likely to be a mistake. It is more probable that <code>.skip(1)</code> was intended to skip the first element, or if not, the <code>.skip(0)</code> call should be removed to avoid confusion and make the code clearer.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-rust" data-lang="rust">let v = vec![1, 2, 3];
let x = v.iter().skip(0).collect::&lt;Vec&lt;_&gt;&gt;(); // Noncompliant: .skip(0) does nothing</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-rust" data-lang="rust">let v = vec![1, 2, 3];
let x = v.iter().collect::&lt;Vec&lt;_&gt;&gt;(); // Compliant: No .skip(0) call makes the intent clear</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Clippy Lints - <a href="https://rust-lang.github.io/rust-clippy/master/index.html#iter_skip_zero" class="bare">https://rust-lang.github.io/rust-clippy/master/index.html#iter_skip_zero</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>