Shared naming conventions allow teams to collaborate effectively. This rule raises an issue when the a pom's ``++groupId++`` does not match the provided regular expression.


== Noncompliant Code Example

With the default regular expression: ``++(com|org)(\.[a-z][a-z-0-9]*)+++``

----
<project ...>
  <groupId>myCo</groupId>  <!-- Noncompliant -->

  <!-- ... -->
</project>
----


== Compliant Solution

----
<project ...>
  <groupId>com.myco</groupId>

  <!-- ... -->
</project>
----



ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

include::parameters.adoc[]

include::highlighting.adoc[]

'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
