== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,dart]
----
void doSomething() {
  // TODO
}
----

include::../see.adoc[]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

The text from the `TODO` tag, included, up to:

* the end of line, for a single-line comment or for a multi-line comment continuing on the next line
* the closing comment tag `*/`, in the case of a multi-line comment

Whitespace characters before the closing comment tag are not included.

=== Highlighting

Same as message.

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
