<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Naming conventions allow teams to collaborate effectively. This rule checks that exception names match a given regular expression.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="paragraph">
<p>With the default regular expression <code>[a-zA-Z]([a-zA-Z0-9_]*[a-zA-Z0-9])?</code>:</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-text" data-lang="text">DECLARE
  my-Exception_ EXCEPTION; -- Noncompliant
BEGIN
  NULL;
END;
/</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-text" data-lang="text">DECLARE
  myException EXCEPTION;
BEGIN
  NULL;
END;
/</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Rename this exception to match the regular expression "xxx".</p>
</div>
</div>
<div class="sect2">
<h3 id="_parameters">Parameters</h3>
<div class="sidebarblock">
<div class="content">
<div class="title">regexp</div>
<div class="listingblock">
<div class="content">
<pre>[a-zA-Z]([a-zA-Z0-9_]*[a-zA-Z0-9])?</pre>
</div>
</div>
<div class="paragraph">
<p>The regular expression the name should match</p>
</div>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_10_nov_2015_130742_ann_campbell_wrote">on 10 Nov 2015, 13:07:42 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~nicolas.peru] I&#8217;ve targeted Java for this rule. Note that we&#8217;ll want to add a companion exception to <a data-rspec-id="S101" class="rspec-auto-link">S101</a> when this is implemented. I&#8217;ll update <a data-rspec-id="S101" class="rspec-auto-link">RSPEC-101</a> when we get to that point.</p>
</div>
</div>
</div>
</div>