== Why is this an issue?

Naming conventions allow teams to collaborate effectively. This rule checks that exception names match a given regular expression.


=== Noncompliant code example

With the default regular expression ``++[a-zA-Z]([a-zA-Z0-9_]*[a-zA-Z0-9])?++``:

[source,text]
----
DECLARE
  my-Exception_ EXCEPTION; -- Noncompliant
BEGIN
  NULL;
END;
/
----


=== Compliant solution

[source,text]
----
DECLARE
  myException EXCEPTION;
BEGIN
  NULL;
END;
/
----

