<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Fields should not be overridden in derived classes.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Unlike other languages, Dart allows to <a href="https://dart.dev/language/extend#overriding-members">override</a> in derived classes any instance member of a class, including fields.</p>
</div>
<div class="paragraph">
<p>Overriding a field in a derived class results in a "double storage": the field in the base class is still present, but it is shadowed by the field in the derived class. So the same name refers to two different storage areas, in two different contexts.</p>
</div>
<div class="paragraph">
<p>That can lead to confusion and unexpected behavior, because, depending on the context, two completely different values are accessed using the same name.</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-dart" data-lang="dart">class BaseClass {
    int field = 0;

    void printField() =&gt; print(field);
}

class DerivedClass extends BaseClass {
    @override int field = 1;

    @override void printField() {
        print(super.field); // Prints the field from the base class
        print(field);       // Prints the field from the derived class
    }
}

void main() {
    BaseClass().printField();    // Prints 0
    DerivedClass().printField(); // Prints 0 and 1
}</code></pre>
</div>
</div>
<div class="paragraph">
<p>Moreover, if no explicit way to access to the field is provided in the base class, the derived class would not be able to access the field at all. This is hardly ever the intended behavior.</p>
</div>
<div class="paragraph">
<p>More likely, overriding a field was a mistake, often not caught by the developer since the compiler doesn&#8217;t enforce the presence of the <code>@override</code> annotation.</p>
</div>
<div class="paragraph">
<p>The rule also applies to mixins, but not to extensions methods or types, since extensions cannot contain instance variable fields.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>The solution depends on the original intent of the field override:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>if the field override was unintentional, for example the result of a copy-paste error, the field in the derived class should be removed, if it is a duplicate, or renamed, if it semantically a new field</p>
</li>
<li>
<p>if the field override was intentional, the field should be wrapped into a property, and the derived class should override the accessors to the field, instead of the field itself</p>
</li>
</ul>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-dart" data-lang="dart">class BaseClass {
    int field = 0;
}

class DerivedClass extends BaseClass {
    @override int field = 1;
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-dart" data-lang="dart">class BaseClass {
    int field = 0;
}

class DerivedClass extends BaseClass {
    // Field removed as it was a duplicate
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_noncompliant_code_example_2">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-dart" data-lang="dart">class BaseClass {
    int field = 0;
}

class DerivedClass extends BaseClass {
    @override int field = 1;
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution_2">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-dart" data-lang="dart">class BaseClass {
    int field = 0;
}

class DerivedClass extends BaseClass {
    int differentField = 1; // Field renamed to better reflect its purpose
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_noncompliant_code_example_3">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-dart" data-lang="dart">class BaseClass {
    int field = 0;
}

class DerivedClass extends BaseClass {
    @override int field = 1;
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution_3">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-dart" data-lang="dart">class BaseClass {
    int _field = 0;
    // Property wrapping the field
    int get field =&gt; _field;
    set field(int value) =&gt; _field = value;
}

class DerivedClass extends BaseClass {
    // Override of the property, rather than the field
    @override int get field =&gt; ...
    @override set field(int value) =&gt; ...
}</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Dart Docs - <a href="https://dart.dev/tools/linter-rules/overridden_fields">Dart Linter rule - overridden_fields</a></p>
</li>
<li>
<p>Dart Docs - <a href="https://dart.dev/language/extend#overriding-members">Language - Overriding members</a></p>
</li>
</ul>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="ulist">
<ul>
<li>
<p>Field overrides a field inherited from '&lt;base_type_name&gt;'.</p>
</li>
</ul>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="ulist">
<ul>
<li>
<p>The identifier of the overriding field in the derived class.</p>
</li>
</ul>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
</div>
</div>