== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

With the default threshold of 3:


[source,swift]
----
func myMethod() -> Bool { // Noncompliant as there are 4 return statements
  if condition1 {
    return true
  } else {
    if condition2 {
      return false
    } else {
      return true
    }
  }
  return false
}
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../parameters.adoc[]

include::../highlighting.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
