This rule raises an issue when a method annotated with `@future` is not declared as `static` or does not have a `void` return type.

== Why is this an issue?

Methods annotated with `@future` in Apex execute asynchronously in a separate thread when system resources become available. This asynchronous execution model has two fundamental requirements:

First, future methods must be `static` because they cannot rely on instance state. When a future method executes, it runs independently of the original calling context, potentially long after the original object instance has been destroyed or modified. Static methods ensure the code can execute without requiring access to instance variables or methods.

Second, future methods must return `void` because they cannot return values to the calling code. The calling thread continues execution immediately after invoking the future method, without waiting for its completion. By the time the future method actually executes, the original caller has already moved on, making it impossible to receive any return value.

These are not just recommendations but strict compilation requirements enforced by the Apex compiler. Any attempt to declare a future method as non-static or with a non-void return type will result in compilation errors, preventing the code from being saved or deployed.

=== What is the potential impact?

Violating this rule will cause compilation errors, preventing the code from being saved, deployed, or executed. This breaks the build process and blocks development progress until the method signature is corrected.

== How to fix it

Declare the method as static and change the return type to void. If you need to return data, consider using alternative approaches like updating database records or sending platform events.

=== Code examples

==== Noncompliant code example

[source,apex,diff-id=1,diff-type=noncompliant]
----
@future
public String processData(String data) { // Noncompliant
    // Process data
    return "processed";
}
----

==== Compliant solution

[source,apex,diff-id=1,diff-type=compliant]
----
@future
public static void processData(String data) {
    // Process data
    // Store results in database or send platform events if needed
    System.debug('Data processed: ' + data);
}
----

== Resources

=== Documentation

 * Future Methods - https://developer.salesforce.com/docs/atlas.en-us.apexcode.meta/apexcode/apex_invoking_future_methods.htm[Official Salesforce documentation on future methods and their requirements]

 * Asynchronous Apex - https://developer.salesforce.com/docs/atlas.en-us.apexcode.meta/apexcode/apex_async_overview.htm[Overview of asynchronous processing options in Apex]
