== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

With default provided regular expression: ^([A-Z0-9_]{empty}*|[a-z0-9_]*)$


[source,javascript]
----
function DoSomething(){...} 
----

=== Compliant solution

[source,javascript]
----
function doSomething(){...} 
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../parameters.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
