include::../description.adoc[]

== Noncompliant Code Example

With default provided regular expression: ^([A-Z0-9_]{empty}*|[a-z0-9_]*)$


----
function DoSomething(){...} 
----

== Compliant Solution

----
function doSomething(){...} 
----
