include::../description.adoc[]

== Noncompliant Code Example

With default provided regular expression: ^([A-Z0-9_]{empty}*|[a-z0-9_]*)$


----
function DoSomething(){...} 
----

== Compliant Solution

----
function doSomething(){...} 
----

ifdef::rspecator-view[]
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::rspecator-view[]
