== How to fix it in BouncyCastle

=== Code examples

include::../../common/fix/code-rationale.adoc[]

==== Noncompliant code example

[source,vbnet,diff-id=1,diff-type=noncompliant]
----
Dim AesFast As new AesFastEngine() ' Noncompliant
----

==== Compliant solution

[source,vbnet,diff-id=1,diff-type=compliant]
----
Dim AES As new AESEngine()
----

=== How does this work?

include::../../common/fix/strong-cryptography.adoc[]

