== How to fix it in BouncyCastle

=== Code examples

include::../../common/fix/code-rationale.adoc[]

==== Noncompliant code example

[source,vbnet,diff-id=1,diff-type=noncompliant]
----
```suggestion
Imports Org.BouncyCastle.Crypto.Engines
Imports Org.BouncyCastle.Crypto.Parameters

Public Sub Encrypt()
    Dim AesFast As new AesFastEngine() ' Noncompliant
End Sub
----

==== Compliant solution

[source,vbnet,diff-id=1,diff-type=compliant]
----
Imports Org.BouncyCastle.Crypto.Engines
Imports Org.BouncyCastle.Crypto.Parameters

```suggestion
Public Sub Encrypt()
    Dim AES As new AESEngine()
End Sub
----

=== How does this work?

include::../../common/fix/strong-cryptography.adoc[]

