== How to fix it in .NET

=== Code examples

include::../../common/fix/code-rationale.adoc[]

==== Noncompliant code example

[source,vbnet,diff-id=1,diff-type=noncompliant]
----
Dim SimpleDES As New DESCryptoServiceProvider() ' Noncompliant
----

==== Compliant solution

[source,vbnet,diff-id=1,diff-type=compliant]
----
Dim AES As new AesCryptoServiceProvider()
----

=== How does this work?

include::../../common/fix/strong-cryptography.adoc[]

