<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Having duplicate environment variable names for a container can lead to unexpected behavior, as the last declared variable will overwrite any previous ones with the same name. This rule aims to detect and prevent such situations, thereby enhancing the clarity and predictability of Kubernetes configurations.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>If an environment variable of a container is declared multiple times with different values, it can become unclear which value will be used by the application and can lead to unpredictable behavior. Also, if an issue arises related to the environment variable, debugging can be challenging. The problem could be due to the overwritten variable, but this might take time to be noticeable.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Identify the duplicate environment variables for a container and remove the duplicates manually. If the same variable is needed with different values, consider using a different name for each variable.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-text" data-lang="text">apiVersion: v1
kind: Pod
metadata:
  name: env-example
spec:
  containers:
  - name: nginx
    image: nginx:1.7.9
    env:
    - name: VAR1 # Noncompliant
      value: "value1"
    - name: VAR1 # Noncompliant
      value: "value2"</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-text" data-lang="text">apiVersion: v1
kind: Pod
metadata:
  name: env-example
spec:
  containers:
  - name: nginx
    image: nginx:1.7.9
    env:
    - name: VAR1
      value: "value1"
    - name: VAR2
      value: "value2"</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p><a href="https://kubernetes.io/docs/tasks/inject-data-application/define-environment-variable-container/">Kubernetes - Environment Variables for a Container</a></p>
</li>
</ul>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Resolve the duplication of this environment variable.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="ulist">
<ul>
<li>
<p>Highlight all the locations where duplicated environment variables are being used.</p>
</li>
</ul>
</div>
</div>
</div>
</div>