<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when <code>Math.pow()</code> is cast to an integer type (<code>int</code> or <code>long</code>) and both arguments are integers.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>When you need to calculate powers of integers, casting <code>Math.pow()</code> to an integer type causes problems.</p>
</div>
<div class="paragraph">
<p><code>Math.pow()</code> returns a <code>double</code>, which means the calculation uses floating-point arithmetic. When you cast this result back to an integer, you get:</p>
</div>
<div class="ulist">
<ul>
<li>
<p><strong>Silent overflow</strong>: Results exceeding the integer type&#8217;s maximum value wrap around without warning</p>
</li>
<li>
<p><strong>Precision loss</strong>: Floating-point arithmetic introduces rounding errors</p>
</li>
<li>
<p><strong>Performance overhead</strong>: Converting integers to doubles and back is less efficient than direct integer operations</p>
</li>
</ul>
</div>
<div class="paragraph">
<p><code>Math.powExact()</code> is more efficient and avoids precision loss by working directly with integers.</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="paragraph">
<p>Silent overflow or precision loss can lead to incorrect calculations in your application. This is particularly dangerous in financial calculations, cryptographic operations, or any domain where mathematical accuracy is essential. The bugs may go unnoticed during testing but cause problems in production with larger input values.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Replace the cast of <code>Math.pow()</code> with <code>Math.powExact()</code>. This method is designed specifically for integer exponentiation and will throw an <code>ArithmeticException</code> if the result overflows, making errors explicit rather than silent.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">public int calculatePower(int base, int exponent) {
    return (int) Math.pow(base, exponent); // Noncompliant
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">public int calculatePower(int base, int exponent) {
    return Math.powExact(base, exponent);
}</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Math.powExact() - Java Documentation - <a href="https://docs.oracle.com/en/java/javase/25/docs/api/java.base/java/lang/Math.html#powExact(int,int)">Official documentation for Math.powExact methods</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>