=== relates to: S949

=== on 16 Oct 2014, 13:02:35 Ann Campbell wrote:
\[~samuel.mercier] please:

* fill in the appropriate reference field(s).
* provide a See section.
* use double curly braces around code in the description, e.g. NULL.
* use the standard section titles: Noncompliant Code Example, Exception*s*

=== on 17 Oct 2014, 15:42:00 Ann Campbell wrote:
\[~freddy.mallet] this rule an RSPEC-949 are 2 sides of the same coin

