== Why is this an issue?

In {cpp}, the literal 0 is both an integer type and the null-pointer-constant. To meet developer expectations, ``++NULL++`` should be used as the null-pointer-constant, and 0 for the integer zero.

Note: as a result of this rule, ``++NULL++`` is considered to have pointer type.


=== Noncompliant code example

[source,cpp]
----
#include <cstddef>

void f1 ( int32_t );
void f2 ( int32_t * );
void f3 ( )
{
  f1 ( NULL ); // Noncompliant, "NULL" used as an integer
  f2 ( NULL ); // Compliant, pointer expected
}
----


=== Compliant solution

[source,cpp]
----
#include <cstddef>

void f1 ( int32_t );
void f2 ( int32_t * );
void f3 ( )
{
  f1 ( 0 ); // Compliant, integer expected
}
----


== Resources

* MISRA {cpp}:2008, 4-10-1


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Replace this use of "NULL" with the integer constant '0'.


'''
== Comments And Links
(visible only on this page)

=== relates to: S949

=== on 16 Oct 2014, 13:02:35 Ann Campbell wrote:
\[~samuel.mercier] please:

* fill in the appropriate reference field(s).
* provide a See section.
* use double curly braces around code in the description, e.g. NULL.
* use the standard section titles: Noncompliant Code Example, Exception*s*

=== on 17 Oct 2014, 15:42:00 Ann Campbell wrote:
\[~freddy.mallet] this rule an RSPEC-949 are 2 sides of the same coin

endif::env-github,rspecator-view[]
