This rule raises an issue when Rails associations use deprecated hash-style options like `:order` or `:uniq` instead of modern lambda scope blocks.

== Why is this an issue?

Rails 4 deprecated hash-style options like `:order` and `:uniq` in association declarations in favor of lambda scope blocks. These deprecated options generate deprecation warnings and may be removed in future Rails versions.

The old hash syntax has several limitations:

* Limited flexibility for complex queries
* Inconsistent with Rails' scoping conventions
* Cannot chain multiple scope methods easily
* Generates deprecation warnings in Rails 4+

Lambda scope blocks provide a more powerful and consistent approach:

* Allow chaining multiple scope methods like `where`, `order`, and `uniq`
* Follow Rails' modern scoping conventions
* Provide better readability and maintainability
* Enable more complex query logic when needed

Using deprecated options makes your codebase less maintainable and may cause issues when upgrading Rails versions.

=== What is the potential impact?

Using deprecated association options can lead to:

* Deprecation warnings cluttering your application logs
* Potential breaking changes when upgrading Rails versions
* Reduced code maintainability and consistency
* Limited flexibility for extending association logic

== How to fix it in Rails

Replace the deprecated `:order` option with a lambda scope block using the `order` method. Place the lambda block before other association options.

=== Code examples

==== Noncompliant code example

[source,ruby,diff-id=1,diff-type=noncompliant]
----
class RelatedList < ActiveRecord::Base
  has_many :contents, :order => :position # Noncompliant
  has_many :tasks, order: 'priority desc', dependent: :destroy # Noncompliant
end
----

==== Compliant solution

[source,ruby,diff-id=1,diff-type=compliant]
----
class RelatedList < ActiveRecord::Base
  has_many :contents, -> { order(:position) }
  has_many :tasks, -> { order(priority: :desc) }, dependent: :destroy
end
----

Replace the deprecated `:uniq` option with a lambda scope block using the `uniq` method. You can chain multiple scope methods together.

=== Code examples

==== Noncompliant code example

[source,ruby,diff-id=2,diff-type=noncompliant]
----
class Goal < ActiveRecord::Base
  has_many :donors, through: :donations, uniq: true # Noncompliant
  has_many :contributors, through: :donations, uniq: true, order: "name" # Noncompliant
end
----

==== Compliant solution

[source,ruby,diff-id=2,diff-type=compliant]
----
class Goal < ActiveRecord::Base
  has_many :donors, -> { uniq }, through: :donations
  has_many :contributors, -> { order(:name).uniq }, through: :donations
end
----

== Resources

=== Documentation

 * Rails Association Basics - https://guides.rubyonrails.org/association_basics.html[Official Rails guide covering association declarations and scoping]

 * Active Record Query Interface - https://guides.rubyonrails.org/active_record_querying.html[Rails guide explaining query methods like order and uniq]
