== Why is this an issue?

Spring Data Repository supports paging for queries, allowing you to return results in small, manageable chunks rather than retrieving an entire large result set.

The conventional approach to paginating data in Spring is to use the `Pageable` interface to control pagination and to store the query results into a `Page` or `Slice`.
If a query method in a `Repository` returns a `Page` or `Slice` without taking a `Pageable` as an input, it raises a runtime exception.

This rule raises an issue on query methods that return a `Page` or `Slice` without taking a `Pageable` as an input.

== How to fix it

Ensure that query methods returning a `Page` or `Slice` include a `Pageable` parameter in their method signature.

=== Code examples

==== Noncompliant code example

[source,java,diff-id=1,diff-type=noncompliant]
----
public Page<Item> findItems() { //non compliant, no Pageable parameter
    // query
}
----

==== Compliant solution

[source,java,diff-id=1,diff-type=compliant]
----
public Page<Item> findItems(Pageable pageable) {
    // query
}
----

== Resources
=== Documentation
* Spring - https://docs.spring.io/spring-data/jpa/reference/jpa/query-methods.html[JPA Query Methods]
* Spring - https://docs.spring.io/spring-data/jpa/reference/repositories/query-methods-details.html#repositories.paging-and-sorting[Defining Query Methods]

=== Articles & blog posts
* Spring Guides - https://reflectoring.io/spring-boot-paging/[Paging with Spring Boot]

