<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Using the output of a non-boolean assignment as a condition is highly suspicious. Either it was done accidentally, and the assignment operator <code>=</code> was used in place of the equality operator <code>==</code>, or it was done with the intent to rely on the automatic conversion of the non-boolean value to a boolean, which could be confusing for maintainers.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-javascript" data-lang="javascript">function process(var a) {
  var BAD_OPS = 3;
  if( a = BAD_OPS) {...}
}</code></pre>
</div>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-javascript" data-lang="javascript">function process(var a) {
  var BAD_OPS = "UNKNOWN";
  if( a = BAD_OPS) {...}
}</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Refactor this suspicious code to not use the result of this non-boolean assignment as a condition</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>The overall non-boolean assignment</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_relates_to_s1121">relates to: <a data-rspec-id="S1121" class="rspec-auto-link">S1121</a></h3>

</div>
<div class="sect2">
<h3 id="_on_21_mar_2017_095054_freddy_mallet_wrote">on 21 Mar 2017, 09:50:54 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>Before C99, the C language didn&#8217;t have any boolean type so applying this rule on C might raise a lot of false-positives.</p>
</div>
</div>
</div>
</div>