include::../description.adoc[]

== Noncompliant Code Example

The following code snippet illustrates this rule with the default threshold of 5:

----
switch myVariable {
  case 0: // 6 lines till next case
    methodCall1("")
    methodCall2("")
    methodCall3("")
    methodCall4("")
    methodCall5("")
  case 1:
  ...
}
----

== Compliant Solution

----
switch myVariable {
  case 0:
    doSomething()
  case 1:
  ...
}
...
func doSomething(){
    methodCall1("")
    methodCall2("")
    methodCall3("")
    methodCall4("")
    methodCall5("")
}
----

ifdef::env-github,rspecator-view[]
'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
