To avoid holding more connections than necessary when using `HttpClient`, `DocumentClient` or Azure Storage clients, consider:

* Creating a single, static client that every function invocation can use.
* Using a static client in a shared helper class if different functions use the same service.
* Use the client with dependency injection.

// If you want to factorize the description uncomment the following line and create the file.
//include::../description.adoc[]

== Noncompliant Code Example

[source,csharp]
----
    public class HttpExample
    {

        [FunctionName("HttpExample")]
        public async Task<IActionResult> Run([HttpTrigger(AuthorizationLevel.Anonymous, "get", Route = null)] HttpRequest request, ILogger log)
        {
            HttpClient httpClient = new HttpClient(); // Noncompliant

            var response = await httpClient.GetAsync("https://example.com");
            // rest of the function
        }
    }
----

== Compliant Solution

[source,csharp]
----
    public class HttpExample
    {
        // Create a single, static HttpClient
        private static HttpClient httpClient = new HttpClient();

        [FunctionName("HttpExample")]
        public static async Task<IActionResult> Run([HttpTrigger(AuthorizationLevel.Anonymous, "get", Route = null)] HttpRequest request)
        {
            var response = await httpClient.GetAsync("https://example.com");
            // rest of the function
        }
    }
----

== See

* https://docs.microsoft.com/en-us/azure/azure-functions/manage-connections?tabs=csharp#static-clients[Microsoft docs] - Manage connections in Azure Functions: Static Clients