include::../description.adoc[]

== Noncompliant Code Example

[source,docker]
----
RUN --mount=type=secret,id=build_secret,mode=0777 ./installer.sh # Noncompliant
----

== Compliant Solution

[source,docker]
----
RUN --mount=type=secret,id=build_secret,uid=1000 ./installer.sh
----

include::../see.adoc[]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]


'''
endif::env-github,rspecator-view[]
