== Why is this an issue?

include::../description.adoc[]

== How to fix it

Rename packages with the expected naming convention

=== Code examples

==== Noncompliant code example

With the default regular expression ``++[a-zA-Z]([a-zA-Z_]*[a-zA-Z])?++``:

[source,plsql,diff-id=1,diff-type=noncompliant]
----
CREATE OR REPLACE PACKAGE invalid_package_ AS
  PROCEDURE display_message;
END invalid_package_;
----

==== Compliant solution

[source,plsql,diff-id=1,diff-type=compliant]
----
CREATE OR REPLACE PACKAGE valid_package AS
  PROCEDURE display_message;
END valid_package;
----

include::./rspecator.adoc[]
