== Why is this an issue?

When using a ``++PRINTER++`` defined file, an ``++INFDS++`` keyword should be used to read the file's line count from ``++*PSSR++`` positions 367-368.


=== Noncompliant code example

[source,rpg]
----
 FCSO602R1  O    E             PRINTER USROPN 
 or ... 
 FCSO602R1  O    E             PRINTER USROPN 
 F                                     INFDS(WSFD01) 
 F                                     INFSR(*PSSR) 

 D WSFD01          DS 
 D  W1ST01           *STATUS 
 D  W1SPNO               123    124B 0 
----


=== Compliant solution

[source,rpg]
----
 FCSO602R1  O    E             PRINTER USROPN 
 F                                     INFDS(WSFD01) 
 F                                     INFSR(*PSSR) 

 D WSFD01          DS 
 D  W1ST01           *STATUS 
 D  W1SPNO               123    124B 0 
 D  W1LNCT               367    368B 0 
 D  W1PGNO               369    372B 0 
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Include an ``++INFDS++`` keyword with a reference to position 367-368.


endif::env-github,rspecator-view[]
