== Why is this an issue?

Parentheses can disambiguate the order of operations in complex expressions and make the code easier to understand.

[source,python]
----
a = (b * c) + (d * e) # Compliant: the intent is clear.
----

Redundant parentheses are parenthesis that
do not change the behavior of the code, and
do not clarify the intent.
They can mislead and complexify the code.
They should be removed.

=== Noncompliant code example

[source,python]
----
return ((3))        # Noncompliant
return ((x + 1))    # Noncompliant
x = ((y / 2)) + 1   # Noncompliant
----

=== Compliant solution

[source,python]
----
return 3
return (3)
return x + 1
return (x + 1)
x = y / 2 + 1
x = (y / 2) + 1
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Remove those redundant parentheses.

include::../highlighting.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
