Creating temporary primitive wrapper objects only for ``++String++`` conversion or the use of the ``++compareTo++`` method is inefficient.

Instead, the static ``++toString()++`` or ``++compare++`` method of the primitive wrapper class should be used.


== Noncompliant Code Example

----
new Integer(myInteger).toString();  // Noncompliant
----


== Compliant Solution

----
Integer.toString(myInteger);        // Compliant
----



ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
