== Why is this an issue?

The use of ``++UNDERLINE++`` makes normal text printed to the screen difficult to read, and prevents titles from being read at all. For these reasons, ``++UNDERLINE++`` should not be used.


=== Noncompliant code example

[source,rpg]
----
     R DETAIL1
      * Noncompliant; next line contains UNDERLINE
       NAME          25      2  2UNDERLINE
       ADD1          25      3  2
     R DETAIL3
       ADD2          25         2SPACEB(1)
----


=== Compliant solution

[source,rpg]
----
     R DETAIL1
       NAME          25      2  2
       ADD1          25      3  2
     R DETAIL3
       ADD2          25         2SPACEB(1)
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
