== Why is this an issue?

The use of ``++UNDERLINE++`` makes normal text printed to the screen difficult to read, and prevents titles from being read at all. For these reasons, ``++UNDERLINE++`` should not be used.


=== Noncompliant code example

[source,rpg]
----
     R DETAIL1
      * Noncompliant; next line contains UNDERLINE
       NAME          25      2  2UNDERLINE
       ADD1          25      3  2
     R DETAIL3
       ADD2          25         2SPACEB(1)
----


=== Compliant solution

[source,rpg]
----
     R DETAIL1
       NAME          25      2  2
       ADD1          25      3  2
     R DETAIL3
       ADD2          25         2SPACEB(1)
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Remove this use of "UNDERLINE".


'''
== Comments And Links
(visible only on this page)

=== on 22 Sep 2014, 06:50:24 Ann Campbell wrote:
Note that this rule is actually for DDS, the record-format specification for RPG, rather than for RPG itself.

=== on 29 Sep 2014, 13:13:15 Ann Campbell wrote:
Note that this rule is for DDS

endif::env-github,rspecator-view[]
