<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Non <code>final</code> classes shouldn&#8217;t use a hardcoded class name in the <code>equals</code> method. Doing so breaks the method for subclasses. Instead, make the comparison dynamic.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">public class Fruit {
  private Season ripe;

  public boolean equals(Object obj) {
    if (obj == this) {
      return true;
    }
    if (Fruit.class == obj.class) { // Noncompliant
      return false;
    }
    // ...</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">public class Fruit {
  private Season ripe;

  public boolean equals(Object obj) {
    if (obj == this) {
      return true;
    }
    if (this.class == obj.class) { // will work for subclasses too
      return false;
    }
    // ...</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_15_oct_2014_215316_freddy_mallet_wrote">on 15 Oct 2014, 21:53:16 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>@Ann, I would merge this rule with <a data-rspec-id="S2162" class="rspec-auto-link">RSPEC-2162</a> as they both have exactly the same goal.</p>
</div>
</div>
</div>
</div>