<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when a finalizer assign <code>null</code> to fields of the instance it is called on.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>In the Java object lifecycle, the <code>finalize</code> method for an instance is called
after the garbage collector has determined that the instance can be removed from the object heap.
Therefore, it is unnecessary to implement a finalizer to set instance fields explicitly to <code>null</code>
to tell the garbage collector that the instance no longer needs them.</p>
</div>
<div class="paragraph">
<p>In the worst case, implementing <code>finalize</code> is even counterproductive because it might accidentally create new references
from other (living) objects on the heap to the collectible instance, thus, reviving it.</p>
</div>
<div class="paragraph">
<p><strong>Important note about finalizers:</strong></p>
</div>
<div class="paragraph">
<p>There are no guarantees when the Java Runtime will call the <code>finalize</code> method or whether it will be called at all.</p>
</div>
<div class="paragraph">
<p>Using finalizers is, therefore, a bad practice. They should never be used to free resources,
such as closing streams, freeing locks, or freeing native system resources.
Consider other freeing mechanisms instead, such as an explicit <code>close</code>, <code>unlock</code>, or <code>free</code> method in your class.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Remove assignments from your finalizer that assign <code>null</code> to fields of the instance the finalizer is called on.
When this leaves you with an empty finalizer body, remove the finalizer.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">public class Foo {
  private String name;

  @Override
  void finalize() {
    name = null;  // Noncompliant, instance will be removed anyway
  }
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">public class Foo { // Compliant
  private String name;
}</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p><a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Object.html#finalize--">Java SE 8 API Specification - Object.finalize</a></p>
</li>
</ul>
</div>
</div>
<div class="sect2">
<h3 id="_articles_blog_posts">Articles &amp; blog posts</h3>
<div class="ulist">
<ul>
<li>
<p><a href="https://howtodoinjava.com/java/basics/why-not-to-use-finalize-method-in-java">Java finalize() â€“ Why We Should Not Use It? - Lokesh Gupta</a></p>
</li>
<li>
<p><a href="https://www.baeldung.com/java-finalize">A Guide to the finalize Method in Java - Baeldung</a></p>
</li>
</ul>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove this nullification of "xxx".</p>
</div>
</div>
</div>
</div>