== How to fix it in CommonCrypto

=== Code examples

==== Noncompliant code example

include::../../common/fix/aes-noncompliant-example.adoc[]

[source,swift]
----
import CommonCrypto

func encrypt(key: Data, iv: Data, plaintext: Data) -> Data? {
    var ciphertext = Data(count: plaintext.count + kCCBlockSizeAES128)
    var bytesEncrypted = 0

    let status = ciphertext.withUnsafeMutableBytes { out_buf in
        plaintext.withUnsafeBytes { in_buf in
            iv.withUnsafeBytes { iv_buf in
                key.withUnsafeBytes { key_buf in
                    CCCrypt(CCOperation(kCCEncrypt),
                        CCAlgorithm(kCCAlgorithmAES),
                        CCOptions(kCCOptionECBMode),  // Noncompliant
                        key_buf.baseAddress, key.count, iv_buf.baseAddress,
                        in_buf.baseAddress, plaintext.count,
                        out_buf.baseAddress, ciphertext.count,
                        &bytesEncrypted)
                }
            }
        }
    }

    if status == kCCSuccess {
        ciphertext.count = bytesEncrypted
        return ciphertext
    }
    return nil
}
----


==== Compliant solution

The recommended AES mode, AES-GCM, is not supported by CommonCrypto. It is recommended to use https://developer.apple.com/documentation/cryptokit[CryptoKit] instead.

=== How does this work?

include::../../common/fix/fix.adoc[]
