<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Having robust documentation just makes life easier. That applies to XSD&#8217;s as well. Each one should include a description of its intended use, and perhaps versioning.</p>
</div>
<div class="paragraph">
<p>This rule raises an issue when there&#8217;s neither a comment before the <code>&lt;xs:schema&gt;</code> tag, nor <code>&lt;xsd:annotation&gt;&lt;xsd:documentation&gt;...&lt;/xsd:documentation&gt;&lt;/xsd:annotation&gt;</code> just inside the <code>&lt;xs:schema&gt;</code> element.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-xml" data-lang="xml">&lt;xs:schema targetNamespace="http://www.codeSamples.com/fruit"
           xmlns:xs="http://www.w3.org/2001/XMLSchema"
           elementFormDefault="qualified" version="1"&gt;
  &lt;xs:element name="fruit"&gt;
    ...
  &lt;/xs:element&gt;
&lt;/xs:schema&gt;</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-xml" data-lang="xml">&lt;-- This schema is intended... --&gt;
&lt;xs:schema targetNamespace="http://www.codeSamples.com/fruit"
           xmlns:xs="http://www.w3.org/2001/XMLSchema"
           elementFormDefault="qualified" version="1"&gt;
  &lt;xs:element name="fruit"&gt;
    ...
  &lt;/xs:element&gt;
&lt;/xs:schema&gt;</code></pre>
</div>
</div>
<div class="paragraph">
<p>or</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-xml" data-lang="xml">&lt;xs:schema targetNamespace="http://www.codeSamples.com/fruit"
           xmlns:xs="http://www.w3.org/2001/XMLSchema"
           elementFormDefault="qualified" version="1"&gt;
  &lt;xsd:annotation&gt;
    &lt;xsd:documentation&gt;
      This schema is intended...
    &lt;/xsd:documentation:
  &lt;/xsd:annotation&gt;
  &lt;xs:element name="fruit"&gt;
    ...
  &lt;/xs:element&gt;
&lt;/xs:schema&gt;</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Add a description to document this schema.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p><code>&lt;xs:schema&gt;</code></p>
</div>
</div>
</div>
</div>