The use of positional notation to describe the overlapping of one field on another can quickly become confusing, particularly in large, or complicated data structures. Instead, the `+OVERLAY+` keyword should be used to make such overlaps more readily identifiable.


== Noncompliant Code Example

----
     D Message         DS                  QUALIFIED
     D   Id                           7
      * Noncompliant
     D   Type                  1      3
----


== Compliant Solution

----
     D Message         DS                  QUALIFIED
     D   Id                           7
     D   Type                         3    OVERLAY(Id)
----

