This rule raises an issue when a `return` statement is used inside a block (created with `{}` or `do...end`).

== Why is this an issue?

In Ruby, blocks are closures that capture the lexical environment where they are defined. When you use `return` inside a block, it doesn't return from the block itself - it returns from the method that lexically contains the block definition.

This behavior can lead to two main problems:

* *LocalJumpError at runtime*: If the block is called outside the context of its defining method, Ruby raises a `LocalJumpError` because there's no method to return from.
* *Unexpected early method returns*: When the block is executed within its defining method, the `return` statement causes the entire method to exit immediately, potentially skipping important cleanup code or logic.

This confusion often arises because developers expect blocks to behave like methods, but blocks are actually more like anonymous functions that share the same scope as their surrounding method.

The distinction becomes clearer when you understand that Ruby has different types of callable objects:

* *Blocks*: Created with `{}` or `do...end`, they share scope with the surrounding method
* *Procs*: Objects created from blocks, they retain the same return behavior as blocks
* *Lambdas*: Special procs that have their own scope and handle `return` statements locally

=== What is the potential impact?

Using `return` in blocks can cause applications to crash with `LocalJumpError` exceptions at runtime, making the code unreliable. It can also lead to subtle bugs where methods exit unexpectedly, bypassing important logic, cleanup operations, or error handling code. This makes the application's behavior unpredictable and difficult to debug.

== How to fix it

Use implicit returns by making the last expression in the block the value you want to return. This is the most idiomatic Ruby approach.

=== Code examples

==== Noncompliant code example

[source,ruby,diff-id=1,diff-type=noncompliant]
----
numbers = (1..5).collect { |n|
  if n.even?
    return n * 2  # Noncompliant
  else
    return n      # Noncompliant
  end
}
----

==== Compliant solution

[source,ruby,diff-id=1,diff-type=compliant]
----
numbers = (1..5).collect { |n|
  if n.even?
    n * 2  # implicit return
  else
    n      # implicit return
  end
}
----

== Resources

=== Documentation

 * Ruby Documentation - Blocks, Procs and Lambdas - https://ruby-doc.org/core/Proc.html[Official Ruby documentation explaining the differences between blocks, procs, and lambdas]

 * Ruby Style Guide - Blocks - https://rubystyle.guide/#no-explicit-return[Community style guide recommendations for block usage and return statements]
