This rule raises an issue when an instance method is called directly on a class instead of on an instance of that class.

== Why is this an issue?

In Ruby, there are two types of methods: instance methods and class methods. Instance methods are defined without the `self.` prefix and must be called on object instances. Class methods are defined with the `self.` prefix and can be called directly on the class.

When you try to call an instance method directly on a class, Ruby cannot find the method because it only exists in the context of an instance. This results in a `NoMethodError` at runtime.

This error often occurs when developers are unsure about the difference between instance and class methods, or when they change a method from class to instance (or vice versa) but forget to update all the call sites.

The distinction is important because:

* Instance methods have access to instance variables (`@variable`) and the specific state of an object
* Class methods have access to class variables (`@@variable`) and class-level functionality
* Each serves different purposes in object-oriented design

=== What is the potential impact?

This issue will cause a `NoMethodError` exception at runtime, potentially crashing the application. The error occurs when the problematic code path is executed, which might not be immediately apparent during development if the code isn't thoroughly tested.

In production environments, this can lead to:

* Application crashes and service interruptions
* Poor user experience due to unexpected errors
* Difficulty debugging, especially if the error occurs in complex call chains

== How to fix it

Convert the instance method to a class method by adding the `self.` prefix. This approach is suitable when the method doesn't need access to instance-specific data.

=== Code examples

==== Noncompliant code example

[source,ruby,diff-id=1,diff-type=noncompliant]
----
class Calculator
  def add(a, b)  # Noncompliant: instance method
    a + b
  end
end

result = Calculator.add(2, 3)  # Noncompliant: calling instance method on class
----

==== Compliant solution

[source,ruby,diff-id=1,diff-type=compliant]
----
class Calculator
  def self.add(a, b)  # Class method with self. prefix
    a + b
  end
end

result = Calculator.add(2, 3)  # Correct: calling class method on class
----

== Resources

=== Documentation

 * Ruby Class and Instance Methods - https://ruby-doc.org/core/doc/syntax/methods_rdoc.html[Official Ruby documentation explaining the difference between class and instance methods]

 * Ruby Classes and Objects - https://ruby-doc.org/core/Class.html[Ruby documentation on classes and object instantiation]
