<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Defining variables in the minimum block scope possible reduces the visibility of those variables and therefore reduces the possibility that these identifiers will be used accidentally. A corollary of this is that global objects (including singleton function objects) shall be used in more than one function.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">void f ( int32_t k )
{
  int32_t i = k * k; // Noncompliant, visibility could be reduced
  {
    std::cout &lt;&lt; i &lt;&lt; std::endl;
  }
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">void f ( int32_t k )
{
  {
    int32_t i = k * k; // Compliant
    std::cout &lt;&lt; i &lt;&lt; std::endl;
  }
}</code></pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p>MISRA C&#43;&#43;:2008, 3-4-1</p>
</li>
</ul>
</div>
<hr>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Scope of this declaration should be reduced.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_relates_to_s1899">relates to: <a data-rspec-id="S1899" class="rspec-auto-link">S1899</a></h3>

</div>
<div class="sect2">
<h3 id="_relates_to_s5955">relates to: <a data-rspec-id="S5955" class="rspec-auto-link">S5955</a></h3>

</div>
<div class="sect2">
<h3 id="_is_related_to_s3798">is related to: <a data-rspec-id="S3798" class="rspec-auto-link">S3798</a></h3>

</div>
<div class="sect2">
<h3 id="_relates_to_s831">relates to: <a data-rspec-id="S831" class="rspec-auto-link">S831</a></h3>

</div>
<div class="sect2">
<h3 id="_is_related_to_s825">is related to: <a data-rspec-id="S825" class="rspec-auto-link">S825</a></h3>

</div>
<div class="sect2">
<h3 id="_on_19_feb_2016_224544_evgeny_mandrikov_wrote">on 19 Feb 2016, 22:45:44 Evgeny Mandrikov wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] looks very similar to <a data-rspec-id="S2392" class="rspec-auto-link">RSPEC-2392</a></p>
</div>
</div>
<div class="sect2">
<h3 id="_on_22_feb_2016_165223_ann_campbell_wrote">on 22 Feb 2016, 16:52:23 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>Actually [~evgeny.mandrikov] <a data-rspec-id="S2392" class="rspec-auto-link">RSPEC-2392</a> used to be a subtask of <a data-rspec-id="S1899" class="rspec-auto-link">RSPEC-1899</a> but was promoted because <a data-rspec-id="S1899" class="rspec-auto-link">RSPEC-1899</a>, like this rule, is only about moving vars down from global to local scope. In JavaScript they can move in both directions.</p>
</div>
<div class="paragraph">
<p>And because this is a MISRA rule, there are of course related rules (<a data-rspec-id="S825" class="rspec-auto-link">RSPEC-825</a>). Since neither this rule nor <a data-rspec-id="S825" class="rspec-auto-link">RSPEC-825</a> is implemented yet, would you like to roll them all together?</p>
</div>
</div>
</div>
</div>