== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,go]
----
func add(x, y int) int {                   
	return x + y // Noncompliant
	z := x + y // dead code
}
----

=== Compliant solution

[source,go]
----
func add(x, y int) int {
	return x + y // Compliant
}
----

include::../see.adoc[]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../highlighting.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
