== Why is this an issue?

Functions declared with the ``++set++`` keyword will automatically return the values they were passed. Thus any value explicitly returned from a setter will be ignored, and explicitly returning a value is an error.


=== Noncompliant code example

[source,javascript]
----
var person = {
  // ...
  set name(name) {
    this.name = name;
    return 42;  // Noncompliant
  }
}
----


=== Compliant solution

[source,javascript]
----
var person = {
  // ...
  set name(name) {
    this.name = name;
  }
}
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Consider removing this return statement; it will be ignored.


'''
== Comments And Links
(visible only on this page)

=== on 21 Apr 2017, 14:41:26 Elena Vilchik wrote:
\[~jeanchristophe.collet] Could you please update message of this rule so that it is a bit softer and gives some details of the problem? (issue \https://github.com/SonarSource/sonar-javascript/issues/572)

endif::env-github,rspecator-view[]
