== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,csharp]
----
int GetBestNumber()
{
  return 12;  // Noncompliant
}
----

=== Compliant solution

[source,csharp]
----
const int BestNumber = 12;
----
or

[source,csharp]
----
static readonly int BestNumber = 12;
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../highlighting.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
