include::../description.adoc[]

== Noncompliant Code Example

----
int GetBestNumber()
{
  return 12;  // Noncompliant
}
----

== Compliant Solution

----
const int BestNumber = 12;
----
or

----
static readonly int BestNumber = 12;
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../highlighting.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
