Host variables are used for passing data back and forth from the database to a COBOL program via SQL. In order to keep that transmission clear and free of errors, the format of each host variable should match its corresponding database column type. 


This rule raises an issue when column and corresponding host variable don't match in terms of numeric vs character data, and when the host variable is smaller than the column width.


== Noncompliant Code Example

----
TODO
----


== Compliant Solution

----
TODO
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

include::highlighting.adoc[]

'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
