
== Why is this an issue?

The preprocessor performs two main tasks: Handling preprocessor directives and expanding macros. Trying to do both simultaneously can be problematic: when a macro argument contains something that could be interpreted as a preprocessor directive, the behavior is undefined.

== How to fix it

=== Code examples

==== Noncompliant code example

[source,cpp]
----
#define LOG(msg) if(logEnabled) log(msg);
int main()
{
   LOG(
#ifdef __cplusplus // Noncompliant, undefined behavior
   "This library is built in C++ mode"
#else
   "This library is built in C mode"
#endif
    );
}
----

==== Compliant solution

The solution to avoid using proprocessing directives in a macro argument is often to use them before calling the macro. For instance, the previous code could be rewritten:
[source,cpp]
----
#define LOG(msg) if(logEnabled) log(msg);
int main()
{
#ifdef __cplusplus
   #define MODE "C++"
#else
   #define MODE "C"
#endif

   LOG("This library is built in " MODE " mode");

#undef MODE
}

----

== Resources
=== External coding guidelines
// MISRA 23 : Replace by an automated mechanism M23_043
* MISRA {cpp}:2023, 19.3.5 - Tokens that look like a preprocessing directive shall not occur within a macro argument
* MISRA C:2012, 20.6 - Tokens that look like a preprocessing directive shall not occur within a macro argument
* https://wiki.sei.cmu.edu/confluence/display/c/PRE32-C.+Do+not+use+preprocessor+directives+in+invocations+of+function-like+macros[CERT, PRE32-C.] - Do not use preprocessor directives in invocations of function-like macros

