
include::../../../shared_content/secrets/description.adoc[]

== Why is this an issue?

include::../../../shared_content/secrets/rationale.adoc[]

=== What is the potential impact?

// Set value that can be used to refer to the type of secret in, for example:
// "An attacker can use this {secret_type} to ..."
:secret_type: credentials

// Where possible, use predefined content for common impacts. This content can
// be found in the folder "shared_content/secrets/impact".
// When using predefined content, search for any required variables to be set and include them in this file.
// Not adding them will not trigger warnings.

include::../../../shared_content/secrets/impact/oauth_token_compromise.adoc[]

=== Secret storage best practice

It is best practice to avoid hard-coding secrets into an application. This is
true even in situations where the secret cannot be kept completely safe, such as
where is must be distributed as part of a client application.

Storing the secret outside of the application code makes it easier to manage
which secret is being used. For example, it can help to ensure that a production
secret is not accidentally used during development.

== How to fix it

include::../../../shared_content/secrets/fix/revoke.adoc[]

include::../../../shared_content/secrets/fix/vault.adoc[]

include::../../../shared_content/secrets/fix/oauth_pkce.adoc[]

**Disable the implicit grant flow**

Prior to the introduction of PKCE, the implicit grant flow was the recommended
solution for applications which cannot secure a `client_secret`. Dropbox allows
the implicit flow to be used for legacy compatibility purposes but PKCE should
be used for all new applications. The ability to use the implicit grant flow can
be disabled in the Dropbox App Console.

=== Code examples

==== Noncompliant code example

[source,java,diff-id=1,diff-type=noncompliant,subs="attributes"]
----
props.set("dropbox.app_key", "vqg7x6qd2pviu4r")
props.set("dropbox.app_secret", "d5r2k2le2ixosna") // Noncompliant
----

==== Compliant solution

[source,java,diff-id=1,diff-type=compliant,subs="attributes"]
----
props.set("dropbox.app_key", System.getenv("DROPBOX_APP_KEY"))
props.set("dropbox.app_secret", System.getenv("DROPBOX_APP_SECRET"))
----

//=== How does this work?

//=== Pitfalls

//=== Going the extra mile

== Resources

=== Documentation

* Dropbox - https://www.dropbox.com/lp/developers/reference/oauth-guide[OAuth Guide]
* IETF Datatracker - https://datatracker.ietf.org/doc/html/rfc7636[RFC 7636: Proof Key for Code Exchange]
* IETF Datatracker - https://datatracker.ietf.org/doc/html/rfc6749#section-4.2[RFC 6749: The OAuth 2.0 Authorization Framework] - Implicit Grant

include::../../../shared_content/secrets/resources/standards.adoc[]

//=== Benchmarks
