== How to fix it in Spring

=== Code examples

In a Spring Security's context, session fixation protection is enabled by default but can be disabled with ``++sessionFixation().none()++`` method.



==== Noncompliant code example

[source,java,diff-id=1,diff-type=noncompliant]
----
@Override
protected void configure(HttpSecurity http) throws Exception {
   http.sessionManagement()
     .sessionFixation().none(); // Noncompliant: the existing session will continue
}
----

==== Compliant solution

In a Spring Security's context, session fixation protection can be enabled as follows:

[source,java,diff-id=1,diff-type=compliant]
----
@Override
protected void configure(HttpSecurity http) throws Exception {
  http.sessionManagement()
     .sessionFixation().migrateSession(); // Compliant: a new session is created, the old one is invalidated and the attributes from the old session are copied over.
}
----
