== How to fix it in Spring

=== Code examples

In a Spring Security's context, session fixation protection is enabled by default but can be disabled with ``++sessionFixation().none()++`` method. Session fixation protection can also be enabled explicitly with ``++migrateSession()++`` but is superfluous in most cases.

==== Noncompliant code example

[source,java,diff-id=1,diff-type=noncompliant]
----
@Override
protected void configure(HttpSecurity http) throws Exception {
   http.sessionManagement()
     .sessionFixation().none(); // Noncompliant: the existing session will continue
}
----

==== Compliant solution

[source,java,diff-id=1,diff-type=compliant]
----
@Override
protected void configure(HttpSecurity http) throws Exception {
  http.sessionManagement()
     .sessionFixation().migrateSession();
}
----

=== How does this work?

include::../../common/fix/new-session.adoc[]

In Spring, calling ``++migrateSession()++`` explicitly is not necessary in most cases, as it automatically migrates session attributes to a new session upon successful authentication. The ``++migrateSession()++`` method is typically explicitly used when you want to manually trigger the migration of session attributes to a new session outside of the authentication process.
