<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p><code>GO TO</code> should not be used to transfer control outside the current module, because any implied <code>EXIT</code> points will then be ignored. A module is either a section, a paragraph, or a set of paragraphs called with the <code>PERFORM ... THRU ...</code> statement.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cobol" data-lang="cobol">         PERFORM PARAGRAPH1 THRU PARAGRAPH3.     &gt; code contained between PARAGRAPH1 and PARAGRAPH3 is now considered as a module
         EXIT PROGRAM.

        PARAGRAPH1.
         MOVE A TO B.
         IF SOMETHING
           GO TO PARAGRAPH3     &gt;OK
         END-IF.
         IF SOMETHING-ELSE
           GO TO PARAGRAPH4     &gt;NOK as we leave the module called with "PERFORM PARGRAPH1 THRU PARAGRAPH3" statement
         END-IF.

        PARAGRAPH2.
         MOVE A TO B.

        PARAGRAPH3.
         EXIT.

        PARAGRAPH4.</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Refactor this code to stop using GO TO as an unstructured control flow statement.</p>
</div>
</div>
</div>
</div>