include::../summary.adoc[]

== Why is this an issue?

include::../description.adoc[]

== How to fix it

=== Code examples

==== Noncompliant code example

[source,yaml,diff-id=1,diff-type=noncompliant]
----
name: Cleanup

on: push

jobs:
  cleanup:
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v4
      - name: Remove files
        run: rm *  # Noncompliant
----

==== Compliant solution

[source,yaml,diff-id=1,diff-type=compliant]
----
name: Cleanup

on: push

jobs:
  cleanup:
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v4
      - name: Remove files
        run: |
          rm ./*
          rm -- *
----

== Resources

include::../resources.adoc[]

ifdef::env-github,rspecator-view[]
'''
== Implementation Specification
(visible only on this page)

=== Message

include::../message.adoc[]

=== Highlighting

include::../highlighting.adoc[]

'''
endif::env-github,rspecator-view[]
