<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>It is unspecified behaviour if the evaluation of an expression with <code>enum</code> underlying type yields a value which does not correspond to one of the enumerators of the enumeration.</p>
</div>
<div class="paragraph">
<p>Additionally, other rules in this standard assume that objects of <code>enum</code> type only contain values corresponding to the enumerators. This rule ensures the validity of these assumptions.</p>
</div>
<div class="paragraph">
<p>One way of ensuring compliance when converting to an enumeration is to use a switch statement.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">enum Numbers {
  ONE, TWO
};

int function() {
        enum Numbers c = (enum Numbers)(TWO + TWO); // Noncompliant, operation might yield an invalid value.
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">enum Numbers {
  ONE, TWO
};

enum Numbers convert ( int16_t v )
{
  switch ( v )
  {
  case 0:
    return ONE;
  case 1:
    return TWO;
  default:
    throw ENUM_ERROR;
  }
}

int function() {
  enum Numbers c = convert(ONE + TWO); // Compliant
}</code></pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p>MISRA C&#43;&#43;:2008, 7-2-1</p>
</li>
</ul>
</div>
<hr>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Refactor this conversion from "xxx" to "yyy".</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_23_oct_2014_161707_ann_campbell_wrote">on 23 Oct 2014, 16:17:07 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~samuel.mercier] FYI, I&#8217;ve updated the code examples.</p>
</div>
<div class="paragraph">
<p>Noncompliant, I changed to actually yield an invalid valid. Since enums start at 0, ONE + TWO yields TWO, which is clearly valid. But TWO + TWO does not.</p>
</div>
<div class="paragraph">
<p>Also modified compliant to fix a typo.</p>
</div>
</div>
</div>
</div>