== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,cpp]
----
void func(int n, int m) {

  switch (n) {
    case 1:
      // ...
    case 2:
      // ...
    case 3:
      switch (m) {  // Noncompliant
    case 4:  // Bad indentation makes this particularly hard to read properly
      // ...
    case 5:
      // ...
    case 6:
      // ...
    }
    case 4:
      // ...
    default:
      // ...
  }
}
----

=== Compliant solution

[source,cpp]
----
void func(int n, int m) {

  switch (n) {
    case 1:
      // ...
    case 2:
      // ...
    case 3:
      int m2 = handle_m(m);
    case 4:
      // ...
    default:
      // ...
  }
}
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

=== on 2 Oct 2014, 14:01:36 Ann Campbell wrote:
\[~evgeny.mandrikov] why duplicate the description here? The purpose of these sub-tasks is to hold variations.

=== on 2 Oct 2014, 14:15:59 Evgeny Mandrikov wrote:
\[~ann.campbell.2] there is variation - "[subroutine|method|function]" replaced by "function".

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
