== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,php]
----
$a = $b + $c; // This is a trailing comment that can be very very long
----

=== Compliant solution

[source,php]
----
// This very long comment is better placed before the line of code
$a = $b + $c;
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

=== Parameters

.legalTrailingCommentPattern
****

----
^(//|#)\s*+[^\s]++$
----

Pattern for text of trailing comments that are allowed. By default, comments containing only one word.
****


'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
