This rule raises an issue when a method is defined with a reserved keyword name surrounded by quotes.

== Why is this an issue?

While Groovy allows defining methods with reserved keyword names by surrounding them with quotes, this practice creates several problems:

* *Reduced readability*: Quoted method names look unusual and can confuse developers who are not familiar with this Groovy feature.
* *Special calling syntax*: Methods with quoted keyword names require qualified calls using the `this.` prefix, making the calling code more verbose and inconsistent.
* *Maintenance burden*: Code becomes harder to understand and maintain when method names don't clearly express their purpose.
* *Inconsistent style*: Mixing quoted and unquoted method names creates inconsistent code style within a project.

The Groovy documentation itself states that "using such names might be confusing and is often best to avoid." This feature was primarily intended for specific Java integration scenarios and Domain Specific Language (DSL) use cases, not for general application development.

=== What is the potential impact?

Using reserved keywords as method names can lead to:

* *Confusion for team members*: Developers may not understand why certain methods require special calling syntax
* *Increased cognitive load*: Reading and understanding code becomes more difficult
* *Maintenance issues*: Future developers may struggle to work with such code
* *Inconsistent codebase*: Mixed naming conventions reduce overall code quality

== How to fix it

Replace the quoted reserved keyword with a descriptive method name that clearly indicates the method's purpose. Use standard naming conventions like camelCase for better readability.

=== Code examples

==== Noncompliant code example

[source,groovy,diff-id=1,diff-type=noncompliant]
----
def "abstract"() { // Noncompliant
    return true
}

// Requires special qualification when called
this.abstract()
----

==== Compliant solution

[source,groovy,diff-id=1,diff-type=compliant]
----
def isAbstract() {
    return true
}

// Clear and readable calling syntax
isAbstract()
----

== Resources

=== Documentation

 * Groovy Language Documentation - Keywords - https://groovy-lang.org/syntax.html#_keywords[Official Groovy documentation explaining reserved keywords and their usage]

 * Groovy Language Documentation - Identifiers - https://groovy-lang.org/syntax.html#_identifiers[Official documentation on identifiers and quoted identifiers in Groovy]

=== Related rules

 * RSPEC-4413 - https://rules.sonarsource.com/tsql/RSPEC-4413/[Similar rule for T-SQL about avoiding reserved keywords as identifiers]

 * RSPEC-1527 - https://rules.sonarsource.com/javascript/RSPEC-1527/[JavaScript rule about avoiding future reserved words as identifiers]
