<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Catching an exception class by value rather than by reference can cause several problems:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>Memory is allocated unnecessarily: catching by value creates a copy of the exception object, which is destroyed at the exit of the catch block.</p>
</li>
<li>
<p>Slicing occurs: the copy will be an instance of the exception base class rather than the potentially more specific exception class initially caught. So it may lead to a loss of precision as any additional data or functionality offered by the subclass will not be accessible.</p>
</li>
<li>
<p>Copying the exception class may throw an exception, leading to unexpected behavior.</p>
</li>
</ul>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when an exception is caught by value instead of by reference.</p>
</div>
<div class="paragraph">
<p>It is also best to avoid catching an exception by pointer (implying throwing by pointer) because it means the catch clause must manage the memory held by the pointer. The rule <a data-rspec-id="S1035" class="rspec-auto-link">S1035</a> detects this situation.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">try {
  // ...
} catch (ExceptionClass ex) { // Noncompliant
  //...
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">try {
  // ...
} catch (ExceptionClass &amp;ex) {
  //...
}</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_standards">Standards</h3>
<div class="ulist">
<ul>
<li>
<p>CERT - <a href="https://wiki.sei.cmu.edu/confluence/x/SXs-BQ">ERR61-CPP. Catch exceptions by lvalue reference</a></p>
</li>
</ul>
</div>
</div>
<div class="sect2">
<h3 id="_external_coding_guidelines">External coding guidelines</h3>
<div class="ulist">
<ul>
<li>
<p>C&#43;&#43; Core Guidelines - <a href="https://github.com/isocpp/CppCoreGuidelines/blob/e49158a/CppCoreGuidelines.md#e15-throw-by-value-catch-exceptions-from-a-hierarchy-by-reference">E.15: Throw by value, catch exceptions from a hierarchy by reference</a></p>
</li>
<li>
<p>MISRA C&#43;&#43;:2008, 15-3-5 - A class type exception shall always be caught by reference</p>
</li>
</ul>
</div>
</div>
<div class="sect2">
<h3 id="_related_rules">Related rules</h3>
<div class="ulist">
<ul>
<li>
<p><a data-rspec-id="S1035" class="rspec-auto-link">S1035</a> - An exception object should not have a pointer type</p>
</li>
</ul>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Catch the exception by reference.</p>
</div>
</div>
</div>
</div>