Getters that recursively call themselves are likely a bug and should be changed, to not be recursive.

== Why is this an issue?

Unlike a method or a function, a getter doesn't receive any input parameter. As a consequence, when invoked, it has no way to break the recursion, based on its input. Therefore, if a getter calls itself, it will most likely cause an infinite loop, resulting into a stack overflow.

This rule detects direct recursion invocations in the context of instance and static getters of classes, mixins, extensions, and extension types.

== How to fix it

It's likely that the getter for `x` was supposed to access a private backing field ``++_x++``, but instead calls itself. To fix this, change the getter to return the backing field.

=== Code examples

==== Noncompliant code example

[source,dart,diff-id=1,diff-type=noncompliant]
----
class Person {
    String _name = '';
    String get name => name; // Noncompliant
    set name(String value) => name = value;
}
----

==== Compliant solution

[source,dart,diff-id=1,diff-type=compliant]
----
class Person {
    String _name = '';
    String get name => _name;
    set name(String value) => name = value;
}
----

== Resources

=== Documentation

* Dart Docs - https://dart.dev/tools/linter-rules/recursive_getters[Dart Linter rule - recursive_getters]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

* The getter 'getter' recursively returns itself.

=== Highlighting

The recursive call.

'''
== Comments And Links
(visible only on this page)

endif::env-github,rspecator-view[]
