<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>The storage layout used for floating-point values may vary from one compiler to another, and therefore no floating-point comparisons or manipulations should be made which rely directly on the way the values are stored. The in-built operators and functions, which hide the storage details from the developer, should be used.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">float32_t My_fabs ( float32_t f )
{
  uint8_t * pB = reinterpret_cast&lt; uint8_t * &gt;( &amp;f );
  *( pB + 3 ) &amp;= 0x7f; // Non-compliant â€“ generate the absolute value of an IEEE-754 float value.
  return f;
}</code></pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p>MISRA C:2004, 12.12 - The underlying bit representations of floating-point values shall not be used.</p>
</li>
<li>
<p>MISRA C&#43;&#43;:2008, 3-9-3 - The underlying big representations of floating-point values shall not be used.</p>
</li>
<li>
<p><a href="https://wiki.sei.cmu.edu/confluence/x/kdUxBQ">CERT, FLP37-C.</a> - Do not use object representations to compare floating-point values</p>
</li>
</ul>
</div>
<hr>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Use a built-in operator or a function to access or manipulate this [float|double].</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_17_oct_2014_154505_ann_campbell_wrote">on 17 Oct 2014, 15:45:05 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>FYI [~samuel.mercier] I&#8217;ve moved this to Compiler-related Portability</p>
</div>
</div>
</div>
</div>