Backticks are a deprecated alias for ``++repr()++``. Don't use them any more, the syntax was removed in Python 3.0.


== Noncompliant Code Example

[source,python]
----
return `num`  # Noncompliant
----


== Compliant Solution

[source,python]
----
return repr(num)
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

endif::env-github,rspecator-view[]
