== Why is this an issue?

Backticks are a deprecated alias for ``++repr()++``. Don't use them any more, the syntax was removed in Python 3.0.


=== Noncompliant code example

[source,python]
----
return `num`  # Noncompliant
----


=== Compliant solution

[source,python]
----
return repr(num)
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Use "repr" instead.


endif::env-github,rspecator-view[]
