``++std::string++`` and ``++std::string_view++`` find-related methods return ``++npos++`` if the substring was not found. Therefore, their returned values should never be treated as boolean. This rule raises an issue when objects returned from the following methods are treated as if they were of boolean type:


* ``++find++``
* ``++rfind++``
* ``++find_first_of++``
* ``++find_last_of++``
* ``++find_first_not_of++``
* ``++find_last_not_of++``


== Noncompliant Code Example

----
void f() {
  std::string s = "";
  if (s.find("42")) { // Noncompliant
    // this branch is taken even if "s" doesn't contain "42"
  }
}
----


== Compliant Solution

----
void f() {
  std::string s = "";
  if (s.find("42") != std::string::npos) {
    // this branch is correctly not taken
  }
}
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

endif::env-github,rspecator-view[]
