Regardless of scope, no identifier with ``++static++`` storage duration should be re-used across any source files in the project. This includes objects or functions with external linkage and any objects or functions with the static storage class specifier.


While the compiler can understand this and is in no way confused, the possibility exists for the developer to incorrectly associate unrelated variables with the same name.


== Noncompliant Code Example

----
// file1.c
extern int value;

// file2.c
static int value = 0; // Noncompliant
----


== See

* MISRA C:2004, 5.5
* MISRA {cpp}:2008, 2-10-5

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

endif::env-github,rspecator-view[]
