<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when the name of a local variable matches the name of a builtin.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Defining a variable with the same name as a built-in symbol will "shadow" it. That means that the builtin will no longer be accessible through its original name, having locally been replaced by the variable.</p>
</div>
<div class="paragraph">
<p>Shadowing a builtin makes the code more difficult to read and maintain. It may also be a source of bugs as you can reference the builtin by mistake.</p>
</div>
<div class="paragraph">
<p>It is sometimes acceptable to shadow a builtin to improve the readability of a public API or to support multiple versions of a library. In these cases, benefits are greater than the maintainability cost. This should, however, be done with care.</p>
</div>
<div class="paragraph">
<p>It is generally not a good practice to shadow builtins with variables which are local to a function or method. These variables are not public and can easily be renamed, thus reducing the confusion and making the code less error-prone.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">def a_function():
    int = 42  # Noncompliant; int is a builtin</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">def a_function():
    value = 42</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Python documentation - <a href="https://docs.python.org/3.8/library/stdtypes.html">Built-in Types</a></p>
</li>
<li>
<p>Python documentation - <a href="https://docs.python.org/3/library/functions.html">Built-in Functions</a></p>
</li>
</ul>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Rename this variable; it shadows a builtin.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>Primary:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>The first definition of a variable shadowing a builtin in a local scope (function)</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>Secondary:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>Any other assignment to the variable.</p>
</li>
<li>
<p>message: "Variable also assigned here."</p>
</li>
</ul>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_relates_to_s2137">relates to: <a data-rspec-id="S2137" class="rspec-auto-link">S2137</a></h3>

</div>
<div class="sect2">
<h3 id="_on_30_apr_2020_191203_nicolas_harraudeau_wrote">on 30 Apr 2020, 19:12:03 Nicolas Harraudeau wrote:</h3>
<div class="paragraph">
<p>This rule is similar to <a data-rspec-id="S2137" class="rspec-auto-link">RSPEC-2137</a> but it is a code smell because shadowing builtins will not make your code crash most of the time.</p>
</div>
</div>
</div>
</div>