== Why is this an issue?

Object literal syntax, which initializes an object's properties inside the object declaration is cleaner and clearer than the alternative: creating an empty object, and then giving it properties one by one.


An issue is raised when the following pattern is met:


* An empty object is created.
* A consecutive single-line statement adds a property to the created object.


=== Noncompliant code example

[source,javascript]
----
let person = {};  // Noncompliant
person.firstName = "John";
person.middleInitial = "Q";
person.lastName = "Public";
----


=== Compliant solution

[source,javascript]
----
let person = {
  firstName: "John",
  middleInitial: "Q",
  lastName: "Public",
}
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
