include::../description.adoc[]

== Noncompliant Code Example

[source,cobol]
----
SELECT * 
       FROM persons 
       INTO newyorkers 
       WHERE city = 'NEW YORK'
----

== Compliant Solution

[source,cobol]
----
SELECT firstname, lastname 
       FROM persons 
       INTO newyorkers
       WHERE city = 'NEW YORK'
----
ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

endif::env-github,rspecator-view[]
