include::../description_with_reference_to_s5998.adoc[]

== Noncompliant Code Example

----
Pattern.compile("a|b|c"); // Noncompliant
----


== Compliant Solution

----
Pattern.compile("[abc]");
// or
Pattern.compile("[a-c]");
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

include::highlighting.adoc[]

endif::env-github,rspecator-view[]
