== Why is this an issue?

include::../description_with_reference_to_s5998.adoc[]

=== Noncompliant code example

[source,java]
----
Pattern.compile("a|b|c"); // Noncompliant
----


=== Compliant solution

[source,java]
----
Pattern.compile("[abc]");
// or
Pattern.compile("[a-c]");
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Replace this alternation with a character class.


=== Highlighting

The alternation


endif::env-github,rspecator-view[]
