include::../description.adoc[]

== Noncompliant Code Example

----
Pattern.compile("a|b|c"); // Noncompliant
----


== Compliant Solution

----
Pattern.compile("[abc]");
// or
Pattern.compile("[a-c]");
----

