<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>As stated in the documentation, the methods <a href="https://docs.oracle.com/javase/8/docs/api/java/util/Objects.html#isNull-java.lang.Object-">Objects.isNull</a> and <a href="https://docs.oracle.com/javase/8/docs/api/java/util/Objects.html#nonNull-java.lang.Object-">Objects.nonNull</a> only exist to be used as <a href="https://docs.oracle.com/javase/8/docs/api/java/util/function/Predicate.html">Predicate</a>. When a predicate is not needed, a simple null check should be used, resulting in more concise and readable code.</p>
</div>
<div class="paragraph">
<p>This rule reports an issue when one of the two methods is not used as a Predicate.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">if (Objects.isNull(object))

if (Objects.nonNull(object))</code></pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">if (object == null)

if (object != null)

list.stream().filter(Object::nonNull)</code></pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_see">See</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p><a href="https://docs.oracle.com/javase/8/docs/api/java/util/Objects.html">Oracle Java 8, Objects</a></p>
</li>
</ul>
</div>
</div>
</div>