== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,php]
----
function compute($a, $a, $c) { // Noncompliant
}
----

=== Compliant solution

[source,php]
----
function compute($a, $b, $c) { // Compliant
}
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
