include::../description.adoc[]

== Noncompliant Code Example

----
function compute($a, $a, $c) { // Noncompliant
}
----

== Compliant Solution

----
function compute($a, $b, $c) { // Compliant
}
----

ifdef::env-github,rspecator-view[]
'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
