== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,javascript]
----
var CryptoJS = require("crypto-js");

var hash = CryptoJS.MD5("Message");
...
var hash = CryptoJS.SHA1("Message");
----

=== Compliant solution

[source,javascript]
----
var hash = CryptoJS.SHA256("Message");
----

include::../see.adoc[]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
