== Why is this an issue?

``++NotImplemented++`` is a constant which is intended to be used only by comparison methods such as ``++__lt__++``. Use it instead of  ``++NotImplementedError++``, which is an exception, and callers will have a hard time using your code.


=== Noncompliant code example

[source,python]
----
class MyClass:
    def do_something(self):
        raise NotImplemented("Haven't gotten this far yet.")  #Noncompliant
----


=== Compliant solution

[source,python]
----
class MyClass:
    def do_something(self):
        raise NotImplementedError("Haven't gotten this far yet.")  
----



ifdef::env-github,rspecator-view[]
'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
