include::../common/why.adoc[]

include::../common/how.adoc[]

=== Code examples

==== Noncompliant code example

[source,javascript,diff-id=1,diff-type=noncompliant]
----
function MyImage() {
    return <img src="sunrise.jpg" alt="image of a sunrise" />; // Noncompliant: "Image, image of a sunrise"
}
----

==== Compliant solution

[source,javascript,diff-id=1,diff-type=compliant]
----
function MyImage() {
    return <img src="sunrise.jpg" alt="a sunrise over a mountain range" />;
}
----

== Resources
=== Documentation

* MDN web docs - https://developer.mozilla.org/en-US/docs/Web/HTML/Element/img[img element]
* MDN web docs - https://developer.mozilla.org/en-US/docs/Web/API/HTMLImageElement/alt[alt property]
* WebAIM - https://webaim.org/techniques/alttext/[Alternative Text]
