include::../../../shared_content/secrets/description.adoc[]

== Why is this an issue?

include::../../../shared_content/secrets/rationale.adoc[]

When an attacker gains access to a Docker Hub personal access token and the username of the account, they can gain access to all container images accessible to that account.

=== What is the potential impact?

In case of a leaked token, an attacker can read out all private images and are also able to store new malicious images in the registry. This can have multiple severe consequences.

==== Compromise of sensitive source code

Docker Hub is often used to store private container images. If a personal access token is leaked, it can be used by unauthorized individuals to gain access to these images. Not only does this allow a malicious person to access and use internal projects, but it can also enable them to leak sensitive source code, proprietary binaries, and other confidential resources belonging to these projects. This can lead to intellectual property theft, unauthorized modifications, or even sabotage of your software.

If these private images contain other secrets, it might even lead to further breaches in the organization's services.

include::../../../shared_content/secrets/impact/supply_chain_attack.adoc[]

== How to fix it

include::../../../shared_content/secrets/fix/revoke.adoc[]

include::../../../shared_content/secrets/fix/vault.adoc[]

=== Code examples

==== Noncompliant code example

[source,yaml,diff-id=1,diff-type=noncompliant]
----
steps:
  - name: Login to DockerHub
    uses: docker/login-action@v2
    with:
      username: mobythewhale
      password: dckr_pat_cq7wQZcv9xZkVlxMhDTcTV00CDo
----

==== Compliant solution

[source,yaml,diff-id=1,diff-type=compliant]
----
steps:
  - name: Login to DockerHub
    uses: docker/login-action@v2
    with:
      username: ${{ secrets.dockerUsername }}
      password: ${{ secrets.dockerAccessToken }}
----

//=== How does this work?

//=== Pitfalls

//=== Going the extra mile

== Resources

=== Documentation

Docker docs - https://docs.docker.com/docker-hub/access-tokens/[Create and manage access tokens]

include::../../../shared_content/secrets/resources/standards.adoc[]

//=== Benchmarks
