== Why is this an issue?

This includes file and I/O functions ``++fgetpos++``, ``++fopen++``, ``++ftell++``, ``++gets++``, ``++perror++``, ``++remove++``, ``++rename++``, etc.


Streams and file I/O have a large number of unspecified, undefined and implementation-defined behaviors associated with them.


=== Noncompliant code example

[source,cpp]
----
#include <cstdio>  /* Noncompliant */

void fn()
{
  char_t array[10];
  gets(array);           /* Can lead to buffer over-run */
}
----


== Resources

* MISRA {cpp}:2008, 27-0-1 - The stream input/output library _<cstdio>_ shall not be used.


ifdef::env-github,rspecator-view[]
'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
