<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Using both <code>SPACEA</code> (space-after) and <code>SPACEB</code> (space-before) in the same printer file can make it difficult to understand how many blank lines will be printed where. Therefore either one command or the other should be used but not both.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-rpg" data-lang="rpg">A          R REP001                    TEXT('Detail line.       ')
A                                      SPACEB(1)
A                                      SPACEA(1)
A                                     2'Additional Details for Student :'
A            VAR01          6        36TEXT('Additional Details for +
A                                          Student')</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-rpg" data-lang="rpg">A          R REP001                    TEXT('Detail line.       ')
A                                      SPACEB(1)
A                                     2'Additional Details for Student :'
A            VAR01          6        36TEXT('Additional Details for +
A                                          Student')
A                                      SPACEB(1)</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>There are n SPACEA and m SPACEB commands in this file.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_22_sep_2014_065100_ann_campbell_wrote">on 22 Sep 2014, 06:51:00 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>Note that this rule is actually for DDS, the record-format specification for RPG, rather than for RPG itself.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_12_oct_2014_175937_freddy_mallet_wrote">on 12 Oct 2014, 17:59:37 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>@Ann, what do you mean by DDS because anyway we&#8217;re going to analyze only RPG programs ?</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_12_oct_2014_222044_ann_campbell_wrote">on 12 Oct 2014, 22:20:44 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~pierre-yves.nicolas] has said that he needs to analyze DDS eventually to have a full symbol table for RPG since DDS files describe data structures [~freddy.mallet]</p>
</div>
</div>
</div>
</div>