<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>TODO</p>
</div>
<div class="paragraph">
<p>Permettre une gestion standardisée de la gestion des dates dans les traitements. Tous les traitements doivent utiliser le module MGDATR03 s’ils ont besoin d’utiliser la date du jour.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cobol" data-lang="cobol">ACCEPT … FROM DATE
ACCEPT … FROM TIME
MOVE CURRENT-DATE TO …</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_27_may_2014_091810_freddy_mallet_wrote">on 27 May 2014, 09:18:10 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>This rule is specific to the BNP context. We should provide an XPath rule to get the same result.</p>
</div>
</div>
</div>
</div>