=== is related to: S948

=== on 16 Oct 2014, 13:13:53 Ann Campbell wrote:
\[~samuel.mercier] please:

* fill in the appropriate reference field(s).
* provide a See section.
* use double curly braces around code in the description, e.g. NULL.
* use the standard section titles: Noncompliant Code Example

=== on 17 Oct 2014, 09:21:22 Samuel Mercier wrote:
\[~ann.campbell.2] about the standard section titles for code, I wonder if it is appropriate.


The first line is compliant, and we want to show it. Since it is compliant it shouldn't be in the noncompliant part. But since it is not a solution, it shouldn't be in the Solution either. Also the current example is a copy-paste of the MISRA document, so we currently stick better to the spec.

