== Why is this an issue?

In {cpp}, the literal 0 is both an integer type and the null-pointer-constant. To meet developer expectations, ``++NULL++`` should be used as the null-pointer-constant, and 0 for the integer zero.


=== Noncompliant code example

[source,cpp]
----
#include <cstddef>

void f1 ( int32_t );
void f2 ( int32_t * );
void f3 ( )
{ 
  f1 ( 0 ); // Compliant, integer expected
  f2 ( 0 ); // Noncompliant, 0 used as the null pointer constant
  f2 ( NULL ); // Compliant, pointer expected
}
----


== Resources

* MISRA {cpp}:2008, 4-10-2 - Literal zero (0) shall not be used as the null-pointer-constant.


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Replace this use of 0 with "NULL".


'''
== Comments And Links
(visible only on this page)

=== is related to: S948

=== on 16 Oct 2014, 13:13:53 Ann Campbell wrote:
\[~samuel.mercier] please:

* fill in the appropriate reference field(s).
* provide a See section.
* use double curly braces around code in the description, e.g. NULL.
* use the standard section titles: Noncompliant Code Example

=== on 17 Oct 2014, 09:21:22 Samuel Mercier wrote:
\[~ann.campbell.2] about the standard section titles for code, I wonder if it is appropriate.


The first line is compliant, and we want to show it. Since it is compliant it shouldn't be in the noncompliant part. But since it is not a solution, it shouldn't be in the Solution either. Also the current example is a copy-paste of the MISRA document, so we currently stick better to the spec.

endif::env-github,rspecator-view[]
