This rule raises an issue when a class method is called on an instance, or when an instance method is called on a class.

== Why is this an issue?

Ruby distinguishes between class methods and instance methods based on how they are defined and called.

Class methods are defined using `def self.method_name`, `class << self`, or `def ClassName.method_name`. These methods belong to the class itself and should be called directly on the class.

Instance methods are defined with `def method_name` inside a class. These methods belong to individual instances of the class and should be called on object instances.

When you call a class method on an instance, Ruby cannot find the method because it only exists on the class level. Similarly, calling an instance method on a class will fail because the method requires an instance context.

This mismatch results in a `NoMethodError` at runtime, which can cause application crashes and unexpected behavior.

=== What is the potential impact?

This issue causes runtime errors that crash the application with `NoMethodError: undefined method` exceptions. These errors can:

* Break application functionality and user workflows
* Cause service outages if not caught by error handling
* Lead to poor user experience with unexpected crashes
* Make debugging more difficult when the error occurs in production

== How to fix it

Call the class method on the class itself, not on an instance. Class methods are shared across all instances and should be accessed through the class.

=== Code examples

==== Noncompliant code example

[source,ruby,diff-id=1,diff-type=noncompliant]
----
class User
  def self.find_active
    where(active: true)
  end
end

user = User.new
active_users = user.find_active  # Noncompliant
----

==== Compliant solution

[source,ruby,diff-id=1,diff-type=compliant]
----
class User
  def self.find_active
    where(active: true)
  end
end

active_users = User.find_active
----

== Resources

=== Documentation

 * Ruby Class and Instance Methods - https://ruby-doc.org/core/doc/syntax/methods_rdoc.html[Official Ruby documentation on method definitions and calling conventions]

 * Understanding Ruby Class Methods vs Instance Methods - https://www.rubyguides.com/2018/11/class-vs-instance-methods/[Comprehensive guide explaining the differences between class and instance methods in Ruby]
