<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>In Dart, there are no keywords like <code>public</code>, <code>protected</code>, and <code>private</code>. To distinguish between public and private there&#8217;s a convention, if an identifier starts with an underscore (_), it&#8217;s private to its library. In the context of library prefixes there&#8217;s no notion of private, so starting it with underscore will only be source of confusion for the developers. That&#8217;s why it&#8217;s recommended to not start the library prefix with an underscore <code>_</code>.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-dart" data-lang="dart">import 'dart:io' as _io;
import 'dart:math' as _math;
import 'dart:js' as _js;</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-dart" data-lang="dart">import 'dart:io' as io;
import 'dart:math' as math;
import 'dart:js' as js;</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Dart Docs - <a href="https://dart.dev/tools/linter-rules/no_leading_underscores_for_library_prefixes">Dart Linter rule - no_leading_underscores_for_library_prefixes</a></p>
</li>
</ul>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="ulist">
<ul>
<li>
<p>The library prefix &lt;prefix&gt; starts with an underscore.</p>
</li>
</ul>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>Library prefix</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
</div>
</div>