This rule raises an issue when a comprehension is used to only copy a collection. Instead, the respective constructor should be used.

== Why is this an issue?

Python comprehensions are a concise way to create new collections while transforming or filtering elements. However, using a comprehension that copies elements from one collection to another without any transformation is less readable than using the respective constructor directly.

Therefore, when a comprehension is only copying elements, use the appropriate constructor instead:

    * Replace `[x for x in iterable]` with `list(iterable)`
    * Replace `{x for x in iterable}` with `set(iterable)`
    * Replace `{k: v for k, v in iterable.items()}` with `dict(iterable)`

== How to fix it

Replace comprehensions that copy elements from one collection to another with the respective constructor.

=== Code examples

==== Noncompliant code example

[source,python,diff-id=1,diff-type=noncompliant]
----
some_list = [1, 2, 3, 2]
[x for x in some_list] # Noncompliant
----

==== Compliant solution

[source,python,diff-id=1,diff-type=compliant]
----
some_list = [1, 2, 3, 2]
list(some_list) # Compliant
----

== Resources
=== Documentation

* Python Documentation - https://docs.python.org/3/library/stdtypes.html#dict[dict()]
* Python Documentation - https://docs.python.org/3/library/stdtypes.html#list[list()]
* Python Documentation - https://docs.python.org/3/library/stdtypes.html#set[set()]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

Determining whether a comprehension is just copying a collection or is modifying it in some way is not trivial. If all of the following conditions are met, the comprehension does not modify its collection:

* Filtering comprehensions
* Transforming comprehensions (e.g., multiplying each element by 2)
    * Dict comprehensions can also flip keys and values (e.g., `{v: k for k, v in d.items()}`)
* No multiple `for` clauses

Keep in mind that this rule also applies to constructors with a generator expression. For example `list(x for x in some_list)` should also raise.

=== Message
Replace this comprehension with `\{constructor}(\{iterable})`.

=== Highlighting
The comprehension or generator expression.

endif::env-github,rspecator-view[]
