``++!important++`` within keyframes declarations is completely ignored in some browsers and therefore it should not be used to be consistent among all browsers.


== Noncompliant Code Example

[source,css]
----
@keyframes kf {
  from { margin-top: 50px; }
  50%  { margin-top: 150px !important; } /* Noncompliant; ignored */
  to   { margin-top: 100px; }
}
----


== Compliant Solution

[source,css]
----
@keyframes kf {
  from { margin-top: 50px; }
  50%  { margin-top: 150px; }
  to   { margin-top: 100px; }
}
----


== See

* https://developer.mozilla.org/en-US/docs/Web/CSS/@keyframes#!important_in_a_keyframe[Mozilla Web Technology for Developers] - ``++!important++`` in a keyframe

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

include::highlighting.adoc[]

endif::env-github,rspecator-view[]
