== How to fix it

=== Code examples

include::../../common/fix/code-rationale.adoc[]

==== Noncompliant code example

[source,yaml,diff-id=1,diff-type=noncompliant]
----
name: Example

on:
  pull_request:

jobs:
  main:
    runs-on: ubuntu-latest
    steps:
      - name: example
        # Sensitive
        run: curl --insecure -O https://expired.example.com/downloads/install.sh
----

==== Compliant solution

[source,yaml,diff-id=1,diff-type=compliant]
----
name: Example

on:
  pull_request:

jobs:
  main:
    runs-on: ubuntu-latest
    steps:
      - name: example
        run: curl -O https://new.example.com/downloads/install.sh
----

==== Noncompliant code example

[source,yaml,diff-id=2,diff-type=noncompliant]
----
name: Example

on:
  pull_request:

jobs:
  main:
    runs-on: ubuntu-latest
    steps:
      - name: example
        # Sensitive
        run: wget --no-check-certificate https://expired.example.com/downloads/install.sh
----

==== Compliant solution

[source,yaml,diff-id=2,diff-type=compliant]
----
name: Example

on:
  pull_request:

jobs:
  main:
    runs-on: ubuntu-latest
    steps:
      - name: example
        run: wget https://new.example.com/downloads/install.sh
----

=== How does this work?

include::../../common/fix/validation.adoc[]
