Re-exporting is a feature that allows a file to import a component (object, function or primitive) from a module and add it to the exported components of the file's module. However, as the language allows a file to export a component without explicitly importing it, a reader may mistake a re-export with a simple export. 

== Noncompliant Code Example

----
export { foo } from 'someModule';
----

== Compliant Solution

----
import { foo } from 'someModule';
export { foo };
----
