Re-exporting is a feature that allows a file to import a component (object, function or primitive) from a module and add it to the exported components of the file's module. However, as the language allows a file to export a component without explicitly importing it, a reader may mistake a re-export with a simple export. 


== Noncompliant Code Example

[source,javascript]
----
export { foo } from 'someModule';
----


== Compliant Solution

[source,javascript]
----
import { foo } from 'someModule';
export { foo };
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

include::highlighting.adoc[]

endif::env-github,rspecator-view[]
