<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>The <code>DD DATA</code> statement continues reading in-stream data until it reaches the end delimiter (<code>/*</code> or the delimiter specified by the <code>DLM</code> parameter) or until it hits the end-of-file. This can lead to other JCL statements being mistakenly included in the data stream.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Always code a delimiter for <code>DD DATA</code> statements, either by using the default delimiter <code>/*</code> or coding a custom delimiter using the <code>DLM</code> parameter.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-jcl" data-lang="jcl">//STEP1 EXEC PGM=DOTHING
//SYSIN DD DATA
//ALPHA JOB ,MSGLEVEL=(1,1)
//NOPE  EXEC PGM=IEFBR14
//STEP2 EXEC PGM=IEFBR14
//* End of file</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-jcl" data-lang="jcl">//STEP1 EXEC PGM=DOTHING
//SYSIN DD DATA,DLM='&gt;&lt;'
//ALPHA JOB ,MSGLEVEL=(1,1)
//NOPE  EXEC PGM=IEFBR14
&gt;&lt;
//STEP2 EXEC PGM=IEFBR14
//* End of file</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p><a href="https://www.ibm.com/docs/en/zos/3.1.0?topic=statement-data-parameter">IBM z/OS MVS JCL Reference - DD DATA parameter</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>