:identifier_capital_plural: Local variables and function parameters
:identifier: local variable and function parameter
:identifier_plural: local variables and function parameters
:identifier_or: local variable or function parameter
:regex: ^\$[a-z][a-zA-Z0-9]*$


include::../introduction.adoc[]

include::../why-is-this-an-issue.adoc[]

=== Exceptions

The rule does not raise an issue on the usages of the blank identifier `$_` or the root context `$`.

include::../what-is-the-potential-impact.adoc[]

include::../how-to-fix-it.adoc[]

As a default, we suggest using the camelCase naming convention, as this is widely adopted for variables in Helm.


=== Code examples

==== Noncompliant code example

With the default regular expression ``{regex}``:

[source,text,diff-id=1,diff-type=noncompliant]
----
{{- $my_Variable := .Values.myVariableValue -}} # Noncompliant
----

[source,text,diff-id=2,diff-type=noncompliant]
----
apiVersion: v1
kind: ConfigMap
metadata:
  name: {{ .Release.Name }}-configmap
data:
  myvalue: "Hello World"
  {{- range $KeyNc, $VAL_NC := .Values.favorite }} # Noncompliant
  {{ $KeyNc }}: {{ $VAL_NC | quote }}
  {{- end }}
----

==== Compliant solution

[source,text,diff-id=1,diff-type=compliant]
----
{{- $myVariable := .Values.myVariableValue -}}
----

[source,text,diff-id=2,diff-type=compliant]
----
apiVersion: v1
kind: ConfigMap
metadata:
  name: {{ .Release.Name }}-configmap
data:
  myvalue: "Hello World"
  {{- range $keyC, $valC := .Values.favorite }}
  {{ $keyC }}: {{ $valC | quote }}
  {{- end }}
----

[source,text]
----
# Usage of blank identifier
{{- $_ := set .Values.ingress.annotations "kubernetes.io/ingress.class" .Values.ingress.className -}}

# Usage of root context
{{- $ := . -}}
----
== Resources

=== Documentation

* Helm documentation - https://helm.sh/docs/chart_template_guide/variables/[Helm - Variables]
* Go documentation - https://go.dev/doc/effective_go#names[Effective Go - Names]
* Wikipedia - https://en.wikipedia.org/wiki/Naming_convention_(programming)[Naming Convention (programming)]


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../parameters.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
