<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>The <code>LONG</code> and <code>LONG RAW</code> datatypes are deprecated and Oracle recommends to migrate them to the LOB datatypes <code>CLOB</code>, <code>NCLOB</code> or <code>BLOB</code>.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-sql" data-lang="sql">CREATE TABLE images(
  data LONG RAW
);</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-sql" data-lang="sql">CREATE TABLE images(
  data BLOB
);</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Migrate this "[LONG|LONG RAW]" datatype to a ["CLOB" or "NCLOB"|"BLOB"].</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_14_mar_2014_111736_dinesh_bolkensteyn_wrote">on 14 Mar 2014, 11:17:36 Dinesh Bolkensteyn wrote:</h3>
<div class="paragraph">
<p>see</p>
</div>
<div class="paragraph">
<p><a href="http://www.oracle-developer.net/display.php?id=430" class="bare">http://www.oracle-developer.net/display.php?id=430</a></p>
</div>
<div class="paragraph">
<p><a href="http://docs.oracle.com/cd/B28359_01/server.111/b28318/datatype.htm#CNCPT613" class="bare">http://docs.oracle.com/cd/B28359_01/server.111/b28318/datatype.htm#CNCPT613</a> and http://docs.oracle.com/cd/B28359_01/server.111/b28318/datatype.htm#CNCPT1831</p>
</div>
<div class="paragraph">
<p><a href="http://stackoverflow.com/questions/12722636/what-is-the-difference-between-long-and-long-raw-data-types-in-oracle" class="bare">http://stackoverflow.com/questions/12722636/what-is-the-difference-between-long-and-long-raw-data-types-in-oracle</a></p>
</div>
</div>
</div>
</div>