Octal escapes are included in a limited form. If the first digit is a 0, or if there are three octal digits, it is considered an octal escape. Otherwise, it is a group reference. As for string literals, octal escapes are always at most three digits in length.

== Why is this an issue?
Octal escapes character codes are easily confused with back-references.

== How to fix it
Use hexadecimal escapes to avoid confusion.

=== Code examples

==== Noncompliant code example

[source,python]
----
import re

match = re.match(r"\101", "A")
----

==== Compliant solution

[source,python]
----
import re

match = re.match(r"\x41", "A")
----

include::../implementation.adoc[]
