Content that doesn't change or that doesn't change often should be included using a mechanism which won't try to interpret it. Specifically, ``++<%@ include file="..." %>++``, which includes the file in the JSP servlet translation phase (i.e. it happens once), should be used instead of ``++<jsp:include page="..." />++``, which includes the page on the file, when the content is being served to the user.


== Noncompliant Code Example

----
<jsp:include page="header.jsp">  <!-- Noncompliant -->
----


== Compliant Solution

----
<%@ include file="header.jsp" %>
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

endif::env-github,rspecator-view[]
