<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>The <code>Monitor.Pulse</code> call releases the object on which it was called and wakes up the first thread waiting for the lock on that object. Significantly, it only releases <em>one</em> lock, and if multiple locks are held when it is called deadlocks could result.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-csharp" data-lang="csharp">public void doSomething(Object obj)
{
  lock (this) //first lock
  {
    lock (obj) {  // second lock
      // ...
      Monitor.Pulse(obj); // Noncompliant; only the second lock is released
    }
  }
}</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove this "Monitor.Pulse" call; it only releases the lock on "xxx", leaving the one on "yyy" intact.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_16_jun_2015_112135_tamas_vajk_wrote">on 16 Jun 2015, 11:21:35 Tamas Vajk wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2], could you please go through this C# specific version of the rule?</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_16_jun_2015_133503_ann_campbell_wrote">on 16 Jun 2015, 13:35:03 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>Looks good [~tamas.vajk]</p>
</div>
</div>
<div class="sect2">
<h3 id="_is_related_to_s2446">is related to: <a data-rspec-id="S2446" class="rspec-auto-link">S2446</a></h3>

</div>
<div class="sect2">
<h3 id="_on_11_jun_2015_185101_ann_campbell_wrote">on 11 Jun 2015, 18:51:01 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>CodePro: Notify method invoked while two locks are held</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_11_jun_2015_185146_ann_campbell_wrote">on 11 Jun 2015, 18:51:46 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>FYI [~tamas.vajk]. I wrote this for Java but I&#8217;m guessing it applies to C# as well, so I&#8217;ve targeted it to the language. Simply un-target it if it&#8217;s not, please.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_23_jan_2017_144408_tibor_blenessy_wrote">on 23 Jan 2017, 14:44:08 Tibor Blenessy wrote:</h3>
<div class="paragraph">
<p><code>notify</code> call is already covered in <a data-rspec-id="S2446" class="rspec-auto-link">RSPEC-2446</a> and there will be two issues raised on notify call in some cases. Therefore I will remove it from the implementation of this rule and change this RSPEC accordingly.</p>
</div>
</div>
</div>
</div>