Casting a base-class pointer/reference to a derived-class pointer/reference is commonly referred to as downcasting.


Downcasting can only be done using an explicit cast. It is usually achieved safely using ``++dynamic_cast++`` or unsafely using ``++static_cast++``.


``++dynamic_cast++`` achieve safe downcasting by performing a run-time check:

* If the cast memory contains an object of the expected derived type, the check succeeds. The result of the ``++dynamic_cast++`` points/refers to the derived object. 
* If the cast memory doesn't contain an object of the expected derived type, the check fails. If the ``++dynamic_cast++`` is used on a pointer, ``++nullptr++`` is returned. If it was used on a reference, ``++std::bad_cast++`` is thrown.

``++static_cast++`` doesn't do any runtime check; it is inherently unsafe. If the cast memory doesn't contain an object of the expected derived type,  your program enters the undefined behavior territory.


As a consequence, ``++dynamic_cast++`` should be preferred when downcasting.


Nonetheless, if you find yourself in a scenario where you want to avoid using ``++dynamic_cast++``, make sure that all the following conditions are met:

* You want to avoid performing run-time checks(e.g. for performance reasons).
* You have an alternative way to guarantee that the cast memory contains an object of the expected derived type.
* No virtual inheritance is involved.


== Noncompliant Code Example

----
struct Shape {
  ...
};

struct Rectangle : public Shape {
  double width;
  double height;
};

struct Circle : public Shape {
  double radius;
};

double computeArea(const Shape* shape) {
  const auto* rectangle = static_cast<const Rectangle*>(shape); // Noncompliant
  return rectangle->width * rectangle->height;
}
----


== Compliant Solution

----
struct Shape {
  ...
};

struct Rectangle : public Shape {
  double width;
  double height;
};

struct Circle : public Shape {
  int radius;
};

double computeArea(const Shape* shape) {
  if(const auto* rectangle = dynamic_cast<const Rectangle*>(shape)) { // Compliant
    return rectangle->width * rectangle->height; 
  }
  return 0;
}
----


== See

* https://github.com/isocpp/CppCoreGuidelines/blob/c553535fb8dda2839d13ab5f807ffbc66b63d67b/CppCoreGuidelines.md#type2-dont-use-static_cast-downcasts-use-dynamic_cast-instead[{cpp} Core Guidelines - Type safety profile - Type.2] - Don’t use static_cast to downcast. Use dynamic_cast instead.


ifdef::env-github,rspecator-view[]
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
