include::../description.adoc[]

== Noncompliant Code Example

----
my_number = 023 # Noncompliant. my_number will hold 19, not 23 - was this really expected?
----

== Compliant Solution

----
my_number = 23
----

== Exceptions

This rule does not apply to values smaller than 8 and octal values having 3 digits, since 3 digits octal values are often used as file permission masks.

Example:

----
permission_mask = 0777
day_of_month = 03
----

include::../see.adoc[]
