== Why is this an issue?

Using ``++MaybeUninit::uninit().assume_init()++`` results in undefined behavior for most types, since it bypasses initialization and assumes the data is valid. This can lead to unpredictable results and hard-to-diagnose bugs.

=== Code examples

==== Noncompliant code example

[source,rust,diff-id=1,diff-type=noncompliant]
----
use std::mem::MaybeUninit;

let _: usize = unsafe { MaybeUninit::uninit().assume_init() }; // Noncompliant: Unsafe code resulting in undefined behavior.
----

==== Compliant solution

[source,rust,diff-id=1,diff-type=compliant]
----
use std::mem::MaybeUninit;

let _: [MaybeUninit<bool>; 5] = unsafe { MaybeUninit::uninit().assume_init() }; // Compliant: Allowed for tuples/arrays of MaybeUninit.
----

== Resources
=== Documentation

* Clippy Lints - https://rust-lang.github.io/rust-clippy/master/index.html#uninit_assumed_init
