include::../description.adoc[]

== Noncompliant Code Example

With the default threshold of 3:

----
BEGIN
  prepare('action1');
  execute('action1');
  release('action1');
END;
/
----

== Compliant Solution

----
DECLARE
  action CONSTANT VARCHAR2(7) := 'action1';
BEGIN
  prepare(action);
  execute(action);
  release(action);
END;
/
----

include::../exceptions.adoc[]
