== Why is this an issue?

There's no point in including the default value of a column in an insert statement. It simply clutters the code to no additional benefit.


*Note* that this rule raises issues only when a database catalog is provided during the SonarQube analysis.


=== Noncompliant code example

With the table ``++PRODUCT++`` having a column ``++INV_COUNT++`` with default ``++0++``:

[source,cobol]
----
EXEC SQL
  INSERT INTO PRODUCT
  (
    NAME,
    INV_COUNT  -- Noncompliant
  )
  VALUES
  (
    :PROD-NAME,
    0  -- this is the default value
  )
END-EXEC
----


=== Compliant solution

[source,cobol]
----
EXEC SQL
  INSERT INTO PRODUCT
  (
    NAME
  )
  VALUES
  (
    :PROD-NAME
  )
END-EXEC
----



ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Remove column "xxx" from this insert; "yyy" is the default value.


=== Highlighting

* primary: column name
* secondary: column value
** message: "XXX' is the default value.


'''
== Comments And Links
(visible only on this page)

=== is related to: S3484

endif::env-github,rspecator-view[]
