When converting a string representation of a date and time to a `DateTime` object with the method `DateTime.Parse` you should always provide an `IFormatProvider` parameter.
If you don't provide it then the method will use the machines Culture Info and in case the given string is not following the machine's Culture then you'll have an object that does not
match the the string representation or an unexpected runtime error.

== Why is this an issue?

If you try to parse a string representation of date and time that does not have the expected format based on the machine's culture you'll get `DateTime` object that
does not match the input or a runtime error.


== How to fix it
Alway use a `DateTime.Parse` overload where you can provide an `IFormatProvider` parameter.

=== Code examples

==== Noncompliant code example

[source,text,diff-id=1,diff-type=noncompliant]
----
var dateTimeString = "4/12/2023 4:05:48 PM"; // this is a en-US format string - 12 of April 2023
var dateTimeObject = DateTime.Parse(dateTimeString); // In a machine with European local format is parsed as 4th of December

var dateTimeString2 = "4/13/2023 4:05:48 PM"; // this is a en-US format string - 13 of April 2023
var dateTimeObject2 = DateTime.Parse(dateTimeString2); // Runtime Error
----

==== Compliant solution

[source,text,diff-id=1,diff-type=compliant]
----
var dateTimeString = "4/12/2023 4:05:48 PM"; // this is a en-US format string - 12 of April 2023
var dateTimeObject = DateTime.Parse(dateTimeString, new CultureInfo("en-US"));

var dateTimeString2 = "4/13/2023 4:05:48 PM"; // this is a en-US format string - 13 of April 2023
var dateTimeObject2 = DateTime.Parse(dateTimeString2, new CultureInfo("en-US"))
----


== Resources

=== Documentation

* https://learn.microsoft.com/en-us/dotnet/api/system.datetime.parse[DateTime.Parse method]
* https://learn.microsoft.com/en-us/dotnet/api/system.globalization.cultureinfo[Culture Info class documentation]
* https://learn.microsoft.com/en-us/dotnet/standard/base-types/standard-date-and-time-format-strings[Standard date and time format strings]

