When a `System.Globalization.CultureInfo` or `IFormatProvider` object is not supplied, then `DateTime.Parse` method
is using the machines Culture Info and in case the DateTime string is not following the machine's Culture then you'll have an unexpected error.

To avoid this you should always Parse a DateTime string by providing the Culture info or if you are not sure you can always use `CultureInfo.InvariantCulture`.

// If you want to factorize the description uncomment the following line and create the file.
//include::../description.adoc[]

== Why is this an issue?

If you try to parse a `DateTime` string that does not have the expected format based on the machine's culture you'll get a runtime error.


== How to fix it
Alway use a `DateTime.Parse` overload where you can provide the CultureInfo or the IFormatProvider. In case you are unsure you can always use `CultureInfo.InvariantCulture`.

=== Code examples

==== Noncompliant code example

[source,text,diff-id=1,diff-type=noncompliant]
----
FIXME
----

==== Compliant solution

[source,text,diff-id=1,diff-type=compliant]
----
FIXME
----


//== Resources
//=== Documentation
//=== Articles & blog posts
//=== Conference presentations
//=== Standards
