A macro definition should not be redefined without marking that intent specifically by un-defining it first.


== Noncompliant Code Example

----
#define A 1
#define A 2
----


== Compliant Solution

----
#define A 1
#undef A
#define A 2
----


== Exceptions

If the redefinition has the same value as the original one. This is consistent with most C compilers warnings.

----
#define A 1
#define A 1
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

include::highlighting.adoc[]

'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
