Using a unique ``++KLIST++`` to access files ensures that the correct key is used at all times and creates a standard method of accessing the data throughout the code, simplifying maintenance and improving readability. 


This rule flags non-compliant instances of ``++CHAIN++``, ``++DELETE++``, ``++READE++``, ``++READPE++``, ``++SETGT++``, and ``++SETLL++`` operations.


== Noncompliant Code Example

----
C     'D001 B'      CHAIN     KYCUSTN                            20
... 
C     DEPT          CHAIN     KYCUSTN                            20
----


== Compliant Solution

----
C     DEPTKY        KLIST
C                   KFLD                    DEPT
C                   KFLD                    SHIFT 

C     DEPTKY        CHAIN     KYCUSTN                            20
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

endif::env-github,rspecator-view[]
