include::../description.adoc[]

include::../recommended.adoc[]

== Noncompliant Code Example

An ingress rule allowing all inbound SSH traffic:

[source,cloudformation]
----
MySecurityGroup:
    Type: "AWS::EC2::SecurityGroup"
    Properties:
      GroupDescription: "noncompliant"
      VpcId: !Ref myVPC
      SecurityGroupIngress:
        - IpProtocol: tcp
          FromPort: 22
          ToPort: 22 # SSH traffic
          CidrIp: "0.0.0.0/0" # from all IP addresses is authorized
----

== Compliant Solution

An ingress rule allowing inbound SSH traffic from specific IP addresses:

[source,cloudformation]
----
MySecurityGroup:
    Type: "AWS::EC2::SecurityGroup"
    Properties:
      GroupDescription: "compliant"
      VpcId: !Ref myVPC
      SecurityGroupIngress:
        - IpProtocol: tcp
          FromPort: 22
          ToPort: 22
          CidrIp: "1.2.3.0/24"
----

include::../see.adoc[]
