include::../description.adoc[]

== Noncompliant Code Example

With the default regular expression ``++^[A-Z][A-Z0-9]*(_[A-Z0-9]+)*$++``:

----
public class MyClass {
  public static final int first = 1;
}

public enum MyEnum {
  first;
}
----

== Compliant Solution

----
public class MyClass {
  public static final int FIRST = 1; 
}

public enum MyEnum {
  FIRST;
}
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../parameters.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
