== Why is this an issue?

The `Collection.toArray()` method returns an `Object[]` when no arguments are provided to it.
This can lead to a `ClassCastException` at runtime if you try to cast the returned array to an array of a specific class.
Instead, you should use this method by also providing an array of the desired type as the parameter.

=== Code examples

==== Noncompliant code example

[source,java,diff-id=1,diff-type=noncompliant]
----
public String [] getStringArray(List<String> strings) {
  return (String []) strings.toArray();  // Noncompliant, a ClassCastException will be thrown here
}
----

==== Compliant solution

[source,java,diff-id=1,diff-type=compliant]
----
public String [] getStringArray(List<String> strings) {
  return strings.toArray(new String[0]);
}
----

== Resources

* https://docs.oracle.com/javase/8/docs/api/java/util/Collection.html#toArray-T:A-[Collection.toArray()]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Pass "new xxx[0]" as an argument to "toArray".


endif::env-github,rspecator-view[]
