== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,java]
----
"ab{1,1}c"
"ab{1}c"
"ab{0,0}c"
"ab{0}c"
----

=== Compliant solution

[source,java]
----
"abc"
"ac"
----

include::../implementation.adoc[]
