include::../description.adoc[]

== Noncompliant Code Example

[source,java]
----
"ab{1,1}c"
"ab{1}c"
"ab{0,0}c"
"ab{0}c"
----

== Compliant Solution

[source,java]
----
"abc"
"ac"
----

include::../implementation.adoc[]
