<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>When an async block or function returns a value that is itself awaitable (like a <code>Future</code>), it often indicates that the developer forgot to await that value. This creates a nested future that must be awaited twice to get the actual result, which is rarely the intended behavior. Missing an await can lead to unexpected behavior where async operations never actually execute, nested futures that require multiple awaits to resolve, hard-to-debug problems, potential deadlocks, or blocking in async contexts.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-rust" data-lang="rust">async fn foo() {}

fn bar() {
    let x = async {
        foo() // Noncompliant: returns a future that needs to be awaited
    };
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-rust" data-lang="rust">async fn foo() {}

fn bar() {
    let x = async {
        foo().await // Properly awaits the inner future
    };
}</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Clippy Lints - <a href="https://rust-lang.github.io/rust-clippy/master/index.html#async_yields_async" class="bare">https://rust-lang.github.io/rust-clippy/master/index.html#async_yields_async</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>