The ``++/EJECT++`` compiler directive adds a page break when the code is printed, and should be used at the end of each ``++F++``, ``++D++``, and ``++C++`` specification section to make the structure of the code clearer and to enhance overall readability.


== Noncompliant Code Example

----
F* Conversion Job Activity File (CV)                   
F*                                                     
FCVJBAVP   IF   E           K DISK                     
F* Conversion Job Dependency File (CZ)                 
F*                                                     
FCVJBDPP   IF   E           K DISK                     
F*                                           
DP0RTCD           S              1                     
DP0JBNM           S             10A                     
DP0ERMG           S            100A                     
DK0JBNM           S                   LIKE(CVJBNM)     
DK1JBNM           S                   LIKE(CVJBNM)     
DJST              S             10A   DIM(10)           
DI                S              3  0 INZ               
C                                                       
C                   EXSR      SR999                     
C                                                       
C                   EXSR      SR100                     
----


== Compliant Solution

----
F* Conversion Job Activity File (CV)                   
F*                                                     
FCVJBAVP   IF   E           K DISK                     
F* Conversion Job Dependency File (CZ)                 
F*                                                     
FCVJBDPP   IF   E           K DISK                     
F*                                           
F/EJECT           
DP0RTCD           S              1                     
DP0JBNM           S             10A                     
DP0ERMG           S            100A                     
DK0JBNM           S                   LIKE(CVJBNM)     
DK1JBNM           S                   LIKE(CVJBNM)     
DJST              S             10A   DIM(10)           
DI                S              3  0 INZ               
D/EJECT 
C                                                       
C                   EXSR      SR999                     
C                                                       
C                   EXSR      SR100                     
C/EJECT  
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

endif::env-github,rspecator-view[]
