== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,swift]
----
if condition1 {
  if condition2 {
    doSomething()
  }
}

if let y = someOptional {
  if x > 0 {
    doSomething()
  } 
}
----

=== Compliant solution

[source,swift]
----
if condition1 && condition2 {
  doSomething()
}

if let y = someOptional where x > 0 {
  doSomething()
}

if x > 0, let y = someOptional {
  doSomething()
}
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../highlighting.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
