include::../description.adoc[]

== Noncompliant Code Example

----
strlen($name); // Noncompliant; "strlen" has no side effect
----

== Compliant Solution

----
$length = strlen($name);
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
