include::../description.adoc[]

== Noncompliant Code Example

----
strlen($name); // Noncompliant; "strlen" has no side effect
----

== Compliant Solution

----
$length = strlen($name);
----

include::../see.adoc[]

ifdef::env-github,rspecator-view[]
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
