== Why is this an issue?

When the call to a function doesn't have any side effect, what is the point of making the call if the results are ignored? In such cases, either the function call is useless and should be dropped, or the source code doesn't behave as expected.


=== Noncompliant code example

[source,php]
----
strlen($name); // Noncompliant; "strlen" has no side effect
----

=== Compliant solution

[source,php]
----
$length = strlen($name);
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
