== Why is this an issue?

Creating a new ``++Exception++`` without actually throwing it is useless and is probably due to a mistake.

=== Noncompliant code example

[source,csharp]
----
if (x < 0)
{
  new ArgumentException("x must be nonnegative");
}
----

=== Compliant solution

[source,csharp]
----
if (x < 0)
{
  throw new ArgumentException("x must be nonnegative");
}
----
ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../highlighting.adoc[]

endif::env-github,rspecator-view[]
