== Why is this an issue?

include::../why.adoc[]

=== What is the potential impact?

include::../impact.adoc[]

== How to fix it

include::../how.adoc[]

=== Code examples

==== Noncompliant code example

[source,csharp,diff-id=1,diff-type=noncompliant]
----
void LogDateTime()
{
    using var streamWriter = new StreamWriter("logs.txt", true);
    streamWriter.WriteLine($"DateTime:{DateTime.Now.ToString("o")}"); // This log won't have any meaning if it's reconstructed in a machine in a different timezone.
}
----

==== Compliant solution

[source,csharp,diff-id=1,diff-type=compliant]
----
void LogDateTime()
{
    using var streamWriter = new StreamWriter("logs.txt", true);
    streamWriter.WriteLine($"DateTime:{DateTime.UtcNow.ToString("o")}");
}
----

== Resources

include::../how.adoc[]
