== Why is this an issue?

The property `DateTime.Now` returns the current date and time expressed in the machines's local time, without containing any timezone-related data (e.g. offset).
This means that you cannot reconstruct what the time would be in another time zone, and even if it was possible, it wouldn't make a difference, since `DateTime.Kind` stops being `Local` and turns to `Unspecified` as soon as it crosses a machine's boundary.

Instead, you should use record the `DateTime` instants in UTC which tells you the date and time as it would be in Coordinated Universal Time that is not affected by changes such as daylight savings.

Generally, unless you want to display the Date and Time to a user on their local machine you should always use UTC (e.g., when storing dates, using them for calculations).


=== What is the potential impact?
You can end up with `DateTime` instants that have no meaning for noone expect the machine that they were recorded.
Additionally, by using UTC you get an unambiguous representation of an instant as you can get from any UTC instant to the equivalent local time. This operation isn't reversible as some local times are ambiguous and can be matched to more than one UTC instant (for example due to daylight savings).


== How to fix it
The fix is to use `DateTime.UtcNow`, `DateTimeOffSet.Now` (as it contains offset information) or `DateTimeOffSet.UtcNow`.

=== Code examples

==== Noncompliant code example

[source,csharp]
----
FIXME
----

==== Compliant solution

[source,csharp]
----
FIXME
----

//== Resources
//=== Documentation
//=== Articles & blog posts
//=== Conference presentations
//=== Standards
