== Why is this an issue?

The property `DateTime.Now` returns the current date and time expressed in the machine's local time without containing any timezone-related data (e.g., offset).
Not having this data means you cannot reconstruct the time in another time zone and potentially display it or use it in a machine other than the one in which it was recorded.
Even if possible, it wouldn't make a difference since `DateTime.Kind` stops being `Local` and turns to `Unspecified` as soon as it crosses a machine's boundary.

Instead, you should record the `DateTime` instants in UTC, which tells you the date and time as it would be in Coordinated Universal Time that is not affected by changes such as daylight savings.

Generally, unless you want to only display the Date and Time to a user on their local machine, you should always use UTC (e.g., when storing dates in a DB, using them for calculations).



=== What is the potential impact?

You can end up with `DateTime` instants that have no meaning for anyone except the machine they were recorded in.
Additionally, by using UTC, you get an unambiguous representation of an instant, as you can get from any UTC instant to the equivalent local time. This operation isn't reversible as some local times are ambiguous and can be matched to more than one UTC instant (for example, due to daylight savings).

== How to fix it

The fix is to use `DateTime.UtcNow`, `DateTimeOffSet.Now` (as it contains offset information) or, `DateTimeOffSet.UtcNow`.

=== Code examples

==== Noncompliant code example

[source,csharp]
----
void LogDateTime()
{
    using var streamWriter = new StreamWriter("logs.txt", true);
    streamWriter.WriteLine($"DateTime:{DateTime.Now.ToString("dd MMM yyyy HH: mm:ss")}"); // Noncompliant, this log won't have any meaning outside the machine it was recorded in
}
----

==== Compliant solution

[source,csharp]
----
void LogDateTime()
{
    using var streamWriter = new StreamWriter("logs.txt", true);
    streamWriter.WriteLine($"DateTime:{DateTime.UtcNow.ToString("dd MMM yyyy HH: mm:ss")}"); // Compliant
}
----

=== Documentation
https://learn.microsoft.com/en-us/dotnet/api/system.datetime.now
https://learn.microsoft.com/en-us/dotnet/api/system.datetime.utcnow
https://learn.microsoft.com/en-us/dotnet/api/system.datetimeoffset
https://learn.microsoft.com/en-us/dotnet/standard/datetime/resolve-ambiguous-times


=== Articles & blog posts
https://stackoverflow.com/questions/2580478/storing-date-times-as-utc-in-database
