<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>In records, introduced in Java 16, there are 2 ways to write a custom constructor: canonical and compact.</p>
</div>
<div class="paragraph">
<p>A canonical constructor is an ordinary constructor with arguments for all private fields. The default implementation just provides initialization for all private fields and there is no need to write it manually. You might want to write it yourself when you need to customize the constructor logic.</p>
</div>
<div class="paragraph">
<p>A compact constructor doesn&#8217;t have parameters defined explicitly, parentheses are omitted and access to private fields is not possible (even via <code>this</code>). The compact constructor has access to the constructor&#8217;s arguments and its body is executed right before the field initialization. It&#8217;s a perfect place to provide validation.</p>
</div>
<div class="paragraph">
<p>This rule reports an issue when a canonical constructor can be easily replaced by a compact version when these requirements are met:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>the last statements are trivial field initializations</p>
</li>
<li>
<p>no statement reads from fields or components</p>
</li>
<li>
<p>there are other statements in the constructor (case covered by <a data-rspec-id="S6207" class="rspec-auto-link">S6207</a>: redundant constructors in records)</p>
</li>
</ul>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">record Person(String name, int age) {
  Person(String name, int age) { // Noncompliant
    if (age &lt; 0) {
      throw new IllegalArgumentException("Negative age");
    }
    this.name = name;
    this.age = age;
  }
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">record Person(String name, int age) {
  Person { // Compliant
    if (age &lt; 0) {
      throw new IllegalArgumentException("Negative age");
    }
  }
}</code></pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p><a href="https://docs.oracle.com/javase/specs/jls/se16/html/jls-8.html#jls-8.10">Records specification</a></p>
</li>
</ul>
</div>
<hr>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Replace this usage of a 'canonical' constructor with a more concise 'compact' version</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>canonical constructor declaration</p>
</div>
</div>
</div>
</div>