Creating an object for the sole purpose of calling ``++getClass++`` on it is a waste of memory and cycles. Instead, simply use the class' ``++.class++`` property.


== Noncompliant Code Example

----
MyObject myOb = new MyObject();  // Noncompliant
Class c = myOb.getClass(); 
----


== Compliant Solution

----
Class c = MyObject.class; 
----



ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
