== Why is this an issue?

This rule applies whenever an `if` statement is followed by one or more `elif` statements; the final `elif` should be followed by an `else` statement.

The requirement for a final `else` statement is defensive programming.

The `else` statement should either take appropriate action or contain a suitable comment as to why no action is taken. This is consistent with the requirement to have a final `*)` clause in a `case` statement.

== How to fix it

=== Code examples

==== Noncompliant code example

[source,bash,diff-id=1,diff-type=noncompliant]
----
if [[ $x -eq 0 ]]; then
  do_something
elif [[ $x -eq 1 ]]; then
  do_something_else
fi
----

==== Compliant solution

[source,bash,diff-id=1,diff-type=compliant]
----
if [[ $x -eq 0 ]]; then
  do_something
elif [[ $x -eq 1 ]]; then
  do_something_else
else
  echo "Unexpected value: $x"
  exit 1
fi
----
