<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Positive <code>tabIndex</code> values can disrupt the natural tab order of the webpage. This can be confusing for screen reader users who rely on a logical tab order to navigate through the content. If the tab order doesn&#8217;t match the visual or logical order of elements, users may struggle to understand the page structure.</p>
</div>
<div class="paragraph">
<p>Therefore, it&#8217;s recommended to avoid using positive <code>tabIndex</code> values.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>If you need to make an element focusable that isn&#8217;t by default (like a <code>&lt;div&gt;</code> or <code>&lt;span&gt;</code>), you can use <code>tabIndex="0"</code>. This will add the element to the natural tab order based on its position in the HTML. Otherwise, either remove the <code>tabIndex</code> value or use <code>tabIndex="-1"</code> to remove the element from the tab order.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-html" data-lang="html">&lt;div&gt;
    &lt;span tabIndex="5"&gt;foo&lt;/span&gt; &lt;!-- Noncompliant --&gt;
    &lt;span tabIndex="3"&gt;bar&lt;/span&gt; &lt;!-- Noncompliant --&gt;
    &lt;span tabIndex="1"&gt;baz&lt;/span&gt; &lt;!-- Noncompliant --&gt;
    &lt;span tabIndex="2"&gt;qux&lt;/span&gt; &lt;!-- Noncompliant --&gt;
&lt;/div&gt;</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-html" data-lang="html">&lt;div&gt;
    &lt;span tabIndex="0"&gt;foo&lt;/span&gt;
    &lt;span tabIndex="-1"&gt;bar&lt;/span&gt;
    &lt;span&gt;qux&lt;/span&gt;
&lt;/div&gt;</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>MDN web docs - <a href="https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/tabindex">tabindex</a></p>
</li>
<li>
<p>WCAG - <a href="https://www.w3.org/WAI/WCAG21/Understanding/focus-order">Focus Order</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>