This rule raises an issue when unquoted symbol keys contain hyphens, which causes Ruby syntax errors.

== Why is this an issue?

In Ruby, symbols are identifiers that start with a colon (`:symbol`). When you use a hyphen in an unquoted symbol like `:data-role`, Ruby interprets the hyphen as a subtraction operator instead of part of the symbol name.

This means Ruby tries to subtract the variable `role` from the variable `data`, which typically results in an "undefined variable" error since these variables don't exist.

This issue commonly occurs in Rails applications when developers try to add HTML data attributes to helper methods like `link_to` or `content_tag`. The syntax `:data-role => "button"` looks intuitive but is invalid Ruby code.

The problem stems from Ruby's parsing rules: unquoted symbols can only contain letters, numbers, underscores, and certain special characters. Hyphens are not allowed because they conflict with the subtraction operator.

=== What is the potential impact?

This issue causes immediate syntax errors that prevent the application from running. The error messages can be confusing to developers, especially beginners, as they mention "undefined variable" rather than explaining the symbol syntax problem.

In development, this leads to:

* Application crashes with cryptic error messages
* Wasted debugging time trying to understand why variables are "undefined"
* Frustration for developers learning Ruby or Rails

The impact is typically caught early since the code won't run, but it can slow down development and create confusion about proper Ruby syntax.

== How to fix it

Quote the symbol using double quotes. This creates a valid symbol that can contain hyphens.

=== Code examples

==== Noncompliant code example

[source,ruby,diff-id=1,diff-type=noncompliant]
----
options = {
  :jquery-ui => true,  # Noncompliant
  :data-role => "button"  # Noncompliant
}
----

==== Compliant solution

[source,ruby,diff-id=1,diff-type=compliant]
----
options = {
  :"jquery-ui" => true,
  :"data-role" => "button"
}
----

== How to fix it in Rails

Use the :data hash option for HTML data attributes. This is the recommended Rails approach that automatically converts underscores to hyphens in the generated HTML.

=== Code examples

==== Noncompliant code example

[source,ruby,diff-id=2,diff-type=noncompliant]
----
link_to "Play", game_path, :data-role => "button" # Noncompliant
----

==== Compliant solution

[source,ruby,diff-id=2,diff-type=compliant]
----
link_to "Play", game_path, data: { role: "button" }
----

== Resources

=== Documentation

 * Ruby Symbol Documentation - https://ruby-doc.org/core/Symbol.html[Official Ruby documentation explaining symbol syntax and usage]

 * Rails HTML Options Guide - https://guides.rubyonrails.org/form_helpers.html#using-html-options[Rails guide explaining how to use HTML options including data attributes]
