With Java 21 `java.lanf.String` was updated with method `String.indexOf(int ch, int fromIndex, int toIndex)` that allows to search for a character in a substring of a string.

== Why is this an issue?

The method `String.indexOf(int ch, int fromIndex, int toIndex)` is now a better alternative to `String.indexOf(int ch, int fromIndex)`, as it allows to restrict the search even more and checks for out-of-bounds inputs.

This rule raises an issue when `String.indexOf(int ch, int fromIndex)` is used.

== How to fix it

Replace `indexOf(int char, int fromIndex)` with `indexOf(int ch, int fromIndex, int toIndex)`.

=== Code examples

==== Noncompliant code example

[source,java,diff-id=1,diff-type=noncompliant]
----
x.indexOf(':', 10); // Noncompliant, use indexOf(int ch, int fromIndex, int toIndex) instead
----

==== Compliant solution

[source,java,diff-id=1,diff-type=compliant]
----
x.indexOf(':', 10, x.length());
----
