This rule raises an issue when the parameters ``++how++``, ``++on++`` and ``++validate++`` are not specified when using ``++pandas.merge++`` or ``++pandas.join++``.


== Why is this an issue?

The Pandas library provides a user-friendly API to concatenate two data frames together with the methods ``++merge++`` and ``++join++``.

While these methods provide sensible default parameters, using them without arguments makes the developer's intention unclear and thus the code harder to maintain.


== How to fix it

To fix this issue provide the parameters ``++how++``, ``++on++`` and ``++validate++`` to methods ``++pd.merge++`` or ``++pd.join++``.

=== Code examples

==== Noncompliant code example

[source,python,diff-id=1,diff-type=noncompliant]
----
import pandas as pd

def merge_dfs(age_df:pd.DataFrame, name_df:pd.DataFrame):
  return age_df.merge(name_df) # Noncompliant: it is unclear on which column the merge could happen, as well as what is the expected result.

----

==== Compliant solution

[source,python,diff-id=1,diff-type=compliant]
----
import pandas as pd

def merge_dfs(age_df:pd.DataFrame, name_df:pd.DataFrame):
  return age_df.merge(
          name_df,
          on="user_id",
          how="inner",
          validate="1:1"
         ) # Compliant
----

//=== How does this work?

//=== Pitfalls

//=== Going the extra mile

== Resources

=== Documentation

* Pandas Documentation - https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.merge.html#pandas-dataframe-merge[pandas.DataFrame.merge]
* Pandas Documentation - https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.join.html#pandas-dataframe-join[pandas.DataFrame.join]

//=== Articles & blog posts
//=== Conference presentations
//=== Standards
//=== External coding guidelines
//=== Benchmarks
