<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Using <code>EXIT</code> and <code>CHECK</code> in <code>SELECT</code> statements to stop the execution of <code>SELECT</code> loop is an expensive and ineffective way to filter data. Filtering should be part of the <code>SELECT</code> loop themselves. Most of the time conditions located in a <code>CHECK</code> statement should be moved to the <code>WHERE</code> clause, and the <code>EXIT</code> statement should typically be replaced by an <code>UP TO 1 ROW</code> clause.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-abap" data-lang="abap">SELECT * FROM SBOOK INTO SBOOK_WA.
  CHECK: SBOOK_WAS-CARRID = 'LH' AND SBOOK_WAS-CONNID = '0400'. "Noncompliant
ENDSELECT.</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-abap" data-lang="abap">SELECT * FROM SBOOK INTO SBOOK_WA WHERE CARRID = 'LH' AND CONNID = '0400'.
ENDSELECT.</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Refactor this piece of code to not stop the execution of the nesting "SELECT" loop before having read all the elements</p>
</div>
</div>
</div>
</div>