This rule reports when a class, namespace or other source code component violates the architectural constraints defined for a project.

== Why is this an issue?

There is a reference from one source code component to another that is prohibited by the architectural constraints defined for the project.
These definitions are made by the maintainers of the project in the architecture configuration file.

=== What is the potential impact?

Over time, codebases often drift from the intended project architecture,
gradually degrading the code structure and mis-aligning it with the original design.

This misalignment reduces visibility and control over daily decisions affecting the architecture.
As these small decisions accumulate, the codebase becomes harder to understand, and the architecture grows increasingly complex and unstructured.

== How to fix it

Refactor your source code to adhere to the architectural constraints.
The specific approach will depend on your project architecture and the code violating the constraints.
For example, you might replace a function call with an alternative function call from another software layer
or create a new function in that layer to maintain a clean architecture.

=== Code examples

Assuming that for source code components in directory `./src/Example/Panels`, direct access to
components in repositories directory `./src/Example/Repos` is constrained and should use a Service instead:

==== Noncompliant code example

[source,csharp,diff-id=1,diff-type=noncompliant]
----
using Example.Repos;

namespace Example.Panels;

public class ShowCustomersPanel
{
    CustomerRepo customerRepo = ...;
}
----

==== Compliant solution

[source,csharp,diff-id=1,diff-type=compliant]
----
using Example.Services;

namespace Example.Panels;

public class ShowCustomersPanel
{
    CustomerService customerService = ...;
}
----

=== Documentation

- Defining architectural constraints for SonarQube
