This rule prevents using the `void` type in ways that could be misleading.

You shouldn't use the void type outside of function returns and generics.

// If you want to factorize the description uncomment the following line and create the file.
//include::../description.adoc[]

== Why is this an issue?

The `void` type in other places makes no sense because only `undefined` or `null` values can be assigned.

//=== What is the potential impact?

== How to fix it

You should use the type `undefined` instead.

=== Code examples

==== Noncompliant code example

[source,typescript,diff-id=1,diff-type=noncompliant]
----
type PossibleValues = string | number | void;
----

==== Compliant solution

[source,typescript,diff-id=1,diff-type=compliant]
----
type PossibleValues = string | number | undefined;
----

//=== How does this work?

//=== Pitfalls

//=== Going the extra mile


== Resources
=== Documentation

* https://www.typescriptlang.org/docs/handbook/2/functions.html#void[TypeScript void type]
* https://www.typescriptlang.org/docs/handbook/2/functions.html#return-type-void[TypeScript void function return type]

//=== Articles & blog posts
//=== Conference presentations
//=== Standards
