<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when timezone-naive time methods like <code>Time.now</code>, <code>DateTime.now</code>, or manual timezone manipulations are used instead of Rails' timezone-aware alternatives.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Using timezone-naive time methods in Ruby/Rails applications can lead to serious timezone-related bugs and inconsistent behavior across different environments.</p>
</div>
<div class="paragraph">
<p>When you use methods like <code>Time.now</code> or <code>DateTime.now</code>, they return the system&#8217;s local time, which ignores your Rails application&#8217;s configured timezone (<code>Time.zone</code> or <code>config.time_zone</code>). This creates several problems:</p>
</div>
<div class="ulist">
<ul>
<li>
<p><strong>Environment inconsistency</strong>: Your development machine, staging server, and production server might be in different timezones, causing the same code to behave differently in each environment.</p>
</li>
<li>
<p><strong>User timezone issues</strong>: When dealing with users in different timezones, system time methods don&#8217;t respect user-specific timezone preferences.</p>
</li>
<li>
<p><strong>Daylight saving time bugs</strong>: Manual timezone calculations often fail to handle daylight saving time transitions correctly, leading to off-by-one-hour errors twice a year.</p>
</li>
<li>
<p><strong>Database storage problems</strong>: Storing timezone-naive timestamps in the database can cause confusion about when events actually occurred.</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>Rails provides timezone-aware methods specifically to solve these problems. Methods like <code>Time.current</code>, <code>Time.zone.now</code>, and <code>Time.zone.parse</code> respect your application&#8217;s timezone configuration and handle complex timezone logic correctly.</p>
</div>
<div class="paragraph">
<p>Manual timezone manipulations, such as modifying time arrays or adding raw offsets, are particularly dangerous because they bypass Rails' sophisticated timezone handling and can easily introduce subtle bugs that only appear under specific conditions.</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="paragraph">
<p>Using timezone-naive time methods can cause:</p>
</div>
<div class="ulist">
<ul>
<li>
<p><strong>Data corruption</strong>: Timestamps stored with incorrect timezone information</p>
</li>
<li>
<p><strong>Business logic errors</strong>: Scheduled tasks running at wrong times, incorrect time-based calculations</p>
</li>
<li>
<p><strong>User experience issues</strong>: Events displayed at wrong times for users in different timezones</p>
</li>
<li>
<p><strong>Compliance problems</strong>: Incorrect audit trails and logging timestamps</p>
</li>
<li>
<p><strong>Difficult debugging</strong>: Timezone-related bugs that only appear in certain environments or during daylight saving time transitions</p>
</li>
</ul>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it_in_rails">How to fix it in Rails</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Replace <code>Time.now</code> and <code>DateTime.now</code> with timezone-aware alternatives like <code>Time.current</code> or <code>Time.zone.now</code>.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-ruby" data-lang="ruby">current_time = Time.now # Noncompliant
current_datetime = DateTime.now # Noncompliant</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-ruby" data-lang="ruby">current_time = Time.current
current_datetime = DateTime.current
# or
current_time = Time.zone.now</code></pre>
</div>
</div>
<div class="paragraph">
<p>Use <code>Time.zone.local</code> and <code>Time.zone.parse</code> instead of <code>DateTime.new</code> and <code>DateTime.parse</code> for creating timezone-aware time objects.</p>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_code_examples_2">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example_2">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-ruby" data-lang="ruby">event_time = DateTime.new(2012, 7, 11, 20, 10, 0) # Noncompliant
parsed_date = DateTime.parse(params[:date]) # Noncompliant</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution_2">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-ruby" data-lang="ruby">Time.zone = "Pacific Time (US &amp; Canada)"
event_time = Time.zone.local(2012, 7, 11, 20, 10, 0)
parsed_date = Time.zone.parse(params[:date])</code></pre>
</div>
</div>
<div class="paragraph">
<p>Use Rails' <code>in_time_zone()</code> method instead of manual timezone calculations or array manipulations.</p>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_code_examples_3">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example_3">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-ruby" data-lang="ruby"># Manual array manipulation
temp = input_date.to_time.to_a # Noncompliant
temp[8] = true
temp[9] = "Eastern Daylight Time"
input_date = Time.local(*temp)

# Manual offset calculation
send_time = Time.now + user.timezone_offset.hours # Noncompliant</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution_3">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-ruby" data-lang="ruby"># Use Rails timezone conversion methods
utc_time = Time.current
eastern_time = utc_time.in_time_zone('Eastern Time (US &amp; Canada)')

# Proper timezone conversion
send_time = Time.current.in_time_zone(user.timezone)</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Rails Time Zone Guide - <a href="https://guides.rubyonrails.org/configuring.html#configuring-time-zones">Official Rails documentation on configuring and working with time zones</a></p>
</li>
<li>
<p>Rails API: Time Zone - <a href="https://api.rubyonrails.org/classes/ActiveSupport/TimeZone.html">API documentation for ActiveSupport::TimeZone class</a></p>
</li>
<li>
<p>Working with Time Zones in Rails - <a href="https://thoughtbot.com/blog/its-about-time-zones">Comprehensive guide on handling time zones correctly in Rails applications</a></p>
</li>
</ul>
</div>
</div>
<div class="sect2">
<h3 id="_standards">Standards</h3>
<div class="ulist">
<ul>
<li>
<p>ISO 8601 - <a href="https://en.wikipedia.org/wiki/ISO_8601">International standard for date and time representation</a></p>
</li>
</ul>
</div>
</div>
<div class="sect2">
<h3 id="_related_rules">Related rules</h3>
<div class="ulist">
<ul>
<li>
<p><a data-rspec-id="S6903" class="rspec-auto-link">RSPEC-6903</a> - <a href="https://rules.sonarsource.com/python/RSPEC-6903/">Python rule for timezone-aware datetime usage</a></p>
</li>
<li>
<p><a data-rspec-id="S6566" class="rspec-auto-link">RSPEC-6566</a> - <a href="https://rules.sonarsource.com/csharp/RSPEC-6566/">C# rule for timezone-aware DateTime usage</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>