include::../description.adoc[]

== Noncompliant Code Example

[source,javascript]
----
"a[b]c"
"[\\^]"
----

== Compliant Solution

[source,javascript]
----
"abc"
"\\^"
"a[*]c" // Compliant, see Exceptions
----

include::../exceptions.adoc[]

include::../implementation.adoc[]
