== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,javascript]
----
/a[b]c/
/[\^]/
----

=== Compliant solution

[source,javascript]
----
/abc/
/\^/
/a[*]c/ // Compliant, see Exceptions
----

include::../exceptions.adoc[]

include::../implementation.adoc[]
