<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when a comprehension is used to only copy a collection. Instead, the respective constructor should be used.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Python comprehensions are a concise way to create new collections while transforming or filtering elements. However, using a comprehension that copies elements from one collection to another without any transformation is less readable than using the respective constructor directly.</p>
</div>
<div class="paragraph">
<p>Therefore, when a comprehension is only copying elements, use the appropriate constructor instead:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>Replace <code>[x for x in iterable]</code> with <code>list(iterable)</code></p>
</li>
<li>
<p>Replace <code>{x for x in iterable}</code> with <code>set(iterable)</code></p>
</li>
<li>
<p>Replace <code>{k: v for k, v in iterable.items()}</code> with <code>dict(iterable)</code></p>
</li>
</ul>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Replace comprehensions that copy elements from one collection to another with the respective constructor.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">some_list = [1, 2, 3, 2]
[x for x in some_list] # Noncompliant</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">some_list = [1, 2, 3, 2]
list(some_list) # Compliant</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Python Documentation - <a href="https://docs.python.org/3/library/stdtypes.html#dict">dict()</a></p>
</li>
<li>
<p>Python Documentation - <a href="https://docs.python.org/3/library/stdtypes.html#list">list()</a></p>
</li>
<li>
<p>Python Documentation - <a href="https://docs.python.org/3/library/stdtypes.html#set">set()</a></p>
</li>
</ul>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="paragraph">
<p>Determining whether a comprehension is just copying a collection or is modifying it in some way is not trivial. If all of the following conditions are met, the comprehension does not modify its collection:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>Filtering comprehensions</p>
</li>
<li>
<p>Transforming comprehensions (e.g., multiplying each element by 2)</p>
</li>
<li>
<p>Dict comprehensions can also flip keys and values (e.g., <code>{v: k for k, v in d.items()}</code>)</p>
</li>
<li>
<p>No multiple <code>for</code> clauses</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>Keep in mind that this rule also applies to constructors with a generator expression. For example <code>list(x for x in some_list)</code> should also raise.</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Replace this comprehension with <code>{constructor}({iterable})</code>.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>The comprehension or generator expression.</p>
</div>
</div>
</div>
</div>