<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when a method name does not follow camelCase conventions, such as starting with an uppercase letter or using underscores.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Consistent method naming conventions are essential for code readability and maintainability. The camelCase convention, where method names start with a lowercase letter followed by mixed case letters and numbers, is widely adopted in many programming languages including Groovy.</p>
</div>
<div class="paragraph">
<p>When method names don&#8217;t follow this convention, several problems arise:</p>
</div>
<div class="ulist">
<ul>
<li>
<p><strong>Reduced readability</strong>: Inconsistent naming makes code harder to scan and understand quickly</p>
</li>
<li>
<p><strong>Team confusion</strong>: Different naming styles within the same codebase create cognitive overhead for developers</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>Method names that start with uppercase letters can be particularly confusing because they resemble class names, potentially misleading developers about the code&#8217;s structure. Similarly, underscore-separated names (snake_case) are more common in other languages and can make Groovy code feel inconsistent with language conventions.</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="paragraph">
<p>While this issue doesn&#8217;t directly affect application functionality, it impacts code maintainability and team productivity. Inconsistent naming conventions make code reviews more difficult, reduce code readability, and can slow down development as team members need extra time to understand non-standard naming patterns.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Rename the method to follow camelCase convention.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-groovy" data-lang="groovy">def MyMethod() { // Noncompliant
    return
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-groovy" data-lang="groovy">def myMethod() { // Compliant
    return
}</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">

</div>
</div>