When two or more components or classes are heavily dependent on each other, it is said that the class is tightly coupled. This makes modifying or maintaining one class difficult without affecting the others.
// If you want to factorize the description uncomment the following line and create the file.
//include::../description.adoc[]

== Why is this an issue?
This results in a rigid and inflexible design that can lead to issues such as code duplication, poor scalability, and reduced code reusability.

//=== What is the potential impact?

== How to fix it
//== How to fix it in FRAMEWORK NAME
To avoid tight coupling in Java, it's important to follow good design principles such as SOLID, dependency injection, and loose coupling. This can help make the code more modular, flexible, and easier to maintain and modify over time.

=== Code examples

==== Noncompliant code example

[source,java]
----
class Subject {
    Topic t = new Topic();
    public void startReading() {
        t.understand();
    }
}
class Topic {
    public void understand() {
        System.out.println("Understand");
    }
}
----

==== Compliant solution

[source,java]
----
public interface Topic {
    void understand();
}
class Topic1 implements Topic {
    public void understand() {
        System.out.println("Understand 1");
    }
}
class Topic2 implements Topic {
    public void understand() {
        System.out.println("Understand 2");
    }
}
public class Subject {
    public static void main(String[] args) {
        Topic t = new Topic1();
        t.understand();
    }
}
----

//=== How does this work?

//=== Pitfalls

//=== Going the extra mile


//== Resources
//=== Documentation
//=== Articles & blog posts
//=== Conference presentations
//=== Standards