== Why is this an issue?

The use of ``++org.fest.assertions.Assertions.assertThat++`` by itself does nothing. You must combine it with another method that actually tests the value in use.


=== Noncompliant code example

[source,java]
----
assertThat(name);  // Noncompliant
----


=== Compliant solution

[source,java]
----
assertThat(name).isNotNull();
----


ifdef::env-github,rspecator-view[]
'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
