The use of ``++org.fest.assertions.Assertions.assertThat++`` by itself does nothing. You must combine it with another method that actually tests the value in use.


== Noncompliant Code Example

----
assertThat(name);  // Noncompliant
----


== Compliant Solution

----
assertThat(name).isNotNull();
----


ifdef::env-github,rspecator-view[]
'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
