<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Shared naming conventions allow teams to collaborate effectively. This rule raises an issue when the a pom&#8217;s <code>groupId</code> does not match the provided regular expression.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="paragraph">
<p>With the default regular expression: <code>(com|org)(\.[a-z][a-z-0-9]*)+</code></p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-xml" data-lang="xml">&lt;project ...&gt;
  &lt;groupId&gt;myCo&lt;/groupId&gt;  &lt;!-- Noncompliant --&gt;

  &lt;!-- ... --&gt;
&lt;/project&gt;</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-xml" data-lang="xml">&lt;project ...&gt;
  &lt;groupId&gt;com.myco&lt;/groupId&gt;

  &lt;!-- ... --&gt;
&lt;/project&gt;</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Update this "groupId" to match the provided regular expression: xxx.</p>
</div>
</div>
<div class="sect2">
<h3 id="_parameters">Parameters</h3>
<div class="sidebarblock">
<div class="content">
<div class="title">regex</div>
<div class="listingblock">
<div class="content">
<pre>(com|org)(\.[a-z][a-z-0-9]*)+</pre>
</div>
</div>
<div class="paragraph">
<p>The regular expression the "groupId" should match</p>
</div>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>groupId value</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_30_nov_2015_140629_michael_gumowski_wrote">on 30 Nov 2015, 14:06:29 Michael Gumowski wrote:</h3>
<div class="paragraph">
<p>LGTM [~ann.campbell.2]!</p>
</div>
<div class="paragraph">
<p>Note that I&#8217;m not sure about your regex. You can have more that one group, separated by dots, after the first <code>com</code>/<code>org</code>.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_30_nov_2015_160030_ann_campbell_wrote">on 30 Nov 2015, 16:00:30 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>Uhm&#8230;&#8203; yeah [~michael.gumowski]. That&#8217;s what I intended.</p>
</div>
</div>
</div>
</div>