== Why is this an issue?

Sharing naming conventions allows teams to collaborate efficiently. This rule checks that targetNamespaces match a provided regular expression.


=== Noncompliant code example

With the default regular expression: \http://www+

[source,xml]
----
<xs:schema xmlns:xs="http://www.w3.org/2001/XMLSchema"
           elementFormDefault="qualified" version="1"
            targetNamespace="fruit">  <!-- Noncompliant -->
...
----


=== Compliant solution

[source,xml]
----
<xs:schema xmlns:xs="http://www.w3.org/2001/XMLSchema"
           elementFormDefault="qualified" version="1"
           targetNamespace="http://www.codeSamples.com/fruit">
...
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Rename this "targetNamespace" to match the regular expression: ${format}.


=== Parameters

.format
****

----
^http:\/{2}www\.*+\/.+
----

Regular expression used to check targetNamespaces against
****


=== Highlighting

Namespace value


endif::env-github,rspecator-view[]
