include::../description.adoc[]

== Noncompliant Code Example

----
public class Foo {
  private var foo : String

  public func getFoo() -> String {
     return foo
  }

  //...

}

var foo = Foo()
foo.getFoo() // what does this return?
----

== Compliant Solution

----
public class Foo {
  private var name : String 

  public func getName() -> String { 
      return name 
  }

  //...

}

var foo = Foo();
foo.getName()
----
ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

endif::env-github,rspecator-view[]
