include::../summary.adoc[]

== Why is this an issue?

include::../description.adoc[]

=== Code examples

==== Noncompliant code example

[source,swift,diff-id=1,diff-type=noncompliant]
----
public class Foo {
  private var foo : String

  public func getFoo() -> String {
     return foo
  }

}

var foo = Foo()
foo.getFoo() // what does this return?
----

==== Compliant solution

[source,swift,diff-id=1,diff-type=compliant]
----
public class Foo {
  private var name : String 

  public func getName() -> String { 
      return name 
  }

}

var foo = Foo();
foo.getName()
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

endif::env-github,rspecator-view[]
