== Why is this an issue?

_Forwarding references_ are a special kind of references that both ignore and preserve the _value category_ of a function argument, making it possible to forward it by means of ``++std::forward++``.

Any code using such a reference for any other purpose than forwarding is actually ignoring rvalue-ness and const-ness of the associated parameter.


=== Noncompliant code example

[source,cpp]
----
#include <utility>
#include <string>
#include <iostream>

template<typename TP> void f( TP&& arg ) {
    std::string s(arg);
}

int main() {
    std::string s("test");
    f(std::move(s));
    std::cout<<"f:"<<s<<std::endl; // output is "f:test"

    return 0;
}
----


=== Compliant solution

[source,cpp]
----
#include <utility>
#include <string>
#include <iostream>

template<typename TP> void f( TP&& arg ) {
    std::string s(std::forward<TP>(arg));
}

int main() {
    std::string s("test");
    f(std::move(s));
    std::cout<<"f:"<<s<<std::endl; // output is "f:"

    return 0;
}
----


== Resources

* https://github.com/isocpp/CppCoreGuidelines/blob/036324/CppCoreGuidelines.md#f19-for-forward-parameters-pass-by-tp-and-only-stdforward-the-parameter[{cpp} Core Guidelines F.19] - For “forward” parameters, pass by TP&& and only std::forward the parameter

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Use only "std::forward" on forwarding references.


endif::env-github,rspecator-view[]
