== Why is this an issue?

_Forwarding references_ are a special kind of references that both ignore and preserve the _value category_ of a function argument, making it possible to forward it by means of ``++std::forward++``.

Any code using such a reference for any purpose other than forwarding is actually ignoring the rvalue-ness and const-ness of the associated parameter.


=== Noncompliant code example

[source,cpp,diff-id=1,diff-type=noncompliant]
----
#include <iostream>
#include <string>
#include <utility>
#include <vector>

class Registry {
  std::vector<std::string> names;

 public:
  template <typename StringLike>
  void addName(StringLike&& arg) {
    names.emplace_back(arg);
  }
};

void example() {
  Registry r;

  std::string name = "example";
  r.addName(std::move(name));
  std::cout << "name:" << name << std::endl;
  // output is "name:example"
}

----

In the above example, instead of moving the content of `name` is not moved into the vector, a copy is made.

=== Compliant solution

[source,cpp,diff-id=1,diff-type=compliant]
----
#include <iostream>
#include <string>
#include <utility>
#include <vector>

class Registry {
  std::vector<std::string> names;

 public:
  template <typename StringLike>
  void addName(StringLike&& arg) {
    names.emplace_back(std::forward<StringLike>(arg));
  }
};

void example() {
  Registry r;

  std::string name = "example";
  r.addName(std::move(name));
  std::cout << "name:" << name << std::endl;
  // output is "name:"
}

----


== Resources

=== Documentation

* {cpp} reference - https://en.cppreference.com/w/cpp/utility/forward[`std::forward`]

=== External coding guidelines

* {cpp} Core Guidelines - https://github.com/isocpp/CppCoreGuidelines/blob/e49158a/CppCoreGuidelines.md#f19-for-forward-parameters-pass-by-tp-and-only-stdforward-the-parameter[F.19: For "forward" parameters, pass by `TP&&` and only `std::forward` the parameter]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Use only "std::forward" on forwarding references.


endif::env-github,rspecator-view[]
