== Why is this an issue?

_Forwarding references_ are a special kind of references that both ignore and preserve the _value category_ of a function argument, making it possible to forward it by using ``++std::forward++`` or ``++std::forward_like++``.

Any code using such a reference for any purpose other than forwarding is actually ignoring the rvalue-ness and const-ness of the associated parameter.


=== Noncompliant code example

[source,cpp,diff-id=1,diff-type=noncompliant]
----
class Registry {
  std::vector<std::string> names;

 public:
  template <typename StringLike>
  void addName(StringLike&& arg) {
    names.emplace_back(arg); // Not forwarded
  }
};

void example() {
  Registry r;

  std::string name = "example";
  r.addName(std::move(name));
  std::cout << "name:" << name << std::endl;
  // output is "name:example"
}
----

In this example, the intent is to move the content of `name` into the vector, but instead a copy is made.

=== Compliant solution

[source,cpp,diff-id=1,diff-type=compliant]
----
class Registry {
  std::vector<std::string> names;

 public:
  template <typename StringLike>
  void addName(StringLike&& arg) {
    names.emplace_back(std::forward<StringLike>(arg));
  }
};

void example() {
  Registry r;

  std::string name = "example";
  r.addName(std::move(name));
  std::cout << "name:" << name << std::endl;
  // output can be anything: name has been moved-from
}
----

``++std::forward_like++`` is available since {cpp}23 and is useful when you want to forward a subobject of a forwarding reference.

[source,cpp]
----
class Registry {
  // ...
  template <typename PairOfStrings>
  void addNames(PairOfStrings&& arg) {
    addName(std::forward_like<PairOfStrings>(arg.second)); // We don't care about arg.first
  }
};
----


== Resources

=== Documentation

* {cpp} reference - https://en.cppreference.com/w/cpp/utility/forward[`std::forward`]
* {cpp} reference - https://en.cppreference.com/w/cpp/utility/forward_like[``++std::forward_like++``]

=== External coding guidelines

* {ccg_F_19}

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Use only "std::forward" on forwarding references.


endif::env-github,rspecator-view[]
