== Why is this an issue?

include::../description.adoc[]

=== Exceptions

To prevent generating some false-positives, literals having 5 or less characters are excluded as well as literals containing only letters, digits and '_'.

== How to fix it

include::../howtofix.adoc[]

=== Code examples

==== Noncompliant code example

With the default threshold of 3:

[source,dart,diff-id=1,diff-type=noncompliant]
----
class A {
  void run() {
    prepare('string literal');    // Noncompliant - "string literal" is duplicated 3 times
    execute('string literal');
    release('string literal');
  }
}
----

==== Compliant solution

[source,dart,diff-id=1,diff-type=compliant]
----
class A {
  static const _const = 'string literal';

  void run() {
    prepare(_const);    // Compliant
    execute(_const);
    release(_const);
  }
}
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../parameters.adoc[]

include::../highlighting.adoc[]

'''


endif::env-github,rspecator-view[]
