<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Using a <code>NUMBER</code> to store an integer is less performant than using a <code>PLS_INTEGER</code>. <code>PLS_INTEGER</code>s require less storage than <code>NUMBER</code>s, and benefit from the use of hardware math, as opposed to the library math required for <code>NUMBER</code>s. Even more performant is the <code>SIMPLE_INTEGER</code> subtype of <code>PLS_INTEGER</code>. However, changing to either of these types is only appropriate under certain circumstances.</p>
</div>
<div class="paragraph">
<p><code>PLS_INTEGER</code> is only a candidate for <code>NUMBER</code> with a scale of up to 9.</p>
</div>
<div class="paragraph">
<p><code>SIMPLE_INTEGER</code> has the same size limitation, in addition to it&#8217;s <code>NOT NULL</code> constraint and lack of overflow checking.</p>
</div>
<div class="paragraph">
<p>This rule raises an issue when a <code>NUMBER</code> is declared with a scale of 9 or less.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-sql" data-lang="sql">DECLARE
    son NUMBER(1);      -- Noncompliant
    rumbo NUMBER(9);  -- Noncompliant
    conga Number(10);   -- Ignored; falls outside the PLS_INTEGER range
    compalsa PLS_INTEGER;</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-sql" data-lang="sql">DECLARE
    son SIMPLE_INTEGER;
    rumbo PLS_INTEGER;
    conga Number(10);   -- Ignored; falls outside the PLS_INTEGER range
    compalsa PLS_INTEGER;</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Use the "PLS_INTEGER" type here; it&#8217;s more performant.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p><code>NUMBER(1-9)</code></p>
</div>
</div>
</div>
</div>