<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>To check the type of an object there are at least two options:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>The simplest and shortest one with help of the <code>instanceof</code> operator</p>
</li>
<li>
<p>The cumbersome and error-prone one with help of the <code>Class.isAssignableFrom(...)</code> method</p>
</li>
</ul>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">if (MyClass.class.isAssignableFrom(x.getClass())) { // Noncompliant
  MyClass mc = (MyClass)x;
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">if (x instanceof MyClass.class) {
  MyClass mc = (MyClass)x;
}</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Use "instanceof" instead.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_is_related_to_s3610">is related to: <a data-rspec-id="S3610" class="rspec-auto-link">S3610</a></h3>

</div>
<div class="sect2">
<h3 id="_on_11_may_2015_094123_dinesh_bolkensteyn_wrote">on 11 May 2015, 09:41:23 Dinesh Bolkensteyn wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] I&#8217;ve updated the rule for C# - the previous example did not match what was detected in Java.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_11_may_2015_094201_dinesh_bolkensteyn_wrote">on 11 May 2015, 09:42:01 Dinesh Bolkensteyn wrote:</h3>
<div class="paragraph">
<p>LGTM</p>
</div>
</div>
</div>
</div>