<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>C&#43;&#43;20  introduces the <code>consteval</code> keyword, which enforces that a function is always evaluated at compile time.</p>
</div>
<div class="paragraph">
<p><code>constexpr</code> functions can produce a compile-time constant in some contexts and when called with appropriate arguments, but they can also be invoked at run-time.
<code>consteval</code> functions must always be evaluated at compile-time. If that cannot happen, a compilation error will occur.</p>
</div>
<div class="paragraph">
<p>User-defined literal operators are designed to be called with constant operands known at compile time.
Therefore, if these are intended to be evaluated at compile time with <code>constexpr</code>, <code>consteval</code> should be used instead to enforce compile time evaluation.
This guarantees that no code is evaluated at run-time, and also enables error detection at compile-time.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">constexpr unsigned char operator ""_u8(unsigned long long value) {
  if (value &gt;= 256u) { throw std::overflow_error("Overflow on literal"); }
  return static_cast&lt;unsigned char&gt;(value);
}

void f() {
  auto const a = 128_u8; // evaluated at compile time
  auto const b = 512_u8; // evaluated at runtime, throws when evaluated
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">consteval unsigned char operator ""_u8(unsigned long long value) {
  if (value &gt;= 256u) { throw std::overflow_error("Overflow on literal"); }
  return static_cast&lt;unsigned char&gt;(value);
}

void f() {
  auto const a = 128_u8; // evaluated at compile time
  auto const b = 512_u8; // compilation error, the bug is detected
}</code></pre>
</div>
</div>
</div>
</div>
</div>