Trying to access a dictionary key that does not exist will raise a `KeyError`.

When trying to access or remove a a key that may not be there, several solutions are possible:

* Use the `get()` method instead of a subscription. It will return `None` for a missing key instead of raising a `KeyError`.
* Use the `setdefault()` method to provide keys that may be missing with a default value.
* Check that the key is present in the dictionary with the `if key in dict:` construct.
* Use a `try/except` block and handle the potential `KeyError`.
* Use a `defaultdict` instead of a regular `dict` object. 

== Noncompliant Code Example

[source,python]
----
def remove_k2_noncompliant(param):
    param.pop('k2')

def noncompliant():
    my_dict = {'k1': 42}
    remove_k2(my_dict)  # Noncompliant
----
== Compliant Solution

[source,python]
----
def remove_k2_compliant(param):
    if 'k2' in param:
        param.pop('k2')

def noncompliant():
    my_dict = {'k1': 42}
    remove_k2(my_dict)
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

'''
endif::env-github,rspecator-view[]