<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>In Java, the <code>Thread</code> class represents a thread of execution.
Synchronization between threads is typically achieved using objects or shared resources.</p>
</div>
<div class="paragraph">
<p>The methods <code>wait(&#8230;&#8203;)</code>, <code>notify()</code>, and <code>notifyAll()</code> are related to the underlying object&#8217;s monitor
and are designed to be called on objects that act as locks or monitors for synchronization.
These methods are available on Java <code>Object</code> and, therefore, automatically inherited by all objects, including <code>Thread.</code></p>
</div>
<div class="paragraph">
<p>Calling these methods on a <code>Thread</code> may corrupt the behavior of the JVM, which relies on them to change the state
of the thread (<code>BLOCKED,</code> <code>WAITING,</code>&#8230;&#8203;).</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">Thread myThread = new Thread(new RunnableJob());
...
myThread.wait(); // Noncompliant</code></pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p><a href="https://docs.oracle.com/en/java/javase/20/docs/api/java.base/java/lang/Thread.html">Oracle Java SE - Thread</a></p>
</li>
<li>
<p><a href="https://docs.oracle.com/en/java/javase/20/docs/api/java.base/java/lang/Object.html">Oracle Java SE - Object</a></p>
</li>
</ul>
</div>
<hr>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Refactor the synchronisation mechanism to not use a Thread instance as a monitor.</p>
</div>
</div>
</div>
</div>