== Why is this an issue?

``++UPDATE dbtab SET ...++`` without a ``++WHERE++`` condition changes all the entries of the table. Check whether dataset to be changed can be limited by a suitable ``++WHERE++`` condition.


=== Noncompliant code example

[source,abap]
----
UPDATE COUNTRIES SET NAME=country_name.
----


=== Compliant solution

[source,abap]
----
UPDATE COUNTRIES SET NAME=country_name WHERE CODE=country_code.
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Add a "WHERE" clause to this "UPDATE" statement to prevent updating all rows in table "XXX".


endif::env-github,rspecator-view[]
