The ``++strerror++`` function returns a pointer to a buffer that is only valid until the function is called again, including from another thread. Which means that in practice, for any multithread program, it's not possible to use it properly.


One safe alternative is ``++strerror_s++``, provided in annex K of C11. To have access to it, you need a standard library that supports it (this can be tested with the macro ``++__STDC_LIB_EXT1__++``), and you need to enable it by defining the macro ``++__STDC_WANT_LIB_EXT1__++`` before including ``++<string.h>++``. ``++strerror_s++`` takes as an argument a buffer that will store the error message. Iworks together with the ``++strerrorlen_s++`` function, which can tell you the required buffer size to store the error.


Some environment also provide the ``++strerror_r++`` function, which works in a way similar to ``++strerror_s++``, except there is now function that can provide you with the needed buffer size (but the return value will tell you if the buffer was large enough): Either you accept to have a truncated message if the message is too long, or you should call this function in a loop with increasing buffer size until it succeeds.


== Noncompliant Code Example

----
int f(char *path) {
  FILE * fp = fopen(path,"r");
  if (fp == NULL) {
    // errno itself has thread storage duration
    char *errorMsg = strerror(errno); // Noncompliant, might be changed by another thread
    printf("Error: %s\n", errorMsg);
  }
}
----


== Compliant Solution

----
int f(char *path) {
  FILE * fp = fopen(path,"r");
  if (fp == NULL) {
    // errno itself has thread storage duration
    int fileError = errno;
    size_t errorLen = strerrorlen_s(fileError) +1; // For the final null character
    char *errorMsg = malloc(errorLen);
    strerror_s(errorMsg, errorLen, fileError)
    printf("Error: %s\n", errorMsg);
    free(errorMsg);
  }
}
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

endif::env-github,rspecator-view[]
