There's no point in creating an array solely for the purpose of passing it to a `params` parameter. Simply pass the elements directly. They will be consolidated into an array automatically.

== Noncompliant Code Example

[source,csharp]
----
public void CallTheThing() 
{
    DoTheThing(new string[] { "s1", "s2" });  // Noncompliant: unnecessary
    DoTheThing(new string[12]);  // Compliant
}

public void DoTheThing(params string[] args)
{
    // ...
}
----

== Compliant Solution

[source,csharp]
----
public void CallTheThing()
{
    DoTheThing("s1", "s2");
    DoTheThing(new string[12]);
}

public void DoTheThing(params string[] args)
{
    // ...
}
----
