There's no point in creating an array solely for the purpose of passing it as a params (``++...++``) argument; params keyword allow to pass a variable number of parameters that will behave exactly like an array variable inside the method implementation. Simply pass the elements directly.

== Noncompliant Code Example

[source,csharp]
----
public void CallTheThing() {
	//...
	DoTheThing(new string[] { "s1", "s2"});  // Noncompliant: unnecessary
	DoTheThing(new string[12]);  // Compliant
	// ...
}

public void DoTheThing (params string[] args) {
	// ...
}
----

== Compliant Solution

[source,csharp]
----
public void CallTheThing()
{
	//...
	DoTheThing("s1", "s2");
	DoTheThing(new string[12]);
	// ...
}

public void DoTheThing(params string[] args)
{
	// ...
}
----
