== Why is this an issue?

When two methods have the same implementation, either it was a mistake - something else was intended - or the duplication was intentional, but may be confusing to maintainers. In the latter case, one implementation should invoke the other.


This rule raises an exception when two methods implemented inside the class definition share the same implementation.

=== Noncompliant code example

[source,cpp,diff-id=1,diff-type=noncompliant]
----
class Point {
  int x;
  int y;

// .....

public:
  void setX(int v) {
    if (v >= 0 && v < MAX_X) {
      x = v;
      return;
    }
    error();
  }

  void setY(int v) {  // Noncompliant
    if (v >= 0 && v < MAX_X) {
      x = v;
      return;
    }
    error();
  }
};
----

=== Compliant solution

[source,cpp,diff-id=1,diff-type=compliant]
----
class Point {
  int x;
  int y;

// .....

public:
 void setX(int v) {
    if (v >= 0 && v < MAX_X) {
      x = v;
      return;
    }
    error();
  }

  void setY(int v) {
    if (v >= 0 && v < MAX_Y) {
      y = v;
      return;
    }
    error();
  }
};
----

=== Exceptions

Empty methods, methods with the same name (overload) and methods with only one statement are ignored.

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../highlighting.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
