This is an issue when a variable is declared with an explicit type that can be clearly inferred from a typed expression on the right-hand side.

== Why is this an issue?

Go's type inference system can automatically determine variable types from their initialization values. When the right-hand side of a declaration contains a typed expression, explicitly declaring the type creates unnecessary visual noise and reduces code readability.

The Go language encourages using the short variable declaration syntax (`:=`) when the type can be inferred. This makes code more concise and follows Go's design philosophy of simplicity and clarity.

Redundant type declarations can make code harder to read and maintain, especially in larger codebases where reducing visual clutter improves comprehension.

=== What is the potential impact?

This issue affects code readability and maintainability. While it doesn't cause functional problems, it goes against Go's idiomatic style and can make code unnecessarily verbose. Following Go's conventions for type inference helps create cleaner, more readable code that aligns with community standards.

== How to fix it

Replace the explicit type declaration with the short variable declaration syntax (`:=`) when the type can be clearly inferred from the right-hand side.

=== Code examples

==== Noncompliant code example

[source,go,diff-id=1,diff-type=noncompliant]
----
var i int = 42
var j int = i // Noncompliant: type is redundant
----

==== Compliant solution

[source,go,diff-id=1,diff-type=compliant]
----
var i int = 42
j := i // Type is inferred as int
----

== Resources

=== Documentation

 * Go Tour - Type Inference - https://go.dev/tour/basics/14[Official Go documentation explaining type inference and variable declarations]

 * Go Tour - Short Variable Declarations - https://go.dev/tour/basics/10[Official Go documentation on short variable declaration syntax]

 * Effective Go - Variables - https://go.dev/doc/effective_go#variables[Best practices for variable declarations in Go]

=== Related rules

 * S3257 - https://rules.sonarsource.com/javascript/RSPEC-3257/[JavaScript rule for avoiding redundant type declarations]

 * S6212 - https://rules.sonarsource.com/java/RSPEC-6212/[Java rule for using var when type is obvious]
