<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when the dimensions of the input arrays for the <code>numpy.broadcast</code> function are not compatible for matrix broadcasting.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>The <code>numpy.broadcast</code> function performs matrix broadcasting between two arrays, this means that
the output will contain a smaller matrix stretched to the shape of the bigger one.</p>
</div>
<div class="paragraph">
<p>For example:</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">import numpy as np

x = np.array([[0, 0 , 0], [10, 10, 10], [20, 20, 20], [30, 30, 30]])
y = np.array([1,2,3])
b = np.broadcast(x, y)

out = np.empty(b.shape)
out.flat = [v for (u, v) in b]
print(out)</code></pre>
</div>
</div>
<div class="paragraph">
<p>the output will be</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-text" data-lang="text">[[1. 2. 3.]
 [1. 2. 3.]
 [1. 2. 3.]
 [1. 2. 3.]]</code></pre>
</div>
</div>
<div class="paragraph">
<p>This broadcasting mechanism is also used silently when performing basic matrix math operations
such as a multiplication (<code>*</code>) or an addition (<code>+</code>) of two matrices.</p>
</div>
<div class="paragraph">
<p>For a matrix broadcasting to be valid, the number of rows in the matrix must be equal to the number of rows in the second matrix or has just one row.
You can use <code>numpy.shape()</code> function to check the dimensions of the arrays before performing the operation.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>To fix this issue you should make sure that two broadcasted matrices are compatible</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">import numpy as np

a = np.array([[1, 2], [3, 4]])
b = np.array([5, 6, 7])
c = a * b # Noncompliant: ValueError: operands could not be broadcast together with shapes (2,2) (3,)
d = np.broadcast(a, b) # Noncompliant: ValueError: shape mismatch: objects cannot be broadcast to a single shape.  Mismatch is between arg 0 with shape (2, 2) and arg 1 with shape (3,).</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">import numpy as np

a = np.array([[1, 2], [3, 4]])
b = np.array([5, 6])
c = a * b # Compliant
d = np.broadcast(a, b) # Compliant</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>NumPy Documentation - <a href="https://numpy.org/doc/stable/user/basics.broadcasting.html">Broadcasting</a></p>
</li>
<li>
<p>NumPy Documentation - <a href="https://numpy.org/doc/stable/reference/generated/numpy.broadcast.html#numpy-broadcast">numpy.broadcast</a></p>
</li>
</ul>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<hr>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
</div>
</div>