<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Usually, it is the container data structures that define <code>operator[]</code> to mimic the traditional C-array interface. The subscript operator looks up an element in the container. It can return the element by value or by (optionally <code>const</code>) reference. You need a non-<code>const</code> subscript operator returning non-<code>const</code> reference to the element to be able to modify it. You need a <code>const</code> subscript operator to make a <code>const</code>-qualified instance of the container useful and enable the read-access with a consistent lookup interface.</p>
</div>
<div class="paragraph">
<p>This rule raises an issue on a class that has a non-<code>const</code> <code>operator[]</code> overload that does not modify <code>*this</code> object but lacks a <code>const</code> overload.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">template&lt;typename T&gt;
class MyVector {
public:
  T&amp; operator[] (unsigned index); // Noncompliant, missing const overload
};</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">template&lt;typename T&gt;
class MyArray {
public:
  const T&amp; operator[] (unsigned index) const; // Compliant
  // const-only operator and the compiler will tell you
  // when you try to use it in a non-const context and need an overload
};
template&lt;typename T&gt;
class MyArray {
public:
  T const&amp; operator[] (unsigned index) const;
  T&amp; operator[] (unsigned index); // Compliant, const-overloaded
};</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_exceptions">Exceptions</h3>
<div class="paragraph">
<p>If <code>operator[]</code> might insert a new key, as it is done in <code>std::map</code>, or modifies the container in another externally-visible manner, there is no intuitive <code>const</code> overload for it.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p><a href="https://isocpp.org/wiki/faq/const-correctness#const-overloading">Isocpp - What&#8217;s the deal with "const-overloading"?</a></p>
</li>
</ul>
</div>
</div>
</div>