=== Code examples

==== Noncompliant code example

[source,csharp,diff-id=1,diff-type=noncompliant]
----
string Interpolate(string value) =>
    FormattableString.Invariant($"Value: {value}");
----

[source,csharp,diff-id=2,diff-type=noncompliant]
----
string Interpolate(string value) =>
    FormattableString.CurrentCulture($"Value: {value}");
----

==== Compliant solution

[source,csharp,diff-id=1,diff-type=compliant]
----
string Interpolate(string value) =>
    string.Create(CultureInfo.InvariantCulture, $"Value: {value}");
----

[source,csharp,diff-id=2,diff-type=compliant]
----
string Interpolate(string value) =>
    string.Create(CultureInfo.CurrentCulture, $"Value: {value}");
----
