== Why is this an issue?

When there is only one statement in a chain, the chain syntax can be omitted, which simplifies the code.


=== Noncompliant code example

[source,abap]
----
CLEAR: w_alvvr.
----


=== Compliant solution

[source,abap]
----
CLEAR w_alvvr.
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Remove the colon from this statement


endif::env-github,rspecator-view[]
