<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Most people will reasonably assume that a class with "Test" in the name will contain test methods. Writing "Test" classes without tests will inevitably lead to confusion, and perhaps to gaps in test coverage. Such classes should either be renamed or corrected.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-text" data-lang="text">public class MathTest {
  private List questions;
  private int averageScore;
  private int highScore;
  // ...
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-text" data-lang="text">public class MathExam {
  private List questions;
  private int averageScore;
  private int highScore;
  // ...
}</code></pre>
</div>
</div>
</div>
</div>
</div>