== Why is this an issue?

``++std::format++``, introduced by {cpp}20, enables straightforward string construction out of values of various types.


Before {cpp}20, one popular way to obtain the same result was the conversion of the values with ``++std::to_string++`` and piecewise string concatenation.


``++std::format++`` is strictly superior. It is more efficient because it constructs the string in-place instead of copying substrings one by one. It is also often shorter and easier to read because the format pattern is presented in a single piece and not scattered across the concatenation expression.


This rule reports string concatenation cases that can be replaced by ``++std::format++`` to improve performance and readability.


=== Noncompliant code example

[source,cpp]
----
std::string greeting(int n) {
  return "Hello, player " + std::to_string(n) + "."; // Noncompliant
}
----


=== Compliant solution

[source,cpp]
----
std::string greeting(int n) {
  return std::format("Hello, player {}.", n); // Compliant
}
----

== Resources

=== Documentation

* {cpp} reference - https://en.cppreference.com/w/cpp/utility/format/format[`std::format`]
* {cpp} reference - https://en.cppreference.com/w/cpp/string/basic_string/to_string[``++std::to_string++``]

=== Articles & blog posts

* {cpp} Stories - https://www.cppstories.com/2022/custom-stdformat-cpp20/[Formatting Custom types with std::format from {cpp}20]

=== Related rules

* S6484 - Concatenated "std::format" outputs should be replaced by a single invocation
* S6494 - {cpp} formatting functions should be used instead of C printf-like functions
* S6495 - "std::format" should be used instead of standard output manipulators
