``++std::format++``, introduced by {cpp}20, enables straightforward string construction out of values of various types.


Before {cpp}20, one popular way to obtain the same result was the conversion of the values with ``++std::to_string++`` and piecewise string concatenation.


``++std::format++`` is strictly superior. It is more efficient because it constructs the string in-place instead of copying substrings one by one. It is also often shorter and easier to read because the format pattern is presented in a single piece and not scattered across the concatenation expression.


This rule reports string concatenation cases that can be replaced by ``++std::format++`` and gain in speed and readability.


== Noncompliant Code Example

[source,cpp]
----
std::string greeting(int n) {
  return "Hello, player " + std::to_string(n) + "."; // Noncompliant
}
----


== Compliant Solution

[source,cpp]
----
std::string greeting(int n) {
  return std::format("Hello, player {}.", n); // Compliant
}
----

