== Why is this an issue?

include::description.adoc[]

=== Noncompliant code example

[source,text]
----
public String getReadableStatus(Job j) {
  return j.isRunning() ? "Running" : j.hasErrors() ? "Failed" : "Succeeded";  // Noncompliant
}
----


=== Compliant solution

[source,text]
----
public String getReadableStatus(Job j) {
  if (j.isRunning()) {
    return "Running";
  }
  return j.hasErrors() ? "Failed" : "Succeeded";
}
----

