<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>In Kotlin, <code>Flow</code> represents a cold stream concept. Similar to <code>Stream</code> in Java or <code>Sequence</code> in Kotlin, we can manipulate the data inside the flow (filter, transform, collect, etc). The <code>Flow</code> API, just like <code>Stream</code> and <code>Sequence</code>, offers two types of operations: intermediate and terminal. Intermediate operations again return a <code>Flow</code> instance, all other operations are considered terminal. As flows are naturally lazy, no operations will actually be started until a terminal operation is called.</p>
</div>
<div class="paragraph">
<p>This rule reports an issue when the result of an intermediate operation on <code>Flow</code> is left unused.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-kotlin" data-lang="kotlin">suspend fun main() {
    val flow = flow {
        emit(1)
        emit(2)
        emit(3)
    }

    flow.take(2) // Noncompliant, the result of this operation is never used
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-kotlin" data-lang="kotlin">suspend fun main() {
    val flow = flow {
        emit(1)
        emit(2)
        emit(3)
    }

    flow.take(2).collect { println(it) } // Compliant, collect is a terminal operation
}</code></pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p><a href="https://kotlinlang.org/docs/flow.html">Flow documentation</a></p>
</li>
</ul>
</div>
</div>
</div>