<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Having two methods in the same class with the same body is suspicious and potentially confusing. If it is due to a bad copy/paste, it should be fixed. If it&#8217;s on purpose, it is clearer for the second method to simply call the first.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-text" data-lang="text">public void doTheThing() {
  this.x = 4;
}

public void doTheOtherThing {  // Noncompliant
  this.x = 4;
}</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_duplicates_s4144">duplicates: <a data-rspec-id="S4144" class="rspec-auto-link">S4144</a></h3>

</div>
<div class="sect2">
<h3 id="_on_18_jan_2016_125920_ann_campbell_wrote">on 18 Jan 2016, 12:59:20 Ann Campbell wrote:</h3>
<div class="paragraph">
<p><a href="http://www.viva64.com/en/b/0363/#ID0EVHBI" class="bare">http://www.viva64.com/en/b/0363/#ID0EVHBI</a></p>
</div>
</div>
<div class="sect2">
<h3 id="_on_5_oct_2017_213127_freddy_mallet_wrote">on 5 Oct 2017, 21:31:27 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>Due to the overlap with <a data-rspec-id="S4144" class="rspec-auto-link">RSPEC-4144</a> I would close this RSPEC [~ann.campbell.2] and [~jeanchristophe.collet].</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_6_oct_2017_131320_jean_christophe_collet_wrote">on 6 Oct 2017, 13:13:20 Jean-Christophe Collet wrote:</h3>
<div class="paragraph">
<p>I agree, it&#8217;s a clear duplicate.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_6_oct_2017_144343_jean_christophe_collet_wrote">on 6 Oct 2017, 14:43:43 Jean-Christophe Collet wrote:</h3>
<div class="paragraph">
<p>Duplicate of <a data-rspec-id="S4144" class="rspec-auto-link">RSPEC-4144</a></p>
</div>
</div>
</div>
</div>