== Why is this an issue?

include::description.adoc[]

=== Noncompliant code example

[source,text]
----
if (myList.size() >= 0) { ... }

if (myList.size() < 0) { ... }

boolean result = myArray.length >= 0;

if (0 > myArray.length) { ... }
----


=== Compliant solution

[source,text]
----
if (!myList.isEmpty()) { ... }

if (myArray.length >= 42) { ... }
----

