== Why is this an issue?

Arguments evaluation order in a function call is not specified:

* Before {cpp}17, the evaluation of each argument was unsequenced with the evaluation of other arguments, which can lead to undefined behavior if the same value is modified in several arguments,
* After {cpp}17, it is sequenced, but in an unspecified order: the behavior is not longer undefined, but the values passed to the function will be non portable.

Both cases should be avoided, because the code will probably not be what was expected.


=== Noncompliant code example

[source,cpp]
----
void f(int i, int j);

void g() {
  int i = 0;
  f(++i, ++i); // Noncompliant, the call could either be f(1,2) or f(2,1) (since C++17) or undefined behavior (before C++17)
}
----


=== Exceptions

This rule does not apply to overloaded operators because they respect the sequencing order rules of the operator they overload (since {cpp}17).


== Resources

* {cpp} Core Guidelines - https://github.com/isocpp/CppCoreGuidelines/blob/e49158a/CppCoreGuidelines.md#es44-dont-depend-on-order-of-evaluation-of-function-arguments[ES.44: Don't depend on order of evaluation of function arguments]
* {cpp} reference - https://en.cppreference.com/w/cpp/language/eval_order[cppreference.com - order of evaluation]


ifdef::env-github,rspecator-view[]
'''
== Comments And Links
(visible only on this page)

=== on 25 Oct 2019, 15:57:41 Geoffray Adde wrote:
I think we should clarify what we are going to raise because this is a quite complex topic (cf. the second link in the section See).

=== on 8 Nov 2019, 18:47:12 Lo√Øc Joly wrote:
\[~amelie.renard]

I wonder why we did not select the more generic rule ES.43 Avoid expressions with undefined order of evaluation, even if for now only the case of function arguments is detected? 


Moreover, the exception only applies to {cpp}17, maybe we should detect it for old code?

endif::env-github,rspecator-view[]
