<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>There is no good reason to keep an empty and therefore valueless section. Such sections should be removed.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cobol" data-lang="cobol">FIRST SECTION.
  MOVE A TO B.

SECOND SECTION.  *&gt; Noncompliant; empty

THIRD SECTION.

someParagraph.
  DISPLAY B.</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cobol" data-lang="cobol">FIRST SECTION.
  MOVE A TO B.

THIRD SECTION.

someParagraph.
  DISPLAY B.</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove this useless section declaration.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_9_oct_2013_001313_ann_campbell_wrote">on 9 Oct 2013, 00:13:13 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>Again, should I be adding the compliant version?</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_20_nov_2013_101837_dinesh_bolkensteyn_wrote">on 20 Nov 2013, 10:18:37 Dinesh Bolkensteyn wrote:</h3>
<div class="paragraph">
<p>I think you definitely should [~ann.campbell.2].</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_20_nov_2013_101936_dinesh_bolkensteyn_wrote">on 20 Nov 2013, 10:19:36 Dinesh Bolkensteyn wrote:</h3>
<div class="paragraph">
<p>Everything must be challengable and challenged, and asking a lot of questions slows every one down, so it&#8217;s better to just go ahead and do the evolutions we believe are right.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_20_nov_2013_102336_dinesh_bolkensteyn_wrote">on 20 Nov 2013, 10:23:36 Dinesh Bolkensteyn wrote:</h3>
<div class="paragraph">
<p>As per our guidelines (http://confluence.internal.sonarsource.com/display/DEV/Guidelines+-+Design), I removed the "isn&#8217;t" contraction.</p>
</div>
</div>
</div>
</div>