``++StringBuilder++`` is more efficient than string concatenation, especially when the operator is repeated over and over as in loops.

== Noncompliant Code Example

[source,cs]
----
string str = "";
for (int i = 0; i < arrayOfStrings.Length ; ++i) 
{
  str = str + arrayOfStrings[i];
}
----

== Compliant Solution

[source,cs]
----
StringBuilder bld = new StringBuilder();
for (int i = 0; i < arrayOfStrings.Length; ++i) 
{
  bld.Append(arrayOfStrings[i]);
}
string str = bld.ToString();
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
