== How to fix it in MSTest

Remove the `ExpectedException` attribute in favor of using the https://learn.microsoft.com/en-us/dotnet/api/microsoft.visualstudio.testtools.unittesting.assert.throwsexception[Assert.ThrowsException] assertion.

=== Code examples

==== Noncompliant code example

[source,csharp,diff-id=2,diff-type=noncompliant]
----
[Test]
[ExpectedException(typeof(ArgumentNullException))]  // Noncompliant
public void Method_NullParam()
{
    var sut = new MyService();
    sut.Method(null);
}
----

==== Compliant solution

[source,csharp,diff-id=2,diff-type=compliant]
----
[Test]
public void Method_NullParam()
{
    var sut = new MyService();
    Assert.ThrowsException<ArgumentNullException>(() => sut.Method(null));
}
----
