== Why is this an issue?

It can be useful to use in-code notation to suppress issues, but when those suppressions are no longer relevant they become a potential source of confusion and should be removed.


=== Noncompliant code example

[source,java]
----
@SuppressWarnings("squid:S4174") // Noncompliant
public void doSomething() {
  final int LOCAL = 42;  // S4174 is about naming of local constants but there's nothing wrong here
----


=== Compliant solution

[source,java]
----
public void doSomething() {
  final int LOCAL = 42;  // S4174 is about naming of local constants but there's nothing wrong here
----



ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Remove this irrelevant "xxx", no such warning would be raised without it.


=== Highlighting

the suppression annotation/marking


'''
== Comments And Links
(visible only on this page)

=== is related to: S4632

endif::env-github,rspecator-view[]
