It can be useful to use in-code notation to suppress issues, but when those suppressions are no longer relevant they become a potential source of confusion and should be removed.


== Noncompliant Code Example

----
@SuppressWarnings("squid:S4174") // Noncompliant
public void doSomething() {
  final int LOCAL = 42;  // S4174 is about naming of local constants but there's nothing wrong here
----


== Compliant Solution

----
public void doSomething() {
  final int LOCAL = 42;  // S4174 is about naming of local constants but there's nothing wrong here
----



ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

include::highlighting.adoc[]

'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
