== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,javascript]
----
function f() {
  try {
    doSomething();
  } catch (err) {
  }
}
----

=== Compliant solution

[source,javascript]
----
function f() {
  try {
    doSomething();
  } catch (err) {
    console.log(`Exception while doing something: ${err}`);
  }
}
----

include::../see.adoc[]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
