<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Type parameters that aren&#8217;t used are dead code, which can only distract and possibly confuse developers during maintenance. Therefore, unused type parameters should be removed.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">int &lt;T&gt; Add(int a, int b) // Noncompliant; &lt;T&gt; is ignored
{
  return a + b;
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">int Add(int a, int b)
{
  return a + b;
}</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>"xxx" is not used in the [method|type|class].</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_6_jan_2015_105615_dinesh_bolkensteyn_wrote">on 6 Jan 2015, 10:56:15 Dinesh Bolkensteyn wrote:</h3>
<div class="paragraph">
<p>I think this might make sense for Java (generics) as well [~ann.campbell.2], and perhaps also C&#43;&#43; (templates).</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_6_jan_2015_121809_ann_campbell_wrote">on 6 Jan 2015, 12:18:09 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~dinesh.bolkensteyn] my understanding of this (which is admittedly very shaky) is that in the Noncompliant example there are actually 3 params: a, b, and an as-yet-unnamed param of type T. So I&#8217;m not sure how this translates to Java. But I&#8217;m happy to add C&#43;&#43; (never used templates, so I&#8217;ll take your word for it. :-) )</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_6_jan_2015_122458_dinesh_bolkensteyn_wrote">on 6 Jan 2015, 12:24:58 Dinesh Bolkensteyn wrote:</h3>
<div class="paragraph">
<p>Well, here would be a Java example:</p>
</div>
<div class="listingblock">
<div class="content">
<pre>class MyClass {
  public static &lt;T&gt; void MyMethod() {} // Isn't &lt;T&gt; unused here too?
}</pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_on_6_jan_2015_192051_ann_campbell_wrote">on 6 Jan 2015, 19:20:51 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>But it&#8217;s not a parameter, [~dinesh.bolkensteyn]&#8230;&#8203;?</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_7_jan_2015_075216_dinesh_bolkensteyn_wrote">on 7 Jan 2015, 07:52:16 Dinesh Bolkensteyn wrote:</h3>
<div class="paragraph">
<p>Let&#8217;s have a short call on this [~ann.campbell.2]</p>
</div>
</div>
</div>
</div>