<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Method references are specialized lambda expressions for methods that already have a name. They take the form of <code>objectReference::methodName</code> or <code>ClassName::methodName</code>. When a lambda does nothing but call an existing method, it can be much clearer to use a method reference instead.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">Set roster = transferElements(rosterSource,
      () -&gt; { return new HashSet&lt;&gt;(); }  // Noncompliant
);</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">Set roster = transferElements(rosterSource, HashSet::new);</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_duplicates_s1612">duplicates: <a data-rspec-id="S1612" class="rspec-auto-link">S1612</a></h3>

</div>
</div>
</div>