== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,cobol]
----
UPDATE books
SET title = 'unknown'
WHERE title = NULL -- Noncompliant 
----

=== Compliant solution

[source,cobol]
----
UPDATE books
SET title = 'unknown'
WHERE title IS NULL
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
