include::../description.adoc[]

== Noncompliant Code Example

----
UPDATE books
SET title = 'unknown'
WHERE title = NULL -- Noncompliant 
----

== Compliant Solution

----
UPDATE books
SET title = 'unknown'
WHERE title IS NULL
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
