include::../description.adoc[]

== Why is this an issue?

include::../impact.adoc[]

== How to fix it in Android

include::../common/fix/android.adoc[]

=== Code examples

==== Noncompliant code example

[source,kotlin,diff-id=1,diff-type=noncompliant]
----
class LoginActivity : Activity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        val username = findViewById<EditText>(R.id.username).text.toString()
        val password = findViewById<EditText>(R.id.password).text.toString()

        Log.i("LoginAttempt", "Login attempt: $username with password: $password") // Noncompliant
    }
}
----

==== Compliant solution

The simplest way to fix this is to not log sensitive data.

[source,kotlin,diff-id=1,diff-type=compliant]
----
class LoginActivity : Activity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        val username = findViewById<EditText>(R.id.username).text.toString()
        val password = findViewById<EditText>(R.id.password).text.toString()

        Log.i("LoginAttempt", "Login attempt: $username")
    }
}
----

include::../common/fix/proguard.adoc[]

include::../common/resources.adoc[]
