<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Both <code>FixedSizeList</code> from the Commons library, and the list returned from <code>Arrays.asList</code> offer <code>add</code> and <code>remove</code> methods (as required by the <code>List</code> interface they implement), but neither truly supports their use. Both list types have fixed lengths and will throw errors if an <code>add</code> or <code>remove</code> method, or any of their variations, is called.</p>
</div>
<div class="paragraph">
<p>This rule raises an issue when one of the following methods is invoked on a <code>FixedSizeList</code> instance:</p>
</div>
<div class="ulist">
<ul>
<li>
<p><code>add</code></p>
</li>
<li>
<p><code>addAll</code></p>
</li>
<li>
<p><code>clear</code></p>
</li>
<li>
<p><code>remove</code></p>
</li>
<li>
<p><code>removeAll</code></p>
</li>
</ul>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">List&lt;String&gt; strings = Arrays.asList("Hello");
strings.add("world");  // Noncompliant</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">List&lt;String&gt; strings = Arrays.asList("Hello", "world");</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove this call to "add"; it will cause an exception for this fixed-length list.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p><code>add</code></p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_21_sep_2017_153025_ann_campbell_wrote">on 21 Sep 2017, 15:30:25 Ann Campbell wrote:</h3>
<div class="paragraph">
<p><a href="https://groups.google.com/forum/?pli=1#!topic/sonarqube/OGktfbSPbdg" class="bare">https://groups.google.com/forum/?pli=1#!topic/sonarqube/OGktfbSPbdg</a></p>
</div>
</div>
<div class="sect2">
<h3 id="_on_21_sep_2017_155801_pierre_yves_nicolas_wrote">on 21 Sep 2017, 15:58:01 Pierre-Yves Nicolas wrote:</h3>
<div class="paragraph">
<p>What about other methods? <code>addAll</code>, <code>remove</code>, <code>clear</code>&#8230;&#8203;</p>
</div>
</div>
</div>
</div>