<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>TO BE TRANSLATED :</p>
</div>
<div class="paragraph">
<p>L’objectif est de laisser la détermination taille des blocs physiques de fichier au niveau des JCL plutôt que de l’imposer par programme. Cela permet d’optimiser cette taille sans avoir à recompiler/livrer le programme</p>
</div>
<div class="paragraph">
<p>Que doit-on contrôler?</p>
</div>
<div class="paragraph">
<p>La présence de BLOC CONTAINS 0 RECORDS (ou BLOC 0, BLOC CONTAINS 0, etc)</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cobol" data-lang="cobol">FD  MONFICHIER
    RECORD CONTAINS 10 CHARACTERS.
01  ENR-LU         PIC X(10).</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cobol" data-lang="cobol">FD  MONFICHIER
    BLOC 0.
01  ENR-LU         PIC X(10).</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_27_may_2014_092736_freddy_mallet_wrote">on 27 May 2014, 09:27:36 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>Rule specific to the BNP context. We should provide an XPath rule to get the same context.</p>
</div>
</div>
</div>
</div>