include::../description.adoc[]

== Noncompliant Code Example

[source,cpp]
----
void save() {
  try {
    saveDocument();
  } catch (const std::exception& ex) {
  }
}
----

== Compliant Solution

[source,cpp]
----
void save() {
  try {
    saveDocument();
  } catch (const std::exception& ex) {
    log << "Exception while saving the document: " << ex.what();
  }
}
----

include::../see.adoc[]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
