include::../description.adoc[]

== Noncompliant Code Example

----
void save() {
  try {
    saveDocument();
  } catch (const std::exception& ex) {
  }
}
----

== Compliant Solution

----
void save() {
  try {
    saveDocument();
  } catch (const std::exception& ex) {
    log << "Exception while saving the document: " << ex.what();
  }
}
----

include::../see.adoc[]

ifdef::env-github,rspecator-view[]
'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
