== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,cpp]
----
void save() {
  try {
    saveDocument();
  } catch (const std::exception& ex) {
  }
}
----

=== Compliant solution

[source,cpp]
----
void save() {
  try {
    saveDocument();
  } catch (const std::exception& ex) {
    log << "Exception while saving the document: " << ex.what();
  }
}
----

include::../see.adoc[]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Handle this exception or don't catch it at all.


'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
