=== Noncompliant code example

[source,text]
----
var colors = [];  // Noncompliant
colors[1] = "red";  // Oops! Explicit initialization means that the 0th element is left empty
colors[2] = "green";
colors[2] = "blue"; // Oops again! "green" overwritten
----
