== Why is this an issue?

Correctly updating a ``++static++`` field from a non-static method is tricky to get right and could easily lead to bugs if there are multiple class instances  and/or multiple threads in play. 


This rule raises an issue each time a ``++static++`` field is updated from a non-static method or property.

=== Noncompliant code example

[source,csharp]
----
public class MyClass 
{
  private static int count = 0;

  public void DoSomething() 
  {
    //...
    count++;  // Noncompliant
  }
}
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

* Make the enclosing instance [method|property] "static" or remove this set on the "static" field.
* Remove this set, which updates a "static" field from an instance [method|property].


=== Highlighting

Primary: static field assignment

Secondary: static field declaration


'''
== Comments And Links
(visible only on this page)

=== on 24 Jun 2015, 07:03:57 Tamas Vajk wrote:
\[~ann.campbell.2] I've created this subtask, because in C# this rule can also apply for properties. Furthermore, the ``++synchronized static++`` part doesn't work for C#.

=== on 24 Jun 2015, 14:01:07 Ann Campbell wrote:
looks good [~tamas.vajk]

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
