This rule raises an issue when SOQL queries filter records using the `LastModifiedDate` field in WHERE clauses.

== Why is this an issue?

In Salesforce, query performance depends heavily on whether the query can use indexed fields. The platform automatically creates indexes on certain system fields to speed up data retrieval.

`LastModifiedDate` is not indexed by default, which means queries filtering on this field require full table scans. When your organization has large amounts of data (over 100,000 records), these non-selective queries can become extremely slow or even fail due to governor limits.

`SystemModStamp`, on the other hand, is automatically indexed by Salesforce. Both fields track when records were last modified, but `SystemModStamp` provides the same filtering capability with much better performance.

The difference becomes critical in production environments with large datasets, where non-selective queries can cause timeouts, poor user experience, and may trigger Salesforce's non-selective query prevention mechanisms.

=== What is the potential impact?

Using `LastModifiedDate` in SOQL queries can lead to:

* Slow query execution times, especially with large datasets
* Potential query timeouts and failures
* Poor application performance and user experience
* Risk of hitting Salesforce governor limits
* Non-selective query errors when querying objects with more than 100,000 records

== How to fix it

Replace `LastModifiedDate` with `SystemModStamp` in your SOQL WHERE clauses. Both fields serve the same purpose for filtering recently modified records, but `SystemModStamp` is indexed and provides better performance.

=== Code examples

==== Noncompliant code example

[source,apex,diff-id=1,diff-type=noncompliant]
----
List<Account> accounts = [SELECT Id, Name FROM Account WHERE LastModifiedDate > :yesterday]; // Noncompliant
----

==== Compliant solution

[source,apex,diff-id=1,diff-type=compliant]
----
List<Account> accounts = [SELECT Id, Name FROM Account WHERE SystemModStamp > :yesterday];
----

== Resources

=== Documentation

 * SOQL Best Practices - https://www.apexhours.com/soql-best-practices[Comprehensive guide to writing efficient SOQL queries in Salesforce]

 * Query Plan Tool - https://help.salesforce.com/s/articleView?id=sf.code_dev_console_query_plan.htm[Salesforce documentation on using the Query Plan tool to optimize queries]

 * Large Data Volumes Best Practices - https://help.salesforce.com/s/articleView?id=sf.dev_guides_large_data_volumes.htm[Salesforce guide for handling large data volumes efficiently]

=== Standards

 * Salesforce Performance Guidelines - https://developer.salesforce.com/docs/atlas.en-us.apexcode.meta/apexcode/apex_gov_limits.htm[Official Salesforce guidelines for writing performant Apex code]
