== Why is this an issue?

Nested comments are confusing and can lead maintainers to misunderstand which code is active.


=== Noncompliant code example

[source,swift]
----
/*
  This is a comment block.
  It may be difficult to figure out that the following line of code is actually commented


variable = function_call();
/* variable contains the result. Noncompliant; inner comment */
*/
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
