== Why is this an issue?

There is no point in creating a ``++const++`` reference to a literal numeric value. Most likely the intent was not to create a reference, but a constant value.


=== Noncompliant code example

[source,cpp]
----
const int & weekdayCount = 7;  // Noncompliant 
----


=== Compliant solution

[source,cpp]
----
const int weekdayCount = 7;
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
