== Why is this an issue?

There is no point in creating a ``++const++`` reference to a literal numeric value. Most likely the intent was not to create a reference, but a constant value.


=== Noncompliant code example

[source,cpp]
----
const int & weekdayCount = 7;  // Noncompliant 
----


=== Compliant solution

[source,cpp]
----
const int weekdayCount = 7;
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Make this reference a constant value by removing the "&".


'''
== Comments And Links
(visible only on this page)

=== on 8 Apr 2015, 19:00:31 Evgeny Mandrikov wrote:
\[~ann.campbell.2] actually

{noformat}

int & weekdayCount = 7;

{noformat}

not allowed per standard, so I deleted part about removal of "const" qualifier from message.

endif::env-github,rspecator-view[]
