<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>The use of punctuation characters to separate subgroups in a number can make the number more readable. For instance consider 1,000,000,000 versus 1000000000. But when the grouping is irregular, such as 1,000,00,000; it indicates an error.</p>
</div>
<div class="paragraph">
<p>This rule raises an issue when underscores (<code>_</code>) are used to break a number into irregular subgroups.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-text" data-lang="text">int thousand = 100_0;
int tenThousand = 100_00;
int million = 1_000_00_000;</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-text" data-lang="text">int thousand = 1000;
int tenThousand = 10_000;
int tenThousandWithout = 10000;
int duos = 1_00_00;
int million = 100_000_000;</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Review this number; its irregular pattern indicates an error.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>the number</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_relates_to_s2148">relates to: <a data-rspec-id="S2148" class="rspec-auto-link">S2148</a></h3>

</div>
<div class="sect2">
<h3 id="_on_27_jun_2018_092817_philipp_menke_wrote">on 27 Jun 2018, 09:28:17 Philipp Menke wrote:</h3>
<div class="paragraph">
<p>This rule may though mark groupings as erroneous, which are perfectly valid in large parts of the world. E.g. the <a href="https://en.wikipedia.org/wiki/Indian_numbering_system">indian numbering system</a> uses groups of 2 and 3 digits (as famously explained on <a href="https://www.youtube.com/watch?v=l4bmZ1gRqCc&amp;t=275">Numberphile</a>).</p>
</div>
<div class="paragraph">
<p>Were such systems left out intentionally, or just not thought of on rule definition?</p>
</div>
<div class="paragraph">
<p>Even though i&#8217;m not really affected by this, i felt obligated to bring this up.</p>
</div>
<div class="paragraph">
<p>Kind Regards,</p>
</div>
<div class="paragraph">
<p>Philipp</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_3_jul_2018_222315_ann_campbell_wrote">on 3 Jul 2018, 22:23:15 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>In fact, we had long debates about this very point [~pmenke] and decided to go with the rule anyway.</p>
</div>
</div>
</div>
</div>