== Why is this an issue?

Strings are immutable objects, so concatenation doesn't simply add the new String to the end of the existing string. Instead, in each loop iteration, the first String is converted to an intermediate object type, the second string is appended, and then the intermediate object is converted back to a String. Further, performance of these intermediate operations degrades as the String gets longer. Therefore, the use of StringBuilder is preferred.


=== Noncompliant code example

[source,java]
----
String str = "";
for (int i = 0; i < arrayOfStrings.length ; ++i) {
  str = str + arrayOfStrings[i];
}
----


=== Compliant solution

[source,java]
----
StringBuilder bld = new StringBuilder();
  for (int i = 0; i < arrayOfStrings.length; ++i) {
    bld.append(arrayOfStrings[i]);
  }
  String str = bld.toString();
----


[options="header"]
|===
| Method| size| Runtime| Average time| Error margin
| plus| 100| Temurin 21| 4.19 &micro;s/op| ±0.34 &micro;s/op
| plus| 1000| Temurin 21| 377.08 &micro;s/op| ±17.36 &micro;s/op
| plus| 10000| Temurin 21| 40221.49 &micro;s/op| ±1342.76 &micro;s/op
| plus| 100000| Temurin 21| 5286840.53 &micro;s/op| ±185796.75 &micro;s/op
| stringBuilder| 100| Temurin 21| 0.97 &micro;s/op| ±0.03 &micro;s/op
| stringBuilder| 1000| Temurin 21| 10.25 &micro;s/op| ±1.64 &micro;s/op
| stringBuilder| 10000| Temurin 21| 93.27 &micro;s/op| ±16.05 &micro;s/op
| stringBuilder| 100000| Temurin 21| 1019.91 &micro;s/op| ±69.58 &micro;s/op
|===


*Benchmarking code*

The results were generated by running the following snippet with https://github.com/openjdk/jmh[JMH].

[source,java]
----
@Param({"100", "1000", "10000", "100000"})
int size;
private String word = "append";

@Benchmark
public String plus() {
  String str = "";
  for (int i = 0; i < size; i++) {
    str = str + word;
  }
  return str;
}

@Benchmark
public String stringBuilder() {
  StringBuilder builder = new StringBuilder();
  for (int i = 0; i < size; i++) {
    builder.append(word);
  }
  return builder.toString();
}
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

=== on 21 Nov 2024, 16:48:00 Alban Auzeill wrote:
[test-code-support-investigation-for-java] Decision for scope: Main -> All.

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
