include::../common/rule.adoc[tag=header]

=== Code examples

==== Noncompliant code example

[source,html,diff-id=1,diff-type=noncompliant]
----
<a href="javascript:void(0)" onClick="foo">Perform action</a>
<a href="#" onClick="foo">Perform action</a>
<a onClick="foo">Perform action</a>
----

==== Compliant solution

[source,html,diff-id=1,diff-type=compliant]
----
<button onClick="foo">Perform action</button>
<a href="#section" onClick="foo" />
----

include::../common/rule.adoc[tag=footer]
