== Why is this an issue?

Using a data value clause in the ``++LINKAGE SECTION++`` can lead to unexpected behavior at runtime.


=== Noncompliant code example

[source,cobol]
----
LINKAGE SECTION.

01 CAR-ID PIC X(20) VALUE IS "VOLVO".   *> Noncompliant

01  EMP-TYPE     PIC X.
   88   FULL-TIME-EMPLOYEE VALUE "F".   *> Compliant; this is a condition name
   88   PART-TIME-EMPLOYEE VALUE "P".

01 TRAIN-ID PIC X(20)
----

