<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule checks that the specified attributes are not present in HTML tags.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="paragraph">
<p>For a disallowed attribute list of: <code>a.name</code>:</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-html" data-lang="html">&lt;a href="blah.com" name="Blah link"&gt; &lt;!-- Noncompliant; name attribute is used --&gt;</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-html" data-lang="html">&lt;a href="blah.com"&gt;</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove the "xxx" attribute from the "yyy" tag</p>
</div>
</div>
<div class="sect2">
<h3 id="_parameters">Parameters</h3>
<div class="sidebarblock">
<div class="content">
<div class="title">attributes</div>
<div class="paragraph">
<p>Comma-separated list of tag.attributes that are not allowed. E.G. a.name forbids a "name" attribute in an "a" tag.</p>
</div>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_22_aug_2014_140456_freddy_mallet_wrote">on 22 Aug 2014, 14:04:56 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>@Ann, looks like the list of default values for property 'attributes' is missing.</p>
</div>
<div class="paragraph">
<p>BTW, this was not a brand new rule &amp; is not related to HTML 5. It&#8217;s a catch-up entry for an old rule that was already implemented in the Web plugin but had no RSpec.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_22_aug_2014_193014_ann_campbell_wrote">on 22 Aug 2014, 19:30:14 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~freddy.mallet] must there always be a default value? This rule is not on by default, and is a user-specified "don&#8217;t use this attribute" rule.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_25_aug_2014_074007_freddy_mallet_wrote">on 25 Aug 2014, 07:40:07 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>Let&#8217;s say [~ann.campbell.2] that all mandatory properties should have a default value out of the box to prevent generating some analysis failure when activating the rule without providing a property value. But indeed here this is a template rule, so everything is fine.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_25_aug_2014_140328_ann_campbell_wrote">on 25 Aug 2014, 14:03:28 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>In the interest of completeness: this RSpec created to document an existing rule in the Web plugin which is neither fish nor fowl (i.e. not a template rule and no defaults)</p>
</div>
</div>
</div>
</div>