<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Typically, backslashes are seen only as part of escape sequences. Therefore, the use of a backslash outside of a raw string or escape sequence looks suspiciously like a broken escape sequence.</p>
</div>
<div class="paragraph">
<p>Characters recognized as escape-able are: <code>abfnrtvox\'"</code></p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">s = "Hello \world."
t = "Nice to \ meet you"
u = "Let's have \ lunch"</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">s = "Hello world."
t = "Nice to \\ meet you"
u = r"Let's have \ lunch"  // raw string</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove this "\", add another "\" to escape it, or make this a raw string.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_4_nov_2014_155900_ann_campbell_wrote">on 4 Nov 2014, 15:59:00 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>pylint:W1401</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_17_apr_2020_151827_nicolas_harraudeau_wrote">on 17 Apr 2020, 15:18:27 Nicolas Harraudeau wrote:</h3>
<div class="paragraph">
<p>Deprecating this rule.</p>
</div>
<div class="paragraph">
<p>It raises many issues and most of them look like False Positives. The most common false positives are strings  containing regular expressions. Example:</p>
</div>
<div class="listingblock">
<div class="content">
<pre>import re
re.sub('\(', '[', '(test)')  # False Positive</pre>
</div>
</div>
<div class="paragraph">
<p>This rule raises 380+ issues just on Pypy.</p>
</div>
<div class="paragraph">
<p>We could try to reduce the number of false positives by focusing only on strings passed to functions which do not use backslashes, such as <code>print</code> or <code>Exception</code>, but the value seems to be low. The worst that could happen is that the backslash is displayed as part of the string or that a few characters are misinterpreted:</p>
</div>
<div class="listingblock">
<div class="content">
<pre>In [36]: raise Exception("\.")
---------------------------------------------------------------------------
Exception                                 Traceback (most recent call last)
&lt;ipython-input-36-de3222772775&gt; in &lt;module&gt;
----&gt; 1 raise Exception("\.")

Exception: \.


In [41]: raise Exception("\234324")
---------------------------------------------------------------------------
Exception                                 Traceback (most recent call last)
&lt;ipython-input-41-d0bd825b64dd&gt; in &lt;module&gt;
----&gt; 1 raise Exception("\234324")

Exception: Âœ324</pre>
</div>
</div>
</div>
</div>
</div>