<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Because <code>case</code> statements in a Swift <code>switch</code> do not fall through, there is no need to use <code>break</code> at the end of a <code>case</code> unless it would otherwise be empty. Since an empty <code>case</code> isn&#8217;t allowed, an explicit <code>break</code> is needed to make such code compilable. There is no other reason to use <code>break</code> in a <code>case</code>.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-swift" data-lang="swift">switch weekday {
  case sunday:
    break
  case monday:
    getUpEarly()
    break  // Noncompliant
  case tuesday
    // ...
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-swift" data-lang="swift">switch weekday {
  case sunday:
    break
  case monday:
    getUpEarly()
  case tuesday
    // ...
}</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove this extraneous "break".</p>
</div>
</div>
</div>
</div>