== Why is this an issue?

The prolog header is the following piece of code that some XML documents start with:

----
<?xml version="y.x" encoding="zzzzz"?>
----

When the prolog exists in an XML document, it should be at the beginning of the file to enable programs to determine the encoding of non-UTF-8, non-UTF-16 files.


=== Noncompliant code example

[source,xml]
----
<!-- Generated file -->  <!--  Noncompliant  -->
<?xml version="1.0" encoding="UTF-8"?>
<firstNode>
  content
</firstNode>
----


=== Compliant solution

[source,xml]
----
<?xml version="1.0" encoding="UTF-8"?>
<!-- Generated file -->
<firstNode>
  content
</firstNode>
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Remove all characters located before "<?xml".


endif::env-github,rspecator-view[]
