== Why is this an issue?

This rule raised an issue when a component is used in the ``++render()++`` method and not ``++import++``ed or defined. 


=== Noncompliant code example

[source,javascript]
----
<Rating value={this.props.measure.reliability_rating} />;
----


=== Compliant solution

[source,javascript]
----
import Rating from './Rating'
[...]
<Rating value={this.props.measure.reliability_rating} />
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Add the missing import of "XXX".


=== Highlighting

Primary: first occurrence of the component usage

Secondary: all other places in the file where this component is used


endif::env-github,rspecator-view[]
