== Why is this an issue?

In Blazor, when a https://learn.microsoft.com/en-us/aspnet/core/blazor/fundamentals/routing#route-parameters[route parameter] is used in a https://learn.microsoft.com/en-us/aspnet/core/blazor/fundamentals/routing#route-templates[routed component], the router will populate the https://learn.microsoft.com/en-us/aspnet/core/blazor/components#component-parameters[component parameter] with the same name.

If the routed component does not contain a matching parameter, a runtime error will be raised:

[source,text]
----
Unhandled exception rendering component: Object of type '<component-type>' does not have a property matching the name '<route-parameter-name>'.
----

== How to fix it

Ensure to have a component parameter that matches the route parameter name, or remove the route parameter if not needed or used.

=== Code examples

==== Noncompliant code example

[source,csharp,diff-id=1,diff-type=noncompliant]
----
@page "/user/{Id:int}" // Noncompliant

@code {
}
----

==== Compliant solution

[source,csharp,diff-id=1,diff-type=compliant]
----
@page "/user/{Id:int}" // Compliant

@code {
    [Parameter]
    public int Id { get; set; }
}
----

== Resources

=== Documentation

* Microsoft Learn - https://learn.microsoft.com/en-us/aspnet/core/blazor/fundamentals/routing#route-parameters[ASP.NET Core Blazor routing and navigation - Route parameters]
* Microsoft Learn - https://learn.microsoft.com/en-us/aspnet/core/blazor/fundamentals/routing#route-templates[ASP.NET Core Blazor routing and navigation - Route templates]
* Microsoft Learn - https://learn.microsoft.com/en-us/aspnet/core/blazor/components#component-parameters[ASP.NET Core Razor components - Component parameters]
