== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,python]
----
r"foo()" # Noncompliant, will match only 'foo'
----

=== Compliant solution

[source,python]
----
r"foo\(\)" # Matches 'foo()'
----

include::../implementation.adoc[]
