include::../description.adoc[]

== Noncompliant Code Example

[source,python]
----
r"foo()" # Noncompliant, will match only 'foo'
----

== Compliant Solution

[source,python]
----
r"foo\(\)" # Matches 'foo()'
----

include::../implementation.adoc[]
