<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>In a multithreaded environment, a thread may need to wait for a particular condition to become true.
One way of pausing execution in Java is <code>Thread.sleep(&#8230;&#8203;)</code>.</p>
</div>
<div class="paragraph">
<p>If a thread that holds a lock calls <code>Thread.sleep(&#8230;&#8203;)</code>, no other thread can acquire said lock.
This can lead to performance and scalability issues, in the worst case leading to deadlocks.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Call <code>wait(&#8230;&#8203;)</code> on the monitor object instead of using <code>Thread.sleep(&#8230;&#8203;)</code>.
While <code>wait(&#8230;&#8203;)</code> is executed, the lock is temporarily released and hence other threads can run in the meantime.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">public void doSomething(){
  synchronized(monitor) {
    while(notReady()){
      Thread.sleep(200); // Noncompliant, any other thread synchronizing on monitor is blocked from running while the first thread sleeps.
    }
    process();
  }
  ...
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">public void doSomething(){
  synchronized(monitor) {
    while(notReady()){
      monitor.wait(200); // Compliant, the current monitor is released.
    }
    process();
  }
  ...
}</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p><a href="https://wiki.sei.cmu.edu/confluence/x/YTdGBQ">CERT, LCK09-J.</a> - Do not perform operations that can block while holding a lock</p>
</li>
</ul>
</div>
<hr>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Replace the call to "Thread.sleep(&#8230;&#8203;)" with a call to "wait(&#8230;&#8203;)".</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_15_feb_2016_171718_ann_campbell_wrote">on 15 Feb 2016, 17:17:18 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>This RSpec exlicitly <em>not</em> expanded to cover full scope of CERT, LCK09-J. because the other operations <em>can</em> lead to problems (but may not), whereas for <code>sleep</code> there is a problem sure.</p>
</div>
</div>
</div>
</div>