A ``++CATCH++`` clause that only rethrows the caught exception has the same effect as omitting the ``++CATCH++`` altogether and letting it bubble up automatically, but with more code and the additional detriment of leaving maintainers scratching their heads. 


Such clauses should either be eliminated or populated with the appropriate logic.

== Noncompliant Code Example

[source,sql]
----
BEGIN TRY
    SELECT 1/0;
END TRY
BEGIN CATCH -- Noncompliant
    THROW;
END CATCH;
----

== Compliant Solution

[source,sql]
----
BEGIN TRY
    SELECT 1/0;
END TRY
BEGIN CATCH
    EXECUTE usp_GetErrorInfo; 
    THROW;
END CATCH;
----
or

----
SELECT 1/0;
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
