To improve the code readability, the explicit line continuation character, ``_``, should not be used. Instead, it is better to break lines after an operator.


== Noncompliant Code Example

----
Module Module1
    Sub Main()
        ' Noncompliant
        Console.WriteLine("Hello" _
                          & "world")
    End Sub
End Module
----


== Compliant Solution

----
Module Module1
    Sub Main()

        Console.WriteLine("Hello" &
                          "world")
    End Sub
End Module
----

