In Swift, the semicolon (``++;++``) is optional as a statement separator, but omitting semicolons can be confusing.

== Noncompliant Code Example

----
var x = 1
----

== Compliant Solution

----
var x = 1;
----

ifdef::rspecator-view[]
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::rspecator-view[]
