== Why is this an issue?

In Swift, the semicolon (``++;++``) is optional as a statement separator, but omitting semicolons can be confusing.

=== Noncompliant code example

[source,swift]
----
var x = 1
----

=== Compliant solution

[source,swift]
----
var x = 1;
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
