include::../common/rule.adoc[tag=header]

== How to fix it

Make sure that DOM elements with the `aria-activedescendant` property have a `tabindex` property, or use an element with an inherent one.

=== Code examples

==== Noncompliant code example

[source,html,diff-id=1,diff-type=noncompliant]
----
<div aria-activedescendant="descendantId">
  <div id="descendantId"></div>
</div>
----

==== Compliant solution

[source,html,diff-id=1,diff-type=compliant]
----
<div aria-activedescendant="descendantId" tabIndex="0">
  <div id="descendantId"></div>
</div>
----

include::../common/rule.adoc[tag=footer]