== Why is this an issue?

Any insert which omits a value for a `NOT NULL` column in a database table will be automatically rejected by the database unless a default value has been specified for the column.

include::../../../shared_content/plsql/data_dictionary.adoc[]

== How to fix it

Ensure that all `NOT NULL` columns have a value specified in the `INSERT` statement.

=== Code examples

Given the following table:

[source,sql]
----
CREATE TABLE MY_TABLE (
  N1 NUMBER NOT NULL,
  N2 VARCHAR2(50) NOT NULL,
  N3 VARCHAR2(50) DEFAULT 'Default Value'
);
----

==== Noncompliant code example

[source,sql,diff-id=1,diff-type=noncompliant]
----
INSERT INTO MY_TABLE  -- Noncompliant; N2 value omitted
(
  N1
)
VALUES
(
  1
)
----

=== Compliant solution

[source,sql,diff-id=1,diff-type=compliant]
----
INSERT INTO MY_TABLE  -- Compliant; N3 has a default value
(
  N1
)
(
  N1,
  N2
)
VALUES
(
  1,
  'Paul'
)
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../highlighting.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
