<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>88-level variables, also known as "condition name" variables, represent possible values of the "conditional variables" they&#8217;re tied to. An unused "condition name" variable is dead code. Such variables should be removed to increase the maintainability of the program.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cobol" data-lang="cobol">01 COLOR PIC X.
  88 COL-YELLOW VALUE 'Y'.
  88 COL-GREEN VALUE 'G'. *&gt; Noncompliant; not used
  88 COL-RED VALUE 'R'.

* ...
IF COL-YELLOW
* ...
END-IF
IF COL-RED
* ...
END-IF</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cobol" data-lang="cobol">01 COLOR PIC X.
  88 COL-YELLOW VALUE 'Y'.
  88 COL-RED VALUE 'R'.

* ...
IF COL-YELLOW
* ...
END-IF
IF COL-RED
* ...
END-IF</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>This condition name variable "XXXX" is unused and so should be removed.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>The unused condition name.</p>
</div>
</div>
</div>
</div>