<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Initializing a vector with a chain of <code>push</code> invocations is less efficient than making the initialization with a single vector literal. Each invocation has a cost in itself.</p>
</div>
<div class="paragraph">
<p>On top of the performance argument, the vector literal is more concise and easier to read.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Introduce a single vector literal that includes all the elements that are added to the vector as arguments of the <code>push</code> invocations.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-rust" data-lang="rust">let mut v = Vec::new();
v.push(0);
v.push(1);
v.push(2);</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-rust" data-lang="rust">let v = vec![0, 1, 2];</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Clippy Lints - <a href="https://rust-lang.github.io/rust-clippy/master/index.html#vec_init_then_push" class="bare">https://rust-lang.github.io/rust-clippy/master/index.html#vec_init_then_push</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>