== Why is this an issue?

In C#, the https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/operators/type-testing-and-cast#is-operator[`is` type testing operator] can be used to check if the run-time type of an object is compatible with a given type. If the object is not null, then the `is` operator performs a cast, and so performing another cast following the check result is redundant.

This can impact:

* Performance: Performing the type check and cast separately can lead to minor performance issues. While this might not be noticeable in small applications, it can add up in larger, more complex systems.
* Readability: The code is less readable and less clean because it requires two lines (and two operations) to achieve something that could be done in one.

== How to fix it

Use https://learn.microsoft.com/en-us/dotnet/csharp/fundamentals/functional/pattern-matching[pattern macthing] to perform the check and retrieve the cast result.

=== Code examples

==== Noncompliant code example

[source,csharp,diff-id=1,diff-type=noncompliant]
----
if (x is Fruit)  // Noncompliant
{
  var f = (Fruit)x; // or x as Fruit
  // ...
}
----


==== Compliant solution

[source,csharp,diff-id=1,diff-type=compliant]
----
if (x is Fruit fruit)
{
  // ...
}
----

== Resources

=== Documentation

* Microsoft Learn - https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/operators/type-testing-and-cast[Type-testing operators and cast expressions - `is`, `as`, `typeof` and casts]
* Microsoft Learn - https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/operators/is[is operator (C# reference)]
* Microsoft Learn - https://learn.microsoft.com/en-us/dotnet/csharp/fundamentals/functional/pattern-matching[Pattern matching overview]

include::../rspecator.adoc[]
