<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when a <code>@Grab</code> annotation uses the verbose format with named parameters (<code>group</code>, <code>module</code>, <code>version</code>) but no additional parameters are specified.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>The <code>@Grab</code> annotation in Groovy supports two formats for specifying dependencies: a verbose format with named parameters and a concise shorthand format using the Maven coordinate style.</p>
</div>
<div class="paragraph">
<p>The verbose format requires explicitly naming each parameter: <code>@Grab(group='org.springframework', module='spring-orm', version='5.2.8.RELEASE')</code></p>
</div>
<div class="paragraph">
<p>The shorthand format uses the familiar Maven coordinate pattern: <code>@Grab('org.springframework:spring-orm:5.2.8.RELEASE')</code></p>
</div>
<div class="paragraph">
<p>When only the basic dependency information (group, module, version) is needed, the shorthand notation offers several advantages:</p>
</div>
<div class="ulist">
<ul>
<li>
<p><strong>Improved readability</strong>: The shorthand format is more compact and easier to scan visually</p>
</li>
<li>
<p><strong>Familiar syntax</strong>: It follows the same pattern used in Maven, Gradle, and other build tools</p>
</li>
<li>
<p><strong>Reduced verbosity</strong>: Less code to write and maintain</p>
</li>
<li>
<p><strong>Consistency</strong>: Aligns with common practices in the Java ecosystem</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>The verbose format should be reserved for cases where additional parameters are needed, such as <code>classifier</code>, <code>transitive</code>, <code>changing</code>, or <code>conf</code> options.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Replace the verbose @Grab annotation with the shorthand notation using the 'group:module:version' format.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-groovy" data-lang="groovy">@Grab(group='org.springframework', module='spring-orm', version='5.2.8.RELEASE') // Noncompliant
import org.springframework.jdbc.core.JdbcTemplate</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-groovy" data-lang="groovy">@Grab('org.springframework:spring-orm:5.2.8.RELEASE') // Compliant
import org.springframework.jdbc.core.JdbcTemplate</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Groovy documentation - <a href="https://groovy-lang.org/grape.html">The Grape dependency manager</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>