//// This rule is superseded by RSPEC-5547
//// Please consider implementing this latter instead.
include::../description.adoc[]

== Noncompliant Code Example

----
CCCryptorStatus cryptStatus = CCCrypt(kCCEncrypt,
                                      kCCAlgorithmDES, // Noncompliant
                                      kCCOptionPKCS7Padding | kCCOptionECBMode,
                                      keyPtr,
                                      kCCKeySizeDES,
                                      NULL,
                                      [self bytes],
                                      dataLength,
                                      buffer,
                                      bufferSize
                                      &numBytesEncrypted);
----

include::../compliant.adoc[]

include::../see.adoc[]

ifdef::env-github,rspecator-view[]
'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
