== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,cobol]
----
       IDENTIFICATION DIVISION.
       PROGRAM-ID. foo.

       PROCEDURE DIVISION.

      * Noncompliant, 4 operands are found, higher than the maximum allowed 3
           IF WS-FOO(1) = 1 OR
              WS-FOO(2) = 2 OR
              WS-FOO(3) = 3 OR
              WS-BAR = 4 OR
              WS-BAZ = 5 OR
              WS-QUX = 42
           END-IF.
       END PROGRAM foo.
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../parameters.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
