== Why is this an issue?

While the properties of a ``++readonly++`` reference type field can still be changed after initialization, those of a ``++readonly++`` value field, such as a ``++struct++``, cannot. 


If the member could be either a ``++class++`` or a ``++struct++`` then assignment to its properties could be unreliable, working sometimes but not others.


=== Noncompliant code example

[source,csharp]
----
interface IPoint
{
  int X { get; set; }
  int Y { get; set; }
}

class PointManager<T> where T: IPoint 
{
  readonly T point;  // this could be a struct
  public PointManager(T point)
  {
    this.point = point;
  }

  public void MovePointVertically(int newX)
  {
    point.X = newX; //Noncompliant; if point is a struct, then nothing happened
    Console.WriteLine(point.X);
  }
}
----


=== Compliant solution

[source,csharp]
----
interface IPoint
{
  int X { get; set; }
  int Y { get; set; }
}

class PointManager<T> where T : IPoint 
{
  readonly T point;  // this could be a struct
  public PointManager(T point)
  {
    this.point = point;
  }

  public void MovePointVertically(int newX) // assignment has been removed
  {
    Console.WriteLine(point.X);
  }
}
----

or


[source,csharp]
----
interface IPoint
{
  int X { get; set; }
  int Y { get; set; }
}

class PointManager<T> where T : class, IPoint 
{
  readonly T point;  // this can only be a class
  public PointManager(T point)
  {
    this.point = point;
  }

  public void MovePointVertically(int newX)
  {
    point.X = newX;  // this assignment is guaranteed to work
    Console.WriteLine(point.X);
  }
}
----



ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Restrict "x" to be a reference type or remove this assignment of "y"; it is useless if "x" is a value type.


'''
== Comments And Links
(visible only on this page)

=== on 13 May 2015, 20:13:12 Ann Campbell wrote:
\[~tamas.vajk]

* I've got the name of the equivalent R# rule, but I'm not sure how to find its key...
* I suspect I may have missed some of the subtleties intended in this rule. Feel free to start an edit iteration

=== on 18 May 2015, 09:15:33 Tamas Vajk wrote:
Updated the rule. Could you please check its language, and if it makes sense to you too?

=== on 21 May 2015, 14:34:22 Ann Campbell wrote:
see what you think [~tamas.vajk]

=== on 22 May 2015, 11:28:16 Tamas Vajk wrote:
Looks good, I adjusted some wording in the first sentence of the description

=== on 22 May 2015, 12:10:14 Ann Campbell wrote:
Thanks [~tamas.vajk]. Looks good

endif::env-github,rspecator-view[]
