This rule raises an issue when an Apex test method (annotated with `@isTest`) does not contain any assertion statements (`System.assert`, `System.assertEquals`, or `System.assertNotEquals`).

== Why is this an issue?

Test methods without assertions provide no validation of functionality and serve only to achieve code coverage without ensuring correctness. The primary purpose of unit testing is to verify that code behaves as expected under various conditions.

When a test method executes code but doesn't assert any expected outcomes, it fails to fulfill this fundamental purpose. Such tests can give developers a false sense of security, as they appear to be testing the code but actually provide no verification that the functionality works correctly.

Meaningful assertions validate specific expected behaviors, such as:

* Verifying that objects are created with correct field values
* Ensuring that calculations produce expected results
* Confirming that exceptions are thrown under appropriate conditions
* Checking that database operations complete successfully

Without these validations, bugs can easily slip through the testing process undetected. A test that simply executes code without verifying results is essentially worthless for quality assurance purposes.

=== What is the potential impact?

Tests without assertions can lead to several serious consequences:

* **False confidence**: Developers may believe their code is properly tested when it actually lacks meaningful validation
* **Undetected bugs**: Issues in the code may go unnoticed because no assertions verify correct behavior
* **Poor code quality**: Without proper validation, defects can make it to production environments
* **Maintenance difficulties**: When code changes break functionality, tests without assertions won't catch the regressions
* **Wasted effort**: Time spent writing and maintaining tests that provide no actual verification value

== How to fix it

Add meaningful assertion statements to verify the expected behavior of your test. Use System.assertEquals to check that actual values match expected values, System.assertNotEquals to verify values are different from unexpected ones, and System.assert for boolean conditions.

=== Code examples

==== Noncompliant code example

[source,apex,diff-id=1,diff-type=noncompliant]
----
@isTest
static void testAccountCreation() {
    Account acc = new Account(Name = 'Test');
    insert acc;
    // Noncompliant - no assertions to verify behavior
}
----

==== Compliant solution

[source,apex,diff-id=1,diff-type=compliant]
----
@isTest
static void testAccountCreation() {
    Account acc = new Account(Name = 'Test');
    insert acc;
    
    Account insertedAccount = [SELECT Id, Name FROM Account WHERE Id = :acc.Id];
    System.assertEquals('Test', insertedAccount.Name, 'Account name should match expected value');
    System.assertNotEquals(null, insertedAccount.Id, 'Account should have been assigned an ID');
}
----

== Resources

=== Documentation

 * Apex Testing Best Practices - https://developer.salesforce.com/docs/atlas.en-us.apexcode.meta/apexcode/apex_testing_best_practices.htm[Official Salesforce documentation on Apex testing best practices including assertion usage]

 * System Class Assert Methods - https://developer.salesforce.com/docs/atlas.en-us.apexcode.meta/apexcode/apex_methods_system_system.htm[Documentation for System.assert, System.assertEquals, and System.assertNotEquals methods]

=== Standards

 * OWASP ASVS-14.2.1 - https://owasp.org/www-project-application-security-verification-standard/[Application Security Verification Standard requirements for unit test coverage and effectiveness]

=== Related rules

 * RSPEC-2699 - https://rules.sonarsource.com/csharp/RSPEC-2699/[Tests should include assertions (C# implementation of the same principle)]
