include::../description.adoc[]

== Noncompliant Code Example

----
int divide(int divisor, int dividend) {
  return divisor / dividend;
}

void doTheThing() {
  int divisor = 15;
  int dividend = 5;

  int result = divide(dividend, divisor);  // Noncompliant; operation succeeds, but result is unexpected
  //...
}
----

== Compliant Solution

----
int divide(int divisor, int dividend) {
  return divisor / dividend;
}

void doTheThing() {
  int divisor = 15;
  int dividend = 5;

  int result = divide(divisor, dividend);
  //...
}
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../highlighting.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
