== Why is this an issue?

When a loop variable is not used outside of a loop, it should be declared inside the loop declaration:

* It improves readability. The scope of the variable is clearly defined.
* It reduces the number of mistakes. The variable can't be accidentally misused outside of the loop.
* Resources are not retained longer than necessary.


=== Noncompliant code example

[source,cpp,diff-id=1,diff-type=noncompliant]
----
void f() {
  int i = 0; // Noncompliant: i is not used outside of the loop
  for (i = 0; i < 10; ++i) {
    std::cout << i << std::endl;
  }
}
----


=== Compliant solution

[source,cpp,diff-id=1,diff-type=compliant]
----
void f() {
  for (int i = 0; i < 10; ++i) {
    std::cout << i << std::endl;
  }
}
----


== Resources

* {cpp} Core Guidelines - https://github.com/isocpp/CppCoreGuidelines/blob/036324/CppCoreGuidelines.md#es5-keep-scopes-small[ES.5 - Keep scopes small]


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Declare the variable XXX inside the loop.


'''
== Comments And Links
(visible only on this page)

=== is related to: S806

endif::env-github,rspecator-view[]
