When a loop variable is not used outside of a loop, it should be declared inside the loop declaration:

* It improves readability, the scope of the variable is clearly defined
* It reduces the number of mistakes, the variable can't be accidentally misused outside of the loop
* Resources are not retained longer than necessary


== Noncompliant Code Example

----
using namespace std;

void f() {
  int i = 0; // Noncompliant: i is not used outside of the loop
  for (i = 0; i < 10; ++i) {
    cout << i << endl;
  }
}
----


== Compliant Solution

----
using namespace std;

void f() {
  for (int i = 0; i < 10; ++i) {
    cout << i << endl;
  }
}
----


== See

* https://github.com/isocpp/CppCoreGuidelines/blob/c553535fb8dda2839d13ab5f807ffbc66b63d67b/CppCoreGuidelines.md#es5-keep-scopes-small[{cpp} Core Guidelines ES.5] - Keep scopes small


ifdef::env-github,rspecator-view[]
'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
