<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>The point of using an optional is to signal that the value may be <code>nil</code> and to provide graceful ways of dealing with it if it is <code>nil</code>. While implicitly unwrapped optionals still provide means of dealing with <code>nil</code> values, they also signal that the value won&#8217;t be <code>nil</code>, and unwrap it automatically. In addition to sending a decidedly mixed signal, this could lead to runtime errors if the value ever is <code>nil</code>.</p>
</div>
<div class="paragraph">
<p>It is safest, and clearest to use either an optional or a plain type and avoid the boggy middle ground of implicitly unwrapped optionals.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-swift" data-lang="swift">var greeting : String!  // Noncompliant

println(greeting)  // At this point the value is nil. Runtime error results</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-swift" data-lang="swift">var greeting : String?

if let howdy = greeting {
  println(howdy)
}</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Convert this implicitly unwrapped optional to a plain optional.</p>
</div>
</div>
</div>
</div>