<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>For readability, the indentation of source code comments should match the indentation of the commented code.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-text" data-lang="text">    " GAC -  13 June 13 - output user data
WRITE: / 'Firstname'.
" GAC -  13 June 13 - output user data
    WRITE: / 'Lasttname'.</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-text" data-lang="text">" GAC -  13 June 13 - output user data
WRITE: / 'Firstname'.
    " GAC -  13 June 13 - output user data
    WRITE: / 'Lasttname'.</code></pre>
</div>
</div>
</div>
</div>
</div>