== Why is this an issue?

For readability, the indentation of source code comments should match the indentation of the commented code.


=== Noncompliant code example

[source,text]
----
    " GAC -  13 June 13 - output user data
WRITE: / 'Firstname'.
" GAC -  13 June 13 - output user data
    WRITE: / 'Lasttname'.
----


=== Compliant solution

[source,text]
----
" GAC -  13 June 13 - output user data
WRITE: / 'Firstname'.
    " GAC -  13 June 13 - output user data
    WRITE: / 'Lasttname'.
----

