For readability, the indentation of source code comments should match the indentation of the commented code.


== Noncompliant Code Example

----
    " GAC -  13 June 13 - output user data
WRITE: / 'Firstname'.
" GAC -  13 June 13 - output user data
    WRITE: / 'Lasttname'.
----


== Compliant Solution

----
" GAC -  13 June 13 - output user data
WRITE: / 'Firstname'.
    " GAC -  13 June 13 - output user data
    WRITE: / 'Lasttname'.
----

