<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>In applications where the accepted practice is to log an <code>Exception</code> and then rethrow it, you end up with miles-long logs that contain multiple instances of the same exception. In multi-threaded applications debugging this type of log can be particularly hellish because messages from other threads will be interwoven with the repetitions of the logged-and-thrown <code>Exception</code>. Instead, exceptions should be either logged or rethrown, not both.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-text" data-lang="text">catch (SQLException e) {
  ...
  LOGGER.log(Level.ERROR,  contextInfo, e);
  throw new MySQLException(contextInfo, e);
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-text" data-lang="text">catch (SQLException e) {
  ...
  throw new MySQLException(contextInfo, e);
}</code></pre>
</div>
</div>
<div class="paragraph">
<p>or</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-text" data-lang="text">catch (SQLException e) {
  ...
  LOGGER.log(Level.ERROR,  contextInfo, e);
  // handle exception...
}</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Either log this exception and handle it, or rethrow it with some contextual information.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="ulist">
<ul>
<li>
<p>Primary: The exception caught</p>
</li>
<li>
<p>Secondary 1: The logging statement:</p>
<div class="ulist">
<ul>
<li>
<p>message: 'Logging statement.'</p>
</li>
</ul>
</div>
</li>
<li>
<p>Secondary 2: The <code>throw</code> statement</p>
<div class="ulist">
<ul>
<li>
<p>message: 'Thrown exception.'</p>
</li>
</ul>
</div>
</li>
</ul>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_21_nov_2024_164800_alban_auzeill_wrote">on 21 Nov 2024, 16:48:00 Alban Auzeill wrote:</h3>
<div class="paragraph">
<p>[test-code-support-investigation-for-java] Decision for scope: Keep 'Main'. Test don&#8217;t always want to capture the logs, throwing any "Boom!" exception is enough.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_11_oct_2014_120959_freddy_mallet_wrote">on 11 Oct 2014, 12:09:59 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>Sounds good to me!</p>
</div>
</div>
</div>
</div>