<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Comparisons of dissimilar types will always return <code>false</code>. The comparison and all its dependent code can simply be removed. This includes:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>comparing an object with <code>null</code></p>
</li>
<li>
<p>comparing an object with an unrelated primitive (e.g. a <code>String</code> with an <code>int</code>)</p>
</li>
<li>
<p>comparing unrelated types</p>
</li>
</ul>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-dart" data-lang="dart">void f() {
  var a = "Hello, World!";
  if (a == 42) { // Noncompliant: comparing a String with an int
    print("BOOM!");
  }
}</code></pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Dart Docs - <a href="https://dart.dev/tools/linter-rules/unrelated_type_equality_checks">Dart Linter rule - unrelated_type_equality_checks</a></p>
</li>
</ul>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>The type of the right operand ('bool') isn&#8217;t a subtype or a supertype of the left operand ('int').</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>The <code>==</code> operator or <code>!=</code> operator.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
</div>
</div>