<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>In some cases a comparison with operators <code>==</code>, or <code>!=</code> will always return True or always return False. When this happens, the comparison and all its dependent code can simply be removed. This includes:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>comparing unrelated builtin types such as string and integer.</p>
</li>
<li>
<p>comparing class instances which do not implement <code>__eq__</code> or <code>__ne__</code> to an object of a different type (builtin or from an unrelated class which also doesn&#8217;t implement <code>__eq__</code> or <code>__ne__</code>).</p>
</li>
</ul>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">foo = 1 == "1"  # Noncompliant. Always False.

foo = 1 != "1"  # Noncompliant. Always True.

class A:
    pass

myvar = A() == 1  # Noncompliant. Always False.
myvar = A() != 1  # Noncompliant. Always True.</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">foo = 1 == int("1")

foo = str(1) != "1"

class Eq:
    def __eq__(self, other):
        return True

myvar = Eq() == 1
myvar = 1 == Eq()
myvar = Eq() != 1  # Ok. "__ne__" calls "__eq__" by default
myvar = 1 != Eq()</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="ulist">
<ul>
<li>
<p>Remove this equality check between incompatible types X and Y; it will always return True/False</p>
</li>
</ul>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_deprecates_s1294">deprecates: <a data-rspec-id="S1294" class="rspec-auto-link">S1294</a></h3>

</div>
<div class="sect2">
<h3 id="_deprecates_s1318">deprecates: <a data-rspec-id="S1318" class="rspec-auto-link">S1318</a></h3>

</div>
<div class="sect2">
<h3 id="_is_related_to_s5845">is related to: <a data-rspec-id="S5845" class="rspec-auto-link">S5845</a></h3>

</div>
<div class="sect2">
<h3 id="_on_15_oct_2014_220422_freddy_mallet_wrote">on 15 Oct 2014, 22:04:22 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>My 2 cents:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>I would prefer the "Reliability" SQALE category because such kind of comparison can&#8217;t be done on purpose</p>
</li>
<li>
<p>Providing a compliant solution is valueless and can even be misleading</p>
</li>
<li>
<p>I would extend the Noncompliant Code Example to provide an example for each kind silly equality checks</p>
</li>
</ul>
</div>
</div>
<div class="sect2">
<h3 id="_on_21_oct_2014_151606_nicolas_peru_wrote">on 21 Oct 2014, 15:16:06 Nicolas Peru wrote:</h3>
<div class="paragraph">
<p>How would we detect cases such as  :</p>
</div>
<div class="listingblock">
<div class="content">
<pre>Interface I {}
class A extends B implements I{}
class B {}
...
B b = new A();
I i;
b.equals(i); //On this call B and I have nothing in common in the class hierarchy, but this can return true.</pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_on_22_oct_2014_132905_ann_campbell_wrote">on 22 Oct 2014, 13:29:05 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>Let me know which cases you want to cover</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_22_apr_2015_163644_samuel_mercier_wrote">on 22 Apr 2015, 16:36:44 Samuel Mercier wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] As a first step I will implement the following:</p>
</div>
<div class="paragraph">
<p>EC: equals() used to compare array and non-array (EC_ARRAY_AND_NONARRAY)</p>
</div>
<div class="listingblock">
<div class="content">
<pre>Object[] array = ...;
MyObject object = ...;
array.equals(object); // Always false
object.equals(array); // Could return true, but following specification of equals this should always return false (different message to prevent misleading ?)
-&gt; 'contract of 'equals' requires false as return value when the passed argument has type 'array''</pre>
</div>
</div>
<div class="paragraph">
<p>EC: equals(&#8230;&#8203;) used to compare incompatible arrays (EC_INCOMPATIBLE_ARRAY_COMPARE)</p>
</div>
<div class="listingblock">
<div class="content">
<pre>Object[] objects = ...;
String[] strings = ...;
objects.equals(strings); // Always false: message should probably suggest usage of java.util.Arrays.equals(Object[], Object[]) to compare contents</pre>
</div>
</div>
<div class="paragraph">
<p>EC: Call to equals(null) (EC_NULL_ARG)</p>
</div>
<div class="listingblock">
<div class="content">
<pre>Object object = ...;
object.equals(null); // could return true, but following specification of equals this should always return false
-&gt; 'contract of 'equals' requires false as return value when null is passed as argument</pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_on_22_apr_2015_164223_samuel_mercier_wrote">on 22 Apr 2015, 16:42:23 Samuel Mercier wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] As stated by Nico it is easy to build examples which generate false positives.</p>
</div>
<div class="paragraph">
<p>For this reason I think EC_UNRELATED_CLASS_AND_INTERFACE, EC_UNRELATED_INTERFACES and EC_UNRELATED_TYPES should not be covered by this RSPEC, because the message and description would be misleading.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_23_apr_2015_153906_ann_campbell_wrote">on 23 Apr 2015, 15:39:06 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>Per our discussion, I&#8217;ve left EC_UNRELATED_CLASS_AND_INTERFACE, EC_UNRELATED_INTERFACES and EC_UNRELATED_TYPES in this RSpec since, as you pointed out [~samuel.mercier] EC_ARRAY_AND_NONARRAY, EC_INCOMPATIBLE_ARRAY_COMPARE are more specific examples of the general cases - i.e. it would eventually be confusing for users to have this as 2 separate rules.</p>
</div>
<div class="paragraph">
<p>Note that the FB description indicates that the FB check includes an examination of the relevant <code>equals</code> methods and limits issue raising to when the comparison will always return false or not be symmetric.</p>
</div>
</div>
</div>
</div>