Portability issues may restrict which characters should be used in an identifier. 

This rule checks identifier names against a regular expression of disallowed characters. Due to a technical limitation, the COBOL analyzer is not able for the time-being to differentiate lowercase from uppercase characters.


== Noncompliant Code Example

With the default regular expression `+[^a-zA-Z0-9-]+`:

----
MOVE DATA-1 TO DATA_2 *> Noncompliant; '_' not allowed
----


== Compliant Solution

----
MOVE DATA-1 TO DATA-2 
----

