<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>There is no reason to add a const qualifier to the return type of a function that returns by <strong>value</strong>.</p>
</div>
<div class="paragraph">
<p>At best, it will be superfluous. At worst, it will be a performance bug: it can prevent move operation (when copy elision doesnâ€™t take place). When an object is const-qualified the copy constructor/assignment will be a better match than the move constructor/assignment.</p>
</div>
<div class="paragraph">
<p>One might think about adding this qualifier in order to forbid the call of unintended functions on the returned object. A common example is to avoid unintended assignments:</p>
</div>
<div class="listingblock">
<div class="content">
<pre>X x1, x2, x3;
if (x1 + x2 = x3) { // Compiler will complain since const object cannot be assigned. Should be "x1 + x2 == x3"
...
}</pre>
</div>
</div>
<div class="paragraph">
<p>C&#43;&#43;11 introduced reference qualifiers for member functions. This feature provides a better approach to forbid calling unintended functions:</p>
</div>
<div class="listingblock">
<div class="content">
<pre>struct X {
X&amp; operator=(const X&amp; other) &amp;;
};
...
X x1, x2, x3;
if (x1 + x2 = x3) { // Compiler will complain since assignment cannot be called on r-value.  Should be "x1 + x2 == x3"
...
}</pre>
</div>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">class A {...};
const A f();  // Noncompliant</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">class A {...};
A f();</code></pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p>C&#43;&#43; Core Guidelines - <a href="https://github.com/isocpp/CppCoreGuidelines/blob/e49158a/CppCoreGuidelines.md#f20-for-out-output-values-prefer-return-values-to-output-parameters">F.20: For "out" output values, prefer return values to output parameters</a></p>
</li>
</ul>
</div>
</div>
</div>