== Why is this an issue?

In the `pubspec.yaml` file there's a possibility to specify urls in various attributes: homepage, repository, issue_tracker, dependency repositories, etc. It is important to make sure that secure protocol is used in these urls. The `https` provide an additional layer of security by encrypting the transferred data and checking the TLS/SSL certificates. It is generally recommended to avoid using `http` or `git` protocols.

== How to fix it
Replace unsecure `http` or `git` protocols with `https`.

=== Code examples

==== Noncompliant code example

[source,yaml,diff-id=1,diff-type=noncompliant]
----
repository: http://github.com/flutter/flutter
----

==== Compliant solution

[source,yaml,diff-id=1,diff-type=compliant]
----
repository: https://github.com/flutter/flutter
----

==== Noncompliant code example

[source,yaml,diff-id=2,diff-type=noncompliant]
----
git:
  url: git://github.com/flutter/flutter.git
----

==== Compliant solution

[source,yaml,diff-id=2,diff-type=compliant]
----
git:
  url: https://github.com/flutter/flutter.git
----

== Resources
=== Documentation

* Dart Docs - https://dart.dev/tools/linter-rules/secure_pubspec_urls[Dart Linter rule - secure_pubspec_urls]
* Dart Docs - https://dart.dev/tools/pub/pubspec[pubspec]
* Wikipedia - https://en.wikipedia.org/wiki/HTTPS[HTTPS]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

* The (<http>|<git>) protocol shouldn't be used because it isn't secure.

=== Highlighting

The whole url.

'''
== Comments And Links
(visible only on this page)

endif::env-github,rspecator-view[]
