== Why is this an issue?

Transport Layer Security (TLS) provides a secure channel between systems communicating over the internet by encrypting the data sent between them. TLS can be used with `http` and `git` URI schemes to build secure `https` and `git+https` channels respectively. 

These secure protocols are essential for the confidentiality and the integrity of the communication performed when supply chain operations are performed, such as `dart pub get`, `flutter pub get`, `dart pub upgrade`, etc.

In the https://dart.dev/tools/pub/pubspec[`pubspec.yaml` file], which is a file used by several Dart/Flutter commands, there's a possibility to specify URIs in various attributes: homepage, repository, issue_tracker, dependency repositories, etc. 

It is important that secure protocols are used in all these URIs. The `https` protocol provides an additional layer of security by encrypting the transferred data and verifying the server's identity.

It is generally recommended not to use `http` or `git` URIs that will be used by Dart/Flutter commands to gather external data during any stage of the application build.

include::../common/impact.adoc[]

== How to fix it

Replace insecure `http` or `git` URI schemas with `https`.

=== Code examples

==== Noncompliant code example

[source,yaml,diff-id=1,diff-type=noncompliant]
----
repository: http://github.com/flutter/flutter
----

==== Compliant solution

[source,yaml,diff-id=1,diff-type=compliant]
----
repository: https://github.com/flutter/flutter
----

==== Noncompliant code example

[source,yaml,diff-id=2,diff-type=noncompliant]
----
git:
  url: git://github.com/flutter/flutter.git
----

==== Compliant solution

[source,yaml,diff-id=2,diff-type=compliant]
----
git:
  url: https://github.com/flutter/flutter.git
----

== Resources

=== Documentation

* Dart Docs - https://dart.dev/tools/linter-rules/secure_pubspec_urls[Dart Linter rule - secure_pubspec_urls]
* Dart Docs - https://dart.dev/tools/pub/pubspec[pubspec]
* Wikipedia - https://en.wikipedia.org/wiki/HTTPS[HTTPS]

include::../common/standards.adoc[]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

* The (<http>|<git>) protocol shouldn't be used because it isn't secure.

=== Highlighting

The whole url.

'''
== Comments And Links
(visible only on this page)

endif::env-github,rspecator-view[]
