<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>The use of DOM attributes that are only supported by specific browsers can doom your website to obsolescence as the browser market shifts or the vendor that introduced the extension abandons it. Instead, it&#8217;s always best to stick to standard attributes so that your site works for as broad a cross section of users as possible.</p>
</div>
<div class="paragraph">
<p>This rule raises an issue when <code>innerText</code> is used.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-javascript" data-lang="javascript">function setData(element) {
  element.innerText = "Hello World!"; // Noncompliant
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-javascript" data-lang="javascript">function setData(element) {
  element.textContent = "Hello World!";
}</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove this use of "innerText".</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_7_apr_2015_080643_linda_martin_wrote">on 7 Apr 2015, 08:06:43 Linda Martin wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] Is the <code>inner.Text</code> - with the "." - in the description a typo ? Or does it refers to something else ? Just want to be sure I&#8217;m not removing meaning that I&#8217;m not aware of.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_9_apr_2015_120042_ann_campbell_wrote">on 9 Apr 2015, 12:00:42 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~linda.martin] I would describe that as less a "typo" and more a "brain fart" :-)</p>
</div>
<div class="paragraph">
<p>Fixed.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_19_may_2015_155337_linda_martin_wrote">on 19 May 2015, 15:53:37 Linda Martin wrote:</h3>
<div class="paragraph">
<p>Haha! Reviewe.</p>
</div>
</div>
</div>
</div>