<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when <code>redirect_to :back</code> is used without proper error handling or fallback mechanisms.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Using <code>redirect_to :back</code> without proper fallback handling can cause your Rails application to crash with an <code>ActionController::RedirectBackError</code> exception.</p>
</div>
<div class="paragraph">
<p>This error occurs when the HTTP_REFERER header is not present in the request. The HTTP_REFERER header contains the URL of the page that linked to the current page, but it&#8217;s not always available:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>Browsers may not send the referrer header for privacy reasons</p>
</li>
<li>
<p>Users might access the page directly through bookmarks or by typing the URL</p>
</li>
<li>
<p>Automated tests often don&#8217;t set the HTTP_REFERER header</p>
</li>
<li>
<p>Some security tools and browser extensions block referrer headers</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>When <code>redirect_to :back</code> is called without a referrer, Rails throws an exception instead of gracefully handling the situation. This creates a poor user experience and can make your application appear broken.</p>
</div>
<div class="paragraph">
<p>Rails 5 introduced the <code>redirect_back</code> method specifically to address this issue by requiring developers to specify a fallback location.</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="paragraph">
<p>When the HTTP_REFERER header is missing, the application will crash with an <code>ActionController::RedirectBackError</code> exception. This results in a 500 Internal Server Error being displayed to users instead of a graceful redirect, creating a poor user experience and potentially exposing error details in production environments.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Create a helper method that checks for the presence of a referrer before attempting to redirect back.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-ruby" data-lang="ruby">def cancel_action
  redirect_to :back # Noncompliant
end</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-ruby" data-lang="ruby">def cancel_action
  redirect_to_back_or_default(root_path)
end

private

def redirect_to_back_or_default(default)
  if request.referer.present?
    redirect_to :back
  else
    redirect_to default
  end
end</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it_in_rails_5">How to fix it in Rails 5+</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Use <code>redirect_back</code> with a <code>fallback_location</code> parameter. This method automatically handles missing referrer headers by redirecting to the specified fallback location.</p>
</div>
<div class="sect2">
<h3 id="_code_examples_2">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example_2">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-ruby" data-lang="ruby">def update
  if @user.update(user_params)
    redirect_to :back # Noncompliant
  else
    render :edit
  end
end</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution_2">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-ruby" data-lang="ruby">def update
  if @user.update(user_params)
    redirect_back fallback_location: root_path
  else
    render :edit
  end
end</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it_in_rails_5_2">How to fix it in Rails &lt; 5</h2>
<div class="sectionbody">
<div class="paragraph">
<p>For older Rails versions, wrap <code>redirect_to :back</code> in a rescue block to handle the exception and provide a fallback redirect.</p>
</div>
<div class="sect2">
<h3 id="_code_examples_3">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example_3">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-ruby" data-lang="ruby">def destroy
  @post.destroy
  redirect_to :back # Noncompliant
end</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution_3">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-ruby" data-lang="ruby">def destroy
  @post.destroy
  redirect_to :back
rescue ActionController::RedirectBackError
  redirect_to posts_path
end</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Rails redirect_back documentation - <a href="https://api.rubyonrails.org/classes/ActionController/Redirecting.html#method-i-redirect_back">Official Rails documentation for the redirect_back method</a></p>
</li>
<li>
<p>ActionController::Redirecting - <a href="https://api.rubyonrails.org/classes/ActionController/Redirecting.html">Complete documentation for Rails redirecting methods</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>