include::../description.adoc[]

== Noncompliant Code Example

----
int _A;
----

== Compliant Solution

----
int a;
----

ifdef::env-github,rspecator-view[]
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
