== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,cpp]
----
int _A;
----

=== Compliant solution

[source,cpp]
----
int a;
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

=== Parameters

.format
****

----
[a-z][a-z_]*
----

Regular expression used to check the names against.
****


'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
