== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,java]
----
String password = Base64.decode(retrievePassword());
DriverManager.getConnection(url, usr, password);  // Noncompliant
----

=== Compliant solution

[source,java]
----
Cipher cipher = Cipher.getInstance("RSA/ECB/OAEPWithMD5AndMGF1Padding");
cipher.init(Cipher.DECRYPT_MODE, privateKey);  // key management out of scope for this example
cipher.doFinal(retrievePassword()));
----

include::../see.adoc[]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
