Appending '$' to any identifier forces it to the String data type, but this usage is obsolete and should only appear when differentiating string-specific functions from variant functions, such as ``++Left$()++`` versus ``++Left()++``.


Instead, Strings should be explicitly declared with the String datatype.


== Noncompliant Code Example

----
Dim Text$ As String
----


== Compliant Solution

----
Dim Text As String
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

endif::env-github,rspecator-view[]
