== Why is this an issue?

Appending '$' to any identifier forces it to the String data type, but this usage is obsolete and should only appear when differentiating string-specific functions from variant functions, such as ``++Left$()++`` versus ``++Left()++``.


Instead, Strings should be explicitly declared with the String datatype.


=== Noncompliant code example

[source,vb6]
----
Dim Text$ As String
----


=== Compliant solution

[source,vb6]
----
Dim Text As String
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Remove the '$' from "XXX" and make sure it is defined with the String datatype


endif::env-github,rspecator-view[]
