include::../description.adoc[]

== Noncompliant Code Example

With default provided regular expression: ``++^([A-Z0-9_]*|[a-z0-9_]*)$++``

[source,abap]
----
DEFINE MyMacro.
  ...
END-OF-DEFINITION.
----

== Compliant Solution

[source,abap]
----
DEFINE my_macro.
  ...
END-OF-DEFINITION.
----
ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../parameters.adoc[]

endif::env-github,rspecator-view[]
