== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

With default provided regular expression: ``++^([A-Z0-9_]*|[a-z0-9_]*)$++``

[source,abap]
----
DEFINE MyMacro.
  ...
END-OF-DEFINITION.
----

=== Compliant solution

[source,abap]
----
DEFINE my_macro.
  ...
END-OF-DEFINITION.
----
ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

=== Parameters

.format
****

----
^([A-Z0-9_]*|[a-z0-9_]*)$
----

Regular expression used to check the macro names against.
****


endif::env-github,rspecator-view[]
