== Why is this an issue?

``++CREATE_TIMER++`` is generating network traffic each time the timer is fired. It's probably totally fine for a timer being executed every hour but generally, this is used to provide clocks components that are going to generate network traffic every second or more.

It is recommended by Oracle to examine timers and replace them with JavaBeans.


=== Noncompliant code example

[source,sql]
----
BEGIN
  timer := CREATE_TIMER('foo', 1000, REPEAR)
ENDl
----


== Resources

* https://docs.oracle.com/cd/A97335_02/apps.102/a86202/chap11.htm[11.3.4 Other Techniques to Improve Performance]


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Replace "CREATE_TIMER" by a JavaBean timer executed on Oracle Forms client side.


endif::env-github,rspecator-view[]
