Positional parameters must be specified before any keyword parameters.

== Why is this an issue?

JCL requires positional parameters to be specified before any keyword parameters. Having keyword parameters defined before a positional parameter will trigger a JCL error.

== How to fix it

=== Code examples

==== Noncompliant code example

[source,jcl,diff-id=1,diff-type=noncompliant]
----
//MYJOB JOB CLASS=X,'TRIGGER'
----

==== Compliant solution

[source,text,diff-id=1,diff-type=compliant]
----
//MYJOB JOB 'TRIGGER',CLASS=X
----

== Resources

=== Documentation

* https://www.ibm.com/docs/en/zos/3.1.0?topic=messages-iefc006i[IBM Reference - IEFC006I]
* https://www.ibm.com/docs/en/zos/3.1.0?topic=fields-parameter-field[IBM Reference - Parameter field]
