include::../description.adoc[]

== Noncompliant Code Example

An arbitrary command can be executed:

----
$cmd = $_GET["cmd"];

// If the command "/sbin/shutdown" is executed and the web server is running as root,
// then the machine running the web server will be shut down and become unavailable for future requests

exec($cmd); // Noncompliant
----

== Compliant Solution


Restrict the command to be executed with an allowlist:

----
$cmd = $_GET["cmd"];

if ($cmd === "/usr/bin/stat" || $cmd === "/usr/bin/cat") {
   exec($cmd." /tmp/test.txt", $output);
}
----

include::../see.adoc[]

ifdef::env-github,rspecator-view[]
'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
