FIXME: add a description

// If you want to factorize the description uncomment the following line and create the file.
//include::../description.adoc[]

== Noncompliant Code Example

----
FIXME
----

== Compliant Solution

----
FIXME
----

== See

FIXME: A list of links

ifdef::env-github,rspecator-view[]

'''

== Hello, this header is hidden in the end

but displayed here (feel special!)
endif::env-github,rspecator-view[]
