== Why is this an issue?

Some types are not very well suited for use in a bit-field, because their behavior is implementation-defined. When defining a bit-field, you should stick to the following safe and portable types:


* In C: ``++signed short++``, ``++unsigned short++``, ``++signed char++``, ``++unsigned char++``, ``++signed int++``, ``++unsigned int++`` or ``++_Bool++``
* In {cpp} before {cpp}14: all enumerated types, as well as ``++signed short++``, ``++unsigned short++``, ``++signed char++``, ``++unsigned char++``, ``++signed int++``, ``++unsigned int++``, ``++signed long++``, ``++unsigned long++``, ``++signed long long++``, ``++unsigned long long++````++ or bool++``
* In {cpp} starting at {cpp}14:  all enumerated and integral types


=== Noncompliant code example

[source,cpp]
----
// Assuming we are in C
int b:3; // Noncompliant - may have the range of values 0..7 or -4..3
----


=== Compliant solution

[source,cpp]
----
unsigned int b:3;
----


== Resources

* MISRA C:2004, 6.4 - Bit fields shall only be defined to be of type _unsigned int_ or _signed int_.
* MISRA {cpp}:2008, 9-6-2 - Bit-fields shall be either _bool_ type or an explicitly _unsigned_ or _signed_ integral type.
* MISRA C:2012, 6.1 - Bit-fields shall only be declared with an appropriate type
* https://wiki.sei.cmu.edu/confluence/x/VNYxBQ[CERT, INT12-C.] - Do not make assumptions about the type of a plain int bit-field when used in an expression


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Change type of bit field.


'''
== Comments And Links
(visible only on this page)

=== is duplicated by: S1008

=== on 28 Sep 2014, 21:00:35 Evgeny Mandrikov wrote:
What about MISRA {cpp}:2008 9-6-2?

=== on 3 Oct 2014, 19:11:05 Ann Campbell wrote:
\[~evgeny.mandrikov] 


First, thanks for finding the {cpp} analog. I'm guessing it was the difference between "Bit fields" +(C)+ and "Bit-fields" ({cpp}) which kept me from finding it when I looked.


Second, MISRA {cpp}:2008, 9-6-2 broadens the scope to include ``++bool++`` as acceptable. Since C has no ``++bool++`` type, how would you like to proceed? Define an exception?

=== on 3 Oct 2014, 19:14:53 Evgeny Mandrikov wrote:
\[~ann.campbell.2] C11 has _Bool type.

=== on 8 Dec 2014, 22:45:11 Evgeny Mandrikov wrote:
Note to myself - see CERT INT12-C.

=== on 9 Dec 2014, 21:47:03 Evgeny Mandrikov wrote:
Note to myself:


in {cpp}11 6.9p3:

____
It is implementation-defined whether a plain (neither explicitly signed nor unsigned) char, short, int, long, or long long bit-field is signed or unsigned.

____
whereas in working draft N4140 of {cpp}14 there is no such statement and in section C.1.8:

____
Change++:++ Bit-fields of type plain int are signed.

Rationale++:++ Leaving the choice of signedness to implementations could lead to inconsistent definitions of template specializations. For consistency, the implementation freedom was eliminated for non-dependent types, too.

Effect on original feature: The choise is implementation-defined in C, but not so in {cpp}.

Difficulty of converting: Syntactic transformation.

How widely used: Seldom.

____

=== on 16 Dec 2014, 09:49:21 Evgeny Mandrikov wrote:
Note that description was tuned for C (CPP-638), but must be tuned for {cpp} at a time of implementation.

endif::env-github,rspecator-view[]
