<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This is an issue when using the <code>junit</code> step in Jenkins pipelines without specifying the <code>testResults</code> parameter.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>The <code>junit</code> step in Jenkins pipelines is used to publish test results and provide visual feedback about test outcomes. When you use this step without specifying the <code>testResults</code> parameter, Jenkins cannot locate the test result files.</p>
</div>
<div class="paragraph">
<p>The <code>testResults</code> parameter tells Jenkins where to find the XML test report files using file patterns (like <code>**/test-results/*.xml</code>). Without this parameter, the step will fail to collect any test results, even if your tests ran successfully.</p>
</div>
<div class="paragraph">
<p>This leads to several problems:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>Jenkins cannot display test trends and historical data</p>
</li>
<li>
<p>Test failures won&#8217;t be properly reported in the build status</p>
</li>
<li>
<p>GitHub integration and other SCM platforms won&#8217;t receive test result notifications</p>
</li>
<li>
<p>Developers lose visibility into which specific tests are failing</p>
</li>
<li>
<p>Build health scoring based on test results won&#8217;t work correctly</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>The Jenkins documentation consistently shows examples where <code>testResults</code> is provided alongside other configuration options, indicating this is essential for proper functionality.</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="paragraph">
<p>Without the <code>testResults</code> parameter, your Jenkins pipeline will lose test result visibility. This means you won&#8217;t see test trends, failure details, or proper build status reporting. Your team will miss critical feedback about code quality and test coverage.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Add the <code>testResults</code> parameter to specify where Jenkins should look for test result XML files. Use file patterns that match your test output location.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-groovy" data-lang="groovy">junit skipMarkingBuildUnstable: true // Noncompliant</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-groovy" data-lang="groovy">junit skipMarkingBuildUnstable: true, testResults: 'test-results.xml'</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Jenkins JUnit Plugin Documentation - <a href="https://plugins.jenkins.io/junit/">Official documentation for the Jenkins JUnit plugin with configuration examples</a></p>
</li>
<li>
<p>Jenkins Pipeline Steps Reference - JUnit - <a href="https://www.jenkins.io/doc/pipeline/steps/junit/">Pipeline step reference for the junit step with parameter details</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>