== Why is this an issue?

Every function should be commented to explain its goal and how it works. This non-empty comment must be located before the function definition.


=== Noncompliant code example

[source,abap]
----
FUNCTION my_function.
  ...
ENDFUNCTION. 
----

=== Compliant solution

[source,abap]
----
* here is  
* my comment
FUNCTION my_function.
  ...
ENDFUNCTION. 
----
ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Document this function 'XXX'.


endif::env-github,rspecator-view[]
