include::../description.adoc[]

== Noncompliant Code Example

[source,abap]
----
FUNCTION my_function.
  ...
ENDFUNCTION. 
----

== Compliant Solution

[source,abap]
----
* here is  
* my comment
FUNCTION my_function.
  ...
ENDFUNCTION. 
----
ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

endif::env-github,rspecator-view[]
