== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,abap]
----
FUNCTION my_function.
  ...
ENDFUNCTION. 
----

=== Compliant solution

[source,abap]
----
* here is  
* my comment
FUNCTION my_function.
  ...
ENDFUNCTION. 
----
ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

endif::env-github,rspecator-view[]
