This rule raises an issue when an object is created but not assigned to a variable, and later code attempts to reference an undefined variable that would logically hold that object.

== Why is this an issue?

When you create an object in Ruby using constructor methods like `new` or `create`, the object exists in memory but disappears immediately if not assigned to a variable. If you later try to reference a variable that you assume holds this object, Ruby will raise a `NameError: undefined local variable` exception.

This commonly happens when developers are learning Ruby or Rails and forget the assignment step. For example, calling `User.new(name: "John")` creates a User object, but without assignment (`user = User.new(name: "John")`), the object cannot be accessed later.

The Ruby interpreter has no way to guess that when you reference `user` later, you meant the User object you created earlier. Each variable must be explicitly defined before use.

=== What is the potential impact?

This issue will cause your application to crash at runtime with a `NameError` exception. The error will occur every time the code path is executed, making the affected functionality completely unusable until fixed.

== How to fix it

Assign the created object to a variable so it can be referenced later.

=== Code examples

==== Noncompliant code example

[source,ruby,diff-id=1,diff-type=noncompliant]
----
# Object created but not assigned
User.new(name: "John", email: "john@example.com")
# Later attempt to use undefined variable
user.save  # Noncompliant
----

==== Compliant solution

[source,ruby,diff-id=1,diff-type=compliant]
----
# Properly assign object to variable
user = User.new(name: "John", email: "john@example.com")
user.save
----

== Resources

=== Documentation

 * Ruby Variables - https://ruby-doc.org/core/doc/syntax/assignment_rdoc.html[Official Ruby documentation on variable assignment]

 * Ruby NameError - https://ruby-doc.org/core/NameError.html[Documentation for NameError exceptions in Ruby]

=== Related rules

 * S3827 - https://rules.sonarsource.com/javascript/RSPEC-3827[Similar rule for JavaScript detecting unused object creation]
