This rule raises an issue when a generator is passed to `np.array`.

== Why is this an issue?

The creation of a NumPy array can be done in several ways, for example by passing a Python list to the `np.array` function.
Another way would be to pass a generator to the `np.array` function, doing so create a 0-dimensional array of objects and may not be the intended goal.
This NumPy array will have a have a data type (dtype) of `object` and could hold any Python objects. 

One of the characteristic of NumPy arrays is to be homogeneous, meaning all its elements are of the same type.
Creating an array of objects allows the user to create heterogeneous array without raising any errors.
Creating such an array can lead to bugs further in the program.

[source,python,diff-id=1,diff-type=noncompliant]
----
arr = np.array(x**2 for x in range(10))

arr.reshape(1)
arr.resize(2)
arr.put(indices=1, values=3) # No issues raised.

----

The NumPy array `arr` shown above now holds 2 values: a generator and the number 3.


== How to fix it

To fix this issue pass a Python list instead of a generator to the `np.array` function or 
explicitly type hint the generator with `Any`, showing the intention to create a 0-dimensional array of objects.

=== Code examples

==== Noncompliant code example

[source,python,diff-id=2,diff-type=noncompliant]
----
arr = np.array(x**2 for x in range(10)) # Noncompliant: the resulting array will be of the data type: object.

gen = (x*2 for x in range(5))
arr = np.array(gen) # Noncompliant: the resulting array will be of the data type: object.
----

==== Compliant solution

[source,python,diff-id=2,diff-type=compliant]
----
from typing import Any

arr = np.array([x**2 for x in range(10)]) # Compliant

gen: Any = (x*2 for x in range(5))
arr = np.array(gen) # Compliant
----

//=== How does this work?

//=== Pitfalls

//=== Going the extra mile


== Resources

=== Documentation

NumPy Documentation - https://numpy.org/doc/stable/reference/typing.html#arraylike[ArrayLike]

//=== Articles & blog posts
//=== Conference presentations
//=== Standards
//=== External coding guidelines
//=== Benchmarks
