include::../description.adoc[]

== Noncompliant Code Example

----
foo#: proc options(main); /* Non-Compliant */
end;
----

== Compliant Solution

----
foo: proc options(main);  /* Compliant */
end;
----

ifdef::env-github,rspecator-view[]
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
