== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,pli]
----
foo#: proc options(main); /* Non-Compliant */
end;
----

=== Compliant solution

[source,pli]
----
foo: proc options(main);  /* Compliant */
end;
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

=== Parameters

.regularExpression
****

----
[a-zA-Z0-9_]++
----

Regular expression used to check the function names against
****


'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
