include::../description.adoc[]

== Noncompliant Code Example

[source,pli]
----
foo#: proc options(main); /* Non-Compliant */
end;
----

== Compliant Solution

[source,pli]
----
foo: proc options(main);  /* Compliant */
end;
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::parameters.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
