== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,sql]
----
SELECT IIF(@status = 'RUNNING', 'Running', IIF(@hasError, 'Failed', 'Succeeded')) -- Noncompliant
----


=== Compliant solution

[source,sql]
----
IF @status = 'RUNNING'
    SELECT 'Running';
ELSE
    SELECT IIF(@hasError, 'Failed', 'Succeeded');
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../highlighting.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
