<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Fields marked with <code>System.Runtime.Serialization.OptionalFieldAttribute</code> are serialized just like any other field. But such fields are ignored on deserialization, and retain the default values associated with their types. Therefore, deserialization event handlers should be declared to set such fields during the deserialization process.</p>
</div>
<div class="paragraph">
<p>This rule raises when at least one field with the <code>System.Runtime.Serialization.OptionalFieldAttribute</code> attribute is declared but one (or both) of the following event handlers <code>System.Runtime.Serialization.OnDeserializingAttribute</code> or <code>System.Runtime.Serialization.OnDeserializedAttribute</code> are not present.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-vbnet" data-lang="vbnet">&lt;Serializable&gt;
Public Class Foo ' Noncompliant
    &lt;OptionalField(VersionAdded:=2)&gt;
    Private optionalField As Integer = 5
End Class</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-vbnet" data-lang="vbnet">&lt;Serializable&gt;
Public Class Foo
    &lt;OptionalField(VersionAdded:=2)&gt;
    Private optionalField As Integer = 5

    &lt;OnDeserializing&gt;
    Private Sub OnDeserializing(ByVal context As StreamingContext)
        optionalField = 5
    End Sub

    &lt;OnDeserialized&gt;
    Private Sub OnDeserialized(ByVal context As StreamingContext)
    End Sub
End Class</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="ulist">
<ul>
<li>
<p>Add the missing "(OnDeserializingAttribute|OnDeserializedAttribute)" event handler.</p>
</li>
<li>
<p>Add deserialization event handlers.</p>
</li>
</ul>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>type name</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_16_mar_2017_131330_ann_campbell_wrote">on 16 Mar 2017, 13:13:30 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~amaury.leve] I&#8217;ve done a rather heavy edit. Please double-check me. In particular, I&#8217;ve updated <em>where</em> (and therefore <em>how many</em>) issues should be raised. I&#8217;ve also tried to clarify some language that was initially unclear to me. Please verify that the description still says what it should.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_16_mar_2017_134758_amaury_levé_wrote">on 16 Mar 2017, 13:47:58 Amaury Levé wrote:</h3>
<div class="paragraph">
<p>I am not a big fan of reporting on the <code>OptionalField</code> field because if the class has multiples fields with this attribute we should still report only once. The event handler method is going to take care of all parameters at once.</p>
</div>
<div class="paragraph">
<p>I would get back to reporting on the class itself. WDYT?</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_16_mar_2017_135448_ann_campbell_wrote">on 16 Mar 2017, 13:54:48 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>Done [~amaury.leve]</p>
</div>
</div>
</div>
</div>