== Why is this an issue?

Finalizers come with a performance cost due to the overhead of tracking the life cycle of objects. An empty one is consequently costly with no benefit or justification.


=== Noncompliant code example

[source,csharp]
----
public class Foo 
{
    ~Foo() // Noncompliant
    { 
    }
}
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Remove this empty finalyzer.


=== Highlighting

Finalizer declaration


endif::env-github,rspecator-view[]
