<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Declaring multiple variables on one line is difficult to read.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-swift" data-lang="swift">var i = 1, j = 2</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-swift" data-lang="swift">var i = 1
var j = 2</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Declare "XXX" on a separate line</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_9_may_2016_154807_evgeny_mandrikov_wrote">on 9 May 2016, 15:48:07 Evgeny Mandrikov wrote:</h3>
<div class="paragraph">
<p>C/C&#43;&#43; already implements <a data-rspec-id="S1004" class="rspec-auto-link">RSPEC-1004</a>.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_16_may_2016_211240_ann_campbell_wrote">on 16 May 2016, 21:12:40 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~evgeny.mandrikov] since it does so with a legacy key, what do you think of moving the key to this RSpec &amp; maybe making <a data-rspec-id="S1004" class="rspec-auto-link">RSPEC-1004</a> a subtask?</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_4_jul_2016_201028_evgeny_mandrikov_wrote">on 4 Jul 2016, 20:10:28 Evgeny Mandrikov wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] sounds good</p>
</div>
</div>
</div>
</div>