== Why is this an issue?

include::../description.adoc[]

== How to fix it

include::../howtofix.adoc[]

=== Code examples

==== Noncompliant code example

[source,bash,diff-id=1,diff-type=noncompliant]
----
prepare "creating customer" # Noncompliant
execute "creating customer" # Noncompliant
release "creating customer" # Noncompliant
----

==== Compliant solution

[source,bash,diff-id=1,diff-type=compliant]
----
readonly message="creating customer"
prepare "$message"
execute "$message"
release "$message"
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../parameters.adoc[]

include::../highlighting.adoc[]

'''

endif::env-github,rspecator-view[]
