<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>As soon as the <code>yield</code> keyword is used the enclosing method or function becomes a generator. Thus <code>yield</code> should never be used in a function or method which is not intended to be a generator.</p>
</div>
<div class="paragraph">
<p>This rule raises an issue when <code>yield from</code> or <code>yield</code> are used in a function or method which is not a generator because:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>the function/method&#8217;s return type annotation is not <a href="https://docs.python.org/3/library/typing.html#typing.Generator"><code>typing.Generator[...]</code></a></p>
</li>
<li>
<p>it is a special method which can never be a generator (ex: <code>__init__</code>).</p>
</li>
</ul>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">class A:
    def __init__(self, value):
        self.value = value
        yield value  # Noncompliant

def mylist2() -&gt; List[str]:
    yield ['string']  # Noncompliant. Return should be used instead of yield

def generator_ok() -&gt; Generator[int, float, str]:
    sent = yield 42
    return '42'</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">class A:
    def __init__(self, value):
        self.value = value

def mylist2() -&gt; List[str]:
    return ['string']

def generator_ok() -&gt; Generator[int, float, str]:
    sent = yield 42
    return '42'</code></pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p><a href="https://docs.python.org/3/library/typing.html">Python documentation - Support for type hints</a></p>
</li>
<li>
<p><a href="https://docs.python.org/3/reference/datamodel.html#special-method-names">Python documentation - Special method names</a></p>
</li>
</ul>
</div>
<hr>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="ulist">
<ul>
<li>
<p>Remove this "yield" statement</p>
</li>
<li>
<p>Replace this "yield" keyword with "return" or change the return type annotation.</p>
</li>
</ul>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>Primary: the "yield" keyword</p>
</div>
<div class="paragraph">
<p>Secondary: function/method&#8217;s return type annotation if there is one</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_is_related_to_s2734">is related to: <a data-rspec-id="S2734" class="rspec-auto-link">S2734</a></h3>

</div>
</div>
</div>