== Why is this an issue?

This rule allows banning the usage of certain data set names.

=== Code examples

==== Noncompliant code example

Given the parameters:

* programs: SORT,ICEMAN
* ddNameField: SORTOUT
* allowedValues: AB\.[^.]*\.DEF

[source,jcl,diff-id=1,diff-type=noncompliant]
----
//EXEC PGM=SORT
//SORTOUT DD DSN=AB.FOOBAR.XYZ
----

==== Compliant solution

[source,jcl,diff-id=1,diff-type=compliant]
----
//EXEC PGM=SORT
//SORTOUT DD DSN=AB.FOOBAR.DEF
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Parameters

.programs
****

Comma-separated list of programs where the data set names should be checked.
This parameter is optional. If absent, all DD statements regardless of procedure steps are checked.
****

.ddNameField
****

Name field of the DD statement where the data set name should be checked.
****

.allowedValues
****

Regular expression to match valid data set names.
****

'''

endif::env-github,rspecator-view[]