== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,csharp]
----
enum FooFlags // Noncompliant
{
    Foo = 1
    Bar = 2
    Baz = 4
}
----

=== Compliant solution

[source,csharp]
----
enum Foo
{
    Foo = 1
    Bar = 2
    Baz = 4
}
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
