include::../description.adoc[]

== Noncompliant Code Example

----
enum FooFlags // Noncompliant
{
    Foo = 1
    Bar = 2
    Baz = 4
}
----

== Compliant Solution

----
enum Foo
{
    Foo = 1
    Bar = 2
    Baz = 4
}
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
