include::../description.adoc[]

== Noncompliant Code Example

----
enum FooFlags // Noncompliant
{
    Foo = 1
    Bar = 2
    Baz = 4
}
----

== Compliant Solution

----
enum Foo
{
    Foo = 1
    Bar = 2
    Baz = 4
}
----

ifdef::rspecator-view[]
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::rspecator-view[]
