<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>In order to make your site usable by as many people as possible, subtitles should be provided for videos.</p>
</div>
<div class="paragraph">
<p>This rule raises an issue when a <code>video</code> does not include at least one <code>&lt;track/&gt;</code> tag with the <code>kind</code> attribute set to <code>captions</code>, or <code>descriptions</code> or at the very least <code>subtitles</code>.</p>
</div>
<div class="paragraph">
<p>Note that the <code>subtitles</code> kind is not meant for accessibility but for translation. The kind <code>captions</code> targets people with hearing impairment, whereas <code>descriptions</code> targets people with vision impairment.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-html" data-lang="html">&lt;video id="video" controls preload="metadata"&gt;
   &lt;source src="resources/myvideo.mp4" type="video/mp4"&gt;
   &lt;source src="resources/myvideo.webm" type="video/webm"&gt;
&lt;/video&gt;</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-html" data-lang="html">&lt;video id="video" controls preload="metadata"&gt;
   &lt;source src="resources/myvideo.mp4" type="video/mp4"&gt;
   &lt;source src="resources/myvideo.webm" type="video/webm"&gt;
   &lt;track label="English" kind="captions" srclang="en" src="resources/myvideo-en.vtt" default&gt;
   &lt;track label="Deutsch" kind="captions" srclang="de" src="resources/myvideo-de.vtt"&gt;
   &lt;track label="Espa√±ol" kind="captions" srclang="es" src="resources/myvideo-es.vtt"&gt;
&lt;/video&gt;</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Add subtitles and description files for this video.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p><code>video</code> in opening tag</p>
</div>
</div>
</div>
</div>