Since {cpp}03, a member function that is contained within a class definition is by definition ``inline``, so an using an ``inline`` specifier on such functions is redundant.


== Noncompliant Code Example

----
class Foo {
  inline void method() { // Noncompliant
    // ...
  }
};
----


== Compliant Solution

----
class Foo {
  void method() {
    // ...
  }
};
----


