<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">class Base {
  virtual f();
}

class Derived : public Base {
  virtual f(int); // Noncompliant; hides "Base::f"
}</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>Entire method signature</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_6_sep_2016_090712_ann_campbell_wrote">on 6 Sep 2016, 09:07:12 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~evgeny.mandrikov] there is no OOP03-CPP. &#8230;&#8203;?</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_6_sep_2016_092836_evgeny_mandrikov_wrote">on 6 Sep 2016, 09:28:36 Evgeny Mandrikov wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] https://www.securecoding.cert.org/confluence/display/cplusplus/OOP03-CPP.+Prefer+not+to+overload+virtual+functions</p>
</div>
</div>
</div>
</div>