This rule raises an issue when legacy `numpy.random.RandomState` is used.

== Why is this an issue?

Using a predictable seed is a common best practice when using NumPy to create reproducible results. To that end, using `np.random.seed(number)` to set the seed of the global `numpy.random.RandomState` has been the privileged solution for a long time.

`numpy.random.RandomState` and its associated methods rely on a global state, which may be problematic when threads or other forms of concurrency are involved. The global state may be altered and the global seed may be reset at various points in the program (for instance, through an imported package or script), which would lead to irreproducible results.

Instead, the preferred best practice to generate reproducible pseudorandom numbers is to instantiate a `numpy.random.Generator` object with a seed and reuse it in different parts of the code. This avoids the reliance on a global state. Whenever a new seed is needed, a new generator may be created instead of mutating a global state.

== How to fix it

To fix this issue, replace usages of `numpy.random.RandomState` to `numpy.random.Generator`.

=== Code examples

==== Noncompliant code example

[source,python,diff-id=1,diff-type=noncompliant]
----
import numpy as np
def foo():
    np.random.seed(42)
    x = np.random.randn()  # Noncompliant: this relies on numpy.random.RandomState, which is deprecated
----

==== Compliant solution

[source,python,diff-id=1,diff-type=compliant]
----
import numpy as np
def foo():
    generator = np.random.default_rng(42)
    x = generator.standard_normal()
----

== Resources
=== Documentation

* NumPy Documentation - https://numpy.org/doc/stable/reference/random/generator.html#random-generator[Random Generator]
* NumPy Documentation - https://numpy.org/doc/stable/reference/random/legacy.html#legacy-random-generation[Legacy Random Generation]
* NumPy Documentation - https://numpy.org/neps/nep-0019-rng-policy.html[NEP19 RNG Policy]
