This rule raises an issue when ``++@isTest++`` annotations use ``++seeAllData=true++`` at the class or method level.

== Why is this an issue?

Using ``++seeAllData=true++`` in test classes or methods creates dependencies on existing data in the Salesforce org, which violates the fundamental principle of test isolation.

When tests rely on existing org data, they become unreliable and unpredictable. The same test might pass in one environment but fail in another, simply because the required data exists in one org but not the other. Even worse, tests that have been working for months can suddenly start failing when someone deletes or modifies the records they depend on.

Test isolation ensures that each test is self-contained and creates its own mock data. This approach guarantees that tests will behave consistently across different environments - development, staging, and production orgs.

Additionally, there's a technical restriction in Apex: ``++seeAllData=true++`` cannot be used when `@testSetup` methods are present in the same test class. This combination will cause a compilation error, preventing the code from being deployed.

=== What is the potential impact?

Tests become unreliable and may fail unexpectedly when dependent org data is modified or deleted. This can block deployments and create maintenance overhead. Tests may also behave differently across environments, making it difficult to ensure consistent application behavior.

== How to fix it

Remove ``++seeAllData=true++`` from the test method and create the necessary test data within the test method itself.

=== Code examples

==== Noncompliant code example

[source,apex,diff-id=1,diff-type=noncompliant]
----
@isTest(seeAllData=true)
static void testAccountProcessing() {
    Account acc = [SELECT Id, Name FROM Account LIMIT 1]; // Noncompliant
    // test logic using existing data
}
----

==== Compliant solution

[source,apex,diff-id=1,diff-type=compliant]
----
@isTest
static void testAccountProcessing() {
    Account acc = new Account(Name='Test Account');
    insert acc;
    Account insertedAcc = [SELECT Id, Name FROM Account LIMIT 1];
    // test logic using created data
}
----

== Resources

=== Documentation

 * Apex Testing Best Practices - https://developer.salesforce.com/docs/atlas.en-us.apexcode.meta/apexcode/apex_testing_best_practices.htm[Official Salesforce documentation on Apex testing best practices]

 * Test Class Annotation - https://developer.salesforce.com/docs/atlas.en-us.apexcode.meta/apexcode/apex_classes_annotation_isTest.htm[Documentation on the @isTest annotation and its parameters]

=== Standards

 * Salesforce Best Practices - https://developer.salesforce.com/blogs/developer-relations/2015/01/apex-best-practices-15-apex-commandments.html[Industry best practices for Apex development including test isolation principles]
