==== Noncompliant code example

[source,vbnet,diff-id=2,diff-type=noncompliant]
----
If (Date.Now - lastRefresh).TotalMilliseconds > MinRefreshInterval Then
    lastRefresh = Date.Now
    ' Refresh
End If
----

==== Compliant solution

[source,vbnet,diff-id=2,diff-type=compliant]
----
If (Date.UtcNow - lastRefresh).TotalMilliseconds > MinRefreshInterval Then
    lastRefresh = Date.UtcNow
    ' Refresh
End If
----