== Why is this an issue?

include::../firstWhy.adoc[]

[source,vbnet]
----
account1.updatedAt = DateTime.UtcNow ' First call, might be N ticks
account2.updatedAt = DateTime.UtcNow ' Second call, this one could be N + 1 ticks, but also N.
Console.WriteLine($"The accounts have been updated at {DateTime.UtcNow}") ' Third call, this can also be N, N + 1 or N + 2 ticks
----

include::../secondWhy.adoc[]

[source,vbnet]
----
Private start = DateTime.Now ' First call, on March 26th 2:59 am
' Do stuff
Private elapsed = CInt((DateTime.Now - start).TotalMilliseconds) ' Second call happens 2 minutes later but `Now` is March 26th, 4:01 am as there's a shift to summer time
----

== How to fix it

include::../howToFix.adoc[]

=== Code examples

==== Noncompliant code example

[source,vbnet,diff-id=1,diff-type=noncompliant]
----
account1.updatedAt = DateTime.UtcNow ' Noncompliant
account2.updatedAt = DateTime.UtcNow ' Secondary
Console.WriteLine($"The accounts have been updated at {DateTime.UtcNow}") ' Secondary
----

==== Compliant solution

[source,vbnet,diff-id=1,diff-type=compliant]
----
Dim now = DateTime.UtcNow ' Compliant
account1.updatedAt = now
account2.updatedAt = now
Console.WriteLine($"The accounts have been updated at {now}")
----

==== Noncompliant code example

[source,vbnet,diff-id=1,diff-type=noncompliant]
----
If (DateTime.UtcNow - latest).TotalMillisecond > 100 Then ' Noncompliant
    latest = DateTime.UtcNow ' Secondary
End If
----

==== Compliant solution

[source,vbnet,diff-id=1,diff-type=compliant]
----
Dim utcNow = DateTime.UtcNow ' Compliant

If (utcNow - latest).TotalMillisecond > 100 Then ' Compliant
    latest = utcNow
End If
----

or

[source,vbnet,diff-id=1,diff-type=compliant]
----
Dim stopWatch = Stopwatch.StartNew() ' Compliant

If stopWatch.ElapsedMilliseconds > 100 Then
    stopWatch.Restart()
End If
----

==== Noncompliant code example

[source,vbnet,diff-id=1,diff-type=noncompliant]
----
Dim start = DateTime.Now ' Noncompliant
' Method to be benchmarked
Console.WriteLine($"{CInt((DateTime.Now - start).TotalMilliseconds)} ms") ' Secondary
----

==== Compliant solution

[source,vbnet,diff-id=1,diff-type=compliant]
----
Dim stopWatch = Stopwatch.StartNew() ' Compliant
' Method to be benchmarked
stopWatch.Stop()
Console.WriteLine($"{stopWatch.ElapsedMilliseconds} ms")
----

include::../resources.adoc[]