== Why is this an issue?

Calling Security.allowDomain("*") lets any domain cross-script into the domain of this SWF and exercise its functionality.


=== Noncompliant code example

[source,flex]
----
Security.allowDomain("*");
----


=== Compliant solution

[source,flex]
----
Security.allowDomain("www.myDomain.com");
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
