Calling Security.allowDomain("*") lets any domain cross-script into the domain of this SWF and exercise its functionality.


== Noncompliant Code Example

----
Security.allowDomain("*");
----


== Compliant Solution

----
Security.allowDomain("www.myDomain.com");
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
