include::../why.adoc[]

Note that this rule does not take reflection into account, which means that issues will be raised on `private` functions that are only accessed using the reflection API.

=== Exceptions

This rule doesn't raise issues for:

* annotated methods 
* methods with parameters that are annotated with `@javax.enterprise.event.Observes` 

=== Code examples

=== Noncompliant code example

[source,kotlin,diff-id=1,diff-type=noncompliant]
----
class Foo: Serializable {
  private fun unusedMethod() {...}
  private fun writeObject(s: ObjectOutputStream) {...}  // Compliant, relates to the serialization mechanism
  private fun readObject(s: ObjectOutputStream) {...}  // Compliant, relates to the serialization mechanism
}
----

=== Compliant solution

[source,kotlin,diff-id=1,diff-type=compliant]
----
class Foo: Serializable {
  private fun writeObject(s: ObjectOutputStream) {...}  // Compliant, relates to the serialization mechanism
  private fun readObject(s: ObjectOutputStream) {...}  // Compliant, relates to the serialization mechanism
}
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
