<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p><code>NVL</code> was introduced by Oracle Database during the 80&#8217;s. <code>COALESCE</code> is a more modern function part of ANSI-92 standard than can replace <code>NVL</code>. Use of <code>COALESCE</code> should be preferred to <code>NVL</code> for performance reason if the two parameters provided have the same type. COALESCE is running faster than NVL thanks to its short-circuit evaluation of the parameters.</p>
</div>
<div class="paragraph">
<p>In order to avoid "ORA-00932: inconsistent datatypes" error, double-check the two arguments of the NVL function have the same type before switching to COALESCE.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-sql" data-lang="sql">SELECT employee_uuid, NVL(TO_CHAR(bonus_pct), 'not this year') "Bonus" FROM employees</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-sql" data-lang="sql">SELECT employee_uuid, COALESCE(TO_CHAR(bonus_pct), 'not this year') "Bonus" FROM employees</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_3_apr_2018_091244_alexandre_gigleux_wrote">on 3 Apr 2018, 09:12:44 Alexandre Gigleux wrote:</h3>
<div class="paragraph">
<p>It has been decided to not implement this rule because the performance gain to move to COALESCE is not guaranteed.</p>
</div>
<div class="paragraph">
<p>Example: https://jonathanlewis.wordpress.com/2018/02/13/coalesce-v-nvl/</p>
</div>
</div>
</div>
</div>