== Why is this an issue?

Nested control flow statements ``++IF++``, ``++CASE++``, ``++DO++``, ``++LOOP++``, ``++SELECT++``, ``++WHILE++`` and ``++PROVIDE++``  are often key ingredients in creating
what's known as "Spaghetti code". This code smell can make your program difficult to understand and maintain.

When numerous control structures are placed inside one another, the code becomes a tangled, complex web.
This significantly reduces the code's readability and maintainability, and it also complicates the testing process.

== How to fix it

=== Code examples

The following example demonstrates the behavior of the rule with the default threshold of 3 levels of nesting:

==== Noncompliant code example

[source,abap]
----
IF param1 = 2.
  IF param2 = 4.
    DO 3 TIMES.    "Compliant - depth = 3, not exceeding the limit
      IF sy-index = 2.   "Noncompliant - depth = 4
        CONTINUE.
      ENDIF.
      WRITE sy-index.
    ENDDO.
  ENDIF.
ENDIF.
----

include::../resources.adoc[]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../parameters.adoc[]

include::../highlighting.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
