<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p><code>UNSTRING</code> and <code>STRING</code> statements can be used on one side to divide a string into sub-strings and on the other side to concatenate the contents of several strings into one single date item.</p>
</div>
<div class="paragraph">
<p>As useful those statements can be, the other way to get the same result in a more declarative and maintainable way is to extend the data items structure.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="paragraph">
<p>Waiting for feedback from BNP</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_27_may_2014_071837_freddy_mallet_wrote">on 27 May 2014, 07:18:37 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>I&#8217;m closing "Won&#8217;t Fix" this RSPEC as this will lead to generate lot of false-positives.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_9_jun_2014_122634_freddy_mallet_wrote">on 9 Jun 2014, 12:26:34 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>We should provide the XPath expression.</p>
</div>
</div>
</div>
</div>