<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Using boxed type suggests that <code>null</code> is a possible value for the variable. Use of the primitive type should be preferred if this is not the case to avoid any confusion about possible values variable can contain.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">Integer x = 5;</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">int x = 5;</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Use ${primitiveType} instead of ${boxedType}</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>variable declaration</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_7_nov_2018_144549_alexandre_gigleux_wrote">on 7 Nov 2018, 14:45:49 Alexandre Gigleux wrote:</h3>
<div class="paragraph">
<p>\[~tibor.blenessy] Can you review the title of this RSPEC to make it compliant with our title policy?</p>
</div>
<div class="paragraph">
<p>I suggest something like: "Primitive types for non-null values should preferred to boxed types".</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_15_nov_2018_105137_alexandre_gigleux_wrote">on 15 Nov 2018, 10:51:37 Alexandre Gigleux wrote:</h3>
<div class="paragraph">
<p>\[~tibor.blenessy]</p>
</div>
<div class="paragraph">
<p>I believe it would be great to write a sentence or two to explain what wrong can happen. Why this is so bad to used boxed type? Is it a performance problem, a memory problem?</p>
</div>
<div class="paragraph">
<p>Also in the Java code sample, we don&#8217;t see/understand why this is bad to used boxed type. I believe you should had a <code>if x == null</code> to justify.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_18_nov_2018_210309_tibor_blenessy_wrote">on 18 Nov 2018, 21:03:09 Tibor Blenessy wrote:</h3>
<div class="paragraph">
<p>\[~alexandre.gigleux] I don&#8217;t understand the question, nothing wrong can happen, it&#8217;s just confusing, that&#8217;s what is explained in the rspec. Call me if we need to discuss further</p>
</div>
</div>
</div>
</div>