== Why is this an issue?

This rule allows you to track the usage of the ``++SuppressMessage++`` attributes and ``++#pragma warning disable++`` mechanism.

=== Noncompliant code example

[source,csharp]
----
[SuppressMessage("", "S100")]
...

#pragma warning disable S100
...
#pragma warning restore S100
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Do not suppress issues.


'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
