This rule allows you to track the usage of the ``++SuppressMessage++`` attributes and ``++#pragma warning disable++`` mechanism.

== Noncompliant Code Example

----
[SuppressMessage("", "S100")]
...

#pragma warning disable S100
...
#pragma warning restore S100
----
