<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>C&#43;&#43;17 introduced structured binding, a syntax that initializes multiple entities by elements or members of an object. It is handy to emulate several return values from a function.</p>
</div>
<div class="paragraph">
<p>Suppose you have a function that returns a pair:</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">std::pair&lt;std::string, std::string&gt; getPair();</code></pre>
</div>
</div>
<div class="paragraph">
<p>Structured binding allows you to bind the members of the <code>std::pair</code> class directly to new names:</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">auto [firstName, lastName] = getPair();</code></pre>
</div>
</div>
<div class="paragraph">
<p>The names <code>firstName</code> and <code>lastName</code> are called structured bindings. As you can see, structured binding makes the code more readable as it allows binding values to names that carry information about their purpose.</p>
</div>
<div class="paragraph">
<p>Structured binding works with:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>Raw arrays, by binding a name to each element</p>
</li>
<li>
<p>Any type that has a tuple-like API</p>
</li>
<li>
<p>Classes and structures where all non-static data members are publicly accessible</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>This rule will detect places where <code>std::pair</code> and <code>std::tuple</code> can be effortlessly replaced by a structured binding.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">void printingMap(const std::map&lt;int, std::string&gt;&amp; map) {
  for (const auto&amp; elem : map) { // Noncompliant
    std::cout &lt;&lt; elem.first &lt;&lt; ": " &lt;&lt; elem.second &lt;&lt; "\n";
  }
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">void printingMap(const std::map&lt;int, std::string&gt;&amp; map) {
  for (const auto&amp; [key, value] : map) { // Compliant
    std::cout &lt;&lt; key &lt;&lt; ": " &lt;&lt; value &lt;&lt; "\n";
  }
}</code></pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>C&#43;&#43; reference - <a href="https://en.cppreference.com/w/cpp/language/structured_binding">Structured binding declaration</a></p>
</li>
</ul>
</div>
</div>
<div class="sect2">
<h3 id="_external_coding_guidelines">External coding guidelines</h3>
<div class="ulist">
<ul>
<li>
<p>C++ Core Guidelines - <a href="https://github.com/isocpp/CppCoreGuidelines/blob/e49158a/CppCoreGuidelines.md#f21-to-return-multiple-out-values-prefer-returning-a-struct-or-tuple">F.21: To return multiple "out" values, prefer returning a struct or tuple</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>