``++SQL COUNT(..), MIN(..), MAX(..), SUM(..), AVG(..)++`` aggregate functions cause the SAP table buffer to be bypassed, so the use of these functions can lead to performance issues.


== Noncompliant Code Example

[source,abap]
----
SELECT COUNT(*) 
       FROM persons 
       INTO count 
       WHERE city = 'NEW YORK'. 
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

endif::env-github,rspecator-view[]
