<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Non-static variables initialized with <code>INZ()</code> are only reliably initialized on the first run through the code in a  "terminate and stay resident" (TSR) program. Subsequent calls to the program within the same job do not re-initialize the variable with the value from <code>INZ()</code> unless the last record indicator is set at the end of the program.</p>
</div>
<div class="paragraph">
<p>Without setting the last record indicator, the assumptions the code makes about the variable&#8217;s initial value will be wrong every time but one, potentially leading to bad program behavior.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-rpg" data-lang="rpg">     D TESTINZ         PR                  extpgm('TESTINZ')
     D  Parm1                        15A
     D  Parm2                        15A   CONST

     DMUTVAR           S             15A   INZ('ABC')  // Noncompliant; *inlr not set

     D TESTINZ         PI
     D  Parm1                        15A
     D  Parm2                        15A   CONST
      /Free
       IF %PARMS &gt; 1 and Parm2 &lt;&gt; '';
         MUTVAR=Parm2;
       ENDIF;
       DSPLY(E) ('MUTVAR:' + MUTVAR);
       Parm1=MUTVAR;
       return;
       /End-free</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-rpg" data-lang="rpg">     D TESTINZ         PR                  extpgm('TESTINZ')
     D  Parm1                        15A
     D  Parm2                        15A   CONST

     DMUTVAR           S             15A   INZ('ABC')

     D TESTINZ         PI
     D  Parm1                        15A
     D  Parm2                        15A   CONST
      /Free
       IF %PARMS &gt; 1 and Parm2 &lt;&gt; '';
         MUTVAR=Parm2;
       ENDIF;
       DSPLY(E) ('MUTVAR:' + MUTVAR);
       Parm1=MUTVAR;
       *inlr = *on;
       return;
       /End-free</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove this use of "Inz()", and set "xxx"'s default value at the beginning of the program.</p>
</div>
</div>
</div>
</div>