== Why is this an issue?

Performing math on variables that are declared - explicitly or implicitly - as ``++DISPLAY++`` or ``++NATIONAL++`` is much less efficient than on ``++COMPUTATIONAL++``, ``++COMP++``, or ``++BINARY++`` variables. That's because ``++COMP++`` variables, for instance, are defined for binary storage, which makes math on them more efficient. That's why values that are going to be used primarily for math should be declared with a math type. When math isn't a primary use, it may not make sense to change the declared type, but ``++MOVE++``ing the value to a ``++COMP++`` variable and performing the math on it instead would.


It is important to note however, that ``++COMPUTATIONAL++``, ``++COMP++``, and ``++BINARY++`` formats should be used with caution if the variable will be passed to other systems which may not use the same storage format.


=== Noncompliant code example

[source,cobol]
----
01 W-AMOUNT-VALUE PIC 9(17).
01 W-AMOUNT-DECIMAL PIC 9.

COMPUTE W-CONV-AMOUNT = W-AMOUNT-VALUE * 10 ** W-AMOUNT-DECIMAL  *> Noncompliant
----


=== Compliant solution

[source,cobol]
----
01 W-AMOUNT-VALUE PIC 9(17) COMP-5.
01 W-AMOUNT-DECIMAL PIC 9 COMP-5.

COMPUTE W-CONV-AMOUNT = W-AMOUNT-VALUE * 10 ** W-AMOUNT-DECIMAL
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Change the declaration of "xxx" to have a computational format.


=== Highlighting

math operator and variable name


'''
== Comments And Links
(visible only on this page)

=== on 24 Mar 2016, 14:46:38 Ann Campbell wrote:
Open question: what is the best action to take? Is some sort of cast or conversion available or must the declaration be changed?


Also, it would be nice to have an understanding of _why_ it's inefficient.

=== on 29 Apr 2016, 10:13:58 Pierre-Yves Nicolas wrote:
\[~ann.campbell.2], Microfocus has some documentation on https://supportline.microfocus.com/documentation/books/sx51/prwrit.htm[Writing Efficient Programs].

The IBM COBOL Programming guide has a section on http://www.ibm.com/support/knowledgecenter/SS6SG3_6.1.0/com.ibm.cobol61.ent.doc/PGandLR/tasks/tpeff05a.html[Choosing efficient computational data items].


I think we should not activate this rule by default as the real performance impact may depend on the context.

=== on 29 Apr 2016, 16:52:24 Ann Campbell wrote:
\[~pierre-yves.nicolas] thanks for the references! I've expanded the description some based mostly on the Microfocus reference. I'm wondering how broadly it applies though. Specifically, do you think this bit applies beyond Microfocus:?

____
In native code programs, it is more efficient to move integer data items that are not COMP-X or COMP-5 to COMP-5 data items before doing arithmetic operations on them.

____

=== on 29 Apr 2016, 17:24:13 Pierre-Yves Nicolas wrote:
\[~ann.campbell.2], my understanding is that this always applies unless when compiling a program to https://supportline.microfocus.com/documentation/books/sx51/pracmp.htm["intermediate code"]. I think that "intermediate code" is specific to Microfocus because they target multiple operating systems.



=== on 29 Apr 2016, 17:44:20 Ann Campbell wrote:
I think I'm done [~pierre-yves.nicolas]. Take a look.

=== on 2 May 2016, 13:20:11 Pierre-Yves Nicolas wrote:
Looks good, thanks [~ann.campbell.2].

endif::env-github,rspecator-view[]
