<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when the <code>pandas.DataFrame.groupby</code> method is not followed by an aggregation method call.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>A <code>groupby</code> operation involves some combination of splitting the data frame, applying a function, and combining the results. This can be used to group large amounts of data and compute operations on these groups.</p>
</div>
<div class="paragraph">
<p>Even though the <code>DataFrame.groupby</code> method call instantiates the <code>GroupBy</code> object it only verifies that a valid mapping is passed - no real data splitting occurs until it is needed.</p>
</div>
<div class="paragraph">
<p>To perform the data transformation, an aggregation method should be called from the <code>GroupBy</code> object, e.g. <code>apply</code> or <code>sum</code>.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">import pandas as pd

df = pd.DataFrame({
        'X': ['A', 'B', 'A', 'C'],
        'Y': [10, 7, 12, 5]
    })


df.groupby('X') # Noncompliant: groupby itself does nothing</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">import pandas as pd

df = pd.DataFrame({
        'X': ['A', 'B', 'A', 'C'],
        'Y': [10, 7, 12, 5]
    })


df.groupby('X').sum() # Compliant</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Pandas Documentation - <a href="https://pandas.pydata.org/pandas-docs/stable/user_guide/groupby.html#splitting-an-object-into-groups">Group by: split-apply-combine</a></p>
</li>
<li>
<p>Pandas Documentation - <a href="https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.groupby.html#pandas.DataFrame.groupby">DataFrame.groupby method reference</a></p>
</li>
<li>
<p>Pandas Documentation - <a href="https://pandas.pydata.org/docs/reference/groupby.html">GroupBy reference</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>