<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>An unused parameter is a parameter that has been declared but is not used anywhere in the block of code where it is defined.
It is dead code, contributing to unnecessary complexity and leading to confusion when reading the code.
Therefore, it should be removed from your code to maintain clarity and efficiency.</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="paragraph">
<p>Having unused parameters in your code can lead to several issues:</p>
</div>
<div class="ulist">
<ul>
<li>
<p><strong>Decreased Readability</strong>: Unused parameters can make your code more difficult to read.
They add extra lines and complexity, which can distract from the main logic of the code.</p>
</li>
<li>
<p><strong>Misunderstanding</strong>: When other developers read your code, they may wonder why a variable is declared but not used.
This can lead to confusion and misinterpretation of the code&#8217;s intent.</p>
</li>
<li>
<p><strong>Potential for Bugs</strong>: If a parameter is declared but not used, it might indicate a bug or incomplete code.
For example, if you declared a parameter intending to use it in a calculation, but then forgot to do so, your program might not work as expected.</p>
</li>
<li>
<p><strong>Maintenance Issues</strong>: Unused parameters can make code maintenance more difficult.
If a programmer sees an unused parameter, they might think it is a mistake and try to 'fix' the code, potentially introducing new bugs.</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>In summary, unused local parameters can make your code less readable, more confusing, and harder to maintain, and they can potentially lead to bugs.
Therefore, it is best to remove them.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it_in_json_templates">How to fix it in JSON templates</h2>
<div class="sectionbody">
<div class="paragraph">
<p>The fix for this issue is straightforward.
Once you ensure the unused parameter is not part of an incomplete implementation leading to bugs, you just need to remove it.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-json" data-lang="json">{
  "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#",
  "parameters": {
      "unusedParameter": {
          "type": "string"
      },
      "virtualMachinesName": {
          "type": "string"
      }
  },
  "resources": [
    {
      "type": "Microsoft.Compute/virtualMachines",
      "name": "[parameters('virtualMachinesName')]",
      "apiVersion": "2023-09-01",
      "location": "[resourceGroup().location]"
    }
  ]
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-json" data-lang="json">{
  "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#",
  "parameters": {
      "virtualMachinesName": {
          "type": "string"
      }
  },
  "resources": [
    {
      "type": "Microsoft.Compute/virtualMachines",
      "name": "[parameters('virtualMachinesName')]",
      "apiVersion": "2023-09-01",
      "location": "[resourceGroup().location]"
    }
  ]
}</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it_in_bicep">How to fix it in Bicep</h2>
<div class="sectionbody">
<div class="paragraph">
<p>The fix for this issue is straightforward.
Once you ensure the unused parameter is not part of an incomplete implementation leading to bugs, you just need to remove it.</p>
</div>
<div class="sect2">
<h3 id="_code_examples_2">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example_2">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-bicep" data-lang="bicep">param unusedParameter string // Noncompliant
param virtualMachinesName string = '${uniqueString(resourceGroup().id)}'

resource vm 'Microsoft.Compute/virtualMachines@2023-09-01' = {
  name: virtualMachinesName
  location: resourceGroup().location
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution_2">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-bicep" data-lang="bicep">param virtualMachinesName string = '${uniqueString(resourceGroup().id)}'

resource vm 'Microsoft.Compute/virtualMachines@2023-09-01' = {
  name: virtualMachinesName
  location: resourceGroup().location
}</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Microsoft - <a href="https://learn.microsoft.com/en-us/azure/azure-resource-manager/templates/best-practices#parameters">ARM template best practices</a></p>
</li>
<li>
<p>Microsoft - <a href="https://learn.microsoft.com/en-us/azure/azure-resource-manager/bicep/best-practices#names">Best practices for Bicep</a></p>
</li>
<li>
<p>Microsoft - <a href="https://learn.microsoft.com/en-us/azure/azure-resource-manager/templates/parameters">Parameters in ARM templates</a></p>
</li>
<li>
<p>Microsoft - <a href="https://learn.microsoft.com/en-us/azure/azure-resource-manager/bicep/parameters">Parameters in Bicep</a></p>
</li>
</ul>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove the unused parameter "xxxx".</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>Highlight the unused parameter.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
</div>
</div>