include::../description.adoc[]

== Noncompliant Code Example

----
Try
    ' Some work which end up throwing an exception
    Throw New ArgumentException()
Finally
    ' Clean up
    Throw New InvalidOperationException()       ' Noncompliant; will mask the ArgumentException
End Try
----

== Compliant Solution

----
Try
    ' Some work which end up throwing an exception
    Throw New ArgumentException()
Finally
    ' Clean up
End Try
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
