== Why is this an issue?

Caller information attributes: `CallerFilePathAttribute`, `CallerLineNumberAttribute`, and `CallerArgumentExpressionAttribute` provide a way to get information about the caller of a method through optional parameters. But the arguments for these optional parameters are only generated if they are not explicitly defined in the call. Thus, specifying the argument values defeats the purpose of the attributes.


=== Noncompliant code example

[source,text]
----
void TraceMessage(string message,
  [CallerFilePath] string filePath = null,
  [CallerLineNumber] int lineNumber = 0)
{
  /* ... */
}

void MyMethod()
{
  TraceMessage("my message", "A.B.C.Foo.cs", 42); // Noncompliant
}
----


=== Compliant solution

[source,text]
----
void TraceMessage(string message,
  [CallerFilePath] string filePath = "",
  [CallerLineNumber] int lineNumber = 0)
{
  /* ... */
}

void MyMethod()
{
  TraceMessage("my message");
}
----


=== Exceptions

`CallerMemberName` is not checked to avoid False-Positives with WPF/UWP applications.
`System.Diagnostics.Debug.Assert` is excluded. The signature of this method has been updated from .NET9 onwards from `public static void Assert(bool condition, string message)` to `public static void Assert([DoesNotReturnIf(false)] bool condition, [CallerArgumentExpression(nameof(condition))] string? message = null)`.
The default message provided by this method is the failed assertion and its position in the code, however a custom message from the developer is many times preferred as an explanation of the failed assertion.