<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>The <code>String</code> class has a <code>toString</code> method because <code>Object</code> itself does. I.e. it couldn&#8217;t not have the method. But having the method doesn&#8217;t mean it should be used. In fact doing so is worse than pointless since it returns the <code>String</code> itself.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">String str1 = "Now is the time for all good people";
String str2 = str1.toString();</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">String str1 = "Now is the time for all good people";
String str2 = str1;</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_duplicates_s1858">duplicates: <a data-rspec-id="S1858" class="rspec-auto-link">S1858</a></h3>

</div>
<div class="sect2">
<h3 id="_on_10_oct_2014_110835_freddy_mallet_wrote">on 10 Oct 2014, 11:08:35 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>@Ann, my 2 cents:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>I really like this tag "clumsy" :)</p>
</li>
<li>
<p>using the toString() method doesn&#8217;t imply any CPU or memory overhead (just the very low cost of a method call). So I would associate this rule to the category Maintainability &gt; Readability
*</p>
</li>
</ul>
</div>
</div>
</div>
</div>