To ensure future code portability, obsolete keywords should not be used. The following keywords were declared obsolete in the COBOL ANSI-85 standard and removed in the ISO/IEC 1989:2002 standard: 


* Paragraphs: ``++AUTHOR++``, ``++INSTALLATION++``, ``++DATE-WRITTEN++``, ``++DATE-COMPILED++``, ``++SECURITY++``
* Clauses: ``++DATA RECORD(S)++``, ``++LABEL RECORD(S)++``, ``++MEMORY SIZE++``, ``++MULTIPLE FILE (TAPE)++``, ``++RERUN++``, ``++VALUE OF++``, ``++CODE SEGMENT-LIMIT++``
* Statements: ``++ALTER++``, ``++ENTER++``, ``++STOP++`` _literal_, ``++GO TO++`` without an argument
* Phrases: ``++REVERSED++`` phrase of the ``++OPEN++`` statement
* Special registers: ``++DEBUG-ITEM++``
* Sections: Debugging sections
* Declarative: ``++USE FOR DEBUGGING++``

The following keywords were declared obsolete in the ISO/IEC 1989:2002 standard:

* Phrase: ``++DEBUGGING MODE++``
* Clause: ``++PADDING CHARACTER++``


== Noncompliant Code Example

----
IDENTIFICATION DIVISION.
PROGRAM-ID.  AcceptAndDisplay.
AUTHOR.  Michael Coughlan.  *> Noncompliant
----


== Compliant Solution

----
IDENTIFICATION DIVISION.
PROGRAM-ID.  AcceptAndDisplay.
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
