<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This is an issue when a Flask signal handler function does not include a <code>**extra</code> parameter in its signature.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Flask signal handlers that don&#8217;t include a <code>**extra</code> parameter may break when Flask is upgraded to a newer version that introduces additional arguments to signals.</p>
</div>
<div class="paragraph">
<p>Flask uses the Blinker library to implement signals, which are a lightweight way to notify subscribers of certain events during the application lifecycle. When you subscribe to a signal, Flask calls your handler function with specific arguments.</p>
</div>
<div class="paragraph">
<p>The Flask team may add new arguments to signals in future versions to provide additional context or functionality. If your signal handler doesn&#8217;t accept these new arguments through a <code>**extra</code> parameter, Python will raise a TypeError when Flask tries to call your function with the unexpected arguments.</p>
</div>
<div class="paragraph">
<p>This creates a maintenance burden because your application could suddenly break after a Flask upgrade, even though your code hasn&#8217;t changed. The <code>**extra</code> parameter acts as a safety net, allowing your handler to gracefully accept any additional arguments that Flask might introduce.</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="paragraph">
<p>When Flask introduces new signal arguments in a future version, signal handlers without <code>**extra</code> parameters will raise TypeError exceptions. This can cause application crashes, failed request processing, or broken functionality that depends on signal handling.</p>
</div>
<div class="paragraph">
<p>The impact is particularly problematic during Flask upgrades, as previously working code will suddenly fail without any changes to your application logic.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it_in_flask">How to fix it in Flask</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Add a <code>**extra</code> parameter to the signal handler function signature. This parameter will capture any additional arguments that Flask might pass to the signal in future versions.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">from flask import template_rendered

@template_rendered.connect_via(app)
def when_template_rendered(sender, template, context):  # Noncompliant
    print(f'Template {template.name} is rendered')</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">from flask import template_rendered

@template_rendered.connect_via(app)
def when_template_rendered(sender, template, context, **extra):
    print(f'Template {template.name} is rendered')</code></pre>
</div>
</div>
<div class="paragraph">
<p>The same principle applies when using the <code>connect()</code> method directly to subscribe to signals.</p>
</div>
</div>
<div class="sect3">
<h4 id="_noncompliant_code_example_2">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">from flask import request_started

def log_request_start(sender):  # Noncompliant
    print(f'Request started for {sender.name}')

request_started.connect(log_request_start, app)</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution_2">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">from flask import request_started

def log_request_start(sender, **extra):
    print(f'Request started for {sender.name}')

request_started.connect(log_request_start, app)</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Flask Signals Documentation - <a href="https://flask.palletsprojects.com/en/stable/signals/">Official Flask documentation on signals, including the recommendation to use **extra parameters</a></p>
</li>
<li>
<p>Blinker Documentation - <a href="https://blinker.readthedocs.io/">Documentation for the Blinker library that Flask uses to implement signals</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>