Waiting for explanation from BNP. This rule seems to be really specific to BNP


== Noncompliant Code Example

TODO


ifdef::env-github,rspecator-view[]
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
