== Why is this an issue?

Waiting for explanation from BNP. This rule seems to be really specific to BNP


=== Noncompliant code example

TODO


ifdef::env-github,rspecator-view[]
'''
== Comments And Links
(visible only on this page)

=== on 26 May 2014, 15:35:43 Freddy Mallet wrote:
Rule specific to the BNP context and which can be easily implemented with the XPath rule: //cancelStatement

endif::env-github,rspecator-view[]
