== Why is this an issue?

In Dart, nullability is a part of the type system.
By default, any given type `T` is non-nullable.
If you append a "?" to the type, it becomes nullable: `T?`.

When accessing properties or functions of a nullable type, you need to handle the case when the target is `null`.
However, while accessing a non-nullable type, it is redundant to test for `null`, as the compiler statically ensures that the value can never be `null`.
So all the nullability checks on the non-nullable types are considered code smells.

When overriding an equality operator it doesn't make sense to check for null, since you can directly check that the object is of a required type. In this example we can see that `other is A` will be true only if `other` is not `null`.

[source,dart]
----
class A {
  final String? value;

  @override
  operator ==(Object? other) =>
    other is A && value == other.value;
}
----

== How to fix it

=== Code examples

==== Noncompliant code example

[source,dart,diff-id=1,diff-type=noncompliant]
----
class A {
  final String? value;

  @override
  operator ==(Object? other) =>
    other != null && other is A && value == other.value; // Noncompliant, redundant null check
}
----

==== Compliant solution

[source,dart,diff-id=1,diff-type=compliant]
----
class A {
  final String? value;

  @override
  operator ==(Object? other) =>
    other is A && value == other.value;
}
----

== Resources

=== Documentation

* Dart Docs - https://dart.dev/tools/linter-rules/avoid_null_checks_in_equality_operators[Dart Linter rule - avoid_null_checks_in_equality_operators]
* Dart Docs - https://dart.dev/null-safety/understanding-null-safety[Dart Null Safety - Understanding null safety]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Unnecessary null comparison in implementation of '=='.

=== Highlighting

The entire null comparison expression, e.g. `o == null` or `o != null`.

'''
== Comments And Links
(visible only on this page)

endif::env-github,rspecator-view[]
