<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>While it is possible to manually set a primary key value, doing so almost guarantees a key clash at some point. Instead, primary key values should be set by (in descending order of desirability):</p>
</div>
<div class="ulist">
<ul>
<li>
<p>automatic generation by the database via a column definition such as <code>PROD_ID INT NOT NULL GENERATED ALWAYS AS IDENTITY (START WITH 1, INCREMENT BY 1, NO CACHE)</code></p>
</li>
<li>
<p>the <code>Generate_Unique()</code> function</p>
</li>
<li>
<p>a value pulled directly from a sequence, like so: <code>nextval for SEQ_PRODUCT</code></p>
</li>
</ul>
</div>
<div class="paragraph">
<p>This rule raises an issue when an <code>INSERT</code> statement assigns values to identity columns that are configured to always generate their values.</p>
</div>
<div class="paragraph">
<p><strong>Note</strong> That this rule raises issues only when a database catalog is provided during the SonarQube analysis.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cobol" data-lang="cobol">CREATE table my_table (
     column_a integer GENERATED ALWAYS AS IDENTITY primary key not null,
     column_b varchar(50)
);

INSERT into my_table (column_a, column_b)
VALUES (1, 'Hello World');  -- Noncompliant</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Do not set the value[s] of identity column[s] "xxx"[, "yyy"&#8230;&#8203;].</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>Primary: first identity column being set</p>
</div>
<div class="paragraph">
<p>Secondary: additional identity columns being set</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_26_jul_2016_193350_ann_campbell_wrote">on 26 Jul 2016, 19:33:50 Ann Campbell wrote:</h3>
<div class="paragraph">
<p><a href="http://www.ibm.com/developerworks/data/library/techarticle/0205pilaka/0205pilaka2.html" class="bare">http://www.ibm.com/developerworks/data/library/techarticle/0205pilaka/0205pilaka2.html</a></p>
</div>
<div class="paragraph">
<p><a href="http://stackoverflow.com/a/9703877/2662707" class="bare">http://stackoverflow.com/a/9703877/2662707</a></p>
</div>
</div>
</div>
</div>