== Why is this an issue?

This rule allows to make sure that all JSF Expressions are syntactically correct.


=== Noncompliant code example

[source,html]
----
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" 
"http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml"
   xmlns:h="http://java.sun.com/jsf/html">
   <h:body>
     First name 
     <h:outputText value="#{user.firstName && @@}"/>   <!-- Noncompliant -->
   </h:body>
</html> 
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Fix this expression: Unknown function ...

Fix this expression: Error ...


=== Parameters

.functions
****
_STRING_

Comma-separated list of names of functions
****


'''
== Comments And Links
(visible only on this page)

=== on 18 Sep 2013, 10:08:46 Freddy Mallet wrote:
Implemented by \http://jira.codehaus.org/browse/SONARPLUGINS-3155

endif::env-github,rspecator-view[]
