== Why is this an issue?

Using "#pragma warning (default: ...)" resets the warning in question to its default settings, which may not be what the compiler was initially invoked with. Typically, this usage is seen after a warning is turned off, in preparation for code that is known to cause warnings. Instead, the warning's current state should be saved, and then restored after the code in question.


=== Noncompliant code example

[source,cpp]
----
#pragma warning (disable: TheWarning)
#include problem_code.h
#pragma warning (default: TheWarning)
----


=== Compliant solution

[source,cpp]
----
#pragma warning (push)
#include problem_code.h
#pragma warning (pop)
----


== Resources

* https://wiki.sei.cmu.edu/confluence/x/6NUxBQ[CERT, MSC00-C.] - Compile cleanly at high warning levels

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

endif::env-github,rspecator-view[]
