To remove all the elements of a collection ``++c++`` the method ``++c.clear()++`` should be used. Calling the method ``++c.removeAll(c)++`` will lead to the same result but with a high CPU overhead. 


== Noncompliant Code Example

[source,text]
----
c.removeAll(c);
----


== Compliant Solution

[source,text]
----
c.clear();
----

ifdef::env-github,rspecator-view[]

'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]

endif::env-github,rspecator-view[]
