To remove all the elements of a collection ``++c++`` the method ``++c.clear()++`` should be used. Calling the method ``++c.removeAll(c)++`` will lead to the same result but with a high CPU overhead. 


== Noncompliant Code Example

----
c.removeAll(c);
----


== Compliant Solution

----
c.clear();
----

