== Why is this an issue?

To remove all the elements of a collection ``++c++`` the method ``++c.clear()++`` should be used. Calling the method ``++c.removeAll(c)++`` will lead to the same result but with a high CPU overhead. 


=== Noncompliant code example

[source,text]
----
c.removeAll(c);
----


=== Compliant solution

[source,text]
----
c.clear();
----

ifdef::env-github,rspecator-view[]

'''
== Comments And Links
(visible only on this page)

=== duplicates: S2114


endif::env-github,rspecator-view[]
