include::../../../shared_content/secrets/description.adoc[]

== Why is this an issue?

include::../../../shared_content/secrets/rationale.adoc[]

If an attacker gains access to a Shopify app token or a Shopify Partners token, they might be able to compromise the Shopify environment linked to this token. As this environment typically contains both important financial data and the personal information of clients, a breach by a malicious entity could have a serious impact on the organization.

=== What is the potential impact?

Shopify contains both important information about customers, as well as financial information in general. If an attacker manages to get access to either of those through a leaked secret, they could severely impact the business in multiple ways. 

include::../../../shared_content/secrets/impact/personal_data_compromise.adoc[]

Furthermore, the personal identifiable information contained by the Shopify platform could be used for phishing. Not sufficiently protecting the sensitive information of clients, such as addresses, email addresses and even financial information, can directly hurt these clients and will also hurt the reputation of the organization.

include::../../../shared_content/secrets/impact/disclosure_of_financial_data.adoc[]

== How to fix it

include::../../../shared_content/secrets/fix/revoke.adoc[]

include::../../../shared_content/secrets/fix/vault.adoc[]

=== Code examples

==== Noncompliant code example

[source,python,diff-id=1,diff-type=noncompliant]
----
import requests

token = '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'  # Noncompliant
response = requests.get('https://test-shop.myshopify.com/admin/api/2021-07/shop.json', headers={
    'X-Shopify-Access-Token': token,
    'Content-Type': 'application/json'
})
----

==== Compliant solution

[source,python,diff-id=1,diff-type=compliant]
----
import requests

token = os.getenv('SHOPIFY_ACCESS_TOKEN')
response = requests.get('https://test-shop.myshopify.com/admin/api/2021-07/shop.json', headers={
    'X-Shopify-Access-Token': token,
    'Content-Type': 'application/json'
})
----

//=== How does this work?

//=== Pitfalls

//=== Going the extra mile

== Resources

=== Documentation

Shopify.dev docs - https://shopify.dev/docs/apps/auth/admin-app-access-tokens#requirements[Access tokens for custom apps in the Shopify admin]

include::../../../shared_content/secrets/resources/standards.adoc[]
