<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Using <code>mem::replace(&amp;mut _, mem::uninitialized())</code> or <code>mem::replace(&amp;mut _, mem::zeroed())</code> leads to undefined behavior even if the value is overwritten later. This is because the uninitialized value might be observed in the case of a panic, which can lead to unpredictable and dangerous consequences in your program.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-rust" data-lang="rust">use std::mem;

fn may_panic(v: Vec&lt;i32&gt;) -&gt; Vec&lt;i32&gt; { v }

#[allow(deprecated, invalid_value)]
fn myfunc(v: &amp;mut Vec&lt;i32&gt;) {
    let taken_v = unsafe { mem::replace(v, mem::uninitialized()) }; // Noncompliant
    let new_v = may_panic(taken_v); // undefined behavior on panic
    mem::forget(mem::replace(v, new_v));
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-rust" data-lang="rust">use std::mem;
use take_mut::take;

fn may_panic(v: Vec&lt;i32&gt;) -&gt; Vec&lt;i32&gt; { v }

fn myfunc(v: &amp;mut Vec&lt;i32&gt;) {
    let new_v = take(v, |old_v| may_panic(old_v)); // Compliant
    mem::forget(mem::replace(v, new_v));
}</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Clippy Lints - <a href="https://rust-lang.github.io/rust-clippy/master/index.html#mem_replace_with_uninit" class="bare">https://rust-lang.github.io/rust-clippy/master/index.html#mem_replace_with_uninit</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>