== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,cpp]
----
void fun() {
  label: doSomething();
}
----

=== Compliant solution

[source,cpp]
----
void fun() {
  doSomething();
}
----

== Resources

* MISRA C:2012, 2.6 - A function should not contain unused label declarations
* https://wiki.sei.cmu.edu/confluence/x/5dUxBQ[CERT, MSC12-C.] - Detect and remove code that has no effect or is never executed

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
