include::../description.adoc[]

== Noncompliant Code Example

----
void fun() {
  label: doSomething();
}
----

== Compliant Solution

----
void fun() {
  doSomething();
}
----

== See

* MISRA C:2012, 2.6 - A function should not contain unused label declarations
* https://wiki.sei.cmu.edu/confluence/x/5dUxBQ[CERT, MSC12-C.] - Detect and remove code that has no effect or is never executed

ifdef::env-github,rspecator-view[]
'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
