<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule reduces confusion over which constructor will be used, and with what parameters.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="paragraph">
<p>Given the following classes:</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">class V
{
public:
  V ( ) { }
  V ( int32_t i ) { }
};

class C1 : public virtual V
{
public:
 C1 ( ) : V ( 21 ) { }
};

class C2 : public virtual V
{
public:
  C2 ( ) : V ( 42 ) { }
};</code></pre>
</div>
</div>
<div class="paragraph">
<p>The following examples are not compliant:</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">class B : public V
{
  public B( ) { } // Non-compliant - benign
}

class D: public C1, public C2
{
public:
 D ( ) { } // Non-compliant - it is unclear which constructor of V is called
};</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="paragraph">
<p>Given the previous definitions of classes V, C1 and C2 the following definitions of B and D are compliant:</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">class B : public V
{
  public B( ) : V ( ) { } // Compliant
}

class D: public C1, public C2
{
public:
 D ( ) : C1 ( ), C2 ( ), V ( ) { } // Compliant
};</code></pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p>MISRA C&#43;&#43;:2008, 12-1-2 - All constructors of a class should explicitly call a constructor for all of its immediate base classes and all virtual base classes.</p>
</li>
</ul>
</div>
<hr>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Add an explicit call of constructor for [base class xxx::virtual base class xxx]</p>
</div>
</div>
</div>
</div>