=== Noncompliant code example

[source,text]
----
public class Human extends Animal {
  private Date dateOfBirth;

  public Human(Gender gender) {  // Compliant; doesn't perform initialization directly
    this(gender, new Date());
  }

  public Human (Gender gender, Date dob) {  // Noncompliant
    this.dateOfBirth = dob;
  }
}
----
