== How to fix it in SignXML

=== Code examples

The following noncompliant code example verifies an XML signature without
providing a trusted signing authority. This code will accept any signature
created from a generally trusted certificate, for example, a Let's encrypt one.

==== Noncompliant code example

[source,python,diff-id=1,diff-type=noncompliant]
----
from lxml import etree
from signxml import XMLVerifier

xml_file = open("signed.xml", "rb")
xml = etree.parse(xml_file)

XMLVerifier().verify(xml) # Noncompliant
----

==== Compliant solution

[source,python,diff-id=1,diff-type=compliant]
----
from lxml import etree
from signxml import XMLVerifier

xml_file = open("signed.xml", "rb")
xml = etree.parse(xml_file)

cert_file = open("cert.pem", "rb")
cert = cert_file.read()
XMLVerifier().verify(xml, x509_cert=cert)
----

=== How does this work?

Here, the compliant solution provides a trusted certificate to the signature validation function. This will ensure only signatures computed with the
private key associated with the provided certificate will be accepted.

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Change this code to only accept signatures computed from a trusted party.

=== Highlight

When no `expect_config` is provided:

* Highlight the call to the `verify` method.

When an `expect_config` is provided:

* Highlight the `verify` call.
* As a secondary location, highlight the `require_x509=False` in the 
`SignatureConfiguration` instantiation.


'''
endif::env-github,rspecator-view[]
