This rule raises an issue when a struct method doesn't access any fields of the struct, suggesting it could be a standalone function instead.

== Why is this an issue?

Go emphasizes simplicity and directness in code design. When methods are defined on structs but don't actually use the struct's fields or maintain any meaningful state, they create unnecessary complexity.

This pattern often emerges when developers coming from object-oriented languages try to group related functionality using struct methods, similar to how they would use classes. However, Go's philosophy favors functions over methods when no state management is involved.

Stateless struct methods introduce several problems:

* They add cognitive overhead by suggesting there's state management when there isn't
* They require unnecessary struct instantiation
* They make the code less idiomatic and harder to understand for Go developers
* They can mislead other developers about the purpose and design of the struct

Go's design encourages using the simplest construct that solves the problem. If functionality doesn't require access to struct fields, a function is the more appropriate choice.

=== What is the potential impact?

Using struct methods unnecessarily reduces code clarity and maintainability. It creates false complexity that makes the codebase harder to understand and maintain. The extra indirection through struct instantiation also adds minor performance overhead without any benefit.

== How to fix it

Convert the struct method to a standalone function. Remove the receiver parameter and make the function independent.

=== Code examples

==== Noncompliant code example

[source,go,diff-id=1,diff-type=noncompliant]
----
type Calculator struct{}

func (c Calculator) Add(a, b int) int { // Noncompliant
    return a + b
}
----

==== Compliant solution

[source,go,diff-id=1,diff-type=compliant]
----
func Add(a, b int) int {
    return a + b
}
----

== Resources

=== Documentation

 * Effective Go - Methods - https://golang.org/doc/effective_go#methods[Official Go documentation on when and how to use methods effectively]

 * Go Code Review Comments - Receiver Type - https://github.com/golang/go/wiki/CodeReviewComments#receiver-type[Guidelines on choosing appropriate receiver types for methods]

=== Standards

 * Go Best Practices - Simplicity - https://golang.org/doc/effective_go#simplicity[Go's philosophy emphasizing simplicity and avoiding unnecessary abstractions]
