== How to fix it

=== Code examples

include::../../common/fix/code-rationale.adoc[]

==== Noncompliant code example

[source,vbnet,diff-id=1,diff-type=noncompliant]
----
Imports System.Text.RegularExpressions
			
Public Module Example
    Public Sub Validate(Input As String)
        Dim rx As New Regex("^(a+)+$", RegexOptions.None) ' Noncompliant
        Dim matches as MatchCollection = rx.Matches(Input)
    End Sub
End Module
----

[source,vbnet,diff-id=2,diff-type=noncompliant]
----
Imports System.Text.RegularExpressions
				
Public Module Example
    Public Sub Validate(Input As String)
        Dim Matches as Boolean = Regex.IsMatch(Input, "^(a+)+$", RegexOptions.None) ' Noncompliant
    End Sub
End Module
----

==== Compliant solution

[source,vbnet,diff-id=1,diff-type=compliant]
----
Imports System
Imports System.Text.RegularExpressions
			
Public Module Example
    Public Sub Validate(Input As String)
        Dim rx As New Regex("^(a+)+$", RegexOptions.None, TimeSpan.FromMilliseconds(100))
        Dim matches as MatchCollection = rx.Matches(Input)
    End Sub
End Module
----

[source,vbnet,diff-id=2,diff-type=compliant]
----
Imports System
Imports System.Text.RegularExpressions
				
Public Module Example
    Public Sub Validate(Input As String)
        Dim Matches as Boolean = Regex.IsMatch(Input, "^(a+)+$", RegexOptions.NonBacktracking, TimeSpan.FromMilliseconds(100))
    End Sub
End Module
----

=== How does this work?

It is recommended to set a time-out value in all regular expression
pattern-matching operations to avoid potential cases of catastrophic
backtracking. This can be done by specifying a time span through the
`matchTimeout` argument as shown in the compliant solutions, or by specifying a
default time-out with the `REGEX_DEFAULT_MATCH_TIMEOUT` property:

[source,vbnet]
----
AppDomain.CurrentDomain.SetData("REGEX_DEFAULT_MATCH_TIMEOUT", TimeSpan.FromMilliseconds(100))
----

Additionally, starting from .NET 7, `RegexOptions.NonBacktracking` can be
specified to enable matching using an approach that avoids backtracking and
guarantees linear-time processing in the length of the input.
