<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when environment (exported) variables don&#8217;t use ALL_CAPS naming or when local variables don&#8217;t use lower_case naming.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Shell scripts benefit from consistent variable naming conventions that immediately communicate the scope and purpose of variables.</p>
</div>
<div class="paragraph">
<p>Environment variables are typically shared across processes and should stand out in code. The ALL_CAPS convention makes exported variables easily recognizable and follows long-established Unix traditions. This helps developers quickly identify which variables affect the broader system environment.</p>
</div>
<div class="paragraph">
<p>Local variables, on the other hand, are used within specific functions or script sections. Using lower_case naming distinguishes them from environment variables and makes the code&#8217;s intent clearer. This separation prevents confusion about variable scope and reduces the risk of accidentally overriding important environment variables.</p>
</div>
<div class="paragraph">
<p>Inconsistent naming conventions make scripts harder to read and maintain. When variables don&#8217;t follow expected patterns, developers must spend extra time understanding the code&#8217;s structure and may make incorrect assumptions about variable behavior.</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="paragraph">
<p>Inconsistent variable naming reduces code readability and can lead to maintenance issues. Developers may accidentally override environment variables or misunderstand variable scope, potentially causing unexpected behavior in scripts or system configurations.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Use ALL_CAPS for exported environment variables and lower_case for local variables. This follows established shell scripting conventions and improves code clarity.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-shell" data-lang="shell">export my_path="/usr/bin"  # Noncompliant
local MyVar="value"        # Noncompliant</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-shell" data-lang="shell">export MY_PATH="/usr/bin"
local my_var="value"</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Bash Best Practices - Variables - <a href="https://bertvv.github.io/cheat-sheets/Bash.html">Comprehensive guide to Bash scripting best practices including variable naming conventions</a></p>
</li>
<li>
<p>Google Shell Style Guide - <a href="https://google.github.io/styleguide/shellguide.html">Google&#8217;s style guide for shell scripting with detailed naming conventions</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>