<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when a dictionary unpacking operation is performed on an invalid mapping object.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Dictionary unpacking allows you to pass the key-value pairs of a dictionary as keyword arguments to a function or merge dictionaries:</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">def foo(a, b):
    print(a+b)

my_dict = {"a": 1, "b": 2}
foo(**my_dict) # will print 3</code></pre>
</div>
</div>
<div class="paragraph">
<p>Dictionary unpacking requires an object with methods <code>__getitem__</code> and <code>keys</code> or <code>__getitem__</code> and <code>__getattr__</code>. This is the case for any <a href="https://docs.python.org/3/glossary.html#term-mapping">mapping</a> object such as <code>dict</code>. Using an object which doesn&#8217;t have these methods will raise a <code>TypeError</code>.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">class A:
    pass

{'a': 10, 'b': 20, **A()}  # Noncompliant</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">class A:
  def __getitem__(self, key):
      return 2

  def keys(self):
      return ['1','2','3']

{'a': 10, 'b': 20, **A()}  # =&gt; {'a': 10, 'b': 20, '1': 2, '2': 2, '3': 2}</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Python Documentation - <a href="https://docs.python.org/3/glossary.html#term-mapping">mapping</a></p>
</li>
</ul>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>X is of type Y and cannot be unpacked with "**". Use a "mapping" object instead.</p>
</div>
</div>
</div>
</div>