include::../description.adoc[]

== Noncompliant Code Example

----
for ($i = 0; $i < 10; $i++) {
  echo $i;
  if(condition) {
    $i = 20;
  }
}
----

== Compliant Solution

----
for ($i = 0; $i < 10; $i++) {
  echo $i;
}
----

ifdef::rspecator-view[]
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::rspecator-view[]
