== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,php]
----
for ($i = 0; $i < 10; $i++) {
  echo $i;
  if(condition) {
    $i = 20;
  }
}
----

=== Compliant solution

[source,php]
----
for ($i = 0; $i < 10; $i++) {
  echo $i;
}
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../parameters.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
