== Why is this an issue?

The Dockerfile instructions are not case-sensitive. However, they should be in upper case to distinguish them from arguments more easily. Shared coding conventions allow teams to collaborate effectively.

=== Noncompliant code example

[source,docker]
----
from ubuntu:22.04 as jammy
----

=== Compliant solution

[source,docker]
----
FROM ubuntu:22.04 AS jammy
----
