<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Using <code>Type.Assembly</code> to get the current assembly is nearly free in terms of performance; it&#8217;s a simple property access. On the other hand, <code>Assembly.GetExecutingAssembly()</code> can take up to 30 times as long because it walks up the call stack to find the assembly.</p>
</div>
<div class="paragraph">
<p>Note that <code>Assembly.GetExecutingAssembly()</code> is different than <code>Type.Assembly</code> because it dynamically returns the assembly that contains the startup object of the currently executed application. For example, if executed from an application it will return the application assembly, but if executed from a unit test project it could return the unit test assembly. <code>Type.Assembly</code> always returns the assembly that contains the specified type.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-text" data-lang="text">public class Example
{
   public static void Main()
   {
      Assembly assem = Assembly.GetExecutingAssembly(); // Noncompliant
      Console.WriteLine("Assembly name: {0}", assem.FullName);
   }
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-text" data-lang="text">public class Example
{
   public static void Main()
   {
      Assembly assem = typeof(Example).Assembly; // Here we use the type of the current class
      Console.WriteLine("Assembly name: {0}", assem.FullName);
   }
}</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Replace this call to "Assembly.GetExecutingAssembly()" with "typeof(XXX).Assembly".</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p><code>Assembly.GetExecutingAssembly()</code></p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_9_mar_2017_165833_ann_campbell_wrote">on 9 Mar 2017, 16:58:33 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~amaury.leve] Assembly.GetExecutingAssembly() walks up the call stack. What does Type.Assembly do?</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_9_mar_2017_171600_amaury_levé_wrote">on 9 Mar 2017, 17:16:00 Amaury Levé wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] It is a simple property so nearly a cost-free access.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_10_mar_2017_151328_ann_campbell_wrote">on 10 Mar 2017, 15:13:28 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>I&#8217;ve reworded this [~amaury.leve]. Please double-check me for accuracy.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_10_mar_2017_153708_amaury_levé_wrote">on 10 Mar 2017, 15:37:08 Amaury Levé wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] LGTM</p>
</div>
</div>
</div>
</div>