== Why is this an issue?

include::description.adoc[]

=== Noncompliant code example

With the default regular expression ``++^[A-Z][a-zA-Z0-9]*$++``:

[source,text]
----
public interface myInterface {...} // Noncompliant
----


=== Compliant solution

[source,text]
----
public interface MyInterface {...}
----

