<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Sharing some naming conventions is a key point to make it possible for a team to efficiently collaborate.</p>
</div>
<div class="paragraph">
<p>This rule allows to check that all interface names match a provided regular expression.</p>
</div>
<div class="paragraph">
<p>The default configuration is the one recommended by Microsoft:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>Must start with an upper case 'I' character, e.g. IFoo</p>
</li>
<li>
<p>Followed by Pascal casing, starting with an upper case character, e.g. IEnumerable</p>
</li>
<li>
<p>Short abbreviations of 2 letters can be capitalized, e.g. IFooID</p>
</li>
<li>
<p>Longer abbreviations need to be lower cased, e.g. IFooHtml</p>
</li>
</ul>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="paragraph">
<p>With the default regular expression <code>^I([A-Z]{1,3}[a-z0-9]+)*([A-Z]{2})?$</code>:</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-vbnet" data-lang="vbnet">Interface Foo  ' Noncompliant
End Interface</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-vbnet" data-lang="vbnet">Interface IFoo ' Compliant
End Interface</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Rename this interface name to match the regular expression ${format}</p>
</div>
</div>
<div class="sect2">
<h3 id="_parameters">Parameters</h3>
<div class="sidebarblock">
<div class="content">
<div class="title">format</div>
<div class="paragraph">
<p><em>STRING</em></p>
</div>
<div class="listingblock">
<div class="content">
<pre>^I([A-Z]{1,3}[a-z0-9]+)*([A-Z]{2})?$</pre>
</div>
</div>
<div class="paragraph">
<p>Regular expression used to check the interface names against.</p>
</div>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_20_may_2013_131748_fabrice_bellingard_wrote">on 20 May 2013, 13:17:48 Fabrice Bellingard wrote:</h3>
<div class="paragraph">
<p>Implementation: http://jira.codehaus.org/browse/SONARJAVA-143</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_1_jul_2015_095931_evgeny_mandrikov_wrote">on 1 Jul 2015, 09:59:31 Evgeny Mandrikov wrote:</h3>
<div class="paragraph">
<p>In C&#43;&#43; everything is classes, so there is no clear definition of "interface", while there is definition for Objective-C, so RSPEC retargeted.</p>
</div>
</div>
</div>
</div>