<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>When the same condition is checked twice in quick succession, it is probably inefficient, and likely an error. Usually, the intent was to check a different condition.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-text" data-lang="text">if (a.length() &gt; 5) {
  //...
}
if (a.length() &gt; 5) {  // Noncompliant
  // ...
}</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_duplicates_s2760">duplicates: <a data-rspec-id="S2760" class="rspec-auto-link">S2760</a></h3>

</div>
<div class="sect2">
<h3 id="_on_25_mar_2015_202907_evgeny_mandrikov_wrote">on 25 Mar 2015, 20:29:07 Evgeny Mandrikov wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] IMO title "The conditional expressions of the 'if' operators situated alongside each other should not be identical" is better, because it is more precise - current one can be interpreted as targeting case of "if" followed by "while", but this is not the intention of this rule.</p>
</div>
</div>
</div>
</div>