== Why is this an issue?

include::../why-dotnet.adoc[]

== How to fix it

include::../how-dotnet.adoc[]

=== Code examples

==== Noncompliant code example

[source,csharp]
----
public class Program
{
    public void WriteMatrix(int[][] matrix) // Non-Compliant
    {
    }
}
----

==== Compliant solution

[source,csharp]
----
public class Matrix
{
    // ...
}

public class Program
{
    public void WriteMatrix(Matrix matrix) // Compliant
    {
    }
}
----

include::../resources-dotnet.adoc[]

include::../rspecator.adoc[]
