include::../description.adoc[]

== Noncompliant Code Example

[source,csharp]
----
public class Program
{
    public void WriteMatrix(int[][] matrix) // Non-Compliant
    {
    }
}
----

== Compliant Solution

[source,csharp]
----
public class Matrix
{
    // ...
}

public class Program
{
    public void WriteMatrix(Matrix matrix) // Compliant
    {
    }
}
----
ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

endif::env-github,rspecator-view[]
