The requirement for an ``++OTHERS++`` clause is defensive programming. The clause should either take appropriate action, or contain a suitable comment as to why no action is taken. 

== Noncompliant Code Example

----
CASE SY-INDEX.   // Noncompliant; missing WHEN OTHERS clause
  WHEN ONE.
    WRITE  'One'. 
  WHEN 2.
    WRITE   'Two'.
ENDCASE.
----

== Compliant Solution

----
CASE SY-INDEX. 
  WHEN ONE.
    WRITE  'One'. 
  WHEN 2.
    WRITE   'Two'.
  WHEN OTHERS. // Compliant
    WRITE 'Unexpected result'
ENDCASE.

CASE SY-INDEX.
  WHEN OTHERS.   // Compliant
    WRITE 'Unexpected result'
  WHEN ONE.
    WRITE  'One'. 
  WHEN 2.
    WRITE   'Two'.
ENDCASE.
----

include::../see.adoc[]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
