include::../../../shared_content/secrets/description.adoc[]

== Why is this an issue?

include::../../../shared_content/secrets/rationale.adoc[]

If an attacker gains access to a Figma personal access token, they might be able to compromise the data that is accessible to the linked Figma account. By doing so, it might be possible for confidential data to be leaked by the attacker.

=== What is the potential impact?

Below are some real-world scenarios that may occur when a malicious entity manages to retrieve a leaked Figma personal access token.

==== Compromise of business-critical data

An attacker can use a personal access token to gain unauthorized access to your company's Figma projects and designs. This can include confidential client data, proprietary design assets, or any other intellectual property stored in Figma. With unauthorized access, the attacker can download and share this sensitive data, potentially leading to data breaches, intellectual property theft, or other forms of unauthorized disclosure.

==== Unauthorized actions in Figma environment

With a leaked Figma personal access token, an attacker can perform various actions on behalf of your company within the Figma workspace. This can include creating, modifying, or deleting projects, files, or team members. By impersonating authorized users, the attacker can manipulate your company's design assets or disrupt the design workflow. This can result in unauthorized changes and data loss.

== How to fix it

include::../../../shared_content/secrets/fix/revoke.adoc[]

include::../../../shared_content/secrets/fix/vault.adoc[]

=== Code examples

==== Noncompliant code example

[source,python,diff-id=1,diff-type=noncompliant]
----
import requests

token = 'figd_OLDXZWOP4fxW4c9ER0xzxRda96M-f0eFwZpFQjHJ'  # Noncompliant
response = requests.get('https://api.figma.com/v1/me', headers={
    'X-FIGMA-TOKEN': token,
    'Content-Type': 'application/json'
})
----

==== Compliant solution

[source,python,diff-id=1,diff-type=compliant]
----
import requests

token = os.getenv('FIGMA_PERSONAL_ACCESS_TOKEN')
response = requests.get('https://api.figma.com/v1/me', headers={
    'X-FIGMA-TOKEN': token,
    'Content-Type': 'application/json'
})
----

//=== How does this work?

//=== Pitfalls

//=== Going the extra mile

== Resources

=== Documentation

Figma Developers - https://www.figma.com/developers/api#access-tokens[Access tokens]

include::../../../shared_content/secrets/resources/standards.adoc[]
