include::../description.adoc[]

== Noncompliant Code Example

[source,swift]
----
switch (variable) {
  case 0:
    doSomething();
  default:
    doSomethingElse();
}
----

== Compliant Solution

[source,swift]
----
if (variable == 0) {
  doSomething();
} else {
  doSomethingElse();
}
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
