== Why is this an issue?

``++CASE++`` can contain a ``++WHEN OTHERS++`` clause for various reasons: to handle unexpected values, to show that all the cases were properly considered.

For readability purpose, to help a developer to quickly find the default behavior of a ``++CASE++`` statement, it is recommended to put the ``++WHEN OTHERS++`` clause at the end of the ``++CASE++`` statement. This rule raises an issue if the ``++WHEN OTHERS++`` clause is not the last one of the ``++CASE++``'s cases.

=== Noncompliant code example

[source,abap]
----
CASE SY-INDEX.
  WHEN OTHERS.   // Noncompliant; WHEN OTHERS should be last statement
    WRITE 'Unexpected result'
  WHEN ONE.
    WRITE  'One'. 
  WHEN 2.
    WRITE   'Two'.
ENDCASE.
----

=== Compliant solution

[source,abap]
----
CASE SY-INDEX. 
  WHEN ONE.
    WRITE  'One'. 
  WHEN 2.
    WRITE   'Two'.
  WHEN OTHERS. // Compliant
    WRITE 'Unexpected result'
ENDCASE.
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
