This rule raises an issue when multiple related variables are declared at package level using separate `var` statements instead of a grouped declaration block.

== Why is this an issue?

In Go, variable declarations can be "factored" into blocks using parentheses, similar to import statements. This is considered idiomatic Go style and provides several benefits.

When you declare multiple related variables at package level using separate `var` statements, the code becomes harder to read and maintain. Each variable appears isolated, making it difficult to understand which variables are conceptually related or serve similar purposes.

Grouped variable declarations improve code organization by:

* Visually grouping related variables together
* Reducing visual clutter from repeated `var` keywords
* Making it easier to scan and understand the package's global state
* Following Go's established conventions for factoring declarations

This pattern is especially important at package level where variables often represent configuration, state, or related data that should be understood as a cohesive unit.

=== What is the potential impact?

This issue affects code maintainability and readability. While it doesn't cause functional problems, scattered variable declarations make the codebase harder to navigate and understand, especially for new team members or when reviewing code changes.

== How to fix it

Group related package-level variables into a single declaration block using parentheses. This creates a clear visual grouping and follows Go's idiomatic style.

=== Code examples

==== Noncompliant code example

[source,go,diff-id=1,diff-type=noncompliant]
----
package main

var name string
var age int
var active bool // Noncompliant

func main() {
}
----

==== Compliant solution

[source,go,diff-id=1,diff-type=compliant]
----
package main

var (
    name   string
    age    int
    active bool
)

func main() {
}
----

== Resources

=== Documentation

 * A Tour of Go - Basic Types - https://go.dev/tour/basics/11[Official Go tutorial explaining that variable declarations may be factored into blocks]

 * A Tour of Go - Variables - https://go.dev/tour/basics/8[Introduction to Go variable declarations and package-level variables]

 * Effective Go - Names - https://go.dev/doc/effective_go#names[Go's official style guide for naming and organizing code elements]
