<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when multiple related variables are declared at package level using separate <code>var</code> statements instead of a grouped declaration block.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>In Go, variable declarations can be "factored" into blocks using parentheses, similar to import statements. This is considered idiomatic Go style and provides several benefits.</p>
</div>
<div class="paragraph">
<p>When you declare multiple related variables at package level using separate <code>var</code> statements, the code becomes harder to read and maintain. Each variable appears isolated, making it difficult to understand which variables are conceptually related or serve similar purposes.</p>
</div>
<div class="paragraph">
<p>Grouped variable declarations improve code organization by:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>Visually grouping related variables together</p>
</li>
<li>
<p>Reducing visual clutter from repeated <code>var</code> keywords</p>
</li>
<li>
<p>Making it easier to scan and understand the package&#8217;s global state</p>
</li>
<li>
<p>Following Go&#8217;s established conventions for factoring declarations</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>This pattern is especially important at package level where variables often represent configuration, state, or related data that should be understood as a cohesive unit.</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="paragraph">
<p>This issue affects code maintainability and readability. While it doesn&#8217;t cause functional problems, scattered variable declarations make the codebase harder to navigate and understand, especially for new team members or when reviewing code changes.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Group related package-level variables into a single declaration block using parentheses. This creates a clear visual grouping and follows Go&#8217;s idiomatic style.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-go" data-lang="go">package main

var name string
var age int
var active bool // Noncompliant

func main() {
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-go" data-lang="go">package main

var (
    name   string
    age    int
    active bool
)

func main() {
}</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>A Tour of Go - Basic Types - <a href="https://go.dev/tour/basics/11">Official Go tutorial explaining that variable declarations may be factored into blocks</a></p>
</li>
<li>
<p>A Tour of Go - Variables - <a href="https://go.dev/tour/basics/8">Introduction to Go variable declarations and package-level variables</a></p>
</li>
<li>
<p>Effective Go - Names - <a href="https://go.dev/doc/effective_go#names">Go&#8217;s official style guide for naming and organizing code elements</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>