<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p><strong>This rule is deprecated, and will eventually be removed.</strong></p>
</div>
<div class="paragraph">
<p>Component parameters can only receive query parameter values in routable components with an @page directive.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p><a href="https://learn.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.components.supplyparameterfromqueryattribute">SupplyParameterFromQuery</a> attribute is used to specify that a component parameter of a routable component comes from the <a href="https://en.wikipedia.org/wiki/Query_string">query string</a>.</p>
</div>
<div class="paragraph">
<p>In the case of non-routable components, the <code>SupplyParameterFromQuery</code> does not contribute to the functionality, and removing it will not affect the behavior.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Either make the component routable or remove the <code>SupplyParameterFromQuery</code> attribute.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-csharp" data-lang="csharp">&lt;h3&gt;Component&lt;/h3&gt;

@code {
    [Parameter]
    [SupplyParameterFromQuery]  // Noncompliant
    public bool Param { get; set; }
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-csharp" data-lang="csharp">@page "/component"

&lt;h3&gt;Component&lt;/h3&gt;

@code {
    [Parameter]
    [SupplyParameterFromQuery]  // Compliant
    public bool Param { get; set; }
}</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Microsoft Learn - <a href="https://learn.microsoft.com/en-us/aspnet/core/blazor/fundamentals/routing#query-strings">Query strings</a></p>
</li>
<li>
<p>Microsoft Learn - <a href="https://learn.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.components.supplyparameterfromqueryattribute">SupplyParameterFromQueryAttribute Class</a></p>
</li>
<li>
<p>Wikipedia - <a href="https://en.wikipedia.org/wiki/Query_string">query string</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>