== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,csharp]
----
class Foo 
{
  public int myField;

  public void DoSomething() 
  {
    int myField = 0;  // Noncompliant
    ...
  }
}
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Rename "${elementName}" which hides the (field|property) with the same name.


include::../highlighting.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
