include::../description.adoc[]

== Noncompliant Code Example

----
class Foo 
{
  public int myField;

  public void DoSomething() 
  {
    int myField = 0;  // Noncompliant
    ...
  }
}
----

include::../see.adoc[]

ifdef::env-github,rspecator-view[]
'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
