include::../description.adoc[]

== Noncompliant Code Example

----
CREATE TABLE employee
(
  employee_id INTEGER NOT NULL,
  first_name VARCHAR2(42) NOT NULL,
  last_name VARCHAR2(42) NOT NULL  
);
----

== Compliant Solution

----
CREATE TABLE employee
(
  employee_id INTEGER PRIMARY KEY,
  first_name VARCHAR2(42) NOT NULL,
  last_name VARCHAR2(42) NOT NULL  
);
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
