== Why is this an issue?

An unordered pair is easy to handle, but the longer an unordered set gets, the harder it is to deal with.


This rule raises an issue when a tuple of more than the allowed number of elements is returned.


=== Noncompliant code example

With the default threshold of 2

[source,text]
----
func doTheThing() -> (Int, Int, Int) {}  // Noncompliant
----


=== Compliant solution

[source,text]
----
func doTheThing() -> MyIntStruct {}
----

