An unordered pair is easy to handle, but the longer an unordered set gets, the harder it is to deal with.


This rule raises an issue when a tuple of more than the allowed number of elements is returned.


== Noncompliant Code Example

With the default threshold of 2

----
func doTheThing() -> (Int, Int, Int) {}  // Noncompliant
----


== Compliant Solution

----
func doTheThing() -> MyIntStruct {}
----

