<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Instances of classes that do not derive from the "Throwable" interface cannot be used in a PHP "throw" statement. Thus, it does not make sense to try to catch such objects within a "try-catch" block.</p>
</div>
<div class="paragraph">
<p>Many built-in exceptions such as "Exception" and the SPL exception classes do implement the "Throwable" interface and can be extended when creating custom exceptions.</p>
</div>
<div class="paragraph">
<p>This rule raises an issue when the classes used to specify the type of objects to be caught in a "try-catch" block do not derive from "Throwable" .</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-php" data-lang="php">class NoThrowable {}

try {
    foo();
} catch (NoThrowable $e) { // Noncompliant
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-php" data-lang="php">&lt;?php

class SomeThrowable implements Throwable {
    // Implementation of the Throwable methods
}

try {
    foo();
} catch (SomeThrowable $e) { // Compliant
}

class SomeCustomException extends Exception {}

try {
    foo();
} catch (SomeCustomException $e) { // Compliant
}</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Change this type to be a class deriving from "Throwable".</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="ulist">
<ul>
<li>
<p>Primary: The caught class name</p>
</li>
<li>
<p>Secondary: The name of the class in its class definition.</p>
<div class="ulist">
<ul>
<li>
<p>message: 'Class definition.'</p>
</li>
</ul>
</div>
</li>
</ul>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_relates_to_s5632">relates to: <a data-rspec-id="S5632" class="rspec-auto-link">S5632</a></h3>

</div>
<div class="sect2">
<h3 id="_on_5_mar_2020_141755_nicolas_harraudeau_wrote">on 5 Mar 2020, 14:17:55 Nicolas Harraudeau wrote:</h3>
<div class="paragraph">
<p>Even if it is possible to catch a class which is not a BaseException in python 2, this rule is still a bug for the following reasons:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>BaseException exists in Python 2 too. There is no reason not to use it. It is advised to use exceptions derived from BaseException since python 2.5 released in 2006.</p>
</li>
<li>
<p>Python 2 is not maintained anymore and there are tools enabling developpers to migrate from python 2 to python 3. When developers migrate this will become a bug.</p>
</li>
<li>
<p>We already have <a data-rspec-id="S5632" class="rspec-auto-link">RSPEC-5632</a> as a Bug and nobody complains. Letâ€™s remain consistent. If we change the rule to code smell we have to change this other rule too.</p>
</li>
<li>
<p>old style classes are the only exception in python 2. Trying to raise a string or a list will fail in in both python 2 and python 3. This exception would be a corner case.</p>
</li>
</ul>
</div>
</div>
</div>
</div>