<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>The TypeScript programming language supports <em>generics</em>, a programming construct for creating reusable components, that is, components that can work over various types rather than a single one. Sometimes, we need to limit this genericity to a specific set of types, typically when we know these types share common capabilities, e.g., a <code>length</code> property. To this end, the language provides the <code>extends</code> clause to describe type constraints on type parameters, whether for classes, interfaces, type aliases, or functions.</p>
</div>
<div class="paragraph">
<p>By default, a type parameter extends the <code>any</code> type. It is therefore redundant to explicitly extend from <code>any</code> and should be removed accordingly.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Fixing such an issue involves removing the redundant type constraint to <code>any</code>.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-typescript" data-lang="typescript">class C&lt;T extends any&gt; {
    // ...
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-typescript" data-lang="typescript">class C&lt;T&gt; {
    // ...
}</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p><a href="https://typescript-eslint.io/">typescript-eslint</a> - Rule <a href="https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/docs/rules/no-unnecessary-type-constraint.mdx">no-unnecessary-type-constraint</a></p>
</li>
<li>
<p>TypeScript Documentation - <a href="https://www.typescriptlang.org/docs/handbook/2/generics.html">Generics</a></p>
</li>
<li>
<p>TypeScript Documentation - <a href="https://www.typescriptlang.org/docs/handbook/2/generics.html#generic-constraints">Generic Constraints</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>