<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>The <code>nodiscard</code> attribute can be used with or without explanations, but explaining why a result should not be discarded can only improve oneâ€™s understanding of the code, and would prevent developers from wasting time figuring those things out by themselves.</p>
</div>
<div class="paragraph">
<p>This rule raises an issue when <code>nodiscard</code> is used on function without any explanation.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">[[nodiscard]] std::vector&lt;int&gt; generateRandomValues(int count); // Noncompliant

generateRandomValues(100);</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">[[nodiscard("Computation of values is expensive")]] std::vector&lt;int&gt; generateRandomValues(int count);

generateRandomValues(100);</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_is_related_to_s6222">is related to: <a data-rspec-id="S6222" class="rspec-auto-link">S6222</a></h3>

</div>
</div>
</div>