include::../description.adoc[]


== Noncompliant Code Example

----
"cc̈d̈d".replaceAll("[c̈d̈]", "X"); // Noncompliant, print "XXXXXX" instead of expected "cXXd".
----


== Compliant Solution

----
"cc̈d̈d".replaceAll("c̈|d̈", "X"); // print "cXXd"
----
