include::../description.adoc[]


== Noncompliant Code Example

[source,java]
----
"cc̈d̈d".replaceAll("[c̈d̈]", "X"); // Noncompliant, print "XXXXXX" instead of expected "cXXd".
----


== Compliant Solution

[source,java]
----
"cc̈d̈d".replaceAll("c̈|d̈", "X"); // print "cXXd"
----
ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../highlighting.adoc[]

endif::env-github,rspecator-view[]
