This rule checks that the disallowed tag libraries are not used.

== Noncompliant Code Example

Using the default parameter for this rule, "http://java.sun.com/jstl/sql":

----
<%@ taglib uri="http://java.sun.com/jstl/sql" prefix="prefixOfTag" > <!-- Noncompliant -->
<jsp:directive.taglib uri="http://java.sun.com/jstl/sql" prefix="prefixOfTag" /> <!-- Noncompliant -->
----
