== Why is this an issue?

This rule checks that the disallowed tag libraries are not used.


=== Noncompliant code example

Using the default parameter for this rule, "http://java.sun.com/jstl/sql":

[source,html]
----
<%@ taglib uri="http://java.sun.com/jstl/sql" prefix="prefixOfTag" > <!-- Noncompliant -->
<jsp:directive.taglib uri="http://java.sun.com/jstl/sql" prefix="prefixOfTag" /> <!-- Noncompliant -->
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Remove the use of "xxx".


=== Parameters

.tagLibs
****

----
http://java.sun.com/jstl/sql
----

Comma-separated list of URIs of disallowed taglibs
****


endif::env-github,rspecator-view[]
