<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Name space and scope are different. This rule is not concerned with scope.</p>
</div>
<div class="paragraph">
<p>ISO C defines a number of different name spaces (see ISO 9899:1990 6.1.2.3). It is technically possible to use the same name in separate name spaces to represent completely different items. However this practice is deprecated because of the confusion it can cause, and so names should not be reused, even in separate name spaces.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">struct { int16_t key; int16_t value; } record;
int16_t value; // Noncompliant, 2nd use of "value"
record.key = 1;
value = 0; // should have been record.value

struct device_q { struct device_q *next; /* ... */ } // Compliant, since "next" must be prefixed.
devices[N_DEVICES];
struct task_q { struct task_q *next; /* ... */ } // Compliant, since "next" must be prefixed.
tasks[N_TASKS];
devices[0].next = &amp;devices[1];
tasks[0].next = &amp;tasks[1];</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">struct { int16_t key; int16_t value; } record;
int16_t val; // Compliant
record.key = 1;
value = 0; // Compile time error</code></pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p>MISRA C:2004, 5.6</p>
</li>
</ul>
</div>
<hr>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>The name "xxx" collides with a previous declaration and should be renamed.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_is_related_to_s808">is related to: <a data-rspec-id="S808" class="rspec-auto-link">S808</a></h3>

</div>
<div class="sect2">
<h3 id="_on_21_oct_2014_182210_ann_campbell_wrote">on 21 Oct 2014, 18:22:10 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~samuel.mercier] please</p>
</div>
<div class="ulist">
<ul>
<li>
<p>combine the code samples in the Noncompliant Code Example</p>
</li>
<li>
<p>move the code commentary into the code in the form of comments if needed</p>
</li>
<li>
<p>reserve "See also" for something that&#8217;s relevant but tangential</p>
</li>
<li>
<p>use h3 for See also, rather than h2</p>
</li>
<li>
<p>include ISO references in the main "See" list.</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>In fact, I would say that if you leave the ISO citations in the text, you don&#8217;t need to include them in the references (See) section, and conversely if you want to list them in the references you should remove them from the text.</p>
</div>
</div>
</div>
</div>