== Why is this an issue?

The purpose of an abstract class is to provide some heritable behaviors while also defining methods which must be implemented by sub-classes.

A `class` with only `abstract` methods and no inheritable behavior should be converted to an `interface`.

=== Noncompliant code example

[source,csharp]
----
public abstract class Animal // Noncompliant; should be an interface
{
  abstract void Move();
  abstract void Feed();
}
----

=== Compliant solution

[source,csharp]
----
public interface Animal
{
  void Move();
  void Feed();
}

public abstract class Lamp
{
  private bool switchLamp = false;

  public abstract void Glow();

  public void FlipSwitch()
  {
    switchLamp = !switchLamp;
    if (switchLamp)
    {
      Glow();
    }
  }
}
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Convert this "abstract" (class|record) to an interface.


'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
