include::../description.adoc[]

== Why is this an issue?
To use these characters as what they match themselves you can use metacharacters escaping, but it makes it harder to read and understand the regular expression.

== How to fix it
Instead of escaping it is better to wrap those characters in character class.

Note: not all the metacharacters could be wrapped in character class.

List of metacharacters could be wrapped:
[source]
----
. $ * + ? { } \ | ( )
----
List of metacharacters couldn't be wrapped without escaping
[source]
----
[ ] ^
----
=== Code examples

==== Noncompliant code example

[source,python]
----
import re
pattern = re.compile("\d+\.\d+", re.DOTALL)
----

==== Compliant solution

[source,python]
----
import re
pattern = re.compile("\d+[.]\d+", re.DOTALL)
----

include::../implementation.adoc[]
