<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>To ensure future code portability, obsolete keywords should not be used. The following keywords were declared obsolete in the COBOL ANSI-85 standard and removed in the ISO/IEC 1989:2002 standard:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>Paragraphs: <code>AUTHOR</code>, <code>INSTALLATION</code>, <code>DATE-WRITTEN</code>, <code>DATE-COMPILED</code>, <code>SECURITY</code></p>
</li>
<li>
<p>Clauses: <code>DATA RECORD(S)</code>, <code>LABEL RECORD(S)</code>, <code>MEMORY SIZE</code>, <code>MULTIPLE FILE (TAPE)</code>, <code>RERUN</code>, <code>VALUE OF</code>, <code>CODE SEGMENT-LIMIT</code></p>
</li>
<li>
<p>Statements: <code>ALTER</code>, <code>ENTER</code>, <code>STOP</code> <em>literal</em>, <code>GO TO</code> without an argument</p>
</li>
<li>
<p>Phrases: <code>REVERSED</code> phrase of the <code>OPEN</code> statement</p>
</li>
<li>
<p>Special registers: <code>DEBUG-ITEM</code></p>
</li>
<li>
<p>Sections: Debugging sections</p>
</li>
<li>
<p>Declarative: <code>USE FOR DEBUGGING</code></p>
</li>
</ul>
</div>
<div class="paragraph">
<p>The following keywords were declared obsolete in the ISO/IEC 1989:2002 standard:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>Phrase: <code>DEBUGGING MODE</code></p>
</li>
<li>
<p>Clause: <code>PADDING CHARACTER</code></p>
</li>
</ul>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cobol" data-lang="cobol">IDENTIFICATION DIVISION.
PROGRAM-ID.  AcceptAndDisplay.
AUTHOR.  Michael Coughlan.  *&gt; Noncompliant</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cobol" data-lang="cobol">IDENTIFICATION DIVISION.
PROGRAM-ID.  AcceptAndDisplay.</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>"XXX" is obsolete.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_8_dec_2014_142450_pierre_yves_nicolas_wrote">on 8 Dec 2014, 14:24:50 Pierre-Yves Nicolas wrote:</h3>
<div class="paragraph">
<p>Other language elements were categorized as obsolete in COBOL 85 and removed in COBOL 2002. See:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>http://www-01.ibm.com/support/knowledgecenter/SSQ2R2_8.5.1/com.ibm.etools.cbl.win.doc/topics/rlpreobs.htm?lang=en</p>
</li>
<li>
<p>http://supportline.microfocus.com/documentation/books/sx22sp1/atobsl.htm</p>
</li>
</ul>
</div>
</div>
</div>
</div>