== Why is this an issue?

Never use ``++with++`` statements, since they decrease readability. When you do not specify a variable's scope, you do not always know where you are setting properties, so your code can be confusing.

=== Noncompliant code example

[source,flex]
----
with (foo) { // Noncompliant
  return x;  // is it a property of foo or local variable ?
}
----

=== Compliant solution

[source,flex]
----
return foo.x;
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

=== on 5 Jan 2015, 18:43:02 Ann Campbell wrote:
Note that this subtask variation of SQALE is an exception to the rule - both JavaScript and Flex plugins had implemented the rule with the same legacy key, but with different SQALE settings.

endif::env-github,rspecator-view[]
