== Why is this an issue?

Never use ``++with++`` statements, since they decrease readability. When you do not specify a variable's scope, you do not always know where you are setting properties, so your code can be confusing.

=== Noncompliant code example

[source,flex]
----
with (foo) { // Noncompliant
  return x;  // is it a property of foo or local variable ?
}
----

=== Compliant solution

[source,flex]
----
return foo.x;
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
