Never use <code>with</code> statements, since they decrease readability. When you do not specify a variable's scope, you do not always know where you are setting properties, so your code can be confusing.

== Noncompliant Code Example

----
with (foo) { // Noncompliant
  return x;  // is it a property of foo or local variable ?
}
----

== Compliant Solution

----
return foo.x;
----
