<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when Apex code lacks proper spacing around operators, after commas, or uses inconsistent indentation in nested blocks like if-else statements, loops, and try-catch blocks.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Consistent spacing and indentation are fundamental to code readability and maintainability. When code lacks proper formatting, it becomes difficult for developers to quickly understand the structure and logic.</p>
</div>
<div class="paragraph">
<p>Spacing around operators and after commas helps separate different parts of expressions, making them easier to parse visually. For example, <code>i=0;i&lt;10;i` is much harder to read than `i = 0; i &lt; 10; i</code>.</p>
</div>
<div class="paragraph">
<p>Proper indentation shows the hierarchical structure of nested code blocks. When if-else statements, loops, and other control structures are properly indented, developers can immediately see which statements belong to which block. This is especially important in complex nested structures where misaligned code can lead to confusion about the program flow.</p>
</div>
<div class="paragraph">
<p>Inconsistent formatting also makes code reviews more difficult, as reviewers spend time deciphering poorly formatted code instead of focusing on logic and potential issues. In team environments, consistent formatting standards help ensure that all developers can easily read and modify each other&#8217;s code.</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="paragraph">
<p>Poor formatting reduces code readability and makes maintenance more difficult. Developers spend extra time understanding poorly formatted code, which slows down development and increases the likelihood of introducing bugs during modifications. In team environments, inconsistent formatting can lead to unnecessary merge conflicts and reduced collaboration efficiency.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Add spaces around operators and after commas in expressions and statements.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-apex" data-lang="apex">for(i=0;i&lt;10;i++){
    System.debug("Value"+i);
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-apex" data-lang="apex">for (i = 0; i &lt; 10; i++) {
    System.debug("Value" + i);
}</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Apex Code Layouts and Formatting Best Practices - <a href="https://www.apexhours.com/code-layouts-and-formatting-salesforce">Comprehensive guide on Apex code formatting standards including spacing, indentation, and layout best practices</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>