<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>According to Hibernate&#8217;s documentation:</p>
</div>
<div class="quoteblock">
<blockquote>
<div class="paragraph">
<p>Hibernate&#8217;s own connection pooling algorithm is, &#8230;&#8203; quite rudimentary. It is intended to help you get started and is <em>not intended for use in a production system</em>, or even for performance testing. You should use a third party pool for best performance and stability.</p>
</div>
</blockquote>
</div>
<div class="paragraph">
<p>This rule raises an issue when a <code>hibernate.connection.pool_size</code> value is found in a <em>hibernate.cfg.xml</em> or <em>hibernate.properties</em> file.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">&lt;hibernate-configuration&gt;
  &lt;session-factory&gt;
    &lt;!-- ... --&gt;
    &lt;property name="connection.pool_size"&gt;10&lt;/property&gt;  &lt;!-- Noncompliant --&gt;
  &lt;/session-factory&gt;
&lt;/hibernate-configuration&gt;</code></pre>
</div>
</div>
<div class="paragraph">
<p>or</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">// ...
hibernate.connection.pool_size=10  // Noncompliant</code></pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p><a href="https://docs.jboss.org/hibernate/orm/3.3/reference/en-US/html/session-configuration.html#configuration-hibernatejdbc">Hibernate, Session Configuration</a> - JDBC Connections</p>
</li>
</ul>
</div>
<hr>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Configure a third-party connection pool.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>primary:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>property name and value for xml</p>
</li>
<li>
<p>whole line for property</p>
</li>
</ul>
</div>
</div>
</div>
</div>