<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>In most databases, the SQL <code>TRUNCATE</code> statement has the following side effects:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>The table being truncated is locked during the operation</p>
</li>
<li>
<p>The transaction log is not updated during the operation, which means that a TRUNCATE operation can&#8217;t be part of a transaction.</p>
</li>
<li>
<p><code>DELETE</code> triggers are not fired during truncation</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>For these reasons use of the <code>TRUNCATE</code> statement may not be permitted in some environments. This rule flags all uses of the <code>TRUNCATE</code> statement.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-text" data-lang="text">EXEC SQL TRUNCATE MY_TABLE END-EXEC.</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-text" data-lang="text">EXEC SQL DELETE FROM MY_TABLE END-EXEC.</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_14_may_2014_093240_freddy_mallet_wrote">on 14 May 2014, 09:32:40 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>Waiting for additional explanations from BNP</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_27_may_2014_130331_ann_campbell_wrote">on 27 May 2014, 13:03:31 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>I&#8217;ve expanded this one slightly - found a third reason not to truncate online and added a compliant solution. Please double-check me.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_10_jul_2014_072839_freddy_mallet_wrote">on 10 Jul 2014, 07:28:39 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>We&#8217;ve finally decided to not support this controversial rule.</p>
</div>
</div>
</div>
</div>