== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,java]
----
void foo() {
  outer: //label is not used.
  for(int i = 0; i<10; i++) {
    break;
  }
}
----

=== Compliant solution

[source,java]
----
void foo() {
  for(int i = 0; i<10; i++) {
    break;
  }
}
----

== Resources

* https://wiki.sei.cmu.edu/confluence/x/5dUxBQ[CERT, MSC12-C.] - Detect and remove code that has no effect or is never executed

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
