== Why is this an issue?

While it's perfectly legal to give a class the same simple name as a class in another package that it extends or interface it implements, it's confusing and could cause problems in the future. 


=== Noncompliant code example

[source,java]
----
package my.mypackage;

public class Foo implements a.b.Foo { // Noncompliant
----


=== Compliant solution

[source,java]
----
package my.mypackage;

public class FooJr implements a.b.Foo {
----



ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Rename this class.


'''
== Comments And Links
(visible only on this page)

=== on 21 Oct 2014, 14:08:46 Nicolas Peru wrote:
LGTM

endif::env-github,rspecator-view[]
