FIXME: add a description

// If you want to factorize the description uncomment the following line and create the file.
//include::../description.adoc[]

== Why is this an issue?

This rule is not really a rule, but a demonstration of the features from Asciidoc that can appear in a rule description.

More specifically, its "How to fix it" section contains several frameworks.


== How to fix it in Razor

=== Code examples

Some text for the Razor framework. Should be displayed in its own subtab.

==== Noncompliant code example

[source,text,diff-id=1,diff-type=noncompliant]
----
FIXME
----

==== Compliant solution

[source,text,diff-id=1,diff-type=compliant]
----
FIXED
----

== How to fix it in Python Standard Library

=== Code examples

Some text for the Python Standard Library framework. Should be displayed in its own subtab.

==== Noncompliant code example

[source,text,diff-id=1,diff-type=noncompliant]
----
FIXME
----

==== Compliant solution

[source,text,diff-id=1,diff-type=compliant]
----
FIXED
----
