== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,javascript]
----
function computeDurationInMilliseconds(hours, minutes, seconds) {
  let duration = (((hours * 60) + minutes) * 60 + seconds ) * 1000;
  return duration;
}
----

=== Compliant solution

[source,javascript]
----
function computeDurationInMilliseconds(hours, minutes, seconds) {
  return (((hours * 60) + minutes) * 60 + seconds ) * 1000;
}
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
