<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Most letters and characters simply match themselves. However, there are some characters called metacharacters, that donâ€™t match themselves. Instead, they indicate that some pattern should be matched, or they repeat or change parts of the regular expression.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>To use these characters as what they match themselves you can use metacharacters escaping, but it makes it harder to read and understand the regular expression.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Instead of escaping it is better to wrap those characters in character class.</p>
</div>
<div class="paragraph">
<p>Note: not all the metacharacters could be wrapped in character class.</p>
</div>
<div class="paragraph">
<p>List of metacharacters could be wrapped:</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code>. $ * + ? { } \ | ( )</code></pre>
</div>
</div>
<div class="paragraph">
<p>List of metacharacters couldn&#8217;t be wrapped without escaping</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code>[ ] ^</code></pre>
</div>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">import re
pattern = re.compile("\d+\.\d+", re.DOTALL)</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">import re
pattern = re.compile("\d+[.]\d+", re.DOTALL)</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_message">Message</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Wrap metacharacter with class character.
(x should be dynamic and be the replacement we propose)</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_highlighting">Highlighting</h2>
<div class="sectionbody">
<div class="paragraph">
<p>The metacharacter with escaping character</p>
</div>
<hr>
</div>
</div>