<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Non-interactive HTML elements, such as <code>&lt;div&gt;</code> and <code>&lt;span&gt;</code>, are not designed to have event handlers. When these elements are given event handlers, it can lead to accessibility issues.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Attaching event handlers to non-interactive HTML elements can lead to significant accessibility issues. These elements, such as <code>&lt;div&gt;</code> and <code>&lt;span&gt;</code>, are not designed to interact with assistive technologies like screen readers, making it difficult for users with disabilities to navigate and interact with the website. Additionally, these elements may not be focusable or provide visual feedback when interacted with, resulting in a confusing and potentially frustrating user experience. Therefore, to maintain an accessible and user-friendly website, event handlers should be used exclusively with interactive elements.</p>
</div>
<div class="paragraph">
<p>The rule only considers the handlers <code>onClick</code>, <code>onMouseDown</code>, <code>onMouseUp</code>, <code>onKeyPress</code>, <code>onKeyDown</code>, and <code>onKeyUp</code>.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>To fix this issue, remove the event handler from the non-interactive element and attach it to an interactive element instead. If the element is not interactive, it should not have an event handler.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-js" data-lang="js">&lt;li onClick={() =&gt; void 0} /&gt;
&lt;div onClick={() =&gt; void 0} role="listitem" /&gt;</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-js" data-lang="js">&lt;div onClick={() =&gt; void 0} role="button" /&gt;
&lt;div onClick={() =&gt; void 0} role="presentation" /&gt;
&lt;input type="text" onClick={() =&gt; void 0} /&gt; // Interactive element does not require role.
&lt;button onClick={() =&gt; void 0} className="foo" /&gt; // button is interactive.
&lt;div onClick={() =&gt; void 0} role="button" aria-hidden /&gt; // This is hidden from the screenreader.</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p><a href="https://github.com/jsx-eslint/eslint-plugin-jsx-a11y">eslint-plugin-jsx-a11y</a> - Rule <a href="https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/blob/HEAD/docs/rules/no-noninteractive-element-interactions.md">no-noninteractive-element-interactions</a></p>
</li>
<li>
<p>WCAG - <a href="https://www.w3.org/TR/wai-aria-practices-1.1/#aria_ex">WAI-ARIA Authoring Practices Guide - Design Patterns and Widgets</a></p>
</li>
<li>
<p>MDN - <a href="https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Techniques/Using_the_button_role#Keyboard_and_focus">ARIA Techniques</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>