Code is clearest when each statement has its own line. Nonetheless, it is a common pattern to combine on the same line an ``++if++`` and its resulting _then_ statement. However, when an ``++if++`` is placed on the same line as the closing ``++}++`` from a preceding ``++else++`` or ``++elseif++``, it is either an error - ``++else++`` is missing - or the invitation to a future error as maintainers fail to understand that the two statements are unconnected.

== Noncompliant Code Example

----
if ($condition1) {
  // ...
} if ($condition2) {  // Noncompliant
  //...
}
----

== Compliant Solution

----
if ($condition1) {
  // ...
} elseif ($condition2) { 
  //...
}
----
Or

----
if ($condition1) {
  // ...
} 

if ($condition2) {
  //...
}
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../highlighting.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
