== Why is this an issue?

Returning ``++null++`` from a non-``++async++`` ``++Task++``/``++Task<T>++`` method will cause a ``++NullReferenceException++`` at runtime. This problem can be avoided by returning ``++Task.FromResult<T>(null)++`` instead.


=== Noncompliant code example

[source,csharp]
----
public Task<object> GetFooAsync()
{
    return null; // Noncompliant
}
----


=== Compliant solution

[source,csharp]
----
public Task<object> GetFooAsync()
{
    return Task.FromResult<object>(null);
}
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../highlighting.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
