include::../description.adoc[]

== Noncompliant Code Example

[source,swift]
----
do {
    try loadImage(imagePath)
} catch { // Noncompliant
    throw error
}
----

== Compliant Solution

[source,swift]
----
do {
    try loadImage(imagePath)
} catch {
    handleImageError(error)
    throw error
}
----
or

[source,swift]
----
try loadImage(imagePath)
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
