include::../description.adoc[]

== Noncompliant Code Example

[source,python]
----
r"a[b]c"
----

== Compliant Solution

[source,python]
----
r"abc"
----

include::../exceptions.adoc[]

include::../implementation.adoc[]
