== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,python]
----
r"a[b]c"
----

=== Compliant solution

[source,python]
----
r"abc"
----

include::../exceptions.adoc[]

include::../implementation.adoc[]
