== How to fix it in Apache Commons

=== Code examples

include::../../common/fix/code-rationale.adoc[]

==== Noncompliant code example

[source,kotlin,diff-id=1,diff-type=noncompliant]
----
@Controller
class ExampleController
{
    @GetMapping("/exec")
    fun exec(@RequestParam("command") command: String?) {
        val cmd = CommandLine(command)
        val executor = DefaultExecutor.Builder().get()
        executor.execute(cmd)
    }
}
----

==== Compliant solution

[source,kotlin,diff-id=1,diff-type=compliant]
----
@Controller
class ExampleController
{
    companion object {
        val ALLOWED_COMMANDS = arrayOf("/usr/bin/ls", "/usr/bin/cat")
    }

    @GetMapping("/exec")
    fun exec(@RequestParam("command") command: String?) {
        if (ALLOWED_COMMANDS.contains(command)) {
            val cmd = CommandLine(command)
            val executor = DefaultExecutor.Builder().get()
            executor.execute(cmd)
        }
    }
}
----

=== How does this work?

include::../../common/fix/introduction.adoc[]

include::../../common/fix/pre-approved-list.adoc[]

:sanitizationLib: org.apache.commons.exec.CommandLine.addArguments(String[] addArguments)

include::../../common/fix/sanitize-meta-characters.adoc[]

Here `{sanitizationLib}` takes care of escaping the passed arguments and
internally creates a single string given to the operating system to be
executed.
