== How to fix it in Java Lang Package

=== Code examples

include::../../common/fix/code-rationale.adoc[]

==== Noncompliant code example

[source,kotlin,diff-id=11,diff-type=noncompliant]
----
@Controller
class ExampleController
{
    @GetMapping("/exec")
    fun exec(@RequestParam("command") command: String?) {
        Runtime.getRuntime().exec(command)
    }
}
----

==== Compliant solution

[source,kotlin,diff-id=11,diff-type=compliant]
----
@Controller
class ExampleController
{
    companion object {
        val ALLOWED_COMMANDS = arrayOf("/usr/bin/ls", "/usr/bin/cat")
    }

    @GetMapping("/exec")
    fun exec(@RequestParam("command") command: String?) {
        if (ALLOWED_COMMANDS.contains(command)) {
            Runtime.getRuntime().exec(command)
        }
    }
}
----

++java.lang.Runtime++ is sometimes used over ++java.lang.ProcessBuilder++ due to ease of use. Flexibility in methods often introduces security issues as edge cases are easily missed. The compliant solution logic is also applied to ++java.lang.ProcessBuilder++.

=== How does this work?

include::../../common/fix/introduction.adoc[]

include::../../common/fix/pre-approved-list.adoc[]

:sanitizationLib: java.lang.Runtime.exec(cmdarray: Array<String>)

include::../../common/fix/sanitize-meta-characters.adoc[]

Here `{sanitizationLib}` takes care of escaping the passed arguments and
internally creates a single string given to the operating system to be
executed.
