<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>In Javadoc code snippets introduced with Java 18, attributes' values can be quoted with single-quote (') characters or double-quote (") characters. Simple values, such as identifiers or numbers need not be quoted. According to Javadoc&#8217;s specification, escape sequences are not supported in attribute value. Using them will therefore lead to unpredictable behaviors when parsing the snippet, at best, and "spurious markup" warnings being reported.</p>
</div>
<div class="paragraph">
<p>If there is a need to use double-quotes within the attribute already using double-quotes, consider using single-quotes to delimit the value instead of escaping quotes. Proceed the other way around in case of single-quote need.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">/**
 * {@snippet :
 *   public static void main(String... args) {
 *    System.out.println("Hello World!");  // @link regex="\".+\"" target="java.lang.String"
 *   }
 * }
 */
void foo();

/**
 * {@snippet :
 *   public static void main(String... args) {
 *    System.out.println('c');  // @link regex='\'.+\'' target="java.lang.Character"
 *   }
 * }
 */
void bar();</code></pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">/**
 * {@snippet :
 *   public static void main(String... args) {
 *    System.out.println("Hello World!");  // @link regex='".+"' target="java.lang.String"
 *   }
 * }
 */
void foo();

/**
 * {@snippet :
 *   public static void main(String... args) {
 *    System.out.println('c');  // @link regex="'.+'" target="java.lang.Character"
 *   }
 * }
 */
void bar();</code></pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_see">See</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Oracleâ€™s Programmer&#8217;s Guide to Snippets: <a href="https://docs.oracle.com/en/java/javase/18/code-snippet/index.html">https://docs.oracle.com/en/java/javase/18/code-snippet/index.html</a></p>
</div>
<hr>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="ulist">
<ul>
<li>
<p>In case of escaped double quotes: Use single-quotes to delimit the value instead of escaping double-quotes.</p>
</li>
<li>
<p>In case of escaped single quotes: Use double-quotes to delimit the value instead of escaping single-quotes.</p>
</li>
</ul>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="ulist">
<ul>
<li>
<p>Primary: attribute value including the quotes</p>
</li>
</ul>
</div>
</div>
<div class="sect2">
<h3 id="_quickfix">Quickfix</h3>
<div class="paragraph">
<p>Only possible if there is no mix of quotes and double quotes within the value (this should not compile on javadoc side anyway)</p>
</div>
<div class="ulist">
<ul>
<li>
<p>In case of escaped double quotes: Replace starting and ending double-quotes by single-quotes, and remove escapes from double-quotes within the value.</p>
</li>
<li>
<p>In case of escaped single quotes: Replace starting and ending single-quotes by double-quotes, and remove escapes from single-quotes within the value.</p>
</li>
</ul>
</div>
</div>
</div>
</div>