== Why is this an issue?

Shadowing fields or ``++enum++`` ``++case++``s with a local variable is a bad practice that reduces code readability: It makes it confusing to know whether the field or the variable is being used.

=== Noncompliant code example

[source,swift]
----
public class Foo {
  public var myField:Int = 0

  public func doSomething() {
    var myField = 0
    ...
  }
}
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Rename "${elementName}" which has the same name as the (field|case) declared at line {}.


=== Highlighting

* Primary: shadowing variable
* Secondary: shadowed field/case
** message: 'Shadowed field/case.'


'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
