== How to fix it in a Servlet

=== Code examples

include::../../common/fix/code-rationale.adoc[]

==== Noncompliant code example

[source,kotlin,diff-id=1,diff-type=noncompliant]
----
import javax.servlet.http.Cookie

fun doGet(request: HttpServletRequest, response: HttpServletResponse) {
    val cookieOpt = Arrays.stream(request.cookies)
        .filter { it.name == "jsessionid" }
        .findFirst()

    if (!cookieOpt.isPresent) {
        val cookie = request.getParameter("cookie")
        val cookieObj = Cookie("jsessionid", cookie) // Noncompliant
        response.addCookie(cookieObj)
    }
    response.sendRedirect("/welcome.jsp")
}
----

==== Compliant solution

[source,kotlin,diff-id=1,diff-type=compliant]
----
fun doGet(request: HttpServletRequest, response: HttpServletResponse) {
    val cookieOpt = Arrays.stream(request.cookies)
        .filter { it.name == "jsessionid" }
        .findFirst()

    if (!cookieOpt.isPresent) {
        response.sendRedirect("/getCookie.jsp")
    } else {
        response.sendRedirect("/welcome.jsp")
    }
}
----

include::../../common/fix/how-does-this-work.adoc[]
