== Why is this an issue?

Using keywords as variable names may yield incomprehensible code, and should be avoided.

=== Noncompliant code example

[source,abap]
----
DATA:  wa_struct TYPE struct,
       name       TYPE string,
       dob          TYPE string,
       aliases     TYPE string,    " ALIASES is a keyword
----

=== Compliant solution

[source,abap]
----
DATA:  wa_struct TYPE struct,
       name         TYPE string,
       dob            TYPE string,
       alt_names  TYPE string,
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
