== How to fix it in Java SE

=== Code examples

include::../../common/fix/code-rationale.adoc[]

==== Noncompliant code example

[source,java,diff-id=1,diff-type=noncompliant]
----
import java.lang.Class;
import java.lang.reflect.Constructor;
import java.lang.reflect.Method;

@RestController
public class EffectController
{
    @GetMapping(value = "/filter/apply")
    @ResponseBody
    public ResponseEntity<String> apply(@RequestParam("effect") String effectName)
    {
        try
        {
            Class effectClass                = Class.forName(effectName);  // Noncompliant
            Constructor<?> effectConstructor = effectClass.getConstructor();
            Object EffectObject              = effectConstructor.newInstance();
            Method applyMethod               = effectClass.getMethod("applyFilter");

            boolean result = (boolean) applyMethod.invoke(EffectObject);

        } catch (Exception e) {}

        if (result)
        {
            return new ResponseEntity<>("Filter Applied", HttpStatus.OK);
        }
        else
        {
            return new ResponseEntity<>("Filter Failure", HttpStatus.FORBIDDEN);
        }
    }
}
----

==== Compliant solution

[source,java,diff-id=1,diff-type=compliant]
----
import java.lang.Class;
import java.lang.reflect.Constructor;
import java.lang.reflect.Method;

@RestController
public class EffectController
{
    private static Set<String> EFFECT_ALLOW_LIST = new HashSet<>();

    static
    {
        allowList.add("SepiaEffect");
        allowList.add("BlackAndWhiteEffect");
        allowList.add("WaterColorEffect");
        allowList.add("OilPaintingEffect");
    }

    @GetMapping(value = "/filter/apply")
    @ResponseBody
    public ResponseEntity<String> apply(@RequestParam("effect") String effectName)
    {
        if (!EFFECT_ALLOW_LIST.contains(effectName)) {
            return new ResponseEntity<>("Filter Failure", HttpStatus.FORBIDDEN);
        }

        try
        {
            Class effectClass                = Class.forName(effectName);
            Constructor<?> effectConstructor = effectClass.getConstructor();
            Object EffectObject              = effectConstructor.newInstance();
            Method applyMethod               = effectClass.getMethod("applyFilter");

            boolean result = (boolean) applyMethod.invoke(EffectObject);

        } catch (Exception e) {}

        if (result) {
            return new ResponseEntity<>("Filter Applied", HttpStatus.OK);
        }
        else {
            return new ResponseEntity<>("Filter Failure", HttpStatus.FORBIDDEN);
        }
    }
}
----

=== How does this work?

include::../../common/fix/pre-approved-list.adoc[]

