<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>An invalid color definition may be interpreted incorrectly, which is likely to have unintended impacts on the expected look and feel of the UI. That&#8217;s why it&#8217;s important to use a 8-digit hexadecimal integer <code>(0xFFFFFFFF)</code> when creating a <code>Color</code> object with <code>Color(int)</code> constructor.
If only 6-digit hex value was used, the 2 leading digits will be considered <code>0</code> and interpreted as fully-transparent. Most likely this is not the desired behavior. To avoid such mistakes, either use a different constructor, or use a 8-digit hexadecimal value.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-dart" data-lang="dart">Color c1 = const Color(0x42A5F5);
Color c2 = const Color(42);</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-dart" data-lang="dart">Color c1 = const Color(0xFF42A5F5);
Color c2 = const Color.fromARGB(0xFF, 0x42, 0xA5, 0xF5);</code></pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p>Dart docs - <a href="https://dart.dev/tools/linter-rules/use_full_hex_values_for_flutter_colors">Dart Linter rule - use_full_hex_values_for_flutter_colors</a></p>
</li>
<li>
<p>Dart docs - <a href="https://api.flutter.dev/flutter/dart-ui/Color-class.html">Color class</a></p>
</li>
</ul>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="ulist">
<ul>
<li>
<p>"Instances of 'Color' should be created using an 8-digit hexadecimal integer (such as '0xFFFFFFFF')."</p>
</li>
</ul>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>The int literal in the argument</p>
</div>
<hr>
</div>
</div>
</div>