An empty comment is likely to be a mistake and doesn't help to improve the readability of the code. For these reasons, it should be removed.

== Why is this an issue?

=== Noncompliant code example

[source,vbnet]
----
'

'''
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

endif::env-github,rspecator-view[]
