include::../single-line-description.adoc

== Why is this an issue?

=== Noncompliant code example

[source,vbnet]
----
'

'''
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

endif::env-github,rspecator-view[]
