
The `apiVersion` of a resource is used to determine the configurable properties and values for the resource.
Setting it as a parameter or variable is not recommended, as it can lead to unexpected behaviors and deployment failures.

== Why is this an issue?

In Azure, different API versions of a resource can have changing properties and values.

Using a variable or parameter for the `apiVersion` for a resource may seem like good practice to always stay up to date with
the latest version.
This can lead to unexpected behaviors like deployment failures,
when the API version you set for a resource doesn't match the properties in your template.


== How to fix it in ARM Templates

To avoid these issues, it is recommended to set the `apiVersion` to a hard-coded value for the resource type.

When creating a new template, we suggest you use the latest API version for a resource type.

To determine which version to use, you can refer to the template reference of the official documentation linked below.
Make sure to choose a version that supports all the features you need.

When your template works as expected, we recommend you continue using the same API version.
Using the same API version means you don't have to worry about breaking changes that might be introduced in later versions.


=== Code examples

==== Noncompliant code example

[source,json,diff-id=1,diff-type=noncompliant]
----
{
  "parameters": {
    "customApiVersion": {
      "type": "string"
    }
  },
  "resources": [
    {
      "apiVersion": "[parameters('customApiVersion')]",
      "type": "Microsoft.Compute/virtualMachines",
      "name": "nonCompliantResourceWithParameterApiVersion"
    }
  ]
}
----

[source,json]
----
{
  "variables": {
    "customApiVersion": "[first(providers(‘Microsoft.Compute’,’virtualMachines’).apiVersions)]"
  },
  "resources": [
    {
      "apiVersion": "[variables('customApiVersion')]",
      "type": "Microsoft.Compute/virtualMachines",
      "name": "nonCompliantResourceWithVariableApiVersion"
    }
  ]
}
----


==== Compliant solution

[source,json,diff-id=1,diff-type=compliant]
----
{
  "resources": [
    {
      "apiVersion": "2020-06-01",
      "type": "Microsoft.Compute/virtualMachines",
      "name": "compliantResource"
    }
  ]
}
----

== Resources
=== Documentation

* https://learn.microsoft.com/en-us/azure/azure-resource-manager/templates/best-practices#api-version[ARM template best practices: API version]
* https://learn.microsoft.com/en-us/azure/templates/[Resource Template Reference]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Use a hardcoded value for the `apiVersion` of this resource.


=== Highlighting

* Highlight `apiVersion` of a resource where a parameter or variable is used.


endif::env-github,rspecator-view[]
