<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>In Kotlin, you must override either both or neither of the <code>equals(Any?)</code> and <code>hashCode()</code> methods in order to keep the contract between the two:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>Whenever the <code>hashCode</code> method is invoked on the same object more than once, it must consistently return the same integer, provided no information used in <code>equals</code> comparisons on the object is modified.</p>
</li>
<li>
<p>If two objects are equal according to the <code>equals</code> method, then calling the <code>hashCode</code> method on each of the two objects must produce the same integer result.</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>By overriding only one of the two methods with a non-trivial implementation, this contract is almost certainly broken.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-kotlin" data-lang="kotlin">class MyClass {    // Noncompliant - should also override "hashCode()"

  override fun equals(other: Any?): Boolean {
    /* ... */
  }

}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-kotlin" data-lang="kotlin">class MyClass {    // Compliant

  override fun equals(other: Any?): Boolean {
    /* ... */
  }

  override fun hashCode(): Int {
    /* ... */
  }

}</code></pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p><a href="https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/-any/hash-code.html"><code>hashCode</code> - kotlinlang.org</a></p>
</li>
</ul>
</div>
</div>
</div>