<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Because Visual Basic 6 does not automatically unload <code>Form</code> s after a program terminates, it is possible for them to remain in memory. Even after <code>Unload</code> ing, they may stay in memory if references remain. Therefore <code>Form</code> s should be explicitly <code>Unload</code> ed, and their references set to <code>Nothing</code>.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-vb6" data-lang="vb6">dim f1 as new form
f1.show
...
f1.hide</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-vb6" data-lang="vb6">dim f1 as new form
f1.show
...
f1.hide
Unload f1
Set f1 = Nothing</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>"Unload" this form and set its reference to "Nothing"</p>
</div>
</div>
</div>
</div>