include::../description.adoc[]

== Noncompliant Code Example

----
class MyClass {
  private:
    int foo = 42;  // Noncompliant, foo is unused

  public:
    int compute(int a) {
      return a * 42;
    }
};
----

== Compliant Solution

----
class MyClass {
  public:
    int compute(int a) {
      return a * 42;
    }
};
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
