== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,cpp]
----
class MyClass {
  private:
    int foo = 42;  // Noncompliant, foo is unused

  public:
    int compute(int a) {
      return a * 42;
    }
};
----

=== Compliant solution

[source,cpp]
----
class MyClass {
  public:
    int compute(int a) {
      return a * 42;
    }
};
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
