//// This rule is superseded by RSPEC-2259
//// Please consider implementing this latter instead.
When either the equality operator in a test for ``++null++`` or ``++undefined++``, or the logical operator that follows it is reversed, the code has the appearance of safely null-testing the object before dereferencing it. Unfortunately the effect is just the opposite - the object is null-tested and then dereferenced only if it is ``++null++``/``++undefined++``, leading to a guaranteed ``++TypeError++``.

== Noncompliant Code Example

----
if (str == null && str.length == 0) {
  console.log("String is empty");
}

if (str == undefined && str.length == 0) {
  console.log("String is empty");
}

if (str != null || str.length > 0) {
  console.log("String is not empty");
}

if (str != undefined || str.length > 0) {
  console.log("String is not empty");
}
----

== Compliant Solution

----
if (str != null && str.length == 0) {
  console.log("String is empty");
}

if (str != undefined && str.length == 0) {
  console.log("String is empty");
}

if (str == null || str.length > 0) {
  console.log("String is not empty");
}

if (str == undefined || str.length > 0) {
  console.log("String is not empty");
}
----
