<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Promises need to be resolved or awaited to return the expected value, otherwise, they return the promise object.</p>
</div>
<div class="paragraph">
<p><span class="underline"><strong>Unresolved promises:</strong></span></p>
</div>
<div class="paragraph">
<p>Forgetting to await a promise is a frequent mistake. There are places where the use of a promise object is confusing or unclear because the developer forgot to resolve it.</p>
</div>
<div class="paragraph">
<p>This rule forbids returning promises where another type is expected such as in:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>conditionals</p>
</li>
<li>
<p>void returns</p>
</li>
<li>
<p>spread operators</p>
</li>
</ul>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="paragraph">
<p>Using a promise instead of its resolved value can have unexpected results leading to bugs.</p>
</div>
<div class="ulist">
<ul>
<li>
<p>In conditionals, it will always return a truthy value.</p>
</li>
<li>
<p>In places where the expected type is void, returning a promise is often a mistake.</p>
</li>
<li>
<p>Using the spread operator on a promise will raise an exception.</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>The executor function of a promise can also be an async function. However, this usually denotes a mistake:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>If an async executor function throws an error, the error won&#8217;t cause the created promise to reject and will be lost. Therefore, this could make it difficult to debug and handle runtime errors.</p>
</li>
<li>
<p>If a promise executor function is using <code>await</code>, this means that it&#8217;s not necessary to use the <code>Promise</code> constructor, or the scope of the Promise constructor can be reduced.</p>
</li>
</ul>
</div>
</div>
<div class="sect2">
<h3 id="_exceptions">Exceptions</h3>
<div class="paragraph">
<p>This rule can be ignored for promises that you know will always resolve like timers.</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-javascript" data-lang="javascript">await new Promise(resolve =&gt; time.setTimeout(1000));</code></pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>If you mistakenly treated a promise as its resolved value, you can ensure it is properly resolved by using await or resolve on the promise. In some cases, you may need to use an "immediately invoked function expression" (IIFE):</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-javascript" data-lang="javascript">(async function foo() {
  const result = await bar();
  // work with result
})();</code></pre>
</div>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-javascript" data-lang="javascript">const promise = new Promise((resolve, reject) =&gt; {
  // ...
  resolve(false)
});
if (promise) {
  // ...
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-javascript" data-lang="javascript">const promise = new Promise((resolve, reject) =&gt; {
  // ...
  resolve(false)
});
if (await promise) {
  // ...
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_noncompliant_code_example_2">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-javascript" data-lang="javascript">const p = new Promise(async (resolve, reject) =&gt; {
  doSomething('Hey, there!', function(error, result) {
    if (error) {
      reject(error);
      return;
    }
    await saveResult(result)
    resolve(result);
  });
});

await p;</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution_2">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-javascript" data-lang="javascript">const p = new Promise((resolve, reject) =&gt; {
  doSomething('Hey, there!', function(error, result) {
    if (error) {
      reject(error);
      return;
    }
    resolve(result);
  });
});

const result = await p;
await saveResult(result);</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_noncompliant_code_example_3">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-javascript" data-lang="javascript">apiCalls.forEach(async (apiCall) =&gt; {
  await apiCall.send();
});</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution_3">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-javascript" data-lang="javascript">for (const apiCall of apiCalls) {
  await apiCall.send();
}</code></pre>
</div>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_how_does_this_work">How does this work?</h3>
<div class="paragraph">
<p>In JavaScript, a promise is a mechanism to perform tasks asynchronously.
To this end, the language provides the <code>Promise</code> object which represents the eventual completion or
failure of an asynchronous operation and its resulting value.
A promise can be created with the <code>Promise</code> constructor accepting an executor function as an argument,
which has <code>resolve</code> and <code>reject</code> parameters that are invoked when the promise completes or fails.</p>
</div>
<div class="paragraph">
<p>The logic of the promise is executed when it is called, however, its result is obtained only when the promise is resolved or awaited.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p><a href="https://typescript-eslint.io/">typescript-eslint</a> - Rule <a href="https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/docs/rules/no-misused-promises.mdx">no-misused-promises</a></p>
</li>
<li>
<p><a href="https://eslint.org">eslint</a> - Rule <a href="https://eslint.org/docs/latest/rules/no-async-promise-executor">no-async-promise-executor</a></p>
</li>
<li>
<p>MDN web docs - <a href="https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise">Promise</a></p>
</li>
<li>
<p>MDN web docs - <a href="https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Using_promises">Using promises</a></p>
</li>
<li>
<p>MDN web docs - <a href="https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/async_function">Async function</a></p>
</li>
<li>
<p>MDN web docs - <a href="https://developer.mozilla.org/en-US/docs/Glossary/IIFE">IIFE</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>