<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Input data that is not a JCL statement implicitly generate a <code>DD *</code> statement with the name <code>SYSIN</code>.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This behavior is implicit and may lead to confusion or unexpected behavior.</p>
</div>
<div class="paragraph">
<p>For example, this code:</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-jcl" data-lang="jcl">//MYDD DD DSN=TEST
Some in-stream data
// DD DSN=CONCAT-DD</code></pre>
</div>
</div>
<div class="paragraph">
<p>Gets implicitly transformed into:</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-jcl" data-lang="jcl">//MYDD DD DSN=TEST
//SYSIN DD *
Some in-stream data
// DD DSN=CONCAT-DD</code></pre>
</div>
</div>
<div class="paragraph">
<p>So the last DD statement is a concatenation of the implicit <code>//SYSIN DD *</code> statement instead of the explicit <code>//MYDD DD</code> statement.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Add explicitly the <code>//SYSIN DD *</code> before the in-stream data.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-jcl" data-lang="jcl">//STEP1     EXEC PGM=SORT
//SYSOUT   DD   SYSOUT=*
//SORTIN   DD   DSN=A123456.SORTIN,DISP=SHR
//SORTOUT  DD   DSN=A123456.SORTOUT,DISP=OLD
  SORT FIELDS=(1,12,CH,A)</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-jcl" data-lang="jcl">//STEP1     EXEC PGM=SORT
//SYSOUT   DD   SYSOUT=*
//SORTIN   DD   DSN=A123456.SORTIN,DISP=SHR
//SORTOUT  DD   DSN=A123456.SORTOUT,DISP=OLD
//SYSIN    DD   *
  SORT FIELDS=(1,12,CH,A)
/*</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p><a href="https://www.ibm.com/docs/en/zos/3.1.0?topic=parameter-location-in-jcl">IBM reference - DD * location in the JCL</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>