<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>It is needlessly complex to invert the result of a boolean comparison. The opposite comparison should be made instead.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-javascript" data-lang="javascript">if (!(a === 2)) { ... }  // Noncompliant</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-javascript" data-lang="javascript">if (a !== 2) { ... }</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Use the opposite operator ("XX") instead.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_12_nov_2015_182359_linda_martin_wrote">on 12 Nov 2015, 18:23:59 Linda Martin wrote:</h3>
<div class="paragraph">
<p>OK!</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_27_sep_2019_155521_sergey_adamovski_wrote">on 27 Sep 2019, 15:55:21 Sergey Adamovski wrote:</h3>
<div class="paragraph">
<p>It is wrong that "<code>!(a&gt;2)</code>" is equivalent to "<code>a&lt;=2</code>".</p>
</div>
<div class="paragraph">
<p>In case a == NaN (not a number), any compare to it will give "false". <code>(a&gt;2) == false</code> and <code>(a&lt;=2) == false</code>. Therefore, "!(a&gt;2)" will be <code>true</code> while "a+&#8656;+2" <code>false</code>.</p>
</div>
<div class="paragraph">
<p>Equivalent of "<code>!(a&gt;2)</code>" would be "<code>(a&lt;=2) || (double.IsNaN(a))</code>" or "<code>(a&lt;=2) || (float.IsNaN(a))</code>" which is easier to read but longer to execute.</p>
</div>
<div class="paragraph">
<p>One can</p>
</div>
<div class="ulist">
<ul>
<li>
<p>check whether comparison contains any float/double values that can be Not-A-Number <em>or</em></p>
</li>
<li>
<p>at least warn a user that it is not the same <em>or</em></p>
</li>
<li>
<p>remove this restriction</p>
</li>
</ul>
</div>
</div>
</div>
</div>