<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>When a type implements <code>Borrow&lt;T&gt;</code>, it should provide consistent behavior between the borrowed and owned values, especially regarding <code>Eq</code>, <code>Ord</code>, and <code>Hash</code>. However, <code>str</code> and <code>[u8]</code> have different Hash implementations, leading to inconsistent hash values for the same underlying data when accessed through different Borrow implementations. This violates the principle that <code>hash(x) == hashx as Borrow&lt;[u8]&gt;).borrow( == hashx as Borrow&lt;str&gt;).borrow(</code> must hold, and can cause issues with hash-based collections and comparisons.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-rust" data-lang="rust">use std::borrow::Borrow;
use std::hash::{Hash, Hasher};

struct ExampleType {
    data: String,
}

impl Hash for ExampleType {
    fn hash&lt;H: Hasher&gt;(&amp;self, state: &amp;mut H) {
        self.data.hash(state); // Noncompliant: Inconsistent Hash
    }
}

impl Borrow&lt;str&gt; for ExampleType {
    fn borrow(&amp;self) -&gt; &amp;str {
        &amp;self.data
    }
}

impl Borrow&lt;[u8]&gt; for ExampleType {
    fn borrow(&amp;self) -&gt; &amp;[u8] {
        self.data.as_bytes()
    }
}</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Clippy Lints - <a href="https://rust-lang.github.io/rust-clippy/master/index.html#impl_hash_borrow_with_str_and_bytes" class="bare">https://rust-lang.github.io/rust-clippy/master/index.html#impl_hash_borrow_with_str_and_bytes</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>