<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when a class derives from one of the following exception classes: <code>BaseException</code>, <code>KeyboardInterrupt</code>, <code>SystemExit</code> or <code>GeneratorExit</code>.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p><a href="https://docs.python.org/3/library/exceptions.html#SystemExit"><code>SystemExit</code></a> is raised when <a href="https://docs.python.org/3/library/sys.html#sys.exit"><code>sys.exit()</code></a> is called. <a href="https://docs.python.org/3/library/exceptions.html#KeyboardInterrupt"><code>KeyboardInterrupt</code></a> is raised when the user asks the program to stop by pressing interrupt keys. Both exceptions are expected to propagate up until the application stops.</p>
</div>
<div class="paragraph">
<p>In order to avoid catching <code>SystemExit</code> and <code>KeyboardInterrupt</code> by mistake, <a href="https://www.python.org/dev/peps/pep-0352/#exception-hierarchy">PEP-352</a> created the root class <code>BaseException</code> from which <code>SystemExit</code>, <code>KeyboardInterrupt</code> and <code>Exception</code> derive. Thus developers can use <code>except Exception:</code> without preventing the software from stopping.</p>
</div>
<div class="paragraph">
<p>The <code>GeneratorExit</code> class also derives from <code>BaseException</code> as it is not really an error and is not supposed to be caught by user code.</p>
</div>
<div class="paragraph">
<p>As said in <a href="https://docs.python.org/3/library/exceptions.html#BaseException">Python&#8217;s documentation</a>, user-defined exceptions are not supposed to inherit directly from <code>BaseException</code>. They should instead inherit from <code>Exception</code> or one of its subclasses.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">class MyException(BaseException):  # Noncompliant
    pass</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">class MyException(Exception):
    pass</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>PEP 352 â€“ <a href="https://www.python.org/dev/peps/pep-0352/#exception-hierarchy-changes">Required Superclass for Exceptions</a></p>
</li>
<li>
<p>Python Documentation - <a href="https://docs.python.org/3/library/exceptions.html#BaseException">BaseException class</a></p>
</li>
</ul>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Derive this class from "Exception" instead of BaseException/GeneratorExit/KeyboardInterrupt/SystemExit</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>The forbidden parent class reference.</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-text" data-lang="text">class Foo(BaseException)
          ^^^^^^^^^^^^^</code></pre>
</div>
</div>
</div>
</div>
</div>