This rule raises an issue when `readLine()` is called on Reader implementations that don't support the `mark()` operation, such as `StringReader`.

== Why is this an issue?

The `readLine()` method in Groovy is designed to work optimally with Readers that support the `mark()` operation, like `BufferedReader`. When used with Readers that don't support marking (such as `StringReader`), it falls back to a less reliable implementation.

This fallback behavior has a significant limitation: it doesn't correctly detect multi-character line termination sequences. Specifically, it may not properly handle carriage return followed by line feed (`\r\n`), which is the standard line ending on Windows systems.

This can lead to:

* Incorrect line parsing where a single logical line gets split into multiple lines
* Data corruption when processing text files with mixed line endings
* Inconsistent behavior across different operating systems
* Silent failures that are difficult to detect during testing

The issue is particularly problematic because it may work correctly with simple test data but fail with real-world input containing various line ending formats.

=== What is the potential impact?

Using `readLine()` with unsupported Readers can cause data corruption and inconsistent text processing. Lines may be incorrectly split, leading to malformed data processing and potential application errors. This is especially problematic when handling files from different operating systems or user-generated content with mixed line endings.

== How to fix it

Replace `readLine()` with `eachLine()` for iterating through lines. The `eachLine()` method correctly handles all line termination formats and automatically closes the Reader.

=== Code examples

==== Noncompliant code example

[source,groovy,diff-id=1,diff-type=noncompliant]
----
def reader = new StringReader(text)
def line = reader.readLine() // Noncompliant
while (line != null) {
    processLine(line)
    line = reader.readLine()
}
----

==== Compliant solution

[source,groovy,diff-id=1,diff-type=compliant]
----
new StringReader(text).eachLine { line ->
    processLine(line)
}
----

== Resources

=== Documentation

 * Groovy Reader Documentation - http://docs.groovy-lang.org/latest/html/groovy-jdk/java/io/Reader.html[Official Groovy documentation for Reader methods, including the readLine() warning and recommended alternatives]
