<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Conversion of a function pointer to a different type of pointer results in undefined behaviour. This means, for example, that a pointer to a function cannot be converted to a pointer to a different type of function.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">int f(int a)
{
  float (*p)(float) = (float (*)(float)) &amp; f; // Noncompliant
}</code></pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p>MISRA C:2004, 11.1 - Conversions shall not be performed between a pointer to a function and any type other than an integral type.</p>
</li>
<li>
<p>MISRA C&#43;&#43;:2008, 5-2-6 - A cast shall not convert a pointer to a function to any other pointer type, including a pointer to function type.</p>
</li>
<li>
<p>MISRA C:2012, 11.1 - Conversions shall not be performed between a pointer to a function and any other type</p>
</li>
</ul>
</div>
<hr>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove this hazardous cast.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_relates_to_s856">relates to: <a data-rspec-id="S856" class="rspec-auto-link">S856</a></h3>

</div>
<div class="sect2">
<h3 id="_relates_to_s1944">relates to: <a data-rspec-id="S1944" class="rspec-auto-link">S1944</a></h3>

</div>
<div class="sect2">
<h3 id="_on_3_oct_2014_184028_ann_campbell_wrote">on 3 Oct 2014, 18:40:28 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~evgeny.mandrikov]</p>
</div>
<div class="paragraph">
<p>MISRA C:2004 says: "Conversions shall not be performed between a pointer to a function and any type <strong>other than an integral type</strong>" (emphasis mine), meaning a cast to an integral type is allowed.</p>
</div>
<div class="paragraph">
<p>MISRA C&#43;&#43;:2008: "A cast should not convert a pointer type to an integral type." I.e. a direct contradiction of the second part of the C:2004 rule.</p>
</div>
<div class="paragraph">
<p>MISRA C:2012 backs up C&#43;&#43;:2008: "Conversions shall not be performed between a pointer to a function and any other type."</p>
</div>
<div class="paragraph">
<p>So&#8230;&#8203; what would you like written up?</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_23_oct_2014_092923_samuel_mercier_wrote">on 23 Oct 2014, 09:29:23 Samuel Mercier wrote:</h3>
<div class="paragraph">
<p>Rule C&#43;&#43; 5-2-9 extracted to another ticket, since its scope is different.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_23_oct_2014_093506_samuel_mercier_wrote">on 23 Oct 2014, 09:35:06 Samuel Mercier wrote:</h3>
<div class="paragraph">
<p>Don&#8217;t get the point of MISRA 2004 : conversion from function pointer to integral type is hazardous&#8230;&#8203;</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_23_oct_2014_150710_ann_campbell_wrote">on 23 Oct 2014, 15:07:10 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~samuel.mercier] please add a See section</p>
</div>
</div>
</div>
</div>