== Why is this an issue?

include::description.adoc[]

=== Noncompliant code example

[source,text]
----
SELECT * 
       FROM persons 
       INTO newyorkers 
       WHERE city = 'NEW YORK'. 
----


=== Compliant solution

[source,text]
----
SELECT firstname, lastname 
       FROM persons 
       INTO newyorkers
       WHERE city = 'NEW YORK'. 
----

