<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p><code>SELECT *</code> should be avoided because it releases control of the returned columns and could therefore lead to errors and potentially to performance issues.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-sql" data-lang="sql">SELECT *     -- Noncompliant
       FROM persons
       WHERE city = 'NEW YORK'</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-sql" data-lang="sql">SELECT firstname, lastname
       FROM persons
       WHERE city = 'NEW YORK'</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_exceptions">Exceptions</h3>
<div class="paragraph">
<p>The following cases are ignored by this rule:</p>
</div>
<div class="ulist">
<ul>
<li>
<p><code>SELECT</code> from temporary tables: <code>SELECT * FROM #temp1</code></p>
</li>
<li>
<p><code>SELECT</code> using common table expressions: <code>WITH A AS (SELECT C1 FROM T1) SELECT * FROM A;</code></p>
</li>
<li>
<p>Inside another <code>SELECT</code>: <code>SELECT C1 FROM T1 WHERE C2 IN (SELECT * FROM T2)</code></p>
</li>
<li>
<p>Inside <code>INSERT</code>: <code>INSERT INTO T1 SELECT * FROM T2</code></p>
</li>
<li>
<p>Inside <code>CREATE TABLE</code>: <code>CREATE TABLE T1 WITH (C1 = C2) AS SELECT * FROM T2</code></p>
</li>
<li>
<p><code>SELECT</code> from rowset providers: <code>SELECT *  FROM OPENXML (@idoc, '/ROOT/Customer',1)</code></p>
</li>
<li>
<p><code>SELECT INTO</code>: <code>SELECT * INTO NEW_TABLE FROM T1</code></p>
</li>
<li>
<p><code>SELECT</code> from variable table: <code>SELECT * FROM @table1</code></p>
</li>
<li>
<p><code>SELECT</code> from derived table: <code>SELECT A.* FROM (SELECT X FROM T1) A INNER JOIN B ON A.X = B.X</code></p>
</li>
</ul>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Replace the use of "*" with a list of the columns to select.</p>
</div>
</div>
</div>
</div>