== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,swift]
----
whileLoopLabel: while x > 0 {    // Noncompliant
    x -= 1
}
----

=== Compliant solution

[source,swift]
----
while x > 0 {
    x -= 1
}
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
