<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when <code>System.abortJob()</code> is called on batch jobs without first filtering by job status.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>When you call <code>System.abortJob()</code> without checking the job status first, you may attempt to abort jobs that are already completed, failed, or aborted. This creates unnecessary overhead and can lead to confusing error messages.</p>
</div>
<div class="paragraph">
<p>Batch jobs in Salesforce can have various statuses: Queued, Processing, Completed, Failed, Aborted, and others. Only jobs with "Queued" or "Processing" status can be successfully aborted. Attempting to abort jobs in other states will either fail silently or throw exceptions, depending on the specific status.</p>
</div>
<div class="paragraph">
<p>This pattern often occurs when developers query all batch jobs and then loop through them calling <code>System.abortJob()</code> without consideration for the current state. While this might seem like a simple approach, it wastes system resources and can mask real issues when legitimate abort operations fail.</p>
</div>
<div class="paragraph">
<p>A better approach is to filter jobs by their status in the SOQL query itself, ensuring you only attempt to abort jobs that are actually running or waiting to run. This makes your code more efficient and your intentions clearer.</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="paragraph">
<p>Attempting to abort jobs that cannot be aborted wastes system resources and may generate unnecessary error logs. It can also mask legitimate failures when trying to abort jobs that should be abortable.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Filter batch jobs by status in your SOQL query to only include jobs that can be aborted (Queued or Processing status).</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-apex" data-lang="apex">List&lt;AsyncApexJob&gt; jobs = [SELECT Id FROM AsyncApexJob WHERE JobType = 'BatchApex'];
for (AsyncApexJob job : jobs) {
    System.abortJob(job.Id); // Noncompliant
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-apex" data-lang="apex">List&lt;AsyncApexJob&gt; jobs = [SELECT Id, Status FROM AsyncApexJob
                           WHERE JobType = 'BatchApex'
                           AND Status IN ('Queued', 'Processing')];
for (AsyncApexJob job : jobs) {
    System.abortJob(job.Id);
}</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>AsyncApexJob Object Reference - <a href="https://developer.salesforce.com/docs/atlas.en-us.object_reference.meta/object_reference/sforce_api_objects_asyncapexjob.htm">Official Salesforce documentation for the AsyncApexJob object and its Status field values</a></p>
</li>
<li>
<p>System.abortJob() Method - <a href="https://developer.salesforce.com/docs/atlas.en-us.apexref.meta/apexref/apex_methods_system_system.htm#apex_System_System_abortJob_1">Documentation for the System.abortJob() method and its behavior</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>