<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>The long suffix should always be written in uppercase, i.e. 'L', as the lowercase 'l' can easily be confused with the digit one '1'.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">return 10l  // Noncompliant; easily confused with one zero one</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">return 10L</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Replace this lower case "l" long suffix by an upper case "L".</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_relates_to_s818">relates to: <a data-rspec-id="S818" class="rspec-auto-link">S818</a></h3>

</div>
<div class="sect2">
<h3 id="_on_12_jul_2013_154305_dinesh_bolkensteyn_wrote">on 12 Jul 2013, 15:43:05 Dinesh Bolkensteyn wrote:</h3>
<div class="paragraph">
<p>Implemented by http://jira.codehaus.org/browse/SONARJAVA-208</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_13_apr_2015_182857_evgeny_mandrikov_wrote">on 13 Apr 2015, 18:28:57 Evgeny Mandrikov wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] Looks like duplicate of <a data-rspec-id="S818" class="rspec-auto-link">RSPEC-818</a></p>
</div>
</div>
<div class="sect2">
<h3 id="_on_20_apr_2015_135212_ann_campbell_wrote">on 20 Apr 2015, 13:52:12 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>deprecated</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_21_jun_2016_154137_elena_vilchik_wrote">on 21 Jun 2016, 15:41:37 Elena Vilchik wrote:</h3>
<div class="paragraph">
<p><code>l</code>/<code>L</code> is irrelevant for Python as this feature was dropped since Python 3.0</p>
</div>
</div>
</div>
</div>