include::../rule.adoc[]

=== Code examples

==== Noncompliant code example

With the default threshold of 3:

[source,rpg,diff-id=1,diff-type=noncompliant]
----
C                   MOVEL     'CLEAR'       W1CLR                           Noncompliant
C                   MOVEL     '*DECOD'      W1DCDE
C                   MOVEL     '*ERROR'      W1ERR 
C                   MOVEL     '*EXIT '      W1EXIT
C                   MOVEL     'CLEAR'       W1FIRT
C                   MOVEL     'CLEAR'       W1HELP
----

[source,rpg,diff-id=2,diff-type=noncompliant]
----
 /free
   W1CLR = 'CLEAR'; // Noncompliant
   W1FIRT = 'CLEAR';
   W1HELP = 'CLEAR';
 /end-free
----

==== Compliant solution

[source,rpg,diff-id=1,diff-type=compliant]
----
D W0Clr         C                   CONST('CLEAR')
C                   MOVEL     W0Clr         W1CLR
C                   MOVEL     '*DECOD'      W1DCDE
C                   MOVEL     '*ERROR'      W1ERR
C                   MOVEL     '*EXIT '      W1EXIT
C                   MOVEL     W0Clr         W1FIRT
C                   MOVEL     W0Clr         W1HELP
----

[source,rpg,diff-id=2,diff-type=compliant]
----
D W0Clr         C                   CONST('CLEAR')
 /free
   W1CLR = W0Clr;
   W1FIRT = W0Clr;
   W1HELP = W0Clr;
 /end-free
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../parameters.adoc[]

include::../highlighting.adoc[]

'''


endif::env-github,rspecator-view[]
