<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>In general, <code>async</code> <code>void</code> test methods are not executed by test frameworks, therefore it&#8217;s better to avoid them altogether.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-csharp" data-lang="csharp">    [TestMethod]
    public async void MyIgnoredTestMethod()  // Noncompliant
    { /* ... */ }</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_duplicates_s3433">duplicates: <a data-rspec-id="S3433" class="rspec-auto-link">S3433</a></h3>

</div>
<div class="sect2">
<h3 id="_on_1_dec_2015_143237_tamas_vajk_wrote">on 1 Dec 2015, 14:32:37 Tamas Vajk wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] I changed the description to not mention <code>[TestMethod]</code> attribute, because that is framework specific. In NUnit, xUnit, and MS test frameworks there are different attributes for marking test methods.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_1_dec_2015_150204_ann_campbell_wrote">on 1 Dec 2015, 15:02:04 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>Thanks [~tamas.vajk]</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_31_jul_2017_160705_jean_christophe_collet_wrote">on 31 Jul 2017, 16:07:05 Jean-Christophe Collet wrote:</h3>
<div class="paragraph">
<p>Merged with <a data-rspec-id="S3433" class="rspec-auto-link">RSPEC-3433</a></p>
</div>
</div>
</div>
</div>