== Why is this an issue?

While both ``++new++`` and ``++Create++`` will yield the same ``++Tuple++``, the ``++Create++`` method automatically infers the argument types, yielding sleeker, more readable code.


=== Noncompliant code example

[source,text]
----
var fastAnimal = new Tuple<string, int>("Cheetah", 93);  // Noncompliant
----


=== Compliant solution

[source,text]
----
var fastAnimal = Tuple.Create("Cheetah", 93);
----

