While both ``++new++`` and ``++Create++`` will yield the same ``++Tuple++``, the ``++Create++`` method automatically infers the argument types, yielding sleeker, more readable code.


== Noncompliant Code Example

[source,text]
----
var fastAnimal = new Tuple<string, int>("Cheetah", 93);  // Noncompliant
----


== Compliant Solution

----
var fastAnimal = Tuple.Create("Cheetah", 93);
----

