<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This is an issue when creating triggers for Change Data Capture events using incorrect object naming patterns that will cause compilation errors.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Change Data Capture (CDC) in Salesforce requires specific naming conventions for event objects to function correctly. The platform automatically generates these event objects based on the underlying Salesforce objects, and triggers must reference them using the exact naming pattern.</p>
</div>
<div class="paragraph">
<p>For standard Salesforce objects (like Account, Contact, Opportunity), the event object name follows the pattern <code>StandardObjectNameChangeEvent</code>. For example, changes to Account records generate <code>AccountChangeEvent</code> objects.</p>
</div>
</div>
For custom objects (those ending with c), the event object name follows the pattern <code>CustomObjectNameChangeEvent</code>. For example, if you have a custom object called Employeec, its change events are captured in <code>EmployeeChangeEvent</code>.

<div class="paragraph">
<p>Using incorrect naming conventions will result in compilation errors because the Salesforce platform cannot find the referenced event object. This prevents the trigger from being saved or deployed, blocking the implementation of asynchronous processing logic that depends on Change Data Capture.</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="paragraph">
<p>Using incorrect naming conventions for Change Data Capture event objects will cause compilation errors, preventing the trigger from being deployed or executed. This blocks the implementation of real-time data synchronization and asynchronous processing workflows that depend on Change Data Capture functionality.</p>
</div>
</div>


<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>For standard Salesforce objects, use the naming pattern <code>StandardObjectNameChangeEvent</code> without any underscores.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-apex" data-lang="apex">trigger AsyncAccountTriggers on Account__ChangeEvent (after insert) {
    // Incorrect naming for standard object - includes unnecessary underscores
    // This will cause a compilation error
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-apex" data-lang="apex">trigger AsyncAccountTriggers on AccountChangeEvent (after insert) {
    // Correct naming for standard object
    // Standard objects use: ObjectNameChangeEvent
}</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Change Data Capture Developer Guide - <a href="https://developer.salesforce.com/docs/atlas.en-us.change_data_capture.meta/change_data_capture/">Official Salesforce documentation for Change Data Capture implementation and best practices</a></p>
</li>
<li>
<p>Apex Triggers for Change Events - <a href="https://developer.salesforce.com/docs/atlas.en-us.change_data_capture.meta/change_data_capture/cdc_trigger.htm">Documentation on creating Apex triggers for Change Data Capture events</a></p>
</li>
</ul>
</div>
</div>
</div>
</div></div>