Using both ``++SPACEA++`` (space-after) and ``++SPACEB++`` (space-before) in the same printer file can make it difficult to understand how many blank lines will be printed where. Therefore either one command or the other should be used but not both.


== Noncompliant Code Example

----
A          R REP001                    TEXT('Detail line.       ')        
A                                      SPACEB(1)                          
A                                      SPACEA(1)                          
A                                     2'Additional Details for Student :'
A            VAR01          6        36TEXT('Additional Details for +     
A                                          Student')
----


== Compliant Solution

----
A          R REP001                    TEXT('Detail line.       ')        
A                                      SPACEB(1)                                
A                                     2'Additional Details for Student :'
A            VAR01          6        36TEXT('Additional Details for +     
A                                          Student')
A                                      SPACEB(1)
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
