Shared coding conventions allow teams to collaborate efficiently. This rule checks that all enumeration values match a provided regular expression.

== Noncompliant Code Example

With default provided regular expression:

----
enum SomeEnumeration {
    some  // Non-Compliant
};
----

== Compliant Solution

----
enum SomeEnumeration {
    SOME
};
----
ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

include::../parameters.adoc[]

endif::env-github,rspecator-view[]
