== Why is this an issue?

Shared coding conventions allow teams to collaborate efficiently. This rule checks that all enumeration values match a provided regular expression.

=== Noncompliant code example

With default provided regular expression:

[source,cpp]
----
enum SomeEnumeration {
    some  // Non-Compliant
};
----

=== Compliant solution

[source,cpp]
----
enum SomeEnumeration {
    SOME
};
----
ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

key: format 

description: Regular expression used to check the enumeration value names against.

default: ^[A-Z][A-Z_0-9]*$


include::../parameters.adoc[]

endif::env-github,rspecator-view[]
