<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>In JCL, some statements require to have a label or name defined, otherwise it will trigger a JCL error.
The statements that require a name/label are:
* CNTL
* JOB
* NOTIFY
* OUTPUT
* in-stream PROC</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-jcl" data-lang="jcl">//MYJOB   JOB DXXX
//        PROC
//STEP1   EXEC PGM=MYPGM
//        PEND</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-jcl" data-lang="jcl">//MYJOB   JOB DXXX
//TRPOC   PROC
//STEP1   EXEC PGM=MYPGM
//        PEND</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p><a href="https://www.ibm.com/docs/en/zos/3.1.0?topic=description-label-field">IBM Reference - CNTL Label field</a></p>
</li>
<li>
<p><a href="https://www.ibm.com/docs/en/zos/3.1.0?topic=d-name-field-5">IBM Reference - JOB Name field</a></p>
</li>
<li>
<p><a href="https://www.ibm.com/docs/en/zos/3.1.0?topic=statement-label-field">IBM Reference - NOTIFY Label field</a></p>
</li>
<li>
<p><a href="https://www.ibm.com/docs/en/zos/3.1.0?topic=d-name-field-6">IBM Reference - OUTPUT Name field</a></p>
</li>
<li>
<p><a href="https://www.ibm.com/docs/en/zos/3.1.0?topic=d-name-field-8">IBM reference - PROC statement - Name field</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>