include::../description.adoc[]

== How to fix it in Gradle

=== Code examples

==== Noncompliant code example

[source,kotlin,diff-id=1,diff-type=noncompliant]
----
dependencies {
    implementation("com.example:a-dependency:1.0")
}

configurations { 
    all {
        resolutionStrategy {
            disableDependencyVerification() // Noncompliant: dependency verification is disabled
        }
    }
}
----

Absence of a `verification-metadata.xml` file in the `gradle` directory of your project will also result in the Gradle build not verifying the integrity of the dependencies.


==== Compliant solution

[source,kotlin,diff-id=1,diff-type=compliant]
----
dependencies {
    implementation("com.example:adependency:1.0")
}
----

Create a `verification-metadata.xml` in the `gradle` directory of your project.
Use `./gradlew --write-verification-metadata pgp,sha256 --export-keys` to bootstrap the file content.

[source,xml]
----
<verification-metadata 
    xmlns="https://schema.gradle.org/dependency-verification" 
    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" 
    xsi:schemaLocation="https://schema.gradle.org/dependency-verification https://schema.gradle.org/dependency-verification/dependency-verification-1.3.xsd">
   <configuration>
      <verify-metadata>true</verify-metadata>
      <verify-signatures>true</verify-signatures>
      <keyring-format>armored</keyring-format>
      <trusted-keys>
         <trusted-key id="FD8190C7D72E7DCD42582B1042677B9FC1DC2161" group="com.example" name="adependency"/>
      </trusted-keys>
   </configuration>
</verification-metadata>
----

== How does this work?

include::../common/fix/gradle-verification.adoc[]

== Resources

include::../common/resources/documentation.adoc[]

include::../common/resources/standards.adoc[]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

* Create a `verification-metadata.xml` file to verify these dependencies against a known checksum or signature
* Remove this call to `disableDependencyVerification()` so that dependencies are verified

=== Highlighting

* Highlight the `dependencies` block
* Highlight `disableDependencyVerification()`


endif::env-github,rspecator-view[]
