Assignment of ``++false++`` to a pointer type is implicitly converted to a ``++NULL++`` assignment.


== Noncompliant Code Example

----
void foo(bool *p) {
  p = false;  // Noncompliant
}
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
