== Why is this an issue?

Assignment of ``++false++`` to a pointer type is implicitly converted to a ``++NULL++`` assignment.


=== Noncompliant code example

[source,cpp]
----
void foo(bool *p) {
  p = false;  // Noncompliant
}
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Remove this "false" assignment to a pointer type.


'''
== Comments And Links
(visible only on this page)

=== on 27 Jul 2015, 12:51:59 Ann Campbell wrote:
\[~massimo.paladin] I've upgraded this to Critical/bug. See what you think.

=== on 28 Aug 2015, 12:25:32 Massimo PALADIN wrote:
\[~evgeny.mandrikov] could you please verify?

=== on 28 Aug 2015, 13:56:21 Evgeny Mandrikov wrote:
\[~massimo.paladin] looks good

endif::env-github,rspecator-view[]
