== Why is this an issue?

include::../rationale.adoc[]

include::../impact.adoc[]

== How to fix it

include::../common/how-to-fix-it/intro.adoc[]

=== Code examples

==== Noncompliant code example

An ingress rule allowing all inbound SSH traffic:

[source,yaml,diff-id=1,diff-type=noncompliant]
----
---
- name: Configure iptables rules
  hosts: all
  become: yes
  tasks:
    - name: Allow incoming SSH connections on port 22
      ansible.builtin.iptables:
        chain: INPUT
        protocol: tcp
        destination_port: "22"   # Noncompliant
        jump: ACCEPT

    - name: Allow existing connections
      ansible.builtin.iptables:
        chain: INPUT
        protocol: tcp
        ctstate: ESTABLISHED,RELATED
        jump: ACCEPT

    - name: Drop all other incoming traffic
      ansible.builtin.iptables:
        chain: INPUT
        jump: DROP
----

==== Compliant solution

An ingress rule allowing inbound SSH traffic from specific IP addresses:

[source,yaml,diff-id=1,diff-type=compliant]
----
---
- name: Configure iptables rules
  hosts: all
  become: yes
  tasks:
    - name: Allow incoming SSH connections on port 22
      ansible.builtin.iptables:
        chain: INPUT
        protocol: tcp
        destination_port: "22"
        source: "42.42.42.42"
        jump: ACCEPT

    - name: Allow existing connections
      ansible.builtin.iptables:
        chain: INPUT
        protocol: tcp
        ctstate: ESTABLISHED,RELATED
        jump: ACCEPT

    - name: Drop all other incoming traffic
      ansible.builtin.iptables:
        chain: INPUT
        jump: DROP
----

== Resources

=== Documentation

Ansible Community Documentation - https://docs.ansible.com/ansible/latest/collections/ansible/builtin/iptables_module.html[Modify iptables rules]

include::../common/resources/articles.adoc[]

include::../common/resources/presentations.adoc[]

include::../common/resources/standards.adoc[]


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

* Change this code to restrict the source to a subset of trusted IP addresses.

=== Highlighting

Highlight the value of the `destination_port` key in the `ansible.builtin.iptables` task.

'''

endif::env-github,rspecator-view[]
