<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>There is no point in adding unused strings to the code. If the first line of a function or a class is a string, it&#8217;s considered documentation, but otherwise short strings (enclosed in single quotes <code>'</code> or double quotes <code>"</code>) that aren&#8217;t either assigned to variables or used in function calls or expressions are considered confusing cruft, and should be removed.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-text" data-lang="text">def fun():
    'documentation string'    # Compliant
    print('Hello, world')
    'Hello, world'    # Noncompliant; has no effect
    ''' Long strings are ignored '''   # Compliant</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-text" data-lang="text">def fun():
    'documentation string'    # Compliant
    print('Hello, world')
    ''' Long strings are ignored '''   # Compliant</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove this useless string.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_24_apr_2015_090847_elena_vilchik_wrote">on 24 Apr 2015, 09:08:47 Elena Vilchik wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] Please check description, i made some small changes</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_28_apr_2015_153731_ann_campbell_wrote">on 28 Apr 2015, 15:37:31 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>looks good [~elena.vilchik]</p>
</div>
</div>
</div>
</div>