== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,sql]
----
DELETE FROM countries
UPDATE employee SET status = 'retired' FROM table1 AS employee
----

=== Compliant solution

[source,sql]
----
TRUNCATE TABLE countries
DELETE FROM countries WHERE CODE = @country_code
UPDATE employee SET status = 'retired' FROM table1 AS employee WHERE age > @maxAge
----

=== Exceptions

No issue is reported for statements made on temporary tables.

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
