include::../description.adoc[]

== Noncompliant Code Example

----
DELETE FROM countries
UPDATE employee SET status = 'retired' FROM table1 AS employee
----

== Compliant Solution

----
TRUNCATE TABLE countries
DELETE FROM countries WHERE CODE = @country_code
UPDATE employee SET status = 'retired' FROM table1 AS employee WHERE age > @maxAge
----

== Exceptions

No issue is reported for statements made on temporary tables.
