<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>A Single Abstract Method (SAM) interface is a Java interface containing only one method. The Java API is full of SAM interfaces, such as <code>java.lang.Runnable</code>, <code>java.awt.event.ActionListener</code>, <code>java.util.Comparator</code> and <code>java.util.concurrent.Callable</code>. SAM interfaces have a special place in Java 8 because they can be implemented using Lambda expressions or Method references.</p>
</div>
<div class="paragraph">
<p>Using <code>@FunctionalInterface</code> forces a compile break when an additional, non-overriding abstract method is added to a SAM, which would break the use of Lambda implementations.</p>
</div>
<div class="paragraph">
<p><strong>Note</strong> that this rule is automatically disabled when the project&#8217;s <code>sonar.java.source</code> is lower than <code>8</code>.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">public interface Changeable&lt;T&gt; {
  public void change(T o);
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">@FunctionalInterface
public interface Changeable&lt;T&gt; {
  public void change(T o);
}</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Annotate the "XXXX" interface with the "@FunctionalInterface" annotation.  [(sonar.java.source not set. Assuming 8 or greater.)]</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_25_feb_2014_233107_freddy_mallet_wrote">on 25 Feb 2014, 23:31:07 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>Is implemented by http://jira.codehaus.org/browse/SONARJAVA-462</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_26_feb_2014_015025_ann_campbell_wrote">on 26 Feb 2014, 01:50:25 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>I expanded a little here. Double-check me, please.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_23_oct_2015_132555_michael_gumowski_wrote">on 23 Oct 2015, 13:25:55 Michael Gumowski wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] I activated the rule by default, as we are now able to execute it only if the java version of the source code is java8+!</p>
</div>
<div class="paragraph">
<p>I also added code tags into the description.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_19_nov_2015_172540_michael_gumowski_wrote">on 19 Nov 2015, 17:25:40 Michael Gumowski wrote:</h3>
<div class="paragraph">
<p>Fixed typo in error message (see commit: https://github.com/SonarSource/sonar-java/commit/ef867a2aa66d071227bb00c2449cef2cceb8a738)</p>
</div>
</div>
</div>
</div>