``++UNSTRING++`` and ``++STRING++`` statements can be used on one side to divide a string into sub-strings and on the other side to concatenate the contents of several strings into one single date item. 


As useful those statements can be, the other way to get the same result in a more declarative and maintainable way is to extend the data items structure.


== Noncompliant Code Example

Waiting for feedback from BNP


ifdef::env-github,rspecator-view[]
'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
