== Why is this an issue?

``++UNSTRING++`` and ``++STRING++`` statements can be used on one side to divide a string into sub-strings and on the other side to concatenate the contents of several strings into one single date item. 


As useful those statements can be, the other way to get the same result in a more declarative and maintainable way is to extend the data items structure.


=== Noncompliant code example

Waiting for feedback from BNP


ifdef::env-github,rspecator-view[]
'''
== Comments And Links
(visible only on this page)

=== on 27 May 2014, 07:18:37 Freddy Mallet wrote:
I'm closing "Won't Fix" this RSPEC as this will lead to generate lot of false-positives.

=== on 9 Jun 2014, 12:26:34 Freddy Mallet wrote:
We should provide the XPath expression.

endif::env-github,rspecator-view[]
