=== on 19 Sep 2014, 13:39:33 Freddy Mallet wrote:
@Ann, I would improve the "Noncompliant Code Example" because when we don't know that's the type of time_limit variable is int, the example is meaningless. 

