include::../description.adoc[]

== Noncompliant Code Example

[source,php]
----
$name = $_GET["name"];
echo "Welcome $name"; // Noncompliant
----

== Compliant Solution

[source,php]
----
$name = $_GET["name"];
$safename = htmlspecialchars($name);
echo "Welcome $safename";
----

include::../see.adoc[]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../highlighting.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
