<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>According to the documentation,</p>
</div>
<div class="quoteblock">
<blockquote>
<div class="paragraph">
<p>A program may produce unpredictable results if it attempts to distinguish two references to equal values of a value-based class, whether directly via reference equality or indirectly via an appeal to synchronization&#8230;&#8203;</p>
</div>
</blockquote>
</div>
<div class="paragraph">
<p>This is because value-based classes are intended to be wrappers for value types, which will be primitive-like collections of data (similar to <code>struct</code>s in other languages) that will come in future versions of Java.</p>
</div>
<div class="quoteblock">
<blockquote>
<div class="paragraph">
<p>Instances of a value-based class &#8230;&#8203;</p>
</div>
<div class="ulist">
<ul>
<li>
<p>do not have accessible constructors, but are instead instantiated through factory methods which make no commitment as to the identity of returned instances;</p>
</li>
</ul>
</div>
</blockquote>
</div>
<div class="paragraph">
<p>This means that you can&#8217;t be sure you&#8217;re the only one trying to lock on any given instance of a value-based class, opening your code up to contention and deadlock issues.</p>
</div>
<div class="paragraph">
<p>Under Java 8 breaking this rule may not actually break your code, but there are no guarantees of the behavior beyond that.</p>
</div>
<div class="paragraph">
<p>This rule raises an issue when a known value-based class is used for synchronization. That includes all the classes in the <code>java.time</code> package except <code>Clock</code>; the date classes for alternate calendars, <code>HijrahDate</code>, <code>JapaneseDate</code>, <code>MinguoDate</code>, <code>ThaiBuddhistDate</code>; and the optional classes: <code>Optional</code>, <code>OptionalDouble</code>, <code>OptionalLong</code>, <code>OptionalInt</code>.</p>
</div>
<div class="paragraph">
<p><strong>Note</strong> that this rule is automatically disabled when the project&#8217;s <code>sonar.java.source</code> is lower than <code>8</code>.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">Optional&lt;Foo&gt; fOpt = doSomething();
synchronized (fOpt) {  // Noncompliant
  // ...
}</code></pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p><a href="https://docs.oracle.com/javase/8/docs/api/java/lang/doc-files/ValueBased.html">Value-based classes</a></p>
</li>
</ul>
</div>
<hr>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Synchronize on a non-value-based object; synchronizing on a "xxx" could lead to contention.</p>
</div>
</div>
</div>
</div>