Arrow functions ``++=>++`` use a syntax similar to certain comparison operators (``++<=++``, ``++>=++``). This can create confusion when used in certain contexts.


== Noncompliant Code Example

----
let foo = (x) => x ? "Africa" : "Asia";   // Noncompliant
----


== Compliant Solution

----
let foo = (x) => { return x ? "Africa" : "Asia"; };
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

include::highlighting.adoc[]

endif::env-github,rspecator-view[]
