== Why is this an issue?

Arrow functions ``++=>++`` use a syntax similar to certain comparison operators (``++<=++``, ``++>=++``). This can create confusion when used in certain contexts.


=== Noncompliant code example

[source,javascript]
----
let foo = (x) => x ? "Africa" : "Asia";   // Noncompliant
----


=== Compliant solution

[source,javascript]
----
let foo = (x) => { return x ? "Africa" : "Asia"; };
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Refactor this code to disambiguate the use of arrow function.


=== Highlighting

The arrow function "=>"


endif::env-github,rspecator-view[]
