<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>While a file I/O call will compile without a file name, such code is almost always an error. The omission of the file name, can cause unexpected results in multi-file programs.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-rpg" data-lang="rpg">  * Noncompliant
C                   READ                    DS1                      90</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-rpg" data-lang="rpg">C                   READ      PF0001        DS1                      90</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_7_apr_2015_092211_pierre_yves_nicolas_wrote">on 7 Apr 2015, 09:22:11 Pierre-Yves Nicolas wrote:</h3>
<div class="paragraph">
<p>I don&#8217;t manage to compile the noncompliant code example.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_8_apr_2015_151444_ann_campbell_wrote">on 8 Apr 2015, 15:14:44 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~pierre-yves.nicolas] this rule is based on this: https://scs.senecac.on.ca/~timothy.mckenna/RPG544/rpg_cls5.html</p>
</div>
<div class="paragraph">
<p>Perhaps this only applies to some (one?) compilers&#8230;&#8203;?</p>
</div>
<div class="paragraph">
<p>I&#8217;m fine to drop this rule if you like.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_8_apr_2015_154119_ann_campbell_wrote">on 8 Apr 2015, 15:41:19 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>BTW, I&#8217;ve just spent probably 20min trying to get the author&#8217;s email address. Apparently it&#8217;s only available if you&#8217;re a student of Seneca College. :-(</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_16_apr_2015_094509_pierre_yves_nicolas_wrote">on 16 Apr 2015, 09:45:09 Pierre-Yves Nicolas wrote:</h3>
<div class="paragraph">
<p>All RPG IV references I could find define the "name" operand as required (e.g. http://publib.boulder.ibm.com/iseries/v5r1/ic2924/books/c0925083696.htm#HDRZZREAD).</p>
</div>
<div class="paragraph">
<p>Maybe this rule applied to RPG III, but we don&#8217;t support RPG III.</p>
</div>
<div class="paragraph">
<p>I would drop this rule.</p>
</div>
</div>
</div>
</div>