include::../description.adoc[]

== Noncompliant Code Example

[source,vbnet]
----
Public Sub DoSomething(Input As String)
    Dim Rx As New Regex("[A")                   ' Noncompliant
    Dim Match = Regex.Match(Input, "[A")        ' Noncompliant
    Dim Matches = Regex.Matches(Input, "[A")    ' Noncompliant
    Dim Replace = Regex.Replace(Input, "[A", "replacement")     ' Noncompliant
    Dim Split = Regex.Split(Input, "[A")        ' Noncompliant
    If (Regex.IsMatch(Input, "[A")) Then        ' Noncompliant
    End If
End Sub
----

== Compliant Solution

[source,vbnet]
----
Public Sub DoSomething(Input As String)
    Dim Rx As New Regex("[A-Z]")
    Dim Match = Regex.Match(Input, "[A-Z]")
    Dim Matches = Regex.Matches(Input, "[A-Z]")
    Dim Replace = Regex.Replace(Input, "[A-Z]", "replacement")
    Dim Split = Regex.Split(Input, "[A-Z]")
    If (Regex.IsMatch(Input, "[A-Z]")) Then
    End If
End Sub
----

include::../rspecator.adoc[]
