Nested components are recreated on each render because React does not know that they haven't changed. This hurts performance as the components are recreated at each render. However, this can also hide surprising bugs where the state of the nested components is lost between renders. Trying to `useCallback` hooks for child components should also be discouraged. You should refactor this code to define the component on its own, passing props if required.

== Noncompliant Code Example

[source,javascript]
----
function Component() {
  function UnstableNestedComponent() {
    return <div/>;
  }

  return (
    <div>
      <UnstableNestedComponent />
    </div>
  );
}
----

[source,javascript]
----
function SomeComponent({ footer: Footer }) {
  return (
    <div>
      <Footer />
    </div>
  );
}

function Component() {
  return (
    <div>
      <SomeComponent footer={ () => <div /> } />
    </div>
  );
}
----

[source,javascript]
----
class Component extends React.Component {
  render() {
    function UnstableNestedComponent() {
      return <div />;
    }

    return (
      <div>
        <UnstableNestedComponent />
      </div>
    );
  }
}
----

== Compliant Solution

[source,javascript]
----
function OutsideDefinedComponent(props) {
  return <div />;
}

function Component() {
  return (
    <div>
      <OutsideDefinedComponent />
    </div>
  );
}
----

[source,javascript]
----
function Component() {
  return <SomeComponent footer={ <div /> } />;
}
----

== See

TODO
