== Why is this an issue?

While named function expressions might be useful for debugging purposes, some browsers do not support them correctly (for example Internet Explorer 8).


=== Noncompliant code example

[source,javascript]
----
f = function fun(){}; // Noncompliant;  named function expression
----


=== Compliant solution

[source,javascript]
----
fun = function(){}; // Compliant; function expression
----


=== Exceptions

ECMAScript 6 generator functions are excluded from this rule.

[source,javascript]
----
function* f() {} // Compliant; generator function.
----


ifdef::env-github,rspecator-view[]
'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
