<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>In JCL, certain parameters are incompatible with each other. For example, in the <code>DD</code> statement, the <code>DYNAM</code> parameter should not have any parameters coded after it, or the <code>DLM</code> parameter is only meaningful with <code>DD *</code> and <code>DD DATA</code>. Using incompatible parameters can lead to unexpected behavior or JCL errors. It can also make the code harder to understand and maintain.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Remove one of the incompatible parameters from the JCL statement. Refer to the issue message and the JCL language reference to find the incompatible parameter pairs.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="paragraph">
<p>In the following example, the <code>DLM</code> parameter is only compatible with <code>DD *</code> and <code>DD DATA</code>.
In addition, <code>SYSOUT</code> and <code>VOLUME</code> are incompatible parameters and coding both of them on the same <code>DD</code> statement will lead to a JCL error.</p>
</div>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-jcl" data-lang="jcl">//STEP1 EXEC PGM=DOTHING
//SYSIN DD DSN=A.B.C,DLM='AA' &lt;-- Noncompliant
//SYSPRINT SYSOUT=*,VOLUME=SER=339006 &lt;-- Noncompliant</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-jcl" data-lang="jcl">//STEP1 EXEC PGM=DOTHING
//SYSIN DD DSN=A.B.C
//SYSPRINT SYSOUT=*</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p><a href="https://www.ibm.com/docs/en/zos/3.1.0?topic=mvs-zos-jcl-reference">z/OS MVS JCL Reference</a></p>
</li>
</ul>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="ulist">
<ul>
<li>
<p>Primary: Remove this '%s' parameter; it is incompatible with an earlier parameter '%s'.</p>
</li>
<li>
<p>Primary for DLM: Remove this 'DLM' parameter; it is only compatible with in-stream data sets.</p>
</li>
<li>
<p>Secondary: Incompatible parameter.</p>
</li>
</ul>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>The primary location should be the parameter, with a secondary location on the earlier parameter.</p>
</div>
</div>
</div>
</div>