<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>When only the condition expression is defined in a <code>for</code> loop, and the initialization and increment expressions are missing, a <code>while</code> loop should be used instead to increase readability.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-text" data-lang="text">for (;condition;) { /*...*/ }</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-text" data-lang="text">while (condition) { /*...*/ }</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Replace this "for" loop with a "while" loop.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_7_jan_2014_105949_freddy_mallet_wrote">on 7 Jan 2014, 10:59:49 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>Is implemented by http://jira.codehaus.org/browse/SONARPLUGINS-3365 for JavaScript</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_8_jan_2014_004329_ann_campbell_wrote">on 8 Jan 2014, 00:43:29 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>I&#8217;m done with this one unless there&#8217;s a decision on whether to quote keywords.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_12_feb_2014_221349_freddy_mallet_wrote">on 12 Feb 2014, 22:13:49 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>Is implemented by http://jira.codehaus.org/browse/SONARPLUGINS-3475 for PHP</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_6_mar_2018_175210_tibor_blenessy_wrote">on 6 Mar 2018, 17:52:10 Tibor Blenessy wrote:</h3>
<div class="paragraph">
<p>This rule is irrelevant for Go, because there are no <code>while</code> loops</p>
</div>
</div>
</div>
</div>