== Why is this an issue?

When you call `empty()` or `is_empty()`, it clearly communicates the intention of the code, which is to check if the collection is empty. Using `size() == 0` for this purpose is less direct and makes the code slightly more complex.

Moreover, depending on the implementation, the `size()`, `length()`, or `count()` methods can have a time complexity of `O(n)` where `n` is the number of elements in the collection. On the other hand, `empty()` and `is_empty()` simply check if there is at least one element in the collection, which is a constant time operation, `O(1)`.

Using `empty()` or `is_empty()` to test for emptiness should therefore be preferred over `size()`, `length()` or `count()`.

== How to fix it

Test the emptiness of your collection using `empty()` or `is_empty()`.

=== Code examples

==== Noncompliant code example

[source,cpp,diff-id=1,diff-type=noncompliant]
----
void fun(const std::vector<int> &myVector) {
  if (myVector.size() == 0) { // Noncompliant
    // do something
  }
}
----

==== Compliant solution

[source,cpp,diff-id=1,diff-type=compliant]
----
void fun(const std::vector<int> &myVector) {
  if (myVector.empty()) {
    // do something
  }
}
----

== Resources

=== External coding guidelines

* {cpp} Core Guidelines - https://github.com/isocpp/CppCoreGuidelines/blob/036324/CppCoreGuidelines.md#reason-382[T.143 - Don't write unintentionally nongeneric code]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Use empty() or is_empty() to check whether the container is empty or not.


'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
