== Why is this an issue?

Using a ``++size()++``, ``++length()++`` or ``++count()++`` method to test for emptiness works, but using an ``++empty()++`` or ``++is_empty()++`` method makes the code more readable and can be more performant. The time complexity of any ``++empty()++``/``++is_empty()++`` method implementation should be ``++O(1)++``, whereas some implementations of ``++size()++``, ``++length()++`` or ``++count()++`` can be ``++O(n)++``.

=== Noncompliant code example

[source,cpp]
----
void fun(const std::vector<int> &myVector) {
  if (myVector.size() == 0) { // Noncompliant
    // do something
  }
}
----

=== Compliant solution

[source,cpp]
----
void fun(const std::vector<int> &myVector) {
  if (myVector.empty()) {
    // do something
  }
}
----

== Resources

* https://github.com/isocpp/CppCoreGuidelines/blob/c553535fb8dda2839d13ab5f807ffbc66b63d67b/CppCoreGuidelines.md#enforcement-324[{cpp} Core Guidelines - T.143] - Don't write unintentionally nongeneric code

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Use empty() or is_empty() to check whether the container is empty or not.


'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
