== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,sql]
----
IF @x > 0 SET @x = 0; IF @y > 0 SET @y = 0; -- Noncompliant
----

=== Compliant solution

[source,sql]
----
IF @x > 0 SET @x = 0;
IF @y > 0 SET @y = 0;
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../highlighting.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
