include::../description.adoc[]

== Noncompliant Code Example

----
IF @x > 0 SET @x = 0; IF @y > 0 SET @y = 0; -- Noncompliant
----

== Compliant Solution

----
IF @x > 0 SET @x = 0;
IF @y > 0 SET @y = 0;
----

ifdef::env-github,rspecator-view[]
'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
