<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Flexible array members are most likely to be used in conjunction with dynamic memory allocation.</p>
</div>
<div class="paragraph">
<p>The presence of flexible array members modifies the behaviour of the <code>sizeof</code> operator in ways that might not be expected by a programmer. The assignment of a structure that contains a flexible array member to another structure of the same type may not behave in the expected manner as it copies only those elements up to but not including the start of the flexible array member.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">#include &lt;stdlib.h&gt;
struct s
{
  uint16_t len;
  uint32_t data[ ]; // Noncompliant - flexible array member
} str;

struct s *copy ( struct s *s1 )
{
  struct s *s2 = malloc ( sizeof ( struct s ) + ( s1-&gt;len * sizeof ( uint32_t ) ) );
  /* Omit malloc ( ) return check for brevity */
  *s2 = *s1; /* Only copies s1-&gt;len */
  return s2;
}</code></pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p>MISRA C:2012, 18.7 - Flexible array members shall not be declared.</p>
</li>
</ul>
</div>
<hr>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove this flexible array member.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_5_jan_2015_125950_samuel_mercier_wrote">on 5 Jan 2015, 12:59:50 Samuel Mercier wrote:</h3>
<div class="paragraph">
<p>I created this rule from MISRA 2012. Can you verify? Thx</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_6_jan_2015_193133_ann_campbell_wrote">on 6 Jan 2015, 19:31:33 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~samuel.mercier], I&#8217;m not sure I&#8217;d have made it on by default, but other than that it looks good.</p>
</div>
</div>
</div>
</div>