<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Getters provide a way to enforce encapsulation by providing methods that give controlled access to struct fields. However, in structs with multiple fields, it is not unusual that copy and paste is used to quickly create the needed getters, which can result in the wrong field being accessed by a getter.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>To fix the issue of incorrect field access in getters, ensure that each getter method correctly accesses the intended field.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-rust" data-lang="rust">struct MyStruct {
    field1: i32,
    field2: i32,
}

impl MyStruct {
    // Incorrectly accessing field2 instead of field1
    fn get_field1(&amp;self) -&gt; i32 {
        self.field2
    }

    fn get_field2(&amp;self) -&gt; i32 {
        self.field2
    }
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-rust" data-lang="rust">struct MyStruct {
    field1: i32,
    field2: i32,
}

impl MyStruct {
    // Correctly accessing field1
    fn get_field1(&amp;self) -&gt; i32 {
        self.field1
    }

    fn get_field2(&amp;self) -&gt; i32 {
        self.field2
    }
}</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Clippy Lints - <a href="https://rust-lang.github.io/rust-clippy/master/index.html#misnamed_getters" class="bare">https://rust-lang.github.io/rust-clippy/master/index.html#misnamed_getters</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>