include::../common/rule.adoc[tag=header]

=== Code examples

==== Noncompliant code example

[source,javascript,diff-id=1,diff-type=noncompliant]
----
const MyComponent = () => {
    return <>
        <a href="javascript:void(0)" onClick={foo}>Perform action</a>
        <a href="#" onClick={foo}>Perform action</a>
        <a onClick={foo}>Perform action</a>
    </>;
};
----

==== Compliant solution

[source,javascript,diff-id=1,diff-type=compliant]
----
const MyComponent = () => {
    return <>
      <button onClick={foo}>Perform action</button>
      <a href="#section" onClick={foo}>Perform action</a>
    </>;
};
----

== Resources
=== Documentation

* WebAIM - https://webaim.org/techniques/hypertext/[Introduction to Links and Hypertext]
* https://marcysutton.com/links-vs-buttons-in-modern-web-applications/[Links vs. Buttons in Modern Web Applications]
* https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/blob/main/docs/rules/anchor-is-valid.md#jsx-a11yanchor-is-valid:~:text=Using%20ARIA%20%2D%20Notes%20on%20ARIA%20use%20in%20HTML[Using ARIA - Notes on ARIA use in HTML]
