<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Running <code>update</code> of your package manager in a single <code>RUN</code> instruction stores the cache index in the file system.
This cache is not needed for the installed software to work properly.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Leaving unnecessary files in Docker image increases its size.
The Docker images should be small and only contain necessary data.
The cache index is obsolete after installation.</p>
</div>
<div class="sect2">
<h3 id="_exceptions">Exceptions</h3>
<div class="paragraph">
<p>The rule does not raise when the update is followed by a <code>gdebi</code> package installation, as it can still install required dependencies when installing from a file, which would require the <code>apt</code> cache to be updated.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-docker" data-lang="docker">RUN apk update
RUN apt-get update
RUN aptitude update</code></pre>
</div>
</div>
<div class="paragraph">
<p>Here each line represents an update command for the most popular package managers.
Each of them stores the cache index in the newly created layer.</p>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-docker" data-lang="docker">RUN apk update &amp;&amp; apk add ...
RUN apt-get update &amp;&amp; apt-get install ...
RUN apt-get update &amp;&amp; gdebi
RUN aptitude update &amp;&amp; aptitude install ...</code></pre>
</div>
</div>
<div class="paragraph">
<p>Here in each line after the update, the package installation is executed.
However, it happens in single <code>RUN</code> instruction so only one layer is created.
After installing all packages the cleanup of the cache index should be done.
For more details please see rule <a data-rspec-id="S6587" class="rspec-auto-link">S6587</a>.</p>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_how_does_this_work">How does this work?</h3>
<div class="paragraph">
<p>Each execution of <code>RUN</code> instruction creates a new layer in Docker.
If a single command <code>apt-get update</code> or equivalent is executed, the cache is stored in the new layer.
This increases the size of the final image.
Even removing those cache in the next <code>RUN</code> instruction doesn&#8217;t decrease the size of the final image.
This overhead is not needed in the Docker image.
Updating the cache and installing packages should be executed in one step (one <code>RUN</code> instruction).</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p><a href="https://docs.docker.com/develop/develop-images/dockerfile_best-practices/#run">RUN - Best practices for writing Dockerfiles</a></p>
</li>
</ul>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Update the cache and install packages in a single RUN instruction.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>Highlight the entire update command.</p>
</div>
<hr>
</div>
</div>
</div>