<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when the same import statement appears multiple times in a Groovy file.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Duplicate import statements create unnecessary code clutter without providing any functional benefit.</p>
</div>
<div class="paragraph">
<p>While duplicate imports don&#8217;t cause compilation errors or runtime issues in Groovy, they make the code harder to read and maintain.</p>
</div>
<div class="paragraph">
<p>Keeping import statements clean and minimal helps developers quickly understand the dependencies of a file and reduces visual noise in the codebase.</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="paragraph">
<p>This issue has minimal impact on functionality but affects code readability and maintainability. Clean import sections help developers understand file dependencies more quickly.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Remove the duplicate import statement. Keep only one instance of each import.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-groovy" data-lang="groovy">import java.util.List
import java.util.Map
import java.util.List // Noncompliant</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-groovy" data-lang="groovy">import java.util.List
import java.util.Map</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>