== Why is this an issue?

Since {cpp}11, type aliases can be declared via ``++using++`` or ``++typedef++``. ``++using++`` should be preferred as  more readable because you see the new name/alias first.

In addition, ``++using++`` can be templated, which makes it applicable to more situations than ``++typedef++``.


=== Noncompliant code example

[source,cpp]
----
typedef void (*FunctionPointerType)(int);
----


=== Compliant solution

[source,cpp]
----
using FunctionPointerType = void (*)(int);
----


== Resources

* https://github.com/isocpp/CppCoreGuidelines/blob/036324/CppCoreGuidelines.md#t43-prefer-using-over-typedef-for-defining-aliases[{cpp} Core Guidelines - T.43] - Prefer using over typedef for defining aliases


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Use 'using' instead of 'typedef' for this type alias


=== Highlighting

The typedef statement


'''
== Comments And Links
(visible only on this page)

=== on 26 Aug 2019, 22:03:26 Lo√Øc Joly wrote:
Can you please review my changes

=== on 27 Aug 2019, 17:16:11 Abbas Sabra wrote:
Looks good to me

=== on 9 Sep 2019, 20:43:09 Ann Campbell wrote:
\[~abbas.sabra], [~loic.joly] the `user-experience` tag is generally reserved for end-user experience, not the experience of API users. :-D

I suggest instead 'design' which has been used for API and overall program design.

endif::env-github,rspecator-view[]
