Since {cpp}11, type aliases can be declared via ``++using++`` or ``++typedef++``. ``++using++`` should be preferred as  more readable because you see the new name/alias first.

In addition, ``++using++`` can be templated, which makes it applicable to more situations than ``++typedef++``.


== Noncompliant Code Example

----
typedef void (*FunctionPointerType)(int);
----


== Compliant Solution

----
using FunctionPointerType = void (*)(int);
----


== See

* https://github.com/isocpp/CppCoreGuidelines/blob/036324/CppCoreGuidelines.md#t43-prefer-using-over-typedef-for-defining-aliases[{cpp} Core Guidelines - T.43] - Prefer using over typedef for defining aliases


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

include::highlighting.adoc[]

'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
