include::../description.adoc[]

== Noncompliant Code Example

With default provided regular expression: ^[a-z][a-zA-Z0-9]*$
----
function DoSomething(){...} 
----

== Compliant Solution

----
function doSomething(){...} 
----
