include::../description.adoc[]

== Noncompliant Code Example

With default provided regular expression: ^[a-z][a-zA-Z0-9]*$

----
function DoSomething(){...} 
----

== Compliant Solution

----
function doSomething(){...} 
----

ifdef::env-github,rspecator-view[]
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
