<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>There is no need to declare the same dependency or plugin twice in a project. In fact, doing so is likely to cause errors in the future when maintainers try to change or upgrade the plugin or dependency.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">    &lt;dependency&gt;
      &lt;groupId&gt;com.mygroup&lt;/groupId&gt;
      &lt;artifactId&gt;myartifact&lt;/artifactId&gt;
      &lt;version&gt;1.0&lt;/version&gt;
      &lt;scope&gt;runtime&lt;/scope&gt;
    &lt;/dependency&gt;
    &lt;!-- ... --&gt;
    &lt;dependency&gt;  &lt;!-- Noncompliant --&gt;
      &lt;groupId&gt;com.mygroup&lt;/groupId&gt;
      &lt;artifactId&gt;myartifact&lt;/artifactId&gt;
      &lt;version&gt;1.0&lt;/version&gt;
      &lt;type&gt;jar&lt;/type&gt;
    &lt;/dependency&gt;</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove this [dependency|plugin] declaration, it duplicates the one on line n.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="ulist">
<ul>
<li>
<p>Primary: &lt;dependency&gt;|&lt;plugin&gt;</p>
</li>
<li>
<p>Secondary: through to &lt;/dependency&gt;|&lt;plugin&gt;</p>
</li>
</ul>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_19_jan_2016_114916_ann_campbell_wrote">on 19 Jan 2016, 11:49:16 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>IntelliJ Duplicate Dependencies, Duplicate plugin declaration</p>
</div>
</div>
</div>
</div>