== Why is this an issue?

Using the ``++=++`` operator to copy the content of an internal table is more efficient than using ``++LOOP++`` + ``++APPEND++`` statements.


=== Noncompliant code example

[source,abap]
----
REFRESH ITAB2.
LOOP AT ITAB1 INTO WA.
  APPEND WA TO ITAB2.
ENDLOOP.
----


=== Compliant solution

[source,abap]
----
ITAB2[] = ITAB1[]
----


ifdef::env-github,rspecator-view[]
'''
== Comments And Links
(visible only on this page)

=== duplicates: S1671

endif::env-github,rspecator-view[]
