Using the ``++=++`` operator to copy the content of an internal table is more efficient than using ``++LOOP++`` + ``++APPEND++`` statements.


== Noncompliant Code Example

----
REFRESH ITAB2.
LOOP AT ITAB1 INTO WA.
  APPEND WA TO ITAB2.
ENDLOOP.
----


== Compliant Solution

----
ITAB2[] = ITAB1[]
----


ifdef::env-github,rspecator-view[]
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
