This rule raises an issue when a public method uses 'def' for parameter types or return types instead of explicit type declarations.

== Why is this an issue?

Using 'def' in public method signatures reduces API clarity and makes it difficult for users to understand the expected parameter types and return types. While 'def' is acceptable for local variables within method scope, it becomes problematic in public APIs because:

* 'def' gets converted to 'Object' in the method signature, obscuring the actual expected types
* API consumers cannot easily determine what types to pass as arguments
* IDE support and code completion become less effective
* The method contract becomes unclear, leading to potential runtime errors
* Documentation and maintainability suffer when type information is implicit

The Groovy documentation explicitly states that omitting types is considered bad practice for public APIs, as it makes the code harder to understand and use correctly.

=== What is the potential impact?

Poor API usability, increased likelihood of runtime type errors, reduced IDE support, and decreased code maintainability. Users of the API may struggle to understand how to call the method correctly, leading to integration issues and debugging difficulties.

== How to fix it

Replace 'def' with explicit type declarations for both parameters and return types in public methods. Use the most specific type that accurately represents the expected input or output.

=== Code examples

==== Noncompliant code example

[source,groovy,diff-id=1,diff-type=noncompliant]
----
public def processData(def input) {
    return input.toString().toUpperCase()
}
----

==== Compliant solution

[source,groovy,diff-id=1,diff-type=compliant]
----
public String processData(Object input) {
    return input.toString().toUpperCase()
}
----

== Resources

=== Documentation

 * Groovy Language Documentation - Optional Typing - https://groovy-lang.org/semantics.html#_optional_typing[Official Groovy documentation explaining when to use explicit types vs 'def' keyword]

 * Groovy Best Practices - Method Signatures - https://groovy-lang.org/style-guide.html[Style guide recommendations for writing clear and maintainable Groovy code]

=== Standards

 * Clean Code - Chapter 3: Functions - https://www.oreilly.com/library/view/clean-code-a/9780136083238/[Principles for writing clean, understandable function signatures]
