<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when <code>readLine()</code> is called on Reader implementations that don&#8217;t support the <code>mark()</code> operation, such as <code>StringReader</code>.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>The <code>readLine()</code> method in Groovy is designed to work optimally with Readers that support the <code>mark()</code> operation, like <code>BufferedReader</code>. When used with Readers that don&#8217;t support marking (such as <code>StringReader</code>), it falls back to a less reliable implementation.</p>
</div>
<div class="paragraph">
<p>This fallback behavior has a significant limitation: it doesn&#8217;t correctly detect multi-character line termination sequences. Specifically, it may not properly handle carriage return followed by line feed (<code>\r\n</code>), which is the standard line ending on Windows systems.</p>
</div>
<div class="paragraph">
<p>This can lead to:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>Incorrect line parsing where a single logical line gets split into multiple lines</p>
</li>
<li>
<p>Data corruption when processing text files with mixed line endings</p>
</li>
<li>
<p>Inconsistent behavior across different operating systems</p>
</li>
<li>
<p>Silent failures that are difficult to detect during testing</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>The issue is particularly problematic because it may work correctly with simple test data but fail with real-world input containing various line ending formats.</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="paragraph">
<p>Using <code>readLine()</code> with unsupported Readers can cause data corruption and inconsistent text processing. Lines may be incorrectly split, leading to malformed data processing and potential application errors. This is especially problematic when handling files from different operating systems or user-generated content with mixed line endings.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Replace <code>readLine()</code> with <code>eachLine()</code> for iterating through lines. The <code>eachLine()</code> method correctly handles all line termination formats and automatically closes the Reader.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-groovy" data-lang="groovy">def reader = new StringReader(text)
def line = reader.readLine() // Noncompliant
while (line != null) {
    processLine(line)
    line = reader.readLine()
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-groovy" data-lang="groovy">new StringReader(text).eachLine { line -&gt;
    processLine(line)
}</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Groovy Reader Documentation - <a href="http://docs.groovy-lang.org/latest/html/groovy-jdk/java/io/Reader.html">Official Groovy documentation for Reader methods, including the readLine() warning and recommended alternatives</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>