<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Since Java 9, <code>@Deprecated</code> has two additional arguments to the annotation:</p>
</div>
<div class="ulist">
<ul>
<li>
<p><code>since</code> allows you to describe when the deprecation took place</p>
</li>
<li>
<p><code>forRemoval</code>, indicates whether the deprecated element will be removed at some future date</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>In order to ease the maintainers work, it is recommended to always add one or both of these arguments.</p>
</div>
<div class="paragraph">
<p>This rule reports an issue when <code>@Deprecated</code> is used without any argument.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">@Deprecated</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">@Deprecated(since="4.2", forRemoval=true)</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_exceptions">Exceptions</h3>
<div class="paragraph">
<p>The members and methods of a deprecated class or interface are ignored by this rule. The classes and interfaces themselves are still subject to it.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_related_rules">Related rules</h3>
<div class="ulist">
<ul>
<li>
<p><a data-rspec-id="S1123" class="rspec-auto-link">S1123</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>