It is a best practice to have variable declarations outside of the loop. Additionally, declaring variables inside a loop is slightly less efficient because memory management is then performed with each iteration of the loop.


== Noncompliant Code Example

----
BEGIN
  FOR i IN 1..10 
  LOOP
    DECLARE
      variableX NUMBER:= 10;
    BEGIN
        variableX:= variableX+i;
        dbms_output.put_line(variableX);
    END;
 END LOOP;
END;
----


== Compliant Solution

----
DECLARE
  variableX NUMBER:= 0;
BEGIN
  FOR i IN 1..10
  LOOP
    variableX:= variableX+1;
    dbms_output.put_line(variableX);
  END LOOP;
END;
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

include::highlighting.adoc[]

endif::env-github,rspecator-view[]
