include::../description.adoc[]

== Noncompliant Code Example

----
function fun($a) {
  $i = 10;
  return $i + $a;
  $i++;             // dead code
}
----

== Compliant Solution

----
function fun($a) {
  $i = 10;
  return $i + $a;
}
----

include::../see.adoc[]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../highlighting.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
