<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>The rule targets test methods that lack an assertion and consist solely of an action and, optionally, a setup.</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-csharp" data-lang="csharp">[TestMethod]
public void Add_SingleNumber_ReturnsSameNumber()
{
    var stringCalculator = new StringCalculator();
    var actual = stringCalculator.Add("0");
}</code></pre>
</div>
</div>
<div class="paragraph">
<p>Such tests only verify that the system under test does not throw any exceptions without providing any guarantees regarding the code&#8217;s behavior under test.
Those tests increase the coverage without enforcing anything on the covered code, resulting in a false sense of security.</p>
</div>
<div class="paragraph">
<p>The rule identifies a potential issue when no assertions are present in tests utilizing the following frameworks:</p>
</div>
<div class="ulist">
<ul>
<li>
<p><code>MSTest</code></p>
</li>
<li>
<p><code>NUnit</code></p>
</li>
<li>
<p><code>xUnit</code></p>
</li>
<li>
<p><code>FluentAssertions</code> (4.x and 5.x)</p>
</li>
<li>
<p><code>NFluent</code></p>
</li>
<li>
<p><code>NSubstitute</code></p>
</li>
<li>
<p><code>Moq</code></p>
</li>
<li>
<p><code>Shoudly</code></p>
</li>
</ul>
</div>
<div class="paragraph">
<p>By enforcing the presence of assertions, this rule aims to enhance the reliability and comprehensiveness of tests by ensuring that they provide meaningful validation of the expected behavior.</p>
</div>
<div class="sect2">
<h3 id="_exceptions">Exceptions</h3>
<div class="paragraph">
<p>Test methods that include a call to a custom assertion method will not raise any issues.</p>
</div>
</div>
<div class="sect2">
<h3 id="_how_can_you_fix_it">How can you fix it?</h3>
<div class="paragraph">
<p>To address this issue, you should include assertions to validate the expected behavior.
Choose an appropriate assertion method provided by your testing framework (such as MSTest, NUnit, xUnit) or select a suitable assertion library like FluentAssertions, NFluent, NSubstitute, Moq, or Shouldly.</p>
</div>
<div class="paragraph">
<p>In addition to using built-in assertion methods, you also have the option to create custom assertion methods. To do this, declare an attribute named <code>[AssertionMethodAttribute]</code> and apply it to the respective method. This allows you to encapsulate specific validation logic within your custom assertion methods without raising the issue.
Here&#8217;s an example:</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-csharp" data-lang="csharp">using System;
using Microsoft.VisualStudio.TestTools.UnitTesting;

[TestClass]
public class CustomTestExample
{
    [TestMethod]
    public void Add_SingleNumber_ReturnsSameNumber()
    {
        var stringCalculator = new StringCalculator();
        var actual = stringCalculator.Add("0");
        Validator.AssertCustomEquality(0, actual); // Compliant
    }
}

public static class Validator
{
    [AssertionMethod]
    public static void AssertCustomEquality(int expected, int actual)
    {
        // ...
    }
}

public class AssertionMethodAttribute : Attribute { }</code></pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it_in_mstest">How to fix it in MSTest</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-csharp" data-lang="csharp">[TestMethod]
public void Add_SingleNumber_ReturnsSameNumber()
{
    var stringCalculator = new StringCalculator();
    var actual = stringCalculator.Add("0");
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-csharp" data-lang="csharp">[TestMethod]
public void Add_SingleNumber_ReturnsSameNumber()
{
    var stringCalculator = new StringCalculator();
    var actual = stringCalculator.Add("0");
    Assert.AreEqual(0, actual);
}</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it_in_nunit">How to fix it in NUnit</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_code_examples_2">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example_2">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-csharp" data-lang="csharp">[Test]
public void Add_SingleNumber_ReturnsSameNumber()
{
    var stringCalculator = new StringCalculator();
    var actual = stringCalculator.Add("0");
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution_2">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-csharp" data-lang="csharp">[Test]
public void Add_SingleNumber_ReturnsSameNumber()
{
    var stringCalculator = new StringCalculator();
    var actual = stringCalculator.Add("0");
    Assert.That(0, Is.EqualTo(actual));
}</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it_in_xunit">How to fix it in xUnit</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_code_examples_3">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example_3">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-csharp" data-lang="csharp">[Fact]
public void Add_SingleNumber_ReturnsSameNumber()
{
    var stringCalculator = new StringCalculator();
    var actual = stringCalculator.Add("0");
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution_3">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-csharp" data-lang="csharp">[Fact]
public void Add_SingleNumber_ReturnsSameNumber()
{
    var stringCalculator = new StringCalculator();
    var actual = stringCalculator.Add("0");
    Assert.Equal(0, actual);
}</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p><a href="https://learn.microsoft.com/en-us/dotnet/core/testing/unit-testing-with-mstest">Unit testing C# with MSTest</a></p>
</li>
<li>
<p><a href="https://learn.microsoft.com/en-us/dotnet/core/testing/unit-testing-with-nunit">Unit testing C# with NUnit</a></p>
</li>
<li>
<p><a href="https://learn.microsoft.com/en-us/dotnet/core/testing/unit-testing-with-dotnet-test">Unit testing C# with xUnit</a></p>
</li>
</ul>
</div>
</div>
<div class="sect2">
<h3 id="_articles_blog_posts">Articles &amp; blog posts</h3>
<div class="ulist">
<ul>
<li>
<p><a href="https://www.everydayunittesting.com/2017/03/unit-testing-anti-pattern-not-asserting.html">Unit Testing anti-pattern</a></p>
</li>
</ul>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Add at least one assertion to this test case.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_2_may_2019_144407_andrei_epure_wrote">on 2 May 2019, 14:44:07 Andrei Epure wrote:</h3>
<div class="paragraph">
<p>This is implemented in <a href="https://github.com/SonarSource/sonar-dotnet/blob/7.13.0.8313/sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/TestMethodShouldContainAssertion.cs">TestMethodShouldContainAssertion </a></p>
</div>
</div>
<div class="sect2">
<h3 id="_on_9_apr_2015_074738_nicolas_peru_wrote">on 9 Apr 2015, 07:47:38 Nicolas Peru wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] I would maybe include a junit3 example in this rule. Otherwise looks fine.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_9_apr_2015_124716_ann_campbell_wrote">on 9 Apr 2015, 12:47:16 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>Per our discussion [~nicolas.peru] I&#8217;m going to skip an explicit JUnit 3 example</p>
</div>
</div>
</div>
</div>