<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>A test case without assertions ensures only that no exceptions are thrown. Beyond basic runnability, it ensures nothing about the behavior of the code under test.</p>
</div>
<div class="paragraph">
<p>This rule raised an issue when no assertions are found within a PHPUnit test method.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-php" data-lang="php">public function testDoSomething() {  // Compliant
  $myClass = new MyClass();
  $myClass-&gt;getSomething();
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-php" data-lang="php">public function testDoSomething() {  // Noncompliant
  $myClass = new MyClass();
  $this-&gt;assertEquals("foo", $myClass-&gt;getSomething());
}</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Add at least one assertion to this test case.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_9_apr_2015_074738_nicolas_peru_wrote">on 9 Apr 2015, 07:47:38 Nicolas Peru wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] I would maybe include a junit3 example in this rule. Otherwise looks fine.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_9_apr_2015_124716_ann_campbell_wrote">on 9 Apr 2015, 12:47:16 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>Per our discussion [~nicolas.peru] I&#8217;m going to skip an explicit JUnit 3 example</p>
</div>
</div>
</div>
</div>