== Why is this an issue?

This rule checks that the specified attributes are present in HTML tags.


=== Noncompliant code example

Given a required attribute list of: ``++img.alt,img.height,img.width++``:

[source,html]
----
<img src="/images/queen.png"> <!-- Noncompliant; missing all required attributes -->
----


=== Compliant solution

[source,html]
----
<img src="/images/queen.png" width="60" height="85" alt="Elizabeth II">
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

include::parameters.adoc[]

'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
