== Why is this an issue?

This rule checks that the specified attributes are present in HTML tags.


=== Noncompliant code example

Given a required attribute list of: ``++img.alt,img.height,img.width++``:

[source,html]
----
<img src="/images/queen.png"> <!-- Noncompliant; missing all required attributes -->
----


=== Compliant solution

[source,html]
----
<img src="/images/queen.png" width="60" height="85" alt="Elizabeth II">
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Add the missing "xxx" attribute to element "yyy".


=== Parameters

.attributes
****

Comma-separated list of tag.attributes that are required. E.G. specify img.alt to require an "alt" attribute in an "img" tag.
****


'''
== Comments And Links
(visible only on this page)

=== on 22 Aug 2014, 10:22:35 Freddy Mallet wrote:
My feedback @Ann:

* I would remove the SQALE mapping because the current one might be misleading
* The default values of the 'attributes' property is missing

=== on 25 Aug 2014, 14:03:41 Ann Campbell wrote:
In the interest of completeness: this RSpec created to document an existing rule in the Web plugin which is neither fish nor fowl (i.e. not a template rule and no defaults)

endif::env-github,rspecator-view[]
