This rule checks that the specified attributes are present in HTML tags.


== Noncompliant Code Example

Given a required attribute list of: ``++img.alt,img.height,img.width++``:

----
<img src="/images/queen.png"> <!-- Noncompliant; missing all required attributes -->
----


== Compliant Solution

----
<img src="/images/queen.png" width="60" height="85" alt="Elizabeth II">
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

include::parameters.adoc[]

'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
