include::../description.adoc[]

== Noncompliant Code Example

----
if condition {
  doSomething()} 
----

== Compliant Solution

----
if condition {
  doSomething()
} 
----

== Exceptions

When blocks are inlined (open and close curly braces on the same line), no issue is triggered.

----
if condition {doSomething()} 
----

ifdef::env-github,rspecator-view[]
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
