== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,swift]
----
if condition {
  doSomething()} 
----

=== Compliant solution

[source,swift]
----
if condition {
  doSomething()
} 
----

=== Exceptions

When blocks are inlined (open and close curly braces on the same line), no issue is triggered.

[source,swift]
----
if condition {doSomething()} 
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
