This rule raises an issue when `<audio>` or `<video>` elements have the `autoplay` attribute.

== Why is this an issue?

Media content that plays automatically can create significant barriers for users with disabilities and negatively impact the user experience.

When audio or video starts playing without user interaction, it can:

* Interfere with screen readers, making it difficult for visually impaired users to navigate the page
* Startle users with hearing aids or cochlear implants
* Cause problems for users with attention disorders who may be distracted by unexpected audio
* Create confusion for users with cognitive disabilities
* Consume mobile data unexpectedly
* Violate user expectations about page behavior

Autoplay also conflicts with the principle that users should control their browsing experience. Many browsers now block autoplay by default, but relying on browser behavior rather than following accessibility guidelines is not a reliable approach.

=== What is the potential impact?

Users with disabilities may be unable to use the website effectively. Screen reader users may lose their place in the content, and users with cognitive or attention disorders may become disoriented. This creates legal compliance risks under accessibility regulations like the Americans with Disabilities Act (ADA) and similar laws worldwide.

== How to fix it

Remove the `autoplay` attribute and provide visible controls for users to start playback manually. Include the `controls` attribute to ensure users can control playback.

=== Code examples

==== Noncompliant code example

[source,html,diff-id=1,diff-type=noncompliant]
----
<video autoplay src="welcome.mp4"></video> <!-- Noncompliant -->
<audio autoplay controls>
  <source src="background.mp3" type="audio/mpeg"> <!-- Noncompliant -->
</audio>
----

==== Compliant solution

[source,html,diff-id=1,diff-type=compliant]
----
<video controls src="welcome.mp4"></video>
<audio controls>
  <source src="background.mp3" type="audio/mpeg">
</audio>
----

== Resources

=== Documentation

 * WCAG 2.2 - Audio Control - https://www.w3.org/TR/WCAG22/#audio-control[Web Content Accessibility Guidelines on controlling audio content]

 * MDN - autoplay attribute - https://developer.mozilla.org/en-US/docs/Web/HTML/Element/video#autoplay[Technical documentation for the autoplay attribute]

 * WebAIM - Multimedia Accessibility - https://webaim.org/techniques/multimedia/[Guidelines for making multimedia content accessible]

=== Standards

 * WCAG 2.2 - 1.4.2 Audio Control - https://www.w3.org/TR/WCAG22/#audio-control[Success criterion requiring user control over audio that plays automatically]
