<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Identifiers declared with <a href="https://www.ibm.com/support/knowledgecenter/en/SSY2V3_5.1.0/com.ibm.ent.pl1.zos.doc/lr/value.html"><code>STATIC</code></a> and <code>INIT</code> are initialized only once and may be later modified. Modifying such an identifier will fail at runtime when a CICS application is <a href="https://www.ibm.com/docs/en/cics-ts/5.6?topic=applications-multithreading-reentrant-quasi-reentrant-threadsafe-programs">executed as reentrant</a>. It is safer to declare the value as <a href="https://www.ibm.com/support/knowledgecenter/en/SSY2V3_5.1.0/com.ibm.ent.pl1.zos.doc/lr/value.html">a named constant</a> (using the keyword <code>VALUE</code>) because any modification will be reported at compilation time.</p>
</div>
<div class="paragraph">
<p>This rule raises an issue when an identifier is declared as <code>STATIC</code> with <code>INIT</code> or <code>INITIAL</code> and has one of the following types:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>CHARACTER/CHAR</p>
</li>
<li>
<p>BINARY/BIN</p>
</li>
<li>
<p>DECIMAL/DEC</p>
</li>
<li>
<p>FLOAT</p>
</li>
<li>
<p>POINTER/PTR</p>
</li>
<li>
<p>OFFSET</p>
</li>
</ul>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-pli" data-lang="pli">dcl returnCodeOk fixed bin(15) static init (00); /* Noncompliant */
/* or */
dcl actions char(05) init('12345') static; /* Noncompliant */
dcl actions_in char(01);
....
valid_action = (verify(action_in, actions) = 0);</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-pli" data-lang="pli">dcl returnCodeOk fixed bin(15) value (00);
/* or */
dcl actions char(05) value('12345');
dcl actions_in char(01);
....
valid_action = (verify(action_in, actions) = 0);</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Use "VALUE" instead of "INIT STATIC".</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>The declaration</p>
</div>
</div>
</div>
</div>