<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when a date format string has the 24-hour code with the AM/PM marker or the 12-hour code without the AM/PM marker.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>The <code>%p</code> directive in the <code>strftime</code> method is used to represent the AM/PM marker in a time string. It is commonly used in conjunction with the <code>%I</code> directive, which represents the hour in a 12-hour clock.</p>
</div>
<div class="paragraph">
<p>Using the 24-hour format directive with an AM/PM marker can lead to unwanted results e.g.:</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">time_string = time(16,0).strftime("%H:%M %p")
print(time_string)</code></pre>
</div>
</div>
<div class="paragraph">
<p>will print <code>16:00 PM</code> which makes no sense.</p>
</div>
<div class="paragraph">
<p>On the other hand the AM/PM marker is needed when the 12-hour format directive is used to show complete information about an hour e.g.:</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">time_string = time(16,0).strftime("%I:%M")
print(time_string)</code></pre>
</div>
</div>
<div class="paragraph">
<p>will print 04:00 without indicating if the time is in the morning or the afternoon.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Use either the 12-hour time format with an AM/PM marker or the 24-hour format without.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">from datetime import time

def foo():
    t = time(16, 0)
    formatted_time1 = t.strftime("%H:%M %p") # Noncompliant: 16:00 PM
    formatted_time2 = t.strftime("%I:%M") # Noncompliant: 04:00</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">from datetime import time

def foo():
    t = time(16, 0)
    formatted_time1 = t.strftime("%I:%M %p") # OK: 04:00 PM
    formatted_time2 = t.strftime("%H:%M") # OK: 16:00</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Python documentation - <a href="https://docs.python.org/3/library/datetime.html#strftime-and-strptime-format-codes">strftime() and strptime() Format Codes</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>