== Why is this an issue?

Shared coding conventions allow teams to collaborate efficiently. This rule checks that curly braces are omitted from interfaces with no instance variables. 


Using curly braces in such a situation means that the reader of the code must pause to find the close curly brace before understanding that there are no variables. On the other hand, omitting the curly braces is a quick, clear indicator that there are no variables. 


=== Noncompliant code example

[source,cpp]
----
@interface Foo : NSObject { // Noncompliant
}

-(void) doSomething;

@end
----


=== Compliant solution

[source,cpp]
----
@interface Foo : NSObject

-(void) doSomething;

@end
----



ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
