<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Theoretically, <code>CASE</code>, <code>COALESCE</code>, and <code>IIF</code> evaluate conditions sequentially, stopping with the first one that evaluates to <code>TRUE</code>. In reality, these expressions evaluate aggregate functions and non-native service calls such as <code>CONTAIN</code> and <code>FREETEXT</code> first, and <em>then</em> pass the results into the expression. That means that if you&#8217;re relying on short-circuit behavior to avoid runtime errors with these arguments, you will not get it. You can work around the issue by wrapping such calls in a subselect.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-sql" data-lang="sql">SELECT
  CASE
    WHEN @i = 0
      THEN 1
    ELSE MIN(1/@i)  -- Noncompliant
  END;</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-sql" data-lang="sql">SELECT
  CASE
    WHEN @i = 0
      THEN 1
    ELSE (SELECT MIN(1/@i))
  END;</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Move this [aggregate function|non-native service] into a sub-select to avoid out-of-order evaluation.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>aggregate function / non-native service call</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_19_jul_2017_164308_ann_campbell_wrote">on 19 Jul 2017, 16:43:08 Ann Campbell wrote:</h3>
<div class="paragraph">
<p><a href="https://sqlperformance.com/2014/06/t-sql-queries/dirty-secrets-of-the-case-expression" class="bare">https://sqlperformance.com/2014/06/t-sql-queries/dirty-secrets-of-the-case-expression</a></p>
</div>
<div class="paragraph">
<p><a href="https://connect.microsoft.com/SQLServer/feedback/details/690017/case-coalesce-wont-always-evaluate-in-textual-order" class="bare">https://connect.microsoft.com/SQLServer/feedback/details/690017/case-coalesce-wont-always-evaluate-in-textual-order</a></p>
</div>
<div class="paragraph">
<p><a href="https://connect.microsoft.com/SQLServer/feedback/details/691535/aggregates-dont-follow-the-semantics-of-case" class="bare">https://connect.microsoft.com/SQLServer/feedback/details/691535/aggregates-dont-follow-the-semantics-of-case</a></p>
</div>
<div class="paragraph">
<p><a href="https://connect.microsoft.com/SQLServer/feedback/details/780132/" class="bare">https://connect.microsoft.com/SQLServer/feedback/details/780132/</a></p>
</div>
<div class="paragraph">
<p><a href="https://learningintheopen.org/2013/11/22/technical-microsoft-sql-server-full-text-search-and-the-problem-with-nullempty-predicates/" class="bare">https://learningintheopen.org/2013/11/22/technical-microsoft-sql-server-full-text-search-and-the-problem-with-nullempty-predicates/</a></p>
</div>
</div>
</div>
</div>