This rule raises an issue when the AWS Lambda handler function is declared with `async`.

== Why is this an issue?

The standard AWS Lambda Python runtime is designed to invoke a synchronous handler function. 
While Python's `asyncio` library can be used inside a synchronous handler (by calling `asyncio.run()`), 
the handler function itself cannot be declared with `async def`. Doing so will lead to a runtime error.


== How to fix it

To fix this issue, remove the `async` keyword from the handler function definition.
The asynchronous part of the code can be moved as well 

=== Code examples

==== Noncompliant code example

[source,python,diff-id=1,diff-type=noncompliant]
----
def some_logic():
  ...

async def lambda_handler(event, context): # Noncompliant: the handler is defined with async
  result = some_logic()
  return {"status": result}
----

==== Compliant solution

[source,python,diff-id=1,diff-type=compliant]
----
import asyncio

async def some_logic():
  ...

def lambda_handler(event, context):
  result = asyncio.run(some_logic())
  return {"status": result}
----

//=== How does this work?

//=== Pitfalls

//=== Going the extra mile


//== Resources
//=== Documentation
//=== Articles & blog posts
//=== Conference presentations
//=== Standards
//=== External coding guidelines
//=== Benchmarks
