== Why is this an issue?

It is undefined behaviour if an identifier is used for which multiple definitions exist (in different translation units) or no definition exists at all. With the exception of templates and inline functions, multiple definitions in different translation units are not permitted, even if the definitions are the same.


=== Noncompliant code example

[source,cpp]
----
// file1.cpp
int32_t i = 0;

// file2.cpp
int32_t i = 1; // Noncompliant, redefinition
----


== Resources

* MISRA C:2004, 8.9
* MISRA {cpp}:2008, 3-2-4
* MISRA C:2012, 8.6


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
