<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Dependency injection is a software design pattern in which one or more dependencies (or services) are injected, or passed by reference, into a dependent object (or client) and are made part of the client&#8217;s state. The pattern separates the creation of a client&#8217;s dependencies from its own behavior, which allows program designs to be loosely coupled and to follow the dependency inversion and single responsibility principles.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-text" data-lang="text">class SomeClass {

  public function __construct() {
    $this-&gt;object = new SomeOtherClass();  // Noncompliant
  }
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-text" data-lang="text">class SomeClass {

  public function __construct(SomeOtherClass $object) {
    $this-&gt;object = $object;
  }
}</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove this object instantiation; make "xxx" a parameter instead.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_15_apr_2015_121547_linda_martin_wrote">on 15 Apr 2015, 12:15:47 Linda Martin wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] assign for review.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_20_apr_2015_114808_ann_campbell_wrote">on 20 Apr 2015, 11:48:08 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>Updated [~linda.martin]. You&#8217;ll want to check it over. Also, 30min seemed like a high remediation cost to me &amp; I&#8217;ve downgraded. Feel free to re-upgrade if you disagree.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_19_may_2015_142628_linda_martin_wrote">on 19 May 2015, 14:26:28 Linda Martin wrote:</h3>
<div class="paragraph">
<p>Reviewed.</p>
</div>
</div>
</div>
</div>