== Why is this an issue?

Calling an overridable method from a constructor could result in failures or strange behaviors when instantiating a subclass which overrides the method.


For example:

* The subclass class constructor starts by calling the parent class constructor.
* The parent class constructor calls the method, which has been overridden in the child class.
* If the behavior of the child class method depends on fields that are initialized in the child class constructor, unexpected behavior (like a ``++NullReferenceException++``) can result, because the fields aren't initialized yet.

=== Noncompliant code example

[source,csharp]
----
public class Parent 
{
  public Parent() 
  {
    DoSomething();  // Noncompliant
  }

  public virtual void DoSomething() // can be overridden
  {  
    ...
  }
}

public class Child : Parent 
{
  private string foo;

  public Child(string foo) // leads to call DoSomething() in Parent constructor which triggers a NullReferenceException as foo has not yet been initialized
  {
    this.foo = foo;
  }

  public override void DoSomething() 
  {
    Console.WriteLine(this.foo.Length);
  }
}
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

=== on 23 Jun 2015, 07:47:21 Tamas Vajk wrote:
\[~ann.campbell.2] Other then changing the code sample and removing the "by contract" from the first bullet point, I've added this rule to the SonarQube Way profile. I think this is a really useful rule, which should be enabled by default.


The main difference between C# and Java in this case is that you need to opt in for overridable methods. So you explicitly need to say that ``++DoSomething++`` is ``++virtual++``. So in case of Java you might have just forget to add the ``++final++`` to the method, but in C# you explicitly stated that you are calling and overridable method from the constructor.

=== on 23 Jun 2015, 11:51:47 Ann Campbell wrote:
looks good to me [~tamas.vajk]

endif::env-github,rspecator-view[]
