<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when an exception class is duplicated in an <code>except</code> statement, or when an exception class has a parent class in the same <code>except</code> statement.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>In Python it is possible to catch multiple types of exception in a single <code>except</code> statement using a tuple of the exceptions.</p>
</div>
<div class="paragraph">
<p>Repeating an exception class in a single <code>except</code> statement will not fail but it does not have any effect. Either the exception class is not the one which should be caught, or it is duplicated code which should be removed.</p>
</div>
<div class="paragraph">
<p>Having a subclass and a parent class in the same <code>except</code> statement does not provide any benefit either. It is enough to keep only the parent class.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">try:
    ...
except (TypeError, TypeError):  # Noncompliant: duplicated code or incorrect exception class.
    print("Foo")

try:
    ...
except (NotImplementedError, RuntimeError):  # Noncompliant: NotImplementedError inherits from RuntimeError.
    print("Foo")</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">try:
    ...
except (TypeError, ValueError):
    print("Foo")

try:
    ...
except RuntimeError:
    print("Foo")</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Python Documentation - <a href="https://docs.python.org/3/tutorial/errors.html#handling-exceptions">Handling Exceptions</a></p>
</li>
<li>
<p>Python Documentation - <a href="https://docs.python.org/3/library/exceptions.html#exception-hierarchy">Exception hierarchy</a></p>
</li>
</ul>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="ulist">
<ul>
<li>
<p>Remove this duplicate Exception class.</p>
</li>
<li>
<p>Remove this useless Exception class; it derives from class XXX which is already caught.</p>
</li>
</ul>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>If the class is a duplicate:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>Primary:</p>
<div class="ulist">
<ul>
<li>
<p>The first time the Exception class appears in the "except" statement.</p>
</li>
</ul>
</div>
</li>
<li>
<p>Secondary:</p>
<div class="ulist">
<ul>
<li>
<p>All the duplicates
message: 'Duplicate.'</p>
</li>
</ul>
</div>
</li>
</ul>
</div>
<div class="paragraph">
<p>If the class inherit from another class in the same "except" statement:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>Primary:</p>
<div class="ulist">
<ul>
<li>
<p>The derived class.</p>
</li>
</ul>
</div>
</li>
<li>
<p>Secondary:</p>
<div class="ulist">
<ul>
<li>
<p>The parent class.
message: 'Parent class.'</p>
</li>
</ul>
</div>
</li>
</ul>
</div>
</div>
</div>
</div>