== Why is this an issue?

The if-`null` operator `??` allows providing a default value when dealing with `null`. It only coalesces when the original value is `null`. Therefore, it is safer and shorter than relying upon testing against `null` or `undefined` explicitly.

This rule reports when conditionals (`?`) can be safely replaced with coalescing (`??`).

== How to fix it

Rewrite the expression using `??` on the unchecked operands.

=== Code examples

==== Noncompliant code example

[source,dart,diff-id=1,diff-type=noncompliant]
----
int either(int? x, int y) {
  return x != null ? x : y;
}
----

==== Compliant solution

[source,dart,diff-id=1,diff-type=compliant]
----
int either(int? x, int y) {
  return x ?? y;
}
----

== Resources

* Dart Docs https://dart.dev/tools/linter-rules/prefer_if_null_operators[Dart Linter rule - prefer_if_null_operators]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

* "Use the '??' operator rather than '?:' when testing for 'null'."

=== Highlighting

The whole ternary operator.

'''

endif::env-github,rspecator-view[]
