The use of unions to access an object in different ways may result in the data being misinterpreted. Therefore, this rule prohibits the use of unions for any purpose.


== Noncompliant Code Example

----
union U1 { // Noncompliant
    float j;
    int i;
};
----


== See

* MISRA C:2004, 18.4 - Unions shall not be used.
* MISRA {cpp}:2008, 9-5-1 - Unions shall not be used.
* MISRA C:2012, 19.2 - The union keyword should not be used


ifdef::env-github,rspecator-view[]
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
