== Why is this an issue?

Having multiple cases in a ``++switch++`` with the same condition is confusing at best. At worst, it's a bug that is likely to induce further bugs as the code is maintained. 


If the first case ends with a break, the second case will never be executed, rendering it dead code. Worse there is the risk in this situation that future maintenance will be done on the dead case, rather than on the one that's actually used.


On the other hand, if the first case does not end with a break, both cases will be executed, but future maintainers may not notice that.


=== Noncompliant code example

[source,flex]
----
switch(i) {
  case 1:
    //...
    break;
  case 5:
    //...
    break;
  case 3:
    //...
    break;
  case 1:  // Noncompliant
    //...
    break;
}
----


=== Compliant solution

[source,flex]
----
switch(i) {
  case 1:
    //...
    break;
  case 5:
    //...
    break;
  case 3:
    //...
    break;
}
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

This case duplicates the case on line n with condition "x".


endif::env-github,rspecator-view[]
