<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Even if there&#8217;s no data to be returned from a procedure, each procedure should at least return a boolean to indicate when procedure execution failed, so callers can respond accordingly.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-rpg" data-lang="rpg">     D MyProc          PI
     D   param1                      10A   const
      /free
        // do work...
        return;  // Noncompliant
      /end-free</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-rpg" data-lang="rpg">     D MyProc          PI              N
     D   param1                      10A   const
      /free
        // do work...
        return *ON;
      /end-free</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Return something from this procedure.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_2_apr_2015_185505_ann_campbell_wrote">on 2 Apr 2015, 18:55:05 Ann Campbell wrote:</h3>
<div class="paragraph">
<p><a href="http://www.bmeyers.net/faqs/14-tips/32-rpg-iv-style?start=2" class="bare">http://www.bmeyers.net/faqs/14-tips/32-rpg-iv-style?start=2</a></p>
</div>
</div>
<div class="sect2">
<h3 id="_on_16_apr_2015_093333_pierre_yves_nicolas_wrote">on 16 Apr 2015, 09:33:33 Pierre-Yves Nicolas wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] I would not activate this rule by default, especially with a critical severity.</p>
</div>
</div>
</div>
</div>