``++CASE++`` statements are useful when there are many different cases depending on the value of the same expression.

For just one or two cases however, the code will be more readable with ``++IF++`` statements.

== Noncompliant Code Example

----
CASE SY-INDEX.
  WHEN ONE.
    WRITE  'One'.
  WHEN 2.
    WRITE  'Two'.
ENDCASE.
----

== Compliant Solution

----
CASE SY-INDEX.
  WHEN ONE.
    WRITE  'One'.
  WHEN 2.
    WRITE  'Two'.
  WHEN OTHERS.
    WRITE 'Unexpected result'
ENDCASE.
----
