== Why is this an issue?

``++CASE++`` statements are useful when there are many different cases depending on the value of the same expression.

For just one or two cases however, the code will be more readable with ``++IF++`` statements.

=== Noncompliant code example

[source,abap]
----
CASE SY-INDEX.
  WHEN ONE.
    WRITE  'One'.
  WHEN 2.
    WRITE  'Two'.
ENDCASE.
----

=== Compliant solution

[source,abap]
----
CASE SY-INDEX.
  WHEN ONE.
    WRITE  'One'.
  WHEN 2.
    WRITE  'Two'.
  WHEN OTHERS.
    WRITE 'Unexpected result'
ENDCASE.
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
