== Why is this an issue?

include::../description.adoc[]


[source,shell]
----
if [ $condition1 ] ; then
  if [ $condition2 ] ; then             # Noncompliant
    # ... 
  fi
fi
----

Will be more readable as

[source,shell]
----
if [ $condition1 && $condition2 ] ; then
    # ... 
fi
----


== How to fix it

If merging the conditions seems to result in a more complex code, extracting the condition or part of it in a named function or variable is a better approach to fix readability.

=== Code examples

==== Noncompliant code example

[source,shell]
----
if [ "$file" != ""] ; then
  if [ $is_file || $is_directory ] ; then       // Noncompliant
    # ... 
  fi
fi
----

==== Compliant solution

[source,shell]
----
is_file_or_directory = $is_file || $is_directory
if [ "$file" != "" && is_file_or_directory ] ; then
  #...
fi
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../highlighting.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
