<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>According to the C&#43;&#43; standard, <code>this</code> can never be null, so comparisons of the two are pointless at best. At worst, because of compiler optimizations, such comparisons could lead to null pointer dereferences or obscure, difficult-to-diagnose errors in production.</p>
</div>
<div class="paragraph">
<p>This rule raises an issue when <code>this</code> is compared to <code>nullptr</code> or 0 or anything <code>#defined</code> as <code>nullptr</code> or 0, such as <code>NULL</code> in most environments.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">class MyClass {
  string name;

  string GetName() {
    if (this != 0) {  // Noncompliant
      return name;
    }
    return 0;
  }
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">class MyClass {
  string name;

  string GetName() {
    return name;
  }
}</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove comparison of "this" with null.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_8_apr_2015_170324_evgeny_mandrikov_wrote">on 8 Apr 2015, 17:03:24 Evgeny Mandrikov wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] in C&#43;&#43; compared to Java, notion of "null" exists in a bit different form(s) - see http://en.cppreference.com/w/cpp/types/NULL</p>
</div>
<div class="paragraph">
<p>So actually wonder - whether we should use term (styled as code) "null", or "nullptr" or "0"? WDYT?</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_8_apr_2015_191045_ann_campbell_wrote">on 8 Apr 2015, 19:10:45 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>I&#8217;ve removed the code formatting on "null" but don&#8217;t want to use 0 instead here because \"<code>this</code> can never be 0" doesn&#8217;t read as smoothly (i.e. takes a little more thought to understand IMO) than \"<code>this</code> can never be null".</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_28_may_2015_122238_evgeny_mandrikov_wrote">on 28 May 2015, 12:22:38 Evgeny Mandrikov wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] I changed title and message according to your last comment, added literal "nullptr" into description and did simplification of examples. Could you please check?</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_29_may_2015_123458_ann_campbell_wrote">on 29 May 2015, 12:34:58 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>looks good [~evgeny.mandrikov]</p>
</div>
</div>
</div>
</div>