include::../description.adoc[]

== Noncompliant Code Example

With the default regular expression ``++^[A-Z][a-zA-Z0-9]*$++``:

----
interface myInterface {...} // Noncompliant
----

== Compliant Solution

----
interface MyInterface {...}
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../parameters.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
