<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>When a class has all <code>final</code> fields, the compiler ensures that the object&#8217;s state remains constant. It also enforces a clear design intent of
immutability, making the class easier to reason about and use correctly.</p>
</div>
<div class="paragraph">
<p>Exceptions are meant to represent the application&#8217;s state at the point at which an error occurred.
Making all fields in an <code>Exception</code> class <code>final</code> ensures that these class fields do not change after initialization.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">public class MyException extends Exception {

  private int status;                        // Noncompliant

  public MyException(String message) {
    super(message);
  }

  public int getStatus() {
    return status;
  }

  public void setStatus(int status) {
    this.status = status;
  }

}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">public class MyException extends Exception {

  private final int status;                  // Compliant

  public MyException(String message, int status) {
    super(message);
    this.status = status;
  }

  public int getStatus() {
    return status;
  }

}</code></pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p>Effective Java 3rd Edition, Joshua Bloch - Exceptions - Item 76 : Strive for failure atomicity</p>
</li>
</ul>
</div>
<hr>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Make this field "xxxx" final.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_30_jul_2013_165545_freddy_mallet_wrote">on 30 Jul 2013, 16:55:45 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>Is implemented by http://jira.codehaus.org/browse/SONARJAVA-254</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_14_mar_2017_114035_amaury_levé_wrote">on 14 Mar 2017, 11:40:35 Amaury Levé wrote:</h3>
<div class="paragraph">
<p>\[~valeri.hristov] This seems to be a nice rule to implement for C#. WDYT?</p>
</div>
</div>
</div>
</div>