== How to fix it in Cryptodome

=== Code examples

include::../../common/fix/code-rationale.adoc[]

==== Noncompliant code example

[source,python,diff-id=1,diff-type=noncompliant]
----
from Crypto.Cipher import DES # pycryptodome
from Cryptodome.Cipher import DES # pycryptodomex

cipher = DES.new(key, DES.MODE_OFB) # Noncompliant
----

==== Compliant solution

[source,python,diff-id=1,diff-type=compliant]
----
sodium_crypto_aead_aes256gcm_encrypt($plaintext, '', $nonce, $key);
----

=== How does this work?

include::../../common/fix/strong-cryptography.adoc[]

