== How to fix it in PyCrypto

=== Code examples

include::../../common/fix/code-rationale.adoc[]

==== Noncompliant code example

[source,python,diff-id=21,diff-type=noncompliant]
----
from Crypto.Cipher import DES

cipher = DES.new(key, DES.MODE_OFB) # Noncompliant
----

==== Compliant solution

PyCrypto is deprecated, thus it is recommended to use another library like PyCryptodome.

[source,python,diff-id=21,diff-type=compliant]
----
from Crypto.Cipher import AES # pycryptodome
from Cryptodome.Cipher import AES # pycryptodomex

cipher = AES.new(key, AES.MODE_CCM)
----

=== How does this work?

include::../../common/fix/strong-cryptography.adoc[]

