== How to fix it in PyCrypto

=== Code examples

include::../../common/fix/code-rationale.adoc[]

==== Noncompliant code example

[source,python,diff-id=21,diff-type=noncompliant]
----
from Crypto.Cipher import DES

cipher = DES.new(key) # Noncompliant
----

==== Compliant solution

PyCrypto is deprecated, thus it is recommended to use another library like pyca.

[source,python,diff-id=21,diff-type=compliant]
----
from cryptography.hazmat.primitives.ciphers import Cipher, algorithms, modes
from cryptography.hazmat.backends import default_backend

cipher = Cipher(algorithms.AES(key), modes.CBC(iv), backend=default_backend())
----

=== How does this work?

include::../../common/fix/strong-cryptography.adoc[]

