== How to fix it in pyDes

=== Code examples

include::../../common/fix/code-rationale.adoc[]

==== Noncompliant code example

[source,python,diff-id=31,diff-type=noncompliant]
----
import pyDes

cipher = pyDes.des(key) # Noncompliant
----

==== Compliant solution

Since pyDes only provides DES, it is recommended to use another library like ``pyca/cryptography``.

[source,python,diff-id=31,diff-type=compliant]
----
from cryptography.hazmat.primitives.ciphers import Cipher, algorithms, modes

cipher = Cipher(algorithms.AES(key), modes.CTR(nonce))
----

In this example, the counter mode (CTR) of AES is used. This mode takes a cryptographic nonce (a **n**umber used only **once**) as its initialization vector (IV). This value must never be reused, as doing so allows attackers to decrypt any message encrypted with the same key.


=== How does this work?

include::../../common/fix/strong-cryptography.adoc[]

