== How to fix it in PyOpenSSL

=== Code examples

include::../../common/fix/code-rationale.adoc[]

==== Noncompliant code example

[source,python,diff-id=51,diff-type=noncompliant]
----
from OpenSSL import SSL

ciphers = 'RC4-SHA:RC4-MD5'
ctx = SSL.Context(SSL.TLS1_3_VERSION)
ctx.set_cipher_list(ciphers.encode())  # Noncompliant
----

==== Compliant solution

[source,python,diff-id=51,diff-type=compliant]
----
from OpenSSL import SSL

ctx = SSL.Context(SSL.TLS1_3_VERSION)
----

=== How does this work?

It is recommended to not override the ciphers but instead, use the secure
default ciphers of the module, as they might change over time.
