== How to fix it in ssl

=== Code examples

include::../../common/fix/code-rationale.adoc[]

==== Noncompliant code example

[source,python,diff-id=41,diff-type=noncompliant]
----
import ssl

ciphers = 'RC4-SHA:RC4-MD5'
ctx = ssl.create_default_context()
ctx.set_ciphers(ciphers)  # Noncompliant
----

==== Compliant solution

[source,python,diff-id=41,diff-type=compliant]
----
import ssl

ctx = ssl.create_default_context()
----

=== How does this work?


It is recommended to not override the ciphers but instead, use the secure default ciphers of the module, as they might change over time. If specific cipher suites need to be enabled or disabled, then this is also possible by adding them after the `DEFAULT` cipher suite string.

For example, `DEFAULT:!RSA:!SHA` enables all default cipher suites except those using RSA and SHA1. `DEFAULT:HIGH+AESGCM` enables all default cipher suites, as well as all high encryption cipher suites that use AES-GCM.

More information about possible options can be found in the https://www.openssl.org/docs/man1.1.1/man1/ciphers.html#CIPHER-LIST-FORMAT[OpenSSL documentation].
