<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>String interpolation should be used only when there is a mix of variables and string in the same expression.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>String interpolation is a powerful feature in Dart that allows you to embed expressions within string literals.</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-dart" data-lang="dart">final expression = 'expressions';
print('a string mixing ${"literals"} and ${2 + 1} ${expressions}');
// Prints 'a string mixing literals and 3 expressions'</code></pre>
</div>
</div>
<div class="paragraph">
<p>However, using string interpolation where it is not needed can make the code harder to read and maintain.</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-dart" data-lang="dart">final aString = 'a string';
print('$aString'); // Unnecessary string interpolation</code></pre>
</div>
</div>
<div class="paragraph">
<p>It is recommended to use string interpolation only when there is a mix of variables and strings in the same expression.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Remove the string interpolation and use the interpolated string directly.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-dart" data-lang="dart">final aString = 'a string';
print('$aString'); // Non compliant</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-dart" data-lang="dart">final aString = 'a string';
print(aString);</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Dart Docs - <a href="https://dart.dev/tools/linter-rules/unnecessary_string_interpolations">Linter rule - unnecessary_string_interpolations</a></p>
</li>
<li>
<p>Dart Docs - <a href="https://dart.dev/language/built-in-types#strings">Language - Strings</a></p>
</li>
</ul>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Unnecessary use of string interpolation.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>The entire string, including the string delimiters (<code>'</code>, <code>"</code>, <code>'''</code> or <code>"""</code>).</p>
</div>
</div>
</div>
</div>