== How to fix it in OpenSSL

=== Code examples

include::../../common/fix/code-rationale.adoc[]

:cert_variable_name: SSL_set1_host
:cert_variable_safe_value: the name of the expected host 

include::../../common/fix/code-rationale-explicit.adoc[]

In case the hostnames are expected not to be prealably known, it is important
to still check the hostnames depending on what is presented in their
certificates.

==== Noncompliant code example

[source,cpp,diff-id=1,diff-type=noncompliant]
----
#include <openssl/ssl.h>

void connect() {
    const SSL_METHOD *method = TLS_method();
    SSL_CTX *ctx = SSL_CTX_new(method);
    SSL *ssl = SSL_new(ctx);

    SSL_connect(ssl); // Noncompliant
}
----

==== Compliant solution

[source,cpp,diff-id=1,diff-type=compliant]
----
#include <openssl/ssl.h>

void connect() {
    const SSL_METHOD *method = TLS_method();
    SSL_CTX *ctx = SSL_CTX_new(method);
    SSL *ssl = SSL_new(ctx);

    SSL_set1_host(ssl, "smtp.example.com");
    SSL_connect(ssl);
}
----

=== How does this work?

include::../../common/fix/validation.adoc[]
