== How to fix it in OpenSSL

=== Code examples

include::../../common/fix/code-rationale.adoc[]

TODO DEFAULT STUFF

==== Noncompliant code example

[source,cpp,diff-id=1,diff-type=noncompliant]
----
#include <openssl/ssl.h>

void connect() {
    const SSL_METHOD *method = TLS_method();
    SSL_CTX *ctx = SSL_CTX_new(method);
    SSL *ssl = SSL_new(ctx);

    SSL_connect(ssl); // Noncompliant
}
----

==== Compliant solution

[source,cpp,diff-id=1,diff-type=compliant]
----
#include <openssl/ssl.h>

void connect() {
    const SSL_METHOD *method = TLS_method();
    SSL_CTX *ctx = SSL_CTX_new(method);
    SSL *ssl = SSL_new(ctx);

    SSL_set1_host(ssl, HOST_NAME);
    SSL_connect(ssl);
}
----

=== How does this work?

include::../../common/fix/validation.adoc[]
