This rule raises an issue when static imports are declared after regular (non-static) imports in a source file.

== Why is this an issue?

Import statements should follow a consistent order to keep code readable and easy to scan. In Java and Groovy, the usual convention is to place all static imports before regular imports.

== How to fix it

Place all static imports together before regular imports in the import section.

=== Code examples

==== Noncompliant code example

[source,groovy,diff-id=1,diff-type=noncompliant]
----
import java.util.ArrayList
import static java.lang.Math.PI // Noncompliant
----

==== Compliant solution

[source,groovy,diff-id=1,diff-type=compliant]
----
import static java.lang.Math.PI
import java.util.ArrayList
----


