Methods that are never executed are dead code: unnecessary, inoperative code that should be removed.Cleaning out dead code decreases the size of the maintained codebase, making it easier to understand the program and preventing bugs from being introduced.

Note that this rule checks only the component class scope, which means that issues will be raised on the methods that are only accessed outside of the component class.

== Noncompliant Code Example

[source,javascript]
----
FIXME
----

== Compliant Solution

[source,javascript]
----
FIXME
----

== See

FIXME: A list of links