This rule checks that the specified attributes are not present in HTML tags.


== Noncompliant Code Example

For a disallowed attribute list of: ``++a.name++``:

----
<a href="blah.com" name="Blah link"> <!-- Noncompliant; name attribute is used -->
----


== Compliant Solution

----
<a href="blah.com">
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

include::parameters.adoc[]

'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
