== Why is this an issue?

This rule checks that the specified attributes are not present in HTML tags.


=== Noncompliant code example

For a disallowed attribute list of: ``++a.name++``:

[source,html]
----
<a href="blah.com" name="Blah link"> <!-- Noncompliant; name attribute is used -->
----


=== Compliant solution

[source,html]
----
<a href="blah.com">
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Remove the "xxx" attribute from the "yyy" tag


=== Parameters

.attributes
****

Comma-separated list of tag.attributes that are not allowed. E.G. a.name forbids a "name" attribute in an "a" tag.
****


'''
== Comments And Links
(visible only on this page)

=== on 22 Aug 2014, 14:04:56 Freddy Mallet wrote:
@Ann, looks like the list of default values for property 'attributes' is missing. 


BTW, this was not a brand new rule & is not related to HTML 5. It's a catch-up entry for an old rule that was already implemented in the Web plugin but had no RSpec. 

=== on 22 Aug 2014, 19:30:14 Ann Campbell wrote:
\[~freddy.mallet] must there always be a default value? This rule is not on by default, and is a user-specified "don't use this attribute" rule.

=== on 25 Aug 2014, 07:40:07 Freddy Mallet wrote:
Let's say [~ann.campbell.2] that all mandatory properties should have a default value out of the box to prevent generating some analysis failure when activating the rule without providing a property value. But indeed here this is a template rule, so everything is fine. 

=== on 25 Aug 2014, 14:03:28 Ann Campbell wrote:
In the interest of completeness: this RSpec created to document an existing rule in the Web plugin which is neither fish nor fowl (i.e. not a template rule and no defaults)

endif::env-github,rspecator-view[]
