<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>An abstract class represents the interface part of a hierarchy. Invoking the copy constructor from the top of such a hierarchy bypasses the underlying implementation resulting in only the base sub-objects being copied.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">class B1
{
public:
  B1 ( );
  B1 &amp; operator= ( B1 const &amp; rhs ); // Noncompliant
};

class D1 : public B1
{
public:
  D1 &amp; operator= ( D1 const &amp; rhs );
private:
  int32_t member;
};

void f1( B1 &amp; b1, B1 &amp; b2 )
{
  b1 = b2; // operator in B1 will be called: field "member" will not be copied
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">class B1
{
public:
  B1 ( );
  protected B1 &amp; operator= ( B1 const &amp; rhs ); // Compliant
};

class D1 : public B1
{
public:
  D1 &amp; operator= ( D1 const &amp; rhs );
private:
  int32_t member;
};

void f1( B1 &amp; b1, B1 &amp; b2 )
{
  b1 = b2; // Compiler error will be reported
}</code></pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p>MISRA C&#43;&#43;:2008, 12-8-2 - The copy assignment operator shall be declared protected or private in an abstract class.</p>
</li>
</ul>
</div>
<hr>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Mark this copy constructor of an abstract class "protected" or "private".</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_is_related_to_s3657">is related to: <a data-rspec-id="S3657" class="rspec-auto-link">S3657</a></h3>

</div>
</div>
</div>