<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>In java 7 to 9, <code>FileInputStream</code> and <code>FileOutputStream</code> rely on finalization to perform final closes if the stream is not already closed. Whether or not the stream is already closed, the finalizer will be called, resulting in extra work for the garbage collector. This can easily be avoided using the <code>Files</code> API.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">try(FileInputStream fis = new FileInputStream(...)) {  // Noncompliant
} finally {
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">try(InputStream is = Files.newInputStream(...)) {
} finally {
}</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Use "Files.newXXXStream" here instead for better performance.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_21_feb_2019_113251_tobias_gruetzmacher_wrote">on 21 Feb 2019, 11:32:51 Tobias Gruetzmacher wrote:</h3>
<div class="paragraph">
<p>Since I just debugged an issue with that class, I&#8217;d propose to add <code>java.util.zip.ZipFile</code> to the "dangerous" classes with finalizer&#8230;&#8203; Using an <code>java.util.zip.ZipInputStream</code> avoids the finalizer.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_21_feb_2019_115644_alexandre_gigleux_wrote">on 21 Feb 2019, 11:56:44 Alexandre Gigleux wrote:</h3>
<div class="paragraph">
<p>Hello [~tobix],</p>
</div>
<div class="paragraph">
<p>Can you open a thread on https://community.sonarsource.com/c/suggestions/rules so we can get more details about the problem you faced? It&#8217;s much more convenient than using JIRA Comments.</p>
</div>
<div class="paragraph">
<p>Thx</p>
</div>
</div>
</div>
</div>