This ``++BYPASSING BUFFER++`` clause explicitly switches off SAP table buffering, so the ``++SELECT++`` reads data directly from the database.

By definition, using this clause can lead to performance issues, which is why its use must be strongly indicated.


== Noncompliant Code Example

----
SELECT * 
INTO US_PERSONS
FROM PERSONS
BYPASSING BUFFER
WHERE CITY EQ 'US'
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

endif::env-github,rspecator-view[]
