== Why is this an issue?

This ``++BYPASSING BUFFER++`` clause explicitly switches off SAP table buffering, so the ``++SELECT++`` reads data directly from the database.

By definition, using this clause can lead to performance issues, which is why its use must be strongly indicated.


=== Noncompliant code example

[source,abap]
----
SELECT * 
INTO US_PERSONS
FROM PERSONS
BYPASSING BUFFER
WHERE CITY EQ 'US'
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Make sure this use of a "BYPASSING BUFFER" clause is strongly indicated.


endif::env-github,rspecator-view[]
