This rule raises an issue when a test method (annotated with ``++@isTest++`` or using the ``++testMethod++`` keyword) is declared with one or more parameters.

== Why is this an issue?

The Apex testing framework has strict requirements for test method signatures. Test methods must not take any arguments to be recognized and executed by the framework.

When a test method is declared with parameters, the Apex testing framework will not recognize it as a valid test method. This means the test will be silently ignored during test execution, leading to several problems:

* The test will never run, providing false confidence that your code is working correctly
* Code coverage metrics will be inaccurate since the test isn't actually executing
* Bugs in the code being tested may go undetected
* Deployment to production may fail if actual code coverage falls below required thresholds

The framework expects test methods to follow a specific syntax pattern that does not include parameters. This design ensures that tests are self-contained and can be executed automatically without requiring external input.

=== What is the potential impact?

Test methods with parameters will not be executed by the Apex testing framework, leading to false confidence in code quality and potentially allowing buggy code to be deployed to production. This can result in runtime failures and reduced application reliability.

== How to fix it

Remove all parameters from the test method signature. Test methods should be self-contained and not require external input.

=== Code examples

==== Noncompliant code example

[source,apex,diff-id=1,diff-type=noncompliant]
----
@isTest
static void testMethod(String param) { // Noncompliant
    // test code
}
----

==== Compliant solution

[source,apex,diff-id=1,diff-type=compliant]
----
@isTest
static void testMethod() {
    // test code
}
----

== Resources

=== Documentation

 * Apex Testing Framework - https://developer.salesforce.com/docs/atlas.en-us.apexcode.meta/apexcode/apex_testing.htm[Official Salesforce documentation on the Apex testing framework and test method requirements]

 * Test Method Syntax - https://developer.salesforce.com/docs/atlas.en-us.apexcode.meta/apexcode/apex_testing_utility_classes.htm[Detailed documentation on proper test method syntax and structure]
