== Why is this an issue?

Using a named constant to refer to an indicator makes the content of the field clearer, and therefore makes the code easier to read and maintain. 

=== Noncompliant code example

[source,rpg]
----
      /Free

       If *In(25);  // Noncompliant
        // Process contents...
       EndIf;
----

=== Compliant solution

[source,rpg]
----
     D accountTotal    c                   25

      /Free

       If *In(accountTotal);
        // Process contents...
       EndIf;
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Use a named constant for this reference to XXX.


'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
