include::../description.adoc[]

include::../how-to-fix-it.adoc[]

=== Code examples

==== Noncompliant code example

[source,rpg,diff-id=1,diff-type=noncompliant]
----
      /Free

       If *In(25);  // Noncompliant
        // Process contents...
       EndIf;
----

==== Compliant solution

[source,rpg,diff-id=1,diff-type=compliant]
----
     D accountTotal    c                   25

      /Free

       If *In(accountTotal);
        // Process contents...
       EndIf;
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Use a named constant for this reference to XXX.


'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
