Comparisons that always evaluate to true or to false, logical expressions that either always or never short-circuit and comparisons to a newly constructed object should not be used.

== Why is this an issue?

An expression that always produces the same result, regardless of the inputs, is unnecessary and likely indicates a programmer's error. This can come from 

- confusing operator precedence
- expecting strict equality between different types
- expecting objects to be compared by value
- expecting empty objects to be `false` or `null`
- mistyping `>=` for `=>`



=== Code examples

==== Noncompliant code example

[source,javascript,diff-id=1,diff-type=noncompliant]
----
!foo == null;
a + b ?? c; 
x === [];
----

==== Compliant solution

[source,javascript,diff-id=1,diff-type=compliant]
----
foo != null;
a + (b ?? c); 
----
