<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This is an issue when a FastAPI endpoint is configured with <code>status_code=204</code> but does not explicitly return <code>None</code> or use the <code>Response</code> class to ensure an empty body.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>The HTTP 204 (No Content) status code indicates that the server successfully processed the request but is not returning any content. According to <a href="https://tools.ietf.org/html/rfc7231#section-6.3.5">RFC 7231</a>, a 204 response must not contain a message body.</p>
</div>
<div class="paragraph">
<p>In FastAPI, if you define an endpoint with <code>status_code=204</code> but don&#8217;t explicitly handle the response, the framework may send a small response body (typically 4 bytes containing <code>null</code>). This violates the HTTP specification and can cause issues with:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>HTTP clients that strictly validate response formats</p>
</li>
<li>
<p>Proxies and caching layers that expect empty bodies for 204 responses</p>
</li>
<li>
<p>API consumers relying on standard HTTP behavior</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>The problem occurs because FastAPI&#8217;s default serialization behavior attempts to serialize the return value, even when the status code indicates no content should be sent.</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="paragraph">
<p>This issue can lead to:</p>
</div>
<div class="ulist">
<ul>
<li>
<p><strong>Protocol violations</strong>: The response doesn&#8217;t comply with HTTP specifications, which may cause unexpected behavior in strict HTTP clients</p>
</li>
<li>
<p><strong>Interoperability problems</strong>: Some clients, proxies, or middleware may reject or mishandle responses that have a body with a 204 status code</p>
</li>
<li>
<p><strong>Confusion for API consumers</strong>: Developers using your API may be confused by receiving content when the status code explicitly indicates there should be none</p>
</li>
</ul>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it_in_fastapi">How to fix it in FastAPI</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Return <code>None</code> explicitly from the endpoint function. This ensures FastAPI sends an empty response body.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">from fastapi import FastAPI

app = FastAPI()

@app.delete("/resource/{id}", status_code=204)
def delete_resource(id: int):
    # Delete the resource
    ...  # Noncompliant</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">from fastapi import FastAPI

app = FastAPI()

@app.delete("/resource/{id}", status_code=204)
def delete_resource(id: int):
    # Delete the resource
    return None</code></pre>
</div>
</div>
<div class="paragraph">
<p>Use the <code>Response</code> class with <code>status_code=204</code>. This gives you explicit control over the response and guarantees no content is sent.</p>
</div>
</div>
<div class="sect3">
<h4 id="_noncompliant_code_example_2">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">from fastapi import FastAPI

app = FastAPI()

@app.put("/resource/{id}/archive", status_code=204)
def archive_resource(id: int):
    # Archive the resource
    ...  # Noncompliant</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution_2">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">from fastapi import FastAPI, Response

app = FastAPI()

@app.put("/resource/{id}/archive", status_code=204)
def archive_resource(id: int):
    # Archive the resource
    return Response(status_code=204)</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>FastAPI Response Status Code - <a href="https://fastapi.tiangolo.com/advanced/response-change-status-code/">Official FastAPI documentation on changing response status codes</a></p>
</li>
<li>
<p>FastAPI Custom Response - <a href="https://fastapi.tiangolo.com/advanced/custom-response/">Official FastAPI documentation on custom responses</a></p>
</li>
<li>
<p>RFC 7231 - HTTP/1.1 Semantics: 204 No Content - <a href="https://tools.ietf.org/html/rfc7231#section-6.3.5">HTTP specification defining the 204 No Content status code</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>