include::../description.adoc[]

== Noncompliant Code Example

----
fun main() {
  ...
  System.runFinalizersOnExit(true)  // Noncompliant
  ...
}
----


== Compliant Solution

----
fun main() {
    Runtime.getRuntime().addShutdownHook(object : Thread() {
        override fun run() {
            doSomething()
        }
    })
}
----


include::../description-tail.adoc[]
