== How to fix it in AesManaged

=== Code examples

==== Noncompliant code example

[source,vbnet,diff-id=1,diff-type=noncompliant]
----
----

==== Compliant solution

[source,vbnet,diff-id=1,diff-type=compliant]
----
----

=== How does this work?


=== Pitfalls


=== Going the extra mile


