== How to fix it in .Net

=== Code examples

==== Noncompliant code example

include::../../common/fix/aes-noncompliant-example.adoc[]

[source,csharp,diff-id=1,diff-type=noncompliant]
----
Imports System.Security.Cryptography

New AesManaged With {
    .KeySize   = 128,
    .BlockSize = 128,
    .Mode      = CipherMode.ECB,   ' Noncompliant
    .Padding   = PaddingMode.PKCS7
}
----

include::../../common/fix/rsa-noncompliant-example.adoc[]

[source,csharp,diff-id=2,diff-type=noncompliant]
----
Imports System.Security.Cryptography

Dim RsaCsp = New RSACryptoServiceProvider()
RsaCsp.Encrypt(data, False) ' Noncompliant
----


==== Compliant solution

include::../../common/fix/aes-compliant-example.adoc[]

[source,csharp,diff-id=1,diff-type=compliant]
----
Imports System.Security.Cryptography

New AesGcm(key)
----

BouncyCastle can also be used:

[source,csharp]
----
Imports Org.BouncyCastle.Crypto
Imports Org.BouncyCastle.Crypto.Engines
Imports Org.BouncyCastle.Crypto.Modes
Imports Org.BouncyCastle.Crypto.Parameters
                    
Dim GcmBc As GcmBlockCipher = New GcmBlockCipher(New AesFastEngine())

GcmBc.Init(True, New AeadParameters(New KeyParameter(key), 128, iv, Nothing));
----

include::../../common/fix/rsa-compliant-example.adoc[]

[source,csharp,diff-id=2,diff-type=compliant]
----
Imports System.Security.Cryptography

Dim RsaCsp = New RSACryptoServiceProvider()
RsaCsp.Encrypt(data, True)
----

=== How does this work?

include::../../common/fix/fix.adoc[] 
