<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Stream operations are divided into intermediate and terminal operations, and are combined to form stream pipelines.
A stream should be operated on (invoking an intermediate or terminal stream operation) only once.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">Stream&lt;Widget&gt; pipeline = widgets.stream().filter(b -&gt; b.getColor() == Color.RED);
var res1 = pipeline.findAny();
var res2 = pipeline.mapToInt(b -&gt; b.getWeight()).sum(); // Noncompliant</code></pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="paragraph">
<p><a href="https://docs.oracle.com/javase/8/docs/api/java/util/stream/package-summary.html#StreamOps">Stream Operations</a></p>
</div>
<hr>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Refactor this code so that this consumed stream pipeline is not reused.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>Primary: Operation invoked on consumed stream</p>
</div>
<div class="paragraph">
<p>Secondary: The previous terminal operation on that stream</p>
</div>
</div>
</div>
</div>