== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

With a maximum number of 4 parameters:

[source,go]
----
func foo(p1 int, p2 int, p3 int, p4 int, p5 int) { // Noncompliant
  // ...
}
----

=== Compliant solution

[source,go]
----
func foo(p1 int, p2 int, p3 int, p4 int) {
  // ...
}
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../parameters.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
