== Why is this an issue?

Using a function in PHP with the same name as the nesting class was historically used to declare a class constructor.
However, as of PHP 8.0.0, this declaration is discouraged and will provoke an `E_DEPRECATED` error, albeit it functions as a constructor.

Instead, users should explicitly define the constructor by declaring a `\__construct(...)` function.
However, if both styles are present in the same class, PHP will treat the `__construct` function as the class constructor, which can cause unintended behavior.

Adhering to this convention improves readability and maintainability by ensuring that the constructor declaration is named uniformly throughout the codebase.

=== Noncompliant code example

[source,php,diff-id=1,diff-type=noncompliant]
----
class Foo {
  function Foo() {...}
}
----

=== Compliant solution

[source,php,diff-id=1,diff-type=compliant]
----
class Foo {
  function __construct() {...}
}
----

== Resources

=== Documentation

* https://www.php.net/manual/en/language.oop5.decon.php[PHP Manual - Constructors and Destructors]
* https://www.phptutorial.net/php-oop/php-constructors/[PHP Tutorial - Constructors]

ifdef::env-github,rspecator-view[]

'''

== Implementation Specification

(visible only on this page)

include::message.adoc[]

'''

== Comments And Links

(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
