=== Code examples

==== Noncompliant code example

[source,csharp,diff-id=1,diff-type=noncompliant]
----
bool StartsWithA(string s) 
    => s.StartsWith("A");
----

[source,csharp,diff-id=2,diff-type=noncompliant]
----
bool EndsWithA(string s) 
    => s.EndsWith("A");
----

==== Compliant solution

[source,csharp,diff-id=1,diff-type=compliant]
----
bool StartsWithA(string s) 
    => s.StartsWith('A');
----

[source,csharp,diff-id=2,diff-type=compliant]
----
bool EndsWithA(string s) 
    => s.EndsWith('A');
----
