== Why is this an issue?

Using ``++auto++`` when the type that would be deduced is a pointer type can cause confusion. It is much better to specify the pointer part outside of ``++auto++``.


=== Noncompliant code example

[source,cpp]
----
auto item = new Item(); // Noncompliant
----


=== Compliant solution

[source,cpp]
----
auto* item = new Item(); 
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

include::highlighting.adoc[]

endif::env-github,rspecator-view[]
