Setting the environment variable using `ENV` instruction creates a new layer in the Docker image.
Calling `RUN unset <env-variable>` creates a new layer, but the environment variable can be still dumped from the previous layer.

== Why is this an issue?

The environment variables contain very often secrets, tokens, and other sensitive information.
They are present in the containers and could be dumped anytime.
Calling `unset` doesn't prevent this information from being hidden for other commands.

== How to fix it

To prevent runtime access, the environment variable can be set and unset in single `RUN` instruction.

=== Code examples

==== Noncompliant code example

[source,docker,diff-id=1,diff-type=noncompliant]
----
ENV $ADMIN_USER
RUN unset $ADMIN_USER
----

==== Compliant solution

[source,docker,diff-id=1,diff-type=compliant]
----
RUN export ADMIN_USER="admin" \
    && ... \
    && unset ADMIN_USER
----

=== How does this work?

In this example, the visibility of `ADMIN_USER` is only limited to the single layer.
However, the value can be still extracted from the image.
The best solution is to use `ARG` instead.

== Resources
=== Documentation

https://docs.docker.com/engine/reference/builder/#env[ENV - Dockerfile reference]
https://docs.docker.com/develop/develop-images/dockerfile_best-practices/#env[ENV - Best practices for writing Dockerfiles]

ifdef::env-github,rspecator-view[]
'''
== Implementation Specification
(visible only on this page)

=== Message

Use ARG instruction or set & unset environment variable in a single layer.

=== Highlighting

Highlight the `unset` command with the environment variable.

'''
endif::env-github,rspecator-view[]
