This rule raises an issue when error messages are printed to stdout instead of stderr.

== Why is this an issue?

Error messages should be sent to stderr (file descriptor 2) to properly separate error output from normal program output.

When error messages are sent to stdout, they become mixed with the program's regular output. This creates several problems:

* **Pipeline interference**: If the script's output is piped to another command, error messages will be processed as if they were normal data, potentially causing the downstream command to fail or produce incorrect results.

* **Output redirection issues**: When stdout is redirected to a file (e.g., `script.sh > output.txt`), error messages end up in the output file instead of being displayed to the user, making debugging difficult.

* **Automated processing problems**: Scripts used in automated systems or CI/CD pipelines need clear separation between normal output and error messages for proper error handling and logging.

* **User experience**: Users expect error messages to appear on their terminal even when redirecting normal output, which only happens when errors go to stderr.

By convention, stderr is specifically designed for error messages, warnings, and diagnostic information, while stdout is reserved for the program's primary output.

=== What is the potential impact?

Error messages mixed with normal output can cause downstream commands in pipelines to fail, make debugging difficult when output is redirected, and break automated systems that rely on proper separation of normal output and error messages.

== How to fix it

Redirect error messages to stderr using `>&2` with echo commands.

=== Code examples

==== Noncompliant code example

[source,shell,diff-id=1,diff-type=noncompliant]
----
echo "Error: file not found" # Noncompliant
----

==== Compliant solution

[source,shell,diff-id=1,diff-type=compliant]
----
echo "Error: file not found" >&2
----

== Resources

=== Documentation

 * Bash Manual - Redirections - https://www.gnu.org/software/bash/manual/html_node/Redirections.html[Official documentation on bash redirection operators including stderr redirection]
 * Advanced Bash-Scripting Guide - I/O Redirection - https://tldp.org/LDP/abs/html/io-redirection.html[Comprehensive guide to input/output redirection in bash scripts]

=== Standards

 * POSIX.1-2017 - Standard Streams - https://pubs.opengroup.org/onlinepubs/9699919799/basedefs/V1_chap03.html#tag_03_395[POSIX specification defining standard input, output, and error streams]

