== Why is this an issue?

Null statements should not normally be included deliberately, but where they are used, they shall appear on a line by themselves. White-space characters may precede the null statement to preserve indentation. If a comment follows the null statement, then at least one white-space character shall separate the null statement from the comment. The use of a white-space character to separate the null statement from any following comment is required on the grounds that it provides an important visual cue to reviewers. Following this rule enables a static checking tool to warn of null statements appearing on a line with other text, which would normally indicate a programming error.


=== Noncompliant code example

[source,cpp]
----
while ( ( port & 0x80 ) == 0 )
{
  /* wait for pin */ ; // Noncompliant, comment before ;
  ;// wait for pin - Noncompliant, no white-space char after ;
}
----

=== Compliant solution

[source,cpp]
----
while ( ( port & 0x80 ) == 0 )
{
  ; // wait for pin - Compliant
}
----

== Resources

* MISRA C:2004, 14.3
* MISRA {cpp}:2008, 6-2-3

ifdef::env-github,rspecator-view[]
'''
== Comments And Links
(visible only on this page)

=== is related to: S139

=== is related to: S1116

=== on 20 Oct 2014, 14:09:28 Samuel Mercier wrote:
Probably not a rule on its own; we should rather update implementation of RSPEC-139 and RSPEC-1116

=== on 20 Oct 2014, 14:50:16 Ann Campbell wrote:
This rule seems to have little faith in the ability of static checking tools. Perhaps that's why it was dropped in the 2012 version. IMO, we should consider this fully covered by RSPEC-1116.


cc [~freddy.mallet]

=== on 25 Nov 2014, 20:48:29 Freddy Mallet wrote:
Fine for me [~ann.campbell.2] to close this ticket with resolution "Won't Fix"

=== on 2 Dec 2014, 14:13:28 Ann Campbell wrote:
References moved to RSPEC-1116

endif::env-github,rspecator-view[]
