== Why is this an issue?

include::../description.adoc[]

== How to fix it

=== Code examples

The examples below enforce the following regular expression ``++^[a-z][a-zA-Z0-9]*$++``:

==== Noncompliant code example

[source,java,diff-id=1,diff-type=noncompliant]
----
class MyClass {
   private int my_field;
}
----

==== Compliant solution

[source,java,diff-id=1,diff-type=compliant]
----
class MyClass {
   private int myField;
}
----

== Resources

=== Documentation

* Oracle - https://www.oracle.com/java/technologies/javase/codeconventions-namingconventions.html[Java SE Naming Conventions]
* Geeksforgeeks - https://www.geeksforgeeks.org/java-naming-conventions/[Java Naming Conventions]
* Wikipedia - https://en.wikipedia.org/wiki/Naming_convention_(programming)[Naming Convention (programming)]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../parameters.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
