include::../description.adoc[]

== Noncompliant Code Example

----
for (var i:int = 0; i < 10; i++) {
  var temp:MyObj = new MyObject();  // Noncompliant
  //...  
}
----

== Compliant Solution

----
var temp:MyObj = new MyObject();
for (var i:int = 0; i < 10; i++) {
  //...  
}
----

ifdef::env-github,rspecator-view[]
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
