== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,flex]
----
for (var i:int = 0; i < 10; i++) {
  var temp:MyObj = new MyObject();  // Noncompliant
  //...  
}
----

=== Compliant solution

[source,flex]
----
var temp:MyObj = new MyObject();
for (var i:int = 0; i < 10; i++) {
  //...  
}
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
