<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Dependency injection frameworks such as Spring, Quarkus, and others support dependency injection by using annotations such as <code>@Inject</code> and <code>@Autowired</code>.
These annotations can be used to inject beans via constructor, setter, and field injection.</p>
</div>
<div class="paragraph">
<p>Generally speaking, field injection is discouraged.
It allows the creation of objects in an invalid state and makes testing more difficult.
The dependencies are not explicit when instantiating a class that uses field injection.</p>
</div>
<div class="paragraph">
<p>In addition, field injection is not compatible with final fields.
Keeping dependencies immutable where possible makes the code easier to understand, easing development and maintenance.</p>
</div>
<div class="paragraph">
<p>Finally, because values are injected into fields after the object has been constructed, they cannot be used to initialize other non-injected fields inline.</p>
</div>
<div class="paragraph">
<p>This rule raises an issue when the <code>@Autowired</code> or <code>@Inject</code> annotations are used on a field.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Use constructor injection instead.</p>
</div>
<div class="paragraph">
<p>By using constructor injection, the dependencies are explicit and must be passed during an object&#8217;s construction.
This avoids the possibility of instantiating an object in an invalid state and makes types more testable.
Fields can be declared final, which makes the code easier to understand, as dependencies don&#8217;t change after instantiation.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-text" data-lang="text">public class SomeService {
    @Autowired
    private SomeDependency someDependency; // Noncompliant

    private String name = someDependency.getName(); // Will throw a NullPointerException
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-text" data-lang="text">public class SomeService {
    private final SomeDependency someDependency;
    private final String name;

    @Autowired
    public SomeService(SomeDependency someDependency) {
        this.someDependency = someDependency;
        name = someDependency.getName();
    }
}</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_articles_blog_posts">Articles &amp; blog posts</h3>
<div class="ulist">
<ul>
<li>
<p>Baeldung - <a href="https://www.baeldung.com/java-spring-field-injection-cons">Why Is Field Injection Not Recommended?</a></p>
</li>
<li>
<p>Baeldung - <a href="https://www.baeldung.com/constructor-injection-in-spring">Constructor Dependency Injection in Spring</a></p>
</li>
<li>
<p>Oliver Drotbohm - <a href="https://odrotbohm.de/2013/11/why-field-injection-is-evil/">Why field injection is evil</a></p>
</li>
<li>
<p>GitHub Discussions - <a href="https://github.com/quarkusio/quarkus/discussions/38474">Field injection in quarkus</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>