The basic numeric types ``++char++``, ``++int++``, ``++short++``, ``++long++``, ``++float++``, ``++double++``, and ``++long double++`` should not be used. Instead, specific-length typedefs should be. This rule helps to clarify the size of the storage, but does not guarantee portability because of the asymmetric behavior of integral promotion.


Note that it is still important to understand the integer size of the implementation, and developers should be aware of the actual implementation of the typedefs under these definitions.


== Noncompliant Code Example

----
int function(unsigned short a) // Noncompliant
{
  // ...
}
----


== Compliant Solution

----
#include <stdint.h>
int32_t function(uint16_t a) // Compliant
{
  // ...
}
----


== See

* MISRA C:2004, 6.3 - typedefs that indicate size and signedness should be used in place of the basic types
* MISRA {cpp}:2008, 3-9-2 - typedefs that indicate size and signedness should be used in place of the basic numerical types

=== See Also

* MISRA {cpp} 2008 Section 6.5.0 on integral promotion


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
