== Why is this an issue?

The basic numeric types ``++char++``, ``++int++``, ``++short++``, ``++long++``, ``++float++``, ``++double++``, and ``++long double++`` should not be used. Instead, specific-length typedefs should be. This rule helps to clarify the size of the storage, but does not guarantee portability because of the asymmetric behavior of integral promotion.


Note that it is still important to understand the integer size of the implementation, and developers should be aware of the actual implementation of the typedefs under these definitions.


=== Noncompliant code example

[source,cpp]
----
int function(unsigned short a) // Noncompliant
{
  // ...
}
----


=== Compliant solution

[source,cpp]
----
#include <stdint.h>
int32_t function(uint16_t a) // Compliant
{
  // ...
}
----

== Resources

* MISRA C:2004, 6.3 - typedefs that indicate size and signedness should be used in place of the basic types
* MISRA {cpp}:2008, 3-9-2 - typedefs that indicate size and signedness should be used in place of the basic numerical types

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Replace this use of 'xxx' with a typedef.


'''
== Comments And Links
(visible only on this page)

=== on 17 Oct 2014, 12:46:04 Ann Campbell wrote:
FYI [~samuel.mercier] I've changed this to Understandability since following the rule "helps to clarify"...

endif::env-github,rspecator-view[]
