=== Code examples

==== Noncompliant code example

[source,vbnet,diff-id=1,diff-type=noncompliant]
----
Structure Point
    Public Property X As Integer
    Public Property Y As Integer
End Structure
----

==== Compliant solution

[source,vbnet,diff-id=1,diff-type=compliant]
----
Structure Point
    Public Property X As Integer
    Public Property Y As Integer

    Public Overloads Function Equals(other As Point) As Boolean
        Return X = other.X AndAlso Y = other.Y
    End Function

    Public Overrides Function Equals(obj As Object) As Boolean
        If TypeOf obj Is Point Then
            Return Equals(DirectCast(obj, Point))
        End If

        Return False
    End Function

    Public Overrides Function GetHashCode() As Integer
        Return (X.GetHashCode() * 397) Xor Y.GetHashCode()
    End Function
End Structure
----
