== Why is this an issue?

If an exception is thrown when constructing the exception object, or when evaluating the assignment expression that initializes the exception object, it is that exception that propagates in preference to the one that was about to be thrown. This may be inconsistent with developer expectations.


=== Noncompliant code example

[source,cpp]
----
// construction of E2 causes an exception to be thrown
class E2
{
  public:
  E2 ( )
  {
    throw 10;
  }
};

try
{
  if ( ... )
  {
    throw E2 ( ); // Non-compliant – int exception thrown when constructing the E2 object
  }
}
----


=== Compliant solution

[source,cpp]
----
class E
{
  public:
  E ( ) { } // Assume constructor cannot cause an exception
};

try
{
  if ( ... )
  {
    throw E ( ); // Compliant – no exception thrown when constructing the object
  }
}
----


== Resources

* MISRA {cpp}:2008, 15-1-1


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

An exception could be thrown during construction of "xxx".


'''
== Comments And Links
(visible only on this page)

=== on 14 Oct 2014, 17:56:36 Ann Campbell wrote:
\[~samuel.mercier] please correct the following

* Add a See section to the description listing the appropriate MISRA number
* Fill in Applicability.
* Make sure the appropriate MISRA C and MISRA {cpp} fields on the references tab are filled in

endif::env-github,rspecator-view[]
