One can use logical AND (`&&`) operator to conditionally render in React (aka short-circuit evaluation). `myCondition && <MyElement />` will return `<MyElement />` if `myCondition` is `true` and `false` otherwise. As React renders boolean values to nothing, when condition is `false`, nothing will be rendered as intended. But if condition has a non-boolean value which evaluates to `false` (e.g. `0`), that value will leak.
Note that along with short-circuit evaluation, in order to achieve conditional rendering it's possible to use ternary operator (`myCondition ? <MyElement /> : null`), which is less error-prone in this case as both return values are explicit.

== Noncompliant Code Example

[source,javascript]
----
function Profile(props) {
  return <div>
    <h1>{ props.username }</h1>
    { props.orders && <Orders /> }
  </div>
}
----

== Compliant Solution

[source,javascript]
----
function Profile(props) {
  return <div>
    <h1>{ props.username }</h1>
    { props.orders > 0 && <Orders /> }
  </div>
}
----
or
----
function Profile(props) {
  return <div>
    <h1>{ props.username }</h1>
    { props.orders ? <Orders /> : null }
  </div>
}
----
== See

* https://reactjs.org/docs/conditional-rendering.html[Conditional Rendering] - conditional rendering in React