<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Even if it is legal, mixing case and non-case labels in the body of a switch statement is very confusing and can even be the result of a typing error.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="paragraph">
<p>Case 1, the code is syntactically correct but the behavior is not the expected one</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-javascript" data-lang="javascript">switch (day) {
  case MONDAY:
  case TUESDAY:
  WEDNESDAY:   // instead of "case WEDNESDAY"
    doSomething();
    break;
  ...
}</code></pre>
</div>
</div>
<div class="paragraph">
<p>Case 2, the code is correct and behaves as expected but is hardly readable</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-javascript" data-lang="javascript">switch (day) {
  case MONDAY:
    break;
  case TUESDAY:
    foo:for(i = 0 ; i &lt; X ; i++) {
         /* ... */
        break foo;  // this break statement doesn't relate to the nesting case TUESDAY
         /* ... */
    }
    break;
    /* ... */
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="paragraph">
<p>Case 1</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-javascript" data-lang="javascript">switch (day) {
  case MONDAY:
  case TUESDAY:
  case WEDNESDAY:
    doSomething();
    break;
  ...
}</code></pre>
</div>
</div>
<div class="paragraph">
<p>Case 2</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-javascript" data-lang="javascript">switch (day) {
  case MONDAY:
    break;
  case TUESDAY:
    compute(args); // put the content of the labelled "for" statement in a dedicated method
    break;

    /* ... */
}</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove this misleading "xxxx" label.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_is_duplicated_by_s915">is duplicated by: <a data-rspec-id="S915" class="rspec-auto-link">S915</a></h3>

</div>
<div class="sect2">
<h3 id="_relates_to_s1119">relates to: <a data-rspec-id="S1119" class="rspec-auto-link">S1119</a></h3>

</div>
<div class="sect2">
<h3 id="_relates_to_s1439">relates to: <a data-rspec-id="S1439" class="rspec-auto-link">S1439</a></h3>

</div>
<div class="sect2">
<h3 id="_on_20_aug_2013_195456_freddy_mallet_wrote">on 20 Aug 2013, 19:54:56 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>Is implemented by http://jira.codehaus.org/browse/SONARJAVA-309</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_7_jan_2014_110344_freddy_mallet_wrote">on 7 Jan 2014, 11:03:44 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>Is implemented by http://jira.codehaus.org/browse/SONARPLUGINS-3366 for JavaScript</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_29_sep_2014_181245_evgeny_mandrikov_wrote">on 29 Sep 2014, 18:12:45 Evgeny Mandrikov wrote:</h3>
<div class="paragraph">
<p>In C Family of languages "break" cannot be used to jump to label, so created dedicated sub-task containing only first example.</p>
</div>
</div>
</div>
</div>