Shared coding conventions allow teams to collaborate effectively. This rule raises an issue when the use of spaces within a template does not conform to the configured style.


== Noncompliant Code Example

With the configured default forbidding spaces:

----
`Hello, ${world }!`; // Noncompliant
----


== Compliant Solution

----
`Hello, ${world}!`;
----


