== Why is this an issue?

Shared coding conventions allow teams to collaborate effectively. This rule raises an issue when the use of spaces within a template does not conform to the configured style.


=== Noncompliant code example

With the configured default forbidding spaces:

[source,javascript]
----
`Hello, ${world }!`; // Noncompliant
----


=== Compliant solution

[source,javascript]
----
`Hello, ${world}!`;
----


