include::../common/why.adoc[]

include::../common/how.adoc[]

=== Code examples

==== Noncompliant code example

[source,html,diff-id=1,diff-type=noncompliant]
----
<div onClick="showMessage('Hello')" /> <!-- Noncompliant -->
----

==== Compliant solution

[source,html,diff-id=1,diff-type=compliant]
----
<div onClick="showMessage('Hello')" role="button" />
----

include::../common/resources.adoc[]
