The use of wrapper objects for primitive types is gratuitous, confusing and dangerous. If you use a wrapper object constructor for type conversion, just remove the ``++new++`` keyword, and you'll get a primitive value automatically. If you use a wrapper object as a way to add properties to a primitive, you should re-think the design. Such uses are considered bad practice, and should be refactored.


== Noncompliant Code Example

[source,javascript]
----
let x = new Number("0");
if (x) {
  alert('hi');  // Shows 'hi'.
}
----


== Compliant Solution

[source,javascript]
----
let x = Number("0");
if (x) {
  alert('hi');
}
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
