<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Using <code>repr(isize/usize)</code> for C-like enumerations with values that don&#8217;t fit into an <code>i32</code> can lead to truncated variant values on 32-bit architectures, potentially causing unexpected behavior.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-rust" data-lang="rust">#[repr(usize)]
enum NonPortable {
    X = 0x1_0000_0000, // Noncompliant
    Y = 0,
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-rust" data-lang="rust">#[repr(u64)]
enum Portable {
    X = 0x1_0000_0000, // Compliant
    Y = 0,
}</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Clippy Lints - <a href="https://rust-lang.github.io/rust-clippy/master/index.html#enum_clike_unportable_variant" class="bare">https://rust-lang.github.io/rust-clippy/master/index.html#enum_clike_unportable_variant</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>