== Why is this an issue?

There are two ways to write lambdas that contain single statement, but one is definitely more compact and readable than the other.


*Note* that this rule is automatically disabled when the project's ``++sonar.java.source++`` is lower than ``++8++``.


=== Noncompliant code example

[source,java]
----
x -> {System.out.println(x+1);}
(a, b) -> { return a+b; }
----


=== Compliant solution

[source,java]
----
x -> System.out.println(x+1)
(a, b) -> a+b    //For return statement, the return keyword should also be dropped
----



ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
