== Why is this an issue?

There are two ways to write lambdas that contain single statement, but one is definitely more compact and readable than the other.


*Note* that this rule is automatically disabled when the project's ``++sonar.java.source++`` is lower than ``++8++``.


=== Noncompliant code example

[source,java]
----
x -> {System.out.println(x+1);}
(a, b) -> { return a+b; }
----


=== Compliant solution

[source,java]
----
x -> System.out.println(x+1)
(a, b) -> a+b    //For return statement, the return keyword should also be dropped
----



ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Remove the useless curly braces around this statement[ and then remove useless return keyword].  [(sonar.java.source not set. Assuming 8 or greater.)]


'''
== Comments And Links
(visible only on this page)

=== on 21 Feb 2014, 16:03:58 Freddy Mallet wrote:
Is implemented by \http://jira.codehaus.org/browse/SONARJAVA-453

endif::env-github,rspecator-view[]
