<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Class members are initialized in the order in which they are declared in the class,
not the order in which they appear in the class initializer list.
To avoid errors caused by order-dependent initialization,
the order of members in the initialization list should match the order in which members are declared in a class.</p>
</div>
<div class="paragraph">
<p>The initialization order, as described <a href="https://en.cppreference.com/w/cpp/language/constructor#Initialization_order">here</a>, is:</p>
</div>
<div class="olist arabic">
<ol class="arabic">
<li>
<p>If the constructor is for the most-derived class, virtual bases are initialized in the order in which they appear in depth-first left-to-right traversal of the base class declarations (left-to-right refers to the appearance in base-specifier lists)</p>
</li>
<li>
<p>Then, direct bases are initialized in left-to-right order as they appear in this class&#8217;s base-specifier list</p>
</li>
<li>
<p>Then, non-static data members are initialized in order of declaration in the class definition.</p>
</li>
</ol>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">#include &lt;iostream&gt;

struct A {
  A(int num) {
    std::cout &lt;&lt; "A(num = " &lt;&lt; num &lt;&lt; ")" &lt;&lt; std::endl;
  }
};

struct B {
  int b;
};

class C : public A, B {
public:
  int x;
  int y;

  C(int i) : B{i}, A{b}, y(i), x(y + 1) { }  // Noncompliant
};

int main() {
  C c(1); // Undefined behavior, might print "A(num = 0)"
  std::cout &lt;&lt; c.x &lt;&lt; " " &lt;&lt; c.y &lt;&lt; std::endl;  // might print "1 1"
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">#include &lt;iostream&gt;

struct A {
  A(int num) {
    std::cout &lt;&lt; "A(num = " &lt;&lt; num &lt;&lt; ")" &lt;&lt; std::endl;
  }
};

struct B {
  int b;
};

class C : public A, B {
public:
  int x;
  int y;

  C(int i) : A{i}, B{i}, x(i + 1), y(i) { }
};

int main() {
  C c(1); // prints "A(num = 1)"
  std::cout &lt;&lt; c.x &lt;&lt; " " &lt;&lt; c.y &lt;&lt; std::endl;  // prints "2 1"
}</code></pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p><a href="https://wiki.sei.cmu.edu/confluence/x/dXw-BQ">CERT, OOP53-CPP.</a> - Write constructor member initializers in the canonical order</p>
</li>
<li>
<p>C++ Core Guidelines - <a href="https://github.com/isocpp/CppCoreGuidelines/blob/e49158a/CppCoreGuidelines.md#c47-define-and-initialize-member-variables-in-the-order-of-member-declaration">C.47: Define and initialize member variables in the order of member declaration</a></p>
</li>
</ul>
</div>
<hr>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Reorder these initializers to match their declaration order.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_is_duplicated_by_s1266">is duplicated by: <a data-rspec-id="S1266" class="rspec-auto-link">S1266</a></h3>

</div>
</div>
</div>