== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,flex]
----
for (var i = 0; i < 10; i++) {
  ...
  i = i - 1; // Noncompliant 
  ...
} 

for (var i = 0; i < getMaximumNumber(); i++) {...} 
----

=== Compliant solution

[source,flex]
----
int stopCondition = getMaximumNumber();
for (var i = 0; i < stopCondition; i++) {...} 
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../parameters.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
