include::../description.adoc[]

== Noncompliant Code Example

----
for (var i = 0; i < 10; i++) {
  ...
  i = i - 1; // Noncompliant 
  ...
} 

for (var i = 0; i < getMaximumNumber(); i++) {...} 
----

== Compliant Solution

----
int stopCondition = getMaximumNumber();
for (var i = 0; i < stopCondition; i++) {...} 
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../parameters.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
