
== Why is this an issue?
Having a manual `PartialEq` implementation for types with a derived `Hash` can lead to inconsistencies. The contract `k1 == k2 â‡’ hash(k1) == hash(k2)` must always hold. Inconsistencies can cause undefined behaviors, especially when these types are used in collections such as `HashMap`.


=== Code examples

==== Noncompliant code example
[source,rust,diff-id=1,diff-type=noncompliant]
----
#[derive(Hash)]
struct Foo;

impl PartialEq for Foo {
    fn eq(&self, other: &Self) -> bool {
        // Some custom equality logic
        true // Noncompliant
    }
}
----

==== Compliant solution

[source,rust,diff-id=1,diff-type=compliant]
----
#[derive(Hash, PartialEq)]
struct Foo;
----

== Resources
=== Documentation

* Clippy Lints - https://rust-lang.github.io/rust-clippy/master/index.html#derived_hash_with_manual_eq
