<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Once you&#8217;ve made a selection, you typically want to know whether it actually found anything. Since selectors <em>always</em> return an object (the set of selected DOM elements), the best way to see whether your selection found anything is to test the returned object&#8217;s <code>.length</code> property.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-javascript" data-lang="javascript">if ( $( "div.foo" ) ) {  // Noncompliant
  // this code always runs, even when the selector didn't match any elements
  // ...
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-javascript" data-lang="javascript">// Testing whether a selection contains elements.
if ( $( "div.foo" ).length &gt; 0) {
  // this code only runs if elements were found
  //  ...
}</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Use the "length" property to see whether this selection contains elements.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_26_mar_2015_171146_linda_martin_wrote">on 26 Mar 2015, 17:11:46 Linda Martin wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] Assigned for review and completion. The description is a copy/paste of https://learn.jquery.com/using-jquery-core/selecting-elements/, I was afraid that if I started to rephrase it would not make any sens anymore.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_27_mar_2015_152413_ann_campbell_wrote">on 27 Mar 2015, 15:24:13 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>double-check me, please [~linda.martin]</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_19_may_2015_155400_linda_martin_wrote">on 19 May 2015, 15:54:00 Linda Martin wrote:</h3>
<div class="paragraph">
<p>Reviewed.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_24_apr_2017_102445_elena_vilchik_wrote">on 24 Apr 2017, 10:24:45 Elena Vilchik wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] why minor?</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_1_nov_2019_172953_elena_vilchik_wrote">on 1 Nov 2019, 17:29:53 Elena Vilchik wrote:</h3>
<div class="paragraph">
<p>See https://github.com/SonarSource/SonarJS/issues/1698</p>
</div>
</div>
</div>
</div>