== Why is this an issue?

include::description.adoc[]

=== Noncompliant code example

[source,text]
----
if (condition)  // Noncompliant
doTheThing();

doTheOtherThing();
somethingElseEntirely();

foo();
----


=== Compliant solution

[source,text]
----
if (condition)
  doTheThing();

doTheOtherThing();
somethingElseEntirely();

foo();
----

