== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,swift]
----
if x == 0 {
  doSomething()
} else if x == 1 {
  doSomethingElse()
} 
----

=== Compliant solution

[source,swift]
----
if x == 0 {
  doSomething()
} else if x == 1 {
  doSomethingElse()
} else {
  NSException(name:"IllegalStateException", reason:"Unreachable else clause is reached", userInfo:nil).raise()
}
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
