include::../description.adoc[]

== Noncompliant Code Example

----
if x == 0 {
  doSomething()
} else if x == 1 {
  doSomethingElse()
} 
----

== Compliant Solution

----
if x == 0 {
  doSomething()
} else if x == 1 {
  doSomethingElse()
} else {
  NSException(name:"IllegalStateException", reason:"Unreachable else clause is reached", userInfo:nil).raise()
}
----

include::../see.adoc[]

ifdef::env-github,rspecator-view[]
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
