<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when Rails associations use deprecated hash-style options like <code>:order</code> or <code>:uniq</code> instead of modern lambda scope blocks.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Rails 4 deprecated hash-style options like <code>:order</code> and <code>:uniq</code> in association declarations in favor of lambda scope blocks. These deprecated options generate deprecation warnings and may be removed in future Rails versions.</p>
</div>
<div class="paragraph">
<p>The old hash syntax has several limitations:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>Limited flexibility for complex queries</p>
</li>
<li>
<p>Inconsistent with Rails' scoping conventions</p>
</li>
<li>
<p>Cannot chain multiple scope methods easily</p>
</li>
<li>
<p>Generates deprecation warnings in Rails 4+</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>Lambda scope blocks provide a more powerful and consistent approach:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>Allow chaining multiple scope methods like <code>where</code>, <code>order</code>, and <code>uniq</code></p>
</li>
<li>
<p>Follow Rails' modern scoping conventions</p>
</li>
<li>
<p>Provide better readability and maintainability</p>
</li>
<li>
<p>Enable more complex query logic when needed</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>Using deprecated options makes your codebase less maintainable and may cause issues when upgrading Rails versions.</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="paragraph">
<p>Using deprecated association options can lead to:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>Deprecation warnings cluttering your application logs</p>
</li>
<li>
<p>Potential breaking changes when upgrading Rails versions</p>
</li>
<li>
<p>Reduced code maintainability and consistency</p>
</li>
<li>
<p>Limited flexibility for extending association logic</p>
</li>
</ul>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it_in_rails">How to fix it in Rails</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Replace the deprecated <code>:order</code> option with a lambda scope block using the <code>order</code> method. Place the lambda block before other association options.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-ruby" data-lang="ruby">class RelatedList &lt; ActiveRecord::Base
  has_many :contents, :order =&gt; :position # Noncompliant
  has_many :tasks, order: 'priority desc', dependent: :destroy # Noncompliant
end</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-ruby" data-lang="ruby">class RelatedList &lt; ActiveRecord::Base
  has_many :contents, -&gt; { order(:position) }
  has_many :tasks, -&gt; { order(priority: :desc) }, dependent: :destroy
end</code></pre>
</div>
</div>
<div class="paragraph">
<p>Replace the deprecated <code>:uniq</code> option with a lambda scope block using the <code>uniq</code> method. You can chain multiple scope methods together.</p>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_code_examples_2">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example_2">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-ruby" data-lang="ruby">class Goal &lt; ActiveRecord::Base
  has_many :donors, through: :donations, uniq: true # Noncompliant
  has_many :contributors, through: :donations, uniq: true, order: "name" # Noncompliant
end</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution_2">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-ruby" data-lang="ruby">class Goal &lt; ActiveRecord::Base
  has_many :donors, -&gt; { uniq }, through: :donations
  has_many :contributors, -&gt; { order(:name).uniq }, through: :donations
end</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Rails Association Basics - <a href="https://guides.rubyonrails.org/association_basics.html">Official Rails guide covering association declarations and scoping</a></p>
</li>
<li>
<p>Active Record Query Interface - <a href="https://guides.rubyonrails.org/active_record_querying.html">Rails guide explaining query methods like order and uniq</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>