== Why is this an issue?

``++Optional++`` value can hold either a value or not. The value held in the ``++Optional++`` can be accessed using the ``++get()++`` method, but it will throw a 

``++NoSuchElementException++`` if there is no value present. To avoid the exception, calling the ``++isPresent()++`` or ``++! isEmpty()++`` method should always be done before any call to ``++get()++``.


Alternatively, note that other methods such as ``++orElse(...)++``, ``++orElseGet(...)++`` or ``++orElseThrow(...)++`` can be used to specify what to do with an empty ``++Optional++``.

== How to fix it

=== Code examples

==== Noncompliant code example

[source,java,diff-id=1,diff-type=noncompliant]
----
public static Optional<String> getOptionalValue(boolean condition) {
  if (condition) {
    return Optional.of("Value text");
  }
  return Optional.empty();
}

public void example(boolean condition) {
  Optional<String> optional = getOptionalValue(condition);
  String value = optional.get(); // Noncompliant; getOptionalValue may return an empty optional
}
----

==== Compliant solution

[source,java,diff-id=1,diff-type=compliant]
----
public static Optional<String> getOptionalValue(boolean condition) {
  if (condition) {
    return Optional.of("Value text");
  }
  return Optional.empty();
}

public void example(boolean condition) {
  getOptionalValue(condition).ifPresent(value -> {
    // Do something with the value
  });
}
----

or

[source,java]
----
Optional<String> value = this.getOptionalValue();

// ...

if (value.isPresent()) {
  String stringValue = value.get();
}
----

or

[source,java]
----
Optional<String> value = this.getOptionalValue();

// ...

String stringValue = value.orElse("default");
----

[source,java]
----
Optional<String> optional = methodThatReturnsOptional();
if (optional.isEmpty()) {
  throw new NotFoundException();
}
String value = optional.get();
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

call "xxx.isPresent()" before accessing the value.


'''

endif::env-github,rspecator-view[]


