== Why is this an issue?

Using the same value on both sides of certain operators is a code defect. In the case of logical operators, it is either a copy/paste error and, therefore, a bug, or it is simply duplicated code and should be simplified. For bitwise operators and most binary mathematical operators, having the same value on both sides of an operator yields predictable results and should be simplified as well to avoid further code defects.

This rule raises for the following operators.

* https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/operators/equality-operators[Equality operators] (`==` and `!=`)
* https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/operators/comparison-operators[Comparison operators] (`< =`, `<`, `>`, `>=`)
* The following https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/operators/boolean-logical-operators[Logical Operators]: 

** Logical OR (`|` )
** Conditional logical OR (`||`)
** Logical AND (`&`)
** Conditional logical AND (`&&`)
** Logical exclusive OR (`^`)

* The following https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/operators/arithmetic-operators[arithmetic operators]: 
** Subtraction (`-`)
** Division (`\`)
** Remainder operator (`%`)
** Subtraction assignment operator (`-=`)
** Divide assignment operator (`\=`)

=== Exceptions

This rule ignores the following operators:

* Multiplication (*)
* Addition (+)
* Assignment (=)
* https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/operators/bitwise-and-shift-operators#left-shift-operator-[Left-shift (<<)]
* https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/operators/bitwise-and-shift-operators#right-shift-operator-[Right-shift (>>)]

=== Code examples

==== Noncompliant code example

[source,csharp]
----
if ( a == a ) // always true
{ 
  doZ();
}
if ( a != a ) // always false
{
  doY();
}
if ( a == b && a == b ) // if the first one is true, the second one is too
{ 
  doX();
}
if ( a == b || a == b ) // if the first one is true, the second one is too
{ 
  doW();
}

int j = 5 / 5; // always 1
int k = 5 - 5; // always 0

c.Equals(c);    // always true
Object.Equals(c, c); // always true
----

== Resources

=== Documentation
* https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/operators/arithmetic-operators[Arithmetic Operators]
* https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/operators/boolean-logical-operators[Boolean logical operators]
* https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/operators/bitwise-and-shift-operators[Bitwise and shift operators]
* https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/operators/equality-operators[Equality operators - test if two objects are equal or not]
* https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/operators/comparison-operators[Comparison operators]
* https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/operators/assignment-operator[Assignment operators]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

* Correct one of the identical expressions on both sides of operator '{0}'.
* Change one instance of '{0}' to a different value; comparing '{0}' to itself always returns true.


include::../highlighting.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
