== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,csharp]
----
if ( a == a ) // always true
{ 
  doZ();
}
if ( a != a ) // always false
{
  doY();
}
if ( a == b && a == b ) // if the first one is true, the second one is too
{ 
  doX();
}
if ( a == b || a == b ) // if the first one is true, the second one is too
{ 
  doW();
}

int j = 5 / 5; //always 1
int k = 5 - 5; // always 0

c.Equals(c);    //always true
Object.Equals(c, c); //always true
----

=== Exceptions

This rule ignores `*`, `+`, `=`, `<<`,  and `>>` operators.

== Resources

=== Documentation
* https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/operators/bitwise-and-shift-operators[Bitwise and shift operators]
* https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/operators/arithmetic-operators[https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/operators/arithmetic-operators]
* https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/ref[`ref` keyword]
* https://learn.microsoft.com/en-us/dotnet/api/system.string.intern[String.Intern(String) Method]
* https://learn.microsoft.com/en-us/dotnet/csharp/programming-guide/concepts/linq/[LINQ],
* https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/builtin-types/built-in-types[build-in types]
* https://learn.microsoft.com/en-us/archive/msdn-magazine/2017/march/net-framework-immutable-collections[Immutable collections]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

* Correct one of the identical expressions on both sides of operator '{0}'.
* Change one instance of '{0}' to a different value; comparing '{0}' to itself always returns true.


include::../highlighting.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
