This rule raises an issue when semicolons are used to terminate statements in Go code where they are not required by the language specification.

== Why is this an issue?

Go has automatic semicolon insertion rules that make explicit semicolons unnecessary in most cases. The Go language specification defines that semicolons are automatically inserted after a line's final token if that token is an identifier, literal, or one of the keywords `break`, `continue`, `fallthrough`, or `return`, or one of the operators `++`, `--`, `)`, `]`, or `}`.

Using unnecessary semicolons goes against Go's design philosophy of clean, readable code. It can make code look cluttered and may confuse developers who are learning Go's syntax rules. The automatic semicolon insertion mechanism is a deliberate language feature that allows Go code to be more concise and readable.

Additionally, unnecessary semicolons can indicate that a developer is applying conventions from other programming languages (like C, Java, or JavaScript) rather than following Go's idiomatic style. This can lead to inconsistent code style within a Go project.

=== What is the potential impact?

While unnecessary semicolons do not cause runtime errors or compilation failures, they impact code quality and maintainability:

* **Readability**: Code becomes more cluttered and less idiomatic
* **Consistency**: Mixed usage of semicolons creates inconsistent code style
* **Learning curve**: New Go developers may be confused about when semicolons are actually required
* **Code reviews**: Unnecessary semicolons can distract from more important issues during code reviews

== How to fix it

Remove unnecessary semicolons from the end of statements. Go's automatic semicolon insertion will handle statement termination.

=== Code examples

==== Noncompliant code example

[source,go,diff-id=1,diff-type=noncompliant]
----
package main

import "fmt";

func main() {
    x := 42;
    fmt.Println(x);
    return;
}
----

==== Compliant solution

[source,go,diff-id=1,diff-type=compliant]
----
package main

import "fmt"

func main() {
    x := 42
    fmt.Println(x)
    return
}
----

== Resources

=== Documentation

 * Go Language Specification - Semicolons - https://go.dev/ref/spec#Semicolons[Official Go specification explaining automatic semicolon insertion rules]

 * Effective Go - Semicolons - https://go.dev/doc/effective_go#semicolons[Best practices for semicolon usage in Go code]
