Deprecated instructions should be replaced by other suggested instructions.

== Why is this an issue?

The languages and technologies evolve and deprecates some features.
At some day they may be removed.
It is important to use up-yo-date constructs to avoid issues by migration to newer version of technology.

== How to fix it

=== Code examples

==== Noncompliant code example

[source,docker,diff-id=1,diff-type=noncompliant]
----
MAINTAINER bob
----

==== Compliant solution

[source,docker,diff-id=1,diff-type=compliant]
----
LABEL org.opencontainers.image.authors="bob"
----

=== How does this work?

The `LABEL` instruction is a much more flexible than `MAINTAINER` and should be used instead.

== Resources
=== Documentation

* https://docs.docker.com/engine/deprecated/#maintainer-in-dockerfile[Deprecated Engine Features]
* https://docs.docker.com/engine/reference/builder/#maintainer-deprecated[MAINTAINER (deprecated)ðŸ”— - Dockerfile reference]

ifdef::env-github,rspecator-view[]
'''
== Implementation Specification
(visible only on this page)

=== Message

Replace deprecated instruction with a new equivalent.

=== Highlighting

Highlight the entire deprecated instruction.

'''
endif::env-github,rspecator-view[]

