<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p><em>Forwarding references</em> are a special kind of references that both ignore and preserve the <em>value category</em> of a function argument, making it possible to forward it by using <code>std::forward</code> or <code>std::forward_like</code>.</p>
</div>
<div class="paragraph">
<p>Any code using such a reference for any purpose other than forwarding is actually ignoring the rvalue-ness and const-ness of the associated parameter.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">class Registry {
  std::vector&lt;std::string&gt; names;

 public:
  template &lt;typename StringLike&gt;
  void addName(StringLike&amp;&amp; arg) {
    names.emplace_back(arg); // Not forwarded
  }
};

void example() {
  Registry r;

  std::string name = "example";
  r.addName(std::move(name));
  std::cout &lt;&lt; "name:" &lt;&lt; name &lt;&lt; std::endl;
  // output is "name:example"
}</code></pre>
</div>
</div>
<div class="paragraph">
<p>In this example, the intent is to move the content of <code>name</code> into the vector, but instead a copy is made.</p>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">class Registry {
  std::vector&lt;std::string&gt; names;

 public:
  template &lt;typename StringLike&gt;
  void addName(StringLike&amp;&amp; arg) {
    names.emplace_back(std::forward&lt;StringLike&gt;(arg));
  }
};

void example() {
  Registry r;

  std::string name = "example";
  r.addName(std::move(name));
  std::cout &lt;&lt; "name:" &lt;&lt; name &lt;&lt; std::endl;
  // output can be anything: name has been moved-from
}</code></pre>
</div>
</div>
<div class="paragraph">
<p><code>std::forward_like</code> is available since C&#43;&#43;23 and is useful when you want to forward a subobject of a forwarding reference.</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">class Registry {
  // ...
  template &lt;typename PairOfStrings&gt;
  void addNames(PairOfStrings&amp;&amp; arg) {
    addName(std::forward_like&lt;PairOfStrings&gt;(arg.second)); // We don't care about arg.first
  }
};</code></pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>C&#43;&#43; reference - <a href="https://en.cppreference.com/w/cpp/utility/forward"><code>std::forward</code></a></p>
</li>
<li>
<p>C&#43;&#43; reference - <a href="https://en.cppreference.com/w/cpp/utility/forward_like"><code>std::forward_like</code></a></p>
</li>
</ul>
</div>
</div>
<div class="sect2">
<h3 id="_external_coding_guidelines">External coding guidelines</h3>
<div class="ulist">
<ul>
<li>
<p>C&#43;&#43; Core Guidelines - <a href="https://github.com/isocpp/CppCoreGuidelines/blob/e49158a/CppCoreGuidelines.md#f19-for-forward-parameters-pass-by-tp-and-only-stdforward-the-parameter">F.19: For "forward" parameters, pass by <code>TP&amp;&amp;</code> and only <code>std::forward</code> the parameter</a></p>
</li>
</ul>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Use only "std::forward" on forwarding references.</p>
</div>
</div>
</div>
</div>