== Why is this an issue?

`Double.longBitsToDouble` expects a 64-bit `long` argument to interpret the bits as a double value correctly.
This allows the method to convert the bit pattern into its corresponding floating-point representation accurately.

If you were to pass a smaller data type, such as an `int`, the method would receive an insufficient number of bits. As a result, the cast
to long would not be correct, and the conversion into a double would not produce the expected result.

The `Double.longBitsToDouble` would produce unexpected results when the arguments have a different behavior when being cast to long.
Unpredictable cast to long is likely to happen when a binary representation of a number is being used, because the "sign bit",
i.e. the most significant bit, is set; while for Java singed data types, i.e. int, short, etc..., the most significant bit is reserved for the sign.

=== Noncompliant code example

[source,java,diff-id=1,diff-type=noncompliant]
----
int i = 0x80003800;
Double.longBitsToDouble(i);   // Noncompliant - NaN
----

=== Compliant solution
`i` represents 64-bit floating point pattern (IEEE-754 Double):
[source,java,diff-id=1,diff-type=compliant]
----
long i = 0x80003800L;
Double.longBitsToDouble(i);   // Compliant - 1.0610049784E-314
----

`i` represents 32-bit floating point pattern (IEEE-754 Single):
[source,java,diff-id=1,diff-type=compliant]
----
int i = 0x80003800;
Float.intBitsToFloat(i);      // Compliant, -2.0089E-41
----

== Resources

=== Documentation
* https://docs.oracle.com/en/java/javase/20/docs/api/java.base/java/lang/Double.html#doubleToLongBits(double)[Oracle Java SE - Double.doubleToLongBits]

=== Articles & blog posts
* https://en.wikipedia.org/wiki/Double-precision_floating-point_format[Wikipedia - Double Precision floating point format]
* https://en.wikipedia.org/wiki/Single-precision_floating-point_format[Wikipedia - Single Precision floating point format]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Remove this "Double.longBitsToDouble" call.


'''
== Comments And Links
(visible only on this page)

=== on 10 Oct 2014, 14:07:37 Freddy Mallet wrote:
Perfect !

endif::env-github,rspecator-view[]
