== Why is this an issue?

`Double.longBitsToDouble` expects a 64-bit `long` argument to interpret the bits as a double value correctly.
This allows the method to convert the bit pattern into its corresponding floating-point representation accurately.

If you were to pass a smaller data type, such as an `int`, the method would receive an insufficient number of bits. As a result, the bit
the pattern would be incomplete, and the conversion into a double would not produce the expected result.

=== Noncompliant code example

[source,java]
----
int i = 42;
double d = Double.longBitsToDouble(i);  // Noncompliant
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Remove this "Double.longBitsToDouble" call.


'''
== Comments And Links
(visible only on this page)

=== on 10 Oct 2014, 14:07:37 Freddy Mallet wrote:
Perfect !

endif::env-github,rspecator-view[]
