== Why is this an issue?

Non-static initializers are rarely used, and can be confusing for most developers because they only run when new class instances are created. When possible, non-static initializers should be refactored into standard constructors or field initializers.


=== Noncompliant code example

[source,java]
----
class MyClass {
  private static final Map<String, String> MY_MAP = new HashMap<String, String>() {

    // Noncompliant - HashMap should be extended only to add behavior, not for initialization
    {
      put("a", "b");
    }

  };
}
----


=== Compliant solution

[source,java]
----
class MyClass {
  private static final Map<String, String> MY_MAP = new HashMap<String, String>();

  static {
    MY_MAP.put("a", "b");
  }
}
----
or using Java 9 ``++Map.of++``:

[source,java]
----
class MyClass {
  // Compliant
  private static final Map<String, String> MY_MAP = java.util.Map.of("a", "b");
}
----
or using Guava:

[source,java]
----
class MyClass {
  // Compliant
  private static final Map<String, String> MY_MAP = ImmutableMap.of("a", "b");
}
----



ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Move the contents of this initializer to a standard constructor or to field initializers.


'''
== Comments And Links
(visible only on this page)

=== is duplicated by: S3476

=== on 31 Jul 2013, 14:43:50 Freddy Mallet wrote:
Is implemented by \http://jira.codehaus.org/browse/SONARJAVA-260

endif::env-github,rspecator-view[]
