<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Tables used for layout should not include semantic markup, such as <code>&lt;th&gt;</code> elements, as it can confuse assistive technologies. At best this information is ignored by screen readers and makes the code less maintainable. However it can also confuse some screen readers and reduce the web page accessibility.</p>
</div>
<div class="paragraph">
<p>This rule raises an issue when a <code>&lt;table&gt;</code> element containing the <code>role</code> attribute set to <code>"presentation"</code> or <code>"none"</code> also contains any of:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>a <code>&lt;caption&gt;</code> element</p>
</li>
<li>
<p>a <code>&lt;th&gt;</code> element</p>
</li>
<li>
<p>a non-empty <code>summary</code> attribute</p>
</li>
<li>
<p>an <code>&lt;td&gt;</code> element with a <code>headers</code> or <code>scope</code> attribute</p>
</li>
</ul>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-html" data-lang="html">&lt;table role="presentation" summary="bla"&gt; &lt;!-- Noncompliant --&gt;
  &lt;caption&gt;People&lt;/caption&gt; &lt;!-- Noncompliant --&gt;
  &lt;tr&gt;
    &lt;td&gt;&lt;/td&gt;
    &lt;th&gt;Name&lt;/th&gt; &lt;!-- Noncompliant --&gt;
    &lt;th id="myid1"&gt;Age&lt;/th&gt; &lt;!-- Noncompliant --&gt;
  &lt;/tr&gt;
  &lt;tr&gt;
    &lt;td scope="row"&gt;1&lt;/td&gt; &lt;!-- Noncompliant --&gt;
    &lt;td&gt;John Doe&lt;/td&gt;
    &lt;td&gt;24&lt;/td&gt;
  &lt;/tr&gt;
  &lt;tr&gt;
    &lt;td id="myid2"&gt;2&lt;/td&gt;
    &lt;td headers="myid1 myid2"&gt;Alice Doe&lt;/td&gt; &lt;!-- Noncompliant --&gt;
    &lt;td&gt;54&lt;/td&gt;
  &lt;/tr&gt;
&lt;/table&gt;</code></pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p><a href="https://www.w3.org/WAI/WCAG21/quickref/?versions=2.0#qr-content-structure-separation-programmatic">WCAG2, 1.3.1</a> - Info and Relationships</p>
</li>
<li>
<p><a href="https://www.w3.org/TR/WCAG20-TECHS/F46.html">WCAG2, F46</a> - Failure of Success Criterion 1.3.1 due to using th elements, caption elements, or non-empty summary attributes in layout tables</p>
</li>
</ul>
</div>
<hr>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove this "&lt;X&gt;" element</p>
</div>
<div class="paragraph">
<p>Remove this "Y" attribute</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>The forbidden opening tag, without its content, or the forbidden attribute</p>
</div>
</div>
</div>
</div>