This rule raises an issue when the `go_package` option in a Protocol Buffer file uses a simple package name instead of a full module path.

== Why is this an issue?

When working with Protocol Buffers in Go, the `go_package` option tells the Protocol Buffer compiler where to place the generated Go code and what import path to use.

Using simple package names like `"mypackage"` instead of full module paths creates several problems:

* **Import conflicts**: Simple names can clash with other packages in your project or dependencies
* **Module system incompatibility**: Go's module system expects full import paths that include the module name
* **Code generation issues**: The protobuf compiler may generate code in unexpected locations
* **Maintainability problems**: Other developers cannot easily understand where the generated code belongs

Go's module system is designed around full import paths that typically include a domain name and repository path. This ensures global uniqueness and makes dependencies clear.

=== What is the potential impact?

Using simple package names can lead to import conflicts, make code harder to maintain, and cause issues with Go's module system. Generated code may be placed in unexpected locations, making it difficult for other developers to understand the project structure.

== How to fix it

Use a full module path that includes your domain and repository path instead of a simple package name. The path should follow Go's module naming conventions.

=== Code examples

==== Noncompliant code example

[source,protobuf,diff-id=1,diff-type=noncompliant]
----
syntax = "proto3";

option go_package = "mypackage"; // Noncompliant

message User {
  string name = 1;
}
----

==== Compliant solution

[source,protobuf,diff-id=1,diff-type=compliant]
----
syntax = "proto3";

option go_package = "github.com/myorg/myproject/api/v1";

message User {
  string name = 1;
}
----

== Resources

=== Documentation

 * Protocol Buffers Go Tutorial - https://protobuf.dev/getting-started/gotutorial/[Official Protocol Buffers tutorial for Go, including go_package option usage]

 * Go Modules Reference - https://go.dev/ref/mod[Official Go modules documentation explaining module paths and naming conventions]

 * Protocol Buffers Language Guide - https://protobuf.dev/programming-guides/proto3/[Comprehensive guide to Protocol Buffers syntax and options]
