== How to fix it in Core PHP

=== Code examples

include::../../common/fix/code-rationale.adoc[]

==== Noncompliant code example

[source,php,diff-id=1,diff-type=noncompliant]
----
function checkCookie()
{
    if (!isset($_COOKIE['PHPSESSID'])) {
        $value = $_GET['cookie'];
        setcookie('PHPSESSID', $value); // Noncompliant
    }

    header('Location: /welcome.php');
}
----

==== Compliant solution

[source,php,diff-id=1,diff-type=compliant]
----
function checkCookie()
{
    if (!isset($_COOKIE['PHPSESSID'])) {
        header('Location: /getcookie.php');
    }

    header('Location: /welcome.php');
}
----

include::../../common/fix/how-does-this-work.adoc[]
