== How to fix it in Symfony

=== Code examples

include::../../common/fix/code-rationale.adoc[]

==== Noncompliant code example

[source,php,diff-id=21,diff-type=noncompliant]
----
use Symfony\Component\HttpFoundation\Cookie;
use Symfony\Component\HttpFoundation\Request;
use Symfony\Component\HttpFoundation\Response;

public function checkCookie(Request $request): Response
{
    $response = $this->render('/welcome.html');

    if (!$request->cookies->has('PHPSESSID')) {
        $value = $request->query->get('cookie');
        $cookie = Cookie::create('PHPSESSID', $value);
        $response->headers->setCookie($cookie); // Noncompliant
    }

    return $response;
}
----

==== Compliant solution

[source,php,diff-id=21,diff-type=compliant]
----
use Symfony\Component\HttpFoundation\Request;
use Symfony\Component\HttpFoundation\Response;

public function checkCookie(Request $request): Response
{
    $response = $this->render('/welcome.html');

    if (!$request->cookies->has('PHPSESSID')) {
        return $this->redirectToRoute('getcookie');
    }

    return $response;
}
----

include::../../common/fix/how-does-this-work.adoc[]
