== Why is this an issue?

The use of an undefined escape sequence leads to undefined behavior. 

Escape sequences supported in every C and {cpp} standard are: `\n`, `\t`, `\v`, `\b`, `\r`, `\f`, `\a`, `\\`, `\?`, `\'`, `\"`, `\<Octal Number>`, and `\x<Hexadecimal Number>`.

C99 and {cpp}11 introduced escape sequences for Unicode characters in the form: `\u<4 Hexadecimal Digits>` and `\U<8 Hexidecimal Digits>`.

{cpp}23 supports named escape sequences for Unicode characters in the form `\N{<Name of Character>}` and a delimited version of escape sequences is supported: `\o{<Octal Number>}`, `\x{<Hexadecimal Number>}`, and `\u{<Hexadecimal Number>}`.


=== Noncompliant code example

[source,cpp]
----
const char_t a[ 2 ] = "\k";   // Noncompliant
const char_t b[ 2 ] = "\b";   // Compliant
----


== Resources

* MISRA C:2004, 4.1 - Only those escape sequences that are defined in ISO C standard shall be used.
* MISRA {cpp}:2008, 2-13-1 - Only those escape sequences that are defined in ISO/IEC 14882:2003 shall be used.


ifdef::env-github,rspecator-view[]
'''
== Comments And Links
(visible only on this page)

=== on 6 Apr 2015, 13:18:07 Evgeny Mandrikov wrote:
\[~ann.campbell.2] implementation seems more complete (SQALE, description) than this spec.

=== on 13 Apr 2015, 19:39:41 Evgeny Mandrikov wrote:
\[~ann.campbell.2] I'm wondering why not active by default as it is currently in implementation?

endif::env-github,rspecator-view[]
