<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>The actions performed by a task are often too complex or specific,
and the task name cannot describe the task&#8217;s action sufficiently well.
Moreover, build scripts usually consist of many tasks - whether explicitly defined or implicitly by the applied plugins -
so a structure is required to help the user navigate through existing tasks.</p>
</div>
<div class="paragraph">
<p>Therefore, tasks should define a <code>description</code> and <code>group</code> to provide documentation and categorization for the user.</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="sect3">
<h4 id="_improve_task_discoverability">Improve task discoverability</h4>
<div class="paragraph">
<p>Users can find tasks easier and understand their purpose if they contain a description and are grouped into categories.</p>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Assign the <code>description</code> and <code>group</code> properties in your task configuration.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-kotlin" data-lang="kotlin">tasks.register&lt;DocsGenerate&gt;("generateHtmlDocs") { // Noncompliant, does neither define "description" and "group"
    title.set("Project docs")
    outputDir.set(layout.buildDirectory.dir("docs"))
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-kotlin" data-lang="kotlin">tasks.register&lt;DocsGenerate&gt;("generateHtmlDocs") { // Compliant, defines "description" and "group"
    description = "Generates the HTML documentation for this project."
    group = JavaBasePlugin.DOCUMENTATION_GROUP
    title.set("Project docs")
    outputDir.set(layout.buildDirectory.dir("docs"))
}</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p><a href="https://docs.gradle.org/current/userguide/authoring_maintainable_build_scripts.html#sec:improving_task_discoverability">Gradle Documentation - Improve task discoverability</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>