This rule raises an issue when a class contains test methods but is not annotated with `@isTest` at the class level.

== Why is this an issue?

In Salesforce, test classes serve a specific purpose and should be properly identified using the `@isTest` annotation. This annotation provides two key benefits:

First, it excludes the test class from your organization's 6MB Apex code storage limit. Without this annotation, test classes count toward your storage quota, which can become problematic as your test suite grows.

Second, it clearly identifies the class as a test class, making your codebase more maintainable and easier to understand. This explicit marking helps developers quickly distinguish between production code and test code.

When a class contains test methods (either using the `testMethod` keyword or methods annotated with `@isTest`) but lacks the class-level `@isTest` annotation, it may consume unnecessary storage space and lacks clear identification as a test class.

=== What is the potential impact?

Without the `@isTest` annotation, test classes count toward your organization's 6MB Apex code storage limit. This can lead to storage quota issues as your test suite grows, potentially preventing you from deploying new code when the limit is reached.

Additionally, the lack of clear test class identification makes code maintenance more difficult and can confuse developers about the purpose of the class.

== How to fix it

Add the `@isTest` annotation at the class level to properly identify the class as a test class and exclude it from code storage limits.

=== Code examples

==== Noncompliant code example

[source,apex,diff-id=1,diff-type=noncompliant]
----
public class AccountTest {
    static void testAccountCreation() { // Noncompliant
        // test logic
    }
}
----

==== Compliant solution

[source,apex,diff-id=1,diff-type=compliant]
----
@isTest
public class AccountTest {
    static void testAccountCreation() {
        // test logic
    }
}
----

== Resources

=== Documentation

 * Apex Test Class Best Practices - https://www.apexhours.com/apex-test-class-best-practices[Comprehensive guide to Apex test class best practices including the use of @isTest annotation]

 * Apex Unit Test Tips - https://www.apexhours.com/12-apex-unit-test-tips-every-salesforce-developer-should-know[Essential tips for Apex unit testing including code storage limits and @isTest usage]
