This rule raises an issue when a `list` literal `[]` is used for a sequence that is defined once and never mutated. It is less efficient and less semantically clear than using a `tuple` literal `()`.

== Why is this an issue?

Python distinguishes between mutable sequences (like `list`) and immutable sequences (like `tuple`).
* **Lists (`[]`):** Can be modified after creation e.g. elements replaced, added, removed. This flexibility comes with slightly higher memory overhead and potentially slower construction and indexing compared to tuples.
* **Tuples (`()`):** Cannot be modified after creation. This immutability allows for internal optimizations, making them generally faster to create, consume less memory, and slightly faster to index.

If you define a sequence of elements e.g., configuration constants, fixed coordinates, unchanging data points using a list literal `[...]` but never modify that list later in its scope, you are missing out on the benefits of tuples. Using a tuple in such cases:
* **Signals Intent:** Clearly communicates to other developers (and static analysis tools) that the sequence is intended to be fixed and should not be changed.
* **Performance:** Can provide minor performance improvements due to tuple optimizations.
* **Safety:** Prevents accidental modification of the sequence.

Mutation is typically identified by subscript/slice assignments (`my_list[0] = x`, `my_list[1:3] = y`) or calls to list methods that modify the list in-place (e.g., `.append()`, `.extend()`, `.insert()`, `.pop()`, `.remove()`, `.sort()`, `.reverse()`). If none of these operations are applied to the list within its accessible scope, it could likely have been defined as a tuple.

== How to fix it

If a sequence is created using a list literal `[...]` but is never mutated within its scope, change the definition to use a tuple literal `(...)` instead.

=== Code examples

==== Noncompliant code example

[source,python,diff-id=1,diff-type=noncompliant]
----
def foo():
    formats = ["jpeg", "png", "gif", "webp"] # Noncompliant: This list is never modified after creation
    for fmt in formats:
        print(f"- {fmt}")
----

==== Compliant solution

[source,python,diff-id=1,diff-type=compliant]
----
def foo():
    formats = ("jpeg", "png", "gif", "webp") # Compliant: Use tuple for a fixed sequence.
    for fmt in formats:
        print(f"- {fmt}")
----

== Resources

=== Documentation

* Python3 documentation - https://docs.python.org/3/library/stdtypes.html#tuples[Python Documentation: Tuples]
* Python3 documentation - https://docs.python.org/3/library/stdtypes.html#lists[Python Documentation: Tuples]
* Python wiki - https://wiki.python.org/moin/PythonSpeed/PerformanceTips#TuplesversusLists[Python Wiki: Performance Tips - Tuples versus Lists]

