<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Commented-out code distracts the focus from the actual executed code. It creates a noise that increases maintenance code. And because it is never executed, it quickly becomes out of date and invalid.</p>
</div>
<div class="paragraph">
<p>Commented-out code should be deleted and can be retrieved from source control history if required.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Delete the commented out code.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-csharp" data-lang="csharp">void Method(string s)
{
    // if (s.StartsWith('A'))
    // {
    //     s = s.Substring(1);
    // }

    // Do something...
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-csharp" data-lang="csharp">void Method(string s)
{
    // Do something...
}</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove this commented out code.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_is_duplicated_by_s789">is duplicated by: <a data-rspec-id="S789" class="rspec-auto-link">S789</a></h3>

</div>
<div class="sect2">
<h3 id="_on_23_feb_2014_233014_freddy_mallet_wrote">on 23 Feb 2014, 23:30:14 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>Is implemented by http://jira.codehaus.org/browse/SONARPLUGINS-3515 for PHP</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_8_nov_2018_141636_elena_vilchik_wrote">on 8 Nov 2018, 14:16:36 Elena Vilchik wrote:</h3>
<div class="paragraph">
<p><a href="https://github.com/SonarSource/eslint-plugin-sonarjs/issues/45" class="bare">https://github.com/SonarSource/eslint-plugin-sonarjs/issues/45</a></p>
</div>
</div>
</div>
</div>