== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

With the default threshold of 3:

[source,scala]
----
public def run() {
  prepare("action random1")    // Noncompliant - "action random1" is duplicated 3 times
  execute("action random1")
  release("action random1")
}
----

=== Compliant solution

[source,scala]
----
public def run() {
  val action = "action random1"
  prepare(action)
  execute(action)
  release(action)
}
----

=== Exceptions

To prevent generating some false-positives, literals having 5 or less characters are excluded as well as literals containing only letters, digits and '_'.

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../parameters.adoc[]

include::../highlighting.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
