== Why is this an issue?

include::../description.adoc[]

=== Exceptions

To prevent generating some false-positives, literals having 5 or less characters are excluded as well as literals containing only letters, digits and '_'.

== How to fix it

include::../howtofix.adoc[]

=== Code examples

==== Noncompliant code example

With the default threshold of 3:

[source,scala,diff-id=1,diff-type=noncompliant]
----
public def run() {
  prepare("action random1")    // Noncompliant - "action random1" is duplicated 3 times
  execute("action random1")
  release("action random1")
}
----

==== Compliant solution

[source,scala,diff-id=1,diff-type=compliant]
----
public def run() {
  val Action = "action random1"
  prepare(Action)
  execute(Action)
  release(Action)
}
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../parameters.adoc[]

include::../highlighting.adoc[]

'''


endif::env-github,rspecator-view[]
