== Why is this an issue?

The types of the arguments to built-in functions are specified in the JavaScript language specifications. Calls to these functions should conform to the documented types, otherwise the result will most likely not be what was expected (e.g.: the call would always return ``++false++``).


=== Noncompliant code example

[source,javascript]
----
const isTooSmall = Math.abs(x < 0.0042);
----


=== Compliant solution

[source,javascript]
----
const isTooSmall = Math.abs(x) < 0.0042;
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

include::highlighting.adoc[]

endif::env-github,rspecator-view[]
