This rule raises an issue when the `include?` method is called with a regular expression argument instead of a string.

== Why is this an issue?

The `include?` method in Ruby only accepts string arguments. When you pass a regular expression to `include?`, Ruby raises a `TypeError` with the message "can't convert regexp to string".

This error occurs at runtime and will cause your application to crash. The `include?` method is designed to check if a string contains another string as a substring, not to perform pattern matching.

For pattern matching with regular expressions, Ruby provides several dedicated methods that are designed for this purpose and handle regex objects correctly.

=== What is the potential impact?

This issue causes a runtime `TypeError` that will crash the application when the problematic code is executed. This can lead to service disruption and poor user experience.

== How to fix it in Ruby

Use the `=~` operator for simple pattern matching that returns the position of the match or `nil` if no match is found.

=== Code examples

==== Noncompliant code example

[source,ruby,diff-id=1,diff-type=noncompliant]
----
if text.include?(/\d+/) # Noncompliant
  puts "Found digits"
end
----

==== Compliant solution

[source,ruby,diff-id=1,diff-type=compliant]
----
if text =~ /\d+/
  puts "Found digits"
end
----

== How to fix it in Rails

In Rails applications, use `match().present?` for a clear boolean result that handles `nil` values gracefully.

=== Code examples

==== Noncompliant code example

[source,ruby,diff-id=2,diff-type=noncompliant]
----
if user_input.include?(/[A-Z]/) # Noncompliant
  puts "Contains uppercase letters"
end
----

==== Compliant solution

[source,ruby,diff-id=2,diff-type=compliant]
----
if user_input.match(/[A-Z]/).present?
  puts "Contains uppercase letters"
end
----

== Resources

=== Documentation

 * Ruby String#include? documentation - https://ruby-doc.org/core/String.html#method-i-include-3F[Official documentation for the include? method]

 * Ruby Regexp class documentation - https://ruby-doc.org/core/Regexp.html[Official documentation for regular expressions in Ruby]

 * Ruby String pattern matching methods - https://ruby-doc.org/core/String.html#method-i-match[Documentation for String#match and related pattern matching methods]
