== Why is this an issue?

Declaring multiple variable on one line is difficult to read.

=== Noncompliant code example

[source,vbnet]
----
Module Module1
  Public Const AAA As Integer = 5, BBB = 42, CCC As String = "foo"  ' Noncompliant
End Module
----

=== Compliant solution

[source,vbnet]
----
Module Module1
  Public Const AAA As Integer = 5
  Public Const BBB = 42
  Public Const CCC as String = "foo"
End Module
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Declare "XXX" in a separate statement.


'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
