<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Being a dynamically typed language, the Python interpreter only does type checking during runtime. Getting the typing right is important as certain operations may result in a <code>TypeError</code>.</p>
</div>
<div class="paragraph">
<p>Type hints can be used to clarify the expected parameters of a function, enabling developers to better document its contract. Applying them consistently makes the code easier to read and understand.</p>
</div>
<div class="paragraph">
<p>In addition, type hints allow some development environments to offer better autocompletion and improve the precision of static analysis tools.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Add a type hint to the function parameter.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">def hello(name) -&gt; str:
    return 'Hello ' + name</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">def hello(name: str) -&gt; str:
    return 'Hello ' + name</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p><a href="https://docs.python.org/3/library/typing.html">Python3 documentation on type hints</a></p>
</li>
<li>
<p><a href="https://peps.python.org/pep-0484">PEP-484 - Type Hints</a></p>
</li>
<li>
<p><a href="https://mypy.readthedocs.io/en/stable/cheat_sheet_py3.html">Cheat sheet on type annotations as part of the mypy documentation</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>