Calling the ``++Not++`` operator twice does nothing: the second invocation undoes the first. Either this is a bug, if the operator was actually meant to be called once, or misleading if done on purpose.

== Noncompliant Code Example

----
Dim b As Boolean = False
Dim c As Boolean = Not Not b 'Noncompliant
----

== Compliant Solution

----
Dim b As Boolean = False
Dim c As Boolean = b 'Compliant
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
