== Why is this an issue?

Contrary to possible developer expectations, a template constructor will not suppress the compiler-generated copy constructor. This may lead to incorrect copy semantics for members requiring deep copies.


=== Noncompliant code example

[source,cpp]
----
class A
{
public:
  A ( );
  // A ( A const & rhs ); Example 1 - implicitly generated
  template <typename T> 
  A ( T const & rhs ) // Example 2 
  : i ( new int32_t )
  {
   *i = *rhs.i;
  }
private:
  int32_t * i; // Member requires deep copy
};

void f ( A const & a1 )
{
  A a2 ( a1 ); // Noncompliant, unexpectedly uses Example 1, which will result in a shallow copy of 'i', instead of a deep copy
}
----


== Resources

* MISRA {cpp}:2008, 14-5-2


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Add an explicit copy constructor to this class.


'''
== Comments And Links
(visible only on this page)

=== is related to: S1026

=== on 21 Oct 2014, 19:12:23 Ann Campbell wrote:
\[~samuel.mercier] please incorporate the text under the Noncompliant header either into the code as comments or the description. Also as a "contrary to developer expectations" I would not tie this to Maintainability/Understandability. I think I see your direction on that, but when the original developer doesn't understand what (s)he's coded, then I'd call it Reliability.

endif::env-github,rspecator-view[]
