== Why is this an issue?

include::../why.adoc[]

== How to fix it

=== Code examples

==== Noncompliant code example

[source,vbnet,diff-id=1,diff-type=noncompliant]
----
<Export(GetType(ISomeType))>
Public Class SomeType  ' Noncompliant: doesn't implement 'ISomeType'.
End Class
----

==== Compliant solution

[source,vbnet,diff-id=1,diff-type=compliant]
----
<Export(GetType(ISomeType))>
Public Class SomeType
    Inherits ISomeType
End Class
----

include::../resources.adoc[]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../highlighting.adoc[]

endif::env-github,rspecator-view[]
