include::../description.adoc[]

== Noncompliant Code Example

----
<Export(GetType(ISomeType))>
Public Class SomeType  // Noncompliant; doesn't implement 'ISomeType'.
End Class
----

== Compliant Solution

----
<Export(GetType(ISomeType))>
Public Class SomeType
    Inherits ISomeType
End Class
----
ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../highlighting.adoc[]

endif::env-github,rspecator-view[]
