<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>When an object has a field annotated with <code>ThreadStatic</code>, that field is shared within a given thread, but unique across threads. Since a class' static initializer is only invoked for the first thread created, it also means that only the first thread will have the expected initial values.</p>
</div>
<div class="paragraph">
<p>Instead, allow such fields to be initialized to their default values or make the initialization lazy.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-csharp" data-lang="csharp">public class Foo
{
  [ThreadStatic]
  public static object PerThreadObject = new object(); // Noncompliant. Will be null in all the threads except the first one.
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-csharp" data-lang="csharp">public class Foo
{
  [ThreadStatic]
  public static object _perThreadObject;
  public static object PerThreadObject
  {
    get
    {
      if (_perThreadObject == null)
      {
        _perThreadObject = new object();
      }
      return _perThreadObject;
    }
  }
}</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove this initialization of "xxx" or make it lazy.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_3_jun_2015_160450_ann_campbell_wrote">on 3 Jun 2015, 16:04:50 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~tamas.vajk] could you supply the code samples please? I think I could stumble through a basic example, but IMO we need to show 2 compliant versions: one with no initializer and the other with lazy initialization &amp; I have little confidence about getting the latter right.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_8_jun_2015_085745_tamas_vajk_wrote">on 8 Jun 2015, 08:57:45 Tamas Vajk wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] I&#8217;ve added the code sample</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_8_jun_2015_120246_ann_campbell_wrote">on 8 Jun 2015, 12:02:46 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>Thanks [~tamas.vajk]</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_15_jun_2015_121242_tamas_vajk_wrote">on 15 Jun 2015, 12:12:42 Tamas Vajk wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] I&#8217;ve removed the "that&#8217;s a good thing" remark from the description.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_15_jun_2015_140047_ann_campbell_wrote">on 15 Jun 2015, 14:00:47 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>okay, fine [~tamas.vajk]</p>
</div>
</div>
</div>
</div>