include::../description.adoc[]

== Noncompliant Code Example

With the default threshold of 3:


----
switch (myVariable) 
{
    case 0: // Noncompliant: 5 statements in the case
        methodCall1("");
        methodCall2("");
        methodCall3("");
        methodCall4("");
        break;
    case 1:
        ...
}
----

== Compliant Solution

----
switch (myVariable) 
{
    case 0:                  
        DoSomething()
        break;
    case 1:
        ...
}
...
private void DoSomething()
{
    methodCall1("");
    methodCall2("");
    methodCall3("");
    methodCall4("");
}
----
