In Azure Resource Manager (ARM) templates, it is possible to set `allowedValues` for various parameters.
This field is used to define a list of valid values for a parameter.
This is a way to restrict the input values provided when deploying or updating a resource using the template.

However, when it comes to a parameter defining the `location` of a resource, this practice can lead to a code smell.
Users may be unable to deploy such a template if their desired location is not included in the `allowedValues`.

== Why is this an issue?

In Azure Resource Manager (ARM) templates, it is possible to set `allowedValues` for various parameters to limit the options and maintain control.
However, when it comes to a parameter defining the `location` of a resource, this practice can lead to a code smell.
Specifically, setting `allowedValues` for a location parameter can cause issues because the locations list might not be exhaustive or suitable for all users.
Users may be unable to deploy such a template if their desired location is not included in the `allowedValues`, causing inconvenience and potential delays in their work.

== How to fix it in ARM Templates

Remove `allowedValues` for the parameter specifying the location.

=== Code examples

==== Noncompliant code example

[source,json,diff-id=1,diff-type=noncompliant]
----
{
  "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentParameters.json#",
  "contentVersion": "1.0.0.0",
  "parameters": {
      "location": {
          "type": "string",
          "metadata": {
              "description": "The location in which the resources should be deployed."
          },
          "defaultValue": "[resourceGroup().location]",
          "allowedValues": [
              "eastus",
              "westus",
              "northeurope",
              "westeurope",
              "southeastasia"
          ]
      }
  }
}
----

==== Compliant solution

[source,json,diff-id=1,diff-type=compliant]
----
{
  "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentParameters.json#",
  "contentVersion": "1.0.0.0",
  "parameters": {
      "location": {
          "type": "string",
          "metadata": {
              "description": "The location in which the resources should be deployed."
          },
          "defaultValue": "[resourceGroup().location]"
      }
  }
}
----

== How to fix it in Bicep

Remove `allowedValues` for the parameter specifying the location.

=== Code examples

==== Noncompliant code example

[source,bicep,diff-id=2,diff-type=noncompliant]
----
@allowed([
  'eastus'
  'westus'
  'northeurope'
  'westeurope'
  'southeastasia'
])  // Noncompliant
param location string = resourceGroup().location
----

==== Compliant solution

[source,bicep,diff-id=2,diff-type=compliant]
----
param location string = resourceGroup().location
----

== Resources

=== Documentation

* Azure Resource Manager - https://learn.microsoft.com/en-us/azure/azure-resource-manager/templates/best-practices#location-recommendations-for-parameters[Location recommendations for parameters]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

For ARM Templates: Remove this `allowedValues` property from the parameter specifying the location.

For Bicep: Remove this `@allowed` decorator from the parameter specifying the location.

=== Highlighting

In case of ARM Tempates , highlight the `allowedValues` property in the parameter specifying the location.

In case of Bicep, highlight the `@allowed` decorator above the parameter specifying the location.

endif::env-github,rspecator-view[]