In Azure Resource Manager (ARM) templates, it is a common practice to set `allowedValues` for various parameters to limit the options and maintain control. However, when it comes to the "location" parameter, this practice can lead to a code smell. Users may be unable to deploy such a template if their desired location is not included in the `allowedValues`.

== Why is this an issue?

In Azure Resource Manager (ARM) templates, it is a common practice to set `allowedValues` for various parameters to limit the options and maintain control. However, when it comes to the "location" parameter, this practice can lead to a code smell. Specifically, setting `allowedValues` for a location parameter can cause issues because the locations you specify may not be available in all clouds. Users may be unable to deploy such a template if their desired location is not included in the `allowedValues`, causing inconvenience and potential delays in their work.

== How to fix it in ARM Templates

Remove `allowedValues` for the location parameter.

=== Code examples

==== Noncompliant code example

[source,json,diff-id=1,diff-type=noncompliant]
----
{
  "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentParameters.json#",
  "contentVersion": "1.0.0.0",
  "parameters": {
      "location": {
          "type": "string",
          "metadata": {
              "description": "The location in which the resources should be deployed."
          },
          "defaultValue": "[resourceGroup().location]",
          "allowedValues": [
              "eastus",
              "westus",
              "northeurope",
              "westeurope",
              "southeastasia"
          ]
      }
  }
}
----

==== Compliant solution

[source,json,diff-id=1,diff-type=compliant]
----
{
  "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentParameters.json#",
  "contentVersion": "1.0.0.0",
  "parameters": {
      "location": {
          "type": "string",
          "metadata": {
              "description": "The location in which the resources should be deployed."
          },
          "defaultValue": "[resourceGroup().location]"
      }
  }
}
----

== How to fix it in Bicep

Remove `allowedValues` for the location parameter.

=== Code examples

==== Noncompliant code example

[source,bicep,diff-id=2,diff-type=noncompliant]
----
param location string {
  allowed: [
    'eastus'
    'westus'
    'northeurope'
    'westeurope'
    'southeastasia'
  ]
}
----

==== Compliant solution

[source,bicep,diff-id=2,diff-type=compliant]
----
param location string
----

== Resources
=== Documentation
* https://learn.microsoft.com/en-us/azure/azure-resource-manager/templates/best-practices#location-recommendations-for-parameters/[Location recommendations for parameters - Azure Resource Manager]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message
Remove this `allowedValues` property from the `location` parameter.

=== Highlighting
Highlight the `allowedValues` property in the `location` parameter.

endif::env-github,rspecator-view[]