== Why is this an issue?

In Blazor, recreating https://learn.microsoft.com/en-us/aspnet/core/blazor/components/event-handling#lambda-expressions[lambda expression delegates] for elements or components in a loop can lead to performance issues, especially when rendering a large number of elements.

For instance, assigning a delegate to the `@onclick` event of each button in a set can slow down rendering speed and negatively impact user experience as Blazor needs to rebuild all the delegates each time the buttons are rendered.

== How to fix it

Ensure to not use a delegate in elements rendered in loops by using a collection of objects containing the delegate as an https://learn.microsoft.com/en-us/dotnet/api/system.action[Action].

=== Code examples

==== Noncompliant code example

[source,csharp,diff-id=1,diff-type=noncompliant]
----
@for (var i = 1; i < 100; i++)
{
    var buttonNumber = i;

    <button @onclick="@(e => DoAction(e, buttonNumber))">
        Button #buttonNumber
    </button>
}

@code {
    private void DoAction(MouseEventArgs e, int button)
    {
        // Do something here
    }
}
----

==== Compliant solution

[source,csharp,diff-id=1,diff-type=compliant]
----
@foreach (var button in Buttons)
{
    <button @key="button.Id" @onclick="button.Action">
        Button #@button.Id
    </button>
}

@code {
    private List<Button> Buttons { get; set; } = new();

    protected override void OnInitialized()
    {
        for (var i = 0; i < 100; i++)
        {
            var button = new Button();

            button.Action = (e) => DoAction(e, button);

            Buttons.Add(button);
        }
    }

    private void DoAction(MouseEventArgs e, Button button)
    {
        // Do something here
    }

    private class Button
    {
        public string? Id { get; } = Guid.NewGuid().ToString();
        public Action<MouseEventArgs> Action { get; set; } = e => { };
    }
}
----

== Resources

=== Documentation

* Microsoft Learn - https://learn.microsoft.com/en-us/aspnet/core/blazor/performance#avoid-recreating-delegates-for-many-repeated-elements-or-components[ASP.NET Core Blazor performance best practices]
* Microsoft Learn - https://learn.microsoft.com/en-us/aspnet/core/blazor/components/event-handling#lambda-expressions[ASP.NET Core Blazor event handling]

