== Why is this an issue?

In Blazor, recreating https://learn.microsoft.com/en-us/aspnet/core/blazor/components/event-handling#lambda-expressions[lambda expression delegates] for elements or components in a loop can lead to performance issues, especially when rendering a large number of elements.

For instance, assigning a delegate to the `@onclick` event of each button in a set can slow down rendering speed and negatively impact user experience and performance as Blazor needs to rebuild all the delegates each time the buttons are rendered.

== How to fix it

Ensure to not use a delegate in elements rendered in loops by using a collection of objects containing the delegate as an https://learn.microsoft.com/en-us/dotnet/api/system.action[Action].

=== Code examples

==== Noncompliant code example

[source,csharp,diff-id=1,diff-type=noncompliant]
----
@for (var i = 1; i < 100; i++)
{
    var buttonNumber = i;

    <button @onclick="@(e => DoAction(e, buttonNumber))">
        Button #buttonNumber
    </button>
}

@code {
    private void DoAction(MouseEventArgs e, int button)
    {
        // Do something here
    }
}
----

==== Compliant solution

[source,csharp,diff-id=1,diff-type=compliant]
----
@foreach (var button in Buttons)
{
    <button @key="button.Id" @onclick="button.Action">
        Button #@button.Id
    </button>
}

@code {
    private List<Button> Buttons { get; set; } = new();

    protected override void OnInitialized()
    {
        for (var i = 0; i < 100; i++)
        {
            var button = new Button();

            button.Action = (e) => DoAction(e, button);

            Buttons.Add(button);
        }
    }

    private void DoAction(MouseEventArgs e, Button button)
    {
        // Do something here
    }

    private class Button
    {
        public string? Id { get; } = Guid.NewGuid().ToString();
        public Action<MouseEventArgs> Action { get; set; } = e => { };
    }
}
----

== Resources

=== Documentation

* Microsoft Learn - https://learn.microsoft.com/en-us/aspnet/core/blazor/performance#avoid-recreating-delegates-for-many-repeated-elements-or-components[ASP.NET Core Blazor performance best practices]
* Microsoft Learn - https://learn.microsoft.com/en-us/aspnet/core/blazor/components/event-handling#lambda-expressions[ASP.NET Core Blazor event handling]

=== Benchmarks

The results were generated by running the following snippet with https://github.com/dotnet/BenchmarkDotNet[BenchmarkDotNet] and https://github.com/egil/Benchmark.Blazor/tree/main[Benchmark.Blazor]:

[options="header"]
|===
| Method      | NbButtonRendered | Mean       | StdDev    | Ratio
| UseDelegate | 10               |   6.603 us | 0.0483 us |  1.00
| UseAction   | 10               |   1.994 us | 0.0592 us |  0.29
| UseDelegate | 100              |  50.666 us | 0.5449 us |  1.00
| UseAction   | 100              |   2.016 us | 0.0346 us |  0.04
| UseDelegate | 1000             | 512.513 us | 9.7561 us | 1.000
| UseAction   | 1000             |   2.005 us | 0.0243 us | 0.004
|===

Hardware configuration:

[source,text]
----
BenchmarkDotNet v0.13.9+228a464e8be6c580ad9408e98f18813f6407fb5a, Windows 10 (10.0.19045.3448/22H2/2022Update)
12th Gen Intel Core i7-12800H, 1 CPU, 20 logical and 14 physical cores
.NET SDK 8.0.100-rc.1.23463.5
  [Host]   : .NET 7.0.11 (7.0.1123.42427), X64 RyuJIT AVX2
  .NET 7.0 : .NET 7.0.11 (7.0.1123.42427), X64 RyuJIT AVX2
----