include::../../../shared_content/secrets/description.adoc[]

== Why is this an issue?

include::../../../shared_content/secrets/rationale.adoc[]

If an attacker gains access to an NPM access token, they might be able to gain access to any private package linked to this token.

=== What is the potential impact?

The exact impact of the compromise of an NPM access token varies depending on the permissions granted to this token. It can range from loss of sensitive data and source code to severe supply chain attacks.

include::../../../shared_content/secrets/impact/source_code_compromise.adoc[]

include::../../../shared_content/secrets/impact/supply_chain_attack.adoc[]

== How to fix it

include::../../../shared_content/secrets/fix/revoke.adoc[]

include::../../../shared_content/secrets/fix/vault.adoc[]

=== Code examples

==== Noncompliant code example

[source,yaml,diff-id=1,diff-type=noncompliant]
----
steps:
    - run: |
        npm install
    - env:
        NPM_TOKEN: npm_tCEMceczuiTXKQaBjGIaAezYQ63PqI972ANG
----

==== Compliant solution

[source,yaml,diff-id=1,diff-type=compliant]
----
steps:
    - run: |
        npm install
    - env:
        NPM_TOKEN: ${{ secrets.NPM_TOKEN }}
----

//=== How does this work?

//=== Pitfalls

=== Going the extra mile

include::../../../shared_content/secrets/extra_mile/permissions_scope.adoc[]

== Resources

=== Documentation

npm Docs - https://docs.npmjs.com/revoking-access-tokens[Revoking access tokens]

include::../../../shared_content/secrets/resources/standards.adoc[]
