== Why is this an issue?

This rule allows to ban declaration of some namespaces in the root element of XHML documents.


=== Noncompliant code example

With a "namespaces" value of "http://java.sun.com/jsf/facelets":

[source,html]
----
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml"
        xmlns:ui="http://java.sun.com/jsf/facelets"    <!-- Noncompliant -->
        xmlns:h="http://java.sun.com/jsf/html"  
        xmlns:f="http://java.sun.com/jsf/core">
----


=== Compliant solution

[source,html]
----
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml"
        xmlns:h="http://java.sun.com/jsf/html"  
        xmlns:f="http://java.sun.com/jsf/core">
----



ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Using "xxxxx" namespace is not allowed.


=== Parameters

.namespaces
****

Comma separated list of namespaces
****


'''
== Comments And Links
(visible only on this page)

=== on 18 Sep 2013, 09:49:58 Freddy Mallet wrote:
Is implemented by \http://jira.codehaus.org/browse/SONARPLUGINS-3154

endif::env-github,rspecator-view[]
