<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>The use of <code>options( *nopass )</code> on a parameter makes it optional, so it should always be checked before use.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-rpg" data-lang="rpg">       DCL-PR VIOLATION1 EXTPGM;
         LASTNAME Char(10);
         FIRSTNAME Char(10) OPTIONS(*NOPASS:*OMIT);
       END-PR;

       DCL-PI VIOLATION1;
         LASTNAME Char(10);
         FIRSTNAME Char(10) OPTIONS(*NOPASS:*OMIT);
       END-PI;

       dsply FIRSTNAME;
       dsply LASTNAME;  // Noncompliant; LASTNAME is optional
       *inlr = *on;</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-rpg" data-lang="rpg">       DCL-PR VIOLATION1 EXTPGM;
         LASTNAME Char(10);
         FIRSTNAME Char(10) OPTIONS(*NOPASS:*OMIT);
       END-PR;

       DCL-PI VIOLATION1;
         LASTNAME Char(10);
         FIRSTNAME Char(10) OPTIONS(*NOPASS:*OMIT);
       END-PI;

       dsply FIRSTNAME;
       IF %PARMS &gt; 1;
         dsply LASTNAME;
       EndIf;
       *inlr = *on;</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Make sure this optional parameter was actually passed before you use it.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>Parameter name</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_28_oct_2015_201015_ann_campbell_wrote">on 28 Oct 2015, 20:10:15 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>Source: user suggestion with background from http://www.itjungle.com/fhg/fhg010505-story02.html</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_29_oct_2015_134746_pierre_yves_nicolas_wrote">on 29 Oct 2015, 13:47:46 Pierre-Yves Nicolas wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] This rule looks more like a bug detection rule than something about api-design or readability.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_29_oct_2015_140211_ann_campbell_wrote">on 29 Oct 2015, 14:02:11 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>You&#8217;re right [~pierre-yves.nicolas]. Adjusted.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_29_oct_2015_140755_pierre_yves_nicolas_wrote">on 29 Oct 2015, 14:07:55 Pierre-Yves Nicolas wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] Should we also update the SQALE characteristic?</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_29_oct_2015_142029_ann_campbell_wrote">on 29 Oct 2015, 14:20:29 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>Thanks [~pierre-yves.nicolas]. That&#8217;s done too now.</p>
</div>
</div>
</div>
</div>