<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when a function name doesn&#8217;t respect the naming convention.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Shared coding conventions allow teams to collaborate efficiently. This rule checks that all function names match a provided regular expression.</p>
</div>
<div class="paragraph">
<p>According to PEP8, function names should be lowercase, with words separated by underscores as necessary to improve readability. This convention is known as "snake_case." For example: <code>calculate_area</code>, <code>print_hello</code>, <code>process_data</code></p>
</div>
<div class="paragraph">
<p>To fix this, respect the naming convention for the function. The default naming convention is snake case, as recommended by PEP8. Other naming conventions can be defined through the <code>format</code> rule parameter.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="paragraph">
<p>With the default provided regular expression: <code>^[a-z_][a-z0-9_]*$</code></p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">def MyFunction(a,b):
    ...</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">def my_function(a,b):
    ...</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>PEP8 - <a href="https://peps.python.org/pep-0008/#function-and-variable-names">Function and Variable Names</a></p>
</li>
</ul>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Rename (function|procedure) "XXXX" to match the regular expression "${format}".</p>
</div>
</div>
<div class="sect2">
<h3 id="_parameters">Parameters</h3>
<div class="sidebarblock">
<div class="content">
<div class="title">format</div>
<div class="listingblock">
<div class="content">
<pre>^[a-z_][a-z0-9_]*$</pre>
</div>
</div>
<div class="paragraph">
<p>Regular expression used to check the function names against</p>
</div>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_relates_to_s100">relates to: <a data-rspec-id="S100" class="rspec-auto-link">S100</a></h3>

</div>
</div>
</div>