<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p><code>NotImplementedException</code> is often used to mark methods which must be implemented for the overall functionality to be complete, but which the developer wants to implement later. That&#8217;s as opposed to the <code>NotSupportedException</code> which is thrown by methods which are required by base classes or interfaces, but which are not appropriate to the current class.</p>
</div>
<div class="paragraph">
<p>This rule raises an exception when <code>NotImplementedException</code> is thrown.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-text" data-lang="text">void doTheThing()
{
    throw new NotImplementedException();
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_exceptions">Exceptions</h3>
<div class="paragraph">
<p>Exceptions derived from <code>NotImplementedException</code> are ignored.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Implement this method or throw 'NotSupportedException' instead.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p><code>throw new NotImplementedException()</code></p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_12_sep_2016_171407_ann_campbell_wrote">on 12 Sep 2016, 17:14:07 Ann Campbell wrote:</h3>
<div class="ulist">
<ul>
<li>
<p>.Net: <a href="https://msdn.microsoft.com/en-us/system.notimplementedexception">NotImplementedException</a></p>
</li>
<li>
<p>Python: <a href="https://docs.python.org/2/library/exceptions.html#exceptions.NotImplementedError">NotImplementedError</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>