== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,cpp]
----
try {
  saveDocument();
} catch (const std::exception& e) { // Noncompliant
  throw;
}
----

=== Compliant solution

[source,cpp]
----
try {
  saveDocument();
} catch (const std::exception& e) { // Compliant
  log << e.what();
  throw;
}
----
or

[source,cpp]
----
saveDocument();
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
