include::../description.adoc[]

== Noncompliant Code Example

[source,cpp]
----
try {
  saveDocument();
} catch (const std::exception& e) { // Noncompliant
  throw;
}
----

== Compliant Solution

[source,cpp]
----
try {
  saveDocument();
} catch (const std::exception& e) { // Compliant
  log << e.what();
  throw;
}
----
or

----
saveDocument();
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
