<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>The output of an <code>as</code> cast will be null if the input to the cast cannot safely be cast to the desired type. So it makes sense that after such a cast you would null-check the output. But it doesn&#8217;t make sense to check the input.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-csharp" data-lang="csharp">void DoTheThing(Toy toy)
{
  Ball ball = toy as Ball;
  if (toy != null) // Noncompliant
  {
    //...
  }
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-csharp" data-lang="csharp">void DoTheThing(Toy toy)
{
  Ball ball = toy as Ball;
  if (ball != null)
  {
    //...
  }
}</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_18_jan_2016_113322_ann_campbell_wrote">on 18 Jan 2016, 11:33:22 Ann Campbell wrote:</h3>
<div class="paragraph">
<p><a href="http://www.viva64.com/en/b/0363/#ID0EVHBI" class="bare">http://www.viva64.com/en/b/0363/#ID0EVHBI</a></p>
</div>
</div>
<div class="sect2">
<h3 id="_on_8_feb_2016_134013_ann_campbell_wrote">on 8 Feb 2016, 13:40:13 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>subsumed by <a data-rspec-id="S2583" class="rspec-auto-link">RSPEC-2583</a></p>
</div>
</div>
</div>
</div>