== Why is this an issue?

Shared naming conventions allow teams to collaborate efficiently. This rule checks that all method parameters follow a naming convention.


=== Noncompliant code example

Using the default importing  and returning regular expressions ``++I_[a-zA-Z_\-0-9]+++``, and ``++R_[a-zA-Z_\-0-9]+++``:

[source,text]
----
    METHODS m_fact IMPORTING i1  TYPE i
                       value(i2) TYPE i
                   RETURNING value(factorial) TYPE i.
----


=== Compliant solution

[source,text]
----
    METHODS m_fact IMPORTING I_1  TYPE i
                       value(i2) TYPE i
                   RETURNING value(R_factorial) TYPE i.
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../parameters.adoc[]

endif::env-github,rspecator-view[]
