== Why is this an issue?

It is common for methods to check the value of their parameters or the state of
their associated object and throw an exception when one of them does not match a
given condition.
Those conditions are usually mentioned in the javadoc of the method.

This rule raises an issue when it detects that a method call will trigger one of
the following exceptions:

* `java.lang.IllegalArgumentException`,
* `java.lang.IllegalStateException`.


=== Noncompliant code example

[source,java]
----
  private void addFirst(Collection<String> collection, String element) {
    if (collection instanceof List) {
      ((List<String>) collection).add(0, element);
    } else {
      throw new IllegalArgumentException();
    }
  }

  public void caller() {
    // Noncompliant: triggers an IllegalArgumentException because HashSet doesn't implement List
    addFirst(new HashSet<>(), "hello");
  }
----

