include::../description.adoc[]

include::../how-to-fix-it.adoc[]

=== Code examples

==== Noncompliant code example

[source,vb6,diff-id=1,diff-type=noncompliant]
----
Function blnCheckSize(dblParameter As Double) As Boolean
  If dblParameter > 1024 Then      ' Noncompliant, 1024 is a magic number
    blnCheckSize = True
  End If
End Function
----

==== Compliant solution

[source,vb6,diff-id=1,diff-type=compliant]
----
Function blnCheckSize(dblParameter As Double) As Boolean
  Dim threshold As Integer = 1024  ' Compliant

  If dblParameter > threshold Then
    blnCheckSize = True
  End If
End Function
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
