include::../description.adoc[]

== Noncompliant Code Example

[source,cs]
----
public int Foo
{
    get
    {
        throw new Exception(); // Noncompliant
    }
}
----

== Compliant Solution

[source,cs]
----
public int Foo
{
    get
    {
        return 42;
    }
}
----

include::../exceptions.adoc[]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
