== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,csharp]
----
public int Foo
{
    get
    {
        throw new Exception(); // Noncompliant
    }
}
----

=== Compliant solution

[source,csharp]
----
public int Foo
{
    get
    {
        return 42;
    }
}
----

include::../exceptions.adoc[]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
