<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>For better readability, column aliases should be used with the <code>AS</code> keyword. If it is missing, it could be misread as another column being selected.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-sql" data-lang="sql">DECLARE
  result DUAL.dummy%TYPE;
BEGIN
  SELECT
    dummy d -- Non-Compliant - could be misread as selecting both "dummy" and a column "d"
  INTO
    result
  FROM
    DUAL;
END;
/</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-sql" data-lang="sql">DECLARE
  result DUAL.dummy%TYPE;
BEGIN
  SELECT
    dummy AS d -- Compliant
  INTO
    result
  FROM
    DUAL;
END;
/</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Add the "AS" keyword to this alias of "xxx".</p>
</div>
</div>
</div>
</div>