For better readability, column aliases should be used with the ``++AS++`` keyword. If it is missing, it could be misread as another column being selected.


== Noncompliant Code Example

----
DECLARE
  result DUAL.dummy%TYPE;
BEGIN
  SELECT
    dummy d -- Non-Compliant - could be misread as selecting both "dummy" and a column "d"
  INTO
    result
  FROM
    DUAL;
END;
/
----


== Compliant Solution

----
DECLARE
  result DUAL.dummy%TYPE;
BEGIN
  SELECT
    dummy AS d -- Compliant
  INTO
    result
  FROM
    DUAL;
END;
/
----

