<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>The Kotlin <code>when</code> statement can not only be used to compare a selector against a list of values,
but also to select from a list of conditions.</p>
</div>
<div class="paragraph">
<p>This is a more readable and elegant alternative to a chain of <code>if</code> statements and should be used instead.</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="sect3">
<h4 id="_readability_and_understanding">Readability and Understanding</h4>
<div class="paragraph">
<p>This change makes it easier to understand a function because it will reduce its complexity.
This is because a single <code>when</code> statement has less cognitive complexity than multiple <code>if</code> statements, even if it has as many cases.</p>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Refactor the chain of <code>if</code> statements into a list of conditions for a <code>when</code> statement.
Note that the order of the conditions must stay the same as in the <code>if</code> chain.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-kotlin" data-lang="kotlin">fun dispatchFunction(instance: Any) {
    if (instance is Foo) {
        instance.fooFunction()
    } else if (instance is Bar) {
        instance.barFunction()
    } else if (instance is Boo) {
        instance.booFunction()
    } else {
        throw IllegalArgumentException()
    }
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-kotlin" data-lang="kotlin">fun dispatchFunction(instance: Any) {
    when (instance) {
        is Foo -&gt; instance.fooFunction()
        is Bar -&gt; instance.barFunction()
        is Boo -&gt; instance.booFunction()
        else -&gt; throw IllegalArgumentException()
    }
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_noncompliant_code_example_2">Noncompliant code example</h4>
<div class="paragraph">
<p>When <code>threshold = 2</code></p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-kotlin" data-lang="kotlin">fun compare(a: Int, b: Int): Int {
    // ...
    return if (a &gt; b) { // Noncompliant
        1
    } else if (a &lt; b) {
        -1
    } else {
        0
    }
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution_2">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-kotlin" data-lang="kotlin">fun compare(a: Int, b: Int): Int {
    // ...
    return when { // Compliant
        a &gt; b -&gt; 1
        a &lt; b -&gt; -1
        else -&gt; 0
    }
}</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p><a href="https://kotlinlang.org/docs/control-flow.html#when-expression">Kotlin Docs, When expression</a></p>
</li>
</ul>
</div>
</div>
<div class="sect2">
<h3 id="_articles_blog_posts">Articles &amp; blog posts</h3>
<div class="ulist">
<ul>
<li>
<p><a href="https://www.baeldung.com/kotlin/when">Baeldung, Guide to the “when{}” Block in Kotlin</a></p>
</li>
<li>
<p><a href="https://superkotlin.com/kotlin-when-statement">Kotlin when: A switch with Superpowers</a></p>
</li>
<li>
<p><a href="https://www.sonarsource.com/resources/cognitive-complexity">G. Ann Campbell, Cognitive Complexity</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_parameters">Parameters</h3>
<div class="sidebarblock">
<div class="content">
<div class="title">threshold</div>
<div class="listingblock">
<div class="content">
<pre>3</pre>
</div>
</div>
<div class="paragraph">
<p>Number of "if" after which the chain should be replaced by a "when" statement.</p>
</div>
</div>
</div>
</div>
</div>
</div>