<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Shared coding conventions make it possible for a team to efficiently collaborate. This rule makes it mandatory to place a close curly brace at the beginning of a line.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-csharp" data-lang="csharp">if(condition)
{
  doSomething();}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-csharp" data-lang="csharp">if(condition)
{
  doSomething();
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_exceptions">Exceptions</h3>
<div class="paragraph">
<p>When blocks are inlined (open and close curly braces on the same line), no issue is triggered.</p>
</div>
<div class="listingblock">
<div class="content">
<pre>if(condition) {doSomething();}</pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Move this closing curly brace to the next line.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_relates_to_s1107">relates to: <a data-rspec-id="S1107" class="rspec-auto-link">S1107</a></h3>

</div>
<div class="sect2">
<h3 id="_relates_to_s1108">relates to: <a data-rspec-id="S1108" class="rspec-auto-link">S1108</a></h3>

</div>
<div class="sect2">
<h3 id="_on_3_jul_2013_160542_freddy_mallet_wrote">on 3 Jul 2013, 16:05:42 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>Is implemented by http://jira.codehaus.org/browse/SONARJAVA-193</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_8_jul_2013_072839_dinesh_bolkensteyn_wrote">on 8 Jul 2013, 07:28:39 Dinesh Bolkensteyn wrote:</h3>
<div class="paragraph">
<p>I don&#8217;t think this rule was meant to be enabled by default</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_8_jul_2013_072945_dinesh_bolkensteyn_wrote">on 8 Jul 2013, 07:29:45 Dinesh Bolkensteyn wrote:</h3>
<div class="paragraph">
<p>It seems that all code snippets do not compile as the miss the mandatory '()' on 'myMethod'</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_18_jul_2013_070717_dinesh_bolkensteyn_wrote">on 18 Jul 2013, 07:07:17 Dinesh Bolkensteyn wrote:</h3>
<div class="paragraph">
<p>I&#8217;ve upaded the title following Ann&#8217;s change, and to be consistent with the rules on left curly braces.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_19_feb_2014_075404_freddy_mallet_wrote">on 19 Feb 2014, 07:54:04 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>Is implemented by http://jira.codehaus.org/browse/SONARPLUGINS-3487 for PHP</p>
</div>
</div>
</div>
</div>