== Why is this an issue?

include::description.adoc[]

=== Noncompliant code example

[source,text]
----
if(condition) {
  doSomething();} 
----


=== Compliant solution

[source,text]
----
if(condition) {
  doSomething();
} 
----


include::exceptions.adoc[]
