This rule raises an issue when an empty string literal is concatenated with another value using the `+` operator.

== Why is this an issue?

Concatenating an empty string with another value is an inefficient way to convert that value to a string. This pattern creates unnecessary string objects and obscures the developer's intent.

In Groovy, when you use `'' + someValue`, the runtime must:

* Create a new empty string object
* Convert `someValue` to a string representation
* Concatenate the empty string with the converted value
* Return the result

This approach is less efficient than direct string conversion methods and makes the code less readable. The intent (converting a value to a string) is not immediately clear to other developers reading the code.

Moreover, this pattern can lead to unexpected behavior with null values, as `'' + null` results in the string `"null"` rather than an empty string or a null pointer exception that might be more appropriate depending on the context.

== How to fix it

Replace empty string concatenation with explicit `toString()` calls. For potentially null values, use safe navigation with null coalescing to handle null cases appropriately.

=== Code examples

==== Noncompliant code example

[source,groovy,diff-id=1,diff-type=noncompliant]
----
def variable = method('' + property)
----

==== Compliant solution

[source,groovy,diff-id=1,diff-type=compliant]
----
def variable = method(property?.toString() ?: "")
----

== Resources

=== Documentation

 * Groovy documentation - https://groovy-lang.org/syntax.html#_string_summary_table[String handling and conversion methods]

 * Groovy documentation - https://groovy-lang.org/operators.html#_safe_navigation_operator[Safe navigation operator]
