include::../description.adoc[]

include::../ask-yourself.adoc[]

include::../recommended.adoc[]

== Noncompliant Code Example

----
FIXME
----

== Compliant Solution

----
FIXME
----

== See

include::../see.adoc[]

ifdef::env-github,rspecator-view[]
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../highlighting.adoc[]
endif::env-github,rspecator-view[]