include::../description.adoc[]

include::../ask-yourself.adoc[]

include::../recommended.adoc[]

== Noncompliant Code Example

----
<application
    android:allowBackup="true"> <!-- Sensitive -->
</application>
----

== Compliant Solution

Disable application backup.
----
<application
    android:allowBackup="false">
</application>
----

If targeting Android 6.0 or above (API level 23), define files to include/exclude from the application backup.
<application
    android:allowBackup="true"
    android:fullBackupContent="@xml/backup.xml">
</application>

== See

include::../see.adoc[]

ifdef::env-github,rspecator-view[]
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../highlighting.adoc[]
endif::env-github,rspecator-view[]