<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p><code>StringBuffer</code> and <code>StringBuilder</code> instances that are <code>append</code>ed but never <code>toString</code>ed needlessly clutter the code, and worse are a drag on performance. Either they should be removed, or the missing <code>toString</code> call added.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">public void doSomething(List&lt;String&gt; strings) {

  StringBuilder sb = new StringBuilder();  // Noncompliant
  sb.append("Got: ");
  for (String str : strings) {
    sb.append(str).append(", ");
    // ...
  }
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">public void doSomething(List&lt;String&gt; strings) {

  for (String str : strings) {
    // ...
  }
}</code></pre>
</div>
</div>
<div class="paragraph">
<p>or</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">public void doSomething(List&lt;String&gt; strings) {

  StringBuilder sb = new StringBuilder();
  sb.append("Got: ");
  for (String str : strings) {
    sb.append(str).append(", ");
    // ...
  }

  LOGGER.info(sb.toString);
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_exceptions">Exceptions</h3>
<div class="paragraph">
<p>This rule ignores <code>StringBuffer</code>s and <code>StringBuilder</code>s that are passed as method arguments on the grounds that they are likely <code>toString</code>ed there.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove this "String[Buffer|Builder]"; it is appended to but ".toString" is never called.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_relates_to_s3075">relates to: <a data-rspec-id="S3075" class="rspec-auto-link">S3075</a></h3>

</div>
<div class="sect2">
<h3 id="_on_15_jun_2015_153013_ann_campbell_wrote">on 15 Jun 2015, 15:30:13 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>CodePro: Unused StringBuffer, UnusedStringBuilder</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_15_jun_2015_201752_nicolas_peru_wrote">on 15 Jun 2015, 20:17:52 Nicolas Peru wrote:</h3>
<div class="paragraph">
<p>Looks good</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_18_sep_2015_124418_ann_campbell_wrote">on 18 Sep 2015, 12:44:18 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>fyi [~tamas.vajk]</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_15_oct_2015_071206_tamas_vajk_wrote">on 15 Oct 2015, 07:12:06 Tamas Vajk wrote:</h3>
<div class="paragraph">
<p>Thanks [~ann.campbell.2]. I see one problem with this rule. We might pass the <code>StringBuilder</code> to a method, which internally calls <code>ToString</code>, but at the declaring scope we won&#8217;t know this. So if the <code>StringBuilder</code> is passed to a method we should not report on it.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_15_oct_2015_113213_ann_campbell_wrote">on 15 Oct 2015, 11:32:13 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>Thanks [~tamas.vajk], I&#8217;ve added an explicit exception.</p>
</div>
</div>
</div>
</div>