<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p><code>StringBuilder</code> instances that never build a <code>string</code> clutter the code and worse are a drag on performance. Either they should be removed, or the missing <code>ToString()</code> call should be added.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-vbnet" data-lang="vbnet">Public Sub DoSomething(ByVal strings As List(Of String))
    Dim sb As StringBuilder = New StringBuilder() ' Noncompliant
    sb.Append("Got: ")

    For Each str As String In strings
        sb.Append(str).Append(", ")
    Next
End Sub</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-vbnet" data-lang="vbnet">Public Sub DoSomething(ByVal strings As List(Of String))
    For Each str As String In strings
    Next
End Sub</code></pre>
</div>
</div>
<div class="paragraph">
<p>or</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-vbnet" data-lang="vbnet">Public Sub DoSomething(ByVal strings As List(Of String))
    Dim sb As StringBuilder = New StringBuilder()
    sb.Append("Got: ")

    For Each str As String In strings
        sb.Append(str).Append(", ")
    Next

    My.Application.Log.WriteEntry(sb.ToString())
End Sub</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_exceptions">Exceptions</h3>
<div class="paragraph">
<p>No issue is reported when <code>StringBuilder</code> is:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>Accessed through <code>sb.CopyTo()</code>, <code>sb.GetChunks()</code>, <code>sb.Length</code>, or <code>sb(index)</code>.</p>
</li>
<li>
<p>Passed as a method argument, on the grounds that it will likely be accessed through a <code>ToString()</code> invocation there.</p>
</li>
<li>
<p>Passed in as a parameter to the current method, on the grounds that the callee will materialize the string.</p>
</li>
<li>
<p>Retrieved by a custom function (<code>Dim sb As StringBuilder = GetStringBuilder()</code>).</p>
</li>
<li>
<p>Returned by the method.</p>
</li>
</ul>
</div>
</div>
</div>
</div>