== How to fix it in ``++DBMS_CRYPTO++``

=== Code examples

include::../../common/fix/code-rationale.adoc[]

==== Noncompliant code example

[source,sql,diff-id=1,diff-type=noncompliant]
----
PLS_INTEGER := DBMS_CRYPTO.ENCRYPT_DES              // Noncompliant
                           + DBMS_CRYPTO.CHAIN_CBC
                           + DBMS_CRYPTO.PAD_PKCS5;
----

==== Compliant solution

[source,sql,diff-id=1,diff-type=compliant]
----
PLS_INTEGER := DBMS_CRYPTO.ENCRYPT_AES256
                           + DBMS_CRYPTO.CHAIN_CBC
                           + DBMS_CRYPTO.PAD_PKCS5;
----

=== How does this work?

include::../../common/fix/strong-cryptography.adoc[]

