include::../why-dotnet.adoc[]

=== Code examples

==== Noncompliant code example

[source,csharp,diff-id=1,diff-type=noncompliant]
----
public void Method(ILogger logger, bool parameter)
{
    logger.DebugFormat($"The value of the parameter is: {parameter}.");
}
----

==== Compliant solution

[source,csharp,diff-id=1,diff-type=compliant]
----
public void Method(ILogger logger, bool parameter)
{
    logger.DebugFormat("The value of the parameter is: {Parameter}.", parameter);
}
----

include::../resources-dotnet.adoc[]