<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>The <code>raise a, b</code> syntax is deprecated in Python 3, and should no longer be used.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">try:
  if not is_okay:
    raise Exception, 'It\'s not okay'  # Noncompliant
except Exception as e:
  # ...</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">try:
  if not is_okay:
    raise Exception('It\'s not okay')
except Exception as e:
  # ...</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Use "x" to raise an exception here.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_21_apr_2015_090709_elena_vilchik_wrote">on 21 Apr 2015, 09:07:09 Elena Vilchik wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] I changed the message. For code example it could be "Use "Exception('It&#8217;s not okay')" to raise an exception here." wdyt?</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_21_apr_2015_092419_ann_campbell_wrote">on 21 Apr 2015, 09:24:19 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~elena.vilchik] that message is fine with me. Note, however, that in the past we&#8217;ve steered away from such messages because of potential message length. Limitations in the platform - which have since been removed - were partly responsible for that.</p>
</div>
</div>
</div>
</div>