The ``++raise a, b++`` syntax is deprecated in Python 3, and should no longer be used.


== Noncompliant Code Example

----
try:
  if not is_okay:
    raise Exception, 'It\'s not okay'  # Noncompliant
except Exception as e:
  # ...
----


== Compliant Solution

----
try:
  if not is_okay:
    raise Exception('It\'s not okay')
except Exception as e:
  # ...
----


