<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>The <code>elseif</code> keyword is preferred over the <code>else if</code> keywords in PHP due to its consistency with other programming languages.</p>
</div>
<div class="paragraph">
<p><code>elseif</code> is a single token, making it more efficient for the PHP parser to process. Additionally, using <code>elseif</code> encourages developers to follow a unified coding style and maintain a consistent syntax throughout their PHP codebase.</p>
</div>
<div class="paragraph">
<p>Overall, the use of <code>elseif</code> improves code clarity, reduces potential errors, and enhances the overall maintainability of PHP projects.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-php" data-lang="php">if ($expr1) {
  ...
} else if ($expr2) {
  ...
} else {...}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-php" data-lang="php">if ($expr1) {
  ...
} elseif ($expr2) {
  ...
} else {...}</code></pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p><a href="https://www.php.net/manual/de/control-structures.elseif.php">PHP Manual - elseif/else if</a></p>
</li>
</ul>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Replace this "else if" keyword sequence by "elseif" keyword.</p>
</div>
</div>
</div>
</div>