== Why is this an issue?

An inherited member function can be hidden in a derived class and that creates a class that behaves differently depending on which interface is used to manipulate it.


Overriding happens when the inherited method is virtual and a method declared in the derived class uses the same identifier as well as the same signature (the return types can be different, as long as they are covariant). However, if the inherited method is non-virtual or if the two declarations of the method do not share the same signature, the method of the base class will be hidden.


Such a class increases the inheritance complexity, and confuses consumers with its non-polymorphic behavior, which can lead to errors.


=== Noncompliant code example

[source,cpp]
----
class Base {
public:
  void shutdown();
  virtual void log(int a);
};

class Derived : public Base {
public:
  void shutdown(); //Noncompliant
  void log(float a); //Noncompliant
};

void stopServer(Base *obj, Derived *obj2) {
  obj->shutdown(); // always calls Base::shutdown even if the given object's type is Derived
  obj->log(2); // calls Base::log(int) even if the given object's type is Derived
  obj2->shutdown(); // calls Derived::shutdown
  obj2->log(2); // calls Derived::log(float), even if this requires a conversion int->float.
}
----


=== Compliant solution

[source,cpp]
----
class Base {
public:
  void shutdown();
  virtual void log(int a);
};

class Derived : public Base {
public:
  void shutdownAndUpdate(); // Define a method with a different name
  void log(int a) override; // Or make the method a proper override
};

void stopServer(Base *obj) {
  obj->shutdown(); // calls Base::shutdown and there is no confusion
  obj->log(2); // calls Derived::log(int) if the given object's type is Derived
}
----


== Resources

* {cpp} Core Guidelines - https://github.com/isocpp/CppCoreGuidelines/blob/e49158a/CppCoreGuidelines.md#c138-create-an-overload-set-for-a-derived-class-and-its-bases-with-using[C.138: Create an overload set for a derived class and its bases with `using`]


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Rename this member function so that it doesn't hide an inherited non-virtual function, or make it virtual in the base class "XXX".

"XXX" hides overloaded virtual functions


=== Highlighting

Function name


'''
== Comments And Links
(visible only on this page)

=== on 2 May 2016, 16:32:04 Ann Campbell wrote:
I've made some small edits [~alban.auzeill], but basically this looks good.

endif::env-github,rspecator-view[]
