For readability, SAP recommends that asterisks (``++*++``) only be used to comment out header lines and code. Commentary should be commented using a double quote (``++"++``)


== Noncompliant Code Example

----
* GAC -  13 June 13 - output user data
* WRITE: / 'Firstname'.
----


== Compliant Solution

----
" GAC -  13 June 13 - output user data
* WRITE: / 'Firstname'.
----


== Exceptions

This rule ignores code that is included in a commentary block and commented as commentary.

----
" This is a commentary block. It could go on for a number of lines.
" As we see in the code sample below...
" WRITE: / 'Firstname'.
" It would have been better to comment the line above as code (i.e. with an asterisk) 
" but not doing so is accepted.
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

endif::env-github,rspecator-view[]
