== Why is this an issue?

For readability, SAP recommends that asterisks (``++*++``) only be used to comment out header lines and code. Commentary should be commented using a double quote (``++"++``)


=== Noncompliant code example

[source,abap]
----
* GAC -  13 June 13 - output user data
* WRITE: / 'Firstname'.
----


=== Compliant solution

[source,abap]
----
" GAC -  13 June 13 - output user data
* WRITE: / 'Firstname'.
----


=== Exceptions

This rule ignores code that is included in a commentary block and commented as commentary.

[source,abap]
----
" This is a commentary block. It could go on for a number of lines.
" As we see in the code sample below...
" WRITE: / 'Firstname'.
" It would have been better to comment the line above as code (i.e. with an asterisk) 
" but not doing so is accepted.
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Use a double-quote to mark this comment line instead of an asterisk


endif::env-github,rspecator-view[]
