<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when a function or method is called with an argument of a different type than the one described in its type annotations.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>The CPython interpreter does not check types of arguments when functions are called.
However, a function can express the type it expects for each argument in its documentation or
by using <a href="https://www.python.org/dev/peps/pep-0484/">Type Hints</a>.
While the code may initially work as intended, not respecting the contract of an API may lead to bugs later
when its implementation evolves or when type checks are added (i.e. with <code>isinstance</code>).</p>
</div>
<div class="paragraph">
<p>This rule also checks argument types for built-in functions.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">def func(var: str):
    pass

func(42)  # Noncompliant: 42 is not of type str.

round("not a number")  # Noncompliant: the builtin function round requires a number as first parameter.</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">def func(var: str):
    pass

func("42")

round(1.2)</code></pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Python documentation - <a href="https://docs.python.org/3/library/functions.html#built-in-funcs">builtins</a></p>
</li>
<li>
<p>Python documentation - <a href="https://docs.python.org/3/library/typing.html">typing â€” Support for type hints</a></p>
</li>
<li>
<p>PEP 484 - <a href="https://www.python.org/dev/peps/pep-0484/">Type Hints</a></p>
</li>
</ul>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="ulist">
<ul>
<li>
<p>Change this argument of type XXX; Function FFF expects type YYY</p>
</li>
</ul>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>Primary: the expression provided as argument</p>
</div>
<div class="paragraph">
<p>Secondary:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>location: definition of the function called</p>
</li>
<li>
<p>message: "Function definition"</p>
</li>
</ul>
</div>
</div>
</div>
</div>