== Why is this an issue?

Automatic variables are those that are allocated automatically when program flow enters the object's scope, and deallocated automatically when program flow leaves. Therefore returning the address of an automatic variable or object is an error because by the time the calling code attempts to use the value at the returned address, it no longer exists.


Unfortunately, this type of error is not always immediately evident. While the relevant memory has been deallocated, it may not have been overwritten by the time the object is dereferenced, thus leading to unpredictable behavior because sometimes the dereference is fully successful, sometimes it is partially successful (only parts of the object have been overwritten) and other times the dereference is a complete failure.


=== Noncompliant code example

[source,cpp]
----
int * divide (int numerator, int denominator) {
  int result = numerator / denominator;
  return (&result);
} 
----


=== Compliant solution

[source,cpp]
----
int divide (int numerator, int denominator) {
  int result = numerator / denominator;
  return result;
} 
----


== Resources

* MISRA {cpp}:2008, 7-5-1 - A function shall not return a reference or a pointer to an automatic variable (including parameters), defined within the function.
* https://wiki.sei.cmu.edu/confluence/x/UtcxBQ[CERT, DCL30-C.] - Declare objects with appropriate storage durations


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

XXX will be deallocated automatically when program flow leaves this function.


'''
== Comments And Links
(visible only on this page)

=== relates to: S2997

=== relates to: S946

=== on 28 Sep 2014, 21:05:43 Evgeny Mandrikov wrote:
What about MISRA C:2004 17.6 and MISRA C:2012 18.6, i.e. RSPEC-946?

=== on 2 Oct 2014, 20:32:31 Ann Campbell wrote:
\[~evgeny.mandrikov] I'd say the C rules are related but have a larger scope. 


----
typedef struct {
  int *a;
} my_struct;

void doBad(my_struct *ms) {
  int i;
  my->a = &i;  // Violates C rule but not C++ rule
}
----

Would you like to cover the larger scope instead?

endif::env-github,rspecator-view[]
