The goal of this coding rule is to make it obvious at first look that a class ``extends`` some other classes and/or ``implements`` some interfaces. The names of extended classes or implemented interfaces can be located on next lines. 


== Noncompliant Code Example

----
<?php
class ClassName 
extends ParentClass {...}
----


== Compliant Solution

----
<?php
class ClassName extends ParentClass {...}
----

or 

----
<?php
class ClassName extends 
ParentClass {...}
----

