== Why is this an issue?

The goal of this coding rule is to make it obvious at first look that a class ``++extends++`` some other classes and/or ``++implements++`` some interfaces. The names of extended classes or implemented interfaces can be located on next lines. 


=== Noncompliant code example

[source,text]
----
<?php
class ClassName 
extends ParentClass {...}
----


=== Compliant solution

[source,text]
----
<?php
class ClassName extends ParentClass {...}
----

or 


[source,text]
----
<?php
class ClassName extends 
ParentClass {...}
----

