== Why is this an issue?

Declaring a member function static or const limits its access to the non-static data members.


This helps to prevent unintentional modification of the data, and facilitates compliance with MISRA {cpp} 2008 Rule 7–1–1 (A variable which is not modified shall be const qualified).


=== Noncompliant code example

[source,cpp]
----
class A
{
public:
  int16_t f1 ( ) // Non-compliant – can be const
  {
    return m_i;
  }
  int16_t f2 ( ) // Non-compliant – can be static
  {
    return m_s;
  }
private:
  int16_t m_i;
  static int16_t m_s;
};
----


=== Compliant solution

[source,cpp]
----
class A
{
public:
  int16_t const f1 ( ) // Compliant
  {
    return m_i;
  }
  static int16_t f2 ( ) // Compliant
  {
    return m_s;
  }
  int16_t f3 ( ) // Compliant – cannot be const or static
  {
    return ++m_i;
  }
private:
  int16_t m_i;
  static int16_t m_s;
};
----


== Resources

* MISRA {cpp}:2008, 9-3-3


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
