== Why is this an issue?

include::../description.adoc[]

== How to fix it

Rename packages with the expected naming convention

=== Code examples

==== Noncompliant code example

With the default regular expression ``++^[a-z]+(\.[a-z][a-z0-9]*)*$++``:

[source,flex,diff-id=1,diff-type=noncompliant]
----
package org.Example {  // Noncompliant
...
}
----

==== Compliant solution

[source,flex,diff-id=1,diff-type=compliant]
----
package org.example { 
...
}
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

=== Parameters

.format
****

----
^[a-z]+(\.[a-z][a-z0-9]*)*$
----

Regular expression used to check the package names against.
****


'''

endif::env-github,rspecator-view[]
