include::../description.adoc[]

== Noncompliant Code Example

With the default regular expression `+^[a-z]+(\.[a-z][a-z0-9]*)*$+`:

----
package org.Example {  // Noncompliant
...
}
----

== Compliant Solution

----
package org.example { 
...
}
----
