== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

With the default regular expression ``++^[a-z]+(\.[a-z][a-z0-9]*)*$++``:

[source,flex]
----
package org.Example {  // Noncompliant
...
}
----

=== Compliant solution

[source,flex]
----
package org.example { 
...
}
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../parameters.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
