include::../description.adoc[]

== Noncompliant Code Example

With the default regular expression ``++^[a-z]+(\.[a-z][a-z0-9]*)*$++``:

----
package org.Example {  // Noncompliant
...
}
----

== Compliant Solution

----
package org.example { 
...
}
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../parameters.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
