This rule raises an issue when Database DML operations return SaveResult objects that are not checked for errors using the `isSuccess()` method.

== Why is this an issue?

When performing Database DML operations in Apex (such as `Database.insert()`, `Database.update()`, `Database.upsert()`, or `Database.delete()`), these methods return `SaveResult` objects that contain information about whether the operation succeeded or failed.

If you don't check these results for errors, your code may experience silent failures. This means that when a DML operation fails, your application continues running as if nothing went wrong, but the expected data changes never actually happened.

Silent failures can lead to serious problems:

* Data integrity issues where records are not created or updated as expected
* Business processes that appear to complete successfully but actually fail partway through
* Difficult-to-debug issues where the root cause is hidden
* Poor user experience when operations seem to work but don't actually save data

The `SaveResult` object provides two key methods: `isSuccess()` to check if the operation succeeded, and `getErrors()` to retrieve detailed error information when it fails. By checking these results, you can handle errors appropriately, log meaningful error messages, and provide proper feedback to users.

=== What is the potential impact?

Silent failures in DML operations can cause data integrity issues, incomplete business processes, and poor user experience. Users may believe their data was saved when it actually wasn't, leading to lost work and confusion.

== How to fix it

Check the SaveResult using isSuccess() and handle any errors by iterating through getErrors(). This ensures you catch and properly handle any DML failures.

=== Code examples

==== Noncompliant code example

[source,apex,diff-id=1,diff-type=noncompliant]
----
Database.SaveResult sr = Database.insert(recordToInsert);
// No error checking - silent failure possible
----

==== Compliant solution

[source,apex,diff-id=1,diff-type=compliant]
----
Database.SaveResult sr = Database.insert(recordToInsert);
if (!sr.isSuccess()) {
    for (Database.Error err : sr.getErrors()) {
        System.debug('Error: ' + err.getMessage());
        // Handle the error appropriately
    }
}
----

== Resources

=== Documentation

 * Database Methods - https://developer.salesforce.com/docs/atlas.en-us.apexref.meta/apexref/apex_methods_system_database.htm[Official Salesforce documentation for Database class methods including SaveResult handling]

 * DML Operations - https://developer.salesforce.com/docs/atlas.en-us.apexcode.meta/apexcode/apex_dml.htm[Comprehensive guide to DML operations in Apex including error handling best practices]
