<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when "f-strings" are deeply nested.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Through <a href="https://peps.python.org/pep-0701/">PEP 701</a>, Python 3.12 lifts restrictions on how to construct "f-strings".</p>
</div>
<div class="paragraph">
<p>Prior to Python 3.12, it was not possible to reuse string quotes when nesting "f-strings". Therefore, the maximum level of nesting was:</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">f"""{f'''{f'{f"{1+1}"}'}'''}"""</code></pre>
</div>
</div>
<div class="paragraph">
<p>It is now possible to arbitrarily nest "f-strings" by reusing string quotes. The following snippet is therefore valid:</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">f"{f"{f"{f"{f"{f"{1+1}"}"}"}"}"}"</code></pre>
</div>
</div>
<div class="paragraph">
<p>It is, however, not recommended to nest "f-strings" too deeply as this would make the code confusing and hard to maintain.</p>
</div>
<div class="paragraph">
<p>This rule will raise an issue when "f-string" literals are nested 3 times or more.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>To fix this issue, refactor the code to avoid nesting "f-string" literals too deeply. This may be done by introducing new variables to store intermediate results.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">hello = "Hello"
name = "John"
my_string = f"{f"{f"{hello}"},"} {name}!" # Noncompliant: deep nesting of "f-strings" is confusing</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">hello = "Hello"
name = "John"
greeting = f"{f"{hello}"},"
my_string = f"{greeting} {name}!" # Compliant</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>PEP 701 - <a href="https://peps.python.org/pep-0701/">Syntactic formalization of "f-strings"</a></p>
</li>
<li>
<p>Python Release Notes - <a href="https://docs.python.org/3/whatsnew/3.12.html#what-s-new-in-python-3-12">Whatâ€™s New In Python 3.12</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>