== Why is this an issue?

Since {cpp}03, a member function that is contained within a class definition is by definition ``++inline++``, so an using an ``++inline++`` specifier on such functions is redundant.


=== Noncompliant code example

[source,cpp]
----
class Foo {
  inline void method() { // Noncompliant
    // ...
  }
};
----


=== Compliant solution

[source,cpp]
----
class Foo {
  void method() {
    // ...
  }
};
----



ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

include::highlighting.adoc[]

'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
