== Why is this an issue?

Since {cpp}03, a member function that is contained within a class definition is by definition ``++inline++``, so an using an ``++inline++`` specifier on such functions is redundant.


=== Noncompliant code example

[source,cpp]
----
class Foo {
  inline void method() { // Noncompliant
    // ...
  }
};
----


=== Compliant solution

[source,cpp]
----
class Foo {
  void method() {
    // ...
  }
};
----



ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Remove this "inline" specifier; member functions defined within a class definition are by nature "inline".


=== Highlighting

inline specifier


'''
== Comments And Links
(visible only on this page)

=== on 2 Mar 2016, 17:09:29 Ann Campbell wrote:
\[~alban.auzeill] please double-check my changes to the description.

endif::env-github,rspecator-view[]
