<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Theoretically, the use of the <code>static</code> keyword on the size of an array parameter means you can assume that only arrays of at least that size will be passed as arguments to the function. I.e. a function parameter of <code>int my_array[static 10]</code> means that <code>my_array</code> will always be <em>at least</em> 10 elements long. If it is not, the behavior is undefined.</p>
</div>
<div class="paragraph">
<p>In practice, the use of <code>static</code> on the size of an array parameter means the compiler might issue a warning if a noncompliant array is passed to the function - a warning that might or might not be ignored. Therefore, in practice the use of <code>static</code> on an array parameter&#8217;s size merely lends a false sense of security, and <code>static</code> should not be used in this context.</p>
</div>
<div class="paragraph">
<p>Note that for some compiler/processor combinations, more efficient code can be generated when <code>static</code> is used, but these combinations are limited, and the benefit does not outweigh the cost.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">int total (int size, int my_array[static 10]) {...}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">int total (int size, int my_array[10]) {...}</code></pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p>MISRA C:2012, 17.6 - The declaration of an array parameter shall not contain the static keyword between the []</p>
</li>
</ul>
</div>
<hr>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove the "static" keyword on the "xxx" array&#8217;s size</p>
</div>
</div>
</div>
</div>