== How to fix it in cURL

=== Code examples

==== Noncompliant code example

[source,docker,diff-id=1,diff-type=noncompliant]
----
FROM ubuntu:22.04

# Noncompliant
RUN curl --tlsv1.0 -O https://tlsv1-0.example.com/downloads/install.sh
----

==== Compliant solution

[source,docker,diff-id=1,diff-type=compliant]
----
FROM ubuntu:22.04

RUN curl --tlsv1.2 -O https://tlsv1-3.example.com/downloads/install.sh
----

=== How does this work?

include::../../common/fix/fix.adoc[]
