== How to fix it in Wget

=== Code examples

==== Noncompliant code example

[source,docker,diff-id=11,diff-type=noncompliant]
----
FROM ubuntu:22.04

# Noncompliant
RUN wget --secure-protocol TLSv1_1 https://example.com/downloads/install.sh
----

==== Compliant solution

[source,docker,diff-id=11,diff-type=compliant]
----
FROM ubuntu:22.04

RUN wget --secure-protocol TLSv1_2 https://example.com/downloads/install.sh
----

=== How does this work?

include::../../common/fix/fix.adoc[]
