<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>When using a floating-point <code>while</code> loop counter, an accumulation of rounding errors may result in a mismatch between the expected and actual number of iterations.</p>
</div>
<div class="paragraph">
<p>Even if floating-point loop counters appears to behave correctly on one implementation, it may give a different number of iterations on another implementation.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>To fix the issue, use an integer loop counter instead. This avoids the accumulation of rounding errors and ensures that the loop iterates the expected number of times.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-rust" data-lang="rust">fn main() {
    let mut counter = 0.0;
    while counter &lt; 10.0 {
        println!("Counter: {}", counter);
        counter += 1.0;
    }
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-rust" data-lang="rust">fn main() {
    let mut counter = 0;
    while counter &lt; 10 {
        println!("Counter: {}", counter);
        counter += 1;
    }
}</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Clippy Lints - <a href="https://rust-lang.github.io/rust-clippy/master/index.html#while_float" class="bare">https://rust-lang.github.io/rust-clippy/master/index.html#while_float</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>