== Why is this an issue?

There's almost no point in returning results from a trigger. Because there is generally no expectation that triggers will return anything, any such results are likely to be ignored nearly all the time. 


This rule raises an issue when ``++PRINT++`` is used inside a trigger, and when ``++SELECT++``, or ``++FETCH++`` is used inside a trigger without the retrieved data subsequently being used inside the trigger itself.


=== Noncompliant code example

[source,text]
----
ALTER TRIGGER TRG_myTrigger
...
  PRINT @diagnostic_message -- Noncompliant
----

