<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Pointers are a powerful tool, but they can be difficult to use correctly, leading to memory leaks and double deletion. Further, they&#8217;re not usually needed in C&#43;&#43; because the language offers abstractions that handle the more difficult aspects of using pointers for you.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">int *pi = new int;
int *parr = new int[1024];
char *str = new char[1024];</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">int i;
std::array&lt;int, 1024&gt; arr;
std:string str;</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_30_jul_2014_201420_freddy_mallet_wrote">on 30 Jul 2014, 20:14:20 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>I expect this rule to be really controversial and to generate a lot of noisy discussions. But if we decide to go ahead :</p>
</div>
<div class="ulist">
<ul>
<li>
<p>By definition this rule should not be activated by default otherwise at least half of the C&#43;&#43; users evaluating the SonarSource plugin will consider the plugin has valueless because generating a lot of noise.</p>
</li>
<li>
<p>I would decrease the severity to "Major" and classify the rule in the "Maintainability" category.</p>
</li>
</ul>
</div>
</div>
</div>
</div>