Element selections that could be matched anywhere in the document can be very slow. That's why use of the universal selector, `+*+`, should be limited; it explicitly specifies that the match could be anywhere.


== Noncompliant Code Example

----
$( ".buttons > *" );  // Noncompliant; extremely expensive
----


== Compliant Solution

----
$( ".buttons" ).children(); // Compliant
----

