<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p><code>Thread.yield</code> is intended to hint to the processor that the current thread is willing to suspended in favor of another thread. Unfortunately, it doesn&#8217;t have the same results across platforms, thus marring the cross-platform compatibility of any application that uses it.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">public void doSomething() {
  // ...
  Thread.yield();  // Noncompliant
  // ...
}</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove this use of "Thread.yield".</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_11_jun_2015_194314_ann_campbell_wrote">on 11 Jun 2015, 19:43:14 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>CodePro: Disallow Yield Usage</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_16_jun_2015_170858_nicolas_peru_wrote">on 16 Jun 2015, 17:08:58 Nicolas Peru wrote:</h3>
<div class="paragraph">
<p>Looks good</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_16_aug_2016_165257_ann_campbell_wrote">on 16 Aug 2016, 16:52:57 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>Note to self: http://stackoverflow.com/questions/38978112/thread-yield-still-a-cross-platform-minefield</p>
</div>
</div>
</div>
</div>