<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Placing the declarations of objects and functions with external linkage in a header file documents that they are intended to be accessible from other translation units.</p>
</div>
<div class="paragraph">
<p>If external linkage is not required, then the object or function shall either be declared in an unnamed namespace or declared <code>static</code>.</p>
</div>
<div class="paragraph">
<p>This will reduce the visibility of objects and functions, which is considered to be good practice.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">// header.hpp
extern int32_t a1;
extern void f3 ( );

// file1.cpp
#include "header.hpp"
int32_t a1 = 0; // Compliant, implicitly "extern" and declared in a header
int32_t a2 = 0; // Non-compliant, implicitly "extern" but not declared in a header
static int32_t a3 = 0; // Compliant, "static"

namespace
{
 int32_t a4 = 0; // Compliant, in an unnamed namespace
 void f1 () { } // Compliant, in an unnamed namespace
}

static void f2 ( ) { } // Compliant, "static"

void f3 ( ) { } // Compliant, implicitly "extern" and declared in a header
void f4 ( ) { } // Non-compliant, implicitly "extern" but not declared in a header

void main ( ) { } // Compliant by exception</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_exceptions">Exceptions</h3>
<div class="paragraph">
<p>This rule does not apply to main, or to members of unnamed namespaces.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p>MISRA C&#43;&#43;:2008, 3-3-1</p>
</li>
</ul>
</div>
<hr>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_duplicates_s831">duplicates: <a data-rspec-id="S831" class="rspec-auto-link">S831</a></h3>

</div>
</div>
</div>