Ternary operator should not be used to select between two boolean values, or instead of a logical `OR` operation.

== Why is this an issue?

Ternary expressions are often difficult to read, so if a simpler syntax exists, it should be used instead of a ternary expression. This happens when  

 - the expression returns two boolean values
 - the same value is used for both the conditional test and the consequent

== How to fix it

Replace the ternary expression with the condition part only, or convert it to a logical `OR` operation.

=== Code examples

==== Noncompliant code example

[source,javascript,diff-id=1,diff-type=noncompliant]
----
let isGood = value > 0 ? true : false; // Non-compliant, replace with value > 0
let isBad = value > 0 ? false : true; // Non-compliant, replace with !(value > 0)

let a = x ? x : y;  // Non-compliant, replace with x || y
----

==== Compliant solution

[source,javascript,diff-id=1,diff-type=compliant]
----
let result = value > 0 ? 'good': 'bad';
let a = x ? y : z;
----
