== Why is this an issue?

Starting in Java 23, comments beginning with three slashes ``++///++`` are interpreted as JavaDoc comments using Markdown syntax.

In Java 22 and earlier, comments starting with more than 2 slashes were treated as normal comments.
Accidentally writing comments with three or more slashes can lead to unintended JavaDoc being generated, when migrating to Java 23.

=== What is the potential impact?

* accidental JavaDoc generation where normal comments were intended.
* misleading or broken documentation output.
* increased maintenance burden when upgrading to Java 23 or later.

== How to fix it

In versions of Java prior to 23, all comments should always start with exactly 2 slashes, and from Java 23 forward they should not start with more than 3.

=== Code examples

==== Noncompliant code example

The following code will generate misleading Javadoc comments if migrated to Java 23:

[source,java,diff-id=1,diff-type=noncompliant]
----
/// Some comment for the developers
public abstract void foo();
//// public void foo(String s){}
public void foo(){}
----

==== Compliant solution

[source,java,diff-id=1,diff-type=compliant]
----
// Some comment for the developers
public abstract void foo();
// public void foo(String s){}
public void foo(){}
----

== Resources
=== Documentation

* https://openjdk.org/jeps/467[JEP 467: Markdown Documentation Comments]

