include::../description.adoc[]

== Noncompliant Code Example

----
CREATE TABLE notnull(col1 INTEGER NOT NULL)
INSERT notnull VALUES (@value) -- FIXME ensure @value is not null
----

include::../see.adoc[]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
