== Why is this an issue?

Naming the parameters in a function prototype helps identify how they'll be used by the function, thereby acting as a thin layer of documentation for the function.


=== Noncompliant code example

[source,cpp]
----
void divide (int, int);
----


=== Compliant solution

[source,cpp]
----
void divide (int numerator, int denominator);
----


== Resources

* MISRA C:2004, 16.3 - Identifiers shall be given for all of the parameters in a function prototype declaration
* MISRA C:2012, 8.2 - Function types shall be in prototype form with named parameters


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
