== Why is this an issue?

L’instruction INSERT/SELECT correspond à des mises à jour de masse sans possibilité de prendre des commits intermédiaires. Ce qui est très dommageable en terme d’accès concurrents sur des environnements 24/24 7/7. 


=== Noncompliant code example

[source,cobol]
----
EXEC SQL INSERT INTO TESTPROJ (PROJNO, PROJNAME, DEPTNO)
		              SELECT PROJNO, PROJNAME, DEPTNO
		              FROM S1TESTPR
		              WHERE DEPTNO LIKE ‘B%’ END-EXEC.
----


ifdef::env-github,rspecator-view[]
'''
== Comments And Links
(visible only on this page)

=== on 27 May 2014, 09:29:14 Freddy Mallet wrote:
This rule is specific to the BNP context. We should provide an XPath rule to get the same result.

endif::env-github,rspecator-view[]
