L’instruction INSERT/SELECT correspond à des mises à jour de masse sans possibilité de prendre des commits intermédiaires. Ce qui est très dommageable en terme d’accès concurrents sur des environnements 24/24 7/7. 


== Noncompliant Code Example

[source,cobol]
----
EXEC SQL INSERT INTO TESTPROJ (PROJNO, PROJNAME, DEPTNO)
		              SELECT PROJNO, PROJNAME, DEPTNO
		              FROM S1TESTPR
		              WHERE DEPTNO LIKE ‘B%’ END-EXEC.
----


ifdef::env-github,rspecator-view[]
'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
