== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,cpp]
----
int i;
for (i = 0; i < 10; i++) {
  if (i == 5) {
    continue;  /* Noncompliant */
  }
  printf("i = %d\n", i);
}
----


=== Compliant solution

[source,cpp]
----
int i;
for (i = 0; i < 10; i++) {
  if (i != 5) {
    printf("i = %d\n", i);
  }
}
----


== Resources

* MISRA C:2004, 14.5 - The continue statement shall not be used.

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
