== Why is this an issue?

Restricting the number of ``++CONTINUE++``, ``++EXIT++`` and ``++CHECK++`` statements in a loop is done in the interest of good structured programming.


One ``++CONTINUE++``, ``++EXIT++`` and ``++CHECK++`` statement is acceptable in a loop, since it facilitates optimal coding. If there is more than one, the code should be refactored to increase readability.

=== Noncompliant code example

[source,abap]
----
DO counter TIMES.
  IF sy-index = 2.
    CONTINUE.
  ENDIF.
  IF sy-index = 10.
    EXIT.
  ENDIF.
  WRITE sy-index.
ENDDO.
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../highlighting.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
