== Why is this an issue?

:link-with-uscores1: https://docs.python.org/3/reference/datamodel.html#object.__setitem__

Storing a value inside a collection at a given key or index and then unconditionally overwriting it without reading the initial value is a case of "dead store".

At best it is a useless operation that will confuse the reader, most often it is an error and not what the developer intended to do.

This rule raises an issue when the {link-with-uscores1}[``++__setitem__++``] method of the same object is called multiple times with the same index, slice or key without any other action done between the calls.

=== Noncompliant code example

[source,python,diff-id=1,diff-type=noncompliant]
----
def swap(mylist, index1, index2):
    tmp = mylist[index2]
    mylist[index2] = mylist[index1]
    mylist[index2] = tmp  # Noncompliant

list2 = [0,1,2,3,4,5,6,7,8,9]
list2[3:5] = [42,42]
list2[3:5] = [42,42]  # Noncompliant

mymap = {'a': {}}
mymap['a']['b'] = 42
mymap['a']['b'] = 42  # Noncompliant
----

=== Compliant solution

[source,python,diff-id=1,diff-type=compliant]
----
def swap(mylist, index1, index2):
    tmp = mylist[index2]
    mylist[index2] = mylist[index1]
    mylist[index1] = tmp

list2 = [0,1,2,3,4,5,6,7,8,9]
list2[3:5] = [42,42]

mymap = {'a': {}}
mymap['a']['b'] = 42
mymap['a']['c'] = 42
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../highlighting.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
