== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,python,diff-id=1,diff-type=noncompliant]
----
def swap(mylist, index1, index2):
    tmp = mylist[index2]
    mylist[index2] = mylist[index1]
    mylist[index2] = tmp  # Noncompliant

list2 = [0,1,2,3,4,5,6,7,8,9]
list2[3:5] = [42,42]
list2[3:5] = [42,42]  # Noncompliant

mymap = {'a': {}}
mymap['a']['b'] = 42
mymap['a']['b'] = 42  # Noncompliant
----

=== Compliant solution

[source,python,diff-id=1,diff-type=compliant]
----
def swap(mylist, index1, index2):
    tmp = mylist[index2]
    mylist[index2] = mylist[index1]
    mylist[index1] = tmp

list2 = [0,1,2,3,4,5,6,7,8,9]
list2[3:5] = [42,42]

mymap = {'a': {}}
mymap['a']['b'] = 42
mymap['a']['c'] = 42
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../highlighting.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
