<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>When multiple, adjacent <code>try</code> statements have duplicate <code>catch</code> and/or <code>finally</code> blocks, they should be merged to consolidate the <code>catch/finally</code> logic for cleaner, more readable code. Note that this applies even when there is intervening code outside any <code>try</code> block.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-text" data-lang="text">try
{
  DoTheFirstThing(a, b);
}
catch (InvalidOperationException ex)
{
  HandleException(ex);
}

DoSomeOtherStuff();

try  // Noncompliant; catch is identical to previous
{
  DoTheSecondThing();
}
catch (InvalidOperationException ex)
{
  HandleException(ex);
}

try  // Compliant; catch handles exception differently
{
  DoTheThirdThing(a);
}
catch (InvalidOperationException ex)
{
  LogAndDie(ex);
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-text" data-lang="text">try
{
  DoTheFirstThing(a, b);
  DoSomeOtherStuff();
  DoTheSecondThing();
}
catch (InvalidOperationException ex)
{
  HandleException(ex);
}

try  // Compliant; catch handles exception differently
{
  DoTheThirdThing(a);
}
catch (InvalidOperationException ex)
{
  LogAndDie(ex);
}</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Combine this "try" with the one starting on line nnn.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_18_dec_2020_100932_andrei_epure_wrote">on 18 Dec 2020, 10:09:32 Andrei Epure wrote:</h3>
<div class="paragraph">
<p>I opened this sub-task just to remove the rule from SonarWay for C#, as it has some controversies. See the discussion on <a href="https://github.com/SonarSource/sonar-dotnet/issues/2490">#2490</a>.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_11_may_2015_144508_dinesh_bolkensteyn_wrote">on 11 May 2015, 14:45:08 Dinesh Bolkensteyn wrote:</h3>
<div class="paragraph">
<p>LGTM , but I&#8217;d increase the SQALE cost a bit (to 20 or 30 minutes), because the refactoring is not functionally equivalent (in the first example, of <code>DoTheFirstThing()</code> fails, then the 2nd function is executed anyway - whereas after the refactoring the execution stops on the first error).</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_12_may_2015_111451_ann_campbell_wrote">on 12 May 2015, 11:14:51 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>remediation cost updated</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_12_may_2015_113212_dinesh_bolkensteyn_wrote">on 12 May 2015, 11:32:12 Dinesh Bolkensteyn wrote:</h3>
<div class="paragraph">
<p>LGTM</p>
</div>
</div>
</div>
</div>