== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,flex]
----
public class MyClass {
  private var foo:int = 4;                       //foo is unused

  public function compute(a:int):int{
    return a * 4;
  }
}
----

=== Compliant solution

[source,flex]
----
public class MyClass {

  public function compute(a:int):int{
    return a * 4;
  }
}
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
