include::../description.adoc[]

== Noncompliant Code Example

----
public class MyClass {
  private var foo:int = 4;                       //foo is unused

  public function compute(a:int):int{
    return a * 4;
  }
}
----

== Compliant Solution

----
public class MyClass {

  public function compute(a:int):int{
    return a * 4;
  }
}
----

ifdef::rspecator-view[]
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::rspecator-view[]
