<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when the <code>__all__</code> property of a module contains objects that aren&#8217;t strings.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>The <code>__all__</code> property of a module is used to define the list of names that will be imported when performing a wildcard import of this module, i.e. when <code>from mymodule import *</code> is used.</p>
</div>
<div class="paragraph">
<p>In the following example:</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python"># mymodule.py
def foo(): ...
def bar(): ...
__all__ = ["foo"]</code></pre>
</div>
</div>
<div class="paragraph">
<p>Executing <code>from mymodule import *</code> from a different module will only import <code>foo</code>.</p>
</div>
<div class="paragraph">
<p>This list can only contain strings. If something other than a string is listed, a <code>TypeError</code> will be raised when trying to perform a wildcard import of the module.</p>
</div>
<div class="paragraph">
<p>To fix this issue, make sure that all properties listed in <code>__all__</code> are strings.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">class MyClass:
    pass

__all__ = [
    MyClass  # Noncompliant: wildcard import will raise a TypeError
]</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">class MyClass:
    pass

__all__ = [
    "MyClass"
]</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p><a href="https://docs.python.org/3/tutorial/modules.html#importing-from-a-package">Python documentation  - Importing * From a Package</a></p>
</li>
</ul>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Replace this symbol with a string; "<em>all</em>" can only contain strings.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>Primary: the wrong symbol in "<em>all</em>"</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_is_related_to_s5807">is related to: <a data-rspec-id="S5807" class="rspec-auto-link">S5807</a></h3>

</div>
</div>
</div>