<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when test code directly accesses or stubs private methods using techniques like <code>send()</code>, <code>instance_eval</code>, or method mocking.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Private methods are implementation details that should remain hidden from external code, including tests. When you test private methods directly, you create several problems:</p>
</div>
<div class="paragraph">
<p><strong>Brittle tests</strong>: Tests that depend on private method names and signatures break whenever you refactor the internal implementation, even when the public behavior remains unchanged.</p>
</div>
<div class="paragraph">
<p><strong>Violated encapsulation</strong>: Private methods exist to hide complexity and implementation details. Testing them directly breaks the encapsulation principle and exposes internal workings that should remain private.</p>
</div>
<div class="paragraph">
<p><strong>Harder maintenance</strong>: When private methods are tested directly, changing internal implementation becomes much more difficult because you need to update both the code and the tests, even for internal refactoring.</p>
</div>
<div class="paragraph">
<p><strong>Poor test design</strong>: Direct testing of private methods often indicates that your tests are focusing on "how" the code works rather than "what" it does. Good tests should verify behavior, not implementation.</p>
</div>
<div class="paragraph">
<p>The correct approach is to test private methods indirectly through the public methods that use them. This way, your tests verify that the overall behavior is correct without being tied to specific implementation details.</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="paragraph">
<p>Testing private methods directly makes your test suite fragile and harder to maintain. When you refactor code, you&#8217;ll need to update tests even when the public behavior hasn&#8217;t changed. This slows down development and makes refactoring risky, ultimately leading to code that becomes harder to evolve over time.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it_in_rspec">How to fix it in RSpec</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Instead of testing private methods directly, test them through the public methods that use them. Focus on verifying the overall behavior rather than internal implementation details.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-ruby" data-lang="ruby">class Calculator
  def add_and_multiply(a, b, multiplier)
    result = add_numbers(a, b)
    result * multiplier
  end

  private

  def add_numbers(a, b)
    a + b
  end
end

RSpec.describe Calculator do
  it "calls private method" do
    calculator = Calculator.new
    expect(calculator).to receive(:add_numbers).and_return(5) # Noncompliant
    calculator.add_and_multiply(2, 3, 4)
  end

  it "tests private method behavior" do
    calculator = Calculator.new
    result = calculator.send(:add_numbers, 2, 3) # Noncompliant
    expect(result).to eq(5)
  end
end</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-ruby" data-lang="ruby">class Calculator
  def add_and_multiply(a, b, multiplier)
    result = add_numbers(a, b)
    result * multiplier
  end

  private

  def add_numbers(a, b)
    a + b
  end
end

RSpec.describe Calculator do
  it "adds and multiplies correctly" do
    calculator = Calculator.new
    result = calculator.add_and_multiply(2, 3, 4)
    expect(result).to eq(20)  # Testing behavior, not implementation
  end
end</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it_in_rails">How to fix it in Rails</h2>
<div class="sectionbody">
<div class="paragraph">
<p>In controller tests, avoid accessing private methods directly. Instead, test the controller actions that use those private methods, verifying the overall behavior and response.</p>
</div>
<div class="sect2">
<h3 id="_code_examples_2">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example_2">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-ruby" data-lang="ruby">describe AccountController do
  let(:controller) { AccountController.new }

  it "tests private method directly" do
    result = controller.send(:current_account) # Noncompliant
    expect(result).to eq(expected_account)

    account = controller.instance_eval { current_account } # Noncompliant
    expect(account).to be_present
  end
end</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution_2">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-ruby" data-lang="ruby">describe AccountController do
  let(:user) { create(:user) }
  let(:controller) { AccountController.new }

  before { allow(controller).to receive(:current_user).and_return(user) }

  describe "#index" do
    it "renders successfully" do
      get :index
      expect(response).to be_successful
    end
  end
end</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>RSpec Best Practices - <a href="https://rspec.info/documentation/">Official RSpec documentation with testing best practices</a></p>
</li>
<li>
<p>Ruby Private Methods - <a href="https://ruby-doc.org/core/doc/syntax/methods_rdoc.html#label-Method+Visibility">Ruby documentation on method visibility and private methods</a></p>
</li>
</ul>
</div>
</div>
<div class="sect2">
<h3 id="_standards">Standards</h3>
<div class="ulist">
<ul>
<li>
<p>SOLID Principles - Encapsulation - <a href="https://en.wikipedia.org/wiki/Encapsulation_(computer_programming)">Object-oriented programming principle that bundles data and methods together while restricting access to internal details</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>