== Why is this an issue?

There are several reasons for a function not to have a function body:

* It is an unintentional omission, and should be fixed to prevent an unexpected behavior in production.
* It is not yet, or never will be, supported. In this case an exception should be thrown in languages where that mechanism is available.
* The method is an intentionally-blank override. In this case a nested comment should explain the reason for the blank override.

=== Noncompliant code example

[source,javascript]
----
function foo() {
}

var foo = () => {};
----

=== Compliant solution

[source,javascript]
----
function foo() {
    // This is intentional
}

var foo = () => {
    do_something();
};
----

=== Exceptions

* This rule does not apply to function expressions and arrow functions as they can denote default values.

[source,javascript]
----
static defaultProps = {
  listStyle: () => {}
};
----

* The rule allows for empty functions with a name starting with the prefix ``++on++`` like ``++onClick++``.

[source,javascript]
----
function onClick() {
}
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
