There are several reasons for a function not to have a function body:

* It is an unintentional omission, and should be fixed to prevent an unexpected behavior in production.
* It is not yet, or never will be, supported. In this case an exception should be thrown in languages where that mechanism is available.
* The method is an intentionally-blank override. In this case a nested comment should explain the reason for the blank override.

== Noncompliant Code Example

----
function foo() {
}

var foo = () => {};
----

== Compliant Solution

----
function foo() {
    // This is intentional
}

var foo = () => {
    do_something();
};
----

ifdef::env-github,rspecator-view[]
'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
