include::../description.adoc[]

== How to fix it in Gradle

=== Code examples

==== Noncompliant code example

[source,xml,diff-id=1,diff-type=noncompliant]
----
<verification-metadata 
    xmlns="https://schema.gradle.org/dependency-verification" 
    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" 
    xsi:schemaLocation="https://schema.gradle.org/dependency-verification https://schema.gradle.org/dependency-verification/dependency-verification-1.3.xsd">
   <configuration>
      <verify-metadata>true</verify-metadata>
      <verify-signatures>false</verify-signatures>
      <keyring-format>armored</keyring-format>
      <trusted-keys>
         <trusted-key id="FD8190C7D72E7DCD42582B1042677B9FC1DC2161" group="com.example" name="adependency"/>
      </trusted-keys>
   </configuration>
</verification-metadata>
----

This configuration will verify the integrity of the dependency but not its authenticity (see "How does this work?" section).

==== Compliant solution

[source,xml,diff-id=1,diff-type=compliant]
----
<verification-metadata 
    xmlns="https://schema.gradle.org/dependency-verification" 
    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" 
    xsi:schemaLocation="https://schema.gradle.org/dependency-verification https://schema.gradle.org/dependency-verification/dependency-verification-1.3.xsd">
   <configuration>
      <verify-metadata>true</verify-metadata>
      <verify-signatures>true</verify-signatures>
      <keyring-format>armored</keyring-format>
      <trusted-keys>
         <trusted-key id="FD8190C7D72E7DCD42582B1042677B9FC1DC2161" group="com.example" name="adependency"/>
      </trusted-keys>
   </configuration>
</verification-metadata>
----

Signature verification will ensure both integrity and authenticity of the dependency.

== How does this work?

include::../common/fix/gradle-verification.adoc[]

== Resources

include::../common/resources/documentation.adoc[]

include::../common/resources/standards.adoc[]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

* Change this value to enable verification of dependencies signatures

=== Highlighting

* The value inbetween `<verify-signatures>` tags

endif::env-github,rspecator-view[]
