<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Stateless stream gatherers, which process each element of the stream independently, don&#8217;t need an initializer function.
That&#8217;s because the role of the initializer function is to set up the gatherer&#8217;s private state object.</p>
</div>
<div class="paragraph">
<p>The gatherer interface provides factory methods that create gatherers without taking an initializer as an argument:</p>
</div>
<div class="ulist">
<ul>
<li>
<p><code>ofSequential(Gatherer.Integrator&lt;Void,T,R&gt; integrator)</code></p>
</li>
<li>
<p><code>ofSequential(Gatherer.Integrator&lt;Void,T,R&gt; integrator, BiConsumer&lt;Void, Gatherer.Downstream&lt;? super R&gt;&gt; finisher)</code></p>
</li>
</ul>
</div>
<div class="paragraph">
<p>Using these specific factory methods to create stateless gatherers is preferable because it clearly communicates their stateless nature.</p>
</div>
<div class="paragraph">
<p>This rule reports the use of <code>ofSequential(Gatherer.Integrator&lt;Void,T,R&gt; integrator, BiConsumer&lt;Void, Gatherer.Downstream&lt;? super R&gt;&gt; finisher)</code> when it returns a stateless gatherer.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">private static Gatherer inRange(int start, int end) {
    return Gatherer.&lt;Integer, Void, Integer&gt;ofSequential(
      () -&gt; null, // Noncompliant: unnecessary initializer
      (_, element, downstream) -&gt; {
        if (element &gt;= start &amp;&amp; element &lt;= end)
          return downstream.push(element - start);
        return !downstream.isRejecting();
      },
      (_, downstream) -&gt; downstream.push(-1)
    );
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">private static Gatherer inRange(int start, int end) {
    return Gatherer.&lt;Integer, Integer&gt;ofSequential(
      (_, element, downstream) -&gt; {
        if (element &gt;= start &amp;&amp; element &lt;= end)
          return downstream.push(element - start);
        return !downstream.isRejecting();
      },
      (_, downstream) -&gt; downstream.push(-1)
    );
}</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>