<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>JavaScript regular expressions provide  Unicode character classes and Unicode property escapes for matching characters based on their Unicode values and Unicode properties respectively. When using Unicode property escapes like <code>\p{Alpha}</code> without the <code>u</code> flag, the regular expression will not match alphabetic characters but rather the <code>'\p{Alpha}'</code> string literal, which is likely a mistake.</p>
</div>
<div class="paragraph">
<p>This rules raises an issue when Unicode character classes and Unicode property escapes are used without the <code>u</code> flag.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-javascript" data-lang="javascript">/\u{1234}/
/\p{Alpha}/</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-javascript" data-lang="javascript">/\u{1234}/u
/\p{Alpha}/u</code></pre>
</div>
</div>
</div>
</div>
</div>