<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>When using POSIX classes like <code>\p{Alpha}</code> without the <code>(?U)</code> to include Unicode characters or when using hard-coded character classes like <code>"[a-zA-Z]"</code>, letters outside of the ASCII range, such as umlauts, accented letters or letter from non-Latin languages, won&#8217;t be matched. This may cause code to incorrectly handle input containing such letters.</p>
</div>
<div class="paragraph">
<p>To correctly handle non-ASCII input, it is recommended to use Unicode classes like <code>\p{IsAlphabetic}</code>. When using POSIX classes, Unicode support should be enabled by using <code>(?U)</code> inside the regex.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-kotlin" data-lang="kotlin">Regex("[a-zA-Z]")
Regex("\\p{Alpha}")
Regex("""\p{Alpha}""")</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-kotlin" data-lang="kotlin">Regex("""\p{IsAlphabetic}""") // matches all letters from all languages
Regex("""\p{IsLatin}""") // matches latin letters, including umlauts and other non-ASCII variations
Regex("""(?U)\p{Alpha}""")
Regex("(?U)\\p{Alpha}")</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="ulist">
<ul>
<li>
<p>when using plain character classes: Replace this character range with a Unicode-aware character class.</p>
</li>
<li>
<p>when using POSIX classes: Use a Unicode-aware alternative or "(?U)".</p>
</li>
</ul>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>The character class</p>
</div>
</div>
</div>
</div>