<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>When using metacharacters like <code>\w</code> without the Unicode flag or when using hard-coded character classes like <code>[a-zA-Z]</code>,
letters outside of the ASCII range, such as umlauts, accented letters, or letters from non-Latin languages, won’t be matched.
This may cause code to incorrectly handle input containing such letters.</p>
</div>
<div class="paragraph">
<p>To correctly handle non-ASCII input, it is recommended to use the Unicode flag <code>\u</code> or Unicode character properties like <code>\p{L}</code>.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-php" data-lang="php">preg_match("/[a-zA-Z]/", "ö"); // returns 0
preg_match("/\w/", "ö"); // returns 0</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-php" data-lang="php">preg_match("/\w/u", "ö"); // returns 1
preg_match("/\p{L}/", "ö"); // return 1</code></pre>
</div>
</div>
</div>
</div>
</div>