This rule raises an issue when a special method is defined with an unexpected number of parameters.

== Why is this an issue?

Python allows developers to customize how code is interpreted by defining special methods (also called magic methods). For example, it is possible to define the objects own truthiness or falsiness by iverriding ``__bool__`` method. It is invoked when the built-in ``bool()`` function is called on an object. The ``bool()`` function returns ``True`` or ``False`` based on the truth value of the object.

Each special method expects a specific return type. The Python interpreter will call these methods. Calls to a special method will throw a ``++TypeError++`` if it there is returning value with incorrect type.

== How to fix it

Make sure to return value with correct type defined in the Python documentation for each special methods.

=== Code examples

==== Noncompliant code example

[source,python,diff-id=1,diff-type=noncompliant]
----
class MyClass:
    def __init__(self, value: int) -> None:
        self.value = value

    def __bool__(self) -> bool:
        return self.value # Noncompliant: incorrect return type. Boolean value expected.

obj1 = MyClass(5)
print(bool(obj1)) # TypeError: __bool__ should return bool, returned int
----

==== Compliant solution 

[source,python,diff-id=1,diff-type=compliant]
----
class MyClass:
    def __init__(self, value: int) -> None:
        self.value = value

    def __bool__(self) -> bool:
        return self.value > 0

obj1 = MyClass(5)
print(bool(obj1))
----

== Resources

=== Documentation

* https://docs.python.org/3/reference/datamodel.html#special-method-names[Special method names] - Python special methods 
* https://docs.python.org/3/library/copy.html[Copy module] - Documentation of ``++__copy__++`` and ``++__deepcopy__++``


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

* Make sure this method returns a value of type XXX.


=== Highlighting

Primary: The method name.

Secondary: The return statement if exists.


'''
== Comments And Links
(visible only on this page)


endif::env-github,rspecator-view[]
