include::../description.adoc[]

== Noncompliant Code Example

----
// in web.xml
<web-app  ...>
  <!--  ...  -->
  <login-config>
    <auth-method>BASIC</auth-method>
  </login-config>
</web-app>
----

== Exceptions

The rule will not raise any issue if HTTPS is enabled, on any URL-pattern.

----
<web-app  ...>
  <!--  ...  -->
  <security-constraint>
    <web-resource-collection>
      <web-resource-name>HTTPS enabled</web-resource-name>
      <url-pattern>/*</url-pattern>
    </web-resource-collection>
    <user-data-constraint>
      <transport-guarantee>CONFIDENTIAL</transport-guarantee>
    </user-data-constraint>
  </security-constraint>
</web-app>
----

include::../see.adoc[]

ifdef::env-github,rspecator-view[]
'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
