<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p><code>forEach</code> function in Dart allows to iterate over collection and do some actions with their elements. This is convenient and more concise if you can pass the reference to the action directly through argument. For example this code:</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-dart" data-lang="dart">void actOnList(List&lt;String&gt; list, Function(String) action) {
  list.forEach(action);
}</code></pre>
</div>
</div>
<div class="paragraph">
<p>looks more concise than this code:</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-dart" data-lang="dart">void actOnList(List&lt;String&gt; list, Function(String) action) {
  for(String s in list) {
    action(s);
  }
}</code></pre>
</div>
</div>
<div class="paragraph">
<p>However, if you want to have a function literal as <code>forEach</code> argument, you might face some limitations:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>If you place <code>return</code> in the function literal, it will only return the value for that iteration, while you might want to return from the function.</p>
</li>
<li>
<p>The bode of the function literal can&#8217;t contain <code>await</code>, while ordinary <code>for</code> loop can contain it.</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>As a consequence, if you want to use <code>forEach</code> with a function literal as an argument, it&#8217;s better to use a simple <code>for</code> loop instead.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Replace the <code>forEach</code> with a <code>for</code> loop or use reference to a function instead of function literal.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-dart" data-lang="dart">void f1(List&lt;String&gt; list) {
  list.forEach((s) =&gt; print(s));
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-dart" data-lang="dart">void f1(List&lt;String&gt; list) {
  list.forEach(print);
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_noncompliant_code_example_2">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-dart" data-lang="dart">void f2(List&lt;String&gt; list) {
  list.forEach((s) {
    if (s.length &gt; 5) {
      print("foo");
    } else {
      print("bar");
    }
  });
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution_2">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-dart" data-lang="dart">void f2(List&lt;String&gt; list) {
  for(String s in list) {
    if (s.length &gt; 5) {
      print("foo");
    } else {
      print("bar");
    }
  }
}</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Dart Docs - <a href="https://dart.dev/tools/linter-rules/avoid_function_literals_in_foreach_calls">Dart Linter rule - avoid_function_literals_in_foreach_calls</a></p>
</li>
</ul>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="ulist">
<ul>
<li>
<p>Function literals shouldn&#8217;t be passed to 'forEach'.</p>
</li>
</ul>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p><code>forEach</code> function</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
</div>
</div>