== Why is this an issue?

include::../rationale.adoc[]

include::../impact.adoc[]

== How to fix it

=== Code examples

include::../common/fix/code-rationale.adoc[]

==== Noncompliant code example

[source,go,diff-id=1,diff-type=noncompliant]
----
func getName(db *sql.DB, id string) (string, error) {
    var name string
    row := db.QueryRow("SELECT name FROM users WHERE id = " + id) // Noncompliant

    if err := row.Scan(&name); err != nil {
        if err == sql.ErrNoRows {
            return name, fmt.Errorf("No name found for id %s", id)
        }
    }

    return name, nil
}
----

==== Compliant solution

[source,go,diff-id=1,diff-type=compliant]
----
func getName(db *sql.DB, id string) (string, error) {
    var name string
    row := db.QueryRow("SELECT name FROM users WHERE id = ?", id)

    if err := row.Scan(&name); err != nil {
        if err == sql.ErrNoRows {
            return name, fmt.Errorf("No name found for id %s", id)
        }
    }

    return name, nil
}
----

=== How does this work?

include::../common/fix/how.adoc[]

== Resources

include::../common/resources/standards.adoc[]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../highlighting.adoc[]

endif::env-github,rspecator-view[]
