<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Marking a variable with <code>IBOutlet</code> allows it to be connected with a Storyboard component through the Interface Builder. Allowing such a variable to be accessed outside the class, may result in other classes making assignments that override the automatic dependency injection from the Storyboard itself.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-swift" data-lang="swift">  @IBOutlet var label: UILabel!  // Noncompliant</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-swift" data-lang="swift">  @IBOutlet private var label: UILabel!</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Make this "IBOutlet" variable private.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>variable declaration</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_5_sep_2017_185631_ann_campbell_wrote">on 5 Sep 2017, 18:56:31 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>I&#8217;ve made some small edits [~carlo.bottiglieri], but not substantially changed your meaning, I think.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_6_sep_2017_101752_carlo_bottiglieri_wrote">on 6 Sep 2017, 10:17:52 Carlo Bottiglieri wrote:</h3>
<div class="paragraph">
<p>Thanks a lot [~ann.campbell.2]</p>
</div>
</div>
</div>
</div>