== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,swift]
----
if condition
{
  doSomething()
}
----

=== Compliant solution

[source,swift]
----
if condition {
  doSomething()
}
----

=== Exceptions

Closure and inlined blocks (left and right curly braces on the same line) are ignored by this rule. 


[source,swift]
----
if condition {doSomething()} // Compliant
reversed = sorted(
    names, 
    { (s1: String, s2: String) -> Bool in  // Compliant
        return s1 > s2
    })
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
