include::../description.adoc[]

== Noncompliant Code Example

----
if condition
{
  doSomething()
}
----

== Compliant Solution

----
if condition {
  doSomething()
}
----

== Exceptions

Closure and inlined blocks (left and right curly braces on the same line) are ignored by this rule. 


----
if condition {doSomething()} // Compliant
reversed = sorted(
    names, 
    { (s1: String, s2: String) -> Bool in  // Compliant
        return s1 > s2
    })
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
