== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,go]
----
if len(myArr) >= 0 { ... }

if len(myArr) < 0 { ... }

var result = len(myArr) >= 0
----

=== Compliant solution

[source,go]
----
if len(myArr) > 0 { ... }

if len(myArr) < 42 { ... }
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../highlighting.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
