<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Non-interactive DOM elements are those that do not have built-in interactivity or do not require user interaction. Examples include <code>&lt;div&gt;</code>, <code>&lt;p&gt;</code>, <code>&lt;img&gt;</code>, <code>&lt;h1&gt;</code> to <code>&lt;h6&gt;</code>, and <code>&lt;ul&gt;</code>, among others. These elements are typically used to structure content and layout but do not have any inherent interactive behavior.</p>
</div>
<div class="paragraph">
<p>Interactive ARIA roles, on the other hand, are used to make elements interactive and accessible. These roles include <code>button</code>, <code>link</code>, <code>checkbox</code>, <code>menuitem</code>, <code>tab</code>, and others. They are used to communicate the type of interaction that users can expect from an element.</p>
</div>
<div class="paragraph">
<p>Non-interactive DOM elements should not use interactive ARIA roles because it can confuse assistive technologies and their users. For example, if a <code>&lt;div&gt;</code> (a non-interactive element) is given an interactive role like "button", assistive technologies like screen readers will announce it as a button. However, since <code>&lt;div&gt;</code> doesn&#8217;t have the inherent behavior of a button, it can confuse users who expect it to behave like a button when interacted with.</p>
</div>
<div class="paragraph">
<p>This can lead to a poor user experience and can make the website less accessible for users relying on assistive technologies. Therefore, it&#8217;s important to ensure that non-interactive DOM elements are not given interactive ARIA roles.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Ensure that non-interactive DOM elements do not use interactive ARIA roles.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-html" data-lang="html">&lt;li role="button"&gt;Foo&lt;/li&gt; &lt;!--Noncompliant; "li" isn't interactive, but "button" is--&gt;</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-html" data-lang="html">&lt;li role="listitem"&gt;Foo&lt;/li&gt;</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>WCAG - <a href="https://www.w3.org/WAI/WCAG21/Understanding/name-role-value">Name, Role, Value</a></p>
</li>
<li>
<p>WCAG - <a href="https://www.w3.org/TR/wai-aria-1.1/#usage_intro">WAI-ARIA Roles</a></p>
</li>
<li>
<p>MDN web docs - <a href="https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles">WAI-ARIA Roles</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>