include::../description.adoc[]

== Noncompliant Code Example

----
If a And b Then
  If c And d Then
    doSomething()
  End If
End If
----

== Compliant Solution

----
If a And b And c And d Then
  doSomething()
End If
----

ifdef::env-github,rspecator-view[]
'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
