== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,vb6]
----
If a And b Then
  If c And d Then
    doSomething()
  End If
End If
----

=== Compliant solution

[source,vb6]
----
If a And b And c And d Then
  doSomething()
End If
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../highlighting.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
