include::../description.adoc[]

== Noncompliant Code Example

[source,vbnet]
----
Dim emailPattern As New Regex(".+@.+", RegexOptions.None)
Dim isNumber as Boolean = Regex.IsMatch(input, "[0-9]+")
Dim isLetterA as Boolean = Regex.IsMatch(input, "(a+)+");
----

== Compliant Solution

[source,vbnet]
----
Dim emailPattern As New Regex(".+@.+", RegexOptions.None, TimeSpan.FromMilliseconds(1))
Dim isNumber as Boolean = Regex.IsMatch(input, "[0-9]+", RegexOptions.None, TimeSpan.FromMilliseconds(1))
Dim isLetterA As Boolean = Regex.IsMatch(input, "(a+)+", RegexOptions.NonBacktracking) '.Net 7 And above
----

== See

include::../see.adoc[]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../highlighting.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]