<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This is an issue when pagination methods like <code>paginate</code> are called but their results are not assigned, or when paginated collections are subsequently overwritten with regular arrays or ActiveRecord relations.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Pagination libraries like will_paginate and kaminari return special collection objects that extend regular arrays with pagination-specific methods such as <code>total_pages</code>, <code>current_page</code>, <code>per_page</code>, and <code>offset</code>. These methods are essential for rendering pagination controls in views.</p>
</div>
<div class="paragraph">
<p>When you call pagination methods without assigning their results, or when you overwrite a paginated collection with a regular array or ActiveRecord relation, you lose access to these pagination methods. This leads to runtime errors when your views try to access pagination information.</p>
</div>
<div class="paragraph">
<p>For example, calling <code>@locations.paginate(page: 1, per_page: 2)</code> without assignment means the paginated result is discarded, and <code>@locations</code> remains a regular array. Similarly, operations like <code>map</code>, <code>select</code>, or additional ActiveRecord queries after pagination will return regular collections, replacing the paginated one.</p>
</div>
<div class="paragraph">
<p>This issue is particularly common when developers chain operations or perform additional filtering after pagination, not realizing that these operations destroy the pagination wrapper.</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="paragraph">
<p>When paginated collections are overwritten, views that depend on pagination methods will fail with <code>NoMethodError</code> exceptions. This breaks the user interface and can cause application crashes in production.</p>
</div>
<div class="paragraph">
<p>Additionally, losing pagination functionality means users cannot navigate through large datasets effectively, leading to poor user experience and potential performance issues if all records are loaded at once.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Always assign the result of pagination methods to preserve the paginated collection object.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-ruby" data-lang="ruby">@locations = Location.find_all_by_city("springfield")
@locations.paginate(page: 1, per_page: 2)  # Noncompliant</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-ruby" data-lang="ruby">@locations = Location.where(city: "springfield")
@locations = @locations.paginate(page: params[:page], per_page: 10)</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>will_paginate documentation - <a href="https://github.com/mislav/will_paginate">Official documentation for the will_paginate gem</a></p>
</li>
<li>
<p>Kaminari documentation - <a href="https://github.com/kaminari/kaminari">Official documentation for the Kaminari pagination gem</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>