== How to fix it in fasthttp

=== Code examples

include::../../common/fix/code-rationale.adoc[]

==== Noncompliant code example

[source,go,diff-id=111,diff-type=noncompliant]
----
import (
	"fmt"

	"github.com/valyala/fasthttp"
)

func Example(ctx *fasthttp.RequestCtx) {
	urlBytes := ctx.FormValue("url")
	url := string(urlBytes)

	statusCode, body, err := fasthttp.Get(nil, url)
	if err != nil {
		fmt.Fprintf(ctx, "Error when making request: %s\n", err.Error())
		ctx.SetStatusCode(fasthttp.StatusInternalServerError)
		return
	}

	if statusCode != fasthttp.StatusOK {
		fmt.Fprintf(ctx, "Request failed with status code: %d\n", statusCode)
		ctx.SetStatusCode(statusCode)
		return
	}

	fmt.Fprintf(ctx, "Response: %s", string(body))
	ctx.SetStatusCode(fasthttp.StatusOK)
}
----

==== Compliant solution

[source,go,diff-id=111,diff-type=compliant]
----
import (
	"fmt"

	"github.com/valyala/fasthttp"
)

func Example(ctx *fasthttp.RequestCtx) {
	urlBytes := ctx.FormValue("url")
	url := string(urlBytes)

	allowedURLs := map[string]bool{
		"https://example.com":  true,
		"https://example.com/": true,
	}
	if _, ok := allowedURLs[url]; !ok {
		fmt.Fprint(ctx, "Specified host is not allowed")
		ctx.SetStatusCode(fasthttp.StatusBadRequest)
		return
	}

	statusCode, body, err := fasthttp.Get(nil, url)
	if err != nil {
		fmt.Fprintf(ctx, "Error when making request: %s\n", err.Error())
		ctx.SetStatusCode(fasthttp.StatusInternalServerError)
		return
	}

	if statusCode != fasthttp.StatusOK {
		fmt.Fprintf(ctx, "Request failed with status code: %d\n", statusCode)
		ctx.SetStatusCode(statusCode)
		return
	}

	fmt.Fprintf(ctx, "Response: %s", string(body))
	ctx.SetStatusCode(fasthttp.StatusOK)
}
----

=== How does this work?

include::../../common/fix/pre-approved-list.adoc[]

The compliant code example uses such an approach.

=== Pitfalls

include::../../common/pitfalls/starts-with.adoc[]
