== How to fix it in Fiber

=== Code examples

include::../../common/fix/code-rationale.adoc[]

==== Noncompliant code example

[source,go,diff-id=11,diff-type=noncompliant]
----
import (
	"github.com/gofiber/fiber/v2"
)

func Example(c *fiber.Ctx) error {
	url := c.Query("url")
	agent := fiber.Get(url) // Noncompliant
	statusCode, body, errs := agent.String()
	if len(errs) > 0 {
		return c.Status(fiber.StatusInternalServerError).JSON(fiber.Map{
			"errs": errs,
		})
	}
	return c.Status(statusCode).SendString(body)
}
----

==== Compliant solution

[source,go,diff-id=11,diff-type=compliant]
----
import (
	"strings"

	"github.com/gofiber/fiber/v2"
)

func Example(c *fiber.Ctx) error {
	url := c.Query("url")
	if !strings.HasPrefix(url, "https://example.com/") {
		return c.SendString("Specified url is not allowed")
	}
	agent := fiber.Get(url)
	statusCode, body, errs := agent.String()
	if len(errs) > 0 {
		return c.Status(fiber.StatusInternalServerError).JSON(fiber.Map{
			"errs": errs,
		})
	}
	return c.Status(statusCode).SendString(body)
}
----

=== How does this work?

include::../../common/fix/pre-approved-list.adoc[]

The compliant code example uses such an approach.

=== Pitfalls

include::../../common/pitfalls/starts-with.adoc[]
