== How to fix it in Go Standard Library

=== Code examples

include::../../common/fix/code-rationale.adoc[]

==== Noncompliant code example

[source,go,diff-id=1,diff-type=noncompliant]
----
import (
	"fmt"
	"io"
	"net/http"
)

func Example(w http.ResponseWriter, req *http.Request) {
	url := req.FormValue("url")
	resp, err := http.Get(url) // Noncompliant
	if err != nil {
		fmt.Fprintf(w, "Error when reading response: %s\n", err.Error())
		return
	}
	defer resp.Body.Close()
	respBytes, err := io.ReadAll(resp.Body)
	if err != nil {
		fmt.Fprintf(w, "Error when reading response body: %s\n", err.Error())
		return
	}
	fmt.Fprintf(w, "Response: %s", string(respBytes))
}

----

==== Compliant solution

[source,go,diff-id=1,diff-type=compliant]
----
import (
	"fmt"
	"io"
	"net/http"
	"strings"
)

func Example(w http.ResponseWriter, req *http.Request) {
	url := req.FormValue("url")
	if !strings.HasPrefix(url, "https://example.com/") {
		fmt.Fprintln(w, "Specified host is not allowed")
		return
	}
	resp, err := http.Get(url)
	if err != nil {
		fmt.Fprintf(w, "Error when reading response: %s\n", err.Error())
		return
	}
	defer resp.Body.Close()
	respBytes, err := io.ReadAll(resp.Body)
	if err != nil {
		fmt.Fprintf(w, "Error when reading response body: %s\n", err.Error())
		return
	}
	fmt.Fprintf(w, "Response: %s", string(respBytes))
}

----

=== How does this work?

include::../../common/fix/pre-approved-list.adoc[]

The compliant code example uses such an approach.

=== Pitfalls

include::../../common/pitfalls/starts-with.adoc[]
