<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Shadowing occurs when a local variable has the same name as a variable or a field in an outer scope.</p>
</div>
<div class="paragraph">
<p>This can lead to three main problems:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>Confusion:
The same name can refer to different variables in different parts of the scope, making the code hard to read and understand.</p>
</li>
<li>
<p>Unintended Behavior:
You might accidentally use the wrong variable, leading to hard-to-detect bugs.</p>
</li>
<li>
<p>Maintenance Issues:
If the inner variable is removed or renamed, the code&#8217;s behavior might change unexpectedly because the outer variable is now being used.</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>To avoid these problems, rename the shadowing, shadowed, or both identifiers to accurately represent their purpose with unique and meaningful names.</p>
</div>
<div class="paragraph">
<p>This rule focuses on variables in methods that shadow a field.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-flex" data-lang="flex">class Foo {
  public var myField:int;

  public function doSomething():String {
    var myField:int = 0; // Noncompliant
    //...
  }
}</code></pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_deprecates_s1524">deprecates: <a data-rspec-id="S1524" class="rspec-auto-link">S1524</a></h3>

</div>
<div class="sect2">
<h3 id="_relates_to_s801">relates to: <a data-rspec-id="S801" class="rspec-auto-link">S801</a></h3>

</div>
<div class="sect2">
<h3 id="_on_11_jul_2013_100430_dinesh_bolkensteyn_wrote">on 11 Jul 2013, 10:04:30 Dinesh Bolkensteyn wrote:</h3>
<div class="paragraph">
<p>Ignore constructors</p>
</div>
<div class="paragraph">
<p>Ignore setters, method which start with 'set'</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_15_jul_2013_135906_dinesh_bolkensteyn_wrote">on 15 Jul 2013, 13:59:06 Dinesh Bolkensteyn wrote:</h3>
<div class="paragraph">
<p>Implemented by http://jira.codehaus.org/browse/SONARJAVA-211</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_31_oct_2013_163605_freddy_mallet_wrote">on 31 Oct 2013, 16:36:05 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>Is implemented by http://jira.codehaus.org/browse/SONARPLUGINS-3232 for Flex</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_23_feb_2014_224904_freddy_mallet_wrote">on 23 Feb 2014, 22:49:04 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>Is implemented by http://jira.codehaus.org/browse/SONARPLUGINS-3504 for PHP</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_12_mar_2019_164940_alexandre_gigleux_wrote">on 12 Mar 2019, 16:49:40 Alexandre Gigleux wrote:</h3>
<div class="paragraph">
<p>Decision was taken to deprecate in favor of <a data-rspec-id="S1117" class="rspec-auto-link">RSPEC-1117</a>. As a consequence the following maintenance tickets should be implemented:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>https://jira.sonarsource.com/browse/SONARPLSQL-722</p>
</li>
<li>
<p>https://github.com/SonarSource/SonarTS/issues/838</p>
</li>
<li>
<p>https://github.com/SonarSource/SonarJS/issues/1214</p>
</li>
</ul>
</div>
</div>
</div>
</div>