<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when code attempts to access hash elements using dot notation (e.g., <code>hash.key</code>) instead of the correct bracket notation (e.g., <code>hash[:key]</code> or <code>hash['key']</code>).</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>In Ruby, hashes and objects use different syntax for accessing their contents. Dot notation (<code>object.method</code>) is used to call methods on objects, while bracket notation (<code>hash[:key]</code> or <code>hash['key']</code>) is the correct way to access hash elements.</p>
</div>
<div class="paragraph">
<p>When you try to use dot notation on a hash, Ruby interprets this as a method call. Since hashes don&#8217;t have methods with those names, this results in a <code>NoMethodError</code> at runtime.</p>
</div>
<div class="paragraph">
<p>This confusion often happens because:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>Other programming languages may use dot notation for both objects and hash-like structures</p>
</li>
<li>
<p>The syntax looks similar to accessing object properties in languages like JavaScript</p>
</li>
<li>
<p>New Ruby developers may not yet understand the distinction between hashes and objects</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>The error will occur immediately when the code runs, making this a reliability issue that prevents the application from functioning correctly.</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="paragraph">
<p>Using dot notation on hashes causes immediate runtime failures with <code>NoMethodError</code> exceptions. This breaks application functionality and can lead to:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>Application crashes when the problematic code is executed</p>
</li>
<li>
<p>Poor user experience due to unexpected errors</p>
</li>
<li>
<p>Debugging time spent tracking down syntax mistakes</p>
</li>
<li>
<p>Potential data processing failures if hash access is part of critical workflows</p>
</li>
</ul>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Replace dot notation with bracket notation using the appropriate key type (symbol or string) that matches your hash definition.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-ruby" data-lang="ruby">person = {
  :name =&gt; "John",
  :age =&gt; 30
}

# This will raise NoMethodError
puts person.name  # Noncompliant
puts person.age   # Noncompliant</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-ruby" data-lang="ruby">person = {
  :name =&gt; "John",
  :age =&gt; 30
}

# Use bracket notation with symbols
puts person[:name]
puts person[:age]</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Ruby Hash Documentation - <a href="https://ruby-doc.org/core/Hash.html">Official Ruby documentation for Hash class and its methods</a></p>
</li>
<li>
<p>Ruby Hash Access Methods - <a href="https://ruby-doc.org/core/Hash.html#method-i-5B-5D">Documentation for the bracket notation method used to access hash elements</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>