There are several reasons for a method not to have a method body:

* It is an unintentional omission, and should be fixed to prevent an unexpected behavior in production.
* It is not yet, or never will be, supported. In this case an `+UnsupportedOperationException+` should be thrown.
* The method is an intentionally-blank override. In this case a nested comment should explain the reason for the blank override.

== Noncompliant Code Example

----
public void doSomething() {
}

public void doSomethingElse() {
}
----

== Compliant Solution

----
@Override
public void doSomething() {
  // Do nothing because of X and Y.
}

@Override
public void doSomethingElse() {
  throw new UnsupportedOperationException();
}
----

== Exceptions

Default (no-argument) constructors are ignored when there are other constructors in the class, as are empty methods in abstract classes.

----
public abstract class Animal {
  void speak() {  // default implementation ignored
  }
}
----
