<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>The semantics of <code>Thread</code> and <code>Runnable</code> are different, and while it is technically correct to use <code>Thread</code> where a <code>Runnable</code> is expected, it is a bad practice to do so.</p>
</div>
<div class="paragraph">
<p>The crux of the issue is that <code>Thread</code> is a larger concept than <code>Runnable</code>.
A <code>Runnable</code> represents a task.
A <code>Thread</code> represents a task and its execution management (ie: how it should behave when started, stopped, resumed, &#8230;&#8203;).
It is both a task and a lifecycle management.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">public static void main(String[] args) {
	Thread runnable = new Thread() {
		@Override
		public void run() { /* ... */ }
	};
	new Thread(runnable).start();  // Noncompliant
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">public static void main(String[] args) {
	Runnable runnable = new Runnable() {
		@Override
		public void run() { /* ... */ }
	};
	new Thread(runnable).start();
}</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="ulist">
<ul>
<li>
<p>Argument n is a "Thread".</p>
</li>
<li>
<p>"XXX" is a "Thread".</p>
</li>
</ul>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_20_jan_2015_162005_ann_campbell_wrote">on 20 Jan 2015, 16:20:05 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~nicolas.peru] I&#8217;d like to add a simile to illustrate the issue. What do you think of this?</p>
</div>
<div class="quoteblock">
<blockquote>
<div class="paragraph">
<p>It would be something like asking Sir Edmund Hillary to take a hike up the nearest hill with you.</p>
</div>
</blockquote>
</div>
</div>
</div>
</div>