include::../description.adoc[]

== Noncompliant Code Example

----
private int Divide(int numerator, int denominator) 
{
    return numerator / denominator;  // FIXME denominator value might be  0
}
----

include::../see.adoc[]

ifdef::env-github,rspecator-view[]
'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
