== Why is this an issue?

include::../description.adoc[]

Note that this rule will raise issues on ``++a == a++`` and ``++a != a++`` expressions which are sometime used to detect ``++NaN++`` values.
It is recommended to use instead https://pkg.go.dev/math#IsNaN[``++math.IsNaN()++``] or an equivalent function.
This will improve code readability.

=== Noncompliant code example

[source,go]
----
func main() {
  v1 := (true && false) && (true && false) // Noncompliant
}
----

=== Compliant solution

[source,go]
----
func main() {
  v1 := (true && false) // Compliant
}
----

=== Exceptions

This rule ignores ``++*++``, ``+``, ``++<<++`` and ``++=++``. 

include::../see.adoc[]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../highlighting.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
