<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>In Azure Resource Manager (ARM) templates, dependencies between resources can be defined in two ways: implicitly or explicitly.
We recommend not adding explicit dependencies when they are already defined implicitly to avoid redundancy and make the template more straightforward to read.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>In Azure Resource Manager (ARM) templates, dependencies between resources can be defined in two ways: implicitly or explicitly.
An implicit dependency is set when you use the <code>reference</code> function and pass in the resource name.
An explicit dependency is defined when you add a <code>dependsOn</code> element.
However, a code smell arises when these dependencies are used simultaneously for the same resources.
This redundancy is unnecessary and can lead to confusion.
Therefore, to maintain clarity and efficiency in your code, it is best to omit explicit dependencies when they are already defined implicitly.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it_in_json_templates">How to fix it in JSON templates</h2>
<div class="sectionbody">
<div class="paragraph">
<p>If a resource references another with a <code>reference</code> function, remove the <code>dependsOn</code> element if it points to the same resource.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-json" data-lang="json">{
  "apiVersion": "2019-04-01",
  "type": "Microsoft.Network/networkInterfaces",
  "name": "exampleNic",
  "location": "[resourceGroup().location]",
  "dependsOn": [
    "[resourceId('Microsoft.Network/virtualNetworks', 'exampleVNet')]"
  ],
  "properties": {
    "ipConfigurations": [
      {
        "name": "ipconfig1",
        "properties": {
          "subnet": {
            "id": "[reference('Microsoft.Network/virtualNetworks/exampleVNet/subnets/exampleSubnet').id]"
          }
        }
      }
    ]
  }
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-json" data-lang="json">{
  "apiVersion": "2019-04-01",
  "type": "Microsoft.Network/networkInterfaces",
  "name": "exampleNic",
  "location": "[resourceGroup().location]",
  "properties": {
    "ipConfigurations": [
      {
        "name": "ipconfig1",
        "properties": {
          "subnet": {
            "id": "[reference('Microsoft.Network/virtualNetworks/exampleVNet/subnets/exampleSubnet').id]"
          }
        }
      }
    ]
  }
}</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it_in_bicep">How to fix it in Bicep</h2>
<div class="sectionbody">
<div class="paragraph">
<p>If a resource references another resource by a symbolic name or with a <code>reference</code> function, remove the <code>dependsOn</code> element if it points to the same resource.
Note: Bicep provides the reference function, but in most cases, the symbolic name should be preferred over it.</p>
</div>
<div class="sect2">
<h3 id="_code_examples_2">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example_2">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-bicep" data-lang="bicep">resource exampleVNet 'Microsoft.Network/virtualNetworks@2020-06-01' existing = {
  name: 'exampleVNet'
}

resource exampleSubnet 'Microsoft.Network/virtualNetworks/subnets@2020-06-01' existing = {
  parent: exampleVNet
  name: 'exampleSubnet'
}

resource exampleNic 'Microsoft.Network/networkInterfaces@2020-06-01' = {
  name: 'exampleNic'
  location: resourceGroup().location
  dependsOn: [
    exampleSubnet  // Noncompliant
  ]
  properties: {
    ipConfigurations: [
      {
        name: 'ipconfig1'
        properties: {
          subnet: {
            id: exampleSubnet.id
          }
        }
      }
    ]
  }
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compilant_solution">Compilant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-bicep" data-lang="bicep">resource exampleVNet 'Microsoft.Network/virtualNetworks@2020-06-01' existing = {
  name: 'exampleVNet'
}

resource exampleSubnet 'Microsoft.Network/virtualNetworks/subnets@2020-06-01' existing = {
  parent: exampleVNet
  name: 'exampleSubnet'
}

resource exampleNic 'Microsoft.Network/networkInterfaces@2020-06-01' = {
  name: 'exampleNic'
  location: resourceGroup().location
  properties: {
    ipConfigurations: [
      {
        name: 'ipconfig1'
        properties: {
          subnet: {
            id: exampleSubnet.id
          }
        }
      }
    ]
  }
}</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Azure Resource Manager - <a href="https://learn.microsoft.com/en-us/azure/azure-resource-manager/templates/best-practices#resource-dependencies">Best practices for resource dependencies in ARM templates</a></p>
</li>
<li>
<p>Azure Resource Manager - <a href="https://learn.microsoft.com/en-us/azure/azure-resource-manager/templates/resource-dependency#reference-and-list-functions">Reference and list functions in ARM templates</a></p>
</li>
<li>
<p>Azure Resource Manager - <a href="https://learn.microsoft.com/en-us/azure/azure-resource-manager/bicep/best-practices#resource-definitions">Best practices for resource definitions in Bicep</a></p>
</li>
<li>
<p>Azure Resource Manager - <a href="https://learn.microsoft.com/en-us/azure/azure-resource-manager/bicep/bicep-functions-resource#reference">Reference function in Bicep</a></p>
</li>
</ul>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove this explicit dependency as it is already defined implicitly.</p>
</div>
<div class="paragraph">
<p>Secondary location message: Implicit dependency is created via the "reference" function.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>Highlight the redundant <code>dependsOn</code> element.</p>
</div>
</div>
</div>
</div>