<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Ternary expressions, while concise, can often lead to code that is difficult to read and understand, especially when they are nested or complex.
Prioritizing readability fosters maintainability and reduces the likelihood of bugs.
Therefore, they should be removed in favor of more explicit control structures, such as <code>if</code>/<code>else</code> statements, to improve the clarity and readability of the code.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">System.out.println(i&gt;10?"yes":"no");  // Noncompliant</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">if (i &gt; 10) {
  System.out.println("yes");
} else {
  System.out.println("no");
}</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Convert this usage of the ternary operator to an "if"/"else" structure.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_deprecates_s1521">deprecates: <a data-rspec-id="S1521" class="rspec-auto-link">S1521</a></h3>

</div>
<div class="sect2">
<h3 id="_on_7_apr_2016_093902_elena_vilchik_wrote">on 7 Apr 2016, 09:39:02 Elena Vilchik wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] Shouldn&#8217;t it be <code>its</code> instead of <code>it's</code> in description?</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_7_apr_2016_134735_ann_campbell_wrote">on 7 Apr 2016, 13:47:35 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>Fixed. Thanks [~elena.vilchik]. That&#8217;s one I chronically screw up</p>
</div>
</div>
</div>
</div>