<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This includes file and I/O functions <code>fgetpos</code>, <code>fopen</code>, <code>ftell</code>, <code>gets</code>, <code>perror</code>, <code>remove</code>, <code>rename</code> and <code>ungetc</code>.</p>
</div>
<div class="paragraph">
<p>Streams and file I/O have a large number of unspecified, undefined and implementation-defined behaviors associated with them. It is assumed within MISRA C that they will not normally be needed in production code in embedded systems.</p>
</div>
<div class="paragraph">
<p>If any of the features of <code>stdio.h</code> need to be used in production code, then the issues associated with the features need to be understood.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">#include &lt;stdio.h&gt; /* Noncompliant */</code></pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p>MISRA C:2004, 20.9 - The input/output library &lt;stdio.h&gt; shall not be used in production code.</p>
</li>
<li>
<p>MISRA C&#43;&#43;:2008, 27-0-1 - The stream input/output library &lt;cstdio&gt; shall not be used.</p>
</li>
<li>
<p>MISRA C:2012, 21.6 - The Standard Library input/output functions shall not be used</p>
</li>
</ul>
</div>
<hr>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_is_related_to_s1055">is related to: <a data-rspec-id="S1055" class="rspec-auto-link">S1055</a></h3>

</div>
<div class="sect2">
<h3 id="_on_31_mar_2015_190258_evgeny_mandrikov_wrote">on 31 Mar 2015, 19:02:58 Evgeny Mandrikov wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] implementation seems more complete (SQALE, description) than this spec.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_13_apr_2015_194720_evgeny_mandrikov_wrote">on 13 Apr 2015, 19:47:20 Evgeny Mandrikov wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] I&#8217;m wondering why blocker, but not active by default? Note that in implementation currently major and not active.</p>
</div>
</div>
</div>
</div>