<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>ORACLE 11g introduced the <code>SIMPLE_INTEGER</code> data type, which is a sub-type of <code>PLS_INTEGER</code>, and covers the same range. There are three main differences between the two types:</p>
</div>
<div class="ulist">
<ul>
<li>
<p><code>SIMPLE_INTEGER</code> is always <code>NOT NULL</code>. So when the value of the declared variable is never going to be null, you can declare it as <code>SIMPLE_INTEGER</code>.</p>
</li>
<li>
<p>You will never face a numeric overflow using <code>SIMPLE_INTEGER</code> because this data type wraps around without giving any error.</p>
</li>
<li>
<p>The <code>SIMPLE_INTEGER</code> data type gives a major performance boost over <code>PLS_INTEGER</code> when the code is compiled in "NATIVE" mode, because arithmetic operations on <code>SIMPLE_INTEGER</code> type are performed directly at the hardware level.</p>
</li>
</ul>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-sql" data-lang="sql">DECLARE
  v1 PLS_INTEGER; -- Noncompliant
  v2 VARCHAR2(10);
BEGIN
  NULL;
END;
/</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-sql" data-lang="sql">DECLARE
  v1 SIMPLE_INTEGER := 42;
  v2 VARCHAR2(10);
BEGIN
  NULL;
END;
/</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Convert this variable to a "SIMPLE_INTEGER".</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_27_jul_2017_135022_ann_campbell_wrote">on 27 Jul 2017, 13:50:22 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~michal.budzowski] there is a clash between this rule and <a data-rspec-id="S4081" class="rspec-auto-link">RSPEC-4081</a>, but <a data-rspec-id="S4081" class="rspec-auto-link">RSPEC-4081</a> wasn&#8217;t on by default. Nonetheless, I&#8217;ve removed this one from the default profile.</p>
</div>
</div>
</div>
</div>