== Why is this an issue?

include::../description.adoc[]


=== Noncompliant code example

[source,java]
----
"[ab]|a"   // The "|a" is redundant because "[ab]" already matches "a"
".*|a"     // .* matches everything, so any other alternative is redundant
----


=== Compliant solution

[source,java]
----
"[ab]"
".*"
----
ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

include::highlighting.adoc[]

endif::env-github,rspecator-view[]
