<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Error handling should be defined on file operations to ensure correct processing of program failures rather than defaulting to an abnormal program end. This rule checks two things:</p>
</div>
<div class="ulist">
<ul>
<li>
<p><code>INFSR</code> error handling is defined on file operations.</p>
</li>
<li>
<p>error handling is delegated to an appropriately named routine.</p>
</li>
</ul>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="paragraph">
<p>Given the default parameter value, <code>^*PSSR$</code>:</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-rpg" data-lang="rpg">     F* Noncompliant
     FIPBDCCP   IF   E           K DISK

     FEP210     CF   E             WORKSTN
     F                                     SFILE(EP21003S:RELN03)

     FEP471R1   O    E             PRINTER
     F                                     USROPN
     F                                     INFDS(W1SF01)</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-rpg" data-lang="rpg">     FIPBDCCP   IF   E           K DISK    INFSR(*PSSR)

     FEP210     CF   E             WORKSTN INFSR(*PSSR)
     F                                     SFILE(EP21003S:RELN03)

     FEP471R1   O    E             PRINTER INFSR(*PSSR)
     F                                     USROPN
     F                                     INFDS(W1SF01)</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Add the error handling "INFSR(xxxx)" keyword to this F spec</p>
</div>
</div>
<div class="sect2">
<h3 id="_parameters">Parameters</h3>
<div class="sidebarblock">
<div class="content">
<div class="title">format_error_handling</div>
<div class="listingblock">
<div class="content">
<pre>^{empty}*PSSR$</pre>
</div>
</div>
<div class="paragraph">
<p>Regular expression describing the names of valid error handling routines</p>
</div>
</div>
</div>
</div>
</div>
</div>