Error handling should be defined on file operations to ensure correct processing of program failures rather than defaulting to an abnormal program end. This rule checks two things: 

* <code>INFSR</code> error handling is defined on file operations.
* error handling is delegated to an appropriately named routine.


== Noncompliant Code Example

Given the default parameter value, <code>^\*PSSR$</code>:

----
     F* Noncompliant
     FIPBDCCP   IF   E           K DISK

     FEP210     CF   E             WORKSTN
     F                                     SFILE(EP21003S:RELN03)     

     FEP471R1   O    E             PRINTER 
     F                                     USROPN
     F                                     INFDS(W1SF01)
----


== Compliant Solution

----
     FIPBDCCP   IF   E           K DISK    INFSR(*PSSR)

     FEP210     CF   E             WORKSTN INFSR(*PSSR)
     F                                     SFILE(EP21003S:RELN03)

     FEP471R1   O    E             PRINTER INFSR(*PSSR)
     F                                     USROPN
     F                                     INFDS(W1SF01)
----

