<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>When using the <code>Stream</code> API, call chains should be simplified as much as possible to improve readability and maintainability.</p>
</div>
<div class="paragraph">
<p>This rule raises an issue when one of the following substitution can be made:</p>
</div>
<table class="tableblock frame-all grid-all stretch">
<colgroup>
<col style="width: 50%;">
<col style="width: 50%;">
</colgroup>
<thead>
<tr>
<th class="tableblock halign-center valign-top">Original</th>
<th class="tableblock halign-center valign-top">Preferred</th>
</tr>
</thead>
<tbody>
<tr>
<td class="tableblock halign-center valign-top"><p class="tableblock"><code>stream.collect(counting())</code></p></td>
<td class="tableblock halign-center valign-top"><p class="tableblock"><code>stream.count()</code></p></td>
</tr>
<tr>
<td class="tableblock halign-center valign-top"><p class="tableblock"><code>stream.collect(maxBy(comparator))</code></p></td>
<td class="tableblock halign-center valign-top"><p class="tableblock"><code>stream.max(comparator)</code></p></td>
</tr>
<tr>
<td class="tableblock halign-center valign-top"><p class="tableblock"><code>stream.collect(minBy(comparator))</code></p></td>
<td class="tableblock halign-center valign-top"><p class="tableblock"><code>stream.min(comparator)</code></p></td>
</tr>
<tr>
<td class="tableblock halign-center valign-top"><p class="tableblock"><code>stream.collect(mapping(mapper))</code></p></td>
<td class="tableblock halign-center valign-top"><p class="tableblock"><code>stream.map(mapper).collect()</code></p></td>
</tr>
<tr>
<td class="tableblock halign-center valign-top"><p class="tableblock"><code>stream.collect(reducing(...))</code></p></td>
<td class="tableblock halign-center valign-top"><p class="tableblock"><code>stream.reduce(...)</code></p></td>
</tr>
<tr>
<td class="tableblock halign-center valign-top"><p class="tableblock"><code>stream.collect(summingInt(mapper))</code></p></td>
<td class="tableblock halign-center valign-top"><p class="tableblock"><code>stream.mapToInt(mapper).sum()</code></p></td>
</tr>
<tr>
<td class="tableblock halign-center valign-top"><p class="tableblock"><code>stream.collect(summingLong(mapper))</code></p></td>
<td class="tableblock halign-center valign-top"><p class="tableblock"><code>stream.mapToLong(mapper).sum()</code></p></td>
</tr>
<tr>
<td class="tableblock halign-center valign-top"><p class="tableblock"><code>stream.collect(summingDouble(mapper))</code></p></td>
<td class="tableblock halign-center valign-top"><p class="tableblock"><code>stream.mapToDouble(mapper).sum()</code></p></td>
</tr>
</tbody>
</table>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">int count = stream.collect(counting());  // Noncompliant</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">int count = stream.count();</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Replace this "collect(xxx())" call by "yyy"</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>The "stream.collect()" call</p>
</div>
</div>
</div>
</div>