Statement name should be unique.

== Why is this an issue?

In JCL, the name of some statement must be unique, otherwise it may cause an error or an unexpected behavior.
IBM reference state that the JOB statement name should be unique, and that each of the following statements should have a unique name withing their job:
* NOTIFY
* COMMAND
* ENDCNTL
* EXEC
* IF / THEN / ELSE / ENDIF
* INCLUDE
* OUTPUT (job-level or step-level)
* PROC (only for in-stream PROC)
* SET
* XMIT

== How to fix it

=== Code examples

==== Noncompliant code example

[source,text,diff-id=1,diff-type=noncompliant]
----
//MYJOB JOB TEST1
//* [...]
//MYJOB JOB TEST2
----

[source,text,diff-id=2,diff-type=noncompliant]
----
//MYJOB JOB TEST
//SEND COMMAND 'SEND ''This message will be sent to the user BOB'',USER=(BOB),'
//* [...]
//SEND COMMAND 'SEND ''Another message to be sent to user BOB'',USER=(BOB),'
----

==== Compliant solution

[source,text,diff-id=1,diff-type=compliant]
----
//MYJOB1 JOB TEST1
//* [...]
//MYJOB2 JOB TEST2
----

[source,text,diff-id=2,diff-type=compliant]
----
//MYJOB JOB TEST
//SEND1 COMMAND 'SEND ''This message will be sent to the user BOB'',USER=(BOB),'
//* [...]
//SEND2 COMMAND 'SEND ''Another message to be sent to user BOB'',USER=(BOB),'
----

== Resources

=== Documentation

* https://www.ibm.com/docs/en/zos/3.1.0?topic=statement-label-field[IBM Reference - NOTIFY Label field]
* https://www.ibm.com/docs/en/zos/3.1.0?topic=description-name-field[IBM Reference - COMMAND Name field]
* https://www.ibm.com/docs/en/zos/3.1.0?topic=d-label-field[IBM Reference - ENDCNTL Label field]
* https://www.ibm.com/docs/en/zos/3.1.0?topic=d-name-field-1[IBM Reference - EXEC Name field]
* https://www.ibm.com/docs/en/zos/3.1.0?topic=d-name-field-2[IBM Reference - IF/THEN/ELD/ENDIF Name field]
* https://www.ibm.com/docs/en/zos/3.1.0?topic=d-name-field-3[IBM Reference - INCLUDE Name field]
* https://www.ibm.com/docs/en/zos/3.1.0?topic=d-name-field-6[IBM Reference - OUTPUT Name field]
* https://www.ibm.com/docs/en/zos/3.1.0?topic=d-name-field-8[IBM Reference - PROC Name field]
* https://www.ibm.com/docs/en/zos/3.1.0?topic=d-name-field-10[IBM Reference - SET Name field]
* https://www.ibm.com/docs/en/zos/3.1.0?topic=d-name-field-11[IBM Reference - XMIT Name field]
