== Why is this an issue?

Step names should be unique within a job. Duplicate step names can lead to unpredictable behavior when using step references and overrides.

== How to fix it

=== Code examples

==== Noncompliant code example

[source,text,diff-id=1,diff-type=noncompliant]
----
//MYJOB JOB TEST1
//STEP  EXEC PGM=PROG1   <-- Noncompliant
//STEP  EXEC PGM=PROG2   <-- Noncompliant
----

==== Compliant solution

[source,text,diff-id=1,diff-type=compliant]
----
//MYJOB JOB TEST1
//STEP1 EXEC PGM=PROG1
//STEP2 EXEC PGM=PROG2
----

== Resources

=== Documentation

* https://www.ibm.com/docs/en/zos/3.1.0?topic=d-name-field-1[IBM Reference - EXEC Name field]
