<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Various aspects of <code>ctime</code> are implementation-defined or unspecified, such as the formats of times.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">#include &lt;ctime&gt;  /* Noncompliant */

void f()
{
  clock();
}</code></pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p>MISRA C&#43;&#43;:2008, 18-0-4 - The time handling functions of library &lt;ctime&gt; shall not be used.</p>
</li>
</ul>
</div>
<hr>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_31_mar_2015_190027_evgeny_mandrikov_wrote">on 31 Mar 2015, 19:00:27 Evgeny Mandrikov wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] implementation seems more complete (SQALE, description) than this spec.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_9_apr_2015_140305_evgeny_mandrikov_wrote">on 9 Apr 2015, 14:03:05 Evgeny Mandrikov wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] Compilers not mentioned in MISRA, so I have real doubts that there is dependency on compiler - most likely MISRA refers to differences in implementation of standard library and compilers can be used with various libraries.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_10_apr_2015_131636_ann_campbell_wrote">on 10 Apr 2015, 13:16:36 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~evgeny.mandrikov] my understanding of "undefined behavior" is that what actually happens is left up to the compiler.</p>
</div>
<div class="paragraph">
<p>If you like, I can revert this to the MISRA text verbatim, but I was trying to give a little more value.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_13_apr_2015_194125_evgeny_mandrikov_wrote">on 13 Apr 2015, 19:41:25 Evgeny Mandrikov wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] I&#8217;m wondering why blocker, but not active by default? Note that in implementation currently major and not active.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_13_apr_2015_194424_evgeny_mandrikov_wrote">on 13 Apr 2015, 19:44:24 Evgeny Mandrikov wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] And answering on previous question - I strongly believe that in this is case "undefined behavior" means that it is left up to the implementation of standard library, which is not necessarily comes with compiler.</p>
</div>
</div>
</div>
</div>