Various aspects of ``ctime`` are implementation-defined or unspecified, such as the formats of times. 


== Noncompliant Code Example

----
#include <ctime>  /* Noncompliant */

void f()
{
  clock();
}
----


== See

* MISRA {cpp}:2008, 18-0-4 - The time handling functions of library <ctime> shall not be used.

