Detect missing double quotes to prevent globbing and word splitting.

== Why is this an issue?

Variable references and command substitutions undergo word splitting and pathname expansion (globbing).

This causes issues if the variable contains whitespaces or shell pathname expansion (glob) characters like `*`.

=== What is the potential impact?

This issue can lead to bugs if the variable contains sensitive characters, which may not be interpreted correctly.

== How to fix it
Add the missing double quotes to this variable.

=== Code examples

==== Noncompliant code example

[source,docker,diff-id=1,diff-type=noncompliant]
----
RUN $test
----

[source,docker,diff-id=1,diff-type=compliant]
----
RUN ./exec $test
----

Demonstration using the `echo` command:
[source,docker,diff-id=1,diff-type=noncompliant]
----
test="command t*.sh"
echo $test
----
This will print `"RUN command temp1.sh temp2.sh"`, as `*` is substituted to matching files in the current folder.
In this example the folder contains the files `temp1.sh` and `temp2.sh`

==== Compliant solution

[source,docker,diff-id=1,diff-type=compliant]
----
RUN "$test"
----

[source,docker,diff-id=1,diff-type=compliant]
----
RUN ./exec "$test"
----

Demonstration using the `echo` command:
[source,docker,diff-id=1,diff-type=noncompliant]
----
test="command t*.sh"
echo "$test"
----
This will print `"RUN command t*.sh"`, as intended.


== Resources

=== Documentation

* https://tldp.org/LDP/abs/html/globbingref.html[Globbing]
* https://dwheeler.com/essays/filenames-in-shell.html#doublequote[Filenames and Pathnames in Shell: How to do it Correctly]

ifdef::env-github,rspecator-view[]
'''
== Implementation Specification
(visible only on this page)

=== Message

Add the missing double quotes to this variable, as it can lead to unexpected behaviour.

=== Highlighting

Highlight the entire command which is using unquoted variables.

'''
endif::env-github,rspecator-view[]
