Detect missing double quotes to prevent globbing and word splitting.

== Why is this an issue?

FIXME: remove the unused optional headers (that are commented out)

Variable references and command substitutions undergo shell field splitting and pathname expansion.

This causes issues if the variable contains whitespaces or shell pathname expansion (glob) characters like `*`.

[source,shell,diff-id=1,diff-type=noncompliant]
----
splittingIssue=" Hello      World "
echo $splitting
----

Will lead to print `"Hello World"`, missing most of the whitespaces.

[source,shell,diff-id=1,diff-type=noncompliant]
----
globbingIssue="Hello * World"
echo $globbingIssue
----

Will lead to print `"Hello file1.txt file2.txt World"`, whereas `*` will be expanded to all files in the respective folder.

=== What is the potential impact?

This issue can lead to bugs, if the variable contains sensitive characters which may not be interpreted correctly.

== How to fix it
//== How to fix it in FRAMEWORK NAME
Add the missing double quotes to this variable.

=== Code examples

==== Noncompliant code example

[source,text,diff-id=1,diff-type=noncompliant]
----
RUN $test
----

----
RUN ./exec $test
----

==== Compliant solution

[source,text,diff-id=1,diff-type=compliant]
----
RUN "$test"
----

----
RUN ./exec "$test"
----


== Resources

=== Documentation

* https://tldp.org/LDP/abs/html/globbingref.html[Globbing]
* https://dwheeler.com/essays/filenames-in-shell.html#doublequote[Filenames and Pathnames in Shell: How to do it Correctly]
