Detect missing double quotes to prevent globbing and word splitting.

== Why is this an issue?

Variable references and command substitutions go through word splitting and pathname expansion (globbing).

This causes issues if the variable contains whitespaces or shell pathname expansion (glob) characters like `*`.

=== What is the potential impact?

This issue can lead to bugs if the variable contains sensitive characters, which may not be interpreted correctly.

== How to fix it
Add the missing double quotes to this variable.

=== Code examples

==== Noncompliant code example

[source,docker,diff-id=1,diff-type=noncompliant]
----
RUN $test
----

[source,docker,diff-id=2,diff-type=noncompliant]
----
RUN ./exec $test
----

Demonstration using the `echo` command:
[source,docker,diff-id=3,diff-type=noncompliant]
----
ARG test="command t*.sh"
RUN echo $test
----
This will print `"command temp1.sh temp2.sh"`, as `*` is substituted to matching files in the current folder.
In this example the folder contains the files `temp1.sh` and `temp2.sh`

==== Compliant solution

[source,docker,diff-id=1,diff-type=compliant]
----
RUN "$test"
----

[source,docker,diff-id=2,diff-type=compliant]
----
RUN ./exec "$test"
----

Demonstration using the `echo` command:
[source,docker,diff-id=3,diff-type=compliant]
----
ARG test="command t*.sh"
RUN echo "$test"
----
This will print `"command t*.sh"`, as intended.


== Resources

=== Documentation

* https://tldp.org/LDP/abs/html/globbingref.html[Linux Documentation Project - Globbing]
* https://dwheeler.com/essays/filenames-in-shell.html#doublequote[Filenames and Pathnames in Shell: How to do it Correctly]
* https://www.shellcheck.net/wiki/SC2086[ShellCheck SC2086]

ifdef::env-github,rspecator-view[]
'''
== Implementation Specification
(visible only on this page)

=== Message

Add the missing double quotes to this variable, as it can lead to unexpected behaviour.

=== Highlighting

Highlight the entire command which is using unquoted variables.

'''
endif::env-github,rspecator-view[]
