== How to fix it in .NET

=== Code examples

include::../../common/fix/code-rationale.adoc[]

==== Noncompliant code example

[source,text,diff-id=1,diff-type=noncompliant]
----
using System.Diagnostics;

public class ExampleController : Controller
{
    public void Example(string name, string value)
    {
        Environment.SetEnvironmentVariable(name, value); // noncompliant
    }
}
----

==== Compliant solution

[source,text,diff-id=1,diff-type=compliant]
----
using System.Diagnostics;
using System.Text.RegularExpressions;

public class ExampleController : Controller
{
    public void Example(string value)
    {
        string pattern = "^[a-zA-Z0-9]*$";
        Match m = Regex.Match(value, pattern);
        if (m.Success) {
            Environment.SetEnvironmentVariable("ENV_VAR", value);
        }
    }
}
----

include::../../common/fix/how-does-this-work.adoc[]