== Why is this an issue?

Unsigned integers have different arithmetic operators than signed ones - operators that few developers understand. Therefore, signed types should be preferred where possible.


=== Noncompliant code example

[source,vbnet]
----
Module Module1
    Sub Main()
        Dim foo1 As UShort   ' Noncompliant
        Dim foo2 As UInteger ' Noncompliant
        Dim foo3 As ULong    ' Noncompliant
    End Sub
End Module
----


=== Compliant solution

[source,vbnet]
----
Module Module1
    Sub Main()
        Dim foo1 As Short
        Dim foo2 As Integer
        Dim foo3 As Long   
    End Sub
End Module
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Change this unsigned type to "xxx".


'''
== Comments And Links
(visible only on this page)

=== on 17 Oct 2016, 13:53:41 Ann Campbell wrote:
If only you had asked me this 10 months ago [~tamas.vajk]. Perhaps this is a rule that was present in the old version of the plugin?

=== on 17 Oct 2016, 14:07:20 Tamas Vajk wrote:
\[~ann.campbell.2] Okay, then I'm going to disable it by default.

endif::env-github,rspecator-view[]
