<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when error messages are printed to stdout instead of stderr.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Error messages should be sent to stderr (file descriptor 2) to properly separate error output from normal program output.</p>
</div>
<div class="paragraph">
<p>When error messages are sent to stdout, they become mixed with the program&#8217;s regular output. This creates several problems:</p>
</div>
<div class="ulist">
<ul>
<li>
<p><strong>Pipeline interference</strong>: If the script&#8217;s output is piped to another command, error messages will be processed as if they were normal data, potentially causing the downstream command to fail or produce incorrect results.</p>
</li>
<li>
<p><strong>Output redirection issues</strong>: When stdout is redirected to a file (e.g., <code>script.sh &gt; output.txt</code>), error messages end up in the output file instead of being displayed to the user, making debugging difficult.</p>
</li>
<li>
<p><strong>Automated processing problems</strong>: Scripts used in automated systems or CI/CD pipelines need clear separation between normal output and error messages for proper error handling and logging.</p>
</li>
<li>
<p><strong>User experience</strong>: Users expect error messages to appear on their terminal even when redirecting normal output, which only happens when errors go to stderr.</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>By convention, stderr is specifically designed for error messages, warnings, and diagnostic information, while stdout is reserved for the program&#8217;s primary output.</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="paragraph">
<p>Error messages mixed with normal output can cause downstream commands in pipelines to fail, make debugging difficult when output is redirected, and break automated systems that rely on proper separation of normal output and error messages.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Redirect error messages to stderr using <code>&gt;&amp;2</code> with echo commands.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-shell" data-lang="shell">echo "Error: file not found" # Noncompliant</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-shell" data-lang="shell">echo "Error: file not found" &gt;&amp;2</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Bash Manual - Redirections - <a href="https://www.gnu.org/software/bash/manual/html_node/Redirections.html">Official documentation on bash redirection operators including stderr redirection</a></p>
</li>
<li>
<p>Advanced Bash-Scripting Guide - I/O Redirection - <a href="https://tldp.org/LDP/abs/html/io-redirection.html">Comprehensive guide to input/output redirection in bash scripts</a></p>
</li>
</ul>
</div>
</div>
<div class="sect2">
<h3 id="_standards">Standards</h3>
<div class="ulist">
<ul>
<li>
<p>POSIX.1-2017 - Standard Streams - <a href="https://pubs.opengroup.org/onlinepubs/9699919799/basedefs/V1_chap03.html#tag_03_395">POSIX specification defining standard input, output, and error streams</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>