include::../description.adoc[]

== Noncompliant Code Example

[source,java]
----
"foo()"  // Noncompliant, will match only 'foo'
----

== Compliant Solution

[source,java]
----
"foo\\(\\)"  // Matches 'foo()'
----

include::../implementation.adoc[]
