== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,java]
----
"foo()"  // Noncompliant, will match only 'foo'
----

=== Compliant solution

[source,java]
----
"foo\\(\\)"  // Matches 'foo()'
----

include::../implementation.adoc[]
