<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Regular expressions are patterns used to match and manipulate strings based on specific rules.</p>
</div>
<div class="paragraph">
<p>The pattern provided to PHP regular expression functions is required to be enclosed in valid delimiters to be working correctly.</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="paragraph">
<p>Failing to enclose the pattern with valid delimiters will result in a PHP warning and the pattern never matching.
Since the warning only appears during runtime when the pattern is evaluated, such a mistake risks to get unnoticed into production.</p>
</div>
<div class="paragraph">
<p>The use of suppressed warnings in PHP can further complicate this issue.
As regex pattern are often used for validation or sanitization, the suppress warnings feature can obscure any problems regarding incorrect patterns.
As the incorrect patterns are not recognized and have no effect, this could lead to more serious security issues.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it_in_core_php">How to fix it in Core PHP</h2>
<div class="sectionbody">
<div class="paragraph">
<p>The provided pattern should be enclosed in valid delimiters.
A delimiter can be any character that is not alphanumeric, a backslash, or a whitespace.
Bracket style delimiters are also allowed.
Further information about possible delimiters can be found in the provided resources.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-php" data-lang="php">// Condition will always evaluate to false
if (preg_match("/.*", $input)) {
    echo "true";
} else {
    echo "false";
}

// Unclosed bracket delimiters
$result = preg_match("[abc", $input);</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-php" data-lang="php">if (preg_match("/.*/", $input)) {
    echo "true";
} else {
    echo "false";
}

$result = preg_match("[abc]", $input);</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p><a href="https://www.php.net/manual/en/regexp.reference.delimiters.php">Delimiters</a> - PHP Documentation</p>
</li>
</ul>
</div>
<hr>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="ulist">
<ul>
<li>
<p>In case there is no opening delimiter: Add delimiters to this regular expression.</p>
</li>
<li>
<p>In case there is no closing delimiter (x being the appropriate end delimiter): Add the missing "x" delimiter to this regular expression.</p>
</li>
</ul>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>The string representing the regular expression.</p>
</div>
</div>
</div>
</div>