``++DATA BEGIN OF ... OCCURS++`` has been deprecated and will eventually be removed. All usages should be replaced.


== Noncompliant Code Example

----
DATA BEGIN OF itab OCCURS n. "Noncompliant
  ... 
DATA END OF itab [VALID BETWEEN intlim1 AND intlim2]. 
----


== Compliant Solution

----
DATA BEGIN OF wa. 
       ... 
DATA END OF wa. 
DATA itab LIKE TABLE OF wa.
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

endif::env-github,rspecator-view[]
