== Why is this an issue?

include::../why.adoc[]

=== Exceptions

`equals` and `hashCode` methods are ignored because they might be automatically generated and might end up being difficult to understand, especially in the presence of many fields.

== How to fix it

include::../how.adoc[]

=== Code examples

==== Noncompliant code example

From org.sonar.api.utils.WildcardPattern.java in SonarQube

As shown below, the more a condition is nested, the more it increases complexity.

[source,java]
----

  private static String toRegexp(String antPattern,
                                 String directorySeparator) {
    final String escapedDirectorySeparator = '\\' + directorySeparator;
    final StringBuilder sb = new StringBuilder(antPattern.length());
    sb.append('^');
    int i = antPattern.startsWith("/") ||                                  // +1
    antPattern.startsWith("\\") ? 1 : 0;                                   // +1
    while (i < antPattern.length()) {                                      // +1
      final char ch = antPattern.charAt(i);
      if (SPECIAL_CHARS.indexOf(ch) != -1) {                               // +2 (nesting = 1)
        sb.append('\\').append(ch);
      } else if (ch == '*') {                                              // +1
        if (i + 1 < antPattern.length()                                    // +3 (nesting = 2)
          && antPattern.charAt(i + 1) == '*') {                            // +1
          if (i + 2 < antPattern.length()                                  // +4 (nesting = 3)
            && isSlash(antPattern.charAt(i + 2))) {                        // +1
            sb.append("(?:.*")
                 .append(escapedDirectorySeparator).append("|)");
            i += 2;
          } else {                                                         // +1
            sb.append(".*");
            i += 1;
          }
        } else {                                                           // +1
          sb.append("[^").append(escapedDirectorySeparator).append("]*?");
        }
      } else if (ch == '?') {                                              // +1
        sb.append("[^").append(escapedDirectorySeparator).append("]");
      } else if (isSlash(ch)) {                                            // +1
        sb.append(escapedDirectorySeparator);
      } else {                                                             // +1
        sb.append(ch);
      }
      i++;
    }
    sb.append('$');
    return sb.toString();
  }                                                                        // total complexity = 20
----

=== Pitfalls

As code is complex, ensure that you have unit tests that cover the code before refactoring.

include::../resources.adoc[]

include::../rspecator.adoc[]
