Nested code blocks can be used to create a new scope: Variables declared within that block cannot be accessed from the outside, and their lifetime end at the end of the block.

While this might seem convenient, using this feature in a function often indicates that it has too many responsibilities and should be refactored into smaller functions.


== Noncompliant Code Example

[source,cpp]
----

void f(Cache &c, int data) {
  int value;
  { // Noncompliant
    std::scoped_lock l(c.getMutex());
    if (c.hasKey(data)) {
      value = c.get(data);
    } else {
      value = compute(data);
      c.set(data, value);
    }
  } // Releases the mutex
  // Use value
}
----

== Compliant Solution

[source,cpp]
----
int getValue(Cache &c, int data) {
  std::scoped_lock l(c.getMutex());
  if (c.hasKey(data)) {
    return c.get(data);
  } else {
    value = compute(data);
    c.set(data, value);
    return value;
  }
}

void f(Cache &c, int data) {
  int value = getValue(c, data);
  // Use value
}
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
