Use either the ``++// ...++`` or ``++/* ... */++`` comment syntax, but be consistent and do not mix them within the same file.


== Noncompliant Code Example

----
/* Noncompliant; both comment syntaxes are used in the same file */
// Foo
/* Bar */
----


== Compliant Solution

----
// Compliant; uniform comment syntax
// Foo
// Bar
----


== See

* https://wiki.sei.cmu.edu/confluence/x/0zVGBQ[CERT, MSC55-J.] - Use comments consistently and in a readable fashion


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
