== Why is this an issue?

Use either the ``++// ...++`` or ``++/* ... */++`` comment syntax, but be consistent and do not mix them within the same file.


=== Noncompliant code example

[source,cpp]
----
/* Noncompliant; both comment syntaxes are used in the same file */
// Foo
/* Bar */
----


=== Compliant solution

[source,cpp]
----
// Compliant; uniform comment syntax
// Foo
// Bar
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Use either // or /* ... */ comments, but do not mix them (see lines X and Y).


'''
== Comments And Links
(visible only on this page)

=== relates to: S1103

=== on 15 Feb 2018, 15:05:43 Alban Auzeill wrote:
Irrelevant because:

ABAP has ``++*++`` and ``++"++`` comments, but for two different locations, not exchangeable.

Java, JavaScript, Flex, PHP, Swift, TypeScript have ``++/*++`` and ``++//++``, but ``++/**++`` is used for api doc and ``++//++`` in the code.

Python has ``++#++`` and ``++"""++`` comments, but for two different usages, piece of code and docstrings.

C#, PL/SQL, T-SQL the rule [RSPEC-2544] is more appropriate.



=== on 7 Mar 2018, 17:17:20 Janos Gyerik wrote:
Irrelevant for Go, there is a more specific, different recommendation for comments, explained in https://golang.org/doc/effective_go.html#commentary[Effective Go], that probably deserves it's own rule.

endif::env-github,rspecator-view[]
