== Why is this an issue?

Condition operators can quickly become hard to be read.


=== Noncompliant code example

[source,text]
----
function sayHello() {
  var a = (a === 'A') ? 'is A' : 'is not A'; // Noncompliant
}
----

