Having a bunch of standalone functions or procedures reduces maintainability because it becomes harder to find them and to see how they are related. Instead, they should be logically grouped into meaningful packages.


== Noncompliant Code Example

----
CREATE PROCEDURE show_name(name VARCHAR2) AS -- Non-Compliant
BEGIN
  DBMS_OUTPUT.PUT_LINE('Name: ' || name);
END;
/

DROP PROCEDURE show_name;
----


== Compliant Solution

----
CREATE PACKAGE employee AS
  PROCEDURE show_name;
END;
/

CREATE PACKAGE BODY employee AS
  name VARCHAR2(42);

  PROCEDURE show_name AS  -- Compliant
  BEGIN
    DBMS_OUTPUT.PUT_LINE('Name: ' || name);
  END;
END;
/

DROP PACKAGE employee;
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

endif::env-github,rspecator-view[]
