<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when a function returns a nil pointer that gets implicitly converted to an interface type, creating a non-nil interface containing a nil value.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>In Go, interfaces have two components: a type and a value. When you assign a nil pointer to an interface, the interface becomes non-nil because it contains type information, even though the underlying value is nil.</p>
</div>
<div class="paragraph">
<p>This creates a common gotcha where <code>err != nil</code> returns <code>true</code> even when you intended to return "no error". The interface is not nil because it contains a nil pointer of a specific type, rather than being a true nil interface.</p>
</div>
<div class="paragraph">
<p>This behavior can lead to:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>Logic errors where error checking fails unexpectedly</p>
</li>
<li>
<p>Difficult-to-debug issues where errors appear to exist when they shouldn&#8217;t</p>
</li>
<li>
<p>Inconsistent behavior compared to returning a true nil interface</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>The problem occurs most commonly with error handling, but can affect any interface type. When a function should return "no error" or "no value", it should return a true nil interface rather than a nil pointer wrapped in an interface.</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="paragraph">
<p>This issue can cause logic errors in error handling code, leading to incorrect program behavior. Applications may incorrectly treat successful operations as failures, potentially causing unnecessary error handling, logging, or user-facing error messages. In critical systems, this could lead to improper error recovery or system instability.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Return nil directly instead of returning a nil pointer that gets wrapped in an interface. When you want to indicate "no error" or "no value", return the zero value of the interface type (nil) rather than a nil pointer of a concrete type.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-go" data-lang="go">func processData() error {
    var err *MyError
    if someCondition {
        err = &amp;MyError{message: "something went wrong"}
    }
    return err // Noncompliant: returns nil pointer wrapped in interface
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-go" data-lang="go">func processData() error {
    if someCondition {
        return &amp;MyError{message: "something went wrong"}
    }
    return nil // Returns true nil interface
}</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Go FAQ - Why is my nil error value not equal to nil? - <a href="https://golang.org/doc/faq#nil_error">Official Go documentation explaining the nil interface gotcha</a></p>
</li>
<li>
<p>The Go Programming Language Specification - Interface types - <a href="https://golang.org/ref/spec#Interface_types">Official specification for Go interface types and their behavior</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>