This rule raises when a Scikit-learn model is instantiated without specifying all the hyperparameters. 

// If you want to factorize the description uncomment the following line and create the file.
//include::../description.adoc[]

== Why is this an issue?

When instantiating a Scikit-learn estimator, it will use default values for the hyperparameters that are not specified.
Relying on the default values can lead to non-reproducible results across diffferent versions of the library. 

Furthermore, the default values might not be the best choice for the specific problem at hand and can lead to suboptimal performance.


== How to fix it
Specify the hyperparameters when instantiating the estimator.
=== Code examples

==== Noncompliant code example

[source,python,diff-id=1,diff-type=noncompliant]
----
from sklearn.neighbors import KNeighborsClassifier

clf = KNeighborsClassifier() # Noncompliant
----

==== Compliant solution

[source,python,diff-id=1,diff-type=compliant]
----
from sklearn.neighbors import KNeighborsClassifier

clf = KNeighborsClassifier( # Compliant
    n_neighbors=5,
    weights='distance',
    algorithm='auto',
    leaf_size=30,
    p=2,
    metric='minkowski',
)
----

//=== How does this work?

//=== Pitfalls

//=== Going the extra mile

// ifdef::env-github,rspecator-view[]

(visible only on this page)

== Implementation specification 

Implementation will be quite tricky if we want to avoid false positives.

Abort if : 

- In a Pipeline/make_pipeline used for hyperparameter search

https://github.com/MarkHaakman/dslinter/blob/81b40914680c2a2108715c8fba543b4449366912/scripts/hyperparameters.py#L5-L26[List of DSLinter estimators]

Possible baby step : only check for some estimators ( for exemple the meta-learners)

Ignore parameters : 

- n_jobs

- that ends in `param` ?

=== Message 

Specify all hyperparameters when instantiating a Scikit-learn estimator.

=== Issue location

Primary : name of the estimator

No secondary location

// endif::env-github,rspecator-view[]

//== Resources
//=== Documentation
//=== Articles & blog posts
//=== Conference presentations
//=== Standards
//=== External coding guidelines
//=== Benchmarks
