== Why is this an issue?

Either use only spaces or only tabs for the indentation of a text block. Mixing white space will lead to a result with irregular indentation.


=== Noncompliant code example

[source,java]
----
String textBlock = """
        this is
<tab>text block!
        !!!!
      """;
----


=== Compliant solution

[source,java]
----
String textBlock = """
        this is
        text block!
        !!!!
      """;
----


== Resources

* https://openjdk.java.net/jeps/378[JEP 378: Text Blocks]
* https://openjdk.org/projects/amber/guides/text-blocks-guide[Programmer's Guide To Text Blocks], by Jim Laskey and Stuart Marks


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

include::highlighting.adoc[]

endif::env-github,rspecator-view[]
