Either use only spaces or only tabs for the indentation of a text block. Mixing white space will lead to a result with irregular indentation.


== Noncompliant Code Example

----
String textBlock = """
        this is 
<tab>text block!
        !!!!
      """;  
----


== Compliant Solution

----
String textBlock = """
        this is 
        text block!
        !!!!
      """;
----


== See

* https://openjdk.java.net/jeps/378[JEP 378: Text Blocks]
* https://cr.openjdk.java.net/~jlaskey/Strings/TextBlocksGuide_v9.html[Programmer's Guide To Text Blocks], by Jim Laskey and Stuart Marks


