<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>While it is possible to define a <code>struct</code> with member functions in C&#43;&#43;, the general expectation is that <code>struct</code>s only aggregate data, while classes are used for fully encapsulated abstractions, containing data and methods. Thus, including a member function in a <code>struct</code> is likely to lead to confusion at best and should be avoided.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">struct S
{
  S(int x, int y) : x(x), y(y) {}
  int x;
  int y;
  public:
    int fun(); // Noncompliant
};</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">struct S
{
  S(int x, int y) : x(x), y(y) {}
  int x;
  int y;
};</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_exceptions">Exceptions</h3>
<div class="ulist">
<ul>
<li>
<p>Constructors are ignored for this rule, because they are the only way that a <code>struct</code> designer can enforce that all fields of a <code>struct</code> are correctly initialized.</p>
</li>
<li>
<p>Overload of the function operator '()'. It is a common practice to use struct as a functor.</p>
</li>
</ul>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Promote this "struct" to a "class" or remove its member functions</p>
</div>
</div>
</div>
</div>