Because ``++RESULT_CACHE++``-enabled functions increase memory consumption, one should double-check that the gain in performances is significant, and avoid over-using this feature in general.


== Noncompliant Code Example

----
CREATE FUNCTION myFastFunction RETURN PLS_INTEGER RESULT_CACHE AS -- Noncompliant
BEGIN
  RETURN 42;
END;
/

DROP FUNCTION myFastFunction;
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

endif::env-github,rspecator-view[]
