== Why is this an issue?

Because ``++RESULT_CACHE++``-enabled functions increase memory consumption, one should double-check that the gain in performances is significant, and avoid over-using this feature in general.


=== Noncompliant code example

[source,sql]
----
CREATE FUNCTION myFastFunction RETURN PLS_INTEGER RESULT_CACHE AS -- Noncompliant
BEGIN
  RETURN 42;
END;
/

DROP FUNCTION myFastFunction;
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Remove this use of "RESULT_CACHE"


endif::env-github,rspecator-view[]
