<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Since C&#43;&#43;11, type aliases can be declared via <code>using</code> or <code>typedef</code>. <code>using</code> should be preferred as  more readable because you see the new name/alias first.</p>
</div>
<div class="paragraph">
<p>In addition, <code>using</code> can be templated, which makes it applicable to more situations than <code>typedef</code>.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">typedef void (*FunctionPointerType)(int);</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">using FunctionPointerType = void (*)(int);</code></pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p>C++ Core Guidelines - <a href="https://github.com/isocpp/CppCoreGuidelines/blob/e49158a/CppCoreGuidelines.md#t43-prefer-using-over-typedef-for-defining-aliases">T.43: Prefer <code>using</code> over <code>typedef</code> for defining aliases</a></p>
</li>
</ul>
</div>
<hr>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Use 'using' instead of 'typedef' for this type alias</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>The typedef statement</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_26_aug_2019_220326_loïc_joly_wrote">on 26 Aug 2019, 22:03:26 Loïc Joly wrote:</h3>
<div class="paragraph">
<p>Can you please review my changes</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_27_aug_2019_171611_abbas_sabra_wrote">on 27 Aug 2019, 17:16:11 Abbas Sabra wrote:</h3>
<div class="paragraph">
<p>Looks good to me</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_9_sep_2019_204309_ann_campbell_wrote">on 9 Sep 2019, 20:43:09 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~abbas.sabra], [~loic.joly] the <code>user-experience</code> tag is generally reserved for end-user experience, not the experience of API users. :-D</p>
</div>
<div class="paragraph">
<p>I suggest instead 'design' which has been used for API and overall program design.</p>
</div>
</div>
</div>
</div>