<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>In Dart type aliases can be declared via function type aliases <code>typedef void F()</code> or generic function type aliases  <code>typedef F = void Function()</code>. Generic function type aliases can be parametrized <code>typedef Compare&lt;T&gt; = int Function(T a, T b);</code>. Function type aliases miss this feature.</p>
</div>
<div class="paragraph">
<p>While it&#8217;s not always needed to have them parametrized, for the sake of readability it&#8217;s recommended to use consistent way of declaring type aliases.</p>
</div>
<div class="paragraph">
<p>Thus, generic function type aliases should be preferred.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-dart" data-lang="dart">typedef void F(int a);</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-dart" data-lang="dart">typedef F = void Function(int a)</code></pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p>Dart Docs - <a href="https://dart.dev/tools/linter-rules/prefer_generic_function_type_aliases">Dart Linter rule - prefer_generic_function_type_aliases</a></p>
</li>
<li>
<p>Dart Docs - <a href="https://dart.dev/language/typedefs">Typedefs</a></p>
</li>
</ul>
</div>
<hr>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="ulist">
<ul>
<li>
<p>Use the generic function type syntax in 'typedef&#8217;s.</p>
</li>
</ul>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>The <code>typedef</code> name: e.g. <code>F</code> in <code>typedef void F();</code></p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
</div>
</div>