include::../description.adoc[]

== Noncompliant Code Example

----
switch ($variable) {
  case 0:
    do_something();
    break;
  default:
    do_something_else();
    break;
}
----

== Compliant Solution

----
if ($variable == 0) {
  do_something();
} else {
  do_something_else();
}
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
