Jump statements (``++return++``, ``++break++`` and ``++continue++``) and ``++throw++`` expressions move control flow out of the current code block. So any statements that come after a jump are dead code.

== Noncompliant Code Example

[source,javascript]
----
function fun(a) {
  var i = 10;
  return i + a; 
  i++;             // Noncompliant; this is never executed
}
----

== Compliant Solution

[source,javascript]
----
function fun(int a) {
  var i = 10;
  return i + a;
}
----

include::../see.adoc[]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

include::../highlighting.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
