<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Using a function in PHP with the same name as the nesting class was historically used to declare a class constructor.
However, as of PHP 8.0.0, this declaration is discouraged and will provoke an <code>E_DEPRECATED</code> error, albeit it functions as a constructor.</p>
</div>
<div class="paragraph">
<p>Instead, users should explicitly define the constructor by declaring a <code>__construct(...)</code> function.
However, if both styles are present in the same class, PHP will treat the <code>__construct</code> function as the class constructor, which can cause unintended behavior.</p>
</div>
<div class="paragraph">
<p>Adhering to this convention improves readability and maintainability by ensuring that the constructor declaration is named uniformly throughout the codebase.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-php" data-lang="php">class Foo {
  function Foo() {...}
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-php" data-lang="php">class Foo {
  function __construct() {...}
}</code></pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p><a href="https://www.php.net/manual/en/language.oop5.decon.php">PHP Manual - Constructors and Destructors</a></p>
</li>
<li>
<p><a href="https://www.phptutorial.net/php-oop/php-constructors/">PHP Tutorial - Constructors</a></p>
</li>
</ul>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="ulist">
<ul>
<li>
<p>Rename this "XXXXX" function to "__construct".</p>
</li>
<li>
<p>Replace this function name "XXXXX", since a "__construct" method has already been defined in this class.</p>
</li>
</ul>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_23_feb_2014_231958_freddy_mallet_wrote">on 23 Feb 2014, 23:19:58 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>Is implemented by http://jira.codehaus.org/browse/SONARPLUGINS-3513 for PHP</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_12_oct_2015_123559_ann_campbell_wrote">on 12 Oct 2015, 12:35:59 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~linda.martin] I&#8217;ve updated your edits. Double-check me.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_13_oct_2015_101857_linda_martin_wrote">on 13 Oct 2015, 10:18:57 Linda Martin wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] Perfect thanks!</p>
</div>
</div>
</div>
</div>