== How to fix it in Java Lang Package

=== Code examples

include::../../common/fix/code-rationale.adoc[]

==== Noncompliant code example

[source,kotlin,diff-id=1,diff-type=noncompliant]
----
fun doGet(request: HttpServletRequest, response: HttpServletResponse?) {
  val r = Runtime.getRuntime()
  val userInput = request.getParameter("example")

  if (userInput != null) {
    val envs = arrayOf(userInput)
    r.exec("/path/to/example", envs)
  } else {
    r.exec("/path/to/example")
  }
}
----

==== Compliant solution

[source,kotlin,diff-id=1,diff-type=compliant]
----
fun doGet(request: HttpServletRequest, response: HttpServletResponse?) {
  val r = Runtime.getRuntime()
  val userInput = request.getParameter("example")

  if (userInput != null && userInput.matches("^[a-zA-Z0-9]*$".toRegex())) {
    val envs = arrayOf("ENV_VAR=$userInput")
    r.exec("/path/to/example", envs)
  } else {
    r.exec("/path/to/example")
  }
}
----

=== How does this work?

include::../../common/fix/pre-approved-list.adoc[] 

