{cpp}11 introduced move-operation: a more efficient way to transfer value from one object to another. This operation is used when the source object is an rvalue, i.e. a temporary or explicitly moved object.


In most situations, the move is performed if a member of an rvalue is accessed (i.e. ``++std::move(val).mem++``). However, in a case where the member itself is declared as an rvalue reference, the referenced object will not be moved. This may lead to unintended copy. This rule helps to identify such cases.


This rule raises an issue when a copy of a member declared with an rvalue reference type is made.

== Noncompliant Code Example

----
void consume(std::pair<std::string, std::string&&>&& p) {
   auto first = std::move(p).first; // Complaint: move is performed
   auto second = std::move(p).second; // Noncompliant: copy is made
}
----

== Compliant Solution

----
void consume(std::pair<std::string, std::string&&>&& p) {
   auto first = std::move(p).first; 
   auto second = std::move(p.second); 
}
// or 
void consume(std::pair<std::string, std::string&&>&& p) {
   auto first = std::get<0>(std::move(p)); 
   auto first = std::get<1>(std::move(p)); 
}
----
