This rule raises an issue when an `if` statement condition is wrapped in parentheses.

== Why is this an issue?

Go's syntax does not require parentheses around `if` statement conditions, unlike many other programming languages. The Go language specification and community style guidelines explicitly favor the cleaner syntax without parentheses.

Using unnecessary parentheses makes the code less idiomatic and can indicate that the developer is not following Go conventions. This can make the code feel foreign to Go developers and reduces consistency across the codebase.

Go's design philosophy emphasizes simplicity and readability. By removing unnecessary syntax elements like these parentheses, the code becomes cleaner and more aligned with the language's intended style.

=== What is the potential impact?

This issue has minimal impact on functionality but affects code consistency and readability. Code that doesn't follow Go idioms may be harder for Go developers to read and maintain.

== How to fix it

Remove the parentheses around the if statement condition. The condition should be written directly after the `if` keyword without wrapping parentheses.

=== Code examples

==== Noncompliant code example

[source,go,diff-id=1,diff-type=noncompliant]
----
if (x > 0) { // Noncompliant
    return "positive"
}
----

==== Compliant solution

[source,go,diff-id=1,diff-type=compliant]
----
if x > 0 {
    return "positive"
}
----

== Resources

=== Documentation

 * Go Tour - If statements - https://go.dev/tour/flowcontrol/5[Official Go tutorial explaining if statement syntax]

 * Effective Go - Control structures - https://go.dev/doc/effective_go#control-structures[Go's official style guide for control structures]
