<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Class templates can be explicitly or partially specialized. But according to the C&#43;&#43; standard, function templates cannot be partially specialized. Under certain conditions, the MicrosoftÂ® compiler will silently ignore the confusing application of partial specialization syntax to a function, but other compilers will raise an error for it and fail compilation.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">template&lt;typename T&gt;
void fun(T p);

template&lt;typename T&gt;
void fun&lt;T&gt;(T p) { // Noncompliant
  // ...
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">template&lt;typename T&gt;
void fun(T p);

template&lt;typename T&gt;
void fun(T p) {
  // ...
}</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove the template arguments from this function template.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="listingblock">
<div class="content">
<pre>template&lt;typename T&gt;
void fun&lt;T&gt;(T p) {
//   ^^^^^^    the template-id</pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_1_aug_2016_161250_tstock_wrote">on 1 Aug 2016, 16:12:50 Tstock wrote:</h3>
<div class="paragraph">
<p>Hi I found a NPE in Java plugin, but I&#8217;ve no permissioin to create an new issue. Can you help me?&#8201;&#8212;&#8201;feel free to delete this post</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_1_aug_2016_164803_ann_campbell_wrote">on 1 Aug 2016, 16:48:03 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>Hi [~tstocknm], thanks for reporting, but this isn&#8217;t the place to do it. :-)</p>
</div>
<div class="paragraph">
<p>Instead, please open a new thread on the <a href="https://groups.google.com/forum/?pli=1#!forum/sonarqube">SonarQube Google Group</a></p>
</div>
</div>
</div>
</div>