<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when a call is made to a non-callable object.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>In order to be callable, a Python class should implement the <a href="https://docs.python.org/3/reference/datamodel.html#object.__call__"><code>__call__</code></a> method. Thanks to this method, an instance of this class will be callable as a function.</p>
</div>
<div class="paragraph">
<p>However, when making a call to a non-callable object, a <code>TypeError</code> will be raised.</p>
</div>
<div class="paragraph">
<p>In order to fix this issue, make sure that the object you are trying to call has a <code>__call__</code> method.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">class MyClass:
    pass

myvar = MyClass()
myvar()  # Noncompliant

none_var = None
none_var()  # Noncompliant</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">class MyClass:
    def __call__(self):
        print("called")

myvar = MyClass()
myvar()</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p><a href="https://docs.python.org/3/reference/datamodel.html#object.__call__">Python documentation - __call__ method</a></p>
</li>
</ul>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Fix this call; XX is not callable.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>Primary: The name before the opening parenthesis</p>
</div>
<div class="ulist">
<ul>
<li>
<p>Secondary 1 (if the call is made on a variable)</p>
<div class="ulist">
<ul>
<li>
<p>location: The last value assigned.</p>
</li>
<li>
<p>message: "Assigned value."</p>
</li>
</ul>
</div>
</li>
<li>
<p>Secondary 2</p>
<div class="ulist">
<ul>
<li>
<p>location: The class definition</p>
</li>
<li>
<p>message: "Definition."</p>
</li>
</ul>
</div>
</li>
</ul>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_relates_to_s2873">relates to: <a data-rspec-id="S2873" class="rspec-auto-link">S2873</a></h3>

</div>
</div>
</div>