This is an issue when the `@TimedInterrupt` annotation is applied to a static method.

== Why is this an issue?

The `@TimedInterrupt` annotation is designed to automatically add timeout checks to methods by inserting interruption checks at strategic points in the code. However, this transformation is currently not compatible with static methods due to implementation limitations in how the transformation processes method contexts.

When `@TimedInterrupt` is applied to a static method, it can lead to unexpected behavior or compilation issues because the transformation logic assumes instance method semantics. The annotation works by injecting timeout verification code that may not function correctly in a static context, potentially causing the timeout mechanism to fail silently or throw unexpected exceptions.

This incompatibility can result in unreliable timeout behavior, making it difficult to control long-running static operations as intended.

=== What is the potential impact?

Using `@TimedInterrupt` on static methods can lead to:

* Unreliable timeout behavior where the intended time limits are not enforced
* Potential compilation errors or runtime exceptions
* Silent failures where the timeout mechanism doesn't work as expected
* Difficulty in controlling long-running static operations
* Unpredictable application behavior when timeouts should have been triggered

== How to fix it

Convert the static method to an instance method.

=== Code examples

==== Noncompliant code example

[source,groovy,diff-id=1,diff-type=noncompliant]
----
class MyClass {
    @TimedInterrupt(value=1, unit=TimeUnit.SECONDS)
    static void staticMethod() { // Noncompliant
        // Long-running operation
    }
}
----

==== Compliant solution

[source,groovy,diff-id=1,diff-type=compliant]
----
class MyClass {
    @TimedInterrupt(value=1, unit=TimeUnit.SECONDS)
    void instanceMethod() {
        // Long-running operation
    }
}
----

== Resources

=== Documentation

 * Groovy documentation - https://groovy-lang.org/metaprogramming.html#xform-TimedInterrupt[TimedInterrupt]
