== Why is this an issue?

This rule allows banning usage of certain statement parameters.

== How to fix it

=== Code examples

==== Noncompliant code example

Given the parameters:

* statementName: INCLUDE
* parameterName: MEMBER
* allowedParameterValues: HELLO,BYE
* nameField: OUTPUT1

[source,jcl,diff-id=1,diff-type=noncompliant]
----
//OUTPUT1 INCLUDE MEMBER=FOOBAR
----

==== Compliant solution

[source,jcl,diff-id=1,diff-type=compliant]
----
//OUTPUT1 INCLUDE MEMBER=HELLO
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Parameters

.operationName
****

Name of the operation in statements where parameters should be checked.
****

.parameterName
****

Name of the parameter whose value should be checked.
****

.allowedParameterValues
****

Comma-separated list of valid values for the given parameter, e.g. SYSOUT1, SYSOUT2.
****

.nameField
****

The name field value of the statement to be checked, e.g. JOBLIB, STEPLIB.
This parameter is optional. If absent, the name field of the statement will not be taken into account.

****

'''

endif::env-github,rspecator-view[]
