<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p><code>#pragma once</code> is a preprocessor directive meant to ensure that a file is only included once in a compilation unit. However, there are several reasons to avoid it:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>It is not part of the standard, which prevents its use in some contexts.</p>
</li>
<li>
<p>Even if it is supported by virtually all compilers, since its behavior is not defined, it may differ between compilers, especially for some corner cases when determining if two files are identical (for instance, in the presence of symbolic links).</p>
</li>
<li>
<p>Its semantic is slightly different from the semantic of an include guard. For instance, if a file is duplicated in two different locations, <code>#pragma once</code> will not prevent multiple inclusion of this file.</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>Note: There used to be a build performance improvement when using <code>#pragma once</code> instead of an include guard because naive implementations of include guards need to parse the full file to get the <code>#endif</code> matching the <code>#if</code>. But most modern compilers specifically detect the include guard pattern and use a dedicated optimization that makes it about as fast as <code>#pragma once</code>.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">// my_header.h
#pragma once
...</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">// my_header.h
#ifndef MY_HEADER
#define MY_HEADER
...
#endif</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove this use of "#pragma once" and use include guards instead</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_30_jul_2014_202537_freddy_mallet_wrote">on 30 Jul 2014, 20:25:37 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>If my feeling is correct :</p>
</div>
<div class="ulist">
<ul>
<li>
<p>The purpose of this rule is mainly to prevent a portability issue from one compiler to another or from one version of the language to another. So I would use the "Portability" category.</p>
</li>
</ul>
</div>
</div>
</div>
</div>