<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Unfortunately, it is possible to make constructor calls recursive. When this happens, you get a class that cannot be instantiated.</p>
</div>
<div class="paragraph">
<p>As a general rule, no constructor should make a call to another constructor in the same class that requires fewer arguments than the calling constructor received. I.e. the constructor that accepts the most arguments is the one that has the fullest picture of how the class should look. It should perform class initialization.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-csharp" data-lang="csharp">class Foo
{
  int start;

  Foo() : this(0)  { }
  Foo(int v) : this()   { }  // Noncompliant
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-csharp" data-lang="csharp">class Foo
{
  int start;

  Foo() : this(0)  { }
  Foo(int v)
  {
    this.count = v;
  }
}</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_8_jun_2015_083932_tamas_vajk_wrote">on 8 Jun 2015, 08:39:32 Tamas Vajk wrote:</h3>
<div class="paragraph">
<p>LGTM, I&#8217;ve only added a "that" in the description. Can you validate that we actually need it. Thanks.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_8_jun_2015_115933_ann_campbell_wrote">on 8 Jun 2015, 11:59:33 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>Looks fine [~tamas.vajk]. Thanks</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_15_jun_2015_065950_tamas_vajk_wrote">on 15 Jun 2015, 06:59:50 Tamas Vajk wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] The default severity was missing, I&#8217;ve added it.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_15_jun_2015_093646_tamas_vajk_wrote">on 15 Jun 2015, 09:36:46 Tamas Vajk wrote:</h3>
<div class="paragraph">
<p>This rule has been added to the C# compiler. Its error ID is CS0768.</p>
</div>
</div>
</div>
</div>