As stated per effective java : 

____
Varargs methods are a convenient way to define methods that require a variable number of arguments, but they should not be overused. They can produce confusing results if used inappropriately.

____

== Noncompliant Code Example

[source,java]
----
void fun ( String... strings )	// Noncompliant
{
  // ...
}
----
