include::../description.adoc[]

== Noncompliant Code Example

----
class Foo {
  public int myField;

  public void doSomething() {
    int myField = 0;
    ...
  }
}
----

include::../see.adoc[]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../highlighting.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
