include::../description.adoc[]

== Noncompliant Code Example

----
function process(var a) {
  var BAD_OPS = 3;
  if( a = BAD_OPS) {...}
}
----

----
function process(var a) {
  var BAD_OPS = "UNKNOWN";
  if( a = BAD_OPS) {...}
}
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../highlighting.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
