== Why is this an issue?

Explicitly specifying a ``++void++`` parameter list is required in C, but optional in {cpp}. Using ``++void++`` for a parameter-less function decreases its readability. The at-a-glance impression is that the function _does_ take a parameter, and it takes a second look to ascertain that it does not. Therefore the more compact notation is preferred.


=== Noncompliant code example

[source,text]
----
int fun(void);

int fun(void) {
  ...
}
----


=== Compliant solution

[source,text]
----
int fun();

int fun() {
  ...
}
----


== Resources

* {ccg_NL_25}

