== Why is this an issue?

Where a data value is to be tested against zero then the test should be made explicit. The exception to this rule is when data represents a Boolean value, even though in C this will in practice be an integer.


This rule is in the interests of clarity, and makes clear the distinction between integers and logical values.


=== Noncompliant code example

[source,cpp]
----
if ( x ) // Noncompliant, unless x is effectively Boolean data
----


=== Compliant solution

[source,cpp]
----
if ( x == 0) // Compliant solution
----


== Resources

* MISRA C:2004, 13.2 - Tests of a value against zero should be made explicit, unless the operand is effectively Boolean.


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Explicitly test this expression against zero.


'''
== Comments And Links
(visible only on this page)

=== is related to: S872

=== on 22 Jan 2019, 11:45:46 Lo√Øc Joly wrote:
This rule is a pure Misra-C rule, and should be retargeted as such. RSPEC-811 & RSPEC-812, together, cover the same situation for Misra-{cpp}.


We need a SonarSource rule that corers the same situation, but with some differences: 

* One rule for both languages, that covers all cases
* Allow also pointer validity checks ``++T* p = f(); if(p) ...++``?
* Allow class types with explicit conversion to bool ``++unique_ptr<T> p = f(); if (p) ...++``?
* ...

We will wait to have started a real Misra initiative to do this.

endif::env-github,rspecator-view[]
