<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Nullability of types is an important part of Dart&#8217;s type system. In order to simplify the work with nullable types in the Dart language there are many null-aware operators (<code>??</code>, <code>??=</code>, <code>?.</code>, <code>!</code>, <code>?..</code>, <code>?[]</code>, <code>...?</code>). Using those operators will make your code more clear and concise.</p>
</div>
<div class="paragraph">
<p>The If-null operator (<code>??</code>), will check if the value on the left side is <code>null</code>, and, if yes, will return the expression on the right. This is an easy way to set default values instead of dealing with <code>null</code>. Thus, this operator doesn&#8217;t make sense if used with <code>null</code> on either sides.</p>
</div>
<div class="paragraph">
<p>For example, in <code>null ?? x</code> the <code>x</code> will always be returned. Using the operator here only introduces confusion and additional cognitive load. In case of <code>x ?? null</code>, again the result will always be the value of <code>x</code>. In both cases the operator can be replaced with just <code>x</code>.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Depending on your context, remove the <code>null ??</code> / <code>null ??</code> part, or replace <code>null</code> with some meaningful value.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-dart" data-lang="dart">void f(String? s) {
  var myString1 = s ?? null;
  var myString2 = null ?? s;
  var myString3 = s ?? null;
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-dart" data-lang="dart">void f(String? s) {
  var my_string1 = s;
  var my_string2 = s;
  var my_string3 = s ?? "default";
}</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Dart Docs - <a href="https://dart.dev/tools/linter-rules/unnecessary_null_in_if_null_operators">Dart Linter rule - unnecessary_null_in_if_null_operators</a></p>
</li>
</ul>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Unnecessary use of '??' with 'null'.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>Only <code>null</code> keyword</p>
</div>
</div>
</div>
</div>