<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule is a more precise version of <a data-rspec-id="S1308" class="rspec-auto-link">S1308</a> preventing the use of <code>GO TO</code>. The flow of a program is already complicated to understand with simple <code>GO TO</code>s. It&#8217;s even worse when a <code>GO TO</code> is executed conditionally like this is the case with <code>GO TO DEPENDING ON</code>.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cobol" data-lang="cobol">PROCEDURE DIVISION.
   ...
   GO TO PARAGRAPH-10
         PARAGRAPH-20
         PARAGRAPH-30
   DEPENDING ON WS-PARA-NUMBER *&gt; Noncompliant
   ...</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cobol" data-lang="cobol">PROCEDURE DIVISION.
   ...
   EVALUATE WS-PARA-NUMBER
      WHEN 1
         PERFORM PARAGRAPH-10
      WHEN 2
         PERFORM PARAGRAPH-20
      WHEN 3
         PERFORM PARAGRAPH-30
      WHEN OTHER
         PERFORM PARAGRAPH-99
   END-EVALUATE
   ...</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Refactor this code to remove the GO TO DEPENDING ON statement.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_relates_to_s1308">relates to: <a data-rspec-id="S1308" class="rspec-auto-link">S1308</a></h3>

</div>
</div>
</div>