<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>When the code under test in a unit test throws an exception, the test itself fails. Therefore, there is no need to surround the tested code with a <code>try</code>-<code>catch</code> structure to detect failure. Instead, you can simply move the exception type to the method signature.</p>
</div>
<div class="paragraph">
<p>This rule raises an issue when there is a fail assertion inside a <code>catch</code> block.</p>
</div>
<div class="paragraph">
<p>Supported frameworks:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>JUnit3</p>
</li>
<li>
<p>JUnit4</p>
</li>
<li>
<p>JUnit5</p>
</li>
<li>
<p>Fest assert</p>
</li>
<li>
<p>AssertJ</p>
</li>
</ul>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">@Test
public void testMethod() {
  try {
            // Some code
  } catch (MyException e) {
    Assert.fail(e.getMessage());  // Noncompliant
  }
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">@Test
public void testMethod() throws MyException {
    // Some code
}</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove this failure assertion and simply add the exception type to the method signature.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="ulist">
<ul>
<li>
<p>Primary: <code>Assert.fail</code></p>
</li>
<li>
<p>Secondary: <code>try</code> and <code>catch (...)</code></p>
</li>
</ul>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_is_related_to_s3431">is related to: <a data-rspec-id="S3431" class="rspec-auto-link">S3431</a></h3>

</div>
<div class="sect2">
<h3 id="_on_27_jun_2016_225357_alix_lourme_wrote">on 27 Jun 2016, 22:53:57 Alix Lourme wrote:</h3>
<div class="paragraph">
<p>@[~ann.campbell.2] : I will update the <a href="https://github.com/SonarSource/sonar-java/pull/902">PR</a> with your last modifications ASAP (I need update an <a href="https://github.com/SonarSource/orchestrator/pull/1">orchestrator PR</a> before) ; but IMO a remediation cost to 1min (perhaps 2) would be better than 5 (this is a default very quick to fix).</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_29_jun_2016_213035_ann_campbell_wrote">on 29 Jun 2016, 21:30:35 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>Just now saw this [~axel3rd]. Remediation updated to 2min.</p>
</div>
</div>
</div>
</div>