<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an error when the <code>dtype</code> parameter is not provided when using <code>pandas.read_csv</code> or <code>pandas.read_table</code>.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>The pandas library provides an easy way to load data from documents hosted locally or remotely, for example with the <code>pandas.read_csv</code> or <code>pandas.read_table</code> functions:</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">import pandas as pd

df = pd.read_csv("my_file.csv")</code></pre>
</div>
</div>
<div class="paragraph">
<p>Pandas will infer the type of each columns of the CSV file and specify the datatype accordingly, making this code perfectly valid.
However this snippet of code does not convey the proper intent of the user, and can raise questions such as:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>What information can I access in <code>df</code>?</p>
</li>
<li>
<p>What are the names of the columns available in <code>df</code>?</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>These questions arise as there are no descriptions of what kind of data is loaded into the data frame, making the code less understandable and harder to maintain.</p>
</div>
<div class="paragraph">
<p>A straightforward way to fix these issues is by providing the schema of the data through the usage of the <code>dtype</code> parameter.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>To fix this issue provide the <code>dtype</code> parameter to the <code>read_csv</code> or <code>read_table</code> function.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">import pandas as pd

def foo():
  return pd.read_csv("my_file.csv") # Noncompliant: it is unclear which type of data the data frame holds.</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">import pandas as pd

def foo():
  return pd.read_csv(
          "my_file.csv",
          dtype={'name': 'str', 'age': 'int'}) # Compliant</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Pandas Documentation - <a href="https://pandas.pydata.org/docs/reference/api/pandas.read_csv.html#pandas-read-csv">pandas.read_csv</a></p>
</li>
<li>
<p>Pandas Documentation - <a href="https://pandas.pydata.org/docs/reference/api/pandas.read_table.html#pandas-read-table">pandas.read_table</a></p>
</li>
<li>
<p>Pandas Documentation - <a href="https://pandas.pydata.org/docs/user_guide/basics.html#dtypes">dtypes</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>