include::../description.adoc[]

include::../why.adoc[]

== How to fix it

=== Code examples

==== Noncompliant code example

[source,shell,diff-id=1,diff-type=noncompliant]
----
#!/bin/bash
apt-get install ca-certificates
aptitude install ca-certificates
apt install ca-certificates
gdebi /tmp/package.deb
----

include::../non-compliant-desc.adoc[]

==== Compliant solution

[source,shell,diff-id=1,diff-type=compliant]
----
#!/bin/bash
apt-get -y install ca-certificates
aptitude -y install ca-certificates
apt -y install ca-certificates
gdebi --n /tmp/package.deb
----

include::../compliant-desc.adoc[]

include::../how.adoc[]

include::../exceptions.adoc[]

== Resources
=== Documentation

include::../general-links.adoc[]

ifdef::env-github,rspecator-view[]
'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../highlighting.adoc[]

'''
endif::env-github,rspecator-view[]
