<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>When a single primitive is tested against three or more values in an <code>if</code>/<code>else if</code> structure, it should be converted to a switch instead for greater readability.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">public void doSomething(int i) {
  if (i == 0) {
    // ...
  }
  else if (i == 1) {
    // ...
  }
  else if (i == 2) {
    // ...
  }
  else if (i == 3) {
    // ...
  }
  else {
    // ...
  }</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">public void doSomething(int i) {
  switch(i) {
    case 0:
      // ...
      break;
    case 1:
      // ...
      break;
    case 2:
      // ...
      break;
    case 3:
      // ...
      break;
    default:
      // ...
      break;
  }</code></pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p>C&#43;&#43; Core Guidelines - <a href="https://github.com/isocpp/CppCoreGuidelines/blob/e49158a/CppCoreGuidelines.md#es70-prefer-a-switch-statement-to-an-if-statement-when-there-is-a-choice">ES.70: Prefer a <code>switch</code>-statement to an <code>if</code>-statement when there is a choice</a></p>
</li>
</ul>
</div>
<hr>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Convert this "if/else if" structure into a "switch".</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_is_duplicated_by_s1073">is duplicated by: <a data-rspec-id="S1073" class="rspec-auto-link">S1073</a></h3>

</div>
<div class="sect2">
<h3 id="_relates_to_s2196">relates to: <a data-rspec-id="S2196" class="rspec-auto-link">S2196</a></h3>

</div>
<div class="sect2">
<h3 id="_on_14_oct_2014_133337_ann_campbell_wrote">on 14 Oct 2014, 13:33:37 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~nicolas.peru] my assumption is that when the analysis target or runtime &lt;= Java 6, this rule will ignore strings. Doable?</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_22_oct_2014_190206_nicolas_peru_wrote">on 22 Oct 2014, 19:02:06 Nicolas Peru wrote:</h3>
<div class="paragraph">
<p>This actually not doable nor planned in fact. Would probably require to split into two rules, one about string with tag java7 and the other one on integers with no particular tags.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_7_apr_2015_225602_evgeny_mandrikov_wrote">on 7 Apr 2015, 22:56:02 Evgeny Mandrikov wrote:</h3>
<div class="paragraph">
<p>\[~nicolas.peru] actually Java analyzer should already know both source and target Java versions (since a long time), so that this knowledge can be utilized in rules.</p>
</div>
</div>
</div>
</div>