<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>When a single primitive is tested against three or more values in an <code>IF</code>, <code>ELSIF</code> chain, it should be converted to a <code>CASE</code> instead for greater readability.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-sql" data-lang="sql">DECLARE
  x PLS_INTEGER := 0;
BEGIN
  IF x = 0 THEN                     -- Noncompliant
    DBMS_OUTPUT.PUT_LINE('x = 0');
  ELSIF x = 1 THEN
    DBMS_OUTPUT.PUT_LINE('x = 1');
  ELSE
    DBMS_OUTPUT.PUT_LINE('x &gt; 1');
  END IF;
END;
/</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-sql" data-lang="sql">DECLARE
  x PLS_INTEGER := 0;
BEGIN
  CASE x
    WHEN 0 THEN
      DBMS_OUTPUT.PUT_LINE('x = 0');
    WHEN 1 THEN
      DBMS_OUTPUT.PUT_LINE('x = 1');
    ELSE
      DBMS_OUTPUT.PUT_LINE('x &gt; 1');
  END CASE;
END;
/</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Convert this "IF/ELSIF" structure into a "CASE".</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_is_duplicated_by_s1073">is duplicated by: <a data-rspec-id="S1073" class="rspec-auto-link">S1073</a></h3>

</div>
<div class="sect2">
<h3 id="_relates_to_s2196">relates to: <a data-rspec-id="S2196" class="rspec-auto-link">S2196</a></h3>

</div>
<div class="sect2">
<h3 id="_on_14_oct_2014_133337_ann_campbell_wrote">on 14 Oct 2014, 13:33:37 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~nicolas.peru] my assumption is that when the analysis target or runtime &lt;= Java 6, this rule will ignore strings. Doable?</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_22_oct_2014_190206_nicolas_peru_wrote">on 22 Oct 2014, 19:02:06 Nicolas Peru wrote:</h3>
<div class="paragraph">
<p>This actually not doable nor planned in fact. Would probably require to split into two rules, one about string with tag java7 and the other one on integers with no particular tags.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_7_apr_2015_225602_evgeny_mandrikov_wrote">on 7 Apr 2015, 22:56:02 Evgeny Mandrikov wrote:</h3>
<div class="paragraph">
<p>\[~nicolas.peru] actually Java analyzer should already know both source and target Java versions (since a long time), so that this knowledge can be utilized in rules.</p>
</div>
</div>
</div>
</div>