== Why is this an issue?

Because the value in a variable of an unsigned type can never be less than zero, testing to see if it is negative is a useless operation which can only confuse future readers of the code.


=== Noncompliant code example

[source,cpp]
----
unsigned int i = 0; // the lowest value this var can have
...
if (i >= 0) { // Noncompliant
  do_x(i);
}
----


=== Compliant solution

[source,cpp]
----
unsigned int i = 0;
...
do_x(i);
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Remove this test of "XX". It is by definition >= 0.


'''
== Comments And Links
(visible only on this page)

=== on 3 Jul 2015, 09:32:39 Massimo PALADIN wrote:
\[~ann.campbell.2] shall we add references which appear in RSPEC-2583?

=== on 6 Jul 2015, 14:51:23 Ann Campbell wrote:
assigning to [~evgeny.mandrikov] pending outcome of merger into RSPEC-2583

=== on 27 Aug 2015, 12:58:14 Evgeny Mandrikov wrote:
\[~ann.campbell.2] with [~freddy.mallet] and [~massimo.paladin] we decided to go ahead without merge for now

endif::env-github,rspecator-view[]
