Because the value in a variable of an unsigned type can never be less than zero, testing to see if it is negative is a useless operation which can only confuse future readers of the code.


== Noncompliant Code Example

----
unsigned int i = 0; // the lowest value this var can have
...
if (i >= 0) { // Noncompliant
  do_x(i);
}
----


== Compliant Solution

----
unsigned int i = 0;
...
do_x(i);
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
