The use of ``++strlen++`` to determine the length of a string to which you are trying to append a null character is an anti-pattern. ``++strlen++`` requires as input an already null-terminated string;Â the result of passing a non-null-terminated string as an input to ``++strlen++`` is undefined. It may even lead to a memory access violation.


== Noncompliant Code Example

----
void f() {
  char *myStr;
  // ...
  myStr[strlen(myStr)] = '\0'; // Noncompliant, strlen requires an already null-terminated string
}
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
