The `RegExp` constructor is used to create a regular expression object for matching text with a pattern. When the pattern is a literal, a regular expression literal can be used instead of the constructor function.

== Noncompliant Code Example

----
new RegExp(/foo/);
new RegExp('bar');
new RegExp('baz', 'i');
new RegExp("\\d+");
new RegExp(`qux|quuz`);
new RegExp(`Dear ${title},`);
----

== Compliant Solution

----
/foo/;
/bar/;
/baz/i;
/\d+/;
/qux|quuz/;
new RegExp(`Dear ${title},`);
----
