<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>In general, the clause <code>INDEXED BY</code> should be used whenever possible when handling COBOL tables. If it&#8217;s not possible, then avoid using a numeric display variable to access the table&#8217;s elements. Instead, use a <code>BINARY</code>/<code>COMP</code> variable, which the processor can handle more efficiently.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cobol" data-lang="cobol">       01 SUBS PIC 9(5).
       01 INVENTORY-RECORD.
          05 Field-A PIC X OCCURS 10000 TIMES.
       ...
       PERFORM VARYING SUBS FROM 1 BY 1 UNTIL SUBS &gt; 10000
         MOVE ITEM1 TO Field-A (SUBS)      *&gt; Noncompliant
       END-PERFORM.</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cobol" data-lang="cobol">       01 SUBS PIC 9(5) COMP.
       01 INVENTORY-RECORD.
          05 Field-A PIC X OCCURS 10000 TIMES.
       ...
       PERFORM VARYING SUBS FROM 1 BY 1 UNTIL SUBS &gt; 10000
         MOVE ITEM1 TO Field-A (SUBS)
       END-PERFORM.</code></pre>
</div>
</div>
<div class="paragraph">
<p>or</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cobol" data-lang="cobol">       01 INVENTORY-RECORD.
          05 Field-A PIC X OCCURS 10000 TIMES INDEXED BY IDX1.
       ...
       PERFORM VARYING IDX1 FROM 1 BY 1 UNTIL IDX1 &gt; 10000
         MOVE ITEM1 TO Field-A (IDX1)
       END-PERFORM.</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Make "xxx" a binary type or use a different variable.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>The table subscript, i.e. <code>(SUBS)</code> in</p>
</div>
<div class="listingblock">
<div class="content">
<pre>MOVE F337-CODE2-SPE (SUBS) TO ...</pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_8_nov_2016_094302_pierre_yves_nicolas_wrote">on 8 Nov 2016, 09:43:02 Pierre-Yves Nicolas wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2], I believe that the compliant solution should include the definition of <code>Row-IDX</code> the same way the non-compliant example includes the definition of <code>SUBS</code>. In fact, I think that this is the only thing which should change between the 2 code samples (similarly as the examples for <a data-rspec-id="S3671" class="rspec-auto-link">RSPEC-3671</a>).</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_8_nov_2016_213634_ann_campbell_wrote">on 8 Nov 2016, 21:36:34 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~pierre-yves.nicolas] I&#8217;ve added the definition of <code>Row-IDX</code> but not otherwise changed the Compliant Solution; both code samples came directly from the original rule requesters.</p>
</div>
<div class="paragraph">
<p>Also, I&#8217;m wondering if the loop format changes based on what you can do with a <code>PIC 99</code> vs a <code>PIC S9(n) BINARY</code></p>
</div>
</div>
<div class="sect2">
<h3 id="_on_21_nov_2016_094854_pierre_yves_nicolas_wrote">on 21 Nov 2016, 09:48:54 Pierre-Yves Nicolas wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] Sorry for replying late. In fact, indexes are usually not declared as data items (see <a href="http://www.ibm.com/support/knowledgecenter/SS6SG3_6.1.0/com.ibm.cobol61.ent.doc/PGandLR/tasks/tptbl12.html">some IBM documentation</a>).</p>
</div>
<div class="paragraph">
<p>I suggest the following changes:</p>
</div>
<div class="paragraph">
<p>Noncompliant Code Example</p>
</div>
<div class="listingblock">
<div class="content">
<pre>       01 SUBS PIC 9(5).
       01 INVENTORY-RECORD.
          05 Field-A PIC X OCCURS 10000 TIMES.
       ...
       PERFORM VARYING SUBS FROM 1 BY 1 UNTIL SUBS &gt; 10000
         MOVE ITEM1 TO Field-A (SUBS)      *&gt; Noncompliant
       END-PERFORM.</pre>
</div>
</div>
<div class="paragraph">
<p>Compliant Solution without <code>INDEXED BY</code></p>
</div>
<div class="listingblock">
<div class="content">
<pre>       01 SUBS PIC 9(5) COMP.
       01 INVENTORY-RECORD.
          05 Field-A PIC X OCCURS 10000 TIMES.
       ...
       PERFORM VARYING SUBS FROM 1 BY 1 UNTIL SUBS &gt; 10000
         MOVE ITEM1 TO Field-A (SUBS)
       END-PERFORM.</pre>
</div>
</div>
<div class="paragraph">
<p>Compliant Solution with <code>INDEXED BY</code></p>
</div>
<div class="listingblock">
<div class="content">
<pre>       01 INVENTORY-RECORD.
          05 Field-A PIC X OCCURS 10000 TIMES INDEXED BY IDX1.
       ...
       PERFORM VARYING IDX1 FROM 1 BY 1 UNTIL IDX1 &gt; 10000
         MOVE ITEM1 TO Field-A (IDX1)
       END-PERFORM.</pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_on_21_nov_2016_161655_ann_campbell_wrote">on 21 Nov 2016, 16:16:55 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>Updated [~pierre-yves.nicolas]</p>
</div>
</div>
</div>
</div>