== Why is this an issue?

Not every data type supports the ``++RANGE++`` or scale constraints. Using these constraints on incompatible types results in an ``++PLS-00572: improper constraint form used++`` exception being raised.


=== Noncompliant code example

[source,sql]
----
DECLARE
  foo INTEGER RANGE 0 .. 42; -- Non-Compliant - raises PLS-00572 as NUMBER does not support the RANGE constraint
BEGIN
  NULL;
END;
/
----


=== Compliant solution

[source,sql]
----
DECLARE
  foo INTEGER; -- Compliant
BEGIN
  NULL;
END;
/
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

endif::env-github,rspecator-view[]
