<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Testing <code>%PARMS</code> before the use of one or more parameters is a clear indication in the code that at least some parameters are treated as optional. However, failure to actually mark those parameters as such with <code>options( *nopass )</code> denies callers the ability to treat them as optional. Since all parameters after the first one explicitly declared <code>options( *nopass )</code> are also implicitly optional, this marking is only required once. However, for clarity you should consider explicitly marking all optional parameters.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-rpg" data-lang="rpg">       DCL-PR VIOLATION1 EXTPGM;  // Noncompliant
         LASTNAME Char(10);
         FIRSTNAME Char(10);
       END-PR;

       DCL-PI VIOLATION1;
         LASTNAME Char(10);
         FIRSTNAME Char(10);
       END-PI;

       dsply FIRSTNAME;
       IF %PARMS &gt; 1;
         dsply LASTNAME;
       EndIf;
       *inlr = *on;</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-rpg" data-lang="rpg">       DCL-PR VIOLATION1 EXTPGM;
         LASTNAME Char(10);
         FIRSTNAME Char(10)  options( *nopass );
       END-PR;

       DCL-PI VIOLATION1;
         LASTNAME Char(10);
         FIRSTNAME Char(10)  options( *nopass );
       END-PI;

       dsply FIRSTNAME;
       IF %PARMS &gt; 1;
         dsply LASTNAME;
       EndIf;
       *inlr = *on;</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Mark optional parameters as such.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>procedure name</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_28_oct_2015_195828_ann_campbell_wrote">on 28 Oct 2015, 19:58:28 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>Source: User suggestion with background from http://www.itjungle.com/fhg/fhg010505-story02.html</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_29_oct_2015_135515_pierre_yves_nicolas_wrote">on 29 Oct 2015, 13:55:15 Pierre-Yves Nicolas wrote:</h3>
<div class="paragraph">
<p>Looks good.</p>
</div>
</div>
</div>
</div>