include::../description.adoc[]

== Noncompliant Code Example

----
<?xml version="1.0" encoding="utf-8"?>
<manifest xmlns:android="http://schemas.android.com/apk/res/android"
    package="com.organization.app">

    <permission
        android:name="android.permission.MYPERMISSION" /> <!-- Sensitive -->

</manifest>
----

== Compliant Solution

----
<?xml version="1.0" encoding="utf-8"?>
<manifest xmlns:android="http://schemas.android.com/apk/res/android"
    package="com.organization.app">

    <permission
        android:name="com.organization.app.permission.MYPERMISSION" />

</manifest> 
----

=== Message

Use a different namespace for the 'xxx' permission.

include::../see.adoc[]
