== Why is this an issue?

include::../description.adoc[]

== How to fix it

=== Code examples

==== Noncompliant code example

[source,java,diff-id=1,diff-type=noncompliant]
----
public void doSomething(String [] strings) {
  for (int i = 0; i < strings.length; i--) { // Noncompliant
    String string = strings[i];  // ArrayIndexOutOfBoundsException when i reaches -1
    //...
  }
----

==== Compliant solution

[source,java,diff-id=1,diff-type=compliant]
----
public void doSomething(String [] strings) {
  for (int i = 0; i < strings.length; i++) {
    String string = strings[i];
    //...
  }
----

== Resources

=== Documentation

* https://wiki.sei.cmu.edu/confluence/x/aTdGBQ[CERT, MSC54-J.] - Avoid inadvertent wrapping of loop counters

include::../rspecator.adoc[]
