Program/file names offer only very limited space for indicating program function, which is why you should take advantage of the ability to specify a program ``++TITLE++``. Omitting the ``++TITLE++`` statement will result in a default, uncommunicative ``++TITLE++`` value being printed at the top of each page of the source listing. Instead, you should write an expressive title that gives a clear impression of the program's function. 


This rule raises an issue when a there is no ``++TITLE++`` before the ``++IDENTIFICATION DIVISION++``. Ideally, ``++TITLE++`` will be the first line of a program, but it cannot be placed before compiler options.


== Noncompliant Code Example

----
      * Copyright (c) 2012 MyCo. All rights reserved.  *> Noncompliant
       IDENTIFICATION DIVISION.
----


== Compliant Solution

----
       TITLE 'IntradayPosition.ExecutePostProcessing'
      * Copyright (c) 2012 MyCo. All rights reserved. 
       IDENTIFICATION DIVISION.
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

include::highlighting.adoc[]

endif::env-github,rspecator-view[]
