<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Any extensible class might have subclasses located in a different package. When that happens, the use of <code>this.getClass().getResource</code> with a relative path would mean that the resource isn&#8217;t found for the child class.</p>
</div>
<div class="paragraph">
<p>Instead, use an absolute path or make the class <code>final</code>.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">public class MyClass {

private URL url = null;

  public MyClass(){
    this.url = this.getClass().getResource("file.txt");  // Noncompliant
  }</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Make this class final, or use an absolute path.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_17_jun_2015_151739_nicolas_peru_wrote">on 17 Jun 2015, 15:17:39 Nicolas Peru wrote:</h3>
<div class="paragraph">
<p>Looks good.</p>
</div>
</div>
</div>
</div>