== Why is this an issue?

Shared coding conventions allow teams to collaborate efficiently. This rule checks that all non-private ``++Const++`` field names comply with the provided regular expression.


The default configuration is the one recommended by Microsoft:

* Pascal casing, starting with an upper case character, e.g. BackColor
* Short abbreviations of 2 letters can be capitalized, e.g. GetID
* Longer abbreviations need to be lower cased, e.g. GetHtml


=== Noncompliant code example

With the default regular expression ``++^([A-Z]{1,3}[a-z0-9]+)*([A-Z]{2})?$++``:

[source,vbnet]
----
Module Module1
    Public Const foo = 0  ' Noncompliant
End Module
----


=== Compliant solution

[source,vbnet]
----
Module Module1
    Public Const Foo = 0  ' Compliant
End Module
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

include::parameters.adoc[]

endif::env-github,rspecator-view[]
