<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>The Java Persistence API specification imposes only a conditional requirement that <code>@Entity</code> classes be <code>Serializable</code>:</p>
</div>
<div class="quoteblock">
<blockquote>
<div class="paragraph">
<p>If an entity instance is to be passed by value as a detached object (e.g., through a remote interface), the entity class must implement the <code>Serializable</code> interface.</p>
</div>
</blockquote>
</div>
<div class="paragraph">
<p>But it&#8217;s best practice to make all such classes <code>Serializable</code> from the start. So this rule raises an issue when an <code>@Entity</code> does not implement <code>Serializable</code>.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">@Entity
pubic class Person {  // Noncompliant
  private String fname;
  // ...</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">@Entity
pubic class Person implements Serializable {
  private String fname;
  // ...</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Make this class "Serialiazable".</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>class name</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_1_dec_2015_111417_michael_gumowski_wrote">on 1 Dec 2015, 11:14:17 Michael Gumowski wrote:</h3>
<div class="paragraph">
<p>LGTM!</p>
</div>
</div>
</div>
</div>