== Why is this an issue?

If wrapped primitive values (e.g. ``++Integers++`` and ``++Floats++``) are used in a ternary operator (e.g. ``++a?b:c++``), both values will be unboxed and coerced to a common type, potentially leading to unexpected results. To avoid this, add an explicit cast to a compatible type.


=== Noncompliant code example

[source,java]
----
Integer i = 123456789;
Float f = 1.0f;
Number n = condition ? i : f;  // Noncompliant; i is coerced to float. n = 1.23456792E8
----


=== Compliant solution

[source,java]
----
Integer i = 123456789;
Float f = 1.0f;
Number n = condition ? (Number) i : f;  // n = 123456789
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Add an explicit cast to match types of operands.


'''
== Comments And Links
(visible only on this page)

=== on 16 Feb 2015, 18:02:37 Michael Gumowski wrote:
Message changed, as the type of the ternary operation can not be resolved at the moment. Indicating the required cast is not yet possible. 

Moreover, other expressions than variables can be used as operand.

endif::env-github,rspecator-view[]
