<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule allows banning usage of certain statement parameters.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="paragraph">
<p>Given the parameters:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>statementName: INCLUDE</p>
</li>
<li>
<p>parameterName: MEMBER</p>
</li>
<li>
<p>allowedParameterValues: HELLO,BYE</p>
</li>
<li>
<p>nameField: OUTPUT1</p>
</li>
</ul>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-jcl" data-lang="jcl">//OUTPUT1 INCLUDE MEMBER=FOOBAR</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-jcl" data-lang="jcl">//OUTPUT1 INCLUDE MEMBER=HELLO</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_parameters">Parameters</h3>
<div class="sidebarblock">
<div class="content">
<div class="title">operationName</div>
<div class="paragraph">
<p>Name of the operation in statements where parameters should be checked.</p>
</div>
</div>
</div>
<div class="sidebarblock">
<div class="content">
<div class="title">parameterName</div>
<div class="paragraph">
<p>Name of the parameter whose value should be checked.</p>
</div>
</div>
</div>
<div class="sidebarblock">
<div class="content">
<div class="title">allowedParameterValues</div>
<div class="paragraph">
<p>Comma-separated list of valid values for the given parameter, e.g. SYSOUT1, SYSOUT2.</p>
</div>
</div>
</div>
<div class="sidebarblock">
<div class="content">
<div class="title">nameField</div>
<div class="paragraph">
<p>The name field value of the statement to be checked, e.g. JOBLIB, STEPLIB.
This parameter is optional. If absent, the name field of the statement will not be taken into account.</p>
</div>
</div>
</div>
<hr>
</div>
</div>
</div>