<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when <code>Map.asBoolean()</code> is called explicitly or when a map is used in a boolean context through implicit conversion.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Using <code>Map.asBoolean()</code> or implicit boolean conversion of maps can be misleading and confusing for developers. These methods only check whether the map is empty, not whether it contains meaningful or valid data.</p>
</div>
<div class="paragraph">
<p>The confusion arises because:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>The method name <code>asBoolean()</code> doesn&#8217;t clearly indicate it&#8217;s checking emptiness</p>
</li>
<li>
<p>Implicit boolean conversion in conditional statements can be misunderstood</p>
</li>
<li>
<p>Developers might expect these methods to validate the content quality, not just presence</p>
</li>
<li>
<p>Code reviewers and maintainers may misinterpret the intent</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>For example, a map containing only null values or placeholder data would still return <code>true</code> with <code>asBoolean()</code>, even though the data might not be meaningful for the application&#8217;s logic.</p>
</div>
<div class="paragraph">
<p>Explicit methods like <code>isEmpty()</code> or <code>size()</code> checks make the code&#8217;s intent crystal clear and improve readability.</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="paragraph">
<p>Using implicit boolean conversion can lead to logical errors where the code appears to check for meaningful data but only verifies the map isn&#8217;t empty. This can cause bugs when maps contain placeholder or invalid data that still passes the boolean check.</p>
</div>
<div class="paragraph">
<p>The unclear intent also reduces code maintainability, as future developers may misunderstand what the condition is actually testing.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Replace explicit calls to <code>asBoolean()</code> with clear emptiness checks using <code>isEmpty()</code> or <code>size()</code>.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-groovy" data-lang="groovy">def map = [:]
if (map.asBoolean()) { // Noncompliant
    println "Map has data"
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-groovy" data-lang="groovy">def map = [:]
if (!map.isEmpty()) {
    println "Map has data"
}</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Groovy Map Documentation - <a href="http://docs.groovy-lang.org/latest/html/groovy-jdk/java/util/Map.html">Official Groovy documentation for Map methods including asBoolean()</a></p>
</li>
<li>
<p>Groovy Truth - <a href="https://groovy-lang.org/semantics.html#Groovy-Truth">Documentation explaining Groovy&#8217;s truth evaluation rules</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>