This rule raises an issue when the dimensions of the input arrays for the ``++numpy.broadcast++`` function are not compatible for matrix broadcasting.

== Why is this an issue?

The ``++numpy.broadcast++`` function performs matrix broadcasting between two arrays, this means that
the output will contain a smaller matrix stretched to the shape of the bigger one.

For example:
[source,python]
----
import numpy as np

x = np.array([[0, 0 , 0], [10, 10, 10], [20, 20, 20], [30, 30, 30]])
y = np.array([1,2,3])
b = np.broadcast(x, y)

out = np.empty(b.shape)
out.flat = [v for (u, v) in b]
print(out)
----

the output will be
[source,text]
----
[[1. 2. 3.]
 [1. 2. 3.]
 [1. 2. 3.]
 [1. 2. 3.]]
----

This broadcasting mechanism is also used silently when performing basic matrix math operations
such as a multiplication (``++*++``) or an addition (``+++++``) of two matrices.

For a matrix broadcasting to be valid, the number of rows in the matrix must be equal to the number of rows in the second matrix or has just one row.
You can use ``++numpy.shape()++`` function to check the dimensions of the arrays before performing the operation.

== How to fix it

To fix this issue you should make sure that two broadcasted matrices are compatible

=== Code examples

==== Noncompliant code example

[source,python,diff-id=1,diff-type=noncompliant]
----
import numpy as np

a = np.array([[1, 2], [3, 4]])
b = np.array([5, 6, 7])
c = a * b # Noncompliant: ValueError: operands could not be broadcast together with shapes (2,2) (3,)
d = np.broadcast(a, b) # Noncompliant: ValueError: shape mismatch: objects cannot be broadcast to a single shape.  Mismatch is between arg 0 with shape (2, 2) and arg 1 with shape (3,).
----


==== Compliant solution

[source,python,diff-id=1,diff-type=compliant]
----
import numpy as np

a = np.array([[1, 2], [3, 4]])
b = np.array([5, 6])
c = a * b # Compliant
d = np.broadcast(a, b) # Compliant
----

== Resources
=== Documentation
* NumPy Documentation - https://numpy.org/doc/stable/user/basics.broadcasting.html[Broadcasting]
* NumPy Documentation - https://numpy.org/doc/stable/reference/generated/numpy.broadcast.html#numpy-broadcast[numpy.broadcast]


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)



'''
== Comments And Links
(visible only on this page)

endif::env-github,rspecator-view[]
