This rule raises an issue when the dimensions of the input arrays for the ``++numpy.broadcast++`` function are not compatible for matrix broadcasting.

== Why is this an issue?

The ``++numpy.broadcast++`` function performs matrix broadcasting between two arrays, could be also called silently during basic matrix math operations
such as a multiplication (``++*++``) or an addition (``+++++``) of two matrixes.
For a matrix broadcasting to be valid, the number of rows in the matrix must be equal to the number of rows in the second matrix or has just one row.
You can use ``++numpy.shape()++`` function to check the dimensions of the arrays before performing the operation.

=== Code examples

==== Noncompliant code example

[source,python,diff-id=1,diff-type=noncompliant]
----
import numpy as np

a = np.array([[1, 2], [3, 4]])
b = np.array([5, 6, 7])
c = a * b # Noncompliant: ValueError: operands could not be broadcast together with shapes (2,2) (3,)
d = np.broadcast(a, b) # Noncompliant: ValueError: shape mismatch: objects cannot be broadcast to a single shape.  Mismatch is between arg 0 with shape (2, 2) and arg 1 with shape (3,).
----


==== Compliant solution

[source,python,diff-id=1,diff-type=compliant]
----
import numpy as np

a = np.array([[1, 2], [3, 4]])
b = np.array([5, 6])
c = a * b # Compliant
d = np.broadcast(a, b) # Compliant
----

== Resources
=== Documentation
* NumPy Documentation - https://numpy.org/doc/stable/user/basics.broadcasting.html[Broadcasting]
* NumPy Documentation - https://numpy.org/doc/stable/reference/generated/numpy.broadcast.html#numpy-broadcast[numpy.broadcast]


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)



'''
== Comments And Links
(visible only on this page)

endif::env-github,rspecator-view[]
