<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when multiple operations performed sequentially on a data frame could be replaced with method chaining.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Pandas provides an easy way to apply multiple operations to a DataFrame through method chaining. When intermediate results are not needed, it is generally recommended to chain methods rather than assigning the result of each operation back to a variable and then applying subsequent operations to that variable.</p>
</div>
<div class="paragraph">
<p>Method chaining leads to clearer and more concise code and is therefore the recommended practice. This rule will raise an issue when three or more consecutive operations are performed on a DataFrame without any use of the intermediate results.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">df_filtered = df[df['A'] &gt; 2]
df_grouped = df_filtered.groupby('B').sum()
df_sorted = df_grouped.sort_values('C', ascending=False)
result = df_sorted.reset_index()  # Noncompliant: The multiple reassignments are not needed.</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">result = (df[df['A'] &gt; 2]
    .groupby('B')
    .sum()
    .sort_values('C', ascending=False)
    .reset_index()
)  # Compliant: method chaining is clearer and more intentional</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_articles_blog_posts">Articles &amp; blog posts</h3>
<div class="ulist">
<ul>
<li>
<p>Practical Data Science - <a href="https://practicaldatascience.co.uk/data-science/how-to-use-method-chaining-in-pandas">How to use method chaining in Pandas</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>