== Why is this an issue?

Using the ``++OPEN++`` file statement is costly, and therefore be avoided inside loops. Instead, the file can be saved into a buffer to increase performance.


=== Noncompliant code example

[source,cobol]
----
  PERFORM UNTIL (NOT DA-OK) OR (Y00CIA-CD-RET-PGM = ZERO)
     OPEN INPUT inventory-file
  END-PERFORM.
----

