== Why is this an issue?

Regardless of scope, no identifier should be re-used across any files in the system.


=== Noncompliant code example

[source,cpp]
----
struct air_speed
{
  uint16_t speed; /* knots */
} * x;

struct gnd_speed
{
  uint16_t speed; /* mph */ // Not Compliant - speed is in different units
} * y;

x->speed = y->speed; // Error: unit conversion required
----


=== Compliant solution

[source,cpp]
----
struct air_speed
{
  uint16_t knotSpeed; /* knots */
} * x;

struct gnd_speed
{
  uint16_t mphSpeed; /* mph */ // Compliant, different name
} * y;

x->knotSpeed = y->mphSpeed; // An error can be deducted from the name of the fields.
----


=== Exceptions

Where an identifier name is used in a header file, and that header file is included in multiple source files, this rule is not violated. The use of a rigorous naming convention can support the implementation of this rule.


== Resources

* MISRA C:2004, 5.7


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Rename this declaration, whose identifier has already been used.


'''
== Comments And Links
(visible only on this page)

=== is related to: S805

=== relates to: S807

=== relates to: S832

=== is related to: S802

=== is related to: S803

=== is related to: S828

=== is related to: S1014

=== on 17 Oct 2014, 15:22:02 Ann Campbell wrote:
\[~freddy.mallet] this rule overlaps a couple other, linked issues.

=== on 17 Oct 2014, 15:23:25 Ann Campbell wrote:
\[~samuel.mercier] as written, this rule makes me sad. It will presumably flag every instance of ``++int i++``.

endif::env-github,rspecator-view[]
