Regardless of scope, no identifier should be re-used across any files in the system.


== Noncompliant Code Example

----
struct air_speed
{
  uint16_t speed; /* knots */
} * x;

struct gnd_speed
{
  uint16_t speed; /* mph */ // Not Compliant - speed is in different units
} * y;

x->speed = y->speed; // Error: unit conversion required
----


== Compliant Solution

----
struct air_speed
{
  uint16_t knotSpeed; /* knots */
} * x;

struct gnd_speed
{
  uint16_t mphSpeed; /* mph */ // Compliant, different name
} * y;

x->knotSpeed = y->mphSpeed; // An error can be deducted from the name of the fields.
----


== Exceptions

Where an identifier name is used in a header file, and that header file is included in multiple source files, this rule is not violated. The use of a rigorous naming convention can support the implementation of this rule.


== See

* MISRA C:2004, 5.7


ifdef::env-github,rspecator-view[]
'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
