== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,python]
----
# pass
try:
    # ...
except:
    pass # Noncompliant

# continue
for i in tab:
    try:
         # ...
    except:
        continue # Noncompliant
----

=== Compliant solution

[source,python]
----
def can_raise_exception():
    try:
        # ...
    except:
        raise # Compliant
----

[source,python]
----
try:
    # ...
except:
    # comment explaining why taking no action is appropriate; Compliant
----

[source,python]
----
try:
    # ...
except:
     logging.error('...')
     pass  # Compliant
----

include::../see.adoc[]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
