include::../description.adoc[]

== Noncompliant Code Example

----
# pass
try:
    # ...
except:
    pass # Noncompliant

# continue
for i in tab:
    try:
         # ...
    except:
        continue # Noncompliant
----

== Compliant Solution

----
def can_raise_exception():
    try:
        # ...
    except:
        raise # Compliant
----

----
try:
    # ...
except:
    # comment explaining why taking no action is appropriate; Compliant
----

----
try:
    # ...
except:
     logging.error('...')
     pass  # Compliant
----

include::../see.adoc[]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
