== Why is this an issue?

Declaring multiple variable on one line is difficult to read.

=== Noncompliant code example

[source,sql]
----
DECLARE @aaa AS INTEGER = 5, @bbb AS INTEGER = 42, @ccc AS CHAR(3) = 'foo'  -- Noncompliant
----

=== Compliant solution

[source,sql]
----
DECLARE @aaa AS INTEGER = 5
DECLARE @bbb AS INTEGER = 42
DECLARE @ccc AS CHAR(3) = 'foo'
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
