Declaring multiple variable on one line is difficult to read.

== Noncompliant Code Example

----
DECLARE @aaa AS INTEGER = 5, @bbb AS INTEGER = 42, @ccc AS CHAR(3) = 'foo'  -- Noncompliant
----

== Compliant Solution

----
DECLARE @aaa AS INTEGER = 5
DECLARE @bbb AS INTEGER = 42
DECLARE @ccc AS CHAR(3) = 'foo'
----

include::../see.adoc[]

ifdef::rspecator-view[]
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::rspecator-view[]
