With Java 8, there's no need to write ``++Comparator++``s that compare primitive values or other ``++Comparable++``s; they can be generated for you using the ``++Comparator.comparing*++`` functions: ``++comparing++``, ``++comparingDouble++``, ``++comparingInt++``, ``++comparingLong++``.


*Note* that this rule is automatically disabled when the project's ``++sonar.java.source++`` is lower than ``++8++``.


== Noncompliant Code Example

----
unparsedFiles.stream()
   .sorted((f1, f2) -> f1.lines - f2.lines)  // Noncompliant
   .limit(30);
----


== Compliant Solution

----
unparsedFiles.stream()
   .sorted(Comparator.comparingInt(UnparsedFile::getLines()))
   .limit(30);
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

include::highlighting.adoc[]

endif::env-github,rspecator-view[]
