<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Marking a class with <a href="https://learn.microsoft.com/en-us/dotnet/api/system.componentmodel.composition.partcreationpolicyattribute"><code>PartCreationPolicy</code></a>(<a href="https://learn.microsoft.com/en-us/dotnet/api/system.componentmodel.composition.creationpolicy"><code>CreationPolicy.Shared</code></a>), which is part of <a href="https://learn.microsoft.com/en-us/dotnet/framework/mef">Managed Extensibility Framework (MEF)</a>, means that a single, shared instance of the exported object will be created. Therefore it doesn&#8217;t make sense to create new instances using the constructor and it will most likely result in unexpected behaviours.</p>
</div>
<div class="paragraph">
<p>This rule raises an issue when a constructor of a class marked shared with a <code>PartCreationPolicyAttribute</code> is invoked.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-vbnet" data-lang="vbnet">&lt;Export(GetType(IFooBar))&gt;
&lt;PartCreationPolicy(CreationPolicy.[Shared])&gt;
Public Class FooBar
    Inherits IFooBar
End Class

Public Class Program
    Public Shared Sub Main()
        Dim fooBar = New FooBar() ' Noncompliant
    End Sub
End Class</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-vbnet" data-lang="vbnet">&lt;Export(GetType(IFooBar))&gt;
&lt;PartCreationPolicy(CreationPolicy.[Shared])&gt;
Public Class FooBar
    Inherits IFooBar
End Class

Public Class Program
    Public Shared Sub Main()
        Dim fooBar = serviceProvider.GetService(Of IFooBar)()
    End Sub
End Class</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p><a href="https://learn.microsoft.com/en-us/dotnet/api/system.componentmodel.composition.partcreationpolicyattribute"><code>PartCreationPolicy</code></a></p>
</li>
<li>
<p><a href="https://learn.microsoft.com/en-us/dotnet/api/system.componentmodel.composition.creationpolicy"><code>CreationPolicy.Shared</code></a></p>
</li>
<li>
<p><a href="https://learn.microsoft.com/en-us/dotnet/framework/mef">Managed Extensibility Framework</a></p>
</li>
</ul>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Refactor this code so that it doesn&#8217;t invoke the constructor of this class.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>Constructor invocation</p>
</div>
</div>
</div>
</div>