<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>When a <code>SELECT</code> returns <code>null</code> from a nullable column, the relevant host variable isn&#8217;t updated; it simply retains its previous value. The only way you&#8217;ll ever know the column value was <code>null</code> is to check the relevant null indicator included in the <code>SELECT</code> for a negative (<code>null</code>) value.</p>
</div>
<div class="paragraph">
<p>This rule raises an issue when a <code>SELECT</code> omits a <code>null</code> indicator for a nullable column.</p>
</div>
<div class="paragraph">
<p><strong>Note</strong> that this rule raises issues only when a database catalog is provided during the SonarQube analysis.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="paragraph">
<p>With the table <code>PRODUCT</code> having a nullable column <code>NAME</code>:</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cobol" data-lang="cobol">EXEC SQL
  SELECT
    PROD_ID,
    NAME
  INTO
    :P-ID,
    :P-NAME                 -- Noncompliant; No null indicator
  FROM PRODUCT
END-EXEC</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cobol" data-lang="cobol">EXEC SQL
  SELECT
    PROD_ID,
    NAME
  INTO
    :P-ID,
    :P-NAME :P-NAME-NULL  -- Compliant
  FROM PRODUCT
END-EXEC</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Add a null indicator for "xxx"; it is nullable.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>Host variable</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_17_may_2016_205942_ann_campbell_wrote">on 17 May 2016, 20:59:42 Ann Campbell wrote:</h3>
<div class="paragraph">
<p><a href="https://www.ibm.com/support/knowledgecenter/SSEPGG_9.5.0/com.ibm.db2.luw.apdv.embed.doc/doc/t0005636.html" class="bare">https://www.ibm.com/support/knowledgecenter/SSEPGG_9.5.0/com.ibm.db2.luw.apdv.embed.doc/doc/t0005636.html</a></p>
</div>
</div>
</div>
</div>