== How to fix it

=== Code examples

include::../../common/fix/code-rationale.adoc[]

Hostname validation is disabled if ``++InsecureSkipVerify++`` is set to `true` for the ``++TLSClientConfig++``.

It is recommended to prefer high-level interfaces (like ``++http.Get()++``), which perform the certificate validation instead of using ``++http.Client++`` directly.

==== Noncompliant code example

[source,go,diff-id=1,diff-type=noncompliant]
----
client := &http.Client{
    Transport: &http.Transport{
        TLSClientConfig: &tls.Config{
            InsecureSkipVerify: true, // Non-compliant
        },
    },
}

client.Get("https://example.com")
----

==== Compliant solution

Usage of high-level interfaces is recommended:
[source,go,diff-id=1,diff-type=compliant]
----
http.Get("https://example.com")
----

=== How does this work?

include::../../common/fix/validation.adoc[]
