== How to fix it

=== Code examples

include::../../common/fix/code-rationale.adoc[]

==== Noncompliant code example

[source,docker,diff-id=1,diff-type=noncompliant]
----
FROM ubuntu:22.04

# Noncompliant
RUN curl --insecure -O https://expired.example.com/downloads/install.sh
----

==== Compliant solution

[source,docker,diff-id=1,diff-type=compliant]
----
FROM ubuntu:22.04

RUN curl -O https://new.example.com/downloads/install.sh
----

=== How does this work?

include::../../common/fix/validation.adoc[]
