include::../../../shared_content/secrets/description.adoc[]

== Why is this an issue?

include::../../../shared_content/secrets/rationale.adoc[]

=== What is the potential impact?

Passwords are often used to authenticate users against database engines. They
are associated with user accounts that are granted specific permissions over the
database and its hosted data.

If a database password leaks to an unintended audience, it can have serious
consequences for the security of your database instance, the data stored within
it, and the applications that rely on it.

include::../../../shared_content/secrets/impact/data_compromise.adoc[]

==== Security downgrade

Applications relying on a database instance can suffer a security downgrade if
an access password is leaked to attackers. Depending on the purposes the
application uses the database for, consequences can range from low-severity
issues, like defacement, to complete compromise.

For example, if the database instance is used as part of the authentication
process of an application, attackers with access to the database will likely be
able to bypass this security mechanism.

== How to fix it

include::../../../shared_content/secrets/fix/revoke.adoc[]

include::../../../shared_content/secrets/fix/vault.adoc[]

=== Code examples

==== Noncompliant code example

[source,java,diff-id=1,diff-type=noncompliant]
----
public static string ConnectionString = "server=database-server;uid=user;pwd=P@ssw0rd;database=ProductionData";
----

==== Compliant solution

[source,java,diff-id=1,diff-type=compliant]
----
public static string ConnectionString = String.format(
    "server=database-server;uid=user;pwd=%s;database=ProductionData",
    System.getenv("DB_PASSWORD")
)
----

//=== How does this work?

//=== Pitfalls

//=== Going the extra mile

== Resources

include::../../../shared_content/secrets/resources/standards.adoc[]

//=== Benchmarks