<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when a CRON expression used with <code>System.schedule</code> is improperly formatted or has a non-zero seconds field.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Salesforce has specific requirements for CRON expressions that differ from standard CRON syntax. The platform requires exactly 6 fields in the format: seconds, minutes, hours, day of month, month, and day of week.</p>
</div>
<div class="paragraph">
<p>The seconds field must always be set to '0' because Salesforce does not support scheduling jobs at the second level. This is a platform limitation that prevents more granular scheduling.</p>
</div>
<div class="paragraph">
<p>When CRON expressions don&#8217;t follow these rules, several problems occur:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>Jobs may fail to schedule without clear error messages</p>
</li>
<li>
<p>Silent failures can leave critical batch processes unscheduled</p>
</li>
<li>
<p>Debugging scheduling issues becomes difficult when the root cause is malformed syntax</p>
</li>
<li>
<p>Standard CRON expressions from other systems won&#8217;t work directly in Salesforce</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>Proper CRON expression formatting ensures reliable job scheduling and prevents unexpected failures in automated processes.</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="paragraph">
<p>Improperly formatted CRON expressions can cause batch jobs and scheduled processes to fail silently, leading to missed data processing, delayed business operations, and difficult-to-diagnose scheduling issues.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Use the correct 6-field CRON format with seconds set to '0'. Ensure all required fields are present: seconds (0), minutes, hours, day of month, month, and day of week.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-apex" data-lang="apex">String cronExp = '30 0 1 1 * ?'; // Noncompliant: seconds should be 0
System.schedule('Monthly Update', cronExp, scheduler);</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-apex" data-lang="apex">String cronExp = '0 0 1 1 * ?'; // Correct: seconds set to 0
System.schedule('Monthly Update', cronExp, scheduler);</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Salesforce Apex Scheduler - <a href="https://developer.salesforce.com/docs/atlas.en-us.apexcode.meta/apexcode/apex_scheduler.htm">Official Salesforce documentation on scheduling Apex jobs with CRON expressions</a></p>
</li>
<li>
<p>System.schedule Method - <a href="https://developer.salesforce.com/docs/atlas.en-us.apexref.meta/apexref/apex_methods_system_system.htm#apex_System_System_schedule">Reference documentation for the System.schedule method and CRON expression syntax</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>