include::../common/description.adoc[]

include::../common/impact.adoc[]

include::../common/fix.adoc[]

=== Code examples

==== Noncompliant code example

[source,sql,diff-id=1,diff-type=noncompliant]
----
SELECT HASHBYTES('SHA1', MyColumn) FROM dbo.MyTable;  -- Noncompliant
----

==== Compliant solution

[source,sql,diff-id=1,diff-type=compliant]
----
SELECT HASHBYTES('SHA2_512', MyColumn) FROM dbo.MyTable;
----

include::../common/fix-pitfalls.adoc[]

include::../common/resources.adoc[]


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../common/message.adoc[]

include::../common/highlighting.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../common/comments-and-links.adoc[]

endif::env-github,rspecator-view[]
