== Why is this an issue?

include::description.adoc[]

=== Noncompliant code example

[source,cpp,diff-id=1,diff-type=noncompliant]
----
switch (param) {
  case 0:
    doSomething();
    break;
  default: // default clause should be the first or last one
    error();
    break;
  case 1:
    doSomethingElse();
    break;
}
----

=== Compliant solution

[source,cpp,diff-id=1,diff-type=compliant]
----
switch (param) {
  case 0:
    doSomething();
    break;
  case 1:
    doSomethingElse();
    break;
  default:
    error();
    break;
}
----
