== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,dart]
----
RegExp(r'([');
RegExp(r'(\\w+-(\\d+)');
----


=== Compliant solution

[source,dart]
----
RegExp(r'\(\[');
RegExp(r'(\\w+)-(\\d+)');
----

== Resources

=== Documentation

* Dart Docs - https://dart.dev/tools/linter-rules/valid_regexps[Dart Lint rule - valid_regexps]
* Dart API Reference - https://api.dart.dev/stable/dart-core/RegExp-class.html[Dart RegExp]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Invalid regular expression syntax.

=== Highlighting

The error inside the regular expression

endif::env-github,rspecator-view[]
