== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,dart]
----
RegExp(r'([');
RegExp(r'(\\w+-(\\d+)');
----


=== Compliant solution

[source,dart]
----
RegExp(r'\(\[');
RegExp(r'(\\w+)-(\\d+)');
----

== Resources

* https://dart.dev/tools/linter-rules/valid_regexps[Dart Lint rule]
* https://api.dart.dev/stable/3.4.4/dart-core/RegExp-class.html[Dart RegExp]
