<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Unsigned right shift (<code>&gt;&gt;&gt;</code>) fills from the left with 0&#8217;s, versus signed right shift (<code>&gt;&gt;</code>), which fills with the sign bit. There&#8217;s no point in using <code>&gt;&gt;&gt;</code> unless you care about the upper bits of the result, so casting a <code>&gt;&gt;&gt;</code>-ed value to a smaller type (which discards the upper bits) is likely a mistake. Either <code>&gt;&gt;</code> should have been used, or the cast was made in error.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-text" data-lang="text">int i = -1;
short s = (short)i &gt;&gt;&gt; 4; /// Noncompliant</code></pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p><a href="https://wiki.sei.cmu.edu/confluence/x/kDZGBQ">CERT, NUM52-J.</a> - Be aware of numeric promotion behavior</p>
</li>
</ul>
</div>
<hr>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove this "xxx" cast.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_29_apr_2015_130647_nicolas_peru_wrote">on 29 Apr 2015, 13:06:47 Nicolas Peru wrote:</h3>
<div class="paragraph">
<p>Critical seems a bit high for this no ?</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_30_apr_2015_063018_ann_campbell_wrote">on 30 Apr 2015, 06:30:18 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~nicolas.peru] I originally set this to critical with the thought that the data loss was accidental, but I&#8217;ve downgraded.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_20_may_2015_150926_nicolas_peru_wrote">on 20 May 2015, 15:09:26 Nicolas Peru wrote:</h3>
<div class="paragraph">
<p>LGTM</p>
</div>
</div>
</div>
</div>