// If you want to factorize the description uncomment the following line and create the file.
//include::../description.adoc[]
The `@Autowired` annotation ensures that beans are correctly injected into fields.
However, the order of initialization of fields can make it so that at given time, an instance field still has a default null value.

== Why is this an issue?

Relying on the default value without checking it can lead to crashes at load time or unexpected behavior at runtime.

=== What is the potential impact?
The runtime of your application may be affected if:

* at startup time, if a bean depends on the class whose initialization fails


== How to fix it
Do not rely on `@Autowired` for initialization fields or check them for potential null values.

=== Code examples

==== Noncompliant code example

[source,java,diff-id=1,diff-type=noncompliant]
----
@Service
class BrickAndMortarStore {
    @Autowired
    private StoreConfiguration configuration;

    private String address = configuration.getAddress(); // Noncompliant, will loading the class
}
----

==== Compliant solution

[source,java,diff-id=1,diff-type=compliant]
----
FIXME
----

//=== How does this work?

//=== Pitfalls

//=== Going the extra mile


//== Resources
//=== Documentation
//=== Articles & blog posts
//=== Conference presentations
//=== Standards
//=== External coding guidelines
//=== Benchmarks
