== Why is this an issue?

Shared coding conventions allow teams to collaborate effectively. This rule flags all Perl-style comments.


=== Noncompliant code example

[source,php]
----
$myvar; # Noncompliant; this comment should have started with "//"
----


=== Compliant solution

[source,php]
----
$myvar; // Compliant; this comment started with "//"
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Use "//" instead of "#" to start this comment


endif::env-github,rspecator-view[]
