Shared coding conventions allow teams to collaborate effectively. This rule flags all Perl-style comments.

== Noncompliant Code Example

----
$myvar; # Noncompliant; this comment should have started with "//"
----

== Compliant Solution

----
$myvar; // Compliant; this comment started with "//"
----
