<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when the <code>instanceof</code> operator is used to check the type of objects that could be Hibernate entities.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Hibernate uses proxy objects for lazy loading to improve performance. When you access a relationship that hasn&#8217;t been loaded yet, Hibernate creates a proxy object instead of the actual entity.</p>
</div>
<div class="paragraph">
<p>These proxy objects extend or implement your entity class, but they are not direct instances of it. This means <code>instanceof</code> checks can fail unexpectedly, even when the proxy represents the correct type.</p>
</div>
<div class="paragraph">
<p>For example, if you have an entity <code>User</code> and Hibernate creates a proxy for it, <code>userProxy instanceof User</code> might return <code>false</code> even though the proxy represents a <code>User</code> object. This can cause your application logic to behave incorrectly, leading to bugs that are difficult to track down.</p>
</div>
<div class="paragraph">
<p>The problem becomes more apparent when checking for interfaces. A Hibernate proxy might implement an interface like <code>DirtyCheckable</code>, but <code>instanceof DirtyCheckable</code> could still return <code>false</code> because the proxy class structure differs from what you expect.</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="paragraph">
<p>Using <code>instanceof</code> with Hibernate entities can cause application logic to fail silently. Methods that should execute might be skipped, leading to incorrect behavior, data inconsistencies, or missing functionality. These bugs are particularly difficult to debug because they may only occur under specific loading conditions.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Use <code>Class.isAssignableFrom()</code> instead of <code>instanceof</code> to check if an object is compatible with a specific type. This method works correctly with Hibernate proxies because it checks the class hierarchy rather than the exact instance type.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-groovy" data-lang="groovy">Object entityObject = ea.getEntity()
boolean isDirtyCheckable = entityObject instanceof DirtyCheckable // Noncompliant</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-groovy" data-lang="groovy">Object entityObject = ea.getEntity()
boolean isDirtyCheckable = DirtyCheckable.class.isAssignableFrom(entityObject.getClass())</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Hibernate User Guide - Proxies - <a href="https://docs.jboss.org/hibernate/orm/current/userguide/html_single/Hibernate_User_Guide.html#pc-proxy">Official Hibernate documentation explaining how proxies work and their implications</a></p>
</li>
<li>
<p>Java Class.isAssignableFrom() Documentation - <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Class.html#isAssignableFrom-java.lang.Class-">Official Java documentation for the isAssignableFrom method</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>