<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>In Azure Resource Manager (ARM) templates, it is possible to set <code>allowedValues</code> for various parameters.
This field is used to define a list of valid values for a parameter.
This is a way to restrict the input values provided when deploying or updating a resource using the template.</p>
</div>
<div class="paragraph">
<p>However, when it comes to a parameter defining the <code>location</code> of a resource, this practice can lead to a code smell.
Users may be unable to deploy such a template if their desired location is not included in the <code>allowedValues</code>.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>In Azure Resource Manager (ARM) templates, it is possible to set <code>allowedValues</code> for various parameters to limit the options and maintain control.
However, when it comes to a parameter defining the <code>location</code> of a resource, this practice can lead to a code smell.
Specifically, setting <code>allowedValues</code> for a location parameter can cause issues because the locations list might not be exhaustive or suitable for all users.
Users may be unable to deploy such a template if their desired location is not included in the <code>allowedValues</code>, causing inconvenience and potential delays in their work.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it_in_json_templates">How to fix it in JSON templates</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Remove <code>allowedValues</code> for the parameter specifying the location.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-json" data-lang="json">{
  "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentParameters.json#",
  "contentVersion": "1.0.0.0",
  "parameters": {
      "location": {
          "type": "string",
          "metadata": {
              "description": "The location in which the resources should be deployed."
          },
          "defaultValue": "[resourceGroup().location]",
          "allowedValues": [
              "eastus",
              "westus",
              "northeurope",
              "westeurope",
              "southeastasia"
          ]
      }
  }
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-json" data-lang="json">{
  "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentParameters.json#",
  "contentVersion": "1.0.0.0",
  "parameters": {
      "location": {
          "type": "string",
          "metadata": {
              "description": "The location in which the resources should be deployed."
          },
          "defaultValue": "[resourceGroup().location]"
      }
  }
}</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it_in_bicep">How to fix it in Bicep</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Remove <code>allowedValues</code> for the parameter specifying the location.</p>
</div>
<div class="sect2">
<h3 id="_code_examples_2">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example_2">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-bicep" data-lang="bicep">@allowed([
  'eastus'
  'westus'
  'northeurope'
  'westeurope'
  'southeastasia'
])  // Noncompliant
param location string = resourceGroup().location</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution_2">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-bicep" data-lang="bicep">param location string = resourceGroup().location</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Azure Resource Manager - <a href="https://learn.microsoft.com/en-us/azure/azure-resource-manager/templates/best-practices#location-recommendations-for-parameters">Location recommendations for parameters</a></p>
</li>
</ul>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>For ARM Templates: Remove this <code>allowedValues</code> property from the parameter specifying the location.</p>
</div>
<div class="paragraph">
<p>For Bicep: Remove this <code>@allowed</code> decorator from the parameter specifying the location.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>In case of ARM Tempates , highlight the <code>allowedValues</code> property in the parameter specifying the location.</p>
</div>
<div class="paragraph">
<p>In case of Bicep, highlight the <code>@allowed</code> decorator above the parameter specifying the location.</p>
</div>
</div>
</div>
</div>