== How to fix it in Apache Commons

=== Code examples

include::../../common/fix/code-rationale.adoc[]

==== Noncompliant code example

[source,kotlin,diff-id=1,diff-type=noncompliant]
----
import org.apache.commons.exec.CommandLine

@Controller
class ExampleController {
    @GetMapping("/find")
    fun find(@RequestParam("filename") filename: String) {
        val cmd = CommandLine("/usr/bin/find . -iname $filename") // Noncompliant
    }
}
----

==== Compliant solution

[source,kotlin,diff-id=1,diff-type=compliant]
----
import org.apache.commons.exec.CommandLine

@Controller
class ExampleController {
    @GetMapping("/find")
    fun find(@RequestParam("filename") filename: String) {
        val cmd = CommandLine("/usr/bin/find")
        cmd.addArguments(arrayOf("/usr/bin/find", ".", "-iname", filename))
    }
}
----

=== How does this work?

include::../../common/fix/introduction.adoc[]

Here `org.apache.commons.exec.CommandLine.addArguments(addArguments: Array<String>)` takes care of escaping the passed arguments and internally
creates a single string given to the operating system to be executed.
