== How to fix it in Java Lang Package

=== Code examples

include::../../common/fix/code-rationale.adoc[]

==== Noncompliant code example

[source,kotlin,diff-id=11,diff-type=noncompliant]
----
@Controller
class ExampleController {
    @GetMapping("/find")
    fun find(@RequestParam("filename") filename: String) {
        Runtime.getRuntime().exec("/usr/bin/find . -iname $filename")
    }
}
----

==== Compliant solution

[source,kotlin,diff-id=11,diff-type=compliant]
----
@Controller
class ExampleController {
    @GetMapping("/find")
    fun find(@RequestParam("filename") filename: String) {
        val cmd = arrayOf("/usr/bin/find", ".", "-iname", filename)
        Runtime.getRuntime().exec(cmd)
    }
}
----

++java.lang.Runtime++ is sometimes used over ++java.lang.ProcessBuilder++ due to ease of use. Flexibility in methods often introduces security issues as edge cases are easily missed. The compliant solution logic is also applied to ++java.lang.ProcessBuilder++.

=== How does this work?

include::../../common/fix/introduction.adoc[]

Here `java.lang.Runtime.exec(cmdarray: Array<String>)` takes care of escaping the passed arguments and internally
creates a single string given to the operating system to be executed.
