==== Noncompliant code example

[source,text,diff-id=1,diff-type=noncompliant]
----
public class Foo { 
  private String foo; 

  public String getFoo() { } 
} 

Foo foo = new Foo(); 
foo.getFoo() // what does this return? 
----
