include::../description.adoc[]

== Noncompliant Code Example

[source,cs]
----
class Program
{
    public int Foo  //Non-Compliant
    {
        set
        {
            // ... some code ...
        }
    }
}
----

== Compliant Solution

[source,cs]
----
class Program
{
    private int foo;

    public void SetFoo(int value)
    {
        // ... some code ...
        foo = value;
    }
}
----

or


[source,cs]
----
class Program
{
  public int Foo { get; set; } // Compliant
}
----
ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

endif::env-github,rspecator-view[]
