== Why is this an issue?

The `Object.wait(...)`, `Object.notify()` and `Object.notifyAll()` methods are used in multithreaded environments to coordinate the access of multiple threads on the same resources.
By contract, a `Thread` needs to hold the lock on the object these methods are being invoked on, otherwise, an `IllegalMonitorException` is thrown.

== How to fix it

To ensure that these methods are called only when the currently running thread is the holder of the lock on the object, you can use the `synchronized` keyword.
`synchronized` can be used to mark entire methods or to define synchronized blocks within methods.
The code inside a synchronized portion of code will only be executed by one thread at a time, and the executing thread will acquire the lock on the object.
This guarantees that when one of the methods mentioned above is invoked by a thread, it will only possibly be the holder of the lock on the object.

=== Code examples

==== Noncompliant code example

[source,java]
----
private void removeElement() {
  while (!suitableCondition()){
    wait(); // Noncompliant, not being inside a `synchronized` block, this could be invoked by a Thread that does not have the right to do it
  }
  ... // Perform removal
}
----

==== Compliant solution

[source,java]
----
private void removeElement() {
  synchronized(obj) {
    while (!suitableCondition()){
      wait(); // Compliant, the `synchronized` block guarantees that Threads running this portion of the code, will be the holders of the object's lock
    }
    ... // Perform removal
  }
}
----

== References

* https://docs.oracle.com/javase/tutorial/essential/concurrency/syncmeth.html[docs.oracle] - Synchronized methods
* https://docs.oracle.com/javase%2F7%2Fdocs%2Fapi%2F%2F/java/lang/Object.html#wait()[docs.oracle] - java.lang.Object class and its methods

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Make this call to "[wait(...)|notify()|notifyAll()]" only inside a synchronized block to be sure to hold the monitor on "[this|xxx]" object.


endif::env-github,rspecator-view[]
