<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>The rules for method resolution are complex and perhaps not properly understood by all coders. The <code>params</code> keyword can make method declarations overlap in non-obvious ways, so that slight changes in the argument types of an invocation can resolve to different methods.</p>
</div>
<div class="paragraph">
<p>This rule raises an issue when an invocation resolves to a method declaration with <code>params</code>, but could also resolve to another non-<code>params</code> method too.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-csharp" data-lang="csharp">public class MyClass
{
    private void Format(string a, params object[] b) { }

    private void Format(object a, object b, object c) { }
}

// ...
MyClass myClass = new MyClass();

myClass.Format("", null, null); // Noncompliant, resolves to the first Format with params, but was that intended?</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Review this call, which partially matches an overload without "params".  The partial match is "XXX".</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_1_jul_2015_175650_ann_campbell_wrote">on 1 Jul 2015, 17:56:50 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>Note that this spec has been mapped to R# rule PossiblyMistakenUseOfParamsMethod after discussion with [~tamas.vajk] because they attack the same problem from different ends, and because it&#8217;s not clear what the message should be if it is the invocation that&#8217;s marked Noncompliant, as with R#.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_2_jul_2015_070710_tamas_vajk_wrote">on 2 Jul 2015, 07:07:10 Tamas Vajk wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] I would put <code>params</code> somewhere in the description. Because we don&#8217;t want to report on all method overloads.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_2_jul_2015_125934_ann_campbell_wrote">on 2 Jul 2015, 12:59:34 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>LGTM [~tamas.vajk]</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_7_jul_2015_123924_ann_campbell_wrote">on 7 Jul 2015, 12:39:24 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>fix message</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_7_jul_2015_140415_ann_campbell_wrote">on 7 Jul 2015, 14:04:15 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>New case added [~tamas.vajk]. Double-check me, please</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_8_jul_2015_085531_tamas_vajk_wrote">on 8 Jul 2015, 08:55:31 Tamas Vajk wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] Looks good</p>
</div>
</div>
</div>
</div>