include::../rationale.adoc[]

include::../how-to-fix-it.adoc[]

=== Code examples

==== Noncompliant code example

[source,kotlin,diff-id=1,diff-type=noncompliant]
----
fun numberOfMinutes(hours: Int): Int {
    val seconds: Int = 0 // Noncompliant - seconds is unused
    val result: Int = hours * 60
    return result
}
----


==== Compliant solution

[source,kotlin,diff-id=1,diff-type=compliant]
----
fun numberOfMinutes(hours: Int): Int {
    val result: Int = hours * 60
    return result
}
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
