<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Shared coding conventions make it possible for a team to collaborate efficiently.</p>
</div>
<div class="paragraph">
<p>This rule makes it mandatory to place a closing curly brace and the next <code>else</code>, <code>catch</code> or <code>finally</code> keyword on two different lines.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-text" data-lang="text">public void myMethod() {
  if(something) {
    executeTask();
  } else if (somethingElse) {          // Noncompliant
    doSomethingElse();
  }
  else {                               // Compliant
     generateError();
  }

  try {
    generateOrder();
  } catch (Exception e) {
    log(e);
  }
  finally {
    closeConnection();
  }
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-text" data-lang="text">public void myMethod() {
  if(something) {
    executeTask();
  }
  else if (somethingElse) {
    doSomethingElse();
  }
  else {
     generateError();
  }

  try {
    generateOrder();
  }
  catch (Exception e) {
    log(e);
  }
  finally {
    closeConnection();
  }
}</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Move this ["else"|"catch"|"finally"] keyword to a new, dedicated line</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_relates_to_s1106">relates to: <a data-rspec-id="S1106" class="rspec-auto-link">S1106</a></h3>

</div>
<div class="sect2">
<h3 id="_relates_to_s1107">relates to: <a data-rspec-id="S1107" class="rspec-auto-link">S1107</a></h3>

</div>
<div class="sect2">
<h3 id="_is_related_to_s1109">is related to: <a data-rspec-id="S1109" class="rspec-auto-link">S1109</a></h3>

</div>
<div class="sect2">
<h3 id="_on_3_jul_2013_155501_freddy_mallet_wrote">on 3 Jul 2013, 15:55:01 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>Is implemented by http://jira.codehaus.org/browse/SONARJAVA-192</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_3_mar_2014_135207_ann_campbell_wrote">on 3 Mar 2014, 13:52:07 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~dinesh.bolkensteyn], is this rule valid for C#?</p>
</div>
</div>
</div>
</div>