<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Passing arguments via an ellipsis bypasses the type checking performed by the compiler. Additionally, passing an argument with non-POD class type leads to undefined behavior. Note that the rule specifies "defined" (and not "declared") so as to permit the use of existing library functions.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">void MyPrintf ( char_t * pFormat, ... )	// Noncompliant
{
  // ...
}</code></pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p>MISRA C:2004, 16.1 - Functions shall not be defined with a variable number of arguments.</p>
</li>
<li>
<p>MISRA C&#43;&#43;:2008, 8-4-1 - Functions shall not be defined using the ellipsis notation.</p>
</li>
<li>
<p><a href="https://wiki.sei.cmu.edu/confluence/x/5ns-BQ">CERT, DCL50-CPP.</a> - Do not define a C-style variadic function</p>
</li>
<li>
<p>C&#43;&#43; Core Guidelines - <a href="https://github.com/isocpp/CppCoreGuidelines/blob/e49158a/CppCoreGuidelines.md#-es34-dont-define-a-c-style-variadic-function">ES.34: Don&#8217;t define a (C-style) variadic function</a></p>
</li>
<li>
<p>C&#43;&#43; Core Guidelines - <a href="https://github.com/isocpp/CppCoreGuidelines/blob/e49158a/CppCoreGuidelines.md#f55-dont-use-va_arg-arguments">F.55: Don&#8217;t use <code>va_arg</code> arguments</a></p>
</li>
</ul>
</div>
</div>
</div>