== Why is this an issue?

Using a "bald" function name is likely a bug. Rather than testing the return value of a function with a ``++void++`` parameter list, it implicitly retrieves the address of that function in memory. If that's truly what's intended, then it should be made explicit with the use of the ``++&++`` (address-of) operator. If it's not, then a parameter list (even an empty one) should be added after the function name.


=== Noncompliant code example

[source,cpp,diff-id=1,diff-type=noncompliant]
----
int func(void) {
  // ...
}

void f2(int a, int b) {
  // ...
  if (func) {  // Noncompliant: tests that the memory address of func() is non-null
    //...
  }
  // ...
}
----


=== Compliant solution

[source,cpp,diff-id=1,diff-type=compliant]
----
int func(void) {
  // ...
}

void f2(int a, int b) {
  // ...
  if (func()) {  // Compliant: tests that the return value of func() > 0
    //...
  }
  // ...
}
----


=== Exceptions

Callback functions are a common occurrence and are usually not passed with a preceding `&`. However, there is little ambiguity, so this rule ignores function identifiers when used as a parameter of a function call.

[source,cpp]
----
void foo() {
  // ...
}

registerEvent(AnEvent, foo); // Compliant by exception
----


== Resources

* MISRA C:2004, 16.9 - A function identifier shall only be used with either a preceding &, or with a parenthesized parameter list, which may be empty.
* MISRA {cpp}:2008, 8-4-4 - A function identifier shall only be used to call the function or it shall be preceded by &.


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Either add a parameter list or the "&" operator to this use of "xxx".


'''
== Comments And Links
(visible only on this page)

=== on 19 Sep 2014, 16:12:20 Freddy Mallet wrote:
@Ann, here I would have reused exactly the MISRA title, description and code snippets which are really meaningful for this rule.

endif::env-github,rspecator-view[]
