//// This rule is superseded by RSPEC-2583
//// Please consider implementing this latter instead.
include::../description.adoc[]

== Noncompliant Code Example

----
if (true)
{
  DoSomething();
}
...
if (false)
{
  DoSomethingElse();
}
----

== Compliant Solution

----
DoSomething();
...
----

include::../see.adoc[]
