include::../description.adoc[]

== Noncompliant Code Example

----
if (true) 
{  
  DoSomething(); 
}
...
if (false) 
{
  DoSomethingElse(); 
}
----

== Compliant Solution

----
DoSomething(); 
...
----

include::../see.adoc[]

ifdef::env-github,rspecator-view[]
'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
