<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Because the evaluation order of <code>#</code> and <code>##</code> are not specified, the results of using them both in the same macro could be unpredictable. Therefore macros should contain at most once instance of either <code>#</code> or <code>##</code>.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">#define NonCompliant(a, b)  # a ## b
int main() {
  std::cout &lt;&lt; NonCompliant(Hello, World);
}</code></pre>
</div>
</div>
<div class="paragraph">
<p>The result of this code is unspecified. It will either print "HelloWorld" or trigger a compilation error. If ## is evaluated first this will print HelloWorld. If # is evaluated first this will cause a compilation error telling thatÂ <strong>"Hello"World</strong> is not a valid preprocessor token.</p>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">#define Stringfy(a) #a
#define Compliant(a, b)  Stringfy(a##b)

int main(){
  std::cout &lt;&lt; Compliant(Hello, World);
}</code></pre>
</div>
</div>
<div class="paragraph">
<p>This example will always print "HelloWorld".</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p>MISRA C:2004, 19.12</p>
</li>
<li>
<p>MISRA C&#43;&#43; 2008, 16-3-1</p>
</li>
<li>
<p>Related: MISRA C:2012, 20.11</p>
</li>
</ul>
</div>
<hr>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Edit this macro to use only a single instance of "<mark>" or a single instance of "</mark>#"</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_is_related_to_s962">is related to: <a data-rspec-id="S962" class="rspec-auto-link">S962</a></h3>

</div>
</div>
</div>