Variant functions are inefficient by nature, and should not be used with strings when string-specific versions, denoted by a ``++$++`` at the end of the name, are available.


This rule flags instances of these functions:

``++Left()++``, ``++Mid()++``, ``++Right()++``, ``++Chr()++``, ``++ChrW()++``, ``++UCase()++``, ``++LCase()++``, ``++LTrim()++``, ``++RTrim()++``, ``++Trim()++``, ``++Space()++``, ``++String()++``, ``++Format()++``, ``++Hex()++``, ``++Oct(),++`` ``++Str()++``, ``++Error++``


== Noncompliant Code Example

----
TrimString = Trim(TestString)
----


== Compliant Solution

----
TrimString = Trim$(TestString)
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

endif::env-github,rspecator-view[]
