<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>A selector parameter is a <code>boolean</code> parameter that&#8217;s used to determine which of two paths to take through a method. Specifying such a parameter may seem innocuous, particularly if it&#8217;s well-named.</p>
</div>
<div class="paragraph">
<p>Unfortunately, developers calling the method won&#8217;t see the parameter name, only its value. They&#8217;ll be forced either to guess at the meaning or to take extra time to look the method up. Additionally, selector parameters create maintenance challenges: adding a third option requires changing the boolean to an enum or similar type, breaking existing code.</p>
</div>
<div class="paragraph">
<p>This rule finds methods with a <code>boolean</code> that is used to determine which path to take through the method.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-text" data-lang="text">function feed(name: string, isHuman: boolean) {
  if (isHuman) {
    // implementation for human
  } else {
    // implementation for animal
  }
}

// Intent is not clear at call site
feed('Max', false); // does this mean not to feed Max?</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="paragraph">
<p>Instead, split the method into separate, clearly named methods:</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-text" data-lang="text">function feedHuman(name: string) {
  offerSushi(name);
}

function feedAnimal(name: string) {
  offerCarrot(name);
}

// Clear intent at call site
feedHuman("Joe");
feedAnimal("Max");</code></pre>
</div>
</div>
<div class="paragraph">
<p>Alternatively, use descriptive object parameters or types:</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-text" data-lang="text">type EntityType = 'human' | 'animal';

function feed(name: string, entityType: EntityType) {
  if (entityType === 'human') {
    // implementation for human
  } else {
    // implementation for animal
  }
}

feed('Max', 'animal');</code></pre>
</div>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-text" data-lang="text">function feed(name: string, { human = true } ) {
  if (human) {
    // implementation for human
  } else {
    // implementation for animal
  }
}

feed('Max', { human: false });</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Provide multiple methods instead of using "xxx" to determine which action to take.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p><a href="https://dev.to/thinkster/code-smell-selector-arguments-57ak" class="bare">https://dev.to/thinkster/code-smell-selector-arguments-57ak</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>