== How to fix it in Cryptodome

=== Code examples

==== Noncompliant code example

include::../../common/fix/aes-noncompliant-example.adoc[]

[source,python,diff-id=21,diff-type=noncompliant]
----
from Crypto.Cipher import AES     # pycryptodome
from Cryptodome.Cipher import AES # pycryptodomex

AES.new(key, AES.MODE_ECB)  # Noncompliant
----

include::../../common/fix/rsa-noncompliant-example.adoc[]

[source,python,diff-id=22,diff-type=noncompliant]
----
from Crypto.Cipher import PKCS1_V1_5     # pycryptodome
from Cryptodome.Cipher import PKCS1_V1_5 # pycryptodomex

PKCS1_v1_5.new(key) # Noncompliant
----


==== Compliant solution

include::../../common/fix/aes-compliant-example.adoc[]

[source,python,diff-id=21,diff-type=compliant]
----
from Crypto.Cipher import AES     # pycryptodome
from Cryptodome.Cipher import AES # pycryptodomex

AES.new(key, AES.MODE_GCM)
----

include::../../common/fix/rsa-compliant-example.adoc[]

[source,python,diff-id=22,diff-type=compliant]
----
from Crypto.Cipher import PKCS1_V1_5     # pycryptodome
from Cryptodome.Cipher import PKCS1_V1_5 # pycryptodomex

PKCS1_OAEP.new(key)
----

=== How does this work?

include::../../common/fix/fix.adoc[]
