== How to fix it in pyDes

=== Code examples

==== Noncompliant code example

[source,python,diff-id=31,diff-type=noncompliant]
----
import pyDes

pyDes.des(key) # Noncompliant
----

==== Compliant solution

Since pyDes only provides DES, it is recommended to use another library like pyca.

[source,python,diff-id=31,diff-type=compliant]
----
from cryptography.hazmat.primitives.ciphers import (
    Cipher,
    algorithms,
    modes,
)
from cryptography.hazmat.backends import default_backend

Cipher(algorithms.AES(key), modes.GCM(iv), backend=default_backend())
----

=== How does this work?

include::../../common/fix/fix.adoc[]
