<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Classes with only <code>private</code> constructors should be marked <code>final</code> to prevent any mistaken extension attempts.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">public class PrivateConstructorClass {  // Noncompliant
  private PrivateConstructorClass() {
    // ...
  }

  public static int magic(){
    return 42;
  }
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">public final class PrivateConstructorClass {  // Compliant
  private PrivateConstructorClass() {
    // ...
  }

  public static int magic(){
    return 42;
  }
}</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Make this class "final" or add a public constructor.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_27_may_2015_084214_freddy_mallet_wrote">on 27 May 2015, 08:42:14 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>My personal feedback:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>I would not activate this rule by default as it is controversial &#8594; some guys are going to say that it&#8217;s useless to flag such classes as final as any way it&#8217;s by design not possible to extend them</p>
</li>
<li>
<p>Due to the previous point I would associate this rule to the SQALE characteristic "Maintainability &gt; Understandability" and not "Changeability - Architecture related changeability"</p>
</li>
</ul>
</div>
</div>
<div class="sect2">
<h3 id="_on_27_may_2015_191627_ann_campbell_wrote">on 27 May 2015, 19:16:27 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>Rule updated</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_1_jun_2015_130632_nicolas_peru_wrote">on 1 Jun 2015, 13:06:32 Nicolas Peru wrote:</h3>
<div class="paragraph">
<p>ok.</p>
</div>
</div>
</div>
</div>