Shared coding conventions allow teams to collaborate efficiently, and consistently spaced code has a strong impact on readability. This rule checks that the number of spaces between a data item's level and its name are consistent. 


== Noncompliant Code Example

[source,text]
----
01 StudentDetails.
   05   StudentId        PIC 9(7).     *< Noncompliant
   05 StudentName. 
      07 FirstName     PIC X(10).
      07  MiddleInitial PIC X.           *< Noncompliant
      07  Surname       PIC X(15).   *< Noncompliant
----


== Compliant Solution

[source,text]
----
01 StudentDetails.
   05 StudentId        PIC 9(7). 
   05 StudentName. 
      07 FirstName     PIC X(10).
      07 MiddleInitial PIC X. 
      07 Surname       PIC X(15).
----

ifdef::env-github,rspecator-view[]

'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]

endif::env-github,rspecator-view[]
