Shared coding conventions allow teams to collaborate efficiently, and consistently spaced code has a strong impact on readability. This rule checks that the number of spaces between a data item's level and its name are consistent. 


== Noncompliant Code Example

----
01 StudentDetails.
   05   StudentId        PIC 9(7).     *< Noncompliant
   05 StudentName. 
      07 FirstName     PIC X(10).
      07  MiddleInitial PIC X.           *< Noncompliant
      07  Surname       PIC X(15).   *< Noncompliant
----


== Compliant Solution

----
01 StudentDetails.
   05 StudentId        PIC 9(7). 
   05 StudentName. 
      07 FirstName     PIC X(10).
      07 MiddleInitial PIC X. 
      07 Surname       PIC X(15).
----

