The path provided here doesn't match the actual path on this file system (e.g. the case is different). While this may work on a particular environment, this is not portable and may fail on a different environment.

== Noncompliant Code Example

----
#include "Foo.h" // Noncompliant, the file name is "foo.h"
----
