<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>While it is technically correct to assign to parameters from within function bodies, it is better to use temporary variables to store intermediate results.</p>
</div>
<div class="paragraph">
<p>Allowing parameters to be assigned to also reduces the code readability as developers will not be able to know whether the original parameter or some temporary variable is being accessed without going through the whole function.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">int glob = 0;
void function (int a) {
  a = glob; // Noncompliant
  ...
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">int glob = 0;
void function (int a) {
  int b = glob;
  ...
}</code></pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p>MISRA C:2012, 17.8 - A function parameter should not be modified</p>
</li>
</ul>
</div>
<hr>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Introduce a new variable or use its initial value before reassigning "xxx".</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>reassignment of var</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_14_oct_2013_125040_dinesh_bolkensteyn_wrote">on 14 Oct 2013, 12:50:40 Dinesh Bolkensteyn wrote:</h3>
<div class="paragraph">
<p>+ constructor + foreach</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_14_oct_2013_131026_dinesh_bolkensteyn_wrote">on 14 Oct 2013, 13:10:26 Dinesh Bolkensteyn wrote:</h3>
<div class="paragraph">
<p>constructors are covered, but not foreach</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_14_oct_2013_131145_dinesh_bolkensteyn_wrote">on 14 Oct 2013, 13:11:45 Dinesh Bolkensteyn wrote:</h3>
<div class="paragraph">
<p>Implemented by http://jira.codehaus.org/browse/SONARJAVA-349</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_15_oct_2013_123419_dinesh_bolkensteyn_wrote">on 15 Oct 2013, 12:34:19 Dinesh Bolkensteyn wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] I&#8217;m slightly changing the title, replacing "overwritten with new assignments" by "reassigned". I had to add a new use case (foreach variables), I guess it&#8217;s worth to be mentioned in the title, but then we must keep the title&#8217;s length reasonable.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_16_oct_2013_232949_ann_campbell_wrote">on 16 Oct 2013, 23:29:49 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>Works for me, [~dinesh.bolkensteyn]</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_13_sep_2014_093057_evgeny_mandrikov_wrote">on 13 Sep 2014, 09:30:57 Evgeny Mandrikov wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] probably relates to MISRA C:2012, 17.8. And so applicable for C.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_15_sep_2014_083830_ann_campbell_wrote">on 15 Sep 2014, 08:38:30 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>Thanks [~evgeny.mandrikov]</p>
</div>
</div>
</div>
</div>