== Why is this an issue?

The `ResultSet.isLast()` method in package `java.sql` returns `true` if the cursor is on the last row of the `ResultSet`.
There are several reasons to avoid using this method:

1. It is optionally available only for result sets of type `ResultSet.TYPE_FORWARD_ONLY`.
   Database drivers will throw an exception if not supported.
2. The method can be expensive to execute as the database driver may need to fetch ahead one row to determine whether the current row is the last in the result set.
   The documentation of the method explicitly mentions this fact.
3. What "the cursor is on the last row" means for an empty `ResultSet` is unclear.
   Database drivers may return `true` or `false` in this case .

`ResultSet.next()` is a good alternative to `ResultSet.isLast()` as it does not have the mentioned issues.
It is always supported and, as per specification, returns `false` for empty ResultSets.

== How to fix it

Replace uses of `ResultSet.isLast()` with `ResultSet.next()`.

=== Code examples

==== Noncompliant code example

[source,java,diff-id=1,diff-type=noncompliant]
----
ResultSet rs = stmt.executeQuery("SELECT name, address FROM PERSON");
while (!rs.isLast()) { // Noncompliant
  // process row
}
----

==== Compliant solution

[source,java,diff-id=1,diff-type=compliant]
----
ResultSet rs = stmt.executeQuery("SELECT name, address FROM PERSON");
while (rs.next()) { // Compliant
  // process row
}
----

== Resources

=== Documentation

* https://docs.oracle.com/javase/8/docs/api/java/sql/ResultSet.html#isLast--[Java SE 8 API Specification: ResultSet.isLast()]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Remove this call to "isLast()".


'''
== Comments And Links
(visible only on this page)

=== on 24 Nov 2014, 19:15:41 Nicolas Peru wrote:
No message. 

Otherwise seems ok.

endif::env-github,rspecator-view[]
