== Why is this an issue?

Signal handling contains implementation-defined and undefined behavior.


=== Noncompliant code example

[source,cpp]
----
#include <signal.h> /* Noncompliant */
----


== Resources

* MISRA C:2004, 20.8  - The signal handling facilities of <signal.h> shall not be used.
* MISRA C:2012, 21.5 - The standard header file <signal.h> shall not be used


ifdef::env-github,rspecator-view[]
'''
== Comments And Links
(visible only on this page)

=== is related to: S1054

=== on 31 Mar 2015, 19:07:23 Evgeny Mandrikov wrote:
\[~ann.campbell.2] implementation seems more complete (SQALE, description) than this spec.

endif::env-github,rspecator-view[]
