include::../rationale.adoc[]

include::../how-to-fix-it.adoc[]

=== Code examples

==== Noncompliant code example

[source,abap,diff-id=1,diff-type=noncompliant]
----
FUNCTION f.
  DATA: LOCAL_1 LIKE BAR.
  DATA: LOCAL_2 LIKE BAR. "Noncompliant - LOCAL_2 is unused

  SELECT * FROM LOCAL_1.

ENDFUNCTION.
----

==== Compliant solution

[source,abap,diff-id=1,diff-type=compliant]
----
FUNCTION f.
  DATA: LOCAL_1 LIKE BAR.

  SELECT * FROM LOCAL_1.

ENDFUNCTION.
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
