include::../description.adoc[]

== Noncompliant Code Example

[source,abap]
----
FUNCTION f.
  DATA: LOCAL_1 LIKE BAR.
  DATA: LOCAL_2 LIKE BAR. "Noncompliant

  SELECT * FROM LOCAL_1.

ENDFUNCTION.
----

== Compliant Solution

[source,abap]
----
FUNCTION f.
  DATA: LOCAL_1 LIKE BAR.

  SELECT * FROM LOCAL_1.

ENDFUNCTION.
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
