== How to fix it in cURL

=== Code examples

==== Noncompliant code example

[source,bash,diff-id=1,diff-type=noncompliant]
----
#!/bin/bash
curl --tlsv1.0 -O https://tlsv1-0.example.com/downloads/install.sh
----

==== Compliant solution

[source,bash,diff-id=1,diff-type=compliant]
----
#!/bin/bash
curl --tlsv1.2 -O https://tlsv1-3.example.com/downloads/install.sh
----

=== How does this work?

include::../../common/fix/fix.adoc[]
