== How to fix it in Wget

=== Code examples

==== Noncompliant code example

[source,bash,diff-id=11,diff-type=noncompliant]
----
#!/bin/bash
wget --secure-protocol TLSv1_1 https://example.com/downloads/install.sh
----

==== Compliant solution

[source,bash,diff-id=11,diff-type=compliant]
----
#!/bin/bash
wget --secure-protocol TLSv1_2 https://example.com/downloads/install.sh
----

=== How does this work?

include::../../common/fix/fix.adoc[]
