<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when a SOQL query contains multiple clauses on a single line, making it difficult to read and understand.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>SOQL queries are an essential part of Apex development, and they can quickly become complex with multiple clauses like SELECT, FROM, WHERE, ORDER BY, and LIMIT. When these clauses are written on a single line, the query becomes harder to read, understand, and maintain.</p>
</div>
<div class="paragraph">
<p>Poor formatting makes it difficult to:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>Quickly identify the different parts of the query</p>
</li>
<li>
<p>Review changes during code reviews</p>
</li>
<li>
<p>Debug issues when queries don&#8217;t return expected results</p>
</li>
<li>
<p>Maintain consistency across the codebase</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>Well-formatted SOQL queries follow a structured approach where each logical section is placed on its own line with proper indentation. This makes the query&#8217;s structure immediately clear and helps developers understand what data is being retrieved and under what conditions.</p>
</div>
<div class="paragraph">
<p>Consistent formatting also improves team collaboration, as all developers can quickly understand and modify queries written by their colleagues.</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="paragraph">
<p>Poor SOQL formatting reduces code readability and maintainability. While it doesn&#8217;t directly impact application functionality, it can slow down development, make code reviews less effective, and increase the likelihood of introducing bugs when modifying queries.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Break the SOQL query into multiple lines, placing each logical clause (SELECT, FROM, WHERE, etc.) on its own line. Align field names and use consistent indentation for better readability.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-apex" data-lang="apex">List&lt;Account&gt; accounts = [SELECT Id, Name, BillingCity FROM Account WHERE Name LIKE '%ApexHours%' ORDER BY Name LIMIT 10]; // Noncompliant</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-apex" data-lang="apex">List&lt;Account&gt; accounts = [
    SELECT Id,
           Name,
           BillingCity
    FROM Account
    WHERE Name LIKE '%ApexHours%'
    ORDER BY Name
    LIMIT 10
];</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>SOQL and SOSL Queries - <a href="https://developer.salesforce.com/docs/atlas.en-us.apexcode.meta/apexcode/langCon_apex_SOQL.htm">Official Salesforce documentation on SOQL queries in Apex</a></p>
</li>
<li>
<p>Apex Code Best Practices - <a href="https://developer.salesforce.com/docs/atlas.en-us.apexcode.meta/apexcode/apex_bestpractices.htm">Salesforce best practices for writing clean and maintainable Apex code</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>