You should set a property of type `DateTime` or `DateTimeOffset` as the primary key of a table only if the date values are guaranteed to be unique.
The rule raises an issue if:
* Entity Framework, or Entity Framework Core dependencies are found; 
* a class contains a `DateTime` or `DateTimeOffset` property that is either:
** named `Id` or `<type name>Id`.
** decorated by the `[Key]` attribute.

== Why is this an issue?

Using a type `DateTime` or `DateTimeOffset` column as the primary key of a table is slightly risky. Two threads could create new objects in the table close enough in sequence for both to have the same time. 
`DateTime` type is also subject to changes such as daylight saving time (DST) transitions, which can also result in duplicate key exceptions. 

== How to fix it

Either use a GUID or the auto generated ID as a primary key.

=== Code examples

==== Noncompliant code example

[source,csharp,diff-id=1,diff-type=noncompliant]
----
internal class Account
{
    public DateTime Id { get; set; }

    public string Name { get; set; }
    public string Surname { get; set; }
}
----

==== Compliant solution

[source,csharp,diff-id=1,diff-type=compliant]
----
internal class Account
{
    public int Id { get; set; }

    public string Name { get; set; }
    public string Surname { get; set; }
}
----

or

==== Noncompliant code example

[source,csharp,diff-id=2,diff-type=noncompliant]
----
internal class Person
{
    [Key]
    public DateTime BirthDate { get; set; }

    public string Name { get; set; }
    public string Surname { get; set; }
}
----

==== Compliant solution

[source,csharp,diff-id=2,diff-type=compliant]
----
internal class Person
{
    public int Id { get; set; }
	
    public DateTime BirthDate { get; set; }
    public string Name { get; set; }
    public string Surname { get; set; }
}
----

== Resources

=== Documentation

* https://learn.microsoft.com/en-us/ef/core/modeling/keys?tabs=data-annotations[Entity Framework keys and data annotation]
* https://learn.microsoft.com/it-it/dotnet/api/system.guid?view=net-7.0[GUID]