include::../dotnet/introduction.adoc[]

== Why is this an issue?

include::../dotnet/why.adoc[]

== How to fix it

include::../dotnet/how.adoc[]

=== Code examples

==== Noncompliant code example

[source,csharp,diff-id=1,diff-type=noncompliant]
----
internal class Account
{
    public DateTime Id { get; set; }

    public string Name { get; set; }
    public string Surname { get; set; }
}
----

==== Compliant solution

[source,csharp,diff-id=1,diff-type=compliant]
----
internal class Account
{
    public Guid Id { get; set; }

    public string Name { get; set; }
    public string Surname { get; set; }
}
----

or

==== Noncompliant code example

[source,csharp,diff-id=2,diff-type=noncompliant]
----
internal class Person
{
    [Key]
    public DateTime BirthDate { get; set; }

    public string Name { get; set; }
    public string Surname { get; set; }
}
----

==== Compliant solution

[source,csharp,diff-id=2,diff-type=compliant]
----
internal class Person
{
    public Guid Id { get; set; }
	
    public DateTime BirthDate { get; set; }
    public string Name { get; set; }
    public string Surname { get; set; }
}
----

== Resources

include::../dotnet/resources.adoc[]