== How to fix it in .NET

=== Code examples

include::../../common/fix/code-rationale.adoc[]

==== Noncompliant code example

[source,csharp]
----
using System.Net;

public void send()
{
    ServicePointManager.ServerCertificateValidationCallback +=
        (sender, certificate, chain, errors) => {
            return true; // Noncompliant
        };
}
----

=== How does this work?

include::../../common/fix/validation.adoc[]
