== How to fix it in System.Net

=== Code examples

include::../../common/fix/code-rationale.adoc[]

==== Noncompliant code example

[source,csharp,diff-id=1,diff-type=noncompliant]
----
ServicePointManager.ServerCertificateValidationCallback +=
    (sender, certificate, chain, errors) => {
        return true; // Noncompliant
    };
----

==== Compliant solution

TODO: you should not try to implement your own validation

=== How does this work?

include::../../common/fix/default-implementation.adoc[]
