In a Zen-like manner, "NULL" is never equal to anything, even itself. Therefore comparisons using equality operators will always return ``++False++``, even when the value actually ``++IS NULL++``.


For that reason, comparison operators should never be used to make comparisons with ``++NULL++``; ``++IS NULL++`` and ``++IS NOT NULL++`` should be used instead.

== Noncompliant Code Example

----
UPDATE books
SET title = 'unknown'
WHERE title = NULL -- Noncompliant 
----

== Compliant Solution

----
UPDATE books
SET title = 'unknown'
WHERE title IS NULL
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
