There is no need to declare the same dependency or plugin twice in a project. In fact, doing so is likely to cause errors in the future when maintainers try to change or upgrade the plugin or dependency.

== Noncompliant Code Example

----
    <dependency>
      <groupId>com.mygroup</groupId>
      <artifactId>myartifact</artifactId>
      <version>1.0</version>
      <scope>runtime</scope>
    </dependency>
    <!-- ... -->
    <dependency>  <!-- Noncompliant -->
      <groupId>com.mygroup</groupId>
      <artifactId>myartifact</artifactId>
      <version>1.0</version>
      <type>jar</type>
    </dependency>
----
