== Why is this an issue?

There is no need to declare the same dependency or plugin twice in a project. In fact, doing so is likely to cause errors in the future when maintainers try to change or upgrade the plugin or dependency.


=== Noncompliant code example

[source,java]
----
    <dependency>
      <groupId>com.mygroup</groupId>
      <artifactId>myartifact</artifactId>
      <version>1.0</version>
      <scope>runtime</scope>
    </dependency>
    <!-- ... -->
    <dependency>  <!-- Noncompliant -->
      <groupId>com.mygroup</groupId>
      <artifactId>myartifact</artifactId>
      <version>1.0</version>
      <type>jar</type>
    </dependency>
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Remove this [dependency|plugin] declaration, it duplicates the one on line n.


=== Highlighting

* Primary: <dependency>|<plugin>
* Secondary: through to </dependency>|<plugin>


'''
== Comments And Links
(visible only on this page)

=== on 19 Jan 2016, 11:49:16 Ann Campbell wrote:
IntelliJ Duplicate Dependencies, Duplicate plugin declaration

endif::env-github,rspecator-view[]
