<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>When inheriting from a class, if you need to inherit its constructors without additional initialization you should prefer <code>using-declaration</code> to inherit all base class&#8217;s constructors instead of writing them by hand.</p>
</div>
<div class="paragraph">
<p><code>using-declaration</code> for inheriting constructor is a C&#43;&#43;11 feature that makes all constructors of the base visible to the overload resolution when initializing the derived class.</p>
</div>
<div class="paragraph">
<p>If you need to change the accessibility of one of the inherited constructors, you can do it by keeping the <code>using-declaration</code> and declaring that constructor explicitly as private:</p>
</div>
<div class="listingblock">
<div class="content">
<pre>class Base {
public:
  Base(int p) {}
  Base(int p1, int p2) {}
};

class Derived : public Base {
  using Base::Base;
  Derived(int p1, int p2);  // Changes constructor to private accessibility
};

int f(){
  Derived b(1); // Base(int p) is visible when initializing the derived class
  Derived b1(1,2); // Compilation error: Base(int p1, int p2) is not visible when initializing the derived class
}</pre>
</div>
</div>
<div class="paragraph">
<p>This rule raises an issue when a constructor inherits the base class constructor without requiring any additional initialization.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">class Base {
public:
  Base(int p) {}
};

class Derived : public Base {
public:
  Derived(int p) : Base(p) {} // Noncompliant
};</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">class Base {
public:
  Base(int p) {}
};

class Derived : public Base {
  using Base::Base;
};</code></pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p>C++ Core Guidelines - <a href="https://github.com/isocpp/CppCoreGuidelines/blob/e49158a/CppCoreGuidelines.md#c52-use-inheriting-constructors-to-import-constructors-into-a-derived-class-that-does-not-need-further-explicit-initialization">C.52: Use inheriting constructors to import constructors into a derived class that does not need further explicit initialization</a></p>
</li>
</ul>
</div>
<hr>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Replace inherited constructors without additional initialization by a "using" declaration.</p>
</div>
</div>
</div>
</div>