=== How to fix it in Knex

include::../../common/fix/code-rationale.adoc[]

==== Noncompliant code example

[source,javascript,diff-id=1,diff-type=noncompliant]
----
const pg = require('knex')({
  client: 'pg',
  connection: process.env.PG_CONNECTION_STRING,
  searchPath: ['knex', 'public'],
});

function (req, res) {
    const user = await knex('users')
        .whereRaw(`user = '${req.query.user}' and pass = '${req.query.pass}'`);

    res.send(JSON.stringify(user));
    res.end();
}
----

==== Compliant solution

[source,javascript,diff-id=1,diff-type=compliant]
----
const pg = require('knex')({
  client: 'pg',
  connection: process.env.PG_CONNECTION_STRING,
  searchPath: ['knex', 'public'],
});

function (req, res) {
    const user = await knex
        .select('users')
        .where('user', user)
        .where('pass', pass);

    res.send(JSON.stringify(user));
    res.end();
}
----

=== How does this work?

:secure_feature: Knex
:unsafe_function: whereRaw()
include::../../common/fix/secure-by-design.adoc[]

