=== How to fix it in Knex

include::../../common/fix/code-rationale.adoc[]

==== Noncompliant code example

[source,javascript,diff-id=1,diff-type=noncompliant]
----
async function index(req, res) {
    const knex = req.app.get('knex');

    let user = await knex('users')
        .whereRaw(`user = '${req.query.user}' and pass = '${req.query.pass}'`);

    res.send(JSON.stringify(user));
    res.end();
}
----

==== Compliant solution

[source,javascript,diff-id=1,diff-type=compliant]
----
async function index(req, res) {
    const knex = req.app.get('knex');
        .select('users')
        .where('user', user)
        .where('pass', pass);

    res.send(JSON.stringify(user));
    res.end();
}
----

=== How does this work?

:secure_feature: Knex
:unsafe_function: whereRaw()
include::../../common/fix/secure-by-design.adoc[]

