=== How to fix it in Sequelize

include::../../common/fix/code-rationale.adoc[]

==== Noncompliant code example

[source,javascript,diff-id=1,diff-type=noncompliant]
----
const sequelize = require("sequelize");

const db = new sequelize();

function (req, res) {
    user = req.query.user;
    pass = req.query.pass;

    const query = "SELECT * FROM users WHERE user = '" + user + "' AND pass = '" + pass + "'"; // Noncompliant

    db.query(query);

}
----

==== Compliant solution

[source,javascript,diff-id=1,diff-type=compliant]
----
const sequelize = require("sequelize");

const db = new sequelize();

function (req, res) {
    user = req.query.user;
    pass = req.query.pass;

    const query = "SELECT * FROM users WHERE user = $user AND pass = $pass";

    db.query(
        query,
        {
            bind: {
                user: user,
                pass: pass,
            },
            type: QueryTypes.SELECT
        }
     );
}
----

=== How does this work?

include::../../common/fix/prepared-statements.adoc[]
