<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>The public API of a framework, plugin, or library is the way its provider intended it to be used.
API stability and compatibility (within the same major version number) of a library are guaranteed only for its public API.</p>
</div>
<div class="paragraph">
<p>Internal APIs are mere implementation details and are prone to breaking changes as the implementation of the library changes.
No guarantees are being made about them. Therefore, users should not use internal APIs, even when visible.</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="sect3">
<h4 id="_code_stability">Code Stability</h4>
<div class="paragraph">
<p>If not fixed, your code might break when the library is upgraded to a new version, even if only the minor version number or the patch number changes.</p>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Replace internal API usage with the public API designed for your use case.
This may imply a refactoring of the affected code if no one-to-one replacement is available in the public API.
If a specific functionality is required, copying the required parts of the implementation into your code may even be better than using the internal API.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-javascript" data-lang="javascript">import { _parseWith } from './node_modules/foo/helpers'</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-javascript" data-lang="javascript">import { parse } from 'foo'</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>