include::../summary.adoc[]

== Why is this an issue?

include::../description.adoc[]

=== Exceptions

The rule does not raise an issue on statements containing only a semicolon (``++;++``).

== How to fix it

include::../how-to-fix.adoc[]

=== Code examples

==== Noncompliant code example

[source,php]
----
function getResult() {
    $result = 42;
    if (shouldBeZero()) {
        $result == 0; // Noncompliant: no side effect, was an assignment intended?
    }
    return $result;
}
----

==== Compliant solution

[source,php]
----
function getResult() {
    $result = 42;
    if (shouldBeZero()) {
        $result = 0; // Compliant
    }
    return $result;
}
----


include::../see.adoc[]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
