<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when hardcoded platform-specific temporary directory paths like <code>/tmp/</code> are used instead of cross-platform alternatives.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Hardcoding platform-specific paths like <code>/tmp/</code> creates portability issues because these paths don&#8217;t exist on all operating systems. The <code>/tmp/</code> directory is common on Unix-like systems (Linux, macOS) but doesn&#8217;t exist on Windows, where temporary files are typically stored in locations like <code>C:\Users\Username\AppData\Local\Temp\</code>.</p>
</div>
<div class="paragraph">
<p>When code with hardcoded Unix paths runs on Windows, it will fail at runtime because the operating system cannot find the specified directory. This leads to application crashes or unexpected behavior when the same codebase is deployed across different platforms.</p>
</div>
<div class="paragraph">
<p>Go provides built-in functions specifically designed to handle cross-platform path operations. The <code>os.TempDir()</code> function returns the appropriate temporary directory for the current operating system, while <code>filepath.Join()</code> ensures proper path separators are used regardless of the platform.</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="paragraph">
<p>Applications will fail at runtime when deployed on operating systems that don&#8217;t have the hardcoded directory structure. This can cause crashes, prevent proper functionality, and create deployment issues across different environments.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Replace hardcoded temporary directory paths with Go&#8217;s cross-platform functions. Use <code>os.TempDir()</code> to get the system&#8217;s temporary directory and <code>filepath.Join()</code> to construct paths properly.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-go" data-lang="go">portFile := fmt.Sprintf("/tmp/testnextnethttp_port_%d", pid) // Noncompliant</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-go" data-lang="go">portFile := filepath.Join(os.TempDir(), fmt.Sprintf("testnextnethttp_port_%d", pid))</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Go os.TempDir() documentation - <a href="https://pkg.go.dev/os#TempDir">Official documentation for the os.TempDir() function</a></p>
</li>
<li>
<p>Go filepath.Join() documentation - <a href="https://pkg.go.dev/path/filepath#Join">Official documentation for the filepath.Join() function</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>