<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>When a SQL query is joining n tables (with n&gt;=2), it is expected to have join conditions defined to determine on which columns these n tables should be joined. At minimum, for n joined tables, the SQL query should contain (n-1) join conditions involving all the joined table to avoid a full cartesian product between the rows of the n tables.</p>
</div>
<div class="paragraph">
<p>Not doing so will imply that too many rows will be returned. If this is not the case and unless this has been done on purpose, the SQL query should be reviewed and missing conditions should be added or useless tables should be removed from the SQL query.</p>
</div>
<div class="paragraph">
<p>This rule is raising no issue when the SQL query is involving <code>CROSS JOIN</code>, <code>NATURAL JOIN</code> statements.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-sql" data-lang="sql">SELECT c.id, c.name, o.id, o.item_id, o.item_quantity
  FROM ORDERS o, CUSTOMERS c; -- Noncompliant; no JOIN condition at all

SELECT c.id, c.name, o.id, o.item_id, o.item_quantity
  FROM ORDERS o
  JOIN CUSTOMERS c ON o.customer_id = o.id; -- Noncompliant; no condition related to CUSTOMERS

SELECT f.name, d.title, l.*
  FROM FOLDERS f, DOCUMENTS d, DOC_LINES l -- Noncompliant; missing at least one condition related to DOC_LINES
 WHERE f.id = d.folder_id;</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-sql" data-lang="sql">SELECT c.id, c.name, o.id, o.item_id, o.item_quantity
  FROM ORDERS o, CUSTOMERS c
 WHERE o.customer_id = c.id; -- Compliant

SELECT c.id, c.name, o.id, o.item_id, o.item_quantity
  FROM ORDERS o
  JOIN CUSTOMERS c ON o.customer_id = c.id; -- Compliant

SELECT f.name, d.title, l.*
  FROM FOLDERS f, DOCUMENTS d, DOC_LINES l
 WHERE f.id = d.folder_id
   AND d.id = l.document_id; -- Compliant</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Add a condition for XXX (table) or remove this table from the SQL query.</p>
</div>
</div>
</div>
</div>