== Why is this an issue?

Unused parameters are misleading. Whatever the value passed to such parameters is, the behavior will be the same.

=== Noncompliant code example

[source,php]
----
function doSomething($a, $b) { // "$a" is unused
  return compute($b);
}
----

=== Compliant solution

[source,php]
----
function doSomething($b) {
  return compute($b);
}
----

=== Exceptions

Functions in classes that override a class or implement interfaces are ignored.

----
class C extends B {

  function doSomething($a, $b) {     // no issue reported on $b
    compute($a);
  }

}
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
