== Why is this an issue?

include::../description.adoc[]

=== Exceptions

Functions in classes that override a class or implement interfaces are ignored.

[source,php]
----
class C extends B {

  function doSomething($a, $b) {     // no issue reported on $b
    compute($a);
  }

}
----

Additionally, methods with specific Magento plugin signatures are ignored because Magento requires these parameters:

* Methods starting with `before` and having exactly 3 parameters
* Methods starting with `around` and having exactly 4 parameters
* Methods starting with `after` and having exactly 2 parameters

[source,php]
----
class ExamplePlugin {

  public function beforeSomeMethod($subject, $arg1, $arg2) {
    // no issue on unused parameters, required by Magento plugin signature
    return;
  }

  public function aroundSomeMethod($subject, callable $proceed, $arg1, $arg2) {
    // no issue on unused parameters, required by Magento plugin signature
    return;
  }

  public function afterSomeMethod($subject, $result) {
    // no issue on unused parameters, required by Magento plugin signature
    return;
  }

}
----

== How to fix it

include::../how-to-fix-it.adoc[]

=== Code examples

==== Noncompliant code example

[source,php,diff-id=1,diff-type=noncompliant]
----
function doSomething($a, $b) { // "$a" is unused
  return compute($b);
}
----

==== Compliant solution

[source,php,diff-id=1,diff-type=compliant]
----
function doSomething($b) {
  return compute($b);
}
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
