<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This is an issue when using a type named <code>scoped</code> in lambda parameter lists in C# 14 or later, where the compiler treats both the modifier and type name as modifiers, causing compilation errors.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>In C# 14, the language introduced the ability to write lambdas with parameter modifiers without specifying parameter types. As part of this enhancement, a breaking change was implemented where <code>scoped</code> is always treated as a modifier in lambda parameter lists.</p>
</div>
<div class="paragraph">
<p>When you have a type named <code>scoped</code> and try to use it with the <code>scoped</code> modifier in a lambda parameter, the compiler interprets both tokens as modifiers rather than recognizing one as a type name. This creates a syntax error because the compiler expects a type name after the modifier.</p>
</div>
<div class="paragraph">
<p>This breaking change affects existing code that was valid in earlier C# versions, particularly when upgrading projects to C# 14. The issue is specific to lambda expressions and doesn&#8217;t affect other contexts where the <code>scoped</code> type name might be used.</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="paragraph">
<p>This issue prevents code compilation when upgrading to C# 14. Projects that previously compiled successfully will fail to build, blocking development and deployment processes until the code is updated.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Escape the type name <code>scoped</code> with the <code>@</code> prefix to distinguish it from the <code>scoped</code> modifier. This tells the compiler to treat it as an identifier rather than a keyword.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-csharp" data-lang="csharp">var lambda = (scoped scoped parameter) =&gt; { /* ... */ }; // Noncompliant

ref struct scoped { }</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-csharp" data-lang="csharp">var lambda = (scoped @scoped parameter) =&gt; { /* ... */ };

ref struct scoped { }</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Breaking changes in Roslyn after .NET 9.0.100 through .NET 10.0.100 - <a href="https://learn.microsoft.com/en-us/dotnet/csharp/whats-new/breaking-changes/compiler%20breaking%20changes%20-%20dotnet%2010">Official Microsoft documentation detailing the breaking change where 'scoped' is always treated as a modifier in lambda parameter lists</a></p>
</li>
<li>
<p>C# Keywords - <a href="https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/">Documentation about C# keywords and how to use the @ prefix to escape them as identifiers</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>