<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>To perform calculations when specifying a CSS property <code>calc()</code> function can be used. This function takes single expression as parameter. When writing this expression some rules must be respected:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>no empty <code>calc()</code></p>
</li>
<li>
<p>there should be an operator between the arguments</p>
</li>
<li>
<p>there should not be any division by zero</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>Otherwise <code>calc()</code> function will be invalid and the entire rule using it will be ignored.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-css" data-lang="css">.btn {
  border: solid black 1px;
  width: calc(100% 80px);  /* Noncompliant */
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-css" data-lang="css">.btn {
  border: solid black 1px;
  width: calc(100% - 80px);
}</code></pre>
</div>
</div>
</div>
</div>
</div>