== Why is this an issue?

``++SYNCHRONIZE++`` was introduced by Oracle as a recovery mechanism in case there is a loss of synchronization between what is in memory and what is displayed. It works but it comes with a price; each call to ``++SYNCHRONIZE++`` generates a network round-trip between the server and the Forms client. Most of the time it should be avoided or used with caution as explicitly stated in the Oracle Forms documentation.


=== Noncompliant code example

[source,sql]
----
SYNCHRONIZE;
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Reconsider the use of "SYNCHRONIZE".


endif::env-github,rspecator-view[]
