include::../description.adoc[]

== Noncompliant Code Example

[source,php]
----
/a[b]c/
----

== Compliant Solution

[source,php]
----
/abc/
----

include::../exceptions.adoc[]

include::../implementation.adoc[]
