== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,php]
----
/a[b]c/
----

=== Compliant solution

[source,php]
----
/abc/
----

include::../exceptions.adoc[]

include::../implementation.adoc[]
