<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when operators, commas, colons, semicolons, or braces lack proper spacing around them.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Proper spacing around operators and punctuation significantly improves code readability and maintainability. When operators like <code>=</code>, <code>+</code>, <code>-</code>, <code>*</code> are cramped together without spaces, the code becomes harder to scan and understand quickly.</p>
</div>
<div class="paragraph">
<p>Consistent spacing follows widely accepted Ruby style conventions and makes code more professional. It helps developers:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>Parse expressions more easily by creating visual separation between operands and operators</p>
</li>
<li>
<p>Spot syntax errors more quickly</p>
</li>
<li>
<p>Maintain consistent formatting across the codebase</p>
</li>
<li>
<p>Reduce cognitive load when reading code</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>Without proper spacing, mathematical expressions and assignments can look cluttered and be prone to misreading. For example, <code>result=x+y*z</code> requires more mental effort to parse than <code>result = x + y * z</code>.</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="paragraph">
<p>Poor spacing reduces code readability and can slow down development and code review processes. While it doesn&#8217;t affect functionality, it makes the codebase harder to maintain and can lead to inconsistent formatting across the project.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Add spaces around operators, after commas, colons and semicolons, and around braces. This creates visual separation that makes the code easier to read and understand.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-ruby" data-lang="ruby">result=x+y*z
hash={key:value,other:data}
array=[1,2,3]</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-ruby" data-lang="ruby">result = x + y * z
hash = { key: value, other: data }
array = [1, 2, 3]</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Shopify Ruby Style Guide - Layout - <a href="https://ruby-style-guide.shopify.dev/#layout">Official Ruby style guide covering spacing conventions around operators and punctuation</a></p>
</li>
<li>
<p>RuboCop Layout/SpaceAroundOperators - <a href="https://docs.rubocop.org/rubocop/cops_layout.html#layoutspacearoundoperators">RuboCop cop that enforces spaces around operators</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>