This rule raises an issue when string slicing is used in condition expressions instead of the `startswith` or `endswith` methods.

== Why is this an issue?
Using the `startswith` and `endswith` methods in Python instead of string slicing offers several advantages:

1. **Readability and Intent:** Using `startswith` and `endswith` methods provides code that is more readable and self-explanatory. It clearly communicates your intention to check if a string starts or ends with a specific pattern. This makes the code more maintainable and easier to understand for other developers.

2. **Flexibility:** The `startswith` and `endswith` methods allow you to check for patterns of varying lengths. With string slicing, you would need to specify the exact length of the substring to compare. However, with the methods, you can pass in a pattern of any length, making your code more flexible and adaptable.

3. **Error Handling:** The methods handle edge cases automatically. If you pass a substring length that exceeds the length of the original string, slicing would raise an `IndexError` exception. On the other hand, the methods gracefully handle such cases and return `False`, avoiding any potential errors.

4. **Performance Optimization:** In some cases, using `startswith` and `endswith` methods can provide better performance. These methods are optimized and implemented in C, which can make them faster than manually slicing the string in Python. Although the performance gain might be negligible for small strings, it can be significant when working with large strings or processing them in a loop.

Overall, using `startswith` and `endswith` methods provides a cleaner, more readable, and error-resistant approach for checking if a string starts or ends with a specific pattern. It promotes code clarity, flexibility, and can potentially improve performance. This is also recommended by the https://peps.python.org/pep-0008/[PEP8] style guide.

== How to fix it

Use `startswith` and `endswith` methods instead of string slicing in condition expressions.

=== Code examples

==== Noncompliant code example

[source,python,diff-id=1,diff-type=noncompliant]
----
message = "Hello, world!"

if message[:5] == "Hello":
    ...

if message[-6:] == "world!":
    ...
----

==== Compliant solution 

[source,python,diff-id=1,diff-type=compliant]
----
message = "Hello, world!"

if message.startswith("Hello"):
    ...

if message.endswith("world!"):
    ...
----

== Resources

=== Documentation

* Python Documentation - https://docs.python.org/3/library/stdtypes.html#string-methods[String Methods]


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

* Use `startswith` here.
* Use `endswith` here.


=== Highlighting

Primary: condition

'''
== Comments And Links
(visible only on this page)


endif::env-github,rspecator-view[]
