<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>88-level variables, also known as "condition name" variables, each have a name, a value or set of values, and a "parent" variable. Those parent variables are called "conditional variables".</p>
</div>
<div class="paragraph">
<p>Each 88-level variable can be seen as a short-cut conditional for testing the value of the parent: <code>IF MY-88</code> will intrinsically return <code>true</code> if the parent value matches <code>MY-88</code>'s value, and <code>false</code> if it does not.</p>
</div>
<div class="paragraph">
<p>Thus, testing a conditional variable against a literal value is redundant and confusing. Just use the 88-levels instead.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cobol" data-lang="cobol">01 COLOR PIC X
  88 YELLOW VALUE 'Y'
  88 GREEN VALUE 'G'
  88 RED VALUE 'R'
...
IF COLOR = 'G' *&gt; Noncompliant
...
END-IF</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cobol" data-lang="cobol">01 COLOR PIC X
  88 YELLOW VALUE 'Y'
  88 GREEN VALUE 'G'
  88 RED VALUE 'R'
...
IF GREEN
...
END-IF</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove this comparison of "xxx" with "yyy" and simply test "zzz" instead.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>operator and both operands</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_29_mar_2016_103801_pierre_yves_nicolas_wrote">on 29 Mar 2016, 10:38:01 Pierre-Yves Nicolas wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] I added code samples.</p>
</div>
<div class="paragraph">
<p>The title should be changed to something like: "Variables which have 88-level variables should not be compared with literals".</p>
</div>
<div class="paragraph">
<p>I think the rule should not check only literal strings but also literal numbers.</p>
</div>
<div class="paragraph">
<p>The official documentations (e.g. <a href="http://www.ibm.com/support/knowledgecenter/SS6SG3_5.1.0/com.ibm.entcobol.doc_5.1/PGandLR/ref/rlddeva2.html">IBM</a>) have some precise names for these "variables":</p>
</div>
<div class="ulist">
<ul>
<li>
<p>88-level variables are called "condition names".</p>
</li>
<li>
<p>variables on which "condition names" are defined are called "conditional variables"</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>However, I&#8217;m not sure that developers use these names.</p>
</div>
</div>
</div>
</div>