<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when the <code>addError()</code> method is called with generic, non-descriptive error messages like "Error", "Invalid", or "Failed".</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Generic error messages provide no actionable information to users when validation fails. When a trigger or validation rule blocks a record save with a message like "Error", users are left confused about what went wrong and how to fix it.</p>
</div>
<div class="paragraph">
<p>Descriptive error messages are essential for good user experience. They should explain what validation failed and ideally provide guidance on how to resolve the issue. This reduces user frustration, decreases support tickets, and helps users complete their tasks successfully.</p>
</div>
<div class="paragraph">
<p>In Salesforce, error messages are often the primary way users understand why their data entry was rejected. A well-crafted error message can turn a frustrating experience into a learning opportunity that guides users toward the correct action.</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="paragraph">
<p>Generic error messages lead to poor user experience and increased support burden. Users may abandon their tasks, enter incorrect data repeatedly, or contact support for help understanding validation failures. This reduces productivity and increases operational costs.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Replace generic error messages with specific, descriptive messages that explain what went wrong and how to resolve the issue. Include relevant field values or context to help users understand the problem.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-apex" data-lang="apex">if (existing.containsKey(accountName)) {
    a.addError('Error'); // Noncompliant
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-apex" data-lang="apex">if (existing.containsKey(accountName)) {
    a.addError('Duplicate Account found: ' + a.Name + '. Please check for existing records before creating new ones.');
}</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Apex Trigger Exceptions - <a href="https://developer.salesforce.com/docs/atlas.en-us.apexcode.meta/apexcode/apex_triggers_exceptions.htm">Official Salesforce documentation on handling exceptions in Apex triggers</a></p>
</li>
<li>
<p>SObject addError Method - <a href="https://developer.salesforce.com/docs/atlas.en-us.apexref.meta/apexref/apex_methods_system_sobject.htm#apex_System_SObject_addError">Reference documentation for the addError method in Apex</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>