<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>In Dart, nullability is a part of the type system.
By default, any given type <code>T</code> is non-nullable.
If you append a "?" to the type, it becomes nullable: <code>T?</code>.</p>
</div>
<div class="paragraph">
<p>When accessing properties or functions of a nullable type, you need to handle the case when the target is <code>null</code>.
However, while accessing a non-nullable type, it is redundant to test for <code>null</code>, as the compiler statically ensures that the value can never be <code>null</code>.
So all the nullability checks on the non-nullable types are considered code smells.</p>
</div>
<div class="paragraph">
<p>When overriding an equality operator it doesn&#8217;t make sense to check for null, since you can directly check that the object is of a required type. In this example we can see that <code>other is A</code> will be true only if <code>other</code> is not <code>null</code>.</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-dart" data-lang="dart">class A {
  final String? value;

  @override
  operator ==(Object? other) =&gt;
    other is A &amp;&amp; value == other.value;
}</code></pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-dart" data-lang="dart">class A {
  final String? value;

  @override
  operator ==(Object? other) =&gt;
    other != null &amp;&amp; other is A &amp;&amp; value == other.value; // Noncompliant, redundant null check
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-dart" data-lang="dart">class A {
  final String? value;

  @override
  operator ==(Object? other) =&gt;
    other is A &amp;&amp; value == other.value;
}</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Dart Docs - <a href="https://dart.dev/tools/linter-rules/avoid_null_checks_in_equality_operators">Dart Linter rule - avoid_null_checks_in_equality_operators</a></p>
</li>
<li>
<p>Dart Docs - <a href="https://dart.dev/null-safety/understanding-null-safety">Dart Null Safety - Understanding null safety</a></p>
</li>
</ul>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Unnecessary null comparison in implementation of '=='.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>The entire null comparison expression, e.g. <code>o == null</code> or <code>o != null</code>.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
</div>
</div>