== Why is this an issue?

include::../rationale.adoc[]

include::../impact.adoc[]

== How to fix it

=== Code examples

include::../common/fix/code-rationale.adoc[]

==== Noncompliant code example

[source,docker,diff-id=1,diff-type=noncompliant]
----
FROM example
# Noncompliant
ENV APP_DEBUG=true
# Noncompliant
ENV ENV=development
CMD /run.sh
----

==== Compliant solution

[source,docker,diff-id=1,diff-type=compliant]
----
FROM example
ENV APP_DEBUG=false
ENV ENV=production
CMD /run.sh
----

=== How does this work?

Do not enable debug features globally.

== Resources

include::../common/resources/standards.adoc[]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

endif::env-github,rspecator-view[]
