<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>In single-threaded environments, the use of <code>this</code> in constructors is normal, and expected. But in multi-threaded environments, it could expose partially-constructed objects to other threads, and should be used with caution.</p>
</div>
<div class="paragraph">
<p>The classic example is a class with a <code>static</code> list of its instances. If the constructor stores <code>this</code> in the list, another thread could access the object before it&#8217;s fully-formed. Even when the storage of <code>this</code> is the last instruction in the constructor, there&#8217;s still a danger if the class is not <code>final</code>. In that case, the initialization of subclasses won&#8217;t be complete before <code>this</code> is exposed.</p>
</div>
<div class="paragraph">
<p>This rule raises an issue when <code>this</code> is assigned to any globally-visible object in a constructor, and when it is passed to the method of another object in a constructor</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-text" data-lang="text">public class Monument {

  public static final List&lt;Monument&gt; ALL_MONUMENTS = new ArrayList()&lt;&gt;;
  // ...

  public Monument(String location, ...) {
    ALL_MONUMENTS.add(this);  // Noncompliant; passed to a method of another object

    this.location = location;
    // ...
  }
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_exceptions">Exceptions</h3>
<div class="paragraph">
<p>This rule ignores instances of assigning <code>this</code> directly to a <code>static</code> field of the same class because that case is covered by <a data-rspec-id="S3010" class="rspec-auto-link">S3010</a> .</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p><a href="https://wiki.sei.cmu.edu/confluence/x/iDdGBQ">CERT, TSM01-J.</a> - Do not let the this reference escape during object construction</p>
</li>
<li>
<p><a href="https://wiki.sei.cmu.edu/confluence/x/vzdGBQ">CERT, TSM03-J.</a> - Do not publish partially initialized objects</p>
</li>
</ul>
</div>
<hr>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Make sure the use of "this" doesn&#8217;t expose partially-constructed instances of this class in multi-threaded environments.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>primary: "this"</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_relates_to_s3010">relates to: <a data-rspec-id="S3010" class="rspec-auto-link">S3010</a></h3>

</div>
<div class="sect2">
<h3 id="_on_9_oct_2015_134938_ann_campbell_wrote">on 9 Oct 2015, 13:49:38 Ann Campbell wrote:</h3>
<div class="paragraph">
<p><a href="http://www.ibm.com/developerworks/library/j-jtp0618/" class="bare">http://www.ibm.com/developerworks/library/j-jtp0618/</a></p>
</div>
</div>
</div>
</div>