=== Noncompliant code example

[source,text]
----
public class Monument {

  public static final List<Monument> ALL_MONUMENTS = new ArrayList()<>;
  // ...

  public Monument(String location, ...) {
    ALL_MONUMENTS.add(this);  // Noncompliant; passed to a method of another object 

    this.location = location;
    // ...
  }
}
----
