{cpp} comments (``++//++``) require fewer keystrokes, and take less space. Perhaps most importantly, they do not have the nesting problems that C-style comments do. Therefore {cpp} comments are preferred.


== Noncompliant Code Example

----
/* this is my comment ... */
----


== Compliant Solution

----
// this is my comment ... 
----


== Exceptions

Because a {cpp} header file may be included by a C source file, header files are ignored by this rule.

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

endif::env-github,rspecator-view[]
