== Why is this an issue?

{cpp} comments (``++//++``) require fewer keystrokes, and take less space. Perhaps most importantly, they do not have the nesting problems that C-style comments do. Therefore {cpp} comments are preferred.


=== Noncompliant code example

[source,cpp]
----
/* this is my comment ... */
----


=== Compliant solution

[source,cpp]
----
// this is my comment ... 
----


=== Exceptions

Because a {cpp} header file may be included by a C source file, header files are ignored by this rule.

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Edit this comment to use the {cpp} format, i.e. "//"


endif::env-github,rspecator-view[]
