<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>In 1978, Brian Kernighan and Dennis Ritchie published the first edition of The C Programming Language. This book, known to C programmers as "K&amp;R", served for many years as an informal specification of the language. The version of C that it describes is commonly referred to as K&amp;R C.</p>
</div>
<div class="paragraph">
<p>The K&amp;R function definition syntax introduced in the book was later deprecated in the ANSI C and ISO C standards. Even though the K&amp;R syntax is still supported in the ISO C11 standard, it&#8217;s not in ISO C&#43;&#43; standard versions and is not considered readable by most C/C&#43;&#43; developers today.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">int foo(a, b)   // Noncompliant K&amp;R C syntax
  int a;
  char* b;
{
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">int foo(int a, char* b) { // Compliant
}</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Move declarations of types of parameters for function into list of parameters.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_20_aug_2013_000447_ann_campbell_wrote">on 20 Aug 2013, 00:04:47 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>I love the background in this one, but it&#8217;s the only rule I&#8217;ve ever noticed like this. Is this kind of deviation desirable? When such background is available for other rules, do we want to include it?</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_20_aug_2013_064443_freddy_mallet_wrote">on 20 Aug 2013, 06:44:43 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>I think so Ann, indeed for me the goal of those rule descriptions is also to help developers learning some new stuff about their language, development environment, &#8230;&#8203;</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_10_sep_2013_111532_evgeny_mandrikov_wrote">on 10 Sep 2013, 11:15:32 Evgeny Mandrikov wrote:</h3>
<div class="paragraph">
<p>Should be noted that K&amp;R style can be used only in function definition, but not in function declaration, which is another reason to not use it.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_21_apr_2015_143339_evgeny_mandrikov_wrote">on 21 Apr 2015, 14:33:39 Evgeny Mandrikov wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] doesn&#8217;t conform to RSPEC standards.</p>
</div>
</div>
</div>
</div>