<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>For fixed-length values, a <code>CHAR</code> field occupies the same amount of disk space as a <code>VARCHAR2</code> field, but for variable-length values <code>CHAR</code> fields use more storage space and make searching more difficult by right-padding values with whitespaces. Therefore <code>VARCHAR2</code> fields are preferred. Similarly, <code>NCHAR</code> should be replaced by <code>NVARCHAR2</code>.</p>
</div>
<div class="paragraph">
<p>Note that for 1-character fields, <code>CHAR</code> is naturally equivalent to <code>VARCHAR2</code>, but the latter is still preferred for consistency.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-sql" data-lang="sql">DECLARE
  var1 CHAR; -- Noncompliant

  var2 CHAR(42); -- Noncompliant

  var3 NCHAR; -- Noncompliant

  var4 NCHAR(42); -- Noncompliant
BEGIN
  NULL;
END;
/</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-sql" data-lang="sql">DECLARE
  var1 VARCHAR2(42);

  var2 VARCHAR2(42);

  var3 NVARCHAR2(42);

  var4 NVARCHAR2(42);
BEGIN
  NULL;
END;
/</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Make "xxx" a "yyy".</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_28_jan_2015_191210_ann_campbell_wrote">on 28 Jan 2015, 19:12:10 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~dinesh.bolkensteyn] this description is copied from Nemo. I think the wording may be off: \"<code>CHAR</code> occupies <strong>the same amount</strong> of disk space and RAM than <code>VARCHAR2</code>&#8230;&#8203;"</p>
</div>
<div class="paragraph">
<p>Will you comment, please?</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_8_may_2015_152402_dinesh_bolkensteyn_wrote">on 8 May 2015, 15:24:02 Dinesh Bolkensteyn wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] I understand the confusion. What is meant is that CHAR uses at least as many bytes as VARCHAR2.</p>
</div>
<div class="paragraph">
<p>Refer to AskTom for the better explanation ;) https://asktom.oracle.com/pls/asktom/f?p=100:11:0::::P11_QUESTION_ID:123212348063</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_11_may_2015_132932_ann_campbell_wrote">on 11 May 2015, 13:29:32 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>please double-check my update [~dinesh.bolkensteyn]</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_11_may_2015_141031_dinesh_bolkensteyn_wrote">on 11 May 2015, 14:10:31 Dinesh Bolkensteyn wrote:</h3>
<div class="paragraph">
<p>Perfect</p>
</div>
</div>
</div>
</div>