This rule raises an issue when code directly accesses elements from string split results by index without first verifying that the element exists.

== Why is this an issue?

When you split a string using methods like `tokenize()` or `split()`, the resulting list size depends on the input content. If the input doesn't contain the expected number of delimiters, accessing elements by index can throw an `IndexOutOfBoundsException` at runtime.

For example, if you expect a string like `"a-b-c"` but receive `"a-b"`, trying to access the third element will cause your application to crash. This is particularly problematic when processing user input, configuration files, or data from external sources where the format might vary.

Groovy provides safer alternatives like multiple assignment that automatically handle cases where fewer elements are available than expected.

=== What is the potential impact?

This issue can cause application crashes and unexpected runtime failures. When processing user input or external data, it can lead to poor user experience and system instability. In server applications, it might cause request failures or service interruptions.

== How to fix it

Use multiple assignment which safely handles missing elements by assigning null to variables when there aren't enough elements in the list.

=== Code examples

==== Noncompliant code example

[source,groovy,diff-id=1,diff-type=noncompliant]
----
def parts = inputString.tokenize("-")
def first = parts.get(0)
def second = parts.get(1)
def third = parts.get(2)  // Noncompliant: may throw IndexOutOfBoundsException
----

==== Compliant solution

[source,groovy,diff-id=1,diff-type=compliant]
----
def parts = inputString.tokenize("-")
def (first, second, third) = parts  // Safe: missing elements become null
----

== Resources

=== Documentation

 * Groovy Multiple Assignment - https://groovy-lang.org/semantics.html#_multiple_assignment[Official documentation on Groovy's multiple assignment feature]

 * Groovy String Processing - https://groovy-lang.org/processing-xml.html#_working_with_strings[Guide to string processing methods in Groovy]

=== Standards

 * CWE 129 - https://cwe.mitre.org/data/definitions/129.html[Improper Validation of Array Index]

=== Related rules

 * S6466 - Collection elements should not be replaced unconditionally (Java)
