== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,javascript]
----
function setName(name) {
    name = name;
}
----

=== Compliant solution

[source,javascript]
----
function setName(name) {
    this.name = name;
}
----

== Resources
=== Documentation

* https://eslint.org[eslint] - Rule https://eslint.org/docs/latest/rules/no-self-assign[no-self-assign]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
