include::../description.adoc[]

== Noncompliant Code Example

----
function setName(name) {
    name = name;
}
----

== Compliant Solution

----
function setName(name) {
    this.name = name;
}
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
