:null-keyword: Nothing
:null-keyword-link: https://learn.microsoft.com/en-us/dotnet/visual-basic/language-reference/nothing
:exceptionCode1: vbnet/exceptionCode1.adoc
:exceptionCode2: vbnet/exceptionCode2.adoc
:exceptionCode3: vbnet/exceptionCode3.adoc

include::../dotnet-why.adoc[]

== How to fix it

=== Code examples

==== Noncompliant code example

[source,vbnet,diff-id=1,diff-type=noncompliant]
----
Public Sub Method()
    Dim O As Object = Nothing
    Console.WriteLine(O.ToString)   ' Noncompliant: 'O' is always Nothing
End Sub
----

==== Compliant solution

[source,vbnet,diff-id=1,diff-type=compliant]
----
Public Sub Method()
    Dim O As New Object
    Console.WriteLine(O.ToString)   ' Compliant: 'O' is not Nothing
End Sub
----

include::../dotnet-resources.adoc[]
* https://learn.microsoft.com/en-us/dotnet/visual-basic/language-reference/nothing[Nothing keyword]

include::../dotnet-rspecator.adoc[]