== Why is this an issue?

The requirement for a final ``++ELSE++`` clause is defensive programming. The clause should either take appropriate action, or contain a suitable comment as to why no action is taken.

=== Noncompliant code example

[source,sql]
----
SELECT
  CASE category
    WHEN 'A' THEN 21
    WHEN 'B' THEN 33
END shipping_cost
FROM product
----

=== Compliant solution

[source,sql]
----
SELECT
  CASE category
    WHEN 'A' THEN 21
    WHEN 'B' THEN 33
    ELSE 42
END shipping_cost
FROM product
----

include::../see.adoc[]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Add an "ELSE" clause to this "CASE" statement.


'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
