== Why is this an issue?

When you call `isEmpty()`, it clearly communicates the intention of the code, which is to check if the collection is empty. Using `size() == 0` for this purpose is less direct and can make the code slightly harder to understand.

Depending on the implementation of the `Collection`, the `size()` method can have a time complexity of `O(n)` where `n` is the number of elements in the collection. On the other hand, `isEmpty()` simply checks if there is at least one element in the collection, which is a constant time operation, `O(1)`.

Using `Collection.isEmpty()` to test for emptiness should therefore be preferred over `Collection.size()`.

== How to fix it

Test the emptiness of your collection using `Collection.isEmpty()`.

=== Code examples

==== Noncompliant code example

[source,java,diff-id=1,diff-type=noncompliant]
----
if (myCollection.size() == 0) { // Noncompliant
  /* ... */
}
----

==== Compliant solution

[source,java,diff-id=1,diff-type=compliant]
----
if (myCollection.isEmpty()) {
  /* ... */
}
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
