== Why is this an issue?

TODO 


Permettre une gestion standardisée de la gestion des dates dans les traitements. Tous les traitements doivent utiliser le module MGDATR03 s’ils ont besoin d’utiliser la date du jour.


=== Noncompliant code example

[source,cobol]
----
ACCEPT … FROM DATE
ACCEPT … FROM TIME
MOVE CURRENT-DATE TO …
----



ifdef::env-github,rspecator-view[]
'''
== Comments And Links
(visible only on this page)

=== on 27 May 2014, 09:18:10 Freddy Mallet wrote:
This rule is specific to the BNP context. We should provide an XPath rule to get the same result.

endif::env-github,rspecator-view[]
