<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>The explicit keyword prevents the constructor from being used to implicitly convert from a fundamental type to the class type.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">class C
{
public:
  C ( int32_t a ) // Noncompliant
  {
  }
};</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">class D
{
public:
  explicit D ( int32_t a )
  {
  }
};</code></pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p>MISRA C&#43;&#43;:2008, 12-1-3</p>
</li>
</ul>
</div>
<hr>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_duplicates_s1709">duplicates: <a data-rspec-id="S1709" class="rspec-auto-link">S1709</a></h3>

</div>
<div class="sect2">
<h3 id="_on_17_nov_2015_214914_evgeny_mandrikov_wrote">on 17 Nov 2015, 21:49:14 Evgeny Mandrikov wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] duplicates <a data-rspec-id="S1709" class="rspec-auto-link">RSPEC-1709</a> ?</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_18_nov_2015_173858_ann_campbell_wrote">on 18 Nov 2015, 17:38:58 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>It sure does [~evgeny.mandrikov]. Closed &amp; MISRA reference added to <a data-rspec-id="S1709" class="rspec-auto-link">RSPEC-1709</a></p>
</div>
</div>
</div>
</div>