== Why is this an issue?

The requirement for a final ``++case _++`` clause is defensive programming. The clause should either take appropriate action, or contain a suitable comment as to why no action is taken.

=== Noncompliant code example

[source,scala]
----
param match {
  case 1 => doSomething
  case 2 => doSomethingElse
}
----

=== Compliant solution

[source,scala]
----
param match {
  case 1 => doSomething
  case 2 => doSomethingElse
  case _ => throw new IllegalArgumentException(s"Unexpected param: $param")
}
----

include::../see.adoc[]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
