<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when the <code>pandas.DataFrame.values</code> is used instead of the <code>pandas.DataFrame.to_numpy()</code> method.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>The <code>values</code> attribute and the <code>to_numpy()</code> method in pandas both provide a way to return a NumPy representation of the <code>DataFrame</code>. However, there are some reasons why the <code>to_numpy()</code> method is recommended over the <code>values</code> attribute:</p>
</div>
<div class="ulist">
<ul>
<li>
<p><strong>Future Compatibility:</strong>
The <code>values</code> attribute is considered a legacy feature, while the <code>to_numpy()</code> is the recommended method to extract data and is considered more future-proof.</p>
</li>
<li>
<p><strong>Data type consistency:</strong>
If the <code>DataFrame</code> has columns with different data types, NumPy will choose a common data type that can hold all the data. This may lead to loss of information, unexpected type conversions, or increased memory usage. The <code>to_numpy()</code> allows you to select the common type manually, passing the <code>dtype</code> argument.</p>
</li>
<li>
<p><strong>View vs Copy:</strong>
The <code>values</code> attribute can return a view or a copy of the data depending on whether the data needs to be transposed. This can lead to confusion when modifying the extracted data. On the other hand, <code>to_numpy()</code> has <code>copy</code> argument allowing to force it always to return a new NumPy array, ensuring that any changes you make won&#8217;t affect the original <code>DataFrame</code>.</p>
</li>
<li>
<p><strong>Missing values control:</strong>
The <code>to_numpy()</code> allows to specify the default value used for missing values in the <code>DataFrame</code>, while the <code>values</code> will always use <code>numpy.nan</code> for missing values.</p>
</li>
</ul>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Use the <code>to_numpy()</code> method instead of the <code>values</code> attribute to get a NumPy representation of the <code>DataFrame</code>.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">import pandas as pd

df = pd.DataFrame({
        'X': ['A', 'B', 'A', 'C'],
        'Y': [10, 7, 12, 5]
    })

arr = df.values # Noncompliant: using the 'values' attribute is not recommended</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">import pandas as pd

df = pd.DataFrame({
        'X': ['A', 'B', 'A', 'C'],
        'Y': [10, 7, 12, 5]
    })

arr = df.to_numpy() # Compliant</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Pandas Documentation - <a href="https://pandas.pydata.org/pandas-docs/version/2.1/reference/api/pandas.DataFrame.to_numpy.html">pandas.DataFrame.to_numpy()</a></p>
</li>
<li>
<p>Pandas Documentation - <a href="https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.values.html">pandas.DataFrame.values</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>