== Why is this an issue?

include::../description.adoc[]


=== Noncompliant code example

[source,php]
----
preg_replace("/[c̈d̈]/", "X", "cc̈d̈d"); // Noncompliant, print "XXXXXX" instead of expected "cXXd".
----


=== Compliant solution

[source,php]
----
preg_replace("/c̈|d̈/", "X", "cc̈d̈d"); // print "cXXd"
----
