include::../description.adoc[]


== Noncompliant Code Example

----
preg_replace("/[c̈d̈]/", "X", "cc̈d̈d"); // Noncompliant, print "XXXXXX" instead of expected "cXXd".
----


== Compliant Solution

----
preg_replace("/c̈|d̈/", "X", "cc̈d̈d"); // print "cXXd"
----
