== Why is this an issue?

Coding conventions allow teams to work efficiently together. This rule checks that the ``++public++`` section of a class is declared first, followed by the ``++protected++`` section, and ending with the ``++private++`` section.


=== Noncompliant code example

[source,cpp]
----
class Point
{
  private: 
    String _color; 
    ...

  protected:
    ...

  public: 
    String getColor(); 
    String setColor(); 
}; 
----


=== Compliant solution

[source,cpp]
----
class Point
{
  public: 
    String getColor(); 
    String setColor(); 

  protected:
    ...

  private: 
    String _color; 
    ...
}; 
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Rearrange this class to follow the required ordering.


endif::env-github,rspecator-view[]
