<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when <code>torch.save()</code> is called with a method reference instead of a method call, such as <code>model.state_dict</code> instead of <code>model.state_dict()</code>.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>When you save a PyTorch model&#8217;s state, you need to save the actual state dictionary, not a reference to the method that returns it.</p>
</div>
<div class="paragraph">
<p>In Python, <code>model.state_dict</code> without parentheses refers to the method object itself, while <code>model.state_dict()</code> with parentheses calls the method and returns the state dictionary.</p>
</div>
<div class="paragraph">
<p>If you save the method reference, PyTorch will serialize a function object instead of the model&#8217;s parameters and buffers. When you later try to load this saved file, PyTorch expects a dictionary but receives a function object, causing an <code>AttributeError: 'function' object has no attribute 'copy'</code>.</p>
</div>
<div class="paragraph">
<p>This error typically occurs during model loading when PyTorch tries to call <code>.copy()</code> on what it expects to be a state dictionary but is actually a function object.</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="paragraph">
<p>This issue causes runtime errors when loading saved models, leading to:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>Application crashes during model loading</p>
</li>
<li>
<p>Failed deployments in production environments</p>
</li>
<li>
<p>Interrupted training workflows</p>
</li>
<li>
<p>Loss of training progress if the error is discovered late</p>
</li>
</ul>
</div>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">import torch
import torch.nn as nn

model = nn.Linear(10, 1)

# Saves method reference instead of state dictionary
torch.save(model.state_dict, 'model.pth')  # Noncompliant</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">import torch
import torch.nn as nn

model = nn.Linear(10, 1)

# Calls method to get and save the state dictionary
torch.save(model.state_dict(), 'model.pth')</code></pre>
</div>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>PyTorch Tutorials - <a href="https://pytorch.org/tutorials/beginner/saving_loading_models.html">Saving and Loading Models</a></p>
</li>
<li>
<p>PyTorch Documentation - <a href="https://pytorch.org/docs/stable/generated/torch.save.html">pytorch.save function</a></p>
</li>
<li>
<p>PyTorch Documentation - <a href="https://pytorch.org/docs/stable/generated/torch.nn.Module.html#torch.nn.Module.state_dict">torch.nn.Module.state_dict</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>