== Why is this an issue?

A ``++LocalConnection++`` object is used to invoke a method in another ``++LocalConnection++`` object, either within a single SWF file or between multiple SWF files. This kind of local connection should be authorized only when the origin (domain) of the other Flex applications is perfectly defined. 


=== Noncompliant code example

[source,flex]
----
localConnection.allowDomain("*");
----


=== Compliant solution

[source,flex]
----
localConnection.allowDomain("www.myDomain.com");
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Replace this wildcard character '*' with a well defined domain


'''
== Comments And Links
(visible only on this page)

=== on 22 Nov 2013, 10:13:21 Freddy Mallet wrote:
Is implemented by \http://jira.codehaus.org/browse/SONARPLUGINS-3276

=== on 24 Nov 2013, 19:10:11 Ann Campbell wrote:
I'm wondering how broadly this rule will apply. Clearly it will flag "*" but what about other wildcard combinations? If so, we might want to take another look at the wording.

endif::env-github,rspecator-view[]
