A ``++LocalConnection++`` object is used to invoke a method in another ``++LocalConnection++`` object, either within a single SWF file or between multiple SWF files. This kind of local connection should be authorized only when the origin (domain) of the other Flex applications is perfectly defined. 


== Noncompliant Code Example

----
localConnection.allowDomain("*");
----


== Compliant Solution

----
localConnection.allowDomain("www.myDomain.com");
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
