== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

With default provided regular expression: ``++^[a-z][a-zA-Z0-9]*$++``:

[source,cpp]
----
void DoSomething (void);
----

=== Compliant solution

[source,cpp]
----
void doSomething (void);
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::parameters.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
