== Why is this an issue?

If a function does not return anything, it makes no sense to use its output. Specifically, passing it to another function, or assigning its "result" to a variable is probably a bug because such functions return nothing, which is probably not what was intended.

=== Noncompliant code example

[source,php]
----
$result = closedir($dir_handle); // Noncompliant, "closedir" does not return anything.
----

=== Compliant solution

[source,php]
----
closedir($dir_handle);
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../highlighting.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
