If a function does not return anything, it makes no sense to use its output. Specifically, passing it to another function, or assigning its "result" to a variable is probably a bug because such functions return nothing, which is probably not what was intended.

== Noncompliant Code Example

----
$result = closedir($dir_handle); // Noncompliant, "closedir" does not return anything.
----

== Compliant Solution

----
closedir($dir_handle);
----
