== Why is this an issue?

If Finalize or an override of Finalize throws an exception, and the runtime is not hosted by an application that overrides the default policy, the runtime terminates the process immediately without graceful cleanup (finally blocks and finalizers are not executed). This behavior ensures process integrity if the finalizer cannot free or destroy resources.


The rule reports on throw statements used in finalizers.

=== Noncompliant code example

[source,csharp]
----
class MyClass
{
    ~MyClass()
    {
        throw new NotImplementedException(); // Noncompliant
    }
}
----

=== Compliant solution

[source,csharp]
----
class MyClass
{
    ~MyClass()
    {
        // no throw
    }
}
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Remove this 'throw' statement.


=== Highlighting

The 'throw' statement.


'''
== Comments And Links
(visible only on this page)

=== on 1 Mar 2018, 17:03:27 Valeri Hristov wrote:
Throwing in IDisposable.Dispose is covered in RSPEC-3877

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
