== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

With a threshold of 5:

[source,csharp]
----
public class Foo    // Noncompliant - Foo depends on too many classes: T1, T2, T3, T4, T5, T6 and T7
{
  private T1 a1;    // Foo is coupled to T1
  private T2 a2;    // Foo is coupled to T2
  private T3 a3;    // Foo is coupled to T3

  public T4 Compute(T5 a, T6 b)    // Foo is coupled to T4, T5 and T6
  {
    T7 result = a.Process(b);    // Foo is coupled to T7
    return result;
  }

  public static class Bar    // Compliant - Bar depends on 2 classes: T8 and T9
  {
    public T8 a8;
    public T9 a9;
  }
}
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Split this (class|record) into smaller and more specialized ones to reduce its dependencies on other types from {} to the maximum authorized {} or less.


=== Parameters

.max
****

----
30
----

Maximum number of types a single type is allowed to depend upon
****


'''
== Comments And Links
(visible only on this page)

=== on 13 Jun 2018, 16:16:48 Valeri Hristov wrote:
Changing the default value for C# to 30, to comply with the one from FxCop

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
