include::../description.adoc[]

== Noncompliant Code Example

----
     C                   IF        X = X
----

----
      /free
        if a = a; // always true
          doZ();
        endif;
        if a <> a; // always false
          doY();
        endif;
        if a = b and a = b; // if the first one is true, the second one is too
          doX();
        endif;
        if a = b or a = b; // if the first one is true, the second one is too
          doV();
        endif;

        j = 5 / 5; //always 1
        k = 5 - 5; //always 0
      /end-free
----

== Exceptions

This rule ignores <code>*</code> and <code>+</code>. 

include::../see.adoc[]
