This rule raises an issue when the `def` keyword is used redundantly with explicit type declarations, visibility modifiers, or other keywords like `static`.

== Why is this an issue?

In Groovy, the `def` keyword is used for dynamic typing, essentially declaring a variable as `Object` type. However, when you already specify an explicit type (like `String`), a visibility modifier (like `private`), or other keywords (like `static`), the `def` keyword becomes redundant.

Using `def` in these contexts creates unnecessary verbosity and can confuse developers about the actual type or scope of the declaration. The Groovy style guide explicitly recommends against this practice, stating that you should "make a choice, either use `def` or a type" but not both.

Redundant `def` usage makes code less idiomatic and harder to read, especially for developers transitioning from Java or those learning Groovy best practices.

=== What is the potential impact?

This issue affects code readability and maintainability. While it doesn't cause functional problems, redundant `def` keywords make the codebase less consistent with Groovy conventions and can mislead developers about the intended typing approach.

== How to fix it

Remove the redundant `def` keyword when an explicit type is already declared.

=== Code examples

==== Noncompliant code example

[source,groovy,diff-id=1,diff-type=noncompliant]
----
def String name = "Guillaume" // Noncompliant
----

==== Compliant solution

[source,groovy,diff-id=1,diff-type=compliant]
----
String name = "Guillaume"
----

== Resources

=== Documentation

 * Groovy Style Guide - Def and Type - https://groovy-lang.org/style-guide.html#_def_and_type[Official Groovy documentation explaining when to use def vs explicit types]

 * CodeNarc Unnecessary Rules - https://codenarc.org/codenarc-rules-unnecessary.html[CodeNarc rules for detecting unnecessary code patterns in Groovy]
