This rule raises an issue when `def` is redundant.

== Why is this an issue?

In Groovy, `def` is used when you intentionally omit an explicit type. When a declaration already includes a type (for example, `String`) or a visibility modifier (for example, `private`), keeping `def` is redundant and makes the code less idiomatic.

Unnecessary `def` appears:
- on fields with a visibility modifier or a type declaration
- on methods with a visibility modifier or an explicit return type
- on variables with an explicit type declaration or a type declaration

This extra keyword adds noise and can confuse readers about whether the code relies on dynamic typing.

=== What is the potential impact?

This is mainly a maintainability issue: redundant keywords reduce readability and make style less consistent across the codebase.

== How to fix it

Remove `def` when the declaration already has an explicit type or a modifier.

=== Code examples

==== Noncompliant code example

[source,groovy,diff-id=1,diff-type=noncompliant]
----
private def String name = "Guillaume" // Noncompliant
----

==== Compliant solution

[source,groovy,diff-id=1,diff-type=compliant]
----
private String name = "Guillaume"
----

== Resources

=== Documentation

 * Groovy documentation - https://groovy-lang.org/style-guide.html#_def_and_type[Style Guide]
