<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Creating a null function pointer using <code>transmute</code> results in undefined behavior, as null function pointers are not valid in Rust. Instead, use <code>Option&lt;fn()&gt;</code> to safely represent a nullable function pointer.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-rust" data-lang="rust">let null_fn: fn() = unsafe { std::mem::transmute(std::ptr::null::&lt;()&gt;()) }; // Noncompliant</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-rust" data-lang="rust">let null_fn: Option&lt;fn()&gt; = None; // Compliant</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Clippy Lints - <a href="https://rust-lang.github.io/rust-clippy/master/index.html#transmute_null_to_fn" class="bare">https://rust-lang.github.io/rust-clippy/master/index.html#transmute_null_to_fn</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>