<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Catching an exception only to immediately rethrow it without doing anything else is useless and misleading.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-text" data-lang="text">try {
  /* ... */
} catch (Exception e) {                   // Non-Compliant
  throw e;
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_exceptions">Exceptions</h3>
<div class="paragraph">
<p>When all instances of a general exception must be handled, but some specific ones not, propagation must be used and so is allowed by this rule.</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-text" data-lang="text">try {
  /* ... */
} catch (RuntimeException e) {            // Compliant - propagation of the unchecked exception
  throw e;
} catch (Exception e) {                   // Compliant - catching of the checked exception
  LOGGER.error("...", e);
}</code></pre>
</div>
</div>
<div class="paragraph">
<p>Throwing the same exception can also makes sense when an action is done before throwing it again.</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-text" data-lang="text">try {
  /* ... */
} catch (MyException e) {            // Compliant - something is done before throwing again the exception
  fixSomething();
  throw e;
}</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_duplicates_s2737">duplicates: <a data-rspec-id="S2737" class="rspec-auto-link">S2737</a></h3>

</div>
</div>
</div>