=== Code examples

==== Noncompliant code example

[source,vbnet,diff-id=1,diff-type=noncompliant]
----
Public Function GetSortedFilteredList(data As IEnumerable(Of Integer)) As IEnumerable(Of Integer)
    Return data.OrderBy(Function(x) x).Where(Function(x) x Mod 2 = 0)
End Function
----

==== Compliant solution

[source,vbnet,diff-id=1,diff-type=compliant]
----
Public Function GetSortedFilteredList(data As IEnumerable(Of Integer)) As IEnumerable(Of Integer)
    Return data.Where(Function(x) x Mod 2 = 0).OrderBy(Function(x) x)
End Function
----
