<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Synchronization can be expensive in terms of time when multiple threads need to pass through the same bottleneck (method with <code>[MethodImpl(MethodImplOptions.Synchronized)]</code>).</p>
</div>
<div class="paragraph">
<p>If you have a piece of code calling a method with <code>[MethodImpl(MethodImplOptions.Synchronized)]</code> attribute once, then it only has to wait its turn to pass through the bottleneck once. But call it in a loop, and your code has to get back in line for the bottleneck over and over.</p>
</div>
<div class="paragraph">
<p>Instead, it would be better to get into the bottleneck, and then do the looping. I.e. consider refactoring the code to perform the loop inside the method.</p>
</div>
<div class="paragraph">
<p>This rule raises an issue when a method with <code>[MethodImpl(MethodImplOptions.Synchronized)]</code> is called in a loop.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-csharp" data-lang="csharp">public void doSomething(int max) {
  for (int i = 0; i &lt; max; i++) {
    doSynchronized(i);  // Noncompliant
  }
}

[MethodImpl(MethodImplOptions.Synchronized)]
public void doSynchronized(int val) {
  // ...
}</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Move the invocation of this "synchronized" method outside the loop.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_16_jun_2015_104655_tamas_vajk_wrote">on 16 Jun 2015, 10:46:55 Tamas Vajk wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] I&#8217;ve added this C# version.</p>
</div>
<div class="paragraph">
<p>I think this attribute is used a lot less frequently than the <code>synchronized</code> language element in Java. It might not be worth implementing.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_16_jun_2015_111908_ann_campbell_wrote">on 16 Jun 2015, 11:19:08 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>Thanks for filling in the C# version [~tamas.vajk]. It&#8217;s completely up to you whether or not to implement. Always feel free to de-target a rule written originally for Java. I add C# based on hunches and guesses.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_12_jun_2015_143822_ann_campbell_wrote">on 12 Jun 2015, 14:38:22 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>CodePro: Invoke Synchronized Method In Loop</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_12_jun_2015_143913_ann_campbell_wrote">on 12 Jun 2015, 14:39:13 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~tamas.vajk] I&#8217;m guessing this rule applies to C# as well as Java. I&#8217;ve targeted it that way on the off chance. Feel free to un-target if I&#8217;m wrong.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_15_jun_2015_202452_nicolas_peru_wrote">on 15 Jun 2015, 20:24:52 Nicolas Peru wrote:</h3>
<div class="paragraph">
<p>Looks good</p>
</div>
</div>
</div>
</div>