<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when trying to perform comparison or arithmetic operations between <code>datetime.date</code> and <code>datetime.datetime</code> objects.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Despite the fact that the <code>datetime.datetime</code> is inherited from the <code>datetime.date</code> they are not compatible from comparison and arithmetical operations perspective.
Trying to compare a <code>datetime.datetime</code> with a <code>datetime.date</code> objects leads to a <code>TypeError</code>.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Make sure that comparison or arithmetic operations are performed between compatible types e.g. convert <code>datetime.datetime</code> to <code>datetime.date</code> or vice versa.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">from datetime import datetime, date

dt = datetime.now()
d = date.today()

if dt &lt; d: # Noncompliant: TypeError: can't compare datetime.datetime to datetime.date
    ...</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">from datetime import datetime, date

dt = datetime.now()
d = date.today()

if dt.date() &lt; d: # OK
    ...</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Python documentation - <a href="https://docs.python.org/3/library/datetime.html#datetime-objects">Datetime Objects</a></p>
</li>
<li>
<p>Python documentation - <a href="https://docs.python.org/3/library/datetime.html#date-objects">Date Objects</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>