<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p><code>CAST(... AS CHAR/VARCHAR)</code> can be a source of incompatibilities between database versions: the behavior of <code>CAST</code> may not be the same depending on the version of the database system. Such incompatibilities can cause unexpected output from applications that <code>CAST</code> decimal data to <code>CHAR</code> or <code>VARCHAR</code>, it&#8217;s therefore best to avoid using <code>CAST(... AS CHAR/VARCHAR)</code>.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cobol" data-lang="cobol">  DELETE product
  WHERE CAST(status_code AS CHAR(2)) = '42' -- Noncompliant</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Avoid using 'CAST(&#8230;&#8203; AS CHAR/VARCHAR)'</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_25_oct_2016_113616_nicolas_bontoux_wrote">on 25 Oct 2016, 11:36:16 Nicolas Bontoux wrote:</h3>
<div class="paragraph">
<p>References:</p>
</div>
<div class="ulist">
<ul>
<li>
<p><a href="http://www.ibm.com/support/knowledgecenter/SSEPEK_11.0.0/sqlref/src/tpc/db2z_castspecification.html">CAST specification</a></p>
</li>
<li>
<p><a href="http://www.ibm.com/support/knowledgecenter/SSEPEK_10.0.0/apsg/src/tpc/db2z_relincompatapplsqlfromv9.html">IBM note on SQL release incompatibilities</a> (see section <em>Changes to string formatting of decimal data</em>)</p>
</li>
</ul>
</div>
</div>
</div>
</div>