== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,vbnet]
----
Dim o As Object = 5
Dim x As Integer = 5

x = o >> 5 ' Noncompliant
x = x >> o ' Noncompliant
----

=== Exceptions

This rule does not raise when the left or the right expression is ``++Nothing++``.

----
x = Nothing >> 5
x = 5 >> Nothing
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../highlighting.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
