<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Ensuring that assembly language code is encapsulated and isolated aids portability. Where assembly language instructions are needed, they shall be encapsulated and isolated in either assembler functions or C&#43;&#43; functions.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">void fn ( void )
{
  DoSomething ( );
  asm ( "NOP" ); // Noncompliant, asm mixed with C/C++ statements
  DoSomething ( );
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">void Delay ( void )
{
  asm ( "NOP" ); // Compliant, asm not mixed with C/C++ statements
}

void fn ( void )
{
  DoSomething ( );
  Delay ( ); // Compliant, Assembler is encapsulated
  DoSomething ( );
}</code></pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p>MISRA C 2004, 2.1 - Assembly language shall be encapsulated and isolated.</p>
</li>
<li>
<p>MISRA C&#43;&#43; 2008, 7-4-3 - Assembly language shall be encapsulated and isolated.</p>
</li>
</ul>
</div>
<hr>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Isolate this statement in a function or macro.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_relates_to_s786">relates to: <a data-rspec-id="S786" class="rspec-auto-link">S786</a></h3>

</div>
</div>
</div>