include::../description.adoc[]

== Noncompliant Code Example

----
if (condition)  // Noncompliant
  executeSomething()
----

== Compliant Solution

----
if (condition) {
  executeSomething()
}
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
