== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,csharp]
----
public void encrypt(byte[] key, byte[] ptxt, byte[] ciphertext, byte[] tag)
{
    var nonce = Encoding.UTF8.GetBytes("7cVgr5cbdCZV"); // The initialization vector is a static value.

    using var cipher = new AesGcm(key);

    cipher.Encrypt(nonce, plaintext, ciphertext, tag); // Noncompliant
}
----

=== Compliant solution

[source,csharp]
----
public void encrypt(byte[] key, byte[] ptxt, byte[] ciphertext, byte[] tag)
{
    var nonce = new byte[AesGcm.NonceByteSizes.MaxSize];
    RandomNumberGenerator.Fill(nonce); // Random 96 bits nonc

    using var cipher = new AesGcm(key);

    cipher.Encrypt(nonce, plaintext, ciphertext, tag);
}
----

include::../see.adoc[]


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

* Primary location
** Use a dynamically-generated initialization vector (IV) to avoid IV-key pair reuse.

* Secondary location
** The initialization vector is a static value.


=== Highlighting

* Primary location
** System.Security.Cryptography.AesGcm.Encrypt call
** System.Security.Cryptography.AesCcm.Encrypt call

* Secondary locations
** nonce variable declaration


endif::env-github,rspecator-view[]