<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule allows banning certain PHP functions.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="paragraph">
<p>Given a parameter of <code>foo</code> for this rule:</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-php" data-lang="php">$arr=array("apple", "pear","banana");
echo foo($arr);  // Noncompliant</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove this use of the "xxx" function.</p>
</div>
</div>
<div class="sect2">
<h3 id="_parameters">Parameters</h3>
<div class="sidebarblock">
<div class="content">
<div class="title">functions</div>
<div class="paragraph">
<p>Comma-separated list of disallowed functions</p>
</div>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_19_sep_2014_151631_freddy_mallet_wrote">on 19 Sep 2014, 15:16:31 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>@Ann, instead of providing a template rule, I would specialize this rule : "Alias functions kept in PHP to preserve a backward compatibility should not be used". Or we could imagine to provide two rules :</p>
</div>
<div class="ulist">
<ul>
<li>
<p>a template rule without any default value</p>
</li>
<li>
<p>a specialized rule targeting alias functions</p>
</li>
</ul>
</div>
</div>
</div>
</div>