<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when the <code>go_package</code> option in a Protocol Buffer file uses a simple package name instead of a full module path.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>When working with Protocol Buffers in Go, the <code>go_package</code> option tells the Protocol Buffer compiler where to place the generated Go code and what import path to use.</p>
</div>
<div class="paragraph">
<p>Using simple package names like <code>"mypackage"</code> instead of full module paths creates several problems:</p>
</div>
<div class="ulist">
<ul>
<li>
<p><strong>Import conflicts</strong>: Simple names can clash with other packages in your project or dependencies</p>
</li>
<li>
<p><strong>Module system incompatibility</strong>: Go&#8217;s module system expects full import paths that include the module name</p>
</li>
<li>
<p><strong>Code generation issues</strong>: The protobuf compiler may generate code in unexpected locations</p>
</li>
<li>
<p><strong>Maintainability problems</strong>: Other developers cannot easily understand where the generated code belongs</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>Go&#8217;s module system is designed around full import paths that typically include a domain name and repository path. This ensures global uniqueness and makes dependencies clear.</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="paragraph">
<p>Using simple package names can lead to import conflicts, make code harder to maintain, and cause issues with Go&#8217;s module system. Generated code may be placed in unexpected locations, making it difficult for other developers to understand the project structure.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Use a full module path that includes your domain and repository path instead of a simple package name. The path should follow Go&#8217;s module naming conventions.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-protobuf" data-lang="protobuf">syntax = "proto3";

option go_package = "mypackage"; // Noncompliant

message User {
  string name = 1;
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-protobuf" data-lang="protobuf">syntax = "proto3";

option go_package = "github.com/myorg/myproject/api/v1";

message User {
  string name = 1;
}</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Protocol Buffers Go Tutorial - <a href="https://protobuf.dev/getting-started/gotutorial/">Official Protocol Buffers tutorial for Go, including go_package option usage</a></p>
</li>
<li>
<p>Go Modules Reference - <a href="https://go.dev/ref/mod">Official Go modules documentation explaining module paths and naming conventions</a></p>
</li>
<li>
<p>Protocol Buffers Language Guide - <a href="https://protobuf.dev/programming-guides/proto3/">Comprehensive guide to Protocol Buffers syntax and options</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>