<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when unquoted symbol keys contain hyphens, which causes Ruby syntax errors.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>In Ruby, symbols are identifiers that start with a colon (<code>:symbol</code>). When you use a hyphen in an unquoted symbol like <code>:data-role</code>, Ruby interprets the hyphen as a subtraction operator instead of part of the symbol name.</p>
</div>
<div class="paragraph">
<p>This means Ruby tries to subtract the variable <code>role</code> from the variable <code>data</code>, which typically results in an "undefined variable" error since these variables don&#8217;t exist.</p>
</div>
<div class="paragraph">
<p>The problem stems from Ruby&#8217;s parsing rules: unquoted symbols can only contain letters, numbers, underscores, and certain special characters. Hyphens are not allowed because they conflict with the subtraction operator.</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="paragraph">
<p>This issue causes immediate syntax errors that prevent the application from running. The error messages can be confusing to developers, especially beginners, as they mention "undefined variable" rather than explaining the symbol syntax problem.</p>
</div>
<div class="paragraph">
<p>In development, this leads to:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>Application crashes with cryptic error messages</p>
</li>
<li>
<p>Wasted debugging time trying to understand why variables are "undefined"</p>
</li>
<li>
<p>Frustration for developers learning Ruby</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>The impact is typically caught early since the code won&#8217;t run, but it can slow down development and create confusion about proper Ruby syntax.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Quote the symbol using double quotes. This creates a valid symbol that can contain hyphens.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-ruby" data-lang="ruby">options = {
  :jquery-ui =&gt; true,  # Noncompliant
  :data-role =&gt; "button"  # Noncompliant
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-ruby" data-lang="ruby">options = {
  :"jquery-ui" =&gt; true,
  :"data-role" =&gt; "button"
}</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Ruby Symbol Documentation - <a href="https://ruby-doc.org/core/Symbol.html">Official Ruby documentation explaining symbol syntax and usage</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>