<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when unused scope-limited definitions are found.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>When a class or function is defined in a parent function or method, it is only visible in this parent function or method&#8217;s scope.
If the defined class or function is not used within this scope it is dead code (unnecessary, inoperative code) that should be removed.</p>
</div>
<div class="paragraph">
<p>Cleaning out dead code decreases the size of the maintained codebase, making it easier to understand the program and preventing bugs from being introduced.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">def parent_function():
    def nested_function():  # Noncompliant: this function is never used in this scope.
        print("nested_function")

    class NestedClass:  # Noncompliant: this class is never used in this scope.
        def __init__(self):
            print("NestedClass")</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">def parent_function():
    class NestedClass:
        def __init__(self):
            print("NestedClass")

    NestedClass()</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove this unused private "XXXX" class/function.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>The class or function&#8217;s name.</p>
</div>
</div>
</div>
</div>