== Why is this an issue?

The long suffix should always be written in uppercase, i.e. 'L', as the lowercase 'l' can easily be confused with the digit one '1'.


=== Noncompliant code example

[source,python]
----
return 10l  // Noncompliant; easily confused with one zero one
----


=== Compliant solution

[source,python]
----
return 10L
----



ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
