== Why is this an issue?

The long suffix should always be written in uppercase, i.e. 'L', as the lowercase 'l' can easily be confused with the digit one '1'.


=== Noncompliant code example

[source,python]
----
return 10l  // Noncompliant; easily confused with one zero one
----


=== Compliant solution

[source,python]
----
return 10L
----



ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Replace this lower case "l" long suffix by an upper case "L".


'''
== Comments And Links
(visible only on this page)

=== relates to: S818

=== on 12 Jul 2013, 15:43:05 Dinesh Bolkensteyn wrote:
Implemented by \http://jira.codehaus.org/browse/SONARJAVA-208

=== on 13 Apr 2015, 18:28:57 Evgeny Mandrikov wrote:
\[~ann.campbell.2] Looks like duplicate of RSPEC-818

=== on 20 Apr 2015, 13:52:12 Ann Campbell wrote:
deprecated

=== on 21 Jun 2016, 15:41:37 Elena Vilchik wrote:
``++l++``/``++L++`` is irrelevant for Python as this feature was dropped since Python 3.0

endif::env-github,rspecator-view[]
