When the size of a variable-length table is ``++DEPENDING ON++`` a non-``++BINARY++``/``++COMP++`` variable, use of that table is inefficient because a conversion must be done every time the table is used.


== Noncompliant Code Example

----
01 VARS
    05 TABLE_SIZE   PIC 9(4).
    05 MY_TABLE OCCURS 1 TO 10
                        DEPENDING ON TABLE_SIZE  *> Noncompliant; TABLE-SIZE isn't BINARY or COMP
                        PIC X(10).
----


== Compliant Solution

----
01 VARS
    05 TABLE_SIZE   PIC 9(4) BINARY.
    05 MY_TABLE OCCURS 1 TO 10
                        DEPENDING ON TABLE_SIZE 
                        PIC X(10).
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

include::highlighting.adoc[]

endif::env-github,rspecator-view[]
