This rule raises an issue on logging calls inside a `Catch` clause that does not pass the raised `Exception`.

include::../why-dotnet.adoc[]

== How to fix it

=== Code examples

==== Noncompliant code example

[source,vbnet,diff-id=1,diff-type=noncompliant]
----
Public Function Save() As Boolean
    Try
        DoSave()
        Return True
    Catch exception As IOException
        Logger.LogError("Saving failed.")            ' Noncompliant: No specifics about the error are logged
        Return False
    End Try
End Function
----

==== Compliant solution

[source,vbnet,diff-id=1,diff-type=compliant]
----
Public Function Save() As Boolean
    Try
        DoSave()
        Return True
    Catch exception As IOException
        logger.LogError(exception, "Saving failed.") ' Compliant: Exception details are logged
        Return False
    End Try
End Function
----

include::../resources-dotnet.adoc[]
