include::../description.adoc[]

== Noncompliant Code Example

----
EXEC SQL
  CREATE TABLE INVENTORY
                 (PARTNO         SMALLINT     NOT NULL,
                  DESCR          VARCHAR(24 ),
                  PRIMARY KEY(PARTNO))
END-EXEC.

EXEC SQL
  DROP TABLE EMPLOYEE RESTRICT
END-EXEC.

EXEC SQL
  ALTER TABLE EQUIPMENT
     DROP COLUMN LOCATION CASCADE
END-EXEC.
----

== Exceptions

Creating global temporary tables, creating indexes on those tables, and then dropping those indices and tables is allowed.

----
EXEC SQL
   DECLARE GLOBAL TEMPORARY TABLE SESSION.TBT09SCO ...
END-EXEC.

EXEC SQL
   CREATE UNIQUE INDEX X1T09SCO ON SESSION.TBT09SCO ...
END-EXEC.

...
EXEC SQL
   DROP INDEX X1T09SCO
END-EXEC.

EXEC SQL
   DROP TABLE SESSION.TBT09SCO
END-EXEC.
----
