{cpp}20 introduces a new text formatting API with the ``<format>`` header,
joining the ``printf`` family of functions -- inherited from C -- and ``iostreams``.
``std::format`` combines the convenience of ``printf``, separating formatting and
arguments, with the type-safety of ``iostreams``.

Before {cpp}20, if you wanted to format an output stream, you had to use standard manipulators that control the output streams.
This approach is very verbose, is often stateful, and is not thread-safe. That is why we recommend replacing them with ``std::format``
when possible.

Some manipulators will have a temporary effect on the output. For example, ``std::setw``. This is due to the resetting of the width property of the stream when most of the ``operator<<``is called.
Other manipulators will have a lasting effect on the output. For example, ``std::boolalpha``. It will set the ``boolalpha`` flag of the outputstream without resetting it.

This rule raises an issue when an outpustream is used with standard manipulators to output some text in a way that can be replaced by ``std::format``.
You should be careful not have undesirable side effects when replacing a manipulator with lasting effect.

== Noncompliant Code Example

[source,cpp]
----
FIXME
----

== Compliant Solution

[source,cpp]
----
FIXME
----

== Exceptions

Manipulators that don't have a direct equivalent in the format library like ``std::quoted``, ``std::put_money``, etc.
