== Why is this an issue?

By implementing class interfaces with member functions the implementation retains more control over how the object state can be modified and helps to allow a class to be maintained without affecting clients. Returning a handle to class-data allows for clients to modify the state of the object without using any interfaces.


=== Noncompliant code example

[source,cpp]
----
class C
{
public:
  int32_t & getA () // Noncompliant, non-const handle to class-data returned.
  {
    return a;
  }
private:
  int32_t a;
};

void b ( C & c )
{
  int32_t & a_ref = c.getA ();
  a_ref = 10; // External modification of private C::a
}
----


=== Compliant solution

[source,cpp]
----
class C
{
public:
  int32_t const & getA () // Compliant
  {
    return a;
  }
  void setA(int32_t a)
  {
    this.a = a;
  }
private:
  int32_t a;
};

void b ( C & c )
{
  int32_t const & a_ref = c.getA ();
  a_ref = 10; // Not allowed, use setter setA instead
}
----


== Resources

* MISRA {cpp}:2008, 9-3-2


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Mark the return type with the "const" qualifier.


'''
== Comments And Links
(visible only on this page)

=== on 15 Oct 2014, 20:51:48 Ann Campbell wrote:
\[~samuel.mercier] please:

* fill in the appropriate reference field(s).
* provide a See section.


=== on 17 Oct 2014, 15:32:33 Ann Campbell wrote:
FYI [~samuel.mercier] I've moved this to Data-related reliability.

endif::env-github,rspecator-view[]
