By implementing class interfaces with member functions the implementation retains more control over how the object state can be modified and helps to allow a class to be maintained without affecting clients. Returning a handle to class-data allows for clients to modify the state of the object without using any interfaces.


== Noncompliant Code Example

----
class C
{
public:
  int32_t & getA () // Noncompliant, non-const handle to class-data returned.
  {
    return a;
  }
private:
  int32_t a;
};

void b ( C & c )
{
  int32_t & a_ref = c.getA ();
  a_ref = 10; // External modification of private C::a
}
----


== Compliant Solution

----
class C
{
public:
  int32_t const & getA () // Compliant
  {
    return a;
  }
  void setA(int32_t a)
  {
    this.a = a;
  }
private:
  int32_t a;
};

void b ( C & c )
{
  int32_t const & a_ref = c.getA ();
  a_ref = 10; // Not allowed, use setter setA instead
}
----


== See

* MISRA {cpp}:2008, 9-3-2


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
