== Why is this an issue?

include::../rationale.adoc[]

include::../impact.adoc[]

== How to fix it

=== Code examples

include::../../common/fix/code-rationale.adoc[]

==== Noncompliant code example

[source,python,diff-id=1,diff-type=noncompliant]
----
from django.db.models.expressions import RawSQL

def search(user):
  RawSQL("SELECT ssn FROM users WHERE id = " + user)  # Noncompliant
----

==== Compliant solution

[source,python,diff-id=1,diff-type=compliant]
----
from django.db import connection

def search(user):
  cursor = connection.cursor(prepared=True)
  cursor.execute("SELECT ssn FROM users WHERE id = %s", (user))
  connection.commit()
----

=== How does this work?

include::../../common/fix/how.adoc[]

== Resources

include::../common/resources/standards.adoc[]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../highlighting.adoc[]

endif::env-github,rspecator-view[]
