<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when a method parameter has a default value that is a non-empty hash literal.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>When you define a method parameter with a default hash containing multiple key-value pairs, Ruby completely replaces the entire default hash when any options are passed. This means that providing just one option will cause all other default values to be lost.</p>
</div>
<div class="paragraph">
<p>For example, if you have a method with <code>options = {age: 27, weight: 160, city: "New York"}</code> as the default, calling the method with <code>{age: 30}</code> will result in <code>options</code> being <code>{age: 30}</code> - the <code>weight</code> and <code>city</code> defaults are completely gone.</p>
</div>
<div class="paragraph">
<p>This behavior is often unexpected and can lead to bugs where methods don&#8217;t work correctly when only some options are provided. The method will receive <code>nil</code> values for the missing keys, which can cause errors or incorrect behavior.</p>
</div>
<div class="paragraph">
<p>The root cause is that Ruby&#8217;s parameter default values are evaluated once when the method is defined, and when you pass an argument, it completely replaces the default value rather than merging with it.</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="paragraph">
<p>This issue can lead to runtime errors when methods try to access missing hash keys, returning <code>nil</code> instead of expected default values. It can also cause incorrect application behavior when only partial configuration is provided, making the code unreliable and harder to debug.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Use an empty hash as the default parameter and merge it with a defaults hash inside the method. This ensures that user-provided options are combined with defaults rather than replacing them entirely.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-ruby" data-lang="ruby">def configure_user(name, options = {age: 27, weight: 160, city: "New York"}) # Noncompliant
  age = options[:age]
  weight = options[:weight]
  city = options[:city]
  puts "#{name}: age=#{age}, weight=#{weight}, city=#{city}"
end</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-ruby" data-lang="ruby">def configure_user(name, options = {})
  defaults = {age: 27, weight: 160, city: "New York"}
  options = defaults.merge(options)

  age = options[:age]
  weight = options[:weight]
  city = options[:city]
  puts "#{name}: age=#{age}, weight=#{weight}, city=#{city}"
end</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Ruby Hash#merge documentation - <a href="https://ruby-doc.org/core/Hash.html#method-i-merge">Official documentation for Hash#merge method used to combine hashes</a></p>
</li>
<li>
<p>Ruby keyword arguments guide - <a href="https://ruby-doc.org/core/doc/syntax/methods_rdoc.html#label-Keyword+Arguments">Official Ruby documentation on keyword arguments as an alternative approach</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>