include::../description.adoc[]

== Noncompliant Code Example

----
var myNumber:int = 010;  // Noncompliant. myNumber will hold 8, not 10 - was this really expected?
----

== Compliant Solution

----
var myNumber:int = 8;
----

include::../see.adoc[]

ifdef::env-github,rspecator-view[]
'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
