This rule raises an issue when code uses the deprecated ``++UserInfo.getRoleAndSubordinates()++`` method, which will be removed in future Salesforce releases.

== Why is this an issue?

Salesforce is phasing out the ``++getRoleAndSubordinates()++`` method as part of the Winter '26 release. This method will stop working in Spring '26, potentially breaking your applications.

The ``++getRoleAndSubordinates()++`` method has been replaced by ``++getRoleAndSubordinatesInternal()++`` to provide better internal handling of role-based sharing groups. Both methods return the same data - a set of user IDs that includes the current user and all users in subordinate roles.

Using the deprecated method creates technical debt and puts your code at risk of runtime failures when Salesforce removes support for the old method.

=== What is the potential impact?

Applications using the deprecated method will experience runtime errors and broken functionality once Salesforce removes support in Spring '26. This can affect:

* SOQL queries that filter records based on role hierarchy
* Security and sharing logic that depends on role-based access
* Reports and dashboards that use role-based data filtering
* Any business processes that rely on hierarchical data access

== How to fix it

Replace ``++UserInfo.getRoleAndSubordinates()++`` with ``++UserInfo.getRoleAndSubordinatesInternal()++`` in SOQL queries. The replacement method provides identical functionality.

=== Code examples

==== Noncompliant code example

[source,apex,diff-id=1,diff-type=noncompliant]
----
List<Account> accounts = [SELECT Id, Name FROM Account WHERE Id IN (
    SELECT AccountId FROM Contact 
    WHERE OwnerId IN :UserInfo.getRoleAndSubordinates() // Noncompliant
)];
----

==== Compliant solution

[source,apex,diff-id=1,diff-type=compliant]
----
List<Account> accounts = [SELECT Id, Name FROM Account WHERE Id IN (
    SELECT AccountId FROM Contact 
    WHERE OwnerId IN :UserInfo.getRoleAndSubordinatesInternal()
)];
----

== Resources

=== Documentation

 * UserInfo Class Documentation - https://developer.salesforce.com/docs/atlas.en-us.apexref.meta/apexref/apex_methods_system_userinfo.htm[Official Salesforce documentation for the UserInfo class and its methods]

 * Winter '26 Release Notes - https://help.salesforce.com/s/articleView?id=release-notes.salesforce_release_notes.htm&release=208&type=5[Salesforce release notes detailing the deprecation of getRoleAndSubordinates()]
