=== on 7 Apr 2015, 09:22:11 Pierre-Yves Nicolas wrote:
I don't manage to compile the noncompliant code example.

=== on 8 Apr 2015, 15:14:44 Ann Campbell wrote:
\[~pierre-yves.nicolas] this rule is based on this: \https://scs.senecac.on.ca/~timothy.mckenna/RPG544/rpg_cls5.html


Perhaps this only applies to some (one?) compilers...?


I'm fine to drop this rule if you like.

=== on 8 Apr 2015, 15:41:19 Ann Campbell wrote:
BTW, I've just spent probably 20min trying to get the author's email address. Apparently it's only available if you're a student of Seneca College. :-(

=== on 16 Apr 2015, 09:45:09 Pierre-Yves Nicolas wrote:
All RPG IV references I could find define the "name" operand as required (e.g. \http://publib.boulder.ibm.com/iseries/v5r1/ic2924/books/c0925083696.htm#HDRZZREAD).

Maybe this rule applied to RPG III, but we don't support RPG III.

I would drop this rule.

