== Why is this an issue?

While a file I/O call will compile without a file name, such code is almost always an error. The omission of the file name, can cause unexpected results in multi-file programs.


=== Noncompliant code example

[source,rpg]
----
  * Noncompliant
C                   READ                    DS1                      90
----


=== Compliant solution

[source,rpg]
----
C                   READ      PF0001        DS1                      90
----


ifdef::env-github,rspecator-view[]
'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
