== Why is this an issue?

Shared coding conventions allow teams to collaborate efficiently. This rule checks that all namespace names match a provided regular expression.


The default configuration is the one recommended by Microsoft:

* Pascal casing, starting with an upper case character, e.g. Microsoft, System
* Short abbreviations of 2 letters can be capitalized, e.g. System.IO
* Longer abbreviations need to be lower cased

=== Noncompliant code example

With the default regular expression: ``++^([A-Z]{1,3}[a-z0-9]+)*([A-Z]{2})?(\.([A-Z]{1,3}[a-z0-9]+)*([A-Z]{2}))*$++``

[source,vbnet]
----
Namespace foo  ' Noncompliant
End Namespace
----

=== Compliant solution

[source,vbnet]
----
Namespace Foo  ' Compliant
End Namespace
----
ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

include::parameters.adoc[]

endif::env-github,rspecator-view[]
