== Why is this an issue?

The ``++<!DOCTYPE>++`` declaration tells the web browser which (X)HTML version is being used on the page, and therefore how to interpret the various elements.

Validators also rely on it to know which rules to enforce.

It should always preceed the ``++<html>++`` tag.


=== Noncompliant code example

[source,html]
----
<html>  <!-- Noncompliant -->
...
</html>
----


=== Compliant solution

[source,html]
----
<!DOCTYPE html>
<html>  <!-- Compliant -->
...
</html>
----



ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Insert a <!DOCTYPE> declaration before this <html> tag.


'''
== Comments And Links
(visible only on this page)

=== on 8 Jul 2013, 18:20:31 Freddy Mallet wrote:
Is implemented by \http://jira.codehaus.org/browse/SONARPLUGINS-2995

endif::env-github,rspecator-view[]
