== Why is this an issue?

The ``++<!DOCTYPE>++`` declaration tells the web browser which (X)HTML version is being used on the page, and therefore how to interpret the various elements.

Validators also rely on it to know which rules to enforce.

It should always preceed the ``++<html>++`` tag.


=== Noncompliant code example

[source,html]
----
<html>  <!-- Noncompliant -->
...
</html>
----


=== Compliant solution

[source,html]
----
<!DOCTYPE html>
<html>  <!-- Compliant -->
...
</html>
----



ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
