== Why is this an issue?

Shared coding conventions allow teams to collaborate effectively. To improve readability, blank lines in a program should not have end-of-line comments.


=== Noncompliant code example

In this example, the second line is blank except for the end-of-line comment, obscuring readability. The seventh line is commented-out, so an end-of-line comment is acceptable.

[source,rpg]
----
     I/COPY VALSRC,VALTEST                                                FIX001
                                                                          FIX001
      ** Data structure for subroutine VALDS                              FIX001
     I            DS                                                      FIX001
     I                                        5   60MMA                   FIX001
     I                                        7   80DDA                   FIX001
      *                                                                   FIX001
     I            DS                                                      FIX001
     I                                        5   60MMB                   FIX001
     I                                        7   80DDB                   FIX001
----


=== Compliant solution

[source,rpg]
----
     I/COPY VALSRC,VALTEST                                                FIX001

      ** Data structure for subroutine VALDS                              FIX001
     I            DS                                                      FIX001
     I                                        5   60MMA                   FIX001
     I                                        7   80DDA                   FIX001
      *                                                                   FIX001
     I            DS                                                      FIX001
     I                                        5   60MMB                   FIX001
     I                                        7   80DDB                   FIX001     
----



ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
