include::../description.adoc[]

== Noncompliant Code Example

----
$someArray[1] = "someValue";
$someArray[1] = "someOtherValue"; // The intention here was probably to write to another array key.
----
