== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,php,diff-id=1,diff-type=noncompliant]
----
$someArray[1] = "someValue";
$someArray[1] = "someOtherValue"; // Noncompliant
----

=== Compliant solution

[source,php,diff-id=1,diff-type=compliant]
----
$someArray[1] = "someValue";
$someArray[2] = "someOtherValue";
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../highlighting.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
