Restricting the number of exits from a loop is done in the interests of good structured programming. One ``++break++`` or ``++goto++`` statement is acceptable in a loop since this allows, for example, for dual-outcome loops or optimal coding.


== Noncompliant Code Example

With the default threshold of 1:

----
for (int i = 0; i < 10; i++) {
  if (...) {
    break;      //  Compliant
  }
  else if (...) {
    break;      //  Non-compliant - second jump from loop
  }
  else {
    ...
  }
}
while (...) {
  if (...) {
    break;      // Compliant
  }
  if (...) {
    break;      // Non-compliant - second jump from loop
  }
}
----


== Compliant Solution

----
for (int i = 0; i < 10; i++) {
  if (...) {
    break;      //  Compliant
  }
}
while (...) {
  if (...) {
    break;    // Compliant
  }
}
----


== See

* MISRA C:2004, 14.6 - For any iteration statement there shall be at most one break statement used for loop termination.
* MISRA {cpp}:2008, 6-6-4 - For any iteration statement there shall be no more than one break or goto statement used for loop termination.
* MISRA C:2012, 15.4 - There should be no more than one break or goto statement used to terminate any iteration statement



ifdef::env-github,rspecator-view[]
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
