<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>The attribute <code>noreturn</code> indicates that a function does not return. This information clarifies the behavior of the function and it allows the compiler to do optimizations.</p>
</div>
<div class="paragraph">
<p>It can also help the compiler (and static analyzer tools, i.e. us) provide better error messages:</p>
</div>
<div class="listingblock">
<div class="content">
<pre>__attribute__((noreturn)) void f();

int g(int b) {
  if (b == 5) {
    f();
    printf("Hello world\n"); // This is dead code, the compiler/static analyzer can now detect it
    // There is no returned value, but it is fine, the compiler/static analyzer knows not to warn about it
  } else {
    return 3;
  }
}</pre>
</div>
</div>
<div class="paragraph">
<p>This rule detects when the attribute <code>noreturn</code> can be added to a function.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">void g() { // Noncompliant
  abort();
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">__attribute__((noreturn)) void g() { // or [[noreturn]] for C++
  abort(); // Compliant
}</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>function could be declared with attribute 'noreturn'</p>
</div>
</div>
</div>
</div>