<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Starting from Java 18, it is possible to include code snippets in a javadoc using the <code>@snippet</code> tag. Inside a snippet, markup commands such as <code>// @highlight</code> or <code>// @replace</code>  are used to respectively highlight or replace a region inside the snippet. These markups should always be put at the end of line. Not doing so can lead to unexpected output (e.g. extra empty line could appear).</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">/**
 * A simple program.
 * {@snippet :
 * class HelloWorld {
 *     public static void main(String... args) {
 *         // @highlight substring="println"     // Noncompliant
 *         System.out.println("Hello World!");
 *     }
 * }
 * }
 */</code></pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">/**
 * A simple program.
 * {@snippet :
 * class HelloWorld {
 *     public static void main(String... args) {
 *         System.out.println("Hello World!");   // @highlight substring="println"
 *     }
 * }
 * }
 */</code></pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_see">See</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Oracleâ€™s Programmer&#8217;s Guide to Snippets: <a href="https://docs.oracle.com/en/&#8230;&#8203;/code-snippet/index.html">https://docs.oracle.com/en/java/javase/18/code-snippet/index.html</a></p>
</div>
<hr>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Move this markup to the end of the line it targets.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>The entirety of the markup</p>
</div>
</div>
</div>
</div>