<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>There are several compilations options available for Visual Basic source code and <code>Option Explicit</code> defines compiler behavior for implicit variable declarations. Not specifying <code>Option Explicit</code> will allow creating a variable by it&#8217;s first usage. This behavior can lead to unexpected runtime errors due to typos in variable names.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p><code>Option Explicit</code> should be added to every individual source file.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-vb6" data-lang="vb6">Sub DoSomething(First As String, Second As String)
    Parameter = Fist        ' New local variable "Fist" is created and assigned to new local variable "Parameter" instead of "First" argument.
    DoSomething(Parameter)
    Parametr = Second       ' "Second" argument is assigned to newly created variable "Parametr" instead of intended "Parameter".
    DoSomething(Parameter)  ' Value of "Parameter" is always Nothing
End Sub</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-vb6" data-lang="vb6">Option Explicit

Sub DoSomething(First As String, Second As String)
    Dim Parameter As String = First
    DoSomething(Parameter)
    Parameter = Second
    DoSomething(Parameter)
End Sub</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>