This rule reports when a class is placed in a location which deviates from the intended architecture.

== Why is this an issue?

This project has an intended architecture defined, which specifies high-level components and their allowed relationships.
This might include that specific classes should be located in specific namespaces or modules, or that namespaces are renamed.
This issue is being raised because this class’s location deviates from the intended architecture.

=== What is the potential impact?

When the code location deviates from the intended architecture, it gradually erodes the project’s structure.
As these deviations accumulate, the code becomes more complex and less consistent with the intent, making future development increasingly challenging.

== How to fix it

Move the relevant class to its intended location or rename the relevant namespace as intended.

=== Code examples

==== Noncompliant code example

If the intended architecture declares developers should use the namespace name `Example.Panel`, not `Example.Panels`, then the following code example is noncompliant:

[source,java,diff-id=1,diff-type=noncompliant]
----
namespace Example.Panels;

public class ShowCustomersPanel
{
    CustomerRepo customerRepo = ...;
}
----

==== Compliant solution

The solution is to move the class to its appropriate namespace, or respectively, to rename the namespace `Panels` to `Panel`.

[source,java,diff-id=1,diff-type=compliant]
----
namespace Example.Panel;

public class ShowCustomersPanel
{
    CustomerRepo customerRepo = ...;
}
----
