<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Function arguments should all have different names to prevent any ambiguity. Indeed, if arguments have the same name, the last duplicated argument hides all the previous arguments with the same name. This hiding makes no sense, reduces understandability and maintainability, and obviously can be error prone.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-php" data-lang="php">function compute($a, $a, $c) { // Noncompliant
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-php" data-lang="php">function compute($a, $b, $c) { // Compliant
}</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Rename the duplicated function parameter(s) "{}".</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_16_oct_2014_094844_linda_martin_wrote">on 16 Oct 2014, 09:48:44 Linda Martin wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] I think the array <code>arguments[i]</code> is specific to JavaScript. I guess the spec should be updated accordingly.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_16_oct_2014_113452_ann_campbell_wrote">on 16 Oct 2014, 11:34:52 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~linda.martin] I&#8217;m confused because the JavaScript subtask has this addendum to the description:</p>
</div>
<div class="quoteblock">
<blockquote>
<div class="paragraph">
<p>Furthermore, in JavaScript strict mode, declaring arguments with the same name produces an error.</p>
</div>
</blockquote>
</div>
</div>
<div class="sect2">
<h3 id="_on_12_nov_2019_113552_elena_vilchik_wrote">on 12 Nov 2019, 11:35:52 Elena Vilchik wrote:</h3>
<div class="paragraph">
<p>Looks like rule was deprecated by mistake, bringing it back to ready</p>
</div>
</div>
</div>
</div>