Proper use of the ``++angular++`` object is crucial when using the AngularJS framework. Misspell an ``++angular++`` function, and depending on the browser you may not get any errors at all, or the error you do get will be distinctly unhelpful.


This rule is intended to catch misspellings of the recognized ``++angular++`` functions. It checks that only the following functions are used:

* ``++directive++``
* ``++module++``
* ``++controller++``
* ``++provider++``
* ``++filter++``


== Noncompliant Code Example

----
angular.moudle('docsSimpleDirectiveLD', []);  // Noncompliant
----


== Compliant Solution

----
angular.module('docsSimpleDirectiveLD', []);  // Compliant
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
