<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p><code>QUALIFIED</code> data structures result in cleaner code because you can&#8217;t reference the fields without using the qualifying name. They also  allow you to have multiple sub-fields with the same name, meaning subfield names don&#8217;t have to be convoluted for uniqueness, and can be expressive instead.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-rpg" data-lang="rpg">      * Noncompliant
     D Employee        DS
     D  EmpId                         7P 0
     D  EFName                       30A
     D  ELName                       30A
     D  EPhone                       11P 0

      * Noncompliant
     D Contractor      DS
     D  CntId                         7P 0
     D  CFName                       30A
     D  CLName                       30A
     D  CPhone                       11P 0

      /free
        EmpId = '000220';
      /end-free</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-rpg" data-lang="rpg">     D Employee        DS                  QUALIFIED
     D  Id                            7P 0
     D  FName                        30A
     D  LName                        30A
     D  Phone                        11P 0

     D Contractor      DS                  QUALIFIED
     D  Id                            7P 0
     D  FName                        30A
     D  LName                        30A
     D  Phone                        11P 0

      /free
        Employee.Id = '000220';
      /end-free</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Mark this data structure "QUALIFIED".</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_2_apr_2015_172840_ann_campbell_wrote">on 2 Apr 2015, 17:28:40 Ann Campbell wrote:</h3>
<div class="paragraph">
<p><a href="http://www.bmeyers.net/faqs/14-tips/32-rpg-iv-style?start=3" class="bare">http://www.bmeyers.net/faqs/14-tips/32-rpg-iv-style?start=3</a></p>
</div>
</div>
</div>
</div>