== Why is this an issue?

The presence of cyclic dependencies between classes is detrimental to software quality attributes such as understandability, testability, 
reusability, and maintainability.

Arguably, in a cycle of classes, there is no reasonable starting point, and none of the classes involved makes sense on its own.

=== What is the potential impact?

Cyclic dependencies increase the coupling between the components of the codebase, making the software's architecture harder to understand and extend.
Also, they increase the likelihood of code to run into recursion problems.

Furthermore, if we try to visualize our codebase as a graph, to understand its structure, having cyclic dependencies could make the task nearly impossible.
On the other hand, the absence of them would result in a tree structure, where every component relies only on underlying components, and understanding the flow of the program will be much easier.

== How to fix it

=== Code examples

==== Noncompliant code example

The following code contains two cycles: Order &rarr; Customer &rarr; Order, and Order &rarr; Product &rarr; Order. Furthermore, both cycles are connected to each other via class Order. This is called _a tangle consisting of two cycles_.

[source,java,diff-id=1,diff-type=noncompliant]
----
class Order {
    public Customer customer;
    public List<Product> products;
}
class Customer {
    public List<Order> orders;
}
class Product {
    public List<Order> orders;
}
----

==== Compliant solution

The issue can be resolved by refactoring the architecture. The dependencies Customer &rarr; Order and Product &rarr; Order have been replaced by two service functions.

[source,java,diff-id=1,diff-type=compliant]
----
class Order {
    public Customer customer;
    public List<Product> products;
}
class Customer { }
class Product { }
interface OrderService {
  List<Order> getOrdersByCustomer(Customer customer);
  List<Order> getOrdersByProduct(Product product);
}
----

== Resources

- Wikipedia - https://en.wikipedia.org/wiki/Acyclic_dependencies_principle[Acyclic dependencies principle]
- STAN - https://stan4j.com/advanced/adp/[Acyclic dependencies principle]
