== Why is this an issue?

The presence of cyclic dependencies between classes is detrimental to software quality attributes such as understandability, testability, 
reusability, and maintainability.

Arguably, in a cycle of classes, there is no reasonable starting point, and none of the classes involved makes sense on its own.

=== What is the potential impact?

Having these cycles greatly raises the coupling between components of the codebase, making it extremely hard to have separate testing, reuse or release.

Furthermore, if we try to visualize our codebase as a graph, to understand its structure, having cyclic dependencies could make the task nearly impossible.
On the other hand, the absence of them would result in a tree structure, where every component relies only on underlying components, and understanding the flow of the program will be much easier.

== How to fix it

=== Code examples

==== Noncompliant code example

[source,java,diff-id=1,diff-type=noncompliant]
----
FIXME
----

==== Compliant solution

[source,java,diff-id=1,diff-type=compliant]
----
FIXME
----

== Resources

- Wikipedia - https://en.wikipedia.org/wiki/Acyclic_dependencies_principle[Acyclic dependencies principle]
- STAN - https://stan4j.com/advanced/adp/[Acyclic dependencies principle]
