Shared coding conventions allow teams to collaborate efficiently. This rule checks that logical file names conform to a provided regular expression.


== Noncompliant Code Example

Given the regular expression ``++FILE-.*++``:

----
 INPUT-OUTPUT SECTION.
 FILE-CONTROL.

     SELECT Y27MVTS       ASSIGN  TO     S1       >Noncompliant
                          FILE STATUS IS FS-S1.
----


== Compliant Solution

----
 INPUT-OUTPUT SECTION.
 FILE-CONTROL.

     SELECT FILE-X345      ASSIGN  TO     S1
                          FILE STATUS IS FS-S1.
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::parameters.adoc[]

endif::env-github,rspecator-view[]
