== Why is this an issue?

Shared coding conventions allow teams to collaborate efficiently. This rule checks that logical file names conform to a provided regular expression.


=== Noncompliant code example

Given the regular expression ``++FILE-.*++``:

[source,cobol]
----
 INPUT-OUTPUT SECTION.
 FILE-CONTROL.

     SELECT Y27MVTS       ASSIGN  TO     S1       >Noncompliant
                          FILE STATUS IS FS-S1.
----


=== Compliant solution

[source,cobol]
----
 INPUT-OUTPUT SECTION.
 FILE-CONTROL.

     SELECT FILE-X345      ASSIGN  TO     S1
                          FILE STATUS IS FS-S1.
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Parameters

.regexPatternString
****

A regular expression to specify the logical file naming convention
****


endif::env-github,rspecator-view[]
