== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,sql]
----
CREATE FUNCTION my_function_ RETURN PLS_INTEGER AS -- Noncompliant
BEGIN
  RETURN 42;
END;
/
----

=== Compliant solution

[source,sql]
----
CREATE FUNCTION my_function RETURN PLS_INTEGER AS
BEGIN
  RETURN 42;
END;
/
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::parameters.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
