== Why is this an issue?

Schema versioning is a big topic, with best practices rather than hard-and-fast rules. One of these best practices is to include the ``++version++`` attribute in the ``++xsd:schema++`` and increment it to indicate minor changes. Nothing breaks if the value changes, but it does give the user a relatively prominent indication that something has changed.


=== Noncompliant code example

[source,xml]
----
<xs:schema targetNamespace="http://www.codeSamples.com/fruit"
           xmlns:xs="http://www.w3.org/2001/XMLSchema"
           elementFormDefault="qualified">  <!-- Noncompliant -->
...
----


=== Compliant solution

[source,xml]
----
<xs:schema targetNamespace="http://www.codeSamples.com/fruit"
           xmlns:xs="http://www.w3.org/2001/XMLSchema"
           elementFormDefault="qualified" version="1">
...
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Add a "version" attribute to this schema.


=== Highlighting

``++xsd:schema++``


endif::env-github,rspecator-view[]
