This rule reports when a source file is placed in to the wrong directory regarding the intended architectural design.

== Why is this an issue?

The software architect may decide that specific files should be located in specific directories, or that directories are misnamed.
These decisions reflect the intended structure of the software and responsibilities of code components (files, directories).
The reported misplaced file violates these design decisions.

=== What is the potential impact?

When new code violates design decisions, the project's structure gradually drift away from its intended architecture.
As these violations accumulate, the code becomes more complex and less consistent with the architectâ€™s original design, making future development and maintenance increasingly challenging.

== How to fix it

Move the proclaimed file to its intended directory, or rename the proclaimed directory as intended.

=== Code examples

==== Noncompliant code example

If the intended design declares developers should use the directory name `panel/`, not `panels/`, then the following code example is noncompliant:

[source,javascript,diff-id=1,diff-type=noncompliant]
----
// panels/ShowCustomersPanel.js

export class ShowCustomersPanel {
  // ...
}
----

==== Compliant solution

The solution is to move the file to its appropriate directory, or respectively, to rename the directory `panels` to `panel`.

[source,javascript,diff-id=1,diff-type=compliant]
----
// panel/ShowCustomersPanel.js

export class ShowCustomersPanel {
  // ...
}
----
