Shared coding conventions allow teams to collaborate effectively. This rule checks that classes extending ``++React.PureComponent++`` or ``++React.Component++`` are named with UpperCamelCase (aka PascalCase).


== Noncompliant Code Example

----
export default class my-super-component extends React.PureComponent { // Noncompliant
 ...
}
----


== Compliant Solution

----
export default class MySuperComponent extends React.PureComponent {
 ...
}
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

include::highlighting.adoc[]

endif::env-github,rspecator-view[]
