== Why is this an issue?

Shared coding conventions allow teams to collaborate effectively. This rule checks that classes extending ``++React.PureComponent++`` or ``++React.Component++`` are named with UpperCamelCase (aka PascalCase).


=== Noncompliant code example

[source,javascript]
----
export default class my-super-component extends React.PureComponent { // Noncompliant
 ...
}
----


=== Compliant solution

[source,javascript]
----
export default class MySuperComponent extends React.PureComponent {
 ...
}
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Rename this class to be compliant with the UpperCamelCase naming convention.


=== Highlighting

Primary: Class name


endif::env-github,rspecator-view[]
