include::../description.adoc[]

== Noncompliant Code Example

----
CASE SY-INDEX.
  WHEN ONE. // 6 lines till next WHEN
    PERFORM sub1.
    PERFORM sub2.
    PERFORM sub3.
    PERFORM sub4.
    PERFORM sub5.
  WHEN 2.
...
ENDCASE.
----

== Compliant Solution

----
CASE SY-INDEX.
  WHEN ONE. 
    PERFORM extractedSub.
  WHEN 2.
...
ENDCASE.

...

FORM extractedSub.
  PERFORM sub1.
  PERFORM sub2.
  PERFORM sub3.
  PERFORM sub4.
  PERFORM sub5.
ENDFORM.
----

ifdef::rspecator-view[]
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::rspecator-view[]
