<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule reports an issue when a lambda invokes another lambda synchronously.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Invoking other Lambdas synchronously from a Lambda is a scalability anti-pattern. Lambdas have a maximum execution time before they timeout (15 minutes as of June 2025). Having to wait for another Lambda to finish its execution could lead to a timeout.</p>
</div>
<div class="paragraph">
<p>A better solution is to generate events that can be consumed asynchronously by other Lambdas.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">import boto3

client = boto3.client('lambda')

def lambda_handler(event, context):
    response = client.invoke(
      FunctionName='target-lambda-function-name',
      InvocationType='RequestResponse' # Noncompliant `RequestResponse` InvocationType means that the lambda call is synchronous
      Payload=json.dumps(event)
    )

    return response['Payload'].read()</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_code_example">Compliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">import boto3

client = boto3.client('lambda')

def lambda_handler(event, context):

    response = client.invoke(
      FunctionName='target-lambda-function-name',
      InvocationType='Event' # Compliant `Event` InvocationType means that the lambda call is asynchronous
      Payload=json.dumps(event)
    )

    return response['Payload'].read()</code></pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it_in_boto3">How to fix it in Boto3</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Using the <code>client.invoke()</code> function, the parameter <code>InvocationType</code> should be set to <code>Event</code> to invoke the function asynchronously by sending events.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p>AWS docs - <a href="https://docs.aws.amazon.com/lambda/latest/dg/best-practices.html">Best practices for working with AWS Lambda functions</a></p>
</li>
<li>
<p>AWS docs - <a href="https://docs.aws.amazon.com/lambda/latest/dg/lambda-runtime-environment.html">Understanding the Lambda execution environment lifecycle</a></p>
</li>
<li>
<p>boto3 - <a href="https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/lambda/client/invoke.html">client.invoke()</a></p>
</li>
</ul>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Avoid synchronous calls to other lambdas</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>Function invocation</p>
</div>
</div>
</div>
</div>