<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>It is fairly normal for COBOL development teams to decide to work either with sections or with paragraphs and to make this choice a standard.</p>
</div>
<div class="paragraph">
<p>When sections are used, it is also normal to define another standard: "End every section definition with an empty paragraph definition, or a paragraph containing only a terminating statement".</p>
</div>
<div class="paragraph">
<p>This empty paragraph can then be jumped to with a <code>GO TO</code> statement to stop the execution of a section.</p>
</div>
<div class="paragraph">
<p>Accepted terminating statements in the otherwise-empty ending paragraph are: <code>EXIT</code>, <code>EXIT PROGRAM</code>, <code>STOP RUN</code>, and <code>GO BACK</code>.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="paragraph">
<p>In this example, an empty paragraph is missing at the end of the first section definition.</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cobol" data-lang="cobol">FIRST_SECTION SECTION.
  ...

SECOND_SECTION SECTION.
  ...
SECOND_SECTION_END.</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cobol" data-lang="cobol">FIRST_SECTION SECTION.
  ...
FIRST_SECTION_END.

SECOND_SECTION SECTION.
  ...
SECOND_SECTION_END.</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Add an empty paragraph at the end of this "XXXX" section.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_relates_to_s1594">relates to: <a data-rspec-id="S1594" class="rspec-auto-link">S1594</a></h3>

</div>
<div class="sect2">
<h3 id="_on_20_nov_2013_123424_dinesh_bolkensteyn_wrote">on 20 Nov 2013, 12:34:24 Dinesh Bolkensteyn wrote:</h3>
<div class="paragraph">
<p>I am wondering how this could conflict with COBOL.UnusedParagraphOrSectionLabelCheck.</p>
</div>
</div>
</div>
</div>