It is fairly normal for COBOL development teams to decide to work either with sections or with paragraphs and to make this choice a standard.
When sections are used, it is also normal to define another standard: "End every section definition with an empty paragraph definition, or a paragraph containing only a terminating statement".
This empty paragraph can then be jumped to with a `+GO TO+` statement to stop the execution of a section.
Accepted terminating statements in the otherwise-empty ending paragraph are: `+EXIT+`, `+EXIT PROGRAM+`, `+STOP RUN+`, and `+GO BACK+`.


== Noncompliant Code Example

In this example, an empty paragraph is missing at the end of the first section definition.

----
FIRST_SECTION SECTION.
  ...

SECOND_SECTION SECTION.
  ...
SECOND_SECTION_END.
----


== Compliant Solution

----
FIRST_SECTION SECTION.
  ...
FIRST_SECTION_END.

SECOND_SECTION SECTION.
  ...
SECOND_SECTION_END.
----


