include::../description.adoc[]

== Noncompliant Code Example

----
function isOdd(x) {
  return x % 2 == 1;  // Noncompliant; if x is an odd negative, x % 2 == -1
}
----

== Compliant Solution

----
function isOdd(int x) {
  return x % 2 != 0;
}
----

include::../see.adoc[]

ifdef::rspecator-view[]
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::rspecator-view[]
