== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,javascript]
----
function isOdd(x) {
  return x % 2 == 1;  // Noncompliant; if x is an odd negative, x % 2 == -1
}
----

=== Compliant solution

[source,javascript]
----
function isOdd(int x) {
  return x % 2 != 0;
}
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
