This rule raises an issue when the `tf.divide` function arguments have incompatible shapes.

== Why is this an issue?

The `tf.divide` function in TensorFlow is used to perform element-wise division between two tensors. It divides the corresponding elements of the first tensor by the corresponding elements of the second tensor, e.g.:

[source,python]
----
x = tf.constant([4,6,8])
y = tf.constant([2,3,4])
----
When you use `tf.divide(x, y)`, TensorFlow will perform element-wise division between `x` and `y`. So, the output would be:
[source]
----
[4 / 2, 6 / 3, 8 / 4] = [2.0 2.0, 2.0]
----
Here, each element of `x` is divided by the corresponding element of `y`.

One limitation to be aware of when using `tf.divide` is that both `x` and `y` must be of compatible shapes. In other words, they must have the same shape or be broadcastable to the same shape. Otherwise it will lead to `InvalidArgumentError`.

== How to fix it
Make sure that the `tf.divide` function arguments have compatible shapes.

=== Code examples

==== Noncompliant code example

[source,python,diff-id=1,diff-type=noncompliant]
----
import tensorflow as tf

x = tf.constant([4, 6, 8])
y = tf.constant([2, 3])
z = tf.divide(x, y)  # Noncompliant: InvalidArgumentError: Incompatible shapes: [3] vs. [2]
----

==== Compliant solution

[source,python,diff-id=1,diff-type=compliant]
----
import tensorflow as tf

x = tf.constant([4, 6, 8])
y = tf.constant([2, 3, 4])
z = tf.divide(x, y)  # OK: [2, 2, 2]
----

//=== How does this work?

//=== Pitfalls

//=== Going the extra mile


== Resources
=== Documentation
* Tensorflow documentation - https://www.tensorflow.org/api_docs/python/tf/math/divide[tf.divide]
* Numpy documentation - https://numpy.org/doc/stable/user/basics.broadcasting.html#broadcasting[Broadcasting]
//=== Articles & blog posts
//=== Conference presentations
//=== Standards
//=== External coding guidelines
//=== Benchmarks
