Assigning between objects that have an overlap in their physical storage leads to undefined behaviour.


== Noncompliant Code Example

----
struct s
{
  int16_t m1 [ 32 ];
};

struct t
{
  int32_t m2;
  struct s m3;
};

void fn ( )
{
  union
  {
    struct s u1;
    struct t u2;
  } a;
  a.u2.m3 = a.u1; // Noncompliant
}
----


== See

* MISRA C:2004, 18.2
* MISRA {cpp}:2008, 0-2-1

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

endif::env-github,rspecator-view[]
