== Why is this an issue?

Assigning between objects that have an overlap in their physical storage leads to undefined behaviour.


=== Noncompliant code example

[source,cpp]
----
struct s
{
  int16_t m1 [ 32 ];
};

struct t
{
  int32_t m2;
  struct s m3;
};

void fn ( )
{
  union
  {
    struct s u1;
    struct t u2;
  } a;
  a.u2.m3 = a.u1; // Noncompliant
}
----


== Resources

* MISRA C:2004, 18.2
* MISRA {cpp}:2008, 0-2-1

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Remove this assignment between overlapping memory locations.


endif::env-github,rspecator-view[]
