Procedural development in general, and ``FORM... ENDFORM``, and ``PERFORM`` specifically, have been been classified as obsolete by SAP and should be avoided. Classes and methods should be used for all new development.


== Noncompliant Code Example

----
FORM fill_table USING    wa   TYPE any 
                CHANGING ptab TYPE INDEX TABLE. 
  APPEND wa TO ptab. 
ENDFORM. 
* ...
PERFORM fill_table IN PROGRAM my_prog.
----

