include::../description.adoc[]

== Noncompliant Code Example

----
class MyException: Exception
{
    public void MyException()
    {
         if (bad_thing) 
         {
             throw new Exception("A bad thing happened");  // Noncompliant
          }
    }
}
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

include::../highlighting.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
