Any statement after a ``++STOP RUN++`` or ``++GOBACK++`` is unreachable and therefore dead code which should be removed.


== Noncompliant Code Example

----
PARAGRAPH1.  
  MOVE A TO B.         
  STOP RUN. 
  MOVE B TO C.
----

or

----
PARAGRAPH1.  
  MOVE A TO B.         
  GOBACK. 
  MOVE B TO C.
----


== Compliant Solution

----
PARAGRAPH1.  
  MOVE A TO B.         
  MOVE B TO C.
  STOP RUN. 
----

or

----
PARAGRAPH1.  
  MOVE A TO B.
  MOVE B TO C.         
  GOBACK. 
----

