== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,csharp]
----
using System.Reflection;

[assembly: AssemblyTitle("MyAssembly")] // Noncompliant

namespace MyLibrary
{
}
----


=== Compliant solution

[source,csharp]
----
using System.Reflection;

[assembly: AssemblyTitle("MyAssembly")] 
[assembly: AssemblyVersion("1.2.125.0")]

namespace MyLibrary
{
}
----


include::../see.adoc[]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../highlighting.adoc[]

endif::env-github,rspecator-view[]
