include::../why.adoc[]

== How to fix it

=== Code examples

==== Noncompliant code example

[source,csharp,diff-id=1,diff-type=noncompliant]
----
using System.Reflection;

[assembly: AssemblyTitle("MyAssembly")] // Noncompliant
namespace MyLibrary
{
    // ...
}
----

==== Compliant solution

[source,csharp,diff-id=1,diff-type=compliant]
----
using System.Reflection;

[assembly: AssemblyTitle("MyAssembly")] 
[assembly: AssemblyVersion("42.1.125.0")]
namespace MyLibrary
{
    // ...
}
----

include::../resources.adoc[]

include::../rspecator.adoc[]