If the second procedure of a PERFORM THRU is not defined after the first one, the source code is semantically incorrect and the program doesn't behave as expected.


== Noncompliant Code Example

----
  PERFORM SECOND-P THRU FIRST-P.
  ...

 FIRST-P.
   ...

 SECOND-P.
   ...
----


== Compliant Solution

----
  PERFORM FIRST-P THRU SECOND-P.
  ...

 FIRST-P.
   ...

 SECOND-P.
   ...
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

endif::env-github,rspecator-view[]
