This rule raises an issue when a function is declared without proper documentation comments that describe its purpose, parameters, and return values.

== Why is this an issue?

Functions without documentation comments make code harder to understand and maintain. When developers encounter an undocumented function, they must read through the entire implementation to understand what it does, what parameters it expects, and what it returns.

This becomes especially problematic in larger codebases where:

* Multiple developers work on the same code
* Functions are called from different parts of the application
* Code needs to be maintained months or years after it was written
* New team members need to understand existing functionality

The Google Shell Style Guide requires function comments to improve code readability and maintainability. Well-documented functions serve as a contract between the function author and its users, making the codebase more professional and easier to work with.

Documentation comments also help with code reviews, debugging, and refactoring by providing clear context about the function's intended behavior.

=== What is the potential impact?

Undocumented functions reduce code maintainability and increase the time needed for code reviews, debugging, and onboarding new developers. This can lead to misunderstandings about function behavior, incorrect usage, and increased development costs over time.

== How to fix it

Add documentation comments above the function declaration. Include a brief description of the function's purpose, document all parameters with their expected types or formats, and describe the return value if applicable.

=== Code examples

==== Noncompliant code example

[source,bash,diff-id=1,diff-type=noncompliant]
----
function process_data() {
  local input="$1"
  local output_file="$2"
  
  if [[ -z "$input" ]]; then
    return 1
  fi
  
  echo "Processing: $input" > "$output_file"
  return 0
} # Noncompliant
----

==== Compliant solution

[source,bash,diff-id=1,diff-type=compliant]
----
# Processes input data and writes it to a file
# Arguments:
#   input: The data to process (string)
#   output_file: Path to the output file (string)
# Returns:
#   0 on success, 1 if input is empty
function process_data() {
  local input="$1"
  local output_file="$2"
  
  if [[ -z "$input" ]]; then
    return 1
  fi
  
  echo "Processing: $input" > "$output_file"
  return 0
}
----

== Resources
=== Documentation

 * Google Shell Style Guide - Comments - https://google.github.io/styleguide/shellguide.html#comments[Official Google Shell Style Guide section on function comments and documentation requirements]
 * BashSupport Pro - Google Style Guide - https://www.bashsupport.com/manual/google-style-guide/[Tool support for Google Shell Style Guide including function comment inspections]

