== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,rpg]
----
SELECT * 
       FROM persons 
       INTO newyorkers 
       WHERE city = 'NEW YORK'
----

=== Compliant solution

[source,rpg]
----
SELECT firstname, lastname 
       FROM persons 
       INTO newyorkers
       WHERE city = 'NEW YORK'
----
ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

endif::env-github,rspecator-view[]
