== Why is this an issue?

Using the same value on either side of a binary operator is almost always a mistake. In the case of logical operators, it is either a copy/paste error and therefore a bug, or it is simply wasted code, and should be simplified. In the case of bitwise operators and most binary mathematical operators, having the same value on both sides of an operator yields predictable results, and should be simplified.


This rule ignores ``++*++``, ``+``. 

=== Noncompliant code example

[source,swift]
----
if a == a { // always true
  doZ()
}
if  a != a  { // always false
  doY()
}
if a == b && a == b { // if the first one is true, the second one is too
  doX()
}
if a == b || a == b { // if the first one is true, the second one is too
  doW()
}

var j = 5 / 5 //always 1
var k = 5 - 5 //always 0
----

=== Exceptions

Left-shifting 1 onto 1 is common in the construction of bit masks, and is ignored.

[source,swift]
----
var i = 1 << 1; // Compliant
var j = a << a; // Noncompliant
----

include::../see.adoc[]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../highlighting.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
