<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p><code>JOIN</code> bypasses the SAP table buffer. Buffered tables should be accessed with the simplest <code>SELECT</code> statements possible so as not to risk bypassing the buffer.</p>
</div>
<div class="paragraph">
<p>If one of the tables in a <code>JOIN</code> is buffered, it would be an advantage to first import the required entries using a <code>SELECT</code> into an internal table <code>itab</code>, and then for example, using the statement <code>SELECT ... FOR ALL ENTRIES IN itab</code> to access further tables.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="paragraph">
<p>For <code>JOIN</code> clauses:</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-abap" data-lang="abap">SELECT s~carrid s~carrname p~connid
       INTO CORRESPONDING FIELDS OF TABLE itab
       FROM scarr AS s
       LEFT OUTER JOIN spfli AS p ON s~carrid   =  p~carrid
            AND p~cityfrom = p_cityfr.</code></pre>
</div>
</div>
<div class="paragraph">
<p>For subqueries</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-abap" data-lang="abap">SELECT  carrname
  INTO  TABLE name_tab
  FROM  scarr
  WHERE EXISTS ( select  *
                   FROM  spfli
                   WHERE carrid   =  scarr~carrid AND
                         cityfrom = 'NEW YORK'        ).</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Make sure this use of  a "JOIN" clause is strongly indicated.</p>
</div>
<div class="paragraph">
<p>Make sure this use of a subquery is strongly indicated.</p>
</div>
</div>
</div>
</div>