JNI (Java Native Interface) code should be used only as a last resort, since part of the point of using Java is to make applications portable, and by definition the use of JNI can reduce portability.

== Noncompliant Code Example

----
System.loadLibrary("nativeStringLib");  // Noncompliant
----
