== Why is this an issue?

JNI (Java Native Interface) code should be used only as a last resort, since part of the point of using Java is to make applications portable, and by definition the use of JNI can reduce portability.


=== Noncompliant code example

[source,java]
----
System.loadLibrary("nativeStringLib");  // Noncompliant
----



ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Remove this use of the Java Native Interface.


'''
== Comments And Links
(visible only on this page)

=== on 16 Jun 2015, 17:06:50 Nicolas Peru wrote:
looks good.

endif::env-github,rspecator-view[]
