<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This is an issue when two or more consecutive function parameters have the same type but are declared with separate type annotations.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Go provides a convenient syntax feature that allows you to group consecutive parameters of the same type together. Instead of repeating the type for each parameter, you can omit the type from all but the last parameter in the group.</p>
</div>
<div class="paragraph">
<p>This feature serves multiple purposes:</p>
</div>
<div class="ulist">
<ul>
<li>
<p><strong>Reduces visual clutter</strong>: Eliminating redundant type declarations makes function signatures cleaner and easier to read.</p>
</li>
<li>
<p><strong>Follows Go idioms</strong>: The Go community and official documentation consistently use this syntax pattern, making it the expected style.</p>
</li>
<li>
<p><strong>Improves maintainability</strong>: When you need to change the type of grouped parameters, you only need to update it in one place.</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>The Go language designers included this feature specifically to make code more concise while maintaining clarity. Using the grouped syntax demonstrates familiarity with Go conventions and produces code that feels natural to other Go developers.</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="paragraph">
<p>This issue affects code readability and consistency. While it doesn&#8217;t impact functionality, using non-idiomatic Go syntax can make code appear less professional and may slow down code reviews as developers notice the style inconsistency.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Group consecutive parameters of the same type by removing the type annotation from all but the last parameter in the group.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-go" data-lang="go">func add(x int, y int) int { // Noncompliant
    return x + y
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-go" data-lang="go">func add(x, y int) int {
    return x + y
}</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>A Tour of Go - Functions continued - <a href="https://go.dev/tour/basics/5">Official Go tutorial explaining how to group consecutive parameters of the same type</a></p>
</li>
<li>
<p>Go&#8217;s Declaration Syntax - <a href="https://go.dev/blog/gos-declaration-syntax">Blog post explaining the reasoning behind Go&#8217;s declaration syntax design</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>