== Why is this an issue?

CSS allows duplicate property names but only the last instance of a duplicated name determines the actual value that will be used for it. Therefore, changing values of other occurrences of a duplicated name will have no effect and may cause misunderstandings and bugs.


This rule ignores ``++$sass++``, ``++@less++``, and ``++var(--custom-property)++`` variable syntaxes.


=== Noncompliant code example

[source,css]
----
a { 
  color: pink; 
  background: orange; 
  color: orange
}
----


=== Compliant solution

[source,css]
----
a { 
  color: pink;
  background: orange
}
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Remove XXX, it is duplicated.


=== Parameters

.ignoreFallbacks
****

----
true
----

Ignore consecutive duplicated properties with different values.
****


=== Highlighting

The duplicated property


endif::env-github,rspecator-view[]
