=== Code examples

==== Noncompliant code example

[source,csharp,diff-id=1,diff-type=noncompliant]
----
bool IsAMatch(string inputString)
{
    var regex = new Regex("^[a-zA-Z]$");

    return regex.IsMatch(inputString);
}
----

==== Compliant solution

[source,csharp,diff-id=1,diff-type=compliant]
----
readonly Regex ipv6Regex = new Regex("^[a-zA-Z]$");

bool IsAMatch(string inputString)
    => ipv6Regex.IsMatch(inputString);
----