=== Code examples

==== Noncompliant code example

[source,csharp,diff-id=1,diff-type=noncompliant]
----
bool Matches(string inputString)
{
    var myRegex = new Regex("^[a-zA-Z]$");
    return myRegex.IsMatch(inputString);
}
----

==== Compliant solution

[source,csharp,diff-id=1,diff-type=compliant]
----
readonly Regex myRegex = new Regex("^[a-zA-Z]$");

bool IsAMatch(string inputString) =>
    ipv6Regex.IsMatch(inputString);
----