== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,java,diff-id=1,diff-type=noncompliant]
----
class Parent {
  public static int counter;
}

class Child extends Parent {
  public Child() {
    Child.counter++;  // Noncompliant
  }
}
----

=== Compliant solution

[source,java,diff-id=1,diff-type=compliant]
----
class Parent {
  public static int counter;
}

class Child extends Parent {
  public Child() {
    Parent.counter++;
  }
}
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Use static access for "X.y".


'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
