== Why is this an issue?

Exceptions are meant to represent the application's state at the point at which an error occurred.


Making all fields in an ``++Exception++`` class ``++final++`` ensures that this state:

* Will be fully defined at the same time the ``++Exception++`` is instantiated,
* Won't be updated or corrupted by a questionable error handler.

This will enable developers to quickly understand what went wrong.

== How to fix it

Declare all the fields in an ``++Exception++`` class as ``++final++``.

=== Code examples

==== Noncompliant code example

[source,java,diff-id=1,diff-type=noncompliant]
----
public class MyException extends Exception {

  private int status;                        // Noncompliant

  public MyException(String message) {
    super(message);
  }

  public int getStatus() {
    return status;
  }

  public void setStatus(int status) {
    this.status = status;
  }

}
----


==== Compliant solution

[source,java,diff-id=1,diff-type=compliant]
----
public class MyException extends Exception {

  private final int status;

  public MyException(String message, int status) {
    super(message);
    this.status = status;
  }

  public int getStatus() {
    return status;
  }

}
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Make this field "xxxx" final.


'''
== Comments And Links
(visible only on this page)

=== on 30 Jul 2013, 16:55:45 Freddy Mallet wrote:
Is implemented by \http://jira.codehaus.org/browse/SONARJAVA-254

=== on 14 Mar 2017, 11:40:35 Amaury Lev√© wrote:
\[~valeri.hristov] This seems to be a nice rule to implement for C#. WDYT?

endif::env-github,rspecator-view[]
