This rule raises an issue when a semicolon is found at the end of a statement in Groovy code.

== Why is this an issue?

Groovy is designed to be more concise and readable than Java. One of its key features is that semicolons are optional at the end of statements.

Using semicolons in Groovy code goes against the language's conventions and style guidelines. The official Groovy style guide explicitly recommends omitting semicolons to make code more idiomatic and cleaner.

When developers come from Java or other C-style languages, they often carry over the habit of using semicolons everywhere. However, in Groovy, this creates unnecessary visual noise and makes the code less idiomatic.

Removing semicolons helps achieve:

* More readable and cleaner code
* Better adherence to Groovy conventions
* Consistency with idiomatic Groovy style
* Reduced visual clutter

=== What is the potential impact?

The impact is primarily on code maintainability and readability. While semicolons don't cause functional issues, they make the code less idiomatic and can indicate that the developer is not following Groovy best practices.

Code with unnecessary semicolons may:

* Appear less professional to Groovy developers
* Create inconsistency in codebases where some files follow Groovy conventions and others don't
* Make the code harder to read due to visual noise

== How to fix it

Remove the semicolon at the end of variable declarations and assignments.

=== Code examples

==== Noncompliant code example

[source,groovy,diff-id=1,diff-type=noncompliant]
----
int count = 5; // Noncompliant
----

==== Compliant solution

[source,groovy,diff-id=1,diff-type=compliant]
----
int count = 5
----

== Resources

=== Documentation

 * Groovy Style Guide - No semicolons - https://groovy-lang.org/style-guide.html#_no_semicolons[Official Groovy style guide section on omitting semicolons]
