``++ID++`` and ``++IDREF++`` are constructs inherited from DTDs, and they suffer from painful drawbacks, mainly:

* ``++ID++``s are too widely scoped: an ``++ID++`` must be unique in the XML document. This means that if an XML document contains ``++<Person>++`` elements and ``++<Dog>++`` elements, both with a "name" attribute of ``++type="xs:ID"++``, then two people can't have the same name. Neither can two dogs, or a person and a dog.
* IDs must comply with tight naming scheme. For example, "123" is not a valid value for an ID.

Instead, ``++KEY++`` and ``++KEYREF++`` should be used. And if you really need to insure uniqueness, use ``++xs:unique++`` to add a uniqueness constraint.


== Noncompliant Code Example

----
ToDo
----


== Compliant Solution

----
ToDo
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

include::highlighting.adoc[]

endif::env-github,rspecator-view[]
