== How to fix it in ASP.NET

=== Code examples

include::../../common/fix/code-rationale.adoc[]

==== Noncompliant code example

[source,csharp,diff-id=1,diff-type=noncompliant]
----
using System.Web;
using System.Web.Mvc;

public class ExampleController: Controller
{
    [HttpGet]
    public IActionResult GetUser(string id)
    {
        string url = "http://example.com/api/user/" + id;
        HttpWebRequest request = (HttpWebRequest)WebRequest.Create(url); // Noncompliant

        return Ok();
    }
}
----

==== Compliant solution

[source,csharp,diff-id=1,diff-type=compliant]
----
using System.Web;
using System.Web.Mvc;

public class ExampleController: Controller
{
    [HttpGet]
    public IActionResult GetUser(string id)
    {
        string url = "http://example.com/api/user/" + HttpUtility.UrlEncode(id);
        HttpWebRequest request = (HttpWebRequest)WebRequest.Create(url);

        return Ok();
    }
}
----

=== How does this work?

include::../../common/fix/encoding.adoc[]
