:identifier_capital_plural: Local variables and parameters
:identifier: local variable and parameter
:identifier_plural: local variables and parameters
:identifier_or: local variable or parameter
:regex: @[a-zA-Z0-9_]*

include::../rule.adoc[]

=== Code examples

==== Noncompliant code example

With the default regular expression ``{regex}``:

[source,sql,diff-id=1,diff-type=noncompliant]
----
CREATE PROCEDURE proc1
    @@var1 INT -- Noncompliant
AS
BEGIN
    DECLARE @var2@ INT; -- Noncompliant
END
----

==== Compliant solution

[source,sql,diff-id=1,diff-type=compliant]
----
CREATE PROCEDURE proc1
    @var1 INT
AS
BEGIN
    DECLARE @var2 INT;
END
----

== Resources

=== Documentation

* Wikipedia - https://en.wikipedia.org/wiki/Naming_convention_(programming)[Naming Convention (programming)]

=== Related rules

* S1542 - Function and procedure names should comply with a naming convention

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../parameters.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
