== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

With the default regular expression ``++@[a-zA-Z0-9_]*++``:

[source,sql]
----
CREATE PROCEDURE proc1
    @@var1 INT -- Noncompliant
AS
BEGIN
    DECLARE @@var2 INT; -- Noncompliant
END
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::parameters.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
