<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Naming the parameters in a function prototype helps identify how they&#8217;ll be used by the function, thereby acting as a thin layer of documentation for the function.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">void divide (int, int);</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">void divide (int numerator, int denominator);</code></pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p>MISRA C:2004, 16.3 - Identifiers shall be given for all of the parameters in a function prototype declaration</p>
</li>
<li>
<p>MISRA C:2012, 8.2 - Function types shall be in prototype form with named parameters</p>
</li>
</ul>
</div>
<hr>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Give names to the parameters of this function prototype.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_relates_to_s819">relates to: <a data-rspec-id="S819" class="rspec-auto-link">S819</a></h3>

</div>
<div class="sect2">
<h3 id="_relates_to_s927">relates to: <a data-rspec-id="S927" class="rspec-auto-link">S927</a></h3>

</div>
<div class="sect2">
<h3 id="_is_related_to_s829">is related to: <a data-rspec-id="S829" class="rspec-auto-link">S829</a></h3>

</div>
</div>
</div>