This rule raises an issue when `require` statements are placed inside methods, classes, or other code blocks instead of at the top level of the file.

== Why is this an issue?

When `require` statements are scattered throughout your code, it becomes difficult to understand what dependencies a file needs at a glance. This creates several problems:

First, it hurts readability. Other developers (including your future self) have to scan through the entire file to understand what libraries and modules are being used. This makes code review and maintenance more time-consuming.

Second, it can lead to performance issues. Ruby's `require` method loads and executes files, which takes time. When requires are placed inside methods that might be called multiple times, you risk loading the same library repeatedly, even though Ruby's require mechanism prevents duplicate loading.

Third, it makes dependency management harder. Build tools, bundlers, and static analysis tools often look for require statements at the top of files to understand project dependencies. Scattered requires can be missed by these tools.

Finally, it goes against Ruby community conventions. The Ruby style guide and most Ruby projects follow the practice of placing all requires at the top of files, making your code inconsistent with established patterns.

=== What is the potential impact?

Scattered require statements reduce code maintainability and readability. They make it harder for developers to understand file dependencies, can impact performance when requires are placed in frequently-called methods, and may cause issues with dependency analysis tools.

== How to fix it

Move all require statements to the top of the file, before any class or module definitions. Group them logically and sort them alphabetically for better organization.

=== Code examples

==== Noncompliant code example

[source,ruby,diff-id=1,diff-type=noncompliant]
----
class ApiClient
  def initialize
    # Class implementation
  end
  
  def fetch_data
    require 'net/http'  # Noncompliant
    response = Net::HTTP.get(URI('https://api.example.com/data'))
    
    require 'json'  # Noncompliant
    JSON.parse(response)
  end
  
  def process_ssl
    require 'openssl'  # Noncompliant
    # SSL processing logic
  end
end
----

==== Compliant solution

[source,ruby,diff-id=1,diff-type=compliant]
----
# All requires at the top of the file
require 'json'
require 'net/http'
require 'openssl'

class ApiClient
  def initialize
    # Class implementation
  end
  
  def fetch_data
    response = Net::HTTP.get(URI('https://api.example.com/data'))
    JSON.parse(response)
  end
  
  def process_ssl
    # SSL processing logic
  end
end
----

== Resources

=== Documentation

 * Ruby Style Guide - Requires - https://rubystyle.guide/#consistent-multi-line-chains[Official Ruby style guide recommendations for organizing require statements]

 * Ruby Documentation - Kernel#require - https://ruby-doc.org/core/Kernel.html#method-i-require[Official documentation for Ruby's require method]
