<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>The <code>strerror</code> function returns a pointer to a buffer that is only valid until the function is called again, including from another thread. Which means that in practice, for any multithread program, it&#8217;s not possible to use it properly.</p>
</div>
<div class="paragraph">
<p>One safe alternative is <code>strerror_s</code>, provided in annex K of C11. To have access to it, you need a standard library that supports it (this can be tested with the macro <code>__STDC_LIB_EXT1__</code>), and you need to enable it by defining the macro <code>__STDC_WANT_LIB_EXT1__</code> before including <code>&lt;string.h&gt;</code>. <code>strerror_s</code> takes as an argument a buffer that will store the error message. Iworks together with the <code>strerrorlen_s</code> function, which can tell you the required buffer size to store the error.</p>
</div>
<div class="paragraph">
<p>Some environment also provide the <code>strerror_r</code> function, which works in a way similar to <code>strerror_s</code>, except there is now function that can provide you with the needed buffer size (but the return value will tell you if the buffer was large enough): Either you accept to have a truncated message if the message is too long, or you should call this function in a loop with increasing buffer size until it succeeds.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">int f(char *path) {
  FILE * fp = fopen(path,"r");
  if (fp == NULL) {
    // errno itself has thread storage duration
    char *errorMsg = strerror(errno); // Noncompliant, might be changed by another thread
    printf("Error: %s\n", errorMsg);
  }
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">int f(char *path) {
  FILE * fp = fopen(path,"r");
  if (fp == NULL) {
    // errno itself has thread storage duration
    int fileError = errno;
    size_t errorLen = strerrorlen_s(fileError) +1; // For the final null character
    char *errorMsg = malloc(errorLen);
    strerror_s(errorMsg, errorLen, fileError)
    printf("Error: %s\n", errorMsg);
    free(errorMsg);
  }
}</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Replace "strerror" by a thread-safe alternative</p>
</div>
</div>
</div>
</div>