Because library names could change from environment to environment, they should never be hard-coded in a program. Instead, you should use a variable to specify the library name and set that variable with data external to the program.


== Noncompliant Code Example

----
CLRPFM   FILE(XPJLIB/FILE01)
----


== Compliant Solution

----
CHGVAR     VAR(&LIB01) VALUE(&PREFIX *CAT 'LIB')
CLRPFM      FILE(&LIB01/FILE01)
----


== Exceptions

Hard-coding references to QTEMP, which is a temporary library, is allowed.

