== Why is this an issue?

Because library names could change from environment to environment, they should never be hard-coded in a program. Instead, you should use a variable to specify the library name and set that variable with data external to the program.


=== Noncompliant code example

[source,rpg]
----
CLRPFM   FILE(XPJLIB/FILE01)
----


=== Compliant solution

[source,rpg]
----
CHGVAR     VAR(&LIB01) VALUE(&PREFIX *CAT 'LIB')
CLRPFM      FILE(&LIB01/FILE01)
----


=== Exceptions

Hard-coding references to QTEMP, which is a temporary library, is allowed.


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
