<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Conditional compilation is generally recognized as a bad practice that is occasionally necessary when dealing with platform-specific code. As much as possible, code should be refactored to minimize or eliminate conditionally-compiled, platform-specific code because even when necessary and well-intentioned, such code segments can leave your codebase in a hopeless tangle.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-swift" data-lang="swift">#if os(OSX) // Noncompliant
    let a = 2
#else
    let a = 3
#endif</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Refactor the code to not need these conditional compilation directives.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_23_jun_2015_075951_elena_vilchik_wrote">on 23 Jun 2015, 07:59:51 Elena Vilchik wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] Since Swift 2.0 (which is not actually released) there is feature "availability checking".</p>
</div>
<div class="paragraph">
<p><a href="http://www.hackingwithswift.com/new-syntax-swift-2-availability-checking" class="bare">http://www.hackingwithswift.com/new-syntax-swift-2-availability-checking</a></p>
</div>
<div class="listingblock">
<div class="content">
<pre>if #available(iOS 9, *) {
    // use UIStackView
} else {
    // show sad face emoji
}</pre>
</div>
</div>
<div class="paragraph">
<p>WDYT May be it&#8217;s worth mentioning in this rule?  (smth like "But to able to run some pieces of code only on specific OS releases you can use availability checking (since Swift 2.0)")</p>
</div>
<div class="paragraph">
<p>UPDATE I&#8217;ve attached screenshot from documentation (it&#8217;s from book and it doesn&#8217;t allow me copy content)</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_23_jun_2015_142221_ann_campbell_wrote">on 23 Jun 2015, 14:22:21 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>I like it [~elena.vilchik] but I&#8217;m wondering about timing. Are you not likely to implement this rule until after Swift 2.0 is released?</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_24_jun_2015_080459_elena_vilchik_wrote">on 24 Jun 2015, 08:04:59 Elena Vilchik wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] ok, let&#8217;s keep it for Swift 2.0 release (ticket created SWIFT-162)</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_10_dec_2015_150200_ann_campbell_wrote">on 10 Dec 2015, 15:02:00 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>expanded per your request [~elena.vilchik]</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_10_dec_2015_151510_elena_vilchik_wrote">on 10 Dec 2015, 15:15:10 Elena Vilchik wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] Actually availability-checking should be compliant :) This piece should help people facing this rule to fix the issue by replacing preprocessor directives by <code>if #available ...</code>.</p>
</div>
</div>
</div>
</div>