<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p><code>getClass</code> should not be used for synchronization in non-<code>final</code> classes because child classes will synchronize on a different object than the parent or each other, allowing multiple threads into the code block at once, despite the <code>synchronized</code> keyword.</p>
</div>
<div class="paragraph">
<p>Instead, hard code the name of the class on which to synchronize or make the class <code>final</code>.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">public class MyClass {
  public void doSomethingSynchronized(){
    synchronized (this.getClass()) {  // Noncompliant
      // ...
    }
  }</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">public class MyClass {
  public void doSomethingSynchronized(){
    synchronized (MyClass.class) {
      // ...
    }
  }</code></pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p><a href="https://wiki.sei.cmu.edu/confluence/x/qTdGBQ">CERT, LCK02-J.</a> - Do not synchronize on the class object returned by getClass()</p>
</li>
</ul>
</div>
<hr>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Synchronize on the static class name instead.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_15_jun_2015_201520_nicolas_peru_wrote">on 15 Jun 2015, 20:15:20 Nicolas Peru wrote:</h3>
<div class="paragraph">
<p>Looks good.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_17_aug_2018_162858_amaury_levé_wrote">on 17 Aug 2018, 16:28:58 Amaury Levé wrote:</h3>
<div class="paragraph">
<p><a href="https://jira.sonarsource.com/browse/RSPEC-2551">RSPEC-2551</a> implements the same behavior for C#.</p>
</div>
</div>
</div>
</div>