<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>If a SQL <code>TABLE</code> is declared but not used in the program, it can be considered dead code and should therefore be removed. This will improve maintainability because developers will not wonder what the variable is used for.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cobol" data-lang="cobol">EXEC SQL
 DECLARE DSN8B10.DEPT TABLE  -- Noncompliant
   ( ...  )
END-EXEC.</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove this table "XXX" declaration; it is never used.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p><code>DECLARE ... TABLE</code></p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_6_oct_2016_114258_pierre_yves_nicolas_wrote">on 6 Oct 2016, 11:42:58 Pierre-Yves Nicolas wrote:</h3>
<div class="paragraph">
<p>IBM reference for DECLARE&#8230;&#8203;TABLE: http://www.ibm.com/support/knowledgecenter/SSEPEK_11.0.0/sqlref/src/tpc/db2z_sql_declaretable.html</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_29_nov_2016_114213_pierre_yves_nicolas_wrote">on 29 Nov 2016, 11:42:13 Pierre-Yves Nicolas wrote:</h3>
<div class="paragraph">
<p>We should not raise an issue when a <code>DECLARE...TABLE</code> is inside a copybook: a COBOL program, which could be in another project, could use this copybook and this <code>DECLARE...TABLE</code> declaration.</p>
</div>
</div>
</div>
</div>