== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,apex]
----
if (booleanMethod() || false) { /* ... */ }
doSomething(!false);
doSomething(booleanMethod() && true);
----

=== Compliant solution

[source,apex]
----
if (booleanMethod()) { /* ... */ }
doSomething(true);
doSomething(booleanMethod());
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
