:true: true
:false: false
:ops: !, &&, ||, ==, ===, !=, !==
== Why is this an issue?

include::../description.adoc[]

== How to fix it

include::../how-to-fix-it.adoc[]

=== Code examples

==== Noncompliant code example

[source,apex,diff-id=1,diff-type=noncompliant]
----
if (booleanMethod() || false) { /* ... */ }
doSomething(!false);
doSomething(booleanMethod() && true);
----

==== Compliant solution

[source,apex,diff-id=1,diff-type=compliant]
----
if (booleanMethod()) { /* ... */ }
doSomething(true);
doSomething(booleanMethod());
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''

endif::env-github,rspecator-view[]
