This rule raises an issue when a method name starts with "is_" and ends with "?".

== Why is this an issue?

In Ruby, predicate methods (methods that return boolean values) should end with a question mark to indicate their boolean nature. Using both the "is_" prefix and the "?" suffix creates redundancy that violates Ruby naming conventions.

The question mark suffix already clearly communicates that the method returns a boolean value. Adding "is_" makes the method name unnecessarily verbose and goes against established Ruby idioms. This redundancy can make code less readable and harder to maintain.

Ruby's naming conventions emphasize clarity and conciseness. Method names like `empty?`, `valid?`, and `present?` are more idiomatic than `is_empty?`, `is_valid?`, and `is_present?`.

=== What is the potential impact?

Using redundant "is_" prefixes in predicate methods reduces code readability and violates Ruby naming conventions. While this doesn't cause runtime errors, it makes the codebase less idiomatic and can confuse developers who expect standard Ruby naming patterns.

This pattern can also make method names unnecessarily long, reducing code clarity and making it harder to follow established Ruby style guides.

== How to fix it in Ruby

Remove the "is_" prefix from predicate methods. The question mark suffix already indicates that the method returns a boolean value.

=== Code examples

==== Noncompliant code example

[source,ruby,diff-id=1,diff-type=noncompliant]
----
def is_empty?
  @items.empty?
end
----

==== Compliant solution

[source,ruby,diff-id=1,diff-type=compliant]
----
def empty?
  @items.empty?
end
----

== How to fix it in Rails

This also applies to field declarations in frameworks like Avo where boolean fields should use the question mark suffix without the "is_" prefix.

=== Code examples

==== Noncompliant code example

[source,ruby,diff-id=2,diff-type=noncompliant]
----
field :is_admin, as: :boolean
----

==== Compliant solution

[source,ruby,diff-id=2,diff-type=compliant]
----
field :admin?, as: :boolean
----

== Resources

=== Documentation

 * Ruby Style Guide - Naming - https://ruby-style-guide.shopify.dev/#naming[Shopify's Ruby Style Guide section on naming conventions, including predicate methods]

 * Ruby Method Naming Conventions - https://docs.ruby-lang.org/en/master/syntax/methods_rdoc.html#label-Method+Names[Official Ruby documentation on method naming conventions]
