== Why is this an issue?

The size of integer required to hold a memory address is implementation-dependent. Therefore, casting a pointer (i.e. a memory address) to any integral data type may result in data loss because the integral type is too small to hold the full address value.


When treating a memory address as integer type is absolutely required, you should be sure to use a large enough type to hold all the data.


=== Noncompliant code example

[source,cpp]
----
int *p;
int addr = ( int ) &p;
----


== Resources

* MISRA C:2004, 11.3 - A cast should not be performed between a pointer type and an integral type.
* MISRA {cpp}:2008, 5-2-9 - A cast should not convert a pointer type to an integral type.
* MISRA C:2012, 11.4 - A conversion should not be performed between a pointer to object and an integer type
* https://wiki.sei.cmu.edu/confluence/x/0dUxBQ[CERT, INT36-C.] - Converting a pointer to integer or integer to pointer


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

An integral type is too small to hold a pointer value.


'''
== Comments And Links
(visible only on this page)

=== is duplicated by: S857

=== is duplicated by: S2192

=== relates to: S2192

=== is related to: S3630

=== on 16 Jun 2014, 18:02:21 Evgeny Mandrikov wrote:
\[~ann.campbell.2] how this relates to RSPEC-857 ?

=== on 16 Jun 2014, 18:37:40 Ann Campbell wrote:
\[~evgeny.mandrikov] are you really asking how, or just pointing out that there is a relationship?  


No idea why I created a new rspec ticket when one already existed. Presumably I simply didn't notice the first one. But since this newer one is fully-fleshed think we should close the other one. Objections?

=== on 16 Jun 2014, 18:41:49 Evgeny Mandrikov wrote:
\[~ann.campbell.2] I'm just pointing out that most probably one should be closed or removed, if you agree.

=== on 28 Sep 2014, 21:19:32 Evgeny Mandrikov wrote:
Provided solution is not compliant, because "long" might be not large enough to hold pointer at least because there is "long long" type. See description in MISRA C:2012 11.4, which shows proper solution for C99 - usage of type "intptr_t", and thus there is also proper solution for {cpp} - usage of "std::intptr_t".


All MISRA descriptions are about "integral types", which is not the same as "int" used in both title and description - quoting CPP11 (ISO/IEC 14882:2011) 3.9.1p7:

____
Types bool, char, char16_t, char32_t, wchar_t, and the signed and unsigned integer types are collectively called integral types.

____


=== on 2 Oct 2014, 20:44:54 Ann Campbell wrote:
Actually [~evgeny.mandrikov] the descriptions all talk about "integer type", but point taken & corrections made.

=== on 2 Oct 2014, 21:07:15 Evgeny Mandrikov wrote:
\[~ann.campbell.2] quoting the same paragraph from the same spec, but few words further:

____
A synonym for integral type is integer type.

____

endif::env-github,rspecator-view[]
