<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when a method is defined with a reserved keyword name surrounded by quotes.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>While Groovy allows defining methods with reserved keyword names by surrounding them with quotes, this practice creates several problems:</p>
</div>
<div class="ulist">
<ul>
<li>
<p><strong>Reduced readability</strong>: Quoted method names look unusual and can confuse developers who are not familiar with this Groovy feature.</p>
</li>
<li>
<p><strong>Special calling syntax</strong>: Methods with quoted keyword names require qualified calls using the <code>this.</code> prefix, making the calling code more verbose and inconsistent.</p>
</li>
<li>
<p><strong>Maintenance burden</strong>: Code becomes harder to understand and maintain when method names don&#8217;t clearly express their purpose.</p>
</li>
<li>
<p><strong>Inconsistent style</strong>: Mixing quoted and unquoted method names creates inconsistent code style within a project.</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>The Groovy documentation itself states that "using such names might be confusing and is often best to avoid." This feature was primarily intended for specific Java integration scenarios and Domain Specific Language (DSL) use cases, not for general application development.</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="paragraph">
<p>Using reserved keywords as method names can lead to:</p>
</div>
<div class="ulist">
<ul>
<li>
<p><strong>Confusion for team members</strong>: Developers may not understand why certain methods require special calling syntax</p>
</li>
<li>
<p><strong>Increased cognitive load</strong>: Reading and understanding code becomes more difficult</p>
</li>
<li>
<p><strong>Maintenance issues</strong>: Future developers may struggle to work with such code</p>
</li>
<li>
<p><strong>Inconsistent codebase</strong>: Mixed naming conventions reduce overall code quality</p>
</li>
</ul>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Replace the quoted reserved keyword with a descriptive method name that clearly indicates the method&#8217;s purpose.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-groovy" data-lang="groovy">def "abstract"() { // Noncompliant
    return true
}

this.abstract()</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-groovy" data-lang="groovy">def isAbstract() {
    return true
}

this.isAbstract()</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Groovy documentation - <a href="https://groovy-lang.org/syntax.html#_keywords">Keywords</a></p>
</li>
<li>
<p>Groovy documentation - <a href="https://groovy-lang.org/syntax.html#_identifiers">Identifiers</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>