== Why is this an issue?

Some types cannot be constrained, and attempting to do so results in the exception ``++PLS-00566: type name "..." cannot be constrained++`` being raised.


=== Noncompliant code example

[source,sql]
----
DECLARE
  foo BLOB(42); -- Noncompliant - raises PLS-00566: type name "BLOB" cannot be constrained
BEGIN
  NULL;
END;
/
----


=== Compliant solution

[source,sql]
----
DECLARE
  foo BLOB;
BEGIN
  NULL;
END;
/
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

endif::env-github,rspecator-view[]
