== Why is this an issue?

ABAP provides the ability to manipulate programs dynamically during execution for instance with statements like ``++INSERT REPORT++`` and ``++GENERATE SUBROUTINE POOL++``. Most of those statements are for internal use within SAP Technology Development and incompatible changes are possible at any time without prior warning or notice. 


This rule raises an issue when any of the following source code processing statements is used: ``++INSERT REPORT++``,  ``++READ REPORT++``, ``++DELETE REPORT++``, ``++EDITOR-CALL FOR REPORT++``, ``++SYNTAX-CHECK FOR itab++``, ``++GENERATE REPORT/SUBROUTINE POOL++``, ``++LOAD REPORT++``, ``++SCAN++``, ``++INSERT TEXTPOOL++``, ``++READ TEXTPOOL++``, ``++DELETE TEXTPOOL++``, ``++EXPORT DYNPRO++``, ``++IMPORT DYNPRO++``, ``++DELETE DYNPRO++``, ``++SYNTAX-CHECK FOR DYNPRO++``, and ``++GENERATE DYNPRO++``.


=== Noncompliant code example

[source,abap]
----
GENERATE REPORT MY_PROG.
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
