include::../description.adoc[]

include::../ask-yourself.adoc[]

include::../recommended.adoc[]

== Noncompliant Code Example

AWS DMS and EC2 instances that have a public IP address assigned to them:

----
resource "aws_instance" "example" {
  associate_public_ip_address = true # Sensitive, by default it's also set to true
}

resource "aws_dms_replication_instance" "example" {
  publicly_accessible          = true # Sensitive, by default it's also set to true
}
----

GCP Compute instance that has a public IP address assigned to it:

----
resource "google_compute_instance" "example" {
  network_interface {
    network = "default"

    access_config {  // Sensitive
      // Ephemeral public IP
    }
  }
}
----

== Compliant Solution

AWS DMS and EC2 instances that do not have a public IP address:

----
resource "aws_instance" "example" {
  associate_public_ip_address = false
}

resource "aws_dms_replication_instance" "example" {
  publicly_accessible          = false
}
----

GCP Compute instance that does not have a public IP address:

----
resource "google_compute_instance" "example" {
  network_interface {
    network = "default"
  }
}
----

include::../see.adoc[]
