<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Custom tasks and plugins should be moved from the build script to the <code>buildSrc</code> directory
to encapsulate custom build logic and separate it against its application in the build script.</p>
</div>
<div class="paragraph">
<p>This helps to keep the build script clean and easy to understand.</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="sect3">
<h4 id="_readability_and_understanding">Readability and Understanding</h4>
<div class="paragraph">
<p>This change makes the build script easier to read and understand due to encapsulation and separation of concerns.</p>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Move custom tasks, plugins, and configuration extensions from your <code>./build.gradle.kts</code> file
to source files in <code>./buildSrc/src/main/kotlin</code>.
You should apply package names and folders like for any other source code.
Please see the attached links for details about how to write custom plugins and the proper use of the <code>./buildSrc</code> directory.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="paragraph">
<p><strong>./build.gradle.kts</strong></p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-kotlin" data-lang="kotlin">open class MyCustomTask : DefaultTask() {
    // ...
}

tasks.register&lt;MyCustomTask&gt;("myCustom")

open class MyCustomExtension {
    var flag: Boolean = false
}

extensions.create&lt;MyCustomExtension&gt;("myCustomExtension")

val myCustomExtension: MyCustomExtension by project

myCustomExtension.flag = true</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="paragraph">
<p><strong>./buildSrc/src/main/kotlin/mypackage/MyCustomTask.kt</strong></p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-kotlin" data-lang="kotlin">package mypackage

import org.gradle.api.DefaultTask

open class MyCustomTask : DefaultTask() {
    // ...
}</code></pre>
</div>
</div>
<div class="paragraph">
<p><strong>./buildSrc/src/main/kotlin/mypackage/MyCustomExtension.kt</strong></p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-kotlin" data-lang="kotlin">package mypackage

open class MyCustomExtension {
    var flag: Boolean = false
}</code></pre>
</div>
</div>
<div class="paragraph">
<p><strong>./build.gradle.kts</strong></p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-kotlin" data-lang="kotlin">import mypackage.MyCustomExtension
import mypackage.MyCustomTask

tasks.register&lt;MyCustomTask&gt;("myCustom")
extensions.create&lt;MyCustomExtension&gt;("myCustomExtension")

val myCustomExtension: MyCustomExtension by project
myCustomExtension.flag = true</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p><a href="https://docs.gradle.org/current/userguide/organizing_gradle_projects.html#sec:build_sources">Gradle Documentation - Use "buildSrc" to abstract imperative logic</a></p>
</li>
<li>
<p><a href="https://docs.gradle.org/current/userguide/custom_tasks.html#custom_tasks">Gradle Documentation - Developing Custom Gradle Task Types</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>