== Why is this an issue?

When using some transaction managers like IBM IMS, each COBOL program is in fact considered a sub-program by the transaction manager. The ``++GOBACK++`` statement returns control to the transaction manager, but using ``++STOP RUN++`` might cause unpredictable results or abnormal termination.


=== Noncompliant code example

[source,cobol]
----
STOP RUN
----


=== Compliant solution

[source,cobol]
----
GOBACK
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

endif::env-github,rspecator-view[]
