<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Functions that accept random numbers of arguments use a rest argument (<code>... argname</code>). This allows you to pass the function each relevant parameter individually. At runtime, those parameters are automatically wrapped in an array. If you pass an array to such a function, it will automatically be re-wrapped in another array, and this double-layering won&#8217;t be expected by the called function. To avoid that double-layering, use the spread operator (<code>...</code>arrayToBeExpanded) in the call to expand the array.</p>
</div>
<div class="paragraph">
<p>This applies both to manually created arrays, and to arguments that were accepted as rest parameters.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-javascript" data-lang="javascript">collect(new Book(), new Book());

function collect(...books: Book[]) {
  buy(books);   // Noncompliant
}

function buy(...things: any[]) {
  console.log(things); // outputs "[ [ Book {}, Book {} ] ]"
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-javascript" data-lang="javascript">collect(new Book(), new Book());

function collect(...books: Book[]) {
  buy(...books);
}

function buy(...things: any[]) {
  console.log(things); // outputs "[ Book {}, Book {} ]"
}</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Use spread operator '&#8230;&#8203;' to pass this argument</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_3_jan_2018_203724_ann_campbell_wrote">on 3 Jan 2018, 20:37:24 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~elena.vilchik] a little text on what "spread" is and why it&#8217;s necessary (when the argument you&#8217;re passing is apparently already an array) would be nice.</p>
</div>
</div>
</div>
</div>