Shared naming conventions allow teams to collaborate efficiently. This rule raises an issue when a test method name does not match the provided regular expression.


== Noncompliant Code Example

With the default value: ``++^test[A-Z][a-zA-Z0-9]*$++``

[source,java]
----
@Test
public void foo() {  // Noncompliant
  //...
}
----


== Compliant Solution

[source,java]
----
@Test
public void testFoo() {
  // ...
}
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

include::parameters.adoc[]

endif::env-github,rspecator-view[]
