== Why is this an issue?

Shared naming conventions allow teams to collaborate efficiently. This rule raises an issue when a test method name does not match the provided regular expression.


=== Noncompliant code example

With the default value: ``++^test[A-Z][a-zA-Z0-9]*$++``

[source,java]
----
@Test
public void foo() {  // Noncompliant
  //...
}
----


=== Compliant solution

[source,java]
----
@Test
public void testFoo() {
  // ...
}
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Rename this method to match the regular expression: yyyy


=== Parameters

.format
****

----
^test[A-Z][a-zA-Z0-9]*$
----

Regular expression the test method names are checked against.
****


endif::env-github,rspecator-view[]
