== Why is this an issue?

This rule allows you to track the use of the PMD suppression comment mechanism. 


=== Noncompliant code example

[source,java]
----
// NOPMD
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Remove this "NOPMD" suppression comment.


'''
== Comments And Links
(visible only on this page)

=== relates to: S1315

=== on 9 Oct 2013, 13:15:50 Freddy Mallet wrote:
Is implemented by \http://jira.codehaus.org/browse/SONARJAVA-340

endif::env-github,rspecator-view[]
