== Why is this an issue?

This rule allows you to track the use of the PMD suppression comment mechanism. 


=== Noncompliant code example

[source,java]
----
// NOPMD
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
