Defining variables in the minimum block scope possible reduces the visibility of those variables and therefore reduces the possibility that these identifiers will be used accidentally. A corollary of this is that global objects (including singleton function objects) shall be used in more than one function.

== Noncompliant Code Example

----
void f ( int32_t k )
{
  int32_t i = k * k; // Noncompliant, visibility could be reduced
  {
    std::cout << i << std::endl;
  }
}
----

== Compliant Solution

----
void f ( int32_t k )
{
  {
    int32_t i = k * k; // Compliant
    std::cout << i << std::endl;
  }
}
----

== See

* MISRA {cpp}:2008, 3-4-1
