<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Externally described files standardize file access, and result in simpler, easier to read code.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-rpg" data-lang="rpg">     FEPBLMSL2  IF   F  100        DISK    INFSR(*PSSR)
     F*
     F*</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-rpg" data-lang="rpg">     F*
     FIPBDCCP   IF   E           K DISK    INFSR(*PSSR)
     F*

     FEP210     CF   E             WORKSTN INFSR(*PSSR)
     F                                     SFILE(EP21003S:RELN03)


     FEP471R1   O    E             PRINTER INFSR(*PSSR)
     F                                     USROPN
     F                                     INFDS(W1SF01)</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Use an "E" indicator for this file and remove any related "I" lines.</p>
</div>
</div>
</div>
</div>