<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>The rule <a data-rspec-id="S1279" class="rspec-auto-link">S1279</a> ("DISPLAY" should not be used) is fully preventing the usage of the <code>DISPLAY</code> statement whereas in same cases it may be useful to output some data to SYSOUT. However, the output of a call to <code>DISPLAY UPON CONSOLE</code> is unlikely to be viewed by someone because the output is displayed on the console and nowadays no one is in front of the console to see if something is happening.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cobol" data-lang="cobol">PROCEDURE DIVISION.
   ...
   DISPLAY "Error Occurred!!!" UPON CONSOLE *&gt; Noncompliant
   ...</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_duplicates_s1279">duplicates: <a data-rspec-id="S1279" class="rspec-auto-link">S1279</a></h3>

</div>
<div class="sect2">
<h3 id="_on_27_sep_2018_152556_pierre_yves_nicolas_wrote">on 27 Sep 2018, 15:25:56 Pierre-Yves Nicolas wrote:</h3>
<div class="paragraph">
<p><a data-rspec-id="S1279" class="rspec-auto-link">RSPEC-1279</a> has a parameter named "checkOnlyDisplayUponConsole".</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_23_nov_2018_135723_alexandre_gigleux_wrote">on 23 Nov 2018, 13:57:23 Alexandre Gigleux wrote:</h3>
<div class="paragraph">
<p>This is a duplicate of <a data-rspec-id="S1279" class="rspec-auto-link">RSPEC-1279</a>. By default <a data-rspec-id="S1279" class="rspec-auto-link">RSPEC-1279</a> is containing a parameter <code>checkOnlyDisplayUponConsole</code> that is ON and prevent the use of "DISPLAY UPON CONSOLE".</p>
</div>
</div>
</div>
</div>