This rule raises an issue when `pluck(:id)` is used on Active Record collections to extract primary key IDs.

== Why is this an issue?

Active Record provides a dedicated `ids` method specifically for extracting primary key IDs from collections. Using `pluck(:id)` is less semantic and doesn't clearly express the intent to retrieve primary key values.

The `ids` method is more expressive and immediately communicates that you're working with primary keys. It follows Rails conventions and makes the code more readable for other developers who are familiar with Rails patterns.

While both methods produce the same result, using the purpose-built `ids` method aligns with Rails best practices and improves code clarity.

=== What is the potential impact?

Using `pluck(:id)` instead of `ids` doesn't cause functional issues, but it makes the code less expressive and harder to understand. Following Rails conventions improves code maintainability and helps other developers quickly understand the intent.

== How to fix it

Replace `pluck(:id)` with the more semantic `ids` method when extracting primary key IDs from Active Record collections.

=== Code examples

==== Noncompliant code example

[source,ruby,diff-id=1,diff-type=noncompliant]
----
User.pluck(:id) # Noncompliant
User.where(active: true).pluck(:id) # Noncompliant
post.comments.pluck(:id) # Noncompliant
----

==== Compliant solution

[source,ruby,diff-id=1,diff-type=compliant]
----
User.ids
User.where(active: true).ids
post.comments.ids
----

== Resources

=== Documentation

 * Rails Style Guide - ids - https://rails.rubystyle.guide/#ids[Official Rails Style Guide recommendation for using ids over pluck(:id)]

 * Active Record Query Interface - https://guides.rubyonrails.org/active_record_querying.html#ids[Rails documentation on the ids method for Active Record queries]
