include::../description.adoc[]

== Noncompliant Code Example

----
public static void doSomething() {
	for(int i = 0; i < 4; i++){                 // Noncompliant, 4 is a magic number
		...
	}
}
----

== Compliant Solution

----
public static final int NUMBER_OF_CYCLES = 4;
public static void doSomething() {
  for(int i = 0; i < NUMBER_OF_CYCLES ; i++){
    ...
  }
}
----

== Exceptions

This rule ignores <code>hashCode</code> methods.
