include::../description.adoc[]

=== Exceptions

This rule ignores ``++hashCode++`` methods.

include::../how-to-fix-it.adoc[]

=== Code examples

==== Noncompliant code example

[source,java,diff-id=1,diff-type=noncompliant]
----
public static void doSomething() {
  for (int i = 0; i < 4; i++) {  // Noncompliant, 4 is a magic number
    ...
  }
}
----

==== Compliant solution

[source,java,diff-id=1,diff-type=compliant]
----
public static final int NUMBER_OF_CYCLES = 4;
public static void doSomething() {
  for (int i = 0; i < NUMBER_OF_CYCLES ; i++) { // Compliant
    ...
  }
}
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
