<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when a synchronous <code>time.sleep()</code> call is used within an asynchronous function.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Synchronous functions like <code>time.sleep()</code> halt the execution of the current thread. In an asynchronous context, which relies on a single-threaded event loop to manage multiple tasks concurrently, calling <code>time.sleep()</code> blocks this entire event loop.
When the event loop is blocked, it cannot switch between tasks, process I/O events, or respond to other operations. Consequently, all other concurrent asynchronous tasks are paused until the <code>time.sleep()</code> call completes. This effectively negates the benefits of using <code>async/await</code> by turning a non-blocking operation (waiting) into a blocking one that freezes the application&#8217;s concurrency model.</p>
</div>
<div class="paragraph">
<p>For instance, if an asynchronous web server uses <code>time.sleep()</code> in a request handler, it won&#8217;t be able to process any other incoming requests until the sleep call completes, leading to poor performance and responsiveness.</p>
</div>
<div class="paragraph">
<p>The correct approach in asynchronous programming is to use non-blocking sleep functions provided by the specific asynchronous framework being used (e.g., <code>asyncio</code>, <code>Trio</code>, <code>AnyIO</code>). These functions give control back to the event loop, allowing it to run other tasks while the current task is "sleeping".</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it_in_asyncio">How to fix it in Asyncio</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">import time
import asyncio

async def sleeping_function():
    time.sleep(1) # Noncompliant

asyncio.run(sleeping_function())</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">import asyncio

async def sleeping_function():
    await asyncio.sleep(1)

asyncio.run(sleeping_function())</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it_in_trio">How to fix it in Trio</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_code_examples_2">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example_2">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">import time
import trio

async def sleeping_function():
    time.sleep(1) # Noncompliant

trio.run(sleeping_function)</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution_2">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">import trio

async def sleeping_function():
    await trio.sleep(1)

trio.run(sleeping_function)</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it_in_anyio">How to fix it in AnyIO</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_code_examples_3">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example_3">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">import time
import anyio

async def sleeping_function():
    time.sleep(1) # Noncompliant

anyio.run(sleeping_function)</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution_3">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">import anyio

async def sleeping_function():
    await anyio.sleep(1)

anyio.run(sleeping_function)</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Asyncio documentation - <a href="https://docs.python.org/3/library/asyncio-task.html#asyncio.sleep">Sleeping</a></p>
</li>
<li>
<p>Trio documentation - <a href="https://trio.readthedocs.io/en/stable/reference-core.html#trio.sleep">Time and clocks</a></p>
</li>
<li>
<p>AnyIO documentation - <a href="https://anyio.readthedocs.io/en/stable/api.html#anyio.sleep">Event loop</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Replace this call to <code>time.sleep()</code> with the appropriate asynchronous sleep function from the framework you are using.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>The <code>time.sleep()</code> call in the function definition.
Secondary location on the async keyword of the function definition.</p>
</div>
</div>
<div class="sect2">
<h3 id="_quickfix">Quickfix</h3>
<div class="paragraph">
<p>No</p>
</div>
</div>
</div>
</div>