Jump statements (`+return+`) move control flow out of the current code block. So any statements that come after a jump are dead code. 

== Noncompliant Code Example

----
def foo(a: Int) {
  val i = 10;
  return a + i;       // Noncompliant 
  bar;                // dead code
}
----

== Compliant Solution

----
def foo(a: Int): Int {
  val i = 10;
  return a + i;
}
----

include::../see.adoc[]
