include::../summary.adoc[]

== Why is this an issue?

Jump statements (``++return++``) move control flow out of the current code block. So any statements that come after a jump are dead code. 

=== Noncompliant code example

[source,scala]
----
def foo(a: Int) {
  val i = 10;
  return a + i;       // Noncompliant 
  bar;                // dead code
}
----

=== Compliant solution

[source,scala]
----
def foo(a: Int): Int {
  val i = 10;
  return a + i;
}
----

include::../see.adoc[]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../highlighting.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
