== Why is this an issue?

Certain functions exist in PHP only as aliases of other functions. These aliases have been made available for backward compatibility, but should really be removed from code. 


This rule looks for uses of the following aliases:

[frame=all]
[cols="^1,^1"]
|===
|Alias|Replacement

|``++chop++``|``++rtrim++``
|``++close++``|``++closedir++``
|``++doubleval++``|``++floatval++``
|``++fputs++``|``++fwrite++``
|``++ini_alter++``|``++ini_set++``
|``++is_double++``|``++is_float++``
|``++is_integer++``|``++is_int++``
|``++is_long++``|``++is_int++``
|``++is_real++``|``++is_float++``
|``++is_writeable++``|``++is_writable++``
|``++join++``|``++implode++``
|``++key_exists++``|``++array_key_exists++``
|``++magic_quotes_runtime++``|``++set_magic_quotes_runtime++``
|``++pos++``|``++current++``
|``++show_source++``|``++highlight_file++``
|``++sizeof++``|``++count++``
|``++strchr++``|``++strstr++``
|===

=== Noncompliant code example

[source,php]
----
$arr=array("apple", "pear","banana");
echo sizeof($arr);  // Noncompliant
----


=== Compliant solution

[source,php]
----
$arr=array("apple", "pear","banana");
echo count($arr);
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Replace this use of "xxx" with "yyy".


'''
== Comments And Links
(visible only on this page)

=== on 2 Jun 2016, 16:40:44 Ann Campbell wrote:
http://php.net/manual/en/aliases.php

endif::env-github,rspecator-view[]
