== How to fix it in .NET

Microsoft's database connection libraries typically provide a connection string builder class. These classes provide a
safe way to add user-supplied values to a connection string.

If no connection string builder is available, user-supplied values must either be validated to ensure that they're not
malicious, or must be properly quoted so that they cannot interfere with other connection string parameters.

=== Code examples

==== Noncompliant code example

[source,csharp,diff-id=1,diff-type=noncompliant]
----
public string ConnectionString { get; set; } = "Server=10.0.0.101;Database=CustomerData";

public SqlConnection ConnectToDatabase(HttpRequest request)
{
    string connectionString = string.Format("{0};User ID={1};Password={2}"
        ConnectionString,
        request.Form["username"],
        request.Form["password"]);

    SqlConnection connection = new SqlConnection();
    connection.ConnectionString = connectionString;
    connection.Open();
    return connection;
}
----

==== Compliant solution

[source,csharp,diff-id=1,diff-type=compliant]
----
public string ConnectionString { get; set; } = "Server=10.0.0.101;Database=CustomerData";

public SqlConnection ConnectToDatabase(HttpRequest request)
{
    SqlConnectionStringBuilder builder = new SqlConnectionStringBuilder(ConnectionString);
    builder.UserID = request.Form["username"];
    builder.Password = request.Form["password"];

    SqlConnection connection = new SqlConnection();
    connection.ConnectionString = builder.ToString();
    connection.Open();
    return connection;
}
----

=== How does this work?

Connection string builders will ensure that all values are safely encoded or quoted when added to the connection string.
