:framework: NUnit
:classAnnotation: TestFixture

== How to fix it in NUnit

include::how.adoc[]

* `Test`
* `TestCase`
* `TestCaseSource`
* `Theory`

=== Code examples

==== Noncompliant code example

[source,csharp,diff-id=11,diff-type=noncompliant]
----
[TestFixture]
public class SomeClassTest { } // Noncompliant: no test
----

==== Compliant solution

[source,csharp,diff-id=11,diff-type=compliant]
----
[TestFixture]
public class SomeClassTest
{
    [Test]
    public void SomeMethodShouldReturnTrue() { }
}
----
