<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>C&#43;&#43;11 introduced "Attributes". They provide a unified syntax to specify additional information about your code.</p>
</div>
<div class="literalblock">
<div class="content">
<pre>They can be applied to various things like types, variables, functions, names, blocks, and translation units.</pre>
</div>
</div>
<div class="literalblock">
<div class="content">
<pre>{cpp} defines some standard attributes like \[[noreturn]], \[[nodiscard]], \[[deprecated]], \[[fallthrough]]...</pre>
</div>
</div>
<div class="paragraph">
<p>Unfortunately, it is possible to use unknown attributes: attributes that are not defined. Your code will compile, but the unknown attribute will be silently ignored. This means that your code will not behave in the way you expected.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">void test1(int p) {
  switch (p) {
    case 0:
    case 1:;
      [[std::fallthrough]]; // Noncompliant [[std::fallthrough]] isn't a defined attribute
    default:
      break;
  }
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">void test1(int p) {
  switch (p) {
    case 0:
    case 1:;
      [[fallthrough]];
    default:
      break;
  }
}</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_7_may_2020_091242_geoffray_adde_wrote">on 7 May 2020, 09:12:42 Geoffray Adde wrote:</h3>
<div class="paragraph">
<p>Whatre you intending to do for compiler-specific attributes? have a whitelist of namespaces?</p>
</div>
<div class="paragraph">
<p>for example:</p>
</div>
<div class="literalblock">
<div class="content">
<pre>https://docs.microsoft.com/en-us/cpp/cpp/attributes?view=vs-2019</pre>
</div>
</div>
</div>
</div>
</div>