This rule reports when a class depends on another class, and has been requested for removal.

== Why is this an issue?

This issue is being raised because a dependency in the code is not desirable since it constitutes an architecture problem.

=== What is the potential impact?

As these problems accumulate, the code becomes more complex and less consistent with the intent, making future development increasingly challenging.

== How to fix it

Remove or refactor the problem dependency.
This may involve replacing the dependency with an allowed one, moving code to appropriate classes or namespaces, or introducing new abstractions. The exact solution depends on your project's specific situation.

=== Code examples

==== Noncompliant code example

If the intended architecture prohibits classes in namespace `Example.Panels` to depend on classes in `Example.Repos`, then the following code example is noncompliant:

[source,java,diff-id=1,diff-type=noncompliant]
----
using Example.Repos;

namespace Example.Panels;

public class ShowCustomersPanel
{
    CustomerRepo customerRepo = ...;
}
----

==== Compliant solution

A solution could be to call into the service layer instead.

[source,java,diff-id=1,diff-type=compliant]
----
using Example.Services;

namespace Example.Panels;

public class ShowCustomersPanel
{
    CustomerService customerService = ...;
}
----
