<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Unlike similar AssertJ methods testing exceptions (<code>assertThatCode()</code>, <code>assertThatExceptionOfType()</code>, &#8230;&#8203;), the <code>assertThatThrownBy()</code> method can be used alone, failing if the code did not raise any exception.</p>
</div>
<div class="paragraph">
<p>Still, only testing that an exception was raised is not enough to guarantee that it was the expected one, and you should test the exception type or content further. In addition, it will make explicit what you are expecting, without relying on side-effects.</p>
</div>
<div class="paragraph">
<p>This rule raises an issue when <code>assertThatThrownBy</code> is used, without testing the exception further.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">assertThatThrownBy(() -&gt; shouldThrow()); // Noncompliant, is it really the exception you expected?</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">assertThatThrownBy(() -&gt; shouldThrow()).isInstanceOf(IOException.class);
//or
assertThatThrownBy(() -&gt; shouldThrow()).hasMessage("My exception");</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Test further the exception raised by this assertThatThrownBy call.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>Method name "assertThatThrownBy".</p>
</div>
</div>
</div>
</div>