This rule raises an issue when non-mutating sorting methods like `sort`, `sort_by`, or `reverse` are called without using their return values.

== Why is this an issue?

Ruby's non-mutating sorting methods (`sort`, `sort_by`, `reverse`) create and return a new sorted collection without modifying the original. When you call these methods without capturing or using their return values, the sorting operation has no effect on your program.

This is a common source of confusion for developers who expect these methods to work like their mutating counterparts (`sort!`, `sort_by!`, `reverse!`). The original collection remains unchanged, which can lead to incorrect program behavior.

For example, if you're trying to display a sorted list to users but ignore the return value of `sort_by`, users will see the data in its original, unsorted order. This can make your application appear broken or unprofessional.

The issue becomes more subtle when the sorting call is buried in larger code blocks, making it harder to spot that the intended sorting never actually happened.

=== What is the potential impact?

Ignoring the return value of non-mutating sort methods means your data remains unsorted, which can lead to:

* Incorrect display of information to users
* Failed assumptions in subsequent code that expects sorted data
* Difficult-to-debug issues where sorting appears to work in some contexts but not others
* Poor user experience when data appears in unexpected order

== How to fix it

Use the mutating version of the method (with `!`) to sort the collection in-place.

=== Code examples

==== Noncompliant code example

[source,ruby,diff-id=1,diff-type=noncompliant]
----
array = [{"value" => 3}, {"value" => 1}, {"value" => 2}]
array.sort_by { |item| item["value"] } # Noncompliant
puts array # Still prints original order
----

==== Compliant solution

[source,ruby,diff-id=1,diff-type=compliant]
----
array = [{"value" => 3}, {"value" => 1}, {"value" => 2}]
array.sort_by! { |item| item["value"] }
puts array # Now prints sorted order
----

== Resources

=== Documentation

 * Ruby Array#sort documentation - https://ruby-doc.org/core/Array.html#method-i-sort[Official documentation for Ruby's Array#sort method]

 * Ruby Array#sort_by documentation - https://ruby-doc.org/core/Array.html#method-i-sort_by[Official documentation for Ruby's Array#sort_by method]

 * Ruby Array#reverse documentation - https://ruby-doc.org/core/Array.html#method-i-reverse[Official documentation for Ruby's Array#reverse method]

=== Related rules

 * S2201 - https://rules.sonarsource.com/csharp/RSPEC-2201/[Return values from functions without side effects should not be ignored (C#)]
