<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This issue indicates that a cast operation will fail and throw a <code>ClassCastException</code>.
To fix it, make sure only references to objects with a compatible type can be cast.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>A cast operation allows an object to be "converted" from one type to another.
The compiler raises an error if it can determine that the target type is incompatible with the declared type of the object, otherwise it accepts the cast.
However, depending on the actual runtime type of the object, a cast operation may fail at runtime.
When a cast operation fails, a <code>ClassCastException</code> is thrown.</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="paragraph">
<p>This type of exception is usually unexpected.
It causes the program to crash or puts it into an inconsistent state.
Therefore, this issue might impact the availability and reliability of your application, or even result in data loss.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>When an object is cast, the code makes assumptions about the type of the object.
A <code>ClassCastException</code> indicates that this assumption has been broken.
If the assumption is reasonable, then some prior logic should ensure that only objects of a compatible type can be cast.
You should try to identify the code responsible for these checks and fix it.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">private String hexString(Object o) {
  return Integer.toHexString((Integer) o); // Noncompliant if hexString is called with a String for example
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="paragraph">
<p>One possible solution is to change <code>hexString</code> to only accept integers and adapt call sites.</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">private String hexString(Integer i) {
  return Integer.toHexString(i);
}</code></pre>
</div>
</div>
<div class="paragraph">
<p>Another one is to return a default value for types that are not <code>Integer</code>.
Here, the <code>if</code> statement with the condition relying on <code>instanceof</code> prevents references to objects with an incompatible type from making it to the cast operation.</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">private String hexString(Object o) {
  if (o instanceof Integer) {
    return Integer.toHexString((Integer) o);
  }
  return "0x0";
}</code></pre>
</div>
</div>
<div class="paragraph">
<p>The solution to adopt depends on which part of the code should be responsible to handle non-Integer types.</p>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_going_the_extra_mile">Going the extra mile</h3>
<div class="paragraph">
<p>Casting is considered an anti-pattern in object-oriented programming.
It is sometimes necessary, but there often is a better alternative.
Consider:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>Polymorphism</p>
</li>
<li>
<p>The visitor design pattern</p>
</li>
<li>
<p>Pattern matching</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>They come with some guarantees from the compiler, making your code more reliable.</p>
</div>
<div class="sect3">
<h4 id="_noncompliant_code_example_2">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">int foo(Shape shape) {
  if (shape instanceof Circle) {
    Circle circle = (Circle) shape;
    // Code for objects of type Circle
  } else if (shape instanceof Square) {
    Square square = (Square) shape;
    // Code for objects of type Square
  }
  // Default
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution_2">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">int foo(Shape shape) {
  return shape.fooValue(...); // Code was moved into subclasses
}</code></pre>
</div>
</div>
<div class="paragraph">
<p>or</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">int foo(Shape shape) {
  return shape.accept(new FooVisitor()); // Code was moved into FooVisitor
}</code></pre>
</div>
</div>
<div class="paragraph">
<p>or</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">// Java 14+ required
int foo(Shape shape) {
  if (shape instanceof Circle c) {
    // Code for objects of type Circle
  } else if (shape instanceof Square s) {
    // Code for objects of type Square
  }
}</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p><a href="https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/lang/ClassCastException.html">ClassCastException</a></p>
</li>
</ul>
</div>
</div>
<div class="sect2">
<h3 id="_articles_blog_posts">Articles &amp; blog posts</h3>
<div class="ulist">
<ul>
<li>
<p><a href="https://refactoring.guru/design-patterns/visitor">Refactoring Guru: Visitor</a></p>
</li>
<li>
<p><a href="https://openjdk.org/jeps/394">JEP 394: Pattern Matching for <code>instanceof</code></a></p>
</li>
</ul>
</div>
</div>
<div class="sect2">
<h3 id="_standards">Standards</h3>
<div class="ulist">
<ul>
<li>
<p><a href="https://docs.oracle.com/javase/specs/jls/se7/html/jls-15.html#jls-15.16">JLS: Cast Expressions</a></p>
</li>
<li>
<p><a href="https://docs.oracle.com/javase/specs/jls/se7/html/jls-15.html#jls-15.20.2">JLS: Type Comparison Operator <code>instanceof</code></a></p>
</li>
</ul>
</div>
</div>
</div>
</div>