This rule raises an issue when a condition-only `for` loop (equivalent to a `while` loop) does not modify any of the variables used in its condition within the loop body, and has no other exit mechanism like `break` or `return`.

== Why is this an issue?

Condition-only loops in Go use the syntax `for condition { }` and are equivalent to `while` loops in other languages. These loops continue executing as long as the condition evaluates to `true`.

For the loop to eventually terminate, something within the loop body must change the state that affects the condition. This typically means:

* Modifying variables used in the condition
* Using control flow statements like `break`, `return`, or `continue`
* Calling functions that modify global state or external resources

When none of these happen, the condition remains unchanged, creating an infinite loop. The loop will run forever, never reaching the code that follows it.

Infinite loops are particularly problematic because they consume CPU resources continuously and make the application unresponsive. In server applications, this can lead to resource exhaustion and system instability.

=== What is the potential impact?

The application will hang indefinitely, consuming CPU resources and becoming unresponsive. This can lead to:

* Poor user experience due to frozen interfaces
* Server overload and potential system crashes
* Resource exhaustion in production environments
* Difficulty in debugging, as the application appears to be running but makes no progress

== How to fix it

Modify the loop condition variable within the loop body to ensure the condition will eventually become false.

=== Code examples

==== Noncompliant code example

[source,go,diff-id=1,diff-type=noncompliant]
----
func main() {
    var value float64 = 100
    for value > 0.5 {
        fmt.Println(value) // Noncompliant
        // value never changes - infinite loop
    }
}
----

==== Compliant solution

[source,go,diff-id=1,diff-type=compliant]
----
func main() {
    var value float64 = 100
    for value > 0.5 {
        value = value / 2  // Modify the condition variable
        fmt.Println(value)
    }
}
----

== Resources

=== Documentation

 * Go Tour - For Loop - https://go.dev/tour/flowcontrol/1[Official Go documentation on for loops and their different forms]

 * Effective Go - Control Structures - https://go.dev/doc/effective_go#control-structures[Best practices for using control structures in Go]
