include::../description.adoc[]

== Noncompliant Code Example

----
void foo()
{
  int x;
  if (x == 42)
  /* Noncompliant - the following nested block is empty */
  {
  }
  else
  {
    printf("x != 42");
  }
}

void bar()
/* Compliant - functions are not nested blocks */
{
}
----

== Compliant Solution

----
void foo()
{
  int x;
  if (x != 42)
  /* Compliant */ 
  {                
    printf("x != 42");
  }
}

/* ... */
----

include::../exceptions.adoc[]
