Starting from Java 18 there is a possibility to write code snippets in javadoc comments. Within snippets, you can target specific parts of the code using a `substring` attribute to be highlighted, linked, or replaced. If the mentioned substring is not present in the line, nothing will be done. You should check that the substring does not contain a typo, or that the substring code initially targeted still exists in the snippet.

== Noncompliant Code Example

[source,java]
----
{@snippet :
   public static void main(String... args) {
       System.out.println("Hello, World!");      // @highlight substing="Non-existing string"
   }
}
----

== Compliant Solution

[source,java]
----
{@snippet :
   public static void main(String... args) {
       System.out.println("Hello, World!");      // @highlight substing="println"
   }
}
----

== See

Oracleâ€™s Programmer's Guide to Snippets: https://docs.oracle.com/en/java/javase/18/code-snippet/index.html[https://docs.oracle.com/en/java/javase/18/code-snippet/index.html]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

include::highlighting.adoc[]

endif::env-github,rspecator-view[]
