Starting Java 18 there is a possibility to write code snippets in javadoc comments. You can also highlight some parts of the code to draw attention to a desired part of a snippet by adding a `substring` attribute. If mentioned substring is not present in the line, nothing will be highlighted, thus it makes no sense to keep such attributes.

== Noncompliant Code Example

[source,java]
----
{@snippet :
   public static void main(String... args) {
       System.out.println("Hello, World!");      // @highlight substing="Non-existing string"
   }
}
----

== Compliant Solution

[source,java]
----
{@snippet :
   public static void main(String... args) {
       System.out.println("Hello, World!");      // @highlight substing="println"
   }
}
----

== See

Oracleâ€™s Programmer's Guide to Snippets: https://docs.oracle.com/en/java/javase/18/code-snippet/index.html[https://docs.oracle.com/en/java/javase/18/code-snippet/index.html]
