== Why is this an issue?

For better readability, it is preferable to use ``++DECLARE++`` instead of the unpronounceable ``++DCL++`` abbreviation.


=== Noncompliant code example

[source,pli]
----
foo: proc options(main);
  dcl i fixed decimal init (0); /* Noncompliant */
end;
----


=== Compliant solution

[source,pli]
----
foo: proc options(main);
  declare i fixed decimal init (0); /* Compliant */
end;
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

endif::env-github,rspecator-view[]
