<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when the walrus operator is used in a way which makes the code confusing, as described in <a href="https://www.python.org/dev/peps/pep-0572/#exceptional-cases">PEP 572</a>.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>The <a href="https://www.python.org/dev/peps/pep-0572">walrus operator</a> <code>:=</code> (also known as "assignment expression") should be used with caution as it can easily make code more difficult to understand and thus maintain. In such case it is advised to refactor the code and use an assignment statement (i.e. <code>=</code>) instead.</p>
</div>
<div class="paragraph">
<p>Reasons why it is better to avoid using the walrus operator in Python:</p>
</div>
<div class="ulist">
<ul>
<li>
<p><strong>Readability:</strong> The walrus operator can lead to more complex and nested expressions, which might reduce the readability of the code, especially for developers who are not familiar with this feature.</p>
</li>
<li>
<p><strong>Compatibility:</strong> If you are working on projects that need to be compatible with older versions of Python (before 3.8), you should avoid using the walrus operator, as it won&#8217;t be available in those versions.</p>
</li>
</ul>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Avoid using the walrus operator for the cases when it is not mandatory.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">v0 = (v1 := f(p))  # Noncompliant: Use an assignment statement ("=") instead; ":=" operator is confusing in this context
f'{(x:=10)}' # Noncompliant: Move this assignment out of the f-string; ":=" operator is confusing in this context</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">v0 = v1 = f(p)
x = 10
f'{x}'</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Python Documentation - <a href="https://www.python.org/dev/peps/pep-0572">PEP 572 â€“ Assignment Expressions</a></p>
</li>
<li>
<p>Python Documentation - <a href="https://www.python.org/dev/peps/pep-0572/#exceptional-cases">PEP 572 - Assignment Expressions - Exceptional cases</a></p>
</li>
</ul>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="ulist">
<ul>
<li>
<p>Use an assignment statement ("=") instead; ":=" operator is confusing in this context</p>
</li>
<li>
<p>Move this assignment out of the f-string; ":=" operator is confusing in this context</p>
</li>
</ul>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>The := operator</p>
</div>
</div>
</div>
</div>