<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Not knowing the <code>Kind</code> of the <code>DateTime</code> object that an application is using can lead to misunderstandings when displaying or comparing them. Explicitly setting the <code>Kind</code> property helps the application to stay consistent, and its maintainers understand what kind of date is being managed.
To achieve this, when instantiating a new <code>DateTime</code> object you should always use a constructor overload that allows you to define the <code>Kind</code> property.</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="paragraph">
<p>Creating the <code>DateTime</code> object without specifying the property <code>Kind</code> will set it to the default value of <code>DateTimeKind.Unspecified</code>. In this case, calling the method <code>ToUniversalTime</code> will assume that <code>Kind</code> is <code>DateTimeKind.Local</code> and calling the method <code>ToLocalTime</code> will assume that it&#8217;s <code>DateTimeKind.Utc</code>.
As a result, you might have mismatched <code>DateTime</code> objects in your application.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>To resolve this issue, use a <a href="https://learn.microsoft.com/en-us/dotnet/api/system.datetime.-ctor">constructor overload</a> that lets you specify the <a href="https://learn.microsoft.com/en-us/dotnet/api/system.datetimekind"><code>DateTimeKind</code></a> when creating the <code>DateTime</code> object. From .Net 6 onwards, use the <a href="https://learn.microsoft.com/en-us/dotnet/api/system.dateonly"><code>DateOnly</code></a> type if the time portion of the date is not relevant.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-csharp" data-lang="csharp">void CreateNewTime()
{
    var birthDate = new DateTime(1994, 7, 5, 16, 23, 42);
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-csharp" data-lang="csharp">void CreateNewTime()
{
    var birthDate = new DateTime(1994, 7, 5, 16, 23, 42, DateTimeKind.Utc);
    // or from .Net 6 onwards, use DateOnly:
    var birthDate = new DateOnly(1994, 7, 5);
}</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Microsoft Learn - <a href="https://learn.microsoft.com/en-us/dotnet/api/system.datetimekind">DateTimeKind Enum</a></p>
</li>
<li>
<p>Microsoft Learn - <a href="https://learn.microsoft.com/en-us/dotnet/api/system.datetime.-ctor">DateTime Constructors</a></p>
</li>
<li>
<p>Microsoft Learn - <a href="https://learn.microsoft.com/en-us/dotnet/api/system.dateonly">DateOnly Struct</a></p>
</li>
<li>
<p>Microsoft Learn - <a href="https://learn.microsoft.com/en-us/dotnet/standard/base-types/how-to-round-trip-date-and-time-values">How to: Round-trip Date and time values</a></p>
</li>
<li>
<p>Microsoft Learn - <a href="https://learn.microsoft.com/en-us/dotnet/standard/design-guidelines/member-overloading">Member Overloading</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>