== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,python]
----
try:
  result = do_the_thing(a)
except:  #Noncompliant; no specific exception caught
  result = None

try:
  result = do_the_thing(a)
except Exception:  #Noncompliant; too generic to be the only exception type caught
  result = None
----

=== Compliant solution

[source,python]
----
try:
  result = do_the_thing(a)
except ZeroDivisionError:
  result = None
except TypeError:
  result = -1
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
