include::../description.adoc[]

== Noncompliant Code Example

----
// The following string will error if there is a whitespace after '\' 
var str = "Hello \ 
World"; 
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
