== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,javascript]
----
// The following string will error if there is a whitespace after '\' 
var str = "Hello \ 
World"; 
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
