Boolean literals should be avoided in comparison expressions ``++==++`` and ``++!=++`` to improve code readability.

This rule also reports on redundant boolean operations.

== Noncompliant Code Example

----
let someValue = "0";
// ...

if (someValue == true) { /* ... */ }
if (someBooleanValue != true) { /* ... */ }
doSomething(!false);
----

== Compliant Solution

----
if (someValue && someValue != "0") { /* ... */ }
if (!someBooleanValue) { /* ... */ }
doSomething(true);
----
