== Why is this an issue?

Code annotated as deprecated should not be used since it will be removed sooner or later.

=== Noncompliant code example

[source,kotlin,diff-id=1,diff-type=noncompliant]
----
@Deprecated("")
interface Old

class Example : Old // Noncompliant
----
