== Why is this an issue?

include::../description.adoc[]

== How to fix it

include::../how_to_fix.adoc[]

=== Code examples

==== Noncompliant code example

[source,kotlin,diff-id=1,diff-type=noncompliant]
----
@Deprecated("This function is deprecated, use newFunction instead", ReplaceWith("newFunction()"))
fun oldFunction() {
    println("This is the old function.")
}

fun newFunction() {
    println("This is the new function.")
}

oldFunction() // Noncompliant: "oldFunction is deprecated"
----

==== Compliant solution

[source,kotlin,diff-id=1,diff-type=compliant]
----
// We deprecate it like this
@Deprecated("This function is deprecated, use newFunction instead", ReplaceWith("newFunction()"))
fun oldFunction() {
    println("This is the old function.")
}

// Here's the new function that should be used instead
fun newFunction() {
    println("This is the new function.")
}

newFunction()
----

include::../see.adoc[]
