== Why is this an issue?

Code annotated as deprecated should not be used since it will be removed sooner or later.

=== Noncompliant code example

[source,kotlin]
----
@Deprecated("")
interface Old

class Example : Old // Noncompliant
----
ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

endif::env-github,rspecator-view[]
