== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,cpp]
----
int i = 0;
loop:
  printf("i = %d\n", i);
  i++;
  if (i < 10){
    goto loop; // Noncompliant
  }
----

=== Compliant solution

[source,cpp]
----
for (int i = 0; i < 10; i++) {
  printf("i = %d\n", i);
}
----

== Resources

* MISRA C:2004, 14.4 - The goto statement shall not be used.
* MISRA C:2012, 15.1 - The goto statement should not be used
* {cpp} Core Guidelines - https://github.com/isocpp/CppCoreGuidelines/blob/e49158a/CppCoreGuidelines.md#es76-avoid-goto[ES.76: Avoid `goto`]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
