include::../description.adoc[]

== Noncompliant Code Example

----
int i = 0;
loop:
  printf("i = %d\n", i);
  i++;
  if (i < 10){
    goto loop; // Noncompliant
  }
----

== Compliant Solution

----
for (int i = 0; i < 10; i++) {
  printf("i = %d\n", i);
}
----

== See

* MISRA C:2004, 14.4 - The goto statement shall not be used.
* MISRA C:2012, 15.1 - The goto statement should not be used
* https://github.com/isocpp/CppCoreGuidelines/blob/036324/CppCoreGuidelines.md#es76-avoid-goto[{cpp} Core Guidelines ES.76]: Avoid goto

ifdef::env-github,rspecator-view[]
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
