This rule raises an issue when a map literal contains the same key more than once.

== Why is this an issue?

When a map literal contains duplicate keys, only the last value associated with that key is retained. All previous values for the same key are silently discarded.

This behavior is rarely intentional and usually indicates a programming error such as:

* Copy-paste mistakes where the same key is accidentally duplicated
* Typos in key names that result in unintended duplicates
* Refactoring errors where old key-value pairs weren't properly removed

The silent nature of this overwriting makes it particularly dangerous, as the code will run without errors but may not behave as expected. This can lead to configuration issues, data loss, or incorrect application behavior that is difficult to debug.

=== What is the potential impact?

The duplicate key issue can cause:

* *Data loss*: Earlier values are silently discarded, potentially losing important configuration or data
* *Logic errors*: The application may use unexpected values, leading to incorrect behavior
* *Debugging difficulties*: Since no error is thrown, the issue may go unnoticed until runtime problems occur
* *Maintenance problems*: Future developers may be confused about which value should be used

== How to fix it

Remove the duplicate key and keep only the intended key-value pair. If both values are needed, use different key names.

=== Code examples

==== Noncompliant code example

[source,groovy,diff-id=1,diff-type=noncompliant]
----
def config = [
    'timeout': 30,
    'retries': 3,
    'timeout': 60  // Noncompliant
]
----

==== Compliant solution

[source,groovy,diff-id=1,diff-type=compliant]
----
def config = [
    'timeout': 60,
    'retries': 3
]
----

== Resources

=== Documentation

 * Groovy Documentation - Maps - https://groovy-lang.org/syntax.html#_maps[Official Groovy documentation on map syntax and behavior]

 * CodeNarc DuplicateMapKey Rule - https://codenarc.org/codenarc-rules-basic.html#duplicatemapkey-rule[CodeNarc rule documentation for detecting duplicate map keys]

=== Related rules

 * S1534 - https://rules.sonarsource.com/javascript/RSPEC-1534/[JavaScript rule for duplicate property names in object literals]

 * S5780 - https://rules.sonarsource.com/python/RSPEC-5780/[Python rule for duplicate keys in dictionary literals]
