<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Starting from C&#43;&#43;20, mathematical constants are defined in the header <code>&lt;numbers&gt;</code>.</p>
</div>
<div class="paragraph">
<p>You should prefer using them instead of hardcoding your own constants.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">const double pi = 3.14159265358979323846; // Noncompliant

auto computeCirclePerimeter(double radius) {
    return 2 * pi * radius;
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">#include &lt;numbers&gt;

auto computeCirclePerimeter(double radius) {
    return 2 * std::numbers::pi * radius;
}</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_is_related_to_s2904">is related to: <a data-rspec-id="S2904" class="rspec-auto-link">S2904</a></h3>

</div>
<div class="sect2">
<h3 id="_is_related_to_s109">is related to: <a data-rspec-id="S109" class="rspec-auto-link">S109</a></h3>

</div>
</div>
</div>