== Why is this an issue?

Unnecessarily verbose type declarations make it harder to read the code, and should be simplified to auto-property declarations when the getters and setters contain no logic other than a simple get/set.


=== Noncompliant code example

[source,csharp]
----
private int myVar;

public int MyProperty
{
  get { return myVar; }
  set { myVar = value; }
}
----


=== Compliant solution

[source,csharp]
----
public int MyProperty { get;  set; }
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Make "xxx" an auto-property.


'''
== Comments And Links
(visible only on this page)

=== on 20 Jul 2015, 11:30:10 Tamas Vajk wrote:
\[~ann.campbell.2] Modified the sample a bit.

endif::env-github,rspecator-view[]
