<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Javadoc code snippets behave like text-blocks in terms of padding. It is the position of the ending curly brace which defines how many spaces should be placed before each line of the code snippet. Consequently, when writing code snippets, it is recommended to align the curly brace closing the snippet with the leftmost character present in the code. This will minimize the number of spaces used in front of each line in the generated javadoc.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">/**
 * {@snippet :
 *   class HelloWorld { ... }
 * } // Noncompliant - aligned with the snippet's closing bracket, but will force 2 spaces before each line of the snippet
 */
void wrongIndentation();</code></pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">/**
 * {@snippet :
 *   class HelloWorld { ... }
 *   } // Compliant - aligned with the leftmost character present in the code, i.e., the 'c' of class. No padding will be generated.
 */
void correctIndentation();</code></pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_see">See</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Oracleâ€™s Programmer&#8217;s Guide to Snippets: <a href="https://docs.oracle.com/&#8230;&#8203;/code-snippet/index.html">https://docs.oracle.com/en/java/javase/18/code-snippet/index.html</a></p>
</div>
<hr>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="ulist">
<ul>
<li>
<p>Align the last closing curly bracket to the leftmost character present in the code by adding N spaces.</p>
</li>
</ul>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="ulist">
<ul>
<li>
<p>Primary: the closing curly brace of the code snippet</p>
</li>
</ul>
</div>
</div>
<div class="sect2">
<h3 id="_quickfix">Quickfix</h3>
<div class="ulist">
<ul>
<li>
<p>Add N spaces in front of the ending curly brace.</p>
</li>
</ul>
</div>
</div>
</div>
</div>