== How to fix it in Bicep

=== Code examples

==== Noncompliant code example

[source,bicep,diff-id=1,diff-type=noncompliant]
----
@secure()
param secureStringWithDefaultValue string = 'S3CR3T' // Noncompliant
----

==== Compliant solution

[source,bicep,diff-id=1,diff-type=compliant]
----
@secure()
param secureStringWithDefaultValue string
----