<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>A function (whether it is part of the standard library, a third party library or a user defined function) may provide some means of indicating the occurrence of an error. This may be via a global error flag, a parametric error flag, a special return value or some other means. Whenever such a mechanism is provided by a function the calling program shall check for the indication of an error as soon as the function returns.</p>
</div>
<div class="paragraph">
<p>Note, however, that the checking of input values to functions is considered a more robust means of error prevention than trying to detect errors after the function has completed.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">extern void fn3 ( int32_t i, bool &amp; flag );
int32_t fn1 ( int32_t i )
{
  int32_t result = 0;
  bool success = false;
  fn3 ( i, success ); // Non-compliant - success not checked
  return result;
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">int32_t fn2 ( int32_t i )
{
  int32_t result = 0;
  bool success = false;
  fn3 ( i, success ); // Compliant - success checked
  if ( !success )
  {
    // ...
  }
  return result;
}</code></pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p>MISRA C:2004, 16.10 - If a function returns error information, then that error information shall be tested.</p>
</li>
<li>
<p>MISRA C&#43;&#43;:2008, 0-3-2 - If a function generates error information, then that error information shall be tested.</p>
</li>
</ul>
</div>
<hr>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Check the returned status of the called function</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_15_oct_2014_072727_samuel_mercier_wrote">on 15 Oct 2014, 07:27:27 Samuel Mercier wrote:</h3>
<div class="paragraph">
<p>probably cannot be implemented, as this requires understanding how the status is returned (parameter or return value)</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_16_oct_2014_144134_ann_campbell_wrote">on 16 Oct 2014, 14:41:34 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~samuel.mercier], the message may need work. The description makes clear that there are a number of places the error indication could be in - not just the return value.</p>
</div>
</div>
</div>
</div>