== Why is this an issue?

Call stacks containing lot of ``++PERFORM++`` statements is a key ingredient for making what's known as "Spaghetti code".

Such code is hard to read, refactor and therefore maintain.


This rule supports both sections and paragraphs. 


=== Noncompliant code example

With a threshold of 3:

[source,cobol]
----
PERFORM FIRST.

FIRST.
  PERFORM SECOND.

SECOND.
  PERFORM THIRD.

THIRD.
  PERFORM FOURTH.  *> Noncompliant

FOURTH.
  DISPLAY something.
----



ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

include::parameters.adoc[]

'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
