== Why is this an issue?

Developers sometimes "reserve" an ``++enum++`` member for future use, i.e. these members won't be used right but they might change in the future. However changing an ``++enum++`` member is a breaking change and can create significant problems. There is no need to reserve an ``++enum++`` member since a new member can be added in the future, and such an addition will usually not be a breaking change.


This rule raises an issue when an ``++enum++`` contains multiple members whose name match the pattern ``++.*reserved.*\d+++``Â (case insensitive) and they all have exactly the same name except for the number at the end.


=== Noncompliant code example

[source,text]
----
using System;

namespace MyLibrary
{
  public enum Color // Noncompliant
  { 
        None, 
        Red, 
        Orange, 
        Yellow,
        Reserved_1,  // reserved
        Reserved_2 // reserved
    }  
}
----

