== Why is this an issue?

While it is technically correct to use a ``++Thread++`` where a ``++Runnable++`` is called for, the semantics of the two objects are different, and mixing them is a bad practice that will likely lead to headaches in the future.


The crux of the issue is that ``++Thread++`` is a larger concept than ``++Runnable++``. A ``++Runnable++`` is an object whose running should be managed. A ``++Thread++`` expects to manage the running of itself or other ``++Runnables++``. 


=== Noncompliant code example

[source,java]
----
	public static void main(String[] args) {
		Thread r =new Thread() {
			int p;
			@Override
			public void run() {
				while(true)
					System.out.println("a");
			}
		};
		new Thread(r).start();  // Noncompliant
----


=== Compliant solution

[source,java]
----
	public static void main(String[] args) {
		Runnable r =new Runnable() {
			int p;
			@Override
			public void run() {
				while(true)
					System.out.println("a");
			}
		};
		new Thread(r).start();
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

* Argument n is a "Thread".
* "XXX" is a "Thread".


'''
== Comments And Links
(visible only on this page)

=== on 20 Jan 2015, 16:20:05 Ann Campbell wrote:
\[~nicolas.peru] I'd like to add a simile to illustrate the issue. What do you think of this?


____
It would be something like asking Sir Edmund Hillary to take a hike up the nearest hill with you.
____

endif::env-github,rspecator-view[]
