=== How to fix it in CakePHP

include::../../common/fix/code-rationale.adoc[]

==== Noncompliant code example

[source,php,diff-id=1,diff-type=noncompliant]
----

----

==== Compliant solution

[source,php,diff-id=1,diff-type=compliant]
----

----

include::../../common/fix/how-does-this-work.adoc[]
