=== How to fix it in Core PHP

include::../../common/fix/code-rationale.adoc[]

==== Noncompliant code example

[source,php,diff-id=1,diff-type=noncompliant]
----
$input = $_GET["input"];
error_log($input); // Noncompliant
----

==== Compliant solution

[source,php,diff-id=1,diff-type=compliant]
----
$input = $_GET["input"];
$badchars = array("\n", "\r", "\t");
if preg_match("[^A-Za-z0-9_\-]", $input) {
  $safeinput = '[' . base64_encode($input) . ']';
} else 
  $safeinput = $input;
}
error_log($safeinput);
----

include::../../common/fix/how-does-this-work.adoc[]

Here, the example compliant code uses the `preg_replace` function to remove any non safe character from the input.
