<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Referencing a column by specifying the schema or the database is deprecated. It is retained temporarily for backward compatibility, but it will eventually be removed from the language. You should only use one part (<code>column_name</code>) or two part (<code>table_name.column_name</code>) references.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-sql" data-lang="sql">SELECT dbo.table1.col1,       -- Noncompliant, three-part column reference
       MY_DB.dbo.table1.col2  -- Noncompliant, four-part column reference
       FROM MY_DB.dbo.table1;

SELECT dbo.table1.name,       -- Noncompliant
       dbo.table2.name        -- Noncompliant
       FROM dbo.table1
       JOIN dbo.table2
         ON dbo.table2.id = dbo.table1.id; -- Noncompliant</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-sql" data-lang="sql">SELECT col1,
       col2
       FROM MY_DB.dbo.table1;

SELECT table1.name,
       table2.name
       FROM dbo.table1
       JOIN dbo.table2
         ON table2.id = table1.id;</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove database or schema reference.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>schema</p>
</div>
</div>
</div>
</div>