== Why is this an issue?

If you have an iterable, such as an array, set, or list, your best option for looping through its values is the ``++for of++`` syntax. Use ``++for in++`` and you'll iterate the properties, rather than the values.


=== Noncompliant code example

[source,javascript]
----
const arr = [4, 3, 2, 1];

for (let value in arr) {  // Noncompliant 
  console.log(value);  // logs 0, 1, 2, 3
}
----


=== Compliant solution

[source,javascript]
----
const arr = [4, 3, 2, 1];

for (let value of arr) { 
  console.log(value); 
}
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

include::highlighting.adoc[]

endif::env-github,rspecator-view[]
