<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p><code>MutableStateFlow</code> and <code>MutableSharedFlow</code> are very convenient for storing and adding updates of some data structures in event-driven paradigm. This is widely used in Android Views for handling updates. While it&#8217;s extremely useful to manage such objects inside some class, it&#8217;s not recommended to expose them outside of the class.</p>
</div>
<div class="paragraph">
<p>When properties of the types <code>MutableStateFlow</code> or <code>MutableSharedFlow</code> are accessible from outside of a class, data updates cannot be verified properly anymore. It is generally recommended to have only one class responsible for updating these flows, otherwise inconsistency issues and problems with maintainability, as well as increased error-proneness may be introduced.</p>
</div>
<div class="paragraph">
<p>To restrict write access, <code>StateFlow</code> or <code>SharedFlow</code> should be used together with private <code>MutableStateFlow</code> or <code>MutableSharedFlow</code> fields.</p>
</div>
<div class="paragraph">
<p>This rule raises an issue when encountering a public or internal property of the type <code>MutableStateFlow</code> or <code>MutableSharedFlow</code>.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-kotlin" data-lang="kotlin">class MyView : ViewModel() {

    val state = MutableStateFlow(State.New)

}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-kotlin" data-lang="kotlin">class MyView : ViewModel() {

    private val _state = MutableStateFlow(State.New)
    val state: StateFlow&lt;LatestNewsUiState&gt; = _uiState

}</code></pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p><a href="https://developer.android.com/kotlin/coroutines/coroutines-best-practices#mutable-types">Android Coroutines Best Practices</a></p>
</li>
</ul>
</div>
</div>
</div>