Curly brace quantifiers in regular expressions can be used to have a more fine-grained control over how many times the character or the sub-expression preceeding them should occur. They can be used to match an expression exactly n times with ``++{n}++``, between n and m times with ``++{n,m}++``, or at least n times with ``++{n,}++``. In some cases, using such a quantifier is superfluous for the semantic of the regular expression, and it can be removed to improve readability. 

- ``++{1,1}++`` and ``++{1}++`` match the expression exactly once. The same behavior can be achieved without the quantifier.
- ``++{0,0}++`` and ``++{0}++`` match the expression zero times. The same behavior can be achieved by removing the expression.

== Noncompliant Code Example

----
"/ab{1,1}c/"
"/ab{1}c/"
"/ab{0,0}c/"
"/ab{0}c/"
----

== Compliant Solution

----
"/abc/"
"/ac/"
----

== See

FIXME: A list of links