<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Filename and pathname should be prefixed to avoid missing filenames starting with a dash when globbing files as program option.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Within the command, arguments and filenames as options are passed as strings.
Programs may misinterpret files as arguments if the file name starts with a single or double dash.
When filenames are specified using glob, files whose names begin with a dash are not included in the scope.
For example, if a file is named "-f" and passed as an argument to a command such as "rm", it may be interpreted as a command line option instead of a file, causing unexpected behaviour.
This issue affects all instructions processing shell commands.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>To avoid this issue, using "./" before the file name will cause it to be interpreted as a file and not an option.
Additionally, adding "--" at the end of command line arguments indicates the end of options and any subsequent arguments will be treated as file names.
However, this fix should only be used if the command supports it as expected.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-docker" data-lang="docker">RUN rm *</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-docker" data-lang="docker">RUN rm ./*
RUN rm -- *</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p><a href="https://dwheeler.com/essays/filenames-in-shell.html">Filenames and Pathnames in Shell: How to do it Correctly</a></p>
</li>
</ul>
</div>
<hr>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Prefix files and paths with <code>./</code> or <code>--</code> when using glob.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>Highlight the entire command which is using glob for file or path option.</p>
</div>
<hr>
</div>
</div>
</div>