== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,swift]
----
if a == b { // Compliant; a reassigned in previous block
  doSomething(b)
}
if a == b {  // Noncompliant; is this really what was intended?
  doTheThing(c)
}
----

=== Compliant solution

[source,swift]
----
if a == b { 
  doTheThing(b)
  doTheThing(c)
}
----
or

[source,swift]
----
if a == b { 
  doTheThing(b)
}
if b == c {
  doTheThing(c)
}
----

include::../exceptions.adoc[]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
