== How to fix it in Django

=== Code examples

include::../../common/fix/code-rationale.adoc[]

==== Noncompliant code example

[source,python,diff-id=1,diff-type=noncompliant]
----
from django.conf import settings

settings.configure(DEBUG=True)  # Noncompliant
settings.configure(DEBUG_PROPAGATE_EXCEPTIONS=True)  # Noncompliant

def custom_config(config):
    settings.configure(default_settings=config, DEBUG=True)  # Noncompliant
----

[source,python,diff-id=2,diff-type=noncompliant]
----
DEBUG = True  # Sensitive
DEBUG_PROPAGATE_EXCEPTIONS = True
----

==== Compliant solution

[source,python,diff-id=1,diff-type=compliant]
----
from django.conf import settings

settings.configure(DEBUG=False)
settings.configure(DEBUG_PROPAGATE_EXCEPTIONS=False)

def custom_config(config):
    settings.configure(default_settings=config, DEBUG=False)
----

[source,python,diff-id=2,diff-type=compliant]
----
DEBUG = False
DEBUG_PROPAGATE_EXCEPTIONS = False
----

=== How does this work?

include::../../common/fix/how.adoc[]
