== How to fix it in Flask

=== Code examples

include::../../common/fix/code-rationale.adoc[]

==== Noncompliant code example

[source,python,diff-id=11,diff-type=noncompliant]
----
from flask import Flask
app = Flask()
app.debug = True  # Noncompliant
app.run(debug=True)  # Noncompliant
----

==== Compliant solution

[source,python,diff-id=11,diff-type=compliant]
----
from flask import Flask
app = Flask()
app.debug = False
app.run(debug=False)
----

=== How does this work?

include::../../common/fix/how.adoc[]
