<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Enabling case-insensitive matching using <code>(?i)</code> within a regex will only apply to characters in the ASCII range. To include characters outside of this range, use the <code>RegexOption.IGNORE_CASE</code> flag. Alternatively, you can use <code>(?iu)</code> or <code>(?iU)</code> within the regex instead.</p>
</div>
<div class="paragraph">
<p>If not done, regular expressions involving non-ASCII letters will still handle those letters as being case sensitive.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>Regex("(?i)söme pättern")
Regex("(?i:söme) pättern")

"(?i)söme pättern".toRegex()
"(?i:söme) pättern".toRegex()</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>Regex("söme pättern", RegexOption.IGNORE_CASE)
Regex("(?iu)söme pättern")
Regex("(?iu:söme) pättern")

"söme pättern".toRegex(RegexOption.IGNORE_CASE)
"(?iu)söme pättern".toRegex()
"(?iu:söme) pättern".toRegex()</pre>
</div>
</div>
<hr>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="ulist">
<ul>
<li>
<p>when Pattern.CASE_INSENTIVE was used: Also use "Pattern.UNICODE_CASE" to correctly handle non-ASCII letters.</p>
</li>
<li>
<p>when the "i" flag was used: Also use the "u" flag to correctly handle non-ASCII letters.</p>
</li>
</ul>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>The use of <code>Pattern.CASE_INSENTIVE</code> or the <code>i</code> in <code>(?i)</code> or <code>(?i:&#8230;&#8203;)</code></p>
</div>
</div>
</div>
</div>