include::../description.adoc[]

== Noncompliant code example

With the default regular expression ``++^[a-z][a-zA-Z0-9]*$++``:

[source, javascript]
----
function DoSomething(){...}  // Noncompliant
----

== Compliant Solution

[source, javascript]
----
function doSomething(){...}
----
