include::../description.adoc[]

== Noncompliant Code Example

With the default regular expression ``++^[a-z][a-zA-Z0-9]*$++``:

----
function DoSomething(){...}  // Noncompliant
----

== Compliant Solution

----
function doSomething(){...}
----
