<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Shared coding conventions allow teams to collaborate effectively. Typing operation codes, keywords, specification codes, compiler directives and built-in functions in upper case makes them more immediately recognizable as such, thereby making the code easier to read.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-text" data-lang="text">     c     test12        plist
     c                   parm                    asdf01
     c                   parm                    asdf02
     c                   parm                    @pdq</code></pre>
</div>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-text" data-lang="text">      /free
        if x = *blank;
          read a;
        endif;
      /end-free</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-text" data-lang="text">     C     test12        PLIST
     C                   PARM                    asdf01
     C                   PARM                    asdf02
     C                   PARM                    @pdq</code></pre>
</div>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-text" data-lang="text">      /free
        IF x = *BLANK;
          READ a;
        ENDIF;
      /end-free</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Correct the case of "xxx".</p>
</div>
</div>
<div class="sect2">
<h3 id="_parameters">Parameters</h3>
<div class="sidebarblock">
<div class="content">
<div class="title">operationCodes</div>
<div class="listingblock">
<div class="content">
<pre>true</pre>
</div>
</div>
<div class="paragraph">
<p>Whether or not to check the case of operation codes.</p>
</div>
</div>
</div>
<div class="sidebarblock">
<div class="content">
<div class="title">keywords</div>
<div class="listingblock">
<div class="content">
<pre>true</pre>
</div>
</div>
<div class="paragraph">
<p>Whether or not to check the case of keywords.</p>
</div>
</div>
</div>
<div class="sidebarblock">
<div class="content">
<div class="title">specificationcodes</div>
<div class="listingblock">
<div class="content">
<pre>false</pre>
</div>
</div>
<div class="paragraph">
<p>Whether or not to check the case of specification codes.</p>
</div>
</div>
</div>
<div class="sidebarblock">
<div class="content">
<div class="title">compilerDirectives</div>
<div class="listingblock">
<div class="content">
<pre>false</pre>
</div>
</div>
<div class="paragraph">
<p>Whether or not to check the case of compiler directives.</p>
</div>
</div>
</div>
<div class="sidebarblock">
<div class="content">
<div class="title">builtInFunctions</div>
<div class="listingblock">
<div class="content">
<pre>false</pre>
</div>
</div>
<div class="paragraph">
<p>Whether or not to check the case of built-in functions</p>
</div>
</div>
</div>
</div>
</div>
</div>