== Why is this an issue?

Shared coding conventions allow teams to collaborate effectively. Typing operation codes, keywords, specification codes, compiler directives and built-in functions in upper case makes them more immediately recognizable as such, thereby making the code easier to read.


=== Noncompliant code example

[source,text]
----
     c     test12        plist
     c                   parm                    asdf01
     c                   parm                    asdf02
     c                   parm                    @pdq
----

[source,text]
----
      /free
        if x = *blank;
          read a;
        endif;
      /end-free
----


=== Compliant solution

[source,text]
----
     C     test12        PLIST
     C                   PARM                    asdf01
     C                   PARM                    asdf02
     C                   PARM                    @pdq
----

[source,text]
----
      /free
        IF x = *BLANK;
          READ a;
        ENDIF;
      /end-free
----


