<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when <code>pluck(:id)</code> is used on Active Record collections to extract primary key IDs.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Active Record provides a dedicated <code>ids</code> method specifically for extracting primary key IDs from collections. Using <code>pluck(:id)</code> is less semantic and doesn&#8217;t clearly express the intent to retrieve primary key values.</p>
</div>
<div class="paragraph">
<p>The <code>ids</code> method is more expressive and immediately communicates that you&#8217;re working with primary keys. It follows Rails conventions and makes the code more readable for other developers who are familiar with Rails patterns.</p>
</div>
<div class="paragraph">
<p>While both methods produce the same result, using the purpose-built <code>ids</code> method aligns with Rails best practices and improves code clarity.</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="paragraph">
<p>Using <code>pluck(:id)</code> instead of <code>ids</code> doesn&#8217;t cause functional issues, but it makes the code less expressive and harder to understand. Following Rails conventions improves code maintainability and helps other developers quickly understand the intent.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Replace <code>pluck(:id)</code> with the more semantic <code>ids</code> method when extracting primary key IDs from Active Record collections.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-ruby" data-lang="ruby">User.pluck(:id) # Noncompliant
User.where(active: true).pluck(:id) # Noncompliant
post.comments.pluck(:id) # Noncompliant</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-ruby" data-lang="ruby">User.ids
User.where(active: true).ids
post.comments.ids</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Rails Style Guide - ids - <a href="https://rails.rubystyle.guide/#ids">Official Rails Style Guide recommendation for using ids over pluck(:id)</a></p>
</li>
<li>
<p>Active Record Query Interface - <a href="https://guides.rubyonrails.org/active_record_querying.html#ids">Rails documentation on the ids method for Active Record queries</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>