<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>The use of <code>org.fest.assertions.Assertions.assertThat</code> by itself does nothing. You must combine it with another method that actually tests the value in use.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">assertThat(name);  // Noncompliant</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">assertThat(name).isNotNull();</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_duplicates_s2970">duplicates: <a data-rspec-id="S2970" class="rspec-auto-link">S2970</a></h3>

</div>
<div class="sect2">
<h3 id="_on_22_apr_2015_160223_nicolas_peru_wrote">on 22 Apr 2015, 16:02:23 Nicolas Peru wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] This rule should precise that it checks for <code>org.fest.assertions.Assertions.assertThat</code> methods. As this is not applicable for junit <code>Assert.assertThat </code></p>
</div>
</div>
</div>
</div>