Type assertion can be done in two ways: with ``++as MyType++`` or with ``++<MyType>++``. But since there is an ambiguity in the latter when using JSX and there is no ambiguity in the former, ``++as++`` is preferred.


== Noncompliant Code Example

----
var foo = <any>"foo";  // Noncompliant
----


== Compliant Solution

----
var foo = "foo" as any;
----


ifdef::env-github,rspecator-view[]
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
