== How to fix it in Go Standard Library

=== Code examples

:code_impact: read

include::../../common/fix/code-rationale.adoc[]

==== Noncompliant code example

[source,go,diff-id=1,diff-type=noncompliant]
----
import (
    "net/http"
    "os"
    "path/filepath"
    "myapp/config"
)

func requestHandler(resp http.ResponseWriter, req *http.Request) {
    staticFilesRoot := config.GetStaticFilesRoot()
    fileName := req.URL.Query().Get("file")
    filePath := filepath.Join(staticFilesRoot, fileName)
    data, err := os.ReadFile(filePath) // Noncompliant
    if err != nil {
        resp.Write(data)
    }
}
----

==== Compliant solution

[source,go,diff-id=1,diff-type=compliant]
----
import (
    "net/http"
    "os"
    "path/filepath"
    "strings"
    "myapp/config"
)

func requestHandler(resp http.ResponseWriter, req *http.Request) {
    staticFilesRoot := config.GetStaticFilesRoot()
    fileName := req.URL.Query().Get("file")
    filePath := filepath.Join(staticFilesRoot, fileName)
    if !strings.HasPrefix(filePath, staticFilesRoot) {
        return
    }
    data, err := os.ReadFile(filePath)
    if err != nil {
        resp.Write(data)
    }
}
----

=== How does this work?

:canonicalization_function: path/filepath.Abs

include::../../common/fix/self-validation.adoc[]

