== Why is this an issue?

An unused parameter is a parameter that has been declared but is not used anywhere in the block of code where it is defined.
It is dead code, contributing to unnecessary complexity and leading to confusion when reading the code.
Therefore, it should be removed from your code to maintain clarity and efficiency.

=== What is the potential impact?

Having unused parameters in your code can lead to several issues:

* *Decreased Readability*: Unused parameters can make your code more difficult to read.
They add extra lines and complexity, which can distract from the main logic of the code.

* *Misunderstanding*: When other developers read your code, they may wonder why a variable is declared but not used.
This can lead to confusion and misinterpretation of the code's intent.

* *Potential for Bugs*: If a parameter is declared but not used, it might indicate a bug or incomplete code.
For example, if you declared a parameter intending to use it in a calculation, but then forgot to do so, your program might not work as expected.

* *Maintenance Issues*: Unused parameters can make code maintenance more difficult.
If a programmer sees an unused parameter, they might think it is a mistake and try to 'fix' the code, potentially introducing new bugs.

In summary, unused local parameters can make your code less readable, more confusing, and harder to maintain, and they can potentially lead to bugs.
Therefore, it is best to remove them.


== How to fix it in JSON templates

include::../how-to-fix-it-text.adoc[]

=== Code examples

==== Noncompliant code example

[source,json,diff-id=1,diff-type=noncompliant]
----
{
  "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#",
  "parameters": {
      "unusedParameter": {
          "type": "string"
      },
      "virtualMachinesName": {
          "type": "string"
      }
  },
  "resources": [
    {
      "type": "Microsoft.Compute/virtualMachines",
      "name": "[parameters('virtualMachinesName')]",
      "apiVersion": "2023-09-01",
      "location": "[resourceGroup().location]"
    }
  ]
}
----

==== Compliant solution

[source,json,diff-id=1,diff-type=compliant]
----
{
  "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#",
  "parameters": {
      "virtualMachinesName": {
          "type": "string"
      }
  },
  "resources": [
    {
      "type": "Microsoft.Compute/virtualMachines",
      "name": "[parameters('virtualMachinesName')]",
      "apiVersion": "2023-09-01",
      "location": "[resourceGroup().location]"
    }
  ]
}
----

== How to fix it in Bicep

include::../how-to-fix-it-text.adoc[]

=== Code examples

==== Noncompliant code example

[source,bicep,diff-id=2,diff-type=noncompliant]
----
param unusedParameter string // Noncompliant
param virtualMachinesName string = '${uniqueString(resourceGroup().id)}'

resource vm 'Microsoft.Compute/virtualMachines@2023-09-01' = {
  name: virtualMachinesName
  location: resourceGroup().location
}
----

==== Compliant solution

[source,bicep,diff-id=2,diff-type=compliant]
----
param virtualMachinesName string = '${uniqueString(resourceGroup().id)}'

resource vm 'Microsoft.Compute/virtualMachines@2023-09-01' = {
  name: virtualMachinesName
  location: resourceGroup().location
}
----

== Resources

=== Documentation

* Microsoft - https://learn.microsoft.com/en-us/azure/azure-resource-manager/templates/best-practices#parameters[ARM template best practices]
* Microsoft - https://learn.microsoft.com/en-us/azure/azure-resource-manager/bicep/best-practices#names[Best practices for Bicep]
* Microsoft - https://learn.microsoft.com/en-us/azure/azure-resource-manager/templates/parameters[Parameters in ARM templates]
* Microsoft - https://learn.microsoft.com/en-us/azure/azure-resource-manager/bicep/parameters[Parameters in Bicep]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Remove the unused parameter "xxxx".

=== Highlighting

Highlight the unused parameter.

'''
== Comments And Links
(visible only on this page)

endif::env-github,rspecator-view[]
