== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,csharp]
----
public class Foo
{
  public int MagicNumber = 42; 
}
----

=== Compliant solution

[source,csharp]
----
public class Foo
{
  public int MagicNumber 
  { 
    get { return 42; }
  }
}
----
or 

[source,csharp]
----
public class Foo
{
  private int MagicNumber = 42; 
}
----

=== Exceptions

``++struct++``s are ignored, as are ``++static++`` and ``++const++`` fields in classes. 

Further, an issue is only raised when the real accessibility is ``++public++``, taking into account the class accessibility.

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

=== on 29 May 2015, 12:29:45 Ann Campbell wrote:
double-check my changes, please [~tamas.vajk]

=== on 29 May 2015, 13:00:48 Tamas Vajk wrote:
\[~ann.campbell.2] Seems good.

=== on 23 Mar 2017, 11:43:27 Ann Campbell wrote:
Done [~amaury.leve]

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
