<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Reading a non-existent property on an object always returns <code>undefined</code>. Doing so is usually an error; either in the name of the property or the type of the variable being accessed.</p>
</div>
<div class="paragraph">
<p>If an attempt is made to access properties of a primitive, the primitive is automatically encased in a primitive-wrapper object for the operation. But being "promoted" to an object doesn&#8217;t mean that the primitive will actually have properties to access. The wrapper object still won&#8217;t have the non-existent property and <code>undefined</code> will be returned instead.</p>
</div>
<div class="paragraph">
<p>This rule raises an issue when an attempt is made to access properties of a primitive. Thus this rule should only be activated when you don&#8217;t use monkey patching for standard objects, like <code>Number</code>, <code>Boolean</code> and <code>String</code>.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-javascript" data-lang="javascript">x = 42;
y = x.length;   // Noncompliant, Number type doesn't have "length" property</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_exceptions">Exceptions</h3>
<div class="paragraph">
<p>The Ember framework introduces a few extensions to <code>String</code>. Since it is a widely used package, the following <code>String</code> properties will not trigger this rule even though they are not built-in:</p>
</div>
<div class="ulist">
<ul>
<li>
<p><code>camelize</code></p>
</li>
<li>
<p><code>capitalize</code></p>
</li>
<li>
<p><code>classify</code></p>
</li>
<li>
<p><code>dasherize</code></p>
</li>
<li>
<p><code>decamelize</code></p>
</li>
<li>
<p><code>fmt</code></p>
</li>
<li>
<p><code>loc</code></p>
</li>
<li>
<p><code>underscore</code></p>
</li>
<li>
<p><code>w</code></p>
</li>
</ul>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_1_nov_2019_173452_elena_vilchik_wrote">on 1 Nov 2019, 17:34:52 Elena Vilchik wrote:</h3>
<div class="paragraph">
<p>See https://github.com/SonarSource/SonarJS/issues/1698</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_21_oct_2020_164443_guillaume_dequenne_wrote">on 21 Oct 2020, 16:44:43 Guillaume Dequenne wrote:</h3>
<div class="paragraph">
<p>Dropping this rule as it has low value. See:</p>
</div>
<div class="paragraph">
<p><a href="https://github.com/SonarSource/SonarJS/issues/2208" class="bare">https://github.com/SonarSource/SonarJS/issues/2208</a></p>
</div>
<div class="paragraph">
<p><a href="https://github.com/SonarSource/SonarJS/issues/1698" class="bare">https://github.com/SonarSource/SonarJS/issues/1698</a></p>
</div>
</div>
</div>
</div>