<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Regardless of scope, no identifier should be re-used across any files in the system.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">struct air_speed
{
  uint16_t speed; /* knots */
} * x;

struct gnd_speed
{
  uint16_t speed; /* mph */ // Not Compliant - speed is in different units
} * y;

x-&gt;speed = y-&gt;speed; // Error: unit conversion required</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">struct air_speed
{
  uint16_t knotSpeed; /* knots */
} * x;

struct gnd_speed
{
  uint16_t mphSpeed; /* mph */ // Compliant, different name
} * y;

x-&gt;knotSpeed = y-&gt;mphSpeed; // An error can be deducted from the name of the fields.</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_exceptions">Exceptions</h3>
<div class="paragraph">
<p>Where an identifier name is used in a header file, and that header file is included in multiple source files, this rule is not violated. The use of a rigorous naming convention can support the implementation of this rule.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p>MISRA C:2004, 5.7</p>
</li>
</ul>
</div>
<hr>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Rename this declaration, whose identifier has already been used.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_is_related_to_s805">is related to: <a data-rspec-id="S805" class="rspec-auto-link">S805</a></h3>

</div>
<div class="sect2">
<h3 id="_relates_to_s807">relates to: <a data-rspec-id="S807" class="rspec-auto-link">S807</a></h3>

</div>
<div class="sect2">
<h3 id="_relates_to_s832">relates to: <a data-rspec-id="S832" class="rspec-auto-link">S832</a></h3>

</div>
<div class="sect2">
<h3 id="_is_related_to_s802">is related to: <a data-rspec-id="S802" class="rspec-auto-link">S802</a></h3>

</div>
<div class="sect2">
<h3 id="_is_related_to_s803">is related to: <a data-rspec-id="S803" class="rspec-auto-link">S803</a></h3>

</div>
<div class="sect2">
<h3 id="_is_related_to_s828">is related to: <a data-rspec-id="S828" class="rspec-auto-link">S828</a></h3>

</div>
<div class="sect2">
<h3 id="_is_related_to_s1014">is related to: <a data-rspec-id="S1014" class="rspec-auto-link">S1014</a></h3>

</div>
<div class="sect2">
<h3 id="_on_17_oct_2014_152202_ann_campbell_wrote">on 17 Oct 2014, 15:22:02 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~freddy.mallet] this rule overlaps a couple other, linked issues.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_17_oct_2014_152325_ann_campbell_wrote">on 17 Oct 2014, 15:23:25 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~samuel.mercier] as written, this rule makes me sad. It will presumably flag every instance of <code>int i</code>.</p>
</div>
</div>
</div>
</div>