== Why is this an issue?

Element selections that could be matched anywhere in the document can be very slow. That's why use of the universal selector, ``++*++``, should be limited; it explicitly specifies that the match could be anywhere.


=== Noncompliant code example

[source,javascript]
----
$( ".buttons > *" );  // Noncompliant; extremely expensive
----


=== Compliant solution

[source,javascript]
----
$( ".buttons" ).children(); // Compliant
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

* Remove the use of this universal selector.


'''
== Comments And Links
(visible only on this page)

=== on 17 Mar 2015, 11:02:13 Linda Martin wrote:
\[~ann.campbell.2] assign for completion and review.



=== on 1 Nov 2019, 17:28:58 Elena Vilchik wrote:
See \https://github.com/SonarSource/SonarJS/issues/1698

endif::env-github,rspecator-view[]
