Element selections that could be matched anywhere in the document can be very slow. That's why use of the universal selector, ``++*++``, should be limited; it explicitly specifies that the match could be anywhere.


== Noncompliant Code Example

----
$( ".buttons > *" );  // Noncompliant; extremely expensive
----


== Compliant Solution

----
$( ".buttons" ).children(); // Compliant
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
