This rule raises an issue when a finalizer sets a field to `null`.

== Why is this an issue?

In the Java object lifecycle,
the `finalize` method
is called only after the garbage collector
has already determined that the object
can be removed from the heap.

It is therefore unnecessary for a finalizer to




Note, however, that whether finalizers will be called at all,
and if so, when specifically they will be called, is up to the implementation of the used Java Runtime Environment.

Using finalizer is therefore a bad practice,
especially when using the `finalize` method to free resources (e.g., closing a stream et c.).





The only purpose for finalizers is
to possibly revive objects
in case




There is no point in setting class fields to ``++null++`` in a finalizer.
If this this is a hint to the garbage collector, it is unnecessary -
the object will be garbage collected anyway -
and doing so may actually cause extra work for the garbage collector.


=== Noncompliant code example

[source,java]
----
public class Foo {
  private String name;

  @Override
  void finalize() {
    name = null;  // Noncompliant; completely unnecessary
  }
}
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Remove this nullification of "xxx".


endif::env-github,rspecator-view[]
