== Noncompliant Code Example

[source,text]
----
class MyClass 
{
  private object obj = new object();

  public void DoSomething() 
  {
    Monitor.Enter(obj); // Noncompliant
    if (IsInitialized())
    {
      // ...
      Monitor.Exit(obj);
    }
  }
}
----
