== Why is this an issue?

include::description.adoc[]

=== Noncompliant code example

[source,text]
----
class MyClass 
{
  private object obj = new object();

  public void DoSomething() 
  {
    Monitor.Enter(obj); // Noncompliant
    if (IsInitialized())
    {
      // ...
      Monitor.Exit(obj);
    }
  }
}
----


=== Compliant solution

[source,text]
----
class MyClass 
{
  private object obj = new object();

  public void DoSomething() 
  {
    lock(obj) // lock() {...} is easier to use than explicit Monitor calls
    {
      if (IsInitialized())
      {
      }
    }
  }
}
----


include::see.adoc[]

