<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Style selectors with a lot of elements are difficult to understand and to maintain. Further, they hamper the maintainability of the underlying documents to which they&#8217;re applied, having, in effect, a double whammy.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-text" data-lang="text">div.main span.login input.btn {  // Noncompliant using the default threshold of 3
  // ...
}</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_6_may_2015_071904_stas_vilchik_wrote">on 6 May 2015, 07:19:04 Stas Vilchik wrote:</h3>
<div class="paragraph">
<p>Looks good</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_20_jul_2015_115000_ann_campbell_wrote">on 20 Jul 2015, 11:50:00 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>formerly targeted to CSS</p>
</div>
</div>
</div>
</div>