== Why is this an issue?

Style selectors with a lot of elements are difficult to understand and to maintain. Further, they hamper the maintainability of the underlying documents to which they're applied, having, in effect, a double whammy.


=== Noncompliant code example

[source,text]
----
div.main span.login input.btn {  // Noncompliant using the default threshold of 3
  // ...
}
----


ifdef::env-github,rspecator-view[]

'''
== Comments And Links
(visible only on this page)

=== on 6 May 2015, 07:19:04 Stas Vilchik wrote:
Looks good

=== on 20 Jul 2015, 11:50:00 Ann Campbell wrote:
formerly targeted to CSS


endif::env-github,rspecator-view[]
