Style selectors with a lot of elements are difficult to understand and to maintain. Further, they hamper the maintainability of the underlying documents to which they're applied, having, in effect, a double whammy.


== Noncompliant Code Example

----
div.main span.login input.btn {  // Noncompliant using the default threshold of 3
  // ...
}
----


