<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Use of <code>File.deleteOnExit()</code> is not recommended for the following reasons:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>The deletion occurs only in the case of a normal JVM shutdown but not when the JVM crashes or is killed.</p>
</li>
<li>
<p>For each file handler, the memory associated with the handler is released only at the end of the process.</p>
</li>
</ul>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">File file = new File("file.txt");
file.deleteOnExit();  // Noncompliant</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove this call to "deleteOnExit".</p>
</div>
</div>
</div>
</div>