This rule raises an issue when synchronous file operations like `open()` are used within asynchronous functions.

== Why is this an issue?

Using synchronous file operations like `open()` in asynchronous code blocks the entire event loop. This undermines the primary advantage of asynchronous programming - the ability to perform concurrent operations without blocking execution.

When an async function makes a synchronous file operation:

* The event loop is completely blocked until the file I/O operation completes
* No other coroutines can run during this time, even if they're ready to execute
* The responsiveness of the application is degraded
* In server applications, this can cause timeouts or failures for other concurrent requests

Instead, async libraries provide mechanisms to handle file operations asynchronously:

* `aiofiles` library for asyncio
* `trio.open_file()` for Trio
* `anyio.open_file()` for AnyIO

Using these constructs allows other tasks to continue executing while waiting for the potentially blocking file operation to complete.

== How to fix it in Asyncio

The standard library's `asyncio` package doesn't provide built-in asynchronous file I/O operations. Use the `aiofiles` library to handle file operations asynchronously.

=== Code examples

==== Noncompliant code example

[source,python,diff-id=1,diff-type=noncompliant]
----
async def read_config():
    with open("config.json", "r") as file:  # Noncompliant
        data = file.read()
    return data
----

==== Compliant solution

[source,python,diff-id=1,diff-type=compliant]
----
import aiofiles

async def read_config():
    async with aiofiles.open("config.json", "r") as file: # Compliant
        data = await file.read()
    return data
----

== How to fix it in Trio

Use `trio.open_file()` to handle file operations asynchronously.

=== Code examples

==== Noncompliant code example

[source,python,diff-id=2,diff-type=noncompliant]
----
async def read_config():
    with open("config.json", "r") as file:  # Noncompliant
        data = file.read()
    return data
----

==== Compliant solution

[source,python,diff-id=2,diff-type=compliant]
----
import trio

async def read_config():
    async with await trio.open_file("config.json", "r") as file: # Compliant
        data = await file.read()
    return data
----

== How to fix it in AnyIO

Use `anyio.open_file()` to handle file operations asynchronously.

=== Code examples

==== Noncompliant code example

[source,python,diff-id=3,diff-type=noncompliant]
----
async def read_config():
    with open("config.json", "r") as file:  # Noncompliant
        data = file.read()
    return data
----

==== Compliant solution

[source,python,diff-id=3,diff-type=compliant]
----
import anyio

async def read_config():
    async with await anyio.open_file("config.json", "r") as file: # Compliant
        data = await file.read()
    return data
----

== Resources

=== Documentation
* Trio - https://trio.readthedocs.io/en/stable/reference-io.html#trio.open_file[trio.open_file() documentation]
* AnyIO - https://anyio.readthedocs.io/en/stable/fileio.html[File I/O in AnyIO]
* Aiofiles - https://github.com/Tinche/aiofiles[aiofiles project]

=== Articles & blog posts
* Python - https://realpython.com/async-io-python/[Async IO in Python]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message
Use an asynchronous file API instead of the synchronous open() in this async function.

=== Highlighting
* Primary locations: the synchronous open() call within an async function
* Secondary locations: the enclosing async function `aync` keyword (message: "this is an asynchronous function")

endif::env-github,rspecator-view[]
