Using `Function.prototype.bind` and arrows functions as attributes will negatively impact performance in React. Each
time the parent is rendered, the function will be re-created and trigger a render of the component causing excessive
renders and more memory use. Wrapping the function in a `useCallback` hook will avoid additional renders. This rule
ignores `Refs`. This rule does not raise findings on DOM nodes since that may require wrapping the DOM in a component.
Still, better performance can be achieved if this rule is respected in DOM nodes too.

== Noncompliant Code Example

[source,javascript]
----
<Component onClick={this._handleClick.bind(this)}></Component>

<Component onClick={() => handleClick()}></Component>
----

== Compliant Solution

[source,javascript]
----

function handleClick() {
    //...
}

<Component onClick={handleClick}></Component>
----

== See

* https://reactjs.org/docs/faq-functions.html[Passing Functions to Components] - React documentation

