Jump statements, such as ``++CHECK++`` and ``++CONTINUE++`` let you change the default flow of program execution, but jump statements that direct the control flow to the original direction are just a waste of keystrokes.

== Noncompliant Code Example

----
  LOOP AT myTable.
    PERFORM form_open USING ...
    CHECK retcode = 0.
    ...
    perform form_close.
    CHECK retcode = 0. "Noncompliant; whatever the result of the check, the loop will continue to the next iteration
  ENDLOOP.
----

== Compliant Solution

----
  LOOP AT myTable.
    PERFORM form_open USING ...
    CHECK retcode = 0.
    ...
    PERFORM form_close.
  ENDLOOP.
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../highlighting.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
