<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when the code cognitive complexity of a function is above a certain threshold.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Cognitive Complexity is a measure of how hard it is to understand the control flow of a unit of code.
Code with high cognitive complexity is hard to read, understand, test, and modify.</p>
</div>
<div class="paragraph">
<p>As a rule of thumb, high cognitive complexity is a sign that the code should be refactored into smaller, easier-to-manage pieces.</p>
</div>
<div class="sect2">
<h3 id="_which_syntax_in_code_does_impact_cognitive_complexity_score">Which syntax in code does impact cognitive complexity score?</h3>
<div class="paragraph">
<p>Here are the core concepts:</p>
</div>
<div class="ulist">
<ul>
<li>
<p><strong>Cognitive complexity is incremented each time the code breaks the normal linear reading flow.</strong><br>
This concerns, for example, loop structures, conditionals, catches, switches, jumps to labels, and conditions mixing multiple operators.</p>
</li>
<li>
<p><strong>Each nesting level increases complexity.</strong><br>
During code reading, the deeper you go through nested layers, the harder it becomes to keep the context in mind.</p>
</li>
<li>
<p><strong>Method calls are free</strong><br>
 A well-picked method name is a summary of multiple lines of code.
 A reader can first explore a high-level view of what the code is performing then go deeper and deeper by looking at called functions content.<br>
<em>Note:</em> This does not apply to recursive calls, those will increment cognitive score.</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>The method of computation is fully detailed in the pdf linked in the resources.</p>
</div>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="paragraph">
<p>Developers spend more time reading and understanding code than writing it. High cognitive complexity slows down changes and increases the cost of maintenance.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Reducing cognitive complexity can be challenging.<br>
Here are a few suggestions:</p>
</div>
<div class="ulist">
<ul>
<li>
<p><strong>Extract complex conditions in a new function.</strong><br>
Mixed operators in condition will increase complexity.
Extracting the condition in a new function with an appropriate name will reduce cognitive load.</p>
</li>
<li>
<p><strong>Break down large functions.</strong><br>
Large functions can be hard to understand and maintain.
If a function is doing too many things, consider breaking it down into smaller, more manageable functions.
Each function should have a single responsibility.</p>
</li>
<li>
<p><strong>Avoid deep nesting by returning early.</strong><br>
To avoid the nesting of conditions, process exceptional cases first and return early.</p>
</li>
<li>
<p><strong>Use null-safe operations (if available in the language).</strong><br>
When available the <code>.?</code> or <code>??</code> operator replaces multiple tests and simplifies the flow.</p>
</li>
</ul>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="paragraph">
<p><strong>Extraction of a complex condition in a new function.</strong></p>
</div>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="paragraph">
<p>The code is using a complex condition and has a cognitive cost of 3.</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-csharp" data-lang="csharp">decimal CalculateFinalPrice(User user, Cart cart)
{
    decimal total = CalculateTotal(cart);
    if (user.HasMembership()               // +1 (if)
        &amp;&amp; user.OrdersCount &gt; 10           // +1 (more than one condition)
        &amp;&amp; user.AccountActive
        &amp;&amp; !user.HasDiscount
        || user.OrdersCount == 1)          // +1 (change of operator in condition)
    {

        total = ApplyDiscount(user, total);
    }
    return total;
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="paragraph">
<p>Even if the cognitive complexity of the whole program did not change, it is easier for a reader to understand the code of the <code>calculateFinalPrice</code> function, which now only has a cognitive cost of 1.</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-csharp" data-lang="csharp">decimal CalculateFinalPrice(User user, Cart cart)
{
    decimal total = CalculateTotal(cart);
    if (IsEligibleForDiscount(user))       // +1 (if)
    {
        total = applyDiscount(user, total);
    }
    return total;
}

bool IsEligibleForDiscount(User user)
{
    return user.HasMembership()
            &amp;&amp; user.OrdersCount &gt; 10       // +1 (more than one condition)
            &amp;&amp; user.AccountActive
            &amp;&amp; !user.HasDiscount
            || user.OrdersCount == 1;      // +1 (change of operator in condition)
}</code></pre>
</div>
</div>
<div class="paragraph">
<p><strong>Break down large functions.</strong></p>
</div>
</div>
<div class="sect3">
<h4 id="_noncompliant_code_example_2">Noncompliant code example</h4>
<div class="paragraph">
<p>For example, consider a function that calculates the total price of a shopping cart, including sales tax and shipping.<br>
<em>Note:</em> The code is simplified here, to illustrate the purpose. Please imagine there is more happening in the <code>foreach</code> loops.</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-csharp" data-lang="csharp">decimal CalculateTotal(Cart cart)
{
    decimal total = 0;
    foreach (Item item in cart.Items) // +1 (foreach)
    {
        total += item.Price;
    }

    // calculateSalesTax
    foreach (Item item in cart.Items) // +1 (foreach)
    {
        total += 0.2m * item.Price;
    }

    //calculateShipping
    total += 5m * cart.Items.Count;

    return total;
}</code></pre>
</div>
</div>
<div class="paragraph">
<p>This function could be refactored into smaller functions:
The complexity is spread over multiple functions and the complex <code>CalculateTotal</code> has now a complexity score of zero.</p>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution_2">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-csharp" data-lang="csharp">decimal CalculateTotal(Cart cart)
{
    decimal total = 0;
    total = CalculateSubtotal(cart, total);
    total += CalculateSalesTax(cart, total);
    total += CalculateShipping(cart, total);
    return total;
}

decimal CalculateSubtotal(Cart cart, decimal total)
{
    foreach (Item item in cart.Items) // +1 (foreach)
    {
        total += item.Price;
    }

    return total;
}

decimal CalculateSalesTax(Cart cart, decimal total)
{
    foreach (Item item in cart.Items)  // +1 (foreach)
    {
        total += 0.2m * item.Price;
    }

    return total;
}

decimal CalculateShipping(Cart cart, decimal total)
{
    total += 5m * cart.Items.Count;
    return total;
}</code></pre>
</div>
</div>
<div class="paragraph">
<p><strong>Avoid deep nesting by returning early.</strong></p>
</div>
</div>
<div class="sect3">
<h4 id="_noncompliant_code_example_3">Noncompliant code example</h4>
<div class="paragraph">
<p>The below code has a cognitive complexity of 6.</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-csharp" data-lang="csharp">decimal CalculateDiscount(decimal price, User user)
{
    if (IsEligibleForDiscount(user))    // +1 ( if )
    {
        if (user.HasMembership())       // +2 ( nested if )
        {
            return price * 0.9m;
        }
        else if (user.OrdersCount == 1) // +1 ( else )
        {
            return price * 0.95m;
        }
        else                            // +1 ( else )
        {
            return price;
        }
    }
    else                                // +1 ( else )
    {
        return price;
    }
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution_3">Compliant solution</h4>
<div class="paragraph">
<p>Checking for the edge case first flattens the <code>if</code> statements and reduces the cognitive complexity to 3.</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-csharp" data-lang="csharp">decimal CalculateDiscount(decimal price, User user)
{
    if (!IsEligibleForDiscount(user)) // +1 ( if )
    {
        return price;
    }

    if (user.HasMembership())         // +1 (  if )
    {
        return price * 0.9m;
    }

    if (user.OrdersCount == 1)        // +1 ( else )
    {
        return price * 0.95m;
    }

    return price;
}</code></pre>
</div>
</div>
<div class="paragraph">
<p><strong>Use the null-conditional operator to access data.</strong></p>
</div>
<div class="paragraph">
<p>In the below code, the cognitive complexity is increased due to the multiple checks required to access the manufacturer&#8217;s name. This can be simplified using the optional chaining operator.</p>
</div>
</div>
<div class="sect3">
<h4 id="_noncompliant_code_example_4">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-csharp" data-lang="csharp">string GetManufacturerName(Product product)
{
    string manufacturerName = null;
    if (product != null &amp;&amp; product.Details != null &amp;&amp;
        product.Details.Manufacturer != null) // +1 (if) +1 (multiple condition)
    {
        manufacturerName = product.Details.Manufacturer.Name;
    }

    if (manufacturerName != null) // +1 (if)
    {
        return manufacturerName;
    }

    return "Unknown";
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution_4">Compliant solution</h4>
<div class="paragraph">
<p>The optional chaining operator will return <code>null</code> if any reference in the chain is <code>null</code>, avoiding multiple checks.
The <code>??</code> operator allows to provide the default value to use.</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-csharp" data-lang="csharp">string GetManufacturerName(Product product)
{
    return product?.Details?.Manufacturer?.Name ?? "Unknown";
}</code></pre>
</div>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_pitfalls">Pitfalls</h3>
<div class="paragraph">
<p>As this code is complex, ensure that you have unit tests that cover the code before refactoring.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Sonar - <a href="https://www.sonarsource.com/docs/CognitiveComplexity.pdf">Cognitive Complexity</a></p>
</li>
</ul>
</div>
</div>
<div class="sect2">
<h3 id="_articles_blog_posts">Articles &amp; blog posts</h3>
<div class="ulist">
<ul>
<li>
<p>Sonar Blog - <a href="https://www.sonarsource.com/blog/5-clean-code-tips-for-reducing-cognitive-complexity/">5 Clean Code Tips for Reducing Cognitive Complexity</a></p>
</li>
</ul>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Refactor this {0} to reduce its Cognitive Complexity from {1} to the {2} allowed.</p>
</div>
<div class="paragraph">
<p>Where {0} is the name of the syntax being analyzed, {1} is the cognitive complexity and {2} is the maximum authorized complexity.</p>
</div>
</div>
<div class="sect2">
<h3 id="_parameters">Parameters</h3>
<div class="sidebarblock">
<div class="content">
<div class="title">threshold</div>
<div class="listingblock">
<div class="content">
<pre>15</pre>
</div>
</div>
<div class="paragraph">
<p>The maximum authorized complexity.</p>
</div>
</div>
</div>
<div class="sidebarblock">
<div class="content">
<div class="title">propertyThreshold</div>
<div class="listingblock">
<div class="content">
<pre>3</pre>
</div>
</div>
<div class="paragraph">
<p>The maximum authorized complexity in a property.</p>
</div>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="ulist">
<ul>
<li>
<p>primary: Method/function name</p>
</li>
<li>
<p>secondary: Keywords &amp; operators that cause increments</p>
</li>
</ul>
</div>
</div>
</div>
</div>