<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when a Rails controller contains multiple render statements for the same format within a single <code>respond_to</code> block.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>In Rails controllers, when you use multiple render calls for the same format within a <code>respond_to</code> block, only the first render statement will execute. All subsequent render calls become unreachable dead code.</p>
</div>
<div class="paragraph">
<p>This happens because Rails processes format blocks sequentially and stops after the first matching format is handled. When you have duplicate format declarations like multiple <code>format.json</code> blocks, the second and subsequent blocks are simply ignored.</p>
</div>
<div class="paragraph">
<p>This creates several problems:</p>
</div>
<div class="ulist">
<ul>
<li>
<p><strong>Dead code</strong>: The unreachable render statements serve no purpose and clutter the codebase</p>
</li>
<li>
<p><strong>Developer confusion</strong>: Other developers might expect all render calls to execute, leading to misunderstandings about the controller&#8217;s behavior</p>
</li>
<li>
<p><strong>Maintenance issues</strong>: Dead code can mislead developers during debugging or feature development</p>
</li>
<li>
<p><strong>Incomplete responses</strong>: The intended data from subsequent render calls will never reach the client</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>The root cause is a misunderstanding of how Rails' <code>respond_to</code> method works. Each format should appear only once per <code>respond_to</code> block.</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="paragraph">
<p>This issue leads to incomplete API responses where only partial data is returned to clients. It creates dead code that confuses developers and makes the codebase harder to maintain. While not a security risk, it can cause application logic errors and unexpected behavior in client applications that depend on the missing data.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it_in_rails">How to fix it in Rails</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Combine multiple data sources into a single response hash. This ensures all intended data is included in the response and eliminates dead code.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-ruby" data-lang="ruby">def calculate_quote
  @moulding = Moulding.find(params[:id])
  @material_costs = MaterialCost.all

  respond_to do |format|
    format.json { render json: @moulding }
    format.json { render json: @material_costs } # Noncompliant: unreachable code
  end
end</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-ruby" data-lang="ruby">def calculate_quote
  @moulding = Moulding.find(params[:id])
  @material_costs = MaterialCost.all

  respond_to do |format|
    format.json { render json: {
      moulding: @moulding,
      material_costs: @material_costs
    }}
  end
end</code></pre>
</div>
</div>
<div class="paragraph">
<p>Use conditional logic within a single format block to determine what to render based on parameters or application state.</p>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_code_examples_2">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example_2">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-ruby" data-lang="ruby">def show_data
  respond_to do |format|
    format.json { render json: @users }
    format.json { render json: @posts } # Noncompliant: unreachable code
  end
end</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution_2">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-ruby" data-lang="ruby">def show_data
  respond_to do |format|
    format.json do
      if params[:type] == 'users'
        render json: @users
      else
        render json: @posts
      end
    end
  end
end</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Rails respond_to method - <a href="https://api.rubyonrails.org/classes/ActionController/MimeResponds.html#method-i-respond_to">Official Rails documentation for the respond_to method</a></p>
</li>
<li>
<p>Rails render method - <a href="https://api.rubyonrails.org/classes/ActionController/Renderer.html#method-i-render">Official Rails documentation for the render method</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>