<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Invoking a method designed to return a string representation of an object which is already a string is a waste of keystrokes. This redundant construction may be optimized by the compiler, but will be confusing in the meantime.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-text" data-lang="text">String message = "hello world";
System.out.println(message.toString()); // Noncompliant;</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-text" data-lang="text">String message = "hello world";
System.out.println(message);</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>there&#8217;s no need to call "toString()" on a string literal.</p>
</div>
<div class="paragraph">
<p>"XXX" is already a string, there&#8217;s no need to call "toString()" on it.</p>
</div>
<div class="paragraph">
<p>"XXX" returns a string, there&#8217;s no need to call "toString()".</p>
</div>
<div class="paragraph">
<p>"XXX" is an array of string, there&#8217;s no need to call "toString()".</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_21_nov_2024_164800_alban_auzeill_wrote">on 21 Nov 2024, 16:48:00 Alban Auzeill wrote:</h3>
<div class="paragraph">
<p>[test-code-support-investigation-for-java] Decision for scope: Main &#8594; All.</p>
</div>
</div>
<div class="sect2">
<h3 id="_is_duplicated_by_s2135">is duplicated by: <a data-rspec-id="S2135" class="rspec-auto-link">S2135</a></h3>

</div>
</div>
</div>