<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Calling the <code>to_string</code> method on a value that is already of type <code>String</code> in Rust is redundant and unnecessary. This redundant call does not change the value and only adds unnecessary complexity to the code. It can also be confusing to readers and may lead to misunderstandings about the code&#8217;s intent.</p>
</div>
<div class="paragraph">
<p>If the intent was to clone the string, it is better to use the <code>clone</code> method instead.</p>
</div>
<div class="paragraph">
<p>Avoiding such redundant calls helps keep the code clean, concise, and more readable.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-rust" data-lang="rust">let message = String::from("hello world");
println!("{}", message.to_string()); // Noncompliant: 'message' is already a String</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-rust" data-lang="rust">let message = String::from("hello world");
println!("{}", message);

// or

let cloned_message = message.clone(); // Use 'clone' if you need a duplicate
println!("{}", cloned_message);</code></pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Clippy Lints - <a href="https://rust-lang.github.io/rust-clippy/master/index.html#string_to_string" class="bare">https://rust-lang.github.io/rust-clippy/master/index.html#string_to_string</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>