<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This is an issue when calling model instance methods or accessing attributes directly on ActiveRecord::Relation objects returned by query methods like <code>where</code>, <code>select</code>, <code>joins</code>, or <code>includes</code>.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>ActiveRecord query methods like <code>where</code>, <code>select</code>, <code>joins</code>, and <code>includes</code> return <code>ActiveRecord::Relation</code> objects, not individual model instances or scalar values. These relation objects represent a database query that hasn&#8217;t been executed yet.</p>
</div>
<div class="paragraph">
<p>When you try to call model instance methods (like <code>name</code>, <code>email</code>, or custom methods) directly on a relation object, Ruby raises a <code>NoMethodError</code> because these methods don&#8217;t exist on the relation class. The relation object is designed to build and execute queries, not to provide access to model data.</p>
</div>
<div class="paragraph">
<p>This is a common source of confusion because:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>Query methods look like they should return the data you want</p>
</li>
<li>
<p>The variable names often suggest a single instance (like <code>user = User.where(id: 1)</code>)</p>
</li>
<li>
<p>Some ActiveRecord methods do return instances directly (like <code>find</code> or <code>create</code>)</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>The error typically manifests as: <code>NoMethodError: undefined method 'attribute_name' for #&lt;ActiveRecord::Relation:0x&#8230;&#8203;&gt;</code></p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="paragraph">
<p>This issue causes runtime exceptions that crash the application when the problematic code is executed. The <code>NoMethodError</code> will halt request processing and potentially expose internal application structure through error messages. In production environments, this leads to poor user experience and may require immediate fixes to restore functionality.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it_in_ruby_on_rails">How to fix it in Ruby on Rails</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Use query termination methods like <code>first</code>, <code>last</code>, or <code>find_by</code> to get a single model instance, then access its attributes.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-ruby" data-lang="ruby"># Calling instance method on relation
user = User.where(id: current_user.id)
print user.name  # Noncompliant</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-ruby" data-lang="ruby"># Get the actual instance first
user = User.where(id: current_user.id).first
if user
  print user.name  # Works on model instance
end</code></pre>
</div>
</div>
<div class="paragraph">
<p>Use <code>find</code> or <code>find_by</code> instead of <code>where().limit(1)</code> when you need a single model instance.</p>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_code_examples_2">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example_2">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-ruby" data-lang="ruby"># limit(1) still returns a relation
person = Person.where(id: 34).limit(1)
puts person.last_name  # Noncompliant</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution_2">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-ruby" data-lang="ruby"># Use find_by to get an actual instance
person = Person.find_by(id: 34)
puts person.last_name if person</code></pre>
</div>
</div>
<div class="paragraph">
<p>Use <code>pluck</code> to extract scalar values directly from the database instead of returning relations.</p>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_code_examples_3">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example_3">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-ruby" data-lang="ruby"># Returns relation instead of email string
def self.get_user_email(user_id)
  User.where(id: user_id).select(:email)  # Noncompliant
end</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution_3">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-ruby" data-lang="ruby"># Extract the actual email value
def self.get_user_email(user_id)
  User.where(id: user_id).pluck(:email).first
end</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>ActiveRecord Query Interface Guide - <a href="https://guides.rubyonrails.org/active_record_querying.html">Official Rails guide covering ActiveRecord query methods and how to properly terminate queries</a></p>
</li>
<li>
<p>ActiveRecord::Relation API Documentation - <a href="https://api.rubyonrails.org/classes/ActiveRecord/Relation.html">Complete API reference for ActiveRecord::Relation class and its methods</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>