There is no need to declare a type parameter when naming a type constraint is not required. Using wildcards makes it easier to read.


== Noncompliant Code Example

----
<T extends MyClass> void foo(List<T> list) {  // Noncompliant, T is used only once
  for (MyClass myObj : list) {
    doSomething(myObj);
  }
}
----


== Compliant Solution

----
void foo(List<? extends MyClass> list) {
  for (MyClass myObj : list) {
    doSomething(myObj);
  }
}
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

include::highlighting.adoc[]

endif::env-github,rspecator-view[]
