<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>PHPUnit provides helper functions and annotations to verify that a given block of code throws an exception and to assert different properties of that exception. The provided helper functions are:</p>
</div>
<div class="ulist">
<ul>
<li>
<p><code>expectException()</code></p>
</li>
<li>
<p><code>expectExceptionCode()</code></p>
</li>
<li>
<p><code>expectExceptionMessage()</code></p>
</li>
<li>
<p><code>expectExceptionMessageRegExp()</code></p>
</li>
</ul>
</div>
<div class="paragraph">
<p>This check raises an issue when the throw of an exception is verified using a <em>try-catch</em> approach instead of relying on the provided helper functions.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-php" data-lang="php">public function testA()
{
    try {
        doSomething();
        $this-&gt;fail("Assertion should have been thrown");
    } catch (MyException $e) {
        assertEquals("Exception message", $e-&gt;getMessage());
    }
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-php" data-lang="php">public function testB()
{
    $this-&gt;expectException(MyException::class);
    $this-&gt;expectExceptionMessage("Exception message");

    doSomething();
}</code></pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="paragraph">
<p>thePHP.cc: <a href="https://thephp.cc/articles/questioning-phpunit-best-practices">Questioning PHPUnit Best Practice</a></p>
</div>
</div>
</div>