== Why is this an issue?

In an enumeration, it is possible to have several enumerators with the same value. This can be useful in many circumstances. But if that happens without an explicit choice from the developer, it can also lead to errors that are hard to diagnose.

When an enumerator has no explicit value, its value will be one more than the preceding enumerator. This rule raises an issue if this value inadvertently clashes with another enumerator value.


=== Noncompliant code example

[source,cpp]
----
enum color { red = 3, blue, green, yellow = 5 }; // Noncompliant, both green and yellow = 5
----


=== Compliant solution

[source,cpp]
----
enum color { red = 3, blue, green, yellow }; // Compliant, different values
enum color { red = 3, blue, green = 5, yellow = 5 }; // Compliant, values explicitly identical
enum color { red = 3, blue, green, yellow = green }; // Compliant, values explicitly identical
----


== Resources

* MISRA {cpp}:2023, 11.6.3 - Within an enumerator list, the value of an implicitly-specified 
enumeration constant shall be unique
* MISRA C:2004, 9.3 - In an enumerator list, the "=" construct shall not be used to explicitly initialize members other than the first, unless all items are explicitly initialized.
* MISRA {cpp}:2008, 8-5-3 - In an enumerator list, the = construct shall not be used to explicitly initialize members other than the first, unless all items are explicitly initialized.

