== Why is this an issue?

You're allowed to make function calls with positionally-identified parameters or with named parameters. In long parameter lists, explicitly naming your parameters both eliminates the risk of swapping parameter values and makes the call clearer to maintainers.


=== Noncompliant code example

With the default threshold of 3:

[source,python]
----
def coordinateSurpriseParty(self, hostName, guestOfHonor, caterer, invitees, peopleNotToInvite):
  # ...

host="John"
guestOfHonor="Fred"
caterer="Barry"
invitees="Thea, Will, Mark, Mary"
peopleToExclude="Susan, Alberta, Frank, Larry"

coordinateSurpriseParty(caterer, host, guestOfHonor, peopleToExclude, invitees)  # Noncompliant; this party will be a train wreck!
----


=== Compliant solution

[source,python]
----
def coordinateSurpriseParty(self, hostName, guestOfHonor, caterer, invitees, peopleNotToInvite):
  # ...

host="John"
guestOfHonor="Fred"
caterer="Barry"
invitees="Thea, Will, Mark, Mary"
peopleToExclude="Susan, Alberta, Frank, Larry"

coordinateSurpriseParty(caterer=caterer, hostName=host, geustOfHonor=guestOfHonor, peopleNotToInvite=peopleToExclude, invitees=invitees)
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

include::parameters.adoc[]

endif::env-github,rspecator-view[]
