<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>If the lifetime of the arguments passed to a lambda function is longer than the lifetime of the lambda itself, these arguments can be passed by reference.</p>
</div>
<div class="paragraph">
<p>Doing so avoids copying potentially big objects and it should be preferred over using copy capture.</p>
</div>
<div class="paragraph">
<p>This rule reports an issue if a lambda passed into an algorithm that uses it locally (all algorithms in headers <code>&lt;algorithm&gt;</code> and <code>&lt;numeric&gt;</code>) captures some values.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">using Vec = vector&lt;int&gt;;

Vec applyPermutation(const Vec&amp; v, const Vec&amp; permutation) {
    assert(v.size() == permutation.size());

    const auto n = v.size();
    Vec result(n);

    // Noncompliant: this will copy the entire v vector for each iteration, resulting in n^2 operations
    transform(permutation.begin(), permutation.end(), back_inserter(result),
      [v](int position){ return v[position]; });

     return result;
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">using Vec = vector&lt;int&gt;;

Vec applyPermutation(const Vec&amp; v, const Vec&amp; permutation) {
    assert(v.size() == permutation.size());

    const auto n = v.size();
    Vec result(n);

    // Compliant: this will NOT copy the entire v vector for each iteration, resulting in n operations
    transform(permutation.begin(), permutation.end(), back_inserter(result),
      [&amp;v](int position){ return v[position]; });

     return result;
}</code></pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p>C++ Core Guidelines - <a href="https://github.com/isocpp/CppCoreGuidelines/blob/e49158a/CppCoreGuidelines.md#f52-prefer-capturing-by-reference-in-lambdas-that-will-be-used-locally-including-passed-to-algorithms">F.52: Prefer capturing by reference in lambdas that will be used locally, including passed to algorithms</a></p>
</li>
</ul>
</div>
</div>
</div>