<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when a GINQ nested query in a select clause does not include a <code>limit 1</code> clause.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>When using nested GINQ queries within select clauses, the subquery may return multiple values. If more than one value is returned, a <code>TooManyValuesException</code> will be thrown at runtime, causing the application to fail.</p>
</div>
<div class="paragraph">
<p>The select clause expects a single value for each row in the result set. When a nested query returns multiple values without proper limiting, the GINQ engine cannot determine which value to use, leading to an exception.</p>
</div>
<div class="paragraph">
<p>This issue typically occurs when:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>The subquery&#8217;s <code>where</code> condition matches multiple records</p>
</li>
<li>
<p>The data source contains duplicate values that satisfy the query conditions</p>
</li>
<li>
<p>The developer assumes the subquery will always return at most one result</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>Using <code>limit 1</code> explicitly restricts the subquery to return only the first matching result, preventing the exception and making the query&#8217;s behavior predictable.</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="paragraph">
<p>The application will throw a <code>TooManyValuesException</code> at runtime when the nested query returns multiple values. This can cause:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>Application crashes or unexpected termination</p>
</li>
<li>
<p>Data processing failures in batch operations</p>
</li>
<li>
<p>Poor user experience due to unhandled exceptions</p>
</li>
<li>
<p>Difficulty in debugging since the error only occurs when specific data conditions are met</p>
</li>
</ul>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Add <code>limit 1</code> to the nested query in the select clause to ensure only one value is returned.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-groovy" data-lang="groovy">GQ {
    from n in [1,2,3]
    select (from m in [2,3,4]
            where m == n
            select m) // Noncompliant
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-groovy" data-lang="groovy">GQ {
    from n in [1,2,3]
    select (from m in [2,3,4]
            where m == n
            limit 1
            select m)
}</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>GINQ Documentation - Nested GINQ in select clause - <a href="https://groovy-lang.org/using-ginq.html#_nested_ginq_in_select_clause">Official Groovy documentation explaining nested GINQ queries and the recommendation to use limit 1</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>