== How to fix it in Android

=== Code examples

==== Noncompliant code example

[source,java,diff-id=11,diff-type=noncompliant]
----
public class ExampleActivity extends AppCompatActivity {

    private EditText passwordEditText;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        EdgeToEdge.enable(this);
        setContentView(R.layout.activity_xml_layout);

        SharedPreferences sharedPreferences = getSharedPreferences("example", Context.MODE_PRIVATE);

        passwordEditText = findViewById(R.id.passwordEditText);
        Button savePasswordButton = findViewById(R.id.savePasswordButton);

        savePasswordButton.setOnClickListener(v -> {
            String password = passwordEditText.getText().toString();
            if (!password.isEmpty()) {
                SharedPreferences.Editor editor = sharedPreferences.edit();
                editor.putString("Password", password);
                editor.apply();
            }
        });
    }
}
----

==== Compliant solution

include::../../common/compliant.adoc[]

[source,java,diff-id=11,diff-type=compliant]
----
public class ExampleActivity extends AppCompatActivity {

    private EditText passwordEditText;
    private CredentialManagerCallback<CreateCredentialResponse, CreateCredentialException> callback;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        EdgeToEdge.enable(this);
        setContentView(R.layout.activity_xml_layout);

        SharedPreferences sharedPreferences = getSharedPreferences("example", Context.MODE_PRIVATE);

        passwordEditText = findViewById(R.id.passwordEditText);
        Button savePasswordButton = findViewById(R.id.savePasswordButton);

        savePasswordButton.setOnClickListener(v -> {
            String password = passwordEditText.getText().toString();
            if (!password.isEmpty()) {
                CredentialManager credentialManager = CredentialManager.create(this);
                CreatePasswordRequest request = new CreatePasswordRequest(
                        "example",
                        password
                );
                credentialManager.createCredentialAsync (this, request,null, Runnable::run, callback);
            }
        });
    }
}
----