<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>The type hint <code>Any</code> represents any possible Python type, on which all operations are possible. This conveys the information that all operations could be possible on a value annotated with <code>Any</code>, making the value dynamically-typed.</p>
</div>
<div class="paragraph">
<p><code>Any</code> as a type hint provides no information about the expected type of the variable or parameter, essentially losing the benefits of type hinting. This can make the code less clear and harder to understand and will reduce the code insight capabilities of IDEs and static analysis tools.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Replace this use of <code>Any</code> with a more specific type hint.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">def foo(arg: Any) -&gt; Any:
    if isinstance(arg, int):
        return arg + 1
    else:
        return arg.upper()</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">def foo(arg: int | str) -&gt; int | str:
    if isinstance(arg, int):
        return arg + 1
    else:
        return arg.upper()</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p><a href="https://docs.python.org/3/library/typing.html#the-any-type">Python3 documentation on Any</a></p>
</li>
<li>
<p><a href="https://mypy.readthedocs.io/en/stable/kinds_of_types.html#the-any-type">Mypy documentation on the Any type</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>