== Why is this an issue?

include::../common/why.adoc[]

=== What is the potential impact?

include::../common/potential_impact.adoc[]

== How to fix it

include::../common/how.adoc[]

=== Code examples

==== Noncompliant code example

[source,javascript,diff-id=1,diff-type=noncompliant]
----
import { _parseWith } from './node_modules/foo/helpers'
----

==== Compliant solution

[source,javascript,diff-id=1,diff-type=compliant]
----
import { parse } from 'foo'
----

//== Resources
//=== Documentation
//=== Articles & blog posts
//=== Conference presentations
//=== Standards
//=== External coding guidelines
//=== Benchmarks
