== Why is this an issue?

Caller information attributes (``++CallerFilePathAttribute++``, ``++CallerLineNumberAttribute++``, ``++CallerMemberNameAttribute++``, and ``++CallerArgumentExpressionAttribute++``) provide a way to get information about the caller of a method through optional parameters. But they only work right if their values aren't provided explicitly. So if you define a method with caller info attributes in the middle of the parameter list, you put your callers in a bad position: they are forced to use named arguments if they want to use the method properly.


=== Noncompliant code example

[source,csharp]
----
void TraceMessage([CallerMemberName] string memberName = "",
  [CallerFilePath] string filePath = "",
  [CallerLineNumber] int lineNumber = 0,
  string message = null)  // Noncompliant
{
  /* ... */
}
----


=== Compliant solution

[source,csharp]
----
void TraceMessage(string message = null,
  [CallerMemberName] string memberName = "",
  [CallerFilePath] string filePath = "",
  [CallerLineNumber] int lineNumber = 0) 
{
  /* ... */
}
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Move "xxx" to the end of the parameter list.


'''
== Comments And Links
(visible only on this page)

=== on 18 Nov 2015, 10:01:25 Tamas Vajk wrote:
\[~ann.campbell.2] I modified the sample code: there can't be any parameter without a default value after parameters with default values. (So I added the ``++null++`` default value to the ``++message++``)


And also modified the last sentence. It is possible to not specify all parameters in a call, but then you force the callers of this method to use named arguments. 

=== on 18 Nov 2015, 19:55:28 Ann Campbell wrote:
Okay, thanks [~tamas.vajk]

endif::env-github,rspecator-view[]
