This is an issue when Go import statements are not organized according to the standard three-group structure: standard library imports first, third-party imports second, and local project imports third, with blank lines separating each group.

== Why is this an issue?

Go has established community conventions for organizing imports that improve code readability and make dependencies immediately clear to developers. The standard three-group structure separates imports by their source and purpose:

**Standard library imports** come first because they are part of Go's core functionality and are always available. These imports use simple package names without dots or slashes (like `fmt`, `os`, `net/http`).

**Third-party imports** come second, separated by a blank line. These are external dependencies from other projects and typically contain domain names or repository paths (like `github.com/user/repo`).

**Local project imports** come last, also separated by a blank line. These are imports from within the same project or module.

This organization provides several benefits:

- **Immediate clarity** about dependencies: Developers can quickly see what external dependencies a file uses
- **Consistent structure** across the codebase makes code easier to navigate
- **Reduced merge conflicts** when multiple developers add imports
- **Tool compatibility** with standard Go formatting tools like `goimports`
- **Community alignment** with widely accepted Go conventions

When imports are mixed together without proper grouping, it becomes harder to understand the dependency structure of the code. This can lead to confusion about which packages are external dependencies versus internal modules, making code maintenance more difficult.

=== What is the potential impact?

Improperly organized imports reduce code readability and make it harder for developers to quickly understand a file's dependencies. This can slow down code reviews, make maintenance more difficult, and create inconsistency across the codebase. While not causing runtime issues, poor import organization violates Go community standards and can make the code appear unprofessional.

== How to fix it

Organize imports into three groups separated by blank lines: standard library imports first, third-party imports second, and local project imports third. Within each group, sort imports alphabetically.

=== Code examples

==== Noncompliant code example

[source,go,diff-id=1,diff-type=noncompliant]
----
package main

import "os"
import "fmt"
import "strings"
import "net/http"

func main() {
    fmt.Println("Hello")
    _ = http.StatusOK
    _ = os.Args
    _ = strings.Join
}
----

==== Compliant solution

[source,go,diff-id=1,diff-type=compliant]
----
package main

import (
    "fmt"
    "net/http"
    "os"
    "strings"
)

func main() {
    fmt.Println("Hello")
    _ = http.StatusOK
    _ = os.Args
    _ = strings.Join
}
----

== Resources

=== Documentation

 * Go Code Review Comments - Imports - https://github.com/golang/go/wiki/CodeReviewComments#imports[Official Go documentation on import organization and formatting conventions]

 * Effective Go - Package names - https://golang.org/doc/effective_go#package-names[Guidelines for Go package naming and import practices]

 * goimports tool documentation - https://pkg.go.dev/golang.org/x/tools/cmd/goimports[Documentation for the goimports tool that automatically organizes imports according to Go conventions]

=== Standards

 * Go Community Standards - https://golang.org/doc/effective_go[Established Go community conventions for code organization and formatting]

=== Related rules

 * RSPEC-XXXX - https://rules.sonarsource.com/go/RSPEC-XXXX[Remove unused imports to keep code clean]
