If you only want one instance that matches certain criteria out of a collection, it's far more efficient to grab the first matching item than it is to fully filter the collection for your criteria and then only use a single value.

== Noncompliant Code Example

----
let one = arr.filter { $0.containsString("yo") }.first  // Noncompliant
----

== Compliant Solution

----
let one = arr.first(where: { $0.containsString("yo") })
----
