<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This is an issue when a variable is declared with an explicit type that can be clearly inferred from a typed expression on the right-hand side.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Go&#8217;s type inference system can automatically determine variable types from their initialization values. When the right-hand side of a declaration contains a typed expression, explicitly declaring the type creates unnecessary visual noise and reduces code readability.</p>
</div>
<div class="paragraph">
<p>The Go language encourages using the short variable declaration syntax (<code>:=</code>) when the type can be inferred. This makes code more concise and follows Go&#8217;s design philosophy of simplicity and clarity.</p>
</div>
<div class="paragraph">
<p>Redundant type declarations can make code harder to read and maintain, especially in larger codebases where reducing visual clutter improves comprehension.</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="paragraph">
<p>This issue affects code readability and maintainability. While it doesn&#8217;t cause functional problems, it goes against Go&#8217;s idiomatic style and can make code unnecessarily verbose. Following Go&#8217;s conventions for type inference helps create cleaner, more readable code that aligns with community standards.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Replace the explicit type declaration with the short variable declaration syntax (<code>:=</code>) when the type can be clearly inferred from the right-hand side.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-go" data-lang="go">var i int = 42
var j int = i // Noncompliant: type is redundant</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-go" data-lang="go">var i int = 42
j := i // Type is inferred as int</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Go Tour - Type Inference - <a href="https://go.dev/tour/basics/14">Official Go documentation explaining type inference and variable declarations</a></p>
</li>
<li>
<p>Go Tour - Short Variable Declarations - <a href="https://go.dev/tour/basics/10">Official Go documentation on short variable declaration syntax</a></p>
</li>
<li>
<p>Effective Go - Variables - <a href="https://go.dev/doc/effective_go#variables">Best practices for variable declarations in Go</a></p>
</li>
</ul>
</div>
</div>
<div class="sect2">
<h3 id="_related_rules">Related rules</h3>
<div class="ulist">
<ul>
<li>
<p><a data-rspec-id="S3257" class="rspec-auto-link">S3257</a> - <a href="https://rules.sonarsource.com/javascript/RSPEC-3257/">JavaScript rule for avoiding redundant type declarations</a></p>
</li>
<li>
<p><a data-rspec-id="S6212" class="rspec-auto-link">S6212</a> - <a href="https://rules.sonarsource.com/java/RSPEC-6212/">Java rule for using var when type is obvious</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>