== Why is this an issue?

Quoted from the Flex documentation :

____
When you define an Array variable in ActionScript, you specify Array as the data type of the variable. However, you cannot specify the data type of the elements of the Array.


To allow the Flex MXML compiler to perform type checking on Array elements, you can use the [ArrayElementType] metadata tag to specify the allowed data type of the Array elements.

____


=== Noncompliant code example

[source,flex]
----
public var newStringProperty:Array;
public var newNumberProperty:Array;
----


=== Compliant solution

[source,flex]
----
[ArrayElementType("String")] 
public var newStringProperty:Array;

[ArrayElementType("Number")] 
public var newNumberProperty:Array;
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Define the element type for this 'xxxxx' array


'''
== Comments And Links
(visible only on this page)

=== on 22 Nov 2013, 13:23:13 Freddy Mallet wrote:
Is implemented by \https://jira.codehaus.org/browse/SONARPLUGINS-3282

endif::env-github,rspecator-view[]
