<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>ViewBag and ViewData dictionaries enable controllers to pass data to their views as weakly typed collections. Reading the provided values is dynamically resolved at runtime without any compile-time checking. This can lead to unexpected behavior, since reading a missing value does not produce an exception.</p>
</div>
<div class="paragraph">
<p>Controllers should pass data to their views via a strongly typed view model class.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-vbnet" data-lang="vbnet">Public Class HomeController
    Inherits Controller

    Public Function Article() As ActionResult
        ViewBag.Title = "Title" ' Noncompliant, model should be used
        ViewData("Text") = "Text" ' Noncompliant, model should be used
        Return View()
    End Function

End Class</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-vbnet" data-lang="vbnet">Public Class ArticleModel

    Public Property Title As String
    Public Property Text As String

End Class

Public Class HomeController
    Inherits Controller

    Public Function Article() As ActionResult
        Dim Model As New ArticleModel With {.Title = "Title", .Text = "Text"}
        Return View(Model)
    End Function

End Class</code></pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="paragraph">
<p><a href="https://docs.microsoft.com/en-us/aspnet/core/mvc/views/overview?view=aspnetcore-3.0#weakly-typed-data-viewdata-viewdata-attribute-and-viewbag">Views in ASP.NET Core MVC - Weakly typed data</a></p>
</div>
<hr>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Replace all uses of "[ViewBag|ViewData]" with model.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>Primary: The first edition of ViewBag/ViewData in a method</p>
</div>
<div class="paragraph">
<p>Secondary:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>location: Every other occurence of ViewBag/ViewData.</p>
</li>
<li>
<p>message: Replace this use of "[ViewBag/ViewData]".</p>
</li>
</ul>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_7_apr_2020_102830_pavel_mikula_wrote">on 7 Apr 2020, 10:28:30 Pavel Mikula wrote:</h3>
<div class="paragraph">
<p>Ref https://github.com/SonarSource/sonar-dotnet/issues/2602</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_30_apr_2020_133918_nicolas_harraudeau_wrote">on 30 Apr 2020, 13:39:18 Nicolas Harraudeau wrote:</h3>
<div class="paragraph">
<p>Note from [~pavel.mikula] about ViewDataAttribute:</p>
</div>
<div class="paragraph">
<p>It can be read from implicit models with strong type checking (good) OR from ViewData dictionary (bad). So it&#8217;s 50% OK and we cannot predict this on C# side of the code. So we should not raise it until we run rules on CSHTML Razor templates.</p>
</div>
</div>
</div>
</div>