include::../description.adoc[]

== Noncompliant Code Example

----
const long b = 0l;      // Noncompliant
----

== Compliant Solution

----
const long b = 0L;
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
