== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,csharp]
----
const long b = 0l;      // Noncompliant
----

=== Compliant solution

[source,csharp]
----
const long b = 0L;
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
