== Why is this an issue?

Explicitly declaring a function's return data type makes it easier to use the function correctly. 

=== Noncompliant code example

[source,vb6]
----
Function BinarySearch(. . .) 
    [. . .]
End Function
----

=== Compliant solution

[source,vb6]
----
Function BinarySearch(. . .) As Boolean
    [. . .]
End Function
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Specify the return data type for XXX


'''
== Comments And Links
(visible only on this page)

=== on 30 Mar 2015, 15:10:18 Evgeny Mandrikov wrote:
\[~ann.campbell.2] I added VB6 as "targeted language" for parent task.

=== on 30 Mar 2015, 18:53:36 Ann Campbell wrote:
Thanks [~evgeny.mandrikov]

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
