== Why is this an issue?

Explicitly declaring a function's return data type makes it easier to use the function correctly. 

=== Noncompliant code example

[source,vb6]
----
Function BinarySearch(. . .) 
    [. . .]
End Function
----

=== Compliant solution

[source,vb6]
----
Function BinarySearch(. . .) As Boolean
    [. . .]
End Function
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
