== Why is this an issue?

There is no good reason to keep an empty and therefore valueless section. Such sections should be removed.


=== Noncompliant code example

[source,cobol]
----
FIRST SECTION.
  MOVE A TO B.

SECOND SECTION.  *> Noncompliant; empty

THIRD SECTION.

someParagraph.
  DISPLAY B.
----


=== Compliant solution

[source,cobol]
----
FIRST SECTION.
  MOVE A TO B.

THIRD SECTION.

someParagraph.
  DISPLAY B.
----



ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Remove this useless section declaration.


'''
== Comments And Links
(visible only on this page)

=== on 9 Oct 2013, 00:13:13 Ann Campbell wrote:
Again, should I be adding the compliant version?

=== on 20 Nov 2013, 10:18:37 Dinesh Bolkensteyn wrote:
I think you definitely should [~ann.campbell.2].

=== on 20 Nov 2013, 10:19:36 Dinesh Bolkensteyn wrote:
Everything must be challengable and challenged, and asking a lot of questions slows every one down, so it's better to just go ahead and do the evolutions we believe are right.

=== on 20 Nov 2013, 10:23:36 Dinesh Bolkensteyn wrote:
As per our guidelines (\http://confluence.internal.sonarsource.com/display/DEV/Guidelines+-+Design), I removed the "isn't" contraction.

endif::env-github,rspecator-view[]
