There is no good reason to keep an empty and therefore valueless section. Such sections should be removed.


== Noncompliant Code Example

----
FIRST SECTION.
  MOVE A TO B.

SECOND SECTION.  *> Noncompliant; empty

THIRD SECTION.

someParagraph.
  DISPLAY B.
----


== Compliant Solution

----
FIRST SECTION.
  MOVE A TO B.

THIRD SECTION.

someParagraph.
  DISPLAY B.
----



ifdef::env-github,rspecator-view[]
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
