String data types, such as ``++VARCHAR2++`` or ``++NVARCHAR2++`` require a size constraint. Omitting the size results in the exception ``++PLS-00215: String length constraints must be in range (1 .. 32767)++`` being raised.


== Noncompliant Code Example

----
DECLARE
  foo VARCHAR2; -- Noncompliant - raises PLS-00215
BEGIN
  NULL;
END;
/
----


== Compliant Solution

----
DECLARE
  foo VARCHAR2(42); -- Compliant
BEGIN
  NULL;
END;
/
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

endif::env-github,rspecator-view[]
