<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This is an issue when an import statement uses an alternative repository path instead of the canonical module path declared in the module&#8217;s go.mod file.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Go modules require import paths to match the canonical module path declared in the module&#8217;s <code>go.mod</code> file. When a module is available through multiple repository URLs, only one path is considered canonical.</p>
</div>
<div class="paragraph">
<p>Using non-canonical import paths can cause several problems:</p>
</div>
<div class="ulist">
<ul>
<li>
<p><strong>Module resolution failures</strong>: The Go toolchain may fail to resolve dependencies correctly when non-canonical paths are used.</p>
</li>
<li>
<p><strong>Inconsistent dependency management</strong>: Different parts of your codebase might reference the same module through different paths, leading to confusion and potential version conflicts.</p>
</li>
<li>
<p><strong>Migration issues</strong>: When migrating existing code to Go modules, non-canonical import paths must be updated to match the canonical module path.</p>
</li>
<li>
<p><strong>Tooling incompatibility</strong>: Go development tools expect canonical paths for proper module analysis and dependency tracking.</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>For example, both <code>github.com/golang/lint</code> and <code>golang.org/x/lint</code> resolve to the same repository, but only <code>golang.org/x/lint</code> is the canonical module path. Using the non-canonical GitHub path will cause issues in module-aware builds.</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="paragraph">
<p>Using non-canonical import paths can lead to build failures, dependency resolution errors, and inconsistent module management. This can prevent successful compilation and deployment of Go applications using modules.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Replace the non-canonical import path with the canonical module path as declared in the module&#8217;s go.mod file.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-go" data-lang="go">import "github.com/golang/lint" // Noncompliant</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-go" data-lang="go">import "golang.org/x/lint"</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Go Modules: Migrating to Go Modules - <a href="https://go.dev/blog/migrating-to-go-modules">Official Go blog post explaining module migration and canonical import paths</a></p>
</li>
<li>
<p>Go Command: Remote Import Paths - <a href="https://go.dev/cmd/go/#hdr-Remote_import_paths">Documentation on how Go resolves remote import paths</a></p>
</li>
<li>
<p>Go Modules Reference - <a href="https://go.dev/ref/mod">Complete reference for Go modules including module paths and versioning</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>