This rule raises an issue when the same property name appears multiple times in a Grails domain class `constraints` block.

== Why is this an issue?

In Grails domain classes, the `constraints` block defines validation rules and database mapping constraints for properties. While Grails allows duplicate constraint names syntactically, this practice creates several problems.

When you define the same property constraint multiple times, only the last definition takes effect. This behavior can be confusing and error-prone, especially in larger constraint blocks where duplicates might not be immediately obvious.

Duplicate constraints often indicate:

* Copy-paste errors during development
* Incomplete refactoring where old constraints weren't removed
* Lack of awareness about existing constraints
* Team collaboration issues where multiple developers modify the same constraints

This pattern makes the code harder to maintain and understand. Future developers (including yourself) may waste time trying to understand why certain constraints don't seem to work, not realizing they've been overridden by later duplicates.

=== What is the potential impact?

Duplicate constraints can lead to unexpected validation behavior where intended constraints are silently overridden. This may result in data integrity issues if stricter constraints are replaced by more lenient ones, or user experience problems if validation rules don't work as expected. The confusion caused by duplicates can also slow down development and debugging efforts.

== How to fix it

Remove the duplicate constraint definition and consolidate all constraints for each property into a single entry. If you need to modify existing constraints, update the original definition rather than adding a new one.

=== Code examples

==== Noncompliant code example

[source,groovy,diff-id=1,diff-type=noncompliant]
----
class Person {
    String firstName
    String lastName
    
    static constraints = {
        firstName nullable: true
        lastName nullable: true, maxSize: 30
        firstName nullable: false // Noncompliant
    }
}
----

==== Compliant solution

[source,groovy,diff-id=1,diff-type=compliant]
----
class Person {
    String firstName
    String lastName
    
    static constraints = {
        firstName nullable: false
        lastName nullable: true, maxSize: 30
    }
}
----

== Resources

=== Documentation

 * Grails Validation Documentation - https://docs.grails.org/latest/guide/validation.html[Official Grails documentation on validation and constraints]

 * Grails Domain Classes Guide - https://docs.grails.org/latest/guide/GORM.html#domainClasses[Comprehensive guide to Grails domain classes and their configuration]

=== Related rules

 * GrailsDuplicateMapping - https://codenarc.org/codenarc-rules-grails.html#grailsduplicatemapping-rule[Detects duplicate names in Grails domain class mapping blocks]
