<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Instances of a <code>Serializable</code> class can be saved out to file and rehydrated at leisure. But that only works when all the values in the class instance are themselves <code>Serializable</code> (or <code>transient</code>). Storing a non-serializable value in a <code>Serializable</code> class will prevent the serialization of that class.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">interface Fruit extends Serializable {...}
class Gooseberry implements Fruit { // Nonserializable because of Thread field
  Thread thread;
}

class Bowl implements Serializable {
  private static final long serialVersionUID = 1;

  private Fruit fruit = new Gooseberry(); //Non-Compliant
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">interface Fruit implements Serializable {...}
class Gooseberry implements Fruit {...} //Serializable

class Bowl implements Serializable {
  private static final long serialVersionUID = 1;

  private Fruit fruit = new Gooseberry(); //Compliant, Gooseberry is serializable
}</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_duplicates_s1948">duplicates: <a data-rspec-id="S1948" class="rspec-auto-link">S1948</a></h3>

</div>
<div class="sect2">
<h3 id="_on_25_sep_2014_085712_ann_campbell_wrote">on 25 Sep 2014, 08:57:12 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~nicolas.peru] please read over this code example &amp; make sure I correctly understood the problem. Not much data in the FB rule description to work from.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_25_sep_2014_113848_nicolas_peru_wrote">on 25 Sep 2014, 11:38:48 Nicolas Peru wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] Example updated, the FB rule refers to storage into fields that are Serializable.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_9_jan_2015_145824_pierre_yves_nicolas_wrote">on 9 Jan 2015, 14:58:24 Pierre-Yves Nicolas wrote:</h3>
<div class="paragraph">
<p>I updated the code examples to reflect what I think is the intent of the FB rule (and I removed a crucial inconsistency).</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_9_jan_2015_151039_pierre_yves_nicolas_wrote">on 9 Jan 2015, 15:10:39 Pierre-Yves Nicolas wrote:</h3>
<div class="paragraph">
<p>If my understanding of this rule is correct, this rule is triggered when:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>the enclosing class implements Serializable (directly or not)</p>
</li>
<li>
<p>the field is declared with a non-Serializable type</p>
</li>
<li>
<p>the assignment assigns an instance of a non-Serializable type which is different from the type used in the field declaration.
Therefore, it only raises issues on assignments to fields which are themselves detected by <a data-rspec-id="S1948" class="rspec-auto-link">RSPEC-1948</a>.</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>As I don&#8217;t see why someone would activate that rule without activating <a data-rspec-id="S1948" class="rspec-auto-link">RSPEC-1948</a>, I don&#8217;t see how useful it is.</p>
</div>
</div>
</div>
</div>