<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>In Transact-SQL, the semicolon statement terminator is in most cases optional. Therefore many developers don&#8217;t use semicolons. However, in some situations missing semicolons may yield insidious errors.</p>
</div>
<div class="paragraph">
<p>Semicolons are required by the ANSI standard, and Microsoft <a href="https://docs.microsoft.com/en-us/sql/t-sql/language-elements/transact-sql-syntax-conventions-transact-sql">recommends</a> the consistent usage of semicolons and might make semicolons mandatory in a future version of SQL Server. Also, semicolons make the code more portable.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="paragraph">
<p>In the code sample below, the exception will never be rethrown because <code>SELECT ERROR_MESSAGE() THROW</code> will be interpreted as a column alias.</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-sql" data-lang="sql">BEGIN TRY
  BEGIN TRAN;
  SELECT 1/0 AS AnException;
  COMMIT;
END TRY
BEGIN CATCH
  SELECT ERROR_MESSAGE()    -- Noncompliant; no exception will be thrown
  THROW
END CATCH</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-sql" data-lang="sql">BEGIN TRY
  BEGIN TRAN;
  SELECT 1/0 AS AnException;
  COMMIT;
END TRY
BEGIN CATCH
  SELECT ERROR_MESSAGE();
  THROW;
END CATCH</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Add a semicolon after this statement.</p>
</div>
<div class="paragraph">
<p>Remove this semicolon.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>statement needing a semicolon added or removed</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_relates_to_s1438">relates to: <a data-rspec-id="S1438" class="rspec-auto-link">S1438</a></h3>

</div>
<div class="sect2">
<h3 id="_on_7_apr_2017_112552_pierre_yves_nicolas_wrote">on 7 Apr 2017, 11:25:52 Pierre-Yves Nicolas wrote:</h3>
<div class="paragraph">
<p>Not sure this should be a "bug detection" rule. That&#8217;s the worst case but it&#8217;s probably rare. The example code comes from http://www.dbdelta.com/always-use-semicolon-statement-terminators/</p>
</div>
</div>
</div>
</div>