<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p><code>NOCOUNT</code> is by default deactivated (OFF) at server level. It means by default, the server will send to the client the number of rows affected by the SQL query executed which is, in most cases, useless because no one will read this information.</p>
</div>
<div class="paragraph">
<p>Deactivating this feature will save some network traffic and improve the execution performance of stored procedures and triggers that&#8217;s why it is recommended to define <code>SET NOCOUNT ON</code> at the beginning of the definition of <code>PROCEDURE</code>s and <code>TRIGGER</code>s, before any query is processed.</p>
</div>
<div class="paragraph">
<p>This rule raises an issue when <code>NOCOUNT</code> is not set or is set to OFF between the beginning of the <code>PROCEDURE</code> (or <code>TRIGGER</code>) definition and the first statement that is not a <code>SET</code>, <code>IF</code> or <code>DECLARE</code>.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-sql" data-lang="sql">CREATE PROCEDURE dbo.MyProc
AS
BEGIN
  DECLARE @var INT;
  SET NOCOUNT OFF; -- Noncompliant; deactivate NOCOUNT
  SELECT COUNT(*) FROM MY_TABLE
END;</code></pre>
</div>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-sql" data-lang="sql">CREATE PROCEDURE dbo.MyProc
AS
BEGIN
  -- Noncompliant; SET NOCOUNT is not specified so behaviour of the procedure execution is based on server configuration (OFF by default)
  SELECT COUNT(*) FROM MY_TABLE
END;</code></pre>
</div>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-sql" data-lang="sql">CREATE PROCEDURE dbo.MyProc
AS
BEGIN
  SELECT COUNT(*) FROM MY_TABLE
  SET NOCOUNT ON -- Noncompliant; SET NOCOUNT is set after select statement
END;</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-sql" data-lang="sql">CREATE PROCEDURE dbo.MyProc(@debug  INT)
AS
BEGIN
  DECLARE @var INT;
  IF @debug = 0
    BEGIN
      SET NOCOUNT ON;
    END
  SELECT COUNT(*) FROM MY_TABLE
END;</code></pre>
</div>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-sql" data-lang="sql">CREATE TRIGGER MyTrigger ON MyTable
AFTER INSERT
AS
BEGIN
  SET NOCOUNT ON;
  [...]
END;</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>\[Add|Set] configuration "NOCOUNT" with expected value "ON".</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>If "SET NOCOUNT ON" is missing, highlight the CREATE PROCEDURE or CREATE TRIGGER</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_3_sep_2018_153536_alexandre_gigleux_wrote">on 3 Sep 2018, 15:35:36 Alexandre Gigleux wrote:</h3>
<div class="literalblock">
<div class="content">
<pre>``++ALTER PROCEDURE++`` should be considered the same as ``++CREATE PROCEDURE++`` by this rule.</pre>
</div>
</div>
</div>
</div>
</div>