== Why is this an issue?

When you use the `DateTime.Now` property as part of a log message renders the log partially meaningless if accessed from a location in a different timezone. This happens because this property returns the current date and time expressed in the machine's local time without containing any timezone-related data (e.g., offset).
A good practice is always to use instants recorded in UTC with `DateTime.UtcNow`, `DateTimeOffset.Now` or `DateTimeOffSet.UtcNow` properties.
It's better because a UTC instant is expressed in Coordinated Universal Time, unaffected by changes such as daylight savings, and the DateTimeOffset gives you an instant with its distance from UTC. This way, you can always reconstruct a local time from these properties. Mind that the opposite is not always possible as this operation isn't reversible as some local times are ambiguous and can be matched to more than one UTC instant (for example, due to daylight savings).
Generally, unless you want to display only the Date and Time locally, you should always use DateTime.UtcNow, DateTimeOffset.UtcNow or DateTimeOffSet.Now.


=== What is the potential impact?

Logs that display local `DateTime` have partially no meaning if they are accessed from different timezones.

== How to fix it

The fix is to use `DateTime.UtcNow`, `DateTimeOffSet.Now` or, `DateTimeOffSet.UtcNow` and reconstruct a local DateTime upon display if needed.

=== Code examples

==== Noncompliant code example

[source,csharp,diff-id=1,diff-type=noncompliant]
----
_logger.LogInformation(DateTime.Now + "Some Event");
----

==== Compliant solution

[source,csharp,diff-id=3,diff-type=compliant]
----
_logger.LogInformation($"{DateTime.UtcNow}: Some Event");
----

==== Compliant solution

[source,csharp,diff-id=3,diff-type=compliant]
----
var eventDateTimeInUtc = DateTime.UtcNow;
_logger.LogInformation($"{eventDateTimeInUtc.ToLocalTime()}: Some Event");
----

== Resources

=== Documentation

https://learn.microsoft.com/en-us/dotnet/api/system.datetime.now[DateTime.Now documentation]
https://learn.microsoft.com/en-us/dotnet/api/system.datetime.utcnow[DateTime.UtcNow documentation]
https://learn.microsoft.com/en-us/dotnet/api/system.datetimeoffset[DateTimeOffset documentation]
https://learn.microsoft.com/en-us/dotnet/standard/datetime/resolve-ambiguous-times[Ambiguous times]
