include::../description.adoc[]

== Noncompliant Code Example

----
// Noncompliant; there are extra spaces after 'String'
var str:String     
----

ifdef::env-github,rspecator-view[]
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
