This rule raises an issue when catch blocks are not ordered from most specific to least specific exception types, making some catch blocks unreachable.

== Why is this an issue?

When multiple catch blocks are used in a try-catch statement, they are evaluated in the order they appear. If a more general exception type (like `Exception`) is placed before a more specific one (like `MathException` or `DMLException`), the specific catch block will never be reached.

This happens because the general exception handler will catch all exceptions that match its type or any of its subtypes. Since most specific exceptions inherit from the general `Exception` class, placing `Exception` first means it will handle all exceptions, leaving the subsequent specific catch blocks unreachable.

Unreachable catch blocks indicate a logical error in exception handling and can lead to:

* Unexpected behavior where specific exception handling logic is never executed
* Debugging difficulties when trying to understand why certain error handling paths are not triggered
* Maintenance issues where developers may not realize that their specific exception handling code is ineffective

=== What is the potential impact?

The unreachable catch blocks will never execute, which means specific exception handling logic is bypassed. This can lead to improper error handling, making it harder to diagnose and respond to different types of exceptions appropriately.

== How to fix it

Reorder the catch blocks so that more specific exception types come before more general ones. Place the general `Exception` catch block last.

=== Code examples

==== Noncompliant code example

[source,apex,diff-id=1,diff-type=noncompliant]
----
try {
    Integer result = 10 / 0;
    insert new Account(Name = 'Test');
} catch (Exception ex) { // Noncompliant - too general, placed first
    System.debug('General exception: ' + ex.getMessage());
} catch (MathException ex) {
    System.debug('Math exception: ' + ex.getMessage());
} catch (DMLException ex) {
    System.debug('DML exception: ' + ex.getMessage());
}
----

==== Compliant solution

[source,apex,diff-id=1,diff-type=compliant]
----
try {
    Integer result = 10 / 0;
    insert new Account(Name = 'Test');
} catch (MathException ex) {
    System.debug('Math exception: ' + ex.getMessage());
} catch (DMLException ex) {
    System.debug('DML exception: ' + ex.getMessage());
} catch (Exception ex) { // General exception comes last
    System.debug('General exception: ' + ex.getMessage());
}
----

== Resources

=== Documentation

 * Apex Exception Handling - https://developer.salesforce.com/docs/atlas.en-us.apexcode.meta/apexcode/apex_exception_definition.htm[Official Salesforce documentation on exception handling in Apex]

 * Exception Handling Best Practices - https://www.apexhours.com/exception-handling-in-apex[Comprehensive guide on exception handling patterns and best practices in Apex]

=== Related rules

 * RSPEC-1045 - https://rules.sonarsource.com/c/RSPEC-1045[Similar rule for C/C++ about catch block ordering]
