<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Free-format syntax is preferred because it is clearer, easier to write, and easier for newcomers to the language than fixed-format syntax.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-rpg" data-lang="rpg">     C                   IF        A &gt; 10
     C                   EVAL      B = 1
     C                   ENDIF</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-rpg" data-lang="rpg">      /free
        if A &gt; 10;
          B = 1;
        endif;
      /end-free</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Convert this calculation block to free-form syntax.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_15_dec_2014_134644_ann_campbell_wrote">on 15 Dec 2014, 13:46:44 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>assigned to you for approval</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_15_dec_2014_141718_pierre_yves_nicolas_wrote">on 15 Dec 2014, 14:17:18 Pierre-Yves Nicolas wrote:</h3>
<div class="paragraph">
<p>Looks good to me.</p>
</div>
<div class="paragraph">
<p>I guess that rule should be fired for a block of lines of code rather than for individual lines. If that&#8217;s the case, then we may have to use a linear remediation function.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_15_dec_2014_144953_ann_campbell_wrote">on 15 Dec 2014, 14:49:53 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>Linear with offset based on the number of lines? What do you think [~pierre-yves.nicolas]?</p>
</div>
<div class="paragraph">
<p>5 min + 5min/line ?</p>
</div>
<div class="paragraph">
<p>For the noncompliant code example, that works out to 20min&#8230;&#8203;</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_15_dec_2014_165931_pierre_yves_nicolas_wrote">on 15 Dec 2014, 16:59:31 Pierre-Yves Nicolas wrote:</h3>
<div class="paragraph">
<p>Ok. We should probably update the message to make it clear that the issue is on a block.</p>
</div>
</div>
</div>
</div>