== Why is this an issue?

`@SpringBootTest` is an annotation provided by the Spring Boot testing framework that is often used for integration testing.
It instructs the testing framework to load and configure the entire Spring application context, including all beans and components defined in your application.

Unit tests in Spring applications that use `@SpringBootTest` should only load the necessary beans to avoid unnecessary resource consumption and slow test execution.
Loading the entire application context can be harmful to test performance.

== How to fix it

When writing unit tests that use `@SpringBootTest`, specify only the beans that are required for the particular test using the classes attribute.
Avoid loading the entire application context unless it is necessary for the specific test case.

=== Code examples

==== Noncompliant code example

[source,java,diff-id=1,diff-type=noncompliant]
----
@SpringBootTest()  // Noncompliant
class MyServiceTest {
  // ...
}
----

==== Compliant solution

[source,java,diff-id=1,diff-type=compliant]
----
@SpringBootTest(classes = {MyServiceTest.TestMyConfig.class, MyService.class, RestConfig.class, ObjectMapper.class})
class MyServiceTest {
  // ...
}
----

== Resources

=== Documentation

* Spring Boot documentation - https://docs.spring.io/spring-boot/docs/current/api/org/springframework/boot/test/context/SpringBootTest.html[@SpringBootTest]
* Spring - https://spring.io/guides/gs/testing-web/[Testing the Web Layer]

=== Articles & blog posts

* Baeldung - https://www.baeldung.com/spring-boot-testing[Testing in Spring Boot]

