<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>The logic of a task&#8217;s actions, like <code>doFirst</code> and <code>doLast</code>, is only executed when the task is executed.
Configuration logic (everything outside the action blocks) is executed in the configuration phase for every build,
whether the task will be executed or not.</p>
</div>
<div class="paragraph">
<p>Code that could also be moved into the action blocks but resides in the configuration logic can
inflict a performance penalty.</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="sect3">
<h4 id="_performance">Performance</h4>
<div class="paragraph">
<p>Moving code from the task&#8217;s configuration logic into the task&#8217;s action blocks can speed up your builds.</p>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Move variable declarations and initializations into action blocks if they are read-only inside the action blocks.
An exception is the read access to the project configurations,
which should only happen during the configuration phase because of the configuration cache.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-kotlin" data-lang="kotlin">tasks.register("printArtifactNames") {
    val libraryNames = configurations.compileClasspath.get().map { it.name } // Non-compliant: "map" is always executed, but the result used in actions only
    doLast {
        logger.quiet(libraryNames.joinToString())
    }
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-kotlin" data-lang="kotlin">tasks.register("printArtifactNames") {
    val compileClasspath: FileCollection = configurations.compileClasspath.get()
    doLast {
        val libraryNames = compileClasspath.map { it.name } // Compliant: defined in  action where it is used
        logger.quiet(libraryNames.joinToString())
    }
}</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p><a href="https://docs.gradle.org/current/userguide/authoring_maintainable_build_scripts.html#sec:minimize_logic_executed_configuration_phase">Gradle Documentation: Minimize logic executed during the configuration phase</a></p>
</li>
<li>
<p><a href="https://docs.gradle.org/current/userguide/configuration_cache.html">Gradle Documentation: Configuration cache</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>