== How to fix it in Beego

=== Code examples

include::../../common/fix/code-rationale.adoc[]

==== Noncompliant code example

[source,go,diff-id=201,diff-type=noncompliant]
----
import (
    "github.com/beego/beego/v2/server/web"
)

func handleGet(c web.Controller) {
    location := r.GetString("url")
    c.Redirect(location, 302)
}
----

==== Compliant solution

[source,go,diff-id=201,diff-type=compliant]
----
import (
    "github.com/beego/beego/v2/server/web"
    "slices"
)

func handleGet(c web.Controller) {
    location := r.GetString("url")

    allowedHosts := []string{
        "https://trusted1.example.com/",
        "https://trusted2.example.com/",
    }

    if slices.Contains(allowedHosts, location) {
        c.Redirect(location, 302)
    }
}
----

include::../../common/fix/how-does-this-work.adoc[]

=== Pitfalls

include::../../common/pitfalls/starts-with.adoc[]

