== How to fix it in fasthttp

=== Code examples

include::../../common/fix/code-rationale.adoc[]

==== Noncompliant code example

[source,go,diff-id=301,diff-type=noncompliant]
----
import (
    "github.com/valyala/fasthttp"
)

func handleGet(c *fasthttp.RequestCtx) {
    location := string(c.QueryArgs().Peek("url"))
    c.Redirect(location, 302)
}
----

==== Compliant solution

[source,go,diff-id=301,diff-type=compliant]
----
import (
    "github.com/valyala/fasthttp"
    "slices"
)

func handleGet(c *fasthttp.RequestCtx) {
    location := string(c.QueryArgs().Peek("url"))

    allowedHosts := []string{
        "https://trusted1.example.com/",
        "https://trusted2.example.com/",
    }

    if slices.Contains(allowedHosts, location) {
        c.Redirect(location, 302)
    }
}
----

include::../../common/fix/how-does-this-work.adoc[]

=== Pitfalls

include::../../common/pitfalls/starts-with.adoc[]

