== How to fix it in Fiber

=== Code examples

include::../../common/fix/code-rationale.adoc[]

==== Noncompliant code example

[source,go,diff-id=401,diff-type=noncompliant]
----
import (
    "github.com/gofiber/fiber/v3"
)

func handleGet(c fiber.Ctx) {
    location := c.Query("url")
    c.Redirect().Status(302).To(location)
}
----

==== Compliant solution

[source,go,diff-id=401,diff-type=compliant]
----
import (
    "github.com/gofiber/fiber/v3"
    "slices"
)

func handleGet(c fiber.Ctx) {
    location := c.Query("url")

    allowedHosts := []string{
        "https://trusted1.example.com/",
        "https://trusted2.example.com/",
    }

    if slices.Contains(allowedHosts, location) {
        c.Redirect().Status(302).To(location)
    }
}
----

include::../../common/fix/how-does-this-work.adoc[]

=== Pitfalls

include::../../common/pitfalls/starts-with.adoc[]

