== How to fix it in Gin

=== Code examples

include::../../common/fix/code-rationale.adoc[]

==== Noncompliant code example

[source,go,diff-id=501,diff-type=noncompliant]
----
import (
    "github.com/gin-gonic/gin"
)

func handleGet(c *gin.Context) {
    location := c.Query("url")
    c.Redirect(303, tainted)
}
----

==== Compliant solution

[source,go,diff-id=501,diff-type=compliant]
----
import (
    "github.com/gin-gonic/gin"
    "slices"
)

func handleGet(c *gin.Context) {
    location := c.Query("url")

    allowedHosts := []string{
        "https://trusted1.example.com/",
        "https://trusted2.example.com/",
    }

    if slices.Contains(allowedHosts, location) {
        c.Redirect(303, tainted)
    }
}
----

include::../../common/fix/how-does-this-work.adoc[]

=== Pitfalls

include::../../common/pitfalls/starts-with.adoc[]

