== How to fix it in Go Standard Library

=== Code examples

include::../../common/fix/code-rationale.adoc[]

==== Noncompliant code example

[source,go,diff-id=101,diff-type=noncompliant]
----
import (
    "net/http"
)

func handleGet(w http.ResponseWriter, r *http.Request) {
    location := r.URL.Query().Get("url")
    http.Redirect(w, r, location, http.StatusFound)
}
----

==== Compliant solution

[source,go,diff-id=101,diff-type=compliant]
----
import (
    "net/http"
    "slices"
)

func handleGet(w http.ResponseWriter, r *http.Request) {
    location := r.URL.Query().Get("url")

    allowedHosts := []string{
        "https://trusted1.example.com/",
        "https://trusted2.example.com/",
    }

    if slices.Contains(allowedHosts, location) {
        http.Redirect(w, r, location, http.StatusFound)
    }
}
----

include::../../common/fix/how-does-this-work.adoc[]

=== Pitfalls

include::../../common/pitfalls/starts-with.adoc[]

