<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when a <code>tensorflow.Variable</code> created inside of a <code>tensorflow.function</code> is not a singleton.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p><code>tensorflow.function</code>s only supports singleton <code>tensorflow.Variable</code>s. This means the variable will be created on the first call
of the <code>tensorflow.function</code> and will be reused across the subsequent calls. Creating a <code>tensorflow.Variable</code> that is not a singleton
will raise a <code>ValueError</code>.</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">import tensorflow as tf

@tf.function
def f(x):
  v = tf.Variable(1.0)
  return v</code></pre>
</div>
</div>
<div class="paragraph">
<p>In the example above each time the function <code>f</code> will be called a new <code>tensorflow.Variable</code> will be created.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>To fix this issue use the singleton pattern to avoid the creation of a <code>tensorflow.Variable</code> multiple times.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">import tensorflow as tf

class Count(tf.Module):
  def __init__(self):
    self.count = None

  @tf.function
  def __call__(self):
    self.count = tf.Variable(0) # Noncompliant: the variable will be re-created for each call
    return self.count.assign_add(1)

c = Count()
c() # ValueError is raised
c()</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">import tensorflow as tf

class Count(tf.Module):
  def __init__(self):
    self.count = None

  @tf.function
  def __call__(self):
    if self.count is None:
      self.count = tf.Variable(0) # Compliant: the variable is created only on during the first call
    return self.count.assign_add(1)

c = Count()
c()
c()</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>TensorFlow Documentation - <a href="https://www.tensorflow.org/guide/function#creating_tfvariables">Creating tf.Variables</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>