<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Trailing closure syntax can only be used with the last argument to a function call. Place a function type parameter anywhere else in the list and you limit the options of the caller.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-swift" data-lang="swift">func foo(p1: Int-&gt;Int, p2: Int){  // Noncompliant; p1 should come at the end
  print(p1(p2))
}

foo({a in a * 2}, 42) // Trailing closure syntax can't be used here</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-swift" data-lang="swift">func foo(p2: Int, p1: Int-&gt;Int){
  print(p1(p2))
}

foo(42) {a in a * 2}</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Move "x" to the end of the parameter list.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_21_may_2015_173356_ann_campbell_wrote">on 21 May 2015, 17:33:56 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~freddy.mallet] what do you think of the SQALE characteristic? I&#8217;m not sure what to pick for rules about making your API easier to use.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_27_may_2015_090112_freddy_mallet_wrote">on 27 May 2015, 09:01:12 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>Hi @Ann, according to the description of this rule the code provided in the "Non-Compliant Code Example" will lead to a runtime error so I would associate this rule to the SQALE "Reliability " characteristic. But at the same time I&#8217;m not sure to fully understand this rule.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_27_may_2015_191352_ann_campbell_wrote">on 27 May 2015, 19:13:52 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>Thanks for weighing in [~freddy.mallet], but what we&#8217;re looking at in the first line of the  Noncompliant example is the function declaration, and as-is, you cannot use trailing closure syntax to call it. (This code made my eyes cross, and [~elena.vilchik] had to talk to me like I was 6 when explaining it.)</p>
</div>
<div class="paragraph">
<p>So this really is just about making your API easier to use, &amp; I&#8217;m afraid the exactly right SQALE category doesn&#8217;t exist for it. :-/</p>
</div>
</div>
</div>
</div>