<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>OS/VS COBOL accepted the <code>ON</code> statement, but IBM Enterprise COBOL does not accept it anymore. The <code>ON</code> statement allows the selective execution of statements it contains. Similar functions are provided in Enterprise COBOL by <code>EVALUATE</code> and <code>IF</code></p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cobol" data-lang="cobol">       IDENTIFICATION DIVISION.
       PROGRAM-ID. foo.

       PROCEDURE DIVISION.
      * Non-Compliant
         ON 1
           DISPLAY 'First time'
         ELSE
           DISPLAY 'Other times'.
       END PROGRAM foo.</code></pre>
</div>
</div>
</div>
</div>
</div>