<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>It is recommended not to declare more than one property per statement for the sake of code readability and maintainability.
Declaring multiple properties in a single statement can make the code harder to understand and debug.
It also increases the risk of introducing errors or overlooking specific property assignments.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>By declaring one property per statement, developers can easily identify and modify individual properties, improving code clarity and reducing potential mistakes.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-php" data-lang="php">&lt;?php
class Foo
{
   private $bar = 1, $bar2 = 2;
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-php" data-lang="php">&lt;?php
class Foo
{
   private $bar1 = 1;
   private $bar2 = 2;
}</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p><a href="https://www.php.net/manual/en/language.oop5.properties.php">PHP Manual - Properties</a></p>
</li>
<li>
<p><a href="https://www.php.net/manual/en/language.operators.assignment.php">PHP Manual - Assignment Operators</a></p>
</li>
</ul>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>{0} property declarations were found in this statement. Reformat the code to declare only one property per statement.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_relates_to_s122">relates to: <a data-rspec-id="S122" class="rspec-auto-link">S122</a></h3>

</div>
<div class="sect2">
<h3 id="_on_27_may_2014_113254_ann_campbell_wrote">on 27 May 2014, 11:32:54 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~linda.martin] how is this different from the standard rule against having more than one statement per line?</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_27_may_2014_151818_linda_martin_wrote">on 27 May 2014, 15:18:18 Linda Martin wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] here we are talking about 2 property declarations in the same statement, so it won&#8217;t be detected by the rule "one statement per line".</p>
</div>
</div>
</div>
</div>