<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>The ability to map a class to a database table has made database interaction in Java a lot easier. But map multiple classes to the same table and you&#8217;ll end up with a classic case of the left hand not knowing what the right hand is doing. In the worse case scenario, it could lead to serious data corruption.</p>
</div>
<div class="paragraph">
<p>This rule raises an issue when multiple classes are mapped to the same table with Hibernate or JPA annotations.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">@Entity  // implicitly mapped to "point" table
public class Point {
  // ..
}

@Entity
@Table(name = "point")  // Noncompliant
public class Spot {
  // ...
}</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>"tableName" is also associated with "classname".</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="ulist">
<ul>
<li>
<p>primary: table name on second class</p>
</li>
</ul>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_6_oct_2015_154945_ann_campbell_wrote">on 6 Oct 2015, 15:49:45 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~nicolas.peru] I don&#8217;t know if this would go in 'implementation details' but this rule should look at both classes and XML configuration files. In Hibernate, at least, you can map classes to tables in XML.</p>
</div>
<div class="paragraph">
<p>Also, since this is a cross-file check, I understand that we won&#8217;t get to even the first half of it soon. :)</p>
</div>
</div>
</div>
</div>