The ``++... IsNot ...++`` syntax is more compact and more readable than the ``++Not ... Is ...++`` syntax.

== Noncompliant Code Example

----
Module Module1
    Sub Main()
        Dim a = Not "a" Is Nothing ' Noncompliant
    End Sub
End Module
----

== Compliant Solution

----
Module Module1
    Sub Main()
        Dim a = "a" IsNot Nothing  ' Compliant
    End Sub
End Module
----
