== Why is this an issue?

The ``++... IsNot ...++`` syntax is more compact and more readable than the ``++Not ... Is ...++`` syntax.


=== Noncompliant code example

[source,vbnet]
----
Module Module1
    Sub Main()
        Dim a = Not "a" Is Nothing ' Noncompliant
    End Sub
End Module
----


=== Compliant solution

[source,vbnet]
----
Module Module1
    Sub Main()
        Dim a = "a" IsNot Nothing  ' Compliant
    End Sub
End Module
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Replace this use of "Not ... Is ..." with "IsNot".


endif::env-github,rspecator-view[]
