<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Manually implementing <code>PartialOrd</code> for a type that derives <code>Ord</code> can lead to inconsistencies, as the implementations of these traits must agree for sorting and other comparisons. According to the trait contract, the following must hold for any type implementing <code>Ord</code>:</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-rust" data-lang="rust">k1.cmp(&amp;k2) == k1.partial_cmp(&amp;k2).unwrap()</code></pre>
</div>
</div>
<div class="paragraph">
<p>Using a default-generated <code>Ord</code> implementation with an explicitly defined <code>PartialOrd</code> is a risky practice that may cause unexpected behavior.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-rust" data-lang="rust">#[derive(Ord, PartialEq, Eq)]
struct Foo;

impl PartialOrd for Foo {
    fn partial_cmp(&amp;self, other: &amp;Self) -&gt; Option&lt;Ordering&gt; {
        None  // Noncompliant: Manually implemented PartialOrd when Ord is derived.
    }
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-rust" data-lang="rust">#[derive(PartialEq, Eq)]
struct Foo;

impl PartialOrd for Foo {
    fn partial_cmp(&amp;self, other: &amp;Foo) -&gt; Option&lt;Ordering&gt; {
        Some(self.cmp(other)) // Compliant: Manual implementation is consistent with Ord.
    }
}

impl Ord for Foo {
    fn cmp(&amp;self, other: &amp;Foo) -&gt; Ordering {
        // Implementation here
    }
}</code></pre>
</div>
</div>
<div class="paragraph">
<p>Or, if a custom ordering is not needed:</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-rust" data-lang="rust">#[derive(Ord, PartialOrd, PartialEq, Eq)]
struct Foo; // Compliant: Both Ord and PartialOrd are derived.</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Clippy Lints - <a href="https://rust-lang.github.io/rust-clippy/master/index.html#derive_ord_xor_partial_ord" class="bare">https://rust-lang.github.io/rust-clippy/master/index.html#derive_ord_xor_partial_ord</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>