<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when ScopedValue.Carrier.where() return values are ignored or when multiple separate ScopedValue instances are used instead of bundling values into a record.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>ScopedValue.Carrier is immutable. The <code>where()</code> method returns a new Carrier object - it doesn&#8217;t modify the existing one. Ignoring the return value means your mapping has no effect.</p>
</div>
<div class="paragraph">
<p>Using multiple separate ScopedValue instances reduces cache efficiency. Java recommends bundling related values into a single record class bound to one ScopedValue instance for better performance.</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="paragraph">
<p>Ignoring <code>where()</code> return values causes scoped values to not bind correctly, leading to <code>NoSuchElementException</code> or unexpected behavior.</p>
</div>
<div class="paragraph">
<p>Multiple separate ScopedValue instances create cache thrashing, causing expensive scope scanning that slows down your application.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Always use the return value of ScopedValue.Carrier.where() since it returns a new immutable Carrier object.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">ScopedValue.Carrier carrier = ScopedValue.where(key1, value1);
carrier.where(key2, value2); // Noncompliant: result not used
carrier.run(() -&gt; { /* operation */ });</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">ScopedValue.Carrier carrier = ScopedValue.where(key1, value1);
ScopedValue.Carrier newCarrier = carrier.where(key2, value2);
newCarrier.run(() -&gt; { /* operation */ });</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>ScopedValue.Carrier JavaDoc - <a href="https://docs.oracle.com/en/java/javase/25/docs/api/java.base/java/lang/ScopedValue.Carrier.html">Official documentation for ScopedValue.Carrier class explaining its immutable nature</a></p>
</li>
<li>
<p>ScopedValue JavaDoc - <a href="https://docs.oracle.com/en/java/javase/25/docs/api/java.base/java/lang/ScopedValue.html">Official documentation for ScopedValue class with performance recommendations</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>