This rule reports when a class depends on another class, deviating from the intended architecture.

== Why is this an issue?

This project has an intended architecture defined, which includes high-level components and their allowed relationships.
This issue is being raised because a dependency in the code is not allowed since it constitutes an architecture deviation.

=== What is the potential impact?

When code deviates from the intended architecture, it gradually erodes the projectâ€™s structure.
As these deviations accumulate, the code becomes more complex and less consistent with the intent, making future development increasingly challenging.

== How to fix it

Remove or refactor the deviating dependency.
This may involve replacing the dependency with an allowed one, moving code to appropriate classes or namespaces, or introducing new abstractions. The exact solution depends on your project's specific situation and the code deviating the intended architecture.

=== Code examples

==== Noncompliant code example

If the intended architecture prohibits classes in namespace `Example.Panels` to depend on classes in `Example.Repos`, then the following code example is noncompliant:

[source,java,diff-id=1,diff-type=noncompliant]
----
using Example.Repos;

namespace Example.Panels;

public class ShowCustomersPanel
{
    CustomerRepo customerRepo = ...;
}
----

==== Compliant solution

A solution could be to call into the service layer instead.

[source,java,diff-id=1,diff-type=compliant]
----
using Example.Services;

namespace Example.Panels;

public class ShowCustomersPanel
{
    CustomerService customerService = ...;
}
----
