<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when an abstract class contains only fields and no methods, making it function as a data container rather than a proper base class.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Abstract classes are designed to serve as base classes that define common behavior for their subclasses. They should contain abstract or virtual methods that subclasses can implement or override.</p>
</div>
<div class="paragraph">
<p>When an abstract class contains only fields and no methods, it&#8217;s not fulfilling its intended purpose. Instead, it&#8217;s acting as a simple data container that cannot be instantiated. This creates confusion about the class&#8217;s role and design intent.</p>
</div>
<div class="paragraph">
<p>Using the <code>abstract</code> keyword for a data-only class is misleading because:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>It suggests that the class is meant for inheritance and method overriding</p>
</li>
<li>
<p>It doesn&#8217;t clearly communicate that the class is simply a non-instantiable data container</p>
</li>
<li>
<p>Other developers might expect to find abstract methods to implement</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>A class that only holds data and should not be instantiated is better served by using a private or protected constructor. This approach clearly communicates the design intent and prevents instantiation without the misleading implications of the <code>abstract</code> keyword.</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="paragraph">
<p>This design issue can lead to confusion among team members about the class&#8217;s intended purpose and usage patterns. It may result in incorrect assumptions about inheritance relationships and make the codebase harder to understand and maintain.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Remove the abstract keyword and add a private or protected constructor to prevent instantiation. This clearly indicates that the class is a data container that should not be instantiated.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-apex" data-lang="apex">public abstract class UserData {
    String username;
    String email;
    Integer userId;
} // Noncompliant</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-apex" data-lang="apex">public class UserData {
    String username;
    String email;
    Integer userId;

    private UserData() {
        // Private constructor prevents instantiation
    }
}</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Apex Classes - <a href="https://developer.salesforce.com/docs/atlas.en-us.apexcode.meta/apexcode/apex_classes.htm">Official Salesforce documentation on Apex classes and abstract classes</a></p>
</li>
<li>
<p>Apex Class Definition - <a href="https://developer.salesforce.com/docs/atlas.en-us.apexcode.meta/apexcode/apex_classes_defining.htm">Salesforce documentation explaining abstract and virtual classes in Apex</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>