<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>TODO</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cobol" data-lang="cobol">PERFORM UNTIL FIN-RGL0
     PERFORM DEBUT-RGL0
     PERFORM UNTIL FIN-RGL1
     PERFORM DEBUT-RGL1
          PERFORM UNTIL FIN-RGL2
           PERFORM DEBUT-RGL2
           PERFORM ALIM-RGL2
           PERFORM FIN-RGL2
     END-PERFORM
     PERFORM FIN-RGL1
END-PERFORM
     PERFORM FIN-RGL0
END-PERFORM</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cobol" data-lang="cobol">PERFORM TRAITEMENT-RGL0 UNTIL FIN-RGL0.
TRAITEMENT-RGL0.
PERFORM DEBUT-RGL0
PERFORM TRAITEMEN-RGL1 UNTIL FIN-RGL1
     PERFORM FIN-RGL0
	.
TRAITEMENT-RGL1.
     PERFORM DEBUT-RGL1
     PERFORM UNTIL FIN-RGL2
           PERFORM DEBUT-RGL2
           PERFORM ALIM-RGL2
           PERFORM FIN-RGL2
     END-PERFORM
     PERFORM FIN-RGL1</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_duplicates_s134">duplicates: <a data-rspec-id="S134" class="rspec-auto-link">S134</a></h3>

</div>
<div class="sect2">
<h3 id="_relates_to_s1306">relates to: <a data-rspec-id="S1306" class="rspec-auto-link">S1306</a></h3>

</div>
<div class="sect2">
<h3 id="_on_13_may_2014_083941_freddy_mallet_wrote">on 13 May 2014, 08:39:41 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>Might in fact duplicates <a data-rspec-id="S134" class="rspec-auto-link">RSPEC-134</a></p>
</div>
</div>
<div class="sect2">
<h3 id="_on_13_may_2014_084901_ann_campbell_wrote">on 13 May 2014, 08:49:01 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>Given the way <a data-rspec-id="S134" class="rspec-auto-link">RSPEC-134</a> was implemented, this is indeed a duplicate.</p>
</div>
</div>
</div>
</div>