== Why is this an issue?

A dynamic class defines an object that can be altered at run time by adding or changing properties and methods. This extremely powerful mechanism should be used very carefully, and only in very limited use cases. 


Indeed, by definition dynamic classes make refactoring difficult and prevent the compiler from raising potential errors at compile time. 


=== Noncompliant code example

[source,flex]
----
dynamic public class DynamicFoo
{...}
----


=== Compliant solution

[source,flex]
----
public class Foo  //Note that the class has been renamed to avoid confusion 
{...}
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
