This rule raises an issue when there are multiple separate `import` statements that could be grouped together in a factored import block.

== Why is this an issue?

Go supports two ways to write import statements: multiple separate statements or a single factored statement that groups imports in parentheses.

While both approaches are syntactically valid, the Go community strongly recommends using factored import statements. This convention is documented in the official Go Tour and is widely adopted across Go codebases.

Factored imports offer several advantages:

* **Better organization**: All imports are grouped in one place, making dependencies easier to see at a glance
* **Cleaner code**: Reduces visual clutter by eliminating repeated `import` keywords
* **Consistency**: Follows the established Go style guidelines that most developers expect
* **Tool compatibility**: Many Go formatting tools like `gofmt` automatically convert separate imports to factored form

Using separate import statements makes code look less polished and can signal unfamiliarity with Go conventions to other developers.

=== What is the potential impact?

This is a style issue that affects code readability and consistency. While it doesn't impact functionality, following Go conventions makes code more maintainable and professional.

== How to fix it

Group multiple separate import statements into a single factored import block using parentheses.

=== Code examples

==== Noncompliant code example

[source,go,diff-id=1,diff-type=noncompliant]
----
import "fmt"
import "math"
import "strings" // Noncompliant
----

==== Compliant solution

[source,go,diff-id=1,diff-type=compliant]
----
import (
    "fmt"
    "math"
    "strings"
)
----

== Resources

=== Documentation

 * A Tour of Go - Imports - https://go.dev/tour/basics/2[Official Go documentation explaining factored import statements]

 * Effective Go - Formatting - https://go.dev/doc/effective_go#formatting[Go's official style guide covering code formatting conventions]
