<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when deprecated Rails dynamic finder methods like <code>find_by_*</code>, <code>find_all_by_*</code>, <code>find_or_initialize_by_*</code>, or <code>find_or_create_by_*</code> are used.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Dynamic finder methods were a Rails feature that allowed querying records using method names like <code>User.find_by_name('John')</code> or <code>Post.find_all_by_status_and_category('published', 'tech')</code>. While convenient, these methods have several problems.</p>
</div>
<div class="paragraph">
<p>First, they rely on Ruby&#8217;s <code>method_missing</code> mechanism. When you call <code>User.find_by_name('John')</code>, Ruby first looks for a method called <code>find_by_name</code> on the User class. When it doesn&#8217;t find one, it calls <code>method_missing</code>, which then parses the method name to understand what you want to query. This extra step adds unnecessary processing overhead compared to explicit methods.</p>
</div>
<div class="paragraph">
<p>Second, dynamic finders become difficult to read and maintain when you have multiple conditions. A method like <code>User.find_all_by_active_and_role_and_created_at(true, 'admin', 1.week.ago)</code> is hard to understand at a glance, and the parameter order becomes important and error-prone.</p>
</div>
<div class="paragraph">
<p>Third, dynamic finders are less flexible for complex queries. You cannot easily add conditions like ranges, OR clauses, or joins using the dynamic syntax.</p>
</div>
<div class="paragraph">
<p>Finally, many of these methods were officially deprecated in Rails 4.0 and removed in later versions. Using them prevents you from upgrading to newer Rails versions and goes against current ActiveRecord conventions.</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="paragraph">
<p>Using deprecated dynamic finder methods can lead to performance degradation due to <code>method_missing</code> overhead, reduced code readability and maintainability, and compatibility issues when upgrading Rails versions. Applications may fail to run on newer Rails versions that have removed these deprecated methods.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it_in_rails">How to fix it in Rails</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Replace <code>find_all_by_*</code> methods with explicit <code>.where</code> clauses. The <code>.where</code> method is more performant, readable, and flexible for complex queries.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-ruby" data-lang="ruby"># Deprecated dynamic finder methods
InfoItem.find_all_by_work_order(self.work_order) # Noncompliant
User.find_all_by_active_and_role(true, 'admin') # Noncompliant</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-ruby" data-lang="ruby"># Use modern ActiveRecord syntax
InfoItem.where(work_order: self.work_order)
User.where(active: true, role: 'admin')</code></pre>
</div>
</div>
<div class="paragraph">
<p>Replace <code>find_or_initialize_by_*</code> and <code>find_or_create_by_*</code> methods with hash-based syntax. This ensures compatibility with modern Rails versions.</p>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_code_examples_2">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example_2">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-ruby" data-lang="ruby"># Deprecated dynamic methods
Progress.find_or_initialize_by_chore_id_and_period(chore.id, period) # Noncompliant
Team.find_or_create_by_name_and_status('Justice League', 'active') # Noncompliant</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution_2">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-ruby" data-lang="ruby"># Use hash-based syntax
Progress.find_or_initialize_by(chore_id: chore.id, period: period)
Team.find_or_create_by(name: 'Justice League', status: 'active')</code></pre>
</div>
</div>
<div class="paragraph">
<p>For complex queries with multiple conditions, modern syntax scales better and is more readable than dynamic finders.</p>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_code_examples_3">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example_3">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-ruby" data-lang="ruby"># Complex dynamic finder becomes unwieldy
User.find_all_by_active_and_role_and_created_at(true, 'admin', 1.week.ago) # Noncompliant</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution_3">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-ruby" data-lang="ruby"># Modern syntax scales better with multiple conditions
User.where(active: true, role: 'admin', created_at: 1.week.ago..Time.now)</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Rails Active Record Query Interface Guide - <a href="https://guides.rubyonrails.org/active_record_querying.html">Official Rails guide covering modern querying methods including where, find_by, and find_or_create_by</a></p>
</li>
<li>
<p>Rails 4.0 Release Notes - Deprecated Features - <a href="https://guides.rubyonrails.org/4_0_release_notes.html#deprecated-features">Official documentation about Rails 4.0 deprecations including dynamic finder methods</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>