include::../description.adoc[]

== Noncompliant Code Example

----
Public Class Sample

    Private myString As String = "foo"

    Public Sub Go()
        SyncLock myString   ' Noncompliant
        End SyncLock
    End Sub

End Class
----

== Compliant Solution

----
Public Class Sample

    Private Shared ReadOnly fLock As New Object

    Public Sub Go()
        SyncLock fLock
        End SyncLock
    End Sub

End Class
----
