include::../why-dotnet.adoc[]

== How to fix it

=== Code examples

==== Noncompliant code example

[source,vbnet,diff-id=1,diff-type=noncompliant]
----
Public Class Sample

    Private myString As String = "foo"

    Public Sub Go()
        SyncLock myString   ' Noncompliant
        End SyncLock
    End Sub

End Class
----

==== Compliant solution

[source,vbnet,diff-id=1,diff-type=compliant]
----
Public Class Sample

    Private Shared ReadOnly fLock As New Object

    Public Sub Go()
        SyncLock fLock
        End SyncLock
    End Sub

End Class
----


include::../rspecator.adoc[]

include::../resources-dotnet.adoc[]