== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,vbnet]
----
Public Class Sample

    Private myString As String = "foo"

    Public Sub Go()
        SyncLock myString   ' Noncompliant
        End SyncLock
    End Sub

End Class
----

=== Compliant solution

[source,vbnet]
----
Public Class Sample

    Private Shared ReadOnly fLock As New Object

    Public Sub Go()
        SyncLock fLock
        End SyncLock
    End Sub

End Class
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../highlighting.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
