In the context of ASP.NET Core MVC web applications, both model binding and model validation are processes that take place prior to the execution of a controller action. It is imperative for the application to examine the `ModelState.IsValid` and respond accordingly.

This rule enforces the developer to validate the model within a controller action, ensuring the application's robustness and reliability.

== Why is this an issue?

Calling `ModelState.IsValid` is necessary because it checks if the submitted data in the HTTP request is valid or not. This method evaluates all the validation attributes applied on your model properties and determines whether the data provided satisfies those validation rules.

=== What is the potential impact?

If `ModelState.IsValid` returns true, it means that the data is valid and the process can continue. If it returns false, it means that the validation failed, indicating that the data is not in the expected format or is missing required information.

In such cases, the controller action should handle this by returning an appropriate response, such as re-displaying the form with error messages. This helps maintain the integrity of the data and provides feedback to the user, enhancing the overall user experience and security of your application.

=== Exceptions

Web API controllers don't have to check `ModelState.IsValid` if they have the `[ApiController]` attribute. In that case, an automatic HTTP `400` response containing error details is returned when model state is invalid.

== How to fix it

=== Code examples

==== Noncompliant code example

[source,csharp,diff-id=1,diff-type=noncompliant]
----
public async Task<IActionResult> Create(Movie movie) // Noncompliant: model validity check is missing
{
    _context.Movies.Add(movie);
    await _context.SaveChangesAsync();

    return RedirectToAction(nameof(Index));
}
----

==== Compliant solution

[source,csharp,diff-id=1,diff-type=compliant]
----
public async Task<IActionResult> Create(Movie movie)
{
    if (!ModelState.IsValid)
    {
        return View(movie);
    }

    _context.Movies.Add(movie);
    await _context.SaveChangesAsync();

    return RedirectToAction(nameof(Index));
}
----

== Resources

=== Documentation

* Microsoft Learn - https://learn.microsoft.com/en-us/aspnet/core/mvc/models/validation

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

ModelState.IsValid should be called in controller actions.

=== Highlighting

Controller action identifier.

endif::env-github,rspecator-view[]
