== Why is this an issue?

include::description.adoc[]

=== Noncompliant code example

[source,text]
----
if (expression) {  
  return true;
} else {
  return false;
}
----


=== Compliant solution

[source,text]
----
return expression; 
----

