<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule leads to greater precision in the definition of local variables by making the developer intention about modifying the variable explicit. The <code>const</code> qualification shall be applied to the object pointed to, not to the pointer, since it is the object itself that is being protected.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">std::string&amp; getString();
void myfunc()
{
  std::string&amp; s = getString(); // Noncompliant
  if (s.size()) {
   std::cout &lt;&lt; s;
  }
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">std::string&amp; getString();
void myfunc () {
  const std::string&amp; x = getString();
  if (s.size()) {
    std::cout &lt;&lt; s;
  }
}</code></pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p>MISRA C:2012, 8.13 - A pointer should point to a const-qualified type whenever possible</p>
</li>
</ul>
</div>
<hr>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Mark "XXX" as const at every possible pointer level.</p>
</div>
</div>
</div>
</div>