Unsigned integer expressions do not strictly overflow, but instead wrap around in a modular way.


Any constant unsigned integer expressions that in effect “overflows” will not be detected by the compiler. Although there may be good reasons at run-time to rely on the modular arithmetic provided by unsigned integer types, the reasons for using it at compile-time to evaluate a constant expression are less obvious. Any instance of an unsigned integer constant expression wrapping around is therefore likely to indicate a programming error.


This rule applies equally to all phases of the translation process. Constant expressions that the compiler chooses to evaluate at compile time are evaluated in such a way that the results are identical to those that would be obtained by evaluation on the target, with the exception of those appearing in conditional preprocessing directives. For such directives, the usual rules of arithmetic apply but the ``++int++`` and ``++unsigned int++`` types behave instead as if they were ``++long++`` and ``++unsigned long++`` respectively.


== Noncompliant Code Example

On a machine with 16-bit ``++int++`` type:

----
#define BASE 65024u
switch ( x )
{
  case BASE + 0u:
    f ( );
    break;
  case BASE + 1u:
    g ( );
    break;
  case BASE + 512u: // Noncompliant, wraps to 0
    h ( );
    break;
}
----


== See

* https://wiki.sei.cmu.edu/confluence/x/bNYxBQ[CERT, INT30-C.] - Ensure that unsigned integer operations do not wrap


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
