<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>If the <code>width</code> and <code>height</code> attributes are set, the space required for the image is reserved immediately by the browser, even before it actually starts to load the image.</p>
</div>
<div class="paragraph">
<p>Without those attributes, the page layout constantly changes as images are loaded until they are all loaded, which can disorient users.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-html" data-lang="html">&lt;img src="logo.png" alt="My Company" /&gt;                           &lt;!-- Non-Compliant --&gt;</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-html" data-lang="html">&lt;img src="logo.png" alt="My Company" width="100" height="50" /&gt;   &lt;!-- Compliant --&gt;</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Add both a 'width' and a 'height' attribute to this image.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_26_jun_2013_151312_dinesh_bolkensteyn_wrote">on 26 Jun 2013, 15:13:12 Dinesh Bolkensteyn wrote:</h3>
<div class="paragraph">
<p>Let&#8217;s also apply to &lt;input type="image" /&gt; tags</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_8_jul_2013_181855_freddy_mallet_wrote">on 8 Jul 2013, 18:18:55 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>Is implemented by http://jira.codehaus.org/browse/SONARPLUGINS-3001</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_30_nov_2018_112144_nicolas_harraudeau_wrote">on 30 Nov 2018, 11:21:44 Nicolas Harraudeau wrote:</h3>
<div class="paragraph">
<p>Reverting the rule name to "Image tags should have "width" and "height" attributes" (removing the requirement for buttons) as this rule&#8217;s implementation does not check buttons and there has been no request or complains from community.</p>
</div>
<div class="paragraph">
<p>Note also that this rule is not enabled on Peach nor on Next, and it is not part of Sonar-Way. Thus the value of this rule seems quite dubious to me. Especially as it is quite common to define the width and height of images in CSS files.</p>
</div>
<div class="paragraph">
<p>\[~alexandre.gigleux] Shouldn&#8217;t we deprecate such a rule?</p>
</div>
</div>
</div>
</div>