Starting with Java 18 there is a possibility to write code snippets in javadoc comments. Previously the construction `<pre>{@code ... }</pre>` was used for this. While you still can use the old approch, snippets give you more possibilities, like easy highlighting, replacing by substring or regex, extracting to a file, linking, etc. 

== Noncompliant Code Example

[source,java]
----
/**
 * The code prints "Hello, World!"
 * <pre>{@code
 *   System.out.println("Hello, World!");
 * }</pre>
 */
public static void main(String... args) {
   ...
}
----

== Compliant Solution

[source,java]
----
/**
 * The code prints "Hello, World!"
 * {@snippet :
 *  public static void main(String... args) {
 *      System.out.println("Hello, World!");
 *  }
 * }
 */
public static void main(String... args) {
   ...
}
----

== See

Oracleâ€™s Programmer's Guide to Snippets: https://docs.oracle.com/en/java/javase/18/code-snippet/index.html[https://docs.oracle.com/en/java/javase/18/code-snippet/index.html]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

include::highlighting.adoc[]

endif::env-github,rspecator-view[]
