include::../description.adoc[]

== Noncompliant Code Example

----
if (x == 0) 
{
    DoSomething();
} 
else if (x == 1) 
{
    DoSomethingElse();
} 
----

== Compliant Solution

----
if (x == 0) 
{
    DoSomething();
} 
else if (x == 1) 
{
    DoSomethingElse();
} 
else 
{
    throw new InvalidOperationException();
}
----

== Exceptions

None

include::../see.adoc[]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
