== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,csharp]
----
if (x == 0) 
{
    DoSomething();
} 
else if (x == 1) 
{
    DoSomethingElse();
} 
----

=== Compliant solution

[source,csharp]
----
if (x == 0) 
{
    DoSomething();
} 
else if (x == 1) 
{
    DoSomethingElse();
} 
else 
{
    throw new InvalidOperationException();
}
----

=== Exceptions

None


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
