String literals written on multiple lines should not be used, especially since the new line characters are *not* part of the string anyway. The concatenation operator `+||+` should be used instead. Moreover, if trailing comments are present, they can be automatically removed by text editors and lead to unexpected results.

== Noncompliant Code Example

----
foo: proc options(main);
  put list ('Hello,      /* Noncompliant; trailing space is not readable and may be stripped */
world');
end;
----

== Compliant Solution

----
foo: proc options(main);
  put list ('Hello, ' ||
'world');
end;
----
