include::../description.adoc[]

== Noncompliant Code Example

----
for ($i = 0; $i < 42; $i++){}  // Empty on purpose or missing piece of code ?
----

include::../exceptions.adoc[]

ifdef::env-github,rspecator-view[]
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
