== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,php]
----
for ($i = 0; $i < 42; $i++){}  // Empty on purpose or missing piece of code ?
----

include::../exceptions.adoc[]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
