== How to fix it in Mcrypt

=== Code examples

This has been deprecated and then removed !

==== Noncompliant code example

[source,php,diff-id=1,diff-type=noncompliant]
----
----

==== Compliant solution

[source,php,diff-id=1,diff-type=compliant]
----
----

=== How does this work?


=== Pitfalls


=== Going the extra mile


