== How to fix it in OpenSSL

=== Code examples

==== Noncompliant code example

include::../../common/fix/aes-noncompliant-example.adoc[]

[source,php,diff-id=11,diff-type=noncompliant]
----
openssl_encrypt($plaintext, "BF-ECB", $key, $options=OPENSSL_RAW_DATA, $iv); // Noncompliant
----

==== Compliant solution

include::../../common/fix/aes-compliant-example.adoc[]

[source,php,diff-id=11,diff-type=compliant]
----
openssl_encrypt($plaintext, "aes-256-gcm", $key, $options=OPENSSL_RAW_DATA, $iv);
----

=== How does this work?

include::../../common/fix/fix.adoc[]
