== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,csharp]
----
void Notify(string company, string office = "QJZ") // Noncompliant
{
}
----

=== Compliant solution

[source,csharp]
----
void Notify(string company)
{
  Notify(company, "QJZ");
}
void Notify(string company, string office)
{
}
----

include::../exceptions.adoc[]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
