<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>The overloading mechanism should be used in place of optional parameters for several reasons:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>Optional parameter values are baked into the method call site code, thus, if a default value has been changed, all referencing assemblies need to be rebuilt, otherwise the original values will be used.</p>
</li>
<li>
<p>The Common Language Specification (CLS) allows compilers to ignore default parameter values, and thus require the caller to explicitly specify the values. For example, if you want to consume a method with default argument from another .NET compatible language (for instance C&#43;&#43;/CLI), you will have to provide all arguments. When using method overloads, you could achieve similar behavior as default arguments.</p>
</li>
<li>
<p>Optional parameters prevent muddying the definition of the function contract. Here is a simple example: if there are two optional parameters, when one is defined, is the second one still optional or mandatory?</p>
</li>
</ul>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-csharp" data-lang="csharp">void Notify(string company, string office = "QJZ") // Noncompliant
{
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-csharp" data-lang="csharp">void Notify(string company)
{
  Notify(company, "QJZ");
}
void Notify(string company, string office)
{
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_exceptions">Exceptions</h3>
<div class="paragraph">
<p>The rule ignores non externally visible methods.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Use the overloading mechanism instead of the optional parameters.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_6_oct_2015_121140_tamas_vajk_wrote">on 6 Oct 2015, 12:11:40 Tamas Vajk wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] The description is not correct. There are optional arguments in C# too.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_6_oct_2015_121248_ann_campbell_wrote">on 6 Oct 2015, 12:12:48 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>Thanks [~tamas.vajk]. Fixed!</p>
</div>
</div>
</div>
</div>