include::../description.adoc[]

== Noncompliant Code Example

----
func compute(a int, b int) {
	sum := a + b
	if  sum > 0 { } // Noncompliant; empty on purpose or missing piece of code?
	fmt.Println("Result:", sum)
}
----

== Compliant Solution

----
func compute(a int, b int) {
	sum := a + b
	if  sum > 0 {
		fmt.Println("Positive result")
	}
	fmt.Println("Result:", sum)
}
----

== Exceptions

When a block contains a comment, this block is not considered to be empty. `+for+` without `+init+` and `+post+` statements with empty blocks are ignored as well.
