== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,go]
----
func compute(a int, b int) {
	sum := a + b
	if  sum > 0 { } // Noncompliant; empty on purpose or missing piece of code?
	fmt.Println("Result:", sum)
}
----

=== Compliant solution

[source,go]
----
func compute(a int, b int) {
	sum := a + b
	if  sum > 0 {
		fmt.Println("Positive result")
	}
	fmt.Println("Result:", sum)
}
----

=== Exceptions

When a block contains a comment, this block is not considered to be empty. ``++for++`` without ``++init++`` and ``++post++`` statements with empty blocks are ignored as well.

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
