include::../summary.adoc[]

== Why is this an issue?

Jump statements (``++EXIT++``, ``++CONTINUE++``, ``++RETURN++``, ``++RAISE++``, and ``++RAISE_APPLICATION_ERROR++``), move control flow out of the current code block. So any statements that come after a jump are dead code. 


This rule detects for statements that follow:

* ``++EXIT++`` without a ``++WHEN++``
* ``++CONTINUE++`` without a ``++WHEN++``
* ``++RETURN++``
* ``++RAISE++``
* ``++RAISE_APPLICATION_ERROR++``

=== Noncompliant code example

[source,sql]
----
SET SERVEROUTPUT ON

BEGIN
  LOOP
    DBMS_OUTPUT.PUT_LINE('This will be printed out');
    EXIT;

    DBMS_OUTPUT.PUT_LINE('This will NEVER be printed out'); -- Non-Compliant
  END LOOP;
END;
/
----

=== Compliant solution

[source,sql]
----
SET SERVEROUTPUT ON

BEGIN
  LOOP
    DBMS_OUTPUT.PUT_LINE('This will be printed out');
    EXIT;
  END LOOP;
END;
/
----

include::../see.adoc[]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../highlighting.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
