<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule verifies that single-line comments are not located at the ends of lines of code. The main idea behind this rule is that in order to be really readable, trailing comments would have to be properly written and formatted (correct alignment, no interference with the visual structure of the code, not too long to be visible) but most often, automatic code formatters would not handle this correctly: the code would end up less readable. Comments are far better placed on the previous empty line of code, where they will always be visible and properly formatted.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-php" data-lang="php">$a = $b + $c; // This is a trailing comment that can be very very long</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-php" data-lang="php">// This very long comment is better placed before the line of code
$a = $b + $c;</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Move this trailing comment on the previous empty line.</p>
</div>
</div>
<div class="sect2">
<h3 id="_parameters">Parameters</h3>
<div class="sidebarblock">
<div class="content">
<div class="title">legalTrailingCommentPattern</div>
<div class="listingblock">
<div class="content">
<pre>^(//|#)\s*+[^\s]++$</pre>
</div>
</div>
<div class="paragraph">
<p>Pattern for text of trailing comments that are allowed. By default, comments containing only one word.</p>
</div>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_relates_to_s906">relates to: <a data-rspec-id="S906" class="rspec-auto-link">S906</a></h3>

</div>
<div class="sect2">
<h3 id="_on_11_jul_2013_172649_dinesh_bolkensteyn_wrote">on 11 Jul 2013, 17:26:49 Dinesh Bolkensteyn wrote:</h3>
<div class="paragraph">
<p>Implemented by https://jira.codehaus.org/browse/SONARJAVA-206</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_12_feb_2014_211520_freddy_mallet_wrote">on 12 Feb 2014, 21:15:20 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>Is implemented by http://jira.codehaus.org/browse/SONARPLUGINS-3461 for PHP</p>
</div>
</div>
</div>
</div>