<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule verifies that single-line comments are not located at the ends of lines of code. The main idea behind this rule is that in order to be really readable, trailing comments would have to be properly written and formatted (correct alignment, no interference with the visual structure of the code, not too long to be visible) but most often, automatic code formatters would not handle this correctly: the code would end up less readable. Comments are far better placed on the previous empty line of code, where they will always be visible and properly formatted.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="paragraph">
<p>With the default comment pattern <code>^'\s*\S+\s*$</code>, which ignores single word comments:</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-vbnet" data-lang="vbnet">Module Module1
  Sub Main()
    Console.WriteLine("Hello, world!") ' Noncompliant - My first program!
    Console.WriteLine("Hello, world!") ' CompliantOneWord
  End Sub
End Module</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-vbnet" data-lang="vbnet">Module Module1
  Sub Main()
    ' Compliant - My first program!
    Console.WriteLine("Hello, world!")
    Console.WriteLine("Hello, world!") ' CompliantOneWord
  End Sub
End Module</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Move this trailing comment on the previous empty line.</p>
</div>
</div>
<div class="sect2">
<h3 id="_parameters">Parameters</h3>
<div class="sidebarblock">
<div class="content">
<div class="title">legalCommentPattern</div>
<div class="listingblock">
<div class="content">
<pre>^'\s*\S+\s*$</pre>
</div>
</div>
<div class="paragraph">
<p>Pattern for text of trailing comments that are allowed.</p>
</div>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_7_sep_2016_175547_ann_campbell_wrote">on 7 Sep 2016, 17:55:47 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~tamas.vajk] IMO we need to remove this exception (and possibly rename the parameter for clarity?)</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_9_sep_2016_081309_tamas_vajk_wrote">on 9 Sep 2016, 08:13:09 Tamas Vajk wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] All the other languages have this exception, and the parameter name is also similar there. Why do you think we&#8217;d need to fix these issues?</p>
</div>
</div>
<div class="sect2">
<h3 id="_relates_to_s906">relates to: <a data-rspec-id="S906" class="rspec-auto-link">S906</a></h3>

</div>
<div class="sect2">
<h3 id="_on_11_jul_2013_172649_dinesh_bolkensteyn_wrote">on 11 Jul 2013, 17:26:49 Dinesh Bolkensteyn wrote:</h3>
<div class="paragraph">
<p>Implemented by https://jira.codehaus.org/browse/SONARJAVA-206</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_12_feb_2014_211520_freddy_mallet_wrote">on 12 Feb 2014, 21:15:20 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>Is implemented by http://jira.codehaus.org/browse/SONARPLUGINS-3461 for PHP</p>
</div>
</div>
</div>
</div>