== Why is this an issue?

``++alert(...)++`` as well as ``++confirm(...)++`` and ``++prompt(...)++`` can be useful for debugging during development, but in production mode this kind of pop-up could expose sensitive information to attackers, and should never be displayed.


=== Noncompliant code example

[source,javascript]
----
if(unexpectedCondition) {
  alert("Unexpected Condition");
}
----


include::../see.adoc[]


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
