<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when <code>DateTimeFormatter.ofPattern()</code> is called repeatedly instead of storing the formatter in a static constant.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Creating <code>DateTimeFormatter</code> instances repeatedly using <code>DateTimeFormatter.ofPattern()</code> is inefficient. Each call to <code>ofPattern()</code> involves parsing and compiling the pattern string, which has computational overhead.</p>
</div>
<div class="paragraph">
<p>Since <code>DateTimeFormatter</code> instances are immutable and thread-safe, they can be safely reused. Store them in <code>static final</code> constants to avoid repeated creation and improve performance.</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="paragraph">
<p>Repeated calls to <code>DateTimeFormatter.ofPattern()</code> can degrade application performance, especially in high-throughput scenarios. The pattern parsing and compilation overhead accumulates over time, leading to increased CPU usage and memory pressure.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Move the DateTimeFormatter creation to a static final constant. This ensures the formatter is created only once when the class is loaded and can be reused safely across all instances and threads.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">public class DateService {
    public String formatDate(LocalDateTime dateTime) {
        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss"); // Noncompliant
        return formatter.format(dateTime);
    }
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">public class DateService {
    private static final DateTimeFormatter FORMATTER = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss");

    public String formatDate(LocalDateTime dateTime) {
        return FORMATTER.format(dateTime);
    }
}</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>DateTimeFormatter (Java Platform SE 8) - <a href="https://docs.oracle.com/javase/8/docs/api/java/time/format/DateTimeFormatter.html">Official Java documentation for DateTimeFormatter, explaining its immutable and thread-safe nature</a></p>
</li>
</ul>
</div>
</div>
<div class="sect2">
<h3 id="_standards">Standards</h3>
<div class="ulist">
<ul>
<li>
<p>Java Performance Best Practices - <a href="https://www.oracle.com/java/technologies/javase/performance-best-practices.html">Oracle&#8217;s guidelines for Java performance optimization, including object reuse patterns</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>