== Why is this an issue?

Complex boolean expressions are hard to read and so to maintain.

=== Noncompliant code example

With the default threshold value of 3


[source,vbnet]
----
If ((condition1 AndAlso condition2) OrElse (condition3 AndAlso condition4)) AndAlso condition5) Then  'Noncompliant
  ...
End If
----

=== Compliant solution

[source,vbnet]
----
If ((MyFirstCondition() OrElse MySecondCondition()) AndAlso MyLastCondition()) Then
  ...
End If
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

=== Parameters

.max
****

----
3
----

Maximum number of allowed conditional operators in an expression
****


'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
