Complex boolean expressions are hard to read and so to maintain.

== Noncompliant Code Example

With the default threshold value of 3


----
If ((condition1 AndAlso condition2) OrElse (condition3 AndAlso condition4)) AndAlso condition5) Then  'Noncompliant
  ...
End If
----

== Compliant Solution

----
If ((MyFirstCondition() OrElse MySecondCondition()) AndAlso MyLastCondition()) Then
  ...
End If
----
