Having long `RUN` instruction on a single line reduce the visibility and make it harder to read and understand.

== Why is this an issue?

A too long `RUN` instruction reduce the code visibility. It is important to split it, if possible by grouping on different lines elements that stand together. +
The usage of the operator `&&` allow to chain multiple operator. Also, the default operator `\` allow to continue the `RUN` instruction on the next line. +
Another way is to use the here documents format, which allow to define multiple instruction together without any chaining operator.

== How to fix it

=== Code examples

==== Noncompliant code example

[source,docker,diff-id=1,diff-type=noncompliant]
----
RUN apt-get update && apt-get install -y package-bar package-baz package-foo
----

==== Compliant solution

[source,docker,diff-id=1,diff-type=compliant]
----
RUN <<EOF
apt-get update
apt-get install -y \
    package-bar \
    package-baz \
    package-foo
EOF
----

== Resources

=== Documentation

* Docker docs - https://docs.docker.com/build/building/best-practices/#run[Building best practices]
