Having long `RUN` instruction on a single line reduce the visibility and make it harder to read and understand.

== Why is this an issue?

In Dockerfiles, `RUN` instruction tend to be extensive and can rapidly become quite long, containing a lot of instruction with parameters and flags. +
In order to keep a good readability and help for future maintenance, it is important to split them in multiple lines. +
It is usually a good idea to group elements that belongs together on the same lines. +
The usage of the operator `&&` allow to chain multiple operator. Also, the default operator `\` allow to continue the `RUN` instruction on the next line. +
Another way is to use the here documents format, which allow to define multiple instruction together without any chaining operator.

== How to fix it

=== Code examples

==== Noncompliant code example

[source,docker,diff-id=1,diff-type=noncompliant]
----
RUN apt-get update && apt-get install -y package-bar package-baz package-foo package-eel
----

==== Compliant solution

[source,docker,diff-id=1,diff-type=compliant]
----
RUN <<EOF
apt-get update
apt-get install -y \
    package-bar \
    package-baz \
    package-foo \
    package-eel
EOF
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Parameters

.maxLength
****
_INT_

----
80
----

The maximum number of characters for a single-line RUN instruction.
****


'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]

== Resources

=== Documentation

* Docker docs - https://docs.docker.com/build/building/best-practices/#run[Building best practices]
