== Why is this an issue?

Multiplying ``++sizeof()++`` with ``++sizeof()++`` indicates a logic error.


=== Noncompliant code example

[source,cpp]
----
sizeof(a) * sizeof(b); // Noncompliant
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Review this sizeof multiplication.


'''
== Comments And Links
(visible only on this page)

=== on 27 Jul 2015, 14:07:48 Ann Campbell wrote:
\[~massimo.paladin] I know you've given me the entirety of the CPPCheck definition, but it's no much to go on. If it indicates a logic error, do you have any idea what may have been intended?

endif::env-github,rspecator-view[]
