== Why is this an issue?

Multiplying ``++sizeof()++`` with ``++sizeof()++`` indicates a logic error.


=== Noncompliant code example

[source,cpp]
----
sizeof(a) * sizeof(b); // Noncompliant
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
