<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>According to the SAP documentation:</p>
</div>
<div class="quoteblock">
<blockquote>
<div class="paragraph">
<p>System functions are only intended for internal usage. Incompatible changes and further development is possible at any time and without warning or notice.</p>
</div>
</blockquote>
</div>
<div class="paragraph">
<p>So calling system C functions using a <code>CALL</code> statement should be avoided.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-abap" data-lang="abap">CALL 'MULTIPLY' ID 'P1'  FIELD '9999'
                ID 'P2'  FIELD '9999'
                ID 'RES' FIELD RESULT.</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove this use of system function 'XXXX'.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_8_jan_2014_155508_freddy_mallet_wrote">on 8 Jan 2014, 15:55:08 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>See ABAP documentation : http://help.sap.com/abapdocu_702/en/abapcall-.htm</p>
</div>
</div>
</div>
</div>