In Javadoc code snippets introduced with Java 18, attributes' values can be quoted with single-quote (') characters or double-quote (") characters. Simple values, such as identifiers or numbers need not be quoted. According to Javadoc's specification, escape sequences are not supported in attribute value. Using them will therefore lead to unpredictable behaviors when parsing the snippet, at best, and "spurious markup" warnings being reported.

If there is a need to use double-quotes within the attribute already using double-quotes, consider using single-quotes to delimit the value instead of escaping quotes. Proceed the other way around in case of single-quote need.

== Noncompliant Code Example

[source,java]
----
/**
 * {@snippet :
 *   public static void main(String... args) {
 *    System.out.println("Hello World!");  // @link regex="\".+\"" target="java.lang.String"
 *   }
 * }
 */
void foo();

/**
 * {@snippet :
 *   public static void main(String... args) {
 *    System.out.println('c');  // @link regex='\'.+\'' target="java.lang.Character"
 *   }
 * }
 */
void bar();
----

== Compliant Solution

[source,java]
----
/**
 * {@snippet :
 *   public static void main(String... args) {
 *    System.out.println("Hello World!");  // @link regex='".+"' target="java.lang.String"
 *   }
 * }
 */
void foo();

/**
 * {@snippet :
 *   public static void main(String... args) {
 *    System.out.println('c');  // @link regex="'.+'" target="java.lang.Character"
 *   }
 * }
 */
void bar();
----

== See

Oracleâ€™s Programmer's Guide to Snippets: https://docs.oracle.com/en/java/javase/18/code-snippet/index.html[https://docs.oracle.com/en/java/javase/18/code-snippet/index.html]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

* In case of escaped double quotes: Use single-quotes to delimit the value instead of escaping double-quotes.
* In case of escaped single quotes: Use double-quotes to delimit the value instead of escaping single-quotes.

=== Highlighting

* Primary: attribute value including the quotes

=== Quickfix

Only possible if there is no mix of quotes and double quotes within the value (this should not compile on javadoc side anyway)

* In case of escaped double quotes: Replace starting and ending double-quotes by single-quotes, and remove escapes from double-quotes within the value.
* In case of escaped single quotes: Replace starting and ending single-quotes by double-quotes, and remove escapes from single-quotes within the value.

endif::env-github,rspecator-view[]
