== Why is this an issue?

When using Spring proxies, calling a method in the same class (e.g. ``++this.aMethod()++``) with an incompatible ``++@Transactional++`` requirement will result in runtime exceptions because Spring only "sees" the caller and makes no provisions for properly invoking the callee. 


Therefore, certain calls should never be made within the same class:

[frame=all]
[cols="^1,^1"]
|===
|From|To

| non-``++@Transactional++`` | MANDATORY, NESTED, REQUIRED, REQUIRES_NEW 
| MANDATORY | NESTED, NEVER, NOT_SUPPORTED, REQUIRES_NEW 
| NESTED | NESTED, NEVER, NOT_SUPPORTED, REQUIRES_NEW 
| NEVER | MANDATORY, NESTED, REQUIRED, REQUIRES_NEW 
| NOT_SUPPORTED | MANDATORY, NESTED, REQUIRED, REQUIRES_NEW 
| REQUIRED or ``++@Transactional++`` | NESTED, NEVER, NOT_SUPPORTED, REQUIRES_NEW 
| REQUIRES_NEW | NESTED, NEVER, NOT_SUPPORTED, REQUIRES_NEW 
| SUPPORTS | MANDATORY, NESTED, NEVER, NOT_SUPPORTED, REQUIRED, REQUIRES_NEW 
|===

=== Noncompliant code example

[source,java]
----

@Override
public void doTheThing() {
  // ...
  actuallyDoTheThing();  // Noncompliant
}

@Override
@Transactional
public void actuallyDoTheThing() {
  // ...
}
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

"xxx's" @Transactional requirement is incompatible with the one for this method.


=== Highlighting

* primary: method call
* secondary: 
** calling method's ``++@Transactional++`` annotation or, if none, calling method signature.
** called method's ``++@Transactional++`` annotation or signature
** message: "Incompatible method definition."


'''
== Comments And Links
(visible only on this page)

=== on 21 Nov 2014, 12:29:50 Freddy Mallet wrote:
I would tag this rule with the label "spring"

=== on 20 Jun 2018, 14:39:00 Alban Auzeill wrote:
@Ann I don't understand why there's a difference between ``++@Transactional++``  and ``++@Transactional(propagation = Propagation.REQUIRED)++`` (the default) ?

And I disagree with incompatible propagations list, this is my proposal:

||From||To||

| non-``++@Transactional++`` | MANDATORY, REQUIRED, REQUIRES_NEW |

| MANDATORY | NESTED, NEVER, NOT_SUPPORTED, REQUIRES_NEW |

| NESTED | MANDATORY, NESTED, NEVER, NOT_SUPPORTED, REQUIRED, REQUIRES_NEW |

| NEVER | MANDATORY, REQUIRED, REQUIRES_NEW |

| NOT_SUPPORTED | MANDATORY, REQUIRED, REQUIRES_NEW |

| REQUIRED or ``++@Transactional++`` | NESTED, NEVER, NOT_SUPPORTED, REQUIRES_NEW |

| REQUIRES_NEW | NESTED, NEVER, NOT_SUPPORTED REQUIRES_NEW |

| SUPPORTS | MANDATORY, NESTED, NEVER, NOT_SUPPORTED, REQUIRED, REQUIRES_NEW |


javadoc https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/transaction/annotation/Propagation.html[org.springframework.transaction.annotation.Propagation]

javadoc https://docs.oracle.com/javaee/7/api/javax/transaction/Transactional.html[javax.transaction.Transactional]

endif::env-github,rspecator-view[]
