As stated in the documentation, the methods https://docs.oracle.com/javase/8/docs/api/java/util/Objects.html#isNull-java.lang.Object-[Objects.isNull] and https://docs.oracle.com/javase/8/docs/api/java/util/Objects.html#nonNull-java.lang.Object-[Objects.nonNull] only exist to be used as https://docs.oracle.com/javase/8/docs/api/java/util/function/Predicate.html[Predicate]. When a predicate is not needed, a simple null check should be used, resulting in more concise and readable code.

This rule reports an issue when one of the two methods is not used as a Predicate.

== Noncompliant Code Example

[source,java]
----
if (Objects.isNull(object))

if (Objects.nonNull(object))
----

== Compliant Solution

[source,java]
----
if (object == null)

if (object != null)

list.stream().filter(Object::nonNull)
----

== See

* https://docs.oracle.com/javase/8/docs/api/java/util/Objects.html[Oracle Java 8, Objects]
