<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Naming conventions play a crucial role in maintaining code clarity and readability.
The uniqueness of bean names in Spring configurations is vital to the clarity and readability of the code.
When two beans share the same name within a configuration, it is not obvious to the reader which bean is being referred to.
This leads to potential misunderstandings and errors.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>To address this issue, ensure each bean within a configuration has a distinct and meaningful name.
Choose names that accurately represent the purpose or functionality of the bean.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">@Configuration
class Config {
  @Bean
  public User user() {
    return currentUser();
  }
  @Bean
  public User user(AuthService auth) { // Noncompliant
    return auth.user();
  }
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">@Configuration
class Config {
  @Bean
  public User user() {
    return currentUser();
  }
  @Bean
  public User userFromAuth(AuthService auth) {
    return auth.user();
  }
}</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p><a href="https://docs.spring.io/spring-framework/reference/core/beans/java/basic-concepts.html">Spring IO - Basic concepts: @Bean and @Configuration</a></p>
</li>
<li>
<p><a href="https://docs.spring.io/spring-framework/reference/core/beans/java/configuration-annotation.html">Spring IO - Using the @Configuration annotation</a></p>
</li>
<li>
<p><a href="https://docs.spring.io/spring-framework/reference/core/beans/java/bean-annotation.html">Spring IO - Using the @Bean annotation</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>