<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Cascade notation in dart is a convenient way of chaining several operations on the same object, while also having access to the instance methods and fields of the cascading object. With cascades you can write such code:</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-dart" data-lang="dart">var person = Person()
  ..firstName = 'John'
  ..lastName = 'Smith'
  ..age = 50;</code></pre>
</div>
</div>
<div class="paragraph">
<p>which will be the equivalent of:</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-dart" data-lang="dart">var person = Person();
person.firstName = 'John';
person.lastName = 'Smith';
person.age = 50;</code></pre>
</div>
</div>
<div class="paragraph">
<p>We can think of the cascade notation as a concise implementation of a builder pattern. Moreover, there&#8217;s a null-safe version of it <code>?..</code> that will only access members if the cascading object isn&#8217;t <code>null</code>.</p>
</div>
<div class="paragraph">
<p>While such notations looks appealing it doesn&#8217;t really make sense to use for a single cascade section. For example, <code>person..read();</code> seems to be an overkill. In this case usual <code>person.read()</code> is much more clear. Sometimes it may also become a source of bugs. Looks at this example:</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-dart" data-lang="dart">var person = Person()..getName(); // here the instance of Person will be returned

var name = Person().getName(); // here the name will be returned</code></pre>
</div>
</div>
<div class="paragraph">
<p>So be careful, when use cascading especially in combination with <code>var</code>. And always make sure the expected value is returned.</p>
</div>
<div class="paragraph">
<p>The other side of this notation os that it might be confused with a range operator <code>..</code> from other languages (like Kotlin). So using it without necessity will make code less readable and as a result less maintainable.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Just replace cascade notation (<code>..</code>) with a single dot (<code>.</code>) or make sure you assign the constructed object.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-dart" data-lang="dart">Person()..doSomething();</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-dart" data-lang="dart">var person = Person()..doSomething(); // if we want to return the Person instance</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_noncompliant_code_example_2">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-dart" data-lang="dart">Person()..doSomething();</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution_2">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-dart" data-lang="dart">Person().doSomething();</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Dart Docs - <a href="https://dart.dev/tools/linter-rules/avoid_single_cascade_in_expression_statements">Dart Linter rule - avoid_single_cascade_in_expression_statements</a></p>
</li>
<li>
<p>Dart Docs - <a href="https://dart.dev/language/operators#cascade-notation">Cascade notation</a></p>
</li>
</ul>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="ulist">
<ul>
<li>
<p>Unnecessary cascade expression.</p>
</li>
</ul>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>Cascade expression</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
</div>
</div>