include::../../../shared_content/secrets/description.adoc[]

== Why is this an issue?

include::../../../shared_content/secrets/rationale.adoc[]

=== What is the potential impact?

A cryptographic private key is a piece of sensitive information that is used in
asymmetric cryptography. They are used in conjunction with public keys to secure
communications and authenticate digital signatures.

Private keys can be used to achieve two main cryptographic operations,
encryption or digital signature. Those operations are the basis of multiple
higher-level security mechanisms such as:

* User authentication.
* Servers authentication, for example in the X509 trust model.
* E-mail encryption.

Disclosing a cryptographic private key to an unintended audience can have severe
security consequences. The exact impact will vary depending on the role of the
key and the assets it protects.

For example, if the key is used in conjunction with an X509 certificate to
authenticate a web server as part of TLS communications, attackers will be able
to impersonate that server. This leads to Man-In-The-Middle-Attacks that would
affect both the confidentiality and integrity of the communications from clients
to that server.

If the key was used as part of e-mail protocols, attackers might be able to send
e-mails on behalf of the key owner or decrypt previously encrypted emails. This
might lead to sensitive information disclosure and reputation loss.

== How to fix it

include::../../../shared_content/secrets/fix/revoke.adoc[]

In most cases, if the key is used as part of a larger trust model (X509, PGP,
etc), it is necessary to issue and publish a revocation certificate. Doing so
will ensure that all people and assets that rely on this key for security
operations are aware of its compromise state and stop trusting it.

include::../../../shared_content/secrets/fix/vault.adoc[]

=== Code examples

==== Noncompliant code example

[source,python,diff-id=1,diff-type=noncompliant]
----
private_key = "-----BEGIN EC PRIVATE KEY-----" \
    "MF8CAQEEGEfVxjrMPigNhGP6DqH6DPeUZPbaoaCCXaAKBggqhkjOPQMBAaE0AzIA" \
    "BCIxho34upZyXDi/AUy/TBisGeh4yKJN7pit9Z+nKs4QajVy97X8W9JdySlbWeRt" \
    "2w==" \
    "-----END EC PRIVATE KEY-----"
----

==== Compliant solution

[source,python,diff-id=1,diff-type=compliant]
----
with open("/path/to/private.key","r") as key_file:
    private_key = key_file.read()
----

//=== How does this work?

//=== Pitfalls

//=== Going the extra mile

== Resources

include::../../../shared_content/secrets/resources/standards.adoc[]

//=== Benchmarks
