<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when an object used as a sequence index does not define an <code>__index__</code> method.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Objects can be used as sequence indexes to access a specific element from the sequence, through the following syntax:</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">my_list = [1, 2, 3, 4]
x = 1
print(my_list[x])  # This will print 2</code></pre>
</div>
</div>
<div class="paragraph">
<p>Whenever an object is used as a sequence index, the Python interpreter calls its <code>__index__</code> method to compute the index that needs to be accessed from the sequence.</p>
</div>
<div class="paragraph">
<p>Any object can be used as sequence index, as long as it defines an <code>__index__</code> method that returns an <code>int</code>. Most commonly, sequence indexes are simply integers.</p>
</div>
<div class="paragraph">
<p>Similarly, sequences can be sliced through the following syntax:</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">my_list = [1, 2, 3, 4]
x = 1
print(my_list[1:3]) # This will print [2, 3]</code></pre>
</div>
</div>
<div class="paragraph">
<p>If an invalid object is used as a sequence index, a <code>TypeError</code> will be raised.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Make sure to use an object that defines an <code>__index__</code> method as sequence indexes.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-text" data-lang="text">def foo():
    my_list = ["spam", "eggs"]
    x = my_list["spam"]  # Noncompliant: a str is not a valid index</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-text" data-lang="text">def foo():
    my_list = ["spam", "eggs"]
    x = my_list[0]</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="paragraph">
<p>Python Documentation - <a href="https://docs.python.org/3/library/operator.html#operator.<em>index</em>"><em>index</em> method</a></p>
</div>
</div>
</div>
</div>