== Why is this an issue?

Although unnecessary ``++using++`` won't change anything to the produced application, removing them:

* Will help readability and maintenance.
* Will help reduce the number of items in the IDE auto-completion list when coding.
* May avoid some name collisions.
* May improve compilation time because the compiler has fewer namespaces to look-up when it resolves types.
* The build will fail if this namespace is removed from the project.

=== Noncompliant code example

[source,csharp]
----
using System.Collections.Generic; // Noncompliant - unnecessary using

namespace Foo
{
    public class Bar
    {
        public Bar(string path)
        {
            File.ReadAllLines(path);
        }
    }
}
----

=== Compliant solution

[source,csharp]
----
using System.IO;

namespace Foo
{
    public class Bar
    {
        public Bar(string path)
        {
            File.ReadAllLines(path);
        }
    }
}
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

* Remove this unused "using".


'''
== Comments And Links
(visible only on this page)

=== on 23 Sep 2019, 16:37:14 Nicolas Harraudeau wrote:
*OUT OF SCOPE*


Duplicate imports are out of scopes as Roslyn already raises an issue in this case. As Roslyn issues are enabled by default thisÂ would simply create duplicates.

=== on 18 Dec 2020, 10:06:15 Andrei Epure wrote:
We are removing this rule from SonarWay due to its performance issues. After the rule gets re-designed to avoid perf issues, (see https://github.com/SonarSource/sonar-dotnet/issues/3761[#3761]), we should bring it back to SonarWay.

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
