<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Dynamic memory allocation &amp; deallocation (e.g. <code>malloc</code> / <code>free</code>) is somewhat expensive. This is particularly true when it happens in a loop. It is good practice to allocate and deallocate memory only when it is needed.</p>
</div>
<div class="paragraph">
<p>This rule raises an issue when one, or more, execution path results in unused allocated memory.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">int foo(int val) {
  int ret;
  char *buf = malloc(100);
  if (!isPresent(val)) {
    snprintf(buf, 100, "msg %d", val);
    addToList(buf);
    ret = 1;
  } else { // Noncompliant: buf unused in this branch
    ret = 2;
  }
  free(buf);
  return ret;
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">int foo(int val) {
  if (!isPresent(val)) {
    char *buf = malloc(100);
    snprintf(buf, 100, "msg %d", val);
    addToList(buf);
    free(buf);
    return 1;
  }
  return 2;
}</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Move this allocation &amp; deallocation into the proper branch(es).</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>Primary: Allocation</p>
</div>
<div class="paragraph">
<p>Secondary: Deallocation</p>
</div>
</div>
</div>
</div>