<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Hibernate&#8217;s lazy loading allows you to retrieve just the data of the current class without being forced to load all its related classes. For instance with lazy loading, you can pull up an instance of a <code>Lecture</code> <code>@Entity</code> without being forced to load all its <code>Student</code>s.</p>
</div>
<div class="paragraph">
<p>But that&#8217;s only if you&#8217;re storing the <code>Student</code>s in a collection. Store them in an array instead, and the benefits of lazy loading are no longer available.</p>
</div>
<div class="paragraph">
<p>This rule raises an issue on each array in <code>@Entity</code> classes.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">@Entity
public class Lecture {

  @OneToMany
  private Student [] attendees;  // Noncompliant
  // ...
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">@Entity
public class Lecture {

  @OneToMany
  private List&lt;Student&gt; attendees;
  // ...
}</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Convert this array to a collection.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="ulist">
<ul>
<li>
<p>primary: array name</p>
</li>
</ul>
</div>
</div>
</div>
</div>