<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises issues when <code>trio.sleep(0)</code> or <code>anyio.sleep(0)</code> is used instead of the more explicit and descriptive <code>trio.lowlevel.checkpoint()</code> or <code>anyio.lowlevel.checkpoint()</code>.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>When using async libraries like <code>trio</code> or <code>anyio</code>, developers sometimes use <code>sleep(0)</code> as a technique to yield control back to the event loop, allowing other pending tasks to run. While this approach technically works, it obscures the actual intent.</p>
</div>
<div class="paragraph">
<p>The <code>sleep(0)</code> pattern is confusing because it suggests waiting for a specific duration (which happens to be 0 seconds), when the actual intention is to create a checkpoint in the code where the current task can yield control to other tasks without any intended delay.</p>
</div>
<div class="paragraph">
<p>Both <code>trio</code> and <code>anyio</code> provide dedicated functions for this exact purpose: <code>trio.lowlevel.checkpoint()</code> and <code>anyio.lowlevel.checkpoint()</code>. These functions clearly communicate the intent to create a checkpoint for task switching.</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="ulist">
<ul>
<li>
<p><strong>Readability</strong>: Code using <code>sleep(0)</code> is less self-documenting and can be confusing to other developers who need to know the non-obvious intent of the code.</p>
</li>
<li>
<p><strong>Maintainability</strong>: The intent behind <code>sleep(0)</code> is not immediately clear to developers reviewing or maintaining the code.</p>
</li>
</ul>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it_in_trio">How to fix it in Trio</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Replace <code>trio.sleep(0)</code> with <code>trio.lowlevel.checkpoint()</code>.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">import trio

async def main():
    await trio.sleep(0)  # Noncompliant

trio.run(main)</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">import trio

async def main():
    await trio.lowlevel.checkpoint()  # Compliant

trio.run(main)</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it_in_anyio">How to fix it in AnyIO</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Replace <code>anyio.sleep(0)</code> with <code>anyio.lowlevel.checkpoint()</code>.</p>
</div>
<div class="sect2">
<h3 id="_code_examples_2">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example_2">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">import anyio

async def main():
    await anyio.sleep(0)  # Noncompliant

anyio.run(main)</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution_2">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">import anyio

async def main():
    await anyio.lowlevel.checkpoint()  # Compliant

anyio.run(main)</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Trio Documentation - <a href="https://trio.readthedocs.io/en/stable/reference-lowlevel.html#trio.lowlevel.checkpoint">trio.lowlevel.checkpoint</a></p>
</li>
<li>
<p>AnyIO Documentation - <a href="https://anyio.readthedocs.io/en/stable/api.html#anyio.lowlevel.checkpoint">anyio.lowlevel.checkpoint</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Use <code>trio.lowlevel.checkpoint()</code> instead of <code>trio.sleep(0)</code>.
Use <code>anyio.lowlevel.checkpoint()</code> instead of <code>anyio.sleep(0)</code>.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>The <code>sleep(0)</code> call.
Secondary location on the async keywork of the function definition.</p>
</div>
</div>
</div>
</div>