<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Collection classes from the <code>java.util.concurrent</code> package have their own concurrency control mechanisms. A concurrent collection is thread-safe, but not governed by a single exclusion lock, therefore using an instance of such a class for synchronization is, at best unnecessary and at worst likely to have unintended consequences.</p>
</div>
<div class="paragraph">
<p>This rule raises an issue when a synchronization lock is used on an instance of one of the collection classes from the <code>java.util.concurrent</code> package.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">public class Foo {
  private ConcurrentHashMap&lt;String,String&gt; map = new ConcurrentHashMap&lt;&gt;();

  public void bar() {
    synchronized(map) {  // Noncompliant
      map.put("foo", "bar");
    }
    // Do something ...
  }
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">public class Foo {
  private ConcurrentHashMap&lt;String,String&gt; map = new ConcurrentHashMap&lt;&gt;();

  public void bar() {
    map.put("foo", "bar");
    // Do something ...
  }
}</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Either remove this "synchronized" or change its target</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>The full synchronized statement</p>
</div>
</div>
</div>
</div>