=== Noncompliant code example

[source,text]
----
for (element = list.first; element != null; element = element->next) { // First termination condition
  if (!matches(element->value)) {                                      // Second termination condition
    break; // Noncompliant
  }

  /* ... */
}
----
