The use of C headers and therefore C functions in a {cpp} program, is sometimes necessary, but should be avoided in favor of {cpp} headers and functions.


== Noncompliant Code Example

----
#include <string.h>
----


== Compliant Solution

----
#include <cstring>
#include <string>
----


== See

* MISRA {cpp}:2008, 18-0-1 - The C library shall not be used.


ifdef::rspecator-view[]
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::rspecator-view[]
