<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>An issue is raised when a FastAPI endpoint uses the generic <code>@app.route()</code> or <code>@router.route()</code> decorator with a <code>methods</code> parameter.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>FastAPI provides specific decorators for each HTTP method (<code>@app.get()</code>, <code>@app.post()</code>, <code>@app.put()</code>, <code>@app.delete()</code>, etc.) that are more explicit and idiomatic than using the generic <code>@app.route()</code> decorator with a <code>methods</code> parameter.</p>
</div>
<div class="paragraph">
<p>The generic <code>route()</code> decorator comes from Flask, where it&#8217;s the standard way to define endpoints. However, FastAPI was designed with a different approach that emphasizes clarity and explicitness.</p>
</div>
<div class="paragraph">
<p>Using specific HTTP method decorators offers several advantages:</p>
</div>
<div class="ulist">
<ul>
<li>
<p><strong>Immediate clarity</strong>: The HTTP method is visible at a glance without needing to read the decorator&#8217;s parameters</p>
</li>
<li>
<p><strong>Less verbose</strong>: The specific decorators require less code and fewer parameters</p>
</li>
<li>
<p><strong>Framework conventions</strong>: This is the standard, documented way to define endpoints in FastAPI</p>
</li>
<li>
<p><strong>Better tooling support</strong>: IDEs and linters can provide better autocomplete and validation with the specific decorators</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>While both approaches work functionally, using the generic <code>route()</code> decorator in FastAPI code suggests unfamiliarity with the framework&#8217;s conventions and can make the codebase harder to read for developers who expect idiomatic FastAPI patterns.</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="paragraph">
<p>Using the generic route decorator instead of specific HTTP method decorators reduces code readability and makes the codebase less maintainable. Developers need to read decorator parameters to understand what HTTP method an endpoint accepts, rather than seeing it immediately in the decorator name.</p>
</div>
<div class="paragraph">
<p>This pattern also signals to other developers that the code may not follow FastAPI best practices, potentially leading to inconsistent coding styles across the project.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it_in_fastapi">How to fix it in FastAPI</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Replace the generic <code>@app.route()</code> decorator with the specific HTTP method decorator that matches the method specified in the <code>methods</code> parameter. FastAPI provides decorators for all standard HTTP methods: <code>@app.get()</code>, <code>@app.post()</code>, <code>@app.put()</code>, <code>@app.delete()</code>, <code>@app.patch()</code>, <code>@app.options()</code>, <code>@app.head()</code>, and <code>@app.trace()</code>.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">@app.route("/users", methods=["GET"])  # Noncompliant
def get_users():
    return {"users": []}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">@app.get("/users")
def get_users():
    return {"users": []}</code></pre>
</div>
</div>
<div class="paragraph">
<p>The same principle applies to POST endpoints. Replace <code>@app.route()</code> with <code>@app.post()</code>.</p>
</div>
</div>
<div class="sect3">
<h4 id="_noncompliant_code_example_2">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">@app.route("/users", methods=["POST"])  # Noncompliant
def create_user(user: User):
    return {"user": user}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution_2">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">@app.post("/users")
def create_user(user: User):
    return {"user": user}</code></pre>
</div>
</div>
<div class="paragraph">
<p>For PUT, DELETE, and other HTTP methods, use the corresponding specific decorator.</p>
</div>
</div>
<div class="sect3">
<h4 id="_noncompliant_code_example_3">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">@app.route("/items/{item_id}", methods=["DELETE"])  # Noncompliant
def delete_item(item_id: int):
    return {"deleted": item_id}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution_3">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">@app.delete("/items/{item_id}")
def delete_item(item_id: int):
    return {"deleted": item_id}</code></pre>
</div>
</div>
<div class="paragraph">
<p>The same pattern applies when using <code>APIRouter</code> instead of the main <code>FastAPI</code> app instance.</p>
</div>
</div>
<div class="sect3">
<h4 id="_noncompliant_code_example_4">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">router = APIRouter()

@router.route("/items", methods=["GET"])  # Noncompliant
def list_items():
    return {"items": []}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution_4">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">router = APIRouter()

@router.get("/items")
def list_items():
    return {"items": []}</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>FastAPI - Path Operation Decorators - <a href="https://fastapi.tiangolo.com/tutorial/first-steps/#path-operation-decorator">Official FastAPI documentation explaining the use of specific HTTP method decorators</a></p>
</li>
<li>
<p>flake8-fastapi - CF001 Rule - <a href="https://pypi.org/project/flake8-fastapi/#cf001---route-decorator-error">Documentation for the flake8-fastapi plugin rule that detects this pattern</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>