<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when the <code>__init__</code> method of a class contains a <code>return</code> or a <code>yield</code> statement.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>By contract, every Python function returns something, even if it is the <code>None</code> value, which can be returned implicitly by omitting the <code>return</code> statement, or explicitly.</p>
</div>
<div class="paragraph">
<p>The <code>__init__</code> method is required to return <code>None</code>. A <code>TypeError</code> will be raised if the <code>__init__</code> method either yields or returns any expression other than <code>None</code>. While explicitly returning an expression that evaluates to <code>None</code> will not raise an error, it is considered bad practice.</p>
</div>
<div class="paragraph">
<p>To fix this issue, make sure that the <code>__init__</code> method does not contain any return statement.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">class MyClass(object):
    def __init__(self):
        self.message = 'Hello'
        return self  # Noncompliant: a TypeError will be raised</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">class MyClass(object):
    def __init__(self):
        self.message = 'Hello'</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p><a href="https://docs.python.org/3/reference/datamodel.html#object.__init__">The <code>__init__</code> method</a></p>
</li>
</ul>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="ulist">
<ul>
<li>
<p>Remove this yield statement.</p>
</li>
<li>
<p>Remove this return value.</p>
</li>
</ul>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_relates_to_s5654">relates to: <a data-rspec-id="S5654" class="rspec-auto-link">S5654</a></h3>

</div>
<div class="sect2">
<h3 id="_on_18_mar_2015_175542_ann_campbell_wrote">on 18 Mar 2015, 17:55:42 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>-E0100: <em>init</em> method is a generator</p>
</div>
<div class="paragraph">
<p>-E0101: Explicit return in <em>init</em></p>
</div>
</div>
</div>
</div>