<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>An <code>ObjectOutputStream</code> writes primitive data types and graphs of Java objects to an <code>OutputStream</code>.
The objects can be read (reconstituted) using an <code>ObjectInputStream</code>.</p>
</div>
<div class="paragraph">
<p>When <code>ObjectOutputStream</code> is used with files opened in append mode, it can cause data corruption and unexpected behavior.
This is because when <code>ObjectOutputStream</code> is created, it writes metadata to the output stream, which can conflict with the existing
metadata when the file is opened in append mode. This can lead to errors and data loss.</p>
</div>
<div class="paragraph">
<p>When used with serialization, an <code>ObjectOutputStream</code> first writes the serialization stream header. This header should appear
once per file at the beginning.
When you&#8217;re trying to read your object(s) back from the file, only the first one will be read successfully, and a <code>StreamCorruptedException</code>
will be thrown after that.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Open the file to use the default action (writes stream header).</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-kotlin" data-lang="kotlin">val fos = FileOutputStream(fileName, true) // fos opened in append mode
val out = ObjectOutputStream(fos) // Noncompliant</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-kotlin" data-lang="kotlin">val fos = FileOutputStream(fileName)
val out = ObjectOutputStream(fos)</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>