== Why is this an issue?

While most script engines support function declarations within blocks, from browser to browser, the implementations are inconsistent with each other.


=== Noncompliant code example

[source,javascript]
----
if (x) {
  function foo() {} //foo is hoisted in Chrome, Firefox and Safari, but not in Edge.
}
----


=== Compliant solution

[source,javascript]
----
if (x) {
  const foo = function() {}
}
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

endif::env-github,rspecator-view[]
