<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>The use of <code>virtual</code> base classes can introduce a number of undefined and potentially confusing behaviours. Therefore, a base class shall only be declared <code>virtual</code> if that base class is to be used as a common base class in a diamond hierarchy.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">class A {};
class B1: public virtual A {}; // Compliant, A is a common base for C
class B2: public virtual A {}; // Compliant, A is a common base for C
class C: public B1, B2 {};
class D: public virtual A {}; // Noncompliant, D is not part of a diamond-shaped hierarchy.</code></pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p>MISRA C&#43;&#43;:2008, 10-1-2 - A base class shall only be declared virtual if it is used in a diamond hierarchy.</p>
</li>
</ul>
</div>
<hr>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove the "virtual" specifier on class "xxx".</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_16_oct_2014_132154_ann_campbell_wrote">on 16 Oct 2014, 13:21:54 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~samuel.mercier] please:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>fill in the appropriate reference field(s).</p>
</li>
<li>
<p>provide a See section.</p>
</li>
<li>
<p>use double curly braces around code in the description</p>
</li>
<li>
<p>use the standard section titles</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>Also, since the description references undefined behavior, I wouldn&#8217;t tie this to Maintainability, but Reliability instead.</p>
</div>
</div>
</div>
</div>