== Why is this an issue?

The ``++for of++`` statement can only iterate over objects that have an ``++iterator++`` property.


=== Noncompliant code example

[source,javascript]
----
let obj = {};
for(let element of obj) {   // Noncompliant
  ...
}
----


=== Compliant solution

[source,javascript]
----
let array = [];
for(let element of array) {
  ...
}
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Use an Iterable in this "for of" loop


=== Highlighting

Primary: for keyword

Secondary: expression


endif::env-github,rspecator-view[]
