Various package and software management applications require manual input for execution confirmation by default.
This confirmation is usually required when installing, updating, or removing programs and packages.
To execute the command without manual input, a general consent can be given.

== Why is this an issue?

If such a command is part of a script that is executed automatically, the execution is aborted because there is no confirming manual input.
Instructions, such as installing programs or updating them, cannot be performed in an automated way.
This applies, among others, to the advanced package tool, 'apt-get'.

== How to fix it

=== Code examples

==== Noncompliant code example

[source,bash,diff-id=1,diff-type=noncompliant]
----
RUN apk add ca-certificates
RUN apt-get install ca-certificates
RUN aptitude install ca-certificates
----

==== Compliant solution

[source,bash,diff-id=1,diff-type=compliant]
----
RUN apk -y add ca-certificates
RUN apt-get install -y ca-certificates
RUN aptitude -y install ca-certificates
----

=== How does this work?

If the `-y` flag set, no manual input is expected, and it is assumed that there is general consent.
For `apk` and `apt-get` also the long versions `--yes` and `--assume-yes` exist.
For `aptitude` also the long version `--assume-yes` exist.

== Resources
=== Documentation

* https://linux.die.net/man/8/apt-get[apt-get - Linux man page]
* https://docs.alpinelinux.org/user-handbook/0.1a/Working/apk.html[apk - Alpine Package Keeper]
* https://wiki.debian.org/Aptitude[aptitude - Linux aptitude command]
]
* https://docs.docker.com/engine/reference/builder/#run[RUN - Docker reference]
* https://docs.docker.com/develop/develop-images/dockerfile_best-practices/#run[RUN - Best practices for writing Dockerfiles]
