Various package and software management applications require manual input for execution confirmation by default.
This confirmation is usually required when installing, updating, or removing programs and packages.
General consent can be given to execute the command without manual input.

== Why is this an issue?

Suppose a package manager invocation is part of a script that is executed automatically, and non-interactive mode is not enabled.
Then, execution is aborted because there is no confirming manual input.
As a result, instructions, such as installation or update of packages, cannot be performed in an automated way.
This applies, among others, to the package manager used in Debian-based systems, Advanced Package Tool (APT).

== How can I fix it?

=== Noncompliant Code Example

[source,docker,diff-id=1,diff-type=noncompliant]
----
RUN apt-get install ca-certificates
RUN aptitude install ca-certificates
RUN apt install ca-certificates
----

=== Compliant Solution

[source,docker,diff-id=1,diff-type=compliant]
----
RUN apt-get install -y ca-certificates
RUN aptitude -y install ca-certificates
RUN apt -y install ca-certificates
----

=== How does this work?

If the `-y` flag is set, no manual input is expected, and the package manager can run non-interactively.
For `apt` and `apt-get`, the long versions `--yes` and `--assume-yes` also exist.
For `aptitude`, the long version `--assume-yes` exists.

== Resources
=== Documentation

* https://linux.die.net/man/8/apt[apt - Linux man page]
* https://linux.die.net/man/8/apt-get[apt-get - Linux man page]
* https://wiki.debian.org/Aptitude[aptitude - Linux aptitude command]
]
* https://docs.docker.com/engine/reference/builder/#run[RUN - Docker reference]
* https://docs.docker.com/develop/develop-images/dockerfile_best-practices/#run[RUN - Best practices for writing Dockerfiles]

ifdef::env-github,rspecator-view[]
'''
== Implementation Specification
(visible only on this page)

=== Message

Add a consent flag so that this command doesn't require user confirmation.

=== Highlighting

Highlight the command where a package manager is executed.

'''
endif::env-github,rspecator-view[]
