include::../description.adoc[]

include::../why.adoc[]

== How to fix it

=== Code examples

==== Noncompliant code example

[source,docker,diff-id=1,diff-type=noncompliant]
----
RUN apt-get install ca-certificates
RUN aptitude install ca-certificates
RUN apt install ca-certificates
RUN gdebi /tmp/package.deb
----

include::../non-compliant-desc.adoc[]

==== Compliant solution

[source,docker,diff-id=1,diff-type=compliant]
----
RUN apt-get -y install ca-certificates
RUN aptitude -y install ca-certificates
RUN apt -y install ca-certificates
RUN gdebi --n /tmp/package.deb
----

include::../compliant-desc.adoc[]

include::../how.adoc[]

== Resources
=== Documentation

include::../general-links.adoc[]

* https://docs.docker.com/engine/reference/builder/#run[RUN - Docker reference]
* https://docs.docker.com/develop/develop-images/dockerfile_best-practices/#run[RUN - Best practices for writing Dockerfiles]

ifdef::env-github,rspecator-view[]
'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../highlighting.adoc[]

'''
endif::env-github,rspecator-view[]
