include::../description.adoc[]

== Noncompliant Code Example

[source,vbnet]
----
Public Sub SetName(name As String)
  name = name
End Sub
----

== Compliant Solution

[source,vbnet]
----
Public Sub SetName(name As String)
  Me.name = name
End Sub
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
