== Why is this an issue?

include::../description.adoc[]

=== Code examples

==== Noncompliant code example

[source,vbnet,diff-id=1,diff-type=noncompliant]
----
Public Sub SetName(name As String)
  name = name ' Noncompliant
End Sub
----

==== Compliant solution

[source,diff-id=1,diff-type=compliant]
----
Public Sub SetName(name As String)
  Me.name = name ' Compliant
End Sub
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
