== Why is this an issue?

Returning the unit type from a closure expecting an `Ord` type is likely a mistake caused by an extra semi-colon. This is because the unit type implements `Ord` but doesn't serve the intended sorting purpose, making the code misleading and potentially incorrect.

=== Code examples

==== Noncompliant code example

[source,rust,diff-id=1,diff-type=noncompliant]
----
let mut twins = vec![(1, 1), (2, 2)];
twins.sort_by_key(|x| { x.1; }); // Noncompliant: Closure returns unit type due to unnecessary semi-colon.
----

==== Compliant solution

[source,rust,diff-id=1,diff-type=compliant]
----
let mut twins = vec![(1, 1), (2, 2)];
twins.sort_by_key(|x| x.1); // Compliant: Closure correctly returns an `Ord` type.
----

== Resources
=== Documentation

* Clippy Lints - https://rust-lang.github.io/rust-clippy/master/index.html#unit_return_expecting_ord
