By default, the WordPress administrator and editor roles can add unfiltered HTML content in various places, such as post content. This includes the capability to add JavaScript code. 

If an account with such a role gets hijacked, this capability can be used to plant malicious JavaScript code that gets executed whenever somebody visits the website.

This default behavior can be deactivated by setting the constant 
DISALLOW_UNFILTERED_HTML to "true" in the WordPress configuration.

== Ask Yourself Whether

* You really need the possibility to add unfiltered HTML with editor or administrator roles.
* There's a chance that the accounts of authorized users get compromised.

== Noncompliant Code Example

----
define( 'DISALLOW_UNFILTERED_HTML', false ); // sensitive
----

== Compliant Solution

----
define( 'DISALLOW_UNFILTERED_HTML', true );
----
