== Why is this an issue?

Use of the bang (``++!++``) operator leads to late binding and results in inefficient code. Use the slightly more verbose dot (``++.++``) notation instead.


=== Noncompliant code example

[source,vb6]
----
MsgBox ObjectWithDefaultProperty!Param
----


=== Compliant solution

[source,vb6]
----
MsgBox ObjectWithDefaultProperty.DefaultProperty("Param")
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

endif::env-github,rspecator-view[]
