Use of the bang (``++!++``) operator leads to late binding and results in inefficient code. Use the slightly more verbose dot (``++.++``) notation instead.


== Noncompliant Code Example

----
MsgBox ObjectWithDefaultProperty!Param
----


== Compliant Solution

----
MsgBox ObjectWithDefaultProperty.DefaultProperty("Param")
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

endif::env-github,rspecator-view[]
