<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Shared coding conventions allow teams to collaborate efficiently. For maximum readability, this rule checks that the levels, names and PICTURE clauses for all items of the same level and which are subordinate to the same item start in the same column.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cobol" data-lang="cobol">01  ZONE1.
    03  ZONE2  PIC X(10).
    03   ZONE3 PIC X(10).    *&gt; Noncompliant; name out of line
     03 ZONE4  PIC X(10).    *&gt; Noncompliant; level out of line
    03  ZONE5   PIC X(10).   *&gt; Noncompliant; PIC out of line</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cobol" data-lang="cobol">01  ZONE1.
    03 ZONE2  PIC X(10).
    03 ZONE3  PIC X(10).
    03 ZONE4  PIC X(10).
    03 ZONE5  PIC X(10).</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="ulist">
<ul>
<li>
<p>Edit this set of data item declarations with the same level so that all (levels|"PIC" clauses|names) start in the same column.</p>
</li>
<li>
<p>Level 01 items should begin in column X</p>
</li>
</ul>
</div>
</div>
<div class="sect2">
<h3 id="_parameters">Parameters</h3>
<div class="sidebarblock">
<div class="content">
<div class="title">level_01_column</div>
<div class="paragraph">
<p>The column in which level 01 items should start. Leave blank to allow level 01 items to start in any column</p>
</div>
</div>
</div>
<div class="sidebarblock">
<div class="content">
<div class="title">check_name_alignment</div>
<div class="paragraph">
<p>Whether to check the alignment of names for items of the same level.</p>
</div>
</div>
</div>
<div class="sidebarblock">
<div class="content">
<div class="title">check_level_alignment</div>
<div class="paragraph">
<p>Whether to check the alignment of level indicators for items of the same level.</p>
</div>
</div>
</div>
<div class="sidebarblock">
<div class="content">
<div class="title">check_pic_alignment</div>
<div class="paragraph">
<p>Whether to check the alignment of "PIC" clauses for items of the same level.</p>
</div>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_is_duplicated_by_s1730">is duplicated by: <a data-rspec-id="S1730" class="rspec-auto-link">S1730</a></h3>

</div>
<div class="sect2">
<h3 id="_is_duplicated_by_s1802">is duplicated by: <a data-rspec-id="S1802" class="rspec-auto-link">S1802</a></h3>

</div>
</div>
</div>