== Why is this an issue?

include::../description.adoc[]

== How to fix it

=== Code examples

The examples below enforce the following regular expression ``++^[a-z][a-zA-Z0-9]*$++``:

==== Noncompliant code example

[source,cpp,diff-id=1,diff-type=noncompliant]
----
class MyClass {
  int my_field; // Noncompliant
};
----

==== Compliant solution

[source,cpp,diff-id=1,diff-type=compliant]
----
class MyClass {
  int myField;  // Compliant
};
----

== Resources

=== Documentation

* Geeksforgeeks - https://www.geeksforgeeks.org/naming-convention-in-c/[Naming convention in C++]
* Wikipedia - https://en.wikipedia.org/wiki/Naming_convention_(programming)[Naming Convention (programming)]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../parameters.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
