== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

With the default regular expression ``++^[a-z][a-zA-Z0-9]*$++``:

[source,cpp]
----
class MyClass {
  int my_field;
};
----

=== Compliant solution

[source,cpp]
----
class MyClass {
  int myField;
};
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../parameters.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
