Because the evaluation order of ``++#++`` and ``++##++`` are not specified, the results of using them both in the same macro could be unpredictable. Therefore macros should contain at most once instance of either ``++#++`` or ``++##++``.


== Noncompliant Code Example

----
#define NonCompliant(a, b)  # a ## b 
int main() {  
  std::cout << NonCompliant(Hello, World);
}
----
The result of this code is unspecified. It will either print "HelloWorld" or trigger a compilation error. If ## is evaluated first this will print HelloWorld. If # is evaluated first this will cause a compilation error telling thatÂ *"Hello"World* is not a valid preprocessor token.


== Compliant Solution

----
#define Stringfy(a) #a
#define Compliant(a, b)  Stringfy(a##b) 

int main(){  
  std::cout << Compliant(Hello, World);
}
----
This example will always print "HelloWorld".


== See

* MISRA C:2004, 19.12
* MISRA {cpp} 2008, 16-3-1
* Related: MISRA C:2012, 20.11


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
