=== on 16 Oct 2014, 12:46:31 Ann Campbell wrote:
\[~samuel.mercier] please:

* fill in the appropriate reference field(s).
* provide a See section.
* use the standard section titles: Noncompliant Code Example, Exception*s*, and heading levels (h2. instead of h3.)
* use the standard section order: description, Noncompliant Code Example, Compliant Solution, Exceptions, See

Also, it's not clear to me why you chose Portability. There is very little to go on in this description, but since it's a "may not meet developer expectations" rule, I would go with Reliability

=== on 21 Oct 2014, 15:41:55 Samuel Mercier wrote:
\[~ann.campbell.2] IMO char is implemented as an integer, so the only questions are about signedness and width.

Those are dependent on the compiler, so it could be considered a portability issue.

But since we have other rules about signedness and data width marked as data issue it probably makes sense to change it.

