<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Host variables are used for passing data back and forth from the database to a COBOL program via SQL. In order to keep that transmission clear and free of errors, the format of each host variable should match its corresponding database column type.</p>
</div>
<div class="paragraph">
<p>This rule raises an issue when column and corresponding host variable don&#8217;t match in terms of numeric vs character data, and when the host variable is smaller than the column width.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cobol" data-lang="cobol">TODO</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cobol" data-lang="cobol">TODO</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Correct the format of "xxx" to match "[relevant bit of column config]".</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="ulist">
<ul>
<li>
<p>primary: host var</p>
</li>
<li>
<p>secondary: column name</p>
<div class="ulist">
<ul>
<li>
<p>message: "X" is "[relevant bit of config]".</p>
</li>
</ul>
</div>
</li>
</ul>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_6_oct_2016_122519_pierre_yves_nicolas_wrote">on 6 Oct 2016, 12:25:19 Pierre-Yves Nicolas wrote:</h3>
<div class="paragraph">
<p>This RSPEC will need to define precisely which DB2 types are "compatible" with which COBOL types.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_13_oct_2016_134330_ann_campbell_wrote">on 13 Oct 2016, 13:43:30 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~pierre-yves.nicolas] take a look at the <a href="http://www.ibm.com/support/knowledgecenter/SSEPEK_10.0.0/apsg/src/tpc/db2z_equivalentcoboldatatypes.html">IBM spec</a> &amp; see if you want to replicate that into the RSpec. Maybe we add that reference to the See section?</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_14_oct_2016_174912_pierre_yves_nicolas_wrote">on 14 Oct 2016, 17:49:12 Pierre-Yves Nicolas wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] The IBM spec definitely helps and shows that it would take a long time to cover all possible cases. We should probably focus on the most common cases that we want to catch. The most basic rule is "don&#8217;t mix numeric with character". We also have to check the length of variables: the "receiving" side should be long enough to store the "sent" value. I&#8217;m not sure how we should describe all of that.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_17_oct_2016_160622_ann_campbell_wrote">on 17 Oct 2016, 16:06:22 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~pierre-yves.nicolas] I&#8217;ve expanded the description slightly</p>
</div>
</div>
</div>
</div>