== How to fix it in Java SE

=== Code examples

include::../../common/fix/code-rationale.adoc[]

==== Noncompliant code example

[source,kotlin,diff-id=11,diff-type=noncompliant]
----
@RestController
public class ApiController {

    @Autowired
    var connection: Connection? = null

    @GetMapping(value = ["/authenticate"])
    @ResponseBody
    @Throws(SQLException::class)
    fun authenticate(
        @RequestParam("user") user: String?,
        @RequestParam("pass") pass: String?,
    ): ResponseEntity<String> {
        val query = "SELECT * FROM users WHERE user = $user AND pass = $pass"

        connection?.prepareStatement(query)?.use { statement ->

            val resultSet: ResultSet = statement.executeQuery(query)
            return if (!resultSet.next()) {
                ResponseEntity("Unauthorized", HttpStatus.UNAUTHORIZED)
            } else {
                ResponseEntity("Authentication Success", HttpStatus.OK)
            }
        }
        return ResponseEntity("Unauthorized", HttpStatus.UNAUTHORIZED)
    }
}
----

==== Compliant solution

[source,kotlin,diff-id=11,diff-type=compliant]
----
@RestController
public class ApiController {

    @Autowired
    var connection: Connection? = null

    @GetMapping(value = ["/authenticate"])
    @ResponseBody
    @Throws(SQLException::class)
    fun authenticate(
        @RequestParam("user") user: String?,
        @RequestParam("pass") pass: String?,
    ): ResponseEntity<String> {
        val query = "SELECT * FROM users WHERE user = ? AND pass = ?"

        connection?.prepareStatement(query)?.use { statement ->
            statement.setString(1, user)
            statement.setString(2, pass)

            val resultSet: ResultSet = statement.executeQuery(query)
            return if (!resultSet.next()) {
                ResponseEntity("Unauthorized", HttpStatus.UNAUTHORIZED)
            } else {
                ResponseEntity("Authentication Success", HttpStatus.OK)
            }
        }
        return ResponseEntity("Unauthorized", HttpStatus.UNAUTHORIZED)
    }
}
----

=== How does this work?

include::../../common/fix/prepared-statements.adoc[]

