== Why is this an issue?

Spring framework 4.3 introduced variants of the ``++@RequestMapping++`` annotation to better represent the semantics of the annotated methods. The use of ``++@GetMapping++``, ``++@PostMapping++``, ``++@PutMapping++``, ``++@PatchMapping++`` and ``++@DeleteMapping++`` should be preferred to the use of the raw ``++@RequestMapping(method = RequestMethod.XYZ)++``.


=== Noncompliant code example

[source,java]
----
@RequestMapping(path = "/greeting", method = RequestMethod.GET) // Noncompliant
public Greeting greeting(@RequestParam(value = "name", defaultValue = "World") String name) {
...
}
----

=== Compliant solution

[source,java]
----
@GetMapping(path = "/greeting") // Compliant
public Greeting greeting(@RequestParam(value = "name", defaultValue = "World") String name) {
...
}
----



ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Replace "@RequestMapping(method = RequestMethod.GET)" with "@GetMapping" 

Replace "@RequestMapping(method = RequestMethod.POST)" with "@PostMapping"

Replace "@RequestMapping(method = RequestMethod.PUT)" with "@PutMapping"

Replace "@RequestMapping(method = RequestMethod.PATCH)" with "@PatchMapping"

Replace "@RequestMapping(method = RequestMethod.DELETE)" with "@DeleteMapping"


=== Highlighting

Primary: "@RequestMapping" 

Secondary:  "RequestMethod.XZY"


'''
== Comments And Links
(visible only on this page)

=== on 2 Mar 2018, 13:54:49 Alexandre Gigleux wrote:
This rule should not raise an issue if the ``++method++`` parameter is not set. RSPEC-3752 is there to catch missing ``++method++``.

endif::env-github,rspecator-view[]
