The goal of this rule is to ban the usage of HTML "style" property to make sure that all CSS styles are defined in CSS classes. Consolidating all styling into classes makes it easier to read, understand and maintain.


== Noncompliant Code Example

----
<body>
  <h1 style="color: blue;">Hello World!</h1>  <!-- Noncompliant --> 
----


== Compliant Solution

----
<head>
  <style>
    h1 {
      color: blue;
    }
  </style>
</head>
<body>
<h1>Hello World!</h1>
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

endif::env-github,rspecator-view[]
