Variables should not be initialized to `undefined`.

== Why is this an issue?

Initializing a variable to `undefined` is unnecessary and should be avoided. A variable will automatically be set to `undefined` if you declare it without initialization, so the initialization code is redundant in this case.

== How to fix it

Declare the variable without initialization.

=== Code examples

==== Noncompliant code example

[source,javascript,diff-id=1,diff-type=noncompliant]
----
var foo = undefined;
let bar = undefined;
----

==== Compliant solution

[source,javascript,diff-id=1,diff-type=compliant]
----
var foo;
let bar;
----
