== Why is this an issue?

This rule allows you to track the usage of `noqa` comments used to suppress specific linting issues.

=== Noncompliant code example

[source,python]
----
... # noqa: S100
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Do not suppress issues.

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
