<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>In Java 8, underscore (<code>_</code>) cannot be used as a parameter to a lambda, and by Java 9, it&#8217;s a keyword. To ensure future portability (not to mention current readability), it should not be used now as an identifier.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">private String _;  // Noncompliant</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">private String foo;</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Rename this (variable|parameter).</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>variable/param name</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_relates_to_s1190">relates to: <a data-rspec-id="S1190" class="rspec-auto-link">S1190</a></h3>

</div>
<div class="sect2">
<h3 id="_on_10_apr_2015_125318_elena_vilchik_wrote">on 10 Apr 2015, 12:53:18 Elena Vilchik wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] I added two full lists of future keywords. Could you verify it?</p>
</div>
<div class="paragraph">
<p><a href="http://www.ecma-international.org/ecma-262/5.1/#sec-7.6.1.2" class="bare">http://www.ecma-international.org/ecma-262/5.1/#sec-7.6.1.2</a> (ES5)</p>
</div>
<div class="paragraph">
<p><a href="http://people.mozilla.org/~jorendorff/es6-draft.html#sec-future-reserved-words" class="bare">http://people.mozilla.org/~jorendorff/es6-draft.html#sec-future-reserved-words</a> (ES6)</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_10_apr_2015_134203_ann_campbell_wrote">on 10 Apr 2015, 13:42:03 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>updated per discussion [~elena.vilchik]</p>
</div>
</div>
</div>
</div>