Where the developer uses new versions of standard library macros or objects (e.g. to enhance functionality or add checks of input values), the modified macro or object shall have a new name.


This is to avoid any confusion as to whether a standard macro or object, or a modified version of them, is being used.


== Noncompliant Code Example

----
#define NULL ( a > b ) // Noncompliant
----


