Where the developer uses new versions of standard library macros or objects (e.g. to enhance functionality or add checks of input values), the modified macro or object shall have a new name.


This is to avoid any confusion as to whether a standard macro or object, or a modified version of them, is being used.


== Noncompliant Code Example

----
#define NULL ( a > b ) // Noncompliant
----


== See

* MISRA C:2004, 20.2
* MISRA {cpp}:2008, 17-0-2
* https://wiki.sei.cmu.edu/confluence/x/GDZGBQ[CERT, DCL01-J.] - Do not reuse public identifiers from the Java Standard Library

