include::../description.adoc[]

include::../why.adoc[]

include::../how.adoc[]

=== Code examples

==== Noncompliant code example

[source,vbnet,diff-id=1,diff-type=noncompliant]
----
Dim dateTimeString = "4/12/2023 4:05:48 PM"
Dim dateTimeObject = DateTime.Parse(dateTimeString) ' This is an en-US format string - 12 of April 2023

Dim dateTimeString2 = "4/13/2023 4:05:48 PM"
Dim dateTimeObject2 = DateTime.Parse(dateTimeString2) ' In a machine with European local format is parsed as 4th of December
----

==== Compliant solution

[source,vbnet,diff-id=1,diff-type=compliant]
----
Dim dateTimeString = "4/12/2023 4:05:48 PM" ' This is an en-US format string - 12 of April 2023
Dim dateTimeObject = DateTime.Parse(dateTimeString, New CultureInfo("en-US"))

Dim dateTimeString2 = "4/13/2023 4:05:48 PM" ' This is an en-US format string - 12 of April 2023
Dim dateTimeObject2 = DateTime.Parse(dateTimeString2, New CultureInfo("en-US"))
----

== Resources

include::../resources.adoc[]
