<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>According to the Java <code>Comparable.compareTo(T o)</code> documentation:</p>
</div>
<div class="quoteblock">
<blockquote>
<div class="paragraph">
<p>It is strongly recommended, but not strictly required that <code>(x.compareTo(y)==0) == (x.equals(y))</code>.</p>
</div>
<div class="paragraph">
<p>Generally speaking, any class that implements the Comparable interface and violates this condition should clearly indicate this fact.</p>
</div>
<div class="paragraph">
<p>The recommended language is "Note: this class has a natural ordering that is inconsistent with equals."</p>
</div>
</blockquote>
</div>
<div class="paragraph">
<p>If this rule is violated, weird and unpredictable failures can occur.</p>
</div>
<div class="paragraph">
<p>For example, in Java 5 the <code>PriorityQueue.remove()</code> method relied on <code>compareTo()</code>, but since Java 6 it has relied on <code>equals()</code>.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">public class Foo implements Comparable&lt;Foo&gt; {
  @Override
  public int compareTo(Foo foo) { /* ... */ }      // Noncompliant as the equals(Object obj) method is not overridden
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">public class Foo implements Comparable&lt;Foo&gt; {
  @Override
  public int compareTo(Foo foo) { /* ... */ }      // Compliant

  @Override
  public boolean equals(Object obj) { /* ... */ }
}</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Override "equals(Object obj)" to comply with the contract of the "compareTo(T o)" method</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_20_aug_2013_140349_freddy_mallet_wrote">on 20 Aug 2013, 14:03:49 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>Is implemented by http://jira.codehaus.org/browse/SONARJAVA-303</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_2_oct_2013_135728_freddy_mallet_wrote">on 2 Oct 2013, 13:57:28 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>Manually tested !</p>
</div>
</div>
</div>
</div>