== Why is this an issue?

A function which does not return a value and which does not have external side effects will only consume time and will not contribute to the generation of any outputs, which may not meet developer expectations.

The following are examples of external side effects:

* Reading or writing to a file, stream, etc.;
* Changing the value of a non local variable;
* Changing the value of an argument having reference type;
* Using a volatile object;
* Raising an exception.


=== Noncompliant code example

[source,cpp]
----
void pointless ( void ) // Non-compliant â€“ no external side effects
{
 int16_t local;
 local = 0;
}
----


== Resources

* MISRA {cpp}:2008, 0-1-8


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Either refactor or remove this function without side effect(s).


'''
== Comments And Links
(visible only on this page)

=== on 16 Oct 2014, 14:25:17 Ann Campbell wrote:
\[~samuel.mercier] FYI, I've switched this to Efficiency because the description says such functions "only consume time and not contribute"

endif::env-github,rspecator-view[]
