This rule raises an issue when accessing properties or methods on objects that might be null using regular dot notation instead of the safe navigation operator.

== Why is this an issue?

In Groovy, accessing properties or methods on null objects using regular dot notation (`.`) will throw a `NullPointerException` at runtime, which can crash your application.

Groovy provides the safe navigation operator (`?.`) specifically to handle this situation safely. When you use `?.` to access a property or method on a null object, it returns `null` instead of throwing an exception.

Using regular dot notation when the object might be null creates a potential runtime failure point that could have been easily avoided. This is particularly problematic in dynamic languages like Groovy where null checks might not be enforced at compile time.

The safe navigation operator makes your code more robust and eliminates a common source of runtime errors without requiring verbose null checks.

=== What is the potential impact?

Using regular dot notation on potentially null objects can cause:

* *Runtime crashes*: `NullPointerException` will terminate your application unexpectedly
* *Poor user experience*: Users may encounter application failures during normal usage
* *Debugging overhead*: NullPointerExceptions can be difficult to trace in complex call stacks
* *Maintenance burden*: Developers need to add explicit null checks throughout the codebase

== How to fix it

Replace regular dot notation with the safe navigation operator when accessing properties or methods on objects that might be null. The safe navigation operator (`?.`) will return null safely instead of throwing an exception.

=== Code examples

==== Noncompliant code example

[source,groovy,diff-id=1,diff-type=noncompliant]
----
def person = Person.find { it.id == 123 }
def name = person.name  // Noncompliant
----

==== Compliant solution

[source,groovy,diff-id=1,diff-type=compliant]
----
def person = Person.find { it.id == 123 }
def name = person?.name  // Returns null safely
----

== Resources

=== Documentation

 * Groovy Operators - Safe Navigation Operator - https://groovy-lang.org/operators.html#_safe_navigation_operator[Official Groovy documentation explaining the safe navigation operator]

 * Groovy Design Patterns - Null Object Pattern - https://groovy-lang.org/design-patterns.html#_null_object_pattern[Groovy documentation showing how safe navigation operator helps avoid null pointer exceptions]

=== Related rules

 * S6582 - https://rules.sonarsource.com/dart/RSPEC-6582/[Optional navigation should be used instead of explicit null testing (Dart)]

 * S6582 - https://rules.sonarsource.com/javascript/RSPEC-6582/[Optional chaining should be used instead of explicit null testing (JavaScript)]
