== Why is this an issue?

Turning on the ``++*srcstmt++`` header option means that any line numbers cited in error statements will actually correspond correctly to the code. Otherwise, object line numbers will be shown, making errors difficult to debug.


The ``++*srcstmt++`` option is also recommended for accurate line numbers during debugging, particularly in combination with ``++*nodebugio++``, (``++H option(*srcstmt:*nodebugio)++``) which prevents debug operations from stopping on I/O.


=== Noncompliant code example

[source,rpg]
----
H*
H* AMENDED BY : G. Ann Campbell
H* DATE       : 16JUL2014      
H* H24 - Miscellaneous Online Maintenance/Enquiry/Printing
H*
F/EJECT
----


=== Compliant solution

[source,rpg]
----
H*
H* AMENDED BY : G. Ann Campbell
H* DATE       : 16JUL2014      
H* H24 - Miscellaneous Online Maintenance/Enquiry/Printing
H*
H option(*srcstmt)
F/EJECT
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Add "H option (*srcstmt)" to this file.


'''
== Comments And Links
(visible only on this page)

=== on 31 Jul 2014, 15:24:06 Freddy Mallet wrote:
\[~ann.campbell.2] If my feeling is correct, there is a typo in the following sentence. I guess you wanted to say "won't be shown" and not "will be shown" ?


____Otherwise, object line numbers will be shown, making errors difficult to debug.____

=== on 31 Jul 2014, 18:55:00 Ann Campbell wrote:
\[~freddy.mallet] nope. At issue is whether source line #s are shown versus object line #s.

endif::env-github,rspecator-view[]
