Turning on the ``++*srcstmt++`` header option means that any line numbers cited in error statements will actually correspond correctly to the code. Otherwise, object line numbers will be shown, making errors difficult to debug.


The ``++*srcstmt++`` option is also recommended for accurate line numbers during debugging, particularly in combination with ``++*nodebugio++``, (``++H option(*srcstmt:*nodebugio)++``) which prevents debug operations from stopping on I/O.


== Noncompliant Code Example

----
H*
H* AMENDED BY : G. Ann Campbell
H* DATE       : 16JUL2014      
H* H24 - Miscellaneous Online Maintenance/Enquiry/Printing
H*
F/EJECT
----


== Compliant Solution

----
H*
H* AMENDED BY : G. Ann Campbell
H* DATE       : 16JUL2014      
H* H24 - Miscellaneous Online Maintenance/Enquiry/Printing
H*
H option(*srcstmt)
F/EJECT
----


ifdef::env-github,rspecator-view[]
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
