<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when 7 or more commands are applied on a data frame.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>The pandas library provides many ways to filter, select, reshape and modify a data frame.
Pandas supports as well method chaining, which means that many <code>DataFrame</code> methods return a modified <code>DataFrame</code>.
This allows the user to chain multiple operations together, making it effortless perform several of them in one line of code:</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">import pandas as pd

schema = {'name':str, 'domain': str, 'revenue': 'Int64'}
joe = pd.read_csv("data.csv", dtype=schema).set_index('name').filter(like='joe', axis=0).groupby('domain').mean().round().sample()</code></pre>
</div>
</div>
<div class="paragraph">
<p>While this code is correct and concise,
it can be challenging to follow its logic and flow, making it harder to debug or modify in the future.</p>
</div>
<div class="paragraph">
<p>To improve code readability, debugging, and maintainability, it is recommended to break down long chains of pandas instructions into smaller, more modular steps.
This can be done with the help of the pandas <code>pipe</code> method, which takes a function as a parameter.
This function takes the data frame as a parameter, operates on it and returns it for further processing.
Grouping complex transformations of a data frame inside a function with a meaningful name can further enhance the readability and maintainability of the code.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>To fix this issue refactor chains of instruction into a function that can be consumed by the <code>pandas.pipe</code> method.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">import pandas as pd

def foo(df: pd.DataFrame):
  return df.set_index('name').filter(like='joe', axis=0).groupby('team').mean().round().sort_values('salary').take([0]) # Noncompliant: too many operations happen on this data frame.</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">import pandas as pd

def select_joes(df):
  return df.set_index('name').filter(like='joe', axis=0)

def compute_mean_salary_per_team(df):
  return df.groupby('team').mean().round()

def foo(df: pd.DataFrame):
  return df.pipe(select_joes).pipe(compute_mean_salary_per_team).sort_values('salary').take([0]) # Compliant</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Pandas Documentation - <a href="https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.pipe.html#pandas-dataframe-pipe">pandas.DataFrame.pipe</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>