This rule raises an issue when environment (exported) variables don't use ALL_CAPS naming or when local variables don't use lower_case naming.

== Why is this an issue?

Shell scripts benefit from consistent variable naming conventions that immediately communicate the scope and purpose of variables.

Environment variables are typically shared across processes and should stand out in code. The ALL_CAPS convention makes exported variables easily recognizable and follows long-established Unix traditions. This helps developers quickly identify which variables affect the broader system environment.

Local variables, on the other hand, are used within specific functions or script sections. Using lower_case naming distinguishes them from environment variables and makes the code's intent clearer. This separation prevents confusion about variable scope and reduces the risk of accidentally overriding important environment variables.

Inconsistent naming conventions make scripts harder to read and maintain. When variables don't follow expected patterns, developers must spend extra time understanding the code's structure and may make incorrect assumptions about variable behavior.

=== What is the potential impact?

Inconsistent variable naming reduces code readability and can lead to maintenance issues. Developers may accidentally override environment variables or misunderstand variable scope, potentially causing unexpected behavior in scripts or system configurations.

== How to fix it

Use ALL_CAPS for exported environment variables and lower_case for local variables. This follows established shell scripting conventions and improves code clarity.

=== Code examples

==== Noncompliant code example

[source,shell,diff-id=1,diff-type=noncompliant]
----
export my_path="/usr/bin"  # Noncompliant
local MyVar="value"        # Noncompliant
----

==== Compliant solution

[source,shell,diff-id=1,diff-type=compliant]
----
export MY_PATH="/usr/bin"
local my_var="value"
----

== Resources

=== Documentation

 * Bash Best Practices - Variables - https://bertvv.github.io/cheat-sheets/Bash.html[Comprehensive guide to Bash scripting best practices including variable naming conventions]
 * Google Shell Style Guide - https://google.github.io/styleguide/shellguide.html[Google's style guide for shell scripting with detailed naming conventions]

