This rule raises an issue when code manually concatenates strings using the `+` operator inside iteration methods like `each`, `for`, or similar loops.

== Why is this an issue?

Manual string concatenation in loops creates performance and reliability problems.

In Ruby, strings are immutable objects. Each time you use the `+` operator to concatenate strings, Ruby creates a completely new string object in memory. When this happens inside a loop, the performance impact grows exponentially with the size of your data.

For example, concatenating 1,000 strings this way creates 1,000 intermediate string objects, each copying all the previous content. This results in O(n²) time complexity instead of the O(n) you would expect.

Beyond performance, manual concatenation is error-prone. If you forget to initialize your string variable, Ruby throws a `NilClass` error when it tries to call the `+` method on `nil`.

Ruby provides better alternatives through its functional programming features. Methods like `map` and `join` are specifically designed for transforming collections into strings efficiently.

=== What is the potential impact?

Performance degrades significantly with larger datasets due to O(n²) time complexity. Memory usage increases unnecessarily from creating intermediate string objects. Runtime errors occur when string variables are not properly initialized, potentially causing application crashes.

== How to fix it

Replace manual string concatenation with `map` and `join`. This approach transforms the collection into an array of strings first, then joins them efficiently in a single operation.

=== Code examples

==== Noncompliant code example

[source,ruby,diff-id=1,diff-type=noncompliant]
----
assets = Asset.where(current_status: ["active"])
string = ""
assets.each do |asset|
  string = string + ":" + asset.movie_title # Noncompliant
end
----

==== Compliant solution

[source,ruby,diff-id=1,diff-type=compliant]
----
assets = Asset.where(current_status: ["active"])
string = assets.map(&:movie_title).join(":")
----

== Resources

=== Documentation

 * Ruby Array#map documentation - https://ruby-doc.org/core/Array.html#method-i-map[Official documentation for the map method used to transform array elements]

 * Ruby Array#join documentation - https://ruby-doc.org/core/Array.html#method-i-join[Official documentation for the join method used to concatenate array elements into a string]

 * Ruby Performance Guide - String Concatenation - https://github.com/JuanitoFatas/fast-ruby#string[Performance benchmarks comparing different string concatenation methods in Ruby]

=== Related rules

 * RSPEC-1643 - https://rules.sonarsource.com/csharp/RSPEC-1643/[Strings should not be concatenated using '+' in a loop (C#)]

 * RSPEC-1643 - https://rules.sonarsource.com/java/RSPEC-1643/[Strings should not be concatenated using '+' in a loop (Java)]
