It is unspecified behaviour if the evaluation of an expression with ``++enum++`` underlying type yields a value which does not correspond to one of the enumerators of the enumeration.


Additionally, other rules in this standard assume that objects of ``++enum++`` type only contain values corresponding to the enumerators. This rule ensures the validity of these assumptions.


One way of ensuring compliance when converting to an enumeration is to use a switch statement.


== Noncompliant Code Example

----
enum Numbers {
  ONE, TWO
};

int function() {
        enum Numbers c = (enum Numbers)(TWO + TWO); // Noncompliant, operation might yield an invalid value.
}
----


== Compliant Solution

----
enum Numbers {
  ONE, TWO
};

enum Numbers convert ( int16_t v )
{
  switch ( v )
  {
  case 0:
    return ONE;
  case 1:
    return TWO;
  default:
    throw ENUM_ERROR;
  }
}

int function() {
  enum Numbers c = convert(ONE + TWO); // Compliant
}
----


== See

* MISRA {cpp}:2008, 7-2-1


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
