== Why is this an issue?

The requirement for a final ``++Case Else++`` clause is defensive programming.

This clause should either take appropriate action or contain a suitable comment as to why no action is taken.

=== Noncompliant code example

[source,vbnet]
----
Select Case param ' Noncompliant - Case Else clause is missing
  Case 0
    DoSomething()
  Case 1
    DoSomethingElse()
End Select
----

=== Compliant solution

[source,vbnet]
----
Select Case param
  Case 0
    DoSomething()
  Case 1
    DoSomethingElse()
  Case Else ' Compliant
    DoSomethingElse()
End Select
----

include::../see.adoc[]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
