This rule raises an issue on logging calls inside a `catch` clause that does not pass the raised `Exception`.

== Why is this an issue?

A log entry should contain all the relevant information about the current execution context. The https://learn.microsoft.com/en-us/dotnet/api/system.exception[Exception] raised in a catch block not only provides the message but also

* the exception type,
* the https://learn.microsoft.com/en-us/dotnet/api/system.exception.stacktrace[stack trace],
* any https://learn.microsoft.com/en-us/dotnet/api/system.exception.innerexception[inner exceptions],
* and more about the cause of the error.

https://learn.microsoft.com/en-us/dotnet/api/microsoft.extensions.logging.loggerextensions[Logging methods] provide overloads that accept an `Exception` as a parameter and 
https://learn.microsoft.com/en-us/dotnet/core/extensions/logging-providers[logging providers] persist the `Exception` in a structured way to facilitate the tracking of system failures. Therefore `Exceptions` should be passed to the logger.

== How to fix it

=== Code examples

==== Noncompliant code example

[source,csharp,diff-id=1,diff-type=noncompliant]
----
public bool Save()
{
    try
    {
        DoSave();
        return true;
    }
    catch(IOException)
    {
        logger.LogError("Saving failed.");             // Noncompliant: No specifics about the error are logged
        return false;
    }
}
----

==== Compliant solution

[source,csharp,diff-id=1,diff-type=compliant]
----
public bool Save()
{
    try
    {
        DoSave();
        return true;
    }
    catch(IOException exception)
    {
        logger.LogError(exception, "Saving failed.");  // Compliant: Exception details are logged
        return false;
    }
}
----

== Resources

=== Documentation

* Microsoft Learn - https://learn.microsoft.com/en-us/dotnet/core/extensions/logging?tabs=command-line#log-exceptions[Log exceptions]
* Microsoft Learn - https://learn.microsoft.com/en-us/dotnet/api/microsoft.extensions.logging.loggerextensions[LoggerExtensions Class]
* Microsoft Learn - https://learn.microsoft.com/en-us/dotnet/core/extensions/logging-providers[Logging providers]
* Microsoft Learn - https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/statements/exception-handling-statements#the-try-catch-statement[The `try-catch` statement]
* Serilog - https://github.com/serilog/serilog/wiki/Getting-Started#example-application[Example application]
* Serilog Analyzer - https://github.com/Suchiman/SerilogAnalyzer#serilog001-exception-usage[`Serilog001`: Exception Usage]