https://learn.microsoft.com/en-us/dotnet/api/microsoft.extensions.logging.loggerextensions[Logging methods] provide overloads that accept an `Exception` as a parameter. This rule raises an issue if there is a logging call inside a `catch` clause that does not pass an `Exception` to the right overload parameter.

== Why is this an issue?

A log entry should contain all the relevant information about the current execution context. In a `catch` block the raised https://learn.microsoft.com/en-us/dotnet/api/system.exception[Exception] not only provides the message but also the exception type, the https://learn.microsoft.com/en-us/dotnet/api/system.exception.stacktrace[stack trace], any https://learn.microsoft.com/en-us/dotnet/api/system.exception.innerexception[inner exceptions], and more about the cause of the error.

https://learn.microsoft.com/en-us/dotnet/core/extensions/logging-providers[Logging providers] persist the information found in the `Exception` in a structured way to facilitate the tracking of system failures and therefore exceptions should be passed to the logger.

== How to fix it

=== Code examples

==== Noncompliant code example

[source,csharp,diff-id=1,diff-type=noncompliant]
----
public bool Save()
{
    try
    {
        DoSave();
        return true;
    }
    catch(IOException)
    {
        logger.LogError("Saving failed.");             // Noncompliant - No specifics about the error are logged
        return false;
    }
}
----

==== Compliant solution

[source,csharp,diff-id=1,diff-type=compliant]
----
public bool Save()
{
    try
    {
        DoSave();
        return true;
    }
    catch(IOException exception)
    {
        logger.LogError(exception, "Saving failed.");  // Compliant - Exception details are logged
        return false;
    }
}
----

== Resources

=== Documentation

* Microsoft Learn - https://learn.microsoft.com/en-us/dotnet/core/extensions/logging?tabs=command-line#log-exceptions[Log exceptions]
* Microsoft Learn - https://learn.microsoft.com/en-us/dotnet/api/microsoft.extensions.logging.loggerextensions[LoggerExtensions Class]
* Microsoft Learn - https://learn.microsoft.com/en-us/dotnet/core/extensions/logging-providers[Logging providers]
* Microsoft Learn - https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/statements/exception-handling-statements#the-try-catch-statement[The `try-catch` statement]
* Serilog - https://github.com/serilog/serilog/wiki/Getting-Started#example-application[Example application]