<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when an unhashable object is used as a set member or dictionary key.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>The hash value of an object is an integer returned by its <code>__hash__</code> method. Objects that are considered equal to each other (as per the <code>__eq__</code> method) should have the same hash value.</p>
</div>
<div class="paragraph">
<p>Whenever using an object as a dictionary key or inserting it into a set, the hash value of that object will be used to derive a bucket in which the object will be inserted.</p>
</div>
<div class="paragraph">
<p>When attempting to insert an unhashable object into a set, a <code>TypeError</code> will be raised instead.</p>
</div>
<div class="paragraph">
<p>If an object defines a <code>__hash__</code> method derived from mutable properties, no <code>TypeError</code> will be raised. However, having a mutable hash value should never be done as this would prevent dictionaries and set from retrieving the object.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Make sure to always use hashable (and immutable) objects as dictionary keys or set members.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">def foo():
    my_list = [1,2,3]
    my_set = {my_list}  # Noncompliant: list is not hashable.</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">def foo():
    my_list = [1,2,3]
    my_set = set(my_list) # This will insert the elements of my_list into a new set.

# OR

def foo():
    my_tuple = (1,2,3)
    my_set = {my_tuple} # tuples are immutable and hashable. They can be inserted into sets.</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Python Documentation - <a href="https://docs.python.org/3/reference/datamodel.html#object.<em>hash</em>">object.<em>hash</em></a></p>
</li>
<li>
<p>Python Documentation - <a href="https://docs.python.org/3/library/functions.html#hash">the hash built-in function</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>