== Why is this an issue?

Leftover empty blocks are usually introduced by mistake. They are useless and prevent readability of the code. They should be removed or completed with real code.


=== Noncompliant code example

[source,css]
----
a { }
----


=== Compliant solution

[source,css]
----
a { color: pink; }
----

