== Why is this an issue?

There's no need to repeat the program name in the free-format definition of an external program.


=== Noncompliant code example

[source,rpg]
----
DCL-PR PGM001 EXTPGM('PGM001'); 
END-PR;
----


=== Compliant solution

[source,rpg]
----
DCL-PR PGM001 EXTPGM ; 
END-PR;
----
or if the program is renamed

[source,rpg]
----
DCL-PR TheProgram EXTPGM('PGM001'); 
END-PR;
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

endif::env-github,rspecator-view[]
