== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,rust,diff-id=1,diff-type=noncompliant]
----
let mut x = 5;
x = x; // Self-assignment - does nothing
----

==== Compliant solution

[source,rust,diff-id=1,diff-type=compliant]
----
let mut x = 5;
// Removed the self-assignment
----

== Resources
=== Documentation

* Clippy Lints - https://rust-lang.github.io/rust-clippy/master/index.html#self_assignment

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
