<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when there are spaces before or after the dot operator (<code>.</code>) in method calls.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Adding spaces around the dot operator in method calls reduces code readability and goes against established Ruby style conventions.</p>
</div>
<div class="paragraph">
<p>The dot operator creates a tight semantic binding between an object and its method. When spaces are added around it, this visual connection is broken, making the code harder to scan and understand quickly.</p>
</div>
<div class="paragraph">
<p>Consistent formatting helps developers read and maintain code more efficiently. Most Ruby style guides, including those from major organizations like Shopify, explicitly recommend avoiding spaces around method call operators.</p>
</div>
<div class="paragraph">
<p>While this formatting issue doesn&#8217;t affect the functionality of the code, it can make codebases appear inconsistent and unprofessional, especially when different team members follow different spacing conventions.</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="paragraph">
<p>This formatting issue reduces code readability and creates inconsistency in the codebase. While it doesn&#8217;t affect functionality, it can slow down code review processes and make the code appear less professional.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Remove the spaces before and after the dot operator in method calls.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-ruby" data-lang="ruby">user . name
data . fetch(:key) . upcase
object . method_name</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-ruby" data-lang="ruby">user.name
data.fetch(:key).upcase
object.method_name</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Shopify Ruby Style Guide - Layout - <a href="https://ruby-style-guide.shopify.dev/#layout">Official Shopify Ruby style guide covering layout and formatting conventions</a></p>
</li>
<li>
<p>RuboCop Layout/SpaceAroundMethodCallOperator - <a href="https://docs.rubocop.org/rubocop/cops_layout.html#layoutspacearoundmethodcalloperator">RuboCop cop documentation for detecting spaces around method call operators</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>