Classes without ``++public static++`` members cannot be used without being instantiated, but classes with only ``++private++`` constructors cannot be instantiated. When a class has only ``++private++`` constructors and no ``++static++`` members, it is useless and should be removed or refactored.


== Noncompliant Code Example

----
public class MyClass {  // Noncompliant

  private MyClass () {
  }

  public double getPi(){
    return 3.14;
  }
}
----


== Compliant Solution

----
public class MyClass {  

  public MyClass () {
  }

  public double getPi(){
    return 3.14;
  }
}
----
or

----
public class MyClass {  

  private MyClass () {
  }

  public static double getPi(){
    return 3.14;
  }
}
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

endif::env-github,rspecator-view[]
