Clean code should not make use of profanities, but instead use neutral and inclusive terminology.

== Why is this an issue?

Code is meant to be understandable and clear to the reader.
Discriminatory and offensive language is often used for expressiveness.
Such language has made its way into technical jargon and code despite being derived
from specific cultural contexts that carry discriminatory connotations.
These terms do not accurately convey the intended meaning in technical contexts
and are often idiomatic expressions that can be confusing or misleading.
Using such terms can perpetuate negative biases and hinder effective communication.

For instance, the term `dummy` is not meaningful in a software context as the term
originally refers to a person who cannot speak, which is irrelevant to code functionality.
More precise and descriptive alternatives, such as placeholder or pseudo,
exist and provide a clearer meaning. By choosing unambiguous and neutral terminology,
we promote inclusivity and improve the clarity and maintainability of code.

Companies with a mandate for inclusivity need to build products for everyone with
the following considerations:

* *Brand reputation*
** The adoption of respectful language in code aligns with evolving industry standards to
position a brand as a leader in innovation and professionalism and avoid reputational damage and backlash.

* *Employee Relations*
** Respectful and inclusive language in code reflects a respectful and inclusive workplace culture.
** Clear language in code fosters better understanding and communication among team members.

* *Legal/Regulatory Compliance*
** Compliance with legal requirements and industry regulations
** Neutral and inclusive terminology minimizes the risk of litigation

== How to fix it

Replace the usage of the prejudiced language with neutral and inclusive terminology.

=== Code examples

==== Noncompliant code example

[source,text,diff-id=1,diff-type=noncompliant]
----
# Used as a dummy value
----

==== Compliant solution

[source,text,diff-id=1,diff-type=compliant]
----
# Used as a placeholder value
----


== Resources
=== Articles & blog posts

* University of Washington - https://itconnect.uw.edu/guides-by-topic/identity-diversity-inclusion/inclusive-language-guide/[IT Inclusive Language Guide]
* Google - https://source.android.com/docs/setup/contribute/respectful-code[Code with respect]
* Android Contributor Guidelines - https://developers.google.com/style/inclusive-documentation[Write inclusive documentation]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Remove the usage of this loaded language and replace it with neutral and inclusive terminology.

=== Highlighting

Highlight the usage of the loaded language.

'''
== Comments And Links
(visible only on this page)

endif::env-github,rspecator-view[]
