This is an issue when using the `junit` step in Jenkins pipelines without specifying the `testResults` parameter.

== Why is this an issue?

The `junit` step in Jenkins pipelines is used to publish test results and provide visual feedback about test outcomes. When you use this step without specifying the `testResults` parameter, Jenkins cannot locate the test result files.

The `testResults` parameter tells Jenkins where to find the XML test report files using file patterns (like `++**/test-results/*.xml++`). Without this parameter, the step will fail to collect any test results, even if your tests ran successfully.

This leads to several problems:

* Jenkins cannot display test trends and historical data
* Test failures won't be properly reported in the build status
* GitHub integration and other SCM platforms won't receive test result notifications
* Developers lose visibility into which specific tests are failing
* Build health scoring based on test results won't work correctly

The Jenkins documentation consistently shows examples where `testResults` is provided alongside other configuration options, indicating this is essential for proper functionality.

=== What is the potential impact?

Without the `testResults` parameter, your Jenkins pipeline will lose test result visibility. This means you won't see test trends, failure details, or proper build status reporting. Your team will miss critical feedback about code quality and test coverage.

== How to fix it

Add the `testResults` parameter to specify where Jenkins should look for test result XML files. Use file patterns that match your test output location.

=== Code examples

==== Noncompliant code example

[source,groovy,diff-id=1,diff-type=noncompliant]
----
junit skipMarkingBuildUnstable: true // Noncompliant
----

==== Compliant solution

[source,groovy,diff-id=1,diff-type=compliant]
----
junit skipMarkingBuildUnstable: true, testResults: 'test-results.xml'
----

== Resources

=== Documentation

 * Jenkins JUnit Plugin Documentation - https://plugins.jenkins.io/junit/[Official documentation for the Jenkins JUnit plugin with configuration examples]

 * Jenkins Pipeline Steps Reference - JUnit - https://www.jenkins.io/doc/pipeline/steps/junit/[Pipeline step reference for the junit step with parameter details]
