<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Defining a nested single-line comment within a multi-line comment invites errors. It may lead a developer to wrongly think that the lines located after the single-line comment are not part of the comment.</p>
</div>
<div class="paragraph">
<p>If a comment starting sequence, <code>/*</code> or <code>//</code>, occurs within a <code>/*</code> comment, is it quite likely to be caused by a missing <code>*/</code> comment ending sequence.</p>
</div>
<div class="paragraph">
<p>If a comment starting sequence occurs within a <code>//</code> comment, it is probably because a region of code has been commented-out using <code>//</code>.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-text" data-lang="text">/* some comment, end comment marker accidentally omitted
// Make sure this function is called in a thread safe context
Perform_Critical_Safety_Function(X);
...
/* this comment is non-compliant */</code></pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p><a href="https://wiki.sei.cmu.edu/confluence/x/k9YxBQ">CERT, MSC04-C.</a> - Use comments consistently and in a readable fashion</p>
</li>
</ul>
</div>
<hr>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove the misleading "[//|/*]" characters.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_deprecates_s788">deprecates: <a data-rspec-id="S788" class="rspec-auto-link">S788</a></h3>

</div>
<div class="sect2">
<h3 id="_deprecates_s2523">deprecates: <a data-rspec-id="S2523" class="rspec-auto-link">S2523</a></h3>

</div>
<div class="sect2">
<h3 id="_is_related_to_s787">is related to: <a data-rspec-id="S787" class="rspec-auto-link">S787</a></h3>

</div>
<div class="sect2">
<h3 id="_is_related_to_s1917">is related to: <a data-rspec-id="S1917" class="rspec-auto-link">S1917</a></h3>

</div>
<div class="sect2">
<h3 id="_on_29_aug_2013_003325_ann_campbell_wrote">on 29 Aug 2013, 00:33:25 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>This is funny. Our spec requires us to break this rule in our explanation of it!</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_29_aug_2013_232627_ann_campbell_wrote">on 29 Aug 2013, 23:26:27 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>I&#8217;ve retained "C style" and "C&#43;&#43; style" with much reduced emphasis.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_4_sep_2013_124516_dinesh_bolkensteyn_wrote">on 4 Sep 2013, 12:45:16 Dinesh Bolkensteyn wrote:</h3>
<div class="paragraph">
<p>This rule will always be trigger on comments such as:</p>
</div>
<div class="listingblock">
<div class="content">
<pre>/*
  http://www.example.org  &lt;&lt; notice the "//"
*/</pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_on_7_jan_2014_112835_freddy_mallet_wrote">on 7 Jan 2014, 11:28:35 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>Implemented by http://jira.codehaus.org/browse/SONARPLUGINS-3372 for JavaScript</p>
</div>
</div>
</div>
</div>