include::description.adoc[]


=== Noncompliant code example

[source,text]
----
^a|b|c$
----


=== Compliant solution

[source,text]
----
^(?:a|b|c)$
----

or


[source,text]
----
^a$|^b$|^c$
----

or, if you do want the anchors to only apply to ``++a++`` and ``++c++`` respectively:


[source,text]
----
(?:^a)|b|(?:c$)
----
ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Group parts of the regex together to make the intended operator precedence explicit.


=== Highlighting

The entire regex


endif::env-github,rspecator-view[]
