include::description.adoc[]


=== Noncompliant code example

[source,text]
----
^a|b|c$
----


=== Compliant solution

[source,text]
----
^(?:a|b|c)$
----

or


[source,text]
----
^a$|^b$|^c$
----

or, if you do want the anchors to only apply to ``++a++`` and ``++c++`` respectively:


[source,text]
----
(?:^a)|b|(?:c$)
----
ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

include::highlighting.adoc[]

endif::env-github,rspecator-view[]
