include::description.adoc[]


== Noncompliant Code Example

----
^a|b|c$
----


== Compliant Solution

----
^(?:a|b|c)$
----

or


----
^a$|^b$|^c$
----

or, if you do want the anchors to only apply to ``++a++`` and ``++c++`` respectively:


----
(?:^a)|b|(?:c$)
----
ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

include::highlighting.adoc[]

endif::env-github,rspecator-view[]
