include::../description.adoc[]

== Noncompliant Code Example

----
public static void DoSomething() 
{
    for(int i = 0; i < 4; i++)  // Noncompliant, 4 is a magic number
    {
        ...
    }
}
----

== Compliant Solution

----
private const int NUMBER_OF_CYCLES = 4;

public static void DoSomething()
{
    for(int i = 0; i < NUMBER_OF_CYCLES ; i++)  //Compliant
    {
        ...
    }
}
----

== Exceptions

This rule doesn't raise an issue when the magic number is used as part of the ``++GetHashCode++`` method or a variable/field declaration.

ifdef::rspecator-view[]
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::rspecator-view[]
