include::../description.adoc[]

== Noncompliant Code Example

----
public static void DoSomething() 
{
    for(int i = 0; i < 4; i++)  // Noncompliant, 4 is a magic number
    {
        ...
    }
}
----

== Compliant Solution

----
private const int NUMBER_OF_CYCLES = 4;

public static void DoSomething()
{
    for(int i = 0; i < NUMBER_OF_CYCLES ; i++)  //Compliant
    {
        ...
    }
}
----

== Exceptions

This rule doesn't raise an issue when the magic number is used as part of:

- the ``++GetHashCode++`` method
- a variable/field declaration
- the single argument of an attribute
- a named argument for a method or attribute
- a constructor call

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
