<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Using contradictory or nonsensical combinations of file open options, such as <code>read(true)</code> with <code>truncate(true)</code>, can lead to code that is confusing and harder to read. In some cases, it can result in runtime errors or undefined behavior, potentially causing data corruption or loss.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-rust" data-lang="rust">use std::fs::OpenOptions;

OpenOptions::new().read(true).truncate(true); // Noncompliant: Invalid combination of file open options.</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-rust" data-lang="rust">use std::fs::OpenOptions;

OpenOptions::new().write(true).truncate(true); // Compliant: Valid combination of file open options.</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Clippy Lints - <a href="https://rust-lang.github.io/rust-clippy/master/index.html#nonsensical_open_options" class="bare">https://rust-lang.github.io/rust-clippy/master/index.html#nonsensical_open_options</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>