Alternation is used to match a single regular expression out of several possible regular expressions. If one of the alternatives is empty it would match any input, which is most probably a mistake.

== Noncompliant Code Example
----
/Jack|Peter|/.test('John'); // returns 'true'
/Jack||Peter/.test('John'); // returns 'true'
----
== Compliant Solution
----
/Jack|Peter/.test('John'); // returns 'false'
----

== Exceptions

Empty alternation can be used to make group inside regex optional like in the following example

----
/mandatory(-optional|)/.test('mandatory'); // returns 'true'
/mandatory(-optional|)/.test('mandatory-optional'); // returns 'true'
----

Rule will not raise issue on such cases. However, if there is quantifier used for the group we will raise the issue because using `|` and quantifier is potentially confusing.

----
/mandatory(-optional|)?/.test('mandatory'); // using both `|` inside the group and `?` for the group.
----
