Alternation is used to match a single regular expression out of several possible regular expressions. Having one of the alternatives being empty would test 'true' on any input and is most probably a mistake.

== Noncompliant Code Example
----
/Jack|Peter|/.test('John'); // returns 'true'
/Jack||Peter/.test('John'); // returns 'true'
----
== Compliant Solution
----
/Jack|Peter/.test('John'); // returns 'false'
----