<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This is an issue when an <code>@InvocableMethod</code> annotation lacks a <code>label</code> and <code>description</code>, or when these attributes contain empty values.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Einstein Copilot is a conversational AI assistant that helps Salesforce users execute commands using natural language. When developers create custom actions using Apex methods annotated with <code>@InvocableMethod</code>, Copilot relies on the <code>label</code> and <code>description</code> attributes to understand what each method does.</p>
</div>
<div class="paragraph">
<p>Without clear and descriptive labels and descriptions, Einstein Copilot cannot effectively match user conversation intents with the appropriate actions. This leads to poor user experience where:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>Users' natural language requests may not trigger the intended custom actions</p>
</li>
<li>
<p>Copilot may suggest irrelevant or incorrect actions</p>
</li>
<li>
<p>The AI assistant becomes less helpful and intuitive to use</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>The <code>label</code> should provide a concise, human-readable name for the action, while the <code>description</code> should explain what the method accomplishes. These descriptions are crucial for Copilot&#8217;s intent matching algorithm to work effectively.</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="paragraph">
<p>Poor intent matching in Einstein Copilot leads to frustrated users who cannot effectively interact with custom actions through natural language. This reduces the value of the conversational AI assistant and may result in users avoiding the feature entirely.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Add meaningful <code>label</code> and <code>description</code> attributes to the <code>@InvocableMethod</code> annotation. The label should be a clear, concise name for the action, and the description should explain what the method does.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-apex" data-lang="apex">@InvocableMethod
public static List&lt;String&gt; sayHello(List&lt;String&gt; name) {
    return new List&lt;String&gt;{ 'Hello ' + name[0] };
} // Noncompliant</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-apex" data-lang="apex">@InvocableMethod(label='Say Hello' description='Say Hello to the user')
public static List&lt;String&gt; sayHello(List&lt;String&gt; name) {
    return new List&lt;String&gt;{ 'Hello ' + name[0] };
}</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>InvocableMethod Annotation - <a href="https://developer.salesforce.com/docs/atlas.en-us.apexcode.meta/apexcode/apex_classes_annotation_InvocableMethod.htm">Official Salesforce documentation for the InvocableMethod annotation</a></p>
</li>
<li>
<p>Einstein Copilot Custom Actions - <a href="https://help.salesforce.com/s/articleView?id=release-notes.rn_einstein_copilot_custom_actions.htm&amp;release=248&amp;type=5">Guide to creating custom actions for Einstein Copilot</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>