== Why is this an issue?

Multiple imports from the same module should be merged together to improve readability.


=== Noncompliant code example

[source,javascript]
----
import { B1 } from 'b';
import { B2 } from 'b'; // Noncompliant
----


=== Compliant solution

[source,javascript]
----
import { B1, B2 } from 'b';
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Merge this import with another one from the same module on line N.


=== Highlighting

Primary: Import #1

Secondary: previous Import to be merged with


endif::env-github,rspecator-view[]
