<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>The C90 standard allows implicit typing of variables and functions, and some C compilers still support legacy code by allowing implicit typing. But it should not be used for new code because it might lead to confusion.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">extern x;
const x;
static fun(void);
typedef ( *pfi ) ( void );</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">extern int16_t x;
const int16_t x;
static int16_t fun(void);
typedef int16_t ( *pfi ) ( void );</code></pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p>MISRA C:2004, 8.2 - Whenever an object or function is declared or defined, its type shall be explicitly stated</p>
</li>
<li>
<p>MISRA C:2012, 8.1 - Types shall be explicitly specified</p>
</li>
<li>
<p><a href="https://wiki.sei.cmu.edu/confluence/x/8NUxBQ">CERT, DCL31-C.</a> - Declare identifiers before using them</p>
</li>
</ul>
</div>
<hr>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Explicitly specify a type for this declaration.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_is_duplicated_by_s1272">is duplicated by: <a data-rspec-id="S1272" class="rspec-auto-link">S1272</a></h3>

</div>
<div class="sect2">
<h3 id="_on_16_oct_2014_142259_ann_campbell_wrote">on 16 Oct 2014, 14:22:59 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~samuel.mercier] please:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>fill in the appropriate reference field(s).</p>
</li>
<li>
<p>provide a text description</p>
</li>
</ul>
</div>
</div>
<div class="sect2">
<h3 id="_on_28_oct_2014_101800_evgeny_mandrikov_wrote">on 28 Oct 2014, 10:18:00 Evgeny Mandrikov wrote:</h3>
<div class="paragraph">
<p>Must be updated to match MISRA C:2012 8.1.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_15_nov_2014_102512_evgeny_mandrikov_wrote">on 15 Nov 2014, 10:25:12 Evgeny Mandrikov wrote:</h3>
<div class="paragraph">
<p>Existing description actually matches with MISRA C:2004 8.1 and not with MISRA C:2004 8.2, so removed.</p>
</div>
</div>
</div>
</div>