<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Explicitly declaring a function&#8217;s return data type makes it easier to use the function correctly.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-vb6" data-lang="vb6">Function BinarySearch(. . .)
    [. . .]
End Function</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-vb6" data-lang="vb6">Function BinarySearch(. . .) As Boolean
    [. . .]
End Function</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Specify the return data type for XXX</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_30_mar_2015_151018_evgeny_mandrikov_wrote">on 30 Mar 2015, 15:10:18 Evgeny Mandrikov wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] I added VB6 as "targeted language" for parent task.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_30_mar_2015_185336_ann_campbell_wrote">on 30 Mar 2015, 18:53:36 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>Thanks [~evgeny.mandrikov]</p>
</div>
</div>
<div class="sect2">
<h3 id="_is_duplicated_by_s1272">is duplicated by: <a data-rspec-id="S1272" class="rspec-auto-link">S1272</a></h3>

</div>
<div class="sect2">
<h3 id="_on_16_oct_2014_142259_ann_campbell_wrote">on 16 Oct 2014, 14:22:59 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~samuel.mercier] please:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>fill in the appropriate reference field(s).</p>
</li>
<li>
<p>provide a text description</p>
</li>
</ul>
</div>
</div>
<div class="sect2">
<h3 id="_on_28_oct_2014_101800_evgeny_mandrikov_wrote">on 28 Oct 2014, 10:18:00 Evgeny Mandrikov wrote:</h3>
<div class="paragraph">
<p>Must be updated to match MISRA C:2012 8.1.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_15_nov_2014_102512_evgeny_mandrikov_wrote">on 15 Nov 2014, 10:25:12 Evgeny Mandrikov wrote:</h3>
<div class="paragraph">
<p>Existing description actually matches with MISRA C:2004 8.1 and not with MISRA C:2004 8.2, so removed.</p>
</div>
</div>
</div>
</div>