<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>In Blazor, when a <a href="https://learn.microsoft.com/en-us/aspnet/core/blazor/fundamentals/routing#route-parameters">route parameter</a> is used in a <a href="https://learn.microsoft.com/en-us/aspnet/core/blazor/fundamentals/routing#route-templates">routed component</a>, the router will populate the <a href="https://learn.microsoft.com/en-us/aspnet/core/blazor/components#component-parameters">component parameter</a> with the same name.</p>
</div>
<div class="paragraph">
<p>If the routed component does not contain a matching parameter, a runtime error will be raised:</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-text" data-lang="text">Unhandled exception rendering component: Object of type '&lt;component-type&gt;' does not have a property matching the name '&lt;route-parameter-name&gt;'.</code></pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Ensure to have a component parameter that matches the route parameter name, or remove the route parameter if not needed or used.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-csharp" data-lang="csharp">@page "/user/{Id:int}" // Noncompliant

@code {
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-csharp" data-lang="csharp">@page "/user/{Id:int}" // Compliant

@code {
    [Parameter]
    public int Id { get; set; }
}</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Microsoft Learn - <a href="https://learn.microsoft.com/en-us/aspnet/core/blazor/fundamentals/routing#route-parameters">ASP.NET Core Blazor routing and navigation - Route parameters</a></p>
</li>
<li>
<p>Microsoft Learn - <a href="https://learn.microsoft.com/en-us/aspnet/core/blazor/fundamentals/routing#route-templates">ASP.NET Core Blazor routing and navigation - Route templates</a></p>
</li>
<li>
<p>Microsoft Learn - <a href="https://learn.microsoft.com/en-us/aspnet/core/blazor/components#component-parameters">ASP.NET Core Razor components - Component parameters</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>