``++fixed bin(31)++`` type has been optimized to run faster than ``++int++`` especially when used in loops. For this reason, ``++fixed bin(31)++`` should be preferred to ``++int++``.

== Noncompliant Code Example

----
dcl i int(2) init (0);
do i = 1 to 20 by 1;
...
end;
----

== Compliant Solution

----
dcl i fixed bin(31) init (0);
do i = 1 to 20 by 1;
...
end;
----
