== Why is this an issue?

``++fixed bin(31)++`` type has been optimized to run faster than ``++int++`` especially when used in loops. For this reason, ``++fixed bin(31)++`` should be preferred to ``++int++``.


=== Noncompliant code example

[source,pli]
----
dcl i int(2) init (0);
do i = 1 to 20 by 1;
...
end;
----


=== Compliant solution

[source,pli]
----
dcl i fixed bin(31) init (0);
do i = 1 to 20 by 1;
...
end;
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Replace INT with FIXED BIN(31)


endif::env-github,rspecator-view[]
