<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Using recursion with paragraphs is not a problem in itself but it potentially exposes the software to endless loop. This is the case when there is no condition to end the recursion or when this condition is always false.</p>
</div>
<div class="paragraph">
<p>This rule raises an issue when a paragraph contains a <code>PERFORM</code> to itself to warn the developer that there is a risk of endless loop. This rule can also be used to fully prevent recursion to be used.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cobol" data-lang="cobol">PROCEDURE DIVISION.
   ...
   PERFORM READ-RELATED-REC-PARA
   ...
READ-RELATED-REC-PARA.
   ...
   CALL MY-MODULE
   IF MORE-RECS
      PERFORM READ-RELATED-REC-PARA
   ...</code></pre>
</div>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cobol" data-lang="cobol">999-ERROR.
      "Write to a log file"
      If "write fails:
         "Display an error message"
         PERFORM 999-ERROR.</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cobol" data-lang="cobol">PROCEDURE DIVISION.
   ...
   PERFORM READ-RELATED-REC-PARA
      UNTIL NO-MORE-RELATED-RECS
      OR MAX-NO-OF-RELATED-RECS
   ...
READ-RELATED-REC-PARA.
   ...
   CALL MY-MODULE
   IF MORE-RECS
      SET MORE-RELATED-RECS TO TRUE
      ADD 1 TO NO-OF-RELATED-RECS
   ELSE
      SET NO-MORE-RELATED-RECS TO TRUE
   END-IF
   ...</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Refactor this code to remove the "PERFORM XXX" to remove the risk of a potential infinite loop.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>PERFORM XXX</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_14_nov_2018_182658_ann_campbell_wrote">on 14 Nov 2018, 18:26:58 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~alexandre.gigleux] this seems to overlap <a data-rspec-id="S925" class="rspec-auto-link">RSPEC-925</a></p>
</div>
</div>
</div>
</div>