<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>HTML &lt;table&gt; elements should not be used for layout purposes as it can confuse screen readers. It is recommended to use CSS instead.</p>
</div>
<div class="paragraph">
<p>This rule raises an issue on every <code>&lt;table&gt;</code> element containing a <code>role</code> attribute set to <code>"presentation"</code> or <code>"none"</code>, which is how <a href="https://www.w3.org/WAI/tutorials/tables/tips/">W3C recommends</a> marks layout tables.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-html" data-lang="html">&lt;table role="presentation"&gt;&lt;!-- Noncompliant --&gt;
&lt;!-- ... --&gt;
&lt;/table&gt;</code></pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p><a href="https://developer.mozilla.org/en-US/docs/Learn/HTML/Introduction_to_HTML/Document_and_website_structure">Document and website structure</a></p>
</li>
</ul>
</div>
</div>
</div>