<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Argument-dependent lookup (ADL) adds additional associated namespaces to the set of scopes searched when lookup is performed for the names of called functions. A generic function found in one of these additional namespaces would be added to the overload set and chosen by overload resolution, which is inconsistent with developer expectation.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">template &lt;typename T&gt;
class B
{
public:
  B operator+ ( long &amp; rhs );
  void f ( )
  {
    *this + 10; // calls NS::operator+ and not B&lt;NS::A&gt;::operator+ when B is instantiated with NS::A
  }
};

namespace NS
{
  class A {
  public:
  };
  template &lt;typename T&gt;
  bool operator+ ( T, int32_t ); // Noncompliant, within associated namespace
}
template class B&lt;NS::A&gt;;</code></pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p>MISRA C&#43;&#43;:2008, 14-5-1</p>
</li>
</ul>
</div>
</div>
</div>