== Why is this an issue?

The multi-line comment syntax ``++/* ... */++`` should not be used for single line comments; the ``++--++`` syntax is more appropriate.


=== Noncompliant code example

[source,text]
----
/*
 These comment lines are Compliant
 comment 1
 comment 2
*/

/*
 This comment is also Compliant
 */

/* This comment is Noncompliant */
----


=== Compliant solution

[source,text]
----
/*
 These comment lines are Compliant
 comment 1
 comment 2
*/

/*
 This comment is also Compliant
 */

-- This comment is compliant 
----

