include::../description.adoc[]

== Noncompliant code example

With provided regular expression ``++^[a-z][a-zA-Z0-9_]*$++``:

[source,apex]
----
public void DoSomething(){...}
----

== Compliant Solution

[source,apex]
----
public void doSomething(){...}
----
