<div class="paragraph">
<p>Shared naming conventions allow teams to collaborate efficiently. This rule checks that all function names match a provided regular expression.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="paragraph">
<p>With provided regular expression <code>^[a-z][a-zA-Z0-9_]*$</code>:</p>
</div>
<div class="listingblock">
<div class="content">
<pre>public void DoSomething(){...}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>public void doSomething(){...}</pre>
</div>
</div>
</div>
</div>