<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule allows banning usage of certain constructors.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="paragraph">
<p>Given parameters:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>className: java.util.Date</p>
</li>
<li>
<p>argumentTypes: java.lang.String</p>
</li>
</ul>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">Date birthday;
birthday = new Date("Sat Sep 27 05:42:21 EDT 1986");  // Noncompliant
birthday = new Date(528176541000L); // Compliant</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove this forbidden initialization</p>
</div>
</div>
<div class="sect2">
<h3 id="_parameters">Parameters</h3>
<div class="sidebarblock">
<div class="content">
<div class="title">className</div>
<div class="paragraph">
<p>Name of the class whose constructor is forbidden. This parameter is mandatory, if absent the rule is disabled.</p>
</div>
</div>
</div>
<div class="sidebarblock">
<div class="content">
<div class="title">argumentTypes</div>
<div class="paragraph">
<p>Comma-delimited list of argument types, E.G. java.lang.String, int[], int</p>
</div>
</div>
</div>
<div class="sidebarblock">
<div class="content">
<div class="title">allOverloads</div>
<div class="paragraph">
<p>Boolean. Set to true to flag all overloads regardless of parameter type (default: false)</p>
</div>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_deprecates_s1212">deprecates: <a data-rspec-id="S1212" class="rspec-auto-link">S1212</a></h3>

</div>
<div class="sect2">
<h3 id="_relates_to_s2253">relates to: <a data-rspec-id="S2253" class="rspec-auto-link">S2253</a></h3>

</div>
</div>
</div>