``++dynamic_cast++`` allow to convert pointers from one class to another, following the class inheritance hierarchy (note that in many cases, the need to perform a ``++dynamic_cast++`` is an indicator of poor class design, for instance not following the Liskov substitution principle).


If the requested conversion is not possible, the result of the conversion is a null pointer to the requested class.

An implication of this is that result of ``++dynamic_cast++`` should be tested before using it to avoid dereferencing a null pointer.


== Noncompliant Code Example

[source,cpp]
----
struct Base {
    virtual ~Base() = default;
};

struct Derived: Base {
    virtual void method() {}
};

void f(Base *b) {
    auto d = dynamic_cast<Derived*>(b);
    d->method(); // Noncompliant, d could be null
}
----


== Compliant Solution

[source,cpp]
----
struct Base {
    virtual ~Base() = default;
};

struct Derived: Base {
    virtual void method() {}
};

void f(Base *b) {
    if (auto d = dynamic_cast<Derived*>(b)) {
        d->method();
    }
}
----


== See

* https://github.com/isocpp/CppCoreGuidelines/blob/036324/CppCoreGuidelines.md#c148-use-dynamic_cast-to-a-pointer-type-when-failure-to-find-the-required-class-is-considered-a-valid-alternative[{cpp} Core Guidelines C.148] - Use dynamic_cast to a pointer type when failure to find the required class is considered a valid alternative


ifdef::env-github,rspecator-view[]
'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
