<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when a name listed in the <code>__all__</code> property of a module has not been defined.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>The <code>__all__</code> property of a module is used to define the list of names that will be imported when performing a wildcard import of this module, i.e. when <code>from mymodule import *</code> is used.</p>
</div>
<div class="paragraph">
<p>In the following example:</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python"># mymodule.py

def foo(): ...
def bar(): ...

__all__ = ["foo"]</code></pre>
</div>
</div>
<div class="paragraph">
<p>Executing <code>from mymodule import *</code> from a different module will only import <code>foo</code>.</p>
</div>
<div class="paragraph">
<p>This list can only reference defined names, otherwise an <code>AttributeError</code> will be raised when the module is imported.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">from mymodule import my_func

__all__ = ["unknown_func"]  # Noncompliant: "unknown_func" is undefined</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">from mymodule import my_func

__all__ = ["my_func"]</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p><a href="https://docs.python.org/3/tutorial/modules.html#importing-from-a-package">Python documentation  - importing * from a package</a></p>
</li>
</ul>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Change or remove this string; "XX" is not defined.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="ulist">
<ul>
<li>
<p>Primary: The string with an undefined name.</p>
</li>
<li>
<p>Secondary: the variable assignment if a variable is used.
message: 'Assigned here.'</p>
</li>
</ul>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_relates_to_s2823">relates to: <a data-rspec-id="S2823" class="rspec-auto-link">S2823</a></h3>

</div>
<div class="sect2">
<h3 id="_relates_to_s3827">relates to: <a data-rspec-id="S3827" class="rspec-auto-link">S3827</a></h3>

</div>
</div>
</div>