<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when a dictionary is iterated over without explicitly calling the <code>.items()</code> method.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>When iterating directly over a dictionary e.g., <code>for k, v in some_dict:</code> or <code>{k: v for k, v in some_dict}</code>, Python iterates over the dictionary&#8217;s keys by default. If the intention is to access both the key and the value, omitting <code>.items()</code> leads to unexpected behavior. In such cases, the <code>k</code> variable would receive the key, and the <code>v</code> variable would attempt to unpack the key itself, which can lead to errors or subtle bugs if the key is iterable, like a <code>string</code>. For example, if a key is a <code>string</code> like <code>"hi"</code>, <code>k</code> would be <code>'h'</code> and <code>v</code> would be <code>'i'</code>.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>To fix this, simply append <code>.items()</code> to your dictionary when iterating.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">some_dict = { "k1": "v1", "k2": "v2"}

{k: v for k, v in some_dict} # Noncompliant: `v` will not receive the value, but the first character of the key</code></pre>
</div>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">some_dict = { "k1": "v1", "k2": "v2"}

for k, v in some_dict: # Noncompliant: `v` will not receive the value, but the first character of the key
  ...</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">some_dict = { "k1": "v1", "k2": "v2"}
{k: v for k, v in some_dict.items()}</code></pre>
</div>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">some_dict = { "k1": "v1", "k2": "v2"}
for k, v in some_dict.items():
  ...</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Python Documentation - <a href="https://docs.python.org/3/tutorial/datastructures.html#looping-techniques">Looping Techniques</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>