There is no good excuse for an empty class. If it's being used simply as a common extension point, it should be replaced with an ``++Interface++``. If it was stubbed in as a placeholder for future development it should be fleshed-out. In any other case, it should be eliminated.

== Noncompliant Code Example

[source,vbnet]
----
Public Class Empty ' Noncompliant

End Class
----

== Compliant Solution

[source,vbnet]
----
Public Interface IEmpty

End Interface
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]