include::../description.adoc[]

== Noncompliant Code Example

[source,vbnet]
----
Public Class Empty ' Noncompliant

End Class
----

== Compliant Solution

[source,vbnet]
----
Public Interface IEmpty

End Interface
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]