== Why is this an issue?

It can be extremely confusing when a ``++for++`` condition tests a variable which is not updated inside the ``++for++`` post statement.

=== Noncompliant code example

[source,go]
----
for i := 1; i <= 10; j++ { // Noncompliant
	// ...
}
----

=== Compliant solution

[source,go]
----
for i := 1; i <= 10; i++ {
	// ...
}
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

This loop's stop condition tests variable which is not incremented in update clause.


include::../highlighting.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
