<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>A class without an explicit extension of object (<code>class ClassName(object)</code>) is considered an old-style class, and <code>__slots__</code> declarations are ignored in old-style classes. Having such a declaration in an old-style class could be confusing for maintainers and lead them to make false assumptions about the class.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">class A:
  __slots__ = ["id"]  # Noncompliant; this is ignored

  def __init__(self):
    self.id = id
    self.name = "name"  # name wasn't declared in __slots__ but there's no error

a = A()</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">class A(object):  # Converting to a new-style class is the preferred method of addressing this issue
  __slots__ = ["id"]

  def __init__(self):
    self.id = id
    self.name = "name"  # "name" is not listed in __slots__, so as expected there is an error in this line

a = A()</code></pre>
</div>
</div>
<div class="paragraph">
<p>or</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">class A:

  def __init__(self):
    self.id = id
    self.name = "name"

a = A()</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Either add "(object)" to this class declaration or remove the "<em>slots</em>" declaration.</p>
</div>
</div>
</div>
</div>