== Why is this an issue?

Using a type parameter when you don't have to simply obfuscates the code. Inserting an unnecessary type parameter in an unparameterized method call will compile, but confuse maintainers. 


=== Noncompliant code example

[source,java]
----
void doTheThing() {
  // ...
}

//...
this.<String>doTheThing();  // Noncompliant
----


=== Compliant solution

[source,java]
----
void doTheThing() {
  // ...
}

//...
this.doTheThing();
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Remove "xxx".


=== Highlighting

Unnecessary type param


endif::env-github,rspecator-view[]
