== Why is this an issue?

Having too many steps in a single job or procedure makes it harder to understand and maintain.
It can also mean that it is aggregating too many responsibilities and should be split.

== How to fix it

=== Code examples

==== Noncompliant code example

With a maximum number of step configured to 3:
[source,jcl,diff-id=1,diff-type=noncompliant]
----
//MYJOB  JOB
//STEP01 EXEC PGM=DOTHING
//STEP02 EXEC PGM=DOTHING
//STEP03 EXEC PGM=DOTHING
//STEP04 EXEC PGM=DOTHING   <-- Noncompliant
----

==== Compliant solution

[source,jcl,diff-id=1,diff-type=compliant]
----
//MYJOB  JOB
//STEP01 EXEC PGM=DOTHING
//STEP02 EXEC PGM=DOTHING
//STEP03 EXEC PGM=DOTHING
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Parameters

.Max
****
----
50
----
Maximum authorized steps in a job.
****

'''
endif::env-github,rspecator-view[]

== Resources

=== Documentation

* https://www.ibm.com/docs/en/zos/3.1.0?topic=tasks-entering-jobs[IBM reference - Entering Jobs]
