<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>The <code>asm</code> declaration is available to all C&#43;&#43; implementations, allowing a consistent mechanism to be used.</p>
</div>
<div class="paragraph">
<p>However, the parameters to <code>asm</code> are still implementation-defined.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">void Delay ( void )
{
#pragma asm
  "NOP" // Noncompliant
#pragma endasm
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">void Delay ( void )
{
  asm ( "NOP" ); // Compliant
}</code></pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p>MISRA C&#43;&#43; 2008, 7-4-2</p>
</li>
</ul>
</div>
<hr>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Refactor this code using the "asm" statement.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_is_related_to_s784">is related to: <a data-rspec-id="S784" class="rspec-auto-link">S784</a></h3>

</div>
<div class="sect2">
<h3 id="_on_16_oct_2014_145627_ann_campbell_wrote">on 16 Oct 2014, 14:56:27 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~samuel.mercier] see also MISRA C:2012, Dir 4.3</p>
</div>
<div class="paragraph">
<p>In general, the 2012 version always has much better descriptions</p>
</div>
<div class="paragraph">
<p>This rule is also defined in MISRA C:2004, 2.1</p>
</div>
</div>
</div>
</div>