This rule reports when a file is placed in a location which deviates from the intended architecture.

== Why is this an issue?

This project has an intended architecture defined, which specifies high-level components and their allowed relationships.
This might include that specific files should be located in specific directories, or that directories are renamed.
This issue is being raised because this file’s location deviates from the intended architecture.

=== What is the potential impact?

When the code location deviates from the intended architecture, it gradually erodes the project’s structure.
As these deviations accumulate, the code becomes more complex and less consistent with the intent, making future development increasingly challenging.

== How to fix it

Move the relevant file to its intended location or rename the relevant directory as intended.

=== Code examples

==== Noncompliant code example

If the intended architecture declares developers should use the directory name `./panel`, not `./panels`, then the following code example is noncompliant:

[source,javascript,diff-id=1,diff-type=noncompliant]
----
# panels/ShowCustomersPanel.js
class ShowCustomersPanel ...
----

==== Compliant solution

The solution is to move the file to its appropriate directory, or respectively, to rename the directory `panels` to `panel`.

[source,javascript,diff-id=1,diff-type=compliant]
----
# panel/ShowCustomersPanel.js
class ShowCustomersPanel ...
----
