<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>The variable is not available in the current scope.
It will be evaluated to an empty value.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>The variables defined by <code>ARG</code> instruction have a scope from the definition to the end of the build stage where it was defined.
If it was defined in the beginning of the Dockerfile (outside of any build stage), then its scope is restricted to only <code>FROM</code> instructions.
Outside of their scope, variables will be resolved to empty string which may lead to unintended behaviour.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-docker" data-lang="docker">ARG SETTINGS
FROM busybox
RUN ./run/setup $SETTINGS</code></pre>
</div>
</div>
<div class="paragraph">
<p>In this case the <code>$SETTINGS</code> variable will be evaluated to empty string.</p>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-docker" data-lang="docker">FROM busybox
ARG SETTINGS
RUN ./run/setup $SETTINGS</code></pre>
</div>
</div>
<div class="paragraph">
<p>In this case when Dockerfile will be built with the flag <code>--build-arg SETTINGS=--some-settings</code> the flag <code>--some-settings</code> will be passed to the <code>RUN</code> instruction.</p>
</div>
</div>
<div class="sect3">
<h4 id="_noncompliant_code_example_2">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-docker" data-lang="docker">ARG SETTINGS="--default-settings"
FROM busybox
RUN ./run/setup $SETTINGS</code></pre>
</div>
</div>
<div class="paragraph">
<p>In this case the <code>$SETTINGS</code> variable will be evaluated to empty string.</p>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution_2">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-docker" data-lang="docker">ARG SETTINGS="--default-settings"
FROM busybox
ARG SETTINGS
RUN ./run/setup $SETTINGS</code></pre>
</div>
</div>
<div class="paragraph">
<p>In this case the flag <code>--default-settings</code> will be passed to <code>RUN</code> instruction (unless another value is provided during build time).</p>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_how_does_this_work">How does this work?</h3>
<div class="paragraph">
<p>The <code>FROM</code> instruction starts a new build stage where variables defined by previous <code>ARG</code> instructions are out of this new scope.
To make it accessible for the build stage they need to be defined after the <code>FROM</code> instruction.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p><a href="https://docs.docker.com/engine/reference/builder/#scope">ARG scope - Dockerfile reference</a></p>
</li>
</ul>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Include the <code>ARG</code> instruction in the build stage where it is used.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>Highlight usage of the variable where it is not accessible.</p>
</div>
<hr>
</div>
</div>
</div>