Jump statements, such as ``++RETURN++`` and ``++CONTINUE++`` let you change the default flow of program execution, but jump statements that direct the control flow to the original direction are just a waste of keystrokes.

== Noncompliant Code Example

----
CREATE PROCEDURE print_numbers AS
BEGIN
  FOR i in 1..4 LOOP
    DBMS_OUTPUT.PUT_LINE(i);
    CONTINUE; -- Noncompliant
  END LOOP;
  RETURN;     -- Noncompliant
END;
----

== Compliant Solution

----
CREATE PROCEDURE print_numbers AS
BEGIN
  FOR i in 1..4 LOOP
    DBMS_OUTPUT.PUT_LINE(i);
  END LOOP;
END;
----

ifdef::rspecator-view[]
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::rspecator-view[]
