include::../description.adoc[]

== Noncompliant Code Example

[source,vbnet]
----
Public Function Transform(Value As Integer) As Integer
    If Value <= 0 Then Return Value
    Dim Number As Integer = Integer.MaxValue
    Return Number + Value       ' Noncompliant
End Function
----

== Compliant Solution

[source,vbnet]
----
Public Function Transform(Value As Integer) As Long
    If Value <= 0 Then Return Value
    Dim Number As Long = Integer.MaxValue
    Return Number + Value
End Function
----

include::../rspecator.adoc[]
