<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule allows banning certain methods.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="paragraph">
<p>Given parameters:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>className:java.lang.String</p>
</li>
<li>
<p>methodName: replace</p>
</li>
<li>
<p>argumentTypes: java.lang.CharSequence, java.lang.CharSequence</p>
</li>
</ul>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-text" data-lang="text">String name;
name.replace("A","a");  // Noncompliant</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove this forbidden call</p>
</div>
</div>
<div class="sect2">
<h3 id="_parameters">Parameters</h3>
<div class="sidebarblock">
<div class="content">
<div class="title">className</div>
<div class="paragraph">
<p>Name of the class whose method is forbidden</p>
</div>
</div>
</div>
<div class="sidebarblock">
<div class="content">
<div class="title">methodName</div>
<div class="paragraph">
<p>Name of the forbidden method</p>
</div>
</div>
</div>
<div class="sidebarblock">
<div class="content">
<div class="title">argumentTypes</div>
<div class="paragraph">
<p>Comma-delimited list of argument types, E.G. java.lang.String, int[], int</p>
</div>
</div>
</div>
<div class="sidebarblock">
<div class="content">
<div class="title">allOverloads</div>
<div class="paragraph">
<p>Boolean. Set to true to flag all overloads regardless of parameter type (default: false)</p>
</div>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_deprecates_s1212">deprecates: <a data-rspec-id="S1212" class="rspec-auto-link">S1212</a></h3>

</div>
<div class="sect2">
<h3 id="_is_related_to_s4011">is related to: <a data-rspec-id="S4011" class="rspec-auto-link">S4011</a></h3>

</div>
</div>
</div>