This rule allows banning certain methods.


== Noncompliant Code Example

Given parameters:

* className:java.lang.String
* methodName: replace
* argumentTypes: java.lang.CharSequence, java.lang.CharSequence

[source,text]
----
String name;
name.replace("A","a");  // Noncompliant
----

