This rule raises an issue when `logging.getLogger()` is called with a string literal instead of `__name__`.

== Why is this an issue?

Using hardcoded strings when creating logger instances prevents proper logger namespacing and organization. The `__name__` variable automatically provides the current module's dotted path (like `myapp.views` or `myproject.utils.helpers`), which creates a hierarchical logger structure that matches your project organization.

This automatic namespacing offers several benefits:

* **Maintainability**: If you move or rename modules, logger names update automatically without code changes
* **Debugging**: Clear, structured logger names help you quickly identify where log messages originate
* **Configuration flexibility**: Django's logging configuration can target specific modules or entire namespaces using the hierarchical structure
* **Collision prevention**: Using `__name__` ensures unique logger names across your project

Hardcoded logger names break this structure and make logging configuration more difficult. You lose the ability to configure logging at different levels of your application hierarchy.

=== What is the potential impact?

Using hardcoded logger names reduces code maintainability and makes logging configuration more complex. It can lead to logger name collisions and makes it harder to trace the source of log messages during debugging.

== How to fix it

Replace the hardcoded string with `__name__` to automatically use the current module's dotted path as the logger name.

=== Code examples

==== Noncompliant code example

[source,python,diff-id=1,diff-type=noncompliant]
----
import logging

logger = logging.getLogger('my_custom_logger')  # Noncompliant
----

==== Compliant solution

[source,python,diff-id=1,diff-type=compliant]
----
import logging

logger = logging.getLogger(__name__)
----

== Resources

=== Documentation

 * Django Logging Documentation - https://docs.djangoproject.com/en/stable/howto/logging/[Official Django guide on configuring and using logging]

 * Python Logging Documentation - https://docs.python.org/3/library/logging.html[Python's built-in logging module documentation]

=== Related rules

 * S3416 - https://rules.sonarsource.com/java/RSPEC-3416/[Loggers should be named for their enclosing classes (Java)]

 * S6672 - https://rules.sonarsource.com/csharp/RSPEC-6672/[Loggers should be named for their enclosing classes (C#)]
