<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when strings or bytes are concatenated implicitly.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Python concatenates adjacent string or byte literals at compile time. It means that <code>"a" "b"</code> is equivalent to <code>"ab"</code>.
This is sometimes used to split a long string on multiple lines. However an implicit string concatenation can also be very confusing.
In the following contexts it might indicate that a comma was forgotten:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>when the two strings are on the same line. This could be interpreted as an incorrectly formatted tuple (parentheses are not mandatory to create a tuple, only the comma is).</p>
</li>
<li>
<p>when the strings are in a list, a set or a tuple.</p>
</li>
</ul>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">def func():
    return "item1" "item2"  # Noncompliant: a comma is missing to return a tuple.

["1"  # Noncompliant: a comma is missing.
 "2",
 "a very"  # Noncompliant: a "+" is missing.
 "long string"]</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">def func():
    return "item1", "item2"

["1",
 "2",
 "a very" +
 "long string"]</code></pre>
</div>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_exceptions">Exceptions</h3>
<div class="paragraph">
<p>This rule will not raise any issues when there is a visible reason for the string concatenation:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>when the quotes used for both strings are different. This can be used to avoid escaping quotes.</p>
</li>
<li>
<p>when the strings or bytes have different prefixes, i.e. "f" for f-strings, "r" for raw, "u" for unicode and no prefix for normal strings.</p>
</li>
<li>
<p>when strings are visibly split to avoid long lines of code, i.e. when the first string ends with a space, punctuation or <code>\n</code>.</p>
</li>
</ul>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>When the strings are on multiple lines:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>"Add a "+" operator to make the string concatenation explicit; or did you forget a comma?"
When the strings are on the same line:</p>
</li>
<li>
<p>"Merge these implicitly concatenated strings; or did you forget a comma?"</p>
</li>
</ul>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>Primary: The end quote of the first concatenated string/bytes literal</p>
</div>
<div class="paragraph">
<p>Secondary:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>location: the starting quote of the next string/bytes/literal</p>
</li>
<li>
<p>no message</p>
</li>
</ul>
</div>
</div>
</div>
</div>