== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,cpp]
----
if (x == 0) {
  doSomething();
} else if (x == 1) {
  doSomethingElse();
}
----

=== Compliant solution

[source,cpp]
----
if (x == 0) {
  doSomething();
} else if (x == 1) {
  doSomethingElse();
} else {
  error();
}
----

=== Exceptions

When all branches of an ``++if++``-``++else if++`` end with ``++return++``, ``++break++`` or ``++throw++``, the code that comes after the ``++if++`` implicitly behaves as if it was in an ``++else++`` clause. This rule will therefore ignore that case.

== Resources

* MISRA C:2004, 14.10 - All if...else if constructs shall be terminated with an else clause.
* MISRA {cpp}:2008, 6-4-2 - All if...else if constructs shall be terminated with an else clause.
* MISRA C:2012, 15.7 - All if...else if constructs shall be terminated with an else statement
* https://wiki.sei.cmu.edu/confluence/x/RtYxBQ[CERT, MSC01-C.] - Strive for logical completeness

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
