<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when hash objects are stored using Redis string methods like <code>set</code> instead of appropriate hash methods.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Redis has different data types, and each type has its own set of operations. When you use <code>set</code> to store a Ruby hash object, Redis converts the hash to its string representation (like <code>"{\"field\"&#8658;\"value\"}"</code>) rather than storing it as a proper Redis hash data structure.</p>
</div>
<div class="paragraph">
<p>This creates several problems:</p>
</div>
<div class="ulist">
<ul>
<li>
<p><strong>Data retrieval issues</strong>: When you retrieve the data with <code>get</code>, you get back a string instead of a hash, requiring manual parsing</p>
</li>
<li>
<p><strong>Loss of Redis hash functionality</strong>: You cannot use Redis hash operations like <code>hget</code>, <code>hset</code>, or <code>hgetall</code> on string-stored data</p>
</li>
<li>
<p><strong>Type conflicts</strong>: If you later try to use hash operations on a key that was stored as a string, Redis will return a "wrong kind of value" error</p>
</li>
<li>
<p><strong>Performance impact</strong>: String serialization and parsing add unnecessary overhead compared to native Redis hash operations</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>Redis hash data structures are specifically designed for storing field-value pairs efficiently and provide atomic operations for individual fields.</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="paragraph">
<p>Using inappropriate Redis methods can lead to data retrieval failures, runtime errors when mixing operation types, and reduced application performance due to unnecessary serialization overhead.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Use Redis hash methods like <code>mapped_hmset</code> and <code>hgetall</code> for storing and retrieving hash data structures.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-ruby" data-lang="ruby">hash = {"field" =&gt; "value", "field2" =&gt; "value2"}
$redis.set('data', hash)  # Noncompliant
data = $redis.get('data') # Returns string, not hash</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-ruby" data-lang="ruby">hash = {"field" =&gt; "value", "field2" =&gt; "value2"}
$redis.mapped_hmset('data', hash)
data = $redis.hgetall('data')</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Redis Hash Commands - <a href="https://redis.io/commands/?group=hash">Official Redis documentation for hash data type commands</a></p>
</li>
<li>
<p>Redis Ruby Client Documentation - <a href="https://github.com/redis/redis-rb">Documentation for the Redis Ruby client library</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>