include::../description.adoc[]

== Noncompliant Code Example

----
if someCondition; puts "hello"; end
----

== Compliant Solution

----
if someCondition
  puts "hello"
end
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../highlighting.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
