<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Type names shouldn&#8217;t be used parameter names of functions and methods.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Dart allows to use type names as parameter names of functions and methods.</p>
</div>
<div class="paragraph">
<p>This can be misleading, as the developer may think that the parameter is of the type corresponding to the specified name, but that is not the case.</p>
</div>
<div class="paragraph">
<p>A scenario leading to this issue is, for example, when the developer forgets to give a name to the parameter, or assumes that omitting the name and only specifying the type would be enough to make the parameter of that type (as in C++). That is not the case, as the parameter would be of type <code>dynamic</code>.</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-dart" data-lang="dart">void foo(int) { // int is a dynamic parameter, not an integer one
  print(int);
}</code></pre>
</div>
</div>
<div class="paragraph">
<p>Dart also allows to use a type name both as type and as a name in the same declaration, which can be additional source of confusion:</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-dart" data-lang="dart">void foo(int int) { // the type may later change, but the name would remain int
  print(int);
}</code></pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Give the parameter a different name, describing its purpose.</p>
</div>
<div class="paragraph">
<p>Make the type of the parameter explicit, if the parameter was not meant to be <code>dynamic</code>.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-dart" data-lang="dart">void foo(int) {
  print(int);
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-dart" data-lang="dart">void foo(String text) {
  print(text);
}</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Dart Docs - <a href="https://dart.dev/tools/linter-rules/avoid_types_as_parameter_names">Dart Linter rule - avoid_types_as_parameter_names</a></p>
</li>
</ul>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="ulist">
<ul>
<li>
<p>The parameter name '&lt;parameter name&gt;' matches a visible type name.</p>
</li>
</ul>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>The identifier of the parameter name, in the parameter list of a function or method declaration: e.g. <code>int</code> in <code>void foo(int) &#8230;&#8203;</code>.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
</div>
</div>