== Why is this an issue?

This rule allows you to track the usage of the ``++@SuppressWarnings++`` mechanism.


=== Noncompliant code example

With a parameter value of "unused" :

[source,java]
----
@SuppressWarnings("unused")
@SuppressWarnings("unchecked")  // Noncompliant
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Suppressing the 'xxxxxx' warning is not allowed.


=== Parameters

.listOfWarnings
****

Comma separated list of warnings that can be suppressed. Example: 'unchecked, cast, all, boxing'. An empty list means that no warning can be suppressed.
****


'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
