== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

With a parameter value of "unused" :

[source,java]
----
@SuppressWarnings("unused")
@SuppressWarnings("unchecked")  // Noncompliant
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::parameters.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
