include::../description.adoc[]

== Noncompliant Code Example

With a parameter value of "unused" :

----
@SuppressWarnings("unused")
@SuppressWarnings("unchecked")  // Noncompliant
----

ifdef::rspecator-view[]
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::rspecator-view[]
