If a string fits on a single line, without concatenation and escaped newlines, you should probably continue to use a string literal.


== Noncompliant Code Example

----
String question = """
              What's the point, really?""";
----


== Compliant Solution

----
String question = "What's the point, really?";
----


== See

* https://openjdk.java.net/jeps/378[JEP 378: Text Blocks]
* https://cr.openjdk.java.net/~jlaskey/Strings/TextBlocksGuide_v9.html[Programmer's Guide To Text Blocks], by Jim Laskey and Stuart Marks


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

endif::env-github,rspecator-view[]
