<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>NUnit <code>TestFixture</code>s may only have one <code>[SetUp]</code> method. Any more than that and the <code>TestFixture</code> will compile but not run.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-csharp" data-lang="csharp">namespace NUnit.Tests
{
  using System;
  using NUnit.Framework;

  [TestFixture]
  public class MyTests
  {
    [SetUp] public void Init()
    { /* ... */ }

    [Setup] public void Prep()  // Noncompliant
    { /* ... */ }
  }
}</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_24_nov_2015_142617_ann_campbell_wrote">on 24 Nov 2015, 14:26:17 Ann Campbell wrote:</h3>
<div class="paragraph">
<p><a href="http://www.nunit.org/index.php?p=setup&amp;r=2.2" class="bare">http://www.nunit.org/index.php?p=setup&amp;r=2.2</a></p>
</div>
</div>
<div class="sect2">
<h3 id="_on_24_nov_2015_151336_ann_campbell_wrote">on 24 Nov 2015, 15:13:36 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>FYI [~tamas.vajk] I&#8217;m closing this based on your comment on MMF-124</p>
</div>
</div>
</div>
</div>