<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Call stacks containing lot of <code>PERFORM</code> statements is a key ingredient for making what&#8217;s known as "Spaghetti code".</p>
</div>
<div class="paragraph">
<p>Such code is hard to read, refactor and therefore maintain.</p>
</div>
<div class="paragraph">
<p>This rule supports both sections and paragraphs.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="paragraph">
<p>With a threshold of 3:</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cobol" data-lang="cobol">PERFORM FIRST.

FIRST.
  PERFORM SECOND.

SECOND.
  PERFORM THIRD.

THIRD.
  PERFORM FOURTH.  *&gt; Noncompliant

FOURTH.
  DISPLAY something.</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>This module can trigger long chain of up to 6 PERFORMs, for example using the ones at line: 123 &#8594; 23 &#8594; 54 &#8594; &#8230;&#8203;</p>
</div>
</div>
<div class="sect2">
<h3 id="_parameters">Parameters</h3>
<div class="sidebarblock">
<div class="content">
<div class="title">max</div>
<div class="listingblock">
<div class="content">
<pre>5</pre>
</div>
</div>
<div class="paragraph">
<p>Maximum allowed call stack</p>
</div>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_is_related_to_s1684">is related to: <a data-rspec-id="S1684" class="rspec-auto-link">S1684</a></h3>

</div>
</div>
</div>