== Why is this an issue?

Shared coding conventions allow teams to collaborate efficiently. This rule checks that type names match the provided regular expression.


=== Noncompliant code example

With the default regular expression ``++[a-zA-Z]([a-zA-Z0-9_]*[a-zA-Z0-9])?++``:

[source,sql]
----
DECLARE
  TYPE Collection_type_ IS VARRAY(42) OF PLS_INTEGER; -- Noncompliant
BEGIN
  NULL;
END;
/
----


=== Compliant solution

[source,sql]
----
DECLARE
  TYPE collectionType IS VARRAY(42) OF PLS_INTEGER;
BEGIN
  NULL;
END;
/
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

include::parameters.adoc[]

endif::env-github,rspecator-view[]
