include::../description.adoc[]

== Noncompliant Code Example

----
public string GetReadableStatus(Job j)
{
  return j.IsRunning ? "Running" : j.HasErrors ? "Failed" : "Succeeded";  // Noncompliant
}
----

== Compliant Solution

----
public string GetReadableStatus(Job j) 
{
  if (j.IsRunning) 
  {
    return "Running";
  }
  return j.HasErrors ? "Failed" : "Succeeded";
}
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../highlighting.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
