include::../rule.adoc[tag=why]

include::../rule.adoc[tag=how]

include::../rule.adoc[tag=examples]

==== Noncompliant code example

[source,html,diff-id=2,diff-type=noncompliant]
----
<label>Favorite food</label>
----

==== Compliant solution

[source,html,diff-id=2,diff-type=compliant]
----
<label>
  <MyCustomInput />
  Favorite food
</label>
----

include::../rule.adoc[tag=resources]
