include::../common/rule.adoc[tag=why]

include::../common/rule.adoc[tag=how]

include::../common/rule.adoc[tag=examples]

==== Noncompliant code example

[source,html,diff-id=2,diff-type=noncompliant]
----
<label>Favorite food</label>
----

==== Compliant solution

[source,html,diff-id=2,diff-type=compliant]
----
<label>
  <MyCustomInput />
  Favorite food
</label>
----

== Resources
=== Documentation

* MDN web docs - https://developer.mozilla.org/en-US/docs/Web/HTML/Element/label[The Label element]
* W3C - https://www.w3.org/WAI/WCAG21/Understanding/info-and-relationships[Info and Relationships]
* W3C - https://www.w3.org/WAI/WCAG21/Understanding/labels-or-instructions[Labels or Instructions]
# end::resources[]
