<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>COBOL programs that include <code>EXEC SQL INCLUDE</code> statements must be pre-processed with an IBM DB2 precompiler or coprocessor before compilation. The precompiler does not require a period after the closing <code>END-EXEC</code>, but the coprocessor does. For portability, a period should always be used.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cobol" data-lang="cobol">EXEC SQL INCLUDE ... END-EXEC  *&gt; Noncompliant; '.' missing</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cobol" data-lang="cobol">EXEC SQL INCLUDE ... END-EXEC.</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Add a period ('.') after this "END-EXEC".</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p><code>END-EXEC</code></p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_13_oct_2016_182241_nicolas_bontoux_wrote">on 13 Oct 2016, 18:22:41 Nicolas Bontoux wrote:</h3>
<div class="paragraph">
<p>References:</p>
</div>
<div class="ulist">
<ul>
<li>
<p><a href="https://www.ibm.com/support/knowledgecenter/SS6SG3_4.2.0/com.ibm.entcobol.doc_4.2/PGandLR/ref/rpdb211.htm">Differences in how the DB2 precompiler and coprocessor behave</a> (section: <em>Period at the end of EXEC SQL INCLUDE statements</em>)</p>
</li>
<li>
<p><a href="http://www.ibm.com/support/knowledgecenter/SS6SG3_4.2.0/com.ibm.entcobol.doc_4.2/MG/igymch1802.htm">Considerations when moving to the coprocessor</a> (section: <em>Period at the end of an EXEC SQL INCLUDE statement</em>)</p>
</li>
<li>
<p><a href="http://www.ibm.com/support/knowledgecenter/SSEPEK_11.0.0/apsg/src/tpc/db2z_sqlstatementscobol.html">Programming COBOL applications that issue SQL statements</a> : <em>For an SQL INCLUDE statement, the DB2 precompiler treats any text that follows the period after END-EXEC, and on the same line as END-EXEC, as a comment. The DB2 coprocessor treats this text as part of the COBOL program syntax.</em></p>
</li>
</ul>
</div>
</div>
</div>
</div>