== Why is this an issue?

Shared naming conventions allow teams to  collaborate efficiently. This rule checks that static non-final field names match a provided regular expression.


=== Noncompliant code example

With the default regular expression ``++^[a-z][a-zA-Z0-9]*$++``:

[source,java]
----
public final class MyClass {
   private static String foo_bar;
}
----


=== Compliant solution

[source,java]
----
class MyClass {
   private static String fooBar;
}
----



ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

include::parameters.adoc[]

'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
