== Why is this an issue?

Shared naming conventions allow teams to  collaborate efficiently. This rule checks that static non-final field names match a provided regular expression.


=== Noncompliant code example

With the default regular expression ``++^[a-z][a-zA-Z0-9]*$++``:

[source,java]
----
public final class MyClass {
   private static String foo_bar;
}
----


=== Compliant solution

[source,java]
----
class MyClass {
   private static String fooBar;
}
----



ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Rename this field "XXX" to match the regular expression ${format}.


=== Parameters

.format
****

----
^[a-z][a-zA-Z0-9]*$
----

Regular expression used to check the field names against.
****


'''
== Comments And Links
(visible only on this page)

=== on 9 Jun 2015, 13:45:21 Ann Campbell wrote:
FYI [~alexandre.gigleux] I've made a few updates.

=== on 9 Jun 2015, 13:48:13 Ann Campbell wrote:
Note for Java: same as RSPEC-116 but for static non-final fields

=== on 12 Jun 2015, 16:04:59 Ann Campbell wrote:
CodePro: Static Field Naming Convention

endif::env-github,rspecator-view[]
