== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,csharp]
----
public class Sample
{
    string myString = "foo";

    public void Go()
    {
        lock (myString) { } // Noncompliant
    }
}
----


=== Compliant solution

[source,csharp]
----
public class Sample
{
    private readonly static string thisLock = new object();

    public void Go()
    {
        lock (thisLock) { }
    }
}
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../highlighting.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
