== Why is this an issue?

The difficulty of understanding an expression increases for each of the ``++And++``, ``++Or++`` and ``++Xor++`` operators it contains.

A single expression's complexity should not become too high to keep the code readable.

=== Noncompliant code example

With the default threshold value of 3

[source,vb6]
----
If ( (condition1 And condition2) Or (condition3 And condition4) Or (condition5 And condition6) )
...
End If
----

=== Compliant solution

[source,vb6]
----
If ( myFirstCondition() Or mySecondCondition() Or myThirdCondition() )
...
End If
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../parameters.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
