The difficulty of understanding an expression increases for each of the <code>And</code>, <code>Or</code> and <code>Xor</code> operators it contains.
A single expression's complexity should not become too high to keep the code readable.

== Noncompliant Code Example

With the default threshold value of 3

----
If ( (condition1 And condition2) Or (condition3 And condition4) Or (condition5 And condition6) )
...
End If
----

== Compliant Solution

----
If ( myFirstCondition() Or mySecondCondition() Or myThirdCondition() )
...
End If
----
