This rule raises an issue when code uses `sort(false)` instead of `toSorted()`, or when the result of a mutating `sort()` call is assigned to a variable.

== Why is this an issue?

Groovy's collection sorting methods have different behaviors that can lead to confusion and subtle bugs.

The `sort()` method mutates the original collection and returns it. When you assign the result of `sort()` to a variable, you might not realize that the original collection has also been modified. This often indicates a misunderstanding of the method's behavior and can cause unintended side effects in your code.

Additionally, using `sort(false)` to create a new sorted collection is less clear than using `toSorted()`. The `false` parameter is not immediately obvious to readers, making the code harder to understand. The `toSorted()` method was introduced specifically to provide a clearer way to express the intent of creating a new sorted collection without modifying the original.

These patterns can lead to:

* Unexpected mutations of collections that should remain unchanged
* Difficult-to-debug issues where collections are modified in unexpected places
* Code that is harder to understand due to unclear method intentions
* Potential race conditions in concurrent code where shared collections are unexpectedly modified

=== What is the potential impact?

Using confusing sorting methods can lead to subtle bugs where collections are unexpectedly modified, making the code harder to debug and maintain. It can also reduce code readability and increase the likelihood of introducing errors when the code is modified later.

== How to fix it

Replace `sort(false)` with `toSorted()` to clearly express the intent of creating a new sorted collection. This makes the code more readable and eliminates the need to remember what the boolean parameter means.

=== Code examples

==== Noncompliant code example

[source,groovy,diff-id=1,diff-type=noncompliant]
----
def originalList = [3, 1, 2]
def sortedList = originalList.sort(false)  // Noncompliant
----

==== Compliant solution

[source,groovy,diff-id=1,diff-type=compliant]
----
def originalList = [3, 1, 2]
def sortedList = originalList.toSorted()  // Clear intention to create new sorted list
----

== Resources

=== Documentation

 * Groovy Collections - Sorting - http://docs.groovy-lang.org/latest/html/groovy-jdk/java/lang/Iterable.html#sort()[Official Groovy documentation for collection sorting methods]

 * Groovy toSorted() method - http://docs.groovy-lang.org/latest/html/groovy-jdk/java/lang/Iterable.html#toSorted()[Documentation for the toSorted() method that creates new sorted collections]

=== Related rules

 * AssignCollectionSort - https://codenarc.org/codenarc-rules-groovyism.html#assigncollectionsort-rule[CodeNarc rule that detects assignment of mutating sort() method results]
