== How to fix it

=== Code examples

include::../../common/fix/code-rationale.adoc[]

==== Noncompliant code example

[source,python,diff-id=1,diff-type=noncompliant]
----
import os
from flask import request

def update_environment():
    env_value = request.args.get('env')
    
    os.environ.update({"LD_PRELOAD": env_value})  # Noncompliant
----

==== Compliant solution

[source,python,diff-id=1,diff-type=compliant]
----
import os
import re
from flask import request

def update_environment():
    env_value = request.args.get('env')
    
    if env_value and re.match(r'^[a-zA-Z0-9]*$', env_value):
        os.environ.update({"SAFE_VAR": env_value})
----

=== How does this work?

include::../../common/fix/pre-approved-list.adoc[]
