== How to fix it

=== Code examples

include::../../common/fix/code-rationale.adoc[]

==== Noncompliant code example

[source,python,diff-id=1,diff-type=noncompliant]
----
import os
import subprocess
from flask import request

def run_command():
    env_value = request.args.get('env')
    
    envs = {**os.environ, "LD_PRELOAD": env_value}
    subprocess.run(["/path/to/executable"], env=envs)  # Noncompliant
----


==== Compliant solution

[source,python,diff-id=1,diff-type=compliant]
----
import os
import re
import subprocess
from flask import request

def run_command():
    env_value = request.args.get('env')
    
    if env_value and re.match(r'^[a-zA-Z0-9]*$', env_value):
        envs = {**os.environ, "SAFE_VAR": env_value}
        subprocess.run(["/path/to/executable"], env=envs)
----

=== How does this work?

include::../../common/fix/pre-approved-list.adoc[]
