<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Overriding a method just to call the overridden function from the base class without performing any other actions can be useless and misleading.</p>
</div>
<div class="paragraph">
<p>There are cases when it is justified because redeclaring the function allows some side effects:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>if a type (return type or a parameter type) is not exactly the same as the super member</p>
</li>
<li>
<p>if the covariant keyword is added to one of the parameters</p>
</li>
<li>
<p>if documentation comments are present on the member</p>
</li>
<li>
<p>if the member has annotations other than <code>@override</code></p>
</li>
<li>
<p>if the member is not annotated with <code>@protected</code>, and the super member is</p>
</li>
</ul>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-dart" data-lang="dart">class Child extends Parent {
  @override
  void foo() {
    super.foo();
  }
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-dart" data-lang="dart">class Child extends Parent {
  @override
  void foo() {
    bar();
  }
}</code></pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p>Dart Docs - <a href="https://dart.dev/tools/linter-rules/unnecessary_overrides">Dart Linter rule - unnecessary_overrides</a></p>
</li>
</ul>
</div>
<hr>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="ulist">
<ul>
<li>
<p>Unnecessary override.</p>
</li>
</ul>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>The identifier of the overridden method.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
</div>
</div>