<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Using <code>FORALL i IN x.first ... x</code>.last or <code>FORALL i IN 1 ... x.count</code> might fail when indexed collections are sparse as Oracle tries to access non-existent element(s) of x. <code>FORALL i IN INDICES OF x</code> syntax will always work including sparse collections. Thus using <code>FORALL i IN INDICES OF x</code> should be preferred as it makes code more robust and easier to review.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-sql" data-lang="sql">FORALL i IN 1 .. l_tab.COUNT  -- Non-Compliant
    INSERT INTO forall_test VALUES l_tab(i);

FORALL i IN l_tab.first .. l_tab.last  -- Non-Compliant
    INSERT INTO forall_test VALUES l_tab(i);</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-sql" data-lang="sql">FORALL i IN INDICES OF l_tab
    INSERT INTO forall_test VALUES l_tab(i);</code></pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p><a href="https://asktom.oracle.com/Misc/oramag/oracle-10g-adds-more-to-forall.html">Oracle 10g Adds More to FORALL</a></p>
</li>
<li>
<p><a href="https://stevenfeuersteinonplsql.blogspot.com/2019/03/using-sparse-collections-with-forall.html">Using sparse collections with FORALL</a> - Steven Feuerstein</p>
</li>
</ul>
</div>
<hr>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Replace "IN &#8230;&#8203;" with "INDICES OF &#8230;&#8203;"</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="ulist">
<ul>
<li>
<p><code>IN x.first ... x.last</code></p>
</li>
<li>
<p><code>1 ... x.count</code></p>
</li>
</ul>
</div>
</div>
</div>
</div>