<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>There is a real, functional difference between a function with an empty parameter list and one with an explicitly <code>void</code> parameter list: It is possible to pass parameters to a function with an empty list; the compiler won&#8217;t complain. That is not the case for a function with a <code>void</code> list. Thus, it is possible, and even easy to invoke empty-list functions incorrectly without knowing it, and thereby introduce the kind of subtle bug that can be very difficult to track down.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">void myfunc ();  // Noncompliant

//...

void otherFunc() {
  int a = 4;
  //...
  myfunc(a); // Compiler allows this
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">void myfunc ( void );

//...

void otherFunc() {
  int a = 4;
  //...
  myfunc(a); // Compiler error!
}</code></pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p>MISRA C:2004, 16.5 - Functions with no parameters shall be declared with parameter type void</p>
</li>
<li>
<p><a href="https://wiki.sei.cmu.edu/confluence/x/t9YxBQ">CERT, DCL20-C.</a> - Explicitly specify void when a function accepts no arguments</p>
</li>
</ul>
</div>
<hr>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Set this function&#8217;s parameter list to "void"</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_relates_to_s1270">relates to: <a data-rspec-id="S1270" class="rspec-auto-link">S1270</a></h3>

</div>
<div class="sect2">
<h3 id="_on_2_sep_2014_162032_ann_campbell_wrote">on 2 Sep 2014, 16:20:32 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>from [~evgeny.mandrikov]:</p>
</div>
<div class="paragraph">
<p>I don&#8217;t like current description, because this rule is not just about little styling. Compare to CERT, which for example contains</p>
</div>
<div class="quoteblock">
<blockquote>
<div class="paragraph">
<p>Failure to declare a void parameter will result in</p>
</div>
<div class="ulist">
<ul>
<li>
<p>An ambiguous functional interface between the caller and callee.</p>
</li>
<li>
<p>Sensitive information outflow.</p>
</li>
</ul>
</div>
</blockquote>
</div>
<div class="paragraph">
<p>Also as a developer I really appreciate references to the language specification in RSPECs.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_19_sep_2014_130835_freddy_mallet_wrote">on 19 Sep 2014, 13:08:35 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>@Ann, I would also associate this rule to the SQALE characteristic "Reliability". Indeed this rule can definitely be associated to "Maintainability" or to "Reliability". But when there is a doubt "Reliability" should be preferred because operational risks matter more than maintainability issue.</p>
</div>
</div>
</div>
</div>