<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Under the covers, Simple <code>CASE</code> expressions are evaluated as searched <code>CASE</code> expressions. That is,</p>
</div>
<div class="listingblock">
<div class="content">
<pre>CASE @foo
WHEN 1 THEN 'a'
WHEN 2 THEN 'b'</pre>
</div>
</div>
<div class="paragraph">
<p>is actually evaluated as</p>
</div>
<div class="listingblock">
<div class="content">
<pre>CASE
WHEN @foo = 1 THEN 'a'
WHEN @foo = 2 THEN 'b'</pre>
</div>
</div>
<div class="paragraph">
<p>In most situations the difference is inconsequential, but when the input expression isn&#8217;t fixed, for instance if <code>RAND()</code> is involved, it is likely to yield unexpected results. For that reason, it is better to evaluate the input expression once, assign it to a variable, and use the variable as the <code>CASE</code>'s input expression.</p>
</div>
<div class="paragraph">
<p>This rule raises an issue when any of the following is used in a <code>CASE</code> input expression: <code>RAND</code>, <code>NEWID</code>, <code>CRYPT_GEN_RANDOM</code>.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-sql" data-lang="sql">CASE CONVERT(SMALLINT, RAND()*@foo)  -- Noncompliant
WHEN 1 THEN 'a'
WHEN 2 THEN 'b'</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-sql" data-lang="sql">DECLARE @bar SMALLINT = CONVERT(SMALLINT, RAND()*@foo)
CASE @bar
WHEN 1 THEN 'a'
WHEN 2 THEN 'b'</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Assign this "CASE" input expression to a variable so that it&#8217;s executed only once, and evaluate the variable instead.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>the case input expression</p>
</div>
</div>
</div>
</div>