<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Waiting for explanation from BNP. This rule seems to be really specific to BNP</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="paragraph">
<p>TODO</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_26_may_2014_153543_freddy_mallet_wrote">on 26 May 2014, 15:35:43 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>Rule specific to the BNP context and which can be easily implemented with the XPath rule: //cancelStatement</p>
</div>
</div>
</div>
</div>