include::../description.adoc[]

== Noncompliant Code Example

----
public func numberOfMinutes(hours:Int) -> Int {
  var seconds = 0   // seconds is never used 
  return hours * 60;
}
----

== Compliant Solution

----
public func numberOfMinutes(hours:Int) -> Int{
  return hours * 60
}
----

== Exceptions

Simple <code>for-in</code> loop counters are ignored by this rule because while they are often legitimately unused, their declaration is required by the syntax.

----
for i in 1...10 {  // Ignored
  print("Hello! ");
}

for (a, b) in someElements {  // Noncompliant; b unused
  print(a)
}
----
