== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,swift]
----
public func numberOfMinutes(hours:Int) -> Int {
  var seconds = 0   // seconds is never used 
  return hours * 60;
}
----

=== Compliant solution

[source,swift]
----
public func numberOfMinutes(hours:Int) -> Int{
  return hours * 60
}
----

=== Exceptions

Simple ``++for-in++`` loop counters are ignored by this rule because while they are often legitimately unused, their declaration is required by the syntax.

----
for i in 1...10 {  // Ignored
  print("Hello! ");
}

for (a, b) in someElements {  // Noncompliant; b unused
  print(a)
}
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
