include::../description.adoc[]

== Noncompliant Code Example

----
func printTen() {                                                                                                                                
	myNumber := 010 // Noncompliant. myNumber will hold 8, not 10 - was this really expected?
	fmt.Println(myNumber)
}
----

== Compliant Solution

----
func printTen() {
	myNumber := 10
	fmt.Println(myNumber)
}
----

include::../see.adoc[]

ifdef::env-github,rspecator-view[]
'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
