== Why is this an issue?

Identifiers declared with https://www.ibm.com/support/knowledgecenter/en/SSY2V3_5.1.0/com.ibm.ent.pl1.zos.doc/lr/value.html[``++STATIC++``] and ``++INIT++`` are initialized only once and may be later modified. Modifying such an identifier will fail at runtime when a CICS application is https://www.ibm.com/docs/en/cics-ts/5.6?topic=applications-multithreading-reentrant-quasi-reentrant-threadsafe-programs[executed as reentrant]. It is safer to declare the value as https://www.ibm.com/support/knowledgecenter/en/SSY2V3_5.1.0/com.ibm.ent.pl1.zos.doc/lr/value.html[a named constant] (using the keyword ``++VALUE++``) because any modification will be reported at compilation time.


This rule raises an issue when an identifier is declared as ``++STATIC++`` with ``++INIT++`` or ``++INITIAL++`` and has one of the following types:

* CHARACTER/CHAR
* BINARY/BIN
* DECIMAL/DEC
* FLOAT
* POINTER/PTR
* OFFSET


=== Noncompliant code example

[source,pli]
----
dcl returnCodeOk fixed bin(15) static init (00); /* Noncompliant */
/* or */
dcl actions char(05) init('12345') static; /* Noncompliant */
dcl actions_in char(01);
....
valid_action = (verify(action_in, actions) = 0);
----


=== Compliant solution

[source,pli]
----
dcl returnCodeOk fixed bin(15) value (00);
/* or */
dcl actions char(05) value('12345');
dcl actions_in char(01);
....
valid_action = (verify(action_in, actions) = 0);
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Use "VALUE" instead of "INIT STATIC".


=== Highlighting

The declaration


endif::env-github,rspecator-view[]
