This rule checks that the specified child elements are present inside the specified parent elements. 


== Noncompliant Code Example

Given a parent/child combination of ``++<head>++``/``++<title>++``:

[source,html]
----
<html>
  <head>
  </head>  <!-- Noncompliant; no title element -->
  <body>
    ...
----


== Compliant Solution

[source,html]
----
<html>
  <head>
    <title>My Page</title>
  </head>
  <body>
    ...
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

include::parameters.adoc[]

endif::env-github,rspecator-view[]
