== Why is this an issue?

This rule checks that the specified child elements are present inside the specified parent elements. 


=== Noncompliant code example

Given a parent/child combination of ``++<head>++``/``++<title>++``:

[source,html]
----
<html>
  <head>
  </head>  <!-- Noncompliant; no title element -->
  <body>
    ...
----


=== Compliant solution

[source,html]
----
<html>
  <head>
    <title>My Page</title>
  </head>
  <body>
    ...
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Add the missing "xxx" element to this "yyy".


=== Parameters

.child
****

Name of the child element
****
.parent
****

Name of the parent element
****


endif::env-github,rspecator-view[]
