By convention, a file that exports only one class, function, or constant should be named for that class, function or constant. Anything else may confuse maintainers.

== Noncompliant Code Example

[source,javascript]
----
// file path: myclass.js  -- Noncompliant
class MyClass {
  // ...
}
export default MyClass;
----

== Compliant Solution

[source,javascript]
----
// file path: MyClass.js
class MyClass {
  // ...
}
export default MyClass;
----

== Exceptions

Case, dots ( ``++.++`` ), underscores ( ``++_++`` ) and dashes  (``++-++``) are ignored from the name comparison. Postfixes in filenames like ``++.dev++`` in ``++my.class.dev.js++`` are also ignored.

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
