== How to fix it in .NET

Microsoft's database connection libraries typically provide a connection string builder class. These classes provide
methods and properties that safely set parameter values.

Connection string builders will only protect you if you use these methods and properties to set parameter values. They
will not help if you are using them to modify a connection string where user-supplied values have already been added.

If no connection string builder is available, user-supplied values must either be validated to ensure that they're not
malicious, or must be properly quoted so that they cannot interfere with other connection string parameters.

=== Code examples

==== Noncompliant code example

[source,vbnet,diff-id=1,diff-type=noncompliant]
----
Public Property ConnectionString As String = "Server=10.0.0.101;Database=CustomerData"

Public Function ConnectToDatabase(request As HttpRequest) As SqlConnection
    Dim connectionString As String = String.Format("{0};User ID={1};Password={2}",
        Me.ConnectionString,
        request.Form("username"),
        request.Form("password"))

    Dim connection As New SqlConnection()
    connection.ConnectionString = connectionString ' Noncompliant
    connection.Open()
    Return connection
End Function
----

==== Compliant solution

[source,vbnet,diff-id=1,diff-type=compliant]
----
Public Property ConnectionString As String = "Server=10.0.0.101;Database=CustomerData"

Public Function ConnectToDatabase(request As HttpRequest) As SqlConnection
    Dim builder As New SqlConnectionStringBuilder(ConnectionString)
    builder.UserID = request.Form("username")
    builder.Password = request.Form("password")

    Dim connection As New SqlConnection()
    connection.ConnectionString = builder.ConnectionString
    connection.Open()
    Return connection
End Function
----

=== How does this work?

Connection string builders will ensure that values are correctly sanitized when adding them to the connection string.
