include::../summary.adoc[]

== Why is this an issue?

include::../description.adoc[]

=== Code examples

==== Noncompliant code example

[source,python,diff-id=1,diff-type=noncompliant]
----
class Foo: 
  foo = '' 

  def getFoo(self):
    ...

foo = Foo()
foo.getFoo() # what does this return?
----

==== Compliant solution

[source,python,diff-id=1,diff-type=compliant]
----
class Foo:
  name = ''

  def getName(self):
    ...

foo = Foo()
foo.getName()
----
ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

endif::env-github,rspecator-view[]
