include::../description.adoc[]

== Noncompliant Code Example

[source,python]
----
class Foo: 
  foo = '' 

  def getFoo(self):
    ...

foo = Foo()
foo.getFoo() # what does this return?
----

== Compliant Solution

[source,python]
----
class Foo:
  name = ''

  def getName(self):
    ...

foo = Foo()
foo.getName()
----
ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

endif::env-github,rspecator-view[]
