== Why is this an issue?

include::../description.adoc[]

=== NonCompliant code example

[source,java]
----
"a[b]c"
"[\\^]"
----

=== Compliant solution

[source,java]
----
"abc"
"\\^"
"a[*]c" // Compliant, see Exceptions
----

include::../exceptions.adoc[]

include::../implementation.adoc[]
