include::../description.adoc[]

== Noncompliant Code Example

[source,java]
----
"a[b]c"
----

== Compliant Solution

[source,java]
----
"abc"
----

include::../exceptions.adoc[]

include::../implementation.adoc[]
