<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Declaring a function <code>constexpr</code> makes it implicitly inline.</p>
</div>
<div class="paragraph">
<p>In that situation, explicitly using the <code>inline</code> keyword would be redundant, and might lead to confusion if it&#8217;s used in some cases but not others. It&#8217;s better to simply omit it.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">inline constexpr int addOne(int n) { return n+1; } // Noncompliant</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">constexpr int addOne(int n) { return n+1; }</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove this redundant 'inline' specifier.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>The <code>inline</code> specifier</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_27_aug_2019_104528_lo誰c_joly_wrote">on 27 Aug 2019, 10:45:28 Lo誰c Joly wrote:</h3>
<div class="paragraph">
<p>\[~abbas.sabra]: Can you please review my changes?</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_27_aug_2019_171408_abbas_sabra_wrote">on 27 Aug 2019, 17:14:08 Abbas Sabra wrote:</h3>
<div class="paragraph">
<p>Looks good to me</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_30_aug_2019_182402_nicolas_harraudeau_wrote">on 30 Aug 2019, 18:24:02 Nicolas Harraudeau wrote:</h3>
<div class="paragraph">
<p>\[~loic.joly] Missing "See" section? This is in the C&#43;&#43; core guidelines MMF so I guess it would link to it?</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_30_aug_2019_222706_lo誰c_joly_wrote">on 30 Aug 2019, 22:27:06 Lo誰c Joly wrote:</h3>
<div class="paragraph">
<p>\[~nicolas.harraudeau] What triggered the idea for this rule is: F.5: If a function is very small and time-critical, declare it inline</p>
</div>
<div class="paragraph">
<p>But since this point is just one line in the core guideline, and a small detail, I don&#8217;t think it&#8217;s worth mentioning&#8230;&#8203;</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_2_sep_2019_150219_nicolas_harraudeau_wrote">on 2 Sep 2019, 15:02:19 Nicolas Harraudeau wrote:</h3>
<div class="paragraph">
<p>\[~loic.joly] Ok it makes sense</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_9_sep_2019_204451_ann_campbell_wrote">on 9 Sep 2019, 20:44:51 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~abbas.sabra] I think you&#8217;re looking for the 'redundant' tag here instead of 'clumsy'. :-D</p>
</div>
</div>
</div>
</div>