<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>When a method is designed to return an invariant value, it may be poor design, but it shouldn&#8217;t adversely affect the outcome of your program. However, when it happens on all paths through the logic, it is surely a bug.</p>
</div>
<div class="paragraph">
<p>This rule raises an issue when a method contains several <code>return</code> statements that all return the same value.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-text" data-lang="text">int foo(int a) {
  int b = 12;
  if (a == 1) {
    return b;
  }
  return b;  // Noncompliant
}</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Refactor this method to not always return the same value.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="ulist">
<ul>
<li>
<p>Primary: method name</p>
</li>
<li>
<p>Secondary: each return statement value</p>
<div class="ulist">
<ul>
<li>
<p>message: "Returned value."</p>
</li>
</ul>
</div>
</li>
</ul>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_relates_to_s3400">relates to: <a data-rspec-id="S3400" class="rspec-auto-link">S3400</a></h3>

</div>
<div class="sect2">
<h3 id="_on_26_mar_2024_081500_philipp_dominik_schubert_wrote">on 26 Mar 2024, 08:15:00 Philipp Dominik Schubert wrote:</h3>
<div class="paragraph">
<p>We created a circle to discuss CFamily&#8217;s rules on dead code and decided to disable this rule in the CFamily analyzer since we believe it provides relatively low value and is not even implemented in CFamily&#8217;s analyzer (cf. <a href="https://sonarsource.atlassian.net/browse/CPP-5165" class="bare">https://sonarsource.atlassian.net/browse/CPP-5165</a>).</p>
</div>
</div>
</div>
</div>