<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Instances of classes that do not derive from the "Throwable" interface cannot be used in a PHP "throw" statement.</p>
</div>
<div class="paragraph">
<p>Many built-in exceptions such as "Exception" and the SPL exception classes do implement the "Throwable" interface and can be extended when creating custom exceptions.</p>
</div>
<div class="paragraph">
<p>This rule raises an issue when an instance of a class that does not implement the "Throwable" interface is used in a "throw" statement .</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-php" data-lang="php">class NoThrowable {}

throw new NoThrowable(); // Noncompliant</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-php" data-lang="php">&lt;?php

class SomeThrowable implements Throwable {
    // Implementation of the Throwable methods
}

throw new SomeThrowable(); // Compliant

class SomeCustomException extends Exception {}

throw new SomeCustomException(); // Compliant{code}</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Throw an object derived from "Throwable".</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="ulist">
<ul>
<li>
<p>Primary: The whole "throw" statement</p>
</li>
<li>
<p>Secondary: The name of the class in its class definition.</p>
<div class="ulist">
<ul>
<li>
<p>message: 'Class definition.'</p>
</li>
</ul>
</div>
</li>
</ul>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_relates_to_s3696">relates to: <a data-rspec-id="S3696" class="rspec-auto-link">S3696</a></h3>

</div>
<div class="sect2">
<h3 id="_is_related_to_s5708">is related to: <a data-rspec-id="S5708" class="rspec-auto-link">S5708</a></h3>

</div>
<div class="sect2">
<h3 id="_on_13_nov_2019_162349_nicolas_harraudeau_wrote">on 13 Nov 2019, 16:23:49 Nicolas Harraudeau wrote:</h3>
<div class="paragraph">
<p>Note: This rule is similar to <a data-rspec-id="S3696" class="rspec-auto-link">RSPEC-3696</a> "Non-exception types should not be thrown" but this one is a Bug instead of a Code Smell because raising a non-exception in Python will raise a TypeError.</p>
</div>
</div>
</div>
</div>