== Why is this an issue?

This rule checks that the specified HTML elements are not present.


=== Noncompliant code example

For a disallowed list of "font,center":

[source,html]
----
<center><font color="red">Hello World!</font></center> <!-- Noncompliant -->
----


=== Compliant solution

[source,html]
----
<div class="centerRed">Hello World!</div>
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Remove this "XXX" element.


=== Parameters

.elements
****

Comma-separated list of names of forbidden elements
****


endif::env-github,rspecator-view[]
