This rule checks that the specified HTML elements are not present.


== Noncompliant Code Example

For a disallowed list of "font,center":

----
<center><font color="red">Hello World!</font></center> <!-- Noncompliant -->
----


== Compliant Solution

----
<div class="centerRed">Hello World!</div>
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

include::parameters.adoc[]

endif::env-github,rspecator-view[]
