include::../description.adoc[]

== Noncompliant Code Example

----
function foo(n, m) {
  switch (n) { 
    case 0:
      switch (m) {  // Noncompliant; nested switch
        // ...
      }
    case 1:
      // ...
    default:
      // ...
  }
}
----

== Compliant Solution

----
function foo(n, m) {
  switch (n) {
    case 0:
      bar(m);      
    case 1:
      // ...
    default:
      // ...
  }
}

function bar(m) {
  switch(m) {
    // ...
  }
}
----
