include::../description.adoc[]

== Noncompliant Code Example

[source,javascript]
----
function foo(n, m) {
  switch (n) { 
    case 0:
      switch (m) {  // Noncompliant; nested switch
        // ...
      }
    case 1:
      // ...
    default:
      // ...
  }
}
----

== Compliant Solution

[source,javascript]
----
function foo(n, m) {
  switch (n) {
    case 0:
      bar(m);      
    case 1:
      // ...
    default:
      // ...
  }
}

function bar(m) {
  switch(m) {
    // ...
  }
}
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
