== How to fix it in Java SE

=== Code examples

include::../../common/fix/code-rationale.adoc[]

==== Noncompliant code example

[source,java,diff-id=11,diff-type=noncompliant]
----
try {
  /* ... */
} catch(Exception e) {
  e.printStackTrace(); // Noncompliant 
}
----

==== Compliant solution

[source,java,diff-id=11,diff-type=compliant]
----
try {
  /* ... */
} catch(Exception e) {
  LOGGER.log("context", e);
}
----

=== How does this work?

``++Throwable.printStackTrace(...)++`` prints a ``++Throwable++`` and its stack
trace to ``++System.Err++`` which is not easily parseable and can expose
sensitive information. Loggers should be used instead to print or store stack
traces.
