== Why is this an issue?

Shared coding conventions allow teams to collaborate efficiently. This rule checks that all local, ``++final++``, initialized, primitive variables, have names that match a provided regular expression.


=== Noncompliant code example

With the default regular expression ``++^[A-Z][A-Z0-9]*(_[A-Z0-9]+)*$++``:

[source,java]
----
public void doSomething() { 
  final int local = 42;    
  ...             
}
----


=== Compliant solution

[source,java]
----
public void doSomething() { 
  final int LOCAL = 42;  
  ...               
}
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

include::parameters.adoc[]

endif::env-github,rspecator-view[]
