Whenever a user clicks on a link that targets an image, the website's navigation menu will be lost.
From a user point of view, it is as if she left the website.
The only way to return to it is using the browser's 'Back' button.
Instead, it is better to create a page which will display the image using the <code><img></code> tag and preserve the navigation menu.

Further, in terms of accessibility, when the image is embedded into a page, content providers are able to provide an alternate text equivalent through the <code>alt</code> attribute.


== Noncompliant Code Example

----
<a href="image.png">...</a>  <!-- Non-Compliant -->
----


== Compliant Solution

----
<a href="page.html">...</a>  <!-- Compliant -->
----

