Variables and fields that are not initialized inline should be explicitly typed, even when the type is correctly inferred at the first initialization, and the variable doesn't end up being `dynamic`.

== Why is this an issue?

Dart allows to declare variables and fields without specifying their type, even when it is not initialized inline. In that case, the type is inferred from the first initialization. 

However, the first initialization can be far from the declaration, making the type of the variable not obvious, just by looking at its declaration. 

[source,dart]
----
void foo() {
    var path; // String or Path?
    // ...
    path = defaultPath;
    // ...
}
----

Moreover, the first initialization expression can be complex, making it hard to understand what the type of the variable is even when the initialization is close enough to the declaration.

[source,dart]
----
void foo() {
    var path; // String or Path?
    // ...
    path = buildPath();
    // ...
}
----

In addition, there are scenarios where type inference is not advisable, like when the desired type is not the narrowest matching initialization:

[source,dart]
----
var x; // x is inferred as an int, becase of its first initialization 
// ...
x = 3;
// ...
x = 4.0; // Error: A value of type 'double' can't be assigned to a variable of type 'int'
----

To avoid confusion and make the intent clear, it's recommended to always specify the type of initialized variables and fields explicitly.

== How to fix it

=== Code examples

==== Noncompliant code example

[source,dart,diff-id=1,diff-type=noncompliant]
----
void foo() {
    var path;
    // ...
}
----

==== Compliant solution

[source,dart,diff-id=1,diff-type=compliant]
----
void foo() {
    Path path;
    // ...
}
----

== Resources

=== Documentation

* Dart Docs - https://dart.dev/tools/linter-rules/prefer_typing_uninitialized_variables[Dart Linter rule - prefer_typing_uninitialized_variables]
* Dart Docs - https://dart.dev/language/type-system#type-inference[Language - Type Inference]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

* An uninitialized variable should have an explicit type annotation.

=== Highlighting

The identifier of the uninitialized variable, in its declaration.

'''
== Comments And Links
(visible only on this page)

endif::env-github,rspecator-view[]
