include::../description.adoc[]

== Noncompliant Code Example

----
if ($x < 0) {
  new foo;  // Noncompliant
}
----

== Compliant Solution

----
$var = NULL;
if ($x < 0) {
  $var = new foo;
}
----

ifdef::env-github,rspecator-view[]
'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
