<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when the <code>validate_indices</code> argument is set for `tf.gather ` function call.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>The <code>tf.gather</code> function allows you to gather slices from a tensor along a specified axis according to the indices provided.
The <code>validate_indices</code> argument is deprecated and setting its value has no effect. Indices are always validated on CPU and never validated on GPU.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">import tensorflow as tf

x = tf.constant([[1, 2], [3, 4]])
y = tf.gather(x, [1], validate_indices=True)  # Noncompliant: validate_indices is deprecated</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">import tensorflow as tf

x = tf.constant([[1, 2], [3, 4]])
y = tf.gather(x, [1])  # OK</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Tensorflow documentation - <a href="https://www.tensorflow.org/api_docs/python/tf/gather">tf.gather</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>