<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Duplicated string literals make the process of refactoring complex and error-prone, as any change would need to be propagated on all occurrences.</p>
</div>
<div class="sect2">
<h3 id="_exceptions">Exceptions</h3>
<div class="paragraph">
<p>Literals with fewer than 7 characters are ignored.</p>
</div>
<div class="paragraph">
<p>Only duplications located in a <code>PROCEDURE DIVISION</code>, not those contained in copybooks are reported.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Use constants to replace the duplicated string literals.
Constants can be referenced from many places, but only need to be updated in a single place.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="paragraph">
<p>With the default threshold of 3:</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cobol" data-lang="cobol">       PROCEDURE DIVISION.

       DISPLAY "Firstname: ".
      *...
       DISPLAY "Firstname: ".
      *...
       DISPLAY "Firstname: ".</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cobol" data-lang="cobol">       WORKING-STORAGE SECTION.
         01 FIRST-NAME-HEADER PIC X(42) VALUE "Firstname: ".
       PROCEDURE DIVISION.

       DISPLAY FIRST-NAME-HEADER
      *...
       DISPLAY FIRST-NAME-HEADER
      *...
       DISPLAY FIRST-NAME-HEADER</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Define a constant instead of duplicating this literal "{string}" {number} times.</p>
</div>
</div>
<div class="sect2">
<h3 id="_parameters">Parameters</h3>
<div class="sidebarblock">
<div class="content">
<div class="title">threshold</div>
<div class="listingblock">
<div class="content">
<pre>3</pre>
</div>
</div>
<div class="paragraph">
<p>Number of times a literal must be duplicated to trigger an issue</p>
</div>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="ulist">
<ul>
<li>
<p>Primary: The first occurrence</p>
</li>
<li>
<p>Secondary: the following occurrences</p>
<div class="ulist">
<ul>
<li>
<p>message: 'Duplication.'</p>
</li>
</ul>
</div>
</li>
</ul>
</div>
<hr>
</div>
</div>
</div>