<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Duplicated string literals make the process of refactoring complex and error-prone, as any change would need to be propagated on all occurrences.</p>
</div>
<div class="sect2">
<h3 id="_exceptions">Exceptions</h3>
<div class="paragraph">
<p>No issue will be raised on:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>strings with less than 5 characters</p>
</li>
<li>
<p>strings with only letters, numbers, underscores, hyphens and periods</p>
</li>
<li>
<p>Laravel validation strings (e.g., <code>"required"</code>, <code>"max:100"</code>, <code>"email:rfc"</code>)</p>
</li>
<li>
<p>Drupal form keys (e.g., <code>"#type"</code>, <code>"#title"</code>)</p>
</li>
</ul>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Use constants to replace the duplicated string literals.
Constants can be referenced from many places, but only need to be updated in a single place.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="paragraph">
<p>With the default threshold of 3:</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-php" data-lang="php">function run() {
  prepare('this is a duplicate'); // Noncompliant - 'this is a duplicate' is duplicated 3 times
  execute('this is a duplicate');
  release('this is a duplicate');
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-php" data-lang="php">MESSAGE = 'this is a duplicate';

function run() {
  prepare(MESSAGE); // Compliant - the duplicated string literal is replaced by a constant and can be safely re-used
  execute(MESSAGE);
  release(MESSAGE);
}</code></pre>
</div>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-php" data-lang="php">$severity = $request-&gt;getParam('severity-score');</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Define a constant instead of duplicating this literal "{string}" {number} times.</p>
</div>
</div>
<div class="sect2">
<h3 id="_parameters">Parameters</h3>
<div class="sidebarblock">
<div class="content">
<div class="title">threshold</div>
<div class="listingblock">
<div class="content">
<pre>3</pre>
</div>
</div>
<div class="paragraph">
<p>Number of times a literal must be duplicated to trigger an issue</p>
</div>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="ulist">
<ul>
<li>
<p>Primary: The first occurrence</p>
</li>
<li>
<p>Secondary: the following occurrences</p>
<div class="ulist">
<ul>
<li>
<p>message: 'Duplication.'</p>
</li>
</ul>
</div>
</li>
</ul>
</div>
<hr>
</div>
</div>
</div>