include::../description.adoc[]


== Noncompliant Code Example

[source,javascript]
----
"cc̈d̈d".replace(/[c̈d̈]/g, "X"); // result is "XXXXXX" and not expected "cXXd"
----


== Compliant Solution

[source,javascript]
----
"cc̈d̈d".replace(/c̈|d̈/g, "X"); // result is "cXXd"
----
