== Why is this an issue?

include::../description.adoc[]


=== Noncompliant code example

[source,javascript]
----
"cc̈d̈d".replace(/[c̈d̈]/g, "X"); // result is "XXXXXX" and not expected "cXXd"
----


=== Compliant solution

[source,javascript]
----
"cc̈d̈d".replace(/c̈|d̈/g, "X"); // result is "cXXd"
----
