<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>When constructing an <code>std::scoped_lock</code>, the constructor arguments are used to list the mutexes that the <code>scoped_lock</code> will lock on the creation and unlock on destruction. It is possible to construct a <code>scoped_lock</code> without any parameter, but in that case, it does absolutely nothing and is just dead code, which was probably not the intent of the user.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">void f1(std::mutex &amp;m) {
  std::scoped_lock lock; // Noncompliant
  // Do some work
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">void f1(std::mutex &amp;m) {
  std::scoped_lock lock {m}; // Compliant
  // Do some work
}

template&lt;class... D&gt;
void processAll(D &amp;...data) {
  scoped_lock lock {data.getMutex()...}; // Compliant, even if the list might be empty in some cases
  // Do some work
}</code></pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p>C&#43;&#43; reference - <a href="https://en.cppreference.com/w/cpp/thread/scoped_lock">std::scoped_lock</a></p>
</li>
</ul>
</div>
<hr>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_relates_to_s5997">relates to: <a data-rspec-id="S5997" class="rspec-auto-link">S5997</a></h3>

</div>
</div>
</div>