This is an issue when for loops use parentheses around components, omit braces around the loop body, or use non-idiomatic patterns for while-like behavior.

== Why is this an issue?

Go has specific syntax requirements for for loops that differ from other programming languages like C, Java, or JavaScript.

Using incorrect syntax can lead to compilation errors or code that doesn't follow Go conventions. Go's for loop syntax is designed to be clean and consistent:

* No parentheses are needed around the three components (init, condition, post)
* Braces are always required around the loop body, even for single statements
* For while-like behavior, Go provides a clean syntax without unnecessary semicolons

When developers come from other languages, they might unconsciously apply familiar syntax patterns that don't work in Go or make the code less idiomatic.

=== What is the potential impact?

Using incorrect for loop syntax will cause compilation errors, preventing the code from running. Even when the code compiles, non-idiomatic patterns make it harder for other Go developers to read and maintain the code.

== How to fix it

Remove parentheses around the for loop components. Go does not require or allow parentheses around the init, condition, and post statements.

=== Code examples

==== Noncompliant code example

[source,go,diff-id=1,diff-type=noncompliant]
----
for (i := 0; i < 10; i++) { // Noncompliant
    fmt.Println(i)
}
----

==== Compliant solution

[source,go,diff-id=1,diff-type=compliant]
----
for i := 0; i < 10; i++ {
    fmt.Println(i)
}
----

== Resources

=== Documentation

 * A Tour of Go - For - https://go.dev/tour/flowcontrol/1[Official Go tutorial explaining for loop syntax and requirements]

 * A Tour of Go - For is Go's "while" - https://go.dev/tour/flowcontrol/3[Official explanation of how Go uses for loops for while-like behavior]
