<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Two classes can have the same simple name if they are in two different packages.</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">package org.foo.domain;

public class User {
  // ..
}</code></pre>
</div>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">package org.foo.presentation;

public class User {
  // ..
}</code></pre>
</div>
</div>
<div class="paragraph">
<p>However, this becomes an issue when a class has the same name as the superclass it extends or the interfaces it
implements, also known as class name shadowing. It is problematic because it can be unclear which class is being
referred to in the code, leading to ambiguity and potential bugs.</p>
</div>
<div class="paragraph">
<p>Therefore, it is recommended to use unique and descriptive class names that do not conflict with the names of the
superclass or interfaces.</p>
</div>
<div class="paragraph">
<p>This rule raises an issue when a class name shadows its superclass or interface names.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Rename the class using a more descriptive name.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">package org.foo.presentation;

public class User implements org.foo.domain.User { // Noncompliant
  // ...
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">package org.foo.presentation;

import org.foo.domain.User;

public class UserView implements User { // Compliant
  // ...
}</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p><a href="https://docs.oracle.com/javase/specs/jls/se17/html/jls-6.html#jls-6.4.1">Java SE Specifications &gt; Java Language Specification &gt; 6.4.1. Shadowing</a></p>
</li>
</ul>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Rename this class.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_21_nov_2024_164800_alban_auzeill_wrote">on 21 Nov 2024, 16:48:00 Alban Auzeill wrote:</h3>
<div class="paragraph">
<p>[test-code-support-investigation-for-java] Decision for scope: Keep 'Main'. Not sure about the value/impact in test.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_21_oct_2014_140846_nicolas_peru_wrote">on 21 Oct 2014, 14:08:46 Nicolas Peru wrote:</h3>
<div class="paragraph">
<p>LGTM</p>
</div>
</div>
</div>
</div>