Jump statements (``++return++``, ``++break++`` and ``++continue++``) move control flow out of the current code block. So any statements that come after a jump are dead code. 

== Noncompliant Code Example

----
fun foo(a: Int): Int {
  var i = 10;
  return a + i;       // Noncompliant 
  i++;                // dead code
}
----

== Compliant Solution

----
fun foo(a: Int): Int {
  var i = 10;
  return a + i;
}
----

include::../see.adoc[]

ifdef::env-github,rspecator-view[]
'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
