== How to fix it in PointyCastle

=== Code examples

include::../../common/fix/code-rationale.adoc[]

==== Noncompliant code example

[source,dart,diff-id=1,diff-type=noncompliant]
----
import 'dart:typed_data';
import 'package:pointycastle/export.dart';

Uint8List encrypt(Uint8List keyBytes, Uint8List ivBytes, Uint8List plainText) {
  final key = KeyParameter(keyBytes);
  final params = ParametersWithIV(key, ivBytes);

  final cipher = BlockCipher('DES'); // Noncompliant

  cipher.init(true, BlockCipherParameters(params, null));
  final encryptedBytes = cipher.process(plainText);

  return encryptedBytes;
}
----

==== Compliant solution

[source,dart,diff-id=1,diff-type=compliant]
----
import 'dart:typed_data';
import 'package:pointycastle/export.dart';

Uint8List encrypt(Uint8List keyBytes, Uint8List ivBytes, Uint8List plainText) {
  final key = KeyParameter(keyBytes);
  final params = ParametersWithIV(key, ivBytes);

  final cipher = GCMBlockCipher(AESEngine());
  final params = AEADParameters(key, 128, ivBytes, Uint8List(0));

  cipher.init(true, params);
  final encryptedBytes = cipher.process(plainText);

  return encryptedBytes;
}
----

=== How does this work?

include::../../common/fix/strong-cryptography.adoc[]

