<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Every section should be commented to explain its goal and how it works. This comment can be placed either just before or just after the section label.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cobol" data-lang="cobol">    UNCOMMENTED-SECTION SECTION.</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cobol" data-lang="cobol">* Some comments
    CORRECTLY-COMMENTED-SECTION SECTION.

    ANOTHER-CORRECTLY-COMMENTED-SECTION SECTION.
* Some comments</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Document this section by adding a comment either before or after the section label.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_relates_to_s1304">relates to: <a data-rspec-id="S1304" class="rspec-auto-link">S1304</a></h3>

</div>
<div class="sect2">
<h3 id="_on_9_oct_2013_001119_ann_campbell_wrote">on 9 Oct 2013, 00:11:19 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>The Noncompliant example for this one embeds the compliant examples. Is this sufficient, or should they be split out? Should I be adding a compliant copy of the example?</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_10_oct_2013_085116_freddy_mallet_wrote">on 10 Oct 2013, 08:51:16 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>I&#8217;ve updated the code the description [~ann.campbell.2]</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_11_nov_2013_125503_dinesh_bolkensteyn_wrote">on 11 Nov 2013, 12:55:03 Dinesh Bolkensteyn wrote:</h3>
<div class="paragraph">
<p>I didn&#8217;t get the double negation part of the title, "Sections should not be left undocumented".</p>
</div>
<div class="paragraph">
<p>I think "Sections should be documented" is much better.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_13_nov_2013_120627_ann_campbell_wrote">on 13 Nov 2013, 12:06:27 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>Agreed, [~dinesh.bolkensteyn]</p>
</div>
</div>
</div>
</div>