This rule raises an issue when a closure is used inside a GString expression in a Jenkins Pipeline script.

== Why is this an issue?

Jenkins Pipeline uses a Continuation Passing Style (CPS) transformation to allow scripts to be paused and resumed. This transformation affects how closures work in the Pipeline environment.

In regular Groovy, you can use closures inside GString expressions to create dynamic strings where the closure is evaluated each time the GString is converted to a String. However, in Jenkins Pipeline scripts, this pattern doesn't work as expected because the closure inside the GString gets CPS-transformed.

When the Pipeline runtime tries to evaluate the GString, it encounters a CPS-transformed closure in a non-CPS context, which causes the CPS interpreter to fail. This results in incorrect behavior and confusing error messages in the Pipeline logs.

The typical error you'll see in the logs looks like:

> expected to call WorkflowScript.echo but wound up catching org.jenkinsci.plugins.workflow.cps.CpsClosure2.call

This happens because the Pipeline runtime cannot properly handle the interaction between the non-CPS GString evaluation and the CPS-transformed closure.

=== What is the potential impact?

The code will not behave as expected and may produce runtime errors or unexpected results. The Pipeline may fail with confusing error messages, making debugging difficult. In some cases, the GString may not evaluate to the expected value, leading to incorrect string interpolation.

== How to fix it

Replace the GString containing a closure with a closure that returns a GString using normal expressions. Call the closure when you need the string value.

=== Code examples

==== Noncompliant code example

[source,groovy,diff-id=1,diff-type=noncompliant]
----
def x = 1
def s = "x = ${-> x}" // Noncompliant
x = 2
echo(s)
----

==== Compliant solution

[source,groovy,diff-id=1,diff-type=compliant]
----
def x = 1
def s = { -> "x = ${x}" }
x = 2
echo(s())
----

== Resources

=== Documentation

 * Jenkins Pipeline CPS Method Mismatches - https://www.jenkins.io/doc/book/pipeline/cps-method-mismatches/[Official Jenkins documentation explaining CPS transformation issues and solutions]

 * Pipeline: Groovy Plugin - https://plugins.jenkins.io/workflow-cps/[Technical background on the Groovy CPS library used by Jenkins Pipeline]
