<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Subprocedures and subroutines are both mechanisms to segregate logic, but subprocedures are preferred for three reasons:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>their local files and variables make maintenance faster and cleaner. They allow you to create variables without worrying about name clashes, and to change fields without worrying about negatively impacting other parts of the program.</p>
</li>
<li>
<p>their local files and variables make code reuse easy.</p>
</li>
<li>
<p>they can be called with parameters as functions, yielding clearer more readable code.</p>
</li>
</ul>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-rpg" data-lang="rpg">     D  FirstName      S             20A
     D  Initial        S              1A
     D  LastName       S             20A
     D  FullName       S             43A

      /free
       FirstName = 'John';
       Initial = 'A';
       LastName = 'Smith';
       EXSR SPFullName;
       DSPLY FullName;
       ...

       begsr SPFullName;
         FullName = FirstName + ' ' + Initial + ' ' + LastName;
       endsr;
      /end-free</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-rpg" data-lang="rpg">      /free
       DSPLY FullName('John':'A':'Smith');
       ...
      /end-free

     P FullName        B
     D FullName        PI            43A
     D  FirstName                    20A   Const
     D  Initial                       1A   Const
     D  LastName                     20A   Const
      /free
       return FirstName + ' ' + Initial + ' ' + LastName;
      /end-Free
     P                 E</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Convert this subroutine into a subprocedure</p>
</div>
</div>
</div>
</div>