<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Labeled blocks are useful, especially when the code is badly indented, to help maintainers match the beginning and ending of each block. When blocks are nested, labeling them can improve the code&#8217;s readability. This rule detects nested block which do not have a start label.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-sql" data-lang="sql">BEGIN -- Compliant, this is not a nested block
  NULL;
END;
/

BEGIN
  BEGIN -- Noncompliant; this nested block has no label
    NULL;
  END;
END;
/

BEGIN
  BEGIN -- Noncompliant; this nested block has only an end label
    NULL;
  END myBlockLabel1;

  &lt;&lt;myBlockLabel2&gt;&gt; -- Compliant
  BEGIN
    NULL;
  END;
END;
/</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-sql" data-lang="sql">BEGIN
  NULL;
END;
/

BEGIN
  BEGIN myBlockLabel0
    NULL;
  END myBlockLabel0;
END;
/

BEGIN
  BEGIN myBlockLabel1
    NULL;
  END myBlockLabel1;

  &lt;&lt;myBlockLabel2&gt;&gt;
  BEGIN
    NULL;
  END;
END;
/</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Add a label to this nested block.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_28_jan_2015_204343_ann_campbell_wrote">on 28 Jan 2015, 20:43:43 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~dinesh.bolkensteyn] this rule description was copied from Nemo &amp; I don&#8217;t understand why <code>&lt;&lt;myBlockLabel2&gt;&gt;</code> is labeled noncompliant. Can you verify for me that it is, please?</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_11_may_2015_114847_dinesh_bolkensteyn_wrote">on 11 May 2015, 11:48:47 Dinesh Bolkensteyn wrote:</h3>
<div class="paragraph">
<p>You&#8217;re right [~ann.campbell.2] - it actually is compliant. I&#8217;ve updated the description.</p>
</div>
<div class="paragraph">
<p>Note that there is another rule that verifies that, whenever there is a starting label, there also is a corresponding ending one.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_11_may_2015_135055_ann_campbell_wrote">on 11 May 2015, 13:50:55 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>corresponding Compliant Solution added</p>
</div>
</div>
</div>
</div>