=== Noncompliant code example

[source,text]
----
int pow(int num, int exponent) {
  if (exponent > 1) {
    num = num * pow(num, exponent-1);  // Noncompliant; direct recursion
  }
  return num;
}
----
