include::../description.adoc[]

== Noncompliant Code Example

----
IF something
  IF something_else
    -- ...
----

== Compliant Solution

----
IF something AND something_else
  -- ...
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../highlighting.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
