== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

With the default threshold of 3:

[source,apex]
----
switch on i { // Compliant - depth = 1
  when 1 {
    /* ... */
    for (Integer i = 0, j = 0; i < 10; i++) { // Compliant - depth = 2
      /* ... */
      Integer j = i + 1;
      if(j == 0){// Compliant - depth = 3, not exceeding the limit
        /* ... */
        while(j < 10){ // Noncompliant - depth = 4
          /* ... */
          if(j == 1){ // Depth = 5, exceeding the limit, but issues are only reported on depth = 4
            /* ... */
          }
        }
      }
    }
  }
}
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../parameters.adoc[]

include::../highlighting.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
