== Why is this an issue?

include::../description.adoc[]

== How to fix it

=== Code examples

The following example demonstrates the behavior of the rule with the default threshold of 3 levels of nesting:

==== Noncompliant code example

[source,apex]
----
if (i == 1) // Compliant - depth = 1
  /* ... */
  for (Integer j = 0; j < 10; j++) { // Compliant - depth = 2
    /* ... */
    Integer k = j + 1;
    if (k == 3){ // Compliant - depth = 3, not exceeding the limit
      /* ... */
      while (k < 10){ // Noncompliant - depth = 4
        /* ... */
        if (k == 1){ // Depth = 5, exceeding the limit, but issues are only reported on depth = 4
          /* ... */
        }
      }
    }
  }
}
----

==== Compliant solution

[source,apex]
----
if (i != 1) {
  return;
}
for (Integer j = 0; j < 10; j++) { // Compliant - depth = 1
  /* ... */
  Integer k = j + 1;
  if (k != 3) {
    continue;
  }
  /* ... */
  while (k < 10) { // Compliant - depth = 2
    /* ... */
    if (k == 1) { // Compliant - depth = 3
      /* ... */
    }
  }
}
----

include::../resources.adoc[]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../parameters.adoc[]

include::../highlighting.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
