<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>It can be expensive to instantiate a new object, and doing so inside a loop is typically an error. Instead, create the object once, before the loop.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-flex" data-lang="flex">for (var i:int = 0; i &lt; 10; i++) {
  var temp:MyObj = new MyObject();  // Noncompliant
  //...
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-flex" data-lang="flex">var temp:MyObj = new MyObject();
for (var i:int = 0; i &lt; 10; i++) {
  //...
}</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Move the instantiation of this "xxx" outside the loop.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_7_apr_2015_232431_evgeny_mandrikov_wrote">on 7 Apr 2015, 23:24:31 Evgeny Mandrikov wrote:</h3>
<div class="paragraph">
<p>I&#8217;m wondering if this rule is really relevant for C# , Java , C/C&#43;&#43;?</p>
</div>
<div class="paragraph">
<p>\[~dinesh.bolkensteyn], [~nicolas.peru], [~samuel.mercier], [~massimo.paladin], [~pierre-yves.nicolas], [~linda.martin] any thoughts?</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_8_apr_2015_051457_dinesh_bolkensteyn_wrote">on 8 Apr 2015, 05:14:57 Dinesh Bolkensteyn wrote:</h3>
<div class="paragraph">
<p>I don&#8217;t think it is [~evgeny.mandrikov]</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_9_apr_2015_130035_nicolas_peru_wrote">on 9 Apr 2015, 13:00:35 Nicolas Peru wrote:</h3>
<div class="paragraph">
<p>Neither for java. this one is a deprecation of a pmd rule which was already controversial : http://stackoverflow.com/questions/17340421/pmd-avoid-instantiating-new-objects-inside-loops</p>
</div>
<div class="paragraph">
<p>So, [~evgeny.mandrikov] I agree with [~dinesh.bolkensteyn] I don&#8217;t think it is worth to implement in Java.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_29_mar_2016_234820_evgeny_mandrikov_wrote">on 29 Mar 2016, 23:48:20 Evgeny Mandrikov wrote:</h3>
<div class="paragraph">
<p>This case is mentioned in "Google C&#43;&#43; Style Guide", so CPP-1448.</p>
</div>
</div>
</div>
</div>