<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when a computation is performed on Python arrays instead of using a NumPy&#8217;s broadcast operations.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Computations over NumPy arrays can be executed more efficiently than computation over Python&#8217;s built-in sequences.
These efficiencies are possible thanks to the nature of the NumPy arrays (fixed size and homogeneous)
and the fact that operations on these arrays that can be done with compiled code.</p>
</div>
<div class="paragraph">
<p>When executing an operation on each elements of an array it is preferable to use NumPy broadcasting capabilities in order to benefit from the performance boost.
NumPy broadcasting operations provide also better readability and maintainability.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>To fix this issue try using NumPy&#8217;s broadcasting approach instead of iterating over each elements of the array.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">import numpy as np

a = np.array([1,2,3,4,5])
b = np.array([9,8,7,6,5])
c = []
for i in range(len(a)):
    c.append(a[i]+b[i]) # Noncompliant: this computation can be done more efficiently with a broadcasting operation.

c = np.array(c)</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">import numpy as np

a = np.array([1,2,3,4,5])
b = np.array([9,8,7,6,5])
c = a + b # Compliant</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>NumPy Documentation - <a href="https://numpy.org/doc/stable/user/whatisnumpy.html#what-is-numpy">What is NumPy</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>