<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This is an issue when a Rails API controller uses the <code>respond_to</code> method without including the <code>ActionController::MimeResponds</code> module.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Rails API controllers inherit from <code>ActionController::API</code>, which is a lightweight version of <code>ActionController::Base</code>. Unlike the full controller, <code>ActionController::API</code> does not include MIME response functionality by default.</p>
</div>
<div class="paragraph">
<p>The <code>respond_to</code> method is part of the <code>ActionController::MimeResponds</code> module, which provides content negotiation capabilities. When you use <code>respond_to</code> in an API controller without including this module, Rails cannot find the method and raises a <code>NoMethodError</code> at runtime.</p>
</div>
<div class="paragraph">
<p>This error typically occurs when developers migrate from full Rails controllers to API-only controllers, or when they follow examples that assume the full controller functionality is available. The error message can be confusing because Ruby suggests <code>respond_to?</code> (the Object method for checking if an object responds to a method) instead of the missing <code>respond_to</code> method.</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="paragraph">
<p>This issue causes a runtime <code>NoMethodError</code> that will crash the application when the affected controller action is called. This leads to:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>Application downtime and poor user experience</p>
</li>
<li>
<p>Failed API requests that clients cannot handle gracefully</p>
</li>
<li>
<p>Potential data loss if the error occurs during critical operations</p>
</li>
<li>
<p>Debugging time spent tracking down what appears to be a missing method</p>
</li>
</ul>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it_in_rails">How to fix it in Rails</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Include the <code>ActionController::MimeResponds</code> module in your API controller to enable the <code>respond_to</code> method.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-ruby" data-lang="ruby">class ApplicationController &lt; ActionController::API
  # Missing: include ActionController::MimeResponds
end

class Api::UsersController &lt; ApplicationController
  def show
    respond_to do |format| # Noncompliant
      format.json { render json: @user }
      format.xml { render xml: @user }
    end
  end
end</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-ruby" data-lang="ruby">class ApplicationController &lt; ActionController::API
  include ActionController::MimeResponds
end

class Api::UsersController &lt; ApplicationController
  def show
    respond_to do |format|
      format.json { render json: @user }
      format.xml { render xml: @user }
    end
  end
end</code></pre>
</div>
</div>
<div class="paragraph">
<p>Alternatively, include the module directly in the specific controller that needs it.</p>
</div>
</div>
<div class="sect3">
<h4 id="_noncompliant_code_example_2">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-ruby" data-lang="ruby">class Api::ProductsController &lt; ActionController::API
  def index
    respond_to do |format| # Noncompliant
      format.json { render json: @products }
    end
  end
end</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution_2">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-ruby" data-lang="ruby">class Api::ProductsController &lt; ActionController::API
  include ActionController::MimeResponds

  def index
    respond_to do |format|
      format.json { render json: @products }
    end
  end
end</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Rails API Documentation - ActionController::API - <a href="https://api.rubyonrails.org/classes/ActionController/API.html">Official Rails documentation for ActionController::API</a></p>
</li>
<li>
<p>Rails API Documentation - ActionController::MimeResponds - <a href="https://api.rubyonrails.org/classes/ActionController/MimeResponds.html">Official Rails documentation for the MimeResponds module</a></p>
</li>
<li>
<p>Rails Guides - Using Rails for API-only Applications - <a href="https://guides.rubyonrails.org/api_app.html">Rails guide explaining API-only applications and their limitations</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>