This is an issue when a class is intended for scheduled execution but either doesn't implement the Schedulable interface or lacks the required execute method with the correct signature.

== Why is this an issue?

In Salesforce, scheduled jobs are a powerful way to automate business processes at specific times or intervals. However, for a class to be properly scheduled and executed by the Salesforce scheduler, it must follow a specific contract.

The Schedulable interface defines this contract. When you implement this interface, you're telling Salesforce that your class can be scheduled for execution. The interface requires exactly one method: `execute(SchedulableContext context)`.

The execute method serves as the entry point for your scheduled logic. The SchedulableContext parameter provides important information about the scheduled job, such as the job ID and trigger details. This context is essential for monitoring, logging, and potentially chaining additional jobs.

If a class doesn't implement the Schedulable interface, Salesforce won't recognize it as a schedulable class, and attempts to schedule it will fail at runtime. Similarly, if the execute method has the wrong signature (missing the SchedulableContext parameter or incorrect access modifier), the scheduler won't be able to invoke it properly.

The execute method must also be declared as `global` to ensure it's accessible by the Salesforce scheduler, which operates outside your organization's namespace.

=== What is the potential impact?

Without proper implementation of the Schedulable interface:

* **Runtime failures**: Attempts to schedule the class will fail with exceptions
* **Broken automation**: Scheduled business processes won't execute as intended
* **Monitoring issues**: Without SchedulableContext, you lose visibility into job execution details
* **Maintenance problems**: Debugging scheduled job issues becomes more difficult without proper context information

== How to fix it

Implement the Schedulable interface and provide a global execute method that accepts a SchedulableContext parameter. This ensures your class can be properly scheduled and executed by Salesforce.

=== Code examples

==== Noncompliant code example

[source,apex,diff-id=1,diff-type=noncompliant]
----
public class MyScheduledJob {
    public void execute() { // Noncompliant
        System.debug('Scheduled execution');
    }
}
----

==== Compliant solution

[source,apex,diff-id=1,diff-type=compliant]
----
public class MyScheduledJob implements Schedulable {
    global void execute(SchedulableContext context) {
        System.debug('Scheduled execution');
    }
}
----

== Resources

=== Documentation

 * Schedulable Interface - https://developer.salesforce.com/docs/atlas.en-us.apexref.meta/apexref/apex_interface_system_schedulable.htm[Official Salesforce documentation for the Schedulable interface]

 * Apex Scheduler - https://developer.salesforce.com/docs/atlas.en-us.apexcode.meta/apexcode/apex_scheduler.htm[Guide to scheduling Apex classes for execution]

 * SchedulableContext Interface - https://developer.salesforce.com/docs/atlas.en-us.apexref.meta/apexref/apex_interface_system_schedulablecontext.htm[Documentation for the SchedulableContext parameter]
