Promises need to be resolved or awaited to return the expected value, otherwise, they return the promise object.

This rule forbids providing promises in conditionals.

// If you want to factorize the description uncomment the following line and create the file.
//include::../description.adoc[]

== Why is this an issue?

Forgetting to await a promise is a frequent mistake. There are places where it makes no sense to use a promise, but the developer forgot to resolve it.

=== What is the potential impact?

Using a promise instead of its resolved values can have unexpected results leading to bugs.
In conditionals, it will always return a truthy value.

== How to fix it

Await the promise so the value gets resolved correctly.

//== How to fix it in FRAMEWORK NAME

=== Code examples

==== Noncompliant code example

[source,javascript,diff-id=1,diff-type=noncompliant]
----
const promise = new Promise((resolve, reject) => resolve(5));
if (promise) {
  // ...
}
----

==== Compliant solution

[source,javascript,diff-id=1,diff-type=compliant]
----
const promise = new Promise((resolve, reject) => resolve(5));
if (promise) {
  // ...
}
----

//=== How does this work?

//=== Pitfalls

//=== Going the extra mile


== Resources

=== Documentation
* https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/async_function[MDN async/await]
* https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Using_promises[MDN using promises]

//=== Articles & blog posts
//=== Conference presentations
//=== Standards
