<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Dependencies should be grouped by their destination, otherwise,
it becomes hard for the reader to see which destination has what dependencies.</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="sect3">
<h4 id="_readability">Readability</h4>
<div class="paragraph">
<p>This change improves readability by making it easier to see which destination has what dependencies.</p>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Reorder your dependencies so that they are grouped by destination.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-kotlin" data-lang="kotlin">dependencies {
    implementation("org.jetbrains.kotlin:kotlin-stdlib")
    implementation("com.google.guava:guava:30.1.1-jre")
    implementation("org.apache.commons:commons-lang3:3.9")

    testImplementation(testLibs.junit.api)
    testRuntimeOnly(testLibs.junit.engine)

    testImplementation(testLibs.assertj.core)
    testImplementation(libs.sonar.plugin.api)

    implementation("log4j:log4j:1.2.17")
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-kotlin" data-lang="kotlin">dependencies {
    implementation("org.jetbrains.kotlin:kotlin-stdlib")
    implementation("com.google.guava:guava:30.1.1-jre")
    implementation("org.apache.commons:commons-lang3:3.9")
    implementation("log4j:log4j:1.2.17")

    testImplementation(testLibs.junit.api)
    testImplementation(testLibs.assertj.core)
    testImplementation(libs.sonar.plugin.api)

    testRuntimeOnly(testLibs.junit.engine)
}</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p><a href="https://github.com/liutikas/gradle-best-practices#keep-your-dependencies-clustered">Best Practices when using Gradle - Keep your dependencies clustered (Liutikas)</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>