include::../description.adoc[]

include::../ask-yourself.adoc[]

include::../recommended.adoc[]

== Noncompliant Code Example

A customer managed policy that grants all permissions by using the wildcard (*) in the ``++Action++`` property:

----
MyPolicy:
    Type: AWS::IAM::ManagedPolicy
    Properties: 
        PolicyDocument: 
            Version: "2012-10-17"
            Statement:
                - Effect: Allow
                Action:
                    - "*" # Sensitive
                Resource:
                    - !Ref MyResource
        Roles: 
            - !Ref MyRole
----

== Compliant Solution

A customer managed policy that lists and grants only the required permissions:

----
MyPolicy:
    Type: AWS::IAM::ManagedPolicy
    Properties: 
        PolicyDocument: 
            Version: "2012-10-17"
            Statement:
                - Effect: Allow
                Action:
                    - "s3:GetObject"
                Resource:
                    - !Ref MyResource
        Roles: 
            - !Ref MyRole
----

include::../see.adoc[]