include::../description.adoc[]

include::../ask-yourself.adoc[]

include::../recommended.adoc[]

== Noncompliant code example

A customer-managed policy that grants all permissions by using the wildcard (*) in the ``++Action++`` property:

[source,cloudformation]
----
MyPolicy:
    Type: AWS::IAM::ManagedPolicy
    Properties: 
        PolicyDocument: 
            Version: "2012-10-17"
            Statement:
                - Effect: Allow
                Action:
                    - "*" # Sensitive
                Resource:
                    - !Ref MyResource
        Roles: 
            - !Ref MyRole
----

== Compliant solution

A customer-managed policy that grants only the required permissions:

[source,cloudformation]
----
MyPolicy:
    Type: AWS::IAM::ManagedPolicy
    Properties: 
        PolicyDocument: 
            Version: "2012-10-17"
            Statement:
                - Effect: Allow
                Action:
                    - "s3:GetObject"
                Resource:
                    - !Ref MyResource
        Roles: 
            - !Ref MyRole
----

include::../see.adoc[]
