=== Code examples

==== Noncompliant code example

[source,csharp,diff-id=1,diff-type=noncompliant]
----
struct Point
{
    public int X { get; set; }
    public int Y { get; set; }
}
----

==== Compliant solution

[source,csharp,diff-id=1,diff-type=compliant]
----
struct Point
{
    public int X { get; set; }
    public int Y { get; set; }

    public bool Equals(Point other) =>
        X == other.X && Y == other.Y;

    public override bool Equals(object obj) =>
        obj is Point other && Equals(other);

    public override int GetHashCode()
    {
        unchecked
        {
            return (X.GetHashCode() * 397) ^ Y.GetHashCode();
        }
    }
}
----
