include::../description.adoc[]

== Noncompliant Code Example

----
If GetTrue() Or GetFalse() Then ' Noncompliant; both sides evaluated
End If
----

== Compliant Solution

----
If GetTrue() OrElse GetFalse() Then ' true short-circuit logic
End If
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
