== Why is this an issue?

include::../description.adoc[]

include::../exceptions.adoc[]

== How to fix it

=== Code examples

==== Noncompliant code example

[source,vbnet,diff-id=1,diff-type=noncompliant]
----
<ServiceContract>
Interface IMyService
    <OperationContract(IsOneWay:=True)>
    Function SomethingHappened(ByVal parameter As Integer) As Integer ' Noncompliant
End Interface
----

==== Compliant solution

[source,vbnet,diff-id=1,diff-type=noncompliant]
----
<ServiceContract>
Interface IMyService
    <OperationContract(IsOneWay:=True)>
    Sub SomethingHappened(ByVal parameter As Integer)
End Interface
----

include::../resources.adoc[]

ifdef::env-github,rspecator-view,env-vscode[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../highlighting.adoc[]

endif::env-github,rspecator-view,env-vscode[]
