include::../description.adoc[]

== Noncompliant Code Example

----
<ServiceContract>
Interface IMyService
    <OperationContract(IsOneWay:=True)>
    Function SomethingHappened(ByVal parameter As Integer) As Integer ' Noncompliant
End Interface
----

== Compliant Solution

----
<ServiceContract>
Interface IMyService
    <OperationContract(IsOneWay:=True)>
    Sub SomethingHappened(ByVal parameter As Integer)
End Interface
----

include::../exceptions.adoc[]
ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../highlighting.adoc[]

endif::env-github,rspecator-view[]
