include::../description.adoc[]

== Noncompliant Code Example

----
if (GetTrue() | GetFalse()) // Noncompliant; both sides evaluated
{
}
----

== Compliant Solution

----
if (GetTrue() || GetFalse()) // true short-circuit logic
{
}
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
