== Why is this an issue?

There is no reason to concatenate literal strings. Doing so is an exercise in reducing code readability. Instead, the strings should be combined.


=== Noncompliant code example

[source,php]
----
$msg = "Hello " . "${name}" . "!";  // Noncompliant
----


=== Compliant solution

[source,php]
----
$msg = "Hello ${name}!";
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
