This rule raises an issue when `String.isEmpty()` is used to validate strings that should also reject whitespace-only content.

== Why is this an issue?

The `String.isEmpty()` method only checks if a string is null or has zero length. It does not consider strings containing only whitespace characters as empty, which can lead to unexpected behavior in validation logic.

For example, a string containing only spaces (`' '`) will return `false` when checked with `isEmpty()`, even though it contains no meaningful content. This can cause issues in:

* User input validation where whitespace-only entries should be rejected
* Data processing where empty fields need to be identified
* Business logic that depends on meaningful string content

The `String.isBlank()` method provides more comprehensive validation by returning `true` for null strings, empty strings, and strings containing only whitespace characters. This makes it the better choice when you need to ensure a string contains actual content.

=== What is the potential impact?

Using `isEmpty()` instead of `isBlank()` can lead to validation bypasses where whitespace-only strings are incorrectly treated as valid input. This may result in poor data quality, unexpected application behavior, or business logic errors when processing user input or data fields.

== How to fix it

Replace `String.isEmpty()` with `String.isBlank()` when you need to validate that a string contains meaningful content, not just check if it's null or empty.

=== Code examples

==== Noncompliant code example

[source,apex,diff-id=1,diff-type=noncompliant]
----
if (!String.isEmpty(inputString)) { // Noncompliant
    processString(inputString);
}
----

==== Compliant solution

[source,apex,diff-id=1,diff-type=compliant]
----
if (!String.isBlank(inputString)) {
    processString(inputString);
}
----

== Resources

=== Documentation

 * Apex String Class Methods - https://developer.salesforce.com/docs/atlas.en-us.apexref.meta/apexref/apex_methods_system_string.htm[Official Salesforce documentation for String class methods including isBlank() and isEmpty()]

=== Related rules

 * RSPEC-7158 - https://rules.sonarsource.com/java/RSPEC-7158/[Java rule for using isBlank() instead of isEmpty() for string validation]
