== How to fix it in Bicep ARM Template

=== Code examples

==== Noncompliant code example

[source,bicep,diff-id=1,diff-type=noncompliant]
----
resource roleDef 'Microsoft.Authorization/roleDefinitions@2022-04-01' = { // Sensitive
  properties: {
    permissions: [
      {
        actions: ['*']
        notActions: []
      }
    ]

    assignableScopes: [
      subscription().id
    ]
  }
}
----

==== Compliant solution

[source,bicep,diff-id=1,diff-type=compliant]
----
resource roleDef 'Microsoft.Authorization/roleDefinitions@2022-04-01' = {
  properties: {
    permissions: [
      {
        actions: ['Microsoft.Compute/*']
        notActions: []
      }
    ]

    assignableScopes: [
      subscription().id
    ]
  }
}
----
