include::../description.adoc[]

To match a literal string, rather than a regular expression, either all special characters should be escaped or methods that donâ€™t use regular expressions should be used.


== Noncompliant Code Example

----
re.compile(r"([")
re.sub(r"([", input, "{")
re.compile(r"(\w+-(\d+)")
----


== Compliant Solution

----
re.compile(r"\(\[")
input.replace("([", "{")
re.compile(r"(\w+)-(\d+)")
----

