== How to fix it in OkHttp

=== Code examples

include::../../common/fix/code-rationale.adoc[]

:cert_method_name: javax.net.ssl.HostnameVerifier.verify()

include::../../common/fix/code-rationale-override.adoc[]

==== Noncompliant code example

[source,java,diff-id=1,diff-type=noncompliant]
----
import javax.net.ssl.HttpsURLConnection
import javax.net.ssl.SSLSession
import javax.net.ssl.HostnameVerifier
import okhttp3.OkHttpClient
import okhttp3.Request
import okhttp3.Response

fun request() {
    val builder = OkHttpClient.Builder()
    builder.hostnameVerifier(object : HostnameVerifier {
      override fun verify(hostname: String?, session: SSLSession?): Boolean { // Noncompliant
        return true
      }
    })

    OkHttpClient client = builder.build()

    Request request = new Request.Builder()
            .url("https://example.com")
            .build()

    Response response = client.newCall(request).execute()
}
----

==== Compliant solution

[source,java,diff-id=1,diff-type=compliant]
----
import javax.net.ssl.HttpsURLConnection
import javax.net.ssl.SSLSession
import javax.net.ssl.HostnameVerifier
import okhttp3.OkHttpClient
import okhttp3.Request
import okhttp3.Response

fun request() {
    val builder = OkHttpClient.Builder()

    OkHttpClient client = builder.build()

    Request request = new Request.Builder()
            .url("https://example.com")
            .build()

    Response response = client.newCall(request).execute()
}
----

=== How does this work?

include::../../common/fix/validation.adoc[]

include::../../common/fix/keytool.adoc[]

