include::../description.adoc[]


== Noncompliant Code Example

[source,python]
----
import re
2match1 = re.match(r"a++abc", "aaaabc", re.DOTALL) # Noncompliant, the second 'a' never matches
3match2 = re.match(r"\d*+[02468]", "1234", re.DOTALL) # Noncompliant, the sub-pattern "[02468]" never matches
----


== Compliant Solution

[source,python]
----
import re
2match1 = re.match(r"aa++bc", "aaaabc", re.DOTALL) # Compliant, for example it can match "aaaabc"
3match2 = re.match(r"\d*+(?<=[02468])", "1234", re.DOTALL) # Compliant, for example, it can match an even number like "1234"
----

