== Why is this an issue?

The type of an integer literal depends on three parameters: the value of the literal, whether it is written in decimal or not, and its suffix.

A literal with a single `L` (or `l`) will usually be of a (possibly `unsigned`) `long` type. But if its value cannot be represented in this type, it will be of a (possibly `unsigned`) `long long` type instead.

In such a case, it is more straightforward to use `LL` in the literal suffix, which unambiguously specifies a `long long` type.

Note: This rule targets classical integer literals, not user-defined literals whose type is entirely determined by their suffix.


== How to fix it

=== Code examples

==== Noncompliant code example

This example assumes that `long` is 32 bits and `long long` is 64 bits. In this case, `12345` fits into a `long`, but `12345678901` does not. Similarly, `0xdeadbeef` fits into an `unsigned long`, but `0xfeeddeadbeef` does not.

[source,cpp,diff-id=1,diff-type=noncompliant]
----
auto i1 = 12345678901L; // Non-compliant, the literal is of type long long
auto i2 = 12345678901UL; // Non-compliant, the literal is of type unsigned long long
auto i3 = 0xfeeddeadbeefL; // Non-compliant, the literal is of type unsigned long long

auto i4 = 12345L; // Compliant, the literal is of type long
auto i5 = 12345UL; // Compliant, the literal is of type unsigned long
auto i6 = 0xdeadbeef; // Compliant, the literal is of type unsigned long
auto i7 = 12345678901; // Compliant, no suffix is present: the compiler is free to decide
----

==== Compliant solution

[source,cpp,diff-id=1,diff-type=compliant]
----
auto i1 = 12345678901LL; // Compliant, the literal is of type long long
auto i2 = 12345678901ULL; // Compliant, the literal is of type unsigned long long
auto i3 = 0xfeeddeadbeefLL; // Compliant, the literal is of type unsigned long long

auto i4 = 12345L; // Compliant, the literal is of type long
auto i5 = 12345UL; // Compliant, the literal is of type unsigned long
auto i6 = 0xdeadbeef; // Compliant, the literal is of type unsigned long
auto i7 = 12345678901; // Compliant, no suffix is present: the compiler is free to decide
----

== Resources
=== Standards
=== External coding guidelines
// MISRA add link to MISRA M023_400
* MISRA {cpp}:2023, 5.13.6 - An integer-literal of type long long shall not use a single L or l in any suffix
