Java 18 introduced the possibility to include code snippets in a javadoc through the usage of the `@snippet` tag. Escape sequences in attributes of the `@snippet` tag should not be used because they are not supported and ignored.


== Noncompliant Code Example ==
[source,java]
----
{@snippet name="\U+039B-lambda":                          // Noncompliant
    public static void main(String... args) {
       System.out.println("Lambda is my name!");
   }
}
----


== Compliant Solution ==
[source,java]
----
{@snippet name="lambda":
    public static void main(String... args) {
       System.out.println("Lambda is my name!");
   }
}
----


== See ==
Oracleâ€™s Programmer's Guide to Snippets: https://docs.oracle.com/.../code-snippet/index.html[https://docs.oracle.com/en/java/javase/18/code-snippet/index.html]


ifdef::env-github,rspecator-view[]

'''

== Implementation Specification ==
(visible only on this page)

=== Message ===

Remove this unnecessary escape sequence

=== Highlighting ===

The escape sequence

endif::env-github,rspecator-view[]
