include::../description.adoc[]

include::../ask-yourself.adoc[]

include::../recommended.adoc[]

== Noncompliant Code Example

[source,docker]
----
FIXME
----

== Compliant Solution

[source,docker]
----
FIXME
----

include::../see.adoc[]
