include::../description.adoc[]

== Noncompliant Code Example

[source,kotlin]
----
foo(); bar();
----

== Compliant Solution

[source,kotlin]
----
foo();
bar();
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../highlighting.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
