This rule raises an issue when a class is declared with the `global` modifier, especially when implementing interfaces that don't require global access.

== Why is this an issue?

The `global` modifier in Apex creates a permanent public API contract that can never be changed or removed once deployed. This is particularly problematic in managed packages where global elements become part of the package's public interface.

Many developers use `global` out of habit or because certain interfaces required it in the past. However, Salesforce has updated many system interfaces (like `Schedulable`, `Batchable`, `WebServiceMock`, and `HttpCalloutMock`) to no longer require the `global` modifier.

When you declare a class as `global`, you're making a commitment that:

* The class signature cannot be changed
* The class cannot be deleted
* All global methods must remain unchanged
* The class becomes part of your org's or package's permanent API

This creates technical debt and reduces flexibility for future development. Once something is global, you're locked into maintaining it forever, even if it's no longer needed.

=== What is the potential impact?

Using unnecessary `global` modifiers creates permanent technical debt by locking classes into unchangeable API contracts. This reduces code maintainability and flexibility, especially in managed packages where global elements become part of the public interface that cannot be modified or removed.

== How to fix it

Remove the `global` modifier and use `public` instead when the class doesn't need to be globally accessible. Most interfaces no longer require global access.

=== Code examples

==== Noncompliant code example

[source,apex,diff-id=1,diff-type=noncompliant]
----
global class JobSyncScheduler implements Schedulable { // Noncompliant
    global void execute(SchedulableContext context) {
        // Implementation
    }
}
----

==== Compliant solution

[source,apex,diff-id=1,diff-type=compliant]
----
public class JobSyncScheduler implements Schedulable {
    public void execute(SchedulableContext context) {
        // Implementation
    }
}
----

== Resources

=== Documentation

 * Apex Developer Guide - Access Modifiers - https://developer.salesforce.com/docs/atlas.en-us.apexcode.meta/apexcode/apex_classes_access_modifiers.htm[Official Salesforce documentation on access modifiers including global usage guidelines]

 * Apex Developer Guide - Batch Apex - https://developer.salesforce.com/docs/atlas.en-us.apexcode.meta/apexcode/apex_batch.htm[Documentation on Batch Apex implementation without global modifier requirements]

 * Apex Developer Guide - Scheduled Apex - https://developer.salesforce.com/docs/atlas.en-us.apexcode.meta/apexcode/apex_scheduler.htm[Documentation on Schedulable interface implementation]
