include::../common/description.adoc[]

include::../common/impact.adoc[]

include::../common/fix.adoc[]

=== Code examples

==== Noncompliant code example

[source,swift,diff-id=1,diff-type=noncompliant]
----
import CryptoSwift

let bytes:Array<UInt8> = [0x01, 0x02, 0x03]
let digest = input.md5()  // Noncompliant
----

==== Compliant solution

[source,swift,diff-id=1,diff-type=compliant]
----
import CryptoSwift

let bytes:Array<UInt8> = [0x01, 0x02, 0x03]
let digest = input.sha512()
----

include::../common/fix-pitfalls.adoc[]

include::../common/resources.adoc[]


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../common/message.adoc[]

include::../common/highlighting.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../common/comments-and-links.adoc[]

endif::env-github,rspecator-view[]
