<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>When a method uses a <code>Gatherer.of(&#8230;&#8203;)</code> factory and provides a combiner function that always throws an exception, this implicitly signals that the gatherer is designed for sequential processing.
The <code>java.util.stream.Gatherer</code> API provides <code>Gatherer.ofSequential(&#8230;&#8203;)</code> factories which clearly indicates that the gatherer is intended for sequential streams.
Using such a factory improves code clarity, makes the intended processing model explicit, and avoids the need for a dummy or throwing combiner.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Use <code>Gatherer.ofSequential</code> instead of <code>Gatherer.of</code>.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">  public static List&lt;Integer&gt; diffWithFirstPositive(List&lt;Integer&gt; list) {
    Gatherer&lt;Integer, AtomicInteger, Integer&gt; gatherer = Gatherer.of(
      () -&gt; new AtomicInteger(-1),
      (state, number, downstream) -&gt; {
        if (state.get() &lt; 0) {
          state.set(number);
          return true;
        }
        return downstream.push(number - state.get());
      },
      (_, _) -&gt; {
        throw new IllegalStateException();
      },
      Gatherer.defaultFinisher());
    return list.stream().gather(gatherer).toList();
  }</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">  public static List&lt;Integer&gt; diffWithFirstPositive(List&lt;Integer&gt; list) {
    Gatherer&lt;Integer, AtomicInteger, Integer&gt; gatherer = Gatherer.ofSequential(
      () -&gt; new AtomicInteger(-1),
      (state, number, downstream) -&gt; {
        if (state.get() &lt; 0) {
          state.set(number);
          return true;
        }
        return downstream.push(number - state.get());
      },
      Gatherer.defaultFinisher());
    return list.stream().gather(gatherer).toList();
  }</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p><a href="https://docs.oracle.com/en/java/javase/24/docs/api/java.base/java/util/stream/Gatherer.html">Gatherer API</a></p>
</li>
<li>
<p><a href="https://docs.oracle.com/en/java/javase/24/core/stream-gatherers.html">Stream gatherers documentation</a></p>
</li>
</ul>
</div>
</div>
</div>