``++offsetof++`` can lead to undefined behavior when the argument types are incompatible or when bit fields are used. Therefore ``++offsetof++`` should be avoided.


== Noncompliant Code Example

----
#include <stddef.h>

struct A
{
  int32_t i;
};

void f1 ( )
{
  offsetof ( A, i ); // Noncompliant
}
----


== See

* MISRA C:2004, 20.6 - The macro offsetof, in library <stddef.h>, shall not be used.
* MISRA {cpp} 2008, 18-2-1 - The macro offsetof, in library <stddef.h>, shall not be used.


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
