== Why is this an issue?

Labels are not commonly used, and many developers do not understand how they work. Moreover, their usage makes the control flow harder to follow, which reduces the code's readability.

=== Noncompliant code example

[source,javascript]
----
myLabel: {
  let x = doSomething();
  if (x > 0) {
    break myLabel;
  }
  doSomethingElse();
}
----

=== Compliant solution

[source,javascript]
----
let x = doSomething();
if (x <= 0) {
  doSomethingElse();
}
----
Â 

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

=== on 14 Mar 2017, 16:53:55 Elena Vilchik wrote:
\[~ann.campbell.2] After SONARJS-953 this rule will not raise issue for any loop. WDYT how this should be in RSPEC? exception?

=== on 14 Mar 2017, 18:35:20 Ann Campbell wrote:
\[~elena.vilchik] I've added an exception. Since I'm not sure I've ever seen any other use of labels outside of COBOL, will you update the code samples?

=== on 14 Mar 2017, 18:51:42 Elena Vilchik wrote:
\[~ann.campbell.2] thanks! done!

=== on 20 Jan 2020, 15:11:56 Tibor Blenessy wrote:
I removed the exception, see main RSPEC-1119 for explanation

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
