<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Private methods are intended for use only within their scope. If these methods return values that are not utilized by any calling functions, it indicates that the return operation is unnecessary. Removing such returns can enhance both efficiency and code clarity.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-csharp" data-lang="csharp">class SomeClass
{
     private int PrivateMethod() =&gt; 42;

     public void PublicMethod()
     {
          PrivateMethod(); // Noncompliant: the result of PrivateMethod is not used
     }
}</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Change return type to "void"; not a single caller uses the returned value.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_is_related_to_s1144">is related to: <a data-rspec-id="S1144" class="rspec-auto-link">S1144</a></h3>

</div>
<div class="sect2">
<h3 id="_on_7_jul_2015_141353_ann_campbell_wrote">on 7 Jul 2015, 14:13:53 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~tamas.vajk] I&#8217;ve left this out of the default profiles on the assumption that it will be controversial. Feel free to change that if you&#8217;re confident about it.</p>
</div>
<div class="paragraph">
<p>Also, I was a little shaky on the terminology. I was tempted to use "DLL" instead of scope, but not confident enough to do it.</p>
</div>
<div class="paragraph">
<p>And finally, I didn&#8217;t try to write any code samples because I think this is the kind of thing (like "Files should not be too long") that is difficult to illustrate with brevity. Jump in if you disagree. :-)</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_8_jul_2015_084639_tamas_vajk_wrote">on 8 Jul 2015, 08:46:39 Tamas Vajk wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] It looks good. I&#8217;ve added it to the SonarQube Way profile. It won&#8217;t generate too much noise because of the restriction on the visibility.</p>
</div>
<div class="paragraph">
<p>A sidenote: whenever you want to write DLL, you should write assembly, because that covers DLLs and EXEs as well. But in this case the scope is better, because for an <code>internal</code> the scope will be the assembly, for a <code>private</code> one it will be the class.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_8_jul_2015_204430_freddy_mallet_wrote">on 8 Jul 2015, 20:44:30 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>@Ann, I would associate this rule to the SQALE characteristic Maintainability &gt; Understandability and I tend to agree with @Tamas : according to the limited scope of this rule, it should not trigger some false-positives. I would even make this RSPEC targeting Java and C&#43;&#43;.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_9_jul_2015_120016_ann_campbell_wrote">on 9 Jul 2015, 12:00:16 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>sqale updated. targeting had already been done.</p>
</div>
</div>
</div>
</div>