This rule raises an issue when text-based file operations are performed without explicitly specifying a charset encoding.

== Why is this an issue?

When reading from or writing to files containing text, the system uses a default charset encoding if none is specified. This default encoding varies between operating systems and system configurations:

* Windows typically uses CP1252 or UTF-16
* Linux and macOS typically use UTF-8
* Some systems may use ISO-8859-1 or other encodings

This creates several problems:

**Portability Issues**: Code that works correctly on one system may fail or produce incorrect results on another system with a different default encoding.

**Data Corruption**: Files containing non-ASCII characters (accented letters, symbols, emojis, or text in non-Latin scripts) may be read incorrectly or written in a way that corrupts the data.

**Silent Failures**: Encoding issues often don't cause immediate exceptions but instead produce garbled text that may only be noticed later in the application lifecycle.

**Maintenance Difficulties**: When charset is not explicit, it becomes harder for developers to understand the intended behavior and troubleshoot encoding-related issues.

Modern applications frequently handle international content, user-generated text, or data from various sources, making explicit charset specification essential for reliable text processing.

=== What is the potential impact?

Files may be corrupted or misread when moved between systems with different default encodings. Non-ASCII characters may appear as question marks, boxes, or other garbled text. In severe cases, this can lead to data loss or application failures when processing international content.

== How to fix it

Specify the charset explicitly when reading file content.

=== Code examples

==== Noncompliant code example

[source,groovy,diff-id=1,diff-type=noncompliant]
----
def content = file.getText() // Noncompliant
----

==== Compliant solution

[source,groovy,diff-id=1,diff-type=compliant]
----
def content = file.getText('UTF-8') // Compliant
----

== Resources
