== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,kotlin]
----
// FIXME denominator value might be  0
fun divide(numerator: Int, denominator: Int): Int = numerator / denominator
----

include::../see.adoc[]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
