include::../description.adoc[]

== Noncompliant Code Example

----
// FIXME denominator value might be  0
fun divide(numerator: Int, denominator: Int): Int = numerator / denominator
----

include::../see.adoc[]

ifdef::env-github,rspecator-view[]
'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
