<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>The values that can be represented by a signed bit field with a length of one bit may not meet developer expectations. For example, according to the C99 Standard, a single-bit signed bit-field has a single (one) sign bit and no (zero) value bits.</p>
</div>
<div class="paragraph">
<p>This rule does not apply to unnamed bit fields, as their values cannot be accessed.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">signed int f:1;  // Noncompliant; there's only room here for the sign</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">unsigned int f:1;</code></pre>
</div>
</div>
<div class="paragraph">
<p>or</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">signed int:1; // unnamed</code></pre>
</div>
</div>
<div class="paragraph">
<p>or</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">signed int f:2;</code></pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p>MISRA C:2004, 6.5 - Bit fields of type signed int shall be at least 2 bits long</p>
</li>
<li>
<p>MISRA C:2012, 6.2 - Single-bit named bit fields shall not be of a signed type</p>
</li>
<li>
<p>MISRA C&#43;&#43;:2008, 9-6-4 - Named bit-fields with signed integer type shall have a length of more than one bit</p>
</li>
</ul>
</div>
<hr>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Review "X" declaration sign and size.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_is_duplicated_by_s1010">is duplicated by: <a data-rspec-id="S1010" class="rspec-auto-link">S1010</a></h3>

</div>
<div class="sect2">
<h3 id="_on_9_dec_2014_205820_evgeny_mandrikov_wrote">on 9 Dec 2014, 20:58:20 Evgeny Mandrikov wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] could you please verify this description, which was built as a combination of MISRA C:2012 and MISRA C&#43;&#43;2008?</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_14_dec_2014_225108_evgeny_mandrikov_wrote">on 14 Dec 2014, 22:51:08 Evgeny Mandrikov wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] I was expecting that you&#8217;ll also specify SQALE model and other parameters, but ok&#8230;&#8203;</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_15_dec_2014_145847_ann_campbell_wrote">on 15 Dec 2014, 14:58:47 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>Sorry [~evgeny.mandrikov], I didn&#8217;t notice the lack of SQALE &amp;etc. I&#8217;ll pay more attention next time.</p>
</div>
</div>
</div>
</div>