== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,vbnet]
----
Public Class [MyClass]
    Private Sub New() // Noncompliant
        ' ...
    End Sub
End Class
----

=== Compliant solution

[source,vbnet]
----
Public Class [MyClass]
    Public Sub New()
        ' ...
    End Sub
End Class
----

include::../exceptions.adoc[]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../highlighting.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
