<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>When the <code>RLSE</code> parameter is specified in a <code>SPACE</code> directive, the system releases unused space that has been allocated for the data set.</p>
</div>
<div class="paragraph">
<p>Omitting this parameter can cause a waste of unneeded space reserved for the data set.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-jcl" data-lang="jcl">//STEP1 EXEC PGM=IEFBR14
//OUT1  DD  DSN=A.B.C,
//          UNIT=3390,VOL=SER=Z19SPL,
//          DISP=(NEW,CATLG),
//          SPACE=(CYL,(1,1)) &lt;-- Noncompliant</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-jcl" data-lang="jcl">//STEP1 EXEC PGM=IEFBR14
//OUT1  DD  DSN=A.B.C,
//          UNIT=3390,VOL=SER=Z19SPL,
//          DISP=(NEW,CATLG),
//          SPACE=(CYL,(1,1),RLSE)</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>IBM Documentation - <a href="https://www.ibm.com/docs/en/zos/3.1.0?topic=statement-space-parameter">SPACE parameter</a></p>
</li>
<li>
<p>IBM Documentation - <a href="https://www.ibm.com/docs/en/zos/3.1.0?topic=definition-system-assignment-space">System assignment of space</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>