<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>When multiple variables are <code>DECLARE</code>d in a single statement, parentheses must be used around the variable list to ensure the variable attributes in the statement are applied to all the variables in the declaration. Without parentheses, the attributes are applied only to the last variable in the list.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-pli" data-lang="pli">DECLARE X,Y FIXED BIN(15);</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-pli" data-lang="pli">DECLARE (X,Y) FIXED BIN(15);</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Add parentheses around this variable list; only "x" has the desired type.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>variable list. E.G. <code>X,Y</code> in: <code>DECLARE X,Y FIXED BIN(15);</code></p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_25_apr_2017_180244_pierre_yves_nicolas_wrote">on 25 Apr 2017, 18:02:44 Pierre-Yves Nicolas wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] Is it a bug or a code smell?</p>
</div>
<div class="paragraph">
<p>If we are careful about potential noise, I think we should either:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>raise issues in all cases and consider this rule as a code smell</p>
</li>
<li>
<p>only raise issues when the intent of the developer is clearly not what the code does and consider the rule as a bug</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>If we go for the 2nd option, we can maybe check how the code is formatted:</p>
</div>
<div class="listingblock">
<div class="content">
<pre>DECLARE X,Y FIXED BIN(15); // Noncompliant
DECLARE
 X,
 Y FIXED BIN(15); // compliant</pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_on_25_apr_2017_180815_ann_campbell_wrote">on 25 Apr 2017, 18:08:15 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~pierre-yves.nicolas] I honestly don&#8217;t know. I suspect that with or without linebreaks, it&#8217;s not the developer&#8217;s intent, but my plan was to implement &amp; evaluate.</p>
</div>
</div>
</div>
</div>