== Why is this an issue?

JUnit is a testing framework for the Java programming language.
It provides a platform to write and execute unit tests for Java code, ensuring the correctness and reliability of the software.

JUnit 4 is the earlier version of the JUnit framework, and it has its distinct ways of writing and executing tests.
However, with the introduction of JUnit 5 (JUnit Jupiter), significant changes and improvements were made to the framework.
The JUnit Vintage engine was created as a compatibility layer to support legacy test code written in JUnit 4.
In a project, you can use both JUnit4 and JUnit5 tests side by side.

== How to fix it

Add `junit-vintage-engine` dependency to the project.

=== Code examples

==== Noncompliant code example

[source,xml,diff-id=1,diff-type=noncompliant]
----
<dependencies>
    <dependency>
        <groupId>org.junit.jupiter</groupId>
        <artifactId>junit-jupiter-engine</artifactId>
        <version>${junit-jupiter.version}</version>
        <scope>test</scope>
    </dependency>  // Noncompliant
</dependencies>
----

==== Compliant solution

[source,xml,diff-id=1,diff-type=compliant]
----
<dependencies>
    <dependency>
        <groupId>org.junit.jupiter</groupId>
        <artifactId>junit-jupiter-engine</artifactId>
        <version>${junit-jupiter.version}</version>
        <scope>test</scope>
    </dependency>
    <dependency>
        <groupId>org.junit.vintage</groupId>
        <artifactId>junit-vintage-engine</artifactId>
        <version>${junit-jupiter.version}</version>
    </dependency>  // Compliant
</dependencies>
----

== Resources
=== Documentation
* https://junit.org/junit5/[JUnit 5]
* https://junit.org/junit4/[JUnit 4]
