<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Named placeholders in <a href="https://messagetemplates.org">message templates</a> should be unique. The meaning of the named placeholders is to store the value of the provided argument under that name, enabling easier log querying. Since the named placeholder is used multiple times, it cannot store the different values uniquely with each name hence not serving its original purpose.
There can be different behaviours when using the same named placeholder multiple times:</p>
</div>
<div class="ulist">
<ul>
<li>
<p><a href="https://www.nuget.org/packages/Microsoft.Extensions.Logging">Microsoft.Extensions.Logging</a> saves the different values under the same name</p>
</li>
<li>
<p><a href="https://www.nuget.org/packages/Serilog">Serilog</a> stores only the latest assigned value</p>
</li>
<li>
<p><a href="https://www.nuget.org/packages/NLog">Nlog</a> makes the name unique by suffixing it with <code>_index</code></p>
</li>
</ul>
</div>
<div class="paragraph">
<p>The rule covers the following logging frameworks:</p>
</div>
<div class="ulist">
<ul>
<li>
<p><a href="https://www.nuget.org/packages/Microsoft.Extensions.Logging">Microsoft.Extensions.Logging</a></p>
</li>
<li>
<p><a href="https://www.nuget.org/packages/Serilog">Serilog</a></p>
</li>
<li>
<p><a href="https://www.nuget.org/packages/NLog">Nlog</a></p>
</li>
</ul>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Assign unique names to each template placeholder.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-csharp" data-lang="csharp">public void Checkout(ILogger logger, User user, Order order)
{
    logger.LogDebug("User {Id} purchased order {Id}", user.Id, order.Id);
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-csharp" data-lang="csharp">public void Checkout(ILogger logger, User user, Order order)
{
    logger.LogDebug("User {UserId} purchased order {OrderId}", user.Id, order.Id);
}</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Message Templates - <a href="https://messagetemplates.org/">Message template specification</a></p>
</li>
<li>
<p>Microsoft Learn - <a href="https://learn.microsoft.com/en-us/dotnet/core/extensions/logging?tabs=command-line#log-message-template-formatting">Log message template formatting</a></p>
</li>
<li>
<p>NLog - <a href="https://github.com/NLog/NLog/wiki/How-to-use-structured-logging">How to use structured logging</a></p>
</li>
<li>
<p>Serilog - <a href="https://github.com/serilog/serilog/wiki/Structured-Data">Structured Data</a></p>
</li>
<li>
<p>Serilog - <a href="https://github.com/Suchiman/SerilogAnalyzer/blob/master/README.md#serilog005-unique-property-name-verifier"><code>Serilog005</code>: Unique Property Name Verifier</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>