== Why is this an issue?

Rather than creating a boxed primitive from a ``++String++`` to extract the primitive value, use the relevant ``++parse++`` method instead. It will be clearer and more efficient.


=== Noncompliant code example

[source,java]
----
String myNum = "12.2";

float f = (new Float(myNum)).floatValue();  // Noncompliant; creates & discards a Float
----


=== Compliant solution

[source,java]
----
String myNum = "12.2";

float f = Float.parseFloat(myNum);
----



ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Use "XXX.parseYyy" for this string-to-yyy conversion.


'''
== Comments And Links
(visible only on this page)

=== on 10 Oct 2014, 13:58:53 Freddy Mallet wrote:
@Ann, for me this RSPEC fully duplicates RSPEC-2131, what's the difference ? Thanks

=== on 10 Oct 2014, 17:55:39 Ann Campbell wrote:
\[~freddy.mallet] they go in opposite directions. This one is string-to-primitive. The other is primitive-to-string. S'okay?

=== on 11 Oct 2014, 12:13:20 Freddy Mallet wrote:
Ok [~ann.campbell.2] !

endif::env-github,rspecator-view[]
