== Why is this an issue?

Rather than creating a boxed primitive from a ``++String++`` to extract the primitive value, use the relevant ``++parse++`` method instead. It will be clearer and more efficient.


=== Noncompliant code example

[source,java]
----
String myNum = "12.2";

float f = (new Float(myNum)).floatValue();  // Noncompliant; creates & discards a Float
----


=== Compliant solution

[source,java]
----
String myNum = "12.2";

float f = Float.parseFloat(myNum);
----



ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
