This rule raises an issue when PyTorch tensors are copied using deprecated methods like `torch.tensor(existing_tensor)` or `tensor.new_tensor(existing_tensor)`.

== Why is this an issue?

PyTorch provides specific methods for copying existing tensors that are more efficient and express intent clearly. The methods `torch.tensor()` and `tensor.new_tensor()` were originally designed to create new tensors from raw data (like lists or arrays), not to copy existing tensor objects.

When you use these methods to copy existing tensors, PyTorch issues a `UserWarning` because the operation is not optimized for this use case. The tensor copying process becomes less efficient, and the code's intent becomes unclear to other developers.

The recommended approach is to use `tensor.detach().clone()` or `tensor.clone().detach()`. These methods clearly indicate that you want to:

* Create a copy of the tensor data (`clone()`)
* Remove the tensor from the computation graph (`detach()`)

This combination is specifically optimized for tensor copying operations and avoids the performance overhead of the deprecated methods.

=== What is the potential impact?

Using deprecated tensor copying methods can lead to:

* Performance degradation due to unoptimized copying operations
* UserWarnings cluttering your application logs
* Unclear code intent, making maintenance more difficult
* Potential compatibility issues in future PyTorch versions

== How to fix it

Replace deprecated tensor copying methods with the recommended `detach().clone()` pattern. If you need gradients, call `++requires_grad_()++` on the result.

=== Code examples

==== Noncompliant code example

[source,python,diff-id=1,diff-type=noncompliant]
----
import torch

x = torch.randn(3, 4, requires_grad=True)
y = torch.tensor(x, requires_grad=True)  # Noncompliant
----

==== Compliant solution

[source,python,diff-id=1,diff-type=compliant]
----
import torch

x = torch.randn(3, 4, requires_grad=True)
y = x.detach().clone().requires_grad_(True)
----

== Resources

=== Documentation

 * PyTorch Tensor Operations - https://pytorch.org/docs/stable/tensors.html[Official PyTorch documentation on tensor operations and methods]

 * PyTorch clone() method - https://pytorch.org/docs/stable/generated/torch.Tensor.clone.html[Documentation for the recommended clone() method]

 * PyTorch detach() method - https://pytorch.org/docs/stable/generated/torch.Tensor.detach.html[Documentation for the detach() method to remove tensors from computation graph]

=== Related rules

 * S7675 - Related rule about PyTorch tensor copying best practices
