include::../common/why.adoc[]

include::../common/how.adoc[]

=== Code examples

==== Noncompliant code example

[source,javascript,diff-id=1,diff-type=noncompliant]
----
function myButton() {
    return <button role="article">Click me!</button>; // Noncompliant; "button" is interactive, but "article" isn't
}
----

==== Compliant solution

[source,javascript,diff-id=1,diff-type=compliant]
----
function myButton() {
    return <button role="button">Click me!</button>;
}
----

include::../common/resources.adoc[]
