include::../common/why.adoc[]

include::../common/how.adoc[]

=== Code examples

==== Noncompliant code example

[source,javascript,diff-id=1,diff-type=noncompliant]
----
function myButton() {
    return <button role="article">Click me!</button>; // Noncompliant; "button" is interactive, but "article" isn't
}
----

==== Compliant solution

[source,javascript,diff-id=1,diff-type=compliant]
----
function myButton() {
    return <button role="button">Click me!</button>;
}
----

== Resources
=== Documentation

* WCAG - https://www.w3.org/WAI/WCAG21/Understanding/name-role-value[Name, Role, Value]
* WCAG - https://www.w3.org/TR/wai-aria-1.1/#usage_intro[WAI-ARIA Roles]
* MDN web docs - https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles[WAI-ARIA Roles]
