== Why is this an issue?

It is undefined behaviour if the ``++'++``, ``++"++``, ``++/*++`` or ``++//++`` characters are used between ``++<++`` and ``++>++`` delimiters or the ``++'++``, ``++/*++`` or ``++//++`` characters are used between the ``++"++`` delimiters in a header name preprocessing token.


=== Noncompliant code example

[source,text]
----
#include "fi'le.h" // Noncompliant; filename contains '
----


== Resources

* MISRA {cpp}:2008, 16-2-4 - The ', ", /* or // characters shall not occur in a header file name.

ifdef::env-github,rspecator-view[]

'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]

endif::env-github,rspecator-view[]
