It is undefined behaviour if the ``++'++``, ``++"++``, ``++/*++`` or ``++//++`` characters are used between ``++<++`` and ``++>++`` delimiters or the ``++'++``, ``++/*++`` or ``++//++`` characters are used between the ``++"++`` delimiters in a header name preprocessing token.


== Noncompliant Code Example

----
#include "fi'le.h" // Noncompliant; filename contains '
----


== See

* MISRA {cpp}:2008, 16-2-4 - The ', ", /* or // characters shall not occur in a header file name.

ifdef::env-github,rspecator-view[]
include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
