<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Read-only fields and properties (properties with only an auto-implemented getter) can only be set in a constructor or as part of their declaration. A read-only member that isn&#8217;t set in either place will retain its default value for the life of the object. At best, such properties clutter the source code. At worst, they are bugs.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-csharp" data-lang="csharp">class Person
{
  int Age { get; }  // Noncompliant; will always be 0.

  Person () {}
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-csharp" data-lang="csharp">class Person
{
  int Age { get; } = 42;

  Person () {}
}</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_11_may_2015_140833_dinesh_bolkensteyn_wrote">on 11 May 2015, 14:08:33 Dinesh Bolkensteyn wrote:</h3>
<div class="paragraph">
<p>LGTM - note that there seems to be a compiler warning for this already (it&#8217;s already reported in Visual Studio).</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_15_mar_2016_094841_tamas_vajk_wrote">on 15 Mar 2016, 09:48:41 Tamas Vajk wrote:</h3>
<div class="paragraph">
<p>\[~dinesh.bolkensteyn] Thanks, I&#8217;ve set the VS reference flag.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_7_apr_2016_145026_tamas_vajk_wrote">on 7 Apr 2016, 14:50:26 Tamas Vajk wrote:</h3>
<div class="paragraph">
<p>Could you check this RSPEC? I modified it from readonly fields to properties.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_7_apr_2016_171120_ann_campbell_wrote">on 7 Apr 2016, 17:11:20 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>Give it another read-through [~tamas.vajk]. I&#8217;ve made some changes.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_15_apr_2016_111337_tamas_vajk_wrote">on 15 Apr 2016, 11:13:37 Tamas Vajk wrote:</h3>
<div class="paragraph">
<p>Note: I added back the fields, because VS doesn&#8217;t report them as you type, it reports them only when you compile the code.</p>
</div>
</div>
</div>
</div>