<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Naming a thread won&#8217;t make it run faster or more reliably, but it will make it easier to deal with if you need to debug the application.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">Thread t1 = new Thread(new Runnable() {
  // ...
};
t1.start();   // Noncompliant; this thread wasn't named</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">Thread t1 = new Thread(new Runnable() {
  // ...
};
t1.setName("t1");
t1.start();</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Name this thread to ease debugging and profiling.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_11_jun_2015_193706_ann_campbell_wrote">on 11 Jun 2015, 19:37:06 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>CodePro: Disallow Unnamed Thread Usage</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_16_jun_2015_170938_nicolas_peru_wrote">on 16 Jun 2015, 17:09:38 Nicolas Peru wrote:</h3>
<div class="paragraph">
<p>Looks good.</p>
</div>
</div>
</div>
</div>