== Why is this an issue?

`null`-aware operators allow to safely access nested properties or methods of an object without having to check for the existence of each intermediate property manually. It provides a concise and safe way to access nested properties or methods without having to write complex and error-prone `null` checks.

This rule flags logical operations that can be safely replaced with the `?.` `null`-aware operator.

== How to fix it

Replace with `?.` `null`-aware operator the logical expression that checks for `null` before accessing the property of an object, the element of an array, or calling a function.

=== Code examples

==== Noncompliant code example

[source,dart,diff-id=1,diff-type=noncompliant]
----
void foo(a) {
    if (a != null) { // Noncompliant
        var x = a.value;
    }
}
----

==== Compliant solution

[source,dart,diff-id=1,diff-type=compliant]
----
void foo(a) {
    var x = a?.value;
}
----

== Resources

* https://dart.dev/tools/linter-rules/prefer_null_aware_operators[Dart Lint rule]
