<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when a variable is declared in an if short statement but is only used in the condition itself, suggesting the declaration might be unnecessary.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Go&#8217;s if short statement allows you to declare and initialize a variable before the condition check. This feature is designed to limit the variable&#8217;s scope to the if-else block, which is useful when you need the variable in multiple parts of the conditional logic.</p>
</div>
<div class="paragraph">
<p>However, if the declared variable is only used in the condition and nowhere else in the if or else blocks, the short statement declaration adds unnecessary complexity. In such cases, you can often simplify the code by moving the expression directly into the condition or restructuring the logic.</p>
</div>
<div class="paragraph">
<p>Using if short statements appropriately helps create cleaner, more readable code by:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>Limiting variable scope to where it&#8217;s actually needed</p>
</li>
<li>
<p>Reducing the number of variables in the broader function scope</p>
</li>
<li>
<p>Making the relationship between variable declaration and usage more explicit</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>When a variable is declared but underutilized, it can confuse readers about the intended purpose and scope of the variable.</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="paragraph">
<p>This issue primarily affects code readability and maintainability. While it doesn&#8217;t cause runtime problems, it can make code harder to understand and maintain. Unnecessary variable declarations can also slightly impact performance in tight loops, though this is typically negligible.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Move the expression directly into the condition when the variable is only used there.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-go" data-lang="go">if x := getValue(); x &gt; 10 { // Noncompliant: x is only used in condition
    fmt.Println("High value")
} else {
    fmt.Println("Low value")
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-go" data-lang="go">if getValue() &gt; 10 {
    fmt.Println("High value")
} else {
    fmt.Println("Low value")
}</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Go Tour - If with a short statement - <a href="https://go.dev/tour/flowcontrol/6">Official Go documentation explaining if short statements</a></p>
</li>
<li>
<p>Go Tour - If and else - <a href="https://go.dev/tour/flowcontrol/7">Official Go documentation on variable scope in if-else blocks</a></p>
</li>
<li>
<p>Effective Go - Control structures - <a href="https://go.dev/doc/effective_go#control-structures">Best practices for Go control structures including if statements</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>