[#floating-point-inc-dec, severity=major, tags=maintainability,pitfall,floating-point]
=== Increment and decrement operators shouldn't be used with floating-point variables

Increment and decrement operators (`{plus}{plus}` and `--`) shouldn't be used with floating-point variables (like `float` or `double`).
While the language allows it, the usage is not idiomatic, and most developers intuitively expect `x{plus}{plus}` to apply to integer types.
Using it on a float violates this common expectation and can lead to misleading code.




// If you want to factorize the description uncomment the following line and create the file.
//include::../description.adoc[]

== Why is this an issue?

Floating-point arithmetic has some non-intuitive properties, which can lead to unexpected bugs.
For example, the following loop will not terminate:

[source,java]
----
for (float x = 16_000_000; x < 17_000_000; x++) {
  // ...
}
// The loop does not terminate
----

This happens because `float` has only 24 bits of precision (mantissa) and once a number gets large enough, adding `1.0` becomes insignificant.
The increment operation effectively does nothing.

The problem would not occur if `int` is used instead of `float`, even though both types occupy 32 bits:

[source,java]
----
for (int x = 16_000_000; x < 17_000_000; x++) {
  // ...
}
// This loop terminates.
----

Using the compound assignment operators (`{plus}=` and `-=`) makes the intent clearer and avoids the surprising use of `{plus}{plus}` and `--` on floating-point types.

== How to fix it
//== How to fix it in FRAMEWORK NAME

=== Code examples

==== Noncompliant code example


[source,java]
----
x++;
y--;
----

==== Compliant solution

[source,java]
----
x += 1.0;
y -= 1.0;
----

//=== How does this work?

//=== Pitfalls

//=== Going the extra mile


//== Resources
//=== Documentation
//=== Articles & blog posts
//=== Conference presentations
//=== Standards
//=== External coding guidelines
//=== Benchmarks
