This rule checks that the specified child elements are present inside the specified parent elements. 


== Noncompliant Code Example

Given a parent/child combination of <code><head></code>/<code><title></code>:
----
<html>
  <head>
  </head>  <!-- Noncompliant; no title element -->
  <body>
    ...
----


== Compliant Solution

----
<html>
  <head>
    <title>My Page</title>
  </head>
  <body>
    ...
----

