<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>C&#43;&#43;23 introduced delimited escape sequences like <code>\x{40}</code> that improve readability and are less error-prone. They should be preferred.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Escape sequences can be used to designate a single character literal or characters in a string literal. Escape sequences followed by a character code are especially handy when the character is non-visible or is not directly representable in the source code encoding. However, before C&#43;&#43;23 the way to write them is convoluted. Let&#8217;s take, for instance, the character <code>@</code>:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>In octal, it is written <code>\100</code>. This notation has two main issues: It&#8217;s not easy to spot octal literals (it&#8217;s just some digits, nothing makes it obvious that they are <em>octal</em> digits), and it&#8217;s not easy to say when the escape sequence ends (see <a data-rspec-id="S2335" class="rspec-auto-link">S2335</a>). This is so confusing that it is commonly advised to avoid octal escape sequences (see <a data-rspec-id="S1314" class="rspec-auto-link">S1314</a>).</p>
</li>
<li>
<p>Hexadecimal sequences are written <code>\x40</code>. While they stand out more prominently, it&#8217;s still difficult to say when they end (see also <a data-rspec-id="S2335" class="rspec-auto-link">S2335</a>).</p>
</li>
<li>
<p>Universal character names, while not strictly speaking escape sequences, have a very similar visual appearance and are written <code>\u0040</code>(exactly 4 hexadecimal digits) or <code>\U00000040</code> (exactly 8 hexadecimal digits). The fact that the number of digits depends on the case of the introducing <code>u</code> can be a source of errors.</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>C&#43;&#43;23 introduces a new way to define those escape sequences, which is more consistent, non-ambiguous, and makes the ending of the literal obvious. It is now possible to write the previous examples respectively <code>\o{100}</code>, <code>\x{40}</code>, <code>\u{40}</code> and <code>\u{40}</code> again (leading zeros are optional).</p>
</div>
<div class="paragraph">
<p>This rule raises an issue when a legacy escape sequence can be replaced with an improved version.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">int main() {
   std::cout &lt;&lt; "\100\x40\u0040\U00000040\n";
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">int main() {
   std::cout &lt;&lt; "\o{100}\x{40}\u{40}\u{40}\n";
}</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>C&#43;&#43; reference - <a href="https://en.cppreference.com/w/cpp/language/escape">Escape sequences</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>