This rule raises an issue when ``++*sql.Rows++`` objects returned from database query operations are not explicitly closed.

== Why is this an issue?

When executing database queries that return multiple rows, Go's ``++database/sql++`` package returns a ``++*sql.Rows++`` object that holds database resources including network connections and server-side cursors.

While the Go standard library does provide automatic cleanup in certain scenarios (such as when ``++rows.Next()++`` returns false and there are no additional result sets), relying solely on this automatic behavior can lead to resource leaks in several situations:

* If an error occurs during row processing and the iteration loop exits early
* If the code returns from the function before fully consuming all rows
* If an exception or panic occurs during row processing
* When working with multiple result sets where automatic cleanup may not occur

These unclosed resources can accumulate over time, leading to:

* Database connection pool exhaustion
* Memory leaks on both client and server sides
* Degraded application performance
* Potential database server resource starvation

Explicit resource management through ``++rows.Close()++`` ensures that database resources are freed immediately and reliably, regardless of how the function exits or whether all rows are consumed.

=== What is the potential impact?

Failing to close database rows can lead to resource leaks that accumulate over time. This can result in database connection pool exhaustion, memory leaks, and degraded application performance. In high-traffic applications, this can eventually cause the application to become unresponsive or crash due to resource starvation.

== How to fix it

Add an explicit call to ``++rows.Close()++`` immediately after checking for query errors. The most common and recommended pattern is to use ``++defer rows.Close()++`` which ensures the rows are closed when the function exits, regardless of how it exits.

=== Code examples

==== Noncompliant code example

[source,go,diff-id=1,diff-type=noncompliant]
----
func getAllUsers(db *sql.DB) error {
    rows, err := db.Query("SELECT * FROM users")
    if err != nil {
        return err
    }
    // Missing rows.Close() // Noncompliant
    for rows.Next() {
        // process rows
    }
    return rows.Err()
}
----

==== Compliant solution

[source,go,diff-id=1,diff-type=compliant]
----
func getAllUsers(db *sql.DB) error {
    rows, err := db.Query("SELECT * FROM users")
    if err != nil {
        return err
    }
    defer rows.Close()
    for rows.Next() {
        // process rows
    }
    return rows.Err()
}
----

== Resources

=== Documentation

 * Go database/sql package documentation - https://pkg.go.dev/database/sql[Official Go documentation for the database/sql package, including details about the Rows type and Close method]

 * Go database/sql tutorial - https://go.dev/doc/tutorial/database-access[Official Go tutorial on accessing databases, demonstrating proper resource management patterns]
