This rule raises an issue when `Jenkins.instance.getItem()` is called with a path that contains folder separators (forward slashes).

== Why is this an issue?

The `getItem()` method in Jenkins only works for jobs at the root level. When you pass a folder path like `'folder_name/job_name'` to `getItem()`, it returns `null` instead of the expected job object.

This happens because `getItem()` treats the entire string as a single job name, not as a path. It doesn't understand folder hierarchies and won't traverse into folders to find the job.

When `getItem()` returns `null` for a job that actually exists in a folder, your script may:

* Throw a null pointer exception when trying to access properties of the job
* Skip important operations because it thinks the job doesn't exist
* Fail silently, making debugging difficult

Jenkins provides `getItemByFullName()` specifically for accessing items using their full path, including folders. This method correctly handles folder hierarchies and returns the job object when it exists.

=== What is the potential impact?

Scripts may fail with null pointer exceptions or behave unexpectedly when they cannot find jobs that exist in folders. This can lead to incomplete automation, failed deployments, or silent failures that are difficult to debug.

== How to fix it

Replace `getItem()` with `getItemByFullName()` when accessing jobs in folders. The `getItemByFullName()` method correctly handles folder paths and will return the job object if it exists.

=== Code examples

==== Noncompliant code example

[source,groovy,diff-id=1,diff-type=noncompliant]
----
// This returns null for jobs in folders
def job = Jenkins.instance.getItem('folder_name/job_name') // Noncompliant
if (job != null) {
    echo "Job found: ${job.name}"
}
----

==== Compliant solution

[source,groovy,diff-id=1,diff-type=compliant]
----
// Use getItemByFullName for jobs in folders
def job = Jenkins.instance.getItemByFullName('folder_name/job_name')
if (job != null) {
    echo "Job found: ${job.name}"
}
----

== Resources

=== Documentation

 * Jenkins API Documentation - getItemByFullName - https://javadoc.jenkins.io/jenkins/model/Jenkins.html#getItemByFullName-java.lang.String-[Official Jenkins API documentation for the getItemByFullName method]

 * Jenkins Script Console Documentation - https://www.jenkins.io/doc/book/managing/script-console/[Guide to using the Jenkins script console for automation]
