== Why is this an issue?

Hashing a unit value doesn't accomplish anything because the implementation of `Hash` for `()` is a no-op. This can lead to confusion or misleading code, as it implies some hashing operation is actually taking place, when in reality it does nothing.

=== Code examples

==== Noncompliant code example

[source,rust,diff-id=1,diff-type=noncompliant]
----
# use std::hash::Hash;
# use std::collections::hash_map::DefaultHasher;
# enum Foo { Empty, WithValue(u8) }
# use Foo::*;
# let mut state = DefaultHasher::new();
# let my_enum = Foo::Empty;
match my_enum {
    Empty => ().hash(&mut state), // Noncompliant
    WithValue(x) => x.hash(&mut state),
}
----

==== Compliant solution

[source,rust,diff-id=1,diff-type=compliant]
----
# use std::hash::Hash;
# use std::collections::hash_map::DefaultHasher;
# enum Foo { Empty, WithValue(u8) }
# use Foo::*;
# let mut state = DefaultHasher::new();
# let my_enum = Foo::Empty;
match my_enum {
    Empty => 0_u8.hash(&mut state), // Compliant
    WithValue(x) => x.hash(&mut state),
}
----

== Resources
=== Documentation

* Clippy Lints - https://rust-lang.github.io/rust-clippy/master/index.html#unit_hash
