include::../description.adoc[]

== Noncompliant Code Example

----
public bool IsOdd(int x) 
{
  return x % 2 == 1;  // Noncompliant; if x is an odd negative, x % 2 == -1
}
----

== Compliant Solution

----
public bool IsOdd(int x) 
{
  return x %2 != 0;
}
----

or

----
public bool IsOdd(uint x) 
{
  return x %2 == 1;
}
----

include::../see.adoc[]
