include::../description.adoc[]

== Noncompliant Code Example

[source,vb6]
----
sampleString  = "Hello World" : MsgBox(sampleString)
----

== Compliant Solution

[source,vb6]
----
sampleString  = "Hello World"
MsgBox(sampleString)
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../highlighting.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
