<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when <code>unless</code> is used with an <code>else</code> clause, or when <code>if !</code> or <code>if not</code> is used for simple negative conditions without an <code>else</code> clause.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Ruby&#8217;s <code>unless</code> statement is designed to make negative conditions more readable, but it can create confusion when misused.</p>
</div>
<div class="paragraph">
<p>When <code>unless</code> is used with an <code>else</code> clause, it creates double negative logic that is hard to understand. The <code>else</code> clause of an <code>unless</code> statement represents the positive case, which is counterintuitive. For example, <code>unless user.admin? &#8230;&#8203; else &#8230;&#8203;</code> means "if the user is not an admin, do this, otherwise (if they are an admin) do that." This mental gymnastics makes code harder to read and maintain.</p>
</div>
<div class="paragraph">
<p>On the other hand, when checking simple negative conditions without an <code>else</code> clause, using <code>if !</code> or <code>if not</code> instead of <code>unless</code> makes the code less idiomatic and slightly harder to read. Ruby&#8217;s <code>unless</code> keyword was specifically created to handle these cases more elegantly.</p>
</div>
<div class="paragraph">
<p>The key principle is: use <code>unless</code> for simple negative conditions, but switch to <code>if !</code> when you need an <code>else</code> clause to maintain clarity.</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="paragraph">
<p>This issue affects code readability and maintainability. Confusing conditional logic can lead to bugs when developers misunderstand the intended behavior, especially during code reviews or when maintaining unfamiliar code. While not a security or performance issue, it can slow down development and increase the likelihood of logical errors.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Replace <code>unless</code> with <code>else</code> clause with an equivalent <code>if</code> statement to eliminate double negative logic.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-ruby" data-lang="ruby">unless user.staff?
  raise SecurityError, 'not authorized'
else  # Noncompliant
  perform_admin_tasks
  generate_reports
end</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-ruby" data-lang="ruby">if user.staff?
  perform_admin_tasks
  generate_reports
else
  raise SecurityError, 'not authorized'
end</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Ruby Style Guide - Conditional Expressions - <a href="https://rubystyle.guide/#unless-for-negatives">Official Ruby style guide recommendations for using unless statements</a></p>
</li>
<li>
<p>Ruby Documentation - Control Expressions - <a href="https://docs.ruby-lang.org/en/master/syntax/control_expressions_rdoc.html">Ruby&#8217;s official documentation on control flow expressions including unless</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>