<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Turning on the <code>*srcstmt</code> header option means that any line numbers cited in error statements will actually correspond correctly to the code. Otherwise, object line numbers will be shown, making errors difficult to debug.</p>
</div>
<div class="paragraph">
<p>The <code>*srcstmt</code> option is also recommended for accurate line numbers during debugging, particularly in combination with <code>*nodebugio</code>, (<code>H option(*srcstmt:*nodebugio)</code>) which prevents debug operations from stopping on I/O.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-rpg" data-lang="rpg">H*
H* AMENDED BY : G. Ann Campbell
H* DATE       : 16JUL2014
H* H24 - Miscellaneous Online Maintenance/Enquiry/Printing
H*
F/EJECT</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-rpg" data-lang="rpg">H*
H* AMENDED BY : G. Ann Campbell
H* DATE       : 16JUL2014
H* H24 - Miscellaneous Online Maintenance/Enquiry/Printing
H*
H option(*srcstmt)
F/EJECT</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Add "H option (*srcstmt)" to this file.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_31_jul_2014_152406_freddy_mallet_wrote">on 31 Jul 2014, 15:24:06 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] If my feeling is correct, there is a typo in the following sentence. I guess you wanted to say "won&#8217;t be shown" and not "will be shown" ?</p>
</div>
<div class="paragraph">
<p><em><em>_Otherwise, object line numbers will be shown, making errors difficult to debug.</em>_</em></p>
</div>
</div>
<div class="sect2">
<h3 id="_on_31_jul_2014_185500_ann_campbell_wrote">on 31 Jul 2014, 18:55:00 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~freddy.mallet] nope. At issue is whether source line #s are shown versus object line #s.</p>
</div>
</div>
</div>
</div>