Creating an extension method that extends ``++Object++`` is not recommended because it makes the method available on _every_ type. Extensions should be applied at the most specialized level possible, and that is very unlikely to be ``++Object++``.

== Noncompliant Code Example

[source,vbnet]
----
Imports System.Runtime.CompilerServices

Module MyExtensions
    <Extension>
    Sub SomeExtension(obj As Object) ' Noncompliant
        ' ...
    End Sub
End Module
----

include::../rspecator.adoc[]