<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Calling <code>DELETE ADJACENT DUPLICATES</code> won&#8217;t reliably do any good if the table hasn&#8217;t first been sorted to put duplicates side by side, since the <code>ADJACENT</code> part of the command looks for multiple rows side-by-side with the same content.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-abap" data-lang="abap">DELETE ADJACENT DUPLICATES FROM ITAB COMPARING LAND.</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-abap" data-lang="abap">SORT ITAB BY LAND.
DELETE ADJACENT DUPLICATES FROM ITAB COMPARING LAND.</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Sort this table before the delete</p>
</div>
</div>
</div>
</div>