This rule raises an issue when an incorrect value is set as an attribute of `datetime.date`, `datetime.time`, or `datetime.datetime`

== Why is this an issue?

Setting of a date atrribute value to a value which is out of the possible values range will lead to the `ValueError`.

== How to fix it
Set attribute values which are in possible values range.

=== Code examples

==== Noncompliant code example

[source,python,diff-id=1,diff-type=noncompliant]
----
def foo():
    return = datetime(year=2024, day=66, month=1, hour=16, minute=1) # ValueError: day is out of range for month
----

==== Compliant solution

[source,python,diff-id=1,diff-type=compliant]
----
def foo():
    dt = datetime(year=2024, day=1, month=1, hour=16, minute=1)
----

//=== How does this work?

//=== Pitfalls

//=== Going the extra mile


== Resources
=== Documentation
* Python documentation - https://docs.python.org/3/library/datetime.html#[daytime]
//=== Articles & blog posts
//=== Conference presentations
//=== Standards
//=== External coding guidelines
//=== Benchmarks
