Shared coding conventions allow teams to collaborate efficiently. This rule checks that first level data item names match a provided regular expression.


== Noncompliant Code Example

Given an regular expression of ``++WS-.*++``:

----
       WORKING-STORAGE SECTION.
            01 WRONG.                                       > Noncompliant; name doesn't match the pattern "WS-.*"
              02  LINK.                                     > Compliant; this is not first level

       LINKAGE SECTION.
             01     DFHCOMMAREA PIC X(1500).                > Compliant; the data item is defined in the LINKAGE SECTION
----


== Compliant Solution

----
       WORKING-STORAGE SECTION.
            01 WS-LINK.
              02  LINK.

       LINKAGE SECTION.
             01     DFHCOMMAREA PIC X(1500).
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::parameters.adoc[]

endif::env-github,rspecator-view[]
