<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Unused cursor parameters are misleading. Whatever the values passed to such parameters, the behavior will be the same.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-sql" data-lang="sql">cursor c_list_emp(pp_country varchar2, pp_status varchar2)  is -- Noncompliant pp_status is not used
   select e.employee_code,
          p.first_name,
          p.last_name,
          e.country
    from persons       p,
    join employee_list e on e.person_id = p.person_id
   where e.country = pp_country;</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-sql" data-lang="sql">cursor c_list_emp(pp_country varchar2, pp_status varchar2)  is
   select e.employee_code,
          p.first_name,
          p.last_name,
          e.country
    from persons       p,
    join employee_list e on e.person_id = p.person_id
   where e.country     = pp_country
     and e.status_code = pp_status; -- use the parameter</code></pre>
</div>
</div>
<div class="paragraph">
<p>or</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-sql" data-lang="sql">cursor c_list_emp(pp_country varchar2)  is -- Remove the parameter
   select e.employee_code,
          p.first_name,
          p.last_name,
          e.country
    from persons       p,
    join employee_list e on e.person_id = p.person_id
   where e.country = pp_country;</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove this unused parameter "{}".</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>The unused parameter</p>
</div>
</div>
</div>
</div>