This rule raises an issue when a method uses an explicit `return` keyword for the last expression that could be returned implicitly.

== Why is this an issue?

In Groovy, the last expression in a method is automatically returned, making the explicit `return` keyword redundant in many cases. This is one of Groovy's key features that distinguishes it from Java and makes code more concise.

Using unnecessary `return` keywords:

* Makes code more verbose than needed
* Goes against Groovy coding conventions and idioms
* Reduces readability by adding visual clutter
* Makes the code look more Java-like rather than taking advantage of Groovy's features

Groovy's implicit return feature is particularly beneficial for:

* Single-expression methods
* Simple getter methods
* Methods that end with a single calculation or operation
* Closures and short utility methods

=== What is the potential impact?

This issue has minimal impact on functionality but affects code maintainability and readability. Removing unnecessary `return` keywords makes the codebase more consistent with Groovy conventions and easier to read for developers familiar with the language.

== How to fix it

Remove the `return` keyword from the last expression in the method. The last expression will be automatically returned by Groovy.

=== Code examples

==== Noncompliant code example

[source,groovy,diff-id=1,diff-type=noncompliant]
----
def getName() {
    return 'John' // Noncompliant
}
----

==== Compliant solution

[source,groovy,diff-id=1,diff-type=compliant]
----
def getName() {
    'John'
}
----

== Resources

=== Documentation

 * Groovy Language Documentation - Methods - https://groovy-lang.org/semantics.html#_return_keyword[Official Groovy documentation explaining implicit returns and when the return keyword is optional]

 * Groovy Style Guide - https://groovy-lang.org/style-guide.html[Official style guide with recommendations for idiomatic Groovy code]

=== Standards

 * Groovy Best Practices - https://groovy-lang.org/style-guide.html#_return_keyword[Groovy community guidelines for writing clean, idiomatic code]

=== Related rules

 * RSPEC-1172 - https://rules.sonarsource.com/java/RSPEC-1172/[Unused method parameters should be removed (related to code simplification)]
