This rule raises an issue when the same variable is compared against multiple constant values using separate if or if-else statements.

== Why is this an issue?

When you need to check the same variable against multiple constant values, using separate if statements creates code that is harder to read and maintain.

Consider this example:

[source,go]
----
if status == "pending" {
    // handle pending
}
if status == "approved" {
    // handle approved  
}
if status == "rejected" {
    // handle rejected
}
----

This pattern has several problems:

* **Poor readability**: The intent to handle different cases of the same variable is not immediately clear
* **Maintenance burden**: Adding new cases requires writing more if statements
* **Missing error handling**: There's no clear way to handle unexpected values
* **Code duplication**: Each if statement repeats the variable name

A switch statement makes the intent much clearer. It shows that you're handling different cases of the same variable, making the code easier to understand and maintain.

Switch statements also encourage better practices like using named constants instead of hardcoded strings and adding default cases for error handling.

=== What is the potential impact?

Using multiple if statements instead of a switch statement reduces code readability and makes maintenance more difficult. It can also lead to bugs when developers forget to handle all possible cases or when adding new cases requires updating multiple locations in the code.

== How to fix it

Replace the multiple if statements with a single switch statement. Use named constants for string literals and add a default case to handle unexpected values.

=== Code examples

==== Noncompliant code example

[source,go,diff-id=1,diff-type=noncompliant]
----
if h.Key == "GOHLAY" {
    // handle GOHLAY
}
if h.Key == "GOHLAY_DELIVERED" {
    // handle GOHLAY_DELIVERED  // Noncompliant
}
if h.Key == "GOHLAY_FAILED" {
    // handle GOHLAY_FAILED
}
----

==== Compliant solution

[source,go,diff-id=1,diff-type=compliant]
----
const (
    keyGohlay = "GOHLAY"
    keyGohlayDelivered = "GOHLAY_DELIVERED"
    keyGohlayFailed = "GOHLAY_FAILED"
)

switch h.Key {
case keyGohlay:
    // handle GOHLAY
case keyGohlayDelivered:
    // handle GOHLAY_DELIVERED
case keyGohlayFailed:
    // handle GOHLAY_FAILED
default:
    return fmt.Errorf("unhandled key: %s", h.Key)
}
----

== Resources

=== Documentation

 * Go Language Specification - Switch statements - https://go.dev/ref/spec#Switch_statements[Official Go documentation on switch statement syntax and behavior]

 * Effective Go - Switch - https://go.dev/doc/effective_go#switch[Best practices for using switch statements in Go]

=== Related rules

 * RSPEC-6880 - https://rules.sonarsource.com/java/RSPEC-6880/[Java rule for replacing if-else chains with switch expressions]
