include::../description.adoc[]

== Noncompliant Code Example

[source,java]
----
public void doTheThing() {
  try {
    // ...
  catch (IOException e) {  // Noncompliant
  }
}
----

== Compliant Solution

[source,java]
----
public void doTheThing() throws IOException {
  // ...
}
----
or

[source,java]
----
public void doTheThing() {
  try {
    // ...
  catch (IOException e) {
    LOGGER.info(e);
  }
}
----
or

[source,java]
----
public void doTheThing() {
  try {
    // ...
  catch (IOException e) {
    // comment explaining why taking no action is appropriate
  }
}
----

== Exceptions

When a block contains a comment, it is not considered to be empty.

include::../see.adoc[]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
