include::../description.adoc[]


== Noncompliant Code Example

----
re.sub(r"[c̈d̈]", "X", "cc̈d̈d") # Noncompliant, print "XXXXXX" instead of expected "cXXd".
----


== Compliant Solution

----
re.sub(r"c̈|d̈", "X", "cc̈d̈d") # print "cXXd"
----
