== Why is this an issue?

This variant of the ``++REFRESH++`` statement is deprecated and should be avoided.


This ``++REFRESH++`` statement initializes the internal table ``++itab++``, reads several rows from the database table ``++dbtab++``, and adds their contents to the internal table ``++itab++``. A ``++SELECT++`` statement should be used instead.


=== Noncompliant code example

[source,abap]
----
TABLES t100. 
DATA itab TYPE STANDARD TABLE OF t100.

t100-sprsl = 'E'. 
t100-arbgb = 'BC'.

REFRESH itab FROM TABLE t100.
----


=== Compliant solution

[source,abap]
----
DATA itab TYPE STANDARD TABLE OF t100.

SELECT * 
       FROM t100 
       INTO TABLE itab 
       WHERE sprsl = 'E' AND 
             arbgb LIKE 'BC%'. 
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
