=== Code examples

==== Noncompliant code example

[source,vbnet,diff-id=1,diff-type=noncompliant]
----
Function StartsWithA(s As String) As Boolean
    Return s.StartsWith("A")
End Function
----

[source,vbnet,diff-id=2,diff-type=noncompliant]
----
Function EndsWithA(s As String) As Boolean
    Return s.EndsWith("A")
End Function
----

==== Compliant solution

[source,vbnet,diff-id=1,diff-type=compliant]
----
Function StartsWithA(s As String) As Boolean
    Return s.StartsWith("A"c)
End Function
----

[source,vbnet,diff-id=2,diff-type=compliant]
----
Function EndsWithA(s As String) As Boolean
    Return s.EndsWith("A"c)
End Function
----
