=== Code examples

==== Noncompliant code example

[source,vbnet,diff-id=1,diff-type=noncompliant]
----
Function StartsWithSlash(s As String) As Boolean
    Return s.StartsWith("/")
End Function
----

[source,vbnet,diff-id=2,diff-type=noncompliant]
----
Function EndsWithSlash(s As String) As Boolean
    Return s.EndsWith("/")
End Function
----

==== Compliant solution

[source,vbnet,diff-id=1,diff-type=compliant]
----
Function StartsWithSlash(s As String) As Boolean
    Return s.StartsWith("/"c)
End Function
----

[source,vbnet,diff-id=2,diff-type=compliant]
----
Function EndsWithSlash(s As String) As Boolean
    Return s.EndsWith("/"c)
End Function
----
