include::../description.adoc


== Noncompliant Code Example

----
class FooIterator : Iterator<Foo>, Iterable<Foo> {

    override fun iterator(): Iterator<Foo> {
        return this // Noncompliant
    } 
    // ...
}
----


== Compliant Solution

----
class FooSequence : Iterable<Foo> {

    override fun iterator(): Iterator<Foo> {
        return object : Iterator<Foo> {
            override fun hasNext(): Boolean {
                //...
            }

            override fun next(): Foo {
                //...
            }
        }
    }
    //...
}
----

