<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Integer literals starting with a zero are octal rather than decimal values. While using octal values is fully supported, most developers do not have experience with them. They may not recognize octal values as such, mistaking them instead for decimal values.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-flex" data-lang="flex">var myNumber:int = 010;  // Noncompliant. myNumber will hold 8, not 10 - was this really expected?</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-flex" data-lang="flex">var myNumber:int = 8;</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Use decimal rather than octal values.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_is_duplicated_by_s815">is duplicated by: <a data-rspec-id="S815" class="rspec-auto-link">S815</a></h3>

</div>
<div class="sect2">
<h3 id="_on_9_oct_2013_162239_freddy_mallet_wrote">on 9 Oct 2013, 16:22:39 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>Is implemented by http://jira.codehaus.org/browse/SONARJAVA-344</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_9_oct_2013_221300_ann_campbell_wrote">on 9 Oct 2013, 22:13:00 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>Why limit this rule to Java? Isn&#8217;t it just as valid in C/C&#43;&#43;/C#, PHP, &amp;etc?</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_30_oct_2013_153751_freddy_mallet_wrote">on 30 Oct 2013, 15:37:51 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>Is implemented by http://jira.codehaus.org/browse/SONARPLUGINS-3216 for Flex</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_14_feb_2018_113352_alban_auzeill_wrote">on 14 Feb 2018, 11:33:52 Alban Auzeill wrote:</h3>
<div class="paragraph">
<p>About support of octal literals in languages:</p>
</div>
<div class="paragraph">
<p>Not supported: ABAP, PL/SQL, RPG, T-SQL</p>
</div>
<div class="paragraph">
<p>Supported without confusion, e.g. decimal value 8 = Cobol(O#10), PL/I('10&#8217;B3), Swift (0o10), TypeScript(0o10), VB.Net(&amp;O10), VB6(&amp;O10)</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_28_oct_2019_145627_nicolas_harraudeau_wrote">on 28 Oct 2019, 14:56:27 Nicolas Harraudeau wrote:</h3>
<div class="paragraph">
<p>This rule has a low value as python 3 as the syntax is clearly different from decimal numbers (ex: 0o555). It is more difficult to confuse it with a decimal number.</p>
</div>
</div>
</div>
</div>