== How to fix it in Python Standard Library

=== Code examples

include::../../common/fix/code-rationale.adoc[]

==== Noncompliant code example

[source,python,diff-id=1,diff-type=noncompliant]
----
parser = xml.sax.make_parser()
myHandler = MyHandler()
parser.setContentHandler(myHandler)
parser.setFeature(feature_external_ges, True) # Noncompliant
parser.parse('xxe.xml')
----

==== Compliant solution

[source,python,diff-id=1,diff-type=compliant]
----
parser = xml.sax.make_parser()
myHandler = MyHandler()
parser.setContentHandler(myHandler)
# The SAX parser does not process general external entities by default since version 3.7.1
parser.setFeature(feature_external_ges, False)
parser.parse('xxe.xml')
----

=== How does this work?

include::../../common/fix/validation.adoc[]
