This rule allows banning usage of certain constructors.


== Noncompliant Code Example

Given parameters:

* className: java.util.Date
* argumentTypes: java.lang.String

----
Date birthday;
birthday = new Date("Sat Sep 27 05:42:21 EDT 1986");  // Noncompliant
birthday = new Date(528176541000L); // Compliant
----


ifdef::rspecator-view[]
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::rspecator-view[]
