FIXME: add a description

// If you want to factorize the description uncomment the following line and create the file.
//include::../description.adoc[]

== Why is this an issue?

FIXME: remove the unused optional headers (that are commented out)

//=== What is the potential impact?

=== Exception

This rule does not raise when explicitly creating a string from `c_str()`.

[source,cpp]
----
void takeString(std::string const& dest);

void func(std::string const& src) {
  takeString(std::string(src.c_str())); // Compliant by exception
}
----

This operation could be done on purpose in the rare case where `src` is a string containing null-terminators and we want to only keep the content up to the first null-terminator in `dst`.

== How to fix it
//== How to fix it in FRAMEWORK NAME

=== Code examples

==== Noncompliant code example

[source,cpp,diff-id=1,diff-type=noncompliant]
----
FIXME
----

==== Compliant solution

[source,cpp,diff-id=1,diff-type=compliant]
----
FIXME
----

//=== How does this work?

//=== Pitfalls

//=== Going the extra mile


//== Resources
//=== Documentation
//=== Articles & blog posts
//=== Conference presentations
//=== Standards
//=== External coding guidelines
//=== Benchmarks
