<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>When a back reference in a regex refers to a capturing group that hasn&#8217;t been defined yet (or at all), it can never be matched and will  fail with an <code>re.error</code> exception</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">import re
pattern1 = re.compile(r"\1(.)") # Noncompliant, group 1 is defined after the back reference
pattern2 = re.compile(r"(.)\2") # Noncompliant, group 2 isn't defined at all
pattern3 = re.compile(r"(.)|\1") # Noncompliant, group 1 and the back reference are in different branches
pattern4 = re.compile(r"(?P&lt;x&gt;.)|(?P=x)") # Noncompliant, group x and the back reference are in different branches</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">import re
pattern1 = re.compile(r"(.)\1")
pattern2 = re.compile(r"(?P&lt;x&gt;.)(?P=x)")</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Change this back reference, so that it refers to a group that can be matched before it.</p>
</div>
<div class="paragraph">
<p>Or:</p>
</div>
<div class="paragraph">
<p>Fix this back reference - it refers to a capturing group that doesn&#8217;t exist.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="ulist">
<ul>
<li>
<p>Primary: The impossible back reference.</p>
</li>
<li>
<p>Secondary: The capturing group to which it refers (if any)</p>
</li>
</ul>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_deprecates_s5840">deprecates: <a data-rspec-id="S5840" class="rspec-auto-link">S5840</a></h3>

</div>
</div>
</div>