<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when a class implements the <code>ASTTransformation</code> interface but is not annotated with <code>@CompileStatic</code>.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>AST transformations are executed during the compilation process and can significantly impact build performance. The Groovy documentation explicitly recommends using <code>@CompileStatic</code> for AST transformation implementations because it improves compiler performance.</p>
</div>
<div class="paragraph">
<p>When AST transformations are not compiled statically, they rely on Groovy&#8217;s dynamic features during compilation, which adds overhead. Since AST transformations can be invoked many times during a single compilation (potentially once per source file for global transformations, or once per annotation occurrence for local transformations), this performance overhead can accumulate and noticeably slow down build times.</p>
</div>
<div class="paragraph">
<p>Using <code>@CompileStatic</code> eliminates this dynamic overhead by ensuring the transformation code itself is optimized at compile time, leading to faster execution when the transformation is applied.</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="paragraph">
<p>Without <code>@CompileStatic</code>, AST transformations will execute more slowly during compilation, leading to increased build times. This performance impact becomes more significant in larger codebases or when transformations are applied frequently. The overhead can accumulate across multiple transformation invocations, making builds noticeably slower.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Add the <code>@CompileStatic</code> annotation to the AST transformation class.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-groovy" data-lang="groovy">@GroovyASTTransformation(phase=CompilePhase.SEMANTIC_ANALYSIS)
class MyTransformation implements ASTTransformation {
    void visit(ASTNode[] nodes, SourceUnit sourceUnit) {
        // transformation logic
    }
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-groovy" data-lang="groovy">@CompileStatic
@GroovyASTTransformation(phase=CompilePhase.SEMANTIC_ANALYSIS)
class MyTransformation implements ASTTransformation {
    void visit(ASTNode[] nodes, SourceUnit sourceUnit) {
        // transformation logic
    }
}</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Groovy documentation - <a href="https://docs.groovy-lang.org/latest/html/documentation/core-semantics.html#compilestatic-annotation">The @CompileStatic annotation</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>