include::../description.adoc[]

include::../ask-yourself.adoc[]

include::../recommended.adoc[]

== Noncompliant Code Example

Update permission is granted for all policies using the wildcard (*) in the ``++Resource++`` property:

----
MyPolicy:
    Type: AWS::IAM::ManagedPolicy
    Properties: 
        PolicyDocument: 
            Version: "2012-10-17"
            Statement:
                - Effect: Allow
                Action:
                    - "iam:CreatePolicyVersion"
                Resource:
                    - "*" # Sensitive
        Roles: 
            - !Ref MyRole
----

== Compliant Solution

Restrict update permission to the appropriate subset of policies:

----
MyPolicy:
    Type: AWS::IAM::ManagedPolicy
    Properties: 
        PolicyDocument: 
            Version: "2012-10-17"
            Statement:
                - Effect: Allow
                Action:
                    - "iam:CreatePolicyVersion"
                Resource:
                    - !Sub "arn:aws:iam::${AWS::AccountId}:policy/team1/*"
        Roles: 
            - !Ref MyRole
----

include::../exceptions.adoc[]

include::../see.adoc[]