== How to fix it in Mcrypt

=== Code examples

==== Noncompliant code example

include::../../common/fix/aes-noncompliant-example.adoc[]

[source,php,diff-id=1,diff-type=noncompliant]
----
mcrypt_encrypt(MCRYPT_DES, $key, $plaintext, "ecb"); // Noncompliant
----

==== Compliant solution

Mcrypt is deprecated and should not be used. You can use
https://www.php.net/manual/en/book.sodium.php[Sodium] instead.

include::../../common/fix/aes-compliant-example.adoc[]

[source,php,diff-id=1,diff-type=compliant]
----
sodium_crypto_aead_aes256gcm_encrypt($plaintext, '', $nonce, $key);
----

=== How does this work?

include::../../common/fix/fix.adoc[]

=== Going the extra mile

include::../../common/extra-mile/pre-quantum.adoc[]

=== Going the extra mile

include::../../common/extra-mile/pre-quantum.adoc[]
