<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Shared naming conventions allow teams to collaborate effectively. This rule raises an issue when a pom&#8217;s <code>artifactId</code> does not match the provided regular expression.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="paragraph">
<p>With the default regular expression: <code>[a-z][a-z-0-9]+</code></p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-xml" data-lang="xml">&lt;project ...&gt;
  &lt;artifactId&gt;My_Project&lt;/artifactId&gt;  &lt;!-- Noncompliant --&gt;

  &lt;!-- ... --&gt;
&lt;/project&gt;</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-xml" data-lang="xml">&lt;project ...&gt;
  &lt;artifactId&gt;my-project&lt;/artifactId&gt;

  &lt;!-- ... --&gt;
&lt;/project&gt;</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Update this "artifactId" to match the provided regular expression: xxx</p>
</div>
</div>
<div class="sect2">
<h3 id="_parameters">Parameters</h3>
<div class="sidebarblock">
<div class="content">
<div class="title">regex</div>
<div class="listingblock">
<div class="content">
<pre>[a-z][a-z-0-9]+</pre>
</div>
</div>
<div class="paragraph">
<p>The regular expression the "artifactId" should match</p>
</div>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>artifactId value</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_25_nov_2015_085745_freddy_mallet_wrote">on 25 Nov 2015, 08:57:45 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>For rules applying to pom.xml Maven file like this one I would use the tag 'maven' instead of 'pom' [~ann.campbell.2] because for a non-Maven users 'pom' doesn&#8217;t mean anything. My 2 cents</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_30_nov_2015_140741_michael_gumowski_wrote">on 30 Nov 2015, 14:07:41 Michael Gumowski wrote:</h3>
<div class="paragraph">
<p>+1 with Freddy&#8217;s comment!</p>
</div>
<div class="paragraph">
<p>Otherwise, LGTM [~ann.campbell.2]!</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_30_nov_2015_184655_ann_campbell_wrote">on 30 Nov 2015, 18:46:55 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>labels updated</p>
</div>
</div>
</div>
</div>