include::../description.adoc[]

== Noncompliant Code Example

[source,java]
----
private void called(int foo) {
  foo==1; // Noncompliant
  if (foo==1) {
    System.out.println("foo\n");
  }
}

public void caller(String [ ] args {
  called(2);
  return 0;
}
----

== Compliant Solution

[source,java]
----
private void called(int foo) {
  foo=1;
  if (foo==1) {
    System.out.println("foo\n");
  }
}

public void caller(String [ ] args {
  called(2);
  return 0;
}
----

include::../see.adoc[]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
