<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>C&#43;&#43;20 extends the <code>using</code> declaration to <code>enum class</code>. <code>using enum</code> introduces all the <code>enum</code> constants into the current scope.</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">enum class rgbaColorChannel { red, green, blue, alpha };

std::string_view toString(rgbaColorChannel channel) {
  switch (channel) {
    using enum rgbaColorChannel;
    case red:   return "red";
    case green: return "green";
    case blue:  return "blue";
    case alpha: return "alpha";
  }
}</code></pre>
</div>
</div>
<div class="paragraph">
<p>As with other <code>using</code> declarations, <code>using enum</code> improves readability when used in small scopes yet it generates confusion in large scopes.</p>
</div>
<div class="paragraph">
<p>The <code>switch</code> statement, as in the example above, is a natural scope for <code>using enum</code>.</p>
</div>
<div class="paragraph">
<p>This rule reports scopes that use a particular <code>enum class</code> extensively and can benefit from <code>using enum</code> declaration. For example, it reports most <code>switch</code> statements applied to an <code>enum</code> value.</p>
</div>
<div class="sect2">
<h3 id="_exceptions">Exceptions</h3>
<div class="paragraph">
<p>The rule does not apply if adding the <code>using enum</code> clause would create a name collision or reduce readability by shadowing a name.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Add <code>using enum</code> inside the relevant scope and replace the qualified identifiers with the corresponding enumerator names.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">enum class DayOfTheWeek { mon, tue, wed, thu, fri, sat, sun };

bool isWorkDay(DayOfTheWeek day) {
  switch(day) { // Noncompliant: case statements are too verbose
    case DayOfTheWeek::mon: return true;
    case DayOfTheWeek::tue: return true;
    case DayOfTheWeek::wed: return true;
    case DayOfTheWeek::thu: return true;
    case DayOfTheWeek::fri: return true;
    case DayOfTheWeek::sat: return false;
    case DayOfTheWeek::sun: return false;
  }
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">enum class DayOfTheWeek { mon, tue, wed, thu, fri, sat, sun };

bool isWorkDay(DayOfTheWeek day) {
  switch(day) {
    using enum DayOfTheWeek;

    case mon: return true;
    case tue: return true;
    case wed: return true;
    case thu: return true;
    case fri: return true;
    case sat: return false;
    case sun: return false;
  }
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_noncompliant_code_example_2">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">enum class rgbaColorChannel { red, green, blue, alpha };

rgbaColorChannel chooseChannel(rgbaColorChannel background, rgbaColorChannel foreground) {
  auto result = background;
  if ((background == rgbaColorChannel::red || background == rgbaColorChannel::green) &amp;&amp; foreground != rgbaColorChannel::red)
    result = foreground;
  return result;
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution_2">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">enum class rgbaColorChannel { red, green, blue, alpha };

rgbaColorChannel chooseChannel(rgbaColorChannel background, rgbaColorChannel foreground) {
  using enum rgbaColorChannel;
  auto result = background;
  if ((background == red || background == green) &amp;&amp; foreground != red)
    result = foreground;
  return result;
}</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>