This rule raises an issue when an Apex class uses a generic name like "Handler", "Processor", or "Manager" without additional context that describes its specific business purpose.

== Why is this an issue?

Generic class names like `Handler` or `Processor` provide no information about what the class actually does or which business domain it serves. This creates several problems:

**Reduced Code Readability**
When developers encounter a class named `Handler`, they must read the entire implementation to understand its purpose. This slows down development and increases cognitive load.

**Maintenance Difficulties**
In complex business applications like Revenue Cloud systems, you might have dozens of handlers for different processes - pricing, billing, contract management, product configuration. Without descriptive names, developers waste time searching for the right class.

**Team Collaboration Issues**
When multiple developers work on the same codebase, generic names make it harder to discuss specific components. "The pricing handler" is clearer than "the handler in the revenue package."

**Business Domain Confusion**
In enterprise applications, different business domains often have similar patterns but serve different purposes. A `PricingProcedureHandler` immediately tells you it's related to pricing logic, while `Handler` could be anything.

Descriptive names act as inline documentation, making your code self-explaining and reducing the need for additional comments or documentation.

=== What is the potential impact?

Poor class naming reduces code maintainability and increases development time. In complex business applications, developers spend more time understanding existing code rather than implementing new features. This leads to slower delivery, increased bug risk, and higher onboarding costs for new team members.

== How to fix it

Replace generic class names with specific names that describe the business purpose and domain. Include the business process or entity the class handles.

=== Code examples

==== Noncompliant code example

[source,apex,diff-id=1,diff-type=noncompliant]
----
public class Handler {
    public void process() {
        // pricing calculation logic
    }
}
----

==== Compliant solution

[source,apex,diff-id=1,diff-type=compliant]
----
public class PricingProcedureHandler {
    public void calculateFinalNetPrice() {
        // pricing calculation logic
    }
}
----

== Resources

=== Documentation

 * Apex Developer Guide - Classes - https://developer.salesforce.com/docs/atlas.en-us.apexcode.meta/apexcode/apex_classes.htm[Official Salesforce documentation on Apex classes and best practices]

 * Clean Code Principles - https://blog.cleancoder.com/uncle-bob/2012/08/13/the-clean-architecture.html[Principles for writing clean, maintainable code including naming conventions]

=== Standards

 * Clean Code - Meaningful Names - https://www.oreilly.com/library/view/clean-code-a/9780136083238/[Industry standard practices for choosing meaningful names in software development]
