<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when a dictionary comprehension is used to create a dictionary where all values are the same constant.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Using a dictionary comprehension to build a dictionary where every key maps to the exact same constant value e.g., <code>{k: 1 for k in keys}</code> is less efficient and less idiomatic than using the <code>dict.fromkeys()</code> class method. <code>dict.fromkeys()</code> is specifically designed for this use case and offers better performance, especially for large iterables, as it avoids the overhead of creating and processing individual key-value pairs in a comprehension.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Rewrite the dictionary comprehension <code>{x: constant for x in iterable}</code> as <code>dict.fromkeys(iterable, constant)</code>. If the constant value is <code>None</code>, you can omit the value argument in <code>dict.fromkeys()</code>, as it defaults to <code>None</code>.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">keys = ['a', 'b', 'c']

dict_comp_one = {k: 1 for k in keys} # Noncompliant</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">keys = ['a', 'b', 'c']

dict_fromkeys_one = dict.fromkeys(keys, 1)</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Python Documentation - <a href="https://docs.python.org/3/library/stdtypes.html#dict.fromkeys">dict.fromkeys</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>