== Why is this an issue?

include::description.adoc[]

=== Noncompliant code example

[source,text]
----
SELECT FIRST_NAME, LAST_NAME, REGION
FROM PERSONS
ORDER BY 2, 1
----


=== Compliant solution

[source,text]
----
SELECT FIRST_NAME, LAST_NAME, REGION
FROM PERSONS
ORDER BY LAST_NAME, FIRST_NAME
----

