== Why is this an issue?

The rules of operator precedence are complicated and can lead to errors. For this reason, parentheses should be used for clarification in complex statements.


This rule raises an issue when more than the allowed number of non-like operators are used in a statement without parentheses to make execution order explicit.

=== Noncompliant code example

With the default threshold of 2:

[source,cobol]
----
COMPUTE WSRESULT = WS1 + 5 * WS2 - WS3**2 END-COMPUTE  *> Noncompliant
COMPUTE WSRESULT2 = WS1 + 5 + WS2 + WS3 + WS4 END-COMPUTE
----

=== Compliant solution

[source,cobol]
----
COMPUTE WSRESULT = WS1 + (5 * WS2) - (WS3**2) END-COMPUTE
COMPUTE WSRESULT2 = WS1 + 5 + WS2 + WS3 + WS4 END-COMPUTE
----

include::../see.adoc[]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::parameters.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
