== Why is this an issue?

Conversion of a function pointer to a different type of pointer results in undefined behaviour. This means, for example, that a pointer to a function cannot be converted to a pointer to a different type of function.


=== Noncompliant code example

[source,cpp]
----
int f(int a)
{
  float (*p)(float) = (float (*)(float)) & f; // Noncompliant
}
----


== Resources

* MISRA C:2004, 11.1 - Conversions shall not be performed between a pointer to a function and any type other than an integral type.
* MISRA {cpp}:2008, 5-2-6 - A cast shall not convert a pointer to a function to any other pointer type, including a pointer to function type.
*  MISRA C:2012, 11.1 - Conversions shall not be performed between a pointer to a function and any other type


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Remove this hazardous cast.


'''
== Comments And Links
(visible only on this page)

=== relates to: S856

=== relates to: S1944

=== on 3 Oct 2014, 18:40:28 Ann Campbell wrote:
\[~evgeny.mandrikov]


MISRA C:2004 says: "Conversions shall not be performed between a pointer to a function and any type *other than an integral type*" (emphasis mine), meaning a cast to an integral type is allowed.


MISRA {cpp}:2008: "A cast should not convert a pointer type to an integral type." I.e. a direct contradiction of the second part of the C:2004 rule.


MISRA C:2012 backs up {cpp}:2008: "Conversions shall not be performed between a pointer to a function and any other type."


So... what would you like written up?

=== on 23 Oct 2014, 09:29:23 Samuel Mercier wrote:
Rule {cpp} 5-2-9 extracted to another ticket, since its scope is different.

=== on 23 Oct 2014, 09:35:06 Samuel Mercier wrote:
Don't get the point of MISRA 2004 : conversion from function pointer to integral type is hazardous...

=== on 23 Oct 2014, 15:07:10 Ann Campbell wrote:
\[~samuel.mercier] please add a See section

endif::env-github,rspecator-view[]
