include::../description.adoc[]


== Noncompliant Code Example

https://cryptography.io[cryptography]

[source,python]
----
from cryptography.hazmat.primitives.ciphers.aead import AESGCM

def encrypt(key, plaintext):
    nonce   = '7cVgr5cbdCZV'.encode('utf-8') # The initialization vector is a static value

    cipher  = AESGCM(key)

    cipher.encrypt(nonce, plaintext, None) # Noncompliant
----

https://pycryptodome.readthedocs.io[pycryptodome]

[source,python]
----
from Crypto.Cipher import AES

def encrypt(key, plaintext):
    nonce   = '7cVgr5cbdCZV'.encode('utf-8') # The initialization vector is a static value

    cipher = AES.new(key, AES.MODE_CCM, nonce) # Initialization vector is configured here
    
    cipher.encrypt(plaintext) # Noncompliant
----

== Compliant Solution

https://cryptography.io[cryptography]

[source,python]
from os import urandom
from cryptography.hazmat.primitives.ciphers.aead import AESGCM
----
def encrypt(key, plaintext):
    nonce   = urandom(12) # Random 96 bit IV

    cipher  = AESGCM(key)

    cipher.encrypt(nonce, plaintext, None)
----

https://pycryptodome.readthedocs.io[pycryptodome]

[source,python]
----
from Crypto.Cipher import AES

def encrypt(key, plaintext):

    cipher  = AES.new(key, AES.MODE_CCM)

    cipher.encrypt_and_digest(plaintext)
----

include::../see.adoc[]


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::./message.adoc[]

include::./highlighting.adoc[]

endif::env-github,rspecator-view[]