== Why is this an issue?

FIXME: add a description

// If you want to factorize the description uncomment the following line and create the file.
//include::../description.adoc[]

=== Noncompliant code example

[source,python]
----
FIXME
----

=== Compliant solution

[source,python]
----
FIXME
----

== Resources

FIXME: A list of links