== Why is this an issue?

When a class in a catch clause does not exist, PHP won't generate an error.
This can lead to challenging debug issues, as the catch clause will have no effect and the root cause may not be obvious.

This problem typically occurs when being in a namespace trying to catch PHP built-in exception classes without escaping to the global namespace or importing the classes.

This rule raises an issue when, being in a namespace, an undefined class belonging to that namespace is caught.

== How to fix it

Escape the caught exception to the global namespace or import the class.

=== Code examples

==== Noncompliant code example

[source,php,diff-id=1,diff-type=noncompliant]
----
namespace Foo\Bar;

try {
    doSomething();
} catch (Exception $e) { // Noncompliant - Exception will never be caught because the class Exception does not exist in the namespace
    echo $e->message;
}
----

==== Compliant solution

[source,php,diff-id=1,diff-type=compliant]
----
namespace Foo\Bar;

try {
    doSomething();
} catch (\Exception $e) { // Compliant: Used by global namespace
    echo $e->message;
}
----

[source,php,diff-id=1,diff-type=compliant]
----
namespace Foo\Bar;

use Exception;

try {
    doSomething();
} catch (Exception $e) { // Compliant: Imported by use statement
    echo $e->message;
}
----

== Resources

=== Documentation

ifdef::env-github,rspecator-view[]

'''

== Implementation Specification

(visible only on this page)

include::message.adoc[]

endif::env-github,rspecator-view[]
