== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,pli]
----
foo: proc options(main);
  declare i fixed decimal init (0);

loopLabel:
  put list (i);
  i = i + 1;
  if i < 10 then go to loopLabel;  /* Noncompliant - usage of the GO TO statement */
end;
----

=== Compliant solution

[source,pli]
----
foo: proc options(main);
  declare i fixed decimal init (0);


  do i = 0 to 9;   /* Compliant */
    put list (i);
  end;
end;
----

=== Exceptions

Exclusions format parameter can be used to allow specific ``++GO TO++`` targets which match the given regular expression.

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::parameters.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
