There are several reasons for a method not to have a method body:


* It is an unintentional omission, and should be fixed.
* It is not yet, or never will be, supported. In this case an error should be thrown.
* The method is an intentionally-blank. In this case a nested comment should explain the reason for the blank method.

== Noncompliant Code Example

----
Sub DoSomething()
End Sub

Function DoSomething() As String
End Function
----

== Compliant Solution

----
Sub DoSomething()
    ' Not implemented because of reason
End Sub

Function DoSomething() As String
    DoSomething = "Value"
End Function
----

ifdef::env-github,rspecator-view[]
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
