== Why is this an issue?

There are several reasons for a method not to have a method body:


* It is an unintentional omission, and should be fixed.
* It is not yet, or never will be, supported. In this case an error should be thrown.
* The method is an intentionally-blank. In this case a nested comment should explain the reason for the blank method.

=== Noncompliant code example

[source,vb6]
----
Sub DoSomething()
End Sub

Function DoSomething() As String
End Function
----

=== Compliant solution

[source,vb6]
----
Sub DoSomething()
    ' Not implemented because of reason
End Sub

Function DoSomething() As String
    DoSomething = "Value"
End Function
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Add a nested comment explaining why this method is empty, throw an error or complete the implementation.


'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
