<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>In Rust, calling <code>Vec::set_len(new_len)</code> directly after allocating or reserving memory using <code>Vec::with_capacity()</code>, <code>Vec::new()</code>, <code>Vec::default()</code>, or <code>Vec::reserve()</code> can lead to the creation of a <code>Vec</code> with uninitialized elements. This is problematic because most safe Rust operations assume initialized data. Using uninitialized data can result in undefined behavior, including memory corruption and unexpected program crashes.</p>
</div>
<div class="paragraph">
<p>This rule detects instances where <code>Vec::set_len()</code> is called directly after allocating or reserving memory with <code>Vec::with_capacity()</code>, <code>Vec::new()</code>, <code>Vec::default()</code>, or <code>Vec::reserve()</code>, without any prior initialization of the vector&#8217;s elements. The rule performs a local analysis and only checks for these calls within adjacent statements.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-rust" data-lang="rust">fn f(reader: &amp;mut dyn std::io::Read) {
    let mut vec: Vec&lt;u8&gt; = Vec::with_capacity(1000);
    unsafe { vec.set_len(1000); } // Noncompliant: Uninitialized vector
    reader.read_exact(&amp;mut vec).unwrap(); // Undefined behavior!
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-rust" data-lang="rust">fn f(reader: &amp;mut dyn std::io::Read) {
    let mut vec: Vec&lt;u8&gt; = vec![0; 1000]; // Properly initialized with zeros
    reader.read_exact(&amp;mut vec).unwrap(); // Safe to use
}</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Clippy Lints - <a href="https://rust-lang.github.io/rust-clippy/master/index.html#uninit_vec" class="bare">https://rust-lang.github.io/rust-clippy/master/index.html#uninit_vec</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>