include::../description.adoc[]

== Noncompliant Code Example

With a maximum number of 4 parameters:

----
fun foo(p1: String, p2: String, p3: String, p4: String, p5: String) { // Noncompliant
  // ...
}
----

== Compliant Solution

----
fun foo(p1: String, p2: String, p3: String, p4: String) { 
  // ...
}
----

== Exceptions

Methods annotated with Spring's ``++@RequestMapping++`` (and related shortcut annotations, like ``++@GetRequest++``) or ``++@JsonCreator++`` may have a lot of parameters, encapsulation being possible. Such methods are therefore ignored.

ifdef::rspecator-view[]
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::rspecator-view[]
