== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

With a maximum number of 4 parameters:

[source,kotlin]
----
fun foo(p1: String, p2: String, p3: String, p4: String, p5: String) { // Noncompliant
  // ...
}
----

=== Compliant solution

[source,kotlin]
----
fun foo(p1: String, p2: String, p3: String, p4: String) { 
  // ...
}
----

=== Exceptions

Methods annotated with Spring's ``++@RequestMapping++`` (and related shortcut annotations, like ``++@GetRequest++``) or ``++@JsonCreator++`` may have a lot of parameters, encapsulation being possible. Such methods are therefore ignored.

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../parameters.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
