<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>The <code>READ TABLE ... WITH KEY ...</code> statement performs a linear search of <code>STANDARD</code> tables, which is very inefficient in most cases.</p>
</div>
<div class="paragraph">
<p>This rule raises an issue when a <code>READ TABLE ... WITH KEY ...</code> statement does not finish with <code>BINARY SEARCH</code>. No issue will be raised for <code>HASHED</code> and <code>SORTED</code> tables.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-abap" data-lang="abap">TYPES BEGIN OF t_mytable,
    myfield TYPE i
END OF t_mytable.

DATA myworkarea TYPE t_mytable.

DATA mytable TYPE STANDARD TABLE OF t_mytable.

SORT mytable BY myfield.

READ TABLE mytable
    WITH KEY myfield = 42
    INTO myworkarea. " Noncompliant</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-abap" data-lang="abap">TYPES BEGIN OF t_mytable,
    myfield TYPE i
END OF t_mytable.

DATA myworkarea TYPE t_mytable.

DATA mytable TYPE STANDARD TABLE OF t_mytable.

SORT mytable BY myfield.

READ TABLE mytable
    WITH KEY myfield = 42
    INTO myworkarea
    BINARY SEARCH. " Compliant

DATA my_hashed_table TYPE HASHED TABLE OF t_mytable
    WITH UNIQUE KEY myfield.

DATA my_sorted_table TYPE SORTED TABLE OF t_mytable
    WITH UNIQUE KEY myfield.

READ TABLE my_hashed_table
    WITH KEY myfield = 42
    INTO myworkarea. " Compliant

READ TABLE my_sorted_table
    WITH KEY myfield = 42
    INTO myworkarea. " Compliant</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Add "BINARY SEARCH" to this "READ" statement.</p>
</div>
</div>
</div>
</div>