The <code>READ TABLE ... WITH KEY ...</code> statement performs a linear search of <code>STANDARD</code> tables, which is very inefficient in most cases.

This rule raises an issue when a <code>READ TABLE ... WITH KEY ...</code> statement does not finish with <code>BINARY SEARCH</code>. No issue will be raised for <code>HASHED</code> and <code>SORTED</code> tables.


== Noncompliant Code Example

----
TYPES BEGIN OF t_mytable, 
    myfield TYPE i
END OF t_mytable. 

DATA myworkarea TYPE t_mytable.

DATA mytable TYPE STANDARD TABLE OF t_mytable.

SORT mytable BY myfield.

READ TABLE mytable 
    WITH KEY myfield = 42
    INTO myworkarea. " Noncompliant
----


== Compliant Solution

----
TYPES BEGIN OF t_mytable, 
    myfield TYPE i
END OF t_mytable. 

DATA myworkarea TYPE t_mytable.

DATA mytable TYPE STANDARD TABLE OF t_mytable.

SORT mytable BY myfield.

READ TABLE mytable 
    WITH KEY myfield = 42
    INTO myworkarea
    BINARY SEARCH. " Compliant

DATA my_hashed_table TYPE HASHED TABLE OF t_mytable 
    WITH UNIQUE KEY myfield.

DATA my_sorted_table TYPE SORTED TABLE OF t_mytable 
    WITH UNIQUE KEY myfield.

READ TABLE my_hashed_table 
    WITH KEY myfield = 42
    INTO myworkarea. " Compliant

READ TABLE my_sorted_table 
    WITH KEY myfield = 42
    INTO myworkarea. " Compliant
----

