<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>A Spring <code>@Controller</code> that uses <code>@SessionAttributes</code> is designed to handle a stateful / multi-post form. Such <code>@Controller</code>s use the specified <code>@SessionAttributes</code> to store data on the server between requests. That data should be cleaned up when the session is over, but unless <code>setComplete()</code> is called on the <code>SessionStatus</code> object from a <code>@RequestMapping</code> method, neither Spring nor the JVM will know it&#8217;s time to do that. Note that the <code>SessionStatus</code> object must be passed to that method as a parameter.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">@Controller
@SessionAttributes("hello")  // Noncompliant: this doesn't get cleaned up
public class HelloWorld {

  @RequestMapping("/greet", method = GET)
  public String greet(String greetee) {

    return "Hello " + greetee;
  }
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">@Controller
@SessionAttributes("hello")
public class HelloWorld {

  @RequestMapping("/greet", method = GET)
  public String greet(String greetee) {

    return "Hello " + greetee;
  }

  @RequestMapping("/goodbye", method = POST)
  public String goodbye(SessionStatus status) {
    //...
    status.setComplete();
  }

}</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Add a call to "setComplete()" on the SessionStatus object in a "@RequestMapping" method.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p><code>@SessionAttributes</code></p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_21_jun_2018_145221_andrei_epure_wrote">on 21 Jun 2018, 14:52:21 Andrei Epure wrote:</h3>
<div class="paragraph">
<p>Note: The RuleAPI will correctly interpret the markdown, even if JIRA does not</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_26_jun_2018_104954_alban_auzeill_wrote">on 26 Jun 2018, 10:49:54 Alban Auzeill wrote:</h3>
<div class="paragraph">
<p>SessionStatus could be a parameter of, not only POST requests, but any kind of @RequestMapping  (like GET requests). Example:</p>
</div>
<div class="paragraph">
<p><a href="https://www.logicbig.com/tutorials/spring-framework/spring-web-mvc/spring-model-attribute-with-session.html">spring-model-attribute-with-session</a> ยง Removing Model Attribute from the session</p>
</div>
<div class="paragraph">
<p>IMO the implementation of this rule should check for each class annotated by @Controller and @SessionAttributes, there&#8217;s at least one call to "sessionStatus.setComplete()" in any method body of this class. Because if such call exist, there&#8217;s other rules to ensure that the code is reachable, the method is used, &#8230;&#8203; So we don&#8217;t need to focus on @RequestMapping, @PostMapping, GET, POST, &#8230;&#8203;</p>
</div>
</div>
</div>
</div>