<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This is an issue when a test class contains multiple methods annotated with <code>@testSetup</code>.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>In Apex, test classes can only contain one <code>@testSetup</code> method. The <code>@testSetup</code> annotation is used to create test data that will be available to all test methods in the class, improving test performance by avoiding duplicate data creation.</p>
</div>
<div class="paragraph">
<p>When you define multiple <code>@testSetup</code> methods in the same test class, the Apex compiler will reject the code and throw a compilation error. This happens because the Salesforce platform expects exactly one setup method per test class to maintain predictable test execution order and data consistency.</p>
</div>
<div class="paragraph">
<p>The <code>@testSetup</code> method runs once before all test methods in the class, creating a clean, consistent data state. Having multiple setup methods would create ambiguity about execution order and could lead to unpredictable test behavior.</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="paragraph">
<p>Multiple <code>@testSetup</code> methods will cause compilation errors, preventing the test class from being deployed or executed. This blocks development progress and can prevent deployments to production environments.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Consolidate all setup logic into a single @testSetup method. Move the code from multiple setup methods into one method and remove the duplicate @testSetup annotations.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-apex" data-lang="apex">@isTest
public class MyTestClass {
    @testSetup
    static void setup1() {
        // setup code
        Account acc = new Account(Name = 'Test Account');
        insert acc;
    }

    @testSetup // Noncompliant
    static void setup2() {
        // more setup code
        Contact con = new Contact(LastName = 'Test Contact');
        insert con;
    }
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-apex" data-lang="apex">@isTest
public class MyTestClass {
    @testSetup
    static void setup() {
        // all setup code in one method
        Account acc = new Account(Name = 'Test Account');
        insert acc;

        Contact con = new Contact(LastName = 'Test Contact');
        insert con;
    }
}</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Apex Testing Framework - <a href="https://developer.salesforce.com/docs/atlas.en-us.apexcode.meta/apexcode/apex_testing_framework.htm">Official Salesforce documentation on Apex testing framework and @testSetup usage</a></p>
</li>
<li>
<p>Unit Testing in Apex - <a href="https://www.apexhours.com/unit-testing-in-apex">Comprehensive guide on Apex unit testing including @testSetup best practices</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>