This rule raises an issue when a method is annotated with ``++@AuraEnabled++`` but has ``++private++`` or ``++protected++`` visibility.

== Why is this an issue?

Lightning Web Components can only import and call Apex methods that are accessible from outside the class. The ``++@AuraEnabled++`` annotation marks a method as available for Lightning components, but the method must also have the proper visibility to be actually accessible.

When a method is marked as ``++private++`` or ``++protected++``, it cannot be imported by Lightning Web Components, even if it has the ``++@AuraEnabled++`` annotation. This creates a mismatch between the developer's intent (making the method available to Lightning components) and the actual accessibility of the method.

The Salesforce platform requires ``++@AuraEnabled++`` methods to be either ``++public++`` or ``++global++`` to ensure they can be properly imported and called from Lightning Web Components using the ES6 import syntax.

=== What is the potential impact?

The method will not be accessible from Lightning Web Components, causing import errors or runtime failures when the component tries to call the method. This can break the functionality of Lightning components that depend on these Apex methods.

== How to fix it

Change the method visibility from ``++private++`` to ``++public++`` to make it accessible from Lightning Web Components.

=== Code examples

==== Noncompliant code example

[source,apex,diff-id=1,diff-type=noncompliant]
----
@AuraEnabled
private static List<Account> getAccounts(String accountName) { // Noncompliant
    return [SELECT Id, Name FROM Account WHERE Name LIKE :accountName];
}
----

==== Compliant solution

[source,apex,diff-id=1,diff-type=compliant]
----
@AuraEnabled
public static List<Account> getAccounts(String accountName) {
    return [SELECT Id, Name FROM Account WHERE Name LIKE :accountName];
}
----

== Resources

=== Documentation

 * Lightning Web Components Developer Guide - Call Apex Methods - https://developer.salesforce.com/docs/platform/lwc/guide/apex.html[Official Salesforce documentation on calling Apex methods from Lightning Web Components]

 * Apex Developer Guide - @AuraEnabled Annotation - https://developer.salesforce.com/docs/atlas.en-us.apexcode.meta/apexcode/apex_classes_annotation_AuraEnabled.htm[Documentation on the @AuraEnabled annotation and its requirements]
