<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when a module uses <code>module_function</code> to create module-level methods.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Using <code>module_function</code> in Ruby modules can make the code&#8217;s intent less clear compared to <code>extend self</code>.</p>
</div>
<div class="paragraph">
<p>When you want to create a module with methods that can be called directly on the module itself, Ruby provides two main approaches: <code>module_function</code> and <code>extend self</code>. While both work, they have different behaviors and implications.</p>
</div>
<div class="paragraph">
<p><code>module_function</code> creates both private instance methods and public module methods. This dual nature can be confusing because it suggests the module might be included in classes, when often the intent is just to have module-level utility methods.</p>
</div>
<div class="paragraph">
<p><code>extend self</code> is more explicit about the intent. It clearly states that the module is extending itself with its own instance methods, making them available as module methods. This approach is more straightforward when all methods in the module should be callable at the module level.</p>
</div>
<div class="paragraph">
<p>The <code>extend self</code> approach is also more consistent with Ruby&#8217;s object model and is recommended by major style guides, including Shopify&#8217;s Ruby Style Guide.</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="paragraph">
<p>Using <code>module_function</code> instead of <code>extend self</code> reduces code clarity and can make the module&#8217;s intended usage less obvious to other developers. While functionally equivalent in many cases, <code>extend self</code> better communicates the intent when creating utility modules with module-level methods.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Replace <code>module_function</code> with <code>extend self</code> at the beginning of the module. This makes the intent clearer and follows Ruby style guide recommendations.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-ruby" data-lang="ruby">module SomeModule
  module_function # Noncompliant

  def some_method
    "Hello from module method"
  end

  def another_method
    "Another module method"
  end
end</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-ruby" data-lang="ruby">module SomeModule
  extend self

  def some_method
    "Hello from module method"
  end

  def another_method
    "Another module method"
  end
end</code></pre>
</div>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Ruby Documentation - Module#extend - <a href="https://ruby-doc.org/core/Module.html#method-i-extend">Official Ruby documentation for the extend method</a></p>
</li>
<li>
<p>Ruby Documentation - Module#module_function - <a href="https://ruby-doc.org/core/Module.html#method-i-module_function">Official Ruby documentation for module_function</a></p>
</li>
<li>
<p>Shopify Ruby Style Guide - <a href="https://github.com/Shopify/ruby-style-guide#module-function">Shopify Ruby Style Guide - Module Function</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>