<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises when a TaskGroup or Nursery body contains only a single call to <code>.start</code> or <code>.start_soon</code> without passing itself as a parameter.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>TaskGroup and Nursery are powerful tools for structured concurrency that automatically manage the lifecycle of multiple concurrent tasks. However, when they are used to spawn only a single task that doesn&#8217;t need the nursery/TaskGroup instance itself, they add unnecessary overhead and complexity.</p>
</div>
<div class="paragraph">
<p>The main issues with using TaskGroup and Nursery for single tasks are:</p>
</div>
<div class="ulist">
<ul>
<li>
<p><strong>Unnecessary overhead</strong>: Creating a nursery or TaskGroup involves additional setup and teardown that serves no purpose when only one task is spawned</p>
</li>
<li>
<p><strong>Misleading code</strong>: The presence of a nursery suggests that multiple tasks will be managed, which can be confusing</p>
</li>
<li>
<p><strong>Reduced readability</strong>: The structured concurrency syntax is more verbose than a simple function call</p>
</li>
</ul>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="ulist">
<ul>
<li>
<p><strong>Performance</strong>: Minor performance overhead from unnecessary nursery/TaskGroup creation</p>
</li>
<li>
<p><strong>Maintainability</strong>: Code is more complex than necessary, making it harder to understand and maintain</p>
</li>
<li>
<p><strong>Code clarity</strong>: The intent of the code is obscured by unnecessary structured concurrency constructs</p>
</li>
</ul>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it_in_asyncio">How to fix it in Asyncio</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Replace the TaskGroup with a direct function call when the TaskGroup body contains only a single <code>create_task()</code> call.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">import asyncio

async def worker_task():
    await asyncio.sleep(1)

async def main():
    # Unnecessary TaskGroup for a single task
    async with asyncio.TaskGroup() as tg:
        tg.create_task(worker_task())</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">import asyncio

async def worker_task():
    await asyncio.sleep(1)

async def main():
    # Direct function call is simpler and more efficient
    await worker_task()</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it_in_trio">How to fix it in Trio</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Replace the nursery with a direct function call when the nursery body contains only a single <code>start_soon()</code> or <code>start()</code> call.</p>
</div>
<div class="sect2">
<h3 id="_code_examples_2">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example_2">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">import trio

async def worker_task():
    await trio.sleep(1)

async def main():
    # Unnecessary nursery for a single task
    async with trio.open_nursery() as nursery:
        nursery.start_soon(worker_task)</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution_2">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">import trio

async def worker_task():
    await trio.sleep(1)

async def main():
    # Direct function call is simpler and more efficient
    await worker_task()</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it_in_anyio">How to fix it in AnyIO</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Replace the task group with a direct function call when the task group body contains only a single <code>start_soon()</code> or <code>start()</code> call.</p>
</div>
<div class="sect2">
<h3 id="_code_examples_3">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example_3">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">import anyio

async def worker_task():
    await anyio.sleep(1)

async def main():
    # Unnecessary task group for a single task
    async with anyio.create_task_group() as tg:
        tg.start_soon(worker_task)</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution_3">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">import anyio

async def worker_task():
    await anyio.sleep(1)

async def main():
    # Direct function call is simpler and more efficient
    await worker_task()</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Asyncio documentation - <a href="https://docs.python.org/3/library/asyncio-task.html#task-groups">Task Groups</a></p>
</li>
<li>
<p>Trio documentation - <a href="https://trio.readthedocs.io/en/latest/reference-core.html#nurseries-and-spawning">Nurseries and spawning</a></p>
</li>
<li>
<p>AnyIO documentation - <a href="https://anyio.readthedocs.io/en/stable/tasks.html#creating-and-managing-tasks">Creating and managing tasks</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Replace the TaskGroup or Nursery with a direct function call.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="ulist">
<ul>
<li>
<p>Primary locations: The task group creation function in the <code>async with</code> statement</p>
</li>
<li>
<p>Secondary locations: - The <code>.start</code> or <code>.start_soon</code> call</p>
<div class="ulist">
<ul>
<li>
<p>the enclosing async function <code>async</code> keyword (message: "This function is async.")</p>
</li>
</ul>
</div>
</li>
</ul>
</div>
</div>
</div>
</div>