Using a unique symbolic <code>PLIST</code> name provides a single definition point for the parameters used by any external routine being called. This ensures that the same information is passed every time the external routine is called, reducing the potential for mistakes in operation. 

This rule flags non-compliant instances of <code>CALL</code> and <code>CALLB</code>.


== Noncompliant Code Example

----
C                   CALLB(D)  'XR01'
C                   PARM                    WSXR07
C                   PARM                    WSFLD1
----


== Compliant Solution

----
C     P0XR07        PLIST                               
C                   PARM                    WSXR07       
C                   PARM                    WSFLD1 

C                   CALLB     'XR07'        P0XR07  
----

