<div id="preamble">
<div class="sectionbody">
<div class="paragraph">
<p>Shared naming conventions allow teams to collaborate efficiently. This rule checks that all function names match a provided regular expression.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_invalid_header">Invalid header</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="paragraph">
<p>With default provided regular expression: <code>^[a-z][a-zA-Z0-9]*$</code>:</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">void DoSomething (void);</code></pre>
</div>
</div>
<div class="listingblock">
<div class="content">
<pre>void DoSomethingElse (void);</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_see">See</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_parameters">Parameters</h3>
<div class="sidebarblock">
<div class="content">
<div class="title">format</div>
<div class="listingblock">
<div class="content">
<pre>^([A-Z0-9_]*|[a-z0-9_]*)$</pre>
</div>
</div>
</div>
</div>
</div>
</div>
</div>