Collection classes from the ``++java.util.concurrent++`` package have their own concurrency control mechanisms. A concurrent collection is thread-safe, but not governed by a single exclusion lock, therefore using an instance of such a class for synchronization is, at best unnecessary and at worst likely to have unintended consequences. 


This rule raises an issue when a synchronization lock is used on an instance of one of the collection classes from the ``++java.util.concurrent++`` package.


== Noncompliant Code Example

----
public class Foo {
  private ConcurrentHashMap<String,String> map = new ConcurrentHashMap<>();

  public void bar() {
    synchronized(map) {  // Noncompliant
      map.put("foo", "bar");
    }
    // Do something ...
  }
}
----


== Compliant Solution

----
public class Foo {
  private ConcurrentHashMap<String,String> map = new ConcurrentHashMap<>();

  public void bar() {
    map.put("foo", "bar");
    // Do something ...
  }
}
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

include::highlighting.adoc[]

endif::env-github,rspecator-view[]
