=== Noncompliant code example

[source,text]
----
if (a == b) { // Compliant; a reassigned in previous block
  doSomething(b);
}
if (a == b) {  // Noncompliant; is this really what was intended?
  doTheThing(c);
}
----
