== Why is this an issue?

include::description.adoc[]

=== Noncompliant code example

[source,text]
----
if (a == b) { // Compliant; a reassigned in previous block
  doSomething(b);
}
if (a == b) {  // Noncompliant; is this really what was intended?
  doTheThing(c);
}
----


=== Compliant solution

[source,text]
----
if (a == b) { 
  doTheThing(b);
  doTheThing(c);
}
----
or

[source,text]
----
if (a == b) { 
  doTheThing(b);
}
if (b == c) {
  doTheThing(c);
}
----


include::exceptions.adoc[]
