<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>The meaning of a boolean parameter may seem perfectly clear when you first write a method call, but that meaning is likely to fade for you over time, and could be completely opaque to those who come behind you.</p>
</div>
<div class="paragraph">
<p>Instead, object literals should be used.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-javascript" data-lang="javascript">widget.repaint(false);  // Noncompliant; does this mean never repaint?</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-javascript" data-lang="javascript">widget.repaint({immediate: false});</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Use an object literal instead.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_relates_to_s2495">relates to: <a data-rspec-id="S2495" class="rspec-auto-link">S2495</a></h3>

</div>
<div class="sect2">
<h3 id="_on_29_jan_2015_184841_ann_campbell_wrote">on 29 Jan 2015, 18:48:41 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>Appropriate for any of your other langauges, [~linda.martin]?</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_13_nov_2015_144841_linda_martin_wrote">on 13 Nov 2015, 14:48:41 Linda Martin wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] Sorry for the delay, I would say yes as it is a general pattern.</p>
</div>
</div>
</div>
</div>