This rule raises an issue when using the two-argument form of `raise` with an explicit `RuntimeError` class name.

== Why is this an issue?

In Ruby, when you use the `raise` method with just a message string, it automatically raises a `RuntimeError` exception. This means that explicitly specifying `RuntimeError` as the exception class is redundant.

The two-argument form `raise RuntimeError, "message"` and the single-argument form `raise "message"` produce exactly the same result. The explicit specification of `RuntimeError` adds unnecessary verbosity to your code without providing any additional functionality or clarity.

Ruby's design philosophy emphasizes conciseness and readability. By omitting the redundant `RuntimeError`, your code becomes more concise while maintaining the same behavior. This follows established Ruby style conventions and makes your code more idiomatic.

The redundant specification can also make the code slightly harder to read, as developers need to process the extra information that doesn't add value. Clean, concise code is generally easier to understand and maintain.

=== What is the potential impact?

This issue has minimal impact on functionality since both forms produce identical behavior. However, it affects code readability and maintainability by introducing unnecessary verbosity. Following consistent style conventions helps maintain clean, readable codebases that are easier for teams to work with.

== How to fix it

Remove the explicit `RuntimeError` class name and use only the message string. Ruby will automatically raise a `RuntimeError` when you provide just a message to the `raise` method.

=== Code examples

==== Noncompliant code example

[source,ruby,diff-id=1,diff-type=noncompliant]
----
raise RuntimeError, "Something went wrong" # Noncompliant
----

==== Compliant solution

[source,ruby,diff-id=1,diff-type=compliant]
----
raise "Something went wrong"
----

== Resources

=== Documentation

 * Ruby Documentation - Exception - https://ruby-doc.org/core/Exception.html[Official Ruby documentation for the Exception class and raise method]

 * Ruby Documentation - Kernel#raise - https://ruby-doc.org/core/Kernel.html#method-i-raise[Official documentation for the raise method showing different usage forms]
