== Why is this an issue?

A function (whether it is part of the standard library, a third party library or a user defined function) may provide some means of indicating the occurrence of an error. This may be via a global error flag, a parametric error flag, a special return value or some other means. Whenever such a mechanism is provided by a function the calling program shall check for the indication of an error as soon as the function returns.


Note, however, that the checking of input values to functions is considered a more robust means of error prevention than trying to detect errors after the function has completed.


=== Noncompliant code example

[source,cpp]
----
extern void fn3 ( int32_t i, bool & flag );
int32_t fn1 ( int32_t i )
{
  int32_t result = 0;
  bool success = false;
  fn3 ( i, success ); // Non-compliant - success not checked
  return result;
}
----


=== Compliant solution

[source,cpp]
----
int32_t fn2 ( int32_t i )
{
  int32_t result = 0;
  bool success = false;
  fn3 ( i, success ); // Compliant - success checked
  if ( !success )
  {
    // ...
  }
  return result;
}
----


== Resources

* MISRA C:2004, 16.10 - If a function returns error information, then that error information shall be tested.
* MISRA {cpp}:2008, 0-3-2 - If a function generates error information, then that error information shall be tested.


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Check the returned status of the called function


'''
== Comments And Links
(visible only on this page)

=== on 15 Oct 2014, 07:27:27 Samuel Mercier wrote:
probably cannot be implemented, as this requires understanding how the status is returned (parameter or return value)

=== on 16 Oct 2014, 14:41:34 Ann Campbell wrote:
\[~samuel.mercier], the message may need work. The description makes clear that there are a number of places the error indication could be in - not just the return value.

endif::env-github,rspecator-view[]
