== Why is this an issue?

It is possible to use the increment operator ``{plus}{plus}``, to set the value of a ``++bool++``({cpp}) or ``++_Bool++``+(C)+ variable to ``++true++``. But this feature has been deprecated in {cpp} since the 1998 version of the standard, removed in {cpp}17, and even where allowed, is simply confusing. 


=== Noncompliant code example

[source,cpp]
----
bool alive;
...
alive++;
----


=== Compliant solution

[source,cpp]
----
bool alive;
...
alive = true;
----


== Resources

*  ISO/IEC 14882:1998, 5.3.2


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
