The ABAP documentation is pretty clear on this subject :

____
This statement is only for 


!!! Internal use in SAP Basis development !!! 


Even within SAP Basis, it may only be used in programs within the ABAP+GUI development group. 


Its use is subject to various restrictions, not all of which may be listed in the documentation. This documentation is intended for internal SAP use within the Basis development group ABAP+GUI. 


Changes and further development, which may be incompatible, may occur at any time, without warning or notice! 

____


== Noncompliant Code Example

----
SYSTEM-CALL CREATE CLASS c. 
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

endif::env-github,rspecator-view[]
