<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Sharing some naming conventions enables teams to collaborate more efficiently. This rule checks that all <code>struct</code> names match a provided regular expression.</p>
</div>
<div class="paragraph">
<p>Using the default regular expression:  "^[A-Z][a-zA-Z0-9]*$"</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-swift" data-lang="swift">struct my_struct {
    var one : Int
    var two : Int
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-swift" data-lang="swift">struct MyStruct {
    var one : Int
    var two : Int
}</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Rename this struct to match the regular expression ${format}</p>
</div>
</div>
<div class="sect2">
<h3 id="_parameters">Parameters</h3>
<div class="sidebarblock">
<div class="content">
<div class="title">format</div>
<div class="paragraph">
<p><em>STRING</em></p>
</div>
<div class="listingblock">
<div class="content">
<pre>^[A-Z][a-zA-Z0-9]*$</pre>
</div>
</div>
<div class="paragraph">
<p>Regular expression struct names should match.</p>
</div>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_29_apr_2014_150324_massimo_paladin_wrote">on 29 Apr 2014, 15:03:24 Massimo PALADIN wrote:</h3>
<div class="paragraph">
<p>I believe that the default value for the regular expression is wrong.</p>
</div>
<div class="paragraph">
<p>From the example I understand you would put small case letters separated by <code>_</code>, are you sure you don&#8217;t want camel case style?</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_29_apr_2014_183533_ann_campbell_wrote">on 29 Apr 2014, 18:35:33 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>I personally prefer camelCase [~massimo.paladin] but set the default regex based on what I thought was the generally accepted standard for C/C&#43;&#43;. However, as I try to re-find what I based that off of, I&#8217;m coming up blank &amp; coming instead to the conclusion that this is holy war territory. If you feel strongly, you can change the regex &amp; flip the code samples. Just either update the RSpec or notify me to do so.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_30_apr_2014_072958_massimo_paladin_wrote">on 30 Apr 2014, 07:29:58 Massimo PALADIN wrote:</h3>
<div class="paragraph">
<p>I would propose to use the same which is used for class names in <a data-rspec-id="S101" class="rspec-auto-link">RSPEC-101</a>.</p>
</div>
<div class="paragraph">
<p>For C&#43;&#43; in practice for classes we have <code>[A-Z][a-zA-Z0-9]+</code>.</p>
</div>
<div class="paragraph">
<p><code>[A-Z][a-zA-Z0-9]+</code> vs <code>[A-Z][a-zA-Z0-9]*</code> I have not much strong preference.</p>
</div>
<div class="paragraph">
<p>If we agree I can modify the description.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_30_apr_2014_114535_ann_campbell_wrote">on 30 Apr 2014, 11:45:35 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>I&#8217;ve updated the spec, [~massimo.paladin].</p>
</div>
</div>
</div>
</div>