== How to fix it in CryptoSwift

=== Code examples

include::../../common/fix/code-rationale.adoc[]

==== Noncompliant code example

[source,swift,diff-id=2,diff-type=noncompliant]
----
import CryptoSwift

func example(password: String) {
    PKCS5.PBKDF2(password: password, salt: [0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00]) // Noncompliant
}
----

==== Compliant solution

[source,swift,diff-id=2,diff-type=compliant]
----
import CryptoSwift

func example(password: String) {
    var secureGenerator = SystemRandomNumberGenerator()
    let randomBytes: [UInt8] = (0..<32).map { _ in secureGenerator.next() }
    PKCS5.PBKDF2(password: password, salt: randomBytes)
}
----

=== How does this work?

include::../../common/fix/salt.adoc[]

Here, the compliant code example ensures the salt is random and has a sufficient
length by calling the `SystemRandomNumberGenerator` function. This one internally uses a
cryptographically secure pseudo random number generator.
