== How to fix it in IDZSwiftCommonCrypto

=== Code examples

include::../../common/fix/code-rationale.adoc[]

==== Noncompliant code example

[source,swift,diff-id=3,diff-type=noncompliant]
----
import IDZSwiftCommonCrypto

func example(password: String) {
    PBKDF.deriveKey(password: password, salt: [0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00], prf: .sha256, rounds: 600000, derivedKeyLength: 32) // Noncompliant
}
----

==== Compliant solution

[source,swift,diff-id=3,diff-type=compliant]
----
import IDZSwiftCommonCrypto

func example(password: String) {
    var secureGenerator = SystemRandomNumberGenerator()
    let randomBytes: [UInt8] = (0..<32).map { _ in secureGenerator.next() 

    PBKDF.deriveKey(password: password, salt: randomSalt, prf: .sha256, rounds: 600000, derivedKeyLength: 32)
}
----

=== How does this work?

include::../../common/fix/salt.adoc[]

Here, the compliant code example ensures the salt is random and has a sufficient
length by calling the `SystemRandomNumberGenerator` function. This one internally uses a
cryptographically secure pseudo random number generator.
