== How to fix it in Swift Crypto

=== Code examples

include::../../common/fix/code-rationale.adoc[]

==== Noncompliant code example

[source,swift,diff-id=4,diff-type=noncompliant]
----
import _CryptoExtras

func example(password: String) {
    KDF.Insecure.PBKDF2.deriveKey(from: password, salt: [0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00], using: .sha256, outputByteCount: 32, rounds: 600000) // Noncompliant
}
----

==== Compliant solution

[source,swift,diff-id=4,diff-type=compliant]
----
import _CryptoExtras

func example(password: String) {
    var secureGenerator = SystemRandomNumberGenerator()
    let randomBytes: [UInt8] = (0..<32).map { _ in secureGenerator.next() 

    KDF.Insecure.PBKDF2.deriveKey(from: password, salt: randomBytes, using: .sha256, outputByteCount: 32, rounds: 600000)
}
----

=== How does this work?

include::../../common/fix/salt.adoc[]

Here, the compliant code example ensures the salt is random and has a sufficient
length by calling the `SystemRandomNumberGenerator` function. This one internally uses a
cryptographically secure pseudo random number generator.
