include::description.adoc[]

== Noncompliant Code Example
[source,terraform]
----
resource "google_compute_ssl_policy" "example" {
  name            = "example"
  min_tls_version = "TLS_1_2" 
  profile         = "COMPATIBLE" # Noncompliant
}
----

== Compliant Solution
[source,terraform]
----
resource "google_compute_ssl_policy" "example" {
  name            = "example"
  min_tls_version = "TLS_1_2" 
  profile         = "RESTRICTED"
}
----

include::see.adoc[]


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

include::highlighting.adoc[]

endif::env-github,rspecator-view[]
