<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when a custom event is created with a name that starts with the prefix 'on'.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>In Lightning Web Components, event handler names conventionally start with 'on' (like <code>onclick</code>, <code>onchange</code>, etc.). When custom event names also start with 'on', it creates confusion and inconsistency.</p>
</div>
<div class="paragraph">
<p>For example, if you create a custom event named 'onselect', the corresponding event handler would need to be named 'ononselect', which is confusing and breaks the expected naming pattern.</p>
</div>
<div class="paragraph">
<p>This naming convention issue makes the code harder to read and understand, especially for developers who are familiar with standard web development practices where the 'on' prefix is reserved for event handlers, not event names themselves.</p>
</div>
<div class="paragraph">
<p>Following consistent naming conventions helps maintain code clarity and makes it easier for team members to understand the codebase.</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="paragraph">
<p>This issue affects code maintainability and readability. While it doesn&#8217;t cause runtime errors, it can lead to confusion among developers and makes the code harder to understand and maintain. It also goes against established Lightning Web Components best practices.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Remove the 'on' prefix from the custom event name. The event name should describe the action without the 'on' prefix.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-javascript" data-lang="javascript">sendData(){
    const customEvt = new CustomEvent('onselect'); // Noncompliant
    this.dispatchEvent(customEvt);
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-javascript" data-lang="javascript">sendData(){
    const customEvt = new CustomEvent('select');
    this.dispatchEvent(customEvt);
}</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Lightning Web Components Best Practices - <a href="https://www.apexhours.com/20-tips-for-lightning-web-components">Comprehensive guide covering 20 tips for Lightning Web Components including naming conventions</a></p>
</li>
<li>
<p>Lightning Web Components Developer Guide - Events - <a href="https://developer.salesforce.com/docs/component-library/documentation/en/lwc/lwc.events">Official Salesforce documentation on handling events in Lightning Web Components</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>