include::../description.adoc[]

== Noncompliant Code Example

----
if (a == b)
{
  doTheThing(b);
}
if (a == b) // Noncompliant; is this really what was intended?
{
  doTheThing(c);
}
----

== Compliant Solution

----
if (a == b)
{
  doTheThing(b);
  doTheThing(c);
}
----
or

----
if (a == b)
{
  doTheThing(b);
}
if (b == c)
{
  doTheThing(c);
}
----

include::../exceptions.adoc[]

ifdef::env-github,rspecator-view[]
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
