== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,csharp]
----
if (a == b)
{
  doTheThing(b);
}
if (a == b) // Noncompliant; is this really what was intended?
{
  doTheThing(c);
}
----

=== Compliant solution

[source,csharp]
----
if (a == b)
{
  doTheThing(b);
  doTheThing(c);
}
----
or

[source,csharp]
----
if (a == b)
{
  doTheThing(b);
}
if (b == c)
{
  doTheThing(c);
}
----

include::../exceptions.adoc[]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

=== on 25 Jun 2015, 13:00:10 Tamas Vajk wrote:
\[~ann.campbell.2] could you check the wording of this language specific exception?

=== on 25 Jun 2015, 17:37:06 Ann Campbell wrote:
I've edited [~tamas.vajk], but I believe it semantically the same. :-)

=== on 26 Jun 2015, 06:31:15 Tamas Vajk wrote:
\[~ann.campbell.2] Thanks.

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
