== Why is this an issue?

In Dart, there are no keywords like `public`, `protected`, and `private`. To distinguish between public and private there's a convention, if an identifier starts with an underscore (_), it's private to its library. In the context of library prefixes there's no notion of private, so starting it with underscore will only be source of confusion for the developers. That's why it's recommended to not start the library prefix with an underscore ``++_++``.

=== Code examples

==== Noncompliant code example

[source,dart,diff-id=1,diff-type=noncompliant]
----
import 'dart:io' as _io;
import 'dart:math' as _math;
import 'dart:js' as _js;
----

==== Compliant solution

[source,dart,diff-id=1,diff-type=compliant]
----
import 'dart:io' as io;
import 'dart:math' as math;
import 'dart:js' as js;
----

== Resources

=== Documentation

* Dart Docs - https://dart.dev/tools/linter-rules/no_leading_underscores_for_library_prefixes[Dart Linter rule - no_leading_underscores_for_library_prefixes]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

* The library prefix <prefix> starts with an underscore.

=== Highlighting

Library prefix

'''
== Comments And Links
(visible only on this page)

endif::env-github,rspecator-view[]

