<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Because semicolons at the ends of statements are optional, starting function call arguments on a separate line makes the code confusing. It could lead to errors and most likely <em>will</em> lead to questions for maintainers.</p>
</div>
<div class="paragraph">
<p>What was the initial intent of the developer?</p>
</div>
<div class="olist arabic">
<ol class="arabic">
<li>
<p>Define a function and then execute some unrelated code inside a closure ?</p>
</li>
<li>
<p>Pass the second function as a parameter to the first one ?</p>
</li>
</ol>
</div>
<div class="paragraph">
<p>The first option will be the one chosen by the JavaScript interpreter.</p>
</div>
<div class="paragraph">
<p>By extension, and to improve readability, any kind of function call argument should not start on new line.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-flex" data-lang="flex">var fn = function () {
  //...
}

(function () { // Noncompliant
  //...
})();</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="paragraph">
<p>Either</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-flex" data-lang="flex">// define a function
var fn = function () {
  //...
}; // &lt;-- semicolon added

// then execute some code inside a closure
(function () {
  //...
})();</code></pre>
</div>
</div>
<div class="paragraph">
<p>Or</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-flex" data-lang="flex">var fn = function () {
  //...
}(function () { // &lt;-- start function call arguments on same line
  //...
})();</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Review this invocation.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>Primary: arguments clause</p>
</div>
<div class="paragraph">
<p>Secondary: callee</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_relates_to_s1438">relates to: <a data-rspec-id="S1438" class="rspec-auto-link">S1438</a></h3>

</div>
<div class="sect2">
<h3 id="_on_26_nov_2013_110322_freddy_mallet_wrote">on 26 Nov 2013, 11:03:22 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>Is implemented by https://jira.codehaus.org/browse/SONARPLUGINS-3292 for Javascript</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_26_nov_2013_110422_freddy_mallet_wrote">on 26 Nov 2013, 11:04:22 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>Is implemented by https://jira.codehaus.org/browse/SONARPLUGINS-3293 for Flex</p>
</div>
</div>
</div>
</div>