<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p><code>GC.SuppressFinalize</code> asks the Common Language Runtime not to call the finalizer of an object. This is useful when implementing the dispose pattern where object finalization is already handled in <code>IDisposable.Dispose</code>. However, it has no effect if there is no finalizer defined in the object&#8217;s type, so using it in such cases is just confusing.</p>
</div>
<div class="paragraph">
<p>This rule raises an issue when <code>GC.SuppressFinalize</code> is called for objects of <code>sealed</code> types without a finalizer.</p>
</div>
<div class="paragraph">
<p><strong>Note:</strong> <a data-rspec-id="S3971" class="rspec-auto-link">S3971</a> is a stricter version of this rule. Typically it makes sense to activate only one of these 2 rules.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-csharp" data-lang="csharp">sealed class MyClass
{
  public void Method()
  {
    ...
    GC.SuppressFinalize(this); //Noncompliant
  }
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-csharp" data-lang="csharp">sealed class MyClass
{
  public void Method()
  {
    ...
  }
}</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove this useless call to "GC.SuppressFinalize".</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_is_related_to_s3971">is related to: <a data-rspec-id="S3971" class="rspec-auto-link">S3971</a></h3>

</div>
<div class="sect2">
<h3 id="_on_7_jul_2015_084300_tamas_vajk_wrote">on 7 Jul 2015, 08:43:00 Tamas Vajk wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] Created the first version, can you go through it? Also, I haven&#8217;t set any SQALE characteristics. Which one fits best?</p>
</div>
</div>
</div>
</div>