== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,ruby]
----
if  !(a == 2)  # Noncompliant
  # ...
end

b = !(a < 10) # Noncompliant
----

=== Compliant solution

[source,ruby]
----
if a != 2
  # ...
end
b = (a >= 10)
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
