== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,javascript]
----
  for (i = 0; i < 10; i++) {
    if (i == 5) {
      continue;  /* Noncompliant */
    }
    alert("i = " + i);
  }
----

=== Compliant solution

[source,javascript]
----
  for (i = 0; i < 10; i++) {
    if (i != 5) {  /* Compliant */
      alert("i = " + i);
    }
  }
----

== Resources
=== Documentation

* https://eslint.org[eslint] - Rule https://eslint.org/docs/latest/rules/no-continue[no-continue]
ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
