This rule raises an issue when an abstract class contains only fields and no methods, making it function as a data container rather than a proper base class.

== Why is this an issue?

Abstract classes are designed to serve as base classes that define common behavior for their subclasses. They should contain abstract or virtual methods that subclasses can implement or override.

When an abstract class contains only fields and no methods, it's not fulfilling its intended purpose. Instead, it's acting as a simple data container that cannot be instantiated. This creates confusion about the class's role and design intent.

Using the `abstract` keyword for a data-only class is misleading because:

* It suggests that the class is meant for inheritance and method overriding
* It doesn't clearly communicate that the class is simply a non-instantiable data container
* Other developers might expect to find abstract methods to implement

A class that only holds data and should not be instantiated is better served by using a private or protected constructor. This approach clearly communicates the design intent and prevents instantiation without the misleading implications of the `abstract` keyword.

=== What is the potential impact?

This design issue can lead to confusion among team members about the class's intended purpose and usage patterns. It may result in incorrect assumptions about inheritance relationships and make the codebase harder to understand and maintain.

== How to fix it

Remove the abstract keyword and add a private or protected constructor to prevent instantiation. This clearly indicates that the class is a data container that should not be instantiated.

=== Code examples

==== Noncompliant code example

[source,apex,diff-id=1,diff-type=noncompliant]
----
public abstract class UserData {
    String username;
    String email;
    Integer userId;
} // Noncompliant
----

==== Compliant solution

[source,apex,diff-id=1,diff-type=compliant]
----
public class UserData {
    String username;
    String email;
    Integer userId;
    
    private UserData() {
        // Private constructor prevents instantiation
    }
}
----

== Resources

=== Documentation

 * Apex Classes - https://developer.salesforce.com/docs/atlas.en-us.apexcode.meta/apexcode/apex_classes.htm[Official Salesforce documentation on Apex classes and abstract classes]

 * Apex Class Definition - https://developer.salesforce.com/docs/atlas.en-us.apexcode.meta/apexcode/apex_classes_defining.htm[Salesforce documentation explaining abstract and virtual classes in Apex]
