<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when multiple independent asynchronous operations are awaited sequentially instead of concurrently.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Awaiting multiple independent asynchronous operations sequentially negates the potential concurrency gains that asynchronous programming provides. If Task B doesn&#8217;t depend on the result of Task A, running them one after the other is inefficient and wastes the opportunity for concurrent execution.</p>
</div>
<div class="paragraph">
<p>This applies to all major Python async frameworks: <code>asyncio</code>, <code>Trio</code>, and <code>AnyIO</code>. Each provides mechanisms to run independent operations concurrently.</p>
</div>
<div class="paragraph">
<p>When independent async operations are awaited sequentially:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>Total execution time is the sum of all individual operation times</p>
</li>
<li>
<p>The event loop cannot interleave execution of the operations</p>
</li>
<li>
<p>Application responsiveness is reduced unnecessarily</p>
</li>
<li>
<p>System resources are underutilized</p>
</li>
</ul>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it_in_asyncio">How to fix it in Asyncio</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Use <code>asyncio.gather()</code> to run operations concurrently.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python"># Noncompliant: Sequential awaiting
result_a = await first_task()
result_b = await second_task()</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python"># Compliant: Concurrent execution
results = await asyncio.gather(
    first_task(),
    second_task()
)</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it_in_trio">How to fix it in Trio</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Use nurseries to manage concurrent execution. Nurseries provide structured concurrency where all spawned tasks are automatically managed.</p>
</div>
<div class="sect2">
<h3 id="_code_examples_2">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example_2">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python"># Noncompliant: Sequential awaiting
result_a = await first_task()
result_b = await second_task()</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution_2">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python"># Compliant: Concurrent execution with nursery
async with trio.open_nursery() as nursery:
    nursery.start_soon(first_task)
    nursery.start_soon(second_task)</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it_in_anyio">How to fix it in AnyIO</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Use task groups to manage concurrent execution. Task groups provide similar functionality to asyncio.gather() but with structured concurrency.</p>
</div>
<div class="sect2">
<h3 id="_code_examples_3">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example_3">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python"># Noncompliant: Sequential awaiting
result_a = await first_task()
result_b = await second_task()</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution_3">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python"># Compliant: Concurrent execution with task group
async with anyio.create_task_group() as tg:
    tg.start_soon(first_task)
    tg.start_soon(second_task)</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Python asyncio - <a href="https://docs.python.org/3/library/asyncio-task.html#asyncio.gather">asyncio.gather()</a></p>
</li>
<li>
<p>Python asyncio - <a href="https://docs.python.org/3/library/asyncio-task.html#asyncio.create_task">asyncio.create_task()</a></p>
</li>
<li>
<p>Trio - <a href="https://trio.readthedocs.io/en/stable/reference-core.html#nurseries-and-spawning">Nurseries and spawning</a></p>
</li>
<li>
<p>AnyIO - <a href="https://anyio.readthedocs.io/en/stable/tasks.html#creating-and-managing-tasks">Creating and managing tasks</a></p>
</li>
</ul>
</div>
</div>
<div class="sect2">
<h3 id="_articles_blog_posts">Articles &amp; blog posts</h3>
<div class="ulist">
<ul>
<li>
<p>Real Python - <a href="https://realpython.com/python-concurrency/">Concurrency and Parallelism in Python</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Await these independent operations concurrently using {<code>asyncio.gather()</code>, nurseries, task groups}.</p>
</div>
<div class="paragraph">
<p>If framework can be detected:
- For asyncio: Await these independent operations concurrently using <code>asyncio.gather()</code>.
- For trio: Use Nursery to run these operations concurrently.
- For anyio: Use TaskGroup to run these operations concurrently.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="ulist">
<ul>
<li>
<p>Primary location: First sequential await of independent operations</p>
</li>
<li>
<p>Secondary locations: Additional sequential awaits in the same context</p>
</li>
<li>
<p>Secondary locations: The async keyword of the enclosing function</p>
</li>
</ul>
</div>
</div>
</div>
</div>