<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Multiple spaces in a regular expression can make it hard to tell how many spaces should be matched. It&#8217;s more readable to use only one space and then indicate with a quantifier how many spaces are expected.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-php" data-lang="php">"/Hello,   world!/"</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-php" data-lang="php">"/Hello, {3}world!/"</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_exceptions">Exceptions</h3>
<div class="paragraph">
<p>If the extended modifier is set, whitespaces are ignored. In this case no issue should be triggered, because the whitespaces may be intended to improve readability.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_message">Message</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Replace spaces with quantifier <code>{n}</code>.</p>
</div>
<div class="paragraph">
<p>(n should be dynamic and be the number of spaces)</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_highlighting">Highlighting</h2>
<div class="sectionbody">
<div class="paragraph">
<p>The spaces that would be replaced with a quantifier. For example in 3 spaces, highlight the last 2.</p>
</div>
<hr>
</div>
</div>