Include guards wrap around the entire contents of a header file and ensure that no matter how many times the file is actually included, its contents are only defined once. Because multiple, potentially conflicting definitions could lead to errors, the use of include guards is a best practice.


== Noncompliant Code Example

----
//my_header.h
...
----


== Compliant Solution

----
//my_header.h
#ifndef MY_HEADER
#define MY_HEADER
...
#endif
----
or 

----
//my_header.h
#if !defined(MY_HEADER)
#define MY_HEADER
 ...
#endif
----


== See

* MISRA C:2004, 19.15
* MISRA {cpp}:2008, 16-2-3 - Include guards shall be provided.
* MISRA C:2012, Directive 4.10
* https://wiki.sei.cmu.edu/confluence/x/G9YxBQ[CERT, PRE06-C] - Enclose header files in an include guard


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
