<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>If the type of a <code>props</code> property is not defined there is a great chance one will wrongly use the value returned by the property and generate a not expected behavior of the application.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-javascript" data-lang="javascript">class Rating extends React.Component {
  render() {
    return ( &lt;div&gt;{this.props.rating}&lt;/div&gt; );
  }
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-javascript" data-lang="javascript">import PropTypes from 'prop-types';

class Rating extends React.Component {
  render() {
    return ( &lt;div&gt;{this.props.rating}&lt;/div&gt; );
  }
}

Rating.propTypes = {
  rating: PropTypes.number.isRequired
};</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_23_feb_2018_115713_alexandre_gigleux_wrote">on 23 Feb 2018, 11:57:13 Alexandre Gigleux wrote:</h3>
<div class="paragraph">
<p>Covered by ESLint for Reach: https://github.com/yannickcr/eslint-plugin-react/blob/HEAD/docs/rules/prop-types.md (react/prop-types)</p>
</div>
</div>
</div>
</div>