== Why is this an issue?

jQuery doesn't cache elements for you. If you've made a selection that you might need to make again, you should save the selection in a variable rather than making the selection repeatedly.


=== Noncompliant code example

With the default threshold of 2

[source,javascript]
----
$( "p" ).hide();
$( "p" ).show();  // Noncompliant
----


=== Compliant solution

[source,javascript]
----
var paragraph = $( "p" );

paragraph.hide();
paragraph.show();
----


=== Exceptions

Stored selections are not updated when the DOM changes. Since variables may need to updated this rule ignores selections that are repeated during an assignment.

[source,javascript]
----
var paragraph = $("p");

// ...

paragraph = $("p");
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

include::parameters.adoc[]

'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
