<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>The exception handling mechanism can provide an effective and clear means of handling error conditions, particularly where a function needs to return both some desired result together with an indication of success or failure. However, because of its ability to transfer control back up the call tree, it can also lead to code that is difficult to understand. Hence it is required that the mechanism is only used to capture behaviour that is in some sense undesirable, and which is not expected to be seen in normal program execution.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-text" data-lang="text">void fn ( )
{
  try
  {
    if ( x &lt; 10 )
    {
      throw ( 10 );
    }
    // Action "A"
  }
  catch ( int32_t y ) // Noncompliant, confusing way of handling x &lt; 10
  {
    // Action "B"
  }
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-text" data-lang="text">if ( x &lt; 10 )
{
  // Action "B"
}
else
{
  // Action "A"
}</code></pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p>MISRA C&#43;&#43;:2008, 15-0-1</p>
</li>
</ul>
</div>
<hr>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove this confusing thrown exception.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_14_oct_2014_135423_samuel_mercier_wrote">on 14 Oct 2014, 13:54:23 Samuel Mercier wrote:</h3>
<div class="paragraph">
<p>This rule cannot be implemented, as it relies on interpretation of the programmer&#8217;s intent.</p>
</div>
<div class="paragraph">
<p>We could however make an approximation by forbidding exceptions to be caught within the same function they are thrown from.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_14_oct_2014_184903_ann_campbell_wrote">on 14 Oct 2014, 18:49:03 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~samuel.mercier] I suspect this Microsoft rule is a variant on the MISRA intent: <a href="http://msdn.microsoft.com/en-us/library/dd264997.aspx">DA0007: Avoid using exceptions for control flow</a></p>
</div>
<div class="paragraph">
<p>I&#8217;ve also just seen an example of an <a href="http://www.gamedev.net/topic/445918-using-exceptions-to-control-the-flow-of-a-program-a-good-idea/">exception used in indirect recursion</a> <em>shudder</em></p>
</div>
</div>
<div class="sect2">
<h3 id="_on_14_oct_2014_201211_freddy_mallet_wrote">on 14 Oct 2014, 20:12:11 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>Indeed [~samuel.mercier], for sure if some exceptions are thrown and caught in the same function body, that means that exceptions are used for control flow. That&#8217;s a good way to partially cover this rule.</p>
</div>
</div>
</div>
</div>