<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p><code>#include_next</code> is a gcc-specific language extension that alters the search path for the specified header file by starting the search from the header file directory <em>after</em> the one in which the directive was encountered. It also ignores the distinction between <code>"file"</code> and <code>&lt;file&gt;</code>. It is typically used when you have two (probably related) header files with the same name, although there is nothing in the extension to enforce or limit the use to same-name files.</p>
</div>
<div class="paragraph">
<p>Use of this extension can be tricky to get right, and is almost never justified. Instead, you should use an absolute path in the <code>#include</code> statement or rename one of the files.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">#include_next "foo.h" // Noncompliant</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">#include "/usr/local/include/foo.h"</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove this use of "#include_next".</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p><code>#include_next</code></p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_27_sep_2016_155104_ann_campbell_wrote">on 27 Sep 2016, 15:51:04 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~jeanchristophe.collet] please explain what <code>#include_next</code> does, and what should be used instead. Please also demonstrate that in a compliant solution</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_27_sep_2016_164720_ann_campbell_wrote">on 27 Sep 2016, 16:47:20 Ann Campbell wrote:</h3>
<div class="paragraph">
<p><a href="https://gcc.gnu.org/onlinedocs/cpp/Wrapper-Headers.html" class="bare">https://gcc.gnu.org/onlinedocs/cpp/Wrapper-Headers.html</a></p>
</div>
</div>
<div class="sect2">
<h3 id="_on_27_sep_2016_165931_ann_campbell_wrote">on 27 Sep 2016, 16:59:31 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>double-check me, please [~jeanchristophe.collet]</p>
</div>
</div>
</div>
</div>