The PHP 4 method of declaring a variable, using the ``++var++`` keyword, was deprecated in early versions of PHP 5. Even though it's not considered deprecated in the most recent versions, it's nonetheless not best practice to use it. When ``++var++`` does appear, it is interpreted as a synonym for ``++public++`` and treated as such. Therefore ``++public++`` should be used instead.


From the PHP Manual:

____
The PHP 4 method of declaring a variable with the var keyword is still supported for compatibility reasons (as a synonym for the public keyword). In PHP 5 before 5.1.3, its usage would generate an E_STRICT warning.

____


== Noncompliant Code Example

----
<?php
class Foo
{
    var $bar = 1;
}
----


== Compliant Solution

----
<?php
class Foo
{
    public $bar = 1;
}
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
