Attempting to make comparisons between pointers will produce undefined behaviour if the two pointers do not point to the same object.


Note: it is permissible to address the next element beyond the end of an array, but accessing this element is not allowed.


== Noncompliant Code Example

----
void f1 ( )
{
 int32_t a1[ 10 ];
 int32_t a2[ 10 ];
 int32_t * p1 = a1;
 if ( p1 < a2 ) // Non-compliant, p1 and a2 are unrelated
 {
 }
}
----


== Compliant Solution

----
void f1 ( )
{
 int32_t a1[ 10 ];
 int32_t * p1 = a1;
 if ( p1 < a1 ) // Compliant, p1 points to an element of a1
 {
 }
}
----


== See

* MISRA C:2004, 17.3 - >, >=, <, +<=+ shall not be applied to pointer types except where they point to the same array.
* MISRA {cpp}:2008, 5-0-18 - >, >=, <, +<=+ shall not be applied to objects of pointer type, except where they point to the same array.


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
