== Why is this an issue?

Attempting to make comparisons between pointers will produce undefined behaviour if the two pointers do not point to the same object.


Note: it is permissible to address the next element beyond the end of an array, but accessing this element is not allowed.


=== Noncompliant code example

[source,cpp]
----
void f1 ( )
{
 int32_t a1[ 10 ];
 int32_t a2[ 10 ];
 int32_t * p1 = a1;
 if ( p1 < a2 ) // Non-compliant, p1 and a2 are unrelated
 {
 }
}
----


=== Compliant solution

[source,cpp]
----
void f1 ( )
{
 int32_t a1[ 10 ];
 int32_t * p1 = a1;
 if ( p1 < a1 ) // Compliant, p1 points to an element of a1
 {
 }
}
----


== Resources

* MISRA C:2004, 17.3 - >, >=, <, +<=+ shall not be applied to pointer types except where they point to the same array.
* MISRA {cpp}:2008, 5-0-18 - >, >=, <, +<=+ shall not be applied to objects of pointer type, except where they point to the same array.


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Remove this illegal comparison of unrelated pointers.


'''
== Comments And Links
(visible only on this page)

=== relates to: S5658

=== on 15 Oct 2014, 20:46:03 Ann Campbell wrote:
\[~samuel.mercier] please:

* fill in the appropriate reference field(s).
* provide a See section.


=== on 17 Oct 2014, 13:42:29 Ann Campbell wrote:
\[~samuel.mercier] I wonder if the message should include the names of the illegally-compared pointers? I ask this only because it could occur in a "busy" line of code.

E.G.

``++if (a < b && c > d) ++`` where a & b are related, but c & d are not.

=== on 17 Jul 2015, 11:15:07 Ann Campbell wrote:
See if you approve of my title edit [~evgeny.mandrikov]

=== on 28 Sep 2015, 20:50:03 Evgeny Mandrikov wrote:
\[~ann.campbell.2] For me "!=" is a comparison, so that original title allows "p1 != p2", even if they don't point on the same array, while new one forbids. What about usage of a term from C and {cpp} specification: <, >, +<=+ and >= are collectively called "relational operators", == and != are called "equality operators"?

=== on 29 Sep 2015, 12:28:16 Ann Campbell wrote:
updated

=== on 4 Feb 2020, 12:13:22 Lo√Øc Joly wrote:
Similar to RSPEC-5658, but strict MISRA version (RSPEC-5658 is broader)

endif::env-github,rspecator-view[]
