== Noncompliant Code Example

----
Class Example
    Private obj As Object = New Object()

    Public Sub DoSomethingWithMonitor()
        Monitor.Enter(obj) ' Noncompliant

        If IsInitialized() Then
            ' ..
            Monitor.[Exit](obj)
        End If
    End Sub

    Private lockObj As ReaderWriterLockSlim = New ReaderWriterLockSlim()

    Public Sub DoSomethingWithReaderWriteLockSlim()
        lockObj.EnterReadLock() ' Noncompliant
        ' do something without releasing in the same method
    End Sub
End Class
----
