== Why is this an issue?

include::../description-dotnet.adoc[]

=== Noncompliant code example

[source,vbnet]
----
Class Example
    Private obj As Object = New Object()

    Public Sub DoSomethingWithMonitor()
        Monitor.Enter(obj) ' Noncompliant

        If IsInitialized() Then
            ' ..
            Monitor.Exit(obj)
        End If
    End Sub

    Private lockObj As ReaderWriterLockSlim = New ReaderWriterLockSlim()

    Public Sub DoSomethingWithReaderWriteLockSlim()
        lockObj.EnterReadLock() ' Noncompliant
        If IsInitialized() Then
            ' ..
            lockObj.ExitReadLock()
        End If
    End Sub
End Class
----


=== Compliant solution

[source,vbnet]
----
Class Example
    Private obj As Object = New Object()

    Public Sub DoSomethingWithMonitor()
        SyncLock obj ' is easier to use than explicit Monitor calls
            If IsInitialized() Then 
            End If
        End SyncLock
    End Sub

    Private lockObj As ReaderWriterLockSlim = New ReaderWriterLockSlim()

    Public Sub DoSomethingWithReaderWriteLockSlim()
        lockObj.EnterReadLock()

        Try
            If IsInitialized() Then
            End If
        Finally
            lockObj.ExitReadLock()
        End Try
    End Sub
End Class
----


include::../see.adoc[]

* https://docs.microsoft.com/en-us/dotnet/standard/threading/overview-of-synchronization-primitives#synchronization-of-access-to-a-shared-resource[Synchronization of access to a shared resource].


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
