include::../description.adoc[]

== Noncompliant Code Example

[source,text]
----
import re
pattern = re.compile(r"\1(.)", re.DOTALL) # Noncompliant, group 1 is defined after the back reference
pattern = re.compile(r"(.)\2", re.DOTALL) # Noncompliant, group 2 isn't defined at all
pattern = re.compile(r"(.)|\1", re.DOTALL) # Noncompliant, group 1 and the back reference are in different branches
pattern = re.compile(r"(?P<x>.)|(?P=x)", re.DOTALL) # Noncompliant, group x and the back reference are in different branches
----


== Compliant Solution

[source,text]
----
import re
pattern = re.compile(r"(.)\1", re.DOTALL)
pattern = re.compile(r"(?P<x>.)(?P=x)", re.DOTALL)
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../highlighting.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
