<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Where a set of operators is overloaded, it is important that the interactions between the operators meet developer expectations.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">class A
{
public:
  A&amp; operator= ( A const &amp; rhs );
};

A &amp; operator += ( A const &amp; lhs, A const &amp; rhs );
A const operator + ( A const &amp; lhs, A const &amp; rhs )  // Noncompliant
{
  return lht -= rhs;
}
void f ( A a1, A a2 )
{
  A x;
  x = a1 + a2;
  a1 += a2;
  if ( x == a1 ) // This should be true, but it's not
  {
  }
}</code></pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p>MISRA C&#43;&#43;:2008, 5-17-1</p>
</li>
</ul>
</div>
<hr>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Refactor the definition of opreators "xxx" and "yyy" to yield semantically equivalent result.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_is_related_to_s3083">is related to: <a data-rspec-id="S3083" class="rspec-auto-link">S3083</a></h3>

</div>
<div class="sect2">
<h3 id="_on_23_oct_2014_142452_samuel_mercier_wrote">on 23 Oct 2014, 14:24:52 Samuel Mercier wrote:</h3>
<div class="paragraph">
<p>We can probably not detect that&#8230;&#8203;</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_23_oct_2014_205408_ann_campbell_wrote">on 23 Oct 2014, 20:54:08 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~samuel.mercier] please follow the conventions on this rule anyway. That or close it.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_17_jul_2015_092237_ann_campbell_wrote">on 17 Jul 2015, 09:22:37 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>Already assigned to you [~evgeny.mandrikov]. Note that I&#8217;ve altered the code sample.</p>
</div>
</div>
</div>
</div>