== Why is this an issue?

In Java 8, underscore (``++_++``) cannot be used as a parameter to a lambda, and by Java 9, it's a keyword. To ensure future portability (not to mention current readability), it should not be used now as an identifier.

=== Noncompliant code example

[source,java]
----
private String _;  // Noncompliant
----

=== Compliant solution

[source,java]
----
private String foo;
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Rename this (variable|parameter).


=== Highlighting

variable/param name


'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
