== Why is this an issue?

AssertJ assertions methods targeting the same object can be chained instead of using multiple ``++assertThat++``. It avoids duplication and increases the clarity of the code.


This rule raises an issue when multiples ``++assertThat++`` target the same tested value.


=== Noncompliant code example

[source,java]
----
assertThat(someList).hasSize(3);
assertThat(someList).contains("something");
----


=== Compliant solution

[source,java]
----
assertThat(someList)
  .hasSize(3)
  .contains("something");
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

include::highlighting.adoc[]

endif::env-github,rspecator-view[]
