
include::../../../shared_content/secrets/description.adoc[]

== Why is this an issue?

include::../../../shared_content/secrets/rationale.adoc[]

=== What is the potential impact?

Commercial editions of the Sidekiq library must be downloaded from a Ruby gem
server run by Contributed Systems. Users of these commercial editions are
provided for credentials to these servers.

If your Sidekiq download credentials are exposed, third parties will be able to
download commercial editions of Sidekiq. Unusual patterns of activity may cause
Contibuted Systems to revoke these credentials, preventing legitimate downloads
of these Sidekiq editions. This can prevent developers from doing their work or
can cause CI/CD pipelines to fail.

== How to fix it

include::../../../shared_content/secrets/fix/revoke.adoc[]

include::../../../shared_content/secrets/fix/vault.adoc[]

=== Code examples

==== Noncompliant code example

[source,docker,diff-id=1,diff-type=noncompliant]
----
FROM ubuntu:22.04
ENV BUNDLE_GEMS__CONTRIBSYS__COM=ec04a635:b5f7de0c

COPY src /src
WORKDIR /src
RUN bundle install
----

==== Compliant solution

[source,docker,diff-id=1,diff-type=compliant]
----
FROM ubuntu:22.04
# docker build --secret id=sidekiq,env=SIDEKIQ_CREDENTIALS ...
RUN export BUNDLE_GEMS__CONTRIBSYS__COM=$(cat /run/secrets/sidekiq)

COPY src /src
WORKDIR /src
RUN bundle install
----

//=== How does this work?

//=== Pitfalls

//=== Going the extra mile

== Resources

=== Documentation

* Sidekiq Wiki - https://github.com/sidekiq/sidekiq/wiki/Commercial-FAQ[Commercial FAQ]

include::../../../shared_content/secrets/resources/standards.adoc[]

//=== Benchmarks
