<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This is required by IEC 61508, under good programming style.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">int function1()
{
  return 3;
}

void function2()
{
  function1();
}

int function3(char* ptr) /* Noncompliant; two explicit returns */
{
  if (ptr == NULL) return -1;

  return 7;
}

void function4(char *ptr) /* Noncompliant; two returns, one explicit and one implicit */
{
  if (1) return;

  printf("hello world!\n");
}</code></pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p>MISRA C:2004, 14.7 - A function shall have a single point of exit at the end of the function.</p>
</li>
<li>
<p>MISRA C&#43;&#43;:2008, 6-6-5 - A function shall have a single point of exit at the end of the function</p>
</li>
<li>
<p>MISRA C:2012, 15.5 - A function should have a single point of exit at the end</p>
</li>
</ul>
</div>
<hr>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_5_sep_2014_141941_evgeny_mandrikov_wrote">on 5 Sep 2014, 14:19:41 Evgeny Mandrikov wrote:</h3>
<div class="paragraph">
<p>This rule is controversial (some people prefer early exits from functions). Also based on our experience - hard to satisfy, and so was deactivated on Dory. Thus proposal - remove from default profile.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_5_sep_2014_175126_ann_campbell_wrote">on 5 Sep 2014, 17:51:26 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~evgeny.mandrikov] I made that change on the 21st of August. I can&#8217;t make it any off-er. :-)</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_9_sep_2014_131357_freddy_mallet_wrote">on 9 Sep 2014, 13:13:57 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>Ticket created on C&#43;&#43; project @Ann and @Evgeny : CPP-814</p>
</div>
</div>
</div>
</div>