<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Procedures, unlike functions, do not return values. The <code>RETURN</code> statement therefore, when used within a procedure, is used to prematurely end the procedure. However, having multiple exit points (i.e. more than the <code>END</code> of the procedure itself), increases the complexity of the procedure and makes it harder to understand and debug.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-sql" data-lang="sql">SET SERVEROUTPUT ON

DECLARE
  PROCEDURE prcoedureWithReturn AS
  BEGIN
    RETURN; -- Noncompliant

    DBMS_OUTPUT.PUT_LINE('prcoedureWithReturn called'); -- This is actually unreachable
  END;
BEGIN
  prcoedureWithReturn;
END;
/</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove this "RETURN" statement.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_5_sep_2014_141941_evgeny_mandrikov_wrote">on 5 Sep 2014, 14:19:41 Evgeny Mandrikov wrote:</h3>
<div class="paragraph">
<p>This rule is controversial (some people prefer early exits from functions). Also based on our experience - hard to satisfy, and so was deactivated on Dory. Thus proposal - remove from default profile.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_5_sep_2014_175126_ann_campbell_wrote">on 5 Sep 2014, 17:51:26 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~evgeny.mandrikov] I made that change on the 21st of August. I can&#8217;t make it any off-er. :-)</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_9_sep_2014_131357_freddy_mallet_wrote">on 9 Sep 2014, 13:13:57 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>Ticket created on C&#43;&#43; project @Ann and @Evgeny : CPP-814</p>
</div>
</div>
</div>
</div>