The ``++print++`` statement was removed in Python 3.0. The built-in function should be used instead.


== Noncompliant Code Example

----
print '1'  # Noncompliant
----


== Compliant Solution

----
print('1') 
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

endif::env-github,rspecator-view[]
