== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,swift]
----
func getReadableStatus(job: Job) -> String {
  return job.isRunning ? "Running" : job.hasErrors ? "Failed" : "Succeeded";  // Noncompliant
}
----

=== Compliant solution

[source,swift]
----
func getReadableStatus(job: Job) -> String {
  let status: String;
  if (job.isRunning) {
     status = "Running";
  }
  else {
     status = job.hasErrors ? "Failed" : "Succeeded"; 
  }
  return status;
}
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

include::../highlighting.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
