Javadoc code snippets behave like text-blocks in terms of padding. It is the position of the ending curly brace which defines how many spaces should be placed before each line of the code snippet. Consequently, when writing code snippets, it is recommended to align the curly brace closing the snippet with the leftmost character present in the code. This will minimize the number of spaces used in front of each line in the generated javadoc.

== Noncompliant Code Example

[source,java]
----
/**
 * {@snippet :
 *   class HelloWorld { ... }
 * } // Noncompliant - aligned with the snippet's closing bracket, but will force 2 spaces before each lines of the snippet
 */
void wrongIndentation();
----

== Compliant Solution

[source,java]
----
/**
 * Highlighting based on a substring on a single line.
 * {@snippet :
 *   class HelloWorld { ... }
 *   } // Compliant - aligned with the last curly braces and the class definition starting column of the code above
 */
void correctIndentation();
----

== See

Oracleâ€™s Programmer's Guide to Snippets: https://docs.oracle.com/en/java/javase/18/code-snippet/index.html[https://docs.oracle.com/en/java/javase/18/code-snippet/index.html]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

* Align the last closing curly bracket by adding N spaces.

=== Highlighting

* Primary: the closing curly brace of the code snippet

=== Quickfix

* Add N spaces in front of the ending curly brace.

endif::env-github,rspecator-view[]
