<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when AWS CloudWatch <code>put_metric_data</code> namespace begins with <code>AWS/</code></p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>AWS CloudWatch has reserved namespaces that begin with 'AWS/' for its own internal services and metrics. These namespaces are used by AWS to publish official service metrics such as EC2 instance metrics, Lambda function metrics, S3 bucket metrics, and others. When you attempt to publish custom metrics using a namespace that begins with 'AWS/', you are essentially trying to use a reserved namespace that conflicts with AWS&#8217;s own metric publishing system.</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="paragraph">
<p>Using AWS service namespaces can lead to metric publishing failures, data corruption, or unexpected behavior in CloudWatch dashboards. It may also cause confusion between your custom metrics and official AWS service metrics, making monitoring and troubleshooting more difficult. Additionally, AWS may reject such metric publications, or they may interfere with existing service metrics.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it_in_boto3">How to fix it in Boto3</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Use a custom namespace that does not begin with 'AWS/' when publishing metrics with CloudWatch. Choose a meaningful namespace that reflects your application or service name, such as 'MyApp/', 'CustomService/', or your organization name.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">import boto3

cloudwatch = boto3.client('cloudwatch')

# Publishing to AWS reserved namespace
cloudwatch.put_metric_data(
    Namespace='AWS/MyCustomService',  # Noncompliant
    MetricData=[
        {
            'MetricName': 'CustomMetric',
            'Value': 123.0
        }
    ]
)</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">import boto3

cloudwatch = boto3.client('cloudwatch')

# Publishing to custom namespace
cloudwatch.put_metric_data(
    Namespace='MyApp/CustomService',  # Compliant
    MetricData=[
        {
            'MetricName': 'CustomMetric',
            'Value': 123.0
        }
    ]
)</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it_in_aiobotocore">How to fix it in aiobotocore</h2>
<div class="sectionbody">
<div class="paragraph">
<p>When using aiobotocore for asynchronous CloudWatch operations, ensure you use a custom namespace that does not start with 'AWS/' to avoid conflicts with AWS reserved namespaces.</p>
</div>
<div class="sect2">
<h3 id="_code_examples_2">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example_2">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">import aiobotocore.session

async def publish_metrics():
    session = aiobotocore.session.get_session()
    async with session.create_client('cloudwatch') as client:
        await client.put_metric_data(
            Namespace='AWS/Lambda/Custom',  # Noncompliant
            MetricData=[
                {
                    'MetricName': 'ProcessingTime',
                    'Value': 45.2
                }
            ]
        )</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution_2">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">import aiobotocore.session

async def publish_metrics():
    session = aiobotocore.session.get_session()
    async with session.create_client('cloudwatch') as client:
        await client.put_metric_data(
            Namespace='MyLambda/Custom',  # Compliant
            MetricData=[
                {
                    'MetricName': 'ProcessingTime',
                    'Value': 45.2
                }
            ]
        )</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>AWS Documentation - <a href="https://docs.aws.amazon.com/AmazonCloudWatch/latest/APIReference/API_PutMetricData.html">AWS CloudWatch PutMetricData API Reference</a></p>
</li>
<li>
<p>AWS Documentation - <a href="https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/cloudwatch_concepts.html#Namespace">CloudWatch Concepts - Namespaces</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Do not use AWS reserved namespace that begins with 'AWS/' for custom metrics.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="ulist">
<ul>
<li>
<p>Primary location: the 'Namespace' parameter value in the put_metric_data call</p>
</li>
</ul>
</div>
</div>
</div>
</div>