== Why is this an issue?

Shared naming conventions allow teams to collaborate efficiently. This rule checks that all element names match a provided regular expression.


=== Noncompliant code example

With the default regular expression:

[source,xml]
----
<xs:element name="Fruit">     <!-- Noncompliant -->
  ...
</xs:element>
----


=== Compliant solution

[source,xml]
----
<xs:element name="fruit">
  ...
</xs:element>
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Rename this element/attribute to match the regular expression: ${format}


=== Parameters

.format
****

----
^[a-z]+([A-Z][a-z0-9]+)*$
----

Regular expression used to check element names against.
****


=== Highlighting

name value


endif::env-github,rspecator-view[]
