== Why is this an issue?

Shared naming conventions allow teams to collaborate efficiently. This rule checks that all element names match a provided regular expression.


=== Noncompliant code example

With the default regular expression:

[source,xml]
----
<xs:element name="Fruit">     <!-- Noncompliant -->
  ...
</xs:element>
----


=== Compliant solution

[source,xml]
----
<xs:element name="fruit">
  ...
</xs:element>
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

include::parameters.adoc[]

include::highlighting.adoc[]

endif::env-github,rspecator-view[]
