Shared naming conventions allow teams to collaborate efficiently. This rule checks that all element names match a provided regular expression.


== Noncompliant Code Example

With the default regular expression:

----
<xs:element name="Fruit">     <!-- Noncompliant -->
  ...
</xs:element>
----


== Compliant Solution

----
<xs:element name="fruit">
  ...
</xs:element>
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

include::parameters.adoc[]

include::highlighting.adoc[]

endif::env-github,rspecator-view[]
