<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Gratuitous boolean expressions are conditions that do not change the evaluation
of a program.
This issue can indicate logical errors and affect the correctness of an
application, as well as its maintainability.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Control flow constructs like <code>if</code>-statements allow the programmer to direct the
flow of a program depending on a boolean expression.
However, if the condition is always true or always false, only one of the
branches will ever be executed.
In that case, the control flow construct and the condition no longer serve a
purpose; they become <em>gratuitous</em>.</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="paragraph">
<p>The presence of gratuitous conditions can indicate a logical error.
For example, the programmer <em>intended</em> to have the program branch into different
paths but made a mistake when formulating the branching condition.
In this case, this issue might result in a bug and thus affect the reliability
of the application.
For instance, it might lead to the computation of incorrect results.</p>
</div>
<div class="paragraph">
<p>Additionally, gratuitous conditions and control flow constructs introduce
unnecessary complexity.
The source code becomes harder to understand, and thus, the application becomes
more difficult to maintain.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Gratuitous boolean expressions are suspicious and should be carefully removed from the code.</p>
</div>
<div class="paragraph">
<p>First, the boolean expression in question should be closely inspected for logical errors.
If a mistake was made, it can be corrected so the condition is no longer gratuitous.</p>
</div>
<div class="paragraph">
<p>If it becomes apparent that the condition is actually unnecessary, it can be removed. The associated control flow construct
(e.g., the <code>if</code>-statement containing the condition) will be adapted or even removed, leaving only the necessary branches.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-rust" data-lang="rust">if a &amp;&amp; b || a { // Noncompliant: `b` is redundant
  // ...
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-rust" data-lang="rust">if a {
  // ...
}</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Clippy Lints - <a href="https://rust-lang.github.io/rust-clippy/master/index.html#overly_complex_bool_expr" class="bare">https://rust-lang.github.io/rust-clippy/master/index.html#overly_complex_bool_expr</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>