<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Case mismatches in pattern arms make some arms unreachable, likely leading to logic errors that can be difficult to debug.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-rust" data-lang="rust">match &amp;*text.to_ascii_lowercase() {
    "foo" =&gt; {},
    "Bar" =&gt; {}, // Noncompliant: This arm is unreachable.
    _ =&gt; {},
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-rust" data-lang="rust">match &amp;*text.to_ascii_lowercase() {
    "foo" =&gt; {},
    "bar" =&gt; {},
    _ =&gt; {},
}</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Clippy Lints - <a href="https://rust-lang.github.io/rust-clippy/master/index.html#match_str_case_mismatch" class="bare">https://rust-lang.github.io/rust-clippy/master/index.html#match_str_case_mismatch</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>