== How to fix it in aiohttp

=== Code examples

include::../../common/fix/code-rationale.adoc[]

==== Noncompliant code example

[source,python,diff-id=21,diff-type=noncompliant]
----
from fastapi import FastAPI, Query
from pydantic.networks import Url
from typing import Annotated
import aiohttp

app = FastAPI()
@app.get('/example')
async def example(url: Annotated[Url, Query()]):
    async with aiohttp.request('GET', url.unicode_string()) as response: # Noncompliant
        return {"response": await response.text()}
----

==== Compliant solution

[source,python,diff-id=21,diff-type=compliant]
----
from fastapi import FastAPI, Query
from pydantic.networks import Url
from typing import Annotated
import aiohttp

SCHEMES_ALLOWLIST = ['https']
DOMAIN_ALLOWLIST = ['trusted1.example.com', 'trusted2.example.com'];

app = FastAPI()

@app.get('/example')
async def example(url: Annotated[Url, Query()]):
    if url.host in DOMAIN_ALLOWLIST and url.scheme in SCHEMES_ALLOWLIST:
        async with aiohttp.request('GET', url.unicode_string()) as response:
            return {"response": await response.text()}
    return {"error": f"URL {url} is not in whitelist"}
----

=== How does this work?

include::../../common/fix/pre-approved-list.adoc[]

The compliant code example uses such an approach.

=== Pitfalls

include::../../common/pitfalls/starts-with.adoc[]
