== How to fix it in HTTPX

=== Code examples

include::../../common/fix/code-rationale.adoc[]

==== Noncompliant code example

[source,python,diff-id=21,diff-type=noncompliant]
----
from fastapi import FastAPI
import httpx

app = FastAPI()

@app.get('/example')
def example(url: str):
    response = httpx.get(url)  # Noncompliant
    return {"response": response.text}
----

==== Compliant solution

[source,python,diff-id=21,diff-type=compliant]
----
from fastapi import FastAPI, Response, status
import httpx
from urllib.parse import urlparse

DOMAINS_ALLOWLIST = ['trusted1.example.com', 'trusted2.example.com']
app = FastAPI()

@app.get('/example')
def example(url: str, response: Response):
    if not urlparse(url).hostname in DOMAINS_ALLOWLIST:
        response.status_code = status.HTTP_400_BAD_REQUEST
        return {"error": f"URL {url} is not in whitelist"}

    r = httpx.get(url)
    return {"response": r.text}
----

=== How does this work?

include::../../common/fix/pre-approved-list.adoc[]

The compliant code example uses such an approach.
HTTPX implicitly validates the scheme as it only allows `http` and `https` by default.

=== Pitfalls

include::../../common/pitfalls/starts-with.adoc[]
