<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>The use of <code>UNDERLINE</code> makes normal text printed to the screen difficult to read, and prevents titles from being read at all. For these reasons, <code>UNDERLINE</code> should not be used.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-rpg" data-lang="rpg">     R DETAIL1
      * Noncompliant; next line contains UNDERLINE
       NAME          25      2  2UNDERLINE
       ADD1          25      3  2
     R DETAIL3
       ADD2          25         2SPACEB(1)</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-rpg" data-lang="rpg">     R DETAIL1
       NAME          25      2  2
       ADD1          25      3  2
     R DETAIL3
       ADD2          25         2SPACEB(1)</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove this use of "UNDERLINE".</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_22_sep_2014_065024_ann_campbell_wrote">on 22 Sep 2014, 06:50:24 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>Note that this rule is actually for DDS, the record-format specification for RPG, rather than for RPG itself.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_29_sep_2014_131315_ann_campbell_wrote">on 29 Sep 2014, 13:13:15 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>Note that this rule is for DDS</p>
</div>
</div>
</div>
</div>