<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when code directly accesses elements from string split results by index without first verifying that the element exists.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>When you split a string using methods like <code>tokenize()</code> or <code>split()</code>, the resulting list size depends on the input content. If the input doesn&#8217;t contain the expected number of delimiters, accessing elements by index can throw an <code>IndexOutOfBoundsException</code> at runtime.</p>
</div>
<div class="paragraph">
<p>For example, if you expect a string like <code>"a-b-c"</code> but receive <code>"a-b"</code>, trying to access the third element will cause your application to crash. This is particularly problematic when processing user input, configuration files, or data from external sources where the format might vary.</p>
</div>
<div class="paragraph">
<p>Groovy provides safer alternatives like multiple assignment that automatically handle cases where fewer elements are available than expected.</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="paragraph">
<p>This issue can cause application crashes and unexpected runtime failures. When processing user input or external data, it can lead to poor user experience and system instability. In server applications, it might cause request failures or service interruptions.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Use multiple assignment which safely handles missing elements by assigning null to variables when there aren&#8217;t enough elements in the list.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-groovy" data-lang="groovy">def parts = inputString.tokenize("-")
def first = parts.get(0)
def second = parts.get(1)
def third = parts.get(2)  // Noncompliant: may throw IndexOutOfBoundsException</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-groovy" data-lang="groovy">def parts = inputString.tokenize("-")
def (first, second, third) = parts  // Safe: missing elements become null</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Groovy Multiple Assignment - <a href="https://groovy-lang.org/semantics.html#_multiple_assignment">Official documentation on Groovy&#8217;s multiple assignment feature</a></p>
</li>
<li>
<p>Groovy String Processing - <a href="https://groovy-lang.org/processing-xml.html#_working_with_strings">Guide to string processing methods in Groovy</a></p>
</li>
</ul>
</div>
</div>
<div class="sect2">
<h3 id="_standards">Standards</h3>
<div class="ulist">
<ul>
<li>
<p>CWE 129 - <a href="https://cwe.mitre.org/data/definitions/129.html">Improper Validation of Array Index</a></p>
</li>
</ul>
</div>
</div>
<div class="sect2">
<h3 id="_related_rules">Related rules</h3>
<div class="ulist">
<ul>
<li>
<p><a data-rspec-id="S6466" class="rspec-auto-link">S6466</a> - Collection elements should not be replaced unconditionally (Java)</p>
</li>
</ul>
</div>
</div>
</div>
</div>