== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,csharp]
----
[Export(typeof(ISomeType))]
public class SomeType // Noncompliant; doesn't implement 'ISomeType'.
{
}
----


=== Compliant solution

[source,csharp]
----
[Export(typeof(ISomeType))]
public class SomeType : ISomeType
{
}
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../highlighting.adoc[]

endif::env-github,rspecator-view[]
