This rule reports when a class is placed in a location which deviates from the intended architecture.

== Why is this an issue?

This project has an intended architecture defined, which specifies high-level components and their allowed relationships.
This might include that specific classes should be located in specific packages or modules, or that packages are renamed.
This issue is being raised because this class’s location deviates from the intended architecture.

=== What is the potential impact?

When the code location deviates from the intended architecture, it gradually erodes the project’s structure.
As these deviations accumulate, the code becomes more complex and less consistent with the intent, making future development increasingly challenging.

== How to fix it

Move the relevant class to its intended location or rename the relevant package as intended.

=== Code examples

==== Noncompliant code example

If the intended architecture declares developers should use the package name `com.example.panel`, not `com.example.panels`, then the following code example is noncompliant:

[source,java,diff-id=1,diff-type=noncompliant]
----
package com.example.panels;

class ShowCustomersPanel extends Panel {
  // ...
}
----

==== Compliant solution

The solution is to move the class to its appropriate package, or respectively, to rename the package `panels` to `panel`.

[source,java,diff-id=1,diff-type=compliant]
----
package com.example.panel;

import com.example.services.CustomerService;

class ShowCustomersPanel extends Panel {
  // ...
}
----
