This rule reports when a class is placed in the wrong package regarding the intended architectural design.

== Why is this an issue?

The software architect may decide that specific classes should be located in specific packages, or that packages are misnamed.
These decisions reflect the intended structure of the software and responsibilities of code components (classes, packages, modules).
The reported misplaced class violates these design decisions.

=== What is the potential impact?

When new code violates design decisions, the project's structure gradually drift away from its intended architecture.
As these violations accumulate, the code becomes more complex and less consistent with the architectâ€™s original design, making future development and maintenance increasingly challenging.

== How to fix it

Move the proclaimed class to its intended package, or rename the proclaimed package as intended.

=== Code examples

==== Noncompliant code example

If the intended design declares developers should use the package name `com.example.panel`, not `com.example.panels`, then the following code example is noncompliant:

[source,java,diff-id=1,diff-type=noncompliant]
----
package com.example.panels;

class ShowCustomersPanel extends Panel {
  // ...
}
----

==== Compliant solution

The solution is to move the class to its appropriate package, or respectively, to rename the package `panels` to `panel`.

[source,java,diff-id=1,diff-type=compliant]
----
package com.example.panel;

import com.example.services.CustomerService;

class ShowCustomersPanel extends Panel {
  // ...
}
----
