This rule raises an issue when inline SOQL queries are assigned to generic `Object` types instead of properly typed SObject collections like `List<SObject>` or `SObject[]`.

== Why is this an issue?

When you execute SOQL queries in Apex, assigning the results to generic `Object` types defeats the purpose of Apex's strong typing system.

SOQL queries return specific SObject types based on the queried object. When you assign these results to an `Object` variable, you lose several important benefits:

* *Type safety*: The compiler cannot verify that you're accessing valid fields or methods, leading to potential runtime errors.
* *IDE support*: Code completion, syntax highlighting, and refactoring tools cannot provide accurate suggestions.
* *Code readability*: Other developers cannot easily understand what type of data the variable contains.
* *Performance*: Additional type casting may be required when working with the data.

Apex provides strong typing specifically to help you write more reliable and maintainable code. By using properly typed collections like `Account[]` or `List<Account>`, you enable the compiler to catch errors early and make your code more self-documenting.

=== What is the potential impact?

Using generic `Object` types for SOQL results can lead to runtime errors when accessing SObject fields or methods. It also reduces code maintainability and makes debugging more difficult, as type-related issues may only surface during execution rather than at compile time.

== How to fix it

Replace the generic `Object` type with the specific SObject type that matches your SOQL query. Use either array notation (`SObject[]`) or List notation (`List<SObject>`).

=== Code examples

==== Noncompliant code example

[source,apex,diff-id=1,diff-type=noncompliant]
----
Object accts = [SELECT Name, Phone FROM Account]; // Noncompliant
----

==== Compliant solution

[source,apex,diff-id=1,diff-type=compliant]
----
Account[] accts = [SELECT Name, Phone FROM Account];
// or alternatively:
List<Account> accts = [SELECT Name, Phone FROM Account];
----

== Resources

=== Documentation

 * Salesforce SOQL and SOSL Reference - https://developer.salesforce.com/docs/atlas.en-us.soql_sosl.meta/soql_sosl/[Official Salesforce documentation for SOQL queries and best practices]

 * Apex Developer Guide - Working with SOQL - https://developer.salesforce.com/docs/atlas.en-us.apexcode.meta/apexcode/langCon_apex_SOQL.htm[Comprehensive guide on using SOQL within Apex code]
