<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Shared coding conventions allow teams to collaborate effectively. To improve readability, blank lines in a program should not have end-of-line comments.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="paragraph">
<p>In this example, the second line is blank except for the end-of-line comment, obscuring readability. The seventh line is commented-out, so an end-of-line comment is acceptable.</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-rpg" data-lang="rpg">     I/COPY VALSRC,VALTEST                                                FIX001
                                                                          FIX001
      ** Data structure for subroutine VALDS                              FIX001
     I            DS                                                      FIX001
     I                                        5   60MMA                   FIX001
     I                                        7   80DDA                   FIX001
      *                                                                   FIX001
     I            DS                                                      FIX001
     I                                        5   60MMB                   FIX001
     I                                        7   80DDB                   FIX001</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-rpg" data-lang="rpg">     I/COPY VALSRC,VALTEST                                                FIX001

      ** Data structure for subroutine VALDS                              FIX001
     I            DS                                                      FIX001
     I                                        5   60MMA                   FIX001
     I                                        7   80DDA                   FIX001
      *                                                                   FIX001
     I            DS                                                      FIX001
     I                                        5   60MMB                   FIX001
     I                                        7   80DDB                   FIX001</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove the comment from the end of this line</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_12_sep_2014_140142_ann_campbell_wrote">on 12 Sep 2014, 14:01:42 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~pierre-yves.nicolas] we spoke briefly about this rule when the request was first made and you said it seemed feasible. Before I send it out to the requester for vetting, I wanted to double-check that you still feel that way.</p>
</div>
</div>
</div>
</div>