Control characters are special, invisible characters in the ASCII range 0-31. These characters are rarely used in JavaScript strings so a regular expression containing these characters is most likely a mistake.

== Noncompliant Code Example

----
var pattern1 = /\x1f/;
var pattern2 = new RegExp("\x1f");
----

== Compliant Solution

----
var pattern1 = /\x20/;
var pattern2 = new RegExp("\x20");
----
