``DISTINCT`` operator causes the ``SELECT`` statement to avoid the SAP buffering and to read directly from the database and not from the buffer on the application server.


== Noncompliant Code Example

----
SELECT DISTINCT carrid
       FROM spfli 
       INTO count 
       WHERE cityto = 'NEW YORK'. 
----

