<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Hashing a unit value doesn&#8217;t accomplish anything because the implementation of <code>Hash</code> for <code>()</code> is a no-op. This can lead to confusion or misleading code, as it implies some hashing operation is actually taking place, when in reality it does nothing.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-rust" data-lang="rust">enum Foo { Empty, WithValue(u8) }
use Foo::*;

let mut state = DefaultHasher::new();
let my_enum = Foo::Empty;

match my_enum {
    Empty =&gt; ().hash(&amp;mut state), // Noncompliant
    WithValue(x) =&gt; x.hash(&amp;mut state),
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-rust" data-lang="rust">enum Foo { Empty, WithValue(u8) }
use Foo::*;

let mut state = DefaultHasher::new();
let my_enum = Foo::Empty;

match my_enum {
    Empty =&gt; 0_u8.hash(&amp;mut state), // Compliant
    WithValue(x) =&gt; x.hash(&amp;mut state),
}</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Clippy Lints - <a href="https://rust-lang.github.io/rust-clippy/master/index.html#unit_hash" class="bare">https://rust-lang.github.io/rust-clippy/master/index.html#unit_hash</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>