There can't be any good reason to do a full table scan on large database tables due to the cost of such operation and the scalability issue that might raise. This rule raises an issue when a ``++SELECT++`` statement doesn't use at least one indexed column in its ``++WHERE++`` clause.


*Note* That this rule raises issues only when a database catalog is provided during the SonarQube analysis.


== Noncompliant Code Example

----
SELECT * FROM USERS WHERE NAME = :name  -- non compliant when NAME column is not indexed
----


== Exceptions

This rule ignores ``++SELECT++`` statements not having any ``++WHERE++`` clauses; they are covered by other rules: S1590, SQL.SelectWithNoWhereClauseCheck


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

include::parameters.adoc[]

include::highlighting.adoc[]

'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
