<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when complex nested anonymous structs are used in field definitions, making code harder to read, reuse, and maintain.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Anonymous structs can make code harder to understand and maintain, especially when they contain multiple fields or are nested within other structs.</p>
</div>
<div class="paragraph">
<p>When you use anonymous structs for complex data structures, you create several problems:</p>
</div>
<div class="ulist">
<ul>
<li>
<p><strong>Poor readability</strong>: Nested anonymous structs create deeply indented code that is difficult to scan and understand quickly.</p>
</li>
<li>
<p><strong>No reusability</strong>: Anonymous structs cannot be reused across different parts of your codebase, leading to code duplication.</p>
</li>
<li>
<p><strong>Testing difficulties</strong>: You cannot easily create instances of anonymous struct types in test files, often forcing you to duplicate the struct definition.</p>
</li>
<li>
<p><strong>Limited documentation</strong>: Anonymous structs cannot have their own methods or be documented separately, reducing code clarity.</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>Named types solve these issues by providing clear, reusable definitions that can be documented, tested, and maintained independently. They also make your code more modular and easier to refactor.</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="paragraph">
<p>Using anonymous structs for complex nested structures reduces code maintainability and readability. It can lead to code duplication across files, make testing more difficult, and create confusion for developers working with the codebase.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Extract the anonymous struct into a named type. Define the named type separately and use it in your struct field definition.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-go" data-lang="go">type Service struct {
	Readiness *struct {
		TimeoutSeconds   int
		PeriodSeconds    int
		HttpGet          *struct { // Noncompliant
			Path string
			Port int
		}
	}
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-go" data-lang="go">type HTTPGetProbe struct {
	Path string
	Port int
}

type ReadinessProbe struct {
	TimeoutSeconds int
	PeriodSeconds  int
	HttpGet        *HTTPGetProbe
}

type Service struct {
	Readiness *ReadinessProbe
}</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Go Language Specification - Struct types - <a href="https://golang.org/ref/spec#Struct_types">Official Go specification for struct types and anonymous structs</a></p>
</li>
<li>
<p>Effective Go - Names - <a href="https://golang.org/doc/effective_go#names">Guidelines for naming conventions in Go, including type names</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>