<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p><code>UNION</code> is significantly less performant compared to <code>UNION ALL</code> because it removes duplicated entries and run an internal <code>DISTINCT</code> to achieve this.</p>
</div>
<div class="paragraph">
<p><code>UNION ALL</code> is not removing duplicates and returns all the results from the queries. It performs faster in most of the cases compared to <code>UNION</code>. Nevertheless, the quantity of data returned by <code>UNION ALL</code> can be significantly more important than with <code>UNION</code>. On slow network, the performance gain to use <code>UNION ALL</code> instead of <code>UNION</code> can be challenged by the time lost to transfer more data than with <code>UNION</code>.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-sql" data-lang="sql">-- if you care about not having duplicated entries, then UNION is the good choice
SELECT EMAIL FROM EMPLOYEES UNION SELECT EMAIL FROM CUSTOMERS</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-sql" data-lang="sql">-- if you don't care about duplicated entries in the results of this UNION, then UNION ALL should be preferred
SELECT EMAIL FROM EMPLOYEES UNION ALL SELECT EMAIL FROM CUSTOMERS</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_23_oct_2018_103607_alexandre_gigleux_wrote">on 23 Oct 2018, 10:36:07 Alexandre Gigleux wrote:</h3>
<div class="paragraph">
<p>Merged with <a data-rspec-id="S1571" class="rspec-auto-link">RSPEC-1571</a> which was better described to cover the case of <code>UNION ALL</code>.</p>
</div>
</div>
</div>
</div>