== Why is this an issue?

Level 77 identifies data items that are not subdivisions of other items, and that have no subdivisions. They are atomic by declaration. To make future subdivision possible, level 01 should be used instead of level 77.


=== Noncompliant code example

[source,cobol]
----
77 CAR            PIC 999.
----


=== Compliant solution

[source,cobol]
----
01 CAR            PIC 999.
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

To allow potential future changes to data items, use level 01 instead of level 77


endif::env-github,rspecator-view[]
