<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when <code>def</code> is redundant.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>In Groovy, <code>def</code> is used when you intentionally omit an explicit type. When a declaration already includes a type (for example, <code>String</code>) or a visibility modifier (for example, <code>private</code>), keeping <code>def</code> is redundant and makes the code less idiomatic.</p>
</div>
<div class="paragraph">
<p>Unnecessary <code>def</code> appears:
- on fields with a visibility modifier or a type declaration
- on methods with a visibility modifier or an explicit return type
- on variables with an explicit type declaration or a type declaration</p>
</div>
<div class="paragraph">
<p>This extra keyword adds noise and can confuse readers about whether the code relies on dynamic typing.</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="paragraph">
<p>This is mainly a maintainability issue: redundant keywords reduce readability and make style less consistent across the codebase.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Remove <code>def</code> when the declaration already has an explicit type or a modifier.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-groovy" data-lang="groovy">private def String name = "Guillaume" // Noncompliant</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-groovy" data-lang="groovy">private String name = "Guillaume"</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Groovy documentation - <a href="https://groovy-lang.org/style-guide.html#_def_and_type">Style Guide</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>