<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>For code compliant with C&#43;&#43;98 or C&#43;&#43;03 standards, declaring overriding virtual functions with the <code>virtual</code> keyword removes the need to check the base class to determine whether a function is virtual.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">class Base
{
  virtual void f();
};
class Derived : public Base
{
  void f(); // Noncompliant, implicitly declared "virtual"
};</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">class Base
{
  virtual void f();
};
class Derived : public Base
{
  virtual void f(); // Compliant, explicitly declared "virtual"
};</code></pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p>MISRA C&#43;&#43;:2008, 10-3-2 - Each overriding virtual function shall be declared with the virtual keyword.</p>
</li>
</ul>
</div>
<div class="sect2">
<h3 id="_related_rules">Related rules</h3>
<div class="ulist">
<ul>
<li>
<p>This rule will only trigger with code compliant with C&#43;&#43;98 and C&#43;&#43;03. For code compliant with C&#43;&#43;11 and above see <a data-rspec-id="S3471" class="rspec-auto-link">S3471</a></p>
</li>
</ul>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>C&#43;&#43;98 or C&#43;&#43;03: Annotate this function with "virtual".</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_is_duplicated_by_s1261">is duplicated by: <a data-rspec-id="S1261" class="rspec-auto-link">S1261</a></h3>

</div>
<div class="sect2">
<h3 id="_is_related_to_s3471">is related to: <a data-rspec-id="S3471" class="rspec-auto-link">S3471</a></h3>

</div>
</div>
</div>