include::../description.adoc[]

== Noncompliant code example

With default provided regular expression: ^[a-z][a-zA-Z0-9]*$

[source,flex]
----
function DoSomething(){...} 
----

== Compliant solution

[source,flex]
----
function doSomething(){...} 
----
