include::../description.adoc[]

== Noncompliant Code Example

With default provided regular expression: ^[a-z][a-zA-Z0-9]*$

[source,flex]
----
function DoSomething(){...} 
----

== Compliant Solution

[source,flex]
----
function doSomething(){...} 
----
