include::../description.adoc[]

== Noncompliant Code Example

[source,abap]
----
IF something.   
  IF somethingElse.
    WRITE / 'hello'.
  ENDIF.
ENDIF.
----

== Compliant Solution

----
IF something and somethingElse.  
  WRITE / 'hello'.
ENDIF.
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../highlighting.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
