include::../description.adoc[]

== Noncompliant Code Example

----
IF something.   
  IF somethingElse.
    WRITE / 'hello'.
  ENDIF.
ENDIF.
----

== Compliant Solution

----
IF something and somethingElse.  
  WRITE / 'hello'.
ENDIF.
----

ifdef::env-github,rspecator-view[]
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
