== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,abap]
----
IF something.   
  IF somethingElse.
    WRITE / 'hello'.
  ENDIF.
ENDIF.
----

=== Compliant solution

[source,abap]
----
IF something and somethingElse.  
  WRITE / 'hello'.
ENDIF.
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../highlighting.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
