== Why is this an issue?

include::description.adoc[]

=== Noncompliant code example

[source,text]
----
interface MyInterface {
  doTheThing(): number;
  doTheOtherThing(): string;
  doTheThing(str: string): string;  // Noncompliant
}
----


=== Compliant solution

[source,text]
----
interface MyInterface {
  doTheThing(): number;
  doTheThing(str: string): string;
  doTheOtherThing(): string;
}
----

