include::../description.adoc[]

== Noncompliant Code Example

[source,cpp]
----
#define foo // Noncompliant
----

== Compliant Solution

[source,cpp]
----
#define FOO
----
ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::parameters.adoc[]

endif::env-github,rspecator-view[]
