== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,cpp]
----
#define foo // Noncompliant
----

=== Compliant solution

[source,cpp]
----
#define FOO
----
ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

=== Parameters

.format
****
_STRING_

----
^[A-Z][A-Z0-9_]*$
----

Regular expression used to check the macro names against.
****


endif::env-github,rspecator-view[]
