== How to fix it in Android

=== Code examples

include::../../common/fix/code-rationale.adoc[]

==== Noncompliant code example

[source,xml,diff-id=1,diff-type=noncompliant]
----
<application
  android:icon="@mipmap/ic_launcher"
  android:label="@string/app_name"
  android:roundIcon="@mipmap/ic_launcher_round"
  android:supportsRtl="true"
  android:debuggable="true"
  android:theme="@style/AppTheme">
</application>  <!-- Noncompliant --> 
----

[source,xml,diff-id=2,diff-type=noncompliant]
----
<configuration>
  <system.web>
    <customErrors mode="Off" />  <!-- Noncompliant -->
  </system.web>
</configuration>
----

==== Compliant solution

[source,xml,diff-id=1,diff-type=compliant]
----
<application
  android:icon="@mipmap/ic_launcher"
  android:label="@string/app_name"
  android:roundIcon="@mipmap/ic_launcher_round"
  android:supportsRtl="true"
  android:debuggable="false"
  android:theme="@style/AppTheme">
</application>
----

[source,xml,diff-id=2,diff-type=compliant]
----
<configuration>
  <system.web>
    <customErrors mode="On" />
  </system.web>
</configuration>
----

=== How does this work?

include::../../common/fix/how.adoc[]
