While most ``++compareTo++`` methods return -1, 0, or 1, some do not, and testing the result of a ``++compareTo++`` against a specific value other than 0 could result in false negatives.


== Noncompliant Code Example

----
if (myClass.compareTo(arg) == -1) {  // Noncompliant
  // ...
}
----


== Compliant Solution

----
if (myClass.compareTo(arg) < 0) {
  // ...
}
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

endif::env-github,rspecator-view[]
