<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when a method name starts with "is_" and ends with "?".</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>In Ruby, predicate methods (methods that return boolean values) should end with a question mark to indicate their boolean nature. Using both the "is_" prefix and the "?" suffix creates redundancy that violates Ruby naming conventions.</p>
</div>
<div class="paragraph">
<p>The question mark suffix already clearly communicates that the method returns a boolean value. Adding "is_" makes the method name unnecessarily verbose and goes against established Ruby idioms. This redundancy can make code less readable and harder to maintain.</p>
</div>
<div class="paragraph">
<p>Ruby&#8217;s naming conventions emphasize clarity and conciseness. Method names like <code>empty?</code>, <code>valid?</code>, and <code>present?</code> are more idiomatic than <code>is_empty?</code>, <code>is_valid?</code>, and <code>is_present?</code>.</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="paragraph">
<p>Using redundant "is_" prefixes in predicate methods reduces code readability and violates Ruby naming conventions. While this doesn&#8217;t cause runtime errors, it makes the codebase less idiomatic and can confuse developers who expect standard Ruby naming patterns.</p>
</div>
<div class="paragraph">
<p>This pattern can also make method names unnecessarily long, reducing code clarity and making it harder to follow established Ruby style guides.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it_in_ruby">How to fix it in Ruby</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Remove the "is_" prefix from predicate methods. The question mark suffix already indicates that the method returns a boolean value.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-ruby" data-lang="ruby">def is_empty?
  @items.empty?
end</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-ruby" data-lang="ruby">def empty?
  @items.empty?
end</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it_in_rails">How to fix it in Rails</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This also applies to field declarations in frameworks like Avo where boolean fields should use the question mark suffix without the "is_" prefix.</p>
</div>
<div class="sect2">
<h3 id="_code_examples_2">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example_2">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-ruby" data-lang="ruby">field :is_admin, as: :boolean</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution_2">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-ruby" data-lang="ruby">field :admin?, as: :boolean</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Ruby Style Guide - Naming - <a href="https://ruby-style-guide.shopify.dev/#naming">Shopify&#8217;s Ruby Style Guide section on naming conventions, including predicate methods</a></p>
</li>
<li>
<p>Ruby Method Naming Conventions - <a href="https://docs.ruby-lang.org/en/master/syntax/methods_rdoc.html#label-Method+Names">Official Ruby documentation on method naming conventions</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>