include::../common/introduction.adoc[]

== Why is this an issue?

include::../common/why-is-this-an-issue.adoc[]

include::../common/why-is-this-an-issue-thread-safety.adoc[]

include::../common/rule-behaviour.adoc[]

=== What is the potential impact?

include::../common/what-is-the-potential-impact.adoc[]

== How to fix it

include::../common/how-to-fix-it.adoc[]

=== Code examples

==== Noncompliant code example

The following example assumes that constant names should match the default
regular expression ``++^[a-z][a-zA-Z0-9]*$++``:

[source,swift,diff-id=1,diff-type=noncompliant]
----
let Pi = 3.14
----

==== Compliant solution

[source,swift,diff-id=1,diff-type=compliant]
----
let pi = 3.14
----

== Resources

=== External coding guidelines

* https://www.swift.org/documentation/api-design-guidelines/#conventions[Case Conventions] in the Swift API Design Guidelines

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../common/message.adoc[]

=== Parameters

.format
****

----
^[a-z][a-zA-Z0-9]*$
----

Regular expression used to check the constant names against.
****


'''
== Comments And Links
(visible only on this page)

=== on 15 May 2015, 09:40:57 Elena Vilchik wrote:
\[~ann.campbell.2] I want to change default parameter value on camelCase (based on examples \https://developer.apple.com/library/mac/documentation/Swift/Conceptual/Swift_Programming_Language/TheBasics.html#//apple_ref/doc/uid/TP40014097-CH5-ID310 and recommendations \http://ericasadun.com/2015/05/05/swift-dont-do-that/ "Donâ€™t use ALL_CAPS").  Any objections?

=== on 15 May 2015, 11:56:52 Ann Campbell wrote:
None whatsoever [~elena.vilchik]. You're the language expert! :-)

include::../common/comments-and-links.adoc[]

endif::env-github,rspecator-view[]
