== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

With the default regular expression ``++^[a-z][a-zA-Z0-9]*$++``:

[source,swift]
----
let Pi = 3.14
----

=== Compliant solution

[source,swift]
----
let pi = 3.14
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

=== Parameters

.format
****

----
^[a-z][a-zA-Z0-9]*$
----

Regular expression used to check the constant names against.
****


'''
== Comments And Links
(visible only on this page)

=== on 15 May 2015, 09:40:57 Elena Vilchik wrote:
\[~ann.campbell.2] I want to change default parameter value on camelCase (based on examples \https://developer.apple.com/library/mac/documentation/Swift/Conceptual/Swift_Programming_Language/TheBasics.html#//apple_ref/doc/uid/TP40014097-CH5-ID310 and recommendations \http://ericasadun.com/2015/05/05/swift-dont-do-that/ "Donâ€™t use ALL_CAPS").  Any objections?

=== on 15 May 2015, 11:56:52 Ann Campbell wrote:
None whatsoever [~elena.vilchik]. You're the language expert! :-)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
