== How to fix it in curl

=== Code examples

include::../../common/fix/code-rationale.adoc[]

==== Noncompliant code example

[source,cpp,diff-id=1,diff-type=noncompliant]
----
#include <curl/curl.h>

CURL *curl;
curl_global_init(CURL_GLOBAL_DEFAULT);

curl = curl_easy_init();
curl_easy_setopt(curl, CURLOPT_URL, "https://example.com/");
curl_easy_setopt(curl, CURLOPT_SSL_VERIFYPEER, 0L); // Noncompliant

curl_easy_perform(curl);
----

==== Compliant solution

[source,cpp,diff-id=1,diff-type=compliant]
----
#include <curl/curl.h>

CURL *curl;
curl_global_init(CURL_GLOBAL_DEFAULT);

curl = curl_easy_init();
curl_easy_setopt(curl, CURLOPT_URL, "https://example.com/");
curl_easy_setopt(curl, CURLOPT_SSL_VERIFYPEER, 1L);

curl_easy_perform(curl);
----

=== How does this work?

include::../../common/fix/default-implementation.adoc[]
