<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>The standard, predefined macros, such as <code>__FILE__</code> and <code>__LINE__</code>, are primarily intended for use by the implementation, and changing them could result in undefined behavior.</p>
</div>
<div class="paragraph">
<p>This rule checks that the following predefined macros are not defined, undefined, or redefined: <code>assert</code>, <code>errno</code>, <code>__FILE__</code>, <code>__LINE__</code>, <code>__TIME__</code>, <code>__DATE__</code>, <code>__TIMESTAMP__</code>, <code>__COUNTER__</code>, <code>__INCLUDE_LEVEL__</code>, <code>__BASE_FILE__</code>, and <code>_Pragma</code>.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">#undef __LINE__</code></pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p>MISRA C:2004, 20.1 - Reserved identifiers, macros and functions in the standard library shall not be defined, redefined, or undefined</p>
</li>
<li>
<p>MISRA C&#43;&#43;:2008, 17-0-1 - Reserved identifiers, macros and functions in the standard library shall not be defined, redefined, or undefined</p>
</li>
<li>
<p>MISRA C:2012, 21.1 - #define and #undef shall not be used on a reserved identifier or reserved macro name</p>
</li>
</ul>
</div>
<hr>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove this XXX directive</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_is_related_to_s979">is related to: <a data-rspec-id="S979" class="rspec-auto-link">S979</a></h3>

</div>
</div>
</div>