== Why is this an issue?

The use of virtual base classes can introduce a number of undefined and potentially confusing behaviours. The use of virtual bases is not recommended.


=== Noncompliant code example

[source,cpp]
----
class B {};
class D: public virtual B {}; // Noncompliant, B is a virtual base
----


== Resources

* MISRA {cpp}:2008, 10-1-1 - Classes should not be derived from virtual bases.


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
