The use of virtual base classes can introduce a number of undefined and potentially confusing behaviours. The use of virtual bases is not recommended.

What is xref:infeasible-path[]?

And what is a <<dataflow-anomaly>>?

== Noncompliant Code Example

[source,cpp]
----
class B {};
class D: public virtual B {}; // Noncompliant, B is a virtual base
----


== See

* MISRA {cpp}:2008, 10-1-1 - Classes should not be derived from virtual bases.

'''

== Glossary

[[infeasible-path]]_infeasible path_::
_Infeasible paths_ occur where there is a syntactic path to a code fragment, but the semantics ensure that the control flow path will not be executed. Example:

[source,cpp]
----
if ( u32 < 0 )
{
   // An unsigned value will never be negative,
   // so code in this block will never be executed.
}
----

[[dataflow-anomaly]]_dataflow anomaly_::
The state of a variable at a point in a program can be described using the following terms:

  * Undefined (U): The value of the variable is indeterminate;
  * Referenced (R): The variable is used in some way (e.g. in an expression);
  * Defined (D): The variable is explicitly initialised or assigned a value.

Given the above, the following _dataflow anomalies_ can be defined:

  * UR dataflow anomaly: Variable not assigned a value before the specified use;
  * DU dataflow anomaly: Variable is assigned a value that is never subsequently used;
  * DD dataflow anomaly: Variable is assigned a value twice with no intermediate use.

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
