<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>While working with bitwise operators <code>&amp;</code> or <code>|</code>, it is easy to make a typo andÂ write the equivalent logical operators <code>&amp;&amp;</code> or <code>||</code>. This rule raises an issue when the right operand of a logical expression  <code>&amp;&amp;</code> or <code>||</code> is a constant of integral type, as the developer probably meant to use the corresponding bitwise operator <code>&amp;</code> or <code>|</code>.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">int fun(int a) {
  return a || 4; // Noncompliant: did you mean to use bitwise operator '|'?
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">int fun(int a) {
 return a | 4;
}</code></pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p>C&#43;&#43; reference - <a href="https://en.cppreference.com/w/cpp/language/operator_arithmetic#Bitwise_logic_operators">Bitwise logic operators</a></p>
</li>
<li>
<p>C&#43;&#43; reference - <a href="https://en.cppreference.com/w/cpp/language/operator_logical">Logical operators</a></p>
</li>
</ul>
</div>
<hr>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Review this logical (&amp;&amp; | ||) expression with constant operand.</p>
</div>
</div>
</div>
</div>