include::../summary.adoc[]

== Why is this an issue?

Jump statements (``++return++``, ``++break++`` and ``++next++``) move control flow out of the current code block. So any statements that come after a jump are dead code. 

=== Noncompliant code example

[source,ruby]
----
def foo(a)
  i = 10
  return a + i    # Noncompliant 
  i += 1          # dead code
end
----

=== Compliant solution

[source,ruby]
----
def foo(a)
  i = 10
  return a + i
end
----

include::../see.adoc[]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../highlighting.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
