== How to fix it in FasterXML

=== Code examples

include::../../common/fix/code-rationale.adoc[]

==== Noncompliant code example

[source,java,diff-id=1,diff-type=noncompliant]
----
import com.fasterxml.jackson.databind.ObjectMapper;

//[...]
public void doGet(HttpServletRequest req, HttpServletResponse resp) throws IOException {
    String tainted = request.getParameter("value");
    String json = "{\"key\":\""+ tainted +"\"}";
    ObjectMapper obj = new ObjectMapper();
    try {
        obj.readValue(json, Object.class); // Noncompliant
    } catch (JsonProcessingException e) {
        response.sendError(400);
    }
}
----

==== Compliant solution

[source,java,diff-id=1,diff-type=compliant]
----
import com.fasterxml.jackson.databind.ObjectMapper;
import com.fasterxml.jackson.databind.node.ObjectNode;

//[...]
public void doGet(HttpServletRequest req, HttpServletResponse resp) throws IOException {
    ObjectMapper obj = new ObjectMapper();
    ObjectNode myObject = obj.createObjectNode();
    myObject.put("key", request.getParameter("value"));
}
----

=== How does this work?

include::../../common/fix/introduction.adoc[]

include::../../common/fix/object.adoc[]

The example compliant code uses the `ObjectNode` class from the Jackson
library to directly build a JSON object from the user input.

include::../../common/fix/casting.adoc[]