== How to fix it in Gson

=== Code examples

include::../../common/fix/code-rationale.adoc[]

==== Noncompliant code example

[source,java,diff-id=11,diff-type=noncompliant]
----
import com.google.gson.Gson;

//[...]
public void doGet(HttpServletRequest req, HttpServletResponse resp) throws IOException {
    String tainted = request.getParameter("value");
    String json = "{\"key\":\""+ tainted +"\"}";
    Gson obj = new Gson();
    obj.fromJson(json, Object.class); // Noncompliant
}
----

==== Compliant solution

[source,java,diff-id=11,diff-type=compliant]
----
import com.google.gson.Gson;
import com.google.gson.JsonObject;
import com.google.gson.JsonSyntaxException;

//[...]
public void doGet(HttpServletRequest req, HttpServletResponse resp) throws IOException {
    try {
        JsonObject json = new JsonObject();
        json.addProperty("key", request.getParameter("value"));
        Gson obj = new Gson();
        obj.fromJson(json, Object.class);
    } catch (JsonSyntaxException e) {
        response.sendError(400);
    }
}
----

=== How does this work?

include::../../common/fix/introduction.adoc[]

include::../../common/fix/object.adoc[]

The example compliant code uses the `JsonObject` class from the Gson library
to directly build an object from the user input.

include::../../common/fix/casting.adoc[]
