=== How to fix it in Java SE

include::../../common/fix/code-rationale.adoc[]

==== Noncompliant code example

[source,java,diff-id=1,diff-type=noncompliant]
----
public void doGet(HttpServletRequest req, HttpServletResponse resp) throws IOException {
    String tainted = request.getParameter("value");
    String json = "{\"key\":\""+ tainted +"\"}";
    JSONObject obj = new JSONObject(json); // Noncompliant
}
----

==== Compliant solution

[source,java,diff-id=1,diff-type=compliant]
----
public void doGet(HttpServletRequest req, HttpServletResponse resp) throws IOException {
    try {
        int tainted = Integer.parseInt(request.getParameter("value"));
        String json = "{\"key\":\""+ tainted +"\"}";
        JSONObject obj = new JSONObject(json);
    } catch (NumberFormatException e) {
        response.sendError(400);
    }
}
----

=== How does this work?

include::../../common/fix/introduction.adoc[]

include::../../common/fix/casting.adoc[]

The example compliant code uses the `Integer.parseInt` function to convert
the tainted user input to a harmless value. The code also catches 
conversion-related exceptions to avoid information leaks in case of error.
