== How to fix it in a Servlet

=== Code examples

include::../../common/fix/code-rationale.adoc[]

==== Noncompliant code example

[source,java,diff-id=21,diff-type=noncompliant]
----
import org.json.JSONObject;

//[...]
public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
  try {
    String tainted = request.getParameter("value");
    String json = "{\"key\":\""+ tainted +"\"}";
    JSONObject obj = new JSONObject(json); // Noncompliant
  } catch (JsonException e) {
    response.sendError(400);
  }
}
----

==== Compliant solution

[source,java,diff-id=21,diff-type=compliant]
----
import org.json.JSONObject;

//[...]
public void doGet(HttpServletRequest request, HttpServletResponse response) {
  JSONObject obj = new JSONObject();
  obj.put("key", request.getParameter("value"));
}
----

=== How does this work?

include::../../common/fix/introduction.adoc[]

include::../../common/fix/object.adoc[]

The example compliant code uses the `org.json` libraries capabilities to
dynamically build a JSON object without string parsing.

include::../../common/fix/casting.adoc[]
