== How to fix it in Spring

=== Code examples

include::../../common/fix/code-rationale.adoc[]

In this example a json string is parsed using the `BasicJsonParser` class from the `org.springframework.boot.json` package. This class is used to parse JSON strings into a `Map` object. Since the attacker can control the json string they can control the keys and values of the `Map` object.

==== Noncompliant code example

[source,java,diff-id=31,diff-type=noncompliant]
----
import java.util.Map;
import org.springframework.boot.json.BasicJsonParser;

@GetMapping("/example")
public Map <String, Object> example(@RequestParam String input) {
    BasicJsonParser parser = new BasicJsonParser();
    String json = String.format("{\"key\":\"%s\"}", input);
    return parser.parseMap(json); // Noncompliant
}
----

==== Compliant solution

[source,java,diff-id=31,diff-type=compliant]
----
import java.util.Map;

@GetMapping("/example")
public Map <String, Object> example(@RequestParam String input) {   
    return Map.of("key", input);
}
----

=== How does this work?

include::../../common/fix/introduction.adoc[]

include::../../common/fix/object.adoc[]

The example compliant code skips the use of a JSON parser and directly adds the user controlled data into a `Map` object. This way the attacker cannot control the keys and values of the `Map` object.

include::../../common/fix/casting.adoc[]
