This rule raises an issue when a Tensorflow function's `indices` argument has out-of-bound values.

== Why is this an issue?
Tensorflow provides various functions producing matrices based on `input` matrices and their `indices`.
For these functions `indices` is an array of numbers referencing input matrix elements.
Values of `indices` that are out-of-bounds for `input` can lead to errors or unexpected behavior.

Functions sensitive to `indices` argument values:

- `tf.gather`
- `tf.gather_nd`
- `tf.scatter_nd`


== How to fix it

=== Code examples

==== Noncompliant code example

[source,python,diff-id=1,diff-type=noncompliant]
----
import tensorflow as tf

input = tf.constant([1, 2, 3])
indices = [1, 0, 3]
result = tf.gather(input, indices)  # Noncompliant: InvalidArgumentError: indices[2] = 3 is not in [0, 3)
----

==== Compliant solution

[source,python,diff-id=1,diff-type=compliant]
----
import tensorflow as tf

input = tf.constant([1, 2, 3])
indices = [1, 0, 2]
result = tf.gather(input, indices)  # OK
----

//=== How does this work?

//=== Pitfalls

//=== Going the extra mile


== Resources
=== Documentation
* Tensorflow documentation - https://www.tensorflow.org/api_docs/python/tf/gather[tf.gather]
* Tensorflow documentation - https://www.tensorflow.org/api_docs/python/tf/gather_nd[tf.gather_nd]
* Tensorflow documentation - https://www.tensorflow.org/api_docs/python/tf/scatter_nd[tf.scatter_nd]
//=== Articles & blog posts
//=== Conference presentations
//=== Standards
//=== External coding guidelines
//=== Benchmarks
