== How to fix it

To fix the issue, the access of the `{null-keyword}` value needs to be prevented by either:

* ensuring the variable has a value, or
* by checking if the value is not `{null-keyword}`

=== Code examples

==== Noncompliant code example

include::{language}/noncompliant-code.adoc[]

==== Compliant solution

include::{language}/compliant-code.adoc[]
