== How to fix it

To fix the issue the access of the `{null-keyword}` value needs to be prevented by either:

* ensuring the variable has a value, or
* by checking if the value is not `{null-keyword}`

=== Code examples

==== Noncompliant code example

include::{noncompliantCode}[]

==== Compliant solution

include::{compliantCode}[]

