== How to fix it

The fix will depend on the context, in any case, it is important to make to not access `{null-keyword}` value:

* by ensuring the variable has a value
* by checking if the value is not `{null-keyword}`

=== Code examples

==== Noncompliant code example

include::{noncompliantCode}[]

==== Compliant solution

include::{compliantCode}[]

