<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Using a final <code>*)</code> pattern in a <code>case</code> statement is defensive programming. The default clause should either handle unexpected values appropriately or include a comment explaining why no action is taken.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-bash" data-lang="bash">case "$param" in  # missing default clause
  0)
    do_something
    ;;
  1)
    do_something_else
    ;;
esac</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-bash" data-lang="bash">case "$param" in
  0)
    do_something
    ;;
  1)
    do_something_else
    ;;
  *)
    error
    ;;
esac</code></pre>
</div>
</div>
</div>
</div>
</div>