The use of an array structure is useless when the array only has one element. Using an array structure anyway can impact performance and decrease the readability of the source code. 


== Noncompliant Code Example

[source,cobol]
----
03 WS-LIBELLE OCCURS 1.
----


== Compliant Solution

----
03 WS-LIBELLE.
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

endif::env-github,rspecator-view[]
