== Why is this an issue?

The use of an array structure is useless when the array only has one element. Using an array structure anyway can impact performance and decrease the readability of the source code. 


=== Noncompliant code example

[source,cobol]
----
03 WS-LIBELLE OCCURS 1.
----


=== Compliant solution

[source,cobol]
----
03 WS-LIBELLE.
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Remove this use of the "OCCURS 1" clause on data item "XXX".


endif::env-github,rspecator-view[]
