<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Hard-coding bounds in FOR loops is a bad practice, just as magic numbers in general are. Often, those magic bounds can be replaced by dynamic values. If that is not possible, replacing the literal number with a constant is still better.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-sql" data-lang="sql">SET SERVEROUTPUT ON

DECLARE
  TYPE myCollectionType IS VARRAY(3) OF VARCHAR2(42);
  myCollection myCollectionType := myCollectionType('David', 'John', 'Richard');

BEGIN

  FOR i IN 2 .. 3 -- Noncompliant; magic numbers used for the loop bounds
  LOOP
    DBMS_OUTPUT.PUT_LINE('name = ' || myCollection(i));
  END LOOP;

  FOR i IN 2 .. myCollection.LAST -- Noncompliant, better but still magic
  LOOP
    DBMS_OUTPUT.PUT_LINE('name = ' || myCollection(i));
  END LOOP;

END;
/</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-sql" data-lang="sql">SET SERVEROUTPUT ON

DECLARE
  TYPE myCollectionType IS VARRAY(3) OF VARCHAR2(42);
  myCollection myCollectionType := myCollectionType('David', 'John', 'Richard');

BEGIN
  FOR i IN myCollection.FIRST .. myCollection.LAST
  LOOP
    DBMS_OUTPUT.PUT_LINE('name = ' || myCollection(i));
  END LOOP;
END;
/</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Use a constant instead of hard-coding "n".</p>
</div>
</div>
<div class="sect2">
<h3 id="_parameters">Parameters</h3>
<div class="sidebarblock">
<div class="content">
<div class="title">allowedExpressions</div>
<div class="listingblock">
<div class="content">
<pre>0,1</pre>
</div>
</div>
<div class="paragraph">
<p>Comma-separated list of allowed magic numbers</p>
</div>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_relates_to_s109">relates to: <a data-rspec-id="S109" class="rspec-auto-link">S109</a></h3>

</div>
</div>
</div>