== Why is this an issue?

Using the https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/readonly[`readonly` keyword] on a field means it can't be changed after initialization. However, that's only partly true when applied to collections or arrays. The `readonly` keyword enforces that another instance can't be assigned to the field, but it cannot keep the contents from being updated. In practice, the field value can be changed, and the use of `readonly` on such a field is misleading, and you're likely not getting the behavior you expect.

This rule raises an issue when a non-private, `readonly` field is an array or collection.

== How to fix it

To fix this, you should either use an https://learn.microsoft.com/en-us/dotnet/api/system.collections.immutable?view=net-7.0[Immutable collection] or remove the `readonly` field to clarify the behavior.

=== Code examples

==== Noncompliant code example


[source,csharp]
----
public class MyClass
{
  public readonly string[] strings;  // Noncompliant

  // ...
}
----


==== Compliant solution

[source,csharp]
----
public class MyClass
{
  public string[] strings;

  // ...
----
or

[source,csharp]
----
public class MyClass
{
  public readonly ImmutableArray<string> strings;

  // ...
----
or

[source,csharp]
----
public class MyClass
{
  private readonly string[] strings;

  // ...
----

== Resources

=== Documentation

* Microsoft Learn - https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/readonly[readonly (C# Reference)]
