== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,sql]
----
CREATE PROCEDURE greet
  @Name varchar(20),
  @Greeting varchar(25) OUTPUT  -- Noncompliant
AS
  DECLARE @Message VARCHAR(45)
  SET @Message = N'Hello ' + RTRIM(@Name);
  PRINT @Message
GO
----

=== Compliant solution

[source,sql]
----
CREATE PROCEDURE greet
  @Name varchar(20),
  @Greeting varchar(25) OUTPUT
AS
  SELECT @Greeting = N'Hello ' + RTRIM(@Name);
GO
----
ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../highlighting.adoc[]

endif::env-github,rspecator-view[]
