== Why is this an issue?

include::../description.adoc[]

Code like

[source,javascript]
----
if (x != undefined) {
  if (y === 2) {
    // ...
  }
}
----

Will be more readable as

[source,javascript]
----
if (x != undefined && y === 2) {
  // ...
}
----


== How to fix it

If merging the conditions seems to result in a more complex code, extracting the condition or part of it in a named function or variable is a better approach to fix readability.

=== Code examples

==== Noncompliant code example

[source,javascript]
----
if (file != undefined) {
  if (file.isFile() || file.isDirectory()) {        // Noncompliant
    /* ... */
  }
}
----

==== Compliant solution

[source,javascript]
----
function isFileOrDirectory(File file) {
  return file.isFile() || file.isDirectory();
}

/* ... */

if (file. != undefined && isFileOrDirectory(file)) { // Compliant
  /* ... */
}
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../highlighting.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
