include::../description.adoc[]

== Noncompliant Code Example

----
if (x != undefined) {
  if (y === 2) {
    // ...
  }
}
----

== Compliant Solution

----
if (x != undefined && y === 2) {
  // ...
}
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../highlighting.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
