== Why is this an issue?

Just as pain is your body's way of telling you something is wrong, errors are PHP's way of telling you there's something you need to fix. Neither pain, nor PHP errors should be ignored.


=== Noncompliant code example

[source,php]
----
@doSomethingDangerous($password);  // Noncompliant; '@' silences errors from function call
----


=== Compliant solution

[source,php]
----
doSomethingDangerous($password);
----



ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
