== Why is this an issue?

While it is legal to place ``++#define++`` and ``++#undef++`` directives anywhere in a source file, placing them outside of the global namespace is misleading since their scope is not actually restricted. This may be inconsistent with developer expectations.


=== Noncompliant code example

[source,cpp]
----
namespace NS
{
  #ifndef MY_HDR
  #define MY_HDR    /* Noncompliant */
  #undef FOO        /* Noncompliant */
  #endif
}
----


=== Compliant solution

[source,cpp]
----
#ifndef MY_HDR
#define MY_HDR
#undef FOO
#endif
----


== Resources

* MISRA C:2004, 19.5 - Macros shall not be #define'd or #undef'd within a block.
* MISRA {cpp}:2008, 16-0-2 - Macros shall only be #define'd or #undef'd in the global namespace.


ifdef::env-github,rspecator-view[]
'''
== Comments And Links
(visible only on this page)

=== on 31 Mar 2015, 19:05:43 Evgeny Mandrikov wrote:
\[~ann.campbell.2] implementation seems more complete (SQALE, description) than this spec.

=== on 13 Apr 2015, 19:51:04 Evgeny Mandrikov wrote:
\[~ann.campbell.2] I'm wondering why blocker, but not active by default? Note that in implementation currently major and active.

endif::env-github,rspecator-view[]
