While it is legal to place ``++#define++`` and ``++#undef++`` directives anywhere in a source file, placing them outside of the global namespace is misleading since their scope is not actually restricted. This may be inconsistent with developer expectations.


== Noncompliant Code Example

----
namespace NS
{
  #ifndef MY_HDR
  #define MY_HDR    /* Noncompliant */
  #undef FOO        /* Noncompliant */
  #endif
}
----


== Compliant Solution

----
#ifndef MY_HDR
#define MY_HDR
#undef FOO
#endif
----


== See

* MISRA C:2004, 19.5 - Macros shall not be #define'd or #undef'd within a block.
* MISRA {cpp}:2008, 16-0-2 - Macros shall only be #define'd or #undef'd in the global namespace.


ifdef::env-github,rspecator-view[]
'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
