<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Unlike class and instance methods, static methods don&#8217;t receive an implicit first argument. Nonetheless naming the first argument <code>self</code> or <code>clz</code> guarantees confusion - either on the part of the original author, who may never understand why the arguments don&#8217;t hold the values he expected, or on that of future maintainers.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">class MyClass:
  @staticmethod
  def s_meth(self, arg1, arg2):  #Noncompliant
    # ...</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">class MyClass:
  @staticmethod
  def s_meth(arg1, arg2):  #Noncompliant
    # ...</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_duplicates_s2854">duplicates: <a data-rspec-id="S2854" class="rspec-auto-link">S2854</a></h3>

</div>
</div>
</div>