== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,kubernetes]
----
apiVersion: v1
kind: Pod
metadata:
  name: test
spec:
  containers: # TODO
  - image: k8s.gcr.io/test-webserver
    name: test-container
    volumeMounts:
    - mountPath: /data
      name: test-volume
----

include::../see.adoc[]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
