== How to fix it in CryptoSwift

=== Code examples

==== Noncompliant code example

[source,swift,diff-id=11,diff-type=noncompliant]
----
import Foundation
import CryptoSwift

func encrypt(data: [UInt8], key: [UInt8]) {
    let aes = try! AES(
        key: key,
        blockMode: CBC(iv: [
            0x01, 0x02, 0x03, 0x04, 0x05, 0x06, 0x07, 0x08, 0x09, 0x0A, 0x0B, 0x0C, 0x0D, 0x0E,
            0x0F, 0x10,
        ]),
        padding: .pkcs7,
    )
    let encrypted = try! aes.encrypt(data)

    // ...
}
----

==== Compliant solution

include::../../common/fix/explicit-fix.adoc[]

[source,swift,diff-id=11,diff-type=compliant]
----
import Foundation
import CryptoSwift

func encrypt(data: [UInt8], key: [UInt8]) {
    let iv = AES.randomIV(16)
    let aes = try! AES(
        key: key,
        blockMode: CBC(iv: iv),
        padding: .pkcs7,
    )
    let encrypted = try! aes.encrypt(data)

    // ...
}
----

=== How does this work?

include::../../common/fix/fix.adoc[]
