<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when backticks (`) are used for command substitution instead of the modern `$()` syntax.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Backticks (`) are an older way to perform command substitution in shell scripts. While they still work, they have several drawbacks compared to the modern `$()` syntax.</p>
</div>
<div class="paragraph">
<p>The main problems with backticks are:</p>
</div>
<div class="ulist">
<ul>
<li>
<p><strong>Poor readability</strong>: Backticks can be hard to distinguish from regular quotes, especially in small fonts or when printed</p>
</li>
<li>
<p><strong>Nesting difficulties</strong>: You cannot easily nest backticks without complex escaping</p>
</li>
<li>
<p><strong>Editor limitations</strong>: Many text editors don&#8217;t highlight backtick syntax as well as <code>$()</code></p>
</li>
</ul>
</div>
<div class="paragraph">
<p>The <code>$()</code> syntax was introduced as part of POSIX standards to address these issues. It provides the same functionality but with better visual clarity and more flexible usage patterns.</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="paragraph">
<p>Using backticks instead of <code>$()</code> reduces code readability and maintainability. Complex command substitutions become harder to understand and modify, especially when nesting is required. This can lead to bugs during script maintenance and makes code reviews more difficult.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Replace backticks with the modern <code>$()</code> syntax for command substitution. This improves readability and allows for easier nesting of commands.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-shell" data-lang="shell">result=`date`                               # Noncompliant
files=`find /path -name "*.txt" | head -5`  # Noncompliant
count=`echo \`ls *.txt | wc -l\``           # Noncompliant</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-shell" data-lang="shell">result=$(date)
files=$(find /path -name "*.txt" | head -5)
count=$(echo $(ls *.txt | wc -l))</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Bash Manual - Command Substitution - <a href="https://www.gnu.org/software/bash/manual/html_node/Command-Substitution.html">Official Bash documentation explaining command substitution syntax</a></p>
</li>
<li>
<p>POSIX Shell Command Language - <a href="https://pubs.opengroup.org/onlinepubs/9699919799/utilities/V3_chap02.html#tag_18_06_03">POSIX standard specification for command substitution</a></p>
</li>
</ul>
</div>
</div>
<div class="sect2">
<h3 id="_standards">Standards</h3>
<div class="ulist">
<ul>
<li>
<p>POSIX.1-2017 Shell Command Language - <a href="https://pubs.opengroup.org/onlinepubs/9699919799/utilities/V3_chap02.html">POSIX standard that defines modern shell syntax including $() command substitution</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>