== Why is this an issue?

In JavaScript, the ``++/++`` (backslash) character indicates that the next character should be treated as a literal character rather than as a special character or string delimiter.
For instance, it is common to escape single quotes inside a string literal using the single quote delimiter like ``++'It\'s a beautiful day'++``. Escaping is only meaningful for special characters.
Escaping non-special charactwers in strings, template literals, and regular expressions doesn't affect their value.
Therefore, useless escapes impact code readability and could even denote a bug in the code if the developer left it by mistake or intended to escape another special character instead.

== How to fix it

You should check if the escape character was not misplaced. Useless character escapes can safely be removed without changing the original behavior.

=== Code examples

==== Noncompliant code example

[source,javascript,diff-id=1,diff-type=noncompliant]
----
const regex = /[\[]/;
const octal = '\8';
const hello = 'Hello, world\!';
const path  = `\/${some}\/${dir}`;
----

==== Compliant solution

[source,javascript,diff-id=1,diff-type=compliant]
----
const regex = /[[]/;
const octal = '8';
const hello = 'Hello, world!';
const path  = `/${some}/${dir}`;
----
