include::../description.adoc[]

== Noncompliant Code Example

----
towns[i] = "London";
towns[i] = "Chicago";  // Noncompliant
----

ifdef::env-github,rspecator-view[]
'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
