== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,cpp,diff-id=1,diff-type=noncompliant]
----
towns[i] = "London";
towns[i] = "Chicago";  // Noncompliant
----

=== Compliant solution

[source,cpp,diff-id=1,diff-type=compliant]
----
towns[i] = "London";
towns[j] = "Chicago";
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../highlighting.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
