include::../description.adoc[]

== Noncompliant Code Example

----
// This can never match because $ and ^ have been switched around
Pattern.compile("$[a-z]+^"); // Noncompliant
----


== Compliant Solution

----
Pattern.compile("^[a-z]+$");
----

ifdef::env-github,rspecator-view[]
'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
