<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Salesforce Governor Limits do not allow more than 10 calls to <code>Messaging.sendEmail</code> in a single transaction. There is a good chance that calling this method in a loop will reach that limit and fail. You can instead send a batch of emails with a single call to <code>Messaging.sendEmail</code>.</p>
</div>
<div class="paragraph">
<p>This rule raises an issue when a call to <code>Messaging.sendEmail</code> is found in a loop.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-apex" data-lang="apex">trigger MyWelcomeTrigger on Contact (after insert) {
    List&lt;Id&gt; toIds = new List&lt;Id&gt;();
    for (Contact contact : trigger.new)
    {
        if(contact.Email != null)
        {
            Messaging.SingleEmailMessage mail = new Messaging.SingleEmailMessage();
        	String[] toAddresses = new String[] { contact.email };
        	mail.setToAddresses(toAddresses);
        	mail.setSubject('Welcome');
        	mail.setPlainTextBody('Welcome');
        	Messaging.sendEmail(new Messaging.SingleEmailMessage[] { mail });  // Noncompliant
        }
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-apex" data-lang="apex">trigger MyWelcomeTrigger on Contact (after insert) {
    List&lt;Id&gt; toIds = new List&lt;Id&gt;();
    for (Contact contact : trigger.new)
    {
        if(contact.Email != null)
        {
            toIds.add(contact.Id);
        }
    }
    string templateName = 'Welcome Email Template';
    EmailTemplate template = [select Id, Name from EmailTemplate where name = :templateName];
    Messaging.MassEmailMessage mail = new Messaging.MassEmailMessage();
    mail.setTargetObjectIds(toIds);
    mail.setTemplateId(template.Id);
    Messaging.sendEmail(new Messaging.MassEmailMessage[] { mail });
}</code></pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p><a href="https://developer.salesforce.com/docs/atlas.en-us.apexcode.meta/apexcode/apex_gov_limits.htm">Execution Governors and Limits</a></p>
</li>
</ul>
</div>
<hr>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Send these emails in batch</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>Primary Location: the call toÂ "Messaging.SendEmail"</p>
</div>
<div class="paragraph">
<p>Secondary Location: the "do", "while" or "for" keyword of the loop.</p>
</div>
</div>
</div>
</div>