<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule is a strict implementation of a MISRA (Motor Industry Software Reliability Association) rule. MISRA defines best practices for developing safety-critical software. You can learn more about this rule in the MISRA documents referenced below.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p>MISRA C&#43;&#43; 2008, 3-3-1</p>
</li>
</ul>
</div>
<hr>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Either declare this [variable||function] in a header or mark it 'static'.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_is_duplicated_by_s827">is duplicated by: <a data-rspec-id="S827" class="rspec-auto-link">S827</a></h3>

</div>
<div class="sect2">
<h3 id="_is_related_to_s806">is related to: <a data-rspec-id="S806" class="rspec-auto-link">S806</a></h3>

</div>
<div class="sect2">
<h3 id="_is_related_to_s830">is related to: <a data-rspec-id="S830" class="rspec-auto-link">S830</a></h3>

</div>
<div class="sect2">
<h3 id="_is_related_to_s832">is related to: <a data-rspec-id="S832" class="rspec-auto-link">S832</a></h3>

</div>
<div class="sect2">
<h3 id="_on_16_oct_2014_125305_ann_campbell_wrote">on 16 Oct 2014, 12:53:05 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~samuel.mercier] please:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>fill in the appropriate reference field(s).</p>
</li>
<li>
<p>provide a See section.</p>
</li>
<li>
<p>use the standard section titles: Noncompliant Code Example, Exception*s*, and heading levels (h2. instead of h3.).</p>
</li>
<li>
<p>use the standard section order: description, Noncompliant Code Example, Compliant Solution, Exceptions, See* SQALE.</p>
</li>
<li>
<p>use our standard: "Noncompliant", rather than "Non-compliant" (code sample).</p>
</li>
</ul>
</div>
</div>
<div class="sect2">
<h3 id="_on_17_oct_2014_155108_ann_campbell_wrote">on 17 Oct 2014, 15:51:08 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~freddy.mallet] this an <a data-rspec-id="S832" class="rspec-auto-link">RSPEC-832</a> overlap to my mind.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_17_oct_2014_190806_ann_campbell_wrote">on 17 Oct 2014, 19:08:06 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>FYI [~samuel.mercier] I&#8217;ve moved this to Data-related reliability - the danger appears to be that a non-static var could be visible (and thus accidentally used!) outside its intended scope. Change it back if I&#8217;m wrong.</p>
</div>
</div>
</div>
</div>