Sometimes there is a need to cancel the execution of the coroutine after some time. You can do it manualy using `delay()` and `cancel()` functions. But this way is verbose and errorprone. There is an easier way to manage timeouts by using functions `withTimeout()` or `withTimeoutOrNull()`.

`withTimeout` function will throw a `TimeoutCancellationException` in case timeout was reached, while `withTimeoutOrNull` will just return null in this case.

This rule raises an issue if timeouts mechanism is implemented manually instead of using appropriate functions.

== Noncompliant Code Example

----
suspend fun main() {
    coroutineScope {
        val job = launch {
            delay(2000L)
            println("Finished")
        }
        delay(500L)
        job.cancel()
    }
}
----

== Compliant Solution

----
suspend fun main() {
    coroutineScope {
        withTimeoutOrNull(1000L){
            delay(2000L)
            println("Finished")
        }
    }
}
----

== See

* https://kotlinlang.org/docs/cancellation-and-timeouts.html[Cancellation and timeouts]
