include::../description.adoc[]

== Noncompliant Code Example

----
a = true;return 0;
----

== Compliant Solution

----
a = true;
return 0;
----

ifdef::env-github,rspecator-view[]
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
