<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when a configured Java package or class is used in a JSP file.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="paragraph">
<p>Given a disallowed list of <code>java.sql,ArrayList</code>:</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-html" data-lang="html">&lt;%@ page import="java.sql.*" %&gt;     &lt;!-- Noncompliant --&gt;
&lt;% java.util.ArrayList clients; %&gt;  &lt;!-- Noncompliant --&gt;
&lt;% java.lang.String name; %&gt;        &lt;!-- Compliant --&gt;</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove the use of "xxx".</p>
</div>
</div>
<div class="sect2">
<h3 id="_parameters">Parameters</h3>
<div class="sidebarblock">
<div class="content">
<div class="title">libraries</div>
<div class="paragraph">
<p>Comma-separated list of Java packages or classes, such as java.sql or java.util.ArrayList</p>
</div>
</div>
</div>
<div class="sidebarblock">
<div class="content">
<div class="title">message</div>
<div class="listingblock">
<div class="content">
<pre>Remove the usage of this library which is not allowed.</pre>
</div>
</div>
<div class="paragraph">
<p>Issue message which is displayed in case of violation</p>
</div>
</div>
</div>
</div>
</div>
</div>