<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Program/file names offer only very limited space for indicating program function, which is why you should take advantage of the ability to specify a program <code>TITLE</code>. Omitting the <code>TITLE</code> statement will result in a default, uncommunicative <code>TITLE</code> value being printed at the top of each page of the source listing. Instead, you should write an expressive title that gives a clear impression of the program&#8217;s function.</p>
</div>
<div class="paragraph">
<p>This rule raises an issue when a there is no <code>TITLE</code> before the <code>IDENTIFICATION DIVISION</code>. Ideally, <code>TITLE</code> will be the first line of a program, but it cannot be placed before compiler options.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cobol" data-lang="cobol">      * Copyright (c) 2012 MyCo. All rights reserved.  *&gt; Noncompliant
       IDENTIFICATION DIVISION.</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cobol" data-lang="cobol">       TITLE 'IntradayPosition.ExecutePostProcessing'
      * Copyright (c) 2012 MyCo. All rights reserved.
       IDENTIFICATION DIVISION.</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="ulist">
<ul>
<li>
<p>Add a title to this program.</p>
</li>
<li>
<p>Move this title to the first line of the file.</p>
</li>
</ul>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p><code>TITLE</code> if it exists on the wrong line, otherwise file level issue</p>
</div>
</div>
</div>
</div>