<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Complex types can be restricted by means of an <code>xs:restriction</code> element. But <code>restriction</code> is anti-inheritance: the derived type has less information than the base type. Further, restriction causes the redefinition of the complex types elements and attributes, which leads to fragility in your XML schema; any change to a super type will require corresponding changes in all restricting subtypes, even those in other schemas. For example, if the base type defines a mandatory "color" element of type <code>xs:int</code>, the restricting type must also define a "color" element whose type cannot be, say, <code>xs:string</code>.</p>
</div>
<div class="paragraph">
<p><code>xs:extension</code>, on the other hand, allows the extending element to inherit the base class' elements and add new ones.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-xml" data-lang="xml">&lt;xs:complexType name="Fruit&gt;
  &lt;xs:sequence&gt;
    &lt;xs:element name="name" type="xs:string"/&gt;
    &lt;xs:element name="color" type="xs:string" minOccurs="0"/&gt;
  &lt;/xs:sequence&gt;
&lt;/xs:complexType&gt;
&lt;xs:complexType name="UncoloredFruit"&gt;
  &lt;xs:complexContent&gt;
    &lt;xs:restriction base="Fruit"&gt;  &lt;!-- Noncompliant --&gt;
      &lt;xs:sequence&gt;
        &lt;xs:element name="name" type="xs:string"/&gt;
      &lt;/xs:sequence&gt;
    &lt;/xs:restriction&gt;
  &lt;/xs:complexContent&gt;
&lt;/xs:complexType&gt;</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-xml" data-lang="xml">&lt;xs:complexType name="FruitBase&gt;
  &lt;xs:sequence&gt;
    &lt;xs:element name="name" type="xs:string"/&gt;
  &lt;/xs:sequence&gt;
&lt;/xs:complexType&gt;
&lt;xs:complexType name="ColoredFruit&gt;
  &lt;xs:sequence&gt;
    &lt;!-- Bottom-up design, using aggregation --&gt;
    &lt;xs:element name="base" type="FruitBase"/&gt;
    &lt;xs:element name="color" type="xs:string" minOccurs="0"/&gt;
  &lt;/xs:sequence&gt;
&lt;/xs:complexType&gt;</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove this "restriction", and consider "extension" instead.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p><code>xs:restriction</code></p>
</div>
</div>
</div>
</div>