<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Arguments evaluation order in a function call is not specified:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>Before C&#43;&#43;17, the evaluation of each argument was unsequenced with the evaluation of other arguments, which can lead to undefined behavior if the same value is modified in several arguments,</p>
</li>
<li>
<p>After C&#43;&#43;17, it is sequenced, but in an unspecified order: the behavior is not longer undefined, but the values passed to the function will be non portable.</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>Both cases should be avoided, because the code will probably not be what was expected.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">void f(int i, int j);

void g() {
  int i = 0;
  f(++i, ++i); // Noncompliant, the call could either be f(1,2) or f(2,1) (since C++17) or undefined behavior (before C++17)
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_exceptions">Exceptions</h3>
<div class="paragraph">
<p>This rule does not apply to overloaded operators because they respect the sequencing order rules of the operator they overload (since C&#43;&#43;17).</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p>C++ Core Guidelines - <a href="https://github.com/isocpp/CppCoreGuidelines/blob/e49158a/CppCoreGuidelines.md#es44-dont-depend-on-order-of-evaluation-of-function-arguments">ES.44: Don&#8217;t depend on order of evaluation of function arguments</a></p>
</li>
<li>
<p>C&#43;&#43; reference - <a href="https://en.cppreference.com/w/cpp/language/eval_order">cppreference.com - order of evaluation</a></p>
</li>
</ul>
</div>
<hr>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_25_oct_2019_155741_geoffray_adde_wrote">on 25 Oct 2019, 15:57:41 Geoffray Adde wrote:</h3>
<div class="paragraph">
<p>I think we should clarify what we are going to raise because this is a quite complex topic (cf. the second link in the section See).</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_8_nov_2019_184712_loïc_joly_wrote">on 8 Nov 2019, 18:47:12 Loïc Joly wrote:</h3>
<div class="paragraph">
<p>\[~amelie.renard]</p>
</div>
<div class="paragraph">
<p>I wonder why we did not select the more generic rule ES.43 Avoid expressions with undefined order of evaluation, even if for now only the case of function arguments is detected?</p>
</div>
<div class="paragraph">
<p>Moreover, the exception only applies to C&#43;&#43;17, maybe we should detect it for old code?</p>
</div>
</div>
</div>
</div>