== Why is this an issue?

Any statement after a ``++STOP RUN++`` or ``++GOBACK++`` is unreachable and therefore dead code which should be removed.


=== Noncompliant code example

[source,cobol]
----
PARAGRAPH1.  
  MOVE A TO B.         
  STOP RUN. 
  MOVE B TO C.
----

or


[source,cobol]
----
PARAGRAPH1.  
  MOVE A TO B.         
  GOBACK. 
  MOVE B TO C.
----


=== Compliant solution

[source,cobol]
----
PARAGRAPH1.  
  MOVE A TO B.         
  MOVE B TO C.
  STOP RUN. 
----

or


[source,cobol]
----
PARAGRAPH1.  
  MOVE A TO B.
  MOVE B TO C.         
  GOBACK. 
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
