<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>A common anti-pattern is to check that a key exists in a dictionary before retrieving its corresponding value and providing a default value otherwise. This pattern works but is less readable than the equivalent call to the built-in dictionary method "get()" with a default value.</p>
</div>
<div class="paragraph">
<p>Note that if a default value is set for every key of the dictionary it is possible to use python&#8217;s defaultdict instead.</p>
</div>
<div class="paragraph">
<p>This rule raises an issue when a key presence is checked before retrieving its value or providing a default value. It only raises an issue when the default value is a hard-coded string, number, list, dictionary or tuple. Computed values will not raise an issue as they can have side-effects.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">result = "default"
if "missing" in mydict:
    result = mydict["missing"]  # Noncompliant

if "missing" in mydict:
    result = mydict["missing"]  # Noncompliant
else:
    result = "default"

if "missing" in mydict:
    result = mydict["missing"]  # Compliant. No issue is raised as generate_value() might have some side-effect.
else:
    result = generate_value()</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">result = mydict.get("missing", "default")

# OR, if "default" is the default value for every key

from collections import defaultdict
mydict = defaultdict(lambda: "default")
result =  mydict["missing"]</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Replace this key check and dictionary access with a call to "get(&#8230;&#8203;, default)"</p>
</div>
</div>
</div>
</div>