<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Because library names could change from environment to environment, they should never be hard-coded in a program. Instead, you should use a variable to specify the library name and set that variable with data external to the program.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-rpg" data-lang="rpg">CLRPFM   FILE(XPJLIB/FILE01)</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-rpg" data-lang="rpg">CHGVAR     VAR(&amp;LIB01) VALUE(&amp;PREFIX *CAT 'LIB')
CLRPFM      FILE(&amp;LIB01/FILE01)</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_exceptions">Exceptions</h3>
<div class="paragraph">
<p>Hard-coding references to QTEMP, which is a temporary library, is allowed.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Use a variable for the library name and set it with data external to the program.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_12_oct_2014_165140_freddy_mallet_wrote">on 12 Oct 2014, 16:51:40 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>Just a feeling @Ann, but I would decrease the severity to Major and would not activate this rule by default.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_14_oct_2014_144834_ann_campbell_wrote">on 14 Oct 2014, 14:48:34 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~freddy.mallet] breaking this rule can (will) cause runtime failures, which is why I&#8217;ve made it critical. I set it to on by default because I&#8217;ve had 2 different companies ask me for the rule (and the second actually helped me specify it).</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_14_oct_2014_210433_freddy_mallet_wrote">on 14 Oct 2014, 21:04:33 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>OK</p>
</div>
</div>
</div>
</div>