=== Noncompliant code example

[source,text]
----
public class Rectangle {
  private int height;
  private int width;  //width is written but never read

  public Rectangle(int height, int width) {
    this.height=height;
    this.width = width;
  }

  public int getArea() {
    return height * height;
  }
}
----
