<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>The <code>switch</code> statement should be used only to clearly define some new branches in the control flow. As soon as a <code>when</code> clause contains too many statements this highly decreases the readability of the overall control flow statement. In such case, the content of the <code>when</code> clause should be extracted into a dedicated function.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="paragraph">
<p>With the threshold set at <code>5</code>:</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-apex" data-lang="apex">public void foo(Integer value) {
  switch on value {
    when 1 {
      methodCall1('');
      methodCall2('');
      methodCall3('');
      methodCall4('');
      methodCall5('');
    }
    when 2 { /* ... */ }
  }
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-apex" data-lang="apex">public void foo(Integer value) {
  switch on value {
    when 1 { doSomething(); }
    when 2 { /* ... */ }
    }
  }
}

private void doSomething() {
  methodCall1('');
  methodCall2('');
  methodCall3('');
  methodCall4('');
  methodCall5('');
}</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Reduce this "[switch/CASE|CASE/WHEN]" number of lines from {} to at most {}, for example by extracting code into a [method|subroutine|function].</p>
</div>
</div>
<div class="sect2">
<h3 id="_parameters">Parameters</h3>
<div class="sidebarblock">
<div class="content">
<div class="title">max</div>
<div class="paragraph">
<p><em>Integer</em></p>
</div>
<div class="listingblock">
<div class="content">
<pre>5</pre>
</div>
</div>
<div class="paragraph">
<p>Maximum number of lines of code</p>
</div>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_is_duplicated_by_s137">is duplicated by: <a data-rspec-id="S137" class="rspec-auto-link">S137</a></h3>

</div>
<div class="sect2">
<h3 id="_on_8_aug_2013_093718_dinesh_bolkensteyn_wrote">on 8 Aug 2013, 09:37:18 Dinesh Bolkensteyn wrote:</h3>
<div class="paragraph">
<p>Implemented by https://jira.codehaus.org/browse/SONARJAVA-238</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_31_oct_2013_164100_freddy_mallet_wrote">on 31 Oct 2013, 16:41:00 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>Is implemented by http://jira.codehaus.org/browse/SONARPLUGINS-3235 for Flex</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_12_feb_2014_175822_freddy_mallet_wrote">on 12 Feb 2014, 17:58:22 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>Is implemented by http://jira.codehaus.org/browse/SONARPLUGINS-3449 for PHP</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_2_jul_2015_081108_evgeny_mandrikov_wrote">on 2 Jul 2015, 08:11:08 Evgeny Mandrikov wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] I find it weird to see colon before the header in description.</p>
</div>
</div>
</div>
</div>