This rule raises an issue when closures rely on implicit parameter declarations instead of using explicit parameter syntax.

== Why is this an issue?

Groovy closures can use implicit parameters, but this can lead to confusion and reduced code readability.

When a closure doesn't explicitly declare parameters, Groovy automatically provides an implicit parameter named `it`. While this might seem convenient, it can create several problems:

* **Unclear intent**: It's not immediately obvious whether the closure is intended to accept parameters or not
* **Accidental parameter acceptance**: A closure intended to accept no parameters will still accept one parameter through the implicit `it`
* **Poor readability**: Using `it` doesn't convey the meaning or purpose of the parameter
* **Maintenance difficulties**: Future developers may not understand what `it` represents in the context

Explicit parameter declarations make the code self-documenting and prevent misunderstandings about the closure's interface.

=== What is the potential impact?

Using implicit parameters can lead to:

* **Runtime errors**: Closures may be called with unexpected arguments when they were intended to accept none
* **Reduced code maintainability**: Other developers may struggle to understand the purpose of parameters
* **Debugging difficulties**: Issues related to parameter usage become harder to trace
* **Code review challenges**: Reviewers cannot easily determine the intended closure interface

== How to fix it

For closures that should accept no parameters, use the explicit empty parameter list syntax `{->}`. This clearly indicates that the closure expects no arguments and will fail if called with arguments.

=== Code examples

==== Noncompliant code example

[source,groovy,diff-id=1,diff-type=noncompliant]
----
def magicNumber = { 42 }
// This closure implicitly accepts one parameter 'it'
// but is intended to accept none
----

==== Compliant solution

[source,groovy,diff-id=1,diff-type=compliant]
----
def magicNumber = {-> 42 }
// Explicitly declares no parameters
// Will fail if called with arguments
----

== Resources

=== Documentation

 * Groovy Closures Documentation - https://groovy-lang.org/closures.html[Official Groovy documentation covering closure syntax, parameters, and best practices]
