<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>While named function expressions might be useful for debugging purposes, some browsers do not support them correctly (for example Internet Explorer 8).</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-javascript" data-lang="javascript">f = function fun(){}; // Noncompliant;  named function expression</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-javascript" data-lang="javascript">fun = function(){}; // Compliant; function expression</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_exceptions">Exceptions</h3>
<div class="paragraph">
<p>ECMAScript 6 generator functions are excluded from this rule.</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-javascript" data-lang="javascript">function* f() {} // Compliant; generator function.</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_1_nov_2019_162641_elena_vilchik_wrote">on 1 Nov 2019, 16:26:41 Elena Vilchik wrote:</h3>
<div class="paragraph">
<p>See https://github.com/SonarSource/SonarJS/issues/1698</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_19_oct_2020_104806_guillaume_dequenne_wrote">on 19 Oct 2020, 10:48:06 Guillaume Dequenne wrote:</h3>
<div class="paragraph">
<p>Dropping this rule as it has low value. See:</p>
</div>
<div class="paragraph">
<p><a href="https://github.com/SonarSource/SonarJS/issues/2208" class="bare">https://github.com/SonarSource/SonarJS/issues/2208</a></p>
</div>
<div class="paragraph">
<p><a href="https://github.com/SonarSource/SonarJS/issues/1698" class="bare">https://github.com/SonarSource/SonarJS/issues/1698</a></p>
</div>
</div>
</div>
</div>