While named function expressions might be useful for debugging purposes, some browsers do not support them correctly (for example Internet Explorer 8).


== Noncompliant Code Example

----
f = function fun(){}; // Noncompliant;  named function expression
----


== Compliant Solution

----
fun = function(){}; // Compliant; function expression
----


== Exceptions

ECMAScript 6 generator functions are excluded from this rule. 

----
function* f() {} // Compliant; generator function.
----

