This rule raises an issue when type aliasing is done without the `type` keyword.

== Why is this an issue?

Since Python 3.12 the keyword `type` is used to defined type aliases. 
It replaces the following construct:

[source,text]
----
from typing import TypeAlias

_T = TypeVar("_T")

MyTypeAlias: TypeAlias = set[_T]
----

Using the new `type` keyword to define type aliases allows for a more concise code and thus better readability.

[source,text]
----
type MyTypeAlias[T] = set[T]
----

== How to fix it

Refactoring the `TypeAlias` assignation with the help of the `type` keyword will fix this issue.

=== Code examples

==== Noncompliant code example

[source,python,diff-id=1,diff-type=noncompliant]
----
from typing import TypeAlias

MyStringAlias: TypeAlias = str # Noncompliant: this TypeAlias can be more concise with the help of the type keyword.

_T = TypeVar("_T")
MyGenericAlias: TypeAlias = list[_T]  # Noncompliant: the type keyword can help replace both the TypeVar and the TypeAlias statements.
----

==== Compliant solution

[source,python,diff-id=1,diff-type=compliant]
----
type MyStringAlias = str # Compliant

type MyGenericAlias[T] = list[T]  # Noncompliant: the type keyword can help replace both the TypeVar and the TypeAlias statements.
----


== Resources
=== Documentation

* Python Documentation - https://docs.python.org/3.12/reference/simple_stmts.html#type[The type statement]
* Python 3.12 Release Notes - https://docs.python.org/3.12/whatsnew/3.12.html#pep-695-type-parameter-syntax[PEP 695: Type Parameter Syntax]

ifdef::env-github,rspecator-view[]

'''

== Implementation Specification
=== Message
(visible only on this page)

Refactor this type alias to use the \"type\" keyword.


endif::env-github,rspecator-view[]
