== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,sql]
----
DECLARE @x INT = (@y / 2 + 1); -- Compliant even if the parentheses are ignored
IF (@x > 0) AND ((@x+@y > 0)) -- Noncompliant
BEGIN
  -- ...
END
----

=== Compliant solution

[source,sql]
----
DECLARE @x INT = (@y / 2 + 1);
IF (@x > 0) AND (@x+@y > 0)
BEGIN
  -- ...
END
----

include::../rspecator.adoc[]
