include::../description.adoc[]

== Noncompliant Code Example

----
DECLARE @x INT = (@y / 2 + 1); -- Compliant even if the parentheses are ignored
IF (@x > 0) AND ((@x+@y > 0)) -- Noncompliant
BEGIN
  -- ...
END
----

== Compliant Solution

----
DECLARE @x INT = (@y / 2 + 1);
IF (@x > 0) AND (@x+@y > 0)
BEGIN
  -- ...
END
----

ifdef::env-github,rspecator-view[]
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
