This rule raise an issue when one of the mandatory label are missing.

== Why is this an issue?

Adding labels to your image help to organize images by project, record licensing information, aid in automation and for other reasons. +
The rule provide the possibility to configure the list of mandatory label that must be present in every Dockerfile.

== How to fix it

=== Code examples

==== Noncompliant code example

With `labels` set to `author,description,version`
[source,docker,diff-id=1,diff-type=noncompliant]
----
FROM ubuntu:22.02
RUN my_command
----

==== Compliant solution

[source,docker,diff-id=1,diff-type=compliant]
----
FROM ubuntu:22.02
LABEL author="John Doe"
LABEL description="Image doing XYZ"
LABEL version=1.0
RUN my_command
----

== Resources

=== Documentation

* Docker docs - https://docs.docker.com/build/building/best-practices/#label[Building best practices - LABEL]
