This rule raise an issue when one of the mandatory label are missing.

== Why is this an issue?

Adding labels to your image help to organize images by project, record licensing information, aid in automation and for other reasons. +
The rule provide the possibility to configure the list of mandatory label that must be present in every Dockerfile.

== How to fix it

=== Code examples

==== Noncompliant code example

With `labels` set to `maintainer,description,version`
[source,docker,diff-id=1,diff-type=noncompliant]
----
FROM ubuntu:22.02
RUN my_command
----

==== Compliant solution

[source,docker,diff-id=1,diff-type=compliant]
----
FROM ubuntu:22.02
LABEL maintainer="John Doe"
LABEL description="Image doing XYZ"
LABEL version=1.0
RUN my_command
----

== Resources

=== Documentation

* Docker docs - https://docs.docker.com/build/building/best-practices/#label[Building best practices - LABEL]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Single label is missing:
Add the missing label `%s`.
Multiple labels are missing:
Add the missing labels; `%s`, `%s` and `%s`.

=== Highlighting

Highlight the first FROM instruction.

'''

.labels
****

----
maintainer
----

A comma separated list of labels that are required for each dockerfile.
****


== Comments And Links
(visible only on this page)

endif::env-github,rspecator-view[]
