<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when a closure is used inside a GString expression in a Jenkins Pipeline script.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Jenkins Pipeline uses a Continuation Passing Style (CPS) transformation to allow scripts to be paused and resumed. This transformation affects how closures work in the Pipeline environment.</p>
</div>
<div class="paragraph">
<p>In regular Groovy, you can use closures inside GString expressions to create dynamic strings where the closure is evaluated each time the GString is converted to a String. However, in Jenkins Pipeline scripts, this pattern doesn&#8217;t work as expected because the closure inside the GString gets CPS-transformed.</p>
</div>
<div class="paragraph">
<p>When the Pipeline runtime tries to evaluate the GString, it encounters a CPS-transformed closure in a non-CPS context, which causes the CPS interpreter to fail. This results in incorrect behavior and confusing error messages in the Pipeline logs.</p>
</div>
<div class="paragraph">
<p>The typical error you&#8217;ll see in the logs looks like:</p>
</div>
<div class="quoteblock">
<blockquote>
<div class="paragraph">
<p>expected to call WorkflowScript.echo but wound up catching org.jenkinsci.plugins.workflow.cps.CpsClosure2.call</p>
</div>
</blockquote>
</div>
<div class="paragraph">
<p>This happens because the Pipeline runtime cannot properly handle the interaction between the non-CPS GString evaluation and the CPS-transformed closure.</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="paragraph">
<p>The code will not behave as expected and may produce runtime errors or unexpected results. The Pipeline may fail with confusing error messages, making debugging difficult. In some cases, the GString may not evaluate to the expected value, leading to incorrect string interpolation.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Replace the GString containing a closure with a closure that returns a GString using normal expressions. Call the closure when you need the string value.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-groovy" data-lang="groovy">def x = 1
def s = "x = ${-&gt; x}" // Noncompliant
x = 2
echo(s)</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-groovy" data-lang="groovy">def x = 1
def s = { -&gt; "x = ${x}" }
x = 2
echo(s())</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Jenkins Pipeline CPS Method Mismatches - <a href="https://www.jenkins.io/doc/book/pipeline/cps-method-mismatches/">Official Jenkins documentation explaining CPS transformation issues and solutions</a></p>
</li>
<li>
<p>Pipeline: Groovy Plugin - <a href="https://plugins.jenkins.io/workflow-cps/">Technical background on the Groovy CPS library used by Jenkins Pipeline</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>