== Why is this an issue?

The requirement for a final ``++else++`` clause is defensive programming. The clause should either take appropriate action, or contain a suitable comment as to why no action is taken.

=== Noncompliant code example

[source,ruby]
----
case param
  when 1
    do_something()
  when 2
    do_something_else()
end
----

=== Compliant solution

[source,ruby]
----
case param
  when 1
    do_something()
  when 2
    do_something_else()
  else
    handle_error('error_message')
end
----

include::../see.adoc[]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
