<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Similar to uncalled functions, un-instantiated class and function templates are a potential source of noise and they may be symptomatic of a more serious problem such as missing paths.</p>
</div>
<div class="paragraph">
<p>Note: Even though a given class template may be instantiated many times, it is possible that some of its member functions are never instantiated.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">template &lt; typename T &gt;
class Sample
{
public:
  void inst_mem ( ) { }
  void uninst_mem ( ) { } // Noncompliant, never instantiated
};

Sample&lt;int64_t&gt; s;
s.inst_mem ( ); // Call to s.inst_mem instantiates the member.
// s.uninst_mem is not called within the program and is not instantiated.</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_exceptions">Exceptions</h3>
<div class="paragraph">
<p>This rule does not apply to libraries.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p>MISRA C&#43;&#43;:2008, 14-7-1</p>
</li>
</ul>
</div>
<hr>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>"xxx" is never instantiated. It should be removed or referenced at least once.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_is_related_to_s1763">is related to: <a data-rspec-id="S1763" class="rspec-auto-link">S1763</a></h3>

</div>
<div class="sect2">
<h3 id="_relates_to_s901">relates to: <a data-rspec-id="S901" class="rspec-auto-link">S901</a></h3>

</div>
<div class="sect2">
<h3 id="_relates_to_s897">relates to: <a data-rspec-id="S897" class="rspec-auto-link">S897</a></h3>

</div>
</div>
</div>