<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>It is not uncommon, for instance when dealing with SQL requests, to have repeated calls to <code>StringBuilder.append()</code> to create a long (sometimes really long) String that will be then passed to the appropriate subsystem (e.g. jdbc.Statement()). This is very undesirable because it makes it more difficult to read, and maintain, the statement in the String due to overlapping syntaxes.</p>
</div>
<div class="paragraph">
<p>It is highly recommended to address such a case with an external text file loaded as a resource.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">sb = new StringBuilder()
    		.append("SELECT CASE ")
    		.append("WHEN year = 'FR' THEN 'FR'")
    		.append("WHEN year = 'SO' THEN 'SO'")
    		.append("WHEN year = 'JR' THEN 'JR'")
    		.append("WHEN year = 'SR' THEN 'SR'")
    		.append("ELSE 'No Year Data' END AS year_group,")
    		.append("COUNT(1) AS count")
    		.append("FROM benn.college_football_players")
    		.append("GROUP BY CASE WHEN year = 'FR' THEN 'FR'")
    		.append("WHEN year = 'SO' THEN 'SO'")
    		.append("WHEN year = 'JR' THEN 'JR'")
    		.append("WHEN year = 'SR' THEN 'SR'")
    		.append("ELSE 'No Year Data' END");</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">InputStream inputStream = this.getClass().getResourceAsStream("MySQLRequest.txt");</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Refactor this code to load this text as an external resource.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>The StringBuilder instance.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_14_nov_2016_150225_tibor_blenessy_wrote">on 14 Nov 2016, 15:02:25 Tibor Blenessy wrote:</h3>
<div class="paragraph">
<p>I think compliant example should show full way to the <code>String</code> , so it is really clear how to replace the code. Also I believe that this case will be quite rare, because usually there will be some embedded variables in the append calls, and then it&#8217;s not so easy to fix (probably it would be necessary to introduce some kind of template engine). Maybe we should just flag this and do not try to offer compliant solution?</p>
</div>
</div>
</div>
</div>