<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Object literal syntax, which initializes an object&#8217;s properties inside the object declaration is cleaner and clearer than the alternative: creating an empty object, and then giving it properties one by one.</p>
</div>
<div class="paragraph">
<p>An issue is raised when the following pattern is met:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>An empty object is created.</p>
</li>
<li>
<p>A consecutive single-line statement adds a property to the created object.</p>
</li>
</ul>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-javascript" data-lang="javascript">let person = {};  // Noncompliant
person.firstName = "John";
person.middleInitial = "Q";
person.lastName = "Public";</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-javascript" data-lang="javascript">let person = {
  firstName: "John",
  middleInitial: "Q",
  lastName: "Public",
}</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Declare one or more properties of this object inside of the object literal syntax instead of using separate statements.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_12_nov_2015_182041_linda_martin_wrote">on 12 Nov 2015, 18:20:41 Linda Martin wrote:</h3>
<div class="paragraph">
<p>\[~elena.vilchik] Same here, not sure of the value.</p>
</div>
</div>
</div>
</div>