<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when the functions <code>datetime.datetime.utcnow</code> or <code>datetime.datetime.utcfromtimestamp</code> are used.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Python&#8217;s <code>datetime</code> API provide several different ways to create <code>datetime</code> objects. One possibility is the to use
<code>datetime.datetime.utcnow</code> or <code>datetime.datetime.utcfromtimestamp</code> functions. The issue with these two functions is they are not time zone aware, even if their name would suggest otherwise.</p>
</div>
<div class="paragraph">
<p>Using these functions could cause issue as they may not behave as expected, for example:</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">from datetime import datetime
timestamp = 1571595618.0
date = datetime.utcfromtimestamp(timestamp)
date_timestamp = date.timestamp()

assert timestamp == date_timestamp</code></pre>
</div>
</div>
<div class="paragraph">
<p>This assertion will fail if the system locale is not set to UTC.
For this reason these 2 functions are deprecated in Python 3.12.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>To fix this issue, prefer the usage of a timezone-aware datetime.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">from datetime import datetime

datetime.utcnow() # Noncompliant

timestamp = 1571595618.0
datetime.utcfromtimestamp(timestamp) # Noncompliant</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">from datetime import datetime, timezone

datetime.now(timezone.utc) # Compliant

timestamp = 1571595618.0
datetime.fromtimestamp(timestamp, timezone.utc) # Compliant</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Python documentation - <a href="https://docs.python.org/3/library/datetime.html#datetime.datetime.utcnow">utcnow reference</a></p>
</li>
<li>
<p>Python documentation - <a href="https://docs.python.org/3/library/datetime.html#datetime.datetime.utcfromtimestamp">utcfromtimestamp reference</a></p>
</li>
</ul>
</div>
</div>
<div class="sect2">
<h3 id="_articles_blog_posts">Articles &amp; blog posts</h3>
<div class="ulist">
<ul>
<li>
<p>Paul Ganssle blog - <a href="https://blog.ganssle.io/articles/2019/11/utcnow.html">Stop using utcnow and utcfromtimestamp</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>