== Why is this an issue?

`std::format` and the related formatting functions provide two different options to pad numerical values up to a specific width:

* Custom character padding: it applies to many types and can align the text to the left, the center, or the right. +
  For example, `std::format("{:*>5}", num)` aligns `num` to the right (`>`) by inserting as many `*` characters as necessary to reach a width of 5.

* Numeric padding with `0`: it applies to most arithmetic types by adding `0` before the width specifier. +
  For example, `std::format("{:05}", num);` adds enough `0` before `num` to align it to the right to reach a width of 5.

Padding negative numeric values to the right results in surprising outputs:

[source,cpp]
----
// Noncompliant: the use of character padding results in "00-10 00-1."
std::format("{:0>5} {:0>#5}", -10, -1.0);

// Compliant: use numeric padding to produce "-0010 -001."
std::format("{:05} {:#05}", -10, -1.0);
----

== How to fix it

This rule raises an issue when a format specification uses `0` as a custom padding character to right-align a value instead of using the numeric padding.

=== Code examples

Use the numeric padding when right-aligning a numeric variable `num` with zeros.

It works with default sign and representation:

==== Noncompliant code example

[source,cpp,diff-id=1,diff-type=noncompliant]
----
std::format("{:0>5}", num)
----

==== Compliant solution

A compliant solution is to remove the `>`:

[source,cpp,diff-id=1,diff-type=compliant]
----
std::format("{:05}", num)
----

Before, the `0` was used as a custom character padding. +
Now, it means numeric padding is used.

Here are some examples of outputs for these two code snippets:

// Note: Asciidoc collapses multiple regular whitespaces in literals.
// To avoid this issue in the tables below, we use EN SPACE Unicode: U+2002.

[cols="1,1,1"]
|===
| Value of `num` | Noncompliant output | Compliant output

| `10`
| `"00010"`
| `"00010"`

| `-10`
| `"00-10"`
| `"-0010"`

| `NAN`
| `"00nan"`
| `"  nan"`

| `-INFINITY`
| `"0-inf"`
| `" -inf"`

|===

Numeric padding also works with a sign specifier (such as `{plus}`) and the alternative form (`#`):

==== Noncompliant code example

[source,cpp,diff-id=2,diff-type=noncompliant]
----
std::format("{:0>+#5", num)
----

==== Compliant solution

A compliant solution is to remove the `>` and move the `0` right before the width specifier (`5`):

[source,cpp,diff-id=2,diff-type=compliant]
----
std::format("{:+#05}", num)
----
[cols="1,1,1"]
|===
| Value of `num` | Noncompliant output | Compliant output

| `10`
| `"0+10."`
| `"+010."`

| `-10`
| `"0-10."`
| `"-010."`

| `NAN`
| `"0+nan"`
| `" +nan"`

| `-INFINITY`
| `"0-inf"`
| `" -inf"`

|===

Numeric passing also works with other specifiers to change the type representation, set its precision, or apply locale-specific formatting.

== Resources

=== Related rules

S6483 detects incompatible use of numeric and character padding specifiers.
