== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,csharp]
----
if(someCondition) DoSomething();
----

=== Compliant solution

[source,csharp]
----
if(someCondition) 
{
  DoSomething();
}
----

=== Exceptions

Anonymous functions containing a single statement are ignored. Block statements are not considered either.


[source,csharp]
----
Func<object, bool> item1 = o => { return true; }; // Compliant
Func<object, bool> item1 = o => { var r = false; return r; }; // Noncompliant
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../highlighting.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
