include::../description.adoc[]

== Noncompliant Code Example

----
if(someCondition) DoSomething();
----

== Compliant Solution

----
if(someCondition) 
{
  DoSomething();
}
----

== Exceptions

Anonymous functions containing a single statement are ignored. Block statements are not considered either.

----
Func<object, bool> item1 = o => { return true; }; // Compliant
Func<object, bool> item1 = o => { var r = false; return r; }; // Noncompliant
----
