<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when there are multiple separate <code>import</code> statements that could be grouped together in a factored import block.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Go supports two ways to write import statements: multiple separate statements or a single factored statement that groups imports in parentheses.</p>
</div>
<div class="paragraph">
<p>While both approaches are syntactically valid, the Go community strongly recommends using factored import statements. This convention is documented in the official Go Tour and is widely adopted across Go codebases.</p>
</div>
<div class="paragraph">
<p>Factored imports offer several advantages:</p>
</div>
<div class="ulist">
<ul>
<li>
<p><strong>Better organization</strong>: All imports are grouped in one place, making dependencies easier to see at a glance</p>
</li>
<li>
<p><strong>Cleaner code</strong>: Reduces visual clutter by eliminating repeated <code>import</code> keywords</p>
</li>
<li>
<p><strong>Consistency</strong>: Follows the established Go style guidelines that most developers expect</p>
</li>
<li>
<p><strong>Tool compatibility</strong>: Many Go formatting tools like <code>gofmt</code> automatically convert separate imports to factored form</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>Using separate import statements makes code look less polished and can signal unfamiliarity with Go conventions to other developers.</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="paragraph">
<p>This is a style issue that affects code readability and consistency. While it doesn&#8217;t impact functionality, following Go conventions makes code more maintainable and professional.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Group multiple separate import statements into a single factored import block using parentheses.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-go" data-lang="go">import "fmt"
import "math"
import "strings" // Noncompliant</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-go" data-lang="go">import (
    "fmt"
    "math"
    "strings"
)</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>A Tour of Go - Imports - <a href="https://go.dev/tour/basics/2">Official Go documentation explaining factored import statements</a></p>
</li>
<li>
<p>Effective Go - Formatting - <a href="https://go.dev/doc/effective_go#formatting">Go&#8217;s official style guide covering code formatting conventions</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>