include::../description.adoc[]

== Noncompliant Code Example

With default provided regular expression: ``^[a-z][a-zA-Z0-9]*$``:

----
function DoSomething(){...} 
----

== Compliant Solution

----
function doSomething(){...} 
----

== Exceptions

Methods with an ``@inheritdoc`` annotation, as well as magic methods (``__construct()``, ``__destruct()``, ``__call()``, ``__callStatic()``, ``__get()``, ``__set()``, ``__isset()``, ``__unset()``, ``__sleep()``, ``__wakeup()``, ``__toString()``, ``__invoke()``, ``__set\_state()``, ``__clone()``, ``__debugInfo()``) are ignored.

----
function __construct(){...}
function __destruct(){...}

/**
 * {@inheritdoc}
 */
function myFunc(){...}
----
