<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Because parameter names could be changed during refactoring, they should not be spelled out literally in strings. Instead, use <code>NameOf()</code>, and the string that&#8217;s output will always be correct.</p>
</div>
<div class="paragraph">
<p>This rule raises an issue when any string in the <code>Throw</code> statement is an exact match for the name of one of the method parameters.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-vbnet" data-lang="vbnet">Public Sub DoSomething(param As Integer, secondParam As String)
    If (param &lt; 0)
        Throw New Exception("param") ' Noncompliant
    End If
    If secondParam is Nothing
      Throw New Exception("secondParam should be valid") ' Noncompliant
    End If
End Sub</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-vbnet" data-lang="vbnet">Public Sub DoSomething(param As Integer, secondParam As String)
    If (param &lt; 0)
        Throw New Exception(NameOf(param))
    End If
    If secondParam is Nothing
      Throw New Exception($"{NameOf(secondParam)} should be valid")
    End If
End Sub</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_exceptions">Exceptions</h3>
<div class="ulist">
<ul>
<li>
<p>The rule doesn&#8217;t raise any issue when using VB.NET &lt; 14.0.</p>
</li>
<li>
<p>When the parameter name is contained in a sentence inside the <code>Throw</code> statement string, the rule will raise an issue only if the parameter name is at least 5 characters long. This is to avoid false positives.</p>
</li>
</ul>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Replace the string 'xxx' with 'nameof(xxx)'.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_8_may_2015_153148_dinesh_bolkensteyn_wrote">on 8 May 2015, 15:31:48 Dinesh Bolkensteyn wrote:</h3>
<div class="paragraph">
<p>LGTM</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_11_sep_2019_112427_andrei_epure_wrote">on 11 Sep 2019, 11:24:27 Andrei Epure wrote:</h3>
<div class="paragraph">
<p>\[~nicolas.harraudeau] - do you see a good reason not to raise an issue when the string is not an exact match? One could include the name of the parameter inside the message like</p>
</div>
<div class="ulist">
<ul>
<li>
<p><code>"someParameter should have value X"</code></p>
</li>
</ul>
</div>
<div class="paragraph">
<p>or using interpolated strings</p>
</div>
<div class="ulist">
<ul>
<li>
<p><code>Â  $"someParameter should have value X but has value {someParameter}"</code></p>
</li>
</ul>
</div>
<div class="paragraph">
<p>As it is, I don&#8217;t find the rule as useful as it could be</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_11_sep_2019_114207_nicolas_harraudeau_wrote">on 11 Sep 2019, 11:42:07 Nicolas Harraudeau wrote:</h3>
<div class="paragraph">
<p>\[~andrei.epure] Both cases you mentioned make sense.</p>
</div>
</div>
</div>
</div>