<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>TypeScript allows all sorts of expressions to initialize enum members. However, as enums create their own scope, using identifiers can lead to unexpected results. The recommendation is thus to use only literal values when defining enum members.</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="paragraph">
<p>An enum member will shadow any variable defined with the same name in an enclosing scope. In the code example below, the <code>day</code> member value should derive from the <code>hour</code> const variable. But it is computed using the enum <code>hour</code> member instead, leading to the result 24 instead of the expected 1440.</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-javascript" data-lang="javascript">const hour = 3600; // hour in seconds

enum DurationInMinutes {
    hour = 60, // hour in minutes
    day = 24 * hour / 60 // Expecting hour in seconds but got hour in minutes
}</code></pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>The recommended approach for enums is to initialize members with literals. Not only are literals easier to read, but they also avoid unexpected results due to the enum scope.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-javascript" data-lang="javascript">const hello = 'Hello';
enum Foo {
  STRING = hello, // Variable
  OBJECT = { hello }.hello.length, // Object
  TEMPLATE = `${hello}, World`, // Template literal
  SET = new Set([hello, 'world']).size, // Constructor
  NUMBER = hello.length + 1, // Expression
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-javascript" data-lang="javascript">enum Foo {
  STRING = 'Hello',
  NUMBER = 0,
}</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p><a href="https://typescript-eslint.io/">typescript-eslint</a> - Rule <a href="https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/docs/rules/prefer-literal-enum-member.mdx">prefer-literal-enum-member</a></p>
</li>
<li>
<p>TypeScript Documentation - <a href="https://www.typescriptlang.org/docs/handbook/enums.html#handbook-content">Enums</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>