<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>In Dart there&#8217;s an ability to use template literals instead of concatenation. Since their use is clearer and more concise, they are preferred.</p>
</div>
<div class="sect2">
<h3 id="_exceptions">Exceptions</h3>
<div class="paragraph">
<p>Concatenation of string literals, without any variables, is allowed, both using <code>+</code> and using adjacent strings. Those are typically used for multiline strings.</p>
</div>
<div class="paragraph">
<p>Raw string literals are also an exception to this rule, since they don&#8217;t support interpolation.</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-dart" data-lang="dart">var s1 =
  'hello\n' +
  'world'; // OK
var s2 =
  'hello\n'
  'world'; // OK
var s3 = r'hello\n' + s1; // OK</code></pre>
</div>
</div>
<div class="paragraph">
<p>The multiline strings like <code>s1</code> and <code>s2</code> above can also be written as follows:</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-dart" data-lang="dart">var s1Alternative = '''
hello
world'''; // OK</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-dart" data-lang="dart">void sayHello(name) {
  print('hello ' + name + '!');  // Noncompliant
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-dart" data-lang="dart">void sayHello(name) {
  print('hello $name!');
}</code></pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Dart Docs - <a href="https://dart.dev/tools/linter-rules/prefer_interpolation_to_compose_strings">Dart Linter rule - prefer_interpolation_to_compose_strings</a></p>
</li>
<li>
<p>Dart Docs - <a href="https://dart.dev/language/built-in-types#strings">Dart language - strings</a></p>
</li>
</ul>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Use interpolation to compose strings and values.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>The entire string concatenation expression, including the operands.</p>
</div>
<div class="paragraph">
<p>If a non-parenthesized sequence of concatenations is detected, a single issue is reported, on the first part of it: e.g. in <code>'a' + s1 + 'a'</code> and <code>s1 + 'a' + s1</code>, only <code>'a' + s1</code> and <code>s1 + 'a'</code> is reported, respectively.</p>
</div>
<div class="paragraph">
<p>An exception is made for single string variables or literals in parentheses, where the outer concatenation is reported: e.g. in <code>'a' + (s1)</code>, the entire expression is reported.</p>
</div>
<div class="paragraph">
<p>In parenthesized expressions, only innermost concatenations are reported: e.g. in <code>'a' + (s1 + 'a')</code>, only <code>s1 + 'a'</code> is reported.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
</div>
</div>