== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,rpg]
----
     C   99W1SCNO        IFEQ      1
     C                   EXSR      SR201
     C                   ELSE
     C* Empty on purpose or missing a piece of code?
     C                   END
----

[source,rpg]
----
      /free
        for i = 1 to n;
        endfor;    
      /end-free
----

include::../exceptions.adoc[]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
