include::../description.adoc[]

== Noncompliant Code Example

----
     C   99W1SCNO        IFEQ      1
     C                   EXSR      SR201
     C                   ELSE
     C* Empty on purpose or missing a piece of code?
     C                   END
----

----
      /free
        for i = 1 to n;
        endfor;    
      /end-free
----

include::../exceptions.adoc[]

ifdef::rspecator-view[]
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::rspecator-view[]
