== Why is this an issue?

An array declared with missing ("sparse") elements is likely to be an error: an extra comma was inserted or perhaps the developer meant to insert the missing value and forgot.


=== Noncompliant code example

[source,javascript]
----
let a = [1, , 3, 6, 9];  // Noncompliant
----


=== Compliant solution

[source,javascript]
----
let a = [1, 3, 6, 9];
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

include::highlighting.adoc[]

'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
