== Why is this an issue?

There is no need to use the ``++final++`` modifier inside a ``++final++`` class. Everything in it is ``++final++`` by default.


=== Noncompliant code example

[source,text]
----
final class MyClass {

  public final String getName() {  // Noncompliant
    return name;
  }
}
----


=== Compliant solution

[source,text]
----
final class MyClass {

  public String getName() {  // Compliant
    return name;
  }
}
----

