== Why is this an issue?

`GoTo` is an unstructured control flow statement. It makes code less readable and maintainable. Structured control flow statements such as `If`, `For`, `While`,  or `Exit` should be used instead.

=== Exceptions

`On Error GoTo` statements are ignored as correct error handling.

== How to fix it

Replace `GoTo` statements with structured control flow statements.

=== Code examples

==== Noncompliant code example

[source,vb6,diff-id=1,diff-type=noncompliant]
----
Sub gotoStatementDemo()
    Dim number As Integer = 1
    Dim sampleString As String
    ' Evaluate number and branch to appropriate label.
    If number = 1 Then GoTo Line1 Else GoTo Line2
Line1:
    sampleString = "Number equals 1"
    GoTo LastLine
Line2:
    ' The following statement never gets executed because number = 1.
    sampleString = "Number equals 2"
LastLine:
    ' Write "Number equals 1" in the Debug window.
    Debug.WriteLine(sampleString)
End Sub
----

==== Compliant solution

[source,vb6,diff-id=1,diff-type=compliant]
----
Sub gotoStatementDemo()
    Dim number As Integer = 1
    Dim sampleString As String
    ' Evaluate number and branch to appropriate label.
    If number = 1 Then
        sampleString = "Number equals 1"
    Else
        sampleString = "Number equals 2"
    End If
    Debug.WriteLine(sampleString)
End Sub
----

== Resources
=== Documentation

* Microsoft Learn - https://learn.microsoft.com/en-us/office/vba/language/reference/user-interface-help/goto-statement[GoTo statement]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
