== Why is this an issue?

Shared naming conventions allow teams to collaborate efficiently. This rule checks that all coroutine names match a provided regular expression.


=== Noncompliant code example

With default provided regular expression: ``++^[a-z][a-zA-Z0-9]*$++``:

[source,cpp]
----
generator<int> Iota(int n = 0) {
  while(true)
    co_yield n++;
}
----

=== Compliant solution

[source,cpp]
----
generator<int> iota(int n = 0) {
  while(true)
    co_yield n++;
}
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Rename coroutine "XXXX" to match the regular expression ${format}.


endif::env-github,rspecator-view[]
