Shared naming conventions allow teams to collaborate efficiently. This rule checks that all coroutine names match a provided regular expression.


== Noncompliant Code Example

With default provided regular expression: ``++^[a-z][a-zA-Z0-9]*$++``:

----
generator<int> Iota(int n = 0) {
  while(true)
    co_yield n++;
}
----

== Compliant Solution

----
generator<int> iota(int n = 0) {
  while(true)
    co_yield n++;
}
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

endif::env-github,rspecator-view[]
