== Why is this an issue?

This rule checks that all Razor component parameters are properly declared before they are used in the code. Using a parameter that is not declared can lead to undefined behavior and runtime exceptions in your program.

[source,text]
----
crit: Microsoft.AspNetCore.Components.WebAssembly.Rendering.WebAssemblyRenderer[100]
      Unhandled exception rendering component: Object of type 'Web.Pages.Samples' does not have a property matching the name 'Other'.
----

=== Exceptions

The rule will not raise when the component has a parameter with https://learn.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.components.parameterattribute.captureunmatchedvalues[CaptureUnmatchedValues] property set to true.

== How to fix it

To prevent runtime exceptions and unexpected behavior, always declare parameters prior to their use.

=== Code examples

==== Noncompliant code example

[source,csharp,diff-id=1,diff-type=noncompliant]
----
@page "/print"
@code {
    [Parameter]
    [SupplyParameterFromQuery]
    public bool BoolParam { get; set; }
}

@* Usage: *@

<Print Other="1">     @* Noncompliant *@
</Print>
----

==== Compliant solution

[source,csharp,diff-id=1,diff-type=compliant]
----
@page "/print"
@code {
    [Parameter]
    [SupplyParameterFromQuery]
    public bool Param { get; set; }
}

@* Usage: *@

<Print Param="true">       @* Compliant *@
</Print>
----

[source,csharp]
----
@page "/print"
@code {
    [Parameter(CaptureUnmatchedValues = true)]
    [SupplyParameterFromQuery]
    public Dictionary<string, object> All { get; set; }
}

@* Usage: *@

<Print Other="true">       @* Compliant *@
</Print>
----

== Resources

=== Documentation

* Microsoft Learn - https://learn.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.components.parameterattribute.captureunmatchedvalues[CaptureUnmatchedValues property]
