== Why is this an issue?

This rule checks that all Razor https://learn.microsoft.com/en-us/aspnet/core/blazor/components/#component-parameters[component parameters] are properly declared before they are used in the code. Using a parameter that is not declared can lead to undefined behavior and runtime exceptions in your program.

[source,text]
----
crit: Microsoft.AspNetCore.Components.WebAssembly.Rendering.WebAssemblyRenderer[100]
      Unhandled exception rendering component: Object of type 'Web.Pages.Samples' does not have a property matching the name 'Other'.
----

=== Exceptions

The rule will not raise when the component has a parameter with https://learn.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.components.parameterattribute.captureunmatchedvalues[CaptureUnmatchedValues] property set to true.

[source,csharp]
----
@code {
    [Parameter(CaptureUnmatchedValues = true)]
    public Dictionary<string, object> All { get; set; }
}

@* Usage: *@

<Print Other="true"/>  @* Compliant *@
----

== How to fix it

To prevent runtime exceptions and unexpected behavior, always declare parameters prior to their use.

=== Code examples

==== Noncompliant code example

[source,csharp,diff-id=1,diff-type=noncompliant]
----
@code {
    [Parameter]
    public bool Param { get; set; }
}

@* Usage: *@

<Print Other="1"/>     @* Noncompliant *@
----

==== Compliant solution

[source,csharp,diff-id=1,diff-type=compliant]
----
@code {
    [Parameter]
    public bool Param { get; set; }
}

@* Usage: *@

<Print Param="true"/>  @* Compliant *@
----

== Resources

=== Documentation

* Microsoft Learn - https://learn.microsoft.com/en-us/aspnet/core/blazor/components/#component-parameters[component parameters]
* Microsoft Learn - https://learn.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.components.parameterattribute.captureunmatchedvalues[CaptureUnmatchedValues property]
