<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This is an issue when using <code>containsWithinBounds()</code> instead of <code>contains()</code> for membership testing.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Groovy <code>Range</code> objects have subtle but important differences in their methods that can lead to unexpected behavior if misused.</p>
</div>
<div class="paragraph">
<p>The <code>containsWithinBounds()</code> method only checks if a value falls between the range boundaries, while <code>contains()</code> checks if the value is actually part of the discrete items in the range. For example, with the range <code>1.5..3</code>, the value <code>2</code> falls within the bounds but is not actually in the range, since the range contains <code>[1.5, 2.5]</code>, incrementing by one from <code>1.5</code>. Using <code>containsWithinBounds(2)</code> returns <code>true</code>, but <code>contains(2)</code> correctly returns <code>false</code>.</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="paragraph">
<p>Using the wrong <code>Range</code> method can cause logic errors in conditional statements, leading to incorrect program behavior.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Use <code>contains()</code> instead of <code>containsWithinBounds()</code> when you need to check if a value is actually part of the range&#8217;s discrete items.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-groovy" data-lang="groovy">def range = 1.5..3
if (range.containsWithinBounds(2)) { // Noncompliant
    println "Found 2 in range"
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-groovy" data-lang="groovy">def range = 1.5..3
if (range.contains(2)) {
    println "Found 2 in range"
}</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Groovy Documentation - <a href="https://docs.groovy-lang.org/latest/html/api/groovy/lang/Range.html">Range</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>