<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This is an issue when a controller defines custom actions (beyond the standard RESTful actions) but these actions are not explicitly defined in the routes file.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Rails follows a specific route matching order when processing requests. When you use <code>resources :posts</code> in your routes file, Rails automatically creates standard RESTful routes including <code>GET /posts/:id</code> for the show action.</p>
</div>
<div class="paragraph">
<p>If you add a custom action like <code>start</code> to your controller without defining a corresponding route, accessing <code>/posts/start</code> will match the show route pattern. Rails will treat "start" as an ID parameter and pass it to the show action, typically resulting in a <code>RecordNotFound</code> error when trying to find a record with ID "start".</p>
</div>
<div class="paragraph">
<p>This happens because Rails matches routes in the order they are defined, and the generic <code>:id</code> parameter in the show route will match any string, including your custom action name.</p>
</div>
<div class="paragraph">
<p>This routing conflict can cause:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>Runtime errors when users try to access the custom action</p>
</li>
<li>
<p>Confusion for developers who expect the custom action to work</p>
</li>
<li>
<p>Unpredictable behavior if the custom action name happens to match an existing record ID</p>
</li>
</ul>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="paragraph">
<p>Users will encounter <code>RecordNotFound</code> errors when trying to access custom controller actions. This can break application functionality and create a poor user experience. The issue may not be immediately apparent during development if the custom action name doesn&#8217;t conflict with existing record IDs.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it_in_rails">How to fix it in Rails</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Define the custom action explicitly in your routes file using collection routes for actions that don&#8217;t operate on a specific resource instance.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-ruby" data-lang="ruby"># routes.rb
Rails.application.routes.draw do
  resources :posts  # Only standard RESTful routes defined
end

# posts_controller.rb
class PostsController &lt; ApplicationController
  def start  # Noncompliant: Custom action without corresponding route
    # This will cause routing conflicts
  end

  def show
    @post = Post.find(params[:id])
  end
end</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-ruby" data-lang="ruby"># routes.rb
Rails.application.routes.draw do
  resources :posts do
    collection do
      get 'start'  # Custom action properly routed
    end
  end
end

# posts_controller.rb
class PostsController &lt; ApplicationController
  def start
    # Custom action implementation
  end

  def show
    @post = Post.find(params[:id])
  end
end</code></pre>
</div>
</div>
<div class="paragraph">
<p>For custom actions that operate on a specific resource instance, use member routes instead of collection routes.</p>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_code_examples_2">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example_2">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-ruby" data-lang="ruby"># routes.rb
Rails.application.routes.draw do
  resources :posts
end

# posts_controller.rb
class PostsController &lt; ApplicationController
  def archive  # Noncompliant: Custom action without corresponding route
    @post = Post.find(params[:id])
    @post.update(archived: true)
  end
end</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution_2">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-ruby" data-lang="ruby"># routes.rb
Rails.application.routes.draw do
  resources :posts do
    member do
      patch 'archive'  # Custom action for specific resource
    end
  end
end

# posts_controller.rb
class PostsController &lt; ApplicationController
  def archive
    @post = Post.find(params[:id])
    @post.update(archived: true)
  end
end</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Rails Routing from the Outside In - <a href="https://guides.rubyonrails.org/routing.html#adding-more-restful-actions">Official Rails guide on adding custom actions to RESTful routes</a></p>
</li>
<li>
<p>Rails Resource Routing - <a href="https://guides.rubyonrails.org/routing.html#resource-routing-the-rails-default">Complete guide to Rails resource routing and RESTful conventions</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>