<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Redirecting standard in and standard out to/from files is bad practice because it contravenes the standard expectation that <code>cin</code> and <code>cout</code> do not relate to files. Additionally, it is less efficient than using file streams such as <code>ifstream</code> and <code>ofstream</code>. For both of these reasons, this practice should be avoided.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">  ifstream in("in.txt");
  cin.rdbuf(in.rdbuf()); // Noncompliant; redirects cin to in.txt

  ofstream out("out.txt");
  cout.rdbuf(out.rdbuf()); //Noncompliant; redirects cout to out.txt

  std::string line;
  while(getline(cin, line))
  {
      cout &lt;&lt; line &lt;&lt; endl;
  }</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">  ifstream in("in.txt");

  ofstream out("out.txt");
  std::string line;
  while (inf &amp;&amp; in &gt;&gt; line)
  {
    out &lt;&lt; line &lt;&lt; endl;
  }</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Use "outf|inf" instead</p>
</div>
</div>
</div>
</div>