<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Assignment of <code>false</code> to a pointer type is implicitly converted to a <code>NULL</code> assignment.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">void foo(bool *p) {
  p = false;  // Noncompliant
}</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove this "false" assignment to a pointer type.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_27_jul_2015_125159_ann_campbell_wrote">on 27 Jul 2015, 12:51:59 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~massimo.paladin] I&#8217;ve upgraded this to Critical/bug. See what you think.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_28_aug_2015_122532_massimo_paladin_wrote">on 28 Aug 2015, 12:25:32 Massimo PALADIN wrote:</h3>
<div class="paragraph">
<p>\[~evgeny.mandrikov] could you please verify?</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_28_aug_2015_135621_evgeny_mandrikov_wrote">on 28 Aug 2015, 13:56:21 Evgeny Mandrikov wrote:</h3>
<div class="paragraph">
<p>\[~massimo.paladin] looks good</p>
</div>
</div>
</div>
</div>