== Why is this an issue?

When ``++@Autowired++`` is used, dependencies need to be resolved when the class is instantiated, which may cause early initialization of beans or lead the context to look in places it shouldn't to find the bean. To avoid this tricky issue and optimize the way the context loads, dependencies should be requested as late as possible. That means using parameter injection instead of field injection for dependencies that are only used in a single ``++@Bean++`` method.


=== Noncompliant code example

[source,java]
----
@Configuration
public class ​FooConfiguration {

  @Autowired private ​DataSource dataSource​;  // Noncompliant

  @Bean
  public ​MyService myService() {
    return new ​MyService(this​.dataSource​);
  }
}
----


=== Compliant solution

[source,java]
----
@Configuration
public class ​FooConfiguration {

 @Bean
  public ​MyService myService(DataSource dataSource) {
    return new ​MyService(dataSource);
  }
}
----


=== Exceptions

Fields used in methods that are called directly by other methods in the application (as opposed to being invoked automatically by the Spring framework) are ignored by this rule so that direct callers don't have to provide the dependencies themselves.


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Inject this field value directly into "xxx", the only method that uses it.


'''
== Comments And Links
(visible only on this page)

=== on 13 Jun 2018, 13:28:53 Andrei Epure wrote:
Are ``++@Autowired++`` setters and constructors on a ``++@Configuration++`` object out of scope of this rule?

endif::env-github,rspecator-view[]
