<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Synchronization is a mechanism used when multithreading in Java to ensure that only one thread executes a given block of code at a time.
This is done to avoid bugs that can occur when multiple threads share a given state and try to manipulate simultaneously.</p>
</div>
<div class="paragraph">
<p>Object serialization is not thread-safe by default.
In a multithreaded environment, one option is to mark <code>writeObject</code> with <code>synchronized</code> to improve thread safety.
It is highly suspicious, however, if <code>writeObject</code> is the only <code>synchronized</code> method in a class.
It may indicate that serialization is not required, as multithreading is not used.
Alternatively, it could also suggest that other methods in the same class have been forgotten to be made thread-safe.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Consider whether this class is used in a multithreaded context.
If it is, ask yourself whether other methods in this class should also be marked as <code>synchronized</code>.
Otherwise, remove the <code>synchronized</code> modifier from this method.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">public class RubberBall implements Serializable {

  private Color color;
  private int diameter;

  public RubberBall(Color color, int diameter) {
    // ...
  }

  public void bounce(float angle, float velocity) {
    // ...
  }

  private synchronized void writeObject(ObjectOutputStream stream) throws IOException { // Noncompliant, "writeObject" is the only synchronized method in this class
    // ...
  }
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">public class RubberBall implements Serializable {

  private Color color;
  private int diameter;

  public RubberBall(Color color, int diameter) {
    // ...
  }

  public void bounce(float angle, float velocity) {
    // ...
  }

  private void writeObject(ObjectOutputStream stream) throws IOException { // Compliant, no methods in this class are synchronized
    // ...
  }
}</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p><a href="https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/io/Serializable.html">Java SE 17 &amp; JDK 17 - Serializable Javadoc</a></p>
</li>
<li>
<p><a href="https://docs.oracle.com/javase/tutorial/essential/concurrency/syncmeth.html">The Java&#8482; Tutorials - Synchronized Methods</a></p>
</li>
</ul>
</div>
<hr>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove this "synchronized" keyword.</p>
</div>
</div>
</div>
</div>