include::../description.adoc[]

== Noncompliant Code Example

https://github.com/libxmljs/libxmljs[libxmljs] module:

----
const libxmljs = require("libxmljs");
var fs = require('fs');

var xml = fs.readFileSync('xxe.xml', 'utf8');

var xmlDoc = libxmljs.parseXmlString(xml, { noblanks: true, noent: true, nocdata: true }); // Noncompliant: noent set to true
----

== Compliant Solution

https://github.com/libxmljs/libxmljs[libxmljs] module:

----
const libxmljs = require("libxmljs");
var fs = require('fs');

var xml = fs.readFileSync('xxe.xml', 'utf8');

var xmlDoc = libxmljs.parseXmlString(xml); // Compliant: noent set to false by default
----

include::../see.adoc[]
