When multiple regions are defined in a Javadoc code snippet, they should not have the same name. Doing so could lead to the wrong region being targeted.


== Noncompliant Code Example
[source,java]
----
{@snippet :
void foo() {
    // @start region="example" // Noncompliant
    System.out.println("This is region 1");
    // @end

    // @start region="example"
    System.out.println("This is region 2");
    // @end
}
}
----



== Compliant Solution
[source,java]
----
{@snippet :
void foo() {
    // @start region="example_1"
    System.out.println("This is region 1");
    // @end

    // @start region="example_2"
    System.out.println("This is region 2");
    // @end
}
}
----



== See
Oracleâ€™s Programmer's Guide to Snippets: https://docs.oracle.com/en/java/javase/18/code-snippet/index.html[https://docs.oracle.com/en/java/javase/18/code-snippet/index.html]
