== Why is this an issue?

Favicons are shown for example in the browser's address bar, bookmark list, or tabs.

They enable users to quickly identify and recognize websites.


=== Noncompliant code example

[source,html]
----
<head>                                                                  <!-- Noncompliant -->
  <title>...<title>
</head>
----


=== Compliant solution

[source,html]
----
<head>
  <title>...<title>
  <link rel="shortcut icon" href="http://example.com/myicon.ico" />
</head>
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Add a 'favicon' declaration in this 'header' tag.


'''
== Comments And Links
(visible only on this page)

=== on 8 Jul 2013, 18:19:22 Freddy Mallet wrote:
Is implemented by \http://jira.codehaus.org/browse/SONARPLUGINS-3000

endif::env-github,rspecator-view[]
