== How to fix it in ASP.NET

=== Code examples

include::../../common/fix/code-rationale.adoc[]

==== Noncompliant code example

[source,vbnet,diff-id=1,diff-type=noncompliant]
----
Imports System.Web
Imports System.Web.Mvc

Public Class ExampleController
    Inherits Controller

    <HttpGet>
    Public Async Function ImageFetch(Location As String) As Task(Of IActionResult)
        Using Stream As Stream = Await Client.GetStreamAsync(Location) ' Noncompliant
            Dim ExampleImage As ExampleImage = Await JsonSerializer.DeserializeAsync(Of ExampleImage)(Stream)

            Return Ok(If(ExampleImage, New ExampleImage()))
        End Using
    End Function
End Class
----

==== Compliant solution

[source,vbnet,diff-id=1,diff-type=compliant]
----
Imports System.Web
Imports System.Web.Mvc

Public Class ExampleController
    Inherits Controller
    Private ReadOnly AllowedSchemes() As String = {"https"}
    Private ReadOnly AllowedDomains() As String = {"trusted1.example.com", "trusted2.example.com"}

    <HttpGet>
    Public Async Function ImageFetch(Location As String) As Task(Of IActionResult)
        Dim Uri As New Uri(Location)

        If Not AllowedDomains.Contains(Uri.Host) AndAlso Not AllowedSchemes.Contains(Uri.Scheme) Then
            Return BadRequest()
        End If

        Using Stream As Stream = Await Client.GetStreamAsync(Location)
            Dim ExampleImage As ExampleImage = Await JsonSerializer.DeserializeAsync(Of ExampleImage)(Stream)

            Return Ok(If(ExampleImage, New ExampleImage()))
        End Using
    End Function
End Class
----

=== How does this work?

include::../../common/fix/pre-approved-list.adoc[]

include::../../common/fix/blacklist.adoc[]

=== Pitfalls

include::../../common/pitfalls/starts-with.adoc[]

include::../../common/pitfalls/blacklist-toctou.adoc[]
