== Why is this an issue?

For better readability, do not put multiple property declarations in the same statement.


=== Noncompliant code example

[source,php]
----
<?php
class Foo
{
   private $bar = 1, $bar2 = 2;
}
----


=== Compliant solution

[source,php]
----
<?php
class Foo
{
   private $bar1 = 1; 
   private $bar2 = 2;
}
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
