== Why is this an issue?

Declaring multiple variable on one line is difficult to read and potentially misleading since the ``++As++`` typing only applies to the last variable on the line. 

=== Noncompliant code example

[source,vb6]
----
Public AAA, BBB, CCC As String ' AAA and BBB are variants
----

=== Compliant solution

[source,vb6]
----
Public AAA as String
Public BBB as String
Public CCC as String
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
