<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>The <code>CONST</code> keyword on a subprocedure&#8217;s parameter indicates that the parameter value will not be changed by the subprocedure. This is not just a nice way to communicate with the programmers who will call the procedure. It also offers performance benefits, because it allows the compiler to produce more optimized code. Further, using <code>CONST</code> means that a field of a similar data type will automatically be converted to the correct type and size for the parameter.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-rpg" data-lang="rpg">     D X               S             15A   INZ('ABC')

     P SubProc1        B
     D SubProc1        PI
     D  Parm1                        15A    // Noncompliant; read-only. Should be CONST
     D  Parm2                        15A
      /Free
       X = Parm1;
       Parm2 = X;
       return;
      /End-free
     P SubProc1        E</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-rpg" data-lang="rpg">     D X               S             15A   INZ('ABC')

     P SubProc1        B
     D SubProc1        PI
     D  Parm1                        15A   CONST
     D  Parm2                        15A
      /Free
       X = Parm1;
       Parm2 = X;
       return;
      /End-free
     P SubProc1        E</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Add the "CONST" keyword to the declaration of "XXX"</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_is_duplicated_by_s1252">is duplicated by: <a data-rspec-id="S1252" class="rspec-auto-link">S1252</a></h3>

</div>
<div class="sect2">
<h3 id="_relates_to_s995">relates to: <a data-rspec-id="S995" class="rspec-auto-link">S995</a></h3>

</div>
</div>
</div>