== Why is this an issue?

Sharing some naming conventions is a key point to make it possible for a team to efficiently collaborate. This rule allows to check that all protocol names match a provided regular expression.

=== Noncompliant code example

With the default regular expression ``++^[A-Z][a-zA-Z0-9]*$++``:

[source,swift]
----
public protocol myProtocol {...} // Noncompliant
----

=== Compliant solution

[source,swift]
----
public protocol MyProtocol {...}
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Rename protocol ${name} to match the regular expression ${format}


=== Parameters

.format
****
_String_

----
^[A-Z][a-zA-Z0-9]*$
----

Regular expression used to check the protocol names against.
****


'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
