Sharing some naming conventions is a key point to make it possible for a team to efficiently collaborate. This rule allows to check that all protocol names match a provided regular expression.

== Noncompliant Code Example

With the default regular expression ``++^[A-Z][a-zA-Z0-9]*$++``:

----
public protocol myProtocol {...} // Noncompliant
----

== Compliant Solution

----
public protocol MyProtocol {...}
----
