include::../description.adoc[]

== Noncompliant Code Example

----
if(condition) 
{
  doSomething();} 
----

== Compliant Solution

----
if(condition) 
{
  doSomething();
} 
----

include::../exceptions.adoc[]

ifdef::env-github,rspecator-view[]
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
