== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,csharp]
----
if(condition) 
{
  doSomething();} 
----

=== Compliant solution

[source,csharp]
----
if(condition) 
{
  doSomething();
} 
----

include::../exceptions.adoc[]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
