<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when <code>return</code> or <code>yield</code> are used outside of a function.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p><code>yield</code> and <code>return</code> only make sense in the context of functions. Using them outside a function raises a <code>SyntaxError</code>.</p>
</div>
<div class="paragraph">
<p>If the goal is to break out of a loop, use <code>break</code> instead of <code>return</code>.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">a = 1
while a &lt; 3:
    if a % 2 == 0:
        return # Noncompliant: return is outside of a function
    a += 1

for n in range(5):
  yield n # Noncompliant: yield is outside of a function</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">a = 1
while a &lt; 3:
    if a % 2 == 0:
        break
    a += 1

def gen():
    for n in range(5):
      yield n</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Python Documentation - <a href="https://docs.python.org/3/reference/expressions.html?highlight=yield#yield-expressions">Yield expressions</a></p>
</li>
<li>
<p>Python Documentation - <a href="https://docs.python.org/3/reference/simple_stmts.html?highlight=return%20tatement#the-return-statement">Return statement</a></p>
</li>
</ul>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove this use of "xxx".</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_17_mar_2015_075940_elena_vilchik_wrote">on 17 Mar 2015, 07:59:40 Elena Vilchik wrote:</h3>
<div class="paragraph">
<p>Pylint rules</p>
</div>
<div class="paragraph">
<p>E0104: Return outside function</p>
</div>
<div class="paragraph">
<p>E0105: Yield outside function</p>
</div>
</div>
</div>
</div>