Using boxed type suggests that ``++null++`` is a possible value for the variable. Use of the primitive type should be preferred if this is not the case to avoid any confusion about possible values variable can contain.


== Noncompliant Code Example

----
Integer x = 5;
----


== Compliant Code Example

----
int x = 5;
----


ifdef::rspecator-view[]
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::rspecator-view[]
