== Why is this an issue?

Using boxed type suggests that ``++null++`` is a possible value for the variable. Use of the primitive type should be preferred if this is not the case to avoid any confusion about possible values variable can contain.


=== Noncompliant code example

[source,java]
----
Integer x = 5;
----


=== Compliant solution

[source,java]
----
int x = 5;
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Use ${primitiveType} instead of ${boxed type}


=== Highlighting

variable declaration


'''
== Comments And Links
(visible only on this page)

=== on 7 Nov 2018, 14:45:49 Alexandre Gigleux wrote:
\[~tibor.blenessy] Can you review the title of this RSPEC to make it compliant with our title policy?

I suggest something like: "Primitive types for non-null values should preferred to boxed types".

=== on 15 Nov 2018, 10:51:37 Alexandre Gigleux wrote:
\[~tibor.blenessy] 

I believe it would be great to write a sentence or two to explain what wrong can happen. Why this is so bad to used boxed type? Is it a performance problem, a memory problem?

Also in the Java code sample, we don't see/understand why this is bad to used boxed type. I believe you should had a ``++if x == null++`` to justify. 

=== on 18 Nov 2018, 21:03:09 Tibor Blenessy wrote:
\[~alexandre.gigleux] I don't understand the question, nothing wrong can happen, it's just confusing, that's what is explained in the rspec. Call me if we need to discuss further

endif::env-github,rspecator-view[]
