== Why is this an issue?

Using boxed type suggests that ``++null++`` is a possible value for the variable. Use of the primitive type should be preferred if this is not the case to avoid any confusion about possible values variable can contain.


=== Noncompliant code example

[source,java]
----
Integer x = 5;
----


=== Compliant solution

[source,java]
----
int x = 5;
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

include::highlighting.adoc[]

'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
