WordPress has a database repair and optimization mode that can be activated by setting WP_ALLOW_REPAIR to true in the configuration.

If activated, the repair page can be accessed by any user, authenticated or not. This makes sense because if the database is corrupt, the authentication mechanism might not work. 

After making use of this feature, it should be deactivated again. Otherwise, malicious users could trigger this potentially costly operation repeatadly slowing down the website, and making it unavailable.

== Ask Yourself Whether
* You really need to always have the database repair more activated.



== Noncompliant Code Example

----
define( 'WP_ALLOW_REPAIR', true ); // Sensitive
----

== Compliant Solution

----
// The default value is false, so the value does not have to be expilicitly set.
define( 'WP_ALLOW_REPAIR', false );
----
