<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when code manually reads from a <code>Reader</code> using a buffer and accumulates the content, instead of using the simpler <code>readAllAsString()</code> or <code>readAllLines()</code> methods available in Java 25+.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Manual buffer-based reading from a <code>Reader</code> involves repetitive boilerplate code that can be simplified with Java 25&#8217;s new methods.</p>
</div>
<div class="paragraph">
<p>Java 25 introduces two new convenience methods:</p>
</div>
<div class="ulist">
<ul>
<li>
<p><code>readAllAsString()</code> - reads all content as a single String</p>
</li>
<li>
<p><code>readAllLines()</code> - reads all content as a List of String lines</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>The traditional approach requires creating containers, allocating buffers, writing loops, and manual buffer management. This pattern is verbose and obscures the code&#8217;s intent.</p>
</div>
<div class="paragraph">
<p>The new methods make the code more concise and the intent clearer.</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="paragraph">
<p>Using manual buffer reading instead of the new convenience methods leads to unnecessary code complexity and reduced maintainability. While functionally equivalent, the verbose approach makes the codebase harder to read and maintain, and increases the chance of implementation errors in buffer handling.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Replace the manual buffer reading loop with a direct call to <code>readAllAsString()</code> when you need all content as a single string.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">StringBuilder content = new StringBuilder();
char[] buffer = new char[4096];
int charsRead;
while ((charsRead = reader.read(buffer)) != -1) { // Noncompliant
    content.append(buffer, 0, charsRead);
}
return content.toString();</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">return reader.readAllAsString();</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Java 25 Reader API Documentation - <a href="https://docs.oracle.com/en/java/javase/25/docs/api/java.base/java/io/Reader.html#readAllAsString()">Official documentation for the readAllAsString() method</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>