Sharing some naming conventions is a key point to make it possible for a team to efficiently collaborate.

This rule allows to check that all class names (along with Objective-C interface, protocol and implementation names) match a provided regular expression.

== Noncompliant Code Example

With default provided regular expression ``++^[A-Z][a-zA-Z0-9]*$++``:

----
// C++
class foo // Noncompliant
{
};

// Objective-C
@interface nonCompliant : NSObject
@end
----

== Compliant Solution

----
// C++
class Foo // Compliant
{
};

// Objective-C
@interface Compliant : NSObject
@end
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::parameters.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
