== How to fix it in libxmljs

=== Code examples

include::../../common/fix/code-rationale.adoc[]

==== Noncompliant code example

[source,javascript,diff-id=1,diff-type=noncompliant]
----
var libxmljs = require('libxmljs');
var fs = require('fs');

var xml = fs.readFileSync('xxe.xml', 'utf8');
libxmljs.parseXmlString(xml, {
                               noblanks: true,
                               noent: true, // Noncompliant
                               nocdata: true
                             }
                       );
----

==== Compliant solution

[source,javascript,diff-id=1,diff-type=compliant]
----
var libxmljs = require('libxmljs');
var fs = require('fs');

var xml = fs.readFileSync('xxe.xml', 'utf8');
// noent is set to false by default
libxmljs.parseXmlString(xml);
----

=== How does this work?

include::../../common/fix/validation.adoc[]
