<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>The <a href="https://api.flutter.dev/flutter/widgets/StatefulWidget/createState.html"><code>createState</code> method of a <code>StatefulWidget</code></a> should return a brand new <a href="https://api.flutter.dev/flutter/widgets/State-class.html"><code>State</code></a> object, and do nothing more than that.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>A Flutter widget is said to be stateful if it has a mutable state that can change over time. More accurately, a widget is stateful when it defines a piece of user interface that may change appearance or behavior over time. Such changes of appearance or behavior are modelled into variations of the internal state of the widget.</p>
</div>
<div class="paragraph">
<p>The state defined in a stateful Flutter widget is <a href="https://docs.flutter.dev/data-and-backend/state-mgmt/ephemeral-vs-app">ephemeral</a>, meaning that it is not persistent and is not to be shared between different instances of the same widget.</p>
</div>
<div class="paragraph">
<p>This is because in most cases, the state of a widget is local and doesn&#8217;t need to be accessed by other widgets in the components tree. Therefore, the Flutter framework can make simplifying assumptions on the lifecycle of such a state, incapsulating it in the <code>StatefulWidget</code> class, that exposes the <a href="https://api.flutter.dev/flutter/widgets/StatefulWidget/createState.html"><code>createState</code> method</a>.</p>
</div>
<div class="paragraph">
<p>The <code>createState</code> method is called by the framework whenever a new instance of the widget is created, and should only create a new state object.</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-dart" data-lang="dart">import 'package:flutter/widgets/framework.dart';

class MyState extends State&lt;MyStatefulWidget&gt; {
  int i1 = 42;

  MyState() { }
}

class MyStatefulWidget extends StatefulWidget {
  @override
  MyState createState() =&gt; MyState();
}</code></pre>
</div>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="paragraph">
<p>Doing more than creating a new instance in the <code>createState</code> method can lead to unexpected behavior. For example, if a state object is cached in a top-level variable and reused across multiple instances of the same widget, a change in the state of one instance will affect all other instances sharing the same state object.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Remove the logic from the <code>createState</code> method, or move it to the appropriate location.</p>
</div>
<div class="ulist">
<ul>
<li>
<p>Any logic that reuses an already existing state object should be removed, as it violates the design of <code>StatefulWidget</code>.</p>
</li>
<li>
<p>Any logic that is related to editing the new <code>State</code> object should be moved to the construction phase of the <code>State</code> or to the <code>State.initState</code> method.</p>
</li>
<li>
<p>Any logic that is temporarily storing the newly created <code>State</code> object should be removed, and the object should be returned directly.</p>
</li>
</ul>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-dart" data-lang="dart">late MyState global;

class MyState extends State&lt;MyStatefulWidget&gt; {
  int i1 = 42;

  MyState() { }
}

class MyStatefulWidget extends StatefulWidget {
  @override
  MyState createState() {
    global = MyState(); // Non compliant: global state object is reused
    return global;
  }
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-dart" data-lang="dart">class MyState extends State&lt;MyStatefulWidget&gt; {
  int i1 = 42;

  MyState() { }
}

class MyStatefulWidget extends StatefulWidget {
  @override
  MyState createState() =&gt; MyState();
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_noncompliant_code_example_2">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-dart" data-lang="dart">class MyState extends State&lt;MyStatefulWidget&gt; {
  int i1 = 42;

  MyState() { }
}

class MyStatefulWidget extends StatefulWidget {
  @override
  MyState createState() {
    final myState = MyState();
    myState.i1 = 43;
    return ;
  }
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution_2">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-dart" data-lang="dart">class MyState extends State&lt;MyStatefulWidget&gt; {
  int i1 = 43;

  MyState() { }
}

class MyStatefulWidget extends StatefulWidget {
  @override
  MyState createState() =&gt; MyState();
}</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Dart Docs - <a href="https://dart.dev/tools/linter-rules/no_logic_in_create_state">Dart Linter rule - no_logic_in_create_state</a></p>
</li>
<li>
<p>Flutter API Reference - <a href="https://docs.flutter.dev/data-and-backend/state-mgmt/ephemeral-vs-app">State management - Differentiate between ephemeral state and app state</a></p>
</li>
<li>
<p>Flutter API Reference - <a href="https://api.flutter.dev/flutter/widgets/StatefulWidget/createState.html">StatefulWidget class - createState abstract method</a></p>
</li>
<li>
<p>Flutter API Reference - <a href="https://api.flutter.dev/flutter/widgets/State-class.html">State class</a></p>
</li>
</ul>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Don&#8217;t put any logic in 'createState'.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>The entire body of the <code>createState</code> method, including the braces, when present.</p>
</div>
</div>
</div>
</div>