<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Extension identifier, when present, should be in PascalCase.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Extensions are a way to add functionality to existing classes. As such, they contain methods, getters, and setters, in the same way classes do.</p>
</div>
<div class="paragraph">
<p>Therefore, it makes sense to adopt the same naming convention as for classes, which is <a href="https://en.wikipedia.org/wiki/Camel_case">PascalCase</a> (a.k.a. UpperCamelCase).</p>
</div>
<div class="paragraph">
<p>This makes the code more homogeneous and doesn&#8217;t mislead developers into thinking that the extension is a variable or a function.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-dart" data-lang="dart">extension my_extension on String { // lower_snake_case
  void myMethod() {
    print('Hello');
  }
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-dart" data-lang="dart">extension MyExtension on String { // UpperCamelCase
  void myMethod() {
    print('Hello');
  }
}</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Dart Docs - <a href="https://dart.dev/tools/linter-rules/camel_case_extensions">Dart Linter rule - camel_case_extensions</a></p>
</li>
<li>
<p>Dart Docs - <a href="https://dart.dev/language/extension-methods">Language - extension methods</a></p>
</li>
<li>
<p>Wikipedia - <a href="https://en.wikipedia.org/wiki/Camel_case">Camel case</a></p>
</li>
</ul>
</div>
</div>
<div class="sect2">
<h3 id="_related_rules">Related rules</h3>
<div class="ulist">
<ul>
<li>
<p><a data-rspec-id="S101" class="rspec-auto-link">S101</a> - Class names should comply with a naming convention</p>
</li>
<li>
<p><a data-rspec-id="S7075" class="rspec-auto-link">S7075</a> - Non-constant names should comply with a naming convention</p>
</li>
</ul>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="ulist">
<ul>
<li>
<p>The extension name '&lt;extension identifier&gt;' isn&#8217;t an UpperCamelCase identifier.</p>
</li>
</ul>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="ulist">
<ul>
<li>
<p>The identifier of the extension.</p>
</li>
</ul>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
</div>
</div>