<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>An Open SQL <code>SELECT</code> statement without an explicit <code>ORDER BY</code> clause will retrieve rows in an unpredictable order. On pool/cluster tables, the current implementation of Open SQL <code>SELECT</code>  returns the result set in the primary key order, but that&#8217;s not the case for transparent tables. That&#8217;s why it&#8217;s safer to always use an <code>ORDER BY</code> clause.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-abap" data-lang="abap">OPEN CURSOR C FOR SELECT * FROM SBOOK WHERE CARRID = 'LH '. "NonCompliant
SELECT * FROM FLIGHTS WHERE FLIGHT_NUMBER = 'LH '."NonCompliant</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-abap" data-lang="abap">OPEN CURSOR C FOR SELECT * FROM SBOOK WHERE CARRID = 'LH '
  ORDER BY PRIMARY KEY.
SELECT * FROM FLIGHTS WHERE FLIGHT_NUMBER = 'LH ' ORDER BY PRIMARY KEY.</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Add an "ORDER BY" clause to this SQL "SELECT" statement.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_relates_to_s1592">relates to: <a data-rspec-id="S1592" class="rspec-auto-link">S1592</a></h3>

</div>
<div class="sect2">
<h3 id="_relates_to_s3270">relates to: <a data-rspec-id="S3270" class="rspec-auto-link">S3270</a></h3>

</div>
<div class="sect2">
<h3 id="_on_2_dec_2014_144156_ann_campbell_wrote">on 2 Dec 2014, 14:41:56 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~freddy.mallet] no code samples?</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_4_dec_2014_094424_freddy_mallet_wrote">on 4 Dec 2014, 09:44:24 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>Done [~ann.campbell.2]!</p>
</div>
</div>
</div>
</div>