This rule raises an issue when ActiveSupport's `try!` method is used instead of Ruby's native safe navigation operator `&.`.

== Why is this an issue?

Ruby 2.3 introduced the safe navigation operator (`&.`) as a native language feature that provides the same functionality as ActiveSupport's `try!` method but with better performance characteristics.

The safe navigation operator is part of Ruby's core language, which means:

* **Better performance**: Native language features are typically faster than library-based implementations
* **Reduced dependencies**: Using built-in language features reduces reliance on external libraries
* **Improved readability**: The `&.` syntax is more concise and widely recognized by Ruby developers
* **Future compatibility**: Native language features are more stable and less likely to change compared to library methods

ActiveSupport's `try!` method was useful before Ruby 2.3, but now that the safe navigation operator is available, it's the preferred approach for safe method chaining.

=== What is the potential impact?

Using `try!` instead of the safe navigation operator can lead to:

* **Performance degradation**: Method-based approaches are slower than native language operators
* **Unnecessary complexity**: Additional dependency on ActiveSupport when native Ruby features suffice
* **Maintenance overhead**: Code that relies on library methods when language features are available

== How to fix it

Replace `try!` method calls with the safe navigation operator `&.`. The safe navigation operator will return `nil` if the receiver is `nil`, otherwise it will call the method.

=== Code examples

==== Noncompliant code example

[source,ruby,diff-id=1,diff-type=noncompliant]
----
obj.try! :fly
user.try! :name
record.try! :save!
----

==== Compliant solution

[source,ruby,diff-id=1,diff-type=compliant]
----
obj&.fly
user&.name
record&.save!
----

== Resources

=== Documentation

 * Ruby Safe Navigation Operator Documentation - https://docs.ruby-lang.org/en/master/syntax/calling_methods_rdoc.html#label-Safe+navigation+operator[Official Ruby documentation explaining the safe navigation operator]

 * Rails Style Guide - Safe Navigation Operator - https://rails.rubystyle.guide/#try-bang[Rails community style guide recommendation for using safe navigation operator over try!]

=== Related rules

 * S6582 - Optional chaining should be preferred (JavaScript equivalent)
