This rule raises an issue when Salesforce object field names are referenced as hardcoded string literals instead of using `@salesforce/schema` imports in Lightning Web Components.

== Why is this an issue?

Using hardcoded field names as strings in Lightning Web Components creates several maintainability and reliability issues.

When you reference Salesforce fields using string literals like `'Name'` or `'CustomField1__c'`, your code becomes fragile. If a field is renamed or deleted in the Salesforce org, these string references will break at runtime, potentially causing your component to fail silently or display errors to users.

The `@salesforce/schema` import syntax provides compile-time validation and proper field reference tracking. When you import fields using this syntax, Salesforce can track these dependencies and prevent field deletion if they're still in use. This also enables better refactoring support - if a field is renamed, the import will automatically reflect the change.

Additionally, using schema imports provides better developer experience with IDE support, including autocomplete and type checking, making your code more maintainable and less prone to typos.

=== What is the potential impact?

Using hardcoded field names can lead to runtime failures when fields are renamed or deleted. Components may fail silently or display errors to end users. The lack of proper field reference tracking makes it difficult to understand field dependencies and can result in broken functionality after org changes.

== How to fix it

Replace hardcoded field name strings with proper @salesforce/schema imports. Import each field individually and use the imported constants instead of string literals.

=== Code examples

==== Noncompliant code example

[source,javascript,diff-id=1,diff-type=noncompliant]
----
import { LightningElement } from 'lwc';

export default class MyComponent extends LightningElement {
    fields = ['Name', 'Phone', 'CustomField1__c']; // Noncompliant
}
----

==== Compliant solution

[source,javascript,diff-id=1,diff-type=compliant]
----
import { LightningElement } from 'lwc';
import NAME_FIELD from '@salesforce/schema/Contact.Name';
import PHONE_FIELD from '@salesforce/schema/Contact.Phone';
import CUSTOM_FIELD1 from '@salesforce/schema/Contact.CustomField1__c';

export default class MyComponent extends LightningElement {
    fields = [NAME_FIELD, PHONE_FIELD, CUSTOM_FIELD1];
}
----

== Resources

=== Documentation

 * Lightning Web Components Developer Guide - Reference Salesforce Objects and Fields - https://developer.salesforce.com/docs/component-library/documentation/en/lwc/lwc.reference_salesforce_modules[Official Salesforce documentation on how to reference Salesforce objects and fields in Lightning Web Components]

 * Lightning Data Service in Lightning Web Components - https://developer.salesforce.com/docs/component-library/documentation/en/lwc/lwc.data_service[Documentation on using Lightning Data Service with proper field references]
