== Why is this an issue?

In Node.js importing modules is doable by providing an absolute path such as `/lib/foo/bar.js`. Doing this restricts the portability of your code, making it specific to your computer's file system and potentially causing issues when the code is distributed, for example, through NPM packages.

== How to fix it

Replace the absolute path with one that is relative to your current file.

=== Code examples

==== Noncompliant code example

[source,js,diff-id=1,diff-type=noncompliant]
----
import { foo } from '/home/project/api/bar.js';
----

==== Compliant solution

[source,text,diff-id=1,diff-type=compliant]
----
import { foo } from '../api/bar.js';
----

== Resources
=== Documentation

* MDN web docs - https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/import[import]
* Node.js docs - https://nodejs.org/api/esm.html[ECMAScript modules]
