== Why is this an issue?

include::../description.adoc[]

=== Code examples

==== Noncompliant code example

[source,javascript,diff-id=1,diff-type=noncompliant]
----
function foo(a) {
  var b = (a === 'A') ? 'is A' : 'is not A'; // Noncompliant
  // ...
}
----

==== Compliant solution

[source,javascript,diff-id=1,diff-type=compliant]
----
function foo(a) {
  var b;
  if (a === 'A') {
    b = 'is A';
  } 
  else {
    b = 'is not A';
  }
  // ...
}
----

== Resources
=== Documentation

* https://eslint.org[eslint] - Rule https://eslint.org/docs/latest/rules/no-ternary[no-ternary]
ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
