include::../description.adoc[]

== Noncompliant Code Example

----
function foo(a) {
  var b = (a === 'A') ? 'is A' : 'is not A'; // Noncompliant
  // ...
}
----

== Compliant Solution

----
function foo(a) {
  var b;
  if (a === 'A') {
    b = 'is A';
  } 
  else {
    b = 'is not A';
  }
  // ...
}
----
