<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Global variables that are only used by a single subprocedure should be declared in that subprocedure, thus limiting both their scope and other procedures' ability to modify the variable unexpectedly.</p>
</div>
<div class="paragraph">
<p>If the variable has been declared in the global scope to preserve its value between calls, add the <code>STATIC</code> keyword to the new local declaration. It will be added to the global heap and retained until the program ends, but only accessible by the subprocedure in which it was declared.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-text" data-lang="text">     D SUBPROC1        PR
     D  Parm1                         5  0 Const

     D X               S              5  0    // Noncompliant; only used in one subproc

      /Free
       SUBPROC1(1234);
       *inlr = *on;
       return;
       /End-free

     P SUBPROC1        B
     D SUBPROC1        PI
     D  Parm1                         5  0 Const
      /Free
       X = Parm1 + 1;                        // X is only used here
       DSPLY(E) X;
      /End-Free
     P                 E</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-text" data-lang="text">     D SUBPROC1        PR
     D  Parm1                         5  0 Const

      /Free
       SUBPROC1(1234);
       *inlr = *on;
       return;
       /End-free

     P SUBPROC1        B
     D SUBPROC1        PI
     D  Parm1                         5  0 Const
     D X               S              5  0
      /Free
       X = Parm1 + 1;
       DSPLY(E) X;
      /End-Free
     P                 E</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Move this declaration of XXX into YYY to make it a local variable.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_relates_to_s1450">relates to: <a data-rspec-id="S1450" class="rspec-auto-link">S1450</a></h3>

</div>
<div class="sect2">
<h3 id="_is_related_to_s806">is related to: <a data-rspec-id="S806" class="rspec-auto-link">S806</a></h3>

</div>
<div class="sect2">
<h3 id="_on_31_jul_2014_151809_freddy_mallet_wrote">on 31 Jul 2014, 15:18:09 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] I would associate this rule to the category "Maintainability &#8594; Understandability". I see the risk associated to such bad coding practice but for me the most important impact is on the maintainability.</p>
</div>
</div>
</div>
</div>