== Why is this an issue?

Labeled blocks are useful, especially when the code is badly indented, to help maintainers match the beginning and ending of each block. When blocks are nested, labeling them can improve the code's readability. This rule detects nested block which do not have a start label.


=== Noncompliant code example

[source,sql]
----
BEGIN -- Compliant, this is not a nested block
  NULL;
END;
/

BEGIN
  BEGIN -- Noncompliant; this nested block has no label
    NULL;
  END;
END;
/

BEGIN
  BEGIN -- Noncompliant; this nested block has only an end label
    NULL;
  END myBlockLabel1;

  <<myBlockLabel2>> -- Compliant
  BEGIN
    NULL;
  END;
END;
/
----


=== Compliant solution

[source,sql]
----
BEGIN 
  NULL;
END;
/

BEGIN
  BEGIN myBlockLabel0
    NULL;
  END myBlockLabel0;
END;
/

BEGIN
  BEGIN myBlockLabel1
    NULL;
  END myBlockLabel1;

  <<myBlockLabel2>>
  BEGIN
    NULL;
  END;
END;
/
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Add a label to this nested block.


'''
== Comments And Links
(visible only on this page)

=== on 28 Jan 2015, 20:43:43 Ann Campbell wrote:
\[~dinesh.bolkensteyn] this rule description was copied from Nemo & I don't understand why ``++<<myBlockLabel2>>++`` is labeled noncompliant. Can you verify for me that it is, please?

=== on 11 May 2015, 11:48:47 Dinesh Bolkensteyn wrote:
You're right [~ann.campbell.2] - it actually is compliant. I've updated the description.


Note that there is another rule that verifies that, whenever there is a starting label, there also is a corresponding ending one.

=== on 11 May 2015, 13:50:55 Ann Campbell wrote:
corresponding Compliant Solution added

endif::env-github,rspecator-view[]
