<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Shared coding conventions allow teams to collaborate efficiently, and consistently indented code has a strong impact on readability. This rule checks that the following spacing conventions are followed:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>01 level declaration should start in a specific column (8 by default)</p>
</li>
<li>
<p>Subordinate data items must be indented from superior data items.</p>
</li>
<li>
<p>In a file, all data item declarations having the same level should start in the same column</p>
</li>
</ul>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-text" data-lang="text">01 StudentDetails.
     05 StudentId        PIC 9(7).    *&lt; Noncompliant: indented too far
   05 StudentName.
      07 FirstName     PIC X(10).
        07 MiddleInitial PIC X.            *&lt; Noncompliant
      07 Surname       PIC X(15).
   05 DateOfBirth.
      07 DayOfBirth    PIC 99.
      07 MonthOfBirth  PIC 99.
    07 YearOfBirth   PIC 9(4).          *&lt; Noncomopliant
   09 CourseCode       PIC X(4).     *&lt; Noncompliant: not indented enough</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-text" data-lang="text">01 StudentDetails.
   05 StudentId        PIC 9(7).
   05 StudentName.
      07 FirstName     PIC X(10).
      07 MiddleInitial PIC X.
      07 Surname       PIC X(15).
   05 DateOfBirth.
      07 DayOfBirth    PIC 99.
      07 MonthOfBirth  PIC 99.
      07 YearOfBirth   PIC 9(4).
         09 CourseCode       PIC X(4).</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_duplicates_s1728">duplicates: <a data-rspec-id="S1728" class="rspec-auto-link">S1728</a></h3>

</div>
<div class="sect2">
<h3 id="_on_14_may_2014_132407_freddy_mallet_wrote">on 14 May 2014, 13:24:07 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>@Ann, I let you review this specification and then notify @Marie-Christine to be sure that is fits her needs.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_8_aug_2014_192340_ann_campbell_wrote">on 8 Aug 2014, 19:23:40 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>rolled into <a data-rspec-id="S1728" class="rspec-auto-link">RSPEC-1728</a></p>
</div>
</div>
</div>
</div>