== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,sql]
----
BEGIN
  SELECT col2, col3
  BULK COLLECT INTO result
  FROM my_table
  ORDER BY
    1 ASC;           -- Noncompliant - if col1 is added to the selected fields, this may break
END;
/
----

=== Compliant solution

[source,sql]
----
BEGIN
  SELECT col2, col3
  BULK COLLECT INTO result
  FROM my_table
  ORDER BY
    col2 ASC;  
END;
/
----
ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

endif::env-github,rspecator-view[]
