== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,dart]
----
void foo() {
  outer: //label is not used.
  for(int i = 0; i < 10; i++) {
    break;
  }
}
----

=== Compliant solution

[source,dart]
----
void foo() {
  for(int i = 0; i < 10; i++) {
    break;
  }
}
----

== Resources

* https://wiki.sei.cmu.edu/confluence/x/5dUxBQ[CERT, MSC12-C.] - Detect and remove code that has no effect or is never executed
