== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,dart,diff-id=1,diff-type=noncompliant]
----
void foo() {
  outer: //label is not used.
  for(int i = 0; i < 10; i++) {
    break;
  }
}
----

=== Compliant solution

[source,dart,diff-id=1,diff-type=compliant]
----
void foo() {
  for(int i = 0; i < 10; i++) {
    break;
  }
}
----

== Resources

* CERT - https://wiki.sei.cmu.edu/confluence/x/5dUxBQ[CERT, MSC12-C.] - Detect and remove code that has no effect or is never executed
* Dart Docs - https://dart.dev/tools/diagnostic-messages?utm_source=dartdev&utm_medium=redir&utm_id=diagcode&utm_content=unnecessary_cast#unused_label[Dart Linter rules - unused_label]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

* The label '<labelName>' isn't used.

`<labelName>` doesn't include the semicolon.

=== Highlighting

The label name, including the semicolon.

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
