This rule raises an issue when operators, commas, colons, semicolons, or braces lack proper spacing around them.

== Why is this an issue?

Proper spacing around operators and punctuation significantly improves code readability and maintainability. When operators like `=`, `+`, `-`, `*` are cramped together without spaces, the code becomes harder to scan and understand quickly.

Consistent spacing follows widely accepted Ruby style conventions and makes code more professional. It helps developers:

* Parse expressions more easily by creating visual separation between operands and operators
* Spot syntax errors more quickly
* Maintain consistent formatting across the codebase
* Reduce cognitive load when reading code

Without proper spacing, mathematical expressions and assignments can look cluttered and be prone to misreading. For example, `result=x+y*z` requires more mental effort to parse than `result = x + y * z`.

=== What is the potential impact?

Poor spacing reduces code readability and can slow down development and code review processes. While it doesn't affect functionality, it makes the codebase harder to maintain and can lead to inconsistent formatting across the project.

== How to fix it

Add spaces around operators, after commas, colons and semicolons, and around braces. This creates visual separation that makes the code easier to read and understand.

=== Code examples

==== Noncompliant code example

[source,ruby,diff-id=1,diff-type=noncompliant]
----
result=x+y*z
hash={key:value,other:data}
array=[1,2,3]
----

==== Compliant solution

[source,ruby,diff-id=1,diff-type=compliant]
----
result = x + y * z
hash = { key: value, other: data }
array = [1, 2, 3]
----

== Resources

=== Documentation

 * Shopify Ruby Style Guide - Layout - https://ruby-style-guide.shopify.dev/#layout[Official Ruby style guide covering spacing conventions around operators and punctuation]

 * RuboCop Layout/SpaceAroundOperators - https://docs.rubocop.org/rubocop/cops_layout.html#layoutspacearoundoperators[RuboCop cop that enforces spaces around operators]
