<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Test frameworks provide a mechanism to skip tests if their prerequisites are not met, by either calling dedicated methods (e.g: <code>unittest.TestCase.skipTest</code>, <code>pytest.skip</code>, &#8230;&#8203;) or using decorators (e.g: <code>unittest.skip</code>, <code>pytest.mark.skip</code>, &#8230;&#8203;)</p>
</div>
<div class="paragraph">
<p>Using a <code>return</code> statement instead will make the test succeed, even though no assertion has been performed. It is therefore better to flag the test as <code>skipped</code> in such situation.</p>
</div>
<div class="paragraph">
<p>This rule raises an issue when a <code>return</code> is performed conditionally at the beginning of a test method.</p>
</div>
<div class="paragraph">
<p>No issue will be raised if the <code>return</code> is unconditional as <a data-rspec-id="S1763" class="rspec-auto-link">S1763</a> already raises an issue in such case.</p>
</div>
<div class="paragraph">
<p>The supported frameworks are <code>Pytest</code> and <code>Unittest</code>.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">import unittest
class MyTest(unittest.TestCase):

    def test_something(self):
        if not external_resource_available():
            return  # Noncompliant
        self.assertEqual(foo(), 42)</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">import unittest
class MyTest(unittest.TestCase):

    def test_something(self):
        if not external_resource_available():
            self.skipTest("prerequisite not met")
        self.assertEqual(foo(), 42)</code></pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p><a href="https://docs.pytest.org/en/latest/how-to/skipping.html">Pytest: skipping test functions</a></p>
</li>
<li>
<p><a href="https://docs.python.org/3/library/unittest.html#skipping-tests-and-expected-failures">Unittest: skipping tests and expected failures</a></p>
</li>
</ul>
</div>
<hr>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Skip this test explicitly.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>The return statement</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_is_related_to_s1763">is related to: <a data-rspec-id="S1763" class="rspec-auto-link">S1763</a></h3>

</div>
</div>
</div>