include::../description.adoc[]

== Noncompliant Code Example

With the default threshold of 3:

----
       PROCEDURE DIVISION.

       DISPLAY "Firstname: ".
      *...
       DISPLAY "Firstname: ".
      *...
       DISPLAY "Firstname: ".
----

== Compliant Solution

----
       WORKING-STORAGE SECTION.
         01 FIRST-NAME-HEADER PIC X(42) VALUE "Firstname: ".
       PROCEDURE DIVISION.

       DISPLAY FIRST-NAME-HEADER
      *...
       DISPLAY FIRST-NAME-HEADER
      *...
       DISPLAY FIRST-NAME-HEADER
----

== Exceptions

Literals with fewer than 7 characters are ignored.


Only duplications located in a ``++PROCEDURE DIVISION++``, not those contained in copybooks are reported.

ifdef::env-github,rspecator-view[]
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
