== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,ruby]
----
def set_name(name)
  name = name
end
----

=== Compliant solution

[source,ruby]
----
def set_name(name)
  @name = name
end
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
