<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when <code>String.isEmpty()</code> is used to validate strings that should also reject whitespace-only content.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>The <code>String.isEmpty()</code> method only checks if a string is null or has zero length. It does not consider strings containing only whitespace characters as empty, which can lead to unexpected behavior in validation logic.</p>
</div>
<div class="paragraph">
<p>For example, a string containing only spaces (<code>' '</code>) will return <code>false</code> when checked with <code>isEmpty()</code>, even though it contains no meaningful content. This can cause issues in:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>User input validation where whitespace-only entries should be rejected</p>
</li>
<li>
<p>Data processing where empty fields need to be identified</p>
</li>
<li>
<p>Business logic that depends on meaningful string content</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>The <code>String.isBlank()</code> method provides more comprehensive validation by returning <code>true</code> for null strings, empty strings, and strings containing only whitespace characters. This makes it the better choice when you need to ensure a string contains actual content.</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="paragraph">
<p>Using <code>isEmpty()</code> instead of <code>isBlank()</code> can lead to validation bypasses where whitespace-only strings are incorrectly treated as valid input. This may result in poor data quality, unexpected application behavior, or business logic errors when processing user input or data fields.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Replace <code>String.isEmpty()</code> with <code>String.isBlank()</code> when you need to validate that a string contains meaningful content, not just check if it&#8217;s null or empty.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-apex" data-lang="apex">if (!String.isEmpty(inputString)) { // Noncompliant
    processString(inputString);
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-apex" data-lang="apex">if (!String.isBlank(inputString)) {
    processString(inputString);
}</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Apex String Class Methods - <a href="https://developer.salesforce.com/docs/atlas.en-us.apexref.meta/apexref/apex_methods_system_string.htm">Official Salesforce documentation for String class methods including isBlank() and isEmpty()</a></p>
</li>
<li>
<p>Apex String.isBlank() Method - <a href="https://developer.salesforce.com/docs/atlas.en-us.apexref.meta/apexref/apex_methods_system_string.htm#apex_System_String_isBlank">Official Salesforce documentation for isBlank()</a></p>
</li>
</ul>
</div>
</div>
<div class="sect2">
<h3 id="_related_rules">Related rules</h3>
<div class="ulist">
<ul>
<li>
<p><a data-rspec-id="S7158" class="rspec-auto-link">S7158</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>