include::../description.adoc[]

== Noncompliant Code Example

With default provided regular expression: ``++^(_|[a-zA-Z0-9]+)$++``:

[source, go]
----
func execute_all() {
...
}
----

== Compliant Solution

[source, go]
----
func executeAll() {
...
}
----
