include::../description.adoc[]

== Noncompliant Code Example

With default provided regular expression: ``++^(_|[a-zA-Z0-9]+)$++``:

----
func execute_all() {
...
}
----

== Compliant Solution

----
func executeAll() {
...
}
----
