include::../description.adoc[]

== Noncompliant code example

With default provided regular expression: ``++^(_|[a-zA-Z0-9]+)$++``:

[source, go]
----
func execute_all() {
...
}
----

== Compliant solution

[source, go]
----
func executeAll() {
...
}
----
