== Why is this an issue?

This rule is a strict implementation of a MISRA (Motor Industry Software Reliability Association) rule. MISRA defines best practices for developing safety-critical software. You can learn more about this rule in the MISRA documents referenced below.


== Resources

* MISRA {cpp} 2008, 3-3-1


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Either declare this [variable||function] in a header or mark it 'static'.


'''
== Comments And Links
(visible only on this page)

=== is duplicated by: S827

=== is related to: S806

=== is related to: S830

=== is related to: S832

=== on 16 Oct 2014, 12:53:05 Ann Campbell wrote:
\[~samuel.mercier] please:

* fill in the appropriate reference field(s).
* provide a See section.
* use the standard section titles: Noncompliant Code Example, Exception*s*, and heading levels (h2. instead of h3.).
* use the standard section order: description, Noncompliant Code Example, Compliant Solution, Exceptions, See* SQALE.
* use our standard: "Noncompliant", rather than "Non-compliant" (code sample).


=== on 17 Oct 2014, 15:51:08 Ann Campbell wrote:
\[~freddy.mallet] this an RSPEC-832 overlap to my mind.

=== on 17 Oct 2014, 19:08:06 Ann Campbell wrote:
FYI [~samuel.mercier] I've moved this to Data-related reliability - the danger appears to be that a non-static var could be visible (and thus accidentally used!) outside its intended scope. Change it back if I'm wrong.

endif::env-github,rspecator-view[]
