== Why is this an issue?

The `Comparable.compareTo` method returns a negative integer, zero, or a positive integer to indicate whether the object is less than, equal to, or greater than the specified object.
The sign of the return value or whether it is zero is what matters, not its magnitude.

Returning a positive or negative constant value other than the basic ones (-1, 0, or 1) lacks justification and indicates a misunderstanding of the principle.
It also misleads readers to think that the concrete value of the returned constant is somehow essential here.

When returning `Integer.MIN_VALUE`, an additional issue is 32-bit two's complement range overflow.
If a caller were to negate the result of `compareTo()`, which may happen to invert the tested condition, this attempt would result in unexpected behavior where the condition is not inverted as intended.

== How to fix it

Replace `Integer.MIN_VALUE` with -1.
Note that you should replace any other negative or positive constant return value with -1, 0, or 1 as well, but this rule is not so restrictive to report these cases as an issue.

=== Code examples

==== Noncompliant code example

[source,java,diff-id=1,diff-type=noncompliant]
----
public int compareTo(Name name) {
  if (condition) {
    return Integer.MIN_VALUE; // Noncompliant
  }
}
----

==== Compliant solution

[source,java,diff-id=1,diff-type=compliant]
----
public int compareTo(Name name) {
  if (condition) {
    return -1; // Compliant
  }
}
----

==== Compliant solution

[source,java]
----
public int compareTo(Name name) {
  if (condition) {
    return -2; // Compliant, but lacks justification
  }
}
----

==== Compliant solution

[source,java]
----
public int compareTo(Name name) {
  if (condition) {
    return hashCode() - name.hashCode(); // Compliant, not a constant
  }
}
----

== Resources

=== Documentation

* https://docs.oracle.com/javase/8/docs/api/java/lang/Comparable.html#compareTo-T-[Java SE 8 API Specification: Comparable.compareTo]
* https://docs.oracle.com/javase/8/docs/api/java/lang/Integer.html#MIN_VALUE[Java SE 8 API Specification: Integer.MIN_VALUE]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Simply return -1.


endif::env-github,rspecator-view[]
