<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>The use of Unicode escape sequences should be reserved for characters that would otherwise be ambiguous, such as unprintable characters.</p>
</div>
<div class="paragraph">
<p>This rule ignores sequences composed entirely of Unicode characters, but otherwise raises an issue for each Unicode character that represents a printable character.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">String prefix = "n\u00E9e"; // Noncompliant</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">String prefix = "n√©e";</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove this Unicode escape sequence and use the character instead.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_27_may_2015_084319_freddy_mallet_wrote">on 27 May 2015, 08:43:19 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>I would not activate this rule by default otherwise we might generate a lot of noise out of the box.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_27_may_2015_191511_ann_campbell_wrote">on 27 May 2015, 19:15:11 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>Inactive by default</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_2_jun_2015_135834_nicolas_peru_wrote">on 2 Jun 2015, 13:58:34 Nicolas Peru wrote:</h3>
<div class="paragraph">
<p>ok for me.</p>
</div>
</div>
</div>
</div>