== Why is this an issue?

When a collection is empty it makes no sense to access or iterate it. Doing so anyway is surely an error; either population was accidentally omitted or the developer doesn't understand the situation.

=== Noncompliant code example

[source,vbnet]
----
Public Sub Method()
    Dim Values As New List(Of String)
    Values.Remove("bar")                ' Noncompliant
    If Values.Contains("foo") Then      ' Noncompliant
    End If
    For Each Value As String In Values  ' Noncompliant
    Next
End Sub
----

==== Compliant solution

[source,vbnet]
---
Public Sub Method()
    Dim Values As List(Of String) = LoadValues()
    Values.Remove("bar")
    If Values.Contains("foo") Then
    End If
    For Each Value As String In Values
    Next
End Sub
---


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../highlighting.adoc[]

endif::env-github,rspecator-view[]
