== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,cpp]
----
for (element = list.first; element != null; element = element->next) { // First termination condition
  if (!matches(element->value)) {                                      // Second termination condition
    break; // Noncompliant
  }

  /* ... */
}
----


=== Compliant solution

[source,cpp]
----
// Compliant
for (element = list.first; element != null && matches(element->value); element = element->next) {
  /* ... */
}
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
