== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

With default provided regular expression:

[source,scala]
----
def DoSomething( ) : Unit = { // Noncompliant
   // ...
}
----

=== Compliant solution

[source,scala]
----
def doSomething( ) : Unit = {
   // ...
}
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
