include::../description.adoc[]

== Noncompliant Code Example

With default provided regular expression:

----
def DoSomething( ) : Unit = { // Noncompliant
   // ...
}
----

== Compliant Solution

----
def doSomething( ) : Unit = {
   // ...
}
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
