include::../description.adoc[]

== Noncompliant Code Example

With default provided regular expression:
----
def DoSomething( ) : Unit = { // Noncompliant
   // ...
}
----

== Compliant Solution

----
def doSomething( ) : Unit = {
   // ...
}
----
