<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Defining an inherent <code>to_string(&amp;self)</code> method on a type that also implements the <code>Display</code> trait can lead to confusion, as this inherent method will overshadow the <code>to_string</code> method automatically provided by the <code>Display</code> trait. The inherent method is less versatile, preventing proper use of the <code>Display</code> trait&#8217;s feature.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-rust" data-lang="rust">use std::fmt;

pub struct A;

impl A {
    pub fn to_string(&amp;self) -&gt; String {
        "I am A".to_string() // Noncompliant: Inherent method shadows `Display::to_string`.
    }
}

impl fmt::Display for A {
    fn fmt(&amp;self, f: &amp;mut fmt::Formatter) -&gt; fmt::Result {
        write!(f, "I am A, too")
    }
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-rust" data-lang="rust">use std::fmt;

pub struct A;

impl fmt::Display for A {
    fn fmt(&amp;self, f: &amp;mut fmt::Formatter) -&gt; fmt::Result {
        write!(f, "I am A")
    }
}</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Clippy Lints - <a href="https://rust-lang.github.io/rust-clippy/master/index.html#inherent_to_string_shadow_display" class="bare">https://rust-lang.github.io/rust-clippy/master/index.html#inherent_to_string_shadow_display</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>