It is very common to pass a collection constructor reference as an argument, for example ``++Collectors.toCollection(ArrayList::new)++`` takes the ``++ArrayList::new++`` constructor. When the method expects a ``++java.util.function.Supplier++`` it is perfectly fine. However when the method argument type is ``++java.util.function.Function++`` it means that an argument will be passed to the constructor.


The first argument of Collections constructors is usually an integer representing its "initial capacity". This is generally not what the developer expects, but the memory allocation is not visible at first glance.


This rule raises an issue when a collection constructor is passed by reference as a ``++java.util.function.Function++`` argument.


== Noncompliant Code Example

----
Arrays.asList(1, 2, 54000).stream().collect(Collectors.toMap(Function.identity(), ArrayList::new)); // Noncompliant, "ArrayList::new" unintentionally refers to "ArrayList(int initialCapacity)" instead of "ArrayList()"
----


== Compliant Solution

----
Arrays.asList(1, 2, 54000).stream().collect(Collectors.toMap(Function.identity(), id -> new ArrayList<>())); // Compliant, explicitly show the usage of "id -> new ArrayList<>()" or "id -> new ArrayList<>(id)"
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

include::highlighting.adoc[]

endif::env-github,rspecator-view[]
