<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when a Python side effect happens inside a <code>tensorflow.function</code>.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Python sides effects such as printing, mutating a list or a global variable, inside of a <code>tensorflow.function</code> may not behave as expected.
Because of the <a href="https://www.tensorflow.org/guide/function#rules_of_tracing">Rules of tracing</a>,
the execution of side effects will depend on the input values of the function and will execute only once per tracing.</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">import tensorflow as tf

@tf.function
def f(x):
  print("A side effect", x)

f(1) # prints "A side effect 1"
f(1) # does not print anything
f(2) # prints "A side effect 2"</code></pre>
</div>
</div>
<div class="paragraph">
<p>The example above depicts the issue encountered when using Python side effects in a <code>tensorflow.function</code>. As a single trace is created per input values,
the second call to <code>f(1)</code> does not output anything to the console.</p>
</div>
<div class="paragraph">
<p>The best practice would be to avoid using Python side effects and prefer the usage of the TensorFlow API with functions such as <code>tf.print</code> or tf.TensorArray`.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>To fix this issue either remove the side effect or use the corresponding TensorFlow function.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">import tensorflow as tf

@tf.function
def f(x):
  print("Printing", x) # Noncompliant print is a side effect</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">import tensorflow as tf

@tf.function
def f(x):
  tf.print("Printing", x) # Compliant</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>TensorFlow Documentation - <a href="https://www.tensorflow.org/guide/function#executing_python_side_effects">Executing Python side effects</a></p>
</li>
<li>
<p>TensorFlow Documentation - <a href="https://www.tensorflow.org/api_docs/python/tf/print">tf.print reference</a></p>
</li>
<li>
<p>TensorFlow Documentation - <a href="https://www.tensorflow.org/api_docs/python/tf/summary">tf.summary reference</a></p>
</li>
<li>
<p>TensorFlow Documentation - <a href="https://www.tensorflow.org/api_docs/python/tf/Variable#methods">tf.Variable methods reference</a></p>
</li>
<li>
<p>TensorFlow Documentation - <a href="https://www.tensorflow.org/api_docs/python/tf/TensorArray">tf.TensorArray reference</a></p>
</li>
<li>
<p>TensorFlow Documentation - <a href="https://www.tensorflow.org/api_docs/python/tf/data">tf.data reference</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>