<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Where the developer uses new versions of standard library functions (e.g. to enhance functionality or add checks of input values), the modified function shall have a new name. However, it is permissible to overload the name to add new parameter types if the functionality is consistent with those of the original. This ensures that the behaviour associated with the name remains consistent. So, for example, if a new version of the <code>sqrt</code> function is written to check that the input is not negative, the new function shall not be named <code>sqrt</code>, but shall be given a new name. It is permissible to add a new <code>sqrt</code> function for a type not present in the library.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">int32_t printf ( int32_t a, int32_t b ) // Noncompliant, printf declared in stdio.h
{
  return ( ( a &gt; b ) ? a : b );
}</code></pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p>MISRA C&#43;&#43;:2008, 17-0-3</p>
</li>
</ul>
</div>
<hr>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>The definition of "xxx" collides with a standard library function and should be renamed.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_is_related_to_s978">is related to: <a data-rspec-id="S978" class="rspec-auto-link">S978</a></h3>

</div>
<div class="sect2">
<h3 id="_on_20_oct_2014_135609_ann_campbell_wrote">on 20 Oct 2014, 13:56:09 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~samuel.mercier] FYI, I&#8217;ve switched this to Reliability since the real danger is that someone will override a std function name to do something the caller doesn&#8217;t expect.</p>
</div>
<div class="paragraph">
<p>Along the same reasoning, I added a 'pitfall' tag.</p>
</div>
</div>
</div>
</div>