{cpp} does not support the nesting of C-style comments even though some compilers support this as a non-portable language extension.

A comment beginning with ``++/*++`` continues until the first ``++*/++`` is encountered.

Any ``++/*++`` occurring inside a comment is a violation of this rule.

== Noncompliant Code Example

----
/* some comment, end comment marker accidentally omitted
Perform_Critical_Safety_Function(X);
/* this "comment" is Noncompliant */
----

In reviewing the code containing the call to the function, the assumption is that it is executed code.

Because of the accidental omission of the end comment marker, the call to Perform_Critical_ Safety_Function will not be executed.

== See

* MISRA C:2004, 2.3 - The character sequence /* shall not be used within a comment.
* MISRA {cpp}:2008, 2-7-1 - The character sequence /* shall not be used within a C-style comment.
* MISRA C:2012, 3.1 - The character sequences /* and // shall not be used within a comment
* https://wiki.sei.cmu.edu/confluence/x/k9YxBQ[CERT, MSC04-C.] - Use comments consistently and in a readable fashion

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
