<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Code is unreachable if there is no syntactic (control flow) path to it. If such code exists, it is unclear if this is intentional or simply that an appropriate path has been accidentally omitted.</p>
</div>
<div class="paragraph">
<p>Compilers may choose not to generate code for these constructs, meaning that, even if the unreachable code is intentional, it may not be present in the final executable code.</p>
</div>
<div class="paragraph">
<p>Missing statements, often caused by editing activities, are a common source of unreachable code.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">int16_t with_unreach ( int16_t para )
{
 int16_t local;
 local = 0;
 switch ( para )
 {
    local = para; // unreachable – Noncompliant
  case 1:
    break;
    para++; // unreachable – Noncompliant
  default:
    break;
  }
  return para;
  para++; // unreachable – Noncompliant
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">int16_t with_unreach ( int16_t para )
{
 int16_t local;
 local = 0;
 switch ( para )
 {
  case 1:
    break;
  default:
    break;
  }
  return para;
}</code></pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p>MISRA C:2004, 14.1</p>
</li>
<li>
<p>MISRA C&#43;&#43;:2008, 0-1-1</p>
</li>
<li>
<p>MISRA C:2012, 2.1</p>
</li>
</ul>
</div>
<hr>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_duplicates_s1763">duplicates: <a data-rspec-id="S1763" class="rspec-auto-link">S1763</a></h3>

</div>
<div class="sect2">
<h3 id="_relates_to_s901">relates to: <a data-rspec-id="S901" class="rspec-auto-link">S901</a></h3>

</div>
<div class="sect2">
<h3 id="_relates_to_s902">relates to: <a data-rspec-id="S902" class="rspec-auto-link">S902</a></h3>

</div>
<div class="sect2">
<h3 id="_on_17_oct_2014_152047_ann_campbell_wrote">on 17 Oct 2014, 15:20:47 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~samuel.mercier] based on the example provided, I&#8217;ve moved this to Reliability</p>
</div>
</div>
</div>
</div>