<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>The <code>enable_dl</code> PHP configuration setting allows PHP extensions to be loaded
dynamically at runtime.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>When dynamic loading is enabled, PHP code can load arbitrary PHP extensions by
calling the <code>dl</code> function. This can be used to bypass restrictions set with
the <code>open_basedir</code> configuration.</p>
</div>
<div class="paragraph">
<p>PHP defaults to allowing dynamic loading.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p><code>enable_dl</code> setting should be set to <code>0</code> in the main PHP configuration.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-php" data-lang="php">; php.ini
enable_dl=1  ; Noncompliant</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-php" data-lang="php">; php.ini
enable_dl=0</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_standards">Standards</h3>
<div class="ulist">
<ul>
<li>
<p>OWASP - <a href="https://owasp.org/Top10/A05_2021-Security_Misconfiguration/">Top 10 2021 Category A5 - Security Misconfiguration</a></p>
</li>
<li>
<p>OWASP - <a href="https://owasp.org/www-project-top-ten/2017/A6_2017-Security_Misconfiguration">Top 10 2017 Category A6 - Security Misconfiguration</a></p>
</li>
</ul>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="ulist">
<ul>
<li>
<p>Explicitly set "enable_dl" to false.</p>
</li>
<li>
<p>Update this "enable_dl" configuration to turn it off.</p>
</li>
</ul>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_1_sep_2015_071209_linda_martin_wrote">on 1 Sep 2015, 07:12:09 Linda Martin wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] I would have the same remark than for: <a data-rspec-id="S3338" class="rspec-auto-link">RSPEC-3338</a>, suggesting the addition of a compliant solution code snippet. WDYT ?</p>
</div>
<div class="paragraph">
<p>Also what would you think about a <code>php-ini</code> tag ?</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_1_sep_2015_135625_ann_campbell_wrote">on 1 Sep 2015, 13:56:25 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>compliant solution added and tag added to all relevant rules [~linda.martin]</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_1_sep_2015_144112_linda_martin_wrote">on 1 Sep 2015, 14:41:12 Linda Martin wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] nice! thank you! LGTM.</p>
</div>
</div>
</div>
</div>