<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>The purpose of checked exceptions is to ensure that errors will be dealt with, either by propagating them or by handling them, but some believe that checked exceptions negatively impact the readability of source code, by spreading this error handling/propagation logic everywhere.</p>
</div>
<div class="paragraph">
<p>This rule verifies that no method throws a new checked exception.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">public void myMethod1() throws CheckedException {
  ...
  throw new CheckedException(message);   // Noncompliant
  ...
  throw new IllegalArgumentException(message); // Compliant; IllegalArgumentException is unchecked
}

public void myMethod2() throws CheckedException {  // Compliant; propagation allowed
  myMethod1();
}</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove this "throw" statement.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_29_jul_2013_160635_freddy_mallet_wrote">on 29 Jul 2013, 16:06:35 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>Is implemented by http://jira.codehaus.org/browse/SONARJAVA-250</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_31_jul_2013_134108_dinesh_bolkensteyn_wrote">on 31 Jul 2013, 13:41:08 Dinesh Bolkensteyn wrote:</h3>
<div class="paragraph">
<p>Not be used is a bit strong, as you&#8217;re still allowed to catch java.io.IOException for example.</p>
</div>
<div class="paragraph">
<p>It&#8217;s just that it should not be part of the API</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_19_aug_2013_073518_dinesh_bolkensteyn_wrote">on 19 Aug 2013, 07:35:18 Dinesh Bolkensteyn wrote:</h3>
<div class="paragraph">
<p>Removed 'Java' from covered languages, as we finally postponed the implementation.</p>
</div>
</div>
</div>
</div>