This rule raises an issue when an HTML meta refresh tag has a delay of less than 20 hours (72,000 seconds).

== Why is this an issue?

Automatic page refreshes can create significant accessibility barriers for users with disabilities. When pages refresh automatically with short delays, several problems arise:

**Cognitive and reading disabilities**: Users who need more time to read and understand content may lose their progress when the page refreshes unexpectedly. This can be frustrating and prevent them from accessing information effectively.

**Screen reader users**: Automatic refreshes can interrupt screen readers mid-sentence, forcing users to start over. This disrupts the reading flow and makes it difficult to consume content.

**Motor disabilities**: Users who navigate slowly using assistive devices may not have enough time to interact with page elements before the refresh occurs.

**Seizure disorders**: Sudden page changes can potentially trigger seizures in susceptible individuals.

The Web Content Accessibility Guidelines (WCAG) 2.2 Success Criterion 2.2.1 requires that timing be adjustable. Specifically, automatic refreshes should either be controllable by the user or have a delay of at least 20 hours, which is considered long enough that users won't be unexpectedly interrupted.

=== What is the potential impact?

Users with disabilities may be unable to read or interact with content before automatic refreshes occur. This can make websites completely inaccessible to some users and create a poor user experience for others. The issue violates WCAG 2.2 accessibility standards and may result in legal compliance problems.

== How to fix it

Remove automatic refresh and implement user-controlled refresh mechanisms instead. Provide buttons or links that allow users to refresh content when they choose to do so.

=== Code examples

==== Noncompliant code example

[source,html,diff-id=1,diff-type=noncompliant]
----
<meta http-equiv="refresh" content="30"> <!-- Noncompliant: refreshes every 30 seconds -->
----

==== Compliant solution

[source,html,diff-id=1,diff-type=compliant]
----
<!-- Remove automatic refresh -->
<button onclick="location.reload()">Refresh Page</button>
<!-- Or provide a manual refresh link -->
<a href="#" onclick="location.reload(); return false;">Update Content</a>
----

== Resources

=== Documentation

 * WCAG 2.2 - Timing Adjustable - https://www.w3.org/TR/WCAG22/#timing-adjustable[Official WCAG 2.2 guidelines on timing adjustability requirements]

 * Deque University - Meta Refresh Rule - https://dequeuniversity.com/rules/axe/4.10/meta-refresh[Detailed explanation of meta refresh accessibility issues and solutions]

 * MDN - Meta Element - https://developer.mozilla.org/en-US/docs/Web/HTML/Element/meta[Technical documentation for the HTML meta element and refresh attribute]

=== Standards

 * WCAG 2.2 Success Criterion 2.2.1 - Timing Adjustable - https://www.w3.org/TR/WCAG22/#timing-adjustable[Accessibility standard requiring user control over timing or delays of at least 20 hours]

=== Related rules

 * ACT-R659A - https://act-rules.github.io/rules/bc659a[ACT rule for meta refresh no delay accessibility testing]

 * ACT-BISZ58 - https://act-rules.github.io/rules/bisz58[ACT rule for meta refresh delay accessibility testing]
