include::../description.adoc[]

== Noncompliant Code Example

[source,apex]
----
private String code = 'bounteous';

public String calculateCode() {
  doTheThing();
  return code;
}

public String getName() {  // Noncompliant
  doTheThing();
  return code;
}
----

== Compliant Solution

[source,apex]
----
private String code = 'bounteous';

public String getCode() {
  doTheThing();
  return code;
}

public String getName() {
  return getCode();
}
----

include::../exceptions.adoc[]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../highlighting.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
