<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises when a <code>torch.autograd.Variable</code> is instantiated.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>The Pytorch Variable API has been deprecated. The behavior of Variables is now provided by the Pytorch tensors and can be controlled with the <code>requires_grad</code> parameter.</p>
</div>
<div class="paragraph">
<p>The Variable API now returns tensors anyway, so there should not be any breaking changes.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Replace the call to <code>torch.autograd.Variable</code> with a call to <code>torch.tensor</code> and set the <code>requires_grad</code> attribute to <code>True</code> if needed.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">import torch

x = torch.autograd.Variable(torch.tensor([1.0]), requires_grad=True) # Noncompliant
x2 = torch.autograd.Variable(torch.tensor([1.0])) # Noncompliant</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">import torch

x = torch.tensor([1.0], requires_grad=True)
x2 = torch.tensor([1.0])</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Pytorch documentation - <a href="https://pytorch.org/docs/stable/autograd.html#variable-deprecated">Variable API</a></p>
</li>
</ul>
</div>
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Should be pretty straighforward to implement.</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Primary : Replace this call with a call to "torch.tensor".</p>
</div>
</div>
<div class="sect2">
<h3 id="_issue_location">Issue location</h3>
<div class="paragraph">
<p>Primary : Name of the function call</p>
</div>
</div>
<div class="sect2">
<h3 id="_quickfix">Quickfix</h3>
<div class="paragraph">
<p>Might be tricky to know how to call the <code>torch.tensor</code> function.
If there is an import like <code>from torch import tensor</code>, then replace with <code>tensor(&#8230;&#8203;)</code>
If not, then replace with <code>torch.tensor(&#8230;&#8203;)</code></p>
</div>
</div>
</div>
</div>