include::../description.adoc[]

== Noncompliant code example

With default provided regular expression:

[source, scala]
----
def DoSomething( ) : Unit = { // Noncompliant
   // ...
}
----

== Compliant Solution

[source, scala]
----
def doSomething( ) : Unit = {
   // ...
}
----
