include::../common/why.adoc[]

include::../common/how.adoc[]

=== Code examples

==== Noncompliant code example

[source,javascript,diff-id=1,diff-type=noncompliant]
----
function myListElement() {
    return <li role="button">Foo</li>; // Noncompliant; "li" isn't interactive, but "button" is
}
----

==== Compliant solution

[source,javascript,diff-id=1,diff-type=compliant]
----
function myListElement() {
    return <li role="listitem">Foo</li>;
}
----

include::../common/resources.adoc[]
