include::../description.adoc[]

== Noncompliant Code Example

With the default regular expression ``++^[a-z][a-zA-Z0-9]*$++``:

----
class MyClass {
  var MyField = 1
}
----

== Compliant Solution

----
class MyClass {
  var myField = 1
}
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../parameters.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
