<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This is an issue when <code>partial</code> is used as a return type without the <code>@</code> escape prefix in C# 14 or later.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Starting with C# 14, the <code>partial</code> keyword becomes a contextual keyword in more places within the language. This expansion means that <code>partial</code> can no longer be used directly as a return type identifier without escaping.</p>
</div>
<div class="paragraph">
<p>When the compiler encounters <code>partial</code> in a return type position, it now interprets it as a modifier rather than a type name. This causes compilation errors in code that previously worked in earlier C# versions.</p>
</div>
<div class="paragraph">
<p>The issue specifically affects method declarations where a custom type named <code>partial</code> is used as the return type. Other uses of the <code>partial</code> type (such as in variable declarations or constructor calls) are not affected by this change.</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="paragraph">
<p>This breaking change will cause compilation failures when upgrading existing codebases to C# 14. The compilation error prevents the application from building, requiring immediate attention to fix the syntax.</p>
</div>
<div class="paragraph">
<p>While this is not a runtime issue, it can block development and deployment processes until resolved. The fix is straightforward but must be applied to all affected method declarations.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Add the <code>@</code> prefix before <code>partial</code> when used as a return type. This tells the compiler to treat <code>partial</code> as an identifier rather than a keyword.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-csharp" data-lang="csharp">class C
{
    partial F() =&gt; new partial(); // Noncompliant
}

class partial { }</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-csharp" data-lang="csharp">class C
{
    @partial F() =&gt; new partial();
}

class partial { }</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>C# Breaking Changes in .NET 10 - <a href="https://learn.microsoft.com/en-us/dotnet/csharp/whats-new/breaking-changes/compiler%20breaking%20changes%20-%20dotnet%2010">Official Microsoft documentation about breaking changes in C# 14, including the partial keyword restrictions</a></p>
</li>
<li>
<p>C# Identifiers - <a href="https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/language-specification/lexical-structure#643-identifiers">C# language specification explaining identifier rules and escaping with the @ prefix</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>