== How to fix it in .NET

=== Code examples

include::../../common/fix/code-rationale.adoc[]

==== Noncompliant code example

[source,vbnet,diff-id=1,diff-type=noncompliant]
----
<Route("NonCompliantArrayList")>
Public Function NonCompliantArrayList() As String
    Dim size As Integer
    Try
        size = Integer.Parse(Request.Query("size"))
    Catch ex As FormatException
        Return "Number format exception while reading size"
    End Try
    Dim arrayList As New ArrayList(size) ' Noncompliant
    Return size.ToString() + " bytes were allocated."
End Function
----

==== Compliant solution

[source,vbnet,diff-id=1,diff-type=compliant]
----
Public Const MAX_ALLOC_SIZE As Integer = 1024

<Route("CompliantArrayList")>
Public Function CompliantArrayList() As String
    Dim size As Integer
    Try
        size = Integer.Parse(Request.Query("size"))
    Catch ex As FormatException
        Return "Number format exception while reading size"
    End Try
    size = Math.Min(size, MAX_ALLOC_SIZE)
    Dim arrayList As New ArrayList(size)
    Return size.ToString() + " bytes were allocated."
End Function
----

=== How does this work?

include::../../common/fix/upper-limit.adoc[]

Here, the example compliant code uses the `Math.Min` function to enforce a
reasonable upper bound to the allocation size. In that case, no more than 1024
bytes can be allocated at a time.

include::../../common/fix/environment-hardening.adoc[]