<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Dart has an <code>is</code> operator to check the type of a variable. This operator can also be used with negation <code>is!</code>. In this case we check that the type of a variable on the left side of the operator is not the one mentioned on the right side.</p>
</div>
<div class="paragraph">
<p>Similar operators also exist in other languages, but they might have different spelling. For example, in Kotlin, the same operator is written this way: <code>!is</code>. Developers familiar with both languages might confuse these 2 operators. Unfortunately, this won&#8217;t lead to a compile time error, but will change the semantic of your code.</p>
</div>
<div class="paragraph">
<p>In Dart, <code>x is! Y</code>, will return <code>true</code> if <code>x</code> is not of type <code>Y</code> and will return <code>false</code>, if <code>x</code> is of type <code>Y</code>.
On the other hand <code>x !is Y</code> will apply null assertion operator (<code>!</code>) to <code>x</code> and then return <code>true</code> if <code>x</code> is of type <code>Y</code>, and return <code>false</code> if it is not.</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="paragraph">
<p>It&#8217;s hard to estimate the impact taking into account that, firstly <code>x!</code> can throw an exception and then, the resulted check will have an absolutely opposite meaning.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>If the intention was to use <code>is!</code>, then replace <code>!is</code> with <code>is!</code>. If null assertion was used on purpose, then, this should be explicit and there must be a space between <code>!</code> and <code>is</code>.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-dart" data-lang="dart">void main(dynamic s) {

  if (s !is String) { // Noncompliant
    print('s is not a String!');
  }

}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-dart" data-lang="dart">void main(dynamic s) {

  if (s is! String) {
    print('s is not a String!');
  }

}</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Dart Docs - <a href="https://dart.dev/language/operators">Operators</a></p>
</li>
</ul>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="ulist">
<ul>
<li>
<p>Use "is!" instead of "!is".</p>
</li>
</ul>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p><code>!</code> and <code>is</code> operators</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
</div>
</div>