include::../description.adoc[]

== Noncompliant Code Example

----
if ( a == a ) { // always true
  do_z();
}
if ( a != a ) { // always false
  do_y();
}
if ( a == b && a == b ) { // if the first one is true, the second one is too
  do_x();
}
if (a == b || a == b ) { // if the first one is true, the second one is too
  do_w();
}

if (5 / 5) { // always 1
  do_v();
}
if (5 - 5) { // always 0
  do_u();
}
----

== Exceptions

The following are ignored:

* The expression ``++1 << 1++``
* When an increment or decrement operator is used, ex: ``*p{plus}{plus} == *p{plus}{plus}``
* Bitwise operators ``++|, &, ^++``

== See

* https://wiki.sei.cmu.edu/confluence/x/5dUxBQ[CERT, MSC12-C.] - Detect and remove code that has no effect or is never executed
* S1656 - Implements a check on ``++=++``. 

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../highlighting.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
