== Why is this an issue?

Using the same value on both sides of a binary operator is a code defect. In the case of logical operators, it is either a copy/paste error and, therefore, a bug, or it is simply duplicated code and should be simplified. In the case of most binary mathematical operators, having the same value on both sides of an operator yields predictable results and should be simplified as well.

=== Exceptions

The following are ignored:

* The expression `1 << 1`
* When an increment or decrement operator is used, ex: ``*p{plus}{plus} == *p{plus}{plus}``
* Bitwise operators `|, &, ^`
* Arithmetic operators `+, *`
* Assignment operators `=, +=, *=`

=== Code examples

==== Noncompliant code example

[source,cpp]
----

extern void nop();

void foo(int a, int b) {
  if ( a == a ) { // always true
    nop();
  }
  if ( a != a ) { // always false
    nop();
  }

  if ( (a == b) && (a == b) ) { // if the first one is true, the second one is too
    nop();
  }
  if ( (a == b) || (a == b) ) { // if the first one is true, the second one is too
    nop();
  }

  if ( 5 / 5 ) { // always 1
    nop();
  }
  if ( 5 - 5 ) { // always 0
    nop();
  }
}
----

== Resources

=== Standards

* https://wiki.sei.cmu.edu/confluence/x/5dUxBQ[CERT, MSC12-C.] - Detect and remove code that has no effect or is never executed

=== Related rules

* S1656 detects self-assignment

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../highlighting.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
