<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p><code>using</code> directives add additional scopes to the set of scopes searched during name lookup. All identifiers in these scopes become visible, increasing the possibility that the identifier found by the compiler does not meet developer expectations.</p>
</div>
<div class="paragraph">
<p><em>Using-declarations</em> or fully-qualified names restricts the set of names considered to only the name explicitly specified, and so these are safer options.</p>
</div>
<div class="sect2">
<h3 id="_exceptions">Exceptions</h3>
<div class="paragraph">
<p>It is not easy to fully qualify the content of the <code>std::literals</code> and <code>std::placeholders</code> namespaces. Therefore, this rule does not raise violations for <code>using</code> directives that target these namespaces or their sub-namespaces.</p>
</div>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">namespace NS1 {
  int f();
}

using namespace NS1; // Noncompliant

void g() {
  f();
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">namespace NS1 {
  int f();
}

void g() {
  NS1::f();
}

// Or

using NS1::f; // Compliant, this is a using declaration

void g() {
  f();
}</code></pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p>MISRA C&#43;&#43;:2008, 7-3-4 - using-directives shall not be used.</p>
</li>
</ul>
</div>
<hr>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_relates_to_s1003">relates to: <a data-rspec-id="S1003" class="rspec-auto-link">S1003</a></h3>

</div>
<div class="sect2">
<h3 id="_on_31_mar_2015_190703_evgeny_mandrikov_wrote">on 31 Mar 2015, 19:07:03 Evgeny Mandrikov wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] implementation seems more complete (SQALE, description) than this spec.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_13_apr_2015_191638_evgeny_mandrikov_wrote">on 13 Apr 2015, 19:16:38 Evgeny Mandrikov wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] Changed spelling of "using-directives" and "using-declarations" to match C&#43;&#43; language specification. Also wondering why blocker, but not active by default? Note that in implementation currently major and active.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_20_apr_2015_140033_ann_campbell_wrote">on 20 Apr 2015, 14:00:33 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>in fact [~evgeny.mandrikov] I don&#8217;t think I had previously edited either of those two attributes</p>
</div>
</div>
</div>
</div>