<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Starting in Java 23, comments beginning with three slashes <code>///</code> are interpreted as JavaDoc comments using Markdown syntax.</p>
</div>
<div class="paragraph">
<p>In Java 22 and earlier, comments starting with more than 2 slashes were treated as normal comments.
Accidentally writing comments with three or more slashes can lead to unintended JavaDoc being generated, when migrating to Java 23.</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="paragraph">
<p>Comments starting with three or more slashes will increase the migration cost when upgrading to Java 23 or later.
Moreover, IDE or other tools may handle such comments as JavaDoc comments if they are not aware of the Java version.</p>
</div>
</div>
<div class="sect2">
<h3 id="_exceptions">Exceptions</h3>
<div class="ulist">
<ul>
<li>
<p>The rule does not raise issues on license headers at the start of files because they will not be considered as JavaDoc comments.</p>
</li>
<li>
<p>The rule only raises issues in projects using Java 17 or above, because those projects will likely be migrated to Java 23 or later.</p>
</li>
</ul>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>In versions of Java prior to 23, comments should not start with more than 2 slashes, and from Java 23 forward they should not start with more than 3.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="paragraph">
<p>The following code will generate unintended JavaDoc comments if migrated to Java 23:</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">/// Some comment for the developers
public abstract void foo();
//// public void foo(String s){}
public void foo(){}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">// Some comment for the developers
public abstract void foo();
// public void foo(String s){}
public void foo(){}</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>OpenJDK - <a href="https://openjdk.org/jeps/467">JEP 467: Markdown Documentation Comments</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>