For optimal code readability, annotation arguments should be specified in the same order that they were declared in the annotation definition.


== Noncompliant Code Example

----
@interface Pet {
    String name();
    String surname(); 
}

@Pet(surname ="", name="") // Noncompliant
----


== Compliant Solution

----
@interface Pet {
    String name();
    String surname(); 
}

@Pet(name ="", surname="") // Compliant
----


ifdef::rspecator-view[]
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::rspecator-view[]
