== Why is this an issue?

A generator without a ``++yield++`` statement is at best confusing, and at worst a bug in your code, since the iterator produced by your code will always be empty. 


=== Noncompliant code example

[source,javascript]
----
function* myGen(a, b) {  // Noncompliant
  let answer = 0;
  answer += a * b;
}
----


=== Compliant solution

[source,javascript]
----
function* myGen(a, b) {
  let answer = 0;
  while (answer < 42) {
    answer += a * b;
    yield answer;
  }
}
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

include::highlighting.adoc[]

'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
