<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>When accessing a non-input-only file with a <code>CHAIN</code>/<code>READx</code> operation, the record is locked and becomes inaccessible to the system, and unreadable other programs. This is fine if you are about to update the record immediately, but if you have data processing to do following the read, it can leave the record unavailable for too long. Therefore, non-input files using the <code>CHAIN</code>, <code>READ</code>, <code>READE</code>, <code>READP</code>, and <code>READPE</code> operation codes should always be accessed with the no lock option, <code>(N)</code>.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-rpg" data-lang="rpg">FDDDLJNL7  UF   E           K DISK    INFSR(*PSSR)
...
C                   READ      DDDLJNL7</code></pre>
</div>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-rpg" data-lang="rpg">FDDDLJNL7  UF   E           K DISK    INFSR(*PSSR)
...
 /free
    read DDDLJNL7;
 /end-free</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-rpg" data-lang="rpg">FDDDLJNL7  UF   E           K DISK    INFSR(*PSSR)
...
C                   READ(N)   DDDLJNL7</code></pre>
</div>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-rpg" data-lang="rpg">FDDDLJNL7  UF   E           K DISK    INFSR(*PSSR)
...
 /free
    read(n) DDDLJNL7;
 /end-free</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Add the "N" option to this file access.</p>
</div>
</div>
</div>
</div>