<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Labeled blocks are useful, especially when the code is badly indented, to match the begin and end of each block. This rule verifies that block start and end labels match, when both are specified.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-sql" data-lang="sql">BEGIN
  NULL;
END; -- Compliant, no labels at all
/

&lt;&lt;myBlockLabel1&gt;&gt;
BEGIN
  NULL;
END; -- Compliant, only starting label
/

BEGIN
  NULL;
END myBlockLabel2; -- Compliant, only ending label
/

&lt;&lt;myBlockLabel3&gt;&gt;
BEGIN
  NULL;
END myBlockLabel4; -- Noncompliant, labels mismatch
/

&lt;&lt;myBlockLabel6&gt;&gt;
&lt;&lt;myBlockLabel6&gt;&gt;
BEGIN
  NULL;
END myBlockLabel6; -- Noncompliant, several starting labels
/</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-sql" data-lang="sql">BEGIN
  NULL;
END;
/

&lt;&lt;myBlockLabel1&gt;&gt;
BEGIN
  NULL;
END;
/

BEGIN
  NULL;
END myBlockLabel2;
/

&lt;&lt;myBlockLabel3&gt;&gt;
BEGIN
  NULL;
END myBlockLabel3;
/

&lt;&lt;myBlockLabel6&gt;&gt;
BEGIN
  NULL;
END myBlockLabel6;
/</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Change this label to "xxx".</p>
</div>
</div>
</div>
</div>