== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,sql]
----
DECLARE
  -- The following line has many trailing whitespaces
  foo VARCHAR2(42) := 'a     
b';
BEGIN
  -- Will misleadingly show 3, counting only the characters 'a', 'b', and the line terminator, but none of the trailing whitespaces
  DBMS_OUTPUT.PUT_LINE(LENGTH(foo));
END;
/
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
