The W3C specifications define the valid media features. Only the official and browser-specific media features should be used to get the expected impact in the final rendering.


== Noncompliant Code Example

----
@media screen and (unknown: 1000px) { .. }
----


== Compliant Solution

----
@media screen and (width: 1000px) { .. }
----


== See

* https://www.w3schools.com/cssref/css3_pr_mediaquery.asp[CSS Specification] - ``++@media++`` rule

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

endif::env-github,rspecator-view[]
