== Why is this an issue?

The W3C specifications define the valid media features. Only the official and browser-specific media features should be used to get the expected impact in the final rendering.


=== Noncompliant code example

[source,css]
----
@media screen and (unknown: 1000px) { .. }
----


=== Compliant solution

[source,css]
----
@media screen and (width: 1000px) { .. }
----


== Resources

* https://www.w3schools.com/cssref/css3_pr_mediaquery.asp[CSS Specification] - ``++@media++`` rule

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Replace XXX by a valid media feature.


endif::env-github,rspecator-view[]
