== Why is this an issue?

It is not common practice to use a logical expression as an array index, the user probably made a typo and misplaced the closing square bracket. This rule flags all array subscriptions where the index is a logical expression.


=== Noncompliant code example

[source,cpp]
----
void f(const int arr[], int a) {
  if (arr[a > 10]) { // Noncompliant, shouldn't it be 'arr[a] > 10' instead?
    // ...
  }
}
----


=== Compliant solution

[source,cpp]
----
void f(const int arr[], int a) {
  if (arr[a] > 10) {
    // ...
  }
}
----


=== Exceptions

Do not raise an issue when the array size is statically known and equal to 2.  

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Review the array subscription index, the closing bracket is probably misplaced.


endif::env-github,rspecator-view[]
