It is not common practice to use a logical expression as an array index, the user probably made a typo and misplaced the closing square bracket. This rule flags all array subscriptions where the index is a logical expression.

== Noncompliant Code Example

----
void f(const int arr[], int a) {
  if (arr[a > 10]) { // Noncompliant, shouldn't it be 'arr[a] > 10' instead?
    // ...
  }
}
----

== Compliant Solution

----
void f(const int arr[], int a) {
  if (arr[a] > 10) {
    // ...
  }
}
----

== Exceptions

Do not raise an issue when the array size is statically known and equal to 2.  
