== How to fix it in Micronaut

=== Code examples

include::../../common/fix/code-rationale.adoc[]

==== Noncompliant code example

[source,java,diff-id=11,diff-type=noncompliant]
----
@Get("/example")
public Object example(@QueryValue @Nonnull String tainted) {
    HttpClient client = HttpClient.create(null);

    HttpRequest request = HttpRequest.GET(tainted);  // Noncompliant

    return client.exchange(request, byte[].class);
}
----

==== Compliant solution

[source,java,diff-id=11,diff-type=compliant]
----
@Get("/example")
public Object example(@QueryValue @Nonnull String tainted) {
    List<String> allowedUrls = new ArrayList<String>();
    allowedUrls.add("https://trusted.example.com");
    allowedUrls.add("https://trusted.example.com:443");

    if (!allowedUrls.contains(tainted)) {
        return HttpResponse.unauthorized();
    }

    HttpClient client = HttpClient.create(null);

    HttpRequest request = HttpRequest.GET(tainted);

    return client.exchange(request, byte[].class);
}
----

=== How does this work?

include::../../common/fix/pre-approved-list.adoc[]

=== Pitfalls

include::../../common/pitfalls/starts-with.adoc[]


