<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>The order in which you <code>close</code> database-releated resources is crucial. Close a <code>Connection</code> first, and depending on the database pooling in use, you may no longer be able to truly reach its <code>Statement</code>s and <code>ResultSet</code>s to close them, even though the calls are made and execute without error.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">Connection conn = null;
PreparedStatement ps = null;
ResultSet rs = null;
try {
  conn = DriverManager.getConnection(connectionString);
  ps = conn.prepareStatement(query);
  rs = ps.executeQuery();
  // ...
} finally {
  try {
    if (conn != null) {
      conn.close();   // Noncompliant; close this last
    }
  } catch (Exception e) {};
  try {
    if (ps != null) {
      ps.close();
    }
  } catch (Exception e) {};
  try {
    if (rs != null) {
      rs.close();
    }
  } catch (Exception e) {};
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">Connection conn = null;
PreparedStatement ps = null;
ResultSet rs = null;
try {
  conn = DriverManager.getConnection(connectionString);
  ps = conn.prepareStatement(query);
  rs = ps.executeQuery();
  // ...
} finally {
  try {
    if (rs != null) {
      rs.close();
    }
  } catch (Exception e) {};
  try {
    if (ps != null) {
      ps.close();
    }
  } catch (Exception e) {};
  try {
    if (conn != null) {
      conn.close();
    }
  } catch (Exception e) {};
}</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Close this "Connection" last.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_16_jun_2015_172824_ann_campbell_wrote">on 16 Jun 2015, 17:28:24 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>CodePro: Close Order</p>
</div>
</div>
</div>
</div>