<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Multiplying the number of <code>RUN</code> instructions increases the final image build time and size.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Each time a <code>RUN</code> instruction is added, a new layer is introduced in the final image.<br>
This has a direct impact on the build time and image size. Chaining commands in a single <code>RUN</code> instruction using <code>&amp;&amp;</code> will use a single layer, thereby reducing the number of layers in the image.<br>
This practice can make Docker images more efficient and easier to manage.<br>
Each layer in a Docker image is essentially a change to the image, like a version control system.<br>
So, fewer layers mean fewer changes to track, which simplifies management and troubleshooting.</p>
</div>
<div class="sect2">
<h3 id="_exceptions">Exceptions</h3>
<div class="paragraph">
<p>In multi-stage builds, the rule only scans instructions that are part of the final image.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-docker" data-lang="docker">RUN curl -SL "https://nodejs.org/dist/v${NODE_VERSION}/node-v${NODE_VERSION}-linux-x64.tar.gz" --output nodejs.tar.gz
RUN echo "$NODE_DOWNLOAD_SHA nodejs.tar.gz" | sha256sum -c -
RUN tar -xzf "nodejs.tar.gz" -C /usr/local --strip-components=1
RUN rm nodejs.tar.gz
RUN ln -s /usr/local/bin/node /usr/local/bin/nodejs</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-docker" data-lang="docker">RUN curl -SL "https://nodejs.org/dist/v${NODE_VERSION}/node-v${NODE_VERSION}-linux-x64.tar.gz" --output nodejs.tar.gz \
&amp;&amp; echo "$NODE_DOWNLOAD_SHA nodejs.tar.gz" | sha256sum -c - \
&amp;&amp; tar -xzf "nodejs.tar.gz" -C /usr/local --strip-components=1 \
&amp;&amp; rm nodejs.tar.gz \
&amp;&amp; ln -s /usr/local/bin/node /usr/local/bin/nodejs</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Dockerfile Best Practices - <a href="https://github.com/dnaprawa/dockerfile-best-practices?tab=readme-ov-file#limit-image-layers-amount">Limit image layers amount</a></p>
</li>
</ul>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Primary locations: Merge this <code>RUN</code> instruction with the consecutive ones.
Secondary locations: consecutive <code>RUN</code> instruction</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>Highlight only the keywords <code>RUN</code> of each consecutive instruction. The primary location is the first <code>RUN</code> instruction, and the secondary locations are the other one followings.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
</div>
</div>