<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p><code>Optional</code> value can hold either a value or not. The value held in the <code>Optional</code> can be accessed using the <code>get()</code> method, but it will throw a</p>
</div>
<div class="paragraph">
<p><code>NoSuchElementException</code> if there is no value present. To avoid the exception, calling the <code>isPresent()</code> or <code>! isEmpty()</code> method should always be done before any call to <code>get()</code>.</p>
</div>
<div class="paragraph">
<p>Alternatively, note that other methods such as <code>orElse(...)</code>, <code>orElseGet(...)</code> or <code>orElseThrow(...)</code> can be used to specify what to do with an empty <code>Optional</code>.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">public static Optional&lt;String&gt; getOptionalValue(boolean condition) {
  if (condition) {
    return Optional.of("Value text");
  }
  return Optional.empty();
}

public void example(boolean condition) {
  Optional&lt;String&gt; optional = getOptionalValue(condition);
  String value = optional.get(); // Noncompliant; getOptionalValue may return an empty optional
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">public static Optional&lt;String&gt; getOptionalValue(boolean condition) {
  if (condition) {
    return Optional.of("Value text");
  }
  return Optional.empty();
}

public void example(boolean condition) {
  getOptionalValue(condition).ifPresent(value -&gt; {
    // Do something with the value
  });
}</code></pre>
</div>
</div>
<div class="paragraph">
<p>or</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">Optional&lt;String&gt; value = this.getOptionalValue();

// ...

if (value.isPresent()) {
  String stringValue = value.get();
}</code></pre>
</div>
</div>
<div class="paragraph">
<p>or</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">Optional&lt;String&gt; value = this.getOptionalValue();

// ...

String stringValue = value.orElse("default");</code></pre>
</div>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">Optional&lt;String&gt; optional = methodThatReturnsOptional();
if (optional.isEmpty()) {
  throw new NotFoundException();
}
String value = optional.get();</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>call "xxx.isPresent()" before accessing the value.</p>
</div>
<hr>
</div>
</div>
</div>