<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Sometimes, you might want to make a thread run indefinitely in the background by not binding it to its creation scope. Even though calling <code>detach()</code> on an <code>std::thread</code> or <code>std::jthread</code> object would satisfy this need, it is not the easiest way to do it: there will be no direct way to monitor and communicate with the detached thread, the <code>std::thread</code> or <code>std::jthread</code> object is no longer associated to any thread.</p>
</div>
<div class="paragraph">
<p>An easier alternative to satisfy this need is giving the thread a global scope. This way the thread will run as long as the program does. The thread will not be bound to any scope. It is also possible to do it by giving the <code>std::thread</code> or <code>std::jthread</code> a scope that is big enough for your use case. For example, the program&#8217;s main function.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">void backgroundTask();
void startBackgroundTask(){
  // Assume you want the thread to run after the end of startBackgroundTask
  std::jthread backgroundThread(backgroundTask);
  backgroundThread.detach(); // Noncompliant
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">void  backgroundTask();
std::jthread backgroundThread;

void startBackgroundTask(){
  // Assume you want the thread to run after the end of startBackgroundTask
  backgroundThread = std::move(std::jthread{backgroundTask});
}</code></pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p>C&#43;&#43; Core Guidelines - <a href="https://github.com/isocpp/CppCoreGuidelines/blob/e49158a/CppCoreGuidelines.md#cp26-dont-detach-a-thread">CP.26: Don&#8217;t <code>detach()</code> a thread</a></p>
</li>
</ul>
</div>
</div>
</div>