== Why is this an issue?

ï»¿Creating an extension method that extends ``++object++`` is not recommended because it makes the method available on _every_ type. Extensions should be applied at the most specialized level possible, and that is very unlikely to be ``++object++``.

=== Noncompliant code example

[source,csharp]
----
public static class MyExtensions
{
    public static void SomeExtension(this object obj) // Noncompliant
    {
        // ...
    }
}
----

include::../rspecator.adoc[]