<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This is an issue when documentation region tags like <code>[START &#8230;&#8203;]</code> and <code>[END &#8230;&#8203;]</code> are not properly paired with matching identifiers, are incorrectly positioned, or don&#8217;t follow consistent naming conventions with product prefixes.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Documentation region tags are special comments used by documentation extraction tools to identify and extract specific code sections for inclusion in documentation. These tools rely on properly formatted and paired tags to function correctly.</p>
</div>
<div class="paragraph">
<p>When region tags are not properly paired with matching identifiers, documentation tools cannot determine where a code section begins or ends. This can result in incomplete or incorrect code samples being extracted for documentation.</p>
</div>
<div class="paragraph">
<p>Incorrect positioning of END tags within code blocks instead of after the closing brace means the extracted code sample will be incomplete, missing important parts of the logical unit.</p>
</div>
<div class="paragraph">
<p>Without consistent naming conventions using product or service prefixes, it becomes difficult to organize and identify which product or service a code region belongs to, especially in large codebases with multiple products or services.</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="paragraph">
<p>Improperly formatted region tags can cause documentation generation tools to fail or produce incorrect documentation. This leads to incomplete or misleading code examples in documentation, which can confuse developers and reduce the quality of technical documentation.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Ensure START and END tags have matching identifiers and are properly positioned around the complete code block.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-go" data-lang="go">// Some code here
// [END example_function] // Noncompliant: missing START tag
func ExampleFunction() {
    // implementation
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-go" data-lang="go">// [START product_example_function]
// Some code here
func ExampleFunction() {
    // implementation
}
// [END product_example_function]</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Go Code Review Comments - <a href="https://github.com/golang/go/wiki/CodeReviewComments">Official Go code review guidelines including comment conventions</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>