<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>A regex should never include a repetitive pattern whose body would match the empty string. This is usually a sign that a part of the regex is redundant or does not do what the author intended.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-php" data-lang="php">"/(?:)*/"      // same as the empty regex, the '*' accomplishes nothing
"/(?:|x)*/"    // same as the empty regex, the alternative has no effect
"/(?:x|)*/"    // same as 'x*', the empty alternative has no effect
"/(?:x*|y*)*/" // same as 'x*', the first alternative would always match, y* is never tried
"/(?:x?)*/"    // same as 'x*'
"/(?:x?)+/"    // same as 'x*'</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-php" data-lang="php">"/x*/"</code></pre>
</div>
</div>
</div>
</div>
</div>