== Why is this an issue?

include::../description.adoc[]

== How to fix it

include::../how-to-fix.adoc[]

=== Code examples

==== Noncompliant code example

[source,java,diff-id=1,diff-type=noncompliant]
----
public long computeDurationInMilliseconds() {
  long duration = (((hours * 60) + minutes) * 60 + seconds) * 1000;
  return duration;
}
----

==== Compliant solution

[source,java,diff-id=1,diff-type=compliant]
----
public long computeDurationInMilliseconds() {
  return (((hours * 60) + minutes) * 60 + seconds) * 1000;
}
----

==== Noncompliant code example

[source,java,diff-id=2,diff-type=noncompliant]
----
public void doSomething() {
  RuntimeException myException = new RuntimeException();
  throw myException;
}
----

==== Compliant solution

[source,java,diff-id=2,diff-type=compliant]
----
public void doSomething() {
  throw new RuntimeException();
}
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
