<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when multiple controller files define classes with identical names in different directories within a Rails application.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Rails uses an autoloading mechanism that maps file paths to class names. When multiple controller files define classes with the same name in different directories, Rails cannot determine which file to load when the class is referenced.</p>
</div>
<div class="paragraph">
<p>This creates several problems:</p>
</div>
<div class="ulist">
<ul>
<li>
<p><strong>Circular dependency errors</strong>: Rails may attempt to load both files simultaneously, creating a circular reference</p>
</li>
<li>
<p><strong>Unpredictable behavior</strong>: The application may load different controller classes depending on the order of file loading</p>
</li>
<li>
<p><strong>Runtime failures</strong>: The application may crash with autoloading errors, especially in production environments</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>Rails expects each class to have a unique fully-qualified name. When controllers are organized in subdirectories, they should use proper namespacing to reflect their directory structure. This ensures that each controller has a distinct identity and can be loaded reliably.</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="paragraph">
<p>The application may experience runtime errors, unpredictable behavior, or complete failure to start. In production environments, this can lead to service outages and difficult-to-debug issues.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Use proper namespacing for controllers in subdirectories. The namespace should match the directory structure to create unique class names.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-ruby" data-lang="ruby"># app/controllers/logs_controller.rb
class LogsController &lt; ApplicationController
  def index
    # Controller logic here
  end
end

# app/controllers/api/logs_controller.rb
class LogsController &lt; ApplicationController # Noncompliant
  def index
    # This creates a circular dependency issue
  end
end</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-ruby" data-lang="ruby"># app/controllers/logs_controller.rb
class LogsController &lt; ApplicationController
  def index
    # Controller logic here
  end
end

# app/controllers/api/logs_controller.rb
class Api::LogsController &lt; ApplicationController
  def index
    # API-specific controller logic here
  end
end</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Rails Autoloading and Reloading Constants - <a href="https://guides.rubyonrails.org/autoloading_and_reloading_constants.html">Official Rails guide explaining how autoloading works and naming conventions</a></p>
</li>
<li>
<p>Rails Routing from the Outside In - <a href="https://guides.rubyonrails.org/routing.html#controller-namespaces-and-routing">Rails guide covering controller namespaces and routing</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>