== Why is this an issue?

Sharing some naming conventions enables teams to collaborate more efficiently. This rule checks that all ``++struct++`` names match a provided regular expression.


=== Noncompliant code example

Using the default regular expression ``++^[A-Z][a-zA-Z0-9]*$++``:

[source,cpp]
----
struct myStruct {
  int one;
  int two;
};
----


=== Compliant solution

[source,cpp]
----
struct MyStruct {
  int one;
  int two;
};
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../parameters.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
