== Why is this an issue?

In Spring applications, application components that expose interfaces should be package protected at most, not ``++public++``. Such reduced visibility helps ensure that the interface is only accessed through the container and not directly. 


=== Noncompliant code example

[source,java]
----
// TODO
----


=== Compliant solution

[source,java]
----
// TODO
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Make this class package protected if possible.


endif::env-github,rspecator-view[]
