In Spring applications, application components that expose interfaces should be package protected at most, not ``++public++``. Such reduced visibility helps ensure that the interface is only accessed through the container and not directly. 


== Noncompliant Code Example

----
// TODO
----


== Compliant Solution

----
// TODO
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

endif::env-github,rspecator-view[]
