include::../common/why.adoc[]

include::../common/how.adoc[]

=== Code examples

==== Noncompliant code example

[source,html,diff-id=1,diff-type=noncompliant]
----
<li onClick="changeText()" />
<div onClick="changeText()" role="listitem" />
----

==== Compliant solution

[source,html,diff-id=1,diff-type=compliant]
----
<div onClick="changeText()" role="button" />
<div onClick="changeText()" role="presentation" />
<input type="text" onClick=changeText() /> <!-- Interactive element does not require role. -->
<button onClick=changeText() className="foo" /> <!-- button is interactive. -->
<div onClick=changeText() role="button" aria-hidden /> <!-- This is hidden from the screenreader. -->
----

include::../common/resources.adoc[]
