include::../description.adoc[]

== Noncompliant Code Example

----
for i in 1...10 { // noncompliant, loop only executes once
  print("i is \(i)") 
  break
}
----

== Compliant Solution

----
for i in 1...10 {
  print("i is \(i)")
}
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../highlighting.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
