include::../description.adoc[]

== Noncompliant Code Example

----
r"(?=a)b" # Noncompliant, the same character can't be equal to 'a' and 'b' at the same time
----


== Compliant Solution

----
r"(?<=a)b"
r"a(?=b)"
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../highlighting.adoc[]

endif::env-github,rspecator-view[]
