== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,python]
----
r"(?=a)b" # Noncompliant, the same character can't be equal to 'a' and 'b' at the same time
----


=== Compliant solution

[source,python]
----
r"(?<=a)b"
r"a(?=b)"
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../highlighting.adoc[]

endif::env-github,rspecator-view[]
