include::../description.adoc[]

== Noncompliant Code Example

With a maximum number of 4 parameters:

----
def do_something(param1, param2, param3, param4, param5):
	...
----

== Compliant Solution

----
def do_something(param1, param2, param3, param4):
	...
----

== Exceptions

The first argument of non-static methods, i.e. ``++self++`` or ``++cls++``, is not counted as it is mandatory and it is passed automatically.

ifdef::env-github,rspecator-view[]
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
