<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Variables of type <code>void</code> should not be assigned a value.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Unlike other languages where <code>void</code> is the absence of a type (and related value), <code>void</code> in Dart is a type, which is not compatible with any other type, since it is conceptually a supertype of <code>Object</code>.</p>
</div>
<div class="paragraph">
<p>Therefore, in Dart it&#8217;s possible to write the following code:</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-dart" data-lang="dart">void x = 3; // A variable of type void can be assigned a value of type int
void f(void x) =&gt; x; // A function that takes a void input argument and returns it as output</code></pre>
</div>
</div>
<div class="paragraph">
<p>The variable <code>x</code> declared above, and the result of <code>f(x)</code> are effectively of type <code>void</code>. However, while <code>void</code> variables can be declared, initialized and even computed, they cannot be used by any function that expects a non-<code>void</code> type, by the very definition of <code>void</code> type.</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-dart" data-lang="dart">print(x); // Error: This expression has type 'void' and can't be used.
print(f(x)); // Error: This expression has type 'void' and can't be used.</code></pre>
</div>
</div>
<div class="paragraph">
<p>Therefore, assigning a value to a variable of type <code>void</code> is almost certainly a mistake.</p>
</div>
<div class="paragraph">
<p>For example, in the cases where the type is not known, the variable should have not been declared as <code>void</code>, but as <code>dynamic</code>:</p>
</div>
<div class="ulist">
<ul>
<li>
<p><code>void</code> variables can be assigned values of any type, but cannot be assigned to any non-<code>void</code> variable</p>
</li>
<li>
<p><code>dynamic</code> variables, like <code>void</code> ones, can be assigned values of any type, and, unlike <code>void</code> variables, can be assigned to any non-<code>dynamic</code> variable</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>Notice that, because <code>void</code> is a type, it can be used as a valid type parameter for a generic class, and that can lead to indirect or "hidden" <code>void</code> assignments:</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-dart" data-lang="dart">class Wrapper&lt;T&gt; {
  T value;
  Wrapper(this.value);
}

void main() {
    Wrapper&lt;void&gt; w = Wrapper(3); // This is a "hidden" assignment to void
    w.value = 3; // This is an explicit assignment to a field of type void
}</code></pre>
</div>
</div>
<div class="paragraph">
<p>Another example of a "hidden" assignment to <code>void</code> is when assigning generic unconstrained type parameters in generic methods:</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-dart" data-lang="dart">void f&lt;T&gt;(T x, T y) {
  y = x; // This is a "hidden" assignment to void
  T z;
  z = x; // This too, is a "hidden" assignment to void
}</code></pre>
</div>
</div>
<div class="paragraph">
<p>However, such scenarios are not reported by the rule because for any non-<code>void</code> type <code>T</code>, the assignment is valid. On the other hand, calling the function <code>f</code> with anything other than <code>void</code> parameters will be reported as an issue:</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-dart" data-lang="dart">f(42, 'a string'); // Both 1st and 2nd parameters are non compliant</code></pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Identify the scenario leading to the assignment of a value to a variable of type <code>void</code> and correct it:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>if the type of the variable is not known, declare it as <code>dynamic</code></p>
</li>
<li>
<p>if the type of the variable was mistakenly declared as <code>void</code>, replace it with the correct type</p>
</li>
</ul>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-dart" data-lang="dart">void x;
x = 3; // Non compliant</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-dart" data-lang="dart">dynamic x; // type of x not known
x = 3;</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_noncompliant_code_example_2">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-dart" data-lang="dart">void x;
x = 3; // Non compliant</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution_2">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-dart" data-lang="dart">int x; // void mistakenly used instead of the actual type
x = 3;</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_noncompliant_code_example_3">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-dart" data-lang="dart">Wrapper&lt;void&gt; w = Wrapper(3); // Non compliant
w.value = 3; // Non compliant</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution_3">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-dart" data-lang="dart">Wrapper&lt;int&gt; w = Wrapper(3);
w.value = 3;</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Dart Docs - <a href="https://dart.dev/tools/linter-rules/void_checks">Dart Linter rule - void_checks</a></p>
</li>
<li>
<p>Medium - <a href="https://medium.com/dartlang/dart-2-legacy-of-the-void-e7afb5f44df0">Dart 2: Legacy of the <code>void</code></a></p>
</li>
<li>
<p>Medium - <a href="https://medium.com/flutter-community/the-curious-case-of-void-in-dart-f0535705e529">The curious case of void in Dart</a></p>
</li>
</ul>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Assignment to a variable of type 'void'.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>The entire assignment expression: e.g. <code>x = 3</code> in <code>x = 3;</code>.</p>
</div>
</div>
</div>
</div>