In a file mixing PHP and HTML, all PHP tags should be closed. Failure to close a tag in this type of file could lead to unexpected output when the file is processed.


== Noncompliant Code Example

----
<?=   // Noncompliant; never closed
$name = "George";

<p> Hello <?= $name ?>.</p>
----


== Compliant Solution

----
<?=
$name = "George";
?>

<p> Hello <?= $name ?>.</p>
----


== Exceptions

This rule does not apply to files that contain only a single opening tag and no inline HTML.


== See

* Related: S1780 - Closing tag "?>" should be omitted on files containing only PHP

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

endif::env-github,rspecator-view[]
