== Why is this an issue?

In a file mixing PHP and HTML, all PHP tags should be closed. Failure to close a tag in this type of file could lead to unexpected output when the file is processed.


=== Noncompliant code example

[source,php]
----
<?=   // Noncompliant; never closed
$name = "George";

<p> Hello <?= $name ?>.</p>
----


=== Compliant solution

[source,php]
----
<?=
$name = "George";
?>

<p> Hello <?= $name ?>.</p>
----


=== Exceptions

This rule does not apply to files that contain only a single opening tag and no inline HTML.


== Resources

* Related: S1780 - Closing tag "?>" should be omitted on files containing only PHP

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Add the missing "?>" to close this tag.


endif::env-github,rspecator-view[]
