<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when the same value is used multiple times when instantiating a set literal.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>By definition, a set cannot hold the same value multiple times.
When instantiating a set literal with the same value repeated multiple times,
only the last occurrence of the duplicated value will remain.</p>
</div>
<div class="paragraph">
<p>Creating a set with redundant elements is prone to errors and confusion. A duplicated value in a set literal should be either:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>modified, as it was mistakenly put in the set instead an actual value which would lead to bugs and errors further in the program.</p>
</li>
<li>
<p>removed, as it was a simple duplication, making the code confusing and difficult to maintain.</p>
</li>
</ul>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-text" data-lang="text">{"one", "two", "one"}  # Noncompliant: the value "one" is duplicated.

def func(a1, a2, a3):
    {a1, a2, a1}  # Noncompliant: the value a1 is duplicated.</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-text" data-lang="text">{"one", "two", "three"}

def func(a1, a2, a3):
    {a1, a2, a3}</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Python Documentation - <a href="https://docs.python.org/3/reference/expressions.html#set-displays">Set displays</a></p>
</li>
</ul>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Change or remove duplicates of value "XXX"</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>Primary: the first occurence of a duplicated value.</p>
</div>
<div class="paragraph">
<p>Secondary:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>location: other occurences of the value.</p>
</li>
<li>
<p>message: "Duplicate value"</p>
</li>
</ul>
</div>
</div>
</div>
</div>