This rule raises an issue when the same value is used multiple times when instantiating a set literal.

== Why is this an issue?

By definition a set cannot hold two identical values. 
When instantiationg a set litteral with the same value repeated multiple times, 
only the last occurence of the duplicated value will remain. 
Duplicated values should be either modified or removed.

=== Noncompliant code example

[source,text,diff-id=1,diff-type=noncompliant]
----
{"one", "two", "one"}  # Noncompliant: the value "one" is duplicated.

def func(a1, a2, a3):
    {a1, a2, a1}  # Noncompliant: the value a1 is duplicated.
----

=== Compliant solution

[source,text,diff-id=1,diff-type=compliant]
----
{"one", "two", "three"}

def func(a1, a2, a3):
    {a1, a2, a3}
----


== Resources

=== Documentation

* Python Documentation - https://docs.python.org/3/reference/expressions.html#set-displays[Set displays]

