== Why is this an issue?

A set cannot have two identical values. When a value is repeated in a set literal, only the last occurence will remain. Thus duplicate values should be either modified or removed.


This rule raises an issue when the same value is used multiple times as a value in a set literal.


=== Noncompliant code example

[source,text]
----
{"one", "two", "one"}  # Noncompliant

def func(a1, a2, a3):
    {a1, a2, a1}  # Noncompliant.
----


=== Compliant solution

[source,text]
----
{"one", "two", "three"}

def func(a1, a2, a3):
    {a1, a2, a3}
----


== Resources

* https://docs.python.org/3/reference/expressions.html#set-displays[Python documentation - Set displays]

