include::../description.adoc[]

== Noncompliant Code Example

----
MOVE '1' TO OCTET-1 MOVE '2' TO OCTET-2
----

== Compliant Solution

----
MOVE '1' TO OCTET-1 
MOVE '2' TO OCTET-2 
----

== Exceptions

Control flow statements with a single nested statement are ignored.


----
IF x > 0 THEN DISPLAY "positive". *> Compliant
----

ifdef::env-github,rspecator-view[]
'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
