<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>TODO and FIXME comments are typically intended to be short-lived; they are placeholders and reminders that programmers leave for themselves. Unfortunately, even with the best of intentions, those comments are not always acted on and removed in a timely manner. Thus, they can become mysterious, lingering cruft in a code base, reducing both readability and understand-ability.</p>
</div>
<div class="paragraph">
<p>This rule flags all FIXME and TODO comments that do not have an attribution matching the specified regular expression immediately after the FIXME or TODO. Ideally, such comments will also contain information about what needs to be fixed or done, but this rule does not enforce that.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python"># TODO</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python"># TODO(ganncamp) per the business partners, more checks needed</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Add a citation of the person who can best explain this comment.</p>
</div>
</div>
<div class="sect2">
<h3 id="_parameters">Parameters</h3>
<div class="sidebarblock">
<div class="content">
<div class="title">pattern</div>
<div class="listingblock">
<div class="content">
<pre>[ ]*\([ _a-zA-Z0-9@.]+\)</pre>
</div>
</div>
<div class="paragraph">
<p>A regular expression defining the pattern that should be present after "TODO" or "FIXME"</p>
</div>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_2_may_2014_152627_massimo_paladin_wrote">on 2 May 2014, 15:26:27 Massimo PALADIN wrote:</h3>
<div class="paragraph">
<p>@ann are you sure about <code>[ \t]</code>? I don&#8217;t really like tab characters on source code. Also, are sure about <code>*</code>? Wouldn&#8217;t it be better <code>?</code>.</p>
</div>
<div class="paragraph">
<p>IMHO the following is not really nice:</p>
</div>
<div class="listingblock">
<div class="content">
<pre>// TODO                                 (author)   should be done</pre>
</div>
</div>
<div class="paragraph">
<p>I would propose:</p>
</div>
<div class="listingblock">
<div class="content">
<pre>[ ]?\([ _a-zA-Z0-9@.]+\)</pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_on_2_may_2014_153333_ann_campbell_wrote">on 2 May 2014, 15:33:33 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~massimo.paladin] I buy your argument on the tab char but would like to allow multiple spaces, so let&#8217;s compromise on</p>
</div>
<div class="listingblock">
<div class="content">
<pre>[ ]*\([ _a-zA-Z0-9@.]+\)</pre>
</div>
</div>
</div>
</div>
</div>