== Why is this an issue?

include::description.adoc[]

=== Noncompliant code example

Using the default regular expression: ``++[ ]*\([ _a-zA-Z0-9@.]+\)++``:

[source,text]
----
// TODO 
----


=== Compliant solution

[source,text]
----
// TODO(ganncamp) per the business partners, more checks needed
----

