<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Bit fields allow the developers to declare a class member with a specific size.</p>
</div>
<div class="paragraph">
<p>However, the size of a bit field is also constrained by its type: even if the specified size is greater than the size of the type, the value of the bit field will not exceed the maximum value of this type. The extra bits will just create unused padding.</p>
</div>
<div class="paragraph">
<p>The incompatibility of the size of the type with the specified size can have two causes: either the specified size is a typo error (that is the most probable cause) or the developer did not realize the size of the type he chose was too small.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">class A {
  unsigned int b : 55; // Noncompliant, specified size is greater than the size of unsigned int
};</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">class A {
  unsigned int b : 32;
};</code></pre>
</div>
</div>
<div class="paragraph">
<p>Or</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">class A {
  unsigned long long int b : 55;
};</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_6_nov_2019_112548_nicolas_harraudeau_wrote">on 6 Nov 2019, 11:25:48 Nicolas Harraudeau wrote:</h3>
<div class="paragraph">
<p>\[~loic.joly] Shouldn&#8217;t this rule be a bug? It seems to me that a scientist setting the wrong bit field size would get unexpected truncations.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_16_dec_2019_111831_loïc_joly_wrote">on 16 Dec 2019, 11:18:31 Loïc Joly wrote:</h3>
<div class="paragraph">
<p>Scientists should not work with bit fields :)</p>
</div>
<div class="paragraph">
<p>Anyways, this looks like a bug, but we may have issues with detection of the runtime environment, maybe leading to false positives&#8230;&#8203;</p>
</div>
<div class="paragraph">
<p>Got to think more about this&#8230;&#8203;</p>
</div>
</div>
</div>
</div>