Always put snippetâ€™s attribute at the end of a line of code. Not doing so can lead to unexpected outputs (i.e. extra empty line could appear).

== Noncompliant Code Example

[source,java]
----
FIXME
----

== Compliant Solution

[source,java]
----
FIXME
----

== See

FIXME: A list of links
