Always put snippet markup tags at the end of line. Not doing so can lead to unexpected rendering of the snippet (e.g. extra empty line could appear).

== Noncompliant Code Example

[source,java]
----
FIXME
----

== Compliant Solution

[source,java]
----
FIXME
----

== See

FIXME: A list of links
