In javadoc comments, always put snippet markups at the end of line. Not doing so can lead to unexpected output (e.g. extra empty line could appear).


== Noncompliant Code Example ==
[source,java]
----
/**
 * A simple program.
 * {@snippet :
 * class HelloWorld {
 *     public static void main(String... args) {
 *         // @highlight substring="println"     // Noncompliant
 *         System.out.println("Hello World!");
 *     }
 * }
 * }
 */

----



== Compliant Solution ==
[source,java]
----
/**
 * A simple program.
 * {@snippet :
 * class HelloWorld {
 *     public static void main(String... args) {
 *         System.out.println("Hello World!");   // @highlight substring="println"
 *     }
 * }
 * }
 */

----


== See ==
Oracleâ€™s Programmer's Guide to Snippets: https://docs.oracle.com/en/.../code-snippet/index.html[https://docs.oracle.com/en/java/javase/18/code-snippet/index.html]


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

include::highlighting.adoc[]

'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
