== Why is this an issue?

Call a non-callable symbol will result in a TypeError at execution.


=== Noncompliant code example

[source,javascript]
----
function foo(a) {
  return a * 2;
}
// ....
var foo = 1;
// ...
foo();   // foo is 1 => TypeError
----


ifdef::env-github,rspecator-view[]
'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
