<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when a method modifies the size of a collection, while the same collection is iterated.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Iterating over a collection using a <code>for-each</code> loop in Java relies on iterators.</p>
</div>
<div class="paragraph">
<p>An iterator is an object that allows you to traverse a collection of elements, such as a list or a dictionary. Iterators are used in <code>for-each</code> loops to iterate over the elements of a collection one at a time.</p>
</div>
<div class="paragraph">
<p>It is important to note that iterators are designed to be read-only. Modifying a collection while iterating over it can cause unexpected behavior, as the iterator may skip over or repeat elements. Therefore, it is important to avoid modifying a collection while iterating over it to ensure that your code behaves as expected.</p>
</div>
<div class="paragraph">
<p>Most JDK collection implementations don&#8217;t support such modification and may throw a <code>ConcurrentModificationException</code>. Even if no such exception is thrown, attempting to modify a collection during iteration could be the source of incorrect or unspecified behaviors in the code.</p>
</div>
<div class="paragraph">
<p>If you still want to modify the collection, it is best to refactor the code and use a second collection (e.g by using streams and filter operations).</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">  public static void foo(List&lt;String&gt; lst) {
    for (String element : lst) {
      if (element.startsWith("x")) {
        lst.remove(element); // Noncompliant: lst size has been modified by "remove" call while it's iterated.
      }
    }
  }</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">  public static void foo(List&lt;String&gt; lst) {
    List&lt;String&gt; toRemove = new ArrayList&lt;&gt;();
    for (String element : lst) {
      if (element.startsWith("x")) {
        toRemove.add(element);
      }
    }
    lst.removeAll(toRemove);
  }</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Java Documentation <a href="https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/util/ConcurrentModificationException.html">ConcurrentModificationException</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>