<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when a method modifies the size of a collection, while the same collection is iterated.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Iterating over a collection using a <code>for</code> loop in Python relies on iterators.</p>
</div>
<div class="paragraph">
<p>An iterator is an object that allows you to traverse a collection of elements, such as a list or a dictionary. Iterators are used in <code>for</code> loops to iterate over the elements of a collection one at a time.</p>
</div>
<div class="paragraph">
<p>When you create an iterator, it keeps track of the current position in the collection and provides a way to access the next element. The <code>next()</code> function is used to retrieve the next element from the iterator. When there are no more elements to iterate over, the <code>next()</code> function raises a StopIteration exception and the iteration stops.</p>
</div>
<div class="paragraph">
<p>It is important to note that iterators are designed to be read-only. Modifying a collection while iterating over it can cause unexpected behavior, as the iterator may skip over or repeat elements. A <code>RuntimeError</code> may also be raised in this situation, with the message <code>changed size during iteration</code>. Therefore, it is important to avoid modifying a collection while iterating over it to ensure that your code behaves as expected.</p>
</div>
<div class="paragraph">
<p>If you still want to modify the collection, it is best to use a second collection or to iterate over a copy of the original collection instead.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">def my_fun():
    my_dict = {'a': 1, 'b': 2, 'c': 3}
    for key in my_dict:
        if my_dict[key] == 'foo':
            my_dict.pop(key) # Noncompliant: this will make the iteration unreliable</code></pre>
</div>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">def my_fun():
    my_dict = {'a': 1, 'b': 2, 'c': 3}
    for key in list(my_dict.keys()):
        if my_dict[key] == 'foo':
            my_dict.pop(key)</code></pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Python Documentation - <a href="https://docs.python.org/3/tutorial/controlflow.html#for-statements"><code>for</code> Statements</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>