This rule raises an issue when private methods or variables are made public solely to enable access from test classes, compromising proper encapsulation.

== Why is this an issue?

Making private members public just for testing breaks encapsulation, one of the fundamental principles of object-oriented programming. When you change a private method or variable to public, you expose internal implementation details that were intentionally hidden.

Encapsulation helps maintain code quality by:

* Hiding internal implementation details from other classes
* Preventing unintended access to sensitive logic or data
* Making it easier to refactor internal code without affecting external dependencies
* Clearly defining the public interface of a class

When private members are made public for testing, they become part of the class's public API, even though they weren't designed for external use. This can lead to:

* Other developers accidentally using these "internal" methods in production code
* Increased coupling between classes
* Difficulty in refactoring, as changing these methods might break code that shouldn't depend on them
* Confusion about what constitutes the intended public interface

Salesforce provides the ``++@TestVisible++`` annotation specifically to address this testing challenge. This annotation allows private members to be accessed from test classes while keeping them private in all other contexts. It's a clean solution that maintains encapsulation while enabling comprehensive testing.

=== What is the potential impact?

Breaking encapsulation can lead to tightly coupled code that is harder to maintain and refactor. It may also result in unintended dependencies on internal implementation details, making the codebase more fragile and prone to bugs when internal logic changes.

== How to fix it

Add the @TestVisible annotation to private methods that need to be accessed in test classes, instead of changing their visibility to public.

=== Code examples

==== Noncompliant code example

[source,apex,diff-id=1,diff-type=noncompliant]
----
public class MyClass {
    public String helperMethod() { // Noncompliant
        // Made public only for testing
        return 'helper logic';
    }
}
----

==== Compliant solution

[source,apex,diff-id=1,diff-type=compliant]
----
public class MyClass {
    @TestVisible
    private String helperMethod() {
        // Accessible in tests but remains private
        return 'helper logic';
    }
}
----

== Resources

=== Documentation

 * Salesforce @TestVisible Annotation - https://developer.salesforce.com/docs/atlas.en-us.apexcode.meta/apexcode/apex_classes_annotation_testvisible.htm[Official Salesforce documentation on using @TestVisible annotation]

 * Apex Testing Best Practices - https://developer.salesforce.com/docs/atlas.en-us.apexcode.meta/apexcode/apex_testing_best_practices.htm[Salesforce guide on best practices for Apex testing]

=== Standards

 * OWASP ASVS 4.0: V14.2.1 - https://owasp.org/www-project-application-security-verification-standard/[Application Security Verification Standard on secure coding practices]

=== Related rules

 * S5803
