<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This is an issue when accessing form data using dictionary-style syntax (<code>request.form['field_name']</code>) for HTML form fields that can have multiple values with the same name.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>HTML forms support multiple fields with the same name to create lists of values. This is commonly used for dynamic forms where users can add multiple phone numbers, addresses, or other repeated information.</p>
</div>
<div class="paragraph">
<p>When a form contains multiple fields with the same name, the Flask <code>request.form</code> object behaves like a dictionary but only returns one value when accessed with bracket notation. This means <code>request.form['field_name']</code> will only return one of the submitted values, silently discarding the others.</p>
</div>
<div class="paragraph">
<p>For example, if a user submits a form with three phone numbers, using <code>request.form['phone_number']</code> will only retrieve one phone number, losing the other two without any error or warning. This creates a poor user experience and can result in important data being lost.</p>
</div>
<div class="paragraph">
<p>The <code>request.form.getlist()</code> method is specifically designed to handle this scenario by returning all values for a given field name as a list, ensuring no data is lost.</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="paragraph">
<p>Data loss occurs when users submit multiple values for the same field. This can lead to:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>Lost user information that needs to be re-entered</p>
</li>
<li>
<p>Incomplete records in the application database</p>
</li>
<li>
<p>Poor user experience and frustration</p>
</li>
<li>
<p>Potential business impact if critical information is lost</p>
</li>
</ul>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it_in_flask">How to fix it in Flask</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Use <code>request.form.getlist()</code> instead of dictionary-style access when the form field can have multiple values. This method returns a list containing all submitted values for the specified field name.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python"># Only gets one phone number, loses others
phone_number = request.form['phone_number']  # Noncompliant</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python"># Gets all phone numbers as a list
phone_numbers = request.form.getlist('phone_number')</code></pre>
</div>
</div>
<div class="paragraph">
<p>When processing multiple related fields (like phone numbers and their types), use <code>getlist()</code> for both fields and process them together using <code>zip()</code>.</p>
</div>
</div>
<div class="sect3">
<h4 id="_noncompliant_code_example_2">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python"># Only processes one phone number and type
phone = request.form['phone_number']  # Noncompliant
phone_type = request.form['phone_type']  # Noncompliant
print(f'Phone: {phone} ({phone_type})')</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution_2">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python"># Processes all phone numbers and types
for number, type_ in zip(
    request.form.getlist('phone_number'),
    request.form.getlist('phone_type')
):
    if number:
        print(f'Phone: {number} ({type_})')</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Flask Request Documentation - <a href="https://flask.palletsprojects.com/en/2.3.x/api/#flask.Request.form">Official Flask documentation for the request.form object and its methods</a></p>
</li>
<li>
<p>Werkzeug MultiDict Documentation - <a href="https://werkzeug.palletsprojects.com/en/2.3.x/datastructures/#werkzeug.datastructures.MultiDict">Documentation for the MultiDict class that underlies request.form</a></p>
</li>
<li>
<p>Dynamic Forms with Flask - <a href="https://blog.miguelgrinberg.com/post/dynamic-forms-with-flask">Comprehensive guide on handling dynamic forms with multiple fields in Flask</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>