<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p><code>std::bit_cast</code> is one of the standard functions working with binary representation. Together with other bit-level functions, it is defined in the <code>&lt;bit&gt;</code> header introduced by C&#43;&#43;20.</p>
</div>
<div class="paragraph">
<p><code>std::bit_cast</code> standardizes the diverse and sub-optimal approaches of reinterpreting a value as being of a different type of the same length, preserving its binary representation.</p>
</div>
<div class="paragraph">
<p>Before C&#43;&#43;20, the correct way to reinterpret a value was a call to <code>std::memcpy</code>, copying the exact binary representation from a variable of one type into a variable of another. Although canonical, the use of <code>std::memcpy</code> might still be confusing; it is verbose, and it might introduce performance overhead if the compiler does not recognize the idiom and does not remove the function call.</p>
</div>
<div class="paragraph">
<p>In contrast, <code>std::bit_cast</code> clearly states the intent and is guaranteed to map to an optimal implementation.</p>
</div>
<div class="paragraph">
<p>This rule reports the uses of <code>std::memcpy</code> that can be replaced by <code>std::bit_cast</code>.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">static_assert(sizeof(float) == sizeof(uint32_t));
float src = 1.0f;
uint32_t dst;
std::memcpy(&amp;dst, &amp;src, sizeof(float)); // Noncompliant: verbose and might incur performance hit</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">float src = 1.0f;
auto dst = std::bit_cast&lt;uint32_t&gt;(src); // Compliant</code></pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>C&#43;&#43; reference - <a href="https://en.cppreference.com/w/cpp/numeric/bit_cast">std::bit_cast</a></p>
</li>
</ul>
</div>
</div>
<div class="sect2">
<h3 id="_related_rules">Related rules</h3>
<div class="paragraph">
<p>Other common patterns predating C&#43;&#43;20:</p>
</div>
<div class="ulist">
<ul>
<li>
<p><a data-rspec-id="S3630" class="rspec-auto-link">S3630</a> - replacing <code>std::reinterpret_cast</code> with <code>std::bit_cast</code>.</p>
</li>
<li>
<p><a data-rspec-id="S871" class="rspec-auto-link">S871</a> - replacing C-style cast with <code>std::bit_cast</code>.</p>
</li>
</ul>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_relates_to_s871">relates to: <a data-rspec-id="S871" class="rspec-auto-link">S871</a></h3>

</div>
<div class="sect2">
<h3 id="_relates_to_s3630">relates to: <a data-rspec-id="S3630" class="rspec-auto-link">S3630</a></h3>

</div>
<div class="sect2">
<h3 id="_is_related_to_s6232">is related to: <a data-rspec-id="S6232" class="rspec-auto-link">S6232</a></h3>

</div>
</div>
</div>