include::../description.adoc[]

== Noncompliant Code Example

----
SELECT name
FROM employee
WHERE EXISTS (SELECT * FROM department WHERE department_id = id AND name = 'Marketing');
----

== Compliant Solution

----
SELECT name
FROM employee INNER JOIN department AS d
  ON department_id = d.id AND d.name = 'Marketing';
----

ifdef::env-github,rspecator-view[]
'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
