<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Multiplying <code>sizeof()</code> with <code>sizeof()</code> indicates a logic error.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">sizeof(a) * sizeof(b); // Noncompliant</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Review this sizeof multiplication.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_27_jul_2015_140748_ann_campbell_wrote">on 27 Jul 2015, 14:07:48 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~massimo.paladin] I know you&#8217;ve given me the entirety of the CPPCheck definition, but it&#8217;s no much to go on. If it indicates a logic error, do you have any idea what may have been intended?</p>
</div>
</div>
</div>
</div>