<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>FOR and WHILE loops are structured control flow statements.</p>
</div>
<div class="paragraph">
<p>A FOR loop will iterate once for each element in the range, and the WHILE iterates for as long as a condition holds.</p>
</div>
<div class="paragraph">
<p>However, inserting an <code>EXIT</code> statement within the loop breaks this structure, reducing the code&#8217;s readability and making it harder to debug.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-sql" data-lang="sql">SET SERVEROUTPUT ON

DECLARE
  TYPE myCollectionType IS VARRAY(10) OF VARCHAR2(42);
  myCollection myCollectionType := myCollectionType('Foo', 'Bar', NULL, 'Baz', 'Qux');

  i PLS_INTEGER;
BEGIN
  i := 1;
  WHILE i &lt;= myCollection.LAST LOOP
    EXIT WHEN myCollection(i) IS NULL; -- Noncompliant, breaks the structure of the WHILE

    DBMS_OUTPUT.PUT_LINE('Got: ' || myCollection(i));
    i := i + 1;
  END LOOP;</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-sql" data-lang="sql">SET SERVEROUTPUT ON

DECLARE
  TYPE myCollectionType IS VARRAY(10) OF VARCHAR2(42);
  myCollection myCollectionType := myCollectionType('Foo', 'Bar', NULL, 'Baz', 'Qux');

  i PLS_INTEGER;
BEGIN
  i := 1;
  WHILE i &lt;= myCollection.LAST AND myCollection(i) IS NOT NULL LOOP
    DBMS_OUTPUT.PUT_LINE('Got: ' || myCollection(i));
    i := i + 1;
  END LOOP;
END;
/</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove this use of "EXIT".</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_is_related_to_s910">is related to: <a data-rspec-id="S910" class="rspec-auto-link">S910</a></h3>

</div>
</div>
</div>