== Why is this an issue?

The ``++yield++`` keyword is used in a generator function to return an ``++IteratorResult++`` to the caller. It has no other purpose, and if found outside such a function will raise a ``++ReferenceError++`` because it is then treated as an identifier.


=== Noncompliant code example

[source,javascript]
----
function foo() { 
  for (var i = 0; i < 5; i++) { 
    yield i * 2; 
  } 
}
----


=== Compliant solution

[source,javascript]
----
function * foo() { 
  for (var i = 0; i < 5; i++) { 
    yield i * 2; 
  } 
}
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Remove this "yield" expression or move it into a generator.


=== Highlighting

"yield" expression


'''
== Comments And Links
(visible only on this page)

=== on 1 Nov 2019, 17:34:27 Elena Vilchik wrote:
See \https://github.com/SonarSource/SonarJS/issues/1698

endif::env-github,rspecator-view[]
