<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>C&#43;&#43; language is highly elastic regarding the placement of type specifiers on type declarations. This makes it possible to place a <code>const</code> qualifier before the type ("west const") or after it ("east const").</p>
</div>
<div class="paragraph">
<p>Mixing both <code>const</code> placement in a single project or, even worse, in a single file, makes the code harder to understand. It will require more effort to notice that the object cannot be modified.</p>
</div>
<div class="paragraph">
<p>This rule checks if the <code>const</code> qualifier placement matches the configuration.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="paragraph">
<p>With default value of "eastConst" parameter.</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">long double const tau = 6.28L; // Noncompliant</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">const long double tau = 6.28L;</code></pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p>C++ Core Guidelines - <a href="https://github.com/isocpp/CppCoreGuidelines/blob/e49158a/CppCoreGuidelines.md#nl26-use-conventional-const-notation">NL.26: Use conventional <code>const</code> notation</a></p>
</li>
</ul>
</div>
<hr>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_parameters">Parameters</h3>
<div class="sidebarblock">
<div class="content">
<div class="title">eastConst</div>
<div class="paragraph">
<p><em>BOOLEAN</em></p>
</div>
<div class="listingblock">
<div class="content">
<pre>false</pre>
</div>
</div>
<div class="paragraph">
<p>Determines if the "const" qualifier should be placed after type</p>
</div>
</div>
</div>
</div>
</div>
</div>