<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>When calling <code>delete</code> on an object of incomplete type, the calling code does not have enough information to do the action properly (it does not know if this object has a trivial or a nontrivial destructor, if it has overloaded the <code>delete</code> operator&#8230;&#8203;). Therefore, deleting a pointer to such an object can lead to undefined behavior.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">class Body;

class Handle {
  public:
    Handle();
    ~Handle() {
      delete impl; // Noncompliant, Body is incomplete
    }
  private:
    Body * impl;
};</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">// In header file
class Body;

class Handle {
  public:
    Handle();
    ~Handle();
    // Add other special member functions to respect the rule of five
  private:
    Body * impl;
};

// In implementation file
#include "Handle.h"
#include "Body.h" // Now Body is complete

Handle::~Handle(){
  delete impl; // Compliant, at this point "Body" is a complete class
}</code></pre>
</div>
</div>
<div class="paragraph">
<p>Or, with modern resource handling:</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">// In header file
class Body;

class Handle {
  public:
    Handle();
    ~Handle();
  private:
    std::unique_ptr&lt;Body&gt; impl; //Compliant
};

// In implementation file
#include "Handle.h"
#include "Body.h" // Now Body is complete

Handle::Handle() : impl{new Body{}} {}
Handle::~Handle() = default; // since "Body" is complete, it can be destroyed by unique_ptr</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>deleting pointer to incomplete type "XXX" may cause undefined behavior</p>
</div>
</div>
</div>
</div>