== Why is this an issue?

The "ManagedEvents" metadata tag allows you to flag an event as being managed. By definition this "ManagedEvents" metadata tag should be used in pair with an "Event" metadata tag. 


=== Noncompliant code example

[source,flex]
----
[Event(name="message", type="my.package.MyEvemt")]
[ManagedEvents("mes")]       //This "mes" event is not defined with the "Event" metadata tag
public class MyClass {...}
----


=== Compliant solution

[source,flex]
----
[Event(name="message", type="my.package.MyEvemt")]
[ManagedEvents("message")]
public class MyClass {...}
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

The managed event "XXXX" is either misspelled or is missing a companion Event metadata tag


'''
== Comments And Links
(visible only on this page)

=== on 22 Nov 2013, 09:52:31 Freddy Mallet wrote:
Is implemented by \http://jira.codehaus.org/browse/SONARPLUGINS-3275

=== on 24 Nov 2013, 19:15:41 Ann Campbell wrote:
Not entirely sure "companion" is the right word here...

endif::env-github,rspecator-view[]
