== Why is this an issue?

The use of a ``++StringBuilder++`` or ``++StringBuffer++`` is supposed to make ``++String++`` assembly more efficient than plain concatenation. So don't ruin the effect by concatenating the arguments to ``++append++``.


=== Noncompliant code example

[source,java,diff-id=1,diff-type=noncompliant]
----
StringBuilder sb = new StringBuilder();
sb.append("foo is: " + getFoo()); // Noncompliant
----


=== Compliant solution

[source,java,diff-id=2,diff-type=compliant]
----
StringBuilder sb = new StringBuilder();
sb.append("foo is: ").append(getFoo());
----

== Resources

=== Benchmarks

[options="header"]
|===
| method| Runtime| Average time| Error margin
| append| Temurin 21| 2.64e-08 s/op| ±6.08e-10 s/op
| concat| Temurin 21| 2.99e-08 s/op| ±6.32e-10 s/op
|===

The results were generated by running the following snippet with https://github.com/openjdk/jmh[jmh]:

[source,java]
----
private String name1 = "John";
private String name2 = "Jane";

@Benchmark
public String concat(){
  StringBuilder sb = new StringBuilder();
  sb.append(name1+", "+name2);
  return sb.toString();
}

@Benchmark
public String append(){
  StringBuilder sb = new StringBuilder();
  sb.append(name1).append(", ").append(name2);
  return sb.toString();
}
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

"append" each concatenated value separately.


endif::env-github,rspecator-view[]
