== Why is this an issue?

The use of a ``++StringBuilder++`` or ``++StringBuffer++`` is supposed to make ``++String++`` assembly more efficient than plain concatenation. So don't ruin the effect by concatenating the arguments to ``++append++``.


=== Noncompliant code example

[source,java]
----
StringBuilder sb = new StringBuilder();
sb.append("foo is: " + getFoo()); // Noncompliant
----


=== Compliant solution

[source,java]
----
StringBuilder sb = new StringBuilder();
sb.append("foo is: ").append(getFoo());
----

== Resources

=== Benchmarks

[options="header"]
|===
| method| Average time| Error margin
| append| 2.64e-08 s/op| ±6.08e-10 s/op
| concat| 2.99e-08 s/op| ±6.32e-10 s/op
|===


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

"append" each concatenated value separately.


endif::env-github,rspecator-view[]
