== Why is this an issue?

The use of a ``++StringBuilder++`` or ``++StringBuffer++`` is supposed to make ``++String++`` assembly more efficient than plain concatenation. So don't ruin the effect by concatenating the arguments to ``++append++``.


=== Noncompliant code example

[source,java]
----
StringBuilder sb = new StringBuilder();
sb.append("foo is: " + getFoo()); // Noncompliant
----


=== Compliant solution

[source,java]
----
StringBuilder sb = new StringBuilder();
sb.append("foo is: ").append(getFoo());
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

"append" each concatenated value separately.


endif::env-github,rspecator-view[]
