<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>The dangling <code>else</code> problem appears when nested <code>if</code>/<code>else</code> statements are written without curly braces. In this case, <code>else</code> is associated with the nearest <code>if</code> but that is not always obvious and sometimes the indentation can also be misleading.</p>
</div>
<div class="paragraph">
<p>This rules reports <code>else</code> statements that are difficult to understand, because they are inside nested <code>if</code> statements without curly braces.</p>
</div>
<div class="paragraph">
<p>Adding curly braces can generally make the code clearer (see rule <a data-rspec-id="S121" class="rspec-auto-link">S121</a> ), and in this situation of dangling <code>else</code>, it really clarifies the intention of the code.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-text" data-lang="text"> if (a)
   if (b)
     d++;
 else     // Noncompliant, is the "else" associated with "if(a)" or "if (b)"? (the answer is "if(b)")
   e++;</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-text" data-lang="text"> if (a) {
   if (b) {
     d++;
   }
 } else { // Compliant, there is no doubt the "else" is associated with "if(a)"
   e++;
 }</code></pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p><a href="https://en.wikipedia.org/wiki/Dangling_else" class="bare">https://en.wikipedia.org/wiki/Dangling_else</a></p>
</li>
</ul>
</div>
<hr>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Add explicit curly braces to avoid dangling else.</p>
</div>
</div>
</div>
</div>