<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>While not mandatory, using the <a href="https://api.dart.dev/dart-core/override-constant.html"><code>@override</code></a> annotation on compliant members (methods, properties, operators) improves readability by making it explicit that members are overridden.</p>
</div>
<div class="paragraph">
<p>Unlike other languages, all methods in Dart are <a href="https://en.wikipedia.org/wiki/Virtual_function"><code>virtual</code></a> by default. So, using the <code>@override</code> annotation prevents accidental overriding of a base class method in a subclass.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-dart" data-lang="dart">class ParentClass {
  bool doSomething(){/*...*/}
}
class FirstChildClass extends ParentClass {
  bool doSomething(){/*...*/}  // Noncompliant
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-dart" data-lang="dart">class ParentClass {
  bool doSomething(){/*...*/}
}
class FirstChildClass extends ParentClass {
  @override
  bool doSomething(){/*...*/}  // Compliant
}</code></pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p>Dart Docs - <a href="https://dart.dev/tools/linter-rules/annotate_overrides">Dart Linter rule - annotate_overrides</a></p>
</li>
<li>
<p>Dart Docs - <a href="https://dart.dev/language/extend#overriding-members">Extend a class - Overriding members</a></p>
</li>
<li>
<p>Dart API Reference - <a href="https://api.dart.dev/dart-core/override-constant.html">Override top-level constant</a></p>
</li>
<li>
<p>Wikipedia - <a href="https://en.wikipedia.org/wiki/Virtual_function">Virtual function</a></p>
</li>
</ul>
</div>
<hr>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>The member '&lt;method name&gt;' overrides an inherited member but isn&#8217;t annotated with '@override'.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>The identifier of the method, property or operator.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_29_jul_2013_155156_freddy_mallet_wrote">on 29 Jul 2013, 15:51:56 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>Is implemented by http://jira.codehaus.org/browse/SONARJAVA-249</p>
</div>
</div>
</div>
</div>