== Why is this an issue?

If ``++Thread.sleep(...)++`` is called when the current thread holds a lock, it could lead to performance and scalability issues, or even worse to deadlocks because the execution of the thread holding the lock is frozen. It's better to call ``++wait(...)++`` on the monitor object to temporarily release the lock and allow other threads to run.


=== Noncompliant code example

[source,java]
----
public void doSomething(){
  synchronized(monitor) {
    while(notReady()){
      Thread.sleep(200);
    }
    process();
  }
  ...
}
----


=== Compliant solution

[source,java]
----
public void doSomething(){
  synchronized(monitor) {
    while(notReady()){
      monitor.wait(200);
    }
    process();
  }
  ...
}
----


== Resources

* https://wiki.sei.cmu.edu/confluence/x/YTdGBQ[CERT, LCK09-J.] - Do not perform operations that can block while holding a lock


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Replace the call to "Thread.sleep(...)" with a call to "wait(...)".


'''
== Comments And Links
(visible only on this page)

=== on 15 Feb 2016, 17:17:18 Ann Campbell wrote:
This RSpec exlicitly _not_ expanded to cover full scope of CERT, LCK09-J. because the other operations _can_ lead to problems (but may not), whereas for ``++sleep++`` there is a problem sure.

endif::env-github,rspecator-view[]
