include::../description.adoc[]

== Noncompliant Code Example

----
<TestMethod>
<ExpectedException(GetType(ArgumentNullException))> ' Noncompliant
Public Sub TestNullArg()
  '...
End Sub
----

== Compliant Solution

----
<TestMethod>
Public Sub TestNullArg()
    Dim CallFailed As Boolean = False
    Try
    ' ...
    Catch ex As Exception
        CallFailed = true
    End Try
    Assert.IsTrue(CallFailed, "Expected call to MyMethod to fail with ArgumentNullException")
End Sub
----

or

----
<TestMethod>
Public Sub TestNullArg()
    Assert.ThrowsException(Of ArgumentNullException)(Sub() ... )
End Sub
----

include::../exceptions.adoc[]

ifdef::env-github,rspecator-view[]
'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
