== How to fix it in NHibernate

=== Code examples

include::../../common/fix/code-rationale.adoc[]

==== Noncompliant code example

[source,csharp,diff-id=1,diff-type=noncompliant]
----
public class ExampleController : Controller
{
    private readonly NHibernate.ISessionFactory SessionFactory;

    public IActionResult Authenticate(string user, string pass)
    {
        using (var session = SessionFactory.OpenSession())
        {
            var hql = "FROM UserModel WHERE User = '" + user + "' AND Pass = '" + pass + "'";

            var query = session.CreateQuery(hql);
            
            var result = query.UniqueResult<UserModel>();
            if (result == null) {
                return Unauthorized();
            }
        }  

        return Ok();
    }
}
----

==== Compliant solution

[source,csharp,diff-id=1,diff-type=compliant]
----
public class ExampleController : Controller
{
    private readonly NHibernate.ISessionFactory SessionFactory;

    public IActionResult Authenticate(string user, string pass)
    {
        using (var session = SessionFactory.OpenSession())
        {
            var hql = "FROM UserModel WHERE User = :user AND Pass = :pass";

            var query = session.CreateQuery(hql);
            query.SetParameter("user", user);
            query.SetParameter("pass", pass);

            var result = query.UniqueResult<UserModel>();
            if (result == null) {
                return Unauthorized();
            }
        }  

        return Ok();
    }
}
----

=== How does this work?

include::../../common/fix/prepared-statements.adoc[]

