== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,swift]
----
if equal(myPoint.x, myPoint.x) {  // Noncompliant
  //...
}
----

=== Compliant solution

[source,swift]
----
if equal(myPoint.x, myPoint.y) {
  //...
}
----
ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../highlighting.adoc[]

endif::env-github,rspecator-view[]
