include::../description.adoc[]

== Noncompliant Code Example

----
if equal(myPoint.x, myPoint.x) {  // Noncompliant
  //...
}
----

== Compliant Solution

----
if equal(myPoint.x, myPoint.y) {
  //...
}
----
ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../highlighting.adoc[]

endif::env-github,rspecator-view[]
