<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when the <code>input_shape</code> is specified in a <code>tensorflow.keras.Model</code> subclass.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Keras provides a full-featured model class called <code>tensorflow.keras.Model</code>. It inherits from <code>tensorflow.keras.layers.Layer</code>, so a Keras model can be used and nested in the same way as Keras layers. Keras models come with extra functionality that makes them easy to train, evaluate, load, save, and even train on multiple machines.</p>
</div>
<div class="paragraph">
<p>As the <code>tensorflow.keras.Model</code> class inherits from the 'tensorflow.keras.layers' you do not need to specify <code>input_shape</code> in a subclassed model; this argument will be ignored.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Do not specify <code>input_shape</code> in a <code>tf.keras.Model</code> subclasses</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">import tensorflow as tf

class MyModel(tf.keras.Model):
    def __init__(self):
        super(MyModel, self).__init__(input_shape=...)  # Noncompliant: this parameter will be ignored</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">import tensorflow as tf

class MyModel(tf.keras.Model):
    def __init__(self):
        super(MyModel, self).__init__()  # OK</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Tensorflow documentation - <a href="https://www.tensorflow.org/guide/intro_to_modules#keras_models">Keras models</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>