<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>In the interests of clean code, getters should be implicit, rather than explicit for properties that have only getters.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-swift" data-lang="swift">var luckyNumber: Int {
  get {  // Noncompliant
    return 7
  }
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-swift" data-lang="swift">var luckyNumber: Int {
  return 7
}</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_duplicates_s2962">duplicates: <a data-rspec-id="S2962" class="rspec-auto-link">S2962</a></h3>

</div>
<div class="sect2">
<h3 id="_on_8_jul_2016_192950_ann_campbell_wrote">on 8 Jul 2016, 19:29:50 Ann Campbell wrote:</h3>
<div class="paragraph">
<p><a href="https://github.com/github/swift-style-guide#prefer-implicit-getters-on-read-only-properties-and-subscripts" class="bare">https://github.com/github/swift-style-guide#prefer-implicit-getters-on-read-only-properties-and-subscripts</a></p>
</div>
</div>
</div>
</div>