The Dockerfile should contain at most one `ENTRYPOINT` and one `CMD` instruction, because only the last one will have an effect.

== Why is this an issue?

Multiple `ENTRYPOINT` or `CMD` instructions in a file can lead to confusion as we may think they are all applied, which is not the case.
Previous `ENTRYPOINT` and `CMD` instructions should be removed to avoid this.

== How to fix it

=== Code examples

==== Noncompliant code example

[source,docker,diff-id=1,diff-type=noncompliant]
----
ENTRYPOINT ignored_entrypoint param1 param2
ENTRYPOINT effective_entrypoint param1 param2

CMD ignored_cmd param1 param2
CMD effective_cmd param1 param2
----

Here we have multiple `ENTRYPOINT` and `CMD` instructions.
The first `ENTRYPOINT` and the first `CMD` instructions will have no effect.
Although this is valid in Docker, this can lead to confusion and be error-prone, as we may expect each `CMD` and `ENTRYPOINT` to have an effect.

==== Compliant solution

[source,docker,diff-id=1,diff-type=compliant]
----
ENTRYPOINT effective_entrypoint param1 param2

CMD effective_cmd param1 param2
----

Here we have only one ENTRYPOINT and one CMD instruction.
Each of them will be considered by the docker container and have a normal effect as we can expect.

== Resources

=== Documentation

* https://docs.docker.com/engine/reference/builder/#entrypoint[ENTRYPOINT - Dockerfile reference]
* https://docs.docker.com/engine/reference/builder/#cmd[CMD - Dockerfile reference]

ifdef::env-github,rspecator-view[]
'''
== Implementation Specification
(visible only on this page)

=== Message

Remove this `CMD` or `ENTRYPOINT` instruction which will be ignored.

=== Highlighting

Highlight the entire `CMD` and `ENTRYPOINT` instruction.

'''
endif::env-github,rspecator-view[]
