Creating a new `+Exception+` without actually throwing it is useless and is probably due to a mistake.

== Noncompliant Code Example

----
if ($x < 0)
  new Exception('$x must be nonnegative');
----

== Compliant Solution

----
if ($x < 0)
  throw new Exception('$x must be nonnegative');
----
