Creating a new ``++Exception++`` without actually throwing it is useless and is probably due to a mistake.

== Noncompliant Code Example

[source,php]
----
if ($x < 0)
  new Exception('$x must be nonnegative');
----

== Compliant Solution

[source,php]
----
if ($x < 0)
  throw new Exception('$x must be nonnegative');
----
ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../highlighting.adoc[]

endif::env-github,rspecator-view[]
