include::../description.adoc[]

=== Exceptions

This rule does not raise an issue when an ASP.NET virtual path is passed as an argument to one of the following:

* methods: `System.Web.HttpServerUtilityBase.MapPath()`, `System.Web.HttpRequestBase.MapPath()`, `System.Web.HttpResponseBase.ApplyAppPathModifier()`, `System.Web.Mvc.UrlHelper.Content()`
* all methods of: `System.Web.VirtualPathUtility`
* constructors of: `Microsoft.AspNetCore.Mvc.VirtualFileResult`, `Microsoft.AspNetCore.Routing.VirtualPathData`

== How to fix it

=== Code examples

==== Noncompliant code example

[source,csharp,diff-id=1,diff-type=noncompliant]
----
public class Foo {
  public List<User> ListUsers() {
    string userListPath = "/home/mylogin/Dev/users.txt"; // Noncompliant
    return ParseUsers(userListPath);
  }
}
----

==== Compliant solution

[source,csharp,diff-id=1,diff-type=compliant]
----
public class Foo {
  // Configuration is a class that returns customizable properties: it can be mocked to be injected during tests. 
  private Configuration config;
  public Foo(Configuration myConfig) {
    this.config = myConfig;
  }
  public List<User> ListUsers() {
    // Find here the way to get the correct folder, in this case using the Configuration object
    string listingFolder = config.GetProperty("myApplication.listingFolder");
    // and use this parameter instead of the hard coded path
    string userListPath = Path.Combine(listingFolder, "users.txt"); // Compliant
    return ParseUsers(userListPath);
  }
}
----

include::../resources.adoc[]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

* Refactor your code not to use hardcoded absolute paths or URIs.
* Remove this hardcoded path-delimiter.
'''

endif::env-github,rspecator-view[]
