<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when a Tensorflow function&#8217;s <code>indices</code> argument has out-of-bound values.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Tensorflow provides various functions producing matrices based on <code>input</code> matrices and their <code>indices</code>.
For these functions <code>indices</code> is an array of numbers referencing input matrix elements.
Values of <code>indices</code> that are out-of-bounds for <code>input</code> can lead to errors or unexpected behavior.</p>
</div>
<div class="paragraph">
<p>Functions sensitive to <code>indices</code> argument values:</p>
</div>
<div class="ulist">
<ul>
<li>
<p><code>tf.gather</code></p>
</li>
<li>
<p><code>tf.gather_nd</code></p>
</li>
<li>
<p><code>tf.scatter_nd</code></p>
</li>
</ul>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">import tensorflow as tf

input = tf.constant([1, 2, 3])
indices = [1, 0, 3]
result = tf.gather(input, indices)  # Noncompliant: InvalidArgumentError: indices[2] = 3 is not in [0, 3)</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">import tensorflow as tf

input = tf.constant([1, 2, 3])
indices = [1, 0, 2]
result = tf.gather(input, indices)  # OK</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Tensorflow documentation - <a href="https://www.tensorflow.org/api_docs/python/tf/gather">tf.gather</a></p>
</li>
<li>
<p>Tensorflow documentation - <a href="https://www.tensorflow.org/api_docs/python/tf/gather_nd">tf.gather_nd</a></p>
</li>
<li>
<p>Tensorflow documentation - <a href="https://www.tensorflow.org/api_docs/python/tf/scatter_nd">tf.scatter_nd</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>