include::../description.adoc[]

== Noncompliant Code Example

[source,csharp]
----
[ExcludeFromCodeCoverage]
public struct HashCode
{
    public static int Combine<T1, T2>(T1 value1, T2 value2)
    {
    }
}
----

== Compliant Solution

[source,csharp]
----
[ExcludeFromCodeCoverage(Justification = "Copy of https://github.com/dotnet/runtime/blob/main/src/libraries/System.Private.CoreLib/src/System/HashCode.cs for the .Net 4.8. Unit tests are in the upstream repo.")]
public struct HashCode
{
    public static int Combine<T1, T2>(T1 value1, T2 value2)
    {
    }
}
----

include::../see.adoc[]