This rule raises an issue when AWS Lambda functions create files in /tmp without cleaning them up before the function completes.

== Why is this an issue?

AWS Lambda provides a temporary file system at /tmp for each execution environment. However, the contents of /tmp can persist across multiple invocations when Lambda reuses the same execution environment (called a "warm start"). When Lambda functions create temporary files without cleaning them up, these files remain accessible to subsequent invocations of the same function instance.

=== What is the potential impact?

Failing to clean up temporary files can lead to data leakage where sensitive information from one invocation becomes accessible to unrelated subsequent invocations. Additionally, it can cause disk space exhaustion since /tmp has limited storage capacity, potentially causing function failures. Finally, subsequent invocations might accidentally use stale data from previous runs, leading to unpredictable behavior and hard-to-debug issues.

== How to fix it

Always clean up temporary files after use by implementing proper file cleanup logic. Use a try...finally block to ensure files are removed even when exceptions occur during processing. Check if the file exists before attempting to remove it to avoid errors.

=== Code examples

==== Noncompliant code example
[source,python,diff-id=1,diff-type=noncompliant]
----
import json
import os

def process_data(event, context):
    file_path = '/tmp/temp_data.txt'  # Noncompliant
    with open(file_path, 'w') as f:
        f.write(json.dumps(event))
    
    # ... processing logic that uses the file ...
    
    return {
        'statusCode': 200,
        'body': 'Data processed successfully.'
    }
----

==== Compliant solution
[source,python,diff-id=1,diff-type=compliant]
----
import json
import os

def process_data(event, context):
    file_path = '/tmp/temp_data.txt'
    try:
        with open(file_path, 'w') as f:
            f.write(json.dumps(event))
        
        # ... processing logic that uses the file ...
        
        return {
            'statusCode': 200,
            'body': 'Data processed successfully.'
        }
    finally:
        if os.path.exists(file_path):
            os.remove(file_path)
----

== Resources

=== Documentation
* https://docs.aws.amazon.com/lambda/latest/dg/configuration-filesystem.html[AWS Lambda - Using Amazon EFS with Lambda]
* https://docs.aws.amazon.com/lambda/latest/dg/python-handler.html[AWS Lambda - Define Lambda function handler in Python]
* https://docs.python.org/3/library/tempfile.html[Python documentation - tempfile module]


ifdef::env-github,rspecator-view[]

== Implementation Specification
(visible only on this page)

=== Message

Clean up this temporary file before the Lambda function completes.

=== Highlighting

* Primary location: the file creation or write operation in /tmp
* Secondary location: the Lambda function definition

endif::env-github,rspecator-view[]