This rule raises an issue when AWS Lambda handlers write to the /tmp directory without cleaning up temporary files before completing invocation.

== Why is this an issue?

AWS Lambda provides a temporary file system at /tmp for each execution environment. While useful for ephemeral storage, the contents of /tmp can persist across multiple invocations of the same Lambda function instance during "warm starts". When Lambda reuses an execution environment, temporary files from previous invocations remain accessible to new invocations.

=== What is the potential impact?

Failing to clean up temporary files can lead to data leakage where sensitive information from one invocation becomes accessible to subsequent unrelated invocations. Additionally, it can cause disk space exhaustion since /tmp has limited size, leading to function failures. Stale data from previous runs may also cause unexpected behavior and hard-to-trace bugs.

== How to fix it

Always clean up temporary files after use by implementing proper cleanup logic. Use a try...finally block to ensure files are removed even if errors occur during processing. For complex scenarios with multiple temporary files, consider using Python's tempfile module combined with proper cleanup.

=== Code examples

==== Noncompliant code example
[source,python,diff-id=1,diff-type=noncompliant]
----
import json
import os

def process_data(event, context):
    file_path = '/tmp/temp_data.txt'
    with open(file_path, 'w') as f:  # Noncompliant
        f.write(json.dumps(event))
    
    # ... processing logic ...
    
    return {
        'statusCode': 200,
        'body': 'Data processed successfully.'
    }
----

==== Compliant solution
[source,python,diff-id=1,diff-type=compliant]
----
import json
import os

def process_data(event, context):
    file_path = '/tmp/temp_data.txt'
    try:
        with open(file_path, 'w') as f:
            f.write(json.dumps(event))
        
        # ... processing logic ...
        
        return {
            'statusCode': 200,
            'body': 'Data processed successfully.'
        }
    finally:
        if os.path.exists(file_path):
            os.remove(file_path)
----

== Resources

=== Documentation
* https://docs.aws.amazon.com/lambda/latest/dg/python-handler.html[AWS Lambda Python handler documentation]
* https://docs.aws.amazon.com/lambda/latest/dg/configuration-filesystem.html[AWS Lambda file system configuration]


ifdef::env-github,rspecator-view[]

== Implementation Specification
(visible only on this page)

=== Message

Clean up this temporary file in /tmp before the Lambda function completes.

=== Highlighting

* Primary location: the file creation operation in /tmp directory
* Secondary location: the Lambda handler function signature

endif::env-github,rspecator-view[]