This rule raises an issue when AWS Lambda functions create files in the /tmp directory without properly cleaning them up before the function completes.

== Why is this an issue?

AWS Lambda provides a temporary file system at /tmp for each execution environment. The /tmp directory persists across multiple invocations of the same Lambda function instance during 'warm starts'. When Lambda functions create temporary files but don't clean them up, the files remain accessible to subsequent invocations. This creates several problems: sensitive data from one invocation might leak to unrelated future invocations, the limited /tmp storage space can fill up causing function failures, and subsequent invocations might accidentally use stale data from previous runs leading to unpredictable behavior.

=== What is the potential impact?

Data leakage between function invocations can expose sensitive information to unauthorized contexts. Disk space exhaustion will cause Lambda functions to fail with storage errors. Using stale data from previous invocations can lead to incorrect processing results and hard-to-debug application issues.

== How to fix it

Always clean up temporary files created in /tmp before the Lambda function completes. Use a try...finally block to ensure cleanup happens even if errors occur during processing. Check if the file exists before attempting to remove it to avoid errors. For complex scenarios with multiple files or directories, consider using Python's tempfile module along with proper cleanup.

=== Code examples

==== Noncompliant code example
[source,python,diff-id=1,diff-type=noncompliant]
----
import json
import os

def process_data(event, context):
    file_path = '/tmp/temp_data.txt'  # Noncompliant
    with open(file_path, 'w') as f:
        f.write(json.dumps(event))
    
    # ... processing logic ...
    
    return {
        'statusCode': 200,
        'body': 'Data processed successfully.'
    }
----

==== Compliant solution
[source,python,diff-id=1,diff-type=compliant]
----
import json
import os

def process_data(event, context):
    file_path = '/tmp/temp_data.txt'
    try:
        with open(file_path, 'w') as f:
            f.write(json.dumps(event))
        
        # ... processing logic ...
        
        return {
            'statusCode': 200,
            'body': 'Data processed successfully.'
        }
    finally:
        if os.path.exists(file_path):
            os.remove(file_path)
----

== Resources

=== Documentation
* https://docs.aws.amazon.com/lambda/latest/dg/python-handler.html[AWS Lambda Python Handler Documentation]
* https://docs.aws.amazon.com/lambda/latest/dg/configuration-filesystem.html[AWS Lambda Temporary File System Configuration]


ifdef::env-github,rspecator-view[]

== Implementation Specification
(visible only on this page)

=== Message

Clean up this temporary file before the Lambda function completes.

=== Highlighting

* Primary location: the file creation operation in /tmp
* Secondary location: the Lambda handler function signature

endif::env-github,rspecator-view[]