:identifier_capital_plural: Functions and block parameters
:identifier: function and block parameter
:identifier_plural: functions and block parameters
:identifier_or: function or block parameter
:regex: ^(@{0,2}[\da-z_]+[!?=]?)|([*+-/%=!><~]+)|(\[]=?)$

include::../rule.adoc[]

=== Code examples

==== Noncompliant code example

With the default regular expression ``{regex}``:

[source,ruby,diff-id=1,diff-type=noncompliant]
----
def show_something(text_Param) # Noncompliant
  localVar = "" # Noncompliant
  puts text_Param + localVar
end
----

==== Compliant solution

[source,ruby,diff-id=1,diff-type=compliant]
----
def show_something(text_param)
  local_var = ""
  puts text_param + local_var
end
----

== Resources

=== Documentation

* Wikipedia - https://en.wikipedia.org/wiki/Naming_convention_(programming)[Naming Convention (programming)]

=== Related rules

* S100 - Method names should comply with a naming convention
* S101 - Class names should comply with a naming convention

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../parameters.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
