<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Having multiple prefixes for the same namespace is highly likely to confuse users and maintainers of the schema.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-xml" data-lang="xml">&lt;xs:schema targetNamespace="http://www.codeSamples.com/fruit"
           elementFormDefault="qualified" version="1"
           xmlns:xs="http://www.w3.org/2001/XMLSchema"
           xmlns:animal="http://www.codeSamples.com/animal"
           xmlns:beast="http://www.codeSamples.com/animal"&gt;    &lt;!-- Noncompliant --&gt;
  ...
&lt;/xs:schema&gt;</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-xml" data-lang="xml">&lt;xs:schema targetNamespace="http://www.codeSamples.com/fruit"
           elementFormDefault="qualified" version="1"
           xmlns:xs="http://www.w3.org/2001/XMLSchema"
           xmlns:animal="http://www.codeSamples.com/animal"&gt;
  ...
&lt;/xs:schema&gt;</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove this redundant namespace listing; this namespace has already been assigned the prefix "xxx".</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>Primary: <code>xmlns:xxx=yyy"</code></p>
</div>
<div class="paragraph">
<p>Secondary: inital definition/prefix assignment</p>
</div>
</div>
</div>
</div>