== How to fix it in request

=== Code examples

include::../../common/fix/code-rationale.adoc[]

==== Noncompliant code example

[source,javascript,diff-id=1,diff-type=noncompliant]
----
const request = require('request');

let socket = request.get({
  url: 'www.example.com',
  secureProtocol: 'TLSv1_2_method',
  rejectUnauthorized: false // Noncompliant
});
----

==== Compliant solution

[source,javascript,diff-id=1,diff-type=compliant]
----
const request = require('request');

let socket = request.get({
  url: 'https://www.example.com/',
  secureProtocol: 'TLSv1_2_method'
  // By default rejectUnauthorized is set to true
});
----

=== How does this work?

include::../../common/fix/default-implementation.adoc[]
