<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>No member function can be invoked on a const-qualified object unless the member function is declared "const".</p>
</div>
<div class="paragraph">
<p>Qualifying member functions that don&#8217;t mutate their object with the "const" qualifier makes your interface easier to understand; you can deduce without diving into implementation if a member function is going to mutate its object.</p>
</div>
<div class="paragraph">
<p>Also, const-qualified member functions make working with const-qualified objects possible. The compiler ensures that only member functions that are declared "const" can be invoked on "const" objects. Avoiding declaring non-mutating member functions const might breakÂ const-correctness: it will not be possible to invoke such non-mutating functions on const-qualified objects.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">class A {
  void f(){ // Noncompliant
    std::cout&lt;&lt; "f doesn't mutate A";
  }
};</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">class A {
  void f() const {
    std::cout&lt;&lt; "f doesn't mutate A";
  }
};</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_exceptions">Exceptions</h3>
<div class="paragraph">
<p>Virtual member functions that don&#8217;t mutate their objects don&#8217;t necessarily need to be declared const. This might be done in order to allow them to be overridden by non-const functions.</p>
</div>
</div>
</div>
</div>