== Why is this an issue?

PHP doesn't fail if two or more `define` statements assign a value to the same constant name.
In such a case, PHP only issues a warning and ignores the second and further `define`.

=== What is the potential impact?

In the best case, the additional `define` statements try to assign the same value to the constant.
This would just entail useless code which can be removed.

In the worst case, the additional `define` statements attempt to assign a different value and are followed by code which wrongly assumes the value of the constant.
This can lead to difficult bugs.

== How to fix it

Remove additional `define` statements and only keep the intended one.

=== Code examples

==== Noncompliant code example

[source,php,diff-id=1,diff-type=noncompliant]
----
define( 'CONSTANT_VALUE', 'old value' );
define( 'SCRIPT_DEBUG', 1 );

// Noncompliant, tries to redefine constant defined 2 lines above
define( 'CONSTANT_VALUE', 'intended value' );
echo CONSTANT_VALUE; // output: 'old value'
----

==== Compliant solution

[source,php,diff-id=1,diff-type=compliant]
----
define( 'SCRIPT_DEBUG', 1 );

// Compliant
define( 'CONSTANT_VALUE', 'intended value' );
echo CONSTANT_VALUE; // output: 'intended value'
----

== Resources

=== Documentation

* https://www.php.net/manual/en/language.constants.syntax.php[PHP Manual - Constants Syntax]
