This rule raised an issue when a `for` loop with `.append()` is used solely to create a copy of an existing list.


== Why is this an issue?

Python provides optimized, built-in ways to create lists from iterables or to copy existing lists:
* The `list(iterable)` constructor efficiently builds a new list containing all elements from the given iterable.
* The `existing_list.copy()` method efficiently creates a shallow copy of an existing list.

Iterating with a `for` loop and appending each element, without modification or filtering, to a new list achieves the same result but is generally:
* Slower: The overhead of the Python `for` loop and repeated `.append()` method calls is typically greater than the optimized internal implementation of `list()` or `.copy()`.
* More Verbose: Requires multiple lines of code compared to a single, clear function or method call.
* Less Idiomatic: Python programmers expect copies to be made using the standard methods.

== How to fix it

Replace the `for` loop with the appropriate built-in method.

If the intent is to create a new list from any iterable, including another list: use `list`:

[source,python]
----
original_list = [1,2,4]
new_list = list(original_list)
---- 

If the intent is to create a shallow copy of an existing list: use `.copy()`.

[source,python]
----
original_list = [1,2,4]
new_list = original_list.copy()
---- 

=== Code examples

==== Noncompliant code example

[source,python,diff-id=1,diff-type=noncompliant]
----
original_list = [10, 20, 30]
copied_list = []

for item in original_list:
    copied_list.append(item) # Noncompliant: the for loop solely copies the list without modification
----

==== Compliant solution

[source,python,diff-id=1,diff-type=compliant]
----
original_list = [10, 20, 30]
copied_list = list(original_list) # Compliant
----

== Resources
=== Documentation

* Python Documentation - https://docs.python.org/3/library/stdtypes.html#lists[`lists`]
* Python Documentation - https://docs.python.org/3/library/copy.html[`copy`]

