== How to fix it in .Net

=== Code examples

==== Noncompliant code example

These samples use a default TLS algorithm, which is a weak cryptographical
algorithm: TLSv1.0.

[source,vbnet,diff-id=1,diff-type=noncompliant]
----
Imports System.Net
Imports System.Security.Authentication
                
Public Module Example
    Public Sub Main()
        ServicePointManager.SecurityProtocol = SecurityProtocolType.Tls ' Noncompliant
    End Sub
End Module
----

[source,vbnet,diff-id=2,diff-type=noncompliant]
----
Imports System.Net.Http
Imports System.Security.Authentication
                
Public Module Example
    Public Sub Main()
        Dim Handler As New HttpClientHandler With { 
            .SslProtocols = SslProtocols.Tls ' Noncompliant
        }
    End Sub
End Module
----

==== Compliant solution

[source,vbnet,diff-id=1,diff-type=compliant]
----
Imports System.Net
Imports System.Security.Authentication
                
Public Module Example
    Public Sub Main()
        ServicePointManager.SecurityProtocol = _
            SecurityProtocolType.Tls12 _
            Or SecurityProtocolType.Tls13
    End Sub
End Module

----

[source,vbnet,diff-id=2,diff-type=compliant]
----
Imports System.Net.Http
Imports System.Security.Authentication
                
Public Module Example
    Public Sub Main()
        Dim Handler As New HttpClientHandler With { 
            .SslProtocols = SslProtocols.Tls12
        }
    End Sub
End Module
----

=== How does this work?

include::../../common/fix/fix.adoc[]
