<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>When an assignment is made in a return statement, the assigned value is returned and the assignment does take place. But if the assigned variable is local to the method, the assignment itself is simply wasted code.</p>
</div>
<div class="paragraph">
<p>Whether the variable is local or not, it makes for confusing code that will leave maintainers scratching their heads, wondering at the real intent.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-text" data-lang="text">public int getNumber() {
  int a = 0;
  return a = 8;  // Noncompliant
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-text" data-lang="text">public int getNumber() {
  int a = 0;
  return 8;
}</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_9_oct_2014_153505_ann_campbell_wrote">on 9 Oct 2014, 15:35:05 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>very tempted to create a 'wat' tag for this &#8230;&#8203;</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_9_oct_2014_163616_nicolas_peru_wrote">on 9 Oct 2014, 16:36:16 Nicolas Peru wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] Just one thing: what about the error message and the number of issues in a case like :</p>
</div>
<div class="listingblock">
<div class="content">
<pre>int foo() {
   return a = b = c = d = e = 8;
}</pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_on_9_oct_2014_170055_nicolas_peru_wrote">on 9 Oct 2014, 17:00:55 Nicolas Peru wrote:</h3>
<div class="paragraph">
<p>Please note that this case is already covered by http://jira.sonarsource.com/browse/RSPEC-1121 so this one is probably not relevant.</p>
</div>
</div>
</div>
</div>