== Why is this an issue?

The W3C specifications say comments should be defined using ``++/* ... */++``. The use of ``++//++`` is not supported on all browsers and can lead to unexpected results.


=== Noncompliant code example

[source,css]
----
// some comment
a { color: pink; }
----


=== Compliant solution

[source,css]
----
/* some comment */
a { color: pink; }
----


=== Exceptions

This rule ignores single line comments in ``++less++`` and ``++scss++`` files.



ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Replace this // by a valid comment: /* ... */


'''
== Comments And Links
(visible only on this page)

=== on 7 Jun 2018, 09:20:01 Ann Campbell wrote:
\[~alexandre.gigleux] the current title reads to me like it's about comment content, not formatting. I suggest: Comment formatting should be valid

endif::env-github,rspecator-view[]
