The W3C specifications say comments should be defined using ``++/* ... */++``. The use of ``++//++`` is not supported on all browsers and can lead to unexpected results.


== Noncompliant Code Example

----
// some comment
a { color: pink; }
----


== Compliant Solution

----
/* some comment */
a { color: pink; }
----


== Exceptions

This rule ignores single line comments in ``++less++`` and ``++scss++`` files.



ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
