This rule raises an issue when `getErrors()` is called on an SObject and the returned list size is immediately checked, instead of first using `hasErrors()` to determine if errors exist.

== Why is this an issue?

When working with SObject error handling, calling `getErrors()` and then checking the size of the returned list is less readable than first checking if errors exist using `hasErrors()`.

The `hasErrors()` method was introduced in Winter '21 specifically to provide a clean way to check if an SObject has any errors before retrieving them. This approach offers several benefits:

* **Improved readability**: The intent is clearer when you first check `hasErrors()` before accessing the actual errors
* **Better performance**: Avoiding the creation and size check of potentially empty lists
* **Consistent pattern**: Following the established pattern of checking existence before accessing collections

When you call `getErrors()` directly and then check `errors.size() > 0`, you're essentially doing what `hasErrors()` already does, but in a more verbose way.

=== What is the potential impact?

This issue has minimal impact on functionality but affects code readability and maintainability. Using the recommended pattern makes the code more expressive and follows Salesforce best practices for error handling.

== How to fix it

Use `hasErrors()` to check if errors exist before calling `getErrors()` to retrieve them. This makes the code more readable and follows the intended usage pattern.

=== Code examples

==== Noncompliant code example

[source,apex,diff-id=1,diff-type=noncompliant]
----
Account a = new Account();
List<Database.Error> errors = a.getErrors(); // Noncompliant
if (errors.size() > 0) {
    // Process errors
    for (Database.Error error : errors) {
        System.debug(error.getMessage());
    }
}
----

==== Compliant solution

[source,apex,diff-id=1,diff-type=compliant]
----
Account a = new Account();
if (a.hasErrors()) {
    List<Database.Error> errors = a.getErrors();
    // Process errors
    for (Database.Error error : errors) {
        System.debug(error.getMessage());
    }
}
----

== Resources

=== Documentation

 * SObject Methods - hasErrors() - https://developer.salesforce.com/docs/atlas.en-us.apexref.meta/apexref/apex_methods_system_sobject.htm#apex_System_SObject_hasErrors[Official Salesforce documentation for the hasErrors() method]

 * SObject Methods - getErrors() - https://developer.salesforce.com/docs/atlas.en-us.apexref.meta/apexref/apex_methods_system_sobject.htm#apex_System_SObject_getErrors[Official Salesforce documentation for the getErrors() method]

 * Apex Testing with SObject Error Methods - https://www.apexhours.com/improve-apex-testing-with-new-sobject-error-methods[Guide on using the new SObject error methods introduced in Winter '21]
