== Why is this an issue?

The ``++foreach++`` statement was introduced in the C# language prior to generics to make it easier to work with the non-generic collections available at that time such as ``++ArrayList++``. The ``++foreach++`` statements allows you to downcast elements of a collection of ``++Object++``s to any other type. The problem is that to achieve the cast, the ``++foreach++`` statements silently performs ``++explicit++`` type conversion, which at runtime can result in an ``++InvalidCastException++``.


C# code iterating on generic collections or arrays should not rely on ``++foreach++`` statement's silent ``++explicit++`` conversions.


=== Noncompliant code example

[source,csharp]
----
public class Fruit { }
public class Orange : Fruit { }
public class Apple : Fruit { }

class MyTest
{
  public void Test()
  {
    var fruitBasket = new List<Fruit>();
    fruitBasket.Add(new Orange());
    fruitBasket.Add(new Orange());
    // fruitBasket.Add(new Apple());  // uncommenting this line will make both foreach below throw an InvalidCastException

    foreach (Fruit fruit in fruitBasket) 
    {
      var orange = (Orange)fruit; // This "explicit" conversion is hidden within the foreach loop below
      ...
    }

    foreach (Orange orange in fruitBasket) // Noncompliant
    {
      ...
    } 
  }
}
----


=== Compliant solution

[source,csharp]
----
var fruitBasket = new List<Orange>();
fruitBasket.Add(new Orange());
fruitBasket.Add(new Orange());
// fruitBasket.Add(new Apple());  // uncommenting this line won't compile

foreach (Orange orange in fruitBasket)
{
  ...
} 
----

or


[source,csharp]
----
var fruitBasket = new List<Fruit>();
fruitBasket.Add(new Orange());
fruitBasket.Add(new Orange());
fruitBasket.Add(new Apple());

foreach (Orange orange in fruitBasket.OfType<Orange>())
{
  ...
} 
----


=== Exceptions

The rule ignores iterations on collections of ``++object++``s. This includes legacy code that uses ``++ArrayList++``. Furthermore, the rule does not report on cases when user defined conversions are being called. 


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Either change the type of "xxx" to "yyy" or iterate on a generic collection of type "zzz".

(e.g. make "element" a "Base".)


'''
== Comments And Links
(visible only on this page)

=== on 1 Jul 2015, 16:23:11 Ann Campbell wrote:
\[~tamas.vajk] as I was writing this it occurred to me that someone might write this code thinking they were filtering the array (or whatever was being ``++foreach++``ed) so that the inner code would only be called for instances of the derived type. 


Am I correct in thinking that we can/should recommend LINQ usage to accomplish that?

=== on 2 Jul 2015, 06:52:37 Tamas Vajk wrote:
\[~ann.campbell.2] Yes, you are right. If the dev wants to filter the list by type, the ``++arr.OfType<Raspberry>()++`` should be used in the ``++foreach++``.

=== on 2 Jul 2015, 12:43:33 Ann Campbell wrote:
Please double-check me [~tamas.vajk]

=== on 2 Jul 2015, 13:44:42 Tamas Vajk wrote:
\[~ann.campbell.2] looks good.

=== on 29 Jul 2015, 11:59:55 Dinesh Bolkensteyn wrote:
Eric Lippert says he implemented this rule on the C# compiler, but then removed it because of the huge amount of false positives in Microsoft's own C# code, so this should be "MAJOR" and "suspicious" at most.


http://ericlippert.com/2013/07/22/why-does-a-foreach-loop-silently-insert-an-explicit-conversion/

endif::env-github,rspecator-view[]
