<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>There&#8217;s no point in redundantly defining an <code>abstract</code> method with the same signature as a method in an <code>interface</code> that the class <code>implements</code>. Any concrete child classes will have to implement the method either way.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-text" data-lang="text">public interface Reportable {
  String getReport();
}

public abstract class AbstractRuleReport implements Reportable{
  public abstract String getReport();  // Noncompliant

  // ...
}</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>"xxx" is defined in the "yyy" interface and can be removed from this class.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_16_jun_2015_132104_nicolas_peru_wrote">on 16 Jun 2015, 13:21:04 Nicolas Peru wrote:</h3>
<div class="paragraph">
<p>Looks good.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_17_jan_2018_151123_michal_barczyk_wrote">on 17 Jan 2018, 15:11:23 Michal Barczyk wrote:</h3>
<div class="paragraph">
<p>In C#, it is mandatory to add method declaration - even if it is abstract. This fails with CS0535:</p>
</div>
<div class="paragraph">
<p>public interface I</p>
</div>
<div class="paragraph">
<p>{</p>
</div>
<div class="literalblock">
<div class="content">
<pre>void Do();</pre>
</div>
</div>
<div class="paragraph">
<p>}</p>
</div>
<div class="paragraph">
<p>public abstract class A : I {}</p>
</div>
</div>
</div>
</div>