The new style of class creation, with the declaration of a parent class, created a unified object model in Python, so that the type of an instantiated class is equal to its class. In Python 2.2-2.7, this is not the case for old-style classes. In Python 3+ all classes are new-style classes. However, since the behavior can differ from 2.2+ to 3+, explicitly inheriting from ``object`` (if there is no better candidate) is recommended.


== Noncompliant Code Example

----
class MyClass():
    pass
----


== Compliant Solution

----
class MyClass(object):
    pass
----


