HashiCorp Vault is a popular open-source tool used for securely storing and accessing sensitive data such as passwords, API keys, certificates, and encryption keys. It provides a centralized solution for managing secrets and helps organizations enforce security best practices.

include::../../../shared_content/secrets/description.adoc[]

== Why is this an issue?

include::../../../shared_content/secrets/rationale.adoc[]

=== What is the potential impact?

If a HashiCorp Vault token is compromised, it can have serious consequences for the security of the system and the sensitive data stored within the Vault. Here are some potential consequences:

include::../../../shared_content/secrets/impact/personal_data_compromise.adoc[]

include::../../../shared_content/secrets/impact/financial_loss.adoc[]

include::../../../shared_content/secrets/impact/non_repudiation.adoc[]

include::../../../shared_content/secrets/impact/security_downgrade.adoc[]

== How to fix it

include::../../../shared_content/secrets/fix/revoke.adoc[]

include::../../../shared_content/secrets/fix/recent_use.adoc[]

include::../../../shared_content/secrets/fix/vault.adoc[]

=== Code examples

==== Noncompliant code example

[source,python,diff-id=1,diff-type=noncompliant]
----
import hvac

client = hvac.Client(url='https://vault.example.com', token='hvb.AAAAAQJyBEVF-vTWUrg0hcoIPuvKjjNxXXZ5MfsYVg2gJ0fGZpVi0IGTFfh4TqsoQIWaocNRXD1qzGXvhIHWJBM_rWU9YJY8sXOYVy_s1JAHasXJwGmZ_fBLJfSG6aCwQkCGwtAhYw') # Noncompliant

secret = client.secrets.kv.v2.read_secret_version(path='secret/myapp')
data = secret['data']
username = data.get('username')
password = data.get('password')
----

==== Compliant solution

[source,python,diff-id=1,diff-type=compliant]
----
import hvac

client = hvac.Client(url='https://vault.example.com', token=os.environ.get('VAULT_TOKEN'))

secret = client.secrets.kv.v2.read_secret_version(path='secret/myapp')
data = secret['data']
username = data.get('username')
password = data.get('password')
----


//=== How does this work?

//=== Pitfalls

//=== Going the extra mile

== Resources

=== Documentation

https://developer.hashicorp.com/vault/docs/concepts/tokens#token-types[Hashicorp API Documentation - Tokens]

https://developer.hashicorp.com/vault/tutorials/tokens/tokens[Hashicorp API Tutorial - Tokens]

https://developer.hashicorp.com/vault/tutorials/tokens/batch-tokens[Hashicorp API Tutorial - Batch tokens]

include::../../../shared_content/secrets/resources/standards.adoc[]

//=== Benchmarks
