include::../rule.adoc[]

=== Code examples

==== Noncompliant code example

With the default threshold of 3:

[source,abap,diff-id=1,diff-type=noncompliant]
----
WRITE: / 'Firstname'.
*...
WRITE: / 'Firstname'.
*...
WRITE: / 'Firstname'.
----

==== Compliant solution

[source,abap,diff-id=1,diff-type=compliant]
----
CONSTANTS: fname    TYPE c LENGTH 9 VALUE 'Firstname',

WRITE: / fname.
*...
WRITE: / fname.
*...
WRITE: / fname.
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../parameters.adoc[]

include::../highlighting.adoc[]

'''

endif::env-github,rspecator-view[]
