== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

With the default threshold of 3:

[source,abap]
----
WRITE: / 'Firstname'.
*...
WRITE: / 'Firstname'.
*...
WRITE: / 'Firstname'.
----

=== Compliant solution

[source,abap]
----
CONSTANTS: fname    TYPE c LENGTH 9 VALUE 'Firstname',

WRITE: / fname.
*...
WRITE: / fname.
*...
WRITE: / fname.
----

include::../exceptions.adoc[]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../parameters.adoc[]

include::../highlighting.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
