include::../description.adoc[]

include::../how-to-fix-it.adoc[]

=== Code examples

==== Noncompliant code example

[source,sql,diff-id=1,diff-type=noncompliant]
----
BEGIN
  FOR l_counter IN 1..5 -- Noncompliant, 5 is a magic number
  LOOP
    ...
  END LOOP;
END;
----

==== Compliant solution

[source,sql,diff-id=1,diff-type=compliant]
----
DECLARE
  co_number_of_cycles CONSTANT NUMBER := 5;
BEGIN
  FOR l_counter IN 1..co_number_of_cycles -- Compliant
  LOOP
    ...
  END LOOP;
END;
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

=== Parameters

.exceptions
****

----
-1,0,1
----

Comma separated list of allowed exceptions
****


'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
