== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,php]
----
"/Hello,   world!/"
----

=== Compliant solution

[source,php]
----
"/Hello, {3}world!/"
----

=== Exceptions

If the extended modifier is set, whitespaces are ignored. In this case no issue should be triggered, because the whitespaces may be intended to improve readability.

include::../implementation.adoc[]
