include::../description.adoc[]

== Noncompliant Code Example

[source,php]
----
"/Hello,   world!/"
----

== Compliant Solution

[source,php]
----
"/Hello, {3}world!/"
----

== Exceptions

If the extended modife is set, whitespaces are ignored. In this case no issue should be triggered,
because the whitespaces may be intended to improve readability.

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

== Message

Replace spaces with quantifier ``++{n}++``.

(n should be dynamic and be the number of spaces) 

== Highlighting

The spaces that would be replaced with a quantifier. For example in 3 spaces, highlight the last 2.

'''

endif::env-github,rspecator-view[]
