include::../description.adoc[]

== Noncompliant Code Example

----
"/Hello,   world!/"
----

== Compliant Solution

----
"/Hello, {3}world!/"
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

== Message

Replace spaces with quantifier ``++{n}++``.

(n should be dynamic and be the number of spaces) 

== Highlighting

The spaces that would be replaced with a quantifier. For example in 3 spaces, highlight the last 2.

'''

endif::env-github,rspecator-view[]