include::../description.adoc[]

== Noncompliant Code Example

[source,php]
----
"/Hello,   world!/"
----

== Compliant Solution

[source,php]
----
"/Hello, {3}world!/"
----

== Exceptions

If the extended modifier is set, whitespaces are ignored. In this case no issue should be triggered, because the whitespaces may be intended to improve readability.

include::../implementation.adoc[]
