== Why is this an issue?

Various aspects of ``++ctime++`` are implementation-defined or unspecified, such as the formats of times. 


=== Noncompliant code example

[source,cpp]
----
#include <ctime>  /* Noncompliant */

void f()
{
  clock();
}
----


== Resources

* MISRA {cpp}:2008, 18-0-4 - The time handling functions of library <ctime> shall not be used.


ifdef::env-github,rspecator-view[]
'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
