== Why is this an issue?

Various aspects of ``++ctime++`` are implementation-defined or unspecified, such as the formats of times. 


=== Noncompliant code example

[source,cpp]
----
#include <ctime>  /* Noncompliant */

void f()
{
  clock();
}
----


== Resources

* MISRA {cpp}:2008, 18-0-4 - The time handling functions of library <ctime> shall not be used.


ifdef::env-github,rspecator-view[]
'''
== Comments And Links
(visible only on this page)

=== on 31 Mar 2015, 19:00:27 Evgeny Mandrikov wrote:
\[~ann.campbell.2] implementation seems more complete (SQALE, description) than this spec.

=== on 9 Apr 2015, 14:03:05 Evgeny Mandrikov wrote:
\[~ann.campbell.2] Compilers not mentioned in MISRA, so I have real doubts that there is dependency on compiler - most likely MISRA refers to differences in implementation of standard library and compilers can be used with various libraries.

=== on 10 Apr 2015, 13:16:36 Ann Campbell wrote:
\[~evgeny.mandrikov] my understanding of "undefined behavior" is that what actually happens is left up to the compiler.


If you like, I can revert this to the MISRA text verbatim, but I was trying to give a little more value.

=== on 13 Apr 2015, 19:41:25 Evgeny Mandrikov wrote:
\[~ann.campbell.2] I'm wondering why blocker, but not active by default? Note that in implementation currently major and not active.

=== on 13 Apr 2015, 19:44:24 Evgeny Mandrikov wrote:
\[~ann.campbell.2] And answering on previous question - I strongly believe that in this is case "undefined behavior" means that it is left up to the implementation of standard library, which is not necessarily comes with compiler.

endif::env-github,rspecator-view[]
