include::../why.adoc[]

include::../how.adoc[]

=== Code examples

==== Noncompliant code example

[source,html,diff-id=1,diff-type=noncompliant]
----
<iframe src="https://openweathermap.org"></iframe> <!-- Noncompliant -->
----

==== Compliant solution

[source,html,diff-id=1,diff-type=compliant]
----
<iframe src="https://openweathermap.org" title="Weather forecasts, nowcasts and history"></iframe>
----

include::../resources.adoc[]
