Python concatenates adjacent string or byte literals at compile time. It means that ``++"a" "b"++`` is equivalent to ``++"ab"++``. This is sometimes used to split a long string on multiple lines. However an implicit string concatenation can also be very confusing. In the following contexts it might indicate that a comma was forgotten:

* when the two strings are on the same line it looks like a badly formatted tuple. Parenthesises are not mandatory to create a tuple, only the comma is.
* when the strings are in a list, set or tuple.


== Noncompliant Code Example

----
def func():
    return "item1" "item2"  # Noncompliant

["1",
 "2"  # Noncompliant
 "3",
 "a very very very"  # Noncompliant
 "very very long string",
 "4"]
----


== Compliant Solution

----
def func():
    return "item1", "item2"

["1",
 "2",
 "3",
 "a very very very" +
 "very very long string",
 "4"]
----


== Exceptions

No issue will be raised when there is a visible reason for the string concatenation:

* when the quotes used for both strings are different. This can be used to avoid escaping quotes
* when the strings or bytes have different prefixes, i.e. "f" for f-strings, "r" for raw, "u" for unicode and no prefix for normal strings.
* when strings are visibly split to avoid long lines of code. (Example: the first string ends with a space, punctuation or ``++\n++``).

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

include::highlighting.adoc[]

endif::env-github,rspecator-view[]
