== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,scala]
----
  if (true) {
    doSomething
  }
  // ...
  if (false) {
    doSomethingElse
  }
----

=== Compliant solution

[source,scala]
----
  doSomething
  // ...
----

include::../see.adoc[]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
