include::../description.adoc[]

== Noncompliant Code Example

[source,scala]
----
  if (true) {
    doSomething
  }
  // ...
  if (false) {
    doSomethingElse
  }
----

== Compliant Solution

[source,scala]
----
  doSomething
  // ...
----

include::../see.adoc[]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
