== How to fix it in Cryptodome

=== Code examples

The example uses a hardcoded IV as a nonce, which causes AES-CCM to be insecure. To fix it, a nonce is randomly generated instead.

==== Noncompliant code example

[source,python,diff-id=101,diff-type=noncompliant]
----
from Crypto.Cipher import AES

def encrypt(key, plaintext):
    nonce   = '7cVgr5cbdCZV'.encode('utf-8')
    cipher = AES.new(key, AES.MODE_CCM, nonce)

    cipher.encrypt(plaintext)  # Noncompliant
----

==== Compliant solution

[source,python,diff-id=101,diff-type=compliant]
----
from Crypto.Cipher import AES

def encrypt(key, plaintext):
    cipher  = AES.new(key, AES.MODE_CCM)

    cipher.encrypt_and_digest(plaintext)
----

=== How does this work?

When using AES-GCM or AES-CCM, NIST recommends a 96-bit length nonce using a deterministic approach or at least 96 bits using a 'Random Bit Generator (RBG)'.

include::../../common/fix/randomized-nonce.adoc[]

include::../../common/fix/deterministic-nonce.adoc[]
