{cpp} language is highly elastic regarding the placement of type specifiers on type declarations. This makes it possible to place a ``++const++`` qualifier before the type ("west const") or after it ("east const").


Mixing both ``++const++`` placement in a single project or, even worse, in a single file, makes the code harder to understand. It will require more effort to notice that the object cannot be modified.


This rule checks if the ``++const++`` qualifier placement matches the configuration.


== Noncompliant Code Example

With default value of "eastConst" parameter.

----
long double const tau = 6.28L; // Noncompliant
----


== Compliant Solution

----
const long double tau = 6.28L;
----


== See

* https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#Rl-const[{cpp} Core Guidelines NL.26] - Use conventional const notation

