<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>While the <code>OCCURS</code> keyword still works, it is slower than its more modern replacement: <code>DIM</code>, which should be used instead.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-rpg" data-lang="rpg">D cussls          ds                  Occurs(200)
D CustNo                         7p 0
D MonthlySls                    11P 2</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-rpg" data-lang="rpg">D cussls          ds                  dim(200)
D CustNo                         7p 0
D MonthlySls                    11P 2</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Use a "DIM" instead.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_8_apr_2015_114045_ann_campbell_wrote">on 8 Apr 2015, 11:40:45 Ann Campbell wrote:</h3>
<div class="paragraph">
<p><a href="http://www.bmeyers.net/faqs/14-tips/32-rpg-iv-style?start=3" class="bare">http://www.bmeyers.net/faqs/14-tips/32-rpg-iv-style?start=3</a></p>
</div>
</div>
<div class="sect2">
<h3 id="_on_21_apr_2015_140047_pierre_yves_nicolas_wrote">on 21 Apr 2015, 14:00:47 Pierre-Yves Nicolas wrote:</h3>
<div class="paragraph">
<p>From what I read, DIM and OCCURS perform similarly.</p>
</div>
<div class="paragraph">
<p><a href="http://forums.iprodeveloper.com/forums/aft/45090" class="bare">http://forums.iprodeveloper.com/forums/aft/45090</a></p>
</div>
<div class="paragraph">
<p>I don&#8217;t think this rule should be both critical and activated by default.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_22_apr_2015_110223_ann_campbell_wrote">on 22 Apr 2015, 11:02:23 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>tough choice, [~pierre-yves.nicolas]. I picked inactive by default</p>
</div>
</div>
</div>
</div>