== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,abap]
----
if param = 1.
  Statement.
elseif param = 2.
  Statement.
elseif param = 1.  // Noncompliant
 Statement.
endif.
----

=== Compliant solution

[source,abap]
----
if param = 1.
  Statement.
elseif param = 2.
  Statement.
elseif param = 3.
 Statement.
endif.
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../highlighting.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
