<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when retrieving a single record by primary key ID using <code>where(id: id).take!</code>, <code>find_by!(id: id)</code>, or <code>find_by_id!(id)</code> instead of the more direct <code>find(id)</code> method.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>ActiveRecord provides the <code>find</code> method specifically for retrieving records by their primary key. This method is more direct, readable, and expressive than using generic query methods like <code>where.take!</code> or <code>find_by!</code> when looking up by ID.</p>
</div>
<div class="paragraph">
<p>Using <code>find</code> makes the intent clearer - you&#8217;re looking for a specific record by its primary key. The alternatives are more verbose and don&#8217;t clearly communicate that you&#8217;re doing a primary key lookup. Additionally, <code>find</code> is optimized for primary key lookups and may offer better performance.</p>
</div>
<div class="paragraph">
<p>All these methods have the same behavior when the record is not found - they raise <code>ActiveRecord::RecordNotFound</code>. However, <code>find</code> is the conventional and preferred approach in the Rails community for this specific use case.</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="paragraph">
<p>Using verbose alternatives to <code>find</code> reduces code readability and doesn&#8217;t follow Rails conventions. While the functionality is equivalent, the code becomes less expressive about its intent and may be slightly less performant for primary key lookups.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it_in_rails">How to fix it in Rails</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Replace <code>where(id: id).take!</code> with <code>find(id)</code> for primary key lookups. Both methods raise <code>ActiveRecord::RecordNotFound</code> when the record is not found, but <code>find</code> is more direct and conventional.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-ruby" data-lang="ruby">User.where(id: user_id).take! # Noncompliant</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-ruby" data-lang="ruby">User.find(user_id)</code></pre>
</div>
</div>
<div class="paragraph">
<p>Replace <code>find_by!(id: id)</code> with <code>find(id)</code> for primary key lookups. The <code>find</code> method is specifically designed for this purpose and is more readable.</p>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_code_examples_2">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example_2">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-ruby" data-lang="ruby">User.find_by!(id: user_id) # Noncompliant</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution_2">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-ruby" data-lang="ruby">User.find(user_id)</code></pre>
</div>
</div>
<div class="paragraph">
<p>Replace the deprecated <code>find_by_id!(id)</code> with <code>find(id)</code>. The <code>find</code> method is the modern, preferred approach for primary key lookups.</p>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_code_examples_3">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example_3">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-ruby" data-lang="ruby">User.find_by_id!(user_id) # Noncompliant</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution_3">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-ruby" data-lang="ruby">User.find(user_id)</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Rails Guides - Active Record Query Interface - <a href="https://guides.rubyonrails.org/active_record_querying.html#retrieving-a-single-object">Official Rails documentation on retrieving single objects, including the find method</a></p>
</li>
<li>
<p>Rails API Documentation - ActiveRecord::FinderMethods#find - <a href="https://api.rubyonrails.org/classes/ActiveRecord/FinderMethods.html#method-i-find">API documentation for the find method in ActiveRecord</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>