This rule raises an issue when a function is declared async but does not use any asynchronous features.


== Why is this an issue?

Declaring a function with `async def` signals that it's a coroutine, typically because it needs to use `await` for non-blocking operations (like I/O), or employs other asynchronous features like `async for` or `async with`.

If none of these asynchronous mechanisms are utilized within the function's body, the `async` declaration is superfluous.

Using `async` unnecessarily can:

* **Reduce Code Clarity:** It misleads developers into thinking the function performs asynchronous operations when it doesn't, making the codebase harder to understand and reason about.
* **Introduce Minor Overhead:** Calling an `async def` function creates a coroutine object. While modern Python is efficient, this still incurs more overhead than a direct synchronous function call if the asynchronicity isn't actually leveraged. The function's body will execute synchronously until an `await` (if any) is encountered. If there's no `await` or other async feature, it essentially runs synchronously but wrapped as a coroutine.
* **Complicate Usage:** Callers must use `await` (or schedule it as a task) to execute the coroutine and get its result. This is an unnecessary ceremony if the function itself isn't truly asynchronous.


== How to fix it

Either remove the `async` keyword, or start using the appropriate asynchronous features.

=== Code examples

==== Noncompliant code example

[source,python,diff-id=1,diff-type=noncompliant]
----
async def my_function():  # Noncompliant
    print("Hello from my function")
----

==== Compliant solution

[source,python,diff-id=1,diff-type=compliant]
----
def my_function():  # Compliant
    print("Hello from my function")
----

==== Noncompliant code example

[source,python,diff-id=2,diff-type=noncompliant]
----
def another_function(): ...

async def my_function():  # Noncompliant
    return another_function()
----

==== Compliant solution

[source,python,diff-id=2,diff-type=compliant]
----
async def another_function(): ...

async def my_function():  # Compliant
    return await another_function()
----


== Resources
=== Articles & blog posts
* https://journal.stuffwithstuff.com/2015/02/01/what-color-is-your-function/[What Color Is Your Function? - Bob Nystrom]

ifdef::env-github,rspecator-view[]
== Implementation Specification
(visible only on this page)

=== Message
Remove the `async` keyword or use the appropriate asynchronous features.

=== Highlighting
Primary: the entire function body
Secondary: the `async` keyword of the function declaration

endif::env-github,rspecator-view[]
