== Why is this an issue?

A `for` loop with a stop condition that can never be reached, such as one with a counter that moves in the wrong direction, will run infinitely. While there are occasions when an infinite loop is intended, the convention is to construct such loops as `while` loops. More typically, an infinite `for` loop is a bug. 

== How to fix it

=== Code examples

==== Noncompliant code example

[source,javascript,diff-id=1,diff-type=noncompliant]
----
for (var i = 0; i < strings.length; i--) { // Noncompliant
  //...
}
----

==== Compliant solution

[source,javascript,diff-id=1,diff-type=compliant]
----
for (var i = 0; i < strings.length; i++) {
  //...
}
----

== Resources

=== Documentation

* https://en.wikipedia.org/wiki/Integer_overflow[Integer overflow]


include::../rspecator.adoc[]