== Why is this an issue?

By default case insensitivity only affects letters in the ASCII range. This can be changed by either passing ``++Pattern.UNICODE_CASE++`` or ``++Pattern.UNICODE_CHARACTER_CLASS++`` as an argument to ``++Pattern.compile++`` or using ``++(?u)++`` or ``++(?U)++`` within the regex.


If not done, regular expressions involving non-ASCII letters will still handle those letters as being case sensitive.


=== Noncompliant code example

[source,java]
----
Pattern.compile("söme pättern", Pattern.CASE_INSENSITIVE);
str.matches("(?i)söme pättern");
str.matches("(?i:söme) pättern");
----


=== Compliant solution

[source,java]
----
Pattern.compile("söme pättern", Pattern.CASE_INSENSITIVE | Pattern.UNICODE_CASE);
str.matches("(?iu)söme pättern");
str.matches("(?iu:söme) pättern");

// UNICODE_CHARACTER_CLASS implies UNICODE_CASE
Pattern.compile("söme pättern", Pattern.CASE_INSENSITIVE | Pattern.UNICODE_CHARACTER_CLASS);
str.matches("(?iU)söme pättern");
str.matches("(?iU:söme) pättern");
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

* when Pattern.CASE_INSENTIVE was used: Also use "Pattern.UNICODE_CASE" to correctly handle non-ASCII letters.
* when the "i" flag was used: Also use the "u" flag to correctly handle non-ASCII letters.


=== Highlighting

The use of ``++Pattern.CASE_INSENTIVE++`` or the ``++i++`` in ``++(?i)++`` or ``++(?i:...)++``


endif::env-github,rspecator-view[]
