<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>When casting a raw pointer to a slice, the produced raw pointer does not correctly update its length metadata, leading to potential issues such as creating a slice with less data than intended or more data, risking undefined behavior upon dereferencing.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Use <code>ptr::slice_from_raw_parts</code> to construct a slice from a data pointer and the correct length.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-rust" data-lang="rust">let a = [1_u8, 2, 3, 4];
let p = &amp;a as *const [u8] as *const [u32];
unsafe {
    println!("{:?}", &amp;*p); // Noncompliant: Undefined behavior
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-rust" data-lang="rust">let a = [1_i32, 2, 3, 4];
let old_ptr = &amp;a as *const [i32];
// The data pointer is cast to a pointer to the target `u8` not `[u8]`
// The length comes from the known length of 4 i32s times the 4 bytes per i32
let new_ptr = core::ptr::slice_from_raw_parts(old_ptr as *const u8, 16);
unsafe {
    println!("{:?}", &amp;*new_ptr);
}</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Clippy Lints - <a href="https://rust-lang.github.io/rust-clippy/master/index.html#cast_slice_different_sizes" class="bare">https://rust-lang.github.io/rust-clippy/master/index.html#cast_slice_different_sizes</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>