Unused parameters are misleading. Whatever the value passed to such parameters is, the behavior will be the same.

== Noncompliant Code Example

----
function doSomething(a:int, b:int):void      // "b" is unused
{
  compute(a);
}
----

== Compliant Solution

----
function doSomething(a:int):void
{
  compute(a);
}
----

== Exceptions

The following cases are ignored

* event handlers.
* overriding methods.
* all methods in classes implementing one or more interfaces.
* methods which are empty or where the body consists of a single comment or a single ``++throw++`` statement (i.e. where the intention is apparently to simulate an abstract class).

----
override function doSomething(a:int):void {    // ignored
  compute(a);
}

...

class AbstractSomething {
  public function doSomething(a:int) {  // ignored
    throw new IllegalOperationError("doSomething() is abstract");
  }

...

interface I {
  function action(a:int, b:int);
}

class C extends I {
  function action(a:int, b:int) { // ignored
    return doSomethignWith(a);
  }
}

function clickHandler(event:MouseEvent):void { // ignored
   trace("click");
}
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
