== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,vbnet]
----
Class Foo
    Public Foo = 42          ' Noncompliant
End Class
----


=== Compliant solution

[source,vbnet]
----
Class Foo
    Public Property Foo = 42 ' Compliant
End Class
----


=== Exceptions

``++Shared++`` and ``++Const++`` fields are ignored.


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
