Triple slash (`///`) should be used for documentation comments, instead of multi-line documentation comments (``++/** ... */++``).

== Why is this an issue?

Dart supports two types of documentation comments: 

* single-line (`/// ...`)
* and multi-line (``++/** ... */++``)

Single-line comments have been chosen as the standard for documentation comments in Dart, since they are generally easier to vertically align and clearer to read.

They are generally easier to align because following lines in multi-line comments are typically indented by the ``++*++`` character, to make it clear that they are part of the same comment block. 

However, this makes each of the following line of the multi-line comment differently indented than the first line, resulting in a higher future maintenance cost to keep it correctly aligned.

[source,dart]
----
/** First comment line
 *  Following comment line, 1 char-indentation
 *  ...
 */
----

== How to fix it

Replace each line of the documentation comment with a single-line comment starting with `\\\`. Remove the ``++*++`` character and related spacing that you may have at the beginning of each of the following line.

=== Code examples

==== Noncompliant code example

[source,dart,diff-id=1,diff-type=noncompliant]
----
/** First comment line
 *  Following comment line, 1 char-indentation
 */
----

==== Compliant solution

[source,dart,diff-id=1,diff-type=compliant]
----
/// First comment line
/// Following comment line, 1 char-indentation
----

== Resources

* Dart Docs - https://dart.dev/tools/linter-rules/slash_for_doc_comments[Dart Linter rule - slash_for_doc_comments]
* Dart Docs - https://dart.dev/language/comments#documentation-comments[Language - Documentation comments]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Use the end-of-line form ('///') for doc comments.

=== Highlighting

The entire comment, including the ``++/**++`` and ``++*/++`` delimiters.

'''
== Comments And Links
(visible only on this page)

endif::env-github,rspecator-view[]
