<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Final fields can&#8217;t be reassigned. There&#8217;s often a need to declare some fields that aren&#8217;t intended to be changed. To avoid confusion, such fields should be marked <code>final</code> to make their intended use explicit, and to prevent future maintainers from inadvertently changing their use.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Mark the given field with the <code>final</code> modifier.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-dart" data-lang="dart">class Person {
  var _birthYear; // Noncompliant, _birthYear never updated in this class
  var _age = 0;

  Person(this.birthYear);

  void updateAge(int newAge) {
    _age = newAge;
  }
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-dart" data-lang="dart">class Person {
  final _birthYear;
  var _age = 0;

  Person(this.birthYear);

  void updateAge(int newAge) {
    _age = newAge;
  }
}</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p><a href="https://dart.dev/tools/linter-rules/prefer_final_fields">Dart Lint rule</a></p>
</li>
</ul>
</div>
</div>
</div>