<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when a method or closure parameter is reassigned to a new value within the method or closure body.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Reassigning method or closure parameters within the method body is a confusing practice that makes code harder to understand and maintain.</p>
</div>
<div class="paragraph">
<p>When you reassign a parameter, you lose access to the original value that was passed to the method. This can lead to several problems:</p>
</div>
<div class="ulist">
<ul>
<li>
<p><strong>Reduced readability</strong>: Other developers reading your code expect parameters to keep their original values throughout the method.</p>
</li>
<li>
<p><strong>Debugging difficulties</strong>: When debugging, you cannot inspect the original parameter value after it has been reassigned.</p>
</li>
<li>
<p><strong>Potential bugs</strong>: You might accidentally use the reassigned value when you meant to use the original parameter value.</p>
</li>
<li>
<p><strong>Violation of expectations</strong>: Parameters are meant to represent input values, not working variables.</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>This practice is particularly problematic in Groovy closures, where parameter reassignment can be even more confusing due to the concise syntax.</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="paragraph">
<p>This issue reduces code readability and maintainability. It can lead to bugs when the original parameter value is needed later in the method, and makes debugging more difficult since the original input values are lost.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Use a temporary variable instead of reassigning the parameter.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-groovy" data-lang="groovy">void processData(String data) {
    data = data.trim() // Noncompliant
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-groovy" data-lang="groovy">void processData(String data) {
    trimmedData = data.trim() // Compliant
}</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">

</div>
</div>