<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when multiple branches of a regex alternative match the same input.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>If an alternative in a regular expression only matches things that are already matched by another alternative, that alternative is redundant and serves no purpose.</p>
</div>
<div class="paragraph">
<p>In the best case this means that the offending subpattern is merely redundant and should be removed. In the worst case it&#8217;s a sign that this regex does not match what it was intended to match and should be reworked.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">r"[ab]|a"   # Noncompliant: the "|a" is redundant because "[ab]" already matches "a"
r".*|a"     # Noncompliant: .* matches everything, so any other alternative is redundant</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">r"[ab]"
r".*"</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>