<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule allows banning the use of some programs or procedures.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="paragraph">
<p>With <code>programName</code> configured with <code>SORT</code>:</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-jcl" data-lang="jcl">//EXEC PGM=SORT
//EXEC PROC=SORT
//EXEC SORT</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove the use of this forbidden program or procedure. Use {replacementProgramName} instead.</p>
</div>
</div>
<div class="sect2">
<h3 id="_parameters">Parameters</h3>
<div class="sidebarblock">
<div class="content">
<div class="title">progName</div>
<div class="paragraph">
<p>Name of the forbidden program or procedure. Use a regex to forbid multiple programs.</p>
</div>
</div>
</div>
<div class="sidebarblock">
<div class="content">
<div class="title">replacementProgramName</div>
<div class="paragraph">
<p>Name of the program to use as a replacement of the forbidden one.</p>
</div>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>First use of forbidden class</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_deprecates_s1212">deprecates: <a data-rspec-id="S1212" class="rspec-auto-link">S1212</a></h3>

</div>
<div class="sect2">
<h3 id="_is_duplicated_by_s5338">is duplicated by: <a data-rspec-id="S5338" class="rspec-auto-link">S5338</a></h3>

</div>
</div>
</div>