== Why is this an issue?

Namespace prefixes that are declared but not used simply clutter the document and confuse maintainers. Such prefixes should be removed.


=== Noncompliant code example

[source,xml]
----
<xs:schema targetNamespace="http://www.codeSamples.com/fruit"
           xmlns:xs="http://www.w3.org/2001/XMLSchema"
           xmlns:animal="http://www.codeSamples.com/animal"
           elementFormDefault="qualified" version="1">  <!-- Noncompliant -->
  <xs:element name="fruit">
    <xs:complexType>
      <xs:sequence>
        ...                        <!-- assume that no use of prefix "animal" is made here -->
      </xs:sequence>
    </xs:complexType>
  </xs:element>
</xs:schema>
----


=== Compliant solution

[source,xml]
----
<xs:schema targetNamespace="http://www.codeSamples.com/fruit"
           xmlns:xs="http://www.w3.org/2001/XMLSchema"
           elementFormDefault="qualified" version="1">
  <xs:element name="fruit">
    <xs:complexType>
      <xs:sequence>
        ...                        <!-- assume that no use of prefix "animal" is made here -->
      </xs:sequence>
    </xs:complexType>
  </xs:element>
</xs:schema>
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Remove this prefix; it isn't used in the XSD.


=== Highlighting

``++xs:some-prefix="some-namespace"++``


endif::env-github,rspecator-view[]
