Namespace prefixes that are declared but not used simply clutter the document and confuse maintainers. Such prefixes should be removed.


== Noncompliant Code Example

----
<xs:schema targetNamespace="http://www.codeSamples.com/fruit"
           xmlns:xs="http://www.w3.org/2001/XMLSchema"
           xmlns:animal="http://www.codeSamples.com/animal"
           elementFormDefault="qualified" version="1">  <!-- Noncompliant -->
  <xs:element name="fruit">
    <xs:complexType>
      <xs:sequence>
        ...                        <!-- assume that no use of prefix "animal" is made here -->
      </xs:sequence>
    </xs:complexType>
  </xs:element>
</xs:schema>
----


== Compliant Solution

----
<xs:schema targetNamespace="http://www.codeSamples.com/fruit"
           xmlns:xs="http://www.w3.org/2001/XMLSchema"
           elementFormDefault="qualified" version="1">
  <xs:element name="fruit">
    <xs:complexType>
      <xs:sequence>
        ...                        <!-- assume that no use of prefix "animal" is made here -->
      </xs:sequence>
    </xs:complexType>
  </xs:element>
</xs:schema>
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

include::highlighting.adoc[]

endif::env-github,rspecator-view[]
