== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,html]
----
<html>                                    <!-- Noncompliant -->
...
</html>
----

=== Compliant solution

[source,html]
----
<!-- Copyright 2013 SonarSource SA -->
<html>
...
</html>
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
