<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p><code>SQL COUNT(..), MIN(..), MAX(..), SUM(..), AVG(..)</code> aggregate functions cause the SAP table buffer to be bypassed, so the use of these functions can lead to performance issues.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-abap" data-lang="abap">SELECT COUNT(*)
       FROM persons
       INTO count
       WHERE city = 'NEW YORK'.</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Double check that use of "XXX(&#8230;&#8203;)" SQL function is strongly indicated here.</p>
</div>
</div>
</div>
</div>