<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>It can be tempting to set <code>elementFormDefault</code> to <code>unqualified</code>, because it allows XML document authors to remove a few prefixes in their files. However this is not a good practice for several reasons:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>The <code>unqualified</code> form doesn&#8217;t convey any functionality.</p>
</li>
<li>
<p>Most XML schemas are <code>qualified</code>, so defining a few <code>unqualified</code> schemas requires the author of an XML document to fully master both cases with no added value.</p>
</li>
<li>
<p>The <code>unqualified</code> form implies that in the XML document, all global elements from the schema must be qualified, and all local elements must be unqualified. So the author must be very cautious regarding the local vs. global nature of each element.</p>
</li>
<li>
<p>The <code>unqualified</code> form implies that in an XML document, the default namespace cannot be used. This stems from the fact that a document&#8217;s default namespace qualifies all prefix-less elements, and an element cannot be both qualified and unqualified. For the XML document author, being prevented from using the default namespace is a real nuisance.</p>
</li>
<li>
<p>In some cases, the <code>unqualified</code> form degrades performance, by requiring the processing tool to look up the <code>unqualified</code> elements in the schema file.</p>
</li>
</ul>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-xml" data-lang="xml">&lt;xs:schema targetNamespace="http://www.codeSamples.com/fruit"
           xmlns:xs="http://www.w3.org/2001/XMLSchema"
           version="1"&gt;                                 &lt;!-- Noncompliant --&gt;
...</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-xml" data-lang="xml">&lt;xs:schema targetNamespace="http://www.codeSamples.com/fruit"
           xmlns:xs="http://www.w3.org/2001/XMLSchema"
           elementFormDefault="qualified" version="1"&gt;
...</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="ulist">
<ul>
<li>
<p>Add an 'elementFormDefault="qualified"' attribute to this schema.</p>
</li>
<li>
<p>Set "elementFormDefault" to "qualified".</p>
</li>
</ul>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="ulist">
<ul>
<li>
<p><code>&lt;xs:schema</code> if no <code>elementFormDefault</code> attribute</p>
</li>
<li>
<p><code>elementFormDefault="unqualified"</code> if attribute present</p>
</li>
</ul>
</div>
</div>
</div>
</div>