<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Using redundant bit masks in comparisons can be misleading because the bit mask does not alter the result of the comparison. This makes the code harder to read and understand. It may also suggest incorrect logic or purpose, increasing the risk of errors.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-rust" data-lang="rust">let x = 1;
if (x | 2 &gt; 3) { // Noncompliant: Bitwise OR with 1 is redundant in this context
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-rust" data-lang="rust">let x = 1;
if (x &gt; 3) { // Compliant: Simplifies the comparison by removing redundant bit operation
}</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Clippy Lints - <a href="https://rust-lang.github.io/rust-clippy/master/index.html#ineffective_bit_mask" class="bare">https://rust-lang.github.io/rust-clippy/master/index.html#ineffective_bit_mask</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>