== How to fix it in Android

=== Code examples

include::../../common/fix/code-rationale.adoc[]

==== Noncompliant code example

[source,kotlin,diff-id=1,diff-type=noncompliant]
----
import android.webkit.WebView

WebView.setWebContentsDebuggingEnabled(true) // Noncompliant
----

==== Compliant solution

[source,kotlin,diff-id=1,diff-type=compliant]
----
import android.webkit.WebView

WebView.setWebContentsDebuggingEnabled(false)
----

=== How does this work?

include::../../common/fix/how.adoc[]
