<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This is an issue when datetime attributes from ActiveRecord models are accessed directly for display purposes without calling <code>.in_time_zone</code> to convert them to the application&#8217;s timezone.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>ActiveRecord stores datetime values in UTC in the database for consistency. However, when these values are displayed to users without timezone conversion, they appear as UTC times instead of the application&#8217;s configured timezone or the user&#8217;s local timezone.</p>
</div>
<div class="paragraph">
<p>This creates several problems:</p>
</div>
<div class="ulist">
<ul>
<li>
<p><strong>User confusion</strong>: Users see times that don&#8217;t match their expectations or local timezone</p>
</li>
<li>
<p><strong>Incorrect time representation</strong>: Business logic that depends on displayed times may be affected</p>
</li>
<li>
<p><strong>Poor user experience</strong>: Applications appear to have incorrect or inconsistent time handling</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>Rails provides the <code>.in_time_zone</code> method specifically to handle this conversion. When called on a datetime attribute, it converts the UTC time to the application&#8217;s configured timezone (set via <code>config.time_zone</code>) or a specified timezone.</p>
</div>
<div class="paragraph">
<p>The issue is particularly common in controllers where datetime attributes are formatted for display, and in view templates where datetime values are directly interpolated.</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="paragraph">
<p>Users will see incorrect times that don&#8217;t match their timezone expectations, leading to confusion and potential business logic errors. This can affect scheduling, reporting, and any time-sensitive operations in the application.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it_in_rails">How to fix it in Rails</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Use <code>.in_time_zone</code> to convert datetime attributes to the application&#8217;s timezone before display or formatting.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-ruby" data-lang="ruby">class OrdersController &lt; ApplicationController
  def show
    @order = Order.find(params[:id])
    # This displays UTC time instead of local timezone
    @formatted_time = @order.placed_at.strftime("%B %d, %Y at %I:%M %p") # Noncompliant
  end
end</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-ruby" data-lang="ruby">class OrdersController &lt; ApplicationController
  def show
    @order = Order.find(params[:id])
    # Convert to application timezone before display
    @formatted_time = @order.placed_at.in_time_zone.strftime("%B %d, %Y at %I:%M %p")
  end
end</code></pre>
</div>
</div>
<div class="paragraph">
<p>In view templates, use <code>.in_time_zone</code> when displaying datetime attributes.</p>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_code_examples_2">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example_2">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-erb" data-lang="erb">&lt;!-- Displays UTC time --&gt;
&lt;%= @order.placed_at %&gt; &lt;!-- Noncompliant --&gt;</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution_2">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-erb" data-lang="erb">&lt;!-- Displays time in application timezone --&gt;
&lt;%= @order.placed_at.in_time_zone %&gt;</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Rails Time Zone Handling - <a href="https://guides.rubyonrails.org/configuring.html#configuring-time-zones">Official Rails guide on configuring and handling time zones</a></p>
</li>
<li>
<p>ActiveSupport TimeZone - <a href="https://api.rubyonrails.org/classes/ActiveSupport/TimeZone.html">API documentation for Rails TimeZone class and methods</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>