<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>According to SQL-92:</p>
</div>
<div class="quoteblock">
<blockquote>
<div class="paragraph">
<p>"X BETWEEN Y AND Z" is equivalent to "X &gt;= Y AND X &lt;= Z"</p>
</div>
</blockquote>
</div>
<div class="paragraph">
<p>Even if the <code>BETWEEN</code> predicate is simply syntactic sugar, using it can improve the readability of a SQL WHERE clause, and is therefore preferred.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-text" data-lang="text">SELECT * FROM PERSONS
WHERE AGE &gt;= 18 and AGE &lt;= 60</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-text" data-lang="text">SELECT * FROM PERSONS
WHERE AGE BETWEEN 18 and 60</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Replace "X&gt;=Y and X+&#8656;+Z" with "X BETWEEN Y AND Z"</p>
</div>
</div>
</div>
</div>