Jump statements, such as ``++return++``, ``++break++`` and ``++continue++`` let you change the default flow of program execution, but jump statements that direct the control flow to the original direction are just a waste of keystrokes.

== Noncompliant Code Example

----
def redundant_jump(x):
    if x == 1:
        print(True)
        return  # NonCompliant
----

== Compliant Solution

----
def redundant_jump(x):
    if x == 1:
        print(True)
----

== Exceptions

* No issue is raised if the jump statement is the only statement of a statement suite:

----
def my_function(x):
    if x > 5:
        do_something()
    elif x == 0:
        return # ok even it could be changed to "pass"
    else:
        do_something_else()
----

* No issue is raised for ``++return None++`` because this was certainly done on purpose to be explicit that a function is really returning ``++None++``.

ifdef::rspecator-view[]
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::rspecator-view[]
