Empty statements, i.e. ``++:++``, are usually introduced by mistake, for example because:

* It was meant to be replaced by an actual statement, but this was forgotten.
* There was a typo which resulted in the colon being doubled, i.e. ``++::++``.

== Noncompliant Code Example

----
Dim i, j As Integer
i = 1 :: j=2   ' Noncompliant - double :
----

== Compliant Solution

----
Dim i, j As Integer
i = 1 : j=2
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
