<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>The classes in the <code>sun.*</code> packages are not part of the official Java API and are not intended for public use. They are internal
implementation details specific to the Oracle JDK (Java Development Kit). Therefore, their availability, behavior, or compatibility is not
guaranteed across different Java implementations or versions.</p>
</div>
<div class="paragraph">
<p>Since these classes are not part of the official Java API, they usually lack proper documentation and support. Finding comprehensive and
up-to-date information about their usage, functionality, and potential limitations can be challenging. This lack of documentation can make
it difficult to understand how to use these classes correctly.</p>
</div>
<div class="paragraph">
<p>Classes in the <code>sun.*</code> packages are often platform-dependent and can vary between different operating systems or Java Virtual Machine (JVM)
implementations. Relying on these classes may lead to code that works on one platform but fails on others, limiting your code&#8217;s portability
and cross-platform compatibility.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">import sun.misc.BASE64Encoder; // Noncompliant</code></pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p><a href="https://www.oracle.com/java/technologies/faq-sun-packages.html">Sun Packages</a></p>
</li>
</ul>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Use classes from the Java API instead of Sun classes.</p>
</div>
</div>
<div class="sect2">
<h3 id="_parameters">Parameters</h3>
<div class="sidebarblock">
<div class="content">
<div class="title">Exclude</div>
<div class="paragraph">
<p><em>String</em></p>
</div>
<div class="paragraph">
<p>Comma separated list of Sun packages to be ignored by this rule. Example: sun.misc</p>
</div>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="ulist">
<ul>
<li>
<p>primary: first <code>sun</code> import</p>
</li>
<li>
<p>secondary: all subsequent <code>sun</code> imports</p>
<div class="ulist">
<ul>
<li>
<p>message: 'Replace also this "Sun" reference.'</p>
</li>
</ul>
</div>
</li>
</ul>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_8_aug_2013_164608_freddy_mallet_wrote">on 8 Aug 2013, 16:46:08 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>Is implemented by http://jira.codehaus.org/browse/SONARJAVA-281</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_2_mar_2017_092505_michael_gumowski_wrote">on 2 Mar 2017, 09:25:05 Michael Gumowski wrote:</h3>
<div class="paragraph">
<p>Move constant cost to linear factor.</p>
</div>
</div>
</div>
</div>