"After" and "Before" prefixes or suffixes should not be used to indicate pre and post events. The concepts of before and after should be given to events using the present and past tense.


== Noncompliant Code Example

----
Class Foo
    Event BeforeClose() ' Noncompliant
    Event AfterClose()  ' Noncompliant
End Class
----


== Compliant Solution

----
Class Foo
    Event Closing()     ' Compliant
    Event Closed()      ' Compliant
End Class
----

