include::../description.adoc[]

== Noncompliant Code Example

With the default threshold of 3:

----
function run() {
  prepare('this is a duplicate'); // Non-Compliant - 'this is a duplicate' is duplicated 3 times
  execute('this is a duplicate');
  release('this is a duplicate');
}
----

== Compliant Solution

----
MESSAGE = 'this is a duplicate';

function run() {
  prepare(MESSAGE);                             
  execute(MESSAGE);
  release(MESSAGE);
}
----

== Exceptions

No issue will be raised on:

* strings with less than 5 characters
* strings with only letters, numbers, underscores and hyphens

----
$severity = $request->getParam('severity-score');{code}
----

ifdef::env-github,rspecator-view[]
'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
