<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when code uses the <code>Duration.withNanos()</code> or <code>Duration.withSeconds()</code> methods.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>The <code>Duration.withNanos()</code> and <code>Duration.withSeconds()</code> methods have confusing behavior that can lead to bugs. These methods only modify one field while keeping the other field unchanged.</p>
</div>
<div class="paragraph">
<p><code>duration.withNanos(newNanos)</code> sets the nanoseconds but keeps the original seconds. <code>duration.withSeconds(newSeconds)</code> sets the seconds but keeps the original nanoseconds.</p>
</div>
<div class="paragraph">
<p>This partial mutation is unexpected and can cause timing bugs when developers assume both fields are set explicitly.</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="paragraph">
<p>Using these methods can introduce timing bugs where durations have unexpected nanosecond or second components, leading to incorrect calculations and comparisons in time-sensitive applications.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Use <code>Duration.ofSeconds()</code> to explicitly set both seconds and nanoseconds.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">Duration duration = Duration.ofSeconds(5, 123456789);
Duration modified = duration.withNanos(987654321); // Noncompliant</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">Duration duration = Duration.ofSeconds(5, 123456789);
Duration modified = Duration.ofSeconds(duration.getSeconds(), 987654321);</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Duration (Java SE 17 &amp; JDK 17) - <a href="https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/time/Duration.html">Official Java documentation for the Duration class</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>