include::../description.adoc[]

== Noncompliant Code Example

----
public String toString () {
  if (this.collection.isEmpty()) {
    return null; // Noncompliant
  } else {
    // ...
----

== Compliant Solution

----
public String toString () {
  if (this.collection.isEmpty()) {
    return "";
  } else {
    // ...
----

include::../see.adoc[]

ifdef::env-github,rspecator-view[]
'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
