<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it_in_asp_net">How to fix it in ASP.NET</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>

</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>

</div>
<div class="sect3">
<h4 id="_noncompliant_code_example_2">Noncompliant code example</h4>

</div>
<div class="sect3">
<h4 id="_compliant_solution_2">Compliant solution</h4>

</div>
</div>
<div class="sect2">
<h3 id="_how_does_this_work">How does this work?</h3>

</div>
<div class="sect2">
<h3 id="_pitfalls">Pitfalls</h3>

</div>
<div class="sect2">
<h3 id="_going_the_extra_mile">Going the extra mile</h3>

</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it_in_razor">How to fix it in Razor</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_pitfalls_2">Pitfalls</h3>

</div>
<div class="sect2">
<h3 id="_going_the_extra_mile_2">Going the extra mile</h3>

</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>

</div>
<div class="sect2">
<h3 id="_articles_blog_posts">Articles &amp; blog posts</h3>

</div>
<div class="sect2">
<h3 id="_conference_presentations">Conference presentations</h3>

</div>
<div class="sect2">
<h3 id="_standards">Standards</h3>

</div>
</div>
</div>