<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Re-assigning a variable to itself is a defect as it has no actual effect and indicates meaning to do something else.
It usually means that:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>The statement is redundant and should be removed</p>
</li>
<li>
<p>The re-assignment is a mistake, and another value or variable was intended for the assignment instead</p>
</li>
</ul>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-csharp" data-lang="csharp">public class Choice {
    private bool selected;

    public void MakeChoice(bool selected)
    {
        selected = selected; // Noncompliant
    }
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-csharp" data-lang="csharp">public class Choice {
    private bool selected;

    public void MakeChoice(bool selected)
    {
        this.selected = selected; // Compliant
    }
}</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Microsoft Learn - <a href="https://learn.microsoft.com/en-us/dotnet/csharp/misc/cs1717">Compiler Warning (level 3) CS1717</a></p>
</li>
</ul>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove or correct this useless self-assignment.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_14_feb_2018_160342_alban_auzeill_wrote">on 14 Feb 2018, 16:03:42 Alban Auzeill wrote:</h3>
<div class="paragraph">
<p>Swift produce a compile error: assigning a variable to itself</p>
</div>
</div>
</div>
</div>