== How to fix it in Micronaut

=== Code examples

include::../../common/fix/code-rationale.adoc[]

==== Noncompliant code example

[source,java,diff-id=10,diff-type=noncompliant]
----
@Get("/Noncompliant/httpresponse")
public HttpResponse<String> noncompliant(@QueryValue("q") String location) throws URISyntaxException {
    URI url = new URI(location);
    return HttpResponse.redirect(url); // Noncompliant  
}
----

==== Compliant solution

[source,java,diff-id=10,diff-type=compliant]
----
@Get("/Compliant/httpresponse")
public HttpResponse<String> compliant(@QueryValue("q") String location) throws URISyntaxException {
    URI url = new URI(location);
    String authority = url.getAuthority();
    if (authority != null && authority.endsWith(".example.com")) {
        return HttpResponse.redirect(url);
    } else {
        return HttpResponse.status(HttpStatus.FORBIDDEN);
    }
}
----

include::../../common/fix/how-does-this-work.adoc[]

=== Pitfalls

include::../../common/pitfalls/starts-with.adoc[]

