== Why is this an issue?

It is considered best-practice to use relative URLs in web pages to prevent having to update the addresses if the web address in use changes. Moreover, if some absolute URLs are missed in such a process, it will obviously impact the user experience. 


=== Noncompliant code example

[source,html]
----
<img src="http://www.myserver.com/smiley.gif" alt="Smiley face" height="42" width="42" />
----


=== Compliant solution

[source,html]
----
<img src="smiley.gif" alt="Smiley face" height="42" width="42" />
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Replace this absolute URI "XXXX" with a relative one, or move this absolute URI to a configuration file.


=== Parameters

.attributes
****

----
a.href,applet.codebase,area.href,base.href,blockquote.cite,body.background,del.cite,form.action,frame.longdesc,frame.src,head.profile,iframe.longdesc,iframe.src,img.longdesc,img.src,img.usemap,input.src,input.usemap,ins.cite,link.href,object.classid,object.codebase,object.data,object.usemap,q.cite,script.src,audio.src,button.formaction,command.icon,embed.src,html.manifest,input.formaction,source.src,video.poster,video.src
----

Comma-separated list of tag.attributes to be checked for absolute URI.
****


'''
== Comments And Links
(visible only on this page)

=== is related to: S1075

endif::env-github,rspecator-view[]
