== How to fix it in Java Net Package

=== Code examples

include::../../common/fix/code-rationale.adoc[]

==== Noncompliant code example

[source,kotlin,diff-id=1,diff-type=noncompliant]
----
import java.net.URL

class RequestHandler {
    fun doGet(req: HttpServletRequest, resp: HttpServletResponse) {
        val location = req.getParameter("url")
        val url = URL(location)
        val connection = url.openConnection()
    }
}
----

==== Compliant solution

[source,kotlin,diff-id=1,diff-type=compliant]
----
class RequestHandler {
    companion object {
        val ALLOWED_HOST = arrayOf("https://trusted1.example.com/", "https://trusted2.example.com/")
    }
    fun doGet(req: HttpServletRequest, resp: HttpServletResponse) {
        val location = req.getParameter("url")
        val url = URL(location)
        if (location in ALLOWED_HOST) {
            var connection = url.openConnection()
        }
    }
}
----

=== How does this work?

include::../../common/fix/pre-approved-list.adoc[]

include::../../common/fix/blacklist.adoc[]

=== Pitfalls

include::../../common/pitfalls/starts-with.adoc[]

include::../../common/pitfalls/blacklist-toctou.adoc[]

