When verifying that code raises an exception, a good practice is to avoid having multiple method calls inside the tested code, to be explicit about what is exactly tested.

When two of the methods can raise the same exception, not respecting this good practice is a bug, since it is not possible to know what is really tested.

== Noncompliant Code Example

----
public function testSomething() 
{ 
  try {
    g(y(1)); // Noncompliant
    $this->fail('RuntimeException is not thrown');
  } catch (RuntimeException $e) {}
}
----

== Compliant Solution

----
public function testSomething() 
{ 
  $y = y(1);
  try {
    g($y);
    $this->fail('RuntimeException is not thrown by g()');
  } catch (RuntimeException $e) {}
}
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../highlighting.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
