include::../description.adoc[]

include::../why.adoc[]

== How to fix it

=== Code examples

==== Noncompliant code example

[source,yaml,diff-id=1,diff-type=noncompliant]
----
name: Setup

on: push

jobs:
  setup:
    runs-on: ubuntu-latest
    steps:
      - name: Install packages
        run: |
          sudo apt-get install ca-certificates
          sudo aptitude install ca-certificates
          sudo apt install ca-certificates
          sudo gdebi /tmp/package.deb
----

include::../non-compliant-desc.adoc[]

==== Compliant solution

[source,yaml,diff-id=1,diff-type=compliant]
----
name: Setup

on: push

jobs:
  setup:
    runs-on: ubuntu-latest
    steps:
      - name: Install packages
        run: |
          sudo apt-get -y install ca-certificates
          sudo aptitude -y install ca-certificates
          sudo apt -y install ca-certificates
          sudo gdebi --n /tmp/package.deb
----

include::../compliant-desc.adoc[]

include::../how.adoc[]

== Resources
=== Documentation

include::../general-links.adoc[]

ifdef::env-github,rspecator-view[]
'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../highlighting.adoc[]

'''
endif::env-github,rspecator-view[]
