<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Fold expressions, introduced in C&#43;&#43;17, are a way to expand a variadic template parameter pack with operators between each pack element. Due to the high flexibility of this construct, many variadic templates that used to be written by a recursive call can now be written in a more direct way.</p>
</div>
<div class="paragraph">
<p>In addition to a usually simpler code, fold expressions results in far less functions instantiated, which can improve compilation time.</p>
</div>
<div class="paragraph">
<p>This rule raises an issue when a recursive template instantiation that could be easily be replaced by a fold expression is detected</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">template&lt;class Cont&gt;
void addElementsToContainer(Cont &amp;C) {
}

template&lt;class Cont, class T, class ...U&gt;
void addElementsToContainer(Cont &amp;C, T &amp;&amp;t, U &amp;&amp;...us) {
    C.push_back(forward&lt;T&gt;(t));
    addElementsToContainer(C, forward&lt;U&gt;(us)...); // Noncompliant recursive call
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">template&lt;class Cont, class ...T&gt;
void addElementsToContainer(Cont &amp;C, T &amp;&amp;...ts) {
    (C.push_back(std::forward&lt;T&gt;(ts)),...); // Compliant fold expression over the operator ','
}</code></pre>
</div>
</div>
</div>
</div>
</div>