<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Shared coding conventions allow teams to collaborate efficiently. This rule checks that data item levels are incremented according to the configured values.</p>
</div>
<div class="paragraph">
<p>Using the default configuration:</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cobol" data-lang="cobol">01 StudentDetails.
   05 StudentName.
      12 FirstName     Pic x(10).   *&gt; Noncompliant, level 10 is expected
      10 MiddleInitial Pic x.
      10 Surname       Pic x(15).
   03 StudentId        Pic 9(7).      *&gt; Noncompliant, only 05 level is expected
                 *&gt; We should not mix two types of incrementation in the same data structure
...
01  Customer-Record.
    03  Customer-Name.
        10 Last-Name         Pic x(17).   *&gt; Noncompliant, level 05 is expected
        05 Filler            Pic x.
        05 Initials          Pic xx.
    03  Part-Order.
        05 Part-Name         Pic x(15).
        05 Part-Color        Pic x(10).
            10 Part-Color-Code        Pic x.   *&gt; Noncompliant, level 07 is expected</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cobol" data-lang="cobol">01 StudentDetails.
   05 StudentName.
      10 FirstName     PIC X(10).
      10 MiddleInitial PIC X.
      10 Surname       PIC X(15).
   05 StudentId        PIC 9(7).
...
01  Customer-Record.
    03  Customer-Name.
        05 Last-Name         Pic x(17).
        05 Filler            Pic x.
        05 Initials          Pic xx.
    03  Part-Order.
        05 Part-Name         Pic x(15).
        05 Part-Color        Pic x(10).
            07 Part-Color-Code        Pic x.</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_16_jun_2014_133706_freddy_mallet_wrote">on 16 Jun 2014, 13:37:06 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>There are too many open questions on this specification to be able to provide a first implementation</p>
</div>
</div>
</div>
</div>