<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when trying to perform comparison or arithmetic operations between <code>offset-naive</code> and <code>offset-aware</code> Datetimes.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p><code>datetime.datetime</code> and <code>datetime.time</code> objects may be categorized as "aware" or "naive" depending on whether or not they include timezone information.</p>
</div>
<div class="paragraph">
<p>Comparison or arithmetic operations between <code>offset-naive</code> and <code>offset-aware</code> datetimes raise a <code>TypeError</code>.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Perform comparison or arithmetic operations only between offset-naive or only between offset-aware Datetimes.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">from datetime import datetime
from pytz import timezone


def compare_tz():
    dt1 = datetime.now(tz=timezone('America/New_York'))
    dt2 = datetime.now()

    if dt1 &lt; dt2: # Noncompliant: TypeError: can't compare offset-naive and offset-aware datetimes
        ...</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">from datetime import datetime
from pytz import timezone


def compare_tz():
    dt1 = datetime.now(tz=timezone('America/New_York'))
    dt2 = datetime.now(timezone.utc)

    if dt1 &lt; dt2:
        ...</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Python documentation - <a href="https://docs.python.org/3/library/datetime.html#aware-and-naive-objects">Aware and Naive Objects</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>