== How to fix it in Java Cryptographic Extension

=== Code examples

include::../../common/fix/code-rationale.adoc[]

==== Noncompliant code example

[source,php,diff-id=1,diff-type=noncompliant]
----
curl_setopt($curl, CURLOPT_SSL_VERIFYPEER, false); // Noncompliant
----

==== Compliant solution

[source,php,diff-id=1,diff-type=noncompliant]
----
// Default value since cURL 7.10
curl_setopt($curl, CURLOPT_SSL_VERIFYPEER, TRUE);
----

=== How does this work?

include::../../common/fix/validation.adoc[]
