== How to fix it in cURL

=== Code examples

include::../../common/fix/code-rationale.adoc[]

==== Noncompliant code example

[source,php,diff-id=1,diff-type=noncompliant]
----
$curl = curl_init();
curl_setopt($ch, CURLOPT_URL, 'https://example.com');
curl_setopt($curl, CURLOPT_SSL_VERIFYPEER, false); // Noncompliant
curl_exec($curl);
----

==== Compliant solution

[source,php,diff-id=1,diff-type=noncompliant]
----
$curl = curl_init();
curl_setopt($ch, CURLOPT_URL, 'https://example.com');
curl_exec($curl);
----

=== How does this work?

include::../../common/fix/validation.adoc[]
