<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Early classes of the Java API, such as <code>Vector</code>, <code>Hashtable</code> and <code>StringBuffer</code>, were synchronized to make them thread-safe.
However, synchronization has a significant negative impact on performance, even when using these collections from a single thread.</p>
</div>
<div class="paragraph">
<p>It is often best to use their non-synchronized counterparts:</p>
</div>
<div class="ulist">
<ul>
<li>
<p><code>ArrayList</code> or <code>LinkedList</code> instead of <code>Vector</code></p>
</li>
<li>
<p><code>Deque</code> instead of <code>Stack</code></p>
</li>
<li>
<p><code>HashMap</code> instead of <code>Hashtable</code></p>
</li>
<li>
<p><code>StringBuilder</code> instead of <code>StringBuffer</code></p>
</li>
</ul>
</div>
<div class="paragraph">
<p>Even when used in synchronized contexts, you should think twice before using their synchronized counterparts, since their usage can be costly.
If you are confident the usage is legitimate, you can safely ignore this warning.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">Vector&lt;Cat&gt; cats = new Vector&lt;&gt;();</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">ArrayList&lt;Cat&gt; cats = new ArrayList&lt;&gt;();</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_exceptions">Exceptions</h3>
<div class="paragraph">
<p>Usage of these synchronized classes is ignored in the signatures of overriding methods.</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">@Override
public Vector getCats() {...} // Compliant</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Replace the synchronized class "{}" by an unsynchronized one such as "{}".</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_relates_to_s1076">relates to: <a data-rspec-id="S1076" class="rspec-auto-link">S1076</a></h3>

</div>
<div class="sect2">
<h3 id="_on_31_jul_2013_151351_dinesh_bolkensteyn_wrote">on 31 Jul 2013, 15:13:51 Dinesh Bolkensteyn wrote:</h3>
<div class="paragraph">
<p>Can also replace UseArrayListInsteadOfVector</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_31_jul_2013_151808_dinesh_bolkensteyn_wrote">on 31 Jul 2013, 15:18:08 Dinesh Bolkensteyn wrote:</h3>
<div class="paragraph">
<p>Implemented by http://jira.codehaus.org/browse/SONARJAVA-236</p>
</div>
</div>
</div>
</div>