Sharing some naming conventions enables teams to collaborate more efficiently. This rule checks that all ``++union++`` names match a provided regular expression.


== Noncompliant Code Example

Using the default regular expression ``++^[A-Z][a-zA-Z0-9]*$++``:

----
union my_union {
    int one;
    int two;
};
----


== Compliant Solution

----
union MyUnion {
    int one;
    int two;
};
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

include::parameters.adoc[]

endif::env-github,rspecator-view[]
