== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

With the default threshold value of 3:

[source,kotlin]
----
if (((condition1 && condition2) || (condition3 && condition4)) && condition5) { ... } 
----

=== Compliant solution

[source,kotlin]
----
if ((myFirstCondition() || mySecondCondition()) && myLastCondition()) { ... }     
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../parameters.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
