<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule allows you to track the usage of the <code>SuppressMessage</code> attributes and <code>#pragma warning disable</code> mechanism.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-csharp" data-lang="csharp">[SuppressMessage("", "S100")]
...

#pragma warning disable S100
...
#pragma warning restore S100</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Do not suppress issues.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_9_oct_2013_130626_freddy_mallet_wrote">on 9 Oct 2013, 13:06:26 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>Is implemented by http://jira.codehaus.org/browse/SONARJAVA-339</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_25_oct_2016_163016_tamas_vajk_wrote">on 25 Oct 2016, 16:30:16 Tamas Vajk wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] I created a dedicated subtask for Java, which only holds the parameter. In C# this rule should work without parameters. Let me know if this change doesn&#8217;t work for the Rule API, or for any other reasons.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_26_oct_2016_170750_ann_campbell_wrote">on 26 Oct 2016, 17:07:50 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>What you&#8217;ve done is just right [~tamas.vajk]</p>
</div>
</div>
</div>
</div>