<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule allows you to track the usage of the <code>@SuppressWarnings</code> mechanism.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="paragraph">
<p>With a parameter value of "unused" :</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">@SuppressWarnings("unused")
@SuppressWarnings("unchecked")  // Noncompliant</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Suppressing the 'xxxxxx' warning is not allowed.</p>
</div>
</div>
<div class="sect2">
<h3 id="_parameters">Parameters</h3>
<div class="sidebarblock">
<div class="content">
<div class="title">listOfWarnings</div>
<div class="paragraph">
<p>Comma separated list of warnings that can be suppressed. Example: 'unchecked, cast, all, boxing'. An empty list means that no warning can be suppressed.</p>
</div>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_9_oct_2013_130626_freddy_mallet_wrote">on 9 Oct 2013, 13:06:26 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>Is implemented by http://jira.codehaus.org/browse/SONARJAVA-339</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_25_oct_2016_163016_tamas_vajk_wrote">on 25 Oct 2016, 16:30:16 Tamas Vajk wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] I created a dedicated subtask for Java, which only holds the parameter. In C# this rule should work without parameters. Let me know if this change doesn&#8217;t work for the Rule API, or for any other reasons.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_26_oct_2016_170750_ann_campbell_wrote">on 26 Oct 2016, 17:07:50 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>What you&#8217;ve done is just right [~tamas.vajk]</p>
</div>
</div>
</div>
</div>