== Why is this an issue?

include::../description.adoc[]


=== Noncompliant code example

[source,python]
----
r"[ab]|a"   # The "|a" is redundant because "[ab]" already matches "a"
r".*|a"     # .* matches everything, so any other alternative is redundant
----


=== Compliant solution

[source,python]
----
r"[ab]"
r".*"
----
