This rule raises an issue when multiple branches of a regex alternative match the same input.

== Why is this an issue?

include::../description.adoc[]


=== Noncompliant code example

[source,python]
----
r"[ab]|a"   # Noncompliant: the "|a" is redundant because "[ab]" already matches "a"
r".*|a"     # Noncompliant: .* matches everything, so any other alternative is redundant
----


=== Compliant solution

[source,python]
----
r"[ab]"
r".*"
----
