include::../description.adoc[]


== Noncompliant Code Example

[source,python]
----
r"[ab]|a"   # The "|a" is redundant because "[ab]" already matches "a"
r".*|a"     # .* matches everything, so any other alternative is redundant
----


== Compliant Solution

[source,python]
----
r"[ab]"
r".*"
----
