<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p><code>throw(NULL)</code> is equivalent to <code>throw(0)</code>, and is therefore caught by an integer handler. However, since <code>NULL</code> is typically used in the context of pointers, developers may expect it to be caught by a pointer-to-type handler. Thus to avoid confusion, zero should be thrown instead of <code>NULL</code>.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">try
{
  throw ( NULL ); // Noncompliant
}
catch ( int32_t i ) // NULL exception handled here
{
  // ...
}
catch ( const char_t * ) // Developer may expect it to be caught here
{
  // ...
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">try
{
  throw ( 0 );
}
catch ( int32_t i )
{
  // ...
}
catch ( const char_t * )
{
  // ...
}</code></pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p>MISRA C&#43;&#43;:2008, 15-1-2</p>
</li>
</ul>
</div>
<hr>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_16_aug_2016_180627_alban_auzeill_wrote">on 16 Aug 2016, 18:06:27 Alban Auzeill wrote:</h3>
<div class="paragraph">
<p>I close this rule in favor of [<a data-rspec-id="S3696" class="rspec-auto-link">RSPEC-3696</a>] that mark as non-compliant NULL and 0 (zero).</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_16_aug_2016_192850_ann_campbell_wrote">on 16 Aug 2016, 19:28:50 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>As an addendum to the previous comment: <code>NULL</code> evaluates to 0, so we can&#8217;t distinguish between <code>throw NULL</code> and <code>throw 0</code>.</p>
</div>
</div>
</div>
</div>