== Why is this an issue?

While not technically incorrect, the omission of ``++BEGIN...END++`` can be misleading, and may lead to the introduction of errors during maintenance.

=== Noncompliant code example

[source,sql]
----
IF @flag = 1 -- Noncompliant
  EXEC something;
----

=== Compliant solution

[source,sql]
----
IF @flag = 1
    BEGIN
        EXEC something;
    END;
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
