include::../description.adoc[]


== Noncompliant Code Example

[source,python]
----
# This can never match because $ and ^ have been switched around
r"$[a-z]+^" # Noncompliant
----


== Compliant Solution

[source,python]
----
r"^[a-z]+$"
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../highlighting.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
