include::../multi-lines-description.adoc[]

== Why is this an issue?

=== Noncompliant code example

[source,ruby]
----
=begin
=end

=begin


=end
----

=== Compliant solution

[source,ruby]
----
=begin
Proper comment or no comment at all
=end
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

endif::env-github,rspecator-view[]
