include::../description.adoc[]

== Noncompliant Code Example

[source,ruby]
----
=begin
=end

=begin


=end
----
ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

endif::env-github,rspecator-view[]
