== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,ruby]
----
=begin
=end

=begin


=end
----
ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

endif::env-github,rspecator-view[]
