include::../description.adoc[]

== Noncompliant Code Example

With the default regular expression ``++^[a-z][a-zA-Z0-9]*$++``:

----
class MyClass {
  $my_field;
}
----

== Compliant Solution

----
class MyClass {
  $myField;
}
----

ifdef::env-github,rspecator-view[]
'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
