== Why is this an issue?

include::../description.adoc[]

=== Code examples

The examples below enforce the following regular expression ``++^[a-z][a-zA-Z0-9]*$++``:

==== Noncompliant code example

[source,php,diff-id=1,diff-type=noncompliant]
----
class MyClass {
  $my_field;
}
----

==== Compliant solution

[source,php,diff-id=1,diff-type=compliant]
----
class MyClass {
  $myField;
}
----

== Resources

=== Documentation

* Wikipedia - https://en.wikipedia.org/wiki/Naming_convention_(programming)[Naming Convention (programming)]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../parameters.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
