== Why is this an issue?

An ``++except++`` clause that only rethrows the caught exception has the same effect as omitting the ``++except++`` altogether and letting it bubble up automatically, but with more code and the additional detriment of leaving maintainers scratching their heads. 


Such clauses should either be eliminated or populated with the appropriate logic.

=== Noncompliant code example

[source,python]
----
a = {}
try:
    a[5]
except KeyError:
    raise  # Noncompliant
----

=== Compliant solution

[source,python]
----
a = {}
try:
    a[5]
except KeyError as e:
    logging.exception('error while accessing the dict')
    raise e
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
