<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Code that contains many macros becomes hard to understand. This is even worse when the set of defined macros is not stable, and you have to know at each point what macros are defined. Therefore, <code>#undef</code> can decrease the readability of macros.</p>
</div>
<div class="paragraph">
<p>However, well-disciplined use of <code>#undef</code> can also improve readability, for instance when defining a macro with a limited scope: The macro is <code>#defined</code>, used a couple of times to reduce code duplication, then immediately <code>#undefed</code>.</p>
</div>
<div class="paragraph">
<p>This rule raises an issue when a <code>#undef</code> undefines a macro that was defined in another file. It will also raise an issue for an <code>#undef</code> directive that tries to undefine a non-existing macro.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">#ifndef MY_HDR
#define MY_HDR
#endif
...
#undef MY_HDR    /* Noncompliant */</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">#define LEVEL(i) int const i = #i
LEVEL(Debug);
LEVEL(Warning);
LEVEL(Error);
#undef LEVEL</code></pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p>MISRA C:2004, 19.6 - #undef shall not be used.</p>
</li>
<li>
<p>MISRA C&#43;&#43;:2008, 16-0-3 - #undef shall not be used.</p>
</li>
<li>
<p>MISRA C:2012, 20.5 - #undef should not be used</p>
</li>
</ul>
</div>
<hr>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_31_mar_2015_190231_evgeny_mandrikov_wrote">on 31 Mar 2015, 19:02:31 Evgeny Mandrikov wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] implementation seems more complete (SQALE, description) than this spec.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_13_apr_2015_192758_evgeny_mandrikov_wrote">on 13 Apr 2015, 19:27:58 Evgeny Mandrikov wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] I&#8217;m wondering why blocker, but not active by default? Note that in implementation currently major and active.</p>
</div>
</div>
</div>
</div>