include::../description.adoc[]

== Noncompliant Code Example

[source,csharp]
----
struct MyStruct  // Noncompliant
{
    public int Value { get; set; }
}
----

== Compliant Solution

[source,csharp]
----
struct MyStruct : IEquatable<MyStruct>
{
    public int Value { get; set; }

    public bool Equals(MyStruct other)
    {
        // ...
    }
}
----

include::../rspecator.adoc[]