== Why is this an issue?

Naming conventions are an important tool in efficient team collaboration. This rule checks that all form names match a regular expression naming convention.


=== Noncompliant code example

With the default regular expression:


[source,abap]
----
FORM MyForm. 
...
ENDFORM.
----


=== Compliant solution

[source,abap]
----
FORM MY_FORM. 
...
ENDFORM.
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

include::parameters.adoc[]

endif::env-github,rspecator-view[]
