== Why is this an issue?

Naming conventions are an important tool in efficient team collaboration. This rule checks that all form names match a regular expression naming convention.


=== Noncompliant code example

With the default regular expression:


[source,abap]
----
FORM MyForm. 
...
ENDFORM.
----


=== Compliant solution

[source,abap]
----
FORM MY_FORM. 
...
ENDFORM.
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Rename form "XXXX" to match the regular expression ${format}


=== Parameters

.format
****

----
^([A-Z0-9_]*|[a-z0-9_]*)$
----

Regular expression used to check the form names against.
****


endif::env-github,rspecator-view[]
