include::../summary.adoc[]

== Why is this an issue?

include::../description.adoc[]

== How to fix it

=== Code examples

==== Noncompliant code example

With default provided regular expression ^([A-Z][A-Za-z]*:)*([A-Z][A-Za-z]*)$:

[source,yaml,diff-id=1,diff-type=noncompliant]
----
AWSTemplateFormatVersion: 2010-09-09
Resources:
  S3Bucket:
    Type: 'AWS::S3::Bucket' 
    Properties: 
      BucketName: "mybucketname"
      Tags: 
        - Key: "anycompany:cost-center"
          Value: "Accounting"
        - Key: "anycompany:EnvironmentType"
          Value: "PROD"
----

==== Compliant solution

[source,yaml,diff-id=1,diff-type=compliant]
----
AWSTemplateFormatVersion: 2010-09-09
Resources:
  S3Bucket:
    Type: 'AWS::S3::Bucket'
    Properties: 
      BucketName: "mybucketname"
      Tags: 
        - Key: "Anycompany:CostCenter"
          Value: "Accounting"
        - Key: "Anycompany:EnvironmentType"
          Value: "PROD"
----

include::../see.adoc[]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../parameters.adoc[]

endif::env-github,rspecator-view[]
