include::../description.adoc[]

== Noncompliant Code Example

----
public function numberOfMinutes(hours:int):int
{
  var seconds:int = 0;  // seconds is never used 
  return hours * 60;
}
----

== Compliant Solution

----
public function numberOfMinutes(hours:int):int
{
  return hours * 60;
}
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
