include::../rationale.adoc[]

include::../how-to-fix-it.adoc[]

=== Code examples

==== Noncompliant code example

[source,flex,diff-id=1,diff-type=noncompliant]
----
public function numberOfMinutes(hours:int):int
{
  var seconds:int = 0;  // Noncompliant - seconds is unused 
  return hours * 60;
}
----

==== Compliant solution

[source,flex,diff-id=1,diff-type=compliant]
----
public function numberOfMinutes(hours:int):int
{
  return hours * 60;
}
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
