Every subroutine should be commented to explain its goal and how it works. This non-empty comment must be located before or after the subroutine definition.

== Noncompliant Code Example

----
C     SR012         BEGSR
C                         ...
C                   ENDSR
----

----
 /free
  begsr SR012;
   ...
  endsr;
 /end-free
----

== Compliant Solution

----
C********************************************************************
C* SR012- Here is my comment
C********************************************************************
C     SR012         BEGSR
C                         ...
C                   ENDSR
----

----
 /free
  // SR012- Here is my comment
  begsr SR012;
   ...
  endsr;
 /end-free
----
ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

endif::env-github,rspecator-view[]
