== Why is this an issue?

Every subroutine should be commented to explain its goal and how it works. This non-empty comment must be located before or after the subroutine definition.

=== Noncompliant code example

[source,rpg]
----
C     SR012         BEGSR
C                         ...
C                   ENDSR
----

[source,rpg]
----
 /free
  begsr SR012;
   ...
  endsr;
 /end-free
----

=== Compliant solution

[source,rpg]
----
C********************************************************************
C* SR012- Here is my comment
C********************************************************************
C     SR012         BEGSR
C                         ...
C                   ENDSR
----

[source,rpg]
----
 /free
  // SR012- Here is my comment
  begsr SR012;
   ...
  endsr;
 /end-free
----
ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Document this subroutine.


endif::env-github,rspecator-view[]
