While the ``++OCCURS++`` keyword still works, it is slower than its more modern replacement: ``++DIM++``, which should be used instead.


== Noncompliant Code Example

----
D cussls          ds                  Occurs(200)  
D CustNo                         7p 0                     
D MonthlySls                    11P 2
----


== Compliant Solution

----
D cussls          ds                  dim(200)   
D CustNo                         7p 0                      
D MonthlySls                    11P 2 
----


ifdef::rspecator-view[]
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::rspecator-view[]
