== Why is this an issue?

While the ``++OCCURS++`` keyword still works, it is slower than its more modern replacement: ``++DIM++``, which should be used instead.


=== Noncompliant code example

[source,rpg]
----
D cussls          ds                  Occurs(200)  
D CustNo                         7p 0                     
D MonthlySls                    11P 2
----


=== Compliant solution

[source,rpg]
----
D cussls          ds                  dim(200)   
D CustNo                         7p 0                      
D MonthlySls                    11P 2 
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Use a "DIM" instead.


'''
== Comments And Links
(visible only on this page)

=== on 8 Apr 2015, 11:40:45 Ann Campbell wrote:
http://www.bmeyers.net/faqs/14-tips/32-rpg-iv-style?start=3

=== on 21 Apr 2015, 14:00:47 Pierre-Yves Nicolas wrote:
From what I read, DIM and OCCURS perform similarly. 

http://forums.iprodeveloper.com/forums/aft/45090

I don't think this rule should be both critical and activated by default.

=== on 22 Apr 2015, 11:02:23 Ann Campbell wrote:
tough choice, [~pierre-yves.nicolas]. I picked inactive by default

endif::env-github,rspecator-view[]
