== Why is this an issue?

While the ``++OCCURS++`` keyword still works, it is slower than its more modern replacement: ``++DIM++``, which should be used instead.


=== Noncompliant code example

[source,rpg]
----
D cussls          ds                  Occurs(200)  
D CustNo                         7p 0                     
D MonthlySls                    11P 2
----


=== Compliant solution

[source,rpg]
----
D cussls          ds                  dim(200)   
D CustNo                         7p 0                      
D MonthlySls                    11P 2 
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
