There is no point in creating a branch in the code only to execute an empty statement. Such code can only be due to the presence of an unexpected trailing semi-colon.


== Noncompliant Code Example

----
if(foo);      //Noncompliant, the semi-colon must be removed
  trigger(action1);  // executes unconditionally
else if (bar);    //Noncompliant, the semi-colon must be removed
  trigger(action2);  // executes unconditionally

while (condition);  // Noncompliant
  doTheThing();  // executes once, unconditionally 
----


== See

* https://www.securecoding.cert.org/confluence/x/7gCTAw[CERT, MSC51-J.] - Do not place a semicolon immediately following an if, for, or while condition
* https://www.securecoding.cert.org/confluence/x/i4FtAg[CERT, EXP15-C.] - Do not place a semicolon on the same line as an if, for, or while statement

