<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Afin de ne pas perturber via la programmation, les paramètres positionnés sur les BIND PACKAGE pour des environnements de type 24/24 7/7, il est important de vérifier que les ordres de SELECT simples ou dans les DECLARE CURSOR que les mots clés WITH RR, WITH RS er WITH CS ne soient pas positionnés. Le seul mot clé accepté est WITH UR.</p>
</div>
<div class="paragraph">
<p>S’assurer que sur les ordres de SELECT (simple ou mis dans des DECLARE CURSOR) que les mots clés WITH RR, WITH RS er WITH CS ne soient pas positionnés.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cobol" data-lang="cobol">EXEC SQL DECLARE CLEC_0 CURSOR WITH HOLD
                    	FOR SELECT	C_BQ
                              		,C_CARNT_ORDREB
                              		,N_REF_ORDREB…..
          FROM       S1ORDCOU
          WHERE
( C_BQ               &gt;=  :TORD-C-BQ               ) AND NOT
( C_BQ                =  :TORD-C-BQ                 AND…..
ORDER BY C_BQ,C_CARNT_ORDREB,N_REF_ORDREB
FOR FETCH ONLY
WITH RR
END-EXEC.</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cobol" data-lang="cobol">EXEC SQL DECLARE CLEC_0 CURSOR WITH HOLD
                    	FOR SELECT	C_BQ
                              		,C_CARNT_ORDREB
                              		,N_REF_ORDREB…..
          FROM       S1ORDCOU
          WHERE
( C_BQ               &gt;=  :TORD-C-BQ               ) AND NOT
( C_BQ                =  :TORD-C-BQ                 AND…..
ORDER BY C_BQ,C_CARNT_ORDREB,N_REF_ORDREB
FOR FETCH ONLY
WITH UR ou RIEN
END-EXEC.</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_27_may_2014_091507_freddy_mallet_wrote">on 27 May 2014, 09:15:07 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>This rule is specific to the BNP context, we should provide the XPath rule allowing to get the same result</p>
</div>
</div>
</div>
</div>