== Why is this an issue?

It's slightly more efficient to append single characters to ``++StringBuffer++`` and ``++StringBuilder++`` instances as ``++char++``s, than as ``++String++``s.


=== Noncompliant code example

[source,java]
----
String f(char c) {
  StringBuilder sb = new StringBuilder();
  sb.append(Character.toString(c)); // Noncompliant
  return sb.toString();
}
----


=== Compliant solution

[source,java]
----
String f(char c) {
  StringBuilder sb = new StringBuilder();
  sb.append(c); // Compliant
  return sb.toString();
}
----

=== Exceptions

The rule doesn't apply to constants, because the difference in performance between appending a constant `char` and a constant `String` is very small.

=== Benchmarks

[options="header"]
|===
| Method| Runtime| Average time| Error margin
| appendConstantChar| Temurin 21| 27.05 ns/op| ±13.85 ns/op
| appendConstantString| Temurin 21| 27.03 ns/op| ±6.40 ns/op
| appendVariableChar| Temurin 21| 33.81 ns/op| ±9.64 ns/op
| appendVariableString| Temurin 21| 51.08 ns/op| ±11.40 ns/op
|===

*Benchmarking code*

The results were generated by running the following snippet with https://github.com/openjdk/jmh[JMH].

[source,java]
----
@State(Scope.Thread)
@BenchmarkMode({Mode.AverageTime})
@OutputTimeUnit(TimeUnit.NANOSECONDS)
public class S3009 {
  StringBuilder builder;
  String stringVar;
  char charVar;
  Random random;

  @Setup(Level.Invocation)
  public  void setupInvocation() {
    charVar = (char) random.nextInt(Character.MAX_VALUE);
    stringVar = String.valueOf(charVar);
    builder = new StringBuilder();
  }

  @Setup(Level.Iteration)
  public void setup() {
    random = new Random();
  }

  @Benchmark
  public void appendConstantChar(Blackhole blackhole) {
    builder.append('a');
    builder.append('a');
    builder.append('a');
    builder.append('a');
    builder.append('a');
    blackhole.consume(builder);
  }

  @Benchmark
  public void appendConstantString(Blackhole blackhole) {
    builder.append("a");
    builder.append("a");
    builder.append("a");
    builder.append("a");
    builder.append("a");
    blackhole.consume(builder);
  }

  @Benchmark
  public void appendVariableString(Blackhole blackhole) {
    builder.append(stringVar);
    builder.append(stringVar);
    builder.append(stringVar);
    builder.append(stringVar);
    builder.append(stringVar);
    blackhole.consume(builder);
  }

  @Benchmark
  public void appendVariableChar(Blackhole blackhole) {
    builder.append(charVar);
    builder.append(charVar);
    builder.append(charVar);
    builder.append(charVar);
    builder.append(charVar);
    blackhole.consume(builder);
  }
}
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Use single quotes around 'x'.


'''
== Comments And Links
(visible only on this page)

=== on 16 Jun 2015, 17:03:28 Nicolas Peru wrote:
I really doubt this one should be activated by default. This is really something that you should activate either, to be anal on your style or because you have a perf critical application that actually cares about this, so any way I would not include it in default profile.

=== on 16 Jun 2015, 17:16:31 Ann Campbell wrote:
done [~nicolas.peru]

endif::env-github,rspecator-view[]
