<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when code manually iterates through SOQL query results to collect record IDs into a collection.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>When you need to extract IDs from SOQL query results, manually looping through records is inefficient and consumes unnecessary CPU time.</p>
</div>
<div class="paragraph">
<p>In Salesforce Apex, CPU time is limited (10 seconds for synchronous operations, 60 seconds for asynchronous). Every loop iteration adds to this CPU consumption, and with large datasets, this can quickly lead to governor limit exceptions.</p>
</div>
<div class="paragraph">
<p>The Map constructor provides a built-in optimization for this common pattern. When you pass a SOQL query to a Map constructor like <code>new Map&lt;Id, SObject&gt;(soqlQuery)</code>, Salesforce automatically creates a map where record IDs become the keys. This happens at the platform level without additional CPU overhead from your code.</p>
</div>
<div class="paragraph">
<p>This approach is not only more performant but also more concise and readable. You can then use <code>keySet()</code> to get the Set of IDs if needed, or work directly with the Map for lookups.</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="paragraph">
<p>Manual ID extraction loops consume unnecessary CPU time, which can lead to governor limit exceptions in environments with large datasets or complex processing. This becomes particularly problematic in batch operations or when multiple automations are running simultaneously.</p>
</div>
<div class="paragraph">
<p>The inefficiency compounds when dealing with hundreds or thousands of records, potentially causing transaction failures and poor user experience.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Replace the manual loop with a Map constructor that takes the SOQL query directly. Use the keySet() method to get the Set of IDs if needed.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-apex" data-lang="apex">List&lt;Account&gt; accList = [SELECT Id, Name FROM Account LIMIT 100];
Set&lt;Id&gt; setIds = new Set&lt;Id&gt;();
for(Account acc : accList) {
    setIds.add(acc.Id); // Noncompliant
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-apex" data-lang="apex">Map&lt;Id, Account&gt; accountMap = new Map&lt;Id, Account&gt;([SELECT Id, Name FROM Account LIMIT 100]);
Set&lt;Id&gt; setIds = accountMap.keySet();</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Apex Developer Guide - Maps - <a href="https://developer.salesforce.com/docs/atlas.en-us.apexcode.meta/apexcode/langCon_apex_collections_maps.htm">Official Salesforce documentation on Map collections and constructors</a></p>
</li>
<li>
<p>Apex CPU Time Limit Best Practices - <a href="https://developer.salesforce.com/docs/atlas.en-us.apexcode.meta/apexcode/apex_gov_limits.htm">Salesforce documentation on governor limits and CPU time optimization</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>