<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p><code>part of</code> directives should be used with strings, rather than with <code>library</code> names.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Often Dart libraries are small enough to be entirely contained in a single file.</p>
</div>
<div class="paragraph">
<p>However, when the file becomes too large and it&#8217;s not possible to split the library into smaller libraries (as advised by the Dart team in the note <a href="https://dart.dev/guides/libraries/create-packages#organizing-a-package">here</a>), Dart defines a <code>part of</code> directive, which allows developers to split the library into multiple smaller chunks.</p>
</div>
<div class="paragraph">
<p>When a library is split into multiple files, each part has to refer the library it belongs to, and it can do so with two similar but alternative syntaxes:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>by name identifier: <code>part of my_library;</code></p>
</li>
<li>
<p>by a URI string: <code>part of 'my_library.dart';</code></p>
</li>
</ul>
</div>
<div class="paragraph">
<p>The second syntax is more precise and flexible, because it explicitly identifies the file that contains the main body of the library, whereas a reference by identifier needs to be resolved to a URI by the Dart compiler.</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="paragraph">
<p>If the <code>part of</code> directive is used with a library name, the Dart compiler may resolve the library incorrectly, if multiple libraries with the same name but different paths exist in the package.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Convert the name identifier into a string, append the <code>.dart</code> extension, and prepend the relative path.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-dart" data-lang="dart">part of my_library;</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-dart" data-lang="dart">part of '../my_library.dart';</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Dart Docs - <a href="https://dart.dev/tools/linter-rules/use_string_in_part_of_directives">Dart Linter rule - use_string_in_part_of_directives</a></p>
</li>
<li>
<p>Dart Docs - <a href="https://dart.dev/guides/libraries/create-packages#organizing-a-package">Packages - Organizing a package</a></p>
</li>
</ul>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>The part-of directive uses a library name.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>The entire <code>part of</code> directive statement, including the semicolon: e.g. <code>part of my_library;</code>.</p>
</div>
</div>
</div>
</div>