Variables declared with ``++const++`` cannot be modified. Unfortunately, attempts to do so don't always raise an error; in a non-ES2015 environment, such an attempt might simply be ignored.


== Noncompliant Code Example

----
const pi = "yes, please";
pi = 3.14;  // Noncompliant
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

include::highlighting.adoc[]

endif::env-github,rspecator-view[]
