<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Overriding a parent class' method implementation with an <code>abstract</code> method is a terrible practice for a number of reasons:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>it blocks invocation of the original class' method by children of the <code>abstract</code> class.</p>
</li>
<li>
<p>it requires the <code>abstract</code> class' children to re-implement (copy/paste?) the original class' logic.</p>
</li>
<li>
<p>it violates the inherited contract.</p>
</li>
</ul>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">public class Parent {
  public int getNumber() {
    return 1;
  }
}

public abstract class AbstractChild {
  abstract public int getNumber();  // Noncompliant
}</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove this "abstract" method override.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_12_jun_2015_153457_ann_campbell_wrote">on 12 Jun 2015, 15:34:57 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>CodePro: Overriding a Non-abstract Method with an Abstract Method</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_12_jun_2015_155029_ann_campbell_wrote">on 12 Jun 2015, 15:50:29 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~nicolas.peru] I&#8217;m wondering if I should have rolled this into <a data-rspec-id="S2177" class="rspec-auto-link">RSPEC-2177</a>, altho I&#8217;m a bit reluctant to roll something that deserves a "dope-slap" tag (this one) into something that&#8217;s mainly an "oops".</p>
</div>
<div class="paragraph">
<p>WDYT?</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_15_jun_2015_202121_nicolas_peru_wrote">on 15 Jun 2015, 20:21:21 Nicolas Peru wrote:</h3>
<div class="paragraph">
<p>This one looks good and intent is fairly different: <a data-rspec-id="S2177" class="rspec-auto-link">RSPEC-2177</a> detects abused overloading whereas this one clearly indicates a bad design. As you point it out, severity is definitely not the same.</p>
</div>
</div>
</div>
</div>