<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This is an issue when using Starlette or FastAPI&#8217;s <code>TestClient</code> with httpx backend and calling HTTP methods like <code>get()</code>, <code>delete()</code>, <code>head()</code>, or <code>options()</code> with body-related parameters (<code>content</code>, <code>data</code>, <code>json</code>, or <code>files</code>).</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>When Starlette version 0.21.0 was released, the <code>TestClient</code> implementation changed from using the <code>requests</code> library to <code>httpx</code>. This change introduced API differences that can break existing test code.</p>
</div>
<div class="paragraph">
<p>In httpx, certain HTTP methods have stricter parameter requirements that align better with HTTP specifications. Specifically, the methods <code>get()</code>, <code>delete()</code>, <code>head()</code>, and <code>options()</code> do not accept body-related parameters like <code>content</code>, <code>data</code>, <code>json</code>, or <code>files</code>. This is because these HTTP methods typically should not include request bodies according to HTTP semantics:</p>
</div>
<div class="ulist">
<ul>
<li>
<p><code>GET</code> and <code>HEAD</code> are defined as safe methods that only retrieve data</p>
</li>
<li>
<p><code>OPTIONS</code> is used for discovering allowed methods</p>
</li>
<li>
<p>While <code>DELETE</code> can technically have a body, it&#8217;s uncommon and not supported by httpx&#8217;s convenience method</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>When you try to use these parameters with the affected methods, the code will fail at runtime with a <code>TypeError</code>, indicating that the method received unexpected keyword arguments. This breaks tests that previously worked with the <code>requests</code>-based implementation.</p>
</div>
<div class="paragraph">
<p>The httpx library provides the generic <code>client.request()</code> method as an alternative when you need to send body data with these HTTP methods, giving you full control over the request parameters.</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="paragraph">
<p>Test code will fail at runtime with a <code>TypeError</code>, preventing your test suite from running successfully. This can block development and deployment pipelines until the issue is resolved.</p>
</div>
<div class="paragraph">
<p>The error typically manifests as:</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-text" data-lang="text">TypeError: TestClient.get() got an unexpected keyword argument 'json'</code></pre>
</div>
</div>
<div class="paragraph">
<p>This issue commonly appears when:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>Upgrading Starlette or FastAPI to versions using httpx</p>
</li>
<li>
<p>Migrating existing test suites to newer framework versions</p>
</li>
<li>
<p>Running tests that worked with older versions of the framework</p>
</li>
</ul>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it_in_starlette">How to fix it in Starlette</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Replace the specific HTTP method call with the generic <code>request()</code> method. This method accepts all parameters and gives you full control over the HTTP request, including the ability to send body data with any HTTP method.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">from starlette.testclient import TestClient
from starlette.applications import Starlette

app = Starlette()
client = TestClient(app)

# Using get() with json parameter
response = client.get('/api/endpoint', json={'key': 'value'})  # Noncompliant</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">from starlette.testclient import TestClient
from starlette.applications import Starlette

app = Starlette()
client = TestClient(app)

# Using request() method instead
response = client.request('GET', '/api/endpoint', json={'key': 'value'})</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it_in_fastapi">How to fix it in FastAPI</h2>
<div class="sectionbody">
<div class="paragraph">
<p>When testing FastAPI endpoints, use <code>client.request()</code> instead of the specific method when you need to include body parameters with <code>GET</code>, <code>DELETE</code>, <code>HEAD</code>, or <code>OPTIONS</code> requests.</p>
</div>
<div class="sect2">
<h3 id="_code_examples_2">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example_2">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">from fastapi import FastAPI
from fastapi.testclient import TestClient

app = FastAPI()
client = TestClient(app)

# Using delete() with data parameter
response = client.delete('/api/resource/123', data='some data')  # Noncompliant</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution_2">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">from fastapi import FastAPI
from fastapi.testclient import TestClient

app = FastAPI()
client = TestClient(app)

# Using request() method for delete with data
response = client.request('DELETE', '/api/resource/123', data='some data')</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>HTTPX Documentation - Request API - <a href="https://www.python-httpx.org/api/#request">Official httpx documentation for the request() method</a></p>
</li>
<li>
<p>Starlette TestClient Documentation - <a href="https://www.starlette.io/testclient/">Official documentation for Starlette&#8217;s TestClient</a></p>
</li>
<li>
<p>FastAPI Testing Documentation - <a href="https://fastapi.tiangolo.com/tutorial/testing/">Official FastAPI guide for testing applications</a></p>
</li>
<li>
<p>bump-testclient Migration Tool - <a href="https://pypi.org/project/bump-testclient/">Automated tool to help migrate TestClient code from requests to httpx</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>