<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Modern browsers ignore unneeded, trailing commas, so there are no negatives to having them unless you&#8217;re supporting an IE 8 application. Since they make adding new properties simpler, their use is preferred. This rule raises an issue when the last item in a multiline construct (array or object literal, import or export statement, function declaration or call) does not end with a trailing comma and does not lie on the same line as the closing curly brace, bracket or parenthesis.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-text" data-lang="text">var joe = {
  fname: "Joe",
  lname: "Smith"      // Noncompliant
};</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-text" data-lang="text">var joe = {
  fname: "Joe",
  lname: "Smith",    // OK
};

var joe = {
  fname: "Joe",
  lname: "Smith"};   // OK</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Add a trailing comma to this item of the list.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p><code>foo: "bar"</code></p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_4_jun_2020_174144_ann_campbell_wrote">on 4 Jun 2020, 17:41:44 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~nicolas.harraudeau] might want to add Python to this. Ref https://www.youtube.com/watch?v=2ct_kO3mCDc</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_4_jun_2020_181706_nicolas_harraudeau_wrote">on 4 Jun 2020, 18:17:06 Nicolas Harraudeau wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] It would make sense yes, but I wouldn&#8217;t add it right now as it will probably create a lot of noise like it does on Javascript, and it is a minor code smell. Let&#8217;s keep it for later.</p>
</div>
</div>
</div>
</div>