include::../description.adoc[]

include::../how-to-fix-it.adoc[]

=== Code examples

==== Noncompliant code example

[source,abap,diff-id=1,diff-type=noncompliant]
----
IF sy-subrc EQ 42.      " Noncompliant - 5 is a magic number
  screen-request = 'OK'.
ENDIF.
----

==== Compliant solution

[source,abap,diff-id=1,diff-type=compliant]
----
answer = 42.
IF sy-subrc EQ answer.  " Compliant
  screen-request = 'OK'.
ENDIF.
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
