== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,abap]
----
IF sy-subrc EQ 42.
  screen-request = 45.
ENDIF.
----

=== Compliant solution

[source,abap]
----
answer = 42.
IF sy-subrc EQ answer.
  screen-request = 45.
ENDIF.
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
