<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when the <code>tf.linalg.inv</code> function argument doesn&#8217;t meet the requirements.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>The <code>tf.linalg.inv</code> is a function used to compute the inverse of a matrix. The argument value should meet the following requirements:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>the matrix should be non-singular</p>
</li>
<li>
<p>the matrix should be square</p>
</li>
<li>
<p>the matrix element should be one of the following types: <code>double</code>, <code>float</code>, <code>half</code>, <code>complex64</code>, <code>complex128</code></p>
</li>
</ul>
</div>
<div class="paragraph">
<p>Otherwise, it will lead to <code>InvalidArgumentError</code></p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">import tensorflow as tf

x = tf.constant([1., 2., 3.])
y = tf.linalg.inv(x)  # Noncompliant: InvalidArgumentError: Input matrix must be square</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">import tensorflow as tf

x = tf.constant([[1., 2.], [3., 4.]])
y = tf.linalg.inv(x)  # OK</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Tensorflow documentation - <a href="https://www.tensorflow.org/api_docs/python/tf/linalg/inv">tf.linalg.inv</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>