This rule raises an issue when `count` is called without arguments on arrays, hashes, or other collections to determine the total number of elements.

== Why is this an issue?

Using `count` without arguments on collections creates unnecessary overhead and unclear intent. The `count` method is designed for conditional counting with blocks or arguments, not simple element counting.

When you call `count` without parameters, it still goes through the enumeration logic internally, which is less efficient than the direct size calculation provided by `size` or `length`. These methods are optimized for simple element counting and directly return the collection's size without iteration.

Moreover, `count` semantically implies that some form of filtering or conditional logic is involved. When readers see `count` in code, they expect to find a block or condition that determines which elements to count. Using `count` for simple element counting can confuse other developers and make the code less readable.

For arrays specifically, `size` has been available since earlier Ruby versions, providing better backward compatibility than `count`, which was added later.

=== What is the potential impact?

Using `count` instead of `size` or `length` for simple element counting can lead to:

* **Performance degradation**: Unnecessary enumeration overhead, especially noticeable with large collections
* **Reduced code clarity**: Other developers may waste time looking for conditional logic that doesn't exist
* **Compatibility issues**: Potential problems with older Ruby versions where `count` might not be available on all collection types

== How to fix it

Replace `count` with `size` when determining the total number of elements in arrays or hashes without conditions.

=== Code examples

==== Noncompliant code example

[source,ruby,diff-id=1,diff-type=noncompliant]
----
users = ['alice', 'bob', 'charlie']
total_users = users.count  # Noncompliant

config = { 'host' => 'localhost', 'port' => 3000 }
config_size = config.count  # Noncompliant
----

==== Compliant solution

[source,ruby,diff-id=1,diff-type=compliant]
----
users = ['alice', 'bob', 'charlie']
total_users = users.size  # Clear intent, optimal performance

config = { 'host' => 'localhost', 'port' => 3000 }
config_size = config.size  # Clear intent, optimal performance
----

== Resources

=== Documentation

 * Ruby Array Documentation - https://ruby-doc.org/core/Array.html[Official Ruby documentation for Array methods including size, length, and count]

 * Ruby Hash Documentation - https://ruby-doc.org/core/Hash.html[Official Ruby documentation for Hash methods including size and length]
