include::../description.adoc[]

== Noncompliant Code Example

Given parameters:

* className:System.Console

----
System.Console.WriteLine("foo");  // Noncompliant
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::parameters.adoc[]

include::../highlighting.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
