== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,vbnet]
----
If condition1 Then
    If condition2 Then ' Noncompliant
        ' ...
    End If
End If
----

=== Compliant solution

[source,vbnet]
----
If condition1 AndAlso condition2 Then
    ' ...
End If
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../highlighting.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
