include::../description.adoc[]

== Noncompliant Code Example

With default provided regular expression: ``++^(_|[a-zA-Z0-9]+)$++``:

[source,go]
----
func execute_all() {
...
}
----

== Compliant Solution

[source,go]
----
func executeAll() {
...
}
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::parameters.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
