== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

With default provided regular expression: ``++^(_|[a-zA-Z0-9]+)$++``:

[source,go]
----
func execute_all() {
...
}
----

=== Compliant solution

[source,go]
----
func executeAll() {
...
}
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

=== Parameters

.format
****
_STRING_

----
^[a-zA-Z0-9]+$
----

Regular expression used to check the function names against
****


'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
