== How to fix it in Fiber

=== Code examples

include::../../common/fix/code-rationale.adoc[]

==== Noncompliant code example

[source,go,diff-id=11,diff-type=noncompliant]
----
import (
	"github.com/gofiber/fiber/v3"
	"github.com/gofiber/fiber/v3/log"
)

app.Get("/example", func(c fiber.Ctx) error {
	taint := c.Query("q")
	log.Infof("User input: ", taint) // Noncompliant
	return c.SendString("Example")
})
----

==== Compliant solution

[source,go,diff-id=11,diff-type=compliant]
----
import (
	"strings"

	"github.com/gofiber/fiber/v3"
	"github.com/gofiber/fiber/v3/log"
)

app.Get("/example", func(c fiber.Ctx) error {
	taint := c.Query("q")
	sanitized := strings.ReplaceAll(strings.ReplaceAll(taint, "\n", "_"), "\r", "_")
	log.Infof("User input: ", sanitized)
	return c.SendString("Example")
})
----

include::../../common/fix/how-does-this-work.adoc[]