== How to fix it in Go Standard Library

=== Code examples

include::../../common/fix/code-rationale.adoc[]

==== Noncompliant code example

[source,go,diff-id=1,diff-type=noncompliant]
----
func handler(w http.ResponseWriter, req *http.Request) {
	userInput := req.FormValue("input")
	log.Println("User input: ", userInput) // Noncompliant
}
----

==== Compliant solution

[source,go,diff-id=1,diff-type=compliant]
----
func handler(w http.ResponseWriter, req *http.Request) {
	userInput := req.FormValue("input")
	sanitizedInput := strings.ReplaceAll(strings.ReplaceAll(userInput, "\n", "_"), "\r", "_")
	log.Println("User input: ", sanitizedInput)
}
----

include::../../common/fix/how-does-this-work.adoc[]