<div class="paragraph">
<p>Shared naming conventions allow teams to collaborate efficiently. This rule checks that all function names match a provided regular expression.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="paragraph">
<p>With default provided regular expression: <code>^[a-z][a-zA-Z0-9]*$</code>:</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">void DoSomething (void);</code></pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-text" data-lang="text">void doSomething (void);</code></pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_see">See</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_parameters">Parameters</h3>
<div class="sidebarblock">
<div class="content">
<div class="title">format</div>
<div class="paragraph">
<p><em>STRING</em></p>
</div>
<div class="listingblock">
<div class="content">
<pre>^([A-Z0-9_]*|[a-z0-9_]*)$</pre>
</div>
</div>
<div class="paragraph">
<p>Regular expression used to check the [method|function|subroutine] names against</p>
</div>
</div>
</div>
<div class="sidebarblock">
<div class="content">
<div class="title">param2</div>
<div class="listingblock">
<div class="content">
<pre>42</pre>
</div>
</div>
<div class="paragraph">
<p>Some integer</p>
</div>
</div>
</div>
</div>
</div>
</div>