include::../description.adoc[]

== Noncompliant Code Example

With default provided regular expression: ``++^[a-z][a-zA-Z0-9]*$++``:

----
void DoSomething (void);
----

== Compliant Solution

----
void doSomething (void);
----

=== Parameters

.format
****
_STRING_

----
^([A-Z0-9_]*|[a-z0-9_]*)$
----

Regular expression used to check the [method|function|subroutine] names against
****

.param2
****
----
42
----

Some integer
****
