include::../description.adoc[]

== Noncompliant Code Example

With default provided regular expression: ``++^[a-z][a-zA-Z0-9]*$++``:

----
void DoSomething (void);
----

== Compliant Solution

----
void doSomething (void);
----
