<body>
<div class="paragraph">
<p>Shared naming conventions allow teams to collaborate efficiently. This rule checks that all function names match a provided regular expression.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="paragraph">
<p>With default provided regular expression: <code>^[a-z][a-zA-Z0-9]*$</code>:</p>
</div>
<div class="listingblock">
<div class="content">
<pre>void DoSomething (void);</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>void doSomething (void);</pre>
</div>
</div>
<div class="sect2">
<h3 id="_parameters">Parameters</h3>
<div class="dlist">
<dl>
<dt class="hdlist1"><strong>format</strong>: STRING [ <code>^([A-Z0-9_]*|[a-z0-9_]*)$</code> ]</dt>
<dd>
<p>Regular expression used to check the [method|function|subroutine] names against</p>
</dd>
<dt class="hdlist1"><strong>param2</strong>: INTEGER [ <code>42</code> ]</dt>
<dd>
<p>Some integer</p>
</dd>
</dl>
</div>
</div>
</div>
</div>
</body>
