<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when manual time calculations use time-related constants like 1000 (milliseconds per second), 60 (seconds per minute), 3600 (seconds per hour), or 86400 (seconds per day) instead of Java&#8217;s built-in time utilities.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Manual time calculations with constants like <code>1000</code>, <code>60</code>, <code>3600</code>, and <code>86400</code> are not self-documenting. Seeing <code>duration * 60 * 1000</code> requires mental calculation to understand it converts minutes to milliseconds. These calculations are also error-prone - it&#8217;s easy to forget a zero or use the wrong conversion factor.</p>
</div>
<div class="paragraph">
<p>Java provides built-in utilities specifically for time operations:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>The <code>TimeUnit</code> enum offers methods like <code>toMillis()</code>, <code>toSeconds()</code>, and <code>convert()</code></p>
</li>
<li>
<p>The <code>Duration</code> class provides similar functionality with a modern API</p>
</li>
<li>
<p>The <code>ChronoUnit</code> enum is available for advanced time operations</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>These utilities make your intent clear. <code>TimeUnit.MINUTES.toMillis(duration)</code> immediately shows you&#8217;re converting minutes to milliseconds, eliminating mental math and calculation errors.</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="paragraph">
<p>Using time-related constants in manual calculations reduces code readability and increases the risk of bugs. Calculation errors in time-sensitive code can lead to incorrect timeouts, scheduling issues, or performance problems in production systems.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Replace manual time calculations with TimeUnit enum methods. Use <code>toMillis()</code>, <code>toSeconds()</code>, or other conversion methods to make your intent clear.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">int minutes = 15;
long millis = minutes * 60 * 1000; // Noncompliant</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">int minutes = 15;
long millis = TimeUnit.MINUTES.toMillis(minutes);</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>TimeUnit (Java SE 17 &amp; JDK 17) - <a href="https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/util/concurrent/TimeUnit.html">Official documentation for the TimeUnit enum with conversion methods</a></p>
</li>
<li>
<p>Duration (Java SE 17 &amp; JDK 17) - <a href="https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/time/Duration.html">Official documentation for the Duration class in the java.time package</a></p>
</li>
<li>
<p>ChronoUnit (Java SE 17 &amp; JDK 17) - <a href="https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/time/temporal/ChronoUnit.html">Official documentation for ChronoUnit enum for advanced time operations</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>