<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>For optimal code readability, annotation arguments should be specified in the same order that they were declared in the annotation definition.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">@interface Pet {
    String name();
    String surname();
}

@Pet(surname ="", name="") // Noncompliant</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">@interface Pet {
    String name();
    String surname();
}

@Pet(name ="", surname="") // Compliant</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Reorder these annotation arguments to match their order of declaration.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_1_sep_2015_123047_ann_campbell_wrote">on 1 Sep 2015, 12:30:47 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>In my editing, [~nicolas.peru], I haven&#8217;t filled in a message. It&#8217;s unclear to me whether you want to raise an issue per argument or an issue per annotation. So I&#8217;ve left that to you to fill in. :-)</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_23_sep_2015_133436_nicolas_peru_wrote">on 23 Sep 2015, 13:34:36 Nicolas Peru wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] can you review message and mark as complete if it is ok for you ?</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_28_sep_2015_180753_ann_campbell_wrote">on 28 Sep 2015, 18:07:53 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>done [~nicolas.peru]</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_1_oct_2019_113216_michael_gumowski_wrote">on 1 Oct 2019, 11:32:16 Michael Gumowski wrote:</h3>
<div class="paragraph">
<p>This rule is deprecated, without alternative. Nothing in JVMLS, or JLS, guarantees that order of annotation members will be preserved once compiled. See SONARJAVA-3194 for more details.</p>
</div>
</div>
</div>
</div>