Sharing some naming conventions is a key point to make it possible for a team to efficiently collaborate.

This rule allows to check that all parameter names match a provided regular expression.


The default configuration is the one recommended by Microsoft:


* Camel casing, starting with a lower case character, e.g. backColor
* Short abbreviations of 2 letters can be capitalized only when not at the beginning, e.g. id, productID
* Longer abbreviations need to be lower cased, e.g. html

== Noncompliant Code Example

With the default regular expression ``++^[a-z][a-z0-9]*([A-Z]{1,3}[a-z0-9]+)*([A-Z]{2})?$++``

----
Module Module1
    Sub GetSomething(ByVal ID As Integer) ' Noncompliant
    End Sub
End Module
----

== Compliant Solution

----
Module Module1
    Sub GetSomething(ByVal id As Integer) ' Compliant
    End Sub
End Module
----
ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::parameters.adoc[]

endif::env-github,rspecator-view[]
