== Why is this an issue?

The ``++parseInt++`` function has two versions, one that takes a base value as a second argument, and one that does not. Unfortunately using the single-arg version can result in unexpected results on older browsers. 


=== Noncompliant code example

[source,javascript]
----
parseInt("010");  // Noncompliant; pre-2013 browsers may return 8 
----


=== Compliant solution

[source,javascript]
----
parseInt("010", 10);
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Add the base to this "parseInt" call.


endif::env-github,rspecator-view[]
