<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when populating a NumPy array with the Python <code>range</code> function.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Python lists and NumPy arrays have a fundamental difference: Python lists can hold different types of elements,
while NumPy arrays are homogeneous, meaning all elements have the same type.
This difference allows for NumPy arrays for better memory management, as they can store data in contiguous memory locations.
It also allows for optimization of certain operations, for example with the help of parallelization.</p>
</div>
<div class="paragraph">
<p>These benefits are lost when a NumPy array is created by first generating a Python list by using the range function.
Using <code>np.arange</code> instead allows for improved readability and yields better performance in terms of memory and speed of execution.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>To fix this issue use the <code>np.arange</code> function to create a new NumPy array, instead of the builtin <code>range</code> function.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">import numpy as np

arr = []
for i in range(10):
    arr.append(i)

arr = np.array(arr) # Noncompliant

arr = np.array([*range(10)]) # Noncompliant</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">import numpy as np

arr = np.arange(10) # Compliant</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>NumPy Documentation - <a href="https://numpy.org/doc/stable/user/basics.creation.html#intrinsic-numpy-array-creation-functions">Intrinsinc NumPy array creation functions</a></p>
</li>
</ul>
</div>
</div>
<div class="sect2">
<h3 id="_articles_blog_posts">Articles &amp; blog posts</h3>
<div class="ulist">
<ul>
<li>
<p>Medium Blog - <a href="https://medium.com/@24littledino/range-python-vs-arange-numpy-3dc2953b9467">Range(Python) vs. Arange(NumPy)</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>