<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>An implicit template specialization does not instantiate every member of the template. Where instantiation of a member would result in an ill-formed program it is not clear that the template should be used with the supplied template-arguments.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">template &lt;typename T&gt;
class A
{
public:
  void f1 ()
  {
    // ...
  }
  void f2 ()
  {
    T t;
    t.x = 0; // Will only work for types that have a .x member
  }
};

void b ()
{
  A&lt;int32_t&gt; a; // A&lt;int32_t&gt;::f2 is not instantiated.
  a.f1 ();
}

template class A&lt;int32_t&gt;; // Noncompliant, instantiation of f2 results in "ill-formed" program.</code></pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p>MISRA C&#43;&#43;:2008, 14-7-2</p>
</li>
</ul>
</div>
<hr>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_21_oct_2014_130033_samuel_mercier_wrote">on 21 Oct 2014, 13:00:33 Samuel Mercier wrote:</h3>
<div class="paragraph">
<p>Not sure I understand correctly this rule, since the line marked non-compliant raises a compile-time error, at least with gcc/clang&#8230;&#8203;</p>
</div>
<div class="paragraph">
<p>Should the instantiation in function b raise an issue?</p>
</div>
</div>
</div>
</div>