<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when a TimeUnit conversion method is called to convert to the same unit as the source. The rule applies to all TimeUnit conversion methods like <code>toSeconds()</code>, <code>toMinutes()</code>, <code>toHours()</code>, etc., when the source and target units are identical (e.g., <code>TimeUnit.SECONDS.toSeconds()</code>).</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Converting a time value from one unit to the same unit is redundant and serves no purpose. This typically indicates a copy-paste error, leftover code from refactoring, or a misunderstanding of TimeUnit conversions.</p>
</div>
<div class="paragraph">
<p>These redundant conversions make code harder to read and may mislead other developers about the intended behavior.</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="paragraph">
<p>This reduces code clarity and may indicate a logical error where a different unit conversion was intended.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Remove the redundant conversion and use the value directly.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">long durationInSeconds = TimeUnit.SECONDS.toSeconds(duration); // Noncompliant</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">long durationInSeconds = duration;</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>TimeUnit (Java SE 17 &amp; JDK 17) - <a href="https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/util/concurrent/TimeUnit.html">Official Java documentation for the TimeUnit enum and its conversion methods</a></p>
</li>
</ul>
</div>
</div>
<div class="sect2">
<h3 id="_related_rules">Related rules</h3>
<div class="ulist">
<ul>
<li>
<p><a data-rspec-id="S1905" class="rspec-auto-link">RSPEC-1905</a> - <a href="https://rules.sonarsource.com/java/RSPEC-1905/">Redundant casts should not be used - similar pattern of removing unnecessary operations</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>