If the ``++width++`` and ``++height++`` attributes are set, the space required for the image is reserved immediately by the browser, even before it actually starts to load the image.

Without those attributes, the page layout constantly changes as images are loaded until they are all loaded, which can disorient users.


== Noncompliant Code Example

[source,html]
----
<img src="logo.png" alt="My Company" />                           <!-- Non-Compliant -->
----


== Compliant Solution

----
<img src="logo.png" alt="My Company" width="100" height="50" />   <!-- Compliant -->
----



ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
