== Why is this an issue?

Where a set of operators is overloaded, it is important that the interactions between the operators meet developer expectations.


=== Noncompliant code example

[source,cpp]
----
class A
{
public:
  A& operator= ( A const & rhs );
};

A & operator += ( A const & lhs, A const & rhs );
A const operator + ( A const & lhs, A const & rhs )  // Noncompliant
{
  return lht -= rhs;
}
void f ( A a1, A a2 )
{
  A x;
  x = a1 + a2;
  a1 += a2;
  if ( x == a1 ) // This should be true, but it's not
  { 
  } 
}
----


== Resources

* MISRA {cpp}:2008, 5-17-1


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Refactor the definition of opreators "xxx" and "yyy" to yield semantically equivalent result.


'''
== Comments And Links
(visible only on this page)

=== is related to: S3083

=== on 23 Oct 2014, 14:24:52 Samuel Mercier wrote:
We can probably not detect that...

=== on 23 Oct 2014, 20:54:08 Ann Campbell wrote:
\[~samuel.mercier] please follow the conventions on this rule anyway. That or close it.

=== on 17 Jul 2015, 09:22:37 Ann Campbell wrote:
Already assigned to you [~evgeny.mandrikov]. Note that I've altered the code sample.

endif::env-github,rspecator-view[]
