<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when date or datetime values are inserted into databases using string concatenation in SQL queries or when using <code>setString()</code> method with PreparedStatement for date parameters.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Using string operations for date values in database operations creates data integrity problems.</p>
</div>
<div class="paragraph">
<p>String-based date handling bypasses proper type validation. When dates are treated as strings, invalid date values can reach the database without validation, potentially causing silent data corruption or insertion failures.</p>
</div>
<div class="paragraph">
<p>Using <code>setString()</code> for date parameters defeats the type safety that PreparedStatement provides. The JDBC driver cannot validate that the string represents a valid date, allowing malformed data to pass through.</p>
</div>
<div class="paragraph">
<p>PreparedStatement offers proper type-safe methods for date handling: <code>setDate()</code> for SQL DATE columns, <code>setTime()</code> for SQL TIME columns, and <code>setTimestamp()</code> for SQL TIMESTAMP columns. These methods ensure proper validation and type conversion.</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="paragraph">
<p>Data integrity issues may result in incorrect date storage, leading to business logic errors, reporting inaccuracies, or application failures when processing date-based operations. Invalid dates may be silently accepted, causing downstream processing problems.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Parse and validate date strings using java.sql.Date for database operations using setDate().</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">String dateStr = "2010-05-01";
PreparedStatement pstmt = connection.prepareStatement("INSERT INTO events (event_date) VALUES (?)");
pstmt.setString(1, dateStr); // Noncompliant
pstmt.executeUpdate();</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">String dateStr = "2010-05-01";
java.sql.Date sqlDate = java.sql.Date.valueOf(dateStr);
PreparedStatement pstmt = connection.prepareStatement("INSERT INTO events (event_date) VALUES (?)");
pstmt.setDate(1, sqlDate);
pstmt.executeUpdate();</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Java SE Documentation - java.sql.PreparedStatement - <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.sql/java/sql/PreparedStatement.html">Complete API documentation for PreparedStatement methods including date setters</a></p>
</li>
<li>
<p>Java SE Documentation - java.sql.Date - <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.sql/java/sql/Date.html">Documentation for java.sql.Date class used for SQL DATE values</a></p>
</li>
<li>
<p>Java SE Documentation - LocalDate - <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/time/LocalDate.html">Modern Java API for date handling without time zone information</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>