== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

With default provided regular expression: ``++^[A-Z][a-zA-Z0-9]*++``

[source,swift]
----
enum someEnumeration { // Non-Compliant
    case Bar
}
----

=== Compliant solution

[source,swift]
----
enum SomeEnumeration {
    case Bar
}
----
ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

include::../parameters.adoc[]

endif::env-github,rspecator-view[]
