== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

With default provided regular expression: ``++^[A-Z][a-zA-Z0-9]*++``

[source,swift]
----
enum someEnumeration { // Non-Compliant
    case Bar
}
----

=== Compliant solution

[source,swift]
----
enum SomeEnumeration {
    case Bar
}
----
ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

key: format

description: Regular expression used to check the enumeration type names against

* default: ^[A-Z][a-zA-Z0-9]*$


=== Parameters

.format
****

----
^([A-Z]{1,3}[a-z0-9]+)*([A-Z]{2})?$
----

Regular expression used to check the enumeration type names against
****


endif::env-github,rspecator-view[]
