include::../description.adoc[]

== Noncompliant Code Example

With default provided regular expression: ``++^[A-Z][a-zA-Z0-9]*++``

----
enum someEnumeration { // Non-Compliant
    case Bar
}
----

== Compliant Solution

----
enum SomeEnumeration {
    case Bar
}
----
ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

include::../parameters.adoc[]

endif::env-github,rspecator-view[]
