== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,python]
----
from pyDes import *

data = "content to be encrypted"
k = des("DESCRYPT", CBC, "\0\0\0\0\0\0\0\0", pad=None, padmode=PAD_PKCS5) // Noncompliant
d = k.encrypt(data)
----

=== Compliant solution

[source,python]
----
import hashlib

m = hashlib.sha256() // Compliant
m.update(data)

d = m.hexdigest()
----

include::../see.adoc[]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
