<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Shared naming conventions allow teams to collaborate efficiently. This rule checks that all method parameters follow a naming convention.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="paragraph">
<p>Using the default importing  and returning regular expressions <code>I_[a-zA-Z_\-0-9]``, and ``R_[a-zA-Z_\-0-9]</code>:</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-abap" data-lang="abap">    METHODS m_fact IMPORTING i1  TYPE i
                       value(i2) TYPE i
                   RETURNING value(factorial) TYPE i.</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-abap" data-lang="abap">    METHODS m_fact IMPORTING I_1  TYPE i
                       value(i2) TYPE i
                   RETURNING value(R_factorial) TYPE i.</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Rename this parameter to comply with the required naming convention: xxx</p>
</div>
</div>
<div class="sect2">
<h3 id="_parameters">Parameters</h3>
<div class="sidebarblock">
<div class="content">
<div class="title">importing_key</div>
<div class="listingblock">
<div class="content">
<pre>I_[a-zA-Z_-0-9]+</pre>
</div>
</div>
<div class="paragraph">
<p>Naming convention for Importing parameter</p>
</div>
</div>
</div>
<div class="sidebarblock">
<div class="content">
<div class="title">exporting_key</div>
<div class="listingblock">
<div class="content">
<pre>E_[a-zA-Z_-0-9]+</pre>
</div>
</div>
<div class="paragraph">
<p>Naming convention for Exporting parameter</p>
</div>
</div>
</div>
<div class="sidebarblock">
<div class="content">
<div class="title">changing_key</div>
<div class="listingblock">
<div class="content">
<pre>C_[a-zA-Z_-0-9]+</pre>
</div>
</div>
<div class="paragraph">
<p>Naming convention for Changing parameter</p>
</div>
</div>
</div>
<div class="sidebarblock">
<div class="content">
<div class="title">returning_key</div>
<div class="listingblock">
<div class="content">
<pre>R_[a-zA-Z_-0-9]+</pre>
</div>
</div>
<div class="paragraph">
<p>Naming convention for Returning parameter</p>
</div>
</div>
</div>
</div>
</div>
</div>