=== Noncompliant code example

Using the default importing  and returning regular expressions ``++I_[a-zA-Z_\-0-9]+++``, and ``++R_[a-zA-Z_\-0-9]+++``:

[source,text]
----
    METHODS m_fact IMPORTING i1  TYPE i
                       value(i2) TYPE i
                   RETURNING value(factorial) TYPE i.
----
