== How to fix it in Android

=== Code examples

==== Noncompliant code example

[source,kotlin,diff-id=21,diff-type=noncompliant]
----
class ExampleActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        val aWebView = WebView(this)
        aWebView.settings.javaScriptEnabled = true
        setContentView(aWebView)

        val name = intent.getStringExtra("name") ?: "Guest"
        aWebView.evaluateJavascript("greeting('$name')", null) // Noncompliant
    }
}
----

==== Compliant solution

[source,kotlin,diff-id=21,diff-type=compliant]
----
class ExampleActivity : AppCompatActivity() {

    // Only allow letters and spaces in the name.
    private val nameRegex = Regex("^[a-zA-Z ]+$")

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        val aWebView = WebView(this)
        aWebView.settings.javaScriptEnabled = true
        setContentView(aWebView)

        val name = intent.getStringExtra("name") ?: "Guest"
        if (name.matches(nameRegex)) {
            aWebView.evaluateJavascript("greeting('$name')", null) // Compliant
        }
    }
}
----

