== Why is this an issue?

{cpp}20 extends the ``++using++`` declaration to ``++enum class++``. ``++using enum++`` introduces all the ``++enum++`` constants into the current scope.

[source,cpp]
----
enum class rgbaColorChannel { red, green, blue, alpha };

std::string_view toString(rgbaColorChannel channel) {
  switch (channel) {
    using enum rgbaColorChannel;
    case red:   return "red";
    case green: return "green";
    case blue:  return "blue";
    case alpha: return "alpha";
  }
}
----

As with other ``++using++`` declarations, ``++using enum++`` improves readability when used in small scopes yet it generates confusion in large scopes.


The ``++switch++`` statement, as in the example above, is a natural scope for ``++using enum++``.


This rule reports scopes that use a particular ``++enum class++`` extensively and can benefit from ``++using enum++`` declaration. For example, it reports most ``++switch++`` statements applied to an ``++enum++`` value.

=== Exceptions

The rule does not apply if adding the `using enum` clause would create a name collision or reduce readability by shadowing a name.

== How to fix it

Add `using enum` inside the relevant scope and replace the qualified identifiers with the corresponding enumerator names.

=== Code examples

==== Noncompliant code example

[source,cpp,diff-id=1,diff-type=noncompliant]
----
enum class DayOfTheWeek { mon, tue, wed, thu, fri, sat, sun };

bool isWorkDay(DayOfTheWeek day) {
  switch(day) { // Noncompliant: case statements are too verbose
    case DayOfTheWeek::mon: return true;
    case DayOfTheWeek::tue: return true;
    case DayOfTheWeek::wed: return true;
    case DayOfTheWeek::thu: return true;
    case DayOfTheWeek::fri: return true;
    case DayOfTheWeek::sat: return false;
    case DayOfTheWeek::sun: return false;
  }
}
----


==== Compliant solution

[source,cpp,diff-id=1,diff-type=compliant]
----
enum class DayOfTheWeek { mon, tue, wed, thu, fri, sat, sun };

bool isWorkDay(DayOfTheWeek day) {
  switch(day) {
    using enum DayOfTheWeek;

    case mon: return true;
    case tue: return true;
    case wed: return true;
    case thu: return true;
    case fri: return true;
    case sat: return false;
    case sun: return false;
  }
}
----

==== Noncompliant code example

[source,cpp,diff-id=2,diff-type=noncompliant]
----
enum class rgbaColorChannel { red, green, blue, alpha };

rgbaColorChannel chooseChannel(rgbaColorChannel background, rgbaColorChannel foreground) {
  auto result = background;
  if ((background == rgbaColorChannel::red || background == rgbaColorChannel::green) && foreground != rgbaColorChannel::red)
    result = foreground;
  return result;
}
----


==== Compliant solution

[source,cpp,diff-id=2,diff-type=compliant]
----
enum class rgbaColorChannel { red, green, blue, alpha };

rgbaColorChannel chooseChannel(rgbaColorChannel background, rgbaColorChannel foreground) {
  using enum rgbaColorChannel;
  auto result = background;
  if ((background == red || background == green) && foreground != red)
    result = foreground;
  return result;
}
----
