== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

With a maximum number of 4 parameters:

[source,csharp]
----
public void doSomething(int param1, int param2, int param3, string param4, long param5)
{
    // ...
}
----

=== Compliant solution

[source,csharp]
----
public void doSomething(int param1, int param2, int param3, string param4)
{
    // ...
}
----

=== Exceptions

[source,csharp]
----
public class BaseClass
{
    public BaseClass(int param1)
    {
        // ...
    }
}

public class DerivedClass : BaseClass
{
    public DerivedClass(int param1, int param2, int param3, string param4, long param5) : base(param1) // Compliant, the parameters intended for the base class constructor do not count in the sum of the parameter list.
    {
        // ...
    }
}
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

\[Constructor|Method|Lambda|Delegate] has {0} parameters, which is greater than the {1} authorized.


include::../parameters.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
