include::../description.adoc[]

== Noncompliant Code Example

With a maximum number of 4 parameters:

[source,csharp]
----
public void doSomething(int param1, int param2, int param3, string param4, long param5)
{
...
}
----

== Compliant Solution

[source,csharp]
----
public void doSomething(int param1, int param2, int param3, string param4)
{
...
}
----

== Exceptions

[source,csharp]
----
public BaseClass(int param1)
{
...
}

public ExtensionClass : BaseClass
{
	public void ExtensionClass1(int param1, int param2, int param3, string param4, long param5) : base(param1) // Compliant, the parameters intended for the base class constructor do not count in the sum of the parameter list.
	{
	...
	}
}

public TestClass(int param1, int param2, int param3, string param4, long param5)
{
	public void TestClass(int param1, int param2, int param3, string param4, long param5) // Noncompliant
	{
	...
	}
}
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

include::../parameters.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
