== Noncompliant Code Example

[source,text]
----
class BaseException { };
class DerivedException: public BaseException { };

try
{
  // ...
}
catch ( BaseException &b ) // Will catch DerivedException as well
{
  // ...
}
catch ( DerivedException &d ) // Noncompliant, the previous handled effectively hides this one
{ 
  // Any code here will be unreachable, 
}
----
