This rule raises an issue when the dimensions of the input arrays for the ``++numpy.dot++`` function are not compatible for matrix multiplication.

== Why is this an issue?

The ``++numpy.dot()++`` function (or equivalently, the @ operator for matrix multiplication) performs matrix multiplication between two arrays.

Matrix multiplication is a binary operation that produces a matrix from two matrices. The resulting matrix, known as the matrix product, has the number of rows of the first and the number of columns of the second matrix.

If ``++A++`` is an ``++m x n++`` matrix and ``++B++`` is an ``++n x p++`` matrix,
then the resulting matrix will be ``++C++`` with ``++m x p++`` shape where each value is calculated by the formula:
[source,text]
----
for i = 1, ..., m and j = 1, ..., p
C[i, j] = A[i, 1] * B[1, j] + A[i, 2] * B[2, j] + ... + A[i, n] * B[n, j]
----

For example:
[source,python]
----
import numpy as np

a = [[1, 2],
     [3, 4],
     [5, 6]]
b = [[1, 2, 3],
     [4, 5, 6]]
# Perform matrix multiplication using np.dot() or @ operator
result = np.dot(a, b)
print(result)
----
The output will be:
[source,text]
----
[[ 9 12 15]
 [19 26 33]
 [29 40 51]]
----

For a matrix multiplication to be valid, the number of columns in the first matrix must be equal to the number of rows in the second matrix.
You can use ``++numpy.shape()++`` function to check the dimensions of the arrays before performing the operation.

== Hot to fix it

Make sure that the number of columns in the first matrix is equal to the number of rows in the second matrix.

=== Code examples

==== Noncompliant code example

[source,python,diff-id=1,diff-type=noncompliant]
----
import numpy as np

a = [[1, 2, 3], [4, 5, 6]]
b = [1, 2, 3, 4]
c = np.dot(a, b) # Noncompliant: ValueError: shapes (2,3) and (4,) not aligned: 3 (dim 1) != 4 (dim 0)
----


==== Compliant solution

[source,python,diff-id=1,diff-type=compliant]
----
import numpy as np

a = [[1, 2, 3], [4, 5, 6]]
b = [1, 2, 3]
c = np.dot(a, b) # Compliant
----

== Resources
=== Documentation
* NumPy Documentation - https://numpy.org/doc/stable/reference/generated/numpy.dot.html[numpy.dot]
* PSU - Matrix algebra review - https://online.stat.psu.edu/statprogram/reviews/matrix-algebra/arithmetic#paragraph--2761[Matrix Multiplication]


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)



'''
== Comments And Links
(visible only on this page)

endif::env-github,rspecator-view[]
