== Why is this an issue?

Certain prefixes are widely recognized and expected to correspond to specific namespaces. For instance, most users will expect "xs" to correspond to "http://www.w3.org/2001/XMLSchema". If you subvert that expected correspondence, you run the risk of severely confusing the users of your XSD.


=== Noncompliant code example

[source,xml]
----
<xs:schema targetNamespace="http://www.codeSamples.com/fruit"
           elementFormDefault="qualified" version="1"
           xmlns:xs="http://www.w3.org/2001/XMLSchema"
           xmlns:wsdl="http://CodeSamples.com/animal>       <!-- Noncompliant -->
...
----


=== Compliant solution

[source,xml]
----
<xs:schema targetNamespace="http://www.codeSamples.com/fruit"
           elementFormDefault="qualified" version="1"
           xmlns:xs="http://www.w3.org/2001/XMLSchema"
           xmlns:animal="http://CodeSamples.com/animal>
...
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Use a different prefix for this namespace.


=== Parameters

.reservedPrefixes
****

----
xml,xmlns,xs,xsd,xsi,wsdl
----

Comma-delimited list of namespace prefixes that should not be used
****


=== Highlighting

``++xmlns:xxx++``


'''
== Comments And Links
(visible only on this page)

=== on 18 Jan 2017, 08:44:23 Ann Campbell wrote:
\[~yves.duboispelerin] I almost want to rename this: Standard namespace prefixes should not be hijacked


WDYT?

endif::env-github,rspecator-view[]
