include::../description.adoc[]

== Noncompliant Code Example

----
if(collection.Count >= 0){...}

if(enumerable.Count() < 0){...}

if(array.Length >= 0){...}

bool result = array.Length >=0;
----

== Compliant Solution

----
if (list.Any()) { ... }

if (list.Count > 0) { ... }

if (array.Length >= 42) { ... }
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../highlighting.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
