<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>SQL Server can be tuned at <code>PROCEDURE</code> and <code>TRIGGER</code> levels thanks to several <code>SET</code> statements that change the current session handling of specific information.</p>
</div>
<div class="paragraph">
<p>This rule raises an issue when expected configuration is not set or is set with an unexpected value between the beginning of the <code>PROCEDURE</code> (or <code>TRIGGER</code>) definition and the first statement that is not a <code>SET</code>, <code>IF</code> or <code>DECLARE</code>.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="paragraph">
<p>When this rule is configured with ARITHABORT and ON.</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-sql" data-lang="sql">CREATE PROCEDURE dbo.MyProc
AS
BEGIN
  SET ARITHABORT OFF; -- Noncompliant; ARITHABORT is OFF
  SELECT COUNT(*) FROM MY_TABLE
END;</code></pre>
</div>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-sql" data-lang="sql">ALTER PROCEDURE dbo.MyProc
AS
BEGIN
  SELECT COUNT(*) FROM MY_TABLE
  SET ARITHABORT ON; -- Noncompliant; ARITHABORT is not set at the beginning of the procedure definition
  [...]
END;</code></pre>
</div>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-sql" data-lang="sql">CREATE PROCEDURE dbo.MyProc
AS
BEGIN
  -- Noncompliant; ARITHABORT is not set at all, so default value of SQL Server will be applied
  SELECT COUNT(*) FROM MY_TABLE
END;</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-sql" data-lang="sql">CREATE PROCEDURE dbo.MyProc(@setConfig INT)
AS
BEGIN
  IF @setConfig=1
    BEGIN
      SET ARITHABORT ON;
    END
  SELECT COUNT(*) FROM MY_TABLE
END;</code></pre>
</div>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-sql" data-lang="sql">ALTER PROCEDURE dbo.MyProc
AS
BEGIN
  DECLARE @var INT;
  SET ARITHABORT ON;
  [...]
END;</code></pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p><a href="https://docs.microsoft.com/en-us/sql/t-sql/statements/set-statements-transact-sql?view=sql-server-2017">SET Statements</a> - SQL Server (Transact-SQL) Documentation</p>
</li>
</ul>
</div>
<hr>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>\[Add|Set] configuration "xxx" with expected value "yyy".</p>
</div>
</div>
<div class="sect2">
<h3 id="_parameters">Parameters</h3>
<div class="sidebarblock">
<div class="content">
<div class="title">configuration identifier</div>
<div class="paragraph">
<p>identifier of the parameter to configure</p>
</div>
</div>
</div>
<div class="sidebarblock">
<div class="content">
<div class="title">value</div>
<div class="paragraph">
<p>value expected for the parameter</p>
</div>
</div>
</div>
</div>
</div>
</div>