<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This is an issue when an ActiveRecord model overrides the <code>to_json</code> method for custom JSON serialization.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Rails uses a two-step process for JSON serialization. First, <code>as_json</code> converts the object to a hash representation. Then, <code>to_json</code> converts that hash to a JSON string.</p>
</div>
<div class="paragraph">
<p>When you override <code>to_json</code> in an ActiveRecord model, you break Rails' JSON rendering pipeline. The <code>render json:</code> method in controllers expects to call <code>as_json</code> on your model, but if you&#8217;ve overridden <code>to_json</code>, this can lead to several problems:</p>
</div>
<div class="ulist">
<ul>
<li>
<p><strong>ArgumentError exceptions</strong>: The <code>to_json</code> method signature may not match what Rails expects, causing "wrong number of arguments" errors.</p>
</li>
<li>
<p><strong>Broken controller rendering</strong>: Using <code>render json: @model</code> may not work as expected because Rails calls <code>as_json</code> internally, not your custom <code>to_json</code> method.</p>
</li>
<li>
<p><strong>Inconsistent behavior</strong>: Your custom serialization will only work when <code>to_json</code> is called directly, not through Rails' standard JSON rendering.</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>The <code>as_json</code> method is specifically designed to be overridden. It receives an options hash that you can merge with your custom options, making it flexible and compatible with Rails' rendering system.</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="paragraph">
<p>Overriding <code>to_json</code> instead of <code>as_json</code> can cause runtime errors in your Rails application, particularly ArgumentError exceptions when using <code>render json:</code> in controllers. This can lead to broken API endpoints and inconsistent JSON serialization behavior across your application.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it_in_rails">How to fix it in Rails</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Replace the <code>to_json</code> method override with an <code>as_json</code> method override. Make sure to accept an options parameter and merge it with your custom options.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-ruby" data-lang="ruby">class User &lt; ActiveRecord::Base
  def to_json # Noncompliant
    super(only: :username, methods: [:foo, :bar])
  end
end</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-ruby" data-lang="ruby">class User &lt; ActiveRecord::Base
  def as_json(options = {})
    super({only: [:username], methods: [:foo, :bar]}.merge(options))
  end
end</code></pre>
</div>
</div>
<div class="paragraph">
<p>Even when accepting an options parameter, overriding <code>to_json</code> is still problematic. Use <code>as_json</code> instead to ensure compatibility with Rails' JSON rendering.</p>
</div>
</div>
<div class="sect3">
<h4 id="_noncompliant_code_example_2">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-ruby" data-lang="ruby">class User &lt; ActiveRecord::Base
  def to_json(options = {}) # Noncompliant
    super(only: :username, methods: [:foo, :bar])
  end
end</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution_2">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-ruby" data-lang="ruby">class User &lt; ActiveRecord::Base
  def as_json(options = {})
    super({only: [:username], methods: [:foo, :bar]}.merge(options))
  end
end</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Rails API Documentation - as_json - <a href="https://api.rubyonrails.org/classes/ActiveModel/Serializers/JSON.html#method-i-as_json">Official Rails documentation for the as_json method</a></p>
</li>
<li>
<p>Rails Guides - Active Model Serialization - <a href="https://guides.rubyonrails.org/active_model_basics.html#serialization">Rails guide explaining JSON serialization in Active Model</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>