== Why is this an issue?

Aligning common keywords in a series of statements makes the code easier to read. therefore, it is better to align the ``++TO++`` keywords in a series of successive MOVE statements.


=== Noncompliant code example

[source,cobol]
----
MOVE "Hi There" TO field
MOVE temp TO b
MOVE 123 TO item
----


=== Compliant solution

[source,cobol]
----
MOVE "Hi There" TO field
MOVE temp       TO b
MOVE 123        TO item
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Align the "TO" keywords in this sequence of "MOVE" statements.


endif::env-github,rspecator-view[]
