Aligning common keywords in a series of statements makes the code easier to read. therefore, it is better to align the ``++TO++`` keywords in a series of successive MOVE statements.


== Noncompliant Code Example

----
MOVE "Hi There" TO field
MOVE temp TO b
MOVE 123 TO item
----


== Compliant Solution

----
MOVE "Hi There" TO field
MOVE temp       TO b
MOVE 123        TO item
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

endif::env-github,rspecator-view[]
