== Why is this an issue?

An array index out-of-bounds panic is a bug class that occurs in Rust when a
program tries to access an array element that does not exist.
This bug can cause your program to crash or behave unexpectedly.

include::../impact.adoc[]

== How to fix it

To fix an array index out of bounds panic in Rust, you should always ensure
that you are accessing array elements within the bounds of the array.

=== Code examples

==== Noncompliant code example

[source,rust,diff-id=1,diff-type=noncompliant]
----
let x = [1, 2, 3, 4];

x[9]; // Out of bounds indexing
----

==== Compliant solution

[source,rust,diff-id=1,diff-type=compliant]
----
let x = [1, 2, 3, 4];

x[0];
----

== Resources
=== Documentation

* Clippy Lints - https://rust-lang.github.io/rust-clippy/master/index.html#out_of_bounds_indexing
