<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>If a <code>&lt;td&gt;</code> cell has a <code>headers</code> attribute, it should reference only IDs of headers in the same column and row.</p>
</div>
<div class="paragraph">
<p>Note that it is usually better to use <code>scope</code> attributes of <code>&lt;th&gt;</code> headers instead of <code>headers</code> attribute. <code>headers</code> attribute requires you to list every corresponding <code>&lt;th&gt;</code> header&#8217;s <code>id</code>, which is error-prone and makes the code less maintainable. See https://www.w3.org/WAI/tutorials/tables/tips/[W3C WAI Web Accessibility Tutorials] for more information.</p>
</div>
<div class="paragraph">
<p>If your table is too complex, it might be better to split it into multiple small tables as it improves both readability and maintainability.</p>
</div>
<div class="paragraph">
<p>This rule raises an issue when the <code>headers</code> attribute of a <code>&lt;td&gt;</code> cell contains IDs which don&#8217;t belong to a header in the same row or column.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-html" data-lang="html">&lt;table border="1"&gt;
&lt;caption&gt;
    Rental price
&lt;/caption&gt;
&lt;thead&gt;
    &lt;tr&gt;
        &lt;td&gt;&lt;/td&gt;
        &lt;th id="small" scope="col"&gt;
            Small car
        &lt;/th&gt;
        &lt;th id="big" scope="col"&gt;
            Big Car
        &lt;/th&gt;
    &lt;/tr&gt;
&lt;/thead&gt;
&lt;tbody&gt;
    &lt;tr&gt;
        &lt;th id="paris" class="span" colspan="3" scope="colgroup"&gt;
            Paris
        &lt;/th&gt;
        &lt;/tr&gt;
        &lt;tr&gt;
        &lt;th headers="paris" id="day1"&gt;
            1 day
        &lt;/th&gt;
        &lt;td headers="paris day1 big"&gt; &lt;!-- Noncompliant, referencing the column "big" instead of "small" --&gt;
            11 euros
        &lt;/td&gt;
        &lt;td headers="berlin day1 big"&gt; &lt;!-- Noncompliant, there is no header with id "berlin" --&gt;
            50 euros
        &lt;/td&gt;
        &lt;/tr&gt;
    &lt;/tr&gt;
&lt;/tbody&gt;
&lt;/table&gt;</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_exceptions">Exceptions</h3>
<div class="paragraph">
<p>No issue will be raised if the <code>headers</code> attribute contains the following characters: {}()[]$. This avoids false positives on variables coming from Javascript frameworks such as Angular.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p><a href="https://www.w3.org/WAI/WCAG21/quickref/?versions=2.0#qr-content-structure-separation-programmatic">WCAG2, 1.3.1</a> - Info and Relationships</p>
</li>
<li>
<p><a href="https://www.w3.org/TR/WCAG20-TECHS/html.html#H43">WCAG2, H43</a> - Using id and headers attributes to associate data cells with header cells in data tables</p>
</li>
</ul>
</div>
<hr>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>id X in "headers" reference the header of another column/row.</p>
</div>
<div class="paragraph">
<p>id X in "headers" does not reference any &lt;th&gt; header.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>The "headers" attribute.</p>
</div>
</div>
</div>
</div>