== Why is this an issue?

Aligning opening and ending words of statements is critical to keep the code readable, especially when blocks contain nested statements.


For ``++IF++`` statements, this rule also checks the alignment of the ``++ELSE++`` word.


=== Noncompliant code example

[source,cobol]
----
IF SOME-STATUS = 1
  DISPLAY something
      END-IF.  *> Noncompliant 
----


=== Compliant solution

[source,cobol]
----
IF SOME-STATUS = 1
  DISPLAY something
END-IF.
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

include::parameters.adoc[]

endif::env-github,rspecator-view[]
