Aligning opening and ending words of statements is critical to keep the code readable, especially when blocks contain nested statements.


For ``++IF++`` statements, this rule also checks the alignment of the ``++ELSE++`` word.

== Noncompliant Code Example

----
IF SOME-STATUS = 1
  DISPLAY something
      END-IF.  *> Noncompliant 
----

== Compliant Solution

----
IF SOME-STATUS = 1
  DISPLAY something
END-IF.
----
