<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>There are two ways to build a link that has the sole purpose of running JavaScript code. The goal of this rule is to ban such patterns in order to support browsing with JavaScript disabled.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-html" data-lang="html">&lt;a href="#" onclick="alert('Clicked!'); return false;"&gt;Run JavaScript Code&lt;/a&gt; &lt;!-- Noncompliant --&gt;
&lt;a href="javascript:void(0)" onclick="alert('Clicked!'); return false;"&gt;Run JavaScript Code&lt;/a&gt;  &lt;!-- Noncompliant --&gt;
&lt;a id="inPageAnchor"&gt;Jump down the page to me&lt;/a&gt;  &lt;!-- Compliant --&gt;</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-html" data-lang="html">&lt;a id="inPageAnchor"&gt;Jump down the page to me&lt;/a&gt;  &lt;!-- Compliant --&gt;</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Replace this 'href' value by a link to an existing page or anchor.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_19_jun_2013_091625_dinesh_bolkensteyn_wrote">on 19 Jun 2013, 09:16:25 Dinesh Bolkensteyn wrote:</h3>
<div class="paragraph">
<p>It seems that there is a difference between the 2 patterns.</p>
</div>
<div class="ulist">
<ul>
<li>
<p># will scroll back up to the top of the page.</p>
</li>
<li>
<p>javascript:void(0) does not affect the scrollbar.</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>But indeed with the return false; in the onclick, both do the same.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_19_jun_2013_094812_dinesh_bolkensteyn_wrote">on 19 Jun 2013, 09:48:12 Dinesh Bolkensteyn wrote:</h3>
<div class="paragraph">
<p>I&#8217;m wondering if this rule should also apply to href="" or not.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_8_jul_2013_182537_freddy_mallet_wrote">on 8 Jul 2013, 18:25:37 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>Is implemented by http://jira.codehaus.org/browse/SONARPLUGINS-2984</p>
</div>
</div>
</div>
</div>