include::../description.adoc[]

== Noncompliant Code Example

[source,java]
----
System.out.println(i>10?"yes":"no");
----

== Compliant Solution

[source,java]
----
if (i > 10) {
  System.out.println(("yes");
} else {
  System.out.println("no");
}
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
