include::../description.adoc[]

== Noncompliant Code Example

----
System.out.println(i>10?"yes":"no");
----

== Compliant Solution

----
if (i > 10) {
  System.out.println(("yes");
} else {
  System.out.println("no");
}
----

ifdef::env-github,rspecator-view[]
'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
