== Why is this an issue?

Shared coding conventions allow teams to collaborate effectively. While it is possible to omit the specification letter before a ``++/COPY++`` statement, it is advisable not to do so because including a specification on the line indicates to other developers what type of statements will be added to the program by the copybook. Further, using a specification letter on the line makes it clearer to those who might be skimming the file that the line is not just a comment, but does contain code.


=== Noncompliant code example

[source,rpg]
----
      /COPY MBR1
----


=== Compliant solution

[source,rpg]
----
     I/COPY MBR1
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Add the missing specification letter to this statement.


endif::env-github,rspecator-view[]
