<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when <code>context.Background()</code> is called in a scope where a context parameter or variable is already available.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>In Go, contexts are designed to carry cancellation signals, deadlines, and request-scoped values across API boundaries. When you call <code>context.Background()</code>, you create a new, empty context that has no parent and will never be cancelled.</p>
</div>
<div class="paragraph">
<p>If your function already receives a context parameter, that context likely carries important information from upstream callers. For example, it might have a timeout set by an HTTP server, or a cancellation signal from a user who closed their browser.</p>
</div>
<div class="paragraph">
<p>When you ignore the existing context and create a new background context instead, you break this chain of communication. Your operation will continue running even if the original request was cancelled, potentially wasting resources and making your application less responsive.</p>
</div>
<div class="paragraph">
<p>This pattern is particularly problematic in web applications, where requests should be cancelled when clients disconnect, and in concurrent programs where operations need to respect cancellation signals.</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="paragraph">
<p>Breaking context propagation can lead to several issues:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>Operations continue running after they should have been cancelled, wasting CPU and memory</p>
</li>
<li>
<p>Timeouts and deadlines are not respected, potentially causing the application to hang</p>
</li>
<li>
<p>Request-scoped values (like trace IDs or user information) are lost</p>
</li>
<li>
<p>The application becomes less responsive and harder to shut down gracefully</p>
</li>
</ul>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Use the existing context parameter instead of creating a new background context. This ensures proper propagation of cancellation signals, deadlines, and values.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-go" data-lang="go">func processData(ctx context.Context, client *Client) error {
    // Using context.Background() ignores the existing context
    result, err := client.Query(context.Background(), "SELECT * FROM table") // Noncompliant
    return err
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-go" data-lang="go">func processData(ctx context.Context, client *Client) error {
    // Reusing the existing ctx parameter
    result, err := client.Query(ctx, "SELECT * FROM table")
    return err
}</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Go Context Package - <a href="https://pkg.go.dev/context">Official documentation for the context package</a></p>
</li>
<li>
<p>Go Blog: Context - <a href="https://go.dev/blog/context">Comprehensive guide to using contexts in Go</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>