Full outer joins aren't in common use, and as a result many developers don't really understand them. Therefore, each use of this language feature should be reviewed.


== Noncompliant Code Example

----
BEGIN
  SELECT *
  BULK COLLECT INTO result
  FROM DUAL d1
  FULL OUTER JOIN DUAL d2 ON d1.dummy != d2.dummy; -- Noncompliant
END;
/
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

endif::env-github,rspecator-view[]
