<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This is an issue when the same Go package is imported multiple times (both directly and with an alias) or when code uses the full package name instead of a defined alias.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Importing the same package multiple times creates redundancy and confusion about which import to use. When a package is imported both with and without an alias, it&#8217;s unclear which approach developers should follow throughout the codebase.</p>
</div>
<div class="paragraph">
<p>Inconsistent usage of import aliases further compounds this problem. If an alias is defined for a package, using the full package name in some places and the alias in others makes the code harder to read and maintain.</p>
</div>
<div class="paragraph">
<p>This inconsistency can lead to:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>Confusion about which import style to use</p>
</li>
<li>
<p>Potential naming conflicts if the same package is referenced differently</p>
</li>
<li>
<p>Reduced code readability and maintainability</p>
</li>
<li>
<p>Unnecessary cognitive load when reading the code</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>Go&#8217;s import system is designed to be explicit and clear. Following consistent import patterns helps maintain this clarity.</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="paragraph">
<p>This issue primarily affects code maintainability and readability. While not a security concern, inconsistent imports can lead to developer confusion, increased cognitive load when reading code, and potential maintenance issues when refactoring or updating dependencies.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Remove redundant imports and keep only the aliased version when a package is imported both directly and with an alias.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-go" data-lang="go">import (
    "cloud.google.com/go/pubsub/v2"
    "cloud.google.com/go/pubsub/v2/apiv1/pubsubpb" // Noncompliant
    pb "cloud.google.com/go/pubsub/v2/apiv1/pubsubpb"
)</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-go" data-lang="go">import (
    "cloud.google.com/go/pubsub/v2"
    pb "cloud.google.com/go/pubsub/v2/apiv1/pubsubpb"
)</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Go Language Specification - Import declarations - <a href="https://golang.org/ref/spec#Import_declarations">Official Go specification for import declarations and package aliases</a></p>
</li>
<li>
<p>Effective Go - Package names - <a href="https://golang.org/doc/effective_go#package-names">Best practices for Go package naming and imports</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>