== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,dart]
----
void save() {
  try {
    saveDocument();
  } catch (exception) {
  }
}
----

=== Compliant solution

[source,dart]
----
void save() {
  try {
    saveDocument();
  } catch (exception) {
    log(exception);
  }
}
----

[source,dart]
----
void save() {
  try {
    saveDocument();
  } catch (_) { // Compliant, ignored intentionally
  }
}
----

[source,dart]
----
void save() {
  try {
    saveDocument();
  } catch (exception) { // Compliant, left a comment
    // ignored intentionally
  }
}
----

include::../see.adoc[]

* https://dart.dev/tools/linter-rules/empty_catches[Dart Lint rule]
