<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>An <code>equals</code> method that unconditionally returns the same answer is an error likely to cause many bugs.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">public class Fruit extends Food {
  private Season ripe;

  public boolean equals(Object obj) {
    return ripe.equals(this);  // Noncompliant
  }</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_23_mar_2018_084224_alexandre_gigleux_wrote">on 23 Mar 2018, 08:42:24 Alexandre Gigleux wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] Do you remember why this one has been closed "Won&#8217;t Fix". It covers 2 FindBugs rules (EQ_ALWAYS_FALSE, EQ_ALWAYS_TRUE), that were not rejected according to http://dist.sonarsource.com/reports/coverage/findbugs.html</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_23_mar_2018_130050_ann_campbell_wrote">on 23 Mar 2018, 13:00:50 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>I think it was in favor of the more sophisticated rules. I&#8217;m not finding a paper trail tho.</p>
</div>
</div>
</div>
</div>