== Why is this an issue?

Member functions can only be used with an instance of a class. But ``++friend++`` functions can be used with an implicitly converted type. So loosening access privileges to ``++friend++`` on overloaded binary operators makes them more flexible. Specifically, with a ``++friend++`` function, the class instance can be on either the right or the left of the operator, but with a member function, it can only be on the left.


This rule raises an issue for all non-``++friend++`` overloaded binary operators except:

* \"``++=++``", \"``++[ ]++``", and \"``++->++``", which cannot be overloaded as ``++friend++`` functions.
* \"``+++=++``", \"``++-=++``", \"``++*=++``", \"``++/=++``", \"``++%=++``", \"``++^=++``", \"``++&=++``", \"``++|=++``", \"``++<<=++``", and \"``++>>=++``", which are not symmetric operators.


=== Noncompliant code example

[source,cpp]
----
bool operator==(const MyClass &RHS);  // Noncompliant
----


=== Compliant solution

[source,cpp]
----
friend bool operator==(const MyClass &LHS, const MyClass &RHS);
----


== Resources

* https://github.com/isocpp/CppCoreGuidelines/blob/036324/CppCoreGuidelines.md#c161-use-nonmember-functions-for-symmetric-operators[{cpp} Core Guidelines C.161] - Use nonmember functions for symmetric operators


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Make this a free function, potentially "friend".


'''
== Comments And Links
(visible only on this page)

=== on 17 Nov 2015, 09:01:34 Evgeny Mandrikov wrote:
\[~ann.campbell.2] could you please help me update this RSPEC - operators ``++=++``, ``++[]++`` and ``++->++`` can't be overloaded as friend functions.

=== on 17 Nov 2015, 14:01:38 Ann Campbell wrote:
See what you think [~evgeny.mandrikov]

=== on 17 Nov 2015, 20:38:54 Evgeny Mandrikov wrote:
\[~ann.campbell.2] LGTM

endif::env-github,rspecator-view[]
