include::../description.adoc[]

== Noncompliant Code Example

[source,vbnet]
----
Structure MyStruct ' Noncompliant

    Public Property Value As Integer

End Structure
----

== Compliant Solution

[source,vbnet]
----
Structure MyStruct
    Implements IEquatable(Of MyStruct)

    Public Property Value As Integer

    Public Overloads Function Equals(other As MyStruct) As Boolean Implements IEquatable(Of MyStruct).Equals
        ' ...
    End Function

End Structure
----

include::../rspecator.adoc[]
