include::../description.adoc[]

== Noncompliant Code Example

----
public double Divide(int divisor, int dividend) 
{
    return divisor/dividend;
}

public void DoTheThing() 
{
    int divisor = 15;
    int dividend = 5;

    double result = Divide(dividend, divisor);  // Noncompliant; operation succeeds, but result is unexpected
    //...
}
----

== Compliant Solution

----
public double Divide(int divisor, int dividend) 
{
    return divisor/dividend;
}

public void DoTheThing() 
{
    int divisor = 15;
    int dividend = 5;

    double result = Divide(divisor, dividend);
    //...
}
----
