== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,csharp]
----
public double Divide(int divisor, int dividend) 
{
    return divisor/dividend;
}

public void DoTheThing() 
{
    int divisor = 15;
    int dividend = 5;

    double result = Divide(dividend, divisor);  // Noncompliant; operation succeeds, but result is unexpected
    //...
}
----

=== Compliant solution

[source,csharp]
----
public double Divide(int divisor, int dividend) 
{
    return divisor/dividend;
}

public void DoTheThing() 
{
    int divisor = 15;
    int dividend = 5;

    double result = Divide(divisor, dividend);
    //...
}
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../highlighting.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
