The first argument to ``++super++`` must be the name of the class making the call. If it's not, the result will be a runtime error.


== Noncompliant Code Example

[source,python]
----
class Person(object):
  #...

class PoliceOfficer(Person):
  def __init__(self, name):
    super().__init__(name)  // Noncompliant
----


== Compliant Solution

[source,python]
----
class Person(object):
  #...

class PoliceOfficer(Person):
  def __init__(self, name):
    super(PoliceOfficer).__init__(name)
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

endif::env-github,rspecator-view[]
