== Why is this an issue?

The first argument to ``++super++`` must be the name of the class making the call. If it's not, the result will be a runtime error.


=== Noncompliant code example

[source,python]
----
class Person(object):
  #...

class PoliceOfficer(Person):
  def __init__(self, name):
    super().__init__(name)  // Noncompliant
----


=== Compliant solution

[source,python]
----
class Person(object):
  #...

class PoliceOfficer(Person):
  def __init__(self, name):
    super(PoliceOfficer).__init__(name)
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

"X" should be the first argument to this method.


endif::env-github,rspecator-view[]
