== Why is this an issue?

include::../common/why.adoc[]

=== What is the potential impact?

include::../common/potential_impact.adoc[]

== How to fix it

include::../common/how.adoc[]

=== Code examples

==== Noncompliant code example

[source,kotlin,diff-id=1,diff-type=noncompliant]
----
if (org.gradle.internal.os.OperatingSystem.current().isWindows) { // Noncompliant, OperatingSystem is part of Gradle internal API
    // ...
}
----

==== Compliant solution

[source,kotlin,diff-id=1,diff-type=compliant]
----
if (System.getProperty("os.name").toLowerCase().contains("windows")) { // Compliant
    // ...
}
----

== Resources

=== Documentation

* https://docs.gradle.org/current/userguide/authoring_maintainable_build_scripts.html#sec:avoiding_gradle_internal_apis[Gradle Documentation - Avoid using internal Gradle APIs]
* https://github.com/liutikas/gradle-best-practices#dont-use-internal-apis[Best Practices when using Gradle - Don't use internal APIs (Liutikas)]
