<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>The <code>&lt;?php</code> tag is used to explicitly mark the start of PHP code in a file.
It is considered the recommended and portable way to open PHP blocks.
On the other hand, the <code>&lt;?=</code> tag is a shorthand for <code>&lt;?php</code> echo and is specifically used to output variables or expressions directly without using the echo statement.
Not using these tags can make the code less readable and harder to maintain, as it deviates from the standard conventions followed by most PHP developers.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-php" data-lang="php">&lt;?
$foo = 1;
?&gt;</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-php" data-lang="php">&lt;?php
$foo = 1;
?&gt;</code></pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p><a href="https://www.php.net/manual/en/language.basic-syntax.phptags.php">PHP Manual - PHP tags</a></p>
</li>
</ul>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Change this opening tag to either "&lt;?php" or "&lt;?=".</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="ulist">
<ul>
<li>
<p>primary: first NC tag</p>
</li>
<li>
<p>secondary: subsequent NC tags</p>
<div class="ulist">
<ul>
<li>
<p>message: none</p>
</li>
</ul>
</div>
</li>
</ul>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_12_oct_2015_120358_ann_campbell_wrote">on 12 Oct 2015, 12:03:58 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>Highlighting added [~linda.martin]. See what you think.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_12_nov_2015_182501_linda_martin_wrote">on 12 Nov 2015, 18:25:01 Linda Martin wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] Perfect! Thanks.</p>
</div>
</div>
</div>
</div>