<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule reports when there is an Architecture Directive to remove a specific usage of a file or folder from a given location.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This issue is being raised because there is a dependency in the code, and an Architecture Directive has been created to remove it.</p>
</div>
<div class="paragraph">
<p>Architecture Directives are defined in the “Architecture” section of the project by an authorized user.</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="paragraph">
<p>Flaws in the architecture constitute unnecessary complexity which affects developer productivity. Removing this flaw makes future development easier by favoring local changes in the code.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Remove or refactor the problematic dependency. This may involve replacing the dependency with an allowed one, moving code to appropriate files or folders, or introducing new abstractions. The exact solution depends on your project’s specific situation.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="paragraph">
<p>The following code contains a circular dependency: <code>order.js</code> &rarr; <code>customer.js</code> &rarr; <code>order.js</code>, and <code>order.js</code> &rarr; <code>product.js</code> &rarr; <code>order.js</code>.
Both cycles are connected through <code>order.js</code>, forming a tangle of two cycles.</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-javascript" data-lang="javascript">// order.js
import { Customer } from './customer.js';
import { Product } from './product.js';

export class Order {
    constructor() {
        this.customer = new Customer();
        this.products = [];
    }
}

// customer.js
import { Order } from './order.js';

export class Customer {
    constructor() {
        this.orders = [];
    }
}

// product.js
import { Order } from './order.js';

export class Product {
    constructor() {
        this.orders = [];
    }
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="paragraph">
<p>The issue can be resolved by refactoring the structure.
Two service functions can replace the dependencies <code>customer.js</code> &rarr; <code>order.js</code> and <code>product.js</code> &rarr; <code>order.js</code>.</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-javascript" data-lang="javascript">// order.js
import { Customer } from './customer.js';
import { Product } from './product.js';

export class Order {
    constructor() {
        this.customer = new Customer();
        this.products = [];
    }
}

// customer.js
export class Customer { }

// product.js
export class Product { }

// orderService.js
export function getOrdersByCustomer(customer) {
    // Implementation to get orders by customer
}

export function getOrdersByProduct(product) {
    // Implementation to get orders by product
}</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>