<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule reports when there is an Architecture Directive to remove a specific usage of a class or package from a given location.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This issue is being raised because there is a dependency in the code, and an Architecture Directive has been created to remove it.</p>
</div>
<div class="paragraph">
<p>Architecture Directives are defined in the “Architecture” section of the project by an authorized user.</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="paragraph">
<p>Flaws in the architecture constitute unnecessary complexity which affects developer productivity. Removing this flaw makes future development easier by favoring local changes in the code.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Remove or refactor the problematic dependency. This may involve replacing the dependency with an allowed one, moving code to appropriate classes or packages, or introducing new abstractions. The exact solution depends on your project’s specific situation.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="paragraph">
<p>The following code contains two cycles: Order &rarr; Customer &rarr; Order, and Order &rarr; Product &rarr; Order. Furthermore, both cycles are connected to each other via class Order. This is called <em>a tangle consisting of two cycles</em>.</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">class Order {
    public Customer customer;
    public List&lt;Product&gt; products;
}
class Customer {
    public List&lt;Order&gt; orders;
}
class Product {
    public List&lt;Order&gt; orders;
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="paragraph">
<p>The issue can be resolved by changing the structure. Two service functions can replace the dependencies Customer &rarr; Order and Product &rarr; Order.</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">class Order {
    public Customer customer;
    public List&lt;Product&gt; products;
}
class Customer { }
class Product { }
interface OrderService {
  List&lt;Order&gt; getOrdersByCustomer(Customer customer);
  List&lt;Order&gt; getOrdersByProduct(Product product);
}</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>