<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule reports when there is an Architecture Directive to remove a specific usage of a file or folder from a given location.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This issue is being raised because there is a dependency in the code, and an Architecture Directive has been created to remove it.</p>
</div>
<div class="paragraph">
<p>Architecture Directives are defined in the “Architecture” section of the project by an authorized user.</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="paragraph">
<p>Flaws in the architecture constitute unnecessary complexity which affects developer productivity. Removing this flaw makes future development easier by favoring local changes in the code.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Remove or refactor the problematic dependency. This may involve replacing the dependency with an allowed one, moving code to appropriate files or folders, or introducing new abstractions. The exact solution depends on your project’s specific situation.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="paragraph">
<p>This example shows a circular dependency between modules <code>order.py</code> and <code>customer.py</code>:</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python"># order.py
from customer import Customer # Noncompliant
class Order:
    def __init__(self, customer: Customer):
        self.customer = customer

# customer.py
from order import Order # Noncompliant
class Customer:
    def __init__(self, order: Order):
        self.order = order</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="paragraph">
<p>The issue can be resolved by refactoring the code structure.
Instead of <code>Customer</code> and <code>Order</code> referencing each other, customers by order and orders by customer can be aggregated using an addition class <code>CustomerOrder</code>.</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python"># order.py
class Order:
    def __init__(self):
        ...

# customer.py
class Customer:
    def __init__(self):
        ...

# customer_order.py
from customer import Customer
from order import Order
class CustomerOrder:
    def __init__(self, customer: Customer, order: Order):
        self.customer = customer
        self.order = order</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>