== Why is this an issue?

C-style underflow/overflow checks will panic in debug builds. Using safe methods like `checked_add` or `overflowing_add` ensures that overflow conditions are explicitly handled, preventing unexpected panics.

=== Code examples

==== Noncompliant code example

[source,rust,diff-id=1,diff-type=noncompliant]
----
let a = 1i32;
let b = 2i32;
if a + b < a {
// Noncompliant: This will panic in debug builds
// handle overflow
}
----

==== Compliant solution

[source,rust,diff-id=1,diff-type=compliant]
----
let a = 1i32;
let b = 2i32;
if a.checked_add(b).is_none() {
// Compliant: Explicitly handle overflow
// handle overflow
}
----

== Resources
=== Documentation

* Clippy Lints - https://rust-lang.github.io/rust-clippy/master/index.html#panicking_overflow_checks
