== Why is this an issue?

According to the SAP documentation:

____
System functions are only intended for internal usage. Incompatible changes and further development is possible at any time and without warning or notice.

____

So calling system C functions using a ``++CALL++`` statement should be avoided.


=== Noncompliant code example

[source,abap]
----
CALL 'MULTIPLY' ID 'P1'  FIELD '9999' 
                ID 'P2'  FIELD '9999' 
                ID 'RES' FIELD RESULT. 
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Remove this use of system function 'XXXX'.


'''
== Comments And Links
(visible only on this page)

=== on 8 Jan 2014, 15:55:08 Freddy Mallet wrote:
See ABAP documentation : \http://help.sap.com/abapdocu_702/en/abapcall-.htm

endif::env-github,rspecator-view[]
