According to the SAP documentation:

____
System functions are only intended for internal usage. Incompatible changes and further development is possible at any time and without warning or notice.

____

So calling system C functions using a ``++CALL++`` statement should be avoided.


== Noncompliant Code Example

----
CALL 'MULTIPLY' ID 'P1'  FIELD '9999' 
                ID 'P2'  FIELD '9999' 
                ID 'RES' FIELD RESULT. 
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
