<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>When a test fails due, for example, to infrastructure issues, you might want to ignore it temporarily. But without some kind of notation about why the test is being ignored, it may never be reactivated. Such tests are difficult to address without comprehensive knowledge of the project, and end up polluting their projects.</p>
</div>
<div class="paragraph">
<p>This rule raises an issue for each skipped test with "<code>unittest.skip</code>" or "<code>pytest.mark.skip</code>" without providing a reason argument.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">import unittest
class MyTest(unittest.TestCase):
    @unittest.skip  # Noncompliant
    def test_something(self): ...</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">import unittest
class MyTest(unittest.TestCase):
    @unittest.skip("need to fix something")
    def test_something(self): ...</code></pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="paragraph">
<p><a href="https://docs.python.org/3/library/unittest.html#skipping-tests-and-expected-failures">Unittest documentation - skipping tests and expected failures</a></p>
</div>
<div class="paragraph">
<p><a href="https://docs.pytest.org/en/latest/how-to/skipping.html#skipping-test-functions">Pytest documentation - skipping test functions</a></p>
</div>
<hr>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Provide a reason for skipping this test.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>Primary:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>the skip annotation or method call (if no argument is provided)</p>
</li>
<li>
<p>the empty string literal (if an empty string is provided as argument)</p>
</li>
</ul>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_deprecates_s3424">deprecates: <a data-rspec-id="S3424" class="rspec-auto-link">S3424</a></h3>

</div>
</div>
</div>