== Why is this an issue?

Using the ``++LIKE++`` operator in SQL ``++WHERE++`` conditions can highly impact the performance of the request. The use of this operator should be strongly indicated.


=== Noncompliant code example

[source,text]
----
SELECT * 
       FROM doktl 
       INTO TABLE text_tab 
       WHERE doktext LIKE srch_str.
----

