== Why is this an issue?

Use of ``++<meta http-equiv="refresh">++`` is discouraged by the World Wide Web Consortium (W3C).

If a user clicks the 'Back' button, some browers will go back to the redirecting page, which will prevent the user from actually going back.


To refresh the page, a better alternative is to use Ajax, to refresh only what needs to be refreshed and not the whole page.

To redirect to another page, using the HTTP response status code 301 'Moved Permanently' and 302 'Found' is a better option.


=== Noncompliant code example

[source,html]
----
<head>
  <meta http-equiv="refresh" content="5">   <!-- Non-Compliant -->
  <meta name="description" content="..."> 
</head>
----


=== Compliant solution

[source,html]
----
<head>
  <meta name="description" content="..."> 
</head>
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Remove this meta refresh tag.


'''
== Comments And Links
(visible only on this page)

=== on 8 Jul 2013, 18:21:05 Freddy Mallet wrote:
Is implemented by \http://jira.codehaus.org/browse/SONARPLUGINS-2994

endif::env-github,rspecator-view[]
