== Why is this an issue?

According to the specification:

____
Nonfinal ``++static++`` class fields are disallowed in EJBs because such fields make an enterprise bean difficult or impossible to distribute.
____


Therefore, all ``++static++`` fields in an EJB should also be ``++final++``.


=== Noncompliant code example

[source,java]
----
@Stateless
public class MyEjb
{
  private static String message;  // Noncompliant
}
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

endif::env-github,rspecator-view[]
