<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>According to the specification:</p>
</div>
<div class="quoteblock">
<blockquote>
<div class="paragraph">
<p>Nonfinal <code>static</code> class fields are disallowed in EJBs because such fields make an enterprise bean difficult or impossible to distribute.</p>
</div>
</blockquote>
</div>
<div class="paragraph">
<p>Therefore, all <code>static</code> fields in an EJB should also be <code>final</code>.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">@Stateless
public class MyEjb
{
  private static String message;  // Noncompliant
}</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Make "xxx" final.</p>
</div>
</div>
</div>
</div>