<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when the AWS Lambda handler function is declared with <code>async</code>.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>The standard AWS Lambda Python runtime is designed to invoke a synchronous handler function.
While Python&#8217;s <code>asyncio</code> library can be used inside a synchronous handler, by calling <code>asyncio.run()</code>,
the handler function itself cannot be declared with <code>async def</code>. Doing so will lead to a runtime error.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>To fix this issue, remove the <code>async</code> keyword from the handler function definition.
The asynchronous part of the code can be moved to its own <code>async</code> function.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">def some_logic():
  ...

async def lambda_handler(event, context): # Noncompliant: the handler is defined with async
  result = some_logic()
  return {"status": result}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">import asyncio

async def some_logic():
  ...

def lambda_handler(event, context):
  result = asyncio.run(some_logic())
  return {"status": result}</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>AWS Documentation - <a href="https://docs.aws.amazon.com/lambda/latest/dg/python-handler.html#python-handler-signature">Valid handler signatures for Python handlers</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>visible only on this page</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove the <code>async</code> keyword from this AWS Lambda handler definition.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>The <code>async</code> keyword</p>
</div>
</div>
</div>
</div>