<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Functions that use <code>yield</code> are known as "generators". Before Python 3.3, generators cannot <code>return</code> values. Similarly, functions that use <code>return</code> cannot use <code>yield</code>. Doing so will cause a <code>SyntaxError</code>.</p>
</div>
<div class="paragraph">
<p>Either upgrade your version of Python to a version &gt;= 3.3, or don&#8217;t use both return and yield in a function.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">def adder(n):
     num = 0
     while num &lt; n:
         yield num
         num += 1
     return num  #Noncompliant</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Use only "return" or only "yield", not both.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_17_mar_2015_084125_elena_vilchik_wrote">on 17 Mar 2015, 08:41:25 Elena Vilchik wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2]</p>
</div>
<div class="olist arabic">
<ol class="arabic">
<li>
<p>We should focus in rule description that "return" contains argument. "return" without argument can be used in generator.</p>
</li>
<li>
<p>About labels: may be we should create tag for syntax error, and to use it in all such python rules. For example "syntax". What do you think?</p>
</li>
</ol>
</div>
</div>
<div class="sect2">
<h3 id="_on_17_mar_2015_174841_ann_campbell_wrote">on 17 Mar 2015, 17:48:41 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~elena.vilchik] I&#8217;ve updated the description. See if it&#8217;s enough.</p>
</div>
<div class="paragraph">
<p>I&#8217;m not wild about the proposed tag.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_27_nov_2019_145209_nicolas_harraudeau_wrote">on 27 Nov 2019, 14:52:09 Nicolas Harraudeau wrote:</h3>
<div class="paragraph">
<p>Deprecating this rule because it is possible to use <code>return</code> in a generator since Python 3.3, which was released on September 29, 2012.</p>
</div>
<div class="paragraph">
<p>The following code:</p>
</div>
<div class="listingblock">
<div class="content">
<pre>return 42</pre>
</div>
</div>
<div class="paragraph">
<p>is equivalent to</p>
</div>
<div class="listingblock">
<div class="content">
<pre>raise StopIteration(42)</pre>
</div>
</div>
<div class="paragraph">
<p>See <a href="https://www.python.org/dev/peps/pep-0380/#formal-semantics">PEP 380</a> for more information.</p>
</div>
<div class="paragraph">
<p>Note also that <a href="https://www.python.org/doc/sunset-python-2/">official python 2 support will be stopped on January 1, 2020</a>, thus there is no point in having this rule enabled by default.</p>
</div>
</div>
</div>
</div>