<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>The <code>const</code> keyword should be used every time a constructor declared as <code>const</code> is invoked in a non-constant context and with constant arguments.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Dart supports a particular type of constructor named <a href="https://dart.dev/language/constructors#constant-constructors">constant constructors</a>, denoted by the word <code>const</code> in front of the constructor name.</p>
</div>
<div class="paragraph">
<p><code>const</code> constructors are used to create objects that are immutable, i.e. their instance variables cannot be changed after the object is created. In order to achieve that, all instance variables of the class must be <code>final</code>, and initialized either inline, or in the <code>const</code> constructor:</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-dart" data-lang="dart">class PointOnYAxis {
  final int x = 0;
  final int y;
  const PointOnYAxis(this.y);
}</code></pre>
</div>
</div>
<div class="paragraph">
<p>When a class defines a <code>const</code> constructor, the Dart compiler has the ability to perform object creation at compile time, creating compile-time constant of non-primitive types, with a process called <strong>canonicalization</strong>.</p>
</div>
<div class="paragraph">
<p>This process later allows the compiler to recognize that two constant objects of the same type are equivalent, and to reuse the same object in memory, resulting in a more efficient memory usage. This optimization, that happens in other languages on immutable primitive types such as <code>int</code>, is extended by Dart to all types that define a <code>const</code> constructor.</p>
</div>
<div class="paragraph">
<p>For that to happen, however, it&#8217;s not enough to just define a <code>const</code> constructor: the intent of creating a compile-time constant needs to be made explicit by using the <code>const</code> keyword:</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-dart" data-lang="dart">var x = const PointOnYAxis(42); // The const keyword makes x a compile-time constant</code></pre>
</div>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="paragraph">
<p>Without the <code>const</code> keyword, the object creation is deferred to runtime, and the canonicalization optimization is not performed, resulting in a new object being created every time the constructor is called.</p>
</div>
<div class="paragraph">
<p>That results in more memory being used and potentially slower execution, especially in scenarios where the object is created multiple times, as in a loop.</p>
</div>
</div>
<div class="sect2">
<h3 id="_exceptions">Exceptions</h3>
<div class="paragraph">
<p>There are scenarios where the <code>const</code> keyword is implicitly assumed, and doesn&#8217;t have to be made explicit. Namely, when the object instantiation happens in a <strong>const context</strong>, there is no need to use the <code>const</code> keyword (see <a data-rspec-id="S3689" class="rspec-auto-link">S3689</a> for further details). That is because the Dart compiler is already building a compile-time constant for a larger expression including the object, and a compile-time constant can only be built of other compile-time constants.</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-dart" data-lang="dart">const x1 = PointOnYAxis(42); // The const constraint is auto-inferred here
const x2 = [
  PointOnYAxis(42), // The const constraint is auto-inferred here
  PointOnYAxis(43), // The const constraint is auto-inferred here
];
var x3 = const {
  'a': PointOnYAxis(42), // The const constraint is auto-inferred here
  'b': PointOnYAxis(43), // The const constraint is auto-inferred here
};</code></pre>
</div>
</div>
<div class="paragraph">
<p>Moreover, a constant constructor can be used as such only when all expressions used in the constructor are also constant expressions. If any of the expressions is not constant, the rule does not apply. For example:</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-dart" data-lang="dart">void aFunction(int i1) {
    var x = PointOnYAxis(i1); // x cannot be const because i1 is not a const expression
}</code></pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Use the <code>const</code> keyword in front of the constructor invocation, or make the context constant.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-dart" data-lang="dart">var x = PointOnYAxis(42); // Non compliant</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-dart" data-lang="dart">var x = const PointOnYAxis(42); // Explicit const constructor invocation</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_noncompliant_code_example_2">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-dart" data-lang="dart">var x = PointOnYAxis(42); // Non compliant</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution_2">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-dart" data-lang="dart">const x = PointOnYAxis(42); // Const context</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Dart Docs - <a href="https://dart.dev/tools/linter-rules/prefer_const_constructors">Dart Linter rule - prefer_const_constructors</a></p>
</li>
<li>
<p>Dart Docs - <a href="https://dart.dev/language/constructors#constant-constructors">Language - Constant constructors</a></p>
</li>
</ul>
</div>
</div>
<div class="sect2">
<h3 id="_related_rules">Related rules</h3>
<div class="ulist">
<ul>
<li>
<p><a data-rspec-id="S3689" class="rspec-auto-link">S3689</a> - "const" modifier should not be redundant</p>
</li>
<li>
<p><a data-rspec-id="S7113" class="rspec-auto-link">S7113</a> - @immutable classes should only have const constructors</p>
</li>
</ul>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Use 'const' with the constructor to improve performance.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>The entire constructor invocation expression: e.g. <code>PointOnYAxis(42)</code> in <code>var x = PointOnYAxis(42);</code>.</p>
</div>
</div>
</div>
</div>