include::../description.adoc[]

== Noncompliant Code Example

With default provided regular expression: ``++^([A-Z0-9_]*|[a-z0-9_]*)$++``

----
FUNCTION MyFunction. 
...
ENDFUNCTION.
----

== Compliant Solution

----
FUNCTION MY_FUNCTION. 
...
ENDFUNCTION.
----

ifdef::env-github,rspecator-view[]
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
