== How to fix it in Argon2

=== Code examples

==== Noncompliant code example

[source,go,diff-id=1,diff-type=noncompliant]
----
import (
	"golang.org/x/crypto/argon2"
)
func passwordHashingArgon2(password string, salt []byte, w http.ResponseWriter) {
	argon2IDHashWeak := argon2.IDKey([]byte(password), salt, 1, 32*1024, 1, 32) // Noncompliant: Use strong parameters
}
----

==== Compliant solution

[source,go,diff-id=1,diff-type=compliant]
----
import (
	"golang.org/x/crypto/argon2"
)
func passwordHashingArgon2(password string, salt []byte, w http.ResponseWriter) {
	argon2IDHashWeak := argon2.IDKey([]byte(password), salt, 1, 64*1024, 4, 32)
}
----

=== How does this work?

include::../../common/fix/argon-parameters.adoc[]

=== Going the extra mile

include::../../common/extra-mile/argon-cli.adoc[]

include::../../common/extra-mile/peppering.adoc[]

