== How to fix it in Bcrypt

=== Code examples

==== Noncompliant code example

[source,go,diff-id=102,diff-type=noncompliant]
----
import (
	"golang.org/x/crypto/bcrypt"
)
func passwordHashingBcrypt(password string) {
	bcrypt.GenerateFromPassword([]byte(password), 4) // Noncompliant
}
----

==== Compliant solution

[source,go,diff-id=102,diff-type=compliant]
----
import (
	"golang.org/x/crypto/bcrypt"
)
func passwordHashingBcrypt(password string) {
	bcrypt.GenerateFromPassword([]byte(password), 12)
}
----

=== How does this work?

include::../../common/fix/bcrypt-parameters.adoc[]

=== Pitfalls

include::../../common/pitfalls/pre-hashing.adoc[]

=== Going the extra mile

include::../../common/extra-mile/peppering.adoc[]

