== How to fix it in Pbkdf2

=== Code examples

==== Noncompliant code example

[source,go,diff-id=1,diff-type=noncompliant]
----
import (
    . "crypto/sha1"
    "golang.org/x/crypto/pbkdf2"
)
func passwordHashingPbkdf2(password string, salt []byte) {
    pbkdf2.Key([]byte(password), salt, 42, 32, New) // Noncompliant
}
----

==== Compliant solution

[source,go,diff-id=1,diff-type=compliant]
----
import (
    . "crypto/sha1"
    "golang.org/x/crypto/pbkdf2"
)
func passwordHashingPbkdf2(password string, salt []byte) {
    pbkdf2.Key([]byte(password), salt, 42*42, 32, New)
}
----

=== How does this work?

include::../../common/fix/pbkdf2-parameters.adoc[]

=== Going the extra mile

include::../../common/extra-mile/peppering.adoc[]

