== How to fix it in Scrypt

=== Code examples

==== Noncompliant code example

[source,go,diff-id=1,diff-type=noncompliant]
----
import (
    "golang.org/x/crypto/scrypt"
)
func passwordHashingScrypt(password string, salt []byte) {
    scrypt.Key([]byte(password), salt, 42, 8, 1, 32) // Noncompliant
}
----

==== Compliant solution

[source,go,diff-id=1,diff-type=compliant]
----
import (
    "golang.org/x/crypto/scrypt"
)
func passwordHashingScrypt(password string, salt []byte) {
    scrypt.Key([]byte(password), salt, 42, 1, 1, 32)
}
----

=== How does this work?

include::../../common/fix/scrypt-parameters.adoc[]

=== Going the extra mile

include::../../common/extra-mile/peppering.adoc[]

