== Why is this an issue?

``++break++`` and ``++continue++`` are unstructured control flow statements which make code harder to read. Additionally, more recent versions of Python raise a SyntaxError when modules containing ``++break++`` or ``++continue++`` outside of a loop are imported.


Therefore, these statements should not be used outside of loops. 


=== Noncompliant code example

[source,python]
----
narg=len(sys.argv)
if narg == 1:
        print('@Usage: input_filename nelements nintervals')
        break
----


=== Compliant solution

[source,python]
----
if narg == 1:
        print('@Usage: input_filename nelements nintervals')
        sys.exit()
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
