<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>For the sake of backward compatibility, Visual Basic .NET continues to offer a set of functions that convert from Object to different primitive types: <code>CChar, CStr, CBool, CDate, CSng, CDbl, CDec, CByte, CSByte, CShort, CUShort, CInt, CUInt, CLng, CULng</code>. However, using these functions is misleading, because it suggests a cast. It is better to cast explicitly using <code>CType()</code>, or use <code>Convert.To()</code> when the value should be converted.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-vbnet" data-lang="vbnet">Public Class Foo
    Public Sub Bar(value as Object)
        Dim stringValue As String = CStr(value)
    End Sub
End Class</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-vbnet" data-lang="vbnet">Public Class Foo
    Public Sub Bar(value as Object)
        Dim stringValue As String = CType(value, String)
    End Sub
End Class</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_1_feb_2017_172805_ann_campbell_wrote">on 1 Feb 2017, 17:28:05 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~valeri.hristov] it&#8217;s not clear to me what the forbidden syntax is. Please at a minimum add 'Noncompliant' comments to the Noncompliant Code Example. Also, from the description</p>
</div>
<div class="quoteblock">
<blockquote>
<div class="paragraph">
<p>the naming is misleading</p>
</div>
</blockquote>
</div>
<div class="paragraph">
<p>&#8230;&#8203;the naming of what? The conversion operators? Also, I don&#8217;t understand your distinction between the old "conversion" and the new "casting". Can you expand on that in the description, pls?</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_1_feb_2017_174950_valeri_hristov_wrote">on 1 Feb 2017, 17:49:50 Valeri Hristov wrote:</h3>
<div class="paragraph">
<p>Sure, I will. Thanks for the guidelines!</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_1_feb_2017_181202_valeri_hristov_wrote">on 1 Feb 2017, 18:12:02 Valeri Hristov wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] This and 3866 are contributed by a community member and I just copy-pasted his descriptions. I am no expert in VB.NET and what he is saying seems reasonable (e.g. the functions are provided just for compatibility, hence we should prefer other methods to cast values to other types), but while reading the MSDN documentation, I began to think that we might not want to have this rule: https://msdn.microsoft.com/en-us/library/s2dy91zy.aspx, see the Remarks section, where they say just the opposite of his description&#8230;&#8203;</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_1_feb_2017_181705_ann_campbell_wrote">on 1 Feb 2017, 18:17:05 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>Agreed [~valeri.hristov]. I&#8217;d say we either flip this rule to '"CType" should <em>not</em> be used' or we close it "Won&#8217;t Fix". WDYT?</p>
</div>
<div class="paragraph">
<p>BTW, it&#8217;s probably worth going back to your community member for a response to the Remarks section.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_6_sep_2017_113510_jean_christophe_collet_wrote">on 6 Sep 2017, 11:35:10 Jean-Christophe Collet wrote:</h3>
<div class="paragraph">
<p>After consideration and some research, it seems very controversial and with little value. So closing this rules as "Won&#8217;t Fix"</p>
</div>
</div>
</div>
</div>