include::../description.adoc[]

== Noncompliant Code Example

[source,vbnet]
----
Namespace MyLibrary  ' Noncompliant

End Namespace
----

== Compliant Solution

[source,vbnet]
----
<Assembly: Runtime.InteropServices.ComVisible(False)>

Namespace MyLibrary

End Namespace
----
ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../highlighting.adoc[]

endif::env-github,rspecator-view[]
