== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,vbnet]
----
Namespace MyLibrary  ' Noncompliant

End Namespace
----

=== Compliant solution

[source,vbnet]
----
<Assembly: Runtime.InteropServices.ComVisible(False)>

Namespace MyLibrary

End Namespace
----
ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../highlighting.adoc[]

endif::env-github,rspecator-view[]
