== Why is this an issue?

Comments are meant to be human readable. Writing valid sentences is a good way to ensure this is achieved. 


Note that XML documentation and trailing comments are not affected by this rule.


=== Noncompliant code example

[source,text]
----
'  tbd
----


=== Compliant solution

[source,text]
----
' To be determined.
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

endif::env-github,rspecator-view[]
