<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Using a data value clause in the <code>LINKAGE SECTION</code> can lead to unexpected behavior at runtime.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cobol" data-lang="cobol">LINKAGE SECTION.

01 CAR-ID PIC X(20) VALUE IS "VOLVO".   *&gt; Noncompliant

01  EMP-TYPE     PIC X.
   88   FULL-TIME-EMPLOYEE VALUE "F".   *&gt; Compliant; this is a condition name
   88   PART-TIME-EMPLOYEE VALUE "P".

01 TRAIN-ID PIC X(20)</code></pre>
</div>
</div>
</div>
</div>
</div>