== Why is this an issue?

The ``++result_cache++`` Oracle hint can vastly improve performance, but it comes at the cost of extra memory consumption, so one should double-check that the gain in performance is significant, and avoid overusing this feature in general.


=== Noncompliant code example

[source,sql]
----
SELECT /*+ result_cache */ * FROM DUAL;  -- Noncompliant
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

endif::env-github,rspecator-view[]
