== Why is this an issue?

There are two possible syntaxes for a lambda having only one input parameter with an inferred type: with and without parentheses around that single parameter. The simpler syntax, without parentheses, is more compact and readable than the one with parentheses, and is therefore preferred.


*Note* that this rule is automatically disabled when the project's ``++sonar.java.source++`` is lower than ``++8++``.


=== Noncompliant code example

[source,java]
----
(x) -> x * 2
----


=== Compliant solution

[source,java]
----
x -> x * 2
----



ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Remove the parentheses around the "XXX" parameter.  [(sonar.java.source not set. Assuming 8 or greater.)]


'''
== Comments And Links
(visible only on this page)

=== on 26 Feb 2014, 00:01:00 Freddy Mallet wrote:
Is implemented by \http://jira.codehaus.org/browse/SONARJAVA-464

endif::env-github,rspecator-view[]
