== Why is this an issue?

Instances of classes that do not derive from the "Throwable" interface cannot be used in a PHP "throw" statement. Thus, it does not make sense to try to catch such objects within a "try-catch" block.


Many built-in exceptions such as "Exception" and the SPL exception classes do implement the "Throwable" interface and can be extended when creating custom exceptions.


This rule raises an issue when the classes used to specify the type of objects to be caught in a "try-catch" block do not derive from "Throwable" .

=== Noncompliant code example

[source,php]
----
class NoThrowable {}

try {
    foo();
} catch (NoThrowable $e) { // Noncompliant
}
----

=== Compliant solution

[source,php]
----
<?php

class SomeThrowable implements Throwable {
    // Implementation of the Throwable methods
}

try {
    foo();
} catch (SomeThrowable $e) { // Compliant
}

class SomeCustomException extends Exception {}

try {
    foo();
} catch (SomeCustomException $e) { // Compliant
}{code}
Â 
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Change this type to be a class deriving from "Throwable".


=== Highlighting

* Primary: The caught class name
* Secondary: The name of the class in its class definition.
** message: 'Class definition.'


'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
