<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>The <code>char</code> type within C&#43;&#43; is defined for use with the implementation character set. It is implementation-defined if <code>char</code> is signed or unsigned, and it is therefore unsuitable for use with numeric data.</p>
</div>
<div class="paragraph">
<p>Character values consist of character literals or strings. A character set maps text characters onto numeric values; the character value is the text itself.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">char c = 10; // Noncompliant</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">char c = 'a'; // Compliant</code></pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p>MISRA C:2004, 6.1 - The plain char type shall be used only for the storage and use of character values.</p>
</li>
<li>
<p>MISRA C&#43;&#43;:2008, 5-0-11 - The plain char type shall only be used for the storage and use of character values.</p>
</li>
</ul>
</div>
<hr>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_duplicates_s810">duplicates: <a data-rspec-id="S810" class="rspec-auto-link">S810</a></h3>

</div>
<div class="sect2">
<h3 id="_relates_to_s810">relates to: <a data-rspec-id="S810" class="rspec-auto-link">S810</a></h3>

</div>
<div class="sect2">
<h3 id="_on_21_oct_2014_182939_ann_campbell_wrote">on 21 Oct 2014, 18:29:39 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~samuel.mercier] please provide a message.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_13_jan_2015_093009_samuel_mercier_wrote">on 13 Jan 2015, 09:30:09 Samuel Mercier wrote:</h3>
<div class="paragraph">
<p>merged with <a data-rspec-id="S810" class="rspec-auto-link">RSPEC-810</a></p>
</div>
</div>
</div>
</div>