== Why is this an issue?

The ``++match++`` statement should be used only to clearly define some new branches in the control flow. As soon as a ``++case++`` clause contains too many statements this highly decreases the readability of the overall control flow statement. In such case, the content of the ``++case++`` clause should be extracted into a dedicated method.

=== Noncompliant code example

With the default threshold of 5:

[source,scala]
----
myVariable match {
  case 0 => // Noncompliant: 6 lines till next case
    methodCall1()
    methodCall2()
    methodCall3()
    methodCall4()
    methodCall5()
  case 1 =>
  // ...
}
----

=== Compliant solution

[source,scala]
----
myVariable match {
  case 0 => doSomething()
  case 1 =>
  // ...
}
// ...
def doSomething(): Unit = {
  methodCall1()
  methodCall2()
  methodCall3()
  methodCall4()
  methodCall5()
}
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../parameters.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
