include::../why-dotnet.adoc[]

=== Code examples

==== Noncompliant code example

[source,vbnet,diff-id=1,diff-type=noncompliant]
----
Public Sub Method(Logger As ILogger, User As User, Order As Order)
    Logger.LogDebug("User {Id} purchased order {Id}", User.Id, Order.Id)
End Sub
----

==== Compliant solution

[source,vbnet,diff-id=1,diff-type=compliant]
----
Public Sub Method(Logger As ILogger, User As User, Order As Order)
    Logger.LogDebug("User {UserId} purchased order {OrderId}", User.Id, Order.Id)
End Sub
----

include::../resources-dotnet.adoc[]