The only possible interest of calling a function which does not have any side effect is the use of its return value. So if this latter is ignored, the call to this function is useless and should be removed.

By adding the ``++nodiscard++`` attribute to the function, such dead code would be highlighted: indeed, if the return value of a function with this attribute is ignored, a warning is raised during compilation.


== Noncompliant Code Example

----
int getValue() const; // Noncompliant
----


== Compliant Solution

----
[[nodiscard]] int getValue() const;
----

ifdef::env-github,rspecator-view[]

'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
