<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p><code>FORMS_DDL</code> command, like every DDL statements, is performing an implicit COMMIT. It should be used only if there is no pending transaction otherwise this transaction is automatically committed without updating the Form statuses. Also, the potentially acquired locks are lost in case of this implicit COMMIT.</p>
</div>
<div class="paragraph">
<p>"FORMS_DDL('COMMIT')" and "FORMS_DDL('ROLLBACK')" should be used with care and most of the time, "COMMIT_FORM" or "ROLLBACK_FORM" should be preferred.</p>
</div>
<div class="paragraph">
<p>Check the Oracle Forms documentation for more details.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-sql" data-lang="sql">FORMS_DDL('COMMIT');</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-sql" data-lang="sql">COMMIT_FORM;</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Replace "FORMS_DDL('COMMIT')" by "COMMIT_FORM"</p>
</div>
<div class="paragraph">
<p>or</p>
</div>
<div class="paragraph">
<p>Replace "FORMS_DDL('ROLLBACK')" by "ROLLBACK_FORM"</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_12_apr_2018_210940_ann_campbell_wrote">on 12 Apr 2018, 21:09:40 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>If possible [~alexandre.gigleux] I would shorten the title to: "FORMS_DDL" should not be used</p>
</div>
</div>
</div>
</div>