<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>The use of the <code>enum.ordinal()</code> method is regarded as suspicious since conceptually, the significance of an <code>enum</code> member is in what it represents, not its position within the <code>enum</code>.</p>
</div>
<div class="paragraph">
<p>According to the method&#8217;s Javadoc:</p>
</div>
<div class="quoteblock">
<blockquote>
<div class="paragraph">
<p>Most programmers will have no use for this method. It is designed for use by sophisticated enum-based data structures, such as <code>EnumSet</code> and <code>EnumMap</code>.</p>
</div>
</blockquote>
</div>
<div class="paragraph">
<p>Further, reliance on a particular member&#8217;s position within an <code>enum</code> is a bug waiting to happen since the position of members within an <code>enum</code> is not generally regarded as significant, and maintainers are just as likely to add new members at the beginning as at the end.</p>
</div>
<div class="paragraph">
<p>Therefore this rule raises an issue for each use of the <code>ordinal</code> method.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">public enum Fruit {
  APPLE, ORANGE, PLUM, GRAPE;
}

public void doTheThing(Fruit fruit) {

  if (fruit.ordinal() == 2) {  // Noncompliant
    // ...
  }
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">public enum Fruit {
  APPLE, ORANGE, PLUM, GRAPE;
}

public void doTheThing(Fruit fruit) {

  if (Fruit.PLUM.equals(fruit)) {
    // ...
  }
}</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Make sure this call to "XXX.ordinal()" is really what&#8217;s intended.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_30_sep_2015_143338_ann_campbell_wrote">on 30 Sep 2015, 14:33:38 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>please review</p>
</div>
</div>
</div>
</div>