<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p><code>goto</code> is an unstructured control flow statement. It makes code less readable and maintainable. Structured control flow statements such as <code>if</code>, <code>for</code>, <code>while</code>, <code>continue</code> or <code>break</code> should be used instead.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">int i = 0;
loop:
  printf("i = %d\n", i);
  i++;
  if (i &lt; 10){
    goto loop; // Noncompliant
  }</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">for (int i = 0; i &lt; 10; i++) {
  printf("i = %d\n", i);
}</code></pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p>MISRA C:2004, 14.4 - The goto statement shall not be used.</p>
</li>
<li>
<p>MISRA C:2012, 15.1 - The goto statement should not be used</p>
</li>
<li>
<p>C++ Core Guidelines - <a href="https://github.com/isocpp/CppCoreGuidelines/blob/e49158a/CppCoreGuidelines.md#es76-avoid-goto">ES.76: Avoid <code>goto</code></a></p>
</li>
</ul>
</div>
<hr>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove this use of "[goto|GOTO|GoTo]".</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_is_related_to_s1119">is related to: <a data-rspec-id="S1119" class="rspec-auto-link">S1119</a></h3>

</div>
<div class="sect2">
<h3 id="_on_19_feb_2014_153652_freddy_mallet_wrote">on 19 Feb 2014, 15:36:52 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>Is implemented by http://jira.codehaus.org/browse/SONARPLUGINS-3494 for PHP</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_9_jun_2015_123533_tamas_vajk_wrote">on 9 Jun 2015, 12:35:33 Tamas Vajk wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] Does this rule apply to C# <code>goto case</code> and <code>goto default</code> statements as well? See https://msdn.microsoft.com/en-us/library/13940fs2.aspx</p>
</div>
<div class="paragraph">
<p>I think it should.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_9_jun_2015_134220_ann_campbell_wrote">on 9 Jun 2015, 13:42:20 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~tamas.vajk] IMO yes</p>
</div>
</div>
</div>
</div>