<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>A <code>GOTO</code> statement is an unstructured change in the control flow. They should be avoided and replaced by structured constructs.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-sql" data-lang="sql">SET SERVEROUTPUT ON

DECLARE
  i PLS_INTEGER := 42;
BEGIN
  IF i &lt; 0 THEN
    GOTO negative; -- Noncompliant
  END IF;

  DBMS_OUTPUT.PUT_LINE('positive');
  goto cleanup; -- Noncompliant

  &lt;&lt;negative&gt;&gt;
  DBMS_OUTPUT.PUT_LINE('negative!');

  &lt;&lt;cleanup&gt;&gt;
  NULL;
END;
/</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-sql" data-lang="sql">SET SERVEROUTPUT ON

DECLARE
  i PLS_INTEGER := 42;
BEGIN
  IF i &lt; 0 THEN
    DBMS_OUTPUT.PUT_LINE('negative!'); -- Compliant
  ELSE
    DBMS_OUTPUT.PUT_LINE('positive');
  END IF;
END;
/</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove this use of "[goto|GOTO|GoTo]".</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_is_related_to_s1119">is related to: <a data-rspec-id="S1119" class="rspec-auto-link">S1119</a></h3>

</div>
<div class="sect2">
<h3 id="_on_19_feb_2014_153652_freddy_mallet_wrote">on 19 Feb 2014, 15:36:52 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>Is implemented by http://jira.codehaus.org/browse/SONARPLUGINS-3494 for PHP</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_9_jun_2015_123533_tamas_vajk_wrote">on 9 Jun 2015, 12:35:33 Tamas Vajk wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] Does this rule apply to C# <code>goto case</code> and <code>goto default</code> statements as well? See https://msdn.microsoft.com/en-us/library/13940fs2.aspx</p>
</div>
<div class="paragraph">
<p>I think it should.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_9_jun_2015_134220_ann_campbell_wrote">on 9 Jun 2015, 13:42:20 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~tamas.vajk] IMO yes</p>
</div>
</div>
</div>
</div>