Included variables may have been set by user input could contain unexpected, and potentially dangerous values. 


== Noncompliant Code Example

[source,php]
----
include $foo;
require $foo;
include $foo . '/config.php';
require $foo . '/config.php';
include "$foo/config.php";
require "$foo/config.php";
include($foo);
require($foo);
include($foo . '/config.php');
require($foo . '/config.php');
include("$foo/config.php");
require("$foo/config.php");

include "./$page.php";
----


== Compliant Solution

[source,php]
----
include "./$page.php";
----

