<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Oracle&#8217;s <code>ROWNUM</code> is a pseudo column that numbers the rows in a result set. Unfortunately, it numbers the rows in the set <em>before</em> ordering is applied. So combining the two in the same query won&#8217;t get you the results you expect. Instead, you should move your selection and ordering into a subquery, and use <code>ROWNUM</code> only on the outer query.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-sql" data-lang="sql">SELECT fname, lname, deptId
FROM employee
WHERE rownum &lt;= 10
ORDER BY salary  -- Noncompliant</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-sql" data-lang="sql">SELECT *
FROM ( SELECT fname, lname, deptId
    FROM employee
    ORDER BY salary
  )
WHERE rownum &lt;= 10</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Order your results in a subquery and apply "rownum" at the level of the outer query.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="ulist">
<ul>
<li>
<p>primary: order by</p>
</li>
<li>
<p>secondary: rownum|row_number()</p>
</li>
</ul>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_27_jun_2017_155756_pierre_yves_nicolas_wrote">on 27 Jun 2017, 15:57:56 Pierre-Yves Nicolas wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] The title may be misleading. Some other possibilities:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>"ROWNUM should not be used at the same query level as ORDER BY"</p>
</li>
<li>
<p>or "ROWNUM should not be used with ORDER BY without a subquery"</p>
</li>
</ul>
</div>
</div>
<div class="sect2">
<h3 id="_on_27_jun_2017_160955_ann_campbell_wrote">on 27 Jun 2017, 16:09:55 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~pierre-yves.nicolas]: "ORDER BY" should not be applied to queries that use "ROWNUM"</p>
</div>
<div class="paragraph">
<p>?</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_27_jun_2017_162611_pierre_yves_nicolas_wrote">on 27 Jun 2017, 16:26:11 Pierre-Yves Nicolas wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] "ORDER BY should not be applied on top of a ROWNUM-based WHERE"? :)</p>
</div>
</div>
</div>
</div>