== How to fix it in Dapper

=== Code examples

include::../../common/fix/code-rationale.adoc[]

==== Noncompliant code example

[source,vbnet,diff-id=1,diff-type=noncompliant]
----
Public Class ExampleController
    Inherits Controller

    Private ReadOnly ConnectionString As String

    Public Function Authenticate(User As String, Pass As String) As IActionResult
        Using Connection As New SqlConnection(ConnectionString)
            Dim Query As String = "SELECT * FROM users WHERE user = '" & User & "' AND pass = '" & Pass & "'"

            Dim Result = Connection.QueryFirst(Of UserEntity)(Query)
            If Result Is Nothing Then
                Unauthorized()
            End If
        End Using
        Return Ok()
    End Function
End Class
----

==== Compliant solution

[source,vbnet,diff-id=1,diff-type=compliant]
----
Public Class ExampleController
    Inherits Controller

    Private ReadOnly ConnectionString As String

    Public Function Authenticate(User As String, Pass As String) As IActionResult
        Using Connection As New SqlConnection(ConnectionString)
            Dim Query As String = "SELECT * FROM users WHERE user = @UserName AND password = @Password"
            Dim Parameters = New With {.UserName = User, .Password = Pass}

            Dim Result = Connection.QueryFirst(Of UserEntity)(Query, Parameters)
            If Result Is Nothing Then
                Unauthorized()
            End If
        End Using
        Return Ok()
    End Function
End Class
----

=== How does this work?

include::../../common/fix/prepared-statements.adoc[]

