== How to fix it in Dapper

=== Code examples

include::../../common/fix/code-rationale.adoc[]

==== Noncompliant code example

[source,vbnet,diff-id=1,diff-type=noncompliant]
----
Public Class ExampleController
    Inherits Controller

    Private ReadOnly ConnectionString As String

    Public Function Authenticate(user As String, pass As String) As IActionResult
        Using connection As New SqlConnection(ConnectionString)
            Dim query As String = "SELECT * FROM users WHERE user = '" + user + "' AND pass = '" + pass + "'"
            
            Dim result = connection.QueryFirst(Of User)(query)
            If result Is Nothing Then
                Unauthorized()
            End If
        End Using
        Return Ok()
    End Function
End Class
----

==== Compliant solution

[source,vbnet,diff-id=1,diff-type=compliant]
----
Public Class ExampleController
    Inherits Controller

    Private ReadOnly ConnectionString As String

    Public Function Authenticate(user As String, pass As String) As IActionResult
        Using connection As New SqlConnection(ConnectionString)
            Dim query As String = "SELECT * FROM users WHERE user = @UserName AND password = @Password"
            Dim parameters = New With {.UserName = user, .Password = pass}
            
            Dim result = connection.QueryFirst(Of User)(query, parameters)
            If result Is Nothing Then
                Unauthorized()
            End If
        End Using
        Return Ok()
    End Function
End Class
----

=== How does this work?

include::../../common/fix/prepared-statements.adoc[]

