== How to fix it in Entity Framework Core

=== Code examples

include::../../common/fix/code-rationale.adoc[]

==== Noncompliant code example

[source,vbnet,diff-id=11,diff-type=noncompliant]
----
Public Class ExampleController
    Inherits Controller

    Private ReadOnly Context As UserAccountContext

    Public Function Authenticate(user As String, pass As String) As IActionResult
        Dim query As String = "SELECT * FROM users WHERE user = '" & user & "' AND pass = '" & pass & "'"

        Dim dbSet As DbSet(Of UserAccount) = New UserAccountContext().Set(Of UserAccount)()
        Dim queryResults As IQueryable(Of UserAccount) = dbSet.FromSqlRaw(query)

        If Not queryResults.Any() Then
            Return Unauthorized()
        End If

        Return Ok()
    End Function
End Class
----

==== Compliant solution

[source,vbnet,diff-id=11,diff-type=compliant]
----
Public Class ExampleController
    Inherits Controller

    Private ReadOnly Context As UserAccountContext

    Public Function Authenticate(user As String, pass As String) As IActionResult
        Dim query As String = "SELECT * FROM users WHERE user = {0} AND pass = {1}"

        Dim dbSet As DbSet(Of UserAccount) = New UserAccountContext().Set(Of UserAccount)()
        Dim queryResults = dbSet.FromSqlRaw(query, user, pass)

        If Not queryResults.Any() Then
            Return Unauthorized()
        End If

        Return Ok()
    End Function
End Class
----

=== How does this work?

include::../../common/fix/prepared-statements.adoc[]

