== How to fix it in Entity Framework Core

=== Code examples

include::../../common/fix/code-rationale.adoc[]

==== Noncompliant code example

[source,vbnet,diff-id=11,diff-type=noncompliant]
----
Public Class ExampleController
    Inherits Controller

    Public Function Authenticate(User As String, Pass As String) As IActionResult
        Dim Query As String = "SELECT * FROM users WHERE user = '" & User & "' AND pass = '" & Pass & "'"

        Dim DbSet As DbSet(Of UserAccount) = New UserAccountContext().Set(Of UserAccount)()
        Dim QueryResults = DbSet.FromSqlRaw(Query)

        If Not QueryResults.Any() Then
            Return Unauthorized()
        End If

        Return Ok()
    End Function
End Class
----

==== Compliant solution

[source,vbnet,diff-id=11,diff-type=compliant]
----
Public Class ExampleController
    Inherits Controller

    Public Function Authenticate(User As String, Pass As String) As IActionResult
        Dim Query As String = "SELECT * FROM users WHERE user = {0} AND pass = {1}"

        Dim DbSet As DbSet(Of UserAccount) = New UserAccountContext().Set(Of UserAccount)()
        Dim QueryResults = DbSet.FromSqlRaw(Query, User, Pass)

        If Not QueryResults.Any() Then
            Return Unauthorized()
        End If

        Return Ok()
    End Function
End Class
----

=== How does this work?

include::../../common/fix/prepared-statements.adoc[]

