include::../description.adoc[]

== Noncompliant Code Example

----
public Integer numberOfMinutes(Integer hours) {
  Integer seconds = 0;   // seconds is never used 
  return hours * 60;
}
----

== Compliant Solution

----
public Integer numberOfMinutes(Integer hours) {
  return hours * 60;
}
----

ifdef::env-github,rspecator-view[]
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
