== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,apex]
----
public Integer numberOfMinutes(Integer hours) {
  Integer seconds = 0;   // seconds is never used 
  return hours * 60;
}
----

=== Compliant solution

[source,apex]
----
public Integer numberOfMinutes(Integer hours) {
  return hours * 60;
}
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
