include::../rationale.adoc[]

include::../how-to-fix-it.adoc[]

=== Code examples

==== Noncompliant code example

[source,apex,diff-id=1,diff-type=noncompliant]
----
public Integer numberOfMinutes(Integer hours) {
  Integer seconds = 0;   // Noncompliant - seconds is unused 
  return hours * 60;
}
----

==== Compliant solution

[source,apex,diff-id=1,diff-type=compliant]
----
public Integer numberOfMinutes(Integer hours) {
  return hours * 60;
}
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
