<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>The use of an undefined escape sequence leads to undefined behavior.</p>
</div>
<div class="paragraph">
<p>Escape sequences supported in every C and C&#43;&#43; standard are: <code>\n</code>, <code>\t</code>, <code>\v</code>, <code>\b</code>, <code>\r</code>, <code>\f</code>, <code>\a</code>, <code>\\</code>, <code>\?</code>, <code>\'</code>, <code>\"</code>, <code>\&lt;Octal Number&gt;</code>, and <code>\x&lt;Hexadecimal Number&gt;</code>.</p>
</div>
<div class="paragraph">
<p>C99 and C&#43;&#43;11 introduced escape sequences for Unicode characters in the form: <code>\u&lt;4 Hexadecimal Digits&gt;</code> and <code>\U&lt;8 Hexidecimal Digits&gt;</code>.</p>
</div>
<div class="paragraph">
<p>C&#43;&#43;23 supports named escape sequences for Unicode characters in the form <code>\N{&lt;Name of Character&gt;}</code> and a delimited version of escape sequences is supported: <code>\o{&lt;Octal Number&gt;}</code>, <code>\x{&lt;Hexadecimal Number&gt;}</code>, and <code>\u{&lt;Hexadecimal Number&gt;}</code>.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">const char_t a[ 2 ] = "\k";   // Noncompliant
const char_t b[ 2 ] = "\b";   // Compliant</code></pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p>MISRA C:2004, 4.1 - Only those escape sequences that are defined in ISO C standard shall be used.</p>
</li>
<li>
<p>MISRA C&#43;&#43;:2008, 2-13-1 - Only those escape sequences that are defined in ISO/IEC 14882:2003 shall be used.</p>
</li>
</ul>
</div>
<hr>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_6_apr_2015_131807_evgeny_mandrikov_wrote">on 6 Apr 2015, 13:18:07 Evgeny Mandrikov wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] implementation seems more complete (SQALE, description) than this spec.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_13_apr_2015_193941_evgeny_mandrikov_wrote">on 13 Apr 2015, 19:39:41 Evgeny Mandrikov wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] I&#8217;m wondering why not active by default as it is currently in implementation?</p>
</div>
</div>
</div>
</div>