<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule applies whenever an <code>if</code> statement is followed by one or more <code>elif</code> statements; the final <code>elif</code> should be followed by an <code>else</code> statement.</p>
</div>
<div class="paragraph">
<p>The requirement for a final <code>else</code> statement is defensive programming.</p>
</div>
<div class="paragraph">
<p>The <code>else</code> statement should either take appropriate action or contain a suitable comment as to why no action is taken. This is consistent with the requirement to have a final <code>*)</code> clause in a <code>case</code> statement.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-bash" data-lang="bash">if [[ $x -eq 0 ]]; then
  do_something
elif [[ $x -eq 1 ]]; then
  do_something_else
fi</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-bash" data-lang="bash">if [[ $x -eq 0 ]]; then
  do_something
elif [[ $x -eq 1 ]]; then
  do_something_else
else
  echo "Unexpected value: $x"
  exit 1
fi</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>