<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule applies whenever an <code>IF</code> statement is followed by one or more <code>ELSE IF</code> statements; the final <code>ELSE IF</code> should be followed by an <code>ELSE</code> statement.</p>
</div>
<div class="paragraph">
<p>The requirement for a final <code>ELSE</code> statement is defensive programming.</p>
</div>
<div class="paragraph">
<p>The <code>ELSE</code> statement should either take appropriate action or contain a suitable comment as to why no action is taken. This is consistent with the requirement to have a final <code>ELSE</code> clause in a <code>CASE</code> statement.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-sql" data-lang="sql">IF @x = 1
  PRINT 'A'
ELSE IF @x = 2
  PRINT 'B'
ELSE IF @x = 3
  PRINT 'C'
-- Noncompliant; final ELSE is missing</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-sql" data-lang="sql">IF @x = 1
  PRINT 'A'
ELSE IF @x = 2
  PRINT 'B'
ELSE IF @x = 3
  PRINT 'C'
ELSE
  PRINT 'Z'</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Add the missing "[else|ELSE]" clause.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_is_duplicated_by_s914">is duplicated by: <a data-rspec-id="S914" class="rspec-auto-link">S914</a></h3>

</div>
<div class="sect2">
<h3 id="_relates_to_s128">relates to: <a data-rspec-id="S128" class="rspec-auto-link">S128</a></h3>

</div>
<div class="sect2">
<h3 id="_is_related_to_s5307">is related to: <a data-rspec-id="S5307" class="rspec-auto-link">S5307</a></h3>

</div>
<div class="sect2">
<h3 id="_on_12_feb_2014_213027_freddy_mallet_wrote">on 12 Feb 2014, 21:30:27 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>Is implemented by http://jira.codehaus.org/browse/SONARPLUGINS-3464 for PHP</p>
</div>
</div>
</div>
</div>