== Why is this an issue?

Every macro should be commented to explain its goal and how it works. This comment can be located just before or after the macro definition.


=== Noncompliant code example

[source,abap]
----
DEFINE my_macro.
  ...
END-OF-DEFINITION. 
----


=== Compliant solution

[source,abap]
----
* here is  
* my comment
DEFINE my_macro.
  ...
END-OF-DEFINITION. 
----

or 

[source,abap]
----
DEFINE my_macro.
* here is  
* my comment
  ...
END-OF-DEFINITION. 
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

endif::env-github,rspecator-view[]
