Every macro should be commented to explain its goal and how it works. This comment can be located just before or after the macro definition.

== Noncompliant Code Example

----
DEFINE my_macro.
  ...
END-OF-DEFINITION. 
----

== Compliant Solution

----
* here is  
* my comment
DEFINE my_macro.
  ...
END-OF-DEFINITION. 
----

or 

----
DEFINE my_macro.
* here is  
* my comment
  ...
END-OF-DEFINITION. 
----
