include::../description.adoc[]

== Noncompliant Code Example

With the default threshold value 3.

----
if (((condition1 && condition2) || (condition3 && condition4)) && condition5) { ... }
----

== Compliant Solution

----
if ((myFirstCondition() || mySecondCondition()) && myLastCondition()) { ... }
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../parameters.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
