When the call to a function doesn't have any side effects, what is the point of making the call if the results are ignored? In such case, either the function call is useless and should be dropped or the source code doesn't behave as expected.


This rule raises an issue when a builtin function or methods which has no side effects is called and its result is not used.

== Noncompliant Code Example

----
myvar = "this is a multiline"
"message from {}".format(sender)  # Noncompliant. The formatted string is not used because the concatenation is not done properly.
----

== Compliant Solution

----
myvar = ("this is a multiline"
"message from {}".format(sender))
----

== Exceptions

No issue will be raised when the function or method call is in a ``++try...except++`` body. This usually indicates that an exception is expected, and this exception is the side-effect.

----
def tryExcept():
    d = {}
    try:
        d[1]
    except IndexError as e:
        pass

    try:
        divmod(1, 0)
    except ZeroDivisionError as e:
        pass
----

== See

* https://docs.python.org/3/library/functions.html[Python documentation - Built-in Functions]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
