<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Comparisons that always evaluate to true or to false, logical expressions that either always or never short-circuit and comparisons to a newly constructed object should not be used.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>An expression that always produces the same result, regardless of the inputs, is unnecessary and likely indicates a programmer&#8217;s error. This can come from</p>
</div>
<div class="ulist">
<ul>
<li>
<p>confusing operator precedence</p>
</li>
<li>
<p>expecting strict equality between different types</p>
</li>
<li>
<p>expecting objects to be compared by value</p>
</li>
<li>
<p>expecting empty objects to be <code>false</code> or <code>null</code></p>
</li>
<li>
<p>mistyping <code>&gt;=</code> for <code>&#8658;</code></p>
</li>
</ul>
</div>
<div class="paragraph">
<p>This can also happen when you put an assignment in a logical sub-expression. While not strictly a bug, this practice is confusing and should be avoided.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Review the code around the issue to find out why the expression always produces the same result. Pay attention to the operator precedence, comparing objects of different types, and comparing objects by reference (not by value!).</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-javascript" data-lang="javascript">!foo == null;
a + b ?? c;
x === [];
(foo=0) &amp;&amp; bar;</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-javascript" data-lang="javascript">foo != null;
a + (b ?? c);
x.length === 0;</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p><a href="https://eslint.org">eslint</a> - Rule <a href="https://eslint.org/docs/latest/rules/no-constant-binary-expression">no-constant-binary-expression</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>