Includes ``++time++``, ``++strftime++``. This library is associated with clock times. Various aspects are implementation dependent or unspecified, such as the formats of times. If any of the facilities of ``++time.h++`` are used, then the exact implementation for the compiler being used must be determined, and a deviation raised.


== Noncompliant Code Example

[source,cpp]
----
#include <time.h>  /* Noncompliant */
----


== See

* MISRA C:2004, 20.12 - The time handling functions of library <time.h> shall not be used.
* MISRA C:2012, 21.10 - The Standard Library time and date functions shall not be used


ifdef::env-github,rspecator-view[]
'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
