== Why is this an issue?

Includes ``++time++``, ``++strftime++``. This library is associated with clock times. Various aspects are implementation dependent or unspecified, such as the formats of times. If any of the facilities of ``++time.h++`` are used, then the exact implementation for the compiler being used must be determined, and a deviation raised.


=== Noncompliant code example

[source,cpp]
----
#include <time.h>  /* Noncompliant */
----


== Resources

* MISRA C:2004, 20.12 - The time handling functions of library <time.h> shall not be used.
* MISRA C:2012, 21.10 - The Standard Library time and date functions shall not be used


ifdef::env-github,rspecator-view[]
'''
== Comments And Links
(visible only on this page)

=== on 31 Mar 2015, 19:05:06 Evgeny Mandrikov wrote:
\[~ann.campbell.2] implementation seems more complete (SQALE, description) than this spec.

=== on 13 Apr 2015, 19:40:44 Evgeny Mandrikov wrote:
\[~ann.campbell.2] I'm wondering why blocker, but not active by default? Note that in implementation currently major and not active.

endif::env-github,rspecator-view[]
