include::../summary.adoc[]

== Why is this an issue?

include::../description.adoc[]

== How to fix it

include::../how-to-fix-it.adoc[]

=== Code examples

==== Noncompliant code example

[source,docker,diff-id=1,diff-type=noncompliant]
----
RUN rm *
----

==== Compliant solution

[source,docker,diff-id=1,diff-type=compliant]
----
RUN rm ./*
RUN rm -- *
----

== Resources

include::../resources.adoc[]

ifdef::env-github,rspecator-view[]
'''
== Implementation Specification
(visible only on this page)

=== Message

include::../message.adoc[]

=== Highlighting

include::../highlighting.adoc[]

'''
endif::env-github,rspecator-view[]
