<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>"A rose by any other name would smell as sweet," but <code>main</code> by any other name would not. Just because a method has the name "main", that doesn&#8217;t make it the entry point to an application. It must also have the correct signature. Specifically, it must be <code>public static void</code> and accept a single <code>String []</code> as an argument.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">public void main(String arg) {  // Noncompliant
  // ...
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">public static void main(String [] args) {
  // ...
}</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Make this method ["public" | "static" | return "void" | accept a "String []" argument].</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_relates_to_s998">relates to: <a data-rspec-id="S998" class="rspec-auto-link">S998</a></h3>

</div>
<div class="sect2">
<h3 id="_on_11_jun_2015_201901_ann_campbell_wrote">on 11 Jun 2015, 20:19:01 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>Code Pro: Entry point method</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_16_jun_2015_131833_nicolas_peru_wrote">on 16 Jun 2015, 13:18:33 Nicolas Peru wrote:</h3>
<div class="paragraph">
<p>Looks good. I really like this first sentence ! :)</p>
</div>
</div>
</div>
</div>