"A rose by any other name would smell as sweet," but `+main+` by any other name would not. Just because a method has the name "main", that doesn't make it the entry point to an application. It must also have the correct signature. Specifically, it must be `+public static void+` and accept a single `+String []+` as an argument.


== Noncompliant Code Example

----
public void main(String arg) {  // Noncompliant
  // ...
}
----


== Compliant Solution

----
public static void main(String [] args) {
  // ...
}
----


