Using more than one ``++OF++`` clause to access a data item can quickly decrease the readability of the source code. Either some ``++OF++`` clauses are optional and should be removed, or there are too many intersections between several data structures and those intersections should be removed.

== Noncompliant Code Example

----
01 EMPLOYEE. 
  05 MOTHER-IN-LAW. 
    10 NAME PIC X(20). 
  05 FATHER-IN-LAW. 
    10 NAME PIC X(20). 
... 
01 CUSTOMER. 
  05 MOTHER-IN-LAW. 
    10 NAME PIC X(20). 
  05 FATHER-IN-LAW. 
    10 NAME PIC X(20). 
... 

MOVE MY_VALUE TO NAME OF MOTHER-IN-LAW OF CUSTOMER 
----
