<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when a PyTorch model state is loaded and <code>torch.nn.Module.eval()</code> or <code>torch.nn.Module.train()</code> is not called.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>When using PyTorch it is common practice to load and save a model&#8217;s state from/to a <code>.pth</code> file.
Doing so allows, for example, to instantiate an untrained model and load learned parameters coming from another pre-trained model.
Once the learned parameters are loaded to the model it is important, before inferencing,
to clearly state the intention by calling <code>torch.nn.Module.eval()</code> method to set the model in evaluation mode
or calling <code>torch.nn.Module.train()</code> to indicate the training will resume.
Failing to call <code>torch.nn.Module.eval()</code> would leave the model in training mode which may not be the intention.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Call the <code>torch.nn.Module.eval()</code> or <code>torch.nn.Module.train()</code> method on the model.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">import torch
import torchvision.models as models

model = models.vgg16()
model.load_state_dict(torch.load('model_weights.pth')) # Noncompliant: model.train() or model.eval() was not called.</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">import torch
import torchvision.models as models

model = models.vgg16()
model.load_state_dict(torch.load('model_weights.pth'))
model.eval()</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>PyTorch Documentation - <a href="https://pytorch.org/docs/master/generated/torch.nn.Module.html#torch.nn.Module.eval">eval - reference</a></p>
</li>
<li>
<p>PyTorch Documentation - <a href="https://pytorch.org/docs/master/generated/torch.nn.Module.html#torch.nn.Module.train">train - reference</a></p>
</li>
<li>
<p>PyTorch Documentation - <a href="https://pytorch.org/docs/stable/notes/autograd.html#evaluation-mode-nn-module-eval">Autograd - Evaluation Mode</a></p>
</li>
</ul>
</div>
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation specification</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Primary : Set the module in training or evaluation mode.</p>
</div>
</div>
<div class="sect2">
<h3 id="_issue_location">Issue location</h3>
<div class="paragraph">
<p>Primary : the call to model.load_state_dict</p>
</div>
</div>
<div class="sect2">
<h3 id="_quickfix">Quickfix</h3>
<div class="paragraph">
<p>No</p>
</div>
</div>
</div>
</div>