The use of positional notation can quickly become confusing, especially in a large or complex data structure. To keep the code clean and understandable, use length notation instead.


== Noncompliant Code Example

----
     D PERSON          DS                  QUALIFIED
      * Noncompliant
     D  FName                  1     30
     D  LName                 31     60
----


== Compliant Solution

----
     D PERSON          DS                  QUALIFIED
     D  FName                        30
     D  LName                        30
----

