This rule raises an issue when `list()`, `set()` or `dict()` is unnecessarily used around a generator expression.

== Why is this an issue?

Using `list()`, `set()`, or `dict()` around a generator expression is redundant when a corresponding comprehension can directly express the same operation. Comprehensions are clearer, more concise, and often more readable than the equivalent constructor/generator expression combination.

This principle applies to all three built-in collection types: `list`, `set`, and `dict`:

* Use `[f(x) for x in foo]` instead of `list(f(x) for x in foo)`
* Use `{f(x) for x in foo}` instead of `set(f(x) for x in foo)`
* Use `{k: v for k, v in items}` instead of `dict((k, v) for k, v in items)`

=== Exceptions
If the generator expression does filter or modify the collection, the rule does not raise.
For example, `list(x for x in foo)` is simply copying the iterable `foo` into a list, which is equivalent to `list(foo)` and covered by a different rule.

== How to fix it

Replace the collection constructor with the appropriate comprehension syntax.

=== Code examples

==== Noncompliant code example

[source,python,diff-id=1,diff-type=noncompliant]
----
def f(x):
    return x * 2

list(f(x) for x in range(5))  # Noncompliant
----

==== Compliant solution

[source,python,diff-id=1,diff-type=compliant]
----
def f(x):
    return x * 2

[f(x) for x in range(5)] # Compliant
----

== Resources
=== Documentation
* Python Documentation - https://docs.python.org/3/tutorial/datastructures.html#list-comprehensions[List Comprehensions]
* Python Documentation - https://docs.python.org/3/tutorial/datastructures.html#dictionaries[Dictionaries]
* Python Documentation - https://docs.python.org/3/tutorial/datastructures.html#sets[Sets]

ifdef::env-github,rspecator-view[]
== Implementation Specification
(visible only on this page)

The rule implementation must only raise if the generator expression is actually changing the iterable. This means that one of the following must be true:

* The generator expression maps items: `f(x) for x in foo` or `x + 1 for x in foo`
    * In case of a dictionary, either the key or the value must be changed, or reversed (`v: k for k, v in items`).
* The generator expression filters items
* There are multiple `for` keywords: `[ (i,j) for i in range(1,3) for j in range(1,5) ]`

=== Message
* Replace list constructor call with a list comprehension.
* Replace set constructor call with a set comprehension.
* Replace dict constructor call with a dictionary comprehension.

=== Highlighting
The list/set/dict constructor call.
endif::env-github,rspecator-view[]
