== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,java]
----
int myNumber = 010; // Noncompliant. myNumber will hold 8, not 10 - was this really expected?
----


=== Compliant solution

[source,java]
----
int myNumber = 8;
----


== Resources

* https://wiki.sei.cmu.edu/confluence/x/atYxBQ[CERT, DCL18-C.] - Do not begin integer constants with 0 when specifying a decimal value
* https://wiki.sei.cmu.edu/confluence/x/7DZGBQ[CERT, DCL50-J.] - Use visually distinct identifiers

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
