Calling the ``++not++`` or ``++~++`` prefix operator twice does nothing: the second invocation undoes the first. Such mistakes are typically caused by accidentally double-tapping the key in question without noticing.


Either this is a bug, if the operator was actually meant to be called once, or misleading if done on purpose.

== Noncompliant Code Example

[source,python]
----
v1 = 0
v2 = False

v3 = ~~v1           # Noncompliant
v4 = not not v2     # Noncompliant
----

== Compliant Solution

[source,python]
----
v1 = 0
v2 = False

v3 = ~v1
v4 = not v2
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
