include::../description.adoc[]

== Noncompliant Code Example

With the default regular expression ``++^[a-z_]+(\.[a-z_][a-z0-9_]*)*$++``:

----
package org.exAmple; // Noncompliant
----

== Compliant Solution

----
package org.example; 
----
