This is an issue when a server action annotated with `@AuraEnabled` performs read-only operations but is not marked as `cacheable=true`.

== Why is this an issue?

Lightning Components can cache server responses on the client side to improve performance and reduce server load. When a server action is idempotent and non-mutating (such as data retrieval operations), it should be marked as cacheable using `@AuraEnabled(cacheable=true)`.

Idempotent operations are those that produce the same result when called multiple times with the same parameters. Read-only operations like SOQL queries are perfect candidates for caching because they don't modify data or have side effects.

Without caching, every component that needs the same data will make separate server calls, even if the data hasn't changed. This creates unnecessary network traffic and increases response times, especially when multiple components on the same page request similar data.

The Lightning Component Framework's storable actions feature automatically handles cache invalidation and ensures data consistency while providing significant performance improvements.

=== What is the potential impact?

Without proper caching, applications experience:

* Increased server load from redundant requests
* Slower page load times and component rendering
* Poor user experience, especially on slower network connections
* Higher resource consumption and potential scalability issues
* Unnecessary network traffic that could be avoided

== How to fix it

Add the `cacheable=true` parameter to the `@AuraEnabled` annotation for methods that perform read-only operations.

=== Code examples

==== Noncompliant code example

[source,apex,diff-id=1,diff-type=noncompliant]
----
@AuraEnabled
public static List<Account> getAccountList() {
    return [SELECT Id, Name FROM Account LIMIT 100]; // Noncompliant
}
----

==== Compliant solution

[source,apex,diff-id=1,diff-type=compliant]
----
@AuraEnabled(cacheable=true)
public static List<Account> getAccountList() {
    return [SELECT Id, Name FROM Account LIMIT 100];
}
----

== Resources

=== Documentation

 * Lightning Components Performance Best Practices - https://developer.salesforce.com/blogs/developer-relations/2017/04/lightning-components-performance-best-practices.html[Comprehensive guide to optimizing Lightning Components performance, including data caching strategies]

 * Caching Data with Storable Actions - https://developer.salesforce.com/blogs/developer-relations/2017/03/lightning-components-best-practices-caching-data-storable-actions.html[Detailed explanation of storable actions and client-side caching in Lightning Components]

 * AuraEnabled Annotation - https://developer.salesforce.com/docs/atlas.en-us.apexcode.meta/apexcode/apex_classes_annotation_AuraEnabled.htm[Official documentation for the @AuraEnabled annotation and its parameters]
