
== Why is this an issue?
Applying the `#[inline]` attribute to trait methods without provided implementations is pointless because only method implementations can be inlined. The attribute is therefore ignored in such cases, leading to unnecessary code complexity.


=== Code examples

==== Noncompliant code example
[source,rust,diff-id=1,diff-type=noncompliant]
----
trait Animal {
    #[inline] // Noncompliant: Inline attribute on trait method without implementation.
    fn name(&self) -> &'static str;
}
----

==== Compliant solution

[source,rust,diff-id=1,diff-type=compliant]
----
trait Animal {
    fn name(&self) -> &'static str;
}
----

== Resources
=== Documentation

* Clippy Lints - https://rust-lang.github.io/rust-clippy/master/index.html#inline_fn_without_body
