include::../description.adoc[]

== Noncompliant Code Example

----
if (a) {
  if (a) { // Noncompliant
    doSomething();
  }
}
----

== Compliant Solution

----
if (a) {
  if (b) {
    doSomething();
  }
}

// or 
if (a) {
  doSomething();
}
----

include::../see.adoc[]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

include::../highlighting.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
