include::../description-common.adoc[]

=== Code examples

==== Noncompliant code example

[source,javascript,diff-id=1,diff-type=noncompliant]
----
if (a) {
  if (a) { // Noncompliant
    doSomething();
  }
}
----

==== Compliant solution

[source,javascript,diff-id=1,diff-type=compliant]
----
if (a) {
  if (b) {
    doSomething();
  }
}

// or 
if (a) {
  doSomething();
}
----

include::../see.adoc[]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Refactor this code so that this expression does not always evaluate to [true|false].


include::../highlighting.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
