The main motivation for this rule is to improve the readability of relevant SQL code. From one database optimiser to another, the performance of <code>IN</code> and <code>OR</code> clauses to specify a list of possible values for a column can be slightly different, but this difference is usually very minor. What is not minor is the difference in readability between the two styles, which is why an <code>IN</code> clause is preferred. 


== Noncompliant Code Example

----
SELECT * FROM PERSONS
WHERE AGE = 10 OR AGE = 13 OR AGE = 18 OR AGE < 5 
----


== Compliant Solution

----
SELECT * FROM PERSONS
WHERE AGE IN (10, 13, 18) OR AGE < 5 
----


