== Why is this an issue?

Shared coding conventions allow teams to collaborate efficiently. This rule checks that paragraph names match a provided regular expression.


=== Noncompliant code example

With a regular expression of ``++[A-Z-]+++``:

[source,cobol]
----
 PROCEDURE DIVISION.

    Do_The_Thing.           *> Noncompliant
----


=== Compliant solution

[source,cobol]
----
 PROCEDURE DIVISION.

    DO-THE-THING           *> Noncompliant
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Parameters

.regexPatternString
****

A regular expression to specify the paragraph naming convention
****


endif::env-github,rspecator-view[]
