Shared coding conventions allow teams to collaborate efficiently. This rule checks that paragraph names match a provided regular expression.


== Noncompliant Code Example

With a regular expression of ``++[A-Z-]+++``:

[source,cobol]
----
 PROCEDURE DIVISION.

    Do_The_Thing.           *> Noncompliant
----


== Compliant Solution

----
 PROCEDURE DIVISION.

    DO-THE-THING           *> Noncompliant
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::parameters.adoc[]

endif::env-github,rspecator-view[]
