== Why is this an issue?

Even though this is syntactically correct, the ``++void++`` return type should not be used in the signature of a constructor. Indeed some developers might be confused by this syntax, believing that the constructor is in fact a standard function. 


=== Noncompliant code example

[source,flex]
----
public class Foo   
{
  public function Foo() : void
  {...}      
}
----


=== Compliant solution

[source,flex]
----
public class Foo   
{
  public function Foo()
  {...}      
}
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Remove the "void" return type from this "XXXX" constructor


'''
== Comments And Links
(visible only on this page)

=== on 31 Oct 2013, 15:29:43 Freddy Mallet wrote:
Is implemented by \http://jira.codehaus.org/browse/SONARPLUGINS-3224

endif::env-github,rspecator-view[]
