== Why is this an issue?

Shared coding conventions allow teams to collaborate efficiently. This rule checks that all namespace names match a provided regular expression.


=== Noncompliant code example

With the default regular expression ``++^[a-z][a-z_0-9]*$++``:

[source,text]
----
namespace Foo // Noncompliant
{
}
----


=== Compliant solution

[source,text]
----
namespace foo
{
}
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Rename "xxx" to match the regex "rrr".


=== Parameters

.format
****

----
^[a-z][a-z_0-9]*$
----

Regular expression used to check the namespace names against
****


endif::env-github,rspecator-view[]
