This rule raises an issue when CSS properties that affect text spacing (`line-height`, `letter-spacing`, `word-spacing`, or `margin`) are declared with `!important` in HTML style attributes.

== Why is this an issue?

Text spacing is crucial for readability, especially for users with dyslexia, visual processing disorders, or other reading difficulties. The WCAG 2.2 Text Spacing criterion requires that users be able to adjust spacing properties to meet their individual needs.

When text spacing properties are marked as `!important` in inline styles, they cannot be overridden by user stylesheets or browser extensions that help with accessibility. This creates a barrier for users who need to customize text spacing to read content effectively.

User agents and assistive technologies rely on the CSS cascade to allow users to apply their own styles. The `!important` declaration in inline styles has the highest specificity in CSS, making it impossible for users to override these values with their own accessibility-focused stylesheets.

This particularly affects users who need to:

* Increase line height for better readability
* Adjust letter spacing to reduce visual crowding
* Modify word spacing to improve text parsing
* Control paragraph margins for better content flow

=== What is the potential impact?

Users with reading difficulties, dyslexia, or visual processing disorders may be unable to customize text spacing to meet their accessibility needs. This can make content completely unreadable for some users, violating WCAG 2.2 Level AA requirements and potentially excluding users from accessing important information or services.

== How to fix it

Remove the `!important` declaration from text spacing properties in style attributes. This allows users to override these values with their own stylesheets when needed for accessibility.

=== Code examples

==== Noncompliant code example

[source,html,diff-id=1,diff-type=noncompliant]
----
<p style="line-height: 1.2 !important; letter-spacing: 0.1em !important;">Text content</p> <!-- Noncompliant -->
----

==== Compliant solution

[source,html,diff-id=1,diff-type=compliant]
----
<p style="line-height: 1.2; letter-spacing: 0.1em;">Text content</p>
----

== Resources

=== Documentation

 * WCAG 2.2 - Text Spacing - https://www.w3.org/TR/WCAG22/#text-spacing[Official WCAG 2.2 guidelines for text spacing accessibility requirements]

 * CSS Cascade and Inheritance - https://developer.mozilla.org/en-US/docs/Web/CSS/Cascade[MDN documentation on CSS cascade and how !important affects specificity]

=== Standards

 * WCAG 2.2-1.4.12-Text Spacing - https://www.w3.org/TR/WCAG22/#text-spacing[Web Content Accessibility Guidelines requirement for customizable text spacing]

=== Related rules

 * axe-avoid-inline-spacing - https://dequeuniversity.com/rules/axe/4.10/avoid-inline-spacing[Axe accessibility rule for avoiding inline spacing with !important]
