== Why is this an issue?

include::../description.adoc[]


=== Noncompliant code example

[source,kotlin]
----
"cc̈d̈d".replace(Regex("[c̈d̈]"), "X") // Noncompliant, print "XXXXXX" instead of expected "cXXd".
----


=== Compliant solution

[source,kotlin]
----
"cc̈d̈d".replace(Regex("c̈|d̈"), "X") // print "cXXd"
----
ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../highlighting.adoc[]

endif::env-github,rspecator-view[]
