== Why is this an issue?

include::../why.adoc[]

=== What is the potential impact?

include::../impact.adoc[]

== How to fix it

include::../how.adoc[]

=== Code examples

==== Noncompliant code example

[source,csharp,diff-id=1,diff-type=noncompliant]
----
void GetEpochTime()
{
    var epochTime = new DateTime(1970, 1, 1);
}
----

==== Compliant solution

[source,csharp,diff-id=1,diff-type=compliant]
----
void GetEpochTime()
{
    var epochTime = DateTime.UnixEpoch;
}
----

== Resources
=== Documentation

include::../documentation.adoc[]