include::../why-dotnet.adoc[]

include::../impact-dotnet.adoc[]

include::../how-dotnet.adoc[]

=== Code examples

==== Noncompliant code example

[source,csharp,diff-id=1,diff-type=noncompliant]
----
void GetEpochTime()
{
    var epochTime = new DateTime(1970, 1, 1);
}
----

==== Compliant solution

[source,csharp,diff-id=1,diff-type=compliant]
----
void GetEpochTime()
{
    var epochTime = DateTime.UnixEpoch;
}
----

include::../resources-dotnet.adoc[]