<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>When several lines must be inserted/updated into an internal table, instead of doing those changes line by line, mass operations should be used because they offer better performance  by design.</p>
</div>
<div class="paragraph">
<p>This rule raises an issue when a single line operation like <code>APPEND</code>, <code>CONCATENATE</code>, and <code>INSERT</code> is performed on an internal table in a loop.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-abap" data-lang="abap">LOOP AT ITAB1 INTO WA.
  APPEND WA TO ITAB2.
ENDLOOP.</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-abap" data-lang="abap">APPEND LINES OF ITAB1 TO ITAB2.</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove the nesting loop structure and use the "XXX LINES OF &#8230;&#8203; TO" mass operation clause</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_is_duplicated_by_s2244">is duplicated by: <a data-rspec-id="S2244" class="rspec-auto-link">S2244</a></h3>

</div>
<div class="sect2">
<h3 id="_on_30_apr_2014_153855_nicolas_peru_wrote">on 30 Apr 2014, 15:38:55 Nicolas Peru wrote:</h3>
<div class="paragraph">
<p>Not really sure of what to do with this rule in fact.</p>
</div>
<div class="paragraph">
<p>Here the example are not equivalent in functional terms : compliant copy STRUC into ITAB and non compliant iterate through a table  to do a write statement.</p>
</div>
<div class="paragraph">
<p>\[~ann.campbell.2] Maybe, you can provide some reference where the rule originated ?</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_30_apr_2014_155332_nicolas_peru_wrote">on 30 Apr 2014, 15:53:32 Nicolas Peru wrote:</h3>
<div class="paragraph">
<p>Rule originated from http://www.dsag.de/fileadmin/media/Leitfaeden/Leitfaden_Best_Practice_Guide_eng/files/assets/basic-html/page20.html</p>
</div>
</div>
</div>
</div>