<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>TypeScript provides enums to allow developers to define a set of named constants under a common type. These enum constants can be assigned values, but it is not mandatory to assign all of them. Those that are not assigned get default values in increasing order starting from zero. As a result, assigning only a subset of enum members can be misleading. For numeric enums in particular, that would create a gap in the numerical order, which could lead to unfortunate bugs. In addition, adding more members or moving around existing ones might change their values.</p>
</div>
<div class="paragraph">
<p>This rule raises an issue whenever an enum assigns a subset of its members, unless the first member only is assigned a numerical literal.</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-typescript" data-lang="typescript">enum Key {
    Up = 1,
    Down,
    Left,
    Right
}</code></pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Either assign a value to all enum members or none of them, or assign a number literal to the first enum member only.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-typescript" data-lang="typescript">enum Digit {
   Zero,
   One,
   ...
   Nine = 9
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-typescript" data-lang="typescript">enum Digit {
    Zero,
    One,
    ...
    Nine
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_noncompliant_code_example_2">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-typescript" data-lang="typescript">enum RGB {
   Red,
   Green,
   Blue = 0b000000000000000011111111
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution_2">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-typescript" data-lang="typescript">enum RGB {
   Red   = 0b111111110000000000000000,
   Green = 0b000000001111111100000000,
   Blue  = 0b000000000000000011111111
}</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p><a href="https://typescript-eslint.io/">typescript-eslint</a> - Rule <a href="https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/docs/rules/prefer-enum-initializers.mdx">prefer-enum-initializers</a></p>
</li>
<li>
<p>TypeScript Documentation - <a href="https://www.typescriptlang.org/docs/handbook/enums.html">Enums</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>