

// If you want to factorize the description uncomment the following line and create the file.
//include::../description.adoc[]

== Why is this an issue?

Numbers in JavaScript are stored as 64-bit floating-point. Numbers that can't be expressed in such a format get rounded before assigning them to a variable.

=== What is the potential impact?

When you are saving a number that is too big or contains a fraction that is not expressible in the 64-bit floating-point format, it gets rounded which will lead to errors when it will be used.

== How to fix it

=== Code examples

==== Noncompliant code example

[source,javascript]
----
const foo = 2312123211345545367
const bar = 0.123456789123456789
----

==== Compliant solution

[source,javascript]
----
const foo = BigInt(2312123211345545367)
// use a library like decimal.js for storing numbers containing lots of decimal digits
----

=== How does this work?

JavaScript numbers are stored in a double-precision 64-bit binary format IEEE 754. There are limits to what can be stored in this format both in huge numbers as in numbers with lots of decimal digits.
When attempting to store a number that can not be represented in this format into a variable, the number gets rounded before storage.

//=== Pitfalls

//=== Going the extra mile

== Resources

=== Documentation

* https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number[Number page on Mozilla documentation]

//=== Articles & blog posts
//=== Conference presentations
//=== Standards
