== Why is this an issue?

Redundant forward declarations simply clutter the code, and like any duplications, should be removed.


=== Noncompliant code example

[source,cpp]
----
struct S {
  // ...
};
// ...
struct S;  // Noncompliant
----


=== Compliant solution

[source,cpp]
----
struct S {
  // ...
};
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

endif::env-github,rspecator-view[]
