== Why is this an issue?

include::../description.adoc[]

=== Code examples

==== Noncompliant code example
Consider the following example that uses `.Capabilities.KubeVersion.GitVersion` which is deprecated. The recommendation is to use `.Capabilities.KubeVersion.Version` instead.

[source,text,diff-id=1,diff-type=noncompliant]
----
apiVersion: v1
kind: ConfigMap
metadata:
  name: {{ .Release.Name }}-configmap
data:
  myvalue: "Hello World"
  kubeVersion: {{ .Capabilities.KubeVersion.GitVersion }} # Noncompliant
----

==== Compliant solution

[source,text,diff-id=1,diff-type=compliant]
----
apiVersion: v1
kind: ConfigMap
metadata:
  name: {{ .Release.Name }}-configmap
data:
  myvalue: "Hello World"
  kubeVersion: {{ .Capabilities.KubeVersion.Version }} # Compliant
----

include::../see.adoc[]

* Go Documentation - https://pkg.go.dev/helm.sh/helm/v3/pkg/chartutil#KubeVersion.GitVersion[GitVersion deprecated]