This rule raises an issue when 5 or more commands are applied on a data frame.

== Why is this an issue?

The pandas library provides many ways to filter, select, reshape and modify a data frame.
All these transformations and instructions are part of the DataFrame API.
This makes it effortless to chain actions and perform several of them in one line of code:

[source,python]
----
import pandas as pd

joe = pd.read_csv("data.csv", dtype={'user_id':'str', 'name':'str'}).set_index("name").filter(like='jo', axis=0).head()
----

While this code is correct and concise, it is difficult to read, understand and maintain. 

One way to go around these difficulties would be to regroup complex transformation a data frame under a function and then use
``++pandas.pipe++`` method.

== How to fix it

To fix this issue refactor chains of instruction into a function that can be consumed by the ``++pandas.pipe++`` method.

=== Code examples

==== Noncompliant code example

[source,python,diff-id=1,diff-type=noncompliant]
----
import pandas as pd

def foo(df: pd.DataFrame):
  return df.set_index("name").filter(like='jo', axis=0).groupby("team")["salary"].mean().head()
----

==== Compliant solution

[source,python,diff-id=1,diff-type=compliant]
----
import pandas as pd

def select_joes(df):
  return df.set_index("name").filter(like='joe', axis=0)

def compute_mean_salary(df_):
  return df.groupby("team")["salary"].mean()

def foo(df: pd.DataFrame):
  return df.pipe(select_joes).pipe(compute_mean_salary).head()
----


//== Resources
//=== Documentation
