React expects a unique identifier `key` in each component to optimize rendering. Missing keys will negatively impact performance and can bring your application to a halt when there are enough elements.

Avoid Array indexes since they are not stable. Instead, use a unique attribute like an id.

== Noncompliant Code Example

[source,javascript]
----
const React = require('react');
class App extends React.Component {
  render() {
    <LoginButton />;
  }
}
----

== Compliant Solution

[source,javascript]
----
const React = require('react');
class App extends React.Component {
  render() {
    <LoginButton key="loginBtn" />;
  }
}
----

== See

* https://reactjs.org/docs/reconciliation.html#recursing-on-children - React API reference
