//// This rule is superseded by RSPEC-2583
//// Please consider implementing this latter instead.
include::../description.adoc[]

== Noncompliant Code Example

----
    If True Then ' Noncompliant
      DoSomething()
    End If

    If False Then ' Noncompliant
      DoSomethingElse()
    End If
----

== Compliant Solution

----
DoSomething();
' ...
----

include::../see.adoc[]
