include::../description.adoc[]

== Noncompliant code example

With the default regular expression ``++^[a-z]+(\.[a-z][a-z0-9]*)*$++``:

----
package org.Example {  // Noncompliant
...
}
----

== Compliant solution

----
package org.example { 
...
}
----
