This rule raises an issue when `unless` is used with an `else` clause, or when `if !` or `if not` is used for simple negative conditions without an `else` clause.

== Why is this an issue?

Ruby's `unless` statement is designed to make negative conditions more readable, but it can create confusion when misused.

When `unless` is used with an `else` clause, it creates double negative logic that is hard to understand. The `else` clause of an `unless` statement represents the positive case, which is counterintuitive. For example, `unless user.admin? ... else ...` means "if the user is not an admin, do this, otherwise (if they are an admin) do that." This mental gymnastics makes code harder to read and maintain.

On the other hand, when checking simple negative conditions without an `else` clause, using `if !` or `if not` instead of `unless` makes the code less idiomatic and slightly harder to read. Ruby's `unless` keyword was specifically created to handle these cases more elegantly.

The key principle is: use `unless` for simple negative conditions, but switch to `if !` when you need an `else` clause to maintain clarity.

=== What is the potential impact?

This issue affects code readability and maintainability. Confusing conditional logic can lead to bugs when developers misunderstand the intended behavior, especially during code reviews or when maintaining unfamiliar code. While not a security or performance issue, it can slow down development and increase the likelihood of logical errors.

== How to fix it

Replace `unless` with `else` clause with an equivalent `if` statement to eliminate double negative logic.

=== Code examples

==== Noncompliant code example

[source,ruby,diff-id=1,diff-type=noncompliant]
----
unless user.staff?
  raise SecurityError, 'not authorized'
else  # Noncompliant
  perform_admin_tasks
  generate_reports
end
----

==== Compliant solution

[source,ruby,diff-id=1,diff-type=compliant]
----
if user.staff?
  perform_admin_tasks
  generate_reports
else
  raise SecurityError, 'not authorized'
end
----

== Resources

=== Documentation

 * Ruby Style Guide - Conditional Expressions - https://rubystyle.guide/#unless-for-negatives[Official Ruby style guide recommendations for using unless statements]

 * Ruby Documentation - Control Expressions - https://docs.ruby-lang.org/en/master/syntax/control_expressions_rdoc.html[Ruby's official documentation on control flow expressions including unless]
