<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Parameter properties let you both create and initialize a member in one place, and omit an explicit member declaration and the assignment of the constructor parameter to the member. To use a parameter property, add an accessibility modifier or <code>readonly</code>, or both in front of the constructor parameter.</p>
</div>
<div class="listingblock">
<div class="content">
<pre>constructor(readonly name: string, private age: number) { // creates 2 initialized members "name" and "age"
}</pre>
</div>
</div>
<div class="paragraph">
<p>While this syntax is very concise it might be confusing for developers who are new to TypeScript.</p>
</div>
<div class="paragraph">
<p>Shared conventions allow teams to collaborate efficiently. This rule checks that either parameter properties are used everywhere or not at all.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="paragraph">
<p>Using the default parameter <code>forceParameterProperties</code> value <code>true</code>:</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-javascript" data-lang="javascript">class Person {
  name: number;
  constructor(name: string) {
    this.name = name; // Noncompliant, parameter property can be used
  }
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-javascript" data-lang="javascript">class Person {
  constructor(public name: string) {
  }
}</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="ulist">
<ul>
<li>
<p>Usage of parameter properties has been disallowed.
or</p>
</li>
<li>
<p>Use parameter properties instead of assigning to members in the constructor body.</p>
</li>
</ul>
</div>
</div>
<div class="sect2">
<h3 id="_parameters">Parameters</h3>
<div class="sidebarblock">
<div class="content">
<div class="title">forceParameterProperties</div>
<div class="listingblock">
<div class="content">
<pre>true</pre>
</div>
</div>
<div class="paragraph">
<p>Forces to use parameter properties when true, forbids their use when false</p>
</div>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="ulist">
<ul>
<li>
<p>parameter property declaration
or</p>
</li>
<li>
<p>assignment statement</p>
</li>
</ul>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_16_feb_2018_142906_ann_campbell_wrote">on 16 Feb 2018, 14:29:06 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~elena.vilchik] message and highlighting are both needed. Since you&#8217;re (currently) the likely implementer, this may seem redundant but it&#8217;s good to have a record. And you might not end up as the implementer after all. :-)</p>
</div>
<div class="paragraph">
<p>Otherwise, nicely done!</p>
</div>
</div>
</div>
</div>