== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,csharp]
----
public class MyClass // Noncompliant
{
  private MyClass() { ... }
}
----


=== Compliant solution

[source,csharp]
----
public class MyClass 
{
  public MyClass() { ... }
}
----


include::../exceptions.adoc[]


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../highlighting.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
