This rule raises an issue when a SOQL query contains multiple clauses on a single line, making it difficult to read and understand.

== Why is this an issue?

SOQL queries are an essential part of Apex development, and they can quickly become complex with multiple clauses like SELECT, FROM, WHERE, ORDER BY, and LIMIT. When these clauses are written on a single line, the query becomes harder to read, understand, and maintain.

Poor formatting makes it difficult to:

* Quickly identify the different parts of the query
* Review changes during code reviews
* Debug issues when queries don't return expected results
* Maintain consistency across the codebase

Well-formatted SOQL queries follow a structured approach where each logical section is placed on its own line with proper indentation. This makes the query's structure immediately clear and helps developers understand what data is being retrieved and under what conditions.

Consistent formatting also improves team collaboration, as all developers can quickly understand and modify queries written by their colleagues.

=== What is the potential impact?

Poor SOQL formatting reduces code readability and maintainability. While it doesn't directly impact application functionality, it can slow down development, make code reviews less effective, and increase the likelihood of introducing bugs when modifying queries.

== How to fix it

Break the SOQL query into multiple lines, placing each logical clause (SELECT, FROM, WHERE, etc.) on its own line. Align field names and use consistent indentation for better readability.

=== Code examples

==== Noncompliant code example

[source,apex,diff-id=1,diff-type=noncompliant]
----
List<Account> accounts = [SELECT Id, Name, BillingCity FROM Account WHERE Name LIKE '%ApexHours%' ORDER BY Name LIMIT 10]; // Noncompliant
----

==== Compliant solution

[source,apex,diff-id=1,diff-type=compliant]
----
List<Account> accounts = [
    SELECT Id,
           Name,
           BillingCity
    FROM Account
    WHERE Name LIKE '%ApexHours%'
    ORDER BY Name
    LIMIT 10
];
----

== Resources

=== Documentation

 * SOQL and SOSL Queries - https://developer.salesforce.com/docs/atlas.en-us.apexcode.meta/apexcode/langCon_apex_SOQL.htm[Official Salesforce documentation on SOQL queries in Apex]

 * Apex Code Best Practices - https://developer.salesforce.com/docs/atlas.en-us.apexcode.meta/apexcode/apex_bestpractices.htm[Salesforce best practices for writing clean and maintainable Apex code]
