This rule raises an issue when the `tf.linalg.inv` function argument doesn't meet the requirements.

== Why is this an issue?
The `tf.linalg.inv` is a function used to compute the inverse of a matrix. The argument value should meet the following requirements:

- the matrix should be non-singular
- the matrix should be square
- the matrix element should be one of the following types: `double`, `float`, `half`, `complex64`, `complex128` 

Otherwise, it will lead to `InvalidArgumentError`

== How to fix it

=== Code examples

==== Noncompliant code example

[source,python,diff-id=1,diff-type=noncompliant]
----
import tensorflow as tf

x = tf.constant([1., 2., 3.])
y = tf.linalg.inv(x)  # Noncompliant: InvalidArgumentError: Input matrix must be square
----

==== Compliant solution

[source,python,diff-id=1,diff-type=compliant]
----
import tensorflow as tf

x = tf.constant([[1., 2.], [3., 4.]])
y = tf.linalg.inv(x)  # OK
----

//=== How does this work?

//=== Pitfalls

//=== Going the extra mile


== Resources
=== Documentation
* Tensorflow documentation - https://www.tensorflow.org/api_docs/python/tf/linalg/inv[tf.linalg.inv]
//=== Articles & blog posts
//=== Conference presentations
//=== Standards
//=== External coding guidelines
//=== Benchmarks
