== How to fix it in go-cmd/cmd

=== Code examples

include::../../common/fix/code-rationale.adoc[]

==== Noncompliant code example

[source,go,diff-id=2,diff-type=noncompliant]
----
import (
	"github.com/go-cmd/cmd"
)

func runCommand(w http.ResponseWriter, r *http.Request) {
	command := r.FormValue("command")
	c := cmd.NewCmd(command)
	status := <-c.Start()
	if status.Error == nil {
		fmt.Fprintf(w, "Command output: %s\n", status.Stdout);
	}
}
----

==== Compliant solution

[source,go,diff-id=2,diff-type=compliant]
----
import (
	"github.com/go-cmd/cmd"
)

func runCommand(w http.ResponseWriter, r *http.Request) {
	command := r.FormValue("command")
	var allowedCommands = map[string]bool{
		"whoami": true,
	}
	if _, ok := allowedCommands[command]; !ok {
		http.Error(w, "Invalid command", http.StatusForbidden)
		return
	}
	c := cmd.NewCmd(command)
	status := <-c.Start()
	if status.Error == nil {
		fmt.Fprintf(w, "Command output: %s\n", status.Stdout);
	}
}
----

=== How does this work?

include::../../common/fix/introduction.adoc[]

include::../../common/fix/pre-approved-list.adoc[]
