== How to fix it in Go Standard Library

=== Code examples

include::../../common/fix/code-rationale.adoc[]

==== Noncompliant code example

[source,go,diff-id=1,diff-type=noncompliant]
----
import (
	"os/exec"
)

func runCommand(w http.ResponseWriter, r *http.Request) {
	command := r.FormValue("command")
	c := exec.Command(command)
	output, err := c.CombinedOutput()
	if err != nil {
		fmt.Fprintf(w, "Command output: %s\n", output);
	}
}
----

==== Compliant solution

[source,go,diff-id=1,diff-type=compliant]
----
import (
	"os/exec"
)

func runCommand(w http.ResponseWriter, r *http.Request) {
	command := r.FormValue("command")
	var allowedCommands = map[string]bool{
		"whoami": true,
	}
	if val, ok := allowedCommands[command]; !ok {
		http.Error(w, "Invalid command", http.StatusForbidden)
		return
	}
	c := exec.Command(command)
	output, err := c.CombinedOutput()
	if err != nil {
		fmt.Fprintf(w, "Command output: %s\n", output);
	}
}
----

=== How does this work?

include::../../common/fix/introduction.adoc[]

include::../../common/fix/pre-approved-list.adoc[]
