There's no point in creating an array solely for the purpose of passing it to a ``ParamArray`` parameter. Simply pass the elements directly. They will be consolidated into an array automatically. 

== Noncompliant Code Example

[source,vbnet]
----
Class SurroundingClass
    Public Sub CallTheThing()
        DoTheThing(New String() {"s1", "s2"}) ' Noncompliant: unnecessary
        DoTheThing(New String(11) {}) ' Compliant
    End Sub

    Public Sub DoTheThing(ParamArray args As String())
        ' Do something
    End Sub
End Class
----

== Compliant Solution

[source,vbnet]
----
Class SurroundingClass
    Public Sub CallTheThing()
        DoTheThing("s1", "s2")
        DoTheThing(New String(11) {})
    End Sub

    Public Sub DoTheThing(ParamArray args As String())
        ' Do something
    End Sub
End Class
----
