When two functions have the same implementation, either it was a mistake - something else was intended - or the duplication was intentional, but may be confusing to maintainers. In the latter case, one implementation should invoke the other.

== Noncompliant Code Example

----
func calculate() {
  doTheThing()
  doOtherThing()
}

func doEverything() {  // Noncompliant
  doTheThing()
  doOtherThing()
}
----

== Compliant Solution

----
func calculate() {
  doTheThing()
  doOtherThing()
}

func doEveryting() {
  calculate()
}
----

== Exceptions

Methods with fewer than 2 statements are ignored.

ifdef::env-github,rspecator-view[]
'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
