== Why is this an issue?

When two functions have the same implementation, either it was a mistake - something else was intended - or the duplication was intentional, but may be confusing to maintainers. In the latter case, one implementation should invoke the other.

=== Noncompliant code example

[source,swift,diff-id=1,diff-type=noncompliant]
----
func calculate() {
  doTheThing()
  doOtherThing()
}

func doEverything() {  // Noncompliant
  doTheThing()
  doOtherThing()
}
----

=== Compliant solution

[source,swift,diff-id=1,diff-type=compliant]
----
func calculate() {
  doTheThing()
  doOtherThing()
}

func doEveryting() {
  calculate()
}
----

=== Exceptions

Methods with fewer than 2 statements are ignored.

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../highlighting.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
