<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Returning the unit type from a closure expecting an <code>Ord</code> type is likely a mistake caused by an extra semi-colon. This is because the unit type implements <code>Ord</code> but doesn&#8217;t serve the intended sorting purpose, making the code misleading and potentially incorrect.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-rust" data-lang="rust">let mut twins = vec![(1, 1), (2, 2)];
twins.sort_by_key(|x| { x.1; }); // Noncompliant: Closure returns unit type due to unnecessary semi-colon.</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-rust" data-lang="rust">let mut twins = vec![(1, 1), (2, 2)];
twins.sort_by_key(|x| x.1); // Compliant: Closure correctly returns an `Ord` type.</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Clippy Lints - <a href="https://rust-lang.github.io/rust-clippy/master/index.html#unit_return_expecting_ord" class="bare">https://rust-lang.github.io/rust-clippy/master/index.html#unit_return_expecting_ord</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>