== Why is this an issue?

include::../why.adoc[]

=== What is the potential impact?

include::../impact.adoc[]

== How to fix it

include::../how.adoc[]

=== Code examples

==== Noncompliant code example

[source,vbnet,diff-id=1,diff-type=noncompliant]
----
Private Sub GetEpochTime()
    Dim epochTime = New DateTime(1970, 1, 1)
End Sub
----

==== Compliant solution

[source,vbnet,diff-id=1,diff-type=compliant]
----
Private Sub GetEpochTime()
    Dim epochTime = DateTime.UnixEpoch
End Sub
----

== Resources
=== Documentation

include::../documentation.adoc[]