include::../why-dotnet.adoc[]

include::../impact-dotnet.adoc[]

include::../how-dotnet.adoc[]

=== Code examples

==== Noncompliant code example

[source,vbnet,diff-id=1,diff-type=noncompliant]
----
Private Sub GetEpochTime()
    Dim epochTime = New DateTime(1970, 1, 1)
End Sub
----

==== Compliant solution

[source,vbnet,diff-id=1,diff-type=compliant]
----
Private Sub GetEpochTime()
    Dim epochTime = DateTime.UnixEpoch
End Sub
----

include::../resources-dotnet.adoc[]