<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Modifying SAP standard tables via an Open SQL statement can result in data corruption and unexpected behavior. This is because a direct modification bypasses all validation mechanisms.</p>
</div>
<div class="paragraph">
<p>Instead, the use of standard functions or SAP Business Add-Ins (BAdIs) is recommended.</p>
</div>
<div class="paragraph">
<p>This rule raises an issue on Open SQL statements <code>INSERT</code>, <code>UPDATE</code>,Â <code>MODIFY</code>, or <code>SELECT ... FOR UPDATE</code> which target an SAP table, i.e. a table with a name starting with a character between "A" and "X", case insensitive.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-abap" data-lang="abap">SELECT my_col FROM my_system_table WHERE my_id = 123 FOR UPDATE. " Noncompliant

SELECT my_col FROM z_custom_table WHERE my_id = 123 FOR UPDATE. " Ignored; modifies a custom table</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Replace this Open SQL statement with a standard function call or an SAP BAdI</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>The table name</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_2_jan_2019_145643_ann_campbell_wrote">on 2 Jan 2019, 14:56:43 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~nicolas.harraudeau] it wasn&#8217;t clear to me until I got to the code sample that the targeted characters are case-insensitive.</p>
</div>
<div class="paragraph">
<p>Also, is the second line in the code sample intended to be the Compliant solution, or just an example of something that&#8217;s ignored by the rule? If the former, it should be moved into a separate, "Compliant Solution" sample, where no "compliant" comment is needed because everything in it is compliant by definition. (And really, even in the Noncompliant example I wouldn&#8217;t use a "compliant" comment.)</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_4_jan_2019_100023_nicolas_harraudeau_wrote">on 4 Jan 2019, 10:00:23 Nicolas Harraudeau wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] Thanks for the review. The second example is just something ignored by the rule. I changed the "Compliant" comment into something more explicit. Let me know if it is still not clear.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_7_jan_2019_085539_alexandre_gigleux_wrote">on 7 Jan 2019, 08:55:39 Alexandre Gigleux wrote:</h3>
<div class="paragraph">
<p>\[~nicolas.harraudeau]</p>
</div>
<div class="ulist">
<ul>
<li>
<p>I renamed "OpenSQL" and "OpenSql" into "Open SQL" because it looks to be the format used by SAP: https://help.sap.com/doc/saphelp_nw70/7.0.31/en-US/fc/eb3969358411d1829f0000e829fbfe/content.htm?no_cache=true</p>
</li>
<li>
<p>Could you add a link here, as a comment, to SAP Badis, because I can&#8217;t find what it is and I would like to learn more about it? Thx</p>
</li>
</ul>
</div>
</div>
</div>
</div>