This rule checks that the specified child tag does not appear as a direct child of the specified parent.


== Noncompliant Code Example

Assuming a parent/child combination of ``++head++``/``++body++``:

----
<head>
  ...
  <body>  <!-- Noncompliant -->
    ...
  </body>
</head>
----


== Compliant Solution

----
<head>
  ...
</head>
----


ifdef::env-github,rspecator-view[]
'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
