== Why is this an issue?

This rule checks that the specified child tag does not appear as a direct child of the specified parent.


=== Noncompliant code example

Assuming a parent/child combination of ``++head++``/``++body++``:

[source,html]
----
<head>
  ...
  <body>  <!-- Noncompliant -->
    ...
  </body>
</head>
----


=== Compliant solution

[source,html]
----
<head>
  ...
</head>
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

include::parameters.adoc[]

'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
