== Why is this an issue?

This rule checks that the specified child tag does not appear as a direct child of the specified parent.


=== Noncompliant code example

Assuming a parent/child combination of ``++head++``/``++body++``:

[source,html]
----
<head>
  ...
  <body>  <!-- Noncompliant -->
    ...
  </body>
</head>
----


=== Compliant solution

[source,html]
----
<head>
  ...
</head>
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Remove this "XXX" tag; it is not a legal child of "YYY".


=== Parameters

.child
****

Name of the child element
****
.parent
****

Name of the parent element
****


'''
== Comments And Links
(visible only on this page)

=== deprecates: S3734

=== on 25 Aug 2014, 06:44:41 Freddy Mallet wrote:
Hi @Ann, is there any default values for the properties of this rule ? Thanks

=== on 25 Aug 2014, 20:17:50 Ann Campbell wrote:
This is the documentation (& cleanup of the format) of an existing Web rule.

endif::env-github,rspecator-view[]
