This rule raises an issue when a class declares a traditional `public static void main(String[] args)` method but does not use the `args` parameter, and the method could be simplified to an instance main method.

== Why is this an issue?

Java 25 introduced instance main methods (JEP 512), allowing simpler entry points. The traditional `public static void main(String[] args)` includes unnecessary complexity when `args` is unused:

* The `public` modifier is not required
* The `static` modifier forces static methods or object instantiation
* The `String[] args` parameter is often unused

The instance main method `void main()` removes this boilerplate while maintaining compatibility.

=== What is the potential impact?

Using the traditional static main method when the simpler instance main method would suffice adds unnecessary complexity and boilerplate that serves no functional purpose.

== How to fix it

Replace with an instance main method when String[] parameter is unused.

=== Code examples

==== Noncompliant code example

[source,java,diff-id=1,diff-type=noncompliant]
----
public class HelloWorld {
    public static void main(String[] args) { // Noncompliant
        System.out.println("Hello, World!");
    }
}
----

==== Compliant solution

[source,java,diff-id=1,diff-type=compliant]
----
class HelloWorld {
    void main() {
        System.out.println("Hello, World!");
    }
}
----

== Resources

=== Documentation

 * JEP 512: Compact Source Files and Instance Main Methods - https://openjdk.org/jeps/512[Official OpenJDK documentation for the instance main methods feature]

=== Related rules

 * RSPEC-3051 - https://rules.sonarsource.com/java/RSPEC-3051/[Unused method parameters should be removed]
