== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,vbnet]
----
Class Person
    Private age As Integer

    <Pure> ' Noncompliant. The method makes a state change
    Private Sub ConfigureAge(ByVal age As Integer)
        Me.age = age
    End Sub

    <Pure>
    Private Sub WriteAge() ' Noncompliant
        Console.WriteLine(Me.age)
    End Sub

End Class
----

=== Compliant solution

[source,vbnet]
----
Class Person
    Private age As Integer

    Private Sub ConfigureAge(ByVal age As Integer)
        Me.age = age
    End Sub
    
    <Pure>
    Private Function Age() As Integer
        Return Me.age
    End Function

    ' or remove Pure attribute from the method

    Private Sub WriteAge()
        Console.WriteLine(Me.age)
    End Sub

End Class
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../highlighting.adoc[]

endif::env-github,rspecator-view[]
