== Why is this an issue?

include::description.adoc[]

=== Noncompliant code example

[source,text]
----
int divide(int numerator, int denominator) {
  return numerator / denominator;              // FIXME denominator value might be  0
}
----


include::see.adoc[]

