== Why is this an issue?

Shared naming conventions allow teams to collaborate efficiently. This rule checks that all job names match a provided regular expression.

=== Noncompliant code example

With the regular expression ``++^[A-Z][A-Z0-9]*$++``:

[source,jcl,diff-id=1,diff-type=noncompliant]
----
//* Noncompliant
//$JOB01  JOB
----

=== Compliant solution

[source,jcl,diff-id=1,diff-type=compliant]
----
//JOB01  JOB
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

=== Parameters

.format
****

----
^[A-Z][A-Z0-9]*$
----

Regular expression used to check the job names against.
****


endif::env-github,rspecator-view[]
