include::../description.adoc[]

== Noncompliant Code Example

[source,php]
----
"/foo()/"  // Noncompliant, will match only 'foo'
----

== Compliant Solution

[source,php]
----
"/foo\(\)/"  // Matches 'foo()'
----

include::../implementation.adoc[]
