include::../description.adoc[]

== Noncompliant Code Example

----
"/foo()/"  // Noncompliant, will match only 'foo'
----

== Compliant Solution

----
"/foo\(\)/"  // Matches 'foo()'
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

== Message

Remove this empty group.

== Highlighting

The empty group should be highlighted.

'''

endif::env-github,rspecator-view[]