== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,php]
----
"/foo()/"  // Noncompliant, will match only 'foo'
----

=== Compliant solution

[source,php]
----
"/foo\(\)/"  // Matches 'foo()'
----

include::../implementation.adoc[]
