=== Noncompliant code example

[source,text]
----
public boolean seek(int [] input) {
  int target = 32;  // Noncompliant
  for (int i : input) {
    if (i == target) {
      return true;
    }
  }
  return false;
}
----
