== Why is this an issue?

Marking a variable that is unchanged after initialization ``++final++`` is an indication to future maintainers that "no this isn't updated, and it's not supposed to be". ``++final++`` should be used in these situations in the interests of code clarity.


=== Noncompliant code example

[source,text]
----
public boolean seek(int [] input) {
  int target = 32;  // Noncompliant
  for (int i : input) {
    if (i == target) {
      return true;
    }
  }
  return false;
}
----


=== Compliant solution

[source,text]
----
public boolean seek(int [] input) {
  final int target = 32;
  for (int i : input) {
    if (i == target) {
      return true;
    }
  }
  return false;
}
----

