<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>In Blazor, the <a href="https://learn.microsoft.com/en-us/aspnet/core/blazor/fundamentals/routing#catch-all-route-parameters">catch-all route parameters</a> allows to capture the rest of the URL path.</p>
</div>
<div class="paragraph">
<p>The catch-all route parameters should be at the end of the URL. Otherwise, the route template will be considered invalid and a runtime error will be raised:</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-text" data-lang="text">Invalid template 'route/{*catchAllRoute}/another/segment'. A catch-all parameter can only appear as the last segment of the route template.
...</code></pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Ensure the catch-all route parameter is at the end of the URL.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-csharp" data-lang="csharp">@page "/route/{*catchAllRoute}/another/segment" // Noncompliant

@code {
    [Parameter]
    public string CatchAllRoute { get; set; }
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-csharp" data-lang="csharp">@page "/route/another/segment/{*catchAllRoute}" // Compliant

@code {
    [Parameter]
    public string CatchAllRoute { get; set; }
}</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Microsoft Learn - <a href="https://learn.microsoft.com/en-us/aspnet/core/blazor/fundamentals/routing#catch-all-route-parameters">ASP.NET Core Blazor routing and navigation</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>