The ``++await++`` keyword can only be used on https://docs.python.org/3/library/asyncio-task.html#awaitables["Awaitable" objects]. Python has three types of awaitables: Future, Task and Coroutines. Calling await on any other object will raise a ``++TypeError++``.


== Noncompliant Code Example

----
import asyncio

def myfunction():
    print("myfunction")

async def otherfunction():
    await myfunction()  # Noncompliant. myfunction is not marked as "async"

asyncio.run(otherfunction())
----


== Compliant Solution

----
import asyncio

async def myfunction():
    print("myfunction")

async def otherfunction():
    await myfunction()

asyncio.run(otherfunction())
----


== See

* https://docs.python.org/3/library/asyncio-task.html[Python documentation - Coroutines and Tasks]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

include::highlighting.adoc[]

endif::env-github,rspecator-view[]
