include::../common/why.adoc[]

include::../common/how.adoc[]

=== Code examples

==== Noncompliant code example

[source,html,diff-id=1,diff-type=noncompliant]
----
<img src="sunrise.jpg" alt="image of a sunrise" /> <!-- Noncompliant: "Image, image of a sunrise" -->
----

==== Compliant solution

[source,html,diff-id=1,diff-type=compliant]
----
<img src="sunrise.jpg" alt="a sunrise over a mountain range" />
----

include::../common/resources.adoc[]
