``++Select Case++`` statements are useful when there are many different cases depending on the value of the same expression.

For just one or two cases however, the code will be more readable with ``++If++`` statements.

== Noncompliant Code Example

----
Select Case number
    Case 1 To 5
        MsgBox("Between 1 and 5, inclusive")
    Case Else
        MsgBox("Not between 1 and 5, inclusive")
End Select
----

== Compliant Solution

----
If ( number >= 1 And number <= 5 ) then
  MsgBox("Between 1 and 5, inclusive")
Else
  MsgBox("Not between 1 and 5, inclusive")
End If
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
