=== Code examples

==== Noncompliant code example

[source,csharp,diff-id=1,diff-type=noncompliant]
----
public IEnumerable<int> GetSortedFilteredList(IEnumerable<int> data) =>
    data.OrderBy(x => x).Where(x => x % 2 == 0);
----

==== Compliant solution

[source,csharp,diff-id=1,diff-type=compliant]
----
public IEnumerable<int> GetSortedFilteredList(IEnumerable<int> data) =>
     data.Where(x => x % 2 == 0).OrderBy(x => x);
----