include::../description.adoc[]

== Noncompliant Code Example

----
if (x == 0) {
  doSomething()
} else if (x == 1) {
  doSomethingElse()
} 
----

== Compliant Solution

----
if (x == 0) {
  doSomething()
} else if (x == 1) {
  doSomethingElse()
} else {
  throw IllegalStateException()
}
----

== Exceptions

When all branches of an ``if``-``else if`` end with ``return``, ``break`` or ``throw``, the code that comes after the ``if`` implicitly behaves as if it was in an ``else`` clause. This rule will therefore ignore that case.

include::../see.adoc[]
