<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Where a data value is to be tested against zero then the test should be made explicit. The exception to this rule is when data represents a Boolean value, even though in C this will in practice be an integer.</p>
</div>
<div class="paragraph">
<p>This rule is in the interests of clarity, and makes clear the distinction between integers and logical values.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">if ( x ) // Noncompliant, unless x is effectively Boolean data</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">if ( x == 0) // Compliant solution</code></pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p>MISRA C:2004, 13.2 - Tests of a value against zero should be made explicit, unless the operand is effectively Boolean.</p>
</li>
</ul>
</div>
<hr>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Explicitly test this expression against zero.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_is_related_to_s872">is related to: <a data-rspec-id="S872" class="rspec-auto-link">S872</a></h3>

</div>
<div class="sect2">
<h3 id="_on_22_jan_2019_114546_loïc_joly_wrote">on 22 Jan 2019, 11:45:46 Loïc Joly wrote:</h3>
<div class="paragraph">
<p>This rule is a pure Misra-C rule, and should be retargeted as such. <a data-rspec-id="S811" class="rspec-auto-link">RSPEC-811</a> &amp; <a data-rspec-id="S812" class="rspec-auto-link">RSPEC-812</a>, together, cover the same situation for Misra-C&#43;&#43;.</p>
</div>
<div class="paragraph">
<p>We need a SonarSource rule that corers the same situation, but with some differences:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>One rule for both languages, that covers all cases</p>
</li>
<li>
<p>Allow also pointer validity checks <code>T* p = f(); if(p) ...</code>?</p>
</li>
<li>
<p>Allow class types with explicit conversion to bool <code>unique_ptr&lt;T&gt; p = f(); if (p) ...</code>?</p>
</li>
<li>
<p>&#8230;&#8203;</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>We will wait to have started a real Misra initiative to do this.</p>
</div>
</div>
</div>
</div>