<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Looping over <code>x.next()</code> is misleading and can introduce subtle bugs. <code>x.next()</code> returns an <code>Option</code> which should be handled properly. When used in a loop, <code>Option</code> implements <code>IntoIterator</code>, resulting in unexpected behavior where only a single element or <code>None</code> might be processed, leading to difficult-to-diagnose errors.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-rust" data-lang="rust">for value in iterator.next() {
    // Noncompliant: Looping directly over `iterator.next()`
    println!("{value}");
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-rust" data-lang="rust">for value in iterator {
    // Compliant: Looping directly over `iterator`
    println!("{value}");
}</code></pre>
</div>
</div>
<div class="paragraph">
<p>Alternatively, handling the Option:</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-rust" data-lang="rust">if let Some(value) = iterator.next() {
    // Process the single value
    println!("{value}");
}</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Clippy Lints - <a href="https://rust-lang.github.io/rust-clippy/master/index.html#iter_next_loop" class="bare">https://rust-lang.github.io/rust-clippy/master/index.html#iter_next_loop</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>