This rule raises an issue when the expression used in an `except` statement is not a class deriving from `BaseException` nor a tuple of such classes.

== Why is this an issue?

In Python 3, attempting to catch in an `except` statement an object which does not derive from `BaseException` will raise a `TypeError`. 

In order to catch multiple exceptions in an `except` statement, a `tuple` of exception classes should be provided. Trying to catch multiple exceptions with a `list` or a `set` will raise a `TypeError`.

*Note*: __In Python 2 it is possible to raise old-style classes, which do not inherit from BaseException.__

== How to fix it

Make sure the expression used in an `expect` statement is an exception which derives from `BaseException` or a tuple of such exceptions.

=== Code examples

==== Noncompliant code example

[source,python]
----
class CustomException:
    """An Invalid exception class."""

try:
    ...
except CustomException:  # Noncompliant: this custom exception does not derive from Exception or BaseException.
    print("exception")
except (None, list()):  # Noncompliant: None and list() are not valid exceptions deriving from BaseException.
    print("exception")

try:
    ...
except [TypeError, ValueError]:  # Noncompliant: lists are not accepted.
    print("exception")
except {TypeError, ValueError}:  # Noncompliant: sets are not accepted.
    print("exception")
----


==== Compliant solution

[source,python]
----
class MyError(Exception):
    pass

try:
    ...
except (MyError, TypeError):
    print("exception")
----

=== Pitfalls

When creating a custom Exception class, note that it should inherit from `Exception`, not `BaseException`. `Exception` allows people to catch all exceptions except the ones explicitly asking the interpreter to stop, such as https://docs.python.org/3/library/exceptions.html#KeyboardInterrupt[`KeyboardInterrupt`] and https://docs.python.org/3/library/exceptions.html#GeneratorExit[`GeneratorExit`] which are not errors per se but program interruption. See https://www.python.org/dev/peps/pep-0352/#exception-hierarchy-changes[PEP 352] and https://docs.python.org/3/library/exceptions.html#exception-hierarchy[Exception Hierarchy] for more information.

== Resources

=== Documentation 

* https://docs.python.org/3/tutorial/errors.html[Errors and Exceptions]
* https://docs.python.org/3/reference/compound_stmts.html#except[The `try` statement]
* https://docs.python.org/3/library/exceptions.html#exception-hierarchy[Exception hierarchy]

=== Standards

* https://www.python.org/dev/peps/pep-0352/#exception-hierarchy-changes[PEP 352 - Required Superclass for Exceptions]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

* Change this expression to be a class deriving from BaseException or a tuple of such classes.


=== Highlighting

The expression which does not resolve in a valid exception class or a tuple of such classes.


'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
