<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Functions and methods with <code>void</code> return type should not return <code>null</code>.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Dart allows to <code>return null;</code> from a function or a method with <code>void</code> return type. That happens because, from a technical standpoint, unlike other languages like Java, every function in Dart returns a value: if you don&#8217;t return a value, <code>null</code> will be returned.</p>
</div>
<div class="paragraph">
<p>From a semantical standpoint, however, this can be misleading, as it may suggest that the function does return an intended value, which is not the case.</p>
</div>
<div class="paragraph">
<p>It is better to consistently use <code>return;</code> and make it clear that in fact the function does not return any intended value.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Replace <code>return null;</code> with <code>return;</code>.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-dart" data-lang="dart">void logIf(bool condition, String message) {
  if (!condition) {
    return null; // Noncompliant
  }

  print(message);
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-dart" data-lang="dart">void logIf(bool condition, String message) {
  if (!condition) {
    return;
  }

  print(message);
}</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Dart Docs - <a href="https://dart.dev/tools/linter-rules/avoid_returning_null_for_void">Dart Linter rule - avoid_returning_null_for_void</a></p>
</li>
</ul>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="ulist">
<ul>
<li>
<p>Don&#8217;t return 'null' from a function with a return type of 'void'.</p>
</li>
<li>
<p>Don&#8217;t return 'null' from a method with a return type of 'void'.</p>
</li>
</ul>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="ulist">
<ul>
<li>
<p>In the case of a function or method with body: the <code>return null;</code> expression, including <code>;</code>.</p>
</li>
<li>
<p>In the case of an arrow function or an arrow method: the <code>&#8658; null;</code> expression, including <code>;</code>.</p>
</li>
</ul>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
</div>
</div>