<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Having an immutable condition in a while loop can result in an infinite loop as the condition will never change, causing the loop to run indefinitely.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-rust" data-lang="rust">let i = 0;
while i &gt; 10 {
    println!("let me loop forever!"); // Noncompliant: 'i' does not change within the loop body
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-rust" data-lang="rust">let mut i = 0;
while i &lt;= 10 {
    println!("looping: {}", i); // Compliant: 'i' gets incremented within the loop body
    i += 1;
}</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Clippy Lints - <a href="https://rust-lang.github.io/rust-clippy/master/index.html#while_immutable_condition" class="bare">https://rust-lang.github.io/rust-clippy/master/index.html#while_immutable_condition</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>