When implementing React Components, their `render` method of should always return, tipically using JSX. The rule will raise an issue when it's missing. If the component has nothing to render, the method should return `null`.

== Noncompliant Code Example

[source,javascript]
----
const React = require('react');
class App extends React.Component {
  render() {
    <div>Contents</div>;
  }
}
----

== Compliant Solution

[source,javascript]
----
const React = require('react');
class App extends React.Component {
  render() {
    return <div>Contents</div>;
  }
}
----

== See

* https://reactjs.org/docs/react-component.html#render[render()] - React API Reference