The `render` should always return. The rule will raise an issue when it's missing.

== Noncompliant Code Example

[source,javascript]
----
const createReactClass = require('create-react-class');
const App1 = createReactClass({
  render() {
    <div>Contents</div>; // Without return, it's an empty statement
  }
});

const React = require('react');
class App2 extends React.Component {
  render() {
    <div>Contents</div>; // Without return, it's an empty statement
  }
}
----

== Compliant Solution

[source,javascript]
----
const createReactClass = require('create-react-class');
var App1 = createReactClass({
  render() {
    return <div>Contents</div>;
  }
});

const React = require('react');
class App2 extends React.Component {
  render() {
    return <div>Contents</div>;
  }
}
----

== See

* https://reactjs.org/docs/react-component.html#render[render()] - React API Reference