React components describe an independent and reusable piece of UI. To define it one can declare a function (functional component) or a class (class component). When implementing a class component, a `render` method is required to define the DOM structure, typically with JSX syntax. Thus having a `render` method without a single `return` statement is usually a mistake. If it's required that the component renders to nothing, the method should return `null`.

== Noncompliant Code Example

[source,javascript]
----
const React = require('react');
class App extends React.Component {
  render() {
    <div>Contents</div>;
  }
}
----

== Compliant Solution

[source,javascript]
----
const React = require('react');
class App extends React.Component {
  render() {
    return <div>Contents</div>;
  }
}
----

== See

* https://reactjs.org/docs/react-component.html#render[render()] - React API Reference