When writing the `render` method in a component it is easy to forget to return the JSX content. This rule will warn if the `return` statement is missing.

== Noncompliant Code Example

[source,javascript]
----
var Hello = createReactClass({
  render() {
    <div>Hello</div>;
  }
});

class Hello extends React.Component {
  render() {
    <div>Hello</div>;
  }
}
----

== Compliant Solution

[source,javascript]
----
var Hello = createReactClass({
  render() {
    return <div>Hello</div>;
  }
});

class Hello extends React.Component {
  render() {
    return <div>Hello</div>;
  }
}
----

== See

* https://github.com/jsx-eslint/eslint-plugin-react/blob/master/docs/rules/require-render-return.md[Enforce ES5 or ES6 class for returning value in render function (react/require-render-return)
] - ESLint rule