<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Null statements should not normally be included deliberately, but where they are used, they shall appear on a line by themselves. White-space characters may precede the null statement to preserve indentation. If a comment follows the null statement, then at least one white-space character shall separate the null statement from the comment. The use of a white-space character to separate the null statement from any following comment is required on the grounds that it provides an important visual cue to reviewers. Following this rule enables a static checking tool to warn of null statements appearing on a line with other text, which would normally indicate a programming error.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">while ( ( port &amp; 0x80 ) == 0 )
{
  /* wait for pin */ ; // Noncompliant, comment before ;
  ;// wait for pin - Noncompliant, no white-space char after ;
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">while ( ( port &amp; 0x80 ) == 0 )
{
  ; // wait for pin - Compliant
}</code></pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p>MISRA C:2004, 14.3</p>
</li>
<li>
<p>MISRA C&#43;&#43;:2008, 6-2-3</p>
</li>
</ul>
</div>
<hr>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_is_related_to_s139">is related to: <a data-rspec-id="S139" class="rspec-auto-link">S139</a></h3>

</div>
<div class="sect2">
<h3 id="_is_related_to_s1116">is related to: <a data-rspec-id="S1116" class="rspec-auto-link">S1116</a></h3>

</div>
<div class="sect2">
<h3 id="_on_20_oct_2014_140928_samuel_mercier_wrote">on 20 Oct 2014, 14:09:28 Samuel Mercier wrote:</h3>
<div class="paragraph">
<p>Probably not a rule on its own; we should rather update implementation of <a data-rspec-id="S139" class="rspec-auto-link">RSPEC-139</a> and <a data-rspec-id="S1116" class="rspec-auto-link">RSPEC-1116</a></p>
</div>
</div>
<div class="sect2">
<h3 id="_on_20_oct_2014_145016_ann_campbell_wrote">on 20 Oct 2014, 14:50:16 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>This rule seems to have little faith in the ability of static checking tools. Perhaps that&#8217;s why it was dropped in the 2012 version. IMO, we should consider this fully covered by <a data-rspec-id="S1116" class="rspec-auto-link">RSPEC-1116</a>.</p>
</div>
<div class="paragraph">
<p>cc [~freddy.mallet]</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_25_nov_2014_204829_freddy_mallet_wrote">on 25 Nov 2014, 20:48:29 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>Fine for me [~ann.campbell.2] to close this ticket with resolution "Won&#8217;t Fix"</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_2_dec_2014_141328_ann_campbell_wrote">on 2 Dec 2014, 14:13:28 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>References moved to <a data-rspec-id="S1116" class="rspec-auto-link">RSPEC-1116</a></p>
</div>
</div>
</div>
</div>