include::../common/indtroduction.adoc[]

== How to fix it in JSX

Remove redundant ARIA role attribute.

=== Code examples

==== Noncompliant code example

[source,javascript,diff-id=1,diff-type=noncompliant]
----
<button role="button" onClick={handleClick}>OK</button>
----

==== Compliant solution

[source,javascript,diff-id=1,diff-type=compliant]
----
<button onClick={handleClick}>OK</button>
----

include::../common/resources.adoc[]