== Why is this an issue?

This rule is not really a rule, but a demonstration of the images that can
appear in a rule description. The Asciidoc docs are copied, but only their code
samples.

=== What is the potential impact?

include::asciidocs.adoc[]

== How to fix it in Spring

=== Code examples

==== Noncompliant code example

[source,text,diff-id=1,diff-type=noncompliant]
----
FIXME
----

==== Compliant solution

[source,text,diff-id=1,diff-type=compliant]
----
FIXED
----

=== How does this work?

include::asciidocs.adoc[]

== How to fix it in Hibernate

=== Code examples

==== Noncompliant code example

[source,text,diff-id=2,diff-type=noncompliant]
----
FIXME
----

==== Compliant solution

[source,text,diff-id=2,diff-type=compliant]
----
FIXED
----

=== How does this work?

include::asciidocs.adoc[]

== Resources

=== Documentation

include::asciidocs.adoc[]
