== Why is this an issue?

This rule is not really a rule, but a demonstration of the images that can
appear in a rule description. The Asciidoc docs are copied, but only their code
samples.

A version containing links is available in
[the initial PR](https://github.com/SonarSource/rspec/blob/b83c150164ae37a77a978d6729c09a5178cbf1aa/rules/S6826/java/asciidocs-links.adoc?plain=1).

=== What is the potential impact?

:random_variable: 42

include::asciidocs.adoc[]

== How to fix it in Spring

=== Code examples

==== Noncompliant code example

[source,text,diff-id=1,diff-type=noncompliant]
----
FIXME
----

==== Compliant solution

[source,text,diff-id=1,diff-type=compliant]
----
FIXED
----

=== How does this work?

:random_variable: 1337

include::asciidocs.adoc[]

== Resources

=== Documentation

* None

