This rule is not really a rule, but a demonstration of the images that can appear in a rule description.

== Why is this an issue?

This rule is not really a rule, but a demonstration of the images that can appear in a rule description.

Inline with Text: The image is placed directly in line with the text. The top
and bottom of the image will align with the text line.

Wrap Text: The image is placed within a block of text, and the text wraps
around the image. This can be on one side of the image or both, depending on
the size of the image and the layout of the page.

Break Text: The image is placed between paragraphs of text, serving as a break
in the text.

Text Over Image: The image serves as a background, and text is placed over it.
This is often used for headers or sections with a large, full-width image.

Align Left, Right, or Center: The image is aligned to the left, right, or
center of the page or text block. Text does not wrap in this setting; instead,
it begins below the image.

Float Left or Right: The image is placed to the left or right of the text, and
the text flows around it. This is similar to text wrapping, but the image can
be moved independently of the text.

Absolute Position: In some advanced cases, you might use CSS to position an
image absolutely within a relative container. This allows for precise control
over the image's position, but it can be complex to implement correctly.


=== What is the potential impact?

== How to fix it in Spring

=== Code examples

==== Noncompliant code example

[source,text,diff-id=1,diff-type=noncompliant]
----
FIXME
----

==== Compliant solution

[source,text,diff-id=1,diff-type=compliant]
----
FIXED
----

== How to fix it in Hibernate

=== Code examples

==== Noncompliant code example

[source,text,diff-id=2,diff-type=noncompliant]
----
FIXME
----

==== Compliant solution

[source,text,diff-id=2,diff-type=compliant]
----
FIXED
----

//=== How does this work?

//=== Pitfalls

//=== Going the extra mile


//== Resources
//=== Documentation
//=== Articles & blog posts
//=== Conference presentations
//=== Standards
//=== External coding guidelines
//=== Benchmarks
