This rule is not really a rule, but a demonstration of the images that can appear in a rule description.

== Why is this an issue?

This rule is not really a rule, but a demonstration of the images that can appear in a rule description.

This image is retrieved from a url with `[BlogPost image, 250, 350]`:

image::https://assets-eu-01.kc-usercontent.com/c196a450-1482-01db-440a-964b8358d017/fd9c3f5b-9e7c-43e7-81a3-dd9ed735e48e/java-jdk-21-lts_blog-index.webp[BlogPost image, 250, 350]

This image is an svg with `[Image,200,200,align="center"]`:

image::icon.svg[Image,200,200,align="center"]

Inline with Text: The image is placed directly in line with the text. The top
and bottom of the image will align with the text line. Here it is, retrieved
from a URL , with `[Blogpost image, 25, 25]`:
image:https://assets-eu-01.kc-usercontent.com/c196a450-1482-01db-440a-964b8358d017/fd9c3f5b-9e7c-43e7-81a3-dd9ed735e48e/java-jdk-21-lts_blog-index.webp[Blogpost image, 25, 25]
and this is some other text.

Here’s an example of a floating block image. The paragraphs or other blocks
that follow the image will float up into the available space next to the image.
The image will also be positioned horizontally in the center of the image
block.

Example 1. A block image pulled to the right and centered within the block (`[Image,200,200,float="right",align="center"]`)

image::image.png[Image,200,200,float="right",align="center"]

Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed maximus ut est nec
porttitor. Vestibulum nec blandit felis. Duis euismod ac nulla nec tempus.
Pellentesque habitant morbi tristique senectus et netus et malesuada fames ac
turpis egestas. Suspendisse dapibus metus metus, ut scelerisque quam posuere
nec. Mauris vitae auctor magna. Fusce feugiat sollicitudin ornare. 


Here’s an example of a floating inline image. The image will float into the upper-right corner of the paragraph text (`[Linux,150,150,float="right"]`).

Example 2. An inline image pulled to the right of the paragraph text

image:image.png[Linux,150,150,float="right"]
You can find Linux everywhere these days!

Sed elementum, justo non ullamcorper tristique, libero odio laoreet sem, auctor
facilisis dolor eros quis lorem. Duis iaculis bibendum odio. Proin a quam
augue. Integer et dictum justo. Maecenas auctor enim nisi, eu eleifend felis
tristique ac. In posuere pharetra mauris ut lobortis. Vivamus sed luctus lacus. 

Break Text: The image is placed between paragraphs of text, serving as a break
in the text.

Text Over Image: The image serves as a background, and text is placed over it.
This is often used for headers or sections with a large, full-width image.

Align Left, Right, or Center: The image is aligned to the left, right, or
center of the page or text block. Text does not wrap in this setting; instead,
it begins below the image.

Float Left or Right: The image is placed to the left or right of the text, and
the text flows around it. This is similar to text wrapping, but the image can
be moved independently of the text.

Absolute Position: In some advanced cases, you might use CSS to position an
image absolutely within a relative container. This allows for precise control
over the image's position, but it can be complex to implement correctly.

Wrap Text: The image is placed within a block of text, and the text wraps
around the image. This can be on one side of the image or both, depending on
the size of the image and the layout of the page.



=== What is the potential impact?

==== Positioning roles
Here are the examples from above, now configured to use roles that map to CSS classes in the default Asciidoctor stylesheet:

Example 3. Block image macro using positioning roles

[.right.text-center]
image::image.png[Tiger,200,200]

Example 4. Inline image macro using positioning role

image:image.png[Sunset,150,150,role=right] What a beautiful sunset!


The following table lists all the roles available out of the box for positioning images.

https://docs.asciidoctor.org/asciidoc/latest/macros/image-position/#positioning-roles


Merely setting the float direction on an image is not sufficient for proper
positioning. That’s because, by default, no space is left between the image and
the text. To alleviate this problem, we’ve added sensible margins to images
that use either the positioning named attributes or roles.

If you want to customize the image styles, perhaps to customize the margins,
you can provide your own additions to the stylesheet (either by using your own
stylesheet that builds on the default stylesheet or by adding the styles to a
docinfo file).

== How to fix it in Spring

=== Code examples

==== Noncompliant code example

[source,text,diff-id=1,diff-type=noncompliant]
----
FIXME
----

==== Compliant solution

[source,text,diff-id=1,diff-type=compliant]
----
FIXED
----

== How to fix it in Hibernate

=== Code examples

==== Noncompliant code example

[source,text,diff-id=2,diff-type=noncompliant]
----
FIXME
----

==== Compliant solution

[source,text,diff-id=2,diff-type=compliant]
----
FIXED
----

//=== How does this work?

//=== Pitfalls

//=== Going the extra mile


//== Resources
//=== Documentation
//=== Articles & blog posts
//=== Conference presentations
//=== Standards
//=== External coding guidelines
//=== Benchmarks
