== Why is this an issue?

This rule is not really a rule, but a demonstration of the images that can
appear in a rule description. The Asciidoc docs are copied, but only their code
samples.

=== What is the potential impact?

:random_variable: 42

include::asciidocs.adoc[]

== How to fix it in Spring

=== Code examples

==== Noncompliant code example

[source,text,diff-id=1,diff-type=noncompliant]
----
FIXME
----

==== Compliant solution

[source,text,diff-id=1,diff-type=compliant]
----
FIXED
----

=== How does this work?

:random_variable: 1337

include::asciidocs.adoc[]

== How to fix it in Hibernate

=== Code examples

This section demonstrates URL-only images.

==== Noncompliant code example

[source,text,diff-id=2,diff-type=noncompliant]
----
FIXME
----

==== Compliant solution

[source,text,diff-id=2,diff-type=compliant]
----
FIXED
----

=== How does this work?

:random_variable: 31337

include::asciidocs-links.adoc[]

== Resources

=== Documentation

:random_variable: 96

include::asciidocs.adoc[]

