<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>L’instruction INSERT/SELECT correspond à des mises à jour de masse sans possibilité de prendre des commits intermédiaires. Ce qui est très dommageable en terme d’accès concurrents sur des environnements 24/24 7/7.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cobol" data-lang="cobol">EXEC SQL INSERT INTO TESTPROJ (PROJNO, PROJNAME, DEPTNO)
		              SELECT PROJNO, PROJNAME, DEPTNO
		              FROM S1TESTPR
		              WHERE DEPTNO LIKE ‘B%’ END-EXEC.</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_27_may_2014_092914_freddy_mallet_wrote">on 27 May 2014, 09:29:14 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>This rule is specific to the BNP context. We should provide an XPath rule to get the same result.</p>
</div>
</div>
</div>
</div>