include::../description.adoc[]

== Noncompliant Code Example

[source,python]
----
r"ab{1,1}c"
r"ab{1}c"
r"ab{0,0}c"
r"ab{0}c"
----

== Compliant Solution

[source,python]
----
r"abc"
r"ac"
----

include::../implementation.adoc[]
