== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,python]
----
r"ab{1,1}c"
r"ab{1}c"
r"ab{0,0}c"
r"ab{0}c"
----

=== Compliant solution

[source,python]
----
r"abc"
r"ac"
----

include::../implementation.adoc[]
