== Why is this an issue?

Dereferencing a null pointer has undefined behavior, and it is particularly harmful if a reference is then bound to the result, because a reference is assumed to refer to a valid object.


=== Noncompliant code example

[source,cpp]
----
void doSomething(A& a);
void f() {
  A* a = nullptr;
  // ...
  doSomething(*a); // Noncompliant
}
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Remove this binding to reference of null pointer dereference.


endif::env-github,rspecator-view[]
