== How to fix it in Java Logging API

=== Code examples

include::../../common/fix/code-rationale.adoc[]

==== Noncompliant code example

[source,kotlin,diff-id=1,diff-type=noncompliant]
----
class Example {
    private val logger: Logger = Logger.getLogger("Logger")

    fun doGet(request: HttpServletRequest) {
        val data = request.getParameter("data")
        if (data != null) {
            logger.log(Level.INFO, "Data: {0} ", data)
        }
    }
}
----

==== Compliant solution

[source,kotlin,diff-id=1,diff-type=compliant]
----
class Example {
    private val logger: Logger = Logger.getLogger("Logger")

    fun doGet(request: HttpServletRequest) {
        var data = request.getParameter("data")
        if (data != null) {
            data = data.replace("[\n\r]".toRegex(), "_")
            logger.log(Level.INFO, "Data: {0} ", data)
        }
    }
}
----

include::../../common/fix/how-does-this-work.adoc[]
