<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule checks that the specified attributes are present in HTML tags.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="paragraph">
<p>Given a required attribute list of: <code>img.alt,img.height,img.width</code>:</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-html" data-lang="html">&lt;img src="/images/queen.png"&gt; &lt;!-- Noncompliant; missing all required attributes --&gt;</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-html" data-lang="html">&lt;img src="/images/queen.png" width="60" height="85" alt="Elizabeth II"&gt;</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Add the missing "xxx" attribute to element "yyy".</p>
</div>
</div>
<div class="sect2">
<h3 id="_parameters">Parameters</h3>
<div class="sidebarblock">
<div class="content">
<div class="title">attributes</div>
<div class="paragraph">
<p>Comma-separated list of tag.attributes that are required. E.G. specify img.alt to require an "alt" attribute in an "img" tag.</p>
</div>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_22_aug_2014_102235_freddy_mallet_wrote">on 22 Aug 2014, 10:22:35 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>My feedback @Ann:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>I would remove the SQALE mapping because the current one might be misleading</p>
</li>
<li>
<p>The default values of the 'attributes' property is missing</p>
</li>
</ul>
</div>
</div>
<div class="sect2">
<h3 id="_on_25_aug_2014_140341_ann_campbell_wrote">on 25 Aug 2014, 14:03:41 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>In the interest of completeness: this RSpec created to document an existing rule in the Web plugin which is neither fish nor fowl (i.e. not a template rule and no defaults)</p>
</div>
</div>
</div>
</div>