include::../description.adoc[]

== Noncompliant Code Example

With the default threshold value of 3


----
if (((condition1 && condition2) || (condition3 && condition4)) && condition5) { ... } 
----

== Compliant Solution

----
if ((MyFirstCondition() || MySecondCondition()) && MyLastCondition()) { ... }     
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../parameters.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
