Use of ``++File.deleteOnExit()++`` is not recommended for the following reasons:

* The deletion occurs only in the case of a normal JVM shutdown but not when the JVM crashes or is killed.
* For each file handler, the memory associated with the handler is released only at the end of the process.


== Noncompliant Code Example

----
File file = new File("file.txt");
file.deleteOnExit();  // Noncompliant
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

endif::env-github,rspecator-view[]
