<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p><code>*</code> and <code>**</code>, conventionally known as <code>*args</code> and <code>**kwargs</code> (although any names can be used), can be used in either a function definition or in a function call. In a function definition, they allow a variable number of arguments to be passed. In a function call, they simply obscure the actual values being used in the call.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">my_arg_list = ("yellow", "green", 42)
my_arg_dictionary = {"first" : "Sam", "next": 4, "last": my_obj}
# many lines of code...

function_with_args(*my_arg_list) #Noncompliant
function_with_named_args(**my_arg_dictionary) #Noncompliant</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">function_with_args("yellow", "green", 42)
function_with_named_args(first = "Sam", next = 4, last = my_obj)</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_21_nov_2019_181310_nicolas_harraudeau_wrote">on 21 Nov 2019, 18:13:10 Nicolas Harraudeau wrote:</h3>
<div class="paragraph">
<p>Closing this rule as it would raise many False Positives. I see multiple cases where passing <code>**kwargs</code> or <code>*args</code> is perfectly fine:</p>
</div>
<div class="paragraph">
<p>1/ building an argument list on the fly. Example:</p>
</div>
<div class="listingblock">
<div class="content">
<pre>options = {}
if condition:
    options[key] = value
myfunction(**options)

# One could argue that this is equivalent:
key = default
if condition:
   key = value
myfunction(key = value)
# But actually this overrides the default value of "key" set in myfunction.</pre>
</div>
</div>
<div class="paragraph">
<p>2/ It is very common to forward arguments you don&#8217;t know much about from a caller to another method/function.</p>
</div>
<div class="listingblock">
<div class="content">
<pre>def myfilter(func):
    def decorator(*args, c=None, **kwargs):
        if c is None or c &lt; 0:
            return None
        return func(c=c, *args, **kwargs)
    return decorator

@myfilter
def myfunction(a, b, *args, c=None):
    return 42
print(myfunction(1,2,c=-1))</pre>
</div>
</div>
</div>
</div>
</div>