Calling ``++Environment.Exit(exitCode)++`` or ``++Application.Exit()++`` terminates the process and returns an exit code to the operating system..

Each of these methods should be used with extreme care, and only when the intent is to stop the whole application.

== Noncompliant Code Example

----
Environment.Exit(0);
Application.Exit();
----

== Exceptions

These methods are ignored inside ``++Main++``.

ifdef::rspecator-view[]
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::rspecator-view[]
