<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>SOQL queries might return too many results and exceed the <a href="https://developer.salesforce.com/docs/atlas.en-us.apexcode.meta/apexcode/apex_gov_limits.htm#total_heap_size_limit_desc">maximum heap size</a>. To avoid this issue the best solution is to use SOQL for loops, i.e. for loops with inline an SOQL query. The loop will then chunk efficiently the query&#8217;s results by calling "query" and "queryMore" methods.</p>
</div>
<div class="paragraph">
<p>This rule raises an issue when an SOQL query is executed and later a for loop iterates over its result.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-apex" data-lang="apex">public class mySOQLLoop {
    public static void myFunction() {
        Task[] tasks = [Select Id, subject from Task];
        for (Task task: tasks) {
            // ...
        }
    }</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-apex" data-lang="apex">public class mySOQLLoop {
    public static void myFunction() {
        for (Task task: [Select Id, subject from Task]) {
            // ...
        }
    }</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_exceptions">Exceptions</h3>
<div class="paragraph">
<p>No issue will be raised when <code>LIMIT X</code> is used where <code>X</code> is a number less than or equal to 200. In this case adding a <code>for</code> loop will not improve the performance.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p><a href="https://developer.salesforce.com/docs/atlas.en-us.apexcode.meta/apexcode/langCon_apex_loops_for_SOQL.htm">SOQL For Loops</a></p>
</li>
</ul>
</div>
<hr>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Refactor this SOQL query and for loop into a single SOQL for loops</p>
</div>
</div>
</div>
</div>