Jump statements (``++BREAK++``, ``++CONTINUE++``, ``++RETURN++``, ``++GOTO++``, and ``++THROW++``), move control flow out of the current code block. So any statements that come after a jump are dead code.

== Noncompliant Code Example

----
CREATE PROCEDURE
    AS
BEGIN
  ...
  RETURN -- Noncompliant, remove following statements

  PRINT 'End'
END
----

== Compliant Solution

----
CREATE PROCEDURE
    AS
BEGIN
  ...
  RETURN
END
----

include::../see.adoc[]

ifdef::env-github,rspecator-view[]
'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
