== Why is this an issue?

Jump statements (``++BREAK++``, ``++CONTINUE++``, ``++RETURN++``, ``++GOTO++``, and ``++THROW++``), move control flow out of the current code block. So any statements that come after a jump are dead code.

=== Noncompliant code example

[source,sql]
----
CREATE PROCEDURE
    AS
BEGIN
  ...
  RETURN -- Noncompliant, remove following statements

  PRINT 'End'
END
----

=== Compliant solution

[source,sql]
----
CREATE PROCEDURE
    AS
BEGIN
  ...
  RETURN
END
----

include::../see.adoc[]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../highlighting.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
