include::../description.adoc[]

== Noncompliant Code Example

[source,php]
----
class MyClass {
  private $foo = 4;                       //foo is unused

  public function compute($a) {
    return $a * 4;
  }
}
----

== Compliant Solution

[source,php]
----
class MyClass {

  public function compute($a) {
    return $a * 4;
  }
}
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
