The `+... = {}+` syntax is more compact, more readable and less error-prone.

== Noncompliant Code Example

----
Module Module1
  Sub Main()
    Dim foo(1) As String   ' Noncompliant
    foo(0) = "foo"
    foo(1) = "bar"
  End Sub
End Module
----

== Compliant Solution

----
Module Module1
  Sub Main()
    Dim foo = {"foo", "bar"}  ' Compliant
  End Sub
End Module
----
