== Why is this an issue?

The ``++... = {}++`` syntax is more compact, more readable and less error-prone.

=== Noncompliant code example

[source,vbnet]
----
Module Module1
  Sub Main()
    Dim foo(1) As String   ' Noncompliant
    foo(0) = "foo"
    foo(1) = "bar"
  End Sub
End Module
----

=== Compliant solution

[source,vbnet]
----
Module Module1
  Sub Main()
    Dim foo = {"foo", "bar"}  ' Compliant
  End Sub
End Module
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Refactor this code to use the "... = {}" syntax.


'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
