== Why is this an issue?

The complexity of an expression is defined by the number of ``++AND++``, ``++OR++``, ``++XOR++`` and ``++EQUIV++`` operators it contains.

A single expression's complexity should not become too high to keep the code readable.

=== Noncompliant code example

With the default threshold value of 3

[source,abap]
----
IF ((condition1 AND condition2) OR (condition3 AND condition4)) AND condition5.
  ...
ENDIF.
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../parameters.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
