<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Overriding a parent class method prevents that method from being called unless an explicit <code>super</code> call is made in the overriding method.
In some cases, not calling the parent method is fine.
However, <code>setUp</code> and <code>tearDown</code> provide some shared logic that is called before all test cases.
This logic may change over the lifetime of your codebase.
To make sure that your test cases are set up and cleaned up consistently, your overriding implementations of <code>setUp</code> and <code>tearDown</code> should call the parent implementations explicitly.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Add an explicit call to <code>super.setUp()</code> and <code>super.tearDown()</code> in the overriding methods.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">public class MyClassTest extends MyAbstractTestCase {

  private MyClass myClass;

  @Override
  protected void setUp() throws Exception {  // Noncompliant
    myClass = new MyClass();
  }
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">public class MyClassTest extends MyAbstractTestCase {

  private MyClass myClass;

  @Override
  protected void setUp() throws Exception {
    super.setUp();
    myClass = new MyClass();
  }
}</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Add a "super.[setUp()|tearDown()]" call to this method.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_24_nov_2014_192721_nicolas_peru_wrote">on 24 Nov 2014, 19:27:21 Nicolas Peru wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2]This is not relevant since junit 4 might be worth mentionning that it only concerns Junit 3 way of writing tests.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_1_dec_2014_172233_ann_campbell_wrote">on 1 Dec 2014, 17:22:33 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>done [~nicolas.peru]</p>
</div>
</div>
</div>
</div>