This rule raises an issue when the `tf.reshape` function `tensor` and `shape` arguments have different amounts of elements.

== Why is this an issue?
The `tf.reshape` returns a new `tf.Tensor` that has the same values as `tensor` in the same order, except with a new shape given by `shape`.

The `tf.reshape` call with different size for `tensor` and `shape` leads to `IllegalArgumentError

== How to fix it
Pass same size `tensor` and `shape` arguments to the `tf.reshape` function call.

=== Code examples

==== Noncompliant code example

[source,python,diff-id=1,diff-type=noncompliant]
----
import tensorflow as tf

tf.reshape([1, 2, 3, 4], [3, 2])  # Nonpompliant: InvalidArgumentError: Input to reshape is a tensor with 4 values, but the requested shape has 6
----

==== Compliant solution

[source,python,diff-id=1,diff-type=compliant]
----
import tensorflow as tf

tf.reshape([1, 2, 3, 4], [2, 2])  # Nonpompliant: InvalidArgumentError: Input to reshape is a tensor with 4 values, but the requested shape has 6
----

//=== How does this work?

//=== Pitfalls

//=== Going the extra mile


== Resources
=== Documentation
* Tensorflow documentation - https://www.tensorflow.org/api_docs/python/tf/reshape[tf.reshape]
//=== Articles & blog posts
//=== Conference presentations
//=== Standards
//=== External coding guidelines
//=== Benchmarks
