This rule raises an issue when the `tf.reshape` function's `tensor` and `shape` number have different amounts of elements.

== Why is this an issue?
The `tf.reshape` function accepts two arguments - `tensor` and `shape`.
The `tensor` argument is a `tf.Tensor` of values that should be reshaped.
The `shape` argument is an array used to explain the shape of the result.
The `tf.reshape` function returns a new `tf.Tensor` that has the same values as `tensor` in the same order, except with a new shape given by `shape`.

Calling the `tf.reshape` function with `tensor` and `shape` having different sizes leads to `IllegalArgumentError

== How to fix it
Make sure the arguments `tensor` and `shape` have the same size when passed to the `tf.reshape` function call.

=== Code examples

==== Noncompliant code example

[source,python,diff-id=1,diff-type=noncompliant]
----
import tensorflow as tf

tf.reshape([1, 2, 3, 4], [3, 2])  # Nonpompliant: InvalidArgumentError: Input to reshape is a tensor with 4 values, but the requested shape has 6
----

==== Compliant solution

[source,python,diff-id=1,diff-type=compliant]
----
import tensorflow as tf

tf.reshape([1, 2, 3, 4], [2, 2])  # OK
----

//=== How does this work?

//=== Pitfalls

//=== Going the extra mile


== Resources
=== Documentation
* Tensorflow documentation - https://www.tensorflow.org/api_docs/python/tf/reshape[tf.reshape]
//=== Articles & blog posts
//=== Conference presentations
//=== Standards
//=== External coding guidelines
//=== Benchmarks
