<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>The implementation of certain <code>ResultSet</code> methods is optional for result sets of type <code>TYPE_FORWARD_ONLY</code>. Even if your current JDBC driver does implement those methods, there&#8217;s no guarantee you won&#8217;t change drivers in the future.</p>
</div>
<div class="paragraph">
<p>This rule looks for invocations of the following methods on <code>TYPE_FORWARD_ONLY</code> <code>ResultSet</code>s:</p>
</div>
<div class="ulist">
<ul>
<li>
<p><code>isBeforeFirst</code></p>
</li>
<li>
<p><code>isAfterLast</code></p>
</li>
<li>
<p><code>isFirst</code></p>
</li>
<li>
<p><code>getRow</code></p>
</li>
</ul>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">Statement stmt = con.createStatement(ResultSet.TYPE_FORWARD_ONLY);
stmt.executeQuery("SELECT name, address FROM PERSON");
ResultSet rs = stmt.getResultSet();

if (rs.isBeforeFirst()) { // Noncompliant

}</code></pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_related_rules">Related rules</h3>
<div class="ulist">
<ul>
<li>
<p><a data-rspec-id="S2232" class="rspec-auto-link">S2232</a> - "ResultSet.isLast()" should not be used</p>
</li>
</ul>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove this call to "xxx".</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_5_dec_2014_075151_nicolas_peru_wrote">on 5 Dec 2014, 07:51:51 Nicolas Peru wrote:</h3>
<div class="paragraph">
<p>This one will require CFG to limit the number of false positive to add some constraints on the "stmt" variable to be able to determine which statement are indeed created with <code>ResultSet.TYPE_FORWARD_ONLY</code></p>
</div>
</div>
</div>
</div>