This is an issue when using time helper methods like `.weeks`, `.days`, `.months`, or `.years` on numeric values without loading the ActiveSupport extensions.

== Why is this an issue?

Ruby's core numeric classes (Integer, Float) do not include time helper methods like `.weeks`, `.days`, `.months`, or `.years`. These convenient methods are provided by ActiveSupport, a library that extends Ruby's core classes with additional functionality.

When you write code like `2.weeks` or `30.days` without loading ActiveSupport, Ruby will raise a `NoMethodError` at runtime because these methods don't exist on the numeric classes. This error only appears when the code is actually executed, making it particularly problematic in production environments.

This issue commonly occurs when:

* Moving code from a Rails application (where ActiveSupport is automatically loaded) to a plain Ruby project
* Writing standalone Ruby scripts that use time calculations
* Creating gems or libraries that need to work independently of Rails

The problem is especially tricky because the code looks correct and natural, but fails unexpectedly at runtime.

=== What is the potential impact?

This issue causes runtime failures that can crash your application when the problematic code is executed. In production environments, this can lead to:

* Application downtime when time-related calculations are performed
* Failed background jobs or scheduled tasks
* Broken user workflows that involve date/time operations
* Difficult debugging since the error only appears during execution, not during code review or static analysis

== How to fix it

Load ActiveSupport's numeric time extensions to enable the helper methods.

=== Code examples

==== Noncompliant code example

[source,ruby,diff-id=1,diff-type=noncompliant]
----
class Job
  def set_expiration
    self.expires_on = Time.now + 2.weeks  # Noncompliant
  end
end
----

==== Compliant solution

[source,ruby,diff-id=1,diff-type=compliant]
----
require 'active_support/core_ext/numeric/time'

class Job
  def set_expiration
    self.expires_on = Time.now + 2.weeks
  end
end
----

== Resources

=== Documentation

 * ActiveSupport Core Extensions - Numeric Time - https://guides.rubyonrails.org/active_support_core_extensions.html#extensions-to-numeric[Official Rails guide explaining ActiveSupport's numeric time extensions]

 * Ruby Time Class Documentation - https://ruby-doc.org/core/Time.html[Official Ruby documentation for the Time class]

 * Ruby Date Class Documentation - https://ruby-doc.org/stdlib/libdoc/date/rdoc/Date.html[Official Ruby documentation for the Date class]
