:operationName: function
:visibility: private

include::../why.adoc[]

=== Code examples

=== Noncompliant code example

[source,dart,diff-id=1,diff-type=noncompliant]
----
class Foo {
  void publicMethod() {
    ...
    _usedPrivateMethod();
    ...
  }
  void _unusedPrivateMethod() {...}
  void _usedPrivateMethod() {...}
}
----

=== Compliant solution

[source,dart,diff-id=1,diff-type=compliant]
----
class Foo {
  void publicMethod() {
    ...
    _usedPrivateMethod();
    ...
  }
  void _usedPrivateMethod() {...}
}
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

* The declaration '<unusedMember>' isn't referenced.

=== Highlighting

The identifier of the unused member.

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
