<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when a method annotated with <code>@future</code> calls another method that is also annotated with <code>@future</code>.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Future methods in Salesforce Apex are designed to execute asynchronously when system resources become available. They run in their own thread and are queued for execution.</p>
</div>
<div class="paragraph">
<p>The Salesforce platform explicitly prohibits future methods from calling other future methods. This restriction exists because:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>Future methods are already executing asynchronously, and chaining them creates unpredictable execution patterns</p>
</li>
<li>
<p>The platform cannot guarantee the order of execution for chained future methods</p>
</li>
<li>
<p>Resource management becomes complex when future methods spawn additional future methods</p>
</li>
<li>
<p>It can lead to cascading asynchronous operations that are difficult to monitor and debug</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>When a future method attempts to call another future method, the Salesforce runtime will throw an exception, causing the operation to fail.</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="paragraph">
<p>Violating this rule will result in a runtime exception when the code executes. The calling future method will fail, potentially leaving business processes incomplete.</p>
</div>
<div class="paragraph">
<p>This can lead to:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>Data inconsistency if the failed operation was part of a larger business process</p>
</li>
<li>
<p>Poor user experience due to unexpected failures</p>
</li>
<li>
<p>Difficulty in debugging and monitoring asynchronous operations</p>
</li>
<li>
<p>Unreliable system behavior in production environments</p>
</li>
</ul>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Replace the future method call with a call to a synchronous helper method that contains the same logic.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-apex" data-lang="apex">@future
public static void method1() {
    method2(); // Noncompliant
}

@future
public static void method2() {
    // Some logic
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-apex" data-lang="apex">@future
public static void method1() {
    // Call synchronous helper method instead
    processData();
}

public static void processData() {
    // Same logic as the original future method
}</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Future Methods - <a href="https://developer.salesforce.com/docs/atlas.en-us.apexcode.meta/apexcode/apex_invoking_future_methods.htm">Official Salesforce documentation on future methods and their limitations</a></p>
</li>
<li>
<p>Asynchronous Apex - <a href="https://developer.salesforce.com/docs/atlas.en-us.apexcode.meta/apexcode/apex_async_overview.htm">Overview of asynchronous processing options in Apex</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>