<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This is an issue when using JodaTime methods <code>plus(long)</code>, <code>minus(long)</code>, or <code>withDurationAdded(long, int)</code> where the time units of the long parameter are unclear.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>JodaTime provides methods that accept <code>long</code> parameters to represent time durations, but these methods are ambiguous because the units are not clear from the method signature.</p>
</div>
<div class="paragraph">
<p>When you call <code>dateTime.plus(5000)</code>, it&#8217;s not obvious whether you&#8217;re adding 5000 milliseconds, seconds, or another unit. This can lead to bugs when developers make incorrect assumptions about time units.</p>
</div>
<div class="paragraph">
<p>The affected methods (<code>plus(long)</code>, <code>minus(long)</code>, and <code>withDurationAdded(long, int)</code>) interpret the <code>long</code> parameter as milliseconds, but this is not evident from the method signature alone.</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="paragraph">
<p>Using ambiguous duration methods can lead to incorrect time calculations in your application. If developers assume the wrong time units, operations like scheduling, timeouts, or time-based logic may behave unexpectedly. This can result in features that don&#8217;t work as intended or, in worst cases, security issues if time-based access controls are misconfigured.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Replace the ambiguous long parameter with an explicit Duration object using Duration.millis(). This makes it clear that the parameter represents milliseconds.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">DateTime dateTime = new DateTime();
DateTime result = dateTime.plus(5000); // Noncompliant</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">DateTime dateTime = new DateTime();
DateTime result = dateTime.plus(Duration.millis(5000));</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>JodaTime Duration Documentation - <a href="https://www.joda.org/joda-time/apidocs/org/joda/time/Duration.html">Official documentation for JodaTime Duration class and its factory methods</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>