include::../description.adoc[]

== Noncompliant Code Example

[source,csharp]
----
using (var tripleDES = new TripleDESCryptoServiceProvider()) //Noncompliant
{
  //...
}
----

== Compliant Solution

[source,csharp]
----
using (var aes = new AesCryptoServiceProvider())
{
  //...
}
----

include::../see.adoc[]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
