//// This rule is superseded by RSPEC-5547
//// Please consider implementing this latter instead.
include::../description.adoc[]

== Noncompliant Code Example

----
using (var tripleDES = new TripleDESCryptoServiceProvider()) //Noncompliant
{
  //...
}
----

== Compliant Solution

----
using (var aes = new AesCryptoServiceProvider())
{
  //...
}
----

include::../see.adoc[]

ifdef::env-github,rspecator-view[]
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
