<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>A common anti-pattern is to check that a key does not exist in a dictionary before adding it with a corresponding value. This pattern works but is less readable than the equivalent call to the built-in dictionary method "setdefault()".</p>
</div>
<div class="paragraph">
<p>Note that if a default value is set for every key of the dictionary it is possible to use python&#8217;s <code>defaultdict</code> instead.</p>
</div>
<div class="paragraph">
<p>This rule raises an issue when a key presence is checked before being set. It only raises an issue when the value is a hard-coded string, number, list, dictionary or tuple. Computed values will not raise an issue as they can have side-effects.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">if "key" not in my_dictionary:
    my_dictionary["key"] = ["a", "b", "c"]  # Noncompliant

if "key" not in my_dictionary:
    my_dictionary["key"] = generate_value()  # Compliant. No issue is raised as generate_value() might have some side-effect.</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">my_dictionary.setdefault("key", ["a", "b", "c"])

# OR, if ["a", "b", "c"] is the default value for every key

from collections import defaultdict
my_dictionary = defaultdict(lambda: ["a", "b", "c"])</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Replace this key check and dictionary update with a call to "setdefault"</p>
</div>
</div>
</div>
</div>