<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>By convention, namespaces within a project should start with the project default namespace, and end with the file&#8217;s position within the project. Anything else may confuse maintainers and callers.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-csharp" data-lang="csharp">// file path: Gui/Screen.cs
namespace Green  // Noncompliant
{
  class Screen
  {
  }
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-csharp" data-lang="csharp">// file path: Gui/Screen.cs
namespace Gui
{
  class Screen
  {
  }
}</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_is_related_to_s3317">is related to: <a data-rspec-id="S3317" class="rspec-auto-link">S3317</a></h3>

</div>
<div class="sect2">
<h3 id="_on_9_jul_2015_141311_ann_campbell_wrote">on 9 Jul 2015, 14:13:11 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~tamas.vajk], I know you said we probably didn&#8217;t need a code example, but I really felt like we needed to illustrate what we were looking for.</p>
</div>
<div class="paragraph">
<p>My code sample is probably wrong. Would you clean it up, please?</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_20_jul_2015_112737_tamas_vajk_wrote">on 20 Jul 2015, 11:27:37 Tamas Vajk wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] I fixed the samples.</p>
</div>
<div class="paragraph">
<p>We could also have a similar rule for class and file name mappings. Or we could just add it to this rule. Class names should match their containing file name.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_20_jul_2015_143045_ann_campbell_wrote">on 20 Jul 2015, 14:30:45 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~tamas.vajk] what about the fact that I can have multiple top-level classes in a file?</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_4_aug_2015_145151_tamas_vajk_wrote">on 4 Aug 2015, 14:51:51 Tamas Vajk wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] You can, but you shouldn&#8217;t. We could check if any of the top-level classes' names match the file name. Also, there is a problem with <code>partial</code> classes. They might be defined in multiple files in the same folder. So we can&#8217;t force the file name to be identical. In this case usually we have a file name in the following format: <em>CLASSNAME.SOMEOTHERNAME.cs</em>, for example: <em>MyForm.Designer.cs</em></p>
</div>
</div>
<div class="sect2">
<h3 id="_on_5_aug_2015_153243_ann_campbell_wrote">on 5 Aug 2015, 15:32:43 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>Thanks [~tamas.vajk]. I created a new rule (which is assigned to you).</p>
</div>
<div class="paragraph">
<p>Note that I&#8217;ve also pulled this phrase from the Noncompliant example section: Given a project default namespace of "MyProject"</p>
</div>
<div class="paragraph">
<p>Since there was no reference to "MyProject" in the code itself.</p>
</div>
</div>
</div>
</div>