<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Two "hash" classes, <code>Hashtable</code>, and <code>ConcurrentHashMap</code> offer <code>contains</code> methods. One might naively assume that the <code>contains</code> method searches both keys and values for its argument. And one would be wrong. Because these legacy methods search only values, they are likely to mislead maintainers even if the original coder understood precisely what&#8217;s going on.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">Hashtable&lt;String,Object&gt; ht = new Hashtable&lt;&gt;();
// ...

if (ht.contains(foo)) {  // Noncompliant
  // ...
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">Hashtable&lt;String,Object&gt; ht = new Hashtable&lt;&gt;();
// ...

if (ht.containsValue(foo)) {
  // ...
}</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Use "containsValue" instead.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p><code>contains</code></p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_14_jul_2016_203916_ann_campbell_wrote">on 14 Jul 2016, 20:39:16 Ann Campbell wrote:</h3>
<div class="paragraph">
<p><a href="https://github.com/google/error-prone/blob/master/docs/bugpattern/HashtableContains.md" class="bare">https://github.com/google/error-prone/blob/master/docs/bugpattern/HashtableContains.md</a></p>
</div>
</div>
<div class="sect2">
<h3 id="_on_15_jul_2016_094830_freddy_mallet_wrote">on 15 Jul 2016, 09:48:30 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>According to following statement [~ann.campbell.2] "they are likely to mislead maintainers even if the original coder understood precisely what&#8217;s going on", this sounds to be more a Critical or Blocker Code Smells than a Bug.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_15_jul_2016_142125_ann_campbell_wrote">on 15 Jul 2016, 14:21:25 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>I disagree [~freddy.mallet]. That statement gives the original coder the benefit of the doubt, but with a significant eye-roll. "<em>even if</em> the original coder understood&#8230;&#8203;"</p>
</div>
<div class="paragraph">
<p>I can reword if you like.</p>
</div>
</div>
</div>
</div>