== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,csharp]
----
public int NumberOfMinutes(int hours)
{
  int seconds = 0;   // seconds is never used 
  return hours * 60;
}
----

=== Compliant solution

[source,csharp]
----
public int NumberOfMinutes(int hours)
{
  return hours * 60;
}
----

=== Exceptions

Unused locally created resources in a ``++using++`` statement are not reported.

----
using(var t = new TestTimer()) // t never used, but compliant.
{
  //...
}
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
