<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Constant members are copied at compile time to the call sites, instead of being fetched at runtime.</p>
</div>
<div class="paragraph">
<p>As an example, say you have a library with a constant <code>Version</code>  member set to <code>1.0</code>, and a client application linked to it. This library is then updated and <code>Version</code> is set to <code>2.0</code>. Unfortunately, even after the old DLL is replaced by the new one,  <code>Version</code> will still be <code>1.0</code> for the client application. In order to see <code>2.0</code>, the client application would need to be rebuilt against the new version of the library.</p>
</div>
<div class="paragraph">
<p>This means that you should use constants to hold values that by definition will never change, such as <code>Zero</code>. In practice, those cases are uncommon, and therefore it is generally better to avoid constant members.</p>
</div>
<div class="paragraph">
<p>This rule only reports issues on public constant fields, which can be reached from outside the defining assembly.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-csharp" data-lang="csharp">public class Foo
{
    public const double Version = 1.0;           // Noncompliant
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-csharp" data-lang="csharp">public class Foo
{
    public static double Version
    {
      get { return 1.0; }
    }
}</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Change this constant to a ["Shared ReadOnly"|"static" read-only] property.</p>
</div>
</div>
</div>
</div>