Sharing some naming conventions is a key point to make it possible for a team to efficiently collaborate. This rule allows to check that all subroutine names match a provided regular expression.

== Noncompliant Code Example

With default provided regular expression <code>^SR[a-zA-Z0-9]*$</code>:
----
C     XXXXX         BEGSR
...
C                   ENDSR
----
----
 /free
   begsr yyyy;
   endsr;
 /end-free{code}
----

== Compliant Solution

----
C     SR102         BEGSR
...
C                   ENDSR
----
----
 /free
   begsr SR103;
   endsr;
 /end-free
----
