Sharing some naming conventions is a key point to make it possible for a team to efficiently collaborate. This rule allows to check that all subroutine names match a provided regular expression.

== Noncompliant Code Example

With default provided regular expression ``++^SR[a-zA-Z0-9]*$++``:

----
C     XXXXX         BEGSR
...
C                   ENDSR
----

----
 /free
   begsr yyyy;
   endsr;
 /end-free{code}
----

== Compliant Solution

----
C     SR102         BEGSR
...
C                   ENDSR
----

----
 /free
   begsr SR103;
   endsr;
 /end-free
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::parameters.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
