<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>When using the <code>Stream</code> API, call chains should be simplified as much as possible. Not only does it make the code easier to read, it also avoid creating unnecessary temporary objects.</p>
</div>
<div class="paragraph">
<p>This rule raises an issue when one of the following substitution is possible:</p>
</div>
<table class="tableblock frame-all grid-all stretch">
<colgroup>
<col style="width: 50%;">
<col style="width: 50%;">
</colgroup>
<thead>
<tr>
<th class="tableblock halign-center valign-top">Original</th>
<th class="tableblock halign-center valign-top">Preferred</th>
</tr>
</thead>
<tbody>
<tr>
<td class="tableblock halign-center valign-top"><p class="tableblock"><code>stream.filter(predicate).findFirst().isPresent()</code></p></td>
<td class="tableblock halign-center valign-top"><p class="tableblock"><code>stream.anyMatch(predicate)</code></p></td>
</tr>
<tr>
<td class="tableblock halign-center valign-top"><p class="tableblock"><code>stream.filter(predicate).findAny().isPresent()</code></p></td>
<td class="tableblock halign-center valign-top"><p class="tableblock"><code>stream.anyMatch(predicate)</code></p></td>
</tr>
<tr>
<td class="tableblock halign-center valign-top"><p class="tableblock"><code>!stream.anyMatch(predicate)</code></p></td>
<td class="tableblock halign-center valign-top"><p class="tableblock"><code>stream.noneMatch(predicate)</code></p></td>
</tr>
<tr>
<td class="tableblock halign-center valign-top"><p class="tableblock"><code>!stream.anyMatch(x -&gt; !(...))</code></p></td>
<td class="tableblock halign-center valign-top"><p class="tableblock"><code>stream.allMatch(...)</code></p></td>
</tr>
<tr>
<td class="tableblock halign-center valign-top"><p class="tableblock"><code>stream.map(mapper).anyMatch(Boolean::booleanValue)</code></p></td>
<td class="tableblock halign-center valign-top"><p class="tableblock"><code>stream.anyMatch(predicate)</code></p></td>
</tr>
</tbody>
</table>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">boolean hasRed = widgets.stream().filter(w -&gt; w.getColor() == RED).findFirst().isPresent(); // Noncompliant</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">boolean hasRed = widgets.stream().anyMatch(w -&gt; w.getColor() == RED);</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Replace this "xxx" chain with "yyy"</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>The chain to be replaced</p>
</div>
</div>
</div>
</div>