<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when code uses the deprecated <code>UserInfo.getRoleAndSubordinates()</code> method, which will be removed in future Salesforce releases.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Salesforce is phasing out the <code>getRoleAndSubordinates()</code> method as part of the Winter '26 release. This method will stop working in Spring '26, potentially breaking your applications.</p>
</div>
<div class="paragraph">
<p>The <code>getRoleAndSubordinates()</code> method has been replaced by <code>getRoleAndSubordinatesInternal()</code> to provide better internal handling of role-based sharing groups. Both methods return the same data - a set of user IDs that includes the current user and all users in subordinate roles.</p>
</div>
<div class="paragraph">
<p>Using the deprecated method creates technical debt and puts your code at risk of runtime failures when Salesforce removes support for the old method.</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="paragraph">
<p>Applications using the deprecated method will experience runtime errors and broken functionality once Salesforce removes support in Spring '26. This can affect:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>SOQL queries that filter records based on role hierarchy</p>
</li>
<li>
<p>Security and sharing logic that depends on role-based access</p>
</li>
<li>
<p>Reports and dashboards that use role-based data filtering</p>
</li>
<li>
<p>Any business processes that rely on hierarchical data access</p>
</li>
</ul>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Replace <code>UserInfo.getRoleAndSubordinates()</code> with <code>UserInfo.getRoleAndSubordinatesInternal()</code> in SOQL queries. The replacement method provides identical functionality.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-apex" data-lang="apex">List&lt;Account&gt; accounts = [SELECT Id, Name FROM Account WHERE Id IN (
    SELECT AccountId FROM Contact
    WHERE OwnerId IN :UserInfo.getRoleAndSubordinates() // Noncompliant
)];</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-apex" data-lang="apex">List&lt;Account&gt; accounts = [SELECT Id, Name FROM Account WHERE Id IN (
    SELECT AccountId FROM Contact
    WHERE OwnerId IN :UserInfo.getRoleAndSubordinatesInternal()
)];</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>UserInfo Class Documentation - <a href="https://developer.salesforce.com/docs/atlas.en-us.apexref.meta/apexref/apex_methods_system_userinfo.htm">Official Salesforce documentation for the UserInfo class and its methods</a></p>
</li>
<li>
<p>Winter '26 Release Notes - <a href="https://help.salesforce.com/s/articleView?id=release-notes.salesforce_release_notes.htm&amp;release=208&amp;type=5">Salesforce release notes detailing the deprecation of getRoleAndSubordinates()</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>