== Why is this an issue?

include::../description.adoc[]

== How to fix it

=== Code examples

==== Noncompliant code example

[source,php,diff-id=1,diff-type=noncompliant]
----
for ($i = 0; $i < $length; $i--) { // Noncompliant
  //...
}
----

==== Compliant solution

[source,php,diff-id=1,diff-type=compliant]
----
for ($i = 0; $i < $length; $i++) {
  //...
}
----

include::../rspecator.adoc[]
