include::../description.adoc[]

== Noncompliant Code Example

----
for ($i = 0; $i < $length; $i--) { // Noncompliant
  //...
}
----

== Compliant Solution

----
for ($i = 0; $i < $length; $i++) {
  //...
}
----

include::../see.adoc[]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
