== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,csharp]
----
// the two statements seems to be attached to the if statement, but that is only true for the first one:
if (condition)
  ExecuteSomething();
  CheckSomething();
----

=== Compliant solution

[source,csharp]
----
if (condition)
{
  ExecuteSomething();
  CheckSomething();
}
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
