Signal handling contains implementation-defined and undefined behavior.


== Noncompliant Code Example

[source,cpp]
----
#include <signal.h> /* Noncompliant */
----


Mentioning S1000 here for testing purposes.

== See

* MISRA C:2004, 20.8  - The signal handling facilities of <signal.h> shall not be used.
* MISRA C:2012, 21.5 - The standard header file <signal.h> shall not be used


ifdef::env-github,rspecator-view[]
'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]

endif::env-github,rspecator-view[]
