== Why is this an issue?

Coroutines, introduced in {cpp}20, are functions in which execution can be suspended and resumed. When a coroutine resumes, it takes over where it left thanks to the _coroutine state_.

A _coroutine state_ is an object which contains all the information a coroutine needs to resume its execution correctly: local variables, copy of the parameters...


This means that if the coroutine has a parameter that is a reference to an object, this object must exist as long as the coroutine is not destroyed. Otherwise, the reference stored in the _coroutine state_ will become a dangling reference and will lead to undefined behavior when the coroutine resumes.


This rule detects when a coroutine parameter becomes a dangling reference. 

To fix this, you can either pass the parameter by value or extend the lifetime of the parameter.


=== Noncompliant code example

[source,cpp]
----
generator<char> spell(const std::string& m) {
    for (char letter : m) {
        co_yield letter;
    }
}

void print() {
    for (char letter : spell("hello world")) { // Noncompliant, parameter becomes a dangling reference
        std::cout << letter << '\n';
    }
}
----


=== Compliant solution

[source,cpp]
----
generator<char> spell(const std::string m) { // Either pass the argument by copy...
    for (char letter : m) {
        co_yield letter;
    }
}

void print() {
    for (char letter : spell("hello world")) {
        std::cout << letter << '\n';
    }
}

// Or

generator<char> spell(const std::string& m) {
    for (char letter : m) {
        co_yield letter;
    }
}

void print() {
    std::string message = "hello world";
    for (char letter : spell(message)) { // ... Or increase the lifetime of the parameter
        std::cout << letter << '\n';
    }
}
----

