Whenever a CICS command is used with a ``++NOHANDE++`` or ``++RESP++`` option, the default CICS exception handling is disabled. The correct approach is then to make sure that every possible exception is handled properly directly in the code. This is done by examining the ``++RESP++`` value or the ``++EIBRESP++`` field value.


It is possible to compare their value to hardcoded numbers corresponding to their binary value, however this makes the code difficult to read and maintain. It is recommended to use instead the DFHRESP built-in translator function which enables the use of equivalent symbolic values.


This rule raises an issue when the ``++EIBRESP++`` field is compared to a hardcoded-number. This rule does not handle ``++RESP++`` values for now.


== Noncompliant Code Example

----
if EIBRESP=36 then /* Noncompliant */
end;

select (EIBRESP);
    when(36) ...;  /* Noncompliant */
end;
----


== Compliant Solution

----
if EIBRESP=DFHRESP(MAPFAIL) then
end;

select (EIBRESP);
    when(DFHRESP(MAPFAIL)) ...;
end;
----


== See

* https://www.ibm.com/docs/en/cics-ts/5.6?topic=conditions-handling-exception-by-inline-code[Handling exception conditions by in-line code]


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

include::highlighting.adoc[]

endif::env-github,rspecator-view[]
