Whenever a CICS command is used with a ``++NOHANDE++`` or ``++RESP++`` option, the default CICS exception handling is disabled. The correct approach then is to ensure that every possible exception is handled correctly directly in the code and to do this, you need to examine the ``++RESP++`` value or the ``++EIBRESP++`` field value.

It is possible to compare the ``++RESP++`` and ``++EIBRESP++`` field values to hard-coded numbers or variables containing numeric values; however, this makes the code difficult to read and maintain. It is recommended to use instead the ``++DFHRESP++`` built-in translator function, which enables the use of the equivalent symbolic values.

This rule raises an issue when the ``++EIBRESP++`` field is compared directly to a variable or hard-coded numeric value that is not wrapped in the ``++DFHRESP++`` function.

This rule does not handle ``++RESP++`` values for now.


== Noncompliant Code Example

[source,pli]
----
if EIBRESP=36 then /* Noncompliant */
end;

if EIBRESP=MAPFAIL then /* Noncompliant */
end;

select (EIBRESP);
    when(36) ...;  /* Noncompliant */
end;
----


== Compliant Solution

[source,pli]
----
if EIBRESP=DFHRESP(36) then
end;

if EIBRESP=DFHRESP(MAPFAIL) then
end;

select (EIBRESP);
    when(DFHRESP(MAPFAIL)) ...;
end;
----


== See

* https://www.ibm.com/docs/en/cics-ts/6.1?topic=conditions-handling-exception-by-inline-code[Handling exception conditions by in-line code]
* https://www.ibm.com/docs/en/cics-ts/5.2?topic=development-exec-interface-block-fields[EXEC interface block fields]
* https://www.ibm.com/docs/en/cics-ts/6.1?topic=code-how-use-resp-resp2-options[How to use the RESP and RESP2 options]


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

include::highlighting.adoc[]

endif::env-github,rspecator-view[]
