== Why is this an issue?

This statement deletes all rows of an internal table itab. This ``++REFRESH++`` statement is deprecated and usage should be avoided.


=== Noncompliant code example

[source,abap]
----
REFRESH itab.
----


=== Compliant solution

[source,abap]
----
CLEAR itab.
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Use "CLEAR" or "FREE" statement instead of the obsolete "REFRESH" statement.


'''
== Comments And Links
(visible only on this page)

=== is related to: S1540

endif::env-github,rspecator-view[]
