This statement deletes all rows of an internal table itab. This ``++REFRESH++`` statement is deprecated and usage should be avoided.


== Noncompliant Code Example

----
REFRESH itab.
----


== Compliant Solution

----
CLEAR itab.
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
