This rule reports when there is an Architecture Directive to remove a specific usage of a class or package from a given location.

== Why is this an issue?

This issue is being raised because there is a dependency in the code, and an Architecture Directive has been created to remove it.

Architecture Directives are defined in the “Architecture” section of the project by an authorized user.

=== What is the potential impact?

Flaws in the architecture constitute unnecessary complexity which affects developer productivity. Removing this flaw makes future development easier by favoring local changes in the code.

== How to fix it

Remove or refactor the problematic dependency. This may involve replacing the dependency with an allowed one, moving code to appropriate classes or packages, or introducing new abstractions. The exact solution depends on your project’s specific situation.

=== Code examples

==== Noncompliant code example

The following code contains two cycles: Order &rarr; Customer &rarr; Order, and Order &rarr; Product &rarr; Order. Furthermore, both cycles are connected to each other via class Order. This is called _a tangle consisting of two cycles_.

[source,java,diff-id=1,diff-type=noncompliant]
----
class Order {
    public Customer customer;
    public List<Product> products;
}
class Customer {
    public List<Order> orders;
}
class Product {
    public List<Order> orders;
}
----

==== Compliant solution

The issue can be resolved by changing the structure. Two service functions can replace the dependencies Customer &rarr; Order and Product &rarr; Order.

[source,java,diff-id=1,diff-type=compliant]
----
class Order {
    public Customer customer;
    public List<Product> products;
}
class Customer { }
class Product { }
interface OrderService {
  List<Order> getOrdersByCustomer(Customer customer);
  List<Order> getOrdersByProduct(Product product);
}
----
