include::../description_with_reference_to_s5998.adoc[]

== Noncompliant Code Example

[source,javascript]
----
/a|b|c/; // Noncompliant
----

== Compliant Solution

[source,javascript]
----
/[abc]/;
// or
/[a-c]/;
----
