== Why is this an issue?

include::../description_with_reference_to_s5998.adoc[]

=== Noncompliant code example

[source,javascript]
----
/a|b|c/; // Noncompliant
----

=== Compliant solution

[source,javascript]
----
/[abc]/;
// or
/[a-c]/;
----
