include::../why-fields.adoc[]

=== Noncompliant code example

[source,php]
----
class Foo {
  public $myField;

  public function doSomething() {
    $myField = 0; // Noncompliant
    ...
  }
}
----

include::../rspecator.adoc[]