include::../description.adoc[]

== Noncompliant Code Example

----
class Foo {
  public $myField;

  public function doSomething() {
    $myField = 0;
    ...
  }
}
----

include::../see.adoc[]

ifdef::env-github,rspecator-view[]
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
