<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>When each of a pair of classes includes the other, a cycle is created. Class cycles make compilation difficult because each class requires the other to build correctly. Breaking a cycle may be as difficult as redesigning the classes in question, or it may be as simple as creating an interface for one of the classes to implement and the other to use.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-text" data-lang="text">// A.java
  import B;
  // ..

// B.java
  import A; // Noncompliant</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-text" data-lang="text">// A.java
  import MyInterface;
  // ..

// MyInterface.java
  interface MyInterface {
    //...

// B.java
  import A;
  import MyInterface;

  public class B implements MyInterface {
    //...</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_15_oct_2014_164320_ann_campbell_wrote">on 15 Oct 2014, 16:43:20 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>Assumptions</p>
</div>
<div class="ulist">
<ul>
<li>
<p>issue will be logged on import statement of individual class</p>
</li>
<li>
<p>cycles within a package will not be included in first implementation</p>
</li>
<li>
<p>false negatives in initial implementation on cycles of more than 2 classes</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>Note, no overlap is intended with <a data-rspec-id="S1196" class="rspec-auto-link">RSPEC-1196</a></p>
</div>
</div>
<div class="sect2">
<h3 id="_on_15_oct_2014_221240_freddy_mallet_wrote">on 15 Oct 2014, 22:12:40 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>According to me this rule is valueless @Nicolas but I let you decide.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_21_oct_2014_152918_nicolas_peru_wrote">on 21 Oct 2014, 15:29:18 Nicolas Peru wrote:</h3>
<div class="paragraph">
<p>I don&#8217;t see the difference with <a data-rspec-id="S1196" class="rspec-auto-link">RSPEC-1196</a> : if you have cycle between classes you will have a cycle between package.</p>
</div>
<div class="paragraph">
<p>If both classes are at the same package level, then I don&#8217;t see why you should not do something like that. This is valid code and I don&#8217;t see any special issue that won&#8217;t be covered by 1196.</p>
</div>
</div>
</div>
</div>