<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>In Dart, there&#8217;s a convention to distinguish between public API and the implementation details. It states:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>Files inside <code>lib/src</code> are considered internal implementation detail, and shouldn&#8217;t be referenced</p>
</li>
<li>
<p>Other files inside the <code>lib</code> directory are publicly accessible and are safe to be imported</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>Introducing the dependency on the internal implementation of other packages is not recommended as it is the subject to change and may introduce maintainability issues and break your code unexpectedly.</p>
</div>
<div class="sect2">
<h3 id="_exceptions">Exceptions</h3>
<div class="paragraph">
<p>It is still safe to reference libraries from <code>lib/src</code> within the same package.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Refactor your code, so you don&#8217;t need to reference other package&#8217;s implementation detail. Check if the needed functionality is available through the Public API.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-dart" data-lang="dart">import 'package:other/src/impl.dart'; // Noncompliant here

void main() {
  doSomethingPrivate(); // this function is coming from 'other' package
}</code></pre>
</div>
</div>
<div class="paragraph">
<p>You can try to find some public API that has the same function:</p>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-dart" data-lang="dart">import 'package:other/api.dart';

void main() {
  doSomethingPublic(); // this function is a public API wrapper of 'doSomethingPrivate'
}</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Dart Docs -https://dart.dev/tools/linter-rules/implementation_imports[Dart Linter rule - implementation_imports]</p>
</li>
<li>
<p>Dart Docs - <a href="https://dart.dev/tools/pub/package-layout#implementation-files">Implementation files</a></p>
</li>
</ul>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="ulist">
<ul>
<li>
<p>Import of a library in the 'lib/src' directory of another package.</p>
</li>
</ul>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>Import statement</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
</div>
</div>