ECMAScript 2015 added the ability to use template literals instead of concatenation. Since their use is clearer and more concise, they are preferred in environments that support ECMAScript 2015.


== Noncompliant Code Example

----
function sayHello(name) {
  console.log("hello " + name);  // Noncompliant
}
----


== Compliant Solution

----
function sayHello(name) {
  console.log(`hello ${name}`);
}
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

include::highlighting.adoc[]

endif::env-github,rspecator-view[]
