== Why is this an issue?

ECMAScript 2015 added the ability to use template literals instead of concatenation. Since their use is clearer and more concise, they are preferred in environments that support ECMAScript 2015.


=== Noncompliant code example

[source,javascript]
----
function sayHello(name) {
  console.log("hello " + name);  // Noncompliant
}
----


=== Compliant solution

[source,javascript]
----
function sayHello(name) {
  console.log(`hello ${name}`);
}
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Convert this concatenation to the use of a template.


=== Highlighting

Entire concatenation expression


endif::env-github,rspecator-view[]
