<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p><code>End</code> statements exit the control flow of the program in an unstructured way. This statement stops code execution immediately without executing <code>Dispose</code> or <code>Finalize</code> methods, or executing <code>Finally</code> blocks. Therefore, it should be avoided.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-vbnet" data-lang="vbnet">Module Module1
    Sub Print(ByVal str As String)
       Try
            ...
            End       ' Noncompliant
        Finally
            ' do something important here
            ...
        End Try
    End Sub
End Module</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove this call to "xxx" or ensure it is really required.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_14_oct_2015_091233_tamas_vajk_wrote">on 14 Oct 2015, 09:12:33 Tamas Vajk wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] This subtask seems to be totally different from its parent task. The parent task exits the execution of the program, while this one is more like using <code>break</code> statements or <code>goto</code>s. Should we map it to something else?</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_14_oct_2015_124359_ann_campbell_wrote">on 14 Oct 2015, 12:43:59 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~tamas.vajk] leaving the code samples aside, the subtask does describe the same thing as the main task if you look at the first half of the description.</p>
</div>
<div class="paragraph">
<p>Do you verify that the descriptive task is accurate and relevant? And if so, can you supply new code samples?</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_15_oct_2015_063359_tamas_vajk_wrote">on 15 Oct 2015, 06:33:59 Tamas Vajk wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] I don&#8217;t think they describe the same think. The main task describes how to end the execution of the program. While the subtask says not to use control flow jumps.</p>
</div>
<div class="paragraph">
<p>Check out these links:</p>
</div>
<div class="paragraph">
<p><a href="https://msdn.microsoft.com/en-us/library/ms157894(v=VS.90).aspx" class="bare">https://msdn.microsoft.com/en-us/library/ms157894(v=VS.90).aspx</a> (<code>Application Exit</code>)</p>
</div>
<div class="paragraph">
<p><a href="https://msdn.microsoft.com/en-us/library/t2at9t47.aspx" class="bare">https://msdn.microsoft.com/en-us/library/t2at9t47.aspx</a> (<code>Exit</code> statements)</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_19_oct_2015_154010_ann_campbell_wrote">on 19 Oct 2015, 15:40:10 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>Fixed after collaboration with [~tamas.vajk]</p>
</div>
</div>
<div class="sect2">
<h3 id="_is_related_to_s1799">is related to: <a data-rspec-id="S1799" class="rspec-auto-link">S1799</a></h3>

</div>
</div>
</div>