<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Starting with Java 18 there is a possibility to write code snippets in javadoc comments. Previously the construction <code>&lt;pre&gt;{@code &#8230;&#8203; }&lt;/pre&gt;</code> was used for this. While you still can use the old approach, snippets give you more possibilities, like easy highlighting, replacing by substring or regex, extracting to a file, linking, etc.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">/**
 * The code prints "Hello, World!"
 * &lt;pre&gt;{@code                                 // Noncompliant
 *   System.out.println("Hello, World!");
 * }&lt;/pre&gt;
 */
public static void main(String... args) {
   ...
}</code></pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">/**
 * The code prints "Hello, World!"
 * {@snippet :
 *  public static void main(String... args) {
 *      System.out.println("Hello, World!");
 *  }
 * }
 */
public static void main(String... args) {
   ...
}</code></pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_see">See</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Oracleâ€™s Programmer&#8217;s Guide to Snippets: <a href="https://docs.oracle.com/en/java/javase/18/code-snippet/index.html">https://docs.oracle.com/en/java/javase/18/code-snippet/index.html</a></p>
</div>
<hr>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Replace this code example with a javadoc @snippet.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="ulist">
<ul>
<li>
<p>Primary: &lt;pre&gt;{@code &#8230;&#8203; }&lt;/pre&gt; construction</p>
</li>
</ul>
</div>
</div>
</div>
</div>