== How to fix it in aiohttp

=== Code examples
The following code uses basic authentication to send out an HTTP request to a protected endpoint.

==== Noncompliant code example

Authentication can be set directly through request headers.

[source,python,diff-id=101,diff-type=noncompliant]
----
from aiohttp import ClientSession
import base64

encoded_creds = base64.b64encode(f"{username}:{password}".encode()).decode()
async with ClientSession(
    headers={
        "Authorization": "Basic " + encoded_creds,
    }
) as session:
    async with session.get("http://api.example.com/foo") as response:  # Noncompliant
        # ...
----

[source,python,diff-id=102,diff-type=noncompliant]
----
from aiohttp import ClientSession
import base64

encoded_creds = base64.b64encode(f"{username}:{password}".encode()).decode()
async with ClientSession() as session:
    async with session.get(  # Noncompliant
        "http://api.example.com/foo"
        headers={
            "Authorization": "Basic " + encoded_creds,
        }
    ) as response:
        # ...
----

Authentication can also be set using an authentication class.

[source,python,diff-id=103,diff-type=noncompliant]
----
from aiohttp import ClientSession, BasicAuth
import base64

async with ClientSession(
    auth=BasicAuth(username, password),
) as session:
    async with session.get("http://api.example.com/foo") as response:  # Noncompliant
        # ...
----

==== Compliant solution

Use an authentication method other than basic authentication.

[source,python,diff-id=101,diff-type=compliant]
----
from aiohttp import ClientSession

async with ClientSession(
    headers={
        "Authorization": "Bearer " + access_token,
    }
) as session:
    async with session.get("http://api.example.com/foo") as response:
        # ...
----

[source,python,diff-id=102,diff-type=compliant]
----
from aiohttp import ClientSession

async with ClientSession() as session:
    async with session.get(
        "http://api.example.com/foo"
        headers={
            "Authorization": "Bearer " + access_token,
        }
    ) as response:
        # ...
----

There are no built-in authentication classes for token-based authentication.

=== How does this work?

include::../../common/fix/token-auth.adoc[]

include::../../common/fix/ssl.adoc[]