== How to fix it in HTTPX

=== Code examples
The following code uses basic authentication to send out an HTTP request to a protected endpoint.

==== Noncompliant code example

Authentication can be set directly through request headers.

[source,python,diff-id=201,diff-type=noncompliant]
----
import httpx
import base64

encoded_creds = base64.b64encode(f"{username}:{password}".encode()).decode()

response = httpx.get(  # Noncompliant
    "http://api.example.com/foo",
    headers = {
        "Authorization": "Basic " + encoded_creds,
    },
)
----

[source,python,diff-id=202,diff-type=noncompliant]
----
import httpx
import base64

encoded_creds = base64.b64encode(f"{username}:{password}".encode()).decode()

with httpx.Client(
    headers = {
        "Authorization": "Basic " + encoded_creds,
    },
) as client:
    response = client.get(  # Noncompliant
        "http://api.example.com/foo"
        # Alternative location for setting request headers
        headers = {
            "Authorization": "Basic " + encoded_creds,
        },
    )
----

[source,python,diff-id=203,diff-type=noncompliant]
----
import httpx
import base64

encoded_creds = base64.b64encode(f"{username}:{password}".encode()).decode()

with httpx.Client() as client:
    request = httpx.request(
        "http://api.example.com/foo",
        headers={
            "Authorization": "Basic " + encoded_creds,
        },
    )
    response = client.send(  # Noncompliant
        request
    )
----

Authentication can also be set using an authentication class.

[source,python]
----
import httpx

response = httpx.get(  # Noncompliant
    "http://api.example.com/foo",
    auth=httpx.BasicAuth(username, password),
)
----

[source,python]
----
import httpx

with httpx.Client(
    auth=httpx.BasicAuth(username, password),
) as client:
    response = client.get(  # Noncompliant
        "http://api.example.com/foo",
        # Alternative location for setting authentication
        auth=httpx.BasicAuth(username, password),
    )
----

[source,python]
----
import httpx

with httpx.Client() as client:
    request = httpx.Request("http://api.example.com/foo")
    response = client.send(  # Noncompliant
        request,
        auth=httpx.BasicAuth(username, password),
    )
----

==== Compliant solution

Use an authentication method other than basic authentication.

[source,python,diff-id=201,diff-type=compliant]
----
import httpx

response = httpx.get(
    "http://api.example.com/foo",
    headers = {
        "Authorization": "Bearer " + access_token,
    },
)
----

[source,python,diff-id=202,diff-type=compliant]
----
import httpx

with httpx.Client(
    headers = {
        "Authorization": "Bearer " + access_token,
    },
) as client:
    response = client.get(
        "http://api.example.com/foo"
        # Alternative location for setting request headers
        headers = {
            "Authorization": "Bearer " + access_token,
        },
    )
----

[source,python,diff-id=203,diff-type=compliant]
----
import httpx

with httpx.Client() as client:
    request = httpx.request(
        "http://api.example.com/foo",
        headers={
            "Authorization": "Bearer " + access_token,
        },
    )
    response = client.send(
        request
    )
----

There are no built-in authentication classes for token-based authentication.

=== How does this work?

include::../../common/fix/token-auth.adoc[]

include::../../common/fix/ssl.adoc[]