== How to fix it in Requests

=== Code examples
The following code uses basic authentication to send out an HTTP request to a protected endpoint.

==== Noncompliant code example

Authentication can be set directly through request headers.

[source,python,diff-id=301,diff-type=noncompliant]
----
import requests
import base64

encoded_creds = base64.b64encode(f"{username}:{password}".encode()).decode()

response = requests.get(  # Noncompliant
    "http://api.example.com/foo",
    headers={
        "Authorization": "Basic " + encoded_creds,
    },
)
----

[source,python,diff-id=302,diff-type=noncompliant]
----
import requests
import base64

encoded_creds = base64.b64encode(f"{username}:{password}".encode()).decode()

with requests.Session(
    headers={
        "Authorization": "Basic" + encoded_creds,
    },
) as session:
    request = requests.Request(
        "http://api.example.com/foo",
        # Alternative location for setting request headers
        headers={
            "Authorization": "Basic" + encoded_creds,
        },
    )
    response = requests.send(
        session.prepare_request(request)  # Noncompliant
    )
----

Authentication can also be set using an authentication class.

[source,python]
----
import requests

response = requests.get(  # Noncompliant
    "http://api.example.com/foo",
    auth=requests.HTTPBasicAuth(username, password),
)
----

[source,python]
----
import requests

with requests.Session(
    auth=requests.HTTPBasicAuth(username, password),
) as session:
    request = requests.Request(
        "http://api.example.com/foo",
        # Alternative location for setting authentication
        auth=requests.HTTPBasicAuth(username, password),
    )
    response = requests.send(
        session.prepare_request(request)  # Noncompliant
    )
----

==== Compliant solution

Use an authentication method other than basic authentication.

[source,python,diff-id=301,diff-type=compliant]
----
import requests

response = requests.get(  # Noncompliant
    "http://api.example.com/foo",
    headers={
        "Authorization": "Bearer " + access_token,
    },
)
----

[source,python,diff-id=302,diff-type=compliant]
----
import requests

with requests.Session(
    headers={
        "Authorization": "Bearer " + access_token,
    },
) as session:
    request = requests.Request(
        "http://api.example.com/foo",
        # Alternative location for setting request headers
        headers={
            "Authorization": "Bearer " + access_token,
        },
    )
    response = requests.send(
        session.prepare_request(request)  # Noncompliant
    )
----

There are no built-in authentication classes for token-based authentication.

=== How does this work?

include::../../common/fix/token-auth.adoc[]

include::../../common/fix/ssl.adoc[]