== How to fix it in Python Standard Library

=== Code examples
The following code uses basic authentication to send out an HTTP request to a protected endpoint.

==== Noncompliant code example

[source,python,diff-id=401,diff-type=noncompliant]
----
from urllib.request import Request, HTTPHandler
import base64

encoded_creds = base64.b64encode(f"{username}:{password}".encode()).decode()

request = Request(
    "http://api.example.com/foo",
    headers={
        "Authorization": "Basic " + encoded_creds,
    }
)
handler = HTTPHandler()
reponse = handler.http_open(request)  # Noncompliant
----

==== Compliant solution

[source,python,diff-id=401,diff-type=compliant]
----
from urllib.request import Request, HTTPHandler

request = Request(
    "http://api.example.com/foo",
    headers={
        "Authorization": "Bearer " + access_token,
    }
)
handler = HTTPHandler()
reponse = handler.http_open(request)
----

=== How does this work?

include::../../common/fix/token-auth.adoc[]

include::../../common/fix/ssl.adoc[]