== How to fix it in urllib3

=== Code examples
The following code uses basic authentication to send out an HTTP request to a protected endpoint.

==== Noncompliant code example

[source,python,diff-id=501,diff-type=noncompliant]
----
import urllib3
import base64

encoded_creds = base64.b64encode(f"{username}:{password}".encode()).decode()

response = urllib3.request(  # Noncompliant
    "GET",
    "http://api.example.com/foo",
    headers={
        "Authorization": "Basic " + encoded_creds,
    },
)
----

==== Compliant solution

[source,python,diff-id=501,diff-type=compliant]
----
import urllib3

response = urllib3.request(
    "GET",
    "http://api.example.com/foo",
    headers={
        "Authorization": "Bearer " + access_token,
    },
)
----

=== How does this work?

include::../../common/fix/token-auth.adoc[]

include::../../common/fix/ssl.adoc[]