This rule raises an issue when FastAPI route functions have query parameters that should be optional but lack proper Union type annotations or default values.

== Why is this an issue?

In FastAPI, query parameters are automatically validated based on their type annotations and default values. When a parameter is intended to be optional, it needs both a Union type annotation (like `Union[str, None]` or `Optional[str]`) and a default value (typically `None`) to work correctly.

Without the Union type annotation, FastAPI treats the parameter as required, even if you provide a default value. Without the default value, Python treats the parameter as mandatory, causing validation errors.

This creates a poor API experience where clients receive unexpected validation errors when they don't provide parameters that should be optional. The API documentation will also incorrectly show these parameters as required, misleading API consumers about the expected interface.

=== What is the potential impact?

API clients will receive validation errors when they don't provide parameters that should be optional. This leads to:

* Poor developer experience for API consumers
* Incorrect API documentation showing optional parameters as required
* Potential integration failures when clients assume parameters are optional
* Reduced API usability and adoption

== How to fix it in FastAPI

Add Union type annotation and default value to make the query parameter truly optional. Use `Union[Type, None] = None` or `Optional[Type] = None` to indicate the parameter can be either the specified type or None.

=== Code examples

==== Noncompliant code example

[source,python,diff-id=1,diff-type=noncompliant]
----
from fastapi import FastAPI

app = FastAPI()

@app.get("/items/{item_id}")
def read_item(item_id: int, q: str):  # Noncompliant
    return {"item_id": item_id, "q": q}
----

==== Compliant solution

[source,python,diff-id=1,diff-type=compliant]
----
from typing import Union
from fastapi import FastAPI

app = FastAPI()

@app.get("/items/{item_id}")
def read_item(item_id: int, q: Union[str, None] = None):
    return {"item_id": item_id, "q": q}
----

Add a default value when you already have the Union type annotation. The default value (typically None) is required to make the parameter optional in the function signature.

=== Code examples

==== Noncompliant code example

[source,python,diff-id=2,diff-type=noncompliant]
----
from typing import Union

@app.get("/items/{item_id}")
def read_item(item_id: int, q: Union[str, None]):  # Noncompliant
    return {"item_id": item_id, "q": q}
----

==== Compliant solution

[source,python,diff-id=2,diff-type=compliant]
----
from typing import Union

@app.get("/items/{item_id}")
def read_item(item_id: int, q: Union[str, None] = None):
    return {"item_id": item_id, "q": q}
----

Use the Optional type alias for cleaner code. `Optional[Type]` is equivalent to `Union[Type, None]` and can make the code more readable.

=== Code examples

==== Noncompliant code example

[source,python,diff-id=3,diff-type=noncompliant]
----
from fastapi import FastAPI

app = FastAPI()

@app.get("/search")
def search_items(query: str, limit: int):  # Noncompliant
    return {"query": query, "limit": limit}
----

==== Compliant solution

[source,python,diff-id=3,diff-type=compliant]
----
from typing import Optional
from fastapi import FastAPI

app = FastAPI()

@app.get("/search")
def search_items(query: Optional[str] = None, limit: Optional[int] = 10):
    return {"query": query, "limit": limit}
----

== Resources

=== Documentation

 * FastAPI Query Parameters - https://fastapi.tiangolo.com/tutorial/query-params/[Official FastAPI documentation on query parameters and optional parameters]

 * FastAPI Query Parameters and String Validations - https://fastapi.tiangolo.com/tutorial/query-params-str-validations/[Advanced query parameter handling and validation in FastAPI]

 * Python typing - Union - https://docs.python.org/3/library/typing.html#typing.Union[Python documentation on Union types for type annotations]
