<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Marking a parameter for output means that callers will expect its value to be updated with a result from the execution of the procedure. Failing to update the parameter before the procedure returns is surely an error.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-sql" data-lang="sql">CREATE OR REPLACE PROCEDURE greet(
  name     IN  VARCHAR2,
  greeting OUT VARCHAR2) -- Noncompliant
AS
  message VARCHAR2(45);
BEGIN
  SELECT 'Hello ' || RTRIM(name) INTO message FROM DUAL;
END;</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-sql" data-lang="sql">CREATE OR REPLACE PROCEDURE greet(
  name     IN  VARCHAR2,
  greeting OUT VARCHAR2)
AS
BEGIN
  SELECT 'Hello ' || RTRIM(name) INTO greeting FROM DUAL;
END;</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Update this output parameter or remove its output designation.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>the parameter name and the output keyword</p>
</div>
</div>
</div>
</div>