== Why is this an issue?

include::description.adoc[]

=== Noncompliant code example

[source,text]
----
switch(ch) 
{
  case 'a' :
    handleA();
    break;
  case 'b' :
    handleB();
    break;
  case 'c' :  // Noncompliant
  default:
    handleTheRest();
    break;
}
----


=== Compliant solution

[source,text]
----
switch(ch) 
{
  case 'a' :
    handleA();
    break;
  case 'b' :
    handleB();
    break;
  default:
    handleTheRest();
    break;
}
----

