This rule raises an issue when a shell function does not end with an explicit `return` statement.

== Why is this an issue?

Shell functions should end with explicit `return` statements to make their completion clear and provide proper exit status handling.

When a function doesn't have an explicit `return` statement, it implicitly returns the exit status of the last command executed. This behavior can be unpredictable and makes the function's intent unclear.

Explicit return statements serve several important purposes:

* **Clarity of Intent**: They make it obvious where and how a function terminates
* **Error Handling**: They allow functions to return specific exit codes (0 for success, non-zero for various error conditions)
* **Consistency**: They follow established shell scripting best practices and coding standards
* **Debugging**: They make it easier to trace function execution and identify issues

Without explicit return statements, functions may accidentally return unexpected exit codes based on the last command's result, potentially causing issues in calling code that checks the function's return value.

=== What is the potential impact?

Functions without explicit return statements can lead to unpredictable behavior when their return values are checked by calling code. This can cause:

* **Logic Errors**: Calling code may make incorrect decisions based on unexpected return values
* **Difficult Debugging**: Issues may be harder to trace when function exit points are implicit
* **Maintenance Problems**: Future modifications to the function may inadvertently change its return behavior
* **Inconsistent Code**: Mixed use of explicit and implicit returns reduces code readability

== How to fix it

Add an explicit `return` statement at the end of the function. Use `return 0` for successful completion or an appropriate non-zero value for error conditions.

=== Code examples

==== Noncompliant code example

[source,bash,diff-id=1,diff-type=noncompliant]
----
my_function() {
  local param="$1"
  echo "processing $param"
}  # Noncompliant
----

==== Compliant solution

[source,bash,diff-id=1,diff-type=compliant]
----
my_function() {
  local param="$1"
  echo "processing $param"
  return 0
}
----

== Resources

=== Documentation

 * Bash Manual - Shell Functions - https://www.gnu.org/software/bash/manual/html_node/Shell-Functions.html[Official Bash manual section on shell functions and return statements]
 * LinuxCommand.org - Bash Script Coding Style Guide - https://linuxcommand.org/lc3_adv_standards.php[Comprehensive shell scripting coding standards including function best practices]

=== Standards

 * POSIX.1-2017 - Shell Command Language - https://pubs.opengroup.org/onlinepubs/9699919799/utilities/V3_chap02.html#tag_18_09_05[POSIX standard for shell function definitions and return statements]

