<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when the <code>tf.reshape</code> function&#8217;s <code>tensor</code> and <code>shape</code> arguments have different number of elements.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>The <code>tf.reshape</code> function accepts two arguments - <code>tensor</code> and <code>shape</code>.
The <code>tensor</code> argument is a <code>tf.Tensor</code> of values that should be reshaped.
The <code>shape</code> argument is an array used to explain the shape of the result.
The <code>tf.reshape</code> function returns a new <code>tf.Tensor</code> that has the same values as <code>tensor</code> in the same order, except with a new shape given by <code>shape</code>.</p>
</div>
<div class="paragraph">
<p>Calling the <code>tf.reshape</code> function with <code>tensor</code> and <code>shape</code> having different sizes leads to <code>IllegalArgumentError`</code></p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Make sure the arguments <code>tensor</code> and <code>shape</code> have the same size when passed to the <code>tf.reshape</code> function call.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">import tensorflow as tf

tf.reshape([1, 2, 3, 4], [3, 2])  # Nonpompliant: InvalidArgumentError: Input to reshape is a tensor with 4 values, but the requested shape has 6</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">import tensorflow as tf

tf.reshape([1, 2, 3, 4], [2, 2])  # OK</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Tensorflow documentation - <a href="https://www.tensorflow.org/api_docs/python/tf/reshape">tf.reshape</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>