This rule raises an issue when `where(...).take` is used to retrieve a single record by attributes.

== Why is this an issue?

Using `where(...).take` for single record retrieval is less semantic and readable compared to `find_by`. The `find_by` method clearly expresses the intent to find a single record by specific attributes and return `nil` if not found.

The `where.take` pattern requires readers to understand that:

* `where` creates a query scope
* `take` retrieves the first record from that scope
* The combination returns `nil` if no record matches

In contrast, `find_by` makes this intent explicit and follows Rails conventions. It's the idiomatic way to retrieve a single record by attributes in ActiveRecord.

Using `find_by` also makes code more consistent with Rails style guidelines and community best practices, improving maintainability and reducing cognitive load for developers familiar with Rails conventions.

=== What is the potential impact?

This issue affects code readability and maintainability. While both approaches have similar performance characteristics, using non-idiomatic patterns can:

* Make code harder to understand for developers familiar with Rails conventions
* Reduce consistency across the codebase
* Increase cognitive load when reading and maintaining the code

The impact is primarily on code quality rather than functionality, as both patterns produce the same results.

== How to fix it

Replace `where(...).take` with `find_by(...)` to make the intent clearer and follow Rails conventions.

=== Code examples

==== Noncompliant code example

[source,ruby,diff-id=1,diff-type=noncompliant]
----
User.where(email: email).take # Noncompliant
----

==== Compliant solution

[source,ruby,diff-id=1,diff-type=compliant]
----
User.find_by(email: email)
----

== Resources

=== Documentation

 * Rails Guides - Active Record Query Interface - https://guides.rubyonrails.org/active_record_querying.html#find_by[Official Rails documentation on using find_by for single record retrieval]

 * ActiveRecord::FinderMethods#find_by - https://api.rubyonrails.org/classes/ActiveRecord/FinderMethods.html#method-i-find_by[API documentation for the find_by method]
