== Why is this an issue?

Externally described files standardize file access, and result in simpler, easier to read code.


=== Noncompliant code example

[source,rpg]
----
     FEPBLMSL2  IF   F  100        DISK    INFSR(*PSSR)                                                                                         
     F*                                                                                                                                                                                                              
     F*                                                                                                                    
----


=== Compliant solution

[source,rpg]
----
     F*                                                                                                                    
     FIPBDCCP   IF   E           K DISK    INFSR(*PSSR)                                                                    
     F*                                                                                                                                                                                          

     FEP210     CF   E             WORKSTN INFSR(*PSSR)                                                                 
     F                                     SFILE(EP21003S:RELN03)                                                       


     FEP471R1   O    E             PRINTER INFSR(*PSSR)                                                              
     F                                     USROPN                                                                    
     F                                     INFDS(W1SF01)                                                             
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Use an "E" indicator for this file and remove any related "I" lines.


endif::env-github,rspecator-view[]
