Externally described files standardize file access, and result in simpler, easier to read code.

== Noncompliant Code Example

----
     FEPBLMSL2  IF   F  100        DISK    INFSR(*PSSR)                                                                                         
     F*                                                                                                                                                                                                              
     F*                                                                                                                    
----

== Compliant Solution

----
     F*                                                                                                                    
     FIPBDCCP   IF   E           K DISK    INFSR(*PSSR)                                                                    
     F*                                                                                                                                                                                          

     FEP210     CF   E             WORKSTN INFSR(*PSSR)                                                                 
     F                                     SFILE(EP21003S:RELN03)                                                       


     FEP471R1   O    E             PRINTER INFSR(*PSSR)                                                              
     F                                     USROPN                                                                    
     F                                     INFDS(W1SF01)                                                             
----
