== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,go]
----
if boolFunc() || false { 
    // ...
}

flag := x && true
----

=== Compliant solution

[source,go]
----
if boolFunc() {
    // ...
}

flag := x
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
