include::../description.adoc[]

== Noncompliant Code Example

----
if boolFunc() || false { 
    // ...
}

flag := x && true
----

== Compliant Solution

----
if boolFunc() {
    // ...
}

flag := x
----

ifdef::env-github,rspecator-view[]
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
