:true: true
:false: false
:ops: !, &&, ||, ==, !=
== Why is this an issue?

include::../description.adoc[]

== How to fix it

include::../how-to-fix-it.adoc[]

=== Code examples

==== Noncompliant code example

[source,go,diff-id=1,diff-type=noncompliant]
----
if boolFunc() || false { 
    // ...
}

flag := x && true
----

==== Compliant solution

[source,go,diff-id=1,diff-type=compliant]
----
if boolFunc() {
    // ...
}

flag := x
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''

endif::env-github,rspecator-view[]
