<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>The complexity of an expression is defined by the number of <code>&amp;&amp;</code>, <code>||</code> and <code>condition ? ifTrue : ifFalse</code> operators it contains.</p>
</div>
<div class="paragraph">
<p>A single expression&#8217;s complexity should not become too high to keep the code readable.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="paragraph">
<p>With the default threshold value 3.</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">if (((condition1 &amp;&amp; condition2) || (condition3 &amp;&amp; condition4)) &amp;&amp; condition5) { ... }</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">if ((myFirstCondition() || mySecondCondition()) &amp;&amp; myLastCondition()) { ... }</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Reduce the number of conditional operators ({}) used in the expression (maximum allowed {}).</p>
</div>
</div>
<div class="sect2">
<h3 id="_parameters">Parameters</h3>
<div class="sidebarblock">
<div class="content">
<div class="title">max</div>
<div class="listingblock">
<div class="content">
<pre>3</pre>
</div>
</div>
<div class="paragraph">
<p>Maximum number of allowed conditional operators in an expression</p>
</div>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_is_duplicated_by_s132">is duplicated by: <a data-rspec-id="S132" class="rspec-auto-link">S132</a></h3>

</div>
<div class="sect2">
<h3 id="_on_25_jul_2013_110532_dinesh_bolkensteyn_wrote">on 25 Jul 2013, 11:05:32 Dinesh Bolkensteyn wrote:</h3>
<div class="paragraph">
<p>Implemented by http://jira.codehaus.org/browse/SONARJAVA-228</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_7_jan_2014_113126_freddy_mallet_wrote">on 7 Jan 2014, 11:31:26 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>Implemented by http://jira.codehaus.org/browse/SONARPLUGINS-3373 for JavaScript</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_12_feb_2014_221005_freddy_mallet_wrote">on 12 Feb 2014, 22:10:05 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>Is implemented by http://jira.codehaus.org/browse/SONARPLUGINS-3474 for PHP</p>
</div>
</div>
</div>
</div>