This rule raises an issue when issue suppression comments have incorrect format or syntax.

== Why is this an issue?

Issue suppression comments like `# NOSONAR` and `# noqa` are essential tools for controlling code analysis. When these comments have incorrect syntax, they may not work as expected, leading to confusion about which issues are actually suppressed.

Python code analysis supports two main suppression formats:
- `# NOSONAR` - SonarQube's suppression comment
- `# noqa` - Python's standard "no quality assurance" comment

Each format has specific syntax rules. When these rules are violated, the suppression might fail silently or behave unexpectedly, making it unclear whether issues are intentionally ignored or accidentally unsuppressed.

=== What is the potential impact?

Incorrectly formatted suppression comments can lead to unintended code analysis behavior. Issues that developers think are suppressed might still be reported, while malformed syntax might cause the analyzer to ignore more issues than intended. This creates confusion during code review and reduces confidence in the analysis results.

== How to fix it

Fix the syntax of issue suppression comments to follow the correct format.

For `# NOSONAR`:
- Use `# NOSONAR` alone to suppress all issues on the line
- Use `# NOSONAR()` with empty parentheses to suppress all issues
- Use `# NOSONAR(rule1, rule2)` to suppress specific rules
- Ensure no empty rule keys in comma-separated lists
- Close all parentheses properly

For `# noqa`:
- Use `# noqa` alone to suppress all issues on the line
- Use `# noqa: rule1,rule2` to suppress specific rules (with or without spaces after colon)
- Ensure no empty rule keys in comma-separated lists
- Use colon (`:`) not other punctuation

=== Code examples

==== Noncompliant code example
[source,python,diff-id=1,diff-type=noncompliant]
----
def example():
    x = 1  # NOSONAR(  # Noncompliant
    y = 2  # NOSONAR(a,)  # Noncompliant
    z = 3  # NOSONAR)(  # Noncompliant
    a = 4  # noqa: ,rule1  # Noncompliant
    b = 5  # noqa- rule1,rule2  # Noncompliant
----

==== Compliant solution
[source,python,diff-id=1,diff-type=compliant]
----
def example():
    x = 1  # NOSONAR
    y = 2  # NOSONAR(a)
    z = 3  # NOSONAR
    a = 4  # noqa: rule1
    b = 5  # noqa: rule1,rule2
----

== Resources

=== Documentation
* SonarQube documentation - https://docs.sonarqube.org/latest/user-guide/issues/#header-4[Managing your code issues]
* Flake8 documentation - https://flake8.pycqa.org/en/latest/user/violations.html#in-line-ignoring-errors[In-line Ignoring Errors]


ifdef::env-github,rspecator-view[]

== Implementation Specification
(visible only on this page)

=== Message

Fix the syntax of this issue suppression comment.

=== Highlighting

* Primary location: the entire suppression comment

endif::env-github,rspecator-view[]