<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>C&#43;&#43;20 introduces the ranges library. A range is a group of items that can be iterated over. It should provide a <code>begin</code> iterator and an <code>end</code> sentinel. All the existing STL containers are ranges.</p>
</div>
<div class="paragraph">
<p>This new library makes working with the STL library much more powerful by introducing range adaptors and much less verbose by introducing a constrained version of most algorithms in the namespace <code>std::ranges</code>. Before the ranges library, you had to specify the <code>begin</code> and <code>end</code> iterators when calling the STL algorithms, even when you want to iterate over the whole container.</p>
</div>
<div class="paragraph">
<p>This rule focuses on making your code less verbose and more readable by suggesting range-based over iterator-based algorithms when convenient.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">auto printEven = [](auto i) {
  if (i % 2 == 0) {
    std::cout &lt;&lt; i;
  }
};

void f1(const std::vector&lt;int&gt;&amp; v) {
  std::for_each(v.begin(), v.end(), printEven); // Noncompliant
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">auto printEven = [](auto i) {
  if (i % 2 == 0) {
    std::cout &lt;&lt; i;
  }
};

void f2(const std::vector&lt;int&gt;&amp; v) {
  std::ranges::for_each(v, printEven); // Compliant
}</code></pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p>C&#43;&#43; reference - <a href="https://en.cppreference.com/w/cpp/ranges">Ranges library</a></p>
</li>
</ul>
</div>
</div>
</div>