<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Whenever a CICS command is used with a <code>NOHANDE</code> or <code>RESP</code> option, the default CICS exception handling is disabled. The correct approach then is to ensure that every possible exception is handled correctly directly in the code and to do this, you need to examine the <code>RESP</code> value or the <code>EIBRESP</code> field value.</p>
</div>
<div class="paragraph">
<p>It is possible to compare the <code>RESP</code> and <code>EIBRESP</code> field values to hard-coded numbers or variables containing numeric values; however, this makes the code difficult to read and maintain. It is recommended to use instead the <code>DFHRESP</code> built-in translator function, which enables the use of the equivalent symbolic values.</p>
</div>
<div class="paragraph">
<p>This rule raises an issue when the <code>EIBRESP</code> field is compared directly to a variable or hard-coded numeric value that is not wrapped in the <code>DFHRESP</code> function.</p>
</div>
<div class="paragraph">
<p>This rule does not handle <code>RESP</code> values for now.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-pli" data-lang="pli">if EIBRESP=36 then /* Noncompliant */
end;

if EIBRESP=MAPFAIL then /* Noncompliant */
end;

select (EIBRESP);
    when(36) ...;  /* Noncompliant */
end;</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-pli" data-lang="pli">if EIBRESP=DFHRESP(36) then
end;

if EIBRESP=DFHRESP(MAPFAIL) then
end;

select (EIBRESP);
    when(DFHRESP(MAPFAIL)) ...;
end;</code></pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p><a href="https://www.ibm.com/docs/en/cics-ts/6.1?topic=conditions-handling-exception-by-inline-code">Handling exception conditions by in-line code</a></p>
</li>
<li>
<p><a href="https://www.ibm.com/docs/en/cics-ts/5.2?topic=development-exec-interface-block-fields">EXEC interface block fields</a></p>
</li>
<li>
<p><a href="https://www.ibm.com/docs/en/cics-ts/6.1?topic=code-how-use-resp-resp2-options">How to use the RESP and RESP2 options</a></p>
</li>
</ul>
</div>
<hr>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>You are comparing "EIBRESP" directly with %s. Replace this value with a call to "DFHRESP(%s)".</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>The line where the EIBRESP field is.</p>
</div>
</div>
</div>
</div>