This is an issue when ternary operators or if statements are used for null-safe default assignments, or when ternary expressions have identical boolean and true expressions that can be simplified.

== Why is this an issue?

The Elvis operator (`?:`) is a Groovy-specific feature that provides a more concise and readable way to handle null-safe operations and default value assignments.

When you use a ternary operator like `user.name ? user.name : 'Anonymous'`, you're duplicating the expression `user.name`. This duplication creates several problems:

* **Code verbosity**: The same expression appears twice, making the code longer than necessary
* **Maintenance risk**: If you need to change the expression (e.g., from `user.name` to `user.displayName`), you must remember to update it in both places
* **Reduced readability**: The duplicated expression makes it harder to quickly understand the intent

Similarly, if statements that check for null or falsy values and assign defaults can be verbose:

```groovy
if (!x) {
    x = 'some value'
}
```

The Elvis operator eliminates these issues by providing a clean, single-expression syntax that clearly expresses the intent: "use this value if it's truthy, otherwise use the default."

=== What is the potential impact?

The duplicated expressions in ternary operators create a higher risk of introducing bugs during refactoring, as developers might forget to update both occurrences of the same expression.

== How to fix it

Replace ternary operators where the condition and true branch are identical with the Elvis operator.

=== Code examples

==== Noncompliant code example

[source,groovy,diff-id=1,diff-type=noncompliant]
----
displayName = user.name ? user.name : 'Anonymous' // Noncompliant
----

==== Compliant solution

[source,groovy,diff-id=1,diff-type=compliant]
----
displayName = user.name ?: 'Anonymous'
----

== Resources

=== Documentation

 * Groovy documentation - https://groovy-lang.org/operators.html#_elvis_operator[Elvis operator]
