">" and "<" can be used in Java both as comparison operators or when working with generics.  To prevent any confusion, extra whitespaces around these characters should be avoided when defining types.


== Noncompliant Code Example

[source,text]
----
public class Foo < A > {...}
...
List < String > myList = new ArrayList < String > ();
...
boolean same = Util. < Integer, String > compare(p1, p2);
----


== Compliant Solution

----
public class Foo<A> {...}
...
List<String> myList = new ArrayList<String>();
...
boolean same = Util.<Integer, String>compare(p1, p2);
----

