== Why is this an issue?

In Kotlin, `if` and `when` statements are expressions, which means they return a value.
This allows for a more concise and functional programming style with less cognitive complexity,
because it results in fewer return points and fewer variable assignments in a function.

If both branches of an `if` statement end with a `return` statement,
the `if` statement should be used as an expression for a `return` statement instead.

=== What is the potential impact?

==== Readability and Understanding

This change makes it easier to understand what a function does, because its complexity will be reduced.
This is because the function does now contain fewer return points that the developer needs to keep track of.

== How to fix it

Use the `if` statement as an expression for a `return` statement.
Remove the `return` keyword from the end of the `if` branch and the `else` branch and move it before the `if` keyword.

=== Code examples

==== Noncompliant code example

[source,kotlin,diff-id=1,diff-type=noncompliant]
----
fun Int.signToString(): String {
    if (this >= 0) {
        return "positive"
    } else {
        return "negative"
    }
}
----

==== Compliant solution

[source,kotlin,diff-id=1,diff-type=compliant]
----
fun Int.signToString(): String {
    return if (this >= 0) {
        "positive"
    } else {
        "negative"
    }
}
----

==== Noncompliant code example

[source,kotlin,diff-id=1,diff-type=noncompliant]
----
fun compareWithSideEffects(a: Float, b: Float): Int {
    if (a < 0) {
        foo()
        return -1
    } else if (a > b) {
        bar()
        return 1
    } else {
        return 0
    }
}
----

==== Compliant solution

[source,kotlin,diff-id=1,diff-type=compliant]
----
fun compareWithSideEffects(a: Float, b: Float): Int {
    return if (a < 0) {
        foo()
        -1
    } else if (a > b) {
        bar()
        1
    } else {
        0
    }
}
----

== Resources

=== Documentation

* https://kotlinlang.org/docs/control-flow.html#if-expression[Kotlin Docs, If expression]

=== Articles & blog posts

* https://www.baeldung.com/kotlin/if-else-expression[Baeldung, If-Else Expression in Kotlin]
* https://www.sonarsource.com/resources/cognitive-complexity/#:~:text=This%20paper%20describes%20Cognitive%20Complexity,Complexity%20precedents%20with%20human%20assessment[G. Ann Campbell, Cognitive Complexity]