== Why is this an issue?

According to Spring documentation, the `@Scheduled` annotation can only be applied to methods without arguments.
Applying @Scheduled to a method with arguments will result in a runtime error.

== How to fix it

Transform method annotated with `@Scheduled` into a no-arg method.

=== Code examples

==== Noncompliant code example

[source,java,diff-id=1,diff-type=noncompliant]
----
public class ExampleService {

    @Scheduled(fixedRate = 5000)
    public void scheduledTask(String param) { // non compliant, method has an argument. It will raise a runtime error.
        // Task implementation
    }
}
----

==== Compliant solution

[source,java,diff-id=1,diff-type=compliant]
----
public class ExampleService {

    @Scheduled(fixedRate = 5000)
    public void scheduledTask() { // compliant, no-arg method
        // Task implementation
    }
}
----

== Resources
=== Documentation
* Spring - https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/scheduling/annotation/Scheduled.html[scheduled]
