== Why is this an issue?

Java 21 introduces the new Sequenced Collections API, which is applicable to all collections with a defined sequence on their elements, such as `LinkedList`, `TreeSet`, and others (* https://openjdk.org/jeps/431[JEP 431]).
For projects using Java 21 and onwards, this API should be utilized instead of workaround implementations that were necessary prior to Java 21.

This rule identifies instances where a workaround is used to iterate over a collection in reverse order, instead of utilizing the reversed view of the collection for iteration.

== How to fix it

Replace the reported statement with a forward-iteration over the reversed view of the collection.

=== Code examples

==== Noncompliant code example

[source,java,diff-id=1,diff-type=noncompliant]
----
void printLastToFirst(List<String> list) {
  for (var it = list.listIterator(list.size()); it.hasPrevious();) {
    var element = it.previous();
    System.out.println(element);
  }
}
----

==== Compliant solution

[source,java,diff-id=1,diff-type=compliant]
----
void printLastToFirst(List<String> list) {
  for (var element: list.reversed()) {
    System.out.println(element);
  }
}
----

== Resources

* https://docs.oracle.com/en/java/javase/21/docs/api/java.base/java/util/SequencedCollection.html[Interface SequencedCollection - Java SE 21 API Documentation]
* https://openjdk.org/jeps/431[JEP 431: Sequenced Collections]
* https://docs.oracle.com/en/java/javase/21/core/creating-sequenced-collections-sets-and-maps.html#GUID-DCFE1D88-A0F5-47DE-A816-AEDA50B97523[Creating Sequenced Collections, Sets, and Maps]
