<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when a public method uses 'def' for parameter types or return types instead of explicit type declarations.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Using 'def' in public method signatures reduces API clarity and makes it difficult for users to understand the expected parameter types and return types. While 'def' is acceptable for local variables within method scope, it becomes problematic in public APIs because:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>'def' gets converted to 'Object' in the method signature, obscuring the actual expected types</p>
</li>
<li>
<p>API consumers cannot easily determine what types to pass as arguments</p>
</li>
<li>
<p>IDE support and code completion become less effective</p>
</li>
<li>
<p>The method contract becomes unclear, leading to potential runtime errors</p>
</li>
<li>
<p>Documentation and maintainability suffer when type information is implicit</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>The Groovy documentation explicitly states that omitting types is considered bad practice for public APIs, as it makes the code harder to understand and use correctly.</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="paragraph">
<p>Poor API usability, increased likelihood of runtime type errors, reduced IDE support, and decreased code maintainability. Users of the API may struggle to understand how to call the method correctly, leading to integration issues and debugging difficulties.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Replace 'def' with explicit type declarations for both parameters and return types in public methods. Use the most specific type that accurately represents the expected input or output.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-groovy" data-lang="groovy">public def processData(def input) {
    return input.toString().toUpperCase()
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-groovy" data-lang="groovy">public String processData(Object input) {
    return input.toString().toUpperCase()
}</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Groovy Language Documentation - Optional Typing - <a href="https://groovy-lang.org/semantics.html#_optional_typing">Official Groovy documentation explaining when to use explicit types vs 'def' keyword</a></p>
</li>
<li>
<p>Groovy Best Practices - Method Signatures - <a href="https://groovy-lang.org/style-guide.html">Style guide recommendations for writing clear and maintainable Groovy code</a></p>
</li>
</ul>
</div>
</div>
<div class="sect2">
<h3 id="_standards">Standards</h3>
<div class="ulist">
<ul>
<li>
<p>Clean Code - Chapter 3: Functions - <a href="https://www.oreilly.com/library/view/clean-code-a/9780136083238/">Principles for writing clean, understandable function signatures</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>