include::../description.adoc[]

== Noncompliant Code Example

With the default regular expression <code>^[a-z][a-zA-Z0-9]*$</code>:

----
function DoSomething(){...}  // Noncompliant
----

== Compliant Solution

----
function doSomething(){...}
----
