== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

With the default regular expression ``++^[a-z][a-zA-Z0-9]*$++``:

[source,javascript]
----
function DoSomething(){...}  // Noncompliant
----

=== Compliant solution

[source,javascript]
----
function doSomething(){...}
----

=== Exceptions

This rule ignores React Functional Components, which are JavaScript functions named with a capital letter and returning a React element (JSX syntax).


[source,javascript]
----
function Welcome() {
  const greeting = 'Hello, World!';

  // ...

  return (
    <div className="Welcome">
      <p>{greeting}</p>
    </div>
  );
}
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::parameters.adoc[]

'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
