Favicons are shown for example in the browser's address bar, bookmark list, or tabs.

They enable users to quickly identify and recognize websites.


== Noncompliant Code Example

----
<head>                                                                  <!-- Noncompliant -->
  <title>...<title>
</head>
----


== Compliant Solution

----
<head>
  <title>...<title>
  <link rel="shortcut icon" href="http://example.com/myicon.ico" />
</head>
----

