include::../../../shared_content/secrets/description.adoc[]

== Why is this an issue?

include::../../../shared_content/secrets/rationale.adoc[]

If an attacker gains access to a Grafana personal access token or Granafa Cloud token, they might be able to compromise the Grafana environment linked to this token. By doing so, it might be possible for business-critical data to be leaked by the attacker.

=== What is the potential impact?

Depending on the permissions given to the secret, the impact might range from the compromise of the data of some dashboards to a full takeover of the Grafana environment.  

:service_name: Grafana

include::../../../shared_content/secrets/impact/data_compromise.adoc[]

include::../../../shared_content/secrets/impact/dataviz_takeover.adoc[]

== How to fix it

include::../../../shared_content/secrets/fix/revoke.adoc[]

include::../../../shared_content/secrets/fix/vault.adoc[]

=== Code examples

==== Noncompliant code example

[source,python,diff-id=1,diff-type=noncompliant]
----
import requests

token = 'glsa_geygSnIfuK5vBG0KgaflRCQfIb8mzaM7_b0999d91'  # Noncompliant
response = requests.get('https://grafana.example.org/api/dashboards/home', headers={
    'Authorization': f'Bearer {token}',
    'Content-Type': 'application/json'
})
----

==== Compliant solution

[source,python,diff-id=1,diff-type=compliant]
----
import requests

token = os.getenv('GRAFANA_SERVICE_ACCOUNT_TOKEN')
response = requests.get('https://grafana.example.org/api/dashboards/home', headers={
    'Authorization': f'Bearer {token}',
    'Content-Type': 'application/json'
})
----

//=== How does this work?

//=== Pitfalls

//=== Going the extra mile

== Resources

=== Documentation

Grafana Documentation - https://grafana.com/docs/grafana/latest/administration/service-accounts/[Service Accounts]

include::../../../shared_content/secrets/resources/standards.adoc[]
