The rule S1279 ("DISPLAY" should not be used) is fully preventing the usage of the ``++DISPLAY++`` statement whereas in same cases it may be useful to output some data to SYSOUT. However, the output of a call to ``++DISPLAY UPON CONSOLE++`` is unlikely to be viewed by someone because the output is displayed on the console and nowadays no one is in front of the console to see if something is happening.


== Noncompliant Code Example

----
PROCEDURE DIVISION.
   ...
   DISPLAY "Error Occurred!!!" UPON CONSOLE *> Noncompliant
   ...
----



ifdef::env-github,rspecator-view[]
'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
