== Why is this an issue?

The rule S1279 ("DISPLAY" should not be used) is fully preventing the usage of the ``++DISPLAY++`` statement whereas in same cases it may be useful to output some data to SYSOUT. However, the output of a call to ``++DISPLAY UPON CONSOLE++`` is unlikely to be viewed by someone because the output is displayed on the console and nowadays no one is in front of the console to see if something is happening.


=== Noncompliant code example

[source,cobol]
----
PROCEDURE DIVISION.
   ...
   DISPLAY "Error Occurred!!!" UPON CONSOLE *> Noncompliant
   ...
----



ifdef::env-github,rspecator-view[]
'''
== Comments And Links
(visible only on this page)

=== duplicates: S1279

=== on 27 Sep 2018, 15:25:56 Pierre-Yves Nicolas wrote:
RSPEC-1279 has a parameter named "checkOnlyDisplayUponConsole".

=== on 23 Nov 2018, 13:57:23 Alexandre Gigleux wrote:
This is a duplicate of RSPEC-1279. By default RSPEC-1279 is containing a parameter ``++checkOnlyDisplayUponConsole++`` that is ON and prevent the use of "DISPLAY UPON CONSOLE".

endif::env-github,rspecator-view[]
