== Why is this an issue?

Using the ``++this++`` keyword inside the scope of an object to refer to the object's properties and methods yields cleaner, clearer code, and helps avoid confusion when there are variables or functions outside the object scope with the same or similar names.


=== Noncompliant code example

[source,javascript]
----
function Person(name, birthdate) {
  this.name = name;
  this.birthdate = birthdate;

  get name() {
    return name; // Noncompliant
  }
}
----


=== Compliant solution

[source,javascript]
----
function Person(name, birthdate) {
  this.name = name;
  this.birthdate = birthdate;

  get name() {
    return this.name;
  }
}
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
