:operationName: method
:visibility: private

include::../why.adoc[]

=== Code examples

==== Noncompliant code example

[source,apex]
----
public class Server {
    public void start() {  // Compliant, publicly available
        log('start');
    }

    private void clear() { // Noncompliant, not used anywhere
        
    }

    private void log(String msg) {  // Compliant, called from 'start()'
        System.debug(msg);
    }
}
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
