<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>When you annotate an <a href="https://learn.microsoft.com/en-us/dotnet/api/system.enum">Enum</a> with the <a href="https://learn.microsoft.com/en-us/dotnet/api/system.flagsattribute">Flags attribute</a>, you must not rely on the values that are automatically set by the language to the <code>Enum</code> members, but you should define the enumeration constants in powers of two (1, 2, 4, 8, and so on). Automatic value initialization will set the first member to zero and increment the value by one for each subsequent member. As a result, you wonâ€™t be able to use the enum members with bitwise operators.</p>
</div>
<div class="sect2">
<h3 id="_exceptions">Exceptions</h3>
<div class="paragraph">
<p>The default initialization of <code>0, 1, 2, 3, 4, &#8230;&#8203;</code> matches <code>0, 1, 2, 4, 8 &#8230;&#8203;</code> in the first three values, so no issue is reported if the first three members of the enumeration are not initialized.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Define enumeration constants in powers of two, that is, 1, 2, 4, 8, and so on.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-vbnet" data-lang="vbnet">&lt;Flags()&gt;
Enum FruitType    ' Non-Compliant
  None
  Banana
  Orange
  Strawberry
End Enum

Module Module1
  Sub Main()
    Dim bananaAndStrawberry = FruitType.Banana Or FruitType.Strawberry
    Console.WriteLine(bananaAndStrawberry.ToString()) ' Will display only "Strawberry"
  End Sub
End Module</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-vbnet" data-lang="vbnet">&lt;Flags()&gt;
Enum FruitType    ' Compliant
  None = 0
  Banana = 1
  Orange = 2
  Strawberry = 4
End Enum

Module Module1
  Sub Main()
    Dim bananaAndStrawberry = FruitType.Banana Or FruitType.Strawberry
    Console.WriteLine(bananaAndStrawberry.ToString()) ' Will display "Banana, Strawberry"
  End Sub
End Module</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p><a href="https://learn.microsoft.com/en-us/dotnet/api/system.enum">Enum Class</a></p>
</li>
<li>
<p><a href="https://learn.microsoft.com/en-us/dotnet/api/system.flagsattribute">FlagsAttribute Class</a></p>
</li>
</ul>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Initialize all the members of this "Flags" enumeration.</p>
</div>
</div>
</div>
</div>