<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Argument-dependent lookup (ADL) adds additional associated namespaces to the set of scopes searched when lookup is performed for the names of called functions. For function templates, ADL is performed at the point of instantiation of the function template, and so it is possible that a function declared after the template may be called.</p>
</div>
<div class="paragraph">
<p>To ensure that ADL does not take place when calling a function with a dependent argument, the postfix-expression denoting the called function can either be a qualified name or a parenthesized expression.</p>
</div>
<div class="paragraph">
<p>Operators with dependent types may also have this problem. In order to avoid ADL in these examples, operators should not be overloaded, or the calls should be changed to use explicit function call syntax and a qualified name or parenthesized expression used, as above.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">void b ( int32_t );

template &lt;typename T&gt;
void f ( T const &amp; t )
{
  b ( t ); // Non-compliant - Calls NS::b declared after f
  ::b ( t ); // Compliant - Calls ::b
  ( b ) ( t ); // Compliant - Calls ::b

  t == t; // Noncompliant - Calls NS::operator== declared after f
  ::operator ==( t, t ); // Compliant - Calls built-in operator==
  ( operator == &lt;T&gt; ) ( t, t ); // Compliant - Calls built-in operator==
}

namespace NS
{
  struct A
  {
    operator int32_t ( ) const;
  };
  void b ( A const &amp; a );
  bool operator== ( A const &amp;, A const &amp; );
}

int main ( )
{
  NS::A a;
  f ( a );
}</code></pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p>MISRA C&#43;&#43;:2008, 14-6-2</p>
</li>
</ul>
</div>
<hr>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>\[Operator|Function] "xxx" is declared after the template declaration.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_24_oct_2014_153024_ann_campbell_wrote">on 24 Oct 2014, 15:30:24 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~samuel.mercier]</p>
</div>
<div class="ulist">
<ul>
<li>
<p>bug or pitfall</p>
</li>
<li>
<p>please use standard section names</p>
</li>
<li>
<p>please incorporate the text between the code samples either into the description or the code itself in the form of comments.</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>Also, the message reads as a bit disembodied. I.e. not attached to either the problem function or the template declaration</p>
</div>
</div>
</div>
</div>