Both ``++php_sapi_name()++`` and the ``++PHP_SAPI++`` constant give the same value. But calling the method is less efficient that simply referencing the constant. 


== Noncompliant Code Example

[source,php]
----
if (php_sapi_name() == 'test') { ... } 
----


== Compliant Solution

[source,php]
----
if (PHP_SAPI == 'test') { ... } 
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

endif::env-github,rspecator-view[]
