== How to fix it in Java Net API

=== Code examples

include::../../common/fix/code-rationale.adoc[]

==== Noncompliant code example

[source,kotlin,diff-id=1,diff-type=noncompliant]
----
@GetMapping("/user")
  fun getUser(@RequestParam(value = "id") id: String): String {
    val url = URL("http://example.com/api/user/" + id) // Noncompliant
    val connection = url.openConnection()
}
----

==== Compliant solution

[source,kotlin,diff-id=1,diff-type=compliant]
----
import java.net.URLEncoder
import java.nio.charset.StandardCharsets

@GetMapping("/user")
  fun getUser(@RequestParam(value = "id") id: String): String {
    val encodedId = URLEncoder.encode(id, StandardCharsets.UTF_8)
    val url = URL("http://example.com/api/user/?id=" + encodedId)
    val connection = url.openConnection()
}
----

=== How does this work?

include::../../common/fix/encoding.adoc[]
