<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>According to the IBM documentation,</p>
</div>
<div class="quoteblock">
<blockquote>
<div class="paragraph">
<p>The Enterprise COBOL compiler now uses the Language Environment STACK storage work area for several functions, including <code>UPPER-CASE</code>, <code>LOWER-CASE</code>, and <code>NATIONAL-OF</code>. If the STACK is allocated below the 16-MB line and a large DSA (Dynamic Save Area) is needed, an insufficient storage error might occur.</p>
</div>
</blockquote>
</div>
<div class="paragraph">
<p>To aid in migrations to IBM Enterprise COBOL 5.x, this rule raises an issue when <code>UPPER-CASE</code>, <code>LOWER-CASE</code>, or <code>NATIONAL-OF</code> is used.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cobol" data-lang="cobol">        MOVE FUNCTION UPPER-CASE(FIRST-NAME) to FIRST-NAME2.  *&gt; Noncompliant
        MOVE FUNCTION LOWER-CASE(FIRST-NAME) to FIRST-NAME2.  *&gt; Noncompliant</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove this use of "xxx".</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>function name</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_1_dec_2015_084000_pierre_yves_nicolas_wrote">on 1 Dec 2015, 08:40:00 Pierre-Yves Nicolas wrote:</h3>
<div class="paragraph">
<p>According to an <a href="http://www-01.ibm.com/support/knowledgecenter/SS6SG3_3.4.0/com.ibm.entcobol.doc_3.4/igym1370.htm">IBM documentation</a>, there may be an issue when switching from IBM COBOL to any version of IBM <strong>Enterprise</strong> COBOL: this documentation is about "Enterprise COBOL for z/OS 3.4.0" and the page starts with "This chapter describes the differences between the IBM COBOL language and the Enterprise COBOL language".</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_1_dec_2015_152246_ann_campbell_wrote">on 1 Dec 2015, 15:22:46 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>I added the word 'Enterprise' [~pierre-yves.nicolas]. S&#8217;okay?</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_1_dec_2015_161132_pierre_yves_nicolas_wrote">on 1 Dec 2015, 16:11:32 Pierre-Yves Nicolas wrote:</h3>
<div class="paragraph">
<p>Ok for me [~ann.campbell.2].</p>
</div>
</div>
</div>
</div>