== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,php]
----
if ($obj == null && $obj->isOpen()) {
  echo "Object is open";
}

if ($obj != null || $obj->isOpen()) {
  echo "Object is not open";
}
----

=== Compliant solution

[source,php]
----
if ($obj == null || $obj->isOpen()) {
  echo "Object is open";
}

if ($obj != null && !$obj->isOpen()) {
  echo "Object is not open";
}
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

=== on 17 Oct 2016, 14:21:05 Yves Dubois-PÃ¨lerin wrote:
Activated the rule as part of SONARPHP-643 (Synchonize metadata of rules with RSPECs).

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
