== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example
HTTP request tools such as `curl`, `wget` and `Invoke-WebRequest` offer the option to disable certificate verification.
The following example successfully requests data from a server with an insecure certificate. Thus, it is possible that the response was intercepted or tampered with by a third party.

[source,docker]
----
FROM ubuntu:22.04

# Noncompliant
RUN curl --insecure -O https://expired.example.com/downloads/install.sh
----

=== Compliant solution
Enabling certificate verification helps to make sure that the created TLS session is secure and cannot be intercepted.
In this example, the option to disable certificate verification is removed, and a request is made to a secure server instead.

[source,docker]
----
FROM ubuntu:22.04

RUN curl -O https://new.example.com/downloads/install.sh
----


include::../see.adoc[]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
