<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Starting from Java 18 there is a possibility to write code snippets in javadoc comments. Within snippets, you can target specific parts of the code using a <code>substring</code> attribute to be highlighted, linked, or replaced. If the mentioned substring is not present in the line, nothing will be done. You should check that the substring does not contain a typo, or that the substring code initially targeted still exists in the snippet.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">{@snippet :
   public static void main(String... args) {
       System.out.println("Hello, World!");      // @highlight substing="Non-existing string"
   }
}</code></pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">{@snippet :
   public static void main(String... args) {
       System.out.println("Hello, World!");      // @highlight substing="println"
   }
}</code></pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_see">See</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Oracleâ€™s Programmer&#8217;s Guide to Snippets: <a href="https://docs.oracle.com/en/java/javase/18/code-snippet/index.html">https://docs.oracle.com/en/java/javase/18/code-snippet/index.html</a></p>
</div>
<hr>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Make sure that this value matches something in the line, or drop the "substring" attribute.
Make sure that this value matches something in the region, or drop the "substring" attribute.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="ulist">
<ul>
<li>
<p>Primary: substing attribute&#8217;s value</p>
</li>
</ul>
</div>
</div>
</div>
</div>