include::../common/description.adoc[]

== How to fix it in Helm

include::../common/how-to-fix.adoc[]

=== Code examples

==== Noncompliant code example
[source,text,diff-id=1,diff-type=noncompliant]
----
apiVersion: v1
kind: Pod
metadata:
  name: example
spec:
  containers:
    - name: firstConatiner
      image: nginx # Noncompliant

    - name: secondContainer
      image: nginx:latest # Noncompliant
----

==== Compliant solution

[source,text,diff-id=1,diff-type=compliant]
----
apiVersion: v1
kind: Pod
metadata:
  name: example
spec:
  containers:
    - name: compliantContainer
      image: nginx:1.14.2 # Compliant, fixed tag

    - name: secondCompliantContainer
      image: nginx@sha256:45b23dee08af5e43a7fea6c4cf9c25ccf269ee113168c19722f87876677c5cb2 # Compliant, SHA of the image
----

include::../common/how-does-this-work.adoc[]

include::../common/extra-mile.adoc[]

== Resources
=== Documentation
* https://helm.sh/docs/chart_best_practices/pods/#images[Helm chart best practices - Images]
* https://kubernetes.io/docs/concepts/containers/images/[Kubernetes - Images]

include::../common/footer.adoc[]
