<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This is an issue when using <code>ThreadLocal</code> for thread-local data storage in Java 25+, where <code>ScopedValue</code> provides a more modern and safer alternative.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Java 25 introduces <code>ScopedValue</code> as a modern replacement for <code>ThreadLocal</code>.</p>
</div>
<div class="paragraph">
<p><code>ThreadLocal</code> has several issues:
* Values persist until explicitly removed, causing memory leaks
* Any code can modify values, breaking encapsulation
* Expensive when creating child threads
* Requires manual cleanup in <code>finally</code> blocks</p>
</div>
<div class="paragraph">
<p><code>ScopedValue</code> provides:
* Automatic cleanup when scope ends
* Immutable values that cannot be modified by callees
* Efficient inheritance for structured concurrency</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="paragraph">
<p>Using <code>ThreadLocal</code> instead of <code>ScopedValue</code> can lead to memory leaks, data corruption, and increased maintenance complexity due to manual cleanup requirements.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Replace ThreadLocal with ScopedValue and use <code>ScopedValue.where().run()</code> for bounded scope execution.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">// Using ThreadLocal with manual cleanup
public class UserContext {
    private static final ThreadLocal&lt;String&gt; USER_ID = new ThreadLocal&lt;&gt;(); // Noncompliant

    public static void setUserId(String userId) {
        USER_ID.set(userId);
    }

    public static String getUserId() {
        return USER_ID.get();
    }

    public void processRequest(String userId) {
        USER_ID.set(userId);
        try {
            doWork();
        } finally {
            USER_ID.remove(); // Manual cleanup required
        }
    }
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">// Using ScopedValue with automatic cleanup
import java.lang.ScopedValue;

public class UserContext {
    public static final ScopedValue&lt;String&gt; USER_ID = ScopedValue.newInstance();

    public static String getUserId() {
        return USER_ID.get();
    }

    public void processRequest(String userId) {
        ScopedValue.where(USER_ID, userId).run(() -&gt; {
            doWork(); // Automatic cleanup when scope ends
        });
    }
}</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>ScopedValue (Java 25) - <a href="https://docs.oracle.com/en/java/javase/25/docs/api/java.base/java/lang/ScopedValue.html">Official Java 25 documentation for ScopedValue API</a></p>
</li>
<li>
<p>ThreadLocal (Java 25) - <a href="https://docs.oracle.com/en/java/javase/25/docs/api/java.base/java/lang/ThreadLocal.html">Official Java 25 documentation for ThreadLocal API</a></p>
</li>
<li>
<p>Java 25 Features - <a href="https://javaalmanac.io/jdk/25">Overview of new features in Java 25 including ScopedValue</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>