<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p><code>dynamic_cast</code> allow to convert pointers from one class to another, following the class inheritance hierarchy (note that in many cases, the need to perform a <code>dynamic_cast</code> is an indicator of poor class design, for instance not following the Liskov substitution principle).</p>
</div>
<div class="paragraph">
<p>If the requested conversion is not possible, the result of the conversion is a null pointer to the requested class.</p>
</div>
<div class="paragraph">
<p>An implication of this is that result of <code>dynamic_cast</code> should be tested before using it to avoid dereferencing a null pointer.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">struct Base {
    virtual ~Base() = default;
};

struct Derived: Base {
    virtual void method() {}
};

void f(Base *b) {
    auto d = dynamic_cast&lt;Derived*&gt;(b);
    d-&gt;method(); // Noncompliant, d could be null
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">struct Base {
    virtual ~Base() = default;
};

struct Derived: Base {
    virtual void method() {}
};

void f(Base *b) {
    if (auto d = dynamic_cast&lt;Derived*&gt;(b)) {
        d-&gt;method();
    }
}</code></pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p>C&#43;&#43; Core Guidelines - <a href="https://github.com/isocpp/CppCoreGuidelines/blob/e49158a/CppCoreGuidelines.md#c148-use-dynamic_cast-to-a-pointer-type-when-failure-to-find-the-required-class-is-considered-a-valid-alternative">C.148: Use <code>dynamic_cast</code> to a pointer type when failure to find the required class is considered a valid alternative</a></p>
</li>
</ul>
</div>
<hr>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_25_oct_2019_112501_loïc_joly_wrote">on 25 Oct 2019, 11:25:01 Loïc Joly wrote:</h3>
<div class="paragraph">
<p>\[~amelie.renard] Please review my changes</p>
</div>
</div>
</div>
</div>