<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when the <code>include?</code> method is called with a regular expression argument instead of a string.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>The <code>include?</code> method in Ruby only accepts string arguments. When you pass a regular expression to <code>include?</code>, Ruby raises a <code>TypeError</code> with the message "can&#8217;t convert regexp to string".</p>
</div>
<div class="paragraph">
<p>This error occurs at runtime and will cause your application to crash. The <code>include?</code> method is designed to check if a string contains another string as a substring, not to perform pattern matching.</p>
</div>
<div class="paragraph">
<p>For pattern matching with regular expressions, Ruby provides several dedicated methods that are designed for this purpose and handle regex objects correctly.</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="paragraph">
<p>This issue causes a runtime <code>TypeError</code> that will crash the application when the problematic code is executed. This can lead to service disruption and poor user experience.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it_in_ruby">How to fix it in Ruby</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Use the <code>=~</code> operator for simple pattern matching that returns the position of the match or <code>nil</code> if no match is found.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-ruby" data-lang="ruby">if text.include?(/\d+/) # Noncompliant
  puts "Found digits"
end</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-ruby" data-lang="ruby">if text =~ /\d+/
  puts "Found digits"
end</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it_in_rails">How to fix it in Rails</h2>
<div class="sectionbody">
<div class="paragraph">
<p>In Rails applications, use <code>match().present?</code> for a clear boolean result that handles <code>nil</code> values gracefully.</p>
</div>
<div class="sect2">
<h3 id="_code_examples_2">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example_2">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-ruby" data-lang="ruby">if user_input.include?(/[A-Z]/) # Noncompliant
  puts "Contains uppercase letters"
end</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution_2">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-ruby" data-lang="ruby">if user_input.match(/[A-Z]/).present?
  puts "Contains uppercase letters"
end</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Ruby String#include? documentation - <a href="https://ruby-doc.org/core/String.html#method-i-include-3F">Official documentation for the include? method</a></p>
</li>
<li>
<p>Ruby Regexp class documentation - <a href="https://ruby-doc.org/core/Regexp.html">Official documentation for regular expressions in Ruby</a></p>
</li>
<li>
<p>Ruby String pattern matching methods - <a href="https://ruby-doc.org/core/String.html#method-i-match">Documentation for String#match and related pattern matching methods</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>