== Why is this an issue?

Re-assigning a variable to itself is a defect as it has no actual effect and indicates meaning to do something else.
It usually means that:

* Either this statement is redundant and should be removed,
* or the re-assignment is a mistake, and another value or variable was intended for the assignment instead

=== Code examples

==== Noncompliant code example

[source,csharp,diff-id=1,diff-type=noncompliant]
----
class DataClass {
  
  static List<int> data;

  public static void SetData(List<int> data)
  {
    data = data; // Noncompliant
  }
}
----

==== Compliant solution

[source,csharp,diff-id=1,diff-type=compliant]
----
class DataClass {
  
  static List<int> data;

  public static void SetData(List<int> data)
  {
    DataClass.data = data; // compliant
  }
}
----


ifdef::env-github,rspecator-view,env-vscode[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view,env-vscode[]
