== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,csharp]
----
public void SetName(string name)
{
  name = name;
}
----

=== Compliant solution

[source,csharp]
----
public void SetName(string name)
{
  this.name = name;
}
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
