include::../description.adoc[]

== Noncompliant Code Example

[source,cs]
----
public void SetName(string name)
{
  name = name;
}
----

== Compliant Solution

[source,cs]
----
public void SetName(string name)
{
  this.name = name;
}
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
