include::../description.adoc[]

== Noncompliant Code Example

----
CREATE PROCEDURE doSomething
	AS
BEGIN
  ...
  -- TODO something
  ...
END
GO
----

include::../see.adoc[]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
