== Why is this an issue?

The use of unnecessary types makes the eye stumble, and inhibits the smooth reading of code. 


=== Noncompliant code example

[source,java]
----
public void doSomething() {

  <String>foo("blah");  // Noncompliant; <String> is inferred
}
public void <T> foo(T t) {
  // ...
}
----


=== Compliant solution

[source,java]
----
public void doSomething() {

  foo("blah");
}
public void <T> foo(T t) {
  // ...
}
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Remove this unneeded "xxx" parameter.


endif::env-github,rspecator-view[]
