== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

With the default regular expression ``++YIF_[A-Za-z]+++``:

[source,abap]
----
INTERFACE MyInterface. "Noncompliant
...
ENDINTERFACE.
----

=== Compliant solution

[source,abap]
----
INTERFACE YIF_myinterface. 
...
ENDINTERFACE.
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::parameters.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
