== Why is this an issue?

include::../description.adoc[]

== How to fix it

include::../howtofix.adoc[]

=== Code examples

==== Noncompliant code example

With the default threshold of 3:

[source,swift,diff-id=1,diff-type=noncompliant]
----
prepare("a message")  // Noncompliant; duplicated 3 times
execute("a message")
release("a message")
----

==== Compliant solution

[source,swift,diff-id=1,diff-type=compliant]
----
let message = "a message"

prepare(message)
execute(message)
release(message)
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../parameters.adoc[]

include::../highlighting.adoc[]

'''

endif::env-github,rspecator-view[]
