include::../description.adoc[]

== Noncompliant Code Example

With the default threshold of 3:

----
prepare("a message")  // Noncompliant; duplicated 3 times
execute("a message")
release("a message")
----

== Compliant Solution

----
let message = "a message"

prepare(message)
execute(message)
release(message)
----

== Exceptions

To prevent generating some false-positives, literals having 5 or less characters are excluded as well as literals containing only letters, digits and '_'.

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../parameters.adoc[]

include::../highlighting.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
