<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p><code>std::string_view</code> is a read-only view over a string, it doesn&#8217;t hold any data, it only holds a pointer to the first character of the string and its length. <code>std::string_view</code> can offer better performance than <code>std::string</code> in several cases:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>no memory allocations are required during construction, it is cheap to pass them by value, no need to pass them by reference</p>
</li>
<li>
<p>no heap allocation when passing a string literal to a <code>std::string_view</code> function argument</p>
</li>
<li>
<p><code>substr</code> operations over a <code>std::string_view</code> do not require memory allocation</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>When using <code>std::string_view</code> you shouldn&#8217;t however forget that:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>it&#8217;s a non-owning range, you should keep into consideration the liveness of the pointed range</p>
</li>
<li>
<p>it doesn&#8217;t guarantee a null-terminated string like <code>std::string</code></p>
</li>
</ul>
</div>
<div class="paragraph">
<p>This rule flags <code>const std::string&amp;</code> function arguments, which can be safely replaced with <code>std::string_view</code> ones when not relying on the null-termination character.</p>
</div>
<div class="paragraph">
<p>Note that, if you are calling <code>substr</code> on the parameter, you may have to modify your code to explicitly cast the result to <code>std::string</code>.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">void fun(const std::string&amp; name) { // Noncompliant, replace const std::string&amp; by std::string_view
  // ...
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">void fun(std::string_view name) {
  // ...
}</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Replace this "const std::string&amp;" by "std::string_view".</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_is_related_to_s6231">is related to: <a data-rspec-id="S6231" class="rspec-auto-link">S6231</a></h3>

</div>
</div>
</div>