== Why is this an issue?

The complexity of an expression is defined by the number of ``++&&++`` and ``++||++`` operators it contains.

A single expression's complexity should not become too high to keep the code readable.

=== Noncompliant code example

With the default threshold value of 3:

[source,scala]
----
if (((condition1 && condition2) || (condition3 && condition4)) && condition5) { ... } 
----

=== Compliant solution

[source,scala]
----
if ((myFirstCondition || mySecondCondition) && myLastCondition) { ... }     
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../parameters.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
