This rule raises an issue when `Database.executeBatch()` is called with a scope parameter greater than 2000.

== Why is this an issue?

When executing batch jobs in Salesforce, the scope parameter determines how many records are processed in each batch execution. While Salesforce allows scope values up to 2000 and will automatically chunk larger values into smaller batches, specifying excessively large scope parameters can lead to several problems.

First, larger batch sizes increase the risk of hitting governor limits within a single batch execution. Each batch execution has its own set of governor limits for DML operations, SOQL queries, and CPU time. Processing too many records at once makes it more likely to exceed these limits, causing the batch job to fail.

Second, performance can degrade with larger batch sizes. The Salesforce platform is optimized for processing smaller chunks of data efficiently. Larger batches can lead to longer execution times and increased memory usage, potentially affecting overall system performance.

Finally, using appropriate batch sizes improves error handling and recovery. If a batch fails, smaller scope sizes mean fewer records need to be reprocessed, making debugging and recovery more manageable.

=== What is the potential impact?

Using excessively large batch scope parameters can cause batch jobs to fail due to governor limit violations, degrade system performance, and make error recovery more difficult. This can lead to incomplete data processing and unreliable batch operations.

== How to fix it

Use a scope parameter of 2000 or less. The default value of 200 is often optimal for most use cases.

=== Code examples

==== Noncompliant code example

[source,apex,diff-id=1,diff-type=noncompliant]
----
AccountUpdateBatchJob obj = new AccountUpdateBatchJob();
Database.executeBatch(obj, 5000); // Noncompliant
----

==== Compliant solution

[source,apex,diff-id=1,diff-type=compliant]
----
AccountUpdateBatchJob obj = new AccountUpdateBatchJob();
Database.executeBatch(obj, 200); // Compliant
----

== Resources

=== Documentation

 * Batch Apex Developer Guide - https://developer.salesforce.com/docs/atlas.en-us.apexcode.meta/apexcode/apex_batch.htm[Official Salesforce documentation on Batch Apex implementation and best practices]

 * Database.executeBatch Method - https://developer.salesforce.com/docs/atlas.en-us.apexref.meta/apexref/apex_methods_system_database.htm#apex_System_Database_executeBatch_1[Reference documentation for the Database.executeBatch method and its parameters]
