This rule raises an issue when `each` is used with `break` or `return` statements for searching, existence checking, or membership testing.

== Why is this an issue?

Using `each` with `break` or `return` statements is a common anti-pattern in Ruby that makes code less readable and less idiomatic.

When you use `each` with `break` or `return`, you're essentially implementing search or filtering logic manually. This approach has several problems:

* **Intent is unclear**: Other developers need to read through the entire loop to understand what the code is trying to accomplish
* **More verbose**: The code requires additional variables and control flow statements
* **Less idiomatic**: Ruby provides specific enumerable methods designed for these exact use cases
* **Harder to maintain**: Manual loop control is more error-prone than using built-in methods

Ruby's enumerable methods are designed to express intent clearly. When you see `find`, you immediately know the code is searching for an element. When you see `any?`, you know it's checking for existence. This makes code self-documenting and easier to understand at a glance.

=== What is the potential impact?

This issue primarily affects code maintainability and readability. While the functional behavior remains the same, unclear intent can lead to:

* Increased time spent understanding code during maintenance
* Higher likelihood of introducing bugs when modifying the logic
* Reduced code quality and team productivity
* Difficulty for new team members to understand the codebase

== How to fix it

Replace `each` with `return` with the `find` method when searching for the first matching element.

=== Code examples

==== Noncompliant code example

[source,ruby,diff-id=1,diff-type=noncompliant]
----
def find_user(name)
  users.each do |user|
    return user if user.name == name  # Noncompliant
  end
  nil
end
----

==== Compliant solution

[source,ruby,diff-id=1,diff-type=compliant]
----
def find_user(name)
  users.find { |user| user.name == name }
end
----

== Resources

=== Documentation

 * Ruby Enumerable#find - https://ruby-doc.org/core/Enumerable.html#method-i-find[Official documentation for the find method]

 * Ruby Enumerable#any? - https://ruby-doc.org/core/Enumerable.html#method-i-any-3F[Official documentation for the any? method]

 * Ruby Array#include? - https://ruby-doc.org/core/Array.html#method-i-include-3F[Official documentation for the include? method]
