== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,python]
----
r"[0-99]" # Noncompliant, this won't actually match strings with two digits
r"[0-9.-_]" # Noncompliant, .-_ is a range that already contains 0-9 (as well as various other characters such as capital letters)
----


=== Compliant solution

[source,python]
----
r"[0-9]{1,2}"
r"[0-9.\\-_]"
----

