== Why is this an issue?

In the classfile API introduced by Java 24, there are several versions of the `transformClass` methods.
In particular if the name of the transformed class is unchanged compared to the original class, then it is recommended to use the simplest version of the method that takes only 2 arguments.
This makes the code shorter, clearer, and reduces chances for mistakes.


== How to fix it

Remove the class name argument of the `transformClass` call.

=== Code examples

==== Noncompliant code example

[source,java,diff-id=1,diff-type=noncompliant]
----
public static void transformClassFile(Path path) throws IOException {
    ClassFile classFile = ClassFile.of();
    ClassModel classModel = classFile.parse(path);
    byte[] newBytes = classFile.transformClass(classModel,
      classModel.thisClass().asSymbol(), // Noncompliant
      (classBuilder, classElement) -> {
        if (!(classElement instanceof MethodModel methodModel &&
            methodModel.methodName().stringValue().startsWith("debug"))) {
            classBuilder.with(classElement);
        }
      });
}
----

==== Compliant solution

[source,java,diff-id=1,diff-type=compliant]
----
public static void transformClassFile(Path path) throws IOException {
    ClassFile classFile = ClassFile.of();
    ClassModel classModel = classFile.parse(path);
    byte[] newBytes = classFile.transformClass(classModel,
      (classBuilder, classElement) -> {
        if (!(classElement instanceof MethodModel methodModel &&
            methodModel.methodName().stringValue().startsWith("debug"))) {
            classBuilder.with(classElement);
        }
      });
}
----

== Resources

* https://openjdk.org/jeps/484[JEP 484: Class-File API]
* https://docs.oracle.com/en/java/javase/24/docs/api/java.base/java/lang/classfile/ClassFile.html#transformClass(java.lang.classfile.ClassModel,java.lang.classfile.ClassTransform)[ClassFile API, transformClass(ClassModel, ClassTransform)]
* https://docs.oracle.com/en/java/javase/24/docs/api/java.base/java/lang/classfile/ClassFile.html#transformClass(java.lang.classfile.ClassModel,java.lang.constant.ClassDesc,java.lang.classfile.ClassTransform)[ClassFile API, transformClass(ClassModel, ClassDesc, ClassTransform)]
