This rules raises an issue when `trio.sleep()` or `anyio.sleep()` is called with a duration greater than 24 hours.

== Why is this an issue?

When using `trio.sleep()` or `anyio.sleep()` with very long intervals (greater than 24 hours), the intent is usually to wait indefinitely or for an extremely long period. Both libraries provide dedicated methods specifically designed for this purpose: `trio.sleep_forever()` and `anyio.sleep_forever()`.

Using explicit sleep durations greater than 24 hours has several drawbacks:

* It obscures the developer's intent. A very large number like `86400 * 365` doesn't clearly communicate that the code intends to wait indefinitely.

* It makes code less maintainable, as other developers need to calculate what the large numbers actually represent.

In summary, using `sleep_forever()` is preferable when the intent is to sleep indefinitely as it clearly conveys this purpose, avoiding maintainability issues caused by using arbitrarily large sleep durations.



== How to fix it in Trio

Replace calls to `trio.sleep()` that use intervals greater than 24 hours with calls to `trio.sleep_forever()`.

If the intention is truly to wait for a specific long duration rather than indefinitely, consider expressing that intent more clearly by using named variables or constants.

=== Code examples

==== Noncompliant code example

[source,python,diff-id=1,diff-type=noncompliant]
----
import trio

async def long_wait():
    await trio.sleep(86400 * 365)  # Noncompliant
----

==== Compliant solution

[source,python,diff-id=1,diff-type=compliant]
----
import trio

async def long_wait():
    await trio.sleep_forever()
----

== How to fix it in AnyIO

Replace calls to `anyio.sleep()` that use intervals greater than 24 hours with calls to `anyio.sleep_forever()`.

If the intention is truly to wait for a specific long duration rather than indefinitely, consider expressing that intent more clearly by using named variables or constants.

=== Code examples

==== Noncompliant code example

[source,python,diff-id=2,diff-type=noncompliant]
----
import anyio

async def long_wait():
    await anyio.sleep(86400 * 30)  # Noncompliant
----

==== Compliant solution

[source,python,diff-id=2,diff-type=compliant]
----
import anyio

async def long_wait():
    await anyio.sleep_forever()
----

== Resources

=== Documentation
* Trio - https://trio.readthedocs.io/en/stable/reference-core.html#trio.sleep_forever[sleep_forever() documentation]
* AnyIO - https://anyio.readthedocs.io/en/stable/api.html#anyio.sleep_forever[sleep_forever() documentation]
* Python asyncio - https://docs.python.org/3/library/asyncio-task.html[Tasks and coroutines]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Replace this long sleep with sleep_forever()

=== Highlighting

The entire function call expression:
* Primary locations: the `trio.sleep()` or `anyio.sleep()` call
* Secondary locations: none

endif::env-github,rspecator-view[]
