<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Using element type in class selectors is slower than using only the class selector.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-javascript" data-lang="javascript">var $products = $("div.products");    // Noncompliant - slow</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-javascript" data-lang="javascript">var $products = $(".products");    // Compliant - fast</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove "XXX" in this selector.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_17_mar_2015_102551_linda_martin_wrote">on 17 Mar 2015, 10:25:51 Linda Martin wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] assigning it to you for completion and review.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_16_jan_2020_140329_elena_vilchik_wrote">on 16 Jan 2020, 14:03:29 Elena Vilchik wrote:</h3>
<div class="paragraph">
<p>Deprecating rule related to jquery (in the process of migration to new JS frontend we decided to deprecate Jquery rules, see https://github.com/SonarSource/SonarJS/issues/1698)</p>
</div>
</div>
</div>
</div>