<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Disabling "ANSI_WARNINGS" and/or "ARITHABORT" in a procedure may silence errors, decrease performance, or block index creation.</p>
</div>
<div class="paragraph">
<p>From the documentation (<a href="https://learn.microsoft.com/en-us/sql/t-sql/statements/set-arithabort-transact-sql">ARITHABORT</a>, <a href="https://learn.microsoft.com/en-us/sql/t-sql/statements/set-ansi-warnings-transact-sql">ANSI_WARNINGS</a>), disabling these options could result in (among others):</p>
</div>
<div class="paragraph">
<p><code>SET ANSI_WARNINGS OFF</code></p>
</div>
<div class="ulist">
<ul>
<li>
<p>CREATE, UPDATE, INSERT, and DELETE statements on tables with indexes on computed columns or indexed views will fail</p>
</li>
<li>
<p>No warning issued if null values appear in aggregate functions, such as SUM, AVG, MAX, MIN, STDEV, STDEVP, VAR, VARP, or COUNT</p>
</li>
<li>
<p>The divide-by-zero and arithmetic overflow errors cause null values to be returned, no roll-back</p>
</li>
</ul>
</div>
<div class="paragraph">
<p><code>SET ARITHABORT OFF</code></p>
</div>
<div class="ulist">
<ul>
<li>
<p>It can negatively impact query optimization, leading to performance issues</p>
</li>
<li>
<p>An arithmetic, overflow, divide-by-zero, or domain error, during INSERT, UPDATE, or DELETE statement will cause SQL Server to insert or update a NULL value</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>This rule raises an issue when "ANSI_WARNINGS" and/or "ARITHABORT" are set to <code>OFF</code> in a procedure.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-sql" data-lang="sql">CREATE PROCEDURE myProc
AS
BEGIN
SET ANSI_WARNINGS OFF; -- Noncompliant
SET ARITHABORT OFF; -- Noncompliant
-- ...
END</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-sql" data-lang="sql">CREATE PROCEDURE myProc
AS
BEGIN
-- OK - Default value is ON
END

CREATE PROCEDURE myProc
AS
BEGIN
SET ANSI_WARNINGS, ARITHABORT ON;
END</code></pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p><a href="https://learn.microsoft.com/en-us/sql/t-sql/statements/set-arithabort-transact-sql">Microsoft Documentation: SET ARITHABORT (Transact-SQL)</a></p>
</li>
<li>
<p><a href="https://learn.microsoft.com/en-us/sql/t-sql/statements/set-ansi-warnings-transact-sql">Microsoft Documentation: SET ANSI_WARNINGS (Transact-SQL)</a></p>
</li>
</ul>
</div>
</div>
</div>