``++IMPORT++``ed values should be treated as read-only. Doing otherwise could result in unpleasant surprises down the line.


== Noncompliant Code Example

----
D VAR2             S              5  0 IMPORT
 * Noncompliant
C           MOVE       MYVAR     VAR2

C/FREE
  // Noncompliant
  VAR2=MYVAR;
/END-FREE
----


ifdef::env-github,rspecator-view[]
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
