== Why is this an issue?

``++IMPORT++``ed values should be treated as read-only. Doing otherwise could result in unpleasant surprises down the line.


=== Noncompliant code example

[source,rpg]
----
D VAR2             S              5  0 IMPORT
 * Noncompliant
C           MOVE       MYVAR     VAR2

C/FREE
  // Noncompliant
  VAR2=MYVAR;
/END-FREE
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

"xxx" was imported and should not be updated


'''
== Comments And Links
(visible only on this page)

=== on 2 Apr 2015, 19:22:50 Ann Campbell wrote:
http://www.bmeyers.net/faqs/14-tips/32-rpg-iv-style?start=2

endif::env-github,rspecator-view[]
