== Why is this an issue?

``++IMPORT++``ed values should be treated as read-only. Doing otherwise could result in unpleasant surprises down the line.


=== Noncompliant code example

[source,rpg]
----
D VAR2             S              5  0 IMPORT
 * Noncompliant
C           MOVE       MYVAR     VAR2

C/FREE
  // Noncompliant
  VAR2=MYVAR;
/END-FREE
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
