<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Taking a <code>shared_ptr</code> by r-value reference is unnecessary. If done on purpose, it might imply that <code>unique_ptr</code> is a better choice since it transfers unique ownership.</p>
</div>
<div class="paragraph">
<p>In general, a function should take:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>A copy of a <code>shared_ptr</code>, if the function takes part in the ownership of the managed object</p>
</li>
<li>
<p>A reference to a <code>shared_ptr</code> if the function plans to modify the shared_ptr itself by calling <code>reset</code> or <code>swap</code></p>
</li>
<li>
<p>A reference to a <code>const</code> <code>shared_ptr</code>, if the function might take part in the ownership by copying the reference to another shared_ptr on at least one of its paths</p>
</li>
<li>
<p>A raw pointer/reference to the object, if the function is only interested in the current value of the managed object</p>
</li>
</ul>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">Class Circle{};

void fn(shared_ptr&lt;Circle&gt;&amp;&amp; circle);  // Noncompliant</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">Class Circle{};

void fn(shared_ptr&lt;Circle&gt; circle);</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_exceptions">Exceptions</h3>
<div class="paragraph">
<p>The move constructor of the <code>shared_ptr</code> itself.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p>C&#43;&#43; Core Guidelines - <a href="https://github.com/isocpp/CppCoreGuidelines/blob/e49158a/CppCoreGuidelines.md#r34-take-a-shared_ptrwidget-parameter-to-express-shared-ownership">R.34: Take a <code>shared_ptr&lt;widget&gt;</code> parameter to express shared ownership</a></p>
</li>
</ul>
</div>
</div>
</div>