<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Needing to cast from an <code>interface</code> to a concrete type indicates that something is wrong with the abstractions in use, likely that something is missing from the <code>interface</code>. Instead of casting to a discrete type, the missing functionality should be added to the <code>interface</code>. Otherwise there is the risk of runtime exceptions.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">public interface MyInterface {
  void doStuff();
}

public class MyClass1 implements MyInterface {
  int data;
  public void DoStuff()  {
    // TODO...
  }
}

public static class DowncastExampleProgram {
  static void EntryPoint(MyInterface interfaceRef) {
    MyClass1 class1 = (MyClass1)interfaceRef;  // Noncompliant
    int privateData = class1.data;
  }
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_exceptions">Exceptions</h3>
<div class="paragraph">
<p>Casting to <code>Object</code> doesn&#8217;t raise an issue, because it can never fail.</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">static void EntryPoint(IMyInterface interfaceRef)
{
  Object o = (Object)interfaceRef;
  ...
}</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove this cast and edit the interface to add the missing functionality.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_2_jul_2015_071845_tamas_vajk_wrote">on 2 Jul 2015, 07:18:45 Tamas Vajk wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] LGTM</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_23_jul_2015_075046_tamas_vajk_wrote">on 23 Jul 2015, 07:50:46 Tamas Vajk wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] Added an exception to this rule.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_23_jul_2015_090731_ann_campbell_wrote">on 23 Jul 2015, 09:07:31 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~tamas.vajk] we need to add a 'why' to this exception.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_23_jul_2015_094138_tamas_vajk_wrote">on 23 Jul 2015, 09:41:38 Tamas Vajk wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] Added the reason.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_23_jul_2015_125203_ann_campbell_wrote">on 23 Jul 2015, 12:52:03 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>Thanks [~tamas.vajk]. Looks good</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_19_jan_2016_102915_ann_campbell_wrote">on 19 Jan 2016, 10:29:15 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>Marked Java irrelevant because of the volume of issues raised.</p>
</div>
</div>
</div>
</div>