<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Java 21 introduces <code>case null</code> for <code>switch</code>. It is a more concise and readable way to handle nullability compared to an <code>if</code> statement before a <code>switch</code>.</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">switch (s) {
  case null: /* code if null */
  // ...
}</code></pre>
</div>
</div>
<div class="paragraph">
<p>When the selector expression evaluates to null and the <code>case null</code> is present, switch statements don&#8217;t throw a NullPointerException anymore.
It is not mandatory anymore to test for <code>null</code> before a <code>switch</code>, and  it is more readable to use the <code>case null</code> form.</p>
</div>
<div class="paragraph">
<p>This rule identifies <code>if</code> statements that could be replaced by a more readable <code>case null</code> label.</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">if (s == null) {
  /* code if null */
}
switch (s) {
  // ...
}</code></pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Move the null check inside the switch statement.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">static void f(String s) {
  if (s == null) { // Noncompliant, we can move the null check in the switch statement.
    System.out.println("Oops!");
    return;
  }
  switch (s) {
    case "Foo", "Bar" -&gt; System.out.println("Great");
    default           -&gt; System.out.println("Ok");
   }
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">static void f(String s) {
  switch (s) {
    case null         -&gt; System.out.println("Oops!");
    case "Foo", "Bar" -&gt; System.out.println("Great");
    default           -&gt; System.out.println("Ok");
   }
}</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p><a href="https://openjdk.org/jeps/441">JEP 441: Pattern Matching for Switch</a></p>
</li>
</ul>
</div>
</div>
</div>