== Why is this an issue?

There is no reason to use Javadoc-specific tags in non-Javadoc comments, and doing to could be confusing.


=== Noncompliant code example

[source,java]
----
// FIXME should be moved to {@link ServerPluginRepository#uninstall(String)}  <-- Noncompliant
----


=== Compliant solution

[source,java]
----
// FIXME should be moved to ServerPluginRepository#uninstall(String)
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Remove this "xxx" tag or convert the comment to Javadoc.


'''
== Comments And Links
(visible only on this page)

=== on 8 May 2015, 14:42:36 Ann Campbell wrote:
rule origin: email thread

endif::env-github,rspecator-view[]
