There is no reason to use Javadoc-specific tags in non-Javadoc comments, and doing to could be confusing.


== Noncompliant Code Example

----
// FIXME should be moved to {@link ServerPluginRepository#uninstall(String)}  <-- Noncompliant
----


== Compliant Solution

----
// FIXME should be moved to ServerPluginRepository#uninstall(String)
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
