include::../description.adoc[]


== Noncompliant Code Example

----
r"^a|b|c$"
----


== Compliant Solution

----
r"^(?:a|b|c)$"
----

or


----
r"^a$|^b$|^c$"
----

or, if you do want the anchors to only apply to ``a`` and ``c`` respectively:


----
r"(?:^a)|b|(?:c$)"
----
ifdef::env-github,rspecator-view[]
include::../highlighting.adoc[]
include::../message.adoc[]
endif::env-github,rspecator-view[]
