include::../description.adoc[]


== Noncompliant Code Example

[source,python]
----
r"^a|b|c$"
----


== Compliant Solution

[source,python]
----
r"^(?:a|b|c)$"
----

or


----
r"^a$|^b$|^c$"
----

or, if you do want the anchors to only apply to ``a`` and ``c`` respectively:


----
r"(?:^a)|b|(?:c$)"
----
