include::../description.adoc[]

Additionnaly, one shouldn't use a class to store exclusively static methods. In JavaScript, you can use a module, or better export each method separately.

== Why is this an issue?

Using an empty class serves no purpose and can hinder the readability of the code.

[source,javascript]
----
class Foo {
  static bar: () => {
    // ...
  }
}
----

//=== What is the potential impact?

== How to fix it

You can simply directly import the functions that you wish to make available.

[source,javascript]
----
export bar = () => {
  // ...
}
----

=== Code examples

==== Noncompliant code example

[source,javascript,diff-id=1,diff-type=noncompliant]
----
class Foo {
  static bar: () => {
    // ...
  }
}

class DoAndLog {
  constructor () {
    console.log('I\'m done!');
  }
}
----

==== Compliant solution

[source,javascript,diff-id=1,diff-type=compliant]
----
export bar = () => {
  // ...
}

function doAndLog() {
  console.log('I\'m done!');
}
----

//=== How does this work?

//=== Pitfalls

//=== Going the extra mile

== Resources

=== Documentation

* 

//=== Articles & blog posts
//=== Conference presentations
//=== Standards
