== How to fix it in Node.js

=== Code examples

include::../../common/fix/code-rationale.adoc[]

==== Noncompliant code example

[source,javascript,diff-id=1,diff-type=noncompliant]
----
const express = require('express');
const errorhandler = require('errorhandler');

let app = express();
app.use(errorhandler()); // Noncompliant
----

==== Compliant solution

[source,javascript,diff-id=1,diff-type=compliant]
----
const express = require('express');
const errorhandler = require('errorhandler');

let app = express();

if (process.env.NODE_ENV === 'development') {
  app.use(errorhandler());
}
----

=== How does this work?

include::../../common/fix/how.adoc[]

For example, in Node.js you can check the value of `process.env.NODE_ENV` to
determine if debug settings should be enabled.

