<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Shared naming conventions allow teams to collaborate efficiently.</p>
</div>
<div class="paragraph">
<p>This rule raises an issue when a method or a property name is not snake_cased.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Rename the function so that its name conforms to the snake_case convention.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-bash" data-lang="bash">function doSomething() { # Noncompliant
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-bash" data-lang="bash">function do_something() {
}</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_external_coding_guidelines">External coding guidelines</h3>
<div class="ulist">
<ul>
<li>
<p><a href="https://courses.cs.washington.edu/courses/cse374/22wi/resources/bash_style_guide.html">CSE 374 Bash Style Guide</a></p>
</li>
<li>
<p><a href="https://google.github.io/styleguide/shellguide.html#function-names">Google Shell Style Guide</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Rename [method|function|subroutine|procedure] "XXXX" to match the regular expression ${format}.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_is_related_to_s1542">is related to: <a data-rspec-id="S1542" class="rspec-auto-link">S1542</a></h3>

</div>
<div class="sect2">
<h3 id="_on_20_may_2013_142913_fabrice_bellingard_wrote">on 20 May 2013, 14:29:13 Fabrice Bellingard wrote:</h3>
<div class="paragraph">
<p>Implementation: http://jira.codehaus.org/browse/SONARJAVA-118</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_2_aug_2013_090559_freddy_mallet_wrote">on 2 Aug 2013, 09:05:59 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>Is implemented by http://jira.codehaus.org/browse/SONARPLUGINS-2776</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_12_feb_2014_180827_freddy_mallet_wrote">on 12 Feb 2014, 18:08:27 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>Is implemented by http://jira.codehaus.org/browse/SONARPLUGINS-3455 for PHP</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_6_feb_2018_153050_alban_auzeill_wrote">on 6 Feb 2018, 15:30:50 Alban Auzeill wrote:</h3>
<div class="paragraph">
<p>COBOL use [<a data-rspec-id="S1589" class="rspec-auto-link">RSPEC-1589</a>] instead.</p>
</div>
<div class="paragraph">
<p>PL/I, PL/SQL, T-SQL, VB6 use [<a data-rspec-id="S1542" class="rspec-auto-link">RSPEC-1542</a>] instead.</p>
</div>
</div>
</div>
</div>