== Why is this an issue?

Checker to find use of single quote where double quote is preferred.


=== Noncompliant code example

[source,html]
----
<div id='header'></div>
----


=== Compliant solution

[source,html]
----
<div id="header"></div>
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
