Checker to find use of single quote where double quote is preferred.


== Noncompliant Code Example

----
<div id='header'></div>
----


== Compliant Solution

----
<div id="header"></div>
----


ifdef::env-github,rspecator-view[]
'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
