This rule raises an issue when synchronous HTTP client calls are used within asynchronous functions.

== Why is this an issue?

Using synchronous HTTP clients like `urllib3`, `requests`, or `httpx.Client` in asynchronous code blocks the entire event loop. This undermines the primary advantage of asynchronous programming - the ability to perform concurrent operations without blocking execution.

When an async function makes a synchronous HTTP request:

* The event loop is completely blocked until the HTTP operation completes
* No other coroutines can run during this time, even if they're ready to execute
* The responsiveness of the application is degraded
* In server applications, this can cause timeouts or failures for other concurrent requests

Instead, async-compatible HTTP clients should be used:

* `httpx.AsyncClient` - works with asyncio, Trio, and AnyIO
* `aiohttp.ClientSession` - works with asyncio
* `asks` - works with Trio and asyncio

Using these libraries allows other tasks to continue executing while waiting for HTTP responses, significantly improving application performance and responsiveness.

== How to fix it

Replace synchronous HTTP clients with asynchronous alternatives. The `httpx.AsyncClient` is recommended as it provides a consistent API across asyncio, Trio, and AnyIO frameworks.

=== Code examples

==== Noncompliant code example

[source,python,diff-id=1,diff-type=noncompliant]
----
import requests

async def fetch_data():
    response = requests.get("https://api.example.com/data")  # Noncompliant
    return response.json()
----

==== Compliant solution

Using httpx.AsyncClient (works with asyncio, Trio, and AnyIO):

[source,python,diff-id=1,diff-type=compliant]
----
import httpx

async def fetch_data():
    async with httpx.AsyncClient() as client:
        response = await client.get("https://api.example.com/data")
        return response.json()
----

== Resources

=== Documentation
* HTTPX - https://www.python-httpx.org/async/[Async Usage]
* AIOHTTP - https://docs.aiohttp.org/en/stable/client_quickstart.html[Client Quickstart]
* Asks - https://asks.readthedocs.io/en/latest/[Documentation]

=== Articles & blog posts
* Python - https://realpython.com/async-io-python/[Async IO in Python]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message
Use an async HTTP client in this async function instead of a synchronous one.

=== Highlighting
* Primary locations: the synchronous HTTP client call within an async function
* Secondary locations: the enclosing async function `async` keyword (message: "this is an asynchronous function")

endif::env-github,rspecator-view[]
