=== How to fix it in Flask

include::../../common/fix/code-rationale.adoc[]

==== Noncompliant code example

[source,python,diff-id=1,diff-type=noncompliant]
----
import logging

app = Flask(__name__)

@app.route('/example')
def log():
    input = request.args["input"]
    app.logger.log("%s", input) # Noncompliant
----

==== Compliant solution

[source,python,diff-id=1,diff-type=compliant]
----
import logging

app = Flask(__name__)

@app.route('/example')
def log():
    input = request.args["input"]
    if input.isalnum():
        app.logger.log("%s", input)
----

include::../../common/fix/how-does-this-work.adoc[]
