=== How to fix it in Flask

include::../../common/fix/code-rationale.adoc[]

==== Noncompliant code example

[source,python,diff-id=1,diff-type=noncompliant]
----
import logging

app = Flask(__name__)

@app.route('/example')
def log():
    input = request.args["input"]
    app.logger.critical("%s", input) # Noncompliant
----

==== Compliant solution

[source,python,diff-id=1,diff-type=compliant]
----
import logging

app = Flask(__name__)

@app.route('/example')
def log():
    input = request.args["input"]
    if input.isalnum():
        app.logger.critical("%s", input)
    else:
        app.logger.critical("Invalid Input")
----

include::../../common/fix/how-does-this-work.adoc[]

Here, the example compliant code uses the `isalnum` function to ensure the untrusted data is safe. If not, it performs  [wherever you choose for the encoding) to protect from log injection.
