=== How to fix it in Python Standard Library

include::../../common/fix/code-rationale.adoc[]

==== Noncompliant code example

[source,python,diff-id=1,diff-type=noncompliant]
----
import logging

logger = logging.getLogger(__name__)

@app.route('/example')
def log():
    input = request.args["input"]
    logger.log(logging.CRITICAL, "%s", input) # Noncompliant
----

==== Compliant solution

[source,python,diff-id=1,diff-type=compliant]
----
import logging

logger = logging.getLogger(__name__)

@app.route('/example')
def log():
    input = request.args["input"]
    if input.isalnum():
        logger.log(logging.CRITICAL, "%s", input)
    else:
        logger.log(logging.CRITICAL, "Invalid Input")
----

include::../../common/fix/how-does-this-work.adoc[]

Here, the example compliant code uses the `isalnum` function to ensure the untrusted data is safe. If not, it performs  [wherever you choose for the encoding) to protect from log injection.
