=== How to fix it in Python Standard Library

include::../../common/fix/code-rationale.adoc[]

==== Noncompliant code example

[source,python,diff-id=1,diff-type=noncompliant]
----
import logging

logger = logging.getLogger(__name__)

@app.route('/example')
def log():
    input = request.args["input"]
        logger.log(logging.CRITICAL, "%s", "input")
----

==== Compliant solution

[source,python,diff-id=1,diff-type=compliant]
----
import logging

logger = logging.getLogger(__name__)

@app.route('/example')
def log():
    input = request.args["input"]
    if input.isalnum():
        app.logger.log("%s", input)
    else:
         # Perform some kind of encoding here.
         app.logger.log("%s", safe_input)
----

include::../../common/fix/how-does-this-work.adoc[]

Here, the example compliant code uses the `isalnum` function to ensure the untrusted data is safe. If not, it performs  [wherever you choose for the encoding) to protect from log injection.
