=== How to fix it in Python Standard Library

include::../../common/fix/code-rationale.adoc[]

==== Noncompliant code example

[source,python,diff-id=1,diff-type=noncompliant]
----
import logging

logger = logging.getLogger(__name__)

@app.route('/example')
def log():
    input = request.args["input"]
    logger.log("%s", input) # Noncompliant
----

==== Compliant solution

[source,python,diff-id=1,diff-type=compliant]
----
import logging

logger = logging.getLogger(__name__)

@app.route('/example')
def log():
    input = request.args["input"]
    if input.isalpha():
        logger.log("%s", input)
----

include::../../common/fix/how-does-this-work.adoc[]
