<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Spring <code>@Controller</code>, <code>@Service</code>, and <code>@Repository</code> classes are singletons by default, meaning only one instance of the class is ever instantiated in the application. Typically such a class might have a few <code>static</code> members, such as a logger, but all non-static members should be managed by Spring and supplied via constructor injection rather than by field injection.</p>
</div>
<div class="paragraph">
<p>This rule raise an issue when any non-<code>static</code> member of a Spring component has an injection annotation.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">@Controller
public class HelloWorld {

  @Autowired
  private String name = null; // Noncompliant

}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="paragraph">
<p>As of Spring 4.3</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">@Controller
public class HelloWorld {

  private String name = null;

  HelloWorld(String name) {
    this.name = name;
  }
}</code></pre>
</div>
</div>
<div class="paragraph">
<p>Before Spring 4.3</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">@Controller
public class HelloWorld {

  private String name = null;

  @Autowired
  HelloWorld(String name) {
    this.name = name;
  }
}</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove this annotation and use constructor injection instead.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>Primary:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>Annotation of injected field</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>Secondary (if possible):</p>
</div>
<div class="ulist">
<ul>
<li>
<p>on all the other fields with injection annotation</p>
<div class="ulist">
<ul>
<li>
<p>message: 'Remove also this annotation.'</p>
</li>
</ul>
</div>
</li>
<li>
<p>finally, on constructor (if present)</p>
<div class="ulist">
<ul>
<li>
<p>message: 'Constructor where you can inject these fields.'</p>
</li>
</ul>
</div>
</li>
</ul>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_8_oct_2018_141658_thomas_turrell_croft_wrote">on 8 Oct 2018, 14:16:58 Thomas Turrell-Croft wrote:</h3>
<div class="paragraph">
<p>Spring 4.3 does not required the autowired annotation if there is only one constructor.  See the <a href="https://spring.io/blog/2016/03/04/core-container-refinements-in-spring-framework-4-3#implicit-constructor-injection-for-single-constructor-scenarios">blog post</a> announcing 4.3 for more details.</p>
</div>
<div class="paragraph">
<p>I would suggest that the compliant solution should look more like:</p>
</div>
<div class="listingblock">
<div class="content">
<pre>@Controller
public class HelloWorld {

  private final String name;

  public HelloWorld(String name) {
    this.name = name;
   // ...
  }

  // ...
}</pre>
</div>
</div>
<div class="paragraph">
<p>I believe that the dependancies should be <code>final</code>.</p>
</div>
</div>
</div>
</div>