In Javadoc you can add a reference to a package, class or member with tags `@link` and `@linkplain`. The referenced elements should exist in your classpath, otherwise you won't be able to build a javadoc.


== Noncompliant Code Example

[source,java]
----
/**
 * prints {@link NonExistingType} "Hello World!"  // Noncompliant
 */
public static void main(String[] args) {
    System.out.println("Hello World!");
}
----


== Compliant Solution

[source,java]
----
/**
 * prints {@link String} "Hello World!"
 */
public static void main(String[] args) {
    System.out.println("Hello World!");
}
----


== See

Javadoc Documentation:  https://docs.oracle.com/javase/8/docs/technotes/tools/windows/javadoc.html[https://docs.oracle.com/javase/8/docs/technotes/tools/windows/javadoc.html]

ifdef::env-github,rspecator-view[]

'''

== Implementation Specification
(visible only on this page)


=== Message

Make sure this link points to an existing package, class or member.


=== Highlighting

* Primary: the reference after `@link` or `@linkplain`

endif::env-github,rspecator-view[]
