== Why is this an issue?

Even if there's no data to be returned from a procedure, each procedure should at least return a boolean to indicate when procedure execution failed, so callers can respond accordingly.


=== Noncompliant code example

[source,rpg]
----
     D MyProc          PI
     D   param1                      10A   const
      /free
        // do work...
        return;  // Noncompliant
      /end-free
----


=== Compliant solution

[source,rpg]
----
     D MyProc          PI              N
     D   param1                      10A   const
      /free
        // do work...
        return *ON;
      /end-free
----



ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
