<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>For better readability, <code>END</code> statements must have the same indentation level as their matching opening <code>PROCEDURE</code>, <code>SELECT</code>, and <code>DO</code> statements.</p>
</div>
<div class="paragraph">
<p>This rule does not verify the alignment of the compound statements <code>DO ... END;</code> (type 1) and <code>BEGIN ... END;</code>. They are mostly used together with the <code>IF</code> and <code>ON</code> statements and are subject to different indentation rules.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-pli" data-lang="pli">foo: proc options(main);
  declare i fixed decimal init (0);

  do i = 0 to 9;
    if i = 0 then /* The do was forgotten here */
      put list ('Initialization...');
      put list ('i = 0');
      end; /* Non-Compliant - should be aligned with the DO statement */
  end; /* Non-Compliant - should be aligned with the PROCEDURE statement */

/* The procedure's end seems to be missing */</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-pli" data-lang="pli">foo: proc options(main);
  declare i fixed decimal init (0);

  do i = 0 to 9;
    if i = 0 then do;
      put list ('Initialization...');
      put list ('i = 0');
    end; /* This end's alignment is not verified by this rule */
  end; /* Compliant */
end; /* Compliant */</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>This statement should start at column n.</p>
</div>
</div>
</div>
</div>