== How to fix it in HTTPX

=== Code examples

include::../../common/fix/code-rationale.adoc[]

:cert_variable_name: verify
:cert_variable_unsafe_value: False
:cert_variable_safe_value: True

include::../../common/fix/code-rationale-setting.adoc[]

==== Noncompliant code example

[source,python,diff-id=31,diff-type=noncompliant]
----
import httpx

httpx.get('https://example.com', verify=False)  # Noncompliant
----

==== Compliant solution

[source,python,diff-id=31,diff-type=compliant]
----
import httpx

# By default, certificate validation is enabled
httpx.get('https://example.com')
----

=== How does this work?

include::../../common/fix/validation.adoc[]
