== How to fix it in OpenSSL

=== Code examples

include::../../common/fix/code-rationale.adoc[]

Certificate validation is not enabled by default and has to be explicitly
enabled through `set_verify`. By setting `SSL.VERIFY_NONE`, no certificate validation will occur.

==== Noncompliant code example

[source,python,diff-id=1,diff-type=noncompliant]
----
from OpenSSL import SSL
from OpenSSL.crypto import X509

ctx1 = SSL.Context(SSL.TLSv1_2_METHOD)  # Noncompliant

ctx2 = SSL.Context(SSL.TLSv1_3_METHOD)
ctx2.set_verify(SSL.VERIFY_NONE, verify_callback)  # Noncompliant

def bad_callback(
    conn: SSL.Connection, cert: X509, errnum: int, depth: int, ok: int
) -> bool:
    return True

ctx3 = SSL.Context(SSL.TLS_CLIENT_METHOD)
ctx3.set_verify(SSL.VERIFY_PEER, bad_callback)  # Noncompliant
----

==== Compliant solution

[source,python,diff-id=1,diff-type=compliant]
----
from OpenSSL import SSL

ctx1 = SSL.Context(SSL.TLSv1_2_METHOD)
ctx1.set_verify(SSL.VERIFY_PEER)

ctx2 = SSL.Context(SSL.TLSv1_3_METHOD)
ctx2.set_verify(SSL.VERIFY_PEER | SSL.VERIFY_FAIL_IF_NO_PEER_CERT)

ctx3 = SSL.Context(SSL.TLS_CLIENT_METHOD)
ctx3.set_verify(SSL.VERIFY_PEER | SSL.VERIFY_FAIL_IF_NO_PEER_CERT | SSL.VERIFY_CLIENT_ONCE, verify_callback)
----

By omitting the `verify_callback` argument, OpenSSL will use its built-in verification process. This can be trusted to be secure by default (if the correct flags are set.)

=== How does this work?

include::../../common/fix/validation.adoc[]

