== How to fix it in requests

=== Code examples

include::../../common/fix/code-rationale.adoc[]

==== Noncompliant code example

[source,python,diff-id=1,diff-type=noncompliant]
----
import requests

requests.request('GET', 'https://example.com', verify=False) # Noncompliant
----

==== Compliant solution

[source,python,diff-id=1,diff-type=noncompliant]
----
import requests

# By default, certificate validation is enabled
requests.request('GET', 'https://example.com', verify=True)
----

=== How does this work?

include::../../common/fix/default-implementation.adoc[]
