== How to fix it in Requests

=== Code examples

include::../../common/fix/code-rationale.adoc[]

:cert_variable_name: verify
:cert_variable_unsafe_value: False
:cert_variable_safe_value: True

include::../../common/fix/code-rationale-setting.adoc[]

==== Noncompliant code example

[source,python,diff-id=11,diff-type=noncompliant]
----
import requests

requests.request('GET', 'https://example.com', verify=False) # Noncompliant
----

==== Compliant solution

[source,python,diff-id=11,diff-type=compliant]
----
import requests

# By default, certificate validation is enabled
requests.request('GET', 'https://example.com')
----

=== How does this work?

include::../../common/fix/validation.adoc[]
