== How to fix it in SSL

=== Code examples

include::../../common/fix/code-rationale.adoc[]

==== Noncompliant code example

[source,python,diff-id=1,diff-type=noncompliant]
----
import ssl

ctx1 = ssl._create_unverified_context() # Noncompliant: by default certificate validation is not done
ctx2 = ssl._create_stdlib_context() # Noncompliant: by default certificate validation is not done

ctx3 = ssl.create_default_context()
ctx3.verify_mode = ssl.CERT_NONE # Noncompliant
----

==== Compliant solution

[source,python,diff-id=1,diff-type=noncompliant]
----
import ssl

ctx = ssl.create_default_context()
ctx.verify_mode = ssl.CERT_REQUIRED

# By default, certificate validation is enabled
ctx = ssl._create_default_https_context()
----

=== How does this work?

include::../../common/fix/validation.adoc[]
