== How to fix it in Python Standard Library

=== Code examples

include::../../common/fix/code-rationale.adoc[]

Certificate validation is not enabled by default when ``++_create_unverified_context++`` is used. It is recommended to use ``++create_default_context++`` instead. This method creates secure contexts that will verify certificates and check server hostnames (see S5527 for more information) by default.

==== Noncompliant code example

[source,python,diff-id=21,diff-type=noncompliant]
----
import ssl

ctx1 = ssl._create_unverified_context() # Noncompliant

ctx2 = ssl.create_default_context()
ctx2.verify_mode = ssl.CERT_NONE # Noncompliant
----

==== Compliant solution

[source,python,diff-id=21,diff-type=compliant]
----
import ssl

# By default, certificate validation is enabled
ctx1 = ssl.create_default_context()

# The context can be specified for client or server use
ctx2 = ssl.create_default_context(ssl.Purpose.SERVER_AUTH)
----

=== How does this work?

include::../../common/fix/validation.adoc[]
