== How to fix it in Python Standard Library

=== Code examples

include::../../common/fix/code-rationale.adoc[]

Certificate validation is not enabled by default when ``++_create_unverified_context++``
is used. It is recommended to use ``++create_default_context++`` instead to
create a secure context that validates certificates.

==== Noncompliant code example

[source,python,diff-id=21,diff-type=noncompliant]
----
import ssl

ctx1 = ssl._create_unverified_context() # Noncompliant

ctx2 = ssl.create_default_context()
ctx2.verify_mode = ssl.CERT_NONE # Noncompliant
----

==== Compliant solution

[source,python,diff-id=21,diff-type=compliant]
----
import ssl

ctx1 = ssl.create_default_context()
ctx1.verify_mode = ssl.CERT_REQUIRED

# By default, certificate validation is enabled
ctx2 = ssl.create_default_context()
----

=== How does this work?

include::../../common/fix/validation.adoc[]
