<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>If a compiler implementation detects that a call to a copy constructor is redundant, then it is permitted to omit that call, even if the copy constructor has a side effect other than the construction of a copy of the object. This is called "copy elision".</p>
</div>
<div class="paragraph">
<p>It is therefore important to ensure that a copy constructor does not modify the program state, since the number of such modifications may be indeterminate.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">class A
{
public:
  A ( A const &amp; rhs )
  : m_i ( rhs.m_i )
  {
    ++m_static; // Non-compliant, might not be executed
  }
private:
  int32_t m_i;
  static int32_t m_static;
};</code></pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p>MISRA C&#43;&#43;:2008, 12-8-1</p>
</li>
<li>
<p><a href="https://wiki.sei.cmu.edu/confluence/x/gXs-BQ">CERT, OOP58-CPP.</a> - Copy operations must not mutate the source object</p>
</li>
</ul>
</div>
<hr>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>This code within a copy constructor might not be executed and should be removed.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_14_oct_2014_174957_ann_campbell_wrote">on 14 Oct 2014, 17:49:57 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~samuel.mercier] please correct the following</p>
</div>
<div class="ulist">
<ul>
<li>
<p>There is no message</p>
</li>
<li>
<p>Add a See section to the description listing the appropriate MISRA number</p>
</li>
<li>
<p>Fill in Applicability.</p>
</li>
<li>
<p>Make sure the appropriate MISRA C and MISRA C&#43;&#43; fields on the references tab are filled in</p>
</li>
</ul>
</div>
</div>
<div class="sect2">
<h3 id="_on_17_oct_2014_143802_ann_campbell_wrote">on 17 Oct 2014, 14:38:02 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~samuel.mercier] I&#8217;m tempted to tag this with 'bug'. WDYT?</p>
</div>
</div>
</div>
</div>