include::../summary.adoc[]

== Why is this an issue?

include::../rationale.adoc[]

== How to fix it in XCode

=== Code examples

==== Noncompliant code example

[source,plaintext,diff-id=1,diff-type=noncompliant]
----
{
	/* ... */
	objects = {
		ABCDEF123456789ABCDFE123 /* Release */ = {
			isa = XCBuildConfiguration;
			buildSettings = {
				/* ... */
				SWIFT_ACTIVE_COMPILATION_CONDITIONS = "DEBUG"; /* Noncompliant */
				STRIP_INSTALLED_PRODUCT = NO; /* Noncompliant */
				ENABLE_TESTABILITY = YES; /* Noncompliant */
				/* ... */
			};
			name = Release;
		};
	};
}
----

This snippet of `pbxproj` has a misconfigured "Release" profile where:

* Debug symbols will not be striped from the binary as
** https://developer.apple.com/documentation/xcode/build-settings-reference#Enable-Testability[Enable Testability] is set to "Yes"
** https://developer.apple.com/documentation/xcode/build-settings-reference#Enable-Testability[Strip Linked Product] is set to "No"
* The `DEBUG` https://developer.apple.com/documentation/xcode/build-settings-reference#Active-Compilation-Conditions[Active Compilation Conditions] is set

==== Compliant solution

[source,plaintext,diff-id=1,diff-type=compliant]
----
{
	/* ... */
	objects = {
		ABCDEF123456789ABCDFE123 /* Release */ = {
			isa = XCBuildConfiguration;
			buildSettings = {
				/* ... */
				SWIFT_ACTIVE_COMPILATION_CONDITIONS = "";
				STRIP_INSTALLED_PRODUCT = YES;
				ENABLE_TESTABILITY = NO;
				/* ... */
			};
			name = Release;
		};
	};
}

----

== Resources

=== Documentation

* Apple Developer Documentation - https://developer.apple.com/documentation/xcode/build-settings-reference[Build settings reference]

include::../common/resources/standards.adoc[]

