== Why is this an issue?

include::../description.adoc[]


=== Noncompliant code example

[source,php]
----
"/[ab]|a/"   // The "|a" is redundant because "[ab]" already matches "a"
"/.*|a/"     // .* matches everything, so any other alternative is redundant
----


=== Compliant solution

[source,php]
----
"/[ab]/"
"/.*/"
----
