include::../description.adoc[]


== Noncompliant Code Example

----
"/[ab]|a/"   // The "|a" is redundant because "[ab]" already matches "a"
"/.*|a/"     // .* matches everything, so any other alternative is redundant
----


== Compliant Solution

----
"/[ab]/"
"/.*/"
----