This rule raises an issue when the same import statement appears multiple times in a Groovy file.

== Why is this an issue?

Duplicate import statements create unnecessary code clutter without providing any functional benefit.

While duplicate imports don't cause compilation errors or runtime issues in Groovy, they make the code harder to read and maintain.

Keeping import statements clean and minimal helps developers quickly understand the dependencies of a file and reduces visual noise in the codebase.

=== What is the potential impact?

This issue has minimal impact on functionality but affects code readability and maintainability. Clean import sections help developers understand file dependencies more quickly.

== How to fix it

Remove the duplicate import statement. Keep only one instance of each import.

=== Code examples

==== Noncompliant code example

[source,groovy,diff-id=1,diff-type=noncompliant]
----
import java.util.List
import java.util.Map
import java.util.List // Noncompliant

class MyClass {
    List<String> items
    Map<String, Object> data
}
----

==== Compliant solution

[source,groovy,diff-id=1,diff-type=compliant]
----
import java.util.List
import java.util.Map

class MyClass {
    List<String> items
    Map<String, Object> data
}
----

== Resources

=== Related rules

 * RSPEC-1128 - https://rules.sonarsource.com/css/RSPEC-1128/[Unused imports should be removed (CSS)]
