This rule raises an issue when the same import statement appears multiple times in a Groovy file.

== Why is this an issue?

Duplicate import statements create unnecessary code clutter without providing any functional benefit. They often result from copy-paste operations or merge conflicts that weren't properly cleaned up.

While duplicate imports don't cause compilation errors or runtime issues in Groovy, they make the code harder to read and maintain. They can also indicate poor code organization or incomplete refactoring.

Keeping import statements clean and minimal helps developers quickly understand the dependencies of a file and reduces visual noise in the codebase.

=== What is the potential impact?

This issue has minimal impact on functionality but affects code readability and maintainability. Clean import sections help developers understand file dependencies more quickly.

== How to fix it

Remove the duplicate import statement. Keep only one instance of each import.

=== Code examples

==== Noncompliant code example

[source,groovy,diff-id=1,diff-type=noncompliant]
----
import java.util.List
import java.util.Map
import java.util.List // Noncompliant

class MyClass {
    List<String> items
    Map<String, Object> data
}
----

==== Compliant solution

[source,groovy,diff-id=1,diff-type=compliant]
----
import java.util.List
import java.util.Map

class MyClass {
    List<String> items
    Map<String, Object> data
}
----

== Resources

=== Documentation

 * CodeNarc DuplicateImport Rule - https://codenarc.org/codenarc-rules-imports.html#duplicateimport-rule[Official CodeNarc documentation for the DuplicateImport rule]

 * Groovy Language Documentation - Import Statements - https://groovy-lang.org/structure.html#_imports[Official Groovy documentation on import statements and their usage]

=== Related rules

 * RSPEC-1128 - https://rules.sonarsource.com/css/RSPEC-1128/[Unused imports should be removed (CSS)]
