== Why is this an issue?

The ``++char++`` type within {cpp} is defined for use with the implementation character set. It is implementation-defined if ``++char++`` is signed or unsigned, and it is therefore unsuitable for use with numeric data.


Character values consist of character literals or strings. A character set maps text characters onto numeric values; the character value is the text itself.


=== Noncompliant code example

[source,cpp]
----
char c = 10; // Noncompliant
----


=== Compliant solution

[source,cpp]
----
char c = 'a'; // Compliant
----


== Resources

* MISRA C:2004, 6.1 - The plain char type shall be used only for the storage and use of character values.
* MISRA {cpp}:2008, 5-0-11 - The plain char type shall only be used for the storage and use of character values.


ifdef::env-github,rspecator-view[]
'''
== Comments And Links
(visible only on this page)

=== duplicates: S810

=== relates to: S810

=== on 21 Oct 2014, 18:29:39 Ann Campbell wrote:
\[~samuel.mercier] please provide a message.

=== on 13 Jan 2015, 09:30:09 Samuel Mercier wrote:
merged with RSPEC-810

endif::env-github,rspecator-view[]
