In Dockerfiles, it is recommended to use the exec form for `CMD` and `ENTRYPOINT` instructions. The exec form, which is represented as a JSON array, ensures that the process runs directly without being wrapped in a shell. This allows OS signals like SIGTERM and SIGINT to be received by the process. This practice enhances the reliability and control of your Docker containers.

== Why is this an issue?

Using the shell form instead of the exec form for CMD and ENTRYPOINT instructions in Dockerfiles can lead to several issues. When you use the shell form, the executable runs as a child process to a shell, which does not pass OS signals. This can cause problems when trying to gracefully stop containers because the main process will not receive the signal intended to terminate it. Moreover, the exec form provides more control and predictability over the execution of the command. It does not invoke a command shell, which means it does not have the potential side effects of shell processing. 

== How to fix it

=== Code examples

==== Noncompliant code example

[source,docker,diff-id=1,diff-type=noncompliant]
----
FROM scratch
ENTRYPOINT echo "Welcome!"
----

==== Compliant solution

[source,docker,diff-id=1,diff-type=compliant]
----
FROM scratch
ENTRYPOINT ["echo", "Welcome!"]
----

== Resources
=== Documentation

* Docker Build Checks - https://docs.docker.com/reference/build-checks/json-args-recommended/[JSONArgsRecommended]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Replace this shell form with exec form.

=== Highlighting

Highlight the value of the CMD or ENTRYPOINT instruction.

'''
== Comments And Links
(visible only on this page)

endif::env-github,rspecator-view[]
