== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,javascript]
----
foo(); bar(); // Noncompliant
----

=== Compliant solution

[source,javascript]
----
foo();
bar();
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../highlighting.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
