include::../description.adoc[]

== Noncompliant Code Example

----
foo(); bar(); // Noncompliant
----

== Compliant Solution

----
foo();
bar();
----

ifdef::env-github,rspecator-view[]
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
