This rule raises an issue when a variable is declared in an if short statement but is only used in the condition itself, suggesting the declaration might be unnecessary.

== Why is this an issue?

Go's if short statement allows you to declare and initialize a variable before the condition check. This feature is designed to limit the variable's scope to the if-else block, which is useful when you need the variable in multiple parts of the conditional logic.

However, if the declared variable is only used in the condition and nowhere else in the if or else blocks, the short statement declaration adds unnecessary complexity. In such cases, you can often simplify the code by moving the expression directly into the condition or restructuring the logic.

Using if short statements appropriately helps create cleaner, more readable code by:

* Limiting variable scope to where it's actually needed
* Reducing the number of variables in the broader function scope
* Making the relationship between variable declaration and usage more explicit

When a variable is declared but underutilized, it can confuse readers about the intended purpose and scope of the variable.

=== What is the potential impact?

This issue primarily affects code readability and maintainability. While it doesn't cause runtime problems, it can make code harder to understand and maintain. Unnecessary variable declarations can also slightly impact performance in tight loops, though this is typically negligible.

== How to fix it

Move the expression directly into the condition when the variable is only used there.

=== Code examples

==== Noncompliant code example

[source,go,diff-id=1,diff-type=noncompliant]
----
if x := getValue(); x > 10 { // Noncompliant: x is only used in condition
    fmt.Println("High value")
} else {
    fmt.Println("Low value")
}
----

==== Compliant solution

[source,go,diff-id=1,diff-type=compliant]
----
if getValue() > 10 {
    fmt.Println("High value")
} else {
    fmt.Println("Low value")
}
----

== Resources

=== Documentation

 * Go Tour - If with a short statement - https://go.dev/tour/flowcontrol/6[Official Go documentation explaining if short statements]

 * Go Tour - If and else - https://go.dev/tour/flowcontrol/7[Official Go documentation on variable scope in if-else blocks]

 * Effective Go - Control structures - https://go.dev/doc/effective_go#control-structures[Best practices for Go control structures including if statements]
