== Why is this an issue?

Type hints can be used to communicate the intended type of a given variable. These are not enforced at runtime and not respecting them might not necessarily lead to runtime errors.


It is however confusing and could lead to maintainability issues.


=== Noncompliant code example

[source,python]
----
def my_function():
    my_int: int = "string"  # Noncompliant
----


=== Compliant solution

[source,python]
----
def my_function():
    my_str: str = "string" 
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

include::highlighting.adoc[]

'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
