<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Because the value in a variable of an unsigned type can never be less than zero, testing to see if it is negative is a useless operation which can only confuse future readers of the code.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">unsigned int i = 0; // the lowest value this var can have
...
if (i &gt;= 0) { // Noncompliant
  do_x(i);
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">unsigned int i = 0;
...
do_x(i);</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove this test of "XX". It is by definition &gt;= 0.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_3_jul_2015_093239_massimo_paladin_wrote">on 3 Jul 2015, 09:32:39 Massimo PALADIN wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] shall we add references which appear in <a data-rspec-id="S2583" class="rspec-auto-link">RSPEC-2583</a>?</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_6_jul_2015_145123_ann_campbell_wrote">on 6 Jul 2015, 14:51:23 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>assigning to [~evgeny.mandrikov] pending outcome of merger into <a data-rspec-id="S2583" class="rspec-auto-link">RSPEC-2583</a></p>
</div>
</div>
<div class="sect2">
<h3 id="_on_27_aug_2015_125814_evgeny_mandrikov_wrote">on 27 Aug 2015, 12:58:14 Evgeny Mandrikov wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] with [~freddy.mallet] and [~massimo.paladin] we decided to go ahead without merge for now</p>
</div>
</div>
</div>
</div>