== How to fix it in excon

=== Code examples

include::../../common/fix/code-rationale.adoc[]

==== Noncompliant code example

[source,ruby,diff-id=1,diff-type=noncompliant]
----
Excon.get("https://example.com", ssl_verify_peer: false) # Noncompliant
----

==== Compliant solution

[source,ruby,diff-id=1,diff-type=compliant]
----
Excon.get("https://example.com")
----
