== How to fix it in httpclient

=== Code examples

include::../../common/fix/code-rationale.adoc[]

==== Noncompliant code example

[source,ruby,diff-id=1111,diff-type=noncompliant]
----
client = HTTPClient.new
client.ssl_config.verify_mode = OpenSSL::SSL::VERIFY_NONE # Noncompliant
response = client.get("https://example.com")
----

==== Compliant solution

[source,ruby,diff-id=1111,diff-type=compliant]
----
client = HTTPClient.new
response = client.get("https://example.com")
----
