== How to fix it in httparty

=== Code examples

include::../../common/fix/code-rationale.adoc[]

==== Noncompliant code example

[source,ruby,diff-id=11,diff-type=noncompliant]
----
HTTParty.get("https://example.com", verify: false) # Noncompliant
----

==== Compliant solution

[source,ruby,diff-id=11,diff-type=compliant]
----
HTTParty.get("https://example.com")
----
