== How to fix it in rest-client

=== Code examples

include::../../common/fix/code-rationale.adoc[]

==== Noncompliant code example

[source,ruby,diff-id=111,diff-type=noncompliant]
----
RestClient::Resource.new("https://example.com", verify_ssl: OpenSSL::SSL::VERIFY_NONE) # Noncompliant
----

==== Compliant solution

[source,ruby,diff-id=111,diff-type=compliant]
----
RestClient::Resource.new("https://example.com")
----
