== How to fix it in Ruby Standard Library

=== Code examples

include::../../common/fix/code-rationale.adoc[]

==== Noncompliant code example

[source,ruby,diff-id=1111,diff-type=noncompliant]
----
http = Net::HTTP.new("https://example.com")
http.use_ssl = true
http.verify_mode = OpenSSL::SSL::VERIFY_NONE # Noncompliant
response = http.get("/")
----

==== Compliant solution

[source,ruby,diff-id=1111,diff-type=compliant]
----
http = Net::HTTP.new("https://example.com")
http.use_ssl = true
response = http.get("/")
----
