== How to fix it in typhoeus

=== Code examples

include::../../common/fix/code-rationale.adoc[]

==== Noncompliant code example

[source,ruby,diff-id=1111111,diff-type=noncompliant]
----
Typhoeus.get("https://example.com", ssl_verifypeer: false) # Noncompliant
----

==== Compliant solution

[source,ruby,diff-id=1111111,diff-type=compliant]
----
Typhoeus.get("https://example.com")
----
