Generic types should not be used raw (without type parameters).
To fix this issue, add the type parameters.

== Why is this an issue?

A generic type is a generic class or interface that is parameterized over types.
For example, `java.util.List` has one type parameter: the type of its elements.

When generic types are used raw (without type parameters), the compiler is not able to do generic type checking.
Therefore, you might have unsafe code that fails at runtime.

== How to fix it

You should add type parameters.

=== Code examples

==== Noncompliant code example

[source,java,diff-id=1,diff-type=noncompliant]
----
List integers = new ArrayList<>();

// Some code here could add non-integer objects to the list by mistake.

Integer a = (Integer) integers.get(0);  // Possible ClassCast exception!
----

==== Compliant solution

[source,java,diff-id=1,diff-type=compliant]
----
List<Integer> integers = new ArrayList<>();

// If some code tries to add a non-integer object here it will not compile.

Integer a = integers.get(0);  // No need to cast anymore.
----

== Resources

=== Documentation

* https://docs.oracle.com/javase/tutorial/java/generics/rawTypes.html[Raw types] in the Java Tutorial.


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Provide the parametrised type for this generic.


=== Highlighting

type name


'''
== Comments And Links
(visible only on this page)

=== on 31 Oct 2018, 09:35:37 Nicolas Peru wrote:
\[~alexandre.gigleux] I would suggest title to be reworked to : Don't use raw types. The wording seems dodgy.

=== on 31 Oct 2018, 12:31:09 Ann Campbell wrote:
"Raw types should not be used"?

endif::env-github,rspecator-view[]
