Nested ``++IF...ELSE++``, ``++WHILE++`` and ``++TRY...CATCH++`` statements is a key ingredient for making what's known as "Spaghetti code".


Such code is hard to read, refactor and therefore maintain.

== Noncompliant Code Example

With the default threshold of 4:

----
IF @flag1 = 1 -- Compliant - depth = 1
  BEGIN
    IF @flag2 = 2 -- Compliant - depth = 2
      BEGIN
        WHILE @var1 > 0  -- Compliant - depth = 3
          BEGIN
            IF @flag3 = 3 -- Compliant - depth = 4, not exceeding the limit
              BEGIN
                IF @flag4 = 4 -- Noncompliant - depth = 5
                  BEGIN
                    IF @flag5 = 5 -- Depth = 6, exceeding the limit, but issues are only reported on depth = 5
                      BREAK
                  END
              END
            SET @var1 = @var1 - 1
          END
      END
  END
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::parameters.adoc[]

include::../highlighting.adoc[]

'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
