== Why is this an issue?

Declaring the package and class together has been deprecated since ActionScript 3. The package definition should be declared outside of the class definition even if the old syntax is still supported. 


=== Noncompliant code example

[source,flex]
----
class P.A {...} 
----


=== Compliant solution

[source,flex]
----
package P {    
    class A {...}
}
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Make the Package definition nest the Class definition


'''
== Comments And Links
(visible only on this page)

=== on 5 Nov 2013, 08:41:48 Freddy Mallet wrote:
Is implemented by \https://jira.codehaus.org/browse/SONARPLUGINS-3243

endif::env-github,rspecator-view[]
