== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

With default provided regular expression ``++^[A-Z][a-zA-Z0-9_]*$++``:

[source,apex]
----
class myClass {}
----

=== Compliant solution

[source,apex]
----
class MyClass {}
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
