Kotlin coroutines follow the principle of structured concurrency. This helps in preventing resource leaks and ensures that scopes are only exited once all child coroutines have exited. Hence, structured concurrency enables developers to build concurrent applications while having to worry less about cleaning up concurrent tasks manually.

It is possible to break this concept of structured concurrency in various ways, however. Generally this is not advised, as it can open the door to coroutines being leaked or lost.

This rule raises an issue when it detects that the structured concurrency principles are violated. It avoids reporting on valid use cases and in situations where developers have consciously opted into using delicate APIs (e.g. by using the `@OptIn` annotation) and hence should be aware of the possible pitfalls.
// If you want to factorize the description uncomment the following line and create the file.
//include::../description.adoc[]

== Noncompliant Code Example
----
fun main() {
  GlobalScope.launch {
    // Do some work
  }.join()
}
----

----
val coroutineScope = CoroutineScope(Job())
coroutineScope.launch(Job()) {
  // Do some work
}.join()
----

----
  coroutineScope {
    launch(SupervisorJob()) {
      // Do some work
    }
  }
----

== Compliant Solution
----
@OptIn(DelicateCoroutinesApi::class)
fun main() {
  GlobalScope.launch {
    // Do some work
  }.join()
}
----

----
fun main() {
  runBlocking { worker() }
}

suspend fun worker() {
  coroutineScope {
    launch {
      // Do some work
    }
  }.join()
}
----

----
  val coroutineScope = CoroutineScope(Job())
  coroutineScope.launch {
    // Do some work
  }.join()
----

----
  supervisorScope {
    launch {
      // Do some work
    }
  }
----

== See

* https://kotlinlang.org/docs/coroutines-basics.html#structured-concurrency[Structured concurrency] in the Kotlin docs
* https://kotlin.github.io/kotlinx.coroutines/kotlinx-coroutines-core/kotlinx.coroutines/-global-scope/index.html[GlobalScope documentation]
* https://kotlin.github.io/kotlinx.coroutines/kotlinx-coroutines-core/kotlinx.coroutines/-coroutine-scope/index.html[coroutineScope documentation]
* https://developer.android.com/kotlin/coroutines/coroutines-best-practices[Android coroutines best practices]
