<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p><code>std::move</code> is just a <code>static_cast</code> to an rvalue reference. It produces an rvalue of xvalue category. Calling <code>std::move</code> on an rvalue is redundant and might lead to preventing copy elision <a href="https://jira.sonarsource.com/browse/RSPEC-5274">RSPEC-5274</a></p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">class A {
  std::string s;
public:
  explicit A(std::string s) :s{std::move(s)} {}
  A(const A&amp;) = default;
  A(A&amp;&amp;) = default;
};

A returnAByValue();
A&amp;&amp; returnARvalueRef();
void takeRvalueRefOfA(A&amp;&amp;);

int uncompliant()
{
  takeRvalueRefOfA(std::move(A("I'm an rvalue"))); // the constructed object is already an rvalue
  A a2 = std::move(returnARvalueRef()); // the returned variable is already an rvalue
  A a1 = std::move(returnAByValue()); // the returned variable is already an rvalue. This prevent copy elision since move constructor will be called.
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">class A {
  std::string s;
public:
  explicit A(std::string s) :s{std::move(s)} {}
  A(const A&amp;) = default;
  A(A&amp;&amp;) = default;
};

A returnAByValue();
A&amp;&amp; returnARvalueRef();
void takeRvalueRefOfA(A&amp;&amp;);

int compliant()
{
  takeRvalueRefOfA(A("I'm an rvalue")); // it compiles without the unnecessary std::move
  A a2 = returnARvalueRef(); // move constructor is called without calling std::move
  A a1 = returnAByValue(); // copy elision
}</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_duplicates_s5274">duplicates: <a data-rspec-id="S5274" class="rspec-auto-link">S5274</a></h3>

</div>
<div class="sect2">
<h3 id="_on_2_sep_2019_194612_lo誰c_joly_wrote">on 2 Sep 2019, 19:46:12 Lo誰c Joly wrote:</h3>
<div class="paragraph">
<p>\[~geoffray.adde]: Shouldn&#8217;t it just be merged into <a data-rspec-id="S5274" class="rspec-auto-link">RSPEC-5274</a>?</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_3_sep_2019_173744_lo誰c_joly_wrote">on 3 Sep 2019, 17:37:44 Lo誰c Joly wrote:</h3>
<div class="paragraph">
<p>After discussion with [~geoffray.adde]: Merge it into <a data-rspec-id="S5274" class="rspec-auto-link">RSPEC-5274</a></p>
</div>
</div>
</div>
</div>