== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

With provided regular expression ``++^[a-z][a-zA-Z0-9_]*$++``:

[source,apex]
----
public void DoSomething(){...}
----

=== Compliant solution

[source,apex]
----
public void doSomething(){...}
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
