include::../description.adoc[]

== Noncompliant Code Example

With provided regular expression ``++^[a-z][a-zA-Z0-9_]*$++``:

----
public void DoSomething(){...}
----

== Compliant Solution

----
public void doSomething(){...}
----

ifdef::env-github,rspecator-view[]
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
