include::../description.adoc[]

== Noncompliant Code Example

With provided regular expression `+^[a-z][a-zA-Z0-9_]*$+`:

----
public void DoSomething(){...}
----

== Compliant Solution

----
public void doSomething(){...}
----
