<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>To prevent portability issues <code>!=</code>, <code>!&gt;</code> and <code>!&lt;</code> operators should be replaced by the ANSI standard operators: <code>&lt;&gt;</code>, <code>&lt;=</code> or <code>&gt;=</code>.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-text" data-lang="text">SELECT C1 FROM S1TESTMD WHERE BIRTHDATE != 2000</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-text" data-lang="text">SELECT C1 FROM S1TESTMD WHERE BIRTHDATE &lt;&gt; 2000</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Replace this "XX" operator with the ANSI standard equivalent: "YY" operator.</p>
</div>
</div>
</div>
</div>