== Why is this an issue?

To prevent portability issues ``++!=++``, ``++!>++`` and ``++!<++`` operators should be replaced by the ANSI standard operators: ``++<>++``, ``++<=++`` or ``++>=++``.


=== Noncompliant code example

[source,text]
----
SELECT C1 FROM S1TESTMD WHERE BIRTHDATE != 2000
----


=== Compliant solution

[source,text]
----
SELECT C1 FROM S1TESTMD WHERE BIRTHDATE <> 2000
----


