To prevent portability issues ``++!=++``, ``++!>++`` and ``++!<++`` operators should be replaced by the ANSI standard operators: ``++<>++``, ``++<=++`` or ``++>=++``.


== Noncompliant Code Example

[source,text]
----
SELECT C1 FROM S1TESTMD WHERE BIRTHDATE != 2000
----


== Compliant Solution

[source,text]
----
SELECT C1 FROM S1TESTMD WHERE BIRTHDATE <> 2000
----


