== Why is this an issue?

Strings and integral types are typically used as indexers. When some other type is required, it typically indicates design problems, and potentially a situation where a method should be used instead.


=== Noncompliant code example

[source,csharp]
----
public int this[MyCustomClass index]  // Noncompliant
{  
    // get and set accessors  
}
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Use string, integral, index or range type here, or refactor this indexer into a method.


=== Highlighting

``++this[xxx]++``


endif::env-github,rspecator-view[]
