The W3C specifications define the valid pseudo-element selectors. Only the official and browser-specific pseudo-element selectors should be used to get the expected impact in the final rendering.


== Noncompliant Code Example

----
a::beforre { /* Noncompliant; there is a typo on the word "before" */
...
}
----


== Compliant Solution

----
a::before {
...
}
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

endif::env-github,rspecator-view[]
