== Why is this an issue?

The W3C specifications define the valid pseudo-element selectors. Only the official and browser-specific pseudo-element selectors should be used to get the expected impact in the final rendering.


=== Noncompliant code example

[source,css]
----
a::beforre { /* Noncompliant; there is a typo on the word "before" */
...
}
----


=== Compliant solution

[source,css]
----
a::before {
...
}
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

endif::env-github,rspecator-view[]
