<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>There are some situations in C&#43;&#43; where certain parts of expressions may not be evaluated. If these sub-expressions contain side effects then those side effects may or may not occur, depending on the values of other sub expressions. The operators which can lead to this problem are <code>&amp;&amp;</code> and <code>||</code>, where the evaluation of the right-hand operand is conditional on the value of the left-hand operand. The conditional evaluation of the right-hand operand of one of the logical operators can easily cause problems if the developer relies on a side effect occurring.</p>
</div>
<div class="paragraph">
<p>Operations that cause side effects are:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>accessing a volatile object</p>
</li>
<li>
<p>modifying an object</p>
</li>
<li>
<p>modifying a file</p>
</li>
<li>
<p>calling a function that performs any operations that cause changes in the state of the execution environment of the calling function.</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>This rule raises an issue when there is assignment or the use of the increment/decrement operators in right-hand operands.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">if ( ishigh &amp;&amp; ( x == i++ ) ) // Noncompliant
...
if ( ishigh &amp;&amp; ( x ==  getX() ) ) // Only acceptable if getX() is known to have no side effects</code></pre>
</div>
</div>
<div class="paragraph">
<p>The operations that cause side effects are accessing a volatile object, modifying an object, modifying a file, or calling a function</p>
</div>
<div class="paragraph">
<p>that does any of those operations, which cause changes in the state of the execution environment of the calling function.</p>
</div>
<div class="paragraph">
<p>For the time being, this rule only check that there is no assignment or no use of increment/decrement operators made in right hand operands.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p>MISRA C:2004, 12.4 - The right-hand operand of a logical &amp;&amp; or || operator shall not contain side effects.</p>
</li>
<li>
<p>MISRA C&#43;&#43;:2008, 5-14-1 - The right hand operand of a logical &amp;&amp; or || operator shall not contain side effects.</p>
</li>
<li>
<p>MISRA C:2012, 13.5 - The right hand operand of a logical &amp;&amp; or || operator shall not contain persistent side effects</p>
</li>
<li>
<p><a href="https://wiki.sei.cmu.edu/confluence/x/vdUxBQ">CERT, EXP02-C.</a> - Be aware of the short-circuit behavior of the logical AND and OR operators</p>
</li>
</ul>
</div>
<hr>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_is_related_to_s811">is related to: <a data-rspec-id="S811" class="rspec-auto-link">S811</a></h3>

</div>
<div class="sect2">
<h3 id="_on_31_mar_2015_190314_evgeny_mandrikov_wrote">on 31 Mar 2015, 19:03:14 Evgeny Mandrikov wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] implementation seems more complete (SQALE, description) than this spec.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_13_apr_2015_194626_evgeny_mandrikov_wrote">on 13 Apr 2015, 19:46:26 Evgeny Mandrikov wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] I&#8217;m wondering why blocker, but not active by default? Note that in implementation currently critical and active.</p>
</div>
</div>
</div>
</div>