<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Widget containers wrapping a single widget with no other parameter modifying the behavior of the UI is unnecessary and should be removed.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p><a href="https://api.flutter.dev/flutter/widgets/Container-class.html"><code>Container</code></a> is a <a href="https://api.flutter.dev/flutter/widgets/StatelessWidget-class.html"><code>StatelessWidget</code></a> that wraps a child widget and applies additional properties and transformations to it.</p>
</div>
<div class="paragraph">
<p>For example a container can:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>apply padding and margin</p>
</li>
<li>
<p>add borders</p>
</li>
<li>
<p>setting width and height</p>
</li>
<li>
<p>apply geometrical transformations, such as rotation, translation, and scaling</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>When none of these features are used, the container solely wraps its child widget. In this case the container is unnecessary and can be removed, making the code shorter and the intent clearer.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Remove the <code>Container</code> instance, replacing it with the <code>Widget</code> specified in its <code>child</code> parameter.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-dart" data-lang="dart">Widget widgetBuilder() =&gt;
  Container(child: MyWidget());</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-dart" data-lang="dart">Widget widgetBuilder() =&gt;
  MyWidget();</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Dart Docs - <a href="https://dart.dev/tools/linter-rules/avoid_unnecessary_containers">Dart Linter rule - avoid_unnecessary_containers</a></p>
</li>
<li>
<p>Flutter API Reference - <a href="https://api.flutter.dev/flutter/widgets/Container-class.html">Container class</a></p>
</li>
</ul>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Unnecessary instance of 'Container'.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>The identifier name of the constructor of <code>Container</code>: e.g. <code>Container</code> in <code>returningContainer() &#8658; Container(child: MyWidget())</code>.</p>
</div>
<div class="paragraph">
<p>The <code>new</code> keyword, if used, is not highlighted: e.g. <code>Container</code> in <code>returningContainer() &#8658; new Container(child: MyWidget())</code>.</p>
</div>
<div class="paragraph">
<p>If the "named constructor syntax" is used, the constructor name (<code>new</code>) and the period sign between the class name and the constructor name are also highlighted: e.g. <code>Container.new</code> in <code>returningContainer() &#8658; Container.new(child: MyWidget())</code>.</p>
</div>
</div>
</div>
</div>