Certain mathematical comparisons will always return the same value, and should simply not be made. 

These comparisons will always return ``++false++``:

* comparing a ``++char++`` with a numeric constant that's outside the ``++char++`` range
* comparing a ``++float++`` with a numeric constant that's outside the ``++float++`` range
* comparing a ``++long++`` with a numeric constant that's outside the ``++long++`` range
* comparing a ``++ulong++`` with a numeric constant that's outside the ``++ulong++`` range

== Noncompliant Code Example

[source,text]
----
float f = 42.0f;
const double d = double.MaxValue;
if (f <= d) { } // Noncompliant
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]