Certain mathematical comparisons will always return the same value, and should simply not be made. 

This comparison will always return ``++false++``:

* comparing a ``++float++`` with a ``++double++`` constant that's outside the ``++float++`` range

These will always return ``++true++``:

* comparing ``++aByte <= Byte.MaxValue++`` and ``++aByte >= Byte.MinValue++``
* comparing ``++anInt <= int.MaxValue++`` and ``++anInt >= int.MinValue++``
* comparing ``++aLong <= long.MaxValue++`` and ``++aLong >= long.MinValue++``


== Noncompliant Code Example

[source,text]
----
float f = 42.0f;
const double d = float.MaxValue + 1;
if (f <= d) { } // Noncompliant
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]