<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Double-checked locking can be used for lazy initialization of <code>volatile</code> fields, but only if field assignment is the last step in the <code>synchronized</code> block. Otherwise you run the risk of threads accessing a half-initialized object.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">public class MyClass {

  private volatile List&lt;String&gt; strings;

  public List&lt;String&gt; getStrings() {
    if (strings == null) {  // check#1
      synchronized(MyClass.class) {
        if (strings == null) {
          strings = new ArrayList&lt;&gt;();  // Noncompliant
          strings.add("Hello");  //When threadA gets here, threadB can skip the synchronized block because check#1 is false
          strings.add("World");
        }
      }
    }
    return strings;
  }
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">public class MyClass {

  private volatile List&lt;String&gt; strings;

  public List&lt;String&gt; getStrings() {
    if (strings == null) {  // check#1
      synchronized(MyClass.class) {
        if (strings == null) {
          List&lt;String&gt; tmpList = new ArrayList&lt;&gt;();
          tmpList.add("Hello");
          tmpList.add("World");
          strings = tmpList;
        }
      }
    }
    return strings;
  }
}</code></pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p><a href="https://wiki.sei.cmu.edu/confluence/x/6zdGBQ">CERT, LCK10-J.</a> - Use a correct form of the double-checked locking idiom</p>
</li>
</ul>
</div>
<div class="sect2">
<h3 id="_related_rules">Related rules</h3>
<div class="ulist">
<ul>
<li>
<p><a data-rspec-id="S2168" class="rspec-auto-link">S2168</a> - Double-checked locking should not be used</p>
</li>
</ul>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Fully initialize this new "xxx" before assigning it to the field.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_15_jun_2015_180741_ann_campbell_wrote">on 15 Jun 2015, 18:07:41 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~tamas.vajk] I&#8217;ve targeted this to C# because I think it might be relevant.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_17_aug_2015_065613_tamas_vajk_wrote">on 17 Aug 2015, 06:56:13 Tamas Vajk wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] This seems very specific (to Java), I wouldn&#8217;t target C#. There might be minor differences in the Java and C# memory model when using <code>volatile</code>, which I&#8217;m not aware of.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_17_aug_2015_160828_ann_campbell_wrote">on 17 Aug 2015, 16:08:28 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>Thanks for the review [~tamas.vajk]. Un-targeted.</p>
</div>
</div>
</div>
</div>