Duplication of selectors might indicate a copy-paste mistake. The rule detects the following kinds of duplications:

* within a list of selectors in a single rule set 
* for duplicated selectors in different rule sets within a single stylesheet. 


== Noncompliant Code Example

----
.foo, .bar, .foo { ... }  /* Noncompliant */

.class1 { ... }
.class1 { ... }  /* Noncompliant */
----


== Compliant Solution

----
.foo, .bar { ... } 

.class1 { ... }
.class2 { ... }
----

