<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>The use of factory methods lets you abstract the job you need to do from the specific tool implementations needed to do it with, and helps insulate you from changes.</p>
</div>
<div class="paragraph">
<p>This rule raises an issue when instances of these are instantiated directly:</p>
</div>
<div class="ulist">
<ul>
<li>
<p><code>javax.xml.parsers.DocumentBuilder</code></p>
</li>
<li>
<p><code>javax.xml.parsers.SAXParser</code></p>
</li>
<li>
<p><code>javax.xml.transform.Transformer</code></p>
</li>
<li>
<p><code>org.xml.sax.XMLReader</code></p>
</li>
<li>
<p><code>org.xml.sax.XMLFilter</code></p>
</li>
<li>
<p><code>org.w3c.dom.*</code></p>
</li>
</ul>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">DocumentBuilder builder = new DocumentBuilderImpl();  // Noncompliant</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">DocumentBuilderFactory factory = DocumentBuilderFactory.newInstance();
DocumentBuilder builder = factory.newDocumentBuilder();</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Use a factory method to create this "xxx".</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_15_jun_2015_201410_nicolas_peru_wrote">on 15 Jun 2015, 20:14:10 Nicolas Peru wrote:</h3>
<div class="paragraph">
<p>Looks good</p>
</div>
</div>
</div>
</div>