In asynchronous programming in Kotlin with coroutines there are 2 ways of defining an async function. The first one is using modifier `suspend` in the function declaration and the second one is to create an extension function on `CoroutineScope`.

`suspend` modifier is used for the functions that might take some time to compute the result. Which means that they can potentially suspend the caller coroutine.

For the functions that return result immediately but are starting the coroutine in the background it is suggested to write an extension function for `CoroutineScope`. This means that extension functions on `CoroutineScope` are not supposed to be `suspend`.


== Noncompliant Code Example

----
suspend fun CoroutineScope.f(): Int {
    val resource1 = loadResource1()
    val resource2 = loadResource2()
    return resource1.size + resource2.size
}
----

== Compliant Solution

----
suspend fun f(): Int {
    val resource1 = loadResource1()
    val resource2 = loadResource2()
    return resource1.size + resource2.size
}

fun CoroutineScope.f(): Deferred<Int> = async {
    val resource1 = loadResource1()
    val resource2 = loadResource2()
    resource1.size + resource2.size
}
----

== See

* https://elizarov.medium.com/coroutine-context-and-scope-c8b255d59055[Coroutine Context and Scope]
