== Why is this an issue?

Offering the same experience with the mouse and the keyboard allow users to pick their preferred devices.

Additionally, users of assistive technology will also be able to browse the site even if they cannot use the mouse.

This rules detects the following issues:

 - Enforce `onClick` is accompanied by at least one of the following: `onKeyUp`, `onKeyDown`, `onKeyPress`.
 - Enforce `onmouseover`/`onmouseout` are accompanied by `onfocus`/`onblur`.

=== Code examples

==== Noncompliant code example

[source,text,diff-id=1,diff-type=noncompliant]
----
<div onClick={() => {}} />

<div onMouseOver={ () => {}} }  />
----

==== Compliant solution

[source,text,diff-id=1,diff-type=compliant]
----
<div onClick={() => {}} onKeyDown={this.handleKeyDown} />

<div onMouseOver={ () => {} } onFocus={ () => {} } />
----

== Exceptions

This does not apply for interactive or hidden elements, eg. when using `aria-hidden="true"` attribute.

== Resources

* https://www.w3.org/TR/WCAG20-TECHS/SCR2.html[SCR2: Using redundant keyboard and mouse event handlers]
* https://www.w3.org/TR/WCAG20-TECHS/G90.html[G90: Providing keyboard-triggered event handlers]