<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Python 3.10 introduced a specific syntax using the "or" operator (<code>X | Y</code>) to represent a union of types.
This syntax has the same functionality as <code>typing.Union</code>, but it is more concise and easier to read.</p>
</div>
<div class="paragraph">
<p>Using <code>typing.Union</code> is more verbose and less convenient. It can also create inconsistencies when different parts of the codebase use different syntaxes for the same type.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Replace usages of <code>typing.Union</code> with the union type syntax.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">from typing import Union

def foo(arg: Union[int, str]) -&gt; Union[int, str]:
    if isinstance(arg, int):
        return arg + 1
    else:
        return arg.upper()</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">def foo(arg: int | str) -&gt; int | str:
    if isinstance(arg, int):
        return arg + 1
    else:
        return arg.upper()</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p><a href="https://peps.python.org/pep-0604/">PEP 604 - Allow writing union types as X | Y</a></p>
</li>
<li>
<p><a href="https://docs.python.org/3/library/stdtypes.html#types-union">Python documentation on union type expressions</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Use a union type expression for this type hint.</p>
</div>
</div>
</div>
</div>