This rule raises an issue when a method annotated with `@AuraEnabled` is not declared as `static` but doesn't access any instance variables or methods.

== Why is this an issue?

Lightning components interact with Apex controllers in a stateless manner. When an `@AuraEnabled` method doesn't need access to instance variables or methods, it should be declared as `static` for better performance and to align with Lightning's architecture.

Non-static methods require the creation of an instance of the class, which adds unnecessary overhead. Static methods are called directly on the class without instantiation, making them more efficient for stateless operations like data retrieval or processing.

Additionally, static methods make the code's intent clearer - they signal that the method is a utility function that doesn't depend on object state. This improves code readability and maintainability.

=== What is the potential impact?

Using non-static methods when static ones would suffice can lead to:

* Reduced performance due to unnecessary object instantiation
* Increased memory usage
* Misleading code that suggests instance state is required when it isn't
* Inconsistency with Lightning development best practices

== How to fix it

Add the `static` keyword to the method declaration when the method doesn't access instance variables or methods.

=== Code examples

==== Noncompliant code example

[source,apex,diff-id=1,diff-type=noncompliant]
----
public with sharing class AccountController {
    @AuraEnabled(cacheable=true)
    public List<Account> getAccounts() { // Noncompliant
        return [SELECT Id, Name FROM Account LIMIT 10];
    }
}
----

==== Compliant solution

[source,apex,diff-id=1,diff-type=compliant]
----
public with sharing class AccountController {
    @AuraEnabled(cacheable=true)
    public static List<Account> getAccounts() {
        return [SELECT Id, Name FROM Account LIMIT 10];
    }
}
----

== Resources

=== Documentation

 * Lightning Web Components Developer Guide - Call Apex Methods - https://developer.salesforce.com/docs/component-library/documentation/en/lwc/lwc.apex[Official Salesforce documentation on calling Apex methods from Lightning Web Components]

 * Apex Developer Guide - AuraEnabled Annotation - https://developer.salesforce.com/docs/atlas.en-us.apexcode.meta/apexcode/apex_classes_annotation_AuraEnabled.htm[Documentation on the @AuraEnabled annotation and its usage]

=== Related rules

 * RSPEC-2325 - https://rules.sonarsource.com/java/RSPEC-2325/[Methods and fields that don't access instance data should be static (Java)]
