<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This is an issue when adding a NOT NULL column to an existing table without providing a default value, which will cause the migration to fail if the table contains existing records.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Database migrations are critical operations that must execute successfully during deployment. When you add a NOT NULL column to an existing table without specifying a default value, the database cannot satisfy the constraint for existing rows.</p>
</div>
<div class="paragraph">
<p>Here&#8217;s what happens: the database tries to add the new column to all existing records, but since no default value is provided, it would set the column to NULL for existing rows. However, the NOT NULL constraint prevents this, causing the migration to fail with an error like "Cannot add a NOT NULL column with default value NULL".</p>
</div>
<div class="paragraph">
<p>This problem becomes more severe in production environments where:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>Failed migrations can cause deployment rollbacks</p>
</li>
<li>
<p>Database inconsistencies may occur if the migration partially succeeds</p>
</li>
<li>
<p>Application downtime extends while the issue is resolved</p>
</li>
<li>
<p>Manual database intervention may be required</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>The issue affects all database systems but is particularly common with SQLite, which is stricter about this constraint compared to other databases that might allow the operation in some cases.</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="paragraph">
<p>Failed migrations can cause deployment failures, application downtime, and require manual database intervention to resolve. In production environments, this can lead to extended outages and potential data inconsistencies.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it_in_rails">How to fix it in Rails</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Add a default value when creating the NOT NULL column. This ensures existing records get a valid value that satisfies the constraint.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-ruby" data-lang="ruby">class AddDivisionIdToProfile &lt; ActiveRecord::Migration
  def change
    add_column :profiles, :division_id, :integer, null: false # Noncompliant
  end
end</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-ruby" data-lang="ruby">class AddDivisionIdToProfile &lt; ActiveRecord::Migration
  def change
    add_column :profiles, :division_id, :integer, null: false, default: 0
  end
end</code></pre>
</div>
</div>
<div class="paragraph">
<p>Use a two-step approach: first add the column without constraints, then populate it with appropriate data, and finally add the NOT NULL constraint.</p>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_code_examples_2">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example_2">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-ruby" data-lang="ruby">class AddDivisionIdToProfile &lt; ActiveRecord::Migration
  def change
    add_column :profiles, :division_id, :integer, null: false # Noncompliant
  end
end</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution_2">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-ruby" data-lang="ruby">class AddDivisionIdToProfile &lt; ActiveRecord::Migration
  def change
    add_column :profiles, :division_id, :integer
    # Populate the column with appropriate values
    Profile.update_all(division_id: 1)
    change_column_null :profiles, :division_id, false
  end
end</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Rails Migrations Guide - <a href="https://guides.rubyonrails.org/active_record_migrations.html">Official Rails documentation on database migrations and schema changes</a></p>
</li>
<li>
<p>ActiveRecord Migration Methods - <a href="https://api.rubyonrails.org/classes/ActiveRecord/Migration.html">API documentation for ActiveRecord migration methods including add_column</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>