<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>For any raw pointer passed as an argument, it is not possible to guarantee its validity. Dereferencing an invalid pointer can lead to undefined behavior, causing potential segmentation faults or other critical issues. By marking such functions as <code>unsafe</code>, it notifies the caller that they need to ensure the pointer&#8217;s validity and surround the function call with an <code>unsafe</code> block to acknowledge the potential risks involved.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-rust" data-lang="rust">pub fn foo(x: *const u8) {
    println!("{}", unsafe { *x });
}

// This call "looks" safe but will segfault or worse!
// foo(invalid_ptr);</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-rust" data-lang="rust">pub unsafe fn foo(x: *const u8) {
    println!("{}", unsafe { *x });
}

// This would cause a compiler error for calling without `unsafe`
// foo(invalid_ptr);

// Sound call if the caller knows the pointer is valid
unsafe { foo(valid_ptr); }</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Clippy Lints - <a href="https://rust-lang.github.io/rust-clippy/master/index.html#not_unsafe_ptr_arg_deref" class="bare">https://rust-lang.github.io/rust-clippy/master/index.html#not_unsafe_ptr_arg_deref</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>