<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This is an issue when <code>defined?</code> is used with hash key access (like <code>params[:key]</code> or <code>session[:id]</code>) to check if a key exists or has a meaningful value.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>The <code>defined?</code> operator checks whether an expression can be evaluated without raising an exception, not whether it returns a meaningful value.</p>
</div>
<div class="paragraph">
<p>When you access a hash key using square brackets (<code>hash[:key]</code>), Ruby always returns either the actual value or <code>nil</code> - it never raises an exception for missing keys. This means <code>defined? hash[:key]</code> will always return <code>"method"</code> (which is truthy), regardless of whether the key exists or what value it contains.</p>
</div>
<div class="paragraph">
<p>As a result, conditions using <code>defined?</code> with hash access will always evaluate to true, making your conditional logic ineffective. This can lead to unexpected behavior where code blocks execute when they shouldn&#8217;t, or where you think you&#8217;re checking for key existence but you&#8217;re actually not checking anything meaningful.</p>
</div>
<div class="paragraph">
<p>This pattern is particularly common in Rails applications when checking request parameters, session data, or other hash-like objects.</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="paragraph">
<p>The code will behave unexpectedly because conditional checks will always pass. This can lead to:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>Logic errors where code executes when it shouldn&#8217;t</p>
</li>
<li>
<p>Security issues if the checks were meant to validate required parameters</p>
</li>
<li>
<p>Difficult-to-debug problems since the condition appears correct but doesn&#8217;t work as intended</p>
</li>
</ul>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Use truthiness checks to verify that keys have meaningful (non-nil, non-false) values.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-ruby" data-lang="ruby">if (defined? params[:one]) &amp;&amp; (defined? params[:two]) # Noncompliant
  # This will always execute if params exists
  process_data
end</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-ruby" data-lang="ruby">if params[:one] &amp;&amp; params[:two]
  # Only executes if both values are truthy
  process_data
end</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it_in_rails">How to fix it in Rails</h2>
<div class="sectionbody">
<div class="paragraph">
<p>In Rails applications, use <code>present?</code> to check if parameters have meaningful values (not nil, empty, or blank).</p>
</div>
<div class="sect2">
<h3 id="_code_examples_2">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example_2">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-ruby" data-lang="ruby">if (defined? params[:name]) &amp;&amp; (defined? params[:email]) # Noncompliant
  create_user
end</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution_2">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-ruby" data-lang="ruby">if params[:name].present? &amp;&amp; params[:email].present?
  create_user
end</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Ruby defined? operator documentation - <a href="https://ruby-doc.org/core/doc/syntax/miscellaneous_rdoc.html#label-defined-3F">Official Ruby documentation explaining the defined? operator</a></p>
</li>
<li>
<p>Ruby Hash class documentation - <a href="https://ruby-doc.org/core/Hash.html">Documentation for Ruby Hash methods including has_key? and key?</a></p>
</li>
<li>
<p>Rails Object#present? documentation - <a href="https://api.rubyonrails.org/classes/Object.html#method-i-present-3F">Rails documentation for the present? method</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>