== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

With default provided regular expression: ^[A-Z][a-zA-Z0-9_]*$


[source,vb6]
----
Private Sub doSomething()
...
End Sub
----

=== Compliant solution

[source,vb6]
----
Private Sub DoSomething()
...
End Sub
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::parameters.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
