== Why is this an issue?

include::../../../shared_content/jsts/aria-intro-1.adoc[]

The `aria-activedescendant` attribute is used to enhance the accessibility of composite widgets by managing focus within them. It allows a parent element to retain active document focus while indicating which of its child elements has secondary focus. This attribute is particularly useful in interactive components like search typeahead select lists, where the user can navigate through a list of options while continuing to type in the input field.

This rule checks that DOM elements with the `aria-activedescendant` property either have an inherent tabIndex or declare one.

== How to fix it in JSX

Make sure that DOM elements with the `aria-activedescendant` property have a `tabIndex` property, or use an element with an inherent one.

=== Code examples

==== Noncompliant code example

[source,javascript,diff-id=1,diff-type=noncompliant]
----
<div aria-activedescendant={descendantId}>
  {content}
</div>
----

==== Compliant solution

[source,javascript,diff-id=1,diff-type=compliant]
----
<div aria-activedescendant={descendantId} tabIndex={0}>
  {content}
</div>
----

== Resources
=== Documentation

* https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Techniques[MDN - Using ARIA: Roles, states, and properties]
* https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles[MDN - ARIA roles (Reference)]
* https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-activedescendant[MDN - `aria-activedescendant` attribute]
* https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/tabindex[MDN - `tabIndex` attribute]

=== Standards

* https://www.w3.org/TR/wai-aria-1.2/[W3C - Accessible Rich Internet Applications (WAI-ARIA) 1.2]
* https://www.w3.org/TR/wai-aria/#composite[W3C - Composite role]
