include::../common/rule.adoc[]

== How to fix it in JSX

Make sure that DOM elements with the `aria-activedescendant` property have a `tabIndex` property, or use an element with an inherent one.

=== Code examples

==== Noncompliant code example

[source,javascript,diff-id=1,diff-type=noncompliant]
----
<div aria-activedescendant={descendantId}>
  {content}
</div>
----

==== Compliant solution

[source,javascript,diff-id=1,diff-type=compliant]
----
<div aria-activedescendant={descendantId} tabIndex={0}>
  {content}
</div>
----

== Resources
=== Documentation

* https://github.com/jsx-eslint/eslint-plugin-jsx-a11y[eslint-plugin-jsx-a11y] - Rule https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/blob/HEAD/docs/rules/aria-activedescendant-has-tabindex.md[aria-activedescendant-has-tabindex]
* MDN web docs - https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Techniques[Using ARIA: Roles, states, and properties]
* MDN web docs - https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles[ARIA roles (Reference)]
* MDN web docs - https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-activedescendant[`aria-activedescendant` attribute]
* MDN web docs - https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/tabindex[`tabIndex` attribute]

=== Standards

* W3C - https://www.w3.org/TR/wai-aria-1.2/[Accessible Rich Internet Applications (WAI-ARIA) 1.2]
* W3C - https://www.w3.org/TR/wai-aria/#composite[Composite role]
