<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>In Dart, <code>const</code> is used to declare compile-time constants. It can also be used to declare constant values, typically provided by <a href="https://dart.dev/language/constructors#constant-constructors">constant constructors</a>. This constructor, if used within <code>const</code> context will create an instance as a compile-time constant. This is an example of usage of a constant constructor:</p>
</div>
<div class="paragraph">
<p>Declaration</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-dart" data-lang="dart">class Person {
  final int age;
  final String name;

  const Person(this.age, this.name);
}</code></pre>
</div>
</div>
<div class="paragraph">
<p>Usage</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-dart" data-lang="dart">void f() {
  var p = const Person(40, 'A');
  var family = const [Person(40, 'A'), Person(39, 'B')];
}</code></pre>
</div>
</div>
<div class="paragraph">
<p>When you&#8217;re already inside the <code>const</code> context, there&#8217;s no need to repeat the keyword. So instead of writing <code>const [const Person(40, 'A'), const Person(39, 'B')]</code> you can just write <code>const [Person(40, 'A'), Person(39, 'B')]</code>.</p>
</div>
<div class="paragraph">
<p>This rule raises an issue when <code>const</code> modifier was used within another <code>const</code> context</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-dart" data-lang="dart">void f() {
  var family = const [const Person(40, 'A'), const Person(39, 'B')];
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-dart" data-lang="dart">void f() {
  var family = const [Person(40, 'A'), Person(39, 'B')];
}</code></pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p>Dart Docs - <a href="https://dart.dev/tools/linter-rules/unnecessary_const">Dart Linter rule - unnecessary_const</a></p>
</li>
<li>
<p>Dart Docs - <a href="https://dart.dev/language/constructors#constant-constructors">Constant constructors</a></p>
</li>
<li>
<p>Dart Docs - <a href="https://dart.dev/language/classes">Dart classes</a></p>
</li>
</ul>
</div>
<hr>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Unnecessary 'const' keyword.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>the const keyword</p>
</div>
</div>
</div>
</div>