<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p><code>ini_set</code> changes the value of the given configuration option for the duration of the script&#8217;s execution. While there may be a reason to do this, you should make sure that it&#8217;s a very good reason indeed, because this is the sort of "magic" change which can cause severe teeth-gnashing and hair tearing when the script needs to be debugged.</p>
</div>
<div class="paragraph">
<p>For instance, if the user explicitly turns logging on for a script, but then the script itself uses <code>ini_set('display_errors', 0);</code> to turn logging back off, it is likely that every other aspect of the environment will be examined before, in desperation, the script is read to figure out where the logging is going.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-php" data-lang="php">ini_set('display_errors', 0);  // Noncompliant</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Move this configuration into a configuration file.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_6_may_2015_115931_ann_campbell_wrote">on 6 May 2015, 11:59:31 Ann Campbell wrote:</h3>
<div class="paragraph">
<p><a href="http://www.reddit.com/r/PHPhelp/comments/2kj41j/sensio_php_configuration_should_not_be_changed/" class="bare">http://www.reddit.com/r/PHPhelp/comments/2kj41j/sensio_php_configuration_should_not_be_changed/</a></p>
</div>
</div>
<div class="sect2">
<h3 id="_on_7_may_2015_082104_linda_martin_wrote">on 7 May 2015, 08:21:04 Linda Martin wrote:</h3>
<div class="paragraph">
<p>Reviewed.</p>
</div>
</div>
</div>
</div>