The access level defaults to ``++internal++`` if left unspecified. Since that doesn't make sense for most top-level declarations, access levels should always be specified explicitly, even when ``++internal++`` is what's intended.


This rule raises an issue when the access level is not specified on any top-level declaration.


== Noncompliant Code Example

----
class Foo {  // Noncompliant
  // ...
}
----


== Compliant Solution

----
public class Foo {
  // ...
}
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
