== Why is this an issue?

In the Spring Framework, the @Import annotation is used to import definitions from a configuration class.
When multiple configurations are imported, it is cleaner to use
one `@Import` annotation with all the configurations combined into an array argument
rather than to have an individual `@Import` annotation for each configuration.

== How to fix it

Replace multiple uses of the `@Import` annotation for a class
with one `@Import` annotation that combines all the configurations to import into an array argument

=== Code examples

==== Noncompliant code example

[source,java,diff-id=1,diff-type=noncompliant]
----
@Import(AppConfig.class) // Noncompliant
@Import(DatabaseConfig.class)
public class Application {
    //...
}
----

==== Compliant solution

[source,java,diff-id=1,diff-type=compliant]
----
@Import({AppConfig.class, DatabaseConfig.class}) // Compliant
public class Application {
    //...
}
----

== Resources

=== Documentation

- https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/context/annotation/Import.html[Spring Framework - Annotation Interface Import]

=== Articles & blog posts

- https://www.javaguides.net/2019/03/spring-boot-best-practices.html[Ramesh Fadatare - Spring Boot Best Practices]
