<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>The storage of a packed numeric field is most efficient when you code an odd number of digits in the PICTURE description, so that the leftmost byte is fully used. Packed-decimal items are handled as fixed-point numbers for arithmetic purposes.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cobol" data-lang="cobol"> 01 VAL PIC 9(6) COMP-3.</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cobol" data-lang="cobol"> 01 VAL PIC 9(5) COMP-3.</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Change this field definition to be odd-length.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_4_dec_2018_211156_sam_fox_wrote">on 4 Dec 2018, 21:11:56 Sam Fox wrote:</h3>
<div class="paragraph">
<p>This rule seems to be incorrectly flagging the following:</p>
</div>
<div class="listingblock">
<div class="content">
<pre>10  BENE-GROSS-WK-AMT            PIC S9(9)V9(2) COMP-3
                                 OCCURS 010 TIMES.</pre>
</div>
</div>
<div class="paragraph">
<p>We believe this is due to "OCCURS 10 TIMES" which might inadvertently tripping the even check.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_4_dec_2018_212128_ann_campbell_wrote">on 4 Dec 2018, 21:21:28 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~SMFox] thanks for your report of this false positive and kudos on identifying the related ticket! However, this should really be reported in the <a href="https://community.sonarsource.com">SonarSource Community</a>. Would you mind opening a thread there, please?</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_4_dec_2018_213140_sam_fox_wrote">on 4 Dec 2018, 21:31:40 Sam Fox wrote:</h3>
<div class="paragraph">
<p>Created https://community.sonarsource.com/t/cobol-packed-numeric-fields-should-be-defined-with-odd-length-false-positive/4856</p>
</div>
<div class="paragraph">
<p>Thanks [~ann.campbell.2]!</p>
</div>
</div>
</div>
</div>