<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>When storing numeric fields, using an odd number of digits allows the sign to be included in the storage area without wasting any space.</p>
</div>
<div class="paragraph">
<p>For example the number -1234567, when packed as 7 digits is stored as follows.</p>
</div>
<div class="listingblock">
<div class="content">
<pre>1357
246-</pre>
</div>
</div>
<div class="paragraph">
<p>thus taking up only 4 data slots. but when packed as 8 characters it uses 5 data slots</p>
</div>
<div class="listingblock">
<div class="content">
<pre>02467
0135-</pre>
</div>
</div>
<div class="paragraph">
<p>or when 7 digits, non-packed, it will be stored as 8 data slots, including the sign.</p>
</div>
<div class="listingblock">
<div class="content">
<pre>FFFFFFF6
12345670</pre>
</div>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-rpg" data-lang="rpg">     D NUMFLD          S              8P 0
     D NUMFLD          S              7S 0</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-rpg" data-lang="rpg">     D NUMFLD          S              7P 0</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Change this field definition to be both odd-length and packed.</p>
</div>
</div>
</div>
</div>