include::../description.adoc[]

== Noncompliant Code Example

----
Public Function NumberOfMinutes(ByVal hours As Integer) As Integer
    Dim seconds As Integer = 0 ' Seconds never used
    Return hours * 60
End Function
----

== Compliant Solution

----
Public Function NumberOfMinutes(ByVal hours As Integer) As Integer
    Return hours * 60
End Function
----

== Exceptions

Unused locally created resources in a ``++Using++`` statement are not reported.

----
Using t = New TestTimer()
End Using
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
