include::../rationale.adoc[]

=== Exceptions

Unused locally created resources in a `Using` statement are not reported.

[source,vbnet]
----
Using t = New TestTimer()
End Using
----

include::../how-to-fix-it.adoc[]

=== Code examples

==== Noncompliant code example

[source,vbnet,diff-id=1,diff-type=noncompliant]
----
Public Function NumberOfMinutes(ByVal hours As Integer) As Integer
    Dim seconds As Integer = 0 ' Noncompliant - seconds is unused
    Return hours * 60
End Function
----

==== Compliant solution

[source,vbnet,diff-id=1,diff-type=compliant]
----
Public Function NumberOfMinutes(ByVal hours As Integer) As Integer
    Return hours * 60
End Function
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
