include::../description.adoc[]

== Noncompliant Code Example

----
int max(int p1, int p2, int p3) {
  return p1 > p2 ? (p1 > p3 ? p1 : p3) : (p2 > p3 ? p2 : p3); // Noncompliant
}
----

== Compliant Solution

----
int max(int p1, int p2, int p3) {
  if (p1 > p2) {
    return p1 > p3 ? p1 : p3;
  } else {
    return p2 > p3 ? p2 : p3;
  }
}
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

include::../highlighting.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
