include::../description.adoc[]

== Noncompliant Code Example

[source,cs]
----
for (i = 0; i < 10; j++) // Noncompliant
{  
  // ...
}
----

== Compliant Solution

[source,cs]
----
for (i = 0; i < 10; i++)
{
  // ...
}
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../highlighting.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
