include::../why-dotnet.adoc[]

=== Noncompliant code example

[source,vbnet,diff-id=1,diff-type=noncompliant]
----
Friend Class MyException    ' Noncompliant
    Inherits Exception

    ' ...

End Class
----

=== Compliant solution

[source,vbnet,diff-id=1,diff-type=compliant]
----
Public Class MyException
    Inherits Exception

    ' ...

End Class
----

include::../resources-dotnet.adoc[]

* https://learn.microsoft.com/en-us/dotnet/visual-basic/programming-guide/language-features/declared-elements/access-levels[Access levels]

include::../rspecator.adoc[]
