== Why is this an issue?

The point of having custom exception types is to convey more information than is available in standard types. But custom exception types must be `Public` for that to work.

If a method throws a non-public exception, the best you can do on the caller's side is to `Catch` the closest `Public` base of the class. However, you lose all the information that the new exception type carries.

=== Noncompliant code example

[source,vbnet]
----
Friend Class MyException    ' Noncompliant
    Inherits Exception

    ' ...

End Class
----


=== Compliant solution

[source,vbnet]
----
Public Class MyException    ' Noncompliant
    Inherits Exception

    ' ...

End Class
----

include::../exceptions.adoc[]

include::../see.adoc[]

include::../rspecator.adoc[]
