The point of having custom exception types is to convey more information than is available in standard types. But custom exception types must be `Public` for that to work.

If a method throws a non-public exception, the best you can do on the caller's side is to `Catch` the closest `Public` base of the class. That is, you lose all that custom information you created the exception type to pass.

== Noncompliant Code Example

[source,vbnet]
----
Friend Class MyException    ' Noncompliant
    Inherits Exception

    ' ...

End Class
----


== Compliant Solution

[source,vbnet]
----
Public Class MyException    ' Noncompliant
    Inherits Exception

    ' ...

End Class
----

include::../exceptions.adoc[]
include::../see.adoc[]
include::../rspecator.adoc[]
