== Why is this an issue?

It can be useful to use the in-code comment ``++NOSONAR++`` to suppress issues on given lines, but when those suppressions are no longer relevant they become a potential source of confusion and should be removed.


This rule raises an issue when ``++NOSONAR++`` is used on a line for which no issues would be raised during analysis.


=== Noncompliant code example

[source,text]
----
public class A {
  // This constant definition is valid and not raising any issue
  static final int LOCAL = 42;  // NOSONAR - Noncompliant 
----


=== Compliant solution

[source,text]
----
public class A {
  // This constant definition is valid and not raising any issue
  static final int LOCAL = 42;
----

