include::../description.adoc[]

== Noncompliant Code Example

----
<html>
<!-- FIXME support small device -->
...
</html>
----

include::../see.adoc[]

ifdef::env-github,rspecator-view[]
'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
