<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Using <code>&lt;algorithm&gt;</code> header non-modifying operations <code>all_of</code>, <code>none_of</code> and <code>any_of</code> allows to simplify the code and make it less error-prone.</p>
</div>
<div class="paragraph">
<p>When using versions of the standard between C&#43;&#43;11 and C&#43;&#43;17 included you should use:</p>
</div>
<div class="ulist">
<ul>
<li>
<p><code>std::all_of</code>: return <code>true</code> if all elements in the given range are matching the given predicate, <code>false</code> otherwise</p>
</li>
<li>
<p><code>std::none_of</code>: return <code>true</code> if no elements in the given range are matching the given predicate, <code>false</code> otherwise</p>
</li>
<li>
<p><code>std::any_of</code>: return <code>true</code> if at least one element in the given range is matching the given predicate, <code>false</code> otherwise</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>In C&#43;&#43;20,Â  range-based alternatives have been introduced:</p>
</div>
<div class="ulist">
<ul>
<li>
<p><code>std::ranges::all_of</code></p>
</li>
<li>
<p><code>std::ranges::none_of</code></p>
</li>
<li>
<p><code>std::ranges::any_of</code></p>
</li>
</ul>
</div>
<div class="paragraph">
<p>This rule will detect common patterns that can be replaced by these three STL algorithms.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">bool asDesired(const int v);

bool areAllDesired(std::vector&lt;int&gt; values) {
  for (int val : values) { // Noncompliant, replace it by a call to std::all_of
    if (!asDesired(val)) {
      return false;
    }
  }
  return true;
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">bool asDesired(const int v);

bool areAllDesired2(std::vector&lt;int&gt; values) {
  return std::all_of(std::begin(values), std::end(values), asDesired);
}</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_19_nov_2020_143948_abbas_sabra_wrote">on 19 Nov 2020, 14:39:48 Abbas Sabra wrote:</h3>
<div class="paragraph">
<p>Implemented as part of <a data-rspec-id="S5566" class="rspec-auto-link">S5566</a></p>
</div>
</div>
</div>
</div>