== Why is this an issue?

If an attempt is made to use an identifier in a preprocessor directive, and that identifier has not been defined, the preprocessor will assume the value zero. ``++#ifdef++``, ``++#ifndef++`` and ``++defined()++`` are provided to test the existence of a macro, and are therefore excluded.


=== Noncompliant code example

[source,cpp]
----
#if x < 0 // Noncompliant, x assumed to be zero as it is not defined
// ...
#endif
----


=== Compliant solution

[source,cpp]
----
#ifdef x
#if x < 0 // Compliant, x is guaranteed to be defined.
// ...
#endif
#endif

// ...

#define Y 12

// ...

#if Y < 12 // Compliant, y is defined.
#endif
----


== Resources

* MISRA {cpp}:2008, 16-0-7


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
