== Why is this an issue?

If an attempt is made to use an identifier in a preprocessor directive, and that identifier has not been defined, the preprocessor will assume the value zero. ``++#ifdef++``, ``++#ifndef++`` and ``++defined()++`` are provided to test the existence of a macro, and are therefore excluded.


=== Noncompliant code example

[source,cpp]
----
#if x < 0 // Noncompliant, x assumed to be zero as it is not defined
// ...
#endif
----


=== Compliant solution

[source,cpp]
----
#ifdef x
#if x < 0 // Compliant, x is guaranteed to be defined.
// ...
#endif
#endif

// ...

#define Y 12

// ...

#if Y < 12 // Compliant, y is defined.
#endif
----


== Resources

* MISRA {cpp}:2008, 16-0-7


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

"xxx" must be defined prior expansion.


'''
== Comments And Links
(visible only on this page)

=== relates to: S966

=== on 21 Oct 2014, 19:18:29 Ann Campbell wrote:
\[~samuel.mercier] please incorporate the text under the Compliant heading either into the description or the code sample in the form of comments.

=== on 21 Oct 2014, 19:21:47 Ann Campbell wrote:
\[~samuel.mercier] [~massimo.paladin] [~evgeny.mandrikov] this rule says it's okay to define preprocessing identifiers either in code or compile-time, but in code is preferred. Would a rule like this someday be feasible and/or desirable?:

Title: Preprocessing identifiers should be defined in the code

Message: "xxx" was (defined|overridden) at compile time. 

=== on 24 Oct 2014, 07:34:28 Samuel Mercier wrote:
\[~ann.campbell.2] IMO the two ways serve different purposes.

* You define a macro directly in code when you want to unconditionally enable or disable some features found in a library. A typical example is the LEAN_AND_MEAN macro on Windows.
* You define a macro on the command line to conditionally change the behaviour of the compilation (e.g. target 32bits vs 64bits, or windows vs unix-like).
Detecting if a macro was defined in code or on the command line is feasible. Understanding why it was done this way is not.

We could however write a rule to check if a macro defined on the command line is redefined in code, because in this case the former definition is lost.

endif::env-github,rspecator-view[]
