<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>The Java 8 version of <code>HashMap</code> handles key clashes by storing nodes in a binary tree when more than 11 keys clash with each other, and that tree needs to know the relative order of the keys. If you don&#8217;t provide a <code>compareTo</code> method, <code>System.identityHashCode()</code> will be used as the fallback, and that typically returns a value based on the object&#8217;s memory location, resulting in a performance degradation to <code>O(n)</code> where <code>n</code> is the number of objects at that map location.</p>
</div>
<div class="paragraph">
<p>Therefore, it&#8217;s considered a best practice to implement <code>compareTo</code> in classes that are used as <code>HashMap</code> keys.</p>
</div>
<div class="paragraph">
<p><strong>Note</strong> that this rule is automatically disabled when the project&#8217;s <code>sonar.java.source</code> is lower than <code>8</code>.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">public class Key {
  @Override
  public boolean equals(Object obj) {
    /* ... */
  }

  @Override
  public int hashCode() {
    /* ... */
  }
}

public void doTheThing() {
  Map &lt;Key,String&gt; map = new HashMap&lt;&gt;();  // Noncompliant
  // ...
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">public class Key implements Comparable{
  @Override
  public boolean equals(Object obj) {
    /* ... */
  }

  @Override
  public int hashCode() {
    /* ... */
  }

  @Override
  public int compareTo(Object o) {
    //...
  }
}

public void doTheThing() {
  Map &lt;Key,String&gt; map = new HashMap&lt;&gt;();
  // ...
}</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_3_mar_2016_172744_ann_campbell_wrote">on 3 Mar 2016, 17:27:44 Ann Campbell wrote:</h3>
<div class="paragraph">
<p><a href="http://www.javaspecialists.eu/archive/Issue235.html" class="bare">http://www.javaspecialists.eu/archive/Issue235.html</a></p>
</div>
</div>
</div>
</div>