include::../description.adoc[]


== Noncompliant Code Example

[source,php]
----
/^a|b|c$/
----


== Compliant Solution

[source,php]
----
/^(?:a|b|c)$/
----

or


[source,php]
----
/^a$|^b$|^c$/
----

or, if you do want the anchors to only apply to ``++a++`` and ``++c++`` respectively:


[source,php]
----
/(?:^a)|b|(?:c$)/
----
