include::../description.adoc[]


== Noncompliant Code Example

----
/^a|b|c$/
----


== Compliant Solution

----
/^(?:a|b|c)$/
----

or


----
/^a$|^b$|^c$/
----

or, if you do want the anchors to only apply to ``++a++`` and ``++c++`` respectively:


----
/(?:^a)|b|(?:c$)/
----
