== Why is this an issue?

Overriding ``++operator new++`` typically indicates that custom memory allocation is required for the class. When that's the case, it must be balanced with a custom memory deallocation in a matching ``++operator delete++`` method. Otherwise memory leaks or memory corruption will result.


=== Noncompliant code example

[source,cpp]
----
class AirPlane
{
public:
  void* operator new(size_t size);
  void fly();
};
----


=== Compliant solution

[source,cpp]
----
class AirPlane
{
public:
  void* operator new(size_t size);
  void operator delete(void* deadObject, size_t size);
  void fly();
};
----


== Resources

* https://wiki.sei.cmu.edu/confluence/x/KX0-BQ[CERT, DCL54-CPP.] - Overload allocation and deallocation functions as a pair in the same scope
* https://github.com/isocpp/CppCoreGuidelines/blob/036324/CppCoreGuidelines.md#r15-always-overload-matched-allocationdeallocation-pairs[{cpp} Core Guidelines R.15] - Always overload matched allocation/deallocation pairs


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Add an "operator delete" to this class.


'''
== Comments And Links
(visible only on this page)

=== on 26 May 2015, 18:26:21 Evgeny Mandrikov wrote:
\[~ann.campbell.2] word "class" looks strange for me after "operator delete" in description. Is it a typo or just bad knowledge of english by me?

=== on 27 May 2015, 14:07:28 Ann Campbell wrote:
It's not your English [~evgeny.mandrikov], it's mine. ;)

Check it now.

=== on 27 May 2015, 14:44:46 Evgeny Mandrikov wrote:
LGTM.

endif::env-github,rspecator-view[]
