Taking the address of an object of incomplete type, where the complete type contains a user declared ``++operator &++`` leads to undefined behavior.


== Noncompliant Code Example

----
// A.h
class A
{
public:
  A * operator & ( ); // Noncompliant
};

// f1.cc
class A;
void f ( A & a )
{
  &a; // uses built-in operator &
}

// f2.cc
#include "A.h"
void f2 ( A & a )
{
  &a; // use user-defined operator &
}
----


== See

* MISRA {cpp} 2008, 5-3-3 - The unary & operator shall not be overloaded. 
* https://github.com/isocpp/CppCoreGuidelines/blob/036324/CppCoreGuidelines.md#Ro-address-of[{cpp} Core Guidelines C.166] - Overload unary ``++&++`` only as part of a system of smart pointers and references



ifdef::env-github,rspecator-view[]
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
