=== Code examples

==== Noncompliant code example

[source,csharp,diff-id=1,diff-type=noncompliant]
----
int GetMax(SortedSet<int> data) =>
    data.Max();
----

[source,csharp,diff-id=2,diff-type=noncompliant]
----
int GetMin(SortedSet<int> data) =>
    data.Min();
----

==== Compliant solution

[source,csharp,diff-id=1,diff-type=compliant]
----
int GetMax(SortedSet<int> data) =>
    data.Max;
----

[source,csharp,diff-id=2,diff-type=compliant]
----
int GetMin(SortedSet<int> data) =>
    data.Min;
----
