== Why is this an issue?

In Dart we use constructors to provide an initialization logic for a new object creation. Sometimes we don't need any additional logic, so th constructor is left empty. In such cases the constructor body can be simply removed and the declaration must be terminated with a semicolon.

=== Noncompliant code example

[source,dart,diff-id=1,diff-type=noncompliant]
----
class Person {

  String name;
  int age;

  Person(this.name, this.age) {}
}
----


=== Compliant solution

[source,dart,diff-id=1,diff-type=compliant]
----
class Person {

  String name;
  int age;

  Person(this.name, this.age);
}
----


== Resources

* https://dart.dev/tools/linter-rules/empty_constructor_bodies[Dart Lint rule]
