== How to fix it in Django

=== Code examples

==== Noncompliant code example

The following noncompliant Django REST framework application contains a hard-coded secret that can be exposed unintentionally.

[source,python,diff-id=401,diff-type=noncompliant]
----
INSTALLED_APPS = [
    'django.contrib.admin',
    'django.contrib.auth',
    'django.contrib.contenttypes',
    'django.contrib.sessions',
    'django.contrib.messages',
    'django.contrib.staticfiles',
    'rest_framework',
]

REST_FRAMEWORK = {
    'DEFAULT_PERMISSION_CLASSES': [
        'rest_framework.permissions.DjangoModelPermissionsOrAnonReadOnly',
    ],
    'DEFAULT_AUTHENTICATION_CLASSES': (
      'rest_framework_simplejwt.authentication.JWTAuthentication',
    )
}

SIMPLE_JWT = {
  "ALGORITHM": "HS256",
  "SIGNING_KEY": "secret"  # Noncompliant
}
----

==== Compliant solution

A solution is to set this secret in an environment variable.

[source,python,diff-id=401,diff-type=compliant]
----
INSTALLED_APPS = [
    'django.contrib.admin',
    'django.contrib.auth',
    'django.contrib.contenttypes',
    'django.contrib.sessions',
    'django.contrib.messages',
    'django.contrib.staticfiles',
    'rest_framework',
]

REST_FRAMEWORK = {
    'DEFAULT_PERMISSION_CLASSES': [
        'rest_framework.permissions.DjangoModelPermissionsOrAnonReadOnly',
    ],
    'DEFAULT_AUTHENTICATION_CLASSES': (
      'rest_framework_simplejwt.authentication.JWTAuthentication',
    )
}

SIMPLE_JWT = {
  "ALGORITHM": "HS256",
  "SIGNING_KEY": os.environ["SIGNING_KEY"]
}
----


//=== How does this work?

//=== Pitfalls

=== Going the extra mile

include::../../../../shared_content/secrets/fix/vault.adoc[]

