== How to fix it in Flask

include::revoke.adoc[]

=== Code examples

==== Noncompliant code example

The following noncompliant code contains a hard-coded secret that can be exposed unintentionally.

[source,python,diff-id=101,diff-type=noncompliant]
----
from flask import Flask

app = Flask(__name__)
app.config['JWT_SECRET_KEY'] = secret_key  # Noncompliant
----

==== Compliant solution

A solution is to set this secret in an environment string.

[source,python,diff-id=101,diff-type=compliant]
----
from flask import Flask
import os

app = Flask(__name__)
app.config['JWT_SECRET_KEY'] = os.environ["JWT_SECRET_KEY"]
----


//=== How does this work?

//=== Pitfalls

=== Going the extra mile

include::../../../../shared_content/secrets/fix/vault.adoc[]

