== How to fix it in Hibernate

=== Code examples

include::../../common/fix/code-rationale.adoc[]

==== Noncompliant code example

[source,java,diff-id=1,diff-type=noncompliant]
----
public void search(Session session, String user) {
  Query query = session.createQuery("FROM users where id = " + user); // Noncompliant
}
----

==== Compliant solution

[source,java,diff-id=1,diff-type=compliant]
----
public void search(Session session, String user) {
  Query query = session.createQuery("FROM users where id = ?");
  query = query.setParameter(0, user);
}
----

=== How does this work?

include::../../common/fix/how.adoc[]
