== How to fix it in Java SE

=== Code examples

include::../../common/fix/code-rationale.adoc[]

==== Noncompliant code example

[source,java,diff-id=11,diff-type=noncompliant]
----
public void search(Connection con, String user) throws SQLException {
    Statement stmt = con.createStatement();
    ResultSet rs = stmt.executeQuery("SELECT ssn FROM users WHERE id=" + user); // Noncompliant
}
----

==== Compliant solution

[source,java,diff-id=11,diff-type=compliant]
----
public void search(Connection con, String user) throws SQLException {
    PreparedStatement stmt = con.prepareStatement("SELECT ssn FROM users WHERE id=?");
    stmt.setString(1, user);
    ResultSet rs = stmt.executeQuery();
}
----

=== How does this work?

include::../../common/fix/how.adoc[]
