A results indicator should be used on a ``++CHAIN++`` statement because it leads to cleaner, more readable, and more correct code. Using an indicator results in the conditions ``++record not found++``, and ``++end of file++`` being automatically handled correctly, without the need for further error handling. 


== Noncompliant Code Example

----
C     K0GMCP        CHAIN     SSGMCPR
----


== Compliant Solution

----
C     K0GMCP        CHAIN     SSGMCPR                            81 
----


== Exceptions

Free-form ``++CHAIN++`` statements cannot have indicators, and are ignored.

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

endif::env-github,rspecator-view[]
