<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This is an issue when calling C functions through cgo and checking the errno-based error without first verifying that the function call actually failed.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>When calling C functions through Go&#8217;s cgo, the errno value may be non-zero even when the function call succeeds. This behavior differs from typical Go error handling conventions where a non-nil error always indicates failure.</p>
</div>
<div class="paragraph">
<p>In cgo, C functions return both a result value and an errno-based error. The errno value reflects the state from the last system call that set it, which might not be related to your current function call. Some C functions may internally make system calls that set errno but still succeed overall.</p>
</div>
<div class="paragraph">
<p>If you check the error first without verifying the return value, you might incorrectly treat successful operations as failures. This can lead to:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>False error reporting in your application</p>
</li>
<li>
<p>Unnecessary error handling that interrupts normal program flow</p>
</li>
<li>
<p>Confusion when debugging, as successful operations appear to fail</p>
</li>
<li>
<p>Inconsistent behavior that&#8217;s difficult to reproduce</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>The correct pattern is to first check the return value to determine if the C function succeeded, and only then examine the errno-based error if the function actually failed.</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="paragraph">
<p>Following incorrect error handling patterns can cause applications to report false failures, interrupt normal operation flow, and create confusing debugging scenarios where successful C function calls appear to have failed.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Check the return value of the C function first to determine if it succeeded. Only examine the errno-based error if the return value indicates failure.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-go" data-lang="go">n, err := C.setenv(key, value, 1)
if err != nil { // Noncompliant
    return err
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-go" data-lang="go">n, err := C.setenv(key, value, 1)
if n != 0 {
    // we know the call failed, so it is now valid to use err
    return err
}</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Go cgo documentation - <a href="https://pkg.go.dev/cmd/cgo">Official Go documentation for cgo, including error handling patterns</a></p>
</li>
<li>
<p>Effective Go - Error handling - <a href="https://go.dev/doc/effective_go#errors">Go&#8217;s general error handling conventions</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>