== Why is this an issue?

It is possible to use ``++await++`` on values which are not ``++Promise++``s, but it's useless and misleading. The point of ``++await++`` is to pause execution until the ``++Promise++``'s asynchronous code has run to completion. With anything other than a ``++Promise++``, there's nothing to wait for.


This rule raises an issue when an ``++await++``ed value is guaranteed not to be a ``++Promise++``.


=== Noncompliant code example

[source,javascript]
----
let x = 42;
await x; // Noncompliant
----


=== Compliant solution

[source,javascript]
----
let x = new Promise(resolve => resolve(42));
await x;

let y = p ? 42 : new Promise(resolve => resolve(42));
await y;
----



ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Refactor this redundant 'await' on a non-promise.


'''
== Comments And Links
(visible only on this page)

=== on 26 Jul 2017, 18:02:55 Ann Campbell wrote:
\[~carlo.bottiglieri] this was a bit bare-bones. I've fluffed it up some. Please double-check me.

=== on 26 Jul 2017, 18:26:10 Carlo Bottiglieri wrote:
\[~ann.campbell.2] I made a small change, for the rest it's fine for me, but the rendering of the code highlight is mostly broken in my browser (I checked and the syntax looks correct)

=== on 26 Jul 2017, 19:29:11 Ann Campbell wrote:
Yes [~carlo.bottiglieri], Jira's handling of its own markdown is broken. It will come out fine when RuleAPI generates the description tho.

=== on 27 Jul 2017, 13:42:42 Freddy Mallet wrote:
\[~carlo.bottiglieri] and [~ann.campbell.2], let's go for the creation of the implementation tickets for JS and TS ?

=== on 9 Aug 2017, 18:09:57 Elena Vilchik wrote:
I removed from default profile. While we can add it back if make rule configurable (list of promise-like classes)

endif::env-github,rspecator-view[]
