<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when semicolons are used to terminate statements in Go code where they are not required by the language specification.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Go has automatic semicolon insertion rules that make explicit semicolons unnecessary in most cases. The Go language specification defines that semicolons are automatically inserted after a line&#8217;s final token if that token is an identifier, literal, or one of the keywords <code>break</code>, <code>continue</code>, <code>fallthrough</code>, or <code>return</code>, or one of the operators <code>++</code>, <code>--</code>, <code>)</code>, <code>]</code>, or <code>}</code>.</p>
</div>
<div class="paragraph">
<p>Using unnecessary semicolons goes against Go&#8217;s design philosophy of clean, readable code. It can make code look cluttered and may confuse developers who are learning Go&#8217;s syntax rules. The automatic semicolon insertion mechanism is a deliberate language feature that allows Go code to be more concise and readable.</p>
</div>
<div class="paragraph">
<p>Additionally, unnecessary semicolons can indicate that a developer is applying conventions from other programming languages (like C, Java, or JavaScript) rather than following Go&#8217;s idiomatic style. This can lead to inconsistent code style within a Go project.</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="paragraph">
<p>While unnecessary semicolons do not cause runtime errors or compilation failures, they impact code quality and maintainability:</p>
</div>
<div class="ulist">
<ul>
<li>
<p><strong>Readability</strong>: Code becomes more cluttered and less idiomatic</p>
</li>
<li>
<p><strong>Consistency</strong>: Mixed usage of semicolons creates inconsistent code style</p>
</li>
<li>
<p><strong>Learning curve</strong>: New Go developers may be confused about when semicolons are actually required</p>
</li>
<li>
<p><strong>Code reviews</strong>: Unnecessary semicolons can distract from more important issues during code reviews</p>
</li>
</ul>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Remove unnecessary semicolons from the end of statements. Go&#8217;s automatic semicolon insertion will handle statement termination.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-go" data-lang="go">package main

import "fmt";

func main() {
    x := 42;
    fmt.Println(x);
    return;
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-go" data-lang="go">package main

import "fmt"

func main() {
    x := 42
    fmt.Println(x)
    return
}</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Go Language Specification - Semicolons - <a href="https://go.dev/ref/spec#Semicolons">Official Go specification explaining automatic semicolon insertion rules</a></p>
</li>
<li>
<p>Effective Go - Semicolons - <a href="https://go.dev/doc/effective_go#semicolons">Best practices for semicolon usage in Go code</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>