This rule raises an issue when converting a `Date` object to a UNIX timestamp using `to_time.to_i`, which creates timezone-dependent behavior.

== Why is this an issue?

Converting `Date` objects to UNIX timestamps using `date.to_time.to_i` can produce inconsistent results across different systems and environments.

When you call `to_time` on a `Date` object, Ruby creates a `Time` object using the system's local timezone. This means the same date will produce different timestamps depending on where your code runs. For example, December 25th, 2023 at midnight will have different UNIX timestamps in New York versus Tokyo.

This timezone dependency can cause several problems:

* **Inconsistent behavior**: The same code produces different results on different servers or developer machines
* **Data corruption**: Timestamps may be off by several hours, leading to incorrect time-based calculations
* **Hard-to-debug issues**: Problems may only appear when code runs in production environments with different timezone settings

UNIX timestamps are meant to represent absolute points in time, independent of timezone. When timezone conversion happens unexpectedly, it defeats this purpose and can lead to subtle but serious bugs in time-sensitive applications.

=== What is the potential impact?

Applications may store incorrect timestamps, leading to data inconsistency, failed time-based operations, or incorrect business logic execution. In distributed systems, this can cause synchronization issues between services running in different timezones.

== How to fix it

Use `Time.utc()` with explicit date components to create a UTC timestamp. This approach is explicit about timezone handling and produces consistent results.

=== Code examples

==== Noncompliant code example

[source,ruby,diff-id=1,diff-type=noncompliant]
----
date = Date.new(2023, 12, 25)
timestamp = date.to_time.to_i  # Noncompliant: uses local timezone
----

==== Compliant solution

[source,ruby,diff-id=1,diff-type=compliant]
----
date = Date.new(2023, 12, 25)
timestamp = Time.utc(date.year, date.month, date.day).to_i
----

== Resources

=== Documentation

 * Ruby Date Class Documentation - https://ruby-doc.org/stdlib-3.0.0/libdoc/date/rdoc/Date.html[Official Ruby documentation for the Date class and its methods]

 * Ruby Time Class Documentation - https://ruby-doc.org/core-3.0.0/Time.html[Official Ruby documentation for the Time class and timezone handling]

 * Working with Time Zones in Ruby - https://thoughtbot.com/blog/its-about-time-zones[Comprehensive guide to handling timezones correctly in Ruby applications]
