== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,php]
----
if ($booleanVariable == true) { /* ... */ }      
if ($booleanVariable != true) { /* ... */ }   
if ($booleanVariable || false) { /* ... */ }     
doSomething(!false);
----

=== Compliant solution

[source,php]
----
if ($booleanVariable) { /* ... */ }              
if (!$booleanVariable) { /* ... */ }    
if ($booleanVariable) { /* ... */ }       
doSomething(true);
----

=== Exceptions

The use of literal booleans in comparisons which use identity operators (``++===++`` and ``++!==++``) are ignored.

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

=== on 4 May 2020, 13:31:08 Pierre-Yves Nicolas wrote:
In PHP, this rule should not apply to the operands of ternary operators, see SONARPHP-906.

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
