include::../description.adoc[]

== Noncompliant Code Example

With the default regular expression ``++^[A-Z][A-Z0-9]*(_[A-Z0-9]+)*$++``:

----
define("const1", true);

class Foo {
    const const2 = "bar";
}
----

== Compliant Solution

----
define("CONST1", true);

class Foo {
    const CONST2 = "bar";
}
----

ifdef::env-github,rspecator-view[]
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
