Declaring multiple variables on one line is difficult to read.

== Noncompliant Code Example

----
var i = 1, j = 2
----

== Compliant Solution

----
var i = 1
var j = 2
----

include::../see.adoc[]

ifdef::env-github,rspecator-view[]
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
