== Why is this an issue?

Declaring multiple variables on one line is difficult to read.

=== Noncompliant code example

[source,swift]
----
var i = 1, j = 2
----

=== Compliant solution

[source,swift]
----
var i = 1
var j = 2
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
