Declaring multiple variables on one line is difficult to read.

== Noncompliant Code Example

[source,swift]
----
var i = 1, j = 2
----

== Compliant Solution

[source,swift]
----
var i = 1
var j = 2
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
