== Why is this an issue?

When https://learn.microsoft.com/en-us/dotnet/visual-basic/language-reference/statements/select-case-statement[Select Case] statements have large sets of `Case` clauses, it is usually an attempt to map two sets of data. 

When the `Case` clauses of such `Select Case` statement contain multiple lines, the code becomes hard to read and maintain. 

For example, the https://www.sonarsource.com/docs/CognitiveComplexity.pdf[Cognitive Complexity] is going to be particularly high.

In such scenarios, it's better to rethink the approach to data retrieval, possibly making use of a collection such as a https://learn.microsoft.com/en-us/dotnet/api/system.collections.generic.dictionary-2[Dictionary]. 

On the other hand, when all the `Case` clauses of a `Select Case` statement are single-line, the readability of the code is not necessarily affected. At times, a `Select Case` can be easier to maintain than a collection, especially when data is not statically mapped, as it happens in a dictionary, but needs to be dynamically calculated, for example calling a method.

=== Exceptions

This rule ignores:

* `Select Case` statements over `Enum` arguments
* fall-through cases

== How to fix it

Multiple approaches can be taken to fix this issue.

When data is statically mapped, a possible approach is to store all the cases apart from the `Case Else` one in a dictionary and try to get the matching value by calling the https://learn.microsoft.com/en-us/dotnet/api/system.collections.generic.dictionary-2.trygetvalue[TryGetValue] method.

When data is dynamically calculated and cannot be easily stored in a collection, an alternative approach is to extract the logic of multi-line `Select Case` clauses into separate methods.

=== Code examples

The examples below use the "Maximum number of case" property set to `4`.

==== Noncompliant code example

[source,vbnet,diff-id=1,diff-type=noncompliant]
----
Public Function MapChar(ch As Char, value As Integer) As Integer ' Noncompliant
    Select Case ch
        Case "a"c
            Return 1
        Case "b"c
            Return 2
        Case "c"c
            Return 3
        ' ...
        Case "-"c
            If value > 10 Then
                Return 42
            ElseIf value < 5 AndAlso value > 1 Then
                Return 21
            End If
            Return 99
        Case Else
            Return 1000
    End Select
End Function
----

==== Compliant solution

[source,vbnet,diff-id=1,diff-type=compliant]
----
Public Function MapChar(ch As Char, value As Integer) As Integer
    Select Case ch
        Case "a"c
            Return 1
        Case "b"c
            Return 2
        Case "c"c
            Return 3
        ' ...
        Case "-"c
            Return HandleDash(value)
        Case Else
            Return 1000
    End Select
End Function

Private Function HandleDash(value As Integer) As Integer
    Select Case value
        Case > 10
            Return 42
        Case > 1, < 5
            Return 21
        Case Else
            Return 99
    End Select
End Function
----

include::../resources-dotnet.adoc[]

* https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/statements/selection-statements#the-switch-statement[The `switch` statement]

include::../rspecator-dotnet.adoc[]

