== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,csharp]
----
[ServiceContract]
interface IMyService 
{
  [OperationContract(IsOneWay = true)]
  int SomethingHappened(int parameter); // Noncompliant
}
----


=== Compliant solution

[source,csharp]
----
[ServiceContract]
interface IMyService 
{
  [OperationContract(IsOneWay = true)]
  void SomethingHappened(int parameter);
}
----


include::../exceptions.adoc[]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../highlighting.adoc[]

endif::env-github,rspecator-view[]
