<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>In an enumeration, it is possible to have several enumerators with the same value. This can be useful in many circumstances. But if that happens without an explicit choice from the developer, it can also lead to errors that are hard to diagnose.</p>
</div>
<div class="paragraph">
<p>When an enumerator has no explicit value, its value will be one more than the preceding enumerator. When another enumerator uses the same value, there are three possibilities:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>The other enumerator also has an implicit value. This rule raises an issue since this collision involves no explicit decision.</p>
</li>
<li>
<p>The other enumerator has a value directly defined as equal to an enumerator of this enum. In that case, this rule considers the collision to be a conscious decision and does not raise an issue.</p>
</li>
<li>
<p>The other enumerator has an explicit value defined in another way. In that case, the collision might be accidental, and this rule raises an issue.</p>
</li>
</ul>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">enum color { red = 3, blue, green, yellow = 5 }; // Noncompliant, both green and yellow = 5</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">enum color { red = 3, blue, green, yellow }; // Compliant, different values
enum color { red = 3, blue, green = 5, yellow = 5 }; // Compliant, all identical values are non-implicit
enum color { red = 3, blue, green, yellow = green }; // Compliant, yellow is defined as identical to green</code></pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p>MISRA C&#43;&#43;:2023, 11.6.3 - Within an enumerator list, the value of an implicitly-specified
enumeration constant shall be unique</p>
</li>
<li>
<p>MISRA C:2004, 9.3 - In an enumerator list, the "=" construct shall not be used to explicitly initialize members other than the first, unless all items are explicitly initialized.</p>
</li>
<li>
<p>MISRA C&#43;&#43;:2008, 8-5-3 - In an enumerator list, the = construct shall not be used to explicitly initialize members other than the first, unless all items are explicitly initialized.</p>
</li>
</ul>
</div>
</div>
</div>