<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This is an issue when <code>update</code>, <code>update_attributes</code>, or <code>save</code> methods are called within ActiveRecord save callbacks like <code>after_save</code>, <code>before_save</code>, <code>after_create</code>, or <code>before_create</code>.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Using update methods within save callbacks creates infinite recursion. When you call <code>update</code>, <code>update_attributes</code>, or <code>save</code> inside a callback like <code>after_save</code>, these methods trigger the same callback again, which calls the update method again, and so on.</p>
</div>
<div class="paragraph">
<p>This happens because ActiveRecord&#8217;s update methods go through the full model lifecycle, including running all callbacks. So when your <code>after_save</code> callback calls <code>update</code>, it saves the record again, which triggers <code>after_save</code> again, creating an endless loop.</p>
</div>
<div class="paragraph">
<p>The result is a stack overflow error that crashes your application. Even if the recursion doesn&#8217;t immediately crash the app, it creates unnecessary database queries and severely impacts performance.</p>
</div>
<div class="paragraph">
<p>ActiveRecord provides callbacks specifically to modify data during the save process. The correct approach is to modify attributes directly within these callbacks, which updates the database without triggering additional callbacks.</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="paragraph">
<p>This issue causes stack overflow errors that crash the application. It can also lead to severe performance degradation due to excessive database queries and infinite callback execution. In production environments, this can result in application downtime and poor user experience.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it_in_ruby_on_rails">How to fix it in Ruby on Rails</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Replace update methods with direct attribute assignment in save callbacks. This modifies the attribute without triggering additional callbacks.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-ruby" data-lang="ruby">class User &lt; ActiveRecord::Base
  after_save :calculate_rating

  private

  def calculate_rating
    # These methods trigger callbacks and cause infinite recursion
    self.update_attributes(rating: rating_score / ratings) # Noncompliant
    # or
    self.update(rating: rating_score / ratings) # Noncompliant
    # or
    self.save # Noncompliant
  end
end</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-ruby" data-lang="ruby">class User &lt; ActiveRecord::Base
  after_save :calculate_rating

  private

  def calculate_rating
    # Direct attribute assignment - safe in callbacks
    self.rating = rating_score / ratings if ratings &gt; 0
  end
end</code></pre>
</div>
</div>
<div class="paragraph">
<p>For more complex updates, use <code>update_column</code> or <code>update_columns</code> which skip callbacks and validations, or move the logic to a <code>before_save</code> callback.</p>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_code_examples_2">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example_2">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-ruby" data-lang="ruby">class Product &lt; ActiveRecord::Base
  after_save :update_inventory

  private

  def update_inventory
    self.update(stock_status: calculate_stock_status) # Noncompliant
  end
end</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution_2">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-ruby" data-lang="ruby">class Product &lt; ActiveRecord::Base
  after_save :update_inventory

  private

  def update_inventory
    # Skip callbacks with update_column (use sparingly)
    self.update_column(:stock_status, calculate_stock_status)
    # Or better: move to before_save
  end
end</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>ActiveRecord Callbacks Guide - <a href="https://guides.rubyonrails.org/active_record_callbacks.html">Official Rails guide explaining ActiveRecord callbacks and their lifecycle</a></p>
</li>
<li>
<p>ActiveRecord update_column documentation - <a href="https://api.rubyonrails.org/classes/ActiveRecord/Persistence.html#method-i-update_column">Documentation for update_column method that skips callbacks</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>