== Why is this an issue?

While working with bitwise operators ``++&++`` or ``++|++`` it is easy to make a typo andÂ write the equivalent logical operators ``++&&++`` or ``++||++``. This rule is raising issues when the right operand of a logical expression  ``++&&++`` or ``++||++`` is a constant of integral type, as the developer probably meant to use the corresponding bitwise operator ``++&++`` or ``++|++``.


=== Noncompliant code example

[source,cpp]
----
int fun(int a) {
  return a || 4; // Noncompliant, did you mean to use bitwise operator '|'?
}
----


=== Compliant solution

[source,cpp]
----
int fun(int a) {
 return a | 4;
}
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Review this logical (&& | ||) expression with constant operand.


endif::env-github,rspecator-view[]
