{cpp}11 improves the specification of the parser so that at the end of a template, multiple ``++>++``s will be interpreted as closing the template argument list rather than as the right shift operator or stream extraction operator. It is no longer required to place a space between the two ``++>++``s. 


This rule raises an issue when such useless spaces exist.


== Noncompliant Code Example

----
vector<pair<int,int> > s; // Noncompliant
----


== Compliant Solution

----
vector<pair<int,int>> s;
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

include::highlighting.adoc[]

'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
