== Why is this an issue?

{cpp}11 improves the specification of the parser so that at the end of a template, multiple ``++>++``s will be interpreted as closing the template argument list rather than as the right shift operator or stream extraction operator. It is no longer required to place a space between the two ``++>++``s. 


This rule raises an issue when such useless spaces exist.


=== Noncompliant code example

[source,cpp]
----
vector<pair<int,int> > s; // Noncompliant
----


=== Compliant solution

[source,cpp]
----
vector<pair<int,int>> s;
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Remove the space between the two right angle brackets.


=== Highlighting

space(s)


'''
== Comments And Links
(visible only on this page)

=== on 2 Mar 2016, 17:34:43 Ann Campbell wrote:
Double-check my changes please [~alban.auzeill]

endif::env-github,rspecator-view[]
