== How to fix it in Realm

=== Code examples

In the example below, a local database is opened using a hardcoded key. To fix this, the key is moved to a secure location instead and retrieved using a `getKey()` method.

==== Noncompliant code example

[source,java,diff-id=201,diff-type=noncompliant]
----
String key = "gb09ym9ydoolp3w886d0tciczj6ve9kszqd65u7d126040gwy86xqimjpuuc788g";
RealmConfiguration config = new RealmConfiguration.Builder();
    .encryptionKey(key.toByteArray()) // Noncompliant
    .build();
Realm realm = Realm.getInstance(config);
----

==== Compliant solution

[source,java,diff-id=201,diff-type=compliant]
----
RealmConfiguration config = new RealmConfiguration.Builder()
    .encryptionKey(getKey())
    .build();
Realm realm = Realm.getInstance(config);
----

=== How does this work?

include::../../common/fix/android-apis.adoc[]

include::../../common/fix/dynamic-retrieval.adoc[]

=== Going the extra mile

include::../../common/extra-mile/avoid-local-data.adoc[]
