include::../rule.adoc[]

== Noncompliant Code Example

[source,java]
----
internal class MyException : Exception   // Noncompliant
{
  // ...
}
----


== Compliant Solution

[source,java]
----
public class MyException : Exception
{
  // ...
}
----

include::../see.adoc[]
include::../rspecator.adoc[]