== Why is this an issue?

Prototypes should be used for program calls because their use allows compile-time checking of the number of parameters and the type and size of those parameters. Without the use of a prototype, that checking happens only at run-time. 


A secondary benefit of the use of prototypes and ``++CALLP++``, is that prototypes allow the caller to gain a clearer understanding of the data expected for each parameter because parameter names are purely descriptive.


=== Noncompliant code example

[source,rpg]
----
----


=== Compliant solution

[source,rpg]
----
----


ifdef::env-github,rspecator-view[]
'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
