== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

With the default comment pattern ``++^'\s*\S+\s*$++``, which ignores single word comments:

[source,vbnet]
----
Module Module1
  Sub Main()
    Console.WriteLine("Hello, world!") ' Noncompliant - My first program!
    Console.WriteLine("Hello, world!") ' CompliantOneWord
  End Sub
End Module
----

=== Compliant solution

[source,vbnet]
----
Module Module1
  Sub Main()
    ' Compliant - My first program!
    Console.WriteLine("Hello, world!")
    Console.WriteLine("Hello, world!") ' CompliantOneWord
  End Sub
End Module
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

=== Parameters

.legalCommentPattern
****

----
^'\s*\S+\s*$
----

Pattern for text of trailing comments that are allowed.
****


'''
== Comments And Links
(visible only on this page)

=== on 7 Sep 2016, 17:55:47 Ann Campbell wrote:
\[~tamas.vajk] IMO we need to remove this exception (and possibly rename the parameter for clarity?)

=== on 9 Sep 2016, 08:13:09 Tamas Vajk wrote:
\[~ann.campbell.2] All the other languages have this exception, and the parameter name is also similar there. Why do you think we'd need to fix these issues?

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
