== Why is this an issue?

There is a real, functional difference between a function with an empty parameter list and one with an explicitly ``++void++`` parameter list: It is possible to pass parameters to a function with an empty list; the compiler won't complain. That is not the case for a function with a ``++void++`` list. Thus, it is possible, and even easy to invoke empty-list functions incorrectly without knowing it, and thereby introduce the kind of subtle bug that can be very difficult to track down.


=== Noncompliant code example

[source,cpp]
----
void myfunc ();  // Noncompliant

//...

void otherFunc() {
  int a = 4;
  //...
  myfunc(a); // Compiler allows this
}
----


=== Compliant solution

[source,cpp]
----
void myfunc ( void );

//...

void otherFunc() {
  int a = 4;
  //...
  myfunc(a); // Compiler error!
}
----


== Resources

* MISRA C:2004, 16.5 - Functions with no parameters shall be declared with parameter type void
* https://wiki.sei.cmu.edu/confluence/x/t9YxBQ[CERT, DCL20-C.] - Explicitly specify void when a function accepts no arguments


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Set this function's parameter list to "void"


'''
== Comments And Links
(visible only on this page)

=== relates to: S1270

=== on 2 Sep 2014, 16:20:32 Ann Campbell wrote:
from [~evgeny.mandrikov]:


I don't like current description, because this rule is not just about little styling. Compare to CERT, which for example contains

____
Failure to declare a void parameter will result in

* An ambiguous functional interface between the caller and callee.
* Sensitive information outflow.
____
Also as a developer I really appreciate references to the language specification in RSPECs.

=== on 19 Sep 2014, 13:08:35 Freddy Mallet wrote:
@Ann, I would also associate this rule to the SQALE characteristic "Reliability". Indeed this rule can definitely be associated to "Maintainability" or to "Reliability". But when there is a doubt "Reliability" should be preferred because operational risks matter more than maintainability issue.

endif::env-github,rspecator-view[]
