<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raise an issue when one of the mandatory label are missing.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Adding labels to your image help to organize images by project, record licensing information, aid in automation and for other reasons.<br>
The rule provide the possibility to configure the list of mandatory label that must be present in every Dockerfile.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="paragraph">
<p>With <code>labels</code> set to <code>maintainer,description,version</code></p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-docker" data-lang="docker">FROM ubuntu:22.02
RUN my_command</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-docker" data-lang="docker">FROM ubuntu:22.02
LABEL maintainer="John Doe"
LABEL description="Image doing XYZ"
LABEL version=1.0
RUN my_command</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Docker docs - <a href="https://docs.docker.com/build/building/best-practices/#label">Building best practices - LABEL</a></p>
</li>
</ul>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Single label is missing:
Add the missing label <code>%s</code>.
Multiple labels are missing:
Add the missing labels; <code>%s</code>, <code>%s</code> and <code>%s</code>.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>Highlight the first FROM instruction.</p>
</div>
<hr>
<div class="sidebarblock">
<div class="content">
<div class="title">labels</div>
<div class="listingblock">
<div class="content">
<pre>maintainer</pre>
</div>
</div>
<div class="paragraph">
<p>A comma separated list of labels that are required for each dockerfile.</p>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
</div>
</div>