<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>After the execution of each SQL statement (other than <code>DECLARE CURSOR</code>, <code>DECLARE TABLE</code> and <code>DECLARE VARIABLE</code>), the value of <code>SQLCODE</code> or <code>SQLSTATE</code> should be checked before proceeding. A 0 <code>SQLCODE</code> value means the statement succeeded, a positive value means success with a warning, and a negative value indicates an error. Proceeding without checking could put your program or your data in a bad state.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-text" data-lang="text">EXEC SQL
  SELECT name INTO :username FROM user WHERE id = :userid
END-EXEC.

DISPLAY username.    *&gt; Noncompliant</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-text" data-lang="text">EXEC SQL
  SELECT name INTO :username FROM user WHERE id = :userid
END-EXEC.

IF SQLCODE = 0 THEN
  DISPLAY username
END-IF.</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_exceptions">Exceptions</h3>
<div class="paragraph">
<p>When the value of <code>SQLCODE</code> or <code>SQLSTATE</code> is not checked but transferred to another variable for later use, no issue is raised.</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-text" data-lang="text">EXEC SQL
  SELECT name INTO :username FROM user WHERE id = :userid
END-EXEC.
MOVE SQLCODE TO SQL-RETURN-CODE</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Test "SQLCODE" or "SQLSTATE" before proceeding.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>First statement after <code>end-exec</code></p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_24_mar_2017_111834_pierre_yves_nicolas_wrote">on 24 Mar 2017, 11:18:34 Pierre-Yves Nicolas wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] It seems that we have to accept more than just "SQLCODE". <a href="https://www.ibm.com/support/knowledgecenter/SSSNY3_10.1.0/com.ibm.db2.luw.apdv.embed.doc/doc/c0006155.html">"SQLSTATE" is also standard</a>.</p>
</div>
<div class="paragraph">
<p>Moreover, I could find COBOL code which seems to use a custom code. I think we should add a parameter to this rule to list alternative names which can be tested. This parameter could be named "sqlcode_alternatives" (just an idea). What do you think?</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_24_mar_2017_151350_pierre_yves_nicolas_wrote">on 24 Mar 2017, 15:13:50 Pierre-Yves Nicolas wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] After some investigations, I still don&#8217;t understand how COBOL code can use "custom codes". Let&#8217;s forget the parameter for now and just accept both "SQLCODE" and "SQLSTATE".</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_27_mar_2017_090738_ann_campbell_wrote">on 27 Mar 2017, 09:07:38 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>Updated [~pierre-yves.nicolas]</p>
</div>
</div>
</div>
</div>