== Why is this an issue?

The use of positional notation can quickly become confusing, especially in a large or complex data structure. To keep the code clean and understandable, use length notation instead.


=== Noncompliant code example

[source,rpg]
----
     D PERSON          DS                  QUALIFIED
      * Noncompliant
     D  FName                  1     30
     D  LName                 31     60
----


=== Compliant solution

[source,rpg]
----
     D PERSON          DS                  QUALIFIED
     D  FName                        30
     D  LName                        30
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
