== Why is this an issue?

The use of positional notation can quickly become confusing, especially in a large or complex data structure. To keep the code clean and understandable, use length notation instead.


=== Noncompliant code example

[source,rpg]
----
     D PERSON          DS                  QUALIFIED
      * Noncompliant
     D  FName                  1     30
     D  LName                 31     60
----


=== Compliant solution

[source,rpg]
----
     D PERSON          DS                  QUALIFIED
     D  FName                        30
     D  LName                        30
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Use length notation for this data structure definition.


'''
== Comments And Links
(visible only on this page)

=== on 2 Apr 2015, 19:33:25 Ann Campbell wrote:
http://www.bmeyers.net/faqs/14-tips/32-rpg-iv-style?start=2

endif::env-github,rspecator-view[]
