include::../common/summary.adoc[]

== Why is this an issue?

include::../common/description.adoc[]

== How to fix it

include::../common/fix/how-to-fix-it.adoc[]

Sometimes it's not possible to rewrite the regex to be linear while still matching what you want it to match. This can be especially difficult when matching only part of the input. If you cannot simplify the regular expression, consider one of the following approaches:

* Solve the problem without regular expressions.
* Use an alternative non-backtracking engine such as Google's https://github.com/google/re2[RE2] or https://github.com/uhop/node-re2/[node-re2].
* Use multiple passes. This could mean pre- and/or post-processing the string manually before/after applying the regular expression to it or using multiple regular expressions.
** For example, replace `str.split("\\s*,\\s*")` with `str.split(",")` and then trim the spaces from the strings as a second step.
* When using `RegExp.exec()` and `String.match()`, it is often possible to make the regex infallible by making all the parts that could fail optional. This prevents backtracking but could result in matching values that are not acceptable. Further processing will be needed to find the acceptable values.
** For example the regex `x*y` could be replaced with `x*(y)?` and then the call to `str.match(regex)` could be replaced with `match = str.match(regex)` and `match[1] !== undefined`.

=== Code examples

==== Noncompliant code example

This regular expression will suffer catastrophic backtracking.

[source,js,diff-id=1,diff-type=noncompliant]
----
var regex = /^(a+)+$/;  // Noncompliant
regex.test(
"aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa"+
"aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa"+
"aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa"+
"aaaaaaaaaaaaaaa!"
);
----

==== Compliant solution

Possessive quantifiers do not keep backtracking positions, so can be used to avoid performance issues. JavaScript does not support possessive quantifiers, however, lookahead assertions and backreferences can be used to mimic their behavior.

[source,js,diff-id=1,diff-type=compliant]
----
var regex = /^((?=(a+))\2)+$/;
regex.test(
"aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa"+
"aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa"+
"aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa"+
"aaaaaaaaaaaaaaa!"
);
----

=== How does this work?

include::../common/fix/how-does-it-work.adoc[]


== Resources

=== Articles & blog posts

include::../common/resources/articles.adoc[]

=== Standards

include::../common/resources/standards.adoc[]

=== External coding guidelines

include::../common/resources/external-guidelines.adoc[]


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
