== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,java]
----
if (str == null && str.length() == 0) {
  System.out.println("String is empty");
}

if (str != null || str.length() > 0) {
  System.out.println("String is not empty");
}
----


=== Compliant solution

[source,java]
----
if (str == null || str.length() == 0) {
  System.out.println("String is empty");
}

if (str != null && str.length() > 0) {
  System.out.println("String is not empty");
}
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
