include::../description.adoc[]

== Noncompliant Code Example

----
void process(int a) {
   if(a = BAD_OPS) {...}
}
----

== Exceptions

In {cpp}, it is common to null-check a pointer by using it as a condition. In such cases, if the pointer is ``++NULL++``, the condition evaluates to ``++false++``, and otherwise it is ``++true++``. Such pointer assignments are ignored: 


----
int *ip;
if (ip = get_status() ) {...}
----

CLang is able to detect the same kind of quality issues and allows to switch-off those issues by surrounding the non-boolean assignment with some parentheses. To prevent raising some issues which were intentionally switched off, this rule supports the same coding convention: 


----
if((my_status = get_status())) {...}
----

== See

* MISRA C:2004, 13.1 - Assignment operators shall not be used in expressions that yield a Boolean value

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../highlighting.adoc[]

'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
