== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,cpp]
----
void process(int a) {
   if(a = BAD_OPS) {...}
}
----

=== Exceptions

In {cpp}, it is common to null-check a pointer by using it as a condition. In such cases, if the pointer is ``++NULL++``, the condition evaluates to ``++false++``, and otherwise it is ``++true++``. Such pointer assignments are ignored: 


[source,cpp]
----
int *ip;
if (ip = get_status() ) {...}
----

CLang is able to detect the same kind of quality issues and allows to switch-off those issues by surrounding the non-boolean assignment with some parentheses. To prevent raising some issues which were intentionally switched off, this rule supports the same coding convention: 


[source,cpp]
----
if((my_status = get_status())) {...}
----

== Resources

* MISRA C:2004, 13.1 - Assignment operators shall not be used in expressions that yield a Boolean value

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../highlighting.adoc[]

'''
== Comments And Links
(visible only on this page)

=== on 21 Mar 2017, 11:21:05 Ann Campbell wrote:
\[~freddy.mallet] does this double-parens exception need to be added in the double-parens rule as well?

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
