== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,python]
----
name = name
----

=== Compliant solution

[source,python]
----
name = other.name
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
