<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Using more than one <code>OF</code> clause to access a data item can quickly decrease the readability of the source code. Either some <code>OF</code> clauses are optional and should be removed, or there are too many intersections between several data structures and those intersections should be removed.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cobol" data-lang="cobol">01 EMPLOYEE.
  05 MOTHER-IN-LAW.
    10 NAME PIC X(20).
  05 FATHER-IN-LAW.
    10 NAME PIC X(20).
...
01 CUSTOMER.
  05 MOTHER-IN-LAW.
    10 NAME PIC X(20).
  05 FATHER-IN-LAW.
    10 NAME PIC X(20).
...

MOVE MY_VALUE TO NAME OF MOTHER-IN-LAW OF CUSTOMER</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="ulist">
<ul>
<li>
<p>Refactor the "YYY" data structure  to be able to access to "XXXX" with at most one "OF" clause</p>
</li>
<li>
<p>Remove the optional "OF" clauses from this qualified data name to access to "XXXX" with at most one "OF" clause</p>
</li>
</ul>
</div>
</div>
</div>
</div>