Once set, the value of a Hibernate ``++@Entity++``'s ``++@Id++`` field/column should never be updated. Therefore, setters for such fields should always be ``++private++``.


== Noncompliant Code Example

[source,java]
----
public class Book {

  @Id
  @GeneratedValue
  private int id;

  public void setId(int id) {  // Noncompliant
    this.id = id;
  }
----


== Compliant Solution

[source,java]
----
public class Book {

  @Id
  @GeneratedValue
  private int id;

  private void setId(int id) {
    this.id = id;
  }
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

include::highlighting.adoc[]

endif::env-github,rspecator-view[]
