Using the same value on either side of a binary operator is almost always a mistake. In the case of logical operators, it is either a copy/paste error and therefore a bug, or it is simply wasted code, and should be simplified. In the case of bitwise operators and most binary mathematical operators, having the same value on both sides of an operator yields predictable results, and should be simplified.


Note that this rule will raise issues on ``++a == a++`` and ``++a != a++`` expressions which are sometime used to detect ``++NaN++`` values. It is recommended to use instead https://docs.python.org/3/library/math.html#math.isnan[``++math.isnan++``] or an equivalent function. This will improve code readability.

== Noncompliant Code Example

----
if a == a: # Noncompliant
    work()

if  a != a: # Noncompliant
    work()

if  a == b and a == b: # Noncompliant
    work()

if a == b or a == b: # Noncompliant
    work()

j = 5 / 5 # Noncompliant
k = 5 - 5 # Noncompliant
----

== Exceptions

The following are ignored:

* The expression ``++1 << 1++``

include::../see.adoc[]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../highlighting.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
