include::../description.adoc[]

== Noncompliant Code Example

With the default provided regular expression ``++^([A-Z0-9_]*|[a-z0-9_]*)$++``:

----
CLASS MyClass DEFINITION. 
...
ENDCLASS.
----

== Compliant Solution

----
CLASS MY_CLASS DEFINITION. 
...
ENDCLASS.
----

ifdef::rspecator-view[]
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::rspecator-view[]
