include::../description.adoc[]

== Noncompliant Code Example

With the default provided regular expression ``++^([A-Z0-9_]*|[a-z0-9_]*)$++``:

----
CLASS MyClass DEFINITION. 
...
ENDCLASS.
----

== Compliant Solution

----
CLASS MY_CLASS DEFINITION. 
...
ENDCLASS.
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::parameters.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
