:example_env: DISCORD_OAUTH2_TOKEN
:example_name: discord_oauth2_token
:example_secret: i039C7ZUl170kWqDUP2_NWmzv1ULMjyY
:secret_type: secret

include::../../../shared_content/secrets/description.adoc[]

== Why is this an issue?

include::../../../shared_content/secrets/rationale.adoc[]

=== What is the potential impact?

include::../../../shared_content/secrets/impact/generic_impact.adoc[]

== How to fix it

// 1. Revoke leaked secrets

include::../../../shared_content/secrets/fix/revoke.adoc[]

// 2. Analyze recent use to identify misuse

include::../../../shared_content/secrets/fix/recent_use.adoc[]

// 3. Use a secret vault in the future

include::../../../shared_content/secrets/fix/vault.adoc[]

// 4. Never hard-code secrets

include::../../../shared_content/secrets/fix/default.adoc[]

=== Code examples

==== Noncompliant code example

[source,bash,diff-id=1,diff-type=noncompliant]
----
DISCORD_CLIENT_SECRET=i039C7ZUl170kWqDUP2_NWmzv1ULMjyY // Noncompliant
----

==== Compliant solution

[source,bash,diff-id=1,diff-type=compliant]
----
DISCORD_CLIENT_SECRET=${DISCORD_CLIENT_SECRET}
----

== Resources

include::../../../shared_content/secrets/resources/standards.adoc[]
