== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

With the default regular expression ``++^[_a-z][_a-z0-9]*$++``:

[source,python]
----
class MyClass:
  myField = 1
----

=== Compliant solution

[source,python]
----
class MyClass:
  my_field = 1
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

=== Parameters

.format
****

----
^[_a-z][_a-z0-9]*$
----

Regular expression used to check the field names against.
****


'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
