include::../description.adoc[]

== Noncompliant Code Example

With the default regular expression ``++^[_a-z][_a-z0-9]*$++``:

----
class MyClass:
  myField = 1
----

== Compliant Solution

----
class MyClass:
  my_field = 1
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../parameters.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
