<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Method for creating empty arrays <code>Array.Empty(Of TElement)</code> was introduced in .NET 4.6 to optimize object instantiation and memory allocation. It also improves code readability by making developer&#8217;s intent more explicit. This new method should be preferred over empty array declaration.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-vbnet" data-lang="vbnet">Public Sub Method()
    Dim Values1(-1) As Integer ' Noncompliant
    Dim Values2 As Integer() = New Integer() {} ' Noncompliant
    Dim Values3 As Integer() = {} ' Noncompliant
    Dim Values4() As Integer = {} ' Noncompliant
End Sub</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-vbnet" data-lang="vbnet">Public Sub Method()
    Dim Values1 As Integer() = Array.Empty(Of Integer)
    Dim Values2 As Integer() = Array.Empty(Of Integer)
    Dim Values3 As Integer() = Array.Empty(Of Integer)
    Dim Values4() As Integer = Array.Empty(Of Integer)
End Sub</code></pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p><a href="https://docs.microsoft.com/en-us/dotnet/api/system.array.empty?view=netcore-3.1">.Net documentation - <code>Array.Empty&lt;T&gt;</code> Method</a></p>
</li>
<li>
<p><a href="https://justinvp.com/2015/07/20/array-empty/"><code>Array.Empty&lt;T&gt;() - by Justin Van Patten</code></a></p>
</li>
</ul>
</div>
<hr>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Replace this empty array with Array.Empty(Of TElement).</p>
</div>
</div>
</div>
</div>