== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,csharp]
----
if (x < 0)
  new ArgumentException("x must be nonnegative");
----

=== Compliant solution

[source,csharp]
----
if (x < 0)
  throw new ArgumentException("x must be nonnegative");
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
