<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>It is implementation-defined whether the call stack is unwound before <code>terminate()</code> is called, so the destructors of any automatic objects may or may not be executed.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p>MISRA C&#43;&#43;:2008, 15-5-3</p>
</li>
<li>
<p>MISRA C&#43;&#43;:2008, 15-3-1</p>
</li>
<li>
<p>MISRA C&#43;&#43;:2008, 15-3-2</p>
</li>
<li>
<p>MISRA C&#43;&#43;:2008, 15-3-4</p>
</li>
<li>
<p>MISRA C&#43;&#43;:2008, 15-5-1</p>
</li>
</ul>
</div>
<hr>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_is_related_to_s1049">is related to: <a data-rspec-id="S1049" class="rspec-auto-link">S1049</a></h3>

</div>
<div class="sect2">
<h3 id="_is_related_to_s1040">is related to: <a data-rspec-id="S1040" class="rspec-auto-link">S1040</a></h3>

</div>
<div class="sect2">
<h3 id="_is_related_to_s1041">is related to: <a data-rspec-id="S1041" class="rspec-auto-link">S1041</a></h3>

</div>
<div class="sect2">
<h3 id="_on_29_may_2014_190941_ann_campbell_wrote">on 29 May 2014, 19:09:41 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>Not sure we can write a single rule for this. The MISRA doc simply refers to 15-3-1, 15-3-2, 15-3-4 and 15-5-1: "The situations addressed by these rules cause the program to call terminate() and so exhibit implementation-defined behaviour."</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_14_oct_2014_181732_ann_campbell_wrote">on 14 Oct 2014, 18:17:32 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~samuel.mercier] please correct the following</p>
</div>
<div class="ulist">
<ul>
<li>
<p>Code should be quoted in the message. Also, the message should be a full sentence, with initial capital letter and terminating period.</p>
</li>
<li>
<p>Add a See section to the description listing the appropriate MISRA number</p>
</li>
<li>
<p>Fill in Applicability.</p>
</li>
<li>
<p>Make sure the appropriate MISRA C and MISRA C&#43;&#43; fields on the references tab are filled in</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>Also, while I appreciate that you added a Noncompliant Code Example, what you&#8217;ve added is an <em>explicit</em> call to terminate, not an <em>implicit</em> one. Please see my comment from May.</p>
</div>
</div>
</div>
</div>