== How to fix it in IDZSwiftCommonCrypto

=== Code examples

include::../../common/fix/code-rationale.adoc[]

==== Noncompliant code example

[source,swift,diff-id=1,diff-type=noncompliant]
----
import IDZSwiftCommonCrypto

let algorithm = .des // Noncompliant
----

==== Compliant solution

[source,swift,diff-id=1,diff-type=compliant]
----
import IDZSwiftCommonCrypto

let algorithm = .aes
----

=== How does this work?

include::../../common/fix/strong-cryptography.adoc[]

