<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Kotlin supports getters and setters for properties.
Because this is a built-in language feature, it should be the idiom used
to implement the getter and setter pattern instead of using custom idioms.</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="sect3">
<h4 id="_readability_and_understanding">Readability and Understanding</h4>
<div class="paragraph">
<p>This change makes it easier to understand the code
because this is how delegation is to be used in Kotlin.
When developers share common standards and idioms, they need to spend less effort understanding each other&#8217;s code.</p>
</div>
</div>
<div class="sect3">
<h4 id="_code_redundancy">Code Redundancy</h4>
<div class="paragraph">
<p>Using a built-in language feature or a standard API is always better than a custom implementation,
because the reimplementation of something that already exists is unnecessary.</p>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Replace explicit functions that serve as getters and setters with property getters and setters.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-kotlin" data-lang="kotlin">internal class GettersSetters {
    private val length = 100
    private var index: Int = 0
    private var finished: Boolean = false

    fun getIndex(): Int { // Noncomplient, use property getter
        return min(max(0 ,index), length)
    }

    fun setIndex(value: Int) { // Noncompliant, use property setter
        seek(value)
        index = value
        finished = value &gt;= length
    }

    fun isFinished(): Boolean { // Noncomplient, use property getter
        return finished
    }
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-kotlin" data-lang="kotlin">internal class GettersSetters {
    private val length = 100
    private var _index: Int = 0
    private var _finished: Boolean = false

    var index: Int // Compliant
        get() = min(max(0 ,index), length)
        set(value: Int) {
            seek(value)
            _index = value
            _finished = value &gt;= length
        }

    val finished: Boolean // Compliant
        get() = _finished
}</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p><a href="https://kotlinlang.org/docs/properties.html#getters-and-setters">Kotlin Docs, Getters and setters</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>