 When working with collections that are known to be non-empty, using https://learn.microsoft.com/en-us/dotnet/api/system.linq.enumerable.first[`First`] or https://learn.microsoft.com/en-us/dotnet/api/system.linq.enumerable.single[`Single`] is generally preferred over https://learn.microsoft.com/en-us/dotnet/api/system.linq.enumerable.firstordefault[`FirstOrDefault`] or https://learn.microsoft.com/en-us/dotnet/api/system.linq.enumerable.singleordefault[`SingleOrDefault`].

== Why is this an issue?

Using `FirstOrDefault` or `SingleOrDefault` on collections that are known to be non-empty is an issue due to:

* Code Clarity and Intent: When you use `FirstOrDefault` or `SingleOrDefault`, it implies that the collection might be empty, which can be misleading if you know for certain that it is not. This can confuse other developers who read your code, making it harder for them to understand the actual constraints and behavior of the collection, which leads to confusion and harder to maintain code.

* Error Detection: FirstOrDefault and SingleOrDefault return a default value (`null` for reference types and `default` for value types) if the collection is empty. This can lead to subtle bugs if the default value is not properly handled. For example, you might inadvertently introduce a `NullReferenceException` later in your code if you assume the result is never null or assume that the default value of a value type represents something when in releality it's the result of an empty collection. Using `First` or `Single` will throw an `InvalidOperationException` immediately if the collection does not meet the expected conditions, making it easier to detect and fix issues early in the development process.

=== Code examples

==== Noncompliant code example

[source,csharp,diff-id=1,diff-type=noncompliant]
----
var items = new List<int> { 1, 2, 3 };

int firstItem = items.FirstOrDefault(); // Noncompliant, this implies the collection might be empty, when we know it is not
----

==== Compliant solution

[source,csharp,diff-id=1,diff-type=compliant]
----
var items = new List<int> { 1, 2, 3 };

int firstItem = items.First(); // Compliant
----


== Resources

=== Documentation

* Microsoft Learn - https://learn.microsoft.com/en-us/dotnet/api/system.linq.enumerable.single[`Single`]
* Microsoft Learn - https://learn.microsoft.com/en-us/dotnet/api/system.linq.enumerable.first[`First`]
* Microsoft Learn - https://learn.microsoft.com/en-us/dotnet/api/system.linq.enumerable.singleordefault[`SingleOrDefault`]
* Microsoft Learn - https://learn.microsoft.com/en-us/dotnet/api/system.linq.enumerable.firstordefault[`FirstOrDefault`] 

=== Articles & blog posts

* https://medium.com/@anyanwuraphaelc/first-vs-firstordefault-single-vs-singleordefault-a-high-level-look-d24db17a2bc3[First vs FirstOrDefault, Single vs SingleOrDefault: A High-level Look]

