<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Infinite iteration is usually an error in most cases, leading to programs that never terminate. While there are some acceptable use cases such as event streams, developers often unintentionally create infinite loops, causing resource exhaustion or unresponsive behavior.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-rust" data-lang="rust">use std::iter;

iter::repeat(1_u8).collect::&lt;Vec&lt;_&gt;&gt;(); // Noncompliant: This creates an infinite iterator.</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-rust" data-lang="rust">use std::iter;

iter::repeat(1_u8).take(5).collect::&lt;Vec&lt;_&gt;&gt;(); // Compliant: This creates a finite iterator by taking only the first 5 items.</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Clippy Lints - <a href="https://rust-lang.github.io/rust-clippy/master/index.html#infinite_iter" class="bare">https://rust-lang.github.io/rust-clippy/master/index.html#infinite_iter</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>