Testing equality or nullness with PHPUnit's ``assertTrue()`` or ``assertFalse()`` should be simplified to the corresponding dedicated assertion.

== Noncompliant Code Example

----
assertTrue($a === $b);
assertTrue($a == $b);
assertTrue($a === null);
assertTrue($a !== null);
assertTrue($a !== $b);
assertTrue($a != $b);
assertFalse($a === $b);
assertFalse($a == $b);
----

== Compliant Solution

----
assertEquals($a, $b);
assertSame($a, $b);
assertNull($a);
assertNotNull($a);
assertNotEquals($a, $b);
assertNotSame($a, $b);
assertNotEquals($a, $b);
assertNotSame($a, $b);
----
