This rule raises an issue when performing direct type comparisons instead of using the `isinstance()` function.

== Why is this an issue?

In Python, using the `isinstance()` function is generally preferred over direct type comparison for several reasons:

1. *Compatibility with inheritance:* `isinstance()` considers inheritance hierarchy, whereas direct type comparison does not. This means that `isinstance()` can handle cases where an object belongs to a subclass of the specified type, making your code more flexible and robust. It allows you to write code that can work with objects of different but related types.

2. *Support for duck typing:* Python follows the principle of "duck typing," which focuses on an object's behavior rather than its actual type. `isinstance()` enables you to check if an object has certain behavior (by checking if it belongs to a particular class or subclass) rather than strictly requiring a specific type. This promotes code reusability and enhances the flexibility of your programs.

3. *Code maintainability and extensibility:* By using `isinstance()`, your code becomes more maintainable and extensible. If you directly compare types, you would need to modify your code whenever a new subtype is introduced or the inheritance hierarchy is changed. On the other hand, `isinstance()` allows your code to accommodate new types without requiring any modifications, as long as they exhibit the desired behavior.

4. *Polymorphism and interface-based programming:* `isinstance()` supports polymorphism, which is the ability of different objects to respond to the same method calls. It allows you to design code that interacts with objects based on their shared interface rather than their specific types. This promotes code reuse and modularity, as you can write functions and methods that operate on a range of compatible objects.

5. *Third-party library compatibility:* Many third-party libraries and frameworks in Python rely on `isinstance()` for type checking and handling different types of objects. By using `isinstance()`, your code becomes more compatible with these libraries and frameworks, making it easier to integrate your code into larger projects or collaborate with other developers.

In summary, using `isinstance()` over direct type comparison in Python promotes flexibility, code reusability, maintainability, extensibility, and compatibility with the wider Python ecosystem. It aligns with the principles of object-oriented programming and supports the dynamic nature of Python. It is also recommended by the https://peps.python.org/pep-0008/[PEP8] style guide.

== How to fix it

Use the `isinstance()` function instead of performing direct type comparisons.

=== Code examples

==== Noncompliant code example

[source,python,diff-id=1,diff-type=noncompliant]
----
class MyClass:
  ...

def foo(a):
  if type(a) == MyClass: # Noncompliant
    ...
----

==== Compliant solution 

[source,python,diff-id=1,diff-type=compliant]
----
class MyClass:
  ...

def foo(a):
  if isinstance(a, MyClass):
    ...
----

== Resources

=== Documentation

* Python Documentation - https://docs.python.org/3/library/functions.html#isinstance[isInstance() function]
* Style Guide for Python Code - https://peps.python.org/pep-0008/[PEP8]


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

* Use the `isinstance()` function here.


=== Highlighting

Primary: condition

'''
== Comments And Links
(visible only on this page)


endif::env-github,rspecator-view[]
