<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This is an issue when a Rails migration class name does not match the CamelCase version of its snake_case filename (excluding timestamp and <code>.rb</code> extension).</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Rails uses a naming convention to automatically load migration classes based on their filenames. When you run <code>rails db:migrate</code>, Rails looks at each migration file and tries to load a class whose name matches the CamelCase version of the filename.</p>
</div>
<div class="paragraph">
<p>For example, a file named <code>20210213040840_create_articles.rb</code> should contain a class named <code>CreateArticles</code>. Rails transforms the snake_case filename <code>create_articles</code> into CamelCase <code>CreateArticles</code>.</p>
</div>
<div class="paragraph">
<p>When the class name doesn&#8217;t match this expected pattern, Rails cannot find the class and throws an "Uninitialized constant" error. This prevents the migration from running and can block database updates.</p>
</div>
<div class="paragraph">
<p>The naming convention follows these rules:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>Remove the timestamp prefix (e.g., <code>20210213040840_</code>)</p>
</li>
<li>
<p>Remove the <code>.rb</code> extension</p>
</li>
<li>
<p>Convert the remaining snake_case name to CamelCase</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>This convention ensures that Rails can reliably load migration classes without requiring explicit configuration.</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="paragraph">
<p>When migration class names don&#8217;t match their filenames, Rails cannot load the migration class. This causes "Uninitialized constant" errors that prevent migrations from running, blocking database schema updates and potentially breaking deployment processes.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it_in_rails">How to fix it in Rails</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Rename the migration class to match the CamelCase version of the filename. Convert the snake_case filename (without timestamp and extension) to CamelCase.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-ruby" data-lang="ruby"># File: 20210213040840_create_articles.rb
class CreateArticle &lt; ActiveRecord::Migration[6.0]  # Noncompliant
  def change
    create_table :articles do |t|
      t.string :title
      t.text :body
      t.timestamps
    end
  end
end</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-ruby" data-lang="ruby"># File: 20210213040840_create_articles.rb
class CreateArticles &lt; ActiveRecord::Migration[6.0]
  def change
    create_table :articles do |t|
      t.string :title
      t.text :body
      t.timestamps
    end
  end
end</code></pre>
</div>
</div>
<div class="paragraph">
<p>For more complex filenames with multiple words, ensure each word is properly capitalized in the class name.</p>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_code_examples_2">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example_2">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-ruby" data-lang="ruby"># File: 20210213040840_add_user_profile_settings.rb
class AddUserprofileSettings &lt; ActiveRecord::Migration[6.0]  # Noncompliant
  def change
    add_column :users, :profile_settings, :json
  end
end</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution_2">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-ruby" data-lang="ruby"># File: 20210213040840_add_user_profile_settings.rb
class AddUserProfileSettings &lt; ActiveRecord::Migration[6.0]
  def change
    add_column :users, :profile_settings, :json
  end
end</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Rails Migrations Guide - <a href="https://guides.rubyonrails.org/active_record_migrations.html">Official Rails documentation on Active Record migrations and naming conventions</a></p>
</li>
<li>
<p>Rails API: ActiveRecord::Migration - <a href="https://api.rubyonrails.org/classes/ActiveRecord/Migration.html">API documentation for Rails migration classes</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>