include::../description.adoc[]

== Noncompliant Code Example

[source,php]
----
$i = 0;
loop:
  echo("i = $i");
  $i++;
  if ($i < 10){
    goto loop;     
  }
----

== Compliant Solution

[source,php]
----
for ($i = 0; $i < 10; $i++){
  echo("i = $i");
}
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
