== Why is this an issue?

The use of digraphs may not meet developer expectations.


The digraphs are: 

* ``++<%++``
* ``++%>++``
* ``++<:++``
* ``++:>++``
* ``++%:++``
* ``++%:%:++``


=== Noncompliant code example

[source,cpp]
----
template <typename T>
class A
{
  public:
    template<int32_t i>
    void f2();
};

void f(A<int32_t> * a<:10:>)    /* Noncompliant - usage of '<:' instead of '[' and ':>' instead of ']' */
<%                              /* Noncompliant - usage of '<%' instead of '{' */
  a<:0:>->f2<20>();             /* Noncompliant - usage of '<:' and ':>' */
%>                              /* Noncompliant - usage of '%>' instead of '}' */
----


=== Compliant solution

[source,cpp]
----
/* ... */

void f(A<int32_t> * a[10])      /* Compliant */
{                               /* Compliant */
  a[0]->f2<20>();               /* Compliant */
}                               /* Compliant */
----


== Resources

* MISRA {cpp}:2008, 2-5-1 - Digraphs should not be used.

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Replace this digraph 'xxx' by its equivalent 'yyy'.


endif::env-github,rspecator-view[]
