== Why is this an issue?

include::../rationale.adoc[]

include::../impact.adoc[]

== How to fix it

=== Code examples

include::../common/fix/code-rationale.adoc[]

==== Noncompliant code example

[source,vbnet,diff-id=1,diff-type=noncompliant]
----
Public Sub Search(ByVal context As DbContext, ByVal user As String)
    Dim sensitiveQuery As String = String.Concat("SELECT ssn FROM users WHERE id=", user)
    context.Database.ExecuteSqlCommand(sensitiveQuery) ' Noncompliant
End Sub
----

==== Compliant solution

[source,vbnet,diff-id=1,diff-type=compliant]
----
Public Sub Search(ByVal context As DbContext, ByVal user As String)
    context.Database.ExecuteSqlCommand("SELECT ssn FROM users WHERE id=@p0", user)
End Sub
----

=== How does this work?

include::../common/fix/how.adoc[]

== Resources

include::../common/resources/standards.adoc[]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../highlighting.adoc[]

endif::env-github,rspecator-view[]
