== Why is this an issue?

Even if your document is arranged in many layers, you don't have to step through those layers explicitly to select an element. In fact, it's more efficiently to skim over some of those layers rather than making the JQuery selector engine explicitly step through each one.


=== Noncompliant code example

[source,javascript]
----
$( "div.data table.attendees td.campbell" );  // Noncompliant; too many selectors. There's no need to trace through each layer of the structure
----


=== Compliant solution

[source,javascript]
----
$( ".data td.campbell");
----


ifdef::env-github,rspecator-view[]
'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
