== How to fix it in .NET

=== Code examples

include::../../common/fix/code-rationale.adoc[]

==== Noncompliant code example

[source,vbnet,diff-id=1,diff-type=noncompliant]
----
Imports System.Diagnostics

Public Class ExampleController
    Inherits Controller

    Public Sub Example(Name As String, Value As String)
        Dim Proc As New Process()
        Proc.StartInfo.FileName = "path/to/executable"
        Proc.StartInfo.EnvironmentVariables.Add(Name, Value) ' Noncompliant
        Proc.Start()
    End Sub
End Class
----

==== Compliant solution

[source,vbnet,diff-id=1,diff-type=compliant]
----
Imports System.Diagnostics
Imports System.Text.RegularExpressions

Public Class ExampleController
    Inherits Controller

    Public Sub Example(Value As String)
        Dim Proc As New Process()
        Proc.StartInfo.FileName = "path/to/executable"
        Dim Pattern As String = "^[a-zA-Z0-9]*$"
        Dim M As Match = Regex.Match(Value, Pattern)
        If M.Success Then Proc.StartInfo.EnvironmentVariables.Add("ENV_VAR", Value)
        Proc.Start()
    End Sub
End Class
----

=== How does this work?

include::../../common/fix/pre-approved-list.adoc[] 

