== How to fix it in .NET

=== Code examples

include::../../common/fix/code-rationale.adoc[]

==== Noncompliant code example

[source,vbnet,diff-id=1,diff-type=noncompliant]
----
Imports System.Diagnostics

Public Class ExampleController
    Inherits Controller

    Public Sub Example(name As String, value As String)
        Dim proc As New Process()
        proc.StartInfo.FileName = "path/to/executable"
        proc.StartInfo.EnvironmentVariables.Add(name, value) ' Noncompliant
        proc.Start()
    End Sub
End Class
----

==== Compliant solution

[source,vbnet,diff-id=1,diff-type=compliant]
----
Imports System.Diagnostics
Imports System.Text.RegularExpressions

Public Class ExampleController
    Inherits Controller

    Public Sub Example(value As String)
        Dim proc As New Process()
        proc.StartInfo.FileName = "path/to/executable"
        Dim pattern As String = "^[a-zA-Z0-9]*$"
        Dim m As Match = Regex.Match(value, pattern)
        If m.Success Then
            proc.StartInfo.EnvironmentVariables.Add("ENV_VAR", value)
        End If
        proc.Start()
    End Sub
End Class
----

=== How does this work?

include::../../common/fix/pre-approved-list.adoc[] 

