== Why is this an issue?

In Android applications, unnecessary CPU usage can increase power consumption, negatively impacting the device's battery life.
To prevent excessive drain, Android devices enter an idle state when not in use.
Developers should be cautious about keeping the CPU active unnecessarily, and if required, they should employ the appropriate mechanisms provided by the Android framework.

`PowerManager.WakeLock` is a mechanism to indicate that your application needs to have the device stay on.
`acquire()` is a method to acquire the wake lock and force the device to stay on at the requested level when the wake lock was created.
`release()` is a method to release the wake lock and allow the device to sleep again.
Doing this as soon as possible is very important to avoid running down the device's battery excessively.

== How to fix it

After every call to `acquire()`, there should be a corresponding call to `release()` or `acquire(long timeout)` with a timeout.
The timeout should be set to the reasonable amount of time the application needs to hold the wake lock.

=== Code examples

==== Noncompliant code example

[source,java,diff-id=1,diff-type=noncompliant]
----
public void onCreate(Bundle savedInstanceState) {
  PowerManager powerManager = (PowerManager) getSystemService(Context.POWER_SERVICE);
  PowerManager.WakeLock wakeLock = powerManager.newWakeLock(PowerManager.PARTIAL_WAKE_LOCK, "My Wake Lock");
  wakeLock.acquire(); // Noncompliant
  // do some work...
}
----

==== Compliant solution

[source,java,diff-id=1,diff-type=compliant]
----
public void onCreate(Bundle savedInstanceState) {
  PowerManager powerManager = (PowerManager) getSystemService(Context.POWER_SERVICE);
  PowerManager.WakeLock wakeLock = powerManager.newWakeLock(PowerManager.PARTIAL_WAKE_LOCK, "My Wake Lock");
  wakeLock.acquire(); // Compliant
  // do some work...
  wakeLock.release();
}
----

== Resources

=== Documentation

* https://developer.android.com/reference/android/os/PowerManager.WakeLock[Android - PowerManager.WakeLock]

=== Articles &blog posts

* https://developer.android.com/develop/background-work/background-tasks/scheduling/wakelock[Android - WakeLock]

