<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when variables are declared but never used, assigned but never read, or when pointer variables are never dereferenced.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Unused variables and ineffectual assignments create dead code that serves no purpose in program execution. This dead code:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>Reduces code readability by cluttering the codebase with unnecessary declarations</p>
</li>
<li>
<p>Can confuse other developers who might assume the variable serves a purpose</p>
</li>
<li>
<p>May indicate incomplete implementations or leftover code from refactoring</p>
</li>
<li>
<p>Increases maintenance burden without providing any value</p>
</li>
<li>
<p>Can mask real issues where a variable was intended to be used but was forgotten</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>In Go, the compiler will catch completely unused variables, but it won&#8217;t detect cases where variables are assigned but never read, or where pointer variables are never dereferenced. These patterns often indicate logic errors or incomplete code that should be addressed.</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="paragraph">
<p>While unused variables don&#8217;t directly impact runtime performance or security, they reduce code quality and maintainability. Dead code can hide bugs, make code reviews more difficult, and create confusion for developers working on the codebase.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Remove the unused variable declaration entirely when the variable serves no purpose.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-go" data-lang="go">func processData() {
    result := calculateValue() // Noncompliant
    fmt.Println("Processing complete")
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-go" data-lang="go">func processData() {
    fmt.Println("Processing complete")
}</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Go Tour - Variables - <a href="https://go.dev/tour/basics/8">Official Go tutorial covering variable declarations and usage</a></p>
</li>
<li>
<p>Go Tour - Pointers - <a href="https://go.dev/tour/moretypes/1">Official Go tutorial explaining pointer syntax and dereferencing</a></p>
</li>
<li>
<p>Effective Go - Blank identifier - <a href="https://go.dev/doc/effective_go#blank">Official documentation on using the blank identifier for unused values</a></p>
</li>
</ul>
</div>
</div>
<div class="sect2">
<h3 id="_standards">Standards</h3>
<div class="ulist">
<ul>
<li>
<p>Go Code Review Comments - <a href="https://github.com/golang/go/wiki/CodeReviewComments">Official Go code review guidelines that emphasize clean, readable code</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>