== Why is this an issue?

Free-format syntax is preferred because it is clearer, easier to write, and easier for newcomers to the language than fixed-format syntax. 


=== Noncompliant code example

[source,rpg]
----
     C                   IF        A > 10
     C                   EVAL      B = 1
     C                   ENDIF
----


=== Compliant solution

[source,rpg]
----
      /free
        if A > 10;
          B = 1;
        endif;
      /end-free
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
