Free-format syntax is preferred because it is clearer, easier to write, and easier for newcomers to the language than fixed-format syntax. 


== Noncompliant Code Example

----
     C                   IF        A > 10
     C                   EVAL      B = 1
     C                   ENDIF
----


== Compliant Solution

----
      /free
        if A > 10;
          B = 1;
        endif;
      /end-free
----


ifdef::env-github,rspecator-view[]
'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
