`StringBuilder` instances that never build a `string` clutter the code and worse are a drag on performance. Either they should be removed, or the missing `ToString()` call should be added.

== Noncompliant Code Example

[source,csharp]
----
public void DoSomething(List<string> strings) {
  var sb = new StringBuilder();  // Noncompliant
  sb.Append("Got: ");
  foreach(var str in strings) {
    sb.Append(str).Append(", ");
    // ...
  }
}
----

== Compliant Solution

[source,csharp]
----
public void DoSomething(List<string> strings) {
  foreach(var str in strings) {
    // ...
  }
}
----
or
[source,csharp]
----
public void DoSomething(List<string> strings) {
  var sb = new StringBuilder();
  sb.Append("Got: ");
  foreach(var str in strings) {
    sb.Append(str).Append(", ");
    // ...
  }
  logger.LogInformation(sb.toString());
}
----

== Exceptions

No issue is reported when `StringBuilder` is:

* Accessed through:
** A `CopyTo()` or `GetChunks()` invocation.
** An indexer (`sb[0]`).
** The `.Length` expression.
* Passed as a method argument, on the grounds that it will likely be accessed through a `ToString()` invocation there.
* Passed in as a parameter to the current method, on the grounds that the callee will materialize the string.
* Retrieved by a custom function (`var sb = GetStringBuilder();`).
* Returned by the method.