`StringBuilder` instances that are never `ToString`ed needlessly clutter the code, and worse are a drag on performance. Either they should be removed, or the missing `ToString` call added.

== Noncompliant Code Example

[source,csharp]
----
public void DoSomething(List<string> strings) {
  var sb = new StringBuilder();  // Noncompliant
  sb.Append("Got: ");
  foreach(var str in strings) {
    sb.Append(str).Append(", ");
    // ...
  }
}
----

== Compliant Solution

[source,csharp]
----
public void DoSomething(List<string> strings) {
  foreach(var str in strings) {
    // ...
  }
}
----
or
[source,csharp]
----
public void DoSomething(List<string> strings) {
  var sb = new StringBuilder();
  sb.Append("Got: ");
  foreach(var str in strings) {
    sb.Append(str).Append(", ");
    // ...
  }
  _logger.LogInformation(sb.toString, DateTimeOffset.UtcNow);
}
----

== Exceptions

No issue is reported when `StringBuilder` is:

* accessed through a ``CopyTo()`` or ``GetChunks()`` invocation.
[source,csharp]
----
var sb = new StringBuilder(); // Compliant
sb.GetChunks();
----
* passed as method argument, on the grounds that it will likely `ToString`ed there.
* a parameter of the current method
* retrieved by a custom function
[source,csharp]
----
var sb = GetStringBuilder(); // Compliant
----
* returned by the method