`StringBuilder` instances that are not accessed through a `ToString()` invocation needlessly clutter the code, and worse are a drag on performance. Either they should be removed, or the missing `ToString()` call should be added.

== Noncompliant Code Example

[source,csharp]
----
public void DoSomething(List<string> strings) {
  var sb = new StringBuilder();  // Noncompliant
  sb.Append("Got: ");
  foreach(var str in strings) {
    sb.Append(str).Append(", ");
    // ...
  }
}
----

== Compliant Solution

[source,csharp]
----
public void DoSomething(List<string> strings) {
  foreach(var str in strings) {
    // ...
  }
}
----
or
[source,csharp]
----
public void DoSomething(List<string> strings) {
  var sb = new StringBuilder();
  sb.Append("Got: ");
  foreach(var str in strings) {
    sb.Append(str).Append(", ");
    // ...
  }
  _logger.LogInformation(sb.toString);
}
----

== Exceptions

No issue is reported when `StringBuilder` is:

* Accessed through a `CopyTo()` or `GetChunks()` invocation.
* Passed as method argument, on the grounds that it will likely accessed through a `ToString()` invocation there.
* A parameter of the current method.
* Retrieved by a custom function (`var sb = GetStringBuilder();`).
* Returned by the method.