<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when a method uses an explicit <code>return</code> keyword for the last expression that could be returned implicitly.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>In Groovy, the last expression in a method is automatically returned, making the explicit <code>return</code> keyword redundant in many cases. This is one of Groovy&#8217;s key features that distinguishes it from Java and makes code more concise.</p>
</div>
<div class="paragraph">
<p>Using unnecessary <code>return</code> keywords:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>Makes code more verbose than needed</p>
</li>
<li>
<p>Goes against Groovy coding conventions and idioms</p>
</li>
<li>
<p>Reduces readability by adding visual clutter</p>
</li>
<li>
<p>Makes the code look more Java-like rather than taking advantage of Groovy&#8217;s features</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>Groovy&#8217;s implicit return feature is particularly beneficial for:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>Single-expression methods</p>
</li>
<li>
<p>Simple getter methods</p>
</li>
<li>
<p>Methods that end with a single calculation or operation</p>
</li>
<li>
<p>Closures and short utility methods</p>
</li>
</ul>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="paragraph">
<p>This issue has minimal impact on functionality but affects code maintainability and readability. Removing unnecessary <code>return</code> keywords makes the codebase more consistent with Groovy conventions and easier to read for developers familiar with the language.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Remove the <code>return</code> keyword from the last expression in the method. The last expression will be automatically returned by Groovy.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-groovy" data-lang="groovy">def getName() {
    return 'John' // Noncompliant
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-groovy" data-lang="groovy">def getName() {
    'John'
}</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Groovy Language Documentation - Methods - <a href="https://groovy-lang.org/semantics.html#_return_keyword">Official Groovy documentation explaining implicit returns and when the return keyword is optional</a></p>
</li>
<li>
<p>Groovy Style Guide - <a href="https://groovy-lang.org/style-guide.html">Official style guide with recommendations for idiomatic Groovy code</a></p>
</li>
</ul>
</div>
</div>
<div class="sect2">
<h3 id="_standards">Standards</h3>
<div class="ulist">
<ul>
<li>
<p>Groovy Best Practices - <a href="https://groovy-lang.org/style-guide.html#_return_keyword">Groovy community guidelines for writing clean, idiomatic code</a></p>
</li>
</ul>
</div>
</div>
<div class="sect2">
<h3 id="_related_rules">Related rules</h3>
<div class="ulist">
<ul>
<li>
<p><a data-rspec-id="S1172" class="rspec-auto-link">RSPEC-1172</a> - <a href="https://rules.sonarsource.com/java/RSPEC-1172/">Unused method parameters should be removed (related to code simplification)</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>