This rule raises an issue when retrieving a single record by primary key ID using `where(id: id).take!`, `find_by!(id: id)`, or `find_by_id!(id)` instead of the more direct `find(id)` method.

== Why is this an issue?

ActiveRecord provides the `find` method specifically for retrieving records by their primary key. This method is more direct, readable, and expressive than using generic query methods like `where.take!` or `find_by!` when looking up by ID.

Using `find` makes the intent clearer - you're looking for a specific record by its primary key. The alternatives are more verbose and don't clearly communicate that you're doing a primary key lookup. Additionally, `find` is optimized for primary key lookups and may offer better performance.

All these methods have the same behavior when the record is not found - they raise `ActiveRecord::RecordNotFound`. However, `find` is the conventional and preferred approach in the Rails community for this specific use case.

=== What is the potential impact?

Using verbose alternatives to `find` reduces code readability and doesn't follow Rails conventions. While the functionality is equivalent, the code becomes less expressive about its intent and may be slightly less performant for primary key lookups.

== How to fix it in Rails

Replace `where(id: id).take!` with `find(id)` for primary key lookups. Both methods raise `ActiveRecord::RecordNotFound` when the record is not found, but `find` is more direct and conventional.

=== Code examples

==== Noncompliant code example

[source,ruby,diff-id=1,diff-type=noncompliant]
----
User.where(id: user_id).take! # Noncompliant
----

==== Compliant solution

[source,ruby,diff-id=1,diff-type=compliant]
----
User.find(user_id)
----

Replace `find_by!(id: id)` with `find(id)` for primary key lookups. The `find` method is specifically designed for this purpose and is more readable.

=== Code examples

==== Noncompliant code example

[source,ruby,diff-id=2,diff-type=noncompliant]
----
User.find_by!(id: user_id) # Noncompliant
----

==== Compliant solution

[source,ruby,diff-id=2,diff-type=compliant]
----
User.find(user_id)
----

Replace the deprecated `find_by_id!(id)` with `find(id)`. The `find` method is the modern, preferred approach for primary key lookups.

=== Code examples

==== Noncompliant code example

[source,ruby,diff-id=3,diff-type=noncompliant]
----
User.find_by_id!(user_id) # Noncompliant
----

==== Compliant solution

[source,ruby,diff-id=3,diff-type=compliant]
----
User.find(user_id)
----

== Resources

=== Documentation

 * Rails Guides - Active Record Query Interface - https://guides.rubyonrails.org/active_record_querying.html#retrieving-a-single-object[Official Rails documentation on retrieving single objects, including the find method]

 * Rails API Documentation - ActiveRecord::FinderMethods#find - https://api.rubyonrails.org/classes/ActiveRecord/FinderMethods.html#method-i-find[API documentation for the find method in ActiveRecord]
