<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when a class contains test methods but is not annotated with <code>@isTest</code> at the class level.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>In Salesforce, test classes serve a specific purpose and should be properly identified using the <code>@isTest</code> annotation. This annotation provides two key benefits:</p>
</div>
<div class="paragraph">
<p>First, it excludes the test class from your organization&#8217;s 6MB Apex code storage limit. Without this annotation, test classes count toward your storage quota, which can become problematic as your test suite grows.</p>
</div>
<div class="paragraph">
<p>Second, it clearly identifies the class as a test class, making your codebase more maintainable and easier to understand. This explicit marking helps developers quickly distinguish between production code and test code.</p>
</div>
<div class="paragraph">
<p>When a class contains test methods (either using the <code>testMethod</code> keyword or methods annotated with <code>@isTest</code>) but lacks the class-level <code>@isTest</code> annotation, it may consume unnecessary storage space and lacks clear identification as a test class.</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="paragraph">
<p>Without the <code>@isTest</code> annotation, test classes count toward your organization&#8217;s 6MB Apex code storage limit. This can lead to storage quota issues as your test suite grows, potentially preventing you from deploying new code when the limit is reached.</p>
</div>
<div class="paragraph">
<p>Additionally, the lack of clear test class identification makes code maintenance more difficult and can confuse developers about the purpose of the class.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Add the <code>@isTest</code> annotation at the class level to properly identify the class as a test class and exclude it from code storage limits.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-apex" data-lang="apex">public class AccountTest {
    static void testAccountCreation() { // Noncompliant
        // test logic
    }
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-apex" data-lang="apex">@isTest
public class AccountTest {
    static void testAccountCreation() {
        // test logic
    }
}</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Apex Test Class Best Practices - <a href="https://www.apexhours.com/apex-test-class-best-practices">Comprehensive guide to Apex test class best practices including the use of @isTest annotation</a></p>
</li>
<li>
<p>Apex Unit Test Tips - <a href="https://www.apexhours.com/12-apex-unit-test-tips-every-salesforce-developer-should-know">Essential tips for Apex unit testing including code storage limits and @isTest usage</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>