<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>A method or class with too many type parameters has likely aggregated too many responsibilities and should be split.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="paragraph">
<p>With the default parameter value of 2:</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-text" data-lang="text">&lt;S, T, U, V&gt; void foo() {} // Noncompliant; not really readable
&lt;String, Integer, Object, String&gt;foo(); // especially on invocations</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Reduce the number of generic parameters in the '{type name}' [class|record|struct|interface] to no more than the {max} authorized.</p>
</div>
<div class="paragraph">
<p>Reduce the number of generic parameters in the '{type name}.{method name}' method to no more than the {maxMethod} authorized.</p>
</div>
</div>
<div class="sect2">
<h3 id="_parameters">Parameters</h3>
<div class="sidebarblock">
<div class="content">
<div class="title">max</div>
<div class="listingblock">
<div class="content">
<pre>2</pre>
</div>
</div>
<div class="paragraph">
<p>Maximum authorized number of generic parameters.</p>
</div>
</div>
</div>
<div class="sidebarblock">
<div class="content">
<div class="title">maxMethod</div>
<div class="listingblock">
<div class="content">
<pre>3</pre>
</div>
</div>
<div class="paragraph">
<p>Maximum authorized number of generic parameters for methods.</p>
</div>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_6_nov_2017_164128_amaury_levé_wrote">on 6 Nov 2017, 16:41:28 Amaury Levé wrote:</h3>
<div class="paragraph">
<p>Could you review the change? Just to make sure I have declared the 2 parameters correctly.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_6_nov_2017_194533_ann_campbell_wrote">on 6 Nov 2017, 19:45:33 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>Looks good [~amaury.leve]</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_20_jan_2015_154140_ann_campbell_wrote">on 20 Jan 2015, 15:41:40 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~nicolas.peru], you&#8217;re going to have to help me out with the whys and wherefores on this one.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_20_jan_2015_160001_freddy_mallet_wrote">on 20 Jan 2015, 16:00:01 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>This rule seems to duplicate <a data-rspec-id="S107" class="rspec-auto-link">RSPEC-107</a> ???</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_21_jan_2015_173938_nicolas_peru_wrote">on 21 Jan 2015, 17:39:38 Nicolas Peru wrote:</h3>
<div class="paragraph">
<p>\[~freddy.mallet] This is about Type parameters not parameters.</p>
</div>
<div class="paragraph">
<p>\[<sub>ann.campbell.2][</sub>freddy.mallet] For me this is the same reasons as for <a data-rspec-id="S107" class="rspec-auto-link">RSPEC-107</a> : if you have too many type parameters then your class/method is probably doing too many things.</p>
</div>
<div class="paragraph">
<p>Another reason to avoid too many type params is readability :</p>
</div>
<div class="listingblock">
<div class="content">
<pre>&lt;S, T, U, V&gt; void foo() {} //not really readable
&lt;String, Integer, Object, String&gt;foo(); // especially on invocations</pre>
</div>
</div>
</div>
</div>
</div>