<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>When two locks are held simultaneously, a <code>wait</code> call only releases one of them. The other will be held until some other thread requests a lock on the awaited object.</p>
</div>
<div class="paragraph">
<p>If no unrelated code tries to lock on that object, then all other threads will be locked out, resulting in a deadlock.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-text" data-lang="text">synchronized (this.mon1) {  // threadB can't enter this block to request this.mon2 lock &amp; release threadA
	synchronized (this.mon2) {
		this.mon2.wait();  // Noncompliant; threadA is stuck here holding lock on this.mon1
	}
}</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_16_jun_2015_133420_ann_campbell_wrote">on 16 Jun 2015, 13:34:20 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>FYI [~tamas.vajk]</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_16_jun_2015_133603_ann_campbell_wrote">on 16 Jun 2015, 13:36:03 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~nicolas.peru] &amp; [~tamas.vajk] should this "potential deadlock" rule be combined with <a data-rspec-id="S3046" class="rspec-auto-link">RSPEC-3046</a>: '"notify" should not be called when multiple locks are held'?</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_7_nov_2016_170931_tibor_blenessy_wrote">on 7 Nov 2016, 17:09:31 Tibor Blenessy wrote:</h3>
<div class="paragraph">
<p>I will start to work on this in my current sprint. Is there any feedback from [~nicolas.peru] or [~tamas.vajk] regarding [~ann.campbell.2] question?</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_14_nov_2016_173152_tibor_blenessy_wrote">on 14 Nov 2016, 17:31:52 Tibor Blenessy wrote:</h3>
<div class="paragraph">
<p>Looking little bit more into this, I would say that it would be indeed better to merge this rule with <a data-rspec-id="S3046" class="rspec-auto-link">RSPEC-3046</a> , since wait/notify are closely cooperating and description could benefit from having both concepts described together.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_14_nov_2016_175942_ann_campbell_wrote">on 14 Nov 2016, 17:59:42 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>rolled into <a data-rspec-id="S3046" class="rspec-auto-link">RSPEC-3046</a></p>
</div>
</div>
</div>
</div>