<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Some COBOL compilers such as IBM one will assume that the minimum value of <code>OCCURS DEPENDING ON</code> is 1 but nothing is enforcing that and this behaviour can change or be different when using another compiler.</p>
</div>
<div class="paragraph">
<p>Setting the minimum value of <code>OCCURS DEPENDING ON</code> makes the code more readable, and less dependent on compiler.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cobol" data-lang="cobol">01 MY-TABLE-COUNT PIC S9(4) BINARY.
01 MY-TABLE.
   03 MY-ITEM OCCURS 500 TIMES          *&gt; Noncompliant
       DEPENDING ON MY-TABLE-COUNT.
      05 MY-FIELD-01 PIC X(08).
      05 MY-FIELD-02 PIC 9(05).</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cobol" data-lang="cobol">01 MY-TABLE-COUNT PIC S9(4) BINARY.
01 MY-TABLE.
   03 MY-ITEM OCCURS 1 TO 500 TIMES          *&gt; Compliant; minimum value is 1
       DEPENDING ON MY-TABLE-COUNT.
      05 MY-FIELD-01 PIC X(08).
      05 MY-FIELD-02 PIC 9(05).</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Add a minimum value to this "OCCURS DEPENDING [ON]" clause.</p>
</div>
</div>
</div>
</div>