An invalid color definition will by default be interpreted as black, which is likely to have unintended impacts on the expected look and feel of the website.


This rule raises an issue when a color definition (``++color++``, ``++background-color++``) is not valid. The color definition is considered valid when it is made of hexadecimal characters:

* longhand: 6 or 8 characters (when alpha is defined)
* shorthand variant: 3 or 4 characters (when alpha is defined)


== Noncompliant Code Example

[source,css]
----
a {
 color: #3c; /* Noncompliant; shorthand should be made of 3 characters */
}
div {
  background-color: #3cb371a; /* Noncompliant; alpha should have 2 characters */
}
----


== Compliant Solution

[source,css]
----
a {
 color: #3cc;
}
div {
  background-color: #3cb371ac;
}
----


== See

* https://developer.mozilla.org/en-US/docs/Web/CSS/color_value[Mozilla Web Technology for Developers] - CSS Color


ifdef::env-github,rspecator-view[]
'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
