<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Memory that is allocated with <code>new T[n]</code> <em>must</em> be freed with <code>delete[]</code>. Leave out the <code>[]</code>, and the likely result is heap corruption or, as a best-case scenario, premature program termination.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">char *cp = new char[10];
// ...
delete cp;  // Noncompliant</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">char *cp = new char[10];
// ...
delete[] cp;</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_duplicates_s1232">duplicates: <a data-rspec-id="S1232" class="rspec-auto-link">S1232</a></h3>

</div>
<div class="sect2">
<h3 id="_on_10_mar_2016_222947_evgeny_mandrikov_wrote">on 10 Mar 2016, 22:29:47 Evgeny Mandrikov wrote:</h3>
<div class="paragraph">
<p>Looks like duplicate of <a data-rspec-id="S1232" class="rspec-auto-link">RSPEC-1232</a></p>
</div>
</div>
<div class="sect2">
<h3 id="_on_11_mar_2016_083155_ann_campbell_wrote">on 11 Mar 2016, 08:31:55 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>Thanks [~evgeny.mandrikov], you&#8217;re right.</p>
</div>
</div>
</div>
</div>