== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,javascript]
----
try {
  doSomething();
} catch (ex) {  // Noncompliant
  throw ex;
}
----

=== Compliant solution

[source,javascript]
----
try {
  doSomething();
} catch (ex) {
  console.err(ex);
  throw ex;
}
----
or

[source,javascript]
----
doSomething();
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
