include::../description.adoc[]

== Noncompliant Code Example

[source,javascript]
----
try {
  doSomething();
} catch (ex) {  // Noncompliant
  throw ex;
}
----

== Compliant Solution

[source,javascript]
----
try {
  doSomething();
} catch (ex) {
  console.err(ex);
  throw ex;
}
----
or

[source,javascript]
----
doSomething();
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
