include::../description.adoc[]

== Noncompliant Code Example

[source,go]
----
foo(); bar() // Noncompliant
----

== Compliant Solution

[source,go]
----
foo()
bar()
----

== Exceptions

Control flow statements with a single nested statement are ignored.


----
if condition { doSomething() } // Compliant
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../highlighting.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
