include::../description.adoc[]

== Noncompliant Code Example

----
foo(); bar() // Noncompliant
----

== Compliant Solution

----
foo()
bar()
----

== Exceptions

Control flow statements with a single nested statement are ignored.


----
if condition { doSomething() } // Compliant
----
