== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,go]
----
foo(); bar() // Noncompliant
----

=== Compliant solution

[source,go]
----
foo()
bar()
----

=== Exceptions

Control flow statements with a single nested statement are ignored.


----
if condition { doSomething() } // Compliant
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../highlighting.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
