<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>The Java Language Specification recommends listing modifiers in the following order:</p>
</div>
<div class="olist arabic">
<ol class="arabic">
<li>
<p>Annotations</p>
</li>
<li>
<p>public</p>
</li>
<li>
<p>protected</p>
</li>
<li>
<p>private</p>
</li>
<li>
<p>abstract</p>
</li>
<li>
<p>static</p>
</li>
<li>
<p>final</p>
</li>
<li>
<p>transient</p>
</li>
<li>
<p>volatile</p>
</li>
<li>
<p>synchronized</p>
</li>
<li>
<p>native</p>
</li>
<li>
<p>default</p>
</li>
<li>
<p>strictfp</p>
</li>
</ol>
</div>
<div class="paragraph">
<p>Not following this convention has no technical impact, but will reduce the code&#8217;s readability because most developers are used to the standard order.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">static public void main(String[] args) {   // Noncompliant
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">public static void main(String[] args) {   // Compliant
}</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Reorder the modifiers to comply with the [Java Language Specification|PSR2 standard].</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_is_duplicated_by_s1789">is duplicated by: <a data-rspec-id="S1789" class="rspec-auto-link">S1789</a></h3>

</div>
</div>
</div>