== How to fix it in a Servlet

=== Code examples

include::../../common/fix/code-rationale.adoc[]

==== Noncompliant code example

[source,kotlin,diff-id=1,diff-type=noncompliant]
----
class RequestHandler {
    fun doGet(req: HttpServletRequest, resp: HttpServletResponse) {
        val location = req.getParameter("url")
        resp.sendRedirect(location)
    }
}
----

==== Compliant solution

[source,kotlin,diff-id=1,diff-type=compliant]
----
class RequestHandler {
    companion object {
        val ALLOWED_HOSTS = arrayOf("https://trusted1.example.com/", "https://trusted2.example.com/")
    }
    fun doGet(req: HttpServletRequest, resp: HttpServletResponse) {
        val location = req.getParameter("url")
        if (location in ALLOWED_HOSTS) {
            resp.sendRedirect(location)
        }
    }
}
----

include::../../common/fix/how-does-this-work.adoc[]

=== Pitfalls

include::../../common/pitfalls/starts-with.adoc[]

