This is an issue when a Rails migration class name does not match the CamelCase version of its snake_case filename (excluding timestamp and `.rb` extension).

== Why is this an issue?

Rails uses a naming convention to automatically load migration classes based on their filenames. When you run `rails db:migrate`, Rails looks at each migration file and tries to load a class whose name matches the CamelCase version of the filename.

For example, a file named `20210213040840_create_articles.rb` should contain a class named `CreateArticles`. Rails transforms the snake_case filename `create_articles` into CamelCase `CreateArticles`.

When the class name doesn't match this expected pattern, Rails cannot find the class and throws an "Uninitialized constant" error. This prevents the migration from running and can block database updates.

The naming convention follows these rules:

* Remove the timestamp prefix (e.g., ``++20210213040840_++``)
* Remove the `.rb` extension
* Convert the remaining snake_case name to CamelCase

This convention ensures that Rails can reliably load migration classes without requiring explicit configuration.

=== What is the potential impact?

When migration class names don't match their filenames, Rails cannot load the migration class. This causes "Uninitialized constant" errors that prevent migrations from running, blocking database schema updates and potentially breaking deployment processes.

== How to fix it in Rails

Rename the migration class to match the CamelCase version of the filename. Convert the snake_case filename (without timestamp and extension) to CamelCase.

=== Code examples

==== Noncompliant code example

[source,ruby,diff-id=1,diff-type=noncompliant]
----
# File: 20210213040840_create_articles.rb
class CreateArticle < ActiveRecord::Migration[6.0]  # Noncompliant
  def change
    create_table :articles do |t|
      t.string :title
      t.text :body
      t.timestamps
    end
  end
end
----

==== Compliant solution

[source,ruby,diff-id=1,diff-type=compliant]
----
# File: 20210213040840_create_articles.rb
class CreateArticles < ActiveRecord::Migration[6.0]
  def change
    create_table :articles do |t|
      t.string :title
      t.text :body
      t.timestamps
    end
  end
end
----

For more complex filenames with multiple words, ensure each word is properly capitalized in the class name.

=== Code examples

==== Noncompliant code example

[source,ruby,diff-id=2,diff-type=noncompliant]
----
# File: 20210213040840_add_user_profile_settings.rb
class AddUserprofileSettings < ActiveRecord::Migration[6.0]  # Noncompliant
  def change
    add_column :users, :profile_settings, :json
  end
end
----

==== Compliant solution

[source,ruby,diff-id=2,diff-type=compliant]
----
# File: 20210213040840_add_user_profile_settings.rb
class AddUserProfileSettings < ActiveRecord::Migration[6.0]
  def change
    add_column :users, :profile_settings, :json
  end
end
----

== Resources

=== Documentation

 * Rails Migrations Guide - https://guides.rubyonrails.org/active_record_migrations.html[Official Rails documentation on Active Record migrations and naming conventions]

 * Rails API: ActiveRecord::Migration - https://api.rubyonrails.org/classes/ActiveRecord/Migration.html[API documentation for Rails migration classes]
