<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule checks that all string literals use the same kind of quotes.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="paragraph">
<p>Using the parameter default (forcing single quotes):</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-javascript" data-lang="javascript">var firstParameter = "something"; // Noncompliant</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-javascript" data-lang="javascript">var firstParameter = 'something';</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_exceptions">Exceptions</h3>
<div class="paragraph">
<p>Strings that contain quotes are ignored.</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-javascript" data-lang="javascript">let heSaid = "Then he said 'What?'."  // ignored
let sheSaid = '"Whatever!" she replied.'  // ignored</code></pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p><a href="https://eslint.org">eslint</a> - Rule <a href="https://eslint.org/docs/latest/rules/quotes">quotes</a></p>
</li>
</ul>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Use string literals quotes consistently</p>
</div>
</div>
<div class="sect2">
<h3 id="_parameters">Parameters</h3>
<div class="sidebarblock">
<div class="content">
<div class="title">singleQuotes</div>
<div class="listingblock">
<div class="content">
<pre>true</pre>
</div>
</div>
<div class="paragraph">
<p>Set to true to require single quotes, false for double quotes.</p>
</div>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_1_nov_2016_170409_elena_vilchik_wrote">on 1 Nov 2016, 17:04:09 Elena Vilchik wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] Could you please add exceptions to this rule (they are both implemented already)?</p>
</div>
<div class="ulist">
<ul>
<li>
<p>if string literal contains <code>'</code> (SONARJS-796)</p>
</li>
<li>
<p>if string literal is part of JSX (SONARJS-681)
Thanks!</p>
</li>
</ul>
</div>
</div>
<div class="sect2">
<h3 id="_on_1_nov_2016_182118_ann_campbell_wrote">on 1 Nov 2016, 18:21:18 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>Done [~elena.vilchik]</p>
</div>
</div>
</div>
</div>