<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This reports an issue when ternary operators or if statements are used for null-safe default assignments, or when ternary expressions have identical boolean and true expressions that can be simplified.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>The Elvis operator (<code>?:</code>) is a Groovy-specific feature that provides a more concise and readable way to handle null-safe operations and default value assignments.</p>
</div>
<div class="paragraph">
<p>When you use a ternary operator like <code>user.name ? user.name : 'Anonymous'</code>, you&#8217;re duplicating the expression <code>user.name</code>. This duplication creates several problems:</p>
</div>
<div class="ulist">
<ul>
<li>
<p><strong>Code verbosity</strong>: The same expression appears twice, making the code longer than necessary</p>
</li>
<li>
<p><strong>Maintenance risk</strong>: If you need to change the expression (e.g., from <code>user.name</code> to <code>user.displayName</code>), you must remember to update it in both places</p>
</li>
<li>
<p><strong>Reduced readability</strong>: The duplicated expression makes it harder to quickly understand the intent</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>Similarly, if statements that check for null or falsy values and assign defaults can be verbose:</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-groovy" data-lang="groovy">if (!x) {
    x = 'some value'
}</code></pre>
</div>
</div>
<div class="paragraph">
<p>The Elvis operator eliminates these issues by providing a clean, single-expression syntax that clearly expresses the intent: "use this value if it&#8217;s truthy, otherwise use the default."</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="paragraph">
<p>The duplicated expressions in ternary operators create a higher risk of introducing bugs during refactoring, as developers might forget to update both occurrences of the same expression.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Replace ternary operators where the condition and true branch are identical with the Elvis operator.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-groovy" data-lang="groovy">displayName = user.name ? user.name : 'Anonymous' // Noncompliant</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-groovy" data-lang="groovy">displayName = user.name ?: 'Anonymous'</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Groovy documentation - <a href="https://groovy-lang.org/operators.html#_elvis_operator">Elvis operator</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>