<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>A cleanly coded web application will have a  clear separation of concerns, with business logic in the <code>@Service</code> layer, and communication with other systems in the data access layer.</p>
</div>
<div class="paragraph">
<p>To help enforce such a separation of concerns, this rule raises an issue when a <code>@Service</code> class has <code>RestTemplate</code>, <code>JmsTemplate</code>, <code>WebServiceTemplate</code>, <code>JdbcTemplate</code>, or <code>DataSource</code> member.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">@Service ("greetingmanager")
public class GreetingManagerImpl implements GreetingManager
{
    @Autowired
    DataSource ds;  // Noncompliant</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">@Service ("greetingmanager")
public class GreetingManagerImpl implements GreetingManager
{
    @Autowired
    GreetingDao gdao;  // DataSource and its use have been moved here</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Move this "xxTypexx" member and its use to a DAO class.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>member declaration</p>
</div>
</div>
</div>
</div>