<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>It is undefined behavior if the '\' character is used between <code>&lt;</code> and <code>&gt;</code> delimiters or between the <code>"</code> delimiters in a header name preprocessing token.</p>
</div>
<div class="paragraph">
<p>Note that this rule is only advisory, since some environments use '\' as a file name delimiter. However, compilers for those environments often support the use of '/' in <code>#include</code> directives.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-text" data-lang="text">#include "foo\bar.h" /* Noncompliant */</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-text" data-lang="text">#include "foo/bar.h" /* Compliant */</code></pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p>MISRA C&#43;&#43;:2008, 16-2-5 - The \ character should not occur in a header file name.</p>
</li>
</ul>
</div>
<hr>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_31_mar_2015_190607_evgeny_mandrikov_wrote">on 31 Mar 2015, 19:06:07 Evgeny Mandrikov wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] implementation seems more complete (SQALE, description) than this spec.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_6_apr_2015_141515_evgeny_mandrikov_wrote">on 6 Apr 2015, 14:15:15 Evgeny Mandrikov wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] I&#8217;m wondering if SQALE is missing on purpose or not?</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_8_apr_2015_161834_ann_campbell_wrote">on 8 Apr 2015, 16:18:34 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>Nope [~evgeny.mandrikov]. Just an oversight</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_13_apr_2015_192617_evgeny_mandrikov_wrote">on 13 Apr 2015, 19:26:17 Evgeny Mandrikov wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] looks like this rule completely covered by <a data-rspec-id="S955" class="rspec-auto-link">RSPEC-955</a>, so what do you think about CPP-1147?</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_20_apr_2015_140327_ann_campbell_wrote">on 20 Apr 2015, 14:03:27 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>deprecation added [~evgeny.mandrikov]</p>
</div>
</div>
</div>
</div>