The top-most level of an Azure Function code should include a try/catch block to capture and log errors so you can monitor the health of the application effectively.

// If you want to factorize the description uncomment the following line and create the file.
//include::../description.adoc[]

== Noncompliant Code Example

[source,csharp]
----
public static async Task RunAsync([EventHubTrigger(eventHubName: "events", Connection = "EventHub")] EventData[] events)
{
    // Noncompliant
    foreach (var event in events)
    {
        // Some code
    }
}
----

== Compliant Solution

[source,csharp]
----
public static async Task RunAsync([EventHubTrigger(eventHubName: "events", Connection = "EventHub")] EventData[] events, TraceWriter log)
{
    foreach (var event in events)
    {
        try
        {
            // Some code
        }
        catch (Exception ex)
        {
            log.Info($"Caught exception: {ex.Message}");
        }
    }
}
----

== See

* https://docs.microsoft.com/en-us/azure/azure-functions/functions-bindings-error-pages?tabs=csharp[Azure Functions error handling and retries]