== How to fix it in API Gateway

=== Code examples

These code samples illustrate how to fix this issue in both APIGateway and
ApiGatewayV2.

==== Noncompliant code example

[source,cloudformation,diff-id=1,diff-type=noncompliant]
----
AWSTemplateFormatVersion: '2010-09-09'
Resources:
  CustomApi:
    Type: AWS::ApiGateway::DomainName
    Properties:
      SecurityPolicy: "TLS_1_0"  # Noncompliant
----

==== Compliant solution

[source,cloudformation,diff-id=1,diff-type=compliant]
----
AWSTemplateFormatVersion: '2010-09-09'
Resources:
  CustomApi:
    Type: AWS::ApiGateway::DomainName
    Properties:
      SecurityPolicy: "TLS_1_2"
----

=== How does this work?

include::../../common/fix/fix.adoc[]
