== How to fix it in OpenSearch

=== Code examples

==== Noncompliant code example

[source,cloudformation,diff-id=21,diff-type=noncompliant]
----
AWSTemplateFormatVersion: 2010-09-09
Resources:
  Example:
    Type: AWS::OpenSearchService::Domain
    Properties:
      DomainName: example
      DomainEndpointOptions:
        EnforceHTTPS: true
        TLSSecurityPolicy: "Policy-Min-TLS-1-0-2019-07"  # Noncompliant
----

==== Compliant solution

[source,cloudformation,diff-id=21,diff-type=compliant]
----
AWSTemplateFormatVersion: 2010-09-09
Resources:
  Example:
    Type: AWS::OpenSearchService::Domain
    Properties:
      DomainName: example
      DomainEndpointOptions:
        EnforceHTTPS: true
        TLSSecurityPolicy: "Policy-Min-TLS-1-2-2019-07"
----

=== How does this work?

include::../../common/fix/fix.adoc[]

