Shared coding conventions allow teams to collaborate efficiently. This rule checks that file-code names conform to a specified regular expression.


== Noncompliant Code Example

Given a pattern of ``++FC-.*++``:

----
 INPUT-OUTPUT SECTION.
 FILE-CONTROL.

     SELECT Y27MVTS       ASSIGN  TO     S1        >Noncompliant; S1 doesn't match "FC-.*" pattern
                          FILE STATUS IS S1.
----


== Compliant Solution

----
 INPUT-OUTPUT SECTION.
 FILE-CONTROL.

     SELECT Y27MVTS     ASSIGN  TO     FC-S1     >OK as FC-S1 matches "FC-.*" pattern
                          FILE STATUS IS FS-S1.
----

