== Why is this an issue?

Promises need to be resolved or awaited to return the expected value, otherwise, they return the promise object.

[.underline]#*Unresolved promises:*#

Forgetting to await a promise is a frequent mistake. There are places where the use of a promise object is confusing or unclear because the developer forgot to resolve it.

This rule forbids returning promises where another type is expected such as in:

* conditionals
* void returns
* spread operators

[.underline]#*Missing error handling:*#

Promises that aren't resolved will finish running at an unexpected time. When they do not contain error handling and they raise an exception, the associated call stack will be wrong.

=== What is the potential impact?

Using a promise instead of its resolved value can have unexpected results leading to bugs.

* In conditionals, it will always return a truthy value.
* In places where the expected type is void, returning a promise is often a mistake.
* Using the spread operator on a promise will raise an exception.

The executor function of a promise can also be an async function. However, this usually denotes a mistake:

* If an async executor function throws an error, the error won't cause the created promise to reject and will be lost. Therefore, this could make it difficult to debug and handle runtime errors.
* If a promise executor function is using `await`, this means that it's not necessary to use the `Promise` constructor, or the scope of the Promise constructor can be reduced.

=== Exceptions

This rule can be ignored for promises that you know will always resolve like timers.

[source,javascript]
----
await new Promise(resolve => time.setTimeout(1000));
----

== How to fix it

If you mistakenly treated a promise as its resolved value, you can ensure it is properly resolved by using await or resolve on the promise. In some cases, you may need to use an "immediately invoked function expression" (IIFE):

[source,javascript]
----
(async function foo() {
  const result = await bar();
  // work with result
})();
----

If you forgot to handle the promise rejection, you should handle it either with `.catch(error => {/*...*/})` or `.then(result => {}, error => {})`.

//== How to fix it in FRAMEWORK NAME

=== Code examples

==== Noncompliant code example

[source,javascript,diff-id=1,diff-type=noncompliant]
----
const promise = new Promise((resolve, reject) => {
  // ...
  resolve(false)
});
if (promise) {
  // ...
}
----

==== Compliant solution

[source,javascript,diff-id=1,diff-type=compliant]
----
const promise = new Promise((resolve, reject) => {
  // ...
  resolve(false)
});
if (await promise) {
  // ...
}
----

==== Noncompliant code example

[source,javascript,diff-id=2,diff-type=noncompliant]
----
const p = new Promise(async (resolve, reject) => {
  doSomething('Hey, there!', function(error, result) {
    if (error) {
      reject(error);
      return;
    }
    await saveResult(result)
    resolve(result);
  });
});

await p;
----

==== Compliant solution

[source,javascript,diff-id=2,diff-type=compliant]
----
const p = new Promise((resolve, reject) => {
  doSomething('Hey, there!', function(error, result) {
    if (error) {
      reject(error);
      return;
    }
    resolve(result);
  });
});

const result = await p;
await saveResult(result);
----

==== Noncompliant code example

[source,javascript,diff-id=3,diff-type=noncompliant]
----
apiCalls.forEach(async (apiCall) => {
  await apiCall.send();
});
----

==== Compliant solution

[source,javascript,diff-id=3,diff-type=compliant]
----
for (const apiCall of apiCalls) {
  await apiCall.send();
}
----

==== Noncompliant code example

[source,javascript,diff-id=4,diff-type=noncompliant]
----
fetch('https://api.com/entries')
  .then(result => {
    // ...
  });
----

==== Compliant solution

[source,javascript,diff-id=4,diff-type=compliant]
----
fetch('https://api.com/entries')
  .then(result => {
    // ...
  })
  .catch(error => {
    // ...
  });
----

=== How does this work?

In JavaScript, a promise is a mechanism to perform tasks asynchronously.
To this end, the language provides the `Promise` object which represents the eventual completion or
failure of an asynchronous operation and its resulting value.
A promise can be created with the `Promise` constructor accepting an executor function as an argument,
which has `resolve` and `reject` parameters that are invoked when the promise completes or fails.

The logic of the promise is executed when it is called, however, its result is obtained only when the promise is resolved or awaited.

//=== Pitfalls
//=== Going the extra mile

== Resources

=== Documentation
* https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise[MDN Promise]
* https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Using_promises[MDN Using promises]
* https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/async_function[MDN Async function]
* https://developer.mozilla.org/en-US/docs/Glossary/IIFE[MDN IIFE]

//=== Articles & blog posts
//=== Conference presentations
//=== Standards
