This is an issue when using Rails numericality validations without explicitly specifying `allow_nil` or `allow_blank` options.

== Why is this an issue?

Rails numericality validations reject nil values by default, even when presence validation is not specified. This behavior can be surprising and lead to unexpected validation failures.

When a field with numericality validation is left empty (nil), Rails will raise a validation error saying the field "is not a number". This happens because the numericality validator tries to convert nil to a number and fails.

This implicit behavior creates several problems:

* *Unexpected validation failures*: Fields that seem optional will cause validation errors when left empty
* *Unclear code intent*: Other developers cannot tell if nil values should be allowed or rejected
* *Runtime errors*: Applications may fail when processing forms with empty numeric fields
* *Poor user experience*: Users receive confusing error messages about fields they intentionally left blank

The solution is to always be explicit about how numericality validations should handle nil and blank values. This makes the code's intent clear and prevents unexpected behavior.

=== What is the potential impact?

Applications may experience unexpected validation failures when users leave numeric fields empty. This can lead to poor user experience, failed form submissions, and runtime errors in production. The implicit behavior also makes code harder to understand and maintain.

== How to fix it

Add `allow_nil: true` to explicitly allow nil values, or `allow_blank: true` to allow both nil and empty strings. Choose based on whether you want to accept empty form fields.

=== Code examples

==== Noncompliant code example

[source,ruby,diff-id=1,diff-type=noncompliant]
----
class Product < ApplicationRecord
  validates :price, numericality: true # Noncompliant
  validates :rating, numericality: { greater_than: 0 } # Noncompliant
  validates_numericality_of :score # Noncompliant
end
----

==== Compliant solution

[source,ruby,diff-id=1,diff-type=compliant]
----
class Product < ApplicationRecord
  validates :price, numericality: true, allow_nil: true
  validates :rating, numericality: { greater_than: 0, allow_blank: true }
  validates :score, numericality: { allow_nil: true, greater_than_or_equal_to: 0 }
end
----

== Resources

=== Documentation

 * Rails Validation Helpers - numericality - https://guides.rubyonrails.org/active_record_validations.html#numericality[Official Rails documentation on numericality validation options]

 * Rails API - validates_numericality_of - https://api.rubyonrails.org/classes/ActiveModel/Validations/HelperMethods.html#method-i-validates_numericality_of[API documentation for numericality validation helper methods]
