<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Using checked exceptions forces method callers to deal with errors, either by propagating them or by handling them. Throwing exceptions makes them fully part of the API of the method.</p>
</div>
<div class="paragraph">
<p>But to keep the complexity for callers reasonable, methods should not throw more than one kind of checked exception.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">public void delete() throws IOException, SQLException {      // Noncompliant
  /* ... */
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">public void delete() throws SomeApplicationLevelException {
  /* ... */
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_exceptions">Exceptions</h3>
<div class="paragraph">
<p>Overriding methods are not checked by this rule and are allowed to throw several checked exceptions.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Refactor this method to throw at most one checked exception instead of: {Ex1, Ex2, Ex3, etc.}</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_29_jul_2013_152034_freddy_mallet_wrote">on 29 Jul 2013, 15:20:34 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>Is implemented by http://jira.codehaus.org/browse/SONARJAVA-248</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_29_jul_2013_234851_ann_campbell_wrote">on 29 Jul 2013, 23:48:51 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>There&#8217;s no compliant example</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_31_jul_2013_162416_dinesh_bolkensteyn_wrote">on 31 Jul 2013, 16:24:16 Dinesh Bolkensteyn wrote:</h3>
<div class="paragraph">
<p>'quickly' is a magic filler word ;-)</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_31_jul_2013_171016_dinesh_bolkensteyn_wrote">on 31 Jul 2013, 17:10:16 Dinesh Bolkensteyn wrote:</h3>
<div class="paragraph">
<p>\[~freddy.mallet]</p>
</div>
<div class="paragraph">
<p>It seems that the checkstyle rule verifies the number of throws <strong>statements</strong>?</p>
</div>
<div class="paragraph">
<p>Are you sure this is a replacement?</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_21_aug_2015_151703_ann_campbell_wrote">on 21 Aug 2015, 15:17:03 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>FYI [~tamas.vajk] I&#8217;ve set C# to targeted on this rule</p>
</div>
</div>
</div>
</div>