<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Nullability annotations in Java are used to indicate whether a variable or parameter can be assigned a null value or not.
These annotations help to prevent Null Pointer Exceptions and improve the reliability of code.</p>
</div>
<div class="paragraph">
<p>Redundant nullability annotations can clutter the code and make it harder to read and understand.
When a nullability annotation is already implied by the context or by other annotations, explicitly adding it again only adds noise and makes the code less clear.</p>
</div>
<div class="paragraph">
<p>Removing them improves code readability, maintainability, reduces the risk of inconsistencies, and ensures that the remaining annotations carry meaningful information.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="paragraph">
<p>JSpecify code example:</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">@NullMarked
class MyClass {

  public void method(@NonNull Object o) { // Noncompliant: @NonNull is redundant here
    // ...
  }
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">@NullMarked
class MyClass {

  public void method(Object o) {
    // ...
  }
}</code></pre>
</div>
</div>
</div>
</div>
</div>