<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>It is both suspicious and unusual to make direct assignments to super-class fields from a class constructor. Choosing to make such direct assignments, rather than invoking the <code>super</code> constructor that sets them, potentially bypasses important initialization.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-text" data-lang="text">public class Parent {

  protected int start;
  public Parent() {
    //...
  }

  public Parent(int start) {
    this.start = start;
  }
}

public class Child extends Parent {

  public Child(int start) {
    this.start = start;  // Noncompliant
  }
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-text" data-lang="text">public class Parent {

  protected int start;
  public Parent() {
    //...
  }

  public Parent(int start) {
    this.start = start;
  }
}

public class Child extends Parent {

  public Child(int start) {
    super(start);
  }
}</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_20_jan_2016_021050_freddy_mallet_wrote">on 20 Jan 2016, 02:10:50 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>Difficult to understand the purpose of this rule as if a parent field is accessible from a child why it should be a bad pattern to access to this parent field from a child constructor ?</p>
</div>
</div>
</div>
</div>