<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>By contract, chaining the 'Address of' operator <code>&amp;</code> with the 'Indirection' operator <code>*</code> results in a return to the initial value. Thus, such combinations are confusing at best, and bugs at worst.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">int *ptr = ...;
int *result1 = &amp;(*ptr); //Noncompliant
int *result2 = &amp;*ptr; //Noncompliant

int value = 4;
int result3 = *(&amp;value); //Noncompliant
int result4 = *&amp;value; //Noncompliant</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">int *ptr = ...;
int *result1 = ptr;
int *result2 = ptr;

int value = 4;
int result3 = value;
int result4 = value;</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_exceptions">Exceptions</h3>
<div class="paragraph">
<p>No issue is raised when the <code>*</code> or <code>&amp;</code> operators are overloaded or when both operators are not located in the same piece of code (one being generated by a macro expansion and the other one located in the main source code for instance).</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove this useless sequence of pointer operators: "xxx".</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>The sequence of pointer operators</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_relates_to_s2761">relates to: <a data-rspec-id="S2761" class="rspec-auto-link">S2761</a></h3>

</div>
<div class="sect2">
<h3 id="_on_19_jan_2016_110848_alban_auzeill_wrote">on 19 Jan 2016, 11:08:48 Alban Auzeill wrote:</h3>
<div class="paragraph">
<p>About the Labels: "bug"</p>
</div>
<div class="paragraph">
<p>I can&#8217;t find one case where it can be a bug. In my view, it&#8217;s more: brain-overload and perhaps performance</p>
</div>
<div class="paragraph">
<p>About "SQALE Characteristic": Reliability - Logic related reliability</p>
</div>
<div class="paragraph">
<p>In my view it&#8217;s more: understandability or readability</p>
</div>
</div>
</div>
</div>