== Why is this an issue?

In Dart type aliases can be declared via function type aliases `typedef void F()` or generic function type aliases  `typedef F = void Function()`. Generic function type aliases can be parametrized `typedef Compare<T> = int Function(T a, T b);`. Function type aliases miss this feature.
While it's not always needed to have them parametrized, for the sake of readability it's recommended to use consistent way of declaring type aliases. Thus, generic function type aliases should be preferred.

=== Noncompliant code example

[source,dart]
----
typedef void F(int a);
----


=== Compliant solution

[source,dart]
----
typedef F = void Function(int a)
----


== Resources

* Dart Docs - https://dart.dev/tools/linter-rules/prefer_generic_function_type_aliases[Dart Linter rule - prefer_generic_function_type_aliases]
* Dart Docs - https://dart.dev/language/typedefs[Typedefs]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

* Use the generic function type syntax in 'typedef's.

=== Highlighting

The `typedef` name: e.g. `F` in `typedef void F();` 

'''
== Comments And Links
(visible only on this page)

endif::env-github,rspecator-view[]

