This rule raises an issue when a method parameter has a default value that is a non-empty hash literal.

== Why is this an issue?

When you define a method parameter with a default hash containing multiple key-value pairs, Ruby completely replaces the entire default hash when any options are passed. This means that providing just one option will cause all other default values to be lost.

For example, if you have a method with `options = {age: 27, weight: 160, city: "New York"}` as the default, calling the method with `{age: 30}` will result in `options` being `{age: 30}` - the `weight` and `city` defaults are completely gone.

This behavior is often unexpected and can lead to bugs where methods don't work correctly when only some options are provided. The method will receive `nil` values for the missing keys, which can cause errors or incorrect behavior.

The root cause is that Ruby's parameter default values are evaluated once when the method is defined, and when you pass an argument, it completely replaces the default value rather than merging with it.

=== What is the potential impact?

This issue can lead to runtime errors when methods try to access missing hash keys, returning `nil` instead of expected default values. It can also cause incorrect application behavior when only partial configuration is provided, making the code unreliable and harder to debug.

== How to fix it

Use an empty hash as the default parameter and merge it with a defaults hash inside the method. This ensures that user-provided options are combined with defaults rather than replacing them entirely.

=== Code examples

==== Noncompliant code example

[source,ruby,diff-id=1,diff-type=noncompliant]
----
def configure_user(name, options = {age: 27, weight: 160, city: "New York"}) # Noncompliant
  age = options[:age]
  weight = options[:weight]
  city = options[:city]
  puts "#{name}: age=#{age}, weight=#{weight}, city=#{city}"
end
----

==== Compliant solution

[source,ruby,diff-id=1,diff-type=compliant]
----
def configure_user(name, options = {})
  defaults = {age: 27, weight: 160, city: "New York"}
  options = defaults.merge(options)
  
  age = options[:age]
  weight = options[:weight]
  city = options[:city]
  puts "#{name}: age=#{age}, weight=#{weight}, city=#{city}"
end
----

== Resources

=== Documentation

 * Ruby Hash#merge documentation - https://ruby-doc.org/core/Hash.html#method-i-merge[Official documentation for Hash#merge method used to combine hashes]

 * Ruby keyword arguments guide - https://ruby-doc.org/core/doc/syntax/methods_rdoc.html#label-Keyword+Arguments[Official Ruby documentation on keyword arguments as an alternative approach]
