This rule raises an issue when an incorrect pattern is provided to an `einops` operation.


== Why is this an issue?

The `einops` library provides a powerful and flexible way to manipulate tensors using the Einstein summation convention.
The `einops` uses a different convention than the https://ejenner.com/post/einsum/[traditional] one.
In particular, the axis names can be more than one letter long and are separated by spaces. 


== How to fix it
Correct the syntax of the `einops` operation by balancing the parentheses and following the convention.

=== Code examples

==== Noncompliant code example

[source,python,diff-id=1,diff-type=noncompliant]
----
from einops import rearrange
import torch

x = torch.randn(2, 3, 4, 5)
x2 = rearrange(x, 'b c h w -> b (c h w') # Noncompliant : the parentheses are not balanced

----

==== Compliant solution

[source,python,diff-id=1,diff-type=compliant]
----
from einops import rearrange
import torch

x = torch.randn(2, 3, 4, 5)
x2 = rearrange(x, 'b c h w -> b (c h w)')
----

== Resources
=== Documentation

* `einops` documentation - https://einops.rocks/1-einops-basics/#welcome-to-einops-land[Einops basics]


ifdef::env-github,rspecator-view[]

(visible only on this page)

== Implementation specification 

Implement a parser inspired by the original `einops` one : https://github.com/arogozhnikov/einops/blob/6181e1e95dc58c00a3143c1726da1c6ee0463164/einops/parsing.py#L85

=== Message 

Primary: The syntax of the `einops` operation is incorrect.


=== Issue location

Primary : The pattern

=== Quickfix

No

endif::env-github,rspecator-view[]
