<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Labeled loops are useful, especially when the code is badly indented, to match the begin and end of each loop. This rule verifies that loop start and end labels match, when both are specified.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-sql" data-lang="sql">BEGIN
  LOOP
    EXIT;
  END LOOP; -- Compliant, this loop has no label at all

  &lt;&lt;myLoopLabel1&gt;&gt;
  LOOP
    EXIT;
  END LOOP; -- Compliant, this loop only has a start label

  LOOP
    EXIT;
  END LOOP myLoopLabel2; -- Compliant, this loop only has an end label

  &lt;&lt;myLoopLabel4&gt;&gt;
  LOOP
    EXIT;
  END LOOP myLoopLabel5; -- Noncompliant, label mismatch

  &lt;&lt;myLoopLabel6&gt;&gt;
  &lt;&lt;myLoopLabel7&gt;&gt;
  LOOP
    EXIT;
  END LOOP myLoopLabel7; -- Noncompliant, several start labels mismatch
END;
/</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-sql" data-lang="sql">BEGIN
  LOOP
    EXIT;
  END LOOP;

  &lt;&lt;myLoopLabel1&gt;&gt;
  LOOP
    EXIT;
  END LOOP;

  LOOP
    EXIT;
  END LOOP myLoopLabel2;

  &lt;&lt;myLoopLabel4&gt;&gt;
  LOOP
    EXIT;
  END LOOP myLoopLabel4;

  &lt;&lt;myLoopLabel7&gt;&gt;
  LOOP
    EXIT;
  END LOOP myLoopLabel7;
END;
/</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Change this label to "xxx".</p>
</div>
</div>
</div>
</div>