
Nullable value types can hold either a value or ``++Nothing++``. The value held in the nullable type can be accessed with the ``++Value++`` property, but ``++.Value++`` throws an ``++InvalidOperationException++`` when the value is ``++Nothing++``. To avoid the exception, a nullable type should always be tested before ``++.Value++`` is accessed.

== Noncompliant Code Example

[source,vbnet]
----
Dim nullable As Integer? = Nothing
...
UseValue(nullable.Value)  'Noncompliant
----

== Compliant Solution

[source,vbnet]
----
Dim nullable As Integer? = Nothing
...
If nullable.HasValue Then
  UseValue(nullable.Value)
End If
----

or

[source,vbnet]
----
Dim nullable As Integer? = Nothing
...
If nullable <> Nothing Then
  UseValue(nullable.Value)
End If
----

include::../see.adoc[]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
