
Nullable value types can hold either a value or `Nothing`. The value stored in the nullable type can be accessed with the `Value` property or by casting it to the underlying type. Still, both operations throw an `InvalidOperationException` when the value is `Nothing`. A nullable type should always be tested before `Value` is accessed to avoid the exception.

== Noncompliant Code Example

[source,vbnet]
----
Dim nullable As Integer? = Nothing
Console.WriteLine(nullable.Value)
----
or
[source,vbnet]
----
Dim nullable As Integer? = Nothing
Console.WriteLine(CType(nullable, Integer))
----

== Compliant Solution

[source,vbnet]
----
Dim nullable As Integer? = Nothing
If nullable.HasValue Then
  Console.WriteLine(nullable.Value)
  Console.WriteLine(CType(nullable, Integer))
End If
----

or

[source,vbnet]
----
Dim nullable As Integer? = Nothing
If nullable <> Nothing Then
  Console.WriteLine(nullable.Value)
  Console.WriteLine(CType(nullable, Integer))
End If
----

include::../see.adoc[]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
