include::../description.adoc[]

== Noncompliant Code Example

----
$url = $_GET["url"];
$resp = file_get_contents($url); // Noncompliant
// ...
----

== Compliant Solution

----
$whitelist = array(
  "https://www.sonarsource.com"
);
$url = $_GET["url"];
if (in_array($url, $whitelist)) {
  $resp = file_get_contents($url);
  // ...
}
----

include::../see.adoc[]
ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../highlighting.adoc[]

endif::env-github,rspecator-view[]
