include::../rule.adoc[]

== Noncompliant Code Example

----
AWSTemplateFormatVersion: 2010-09-09
Resources:
  Example:
    Type: AWS::OpenSearchService::Domain
    Properties:
      DomainName: example
      NodeToNodeEncryptionOptions:
        Enabled: false
----

== Compliant Solution

----
AWSTemplateFormatVersion: 2010-09-09
Resources:
  Example:
    Type: AWS::OpenSearchService::Domain
    Properties:
      DomainName: example
      NodeToNodeEncryptionOptions:
        Enabled: true
----

include::../see.adoc[]


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::highlighting.adoc[]

endif::env-github,rspecator-view[]