This rule raises an issue when the walrus operator is used in a way which makes the code confusing, as described in https://www.python.org/dev/peps/pep-0572/#exceptional-cases[PEP 572].

== Why is this an issue?

The https://www.python.org/dev/peps/pep-0572[walrus operator] ``++:=++`` (also known as "assignment expression") should be used with caution as it can easily make code more difficult to understand and thus maintain. In such case it is advised to refactor the code and use an assignment statement (i.e. ``++=++``) instead.

Reasons why it is better to avoid using the walrus operator in Python:

* *Readability:* The walrus operator can lead to more complex and nested expressions, which might reduce the readability of the code, especially for developers who are not familiar with this feature.

* *Code Maintainability:* The use of the walrus operator might make the code harder to maintain and debug, as it can introduce side effects and modify variable values within an expression.

* *Overuse:* It's possible for some developers to overuse the walrus operator, leading to code that is difficult to understand and maintain. Like any feature in a programming language, it's essential to use it judiciously and only when it genuinely improves the code's clarity.

* *Compatibility:* If you are working on projects that need to be compatible with older versions of Python (before 3.8), you should avoid using the walrus operator, as it won't be available in those versions.

* *Alternative Solutions:* In many cases, you can achieve the same result without using the walrus operator, using more straightforward and conventional Python constructs. Opting for familiar constructs might be more beneficial for the broader Python community's understanding and collaboration.

=== Noncompliant code example

[source,python,diff-id=1,diff-type=noncompliant]
----
(v := f(p))  # Noncompliant
v0 = (v1 := f(p))  # Noncompliant

func(a=(b := f(p)))  # Noncompliant
func(a := f(p), b=2)  # Noncompliant
def func(param=(p := 21)):  # Noncompliant
    pass

def func(param: (p := 21) = 3):  # Noncompliant
    pass

f'{(x:=10)}'  # Noncompliant
----


=== Compliant solution

[source,python,diff-id=1,diff-type=compliant]
----
v = f(p)
v0 = v1 = f(p)

value = f(p)
func(a=value)
func(value, b=2)
def func(param=21):
    p = 21

p = 21
def func(param: p = 3):
    pass

x = 10
f'{x}' 
----


== Resources

=== Documentation

* Python Documentation - https://www.python.org/dev/peps/pep-0572[PEP 572 â€“ Assignment Expressions]
* Python Documentation - https://www.python.org/dev/peps/pep-0572/#exceptional-cases[PEP 572 - Assignment Expressions - Exceptional cases]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

* Use an assignment statement ("=") instead; ":=" operator is confusing in this context
* Move this assignment out of the f-string; ":=" operator is confusing in this context


=== Highlighting

The := operator


endif::env-github,rspecator-view[]
