<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>The <code>Files.exists</code> method has noticeably poor performance in JDK 8, and can slow an application significantly when used to check files that don&#8217;t actually exist.</p>
</div>
<div class="paragraph">
<p>The same goes for <code>Files.notExists</code>, <code>Files.isDirectory</code> and <code>Files.isRegularFile</code> from <code>java.nio.file</code> package.</p>
</div>
<div class="paragraph">
<p><strong>Note</strong> that this rule is automatically disabled when the project&#8217;s <code>sonar.java.source</code> is not 8.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">Path myPath;
if(java.nio.file.Files.exists(myPath)) {  // Noncompliant
 // do something
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">Path myPath;
if(myPath.toFile().exists())) {
 // do something
}</code></pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p><a href="https://bugs.openjdk.java.net/browse/JDK-8153414">https://bugs.openjdk.java.net/browse/JDK-8153414</a></p>
</li>
<li>
<p><a href="https://bugs.openjdk.java.net/browse/JDK-8154077">https://bugs.openjdk.java.net/browse/JDK-8154077</a></p>
</li>
</ul>
</div>
<hr>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Replace this with a call to the "toFile().exists()" method</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p><code>Files.exists</code></p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_22_sep_2016_201228_ann_campbell_wrote">on 22 Sep 2016, 20:12:28 Ann Campbell wrote:</h3>
<div class="paragraph">
<p><a href="https://bugs.openjdk.java.net/browse/JDK-8153414" class="bare">https://bugs.openjdk.java.net/browse/JDK-8153414</a></p>
</div>
<div class="paragraph">
<p><a href="https://bugs.openjdk.java.net/browse/JDK-8154077" class="bare">https://bugs.openjdk.java.net/browse/JDK-8154077</a></p>
</div>
</div>
<div class="sect2">
<h3 id="_on_15_nov_2019_123434_tibor_blenessy_wrote">on 15 Nov 2019, 12:34:34 Tibor Blenessy wrote:</h3>
<div class="paragraph">
<p>Removed from default profile because Java 8 is becoming quite old and using Files.exists is in general better approach when using Path API. See also https://community.sonarsource.com/t/sonar-should-flag-calls-to-path-tofile/16664/2</p>
</div>
</div>
</div>
</div>