=== is related to: S808

=== on 21 Oct 2014, 18:22:10 Ann Campbell wrote:
\[~samuel.mercier] please 

* combine the code samples in the Noncompliant Code Example
* move the code commentary into the code in the form of comments if needed
* reserve "See also" for something that's relevant but tangential
* use h3 for See also, rather than h2
* include ISO references in the main "See" list.

In fact, I would say that if you leave the ISO citations in the text, you don't need to include them in the references (See) section, and conversely if you want to list them in the references you should remove them from the text.

