== Why is this an issue?

Name space and scope are different. This rule is not concerned with scope.


ISO C defines a number of different name spaces (see ISO 9899:1990 6.1.2.3). It is technically possible to use the same name in separate name spaces to represent completely different items. However this practice is deprecated because of the confusion it can cause, and so names should not be reused, even in separate name spaces.


=== Noncompliant code example

[source,cpp]
----
struct { int16_t key; int16_t value; } record;
int16_t value; // Noncompliant, 2nd use of "value"
record.key = 1;
value = 0; // should have been record.value

struct device_q { struct device_q *next; /* ... */ } // Compliant, since "next" must be prefixed.
devices[N_DEVICES];
struct task_q { struct task_q *next; /* ... */ } // Compliant, since "next" must be prefixed.
tasks[N_TASKS];
devices[0].next = &devices[1];
tasks[0].next = &tasks[1];
----


=== Compliant solution

[source,cpp]
----
struct { int16_t key; int16_t value; } record;
int16_t val; // Compliant
record.key = 1;
value = 0; // Compile time error
----


== Resources

* MISRA C:2004, 5.6


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

The name "xxx" collides with a previous declaration and should be renamed.


'''
== Comments And Links
(visible only on this page)

=== is related to: S808

=== on 21 Oct 2014, 18:22:10 Ann Campbell wrote:
\[~samuel.mercier] please 

* combine the code samples in the Noncompliant Code Example
* move the code commentary into the code in the form of comments if needed
* reserve "See also" for something that's relevant but tangential
* use h3 for See also, rather than h2
* include ISO references in the main "See" list.

In fact, I would say that if you leave the ISO citations in the text, you don't need to include them in the references (See) section, and conversely if you want to list them in the references you should remove them from the text.

endif::env-github,rspecator-view[]
