== Why is this an issue?

Name space and scope are different. This rule is not concerned with scope.


ISO C defines a number of different name spaces (see ISO 9899:1990 6.1.2.3). It is technically possible to use the same name in separate name spaces to represent completely different items. However this practice is deprecated because of the confusion it can cause, and so names should not be reused, even in separate name spaces.


=== Noncompliant code example

[source,cpp]
----
struct { int16_t key; int16_t value; } record;
int16_t value; // Noncompliant, 2nd use of "value"
record.key = 1;
value = 0; // should have been record.value

struct device_q { struct device_q *next; /* ... */ } // Compliant, since "next" must be prefixed.
devices[N_DEVICES];
struct task_q { struct task_q *next; /* ... */ } // Compliant, since "next" must be prefixed.
tasks[N_TASKS];
devices[0].next = &devices[1];
tasks[0].next = &tasks[1];
----


=== Compliant solution

[source,cpp]
----
struct { int16_t key; int16_t value; } record;
int16_t val; // Compliant
record.key = 1;
value = 0; // Compile time error
----


== Resources

* MISRA C:2004, 5.6


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
