== Why is this an issue?

Duplication of selectors might indicate a copy-paste mistake. The rule detects the following kinds of duplications:

* within a list of selectors in a single rule set 
* for duplicated selectors in different rule sets within a single stylesheet. 


=== Noncompliant code example

[source,css]
----
.foo, .bar, .foo { ... }  /* Noncompliant */

.class1 { ... }
.class1 { ... }  /* Noncompliant */
----


=== Compliant solution

[source,css]
----
.foo, .bar { ... } 

.class1 { ... }
.class2 { ... }
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Remove this duplicated selector.


endif::env-github,rspecator-view[]
