== Why is this an issue?

Since Visual Studio 2010 SP1, the ``++ByVal++`` parameter modifier is implicitly applied, and therefore not required anymore. Removing it from your source code will improve readability.


=== Noncompliant code example

[source,vbnet]
----
Sub Foo(ByVal bar As String)
  ' ...
End Sub
----


=== Compliant solution

[source,vbnet]
----
Sub Foo(bar As String)
  ' ...
End Sub
----



ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Remove this redundant "ByVal" modifier


=== Highlighting

``++ByVal++``


'''
== Comments And Links
(visible only on this page)

=== on 26 Jan 2017, 13:51:17 Valeri Hristov wrote:
References:


Parameter Arrays - \https://msdn.microsoft.com/en-us/library/538f81ec.aspx

Parameter List - \https://msdn.microsoft.com/en-us/library/cbs7z96t.aspx (see also in this article Passing Mechanism in the Rules section)

endif::env-github,rspecator-view[]
