Since Visual Studio 2010 SP1, the ``++ByVal++`` parameter modifier is implicitly applied, and therefore not required anymore. Removing it from your source code will improve readability.


== Noncompliant Code Example

----
Sub Foo(ByVal bar As String)
  ' ...
End Sub
----


== Compliant Solution

----
Sub Foo(bar As String)
  ' ...
End Sub
----


