include::../description.adoc[]

== Noncompliant Code Example

With a maximum number of 4 parameters:

----
void doSomething(int param1, int param2, int param3, int param4, int param5) { 
  ...
}
----

== Compliant Solution

----
void doSomething(int param1, int param2, int param3, int param4) { 
  ...
}
----

== See

* https://github.com/isocpp/CppCoreGuidelines/blob/036324/CppCoreGuidelines.md#i23-keep-the-number-of-function-arguments-low[{cpp} Core Guidelines I.23]: Keep the number of function arguments low

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::parameters.adoc[]

include::highlighting.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
