== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

With a maximum number of 4 parameters:

[source,cpp]
----
void doSomething(int param1, int param2, int param3, int param4, int param5) { 
  ...
}
----

=== Compliant solution

[source,cpp]
----
void doSomething(int param1, int param2, int param3, int param4) { 
  ...
}
----

== Resources

* https://github.com/isocpp/CppCoreGuidelines/blob/036324/CppCoreGuidelines.md#i23-keep-the-number-of-function-arguments-low[{cpp} Core Guidelines I.23]: Keep the number of function arguments low

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

=== Parameters

.max
****
_INTEGER_

----
7
----

Maximum authorized number of parameters
****


=== Highlighting

* Primary: function name
* Additional: unauthorized parameters


'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
