Declaring multiple variables on one line is difficult to read.

== Noncompliant Code Example

----
class MyClass {

  private int a, b;

  public void method(){
    int c; int d;
  }
}
----

== Compliant Solution

----
class MyClass {

  private int a;
  private int b;

  public void method(){
    int c;
    int d;
  }
}
----

include::../see.adoc[]

ifdef::env-github,rspecator-view[]
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
