<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Types are declared in namespaces in order to prevent name collisions and as a way to organize them into the object hierarchy. Types that are defined outside any named namespace are in a global namespace that cannot be referenced in code.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-vbnet" data-lang="vbnet">Public Class Foo
End Class

Public Structure Bar
End Structure</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-vbnet" data-lang="vbnet">Namespace SomeSpace
    Public Class Foo
    End Class

    Public Structure Bar
    End Structure
End Namespace</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Move the type 'XXX' into a named namespace.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>The type identifier.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_relates_to_s997">relates to: <a data-rspec-id="S997" class="rspec-auto-link">S997</a></h3>

</div>
<div class="sect2">
<h3 id="_on_10_mar_2017_134243_ann_campbell_wrote">on 10 Mar 2017, 13:42:43 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~amaury.leve] from the description, this sounds like a bug&#8230;&#8203;?</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_10_mar_2017_154924_amaury_levé_wrote">on 10 Mar 2017, 15:49:24 Amaury Levé wrote:</h3>
<div class="paragraph">
<p>This rule has been implemented as a separate rule in order to define it as a <code>Bug</code>. See <a href="https://jira.sonarsource.com/browse/RSPEC-997">RSPEC-997</a> for the original rule.</p>
</div>
</div>
</div>
</div>