== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,pli]
----
foo: proc options(main);
  declare i fixed decimal init(42); put list (i); /* Noncompliant - there are two statements */
end;
----

=== Compliant solution

[source,pli]
----
foo: proc options(main);
  /* Compliant */
  declare i fixed decimal init(42);
  put list (i);
end;
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../highlighting.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
