<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when a private nested class is never used.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p><code>private</code> classes that are never used are dead code: unnecessary, inoperative code that should be removed. Cleaning out dead code decreases the size of the maintained codebase, making it easier to understand the program and preventing bugs from being introduced.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>

</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">public class TopLevel
{
  private class Nested {...} // Noncompliant: Nested is never used
}</code></pre>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">public class TopLevel
{
  void doSomething() {
    Nested a = new Nested();
    ...
  }
  private class Nested {...}
}</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove this unused private "XXXX" class.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>The name of the unused private class</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_relates_to_s1068">relates to: <a data-rspec-id="S1068" class="rspec-auto-link">S1068</a></h3>

</div>
<div class="sect2">
<h3 id="_relates_to_s1144">relates to: <a data-rspec-id="S1144" class="rspec-auto-link">S1144</a></h3>

</div>
<div class="sect2">
<h3 id="_on_21_jul_2017_130722_amaury_levé_wrote">on 21 Jul 2017, 13:07:22 Amaury Levé wrote:</h3>
<div class="paragraph">
<p>Covered by <a data-rspec-id="S1144" class="rspec-auto-link">S1144</a> for C#</p>
</div>
</div>
</div>
</div>