include::../description.adoc[]

== Noncompliant Code Example

[source,vbnet]
----
<ExcludeFromCodeCoverage>
Public Structure HashCode
    Public Shared Function Combine(Of T1, T2)(value1 As T1, value2 As T2) As Integer
        Dim hashCode = -1394996599
        hashCode = hashCode * -1521134295 + EqualityComparer(Of T1).[Default].GetHashCode(value1)
        hashCode = hashCode * -1521134295 + EqualityComparer(Of T2).[Default].GetHashCode(value2)
        Return hashCode
    End Function
End Structure
----

== Compliant Solution

[source,vbnet]
----
<ExcludeFromCodeCoverage(Justification:="Code generated by using the Visual Studio refactoring.")>
Public Structure HashCode
    Public Shared Function Combine(Of T1, T2)(value1 As T1, value2 As T2) As Integer
        Dim hashCode = -1394996599
        hashCode = hashCode * -1521134295 + EqualityComparer(Of T1).[Default].GetHashCode(value1)
        hashCode = hashCode * -1521134295 + EqualityComparer(Of T2).[Default].GetHashCode(value2)
        Return hashCode
End Structure
----

include::../see.adoc[]