== Why is this an issue?

COBOL programs that include ``++EXEC SQL INCLUDE++`` statements must be pre-processed with an IBM DB2 precompiler or coprocessor before compilation. The precompiler does not require a period after the closing ``++END-EXEC++``, but the coprocessor does. For portability, a period should always be used.


=== Noncompliant code example

[source,cobol]
----
EXEC SQL INCLUDE ... END-EXEC  *> Noncompliant; '.' missing
----


=== Compliant solution

[source,cobol]
----
EXEC SQL INCLUDE ... END-EXEC.
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

include::highlighting.adoc[]

'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
