== Why is this an issue?

Array literals are more compact than array creation expressions.


=== Noncompliant code example

[source,vbnet]
----
Module Module1
    Sub Main()
        Dim foo = New String() {"a", "b", "c"} ' Noncompliant
    End Sub
End Module
----


=== Compliant solution

[source,vbnet]
----
Module Module1
    Sub Main()
        Dim foo = {"a", "b", "c"}              ' Compliant
    End Sub
End Module
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Use an array literal here instead.


endif::env-github,rspecator-view[]
