include::../description.adoc[]

== Noncompliant Code Example

----
Connection conn = DriverManager.getConnection("jdbc:derby:memory:myDB;create=true", "login", "");
----

== Compliant Solution

----
String password = System.getProperty("database.password");
Connection conn = DriverManager.getConnection("jdbc:derby:memory:myDB;create=true", "login", password);
----

include::../see.adoc[]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::highlighting.adoc[]

'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
