<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>According to the Backbone.js docs</p>
</div>
<div class="quoteblock">
<blockquote>
<div class="paragraph">
<p>The <strong>changed</strong> property is the internal hash containing all the attributes that have changed since the last set. Please do not update <strong>changed</strong> directly since its state is internally maintained by set. A copy of <strong>changed</strong> can be acquired from changedAttributes.</p>
</div>
</blockquote>
</div>
<div class="paragraph">
<p>The <code>changed</code> property is involved in decisions about whether or not a collection should be resorted when it is updated. If you modify it manually, you can break the resorting of the collection.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-javascript" data-lang="javascript">myModel.changed = { myProperty: 1 }; // Non-compliant</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove this update of the "changed" property.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_20_may_2015_123141_linda_martin_wrote">on 20 May 2015, 12:31:41 Linda Martin wrote:</h3>
<div class="paragraph">
<p>OK!</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_1_nov_2019_162943_elena_vilchik_wrote">on 1 Nov 2019, 16:29:43 Elena Vilchik wrote:</h3>
<div class="paragraph">
<p>See https://github.com/SonarSource/SonarJS/issues/1698</p>
</div>
</div>
</div>
</div>