== Why is this an issue?

Using ``++count()++`` to test for emptiness works, but using ``++empty()++`` makes the code more readable and can be more performant

=== Noncompliant code example

[source,php]
----
if (count($a) > 0) { // Noncompliant
 echo $a[0];
}
----

=== Compliant solution

[source,php]
----
if (!empty($a)) {
  echo $a[0];
}
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
