<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>In Dart, there are no keywords like <code>public</code>, <code>protected</code>, and <code>private</code>. If an identifier starts with an underscore (_), it&#8217;s private to its library. So every type starting with underscore(<code>_T</code>) is considered private.</p>
</div>
<div class="paragraph">
<p>Public API of a library is designed to be used outside the library itself. And this API is supposed to be stable and not change often. Changes to the Public API will most likely break the code using it. This is why it is highly recommended to follow some standard processes of deprecation and prepare the users to the breaking change if the one is coming.</p>
</div>
<div class="paragraph">
<p>In case private types appear in public API, the users of the API are forced to depend on those private types too, which is highly discouraged. This is why it is very important to decouple the public API from the implementation. Internal implementation is a subject to change often and it shouldn&#8217;t affect the users of the public API.</p>
</div>
<div class="paragraph">
<p>To keep this decoupling in control, this rule reports if some private types leaked into public API.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Make sure your implementation details don&#8217;t leak into public API. In case it&#8217;s impossible to refactor the code to get rid of the leakage, consider changing the visibility of a private type and adding it to the public API.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-dart" data-lang="dart">_PrivateClass nonCompliantPublicFunction() { // returns private class
  _PrivateClass private = _PrivateClass();
  return private;
}

PublicClass nonCompliantPublicFunction2(_PrivateClass private) {  // takes an instance of a private type as a parameter
  private.runtimeType;
  return PublicClass();
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-dart" data-lang="dart">_PrivateClass _privateFunction() {
  _PrivateClass private = _PrivateClass();
  return private;
}

PublicClass publicFunction() {
  _PrivateClass private = _PrivateClass(); // it's okay to use private types inside the implementation as long as they don't leak
  return PublicClass();
}</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Dart Docs - <a href="https://dart.dev/tools/linter-rules/library_private_types_in_public_api">Dart Linter rule - library_private_types_in_public_api</a></p>
</li>
</ul>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="ulist">
<ul>
<li>
<p>Invalid use of a private type in a public API.</p>
</li>
</ul>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>Type usage</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
</div>
</div>