== How to fix it in Python Standard Library

=== Code examples

include::../../common/fix/code-rationale.adoc[]

==== Noncompliant code example

[source,python,diff-id=1,diff-type=noncompliant]
----
hash = crypt.crypt(password) # Noncompliant
----

==== Compliant solution

[source,python,diff-id=1,diff-type=compliant]
----
salt = crypt.mksalt(crypt.METHOD_SHA256)
hash = crypt.crypt(password, salt)
----

=== How does this work?

include::../../common/fix/salt.adoc[]

