This rule raises an issue when a `@Grab` annotation uses the verbose format with named parameters (`group`, `module`, `version`) but no additional parameters are specified.

== Why is this an issue?

The `@Grab` annotation in Groovy supports two formats for specifying dependencies: a verbose format with named parameters and a concise shorthand format using the Maven coordinate style.

The verbose format requires explicitly naming each parameter:
```
@Grab(group='org.springframework', module='spring-orm', version='5.2.8.RELEASE')
```

The shorthand format uses the familiar Maven coordinate pattern:
```
@Grab('org.springframework:spring-orm:5.2.8.RELEASE')
```

When only the basic dependency information (group, module, version) is needed, the shorthand notation offers several advantages:

* *Improved readability*: The shorthand format is more compact and easier to scan visually
* *Familiar syntax*: It follows the same pattern used in Maven, Gradle, and other build tools
* *Reduced verbosity*: Less code to write and maintain
* *Consistency*: Aligns with common practices in the Java ecosystem

The verbose format should be reserved for cases where additional parameters are needed, such as `classifier`, `transitive`, `changing`, or `conf` options.

== How to fix it

Replace the verbose @Grab annotation with the shorthand notation using the 'group:module:version' format.

=== Code examples

==== Noncompliant code example

[source,groovy,diff-id=1,diff-type=noncompliant]
----
@Grab(group='org.springframework', module='spring-orm', version='5.2.8.RELEASE') // Noncompliant
import org.springframework.jdbc.core.JdbcTemplate
----

==== Compliant solution

[source,groovy,diff-id=1,diff-type=compliant]
----
@Grab('org.springframework:spring-orm:5.2.8.RELEASE') // Compliant
import org.springframework.jdbc.core.JdbcTemplate
----

== Resources

=== Documentation

 * Groovy documentation - https://groovy-lang.org/grape.html[The Grape dependency manager]
