<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>If a local variable or a local function is declared but not used, it is dead code and should be removed. Doing so will improve maintainability because developers will not wonder what the variable or function is used for.</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="sect3">
<h4 id="_dead_code">Dead code</h4>
<div class="paragraph">
<p>An unused variable or local function usually occurs because some logic is no longer required after a code change. In that case, such code becomes unused and never executed.</p>
</div>
<div class="paragraph">
<p>Also, if you are writing code for the front-end, every unused variable or function remaining in your codebase is just extra bytes you have to send over the wire to your users. Unused code bloats your codebase unnecessarily and impacts the performance of your application.</p>
</div>
</div>
<div class="sect3">
<h4 id="_wrong_logic">Wrong logic</h4>
<div class="paragraph">
<p>It could happen that due to a bad copy-paste or autocompletion, the wrong variable is used, while the right one is only declared. In that case, the unused variable should be used instead of deleted from the codebase.</p>
</div>
</div>
<div class="sect3">
<h4 id="_memory_leaks">Memory leaks</h4>
<div class="paragraph">
<p>Finally, unused functions can also cause memory leaks. For example, an unused function can create a closure over a variable that would otherwise be released to the garbage collector.</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-javascript" data-lang="javascript">let theThing = null;
const replaceThing = function () {
  const originalThing = theThing;
  const unused = function () {
    if (originalThing) {
      console.log("hi");
    }
  };
  theThing = {
    longStr: new Array(1000000).join("*"),
    someMethod: function () {
      console.log(someMessage);
    },
  };
};
setInterval(replaceThing, 1000);</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Usually, the fix for this issue is straightforward, you just need to remove the unused variable declaration, or its name from the declaration statement if it is declared along with other variables.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-javascript" data-lang="javascript">function numberOfMinutes(hours) {
  var seconds = 0;   // seconds is never used
  return hours * 60;
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-javascript" data-lang="javascript">function numberOfMinutes(hours) {
  return hours * 60;
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_noncompliant_code_example_2">Noncompliant code example</h4>
<div class="paragraph">
<p>When an array destructuring is used and some element of the array is never referenced, one might simply remove it from the destructuring.</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-javascript" data-lang="javascript">const [_, params] = url.split(path);</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution_2">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-javascript" data-lang="javascript">const [, params] = url.split(path);</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>MDN web docs - <a href="https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment#ignoring_some_returned_values">Destructuring assignment / Ignoring some values</a></p>
</li>
</ul>
</div>
</div>
<div class="sect2">
<h3 id="_articles_blog_posts">Articles &amp; blog posts</h3>
<div class="ulist">
<ul>
<li>
<p><a href="https://www.sonarsource.com/blog/common-typescript-issues-no-3-unused-local-variables-and-functions/">Phil Nash, Common TypeScript Issues NÂº 3: unused local variables and functions</a></p>
</li>
<li>
<p><a href="https://blog.meteor.com/an-interesting-kind-of-javascript-memory-leak-8b47d2e7f156">
David Glasser, An interesting kind of JavaScript memory leak</a></p>
</li>
</ul>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove unused function 'xxx'.</p>
</div>
<div class="paragraph">
<p>Remove the declaration of the unused 'xxx' variable.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments and links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_is_duplicated_by_s896">is duplicated by: <a data-rspec-id="S896" class="rspec-auto-link">S896</a></h3>

</div>
<div class="sect2">
<h3 id="_relates_to_s895">relates to: <a data-rspec-id="S895" class="rspec-auto-link">S895</a></h3>

</div>
<div class="sect2">
<h3 id="_is_related_to_s1854">is related to: <a data-rspec-id="S1854" class="rspec-auto-link">S1854</a></h3>

</div>
<div class="sect2">
<h3 id="_on_2_dec_2013_165945_freddy_mallet_wrote">on 2 Dec 2013, 16:59:45 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>Is implemented by https://jira.codehaus.org/browse/SONARPLUGINS-3304 in Flex</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_3_dec_2013_220449_freddy_mallet_wrote">on 3 Dec 2013, 22:04:49 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>Is implemented by http://jira.codehaus.org/browse/SONARJAVA-392 in Java</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_19_feb_2014_084346_freddy_mallet_wrote">on 19 Feb 2014, 08:43:46 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>Is implemented by http://jira.codehaus.org/browse/SONARPLUGINS-3489 for PHP</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_15_feb_2018_161316_alban_auzeill_wrote">on 15 Feb 2018, 16:13:16 Alban Auzeill wrote:</h3>
<div class="paragraph">
<p>Irrelevant for TypeScript because it is covered by typescript&#8217;s --noUnusedLocals and --noUnusedParameters compiler options.</p>
</div>
<div class="paragraph">
<p>See [SonarTS/issues/67](https://github.com/SonarSource/SonarTS/issues/67)</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_7_mar_2018_172834_janos_gyerik_wrote">on 7 Mar 2018, 17:28:34 Janos Gyerik wrote:</h3>
<div class="paragraph">
<p>Irrelevant for Go: declared but unused variables raise compilation error.</p>
</div>
</div>
</div>
</div>