The complexity of an expression is defined by the number of ``AND`` and ``OR`` operators it contains.
A single expression's complexity should not become too high to keep the code readable.

== Noncompliant Code Example

With the default threshold value of 3

----
IF ((@a = 1 AND @b > 2) OR (@c <> 3 AND @d <= 4)) AND @e IS NULL
  ...
----
