== Why is this an issue?

The complexity of an expression is defined by the number of ``++AND++`` and ``++OR++`` operators it contains.

A single expression's complexity should not become too high to keep the code readable.

=== Noncompliant code example

With the default threshold value of 3

[source,sql]
----
IF ((@a = 1 AND @b > 2) OR (@c <> 3 AND @d <= 4)) AND @e IS NULL
  ...
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../parameters.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
