include::../common/why.adoc[]

include::../common/how.adoc[]

=== Code examples

==== Noncompliant code example

[source,html,diff-id=1,diff-type=noncompliant]
----
<button role="article">Click me!</button> <!-- Noncompliant; "button" is interactive, but "article" isn't -->
----

==== Compliant solution

[source,html,diff-id=1,diff-type=compliant]
----
<button role="button">Click me!</button>
----

include::../common/resources.adoc[]
