<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Having long <code>RUN</code> instruction on a single line reduce the visibility and make it harder to read and understand.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>In Dockerfiles, <code>RUN</code> instruction tend to be extensive and can rapidly become quite long, containing a lot of instruction with parameters and flags.<br>
In order to keep a good readability and help for future maintenance, it is important to split them into multiple lines.<br>
It is usually a good idea to group elements that belong together on the same lines.<br>
The usage of the operator <code>&amp;&amp;</code> allows to chain multiple operators. Also, the default operator <code>\</code> allows to continue the <code>RUN</code> instruction on the next line.<br>
Another way is to use the here documents format, which allows to define multiple instructions together without any chaining operator.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-docker" data-lang="docker">RUN apt-get update &amp;&amp; apt-get install -y package-bar package-baz package-foo package-eel</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-docker" data-lang="docker"># Split shell form
RUN apt-get update &amp;&amp; \
  apt-get install -y \
  package-bar \
  package-baz \
  package-foo \
  package-eel


# Heredoc form
RUN &lt;&lt;EOF
apt-get update
apt-get install -y \
    package-bar \
    package-baz \
    package-foo \
    package-eel
EOF</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Line is too long. Split it into multiple lines using backslash continuations.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>Highlight the whole line of the <code>RUN</code> instruction that is too long.</p>
</div>
</div>
<div class="sect2">
<h3 id="_parameters">Parameters</h3>
<div class="sidebarblock">
<div class="content">
<div class="title">maxLength</div>
<div class="paragraph">
<p><em>INT</em></p>
</div>
<div class="listingblock">
<div class="content">
<pre>120</pre>
</div>
</div>
<div class="paragraph">
<p>The maximum number of characters for a single-line RUN instruction, including the RUN keyword itself.</p>
</div>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Docker docs - <a href="https://docs.docker.com/build/building/best-practices/#run">Building best practices</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>