This rule raises an issue when Apex code lacks proper spacing around operators, after commas, or uses inconsistent indentation in nested blocks like if-else statements, loops, and try-catch blocks.

== Why is this an issue?

Consistent spacing and indentation are fundamental to code readability and maintainability. When code lacks proper formatting, it becomes difficult for developers to quickly understand the structure and logic.

Spacing around operators and after commas helps separate different parts of expressions, making them easier to parse visually. For example, `i=0;i<10;i++` is much harder to read than `i = 0; i < 10; i++`.

Proper indentation shows the hierarchical structure of nested code blocks. When if-else statements, loops, and other control structures are properly indented, developers can immediately see which statements belong to which block. This is especially important in complex nested structures where misaligned code can lead to confusion about the program flow.

Inconsistent formatting also makes code reviews more difficult, as reviewers spend time deciphering poorly formatted code instead of focusing on logic and potential issues. In team environments, consistent formatting standards help ensure that all developers can easily read and modify each other's code.

=== What is the potential impact?

Poor formatting reduces code readability and makes maintenance more difficult. Developers spend extra time understanding poorly formatted code, which slows down development and increases the likelihood of introducing bugs during modifications. In team environments, inconsistent formatting can lead to unnecessary merge conflicts and reduced collaboration efficiency.

== How to fix it

Add spaces around operators and after commas in expressions and statements.

=== Code examples

==== Noncompliant code example

[source,apex,diff-id=1,diff-type=noncompliant]
----
for(i=0;i<10;i++){
    System.debug("Value"+i);
}
----

==== Compliant solution

[source,apex,diff-id=1,diff-type=compliant]
----
for (i = 0; i < 10; i++) {
    System.debug("Value" + i);
}
----

== Resources

=== Documentation

 * Apex Code Layouts and Formatting Best Practices - https://www.apexhours.com/code-layouts-and-formatting-salesforce[Comprehensive guide on Apex code formatting standards including spacing, indentation, and layout best practices]
