<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p><strong>This rule was deprecated for Java before it was implemented.</strong></p>
</div>
<div class="quoteblock">
<blockquote>
<div class="paragraph">
<p>in Java the members are initialized to default values (i.e. zero and null), so the rule doesnâ€™t bring much value.</p>
</div>
</blockquote>
</div>
<div class="paragraph">
<p><a href="https://discuss.sonarsource.com/t/is-the-rule-s2107-implemented-planned-duplicate-for-java/15217/2">discuss post</a></p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Class members that are not assigned a default value and are not initialized in a constructor will be set to null by the compiler. Even if code exists to properly set those members, there is a risk that they will be dereferenced before it is called, resulting in a <code>NullPointerException</code>.</p>
</div>
<div class="paragraph">
<p>Because you cannot guarantee that such classes will always be used properly, class members should always be initialized.</p>
</div>
<div class="paragraph">
<p>This rule flags members which have no default value and which are left uninitialized by at least one class constructor, but which are unconditionally dereferenced somewhere in the code.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">public class Team {

  int limit = 30;
  List&lt;Player&gt; roster;  // Noncompliant; no default &amp; not initialized by constructor
  Person coach;

  public Team (Person coach) { // roster is left uninitialized
    this.coach = coach;
  }

  public void add(Player p) {
    if (roster == null) {
      roster = new ArrayList&lt;Player&gt;();
    }
    roster.add(p);
  }

  public boolean isFull() {  // NPE if called before add()
    return roster.size() &lt; limit;
  }
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">public class Team {

  int limit = 30;
  List&lt;Player&gt; roster = new ArrayList&lt;Player&gt;();
  Person coach;

  public Team (Person coach) {
    this.coach = coach;
  }

  public void add(Player p) {
    roster.add(p);
  }

  // ...</code></pre>
</div>
</div>
<div class="paragraph">
<p>or</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">public class Team {

  int limit = 30;
  List&lt;Player&gt; roster;
  Person coach;

  public Team (Person coach) {
    this.coach = coach;
    this.roster = new ArrayList&lt;Player&gt;();
  }

  public void add(Player p) {
    roster.add(p);
  }

  // ...</code></pre>
</div>
</div>
</div>
</div>
</div>