<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>In PHP, the <code>define</code> function allows you to define a named constant with a specific value, which cannot be changed later in the code.
Once a constant has been defined, it can be used throughout the entire script, including in function and class definitions.</p>
</div>
<div class="paragraph">
<p>Assigning a value to the same constant name using two or more <code>define</code> statements does not cause PHP to fail.
In such a case, PHP only issues a warning and ignores the second and further <code>define</code>.</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="paragraph">
<p>Using duplicate define statements can lead to unnecessary code that can be safely removed in the best case scenario.</p>
</div>
<div class="paragraph">
<p>It is important to note that in the worst case, duplicate define statements may assign different values, which can result in hard-to-debug issues when other parts of the code make incorrect assumptions about the constant&#8217;s value.</p>
</div>
<div class="paragraph">
<p>It is advisable to avoid duplicate define statements to prevent potential unexpected behavior and to ensure code clarity and correctness.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Remove duplicate <code>define</code> statements and only keep the intended one.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-php" data-lang="php">define( 'CONSTANT_VALUE', 'old value' );
define( 'SCRIPT_DEBUG', 1 );

// Noncompliant, tries to redefine constant defined 2 lines above
define( 'CONSTANT_VALUE', 'intended value' );
echo CONSTANT_VALUE; // output: 'old value'</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-php" data-lang="php">define( 'SCRIPT_DEBUG', 1 );

// Compliant
define( 'CONSTANT_VALUE', 'intended value' );
echo CONSTANT_VALUE; // output: 'intended value'</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p><a href="https://www.php.net/manual/en/language.constants.syntax.php">PHP Manual - Constants Syntax</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>