This rule raises an issue when a `TypeVar` is used in a function instead of the generic function type syntax.

== Why is this an issue?

Prior to Python 3.12 functions using generic types were created as follows:

[source,python]
----
from typing import TypeVar

_T = TypeVar("_T")

def func(a: _T, b: _T) -> _T:
    ...
----

This snippet of code can be confusing and difficult to read. This is why is it a good idea to 
use the new type parameter syntax of Python 3.12 that allows for a more concise and more readable code:

[source,python]
----
def func[T](a: T, b: T) -> T:
    ...
----



== How to fix it

To fix this issue replace the usage a `TypeVar` as a function parameter type with the generic type parameter syntax.

=== Code examples

==== Noncompliant code example

[source,python,diff-id=1,diff-type=noncompliant]
----
from typing import TypeVar

_T = TypeVar("_T", bound=str)

def func(a: _T, b: int) -> _T: # Noncompliant: the usage of a TypeVar could be simplified with a generic type parameter.
    ...
----

==== Compliant solution

[source,python,diff-id=1,diff-type=compliant]
----
def func[T: str](a: T, b: int) -> T: # Compliant: the code is clear and concise.
    ...
----

//=== How does this work?

//=== Pitfalls

//=== Going the extra mile


//== Resources
//=== Documentation
//=== Articles & blog posts
//=== Conference presentations
//=== Standards
//=== External coding guidelines
//=== Benchmarks
