== How to fix it in Botan

=== Code examples

include::../../common/fix/code-rationale.adoc[]

==== Noncompliant code example

[source,cpp,diff-id=1,diff-type=noncompliant]
----
#include <botan/cipher_mode.h>

void encrypt() {
  Botan::Cipher_Mode::create("DES/CBC/PKCS7", Botan::ENCRYPTION); // Noncompliant
}
----

==== Compliant solution

[source,cpp,diff-id=1,diff-type=compliant]
----
#include <botan/cipher_mode.h>

void encrypt() {
  Botan::Cipher_Mode::create("AES-256/GCM", Botan::ENCRYPTION);
}
----

=== How does this work?

include::../../common/fix/strong-cryptography.adoc[]

