== How to fix it in CryptoPP

=== Code examples

include::../../common/fix/code-rationale.adoc[]

==== Noncompliant code example

[source,cpp,diff-id=21,diff-type=noncompliant]
----
#include <cryptopp/des.h>

void encrypt() {
  CryptoPP::DES::Encryption(key, sizeof(key)); // Noncompliant
}
----

==== Compliant solution

[source,cpp,diff-id=21,diff-type=compliant]
----
#include <cryptopp/aes.h>

void encrypt() {
  CryptoPP::AES::Encryption(key, sizeof(key));
}
----

=== How does this work?

include::../../common/fix/strong-cryptography.adoc[]

