== How to fix it in OpenSSL

=== Code examples

include::../../common/fix/code-rationale.adoc[]

==== Noncompliant code example

[source,cpp,diff-id=31,diff-type=noncompliant]
----
#include <openssl/evp.h>

void encrypt() {
  EVP_des_cbc(); // Noncompliant
}
----

==== Compliant solution

[source,cpp,diff-id=31,diff-type=compliant]
----
#include <openssl/evp.h>

void encrypt() {
  EVP_aes_128_gcm();
}
----

=== How does this work?

include::../../common/fix/strong-cryptography.adoc[]

