<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p><code>NCHAR</code> and <code>NVARCHAR2</code> lengths must be given in characters, not bytes. This is partly because a single character may occupy more than a single byte in memory. Specify the field length in bytes, and theoretically your value could overrun the field, but instead Oracle simply refuses to run the code. Specify it in characters, and Oracle will allocate the appropriate number of bytes to store the requested number of characters. Trying to specify the length semantics in bytes will result in the <code>PLS-00639: NCHAR/NVARCHAR2 cannot be byte length semantics</code> exception being raised.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-sql" data-lang="sql">DECLARE
  foo NCHAR(42 BYTE); -- Noncompliant - raises PLS-00639
BEGIN
  NULL;
END;
/</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-sql" data-lang="sql">DECLARE
  foo NCHAR(42);      -- Compliant
  bar NCHAR(42 CHAR); -- Also compliant, as an alternative
BEGIN
  NULL;
END;
/</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Specify the size of "xxx" in chars.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_8_may_2015_150611_dinesh_bolkensteyn_wrote">on 8 May 2015, 15:06:11 Dinesh Bolkensteyn wrote:</h3>
<div class="paragraph">
<p>LGTM, except the <code> Specify the field length in bytes, and your value could overrun the field.</code> bit.</p>
</div>
<div class="paragraph">
<p>There is no buffer overflow risk, as Oracle refuses to run that code.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_8_may_2015_173943_ann_campbell_wrote">on 8 May 2015, 17:39:43 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>Updated. See if it&#8217;s okay please [~dinesh.bolkensteyn]</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_11_may_2015_060017_dinesh_bolkensteyn_wrote">on 11 May 2015, 06:00:17 Dinesh Bolkensteyn wrote:</h3>
<div class="paragraph">
<p>LGTM</p>
</div>
</div>
</div>
</div>