== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,vbnet,diff-id=1,diff-type=noncompliant]
----
towns.Item(x) = "London"
towns.Item(x) = "Chicago";  // Noncompliant
----

=== Compliant solution

[source,vbnet,diff-id=1,diff-type=compliant]
----
towns.Item(x) = "London"
towns.Item(y) = "Chicago";
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../highlighting.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
