=== Noncompliant code example

With the table ``++MY_TABLE++`` having a ``++NOT NULL++`` column ``++N2++`` without default value and a ``++NOT NULL++`` column ``++N3++`` with default value:

[source,text]
----
EXEC SQL
  INSERT INTO MY_TABLE  *> Noncompliant; N2 value omitted
  (
    N1
  )
  VALUES 
  (
    :ITQ1-NUMBER,
  )
END-EXEC.
----
