<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Any insert which omits a value for a <code>NOT NULL</code> column in a database table will be automatically rejected by the database unless a default value has been specified for the column.</p>
</div>
<div class="sect2">
<h3 id="_noteworthy">Noteworthy</h3>
<div class="paragraph">
<p>This rule raises issues only when a <strong>Data Dictionary</strong> is provided during the analysis. See&#160;<a href="https://docs.sonarqube.org/latest/analysis/languages/plsql/" class="bare">https://docs.sonarqube.org/latest/analysis/languages/plsql/</a></p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Ensure that all <code>NOT NULL</code> columns have a value specified in the <code>INSERT</code> statement.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="paragraph">
<p>Given the following table:</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-sql" data-lang="sql">CREATE TABLE MY_TABLE (
  N1 NUMBER NOT NULL,
  N2 VARCHAR2(50) NOT NULL,
  N3 VARCHAR2(50) DEFAULT 'Default Value'
);</code></pre>
</div>
</div>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-sql" data-lang="sql">INSERT INTO MY_TABLE  -- Noncompliant; N2 value omitted
(
  N1
)
VALUES
(
  1
)</code></pre>
</div>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-sql" data-lang="sql">INSERT INTO MY_TABLE  -- Compliant; N3 has a default value
(
  N1,
  N2
)
VALUES
(
  1,
  'Paul'
)</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Include a value for "xxx" in this insert.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>table name</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_is_related_to_s3484">is related to: <a data-rspec-id="S3484" class="rspec-auto-link">S3484</a></h3>

</div>
</div>
</div>