<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Serializing a non-<code>static</code> inner class will result in an attempt at serializing the outer class as well. If the outer class is actually serializable, then the serialization will succeed but possibly write out far more data than was intended.</p>
</div>
<div class="paragraph">
<p>Making the inner class <code>static</code> (i.e. "nested") avoids this problem, therefore inner classes should be <code>static</code> if possible. However, you should be aware that there are semantic differences between an inner class and a nested one:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>an inner class can only be instantiated within the context of an instance of the outer class.</p>
</li>
<li>
<p>a nested (<code>static</code>) class can be instantiated independently of the outer class.</p>
</li>
</ul>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">public class Raspberry implements Serializable {
  // ...

  public class Drupelet implements Serializable {  // Noncompliant; output may be too large
    // ...
  }
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">public class Raspberry implements Serializable {
  // ...

  public static class Drupelet implements Serializable {
    // ...
  }
}</code></pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p><a href="https://wiki.sei.cmu.edu/confluence/x/ZTdGBQ">CERT, SER05-J.</a> - Do not serialize instances of inner classes</p>
</li>
</ul>
</div>
<hr>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Make this inner class static.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_25_sep_2014_091910_ann_campbell_wrote">on 25 Sep 2014, 09:19:10 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~nicolas.peru] FB divides this into two rules: one for <code>Serializable</code> outer classes and one for non-serializable outer classes.</p>
</div>
<div class="paragraph">
<p>I really debated one rule or two because the problems caused are different in magnitude, but since the title and the recommendation are the same for both I combined. Let me know if you&#8217;d like me to split.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_25_sep_2014_115559_nicolas_peru_wrote">on 25 Sep 2014, 11:55:59 Nicolas Peru wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] Detection of the issue is the same : (non static inner class) the mitigation is different depending on the <code>Serializability</code> of the outer class.</p>
</div>
<div class="literalblock">
<div class="content">
<pre>Since one case will result in a runtime error (critical) and the other one in a (major) recommandation I tend to think they should be split with different severity (but implementation will probably be shared).</pre>
</div>
</div>
</div>
</div>
</div>