<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>In Java 15 Text Blocks are now official and can be used. The most common pattern for multiline strings in Java &lt; 15 was to write String concatenation. Now it&#8217;s possible to do it in a more natural way using Text Blocks.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">String textBlock =
               "&lt;html&gt;\n" +
               "    &lt;body&gt;\n" +
               "        &lt;tag&gt;\n" +
               "        &lt;/tag&gt;\n" +
               "    &lt;/body&gt;\n" +
               "&lt;/html&gt;";</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">String textBlock = """
        &lt;html&gt;
            &lt;body&gt;
                &lt;tag&gt;
                &lt;/tag&gt;
            &lt;/body&gt;
        &lt;/html&gt;""";</code></pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p><a href="https://openjdk.java.net/jeps/378">JEP 378: Text Blocks</a></p>
</li>
<li>
<p><a href="https://openjdk.org/projects/amber/guides/text-blocks-guide">Programmer&#8217;s Guide To Text Blocks</a>, by Jim Laskey and Stuart Marks</p>
</li>
</ul>
</div>
<hr>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Replace this String concatenation with Text block.</p>
</div>
</div>
</div>
</div>