== Why is this an issue?

include::../description_without_reference_to_s5998.adoc[]

=== Noncompliant code example

[source,python]
----
r"a|b|c" # Noncompliant
----

=== Compliant solution

[source,python]
----
r"[abc]"
# or
r"[a-c]"
----
