include::../description_without_reference_to_s5998.adoc[]

== Noncompliant Code Example

[source,python]
----
r"a|b|c" # Noncompliant
----

== Compliant Solution

[source,python]
----
r"[abc]"
# or
r"[a-c]"
----
