<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>A Spring <code>singleton</code> bean may be used by many threads at once, and the use of instance (non-<code>static</code>) variables could cause concurrency issues.</p>
</div>
<div class="paragraph">
<p>This rule applies to classes with the following annotations: <code>@Service</code>, <code>@Component</code>, <code>@Repository</code>, <code>@Scope("singleton")</code></p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">@Service("animalService")
public class AnimalService {
private int age = 1;              // Noncompliant
private static int count = 0; // Compliant; static

@Inject
private AnimalDAO animalDAO;  // Compliant; managed by Spring
...
}</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Make "xxx" static, remove it, or give the class "prototype" scope.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_2_mar_2018_141529_alexandre_gigleux_wrote">on 2 Mar 2018, 14:15:29 Alexandre Gigleux wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] This RSPEC is obsolete because better described <a data-rspec-id="S3749" class="rspec-auto-link">RSPEC-3749</a> and already implemented by SONARJAVA-2168.</p>
</div>
<div class="paragraph">
<p>Shall we simply "Close Won&#8217;t Fix" this RSPEC or delete it?</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_2_mar_2018_151459_ann_campbell_wrote">on 2 Mar 2018, 15:14:59 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~alexandre.gigleux] the scope is slightly different:</p>
</div>
<div class="paragraph">
<p>||this||<a data-rspec-id="S3749" class="rspec-auto-link">RSPEC-3749</a>||</p>
</div>
<div class="paragraph">
<p>|@Service|@Service|</p>
</div>
<div class="paragraph">
<p>|@Component| |</p>
</div>
<div class="paragraph">
<p>|@Repository|@Repository|</p>
</div>
<div class="paragraph">
<p>|@Scope("singleton")| |</p>
</div>
<div class="paragraph">
<p>| |@Controller|</p>
</div>
<div class="paragraph">
<p>Should the missing elements be added to <a data-rspec-id="S3749" class="rspec-auto-link">RSPEC-3749</a> or just dropped? Either way, I&#8217;m fine to Won&#8217;t-Fix this.</p>
</div>
</div>
</div>
</div>