<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>The <code>TO_NUMBER</code> function is converting the value of <code>BINARY_FLOAT</code>, <code>BINARY_DOUBLE</code>, <code>CHAR</code>, <code>VARCHAR2</code>, <code>NCHAR</code>, or <code>NVARCHAR2</code> datatype to a value of <code>NUMBER</code> datatype.</p>
</div>
<div class="paragraph">
<p>Without providing the format of the input, <code>TO_NUMBER</code> will consider the provided value is compliant with the default format. Relying on a default configuration is a source of error because it creates a dependency between the code and the configuration of the ORACLE server where this code is deployed.</p>
</div>
<div class="paragraph">
<p>The behaviour of the <code>TO_NUMBER</code> function will certainly not be the expected one if the configuration of the ORACLE server is changing.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="paragraph">
<p>The following code with return 0 on an ORACLE server running with its default US configuration with p_string = "2,540"</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-sql" data-lang="sql">IF ( TO_NUMBER(p_string) &gt;= 0 and TO_NUMBER(p_string) &lt;= TO_NUMBER('50.00') ) THEN
  RETURN 1;
ELSE
  RETURN 0;
END IF;</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="paragraph">
<p>The following code with return 1 on an ORACLE server running with its default FR configuration with p_string = "2,540" because the comma will be interpreted as decimal separator instead of thousand separator.</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-sql" data-lang="sql">IF ( TO_NUMBER(p_string, '99.99') &gt;= 0 and TO_NUMBER(p_string, '99.99') &lt;= TO_NUMBER('50.00','99.99') ) THEN
  RETURN 1;
ELSE
  RETURN 0;
END IF;</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Provide the format of the input value.</p>
</div>
</div>
</div>
</div>