include::../summary.adoc[]

== Why is this an issue?

include::../rationale.adoc[]

== How to fix it
//== How to fix it in FRAMEWORK NAME

=== Code examples

==== Noncompliant code example

[source,python,diff-id=1,diff-type=noncompliant]
----
from graphql_server.flask import GraphQLView

app.add_url_rule("/api",
    view_func=GraphQLView.as_view(  # Noncompliant
        name="api",
        schema=schema,
    )
)
----

==== Compliant solution

[source,python,diff-id=1,diff-type=compliant]
----
from graphql_server.flask import GraphQLView
from graphql.validation import NoSchemaIntrospectionCustomRule

app.add_url_rule("/api",
    view_func=GraphQLView.as_view(
        name="api",
        schema=schema,
        validation_rules=[
            NoSchemaIntrospectionCustomRule
        ]
    )
)
----

=== How does this work?

include::../common/fix/disable.adoc[]

//=== Pitfalls

//=== Going the extra mile


== Resources

include::../common/resources/standards.adoc[]

//=== Documentation
//=== Articles & blog posts
//=== Conference presentations

=== Standards

//=== External coding guidelines
//=== Benchmarks
