//// This rule is superseded by RSPEC-5547
//// Please consider implementing this latter instead.
include::../description.adoc[]

== Noncompliant Code Example

----
PLS_INTEGER := DBMS_CRYPTO.ENCRYPT_DES
                           + DBMS_CRYPTO.CHAIN_CBC
                           + DBMS_CRYPTO.PAD_PKCS5;
----

== Compliant Solution

----
PLS_INTEGER := DBMS_CRYPTO.ENCRYPT_AES256
                           + DBMS_CRYPTO.CHAIN_CBC
                           + DBMS_CRYPTO.PAD_PKCS5;
----

include::../see.adoc[]
