<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when synchronous file operations like <code>open()</code> are used within asynchronous functions.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Using synchronous file operations like <code>open()</code> in asynchronous code blocks the entire event loop. This undermines the primary advantage of asynchronous programming - the ability to perform concurrent operations without blocking execution.</p>
</div>
<div class="paragraph">
<p>When an async function makes a synchronous file operation:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>The event loop is completely blocked until the file I/O operation completes</p>
</li>
<li>
<p>No other coroutines can run during this time, even if they&#8217;re ready to execute</p>
</li>
<li>
<p>The responsiveness of the application is degraded</p>
</li>
<li>
<p>In server applications, this can cause timeouts or failures for other concurrent requests</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>Instead, async libraries provide mechanisms to handle file operations asynchronously:</p>
</div>
<div class="ulist">
<ul>
<li>
<p><code>aiofiles</code> library for asyncio</p>
</li>
<li>
<p><code>trio.open_file()</code> for Trio</p>
</li>
<li>
<p><code>anyio.open_file()</code> for AnyIO</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>Using these constructs allows other tasks to continue executing while waiting for the potentially blocking file operation to complete.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it_in_asyncio">How to fix it in Asyncio</h2>
<div class="sectionbody">
<div class="paragraph">
<p>The standard library&#8217;s <code>asyncio</code> package doesn&#8217;t provide built-in asynchronous file I/O operations. Use the <code>aiofiles</code> library to handle file operations asynchronously.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">async def read_config():
    with open("config.json", "r") as file:  # Noncompliant
        data = file.read()
    return data</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">import aiofiles

async def read_config():
    async with aiofiles.open("config.json", "r") as file: # Compliant
        data = await file.read()
    return data</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it_in_trio">How to fix it in Trio</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Use <code>trio.open_file()</code> to handle file operations asynchronously.</p>
</div>
<div class="sect2">
<h3 id="_code_examples_2">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example_2">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">async def read_config():
    with open("config.json", "r") as file:  # Noncompliant
        data = file.read()
    return data</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution_2">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">import trio

async def read_config():
    async with await trio.open_file("config.json", "r") as file: # Compliant
        data = await file.read()
    return data</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it_in_anyio">How to fix it in AnyIO</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Use <code>anyio.open_file()</code> to handle file operations asynchronously.</p>
</div>
<div class="sect2">
<h3 id="_code_examples_3">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example_3">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">async def read_config():
    with open("config.json", "r") as file:  # Noncompliant
        data = file.read()
    return data</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution_3">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">import anyio

async def read_config():
    async with await anyio.open_file("config.json", "r") as file: # Compliant
        data = await file.read()
    return data</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Trio - <a href="https://trio.readthedocs.io/en/stable/reference-io.html#trio.open_file">trio.open_file() documentation</a></p>
</li>
<li>
<p>AnyIO - <a href="https://anyio.readthedocs.io/en/stable/fileio.html">File I/O in AnyIO</a></p>
</li>
<li>
<p>Aiofiles - <a href="https://github.com/Tinche/aiofiles">aiofiles project</a></p>
</li>
</ul>
</div>
</div>
<div class="sect2">
<h3 id="_articles_blog_posts">Articles &amp; blog posts</h3>
<div class="ulist">
<ul>
<li>
<p>Python - <a href="https://realpython.com/async-io-python/">Async IO in Python</a></p>
</li>
</ul>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Use an asynchronous file API instead of the synchronous open() in this async function.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="ulist">
<ul>
<li>
<p>Primary locations: the synchronous open() call within an async function</p>
</li>
<li>
<p>Secondary locations: the enclosing async function <code>aync</code> keyword (message: "this is an asynchronous function")</p>
</li>
</ul>
</div>
</div>
</div>
</div>