<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>The unit type <code>()</code> is always equal to itself and using it in comparisons is either redundant or a mistake, often caused by accidental semicolon placement.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-rust" data-lang="rust">fn foo() {}
fn bar() {}
fn baz() {}

if {
    foo();
} == {
    bar();
} {
    baz();
} // Noncompliant: Comparing unit values.</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-rust" data-lang="rust">fn foo() {}
fn bar() {}
fn baz() {}

foo();
bar();
baz(); // Compliant: No unit comparison.</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Clippy Lints - <a href="https://rust-lang.github.io/rust-clippy/master/index.html#unit_cmp" class="bare">https://rust-lang.github.io/rust-clippy/master/index.html#unit_cmp</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>