== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,xml]
----
<configuration>
  <system.web>
    <authentication mode="Forms">
      <forms name="customer_login" timeout="30" loginUrl="~/Login.aspx">
        <credentials passwordFormat="Clear"> <!-- Noncompliant -->
          <user name="admin" password="password" />
        </credentials>
      </forms>
    </authentication>
  </system.web>
</configuration>
----

=== Compliant solution

This form of authentication does not support any secure format, thus its use
is highly discouraged.

include::../see.adoc[]

* Microsoft Learn - https://learn.microsoft.com/en-us/previous-versions/dotnet/netframework-4.0/e01fc50a%28v%3Dvs.100%29[credentials Element for forms for authentication]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
