include::../summary.adoc[]

== Why is this an issue?

include::../rationale.adoc[]

include::../impact.adoc[]

== How to fix it

=== Code examples

==== Noncompliant code example

[source,xml,diff-id=2,diff-type=noncompliant]
----
<configuration>
  <system.web>
    <authentication mode="Forms">
      <forms name="customer_login" timeout="30" loginUrl="~/Login.aspx">
        <credentials passwordFormat="Clear"> <!-- Noncompliant -->
          <user name="admin" password="password" />
        </credentials>
      </forms>
    </authentication>
  </system.web>
</configuration>
----

==== Compliant solution

The form of authentication supports only Clear, SHA1, and MD5. Neither of them
is secure, so its use is highly discouraged. 

Instead, it is recommended to use another API altogether, such as the Membership
API:

[source,xml,diff-id=2,diff-type=compliant]
----
<configuration>
   <system.web>
      <authentication mode="Forms">
         <forms name="401kApp" loginUrl="/login.aspx" />
      </authentication>
      <membership defaultProvider="SqlProvider" hashAlgorithmType="SHA256">
        <providers>
            <add name="SqlProvider"
                type="System.Web.Security.SqlMembershipProvider"
                connectionStringName="YourConnectionString"
                enablePasswordRetrieval="false"
                enablePasswordReset="true"
                requiresQuestionAndAnswer="false"
                requiresUniqueEmail="false"
                maxInvalidPasswordAttempts="5"
                minRequiredPasswordLength="8"
                minRequiredNonalphanumericCharacters="1"
                passwordFormat="Hashed"
                applicationName="/" />
        </providers>
      </membership>
    </system.web>
</configuration>
----

=== How does this work?

include::../common/fix/password-hashing.adoc[]

Unfortunately, out of the .Net box, the algorithms we recommend are not
available. Therefore, we recommend using the strong algorithms available, such
as SHA256.

include::../common/fix/plaintext-password.adoc[]

== Resources

=== Documentation

* Microsoft Learn - https://learn.microsoft.com/en-us/previous-versions/dotnet/netframework-4.0/e01fc50a%28v%3Dvs.100%29[credentials Element for forms for authentication]

include::../common/resources/standards.adoc[]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Use a secure password hashing algorithm.

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
