include::../summary.adoc[]

== Why is this an issue?

include::../rationale.adoc[]

include::../impact.adoc[]

== How to fix it

=== Code examples

==== Noncompliant code example

[source,xml]
----
<configuration>
  <system.web>
    <authentication mode="Forms">
      <forms name="customer_login" timeout="30" loginUrl="~/Login.aspx">
        <credentials passwordFormat="Clear"> <!-- Noncompliant -->
          <user name="admin" password="password" />
        </credentials>
      </forms>
    </authentication>
  </system.web>
</configuration>
----

==== Compliant solution

The ``Form`` mode of authentication supports only Clear, SHA1, and MD5. None of them
are secure, so their use is highly discouraged. 

Instead, it is recommended to use another, modern authentication mechanism such as https://learn.microsoft.com/en-us/aspnet/core/security/authentication/identity?view=aspnetcore-8.0&tabs=visual-studio[ASP.Net Identity].


=== How does this work?

include::../common/fix/password-hashing.adoc[]

Fortunately, ASP.Net Identity supports PBKDF2 and uses it by default.

include::../common/fix/plaintext-password.adoc[]

== Resources

=== Documentation

* Microsoft Learn - https://learn.microsoft.com/en-us/previous-versions/dotnet/netframework-4.0/e01fc50a%28v%3Dvs.100%29[credentials Element for forms for authentication]

include::../common/resources/standards.adoc[]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Use a secure password hashing algorithm.

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
