== Why is this an issue?

Nesting functions can quickly turn your code into "spaghetti code". Such code is hard to read, refactor and therefore to maintain.


=== Noncompliant code example

With the default threshold of 3:

[source,php]
----
function f () {
  function f_inner () {
    function f_inner_inner() { 
      function f_inner_inner_inner() { // Noncompliant
      }
    }
  }
}
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

include::parameters.adoc[]

endif::env-github,rspecator-view[]
