<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>SpEL, the Spring Expression Languages allows developers fine-grained control over the values injected into fields and parameters.
Using the <code>@Value</code> annotation, it is possible to inject values from sources such as system properties.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>The <code>@Value</code> annotation does not guarantee that the property is defined.
Particularly if a field or parameter is annotated as nullable, it indicates that the developer assumes that the property may be undefined.</p>
</div>
<div class="paragraph">
<p>An undefined property may lead to runtime exceptions when the Spring framework tries to inject the autowired dependency during bean creation.</p>
</div>
<div class="paragraph">
<p>This rule raises an issue when a nullable field or parameter is annotated with <code>@Value</code> and no default value is provided.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Add a default value to the <code>@Value</code> annotation.
A default value can be supplied by using the colon (<code>:</code>) operator.
As the field is nullable, the default value should most likely be <code>#{null}</code>.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">@Nullable
@Value("${my.property}") // Noncompliant, no default value is provided, even though the field is nullable
private String myProperty;</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">@Nullable
@Value("${my.property:#{null}}") // Compliant, a default value is provided
private String myProperty;</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_articles_blog_posts">Articles &amp; blog posts</h3>
<div class="ulist">
<ul>
<li>
<p>Baeldung - <a href="https://www.baeldung.com/spring-value-defaults">Using Spring @Value With Defaults</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>