include::../description-dotnet.adoc[]

=== Code examples

==== Noncompliant code example

[source,vbnet,diff-id=1,diff-type=noncompliant]
----
Public Class Example

    Private Shared rwLock As New ReaderWriterLock

    Public Sub AcquireWriterLock()
        rwLock.AcquireWriterLock(2000)  ' Noncompliant, as the lock release is on the callers responsibility
    End Sub

    Public Sub DoSomething()
        ' ...
    End Sub

    Public Sub ReleaseWriterLock()
        rwLock.ReleaseWriterLock()
    End Sub

End Class
----

==== Compliant solution

[source,vbnet,diff-id=1,diff-type=compliant]
----
Public Class Example

    Private Shared rwLock As New ReaderWriterLock

    Public Sub DoSomething()
        rwLock.AcquireWriterLock(2000)  ' Compliant, locks are released in the same method
        Try
            ' ...
        Finally
            rwLock.ReleaseWriterLock()
        End Try
    End Sub

End Class
----

include::../resources-dotnet.adoc[]

include::../rspecator.adoc[]