<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>While it is possible for inheritance to be non-<code>public</code>, it is rarely justified and complicates the use of the derived class. For instance, inherited member visibility is diminished and implicit and <code>static_cast</code> casts from the derived class to the base class will not work.</p>
</div>
<div class="paragraph">
<p>It is sometimes used to limit the base class functionality available in the derived class. When that is the desire, composition should be used instead.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">class B : private A {  // Noncompliant
  // ...
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">class B : public A {
  // ...
}</code></pre>
</div>
</div>
<div class="paragraph">
<p>or</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">class B {
  private:
  A a;
  // ...
}</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Make this inheritance "public".</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>visibility keyword and inherited class name</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_7_jan_2016_201635_alban_auzeill_wrote">on 7 Jan 2016, 20:16:35 Alban Auzeill wrote:</h3>
<div class="paragraph">
<p>In the bellow description there&#8217;s nothing about <strong>protected</strong> inheritance. What I&#8217;ve done in rule&#8217;s implementation, is applying the same logique to <strong>private</strong> and <strong>protected</strong>.</p>
</div>
<div class="paragraph">
<p>Do we have to change the documentation about this?</p>
</div>
<div class="paragraph">
<p>I also distinguish the default inheritance of a class as private and  the default inheritance of a struct as public.</p>
</div>
<div class="paragraph">
<p>My non-compliant test cases are:</p>
</div>
<div class="listingblock">
<div class="content">
<pre>class B : A {
};

class C : protected A {
};

struct D : protected A {
};

class E : private A {
};

struct F : private A {
};</pre>
</div>
</div>
<div class="paragraph">
<p>My compliant test cases are:</p>
</div>
<div class="listingblock">
<div class="content">
<pre>struct B : A {
};

class C : public A {
};

struct D : public A {
};</pre>
</div>
</div>
<div class="paragraph">
<p>Do you agree?</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_7_jan_2016_204038_ann_campbell_wrote">on 7 Jan 2016, 20:40:38 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>Since both the title and description are about "non-public" inheritance, I think the <code>protected</code> case is already adequately covered. Note that the Noncompliant Code Example only shows <code>private</code> inheritance because adding a <code>protected</code> example wouldn&#8217;t have contributed anything significant.</p>
</div>
</div>
</div>
</div>