<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This is an issue when destructive string methods like <code>tr!</code>, <code>gsub!</code>, or <code>sub!</code> are used in contexts where their return value is assigned to a variable, used in method chains, or returned from a method.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Destructive string methods in Ruby (those ending with <code>!</code>) have a special behavior: they return <code>nil</code> when no changes are made to the string, rather than returning the original string.</p>
</div>
<div class="paragraph">
<p>This behavior can cause unexpected runtime errors. For example, if you chain methods after a destructive call that returns <code>nil</code>, you&#8217;ll get a <code>NoMethodError</code> when trying to call methods on <code>nil</code>.</p>
</div>
<div class="paragraph">
<p>The problem is particularly subtle because the code might work correctly in some cases (when the string is actually modified) but fail in others (when no modification occurs). This makes bugs hard to reproduce and debug.</p>
</div>
<div class="paragraph">
<p>Consider this example:</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-ruby" data-lang="ruby">str = "hello world"
result = str.tr!(" ", "_")  # Returns "hello_world"
puts result  # Works fine

str = "helloworld"
result = str.tr!(" ", "_")  # Returns nil (no spaces to replace)
puts result  # Prints nothing - unexpected!</code></pre>
</div>
</div>
<div class="paragraph">
<p>The same code produces different types of results depending on the input data, making the behavior unpredictable.</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="paragraph">
<p>This issue can cause <code>NoMethodError</code> exceptions at runtime when methods are called on <code>nil</code> values. These errors are often intermittent and depend on the input data, making them difficult to catch during testing and potentially causing production failures.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Use the non-destructive variant of the method instead. Non-destructive methods always return a string, making the behavior predictable.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-ruby" data-lang="ruby">str = "color"
result = str.downcase.tr!(" ", "_")  # Noncompliant: returns nil
puts result  # nil - unexpected!</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-ruby" data-lang="ruby">str = "color"
result = str.downcase.tr(" ", "_")  # Always returns a string
puts result  # "color"</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Ruby String Documentation - <a href="https://ruby-doc.org/core/String.html">Official Ruby documentation for String class methods</a></p>
</li>
<li>
<p>Ruby Style Guide - String Methods - <a href="https://rubystyle.guide/#string-mutating-methods">Community style guide recommendations for string methods</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>