<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Allowing an application to dynamically change the structure of a database at runtime is very dangerous because the application can become unstable under unexpected conditions. Best practices dictate that applications only manipulate data.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-sql" data-lang="sql">CREATE TABLE my_table(
  my_column INTEGER
);</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_exceptions">Exceptions</h3>
<div class="paragraph">
<p>Creating global temporary tables, creating indexes on those tables, and then dropping those indices and tables is allowed.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove this DDL statement.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_13_may_2014_153257_freddy_mallet_wrote">on 13 May 2014, 15:32:57 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>Spec updated @Ann to make this spec supporting two exceptions (COBOL-1053)</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_25_nov_2014_095529_pierre_yves_nicolas_wrote">on 25 Nov 2014, 09:55:29 Pierre-Yves Nicolas wrote:</h3>
<div class="paragraph">
<p>We can identify that an index which is dropped is temporary only if it is prefixed with "SESSION.". It is therefore not possible for the exception which is described:</p>
</div>
<div class="paragraph">
<p>{code}DROP INDEX X1T09SCO{code}</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_25_nov_2014_100630_freddy_mallet_wrote">on 25 Nov 2014, 10:06:30 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>\[~pierre-yves.nicolas], this is possible because if you have a look to the exceptions, you&#8217;ll see that the index was created earlier on a temporary table.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_25_nov_2014_101951_pierre_yves_nicolas_wrote">on 25 Nov 2014, 10:19:51 Pierre-Yves Nicolas wrote:</h3>
<div class="paragraph">
<p>Ok, it may be doable but that would make the rule implementation much more complex. What if the create statement is located in another file?</p>
</div>
<div class="paragraph">
<p>Moreover, I suppose that this exception would be rarely used: a temporary index is usually created on a temporary table, and the temporary index is rarely dropped explicitly because it is implicitly dropped when the temporary table is dropped.</p>
</div>
</div>
</div>
</div>