<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This <code>BYPASSING BUFFER</code> clause explicitly switches off SAP table buffering, so the <code>SELECT</code> reads data directly from the database.</p>
</div>
<div class="paragraph">
<p>By definition, using this clause can lead to performance issues, which is why its use must be strongly indicated.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-abap" data-lang="abap">SELECT *
INTO US_PERSONS
FROM PERSONS
BYPASSING BUFFER
WHERE CITY EQ 'US'</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Make sure this use of a "BYPASSING BUFFER" clause is strongly indicated.</p>
</div>
</div>
</div>
</div>