== Why is this an issue?

Namespaces with no lines of code clutter a project and should be removed. 


=== Noncompliant code example

[source,text]
----
namespace MyEmptyNamespace // Noncompliant
{

}
----


