include::../description.adoc[]

include::../recommended.adoc[]

== Noncompliant Code Example

An ingress rule allowing all inbound SSH traffic:

[source,terraform]
----
resource "aws_security_group" "noncompliant" {
  name        = "allow_ssh_noncompliant"
  description = "allow_ssh_noncompliant"
  vpc_id      = aws_vpc.main.id

  ingress {
    description      = "SSH rule"
    from_port        = 22
    to_port          = 22 # SSH traffic
    protocol         = "tcp"
    cidr_blocks      = ["0.0.0.0/0"] # from all IP addresses is authorized
  }
}
----

== Compliant Solution

An ingress rule allowing inbound SSH traffic from specific IP addresses:

----
resource "aws_security_group" "compliant" {
  name        = "allow_ssh_compliant"
  description = "allow_ssh_compliant"
  vpc_id      = aws_vpc.main.id

  ingress {
    description      = "SSH rule"
    from_port        = 22
    to_port          = 22
    protocol         = "tcp"
    cidr_blocks      = ["1.2.3.0/24"] # Compliant 
  }
}
----

include::../see.adoc[]