<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>The if-<code>null</code> operator <code>??</code> allows providing a default value when dealing with <code>null</code>. It only coalesces when the original value is <code>null</code>. Therefore, it is safer and shorter than relying upon testing against <code>null</code> or <code>undefined</code> explicitly.</p>
</div>
<div class="paragraph">
<p>This rule reports when conditionals (<code>?</code>) can be safely replaced with coalescing (<code>??</code>).</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Rewrite the expression using <code>??</code> on the unchecked operands.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-dart" data-lang="dart">int either(int? x, int y) {
  return x != null ? x : y;
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-dart" data-lang="dart">int either(int? x, int y) {
  return x ?? y;
}</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p>Dart Docs <a href="https://dart.dev/tools/linter-rules/prefer_if_null_operators">Dart Linter rule - prefer_if_null_operators</a></p>
</li>
</ul>
</div>
<hr>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="ulist">
<ul>
<li>
<p>"Use the '??' operator rather than '?:' when testing for 'null'."</p>
</li>
</ul>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>The whole ternary operator.</p>
</div>
<hr>
</div>
</div>
</div>