<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>The nullish coalescing operator <code>??</code> allows providing a default value when dealing with <code>null</code> or <code>undefined</code>. It only coalesces when the original value is <code>null</code> or <code>undefined</code>. Therefore, it is safer and shorter than relying upon chaining logical <code>||</code> expressions or testing against <code>null</code> or <code>undefined</code> explicitly.</p>
</div>
<div class="paragraph">
<p>This rule reports when disjunctions (<code>||</code>) and conditionals (<code>?</code>) can be safely replaced with coalescing (<code>??</code>).</p>
</div>
<div class="paragraph">
<p>The TSConfig needs to set <code>strictNullChecks</code> to <code>true</code> for the rule to work properly.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Rewrite the logical expression <code>||</code> using <code>??</code> on the unchecked operands.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-typescript" data-lang="typescript">function either(x: number | undefined, y: number) {
  return x || y;
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-typescript" data-lang="typescript">function either(x: number | undefined, y: number) {
  return x ?? y;
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_noncompliant_code_example_2">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-typescript" data-lang="typescript">function either(x: number | undefined, y: number) {
  return x !== undefined ? x : y;
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution_2">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-typescript" data-lang="typescript">function either(x: number | undefined, y: number) {
  return x ?? y;
}</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p><a href="https://typescript-eslint.io/">typescript-eslint</a> - Rule <a href="https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/docs/rules/prefer-nullish-coalescing.mdx">prefer-nullish-coalescing</a></p>
</li>
<li>
<p><a href="https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing">MDN - Nullish coalescing</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>