<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Just as pain is your body&#8217;s way of telling you something is wrong, errors are PHP&#8217;s way of telling you there&#8217;s something you need to fix. Neither pain, nor PHP errors should be ignored.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-php" data-lang="php">@doSomethingDangerous($password);  // Noncompliant; '@' silences errors from function call</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-php" data-lang="php">doSomethingDangerous($password);</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove the '@' symbol from this function call to un-silence errors.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_relates_to_s2486">relates to: <a data-rspec-id="S2486" class="rspec-auto-link">S2486</a></h3>

</div>
<div class="sect2">
<h3 id="_on_17_sep_2014_113127_freddy_mallet_wrote">on 17 Sep 2014, 11:31:27 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>@ann, I would increase the SQALE remediation cost, as the remediation action should lead to take into account errors.</p>
</div>
</div>
</div>
</div>