<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>The <a href="https://docs.python.org/3/reference/datamodel.html#object.__len__"><code>__len__</code></a> special method enables to define the length of an object. As such it should always return an integer greater than or equal to 0. When the <code>len(...)</code> builtin is called on an object, any other return type will raise a <code>TypeError</code> and a negative integer will raise a <code>ValueError</code>.</p>
</div>
<div class="paragraph">
<p>This rule raises an issue when a <code>__len__</code> method returns an expression which is not an integer greater than or equal to 0.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">class A:
    def __len__(self):
        return 2.0  # Noncompliant

len(A())  # This will raise "TypeError: 'float' object cannot be interpreted as an integer"

class B:
    def __len__(self):
        return -2  # Noncompliant

len(B())  # This will raise "ValueError: __len__() should return &gt;= 0"</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">class A:
    def __len__(self):
        return 2

len(A())  # This will return 2</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_exceptions">Exceptions</h3>
<div class="paragraph">
<p>This rule raises no issue when the type returned by <code>__len__</code> is a class implementing the <a href="https://docs.python.org/3/reference/datamodel.html#object.__index__"><code>__index__</code></a> special method.</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">class Ind:
    def __index__(self):
        return 3

class A:
    def __len__(self):
        return Ind()

len(A())  # This will return 3</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="ulist">
<ul>
<li>
<p>Replace this return value with an integer &gt;= 0; XXX is negative.</p>
</li>
<li>
<p>Replace this return value with an integer &gt;= 0; XXX is of type YYY.</p>
</li>
</ul>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>Primary: The returned expression</p>
</div>
</div>
</div>
</div>