This is an issue when fuzzing functions using `*testing.F` are defined without appropriate build constraints to ensure compatibility with Go versions.

== Why is this an issue?

Go's native fuzzing feature was introduced in Go 1.18. When you write fuzzing functions that use `*testing.F`, `f.Add()`, or `f.Fuzz()` without proper build constraints, the code will fail to compile on Go versions prior to 1.18.

This creates compatibility issues in projects that need to support multiple Go versions. Build failures can occur when:

* Team members use different Go versions
* CI/CD pipelines run on older Go versions
* The project needs to maintain backward compatibility

Build constraints (also called build tags) tell the Go compiler to include or exclude files based on conditions like Go version, operating system, or architecture. For fuzzing code, you need to specify that the file should only be compiled when Go 1.18 or later is available.

=== What is the potential impact?

Without proper build constraints, projects may experience compilation failures when built with Go versions prior to 1.18. This can break CI/CD pipelines, prevent team collaboration across different Go versions, and create deployment issues in environments with older Go installations.

== How to fix it

Add build constraints at the top of files containing fuzzing functions. Use both the new `//go:build` syntax and the legacy `// +build` syntax for maximum compatibility.

=== Code examples

==== Noncompliant code example

[source,go,diff-id=1,diff-type=noncompliant]
----
package fuzz

import (
    "testing"
)

func FuzzExample(f *testing.F) { // Noncompliant
    f.Add("test")
    f.Fuzz(func(t *testing.T, s string) {
        // fuzzing logic
    })
}
----

==== Compliant solution

[source,go,diff-id=1,diff-type=compliant]
----
//go:build go1.18
// +build go1.18

package fuzz

import (
    "testing"
)

func FuzzExample(f *testing.F) {
    f.Add("test")
    f.Fuzz(func(t *testing.T, s string) {
        // fuzzing logic
    })
}
----

== Resources

=== Documentation

 * Go Fuzzing Documentation - https://go.dev/doc/tutorial/fuzz[Official Go tutorial on fuzzing with examples and best practices]

 * Build Constraints Documentation - https://pkg.go.dev/go/build#hdr-Build_Constraints[Official documentation on Go build constraints and build tags]

 * Go 1.18 Fuzzing Blog Post - https://go.dev/blog/fuzz-beta[Announcement and introduction to Go's native fuzzing feature]
