Sharing some naming conventions is a key point to make it possible for a team to efficiently collaborate. This rule allows to check that all method names match a provided regular expression.

== Noncompliant Code Example

With default provided regular expression: ``++^[a-z_][a-z0-9_]*$++``

----
class MyClass:
    def MyMethod(a,b):
        ...
----

== Compliant Solution

----
class MyClass:
    def my_method(a,b):
        ...
----
