== Why is this an issue?

Nullability of types is an important part of Dart's type system. In order to simplify the work with nullable types in Dart language there are many null-aware operators (`??`, `??=`, `?.`, `!`, `?..`, `?[]`, `...?`). Using those operators, will make your code more clear and concise.

Null-aware assignment (`??=`), will check if the value on the left side is `null`, and, if yes, will perform the assignment. This is an easy way to set default values instead of dealing with `null`. Thus, this operator doesn't make sense if used with `null` on the right side. For example, `x ??= null` will be reported as an issue.

=== Exceptions

The rule doesn't apply if property access is used. For example:

[source,dart]
----
class Person {
  String? name;
  int? age;

  void g() {
    name ??= null; // Ok, field
    age ??= null; // Ok, field
  }
}
----

== How to fix it

Remove the assignment or change the new value to not be `null`.

=== Code examples

==== Noncompliant code example

[source,dart,diff-id=1,diff-type=noncompliant]
----
void f() {
  var x;
  x ??= null; // Noncompliant
}
----

==== Compliant solution

[source,dart,diff-id=1,diff-type=compliant]
----
void f() {
  var x;
  x ??= 0;
}
----

== Resources

=== Documentation

* Dart Docs - https://dart.dev/tools/linter-rules/unnecessary_null_aware_assignments[Dart Linter rule - unnecessary_null_aware_assignments]


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Unnecessary assignment of 'null'.

=== Highlighting

The whole assignment statement

endif::env-github,rspecator-view[]
