== Why is this an issue?

A selector parameter is a ``++boolean++`` parameter that's used to determine which of two paths to take through a method. Specifying such a parameter may seem innocuous, particularly if it's well-named.


Unfortunately, developers calling the method won't see the parameter name, only its value. They'll be forced either to guess at the meaning or to take extra time to look the method up.

This rule finds methods with a ``++boolean++`` that's used to determine which path to take through the method.

=== Noncompliant code example

[source,text,typescript]
----
function feedHuman(name: string) {
  offerSushi(name);
}

function feedAnimal(name: string) {
  offerCarrot(name);
}


function feed(name: string, isHuman: boolean) {
  if (isHuman) {
    feedHuman(name);
  } else {
    feedAnimal(name);
  }
}

// ...
function handleFeeding() {
  feed('Max', false); // does this mean not to feed Max?
}
----

=== Compliant solution

Instead, use types or object properties so that the intent is clear.

[source,text,typescript]
----
function feed(name: string, entityType: 'human' | 'animal') {
  if (entityType === 'human') {
    feedHuman(name);
  } else {
    feedAnimal(name);
  }
}

function handleFeeding() {
  feed('Max', 'animal');
}
----
[source,text,javascript]
----
function feed(name: string, { human = true } ) {
  if (human) {
    feedHuman(name);
  } else {
    feedAnimal(name);
  }
}

function handleFeeding() {
  feed('Max', { human: false });
}
----
