== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,javascript]
----
var a1 = b + c; // This is a trailing comment that can be very very long
----

=== Compliant solution

[source,javascript]
----
// This very long comment is better placed before the line of code
var a2 = b + c;
----

== Resources
=== Documentation

* https://eslint.org[eslint] - Rule https://eslint.org/docs/latest/rules/line-comment-position[line-comment-position]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../parameters.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
