== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,cpp]
----
class MyClass {
public :
  static void Mymethod() {
    // ...
  }
};

MyClass* pmyclass = new MyClass();
pmyclass->Mymethod(); // Noncompliant
----

=== Compliant solution

[source,cpp]
----
class MyClass {
public :
  static Mymethod() {
    // ...
  }
};

Myclass::Mymethod();
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Replace "[.|->]" with "::" for access to "XXX".


'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
