include::../description.adoc[]

== Noncompliant Code Example

[source,cpp]
----
class MyClass {
public :
  static void Mymethod() {
    // ...
  }
};

MyClass* pmyclass = new MyClass();
pmyclass->Mymethod(); // Noncompliant
----

== Compliant Solution

[source,cpp]
----
class MyClass {
public :
  static Mymethod() {
    // ...
  }
};

Myclass::Mymethod();
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
