<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>While you can perform arithmetic on a <code>time_t</code> type, you may not get the results you expect from the operation because the way that a time is encoded within the type is unspecified. Therefore there is no safe way to manually perform arithmetic on <code>time_t</code> variables despite the fact that the ISO C standard defines <code>time_t</code> as an "arithmetic type". The relevant function calls should be used instead.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">int max_seconds = 30;
time_t start = time(NULL);

while (time(NULL) &lt; start + max_seconds) {  // Noncompliant
  //...
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">int max_seconds = 30;
time_t start = time(NULL);
time_t now = start;

while (difftime(now, start) &lt; max_seconds) {
  now = time(NULL);
  //...
}</code></pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p><a href="https://wiki.sei.cmu.edu/confluence/x/09UxBQ">CERT, MSC05-C.</a> - Do not manipulate <code>time_t</code> typed values directly</p>
</li>
</ul>
</div>
<hr>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Use the correct function to perform this calculation.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_19_sep_2014_133933_freddy_mallet_wrote">on 19 Sep 2014, 13:39:33 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>@Ann, I would improve the "Noncompliant Code Example" because when we don&#8217;t know that&#8217;s the type of time_limit variable is int, the example is meaningless.</p>
</div>
</div>
</div>
</div>