include::../description.adoc[]

== Noncompliant Code Example

----
fun doSomething() {
    var name = ""
    // ...
    name = name
}
----

== Compliant Solution

----
fun doSomething() {
    var name = ""
    // ...
    this.name = name
}
----

include::../see.adoc[]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
