== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,kotlin]
----
fun doSomething() {
    var name = ""
    // ...
    name = name
}
----

=== Compliant solution

[source,kotlin]
----
fun doSomething() {
    var name = ""
    // ...
    this.name = name
}
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
