== Why is this an issue?

Not using a ``++DO ... END++`` compound statement can be error-prone.

=== Noncompliant code example

[source,pli]
----
foo: proc options(main);
  declare i fixed decimal init (0);

  if i = 42 then
    put list ('The answer is... '); /* Noncompliant; should be enclosed in DO ... END*/
    put list ('42!'); /* Always unconditionally be executed! Indentation level is misleading. */
end;
----

=== Compliant solution

[source,pli]
----
foo: proc options(main);
  declare i fixed decimal init (0);

  if i = 42 then do;
    put list ('The answer is... '); /* Compliant */
    put list ('42!');
  end;
end;
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Add a "DO ... END" to this statement to this "xxx".


'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
