<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>It is undefined behaviour if, for a set of template-arguments, an implicit instantiation is generated by the compiler, and a partial or explicit specialization is declared or defined elsewhere in the program that would match the set of template-arguments.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">// tmpl.h
template &lt;typename T&gt; void bad_tmpl () {}

// tmpl.cc
#include "tmpl.h"
template &lt;&gt; void bad_tmpl&lt;int32_t&gt; () {} // Noncompliant, should be moved to tmpl.h

// f.cc
#include &lt;tmpl.h&gt;
void f ()
{
  bad_tmpl&lt;int32_t&gt; (); // implicit instantiation of primary. explicit instantiation in tmpl.cc would have been used if it were visible.
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">// tmpl.h
template &lt;typename T&gt; void good_tmpl () {}
template &lt;&gt; void bad_tmpl&lt;int32_t&gt; () {} // Compliant

// f.cc
#include &lt;tmpl.h&gt;
void f ()
{
  good_tmpl&lt;int32_t&gt; (); // specialization of good_tmpl&lt;int32_t&gt; is used
}</code></pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p>MISRA C&#43;&#43;:2008, 14-7-3</p>
</li>
</ul>
</div>
<hr>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Move this specialization to "filename", with the primary template.</p>
</div>
</div>
</div>
</div>