``++__all__++`` is used to define the list of module's names that should be imported when ``++from package import *++`` is used. For that reason, it may only contain strings.


== Noncompliant Code Example

[source,python]
----
class MyClass:
    pass

__all__ = [
    MyClass  # Noncompliant
]
----


== Compliant Solution

[source,python]
----
class MyClass:
    pass

__all__ = [
    "MyClass"
]
----


== See

* https://docs.python.org/3/tutorial/modules.html#importing-from-a-package[Python documentation  - Importing * From a Package]



ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

include::highlighting.adoc[]

'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
