== Why is this an issue?

ECMAScript 2015 introduced the ``++let++`` and ``++const++`` keywords for block-scope variable declaration. Using ``++const++`` creates a read-only (constant) variable.


The distinction between the variable types created by ``++var++`` and by ``++let++`` is significant, and a switch to ``++let++`` will help alleviate many of the variable scope issues which have caused confusion in the past. 


Because these new keywords create more precise variable types, they are preferred in environments that support ECMAScript 2015. However, some refactoring may be required by the switch from ``++var++`` to ``++let++``, and you should be aware that they raise ``++SyntaxError++``s in pre-ECMAScript 2015 environments.


This rule raises an issue when ``++var++`` is used instead of ``++const++`` or ``++let++``.


=== Noncompliant code example

[source,javascript]
----
var color = "blue";
var size = 4;
----


=== Compliant solution

[source,javascript]
----
const color = "blue";
let size = 4;
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Replace "var" with "let" or "const"


=== Highlighting

"var"


'''
== Comments And Links
(visible only on this page)

=== is duplicated by: S1252

=== is related to: S994

=== on 19 Jan 2016, 18:06:26 Elena Vilchik wrote:
\[~ann.campbell.2] Could you add to the description that ``++let++`` statement declares a *block scope* local variable?

=== on 19 Jan 2016, 18:08:34 Elena Vilchik wrote:
\[~ann.campbell.2] WDYT about "creates a read-only variable" instead of "creates a true, unupdatable constant"?  

=== on 19 Jan 2016, 18:20:32 Elena Vilchik wrote:
\[~ann.campbell.2] I would change severity on ``++major++`` and replace ``++confusing++`` on ``++bad-practice++`` or even ``++obsolete++``.

=== on 20 Jan 2016, 09:30:28 Elena Vilchik wrote:
\[~ann.campbell.2] looks good! But IMO code sample in description is excessive, and what you can understand from this code sample is that "let" fails with error but "var" is save and works smoothly :)

=== on 20 Jan 2016, 13:02:43 Ann Campbell wrote:
okay [~elena.vilchik]

endif::env-github,rspecator-view[]
