<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Access modifiers define which classes can access properties, variables, methods, and other classes. If an access modifier is not specified, the access level defaults to <code>internal</code>, which grants access to all classes in the same package. This may be what is intended, but it should be specified explicitly to avoid confusion.</p>
</div>
<div class="paragraph">
<p>Available access modifiers are:</p>
</div>
<div class="ulist">
<ul>
<li>
<p><code>internal</code> - access allowed within the same package</p>
</li>
<li>
<p><code>private</code> - access allowed only within the same class</p>
</li>
<li>
<p><code>protected</code> - access allowed to the class and its child classes</p>
</li>
<li>
<p><code>public</code> - unfettered access by all</p>
</li>
</ul>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-flex" data-lang="flex">function checkResources():Boolean {
  ...
  return true;
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-flex" data-lang="flex">public function checkResources():Boolean {
  ...
  return true;
}</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Explicitly declare the visibility of method "XXXX".</p>
</div>
</div>
</div>
</div>