<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Explicitly declaring method visibility provides clarity and improves code readability. This leads to:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>Simplifying access to and use of the methods for other developers, enhancing the maintainability of the codebase.</p>
</li>
<li>
<p>Promoting the principle of encapsulation, allowing you to control the accessibility of methods and prevent unintended access or modifications from external code.</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>By explicitly stating the visibility, you establish a clear contract for how the methods should be interacted with, making it easier to understand and reason about the behavior of the code.</p>
</div>
<div class="paragraph">
<p>Available access modifiers are:</p>
</div>
<div class="ulist">
<ul>
<li>
<p><code>private</code> - access allowed only within the same class</p>
</li>
<li>
<p><code>protected</code> - access allowed to the class and its child classes</p>
</li>
<li>
<p><code>public</code> - unfettered access by all (default)</p>
</li>
</ul>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-php" data-lang="php">function foo(){...}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-php" data-lang="php">private function foo(){...}</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p><a href="https://www.php.net/manual/en/language.oop5.visibility.php">PHP Manual - Visibility</a></p>
</li>
<li>
<p><a href="https://www.w3schools.com/php/php_oop_access_modifiers.asp">W3Schools - Access Modifiers</a></p>
</li>
</ul>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Explicitly declare the visibility of method "XXXX".</p>
</div>
</div>
</div>
</div>