include::../description.adoc[]

== Noncompliant Code Example

With the default regular expression ``++^[a-z][a-zA-Z0-9]*$++``:

----
void doSomething(int my_param) {
  int LOCAL;
  ...
}
----

== Compliant Solution

----
void doSomething(int myParam) {
  int local;
  ...
}
----

== Exceptions

This rule ignores ``++const++`` variables.

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::parameters.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
