== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

With the default regular expression ``++^[a-z][a-zA-Z0-9]*$++``:

[source,cpp]
----
void doSomething(int my_param) {
  int LOCAL;
  ...
}
----

=== Compliant solution

[source,cpp]
----
void doSomething(int myParam) {
  int local;
  ...
}
----

=== Exceptions

This rule ignores ``++const++`` variables.

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

=== Parameters

.format
****
_STRING_

----
^[a-z][a-zA-Z0-9]*$
----

Regular expression used to check the names against.
****


'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
