:identifier_capital_plural: Local variables and function parameters
:identifier: local variable and function parameter
:identifier_plural: local variables and function parameters
:identifier_or: local variable or function parameter
:regex: ^[a-z][a-zA-Z0-9]*$

include::../introduction.adoc[]

include::../why-is-this-an-issue.adoc[]

include::../what-is-the-potential-impact.adoc[]

=== Exceptions

This rule ignores ``++const++`` variables.

include::../how-to-fix-it.adoc[]

=== Code examples

==== Noncompliant code example

With the default regular expression ``{regex}``:

[source,cpp,diff-id=1,diff-type=noncompliant]
----
void doSomething(int my_param) {
  int LOCAL;
  ...
}
----

==== Compliant solution

[source,cpp,diff-id=1,diff-type=compliant]
----
void doSomething(int myParam) {
  int local;
  ...
}
----

== Resources

=== Documentation

* Geeksforgeeks - https://www.geeksforgeeks.org/naming-convention-in-c/[Naming convention in C++]
* Wikipedia - https://en.wikipedia.org/wiki/Naming_convention_(programming)[Naming Convention (programming)]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../parameters.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
