<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p><code>SELECT INTO TABLE</code> is much more efficient than <code>SELECT ... ENDSELECT</code>. <code>SELECT INTO TABLE</code> needs more memory to hold the result set, but in normal situations, this is not a concern. When memory is a concern, the result set can be divided into smaller sets.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-abap" data-lang="abap">SELECT * FROM T006 INTO X006_WA.
  ...
ENDSELECT.</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-abap" data-lang="abap">SELECT * FROM T006 INTO TABLE X006.
LOOP AT X006 INTO X006_WA.
  ...
ENDLOOP.</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Refactor this select to use "SELECT INTO TABLE int1 + LOOP AT int1".</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_2_dec_2014_143921_ann_campbell_wrote">on 2 Dec 2014, 14:39:21 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~freddy.mallet] I&#8217;m confused by this description. <code>SELECT INTO TABLE</code> is more efficient but it needs more memory&#8230;&#8203;?</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_4_dec_2014_094122_freddy_mallet_wrote">on 4 Dec 2014, 09:41:22 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>Indeed @Ann:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>In one case, we we&#8217;re going to consume less memory but the execution time might be really bigger</p>
</li>
<li>
<p>In the other case, we start by quickly loading all information in memory to then process it</p>
</li>
</ul>
</div>
</div>
</div>
</div>