<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when list comprehensions are used as parameters to <code>any()</code> or <code>all()</code> instead of generator expressions as this prevents <code>any()</code> or <code>all()</code> from short-circuiting.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Using a list comprehension inside <code>any()</code> or <code>all()</code> forces the entire list to be created in memory before the check begins. This prevents the short-circuiting behavior that these functions are designed to leverage, where <code>any()</code> stops at the first <code>True</code> and <code>all()</code> stops at the first <code>False</code>.</p>
</div>
<div class="paragraph">
<p>Using a generator expression provides the same functionality while preserving the short-circuiting behavior of these functions. This could save both processing time and memory, especially for large iterables or when the condition has side effects or is computationally expensive.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Use a generator expression instead of a list comprehension.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">numbers = [1, 5, 0, 10]
res_all = all([x &gt; 2 for x in numbers])  # Noncompliant: will evaluate all numbers instead of stopping at "5"</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">numbers = [1, 5, 0, 10]
res_all = all(x &gt; 2 for x in numbers)  # Compliant</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Python Wiki - <a href="https://wiki.python.org/moin/Generators">Generators</a></p>
</li>
<li>
<p>Python Documentation - <a href="https://docs.python.org/3/glossary.html#term-generator">Generator Glossary Entry</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>