include::../description.adoc[]

== Noncompliant Code Example

----
private void DoSomething()
{
    // ...
    Console.WriteLine("so far, so good..."); // Noncompliant
    // ...
}
----

== Exceptions

The following are ignored by this rule:

* Console Applications
* Calls in methods decorated with ``++[Conditional ("DEBUG")]++``
* Calls included in DEBUG preprocessor branches (``++#if DEBUG++``)


== See

* https://owasp.org/Top10/A09_2021-Security_Logging_and_Monitoring_Failures/[OWASP Top 10 2021 Category A9] - Security Logging and Monitoring Failures

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
