<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when an iterable is reversed using slicing, like <code>[::-1]</code>, before being passed to <code>set()</code>, <code>sorted()</code>, or <code>reversed()</code>.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Reversing an iterable using slicing, likle <code>[::-1]</code>, before passing it to <code>set()</code>, <code>sorted()</code>, or <code>reversed()</code> is unnecessary and inefficient. The slicing operation creates a new copy of the iterable in reverse order, which is not needed for the following reasons:</p>
</div>
<div class="ulist">
<ul>
<li>
<p><code>set()</code>: The order of elements in a set is inherently undefined, so reversing the iterable before creating the set has no effect on the final set, and it introduces unnecessary computation.</p>
</li>
<li>
<p><code>sorted()</code>:  The <code>sorted()</code> function has a <code>reverse</code> parameter that provides a more efficient way to sort in descending order. Using slicing to reverse the result of <code>sorted()</code> is less efficient and less readable.</p>
</li>
<li>
<p><code>reversed()</code>: Applying <code>reversed()</code> twice on the same iterable effectively returns the original iterable, if it supports direct iteration. If the iterable is a one-time iterator, then you will need to create an iterator from the original iterable using <code>iter(iterable)</code>. Using slicing adds unnecessary overhead.</p>
</li>
</ul>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>To fix these issues remove the redundant slicing operation:
* <code>set(iterable[::-1])</code>:  Replace with <code>set(iterable)</code>.
* <code>sorted(iterable)[::-1]</code>: Replace with <code>sorted(iterable, reverse=True)</code>.
* <code>reversed(iterable[::-1])</code>: Replace with <code>iterable</code> if it supports direct iteration, or <code>iter(iterable)</code> if it is a one-time iterator.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">iterable = [1, 3, 2]
result = set(iterable[::-1]) # Noncompliant</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">iterable = [1, 3, 2]
result = set(iterable)</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Python Documentation - <a href="https://docs.python.org/3/library/functions.html#sorted">sorted</a></p>
</li>
<li>
<p>Python Documentation - <a href="https://docs.python.org/3/library/functions.html#reversed">reversed</a></p>
</li>
<li>
<p>Python Documentation - <a href="https://docs.python.org/3/library/functions.html#func-set">set</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>