== Why is this an issue?


=== Noncompliant code example

[source,go]
----
var i, j int = 1, 2
----

=== Compliant solution

[source,go]
----
var i int = 1
var j int = 2
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
