== How to fix it in excon

=== Code examples

==== Noncompliant code example

[source,ruby,diff-id=1,diff-type=noncompliant]
----
excon = Excon.new("https://www.example.com", ciphers: "DEFAULT@SECLEVEL=0") # Noncompliant
response = excon.get
----

==== Compliant solution

[source,ruby,diff-id=1,diff-type=compliant]
----
excon = Excon.new("https://www.example.com")
response = excon.get
----


=== How does this work?

include::../../common/fix/fix.adoc[]

include::../../common/fix/openssl-seclevels.adoc[]
