== How to fix it in Faraday

=== Code examples

==== Noncompliant code example

[source,ruby,diff-id=11,diff-type=noncompliant]
----
ssl_options = {
    ciphers: "DEFAULT@SECLEVEL=0" # Noncompliant
    # Optional parameters
}
conn = Faraday.new(url: "https://www.example.com", ssl: ssl_options)
response = conn.get("/")
----

==== Compliant solution

[source,ruby,diff-id=11,diff-type=compliant]
----
ssl_options = {
    # Optional parameters
}
conn = Faraday.new(url: "https://www.example.com", ssl: ssl_options)
response = conn.get("/")
----


=== How does this work?

include::../../common/fix/fix.adoc[]

include::../../common/fix/openssl-seclevels.adoc[]
