== How to fix it in rest-client

=== Code examples

==== Noncompliant code example

[source,ruby,diff-id=111,diff-type=noncompliant]
----
client = RestClient::Resource.new("https://www.example.com", ssl_ciphers: "DEFAULT@SECLEVEL=0") # Noncompliant
response = client.get
----

==== Compliant solution

[source,ruby,diff-id=111,diff-type=compliant]
----
client = RestClient::Resource.new("https://www.example.com")
response = client.get
----


=== How does this work?

include::../../common/fix/fix.adoc[]

include::../../common/fix/openssl-seclevels.adoc[]
