== How to fix it in Ruby Standard Library

=== Code examples

==== Noncompliant code example

[source,ruby,diff-id=1111,diff-type=noncompliant]
----
http = Net::HTTP.new("https://www.example.com")
http.use_ssl = true
http.ciphers = "DEFAULT@SECLEVEL=0" # Noncompliant
response = http.get("/")
----

==== Compliant solution

[source,ruby,diff-id=1111,diff-type=compliant]
----
http = Net::HTTP.new("https://www.example.com")
http.use_ssl = true
response = http.get("/")
----


=== How does this work?

include::../../common/fix/fix.adoc[]

include::../../common/fix/openssl-seclevels.adoc[]
