<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p><code>typedef</code>s are a convenient way of encapsulating complex types in simpler names. However, <code>typedef</code>ing a pointer type is problematic because when <code>const</code> is applied to the defined type, it makes the pointer constant, not the pointed-to value. Even when that is what&#8217;s intended, maintainers may misunderstand.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">struct book {
  int isbn;
  int publishYear;
};
typedef book;
typedef *pBook;  // Noncompliant

void mangleBook(const pBook pb) {
  pb-&gt;isbn = 4;  // this is legal; its the pointer, not the book that's const
  pb-&gt;publishYear=1900;
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_exceptions">Exceptions</h3>
<div class="paragraph">
<p>Function pointers are ignored by this rule.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<hr>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_22_jun_2015_113055_ann_campbell_wrote">on 22 Jun 2015, 11:30:55 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>Why assign this to me?</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_22_jun_2015_152101_evgeny_mandrikov_wrote">on 22 Jun 2015, 15:21:01 Evgeny Mandrikov wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] because counterpart CPP-927 was assigned to you.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_23_jun_2015_114720_ann_campbell_wrote">on 23 Jun 2015, 11:47:20 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~evgeny.mandrikov] I&#8217;ve added an exception text. I don&#8217;t see the need to add a code sample to the exception.</p>
</div>
</div>
</div>
</div>