== Why is this an issue?

include::../rationale.adoc[]

include::../impact.adoc[]

== How to fix it

=== Code examples

include::../common/fix/code-rationale.adoc[]

==== Noncompliant code example

[source,flex]
----
var val:Number = doCalculation();
trace("Calculation result: " + val); // Noncompliant
----

=== How does this work?

The ``++trace()++`` function outputs debug statements, which can be read by
anyone with a debug version of the Flash player. It should not be used to
output sensitive information.

== Resources

include::../common/resources/standards.adoc[]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

endif::env-github,rspecator-view[]
