== How to fix it in Blazor

Implement proper error handling by reporting customized error messages that do not contain a detailed stack trace. Log the exception stack trace if needed.

=== Code examples

==== Noncompliant code example

[source,csharp,diff-id=2,diff-type=noncompliant]
----
@page "/"

<p>@Content</p>  <!-- Noncompliant -->

@code {
    private String Content = "";

    protected override void OnInitialized()
    {
        try
        {
            throw new InvalidOperationException("");
        }
        catch (Exception ex)
        {
            Content = ex.StackTrace;
        }
    }
}
----

==== Compliant solution

[source,csharp,diff-id=2,diff-type=compliant]
----
@page "/"
@using Microsoft.Extensions.Logging
@inject ILogger<Example> Logger

<p>Internal Server Error</p>

@code {
    protected override void OnInitialized()
    {
        try
        {
            throw new InvalidOperationException("");
        }
        catch (Exception ex)
        {
            Logger.LogError(ex.StackTrace);
        }
    }
}
----
