== How to fix it in ASP.NET

Implement proper error handling by reporting customized error messages that do not contain a detailed stack trace. Log the exception stack trace if needed.

=== Code examples

==== Noncompliant code example

[source,vbnet,diff-id=1,diff-type=noncompliant]
----
<ApiController>
<Route("/")>
Public Class StacktraceController
    Inherits ControllerBase

    <HttpGet("Exception")>
    Public Function ExceptionEndpoint() As String
        Try
            Throw New InvalidOperationException("ExceptionMessage")
        Catch ex As Exception
            Return ex.StackTrace ' Noncompliant
        End Try
        Return "Ok"
    End Function
End Class
----

==== Compliant solution

[source,vbnet,diff-id=1,diff-type=compliant]
----
<ApiController>
<Route("/")>
Public Class StacktraceController
    Inherits ControllerBase

    Private ReadOnly Logger As ILogger(Of StacktraceController)

    <HttpGet("Exception")>
    Public Function ExceptionEndpoint() As String
        Try
            Throw New InvalidOperationException("ExceptionMessage")
        Catch Ex As Exception
            Logger.LogError(Ex.StackTrace)
        End Try
        Return "Ok"
    End Function
End Class
----
