<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Catching <code>ExceptionGroup</code> with <code>except*</code> will raise a <code>TypeError</code>.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Python 3.11 introduced <code>except*</code> and <code>ExceptionGroup</code>, making it possible to handle and raise multiple unrelated exceptions simultaneously.</p>
</div>
<div class="paragraph">
<p>In the example below, we gather multiple exceptions in an <code>ExceptionGroup</code>. This <code>ExceptionGroup</code> is then caught by a single except block:</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">try:
  exception_group = ExceptionGroup("Files not found", [FileNotFoundError("file1.py"), FileNotFoundError("file2.py")])

  raise exception_group

except ExceptionGroup as exceptions:
  # Do something with all the exceptions
  pass</code></pre>
</div>
</div>
<div class="paragraph">
<p>To handle differently each type of exceptions present in an <code>ExceptionGroup</code>, we have to use the <code>except*</code> keyword.</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">try:
  exception_group = ExceptionGroup("Operation errors", [ValueError("Value bigger than 100"), TypeError("Type str is not allowed")])

  raise exception_group

except* ValueError as v:
  # Do something with only ValueErrors
  pass

except* TypeError as t:
  # Do something with only TypeErrors
  pass</code></pre>
</div>
</div>
<div class="paragraph">
<p>While it is possible to catch the <code>ExceptionGroup</code> and <code>BaseExceptionGroup</code> types with <code>except</code>, a <code>TypeError</code> will be raised when this is done with <code>except*</code>.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Make sure to use <code>except</code> when catching ExceptionGroup errors.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">try:
    ...
except* ExceptionGroup:  # Noncompliant: TypeError
    pass

try:
    ...
except* (TypeError, ExceptionGroup):  # Noncompliant: TypeError
    pass</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">try:
    ...
except ExceptionGroup:
    pass

try:
    ...
except (TypeError, ExceptionGroup):
    pass</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p><a href="https://peps.python.org/pep-0654/#forbidden-combinations">PEP-654</a> - Forbidden combinations</p>
</li>
</ul>
</div>
</div>
</div>
</div>