== How to fix it in Dom4j

=== Code examples

include::../../common/fix/code-rationale.adoc[]

==== Noncompliant code example

[source,java,diff-id=1,diff-type=noncompliant]
----
SAXReader xmlReader = new SAXReader(); // Noncompliant
----

==== Compliant solution

[source,java,diff-id=1,diff-type=compliant]
----
SAXReader xmlReader = new SAXReader();
xmlReader.setFeature("http://apache.org/xml/features/disallow-doctype-decl", true);
----

=== How does this work?

include::../../common/fix/xxe.adoc[]
