== How to fix it in Jdom2

=== Code examples

include::../../common/fix/code-rationale.adoc[]

==== Noncompliant code example

[source,java,diff-id=1,diff-type=noncompliant]
----
SAXBuilder builder = new SAXBuilder(); // Noncompliant
----

==== Compliant solution

[source,java,diff-id=1,diff-type=compliant]
----
SAXBuilder builder = new SAXBuilder(); 
builder.setProperty(XMLConstants.ACCESS_EXTERNAL_DTD, "");
builder.setProperty(XMLConstants.ACCESS_EXTERNAL_SCHEMA, "");
----

=== How does this work?

include::../../common/fix/validation.adoc[]
