include::../summary.adoc[]

== Why is this an issue?

include::../description.adoc[]

== How to fix it

=== Code examples

==== Noncompliant code example

[source,bash,diff-id=1,diff-type=noncompliant]
----
#!/bin/bash

rm *
----

==== Compliant solution

[source,bash,diff-id=1,diff-type=compliant]
----
#!/bin/bash

rm ./*
rm -- *
----

== Resources

include::../resources.adoc[]

ifdef::env-github,rspecator-view[]
'''
== Implementation Specification
(visible only on this page)

=== Message

include::../message.adoc[]

=== Highlighting

include::../highlighting.adoc[]

'''
endif::env-github,rspecator-view[]
