== How to fix it

When all actions in the controllers have an absolute route, the route should be moved at the controller level.

When only some of the actions in the controller have absolute routes, the absolute route should be extracted into a dedicated controller.

=== Code examples

==== Noncompliant code example

include::{language}/noncompliant-code-complete.adoc[]

==== Compliant solution

include::{language}/compliant-code-complete.adoc[]

==== Noncompliant code example

include::{language}/noncompliant-code-partial.adoc[]

==== Compliant solution

include::{language}/compliant-code-partial.adoc[]
