Characters like ``++'é'++`` can be expressed either as a single code point or as a cluster of the letter ``++'e'++`` and a combining accent mark. Without the ``++CANON_EQ++`` flag, a regex will only match a string in which the characters are expressed in the same way.


== Noncompliant Code Example

----
String s = "e\u0300";
Pattern p = Pattern.compile("é|ë|è"); // Noncompliant
System.out.println(p.matcher(s).replaceAll("e")); // print 'è'
----


== Compliant Solution

----
String s = "e\u0300";
Pattern p = Pattern.compile("é|ë|è", Pattern.CANON_EQ);
System.out.println(p.matcher(s).replaceAll("e")); // print 'e'
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

include::highlighting.adoc[]

endif::env-github,rspecator-view[]
