<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This is an issue when the <code>message</code> option is placed at the top level of a <code>validates</code> method call instead of being nested within a specific validator&#8217;s options hash.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>In Rails ActiveRecord validations, custom error messages must be specified within the options hash of the specific validator they apply to. When you place a <code>message</code> option at the top level of a <code>validates</code> call, Rails treats it as an unknown validator and raises an "Unknown validator: 'message'" error.</p>
</div>
<div class="paragraph">
<p>This happens because Rails expects the <code>validates</code> method to receive validator names (like <code>presence</code>, <code>length</code>, <code>uniqueness</code>) as keys, with their respective options as values. The <code>message</code> option is not a validator itself, but rather a configuration option that belongs to a specific validator.</p>
</div>
<div class="paragraph">
<p>When you write <code>validates :title, presence: true, message: "Custom message"</code>, Rails interprets this as trying to use both a <code>presence</code> validator and a <code>message</code> validator, which doesn&#8217;t exist. The correct approach is to nest the message within the validator&#8217;s options: <code>validates :title, presence: { message: "Custom message" }</code>.</p>
</div>
<div class="paragraph">
<p>This syntax error prevents your Rails application from starting and can be confusing for developers who are learning Rails validation syntax or transitioning from older validation patterns.</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="paragraph">
<p>This issue causes a runtime error that prevents the Rails application from starting. The application will fail to load with an "Unknown validator" error, making it impossible to run the application until the validation syntax is corrected.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it_in_rails">How to fix it in Rails</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Move the <code>message</code> option from the top level into the specific validator&#8217;s options hash. If the validator currently uses a boolean value (like <code>presence: true</code>), change it to a hash containing the message.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-ruby" data-lang="ruby">class Story &lt; ActiveRecord::Base
  validates :title, presence: true, message: "Story title is required" # Noncompliant
end</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-ruby" data-lang="ruby">class Story &lt; ActiveRecord::Base
  validates :title, presence: { message: "Story title is required" }
end</code></pre>
</div>
</div>
<div class="paragraph">
<p>When using multiple validators, ensure each validator that needs a custom message has the message nested within its own options hash.</p>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_code_examples_2">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example_2">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-ruby" data-lang="ruby">class Story &lt; ActiveRecord::Base
  validates :content, presence: true, length: { minimum: 10 }, message: "Invalid content" # Noncompliant
end</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution_2">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-ruby" data-lang="ruby">class Story &lt; ActiveRecord::Base
  validates :content, presence: { message: "Content cannot be blank" },
                     length: { minimum: 10, message: "Content too short" }
end</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Rails Active Record Validations Guide - <a href="https://guides.rubyonrails.org/active_record_validations.html#message">Official Rails documentation on validation messages and their proper syntax</a></p>
</li>
<li>
<p>Rails API Documentation - validates - <a href="https://api.rubyonrails.org/classes/ActiveModel/Validations/ClassMethods.html#method-i-validates">API documentation for the validates method and its options</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>