== Why is this an issue?

include::description.adoc[]

=== Noncompliant code example

[source,text]
----
public class BaseClass
{
    public virtual void MyMethod(int i = 1)
    { 
        Console.WriteLine(i);
    }
}

public class DerivedClass : BaseClass
{
    public override void MyMethod(int i = 1)
    {
        // ... 
        base.MyMethod(); // Noncompliant; caller's value is ignored
    }

    static int Main(string[] args) 
    {
        DerivedClass dc = new DerivedClass();
        dc.MyMethod(12);  // prints 1
    }
}
----


=== Compliant solution

[source,text]
----
public class BaseClass
{
    public virtual void MyMethod(int i = 1)
    { 
        Console.WriteLine(i);
    }
}

public class DerivedClass : BaseClass
{
    public override void MyMethod(int i = 1)
    {
        // ... 
        base.MyMethod(i);
    }

    static int Main(string[] args) 
    {
        DerivedClass dc = new DerivedClass();
        dc.MyMethod(12);  // prints 12
    }
}
----

