<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>@VisibleForTesting can be used to mark methods, fields and classes whose visibility restrictions have been relaxed more than necessary for the API to allow for easier unit testing.</p>
</div>
<div class="paragraph">
<p>Access to such methods, fields and classes only possible thanks to this relaxed visibility is fine for test code, but it should be avoided in production code. In production code these methods should be treated as if they are private.</p>
</div>
<div class="paragraph">
<p>Supported framework:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>Guava: <code>com.google.common.annotations.VisibleForTesting</code></p>
</li>
<li>
<p>AssertJ: <code>org.assertj.core.util.VisibleForTesting</code></p>
</li>
<li>
<p>Android: <code>androidx.annotation.VisibleForTesting</code></p>
</li>
<li>
<p>Apache Flink: <code>org.apache.flink.annotation.VisibleForTesting</code></p>
</li>
</ul>
</div>
<div class="paragraph">
<p>or any other annotation named <code>VisibleForTesting</code></p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">/** src/main/java/MyObject.java */

@VisibleForTesting String foo;

/** src/main/java/Service.java */

new MyObject().foo; // Noncompliant, foo is accessed from production code</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">/** src/main/java/MyObject.java */

@VisibleForTesting String foo;

/** src/test/java/MyObjectTest.java */

new MyObject().foo; // Compliant, foo is accessed from test code</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove this usage of {identifier name}, it is annotated with @VisibleForTesting and should not be accessed from production code.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>Visible for testing identifier name (methods or fields)</p>
</div>
</div>
</div>
</div>