== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,vbnet]
----
Select Case variable
    Case 0
        doSomething()
    Case Else
        doSomethingElse()
End Select
----

=== Compliant solution

[source,vbnet]
----
If variable = 0 Then
    doSomething()
Else
    doSomethingElse()
End If
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
