<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>When a transmute is placed within an expression that uses eager evaluation (like <code>bool::then_some</code>), it will execute even if a preceding validity check fails. This can result in creating invalid values, potentially leading to undefined behavior.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Use lazy evaluation (for example by replacing <code>then_some</code> with <code>then</code>) and providing a closure that contains the transmute. This ensures the transmute only occurs if the validity check passes.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-rust" data-lang="rust">#[repr(u8)]
enum Opcode {
    Add = 0,
    Sub = 1,
    Mul = 2,
    Div = 3
}

fn int_to_opcode(op: u8) -&gt; Option&lt;Opcode&gt; {
    (op &lt; 4).then_some(unsafe { std::mem::transmute(op) })
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-rust" data-lang="rust">#[repr(u8)]
enum Opcode {
    Add = 0,
    Sub = 1,
    Mul = 2,
    Div = 3
}

fn int_to_opcode(op: u8) -&gt; Option&lt;Opcode&gt; {
    (op &lt; 4).then(|| unsafe { std::mem::transmute(op) })
}</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Clippy Lints - <a href="https://rust-lang.github.io/rust-clippy/master/index.html#eager_transmute" class="bare">https://rust-lang.github.io/rust-clippy/master/index.html#eager_transmute</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>