== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

With the default regular expression ``++^[a-z][a-zA-Z0-9]*$++``:

[source,java]
----
public void doSomething(int my_param) { 
  int LOCAL;    
  ...             
}
----

=== Compliant solution

[source,java]
----
public void doSomething(int myParam) { 
  int local;  
  ...               
}
----

=== Exceptions

Loop counters are ignored by this rule.

----
for (int i_1 = 0; i_1 < limit; i_1++) {  // Compliant
  // ...
}
----
as well as one-character ``++catch++`` variables: 

----
try {
//... 
} catch (Exception e) { // Compliant
}
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::parameters.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
