<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>A chain of <a href="https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/statements/selection-statements#the-if-statement">if/else if</a> statements is evaluated from top to bottom. At most, only one branch will be executed: the first statement with a condition that evaluates to <code>true</code>.
Therefore, duplicating a condition leads to unreachable code inside the duplicated condition block. Usually, this is due to a copy/paste error.</p>
</div>
<div class="paragraph">
<p>The result of such duplication can lead to unreachable code or even to unexpected behavior.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-csharp" data-lang="csharp">if (param == 1)
{
  OpenWindow();
}
else if (param == 2)
{
  CloseWindow();
}
else if (param == 1) // Noncompliant: condition has already been checked
{
  MoveWindowToTheBackground(); // unreachable code
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-csharp" data-lang="csharp">if (param == 1)
{
  OpenWindow();
}
else if (param == 2)
{
  CloseWindow();
}
else if (param == 3)
{
  MoveWindowToTheBackground();
}</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p><a href="https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/statements/selection-statements#the-if-statement">The if statement</a></p>
</li>
</ul>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>This branch duplicates the one on line n.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="ulist">
<ul>
<li>
<p>primary: 2nd duplicate condition</p>
</li>
<li>
<p>secondary: 1st condition</p>
<div class="ulist">
<ul>
<li>
<p>message: Original</p>
</li>
</ul>
</div>
</li>
</ul>
</div>
<div class="paragraph">
<p>Only the characters part of the conditions should be underlined.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_25_mar_2015_160323_tamas_vajk_wrote">on 25 Mar 2015, 16:03:23 Tamas Vajk wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] should I add the See section for C# as well?</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_25_mar_2015_162448_ann_campbell_wrote">on 25 Mar 2015, 16:24:48 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~tamas.vajk] these language-specific subtasks override portions of their parent tasks, and the description field is treated as 5 separate parts: description, Noncompliant Code Example, Compliant Solution, Exceptions, and references (See). There&#8217;s no need to copy/paste the references to this child ticket, but do include them in the implementation&#8217;s rule description.</p>
</div>
</div>
<div class="sect2">
<h3 id="_deprecates_s1950">deprecates: <a data-rspec-id="S1950" class="rspec-auto-link">S1950</a></h3>

</div>
<div class="sect2">
<h3 id="_is_related_to_s1763">is related to: <a data-rspec-id="S1763" class="rspec-auto-link">S1763</a></h3>

</div>
<div class="sect2">
<h3 id="_is_related_to_s901">is related to: <a data-rspec-id="S901" class="rspec-auto-link">S901</a></h3>

</div>
<div class="sect2">
<h3 id="_is_related_to_s2760">is related to: <a data-rspec-id="S2760" class="rspec-auto-link">S2760</a></h3>

</div>
<div class="sect2">
<h3 id="_on_12_feb_2015_072325_nicolas_peru_wrote">on 12 Feb 2015, 07:23:25 Nicolas Peru wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2]I am updating example and message for Java as a switch with twice the same case won&#8217;t compile cf http://docs.oracle.com/javase/specs/jls/se8/html/jls-14.html#jls-14.11</p>
</div>
<div class="paragraph">
<p>"No two of the case constants associated with the switch statement have the same value."</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_12_feb_2015_134806_ann_campbell_wrote">on 12 Feb 2015, 13:48:06 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~nicolas.peru] I&#8217;ve updated the title for Java as well</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_6_apr_2015_132606_evgeny_mandrikov_wrote">on 6 Apr 2015, 13:26:06 Evgeny Mandrikov wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] same for C-Family as for Java, in fact I suppose that majority of languages forbids duplicate "cases" and minority allows, so might be better to do opposite - generalize "if" and specialize "switch"</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_9_apr_2015_161633_ann_campbell_wrote">on 9 Apr 2015, 16:16:33 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~evgeny.mandrikov] laborious cleanup done to make sure each language has the right title and right description details for the language features.</p>
</div>
<div class="paragraph">
<p>| |if|switch w/dup cond|break|</p>
</div>
<div class="paragraph">
<p>|Java|x|-||</p>
</div>
<div class="paragraph">
<p>|C-Family|x|-||</p>
</div>
<div class="paragraph">
<p>|JavaScript|x|x|x|</p>
</div>
<div class="paragraph">
<p>|PHP'|x|x|x|</p>
</div>
<div class="paragraph">
<p>|PL/SQL|x|x|-|</p>
</div>
<div class="paragraph">
<p>|Cobol|x|x|-|</p>
</div>
<div class="paragraph">
<p>|Swift|x|x|-|</p>
</div>
<div class="paragraph">
<p>|Python|x|-||</p>
</div>
<div class="paragraph">
<p>|C#|x|-|</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_10_apr_2015_075236_evgeny_mandrikov_wrote">on 10 Apr 2015, 07:52:36 Evgeny Mandrikov wrote:</h3>
<div class="paragraph">
<p>Thanks [~ann.campbell.2]!</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_9_mar_2018_112719_tibor_blenessy_wrote">on 9 Mar 2018, 11:27:19 Tibor Blenessy wrote:</h3>
<div class="paragraph">
<p>Just a note: for Go compiler may disallow some trivial duplicated conditions in <code>case</code> clauses, but it is possible in general. https://golang.org/ref/spec#Switch_statements</p>
</div>
</div>
</div>
</div>