<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>With the introduction of Markdown support in Javadoc comments in Java 23, developers should prefer Markdown syntax over legacy HTML tags (e.g., <code>&lt;b&gt;</code>, <code>&lt;i&gt;</code>, <code>&lt;code&gt;</code>, <code>&lt;ul&gt;</code>, <code>&lt;li&gt;</code>, etc.) and legacy Javadoc tags (e.g. <code>{@link java.util.HashMap}</code>, <code>{@code Object}</code>).</p>
</div>
<div class="paragraph">
<p>Mixing HTML and Markdown can lead to readability issues and inconsistencies in rendering across tools.
Markdown is generally more readable, less cluttered and, should be preferred over HTML tags when possible.</p>
</div>
<div class="paragraph">
<p>This rule will raise an issue when an HTML tag or a legacy Javadoc tag inside a markdown comment could be replaced with a slimmer, and more cohesive, syntax.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Here is a list of tags that should be replaced with markdown syntax:</p>
</div>
<div class="ulist">
<ul>
<li>
<p><code>&lt;p&gt;</code> should be replaced with 2 new lines</p>
</li>
<li>
<p><code>&lt;code&gt;MyCode&lt;/code&gt;</code>, <code>{@code MyCode}</code> should be replaced with <code>`MyCode`</code></p>
</li>
<li>
<p><code>&lt;i&gt;italic text&lt;/i&gt;</code>, <code>&lt;em&gt;</code> should be replaced with <code>*italic text*</code></p>
</li>
<li>
<p><code>&lt;b&gt;bold text&lt;/b&gt;</code>, <code>&lt;strong&gt;</code> should be replaced with <code>**bold text**</code></p>
</li>
<li>
<p><code>&lt;ul&gt;&lt;li&gt;list item&lt;/li&gt;&lt;/ul&gt;</code>, <code>&lt;ol&gt;&lt;li&gt;</code> should be replaced with <code>* list item</code></p>
</li>
<li>
<p><code>{@link some.java.Class}</code> should be replaced with <code>[some.java.Class]</code></p>
</li>
<li>
<p><code>{@link #equals(Object) equals}</code> should be replaced with <code>[equals][#equals(Object)]</code></p>
</li>
</ul>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">/// This is a markdown Javadoc comment
/// &lt;b&gt;ExampleClass&lt;/b&gt; is a simple utility for &lt;i&gt;demonstration purposes&lt;/i&gt;.
/// &lt;p&gt;
///     Use &lt;code&gt;ExampleClass.run()&lt;/code&gt; to execute the example.
/// &lt;/p&gt;
/// {@link some.java.Class} for more details
public class ExampleClass {
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">/// This is a markdown Javadoc comment
/// **ExampleClass** is a simple utility for *demonstration purposes*.
///
/// Use `ExampleClass.run()` to execute the example.
/// [some.java.Class] for more details
public class ExampleClass {
}</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p><a href="https://openjdk.org/jeps/467">JEP 467: Markdown Documentation Comments</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>