== Why is this an issue?

Shared coding conventions allow teams to collaborate efficiently. To avoid the confusion that can be caused by tangling two coding languages in the same file, inline HTML should be avoided.


=== Noncompliant code example

[source,php]
----
<?php 
$name = "George";
?>
<p> Hello <?php echo $name ?>!</p>
----


=== Exceptions

File having the extension ``++.phtml++`` are ignored by this rule because they are expected to have mixed PHP and HTML.

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Remove the inline HTML in this file.


endif::env-github,rspecator-view[]
