Shared coding conventions allow teams to collaborate efficiently. To avoid the confusion that can be caused by tangling two coding languages in the same file, inline HTML should be avoided.


== Noncompliant Code Example

----
<?php 
$name = "George";
?>
<p> Hello <?php echo $name ?>!</p>
----


== Exceptions

File having the extension ``++.phtml++`` are ignored by this rule because they are expected to have mixed PHP and HTML.

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

endif::env-github,rspecator-view[]
