<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when ActiveRecord model constructors (<code>new</code>, <code>create</code>, <code>build</code>) receive non-hash arguments such as strings, symbols, or multiple individual parameters.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>ActiveRecord model constructors expect a hash of attributes where keys represent attribute names and values represent the corresponding values. When you pass non-hash arguments, Rails cannot process them correctly because it tries to call hash-specific methods like <code>stringify_keys!</code> on the provided arguments.</p>
</div>
<div class="paragraph">
<p>This leads to runtime errors such as:</p>
</div>
<div class="ulist">
<ul>
<li>
<p><code>ArgumentError: When assigning attributes, you must pass a hash as an argument</code></p>
</li>
<li>
<p><code>undefined method 'stringify_keys!' for "SomeString":String</code></p>
</li>
</ul>
</div>
<div class="paragraph">
<p>These errors occur because Rails internally processes the constructor arguments as attribute hashes. When you pass a string, symbol, or other non-hash data type, Rails fails when it attempts to normalize the keys and values.</p>
</div>
<div class="paragraph">
<p>The problem often arises when developers mistakenly pass:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>Direct parameter values instead of structured attribute hashes</p>
</li>
<li>
<p>Individual symbols thinking they represent attribute names</p>
</li>
<li>
<p>String values directly from form parameters without proper structuring</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>This pattern violates Rails conventions and makes the code fragile and error-prone.</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="paragraph">
<p>This issue causes immediate runtime failures when the affected code executes. The application will crash with an <code>ArgumentError</code>, potentially breaking user workflows and causing service disruptions. In web applications, this typically results in 500 Internal Server Error responses, degrading user experience and system reliability.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it_in_rails">How to fix it in Rails</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Replace non-hash arguments with a proper hash structure where keys are attribute names and values are the corresponding data.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-ruby" data-lang="ruby">def create_car
  # Passing a string directly
  @car = Car.new(params[:car])  # params[:car] is "Honda"
end</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-ruby" data-lang="ruby">def create_car
  # Pass a hash with attribute names as keys
  @car = Car.new(name: params[:car])
end</code></pre>
</div>
</div>
<div class="paragraph">
<p>When passing symbols, convert them to proper hash syntax with the symbol as the key.</p>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_code_examples_2">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example_2">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-ruby" data-lang="ruby">def create_document
  # Passing a symbol directly
  @document = Document.new(:name)  # Noncompliant
end</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution_2">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-ruby" data-lang="ruby">def create_document
  # Use hash syntax with the attribute name
  @document = Document.new(name: params[:name])
end</code></pre>
</div>
</div>
<div class="paragraph">
<p>For multiple attributes, use a single hash instead of multiple individual arguments.</p>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_code_examples_3">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example_3">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-ruby" data-lang="ruby">def create_document
  # Multiple non-hash arguments
  @document = Document.new(:name, :title)  # Noncompliant
end</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution_3">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-ruby" data-lang="ruby">def create_document
  # Single hash with multiple attributes
  @document = Document.new(name: params[:name], title: params[:title])
end</code></pre>
</div>
</div>
<div class="paragraph">
<p>When no specific attributes are needed, call the constructor without arguments or with an empty hash.</p>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_code_examples_4">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example_4">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-ruby" data-lang="ruby">def create_document
  # Incorrect: passing empty string
  @document = Document.new("")  # Noncompliant
end</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution_4">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-ruby" data-lang="ruby">def create_document
  # Correct: no arguments or empty hash
  @document = Document.new
  # or
  @document = Document.new({})
end</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>ActiveRecord Basics - Creating Active Record Objects - <a href="https://guides.rubyonrails.org/active_record_basics.html#creating-active-record-objects">Official Rails guide explaining how to properly create ActiveRecord objects with attribute hashes</a></p>
</li>
<li>
<p>ActiveRecord::Base.new Documentation - <a href="https://api.rubyonrails.org/classes/ActiveRecord/Base.html#method-c-new">API documentation for ActiveRecord::Base.new method and its expected parameters</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>