<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Lint attributes on crate imports and <code>use</code> statements have no effect and may indicate a misunderstanding or mistake in the code. This can lead to confusion or unintended behavior.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-rust" data-lang="rust">#[deny(dead_code)]
extern crate foo;

#[forbid(dead_code)]
use foo::bar; // Noncompliant: Lint attributes are ineffective on imports.</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-rust" data-lang="rust">#[allow(unused_imports)]
use foo::baz;

#[allow(unused_imports)]
#[macro_use]
extern crate baz; // Compliant: Example with appropriate usage.</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Clippy Lints - <a href="https://rust-lang.github.io/rust-clippy/master/index.html#useless_attribute" class="bare">https://rust-lang.github.io/rust-clippy/master/index.html#useless_attribute</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>