== Why is this an issue?

Side effects in a module can lead to unpredictable behavior, making the module harder to understand and debug.
They complicate testing by requiring complex setup and teardown procedures, reducing the module's reusability since its
behavior depends on specific states or external systems. Side effects can also introduce concurrency issues, such as
race conditions, and make the codebase harder to maintain and more prone to breaking. Additionally, they increase the
complexity and reduce the readability of the code, while breaking encapsulation and leading to tighter coupling between
modules. Minimizing side effects helps create more predictable, testable, reusable, and maintainable code.

== How to fix it

One should not include loops, if/switch conditions and function calls in modules.

=== Code examples

==== Noncompliant code example

[source,javascript,diff-id=1,diff-type=noncompliant]
----
export const x = 10;
console.log('x is %s');
----


==== Compliant solution

[source,javascript,diff-id=1,diff-type=compliant]
----
export const x = 10;
----