<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>The preprocessor performs two main tasks: Handling preprocessor directives and expanding macros. Trying to do both simultaneously can be problematic: when a macro argument contains something that could be interpreted as a preprocessor directive, the behavior is undefined.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">#define LOG(msg) if(logEnabled) log(msg);
int main()
{
   LOG(
#ifdef __cplusplus // Noncompliant, undefined behavior
   "This library is built in C++ mode"
#else
   "This library is built in C mode"
#endif
    );
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="paragraph">
<p>The solution to avoid using proprocessing directives in a macro argument is often to use them before calling the macro. For instance, the previous code could be rewritten:</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">#define LOG(msg) if(logEnabled) log(msg);
int main()
{
#ifdef __cplusplus
   #define MODE "C++"
#else
   #define MODE "C"
#endif

   LOG("This library is built in " MODE " mode");

#undef MODE
}</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_external_coding_guidelines">External coding guidelines</h3>
<div class="ulist">
<ul>
<li>
<p>MISRA C&#43;&#43;:2023, 19.3.5 - Tokens that look like a preprocessing directive shall not occur within a macro argument</p>
</li>
<li>
<p>MISRA C:2012, 20.6 - Tokens that look like a preprocessing directive shall not occur within a macro argument</p>
</li>
<li>
<p><a href="https://wiki.sei.cmu.edu/confluence/display/c/PRE32-C.+Do+not+use+preprocessor+directives+in+invocations+of+function-like+macros">CERT, PRE32-C.</a> - Do not use preprocessor directives in invocations of function-like macros</p>
</li>
</ul>
</div>
</div>
</div>
</div>