== Why is this an issue?

The use of dynamic memory can lead to out-of-storage run-time failures, which are undesirable.


The built-in ``++new++`` and ``++delete++`` operators, other than the placement versions, use dynamic heap memory. The functions ``++calloc++``, ``++malloc++``, ``++realloc++`` and ``++free++`` also use dynamic heap memory.


There is a range of unspecified, undefined and implementation-defined behaviour associated with dynamic memory allocation, as well as a number of other potential pitfalls. Dynamic heap memory allocation may lead to memory leaks, data inconsistency, memory exhaustion, non-deterministic behaviour, etc.


Note that some implementations may use dynamic heap memory allocation to implement other functions (for example, functions in the library ``++cstring++``). If this is the case, then these functions shall also be avoided.


=== Noncompliant code example

[source,cpp]
----
int *b;
void initialize()
{
  b = (int*) malloc(1024 * sizeof(int)); // Noncompliant, could lead to an out-of-storage run-time failure.
  if (b == 0)
  {
    // handle case when dynamic allocation failed.
  }
}
----


=== Compliant solution

[source,cpp]
----
int b[1024]; // Compliant solution.
----


== Resources

* MISRA C:2004, 20.4 - Dynamic heap memory allocation shall not be used.
* MISRA {cpp} 2008, 18-4-1 - Dynamic heap memory allocation shall not be used.
* MISRA C:2012, 21.3 The memory allocation and deallocation functions of <stdlib.h> shall not be used


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Remove this usage of function "xxx".


'''
== Comments And Links
(visible only on this page)

=== is related to: S5502

=== on 23 Oct 2014, 16:12:15 Ann Campbell wrote:
\[~samuel.mercier] I know you're working from the MISRA descriptions but I have to ask you to add code examples on this one, both Noncompliant and Compliant. If you can't use malloc/alloc/calloc/realloc and you can't use new, then we have to show what you _can_ use.

=== on 23 Oct 2014, 16:19:11 Samuel Mercier wrote:
\[~ann.campbell.2] I just hope our customers won't read the compliant solution, because as soon as they will require a dynamic size this rule will report issues they can't fix :)

=== on 23 Oct 2014, 20:18:18 Ann Campbell wrote:
\[~samuel.mercier] I was really hoping there was some 3rd way to get memory in C that I didn't know about, but my worst fear has been realized. This rule is: Only automatically-allocated memory should be used.

=== on 24 Oct 2014, 15:50:29 Ann Campbell wrote:
\[~samuel.mercier] FYI, I tightened up the description

endif::env-github,rspecator-view[]
