<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when random number generators do not specify a seed parameter.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Data science and machine learning tasks make extensive use of random number generation. It may, for example, be used for:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>Model initialization</p>
<div class="ulist">
<ul>
<li>
<p>Randomness is used to initialize the parameters of machine learning models. Initializing parameters with random values helps to break symmetry and prevents models from getting stuck in local optima during training. By providing a random starting point, the model can explore different regions of the parameter space and potentially find better solutions.</p>
</li>
</ul>
</div>
</li>
<li>
<p>Regularization techniques</p>
<div class="ulist">
<ul>
<li>
<p>Randomness is used to introduce noise into the learning process. Techniques like dropout and data augmentation use random numbers to randomly drop or modify features or samples during training. This helps to regularize the model, reduce overfitting, and improve generalization performance.</p>
</li>
</ul>
</div>
</li>
<li>
<p>Cross-validation and bootstrapping</p>
<div class="ulist">
<ul>
<li>
<p>Randomness is often used in techniques like cross-validation, where data is split into multiple subsets. By using a predictable seed, the same data splits can be generated, allowing for fair and consistent model evaluation.</p>
</li>
</ul>
</div>
</li>
<li>
<p>Hyperparameter tuning</p>
<div class="ulist">
<ul>
<li>
<p>Many machine learning algorithms have hyperparameters that need to be tuned for optimal performance. Randomness is often used in techniques like random search or Bayesian optimization to explore the hyperparameter space. By using a fixed seed, the same set of hyperparameters can be explored, making the tuning process more controlled and reproducible.</p>
</li>
</ul>
</div>
</li>
<li>
<p>Simulation and synthetic data generation</p>
<div class="ulist">
<ul>
<li>
<p>Randomness is often used in techniques such as data augmentation and synthetic data generation to generate diverse and realistic datasets.</p>
</li>
</ul>
</div>
</li>
</ul>
</div>
<div class="paragraph">
<p>To ensure that results are reproducible, it is important to use a predictable seed in this context.</p>
</div>
<div class="paragraph">
<p>The preferred way to do this in <code>numpy</code> is by instantiating a <code>Generator</code> object, typically through <code>numpy.random.default_rng</code>, which should be provided with a seed parameter.</p>
</div>
<div class="paragraph">
<p>Note that a global seed for <code>RandomState</code> can be set using <code>numpy.random.seed</code> or <code>numpy.seed</code>, this will set the seed for <code>RandomState</code> methods such as <code>numpy.random.randn</code>. This approach is, however, deprecated and <code>Generator</code> should be used instead. This is reported by rule <a data-rspec-id="S6711" class="rspec-auto-link">S6711</a>.</p>
</div>
<div class="sect2">
<h3 id="_exception">Exception</h3>
<div class="paragraph">
<p>In contexts that are not related to data science and machine learning, having a predictable seed may not be the desired behavior. Therefore, this rule will only raise issues if machine learning and data science libraries are being used.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it_in_numpy">How to fix it in Numpy</h2>
<div class="sectionbody">
<div class="paragraph">
<p>To fix this issue, provide a predictable seed to the random number generator.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">import numpy as np

def foo():
    generator = np.random.default_rng()  # Noncompliant: no seed parameter is provided
    x = generator.uniform()</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">import numpy as np

def foo():
    generator = np.random.default_rng(42)  # Compliant
    x = generator.uniform()</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it_in_scikit_learn">How to fix it in Scikit-Learn</h2>
<div class="sectionbody">
<div class="paragraph">
<p>To fix this issue, provide a predictable seed to the estimator or the utility function.</p>
</div>
<div class="sect2">
<h3 id="_code_examples_2">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example_2">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">from sklearn.model_selection import train_test_split
from sklearn.datasets import load_iris

X, y = load_iris(return_X_y=True)
X_train, _, y_train, _ = train_test_split(X, y) # Noncompliant: no seed parameter is provided</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution_2">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">from sklearn.model_selection import train_test_split
from sklearn.datasets import load_iris
import numpy as np

rng = np.random.default_rng(42)
X, y = load_iris(return_X_y=True)
X_train, _, y_train, _ = train_test_split(X, y, random_state=rng.integers(1)) # Compliant</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>NumPy documentation - <a href="https://numpy.org/neps/nep-0019-rng-policy.html">NEP 19 RNG Policy</a></p>
</li>
<li>
<p>Scikit-learn documentation - <a href="https://scikit-learn.org/stable/glossary.html#term-random_state">Glossary random_state</a></p>
</li>
<li>
<p>Scikit-learn documentation - <a href="https://scikit-learn.org/stable/common_pitfalls.html#controlling-randomness">Controlling randomness</a></p>
</li>
</ul>
</div>
</div>
<div class="sect2">
<h3 id="_standards">Standards</h3>
<div class="ulist">
<ul>
<li>
<p>STIG Viewer - <a href="https://stigviewer.com/stigs/application_security_and_development/2024-12-06/finding/V-222642">Application Security and Development: V-222642</a> - The application must not contain embedded authentication data.</p>
</li>
</ul>
</div>
</div>
<div class="sect2">
<h3 id="_related_rules">Related rules</h3>
<div class="ulist">
<ul>
<li>
<p><a data-rspec-id="S6711" class="rspec-auto-link">S6711</a> - <code>numpy.random.Generator</code> should be preferred to <code>numpy.random.RandomState</code></p>
</li>
</ul>
</div>
</div>
</div>
</div>