<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>While <code>:&lt;element_type&gt;</code> and <code>[type="&lt;element_type&gt;"]</code> can both be used in jQuery to select elements by their type, <code>[type="&lt;element_type&gt;"]</code> is far faster because it can take advantage of the native DOM <code>querySelectorAll()</code> method in modern browsers.</p>
</div>
<div class="paragraph">
<p>This rule raises an issue when following selectors are used:</p>
</div>
<div class="ulist">
<ul>
<li>
<p><code>:checkbox</code></p>
</li>
<li>
<p><code>:file</code></p>
</li>
<li>
<p><code>:image</code></p>
</li>
<li>
<p><code>:password</code></p>
</li>
<li>
<p><code>:radio</code></p>
</li>
<li>
<p><code>:reset</code></p>
</li>
<li>
<p><code>:text</code></p>
</li>
</ul>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-javascript" data-lang="javascript">var input = $( "form input:radio" ); // Noncompliant</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-javascript" data-lang="javascript">var input = $( "form input[type=radio]" ); // Compliant</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Use the "[type='{0}']" selector here instead of ":{0}".</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_30_apr_2015_072135_ann_campbell_wrote">on 30 Apr 2015, 07:21:35 Ann Campbell wrote:</h3>
<div class="paragraph">
<p><a href="https://api.jquery.com/radio-selector/" class="bare">https://api.jquery.com/radio-selector/</a></p>
</div>
</div>
<div class="sect2">
<h3 id="_on_30_apr_2015_075705_linda_martin_wrote">on 30 Apr 2015, 07:57:05 Linda Martin wrote:</h3>
<div class="paragraph">
<p>Reviewed!</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_6_may_2015_082955_elena_vilchik_wrote">on 6 May 2015, 08:29:55 Elena Vilchik wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] The same rule applies to 6 more selectors, so I updated rule accordingly. Could you verify it?</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_6_may_2015_114330_ann_campbell_wrote">on 6 May 2015, 11:43:30 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>looks fine to me [~elena.vilchik]</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_1_nov_2019_173035_elena_vilchik_wrote">on 1 Nov 2019, 17:30:35 Elena Vilchik wrote:</h3>
<div class="paragraph">
<p>See https://github.com/SonarSource/SonarJS/issues/1698</p>
</div>
</div>
</div>
</div>