== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

With default provided regular expression ^([A-Z][A-Za-z]*:)*([A-Z][A-Za-z]*)$:


[source,terraform]
----
resource "aws_s3_bucket" "mynoncompliantbucket" {
  bucket = "mybucketname"

  tags = {
    "anycompany:cost-center" = "Accounting" # Noncompliant
  }
}
----

=== Compliant solution

[source,terraform]
----
resource "aws_s3_bucket" "mycompliantbucket" {
  bucket = "mybucketname"

  tags = {
    "AnyCompany:CostCenter" = "Accounting"
  }
}
----

include::../see.adoc[]
ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../parameters.adoc[]

endif::env-github,rspecator-view[]
