include::../description.adoc[]

== Noncompliant Code Example

With default provided regular expression ^([A-Z][A-Za-z]*:)*([A-Z][A-Za-z]*)$:


----
resource "aws_s3_bucket" "mynoncompliantbucket" {
  bucket = "mybucketname"

  tags = {
    "anycompany:cost-center" = "Accounting" # Noncompliant
  }
}
----

== Compliant Solution

----
resource "aws_s3_bucket" "mycompliantbucket" {
  bucket = "mybucketname"

  tags = {
    "AnyCompany:CostCenter" = "Accounting"
  }
}
----

include::../see.adoc[]
