include::../summary.adoc[]

== Why is this an issue?

include::../description.adoc[]

== How to fix it

=== Code examples

==== Noncompliant code example

With default provided format parameter `^([A-Z][A-Za-z]*:)*([A-Z][A-Za-z]*)$`

[source,hcl,diff-id=1,diff-type=noncompliant]
----
resource "aws_s3_bucket" "mynoncompliantbucket" {
  bucket = "mybucketname"

  tags = {
    "anycompany:cost-center" = "Accounting"
  }
}
----

==== Compliant solution

[source,hcl,diff-id=1,diff-type=compliant]
----
resource "aws_s3_bucket" "mycompliantbucket" {
  bucket = "mybucketname"

  tags = {
    "AnyCompany:CostCenter" = "Accounting"
  }
}
----

include::../see.adoc[]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../parameters.adoc[]

endif::env-github,rspecator-view[]
