This rule raises an issue when database column names use SQL reserved words like "release", "order", "group", or "select".

== Why is this an issue?

SQL reserved words are special keywords that have specific meanings in SQL syntax. When you use them as column names, database engines can become confused about whether you're referring to the column or trying to use the SQL command.

This confusion leads to several problems:

* *Syntax errors*: Queries may fail to parse correctly, causing runtime errors
* *Escaping requirements*: You need to wrap column names in backticks or quotes, making queries harder to read
* *Database portability issues*: Different databases have different reserved word lists, so code may work on one database but fail on another
* *Developer confusion*: Other team members may struggle to understand why certain queries need special escaping

The issue is particularly common with words like "order" (used for sorting), "group" (used for grouping), "release" (sounds like a good column name but conflicts with transaction commands), and "select" (the most basic SQL command).

=== What is the potential impact?

Using SQL reserved words as column names can cause application crashes when queries fail to parse. This creates reliability issues and can lead to unexpected downtime. The problem may not surface immediately during development but can appear later when more complex queries are written, making it harder to debug and fix.

== How to fix it

Replace SQL reserved words with more descriptive column names that clearly indicate their purpose. Add context or suffixes to make the names more specific and avoid conflicts.

=== Code examples

==== Noncompliant code example

[source,ruby,diff-id=1,diff-type=noncompliant]
----
class CreateMovies < ActiveRecord::Migration
  def change
    create_table :movies do |t|
      t.string :title
      t.date :release      # Noncompliant
      t.text :description
      t.integer :order     # Noncompliant
      t.timestamps
    end
  end
end
----

==== Compliant solution

[source,ruby,diff-id=1,diff-type=compliant]
----
class CreateMovies < ActiveRecord::Migration
  def change
    create_table :movies do |t|
      t.string :title
      t.date :release_date
      t.text :description
      t.integer :sort_order
      t.timestamps
    end
  end
end
----

== Resources

=== Documentation

 * MySQL Reserved Words - https://dev.mysql.com/doc/refman/8.0/en/keywords.html[Complete list of MySQL reserved words and keywords]

 * PostgreSQL Reserved Words - https://www.postgresql.org/docs/current/sql-keywords-appendix.html[PostgreSQL SQL key words and reserved words reference]

 * ActiveRecord Migrations Guide - https://guides.rubyonrails.org/active_record_migrations.html[Rails guide on creating and managing database migrations]

=== Standards

 * SQL:2016 Standard - https://www.iso.org/standard/63555.html[ISO/IEC 9075 SQL standard defining reserved words across database systems]

=== Related rules

 * RSPEC-4413 - https://rules.sonarsource.com/tsql/RSPEC-4413/[T-SQL rule for avoiding reserved words as identifiers]

 * RSPEC-7187 - https://rules.sonarsource.com/python/RSPEC-7187/[Python rule for avoiding SQL reserved words in database schemas]
