<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>To improve code readability, at least one blank line should separate properties and method definitions.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-text" data-lang="text">Module Module1
    Public Property Foo() As String  ' Compliant
        Get
            Return ""
        End Get
        Set(ByVal value As String)

        End Set
    End Property
    Sub Bar()                        ' Non-Compliant
    End Sub
End Module</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-text" data-lang="text">Module Module1
    Public Property Foo() As String  ' Compliant
        Get
            Return ""
        End Get
        Set(ByVal value As String)

        End Set
    End Property

    Sub Bar()                        ' Compliant
    End Sub
End Module</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Add a blank line after this property.</p>
</div>
</div>
</div>
</div>