To improve code readability, at least one blank line should separate properties and method definitions.


== Noncompliant Code Example

----
Module Module1
    Public Property Foo() As String  ' Compliant
        Get
            Return ""
        End Get
        Set(ByVal value As String)

        End Set
    End Property
    Sub Bar()                        ' Non-Compliant
    End Sub
End Module
----


== Compliant Solution

----
Module Module1
    Public Property Foo() As String  ' Compliant
        Get
            Return ""
        End Get
        Set(ByVal value As String)

        End Set
    End Property

    Sub Bar()                        ' Compliant
    End Sub
End Module
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

endif::env-github,rspecator-view[]
