:operationName: function
:visibility: private

include::../why.adoc[]

=== Code examples

==== Noncompliant code example

[source,swift]
----
class Foo {    
    private func unusedMethod() {...} // Noncompliant
}
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
