<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>A primary key uniquely identifies a row in a database table, and should be considered immutable. Primary key values may be used in foreign keys in other tables, as well as in external systems. Changing such a value, even with the best of motivations, is likely to wreak havoc on the system&#8217;s data integrity and potentially across other systems as well.</p>
</div>
<div class="paragraph">
<p><strong>Note</strong> That this rule raises issues only when a database catalog is provided during the SonarQube analysis.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cobol" data-lang="cobol">UPDATE USERS
  SET USER_ID = :new-id, USER_NAME = :new-name  *&gt; Noncompliant
  WHERE USER_ID = :input</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>"COLUMN_X" is the primary key and should not be updated.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>primary: column name</p>
</div>
<div class="paragraph">
<p>secondary: "UPDATE"</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_11_jan_2016_163548_pierre_yves_nicolas_wrote">on 11 Jan 2016, 16:35:48 Pierre-Yves Nicolas wrote:</h3>
<div class="paragraph">
<p>This rule should catch issues on primary keys, but also on partitioning indexes. This is reflected in the message, but not in the current description.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_14_jan_2016_163230_elena_vilchik_wrote">on 14 Jan 2016, 16:32:30 Elena Vilchik wrote:</h3>
<div class="literalblock">
<div class="content">
<pre>IMO table name ``++ROW++`` is confusing</pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_on_25_jan_2016_152748_pierre_yves_nicolas_wrote">on 25 Jan 2016, 15:27:48 Pierre-Yves Nicolas wrote:</h3>
<div class="paragraph">
<p>It seems that checking partitioning columns is more related to performance. Maybe we could add a rule parameter to deactivate that part of the check.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_26_jan_2016_144136_pierre_yves_nicolas_wrote">on 26 Jan 2016, 14:41:36 Pierre-Yves Nicolas wrote:</h3>
<div class="paragraph">
<p>In fact, we should think about creating a separate rule to check partitioning columns since the motivation is not the same ("performance" vs "data related reliability").</p>
</div>
</div>
</div>
</div>