include::../description.adoc[]

== Noncompliant Code Example

With a maximum number of 4 parameters:

----
public function addData(p1 : int, p2 : int, p3 : int, p4 : int, p5 : int): void  { 
...
}
----

== Compliant Solution

----
public function addData(p1 : int, p2 : int, p3 : int, p4 : int): void  { 
...
}
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::parameters.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
