This rule raises an issue when test methods use SOQL queries to fetch existing organizational data instead of creating their own test data.

== Why is this an issue?

Test classes that rely on existing organizational data create unreliable and environment-dependent tests. When tests query for existing records using SOQL statements like `[SELECT Id FROM Account LIMIT 1]`, they assume that specific data already exists in the org.

This approach leads to several problems:

* *Test fragility*: Tests may pass in one environment but fail in another if the expected data doesn't exist
* *Unpredictable results*: Tests might behave differently based on what data happens to be present
* *Maintenance burden*: Changes to org data can break tests that have nothing to do with the code being modified
* *Poor test isolation*: Tests become interdependent on external state rather than being self-contained

Reliable unit tests follow the principle of test isolation - each test should create exactly the data it needs and not depend on any pre-existing state. This ensures tests are predictable, maintainable, and can run successfully in any environment.

=== What is the potential impact?

Tests that depend on existing org data are unreliable and may fail unexpectedly when deployed to different environments or when org data changes. This reduces confidence in the test suite and can lead to deployment failures or missed bugs in production.

== How to fix it

Replace SOQL queries that fetch existing data with code that creates the required test data. Use the `new` keyword to create objects and `insert` them if needed for your test logic.

=== Code examples

==== Noncompliant code example

[source,apex,diff-id=1,diff-type=noncompliant]
----
@isTest
public class MyTestClass {
    @isTest
    static void testAccountProcessing() {
        Account acc = [SELECT Id, Name FROM Account LIMIT 1]; // Noncompliant
        // test logic using existing data
        processAccount(acc.Id);
    }
}
----

==== Compliant solution

[source,apex,diff-id=1,diff-type=compliant]
----
@isTest
public class MyTestClass {
    @isTest
    static void testAccountProcessing() {
        Account acc = new Account(Name='Test Account');
        insert acc;
        // test logic using created data
        processAccount(acc.Id);
    }
}
----

== Resources

=== Documentation

 * Apex Testing Best Practices - https://developer.salesforce.com/docs/atlas.en-us.apexcode.meta/apexcode/apex_testing_best_practices.htm[Official Salesforce documentation on Apex testing best practices including test data creation]

 * Test Data Creation in Apex - https://developer.salesforce.com/docs/atlas.en-us.apexcode.meta/apexcode/apex_testing_utility_classes.htm[Salesforce guide on creating test data and utility classes for testing]
