include::../description.adoc[]

== Noncompliant Code Example

----
if (i < 10) {} // Empty on purpose or missing piece of code ?
----

include::../exceptions.adoc[]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
