<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>If an abstract class does not contain properties nor actual function implementations but only abstract functions,
it should be replaced with an interface.
This is a good practice because an interface provides more flexibility,
while an abstract class does not offer any benefit in this situation.</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="sect3">
<h4 id="_flexibility_of_the_design">Flexibility of the design</h4>
<div class="paragraph">
<p>Abstract classes are limited in their inheritability compared to interfaces,
while they do not offer any benefit in this situation.
A class can only extend one class but can implement many interfaces.
When APIs are represented by abstract classes instead of interfaces,
the implementing class must have the abstract class as its one (and only) parent class.
Also, it cannot implement more than one API at once.</p>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Either replace the class declaration with an interface declaration,
or add actual function implementations or state properties to the abstract class.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-kotlin" data-lang="kotlin">abstract class Shape { // Noncompliant, abstract class only contains abstract functions
    abstract fun getPath(): Path
    abstract fun getArea(): Double
    abstract fun getBoundingBox(): Rectangle
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-kotlin" data-lang="kotlin">interface Shape { // Compliant, we are using an interface here
    fun getPath(): Path
    fun getArea(): Double
    fun getBoundingBox(): Rectangle
}

// or

abstract class Shape { // Compliant, abstract class has function implementations
    abstract fun getPath(): Path
    abstract fun getArea(): Double
    fun getBoundingBox(): Rectangle {
        // ...
    }
}</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p><a href="https://kotlinlang.org/docs/interfaces.html">Kotlin Docs, Interfaces</a></p>
</li>
</ul>
</div>
</div>
<div class="sect2">
<h3 id="_articles_blog_posts">Articles &amp; blog posts</h3>
<div class="ulist">
<ul>
<li>
<p><a href="https://blog.kotlin-academy.com/abstract-class-vs-interface-in-kotlin-5ab8697c3a14">Kotlin Academy, Abstract class vs interface in Kotlin</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>