== Why is this an issue?

Marking a variable with ``++IBOutlet++`` allows it to be connected with a Storyboard component through the Interface Builder. Allowing such a variable to be accessed outside the class, may result in other classes making assignments that override the automatic dependency injection from the Storyboard itself.


=== Noncompliant code example

[source,swift]
----
  @IBOutlet var label: UILabel!  // Noncompliant
----


=== Compliant solution

[source,swift]
----
  @IBOutlet private var label: UILabel!
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

include::highlighting.adoc[]

'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
