<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>C&#43;&#43;17 introduced <code>as_const</code>: a helper function that converts a value to its corresponding const value succinctly and more explicitly.</p>
</div>
<div class="paragraph">
<p>This is usually done to force an overloaded function call on a non-const object to resolve to the const alternative. Or to instantiate a template with a const type rather than the original non-const one.</p>
</div>
<div class="paragraph">
<p>This rule detects casts that can be replaced by <code>as_const</code>.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">void fn(std::vector&lt;int&gt;&amp; vec);
void fn(const std::vector&lt;int&gt;&amp; vec);

void constFCaller() {
  std::vector&lt;int&gt; vec;
  // Set the content of vec
  // ...
  fn(static_cast&lt;const std::vector&lt;int&gt;&amp;&gt;(vec)); // Noncompliant
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">void fn(std::vector&lt;int&gt;&amp; vec);
void fn(const std::vector&lt;int&gt;&amp; vec);

void constFCaller() {
  std::vector&lt;int&gt; vec;
  // Set the content of vec
  // ...
  fn(as_const(vec)); // Compliant
}</code></pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p>C&#43;&#43; reference - <a href="https://en.cppreference.com/w/cpp/utility/as_const">std::as_const</a></p>
</li>
</ul>
</div>
</div>
</div>