<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>A field marked <code>readonly</code> can only be assigned as part of its declaration or in a constructor. While <code>readonly</code> reference types (e.g. classes) can still have their state changed subsequently, the same is not true of value types such as <code>struct</code> s. Thus, calling a method that updates object state on a <code>readonly</code> value type field simply has no effect (but runs without error!). The result is code that probably doesn&#8217;t do what you thought it did.</p>
</div>
<div class="paragraph">
<p>This rule raises an issue when a method that is not marked <code>[Pure]</code> is invoked on a <code>readonly</code> value type field.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-csharp" data-lang="csharp">public struct S1
{
    public int value;

    public void SetValue()
    {
        value = 10;
    }
}

class Test
{
    static readonly S1 first;
    static S1 second;

    static void Main()
    {
        first.SetValue();  // Noncompliant
        second.SetValue();
        Console.WriteLine(first.value);  // Surprise! This writes 0
        Console.WriteLine(second.value); // This writes 10
    }
}</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove the "readonly" modifier from "xxx", or add the "[Pure]" attribute to "yyy".</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p><code>xxx.impureMethod</code></p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_11_dec_2015_091714_tamas_vajk_wrote">on 11 Dec 2015, 09:17:14 Tamas Vajk wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] I changed the title a bit.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_11_dec_2015_155448_ann_campbell_wrote">on 11 Dec 2015, 15:54:48 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~tamas.vajk] I made one further, minor change</p>
</div>
</div>
</div>
</div>