<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This is an issue when <code>flask.send_file()</code> is called with a file-like object without providing either the <code>mimetype</code> or <code>attachment_filename</code> parameter.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Flask&#8217;s <code>send_file()</code> function needs to determine the content type and filename for HTTP responses. When you pass a file path as a string, Flask can automatically detect these from the file system. However, when you pass a file-like object (such as an open file handle, <code>BytesIO</code>, or <code>StringIO</code>), Flask cannot determine this information automatically.</p>
</div>
<div class="paragraph">
<p>Without the <code>mimetype</code> parameter, Flask doesn&#8217;t know what <code>Content-Type</code> header to set for the response. Without <code>attachment_filename</code>, Flask cannot set the <code>Content-Disposition</code> header properly for downloads.</p>
</div>
<div class="paragraph">
<p>When Flask encounters this situation, it raises a <code>ValueError</code> with a message indicating that it cannot determine the content type. This causes your application to crash with a 500 Internal Server Error, creating a poor user experience.</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="paragraph">
<p>The application will crash with a <code>ValueError</code> when the endpoint is accessed, resulting in a 500 Internal Server Error response. This breaks the file download functionality and creates a poor user experience.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it_in_flask">How to fix it in Flask</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Add the <code>mimetype</code> parameter to specify the content type of the file being sent.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">from flask import send_file

def download_file():
    file_obj = open('data.txt', 'rb')
    return send_file(file_obj)  # Noncompliant</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">from flask import send_file

def download_file():
    file_obj = open('data.txt', 'rb')
    return send_file(file_obj, mimetype='text/plain')</code></pre>
</div>
</div>
<div class="paragraph">
<p>Add the <code>attachment_filename</code> parameter to specify the filename for downloads. This is especially useful for in-memory file objects.</p>
</div>
</div>
<div class="sect3">
<h4 id="_noncompliant_code_example_2">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">from flask import send_file
from io import BytesIO

def download_csv():
    csv_data = BytesIO(b'name,age\nJohn,30')
    return send_file(csv_data)  # Noncompliant</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution_2">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">from flask import send_file
from io import BytesIO

def download_csv():
    csv_data = BytesIO(b'name,age\nJohn,30')
    return send_file(csv_data, attachment_filename='data.csv', as_attachment=True)</code></pre>
</div>
</div>
<div class="paragraph">
<p>You can provide both parameters for complete control over the response headers.</p>
</div>
</div>
<div class="sect3">
<h4 id="_noncompliant_code_example_3">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">from flask import send_file
from io import StringIO

def download_log():
    log_data = StringIO('INFO: Application started')
    return send_file(log_data)  # Noncompliant</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution_3">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">from flask import send_file
from io import StringIO

def download_log():
    log_data = StringIO('INFO: Application started')
    return send_file(log_data, mimetype='text/plain', attachment_filename='app.log')</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Flask send_file documentation - <a href="https://flask.palletsprojects.com/en/2.3.x/api/#flask.send_file">Official Flask documentation for the send_file function</a></p>
</li>
<li>
<p>Flask file uploads and downloads guide - <a href="https://flask.palletsprojects.com/en/2.3.x/patterns/fileuploads/">Flask patterns for handling file uploads and downloads</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>