include::../../../shared_content/secrets/description.adoc[]

== Why is this an issue?

include::../../../shared_content/secrets/rationale.adoc[]

=== What is the potential impact?

include::../common/impact/rationale.adoc[]

include::../../../shared_content/secrets/impact/financial_loss.adoc[]

include::../../../shared_content/secrets/impact/security_downgrade.adoc[]

== How to fix it in ARM Templates

=== Code examples

==== Noncompliant code example

[source,json,diff-id=1,diff-type=noncompliant]
----
{
  "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
  "contentVersion": "1.0.0.0",
  "parameters": {
  },
  "resources": [
    {
      "type": "Microsoft.Sql/servers",
      "apiVersion": "2023-05-01-preview",
      "name": "example-database",
      "properties": [
        {
          "administratorLogin": "admin",
          "administratorLoginPassword": "password"
        }
      ]
    }
  ]
}
----

==== Compliant solution

[source,json,diff-id=1,diff-type=compliant]
----
{
  "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
  "contentVersion": "1.0.0.0",
  "parameters": {
    "adminUsername": {
      "type": "securestring"
    },
    "adminPassword": {
      "type": "securestring"
    }
  },
  "resources": [
    {
      "type": "Microsoft.Sql/servers",
      "apiVersion": "2023-05-01-preview",
      "name": "example-database",
      "properties": [
        {
          "administratorLogin": "[parameters('adminUsername')]",
          "administratorLoginPassword": "[parameters('adminPassword')]"
        }
      ]
    }
  ]
}
----

== How to fix it in Bicep

=== Code examples

==== Noncompliant code example

[source,bicep,diff-id=2,diff-type=noncompliant]
----
resource sqlServer 'Microsoft.Sql/servers@2023-05-01-preview' = {
  name: 'example-database'
  properties: {
    administratorLogin: 'admin'
    administratorLoginPassword: 'password'
  }
}
----

==== Compliant solution

[source,bicep,diff-id=2,diff-type=compliant]
----
@secure()
param adminLogin string
@secure()
param adminPassword string

resource sqlServer 'Microsoft.Sql/servers@2023-05-01-preview' = {
  name: 'example-database'
  properties: {
    administratorLogin: adminLogin
    administratorLoginPassword: adminPassword
  }
}
----

== Resources

=== Documentation

* Microsoft - https://learn.microsoft.com/en-us/azure/azure-resource-manager/templates/best-practices#security-recommendations-for-parameters[ARM template best practices]
* Microsoft - https://learn.microsoft.com/en-us/azure/azure-resource-manager/bicep/best-practices[Best practices for Bicep]
* Microsoft - https://learn.microsoft.com/en-us/azure/templates/microsoft.sql/servers?pivots=deployment-language-arm-template[Microsoft.Sql servers documentation]

include::../common/resources/standards.adoc[]

ifdef::env-github,rspecator-view[]
'''
== Implementation Specification
(visible only on this page)

=== Message
Revoke and change this secret, as it might be compromised.

=== Highlighting

Highlight property name and value

'''
endif::env-github,rspecator-view[]
