<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>ABAP provides the ability to manipulate programs dynamically during execution for instance with statements like <code>INSERT REPORT</code> and <code>GENERATE SUBROUTINE POOL</code>. Most of those statements are for internal use within SAP Technology Development and incompatible changes are possible at any time without prior warning or notice.</p>
</div>
<div class="paragraph">
<p>This rule raises an issue when any of the following source code processing statements is used: <code>INSERT REPORT</code>,  <code>READ REPORT</code>, <code>DELETE REPORT</code>, <code>EDITOR-CALL FOR REPORT</code>, <code>SYNTAX-CHECK FOR itab</code>, <code>GENERATE REPORT/SUBROUTINE POOL</code>, <code>LOAD REPORT</code>, <code>SCAN</code>, <code>INSERT TEXTPOOL</code>, <code>READ TEXTPOOL</code>, <code>DELETE TEXTPOOL</code>, <code>EXPORT DYNPRO</code>, <code>IMPORT DYNPRO</code>, <code>DELETE DYNPRO</code>, <code>SYNTAX-CHECK FOR DYNPRO</code>, and <code>GENERATE DYNPRO</code>.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-abap" data-lang="abap">GENERATE REPORT MY_PROG.</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove this use of the source code processing "XXXX" statement.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_deprecates_s1513">deprecates: <a data-rspec-id="S1513" class="rspec-auto-link">S1513</a></h3>

</div>
<div class="sect2">
<h3 id="_is_duplicated_by_s1509">is duplicated by: <a data-rspec-id="S1509" class="rspec-auto-link">S1509</a></h3>

</div>
</div>
</div>