== Why is this an issue?

include::../rationale.adoc[]

include::../impact.adoc[]

== How to fix it

=== Code examples

include::../../common/fix/code-rationale.adoc[]

==== Noncompliant code example

[source,php,diff-id=1,diff-type=noncompliant]
----
$result = mysql_query("SELECT ssn FROM users WHERE id = " . $name);  // Noncompliant

while ($row = mysql_fetch_object($result)) {
    echo $row->ssn;
}
----

==== Compliant solution

[source,php,diff-id=1,diff-type=compliant]
----
$conn = new PDO('mysql:host=localhost;dbname=db', $username, $password);
$conn->setAttribute(PDO::ATTR_ERRMODE, PDO::ERRMODE_EXCEPTION);    

$stmt = $conn->prepare('SELECT ssn FROM users WHERE id = :id');
$stmt->execute(['id' => $name]);

while ($row = $stmt->fetch(PDO::FETCH_OBJ)) {
    echo $row->ssn;
}
----

=== How does this work?

include::../../common/fix/how.adoc[]

== Resources

include::../common/resources/standards.adoc[]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

endif::env-github,rspecator-view[]
