This rule raises an issue when a coroutine object is created but never awaited.

== Why is this an issue?

Calling a coroutine function (defined with `async def`) returns a coroutine *object*, but does not execute its code. If you forget to use `await`, the coroutine never runs, which can lead to:

* Functions that appear to be called but never actually execute
* Missing operations that were expected to run
* Values that remain as coroutine objects instead of being resolved
* Potential resource leaks when cleanup code in the coroutine doesn't run

This is a common pitfall in asynchronous programming, as the syntax for calling an async function looks the same as calling a regular function, but the execution model is entirely different.

== How to fix it

Always `await` coroutines or use structured concurrency patterns to manage them. Depending on the framework you use, options include:

- Using `await` directly on the coroutine
- Using `asyncio.TaskGroup`, `trio.Nursery`, or similar constructs for concurrent execution
- Creating tasks with `asyncio.create_task()` and ensuring they're properly awaited later

=== Code examples

==== Noncompliant code example

[source,python,diff-id=1,diff-type=noncompliant]
----
async def do_something_async():
    await async_operation()
    return "Done"

async def main():
    result = do_something_async() # Noncompliant: coroutine is created but never executed
    print(f"Result: {result}")  # Prints a coroutine object, not "Done"
----

==== Compliant solution

[source,python,diff-id=1,diff-type=compliant]
----
async def do_something_async():
    await async_operation()
    return "Done"

async def main():
    result = await do_something_async() # Compliant
    print(f"Result: {result}")  # Prints "Done"
----

== Resources
=== Documentation
* Python asyncio - https://docs.python.org/3/library/asyncio-task.html[Tasks and Coroutines]
* Python - https://docs.python.org/3/reference/compound_stmts.html#async-def[Coroutine function definition]

=== Articles & blog posts
* Python - https://realpython.com/async-io-python/[Async IO in Python: A Complete Walkthrough]

ifdef::env-github,rspecator-view[]
== Implementation Specification
(visible only on this page)

=== Message
Await this coroutine or store it in a variable to be awaited later.

=== Highlighting
The coroutine function call that's not being awaited.

endif::env-github,rspecator-view[]

