== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,swift]
----
var a1 = b + c // This is a trailing comment that can be very very long
----

=== Compliant solution

[source,swift]
----
// This very long comment is better placed before the line of code
var a2 = b + c
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::parameters.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
