This is an issue when using a type named `scoped` in lambda parameter lists in C# 14 or later, where the compiler treats both the modifier and type name as modifiers, causing compilation errors.

== Why is this an issue?

In C# 14, the language introduced the ability to write lambdas with parameter modifiers without specifying parameter types. As part of this enhancement, a breaking change was implemented where `scoped` is always treated as a modifier in lambda parameter lists.

When you have a type named `scoped` and try to use it with the `scoped` modifier in a lambda parameter, the compiler interprets both tokens as modifiers rather than recognizing one as a type name. This creates a syntax error because the compiler expects a type name after the modifier.

This breaking change affects existing code that was valid in earlier C# versions, particularly when upgrading projects to C# 14. The issue is specific to lambda expressions and doesn't affect other contexts where the `scoped` type name might be used.

=== What is the potential impact?

This issue prevents code compilation when upgrading to C# 14. Projects that previously compiled successfully will fail to build, blocking development and deployment processes until the code is updated.

== How to fix it

Escape the type name `scoped` with the `@` prefix to distinguish it from the `scoped` modifier. This tells the compiler to treat it as an identifier rather than a keyword.

=== Code examples

==== Noncompliant code example

[source,csharp,diff-id=1,diff-type=noncompliant]
----
var lambda = (scoped scoped parameter) => { /* ... */ }; // Noncompliant

ref struct scoped { }
----

==== Compliant solution

[source,csharp,diff-id=1,diff-type=compliant]
----
var lambda = (scoped @scoped parameter) => { /* ... */ };

ref struct scoped { }
----

== Resources

=== Documentation

 * Breaking changes in Roslyn after .NET 9.0.100 through .NET 10.0.100 - https://learn.microsoft.com/en-us/dotnet/csharp/whats-new/breaking-changes/compiler%20breaking%20changes%20-%20dotnet%2010[Official Microsoft documentation detailing the breaking change where 'scoped' is always treated as a modifier in lambda parameter lists]

 * C# Keywords - https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/[Documentation about C# keywords and how to use the @ prefix to escape them as identifiers]
