Since .NET 6 you don't have to use the `TimeZoneConverter` library to do the conversion between IANA and Windows timezones manually.
.NET 6.0 introduced new Time Zone enhancements, one being the `TimeZoneInfo.FindSystemTimeZoneById(string timezone)` method now accepts as input both IANA and Windows time zone IDs on any operating system with installed time zone data.
`TimeZoneInfo.FindSystemTimeZoneById` will automatically convert its input from IANA to Windows and vice versa if the requested time zone is not found on the system.


// If you want to factorize the description uncomment the following line and create the file.
//include::../description.adoc[]

== Why is this an issue?

Code becomes more complicated and thus more difficult to maintain.


== How to fix it

There's no need to manually translate between time zones - calling `TimeZoneInfo.FindSystemTimeZoneById(string timezone)` where the timezone can be IANA or Windows format, is enough.
Depending on the OS the equivalent time zone will be returned (Windows Time Zones for Windows and IANA timezones for Linux, macOS, Java, Python etc.)

=== Code examples

==== Noncompliant code example

[source,text,diff-id=1,diff-type=noncompliant]
----
// Assuming we are in Windows OS and we need to get the Tokyo Time Zone.
var ianaTimeZone = "Asia/Tokyo";
var windowsTimeZone = TZConvert.IanaToWindows(ianaTimeZone);
TimeZoneInfo tokyoWindowsTimeZone = TimeZoneInfo.FindSystemTimeZoneById(WindowsTimeZone);
----

==== Compliant solution

[source,text,diff-id=1,diff-type=compliant]
----
// Assuming we are in Windows OS and we need to get the Tokyo Time Zone.
var ianaTimeZone = "Asia/Tokyo";
TimeZoneInfo tokyoWindowsTimeZone = TimeZoneInfo.FindSystemTimeZoneById(ianaTimeZone);
----

== Resources

=== Documentation
* https://learn.microsoft.com/en-us/dotnet/api/system.timezoneinfo[TimeZone Info documentation]
* https://devblogs.microsoft.com/dotnet/date-time-and-time-zone-enhancements-in-net-6/[Date, Time, and Time Zone Enhancements in .NET 6]
* https://github.com/mattjohnsonpint/TimeZoneConverter[TimeZoneConverter]

=== Articles & blog posts
* https://codeblog.jonskeet.uk/2022/02/05/whats-up-with-timezoneinfo-on-net-6-part-1/[What's up with TimeZoneInfo on .NET 6?]
