=== relates to: S1155

=== on 2 Dec 2014, 07:35:56 Nicolas Peru wrote:
Doesn't this one duplicates this : \http://jira.sonarsource.com/browse/RSPEC-1155 ?

=== on 2 Dec 2014, 14:44:21 Ann Campbell wrote:
It does now [~nicolas.peru]. It was originally written as "just don't use ``++ConcurrentLinkedQueue++`` - for anything!" in response to an internal email thread.

\[~freddy.mallet]...?

=== on 4 Dec 2014, 10:15:47 Freddy Mallet wrote:
Ok [~nicolas.peru] and [~ann.campbell.2], I've changed the content of the noncompliant code example to prevent any misunderstanding but for me there is no overlap between RSPEC-2250 and RSPEC-1155.

