<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>The time complexity of method calls on collections is not always obvious. For instance, for most collections the <code>size()</code> method takes constant time, but the time required to execute <code>ConcurrentLinkedQueue.size()</code> is O(n), i.e. directly proportional to the number of elements in the collection. When the collection is large, this could therefore be an expensive operation.</p>
</div>
<div class="paragraph">
<p>This rule raises an issue when the following O(n) methods are called outside of constructors on class fields:</p>
</div>
<div class="ulist">
<ul>
<li>
<p><code>ArrayList</code></p>
<div class="ulist">
<ul>
<li>
<p><code>contains</code></p>
</li>
<li>
<p><code>remove</code></p>
</li>
</ul>
</div>
</li>
<li>
<p><code>LinkedList</code></p>
<div class="ulist">
<ul>
<li>
<p><code>get</code></p>
</li>
<li>
<p><code>contains</code></p>
</li>
</ul>
</div>
</li>
<li>
<p><code>ConcurrentLinkedQueue</code></p>
<div class="ulist">
<ul>
<li>
<p><code>size</code></p>
</li>
<li>
<p><code>contains</code></p>
</li>
</ul>
</div>
</li>
<li>
<p><code>ConcurrentLinkedDeque</code></p>
<div class="ulist">
<ul>
<li>
<p><code>size</code></p>
</li>
<li>
<p><code>contains</code></p>
</li>
</ul>
</div>
</li>
<li>
<p><code>CopyOnWriteArrayList</code></p>
<div class="ulist">
<ul>
<li>
<p><code>add</code></p>
</li>
<li>
<p><code>contains</code></p>
</li>
<li>
<p><code>remove</code></p>
</li>
</ul>
</div>
</li>
<li>
<p><code>CopyOnWriteArraySet</code></p>
<div class="ulist">
<ul>
<li>
<p><code>add</code></p>
</li>
<li>
<p><code>contains</code></p>
</li>
<li>
<p><code>remove</code></p>
</li>
</ul>
</div>
</li>
</ul>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-text" data-lang="text">ConcurrentLinkedQueue queue = new ConcurrentLinkedQueue();
//...
log.info("Queue contains " + queue.size() + " elements");  // Noncompliant</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>This call to "xxx" may be a performance hot spot if the collection is large.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_relates_to_s1155">relates to: <a data-rspec-id="S1155" class="rspec-auto-link">S1155</a></h3>

</div>
<div class="sect2">
<h3 id="_on_2_dec_2014_073556_nicolas_peru_wrote">on 2 Dec 2014, 07:35:56 Nicolas Peru wrote:</h3>
<div class="paragraph">
<p>Doesn&#8217;t this one duplicates this : http://jira.sonarsource.com/browse/RSPEC-1155 ?</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_2_dec_2014_144421_ann_campbell_wrote">on 2 Dec 2014, 14:44:21 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>It does now [~nicolas.peru]. It was originally written as "just don&#8217;t use <code>ConcurrentLinkedQueue</code> - for anything!" in response to an internal email thread.</p>
</div>
<div class="paragraph">
<p>\[~freddy.mallet]&#8230;&#8203;?</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_4_dec_2014_101547_freddy_mallet_wrote">on 4 Dec 2014, 10:15:47 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>Ok [~nicolas.peru] and [~ann.campbell.2], I&#8217;ve changed the content of the noncompliant code example to prevent any misunderstanding but for me there is no overlap between <a data-rspec-id="S2250" class="rspec-auto-link">RSPEC-2250</a> and <a data-rspec-id="S1155" class="rspec-auto-link">RSPEC-1155</a>.</p>
</div>
</div>
</div>
</div>