<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>jQuery doesn&#8217;t cache elements for you. If you&#8217;ve made a selection that you might need to make again, you should save the selection in a variable rather than making the selection repeatedly.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="paragraph">
<p>With the default threshold of 2</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-javascript" data-lang="javascript">$( "p" ).hide();
$( "p" ).show();  // Noncompliant</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-javascript" data-lang="javascript">var paragraph = $( "p" );

paragraph.hide();
paragraph.show();</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_exceptions">Exceptions</h3>
<div class="paragraph">
<p>Stored selections are not updated when the DOM changes. Since variables may need to updated this rule ignores selections that are repeated during an assignment.</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-javascript" data-lang="javascript">var paragraph = $("p");

// ...

paragraph = $("p");</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Selection "xxx" is made n times. It should be stored in a variable and reused.</p>
</div>
</div>
<div class="sect2">
<h3 id="_parameters">Parameters</h3>
<div class="sidebarblock">
<div class="content">
<div class="title">threshold</div>
<div class="listingblock">
<div class="content">
<pre>2</pre>
</div>
</div>
<div class="paragraph">
<p>Number of allowed repetition before triggering an issue</p>
</div>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_25_mar_2015_110035_linda_martin_wrote">on 25 Mar 2015, 11:00:35 Linda Martin wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] Assigned for review and completion.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_10_apr_2015_144408_elena_vilchik_wrote">on 10 Apr 2015, 14:44:08 Elena Vilchik wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2], i added parameter as we discussed.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_1_nov_2019_172929_elena_vilchik_wrote">on 1 Nov 2019, 17:29:29 Elena Vilchik wrote:</h3>
<div class="paragraph">
<p>See https://github.com/SonarSource/SonarJS/issues/1698</p>
</div>
</div>
</div>
</div>