== Why is this an issue?

This rule allows banning some modules.

=== Noncompliant code example

With ``++moduleName++`` configured with ``++[a-zA-Z-]*UT123[a-zA-Z]*++``:


[source,cobol]
----
CALL UT123.
CALL UT123L.
CALL UT123-L.
CALL WS-UT123.
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

include::parameters.adoc[]

include::../highlighting.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
