This rule allows banning some modules.

== Noncompliant Code Example

With ``++moduleName++`` configured with ``++[a-zA-Z-]*UT123[a-zA-Z]*++``:


----
CALL UT123.
CALL UT123L.
CALL UT123-L.
CALL WS-UT123.
----

ifdef::env-github,rspecator-view[]
'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
