== Why is this an issue?

This rule allows banning some modules.

=== Noncompliant code example

With ``++moduleName++`` configured with ``++[a-zA-Z-]*UT123[a-zA-Z]*++``:


[source,cobol]
----
CALL UT123.
CALL UT123L.
CALL UT123-L.
CALL WS-UT123.
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Remove the use of this forbidden module. Use {replacementModuleName} instead.


=== Parameters

.moduleName
****

Name of the forbidden module. Use a regex to forbid more widely multiple modules sharing the same root name.
****
.replacementModuleName
****

Name of the module to use as a replacement of the forbidden one.
****


include::../highlighting.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
