Code annotated as deprecated should not be used since it will be removed sooner or later.

== Noncompliant Code Example

----
// C++14 attribute
[[deprecated]]
void fun();

// GNU attribute
__attribute__((deprecated))
void fun();

// Microsoft attribute
__declspec(deprecated)
void fun();

void example() {
  fun(); // Noncompliant
}
----

include::../see.adoc[]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
