This rule raises an issue when the `fetch()` method on an Apex cursor is called with a chunk size larger than 200 records.

== Why is this an issue?

Salesforce enforces strict governor limits on resources like heap size and CPU time within each transaction. These limits prevent any single operation from consuming too many platform resources.

When using Apex cursors to process large datasets, the `fetch()` method retrieves a specified number of records in a single call. If you fetch too many records at once (typically more than 200), you risk exceeding these governor limits, especially:

* **Heap size limit**: Large record sets consume significant memory
* **CPU time limit**: Processing many records can exceed the allowed computation time
* **SOQL query limits**: Though less common with cursors, large fetches can still impact query performance

When governor limits are exceeded, Salesforce throws runtime exceptions that cannot be caught or handled gracefully. This causes the entire transaction to fail, potentially leaving your data in an inconsistent state.

Apex cursors are specifically designed for incremental processing of large datasets. The whole point is to break down big operations into manageable chunks that stay well within platform limits.

=== What is the potential impact?

Exceeding governor limits causes immediate transaction failure with uncatchable runtime exceptions. This can lead to:

* Data processing operations that fail unpredictably
* Incomplete data migrations or bulk operations
* Poor user experience when operations fail without clear error handling
* Difficulty in debugging and maintaining code that works in some environments but fails in others with larger datasets

== How to fix it

Reduce the chunk size to 200 records or fewer. This keeps your operations well within Salesforce governor limits while still processing data efficiently.

=== Code examples

==== Noncompliant code example

[source,apex,diff-id=1,diff-type=noncompliant]
----
List<Contact> scope = locator.fetch(position, 2000); // Noncompliant
----

==== Compliant solution

[source,apex,diff-id=1,diff-type=compliant]
----
List<Contact> scope = locator.fetch(position, 200);
----

== Resources

=== Documentation

 * Apex Cursors for Scalable Data Processing - https://www.apexhours.com/apex-cursors-for-scalable-data-processing-in-salesforce[Comprehensive guide on using Apex cursors with best practices for chunk sizes]

 * Salesforce Governor Limits - https://developer.salesforce.com/docs/atlas.en-us.apexcode.meta/apexcode/apex_gov_limits.htm[Official documentation on Salesforce platform limits and resource management]
