== Why is this an issue?

Shared coding conventions allow teams to collaborate efficiently. This rule checks that all variable names match the provided regular expression. 

=== Noncompliant code example

With the default regular expression ``++[a-zA-Z]([a-zA-Z0-9_]*[a-zA-Z0-9])?++``:

[source,sql]
----
DECLARE
  goodVariable PLS_INTEGER; -- Compliant
  badVariable_ PLS_INTEGER; -- Non-Compliant
BEGIN
  NULL;
END;
/
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Rename this variable to match the provided regular expression: xxx.


=== Parameters

.regexp
****

----
[a-zA-Z]([a-zA-Z0-9_]*[a-zA-Z0-9])?
----

The regular expression the name should match
****


'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
