Shared coding conventions allow teams to collaborate efficiently. This rule checks that all variable names match the provided regular expression. 

== Noncompliant Code Example

With the default regular expression `+[a-zA-Z]([a-zA-Z0-9_]*[a-zA-Z0-9])?+`:

----
DECLARE
  goodVariable PLS_INTEGER; -- Compliant
  badVariable_ PLS_INTEGER; -- Non-Compliant
BEGIN
  NULL;
END;
/
----
