The use of Unicode escape sequences should be reserved for characters that would otherwise be ambiguous, such as unprintable characters.


This rule ignores sequences composed entirely of Unicode characters, but otherwise raises an issue for each Unicode character that represents a printable character.


== Noncompliant Code Example

----
String prefix = "n\u00E9e"; // Noncompliant
----


== Compliant Solution

----
String prefix = "n√©e";
----



ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
