== Why is this an issue?

The number of distinct data items used in a condition (``++IF++``, ``++EVALUATE++``, ...) should not exceed a defined threshold.


=== Noncompliant code example

With the default default threshold of 3:

[source,cobol]
----
IF WS-FOO(1) = 1 OR *> 1st data item
WS-FOO(2) = 2 OR
WS-FOO(3) = 3 OR
WS-BAR = 4 OR *> 2nd data item
WS-BAZ = 5 OR *> 3rd data item
WS-QUX = 42 *> Noncompliant; 4th data item
END-IF.
----


=== Compliant solution

[source,cobol]
----
IF WS-FOO(1) = 1 OR
WS-FOO(2) = 2 OR
WS-FOO(3) = 3 OR
WS-BAR = 4 OR
WS-BAZ = 42
END-IF.
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

???


=== Parameters

.maximumNumberOfOperands
****

----
3
----

Maximum number of distinct data items allowed in a single condition
****


'''
== Comments And Links
(visible only on this page)

=== on 29 May 2015, 14:48:09 Ann Campbell wrote:
double-check me please, [~pierre-yves.nicolas]

=== on 25 Jun 2015, 15:45:26 Pierre-Yves Nicolas wrote:
\[~ann.campbell.2] Why was this RSPEC closed and the legacy key removed?

=== on 25 Jun 2015, 17:25:03 Ann Campbell wrote:
\[~pierre-yves.nicolas] the legacy key was moved to RSPEC-1067, which has the modernized version of the title that comes up on nemo when you search for 'COBOL.ConditionComplexityCheck'. And at that point, this rule seemed like a duplicate, although on closer inspection, perhaps it's a separate issue & this rule should be reopened? (Altho not with that legacy key.)

=== on 26 Jun 2015, 07:28:43 Pierre-Yves Nicolas wrote:
Currently, the COBOL plugin implements the following rule:

* name: Avoid too complex conditions
* key: COBOL.ConditionComplexityCheck
Its behavior is the one described for RSPEC-2982: it's not related to cyclomatic complexity but to "distinct data items".


I can't find any implementation of RSPEC-1067 in the current version of the COBOL plugin.

=== on 26 Jun 2015, 13:32:14 Ann Campbell wrote:
Sorry about that [~pierre-yves.nicolas]. I'm returning this to your queue.

endif::env-github,rspecator-view[]
