include::../description.adoc[]

== Noncompliant Code Example

----
def divide(numerator, denominator)
  return numerator / denominator             # FIXME denominator value might be 0
end
----

include::../see.adoc[]

ifdef::env-github,rspecator-view[]
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
