_Mutexes_ are synchronization primitives that allow to manage concurrency.
Their use requires following a well-defined life-cycle.

* _Mutexes_ need to be initialized (<code>pthread_mutex_init</code>) before being used. Once it is initialized, a _mutex_ is in an _unlocked_ state.
* _Mutexes_ need to be destroyed (<code>pthread_mutex_destroy</code>) to free the associated internal resources. Only _unlocked_ _mutexes_ can be safely destroyed.

Before initialization or after destruction, a mutex is in an uninitialized state.

About this life-cycle, the following patterns should be avoided as they result in an undefined behavior:

* trying to initialize an initialized _mutex_
* trying to destroy an initialized _mutex_ that is in a _locked_ state
* trying to destroy an uninitialized _mutex_
* trying to lock an uninitialized _mutex_
* trying to unlock an uninitialized _mutex_

In C++, it is recommended to wrap mutex creation/destruction in a RAII class, as well as mutex lock/unlock. Those RAII classes will perform the right operations, even in presence of exceptions.


== Noncompliant Code Example

----
pthread_mutex_t mtx1;

void bad1(void)
{
  pthread_mutex_init(&mtx1);
  pthread_mutex_init(&mtx1);
}

void bad2(void)
{
  pthread_mutex_init(&mtx1);
  pthread_mutex_lock(&mtx1);
  pthread_mutex_destroy(&mtx1);
}

void bad3(void)
{
  pthread_mutex_init(&mtx1);
  pthread_mutex_destroy(&mtx1);
  pthread_mutex_destroy(&mtx1);
}

void bad4(void)
{
  pthread_mutex_init(&mtx1);
  pthread_mutex_destroy(&mtx1);
  pthread_mutex_lock(&mtx1);
}

void bad5(void)
{
  pthread_mutex_init(&mtx1);
  pthread_mutex_destroy(&mtx1);
  pthread_mutex_unlock(&mtx1);
}

----


== Compliant Solution

----
pthread_mutex_t mtx1;

void ok1(void)
{
  pthread_mutex_init(&mtx1);
  pthread_mutex_destroy(&mtx1);
}

void ok2(void)
{
  pthread_mutex_init(&mtx1);
  pthread_mutex_lock(&mtx1);
  pthread_mutex_unlock(&mtx1);
  pthread_mutex_destroy(&mtx1);
}
----


== See

* https://pubs.opengroup.org/onlinepubs/009695399/functions/pthread_mutex_destroy.html[The Open Group] pthread_mutex_init, pthread_mutex_destroy

