== How to fix it in Pyca

=== Code examples

==== Noncompliant code example

include::../../common/fix/aes-noncompliant-example.adoc[]

[source,python,diff-id=1,diff-type=noncompliant]
----
from cryptography.hazmat.primitives.ciphers import (
    Cipher, 
    algorithms, 
    modes,
)
from cryptography.hazmat.backends import default_backend

Cipher(algorithms.AES(key), modes.CBC(iv), backend=default_backend()) # Noncompliant
----

include::../../common/fix/rsa-noncompliant-example.adoc[]

[source,python,diff-id=2,diff-type=noncompliant]
----
from cryptography.hazmat.primitives.asymmetric import (
    rsa,
    padding,
)

private_key = rsa.generate_private_key(
  public_exponent=65537,
  key_size=2048,
  backend=default_backend()
)

public_key = private_key.public_key()

public_key.encrypt(
  message,
  padding.PKCS1v15() # Noncompliant
)
----

==== Compliant solution

include::../../common/fix/aes-compliant-example.adoc[]

[source,python,diff-id=1,diff-type=compliant]
----
from cryptography.hazmat.primitives.ciphers import (
    Cipher, 
    algorithms, 
    modes,
)
from cryptography.hazmat.backends import default_backend

Cipher(algorithms.AES(key), modes.GCM(iv), backend=default_backend())
----

[source,python,diff-id=2,diff-type=compliant]
----
from cryptography.hazmat.primitives.asymmetric import (
    rsa,
    padding,
)
from cryptography.hazmat.primitives import hashes

private_key = rsa.generate_private_key(
  public_exponent=65537,
  key_size=2048,
  backend=default_backend()
)

public_key = private_key.public_key()

public_key.encrypt(
  message,
  padding.OAEP(
    mgf=padding.MGF1(algorithm=hashes.SHA256()),
    algorithm=hashes.SHA256(),
    label=None
  )
)
nclude::../../common/fix/fix.adoc[] ----

=== How does this work?

include::../../common/fix/fix.adoc[]


=== Going the extra mile

include::../../common/extra-mile/pre-quantum.adoc[]
