Unnecessary braces in string interpolation should be removed.

== Why is this an issue?

Dart https://dart.dev/language/built-in-types#strings[strings] support interpolation, which allows you to embed expressions within a string.

[source,dart]
----
String expression = 'words';
print('A string containing ${expression}');
----

When the expression is a simple identifier, and the identifier is followed in the string interpolation by a character which is not allowed in an identifier (e.g. whitespace, `:`, `-` but not a letter, digit, ``++_++``, ...), you can omit the curly braces. 

This makes the code more readable and concise.

[source,dart]
----
String expression = 'words';
print('A string containing $expression');
----

== How to fix it

Remove the curly braces.

=== Code examples

==== Noncompliant code example

[source,dart,diff-id=1,diff-type=noncompliant]
----
String expression = 'words';
print('A string containing ${expression}');
----

==== Compliant solution

[source,dart,diff-id=1,diff-type=compliant]
----
String expression = 'words';
print('A string containing $expression');
----

== Resources

=== Documentation

* Dart Docs - https://dart.dev/tools/linter-rules/unnecessary_brace_in_string_interps[Dart Linter rule - unnecessary_brace_in_string_interps]
* Dart Docs - https://dart.dev/language/built-in-types#strings[Language - Strings]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Unnecessary braces in a string interpolation.

=== Highlighting

The interpolation expression, including the ``++$++`` sign and the braces: e.g. ``++${expression}++`` in ``++'A string containing ${expression}'++``.

'''
== Comments And Links
(visible only on this page)

endif::env-github,rspecator-view[]
