== How to fix it in .NET

=== Code examples

include::../../common/fix/code-rationale.adoc[]

==== Noncompliant code example

[source,vbnet]
----
Imports System.Net

Public Sub Send()
    ServicePointManager.ServerCertificateValidationCallback = 
        Function(sender, certificate, chain, errors) True ' Noncompliant
End Sub
----

=== How does this work?

include::../../common/fix/validation.adoc[]
