== How to fix it in .NET

=== Code examples

In the following example, the callback change impacts the entirety of HTTP
requests made by the application.

:cert_method_name: ServerCertificateValidationCallback

include::../../common/fix/code-rationale-override.adoc[]

==== Noncompliant code example

[source,vbnet]
----
Imports System.Net

Public Sub Send()
    ServicePointManager.ServerCertificateValidationCallback = 
        Function(sender, certificate, chain, errors) True ' Noncompliant
End Sub
----

=== How does this work?

include::../../common/fix/validation.adoc[]
