== How to fix it in System.Net

=== Code examples

include::../../common/fix/code-rationale.adoc[]

==== Noncompliant code example

[source,vbnet,diff-id=1,diff-type=noncompliant]
----
ServicePointManager.ServerCertificateValidationCallback = 
    Function(sender, certificate, chain, errors) True ' Noncompliant
----

==== Compliant solution

=== How does this work?

include::../../common/fix/default-implementation.adoc[]
