== How to fix it in System.Net

=== Code examples

include::../../common/fix/code-rationale.adoc[]

==== Noncompliant code example

[source,vbnet]
----
ServicePointManager.ServerCertificateValidationCallback = 
    Function(sender, certificate, chain, errors) True ' Noncompliant
----

=== How does this work?

include::../../common/fix/validation.adoc[]
