<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>In places where key-value pairs are used, a space before the equal sign may lead to unintended behavior.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>A few Docker instructions (like <code>ARG</code>, <code>ENV</code>, <code>LABEL</code>) may contain key-value pairs in form of <code>&lt;key&gt;=&lt;value&gt;</code>.
The equal sign should not be followed by a space.
The space before the equal sign may lead to unintended behavior.
This is critical, especially for multiple key-value pairs, e.g. <code>key1 = value1 key2 = value2</code>, will lead to the key <code>key1</code> with the value <code>= value1 key2 = value2</code>.
In most cases it is unintended.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-docker" data-lang="docker">ENV BUILD_NUMBER =1
RUN echo $BUILD_NUMBER</code></pre>
</div>
</div>
<div class="paragraph">
<p>This will lead to print <code>=1</code>, which is not expected.</p>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-docker" data-lang="docker">ENV BUILD_NUMBER=1
RUN echo $BUILD_NUMBER</code></pre>
</div>
</div>
<div class="paragraph">
<p>This will lead to print <code>1</code>, which is expected.</p>
</div>
</div>
<div class="sect3">
<h4 id="_noncompliant_code_example_2">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-docker" data-lang="docker">ENV USER = bob MODE = all
RUN echo $USER</code></pre>
</div>
</div>
<div class="paragraph">
<p>This will lead to print <code>= bob MODE = all</code>, which is not expected.</p>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution_2">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-docker" data-lang="docker">ENV USER=bob MODE=all
RUN echo $USER</code></pre>
</div>
</div>
<div class="paragraph">
<p>This will lead to print <code>bob</code>, which is expected.</p>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_how_does_this_work">How does this work?</h3>
<div class="paragraph">
<p>The <code>ENV</code> instruction allows alternative syntax <code>ENV &lt;key&gt; &lt;value&gt;</code> and in case of space before equal sign, the <code>=1</code> is evaluated as value.
The <code>LABEL</code> instruction will be also evaluated to <code>=1</code>.
The <code>ARG</code> instruction will cause the build error.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p><a href="https://docs.docker.com/engine/reference/builder/#arg">ARG - Dockerfile reference</a></p>
</li>
<li>
<p><a href="https://docs.docker.com/engine/reference/builder/#env">ENV - Dockerfile reference</a></p>
</li>
<li>
<p><a href="https://docs.docker.com/engine/reference/builder/#label">LABEL - Dockerfile reference</a></p>
</li>
</ul>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove the space before the equal sign in the key-value pair, as it can lead to unexpected behavior.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>Highlight the entire key-value pair containing a space before the equal sign.</p>
</div>
<hr>
</div>
</div>
</div>