<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>While keywords introduced in later standards can legally be used as identifiers in code compiled to earlier standards, doing so will eventually cause problems. Such code will cause compile errors if (when) the compiler is upgraded, and fixing those errors could be difficult and painful.</p>
</div>
<div class="paragraph">
<p>Additionally, such misuse of keywords has the potential to thoroughly confuse people who are unfamiliar with the code base, possibly leading them to introduce additional errors.</p>
</div>
<div class="paragraph">
<p>For these reasons, the earlier this practice is stopped, the better.</p>
</div>
<div class="paragraph">
<p>This rule flags instances of the following keywords used as identifiers:</p>
</div>
<div class="paragraph">
<p><strong>C99</strong></p>
</div>
<div class="paragraph">
<p><code>inline</code>, <code>restrict</code>, <code>_Bool</code>, <code>_Complex</code>, <code>_Noreturn</code>, <code>_Static_assert</code>, <code>_Thread_local</code></p>
</div>
<div class="paragraph">
<p><strong>C11</strong></p>
</div>
<div class="paragraph">
<p><code>_Alignas</code>, <code>_Alignof</code>, <code>_Atomic</code>, <code>_Generic</code>, <code>_Imaginary</code></p>
</div>
<div class="paragraph">
<p><strong>C&#43;&#43;11</strong></p>
</div>
<div class="paragraph">
<p><code>alignas</code>, <code>alignof</code>, <code>char16_t</code>, <code>char32_t</code>, <code>constexpr</code>, <code>decltype</code>, <code>noexcept</code>, <code>nullptr</code>, <code>static_assert</code>, <code>thread_local</code></p>
</div>
<div class="paragraph">
<p><strong>C&#43;&#43;20</strong></p>
</div>
<div class="paragraph">
<p><code>concept</code>, <code>requires</code>, <code>constinit</code>, <code>consteval</code>, <code>co_await</code>, <code>co_return</code>, <code>co_yield</code>, <code>char8_t</code></p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">int inline = 0;</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">int inline_count = 0;</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Rename the "xxx" identifier</p>
</div>
</div>
</div>
</div>