== Why is this an issue?

The ``++<object>++`` tag is used by Internet Explorer 3.0 or later on Windows platforms or any browser that supports the use of the Flash ActiveX control. The ``++<embed>++`` tag is used by Netscape Navigator 2.0 or later, or browsers that support the use of the Netscape-compatible plug-in version of Flash Player.


When an ActiveX-enabled browser loads the HTML page, it reads the values set on the ``++<object>++`` and ignores the ``++<embed>++`` tag. When browsers using the Flash plug-in load the HTML page, they read the values set on the ``++<embed>++`` tag and ignore the ``++<object>++`` tag.


=== Noncompliant code example

[source,html]
----
<object classid="clsid:d27cdb6e-ae6d-11cf-96b8-444553540000" width="100" height="100">      <!-- Non-Compliant -->
  <param name="movie" value="movie_name.swf" />
</object>

<embed src="movie_name.swf"                                                                 <!-- Non-Compliant -->
       width="550"
       height="400"
       type="application/x-shockwave-flash"
       pluginspage="http://www.macromedia.com/go/getflashplayer" />
----


=== Compliant solution

[source,html]
----
<object classid="clsid:d27cdb6e-ae6d-11cf-96b8-444553540000" width="100" height="100">      <!-- Compliant -->
  <param name="movie" value="movie_name.swf" />
  <embed src="movie_name.swf"                                                               <!-- Compliant -->
       width="550"
       height="400"
       type="application/x-shockwave-flash"
       pluginspage="http://www.macromedia.com/go/getflashplayer" />
</object>
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

* Add an "<embed>" tag within this "<object>" one.
* Surround this "<embed>" tag with an "<object>" one.


'''
== Comments And Links
(visible only on this page)

=== on 24 Jun 2013, 09:13:07 Dinesh Bolkensteyn wrote:
There doesn't seem to be a consensus on this, and I am unsure if this rule should be enabled by default.


http://helpx.adobe.com/flash/kb/object-tag-syntax-flash-professional.html does not include the embed tag for instance.

=== on 24 Jun 2013, 15:22:24 Dinesh Bolkensteyn wrote:
IE 3.0, Netscape 2.0, sounds like very serious reasons to comply to this rule ;)


I didn't find anything much better, the provided rationale is a bit blur.

=== on 3 Jul 2013, 16:06:04 Dinesh Bolkensteyn wrote:
I don't think this rule should be enabled by default.


The rationale behind this rule is obsolete, nowdays there are different recommended ways of inserting Flash animations.

=== on 8 Jul 2013, 18:24:26 Freddy Mallet wrote:
Is implemented by \http://jira.codehaus.org/browse/SONARPLUGINS-2987

endif::env-github,rspecator-view[]
