The ``++<object>++`` tag is used by Internet Explorer 3.0 or later on Windows platforms or any browser that supports the use of the Flash ActiveX control. The ``++<embed>++`` tag is used by Netscape Navigator 2.0 or later, or browsers that support the use of the Netscape-compatible plug-in version of Flash Player.


When an ActiveX-enabled browser loads the HTML page, it reads the values set on the ``++<object>++`` and ignores the ``++<embed>++`` tag. When browsers using the Flash plug-in load the HTML page, they read the values set on the ``++<embed>++`` tag and ignore the ``++<object>++`` tag.


== Noncompliant Code Example

----
<object classid="clsid:d27cdb6e-ae6d-11cf-96b8-444553540000" width="100" height="100">      <!-- Non-Compliant -->
  <param name="movie" value="movie_name.swf" />
</object>

<embed src="movie_name.swf"                                                                 <!-- Non-Compliant -->
       width="550"
       height="400"
       type="application/x-shockwave-flash"
       pluginspage="http://www.macromedia.com/go/getflashplayer" />
----


== Compliant Solution

----
<object classid="clsid:d27cdb6e-ae6d-11cf-96b8-444553540000" width="100" height="100">      <!-- Compliant -->
  <param name="movie" value="movie_name.swf" />
  <embed src="movie_name.swf"                                                               <!-- Compliant -->
       width="550"
       height="400"
       type="application/x-shockwave-flash"
       pluginspage="http://www.macromedia.com/go/getflashplayer" />
</object>
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
