include::../why.adoc[]

== How to fix it

=== Code examples

==== Noncompliant code example

[source,vbnet,diff-id=1,diff-type=noncompliant]
----
Dim foo As New Disposable()
foo.Dispose()
foo.Dispose() ' Noncompliant

Using bar As New Disposable()  ' Noncompliant
    bar.Dispose()
End Using
----

==== Compliant solution

[source,vbnet,diff-id=1,diff-type=compliant]
----
Dim foo As New Disposable()
foo.Dispose()

Using bar As New Disposable()  ' Noncompliant

End Using
----


include::resources.adoc[]

include::../rspecator.adoc[]
