<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Since .NET 6 you don&#8217;t have to use the <code>TimeZoneConverter</code> library to manually do the conversion between IANA and Windows timezones.
The .NET 6.0 introduced new Time Zone enhancements, one being the <code>TimeZoneInfo.FindSystemTimeZoneById(string timezone)</code> method now accepts as input both IANA and Windows time zone IDs on any operating system with installed time zone data.
<code>TimeZoneInfo.FindSystemTimeZoneById</code> will automatically convert its input from IANA to Windows and vice versa if the requested time zone is not found on the system.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>The method <code>TimeZoneInfo.FindSystemTimeZoneById(string timezone)</code> can get both IANA and Windows timezones as input and automatically convert one to the other if the requested time zone is not found on the system.
Because one does not need to handle the conversion, the code will be less complex and easier to maintain.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>There&#8217;s no need to translate manually between time zones; it is enough to call <code>TimeZoneInfo.FindSystemTimeZoneById(string timezone)</code>, where the timezone can be IANA or Windows format.
Depending on the OS, the equivalent time zone will be returned (Windows Time Zones for Windows and IANA timezones for Linux, macOS).</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-vbnet" data-lang="vbnet">// Assuming we are in Windows OS and we need to get the Tokyo Time Zone.
Dim ianaTimeZone = "Asia/Tokyo"
Dim windowsTimeZone = TZConvert.IanaToWindows(ianaTimeZone)
Dim tokyoWindowsTimeZone As TimeZoneInfo = TimeZoneInfo.FindSystemTimeZoneById(windowsTimeZone)</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-vbnet" data-lang="vbnet">// Assuming we are in Windows OS and we need to get the Tokyo Time Zone.
Dim ianaTimeZone = "Asia/Tokyo"
Dim tokyoWindowsTimeZone As TimeZoneInfo = TimeZoneInfo.FindSystemTimeZoneById(ianaTimeZone)</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p><a href="https://learn.microsoft.com/en-us/dotnet/api/system.timezoneinfo.findsystemtimezonebyid">TimeZoneInfo.FindSystemTimeZoneById documentation</a></p>
</li>
<li>
<p><a href="https://devblogs.microsoft.com/dotnet/date-time-and-time-zone-enhancements-in-net-6/">Date, Time, and Time Zone Enhancements in .NET 6</a></p>
</li>
<li>
<p><a href="https://github.com/mattjohnsonpint/TimeZoneConverter">TimeZoneConverter</a></p>
</li>
</ul>
</div>
</div>
<div class="sect2">
<h3 id="_articles_blog_posts">Articles &amp; blog posts</h3>
<div class="ulist">
<ul>
<li>
<p><a href="https://codeblog.jonskeet.uk/2022/02/05/whats-up-with-timezoneinfo-on-net-6-part-1/">What&#8217;s up with TimeZoneInfo on .NET 6?</a></p>
</li>
</ul>
</div>
</div>
<div class="sect2">
<h3 id="_standards">Standards</h3>
<div class="ulist">
<ul>
<li>
<p><a href="https://www.iana.org/time-zones">IANA Time Zone Database</a></p>
</li>
<li>
<p><a href="https://learn.microsoft.com/en-us/windows-hardware/manufacture/desktop/default-time-zones?view=windows-11">Windows Time Zones</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>