== Why is this an issue?

In Blazor, the https://learn.microsoft.com/en-us/aspnet/core/blazor/fundamentals/routing#catch-all-route-parameters[catch-all route parameters] allows to capture the rest of the URL path. Blazor does not provide automatic casting so the matching component parameter should be a string.

Failing to do so, a runtime error will occur when navigating to the URL:

[source,text]
----
Unhandled exception rendering component: Unable to set property '<property-name>' on object of type '<component-type>'. The error was: Specified cast is not valid.
...
----

== How to fix it

Ensure the parameter in the component that matches the catch-all route parameter is a string type.

=== Code examples

==== Noncompliant code example

[source,csharp,diff-id=1,diff-type=noncompliant]
----
@page "/route/{*catchAllRoute}"

@code {
    [Parameter]
    public int CatchAllRoute { get; set; } // Noncompliant
}
----

==== Compliant solution

[source,csharp,diff-id=1,diff-type=compliant]
----
@page "/route/{*catchAllRoute}"

@code {
    [Parameter]
    public string CatchAllRoute { get; set; } // Compliant
}
----

== Resources
=== Documentation

* Microsoft Learn - https://learn.microsoft.com/en-us/aspnet/core/blazor/fundamentals/routing#catch-all-route-parameters[ASP.NET Core Blazor routing and navigation]
