<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when a <code>numpy</code> weekmask format is incorrect.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>To allow a datetime to be used in contexts where only certain days of the week are valid, NumPy includes a set of business day functions. <code>Weekmask</code> is used to customize valid business days.</p>
</div>
<div class="paragraph">
<p><code>Weekmask</code> can be specified in several formats:</p>
</div>
<div class="olist arabic">
<ol class="arabic">
<li>
<p>As an array of 7 <code>1</code> or <code>0</code> values, e.g. <code>[1, 1, 1, 1, 1, 0, 0]</code></p>
</li>
<li>
<p>As a string of 7 <code>1</code> or <code>0</code> characters, e.g. <code>"1111100"</code></p>
</li>
<li>
<p>As a string with abbreviations of valid days from this list: <code>Mon Tue Wed Thu Fri Sat Sun</code>, e.g. <code>"Mon Tue Wed Thu Fri"</code></p>
</li>
</ol>
</div>
<div class="paragraph">
<p>Setting an incorrect <code>weekmask</code> leads to <code>ValueError</code>.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Provide a <code>weekmask</code> with correct values.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">import numpy as np

offset = np.busday_offset('2012-05', 1, roll='forward', weekmask='01') # Noncompliant: ValueError</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">import numpy as np

offset = np.busday_offset('2012-05', 1, roll='forward', weekmask='0111100') # OK</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Numpy documentation - <a href="https://numpy.org/doc/stable/reference/arrays.datetime.html#business-day-functionality">Business Day Functionality</a></p>
</li>
<li>
<p>Numpy documentation - <a href="https://numpy.org/doc/stable/reference/arrays.datetime.html#custom-weekmasks">Custom Weekmasks</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>