The W3C specifications define the valid ``++at-rules++``. Only the official and browser-specific ``++at-rules++`` should be used to get the expected impact in the final rendering.


== Noncompliant Code Example

----
@encoding "utf-8";
----


== Compliant Solution

----
@charset "utf-8";
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

include::parameters.adoc[]

endif::env-github,rspecator-view[]
