== Why is this an issue?

The W3C specifications define the valid ``++at-rules++``. Only the official and browser-specific ``++at-rules++`` should be used to get the expected impact in the final rendering.


=== Noncompliant code example

[source,css]
----
@encoding "utf-8";
----


=== Compliant solution

[source,css]
----
@charset "utf-8";
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Replace XXX by a valid "at-rules".


=== Parameters

.ignoreAtRules
****

----
"content,debug,each,else,for,function,if,include,mixin,return,while"
----

comma-separated list of "at-rules" to consider as valid
****


endif::env-github,rspecator-view[]
