<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when the <code>sleep</code> function is used in an asynchronous loop instead of an Event object.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Asynchronous tasks often need to wait for a condition to change or an event to occur.
A simple-looking but inefficient way to achieve this is by polling the condition within a loop, using <code>sleep</code> to pause between checks:</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">while not condition_is_met:
    await asyncio.sleep(0.1) # Noncompliant
# Condition is met, we can proceed</code></pre>
</div>
</div>
<div class="paragraph">
<p>This busy-waiting approach is problematic in asynchronous code because it introduces increased latency.
The task only notices the condition change after the <code>sleep</code> interval expires. If the condition becomes true just after the task starts sleeping, the reaction is delayed.</p>
</div>
<div class="paragraph">
<p>Instead of polling with <code>sleep</code>, use dedicated synchronization primitives like <code>asyncio.Event</code>, <code>trio.Event</code> or <code>anyio.Event</code>.
Using an <code>Event</code> allows a task to efficiently pause (<code>await event.wait()</code>) until another part of the program signals the event (<code>event.set()</code>).
The waiting task consumes almost no resources while paused and reacts immediately when the event is set.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it_in_asyncio">How to fix it in Asyncio</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">import asyncio

SHARED_CONDITION = False

async def worker():
    while not SHARED_CONDITION: # Noncompliant
        await asyncio.sleep(0.01)
    print("Condition is now true")

asyncio.run(worker)</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">import asyncio

SHARED_CONDITION = asyncio.Event()

async def worker():
    await SHARED_CONDITION.wait() # Compliant
    print("Condition is now true")

asyncio.run(worker)</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it_in_trio">How to fix it in Trio</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_code_examples_2">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example_2">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">import trio

SHARED_CONDITION = False

async def worker():
    while not SHARED_CONDITION: # Noncompliant
        await trio.sleep(0.01)
    print("Condition is now true")

trio.run(worker)</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution_2">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">import trio

SHARED_CONDITION = trio.Event()

async def worker():
    await SHARED_CONDITION.wait() # Compliant
    print("Condition is now true")

trio.run(worker)</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it_in_anyio">How to fix it in AnyIO</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_code_examples_3">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example_3">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">import anyio

SHARED_CONDITION = False

async def worker():
    while not SHARED_CONDITION: # Noncompliant
        await anyio.sleep(0.01)
    print("Condition is now true")

anyio.run(worker)</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution_3">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">import anyio

SHARED_CONDITION = anyio.Event()

async def worker():
    await SHARED_CONDITION.wait() # Compliant
    print("Condition is now true")

anyio.run(worker)</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Asyncio documentation - <a href="https://docs.python.org/3/library/asyncio-sync.html#asyncio.Event">Event</a></p>
</li>
<li>
<p>Trio documentation - <a href="https://trio.readthedocs.io/en/stable/reference-core.html#broadcasting-an-event-with-event">Broadcasting an event with Event</a></p>
</li>
<li>
<p>AnyIO documentation - <a href="https://anyio.readthedocs.io/en/stable/synchronization.html#events">Events</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Refactor this loop to use an <code>Event</code> instead of polling with <code>sleep</code>.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>Primary location is the loop condition with a secondary location on the sleep
Secondary location on the async keyword of the function definition.</p>
</div>
</div>
<div class="sect2">
<h3 id="_quickfix">Quickfix</h3>
<div class="paragraph">
<p>No</p>
</div>
</div>
</div>
</div>