== Why is this an issue?

This rule allows you to track the use of the Checkstyle suppression comment mechanism. 


=== Noncompliant code example

[source,java]
----
// CHECKSTYLE:OFF
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Remove this "CHECKSTYLE:OFF" suppression comment.


'''
== Comments And Links
(visible only on this page)

=== is related to: S1310

=== on 11 Oct 2013, 08:23:18 Freddy Mallet wrote:
Is implemented by \http://jira.codehaus.org/browse/SONARJAVA-355

endif::env-github,rspecator-view[]
