Declaring a parameter without specifying its data type leaves the compiler to assign the type that seems the most appropriate - whether it's what you need or not. Therefore you should always specify the data type.


== Noncompliant Code Example

[source,vb6]
----
Sub SubComputeArea(Length, TheWidth)
  [. . .]
End Sub
----


== Compliant Solution

----
Sub SubComputeArea(Length As Integer, TheWidth As Integer)
  [. . .]
End Sub
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

endif::env-github,rspecator-view[]
