<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Batch Apex jobs should execute as fast as possible because other jobs wait in the queue. They will execute faster when their <code>start</code> method returns a QueryLocator object or an iterable which does not use subqueries to include related records. It is more efficient to execute these subqueries separately inside the <code>execute</code> method of the Batch class.</p>
</div>
<div class="paragraph">
<p>This rule raises an issue when, in the context of a Batch class' <code>start</code> method, the  <code>Database.getQueryLocator(query)</code> method is called with a query containing one or more subqueries</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-apex" data-lang="apex">global class MyBatch implements Database.Batchable&lt;sObject&gt; {
    global MyBatch(){
    }

    global Database.QueryLocator start(Database.BatchableContext BC){
        return Database.getQueryLocator([SELECT Id, (SELECT id FROM Contacts) FROM Account]);  // Noncompliant
    }

    global void execute(Database.BatchableContext BC, List&lt;sObject&gt; scope){
        // ...
    }

    global void finish(Database.BatchableContext BC){
        // ...
    }
}</code></pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p><a href="https://developer.salesforce.com/docs/atlas.en-us.apexcode.meta/apexcode/apex_batch_interface.htm">Using Batch Apex</a> (See the "Batch Apex Best Practices" section)</p>
</li>
</ul>
</div>
<hr>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Execute subqueries in the "execute" method</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>The SOQL query</p>
</div>
</div>
</div>
</div>