== How to fix it in Java Cryptographic Extension

=== Code examples

==== Noncompliant code example

[source,java,diff-id=1,diff-type=noncompliant]
----
import javax.net.ssl.SSLContext;
import java.security.NoSuchAlgorithmException;

public static void main(String[] args) {
    try {
        SSLContext.getInstance("TLSv1.1"); // Noncompliant
    } catch (NoSuchAlgorithmException e) {
        // ...
    }
}
----

==== Compliant solution

[source,java,diff-id=1,diff-type=compliant]
----
import javax.net.ssl.SSLContext;
import java.security.NoSuchAlgorithmException;

public static void main(String[] args) {
    try {
        SSLContext.getInstance("TLSv1.2");
    } catch (NoSuchAlgorithmException e) {
        // ...
    }
}
----

=== How does this work?

include::../../common/fix/fix.adoc[]
