== How to fix it in ASP.NET

=== Code examples

include::../../common/fix/code-rationale.adoc[]

==== Noncompliant code example

[source,vbnet,diff-id=1,diff-type=noncompliant]
----
Public Class ExampleController
    Inherits Controller

    <HttpGet>
    Public Function GetUser(Id As String) As IActionResult
        Dim Url As String = "http://example.com/api/user/" & Id
        Dim Request As HttpWebRequest = CType(WebRequest.Create(Url), HttpWebRequest) ' Noncompliant

        Return Ok()
    End Function
End Class
----

==== Compliant solution

[source,vbnet,diff-id=1,diff-type=compliant]
----
Public Class ExampleController
    Inherits Controller

    <HttpGet>
    Public Function GetUser(Id As String) As IActionResult
        Dim Url As String = "http://example.com/api/user/?id=" & HttpUtility.UrlEncode(Id)
        Dim Request As HttpWebRequest = CType(WebRequest.Create(Url), HttpWebRequest)

        Return Ok()
    End Function
End Class
----

=== How does this work?

include::../../common/fix/encoding.adoc[]
