== How to fix it in ASP.NET

=== Code examples

include::../../common/fix/code-rationale.adoc[]

==== Noncompliant code example

[source,vbnet,diff-id=1,diff-type=noncompliant]
----
Imports System.Web
Imports System.Web.Mvc

Public Class ExampleController
    Inherits Controller

    <HttpGet>
    Public Function GetUser(id As String) As IActionResult
        Dim url As String = "http://example.com/api/user/" + id
        Dim request As HttpWebRequest = CType(WebRequest.Create(url), HttpWebRequest) ' Noncompliant

        Return Ok()
    End Function
End Class
----

==== Compliant solution

[source,vbnet,diff-id=1,diff-type=compliant]
----
Imports System.Web
Imports System.Web.Mvc

Public Class ExampleController
    Inherits Controller

    <HttpGet>
    Public Function GetUser(id As String) As IActionResult
        Dim url As String = "http://example.com/api/user/?id=" + HttpUtility.UrlEncode(id)
        Dim request As HttpWebRequest = CType(WebRequest.Create(url), HttpWebRequest)

        Return Ok()
    End Function
End Class
----

=== How does this work?

include::../../common/fix/encoding.adoc[]
