<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Side effects in a module can lead to unpredictable behavior, making the module harder to understand and debug.
They complicate testing by requiring complex setup and teardown procedures, reducing the module&#8217;s reusability since its
behavior depends on specific states or external systems. Side effects can also introduce concurrency issues, such as
race conditions, and make the codebase harder to maintain and more prone to breaking. Additionally, they increase the
complexity and reduce the readability of the code, while breaking encapsulation and leading to tighter coupling between
modules. Minimizing side effects helps create more predictable, testable, reusable, and maintainable code.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-javascript" data-lang="javascript">console.log('hello world');
if (typeof Array.prototype.map !== 'function') {
  Array.prototype.map = () =&gt; {
    // Implement polyfill
  };
}
let s = 0;
for (let i = 0; i &lt; 10; i++) {
  s += i;
}
fetch('/api').then((res) =&gt; res.text());</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-javascript" data-lang="javascript">export default function () {
  console.log('hello world');

  let s = 0;
  for (let i = 0; i &lt; 10; i++) {
    s += i;
  }

  fetch('/api').then((res) =&gt; res.text());
}</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>MDN web docs - <a href="https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Modules">Modules</a></p>
</li>
<li>
<p>MDN web docs - <a href="https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/import">import</a></p>
</li>
<li>
<p>MDN web docs - <a href="https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/export">export</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>