== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,python]
----
r"Hello,   world!"
----

=== Compliant solution

[source,python]
----
r"Hello, {3}world!"
----

=== Exceptions

If the extended modifier `re.X` or `re.VERBOSE` is set, whitespaces are ignored. In this case no issue should be triggered, because the whitespaces may be intended to improve readability.

include::../implementation.adoc[]
