Sharing some naming conventions is a key point to make it possible for a team to efficiently collaborate. This rule allows to check that all subroutine names match a provided regular expression.

== Noncompliant Code Example

With default provided regular expression ``++^SR[a-zA-Z0-9]*$++``:

[source,rpg]
----
C     XXXXX         BEGSR
...
C                   ENDSR
----

[source,rpg]
----
 /free
   begsr yyyy;
   endsr;
 /end-free{code}
----

== Compliant Solution

[source,rpg]
----
C     SR102         BEGSR
...
C                   ENDSR
----

[source,rpg]
----
 /free
   begsr SR103;
   endsr;
 /end-free
----
