<div class="paragraph">
<p>Sharing some naming conventions is a key point to make it possible for a team to efficiently collaborate. This rule allows to check that all subroutine names match a provided regular expression.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="paragraph">
<p>With default provided regular expression <code>^SR[a-zA-Z0-9]*$</code>:</p>
</div>
<div class="listingblock">
<div class="content">
<pre>C     XXXXX         BEGSR
...
C                   ENDSR</pre>
</div>
</div>
<div class="listingblock">
<div class="content">
<pre> /free
   begsr yyyy;
   endsr;
 /end-free{code}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>C     SR102         BEGSR
...
C                   ENDSR</pre>
</div>
</div>
<div class="listingblock">
<div class="content">
<pre> /free
   begsr SR103;
   endsr;
 /end-free</pre>
</div>
</div>
</div>
</div>