This rule raises an issue when code attempts to access hash elements using dot notation (e.g., `hash.key`) instead of the correct bracket notation (e.g., `hash[:key]` or `hash['key']`).

== Why is this an issue?

In Ruby, hashes and objects use different syntax for accessing their contents. Dot notation (`object.method`) is used to call methods on objects, while bracket notation (`hash[:key]` or `hash['key']`) is the correct way to access hash elements.

When you try to use dot notation on a hash, Ruby interprets this as a method call. Since hashes don't have methods with those names, this results in a `NoMethodError` at runtime.

This confusion often happens because:

* Other programming languages may use dot notation for both objects and hash-like structures
* The syntax looks similar to accessing object properties in languages like JavaScript
* New Ruby developers may not yet understand the distinction between hashes and objects

The error will occur immediately when the code runs, making this a reliability issue that prevents the application from functioning correctly.

=== What is the potential impact?

Using dot notation on hashes causes immediate runtime failures with `NoMethodError` exceptions. This breaks application functionality and can lead to:

* Application crashes when the problematic code is executed
* Poor user experience due to unexpected errors
* Debugging time spent tracking down syntax mistakes
* Potential data processing failures if hash access is part of critical workflows

== How to fix it

Replace dot notation with bracket notation using the appropriate key type (symbol or string) that matches your hash definition.

=== Code examples

==== Noncompliant code example

[source,ruby,diff-id=1,diff-type=noncompliant]
----
person = {
  :name => "John",
  :age => 30
}

# This will raise NoMethodError
puts person.name  # Noncompliant
puts person.age   # Noncompliant
----

==== Compliant solution

[source,ruby,diff-id=1,diff-type=compliant]
----
person = {
  :name => "John",
  :age => 30
}

# Use bracket notation with symbols
puts person[:name]
puts person[:age]
----

== Resources

=== Documentation

 * Ruby Hash Documentation - https://ruby-doc.org/core/Hash.html[Official Ruby documentation for Hash class and its methods]

 * Ruby Hash Access Methods - https://ruby-doc.org/core/Hash.html#method-i-5B-5D[Documentation for the bracket notation method used to access hash elements]
