== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,kotlin]
----
when(param) { // missing else
  1 -> doSomething()
  2 -> doSomethingElse()
}
----

=== Compliant solution

[source,kotlin]
----
when(param) {
  1 -> doSomething()
  2 -> doSomethingElse()
  else -> error("myMessage")
}
----

=== Exceptions

If the ``++when++`` clause is used as an expression, Kotlin compiler already enforces that all possible cases are covered with branch conditions. So no issue will be raised in this case.


Example:

[source,kotlin]
----
enum class Day {
    SUNDAY, MONDAY
}
...
val myVal = when(day) {
  MyEnum.SUNDAY -> 1
  MyEnum.MONDAY -> 2
}
----

include::../see.adoc[]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
