== Why is this an issue?

``++std::move++`` is just a ``++static_cast++`` to an rvalue reference. It produces an rvalue of xvalue category. Calling ``++std::move++`` on an rvalue is redundant and might lead to preventing copy elision https://jira.sonarsource.com/browse/RSPEC-5274[RSPEC-5274]


=== Noncompliant code example

[source,cpp]
----
class A {
  std::string s;
public:
  explicit A(std::string s) :s{std::move(s)} {}
  A(const A&) = default;
  A(A&&) = default;
};

A returnAByValue();
A&& returnARvalueRef();
void takeRvalueRefOfA(A&&);

int uncompliant()
{
  takeRvalueRefOfA(std::move(A("I'm an rvalue"))); // the constructed object is already an rvalue
  A a2 = std::move(returnARvalueRef()); // the returned variable is already an rvalue
  A a1 = std::move(returnAByValue()); // the returned variable is already an rvalue. This prevent copy elision since move constructor will be called.
}
----


=== Compliant solution

[source,cpp]
----
class A {
  std::string s;
public:
  explicit A(std::string s) :s{std::move(s)} {}
  A(const A&) = default;
  A(A&&) = default;
};

A returnAByValue();
A&& returnARvalueRef();
void takeRvalueRefOfA(A&&);

int compliant()
{
  takeRvalueRefOfA(A("I'm an rvalue")); // it compiles without the unnecessary std::move
  A a2 = returnARvalueRef(); // move constructor is called without calling std::move
  A a1 = returnAByValue(); // copy elision
}
----



ifdef::env-github,rspecator-view[]
'''
== Comments And Links
(visible only on this page)

=== duplicates: S5274

=== on 2 Sep 2019, 19:46:12 Loïc Joly wrote:
\[~geoffray.adde]: Shouldn't it just be merged into RSPEC-5274?



=== on 3 Sep 2019, 17:37:44 Loïc Joly wrote:
After discussion with [~geoffray.adde]: Merge it into RSPEC-5274

endif::env-github,rspecator-view[]
