<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Immediately dropping a synchronization lock (e.g., <code>mutex</code>, <code>rwlock</code>) after acquiring it using <code>let _ = sync_lock</code> is often unintentional and can lead to subtle bugs. By extending the lock lifetime to the end of the scope using a named variable, the code becomes safer and intentions are clearer. If immediate drop is intended, using <code>std::mem::drop</code> conveys the intention more clearly and reduces error-prone behavior.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-rust" data-lang="rust">let _ = Mutex::new(1).lock(); // Noncompliant: Immediately drops the lock.</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-rust" data-lang="rust">// Extend lock lifetime to end of scope
let _lock = Mutex::new(1).lock(); // Compliant: Lock remains till scope ends.</code></pre>
</div>
</div>
<div class="paragraph">
<p>You can also explicitly drop the lock to convey intention:</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-rust" data-lang="rust">std::mem::drop(Mutex::new(1).lock()); // Compliant: Clearly drops the lock.</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Clippy Lints - <a href="https://rust-lang.github.io/rust-clippy/master/index.html#let_underscore_lock" class="bare">https://rust-lang.github.io/rust-clippy/master/index.html#let_underscore_lock</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>