== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,sql]
----
CREATE TABLE my_table(
  my_column INTEGER
);
----

=== Exceptions

Creating global temporary tables, creating indexes on those tables, and then dropping those indices and tables is allowed.

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
