include::../description.adoc[]

== Noncompliant Code Example

----
CREATE TABLE my_table(
  my_column INTEGER
);
----

== Exceptions

Creating global temporary tables, creating indexes on those tables, and then dropping those indices and tables is allowed.
