This rule raises an issue when code uses 'field' as an identifier in contexts where it conflicts with the new `field` contextual keyword introduced in C# 14.

== Why is this an issue?

C# 14 introduces the `field` contextual keyword for field-backed properties. This keyword allows you to access synthesized backing fields directly within property accessors, simplifying property implementations that need custom logic.

When your code uses 'field' as an identifier, several problems can occur:

* **Compilation errors**: Local variables or parameters named 'field' within property accessors will cause compiler error CS9272, preventing your code from compiling.
* **Semantic confusion**: Existing class members named 'field' may be overshadowed by the synthesized backing field when referenced within property accessors, leading to unexpected behavior.
* **Upgrade barriers**: Code that compiles in earlier C# versions may break when upgrading to C# 14.

The `field` keyword specifically refers to compiler-synthesized backing fields for properties. When the compiler encounters 'field' within a property accessor, it treats it as this special keyword rather than a regular identifier, which can cause conflicts with your existing code.

=== What is the potential impact?

Using 'field' as an identifier can prevent successful compilation when upgrading to C# 14. In property accessors, local variables named 'field' will cause compilation errors, while class members named 'field' may be unexpectedly overshadowed by synthesized backing fields, leading to logic errors and unexpected runtime behavior.

== How to fix it

Rename class members that use 'field' as an identifier to avoid conflicts with the contextual keyword.

=== Code examples

==== Noncompliant code example

[source,csharp,diff-id=1,diff-type=noncompliant]
----
public class MyClass
{
    private string field; // Noncompliant
    
    public string Message
    {
        get => field;
        set => field = value;
    }
}
----

==== Compliant solution

[source,csharp,diff-id=1,diff-type=compliant]
----
public class MyClass
{
    private string _field;
    
    public string Message
    {
        get => _field;
        set => _field = value;
    }
}
----

== Resources

=== Documentation

 * What's new in C# 14 - The field keyword - https://learn.microsoft.com/en-us/dotnet/csharp/whats-new/csharp-14#the-field-keyword[Official Microsoft documentation explaining the field keyword feature in C# 14]

 * Breaking changes in Roslyn - field keyword conflicts - https://learn.microsoft.com/en-us/dotnet/csharp/whats-new/breaking-changes/compiler%20breaking%20changes%20-%20dotnet%2010[Documentation of breaking changes related to the field keyword in C# 14]
