<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>There is no reason to store a value that is never read. Doing so indicates a bug: was the value assigned to the wrong variable? Was some calculation left out of the method? If it is not a bug, it is a waste of cycles and an obfuscation of the code.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-text" data-lang="text">public int getNumber(int a) {
  int i;
  int b = a;

  for (i = 0; i &lt; 10; i++) {
    b *= i;
  }

  i = a;  // Noncompliant; value assigned but not used before return

  return b;
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-text" data-lang="text">public int getNumber(int a) {
  int i;
  int b = a;

  for (i = 0; i &lt; 10; i++) {
    b *= i;
  }

  return b;
}</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove this useless assignment of "xxx".</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_10_oct_2014_140508_freddy_mallet_wrote">on 10 Oct 2014, 14:05:08 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>@Ann, for me this RSPEC duplicates <a data-rspec-id="S1981" class="rspec-auto-link">RSPEC-1981</a></p>
</div>
</div>
</div>
</div>