include::../summary.adoc[]

== Why is this an issue?

Jump statements (``++return++``, ``++break++``, ``++continue++``, and ``++fallthrough++``) move control flow out of the current code block. So any statements that come after a jump are dead code.

=== Noncompliant code example

[source,swift]
----
func fun(a:Int)->Int{
  var i = 10;
  return i + a;
  i++;             // this is never executed
}
----

=== Compliant solution

[source,swift]
----
func fun(a:Int)->Int{
  var i = 10;
  return i + a;
}
----

include::../see.adoc[]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../highlighting.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
