Jump statements (``++return++``, ``++break++``, ``++continue++``, and ``++fallthrough++``) move control flow out of the current code block. So any statements that come after a jump are dead code.

== Noncompliant Code Example

----
func fun(a:Int)->Int{
  var i = 10;
  return i + a;
  i++;             // this is never executed
}
----

== Compliant Solution

----
func fun(a:Int)->Int{
  var i = 10;
  return i + a;
}
----

include::../see.adoc[]

ifdef::env-github,rspecator-view[]
'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
