== Why is this an issue?

Shared naming conventions allow teams to collaborate efficiently. This rule checks that all [report/]program names match a provided regular expression.


=== Noncompliant code example

With the regular expression ``++^([A-Z0-9_]*|[a-z0-9_]*)$++``:

[source,text]
----
    IDENTIFICATION DIVISION.
       PROGRAM-ID. MY-PROGRAM.      *> Noncompliant
----


=== Compliant solution

[source,text]
----
    IDENTIFICATION DIVISION.
       PROGRAM-ID. MY_PROGRAM.
----

