<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>In Gradle, using the <code>GradleBuild</code> task type you can invoke another Gradle build and execute some tasks from another project. See the example below:</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-kotlin" data-lang="kotlin">tasks.register&lt;GradleBuild&gt;("execModuleTask") {
    dir = File("module/build.gradle.kts")
    tasks = listOf("task")
}</code></pre>
</div>
</div>
<div class="paragraph">
<p>In general, it is considered to be a bad practice because there are corner cases where the invoked build behaves differently compared to executing it from the command line or through the Tooling API. This can lead to unexpected results.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Change the way to model the requirement. The appropriate approach depends on the problem at hand. Hereâ€™re some options:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>Model the build as <a href="https://docs.gradle.org/current/userguide/multi_project_builds.html#multi_project_builds">multi-project build</a> if the intention is to execute tasks from different modules as unified build.</p>
</li>
<li>
<p>Use <a href="https://docs.gradle.org/current/userguide/composite_builds.html#composite_builds">composite builds</a> for projects that are physically separated but should occasionally be built as a single unit.</p>
</li>
</ul>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-kotlin" data-lang="kotlin">tasks.register&lt;GradleBuild&gt;("execModuleTask") {
    dir = File("module/build.gradle.kts")
    tasks = listOf("task")
}</code></pre>
</div>
</div>
<div class="paragraph">
<p>In order to execute it you need to run it:</p>
</div>
<div class="listingblock">
<div class="content">
<pre>    ./gradlew execModuleTask</pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="paragraph">
<p>To fix it you can make <code>module</code> your submodule and run it with:</p>
</div>
<div class="listingblock">
<div class="content">
<pre>    ./gradlew :module:task</pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p><a href="https://docs.gradle.org/current/userguide/authoring_maintainable_build_scripts.html#sec:avoiding_use_of_gradlebuild">Avoiding use of GradleBuild</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>