<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>You could you use preprocessor directives to do some pretty wild things, up to and including redefining the syntax of the language. But doing so <em>could</em> result in unexpected behavior, and definitely <em>would</em> result in impossible-to-read code. Therefore the acceptable uses of macros are strictly limited.</p>
</div>
<div class="paragraph">
<p>Specifically, macros may only be used do define a:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>braced initialzer</p>
</li>
<li>
<p>constant</p>
</li>
<li>
<p>parenthesized expression</p>
</li>
<li>
<p>type qualifier</p>
</li>
<li>
<p>storage class specifier</p>
</li>
<li>
<p><code>do</code>-<code>while (0)</code> construct</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>Macros should not be used to define statements or parts of statements except for the use of the <code>do</code>-<code>while (0)</code> construct, or to redefine the syntax of the language. All brackets ( (), {}, and [] ) in a macro must be balanced</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">#define int32_t long /* Noncompliant; use typedef instead */
#define STRTIF if( /* Noncompliant; unbalanced () and language redefinition */
#define XSTAL 10000000 /* Compliant; constant */
#define CLOCK (XSTAL/16) /* Compliant; constant expression */
#define PLUS2(X) ((X) + 2) /* Compliant; macro expanding to expression */
#define STOR extern /* Compliant; storage class specifier */
#define INIT(value) { (value), 0, 0} /* Compliant; braced initialiser */
#define READ_TIME_32() \  /* Compliant; example of do-while-zero */
  do { \
    DISABLE_INTERRUPTS (); \
    time_now = (uint32_t)TIMER_HI &lt;&lt; 16; \
    time_now = time_now | (uint32_t)TIMER_LO; \
    ENABLE_INTERRUPTS (); \
  } while (0)  /* note the semicolon must be omitted here at the end */
...

STRTIF my_bool )  /* Huh? This is unreadable */
{
  ...
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">#define XSTAL 10000000
#define CLOCK (XSTAL/16)
#define PLUS2(X) ((X) + 2)
#define STOR extern
#define INIT(value) { (value), 0, 0}
#define READ_TIME_32() \
  do { \
    DISABLE_INTERRUPTS (); \
    time_now = (uint32_t)TIMER_HI &lt;&lt; 16; \
    time_now = time_now | (uint32_t)TIMER_LO; \
    ENABLE_INTERRUPTS (); \
  } while (0)

typedef int32_t long;

...

if (my_bool)
{
  ...
}</code></pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p>MISRA C:2004, 19.4</p>
</li>
</ul>
</div>
<hr>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove this macro or redefine it to be conformant</p>
</div>
</div>
</div>
</div>