<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Referencing a static member of a subclass from its parent during class initialization, makes the code more fragile and prone to future bugs.
The execution of the program will rely heavily on the order of initialization of classes and their static members.</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="paragraph">
<p>This could create what is known as an "initialization cycle", or even a deadlock in some extreme cases.
Additionally, if the order of the static class members is changed, the behavior of the program might change.
These issues can be very hard to diagnose so it is highly recommended to avoid creating this kind of dependencies.</p>
</div>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">class Parent {
  static int field1 = Child.method(); // Noncompliant
  static int field2 = 42;

  public static void main(String[] args) {
    System.out.println(Parent.field1); // will display "0" instead of "42"
  }
}

class Child extends Parent {
  static int method() {
    return Parent.field2;
  }
}</code></pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p>CERT - <a href="https://wiki.sei.cmu.edu/confluence/display/java/DCL00-J.+Prevent+class+initialization+cycles">DCL00-J. Prevent class initialization cycles</a></p>
</li>
<li>
<p><a href="https://docs.oracle.com/javase/specs/jls/se17/html/jls-12.html#jls-12.4">Section 12.4: Initialization of Classes and Interfaces</a> - Java Language Specification</p>
</li>
</ul>
</div>
<hr>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove this reference to "xxx".</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_07_jun_2023_151813_leonardo_pilastri_wrote">on 07 Jun 2023, 15:18:13 Leonardo Pilastri wrote:</h3>
<div class="paragraph">
<p>The rule is very limited and is mentioning the parent-child class relationship, which is actually not the root cause.
Should be reworked.</p>
</div>
</div>
</div>
</div>