== Why is this an issue?

While it is possible to access a global variable, function or enum constant without using the ``++::++`` operator, it can be considered to be misleading because it might imply to the readers of your code that this is a local or class variable/function/enum constant and not a global one. Being explicit also allows more freedom in naming local variables without the chance of clashing with global names.


=== Noncompliant code example

[source,cpp]
----
int a = 10;

enum Enum {
  C
};

int main()
{
  ...
  int b = a;    // Noncompliant
  Enum d = C;
  ...
}
----


=== Compliant solution

[source,cpp]
----
int a = 10;

enum Enum {
  C
};

int main()
{
  ...
  int b = ::a;    // Compliant
  Enum d = ::C;
  ...
}
----



ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Use "::" to access "XX".


=== Highlighting

The variable or function name.


'''
== Comments And Links
(visible only on this page)

=== relates to: S2209
=== relates to: S3642

endif::env-github,rspecator-view[]
