== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,cpp]
----
void called(int foo) {
  foo==1; // Noncompliant
  if (foo==1) {
    printf("foo\n");
  }
}

int main() {
  called(2);
  return 0;
}
----

=== Compliant solution

[source,cpp]
----
void called(int foo) {
  foo=1;
  if (foo==1) {
    printf("foo\n");
  }
}

int main() {
  called(2);
  return 0;
}
----

include::../see.adoc[]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
