//// This rule is superseded by RSPEC-2583
//// Please consider implementing this latter instead.
include::../description.adoc[]

== Noncompliant Code Example

----
if true
  doSomething()
end
...
if false
  doSomethingElse()
end
----

== Compliant Solution

----
doSomething()
...
----

include::../see.adoc[]
