This rule raises an issue when a `TypeVar` is used as a type parameter in a `type` statement.

== Why is this an issue?

Prior to Python 3.12, generic type aliases were defined as follows:

[source,python]
----
from typing import TypeAlias, TypeVar

_T = TypeVar("_T")

MyAlias: TypeAlias = set[_T]
----

Python 3.12 introduced the `type` statement to facilitate the use of such type aliases, 
allowing for a less confusing and more concise code:

[source,python]
----
type MyAlias[T] = set[T]
----

It is not possible to mix a `TypeVar` with the `type` statement. 
Type checkers will raise an error for the following case:

[source,python]
----
from typing import TypeVar

_T = TypeVar("_T")

type MyAlias[A] = dict[A, _T]  # Type checker error would be raise
----

Mixing the two styles of type variables is a bad practices as it is confusing and my result in unpredictable behavior:

[source,python]
----
from typing import TypeVar

T = TypeVar("T", int, float)

type MyAlias[A] = dict[A, T]  # Type checker error would be raise

def foo[A, T: str](a: MyAlias[A], b: MyAlias[B]):
    ....
----


=== Exceptions

This rule will only raise an issue when the Python version of the analyzed project is set to 3.12 or higher.

== How to fix it

To fix this error use a generic `type` statement and remove the use of the `TypeVar`.

=== Code examples

==== Noncompliant code example

[source,python,diff-id=1,diff-type=noncompliant]
----
from typing import TypeAlias

_T = TypeVar("_T")

type MyAlias = set[_T]  # Noncompliant: a TypeVar is used as part of the type statement
----

==== Compliant solution

[source,python,diff-id=1,diff-type=compliant]
----
type MyAlias[T] = set[T]  # Compliant: the new type statement syntax is used.
----

== Resources
=== Documentation

* Python Documentation - https://docs.python.org/3.12/reference/simple_stmts.html#type[The type statement]
* Python 3.12 Release Notes - https://docs.python.org/3.12/whatsnew/3.12.html#pep-695-type-parameter-syntax[PEP 695: Type Parameter Syntax]
* PEP 695 - https://peps.python.org/pep-0695/[Type Parameter Syntax]


ifdef::env-github,rspecator-view[]

'''

== Implementation Specification
=== Message
(visible only on this page)

Use a generic type parameter instead of a `TypeVar` in this `type` statement.

'''
endif::env-github,rspecator-view[]
