include::../description.adoc[]

== Noncompliant Code Example

With a maximum number of 4 parameters:

----
func doSomething(param1: String, param2: String, param3: String, param4: String, param5: String) {
    // ...
}
----

== Compliant Solution

----
func doSomething(param1: String, param2: String, param3: String, param4: String) {
    // ...
}
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

include::../parameters.adoc[]

'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
