== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

With a maximum number of 4 parameters:

[source,swift]
----
func doSomething(param1: String, param2: String, param3: String, param4: String, param5: String) {
    // ...
}
----

=== Compliant solution

[source,swift]
----
func doSomething(param1: String, param2: String, param3: String, param4: String) {
    // ...
}
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

 [Function|Initializer] X has {0} parameters, which is greater than the {1} authorized.


include::../parameters.adoc[]

'''
== Comments And Links
(visible only on this page)

=== on 7 May 2018, 09:52:00 Andrei Epure wrote:
After an internal discussion, I decided to remove the check on subscripts, as it's too much of a corner case.

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
