include::../description.adoc[]

== Noncompliant Code Example

----
$myNumber = 023; # Noncompliant myNumber will hold 19, not 23 - was this really expected?
----

== Compliant Solution

----
$myNumber = 19;
----

== Exceptions

This rule does not apply to values smaller than 8 and octal values having 3 digits, since 3 digits octal values are often used as file permission masks.
 Example:

----
$permissionMask = 0777;
$dayOfMonth = 03;
----

include::../see.adoc[]
