== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,php]
----
$myNumber = 023; # Noncompliant myNumber will hold 19, not 23 - was this really expected?
----

=== Compliant solution

[source,php]
----
$myNumber = 19;
----

=== Exceptions

This rule does not apply to values smaller than 8 and octal values having 3 digits, since 3 digits octal values are often used as file permission masks.

Example:

[source,php]
----
$permissionMask = 0777;
$dayOfMonth = 03;
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
