According to the Java Language Specification:


____
It is permitted, but discouraged as a matter of style, to redundantly specify the ``++public++`` and/or ``++abstract++`` modifier for a method declared in an interface.

____


== Noncompliant Code Example

----
public interface Task{
  public abstract void execute();
}
----


== Compliant Solution

----
public interface Task{
  void execute();
}
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

endif::env-github,rspecator-view[]
