<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when Rails models override the default primary key using <code>self.primary_key =</code> or when database tables are created without primary keys using <code>id: false</code>.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Rails is built around the convention of using integer auto-incrementing primary keys named <code>id</code>. This convention enables many framework features to work seamlessly.</p>
</div>
<div class="paragraph">
<p>When you override the primary key with <code>self.primary_key = "custom_field"</code>, several problems can occur:</p>
</div>
<div class="ulist">
<ul>
<li>
<p><strong>Foreign key mismatches</strong>: Rails generates foreign keys expecting integer IDs, but your custom primary key might be a string or other type</p>
</li>
<li>
<p><strong>Routing issues</strong>: Rails URL helpers expect integer IDs in routes like <code>/users/123</code></p>
</li>
<li>
<p><strong>Plugin compatibility</strong>: Many Rails gems and plugins assume standard primary keys</p>
</li>
<li>
<p><strong>Association problems</strong>: <code>belongs_to</code> and <code>has_many</code> relationships may not work correctly</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>Creating tables without primary keys using <code>id: false</code> causes different but equally serious issues:</p>
</div>
<div class="ulist">
<ul>
<li>
<p><strong>DELETE operation failures</strong>: ActiveRecord cannot generate proper DELETE queries without a primary key, leading to SQL syntax errors</p>
</li>
<li>
<p><strong>Record identification problems</strong>: The ORM cannot uniquely identify records for updates or deletions</p>
</li>
<li>
<p><strong>Dependent destroy failures</strong>: Cascading deletes and <code>dependent: :destroy</code> options will not work</p>
</li>
<li>
<p><strong>Replication issues</strong>: Database replication systems often require primary keys</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>These problems are particularly common in join tables where developers mistakenly think primary keys are unnecessary.</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="paragraph">
<p>Applications may experience runtime errors during record deletion, foreign key constraint violations, broken associations between models, and compatibility issues with Rails plugins and gems. In production, this can lead to data integrity problems and application crashes.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it_in_rails">How to fix it in Rails</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Remove the custom primary key assignment and let Rails use its default integer <code>id</code> primary key. Update any foreign key references to use the standard naming convention.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-ruby" data-lang="ruby">class State &lt; ActiveRecord::Base
  self.primary_key = "state" # Noncompliant
  has_many :acquisitions
end</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-ruby" data-lang="ruby">class State &lt; ActiveRecord::Base
  # Rails automatically creates an 'id' integer primary key
  has_many :acquisitions
end</code></pre>
</div>
</div>
<div class="paragraph">
<p>Remove the <code>id: false</code> option from table creation to allow Rails to create the default primary key. This ensures proper record identification and deletion operations.</p>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_code_examples_2">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example_2">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-ruby" data-lang="ruby">create_table :regions, id: false do |t| # Noncompliant
  t.belongs_to :company, index: true
  t.belongs_to :province, index: true
end</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution_2">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-ruby" data-lang="ruby">create_table :regions do |t|
  t.belongs_to :company, index: true
  t.belongs_to :province, index: true
  t.timestamps
end</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Rails Active Record Basics - <a href="https://guides.rubyonrails.org/active_record_basics.html#naming-conventions">Official Rails guide explaining naming conventions and primary key defaults</a></p>
</li>
<li>
<p>Rails Migrations Guide - <a href="https://guides.rubyonrails.org/active_record_migrations.html#creating-a-table">Official documentation on creating tables with proper primary keys</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>