<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Having a <code>nillable</code> element in a schema means that element must always be present, whether or not you have data for it. That means that the resulting XML transmissions will be larger than they need to be. On the other hand, <code>minOccurs="0"</code> communicates the same "there is no value" message, with smaller XML messages. For that reason, <code>minOccurs="0"</code> is <em>almost</em> always preferred over <code>nillable="true"</code>. The one case were <code>nillable="true"</code> is valuable is in the case of an array element, where you may have the need to set a placeholder.</p>
</div>
<div class="paragraph">
<p>This rule raises an issue when <code>nillable="true"</code> is used on an element were <code>maxOccurs</code> is 1.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-xml" data-lang="xml">&lt;complexType name="fruitArray" nillable="true"&gt;            &lt;!-- Noncompliant --&gt;
  &lt;sequence&gt;
    &lt;element name="fruit" type="xs:string" nillable="true" maxOccurs="4"/&gt;    &lt;!-- ignored --&gt;
  &lt;/sequence&gt;
&lt;/complexType&gt;</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-xml" data-lang="xml">&lt;complexType name="fruitArray"&gt;
  &lt;sequence&gt;
    &lt;element name="fruit" type="xs:string" minOccurs="0" maxOccurs="4"/&gt;
  &lt;/sequence&gt;
&lt;/complexType&gt;</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Use 'minOccurs="0"' here instead of 'nillable="true"'.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p><code>nillable="true"</code></p>
</div>
</div>
</div>
</div>