== Why is this an issue?

The PSR2 standard recommends listing modifiers in the following order to improve the readability of PHP source code:


. final or abstract
. public or protected or private
. static

=== Noncompliant code example

[source,php]
----
static protected $foo;
... 
public static final function bar(){...}
----

=== Compliant solution

[source,php]
----
protected static $foo;
... 
final public static function bar(){...}
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
