== Why is this an issue?

Since C# 5.0, ``++async++`` and ``++await++`` are contextual keywords. Contextual keywords do have a particular meaning in some contexts, but can still be used as variable names. Keywords, on the other hand, are always reserved, and therefore are not valid variable names. To avoid any confusion though, it is best to not use ``++async++`` and ``++await++`` as identifiers.


=== Noncompliant code example

[source,csharp]
----
int await = 42; // Noncompliant
----


=== Compliant solution

[source,csharp]
----
int someOtherName = 42;
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Rename "xxx" to not use a contextual keyword as an identifier.


'''
== Comments And Links
(visible only on this page)

=== is related to: S1190

=== on 12 Jan 2015, 15:02:43 Freddy Mallet wrote:
I would tend to associate this coding rule to the characteristic "Portability -> Language related Portability" and also to the tag "obsolete"

endif::env-github,rspecator-view[]
