== Why is this an issue?

Shared naming conventions allow teams to collaborate efficiently.

This rule raises an issue when a method or a property name is not snake_cased.

== How to fix it
Rename the function so that its name conforms to the snake_case convention.

=== Code examples

==== Noncompliant code example

[source,bash,diff-id=1,diff-type=noncompliant]
----
function doSomething() {...} // Noncompliant
----

==== Compliant solution

[source,bash,diff-id=1,diff-type=compliant]
----
function do_something() {...}
----

=== External coding guidelines
https://courses.cs.washington.edu/courses/cse374/22wi/resources/bash_style_guide.html[CSE 374 Bash Style Guide]

https://google.github.io/styleguide/shellguide.html#function-names[Google Shell Style Guide]

== Implementation Specification
(visible only on this page)

include::../message.adoc[]

== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
