<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>To aid code readability, all the <code>#include</code> directives in a particular code file should be grouped together near the top of the file. The only items which may precede an <code>#include</code> in a file are other preprocessor directives or comments.</p>
</div>
<div class="paragraph">
<p>Additionally, an <code>#include</code> may appear within an <code>extern "C"</code> block, this can be used for instance to include a C file from a C&#43;&#43; file.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">#include &lt;h1.h&gt; /* Compliant */
int32_t i;
#include &lt;f2.h&gt; /* Noncompliant */</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">#include &lt;h1.h&gt;
#include &lt;f2.h&gt;
extern "C" {
#include &lt;f3.h&gt;
}

int32_t i;</code></pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p>MISRA C:2004, 19.1 - #include statements in a file should only be preceded by other preprocessor directives or comments.</p>
</li>
<li>
<p>MISRA C&#43;&#43;:2008, 16-0-1 - #include directives in a file shall only be preceded by other preprocessor directives or comments.</p>
</li>
<li>
<p>MISRA C:2012, 20.1 - #include directives should only be preceded by preprocessor directives or comments</p>
</li>
</ul>
</div>
<hr>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_31_mar_2015_190055_evgeny_mandrikov_wrote">on 31 Mar 2015, 19:00:55 Evgeny Mandrikov wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] implementation seems more complete (SQALE, description) than this spec.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_13_apr_2015_193557_evgeny_mandrikov_wrote">on 13 Apr 2015, 19:35:57 Evgeny Mandrikov wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] I&#8217;m wondering why blocker, but not active by default? Note that in implementation currently major and active.</p>
</div>
</div>
</div>
</div>