To aid code readability, all the ``#include`` directives in a particular code file should be grouped together near the top of the file. The only items which may precede an ``#include`` in a file are other preprocessor directives or comments.


== Noncompliant Code Example

----
#include <h1.h> /* Compliant */
int32_t i;
#include <f2.h> /* Noncompliant */
----


== Compliant Solution

----
#include <h1.h>
#include <f2.h>

int32_t i;
----


== See

* MISRA C:2004, 19.1 - #include statements in a file should only be preceded by other preprocessor directives or comments.
* MISRA {cpp}:2008, 16-0-1 - #include directives in a file shall only be preceded by other preprocessor directives or comments.
* MISRA C:2012, 20.1 - #include directives should only be preceded by preprocessor directives or comments

