== How to fix it

=== Code examples

In the following code example, ``memchr`` is used to find the first occurrence of the ``@`` character in a buffer, ``array``. The second argument does not match the size of this buffer, so it may find an occurrence of the character after the end of the buffer and read data it is not supposed to. If the returned ``pos`` pointer is later used to write characters, this could unexpectedly overwrite data.

==== Noncompliant code example

[source,cpp,diff-id=1,diff-type=noncompliant]
----
char array[10];
initialize(array);
char *pos = memchr(array, '@', 42); // Noncompliant
----


==== Compliant solution

[source,cpp,diff-id=1,diff-type=compliant]
----
char array[10];
initialize(array);
char *pos = memchr(array, '@', sizeof(array));
----

