== How to fix it

=== Code examples

=== Noncompliant code example

[source,cpp,diff-id=1,diff-type=noncompliant]
----
char array[10];
initialize(array);
void *pos = memchr(array, '@', 42); // Noncompliant, buffer overflow that could expose sensitive data 
----


=== Compliant solution

[source,cpp,diff-id=1,diff-type=compliant]
----
char array[10];
initialize(array);
void *pos = memchr(array, '@', 10);
----

