include::../description.adoc[]

== Noncompliant Code Example

----
class Foo 
{
    static void Main() 
    {
        for (int i = 1; i <= 5; i++) 
        {
            Console.WriteLine(i);
            if (condition) 
            {
               i = 20;
           }
        }
    }
}
----

== Compliant Solution

----
class Foo 
{
    static void Main() 
    {
        for (int i = 1; i <= 5; i++) 
        {
            Console.WriteLine(i);
        }
    }
}
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../parameters.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
