== Why is this an issue?

This rule reduces confusion over which constructor will be used, and with what parameters.

=== Noncompliant code example

Given the following classes:

[source,cpp]
----
class V
{
public:
  V ( ) { }
  V ( int32_t i ) { }
};

class C1 : public virtual V
{
public: 
 C1 ( ) : V ( 21 ) { }
};

class C2 : public virtual V
{
public: 
  C2 ( ) : V ( 42 ) { }
};
----
The following examples are not compliant:

[source,cpp]
----
class B : public V
{
  public B( ) { } // Non-compliant - benign
}

class D: public C1, public C2
{ 
public: 
 D ( ) { } // Non-compliant - it is unclear which constructor of V is called
};
----

=== Compliant solution

Given the previous definitions of classes V, C1 and C2 the following definitions of B and D are compliant:

[source,cpp]
----
class B : public V
{
  public B( ) : V ( ) { } // Compliant
}

class D: public C1, public C2 
{ 
public: 
 D ( ) : C1 ( ), C2 ( ), V ( ) { } // Compliant
};
----

== Resources

* MISRA {cpp}:2008, 12-1-2 - All constructors of a class should explicitly call a constructor for all of its immediate base classes and all virtual base classes.
ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Add an explicit call of constructor for [base class xxx::virtual base class xxx]


endif::env-github,rspecator-view[]
