== Why is this an issue?

Shared naming conventions allow teams to collaborate effectively. This rule raises an issue when a pom's ``++artifactId++`` does not match the provided regular expression.


=== Noncompliant code example

With the default regular expression: ``++[a-z][a-z-0-9]+++``

[source,xml]
----
<project ...>
  <artifactId>My_Project</artifactId>  <!-- Noncompliant -->

  <!-- ... -->
</project>
----


=== Compliant solution

[source,xml]
----
<project ...>
  <artifactId>my-project</artifactId>

  <!-- ... -->
</project>
----



ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Update this "artifactId" to match the provided regular expression: xxx


=== Parameters

.regex
****

----
[a-z][a-z-0-9]+
----

The regular expression the "artifactId" should match
****


=== Highlighting

artifactId value


'''
== Comments And Links
(visible only on this page)

=== on 25 Nov 2015, 08:57:45 Freddy Mallet wrote:
For rules applying to pom.xml Maven file like this one I would use the tag 'maven' instead of 'pom' [~ann.campbell.2] because for a non-Maven users 'pom' doesn't mean anything. My 2 cents

=== on 30 Nov 2015, 14:07:41 Michael Gumowski wrote:
+1 with Freddy's comment!

Otherwise, LGTM [~ann.campbell.2]!

=== on 30 Nov 2015, 18:46:55 Ann Campbell wrote:
labels updated

endif::env-github,rspecator-view[]
