== Why is this an issue?

In Java 15 Text Blocks are now official and can be used. The most common pattern for multiline strings in Java < 15 was to write String concatenation. Now it's possible to do it in a more natural way using Text Blocks.


=== Noncompliant code example

[source,java]
----
String textBlock =
               "<html>\n" +
               "    <body>\n" +
               "        <tag>\n" +
               "        </tag>\n" +
               "    </body>\n" +
               "</html>";
----


=== Compliant solution

[source,java]
----
String textBlock = """
        <html>
            <body>
                <tag>
                </tag>
            </body>
        </html>""";
----


== Resources

* https://openjdk.java.net/jeps/378[JEP 378: Text Blocks]
* https://openjdk.org/projects/amber/guides/text-blocks-guide[Programmer's Guide To Text Blocks], by Jim Laskey and Stuart Marks


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Replace this String concatenation with Text block.


endif::env-github,rspecator-view[]
