In Java 15 Text Blocks are now official and can be used. The most common pattern for multiline strings in Java < 15 was to write String concatenation. Now it's possible to do it in a more natural way using Text Blocks.


== Noncompliant Code Example

[source,java]
----
String textBlock = 
               "<html>\n" +
               "    <body>\n" +
               "        <tag>\n" +
               "        </tag>\n" +
               "    </body>\n" +
               "</html>";
----


== Compliant Solution

[source,java]
----
String textBlock = """
        <html>
            <body>
                <tag>
                </tag>
            </body>
        </html>""";
----


== See

* https://openjdk.java.net/jeps/378[JEP 378: Text Blocks]
* https://cr.openjdk.java.net/~jlaskey/Strings/TextBlocksGuide_v9.html[Programmer's Guide To Text Blocks], by Jim Laskey and Stuart Marks


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

endif::env-github,rspecator-view[]
