include::../description-common.adoc[]

=== Code examples

==== Noncompliant code example

[source,rust,diff-id=1,diff-type=noncompliant]
----
if a && b || a { // Noncompliant: `b` is redundant
  // ...
}
----

==== Compliant solution

[source,rust,diff-id=1,diff-type=compliant]
----
if a {
  // ...
}
----

== Resources
=== Documentation

* Clippy Lints - https://rust-lang.github.io/rust-clippy/master/index.html#overly_complex_bool_expr
