<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when an asynchronous function accepts a <code>timeout</code> parameter instead of using the timeout context managers.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Modern asynchronous Python libraries like <code>asyncio</code>, <code>anyio</code>, and <code>trio</code> promote a principle called <strong>structured concurrency</strong>. A key aspect of this is that the caller of an asynchronous function should be responsible for managing timeouts and cancellation, not the callee.</p>
</div>
<div class="paragraph">
<p>When an <code>async</code> function accepts a <code>timeout</code> parameter, it violates this principle:</p>
</div>
<div class="olist arabic">
<ol class="arabic">
<li>
<p><strong>Coupling between logic and timeout handling</strong>: The function dictates how the timeout is handled internally, rather than letting the caller decide.</p>
</li>
<li>
<p><strong>Preempting caller control</strong>: The caller might want to enforce a different timeout duration or coordinate timeouts across several concurrent operations. An internal timeout parameter makes this difficult or impossible.</p>
</li>
<li>
<p><strong>Reducing composability</strong>: Combining functions that manage their own timeouts can lead to complex and unpredictable behavior, especially when nesting calls or running tasks concurrently under a shared deadline.</p>
</li>
</ol>
</div>
<div class="paragraph">
<p>Instead, the caller should use the timeout features provided by the concurrency library (e.g., <code>async with asyncio.timeout()</code> or <code>with trio.move_on_after()</code>).
This separates the concern of what the function does from how long the caller is willing to wait for it.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it_in_asyncio">How to fix it in Asyncio</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">import asyncio

async def example_function(timeout): # Noncompliant
    await asyncio.sleep(timeout)

async def main():
    await example_function(5)</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">import asyncio

async def example_function():
    await asyncio.sleep(5)

async def main():
    async with asyncio.timeout(5): # Compliant
        await example_function()</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it_in_trio">How to fix it in Trio</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_code_examples_2">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example_2">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">import trio

async def example_function(timeout): # Noncompliant
    await trio.sleep(timeout)

async def main():
    await example_function(5)

trio.run(main)</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution_2">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">import trio

async def example_function():
    await trio.sleep(5)

async def main():
    with trio.move_on_after(5): # Compliant
        await example_function()

trio.run(main)</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it_in_anyio">How to fix it in AnyIO</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_code_examples_3">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example_3">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">import anyio

async def example_function(timeout): # Noncompliant
    await anyio.sleep(timeout)

async def main():
    await example_function(5)

anyio.run(main)</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution_3">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">import anyio

async def example_function():
    await anyio.sleep(5)

async def main():
    with anyio.move_on_after(5): # Compliant
        await example_function()

anyio.run(main)</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Asyncio documentation - <a href="https://docs.python.org/3/library/asyncio-task.html#asyncio.timeout">Timeouts</a></p>
</li>
<li>
<p>Trio documentation - <a href="https://trio.readthedocs.io/en/stable/reference-core.html#cancellation-and-timeouts">Cancellation and timeouts</a></p>
</li>
<li>
<p>AnyIO documentation - <a href="https://anyio.readthedocs.io/en/stable/cancellation.html#timeouts">Timeouts</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove the <code>timeout</code> parameter from the function and use the timeout context managers instead.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>The <code>timeout</code> parameter in the function definition, with an eventual typehint.
Secondary location on the async keyword of the function definition.</p>
</div>
</div>
<div class="sect2">
<h3 id="_quickfix">Quickfix</h3>
<div class="paragraph">
<p>No</p>
</div>
</div>
</div>
</div>