== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,ruby]
----
x = (y / 2 + 1)  # Compliant even if the parenthesis are ignored by the compiler

if a && ((x+y > 0))  # Noncompliant
  # ...
end

return ((x + 1)) # Noncompliant
----

=== Compliant solution

[source,ruby]
----
x = (y / 2 + 1)

if a && (x+y > 0)
  # ...
end

return (x + 1)
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../highlighting.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
