include::../description.adoc[]

== Noncompliant Code Example

----
x = (y / 2 + 1)  # Compliant even if the parenthesis are ignored by the compiler

if a && ((x+y > 0))  # Noncompliant
  # ...
end

return ((x + 1)) # Noncompliant
----

== Compliant Solution

----
x = (y / 2 + 1)

if a && (x+y > 0)
  # ...
end

return (x + 1)
----

ifdef::env-github,rspecator-view[]
'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
