Nested classes, methods and variables that are package-private (default visibility) and never accessed outside the file they are declared in, should be private instead.


== Why is this an issue?

If specific visibility is not defined for a component, then it will be package-private.
This means that we intend to access the component from another file within the same package.
If this is not the case, then the component should be private instead.

//=== Code examples

//==== Noncompliant code example

//[source,text,diff-id=1,diff-type=noncompliant]
//----
//FIXME
//----

//==== Compliant solution

//[source,text,diff-id=1,diff-type=compliant]
//----
//FIXME
//----

//=== How does this work?

//=== Pitfalls

//=== Going the extra mile


//== Resources
//=== Documentation
//=== Articles & blog posts
//=== Conference presentations
//=== Standards
//=== Benchmarks
