<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Whenever there are portions of code that are duplicated and do not depend on the state of their
container class, they can be centralized inside a "utility class".
A utility class is a class that only has static members, hence it should not be instantiated.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>To prevent the class from being instantiated, you should define a non-public constructor.
This will prevent the compiler from implicitly generating a public parameterless constructor.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-dart" data-lang="dart">class StringUtils { // Noncompliant

  static String concatenate(String s1, String s2) {
    return s1 + s2;
  }

}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-dart" data-lang="dart">class StringUtils { // Compliant

  StringUtils._() {
    throw Exception('Utility class');
  }

  static String concatenate(String s1, String s2) {
    return s1 + s2;
  }

}</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Hide this public constructor.</p>
</div>
<div class="paragraph">
<p>Add a private constructor to hide the implicit public one.</p>
</div>
<hr>
</div>
</div>
</div>