This is an issue when `defined?` is used with hash key access (like `params[:key]` or `session[:id]`) to check if a key exists or has a meaningful value.

== Why is this an issue?

The `defined?` operator checks whether an expression can be evaluated without raising an exception, not whether it returns a meaningful value.

When you access a hash key using square brackets (`hash[:key]`), Ruby always returns either the actual value or `nil` - it never raises an exception for missing keys. This means `defined? hash[:key]` will always return `"method"` (which is truthy), regardless of whether the key exists or what value it contains.

As a result, conditions using `defined?` with hash access will always evaluate to true, making your conditional logic ineffective. This can lead to unexpected behavior where code blocks execute when they shouldn't, or where you think you're checking for key existence but you're actually not checking anything meaningful.

This pattern is particularly common in Rails applications when checking request parameters, session data, or other hash-like objects.

=== What is the potential impact?

The code will behave unexpectedly because conditional checks will always pass. This can lead to:

* Logic errors where code executes when it shouldn't
* Security issues if the checks were meant to validate required parameters
* Difficult-to-debug problems since the condition appears correct but doesn't work as intended

== How to fix it

Use truthiness checks to verify that keys have meaningful (non-nil, non-false) values.

=== Code examples

==== Noncompliant code example

[source,ruby,diff-id=1,diff-type=noncompliant]
----
if (defined? params[:one]) && (defined? params[:two]) # Noncompliant
  # This will always execute if params exists
  process_data
end
----

==== Compliant solution

[source,ruby,diff-id=1,diff-type=compliant]
----
if params[:one] && params[:two]
  # Only executes if both values are truthy
  process_data
end
----

== How to fix it in Rails

In Rails applications, use `present?` to check if parameters have meaningful values (not nil, empty, or blank).

=== Code examples

==== Noncompliant code example

[source,ruby,diff-id=2,diff-type=noncompliant]
----
if (defined? params[:name]) && (defined? params[:email]) # Noncompliant
  create_user
end
----

==== Compliant solution

[source,ruby,diff-id=2,diff-type=compliant]
----
if params[:name].present? && params[:email].present?
  create_user
end
----

== Resources

=== Documentation

 * Ruby defined? operator documentation - https://ruby-doc.org/core/doc/syntax/miscellaneous_rdoc.html#label-defined-3F[Official Ruby documentation explaining the defined? operator]

 * Ruby Hash class documentation - https://ruby-doc.org/core/Hash.html[Documentation for Ruby Hash methods including has_key? and key?]

 * Rails Object#present? documentation - https://api.rubyonrails.org/classes/Object.html#method-i-present-3F[Rails documentation for the present? method]
