This rule raises an issue when function name is out of python naming convention (snake case).

== Why is this an issue?

include::../description.adoc[]

According to PEP8 function names should be in all lowercase letters, multiple words should be separated by underscores. This convention is known as "snake_case." For example: ``++calculate_area++``, ``++print_hello++``, ``++process_data++``

=== How to fix it
Use the PEP8 standard of the naming strategy for Python.

=== Noncompliant code example

With the default provided regular expression: ``++^[a-z_][a-z0-9_]*$++``

[source,python,diff-id=1,diff-type=noncompliant]
----
def MyFunction(a,b):
    ...
----

=== Compliant solution

[source,python,diff-id=1,diff-type=compliant]
----
def my_function(a,b):
    ...
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

=== Parameters

.format
****

----
^[a-z_][a-z0-9_]*$
----

Regular expression used to check the function names against
****


'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
