== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

With the default provided regular expression: ``++^[a-z_][a-z0-9_]*$++``

[source,python]
----
def MyFunction(a,b):
    ...
----

=== Compliant solution

[source,python]
----
def my_function(a,b):
    ...
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::parameters.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
