== How to fix it in OpenSSL

=== Code examples

include::../../common/fix/code-rationale.adoc[]

==== Noncompliant code example

[source,php,diff-id=1,diff-type=noncompliant]
----
openssl_encrypt($plaintext, "des-ofb", $key, $options=OPENSSL_RAW_DATA, $iv); // Noncompliant
----

==== Compliant solution

Mcrypt is deprecated and should not be used. You can use
https://www.php.net/manual/en/book.sodium.php[Sodium] instead.

[source,php,diff-id=1,diff-type=compliant]
----
openssl_encrypt($plaintext, "aes-256-gcm", $key, $options=OPENSSL_RAW_DATA, $iv);
----

=== How does this work?

include::../../common/fix/strong-cryptography.adoc[]

