<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Dispatchers should not be hardcoded when using <code>withContext</code> or creating new coroutines using <code>launch</code> or <code>async</code>. Injectable dispatchers ease testing by allowing tests to inject more deterministic dispatchers.</p>
</div>
<div class="paragraph">
<p>You can use default values for the dispatcher constructor arguments to eliminate the need to specify them explicitly in the production caller contexts.</p>
</div>
<div class="paragraph">
<p>This rule raises an issue when it finds a hard-coded dispatcher being used in <code>withContext</code> or when starting new coroutines.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-kotlin" data-lang="kotlin">class ExampleClass {
    suspend fun doSomething() {
        withContext(Dispatchers.Default) { // Noncompliant: hard-coded dispatcher
            ...
        }
    }
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-kotlin" data-lang="kotlin">class ExampleClass(
    private val dispatcher: CoroutineDispatcher = Dispatchers.Default
) {
    suspend fun doSomething() {
        withContext(dispatcher) {
            ...
        }
    }
}</code></pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p><a href="https://developer.android.com/kotlin/coroutines/coroutines-best-practices#inject-dispatchers">Inject dispatchers</a> (Android coroutines best practices)</p>
</li>
</ul>
</div>
</div>
</div>