== Why is this an issue?

Shared coding conventions allow teams to collaborate efficiently. This rule checks that file-code names conform to a specified regular expression.


=== Noncompliant code example

Given a pattern of ``++FC-.*++``:


[source,cobol]
----
 INPUT-OUTPUT SECTION.
 FILE-CONTROL.

     SELECT Y27MVTS       ASSIGN  TO     S1        >Noncompliant; S1 doesn't match "FC-.*" pattern
                          FILE STATUS IS S1.
----


=== Compliant solution

[source,cobol]
----
 INPUT-OUTPUT SECTION.
 FILE-CONTROL.

     SELECT Y27MVTS     ASSIGN  TO     FC-S1     >OK as FC-S1 matches "FC-.*" pattern
                          FILE STATUS IS FS-S1.
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Parameters

.regexPatternString
****

A regular expression to specify the File-Code naming convention
****


endif::env-github,rspecator-view[]
