<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p><code>CREATE_TIMER</code> is generating network traffic each time the timer is fired. It&#8217;s probably totally fine for a timer being executed every hour but generally, this is used to provide clocks components that are going to generate network traffic every second or more.</p>
</div>
<div class="paragraph">
<p>It is recommended by Oracle to examine timers and replace them with JavaBeans.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-sql" data-lang="sql">BEGIN
  timer := CREATE_TIMER('foo', 1000, REPEAR)
ENDl</code></pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p><a href="https://docs.oracle.com/cd/A97335_02/apps.102/a86202/chap11.htm">11.3.4 Other Techniques to Improve Performance</a></p>
</li>
</ul>
</div>
<hr>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Replace "CREATE_TIMER" by a JavaBean timer executed on Oracle Forms client side.</p>
</div>
</div>
</div>
</div>