This rule raises an issue when a CRON expression used with `System.schedule` is improperly formatted or has a non-zero seconds field.

== Why is this an issue?

Salesforce has specific requirements for CRON expressions that differ from standard CRON syntax. The platform requires exactly 6 fields in the format: seconds, minutes, hours, day of month, month, and day of week.

The seconds field must always be set to '0' because Salesforce does not support scheduling jobs at the second level. This is a platform limitation that prevents more granular scheduling.

When CRON expressions don't follow these rules, several problems occur:

* Jobs may fail to schedule without clear error messages
* Silent failures can leave critical batch processes unscheduled
* Debugging scheduling issues becomes difficult when the root cause is malformed syntax
* Standard CRON expressions from other systems won't work directly in Salesforce

Proper CRON expression formatting ensures reliable job scheduling and prevents unexpected failures in automated processes.

=== What is the potential impact?

Improperly formatted CRON expressions can cause batch jobs and scheduled processes to fail silently, leading to missed data processing, delayed business operations, and difficult-to-diagnose scheduling issues.

== How to fix it

Use the correct 6-field CRON format with seconds set to '0'. Ensure all required fields are present: seconds (0), minutes, hours, day of month, month, and day of week.

=== Code examples

==== Noncompliant code example

[source,apex,diff-id=1,diff-type=noncompliant]
----
String cronExp = '30 0 1 1 * ?'; // Noncompliant: seconds should be 0
System.schedule('Monthly Update', cronExp, scheduler);
----

==== Compliant solution

[source,apex,diff-id=1,diff-type=compliant]
----
String cronExp = '0 0 1 1 * ?'; // Correct: seconds set to 0
System.schedule('Monthly Update', cronExp, scheduler);
----

== Resources

=== Documentation

 * Salesforce Apex Scheduler - https://developer.salesforce.com/docs/atlas.en-us.apexcode.meta/apexcode/apex_scheduler.htm[Official Salesforce documentation on scheduling Apex jobs with CRON expressions]

 * System.schedule Method - https://developer.salesforce.com/docs/atlas.en-us.apexref.meta/apexref/apex_methods_system_system.htm#apex_System_System_schedule[Reference documentation for the System.schedule method and CRON expression syntax]
