== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,go]
----
func (user *User) rename(name string) {
  name = name  // Noncompliant
}
----

=== Compliant solution

[source,go]
----
func (user *User) rename(name string) {
  user.name = name
}
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
