include::../description.adoc[]

== Noncompliant Code Example

----
func (user *User) rename(name string) {
  name = name  // Noncompliant
}
----

== Compliant Solution

----
func (user *User) rename(name string) {
  user.name = name
}
----

include::../see.adoc[]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
