This rule raises an issue when a computation is done on Python arrays instead of using a NumPy broadcast operation.

== Why is this an issue?

Computations over NumPy arrays can be executed more efficiently than computation over Python's built-in sequences. 
These performances are possible thanks to: the nature of the NumPy arrays (fixed size and homogeneous) 
and operations on these arrays that can be done with compiled code. 

When executing an operation on each elements of an array it is preferable to use NumPy broadcasting capabilities in order to benefit from the performance boost.
NumPy broadcasting operations provide also better readability and maintainability.

== How to fix it

To fix this issue try using NumPy's broadcasting approach instead of iterating over each elements of the array.

=== Code examples

==== Noncompliant code example

[source,python,diff-id=1,diff-type=noncompliant]
----
a = [1,2,3,4,5]
b = [9,8,7,6,5]
c = []
for i in range(len(a)):
    c.append(a[i]+b[i]) # Noncompliant: this computation can be done more efficiently with a broadcasting operation.
----

==== Compliant solution

[source,python,diff-id=1,diff-type=compliant]
----
import numpy as np 

a = np.array([1,2,3,4,5])
b = np.array([9,8,7,6,5])
c = a + b # Compliant
----

//=== How does this work?

//=== Pitfalls

//=== Going the extra mile


== Resources

=== Documentation

NumPy Documentation - https://numpy.org/doc/stable/user/whatisnumpy.html#what-is-numpy[What is NumPy]


//=== Articles & blog posts
//=== Conference presentations
//=== Standards
//=== External coding guidelines
//=== Benchmarks
