=== How to fix it in Node.js

include::../../common/fix/code-rationale.adoc[]

==== Noncompliant code example

[source,javascript,diff-id=1,diff-type=noncompliant]
----
const { execSync } = require('child_process');

cmd = req.query.cmd
execSync(cmd) // Noncompliant
----

==== Compliant solution

[source,javascript,diff-id=1,diff-type=compliant]
----
const { spawnSync } = require('child_process');

cmdId = parseInt(req.query.cmdId)
host = req.query.host
host = typeof host === "string"? host : "example.org"

DIAG_CMD = [
    {exe:"/bin/ping", args:["-c","1","--"]},
    {exe:"/bin/host", args:["--"]}
]
cmd = DIAG_CMD[cmdId]
spawnSync(cmd.exe, cmd.args.concat(host))
----

=== How does this work?

include::../../common/fix/introduction.adoc[]

include::../../common/fix/pre-approved-list.adoc[]

Here, a static list of trusted commands is used. Users are only allowed to
submit an index in this array in place of a full command name.

:sanitizationLib: child_process.spawn
include::../../common/fix/sanitize-meta-characters.adoc[]

Here, the `spawn` function from `child_process` is used in place of its less
secure `exec` counterpart. It accepts command arguments as an array and
performs a proper escaping of its element before building the command line to
run.

**Important note**: Because JavaScript is a loosely typed language, extra care
should be taken when accepting user-controlled parameters. Here, the `concat`
method, used to add an argument to the command to run, accepts both single
objects and object arrays as input. Ensuring that the `host` parameter is a
`string` is mandatory to prevent multiple command line arguments from being
added.

include::../../common/fix/shell_integration.adoc[]

The `spawn` function that is used here disables shell integration by default.

