There is no reason to concatenate literal strings. Doing so is an exercise is reducing code readability. Instead, the strings should be combined. Similarly, literal strings should not be ``++append++``ed to a ``++StringBuffer++`` or ``++StringBuilder++`` sequentially, but combined into one call.

== Noncompliant Code Example

----
String message = "Hello " + "world" + "!";  // Noncompliant
StringBuilder sb = new StringBuilder();
sb.append("I'm pleased").append(" to meet you.");  //Noncompliant
----

== Compliant Solution

----
String message = "Hello world!";
StringBuilder sb = new StringBuilder();
sb.append("I'm pleased to meet you.");  
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
