``++GOTO++`` is an unstructured control flow statement. It makes code less readable and maintainable, and should only be used to branch to defined ``++ENDSR++`` points within the source. For normal loop functions use ``++ITER++`` or ``++LEAVE++`` instead.

== Noncompliant Code Example

----
C     LOOP1         TAG 
... 
C                   GOTO      LOOP1 
----

ifdef::env-github,rspecator-view[]
'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
