== How to fix it in Laravel

include::how-to-fix-it.adoc[]

=== Code Examples

==== Noncompliant Code Example

[source,php,diff-id=2,diff-type=noncompliant]
----
response()->header('Access-Control-Allow-Origin', "*"); // Noncompliant
----

==== Compliant Solution

[source,php,diff-id=2,diff-type=compliant]
----
response()->header('Access-Control-Allow-Origin', $trusteddomain);
----
