== How to fix it in Core PHP

include::how-to-fix-it.adoc[]

=== Code Examples

==== Noncompliant Code Example

[source,php,diff-id=1,diff-type=noncompliant]
----
header("Access-Control-Allow-Origin: *"); // Noncompliant
----

==== Compliant Solution

[source,php,diff-id=1,diff-type=compliant]
----
header("Access-Control-Allow-Origin: $trusteddomain");
----
