== Why is this an issue?

When exceptions occur, it is usually a bad idea to simply ignore them. Instead, it is better to handle them properly, or at least to log them.


This rule only reports on empty catch clauses that catch generic ``++Exception++``s.

=== Noncompliant code example

[source,csharp]
----
string text = "";
try 
{
    text = File.ReadAllText(fileName);
} 
catch (Exception exc) // Noncompliant
{  
}
----

=== Compliant solution

[source,csharp]
----
string text = "";
try 
{
    text = File.ReadAllText(fileName);
} 
catch (Exception exc) 
{
    logger.Log(exc);
}
----

=== Exceptions

When a block contains a comment, it is not considered to be empty.

include::../see.adoc[]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Handle the exception or explain in a comment why it can be ignored.


'''
== Comments And Links
(visible only on this page)

=== on 23 Mar 2015, 10:07:39 Tamas Vajk wrote:
\[~ann.campbell.2], Could you please check this subtask?

=== on 23 Mar 2015, 11:31:17 Ann Campbell wrote:
\[~tamas.vajk], there's no need to repeat the description in the subtask when it's the same (so I've removed it). Otherwise, this looks great. :-)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
