<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>With Java 8&#8217;s "default method" feature, any abstract class without direct or inherited field should be converted into an interface. However, this change may not be appropriate in libraries or other applications where the class is intended to be used as an API.</p>
</div>
<div class="paragraph">
<p><strong>Note</strong> that this rule is automatically disabled when the project&#8217;s <code>sonar.java.source</code> is lower than <code>8</code>.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">public abstract class Car {
  public abstract void start(Environment c);

  public void stop(Environment c) {
    c.freeze(this);
  }
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">public interface Car {
  public void start(Environment c);

  public default void stop(Environment c) {
    c.freeze(this);
  }
}</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Convert the abstract class "XXXX" into an interface.  [(sonar.java.source not set. Assuming 8 or greater.)]</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_25_feb_2014_234534_freddy_mallet_wrote">on 25 Feb 2014, 23:45:34 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>Is implemented by http://jira.codehaus.org/browse/SONARJAVA-463</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_23_oct_2015_133258_michael_gumowski_wrote">on 23 Oct 2015, 13:32:58 Michael Gumowski wrote:</h3>
<div class="paragraph">
<p>Activated by default in java plugin 3.7, as the check will now only perform the analysis if java8 is detected for the sources.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_14_dec_2015_115341_florian_schneider_wrote">on 14 Dec 2015, 11:53:41 Florian Schneider wrote:</h3>
<div class="paragraph">
<p>Should this rule be modified to accept abstract classes with package private visibility that only contain methods with package private visibility?</p>
</div>
<div class="paragraph">
<p>Refactoring the abstract class as required by the rule to an interface would make the method public API, which might be undesired in some cases.</p>
</div>
</div>
</div>
</div>