<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>The <code>delete</code> operator expects a pointer argument. Passing an object to it may compile and seem to run (with an implicit cast to pointer type), but it can result in unexpected behavior at runtime.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">class CString {
public:
  operator const char*();
  // ...
};

void fun() {
  CString str;
  // ...
  delete str;  // Noncompliant
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">class CString {
public:
  operator const char*();
  // ...
};

void fun() {
  CString *pstr = new CString;
  // ...
  delete pstr;
}</code></pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p>C&#43;&#43; reference - <a href="https://en.cppreference.com/w/cpp/memory/new/operator_delete">operator delete</a></p>
</li>
</ul>
</div>
<hr>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>"xxx" is not a pointer.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_8_apr_2015_165338_evgeny_mandrikov_wrote">on 8 Apr 2015, 16:53:38 Evgeny Mandrikov wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] I feel that word "operator" should be added to title, so what do you think about:</p>
</div>
<div class="paragraph">
<p>Operand of "delete" operator should have pointer type</p>
</div>
<div class="paragraph">
<p>Operator "delete" should be used only for pointers</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_8_apr_2015_184739_evgeny_mandrikov_wrote">on 8 Apr 2015, 18:47:39 Evgeny Mandrikov wrote:</h3>
<div class="paragraph">
<p>LGTM.</p>
</div>
</div>
</div>
</div>