Frames allow different web pages to be put together on the same visual space. Users without disabilities can easily scan the contents of all frames at once. However, visually impaired users using screen readers hear the page content linearly.


The ``++title++`` attribute is used to list all the page's frames, enabling those users to easily navigate among them. Therefore, the ``++<frame>++`` and ``++<iframe>++`` tags should always have a ``++title++`` attribute.

== Noncompliant Code Example

----
<frame src="index.php?p=menu">                                      <-- Non-Compliant -->
<frame src="index.php?p=home" name="contents">                      <-- Non-Compliant -->
----

== Compliant Solution

----
<frame src="index.php?p=menu" title="Navigation menu">              <-- Compliant -->
<frame src="index.php?p=home" title="Main content" name="contents"> <-- Compliant -->
----
