HTML <table> elements should not be used for layout purpose as it can confuse screen readers. It is recommended to use CSS instead.


This rule raises an issue on every ``++<table>++`` element containing a ``++role++``Â attribute set to ``++"presentation"++`` or ``++"none"++``, which is how https://www.w3.org/WAI/tutorials/tables/tips/[W3C recommends] marks layout tables.


== Noncompliant Code Example

----
<table role="presentation">
<!-- ... -->
</table>
----


== See

* https://www.w3schools.com/html/html_layout.asp[HTML Layouts]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

include::highlighting.adoc[]

endif::env-github,rspecator-view[]
