== How to fix it in Java Cryptographic Extension

=== Code examples

==== Noncompliant code example


[source,kotlin,diff-id=1,diff-type=noncompliant]
----
import javax.net.ssl.SSLContext;
import java.security.NoSuchAlgorithmException;

SSLContext.getInstance("TLSv1.1"); // Noncompliant
----

==== Compliant solution

[source,kotlin,diff-id=1,diff-type=compliant]
----
import javax.net.ssl.SSLContext;
import java.security.NoSuchAlgorithmException;

SSLContext.getInstance("TLSv1.2");
----

=== How does this work?

include::../../common/fix/fix.adoc[]
