include::../rationale.adoc[]

=== Special Cases

Variables passed by reference to functions whose name contains `by_ref` are considered used. In bash, some functions accept variable names as plain word arguments (not `$var` expansions) and modify them indirectly. When such a function name contains `by_ref`, word arguments matching declared variable names are counted as legitimate usages.

[source,bash]
----
function process_data() {
  local result status  # Compliant: result and status are used by reference
  get_values_by_ref result status
}
----

include::../how-to-fix-it.adoc[]

=== Code examples

==== Noncompliant code example

[source,bash,diff-id=1,diff-type=noncompliant]
----
function numberOfMinutes(hours) {
  local seconds=0   # Noncompliant: seconds is unused
  echo $(( $hours * 60 ))
}
----

==== Compliant solution

[source,bash,diff-id=1,diff-type=compliant]
----
function numberOfMinutes(hours) {
  echo $(( $hours * 60 ))
}
----
