include::../rationale.adoc[]

include::../how-to-fix-it.adoc[]

=== Code examples

==== Noncompliant code example

[source,bash,diff-id=1,diff-type=noncompliant]
----
function numberOfMinutes(hours) {
  local seconds=0   # Noncompliant: seconds is unused
  echo $(( $hours * 60 ))
}
----

==== Compliant solution

[source,bash,diff-id=1,diff-type=compliant]
----
function numberOfMinutes(hours) {
  echo $(( $hours * 60 ))
}
----
