== Why is this an issue?

Apart from a few operators (notably ``++&&++``, ``++||++``, ``++?:++`` and ``++,++``) the order in which sub-expressions are evaluated is unspecified and can vary. This means that no reliance can be placed on the order of evaluation of sub-expressions and, in particular, no reliance can be placed on the order in which side effects occur. Those points in the evaluation of an expression at which all previous side effects can be guaranteed to have taken place are called “sequence points”. Sequence points and side effects are described in Section 1.9(7) of ISO/IEC 14882:2003 [1].


Note that the “order of evaluation” problem is not solved by the use of parentheses, as this is not a precedence issue.


=== Noncompliant code example

[source,cpp]
----

----


== Resources

* MISRA C:2004, 12.2 - The value of an expression shall be the same under any order of evaluation that the standard permits.
* MISRA {cpp}:2008, 5-0-1 - The value of an expression shall be the same under any order of evaluation that the standard permits.
* https://www.securecoding.cert.org/confluence/x/WQD3[CERT, EXP10-C.] - Do not depend on the order of evaluation of subexpressions or the order in which side effects take place


ifdef::env-github,rspecator-view[]
'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
