The use of C headers and therefore C functions in a {cpp} program, is sometimes necessary, but should be avoided in favor of {cpp} headers and functions.


== Noncompliant Code Example

[source,cpp]
----
#include <string.h>
----


== Compliant Solution

[source,cpp]
----
#include <cstring>
#include <string>
----


ifdef::env-github,rspecator-view[]
'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
