``End`` statements exit the control flow of the program in an unstructured way. This statement stops code execution immediately without executing ``Dispose`` or ``Finalize`` methods, or executing ``Finally`` blocks. Therefore, it should be avoided.

== Noncompliant Code Example

----
Module Module1
    Sub Print(ByVal str As String)
       Try
            ...
            End       ' Noncompliant
        Finally
            ' do something important here
            ...
        End Try
    End Sub
End Module
----
