<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p><code>xs:any</code> and <code>xs:anyAttribute</code> are wildcards; they allow the XML document author to use any element or attribute. By specifying <code>nameSpace</code>s for them, you can restrict what&#8217;s used to elements in the specified name spaces. By default, what&#8217;s used for <code>xs:any</code> and <code>xs:anyAttribute</code> will be validated in a strict manner (<code>processContents="strict"</code>). That is, validation will fail if the element used isn&#8217;t recognized. That makes sense if you&#8217;ve provided a namespace, but not if you haven&#8217;t. Conversely, <code>processContents="lax"</code> allows unrecognized elements to pass validation even if they&#8217;re not recognized. That makes sense if no namespace has been specified, but not if one has been specified.</p>
</div>
<div class="paragraph">
<p>This rule raises an issue when <code>strict</code> processing is used (either explicitly or implicitly) when no namespace is provided, and conversely when <code>lax</code> processing is used with a namespace.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-xml" data-lang="xml">ToDo</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-xml" data-lang="xml">ToDo</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="ulist">
<ul>
<li>
<p>Set the processing of this [attribute|element] to "lax" or add a "nameSpace".</p>
</li>
<li>
<p>Set the processing of this [attribute|element] to "strict" or drop the "nameSpace".</p>
</li>
</ul>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p><code>processContents="xxx"</code> if attribute is present, otherwise <code>namespace</code> attribute</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_18_jan_2017_131721_ann_campbell_wrote">on 18 Jan 2017, 13:17:21 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>see what you think [~yves.duboispelerin]</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_18_jan_2017_135723_yves_dubois_pèlerin_wrote">on 18 Jan 2017, 13:57:23 Yves Dubois-Pèlerin wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] Seems OK!</p>
</div>
</div>
</div>
</div>