<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Overloaded versions of the comma and logical conjunction operators have the semantics of function calls whose sequence point and ordering semantics are different from those of the built-in versions. It may not be clear at the point of use that these operators are overloaded, and so developers may be unaware which semantics apply.</p>
</div>
<div class="paragraph">
<p>Exception: Starting from <em>C&#43;&#43;17</em>, the order of evaluation of the comma operator is defined and identical for the builtin and the overloaded versions. In such circumstances, the comma operator can safely be overloaded.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">#include "util.h"
class A
{
public:
  UtilType getValue ( );
  UtilType setValue ( UtilType const &amp; );
};
void f1 ( A &amp; a1, A &amp; a2 )
{
  a1.getValue ( ) &amp;&amp; a2.setValue ( 0 );	// Short circuiting may occur
}
bool operator &amp;&amp; ( UtilType const &amp;, UtilType const &amp; ); // Noncompliant
void f2 ( A &amp; a1, A &amp; a2 )
{
  a1.getValue ( ) &amp;&amp; a2.setValue ( 0 ); // Both operands evaluated if type returned has overloaded operator&amp;&amp;
}</code></pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p>MISRA C&#43;&#43; 2008, 5-2-11 - The comma operator, &amp;&amp; operator and the || operator shall not be overloaded.</p>
</li>
</ul>
</div>
</div>
</div>