== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,vbnet]
----
<Serializable>
Public Class Foo ' Noncompliant
    <OptionalField(VersionAdded:=2)>
    Private optionalField As Integer = 5
End Class
----

=== Compliant solution

[source,vbnet]
----
<Serializable>
Public Class Foo
    <OptionalField(VersionAdded:=2)>
    Private optionalField As Integer = 5

    <OnDeserializing>
    Private Sub OnDeserializing(ByVal context As StreamingContext)
        optionalField = 5
    End Sub

    <OnDeserialized>
    Private Sub OnDeserialized(ByVal context As StreamingContext)
    End Sub
End Class
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../highlighting.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
