Python interprets ``++False++`` as a "false" boolean value, and ``++True++`` as a "true" boolean value. Assigning them other values could lead to unexpected behaviors.


This rule raises an issue when "True" and "False", with any case, are used as identifiers.


== Noncompliant Code Example

----
True = "red"  # Noncompliant
----


== Compliant Solution

----
color = "red"  # Noncompliant
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
