<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>In order to extend the functionalities of your Gradle build you can add plugins with <code>id("plugin-id")</code>. Gradle maintains some plugins that are known as <a href="https://docs.gradle.org/current/userguide/plugin_reference.html">core plugins</a>. These core plugins can be identified by a short name.</p>
</div>
<div class="paragraph">
<p>When adding core plugins, it is a good practice to use their short name because it is more concise, readable, and less prone to typing mistakes.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Replace <code>id("core-plugin-id")</code> with the core plugin short name.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-kotlin" data-lang="kotlin">plugins {
    id("org.gradle.java")
    id("org.gradle.jacoco")
    id("org.gradle.maven-publish")
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-kotlin" data-lang="kotlin">plugins {
    java
    jacoco
    `maven-publish`
}</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p><a href="https://docs.gradle.org/current/userguide/plugin_reference.html">Gradle core plugins reference</a></p>
</li>
<li>
<p><a href="https://docs.gradle.org/current/userguide/plugins.html">Gradle plugins</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>