<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Weak <code>REF CURSOR</code> types are harder to work with than ones with a return type. Indeed, the compiler&#8217;s type-checker is unable to make some verifications, which are then delayed till runtime.</p>
</div>
<div class="paragraph">
<p>When the use of weak <code>REF CURSOR</code> is required, it is best to use the <code>SYS_REFCURSOR</code> built-in type instead of defining a new one.</p>
</div>
<div class="paragraph">
<p>This rule&#8217;s sysRefCursorAllowed parameter can be used to control whether or not the usage of <code>SYS_REFCURSOR</code> is allowed.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-sql" data-lang="sql">DECLARE
  TYPE dualCursorType IS REF CURSOR;                      -- Noncompliant
  dualCursor dualCursorType;

  otherCursor SYS_REFCURSOR;                              -- Compliant or non-compliant, depending on the "sysRefCursorAllowed" parameter
BEGIN
  otherCursor := dualCursor;                              -- Works
END;
/</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-sql" data-lang="sql">DECLARE
  TYPE dualCursorType IS REF CURSOR RETURN DUAL%ROWTYPE;
  dualCursor dualCursorType;

  TYPE otherCursorType IS REF CURSOR RETURN a%ROWTYPE;
  otherCursor otherCursorType;
BEGIN
  otherCursor := dualCursor;                              -- raises PLS-00382: expression is of wrong type, which makes debugging easier
END;
/</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Add a return type to this "REF CURSOR".</p>
</div>
</div>
<div class="sect2">
<h3 id="_parameters">Parameters</h3>
<div class="sidebarblock">
<div class="content">
<div class="title">sysRefCursorAllowed</div>
<div class="listingblock">
<div class="content">
<pre>false</pre>
</div>
</div>
<div class="paragraph">
<p>Whether or not the use of "SYS_REFCURSOR" is allowed.</p>
</div>
</div>
</div>
</div>
</div>
</div>