include::../description.adoc[]

== Noncompliant Code Example

----
void setValue(int value) {
  value = value;
}
----

== Compliant Solution

----
void setValue(int value) {
  this->value = value;
}
----

== See

* https://wiki.sei.cmu.edu/confluence/x/5dUxBQ[CERT, MSC12-C.] - Detect and remove code that has no effect or is never executed

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
