include::../description.adoc[]

== Noncompliant Code Example

----
void setValue(int value) {
  value = value;
}
----

== Compliant Solution

----
void setValue(int value) {
  this->value = value;
}
----

include::../see.adoc[]

ifdef::env-github,rspecator-view[]
'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
