== Why is this an issue?

In Gradle, using the `GradleBuild` task type you can invoke another Gradle build and execute some tasks from another project. See the example below:

[source,kotlin]
----
tasks.register<GradleBuild>("execModuleTask") {
    dir = File("module/build.gradle.kts")
    tasks = listOf("task")
}
----

In general, it is considered to be a bad practice because there are corner cases where the invoked build behaves differently compared to executing it from the command line or through the Tooling API. This can lead to unexpected results.

== How to fix it

Change the way to model the requirement. The appropriate approach depends on the problem at hand. Hereâ€™re some options:

* Model the build as https://docs.gradle.org/current/userguide/multi_project_builds.html#multi_project_builds[multi-project build] if the intention is to execute tasks from different modules as unified build.
* Use https://docs.gradle.org/current/userguide/composite_builds.html#composite_builds[composite builds] for projects that are physically separated but should occasionally be built as a single unit.

=== Code examples

==== Noncompliant code example

[source,kotlin]
----
tasks.register<GradleBuild>("execModuleTask") {
    dir = File("module/build.gradle.kts")
    tasks = listOf("task")
}
----

In order to execute it you need to run it:

----
    ./gradlew execModuleTask
----

==== Compliant solution

To fix it you can make `module` your submodule and run it with:

----
    ./gradlew :module:task
----


== Resources

=== Documentation

* https://docs.gradle.org/current/userguide/authoring_maintainable_build_scripts.html#sec:avoiding_use_of_gradlebuild[Avoiding use of GradleBuild]

//=== Articles & blog posts
//=== Conference presentations
//=== Standards
//=== Benchmarks
