== Why is this an issue?

include::../description.adoc[]

=== NonCompliant code example

[source,kotlin]
----
@Deprecated("") // Noncompliant
class Example
----
ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

endif::env-github,rspecator-view[]
