include::../description.adoc[]

== Noncompliant Code Example

----
@Deprecated("") // Noncompliant
class Example
----
ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

endif::env-github,rspecator-view[]
