== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,java]
----
# TODO
AWSTemplateFormatVersion: 2010-09-09
----

include::../see.adoc[]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
