<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when code attempts to directly compare a <code>Date</code> object with an <code>ActiveSupport::TimeWithZone</code> object using comparison operators (<code>&lt;</code>, <code>&gt;</code>, <code>&#8656;</code>, <code>&gt;=</code>, <code>==</code>, <code>!=</code>).</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Direct comparison between <code>Date</code> and <code>ActiveSupport::TimeWithZone</code> objects will fail at runtime with an <code>ArgumentError</code>. This happens because Ruby cannot automatically convert between these different time representations.</p>
</div>
<div class="paragraph">
<p><code>Date</code> objects represent calendar dates without time information, while <code>TimeWithZone</code> objects include both date and time with timezone information. When Ruby encounters a comparison operator between these incompatible types, it cannot determine how to perform the comparison and raises an error.</p>
</div>
<div class="paragraph">
<p>This issue is particularly common in Rails applications where:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>Model attributes are often stored as datetime fields (which become <code>TimeWithZone</code> objects)</p>
</li>
<li>
<p>Business logic frequently works with <code>Date.today</code> or other <code>Date</code> objects</p>
</li>
<li>
<p>Calculations involving date arithmetic (like <code>+ age.years</code>) return <code>TimeWithZone</code> objects</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>The error only appears at runtime, making it harder to catch during development and potentially causing production failures.</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="paragraph">
<p>This issue causes immediate runtime failures with <code>ArgumentError: comparison of Date with ActiveSupport::TimeWithZone failed</code>. These errors can break critical application functionality, especially in date-sensitive operations like age calculations, deadline checks, or scheduling logic.</p>
</div>
<div class="paragraph">
<p>Since the error only occurs when the specific code path is executed, it may not be caught during testing if test coverage is incomplete, leading to production outages.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it_in_rails">How to fix it in Rails</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Convert the TimeWithZone object to a Date using <code>.to_date</code> when you only need date-level precision for the comparison.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-ruby" data-lang="ruby">def age(date = nil)
  date = Date.today if date.nil?
  age = 0
  unless date_of_birth.nil?
    age = date.year - date_of_birth.year
    age -= 1 if date &lt; date_of_birth + age.years # Noncompliant
  end
  age
end</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-ruby" data-lang="ruby">def age(date = nil)
  date = Date.today if date.nil?
  age = 0
  unless date_of_birth.nil?
    age = date.year - date_of_birth.year
    age -= 1 if date &lt; (date_of_birth + age.years).to_date
  end
  age
end</code></pre>
</div>
</div>
<div class="paragraph">
<p>Convert the Date object to a Time using <code>.to_time</code> when you need full datetime precision for the comparison.</p>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_code_examples_2">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example_2">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-ruby" data-lang="ruby">def deadline_passed?(deadline_date)
  current_time = Time.current
  deadline_date &gt; current_time # Noncompliant
end</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution_2">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-ruby" data-lang="ruby">def deadline_passed?(deadline_date)
  current_time = Time.current
  deadline_date.to_time &gt; current_time
end</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Ruby Date Class Documentation - <a href="https://ruby-doc.org/stdlib-3.0.0/libdoc/date/rdoc/Date.html">Official Ruby documentation for the Date class and its methods</a></p>
</li>
<li>
<p>Rails TimeWithZone Documentation - <a href="https://api.rubyonrails.org/classes/ActiveSupport/TimeWithZone.html">Rails documentation for ActiveSupport::TimeWithZone class</a></p>
</li>
<li>
<p>Rails Time and Date Handling Guide - <a href="https://guides.rubyonrails.org/active_support_core_extensions.html#time">Rails guide covering time and date extensions and best practices</a></p>
</li>
</ul>
</div>
</div>
<div class="sect2">
<h3 id="_related_rules">Related rules</h3>
<div class="ulist">
<ul>
<li>
<p><a data-rspec-id="S6886" class="rspec-auto-link">S6886</a> - <a href="https://rules.sonarsource.com/python/RSPEC-6886/">Similar rule for Python datetime comparisons</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>