== Why is this an issue?

Using several "--" or "{plus}{plus}" unary operators in the same arithmetic expression can quickly make the expression unreadable. 


=== Noncompliant code example

[source,flex]
----
var j:int = foo++ - --bar;
----


=== Compliant solution

[source,flex]
----
bar--;
var j:int = foo++ - bar;
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
