include::description.adoc[]

== Noncompliant Code Example

[source,python]
----
import re

def foo():
    pattern = re.compile(r"(?P<a>.)")
    matches = pattern.match("abc")
    g = matches.group("b") # Noncompliant - group "b" is not defined
    g = matches.group(2) # Noncompliant - there is no group with this number
----


== Compliant Solution

[source,python]
----
import re

def foo():
    pattern = re.compile(r"(?P<a>.)")
    matches = pattern.match("abc")
    g = matches.group("a")
    g = matches.group(1)
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

include::highlighting.adoc[]

endif::env-github,rspecator-view[]
