<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p><code>@SpringBootTest</code> is an annotation provided by the Spring Boot testing framework that is often used for integration testing.
It instructs the testing framework to load and configure the entire Spring application context, including all beans and components defined in your application.</p>
</div>
<div class="paragraph">
<p>Unit tests in Spring applications that use <code>@SpringBootTest</code> should only load the necessary beans to avoid unnecessary resource consumption and slow test execution.
Loading the entire application context can be harmful to test performance.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>When writing unit tests that use <code>@SpringBootTest</code>, specify only the beans that are required for the particular test using the classes attribute.
Avoid loading the entire application context unless it is necessary for the specific test case.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">@SpringBootTest()  // Noncompliant
class MyServiceTest {
  // ...
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">@SpringBootTest(classes = {MyServiceTest.TestMyConfig.class, MyService.class, RestConfig.class, ObjectMapper.class})
class MyServiceTest {
  // ...
}</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Spring Boot documentation - <a href="https://docs.spring.io/spring-boot/docs/current/api/org/springframework/boot/test/context/SpringBootTest.html">@SpringBootTest</a></p>
</li>
<li>
<p>Spring - <a href="https://spring.io/guides/gs/testing-web/">Testing the Web Layer</a></p>
</li>
</ul>
</div>
</div>
<div class="sect2">
<h3 id="_articles_blog_posts">Articles &amp; blog posts</h3>
<div class="ulist">
<ul>
<li>
<p>Baeldung - <a href="https://www.baeldung.com/spring-boot-testing">Testing in Spring Boot</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>