<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Implementing the special method <code>__ne__</code> is not equivalent to implementing the special method <code>__eq__</code>. By default <code>__ne__</code> will call <code>__eq__</code>, but the default implementation of <code>__eq__</code> does not call <code>__ne__</code>.</p>
</div>
<div class="paragraph">
<p>This rule raises an issue when the special method <code>__ne__</code> is implemented but not the <code>__eq__</code> method.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">class Ne:
    def __ne__(self, other):   # Noncompliant.
        return False

myvar = Ne() == 1  # False. __ne__ is not called
myvar = 1 == Ne()  # False. __ne__ is not called
myvar = Ne() != 1  # False
myvar = 1 != Ne()  # False</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python"># Recommended, as __ne__ delegates to __eq__ by default
class Eq:
    def __eq__(self, other):
        return True

myvar = Eq() == 1  # True
myvar = 1 == Eq()  # True
myvar = Eq() != 1  # False. __eq__ is called
myvar = 1 != Eq()  # False. __eq__ is called

# OR if __ne__ needs a special implementation

class Eq:
    def __eq__(self, other):
        return True

    def __ne__(self, other):
        return False</code></pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p>Python documentation - <a href="https://docs.python.org/3/reference/datamodel.html#object.__ne__">the __ne__ special method</a></p>
</li>
</ul>
</div>
<hr>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>The "<em>ne</em>" method name</p>
</div>
</div>
</div>
</div>