<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>An "EXISTS" statement is generally used to select/update/delete some rows of a table based on the content of columns of other tables.</p>
</div>
<div class="paragraph">
<p>If the "SELECT" statement used as argument of the "EXISTS" statement is always returning "true" for all rows of the main <code>SELECT</code> statement, the <code>EXISTS</code> statement is useless and has the same effect as if it was not there. Still, this is probably not the original intend of the developer to have an <code>EXISTS</code> statement that is always true.</p>
</div>
<div class="paragraph">
<p>As a consequence, the <code>SELECT</code> statement of an <code>EXISTS</code> statement should always contain a <code>WHERE</code> clause.</p>
</div>
<div class="paragraph">
<p>What is true for <code>EXISTS</code> is also true for <code>NOT EXISTS</code>.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-text" data-lang="text">SELECT  *
FROM    sys.[databases] AS [sd]
WHERE EXISTS (SELECT  1
              FROM    [sys].[master_files] AS [mf])</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-text" data-lang="text">SELECT  *
FROM    sys.[databases] AS [sd]
WHERE EXISTS (SELECT  1
              FROM    [sys].[master_files] AS [mf]
              WHERE [mf].[database_id] = [sd].[database_id])</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_exceptions">Exceptions</h3>
<div class="paragraph">
<p>This rule doesn&#8217;t raise an issue when <code>EXISTS</code> is used in the context of a <code>WHILE</code> or a <code>IF</code> statement.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Add a WHERE clause to this SELECT statement to make it more selective.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="ulist">
<ul>
<li>
<p>First: the "SELECT" of the EXISTS/NOT EXISTS statement.</p>
</li>
<li>
<p>Second: the the EXISTS/NOT EXISTS statement
message: 'Calling "EXISTS" on non selective "SELECT" is usually a mistake.'</p>
</li>
</ul>
</div>
</div>
</div>
</div>