This rule raises an issue when `list()` calls are applied to types that are already directly iterable with for-loops or comprehensions.

== Why is this an issue?

When iterating over an already iterable object with a for loop or a comprehension, wrapping it with `list()` adds meaningless clutter that doesn't provide any functional value. Additionally, it creates unnecessary overhead by generating an intermediate list in memory, which inefficiently consumes memory and can degrade performance, especially with large data structures. Iterating directly over the original object is cleaner and more efficient.

== How to fix it

Remove the redundant `list()` call and iterate directly over the original iterable.

=== Code examples

==== Noncompliant code example

[source,python,diff-id=1,diff-type=noncompliant]
----
some_iterable = range(10)
for i in list(some_iterable): # Noncompliant: unnecessary list() call
    print(i)
----

==== Compliant solution

[source,python,diff-id=1,diff-type=compliant]
----
some_iterable = range(10)
for i in some_iterable: # Compliant
    print(i)
----

== Resources
=== Documentation

* Python Documentation - https://docs.python.org/3/glossary.html#term-iterable[Iterable Glossary Entry]
* Python Documentation - https://docs.python.domainunion.de/3/library/stdtypes.html#list[list()]

ifdef::env-github,rspecator-view[]
'''
== Implementation Specification
(visible only on this page)

This rule should raise when an iterable is wrapped in a `list()` call, and used in either a for loop or a comprehension.

=== Message

Remove this unnecessary `list()` call on an already iterable object.

=== Highlighting

The `list()` call applied to the iterable.
endif::env-github,rspecator-view[]
