== Why is this an issue?

Shared coding conventions allow teams to collaborate efficiently. This rule checks that all record field names match the provided regular rexpression.


=== Noncompliant code example

With the default regular expression ``++[a-zA-Z](_?+[a-zA-Z0-9])*+++``:

[source,sql]
----
DECLARE
  TYPE my_type IS RECORD(
    foo__bar PLS_INTEGER   -- Non-Compliant
  );
BEGIN
  NULL;
END;
/
----


=== Compliant solution

[source,sql]
----
DECLARE
  TYPE my_type IS RECORD(
    foo_bar PLS_INTEGER    -- Compliant
  );
BEGIN
  NULL;
END;
/
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

include::parameters.adoc[]

endif::env-github,rspecator-view[]
