include::../description.adoc[]

== Noncompliant Code Example

----
Class Foo        ' Non-Compliant - depends on too many other classes
  Dim a1 As T1   ' Foo is coupled to T1
  Dim a2 As T2   ' Foo is coupled to T2
  Dim a3 As T3   ' Foo is coupled to T3
  Dim a4 As T4   ' etc.
  Dim a5 As T5
  Dim a6 As T6
  Dim a7 As T7
  Dim a8 As T8
  Dim a9 As T9
  Dim a10 As T10
  Dim a11 As T11
  Dim a12 As T12
  Dim a13 As T13
  Dim a14 As T14
  Dim a15 As T15
  Dim a16 As T16
  Dim a17 As T17
  Dim a18 As T18
  Dim a19 As T19
  Dim a20 As T20
  Dim a21 As T21
End Class
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::parameters.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
