<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>The ABAP documentation is pretty clear on this subject :</p>
</div>
<div class="quoteblock">
<blockquote>
<div class="paragraph">
<p>This statement is only for</p>
</div>
<div class="paragraph">
<p>!!! Internal use in SAP Basis development !!!</p>
</div>
<div class="paragraph">
<p>Even within SAP Basis, it may only be used in programs within the ABAP+GUI development group.</p>
</div>
<div class="paragraph">
<p>Its use is subject to various restrictions, not all of which may be listed in the documentation. This documentation is intended for internal SAP use within the Basis development group ABAP+GUI.</p>
</div>
<div class="paragraph">
<p>Changes and further development, which may be incompatible, may occur at any time, without warning or notice!</p>
</div>
</blockquote>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-abap" data-lang="abap">SYSTEM-CALL CREATE CLASS c.</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove any usage of this "SYSTEM-CALL" statement.</p>
</div>
</div>
</div>
</div>