<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when a test method (annotated with <code>@isTest</code> or using the <code>testMethod</code> keyword) is declared with one or more parameters.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>The Apex testing framework has strict requirements for test method signatures. Test methods must not take any arguments to be recognized and executed by the framework.</p>
</div>
<div class="paragraph">
<p>When a test method is declared with parameters, the Apex testing framework will not recognize it as a valid test method. This means the test will be silently ignored during test execution, leading to several problems:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>The test will never run, providing false confidence that your code is working correctly</p>
</li>
<li>
<p>Code coverage metrics will be inaccurate since the test isn&#8217;t actually executing</p>
</li>
<li>
<p>Bugs in the code being tested may go undetected</p>
</li>
<li>
<p>Deployment to production may fail if actual code coverage falls below required thresholds</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>The framework expects test methods to follow a specific syntax pattern that does not include parameters. This design ensures that tests are self-contained and can be executed automatically without requiring external input.</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="paragraph">
<p>Test methods with parameters will not be executed by the Apex testing framework, leading to false confidence in code quality and potentially allowing buggy code to be deployed to production. This can result in runtime failures and reduced application reliability.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Remove all parameters from the test method signature. Test methods should be self-contained and not require external input.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-apex" data-lang="apex">@isTest
static void testMethod(String param) { // Noncompliant
    // test code
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-apex" data-lang="apex">@isTest
static void testMethod() {
    // test code
}</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Apex Testing Framework - <a href="https://developer.salesforce.com/docs/atlas.en-us.apexcode.meta/apexcode/apex_testing.htm">Official Salesforce documentation on the Apex testing framework and test method requirements</a></p>
</li>
<li>
<p>Test Method Syntax - <a href="https://developer.salesforce.com/docs/atlas.en-us.apexcode.meta/apexcode/apex_testing_utility_classes.htm">Detailed documentation on proper test method syntax and structure</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>