<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>A loop with at most one iteration is equivalent to the use of an <code>if</code> statement to conditionally execute one piece of code. No developer expects to find such a use of a loop statement. If the initial intention of the author was really to conditionally execute one piece of code, an <code>if</code> statement should be used instead.</p>
</div>
<div class="paragraph">
<p>At worst that was not the initial intention of the author and so the body of the loop should be fixed to use the nested <code>return</code>, <code>break</code> or <code>throw</code> statements in a more appropriate way.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-rust" data-lang="rust">let mut x = 0;
loop {
    x += 1;
    if x == 1 {
        return;
    }
    break;
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-rust" data-lang="rust">let mut x = 0;
x += 1;
if x == 1 {
    return;
}</code></pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Clippy Lints - <a href="https://rust-lang.github.io/rust-clippy/master/index.html#never_loop" class="bare">https://rust-lang.github.io/rust-clippy/master/index.html#never_loop</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>