<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>A loop with at most one iteration is equivalent to the use of an <code>if</code> statement to conditionally execute one piece of code. No developer expects to find such a use of a loop statement. If the initial intention of the author was really to conditionally execute one piece of code, an <code>if</code> statement should be used instead.</p>
</div>
<div class="paragraph">
<p>At worst that was not the initial intention of the author and so the body of the loop should be fixed to use the nested <code>return</code>, <code>break</code> or <code>throw</code> statements in a more appropriate way.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-swift" data-lang="swift">for i in 1...10 { // noncompliant, loop only executes once
  print("i is \(i)")
  break
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-swift" data-lang="swift">for i in 1...10 {
  print("i is \(i)")
}</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Refactor this loop to do more than one iteration.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="ulist">
<ul>
<li>
<p>primary: first line of the loop statement</p>
</li>
<li>
<p>secondary: all the nested control flow statements <code>return</code>, <code>break</code>, <code>throw</code> or <code>goto</code> leading to have at most one iteration
message: 'The loop stops here.'</p>
</li>
</ul>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_is_duplicated_by_s3554">is duplicated by: <a data-rspec-id="S3554" class="rspec-auto-link">S3554</a></h3>

</div>
<div class="sect2">
<h3 id="_relates_to_s3626">relates to: <a data-rspec-id="S3626" class="rspec-auto-link">S3626</a></h3>

</div>
<div class="sect2">
<h3 id="_is_related_to_s1763">is related to: <a data-rspec-id="S1763" class="rspec-auto-link">S1763</a></h3>

</div>
</div>
</div>