There should not be any statements after `+EXIT PROGRAM+`. Such statements cannot be reached, and are therefore dead code. Dead code makes a program more complex and therefore more difficult to maintain.


== Noncompliant Code Example

----
  PROCEDURE DIVISION.
    PARAGRAPH1.
      MOVE A TO B.
      EXIT PROGRAM.   >NOK as the following "MOVE B TO C" statement will never be called
      MOVE B TO C.
----


== Compliant Solution

----
  PROCEDURE DIVISION.
    PARAGRAPH1.
      MOVE A TO B.
      EXIT PROGRAM.
----


