<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>The information that an enumeration type is actually an enumeration or a set of flags should not be duplicated in its name.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-csharp" data-lang="csharp">enum FooFlags // Noncompliant
{
    Foo = 1
    Bar = 2
    Baz = 4
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-csharp" data-lang="csharp">enum Foo
{
    Foo = 1
    Bar = 2
    Baz = 4
}</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Rename this enumeration to remove the "xxx" suffix.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_deprecates_s2341">deprecates: <a data-rspec-id="S2341" class="rspec-auto-link">S2341</a></h3>

</div>
<div class="sect2">
<h3 id="_on_20_oct_2015_094059_tamas_vajk_wrote">on 20 Oct 2015, 09:40:59 Tamas Vajk wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] Shouldn&#8217;t we merge this RSPEC with https://jira.sonarsource.com/browse/RSPEC-2341?</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_20_oct_2015_122802_ann_campbell_wrote">on 20 Oct 2015, 12:28:02 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>done [~tamas.vajk]</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_20_oct_2015_140638_tamas_vajk_wrote">on 20 Oct 2015, 14:06:38 Tamas Vajk wrote:</h3>
<div class="paragraph">
<p>Thanks [~ann.campbell.2]</p>
</div>
</div>
</div>
</div>