This rule raises an issue when ``++np.where++`` is used with only the condition parameter set.

== Why is this an issue?

The NumPy function ``++np.where++`` provides a way to execute operations on an array under a certain condition. 

[source,python,diff-id=1,diff-type=compliant]
----
import numpy as np

arr = np.array([1,2,3,4])

result = np.where(arr > 2, arr * 2, arr)
----

In the example above the ``++np.where++`` function will multiply by 2 all the elements in the array which satisfy the condition: ``++element > 2++``. 
The elements that do not satisfy the condition will be left untouched.
The ``++result++`` array holds now the values 1, 2, 6 and 8.

It is also possible to call ``++np.where++`` with only the condition parameter set.

[source,python,diff-id=1,diff-type=compliant]
----
import numpy as np

arr = np.array([1,2,3,4])

result = np.where(arr > 2)
----

The code above is perfectly valid NumPy code but may not be the developers intention.
When providing only the condition parameter to the ``++np.where++`` function, it is a shorthand for ``++np.asarray(condition).nonzero()++``.
This means the ``++result++`` variable now holds a tuple 
with the first element being an array of all the indices where the condition ``++arr > 2++`` was satisfied: ``++(array([2,3]),)++``.

If the intention is to find the indices of the elements which satisfy a certain condition it is preferable to use the ``++np.asarray(condition).nonzero()++`` or ``++np.nonzero(condition)++`` function instead.


== How to fix it

To fix this issue provide all three parameters to the ``++np.where++`` function or use the ``++np.nonzero++`` function.

=== Code examples

==== Noncompliant code example

[source,text,diff-id=2,diff-type=noncompliant]
----
import numpy as np

arr = np.array([1,2,3,4])

result = np.where(arr > 2) # Noncompliant: only the condition parameter is provided to the np.where function.
----

==== Compliant solution

[source,text,diff-id=1,diff-type=compliant]
----
import numpy as np

arr = np.array([1,2,3,4])

result = np.where(arr > 2, arr + 1, arr) # Compliant

indices = np.nonzero(arr > 2) # Compliant
----



== Resources
=== Documentation

NumPy Documentation - https://numpy.org/doc/stable/reference/generated/numpy.where.html#numpy-where[numpy.where]
NumPy Documentation - https://numpy.org/doc/stable/reference/generated/numpy.nonzero.html#numpy.nonzero[numpy.nonzero]

