== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,vbnet]
----
If Collection.Count >= 0 Then ...

If Enumerable.Count < 0 Then ...

If array.Length >= 0 Then ...

Dim result As Boolean = Array.Length >= 0
----

=== Compliant solution

[source,vbnet]
----
If list.Count = 0 Then ...
If array.Length >= 42 Then ...
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../highlighting.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
