include::../description.adoc[]

include::../recommended.adoc[]

== Noncompliant Code Example

[source,java]
----
protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws IOException {
  Long time = Long.parseLong(req.getParameter("time"));
  Thread.sleep(time); // Noncompliant
  // ...
}
----

== Compliant Solution

[source,java]
----
protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws IOException {
  Long time = Long.parseLong(req.getParameter("time"));
  Thread.sleep(Math.min(inputTime, 1000));
  // ...
}
----


include::../see.adoc[]

ifdef::env-github,rspecator-view[]
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

endif::env-github,rspecator-view[]