== Why is this an issue?

Deprecated method should be avoided, rather than overridden. Deprecation is a warning that the method has been superseded, and will eventually be removed. The deprecation period allows you to make a smooth transition away from the aging, soon-to-be-retired technology.


=== Noncompliant code example

[source,java]
----
class A {
  @Deprecated
  void foo(){}
}

class B extends A {
  @Override
  void foo(){  // Noncompliant
  }
}
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

"xxx" is deprecated.


'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
