Change the generic type parameter of the `ILogger` interface to match the enclosing type.

=== Noncompliant code example

[source,csharp,diff-id=1,diff-type=noncompliant]
----
class EnclosingType
{
    public EnclosingType(ILogger<AnotherType> logger) // Noncompliant
    {
        // ...
    }
}
----

=== Compliant solution

[source,csharp,diff-id=1,diff-type=compliant]
----
class EnclosingType
{
    public EnclosingType(ILogger<EnclosingType> logger) // Compliant
    {
        // ...
    }
}
----