This rule raises an issue when there are spaces before or after the dot operator (`.`) in method calls.

== Why is this an issue?

Adding spaces around the dot operator in method calls reduces code readability and goes against established Ruby style conventions.

The dot operator creates a tight semantic binding between an object and its method. When spaces are added around it, this visual connection is broken, making the code harder to scan and understand quickly.

Consistent formatting helps developers read and maintain code more efficiently. Most Ruby style guides, including those from major organizations like Shopify, explicitly recommend avoiding spaces around method call operators.

While this formatting issue doesn't affect the functionality of the code, it can make codebases appear inconsistent and unprofessional, especially when different team members follow different spacing conventions.

=== What is the potential impact?

This formatting issue reduces code readability and creates inconsistency in the codebase. While it doesn't affect functionality, it can slow down code review processes and make the code appear less professional.

== How to fix it

Remove the spaces before and after the dot operator in method calls.

=== Code examples

==== Noncompliant code example

[source,ruby,diff-id=1,diff-type=noncompliant]
----
user . name
data . fetch(:key) . upcase
object . method_name
----

==== Compliant solution

[source,ruby,diff-id=1,diff-type=compliant]
----
user.name
data.fetch(:key).upcase
object.method_name
----

== Resources

=== Documentation

 * Shopify Ruby Style Guide - Layout - https://ruby-style-guide.shopify.dev/#layout[Official Shopify Ruby style guide covering layout and formatting conventions]

 * RuboCop Layout/SpaceAroundMethodCallOperator - https://docs.rubocop.org/rubocop/cops_layout.html#layoutspacearoundmethodcalloperator[RuboCop cop documentation for detecting spaces around method call operators]
