<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>There is no good excuse for an empty class. If it&#8217;s being used simply as a common extension point, it should be replaced with an <code>interface</code>. If it was stubbed in as a placeholder for future development it should be fleshed-out. In any other case, it should be eliminated.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-csharp" data-lang="csharp">public class Empty // Noncompliant
{
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-csharp" data-lang="csharp">public interface IEmpty
{
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_exceptions">Exceptions</h3>
<div class="ulist">
<ul>
<li>
<p>Partial classes are ignored entirely, as source generators often use them.</p>
</li>
<li>
<p>Classes with names ending in <code>Command</code>, <code>Message</code>, <code>Event</code>, or <code>Query</code> are ignored as messaging libraries often use them.</p>
</li>
<li>
<p>Subclasses of <code>System.Exception</code> are ignored; even an empty Exception class can provide helpful information by its type name alone.</p>
</li>
<li>
<p>Subclasses of <code>System.Attribute</code> and classes annotated with attributes are ignored.</p>
</li>
<li>
<p>Subclasses of generic classes are ignored, as they can be used for type specialization even when empty.</p>
</li>
<li>
<p>Subclasses of certain framework types — like the <code>PageModel</code> class used in ASP.NET Core Razor Pages — are ignored.</p>
</li>
<li>
<p>Subclass of a class with non-public default constructors are ignored, as they widen the constructor accessibility.</p>
</li>
</ul>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-csharp" data-lang="csharp">using Microsoft.AspNetCore.Mvc.RazorPages;

public class EmptyPageModel: PageModel // Compliant - an empty PageModel can be fully functional, the C# code can be in the cshtml file
{
}</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove this empty class, write its code or make it an "interface".</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_19_jul_2016_160658_evgeny_mandrikov_wrote">on 19 Jul 2016, 16:06:58 Evgeny Mandrikov wrote:</h3>
<div class="paragraph">
<p>Note to myself: study http://stackoverflow.com/questions/4035180/advantages-of-an-empty-class-in-c</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_18_jan_2019_172020_loïc_joly_wrote">on 18 Jan 2019, 17:20:20 Loïc Joly wrote:</h3>
<div class="paragraph">
<p>Empty classes are used in some C&#43;&#43; idioms (mostly as tags), as described in the linked shared by Evgeny. Removing C&#43;&#43; from the list of targeted languages.</p>
</div>
</div>
</div>
</div>