== Noncompliant Code Example

[source,text]
----
using System;

class MyEventProcucer
{
    public static event EventHandler eventFired;
}

public class MyEventSubscriber : IDisposable
{

    public MyEventSubscriber()
    {
        MyEventProcucer.eventFired += c_EventFired;  // Noncompliant.
    }

    static void c_EventFired(object sender, EventArgs e)
    {}

    public void Dispose()
    {}
}
----
