<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Subscribing to events without unsubscribing later on can lead to memory leaks or even duplicate subscriptions, i.e. code which is executed multiple times by mistake.</p>
</div>
<div class="paragraph">
<p>Even if there is no problem right now, the code is more difficult to review and a simple refactoring can create a bug. For example the lifetime of the event publisher could change and prevent subscribers from being garbage collected.</p>
</div>
<div class="paragraph">
<p>There are patterns to automatically unsubscribe, but the simplest and most readable solution remains to unsubscribe from events explicitly using <code>RemoveHandler</code>.</p>
</div>
<div class="paragraph">
<p>This rule raises an issue when a class subscribes to an even using <code>AddHandler</code> without explicitly unsubscribing with <code>RemoveHandler</code>.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-vbnet" data-lang="vbnet">Class MyEventProcucer
    Public Shared Event EventFired As EventHandler
End Class

Public Class MyEventSubscriber
    Implements IDisposable

    Public Sub New()
        AddHandler MyEventProcucer.EventFired, AddressOf c_EventFired  'Noncompliant
    End Sub

    Private Sub c_EventFired(sender As Object, e As EventArgs)
    End Sub

    Public Sub Dispose() Implements IDisposable.Dispose
    End Sub
End Class</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-vbnet" data-lang="vbnet">Class MyEventProcucer
    Public Shared Event EventFired As EventHandler
End Class

Public Class MyEventSubscriber
    Implements IDisposable

    Public Sub New()
        AddHandler MyEventProcucer.EventFired, AddressOf c_EventFired
    End Sub

    Private Sub c_EventFired(ByVal sender As Object, ByVal e As EventArgs)
    End Sub

    Public Sub Dispose() Implements IDisposable.Dispose
        RemoveHandler MyEventProcucer.EventFired, AddressOf c_EventFired
    End Sub
End Class</code></pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p><a href="https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/events/how-to-subscribe-to-and-unsubscribe-from-events#unsubscribing">How to subscribe to and unsubscribe from events (C# Programming Guide)</a></p>
</li>
<li>
<p><a href="https://michaelscodingspot.com/5-techniques-to-avoid-memory-leaks-by-events-in-c-net-you-should-know/">5 Techniques to avoid Memory Leaks by Events in C# .NET you should know</a></p>
</li>
</ul>
</div>
<hr>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Unsubscribe from this event explicitly with "RemoveHandler".</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>The <code>AddHandler</code> call.</p>
</div>
</div>
</div>
</div>