<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>The <code>Any#equals(other: Any?)</code> method is used to compare two objects to see if they are equal.</p>
</div>
<div class="paragraph">
<p>The <code>other</code> parameter&#8217;s type is <code>Any?</code>, this means that an object of any type, as well as <code>null</code>, can be passed as an argument to this method.</p>
</div>
<div class="paragraph">
<p>Any class overriding <code>Any#equals(other: Any?)</code> should respect this contract, accept any object as an argument, and return <code>false</code> when the
argument&#8217;s type differs from the expected type. The <code>other</code> parameter&#8217;s type can be checked using the <code>is</code> operator or by comparing the <code>javaClass</code> field:</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-kotlin" data-lang="kotlin">override fun equals(other: Any?): Boolean {
  // ...
  if (other?.javaClass != this.javaClass) {
    return false
  }
  // ...
}</code></pre>
</div>
</div>
<div class="paragraph">
<p>However, it is an issue to assume that the <code>equals</code> method will only be used to compare objects of the same type. Casting the <code>other</code> parameter without a prior test will throw a <code>ClassCastException</code> instead of returning false.</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-kotlin" data-lang="kotlin">class MyClass {
  override fun equals(other: Any?): Boolean {
    val that = other as MyClass // may throw a ClassCastException
    // ...
  }
  // ...
}</code></pre>
</div>
</div>
<div class="paragraph">
<p>This rule raises an issue when <code>other</code> parameter&#8217;s type has not been tested before a cast operation.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Ensure the <code>other</code> parameter&#8217;s type is checked by comparing <code>other?.javaClass</code> and <code>this.javaClass</code>, or use the <code>is</code> operator to test `other&#8217;s type.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-kotlin" data-lang="kotlin">class MyClass {
  override fun equals(other: Any?): Boolean {
    if (this === other) {
        return true
    }
    val that = other as MyClass // Noncompliant, may throw a ClassCastException
    // ...
  }
  // ...
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-kotlin" data-lang="kotlin">class MyClass {
  override fun equals(other: Any?): Boolean {
    if (this === other) {
        return true
    }
    if (other?.javaClass != this.javaClass) {
      return false
    }
    val that = other as MyClass // Compliant, other's type is MyClass
    // ...
  }
  // ...
}</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p><a href="https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/-any/equals.html">Kotlin Standard Library - Any#equals(other: Any?)</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>