== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,scala]
----
if (file != null) {
  if (file.isFile || file.isDirectory) {
    /* ... */
  }
}
----

=== Compliant solution

[source,scala]
----
if (file != null && isFileOrDirectory(file)) {
  /* ... */
}

def isFileOrDirectory(file: File): Boolean = file.isFile || file.isDirectory
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../highlighting.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
