<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>SAP recommends to keep the result set of any request as small as possible for performance reasons.</p>
</div>
<div class="paragraph">
<p>A <code>SELECT...ENDSELECT</code> request will retrieve multiple records at the same time. Stopping immediately such a request with a <code>CHECK</code>, <code>RETURN</code> or <code>EXIT</code> statement will not reduce the number of retrieved records. If the goal is to retrieve a single record, or check that at least one record exists, it is recommended to add <code>UP TO 1 ROWS</code> or to use <code>SELECT SINGLE</code>.</p>
</div>
<div class="paragraph">
<p>Even if the request already retrieved only one record, adding <code>UP TO 1 ROWS</code> or to using <code>SELECT SINGLE</code> will make the code easier to read.</p>
</div>
<div class="paragraph">
<p>This rule raises an issue when a <code>SELECT...ENDSELECT</code> which has neither <code>SINGLE</code> nor <code>UP TO 1 ROWS</code> and:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>contains only one statement and it is a <code>CHECK</code>, <code>RETURN</code> or <code>EXIT</code></p>
</li>
<li>
<p>or when it is completely empty.</p>
</li>
</ul>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-abap" data-lang="abap">SELECT * FROM sbook INTO ls_book WHERE carrid EQ l_carrid.
    EXIT.
ENDSELECT.
IF sy-subrc NE 0.
    WRITE:/ 'NO RECORD FOUND'.
ENDIF.</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-abap" data-lang="abap">SELECT SINGLE * sbook INTO ls_book WHERE carrid EQ l_carrid.
IF sy-subrc NE 0.
    WRITE:/ 'NO RECORD FOUND'.
ENDIF.</code></pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p><a href="https://help.sap.com/doc/saphelp_nw70/7.0.31/en-US/aa/4734940f1c11d295380000e8353423/content.htm?no_cache=true">SAP documentation - Keep the Result Set Small</a></p>
</li>
</ul>
</div>
<hr>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Add "SELECT SINGLE" or "Up to 1 ROW" to this query; it uses only one record.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>Primary: The SELECT keyword.</p>
</div>
<div class="literalblock">
<div class="content">
<pre>Secondary:</pre>
</div>
</div>
<div class="ulist">
<ul>
<li>
<p>The "RETURN", "CHECK" or "EXIT" statement.</p>
</li>
<li>
<p>message: "XXX" stops the request at the first record.
(XXX is the command name)</p>
</li>
</ul>
</div>
</div>
</div>
</div>