<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Contrary to possible developer expectations, a template assignment operator will not suppress the compiler generated <em>copy assignment operator</em>. This may lead to incorrect copy semantics for members requiring deep copies.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">class A
{
public:
// A &amp; operator= ( A const &amp; rhs ) Example 1 - implicitly generated
// {
// i = rhs.i;
// return *this;
// }
  template &lt;typename T&gt;
  T &amp; operator= ( T const &amp; rhs ) // Example 2
  {
    if ( this != &amp;rhs ) {
      delete i;
        i = new int32_t;
      *i = *rhs.i;
    }
    return *this;
  }
private:
  int32_t * i; // Member requires deep copy
};

void f ( A const &amp; a1, A &amp; a2 )
{
  a2 = a1; // Unexpectedly uses Example 1
}</code></pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p>MISRA C&#43;&#43;:2008, 14-5-3</p>
</li>
</ul>
</div>
<hr>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Add an explicit copy assignment operator to this class.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_relates_to_s1025">relates to: <a data-rspec-id="S1025" class="rspec-auto-link">S1025</a></h3>

</div>
</div>
</div>