include::../why-dotnet.adoc[]

include::../impact-dotnet.adoc[]

include::../how-dotnet.adoc[]

=== Code examples

==== Noncompliant code example

[source,csharp,diff-id=1,diff-type=noncompliant]
----
void CreateNewTime()
{
    var birthDate = new DateTime(1994, 7, 5, 16, 23, 42);
}
----

==== Compliant solution

[source,csharp,diff-id=1,diff-type=compliant]
----
void CreateNewTime()
{
    var birthDate = new DateTime(1994, 7, 5, 16, 23, 42, DateTimeKind.Utc);
    // or from .Net 6 onwards, use DateOnly:
    var birthDate = new DateOnly(1994, 7, 5);
}
----

include::../resources-dotnet.adoc[]