== Why is this an issue?

include::../whyAnIssue.adoc[]

=== What is the potential impact?

include::../impact.adoc[]

== How to fix it

include::../howToFIxIt.adoc[]

=== Code examples

==== Noncompliant code example

[source,csharp]
----
void CreateNewTime()
{
    var birthDate = new DateTime(1994, 7, 5, 16, 23, 42);
}
----

==== Compliant solution

[source,csharp]
----
void CreateNewTime()
{
    var birthDate = new DateTime(1994, 7, 5, 16, 23, 42, DateTimeKind.Utc);
}
----

== Resources
=== Documentation

include::../documentation.adoc[]