<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This is an issue when the HTML meta viewport tag prevents users from zooming by setting <code>user-scalable=no</code> or limiting <code>maximum-scale</code> to 2.0 or less.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>The viewport meta tag controls how web pages are displayed on mobile devices. When it prevents zooming, it creates accessibility barriers for users who need to enlarge text to read content.</p>
</div>
<div class="paragraph">
<p>Many users rely on zoom functionality to make text readable:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>People with visual impairments who need larger text</p>
</li>
<li>
<p>Users with reading difficulties like dyslexia</p>
</li>
<li>
<p>Anyone using a device in bright sunlight or poor lighting conditions</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>The Web Content Accessibility Guidelines (WCAG) 2.2 require that users can resize text up to 200% without assistive technology. When the viewport meta tag blocks zooming with <code>user-scalable=no</code> or restricts it with <code>maximum-scale=1</code>, it violates this requirement.</p>
</div>
<div class="paragraph">
<p>This creates a poor user experience and excludes users who depend on zoom functionality to access your content effectively.</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="paragraph">
<p>Users with visual impairments or reading difficulties may be unable to access your content. This violates accessibility standards and can lead to legal compliance issues in jurisdictions that require web accessibility.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Remove the user-scalable=no attribute to allow users to zoom in and out as needed.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-html" data-lang="html">&lt;meta name="viewport" content="width=device-width, initial-scale=1, user-scalable=no"&gt; &lt;!-- Noncompliant --&gt;</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-html" data-lang="html">&lt;meta name="viewport" content="width=device-width, initial-scale=1"&gt;</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>MDN: Viewport meta tag - <a href="https://developer.mozilla.org/en-US/docs/Web/HTML/Viewport_meta_tag">Complete guide to the viewport meta tag and its attributes</a></p>
</li>
<li>
<p>Deque University: Meta viewport accessibility - <a href="https://dequeuniversity.com/rules/axe/4.10/meta-viewport">Accessibility guidelines for viewport meta tags</a></p>
</li>
</ul>
</div>
</div>
<div class="sect2">
<h3 id="_standards">Standards</h3>
<div class="ulist">
<ul>
<li>
<p>WCAG 2.2: Success Criterion 1.4.4 Resize text - <a href="https://www.w3.org/TR/WCAG22/#resize-text">Text must be resizable up to 200% without assistive technology</a></p>
</li>
<li>
<p>ACT Rules: Meta viewport allows for zoom - <a href="https://act-rules.github.io/rules/b4f0c3">Accessibility conformance testing rule for viewport zoom restrictions</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>