== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,javascript]
----
var myNumber = 010;   // Noncompliant. myNumber will hold 8, not 10 - was this really expected?
----

=== Compliant solution

[source,javascript]
----
var myNumber = 8;
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
