<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises when a control flow statement (<code>return</code>, <code>break</code>, <code>continue</code>) is used inside a TaskGroup or Nursery context manager.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Using control flow statements like <code>return</code>, <code>break</code>, or <code>continue</code> inside async TaskGroup or Nursery blocks leads to counterintuitive behavior that can confuse developers and introduce bugs.</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="sect3">
<h4 id="_deferred_execution_in_taskgroup">Deferred execution in TaskGroup</h4>
<div class="paragraph">
<p>In asyncio&#8217;s TaskGroup, control flow statements don&#8217;t take immediate effect. Instead, they wait for all tasks in the group to complete before executing. This can lead to:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>Unexpected delays when tasks run longer than anticipated</p>
</li>
<li>
<p>Code that appears to exit early but actually continues running</p>
</li>
<li>
<p>Potential infinite loops if background tasks never complete</p>
</li>
</ul>
</div>
</div>
<div class="sect3">
<h4 id="_lost_return_values_in_nurseries">Lost return values in Nurseries</h4>
<div class="paragraph">
<p>In Trio and AnyIO nurseries, return values can be lost if other tasks in the nursery raise exceptions:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>When a background task raises an exception, the return value from the main flow is discarded</p>
</li>
<li>
<p>The nursery&#8217;s exception handling takes precedence over return values</p>
</li>
<li>
<p>Silent data loss that&#8217;s difficult to debug</p>
</li>
</ul>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it_in_asyncio">How to fix it in Asyncio</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Move the control flow statement outside the TaskGroup or Nursery block, and use the appropriate cancellation mechanism before exiting the block.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">import asyncio

async def process():
    async with asyncio.TaskGroup() as tg:
        tg.create_task(background_task())

        if condition():
            return "result"  # Noncompliant: waits for background_task() to complete</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">import asyncio

async def process():
    result = None
    async with asyncio.TaskGroup() as tg:
        task = tg.create_task(background_task())

        if condition():
            result = "result"
            task.cancel()

    return result</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it_in_trio">How to fix it in Trio</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Move the control flow statement outside the Nursery block, and use the appropriate cancellation mechanism before exiting the block.</p>
</div>
<div class="sect2">
<h3 id="_code_examples_2">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example_2">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">import trio

async def process():
    async with trio.open_nursery() as nursery:
        nursery.start_soon(background_task)

        if condition():
            return "result"  # Noncompliant: value may be lost</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution_2">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">import trio

async def process():
    result = None
    async with trio.open_nursery() as nursery:
        nursery.start_soon(background_task)

        if condition():
            result = "result"
            nursery.cancel_scope.cancel()

    return result</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it_in_anyio">How to fix it in AnyIO</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Move the control flow statement outside the TaskGroup block, and use the appropriate cancellation mechanism before exiting the block.</p>
</div>
<div class="sect2">
<h3 id="_code_examples_3">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example_3">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">import anyio

async def process():
    async with anyio.create_task_group() as tg:
        tg.start_soon(background_task)

        if condition():
            return "result"  # Noncompliant: waits for background_task</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution_3">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">import anyio

async def process():
    result = None
    async with anyio.create_task_group() as tg:
        tg.start_soon(background_task)

        if condition():
            result = "result"
            tg.cancel_scope.cancel()

    return result</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Asyncio documentation - <a href="https://docs.python.org/3/library/asyncio-task.html#task-groups">Task Groups</a></p>
</li>
<li>
<p>Trio documentation - <a href="https://trio.readthedocs.io/en/latest/reference-core.html#nurseries-and-spawning">Nurseries and spawning</a></p>
</li>
<li>
<p>AnyIO documentation - <a href="https://anyio.readthedocs.io/en/stable/tasks.html#creating-and-managing-tasks">Creating and managing tasks</a></p>
</li>
<li>
<p>Trio issue #1493 - <a href="https://github.com/python-trio/trio/issues/1493">Returning a value from inside a nursery block behaves counterintuitively</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="ulist">
<ul>
<li>
<p>Refactor the block to eliminate this <code>{return|break|continue}</code> statement.</p>
</li>
</ul>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="ulist">
<ul>
<li>
<p>Primary location: The control flow statement (return/break/continue)</p>
</li>
<li>
<p>Secondary locations: The TaskGroup/Nursery block declaration and the async keyword of the function</p>
</li>
</ul>
</div>
</div>
</div>
</div>