<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Synchronizing at the method, rather than the block level could lead to problems when maintenance adds code to the method, perhaps inadvertently synchronizing it as well. Instead, synchronization should be applied to the smallest feasible block for optimum performance and maintainability.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">public class MyClass() {

  public void synchronized doSomething() { // Noncompliant
    // ...
  }
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">public class MyClass() {

  private Object lockObj = new Object();

  public void doSomething() {
    synchronized(lockObj) {
      // ...
    }
  }
}</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Move synchronization to the block level</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_16_jun_2015_170808_nicolas_peru_wrote">on 16 Jun 2015, 17:08:08 Nicolas Peru wrote:</h3>
<div class="paragraph">
<p>Looks good, wondering with overlap or even catch 22 with some other RSPEC about not synchronizing on <code>this</code></p>
</div>
</div>
<div class="sect2">
<h3 id="_on_16_jun_2015_190600_ann_campbell_wrote">on 16 Jun 2015, 19:06:00 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>The closest I&#8217;m finding is <a data-rspec-id="S3067" class="rspec-auto-link">RSPEC-3067</a> [~nicolas.peru], and unfortunately "this" seems to be suppressed from JIRA&#8217;s indexing :-(</p>
</div>
<div class="paragraph">
<p>If you have a specific RSpec in mind, can you give me another pointer toward it?</p>
</div>
</div>
</div>
</div>