When several lines must be inserted/updated into an internal table, instead of doing those changes line by line, mass operations should be used because they offer better performance  by design.


This rule raises an issue when a single line operation like ``++APPEND++``, ``++CONCATENATE++``, and ``++INSERT++`` is performed on an internal table in a loop. 


== Noncompliant Code Example

----
LOOP AT ITAB1 INTO WA. 
  APPEND WA TO ITAB2. 
ENDLOOP.
----


== Compliant Solution

----
APPEND LINES OF ITAB1 TO ITAB2. 
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
