Internal tables can be sorted without specifying the specific fields on which to sort. However, doing so is inefficient because when a sort key is not specified, the entire row is used in the sort, which can be markedly inefficient.


== Noncompliant Code Example

----
SORT ITAB.
----


== Compliant Solution

----
SORT ITAB BY LAND WEIGHT.
----

