== Why is this an issue?

Internal tables can be sorted without specifying the specific fields on which to sort. However, doing so is inefficient because when a sort key is not specified, the entire row is used in the sort, which can be markedly inefficient.


=== Noncompliant code example

[source,abap]
----
SORT ITAB.
----


=== Compliant solution

[source,abap]
----
SORT ITAB BY LAND WEIGHT.
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Provide the missing sort fields


endif::env-github,rspecator-view[]
