<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when a function is declared async but does not use any asynchronous features.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Declaring a function with <code>async def</code> signals that it&#8217;s a coroutine, typically because it needs to use <code>await</code> for non-blocking operations (like I/O), or employs other asynchronous features like <code>async for</code> or <code>async with</code>.</p>
</div>
<div class="paragraph">
<p>If none of these asynchronous mechanisms are utilized within the function&#8217;s body, the <code>async</code> declaration is superfluous.</p>
</div>
<div class="paragraph">
<p>Using <code>async</code> unnecessarily can:</p>
</div>
<div class="ulist">
<ul>
<li>
<p><strong>Reduce Code Clarity:</strong> It misleads developers into thinking the function performs asynchronous operations when it doesn&#8217;t, making the codebase harder to understand and reason about.</p>
</li>
<li>
<p><strong>Introduce Minor Overhead:</strong> Calling an <code>async def</code> function creates a coroutine object. While modern Python is efficient, this still incurs more overhead than a direct synchronous function call if the asynchronicity isn&#8217;t actually leveraged. The function&#8217;s body will execute synchronously until an <code>await</code> (if any) is encountered. If there&#8217;s no <code>await</code> or other async feature, it essentially runs synchronously but wrapped as a coroutine.</p>
</li>
<li>
<p><strong>Complicate Usage:</strong> Callers must use <code>await</code> (or schedule it as a task) to execute the coroutine and get its result. This is an unnecessary ceremony if the function itself isn&#8217;t truly asynchronous.</p>
</li>
</ul>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Either remove the <code>async</code> keyword, or start using the appropriate asynchronous features.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">async def my_function():  # Noncompliant
    print("Hello from my function")</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">def my_function():  # Compliant
    print("Hello from my function")</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_noncompliant_code_example_2">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">def another_function(): ...

async def my_function():  # Noncompliant
    return another_function()</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution_2">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">async def another_function(): ...

async def my_function():  # Compliant
    return await another_function()</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_articles_blog_posts">Articles &amp; blog posts</h3>
<div class="ulist">
<ul>
<li>
<p><a href="https://journal.stuffwithstuff.com/2015/02/01/what-color-is-your-function/">What Color Is Your Function? - Bob Nystrom</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove the <code>async</code> keyword or use the appropriate asynchronous features.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>Primary: the entire function body
Secondary: the <code>async</code> keyword of the function declaration</p>
</div>
</div>
</div>
</div>