The ``onEnterFrame`` event handler is continually invoked at the frame rate of the SWF file, regardless of which individual movie frame it is set for. Having too many ``onEnterFrame`` handlers can seriously degrade performance. 

If the use of this event handler cannot be avoided entirely, then it should be created as close to its use as possible, and then destroyed as soon as possible afterward.


== Noncompliant Code Example

----
movieClip.onEnterFrame = function () {   // Noncompliant
   // ...
}
----

