<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p><code>std::format</code>, introduced by C&#43;&#43;20, enables straightforward string construction out of values of various types.</p>
</div>
<div class="paragraph">
<p>Before C&#43;&#43;20, one popular way to obtain the same result was the conversion of the values with <code>std::to_string</code> and piecewise string concatenation.</p>
</div>
<div class="paragraph">
<p><code>std::format</code> is strictly superior. It is more efficient because it constructs the string in-place instead of copying substrings one by one. It is also often shorter and easier to read because the format pattern is presented in a single piece and not scattered across the concatenation expression.</p>
</div>
<div class="paragraph">
<p>This rule reports string concatenation cases that can be replaced by <code>std::format</code> to improve performance and readability.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">std::string greeting(int n) {
  return "Hello, player " + std::to_string(n) + "."; // Noncompliant
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">std::string greeting(int n) {
  return std::format("Hello, player {}.", n); // Compliant
}</code></pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>C&#43;&#43; reference - <a href="https://en.cppreference.com/w/cpp/utility/format/format"><code>std::format</code></a></p>
</li>
<li>
<p>C&#43;&#43; reference - <a href="https://en.cppreference.com/w/cpp/string/basic_string/to_string"><code>std::to_string</code></a></p>
</li>
</ul>
</div>
</div>
<div class="sect2">
<h3 id="_articles_blog_posts">Articles &amp; blog posts</h3>
<div class="ulist">
<ul>
<li>
<p>C&#43;&#43; Stories - <a href="https://www.cppstories.com/2022/custom-stdformat-cpp20/">Formatting Custom types with std::format from C&#43;&#43;20</a></p>
</li>
</ul>
</div>
</div>
<div class="sect2">
<h3 id="_related_rules">Related rules</h3>
<div class="ulist">
<ul>
<li>
<p><a data-rspec-id="S6484" class="rspec-auto-link">S6484</a> - Concatenated "std::format" outputs should be replaced by a single invocation</p>
</li>
<li>
<p><a data-rspec-id="S6494" class="rspec-auto-link">S6494</a> - C&#43;&#43; formatting functions should be used instead of C printf-like functions</p>
</li>
<li>
<p><a data-rspec-id="S6495" class="rspec-auto-link">S6495</a> - "std::format" should be used instead of standard output manipulators</p>
</li>
</ul>
</div>
</div>
</div>
</div>