Global variables provide a convenient way of accessing the same value or set of values from multiple places in the code without the need to pass them through the interface and possibly the entire call hierarchy. As a consequence, they can be used to avoid repetition and guarantee that the same value is used in multiple places.


However, modification of such a global variable may potentially change the behavior of any program component which makes it difficult to reason locally about the program's correctness. Additionally, the modification of the same global variable in a multi-threaded program may lead to data races.


This rule raises an issue on the modification of publicly accessible non-atomic global variables.


== Noncompliant Code Example

----
std::size_t operationCounter = 0;

int add(int x, int y) {
  ++operationCounter; // Noncompliant
  return x + y;
}
----


== Compliant Solution

----
class StatsCollector
{
public:
  static void registerOp() {
      ++operationCounter;
  }

private:
  static std::atomic<std::size_t> operationCounter{0};
};

int add(int x, int y) {
  StatsCollector::registerOp();
  return x + y;
}
----


== Exceptions

An issue will not be raised on modifications of a private static member of a class, as the impact of such modification is limited to its members and friends. This makes additional states visible from the class declaration and makes it possible to protect variables from multi-threaded access if necessary.


== See

* S5421 - marking global variables ``++const++`` to prevent modification


ifdef::env-github,rspecator-view[]
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
