<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>If an exception is thrown when constructing the exception object, or when evaluating the assignment expression that initializes the exception object, it is that exception that propagates in preference to the one that was about to be thrown. This may be inconsistent with developer expectations.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">// construction of E2 causes an exception to be thrown
class E2
{
  public:
  E2 ( )
  {
    throw 10;
  }
};

try
{
  if ( ... )
  {
    throw E2 ( ); // Non-compliant – int exception thrown when constructing the E2 object
  }
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">class E
{
  public:
  E ( ) { } // Assume constructor cannot cause an exception
};

try
{
  if ( ... )
  {
    throw E ( ); // Compliant – no exception thrown when constructing the object
  }
}</code></pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p>MISRA C&#43;&#43;:2008, 15-1-1</p>
</li>
</ul>
</div>
<hr>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>An exception could be thrown during construction of "xxx".</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_14_oct_2014_175636_ann_campbell_wrote">on 14 Oct 2014, 17:56:36 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~samuel.mercier] please correct the following</p>
</div>
<div class="ulist">
<ul>
<li>
<p>Add a See section to the description listing the appropriate MISRA number</p>
</li>
<li>
<p>Fill in Applicability.</p>
</li>
<li>
<p>Make sure the appropriate MISRA C and MISRA C&#43;&#43; fields on the references tab are filled in</p>
</li>
</ul>
</div>
</div>
</div>
</div>