An empty multi-line comment is likely to be a mistake and doesn't help to improve the readability of the code. For these reasons, it should be removed.

== Why is this an issue?

=== Noncompliant code example

[source,java]
----
/* */

/*

 */
----

=== Compliant solution

[source,java]
----
/*
Proper comment or no comment at all
*/
----
