== Why is this an issue?

include::multi-lines-description.adoc[]

=== Noncompliant code example

[source,java]
----
/* */

/*

 */
----

=== Compliant solution

[source,java]
----
/*
Proper comment or no comment at all
*/
----
