== Why is this an issue?

An empty multi-line comment doesn't help to improve the readability of the code and might indicate an oversight. A meaningful comment should be added or the comment markers should be removed.

=== Noncompliant code example

[source,java]
----
/* */

/*

 */
----

=== Compliant solution

[source,java]
----
/*
Proper comment or no comment at all
*/
----
