<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>When a method doesn&#8217;t match it&#8217;s <code>super</code> method in visibility (<code>public</code>, <code>protected</code>, &#8230;&#8203;), malicious callers could take advantage of the over-broad access offered by the child class to undermine the application.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-swift" data-lang="swift">public class Parent {

  protected void foo() {
    //...
  }
}

public class Child extends Parent {

  public void foo() {  // Noncompliant
    // ...
    super.foo();
  }
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-swift" data-lang="swift">public class Parent {

  protected void foo() {
    //...
  }
}

public class Child extends Parent {

  protected void foo() {
    // ...
    super.foo();
  }
}</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="ulist">
<ul>
<li>
<p>Decrease the accessibility of this method to "xxx" to match the parent class implementation.</p>
</li>
</ul>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="ulist">
<ul>
<li>
<p>visibility modifier</p>
</li>
</ul>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_31_oct_2018_115713_tibor_blenessy_wrote">on 31 Oct 2018, 11:57:13 Tibor Blenessy wrote:</h3>
<div class="paragraph">
<p>This rule was forked from <a data-rspec-id="S3551" class="rspec-auto-link">RSPEC-3551</a> to allow <a data-rspec-id="S3551" class="rspec-auto-link">RSPEC-3551</a> to focus only on synchronization which is Java specific concept and real bug. I am actually not really sure about the value of this rule, that&#8217;s why it is not included in the default profile. However, I am really sure about the value of <a data-rspec-id="S3551" class="rspec-auto-link">RSPEC-3551</a> so I wanted to remove this fuzzy part from it.</p>
</div>
</div>
</div>
</div>