== Why is this an issue?

According to the W3C specifications:

____
A string cannot directly contain a newline. To include a newline in a string, use an escape representing the line feed character in ISO-10646 (U+000A), such as "\A" or "\00000a".

{empty}[...]

It is possible to break strings over several lines, for aesthetic or other reasons, but in such a case the newline itself has to be escaped with a backslash (\).

____


=== Noncompliant code example

[source,css]
----
a {
  content: "first
    second";     
} 
----


=== Compliant solution

[source,css]
----
a {
  content: "first\Asecond";     
}  
----


== Resources

* https://www.w3.org/TR/CSS2/syndata.html#strings[CSS Specification] - Strings


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

endif::env-github,rspecator-view[]
