== Why is this an issue?

Calling ``++DELETE ADJACENT DUPLICATES++`` won't reliably do any good if the table hasn't first been sorted to put duplicates side by side, since the ``++ADJACENT++`` part of the command looks for multiple rows side-by-side with the same content.


=== Noncompliant code example

[source,abap]
----
DELETE ADJACENT DUPLICATES FROM ITAB COMPARING LAND.
----


=== Compliant solution

[source,abap]
----
SORT ITAB BY LAND.
DELETE ADJACENT DUPLICATES FROM ITAB COMPARING LAND.
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Sort this table before the delete


endif::env-github,rspecator-view[]
