== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,java]
----
int <T> Add(int a, int b) // Noncompliant; <T> is ignored
{
  return a + b; 
}
----

=== Compliant solution

[source,java]
----
int Add(int a, int b)
{
  return a + b; 
}
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
