<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p><code>PERFORM</code> is used to execute a paragraph located somewhere in the program and then, once executed, the execution flow will continue on the line following the PERFORM statement. This is the expected behaviour that can be broken if a <code>GO TO</code> is added in the called paragraph. When mixing <code>PERFORM</code> and <code>GO TO</code> you can quickly be lost on the execution flow and finally don&#8217;t get the one you expect. For this reason, calling <code>PERFORM</code> with paragraphs that used <code>GO TO</code> should be avoided.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cobol" data-lang="cobol">PROCEDURE DIVISION.
DISPLAY-9-LETTERS.
    PERFORM ABC.
    DISPLAY "END OF DISPLAY-9-LETTERS".
    STOP RUN.

    ABC.
        DISPLAY "ABC".
        GO TO XYZ.

    DEF.
        DISPLAY "DEF".

    XYZ.
        DISPLAY "XYZ".</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cobol" data-lang="cobol">PROCEDURE DIVISION.
DISPLAY-9-LETTERS.
    PERFORM ABC.
    PERFORM DEF.
    PERFORM XYZ.
    DISPLAY "END OF DISPLAY-9-LETTERS".
    STOP RUN.

    ABC.
        DISPLAY "ABC".

    DEF.
        DISPLAY "DEF".

    XYZ.
        DISPLAY "XYZ".</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Refactor this code to not use "GO TO" in the paragraph XXX.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>First: Paragraph name used in a PERFORM</p>
</div>
<div class="paragraph">
<p>Second: the GO TO statement of the Paragraph</p>
</div>
</div>
</div>
</div>