<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>When a class with no <code>virtual</code> destructor is used as a base class, surprises can occur if pointers to instances of this class are used. Specifically, if an instance of a derived class is <code>delete</code>d through a pointer to the base type, the behavior is undefined and can lead to resource leaks, crashes or corrupted memory.</p>
</div>
<div class="paragraph">
<p>If it is not expected for base class pointers to be deleted, then the destructor should be made <code>protected</code> to avoid such a misuse.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">class Base { // Noncompliant: no destructor is supplied, and the default version is not virtual
public:
  Base() {}
  virtual void doSomething() {}
};

class Derived : public Base {
}

void f() {
  Base *p = new Derived();
  delete p; // Undefined behavior
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">class Base {
public:
  Base() {}
  virtual ~Base() = default;
  virtual void doSomething() {}
};</code></pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p><a href="https://wiki.sei.cmu.edu/confluence/x/5Xs-BQ">CERT, OOP52-CPP.</a> - Do not delete a polymorphic object without a virtual destructor</p>
</li>
<li>
<p><a href="http://www.gotw.ca/publications/mill18.htm">Virtuality article</a></p>
</li>
<li>
<p>C++ Core Guidelines - <a href="https://github.com/isocpp/CppCoreGuidelines/blob/e49158a/CppCoreGuidelines.md#c35-a-base-class-destructor-should-be-either-public-and-virtual-or-protected-and-non-virtual">C.35: A base class destructor should be either public and virtual, or protected and non-virtual</a></p>
</li>
<li>
<p>C++ Core Guidelines - <a href="https://github.com/isocpp/CppCoreGuidelines/blob/e49158a/CppCoreGuidelines.md#c127-a-class-with-a-virtual-function-should-have-a-virtual-or-protected-destructor">C.127: A class with a virtual function should have a virtual or protected destructor</a></p>
</li>
</ul>
</div>
</div>
</div>