Shared naming conventions allow teams to collaborate effectively. This rule raises an issue when a pom's ``++artifactId++`` does not match the provided regular expression.


== Noncompliant Code Example

With the default regular expression: ``++[a-z][a-z-0-9]+++``

----
<project ...>
  <artifactId>My_Project</artifactId>  <!-- Noncompliant -->

  <!-- ... -->
</project>
----


== Compliant Solution

----
<project ...>
  <artifactId>my-project</artifactId>

  <!-- ... -->
</project>
----



ifdef::rspecator-view[]
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::rspecator-view[]
