This is an issue when destructive string methods like `tr!`, `gsub!`, or `sub!` are used in contexts where their return value is assigned to a variable, used in method chains, or returned from a method.

== Why is this an issue?

Destructive string methods in Ruby (those ending with `!`) have a special behavior: they return `nil` when no changes are made to the string, rather than returning the original string.

This behavior can cause unexpected runtime errors. For example, if you chain methods after a destructive call that returns `nil`, you'll get a `NoMethodError` when trying to call methods on `nil`.

The problem is particularly subtle because the code might work correctly in some cases (when the string is actually modified) but fail in others (when no modification occurs). This makes bugs hard to reproduce and debug.

Consider this example:

[source,ruby]
----
str = "hello world"
result = str.tr!(" ", "_")  # Returns "hello_world"
puts result  # Works fine

str = "helloworld" 
result = str.tr!(" ", "_")  # Returns nil (no spaces to replace)
puts result  # Prints nothing - unexpected!
----

The same code produces different types of results depending on the input data, making the behavior unpredictable.

=== What is the potential impact?

This issue can cause `NoMethodError` exceptions at runtime when methods are called on `nil` values. These errors are often intermittent and depend on the input data, making them difficult to catch during testing and potentially causing production failures.

== How to fix it

Use the non-destructive variant of the method instead. Non-destructive methods always return a string, making the behavior predictable.

=== Code examples

==== Noncompliant code example

[source,ruby,diff-id=1,diff-type=noncompliant]
----
str = "color"
result = str.downcase.tr!(" ", "_")  # Noncompliant: returns nil
puts result  # nil - unexpected!
----

==== Compliant solution

[source,ruby,diff-id=1,diff-type=compliant]
----
str = "color"
result = str.downcase.tr(" ", "_")  # Always returns a string
puts result  # "color"
----

== Resources

=== Documentation

 * Ruby String Documentation - https://ruby-doc.org/core/String.html[Official Ruby documentation for String class methods]

 * Ruby Style Guide - String Methods - https://rubystyle.guide/#string-mutating-methods[Community style guide recommendations for string methods]
