== Why is this an issue?

Spring provides the `@EventListener` annotation to make a method handle some specific types of events.
This allows to skip the implementation of the `ApplicationListener` interface, making it easier to handle events.
However, the `@EventListener` annotation can only be used on methods that have at most one parameter, which should be the specific event that we want to handle.
For multiple types of events, we can use the `classes` argument of the annotation, and provide a list of classes of events that we want to handle.

This rule raises an issue on all methods annotated with `@EventListener` that have more than one parameter.

== How to fix it

=== Code examples

==== Noncompliant code example

[source,java,diff-id=1,diff-type=noncompliant]
----
@EventListener
void handleEvent(CustomerEvent customerEvent, ExceptionalEvent exceptionalEvent) { // Non compliant, this will cause a runtime error
    //... some event handling
}
----

==== Compliant solution

[source,java,diff-id=1,diff-type=compliant]
----
@EventListener(classes = {CustomerEvent.class, ExceptionalEvent.class})
void handleEvent(ApplicationEvent event) { // Only one parameter, of the super type `ApplicationEvent`
    //... some event handling
}
----

== Resources
=== Documentation

Spring API - https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/context/event/EventListener.html[@EventListener]