Virtual and non-virtual functions are not dispatched the same way: for virtual functions, the resolution will be done dynamically while for non-virtual ones, it will be done statically by the compiler based on the type it sees.

Thus, overloading functions with both virtual and non-virtual functions adds a level of complexity to the code because the resolution of these functions will be completely different even though they share the same name. So the code becomes more confusing and more error-prone (for example if a virtual function is mistaken with a non-virtual one).


== Noncompliant Code Example

----
#include <iostream>

struct A {
  virtual void size(double a) {
    std::cout << "A double" << std::endl;
  }
  void size(int b) { // Noncompliant, it overloads size(double) which is virtual
    std::cout << "A int" << std::endl;
  }
};

struct B : A {
  void size(double a) override {
    std::cout << "B double" << std::endl;
  }  
  void size(int b) {
    std::cout << "B int" << std::endl;
  }
};

int main() {
  A* a = new B;
  a->size(2.2); // will display "B double"
  a->size(2); // will display "A int"

  return 0;
}
----


== Compliant Solution

----
#include <iostream>

struct A {
  virtual void size(double a) {
    std::cout << "A double" << std::endl;
  }
  virtual void size(int b) {
    std::cout << "A int" << std::endl;
  }
};

struct B : A {
  void size(double a) override {
    std::cout << "B double" << std::endl;
  }  
  void size(int b) override {
    std::cout << "B int" << std::endl;
  }
};

int main() {
  A* a = new B;
  a->size(2.2); // will display "B double"
  a->size(2); // will display "B int"

  return 0;
}
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

endif::env-github,rspecator-view[]
