== Why is this an issue?

In general, ``++async++`` ``++void++`` test methods are not executed by test frameworks, therefore it's better to avoid them altogether.


=== Noncompliant code example

[source,csharp]
----
    [TestMethod]
    public async void MyIgnoredTestMethod()  // Noncompliant
    { /* ... */ }
----


ifdef::env-github,rspecator-view[]
'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
