In general, ``++async++`` ``++void++`` test methods are not executed by test frameworks, therefore it's better to avoid them altogether.


== Noncompliant Code Example

----
    [TestMethod]
    public async void MyIgnoredTestMethod()  // Noncompliant
    { /* ... */ }
----


ifdef::env-github,rspecator-view[]
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
