<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This issue indicates that an exception will be thrown because a method is used incorrectly.
To fix it, check the requirements of the method and fulfill them.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>It is common for methods to check the value of their parameters or the state of
their associated object and throw an exception when one of them does not match a
given condition.
Those conditions are usually mentioned in the javadoc of the method.</p>
</div>
<div class="paragraph">
<p>This rule raises an issue when it detects that a method call will trigger one of
the following exceptions:</p>
</div>
<div class="ulist">
<ul>
<li>
<p><code>java.lang.IllegalArgumentException</code></p>
</li>
<li>
<p><code>java.lang.IllegalStateException</code></p>
</li>
</ul>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="paragraph">
<p>Issues of this type interrupt the normal execution of a program, causing it to
crash or putting it into an inconsistent state.
Therefore, this issue might impact the availability and reliability of your
application, or even result in data loss.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">/**
 * Set the oven temperature
 * @param temp the temperature in Celsius, between 0 and 250 (inclusive)
 * @throws IllegalArgumentException if the temperature is outside of the supported range
 */
private void setOvenTemperature(int temp) {
  if (temp &lt; 0 || temp &gt; 250) {
    throw new IllegalArgumentException();
  }
  // ...
}

void finishCooking() {
  setOvenTemperature(-3); // Noncompliant
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">/**
 * Set the oven temperature
 * @param temp the temperature in Celsius, between 0 and 250 (inclusive)
 * @throws IllegalArgumentException if the temperature is outside of the supported range
 */
private void setOvenTemperature(int temp) {
  if (temp &lt; 0 || temp &gt; 250) {
    throw new IllegalArgumentException();
  }
  // ...
}

void finishCooking() {
  setOvenTemperature(0);
}</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p><a href="https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/lang/IllegalArgumentException.html">IllegalArgumentException</a></p>
</li>
<li>
<p><a href="https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/lang/IllegalStateException.html">IllegalStateException</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>