Merging collapsible ``++IF++`` statements increases the code's readability.

== Noncompliant Code Example

----
       IF CONDITION1 THEN
         IF CONDITION2 THEN
           ...
         END-IF
       END-IF.
----

== Compliant Solution

----
       IF CONDITION1 AND CONDITION2 THEN
         ...
       END-IF.
----

ifdef::env-github,rspecator-view[]
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
