== Why is this an issue?

Merging collapsible ``++IF++`` statements increases the code's readability.

=== Noncompliant code example

[source,cobol]
----
       IF CONDITION1 THEN
         IF CONDITION2 THEN
           ...
         END-IF
       END-IF.
----

=== Compliant solution

[source,cobol]
----
       IF CONDITION1 AND CONDITION2 THEN
         ...
       END-IF.
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../highlighting.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
