<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>The <code>IEquatable&lt;T&gt;</code> interface has only one method in it: <code>Equals(&lt;T&gt;)</code>. If you&#8217;ve already written <code>Equals(T)</code>, there&#8217;s no reason not to explicitly implement <code>IEquatable&lt;T&gt;</code>. Doing so expands the utility of your class by allowing it to be used where an <code>IEquatable</code> is called for.</p>
</div>
<div class="paragraph">
<p><strong>Note</strong>: Classes that implement <code>IEquatable&lt;T&gt;</code> should also be <code>sealed</code>.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-csharp" data-lang="csharp">class MyClass  // Noncompliant
{
  public bool Equals(MyClass other)
  {
    //...
  }
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-csharp" data-lang="csharp">sealed class MyClass : IEquatable&lt;MyClass&gt;
{
  public override bool Equals(object other)
  {
    return Equals(other as MyClass);
  }

  public bool Equals(MyClass other)
  {
    //...
  }
}</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Implement 'IEquatable&lt;XXX&gt;'.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="ulist">
<ul>
<li>
<p>primary: class declaration</p>
</li>
<li>
<p>secondary: Equals signature</p>
</li>
</ul>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_6_mar_2017_181237_ann_campbell_wrote">on 6 Mar 2017, 18:12:37 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>Updated as discussed [~amaury.leve].  Please double-check me.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_7_mar_2017_100209_amaury_levé_wrote">on 7 Mar 2017, 10:02:09 Amaury Levé wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] Done!</p>
</div>
</div>
</div>
</div>