<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Having a duplication between a field name and a method name is confusing, misleading, and probably an indication that the field name should be updated to be more meaningful.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-text" data-lang="text">public class Foo {
  public int sum;         // Noncompliant, matching sum() method name
  public int sum() {...}
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-text" data-lang="text">public class Foo {
  public int sumTotal;    // Compliant
  public int sum() {...}
}</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_20_aug_2013_204831_freddy_mallet_wrote">on 20 Aug 2013, 20:48:31 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>Is implemented by http://jira.codehaus.org/browse/SONARJAVA-313</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_21_aug_2013_001032_ann_campbell_wrote">on 21 Aug 2013, 00:10:32 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>This description says that in the case of a member name matching a method name, the member should be renamed. However, the non-compliant example shows a member that name-clashes with a private method.</p>
</div>
<div class="paragraph">
<p>In this case, I would be tempted to rename the method - particularly because it&#8217;s private. Should the method in the examples be made public?</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_21_aug_2013_063714_freddy_mallet_wrote">on 21 Aug 2013, 06:37:14 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>Examples updated to make methods public</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_11_oct_2013_152503_dinesh_bolkensteyn_wrote">on 11 Oct 2013, 15:25:03 Dinesh Bolkensteyn wrote:</h3>
<div class="paragraph">
<p>This rule is not valuable as it will trigger lots of violations on fluent APIs:</p>
</div>
<div class="listingblock">
<div class="content">
<pre>class ClassTree {
  private final String name;  // There really is no problem here

  public String name() {      // There really is no problem here
    return name;
  }
}</pre>
</div>
</div>
<div class="paragraph">
<p>Class names and fields/method should never collide because of naming conventions. (although they might collide when compared case insensitively)</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_16_jun_2014_193926_ann_campbell_wrote">on 16 Jun 2014, 19:39:26 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~freddy.mallet] You closed this RSpec "Won&#8217;t Fix" last Friday, 2 minutes before you marked the implementing PHP ticket "manually tested"</p>
</div>
<div class="paragraph">
<p>Surely you closed this in error&#8230;&#8203;?</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_16_jun_2014_194026_ann_campbell_wrote">on 16 Jun 2014, 19:40:26 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~freddy.mallet] never mind.</p>
</div>
</div>
</div>
</div>