== Why is this an issue?

Theoretically, ``++CASE++``, ``++COALESCE++``, and ``++IIF++`` evaluate conditions sequentially, stopping with the first one that evaluates to ``++TRUE++``. In reality, these expressions evaluate aggregate functions and non-native service calls such as ``++CONTAIN++`` and ``++FREETEXT++`` first, and _then_ pass the results into the expression. That means that if you're relying on short-circuit behavior to avoid runtime errors with these arguments, you will not get it. You can work around the issue by wrapping such calls in a subselect.


=== Noncompliant code example

[source,sql]
----
SELECT 
  CASE 
    WHEN @i = 0 
      THEN 1 
    ELSE MIN(1/@i)  -- Noncompliant
  END; 
----


=== Compliant solution

[source,sql]
----
SELECT 
  CASE 
    WHEN @i = 0 
      THEN 1 
    ELSE (SELECT MIN(1/@i))
  END; 
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

include::highlighting.adoc[]

'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
