== Why is this an issue?

String data types (``++char++``, ``++varchar++``, ``++nchar++``, ``++nvarchar++``) default to a size of 1 if no size is specified in the declaration. For ``++char++`` and ``++nchar++`` this is confusing at best, but it is most probably a mistake for ``++varchar++`` and ``++nvarchar++``.


This rule raises an issue when no size is specified for ``++varchar++`` or ``++nvarchar++``.


=== Noncompliant code example

[source,sql]
----
DECLARE @myStr varchar;  -- Noncompliant
----


=== Compliant solution

[source,sql]
----
DECLARE @myStr varchar(255);
----



ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

include::highlighting.adoc[]

'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
