== Why is this an issue?

String data types (``++char++``, ``++varchar++``, ``++nchar++``, ``++nvarchar++``) default to a size of 1 if no size is specified in the declaration. For ``++char++`` and ``++nchar++`` this is confusing at best, but it is most probably a mistake for ``++varchar++`` and ``++nvarchar++``.


This rule raises an issue when no size is specified for ``++varchar++`` or ``++nvarchar++``.


=== Noncompliant code example

[source,sql]
----
DECLARE @myStr varchar;  -- Noncompliant
----


=== Compliant solution

[source,sql]
----
DECLARE @myStr varchar(255);
----



ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Add a size to this [variable|parameter] declaration


=== Highlighting

variable type


'''
== Comments And Links
(visible only on this page)

=== on 6 Jul 2017, 10:28:01 Pierre-Yves Nicolas wrote:
\[~ann.campbell.2] I don't think the rule should cover ``++char++`` and ``++nchar++``. I think it would be quite noisy.

=== on 6 Jul 2017, 11:00:55 Pierre-Yves Nicolas wrote:
\[~ann.campbell.2] Should we update the rule title?

endif::env-github,rspecator-view[]
