== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

With an indent size of 2:

[source,abap]
----
CLASS counter IMPLEMENTATION.
  METHOD set.
  count = set_value.   " Noncompliant, expected to start at column 4
    ENDMETHOD.       " Noncompliant, expected to start at column 2
  METHOD increment.
    ADD 1 TO count.
  ENDMETHOD.
  METHOD get.
    get_value = count.
ENDMETHOD.           " Noncompliant, expected to start at column 2
ENDCLASS.
----

=== Compliant solution

[source,abap]
----
CLASS counter IMPLEMENTATION.
  METHOD set.
    count = set_value. 
  ENDMETHOD.  
  METHOD increment.
    ADD 1 TO count.
  ENDMETHOD.
  METHOD get.
    get_value = count.
  ENDMETHOD.      
ENDCLASS.
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
