<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>A <code>FETCH</code> statement fails when the number of variables does not match the number of columns selected in the CURSOR definition.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-sql" data-lang="sql">DECLARE c1 cursor FOR SELECT FirstName, LastName FROM customer;
OPEN c1;
FETCH NEXT FROM c1 INTO @Name; -- Noncompliant</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-sql" data-lang="sql">DECLARE c1 cursor FOR SELECT FirstName, LastName FROM customer;
OPEN c1;
FETCH NEXT FROM c1 INTO @FirstName, @LastName;</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Refactor this <code>FETCH</code> to select the same number of columns selected in "xxx".</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="ulist">
<ul>
<li>
<p>primary: <code>INTO</code> + column list in <code>FETCH</code> statement</p>
</li>
<li>
<p>secondary: <code>SELECT</code> + column list in cursor</p>
</li>
</ul>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_is_related_to_s3613">is related to: <a data-rspec-id="S3613" class="rspec-auto-link">S3613</a></h3>

</div>
<div class="sect2">
<h3 id="_is_related_to_s3614">is related to: <a data-rspec-id="S3614" class="rspec-auto-link">S3614</a></h3>

</div>
<div class="sect2">
<h3 id="_on_20_jul_2017_152619_pierre_yves_nicolas_wrote">on 20 Jul 2017, 15:26:19 Pierre-Yves Nicolas wrote:</h3>
<div class="paragraph">
<p>According to the Microsoft <a href="https://docs.microsoft.com/en-us/sql/t-sql/language-elements/fetch-transact-sql">reference for FETCH</a>:</p>
</div>
<div class="quoteblock">
<blockquote>
<div class="paragraph">
<p>The number of variables must match the number of columns in the cursor select list.</p>
</div>
</blockquote>
</div>
<div class="paragraph">
<p>The code can be compiled with no issue, for example in a stored procedure. The following error is raised at runtime:</p>
</div>
<div class="listingblock">
<div class="content">
<pre>Msg 16924, Level 16, State 1, Line 61
Cursorfetch: The number of variables declared in the INTO list must match that of selected columns.</pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_on_7_aug_2017_150324_pierre_yves_nicolas_wrote">on 7 Aug 2017, 15:03:24 Pierre-Yves Nicolas wrote:</h3>
<div class="paragraph">
<p>\[~massimo.paladin] Why minor? Such code triggers a failure at execution time.</p>
</div>
</div>
</div>
</div>