:entry-point: static void Main

include::../why.adoc[]

=== Noncompliant code example

[source,csharp]
----
using System;
using System.Windows.Forms;

namespace WinFormsApp
{
    static class Program
    {
        private static void Main() // Noncompliant: the [STAThread] is missing
        {
            ApplicationConfiguration.Initialize();
            Application.Run(new Form1());
        }
    }
}
----


=== Compliant solution

[source,csharp]
----
using System;
using System.Windows.Forms;

namespace WinFormsApp
{
    static class Program
    {
        [STAThread]
        private static void Main() // Compliant: the [STAThread] is present
        {
            ApplicationConfiguration.Initialize();
            Application.Run(new Form1());
        }
    }
}
----

include::../rspecator.adoc[]
