<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p><code>@Configuration</code> is a class-level annotation indicating that an object is a source of bean definitions.
<code>@Configuration</code> classes declare beans through <code>@Bean</code>-annotated methods.
Calls to <code>@Bean</code> methods on <code>@Configuration</code> classes can also be used to define inter-bean dependencies.
The <code>@Bean</code> annotation indicates that a method instantiates, configures, and initializes a new object to be managed by the Spring IoC container.</p>
</div>
<div class="paragraph">
<p>Annotating a method of a bean with <code>@Async</code> will make it execute in a separate thread.
In other words, the caller will not wait for the completion of the called method.</p>
</div>
<div class="paragraph">
<p>The <code>@Async</code> annotation is not supported on methods declared within a <code>@Configuration</code> class.
This is because <code>@Async</code> methods are typically used for asynchronous processing, and they require certain infrastructure to be set up, which may not be available or appropriate in a <code>@Configuration</code> class.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Don&#8217;t use <code>@Async</code> annotations on methods of <code>@Configuration</code> classes.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">@EnableAsync
@Configuration
public class MyConfiguration {

  @Async // Noncompliant - This is not allowed
  public void asyncMethod() {
    // ...
  }
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">@EnableAsync
@Configuration
public class MyConfiguration {

  public void method() {
    // ...
  }
}</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p><a href="https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/scheduling/annotation/Async.html">Spring Framework - @Async</a></p>
</li>
<li>
<p><a href="https://docs.spring.io/spring-framework/reference/core/beans/java/configuration-annotation.html">Spring Framework - Using the @Configuration annotation</a></p>
</li>
<li>
<p><a href="https://docs.spring.io/spring-framework/reference/core/beans/java/basic-concepts.html">Spring Framework - Basic Concepts: @Bean and @Configuration</a></p>
</li>
</ul>
</div>
</div>
<div class="sect2">
<h3 id="_articles_blog_posts">Articles &amp; blog posts</h3>
<div class="ulist">
<ul>
<li>
<p><a href="https://www.baeldung.com/spring-async">Baeldung - How To Do @Async in Spring</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>