== Why is this an issue?

In C#, without constraints on a generic type parameter, both https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/reference-types[reference] and https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/builtin-types/value-types[value] types can be passed. However, comparing this type parameter to `null` can be misleading as value types, like `struct`, can never be null. To avoid this, use https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/operators/default#default-operator[default()] for comparison if you expect a value type. If you expect a reference type, add https://learn.microsoft.com/en-us/dotnet/csharp/programming-guide/generics/constraints-on-type-parameters[constraints] to prevent value types from being passed.

== How to fix it

Use the default operator `default()` to do the comparison or add a constraint to the generic type.

=== Code examples

==== Noncompliant code example

[source,csharp]
----
bool IsDefault<T>(T value)
{
  if (value == null) // Noncompliant
  {
    // ...
  }
  // ...
}
----


==== Compliant solution

[source,csharp]
----
bool IsDefault<T>(T value)
{
  if(object.Equals(value, default(T)))
  {
    // ...
  }
  // ...
}
----

or

[source,csharp]
----
bool IsDefault<T>(T value) where T : class
{
  if (value == null)
  {
    // ...
  }
  // ...
}
----

== Resources

=== Documentation

* Microsoft Learn - https://learn.microsoft.com/en-us/dotnet/csharp/programming-guide/generics/constraints-on-type-parameters[Constraints on type parameters]
* Microsoft Learn - https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/reference-types[Reference types]
* Microsoft Learn - https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/builtin-types/value-types[Value types]
* Microsoft Learn - https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/operators/default#default-operator["`default` operator"]

include::rspecator.adoc[]
