== Why is this an issue?

Most developers expect https://learn.microsoft.com/en-us/dotnet/visual-basic/language-reference/properties[property] access to be as efficient as https://learn.microsoft.com/en-us/dotnet/visual-basic/programming-guide/language-features/objects-and-classes/#fields-and-properties[field] access. However, if a property returns a copy of an array or https://learn.microsoft.com/en-us/dotnet/visual-basic/programming-guide/concepts/collections[collection], it will be much slower than simple field access, contrary to the caller's likely expectations. Therefore, such properties should be refactored into methods so that callers are not surprised by the unexpectedly slower performance.

== How to fix it

=== Code examples

==== Noncompliant code example

[source,vbnet,diff-id=1,diff-type=noncompliant]
----
Module SomeModule
    Dim foo = {"apple", "banana", "orange", "pineapple", "strawberry"}

    ReadOnly Property Foo() As String() ' Noncompliant
        Get
            Dim copy = foo.Clone      ' Expensive call
            Return copy
        End Get
    End Property
End Module
----

==== Compliant solution

[source,vbnet,diff-id=1,diff-type=compliant]
----
Module SomeModule
    Dim foo = {"apple", "banana", "orange", "pineapple", "strawberry"}

    Function GetFoo() As String()
        Dim copy = foo.Clone
        Return copy
    End Function
End Module
----

== Resources

=== Documentation

* https://learn.microsoft.com/en-us/dotnet/visual-basic/language-reference/properties[Properties (Visual Basic)]
* https://learn.microsoft.com/en-us/dotnet/visual-basic/programming-guide/language-features/objects-and-classes/#fields-and-properties[Fields and properties]
* https://learn.microsoft.com/en-us/dotnet/visual-basic/programming-guide/concepts/collections[Collections (Visual Basic)]

include::../rspecator-dotnet.adoc[]