== How to fix it in Swift Crypto

=== Code examples

include::../../common/fix/code-rationale.adoc[]

==== Noncompliant code example

include::../../common/fix/rsa.adoc[]

[source,swift,diff-id=21,diff-type=noncompliant]
----
import Crypto
import _CryptoExtras

do {
    let privateKey = try _RSA.Encryption.PrivateKey(keySize: _RSA.Signing.KeySize(bitCount: 1024))  // Noncompliant
} catch {
    // ...
}
----

==== Compliant solution

[source,swift,diff-id=21,diff-type=compliant]
----
import Crypto
import _CryptoExtras

do {
    let privateKey = try _RSA.Encryption.PrivateKey(keySize: .bits2048)
} catch {
    // ...
}
----

=== How does this work?

include::../../common/fix/fix.adoc[]

=== Going the extra mile

include::../../common/extra-mile/pre-quantum.adoc[]
