<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when a controller includes a helper module or directly calls view helper methods.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>In the Model-View-Controller (MVC) architecture, each component has a specific responsibility. Controllers handle request and response logic, models manage data and business logic, and views handle presentation.</p>
</div>
<div class="paragraph">
<p>View helpers are designed specifically for formatting and presenting data in templates. When controllers include helper modules or call helper methods directly, they violate the separation of concerns principle.</p>
</div>
<div class="paragraph">
<p>This creates several problems:</p>
</div>
<div class="ulist">
<ul>
<li>
<p><strong>Tight coupling</strong>: The controller becomes dependent on view-specific formatting logic</p>
</li>
<li>
<p><strong>Reduced testability</strong>: Controller tests now need to account for view presentation logic</p>
</li>
<li>
<p><strong>Code organization</strong>: Business logic gets mixed with presentation logic, making the codebase harder to maintain</p>
</li>
<li>
<p><strong>Reusability</strong>: Formatting logic tied to controllers cannot be easily reused in other contexts</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>Controllers should focus on coordinating between models and views, not on how data should be formatted for display.</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="paragraph">
<p>This architectural violation can lead to tightly coupled code that is harder to test, maintain, and refactor. It makes the application more fragile to changes and reduces code reusability.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it_in_ruby_on_rails">How to fix it in Ruby on Rails</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Remove the helper include from the controller and move the formatting logic to a model method or service object.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-ruby" data-lang="ruby">class CommentsController &lt; ApplicationController
  include ApplicationHelper # Noncompliant

  def index
    @comments = []
    Comment.find_each do |comment|
      @comments &lt;&lt; {
        id: comment.id,
        content: html_format(comment.content) # Noncompliant
      }
    end
    render json: @comments
  end
end</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-ruby" data-lang="ruby"># Move formatting logic to a service object
class CommentSerializer
  def self.format_for_json(comments)
    comments.map do |comment|
      {
        id: comment.id,
        content: comment.formatted_content
      }
    end
  end
end

class CommentsController &lt; ApplicationController
  def index
    @comments = Comment.all
    render json: CommentSerializer.format_for_json(@comments)
  end
end</code></pre>
</div>
</div>
<div class="paragraph">
<p>Instead of using the helpers object to access view methods, create a dedicated presenter or move the logic to the model.</p>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_code_examples_2">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example_2">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-ruby" data-lang="ruby">class UsersController &lt; ApplicationController
  def show
    @user = User.find(params[:id])
    @formatted_name = helpers.truncate(@user.full_name, length: 20) # Noncompliant
  end
end</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution_2">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-ruby" data-lang="ruby">class User &lt; ApplicationRecord
  def display_name(length: 20)
    full_name.length &gt; length ? "#{full_name[0...length]}..." : full_name
  end
end

class UsersController &lt; ApplicationController
  def show
    @user = User.find(params[:id])
    @formatted_name = @user.display_name
  end
end</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Rails Guides - Action Controller Overview - <a href="https://guides.rubyonrails.org/action_controller_overview.html">Official Rails documentation on controller responsibilities and best practices</a></p>
</li>
<li>
<p>Rails Guides - Action View Helpers - <a href="https://guides.rubyonrails.org/action_view_helpers.html">Documentation on Rails view helpers and their intended use in templates</a></p>
</li>
</ul>
</div>
</div>
<div class="sect2">
<h3 id="_standards">Standards</h3>
<div class="ulist">
<ul>
<li>
<p>SOLID Principles - Single Responsibility Principle - <a href="https://en.wikipedia.org/wiki/Single-responsibility_principle">Each class should have only one reason to change and one responsibility</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>