According to the IBM documentation,


____
The Enterprise COBOL compiler now uses the Language Environment STACK storage work area for several functions, including ``++UPPER-CASE++``, ``++LOWER-CASE++``, and ``++NATIONAL-OF++``. If the STACK is allocated below the 16-MB line and a large DSA (Dynamic Save Area) is needed, an insufficient storage error might occur.
____


To aid in migrations to IBM Enterprise COBOL 5.x, this rule raises an issue when ``++UPPER-CASE++``, ``++LOWER-CASE++``, or ``++NATIONAL-OF++`` is used.


== Noncompliant Code Example

----
        MOVE FUNCTION UPPER-CASE(FIRST-NAME) to FIRST-NAME2.  *> Noncompliant    
        MOVE FUNCTION LOWER-CASE(FIRST-NAME) to FIRST-NAME2.  *> Noncompliant
----



ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

include::highlighting.adoc[]

'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
