== Why is this an issue?

According to the IBM documentation,


____
The Enterprise COBOL compiler now uses the Language Environment STACK storage work area for several functions, including ``++UPPER-CASE++``, ``++LOWER-CASE++``, and ``++NATIONAL-OF++``. If the STACK is allocated below the 16-MB line and a large DSA (Dynamic Save Area) is needed, an insufficient storage error might occur.
____


To aid in migrations to IBM Enterprise COBOL 5.x, this rule raises an issue when ``++UPPER-CASE++``, ``++LOWER-CASE++``, or ``++NATIONAL-OF++`` is used.


=== Noncompliant code example

[source,cobol]
----
        MOVE FUNCTION UPPER-CASE(FIRST-NAME) to FIRST-NAME2.  *> Noncompliant    
        MOVE FUNCTION LOWER-CASE(FIRST-NAME) to FIRST-NAME2.  *> Noncompliant
----



ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Remove this use of "xxx".


=== Highlighting

function name


'''
== Comments And Links
(visible only on this page)

=== on 1 Dec 2015, 08:40:00 Pierre-Yves Nicolas wrote:
According to an http://www-01.ibm.com/support/knowledgecenter/SS6SG3_3.4.0/com.ibm.entcobol.doc_3.4/igym1370.htm[IBM documentation], there may be an issue when switching from IBM COBOL to any version of IBM *Enterprise* COBOL: this documentation is about "Enterprise COBOL for z/OS 3.4.0" and the page starts with "This chapter describes the differences between the IBM COBOL language and the Enterprise COBOL language".

=== on 1 Dec 2015, 15:22:46 Ann Campbell wrote:
I added the word 'Enterprise' [~pierre-yves.nicolas]. S'okay?

=== on 1 Dec 2015, 16:11:32 Pierre-Yves Nicolas wrote:
Ok for me [~ann.campbell.2].

endif::env-github,rspecator-view[]
