== Why is this an issue?

Assigning default values to variables in an initialization section is a good practice as it increases the readability of the code: when starting reading a block, one will know exactly which variables have default values.


=== Noncompliant code example

[source,sql]
----
DECLARE
   var1 integer;
   var2 integer;
   var3 integer;
BEGIN
   var1 := 1; -- Noncompliant
   var2 := 2; -- Noncompliant
   var3 := var1 + var2;
END;
/
----


=== Compliant solution

[source,sql]
----
DECLARE
   var1 integer := 1;
   var2 integer := 2;
   var3 integer;
BEGIN
   var3 := var1 + var2;
END;
/
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Move this devault value assignment in the initialization section


endif::env-github,rspecator-view[]
