The integrator function can pass data downstream using the push method
In return, the stream should return a result indicating whether it accepted the latest value.

Ignoring the value returned by `Downstream.push` might lead to inconsistent behavior of the integrator.

== Why is this an issue?

FIXME: remove the unused optional headers (that are commented out)

//=== What is the potential impact?

== How to fix it
//== How to fix it in FRAMEWORK NAME

=== Code examples

==== Noncompliant code example

[source,java,diff-id=1,diff-type=noncompliant]
----
Gatherer.ofSequential(
  () -> new AtomicInteger(100),
  (state, element, downstream) -> {
      int newState = state.addAndGet(element);
      downstream.push(newState); // Noncompliant: ignoring return value
      return true;
  });
----

==== Compliant solution

[source,java,diff-id=1,diff-type=compliant]
----
Gatherer.ofSequential(
  () -> new AtomicInteger(100),
  (state, element, downstream) -> {
      int newState = state.addAndGet(element);
      return downstream.push(newState); // Compliant
  });
----

[source,java,diff-id=1,diff-type=compliant]
----
Gatherer.ofSequential(
  () -> new AtomicInteger(100),
  (state, element, downstream) -> {
      int newState = state.addAndGet(element);
      if (downstream.push(newState)) {
        // Do something to handle this specific case
        return false;
      }
      return true;
  });
----

//=== How does this work?

//=== Pitfalls

//=== Going the extra mile


//== Resources
//=== Documentation
//=== Articles & blog posts
//=== Conference presentations
//=== Standards
//=== External coding guidelines
//=== Benchmarks
