== Why is this an issue?

By convention, the name of a file should match the name of the top-level class it contains. Anything else may confuse maintainers.


=== Noncompliant code example

[source,text]
----
// file path: Gui/Interface.cs  -- Noncompliant
namespace Gui
{
  class Screen
  {
  }
}
----


=== Compliant solution

[source,text]
----
// file path: Gui/Screen.cs
namespace Gui
{
  class Screen
  {
  }
}
----

