include::../why-dotnet.adoc[]

=== Noncompliant code example

[source,csharp,diff-id=1,diff-type=noncompliant]
----
internal class MyException : Exception   // Noncompliant
{
  // ...
}
----

=== Compliant solution

[source,csharp,diff-id=1,diff-type=compliant]
----
public class MyException : Exception
{
  // ...
}
----

include::../resources-dotnet.adoc[]

* https://learn.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/access-modifiers[Access modifiers]

include::../rspecator.adoc[]
