include::../rule.adoc[]

== Noncompliant Code Example

[source,csharp]
----
internal class MyException : Exception   // Noncompliant
{
  // ...
}
----


== Compliant Solution

[source,csharp]
----
public class MyException : Exception
{
  // ...
}
----


include::../exceptions.adoc[]
include::../see.adoc[]
include::../rspecator.adoc[]