The point of having custom exception types is to convey more information than is available in standard types. But custom exception types must be `public` for that to work.

If a method throws a non-public exception, the best you can do on the caller's side is to `catch` the closest `public` base of the class. That is, you lose all that custom information you created the exception type to pass.

== Noncompliant Code Example

[source,csharp]
----
internal class MyException : Exception   // Noncompliant
{
  // ...
}
----


== Compliant Solution

[source,csharp]
----
public class MyException : Exception
{
  // ...
}
----


include::../exceptions.adoc[]
include::../see.adoc[]
include::../rspecator.adoc[]