<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>It is more efficient to skip the function invocation and use the predefined string constants rather than calling <code>Chr$()</code> or <code>ChrW$()</code> for the following numbers:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>0 - <code>vbNullChar</code></p>
</li>
<li>
<p>8 - <code>vbBack</code></p>
</li>
<li>
<p>9 - <code>vbTab</code></p>
</li>
<li>
<p>10 - <code>vbLf</code></p>
</li>
<li>
<p>11 - <code>vbVerticalTab</code></p>
</li>
<li>
<p>12 - <code>vbFormFeed</code></p>
</li>
<li>
<p>13 - <code>vbCr</code></p>
</li>
<li>
<p>13+10 - <code>vbCrLf</code> | <code>vbNewLine</code></p>
</li>
<li>
<p>34 - <code>""</code></p>
</li>
</ul>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-vb6" data-lang="vb6">Dim myChar As Char
myChar = Chr$(0)</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-vb6" data-lang="vb6">Dim myChar As Char
myChar = vbNullChar</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Use the predefined constant XXX instead</p>
</div>
</div>
</div>
</div>