<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when <code>Jenkins.instance.getItem()</code> is called with a path that contains folder separators (forward slashes).</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>The <code>getItem()</code> method in Jenkins only works for jobs at the root level. When you pass a folder path like <code>'folder_name/job_name'</code> to <code>getItem()</code>, it returns <code>null</code> instead of the expected job object.</p>
</div>
<div class="paragraph">
<p>This happens because <code>getItem()</code> treats the entire string as a single job name, not as a path. It doesn&#8217;t understand folder hierarchies and won&#8217;t traverse into folders to find the job.</p>
</div>
<div class="paragraph">
<p>When <code>getItem()</code> returns <code>null</code> for a job that actually exists in a folder, your script may:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>Throw a null pointer exception when trying to access properties of the job</p>
</li>
<li>
<p>Skip important operations because it thinks the job doesn&#8217;t exist</p>
</li>
<li>
<p>Fail silently, making debugging difficult</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>Jenkins provides <code>getItemByFullName()</code> specifically for accessing items using their full path, including folders. This method correctly handles folder hierarchies and returns the job object when it exists.</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="paragraph">
<p>Scripts may fail with null pointer exceptions or behave unexpectedly when they cannot find jobs that exist in folders. This can lead to incomplete automation, failed deployments, or silent failures that are difficult to debug.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Replace <code>getItem()</code> with <code>getItemByFullName()</code> when accessing jobs in folders. The <code>getItemByFullName()</code> method correctly handles folder paths and will return the job object if it exists.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-groovy" data-lang="groovy">// This returns null for jobs in folders
def job = Jenkins.instance.getItem('folder_name/job_name') // Noncompliant
if (job != null) {
    echo "Job found: ${job.name}"
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-groovy" data-lang="groovy">// Use getItemByFullName for jobs in folders
def job = Jenkins.instance.getItemByFullName('folder_name/job_name')
if (job != null) {
    echo "Job found: ${job.name}"
}</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Jenkins API Documentation - getItemByFullName - <a href="https://javadoc.jenkins.io/jenkins/model/Jenkins.html#getItemByFullName-java.lang.String-">Official Jenkins API documentation for the getItemByFullName method</a></p>
</li>
<li>
<p>Jenkins Script Console Documentation - <a href="https://www.jenkins.io/doc/book/managing/script-console/">Guide to using the Jenkins script console for automation</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>