include::../rationale.adoc[]

include::../how-to-fix-it.adoc[]

=== Code examples

==== Noncompliant code example

[source,sql,diff-id=1,diff-type=noncompliant]
----
declare
  seconds integer := 0; -- Noncompliant - seconds is unused
  hours integer := 2;
begin
  minutes := hours * 60;
  dbms_output.put_line('Number of minutes: ' || minutes);
end;
----


==== Compliant solution

[source,sql,diff-id=1,diff-type=compliant]
----
declare
  hours integer := 2;
begin
  minutes := hours * 60;
  dbms_output.put_line('Number of minutes: ' || minutes);
end;
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Remove the unused local constant/exception/variable "xxx".


'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
