Declaring a variable without specifying its data type leaves the compiler to assign the type that seems the most appropriate - whether it's what you need or not. Therefore you should always specify the data type.

== Noncompliant Code Example

----
Const MyVar = "459"
'...
Dim NumberOfEmployees
----

== Compliant Solution

----
Const MyVar as String = "459"
'...
Dim NumberOfEmployees As Integer
----
