== Why is this an issue?

Declaring a variable without specifying its data type leaves the compiler to assign the type that seems the most appropriate - whether it's what you need or not. Therefore you should always specify the data type.


=== Noncompliant code example

[source,vb6]
----
Const MyVar = "459"
'...
Dim NumberOfEmployees
----


=== Compliant solution

[source,vb6]
----
Const MyVar as String = "459"
'...
Dim NumberOfEmployees As Integer
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

endif::env-github,rspecator-view[]
