<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>"Monster Class" is a class that implements too many functionalities that are not well organized, or cohesive. Monster Classes are not created intentionally, but rather they often arise over time as a result of poor software design or a lack of attention to good programming practices such as modularization and encapsulation.
As a software evolves and new requirements are added, it can be tempting to add new functionality to existing classes rather than creating new classes to handle the new functionality. This leads to classes becoming bloated and difficult to understand and maintain, ultimately resulting in a Monster Class.</p>
</div>
<div class="paragraph">
<p>Here are the characteristics of a Monster Class:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>High number of instance variables and methods</p>
</li>
<li>
<p>Variables and methods that are not related to each other (low cohesion and high coupling)</p>
</li>
<li>
<p>Too many responsibilities, violating the Single Responsibility Principle (aka SRP)</p>
</li>
<li>
<p>Difficult to test, leading to poor test coverage and a higher risk of defects</p>
</li>
</ul>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Monster Classes become monolithic entities, with numerous responsibilities and functionalities packed into a single class. This is problematic because it violates the Single Responsibility Principle, which states that a class should have only one reason to change.</p>
</div>
<div class="paragraph">
<p>When a class has too many responsibilities and functionalities, it becomes difficult to maintain. Changes to one part of the class can unintentionally affect other parts, leading to bugs. Additionally, it can be difficult to test the class, as there may be many different interactions between different parts of the class that need to be considered.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>There is no magical recipe. Monster Classes should be refactored and broken into smaller, more focused classes, each one with a single responsibility.
When a class has only one responsibility, it is easier to reason about its behavior and to make changes to it without affecting other parts of the code.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="paragraph">
<p>With a threshold of 5:</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">class Foo { // class Foo depends on too many classes: T1, T2, T3, T4, T5, T6 and T7
  T1 t1;
  T2 t2;
  T3 t3;

  public T4 compute(T5 a, T6 b) {
    T7 result = a.getResult(b);
    return (T4) result;
  }
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">public class Bar {
    T8 a8;
    T9 a9;
}</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_articles_blog_posts">Articles &amp; blog posts</h3>
<div class="ulist">
<ul>
<li>
<p><a href="https://blog.cleancoder.com/uncle-bob/2014/05/08/SingleReponsibilityPrinciple.html">Single Responsibility Principle (aka SRP)</a></p>
</li>
<li>
<p><a href="https://en.wikipedia.org/wiki/SOLID">SOLID</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>