<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>There is no good reason to create a new object to not do anything with it. Most of the time, this is due to a missing piece of code and so could lead to an unexpected behavior in production.</p>
</div>
<div class="paragraph">
<p>If it was done on purpose because the constructor has side-effects, then that side-effect code should be moved into a separate method and called directly.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">if (x &lt; 0)
  new IllegalArgumentException("x must be nonnegative");</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">if (x &lt; 0)
  throw new IllegalArgumentException("x must be nonnegative");</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Either remove this useless object instantiation of class "XXXX" or use it</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_is_related_to_s905">is related to: <a data-rspec-id="S905" class="rspec-auto-link">S905</a></h3>

</div>
<div class="sect2">
<h3 id="_on_21_aug_2015_152029_freddy_mallet_wrote">on 21 Aug 2015, 15:20:29 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>This RSPEC should be part of <a data-rspec-id="S905" class="rspec-auto-link">RSPEC-905</a> but as this seems to be impossible to not generate some false-positives, we prefer to keep it isolated.</p>
</div>
</div>
</div>
</div>