include::../description.adoc[]

== Noncompliant Code Example

A ``++CryptoObject++``, used to verify the integrity of the authentication, is not used: 

----
// ...
val biometricPrompt: BiometricPrompt = BiometricPrompt(activity, executor, callback)
// ...
biometricPrompt.authenticate(promptInfo) // Noncompliant
----

== Compliant Solution

A ``++CryptoObject++``, used to verify the integrity of the authentication, is used: 

----
// ...
val biometricPrompt: BiometricPrompt = BiometricPrompt(activity, executor, callback)
// ...
biometricPrompt.authenticate(promptInfo, BiometricPrompt.CryptoObject(cipher)) // Compliant

----

include::../see.adoc[]