include::../description.adoc[]

== Noncompliant Code Example

[source,flex]
----
public class Greeter {
  public static var foo:Foo = new Foo(...);
  ...
}
----

== Compliant Solution

[source,flex]
----
public class Greeter {
  public static const FOO:Foo = new Foo(...);
  ...
}
----

include::../see.adoc[]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
