include::../description.adoc[]

== Noncompliant Code Example

----
DECLARE
  myvar CHAR;
BEGIN
  SELECT * INTO myvar FROM DUAL; -- Noncompliant
 END;
/
----

== Compliant Solution

----
DECLARE
  myvar CHAR;
BEGIN
  SELECT dummy INTO myvar FROM DUAL;
END;
/
----

== Exceptions

Wrapper queries using ``ROWNUM`` are ignored.

----
SELECT *
FROM ( SELECT fname, lname, deptId
    FROM employee
    ORDERBY salary
  )
WHERE rownum <= 10
----
