include::../description.adoc[]

== Noncompliant Code Example

[source,sql]
----
DECLARE
  myvar CHAR;
BEGIN
  SELECT * INTO myvar FROM DUAL; -- Noncompliant
 END;
/
----

== Compliant Solution

[source,sql]
----
DECLARE
  myvar CHAR;
BEGIN
  SELECT dummy INTO myvar FROM DUAL;
END;
/
----

== Exceptions

Wrapper queries using ``++ROWNUM++`` are ignored.

----
SELECT *
FROM ( SELECT fname, lname, deptId
    FROM employee
    ORDERBY salary
  )
WHERE rownum <= 10
----
ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

endif::env-github,rspecator-view[]
