<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when an incorrect pattern is provided to an <code>einops</code> operation.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>The <code>einops</code> library provides a powerful and flexible way to manipulate tensors using the Einstein summation convention.
The <code>einops</code> uses a different convention than the <a href="https://ejenner.com/post/einsum/">traditional</a> one.
In particular, the axis names can be more than one letter long and are separated by spaces.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Correct the syntax of the <code>einops</code> operation by balancing the parentheses and following the convention.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">from einops import rearrange
import torch

x = torch.randn(2, 3, 4, 5)
x2 = rearrange(x, 'b c h w -&gt; b (c h w') # Noncompliant : the parentheses are not balanced</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">from einops import rearrange
import torch

x = torch.randn(2, 3, 4, 5)
x2 = rearrange(x, 'b c h w -&gt; b (c h w)')</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p><code>einops</code> documentation - <a href="https://einops.rocks/1-einops-basics/#welcome-to-einops-land">Einops basics</a></p>
</li>
</ul>
</div>
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Implement a parser inspired by the original <code>einops</code> one : <a href="https://github.com/arogozhnikov/einops/blob/6181e1e95dc58c00a3143c1726da1c6ee0463164/einops/parsing.py#L85" class="bare">https://github.com/arogozhnikov/einops/blob/6181e1e95dc58c00a3143c1726da1c6ee0463164/einops/parsing.py#L85</a></p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Primary: The syntax of the <code>einops</code> operation is incorrect.</p>
</div>
</div>
<div class="sect2">
<h3 id="_issue_location">Issue location</h3>
<div class="paragraph">
<p>Primary : The pattern</p>
</div>
</div>
<div class="sect2">
<h3 id="_quickfix">Quickfix</h3>
<div class="paragraph">
<p>No</p>
</div>
</div>
</div>
</div>