<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>A listener can be attached to an object only after it has been constructed. So dispatching an event in a constructor is useless and error prone.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-flex" data-lang="flex">public class MyClass
{
  public function MyClass()
  {
    dispatchEvent( new Event( "uselessEvent" ) );
  }
}</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove this event dispatch from the "XXXX" constructor</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_22_nov_2013_122848_freddy_mallet_wrote">on 22 Nov 2013, 12:28:48 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>Is implemented by http://jira.codehaus.org/browse/SONARPLUGINS-3279</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_24_nov_2013_190422_ann_campbell_wrote">on 24 Nov 2013, 19:04:22 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>I&#8217;m tempted to add a Compliant Solution using [PostConstruct], but I&#8217;m afraid my 3-minute skim of Flex sites doesn&#8217;t adequately qualify me&#8230;&#8203;</p>
</div>
</div>
</div>
</div>