== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,text]
----
Enum FooFlags ' Noncompliant
    Foo = 1
    Bar = 2
    Baz = 4
End Enum
----


=== Compliant solution

[source,text]
----
Enum Foo      ' Compliant
    Foo = 1
    Bar = 2
    Baz = 4
End Enum
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
