include::../description-dotnet.adoc[]

=== Code examples

==== Noncompliant code example

[source,vbnet,diff-id=1,diff-type=noncompliant]
----
Public Class Example

    Private Shared rwLock As New ReaderWriterLock()

    Public Sub Writer()
        rwLock.AcquireWriterLock(2000)
        Try
            ' ...
        Finally
            rwLock.ReleaseReaderLock() ' Noncompliant, will throw runtime exception
        End Try
    End Sub

    Public Sub Reader()
        rwLock.AcquireReaderLock(2000)
        Try
            ' ...
        Finally
            rwLock.ReleaseWriterLock() ' Noncompliant, will throw runtime exception
        End Try
    End Sub

End Class
----

==== Compliant solution

[source,vbnet,diff-id=1,diff-type=compliant]
----
Public Class Example

    Private Shared rwLock As New ReaderWriterLock()

    Public Shared Sub Writer()
        rwLock.AcquireWriterLock(2000)
        Try
            ' ...
        Finally
            rwLock.ReleaseWriterLock()
        End Try
    End Sub

    Public Shared Sub Reader()
        rwLock.AcquireReaderLock(2000)
        Try
            ' ...
        Finally
            rwLock.ReleaseReaderLock()
        End Try
    End Sub
    
End Class
----

include::../resources-dotnet.adoc[]

include::../rspecator.adoc[]