<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Every subroutine should be commented to explain its goal and how it works. This non-empty comment must be located before or after the subroutine definition.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-rpg" data-lang="rpg">C     SR012         BEGSR
C                         ...
C                   ENDSR</code></pre>
</div>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-rpg" data-lang="rpg"> /free
  begsr SR012;
   ...
  endsr;
 /end-free</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-rpg" data-lang="rpg">C********************************************************************
C* SR012- Here is my comment
C********************************************************************
C     SR012         BEGSR
C                         ...
C                   ENDSR</code></pre>
</div>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-rpg" data-lang="rpg"> /free
  // SR012- Here is my comment
  begsr SR012;
   ...
  endsr;
 /end-free</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Document this subroutine.</p>
</div>
</div>
</div>
</div>