<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Octal escape sequences, when used in regular expressions, can easily be mistaken for backreferences. When the use of such sequence is intentional, it is generally better to replace them with Unicode or hexadecimal sequence to avoid any ambiguity.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Using octal escapes in regular expressions can create confusion with backreferences.
Octal escapes are sequences of digits that represent a character in the ASCII table, and they are sometimes used to represent special characters in regular expressions.
However, they can be easily mistaken for backreferences, which are also sequences of digits that represent previously captured groups. This confusion can lead to unexpected results or errors in the regular expression.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Instead of using octal escapes, it is recommended to use other ways to represent special characters in regular expressions. For example, you can use Unicode escape sequences, hexadecimal escape sequences or character classes. By using these alternatives, you can avoid the confusion with backreferences and improve the readability of your regular expressions.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">import re

match = re.match(r"\101", "A")</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">import re

match = re.match(r"\x41", "A")</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_message">Message</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Consider replacing this octal escape sequence with a Unicode or hexadecimal sequence instead.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_highlighting">Highlighting</h2>
<div class="sectionbody">
<div class="paragraph">
<p>The octal escape sequence</p>
</div>
<hr>
</div>
</div>