<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when the <code>tf.divide</code> function arguments have incompatible shapes.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>The <code>tf.divide</code> function in TensorFlow is used to perform element-wise division between two tensors. It divides the corresponding elements of the first tensor by the corresponding elements of the second tensor, e.g.:</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">x = tf.constant([4,6,8])
y = tf.constant([2,3,4])</code></pre>
</div>
</div>
<div class="paragraph">
<p>When you use <code>tf.divide(x, y)</code>, TensorFlow will perform element-wise division between <code>x</code> and <code>y</code>. So, the output would be:</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code>[4 / 2, 6 / 3, 8 / 4] = [2.0 2.0, 2.0]</code></pre>
</div>
</div>
<div class="paragraph">
<p>Here, each element of <code>x</code> is divided by the corresponding element of <code>y</code>.</p>
</div>
<div class="paragraph">
<p>One limitation to be aware of when using <code>tf.divide</code> is that both <code>x</code> and <code>y</code> must be of compatible shapes. In other words, they must have the same shape or be broadcastable to the same shape. Otherwise it will lead to <code>InvalidArgumentError</code>.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Make sure that the <code>tf.divide</code> function arguments have compatible shapes.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">import tensorflow as tf

x = tf.constant([4, 6, 8])
y = tf.constant([2, 3])
z = tf.divide(x, y)  # Noncompliant: InvalidArgumentError: Incompatible shapes: [3] vs. [2]</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">import tensorflow as tf

x = tf.constant([4, 6, 8])
y = tf.constant([2, 3, 4])
z = tf.divide(x, y)  # OK: [2, 2, 2]</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Tensorflow documentation - <a href="https://www.tensorflow.org/api_docs/python/tf/math/divide">tf.divide</a></p>
</li>
<li>
<p>Numpy documentation - <a href="https://numpy.org/doc/stable/user/basics.broadcasting.html#broadcasting">Broadcasting</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>