<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Using <code>unwrap()</code> in conditions where it will always fail leads to unexpected panics and unclear error handling. It&#8217;s likely a mistake and checking that there is indeed a wrapped value and not a none one was intended instead, which would require inverting the condition. However, if panicking was indeed intended, then one should use <code>panic!</code> explicitly to make the intention clear and readable.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-rust" data-lang="rust">let option = Some(0);
fn do_something_with(_x: usize) {}
if option.is_none() {
    do_something_with(option.unwrap()); // Noncompliant: This code will always panic.
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-rust" data-lang="rust">let option = Some(0);
fn do_something_with(_x: usize) {}
if option.is_some() {
    do_something_with(option.unwrap()); // Compliant: Inverted condition to ensure unwrap is safe.
}

// or

if option.is_none() {
    panic!("Option was none when it was expected to be some."); // Compliant: Explicit panic with a message.
}</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Clippy Lints - <a href="https://rust-lang.github.io/rust-clippy/master/index.html#panicking_unwrap" class="bare">https://rust-lang.github.io/rust-clippy/master/index.html#panicking_unwrap</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>