<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when <code>Database.executeBatch()</code> is called with a scope parameter greater than 2000.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>When executing batch jobs in Salesforce, the scope parameter determines how many records are processed in each batch execution. While Salesforce allows scope values up to 2000 and will automatically chunk larger values into smaller batches, specifying excessively large scope parameters can lead to several problems.</p>
</div>
<div class="paragraph">
<p>First, larger batch sizes increase the risk of hitting governor limits within a single batch execution. Each batch execution has its own set of governor limits for DML operations, SOQL queries, and CPU time. Processing too many records at once makes it more likely to exceed these limits, causing the batch job to fail.</p>
</div>
<div class="paragraph">
<p>Second, performance can degrade with larger batch sizes. The Salesforce platform is optimized for processing smaller chunks of data efficiently. Larger batches can lead to longer execution times and increased memory usage, potentially affecting overall system performance.</p>
</div>
<div class="paragraph">
<p>Finally, using appropriate batch sizes improves error handling and recovery. If a batch fails, smaller scope sizes mean fewer records need to be reprocessed, making debugging and recovery more manageable.</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="paragraph">
<p>Using excessively large batch scope parameters can cause batch jobs to fail due to governor limit violations, degrade system performance, and make error recovery more difficult. This can lead to incomplete data processing and unreliable batch operations.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Use a scope parameter of 2000 or less. The default value of 200 is often optimal for most use cases.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-apex" data-lang="apex">AccountUpdateBatchJob obj = new AccountUpdateBatchJob();
Database.executeBatch(obj, 5000); // Noncompliant</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-apex" data-lang="apex">AccountUpdateBatchJob obj = new AccountUpdateBatchJob();
Database.executeBatch(obj, 200); // Compliant</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Batch Apex Developer Guide - <a href="https://developer.salesforce.com/docs/atlas.en-us.apexcode.meta/apexcode/apex_batch.htm">Official Salesforce documentation on Batch Apex implementation and best practices</a></p>
</li>
<li>
<p>Database.executeBatch Method - <a href="https://developer.salesforce.com/docs/atlas.en-us.apexref.meta/apexref/apex_methods_system_database.htm#apex_System_Database_executeBatch_2">Reference documentation for the Database.executeBatch method and its parameters</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>