<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule allows banning the usage of certain data set names.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="paragraph">
<p>Given the parameters:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>programs: SORT,ICEMAN</p>
</li>
<li>
<p>ddNameField: SORTOUT</p>
</li>
<li>
<p>allowedValues: AB\.[^.]*\.DEF</p>
</li>
</ul>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-jcl" data-lang="jcl">//EXEC PGM=SORT
//SORTOUT DD DSN=AB.FOOBAR.XYZ</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-jcl" data-lang="jcl">//EXEC PGM=SORT
//SORTOUT DD DSN=AB.FOOBAR.DEF</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_parameters">Parameters</h3>
<div class="sidebarblock">
<div class="content">
<div class="title">programs</div>
<div class="paragraph">
<p>Comma-separated list of programs where the data set names should be checked.
This parameter is optional. If absent, all DD statements regardless of procedure steps are checked.</p>
</div>
</div>
</div>
<div class="sidebarblock">
<div class="content">
<div class="title">ddNameField</div>
<div class="paragraph">
<p>Name field of the DD statement where the data set name should be checked.</p>
</div>
</div>
</div>
<div class="sidebarblock">
<div class="content">
<div class="title">allowedValues</div>
<div class="paragraph">
<p>Regular expression to match valid data set names.</p>
</div>
</div>
</div>
<hr>
</div>
</div>
</div>