<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>There are several reasons to avoid using this method:</p>
</div>
<div class="olist arabic">
<ol class="arabic">
<li>
<p>It is optionally available only for result sets of type <code>ResultSet.TYPE_FORWARD_ONLY</code>.
Database drivers will throw an exception if not supported.</p>
</li>
<li>
<p>The method can be expensive to execute as the database driver may need to fetch ahead one row to determine whether the current row is the last in the result set.
The documentation of the method explicitly mentions this fact.</p>
</li>
<li>
<p>What "the cursor is on the last row" means for an empty <code>ResultSet</code> is unclear.
Database drivers may return <code>true</code> or <code>false</code> in this case .</p>
</li>
</ol>
</div>
<div class="paragraph">
<p><code>ResultSet.next()</code> is a good alternative to <code>ResultSet.isLast()</code> as it does not have the mentioned issues.
It is always supported and, as per specification, returns <code>false</code> for empty result sets.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Refactor your code to use <code>ResultSet.next()</code> instead of <code>ResultSet.isLast()</code>.
Be cautious of its different semantics and side effects on cursor positioning in the result set.
Verify that your program logic is still valid under these side effects and otherwise adjust it.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">ResultSet results = stmt.executeQuery("SELECT name, address FROM PERSON");
StringBuilder sb = new StringBuilder();
while (results.next() &amp;&amp; !results.isLast()) { // Noncompliant
  sb.append(results.getString("name") + ", ");
}
sb.append(results.getString("name"));
String formattedNames = sb.toString();</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">ResultSet results = stmt.executeQuery("SELECT name, address FROM PERSON");
List&lt;String&gt; names = new ArrayList&lt;&gt;();
while (results.next()) { // Compliant, and program logic refactored
  names.add(results.getString("name"));
}
String formattedNames =  names.stream().collect(Collectors.joining(", "));</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p><a href="https://docs.oracle.com/javase/8/docs/api/java/sql/ResultSet.html#isLast--">Java SE 8 API Specification: ResultSet.isLast()</a></p>
</li>
</ul>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove this call to "isLast()".</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_24_nov_2014_191541_nicolas_peru_wrote">on 24 Nov 2014, 19:15:41 Nicolas Peru wrote:</h3>
<div class="paragraph">
<p>No message.</p>
</div>
<div class="paragraph">
<p>Otherwise seems ok.</p>
</div>
</div>
</div>
</div>