== How to fix it in Laravel

The most restrictive possible permissions should be assigned to files and directories.

=== Code Examples

==== Noncompliant Code Example

[source,php,diff-id=4,diff-type=noncompliant]
----
use Illuminate\Filesystem\Filesystem;

$fs = new Filesystem();
$fs->chmod("foo", 0777); // Noncompliant
----

==== Compliant Solution

[source,php,diff-id=4,diff-type=compliant]
----
use Illuminate\Filesystem\Filesystem;

$fs = new Filesystem();
$fs->chmod("foo", 0750); // Compliant
----
