== How to fix it in PHP

The most restrictive possible permissions should be assigned to files and directories.


=== Code Examples

==== Noncompliant Code Example

[source,php,diff-id=3,diff-type=noncompliant]
----
use Symfony\Component\Filesystem\Filesystem;

$fs = new Filesystem();
$fs->chmod("foo", 0777); // Noncompliant
----

==== Compliant Solution

[source,php,diff-id=3,diff-type=compliant]
----
use Symfony\Component\Filesystem\Filesystem;

$fs = new Filesystem();
$fs->chmod("foo", 0750); // Compliant
----
