== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,java]
----
if (x == 0) {
  doSomething();
} else if (x == 1) {
  doSomethingElse();
} 
----

=== Compliant solution

[source,java]
----
if (x == 0) {
  doSomething();
} else if (x == 1) {
  doSomethingElse();
} else {
  throw new IllegalStateException();
}
----

== Resources

* https://wiki.sei.cmu.edu/confluence/x/RtYxBQ[CERT, MSC01-C.] - Strive for logical completeness
* https://wiki.sei.cmu.edu/confluence/x/jzZGBQ[CERT, MSC57-J.] - Strive for logical completeness

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
