== Why is this an issue?

_Mutexes_ are synchronization primitives that allow to manage concurrency.

Their use requires following a well-defined life-cycle.

* _Mutexes_ need to be initialized (``++pthread_mutex_init++``) before being used. Once it is initialized, a _mutex_ is in an _unlocked_ state.
* _Mutexes_ need to be destroyed (``++pthread_mutex_destroy++``) to free the associated internal resources. Only _unlocked_ _mutexes_ can be safely destroyed.

Before initialization or after destruction, a mutex is in an uninitialized state.


About this life-cycle, the following patterns should be avoided as they result in an undefined behavior:

* trying to initialize an initialized _mutex_
* trying to destroy an initialized _mutex_ that is in a _locked_ state
* trying to destroy an uninitialized _mutex_
* trying to lock an uninitialized _mutex_
* trying to unlock an uninitialized _mutex_

In {cpp}, it is recommended to wrap mutex creation/destruction in a RAII class, as well as mutex lock/unlock. Those RAII classes will perform the right operations, even in presence of exceptions.


=== Noncompliant code example

[source,cpp]
----
pthread_mutex_t mtx1;

void bad1(void)
{
  pthread_mutex_init(&mtx1);
  pthread_mutex_init(&mtx1);
}

void bad2(void)
{
  pthread_mutex_init(&mtx1);
  pthread_mutex_lock(&mtx1);
  pthread_mutex_destroy(&mtx1);
}

void bad3(void)
{
  pthread_mutex_init(&mtx1);
  pthread_mutex_destroy(&mtx1);
  pthread_mutex_destroy(&mtx1);
}

void bad4(void)
{
  pthread_mutex_init(&mtx1);
  pthread_mutex_destroy(&mtx1);
  pthread_mutex_lock(&mtx1);
}

void bad5(void)
{
  pthread_mutex_init(&mtx1);
  pthread_mutex_destroy(&mtx1);
  pthread_mutex_unlock(&mtx1);
}
----


=== Compliant solution

[source,cpp]
----
pthread_mutex_t mtx1;

void ok1(void)
{
  pthread_mutex_init(&mtx1);
  pthread_mutex_destroy(&mtx1);
}

void ok2(void)
{
  pthread_mutex_init(&mtx1);
  pthread_mutex_lock(&mtx1);
  pthread_mutex_unlock(&mtx1);
  pthread_mutex_destroy(&mtx1);
}
----


== Resources

* The Open Group - https://pubs.opengroup.org/onlinepubs/009695399/functions/pthread_mutex_destroy.html[``++pthread_mutex_init++``, ``++pthread_mutex_destroy++``]


ifdef::env-github,rspecator-view[]
'''
== Comments And Links
(visible only on this page)

=== relates to: S5486

=== relates to: S5489

=== is related to: S5486

=== is related to: S5489

=== on 6 Nov 2019, 23:33:11 Lo√Øc Joly wrote:
\[~geoffray.adde] Can you please review my changes?

endif::env-github,rspecator-view[]
