== Why is this an issue?

When using Backbone.js, the use of ``++has++`` to check attribute existence is clearer and more readable than the use of ``++get++``, and is therefore preferred.


=== Noncompliant code example

[source,javascript]
----
if (myModel.get('myProperty')) { ... }
----


=== Compliant solution

[source,javascript]
----
if (myModel.has('myProperty') { ... }
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

endif::env-github,rspecator-view[]
