When using Backbone.js, the use of ``++has++`` to check attribute existence is clearer and more readable than the use of ``++get++``, and is therefore preferred.

== Noncompliant Code Example

----
if (myModel.get('myProperty')) { ... }
----

== Compliant Solution

----
if (myModel.has('myProperty') { ... }
----
