This rule raises an issue when Java `Instant` methods are called with unsupported temporal units that will cause runtime exceptions.

== Why is this an issue?

Java's `Instant` class represents a specific point in time on the timeline. It only supports temporal units that represent exact durations, not calendar-based units that can vary in length.

The supported temporal units are:

* `NANOS` - nanoseconds
* `MICROS` - microseconds  
* `MILLIS` - milliseconds
* `SECONDS` - seconds
* `MINUTES` - minutes
* `HOURS` - hours
* `HALF_DAYS` - 12-hour periods
* `DAYS` - 24-hour periods

Using unsupported units like `WEEKS`, `MONTHS`, or `YEARS` will throw an `UnsupportedTemporalTypeException` at runtime. These units are not supported because their actual duration can vary - for example, months can have 28, 29, 30, or 31 days.

Since `Instant` represents a precise moment in time, it cannot work with units that don't have a fixed duration. This design ensures that time calculations remain predictable and accurate.

=== What is the potential impact?

Using unsupported temporal units with `Instant` methods will cause runtime crashes with `UnsupportedTemporalTypeException`, leading to service disruptions.

== How to fix it

Replace unsupported temporal units with supported ones, or use `LocalDateTime` for calendar operations.

=== Code examples

==== Noncompliant code example

[source,java,diff-id=1,diff-type=noncompliant]
----
Instant instant = Instant.now();
Instant result = instant.plus(1, ChronoUnit.MONTHS); // Noncompliant
----

==== Compliant solution

[source,java,diff-id=1,diff-type=compliant]
----
Instant instant = Instant.now();
Instant result = instant.plus(30, ChronoUnit.DAYS);

// Or use LocalDateTime for calendar operations
LocalDateTime dateTime = LocalDateTime.now();
LocalDateTime result2 = dateTime.plus(1, ChronoUnit.MONTHS);
----

== Resources

=== Documentation

 * Java Instant Documentation - https://docs.oracle.com/en/java/javase/25/docs/api/java.base/java/time/Instant.html[Official Java documentation for the Instant class]

 * Java ChronoUnit Documentation - https://docs.oracle.com/en/java/javase/25/docs/api/java.base/java/time/temporal/ChronoUnit.html[Official documentation for ChronoUnit enum values]

 * Java Time API Guide - https://docs.oracle.com/javase/tutorial/datetime/[Oracle's tutorial on the Java Date and Time API]
