== Why is this an issue?

Superglobal variables are predefined variables available in all scopes throughout a script. However, accessing them directly is considered bad practice. Instead, they should be accessed through an object or framework that handles sanitation and validation.


=== Noncompliant code example

[source,php]
----
$name = $_POST['name'];
----


=== Compliant solution

[source,php]
----
$name = $this->params()->fromPost('name'); 
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Do not access "XXX" directly.


endif::env-github,rspecator-view[]
