Superglobal variables are predefined variables available in all scopes throughout a script. However, accessing them directly is considered bad practice. Instead, they should be accessed through an object or framework that handles sanitation and validation.


== Noncompliant Code Example

----
$name = $_POST['name'];
----


== Compliant Solution

----
$name = $this->params()->fromPost('name'); 
----


