PHP session tokens are normally transmitted through HTTP cookies. However, for
clients that do not support cookies and when the PHP `session.use_trans_sid`
setting is enabled, those tokens can be transmitted as URL parameters.

== Why is this an issue?

GET URL parameter can be disclosed in a variety of ways:

* Directly in a web browser address bar.
* In navigation history.
* In web servers or intermediate proxies log files.

=== What is the potential impact?

Attackers with access to any of those disclosure locations will be able to see
and steal a victim's session token. They can then use it to log as the user,
impersonate their account, and take advantage of their privileges.

== How to fix it

Disable the `session.use_trans_sid` setting in the main PHP configuration file.

Note that this parameter is off by default. 

=== Code examples

==== Noncompliant code example

[source,ini,diff-id=1,diff-type=noncompliant]
----
; php.ini
session.use_trans_sid=1  ; Noncompliant
----

==== Compliant solution

[source,ini,diff-id=1,diff-type=compliant]
----
; php.ini
session.use_trans_sid=0
----

== Resources

=== Standards

* OWASP - https://owasp.org/Top10/A05_2021-Security_Misconfiguration/[Top 10 2021 Category A5 - Security Misconfiguration]
* OWASP - https://owasp.org/www-project-top-ten/2017/A6_2017-Security_Misconfiguration[Top 10 2017 Category A6 - Security Misconfiguration]


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Set "session.use_trans_sid" to 0 or remove this configuration.


'''
== Comments And Links
(visible only on this page)

=== on 1 Sep 2015, 06:47:24 Linda Martin wrote:
LGTM!

endif::env-github,rspecator-view[]
