This rule raises an issue when a method annotated with ``++@future++`` calls another method that is also annotated with ``++@future++``.

== Why is this an issue?

Future methods in Salesforce Apex are designed to execute asynchronously when system resources become available. They run in their own thread and are queued for execution.

The Salesforce platform explicitly prohibits future methods from calling other future methods. This restriction exists because:

* Future methods are already executing asynchronously, and chaining them creates unpredictable execution patterns
* The platform cannot guarantee the order of execution for chained future methods
* Resource management becomes complex when future methods spawn additional future methods
* It can lead to cascading asynchronous operations that are difficult to monitor and debug

When a future method attempts to call another future method, the Salesforce runtime will throw an exception, causing the operation to fail.

=== What is the potential impact?

Violating this rule will result in a runtime exception when the code executes. The calling future method will fail, potentially leaving business processes incomplete.

This can lead to:

* Data inconsistency if the failed operation was part of a larger business process
* Poor user experience due to unexpected failures
* Difficulty in debugging and monitoring asynchronous operations
* Unreliable system behavior in production environments

== How to fix it

Replace the future method call with a call to a synchronous helper method that contains the same logic.

=== Code examples

==== Noncompliant code example

[source,apex,diff-id=1,diff-type=noncompliant]
----
@future
public static void method1() {
    method2(); // Noncompliant
}

@future
public static void method2() {
    // Some logic
}
----

==== Compliant solution

[source,apex,diff-id=1,diff-type=compliant]
----
@future
public static void method1() {
    // Call synchronous helper method instead
    processData();
}

public static void processData() {
    // Same logic as the original future method
}
----

== Resources

=== Documentation

 * Future Methods - https://developer.salesforce.com/docs/atlas.en-us.apexcode.meta/apexcode/apex_invoking_future_methods.htm[Official Salesforce documentation on future methods and their limitations]

 * Asynchronous Apex - https://developer.salesforce.com/docs/atlas.en-us.apexcode.meta/apexcode/apex_async_overview.htm[Overview of asynchronous processing options in Apex]
