== Why is this an issue?

Testing equality or nullness with JUnit's ``++assertTrue()++`` or ``++assertFalse()++``  should be simplified to the corresponding dedicated assertion.


=== Noncompliant code example

[source,text]
----
Assert.assertTrue(a.equals(b));
Assert.assertTrue(a == b);
Assert.assertTrue(a == null);
Assert.assertTrue(a != null);
Assert.assertFalse(a.equals(b));
----


=== Compliant solution

[source,text]
----
Assert.assertEquals(a, b);
Assert.assertSame(a, b);
Assert.assertNull(a);
Assert.assertNotNull(a);
Assert.assertNotEquals(a, b);
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../highlighting.adoc[]

endif::env-github,rspecator-view[]
