<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Offering the same experience with the mouse and the keyboard allow users to pick their preferred devices.</p>
</div>
<div class="paragraph">
<p>Additionally, users of assistive technology will also be able to browse the site even if they cannot use the mouse.</p>
</div>
<div class="paragraph">
<p>This rules detects the following issues:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>when <code>onClick</code> is not accompanied by at least one of the following: <code>onKeyUp</code>, <code>onKeyDown</code>, <code>onKeyPress</code>.</p>
</li>
<li>
<p>when <code>onmouseover</code>/<code>onmouseout</code> are not paired by <code>onfocus</code>/<code>onblur</code>.</p>
</li>
</ul>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it_in_jsx">How to fix it in JSX</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Add at least one of the following event handlers <code>onKeyUp</code>, <code>onKeyDown</code>, <code>onKeyPress</code> to the element when using <code>onClick</code> event handler.
Add corresponding event handlers <code>onfocus</code>/<code>onblur</code> to the element when using <code>onmouseover</code>/<code>onmouseout</code> event handlers.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-javascript" data-lang="javascript">&lt;div onClick={() =&gt; {}} /&gt;

&lt;div onMouseOver={ () =&gt; {}} }  /&gt;</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-javascript" data-lang="javascript">&lt;div onClick={() =&gt; {}} onKeyDown={this.handleKeyDown} /&gt;

&lt;div onMouseOver={ () =&gt; {} } onFocus={ () =&gt; {} } /&gt;</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_exceptions">Exceptions</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This does not apply for interactive or hidden elements, eg. when using <code>aria-hidden="true"</code> attribute.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p><a href="https://github.com/jsx-eslint/eslint-plugin-jsx-a11y">eslint-plugin-jsx-a11y</a> - Rule <a href="https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/blob/HEAD/docs/rules/mouse-events-have-key-events.md">mouse-events-have-key-events</a></p>
</li>
<li>
<p><a href="https://github.com/jsx-eslint/eslint-plugin-jsx-a11y">eslint-plugin-jsx-a11y</a> - Rule <a href="https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/blob/HEAD/docs/rules/click-events-have-key-events.md">click-events-have-key-events</a></p>
</li>
<li>
<p>SCR2 - <a href="https://www.w3.org/TR/WCAG20-TECHS/SCR2.html">Using redundant keyboard and mouse event handlers</a></p>
</li>
<li>
<p>G90 - <a href="https://www.w3.org/TR/WCAG20-TECHS/G90.html">Providing keyboard-triggered event handlers</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>