== Why is this an issue?

Labeled loops are useful, especially when the code is badly indented, to match the begin and end of each loop. This rule raises an issue when the end of a labeled loop is unlabeled.


=== Noncompliant code example

[source,sql]
----
BEGIN
  <<myLoopLabel1>>
  LOOP
    EXIT;
  END LOOP; -- Noncompliant; this labeled loop has no ending label

  LOOP
    EXIT;
  END LOOP; -- Compliant; not a labeled loop
END;
/
----


=== Compliant solution

[source,sql]
----
BEGIN
  <<myLoopLabel1>>
  LOOP
    EXIT;
  END LOOP myLoopLabel1;

  LOOP
    EXIT;
  END LOOP;
END;
/
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Label this "END".


endif::env-github,rspecator-view[]
