Service account tokens should not be mounted in pods

// If you want to factorize the description uncomment the following line and create the file.
//include::../description.adoc[]

== Why is this an issue?

Service account tokens are Kubernetes secrets created automatically to authenticate applications running inside pods to the API server. If a pod is compromised, an attacker could use this token to gain access to other resources in the cluster.

For example, they could create new pods, modify existing ones, or even delete critical system pods, depending on the permissions associated with the service account. This could lead to a variety of security issues, including data leakage, denial of service, or unauthorized modification of the cluster's resources.

Therefore, it's recommended to disable the automounting of service account tokens when it's not necessary for the application running in the pod. This can be done by setting `automountServiceAccountToken: false` in the pod's specification or in the service account's configuration.

//=== What is the potential impact?

== How to fix it
//== How to fix it in FRAMEWORK NAME

=== Code examples

==== Noncompliant code example

[source,yaml,diff-id=1,diff-type=noncompliant]
----
apiVersion: v1
kind: Pod
metadata:
  name: nginx-noncompliant
spec:
  containers:
  - name: nginx-noncompliant
    image: nginx:1.25.3
----

==== Compliant solution

[source,yaml,diff-id=1,diff-type=compliant]
----
apiVersion: v1
kind: Pod
metadata:
  name: nginx-compliant
spec:
  containers:
  - name: nginx-compliant
    image: nginx:1.25.3
  automountServiceAccountToken: false # Compliant

----

The same can be achieved by setting `automountServiceAccountToken: false` in the service account's configuration:

[source,yaml,diff-id=2,diff-type=compliant]
----
apiVersion: v1
kind: ServiceAccount
metadata:
  name: example-service-account
  namespace: default
automountServiceAccountToken: false
---
apiVersion: v1
kind: Pod
metadata:
  name: nginx-compliant-2
spec:
  containers:
  - name: nginx-compliant-2
    image: nginx:1.25.3
  serviceAccountName: example-service-account # Compliant
----

//=== How does this work?

//=== Pitfalls

//=== Going the extra mile


//== Resources
//=== Documentation
//=== Articles & blog posts
//=== Conference presentations
//=== Standards
//=== External coding guidelines
//=== Benchmarks
