== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,swift]
----
func doSomething(a: Int, b: Int) {     // "b" is unused
  compute(a)
}
----

=== Compliant solution

[source,swift]
----
void doSomething(a: Int) { 
  compute(a)
}
----

=== Exceptions

Override methods are excluded.


----
override doSomething(a: Int, b: Int) {     // no issue reported on b
  compute(a)
}
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
