== Why is this an issue?

include::../description.adoc[]

== How to fix it

include::../how-to-fix-it.adoc[]

=== Code examples

==== Noncompliant code example

[source,swift,diff-id=1,diff-type=noncompliant]
----
class Base {
  func example(
    _ p1: String,
    p2: String,
  ) {
    print(p1)
    print(p2)
  }

  private func example(
    p1: String,
    p2: String,  // "p2" is unused and can be removed
  ) {
    print(p1)
  }
}

class Derived: Base {
  override func example(
    _ p1: String,  // "p1" is unused and can be renamed
    p2: String,    // "p2" is unused and can be renamed
  ) {}
}
----

==== Compliant solution

[source,swift,diff-id=1,diff-type=compliant]
----
class Base {
  func example(
    _ p1: String,
    p2: String,
  ) {
    print(p1)
    print(p2)
  }

  private func example(
    p1: String,
  ) {
    print(p1)
  }
}

class Derived: Base {
  override func example(
    _: String,
    p2 _: String,
  ) {}
}
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
