== How to fix it in cURL

=== Code examples

==== Noncompliant code example

[source,yaml,diff-id=1,diff-type=noncompliant]
----
name: Download

on: push

jobs:
  download:
    runs-on: ubuntu-latest
    steps:
      - name: Download file
        run: curl --tlsv1.0 -O https://tlsv1-0.example.com/downloads/install.sh  # Noncompliant
----

==== Compliant solution

[source,yaml,diff-id=1,diff-type=compliant]
----
name: Download

on: push

jobs:
  download:
    runs-on: ubuntu-latest
    steps:
      - name: Download file
        run: curl --tlsv1.2 -O https://tlsv1-3.example.com/downloads/install.sh
----

=== How does this work?

include::../../common/fix/fix.adoc[]
