== How to fix it in Wget

=== Code examples

==== Noncompliant code example

[source,yaml,diff-id=11,diff-type=noncompliant]
----
name: Download

on: push

jobs:
  download:
    runs-on: ubuntu-latest
    steps:
      - name: Download file
        run: wget --secure-protocol TLSv1_1 https://example.com/downloads/install.sh  # Noncompliant
----

==== Compliant solution

[source,yaml,diff-id=11,diff-type=compliant]
----
name: Download

on: push

jobs:
  download:
    runs-on: ubuntu-latest
    steps:
      - name: Download file
        run: wget --secure-protocol TLSv1_2 https://example.com/downloads/install.sh
----

=== How does this work?

include::../../common/fix/fix.adoc[]
