<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when code uses <code>sort(false)</code> instead of <code>toSorted()</code>, or when the result of a mutating <code>sort()</code> call is assigned to a variable.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Groovy collections can be sorted either by mutating the original collection or by creating a new one. The <code>sort()</code> method sorts the collection in place and returns the same instance. When you assign this result to another variable, it is easy to assume you now have two independent collections, while in reality both variables point to the same, already modified collection.</p>
</div>
<div class="paragraph">
<p>Using <code>sort(false)</code> to get a new sorted collection is also confusing. The meaning of the <code>false</code> argument is not obvious, so readers must remember a Groovy special case instead of understanding the code directly. The <code>toSorted()</code> method was introduced to make this intent explicit: it returns a new sorted collection and keeps the original unchanged.</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="paragraph">
<p>Misusing these sorting methods can hide mutations, which makes defects harder to reproduce, increases maintenance cost, and can compromise reliability when collections are shared between components or threads.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Use <code>toSorted()</code> instead of <code>sort(false)</code> when you need a new sorted collection.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-groovy" data-lang="groovy">def originalList = [3, 1, 2]
def sortedList = originalList.sort(false)  // Noncompliant</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-groovy" data-lang="groovy">def originalList = [3, 1, 2]
def sortedList = originalList.toSorted()  // Clear intention to create new sorted list</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Groovy documentation - <a href="http://docs.groovy-lang.org/latest/html/groovy-jdk/java/lang/Iterable.html#sort()">Iterable sort</a></p>
</li>
<li>
<p>Groovy documentation - <a href="http://docs.groovy-lang.org/latest/html/groovy-jdk/java/lang/Iterable.html#toSorted()">Iterable toSorted</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>