<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>No matter whether the optional value is present or not, <code>Optional::orElse</code>'s argument will always be executed. This is usually not what the developer intended when the content of the <code>orElse()</code> call has side effects. Even when no side effect is involved, the unnecessary computation of the <code>orElse()</code> clause might be a waste of resources.</p>
</div>
<div class="paragraph">
<p>Calls to <a href="https://docs.oracle.com/javase/8/docs/api/java/util/Optional.html#orElse-T-"><code>Optional::orElse</code></a> should be replaced with <a href="https://docs.oracle.com/javase/8/docs/api/java/util/Optional.html#orElseGet-java.util.function.Supplier-"><code>Optional::orElseGet</code></a> whenever the alternative value is not a constant.</p>
</div>
<div class="paragraph">
<p>This rule raises an issue when <code>Optional::orElse</code> is called with an argument that doesn&#8217;t evaluate to a constant value.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">Optional&lt;MyObj&gt; opt = getOptMyObj();
MyObj myObj = opt.orElse(new MyObj());  // Noncompliant</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">Optional&lt;MyObj&gt; opt = getOptMyObj();
MyObj myObj = opt.orElseGet(MyObj::new);
Optional&lt;String&gt; optString = getOptString();
String str = opt.orElse("hello");</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Use "orElseGet" here instead of "orElse".</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p><code>orElse</code></p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_27_jun_2017_173318_ann_campbell_wrote">on 27 Jun 2017, 17:33:18 Ann Campbell wrote:</h3>
<div class="paragraph">
<p><a href="https://groups.google.com/forum/?pli=1#!topic/sonarqube/b_DFJ2ual6E" class="bare">https://groups.google.com/forum/?pli=1#!topic/sonarqube/b_DFJ2ual6E</a></p>
</div>
</div>
</div>
</div>