<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>In Kotlin DSL for Gradle build scripts <code>Provider.get()</code> is used to retrieve a value of a specific type.
The best practice is to retrieve the value only when it is required, that is at the latest possible point, otherwise it can lead to ordering issues or to <code>java.lang.IllegalStateException</code> in case of a missing value.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>In order to fix the issue you can:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>move the <code>Provider.get()</code> inside the task action that requires it</p>
</li>
<li>
<p>or, in case the <code>Procider.get()</code> is required outside a task action, replace it with <code>Provider.map()</code> or <code>Provider.flatMap()</code></p>
</li>
</ul>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-kotlin" data-lang="kotlin">val x = Provider.get(value)

tasks.register("logX") {
    group = JavaBasePlugin.DOCUMENTATION_GROUP
    description = "Logging x value."

    doLast {
        logger.quiet("x = $x")
    }
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-kotlin" data-lang="kotlin">tasks.register("logX") {
    group = JavaBasePlugin.DOCUMENTATION_GROUP
    description = "Logging x value."

    doLast {
        val x = Provider.get(value)
        logger.quiet("x = $x")
    }
}</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p><a href="https://docs.gradle.org/current/javadoc/org/gradle/api/provider/Provider.html">org.gradle.api.provider.Provider</a></p>
</li>
</ul>
</div>
</div>
<div class="sect2">
<h3 id="_standards">Standards</h3>
<div class="ulist">
<ul>
<li>
<p><a href="https://github.com/liutikas/gradle-best-practices#dont-call-get-on-a-provider-outside-a-task-action">Don&#8217;t call get on a Provider outside a task action</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>