== Why is this an issue?

Instantiating a ``++StringBuilder++`` or a ``++StringBuffer++`` with a character is misleading because most Java developers would expect the character to be the initial value of the ``++StringBuffer++``. 

What actually happens is that the int representation of the character is used to determine the initial size of the ``++StringBuffer++``.


=== Noncompliant code example

[source,java]
----
StringBuffer foo = new StringBuffer('x');   //equivalent to StringBuffer foo = new StringBuffer(120);
----


=== Compliant solution

[source,java]
----
StringBuffer foo = new StringBuffer("x");
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Replace the constructor character parameter 'X' with string parameter "X".


'''
== Comments And Links
(visible only on this page)

=== on 15 Oct 2013, 08:54:16 Freddy Mallet wrote:
Is implemented by \http://jira.codehaus.org/browse/SONARJAVA-359

endif::env-github,rspecator-view[]
