By contract, any implementation of the `kotlin.collections.Iterator.next()` method should throw a `kotlin.NoSuchElementException` when the iteration has no more elements. Any other behavior when the iteration is done could lead to unexpected behavior for users of this `Iterator`.


== Noncompliant Code Example

----
class MyIterator : Iterator<String?> {
    // ...
    override fun next(): String? {
        if(!hasNext()){
            return null
        }
        // ...
    }
}
----


== Compliant Solution

----
class MyIterator : Iterator<String?> {
    // ...
    override fun next(): String? {
        if(!hasNext()){
            throw NoSuchElementException()
        }
        // ...
    }
}
----

