<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>There are valid cases for passing a variable multiple times into the same method call, but usually doing so is a mistake, and something else was intended for one of the arguments.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-vbnet" data-lang="vbnet">If compare(myPoint.x, myPoint.x) &lt;&gt; 0 Then ' Noncompliant
    '...
End If

If compare(getNextValue(), getNextValue()) &lt;&gt; 0 Then ' Noncompliant
    '...
End If</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-vbnet" data-lang="vbnet">If compare(myPoint.x, myPoint.y) &lt;&gt; 0 Then
    '...
End If

Dim v1 As Integer = getNextValue()
Dim v2 As Integer = getNextValue()
If compare(v1, v2) &lt;&gt; 0 Then
    '...
End If</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Verify that this is the intended value; it is the same as the [1st|2nd|3rd|nth] argument.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="ulist">
<ul>
<li>
<p>primary: the duplicate argument</p>
</li>
<li>
<p>secondary: the first instance of the argument</p>
<div class="ulist">
<ul>
<li>
<p>message: Original instance</p>
</li>
</ul>
</div>
</li>
</ul>
</div>
</div>
</div>
</div>