<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>TypeScript provides two ways to tell the compiler that a literal value should be typed as a literal type like <code>42</code> rather than the primitive one <code>number</code>:</p>
</div>
<div class="ulist">
<ul>
<li>
<p><code>as const</code> tells TypeScript to infer the literal type automatically</p>
</li>
<li>
<p><code>as T</code> where <code>T</code> denotes a literal type to instruct TypeScript to infer the literal type explicitly</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>In practice, <code>as const</code> is preferred because the type checker doesn&#8217;t need re-typing the literal value.</p>
</div>
<div class="paragraph">
<p>Therefore, the rule flags occurrences of explicit literal types that can be replaced with an <code>as const</code> assertion.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Replace the explicit literal type assertion with <code>as const</code>.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-typescript" data-lang="typescript">class Foo {
  public static foo: 42 = 42; // Noncompliant

  // ...
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-typescript" data-lang="typescript">class Foo {
  public static foo = 42 as const;

  // ...
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_noncompliant_code_example_2">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-typescript" data-lang="typescript">let foo = { bar: 'baz' as 'baz' };</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution_2">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-typescript" data-lang="typescript">let foo = { bar: 'baz' as const };</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p><a href="https://typescript-eslint.io/">typescript-eslint</a> - Rule <a href="https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/docs/rules/prefer-as-const.mdx">prefer-as-const</a></p>
</li>
<li>
<p>TypeScript Documentation - <a href="https://www.typescriptlang.org/docs/handbook/2/everyday-types.html#literal-types">Literal types</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>