<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Flag variables are boolean parameters used to control the behavior of a method. Specifying such a parameter may seem innocuous, particularly if it&#8217;s well named.</p>
</div>
<div class="paragraph">
<p>Unfortunately, the maintainers of the code calling the method won&#8217;t see the parameter name, only its value. They&#8217;ll be forced either to guess at the meaning or to take extra time to look the method up.</p>
</div>
<div class="paragraph">
<p>Instead, separate methods should be written.</p>
</div>
<div class="paragraph">
<p>This rule flags methods where a boolean parameter is used to control the majority of the code in the method.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-text" data-lang="text">public void getDressed(boolean isCold) {

  if (isCold()) {
    turtleneck();
    sweater();
    thickSocks();
    heavyPants();
    boots();
  } else {
    tshirt();
    shorts();
    sandals();
  }
}

getDressed(false); // does this mean go naked?</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-text" data-lang="text">public void getDressedColdWeather() {

  turtleneck();
  sweater();
  thickSocks();
  heavyPants();
  boots();
}

public void getDressedWarmWeather() {
  tshirt();
  shorts();
  sandals();
}

getDressedWarmWeather();</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_duplicates_s2301">duplicates: <a data-rspec-id="S2301" class="rspec-auto-link">S2301</a></h3>

</div>
<div class="sect2">
<h3 id="_is_related_to_s2494">is related to: <a data-rspec-id="S2494" class="rspec-auto-link">S2494</a></h3>

</div>
</div>
</div>