<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when a class declares a traditional <code>public static void main(String[] args)</code> method but does not use the <code>args</code> parameter, and the method could be simplified to an instance main method.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Java 25 introduced instance main methods (JEP 512), allowing simpler entry points. The traditional <code>public static void main(String[] args)</code> includes unnecessary complexity when <code>args</code> is unused:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>The <code>public</code> modifier is not required</p>
</li>
<li>
<p>The <code>static</code> modifier forces static methods or object instantiation</p>
</li>
<li>
<p>The <code>String[] args</code> parameter is often unused</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>The instance main method <code>void main()</code> removes this boilerplate while maintaining compatibility.</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="paragraph">
<p>Using the traditional static main method when the simpler instance main method would suffice adds unnecessary complexity and boilerplate that serves no functional purpose.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Replace with an instance main method when String[] parameter is unused.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">public class HelloWorld {
    public static void main(String[] args) { // Noncompliant
        System.out.println("Hello, World!");
    }
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">class HelloWorld {
    void main() {
        System.out.println("Hello, World!");
    }
}</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>JEP 512: Compact Source Files and Instance Main Methods - <a href="https://openjdk.org/jeps/512">Official OpenJDK documentation for the instance main methods feature</a></p>
</li>
</ul>
</div>
</div>
<div class="sect2">
<h3 id="_related_rules">Related rules</h3>
<div class="ulist">
<ul>
<li>
<p><a data-rspec-id="S3051" class="rspec-auto-link">RSPEC-3051</a> - <a href="https://rules.sonarsource.com/java/RSPEC-3051/">Unused method parameters should be removed</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>