== Why is this an issue?

include::../description.adoc[]

Code like

[source,kotlin]
----
if (condition1) {
  if (condition2) {             // NonCompliant
    /* ... */
  }
}
----

Will be more readable as

[source,kotlin]
----
if (condition1 && condition2) { // Compliant
  /* ... */
}
----


== How to fix it

If merging the conditions seems to result in a more complex code, extracting the condition or part of it in a named function or variable is a better approach to fix readability.

=== Code examples

==== Noncompliant code example

include::../noncompliant.adoc[]

==== Compliant solution

[source,kotlin]
----
if (file != null && isFileOrDirectory(file)) {   // Compliant
  /* ... */
}
fun isFileOrDirectory(file: File): Boolean {
    return file.isFile() || file.isDirectory();
}
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../highlighting.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
