include::../description.adoc[]

== Noncompliant Code Example

----
function MyFunc() {
  // ...
}
----

== Compliant Solution

----
function myFunc() {
  // ...
}
----

ifdef::env-github,rspecator-view[]
'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
