<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Usually <code>IntPtr</code> and <code>UIntPtr</code> fields are used to store pointers to unmanaged resources and the finalizer will free the unmanaged resource pointed to by the pointer fields. If the garbage collector finalises the object while the managed resources are still in use it could lead to serious, hard to diagnose bug. To prevent this from happening the object should be kept alive by calling <code>GC.KeepAlive(this)</code> in methods calling unmanaged code on this pointers.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-csharp" data-lang="csharp">using System;

namespace MyLibrary
{
   class Foo
   {
      private IntPtr res;

      Foo()
      {
         GetRes (res);
      }

      ~Foo()
      {
         FreeRes (res);
      }

      void Bar()  // Noncompliant
      {
         UseRes(res);
      }

      // Methods that would typically make calls to unmanaged code.
      void GetRes(IntPtr p)
      {
        // Allocate the resource ...
      }
      void FreeRes(IntPtr p)
      {
        // Free the resource and set the pointer to null ...
      }
      void UseRes(IntPtr p)
      {
        // Use the resource in unmanaged code ...
      }

   }
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-csharp" data-lang="csharp">using System;

namespace MyLibrary
{
   class Foo
   {
      private IntPtr res;

      Foo()
      {
         GetRes (res);
      }

      ~Foo()
      {
         FreeRes (res);
      }

      void Bar()
      {
         UseRes(res);
         GC.KeepAlive(this);
      }

      // Methods that would typically make calls to unmanaged code.
      void GetRes(IntPtr p)
      {
        // Allocate the resource ...
      }
      void FreeRes(IntPtr p)
      {
        // Free the resource and set the pointer to null ...
      }
      void UseRes(IntPtr p)
      {
        // Use the resource in unmanaged code ...
      }

   }
}</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_28_jan_2019_143446_amaury_levé_wrote">on 28 Jan 2019, 14:34:46 Amaury Levé wrote:</h3>
<div class="paragraph">
<p>We cannot find how to get the original FxCop to report any issue + the description is not precise enough so we cannot find what/how the rule should be implemented.</p>
</div>
<div class="paragraph">
<p>cc [~nicolas.harraudeau]</p>
</div>
</div>
</div>
</div>