<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>In Java, an enum is a special data type that allows you to define a set of constants.
Nested enum types, also known as inner enum types, are enum types that are defined within another class or interface.</p>
</div>
<div class="paragraph">
<p>Nested enum types are implicitly static, so there is no need to declare them <code>static</code> explicitly.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">public class Flower {
  static enum Color { // Noncompliant; static is redundant here
    RED, YELLOW, BLUE, ORANGE
  }
  // ...
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">public class Flower {
  enum Color { // Compliant
    RED, YELLOW, BLUE, ORANGE
  }
  // ...
}</code></pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p><a href="https://docs.oracle.com/javase/specs/jls/se20/html/jls-8.html#jls-8.9">Java Language Specification-8.9</a></p>
</li>
</ul>
</div>
<hr>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove this redundant "static" qualifier; nested enum types are implicitly static.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>Primary Location: the 'static' keyword</p>
</div>
<div class="paragraph">
<p>Secondary Location: the 'enum' keyword</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_31_mar_2015_150956_nicolas_peru_wrote">on 31 Mar 2015, 15:09:56 Nicolas Peru wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2]could be nice to add a link or a quote from the jls : http://docs.oracle.com/javase/specs/jls/se7/html/jls-8.html#jls-8.9</p>
</div>
<div class="quoteblock">
<blockquote>
<div class="paragraph">
<p>Nested enum types are implicitly static. It is permissible to explicitly declare a nested enum type to be static.</p>
</div>
</blockquote>
</div>
</div>
<div class="sect2">
<h3 id="_on_31_mar_2015_174606_ann_campbell_wrote">on 31 Mar 2015, 17:46:06 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~nicolas.peru] why would we quote the docs as saying it&#8217;s permissible to do exactly what this rule says not to do?</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_1_apr_2015_084230_nicolas_peru_wrote">on 1 Apr 2015, 08:42:30 Nicolas Peru wrote:</h3>
<div class="paragraph">
<p>Then let&#8217;s just keep the first part of the sentence :</p>
</div>
<div class="quoteblock">
<blockquote>
<div class="literalblock">
<div class="content">
<pre>Nested enum types are implicitly static.</pre>
</div>
</div>
</blockquote>
</div>
<div class="paragraph">
<p>point is to refer to the correct section of the JLS to let user know we did not invent this.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_1_apr_2015_170642_ann_campbell_wrote">on 1 Apr 2015, 17:06:42 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>Done [~nicolas.peru]</p>
</div>
</div>
</div>
</div>