== How to fix it in .NET

=== Code examples

include::../../common/fix/code-rationale.adoc[]

==== Noncompliant code example

[source,vbnet,diff-id=1,diff-type=noncompliant]
----
Public Class ExampleController
    Inherits Controller

    Public Function Apply(EffectName As String) As IActionResult
        Dim EffectInstance = Activator.CreateInstance(Nothing, EffectName) ' Noncompliant
        Dim EffectPlugin As Object = EffectInstance.Unwrap()

        If CType(EffectPlugin, IEffect).ApplyFilter() Then
            Return Ok()
        Else
            Return StatusCode(500, "Problem occurred")
        End If
    End Function
End Class

Public Interface IEffect
    Function ApplyFilter() As Boolean
End Interface
----

==== Compliant solution

[source,vbnet,diff-id=1,diff-type=compliant]
----
Public Class ExampleController
    Inherits Controller

    Private Shared ReadOnly EffectAllowList() As String = {
        "SepiaEffect",
        "BlackAndWhiteEffect",
        "WaterColorEffect",
        "OilPaintingEffect"
    }

    Public Function Apply(EffectName As String) As IActionResult
        If Not EffectAllowList.Contains(EffectName) Then
            Return BadRequest("Invalid effect name. The effect is not allowed.")
        End If

        Dim EffectInstance = Activator.CreateInstance(Nothing, EffectName)
        Dim EffectPlugin As Object = EffectInstance.Unwrap()

        If CType(EffectPlugin, IEffect).ApplyFilter() Then
            Return Ok()
        Else
            Return StatusCode(500, "Problem occurred")
        End If
    End Function
End Class

Public Interface IEffect
    Function ApplyFilter() As Boolean
End Interface
----

=== How does this work?

include::../../common/fix/pre-approved-list.adoc[]

