include::../description.adoc[]

== Noncompliant Code Example

----
IF something THEN
  IF something_else THEN
    -- ...
  END IF;
END IF;
----

== Compliant Solution

----
IF something AND something_else THEN
  -- ...
END IF;
----

ifdef::env-github,rspecator-view[]
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
