== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,sql]
----
IF something THEN
  IF something_else THEN
    -- ...
  END IF;
END IF;
----

=== Compliant solution

[source,sql]
----
IF something AND something_else THEN
  -- ...
END IF;
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../highlighting.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
