In Kotlin, we're using `suspend` function modifier to declare the function, which might take some time to execute and can potentially suspend the caller thread. The place where such function is called is a "suspension point". Functions marked as `suspend` should have at least one suspension point, otherwise there is no sense to mark suvh function as `suspend`

This rule reports an issue if a function with `suspend` modifier has no calls to other `suspend` functions inside its body.

== Noncompliant Code Example

----
suspend fun function() { // Noncompliant, redundant 'suspend' modifier
    println("Hello!")
}
----

== Compliant Solution

----
fun function() {
    println("Hello!")
}
----

== See

* https://kotlinlang.org/docs/coroutines-guide.html[Coroutines guide]
