Calls to std::format and std::vformat can receive either format strings whose replacement fields are fully indexed or fully non-indexed.

Explicitly indexing replacement fields is useful when arguments are not used in order or are used multiple times. Implicit indexing has the advantage of tersity and simplicity.

Using explicit indexing in a context where implicit indexing would have the same behavior is more verbose, error-prone, and might confuse the reader who might expect the arguments not to be used in order.

== Noncompliant Code Example

[source,cpp]
----
std::string printRectangle(Rectangle const& r) {
    return std::format("[{0},{1}]:({2},{3})", r.x, r.y, r.width, r.height); // Noncompliant
}
----

== Compliant Solution

[source,cpp]
----
std::string printRectangle(Rectangle const& r) {
    return std::format("[{},{}]:({},{})", r.x, r.y, r.width, r.height); // Compliant
}
----
