Calls to std::format and std::vformat can receive either format strings whose replacement fields are fully indexed or fully non-indexed.

Indexed replacement fields are useful when parameters are not used in order or are used multiple times. Non-indexed replacement fields have the advantage of tersity and simplicity.

Using indexed replacement fields in a context where non-indexed ones would have the same behaviour is more verbose, error prone, and might give the impression to the reader that parameters will not be read in order.

== Noncompliant Code Example

[source,cpp]
----
std::string printRectangle(Rectangle const& r) {
    return std::format("[{0},{1}]:({2},{3})", r.x, r.y, r.width, r.height); // noncompliant
}
----

== Compliant Solution

[source,cpp]
----
std::string printRectangle(Rectangle const& r) {
    return std::format("[{},{}]:({},{})", r.x, r.y, r.width, r.height);
}
----
