== Why is this an issue?

The storage of a ``++FIXED DECIMAL++`` is most efficient when you use an odd number of digits, so that the leftmost byte is fully used.

=== Noncompliant code example

[source,pli]
----
 declare total fixed dec(10);
----

=== Compliant solution

[source,pli]
----
 declare total fixed dec(9);
----
ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

endif::env-github,rspecator-view[]
