The storage of a ``++FIXED DECIMAL++`` is most efficient when you use an odd number of digits, so that the leftmost byte is fully used.

== Noncompliant Code Example

[source,pli]
----
 declare total fixed dec(10);
----

== Compliant Solution

[source,pli]
----
 declare total fixed dec(9);
----
ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

endif::env-github,rspecator-view[]
