Although, it is syntactically possible (by using delimiters) to use SQL reserved keywords as identifiers and object names, it is best to not use them to avoid any confusion.


== Noncompliant Code Example

[source,text]
----
CREATE TABLE "SELECT" ( "FROM" INT ); -- Noncompliant

SELECT "FROM" FROM "SELECT";
----


