== Why is this an issue?

Using an ``++ORDER BY++`` clause in a ``++SELECT++`` helps ensure runtime reproducibility. Without one, _some_ databases _may_ return results in a consistent order, but it is likely to be one that appears random to users and is not optimal for processing the data. 


=== Noncompliant code example

[source,text]
----
SELECT model   -- Noncompliant
FROM car
WHERE wheel_count = 4
----


=== Compliant solution

[source,text]
----
SELECT model
FROM car
WHERE wheel_count = 4
ORDER BY year, make
----

