<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>In JCL, the <code>COND</code> code is used to control the execution of job steps based on the completion codes of previous steps.
Omitting the <code>COND</code> code in a subsequent step will lead to the execution of the step even when previous steps have failed.
This can result in unnecessary resource usage, potential data corruption, and can make debugging more difficult.</p>
</div>
<div class="sect2">
<h3 id="_exceptions">Exceptions</h3>
<div class="paragraph">
<p>The rule will not trigger on job steps inside <code>IF/THEN/ELSE/ENDIF</code> statements.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Code a <code>COND</code> parameter on this step and all subsequent steps.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="paragraph">
<p>With the rule parameter <code>steps</code> set to 2:</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-jcl" data-lang="jcl">//MYJOB JOB ,MSGLEVEL=1
//STEP1    EXEC PGM=FIRSTPGM
//STEP2    EXEC PGM=SECONDPGM
//STEP3    EXEC PGM=THIRDPGM</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-jcl" data-lang="jcl">//MYJOB JOB ,MSGLEVEL=1
//STEP1    EXEC PGM=FIRSTPGM
//STEP2    EXEC PGM=SECONDPGM
//STEP3    EXEC PGM=THIRDPGM,COND=(4,LT)</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p><a href="https://www.ibm.com/docs/en/zos/3.1.0?topic=statement-cond-parameter">z/OS MVS JCL Reference - EXEC statement - COND parameter</a></p>
</li>
</ul>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_parameters">Parameters</h3>
<div class="sidebarblock">
<div class="content">
<div class="title">steps</div>
<div class="listingblock">
<div class="content">
<pre>^4$</pre>
</div>
</div>
<div class="paragraph">
<p>Maximum number of steps without COND.</p>
</div>
</div>
</div>
<hr>
</div>
</div>
</div>