Currently, ``++VARCHAR++`` and ``++VARCHAR2++`` are identical data types. But to prevent future changes in behavior, Oracle recommends the use of ``++VARCHAR2++``.


== Noncompliant Code Example

----
DECLARE
  var VARCHAR(42);  -- Noncompliant
BEGIN
  NULL;
END;
/
----


== Compliant Solution

----
DECLARE
  var VARCHAR2(42);  -- Noncompliant
BEGIN
  NULL;
END;
/
----

