== Why is this an issue?

Currently, `VARCHAR` and `VARCHAR2` are identical data types. But to prevent future changes in behavior, Oracle recommends the use of `VARCHAR2`.


=== Noncompliant code example

[source,sql,diff-id=1,diff-type=noncompliant]
----
DECLARE
  var VARCHAR(42);  -- Noncompliant
BEGIN
  NULL;
END;
/
----


=== Compliant solution

[source,sql,diff-id=1,diff-type=compliant]
----
DECLARE
  var VARCHAR2(42);  -- Compliant
BEGIN
  NULL;
END;
/
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Make this variable a "VARCHAR2".


endif::env-github,rspecator-view[]
