include::../common/rule.adoc[tag=header]

=== Code examples

==== Noncompliant code example

[source,javascript,diff-id=1,diff-type=noncompliant]
----
function MyInput() {
    return <input type="text" autocomplete="foo" />; // Noncompliant
}
----

==== Compliant solution

[source,javascript,diff-id=1,diff-type=compliant]
----
function MyInput() {
    return <input type="text" autocomplete="name" />;
}
----

include::../common/rule.adoc[tag=footer]