== Why is this an issue?

This rule allows banning certain statements. 


=== Noncompliant code example

Given a parameter of ``++CANCEL++`` for this rule: 

[source,cobol]
----
DISPLAY "Cancelling action".
CANCEL PROGRAM1. *> Noncompliant
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Remove this use of "xxx".


=== Parameters

.statements
****

Comma-separated list of disallowed statements, such as "EXIT PROGRAM, CANCEL"
****


=== Highlighting

the forbidden statement


endif::env-github,rspecator-view[]
