== Why is this an issue?

This rule allows banning certain statements. 


=== Noncompliant code example

Given a parameter of ``++CANCEL++`` for this rule: 

[source,cobol]
----
DISPLAY "Cancelling action".
CANCEL PROGRAM1. *> Noncompliant
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

include::parameters.adoc[]

include::highlighting.adoc[]

endif::env-github,rspecator-view[]
