<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>When the size of a variable-length table is <code>DEPENDING ON</code> a non-<code>BINARY</code>/<code>COMP</code> variable, use of that table is inefficient because a conversion must be done every time the table is used.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cobol" data-lang="cobol">01 VARS
    05 TABLE_SIZE   PIC 9(4).
    05 MY_TABLE OCCURS 1 TO 10
                        DEPENDING ON TABLE_SIZE  *&gt; Noncompliant; TABLE-SIZE isn't BINARY or COMP
                        PIC X(10).</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cobol" data-lang="cobol">01 VARS
    05 TABLE_SIZE   PIC 9(4) BINARY.
    05 MY_TABLE OCCURS 1 TO 10
                        DEPENDING ON TABLE_SIZE
                        PIC X(10).</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Use a "BINARY" variable to set the size of this table; "xxx" is a non-binary variable.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p><code>xxx</code></p>
</div>
</div>
</div>
</div>