== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,python]
----
if param == 1:
  openWindow()
elif param == 2:
  closeWindow()
elif param == 1:            # Noncompliant
  moveWindowToTheBackground()
----

=== Compliant solution

[source,python]
----
if param == 1:
  openWindow()
elif param == 2:
  closeWindow()
elif param == 3:        
  moveWindowToTheBackground()
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../highlighting.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
