<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>There are several compilations options available for Visual Basic source code and <code>Option Strict</code> defines compiler behavior for implicit data type conversions. Specifying <code>Option Strict Off</code> will allow:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>Implicit narrowing conversions</p>
</li>
<li>
<p>Late binding</p>
</li>
<li>
<p>Implicit typing that results in an <code>Object</code> type</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>This behavior can lead to unexpected runtime errors due to type mismatch or missing members.</p>
</div>
<div class="paragraph">
<p><code>Option Strict</code> can be set in project properties or overridden in individual source files.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-text" data-lang="text">Option Strict Off    ' Noncompliant

Public Class KnownType

    Public ReadOnly Property Name As String

End Class

Public Module MainMod

    Public Function DoSomething(Arg) As String  ' Type for "Arg" argument is not defined.
        Dim Item As KnownType = Arg             ' Implicit narrowing conversion doesn't enforce "Arg" to be of type "KnownType"
        Return Arg.Title                        ' "Title" might not exist in "Arg"
    End Function

End Module</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-text" data-lang="text">Option Strict On

Public Class KnownType

    Public ReadOnly Property Name As String

End Class

Public Module MainMod

    Public Function DoSomething(Arg As KnownType) As String
        Dim Item As KnownType = Arg
        Return Arg.Name
    End Function

End Module</code></pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p><a href="https://docs.microsoft.com/en-us/dotnet/visual-basic/language-reference/statements/option-strict-statement">Visual Basic documentation - Option Strict Statement</a></p>
</li>
</ul>
</div>
<hr>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="ulist">
<ul>
<li>
<p>Change this to 'Option Strict On'.</p>
</li>
<li>
<p>Configure 'Option Strict On' for assembly '{0}'.</p>
</li>
</ul>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>"Option Strict Off" statement</p>
</div>
</div>
</div>
</div>