== Why is this an issue?

When error conditions occur, it is usually a bad idea to simply ignore them. Instead, it is better to handle them properly, or at least to log them.

=== Noncompliant code example

[source,cobol]
----
EXEC CICS
  ...
  INGNORE CONDITION ERROR *> Noncompliant
END-EXEC.
----

include::../compliant.adoc[]

include::../see.adoc[]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
