This rule creates a issue whenever an ``++#error++`` preprocessor directive is reached during the project's analysis. In most cases, this indicates that the preprocessor was badly configured. Some predefined macros or library include paths might be required to fix the configuration.


== Noncompliant Code Example

----
#error This is an error
----

