== Why is this an issue?

This rule creates a issue whenever an ``++#error++`` preprocessor directive is reached during the project's analysis. In most cases, this indicates that the preprocessor was badly configured. Some predefined macros or library include paths might be required to fix the configuration.


=== Noncompliant code example

[source,cpp]
----
#error This is an error
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Review the preprocessor configuration options to not reach this #error directive.


endif::env-github,rspecator-view[]
