== How to fix it in Bcrypt

=== Code examples

==== Noncompliant code example

[source,python,diff-id=201,diff-type=noncompliant]
----
import bcrypt

def hash_password(password):
    return bcrypt.hashpw(password, bcrypt.gensalt(2)) # Noncompliant

def kdf(password, salt):
    return bcrypt.kdf(
    password=password,
    salt=salt,
    desired_key_bytes=32,
    rounds=10, # Noncompliant
    ignore_few_rounds=True)
----

==== Compliant solution

[source,python,diff-id=201,diff-type=compliant]
----
import bcrypt

def hash_password(password):
    return bcrypt.hashpw(password, bcrypt.gensalt())

def kdf(password, salt):
    return bcrypt.kdf(
    password=password,
    salt=salt,
    desired_key_bytes=32,
    rounds=100)
----

=== How does this work?

==== Use a stronger hashing algorithm

In the order, the following are the most secure password hashing algorithms to
date:

. Argon2
. scrypt
. bcrypt
. PBKDF2

Anytime possible, choose the strongest algorithm available. The OWASP documents
potential methods of algorithm upgrades here:
https://cheatsheetseries.owasp.org/cheatsheets/Password_Storage_Cheat_Sheet.html#upgrading-legacy-hashes[Upgrading Legacy Hashes].

==== Choose the right parameters

If bcrypt must be used, the default values of bcrypt are considered safe.

For `hashpw()`, the lib uses 12 rounds of hashing. However, if you want to be
extra cautious, you can increase the number of rounds to 14 or 15.

For `kdf()`, more than 50 rounds should be used. This number is not the same
coefficient as the former because the `kdf()` function uses a different
algorithm.

=== Pitfalls

==== Pre-hashing passwords

As bcrypt has a maximum length input length of 72 bytes for most
implementations, some developers may be tempted to pre-hash the password with a
stronger algorithm before hashing it with bcrypt.

Pre-hashing passwords with bcrypt is not recommended as it can lead to
a specific range of issues. Using a strong salt and a high number of rounds is
enough to protect the password.

More information about this can be found here:
https://cheatsheetseries.owasp.org/cheatsheets/Password_Storage_Cheat_Sheet.html#pre-hashing-passwords-with-bcrypt[Pre-hashing Passwords with Bcrypt].

