== How to fix it in Django

=== Code examples

==== Noncompliant code example

[source,python,diff-id=203,diff-type=noncompliant]
----
# settings.py
PASSWORD_HASHERS = [
    'django.contrib.auth.hashers.SHA1PasswordHasher',  # Noncompliant
    'django.contrib.auth.hashers.MD5PasswordHasher',   # Noncompliant
    'django.contrib.auth.hashers.CryptPasswordHasher', # Noncompliant
]
----

==== Compliant solution

[source,python,diff-id=203,diff-type=compliant]
----
# settings.py
PASSWORD_HASHERS = [
    'django.contrib.auth.hashers.Argon2PasswordHasher',
    'django.contrib.auth.hashers.ScryptPasswordHasher',
    'django.contrib.auth.hashers.BCryptSHA256PasswordHasher',
    'django.contrib.auth.hashers.PBKDF2PasswordHasher',
    'django.contrib.auth.hashers.PBKDF2SHA1PasswordHasher',
]
----

To understand password management in Django,
https://docs.djangoproject.com/en/5.0/topics/auth/passwords/[Password management in Django]
is a good place to start.

=== How does this work?

include::../../common/fix/password-hashing.adoc[]

=== Going the extra mile

include::../../common/extra-mile/peppering.adoc[]

