== How to fix it in Django

=== Code examples

==== Noncompliant code example

[source,python,diff-id=203,diff-type=noncompliant]
----
# settings.py
PASSWORD_HASHERS = [
    'django.contrib.auth.hashers.SHA1PasswordHasher',  # Noncompliant
    'django.contrib.auth.hashers.MD5PasswordHasher',   # Noncompliant
    'django.contrib.auth.hashers.CryptPasswordHasher', # Noncompliant
]
----

==== Compliant solution

[source,python,diff-id=203,diff-type=compliant]
----
# settings.py
PASSWORD_HASHERS = [
    'django.contrib.auth.hashers.Argon2PasswordHasher',
    'django.contrib.auth.hashers.ScryptPasswordHasher',
    'django.contrib.auth.hashers.BCryptSHA256PasswordHasher',
    'django.contrib.auth.hashers.PBKDF2PasswordHasher',
    'django.contrib.auth.hashers.PBKDF2SHA1PasswordHasher',
]
----

=== How does this work?

include::../../common/fix/password-hashing.adoc[]

In the previous example, Argon2 is used as the default password hashing function
by Django. Use the `PASSWORD_HASHERS` variable carefuly. If there is a need to
upgrade, use 
https://docs.djangoproject.com/en/5.0/topics/auth/passwords/#password-upgrading[Django's password upgrade documentation].

=== Going the extra mile

==== Tweaking password hashing parameters

It is possible to change the parameters of the password hashing algorithm to
make it more secure. For example, you can increase the number of iterations or
the length of the salt. +
https://docs.djangoproject.com/en/5.0/topics/auth/passwords/[The Django documentation contains more details about these parameters].

include::../../common/extra-mile/peppering.adoc[]

