== How to fix it in Flask

=== Code examples

==== Noncompliant code example

[source,python,diff-id=204,diff-type=noncompliant]
----
from flask import Flask, request
from flask_bcrypt import Bcrypt

app = Flask(__name__)
bcrypt = Bcrypt(app)

@app.get("/")
def hash():
    password = request.args.get('password', '')
    hashed_password = bcrypt\
        .generate_password_hash(password, rounds=2)\ # Noncompliant
        .decode('utf-8')
    return f"<p>{hashed_password}</p>"
----

==== Compliant solution

[source,python,diff-id=204,diff-type=compliant]
----
from flask import Flask, request
from flask_bcrypt import Bcrypt

app = Flask(__name__)
bcrypt = Bcrypt(app)

@app.get("/")
def hash():
    password = request.args.get('password', '')
    hashed_password = bcrypt.generate_password_hash(password).decode('utf-8')
    return f"<p>{hashed_password}</p>"
----

=== How does this work?

include:: FIX_ME.adoc[]

=== Pitfalls

include:: FIX_ME.adoc[]

=== Going the extra mile

include:: FIX_ME.adoc[]

