== How to fix it in Flask

=== Code examples

==== Noncompliant code example

[source,python,diff-id=204,diff-type=noncompliant]
----
from flask import Flask, request
from flask_bcrypt import Bcrypt

app = Flask(__name__)
bcrypt = Bcrypt(app)

@app.get("/")
def hash():
    password = request.args.get('password', '')
    hashed_password = bcrypt.generate_password_hash(password, rounds=2) # Noncompliant

    return f"<p>{hashed_password.decode('utf-8')}</p>"
----

==== Compliant solution

[source,python,diff-id=204,diff-type=compliant]
----
from flask import Flask, request
from flask_bcrypt import Bcrypt

app = Flask(__name__)
bcrypt = Bcrypt(app)

@app.get("/")
def hash():
    password = request.args.get('password', '')
    hashed_password = bcrypt.generate_password_hash(password)
    return f"<p>{hashed_password.Decode('utf-8')}</p>"
----

=== How does this work?

include::../../common/fix/password-hashing.adoc[]

include::../../common/fix/bcrypt-parameters.adoc[]

=== Pitfalls

include::../../common/pitfalls/pre-hashing.adoc[]

=== Going the extra mile

include::../../common/extra-mile/peppering.adoc[]

