== How to fix it in Hashlib

=== Code examples

==== Noncompliant code example

[source,python,diff-id=206,diff-type=noncompliant]
----
from hashlib import scrypt

def hash_password(password, salt):
    return scrypt(
        password,
        salt,
        n=1 << 10,
        r=8,
        p=2,
        dklen=64) # Noncompliant
----

Code targeting PBKDF2:

[source,python,diff-id=266,diff-type=noncompliant]
----
from hashlib import pbkdf2_hmac

def hash_password(password, salt):
    return pbkdf2_hmac(
        'sha1',
        password,
        salt,
        500_000) # Noncompliant
----


==== Compliant solution

[source,python,diff-id=206,diff-type=compliant]
----
from hashlib import scrypt

def hash_password(password, salt):
    return scrypt(
        password,
        salt,
        n=1 << 17,
        r=8,
        p=1,
        dklen=64)
----

Code targeting PBKDF2:

[source,python,diff-id=266,diff-type=compliant]
----
from hashlib import pbkdf2_hmac

def hash_password(password, salt):
    return pbkdf2_hmac(
        'sha256',
        password,
        salt,
        600_000)
----


=== How does this work?

include::../../common/fix/password-hashing.adoc[]

The following sections provide guidance on the usage of these secure
password-hashing algorithms as provided by hashlib.

include::../../common/fix/scrypt-parameters.adoc[]

The values recommended by OWASP are the following:

[source,python]
----
OWASP_1 = {
    "n": 1 << 17,
    "r": 8,
    "p": 1,
    "dklen": 64,
}

OWASP_2 = {
    "n": 1 << 16,
    "r": 8,
    "p": 2,
    "dklen": 64,
}

OWASP_3 = {
    "n": 1 << 15,
    "r": 8,
    "p": 3,
    "dklen": 64,
}

OWASP_4 = {
    "n": 1 << 14,
    "r": 8,
    "p": 5,
    "dklen": 64,
}

OWASP_5 = {
    "n": 1 << 13,
    "r": 8,
    "p": 10,
    "dklen": 64,
}

# To use these examples, you can use the dictionnary as a ``**kwargs`` variable:
scrypt(password, salt, **OWASP_1)
----

include::../../common/fix/pbkdf2-parameters.adoc[]

=== Pitfalls

include::../../common/pitfalls/pre-hashing.adoc[]

=== Going the extra mile

include::../../common/extra-mile/peppering.adoc[]

