== How to fix it in Passlib

=== Code examples

==== Noncompliant code example

[source,python,diff-id=205,diff-type=noncompliant]
----
from passlib.hash import bcrypt

def hash_password(password):
    return bcrypt.using(rounds=2).hash(password) # Noncompliant
----

==== Compliant solution

[source,python,diff-id=205,diff-type=compliant]
----
from passlib.hash import bcrypt

def hash_password(password):
    return bcrypt.hash(password) # Noncompliant
----

=== How does this work?

include::../common/fix/password-hashing.adoc[]

Note that you should drop Passlib, as it is not maintained anymore. In case you
are forced to use Passlib, the following sections provide guidance on the usage
of these secure password-hashing algorithms.

include::../common/fix/argon-parameters.adoc[]

include::../common/fix/scrypt-parameters.adoc[]

include::../common/fix/bcrypt-parameters.adoc[]

include::../common/fix/pbkdf2-parameters.adoc[]

=== Pitfalls

include::../../common/pitfalls/pre-hashing.adoc[]

As a consequence, the
https://passlib.readthedocs.io/en/stable/lib/passlib.hash.bcrypt_sha256.html[passlib.hash.bcrypt_sha256]
construct and equivalents are not recommended.

=== Going the extra mile

include::../../common/extra-mile/peppering.adoc[]

