== How to fix it in Passlib

=== Code examples

==== Noncompliant code example

Code targeting Argon2:

[source,python,diff-id=205,diff-type=noncompliant]
----
from passlib.hash import argon2

def hash_password(password):
    algorithm = argon2.using(
        salt_size=8,
        digest_size=4,
        time_cost=1,
        memory_cost=8,
        parallelism=1) # Noncompliant

    return algorithm.hash(password)
----

Code targeting PBKDF2:

[source,python,diff-id=255,diff-type=noncompliant]
----
from passlib.hash import pbkdf2_sha256

def hash_password(password):
    algorithm = pbkdf2_sha256 # Noncompliant

    return algorithm.hash(password)
----

==== Compliant solution

For Argon2, this example uses OWASP's first recommended parameter set.

[source,python,diff-id=205,diff-type=compliant]
----
from passlib.hash import argon2

def hash_password(password):
    algorithm = argon2.using(
        type="id",
        salt_size=16,
        digest_size=32,
        time_cost=1,
        memory_cost=47104, # 46 MiB
        parallelism=1)

    return argon2.hash(password)
----

Code targeting PBKDF2:

[source,python,diff-id=255,diff-type=compliant]
----
from passlib.hash import pbkdf2_sha256

def hash_password(password):
    algorithm = pbkdf2_sha256.using(rounds=600_000)

    return algorithm.hash(password)
----

Passlib's default for PBKDF2 is not safe, as it uses a low number of rounds.
You should always specify the number of rounds to use.

=== How does this work?

include::../../common/fix/password-hashing.adoc[]

Note that Passlib is unmaintained and an up-to-date library should be used instead.
In case it is not possible to migrate to another library, the following sections provide
guidance on the usage of these secure password-hashing algorithms in Passlib.

include::../../common/fix/argon-parameters.adoc[]

To use values recommended by the Argon2 authors or by OWASP, you can call `using()` as follows:

[source, python]
----
# RFC 9106: High Memory Variant
RFC_9106_HIGH_MEMORY = argon2.using(
    type="id",
    salt_size=16,
    digest_size=32,
    time_cost=1,
    memory_cost=2097152,  # 2 GiB
    parallelism=4)

OWASP_1 = argon2.using(
        type="id",
        salt_size=16,
        digest_size=32,
        time_cost=1,
        memory_cost=47104, # 46 MiB
        parallelism=1)
----

include::../../common/fix/scrypt-parameters.adoc[]

To use values recommended by OWASP, you can call `using()` as follows:

[source,python]
----
OWASP_1 = scrypt.using(
        rounds=8,
        block_size=1<<17,
        parallelism=1)
----

include::../../common/fix/pbkdf2-parameters.adoc[]

=== Pitfalls

include::../../common/pitfalls/pre-hashing.adoc[]

As a consequence, the
https://passlib.readthedocs.io/en/stable/lib/passlib.hash.bcrypt_sha256.html[passlib.hash.bcrypt_sha256]
construct and equivalents are not recommended.

=== Going the extra mile

include::../../common/extra-mile/argon-cli.adoc[]

include::../../common/extra-mile/peppering.adoc[]

