== How to fix it in Passlib

=== Code examples

==== Noncompliant code example

[source,python,diff-id=205,diff-type=noncompliant]
----
from passlib.hash import argon2

def hash_password(password):
    algorithm = argon2.using(
        salt_size=8,
        digest_size=4,
        time_cost=1,
        memory_cost=8,
        parallelism=1) # Noncompliant

    return algorithm.hash(password)
----

==== Compliant solution

The code samples show how to fix the issue when using the Argon2 algorithm, but
in general the main recommendations are the same: +
Use default parameters unless you have a good reason not to.

This example uses OWASP's first recommended parameter set.

[source,python,diff-id=205,diff-type=compliant]
----
from passlib.hash import argon2

def hash_password(password):
    algorithm = argon2.using(
        type="id",
        salt_size=16,
        digest_size=32,
        time_cost=1,
        memory_cost=47104, # 46 MiB
        parallelism=1)

    return argon2.hash(password)
----

=== How does this work?

include::../common/fix/password-hashing.adoc[]

Note that you should drop Passlib, as it is not maintained anymore. In case you
are forced to use Passlib, the following sections provide guidance on the usage
of these secure password-hashing algorithms.

include::../common/fix/argon-parameters.adoc[]

The values recommended by the Argon2 authors are the two following:

[source, python]
----
# RFC 9106: High Memory Variant
RFC_9106_HIGH_MEMORY = argon2.using(
    type="id",
    salt_size=16,
    digest_size=32,
    time_cost=1,
    memory_cost=2097152,  # 2 GiB
    parallelism=4)

# RFC 9106: Low Memory Variant
RFC_9106_LOW_MEMORY = argon2.using(
    type="id",
    salt_size=16,
    digest_size=32,
    time_cost=3,
    memory_cost=65536,  # 64 MiB
    parallelism=4)
----

The values recommended by the OWASP are the following:

[source, python]
----
OWASP_1 = argon2.using(
        type="id",
        salt_size=16,
        digest_size=32,
        time_cost=1,
        memory_cost=47104, # 46 MiB
        parallelism=1)

OWASP_2 = argon2.using(
        type="id",
        salt_size=16,
        digest_size=32,
        time_cost=2,
        memory_cost=19456,  # 19 MiB
        parallelism=1)

OWASP_3 = argon2.using(
        type="id",
        salt_size=16,
        digest_size=32,
        time_cost=3,
        memory_cost=12288,  # 12 MiB
        parallelism=1)

OWASP_4 = argon2.using(
        type="id",
        salt_size=16,
        digest_size=32,
        time_cost=4,
        memory_cost=9216,  # 9 MiB
        parallelism=1)

OWASP_5 = argon2.using(
        type="id",
        salt_size=16,
        digest_size=32,
        time_cost=5,
        memory_cost=7168,  # 7 MiB
        parallelism=1)
----

include::../common/fix/scrypt-parameters.adoc[]

include::../common/fix/pbkdf2-parameters.adoc[]

=== Pitfalls

include::../../common/pitfalls/pre-hashing.adoc[]

As a consequence, the
https://passlib.readthedocs.io/en/stable/lib/passlib.hash.argon2_sha256.html[passlib.hash.bcrypt_sha256]
construct and equivalents are not recommended.

=== Going the extra mile

include::../../common/extra-mile/argon-cli.adoc[]

include::../../common/extra-mile/peppering.adoc[]

