== How to fix it in Scrypt

=== Code examples

==== Noncompliant code example

[source,python,diff-id=206,diff-type=noncompliant]
----
11
----

==== Compliant solution

[source,python,diff-id=206,diff-type=compliant]
----
22
----

=== How does this work?

include:: FIX_ME.adoc[]

=== Pitfalls

include:: FIX_ME.adoc[]

=== Going the extra mile

include:: FIX_ME.adoc[]

