== How to fix it in Scrypt

=== Code examples

==== Noncompliant code example

[source,python,diff-id=201,diff-type=noncompliant]
----
import scrypt

def hash_password(password, salt):
    return scrypt.hash(
        password,
        salt,
        N=1 << 10,
        r=8,
        p=2,
        buflen=64) # Noncompliant
----

==== Compliant solution

[source,python,diff-id=201,diff-type=compliant]
----
import scrypt

def hash_password(password):
    return scrypt.hash(
        password,
        salt,
        N=1 << 17,
        r=8,
        p=1,
        buflen=64)
----

=== How does this work?

include::../../common/fix/password-hashing.adoc[]

include::../../common/fix/scrypt-parameters.adoc[]

To use values recommended by OWASP, you can use an object crafted as follows:

[source,python]
----
OWASP_1 = {
    "n": 1 << 17,
    "r": 8,
    "p": 1,
    "dklen": 64,
}

# To use this example, you can use the dictionary as a ``**kwargs`` variable:
scrypt.hash(password, salt, **OWASP_1)
----


=== Pitfalls

include::../../common/pitfalls/pre-hashing.adoc[]

=== Going the extra mile

include::../../common/extra-mile/peppering.adoc[]

