== How to fix it in Scrypt

=== Code examples

==== Noncompliant code example

[source,python,diff-id=201,diff-type=noncompliant]
----
import scrypt

def hash_password(password, salt):
    return scrypt.hash(
        password,
        salt,
        N=1 << 10,
        r=8,
        p=2,
        buflen=64) # Noncompliant
----

==== Compliant solution

[source,python,diff-id=201,diff-type=compliant]
----
import scrypt

def hash_password(password):
    return scrypt.hash(
        password,
        salt,
        N=1 << 17,
        r=8,
        p=1,
        buflen=64)
----

=== How does this work?

include::../../common/fix/password-hashing.adoc[]

include::../../common/fix/scrypt-parameters.adoc[]

The values recommended by OWASP are the following:

[source,python]
----
OWASP_1 = {
    "N": 1 << 17,
    "r": 8,
    "p": 1,
    "buflen": 64,
}

OWASP_2 = {
    "N": 1 << 16,
    "r": 8,
    "p": 2,
    "buflen": 64,
}

OWASP_3 = {
    "N": 1 << 15,
    "r": 8,
    "p": 3,
    "buflen": 64,
}

OWASP_4 = {
    "N": 1 << 14,
    "r": 8,
    "p": 5,
    "buflen": 64,
}

OWASP_5 = {
    "N": 1 << 13,
    "r": 8,
    "p": 10,
    "buflen": 64,
}

# To use these examples, you can use the dictionnary as a ``**kwargs`` variable:
scrypt.hash(password, salt, **OWASP_1)
----


=== Pitfalls

include::../../common/pitfalls/pre-hashing.adoc[]

=== Going the extra mile

include::../../common/extra-mile/peppering.adoc[]

