<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when using the two-argument form of <code>raise</code> with an explicit <code>RuntimeError</code> class name.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>In Ruby, when you use the <code>raise</code> method with just a message string, it automatically raises a <code>RuntimeError</code> exception. This means that explicitly specifying <code>RuntimeError</code> as the exception class is redundant.</p>
</div>
<div class="paragraph">
<p>The two-argument form <code>raise RuntimeError, "message"</code> and the single-argument form <code>raise "message"</code> produce exactly the same result. The explicit specification of <code>RuntimeError</code> adds unnecessary verbosity to your code without providing any additional functionality or clarity.</p>
</div>
<div class="paragraph">
<p>Ruby&#8217;s design philosophy emphasizes conciseness and readability. By omitting the redundant <code>RuntimeError</code>, your code becomes more concise while maintaining the same behavior. This follows established Ruby style conventions and makes your code more idiomatic.</p>
</div>
<div class="paragraph">
<p>The redundant specification can also make the code slightly harder to read, as developers need to process the extra information that doesn&#8217;t add value. Clean, concise code is generally easier to understand and maintain.</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="paragraph">
<p>This issue has minimal impact on functionality since both forms produce identical behavior. However, it affects code readability and maintainability by introducing unnecessary verbosity. Following consistent style conventions helps maintain clean, readable codebases that are easier for teams to work with.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Remove the explicit <code>RuntimeError</code> class name and use only the message string. Ruby will automatically raise a <code>RuntimeError</code> when you provide just a message to the <code>raise</code> method.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-ruby" data-lang="ruby">raise RuntimeError, "Something went wrong" # Noncompliant</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-ruby" data-lang="ruby">raise "Something went wrong"</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Ruby Documentation - Exception - <a href="https://ruby-doc.org/core/Exception.html">Official Ruby documentation for the Exception class and raise method</a></p>
</li>
<li>
<p>Ruby Documentation - Kernel#raise - <a href="https://ruby-doc.org/core/Kernel.html#method-i-raise">Official documentation for the raise method showing different usage forms</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>