include::../introduction.adoc[]

== Why is this an issue?

include::../why.adoc[]

=== What is the potential impact?

include::../impact.adoc[]

== How to fix it

include::../how-to-fix.adoc[]

=== Code examples

==== Noncompliant code example

With an indent size of 2:

[source,vb6,diff-id=1,diff-type=noncompliant]
----
  Public Sub OutputObject(oOutput As Object)
    Set oObject = oOutput

  If TypeOf oObject Is Form Then
  oObject.Cls
  ElseIf TypeOf oObject Is PictureBox Then
        oObject.Cls
    ElseIf TypeOf oObject Is ListBox Then
        oObject.Clear
    End If
End Sub
----

==== Compliant solution

[source,vb6,diff-id=1,diff-type=compliant]
----
Public Sub OutputObject(oOutput As Object)
    Set oObject = oOutput

    If TypeOf oObject Is Form Then
        oObject.Cls
    ElseIf TypeOf oObject Is PictureBox Then
        oObject.Cls
    ElseIf TypeOf oObject Is ListBox Then
        oObject.Clear
    End If
End Sub
----

ifdef::env-github,rspecator-view[]

=== Going the extra mile

include::../extra-mile.adoc[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Correct the indentation of this and the following X lines. This line should start at column Y.


'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
