include::../description.adoc[]

== Noncompliant Code Example

With an indent size of 2:

----
  Public Sub OutputObject(oOutput As Object)
    Set oObject = oOutput

  If TypeOf oObject Is Form Then
  oObject.Cls
  ElseIf TypeOf oObject Is PictureBox Then
        oObject.Cls
    ElseIf TypeOf oObject Is ListBox Then
        oObject.Clear
    End If
End Sub
----

== Compliant Solution

----
Public Sub OutputObject(oOutput As Object)
    Set oObject = oOutput

    If TypeOf oObject Is Form Then
        oObject.Cls
    ElseIf TypeOf oObject Is PictureBox Then
        oObject.Cls
    ElseIf TypeOf oObject Is ListBox Then
        oObject.Clear
    End If
End Sub
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
