<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Adding <code>IBInspectable</code> to a properly-typed variable makes it available in Xcode&#8217;s Interface Builder. But that only works if variable type is declared explicitly as one of the following:</p>
</div>
<div class="ulist">
<ul>
<li>
<p><code>Int</code> types, <code>Double</code>, <code>Float</code>, <code>Bool</code></p>
</li>
<li>
<p><code>String</code> (or its optional)</p>
</li>
<li>
<p><code>CGFloat</code>, <code>CGPoint</code>, <code>CGSize</code>, <code>CGRect</code></p>
</li>
<li>
<p><code>UIColor</code>, <code>UIImage</code> (and their optionals)</p>
</li>
<li>
<p><code>NSString</code>,  <code>NSColor</code>, <code>NSImage</code> (and their optionals)</p>
</li>
<li>
<p><code>NSRect</code>, <code>NSPoint</code>, <code>NSSize</code>,</p>
</li>
</ul>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-swift" data-lang="swift">@IBInspectable  // Noncompliant; type is implicit
public var cornerRadius = 2.0 {
  didSet {
     //...
  }
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-swift" data-lang="swift">@IBInspectable
public var cornerRadius: CGFloat  = 2.0 {
  didSet {
     //...
  }
}</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="ulist">
<ul>
<li>
<p>Remove this "IBInspectable" attribute; it should only be applied to variables.</p>
</li>
<li>
<p>Remove this "IBInspectable" attribute or use [a supported|an explicit] type.</p>
</li>
</ul>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p><code>IBInspectable</code></p>
</div>
</div>
</div>
</div>