<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when a call to <code>Object.wait()</code> is not within a <code>while</code> loop.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Calls to <code>Object.wait()</code> should always be placed inside <code>while</code> loops to handle two critical concurrency scenarios properly.</p>
</div>
<div class="paragraph">
<p>First, the Java Virtual Machine can produce <strong>spurious wakeups</strong>. This means that a thread waiting on a condition can be awakened even when the condition has not actually been satisfied. If the <code>wait()</code> call is inside an <code>if</code> statement, the thread will proceed after the spurious wakeup without rechecking the condition, potentially leading to incorrect behavior.</p>
</div>
<div class="paragraph">
<p>Second, <strong>race conditions</strong> can occur when multiple threads are involved. Between the time a condition is checked and the <code>wait()</code> method is called, another thread might change the condition. If the condition was satisfied by another thread before <code>wait()</code> is invoked, the current thread could wait indefinitely for a condition that has already been met.</p>
</div>
<div class="paragraph">
<p>Using a <code>while</code> loop ensures that the condition is rechecked immediately after the thread wakes up, providing protection against both spurious wakeups and race conditions.</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="paragraph">
<p>When <code>wait()</code> calls are not properly guarded by <code>while</code> loops, applications can experience:</p>
</div>
<div class="ulist">
<ul>
<li>
<p><strong>Deadlocks</strong>: Threads may wait indefinitely for conditions that have already been satisfied</p>
</li>
<li>
<p><strong>Data corruption</strong>: Threads may proceed with operations when the required conditions are not actually met</p>
</li>
<li>
<p><strong>Unpredictable behavior</strong>: Spurious wakeups can cause threads to execute code at inappropriate times</p>
</li>
<li>
<p><strong>Race conditions</strong>: Multiple threads may interfere with each other&#8217;s execution in unexpected ways</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>These issues are particularly difficult to debug because they may occur intermittently and depend on timing and thread scheduling.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Replace the <code>if</code> statement with a <code>while</code> loop.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-groovy" data-lang="groovy">synchronized(data) {
    if (!data.isReady()) {
        data.wait() // Noncompliant
    }
    data.process()
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-groovy" data-lang="groovy">synchronized(data) {
    while (!data.isReady()) {
        data.wait() // Compliant
    }
    data.process()
}</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">

</div>
</div>