<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>When a loop variable is not used outside of a loop, it should be declared inside the loop declaration:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>It improves readability. The scope of the variable is clearly defined.</p>
</li>
<li>
<p>It reduces the number of mistakes. The variable can&#8217;t be accidentally misused outside of the loop.</p>
</li>
<li>
<p>Resources are not retained longer than necessary.</p>
</li>
</ul>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">void f() {
  int i = 0; // Noncompliant: i is not used outside of the loop
  for (i = 0; i &lt; 10; ++i) {
    std::cout &lt;&lt; i &lt;&lt; std::endl;
  }
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">void f() {
  for (int i = 0; i &lt; 10; ++i) {
    std::cout &lt;&lt; i &lt;&lt; std::endl;
  }
}</code></pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p>C&#43;&#43; Core Guidelines - <a href="https://github.com/isocpp/CppCoreGuidelines/blob/e49158a/CppCoreGuidelines.md#es5-keep-scopes-small">ES.5: Keep scopes small</a></p>
</li>
</ul>
</div>
<hr>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Declare the variable XXX inside the loop.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_is_related_to_s806">is related to: <a data-rspec-id="S806" class="rspec-auto-link">S806</a></h3>

</div>
</div>
</div>