Every section should be commented to explain its goal and how it works. This comment can be placed either just before or just after the section label.


== Noncompliant Code Example

----
    UNCOMMENTED-SECTION SECTION. 
----


== Compliant Solution

----
* Some comments
    CORRECTLY-COMMENTED-SECTION SECTION.     

    ANOTHER-CORRECTLY-COMMENTED-SECTION SECTION.  
* Some comments
----



ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
