== Why is this an issue?

Using ``++Thread.sleep++`` in a test is just generally a bad idea. It creates brittle tests that can fail unpredictably depending on environment ("Passes on my machine!") or load. Don't rely on timing (use mocks) or use libraries such as ``++Awaitility++`` for asynchroneous testing. 


=== Noncompliant code example

[source,java]
----
@Test
public void testDoTheThing(){

  MyClass myClass = new MyClass();
  myClass.doTheThing();

  Thread.sleep(500);  // Noncompliant
  // assertions...
}
----


=== Compliant solution

[source,java]
----
@Test
public void testDoTheThing(){

  MyClass myClass = new MyClass();
  myClass.doTheThing();

  await().atMost(2, Duration.SECONDS).until(didTheThing());  // Compliant
  // assertions...
}

private Callable<Boolean> didTheThing() {
  return new Callable<Boolean>() {
    public Boolean call() throws Exception {
      // check the condition that must be fulfilled...
    }
  };
}
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Remove this use of "Thread.sleep()".


'''
== Comments And Links
(visible only on this page)

=== on 12 May 2015, 14:52:51 David Gageot wrote:
lgtm

=== on 20 May 2015, 15:08:39 Nicolas Peru wrote:
Ok.

=== on 21 Aug 2015, 09:21:47 Michael Gumowski wrote:
WDYT about changes [~duarte.meneses]?

=== on 21 Aug 2015, 09:29:58 Duarte Meneses wrote:
Looks good.

Only suggestion I have is to put the timeout in the solution longer (2/3s?), otherwise it wouldn't fix the problem of dependency on the environment.

With awaitility we can use larger timeouts but still proceed quickly as soon as the polling conditions is satisfied.

=== on 21 Aug 2015, 09:35:55 Michael Gumowski wrote:
Ok, updated. Thanks for your feedback! Feels free to modify it directly next time!

endif::env-github,rspecator-view[]
