== Why is this an issue?

include::../whyAnIssue.adoc[]

=== What is the potential impact?

include::../impact.adoc[]

== How to fix it

include::../howToFIxIt.adoc[]

=== Code examples

==== Noncompliant code example

[source,csharp]
----
void CreateNewDateAndTime()
{
    var date = new DateTime(1994, 7, 5);
    var time = new DateTime(1, 1, 1, 16, 23, 42);
}
----

==== Compliant solution

[source,csharp]
----
void CreateNewDateAndTime()
{
    var date = new DateOnly(1994, 7, 5);
    var time = new TimeOnly(16, 23, 42);
}
----

== Resources
=== Documentation

include::../documentation.adoc[]
