include::../why-dotnet.adoc[]

include::../impact-dotnet.adoc[]

include::../how-dotnet.adoc[]

=== Code examples

==== Noncompliant code example

[source,csharp,diff-id=1,diff-type=noncompliant]
----
void CreateNewDateAndTime()
{
    var date = new DateTime(1994, 7, 5);
    var time = new DateTime(1, 1, 1, 16, 23, 42);
}
----

==== Compliant solution

[source,csharp,diff-id=1,diff-type=compliant]
----
void CreateNewDateAndTime()
{
    var date = new DateOnly(1994, 7, 5);
    var time = new TimeOnly(16, 23, 42);
}
----

include::../resources-dotnet.adoc[]
