Every subroutine(``++FORM++``) should be commented to explain its goal and how it works. This comment can be located just before or after the form definition.


== Noncompliant Code Example

----
FORM my_form.
  ...
ENDFORM 
----


== Compliant Solution

----
* here is  
* my comment
FORM my_form.
  ...
ENDFORM 
----


or

----
FORM my_form.
* here is  
* my comment
  ...
ENDFORM 
----

