The number of ``++RECORDS++`` or ``++CHARACTERS++`` specified in a ``++BLOCK CONTAINS++`` clause is used to determine block size. Specify ``++10 RECORDS++``, and the block will be exactly 10x the length of the record. But that may not be the right size, depending on the environment. Instead, it is considered a best practice to specify ``++0 RECORDS++``, so the block size will be calculated automatically.


== Noncompliant Code Example

----
       FD OUTFILE1
           BLOCK CONTAINS 32760 RECORDS     >* Noncompliant
           RECORDING MODE V.
       FD OUTFILE2
           BLOCK CONTAINS 1024 CHARACTERS.  >* Noncompliant
----


== Compliant Solution

----
       FD OUTFILE1
           BLOCK CONTAINS 0 RECORDS
           RECORDING MODE V.
       FD OUTFILE2
           BLOCK CONTAINS 0 RECORDS.
----


== Exceptions

``++0 CHARACTERS++`` is compliant.

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

include::highlighting.adoc[]

endif::env-github,rspecator-view[]
