== Why is this an issue?

Throwing ``++swap++``, ``++move operator++`` or ``++default constructor++`` make it very difficult to handle exception or can lead to undefined/inconsistent state.

 If exceptions can be thrown there:

* Creation and destruction of local variables allocated on the stack can throw. It makes that any scope should be protected by a ``++try-catch++`` block, in your own code and in all the dependencies.

* If a move operator/constructor could throw, objects could be left in an inconsistent states as partially moved (just a few members moved). In what state is the source object? the target one?
 Copy does not have this problem as the source object is not modified. It is then much easier to recover from an exception.


* For the same sort of reason as above, a throwing swap could create inconsistent states. If a swap throws, what are the states of the 2 objects. Was any state lost in the process
 The comparison with copy still holds.


Notes:

* If a ``++noexcept++`` function tries to raise an exception, the program will terminate. For the above case, it might often be the best way to cope with an exception arising in a default constructor, a swap or a move.

* If a move constructor/operator can throw, the STL will replace it with a copy if available.

* ``++std::swap++`` is required to be ``++noexcept++``


=== Noncompliant code example

[source,cpp]
----
class S {};
void swap(S& s1, S& s2);
----


=== Compliant solution

[source,cpp]
----
class S {};
void swap(S& s1, S& s2) noexcept;
----


== Resources

* {cpp} reference - https://en.cppreference.com/w/cpp/language/noexcept_spec[Noexcept specifier]
* http://blogs.microsoft.co.il/sasha/2014/08/08/make-move-constructors-no-throw/[Make move constructor nothrow]


ifdef::env-github,rspecator-view[]
'''
== Comments And Links
(visible only on this page)

=== is related to: S1048

=== duplicates: S5018

=== on 28 Aug 2019, 21:37:24 Loïc Joly wrote:
\[~geoffray.adde] I agree for the ``++move++`` part, but we already have RSPEC-5018 for that. I kind of agree for the swap part, but I think it would fit perfectly within RSPEC-5018 that we could just extend a little.

For constructor, I'm not so sure:

* "Creation and destruction of local variables allocated on the stack can throw": Yes, and making the default constructor noexcept will not prevent this
* "It makes that any scope should be protected by a try-catch block": I don't see why
It might be desirable that the moved-from state of an object is equivalent to the default constructed one, and for that, having a noexcept default constructor might help, but this looks more like a low priority code smell to me.


What do you think?




=== on 3 Sep 2019, 10:22:56 Loïc Joly wrote:
After discussion with [~geoffray.adde]: Ok to add swap to RPSEC-5018, and drop the default constructor part. Which means this RSPEC is no longer needed.

endif::env-github,rspecator-view[]
