== Why is this an issue?

Throwing ``++swap++``, ``++move operator++`` or ``++default constructor++`` make it very difficult to handle exception or can lead to undefined/inconsistent state.

 If exceptions can be thrown there:

* Creation and destruction of local variables allocated on the stack can throw. It makes that any scope should be protected by a ``++try-catch++`` block, in your own code and in all the dependencies.

* If a move operator/constructor could throw, objects could be left in an inconsistent states as partially moved (just a few members moved). In what state is the source object? the target one?
 Copy does not have this problem as the source object is not modified. It is then much easier to recover from an exception.


* For the same sort of reason as above, a throwing swap could create inconsistent states. If a swap throws, what are the states of the 2 objects. Was any state lost in the process
 The comparison with copy still holds.


Notes:

* If a ``++noexcept++`` function tries to raise an exception, the program will terminate. For the above case, it might often be the best way to cope with an exception arising in a default constructor, a swap or a move.

* If a move constructor/operator can throw, the STL will replace it with a copy if available.

* ``++std::swap++`` is required to be ``++noexcept++``


=== Noncompliant code example

[source,cpp]
----
class S {};
void swap(S& s1, S& s2);
----


=== Compliant solution

[source,cpp]
----
class S {};
void swap(S& s1, S& s2) noexcept;
----


== Resources

* https://en.cppreference.com/w/cpp/language/noexcept_spec[Noexcept specifier]
* http://blogs.microsoft.co.il/sasha/2014/08/08/make-move-constructors-no-throw/[Make move constructor nothrow]


ifdef::env-github,rspecator-view[]
'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
