<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when <code>Array.new</code> or <code>Hash.new</code> is used without parameters to create empty collections.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Using constructor methods like <code>Array.new</code> and <code>Hash.new</code> without parameters is unnecessarily verbose compared to the literal syntax <code>[]</code> and <code>{}</code>. The literal syntax is more concise, widely recognized as idiomatic Ruby, and preferred by the Ruby community.</p>
</div>
<div class="paragraph">
<p>Literal syntax makes the code more readable and consistent with Ruby conventions. When no parameters are needed, the constructor methods provide no additional functionality over literals, making them redundant.</p>
</div>
<div class="paragraph">
<p>This pattern is particularly important in Ruby codebases where consistency and readability are valued. Most Ruby style guides, including those from major organizations, recommend using literal syntax for empty collections.</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="paragraph">
<p>This issue affects code readability and maintainability. While it doesn&#8217;t impact functionality, inconsistent style can make code harder to read and maintain, especially in team environments where coding standards matter.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Replace <code>Array.new</code> with <code>[]</code> when creating an empty array without parameters.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-ruby" data-lang="ruby">arr = Array.new # Noncompliant</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-ruby" data-lang="ruby">arr = []</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Ruby Array Documentation - <a href="https://ruby-doc.org/core/Array.html">Official Ruby documentation for Array class and literal syntax</a></p>
</li>
<li>
<p>Ruby Hash Documentation - <a href="https://ruby-doc.org/core/Hash.html">Official Ruby documentation for Hash class and literal syntax</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>