include::../description.adoc[]

== Noncompliant Code Example

[source,csharp]
----
public class Outer
{
    public void OuterMethod() 
    {
        Console.WriteLine("Called from an outer method.");
    }

    private static void Print(int num) // Noncompliant - method is only used by inner class, should be moved there
    {
        Console.WriteLine(num);
    }

    public class Inner
    {
        public void SomeMethod()
        {
            Outer.Print(42);
        }
    }
}
----

== Compliant Solution

[source,csharp]
----
public class Outer
{
    public void OuterMethod() 
    {
        Console.WriteLine("Called from an outer method.");
    }

    public class Inner
    {
        public void SomeMethod()
        {
            Print(42);
        }

        private static void Print(int num)
        {
            Console.WriteLine(num);
        }
    }
}
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../highlighting.adoc[]

endif::env-github,rspecator-view[]