<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>There are two ways to define asynchronous functions in Kotlin:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>using the modifier <code>suspend</code> in the function declaration</p>
</li>
<li>
<p>creating an extension function on <code>CoroutineScope</code> (or passing it as a parameter)</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>The <code>suspend</code> modifier is generally used for functions that might take some time to complete. The caller coroutine might be potentially suspended.</p>
</div>
<div class="paragraph">
<p>Functions that return results immediately but start a coroutine in the background should be written as extension functions on <code>CoroutineScope</code>. At the same time, these functions should not be declared <code>suspend</code>, as suspending functions should not leave running background tasks behind.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-kotlin" data-lang="kotlin">suspend fun CoroutineScope.f(): Int {
    val resource1 = loadResource1()
    val resource2 = loadResource2()
    return resource1.size + resource2.size
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="paragraph">
<p>Using <code>suspend</code>:</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-kotlin" data-lang="kotlin">suspend fun f(): Int {
    val resource1 = loadResource1()
    val resource2 = loadResource2()
    return resource1.size + resource2.size
}</code></pre>
</div>
</div>
<div class="paragraph">
<p>Using extension on <code>CoroutineScope</code>:</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-kotlin" data-lang="kotlin">fun CoroutineScope.f(): Deferred&lt;Int&gt; = async {
    val resource1 = loadResource1()
    val resource2 = loadResource2()
    resource1.size + resource2.size
}</code></pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p><a href="https://elizarov.medium.com/coroutine-context-and-scope-c8b255d59055">Coroutine Context and Scope</a></p>
</li>
</ul>
</div>
</div>
</div>