== Why is this an issue?

Whenever more than one line needs to be read, inserted or deleted from a database table, it is more efficient to work with an internal table than to read, insert or delete the lines one by one inside a loop.


=== Noncompliant code example

[source,abap]
----
LOOP AT TAB INTO TAB_WA.
  INSERT INTO CUSTOMERS VALUES TAB_WA.
ENDLOOP.
----


=== Compliant solution

[source,abap]
----
INSERT CUSTOMERS FROM TABLE TAB.
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Remove this "XXXX" SQL statement from the nesting loop and use an internal table to get the same result


endif::env-github,rspecator-view[]
