include::../description.adoc[]

== Noncompliant Code Example

With the default value for the ``++legalTrailingCommentPattern++`` parameter: ``++^'\s*[^\s]+$++``


----
a = b + c   ' This is a trailing comment that could be very very long  -- Noncompliant
d = a + 9 REM This is another trailing comment which could be very very long  -- Noncompliant
e = a * b ' FIXME  -- allowed
----

== Compliant Solution

----
' This very long comment is better placed before the line of code
a = b + c
REM This is very long comment which is better placed before the line of code
d = a + 9
e = a * b ' FIXME 
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../parameters.adoc[]

'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
