== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

With the default value for the ``++legalTrailingCommentPattern++`` parameter: ``++^'\s*[^\s]+$++``


[source,vb6]
----
a = b + c   ' This is a trailing comment that could be very very long  -- Noncompliant
d = a + 9 REM This is another trailing comment which could be very very long  -- Noncompliant
e = a * b ' FIXME  -- allowed
----

=== Compliant solution

[source,vb6]
----
' This very long comment is better placed before the line of code
a = b + c
REM This is very long comment which is better placed before the line of code
d = a + 9
e = a * b ' FIXME 
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

=== Parameters

.legalTrailingCommentPattern
****

----
^'\s*+[^\s]++$
----

Pattern for text of trailing comments that are allowed. By default, comments containing only one word.
****


'''
== Comments And Links
(visible only on this page)

=== on 9 Feb 2018, 20:02:15 Ann Campbell wrote:
\[~michael.gumowski] it seems odd to me to list a parameter in an exception.

=== on 12 Feb 2018, 08:42:53 Michael Gumowski wrote:
\[~ann.campbell.2] I was loosing this exception when synchronizing RSPECs with HTML. I simply added it back in order to keep this behavior visible the same way it was described before (and to not loose it from in the generated html). If you see a better way to mention that behavior, I let you rephrase it, and I'll resync.

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
