include::../description.adoc[]

== Noncompliant Code Example

----
DATA: MyText TYPE string.
----

== Compliant Solution

----
DATA: my_text TYPE string.
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../parameters.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
