== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,abap]
----
DATA: MyText TYPE string.
----

=== Compliant solution

[source,abap]
----
DATA: my_text TYPE string.
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../parameters.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
