== How to fix it in Python Standard Library

=== Code examples

:canonicalization_function1: os.path.abspath
:canonicalization_function2: os.path.realpath

include::../../common/fix/code-rationale.adoc[]

==== Noncompliant code example

[source,python,diff-id=1,diff-type=noncompliant]
----
import os
import zipfile

def extract_zip(zip_path, target_dir):
    with zipfile.ZipFile(zip_path, 'r') as zf:
        for entry in zf.filelist:
            dest = os.path.join(target_dir, entry.filename)  # Noncompliant
            with zf.open(entry) as src, open(dest, 'wb') as out:
                out.write(src.read())
----

==== Compliant solution

[source,python,diff-id=1,diff-type=compliant]
----
import os
import zipfile

def extract_zip(zip_path, target_dir):
    canonical_target = os.path.abspath(target_dir) + os.sep
    with zipfile.ZipFile(zip_path, 'r') as zf:
        for entry in zf.filelist:
            dest = os.path.join(target_dir, entry.filename)
            if not os.path.abspath(dest).startswith(canonical_target):
                raise ValueError("Path traversal detected")
            with zf.open(entry) as src, open(dest, 'wb') as out:
                out.write(src.read())
----

=== How does this work?

include::../../common/fix/how-does-this-work.adoc[]

