When calling `toString()` or coercing into a string an object that doesn't implement its own `toString` method, it returns `[object Object]` which is often not what was intended.

== Why is this an issue?

When using an object into a string context, one would probably want to expose its properties, so obtaining `[object Object]` is probably not the intended behaviour.

== How to fix it

You can simply define a `toString()` method for the object or class.

=== Code examples

==== Noncompliant code example

[source,javascript,diff-id=1,diff-type=noncompliant]
----
class Foo {};
const bar = new Foo();

bar + '';
`Foo: ${bar}`;
bar.toString();
----

==== Compliant solution

[source,javascript,diff-id=1,diff-type=compliant]
----
class Foo {
  toString() {
    return 'Foo';
  }
}
const bar = new Foo();

bar + '';
`Foo: ${bar}`;
bar.toString();
----

==== Noncompliant code example

[source,javascript,diff-id=2,diff-type=noncompliant]
----
const foo = {};
foo + '';
`Foo: ${foo}`;
foo.toString();
----

==== Compliant solution

[source,javascript,diff-id=2,diff-type=compliant]
----
const foo = {
  toString: () => {
    return 'Foo';
  }
}
foo + '';
`Foo: ${foo}`;
foo.toString();
----

//== Resources
//=== Documentation
//=== Articles & blog posts
//=== Conference presentations
//=== Standards
