<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Using <code>LOOP...INTO</code>  with <code>MODIFY</code> statements will put the required record in a work area, process it and put it back in the internal table. It is more efficient to modify the internal table directly by using <code>LOOP...ASSIGNING</code> and field-symbols.</p>
</div>
<div class="paragraph">
<p>This rule raises an issue when a <code>LOOP...INTO</code> contains one or more <code>MODIFY</code> statements.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-abap" data-lang="abap">LOOP AT i_bseg INTO wa_bseg.
    ...
    wa_bseg-sgtxt = 'VALUE'.
    MODIFY i_bseg FROM wa_bseg.
ENDLOOP.</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-abap" data-lang="abap">LOOP AT i_bseg ASSIGNING &lt;fs_bseg&gt;.
  ...
  &lt;fs_bseg&gt;-sgtxt = 'VALUE'.
ENDLOOP.</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Replace this "LOOP INTO" and "MODIFY" with "LOOP ASSIGNING" and field-symbols.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>Primary: the whole LOOP&#8230;&#8203;INTO statement, but not the body of the loop.</p>
</div>
<div class="paragraph">
<p>Secondary:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>Every "MODIFY" statement</p>
</li>
<li>
<p>message: replace this "MODIFY" with a field-symbol</p>
</li>
</ul>
</div>
</div>
</div>
</div>