<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Loggers should be:</p>
</div>
<div class="ulist">
<ul>
<li>
<p><code>private</code>: not accessible outside of their parent classes. If another class needs to log something, it should instantiate its own logger.</p>
</li>
<li>
<p><code>static</code>: not dependent on an instance of a class (an object). When logging something, contextual information can of course be provided in the messages but the logger should be created at class level to prevent creating a logger along with each object.</p>
</li>
<li>
<p><code>const</code>: created once and only once per class.</p>
</li>
</ul>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="paragraph">
<p>With the default regular expression <code>LOG(?:GER)?</code>:</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-flex" data-lang="flex">public const logger:ILogger = LogUtil.getLogger(MyClass);</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-flex" data-lang="flex">private static const LOG:ILogger = LogUtil.getLogger(MyClass);</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Make the logger "XXXX" private static [final|const] and rename it to comply with the format "{}".</p>
</div>
<div class="paragraph">
<p>Make the logger "XXXX" private static [final|const].</p>
</div>
<div class="paragraph">
<p>Rename the "XXX" logger to comply with the format "{}".</p>
</div>
</div>
<div class="sect2">
<h3 id="_parameters">Parameters</h3>
<div class="sidebarblock">
<div class="content">
<div class="title">format</div>
<div class="paragraph">
<p><em>String</em></p>
</div>
<div class="listingblock">
<div class="content">
<pre>LOG(?:GER)?</pre>
</div>
</div>
<div class="paragraph">
<p>Regular expression used to check the logger names against.</p>
</div>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_22_nov_2013_125331_freddy_mallet_wrote">on 22 Nov 2013, 12:53:31 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>Is implemented by http://jira.codehaus.org/browse/SONARPLUGINS-3280 for Flex</p>
</div>
</div>
</div>
</div>