== Why is this an issue?

Regardless of the logging framework in use (Microsoft.Extension.Logging, Serilog, Log4net, NLog, ...â€‹), logger fields should be:

* *{private}*: this restricts access to the logger from outside the enclosing type (class, struct, record...). Using any other access modifier would allow other types to use the logger to log messages in the type where it's defined.
* *{static}*: making the logger field `{static}` will ensure that the lifetime of the object doesn't depend on the lifetime of the instance of the enclosing type.
* *{readonly}*: marking the field as `{readonly}` will prevent modifications to the reference of the logger. This ensures that the reference to the logger remains consistent and doesn't get accidentally reassigned during the lifetime of the enclosing type.

This rule should be activated when https://en.wikipedia.org/wiki/Service_locator_pattern[Service Locator Design pattern] is followed in place of https://en.wikipedia.org/wiki/Dependency_injection[Dependency Injection] for logging.

The rule supports the most popular logging frameworks:

* https://www.nuget.org/packages/Microsoft.Extensions.Logging[Microsoft.Extensions.Logging]
* https://www.nuget.org/packages/Serilog[Serilog]
* https://www.nuget.org/packages/Castle.Core[Castle.Core]
* https://www.nuget.org/packages/NLog[NLog]
* https://www.nuget.org/packages/log4net[log4net]

== How to fix it

Make the logging field `{private static readonly}`.

=== Noncompliant code example

include::{language}/noncompliant.adoc[]

=== Compliant solution

include::{language}/compliant.adoc[]

== Resources

=== Documentation

include::{language}/resources.adoc[]

* Wikipedia - https://en.wikipedia.org/wiki/Service_locator_pattern[Service locator pattern]
* Wikipedia - https://en.wikipedia.org/wiki/Dependency_injection[Dependency injection]

=== Articles & blog posts

* https://stackoverflow.com/questions/968132/c-sharp-private-static-and-readonly[C# `private`, `static`, and `readonly`]

include::rspecator-dotnet.adoc[]