### Summary

In programming, a module should not import itself. This means that a file or a component within your 
code should not try to include or reference itself. When a module imports itself, it can lead to confusion, 
errors, and unexpected behavior in your program. This issue is often a sign of poor code organization and 
can make your code harder to understand and maintain. By avoiding self-imports, you ensure that your code 
remains clean, efficient, and easy to follow.

== Why is this an issue?

When a module imports itself, it essentially performs a no-operation (noop). This means that the import 
statement does nothing useful and serves no purpose.

== How to fix it

Remove the self-import statement.

=== Code examples

==== Noncompliant code example

[source,javascript,diff-id=1,diff-type=noncompliant]
----
// foo.js
import foo from './foo';

const foo = require('./foo');
----

[source,javascript,diff-id=2,diff-type=noncompliant]
----
// index.js
import index from '.';

const index = require('.');
----

==== Compliant solution

[source,javascript,diff-id=1,diff-type=compliant]
----
// foo.js
import bar from './bar';

const bar = require('./bar');
----


== Resources
=== Documentation

* MDN web docs - https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Modules[Modules]
* MDN web docs - https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/import[import]
* Node.js docs - https://nodejs.org/api/modules.html#requireid[Node.js require]