== Why is this an issue?

By definition, primitive types are not Objects and so they can't be ``++null++``. Adding ``++@CheckForNull++`` or ``++@Nullable++`` on primitive types adds confusion and is useless.


This rule raises an issue when ``++@CheckForNull++`` or ``++@Nullable++`` is set on a method returning a primitive type: byte, short, int, long, float, double, boolean, char.


=== Noncompliant code example

[source,java]
----
@CheckForNull
boolean isFoo() {
 ...
}
----


=== Compliant solution

[source,java]
----
boolean isFoo() {
 ...
}
----



ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

include::highlighting.adoc[]

'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
