Because the value returned never changes, it is inefficient to call ``++Asc++``/``++AscW++`` on a String constant. Use the numeric value instead.


== Noncompliant Code Example

----
intCode = Asc("*")
----


== Compliant Solution

----
intCode = 42
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

endif::env-github,rspecator-view[]
