== Why is this an issue?

Because the value returned never changes, it is inefficient to call ``++Asc++``/``++AscW++`` on a String constant. Use the numeric value instead.


=== Noncompliant code example

[source,vb6]
----
intCode = Asc("*")
----


=== Compliant solution

[source,vb6]
----
intCode = 42
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

endif::env-github,rspecator-view[]
