<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Directives in the php.ini files can be of type <code>boolean</code>, <code>integer</code> or <code>string</code>.</p>
</div>
<div class="paragraph">
<p>For <code>boolean</code> acceptable values are <code>0</code>, <code>1</code>, <code>true</code>, <code>false</code>, <code>yes</code>, <code>no</code>, <code>on</code> and <code>off</code>.</p>
</div>
<div class="paragraph">
<p>For <code>integers</code> they can be qualified with <code>k</code>, <code>m</code> or <code>g</code>. E.g. 16k means 16000.</p>
</div>
<div class="paragraph">
<p>The complete list of directive is at http://php.net/manual/en/ini.list.php</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-php" data-lang="php">allow_url_include = 42; Noncompliant, should be a boolean
max_execution_time = "1024"; Noncompliant, should be an integer</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-php" data-lang="php">allow_url_include = false
max_execution_time = 1024</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Change that directive to be of the proper type.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_20_oct_2016_163254_pierre_yves_nicolas_wrote">on 20 Oct 2016, 16:32:54 Pierre-Yves Nicolas wrote:</h3>
<div class="paragraph">
<p>Looks good to me.</p>
</div>
</div>
</div>
</div>