Directives in the php.ini files can be of type ``boolean``, ``integer`` or ``string``.
For ``boolean`` acceptable values are ``0``, ``1``, ``true``, ``false``, ``yes``, ``no``, ``on`` and ``off``.
For ``integers`` they can be qualified with ``k``, ``m`` or ``g``. E.g. 16k means 16000.
The complete list of directive is at http://php.net/manual/en/ini.list.php


== Noncompliant Code Example

----
allow_url_include = 42; Noncompliant, should be a boolean
max_execution_time = "1024"; Noncompliant, should be an integer
----


== Compliant Solution

----
allow_url_include = false
max_execution_time = 1024
----

