== Why is this an issue?

The standard order for ``++using++`` directives is alphabetic with the exception of ``++System++`` directives, which come first for higher visibility. Using a different order may cause maintainers to overlook a directive or misunderstand what's being used.


=== Noncompliant code example

[source,csharp]
----
using C;  // Noncompliant
using System.A.A;
using A;
using D;
using B;
using System;
using System.A;
using System.B;
----


=== Compliant solution

[source,csharp]
----
using System;
using System.A;
using System.A.A;
using System.B;
using A;
using C;
using B;
using D;
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Reorder these "using" directives to put "System" directives first, then sub-sort alphabetically.


=== Highlighting

first out-of-order directive


endif::env-github,rspecator-view[]
