== Why is this an issue?

The meaning of a boolean parameter may seem perfectly clear when you first write a method call, but that meaning is likely to fade for you over time, and could be completely opaque to those who come behind you.


Instead, object literals should be used.


=== Noncompliant code example

[source,javascript]
----
widget.repaint(false);  // Noncompliant; does this mean never repaint?
----


=== Compliant solution

[source,javascript]
----
widget.repaint({immediate: false});
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
