== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,sql]
----
SET SERVEROUTPUT ON

DECLARE
  foo BOOLEAN := TRUE;
BEGIN
  IF foo = FALSE THEN                     -- Noncompliant
    DBMS_OUTPUT.PUT_LINE('foo = false!');
  ELSIF foo = TRUE THEN                   -- Noncompliant
    DBMS_OUTPUT.PUT_LINE('foo = true!');
  END IF;
END;
/
----

=== Compliant solution

[source,sql]
----
DECLARE
  foo BOOLEAN := TRUE;
BEGIN
  IF NOT foo THEN                         -- Compliant
    DBMS_OUTPUT.PUT_LINE('foo = false!');
  ELSIF foo THEN                          -- Compliant
    DBMS_OUTPUT.PUT_LINE('foo = true!');
  END IF;
END;
/
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
