== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,javascript]
----
for (var i = 0; i < length; i++) {}  // Empty on purpose or missing piece of code ?
----

=== Exceptions

When a block contains a comment, this block is not considered to be empty. Moreover ``++catch++`` blocks are ignored.

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
