<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when Rails MVC layers are improperly mixed: models include view helpers or call controller methods, view helpers perform database queries or contain business logic, or controllers mix presentation logic with data access.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>The Model-View-Controller (MVC) pattern is fundamental to Rails architecture and provides clear separation of concerns. Each layer has a specific responsibility:</p>
</div>
<div class="ulist">
<ul>
<li>
<p><strong>Models</strong> handle data and business logic</p>
</li>
<li>
<p><strong>Views</strong> handle presentation and user interface</p>
</li>
<li>
<p><strong>Controllers</strong> orchestrate between models and views</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>When these boundaries are violated, several problems arise:</p>
</div>
<div class="paragraph">
<p><strong>Tight Coupling</strong>: Including view helpers in models creates unnecessary dependencies between the data layer and presentation layer. This makes the code harder to test, maintain, and refactor.</p>
</div>
<div class="paragraph">
<p><strong>Runtime Errors</strong>: View helpers often depend on controller context (like <code>params</code>, <code>request</code>, or <code>current_user</code>). When called from models, these dependencies may be missing, causing <code>NoMethodError</code> or <code>undefined method</code> exceptions.</p>
</div>
<div class="paragraph">
<p><strong>Poor Testability</strong>: Models with view dependencies become harder to unit test because they require view context to be set up. This violates the principle that models should be testable in isolation.</p>
</div>
<div class="paragraph">
<p><strong>Namespace Pollution</strong>: Including helper modules in models adds many unrelated methods to the model&#8217;s interface, making it confusing and harder to understand.</p>
</div>
<div class="paragraph">
<p><strong>Performance Issues</strong>: Database queries in view helpers can lead to N+1 query problems and make it difficult to optimize data loading. View rendering becomes slower and less predictable.</p>
</div>
<div class="paragraph">
<p><strong>Maintenance Burden</strong>: When presentation logic is scattered across models and helpers, making UI changes requires touching multiple layers, increasing the risk of introducing bugs.</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="paragraph">
<p>Violating MVC separation leads to code that is difficult to maintain, test, and debug. Applications become more fragile as changes in one layer unexpectedly break functionality in another. Performance can degrade due to inefficient data loading patterns, and development velocity slows as developers struggle with tightly coupled code.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it_in_rails">How to fix it in Rails</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Move formatting logic from models to presenter objects. Presenters handle the presentation concerns while keeping models focused on data and business logic.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-ruby" data-lang="ruby"># Model including view helpers
class Job &lt; ActiveRecord::Base
  include ActionView::Helpers::NumberHelper
  include ApplicationHelper

  def details
    "Only #{number_to_currency(part_amount_received)} received." # Noncompliant
  end
end</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-ruby" data-lang="ruby"># Use presenter for formatting
class JobPresenter
  include ActionView::Helpers::NumberHelper

  def initialize(job)
    @job = job
  end

  def details
    "Only #{number_to_currency(@job.part_amount_received)} received."
  end
end

class Job &lt; ActiveRecord::Base
  def total_received_percentage
    (part_amount_received.to_f / price) * 100
  end
end</code></pre>
</div>
</div>
<div class="paragraph">
<p>Move database queries and business logic from helpers to models. Keep helpers focused on formatting and presentation logic only.</p>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_code_examples_2">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example_2">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-ruby" data-lang="ruby"># Helper with database queries and business logic
module BooksHelper
  def book_count_for_author(author)
    Book.where(author: author).count # Noncompliant
  end

  def featured_products
    Product.where(featured: true).limit(5).order(:created_at) # Noncompliant
  end
end</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution_2">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-ruby" data-lang="ruby"># Move queries to model, keep formatting in helper
class Book &lt; ActiveRecord::Base
  def self.count_for_author(author)
    where(author: author).count
  end
end

module BooksHelper
  def format_book_count(author, book_count)
    pluralize(book_count, 'book') + " by #{author}"
  end
end</code></pre>
</div>
</div>
<div class="paragraph">
<p>Pass data from controllers to models instead of having models call controller methods. This maintains proper data flow and eliminates tight coupling.</p>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_code_examples_3">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example_3">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-ruby" data-lang="ruby"># Model calling controller methods
class Order &lt; ActiveRecord::Base
  def transfer
    active = ApplicationController.helpers.get_active_gateway(self.cart) # Noncompliant
  end
end</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution_3">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-ruby" data-lang="ruby"># Pass data from controller to model
class ApplicationController &lt; ActionController::Base
  def show
    @order = Order.find(params[:id])
    active_gateway = get_active_gateway(@order.cart)
    @order.transfer(active_gateway)
  end
end

class Order &lt; ActiveRecord::Base
  def transfer(active_gateway)
    # Use passed parameter
  end
end</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Rails MVC Architecture Guide - <a href="https://guides.rubyonrails.org/getting_started.html#mvc-and-you">Official Rails guide explaining the MVC pattern and proper layer separation</a></p>
</li>
<li>
<p>Rails View Helpers Best Practices - <a href="https://avohq.io/glossary/helpers">Comprehensive guide on proper use of view helpers and avoiding common pitfalls</a></p>
</li>
<li>
<p>Presenter Pattern in Rails - <a href="https://nithinbekal.com/posts/rails-presenters/">Guide on using presenter objects to handle view logic without violating MVC separation</a></p>
</li>
</ul>
</div>
</div>
<div class="sect2">
<h3 id="_standards">Standards</h3>
<div class="ulist">
<ul>
<li>
<p>SOLID Principles - Single Responsibility Principle - <a href="https://en.wikipedia.org/wiki/Single-responsibility_principle">Each class should have only one reason to change, supporting proper separation of concerns</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>