include::../description.adoc[]

== Noncompliant Code Example

----
if ($x > 0)  // Noncompliant
doTheThing();
doTheOtherThing();

foo();
----

== Compliant Solution

----
if ($x > 0) {
  doTheThing();
  doTheOtherThing();
}

foo();
----
or

----
if ($x > 0)
    doTheThing();
doTheOtherThing();

foo();
----

ifdef::env-github,rspecator-view[]
'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
