include::../description.adoc[]

== Noncompliant Code Example

[source,php]
----
if ($x > 0)  // Noncompliant
doTheThing();
doTheOtherThing();

foo();
----

== Compliant Solution

[source,php]
----
if ($x > 0) {
  doTheThing();
  doTheOtherThing();
}

foo();
----
or

[source,php]
----
if ($x > 0)
    doTheThing();
doTheOtherThing();

foo();
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../highlighting.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
