<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Having a manual <code>PartialEq</code> implementation for types with a derived <code>Hash</code> can lead to inconsistencies. The contract <code>k1 == k2 â‡’ hash(k1) == hash(k2)</code> must always hold. Inconsistencies can cause undefined behaviors, especially when these types are used in collections such as <code>HashMap</code>.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-rust" data-lang="rust">#[derive(Hash)]
struct Foo;

impl PartialEq for Foo {
    fn eq(&amp;self, other: &amp;Self) -&gt; bool {
        // Some custom equality logic
        true // Noncompliant
    }
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-rust" data-lang="rust">#[derive(Hash, PartialEq)]
struct Foo;</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Clippy Lints - <a href="https://rust-lang.github.io/rust-clippy/master/index.html#derived_hash_with_manual_eq" class="bare">https://rust-lang.github.io/rust-clippy/master/index.html#derived_hash_with_manual_eq</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>