include::../description.adoc[]

== How to fix it

=== Code examples

==== Noncompliant code example

[source,php,diff-id=1,diff-type=noncompliant]
----
function foo() {
    if (true) { // Noncompliant: the condition is always true
        return 1;
    } else {
        return 2;
    }
}
----

==== Compliant solution

[source,php,diff-id=1,diff-type=compliant]
----
function foo() {
    $a = bar();
    if ($a) {
        return 1;
    } else {
        return 2;
    }
}
----

include::../rule.adoc[]
