== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,javascript]
----
const pattern = /Hello,   world!/;
----

=== Compliant solution

[source,javascript]
----
const pattern = /Hello, {3}world!/;
----
