include::../description.adoc[]

== Noncompliant Code Example

[source,javascript]
----
const pattern = /Hello,   world!/;
----

== Compliant Solution

[source,javascript]
----
const pattern = /Hello, {3}world!/;
----
