One of the more error-prone things you can do with a regular expression is to use more than one space. It's very hard to tell how many spaces are intended to be matched.

It's better to use only one space and then specify how many spaces are expected.

== Noncompliant Code Example

----
var re = /foo   bar/;
var re = new RegExp("foo   bar");
----

== Compliant Solution

----
var re = /foo {3}bar/;
var re = new RegExp("foo {3}bar");
----
