<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when <code>list()</code> calls are applied to types that are already directly iterable with for-loops or comprehensions.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>When iterating over an already iterable object with a for loop or a comprehension, wrapping it with <code>list()</code> adds meaningless clutter that doesn&#8217;t provide any functional value. Additionally, it creates unnecessary overhead by generating an intermediate list in memory, which inefficiently consumes memory and can degrade performance, especially with large data structures. Iterating directly over the original object is cleaner and more efficient.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Remove the redundant <code>list()</code> call and iterate directly over the original iterable.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">some_iterable = range(10)
for i in list(some_iterable): # Noncompliant: unnecessary list() call
    print(i)</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">some_iterable = range(10)
for i in some_iterable: # Compliant
    print(i)</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Python Documentation - <a href="https://docs.python.org/3/glossary.html#term-iterable">Iterable Glossary Entry</a></p>
</li>
<li>
<p>Python Documentation - <a href="https://docs.python.domainunion.de/3/library/stdtypes.html#list">list()</a></p>
</li>
</ul>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="paragraph">
<p>This rule should raise when an iterable is wrapped in a <code>list()</code> call, and used in either a for loop or a comprehension.</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove this unnecessary <code>list()</code> call on an already iterable object.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>The <code>list()</code> call applied to the iterable.</p>
</div>
</div>
</div>
</div>