The standard order for `+using+` directives is alphabetic with the exception of `+System+` directives, which come first for higher visibility. Using a different order may cause maintainers to overlook a directive or misunderstand what's being used.


== Noncompliant Code Example

----
using C;  // Noncompliant
using System.A.A;
using A;
using D;
using B;
using System;
using System.A;
using System.B;
----


== Compliant Solution

----
using System;
using System.A;
using System.A.A;
using System.B;
using A;
using C;
using B;
using D;
----

