include::../description.adoc[]

== Noncompliant Code Example

----
Public Class Foo
End Class

Public Structure Bar
End Structure
----

== Compliant Solution

----
Namespace SomeSpace
    Public Class Foo
    End Class

    Public Structure Bar
    End Structure
End Namespace
----

ifdef::env-github,rspecator-view[]
'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
