== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,sql]
----
x := (y / 2 + 1); -- Compliant even if the parentheses are ignored
IF (x > 0) AND ((x+y > 0)) THEN -- Noncompliant
  -- ...
END IF;
----

=== Compliant solution

[source,sql]
----
x := (y / 2 + 1);
IF (x > 0) AND (x+y > 0) THEN
  -- ...
END IF;
----

include::../rspecator.adoc[]
