include::../description.adoc[]

== Noncompliant Code Example

[source,sql]
----
x := (y / 2 + 1); -- Compliant even if the parentheses are ignored
IF (x > 0) AND ((x+y > 0)) THEN -- Noncompliant
  -- ...
END IF;
----

== Compliant Solution

[source,sql]
----
x := (y / 2 + 1);
IF (x > 0) AND (x+y > 0) THEN
  -- ...
END IF;
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../highlighting.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
