== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,csharp]
----
try 
{
  /* some work which end up throwing an exception */
  throw new ArgumentException();
} 
finally 
{
  /* clean up */
  throw new InvalidOperationException();       // Noncompliant; will mask the ArgumentException
}
----

=== Compliant solution

[source,csharp]
----
try
{
  /* some work which end up throwing an exception */
  throw new ArgumentException();
} 
finally 
{
  /* clean up */                       // Compliant
}
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
