include::../description.adoc[]

== Noncompliant Code Example

[source,sql]
----
a := b + c; -- This is a trailing comment that can be very very long
----

== Compliant Solution

[source,sql]
----
-- This very long comment is better placed before the line of code
a := b + c;
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../parameters.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
