According to the Backbone.js docs

____
The *changed* property is the internal hash containing all the attributes that have changed since the last set. Please do not update *changed* directly since its state is internally maintained by set. A copy of *changed* can be acquired from changedAttributes.
____


The ``++changed++`` property is involved in decisions about whether or not a collection should be resorted when it is updated. If you modify it manually, you can break the resorting of the collection.


== Noncompliant Code Example

----
myModel.changed = { myProperty: 1 }; // Non-compliant
----



ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
