<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>It is a best practice in the public part of a class body, to describe only information relevant for reusers of this class, without implementation details like <code>inline</code> specifier.</p>
</div>
<div class="paragraph">
<p>For inline member function defined outside of the class body, this rule verifies that <code>inline</code> is set on the definition and not on the declaration of the function.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">class Foo {
  public:
    inline void method();  // Noncompliant
    // ...
};
void Foo::method() {
  // ...
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">class Foo {
  public:
    void method();
    // ...
};
inline void Foo::method() {
  // ...
}</code></pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p><a href="https://isocpp.org/wiki/faq/inline-functions#where-to-put-inline-keyword" class="bare">https://isocpp.org/wiki/faq/inline-functions#where-to-put-inline-keyword</a></p>
</li>
</ul>
</div>
<hr>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Move this "inline" specifier to the definition of the function.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>inline specifier</p>
</div>
</div>
</div>
</div>