== Why is this an issue?

Nesting too many control flow statements (such as ``++IF++``, ``++PERFORM++``, ``++EVALUATE++``...) should be avoided as it makes the code complex and therefore difficult to maintain.

=== Noncompliant code example

With a threshold (maximum allowed control flow statement nesting depth) of 3:

[source,cobol]
----
IF A = 1
    PERFORM 
        MOVE A TO B
        PERFORM
            IF B = 1 *> Noncompliant
              MOVE "HI" TO S1  
            END-IF
        END-PERFORM
    END-PERFORM
END-IF.
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

=== Parameters

.max
****
_Integer_

----
5
----

Maximum allowed control flow statement nesting depth.
****


include::../highlighting.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
