<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Jump statements, such as <code>CHECK</code> and <code>CONTINUE</code> let you change the default flow of program execution, but jump statements that direct the control flow to the original direction are just a waste of keystrokes.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-abap" data-lang="abap">  LOOP AT myTable.
    PERFORM form_open USING ...
    CHECK retcode = 0.
    ...
    perform form_close.
    CHECK retcode = 0. "Noncompliant; whatever the result of the check, the loop will continue to the next iteration
  ENDLOOP.</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-abap" data-lang="abap">  LOOP AT myTable.
    PERFORM form_open USING ...
    CHECK retcode = 0.
    ...
    PERFORM form_close.
  ENDLOOP.</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove this redundant jump.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>statement</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_is_related_to_s1751">is related to: <a data-rspec-id="S1751" class="rspec-auto-link">S1751</a></h3>

</div>
<div class="sect2">
<h3 id="_is_related_to_s3031">is related to: <a data-rspec-id="S3031" class="rspec-auto-link">S3031</a></h3>

</div>
<div class="sect2">
<h3 id="_on_2_jun_2016_134753_tamas_vajk_wrote">on 2 Jun 2016, 13:47:53 Tamas Vajk wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] I drafted the rule, let me know what you think.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_2_jun_2016_140507_ann_campbell_wrote">on 2 Jun 2016, 14:05:07 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>Note that there is a potential overlap of this rule with <a data-rspec-id="S1751" class="rspec-auto-link">RSPEC-1751</a>: unconditional <code>continue</code> at the end of a loop.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_2_jun_2016_140522_ann_campbell_wrote">on 2 Jun 2016, 14:05:22 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>looks fine [~tamas.vajk]</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_26_mar_2024_081500_philipp_dominik_schubert_wrote">on 26 Mar 2024, 08:15:00 Philipp Dominik Schubert wrote:</h3>
<div class="paragraph">
<p>We created a circle to discuss CFamily&#8217;s rules on dead code and decided to disable this rule in the CFamily analyzer since we believe it provides relatively low value and is not even implemented in CFamily&#8217;s analyzer (cf. <a href="https://sonarsource.atlassian.net/browse/CPP-5165" class="bare">https://sonarsource.atlassian.net/browse/CPP-5165</a>).</p>
</div>
</div>
</div>
</div>