
Gratuitous boolean expressions are conditions that do not change the evaluation
of a program.
This issue can indicate logical errors and affect the correctness of an
application, as well as its maintainability.

== Why is this an issue?

Control flow constructs like `if`-statements allow the programmer to direct the
flow of a program depending on a boolean expression.
However, suppose that condition is always true or always false,
independent of any input or other circumstances.
In that case, the program will effectively always execute only one of the
available execution branches.
Thus, the control flow construct and the condition no longer serve a purpose,
they become _gratuitous_.

=== What is the potential impact?

The presence of gratuitous conditions can indicate a logical error.
I.e., the programmer _did intend_ to have the program branch depending on a
condition, but they made a slight mistake during its formulation.
In this case, this issue might result in a bug and thus affect the reliability
of the application.
For instance, it might lead to the computation of incorrect results.

Additionally, gratuitous conditions and control flow constructs introduce
unnecessary complexity.
The source code becomes harder to understand, and thus, the application becomes
more difficult to maintain.

=== Noncompliant code example

[source,python]
----
def f(b):
    a = True
    if a:  # Noncompliant
        do_something()

    if a and b:  # Noncompliant; "a" is always "True"
        do_something()
----

=== Compliant solution

[source,python]
----
def f(b):
    a = True
    if foo(a):
        do_something()

    if b:
        do_something()
----

include::../see.adoc[]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../highlighting.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
