== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,csharp]
----
if (condition)  // Noncompliant
DoTheThing();

DoTheOtherThing();
SomethingElseEntirely();

Foo();
----

=== Compliant solution

[source,csharp]
----
if (condition)
  DoTheThing();

DoTheOtherThing();
SomethingElseEntirely();

Foo();
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

=== Highlighting

* Primary: ``++if (condition)++``
* Secondary: line of code following the condition


'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
