include::../description.adoc[]

== Noncompliant Code Example

----
if (condition)  // Noncompliant
DoTheThing();

DoTheOtherThing();
SomethingElseEntirely();

Foo();
----

== Compliant Solution

----
if (condition)
  DoTheThing();

DoTheOtherThing();
SomethingElseEntirely();

Foo();
----

ifdef::env-github,rspecator-view[]
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
