== Why is this an issue?

The storage layout used for floating-point values may vary from one compiler to another, and therefore no floating-point comparisons or manipulations should be made which rely directly on the way the values are stored. The in-built operators and functions, which hide the storage details from the developer, should be used.


=== Noncompliant code example

[source,cpp]
----
float32_t My_fabs ( float32_t f )
{
  uint8_t * pB = reinterpret_cast< uint8_t * >( &f );
  *( pB + 3 ) &= 0x7f; // Non-compliant â€“ generate the absolute value of an IEEE-754 float value.
  return f;
}
----


== Resources

* MISRA C:2004, 12.12 - The underlying bit representations of floating-point values shall not be used.
* MISRA {cpp}:2008, 3-9-3 - The underlying big representations of floating-point values shall not be used.
* https://wiki.sei.cmu.edu/confluence/x/kdUxBQ[CERT, FLP37-C.] - Do not use object representations to compare floating-point values


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
