With the introduction of the `@snippet` tag in Java 18, it is possible to include code snippets inside a javadoc. Code inside these snippets should use spaces only or tabs only (i.e., not a mix of both). Not doing so will lead to the inconsistent indentation in the javadoc output.


== Noncompliant Code Example ==
[source,java]
----
{@snippet :
class HelloWorld {                               // Noncompliant - spaces and tabs are mixed
     public static void main(String... args) {
<TAB><TAB>System.out.println("Hello World!");
     }
}
}
----


== Compliant Solution ==

=== Use spaces only ===
[source,java]
----
{@snippet :
class HelloWorld {
     public static void main(String... args) {
        System.out.println("Hello World!");
     }
}
}
----


=== Use tabulations only ===
[source,java]
----
{@snippet :
class HelloWorld {
<TAB>public static void main(String... args) {
<TAB><TAB>System.out.println("Hello World!");
<TAB>}
}
}
----

== See ==
Oracleâ€™s Programmer's Guide to Snippets: https://docs.oracle.com/en/java/javase/18/code-snippet/index.html[https://docs.oracle.com/en/java/javase/18/code-snippet/index.html]


ifdef::env-github,rspecator-view[]

'''

== Implementation Specification ==
(visible only on this page)

include::message.adoc[]

include::highlighting.adoc[]
endif::env-github,rspecator-view[]
