This rule reports when there is an Architecture Directive to remove a specific usage of a file or folder from a given location.

== Why is this an issue?

This issue is being raised because there is a dependency in the code, and an Architecture Directive has been created to remove it.

Architecture Directives are defined in the “Architecture” section of the project by an authorized user.

=== What is the potential impact?

Flaws in the architecture constitute unnecessary complexity which affects developer productivity. Removing this flaw makes future development easier by favoring local changes in the code.

== How to fix it

Remove or refactor the problematic dependency. This may involve replacing the dependency with an allowed one, moving code to appropriate files or folders, or introducing new abstractions. The exact solution depends on your project’s specific situation.

=== Code examples

==== Noncompliant code example

The following code contains a circular dependency: `order.js` &rarr; `customer.js` &rarr; `order.js`, and `order.js` &rarr; `product.js` &rarr; `order.js`.
Both cycles are connected through `order.js`, forming a tangle of two cycles.

[source,javascript,diff-id=1,diff-type=noncompliant]
----
// order.js
import { Customer } from './customer.js';
import { Product } from './product.js';

export class Order {
    constructor() {
        this.customer = new Customer();
        this.products = [];
    }
}

// customer.js
import { Order } from './order.js';

export class Customer {
    constructor() {
        this.orders = [];
    }
}

// product.js
import { Order } from './order.js';

export class Product {
    constructor() {
        this.orders = [];
    }
}
----

==== Compliant solution

The issue can be resolved by refactoring the structure.
Two service functions can replace the dependencies `customer.js` &rarr; `order.js` and `product.js` &rarr; `order.js`.

[source,javascript,diff-id=1,diff-type=compliant]
----
// order.js
import { Customer } from './customer.js';
import { Product } from './product.js';

export class Order {
    constructor() {
        this.customer = new Customer();
        this.products = [];
    }
}

// customer.js
export class Customer { }

// product.js
export class Product { }

// orderService.js
export function getOrdersByCustomer(customer) {
    // Implementation to get orders by customer
}

export function getOrdersByProduct(product) {
    // Implementation to get orders by product
}
----
