This rule raises an issue when using `FormulaEval.FormulaBuilder` without proper configuration for null checking, SObject type specification, or return type declaration.

== Why is this an issue?

Salesforce's Dynamic Formula Evaluation (DFE) feature allows you to evaluate formula expressions at runtime. However, improper configuration of `FormulaEval.FormulaBuilder` can lead to several problems.

When you don't check if the formula string is null or empty before building, your code can throw runtime exceptions. This is especially common when retrieving formulas from Custom Metadata Types, where the formula field might be blank.

Without specifying the SObject type using `withType()`, the formula lacks proper context for field references. This means formulas that reference object fields like `Type` or `AnnualRevenue` may fail to evaluate correctly or throw errors.

Not declaring the return type with `withReturnType()` removes type safety guarantees. The formula evaluation becomes unpredictable, and you might get unexpected data types that cause issues when casting or using the results.

These configuration issues become more critical in production environments where formula strings come from external sources like Custom Metadata Types, and where type mismatches can cause silent failures or unexpected behavior.

=== What is the potential impact?

Missing proper configuration can cause runtime exceptions, unpredictable formula evaluation results, and type casting errors. In production environments, this can lead to failed business logic execution, incorrect calculations, and poor user experience.

== How to fix it

Always implement all three best practices when using FormulaEval.FormulaBuilder: check for null/empty formula strings, specify the SObject type with withType(), and declare the return type with withReturnType().

=== Code examples

==== Noncompliant code example

[source,apex,diff-id=1,diff-type=noncompliant]
----
DiscountCalculationRule__mdt discountRule = [SELECT DiscountFormula__c FROM DiscountCalculationRule__mdt WHERE DeveloperName = 'StandardDiscount' LIMIT 1];
FormulaEval.FormulaBuilder builder = Formula.builder()
    .withFormula(discountRule.DiscountFormula__c); // Noncompliant
----

==== Compliant solution

[source,apex,diff-id=1,diff-type=compliant]
----
DiscountCalculationRule__mdt discountRule = [SELECT DiscountFormula__c FROM DiscountCalculationRule__mdt WHERE DeveloperName = 'StandardDiscount' LIMIT 1];
if (String.isNotBlank(discountRule.DiscountFormula__c)) {
    FormulaEval.FormulaBuilder builder = Formula.builder()
        .withType(Opportunity.SObjectType)
        .withReturnType(FormulaEval.FormulaReturnType.INTEGER)
        .withFormula(discountRule.DiscountFormula__c);
}
----

== Resources

=== Documentation

 * Dynamic Formula Evaluation in Salesforce - https://www.merfantz.com/blog/how-to-dynamically-evaluate-formulas-in-salesforce-apex[Comprehensive guide on implementing Dynamic Formula Evaluation with best practices]

 * Salesforce FormulaEval Class Documentation - https://developer.salesforce.com/docs/atlas.en-us.apexref.meta/apexref/apex_class_FormulaEval.htm[Official Salesforce documentation for the FormulaEval class and its methods]
