<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>The use of <code>this</code> is optional except when redirecting to a named constructor, and when it&#8217;s needed to distinguish between property names and arguments or other variables. For the sake of brevity, <code>this</code> should be omitted when it&#8217;s not strictly required.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-dart" data-lang="dart">class Car {
  int color;

  Car(this.color); // Mandatory to distinguish between 'color' property and 'color' parameter

  void fade() {
    this.color--;  // Noncompliant
  }
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-dart" data-lang="dart">class Car {
  int color;

  Car(this.color); // Mandatory to distinguish between 'color' property and 'color' parameter

  void fade() {
    color--;
  }
}</code></pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p>Dart Docs - <a href="https://dart.dev/tools/linter-rules/unnecessary_this">Dart Linter rule - unnecessary_this</a></p>
</li>
<li>
<p><a href="https://dart.dev/effective-dart/usage#dont-use-this-when-not-needed-to-avoid-shadowing">Effective Dart</a></p>
</li>
</ul>
</div>
<hr>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Unnecessary 'this.' qualifier.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>'this' keyword</p>
</div>
</div>
</div>
</div>