== How to fix it

=== Code examples

==== Noncompliant code example

include::{language}/noncompliant-code-attribute.adoc[]

==== Compliant solution

include::{language}/compliant-code-attribute.adoc[]

==== Noncompliant code example

include::{language}/noncompliant-code-mapcontrollerroute.adoc[]

==== Compliant solution

include::{language}/compliant-code-mapcontrollerroute.adoc[]