== Why is this an issue?

In single Dockerfile instructions like `RUN`, `CMD`, and `ENTRYPOINT` multiple commands can be run, including changing directory `cd` command.
Using `WORKDIR` instead reduces the complexity of the above instructions and makes them easy to read, understand, troubleshoot, and maintain.

=== What is the potential impact?

The Dockerfile instructions like `RUN` allow users to execute longer scripts. See the following example:

[source,docker]
----
RUN cd /tmp && \
  git clone myrepository.com/MyOrganization/project && \
  cd project && \
  make && \
  cd /app/bin
----

In this example, the first `cd /tmp` command can be replaced by `WORKDIR cd /tmp` put above `RUN`.
The last `cd /app/bin` command can be replaced too.
Those actions will reduce the length of the `RUN` instruction, which makes it easier to read and understand.
Sometimes, it is hard to avoid the usage of `cd` command, especially in the middle of a long script, but removing them from the beginning and end is straightforward.
Additionally, most commands work well with absolute paths, so changing directories can be avoided very often.

The `WORKDIR` instruction can be used multiple times in Dockerfile.
It changes the current directory for next instructions (until the following change).
This approach simplifies understanding of what is a current directory.

The same principles apply to `CMD` and `ENTRYPOINT` instructions.
Following this recommendation provides a clear structure for Dockerfiles, making it easier to maintain.

== How to fix it

Ensure that all usages of `cd` command are replaced by `WORKDIR` instruction, where possible.
Also, using absolute paths can be considered for commands that accept them.

=== Code examples

==== Noncompliant code example

[source,docker,diff-id=1,diff-type=noncompliant]
----
RUN cd /app/bin && ./start.sh
----

==== Compliant solution

[source,docker,diff-id=1,diff-type=compliant]
----
WORKDIR app/bin
RUN ./start.sh
----

== Resources

=== Documentation

* https://docs.docker.com/develop/develop-images/dockerfile_best-practices/#workdir[WORKDIR - Best practices for writing Dockerfiles]
* https://docs.docker.com/engine/reference/builder/#workdir[WORKDIR - Dockerfile reference]

ifdef::env-github,rspecator-view[]
'''
== Implementation Specification
(visible only on this page)

=== Message

WORKDIR instruction should be used instead of cd command.

=== Highlighting

Highlight usage of cd command.

'''
endif::env-github,rspecator-view[]
