Titles are important because they are displayed in search engine results as well as the browser's toolbar.


This rule verifies that the ``++<head>++`` tag contains a ``++<title>++`` one, and the ``++<html>++`` tag a ``++<head>++`` one.


== Noncompliant Code Example

----
<html>         <!-- Non-Compliant -->

<body>
...
</body>

</html>
----


== Compliant Solution

----
<html>         <!-- Compliant -->

<head>
<title>Some relevant title</title>
</head>

<body>
...
</body>

</html>
----



ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
