== Why is this an issue?

Titles are important because they are displayed in search engine results as well as the browser's toolbar.


This rule verifies that the ``++<head>++`` tag contains a ``++<title>++`` one, and the ``++<html>++`` tag a ``++<head>++`` one.


=== Noncompliant code example

[source,html]
----
<html>         <!-- Non-Compliant -->

<body>
...
</body>

</html>
----


=== Compliant solution

[source,html]
----
<html>         <!-- Compliant -->

<head>
<title>Some relevant title</title>
</head>

<body>
...
</body>

</html>
----



ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Add a <title> tag to this page.


'''
== Comments And Links
(visible only on this page)

=== on 8 Jul 2013, 18:20:05 Freddy Mallet wrote:
Is implemented by \http://jira.codehaus.org/browse/SONARPLUGINS-2996

endif::env-github,rspecator-view[]
