== Why is this an issue?

``++DISTINCT++`` operator causes the ``++SELECT++`` statement to avoid the SAP buffering and to read directly from the database and not from the buffer on the application server.


=== Noncompliant code example

[source,abap]
----
SELECT DISTINCT carrid
       FROM spfli 
       INTO count 
       WHERE cityto = 'NEW YORK'. 
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Double check that usage of "DISTINCT" operator is required as it leads to bypass SAP buffering.


endif::env-github,rspecator-view[]
