== Why is this an issue?

TO BE TRANSLATED :


L’objectif est de laisser la détermination taille des blocs physiques de fichier au niveau des JCL plutôt que de l’imposer par programme. Cela permet d’optimiser cette taille sans avoir à recompiler/livrer le programme


Que doit-on contrôler?

La présence de BLOC CONTAINS 0 RECORDS (ou BLOC 0, BLOC CONTAINS 0, etc)


=== Noncompliant code example

[source,cobol]
----
FD  MONFICHIER
    RECORD CONTAINS 10 CHARACTERS.
01  ENR-LU         PIC X(10).
----


=== Compliant solution

[source,cobol]
----
FD  MONFICHIER
    BLOC 0.
01  ENR-LU         PIC X(10).
----



ifdef::env-github,rspecator-view[]
'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
