This rule raises an issue when variables are referenced using the unbraced form `$var` instead of the braced form `$\{var}`.

== Why is this an issue?

Using the unbraced form `$var` for variable references can lead to several problems in shell scripts.

First, it creates inconsistency in your code. When some variables use `$var` and others use `$\{var}`, the script becomes harder to read and maintain. Consistent formatting helps developers quickly understand the code structure.

Second, the unbraced form can cause ambiguity issues. Consider a scenario where you want to append text directly after a variable name, like `$varname_suffix`. The shell interpreter cannot determine where the variable name ends and the literal text begins. This often leads to unexpected behavior or empty output.

Third, using braces makes your intentions explicit. The `$\{var}` form clearly delimits the variable name, making the code more readable and less prone to parsing errors. This is especially important in complex expressions or when variables are embedded within strings.
Finally, adopting the braced form as a standard practice prepares your code for more advanced parameter expansion features that shell provides, such as `${var:-default}` or `${var#prefix}`. Starting with consistent bracing makes it easier to enhance your scripts later.

=== What is the potential impact?

Using unbraced variable references can lead to:

* Unexpected behavior when variable names are followed by characters that could be part of the variable name
* Reduced code readability and maintainability
* Inconsistent coding style across the codebase
* Potential bugs when refactoring or extending variable usage

== How to fix it

Replace unbraced variable references with braced form. Wrap the variable name in curly braces `${}`.

=== Code examples

==== Noncompliant code example

[source,shell,diff-id=1,diff-type=noncompliant]
----
echo $var
echo "Hello $name!"
file_path=$dir/file.txt
----

==== Compliant solution

[source,shell,diff-id=1,diff-type=compliant]
----
echo "${var}"
echo "Hello ${name}!"
file_path="${dir}/file.txt"
----

== Resources

=== Documentation

 * Bash Parameter Expansion - https://www.gnu.org/software/bash/manual/html_node/Shell-Parameter-Expansion.html[Official Bash manual section on parameter expansion and variable referencing]
 * Bash Best Practices Guide - https://bertvv.github.io/cheat-sheets/Bash.html[Comprehensive guide to Bash scripting best practices including variable handling]

=== Standards

 * Google Shell Style Guide - https://google.github.io/styleguide/shellguide.html#s7.3-variable-expansion[Google's style guide recommending braced variable expansion]

