== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,csharp]
----
if (str == null && str.Length == 0)
{
  Console.WriteLine("String is empty");
}

if (str != null || str.Length > 0)
{
  Console.WriteLine("String is not empty");
}
----

=== Compliant solution

[source,csharp]
----
if (str == null || str.Length == 0)
{
  Console.WriteLine("String is empty");
}

if (str != null && str.Length > 0)
{
  Console.WriteLine("String is not empty");
}
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
