//// This rule is superseded by RSPEC-2259
//// Please consider implementing this latter instead.
include::../description.adoc[]

== Noncompliant Code Example

----
if (str == null && str.Length == 0)
{
  Console.WriteLine("String is empty");
}

if (str != null || str.Length > 0)
{
  Console.WriteLine("String is not empty");
}
----

== Compliant Solution

----
if (str == null || str.Length == 0)
{
  Console.WriteLine("String is empty");
}

if (str != null && str.Length > 0)
{
  Console.WriteLine("String is not empty");
}
----
