== Why is this an issue?

Using a final `*)` pattern in a `case` statement is defensive programming. The default clause should either handle unexpected values appropriately or include a comment explaining why no action is taken.

=== Noncompliant code example

[source,bash,diff-id=1,diff-type=noncompliant]
----
case "$param" in  # missing default clause
  0)
    do_something
    ;;
  1)
    do_something_else
    ;;
esac
----

=== Compliant solution

[source,bash,diff-id=1,diff-type=compliant]
----
case "$param" in
  0)
    do_something
    ;;
  1)
    do_something_else
    ;;
  *)
    error
    ;;
esac
----
