<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>You should only set a property of a temporal type (like <code>DateTime</code> or <code>DateTimeOffset</code>) as the primary key of a table if the values are guaranteed to be unique.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Using temporal types as the primary key of a table is risky.
When these types are used as primary keys, it usually means that each new key is created with the use of <code>.Now</code> or <code>.UtcNow</code> properties from <code>DateTime</code> and <code>DateTimeOffset</code> classes.
In those cases, duplicate keys exceptions may occur in many ways:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>when entries are added consecutively by a machine with low-enough system clock resolution;</p>
</li>
<li>
<p>when two different threads are inserting entries in close enough sequence for both to have the same time;</p>
</li>
<li>
<p>when changes such as daylight saving time (DST) transitions occur, where values can be repeated the following hour (only for <code>DateTime</code> type);</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>The rule raises an issue if:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>Entity Framework, or Entity Framework Core dependencies are found;</p>
</li>
<li>
<p>a class contains a property either named <code>Id</code>, <code>&lt;type name&gt;Id</code> or decorated by the <code>[Key]</code> or <code>[PrimaryKey]</code> attribute.</p>
</li>
<li>
<p>the key property is of one of the following types:</p>
<div class="ulist">
<ul>
<li>
<p><a href="https://learn.microsoft.com/en-us/dotnet/api/system.datetime">System.DateTime</a></p>
</li>
<li>
<p><a href="https://learn.microsoft.com/en-us/dotnet/api/system.datetimeoffset">System.DateTimeOffset</a></p>
</li>
<li>
<p><a href="https://learn.microsoft.com/en-us/dotnet/api/system.timespan">System.TimeSpan</a></p>
</li>
<li>
<p><a href="https://learn.microsoft.com/en-us/dotnet/api/system.dateonly">System.DateOnly</a></p>
</li>
<li>
<p><a href="https://learn.microsoft.com/en-us/dotnet/api/system.timeonly">System.TimeOnly</a></p>
</li>
</ul>
</div>
</li>
</ul>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Either use a GUID or the auto generated ID as a primary key.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-vbnet" data-lang="vbnet">Public Class Account
    Public Property Id As DateTime

    Public Property Name As String
    Public Property Surname As String
End Class</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-vbnet" data-lang="vbnet">Public Class Account
    Public Property Id As Guid

    Public Property Name As String
    Public Property Surname As String
End Class</code></pre>
</div>
</div>
<div class="paragraph">
<p>or</p>
</div>
</div>
<div class="sect3">
<h4 id="_noncompliant_code_example_2">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-vbnet" data-lang="vbnet">Public Class Person
    &lt;Key&gt;
    Public Property PersonIdentifier As DateTime

    Public Property Name As String
    Public Property Surname As String
End Class</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution_2">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-vbnet" data-lang="vbnet">Public Class Person
    &lt;Key&gt;
    Public Property PersonIdentifier As Guid

    Public Property Name As String
    Public Property Surname As String
End Class</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p><a href="https://learn.microsoft.com/en-us/ef/core/modeling/keys?tabs=data-annotations">Entity Framework keys and data annotation</a></p>
</li>
<li>
<p><a href="https://learn.microsoft.com/en-us/dotnet/api/system.guid">GUID</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>