<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Proper use of the <code>angular</code> object is crucial when using the AngularJS framework. Misspell an <code>angular</code> function, and depending on the browser you may not get any errors at all, or the error you do get will be distinctly unhelpful.</p>
</div>
<div class="paragraph">
<p>This rule is intended to catch misspellings of the recognized <code>angular</code> functions. It checks that only the following functions are used:</p>
</div>
<div class="ulist">
<ul>
<li>
<p><code>directive</code></p>
</li>
<li>
<p><code>module</code></p>
</li>
<li>
<p><code>controller</code></p>
</li>
<li>
<p><code>provider</code></p>
</li>
<li>
<p><code>filter</code></p>
</li>
</ul>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-javascript" data-lang="javascript">angular.moudle('docsSimpleDirectiveLD', []);  // Noncompliant</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-javascript" data-lang="javascript">angular.module('docsSimpleDirectiveLD', []);  // Compliant</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>"xxx" is not a recognized function of the "angular" object.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_9_mar_2015_103636_linda_martin_wrote">on 9 Mar 2015, 10:36:36 Linda Martin wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] I don&#8217;t see the difference between the <em>Compliant</em> and the <em>Noncompliant</em> code snippet.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_9_mar_2015_114834_ann_campbell_wrote">on 9 Mar 2015, 11:48:34 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>That&#8217;s kinda the point [~linda.martin]. Look closely. ;-)</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_9_mar_2015_173827_ann_campbell_wrote">on 9 Mar 2015, 17:38:27 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>Oops. Thanks [~linda.martin]</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_19_may_2015_142723_linda_martin_wrote">on 19 May 2015, 14:27:23 Linda Martin wrote:</h3>
<div class="paragraph">
<p>Reviewed.</p>
</div>
</div>
</div>
</div>