This rule raises an issue when a class declares or overrides the `clone()` method but does not implement the `Cloneable` interface.

== Why is this an issue?

The `Cloneable` interface serves as a marker interface that indicates a class supports cloning operations. When a class implements a `clone()` method without implementing `Cloneable`, it violates the established Java cloning contract.

This violation can lead to several problems:

* *Runtime exceptions*: If the `clone()` method calls `super.clone()` and the class doesn't implement `Cloneable`, it will throw a `CloneNotSupportedException` at runtime.
* *Inconsistent behavior*: Other developers and frameworks expect classes with `clone()` methods to implement `Cloneable`. Without it, the class may not work correctly with cloning utilities or frameworks.
* *Contract violation*: The Java specification states that classes should implement `Cloneable` if they support cloning, making this a violation of the language contract.

The `Cloneable` interface acts as a signal to the JVM and other code that the class has been designed to support cloning operations safely.

=== What is the potential impact?

Without implementing `Cloneable`, calling the `clone()` method may result in a `CloneNotSupportedException` at runtime, causing application crashes. This can lead to unexpected failures in production environments, especially when the cloning functionality is used indirectly through frameworks or libraries.

== How to fix it

Add the `Cloneable` interface to the class declaration.

=== Code examples

==== Noncompliant code example

[source,groovy,diff-id=1,diff-type=noncompliant]
----
class ValueClass {
    ValueClass clone() { // Noncompliant
        return new ValueClass()
    }
}
----

==== Compliant solution

[source,groovy,diff-id=1,diff-type=compliant]
----
class ValueClass implements Cloneable {
    ValueClass clone() {
        return new ValueClass()
    }
}
----

== Resources

=== Documentation

 * Oracle Java Documentation - Object.clone() - https://docs.oracle.com/javase/8/docs/api/java/lang/Object.html#clone--[Official documentation for the clone() method and Cloneable interface]

 * Effective Java - Item 13: Override clone judiciously - https://www.oreilly.com/library/view/effective-java/9780134686097/[Best practices for implementing clone() methods in Java]

=== Related rules

 * S1182 - Classes that override "clone" should be "Cloneable" and call "super.clone()"
