include::../description.adoc[]

== Noncompliant Code Example

----
$s = "";
try {
  $s = readMyFile($fileName);
} catch (Exception $e)  {
  throw $e;  // Noncompliant
}
----

== Compliant Solution

----
$s = "";
try { 
  $s = readMyFile($fileName);
} catch (Exception $e) {
  error_log($e->getMessage());
  throw $e;  // Compliant
}
----

or

----
$s = "";
try { 
  $s = readMyFile($fileName);
} catch (Exception $e) {
  error_log($e->getMessage());
  throw new MyException("an exception occurred", 2, $e);  // Compliant
}
----

or

----
$s = readMyFile($fileName);
----
