== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,php]
----
$s = "";
try {
  $s = readMyFile($fileName);
} catch (Exception $e)  {
  throw $e;  // Noncompliant
}
----

=== Compliant solution

[source,php]
----
$s = "";
try { 
  $s = readMyFile($fileName);
} catch (Exception $e) {
  error_log($e->getMessage());
  throw $e;  // Compliant
}
----
or

[source,php]
----
$s = "";
try { 
  $s = readMyFile($fileName);
} catch (Exception $e) {
  error_log($e->getMessage());
  throw new MyException("an exception occurred", 2, $e);  // Compliant
}
----
or

[source,php]
----
$s = readMyFile($fileName);
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
