<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Having shared code in all branches of an if-statement leads to redundancy, making the code harder to maintain and increasing the risk of errors. It also reduces readability by cluttering the distinct logic of each branch. Extracting shared code into a single location improves maintainability.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>To fix this issue, move the shared code outside of the if-statement branches so that it is executed regardless of the condition.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-rust" data-lang="rust">let result = if condition {
    println!("Hello World");
    42
} else {
    println!("Hello World");
    24
};</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-rust" data-lang="rust">println!("Hello World");
let result = if condition {
    42
} else {
    24
};</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Clippy Lints - <a href="https://rust-lang.github.io/rust-clippy/master/index.html#branches_sharing_code" class="bare">https://rust-lang.github.io/rust-clippy/master/index.html#branches_sharing_code</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>