<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>There is such a thing as software that&#8217;s too helpful, and that&#8217;s the case with DB2&#8217;s implicit casting of host variables used in <code>WHERE</code> clauses. If for instance you compare a varchar column with the value of a numeric host variable, starting with version 10, DB2 will silently convert the numeric value to a string - at a potentially huge performance cost.</p>
</div>
<div class="paragraph">
<p>This rule raises an issue when the type of a variable used with a SQL <code>WHERE</code> clause does not match the underlying type of the column to which it is compared.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cobol" data-lang="cobol">TODO</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Use a "type" variable here to match the "xxx" definition of the column.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p><code>colname = :badHostVar</code></p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_27_jun_2016_100503_pierre_yves_nicolas_wrote">on 27 Jun 2016, 10:05:03 Pierre-Yves Nicolas wrote:</h3>
<div class="paragraph">
<p>I think that this rule may overlap <a data-rspec-id="S3617" class="rspec-auto-link">RSPEC-3617</a>.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_27_jun_2016_215949_ann_campbell_wrote">on 27 Jun 2016, 21:59:49 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>Related, but different [~pierre-yves.nicolas]. This RSpec is about how I send data <em>to</em> the DB. <a data-rspec-id="S3617" class="rspec-auto-link">RSPEC-3617</a> is about how I get data <em>from</em> the DB.</p>
</div>
<div class="paragraph">
<p>They also have different motivations: this is about performance, which could apparently be significantly degraded if you break the rule. <a data-rspec-id="S3617" class="rspec-auto-link">RSPEC-3617</a> is about getting back screwed up data.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_27_jun_2016_220157_ann_campbell_wrote">on 27 Jun 2016, 22:01:57 Ann Campbell wrote:</h3>
<div class="paragraph">
<p><a href="http://www.idug.org/p/fo/et/thread=42706" class="bare">http://www.idug.org/p/fo/et/thread=42706</a></p>
</div>
<div class="paragraph">
<p><a href="http://www.idug.org/p/bl/et/blogaid=274" class="bare">http://www.idug.org/p/bl/et/blogaid=274</a></p>
</div>
<div class="paragraph">
<p><a href="http://it.toolbox.com/blogs/oracle-guide/minitip-10-implicit-vs-explicit-conversion-16441" class="bare">http://it.toolbox.com/blogs/oracle-guide/minitip-10-implicit-vs-explicit-conversion-16441</a></p>
</div>
</div>
</div>
</div>