=== on 25 Mar 2015, 16:03:23 Tamas Vajk wrote:
\[~ann.campbell.2] should I add the See section for C# as well?

=== on 25 Mar 2015, 16:24:48 Ann Campbell wrote:
\[~tamas.vajk] these language-specific subtasks override portions of their parent tasks, and the description field is treated as 5 separate parts: description, Noncompliant Code Example, Compliant Solution, Exceptions, and references (See). There's no need to copy/paste the references to this child ticket, but do include them in the implementation's rule description.

include::../comments-and-links.adoc[]
