include::../description.adoc[]

== Noncompliant Code Example

[source,scala]
----
println("Hello"); println("world!")
----

== Compliant Solution

[source,scala]
----
println("Hello")
println("world!")
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../highlighting.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
