include::../description.adoc[]

== Noncompliant Code Example

----
println("Hello"); println("world!")
----

== Compliant Solution

----
println("Hello")
println("world!")
----

ifdef::env-github,rspecator-view[]
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
