include::../description.adoc[]

== Noncompliant Code Example

----
  $i = 0;
  $i = $i++; // Noncompliant; i is still zero
----

== Compliant Solution

----
  $i = 0;
  $i++; 
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
