include::../description.adoc[]

== Noncompliant Code Example

----
  $i = 0;
  $i = $i++; // Noncompliant; i is still zero
----

== Compliant Solution

----
  $i = 0;
  $i++; 
----

ifdef::rspecator-view[]
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::rspecator-view[]
