== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,php]
----
  $i = 0;
  $i = $i++; // Noncompliant; i is still zero
----

=== Compliant solution

[source,php]
----
  $i = 0;
  $i++; 
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
