<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>On the principle that clearer code is better code, you should explicitly <code>import</code> the things you want to use in a module. Using <code>import *</code> imports everything in the module and risks confusing maintainers. Similarly, <code>export * from "module";</code> imports and then re-exports everything in the module and risks confusing not just maintainers but also the module&#8217;s users.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-javascript" data-lang="javascript">import * as Imported from "aModule";  // Noncompliant</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-javascript" data-lang="javascript">import {aType, aFunction} from "aModule";</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Explicitly import the specific member needed.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_5_feb_2016_153127_elena_vilchik_wrote">on 5 Feb 2016, 15:31:27 Elena Vilchik wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] Actually there is no problems with scopes or name conflicts when using wildcard in JS: all you have imported are properties of <code>Imported</code>. So I would just remove first sentence and keep the second one reworked a bit: "Listing the necessary imports makes clear what&#8217;s wanted."</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_5_feb_2016_163927_ann_campbell_wrote">on 5 Feb 2016, 16:39:27 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>See what you think now [~elena.vilchik]</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_25_nov_2014_211440_freddy_mallet_wrote">on 25 Nov 2014, 21:14:40 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>Hi [~ann.campbell.2], I would not activate this rule by default and would support an exception with static imports.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_26_feb_2016_160532_elena_vilchik_wrote">on 26 Feb 2016, 16:05:32 Elena Vilchik wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] there is one more case we want to cover in this rule:</p>
</div>
<div class="listingblock">
<div class="content">
<pre>export * from "moduleName";</pre>
</div>
</div>
<div class="paragraph">
<p>which is shortcut for importing everything and then exporting it. Could add it tot description?</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_26_feb_2016_161314_ann_campbell_wrote">on 26 Feb 2016, 16:13:14 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>check it out [~elena.vilchik]</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_26_feb_2016_161547_elena_vilchik_wrote">on 26 Feb 2016, 16:15:47 Elena Vilchik wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] thanks!</p>
</div>
</div>
</div>
</div>