<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Importing every public name from a module using a wildcard (<code>use _::*</code>) is a bad idea because:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>It could lead to conflicts between names defined locally and the ones imported.</p>
</li>
<li>
<p>It reduces code readability as developers will have a hard time knowing where names come from.</p>
</li>
<li>
<p>It clutters the local namespace, which makes debugging more difficult.</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>Remember that imported names can change when you update your dependencies. A wildcard import that works today might be broken tomorrow.</p>
</div>
<div class="sect2">
<h3 id="_exceptions">Exceptions</h3>
<div class="paragraph">
<p>No issue will be raised for modules that their name contains <code>prelude </code>. Many crates, including the standard library, provide modules named <code>prelude </code> that are specifically designed for wildcard import.</p>
</div>
<div class="paragraph">
<p>No issue will be raised in test modules. This is defined as any module with <code>test</code> in the name.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>There are two ways to avoid a wildcard import:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>Replace it with <code>use mymodule;</code> and access module members as <code>mymodule::myfunction</code>. If the module name is too long, alias it to a shorter name. Example: <code>use std::collections as col;</code></p>
</li>
<li>
<p>List every imported name. If necessary, import statements can be split on multiple lines using parentheses (preferred solution) or backslashes.</p>
</li>
</ul>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-rust" data-lang="rust">use std::collections::*; // Noncompliant
fn main() {
    let mut map = HashMap::new();
    map.insert(1, 2);
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-rust" data-lang="rust">use std::collections::HashMap;
fn main() {
    let mut map = HashMap::new();
    map.insert(1, 2);
}</code></pre>
</div>
</div>
<div class="paragraph">
<p>Or</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-rust" data-lang="rust">use std::collections::HashMap as HM;
fn main() {
    let mut map = HM::new();
    map.insert(1, 2);
}</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Clippy Lints - <a href="https://rust-lang.github.io/rust-clippy/master/index.html#wildcard_imports" class="bare">https://rust-lang.github.io/rust-clippy/master/index.html#wildcard_imports</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>