Access variable which is not available in the current scope

== Why is this an issue?

The variables defined by `ARG` instruction have a scope from the definition to the end of the build stage where it was defined.

=== What is the potential impact?

The variables will be unresolved which may lead to unintended behavior.

== How to fix it

=== Code examples

==== Noncompliant code example

[source,docker,diff-id=1,diff-type=noncompliant]
----
ARG SETTINGS
FROM busybox
RUN ./run/setup $SETTINGS
----

In this case when Dockerfile will be built with the flag `--build-arg SETTINGS=--some-settings` the flag `--some-settings` will be not passed to the `RUN` instruction.

==== Compliant solution

[source,docker,diff-id=1,diff-type=compliant]
----
FROM busybox
ARG SETTINGS
RUN ./run/setup $SETTINGS
----

In this case when Dockerfile will be built with the flag `--build-arg SETTINGS=--some-settings` the flag `--some-settings` will be passed to the `RUN` instruction.

=== How does this work?

The `FROM` instruction starts a new build stage where variables defined by previous `ARG` instructions are out of this new scope.
To make it accessible for the build stage they need to be defined after `FROM` instruction.

== Resources

=== Documentation

https://docs.docker.com/engine/reference/builder/#scope[ARG scope - Dockerfile reference]

ifdef::env-github,rspecator-view[]
'''
== Implementation Specification
(visible only on this page)

=== Message

Include the `ARG` instruction in the build stage where is used.

=== Highlighting

Highlight usage of the variable where it is not accessible.

'''
endif::env-github,rspecator-view[]
