== Why is this an issue?

Indicators should not be used on "IF" statements because they foster poor logic flow and result in code that is difficult to read and maintain.


=== Noncompliant code example

[source,rpg]
----
C   99FLD1          IFEQ      FLD2
----


=== Compliant solution

[source,rpg]
----
C     FLD1          IFEQ      FLD2
C     *IN99         ANDEQ     *ON
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Move this indicator into an additional clause in the "IF" statement


endif::env-github,rspecator-view[]
