== Why is this an issue?

Code that contains many macros becomes hard to understand. This is even worse when the set of defined macros is not stable, and you have to know at each point what macros are defined. Therefore, ``++#undef++`` can decrease the readability of macros.


However, well-disciplined use of ``++#undef++`` can also improve readability, for instance when defining a macro with a limited scope: The macro is ``++#defined++``, used a couple of times to reduce code duplication, then immediately ``++#undefed++``.


This rule raises an issue when a ``++#undef++`` undefines a macro that was defined in another file. It will also raise an issue for an ``++#undef++`` directive that tries to undefine a non-existing macro.


=== Noncompliant code example

[source,cpp]
----
#ifndef MY_HDR
#define MY_HDR
#endif
...
#undef MY_HDR    /* Noncompliant */
----


=== Compliant solution

[source,cpp]
----
#define LEVEL(i) int const i = #i
LEVEL(Debug);
LEVEL(Warning);
LEVEL(Error);
#undef LEVEL
----


== Resources

* MISRA C:2004, 19.6 - #undef shall not be used.
* MISRA {cpp}:2008, 16-0-3 - #undef shall not be used.
* MISRA C:2012, 20.5 - #undef should not be used


ifdef::env-github,rspecator-view[]
'''
== Comments And Links
(visible only on this page)

=== on 31 Mar 2015, 19:02:31 Evgeny Mandrikov wrote:
\[~ann.campbell.2] implementation seems more complete (SQALE, description) than this spec.

=== on 13 Apr 2015, 19:27:58 Evgeny Mandrikov wrote:
\[~ann.campbell.2] I'm wondering why blocker, but not active by default? Note that in implementation currently major and active.

endif::env-github,rspecator-view[]
