== Why is this an issue?

include::../description-dotnet.adoc[]

=== Noncompliant code example

[source,vbnet]
----
Public Class Car

    <Obsolete>  ' Noncompliant
    Public Sub CrankEngine(Turns As Integer)
        ' ...
    End Sub

End Class
----

=== Compliant solution

[source,vbnet]
----
Public Class Car

    <Obsolete("Replaced by the automatic starter")>
    Public Sub CrankEngine(Turns As Integer)
        ' ...
    End Sub

End Class
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Add an explanation.


'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
