include::../description-dotnet.adoc[]

== Noncompliant Code Example

[source,vbnet]
----
Public Class Car

    <Obsolete>  ' Noncompliant
    Public Sub CrankEngine(Turns As Integer)
        ' ...
    End Sub

End Class
----

== Compliant Solution

[source,vbnet]
----
Public Class Car

    <Obsolete("Replaced by the automatic starter")>
    Public Sub CrankEngine(Turns As Integer)
        ' ...
    End Sub

End Class
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
