Methods with a ``++void++`` return type that neither alter their objects' state nor the states of their arguments are effectively useless. They should either be eliminated, or - more likely - corrected to include the missing side effects.


== Noncompliant Code Example

[source,text]
----
public void whatsThePoint(List<Integer> ints) {  // Noncompliant
  int sum = 0;
  for (Integer i : ints) {
    sum += i.intValue();
  }
}
----


