== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,apex]
----
if ( a == a ) { // always true
  foo();
}
if ( a != a ) { // always false
  foo();
}
if ( a == b && a == b ) { // if the first one is true, the second one is too
  foo();
}
if ( a != b || a <> b ) { // if the first one is true, the second one is too
  foo();
}

Integer i = 5 / 5; // always 1
Integer j = 5 - 5; // always 0
----

=== Exceptions

* This rule ignores ``++*++``, ``+``, ``++^++``, ``++&++``, ``++|++``, ``++=++``, ``++<<++``, ``++>>++``  and ``++>>>++``.

include::../see.adoc[]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../highlighting.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
