include::../description.adoc[]

== Noncompliant Code Example

----
if ( a == a ) { // always true
  foo();
}
if ( a != a ) { // always false
  foo();
}
if ( a == b && a == b ) { // if the first one is true, the second one is too
  foo();
}
if ( a != b || a <> b ) { // if the first one is true, the second one is too
  foo();
}

Integer i = 5 / 5; // always 1
Integer j = 5 - 5; // always 0

----

== Exceptions

* This rule ignores `+*+`, `+++`, `+^+`, `+&+`, `+|+`, `+=+`, `+<<+`, `+>>+`  and `+>>>+`.

include::../see.adoc[]
