== Why is this an issue?

If all the keys in a `Map` are values from a single enum, it is recommended to use an `EnumMap` as the specific implementation.
An EnumMap, which has the advantage of knowing all possible keys in advance, is more efficient compared to other implementations, as it can use a simple array as underlying data structure.


=== Noncompliant code example

[source,java,diff-id=1,diff-type=noncompliant]
----
public class MyClass {

  public enum Color {
    RED, GREEN, BLUE, ORANGE;
  }

  Map<Color, String> colorMap = new HashMap<>(); // Noncompliant
}
----


=== Compliant solution

[source,java,diff-id=1,diff-type=compliant]
----
public class MyClass {

  public enum Color {
    RED, GREEN, BLUE, ORANGE;
  }

  Map<Color, String> colorMap = new EnumMap<>(Color.class); // Compliant
}
----

== Resources
* A Guide to EnumMap[https://www.baeldung.com/java-enum-map]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Convert this Map to an EnumMap.


endif::env-github,rspecator-view[]
