== How to fix it in ASP.NET

=== Code examples

include::../../common/fix/code-rationale.adoc[]

==== Noncompliant code example

[source,vbnet,diff-id=1,diff-type=noncompliant]
----
Imports System.Web
Imports System.Web.Mvc

Public Class ExampleController
    Inherits Controller

    <HttpGet>
    Public Sub Redirect(Url As String)
        Response.Redirect(Url)
    End Sub
End Class
----

==== Compliant solution

[source,vbnet,diff-id=1,diff-type=compliant]
----
Imports System.Web
Imports System.Web.Mvc

Public Class ExampleController
    Inherits Controller

    Private ReadOnly AllowedUrls() As String = {"/", "/login", "/logout"}

    <HttpGet>
    Public Sub Redirect(Url As String)
        If AllowedUrls.Contains(Url) Then
            Response.Redirect(Url)
        End If
    End Sub
End Class
----

include::../../common/fix/how-does-this-work.adoc[]

=== Pitfalls

include::../../common/pitfalls/starts-with.adoc[]
