This rule allows banning certain PHP functions.


== Noncompliant Code Example

Given a parameter of ``++foo++`` for this rule:

----
$arr=array("apple", "pear","banana");
echo foo($arr);  // Noncompliant
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

include::parameters.adoc[]

'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
