== Why is this an issue?

This rule allows banning certain PHP functions.


=== Noncompliant code example

Given a parameter of ``++foo++`` for this rule:

[source,php]
----
$arr=array("apple", "pear","banana");
echo foo($arr);  // Noncompliant
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Remove this use of the "xxx" function.


=== Parameters

.functions
****

Comma-separated list of disallowed functions
****


'''
== Comments And Links
(visible only on this page)

=== on 19 Sep 2014, 15:16:31 Freddy Mallet wrote:
@Ann, instead of providing a template rule, I would specialize this rule : "Alias functions kept in PHP to preserve a backward compatibility should not be used". Or we could imagine to provide two rules :

* a template rule without any default value
* a specialized rule targeting alias functions

endif::env-github,rspecator-view[]
