This rule allows banning certain PHP functions.

== Noncompliant Code Example

Given a parameter of ``++foo++`` for this rule:

----
$arr=array("apple", "pear","banana");
echo foo($arr);  // Noncompliant
----
