An issue is raised when a method overrides a superclass method or implements an interface method but is not annotated with `@Override`.

== Why is this an issue?

The `@Override` annotation serves two important purposes in Groovy code.

First, it makes your intent explicit. When you add `@Override`, you're clearly telling other developers (and your future self) that this method is meant to override a parent method or implement an interface method. This improves code readability and helps with code reviews.

Second, and more importantly, it acts as a safety net during compilation. If you intend to override a method but make a mistake in the method signature (wrong parameter types, misspelled method name, etc.), the compiler will catch this error when you use `@Override`. Without the annotation, your method would simply be treated as a new method rather than an override, which could lead to unexpected behavior.

For example, if you meant to override `equals(Object obj)` but accidentally wrote `equals(String str)`, the `@Override` annotation would cause a compilation error, alerting you to the mistake. Without it, you'd have two different methods, and the original `equals` behavior would remain unchanged.

This is especially valuable in Groovy's dynamic environment where method signatures can be less obvious than in statically typed languages.

=== What is the potential impact?

Without `@Override` annotations, method signature mistakes can go unnoticed, leading to methods that don't actually override parent methods as intended. This can result in unexpected runtime behavior, failed polymorphism, and bugs that are difficult to track down.

== How to fix it

Add the `@Override` annotation to methods that override superclass methods or implement interface methods.

=== Code examples

==== Noncompliant code example

[source,groovy,diff-id=1,diff-type=noncompliant]
----
class ClassOverridingToString {
    String toString() { // Noncompliant
        "ClassOverridingToString"
    }
}
----

==== Compliant solution

[source,groovy,diff-id=1,diff-type=compliant]
----
class ClassOverridingToString {
    @Override
    String toString() {
        "ClassOverridingToString"
    }
}
----

== Resources

=== Documentation

 * Groovy Documentation - Annotations - https://groovy-lang.org/syntax.html#_annotations[Official Groovy documentation on annotations including @Override]

 * Oracle Java Documentation - Override Annotation - https://docs.oracle.com/javase/8/docs/api/java/lang/Override.html[Official documentation for the @Override annotation]

=== Related rules

 * RSPEC-1161 - https://rules.sonarsource.com/java/RSPEC-1161/[Java version of the same rule for @Override annotations]
