This rule raises an issue when a method or closure parameter is reassigned to a new value within the method or closure body.

== Why is this an issue?

Reassigning method or closure parameters within the method body is a confusing practice that makes code harder to understand and maintain.

When you reassign a parameter, you lose access to the original value that was passed to the method. This can lead to several problems:

* **Reduced readability**: Other developers reading your code expect parameters to keep their original values throughout the method.
* **Debugging difficulties**: When debugging, you cannot inspect the original parameter value after it has been reassigned.
* **Potential bugs**: You might accidentally use the reassigned value when you meant to use the original parameter value.
* **Violation of expectations**: Parameters are meant to represent input values, not working variables.

This practice is particularly problematic in Groovy closures, where parameter reassignment can be even more confusing due to the concise syntax.

=== What is the potential impact?

This issue reduces code readability and maintainability. It can lead to bugs when the original parameter value is needed later in the method, and makes debugging more difficult since the original input values are lost.

== How to fix it

Use a temporary variable instead of reassigning the parameter.

=== Code examples

==== Noncompliant code example

[source,groovy,diff-id=1,diff-type=noncompliant]
----
void processData(String data) {
    data = data.trim() // Noncompliant
}
----

==== Compliant solution

[source,groovy,diff-id=1,diff-type=compliant]
----
void processData(String data) {
    trimmedData = data.trim() // Compliant
}
----

== Resources
