<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Contrary to possible developer expectations, a template constructor will not suppress the compiler-generated copy constructor. This may lead to incorrect copy semantics for members requiring deep copies.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">class A
{
public:
  A ( );
  // A ( A const &amp; rhs ); Example 1 - implicitly generated
  template &lt;typename T&gt;
  A ( T const &amp; rhs ) // Example 2
  : i ( new int32_t )
  {
   *i = *rhs.i;
  }
private:
  int32_t * i; // Member requires deep copy
};

void f ( A const &amp; a1 )
{
  A a2 ( a1 ); // Noncompliant, unexpectedly uses Example 1, which will result in a shallow copy of 'i', instead of a deep copy
}</code></pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p>MISRA C&#43;&#43;:2008, 14-5-2</p>
</li>
</ul>
</div>
<hr>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Add an explicit copy constructor to this class.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_is_related_to_s1026">is related to: <a data-rspec-id="S1026" class="rspec-auto-link">S1026</a></h3>

</div>
<div class="sect2">
<h3 id="_on_21_oct_2014_191223_ann_campbell_wrote">on 21 Oct 2014, 19:12:23 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~samuel.mercier] please incorporate the text under the Noncompliant header either into the code as comments or the description. Also as a "contrary to developer expectations" I would not tie this to Maintainability/Understandability. I think I see your direction on that, but when the original developer doesn&#8217;t understand what (s)he&#8217;s coded, then I&#8217;d call it Reliability.</p>
</div>
</div>
</div>
</div>