== Why is this an issue?

Earlier versions of z/OS and certain installations do not support in-stream data in procedures.
This can lead to inconsistencies within the codebase and compatibility issues.

== How to fix it

Remove the uses of `DD *` and `DD DATA` from procedures.
You can add in-stream data to a procedure step by using `DD DUMMY` inside the procedure and overriding that by a `DD *` or `DD DATA` statement in the calling step.

=== Code examples

==== Noncompliant code example

[source,jcl,diff-id=1,diff-type=noncompliant]
----
//MYPROC   PROC
//STEP1    EXEC PGM=DOTHING
//SYSIN    DD *
  Input data
/*
//         PEND
//* Call procedure
//CALLPRC  EXEC PROC=MYPROC
----

==== Compliant solution

[source,jcl,diff-id=1,diff-type=compliant]
----
//MYPROC   PROC
//STEP1    EXEC PGM=DOTHING
//SYSIN    DD DUMMY
//         PEND
//* Call procedure
//CALLPRC  EXEC PROC=MYPROC
//STEP1.SYSIN DD *
  Input data
/*
----
