include::../kubernetes/description.adoc[]

== How to fix it

=== Code examples

==== Noncompliant code example

[source,yaml,diff-id=1,diff-type=noncompliant]
----
- name: Noncompliant example task
  kubernetes.core.k8s:
    state: present
    definition:
      apiVersion: rbac.authorization.k8s.io/v1
      kind: Role
      metadata:
        namespace: default
        name: example-role
      rules:
        - apiGroups: [""]
          resources: ["pods"]
          verbs: ["get"]
        - apiGroups: [""]
          resources: ["pods/exec"] # Noncompliant
          verbs: ["create"]
----

==== Compliant solution

[source,yaml,diff-id=1,diff-type=compliant]
----
- name: Compliant example task
  kubernetes.core.k8s:
    state: present
    definition:
      apiVersion: rbac.authorization.k8s.io/v1
      kind: Role
      metadata:
        namespace: default
        name: example-role
      rules:
        - apiGroups: [""]
          resources: ["pods"]
          verbs: ["get"]
----

=== How does this work?

The `exec` permissions are set by allowing the `create` verb for the `pods/exec` resource. Removing this permission will prevent users and services from executing arbitrary commands within containers.

//=== Pitfalls

//=== Going the extra mile


include::../kubernetes/resources.adoc[]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../kubernetes/message.adoc[]


include::../kubernetes/highlighting.adoc[]


endif::env-github,rspecator-view[]
