<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when an empty string literal is concatenated with another value using the <code>+</code> operator.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Concatenating an empty string with another value is an inefficient way to convert that value to a string. This pattern creates unnecessary string objects and obscures the developer&#8217;s intent.</p>
</div>
<div class="paragraph">
<p>In Groovy, when you use <code>'' + someValue</code>, the runtime must:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>Create a new empty string object</p>
</li>
<li>
<p>Convert <code>someValue</code> to a string representation</p>
</li>
<li>
<p>Concatenate the empty string with the converted value</p>
</li>
<li>
<p>Return the result</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>This approach is less efficient than direct string conversion methods and makes the code less readable. The intent (converting a value to a string) is not immediately clear to other developers reading the code.</p>
</div>
<div class="paragraph">
<p>Moreover, this pattern can lead to unexpected behavior with null values, as <code>'' + null</code> results in the string <code>"null"</code> rather than an empty string or a null pointer exception that might be more appropriate depending on the context.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Replace empty string concatenation with explicit <code>toString()</code> calls. For potentially null values, use safe navigation with null coalescing to handle null cases appropriately.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-groovy" data-lang="groovy">def variable = method('' + property)</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-groovy" data-lang="groovy">def variable = method(property?.toString() ?: "")</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">

</div>
</div>