include::../../../shared_content/secrets/description.adoc[]

== Why is this an issue?

include::../../../shared_content/secrets/rationale.adoc[]

=== What is the potential impact?

Passwords in Redis are used to authenticate users against the database
engine. They are associated with user accounts that are granted specific
permissions over the database and its hosted data.

If a Redis password leaks to an unintended audience, it can have serious
consequences for the security of your database, the data stored within it, and
the applications that rely on it.

:secret_type: credentials


include::../../../shared_content/secrets/impact/data_compromise.adoc[]


== How to fix it

include::../../../shared_content/secrets/fix/revoke.adoc[]

include::../../../shared_content/secrets/fix/vault.adoc[]

=== Code examples

:example_secret: admin:password
:example_name: redis-url
:example_env: REDUS_CREDENTIALS 

==== Noncompliant code example

[source,java,diff-id=1,diff-type=noncompliant,subs="attributes"]
----
props.set("{example_name}", "rediss://{example_secret}@example.com:8080/example")
----

==== Compliant solution

[source,java,diff-id=1,diff-type=compliant,subs="attributes"]
----
props.set("{example_name}", "rediss://"+System.getenv("{example_env}")+"@example.com:8080/example")
----

//=== How does this work?

//=== Pitfalls

//=== Going the extra mile

== Resources

include::../../../shared_content/secrets/resources/standards.adoc[]

//=== Benchmarks
