<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This is an issue when using <code>bytes.Compare(a, b) == 0</code> or <code>bytes.Compare(a, b) != 0</code> to check if two byte slices are equal or not equal.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>The <code>bytes.Compare</code> function is designed for ordering comparisons and returns -1, 0, or 1 to indicate the relative order of two byte slices. When you only need to check equality, <code>bytes.Compare</code> does unnecessary work.</p>
</div>
<div class="paragraph">
<p>The <code>bytes.Equal</code> function is specifically optimized for equality checks. It can return early as soon as it finds a difference, and it may use optimized assembly code for better performance on certain architectures.</p>
</div>
<div class="paragraph">
<p>Using <code>bytes.Compare</code> for equality checks makes your code slower and less clear about its intent. The performance difference can be significant, especially when comparing large byte slices or when the comparison is performed frequently.</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="paragraph">
<p>Using <code>bytes.Compare</code> instead of <code>bytes.Equal</code> for equality checks results in unnecessary performance overhead. The impact increases with the size of the byte slices being compared and the frequency of comparisons in your application.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Replace <code>bytes.Compare(a, b) == 0</code> with <code>bytes.Equal(a, b)</code> for equality checks.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-go" data-lang="go">if bytes.Compare(data1, data2) == 0 { // Noncompliant
    fmt.Println("Equal")
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-go" data-lang="go">if bytes.Equal(data1, data2) {
    fmt.Println("Equal")
}</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>bytes.Equal documentation - <a href="https://pkg.go.dev/bytes#Equal">Official Go documentation for the bytes.Equal function</a></p>
</li>
<li>
<p>bytes.Compare documentation - <a href="https://pkg.go.dev/bytes#Compare">Official Go documentation for the bytes.Compare function</a></p>
</li>
<li>
<p>Go Blog: go fmt your code - <a href="https://go.dev/blog/gofmt#mechanical-source-transformation">Blog post discussing mechanical transformations including the bytes.Compare to bytes.Equal optimization</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>