include::../description.adoc[]

== Noncompliant Code Example

----
UPDATE person
SET name = name;
----

== Compliant Solution

----
UPDATE person
SET name = UPPER(name);
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
