== Why is this an issue?

Because Visual Basic 6 does not automatically unload ``++Form++`` s after a program terminates, it is possible for them to remain in memory. Even after ``++Unload++`` ing, they may stay in memory if references remain. Therefore ``++Form++`` s should be explicitly ``++Unload++`` ed, and their references set to ``++Nothing++``.


=== Noncompliant code example

[source,vb6]
----
dim f1 as new form
f1.show
...
f1.hide
----


=== Compliant solution

[source,vb6]
----
dim f1 as new form
f1.show
...
f1.hide
Unload f1
Set f1 = Nothing
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

endif::env-github,rspecator-view[]
