== Why is this an issue?

The following predefined variables are deprecated and should be replaced by the new versions:

[frame=all]
[cols="^1,^1"]
|===
|Replace|With

|$HTTP_SERVER_VARS|$_SERVER
|$HTTP_GET_VARS|$_GET
|$HTTP_POST_VARS|$_POST
|$HTTP_POST_FILES|$_FILES
|$HTTP_SESSION_VARS|$_SESSION
|$HTTP_ENV_VARS|$_ENV
|$HTTP_COOKIE_VARS|$_COOKIE
|$php_errormsg|error_get_last()
|===

=== Noncompliant code example

[source,php]
----
echo 'Name parameter value: ' . $HTTP_GET_VARS["name"];
----


=== Compliant solution

[source,php]
----
echo 'Name parameter value: ' . $_GET["name"];
----



ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
