include::../description.adoc[]

== Noncompliant Code Example

----
if (x < 0)
  new IllegalArgumentException("x must be nonnegative");
----

== Compliant Solution

----
if (x < 0)
  throw new IllegalArgumentException("x must be nonnegative");
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
