<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>JUnit is a testing framework for the Java programming language.
It provides a platform to write and execute unit tests for Java code, ensuring the correctness and reliability of the software.</p>
</div>
<div class="paragraph">
<p>JUnit 4 is the earlier version of the JUnit framework, and it has its distinct ways of writing and executing tests.
However, with the introduction of JUnit 5 (JUnit Jupiter), significant changes and improvements were made to the framework.
The JUnit Vintage engine was created as a compatibility layer to support legacy test code written in JUnit 4.
In a project, you can use both JUnit4 and JUnit5 tests side by side.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Add <code>junit-vintage-engine</code> dependency to the project.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-xml" data-lang="xml">&lt;dependencies&gt;
    &lt;dependency&gt;
        &lt;groupId&gt;org.junit.jupiter&lt;/groupId&gt;
        &lt;artifactId&gt;junit-jupiter-engine&lt;/artifactId&gt;
        &lt;version&gt;${junit-jupiter.version}&lt;/version&gt;
        &lt;scope&gt;test&lt;/scope&gt;
    &lt;/dependency&gt;  // Noncompliant
&lt;/dependencies&gt;</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-xml" data-lang="xml">&lt;dependencies&gt;
    &lt;dependency&gt;
        &lt;groupId&gt;org.junit.jupiter&lt;/groupId&gt;
        &lt;artifactId&gt;junit-jupiter-engine&lt;/artifactId&gt;
        &lt;version&gt;${junit-jupiter.version}&lt;/version&gt;
        &lt;scope&gt;test&lt;/scope&gt;
    &lt;/dependency&gt;
    &lt;dependency&gt;
        &lt;groupId&gt;org.junit.vintage&lt;/groupId&gt;
        &lt;artifactId&gt;junit-vintage-engine&lt;/artifactId&gt;
        &lt;version&gt;${junit-jupiter.version}&lt;/version&gt;
    &lt;/dependency&gt;  // Compliant
&lt;/dependencies&gt;</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p><a href="https://junit.org/junit5/">JUnit 5</a></p>
</li>
<li>
<p><a href="https://junit.org/junit4/">JUnit 4</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>