<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>The use of digraphs may not meet developer expectations.</p>
</div>
<div class="paragraph">
<p>The digraphs are:</p>
</div>
<div class="ulist">
<ul>
<li>
<p><code>&lt;%</code></p>
</li>
<li>
<p><code>%&gt;</code></p>
</li>
<li>
<p><code>&lt;:</code></p>
</li>
<li>
<p><code>:&gt;</code></p>
</li>
<li>
<p><code>%:</code></p>
</li>
<li>
<p><code>%:%:</code></p>
</li>
</ul>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">template &lt;typename T&gt;
class A
{
  public:
    template&lt;int32_t i&gt;
    void f2();
};

void f(A&lt;int32_t&gt; * a&lt;:10:&gt;)    /* Noncompliant - usage of '&lt;:' instead of '[' and ':&gt;' instead of ']' */
&lt;%                              /* Noncompliant - usage of '&lt;%' instead of '{' */
  a&lt;:0:&gt;-&gt;f2&lt;20&gt;();             /* Noncompliant - usage of '&lt;:' and ':&gt;' */
%&gt;                              /* Noncompliant - usage of '%&gt;' instead of '}' */</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">/* ... */

void f(A&lt;int32_t&gt; * a[10])      /* Compliant */
{                               /* Compliant */
  a[0]-&gt;f2&lt;20&gt;();               /* Compliant */
}                               /* Compliant */</code></pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p>MISRA C&#43;&#43;:2008, 2-5-1 - Digraphs should not be used.</p>
</li>
</ul>
</div>
<hr>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Replace this digraph 'xxx' by its equivalent 'yyy'.</p>
</div>
</div>
</div>
</div>