== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,vbnet]
----
Public Class HomeController
    Inherits Controller

    Public Function Article() As ActionResult
        ViewBag.Title = "Title" ' Noncompliant, model should be used
        ViewData("Text") = "Text" ' Noncompliant, model should be used
        Return View()
    End Function

End Class
----

=== Compliant solution

[source,vbnet]
----
Public Class ArticleModel

    Public Property Title As String
    Public Property Text As String

End Class

Public Class HomeController
    Inherits Controller

    Public Function Article() As ActionResult
        Dim Model As New ArticleModel With {.Title = "Title", .Text = "Text"}
        Return View(Model)
    End Function

End Class
----

include::../see.adoc[]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../highlighting.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
