<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Importing a feature from the <code>__future__</code> module turns on that feature from a future version of Python in your module. The purpose is to allow you to gradually transition to the new features or incompatible changes in future language versions, rather than having to make the entire jump at once.</p>
</div>
<div class="paragraph">
<p>Because such changes must be applied to the entirety of a module to work, putting such imports anywhere but in the beginning of the module doesn&#8217;t make sense. It would mean applying those restrictions to only part of your code. Because that would lead to inconsistencies and massive confusion, it&#8217;s not allowed.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">name = "John"

from __future__ import division # Noncompliant</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">from __future__ import division

name = "John"</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Move this "<em>future</em>" import to the top of the module.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_29_apr_2015_070900_ann_campbell_wrote">on 29 Apr 2015, 07:09:00 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>I&#8217;ve edited your edits [~elena.vilchik]. Double-check me, please</p>
</div>
</div>
</div>
</div>