== Why is this an issue?

When a function parameter has a nullable type, e.g., parameter ``++param++`` in ``++f(?int param)++``, it must be explicitly provided in every function call. A nullable-type parameter has no default value.


=== Noncompliant code example

[source,php]
----
function f(?int param) {}
f();
----


=== Compliant solution

[source,php]
----
function f(?int param) {}
f(0);
----



ifdef::env-github,rspecator-view[]
'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
