This is an issue when a method annotated with ``++@future++`` contains HTTP callout operations but does not specify ``++callout=true++`` in the annotation.

== Why is this an issue?

In Salesforce Apex, future methods run asynchronously and have different execution contexts than regular methods. By default, future methods cannot make HTTP callouts to external web services for security and resource management reasons.

When you attempt to make an HTTP callout from a future method without the ``++callout=true++`` parameter, Salesforce will throw a runtime exception. This happens because the platform needs explicit permission to allow network operations from asynchronous contexts.

The ``++callout=true++`` parameter tells the Salesforce platform that this future method is authorized to make external HTTP requests. Without this parameter, any attempt to use HTTP classes like `Http`, `HttpRequest`, or `HttpResponse` will fail at runtime, potentially breaking your application's functionality.

This restriction exists because HTTP callouts consume system resources and can affect performance. By requiring explicit declaration, Salesforce ensures developers are intentional about which asynchronous methods can access external services.

=== What is the potential impact?

Without the ``++callout=true++`` parameter, the application will throw a runtime exception when the future method attempts to make HTTP callouts. This can cause:

* Integration failures with external systems
* Incomplete business processes that depend on external data
* Poor user experience due to unexpected errors
* Difficult debugging since the error occurs asynchronously

== How to fix it

Add the ``++callout=true++`` parameter to the ``++@future++`` annotation when the method contains HTTP callout operations.

=== Code examples

==== Noncompliant code example

[source,apex,diff-id=1,diff-type=noncompliant]
----
@future
public static void syncWithExternalSystem(String recordId) {
    Http http = new Http(); // Noncompliant
    HttpRequest request = new HttpRequest();
    request.setEndpoint('https://api.example.com/sync');
    request.setMethod('POST');
    HttpResponse response = http.send(request);
}
----

==== Compliant solution

[source,apex,diff-id=1,diff-type=compliant]
----
@future(callout=true)
public static void syncWithExternalSystem(String recordId) {
    Http http = new Http();
    HttpRequest request = new HttpRequest();
    request.setEndpoint('https://api.example.com/sync');
    request.setMethod('POST');
    HttpResponse response = http.send(request);
}
----

== Resources

=== Documentation

 * Salesforce Future Methods Documentation - https://developer.salesforce.com/docs/atlas.en-us.apexcode.meta/apexcode/apex_invoking_future_methods.htm[Official Salesforce documentation on future methods and their usage]

 * Apex HTTP Callouts - https://developer.salesforce.com/docs/atlas.en-us.apexcode.meta/apexcode/apex_callouts_http.htm[Documentation on making HTTP callouts in Apex]
