== Why is this an issue?

include::description.adoc[]

=== Noncompliant code example

[source,text]
----
public class Fruit {
  String plantingSeason;
  //...
}

public class Raspberry {
  String plantingseason;  // Noncompliant
  // ...
}
----


=== Compliant solution

[source,text]
----
public class Fruit {
  String plantingSeason;
  //...
}

public class Raspberry {
  String whenToPlant;
  // ...
}
----
Or

[source,text]
----
public class Fruit {
  String plantingSeason;
  //...
}

public class Raspberry {  // field removed; parent field will be used instead
  // ...
}
----

