This rule reports when a file depends on another file, and has been requested for removal.

== Why is this an issue?

This issue is being raised because a dependency in the code is not desirable since it constitutes an architecture problem.

=== What is the potential impact?

As these problems accumulate, the code becomes more complex and less consistent with the intent, making future development increasingly challenging.

== How to fix it

Remove or refactor the problem dependency.
This may involve replacing the dependency with an allowed one, moving code to appropriate files or directories, or introducing new abstractions. The exact solution depends on your project's specific situation.

=== Code examples

==== Noncompliant code example

If the problem dependency is that files in directory `panels` should not depend on files in `repos`, then the following code example is noncompliant:

[source,javascript,diff-id=1,diff-type=noncompliant]
----
# repos/order_repo.py
class OrderRepo: ...

# panels/foo.py
from repos.order_repo import OrderRepo # Noncompliant

def foo():
    repo = OrderRepo()
----

==== Compliant solution

A solution could be to call into the service layer instead.

[source,javascript,diff-id=1,diff-type=compliant]
----
# repos/order_repo.py
class OrderRepo: ...

# services/order_service.py
from repos.order_repo import OrderRepo

class OrderService:
    def __init__(self):
        self.repo = OrderRepo()

# panels/foo.py
from services.order_service import OrderService

def foo():
    repo = OrderService()
----
