This rule raises an issue in Java 25 when import declarations of different types (module imports, package imports, single-type imports) are mixed without proper grouping by specificity.

== Why is this an issue?

When Java 25 source files contain a mix of different import declaration types, organizing them by specificity significantly improves code readability and maintainability.

Java 25 introduces module imports as a preview feature (JEP 511), expanding the import system to support three types of import declarations with different levels of specificity:

* Module imports (`import module M;`) - least specific, import all exported packages from a module
* Package imports (`import P.*;`) - medium specificity, import all types from a package  
* Single-type imports (`import P.C;`) - most specific, import one particular type

When these different import types are scattered throughout the import section, it becomes harder for developers to:

* Quickly understand what dependencies the code has
* Identify potential naming conflicts between imports
* Maintain and refactor the import section
* Follow the logical hierarchy of Java's import shadowing behavior

The recommended grouping order follows Java's import shadowing rules: more specific imports can override less specific ones. By organizing imports from least specific (module) to most specific (single-type), the code structure mirrors the language's resolution behavior, making it more intuitive for developers to understand which imports take precedence.

=== What is the potential impact?

Poor import organization reduces code readability and makes it harder to maintain dependencies. While this doesn't affect runtime behavior, it can slow down development and increase the likelihood of import-related errors during refactoring.

== How to fix it

Group import declarations by specificity with module imports first, package imports in the middle, and single-type imports last. Add blank lines between different groups for better visual separation.

=== Code examples

==== Noncompliant code example

[source,java,diff-id=1,diff-type=noncompliant]
----
import P1.C1;
import module M1;
import P1.*;
import P2.C2;
import module M2;

class Foo { ... }
----

==== Compliant solution

[source,java,diff-id=1,diff-type=compliant]
----
// Module imports
import module M1;
import module M2;

// Package imports
import P1.*;

// Single-type imports
import P1.C1;
import P2.C2;

class Foo { ... }
----

== Resources

=== Documentation

 * JEP 511: Module Import Declarations - https://openjdk.org/jeps/511[Module import declarations specification for Java 25]
