<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when a <code>tensorflow.function</code> is recursive.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>When defining a <code>tensorflow.function</code> it is generally a bad practice to make this function recursive. TensorFlow does not
support recursive <code>tensorflow.function</code> and will in the majority of cases throw an exception. However it is possible as
well that the execution of such function succeeds, but with multiple tracings which has strong performance implications.
When executing <code>tensorflow.function</code>, the code is split into two distinct stages.
The first stage call <code>tracing</code> creates a new <code>tensorflow.Graph</code>, runs the Python code normally,
but defers the execution of TensorFlow operations (i.e. adding two Tensors). These operations are added to the graph without being ran.
The second stage which is much faster than the first, runs everything that was deferred previously.
Depending on the input of the <code>tensorflow.function</code> the first stage may not be needed, see: <a href="https://www.tensorflow.org/guide/function#rules_of_tracing">Rules of tracing</a>.
Skipping this first stage is what provides the user with TensorFlow&#8217;s high performance.</p>
</div>
<div class="paragraph">
<p>Having a recursive <code>tensorflow.function</code> prevents the user from benefiting of TensorFlow&#8217;s capabilities.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>To fix this issue, refactor the <code>tensorflow.function</code> so that is it not recursive.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">import tensorflow as tf

@tf.function
def factorial(n):
     if n == 1:
        return 1
    else:
        return (n * factorial(n-1)) # Noncompliant: the function is recursive</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">import tensorflow as tf

@tf.function
def factorial(n):
     return tf.exp(tf.lgamma(n + 1)) # Compliant</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>TensorFlow Documentation - <a href="https://www.tensorflow.org/guide/function#recursive_tffunctions_are_not_supported">Recursive tf.functions are not supported</a></p>
</li>
<li>
<p>TensorFlow Documentation - <a href="https://www.tensorflow.org/guide/function#rules_of_tracing">Rules of tracing</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>