== Why is this an issue?

include::../whyAnIssue.adoc[]

=== What is the potential impact?

include::../impact.adoc[]

== How to fix it

include::../howToFixIt.adoc[]

=== Code examples

==== Noncompliant code example

[source,vbnet]
----
Private Sub CreateNewTime()
    Dim birthDate = New DateTime(1994, 7, 5, 16, 23, 42)
End Sub
----

==== Compliant solution

[source,vbnet]
----
Private Sub CreateNewTime()
    Dim birthDate = New DateTime(1994, 7, 5, 16, 23, 42, DateTimeKind.Utc)
End Sub
----

== Resources
=== Documentation

include::../documentation.adoc[]