<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Relative lib imports should be avoided.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>The <a href="https://dart.dev/guides/libraries/create-packages"><code>lib</code> directory</a> of a Dart project is where the main code of the package is stored. Code living directly under the <code>lib</code> directory, when not explicitly declared as private with the <code>__</code> name convention, is considered the public API of the package, and can be imported via <a href="https://dart.dev/guides/packages#importing-libraries-from-packages">package imports</a>.</p>
</div>
<div class="paragraph">
<p>When importing code from a <code>lib</code> directory of a package available in the local file structure, it is recommended to use package imports instead of relative imports.</p>
</div>
<div class="paragraph">
<p>This is because relative imports may cause duplicate imports and may lead to confusion.</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-dart" data-lang="dart">import 'package:my_package/my_module.dart'; // OK
import '../my_module.dart'; // Noncompliant: duplicate of above</code></pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Convert the relative import to a package import.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-dart" data-lang="dart">import '../my_module.dart';</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-dart" data-lang="dart">import 'package:my_package/my_module.dart';</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Dart Docs - <a href="https://dart.dev/tools/linter-rules/avoid_relative_lib_imports">Dart Linter rule - avoid_relative_lib_imports</a></p>
</li>
<li>
<p>Dart Docs - <a href="https://dart.dev/guides/libraries/create-packages">Packages - Create packages</a></p>
</li>
<li>
<p>Dart Docs - <a href="https://dart.dev/guides/packages#importing-libraries-from-packages">Packages - Importing libraries from packages</a></p>
</li>
</ul>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="ulist">
<ul>
<li>
<p>Can&#8217;t use a relative path to import a library in 'lib'.</p>
</li>
</ul>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>The string defining the path of the relative import.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
</div>
</div>