<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Instead of manually recreating dataclasses, <code>dataclasses.replace(...)</code> can be used to recreate dataclasses with modified values.</p>
</div>
<div class="paragraph">
<p>Alternatively, the in Python 3.13 introduced, more general-purpose <code>copy.replace(...)</code> can be used to archive the same result.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">from dataclasses import dataclass

@dataclass
class Point:
    x: int
    y: int

p1 = Point(1, 2)
p2 = Point(p1.x, 3) # Noncompliant</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">from dataclasses import dataclass
import dataclasses
import copy

@dataclass
class Point:
    x: int
    y: int

p1 = Point(1, 2)
p2 = dataclasses.replace(p1, y=3) # Compliant
p3 = copy.replace(p1, y=3) # Compliant</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p><a href="https://docs.python.org/3/library/dataclasses.html#dataclasses.replace" class="bare">https://docs.python.org/3/library/dataclasses.html#dataclasses.replace</a></p>
</li>
<li>
<p><a href="https://docs.python.org/3/library/copy.html#copy.replace" class="bare">https://docs.python.org/3/library/copy.html#copy.replace</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>