include::../description.adoc[]

== Noncompliant Code Example

----
function doSomething() {
  for (let i = 0; i < 4; i++) {    // Noncompliant, 4 is a magic number
    // ...
  }
}
----

== Compliant Solution

----
function doSomething() {
  const numberOfCycles = 4;
  for (let i = 0; i < numberOfCycles; i++) {
    // ...
  }
}
----
