<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when a condition-only <code>for</code> loop (equivalent to a <code>while</code> loop) does not modify any of the variables used in its condition within the loop body, and has no other exit mechanism like <code>break</code> or <code>return</code>.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Condition-only loops in Go use the syntax <code>for condition { }</code> and are equivalent to <code>while</code> loops in other languages. These loops continue executing as long as the condition evaluates to <code>true</code>.</p>
</div>
<div class="paragraph">
<p>For the loop to eventually terminate, something within the loop body must change the state that affects the condition. This typically means:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>Modifying variables used in the condition</p>
</li>
<li>
<p>Using control flow statements like <code>break</code>, <code>return</code>, or <code>continue</code></p>
</li>
<li>
<p>Calling functions that modify global state or external resources</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>When none of these happen, the condition remains unchanged, creating an infinite loop. The loop will run forever, never reaching the code that follows it.</p>
</div>
<div class="paragraph">
<p>Infinite loops are particularly problematic because they consume CPU resources continuously and make the application unresponsive. In server applications, this can lead to resource exhaustion and system instability.</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="paragraph">
<p>The application will hang indefinitely, consuming CPU resources and becoming unresponsive. This can lead to:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>Poor user experience due to frozen interfaces</p>
</li>
<li>
<p>Server overload and potential system crashes</p>
</li>
<li>
<p>Resource exhaustion in production environments</p>
</li>
<li>
<p>Difficulty in debugging, as the application appears to be running but makes no progress</p>
</li>
</ul>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Modify the loop condition variable within the loop body to ensure the condition will eventually become false.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-go" data-lang="go">func main() {
    var value float64 = 100
    for value &gt; 0.5 {
        fmt.Println(value) // Noncompliant
        // value never changes - infinite loop
    }
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-go" data-lang="go">func main() {
    var value float64 = 100
    for value &gt; 0.5 {
        value = value / 2  // Modify the condition variable
        fmt.Println(value)
    }
}</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Go Tour - For Loop - <a href="https://go.dev/tour/flowcontrol/1">Official Go documentation on for loops and their different forms</a></p>
</li>
<li>
<p>Effective Go - Control Structures - <a href="https://go.dev/doc/effective_go#control-structures">Best practices for using control structures in Go</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>