Each file is considered an "external" module. The use of the ``++module++`` keyword creates an internal module, and was used before the ECMAScript 2015 addition of ``++namespace++``s for the same purpose. Now that ``++namespace++`` is available, the use of ``++module++`` is deprecated because it does the same thing, and its use could confuse maintainers unaware of the history of the language.


== Noncompliant Code Example

----
module myMod {  // Noncompliant
  // ...
}
----


== Compliant Solution

----
namespace myMod {
  // ...
}
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

include::highlighting.adoc[]

endif::env-github,rspecator-view[]
