When installing packages an index is cached locally by default.
The index can be used on-the-fly and not stored locally.

== Why is this an issue?

Docker images should only contain the necessary data.
The package index is not needed to good working of the installed software.
Storing an index also increases the size of the Docker image.
It should be reduced to speed up deployments and reduce storage and bandwidth.

== How to fix it

=== Code examples

==== Noncompliant code example

[source,docker,diff-id=1,diff-type=noncompliant]
----
RUN apk --no-cache add nginx
RUN apt-get update \
  && apt-get install -y nginx
RUN aptitude update \
  && aptitude install -y nginx
----

==== Compliant solution

[source,docker,diff-id=1,diff-type=compliant]
----
RUN apk --no-cache add nginx
RUN apt-get update \
  && apt-get install -y nginx \
  && rm -rf /var/cache/apt/archives /var/lib/apt/lists/*
RUN aptitude update \
  && aptitude install -y nginx \
  && rm -rf /var/cache/apt/archives /var/lib/apt/lists/*
----

=== How does this work?

When installing packages using `apk`, `apt-get`, or `aptitude` they store an index in Docker image layer in `/var/lib/apt/lists/`.
Also, some lock files are stored in `/var/cache/apt/archives`.
Those files are not needed to properly execute installed packages and can be removed to reduce image size.

== Resources
=== Documentation

https://docs.docker.com/develop/develop-images/dockerfile_best-practices/#run[RUN - Best practices for writing Dockerfiles]
https://man.archlinux.org/man/apk.8.en[apk man]
https://linux.die.net/man/8/apt-get[apt-get man]
https://linux.die.net/man/8/aptitude[aptitude man]
https://askubuntu.com/questions/1050800/how-do-i-remove-the-apt-package-index[How do I remove the apt package index?]

ifdef::env-github,rspecator-view[]
'''
== Implementation Specification
(visible only on this page)

=== Message

Remove cache after installing packages.

=== Highlighting

Highlight the entire install command.

'''
endif::env-github,rspecator-view[]
