When installing packages an index is cached locally by default.
The index can be used on-the-fly and not stored locally.

== Why is this an issue?

Docker images should only contain necessary data.
The package index is optional for the correct working of the installed software.
Storing an index also increases the size of the Docker image.
It should be reduced to speed up deployments and reduce storage and bandwidth.

== How to fix it

=== Code examples

==== Noncompliant code example

[source,docker,diff-id=1,diff-type=noncompliant]
----
RUN apk add nginx
RUN apt-get update \
  && apt-get install nginx
RUN aptitude update \
  && aptitude install nginx
----

==== Compliant solution

[source,docker,diff-id=1,diff-type=compliant]
----
RUN apk --no-cache add nginx
RUN apk add nginx \
  && rm -rf /var/lib/apt/lists/*
RUN apk add nginx \
  && rm -rf /var/cache/apt/archives /var/lib/apt/lists/*

RUN apt-get update \
  && apt-get install nginx \
  && apt-get clean
RUN apt-get update \
  && apt-get install nginx \
  && rm -rf /var/lib/apt/lists/*
RUN apt-get update \
  && apt-get install nginx \
  && rm -rf /var/cache/apt/archives /var/lib/apt/lists/*

RUN aptitude update \
  && aptitude install nginx \
  && aptitude clean
RUN aptitude update \
  && aptitude install nginx \
  && rm -rf /var/lib/apt/lists/*
RUN aptitude update \
  && aptitude install nginx \
  && rm -rf /var/cache/apt/archives /var/lib/apt/lists/*
----

=== How does this work?

When installing packages using `apk`, `apt-get`, or `aptitude` they store an index in the Docker image layer in `/var/lib/apt/lists/`.
This index is not needed after installation, so it can be removed.
To do that execute the `clean` command of your package manager tool or just run `rm -rf /var/lib/apt/lists/*`.
Additionally, some lock files are stored in `/var/cache/apt/archives`.
They are not removed by the `clean` command, so they need to be removed manually.

== Resources
=== Documentation

* https://docs.docker.com/develop/develop-images/dockerfile_best-practices/#run[RUN - Best practices for writing Dockerfiles]
* https://man.archlinux.org/man/apk.8.en[apk man]
* https://manpages.debian.org/bookworm/apt/apt-get.8.en.html[apt-get man]
* https://manpages.debian.org/testing/aptitude/aptitude.8.en.html[aptitude man]
* https://askubuntu.com/questions/1050800/how-do-i-remove-the-apt-package-index[How do I remove the apt package index?]

ifdef::env-github,rspecator-view[]
'''
== Implementation Specification
(visible only on this page)

=== Message

Remove the cache after installing packages.

=== Highlighting

Highlight the entire `install` command.

'''
endif::env-github,rspecator-view[]
