<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Assertions in Java are used to test assumptions that should hold true at a certain point in the program. An assertion with a condition that is always false will fail, which can lead to unexpected program termination. Moreover, an assertion that is always false often indicates a logic error or misunderstanding in the code, which could lead to further issues.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">  void caller() {
    // Noncompliant: triggers an AssertionError because the array length is 2
    printFifth(new int[2]);
  }

  void printFifth(int[] a) {
    assert a.length == 5;
    System.out.println(a[4]);
  }</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_exceptions">Exceptions</h3>
<div class="paragraph">
<p>No issue is raised for <code>assert false</code> statements, as they are considered intentional failures.</p>
</div>
</div>
</div>
</div>