== Why is this an issue?

HTML-style comments are not part of EcmaScript specification, and should not be used.


=== Noncompliant code example

[source,javascript]
----
<!-- Noncompliant -->
----


=== Compliant solution

[source,javascript]
----
// Compliant
/* Compliant */
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
