== Why is this an issue?

HTML-style comments are not part of EcmaScript specification, and should not be used.


=== Noncompliant code example

[source,javascript]
----
<!-- Noncompliant -->
----


=== Compliant solution

[source,javascript]
----
// Compliant
/* Compliant */
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Replace this HTML-style comment by a standard comment.


'''
== Comments And Links
(visible only on this page)

=== relates to: S1876

=== on 1 Nov 2019, 16:21:27 Elena Vilchik wrote:
See \https://github.com/SonarSource/SonarJS/issues/1698

endif::env-github,rspecator-view[]
