Try to imagine using the standard Flex API without ASDoc. It would be a nightmare, because ASDoc is the only way to understand of the contract of the API.
Documenting an API with ASDoc increases the productivity of the developers use it.

== Noncompliant Code Example

----
public class MyClass {
  public var myLabel:String; 

  public function myMethod(param1:String):Boolean {...}
}
----

== Compliant Solution

----
/**
 * my doc
 */
public class MyClass {
  /**
   * my doc
   */
  public var myLabel:String; 

  /**
   * my doc
   * @param param1 my doc
   * @return my doc
   */
  public function myMethod(param1:String):Boolean {...}
}
----

== Exceptions

Classes or class elements with an ASDoc `+@private+` comment are ignored by this rule.

----
/** 
 * @private  // This class and all its elements are ignored
 */
public class MyClass {  // Compliant

  public var myLabel:String;   // Compliant
}

public class AnotherClass {  // Noncompliant; class not @private and not documented

  /** 
   * @private
   */
  public var name:String;  // Compliant
}
----
