== Why is this an issue?

Try to imagine using the standard Flex API without ASDoc. It would be a nightmare, because ASDoc is the only way to understand of the contract of the API.

Documenting an API with ASDoc increases the productivity of the developers use it.

=== Noncompliant code example

[source,flex]
----
public class MyClass {
  public var myLabel:String; 

  public function myMethod(param1:String):Boolean {...}
}
----

=== Compliant solution

[source,flex]
----
/**
 * my doc
 */
public class MyClass {
  /**
   * my doc
   */
  public var myLabel:String; 

  /**
   * my doc
   * @param param1 my doc
   * @return my doc
   */
  public function myMethod(param1:String):Boolean {...}
}
----

=== Exceptions

Classes or class elements with an ASDoc ``++@private++`` comment are ignored by this rule.


----
/** 
 * @private  // This class and all its elements are ignored
 */
public class MyClass {  // Compliant

  public var myLabel:String;   // Compliant
}

public class AnotherClass {  // Noncompliant; class not @private and not documented

  /** 
   * @private
   */
  public var name:String;  // Compliant
}
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

include::parameters.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
