include::../description.adoc[]

== Noncompliant Code Example

----
def doSomething() = {
    var name = ""
    // ...
    name = name
}
----

== Compliant Solution

----
def doSomething() = {
    var name = ""
    // ...
    this.name = name
}
----


ifdef::env-github,rspecator-view[]
'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
