<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Variant functions are inefficient by nature, and should not be used with strings when string-specific versions, denoted by a <code>$</code> at the end of the name, are available.</p>
</div>
<div class="paragraph">
<p>This rule flags instances of these functions:</p>
</div>
<div class="paragraph">
<p><code>Left()</code>, <code>Mid()</code>, <code>Right()</code>, <code>Chr()</code>, <code>ChrW()</code>, <code>UCase()</code>, <code>LCase()</code>, <code>LTrim()</code>, <code>RTrim()</code>, <code>Trim()</code>, <code>Space()</code>, <code>String()</code>, <code>Format()</code>, <code>Hex()</code>, <code>Oct(),</code> <code>Str()</code>, <code>Error</code></p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-vb6" data-lang="vb6">TrimString = Trim(TestString)</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-vb6" data-lang="vb6">TrimString = Trim$(TestString)</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Replace this use of XXX with XXX$</p>
</div>
</div>
</div>
</div>