== How to fix it in OpenSSL

=== Code examples

Currently, pyOpenSSL does not have any (public) method to enable hostname verification.
Therefore, it is recommended to use another SSL library. The `ssl` module
of the Python standard library can be a great replacement for several use cases.

==== Noncompliant code example

[source,python,diff-id=11,diff-type=noncompliant]
----
import socket
from OpenSSL import SSL

ctx = SSL.Context(SSL.TLSv1_2_METHOD)  # Noncompliant
ctx.set_verify(SSL.VERIFY_PEER, verify_cb)

conn = SSL.Connection(ctx, socket.socket(socket.AF_INET, socket.SOCK_STREAM))
conn.connect(("www.example.com", 443))
conn.send("GET / HTTP/1.0\r\nHost: www.example.com\r\n\r\n".encode())
result = conn.recv(1024)

conn.shutdown()
conn.close()
----

==== Compliant solution

[source,python,diff-id=11,diff-type=compliant]
----
import ssl
import socket

# By default, hostname verification is enabled
ctx = ssl.create_default_context()

with socket.create_connection(("www.example.com", 443)) as sock:
    with ctx.wrap_socket(sock, server_hostname="www.example.com") as conn:
        conn.send(f"GET / HTTP/1.0\r\nHost: www.example.com\r\n\r\n".encode())
        conn.recv(1024)
----

=== How does this work?

include::../../common/fix/validation.adoc[]
