== How to fix it in Python Standard Library

=== Code examples

include::../../common/fix/code-rationale.adoc[]

Certificate validation is not enabled by default when ``++_create_unverified_context++`` or ``++_create_stdlib_context++`` is used. It is recommended to use `create_default_context`, without explicitly setting `check_hostname` to `False`. Doing so creates a secure context that validates both hostnames and certificates. +
If ``++create_default_context++`` cannot be used, then the ``check_hostname`` property must be set to ``True`` on the ``SSLContext``. Doing so also automatically enables certificate verification (see S4830 for more information).

==== Noncompliant code example

[source,python,diff-id=1,diff-type=noncompliant]
----
import ssl

ctx1 = ssl.create_default_context()
ctx1.check_hostname = False  # Noncompliant

ctx2 = ssl.SSLContext(ssl.PROTOCOL_TLS_CLIENT)  # Noncompliant
----

==== Compliant solution

[source,python,diff-id=1,diff-type=compliant]
----
import ssl

ctx1 = ssl.create_default_context()

ctx2 = ssl.SSLContext(ssl.PROTOCOL_TLS_CLIENT)
ctx2.check_hostname = True
----

=== How does this work?

include::../../common/fix/validation.adoc[]
