== How to fix it in Python Standard Library

=== Code examples

include::../../common/fix/code-rationale.adoc[]

Certificate validation is not enabled by default when ``++_create_unverified_context++`` or ``++_create_stdlib_context++`` is used. It is recommended to use ``++create_default_context++`` instead. This method creates secure contexts that will check server hostnames and verify certificates (see S4830 for more information) by default. +
If ``++create_default_context++`` cannot be used, then the ``check_hostname`` property must be set to ``True`` on the ``SSLContext``. Doing so also automatically enables certificate verification for this context.

==== Noncompliant code example

[source,python,diff-id=1,diff-type=noncompliant]
----
import ssl

ctx1 = ssl.create_default_context()
ctx1.check_hostname = False  # Noncompliant

ctx2 = ssl.SSLContext(ssl.PROTOCOL_TLSv1_2)  # Noncompliant
----

==== Compliant solution

[source,python,diff-id=1,diff-type=compliant]
----
import ssl

# By default, hostname verification is enabled
ctx1 = ssl.create_default_context()

ctx2 = ssl.SSLContext(ssl.PROTOCOL_TLSv1_2)
ctx2.check_hostname = True
----

=== How does this work?

include::../../common/fix/validation.adoc[]
