== Why is this an issue?

In Kotlin DSL for Gradle build scripts `Provider.get()` is used to retrieve a value of a specific type. 
The best practice is to retrieve the value only when it is required, that is at the latest possible point, otherwise it can lead to ordering issues or to `java.lang.IllegalStateException` in case of a missing value.

== How to fix it

In order to fix the issue you can:

* move the `Provider.get()` inside the task action that requires it
* or, in case the `Procider.get()` is required outside a task action, replace it with `Provider.map()` or `Provider.flatMap()`

=== Code examples

==== Noncompliant code example

[source,kotlin,diff-id=1,diff-type=noncompliant]
----
val x = Provider.get(value)

tasks.register("logX") {
    group = JavaBasePlugin.DOCUMENTATION_GROUP
    description = "Logging x value."
    
    doLast {
        logger.quiet("x = $x")
    }
}
----

==== Compliant solution

[source,kotlin,diff-id=1,diff-type=compliant]
----
tasks.register("logX") {
    group = JavaBasePlugin.DOCUMENTATION_GROUP
    description = "Logging x value."
    
    doLast {
        val x = Provider.get(value)
        logger.quiet("x = $x")
    }
}
----

== Resources

=== Documentation

* https://docs.gradle.org/current/javadoc/org/gradle/api/provider/Provider.html[org.gradle.api.provider.Provider]

=== Standards

* https://github.com/liutikas/gradle-best-practices#dont-call-get-on-a-provider-outside-a-task-action[Don't call get on a Provider outside a task action]
