<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Spring framework 4.3 introduced variants of the <code>@RequestMapping</code> annotation to better represent the semantics of the annotated methods. The use of <code>@GetMapping</code>, <code>@PostMapping</code>, <code>@PutMapping</code>, <code>@PatchMapping</code> and <code>@DeleteMapping</code> should be preferred to the use of the raw <code>@RequestMapping(method = RequestMethod.XYZ)</code>.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">@RequestMapping(path = "/greeting", method = RequestMethod.GET) // Noncompliant
public Greeting greeting(@RequestParam(value = "name", defaultValue = "World") String name) {
...
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">@GetMapping(path = "/greeting") // Compliant
public Greeting greeting(@RequestParam(value = "name", defaultValue = "World") String name) {
...
}</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Replace "@RequestMapping(method = RequestMethod.GET)" with "@GetMapping"</p>
</div>
<div class="paragraph">
<p>Replace "@RequestMapping(method = RequestMethod.POST)" with "@PostMapping"</p>
</div>
<div class="paragraph">
<p>Replace "@RequestMapping(method = RequestMethod.PUT)" with "@PutMapping"</p>
</div>
<div class="paragraph">
<p>Replace "@RequestMapping(method = RequestMethod.PATCH)" with "@PatchMapping"</p>
</div>
<div class="paragraph">
<p>Replace "@RequestMapping(method = RequestMethod.DELETE)" with "@DeleteMapping"</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>Primary: "@RequestMapping"</p>
</div>
<div class="paragraph">
<p>Secondary:  "RequestMethod.XZY"</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_2_mar_2018_135449_alexandre_gigleux_wrote">on 2 Mar 2018, 13:54:49 Alexandre Gigleux wrote:</h3>
<div class="paragraph">
<p>This rule should not raise an issue if the <code>method</code> parameter is not set. <a data-rspec-id="S3752" class="rspec-auto-link">RSPEC-3752</a> is there to catch missing <code>method</code>.</p>
</div>
</div>
</div>
</div>