<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when an object is created but not assigned to a variable, and later code attempts to reference an undefined variable that would logically hold that object.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>When you create an object in Ruby using constructor methods like <code>new</code> or <code>create</code>, the object exists in memory but disappears immediately if not assigned to a variable. If you later try to reference a variable that you assume holds this object, Ruby will raise a <code>NameError: undefined local variable</code> exception.</p>
</div>
<div class="paragraph">
<p>This commonly happens when developers are learning Ruby or Rails and forget the assignment step. For example, calling <code>User.new(name: "John")</code> creates a User object, but without assignment (<code>user = User.new(name: "John")</code>), the object cannot be accessed later.</p>
</div>
<div class="paragraph">
<p>The Ruby interpreter has no way to guess that when you reference <code>user</code> later, you meant the User object you created earlier. Each variable must be explicitly defined before use.</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="paragraph">
<p>This issue will cause your application to crash at runtime with a <code>NameError</code> exception. The error will occur every time the code path is executed, making the affected functionality completely unusable until fixed.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Assign the created object to a variable so it can be referenced later.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-ruby" data-lang="ruby"># Object created but not assigned
User.new(name: "John", email: "john@example.com")
# Later attempt to use undefined variable
user.save  # Noncompliant</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-ruby" data-lang="ruby"># Properly assign object to variable
user = User.new(name: "John", email: "john@example.com")
user.save</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Ruby Variables - <a href="https://ruby-doc.org/core/doc/syntax/assignment_rdoc.html">Official Ruby documentation on variable assignment</a></p>
</li>
<li>
<p>Ruby NameError - <a href="https://ruby-doc.org/core/NameError.html">Documentation for NameError exceptions in Ruby</a></p>
</li>
</ul>
</div>
</div>
<div class="sect2">
<h3 id="_related_rules">Related rules</h3>
<div class="ulist">
<ul>
<li>
<p><a data-rspec-id="S3827" class="rspec-auto-link">S3827</a> - <a href="https://rules.sonarsource.com/javascript/RSPEC-3827">Similar rule for JavaScript detecting unused object creation</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>