== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,swift]
----
u8a = ++u8b + u8c--
foo = bar++ / 4
----

=== Compliant solution

The following sequence is clearer and therefore safer:

[source,swift]
----
++u8b  
u8a = u8b + u8c
u8c--
foo = bar / 4
bar++
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../highlighting.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
