include::../description.adoc[]

== Noncompliant Code Example

----
u8a = ++u8b + u8c--
foo = bar++ / 4
----

== Compliant Solution

The following sequence is clearer and therefore safer:

----
++u8b  
u8a = u8b + u8c
u8c--
foo = bar / 4
bar++
----

include::../see.adoc[]

ifdef::rspecator-view[]
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::rspecator-view[]
