<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when using an <code>assert</code> statement on a tuple literal.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>When tested for truthiness, a sequence or collection will evaluate to <code>False</code> if it is empty (its <code>__len__</code> method returns 0) and to <code>True</code> if it contains at least one element.</p>
</div>
<div class="paragraph">
<p>Using the <code>assert</code> statement on a tuple literal will therefore always fail if the tuple is empty, and always succeed otherwise.</p>
</div>
<div class="paragraph">
<p>The <code>assert</code> statement does not take parentheses around its parameters. Calling <code>assert(x, y)</code> will test if the tuple <code>(x, y)</code> is True, which is always the case.</p>
</div>
<div class="paragraph">
<p>There are two possible fixes:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>If your intention is to test the first value of the tuple and use the second value as a message, simply remove the parentheses.</p>
</li>
<li>
<p>If your intention is to check that every element of the tuple is <code>True</code>, test each value separately.</p>
</li>
</ul>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">def test_values(a, b):
    assert (a, b)  # Noncompliant: will always be True</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">def test_values(a, b):
    # If you mean to test "a" and use "b" as an error message
    assert a, b

    # If you mean to test the values of "a" and "b"
    assert a and b</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p><a href="https://docs.python.org/3/reference/simple_stmts.html#the-assert-statement">Python documentation - The <code>assert</code> statement</a></p>
</li>
<li>
<p><a href="https://docs.python.org/3/library/stdtypes.html#truth-value-testing">Python documentation - Truth Value Testing</a></p>
</li>
</ul>
</div>
<hr>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Fix this assertion on a tuple literal; did you mean "assert A, B".</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>The tuple parameter</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_deprecates_s1721">deprecates: <a data-rspec-id="S1721" class="rspec-auto-link">S1721</a></h3>

</div>
</div>
</div>