<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>While <code>END</code> will adequately close a statement, it is less clear than the use of the relevant, statement-specific <code>ENDxx</code>. Thus, the statement-specific version is preferred to facilitate code maintenance and enhance clarity.</p>
</div>
<div class="paragraph">
<p>This rule is applied to the following operations: <code>CASxx</code>, <code>DO</code>, <code>DOU</code>, <code>DOUxx</code>, <code>DOW</code>, <code>DOWxx</code>, <code>FOR</code>, <code>IF</code>, <code>IFxx</code> and <code>SELECT</code> groups.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-rpg" data-lang="rpg">C     W0PKEX        DOUEQ     W0ON
...
C                   END

C     W0PKEX        IFEQ      W0ON
...
C                   END</code></pre>
</div>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-rpg" data-lang="rpg"> /free
  if x = 1;
    ...
  end;
 /end-free</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-rpg" data-lang="rpg">C     W0PKEX        DOUEQ     W0ON
...
C                   ENDDO

C     W0PKEX        IFEQ      W0ON
...
C                   ENDIF</code></pre>
</div>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-rpg" data-lang="rpg"> /free
  if x = 1;
    ...
  endif;
 /end-free</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Use the "[ENDxx]" statement to close this [xxx] clause.</p>
</div>
</div>
</div>
</div>