<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when text-based file operations are performed without explicitly specifying a charset encoding.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>When reading from or writing to files containing text, the system uses a default charset encoding if none is specified. This default encoding varies between operating systems and system configurations:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>Windows typically uses CP1252 or UTF-16</p>
</li>
<li>
<p>Linux and macOS typically use UTF-8</p>
</li>
<li>
<p>Some systems may use ISO-8859-1 or other encodings</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>This creates several problems:</p>
</div>
<div class="paragraph">
<p><strong>Portability Issues</strong>: Code that works correctly on one system may fail or produce incorrect results on another system with a different default encoding.</p>
</div>
<div class="paragraph">
<p><strong>Data Corruption</strong>: Files containing non-ASCII characters (accented letters, symbols, emojis, or text in non-Latin scripts) may be read incorrectly or written in a way that corrupts the data.</p>
</div>
<div class="paragraph">
<p><strong>Silent Failures</strong>: Encoding issues often don&#8217;t cause immediate exceptions but instead produce garbled text that may only be noticed later in the application lifecycle.</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="paragraph">
<p>Files may be corrupted or misread when moved between systems with different default encodings. Non-ASCII characters may appear as question marks, boxes, or other garbled text. In severe cases, this can lead to data loss or application failures when processing international content.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Specify the charset explicitly when reading file content.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-groovy" data-lang="groovy">def content = file.getText() // Noncompliant</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-groovy" data-lang="groovy">def content = file.getText('UTF-8') // Compliant</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">

</div>
</div>