<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>After calling CICS commands with the <code>RESP</code> or <code>NOHANDLE</code> options, the return code should be tested.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cobol" data-lang="cobol">  EXEC CICS DELETEQ TS        *&gt; Noncompliant; WS-STATUS should have been tested before the MOVE
    QNAME(WS-TS5FTARF-NAME)
    RESP(WS-STATUS)
  END-EXEC.
  MOVE WS-EIBTASKN (4:4) TO WS-TS5FTAR1-NAME-TSKID.</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cobol" data-lang="cobol">  EXEC CICS DELETEQ TS
    QNAME(WS-TS5FTARF-NAME)
    RESP(WS-STATUS)
  END-EXEC.
  IF WS-STATUS ...

  MOVE WS-EIBTASKN (4:4) TO WS-TS5FTAR1-NAME-TSKID.</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_is_related_to_s1587">is related to: <a data-rspec-id="S1587" class="rspec-auto-link">S1587</a></h3>

</div>
<div class="sect2">
<h3 id="_on_1_feb_2016_112942_pierre_yves_nicolas_wrote">on 1 Feb 2016, 11:29:42 Pierre-Yves Nicolas wrote:</h3>
<div class="paragraph">
<p>\[~freddy.mallet] The title and description of this rule mention <code>NOHANDLE</code> but the current implementation does nothing related to <code>NOHANDLE</code> and it seems that there&#8217;s nothing to test when <code>NOHANDLE</code> is used according to the <a href="https://www-01.ibm.com/support/knowledgecenter/SSGMCP_5.1.0/com.ibm.cics.ts.applicationprogramming.doc/topics/dfhp4_nohandleintro.html">documentation</a>.</p>
</div>
</div>
</div>
</div>