include::../description.adoc[]

== Noncompliant Code Example

----
SET NAME TO NAME.    *> Noncompliant
MOVE NAME TO NAME.   *> Noncompliant
COMPUTE NAME = NAME. *> Noncompliant
EXEC SQL
  UPDATE PERSON
    SET NAME = NAME  -- Noncompliant
  WHERE ID = :PERSON_ID
END-EXEC.
----

== Compliant Solution

----
SET NAME TO NEW_NAME.
MOVE NEW_NAME TO NAME. 
COMPUTE NAME = NEW_NAME.
EXEC SQL
  UPDATE PERSON
    SET NAME = :NEW_NAME
  WHERE ID = :PERSON_ID
END-EXEC.
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
