<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>By default, the Maven Surefire plugin only executes test classes with names that end in "Test" or "TestCase". Name your class "TestClassX.java", for instance, and it will be skipped.</p>
</div>
<div class="paragraph">
<p>This rule raises an issue for each test class with a name not ending in "Test" or "TestCase".</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">public class TestClassX {  // Noncompliant
  @Test
  public void testDoTheThing() {
    //...</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">public class ClassXTest {
  @Test
  public void testDoTheThing() {
    //...</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Rename this class to end with "Test" or "TestCase" to ensure it will be executed by the Maven Surefire plugin.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>primary: the class name</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_7_oct_2015_125209_ann_campbell_wrote">on 7 Oct 2015, 12:52:09 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>Assigned for review&#8230;&#8203;</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_7_oct_2015_125426_ann_campbell_wrote">on 7 Oct 2015, 12:54:26 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~nicolas.peru], [~pawlak] probably has a pull request in the wings for this one</p>
</div>
<div class="paragraph">
<p><a href="https://groups.google.com/forum/#!searchin/sonarqube/non-detectable$20test$20method/sonarqube/xn0O_0Aa9aQ/ysPUTNuOAwAJ" class="bare">https://groups.google.com/forum/#!searchin/sonarqube/non-detectable$20test$20method/sonarqube/xn0O_0Aa9aQ/ysPUTNuOAwAJ</a></p>
</div>
</div>
</div>
</div>