<body class="book">
<div id="header">
<h1>Invalid header level</h1>
</div>
<div id="content">
<div id="preamble">
<div class="sectionbody">
<div class="paragraph">
<p>Shared naming conventions allow teams to collaborate efficiently. This rule checks that all function names match a provided regular expression.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_invalid_header">Invalid header</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="paragraph">
<p>With default provided regular expression: <code>^[a-z][a-zA-Z0-9]*$</code>:</p>
</div>
<div class="listingblock">
<div class="content">
<pre>void DoSomething (void);</pre>
</div>
</div>
<div class="sect2">
<h3 id="_parameters">Parameters</h3>
<div class="paragraph">
<p><strong>format</strong>: STRING [ <code>^([A-Z0-9_]*|[a-z0-9_]*)$</code> ]
  Regular expression used to check the [method|function|subroutine] names against</p>
</div>
<div class="dlist">
<dl>
<dt class="hdlist1"><strong>param2</strong>: INTEGER [ <code>42</code> ]</dt>
<dd>
<p>Some integer</p>
</dd>
</dl>
</div>
</div>
</div>
</div>
</div>
<div id="footer">
<div id="footer-text">
Last updated 2022-01-28 19:05:40 +0100
</div>
</div>
</body>