= Invalid header level

Shared naming conventions allow teams to collaborate efficiently. This rule checks that all function names match a provided regular expression.

== Invalid header

== Noncompliant Code Example

With default provided regular expression: ``++^[a-z][a-zA-Z0-9]*$++``:

----
void DoSomething (void);
----

=== Parameters

*format*: STRING [ `+^([A-Z0-9_]*|[a-z0-9_]*)$+` ]
  Regular expression used to check the [method|function|subroutine] names against

*param2*: INTEGER [ `+42+` ]::
  Some integer
