<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This is an issue when the <code>@TimedInterrupt</code> annotation is applied to a static method.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>The <code>@TimedInterrupt</code> annotation is designed to automatically add timeout checks to methods by inserting interruption checks at strategic points in the code. However, this transformation is currently not compatible with static methods due to implementation limitations in how the transformation processes method contexts.</p>
</div>
<div class="paragraph">
<p>When <code>@TimedInterrupt</code> is applied to a static method, it can lead to unexpected behavior or compilation issues because the transformation logic assumes instance method semantics. The annotation works by injecting timeout verification code that may not function correctly in a static context, potentially causing the timeout mechanism to fail silently or throw unexpected exceptions.</p>
</div>
<div class="paragraph">
<p>This incompatibility can result in unreliable timeout behavior, making it difficult to control long-running static operations as intended.</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="paragraph">
<p>Using <code>@TimedInterrupt</code> on static methods can lead to:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>Unreliable timeout behavior where the intended time limits are not enforced</p>
</li>
<li>
<p>Potential compilation errors or runtime exceptions</p>
</li>
<li>
<p>Silent failures where the timeout mechanism doesn&#8217;t work as expected</p>
</li>
<li>
<p>Difficulty in controlling long-running static operations</p>
</li>
<li>
<p>Unpredictable application behavior when timeouts should have been triggered</p>
</li>
</ul>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Convert the static method to an instance method.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-groovy" data-lang="groovy">class MyClass {
    @TimedInterrupt(value=1, unit=TimeUnit.SECONDS)
    static void staticMethod() { // Noncompliant
        // Long-running operation
    }
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-groovy" data-lang="groovy">class MyClass {
    @TimedInterrupt(value=1, unit=TimeUnit.SECONDS)
    void instanceMethod() {
        // Long-running operation
    }
}</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Groovy documentation - <a href="https://groovy-lang.org/metaprogramming.html#xform-TimedInterrupt">TimedInterrupt</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>