<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Rvalue references were introduced as part of C&#43;&#43;11. They are thus a new feature of the language, and are not yet widely understood. Using them is complicated, and code using rvalue references may be difficult to understand.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">std::vector&lt;int&gt; return_vector(void)
{
    std::vector&lt;int&gt; tmp {1,2,3,4,5};
    return tmp;
}

std::vector&lt;int&gt; &amp;&amp;rval_ref = return_vector(); // Noncompliant</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">std::vector&lt;int&gt; return_vector(void)
{
    std::vector&lt;int&gt; tmp {1,2,3,4,5};
    return tmp;
}

const std::vector&lt;int&gt;&amp; rval_ref = return_vector();</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Rework this logic to eliminate the rvalue reference.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_8_aug_2016_215155_ann_campbell_wrote">on 8 Aug 2016, 21:51:55 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~massimo.paladin] please add a Noncompliant comment to the Noncompliant Code Example</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_15_aug_2016_092021_massimo_paladin_wrote">on 15 Aug 2016, 09:20:21 Massimo PALADIN wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] done.</p>
</div>
</div>
</div>
</div>