<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This is an issue when GString expressions (strings containing <code>${}</code> interpolation or <code>$variable</code> syntax) are used as keys in map literals or when calling map methods like <code>put()</code>.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>GString objects and String objects have different hash codes, even when they represent identical text. This causes maps to fail matching keys when mixing GStrings and Strings.</p>
</div>
<div class="paragraph">
<p>When a GString is used as a map key, lookups with an equivalent String will fail. For example, storing with <code>"${name}"</code> (where <code>name</code> is <code>"John"</code>) and retrieving with <code>"John"</code> returns <code>null</code>.</p>
</div>
<div class="paragraph">
<p>This violates expectations and leads to bugs, especially when keys are created dynamically in one part of the code and used as literals elsewhere.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_what_is_the_potential_impact">What is the potential impact?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Using GString expressions as map keys can cause data loss, logic errors, silent failures, and inconsistent behavior. These issues may result in data corruption, incorrect logic, or reliability problems in production.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Use regular variables or String literals as map keys instead of GString expressions. This ensures consistent hash codes and reliable key lookups.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-groovy" data-lang="groovy">def key = "a"
def m = ["${key}": "letter ${key}"] // Noncompliant
assert m["a"] == null  // fails to find the value</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-groovy" data-lang="groovy">def key = "a"
def m = [(key): "letter ${key}"]
assert m["a"] == "letter a"  // works correctly</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Groovy Documentation - <a href="https://groovy-lang.org/syntax.html#_gstring_and_string_hashcodes">Syntax</a></p>
</li>
<li>
<p>Groovy Documentation - <a href="https://groovy-lang.org/groovy-dev-kit.html#Collections-Maps">Groovy development kit</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>