include::../description.adoc[]

== Noncompliant Code Example

----
COMPUTE x = (y / 2 + 1).  *> Compliant even if the parenthesis are ignored by the compiler
COMPUTE y = 2 * ((x + 1)).  *> Noncompliant
----

== Compliant Solution

----
COMPUTE x = (y / 2 + 1).
COMPUTE y = 2 * (x + 1).
----

ifdef::env-github,rspecator-view[]
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
