== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,cobol]
----
COMPUTE x = (y / 2 + 1).  *> Compliant even if the parenthesis are ignored by the compiler
COMPUTE y = 2 * ((x + 1)).  *> Noncompliant
----

=== Compliant solution

[source,cobol]
----
COMPUTE x = (y / 2 + 1).
COMPUTE y = 2 * (x + 1).
----

include::../rspecator.adoc[]
