<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>A hardcoded file path is a guarantee that eventually the program will fail. It may happen because the paths on the target machine changed or because the application was deployed on an OS other than the one on which it was developed. After all, not every OS has a "C:" drive, just has not every OS has a "/home" directory.</p>
</div>
<div class="paragraph">
<p>This rule checks for hardcoded, absolute paths in <code>Files</code> and all types of input and output streams.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">public void readProperties() {
  File in = new File("C:/myappdir/app.properties");  // Noncompliant
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">public void readProperties(String path) {
  File in = new File(path + "app.properties");
}</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_duplicates_s1075">duplicates: <a data-rspec-id="S1075" class="rspec-auto-link">S1075</a></h3>

</div>
<div class="sect2">
<h3 id="_on_10_oct_2014_143548_freddy_mallet_wrote">on 10 Oct 2014, 14:35:48 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>If my feeling is correct @Ann, this rule is duplicated by <a data-rspec-id="S1075" class="rspec-auto-link">RSPEC-1075</a>. Perhaps we should merge the two into one.</p>
</div>
</div>
</div>
</div>