This rule raises an issue when multiple operations performed sequentially on a data frame could be replaced with method chaining.

== Why is this an issue?

Pandas provides an easy way to apply multiple operations to a DataFrame through method chaining. When intermediate results are not needed, it is generally recommended to chain methods rather than assigning the result of each operation back to a variable and then applying subsequent operations to that variable.

Method chaining leads to clearer and more concise code and is therefore the recommended practice. This rule will raise an issue when three or more consecutive operations are performed on a DataFrame without any use of the intermediate results.

=== Code examples

==== Noncompliant code example

[source,python,diff-id=1,diff-type=noncompliant]
----
df_filtered = df[df['A'] > 2]
df_grouped = df_filtered.groupby('B').sum()
df_sorted = df_grouped.sort_values('C', ascending=False)
result = df_sorted.reset_index()  # Noncompliant: The multiple reassignments are not needed.
----

==== Compliant solution

[source,python,diff-id=1,diff-type=compliant]
----
result = (df[df['A'] > 2]
    .groupby('B')
    .sum()
    .sort_values('C', ascending=False)
    .reset_index()
)  # Compliant: method chaining is clearer and more intentional
----


== Resources
=== Articles & blog posts

* Practical Data Science - https://practicaldatascience.co.uk/data-science/how-to-use-method-chaining-in-pandas[How to use method chaining in Pandas]
