== How to fix it in .NET

=== Code examples

The example uses a hardcoded IV as a nonce, which causes AES-CCM to be insecure. To fix it, a nonce is randomly generated instead.

==== Noncompliant code example

[source,csharp,diff-id=101,diff-type=noncompliant]
----
public void encrypt(byte[] key, byte[] ptxt, byte[] ciphertext, byte[] tag)
{
    var nonce = Encoding.UTF8.GetBytes("7cVgr5cbdCZV");

    using var cipher = new AesGcm(key);

    cipher.Encrypt(nonce, plaintext, ciphertext, tag); // Noncompliant
}
----

==== Compliant solution

[source,csharp,diff-id=101,diff-type=compliant]
----
public void encrypt(byte[] key, byte[] ptxt, byte[] ciphertext, byte[] tag)
{
    var nonce = new byte[AesGcm.NonceByteSizes.MaxSize];
    RandomNumberGenerator.Fill(nonce);

    using var cipher = new AesGcm(key);

    cipher.Encrypt(nonce, plaintext, ciphertext, tag);
}
----

=== How does this work?

When using AES-GCM or AES-CCM, NIST recommends a 96-bit length nonce using a deterministic approach or at least 96 bits using a 'Random Bit Generator (RBG)'.

include::../../common/fix/randomized-nonce.adoc[]

include::../../common/fix/deterministic-nonce.adoc[]
