== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,csharp]
----
private void DoSomething()
{
    // ...
    Console.WriteLine("so far, so good..."); // Noncompliant
    // ...
}
----

=== Exceptions

The following are ignored by this rule:

* Console Applications
* Calls in methods decorated with ``++[Conditional ("DEBUG")]++``
* Calls included in DEBUG preprocessor branches (``++#if DEBUG++``)

include::../see.adoc[]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
