include::../description-dotnet.adoc[]

include::../why-dotnet.adoc[]

include::../how-dotnet.adoc[]

=== Code examples

==== Noncompliant code example

[source,vbnet,diff-id=1,diff-type=noncompliant]
----
logger.LogDebug("User {firstName} logged in", firstName) ' Noncompliant
----

==== Compliant solution

[source,vbnet,diff-id=1,diff-type=compliant]
----
logger.LogDebug("User {FirstName} logged in", firstName) ' Compliant
----

include::../resources-dotnet.adoc[]

include::../rspecator.adoc[]