== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,vbnet,diff-id=1,diff-type=noncompliant]
----
If a AndAlso ((x + y > 0)) Then ' Noncompliant
    ' ...
End If

Return ((x + 1))  ' Noncompliant
----

=== Compliant solution

[source,vbnet,diff-id=1,diff-type=compliant]
----
If a AndAlso x + y > 0 Then
    ' ...
End If

Return x + 1
----

include::../rspecator.adoc[]
