include::../description.adoc[]

== Noncompliant Code Example

----
If a AndAlso ((x + y > 0)) Then ' Noncompliant
    ' ...
End If

Return ((x + 1))  ' Noncompliant
----

== Compliant Solution

----
If a AndAlso x + y > 0 Then
    ' ...
End If

Return (x + 1)
----

ifdef::rspecator-view[]
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::rspecator-view[]
