include::../description.adoc[]

== Noncompliant Code Example

[source,vbnet]
----
If a AndAlso ((x + y > 0)) Then ' Noncompliant
    ' ...
End If

Return ((x + 1))  ' Noncompliant
----

== Compliant Solution

[source,vbnet]
----
If a AndAlso x + y > 0 Then
    ' ...
End If

Return (x + 1)
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../highlighting.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
