<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p><code>Path.of()</code> provides a modern and consistent approach to creating Path instances, aligning with factory methods such as <code>List.of()</code> and <code>Set.of()</code>.
Additionally, <code>Paths.get()</code> may be deprecated in future versions of Java.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Replace <code>Paths.get()</code> with <code>Path.of()</code>.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">Path path = Paths.get("my/file.txt"); // non compliant</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">Path path = Path.of("my/file.txt"); // compliant</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Java Documentation - <a href="https://docs.oracle.com/en/java/javase/21/docs/api/java.base/java/nio/file/Paths.html">Paths</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>