All empty or `null` properties that are not required should be excluded from the template samples.
This includes empty objects `{}`, arrays `[]`, strings `""`, and any property that has a `null` value.

== Why is this an issue?

Empty or null elements are usually introduced by mistake.
They are useless and prevent readability of the code.

=== Exceptions

The top-level JSON template properties: `parameters`, `variables`, `functions`, `resources` and `outputs`.

== How to fix it

They should be removed or completed with real code.

=== Code examples

==== Noncompliant code example

[source,json,diff-id=1,diff-type=noncompliant]
----
{
  "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#",
  "contentVersion": "1.0.0.0",
  "resources": [
    {
      "type": "Microsoft.Storage/storageAccounts",
      "apiVersion": "",
      "name": null,
      "tags": {},
      "dependsOn": []
    }
  ]
}
----

==== Compliant solution

[source,json,diff-id=1,diff-type=compliant]
----
{
  "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#",
  "contentVersion": "1.0.0.0",
  "resources": [
    {
      "type": "Microsoft.Storage/storageAccounts",
      "apiVersion": "2023-01-01",
      "name": "myStorage"
    }
  ]
}
----

== Resources

=== External coding guidelines

* Azure/azure-quickstart-templates - https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md#empty-and-null-properties[best-practices]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Remove this [empty|null] [object|array|string|property] or completed with real code.

=== Highlighting

Highlight property name and empty/null value.

'''


endif::env-github,rspecator-view[]
