== Why is this an issue?

When entering a critical section the program holds a lock that can be hold by a single entity at a time. It is then recommended that the critical section is reduced to the minimum required and executed without any interruption.


=== Noncompliant code example

[source,cpp]
----
void f() {
  std::mutex m;
  // ...
  m.lock();
  // ...
  sleep(1); // Noncompliant
  // ...
  m.unlock();
}
----

