=== How to fix it in Flask

include::../../common/fix/code-rationale.adoc[]

==== Noncompliant code example

[source,python,diff-id=1,diff-type=noncompliant]
----
@app.route("/redirect")
def redirect():
    endpoint = request.args["url"]
    return redirect(endpoint) # Noncompliant
----

==== Compliant solution

[source,python,diff-id=1,diff-type=compliant]
----
@app.route("/redirect")
def redirect():
    endpoint = request.args["url"]
    return redirect(url_for(endpoint))
----

include::../../common/fix/how-does-this-work.adoc[]

=== Pitfalls

include::../../common/pitfalls/starts-with.adoc[]
