Catching group exceptions with `expect*` will raise a Runtime error.

== Why is this an issue?

Python 3.11 introduced ``++except*++`` and ``++ExceptionGroup++`` which allows to handle and raise multiple unrelated exceptions simultaneously.
While it is possible to catch the ExceptionGroup and BaseExceptionGroup types with ``++except++``, a Runtime error will be raised when this is done with ``++except*++``.


== How to fix it

=== Code examples

Group exceptions should only be caught with ``++except++`.

==== Noncompliant code example

[source,python,diff-id=1,diff-type=noncompliant]
----
try:
    ...
except* ExceptionGroup:  # Noncompliant: Runtime error  
    pass

try:
    ...
except* (TypeError, ExceptionGroup):  # Noncompliant: Runtime error
    pass
----

==== Compliant solution

[source,python,diff-id=1,diff-type=compliant]
----
try:
    ...
except ExceptionGroup:
    pass

try:
    ...
except (TypeError, ExceptionGroup):
    pass
----

== Resources

=== Documentation

* https://peps.python.org/pep-0654/#forbidden-combinations[PEP-654] - Forbidden combinations
