== Why is this an issue?

Python 3.11 introduced `except*` and `ExceptionGroup`.
While it's possible to catch the ExceptionGroup and BaseExceptionGroup types with `except``, a Runtime error will be raised when this is done with `except*`.
See PEP 654 : https://peps.python.org/pep-0654/#forbidden-combinations[PEP-654]

=== Noncompliant code example

[source,python]
----
try:
    ...
except* ExceptionGroup:  # Noncompliant
    pass

try:
    ...
except* (TypeError, ExceptionGroup):  # Noncompliant
    pass
----

=== Compliant solution

[source,python]
----
try:
    ...
except ExceptionGroup:
    pass
----

== Resources

https://peps.python.org/pep-0654/#forbidden-combinations[PEP-654]
