== How to fix it in OpenSSL

=== Code examples

==== Noncompliant code example

[source,python,diff-id=11,diff-type=noncompliant]
----
from OpenSSL import SSL

SSL.Context(SSL.SSLv3_METHOD)  # Noncompliant
----

==== Compliant solution

[source,python,diff-id=11,diff-type=compliant]
----
from OpenSSL import SSL

context = SSL.Context(SSL.TLS_SERVER_METHOD)
context.set_min_proto_version(SSL.TLS1_3_VERSION)
----

=== How does this work?

include::../../common/fix/fix.adoc[]
