== How to fix it in Python Standard Library

=== Code examples

==== Noncompliant code example

[source,python,diff-id=21,diff-type=noncompliant]
----
import ssl

ssl.SSLContext(ssl.PROTOCOL_SSLv3) # Noncompliant
----

==== Compliant solution

[source,python,diff-id=21,diff-type=compliant]
----
import ssl

context = ssl.SSLContext(ssl.PROTOCOL_TLS_SERVER)
context.minimum_version = ssl.TLSVersion.TLSv1_3
----

=== How does this work?

include::../../common/fix/fix.adoc[]
