Class templates can be explicitly or partially specialized. But according to the {cpp} standard, function templates cannot be partially specialized. Under certain conditions, the MicrosoftÂ® compiler will silently ignore the confusing application of partial specialization syntax to a function, but other compilers will raise an error for it and fail compilation.


== Noncompliant Code Example

----
template<typename T>
void fun(T p);

template<typename T>
void fun<T>(T p) { // Noncompliant
  // ...
}
----


== Compliant Solution

----
template<typename T>
void fun(T p);

template<typename T>
void fun(T p) {
  // ...
}
----



ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

include::highlighting.adoc[]

'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
