<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>The use of <code>break</code> and <code>continue</code> statements increases the complexity of the control flow and makes it harder to understand the program logic.
In order to keep a good program structure, they should not be applied more than once per loop.</p>
</div>
<div class="paragraph">
<p>This rule reports an issue when there is more than one <code>break</code> or <code>continue</code> statement in a loop.
The code should be refactored to increase readability if there is more than one.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-javascript" data-lang="javascript">for (var i = 1; i &lt;= 10; i++) {  // Noncompliant - 2 continue - one might be tempted to add some logic in between
  if (i % 2 == 0) {
    continue;
  }

  if (i % 3 == 0) {
    continue;
  }

  alert("i = " + i);
}</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Reduce the total number of ["break" and "continue"|"CONTINUE", "EXIT" and "CHECK"] statements in this loop to use one at most.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="ulist">
<ul>
<li>
<p>Primary: the <code>for</code> or <code>while</code> keyword</p>
</li>
<li>
<p>Secondary: each <code>continue</code> and <code>break</code> keyword</p>
<div class="ulist">
<ul>
<li>
<p>message 1: "break" statement.</p>
</li>
<li>
<p>message 2: "continue" statement.</p>
</li>
</ul>
</div>
</li>
</ul>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_is_duplicated_by_s924">is duplicated by: <a data-rspec-id="S924" class="rspec-auto-link">S924</a></h3>

</div>
<div class="sect2">
<h3 id="_is_related_to_s924">is related to: <a data-rspec-id="S924" class="rspec-auto-link">S924</a></h3>

</div>
<div class="sect2">
<h3 id="_on_9_aug_2013_103838_freddy_mallet_wrote">on 9 Aug 2013, 10:38:38 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>Is implemented by http://jira.codehaus.org/browse/SONARJAVA-283</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_10_aug_2013_000012_ann_campbell_wrote">on 10 Aug 2013, 00:00:12 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>The format of this rule title is "Do not&#8230;&#8203;"</p>
</div>
<div class="paragraph">
<p>I thought we were going for "Blah should (not) &#8230;&#8203;"</p>
</div>
<div class="paragraph">
<p>Is either format acceptable, or should I rewrite this one:</p>
</div>
<div class="paragraph">
<p>'Loops should not contain more than a single "break" or "continue" statement' ?</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_12_aug_2013_121319_dinesh_bolkensteyn_wrote">on 12 Aug 2013, 12:13:19 Dinesh Bolkensteyn wrote:</h3>
<div class="paragraph">
<p>Just go ahead and fix whatever needs to be fixed Ann ;-)</p>
</div>
<div class="paragraph">
<p>Indeed you&#8217;re right!</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_12_aug_2013_233207_ann_campbell_wrote">on 12 Aug 2013, 23:32:07 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>As my grandfather used to say: It gets monotonous ;-P</p>
</div>
</div>
</div>
</div>