<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p><code>bool std::uncaught_exception()</code> allows you to know whether a thread is in an exception stack unwinding context. However, its practical functionality was restricted.</p>
</div>
<div class="paragraph">
<p>C&#43;&#43;17 deprecates <code>bool std::uncaught_exception()</code> and introduces <code>int std::uncaught_exceptions()</code> which returns the number of uncaught exceptions. The code example below shows how you can benefit from this new improved function.</p>
</div>
<div class="paragraph">
<p><code>std::uncaught_exception</code> has been removed in C&#43;&#43;20.</p>
</div>
<div class="paragraph">
<p>This rule will flag any usage of <code>std::uncaught_exception</code>.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">class Transaction {

  // ...

  ~Transaction() {
    if (!std::uncaught_exception()) { // Noncompliant, replace std::uncaught_exception by std::uncaught_exceptions
      // commit
    } else {
      // rollback
    }
  }
};</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="paragraph">
<p>The following example shows how <code>std::uncaught_exceptions</code> can be used to determine in <code>~Transaction</code> if a new exception was thrown since <code>t1</code>/<code>t2</code> creation.</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">class Transaction {

  // ...

  ~Transaction() {
    if (initialUncaughtExceptions == std::uncaught_exceptions()) {
      // commit
    } else {
      // rollback
    }
  }

  // ...

  int initialUncaughtExceptions = std::uncaught_exceptions();
};

int f() {
  try {
    Transaction t1;
    // ... something here could throw
  } catch(...) {
    Transaction t2;
    // ... something here could throw
  }
}</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Replace this use of std::uncaught_exception by std::uncaught_exceptions.</p>
</div>
</div>
</div>
</div>