== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,text]
----
using System;

namespace MyLibrary  // Noncompliant
{
}
----


=== Compliant solution

[source,text]
----
using System;

[assembly: System.Runtime.InteropServices.ComVisible(false)]
namespace MyLibrary
{
}
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../highlighting.adoc[]

endif::env-github,rspecator-view[]
