Creating a new ``++Error++`` without actually throwing it is useless and is probably due to a mistake.

== Noncompliant Code Example

----
if (x < 0) {
  new Error("x must be nonnegative");
}
----

== Compliant Solution

----
if (x < 0) {
  throw new Error("x must be nonnegative");
}
----
ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

include::../highlighting.adoc[]

endif::env-github,rspecator-view[]
