== Why is this an issue?

Creating a new ``++Error++`` without actually throwing it is useless and is probably due to a mistake.

=== Noncompliant code example

[source,javascript]
----
if (x < 0) {
  new Error("x must be nonnegative");
}
----

=== Compliant solution

[source,javascript]
----
if (x < 0) {
  throw new Error("x must be nonnegative");
}
----
ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Throw this error or remove this useless statement


include::../highlighting.adoc[]

endif::env-github,rspecator-view[]
