== Why is this an issue?

Trying to access a list or a tuple index that is beyond the size of the list/tuple will result in an `IndexError` and is probably a mistake.

=== Noncompliant code example

[source,python]
----
def noncompliant():
    ls = [1, 2, 3]
    foo(ls[3])  # Noncompliant

----

=== Compliant solution

[source,python]
----
def compliant():
    ls = [1, 2, 3]
    foo(ls[2])

----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

'''

endif::env-github,rspecator-view[]
