include::../description.adoc[]

== Noncompliant Code Example

----
public class Rectangle
{
  private readonly int length;
  private readonly int width;  // width is written but never read

  public Rectangle(int length, int width)
  {
    this.length = length;
    this.width = width;
  }

  public int Surface
  {
    get
    {
      return length * length;
    }
  }
}
----

== Compliant Solution

----
public class Rectangle
{
  private readonly int length;
  private readonly int width;

  public Rectangle(int length, int width)
  {
    this.length = length;
    this.width= width;
  }

  public int Surface
  {
    get
    {
      return length * width;
    }
  }
}
----

include::../see.adoc[]

ifdef::rspecator-view[]
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::rspecator-view[]
