== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,csharp]
----
public class Rectangle
{
  private readonly int length;
  private readonly int width;  // width is written but never read

  public Rectangle(int length, int width)
  {
    this.length = length;
    this.width = width;
  }

  public int Surface
  {
    get
    {
      return length * length;
    }
  }
}
----

=== Compliant solution

[source,csharp]
----
public class Rectangle
{
  private readonly int length;
  private readonly int width;

  public Rectangle(int length, int width)
  {
    this.length = length;
    this.width= width;
  }

  public int Surface
  {
    get
    {
      return length * width;
    }
  }
}
----

include::../see.adoc[]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../highlighting.adoc[]

'''
== Comments And Links
(visible only on this page)

=== on 15 Oct 2019, 10:44:10 Andrei Epure wrote:
Note that:

* the implementation of this rule only considers fields. Other members (e.g. properties)  are considered by the RSPEC-1144 C# implementation.
* to avoid overlap of this rule with RSPEC-1450, the RSPEC-1450 C# implementation has been modified to only raise when the field is written before being read

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
