<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Shared naming conventions allow teams to collaborate effectively. This rule raises an issue when an open curly brace is not placed at the end of a line of code.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-text" data-lang="text">if(condition)
{
  doSomething();
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-text" data-lang="text">if(condition) {
  doSomething();
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_exceptions">Exceptions</h3>
<div class="paragraph">
<p>When blocks are inlined (left and right curly braces on the same line), no issue is triggered.</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-text" data-lang="text">if(condition) {doSomething();}</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Move this open curly brace to the end of the previous line.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_is_related_to_s1106">is related to: <a data-rspec-id="S1106" class="rspec-auto-link">S1106</a></h3>

</div>
<div class="sect2">
<h3 id="_on_3_jul_2013_115730_freddy_mallet_wrote">on 3 Jul 2013, 11:57:30 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>Is implemented by http://jira.codehaus.org/browse/SONARJAVA-189</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_5_jul_2013_162758_dinesh_bolkensteyn_wrote">on 5 Jul 2013, 16:27:58 Dinesh Bolkensteyn wrote:</h3>
<div class="paragraph">
<p>"Compliant as when the condition is split over several lines this rule is switched off" ?</p>
</div>
<div class="paragraph">
<p>Arf this totally stinks&#8230;&#8203; that&#8217;s just way too complex for such a basic rule</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_19_feb_2014_083303_freddy_mallet_wrote">on 19 Feb 2014, 08:33:03 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>Is implemented by http://jira.codehaus.org/browse/SONARPLUGINS-3488 for PHP</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_3_mar_2014_135629_ann_campbell_wrote">on 3 Mar 2014, 13:56:29 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~dinesh.bolkensteyn], does this apply for C#?</p>
</div>
</div>
</div>
</div>