== Why is this an issue?

include::description.adoc[]

=== Noncompliant code example

[source,text]
----
if(condition)
{
  doSomething();
}
----


=== Compliant solution

[source,text]
----
if(condition) {
  doSomething();
}
----


=== Exceptions

When blocks are inlined (left and right curly braces on the same line), no issue is triggered. 


[source,text]
----
if(condition) {doSomething();} 
----

