== Why is this an issue?

The attribute ``++noreturn++`` indicates that a function does not return. 

Using this attribute allows the compilerÂ to do some assumptions that can lead to optimizations. However, if a function with this attribute ever returns, the behavior becomes undefined.


=== Noncompliant code example

[source,cpp]
----
[[noreturn]] void f () {
  while (1) {
    // ...
    if (/* something*/) {
      return; // Noncompliant, this function should not return
    }
  }
}
----


=== Compliant solution

[source,cpp]
----
[[noreturn]] void f() {  // Compliant
  while (true) {
    // ...
  }
}
----
Or

[source,cpp]
----
void f() {  
  while (true) {
    // ... 
    if (/* something*/) { 
      return; // Compliant 
    }
  }
}
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

function "XXX" declared 'noreturn' should not return


endif::env-github,rspecator-view[]
