== Why is this an issue?

Having characters before ``++<?php++`` can cause "Cannot modify header information" errors and similar problems with Ajax requests.


=== Noncompliant code example

[source,php]
----
test<?php  //Noncompliant
// ...
----
and

[source,php]
----
// Noncompliant; newline before opening tag
<?php
// ...
----


=== Compliant solution

[source,php]
----
<?php  
// ...
----



ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
