== Why is this an issue?

Having characters before ``++<?php++`` can cause "Cannot modify header information" errors and similar problems with Ajax requests.


=== Noncompliant code example

[source,php]
----
test<?php  //Noncompliant
// ...
----
and

[source,php]
----
// Noncompliant; newline before opening tag
<?php
// ...
----


=== Compliant solution

[source,php]
----
<?php  
// ...
----



ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Remove the extra characters before the open tag.


'''
== Comments And Links
(visible only on this page)

=== on 19 Sep 2014, 14:30:34 Freddy Mallet wrote:
@Ann, do you know the kind of problems we can face when having some characters before "<?php" ?

endif::env-github,rspecator-view[]
