== Why is this an issue?

include::../description.adoc[]

== How to fix it

include::../how-to-fix-it.adoc[]

=== Code examples

==== Noncompliant code example

[source,apex,diff-id=1,diff-type=noncompliant]
----
void doSomething(Integer a, Integer b) { // Noncompliant, "b" is unused
  compute(a);
}
----

==== Compliant solution

[source,apex,diff-id=1,diff-type=compliant]
----
void doSomething(Integer a) {
  compute(a);
}
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
