== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,apex]
----
void doSomething(Integer a, Integer b) {     // "b" is unused
  compute(a);
}
----

=== Compliant solution

[source,apex]
----
void doSomething(Integer a) {
  compute(a);
}
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
