Input data that is not a JCL statement implicitly generate a `DD *` statement with the name `SYSIN`.

== Why is this an issue?

This behavior is implicit and lead to confusion.
The feature allow flexibility in the syntax that can even lead to unexpected behavior.

For example, this code:
[source,jcl]
----
//MYDD DD DSN=TEST
Some in-stream data
// DD DSN=CONCAT-DD
----

Get implicitly transformed into:
[source,jcl]
----
//MYDD DD DSN=TEST
//SYSIN DD *
Some in-stream data
// DD DSN=CONCAT-DD
----
So the last DD statement is a concatenation of the implicit `//SYSIN DD *` statement instead of the explicit `//MYDD DD` statement.

== How to fix it

Add explicitly the `//SYSIN DD *` before the in-stream data.

=== Code examples

==== Noncompliant code example

[source,jcl,diff-id=1,diff-type=noncompliant]
----
Some in-stream data
----

==== Compliant solution

[source,jcl,diff-id=1,diff-type=compliant]
----
//SYSIN DD *
Some in-stream data
----

== Resources

=== Documentation

* https://techdocs.broadcom.com/us/en/ca-mainframe-software/automation/ca-jclcheck-workload-automation/12-0/reference/jclcheck-messages/cay6000-to-cay6099w/cay6073w-sysin-dd-statement-generated.html[JCLCheckâ„¢ Workload Automation 12.0 - CAY6073W]
