Input data that is not a JCL statement implicitly generate a `DD *` statement with the name `SYSIN`.

== Why is this an issue?

This behavior is implicit and may lead to confusion or unexpected behavior.

For example, this code:
[source,jcl]
----
//MYDD DD DSN=TEST
Some in-stream data
// DD DSN=CONCAT-DD
----

Gets implicitly transformed into:
[source,jcl]
----
//MYDD DD DSN=TEST
//SYSIN DD *
Some in-stream data
// DD DSN=CONCAT-DD
----
So the last DD statement is a concatenation of the implicit `//SYSIN DD *` statement instead of the explicit `//MYDD DD` statement.

== How to fix it

Add explicitly the `//SYSIN DD *` before the in-stream data.

=== Code examples

==== Noncompliant code example

[source,jcl,diff-id=1,diff-type=noncompliant]
----
//STEP1     EXEC PGM=SORT
//SYSOUT   DD   SYSOUT=*
//SORTIN   DD   DSN=A123456.SORTIN,DISP=SHR
//SORTOUT  DD   DSN=A123456.SORTOUT,DISP=OLD
  SORT FIELDS=(1,12,CH,A)
----

==== Compliant solution

[source,jcl,diff-id=1,diff-type=compliant]
----
//STEP1     EXEC PGM=SORT
//SYSOUT   DD   SYSOUT=*
//SORTIN   DD   DSN=A123456.SORTIN,DISP=SHR
//SORTOUT  DD   DSN=A123456.SORTOUT,DISP=OLD
//SYSIN    DD   *
  SORT FIELDS=(1,12,CH,A)
/*
----

== Resources

=== Documentation

* https://www.ibm.com/docs/en/zos/3.1.0?topic=parameter-location-in-jcl[IBM reference - DD * location in the JCL]
