<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>WordPress has a database repair and optimization mode that can be activated by setting <code>WP_ALLOW_REPAIR</code> to <code>true</code> in the configuration.</p>
</div>
<div class="paragraph">
<p>If activated, the repair page can be accessed by any user, authenticated or not. This makes sense because if the database is corrupted, the authentication mechanism might not work.</p>
</div>
<div class="paragraph">
<p>Malicious users could trigger this potentially costly operation repeatadly slowing down the website, and making it unavailable.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>It&#8217;s recommended to enable automatic database repair mode only in case of database corruption. This feature should be deactivated again when the database issue is resolved.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code Examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant Code Example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-php" data-lang="php">define( 'WP_ALLOW_REPAIR', true ); // Noncompliant</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant Solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-php" data-lang="php">// The default value is false, so the value does not have to be expilicitly set.
define( 'WP_ALLOW_REPAIR', false );</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p>OWASP - <a href="https://owasp.org/Top10/A05_2021-Security_Misconfiguration/">Top 10 2021 Category A5 - Security Misconfiguration</a></p>
</li>
<li>
<p>OWASP - <a href="https://owasp.org/Top10/A07_2021-Identification_and_Authentication_Failures/">Top 10 2021 Category A7 - Identification and Authentication Failures</a></p>
</li>
<li>
<p><a href="https://wordpress.org/support/article/editing-wp-config-php/#automatic-database-optimizing">wordpress.org</a> - Automatic Database Optimizing</p>
</li>
<li>
<p>OWASP - <a href="https://owasp.org/www-project-top-ten/2017/A6_2017-Security_Misconfiguration">Top 10 2017 Category A6 - Security Misconfiguration</a></p>
</li>
</ul>
</div>
</div>
</div>