include::../description.adoc[]

== Noncompliant Code Example

With default provided regular expression ``++^[A-Z][a-zA-Z0-9]*$++``:

[source,flex]
----
public class myClass {...}
----

== Compliant Solution

[source,flex]
----
public class MyClass {...}
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::parameters.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
