== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

With default provided regular expression ``++^[A-Z][a-zA-Z0-9]*$++``:

[source,flex]
----
public class myClass {...}
----

=== Compliant solution

[source,flex]
----
public class MyClass {...}
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

=== Parameters

.format
****

----
^[A-Z][a-zA-Z0-9]*$
----

Regular expression used to check the class names against.
****


'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
