include::../description.adoc[]

== Noncompliant Code Example

With default provided regular expression ``++^[A-Z][a-zA-Z0-9]*$++``:

----
public class myClass {...}
----

== Compliant Solution

----
public class MyClass {...}
----

ifdef::env-github,rspecator-view[]
'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
