<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Unlike class and instance methods, static methods don&#8217;t receive an implicit first argument. Nonetheless naming the first argument <code>self</code> or <code>cls</code> guarantees confusion - either on the part of the original author, who may never understand why the arguments don&#8217;t hold the values he expected, or on that of future maintainers.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">class MyClass:
  @staticmethod
  def s_meth(self, arg1, arg2):  #Noncompliant
    # ...</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">class MyClass:
  @staticmethod
  def s_meth(arg1, arg2):  #Noncompliant
    # ...</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove the "xxx" argument from this method definition.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_is_duplicated_by_s2851">is duplicated by: <a data-rspec-id="S2851" class="rspec-auto-link">S2851</a></h3>

</div>
<div class="sect2">
<h3 id="_is_duplicated_by_s2852">is duplicated by: <a data-rspec-id="S2852" class="rspec-auto-link">S2852</a></h3>

</div>
<div class="sect2">
<h3 id="_on_24_apr_2015_090419_elena_vilchik_wrote">on 24 Apr 2015, 09:04:19 Elena Vilchik wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] I would replace <code>bug</code> tag with <code>confusing</code>. Or at least add it. WDYT?</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_28_apr_2015_153624_ann_campbell_wrote">on 28 Apr 2015, 15:36:24 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~elena.vilchik] I&#8217;ve downgraded this from bug to pitfall (meaning it&#8217;s not broken now but you&#8217;ve left a trap for the next guy)</p>
</div>
</div>
</div>
</div>