There is no good reason to declare a field "public" and "static" without also declaring it "final". Most of the time this is a kludge to share a state among several objects. But with this approach, any object can do whatever it wants with the shared state, such as setting it to ``++null++``. 

== Noncompliant Code Example

----
public class Greeter {
  public static Foo foo = new Foo();
  ...
}
----

== Compliant Solution

----
public class Greeter {
  public static final Foo FOO = new Foo();
  ...
}
----

include::../see.adoc[]
