<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Code snippets are useful for adding code examples to javadoc comments. This will help developers to better understand the usage of documented method. However, large code snippets might be confusing and can decrease the readability. So the general rule is to keep snippets as small as possible.</p>
</div>
<div class="paragraph">
<p>Sometimes there are cases where this is not possible to have a small snippet. For these cases we recommend to extract the snippet into a separate file and make it either an external or a hybrid one.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">/**
 * The code prints "Hello, World!"
 * {@snippet :                                        // Noncompliant
 *  public static void main(String... args) {
 *      System.out.println("Hello, World!0");
 *      System.out.println("Hello, World!1");
 *      System.out.println("Hello, World!2");
 *      System.out.println("Hello, World!3");
 *      System.out.println("Hello, World!4");
 *      System.out.println("Hello, World!5");
 *      System.out.println("Hello, World!6");
 *      System.out.println("Hello, World!7");
 *      System.out.println("Hello, World!8");
 *      System.out.println("Hello, World!9");
 *      System.out.println("Hello, World!10");
 *      System.out.println("Hello, World!11");
 *      System.out.println("Hello, World!12");
 *      System.out.println("Hello, World!13");
 *      System.out.println("Hello, World!14");
 *      System.out.println("Hello, World!15");
 *      System.out.println("Hello, World!16");
 *      System.out.println("Hello, World!17");
 *      System.out.println("Hello, World!18");
 *      System.out.println("Hello, World!19");
 *      System.out.println("Hello, World!20");
 *  }
 * }
 */</code></pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">/**
 * The code prints "Hello, World!"
 * {@snippet :
 *  public static void main(String... args) {
 *      System.out.println("Hello, World!");
 *  }
 * }
 */</code></pre>
</div>
</div>
<div class="paragraph">
<p>Or</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">{@snippet file=ExternalSnippet.java }</code></pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_see">See</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Oracleâ€™s Programmer&#8217;s Guide to Snippets: <a href="https://docs.oracle.com/en/java/javase/18/code-snippet/index.html">https://docs.oracle.com/en/java/javase/18/code-snippet/index.html</a></p>
</div>
<hr>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Reduce this code snippet to have less than ${maximumSnippetLength} lines or extract it to a separate file to make it external/hybrid.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="ulist">
<ul>
<li>
<p>Primary: @snippet tag content</p>
</li>
</ul>
</div>
</div>
<div class="sect2">
<h3 id="_parameters">Parameters</h3>
<div class="sidebarblock">
<div class="content">
<div class="title">maximumSnippetLength</div>
<div class="listingblock">
<div class="content">
<pre>20</pre>
</div>
</div>
<div class="paragraph">
<p>The maximum authorized number of lines of a single code snippet.</p>
</div>
</div>
</div>
</div>
</div>
</div>