== Why is this an issue?

Even if all browsers are fault-tolerant, HTML tags should be closed to prevent any unexpected behavior.


=== Noncompliant code example

[source,html]
----
<html>
  <head>
    <title>Test Page    <!-- Noncompliant; title not closed -->
  <!-- Noncompliant; head not closed -->
  <body>
    <em>Emphasized Text  <!-- Noncompliant; em not closed -->
  <!-- Noncompliant; body not closed -->
</html>  
----


=== Compliant solution

[source,html]
----
<html>
  <head>
    <title>Test Page</title>
  </head>
  <body>
    <em>Emphasized Text</em>
  </body>
</html>  
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

The tag "xxx" has no corresponding closing tag.


=== Parameters

.ignoreTags
****

----
HTML,HEAD,BODY,P,DT,DD,LI,OPTION,THEAD,TH,TBODY,TR,TD,TFOOT,COLGROUP,IMG,INPUT,BR,HR,FRAME,AREA,BASE,BASEFONT,COL,ISINDEX,LINK,META,PARAM
----

Ignore tags
****


'''
== Comments And Links
(visible only on this page)

=== on 18 Nov 2013, 10:19:10 Freddy Mallet wrote:
Is implemented by \http://jira.codehaus.org/browse/SONARPLUGINS-3043

endif::env-github,rspecator-view[]
