<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>If a <code>RuntimeException</code> is thrown while code is unit tested, it&#8217;s likely to be thrown again in production. Therefore, test methods should be allowed to <code>throw RuntimeException</code>s so you&#8217;re alerted early to problems in the code.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-text" data-lang="text">@Test
public void testTheThing() {
  try {
    // ...
  } catch (RuntimeException e) {  // Noncompliant
    // ...
  }
}</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_11_dec_2015_204742_ann_campbell_wrote">on 11 Dec 2015, 20:47:42 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>fyi [~tamas.vajk], I&#8217;ve targeted this to C# too.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_19_nov_2020_140438_nicolas_harraudeau_wrote">on 19 Nov 2020, 14:04:38 Nicolas Harraudeau wrote:</h3>
<div class="paragraph">
<p>It&#8217;s not clear what this rule tries to detect. It could be either detecting when RuntimeExceptions are ignored, or detecting when a more specific exception type should be caught. Rule <a data-rspec-id="S2486" class="rspec-auto-link">RSPEC-2486</a> covers the first case, and rule <a data-rspec-id="S5958" class="rspec-auto-link">RSPEC-5958</a> covers the second case. Thus we don&#8217;t seem to need <a data-rspec-id="S3477" class="rspec-auto-link">RSPEC-3477</a>.</p>
</div>
</div>
</div>
</div>