This rule raises an issue when the ``++pandas.DataFrame.groupby++`` method is not followed by an aggregation method call.

== Why is this an issue?

A ``++groupby++`` operation involves some combination of splitting the data frame, applying a function, and combining the results. This can be used to group large amounts of data and compute operations on these groups.

Even though the ``++DataFrame.groupby++`` method call instantiates the ``++GroupBy++`` object it only verifies that a valid mapping is passed - no real data splitting occurs until it is needed.

To perform the data transformation, an aggregation method should be called from the ``++GroupBy++`` object, e.g. ``++apply++`` or ``++sum++``.

=== Code examples

==== Noncompliant code example

[source,python,diff-id=1,diff-type=noncompliant]
----
import pandas as pd

df = pd.DataFrame({
        'X': ['A', 'B', 'A', 'C'],
        'Y': [10, 7, 12, 5]
    })


df.groupby('X') # Noncompliant: groupby itself does nothing
----

==== Compliant solution

[source,python,diff-id=1,diff-type=compliant]
----
import pandas as pd

df = pd.DataFrame({
        'X': ['A', 'B', 'A', 'C'],
        'Y': [10, 7, 12, 5]
    })


df.groupby('X').sum() # Compliant
----


== Resources
=== Documentation
* Pandas Documentation - https://pandas.pydata.org/pandas-docs/stable/user_guide/groupby.html#splitting-an-object-into-groups[Group by: split-apply-combine]
* Pandas Documentation - https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.groupby.html#pandas.DataFrame.groupby[DataFrame.groupby method reference]
* Pandas Documentation - https://pandas.pydata.org/docs/reference/groupby.html[GroupBy reference]
