== Why is this an issue?

``++sizeof()++`` is an alias of ``++count()++`` so the two offer the same results, but ``++sizeof()++`` has a different meaning in other languages and may confuse developers who are new to PHP.


=== Noncompliant code example

[source,php]
----
$fruit=("apple","pear","raspberry","gooseberry","red currant");
echo sizeof($fruit);  // Noncompliant
----


=== Compliant solution

[source,php]
----
$fruit=("apple","pear","raspberry","gooseberry","red currant");
echo count($fruit);
----



ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Use "count()" instead.


'''
== Comments And Links
(visible only on this page)

=== on 20 Jan 2015, 07:59:55 Ann Campbell wrote:
Source: \https://twitter.com/declaassen/status/555301362846937088

=== on 19 May 2015, 15:57:10 Linda Martin wrote:
OK!

endif::env-github,rspecator-view[]
