<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p><code>ID</code> and <code>IDREF</code> are constructs inherited from DTDs, and they suffer from painful drawbacks, mainly:</p>
</div>
<div class="ulist">
<ul>
<li>
<p><code>ID</code>s are too widely scoped: an <code>ID</code> must be unique in the XML document. This means that if an XML document contains <code>&lt;Person&gt;</code> elements and <code>&lt;Dog&gt;</code> elements, both with a "name" attribute of <code>type="xs:ID"</code>, then two people can&#8217;t have the same name. Neither can two dogs, or a person and a dog.</p>
</li>
<li>
<p>IDs must comply with tight naming scheme. For example, "123" is not a valid value for an ID.</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>Instead, <code>KEY</code> and <code>KEYREF</code> should be used. And if you really need to insure uniqueness, use <code>xs:unique</code> to add a uniqueness constraint.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-xml" data-lang="xml">ToDo</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-xml" data-lang="xml">ToDo</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Use "(KEY|KEYREF)" instead.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p><code>ID=</code>|<code>IDREF=</code></p>
</div>
</div>
</div>
</div>