== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,php]
----
$value = $_GET["value"];
header("X-Header: $value"); // Noncompliant
----

=== Compliant solution

[source,php]
----
$value = $_GET["value"];
if (ctype_alnum($value)) {
  header("X-Header: $value"); // Compliant
} else {
  // Error
}
----

include::../see.adoc[]
ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../highlighting.adoc[]

endif::env-github,rspecator-view[]
