<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when S3 operations are performed without verifying bucket ownership using the <code>ExpectedBucketOwner</code> parameter.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>When working with S3 buckets in AWS applications, it&#8217;s essential to verify that you&#8217;re accessing the correct bucket owned by the expected AWS account.
Without proper bucket owner verification, applications may inadvertently interact with unintended S3 buckets due to configuration errors, naming conflicts, or security misconfigurations.
This is particularly critical in multi-account AWS environments where bucket names might be similar across different accounts, or when bucket names are dynamically constructed based on configuration values.
The <code>ExpectedBucketOwner</code> parameter provides a safety mechanism that ensures operations only proceed when the bucket is owned by the specified AWS account ID.</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="paragraph">
<p>Failing to verify bucket ownership exposes systems to security threats. Applications may process data in unintended locations, including test environments or malicious buckets.</p>
</div>
<div class="paragraph">
<p>Data integrity suffers when operations target wrong buckets. Sensitive information could be exposed or corrupted without proper verification.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it_in_boto3">How to fix it in Boto3</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Add the <code>ExpectedBucketOwner</code> parameter to your S3 operations to verify the bucket owner before performing any operations. This parameter should contain the AWS account ID that owns the bucket you expect to access.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">import boto3

s3_client = boto3.client('s3')

def lambda_handler(event, context):
    bucket_name = 'my-production-bucket'

    response = s3_client.get_object(  # Noncompliant
        Bucket=bucket_name,
        Key='data.json'
    )</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">import boto3

s3_client = boto3.client('s3')

def lambda_handler(event, context):
    bucket_name = 'my-production-bucket'
    expected_owner = '123456789012'

    response = s3_client.get_object(
        Bucket=bucket_name,
        Key='data.json',
        ExpectedBucketOwner=expected_owner  # Compliant
    )</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it_in_aiobotocore">How to fix it in aiobotocore</h2>
<div class="sectionbody">
<div class="paragraph">
<p>When using aiobotocore, include the <code>ExpectedBucketOwner</code> parameter in your S3 operations to ensure bucket ownership verification.</p>
</div>
<div class="sect2">
<h3 id="_code_examples_2">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example_2">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">import aiobotocore.session


async def lambda_handler(event, context):
    async with session.create_client('s3') as s3_client:
        bucket_name = 'my-production-bucket'

        response = await s3_client.get_object(  # Noncompliant
            Bucket=bucket_name,
            Key='data.json'
        )</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution_2">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">import aiobotocore.session

session = aiobotocore.session.get_session()

async def lambda_handler(event, context):
    async with session.create_client('s3') as s3_client:
        bucket_name = 'my-production-bucket'
        expected_owner = '123456789012'

        response = await s3_client.get_object(
            Bucket=bucket_name,
            Key='data.json',
            ExpectedBucketOwner=expected_owner  # Compliant
        )</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>boto3 Documentation -  <a href="https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/s3.html#S3.Client.get_object">get_object</a></p>
</li>
<li>
<p>AWS Documentation - <a href="https://docs.aws.amazon.com/AmazonS3/latest/userguide/bucket-owner-condition.html">Verifying bucket ownership with bucket owner condition</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Add &#8216;ExpectedBucketOwner&#8217; parameter to verify S3 bucket ownership.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="ulist">
<ul>
<li>
<p>Primary location: the S3 client method call without <code>ExpectedBucketOwner</code> parameter</p>
</li>
</ul>
</div>
</div>
</div>
</div>