include::../description.adoc[]

== Noncompliant Code Example

----
if ( $a == $a ) { // always true
  doZ();
}
if ( $a != $a ) { // always false
  doY();
}
if ( $a == $b && $a == $b ) { // if the first one is true, the second one is too
  doX();
}
if ( $a == $b || $a == $b ) { // if the first one is true, the second one is too
  doW();
}

$j = 5 / 5; //always 1
$k = 5 - 5; //always 0
----

== Exceptions

Left-shifting 1 onto 1 is common in the construction of bit masks, and is ignored.
----
$i = 1 << 1; // Compliant
$j = $a << $a; // Noncompliant
----

include::../see.adoc[]
