== Why is this an issue?

include::description.adoc[]

=== Noncompliant code example

[source,text]
----
Module Module1
    Public Property Foo() As Integer
        Get
            Throw New Exception  ' Non-Compliant
        End Get
        Set(ByVal value As Integer)
            ' ... some code ...
        End Set
    End Property
End Module
----


=== Compliant solution

[source,text]
----
Module Module1
    Sub SetFoo(ByVal value As Integer)         ' Compliant
        ' ... some code ...
    End Sub
End Module
----


include::exceptions.adoc[]
