This rule raises an issue when arrays are converted to delimited strings using manual concatenation in loops or the `*` operator instead of the `join` method.

== Why is this an issue?

Manual string concatenation for creating delimited lists is problematic for several reasons.

When you concatenate strings in a loop with delimiters, you often end up with trailing separators that need to be manually removed. This creates additional complexity and potential for bugs:

[source,ruby]
----
output = ''
users.each { |user| output += user.name + ", " }
output # Results in "Alice, Bob, Charlie, " - unwanted trailing comma
----

The `*` operator, while functional, is less explicit about intent. When reading `array * ","`, it's not immediately clear that the goal is to create a comma-separated string. The operation looks more like mathematical multiplication than string joining.

Ruby's `Array#join` method was specifically designed for this purpose. It handles edge cases gracefully, never produces trailing delimiters, and clearly expresses the intent to combine array elements into a single string with separators.

Using `join` also provides better performance for large arrays since it doesn't create intermediate string objects during concatenation, unlike the manual approach which creates a new string object for each concatenation operation.

=== What is the potential impact?

Using manual concatenation or the `*` operator instead of `join` can lead to:

* Bugs from trailing delimiters that require manual removal
* Reduced code readability and maintainability
* Performance issues with large arrays due to repeated string object creation
* Code that doesn't follow Ruby idioms and conventions

== How to fix it

Replace manual string concatenation in loops with `map` and `join` to avoid trailing delimiters and improve readability.

=== Code examples

==== Noncompliant code example

[source,ruby,diff-id=1,diff-type=noncompliant]
----
output = ''
users.each { |user| output += user.name + ", " } # Noncompliant
output # Results in unwanted trailing ", "
----

==== Compliant solution

[source,ruby,diff-id=1,diff-type=compliant]
----
# Use map and join for clean delimiter separation
users.map(&:name).join(", ")
----

== Resources

=== Documentation

 * Array#join - Ruby Documentation - https://ruby-doc.org/core/Array.html#method-i-join[Official Ruby documentation for the Array#join method]

 * Ruby Style Guide - String Concatenation - https://rubystyle.guide/#array-join[Community Ruby style guide recommendations for array joining]

=== Standards

 * Ruby Style Guide - Array#join - https://rubystyle.guide/#array-join[Use Array#join over Array#* for joining array elements into strings]

=== Related rules

 * RSPEC-1643 - https://rules.sonarsource.com/java/RSPEC-1643/[Strings should not be concatenated using '+' in a loop (Java)]
