<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when a function call result is used, even though the function does not return anything.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>If a function does not return anything, it makes no sense to use its output. Specifically, passing it to another function, or assigning its "result" to a variable is probably a bug because such functions return nothing, which is probably not what was intended.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-php" data-lang="php">$result = closedir($dir_handle); // Noncompliant, "closedir" does not return anything.</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-php" data-lang="php">closedir($dir_handle);</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>PHP documentation - <a href="https://www.php.net/manual/en/functions.returning-values.php">Returning values</a></p>
</li>
</ul>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove this use of the output from "xxx"; "xxx" doesn&#8217;t return anything.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="ulist">
<ul>
<li>
<p>Primary:</p>
<div class="ulist">
<ul>
<li>
<p><code>x = noReturn(...)</code></p>
</li>
<li>
<p><code>doSomething(noReturn(...))</code></p>
</li>
</ul>
</div>
</li>
<li>
<p>Secondary: function definition</p>
<div class="ulist">
<ul>
<li>
<p>'Function definition.'</p>
</li>
</ul>
</div>
</li>
</ul>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_22_aug_2016_112449_pierre_yves_nicolas_wrote">on 22 Aug 2016, 11:24:49 Pierre-Yves Nicolas wrote:</h3>
<div class="paragraph">
<p>What about using the result of the function for something else?</p>
</div>
<div class="listingblock">
<div class="content">
<pre>function foo() {}
bar(foo());
foo() + 42;
foo().bar(); // that one would throw a TypeError and should be caught with RSPEC-2259</pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_on_22_aug_2016_151043_ann_campbell_wrote">on 22 Aug 2016, 15:10:43 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~pierre-yves.nicolas] I&#8217;ve genericized the rule (altho not the code samples). See what you think.</p>
</div>
<div class="paragraph">
<p>cc [~yves.duboispelerin]</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_22_aug_2016_154744_pierre_yves_nicolas_wrote">on 22 Aug 2016, 15:47:44 Pierre-Yves Nicolas wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] Looks good to me!</p>
</div>
</div>
</div>
</div>