== Why is this an issue?

Excessive logging within a code block can lead to several problems:

* *Log file overload*: generating an overwhelming number of log entries can fill up disk space quickly (thus increasing the storage space cost) and make it challenging to identify important log events promptly.
* *Performance degradation*: writing a large number of log statements can impact the performance of an application, especially when the logs are placed in frequently executed paths.
* *Code readability and maintainability*: excessive logging can clutter the code and increase the code's complexity, making it difficult for developers to identify essential logic.

As code block is considered any of the following:

* `Try`, `Catch` and `Finally` blocks
* `if` and `else` blocks
* `for`, `while`, `do-while`, `foreach` loops blocks
* method body and lambda expression blocks
* property getter and setter blocks

Only the logging statements that are directly within the first level of the aforementioned code blocks will be counted, and any logging statements within nested blocks will not be included in the same count. For example consider the snippet below:

include::{language}/example.adoc[]

The rule will count 2 logging statements that are within the method block (namely `logger.Debug("The operation started")` and `logger.Debug("The operation ended")`). Any statements within nested blocks, such as the `foreach` block will be counted separately.

It is important to note that the rule does not consider the logging level set on the specific logging framework configuration.
The most popular logging frameworks are supported: 

* Nuget package - https://www.nuget.org/packages/Microsoft.Extensions.Logging[Microsoft.Extensions.Logging]
* Nuget package - https://www.nuget.org/packages/Serilog[Serilog]
* Nuget package - https://www.nuget.org/packages/Castle.Core[Castle.Core]
* Nuget package - https://www.nuget.org/packages/NLog[NLog]
* Nuget package - https://www.nuget.org/packages/log4net[log4net]

== How to fix it

Reduce the number of logging calls within the code block by identifying and selecting essential log statements with relevant information, necessary for understanding the flow of execution or diagnosing issues.

=== Code examples

==== Noncompliant code example

With the default max parameter value 2:

include::{language}/noncompliant.adoc[]

==== Compliant solution

With the default max parameter value 2:

include::{language}/compliant.adoc[]

== Resources

=== Documentation

* Microsoft Learn - https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/statements/exception-handling-statements[Exception-handling statements]

=== Articles & blog posts

* https://rolf-engelhard.de/2013/04/logging-anti-patterns-part-ii/[Rolf Engelhard - Logging anti-patterns]

include::rspecator.adoc[]