== Why is this an issue?

Excessive logging within a code block can lead to several problems:

* *Log file overload*: generating an overwhelming number of log entries can fill up disk space quickly (thus increasing the storage space cost) and make it challenging to identify important log events promptly.
* *Performance degradation*: writing a large number of log statements can impact the performance of an application, especially when the logs are placed in frequently executed paths.
* *Code readability and maintainability*: excessive logging can clutter the code and increase the code's complexity, making it difficult for developers to identify essential logic.

The rule supports the most popular logging frameworks: 

* https://www.nuget.org/packages/Microsoft.Extensions.Logging[Microsoft.Extensions.Logging]
* https://www.nuget.org/packages/Serilog[Serilog]
* https://www.nuget.org/packages/Castle.Core[Castle.Core]
* https://www.nuget.org/packages/NLog[NLog]
* https://www.nuget.org/packages/log4net[log4net]

== How to fix it

Reduce the number of logging calls within the code block by identifying and selecting essential leg statements with relevant information, necessary for understanding the flow of execution or diagnosing issues.

=== Code examples

==== Noncompliant code example

With the default max parameter value 4:
include::{language}/noncompliant.adoc[]

==== Compliant solution

With the default max parameter value 4:
include::{language}/compliant.adoc[]

== Resources

=== Documentation

* Microsoft Learn - https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/statements/exception-handling-statements[Exception-handling statements]

=== Articles & blog posts

* https://rolf-engelhard.de/2013/04/logging-anti-patterns-part-ii/[Rolf Engelhard - Logging anti-patterns]

include::rspecator.adoc[]