include::../../../shared_content/secrets/description.adoc[]

== Why is this an issue?

include::../../../shared_content/secrets/rationale.adoc[]

=== What is the potential impact?

The consequences vary greatly depending on the situation and the secret-exposed
audience. Still, two main scenarios should be considered.

include::../../../shared_content/secrets/impact/financial_loss.adoc[]

include::../../../shared_content/secrets/impact/security_downgrade.adoc[]

== How to fix it

include::../../../shared_content/secrets/fix/revoke.adoc[]

include::../../../shared_content/secrets/fix/recent_use.adoc[]

include::../../../shared_content/secrets/fix/vault.adoc[]

=== Code examples

==== Noncompliant code example

[source,python,diff-id=1,diff-type=noncompliant,subs="attributes"]
----
uri = "mongodb://foouser:foopass@example.com/testdb"
----

==== Compliant solution

[source,python,diff-id=1,diff-type=compliant,subs="attributes"]
----
import os

user = os.environ["MONGO_USER"]
password = os.environ["MONGO_PASSWORD"]
uri = f"mongodb://{user}:{password}}@example.com/testdb"
----

//=== How does this work?

//=== Pitfalls

//=== Going the extra mile

== Resources

include::../../../shared_content/secrets/resources/standards.adoc[]

//=== Benchmarks
