== How to fix it in Botan

=== Code examples

include::../../common/fix/code-rationale.adoc[]

==== Noncompliant code example

[source,csharp,diff-id=1,diff-type=noncompliant]
----
AesFastEngine aesFast = new AesFastEngine(); // Noncompliant
----

==== Compliant solution

[source,csharp,diff-id=1,diff-type=compliant]
----
var AES = new AESEngine();
----

=== How does this work?

include::../../common/fix/strong-cryptography.adoc[]

