== How to fix it in BouncyCastle

=== Code examples

include::../../common/fix/code-rationale.adoc[]

==== Noncompliant code example

[source,csharp,diff-id=1,diff-type=noncompliant]
----
using Org.BouncyCastle.Crypto.Engines;
using Org.BouncyCastle.Crypto.Parameters;

public void encrypt()
{
    AesFastEngine aesFast = new AesFastEngine(); // Noncompliant
}
----

==== Compliant solution

[source,csharp,diff-id=1,diff-type=compliant]
----
using Org.BouncyCastle.Crypto.Engines;
using Org.BouncyCastle.Crypto.Parameters;

public void encrypt()
{
    var AES = new AESEngine();
}
----

=== How does this work?

include::../../common/fix/strong-cryptography.adoc[]

