== How to fix it in .NET

=== Code examples

include::../../common/fix/code-rationale.adoc[]

==== Noncompliant code example

[source,csharp,diff-id=1,diff-type=noncompliant]
----
using System.Security.Cryptography;

var simpleDES = new DESCryptoServiceProvider(); // Noncompliant
----

==== Compliant solution

[source,csharp,diff-id=1,diff-type=compliant]
----
using System.Security.Cryptography;

var AES = new AesCryptoServiceProvider();
----

=== How does this work?

include::../../common/fix/strong-cryptography.adoc[]

