== Why is this an issue?

Debug statements are always useful during development. But include them in production code - particularly in code that runs client-side - and you run the risk of inadvertently exposing sensitive information, slowing down the browser, or even erroring-out the site for some users.

=== Noncompliant code example

[source,javascript]
----
console.log(password_entered); // Noncompliant
----

include::../see.adoc[]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

=== on 2 Jun 2015, 15:59:21 Linda Martin wrote:
OK!

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
