include::../description.adoc[]

== Noncompliant Code Example

----
if (condition1) {
  if (condition2) {             // NonCompliant
    /* ... */
  }
}
----

== Compliant Solution

----
if (condition1 && condition2) { // Compliant
  /* ... */
}
----

ifdef::env-github,rspecator-view[]
'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
