== Why is this an issue?

"A rose by any other name would smell as sweet," but ``++main++`` by any other name would not. Just because a method has the name "main", that doesn't make it the entry point to an application. It must also have the correct signature. Specifically, it must be ``++public static void++`` and accept a single ``++String []++`` as an argument.


=== Noncompliant code example

[source,java]
----
public void main(String arg) {  // Noncompliant
  // ...
}
----


=== Compliant solution

[source,java]
----
public static void main(String [] args) {
  // ...
}
----



ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Make this method ["public" | "static" | return "void" | accept a "String []" argument].


'''
== Comments And Links
(visible only on this page)

=== relates to: S998

=== on 11 Jun 2015, 20:19:01 Ann Campbell wrote:
Code Pro: Entry point method

=== on 16 Jun 2015, 13:18:33 Nicolas Peru wrote:
Looks good. I really like this first sentence ! :)

endif::env-github,rspecator-view[]
