== Why is this an issue?

Each element should be on a line to itself.


=== Noncompliant code example

[source,text]
----
<parent><child /></parent>  <!-- Noncompliant -->
----


=== Compliant solution

[source,text]
----
<parent>
  <child />
</parent>
----

