<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rules raises an issue when <code>trio.sleep()</code> or <code>anyio.sleep()</code> is called with a duration greater than 24 hours.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>When using <code>trio.sleep()</code> or <code>anyio.sleep()</code> with very long intervals (greater than 24 hours), the intent is usually to wait indefinitely or for an extremely long period. Both libraries provide dedicated methods specifically designed for this purpose: <code>trio.sleep_forever()</code> and <code>anyio.sleep_forever()</code>.</p>
</div>
<div class="paragraph">
<p>Using explicit sleep durations greater than 24 hours has several drawbacks:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>It obscures the developer&#8217;s intent. A very large number like <code>86400 * 365</code> doesn&#8217;t clearly communicate that the code intends to wait indefinitely.</p>
</li>
<li>
<p>It makes code less maintainable, as other developers need to calculate what the large numbers actually represent.</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>In summary, using <code>sleep_forever()</code> is preferable when the intent is to sleep indefinitely as it clearly conveys this purpose, avoiding maintainability issues caused by using arbitrarily large sleep durations.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it_in_trio">How to fix it in Trio</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Replace calls to <code>trio.sleep()</code> that use intervals greater than 24 hours with calls to <code>trio.sleep_forever()</code>.</p>
</div>
<div class="paragraph">
<p>If the intention is truly to wait for a specific long duration rather than indefinitely, consider expressing that intent more clearly by using named variables or constants.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">import trio

async def long_wait():
    await trio.sleep(86400 * 365)  # Noncompliant</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">import trio

async def long_wait():
    await trio.sleep_forever()</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it_in_anyio">How to fix it in AnyIO</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Replace calls to <code>anyio.sleep()</code> that use intervals greater than 24 hours with calls to <code>anyio.sleep_forever()</code>.</p>
</div>
<div class="paragraph">
<p>If the intention is truly to wait for a specific long duration rather than indefinitely, consider expressing that intent more clearly by using named variables or constants.</p>
</div>
<div class="sect2">
<h3 id="_code_examples_2">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example_2">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">import anyio

async def long_wait():
    await anyio.sleep(86400 * 30)  # Noncompliant</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution_2">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">import anyio

async def long_wait():
    await anyio.sleep_forever()</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Trio - <a href="https://trio.readthedocs.io/en/stable/reference-core.html#trio.sleep_forever">sleep_forever() documentation</a></p>
</li>
<li>
<p>AnyIO - <a href="https://anyio.readthedocs.io/en/stable/api.html#anyio.sleep_forever">sleep_forever() documentation</a></p>
</li>
<li>
<p>Python asyncio - <a href="https://docs.python.org/3/library/asyncio-task.html">Tasks and coroutines</a></p>
</li>
</ul>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Replace this long sleep with sleep_forever()</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>The entire function call expression:
* Primary locations: the <code>trio.sleep()</code> or <code>anyio.sleep()</code> call
* Secondary locations: none</p>
</div>
</div>
</div>
</div>