== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

With a threshold of 5:

[source,php]
----
switch ($var) {
  case 0:  // 6 lines till next case
    methodCall1();
    methodCall2();
    methodCall3();
    methodCall4();
    break;
  default:
    break;
}
----

=== Compliant solution

[source,php]
----
switch ($var) {
  case 0:
    doSomething();
    break;
  default:
    break;  
}

function doSomething(){
  methodCall1("");
  methodCall2("");
  methodCall3("");
  methodCall4("");
}
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

=== Parameters

.max
****
_Integer_

----
10
----

Maximum number of lines of code
****


'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
