<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Declaring a member function static or const limits its access to the non-static data members.</p>
</div>
<div class="paragraph">
<p>This helps to prevent unintentional modification of the data, and facilitates compliance with MISRA C&#43;&#43; 2008 Rule 7–1–1 (A variable which is not modified shall be const qualified).</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">class A
{
public:
  int16_t f1 ( ) // Non-compliant – can be const
  {
    return m_i;
  }
  int16_t f2 ( ) // Non-compliant – can be static
  {
    return m_s;
  }
private:
  int16_t m_i;
  static int16_t m_s;
};</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">class A
{
public:
  int16_t const f1 ( ) // Compliant
  {
    return m_i;
  }
  static int16_t f2 ( ) // Compliant
  {
    return m_s;
  }
  int16_t f3 ( ) // Compliant – cannot be const or static
  {
    return ++m_i;
  }
private:
  int16_t m_i;
  static int16_t m_s;
};</code></pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p>MISRA C&#43;&#43;:2008, 9-3-3</p>
</li>
</ul>
</div>
<hr>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Make this member function ["static"||"const"].</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_is_duplicated_by_s1252">is duplicated by: <a data-rspec-id="S1252" class="rspec-auto-link">S1252</a></h3>

</div>
<div class="sect2">
<h3 id="_on_15_oct_2014_204127_ann_campbell_wrote">on 15 Oct 2014, 20:41:27 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~samuel.mercier] please:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>fill in the appropriate reference field(s)</p>
</li>
<li>
<p>provide a See section</p>
</li>
<li>
<p>make the message a sentence (initial capital letter, terminating period)</p>
</li>
</ul>
</div>
</div>
</div>
</div>