== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,abap]
----
IF A = 0. " Noncompliant
ELSEIF A > 1. " Noncompliant 
ELSE. " Noncompliant
ENDIF.
----

include::../exceptions.adoc[]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
