== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,go]
----
func doNothing() { // Noncompliant
}
----

=== Compliant solution

[source,go]
----
func doNothing() {
  // Do nothing because of X and Y.
}
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Add a nested comment explaining why this function is empty or complete the implementation.


'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
