Modulo by 1 operation always returns 0, it is likely a typo.


== Noncompliant Code Example

----
void f(int value1, int value2) {
  if (fun() % 1 == 1) { // Noncompliant
    // ...
  }
}
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

endif::env-github,rspecator-view[]
