== Why is this an issue?

Unlike class and instance methods, static methods don't receive an implicit first argument. Nonetheless naming the first argument ``++self++`` or ``++cls++`` guarantees confusion - either on the part of the original author, who may never understand why the arguments don't hold the values he expected, or on that of future maintainers.


=== Noncompliant code example

[source,python]
----
class MyClass:
  @staticmethod
  def s_meth(self, arg1, arg2):  #Noncompliant
    # ...
----


=== Compliant solution

[source,python]
----
class MyClass:
  @staticmethod
  def s_meth(arg1, arg2):  #Noncompliant
    # ...
----



ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
