<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when a class method is called on an instance, or when an instance method is called on a class.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Ruby distinguishes between class methods and instance methods based on how they are defined and called.</p>
</div>
<div class="paragraph">
<p>Class methods are defined using <code>def self.method_name</code>, <code>class &lt;&lt; self</code>, or <code>def ClassName.method_name</code>. These methods belong to the class itself and should be called directly on the class.</p>
</div>
<div class="paragraph">
<p>Instance methods are defined with <code>def method_name</code> inside a class. These methods belong to individual instances of the class and should be called on object instances.</p>
</div>
<div class="paragraph">
<p>When you call a class method on an instance, Ruby cannot find the method because it only exists on the class level. Similarly, calling an instance method on a class will fail because the method requires an instance context.</p>
</div>
<div class="paragraph">
<p>This mismatch results in a <code>NoMethodError</code> at runtime, which can cause application crashes and unexpected behavior.</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="paragraph">
<p>This issue causes runtime errors that crash the application with <code>NoMethodError: undefined method</code> exceptions. These errors can:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>Break application functionality and user workflows</p>
</li>
<li>
<p>Cause service outages if not caught by error handling</p>
</li>
<li>
<p>Lead to poor user experience with unexpected crashes</p>
</li>
<li>
<p>Make debugging more difficult when the error occurs in production</p>
</li>
</ul>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Call the class method on the class itself, not on an instance. Class methods are shared across all instances and should be accessed through the class.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-ruby" data-lang="ruby">class User
  def self.find_active
    where(active: true)
  end
end

user = User.new
active_users = user.find_active  # Noncompliant</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-ruby" data-lang="ruby">class User
  def self.find_active
    where(active: true)
  end
end

active_users = User.find_active</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Ruby Class and Instance Methods - <a href="https://ruby-doc.org/core/doc/syntax/methods_rdoc.html">Official Ruby documentation on method definitions and calling conventions</a></p>
</li>
<li>
<p>Understanding Ruby Class Methods vs Instance Methods - <a href="https://www.rubyguides.com/2018/11/class-vs-instance-methods/">Comprehensive guide explaining the differences between class and instance methods in Ruby</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>