== Why is this an issue?

When allocating memory with ``++malloc++``, ``++calloc++`` and ``++realloc++`` it is important to make sure that the size of the allocated memory is compatible with the receiver type.


=== Noncompliant code example

[source,cpp]
----
void f() {
  ...
  int* ptr1 = malloc(sizeof(short)); // Noncompliant
  ...
}
----


=== Compliant solution

[source,cpp]
----
void f() {
  ...
  int* ptr1 = malloc(sizeof(int));
  ...
}
----

