This rule raises an issue when a method in a Spock specification class uses a traditional camelCase name instead of a descriptive string literal.

== Why is this an issue?

Spock is a behavior-driven development (BDD) testing framework that emphasizes readable specifications. One of its key features is the ability to use string literals as method names for feature methods, which makes tests self-documenting and easier to understand.

When you use traditional camelCase method names like `testAddingElement()`, the test purpose is less clear and requires developers to read the implementation to understand what behavior is being tested. This reduces the documentation value of your test suite.

String literal method names like `"adding an element leads to size increase"()` immediately communicate the expected behavior in natural language. This approach:

* Makes test reports more readable for stakeholders
* Improves test documentation without additional comments
* Follows Spock's intended design and conventions
* Helps teams write better behavioral specifications

Using descriptive string literals transforms your test methods into living documentation that clearly expresses the system's expected behavior.

=== What is the potential impact?

Using traditional method names in Spock tests reduces code readability and documentation value. Test suites become harder to understand, especially for team members who need to quickly grasp what behaviors are being tested. This can slow down development and make maintenance more difficult.

== How to fix it

Replace the camelCase method name with a descriptive string literal that clearly explains what behavior is being tested.

=== Code examples

==== Noncompliant code example

[source,groovy,diff-id=1,diff-type=noncompliant]
----
class StackSpec extends Specification {
    def testAddingElement() { // Noncompliant
        given:
        def stack = new Stack()
        
        when:
        stack.push(42)
        
        then:
        stack.size() == 1
    }
}
----

==== Compliant solution

[source,groovy,diff-id=1,diff-type=compliant]
----
class StackSpec extends Specification {
    def "adding an element leads to size increase"() {
        given:
        def stack = new Stack()
        
        when:
        stack.push(42)
        
        then:
        stack.size() == 1
    }
}
----

== Resources

=== Documentation

 * Groovy Testing Guide - Spock Specifications - http://groovy-lang.org/testing.html#_specifications[Official Groovy documentation explaining Spock specifications and the convention of using string literals for feature method names]

 * Spock Framework Documentation - https://spockframework.org/spock/docs/2.3/spock_primer.html[Official Spock framework documentation with examples of proper feature method naming]
