<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>The following order represents what is commonly expected by ReactJS developers.</p>
</div>
<div class="paragraph">
<p>Not following this convention has no technical impact, but will reduce the class&#8217;s readability because most developers are used to this standard order.</p>
</div>
<div class="olist arabic">
<ol class="arabic">
<li>
<p><code>static</code> methods</p>
</li>
<li>
<p>the <code>constructor</code></p>
</li>
<li>
<p><code>getChildContext()</code></p>
</li>
<li>
<p><code>componentWillMount()</code></p>
</li>
<li>
<p><code>componentDidMount()</code></p>
</li>
<li>
<p><code>componentWillReceiveProps()</code></p>
</li>
<li>
<p><code>shouldComponentUpdate()</code></p>
</li>
<li>
<p><code>componentWillUpdate()</code></p>
</li>
<li>
<p><code>componentDidUpdate()</code></p>
</li>
<li>
<p><code>componentWillUnmount()</code></p>
</li>
<li>
<p>clickHandlers or eventHandlers such as <code>onClickSubmit()</code> or <code>onChangeDescription()</code></p>
</li>
<li>
<p>getter methods for render such as <code>getSelectReason()</code> or <code>getFooterContent()</code></p>
</li>
<li>
<p>optional render methods such as <code>renderNavigation()</code> or <code>renderProfilePicture()</code></p>
</li>
<li>
<p><code>render()</code></p>
</li>
</ol>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-javascript" data-lang="javascript">export default class MyApp extends React.PureComponent {

  render() { // Noncompliant
  }
  ...
  componentDidMount() {
  }
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-javascript" data-lang="javascript">export default class MyApp extends React.PureComponent {

  componentDidMount() {
  }
  ...
  render() {
  }
}</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_7_feb_2018_125352_alexandre_gigleux_wrote">on 7 Feb 2018, 12:53:52 Alexandre Gigleux wrote:</h3>
<div class="paragraph">
<p>Reference: https://hackernoon.com/reactjs-component-lifecycle-methods-a-deep-dive-38275d9d13c0</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_23_feb_2018_114104_alexandre_gigleux_wrote">on 23 Feb 2018, 11:41:04 Alexandre Gigleux wrote:</h3>
<div class="paragraph">
<p>This rule is covered by ESLint for React: https://github.com/yannickcr/eslint-plugin-react/blob/HEAD/docs/rules/sort-comp.md (react/sort-comp)</p>
</div>
</div>
</div>
</div>