include::../description.adoc[]

== Noncompliant Code Example

----
private void DoSomething() 
{
  // TODO
}
----

include::../see.adoc[]

ifdef::env-github,rspecator-view[]
'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
