<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Using conditional subtraction <code>if a &gt; b { b - a } else { 0 }</code> can lead to an unintended underflow, which can cause bugs or unexpected behaviors. The <code>saturating_sub</code> method ensures that the subtraction does not underflow by returning zero if the result would have been negative.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-rust" data-lang="rust">let a = 12u32;
let b = 13u32;
let result = if a &gt; b { b - a } else { 0 }; // Noncompliant: Potential underflow condition.</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-rust" data-lang="rust">let a = 12u32;
let b = 13u32;
let result = a.saturating_sub(b); // Compliant: Safe subtraction using saturating_sub.</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Clippy Lints - <a href="https://rust-lang.github.io/rust-clippy/master/index.html#inverted_saturating_sub" class="bare">https://rust-lang.github.io/rust-clippy/master/index.html#inverted_saturating_sub</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>