== Why is this an issue?

Character classes in regular expressions allow you to define sets of characters for matching. An empty character class will not match any character because it has no characters within the brackets. 

It is likely the regular expression does not work as you intended. It's good practice to avoid using empty character classes in regular expressions, as they do not serve any useful purpose in pattern matching.

=== Noncompliant code example

[source,javascript]
----
/^foo[]/.test("foobar"); // false
----

=== Compliant solution

[source,javascript]
----
/^foo/.test("foobar"); // true
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Rework this empty character class that doesn't match anything.

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
