== Why is this an issue?

include::../description.adoc[]

To match a literal string, rather than a regular expression, either all special characters should be escaped or methods that don't use regular expressions should be used.

=== Noncompliant code example

[source,javascript]
----
new RegExp("([");
str.match("([");
----

=== Compliant solution

[source,javascript]
----
new RegExp("\\(\\[");
str.match("\\(\\[");
str.replace("([", "{");
----

include::../rspecator.adoc[]
