<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>A complete declaration of the structure or union shall be included within any translation unit that refers to that structure. See section 6.1.2.5 of ISO 9899:1990 [2] for a full description of incomplete types.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">struct tnode * pt; // tnode is incomplete</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">struct tnode * pt; // tnode is incomplete at this point
struct tnode
{
  int count;
  struct tnode * left;
  struct tnode * right;
}; // type tnode is now complete</code></pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p>MISRA C:2004, 18.1 - All structure and union types shall be complete at the end of a translation unit.</p>
</li>
</ul>
</div>
<hr>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Complete the declaration of [struct|union] 'xxx'.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_17_oct_2014_145340_ann_campbell_wrote">on 17 Oct 2014, 14:53:40 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~samuel.mercier]</p>
</div>
<div class="ulist">
<ul>
<li>
<p>Message: in my experience, structs are often defined in headers, so it may not be a matter of completing so much as providing, i.e. adding the header include that contains the definition</p>
</li>
<li>
<p>Are you sure this is Maintainability? Can bad things happen if the struct definition is not available in the translation unit?</p>
</li>
</ul>
</div>
</div>
<div class="sect2">
<h3 id="_on_22_oct_2014_150910_samuel_mercier_wrote">on 22 Oct 2014, 15:09:10 Samuel Mercier wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2]</p>
</div>
<div class="ulist">
<ul>
<li>
<p>It is a good practice to define a type only once in a header, but this is not the scope of this rule.</p>
</li>
<li>
<p>Incomplete types (known as opaque types) can exist and can be used as long as you don&#8217;t need to know their size or to access a field. Nothing bad can happen. You will only get compile time error if you try to allocate a space for them or try to access a field, so maintenance is fine for me.</p>
</li>
</ul>
</div>
</div>
</div>
</div>