<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p><code>std::string</code> and <code>std::string_view</code> provide functions to search for specific characters or groups of characters within a string. If the search is successful, these functions will return the position of the first character in the string that matches the search pattern. If the search fails, the functions will return the special value <code>npos</code>.</p>
</div>
<div class="paragraph">
<p><code>npos</code> is converted to <code>true</code> when used in a boolean context. Therefore, using the returned value in a boolean context to determine if the search was successful does not work.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>The returned value of the following functions should be compared to <code>npos</code> when used in a boolean context:</p>
</div>
<div class="ulist">
<ul>
<li>
<p><code>find</code></p>
</li>
<li>
<p><code>rfind</code></p>
</li>
<li>
<p><code>find_first_of</code></p>
</li>
<li>
<p><code>find_last_of</code></p>
</li>
<li>
<p><code>find_first_not_of</code></p>
</li>
<li>
<p><code>find_last_not_of</code></p>
</li>
</ul>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">void f() {
  std::string s = "";
  if (s.find("42")) { // Noncompliant: the condition is true even if "s" does not contain "42"
    // ...
  }
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">void f() {
  std::string s = "";
  if (s.find("42") != std::string::npos) {
    // ...
  }
}</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>C&#43;&#43; reference - <a href="https://en.cppreference.com/w/cpp/string/basic_string">Search functions for std::basic_string</a></p>
</li>
<li>
<p>C&#43;&#43; reference - <a href="https://en.cppreference.com/w/cpp/string/basic_string_view">Search functions for std::basic_string_view</a></p>
</li>
<li>
<p>C&#43;&#43; reference - <a href="https://en.cppreference.com/w/cpp/string/basic_string/npos">std::basic_string&lt;CharT,Traits&gt;::npos</a></p>
</li>
<li>
<p>C&#43;&#43; reference - <a href="https://en.cppreference.com/w/cpp/string/basic_string_view/npos">std::basic_string_view&lt;CharT,Traits&gt;::npos</a></p>
</li>
</ul>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Compare the method result with std::(string|string_view)::npos.</p>
</div>
</div>
</div>
</div>