== Why is this an issue?

If a statement name exceeds the maximum size, it will trigger an error and prevent the program from being executed.

== How to fix it

The maximum size of the name/label of statements cannot exceed 8 characters.
The following statements allow to be prefixed by another 8 characters separated by a dot:

* COMMAND
* CNTL
* DD (for override)
* EXEC
* IF/THEN, ELSE and ENDIF
* INCLUDE
* JCLLIB
* PEND

=== Code examples

==== Noncompliant code example

[source,text,diff-id=1,diff-type=noncompliant]
----
//MYEXPORTLAB     EXPORT SYMLIST=(TEST)
//MYEXECPROG.PROG EXEC PGM=ABC
//EXEC.MYPROGRAM  EXEC PGM=ABC
----

==== Compliant solution

[source,text,diff-id=1,diff-type=compliant]
----
//MYEXPORT      EXPORT SYMLIST=(TEST)
//MYEXECPR.PROG EXEC PGM=ABC
//EXEC.MYPRG    EXEC PGM=ABC
----

== Resources

=== Documentation

* https://www.ibm.com/docs/en/zos/3.1.0?topic=description-name-field[IBM Reference - COMMAND name field]
* https://www.ibm.com/docs/en/zos/3.1.0?topic=description-label-field[IBM Reference - CNTL label field]
* https://www.ibm.com/docs/en/zos/3.1.0?topic=d-label-field[IBM Reference - ENDCNTL label field]
* https://www.ibm.com/docs/en/zos/3.1.0?topic=d-name-field[IBM Reference - DD name field]
* https://www.ibm.com/docs/en/zos/3.1.0?topic=d-name-field-1[IBM Reference - EXEC name field]
* https://www.ibm.com/docs/en/zos/3.1.0?topic=d-label-field-1[IBM Reference - EXPORT label field]
* https://www.ibm.com/docs/en/zos/3.1.0?topic=d-name-field-2[IBM Reference - IF/THEN/ELSE/ENDIF name field]
* https://www.ibm.com/docs/en/zos/3.1.0?topic=d-name-field-3[IBM Reference - INCLUDE name field]
* https://www.ibm.com/docs/en/zos/3.1.0?topic=d-name-field-4[IBM Reference - JCLLIB name field]
* https://www.ibm.com/docs/en/zos/3.1.0?topic=d-name-field-5[IBM Reference - JOB name field]
* https://www.ibm.com/docs/en/zos/3.1.0?topic=statement-label-field[IBM Reference - NOTIFY label field]
* https://www.ibm.com/docs/en/zos/3.1.0?topic=d-name-field-6[IBM Reference - OUTPUT name field]
* https://www.ibm.com/docs/en/zos/3.1.0?topic=d-name-field-8[IBM Reference - PROC name field]
* https://www.ibm.com/docs/en/zos/3.1.0?topic=d-name-field-7[IBM Reference - PEND name field]
* https://www.ibm.com/docs/en/zos/3.1.0?topic=d-name-field-9[IBM Reference - SCHEDULE name field]
* https://www.ibm.com/docs/en/zos/3.1.0?topic=d-name-field-10[IBM Reference - SET name field]
* https://www.ibm.com/docs/en/zos/3.1.0?topic=d-name-field-11[IBM Reference - XMIT name field]
