Functions that return a promise should be marked with the `async` keyword.

// If you want to factorize the description uncomment the following line and create the file.
//include::../description.adoc[]

== Why is this an issue?

JavaScript supports both promises and async/await to handle asynchronous code. These different flavors of performing asynchronous tasks can be confusing.

//=== What is the potential impact?

== How to fix it

Marking functions that return a promise with the `async` keyword serves as documentation that improves code readability.

//== How to fix it in FRAMEWORK NAME

=== Code examples

==== Noncompliant code example

[source,js,diff-id=1,diff-type=noncompliant]
----
function foo(param) {
  return Promise.resolve(param);
}
----

==== Compliant solution

[source,js,diff-id=1,diff-type=compliant]
----
async function foo(param) {
  return Promise.resolve(param);
}
----

//=== How does this work?

//=== Pitfalls

//=== Going the extra mile


//== Resources
//=== Documentation
//=== Articles & blog posts
//=== Conference presentations
//=== Standards
