== How to fix it in a Servlet

=== Code examples

include::../../common/fix/code-rationale.adoc[]

==== Noncompliant code example

[source,kotlin,diff-id=1,diff-type=noncompliant]
----
import javax.xml.parsers.DocumentBuilder;
import javax.xml.parsers.DocumentBuilderFactory;

fun doGet(req: HttpServletRequest, resp: HttpServletResponse) {
  val xml = """
     <user>
       <username>
        ${req.getParameter("username")}
       </username>
       <role>user</role>
     </user>
     """

  val factory = DocumentBuilderFactory.newInstance()
  try {
    val builder = factory.newDocumentBuilder()
    builder.parse(InputSource(StringReader(xml))) // Noncompliant
  } catch (e: ParserConfigurationException) {
    resp.sendError(400)
  } catch (e: SAXException) {
    resp.sendError(400)
  }
}
----

==== Compliant solution

[source,kotlin,diff-id=1,diff-type=compliant]
----
import javax.xml.parsers.DocumentBuilder;
import javax.xml.parsers.DocumentBuilderFactory;
import org.w3c.dom.Document;
import org.w3c.dom.Element;

fun doGet(req: HttpServletRequest, resp: HttpServletResponse) {
  val factory = DocumentBuilderFactory.newInstance()
  try {
    val builder = factory.newDocumentBuilder()
    val doc: Document = builder.newDocument()
    val user: Element = doc.createElement("user")
    doc.appendChild(user)

    val usernameElement: Element = doc.createElement("username")
    user.appendChild(usernameElement)
    usernameElement.textContent = req.getParameter("username")

    val role: Element = doc.createElement("role")
    user.appendChild(role)
    role.textContent = "user"
  } catch (e: ParserConfigurationException) {
    resp.sendError(400)
  }
}
----

=== How does this work?

include::../../common/fix/introduction.adoc[]

include::../../common/fix/object.adoc[]

The example compliant code takes advantage of the `javax.xml` and `org.w3c.dom`
libraries capabilities to programmatically build XML documents.

include::../../common/fix/casting.adoc[]

