This is an issue when constructor functions return structs containing map fields that are not explicitly initialized with `make()`.

== Why is this an issue?

In Go, maps have a zero value of `nil`. When a struct contains map fields and these maps are not initialized in the constructor function, they remain `nil`.

Attempting to write to a `nil` map causes a runtime panic with the message "assignment to entry in nil map". This can crash your application unexpectedly, especially when the map assignment happens later in the code, making the root cause harder to trace.

Constructor functions are meant to return fully usable instances of structs. When map fields are left uninitialized, the returned struct is in an incomplete state that can lead to runtime failures.

The `make()` function creates an initialized, empty map that is ready for use. By explicitly initializing map fields in constructors, you ensure that the struct is fully functional immediately after creation.

=== What is the potential impact?

The application will crash with a runtime panic when code attempts to write to the uninitialized map. This can lead to service disruptions and poor user experience, especially if the panic occurs in production environments.

== How to fix it

Initialize map fields explicitly using `make()` in the struct literal returned by the constructor function.

=== Code examples

==== Noncompliant code example

[source,go,diff-id=1,diff-type=noncompliant]
----
func NewGraph() *Graph {
    return &Graph{} // Noncompliant: vertices map is nil
}
----

==== Compliant solution

[source,go,diff-id=1,diff-type=compliant]
----
func NewGraph() *Graph {
    return &Graph{
        vertices: make(map[int][]int),
    }
}
----

== Resources

=== Documentation

 * Go Maps - https://go.dev/blog/maps[Official Go blog post explaining maps and their initialization]

 * Effective Go - Maps - https://go.dev/doc/effective_go#maps[Best practices for using maps in Go]
