== Why is this an issue?

There's no need to end a ``++void++`` method with a ``++return++`` statement. Since nothing is actually being returned, the ``++return++`` can be omitted; ``++void++`` will be returned implicitly.


=== Noncompliant code example

[source,text]
----
public void incrementCount() {
  this.count++;
  return;  // Noncompliant
}
----


=== Compliant solution

[source,text]
----
public void incrementCount() {
  this.count++;
}
----


