There's no need to end a ``++void++`` method with a ``++return++`` statement. Since nothing is actually being returned, the ``++return++`` can be omitted; ``++void++`` will be returned implicitly.


== Noncompliant Code Example

[source,text]
----
public void incrementCount() {
  this.count++;
  return;  // Noncompliant
}
----


== Compliant Solution

[source,text]
----
public void incrementCount() {
  this.count++;
}
----


