<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>It is possible to use the increment operator <code>&#43;&#43;</code>, to set the value of a <code>bool</code>(C&#43;&#43;) or <code>_Bool</code>(C) variable to <code>true</code>. But this feature has been deprecated in C&#43;&#43; since the 1998 version of the standard, removed in C&#43;&#43;17, and even where allowed, is simply confusing.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">bool alive;
...
alive++;</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">bool alive;
...
alive = true;</code></pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p>ISO/IEC 14882:1998, 5.3.2</p>
</li>
</ul>
</div>
<hr>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Explicitly set 'XXX' to 'true'.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_20_mar_2015_221213_evgeny_mandrikov_wrote">on 20 Mar 2015, 22:12:13 Evgeny Mandrikov wrote:</h3>
<div class="paragraph">
<p>Same is allowed in C, but seems that was not deprecated:</p>
</div>
<div class="listingblock">
<div class="content">
<pre>void fun() {

  _Bool b = 0;

  b{plus}{plus};

}</pre>
</div>
</div>
<div class="paragraph">
<p>So [~freddy.mallet] could you please update spec? Also could you please add reference to exact version of standard about deprecation in order to not be unfounded?</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_6_apr_2015_124059_evgeny_mandrikov_wrote">on 6 Apr 2015, 12:40:59 Evgeny Mandrikov wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] description is about "C&#43;&#43;", whereas increment of boolean is allowed in C also.</p>
</div>
</div>
</div>
</div>