== Why is this an issue?

There's no need for multiple, consecutive logging calls. Instead, they should be combined, to save I/O time, especially when the logs are written to file. Further, doing so can improve log output readability in concurrent applications; adjacent logger calls will not necessarily result in adjacent log entries in multithreaded environments.


=== Noncompliant code example

[source,text]
----
int result = doSomething();
LOGGER.log(LEVEL.info, "doSomething() called");
LOGGER.log(LEVEL.debug, "doSomething response: %n", result);  // Noncompliant
----


=== Compliant solution

[source,text]
----
int result = doSomething();
LOGGER.log(LEVEL.info, "doSomething() called. Response: %n", result);
----


