<code>vbNullString</code> is a special constant that denotes a null string (0), while <code>""</code> is a  literal empty string. For most purposes, the two are equivalent, but <code>vbNullString</code> is faster to assign and process, and takes less memory. <code>vbNullString</code> is therefore preferred, however some non-VB APIs or components may not properly handle null strings, and their use should be tested.


== Noncompliant Code Example

----
Text = ""
----


== Compliant Solution

----
Text = vbNullString
----

