== Why is this an issue?

Every paragraph should be commented to explain its goal and how it works. This comment can be placed either just before or just after the paragraph label. Moreover paragraphs used to close a module can be left uncommented.


=== Noncompliant code example

[source,cobol]
----
PROCEDURE DIVISION.

* Some comments
    CORRECTLY-COMMENTED-PARAGRAPH1.             *>  Compliant, the comment is just before

    ANOTHER-CORRECTLY-COMMENTED-PARAGRAPH1.     *>  Compliant, the comment is just after
* Some comments
    ...

    UNCORRECTLY-COMMENTED-PARAGRAPH1.           *> Not Compliant
    ...

*-------
    UNCORRECTLY-COMMENTED-PARAGRAPH2.           *> Not Compliant as the comment is empty
    ...

      PERFORM P1 THRU P2.
    ...

*Some comments                                  *> Compliant
    P1.
      ....


    P2.                                         *> No violation as the this P2 paragraph close a module
       MOVE A TO B.
       ...
       EXIT.
----


ifdef::env-github,rspecator-view[]
'''
== Comments And Links
(visible only on this page)

=== is related to: S1305

endif::env-github,rspecator-view[]
