== Why is this an issue?

:operationName: procedure

include::../description.adoc[]

== How to fix it

=== Code examples

==== Noncompliant code example

[source,jcl,diff-id=1,diff-type=noncompliant]
----
//MYPROC PROC
//       PEND
----

==== Compliant solution

[source,java,diff-id=1,diff-type=compliant]
----
//MYPROC PROC
//RUN    EXEC PGM=MYPROG
//       PEND
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Complete the implementation of this procedure.

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
