== Why is this an issue?

Queries with contradictory ``++WHERE++`` clauses will always return empty result sets. This is clearly a bug.


=== NonCompliant code example

[source,text]
----
SELECT *
FROM fruit
WHERE type='apple' AND type='orange'  -- Noncompliant
----

