Queries with contradictory ``++WHERE++`` clauses will always return empty result sets. This is clearly a bug.


== Noncompliant Code Example

[source,text]
----
SELECT *
FROM fruit
WHERE type='apple' AND type='orange'  -- Noncompliant
----

