<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Queries with contradictory <code>WHERE</code> clauses will always return empty result sets. This is clearly a bug.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-text" data-lang="text">SELECT *
FROM fruit
WHERE type='apple' AND type='orange'  -- Noncompliant</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Cannot match opposite condition.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>*primary: the second contradictory condition</p>
</div>
<div class="paragraph">
<p>secondary: the first contradictory condition</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_11_aug_2017_174952_ann_campbell_wrote">on 11 Aug 2017, 17:49:52 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~pierre-yves.nicolas] I&#8217;ve re-parented this subtask from <a data-rspec-id="S3651" class="rspec-auto-link">RSPEC-3651</a> to <a data-rspec-id="S3633" class="rspec-auto-link">RSPEC-3633</a></p>
</div>
</div>
<div class="sect2">
<h3 id="_relates_to_s2583">relates to: <a data-rspec-id="S2583" class="rspec-auto-link">S2583</a></h3>

</div>
<div class="sect2">
<h3 id="_is_related_to_s3651">is related to: <a data-rspec-id="S3651" class="rspec-auto-link">S3651</a></h3>

</div>
<div class="sect2">
<h3 id="_on_15_jun_2016_101656_ann_campbell_wrote">on 15 Jun 2016, 10:16:56 Ann Campbell wrote:</h3>
<div class="paragraph">
<p><a href="http://users.informatik.uni-halle.de/~brass/sqllint/semerr2.pdf" class="bare">http://users.informatik.uni-halle.de/~brass/sqllint/semerr2.pdf</a></p>
</div>
</div>
</div>
</div>