``++COPY ... SUPPRESS++`` suppresses the inclusion of the copybook contents from the source listing, making it very difficult to gain a complete understanding of what's happening in the code. This could hinder both maintenance and debugging.


== Noncompliant Code Example

----
COPY XX001234 SUPPRESS.  <* Noncompliant
----


== Compliant Solution

----
COPY XX001234.
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

include::highlighting.adoc[]

endif::env-github,rspecator-view[]
