== Why is this an issue?

``++COPY ... SUPPRESS++`` suppresses the inclusion of the copybook contents from the source listing, making it very difficult to gain a complete understanding of what's happening in the code. This could hinder both maintenance and debugging.


=== Noncompliant code example

[source,cobol]
----
COPY XX001234 SUPPRESS.  <* Noncompliant
----


=== Compliant solution

[source,cobol]
----
COPY XX001234.
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Don't "SUPPRESS" inclusion of this library in the code listing.


=== Highlighting

``++SUPPRESS++``


endif::env-github,rspecator-view[]
