<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Serde is a popular framework in the Rust ecosystem for serializing and deserializing data. It provides a flexible and efficient way to convert Rust data structures into various formats (e.g., JSON, YAML) and vice versa.</p>
</div>
<div class="paragraph">
<p>One of the core components of Serde is the <code>Visitor</code> trait, which allows custom deserialization logic by visiting each element of the data structure. According to Serde&#8217;s <a href="https://docs.rs/serde/latest/serde/de/trait.Visitor.html#method.visit_str">documentation</a>, any implementation of the <code>Visitor</code> trait that implements the <code>visit_string</code> method must also implement the <code>visit_str</code> method. Failing to do so can lead to unexpected behavior.</p>
</div>
<div class="paragraph">
<p>This rule ensures that implementations of the <code>Visitor</code> trait adhere to this requirement, promoting correctness and preventing subtle bugs in deserialization logic.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-rust" data-lang="rust">struct A;

impl&lt;'de&gt; serde::de::Visitor&lt;'de&gt; for A {
    type Value = ();

    fn expecting(&amp;self, _: &amp;mut ::std::fmt::Formatter) -&gt; ::std::fmt::Result {
      unimplemented!()
    }

    fn visit_string&lt;E&gt;(self, _v: String) -&gt; Result&lt;Self::Value, E&gt;
    where
        E: serde::de::Error,
    {
      unimplemented!()
    }
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-rust" data-lang="rust">impl&lt;'de&gt; serde::de::Visitor&lt;'de&gt; for A {
    type Value = ();

    fn expecting(&amp;self, _: &amp;mut ::std::fmt::Formatter) -&gt; ::std::fmt::Result {
        unimplemented!()
    }

    fn visit_str&lt;E&gt;(self, _v: &amp;str) -&gt; Result&lt;Self::Value, E&gt;
    where
        E: serde::de::Error,
    {
        unimplemented!()
    }

    fn visit_string&lt;E&gt;(self, _v: String) -&gt; Result&lt;Self::Value, E&gt;
    where
        E: serde::de::Error,
    {
        unimplemented!()
    }
}</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Clippy Lints - <a href="https://rust-lang.github.io/rust-clippy/master/index.html#serde_api_misuse" class="bare">https://rust-lang.github.io/rust-clippy/master/index.html#serde_api_misuse</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>