<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Schema versioning is a big topic, with best practices rather than hard-and-fast rules. One of these best practices is to include the <code>version</code> attribute in the <code>xsd:schema</code> and increment it to indicate minor changes. Nothing breaks if the value changes, but it does give the user a relatively prominent indication that something has changed.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-xml" data-lang="xml">&lt;xs:schema targetNamespace="http://www.codeSamples.com/fruit"
           xmlns:xs="http://www.w3.org/2001/XMLSchema"
           elementFormDefault="qualified"&gt;  &lt;!-- Noncompliant --&gt;
...</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-xml" data-lang="xml">&lt;xs:schema targetNamespace="http://www.codeSamples.com/fruit"
           xmlns:xs="http://www.w3.org/2001/XMLSchema"
           elementFormDefault="qualified" version="1"&gt;
...</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Add a "version" attribute to this schema.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p><code>xsd:schema</code></p>
</div>
</div>
</div>
</div>