<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Python interprets <code>False</code> as a "false" boolean value, and <code>True</code> as a "true" boolean value. Assigning them other values could lead to unexpected behaviors.</p>
</div>
<div class="paragraph">
<p>This rule raises an issue when "True" and "False", with any case, are used as identifiers.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">True = "red"  # Noncompliant</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">color = "red"  # Noncompliant</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Rename this variable.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_17_mar_2015_085602_elena_vilchik_wrote">on 17 Mar 2015, 08:56:02 Elena Vilchik wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] AFAK:</p>
</div>
<div class="olist arabic">
<ol class="arabic">
<li>
<p>"None" can&#8217;t be used as identifier.</p>
</li>
<li>
<p>"True" has value, which is "true".</p>
</li>
</ol>
</div>
</div>
</div>
</div>