include::../rationale.adoc[]

include::../how-to-fix-it.adoc[]

=== Code examples

==== Noncompliant code example

[source,sql,diff-id=1,diff-type=noncompliant]
----
DECLARE
@ClientId INT = 1, -- Noncompliant - @ClientId is unused
@DeliveryId INT = 0;
SELECT 'Date', 'Weight' FROM Claims WHERE Id = @DeliveryId;
GO
----


==== Compliant solution

[source,sql,diff-id=1,diff-type=compliant]
----
DECLARE
@DeliveryId INT = 0;
SELECT 'Date', 'Weight' FROM Claims WHERE Id = @DeliveryId;
GO
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
