include::../rationale.adoc[]

include::../how-to-fix-it.adoc[]

=== Code examples

==== Noncompliant code example

[source,sql,diff-id=1,diff-type=noncompliant]
----
declare 
@ClientId int = 1, -- Noncompliant - @ClientId is unused
@DeliveryId int = 0;
select 'Date', 'Weight' from Claims where Id = @DeliveryId;
go
----


==== Compliant solution

[source,sql,diff-id=1,diff-type=compliant]
----
declare 
@DeliveryId int = 0;
select 'Date', 'Weight' from Claims where Id = @DeliveryId;
go
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
