== Why is this an issue?

When explicitly setting the region on an AWS Client, you should always prefer providing the value from the Enum https://docs.aws.amazon.com/AWSJavaSDK/latest/javadoc/index.html?com/amazonaws/regions/Regions.html[Regions] instead of a hardcoded String. This will allow you to transparently support any change in the API and avoid mistakes.


This rule reports an issue when a hardcoded string is used instead of an available enum value.


=== Noncompliant code example

[source,java]
----
AmazonS3ClientBuilder.standard().withRegion("eu_west_1").build();
----


=== Compliant solution

[source,java]
----
AmazonS3ClientBuilder.standard().withRegion(Regions.EU_WEST_1).build();
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Give the enum value for this region instead.


=== Highlighting

The hardcoded string which could be replaced by the enum.


endif::env-github,rspecator-view[]
