== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,cobol]
----
* Password is "red!"
----

include::../see.adoc[]
ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

endif::env-github,rspecator-view[]
