A complete declaration of the structure or union shall be included within any translation unit that refers to that structure. See section 6.1.2.5 of ISO 9899:1990 [2] for a full description of incomplete types.


== Noncompliant Code Example

----
struct tnode * pt; // tnode is incomplete
----


== Compliant Solution

----
struct tnode * pt; // tnode is incomplete at this point
struct tnode
{
  int count;
  struct tnode * left;
  struct tnode * right;
}; // type tnode is now complete
----


== See

* MISRA C:2004, 18.1 - All structure and union types shall be complete at the end of a translation unit.


ifdef::env-github,rspecator-view[]
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
