== Why is this an issue?

A complete declaration of the structure or union shall be included within any translation unit that refers to that structure. See section 6.1.2.5 of ISO 9899:1990 [2] for a full description of incomplete types.


=== Noncompliant code example

[source,cpp]
----
struct tnode * pt; // tnode is incomplete
----


=== Compliant solution

[source,cpp]
----
struct tnode * pt; // tnode is incomplete at this point
struct tnode
{
  int count;
  struct tnode * left;
  struct tnode * right;
}; // type tnode is now complete
----


== Resources

* MISRA C:2004, 18.1 - All structure and union types shall be complete at the end of a translation unit.


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
