== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,java]
----
public void run(javax.servlet.http.HttpServletRequest request) throws ClassNotFoundException {
    String name = request.getParameter("name");
    Class clazz = Class.forName(name);  // Noncompliant
} 
----

=== Compliant solution

[source,java]
----
public void run(javax.servlet.http.HttpServletRequest request) throws ClassNotFoundException {
    String name = request.getParameter("name");
    if (this.allowed.contains(name)) {
        Class clazz = Class.forName(name);
    }
} 
----

include::../see.adoc[]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../highlighting.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
