== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,sql]
----
IF NOT x <> y THEN   -- Noncompliant
  -- ...
END IF;
----

=== Compliant solution

[source,sql]
----
IF x = y THEN
  -- ...
END IF;
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
