include::../description.adoc[]

== Noncompliant Code Example

----
IF NOT x <> y THEN   -- Noncompliant
  -- ...
END IF;
----

== Compliant Solution

----
IF x = y THEN
  -- ...
END IF;
----

ifdef::env-github,rspecator-view[]
'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
