<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>It may be a good idea to raise an exception in a constructor if you&#8217;re unable to fully flesh the object in question, but not in an <code>exception</code> constructor. If you do, you&#8217;ll interfere with the exception that was originally being thrown. Further, it is highly unlikely that an exception raised in the creation of an exception will be properly handled in the calling code, and the unexpected, unhandled exception will lead to program termination.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-csharp" data-lang="csharp">class MyException: Exception
{
    public void MyException()
    {
         if (bad_thing)
         {
             throw new Exception("A bad thing happened");  // Noncompliant
          }
    }
}</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Avoid throwing exceptions in this constructor.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p><code>throw ...</code></p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_4_aug_2016_173236_ann_campbell_wrote">on 4 Aug 2016, 17:32:36 Ann Campbell wrote:</h3>
<div class="paragraph">
<p><a href="http://www.acodersjourney.com/2016/08/top-15-c-exception-handling-mistakes-avoid/" class="bare">http://www.acodersjourney.com/2016/08/top-15-c-exception-handling-mistakes-avoid/</a></p>
</div>
</div>
<div class="sect2">
<h3 id="_on_14_oct_2019_120342_andrei_epure_wrote">on 14 Oct 2019, 12:03:42 Andrei Epure wrote:</h3>
<div class="paragraph">
<p>This rule has been deprecated for <code>C#</code> and <code>VB.NET</code>, see details in <a href="https://github.com/SonarSource/sonar-dotnet/issues/2468">SonarSource/sonar-dotnet#2468</a> . Due to a bug in rule-api, sub-task deprecation does not work. We&#8217;ve decided to deprecated it at parent RSPEC level due to the fact that it was not implemented in <code>C&#43;&#43;</code>.</p>
</div>
<div class="paragraph">
<p>The rule still makes sense for C&#43;&#43;, so a new RSPEC can be created for it once a decision is taken to implement it.</p>
</div>
</div>
</div>
</div>