Shared coding conventions allow teams to collaborate effectively. This rule checks that labels are on lines by themselves, which makes them more immediately visible as such. 


== Noncompliant Code Example

[source,rpg]
----
LBLVAL:    IF         COND(&ERRFLG = ' ') THEN (DO)
           CHGVAR     VAR(&MSG) VALUE('Validation Completed')
           ENDDO 
----


== Compliant Solution

[source,rpg]
----
LBLVAL:
           IF         COND(&ERRFLG = ' ') THEN (DO)
           CHGVAR     VAR(&MSG) VALUE('Validation Completed')
           ENDDO 
----


ifdef::env-github,rspecator-view[]
'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
