== Why is this an issue?

Shared coding conventions allow teams to collaborate effectively. This rule checks that labels are on lines by themselves, which makes them more immediately visible as such. 


=== Noncompliant code example

[source,rpg]
----
LBLVAL:    IF         COND(&ERRFLG = ' ') THEN (DO)
           CHGVAR     VAR(&MSG) VALUE('Validation Completed')
           ENDDO 
----


=== Compliant solution

[source,rpg]
----
LBLVAL:
           IF         COND(&ERRFLG = ' ') THEN (DO)
           CHGVAR     VAR(&MSG) VALUE('Validation Completed')
           ENDDO 
----


ifdef::env-github,rspecator-view[]
'''
== Comments And Links
(visible only on this page)

=== on 29 Sep 2014, 13:29:43 Ann Campbell wrote:
As it turns out, this rule is applicable to CLP, not RPG.

endif::env-github,rspecator-view[]
