<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>include::../description.adoc</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>class FooIterator : Iterator&lt;Foo&gt;, Iterable&lt;Foo&gt; {

    override fun iterator(): Iterator&lt;Foo&gt; {
        return this // Noncompliant
    }
    // ...
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>class FooSequence : Iterable&lt;Foo&gt; {

    override fun iterator(): Iterator&lt;Foo&gt; {
        return object : Iterator&lt;Foo&gt; {
            override fun hasNext(): Boolean {
                //...
            }

            override fun next(): Foo {
                //...
            }
        }
    }
    //...
}</pre>
</div>
</div>
</div>
</div>