== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,sql]
----
DECLARE @a INT;
DECLARE @b INT = 2;
SET @a = @a; -- Noncompliant
----

=== Compliant solution

[source,sql]
----
DECLARE @a INT;
DECLARE @b INT = 2;
SET @a = @b;
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
