:identifier_capital_plural: Local variables and function parameters
:identifier: local variable and function parameter
:identifier_plural: local variables and function parameters
:identifier_or: local variable or function parameter
:regex: ^[_a-z][a-zA-Z0-9]*$

include::../rule.adoc[]

=== Code examples

==== Noncompliant code example

With the default regular expression ``{regex}``:

[source,flex,diff-id=1,diff-type=noncompliant]
----
public function doSomething(my_param:int):void
{  
  var LOCAL:int;
  ...
}
----

==== Compliant solution

[source,flex,diff-id=1,diff-type=compliant]
----
public function doSomething(myParam):void
{  
  var local;
  ...
}
----

== Resources

=== Documentation

* Wikipedia - https://en.wikipedia.org/wiki/Naming_convention_(programming)[Naming Convention (programming)]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../parameters.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
