include::../description.adoc[]

== Noncompliant Code Example

With the default regular expression ``++^[_a-z][a-zA-Z0-9]*$++``:

----
public function doSomething(my_param:int):void
{  
  var LOCAL:int;
  ...
}
----

== Compliant Solution

----
public function doSomething(myParam):void
{  
  var local;
  ...
}
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../parameters.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
