include::../description.adoc[]

== Noncompliant Code Example

With the default regular expression ``++^[_a-z][a-zA-Z0-9]*$++``:

----
public function doSomething(my_param:int):void
{  
  var LOCAL:int;
  ...
}
----

== Compliant Solution

----
public function doSomething(myParam):void
{  
  var local;
  ...
}
----

ifdef::rspecator-view[]
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::rspecator-view[]
