<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>The name of a method should communicate what it does, and the names of its parameters should indicate how they&#8217;re used. If a method and its parameter have the same name it is an indication that one of these rules of thumb has been broken, if not both. Even if by some trick of language that&#8217;s not the case, it is still likely to confuse callers and maintainers.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-text" data-lang="text">public void Login(string login)  // Noncompliant
{
  //...
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-text" data-lang="text">public void Login(string userName)
{
  //...
}</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Rename the parameter 'xxx' so that it doesn&#8217;t duplicate the method name.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>primary: The parameter name</p>
</div>
<div class="paragraph">
<p>secondary: the method name</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_16_feb_2017_201032_ann_campbell_wrote">on 16 Feb 2017, 20:10:32 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~amaury.leve] it seems like you said something about getting the wrong behavior when the method is called, but I don&#8217;t remember the details. Could you fill that in &amp; double-check my code samples, pls?</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_17_feb_2017_181605_amaury_levé_wrote">on 17 Feb 2017, 18:16:05 Amaury Levé wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] I couldn&#8217;t find any example where it could go really wrong. I think this is more to have a meaningful method name and a meaningful parameter name.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_2_mar_2017_140737_valeri_hristov_wrote">on 2 Mar 2017, 14:07:37 Valeri Hristov wrote:</h3>
<div class="paragraph">
<p>I like the FxCop message more: <code>Change either member or parameter name 'method1' so that one differs from the other</code>. [~ann.campbell.2], what do you think, should we change ours too, or we want be as short as possible?</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_3_mar_2017_083914_freddy_mallet_wrote">on 3 Mar 2017, 08:39:14 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>+1 for your issue message proposal [~valeri.hristov]</p>
</div>
</div>
</div>
</div>