This rule raises an issue when a loop iteration through the dataframe is used instead of vectorized operations.

== Why is this an issue?

Pandas is optimized for vectorized operations, which means it's much faster to apply operations to entire columns or slices of data than to loop through rows. Using a ``++for++`` loop can be significantly slower, especially on large datasets.

Many Pandas methods and functions are designed to work with entire columns or slices of data. When you iterate through rows, you miss out on the full power of Pandas.

Instead of using a ``++for++`` loop to iterate through a DataFrame, it's generally recommended to use vectorized operations and Pandas functions to perform operations on columns or subsets of data. This approach is not only more efficient but also leads to cleaner and more maintainable code.

=== Code examples

==== Noncompliant code example

[source,python,diff-id=1,diff-type=noncompliant]
----
import pandas as pd

df = pd.DataFrame({'X': ['a', 'c', 'c'],
                   'Y': [1, 2, 3]})

for index, row in df.iterrows(): # Noncompliant: iteration over dataframe with loop is slow
  print(f"X: {row['X']}, Y: {row['Y']}")
----

==== Compliant solution

[source,python,diff-id=1,diff-type=compliant]
----
import pandas as pd

df = pd.DataFrame({'X': ['a', 'c', 'c'],
                   'Y': [1, 2, 3]})

df.apply(lambda row: print(f"X: {row['X']}, Y: {row['Y']}"), axis = 1) # Compliant
----



== Resources
=== Documentation
* Pandas Documentation - https://pandas.pydata.org/pandas-docs/stable/user_guide/basics.html#iteration[Iteration]
* Pandas Documentation - https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.apply.html[pandas.DataFrame.apply]
