== How to fix it in AWS OpenSearch

=== Code examples

==== Noncompliant code example

The resource `CfnDomain` uses a weak TLS security policy, by default.

[source,javascript,diff-id=1,diff-type=noncompliant]
----
import { aws_opensearchservice as os } from 'aws-cdk-lib';

new os.CfnDomain(this, 'Example', {
  domainEndpointOptions: {
    enforceHttps: true,
  },
}); // Noncompliant
----

==== Compliant solution


[source,javascript,diff-id=1,diff-type=compliant]
----
import { aws_opensearchservice as os } from 'aws-cdk-lib';

new os.CfnDomain(this, 'Example', {
  domainEndpointOptions: {
    enforceHttps: true,
    tlsSecurityPolicy: 'Policy-Min-TLS-1-2-2019-07',
  },
});
----

=== How does this work?

include::../../common/fix/fix.adoc[]
