== Deprecated or obsolete HTML tags and attributes should not be used

HTML5 has removed a number of tags and attributes that were present in earlier versions of HTML.
These elements and attributes are considered obsolete and should be avoided because:

* They are not supported in modern browsers.
* They can cause inconsistent rendering across different user agents.
* They reduce maintainability and accessibility.
* Their functionality should be implemented using CSS or JavaScript instead.

This rule raises an issue when any deprecated or obsolete HTML tag or attribute is used.

== Noncompliant Code Example

[source,html]
----
<!-- Deprecated tags -->
<center>Centered text</center> <!-- Noncompliant -->
<font color="red">Red text</font> <!-- Noncompliant -->
<marquee>Scrolling text</marquee> <!-- Noncompliant -->

<!-- Deprecated attributes -->
<table border="1"> <!-- Noncompliant -->
  <tr align="center"> <!-- Noncompliant -->
    <td bgcolor="yellow">Cell</td> <!-- Noncompliant -->
  </tr>
</table>
----

== Compliant Code Example

[source,html]
----
<!-- Use CSS for styling and layout -->
<p style="text-align: center;">Centered text</p>
<span style="color: red;">Red text</span>
<div style="overflow: hidden; white-space: nowrap; animation: scroll 10s linear infinite;">
  Scrolling text
</div>

<table style="border: 1px solid black;">
  <tr style="text-align: center;">
    <td style="background-color: yellow;">Cell</td>
  </tr>
</table>
----

== See

* https://developer.mozilla.org/en-US/docs/Web/HTML/Element#obsolete_and_deprecated_elements[MDN Web Docs - Obsolete and deprecated elements]
* https://html.spec.whatwg.org/multipage/obsolete.html[WHATWG HTML Living Standard - Obsolete features]
* https://www.w3.org/TR/html401/struct/global.html#h-7.5.1[W3C HTML 4.01 Specification - Deprecated elements]
* https://www.tutorialspoint.com/html/html_deprecated_tags.htm[TutorialsPoint - HTML Deprecated Tags]