== Why is this an issue?

HTML5 has removed a number of tags that were present in earlier versions of HTML.
These elements are considered obsolete and should be avoided because:

* They are not supported in modern browsers.
* They can cause inconsistent rendering across different user agents.
* They reduce maintainability and accessibility.

This rule raises an issue when any deprecated or obsolete HTML tag is used.

== Noncompliant Code Example

[source,html]
----
<!-- Deprecated tags -->
<center>Centered text</center> <!-- Noncompliant -->
<font color="red">Red text</font> <!-- Noncompliant -->
<marquee>Scrolling text</marquee> <!-- Noncompliant -->
----

== Compliant Code Example

[source,html]
----
<!-- Use CSS for styling and layout -->
<p style="text-align: center;">Centered text</p>
<span style="color: red;">Red text</span>
<div style="overflow: hidden; white-space: nowrap; animation: scroll 10s linear infinite;">
  Scrolling text
</div>
----

== Resources
=== Documentation

* https://developer.mozilla.org/en-US/docs/Web/HTML/Element#obsolete_and_deprecated_elements[MDN Web Docs - Obsolete and deprecated elements]
* https://html.spec.whatwg.org/multipage/obsolete.html[WHATWG HTML Living Standard - Obsolete features]
* https://www.w3.org/TR/html401/struct/global.html#h-7.5.1[W3C HTML 4.01 Specification - Deprecated elements]
* https://www.tutorialspoint.com/html/html_deprecated_tags.htm[TutorialsPoint - HTML Deprecated Tags]