include::../rule.adoc[]

=== Noncompliant code example

[source,python]
----
for d in lib_dirs:
    # NOSONAR: lib_dirs is undefined
    pass
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Is ``++#NOSONAR++`` used to exclude false-positive or to hide real quality flaw?

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
