include::../rule.adoc[]

== Noncompliant Code Example

[source,python]
----
for d in lib_dirs:
    # NOSONAR: lib_dirs is undefined
    pass
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
