== Why is this an issue?

Control characters aren't visible to maintainers, so they should be escaped.

=== Noncompliant code example

[source,cpp]
----
const char* foo = "A	B";  // Noncompliant, contains a tabulation
----

=== Compliant solution

[source,cpp]
----
const char* foo = "A\tB";  // Compliant, use escaped value
----

=== Exceptions

Raw string literals has no escape character mechanism.

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Replace this character by its escaped value "xxx".


=== Highlighting

control character


'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
