<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Java 15 introduced feature of <code>sealed</code> classes. With <code>sealed</code> classes and interfaces you can specify a strict hierarchy of types and restrict possible inheritance.</p>
</div>
<div class="paragraph">
<p>Although this feature can help to make code safer, it is not applicable everywhere. Functional interfaces can not be sealed. This means that if an interface with a single abstract method is declared with a <code>sealed</code> keyword, its implementation can&#8217;t be replaced with a lambda.</p>
</div>
<div class="paragraph">
<p>This rule reports an issue when an interface with a single abstract method is marked <code>sealed</code>.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">public sealed interface F permits ... { // Noncompliant
  void f();
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">public interface F { // Compliant
  void f();
}</code></pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p><a href="https://docs.oracle.com/javase/specs/jls/se16/preview/specs/sealed-classes-jls.html#jls-8.1.1.2">Sealed classes specification</a></p>
</li>
</ul>
</div>
<hr>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove this "sealed" keyword if this interface is supposed to be functional</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>interface declaration</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_22_june_2021_155300_alban_auzeill_wrote">on 22 June 2021, 15:53:00 Alban Auzeill wrote:</h3>
<div class="paragraph">
<p>For the <code>java</code> analyzer we decided to not implement this rule,
see why on <a href="https://github.com/SonarSource/sonar-java/pull/3658"><strong>this github PR</strong></a>
and <a href="https://jira.sonarsource.com/browse/">SONARJAVA-3775</a></p>
</div>
</div>
</div>
</div>