<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>There&#8217;s no point in forcing the overhead of a function or method call for a function that always returns the same constant value. Even worse, the fact that a function call must be made will likely mislead developers who call the method thinking that something more is done. Declare a constant instead.</p>
</div>
<div class="paragraph">
<p>This rule raises an issue on functions that contain only one  statement: the <code>return</code> of a constant value.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-swift" data-lang="swift">func getBestNumber() -&gt; Int {
  return 12  // Noncompliant
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-swift" data-lang="swift">let bestNumber = 12;</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_exceptions">Exceptions</h3>
<div class="paragraph">
<p>Methods which are members of a class having a type inheritance clause are ignored.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove this method and declare a constant for this value.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>returned value</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_is_related_to_s3516">is related to: <a data-rspec-id="S3516" class="rspec-auto-link">S3516</a></h3>

</div>
<div class="sect2">
<h3 id="_on_1_dec_2015_110813_michael_gumowski_wrote">on 1 Dec 2015, 11:08:13 Michael Gumowski wrote:</h3>
<div class="paragraph">
<p>LGTM!</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_5_jan_2016_100718_nicolas_peru_wrote">on 5 Jan 2016, 10:07:18 Nicolas Peru wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] I kinda lack the definition of what is a constant and limitation of cases to be 100% sure of the scope of this rule.</p>
</div>
<div class="paragraph">
<p>What about the following for instance :</p>
</div>
<div class="listingblock">
<div class="content">
<pre>int foo(boolean a) {
   if(a) {
     return 12;
   }
   return 12;
}</pre>
</div>
</div>
<div class="paragraph">
<p>this is actually detected by intellij idea but the following is not :</p>
</div>
<div class="listingblock">
<div class="content">
<pre>  int foo(boolean a) {
    int b = 12;
    if(a) {
      return b;
    }
    return b;
  }</pre>
</div>
</div>
<div class="paragraph">
<p>IMO first implementation should be limited to method with only one return statement and a constant with the following definition:</p>
</div>
<div class="paragraph">
<p>A constant is a literal or a static final primitive variable.</p>
</div>
<div class="paragraph">
<p>Then we might want to improve this using  SE engine to detect cases like I described.</p>
</div>
<div class="paragraph">
<p>WDYT ?</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_5_jan_2016_162117_ann_campbell_wrote">on 5 Jan 2016, 16:21:17 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>Uhm&#8230;&#8203; I thought it was evident from the examples [~nicolas.peru] that that&#8217;s all that was expected.</p>
</div>
</div>
</div>
</div>