<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>The <code>Obsolete</code> attribute can be applied with or without a message argument. Marking something <code>Obsolete</code> without including advice on why it&#8217;s obsolete or what to use instead will lead maintainers to waste time trying to figure those things out.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-vbnet" data-lang="vbnet">Public Class Car

    &lt;Obsolete&gt;  ' Noncompliant
    Public Sub CrankEngine(Turns As Integer)
        ' ...
    End Sub

End Class</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-vbnet" data-lang="vbnet">Public Class Car

    &lt;Obsolete("Replaced by the automatic starter")&gt;
    Public Sub CrankEngine(Turns As Integer)
        ' ...
    End Sub

End Class</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Add an explanation.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_21_jan_2015_174731_nicolas_peru_wrote">on 21 Jan 2015, 17:47:31 Nicolas Peru wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] I don&#8217;t think this is a relevant for java as we already have http://jira.sonarsource.com/browse/RSPEC-1123</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_22_jan_2015_074148_ann_campbell_wrote">on 22 Jan 2015, 07:41:48 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~nicolas.peru] does <a data-rspec-id="S1123" class="rspec-auto-link">RSPEC-1123</a> set an issue when @Deprecated/@deprecated does not include explanatory text?</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_11_may_2015_092036_dinesh_bolkensteyn_wrote">on 11 May 2015, 09:20:36 Dinesh Bolkensteyn wrote:</h3>
<div class="paragraph">
<p>LGTM</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_18_jul_2013_073152_dinesh_bolkensteyn_wrote">on 18 Jul 2013, 07:31:52 Dinesh Bolkensteyn wrote:</h3>
<div class="paragraph">
<p>Implemented by http://jira.codehaus.org/browse/SONARJAVA-214</p>
</div>
</div>
</div>
</div>