<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>The size of integer that is required when a pointer is converted to an integer is implementation-defined. Casting between a pointer and an integer type should be avoided where possible, but may be unavoidable when addressing memory mapped registers or other hardware specific features.</p>
</div>
<div class="paragraph">
<p>Note that C&#43;&#43; does not permit a pointer to be converted to any floating type.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-text" data-lang="text">struct S
{
  int32_t i;
  int32_t j;
};

void f ( S * s )
{
  int32_t p = reinterpret_cast&lt; int32_t &gt;( s ); // Noncompliant
}</code></pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p>MISRA C&#43;&#43;:2008, 5-2-9</p>
</li>
</ul>
</div>
<hr>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_duplicates_s1767">duplicates: <a data-rspec-id="S1767" class="rspec-auto-link">S1767</a></h3>

</div>
<div class="sect2">
<h3 id="_is_related_to_s1767">is related to: <a data-rspec-id="S1767" class="rspec-auto-link">S1767</a></h3>

</div>
<div class="sect2">
<h3 id="_relates_to_s1944">relates to: <a data-rspec-id="S1944" class="rspec-auto-link">S1944</a></h3>

</div>
<div class="sect2">
<h3 id="_on_23_oct_2014_151028_ann_campbell_wrote">on 23 Oct 2014, 15:10:28 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~samuel.mercier] I&#8217;m thinking this is either a "bug" or a "pitfall"&#8230;&#8203;</p>
</div>
</div>
</div>
</div>