include::../common/rule.adoc[tag=header]

=== Code examples

==== Noncompliant code example

[source,javascript,diff-id=1,diff-type=noncompliant]
----
<input type="text" autocomplete="foo" />
----

==== Compliant solution

[source,javascript,diff-id=1,diff-type=compliant]
----
<input type="text" autocomplete="name" />
----

include::../common/rule.adoc[tag=footer]