include::../common/rule.adoc[]

=== Code examples

==== Noncompliant code example

[source,javascript,diff-id=1,diff-type=noncompliant]
----
<input type="text" autocomplete="foo" />
----

==== Compliant solution

[source,javascript,diff-id=1,diff-type=compliant]
----
<input type="text" autocomplete="name" />
----

== Resources
=== Documentation

* WCAG - https://www.w3.org/WAI/WCAG21/Understanding/identify-input-purpose[Identify Input Purpose]
* WCAG - https://www.w3.org/TR/WCAG21/#input-purposes[Input Purposes for User Interface Components]
* HTML Standard - https://html.spec.whatwg.org/multipage/forms.html#enabling-client-side-automatic-filling-of-form-controls[Enabling client-side automatic filling of form controls]
* HTML Standard - https://html.spec.whatwg.org/multipage/form-control-infrastructure.html#attr-fe-autocomplete[Autofilling form controls: the autocomplete attribute]
