include::../description.adoc[]

== Noncompliant Code Example

----
public long ComputeDurationInMilliseconds()
{
  long duration = (((hours * 60) + minutes) * 60 + seconds ) * 1000 ;
  return duration;
}

public void DoSomething()
{
  ApplicationException myException = new ApplicationException();
  throw myException;
}
----

== Compliant Solution

----
public long ComputeDurationInMilliseconds() 
{
  return (((hours * 60) + minutes) * 60 + seconds ) * 1000 ;
}

public void DoSomething() 
{
  throw new ApplicaitonException();
}
----

ifdef::rspecator-view[]
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::rspecator-view[]
