== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,csharp]
----
public long ComputeDurationInMilliseconds()
{
  long duration = (((hours * 60) + minutes) * 60 + seconds ) * 1000 ;
  return duration;
}

public void DoSomething()
{
  ApplicationException myException = new ApplicationException();
  throw myException;
}
----

=== Compliant solution

[source,csharp]
----
public long ComputeDurationInMilliseconds() 
{
  return (((hours * 60) + minutes) * 60 + seconds ) * 1000 ;
}

public void DoSomething() 
{
  throw new ApplicaitonException();
}
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
