<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when a deprecated Numpy alias of a built-in type is used.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>In NumPy, some built-in types such as <code>int</code> have aliases in the form of <code>numpy.int</code>. However, these aliases have been deprecated and should not be used anymore.</p>
</div>
<div class="paragraph">
<p>The following deprecated aliases should be replaced with their built-in alternatives:</p>
</div>
<table class="tableblock frame-all grid-all stretch">
<colgroup>
<col style="width: 50%;">
<col style="width: 50%;">
</colgroup>
<thead>
<tr>
<th class="tableblock halign-center valign-top">Deprecated name</th>
<th class="tableblock halign-center valign-top">Equivalent built-in type</th>
</tr>
</thead>
<tbody>
<tr>
<td class="tableblock halign-center valign-top"><p class="tableblock">numpy.bool</p></td>
<td class="tableblock halign-center valign-top"><p class="tableblock">bool</p></td>
</tr>
<tr>
<td class="tableblock halign-center valign-top"><p class="tableblock">numpy.int</p></td>
<td class="tableblock halign-center valign-top"><p class="tableblock">int</p></td>
</tr>
<tr>
<td class="tableblock halign-center valign-top"><p class="tableblock">numpy.float</p></td>
<td class="tableblock halign-center valign-top"><p class="tableblock">float</p></td>
</tr>
<tr>
<td class="tableblock halign-center valign-top"><p class="tableblock">numpy.complex</p></td>
<td class="tableblock halign-center valign-top"><p class="tableblock">complex</p></td>
</tr>
<tr>
<td class="tableblock halign-center valign-top"><p class="tableblock">numpy.object</p></td>
<td class="tableblock halign-center valign-top"><p class="tableblock">object</p></td>
</tr>
<tr>
<td class="tableblock halign-center valign-top"><p class="tableblock">numpy.str</p></td>
<td class="tableblock halign-center valign-top"><p class="tableblock">str</p></td>
</tr>
<tr>
<td class="tableblock halign-center valign-top"><p class="tableblock">numpy.long</p></td>
<td class="tableblock halign-center valign-top"><p class="tableblock">int</p></td>
</tr>
<tr>
<td class="tableblock halign-center valign-top"><p class="tableblock">numpy.unicode</p></td>
<td class="tableblock halign-center valign-top"><p class="tableblock">str</p></td>
</tr>
</tbody>
</table>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>To fix this issue, make sure to replace deprecated NumPy type aliases with their corresponding built-in types.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">import numpy as np
def foo():
    x = np.int(42)  # Noncompliant: deprecated type alias</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">import numpy as np
def foo():
    x = 42  # Compliant</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>NumPy Documentation - <a href="https://numpy.org/devdocs/user/basics.types.html#basics-types">Data types</a></p>
</li>
<li>
<p>NumPy Documentation - <a href="https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations">Deprecation of aliases of builtin types</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>