== Why is this an issue?

This rule allows banning the use of some programs or procedures.

=== Code examples

==== Noncompliant code example

With ``++programName++`` configured with ``SORT``:


[source,jcl]
----
//EXEC PGM=SORT
//EXEC PROC=SORT
//EXEC SORT
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Remove the use of this forbidden program or procedure. Use {replacementProgramName} instead.


=== Parameters

.progName
****

Name of the forbidden program or procedure. Use a regex to forbid multiple programs.

****
.replacementProgramName
****

Name of the program to use as a replacement of the forbidden one.
****


include::../highlighting.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
