== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,cpp]
----
a = b + c;   // This is a trailing comment that could be very very long
----

=== Compliant solution

[source,cpp]
----
// This very long comment is better placed before the line of code
a = b + c;
----

=== Exceptions

The rule will not raise an issue if the trailing comment is preceded by an opening brace, closing brace (optionally followed by a semicolon) or a preprocessor directive.

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../parameters.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
