include::../description.adoc[]

== Noncompliant Code Example

----
a = b + c;   // This is a trailing comment that could be very very long
----

== Compliant Solution

----
// This very long comment is better placed before the line of code
a = b + c;
----

== Exceptions

The rule will not raise an issue if the trailing comment is preceded by an opening brace, closing brace (optionally followed by a semicolon) or a preprocessor directive.
