== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,sql]
----
IF NOT (@a = 2) -- Noncompliant
BEGIN
  ...
END

IF NOT (@b < 10) -- Noncompliant
BEGIN
  ...
END
----

=== Compliant solution

[source,sql]
----
IF @a <> 2
BEGIN
  ...
END

IF @b >= 10
BEGIN
  ...
END
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
