<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>The <code>ismap</code> attribute in an <code>img</code> tag creates a server-side image map: The browser sends the coordinates of the clicked point to the server.</p>
</div>
<div class="paragraph">
<p>For any person who cannot use a mouse, this form of navigation is inaccessible because it is the position of the cursor on the image that determines the action.</p>
</div>
<div class="paragraph">
<p>On the other hand, client-side image maps, which use the <code>usemap</code> attribute allow for each clickable area to specify an alternate text, enabling accessibility for the blind.</p>
</div>
<div class="paragraph">
<p>Further, in terms of separation of concerns, it is definitely better to leave the task of mapping pixels to links to the client.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-html" data-lang="html">&lt;a href="click_on_world_map.php" target="_self"&gt;
  &lt;img src="world_map.png" ismap&gt;              &lt;!-- Noncompliant --&gt;
&lt;/a&gt;</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-html" data-lang="html">&lt;img src="world_map.png" usemap="#world_map"&gt;

&lt;map name="world_map"&gt;
  &lt;area shape="rect" coords="0,0,10,10" href="france.html" alt="France"&gt;
  &lt;area shape="circle" coords="20,20,10" href="spain.html" alt="Spain"&gt;
  &lt;area shape="circle" coords="30,30,8" href="england.html" alt="England"&gt;
  &lt;!-- ... --&gt;
&lt;/map&gt;</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Use the "map" tag and "area" tags instead.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_is_duplicated_by_s1925">is duplicated by: <a data-rspec-id="S1925" class="rspec-auto-link">S1925</a></h3>

</div>
<div class="sect2">
<h3 id="_on_19_jun_2013_124916_dinesh_bolkensteyn_wrote">on 19 Jun 2013, 12:49:16 Dinesh Bolkensteyn wrote:</h3>
<div class="paragraph">
<p><a href="http://www.jimthatcher.com/webcourse5.htm" class="bare">http://www.jimthatcher.com/webcourse5.htm</a></p>
</div>
</div>
<div class="sect2">
<h3 id="_on_19_jun_2013_132936_dinesh_bolkensteyn_wrote">on 19 Jun 2013, 13:29:36 Dinesh Bolkensteyn wrote:</h3>
<div class="paragraph">
<p>Ann is going to review this description.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_8_jul_2013_182219_freddy_mallet_wrote">on 8 Jul 2013, 18:22:19 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>Is implemented by http://jira.codehaus.org/browse/SONARPLUGINS-2991</p>
</div>
</div>
</div>
</div>