=== Noncompliant code example

[source,text]
----
SET SERVEROUTPUT ON

DECLARE
  name VARCHAR2(42) := ''; /* This is equivalent to name := NULL; */
BEGIN
  IF name = '' THEN /* Noncompliant; equivalent to name = NULL & therefore always FALSE*/
    DBMS_OUTPUT.PUT_LINE('True'); /* Unreachable */
  ELSE
    DBMS_OUTPUT.PUT_LINE('False');
  END IF;
END;
/
----
