<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Calls to <code>std::format</code> and <code>std::vformat</code> can receive either format strings whose replacement fields are fully indexed or fully non-indexed.</p>
</div>
<div class="paragraph">
<p>Explicitly indexing replacement fields is useful when arguments are not used in order or are used multiple times. Implicit indexing has the advantage of being terse and simple.</p>
</div>
<div class="paragraph">
<p>Using explicit indexing in a context where implicit indexing would have the same behavior is more verbose and error-prone. It might confuse the reader, who might expect the arguments not to be used in order.</p>
</div>
<div class="paragraph">
<p>This rule raises an issue on explicit indexing that should be replaced by implicit indexing.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">std::string printRectangle(Rectangle const&amp; r) {
    return std::format("[{0},{1}]:({2},{3})", r.x, r.y, r.width, r.height); // Noncompliant
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">std::string printRectangle(Rectangle const&amp; r) {
    return std::format("[{},{}]:({},{})", r.x, r.y, r.width, r.height); // Compliant
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_exceptions">Exceptions</h3>
<div class="paragraph">
<p>Format strings with nested replacement fields are not considered by this rule as their order is less intuitive.</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">std::string printNDigits() {
    return std::format("{0:*&lt;{1}} {2:*&gt;{3}}", 10, 2, 11, 3); // Compliant
}</code></pre>
</div>
</div>
</div>
</div>
</div>