== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

With the default regular expression ``++^[_a-z][a-zA-Z0-9]*$++``:

[source,flex]
----
class MyClass {
  public var my_field:int;
}
----

=== Compliant solution

[source,flex]
----
public class MyClass {
   public var myField:int;
}
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

=== Parameters

.format
****

----
^[_a-z][a-zA-Z0-9]*$
----

Regular expression used to check the field names against.
****


'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
