== Why is this an issue?

include::../description.adoc[]

=== Code examples

The examples below enforce the following regular expression ``++^[_a-z][a-zA-Z0-9]*$++``:

==== Noncompliant code example

[source,flex,diff-id=1,diff-type=noncompliant]
----
class MyClass {
  public var my_field:int;
}
----

==== Compliant solution

[source,flex,diff-id=1,diff-type=compliant]
----
public class MyClass {
   public var myField:int;
}
----

== Resources

=== Documentation

* Wikipedia - https://en.wikipedia.org/wiki/Naming_convention_(programming)[Naming Convention (programming)]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

=== Parameters

.format
****

----
^[_a-z][a-zA-Z0-9]*$
----

Regular expression used to check the field names against.
****


'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
