== How to fix it in Foundation

=== Code examples

==== Noncompliant code example

TLS 1.1 or lower will only be allowed if an exception has been added to the App Transport Security configuration.
This is enabled for a specific domain using the ``++NSExceptionAllowsInsecureHTTPLoads++`` key.

[source,xml]
----
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
<plist version="1.0">
<dict>
	<key>NSAppTransportSecurity</key>
	<dict>
		<key>NSExceptionDomains</key>
		<dict>
			<key>example.com</key>
			<dict>
				<key>NSExceptionAllowsInsecureHTTPLoads</key> <!-- Noncompliant -->
				<true/>
			</dict>
		</dict>
	</dict>
</dict>
</plist>
----

In addition, ``++URLSessionConfiguration++`` has to be configured to allow TLS 1.1 or lower.

[source,swift,diff-id=1,diff-type=noncompliant]
----
import Foundation

let configuration = URLSessionConfiguration.default
configuration.tlsMinimumSupportedProtocol = .tlsProtocol11  // Noncompliant
----

==== Compliant solution

The easiest solution is to remove the ``++NSExceptionAllowsInsecureHTTPLoads++`` key from the App Transport Security configuration.

In addition, it is a good practice to set the ``++tlsMinimumSupportedProtocol++`` to ``++tlsProtocol12++`` (the default) or higher.

[source,swift,diff-id=1,diff-type=compliant]
----
import Foundation

let configuration = URLSessionConfiguration.default
configuration.tlsMinimumSupportedProtocol = .tlsProtocol12
----

=== How does this work?

include::../../common/fix/fix.adoc[]
