== How to fix it in Foundation

=== Code examples

==== Noncompliant code example

[source,swift,diff-id=1,diff-type=noncompliant]
----
import Foundation

let configuration = URLSessionConfiguration.default
configuration.tlsMinimumSupportedProtocol = .tlsProtocol11  // Noncompliant
----

==== Compliant solution

[source,swift,diff-id=1,diff-type=compliant]
----
import Foundation

let configuration = URLSessionConfiguration.default
configuration.tlsMinimumSupportedProtocol = .tlsProtocol12
----

=== How does this work?

include::../../common/fix/fix.adoc[]
