== How to fix it in WebKit

=== Code examples

==== Noncompliant code example

TLS 1.1 or lower will only be allowed if an exception has been added to the App Transport Security configuration.
This is enabled for a specific domain using the ``++NSExceptionAllowsInsecureHTTPLoads++`` key.

[source,xml]
----
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
<plist version="1.0">
<dict>
	<key>NSAppTransportSecurity</key>
	<dict>
		<key>NSExceptionDomains</key>
		<dict>
			<key>example.com</key>
			<dict>
				<key>NSExceptionAllowsInsecureHTTPLoads</key> <!-- Noncompliant -->
				<true/>
			</dict>
		</dict>
	</dict>
</dict>
</plist>
----

In addition, a ``++WKNavigationDelegate++`` has to be implemented to allow TLS 1.1 or lower.

[source,swift,diff-id=11,diff-type=noncompliant]
----
import WebKit

class NavigationDelegate: WKNavigationDelegate {
    func webView(_ webView: WKWebView, authenticationChallenge challenge: URLAuthenticationChallenge, shouldAllowDeprecatedTLS decisionHandler: @escaping (Bool) -> Void) {
        decisionHandler(true)  // Noncompliant
    }
}
----

==== Compliant solution

The easiest solution is to remove the ``++NSExceptionAllowsInsecureHTTPLoads++`` key from the App Transport Security configuration.

In addition, it is a good practice to not allow deprecated TLS protocols in the ``++WKNavigationDelegate++``.

[source,swift,diff-id=11,diff-type=compliant]
----
import WebKit

class NavigationDelegate: WKNavigationDelegate {
    func webView(_ webView: WKWebView, authenticationChallenge challenge: URLAuthenticationChallenge, shouldAllowDeprecatedTLS decisionHandler: @escaping (Bool) -> Void) {
        decisionHandler(false)
    }
}
----

=== How does this work?

include::../../common/fix/fix.adoc[]
