== How to fix it in WebKit

=== Code examples

==== Noncompliant code example

[source,swift,diff-id=11,diff-type=noncompliant]
----
import WebKit

class NavigationDelegate: WKNavigationDelegate {
    func webView(_ webView: WKWebView, authenticationChallenge challenge: URLAuthenticationChallenge, shouldAllowDeprecatedTLS decisionHandler: @escaping (Bool) -> Void) {
        decisionHandler(true)  // Noncompliant
    }
}
----

==== Compliant solution

[source,swift,diff-id=11,diff-type=compliant]
----
import WebKit

class NavigationDelegate: WKNavigationDelegate {
    func webView(_ webView: WKWebView, authenticationChallenge challenge: URLAuthenticationChallenge, shouldAllowDeprecatedTLS decisionHandler: @escaping (Bool) -> Void) {
        decisionHandler(false)
    }
}
----

=== How does this work?

include::../../common/fix/fix.adoc[]
