:true: True
:false: False
:ops: Not, And, Or, =
== Why is this an issue?

include::../description.adoc[]

== How to fix it

include::../how-to-fix-it.adoc[]

=== Code examples

==== Noncompliant code example

[source,vb6,diff-id=1,diff-type=noncompliant]
----
If BooleanVariable = True Then ...
If BooleanVariable <> True Then ...
If BooleanVariable OR False Then ...
DoSomething(Not False)
----

==== Compliant solution

[source,vb6,diff-id=1,diff-type=compliant]
----
If BooleanVariable Then ...
If Not BooleanVariable Then ...
If BooleanVariable Then ...
DoSomething(True)
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''

endif::env-github,rspecator-view[]
