== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,vb6]
----
If BooleanVariable = True Then ...
If BooleanVariable <> True Then ...
If BooleanVariable OR False Then ...
DoSomething(Not False)
----

=== Compliant solution

[source,vb6]
----
If BooleanVariable Then ...
If Not BooleanVariable Then ...
If BooleanVariable Then ...
DoSomething(True)
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
