This rule raises an issue when configuration hashes use `nil` as default values, omit required keys, or are incomplete in ways that could cause runtime errors.

== Why is this an issue?

Configuration hashes are a common pattern in Ruby applications, especially in frameworks and libraries. When these configurations are incomplete or use problematic default values, they can lead to several issues:

**Nil Default Values Create Runtime Errors**

Using `nil` as a default value in configuration hashes often leads to `NoMethodError` or `TypeError` exceptions at runtime. When the application tries to use these nil values, it may call methods on nil or pass nil to functions expecting specific types.

**Missing Required Keys Cause Unexpected Behavior**

When configuration hashes omit required keys, the application may fall back to implicit defaults that aren't obvious to developers reading the code. This makes the configuration less self-documenting and can lead to surprising behavior when defaults change between versions.

**Incomplete Configurations Lead to Feature Failures**

Some configurations require multiple related keys to function properly. For example, an ordering configuration might need both visibility settings and action definitions. When these are incomplete, features may partially work or fail silently.

**Poor Developer Experience**

Incomplete or unclear configurations make it harder for developers to understand what options are available and how to properly configure features. This leads to more time spent debugging and consulting documentation.

=== What is the potential impact?

Runtime errors from nil method calls, unexpected application behavior due to implicit defaults, feature malfunctions from incomplete configurations, and increased debugging time for developers working with the codebase.

== How to fix it

Replace nil default values with meaningful defaults or explicit handling. Provide a sensible default value that won't cause runtime errors.

=== Code examples

==== Noncompliant code example

[source,ruby,diff-id=1,diff-type=noncompliant]
----
self.profile_photo = {
  source: nil # Noncompliant
}
----

==== Compliant solution

[source,ruby,diff-id=1,diff-type=compliant]
----
self.profile_photo = {
  source: :default_avatar
}
----

== Resources

=== Documentation

 * Ruby Hash Documentation - https://ruby-doc.org/core/Hash.html[Official Ruby documentation for Hash class and methods]

 * Ruby Style Guide - Hashes - https://rubystyle.guide/#hash-literals[Community Ruby style guide recommendations for hash usage]
