include::../description.adoc[]

== Noncompliant Code Example

With default provided regular expression ``++^[A-Z][a-zA-Z0-9]*$++``:

[source,cpp]
----
enum someEnumeration { // Noncompliant
};
----

== Compliant Solution

[source,cpp]
----
enum SomeEnumeration {
};
----
ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::parameters.adoc[]

endif::env-github,rspecator-view[]
