== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

With default provided regular expression ``++^[A-Z][a-zA-Z0-9]*$++``:

[source,cpp]
----
enum someEnumeration { // Noncompliant
};
----

=== Compliant solution

[source,cpp]
----
enum SomeEnumeration {
};
----
ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

=== Parameters

.format
****

----
^[A-Z][a-zA-Z0-9]*$
----

Regular expression used to check the enumeration names against
****


endif::env-github,rspecator-view[]
