include::../common/description.adoc[]

include::../common/impact.adoc[]

include::../common/fix.adoc[]

=== Code examples

==== Noncompliant code example

[source,cpp,diff-id=1,diff-type=noncompliant]
----
#include <botan/hash.h>
// ...

Botan::secure_vector<uint8_t> f(std::string input){
    std::unique_ptr<Botan::HashFunction> hash(Botan::HashFunction::create("MD5"));  // Noncompliant
    return hash->process(input);
}
----

==== Compliant solution

[source,cpp,diff-id=1,diff-type=compliant]
----
#include <botan/hash.h>
// ...

Botan::secure_vector<uint8_t> f(std::string input){
    std::unique_ptr<Botan::HashFunction> hash(Botan::HashFunction::create("SHA-512"));
    return hash->process(input);
}
----

include::../common/fix-pitfalls.adoc[]

include::../common/resources.adoc[]


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../common/message.adoc[]

include::../common/highlighting.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../common/comments-and-links.adoc[]

endif::env-github,rspecator-view[]
