Shared coding conventions allow teams to collaborate efficiently. This rule checks that type specifiers always appear in the following order:

. ``typedef``
. type name, spelling of built-in types with more than one type-specifier:
## signedness - ``signed`` or ``unsigned``
## last single type-specifier or
##* ``short int``
##* ``long int``
##* ``long long int``
##* ``long double``

Since the positioning of the ``const`` keyword is controversial, this rule does not check it.


== Noncompliant Code Example

----
int typedef T;

double long d;
char unsigned ch;
long signed int i;

----


== Compliant Solution

----
typedef int T;

long double d;
unsigned char ch;
signed long int i;

----


== See

* https://github.com/isocpp/CppCoreGuidelines/blob/036324/CppCoreGuidelines.md#Rl-const[C++ Core Guidelines NL.26] - Use conventional ``const`` notation

