Java 8 adds ``++Comparator.comparing++`` to allow the creation of a single-value comparator to be shorthanded into a single call. This cleaner syntax should be preferred.


*Note* that this rule is automatically disabled when the project's ``++sonar.java.source++`` is lower than ``++8++``.


== Noncompliant Code Example

----
Comparator<Foo> compartor = (foo1, foo2) -> foo.getName().compareTo(foo2.getName());  // Noncompliant
----


== Compliant Solution

----
Comparator<Foo> compartor = Comparator.comparing(Foo::getName);
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

include::highlighting.adoc[]

endif::env-github,rspecator-view[]
