Omitting the data type of an attribute or element leaves it to default to <code>xs:any</code> for elements, and <code>xs:anyType</code> for attributes. However there is no reason for making such a significant design decision implicit..


== Noncompliant Code Example

----
<xs:element name="fruit">
  <xs:complexType>
    <xs:sequence>
      <xs:element name="color"/> <!-- Noncompliant -->
    </xs:sequence>
  </xs:complexType>
</xs:element>
----


== Compliant Solution

----
<xs:element name="fruit">
  <xs:complexType>
    <xs:sequence>
      <xs:element name="color" type="xs:string"/>
    </xs:sequence>
  </xs:complexType>
</xs:element>
----

