include::../description.adoc[]

== Noncompliant Code Example

----
foo: proc options(main);
  declare bar# fixed decimal; /* Noncompliant */
end;
----

== Compliant Solution

----
foo: proc options(main);
  declare bar fixed decimal;  /* Compliant */
end;
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::parameters.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
