== Noncompliant Code Example

[source,text]
----
import pytest
def foo(): return 1 / 0
def bar(): return 42
def test_something():
  with pytest.raises(ZeroDivisionError):
    foo()
    assert bar() == 42  # Noncompliant
----
