<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>In PHPUnit, to test that an exception is thrown in a given piece of code, the <em>expectException*()</em> methods or the <em>@expectedException*</em> annotations can be used. For such a test to succeed, something in the test method has to throw an exception with the awaited properties. Having an assertion at the end of such a test method, means that, if the test succeeds, that assertion was never evaluated because an exception was thrown before getting to that point.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-php" data-lang="php">public function testA() {
    $o = new MyClass();
    $this-&gt;expectException(\Exception::class);
    $o-&gt;doSomething();
    $this-&gt;assertTrue($o-&gt;hasProperty()); // Noncompliant - This assertion might never get evaluated
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-php" data-lang="php">public function testA() {
    $o = new MyClass();
    $this-&gt;expectException(\Exception::class);
    $o-&gt;doSomething();
}

public function testB() {
    $o = new MyClass();
    $this-&gt;assertTrue($o-&gt;hasProperty());
}</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>If the with block contains multiple statements: Don&#8217;t perform an assertion here; An exception is expected to be raised before its execution.</p>
</div>
<div class="paragraph">
<p>If the assert is the only statement: Refactor this test; if this assertion&#8217;s argument raises an exception, the assertion will never get executed.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>Primary: the assert keyword.</p>
</div>
<div class="paragraph">
<p>Secondary: the with raises / assertRaises statement.</p>
</div>
</div>
</div>
</div>