<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Using <code>pytest.raises</code> or <code>unittest.TestCase.assertRaises</code> will assert that an exception is raised in the following block. Ending such block in an assertion means that the test can succeed with that last assertion never being executed.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">import pytest
def foo(): return 1 / 0
def bar(): return 42
def test_something():
  with pytest.raises(ZeroDivisionError):
    foo()
    assert bar() == 42  # Noncompliant</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">import pytest
def foo(): return 1 / 0
def bar(): return 42
def test_something():
  with pytest.raises(ZeroDivisionError):
    foo()
  assert bar() == 42</code></pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p><a href="https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertRaises">Unittest: assertRaises</a></p>
</li>
<li>
<p><a href="https://docs.pytest.org/en/stable/assert.html#assertions-about-expected-exceptions">Pytest: assertions about expected exceptions</a></p>
</li>
</ul>
</div>
<hr>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>If the with block contains multiple statements: Don&#8217;t perform an assertion here; An exception is expected to be raised before its execution.</p>
</div>
<div class="paragraph">
<p>If the assert is the only statement: Refactor this test; if this assertion&#8217;s argument raises an exception, the assertion will never get executed.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>Primary: the assert keyword.</p>
</div>
<div class="paragraph">
<p>Secondary: the with raises / assertRaises statement.</p>
</div>
</div>
</div>
</div>