include::../why.adoc[]

Note that this rule does not take reflection into account, which means that issues will be raised on `private` methods that are only accessed using the reflection API.

=== Exceptions

This rule doesn't raise issues for:

* annotated methods 
* methods with parameters that are annotated with `@javax.enterprise.event.Observes` 

=== Code examples

=== Noncompliant code example

[source,java,diff-id=1,diff-type=noncompliant]
----
public class Foo implements Serializable
{
  public static void doSomething() {
    Foo foo = new Foo();
    ...
  }

  private void unusedPrivateMethod() {...}
  private void writeObject(ObjectOutputStream s) {...}  //Compliant, relates to the java serialization mechanism
  private void readObject(ObjectInputStream in) {...}  //Compliant, relates to the java serialization mechanism
}
----

=== Compliant solution

[source,java,diff-id=1,diff-type=compliant]
----
public class Foo implements Serializable
{
  public static void doSomething(){
    Foo foo = new Foo();
    ...
  }

  private void writeObject(ObjectOutputStream s) {...}  //Compliant, relates to the java serialization mechanism
  private void readObject(ObjectInputStream in) {...}  //Compliant, relates to the java serialization mechanism
}
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
