=== Noncompliant code example

[source,text]
----
function foo() {
  let x = bar(1);
  if (x > 0) {
      let x = bar(2); // Noncompliant
      console.log(x);
  } else {
     console.log("Wrong Value");
  }
}
----
