include::../description.adoc[]

== Noncompliant Code Example

----
int numberOfMinutes(int hours) {
  int seconds = 0; // Noncompliant, never used
  return hours * 60;
}
----

== Compliant Solution

----
int numberOfMinutes(int hours) {
  return hours * 60;
}
----

== Exceptions

No issue is raised on local variables having the attribute "unused" and on object declarations with non-empty arguments.
