include::../rationale.adoc[]

=== Exceptions

No issue is raised on local variables with the attribute ``++[[maybe_unused]]++`` or `unused` and object declarations with non-empty arguments.

include::../how-to-fix-it.adoc[]

=== Code examples

==== Noncompliant code example

[source,cpp,diff-id=1,diff-type=noncompliant]
----
int numberOfMinutes(int hours) {
  int seconds = 0; // Noncompliant second is unused
  return hours * 60;
}
----

==== Compliant solution

[source,cpp,diff-id=1,diff-type=compliant]
----
int numberOfMinutes(int hours) {
  return hours * 60;
}
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
