include::../description.adoc[]

== Noncompliant Code Example

----
int numberOfMinutes(int hours) {
  int seconds = 0; // Noncompliant, never used
  return hours * 60;
}
----

== Compliant Solution

----
int numberOfMinutes(int hours) {
  return hours * 60;
}
----

== Exceptions

No issue is raised on local variables having the attribute "unused" and on object declarations with non-empty arguments.

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
