Using a ``++<li>++`` or ``++<dt>++`` item tag outside of a ``++<ul>++``, ``++<ol>++`` or ``++<dl>++`` one does not make sense and indicates a bug.


== Noncompliant Code Example

[source,html]
----
<li>Apple</li>          <!-- Noncompliant -->
<li>Strawberry</li>     <!-- Noncompliant -->

<li>Apple</li>          <!-- Noncompliant -->
<li>Strawberry</li>     <!-- Noncompliant -->

<dt>Apple</dt>          <!-- Noncompliant -->
<dt>Strawberry</dt>     <!-- Noncompliant -->
----


== Compliant Solution

[source,html]
----
<ul>
  <li>Apple</li>
  <li>Strawberry</li>
</ul>

<ol>
  <li>Apple</li>
  <li>Strawberry</li>
</ol>

<dl>
  <dt>Apple</dt>
  <dt>Strawberry</dt>
</dl>
----



ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
