== Why is this an issue?

Using a ``++<li>++`` or ``++<dt>++`` item tag outside of a ``++<ul>++``, ``++<ol>++`` or ``++<dl>++`` one does not make sense and indicates a bug.


=== Noncompliant code example

[source,html]
----
<li>Apple</li>          <!-- Noncompliant -->
<li>Strawberry</li>     <!-- Noncompliant -->

<li>Apple</li>          <!-- Noncompliant -->
<li>Strawberry</li>     <!-- Noncompliant -->

<dt>Apple</dt>          <!-- Noncompliant -->
<dt>Strawberry</dt>     <!-- Noncompliant -->
----


=== Compliant solution

[source,html]
----
<ul>
  <li>Apple</li>
  <li>Strawberry</li>
</ul>

<ol>
  <li>Apple</li>
  <li>Strawberry</li>
</ol>

<dl>
  <dt>Apple</dt>
  <dt>Strawberry</dt>
</dl>
----



ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Surround this {0} item tag by a {1} or {2} container one.


'''
== Comments And Links
(visible only on this page)

=== on 21 Jun 2013, 08:59:48 Dinesh Bolkensteyn wrote:
----
<?php
echo "<ul>\n";
foreach ($array as $element) {
  <li><?php echo $element; ?></li>
}
echo "</ul>\n";
?>
----

should be refactored into:


----
<ul>
<?php
foreach ($array as $element) {
?>
  <li><?php echo $element; ?></li>
<?php
}
?>
</ul>
----

or:



----
<?php
echo "<ul>\n";
foreach ($array as $element) {
  echo "  <li>$element</li>\n";
}
echo "</ul>\n";
?>
----

=== on 21 Jun 2013, 09:28:59 Dinesh Bolkensteyn wrote:
We should be careful with nested lists....

=== on 8 Jul 2013, 18:21:30 Freddy Mallet wrote:
Is implemented by \http://jira.codehaus.org/browse/SONARPLUGINS-2993

endif::env-github,rspecator-view[]
