<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Why use named groups only to never use any of them later on in the code?</p>
</div>
<div class="paragraph">
<p>This rule raises issues every time named groups are:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>referenced while not defined;</p>
</li>
<li>
<p>defined but called elsewhere in the code by their number instead.</p>
</li>
</ul>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">import re

def foo():
    pattern = re.compile(r"(?P&lt;a&gt;.)")
    matches = pattern.match("abc")
    g1 = matches.group("b") # Noncompliant - group "b" is not defined
    g2 = matches.group(1) # Noncompliant - Directly use 'a' instead of its group number.</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">import re

def foo():
    pattern = re.compile(r"(?P&lt;a&gt;.)")
    matches = pattern.match("abc")
    g = matches.group("a")</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="ulist">
<ul>
<li>
<p>In case there is no group with name:</p>
<div class="ulist">
<ul>
<li>
<p>There is no group named 'S' in the regular expression.</p>
</li>
</ul>
</div>
</li>
<li>
<p>In case number access to named group</p>
<div class="ulist">
<ul>
<li>
<p>Directly use 'S' instead of its group number.</p>
</li>
</ul>
</div>
</li>
</ul>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="ulist">
<ul>
<li>
<p>primary: the group number or group name</p>
<div class="ulist">
<ul>
<li>
<p>secondary: named groups</p>
</li>
</ul>
</div>
</li>
</ul>
</div>
</div>
</div>
</div>