== Why is this an issue?

Heading tags are used by search engines and screen reader softwares to construct an outline of the page.

Starting at ``++<h1>++`` and not skipping any level eases this automatic construction.


=== Noncompliant code example

[source,html]
----
<h2>My Title</h2>     <!-- Non-Compliant - h1 is not being used -->

<h3>My Sub Title</h3> <!-- Compliant -->
----


=== Compliant solution

[source,html]
----
<h1>My Title</h1>     <!-- Compliant -->

<h2>My Sub Title</h2> <!-- Compliant -->
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
