<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Call a non-callable symbol will result in a TypeError at execution.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-javascript" data-lang="javascript">function foo(a) {
  return a * 2;
}
// ....
var foo = 1;
// ...
foo();   // foo is 1 =&gt; TypeError</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_duplicates_s2873">duplicates: <a data-rspec-id="S2873" class="rspec-auto-link">S2873</a></h3>

</div>
<div class="sect2">
<h3 id="_on_27_apr_2015_132708_linda_martin_wrote">on 27 Apr 2015, 13:27:08 Linda Martin wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] Assign for completion.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_28_apr_2015_142856_ann_campbell_wrote">on 28 Apr 2015, 14:28:56 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~linda.martin] isn&#8217;t <a data-rspec-id="S2873" class="rspec-auto-link">RSPEC-2873</a> a sub-type of this rule?</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_29_apr_2015_090709_linda_martin_wrote">on 29 Apr 2015, 09:07:09 Linda Martin wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] I have a doubt, maybe we could have a rule checking "unchecked TypeError", or "inappropriate property access or method call" ? (I would prefer the last one) So both can be merged. I agree that it seems weird to have 2 different rules.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_29_apr_2015_125448_ann_campbell_wrote">on 29 Apr 2015, 12:54:48 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~linda.martin], I&#8217;ve updated <a data-rspec-id="S2873" class="rspec-auto-link">RSPEC-2873</a>, and am closing this as a duplicate.</p>
</div>
</div>
</div>
</div>