== How to fix it in FasterXML

=== Code examples

include::../../common/fix/code-rationale.adoc[]

==== Noncompliant code example

[source,kotlin,diff-id=1,diff-type=noncompliant]
----
import com.fasterxml.jackson.databind.ObjectMapper

//[...]
fun doGet(request: HttpServletRequest, response: HttpServletResponse) {
  val tainted = request.getParameter("value")
  val json = "{\"key\":\"$tainted\"}"
  val obj = ObjectMapper()
  try {
    obj.readValue(json, Any::class.java) // Noncompliant
  } catch (e: JsonProcessingException) {
    response.sendError(400)
  }
}
----

==== Compliant solution

[source,kotlin,diff-id=1,diff-type=compliant]
----
import com.fasterxml.jackson.databind.ObjectMapper;
import com.fasterxml.jackson.databind.node.ObjectNode;

//[...]
fun doGet(request: HttpServletRequest, response: HttpServletResponse) {
  val obj = ObjectMapper()
  val myObject = obj.createObjectNode()
  myObject.put("key", request.getParameter("value"))
}
----

=== How does this work?

include::../../common/fix/introduction.adoc[]

include::../../common/fix/object.adoc[]

The example compliant code uses the `ObjectNode` class from the Jackson
library to directly build a JSON object from the user input.

include::../../common/fix/casting.adoc[]
