== How to fix it in Gson

=== Code examples

include::../../common/fix/code-rationale.adoc[]

==== Noncompliant code example

[source,kotlin,diff-id=11,diff-type=noncompliant]
----
import com.google.gson.Gson;

//[...]
fun doGet(request: HttpServletRequest, response: HttpServletResponse) {
  val tainted = request.getParameter("value")
  val json = "{\"key\":\"$tainted\"}"
  val obj = Gson()
  obj.fromJson(json, Any::class.java) // Noncompliant
}
----

==== Compliant solution

[source,kotlin,diff-id=11,diff-type=compliant]
----
import com.google.gson.Gson;
import com.google.gson.JsonObject;
import com.google.gson.JsonSyntaxException;

//[...]
fun doGet(request: HttpServletRequest, response: HttpServletResponse) {
  try {
    val json: JsonObject = JsonObject()
    json.addProperty("key", request.getParameter("value"))
    val obj = Gson()
    obj.fromJson(json, Any::class.java)
  } catch (e: JsonSyntaxException) {
    response.sendError(400)
  }
}
----

=== How does this work?

include::../../common/fix/introduction.adoc[]

include::../../common/fix/object.adoc[]

The example compliant code uses the `JsonObject` class from the Gson library
to directly build an object from the user input.

include::../../common/fix/casting.adoc[]
