== How to fix it in Java SE

=== Code examples

include::../../common/fix/code-rationale.adoc[]

==== Noncompliant code example

[source,kotlin,diff-id=21,diff-type=noncompliant]
----
import org.json.JSONObject;

//[...]
fun doGet(request: HttpServletRequest, response: HttpServletResponse) {
  try {
    val tainted = request.getParameter("value")
    val json = "{\"key\":\"$tainted\"}"
    val obj = JSONObject(json) // Noncompliant
  } catch (e: JsonException) {
    response.sendError(400)
  }
}
----

==== Compliant solution

[source,kotlin,diff-id=21,diff-type=compliant]
----
import org.json.JSONObject;

//[...]
fun doGet(request: HttpServletRequest, response: HttpServletResponse) {
  val obj = JSONObject()
  obj.put("key", request.getParameter("value"))
}
----

=== How does this work?

include::../../common/fix/introduction.adoc[]

include::../../common/fix/object.adoc[]

The example compliant code uses the `org.json` libraries capabilities to
dynamically build a JSON object without string parsing.

include::../../common/fix/casting.adoc[]
