== How to fix it in Spring

=== Code examples

include::../../common/fix/code-rationale.adoc[]

In this example a json string is parsed using the `BasicJsonParser` class from the `org.springframework.boot.json` package. This class is used to parse JSON strings into a `Map` object. Since the attacker can control the json string they can control the keys and values of the `Map` object.

==== Noncompliant code example

[source,kotlin,diff-id=31,diff-type=noncompliant]
----
import org.springframework.boot.json.BasicJsonParser;

//[...]
@GetMapping("/example")
fun example(@RequestParam input: String): Map<String, Any> {
  val parser = BasicJsonParser()
  val json = "{\"key\":\"$input\"}"
  return parser.parseMap(json) // Noncompliant
}
----

==== Compliant solution

[source,kotlin,diff-id=31,diff-type=compliant]
----
@GetMapping("/example")
fun example(@RequestParam input: String): Map<String, Any> {
  return mapOf("key" to input)
}
----

=== How does this work?

include::../../common/fix/introduction.adoc[]

include::../../common/fix/object.adoc[]

The example compliant code skips the use of a JSON parser and directly adds the user controlled data into a `Map` object. This way the attacker cannot control the keys and values of the `Map` object.

include::../../common/fix/casting.adoc[]
