include::../rule.adoc[]

== Noncompliant Code Example

For https://docs.aws.amazon.com/opensearch-service/index.html[Amazon OpenSearch domains]:

----
resource "aws_elasticsearch_domain" "example" {
  domain_name = "example"
  domain_endpoint_options {
    enforce_https = false # Noncompliant
  }
  node_to_node_encryption {
    enabled = false # Noncompliant
  }
}
----

== Compliant Solution

For https://docs.aws.amazon.com/opensearch-service/index.html[Amazon OpenSearch domains]:

----
resource "aws_elasticsearch_domain" "example" {
  domain_name = "example"
  domain_endpoint_options {
    enforce_https = true
  }
  node_to_node_encryption {
    enabled = true
  }
}
----

include::../see.adoc[]


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::highlighting.adoc[]

endif::env-github,rspecator-view[]