include::../rule.adoc[]

== Noncompliant Code Example

----
resource "aws_elasticsearch_domain" "example" {
  domain_name = "example"
  node_to_node_encryption {
    enabled = false # Noncompliant
  }
}
----

== Compliant Solution

----
resource "aws_elasticsearch_domain" "example" {
  domain_name = "example"
  node_to_node_encryption {
    enabled = true
  }
}
----

include::../see.adoc[]


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::highlighting.adoc[]

endif::env-github,rspecator-view[]