<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This is an issue when a class file is located in a subdirectory but the class is not wrapped in the corresponding namespace module that matches the directory structure.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Ruby&#8217;s autoloading mechanism expects the file structure to match the namespace structure. When this convention is not followed, several problems can occur:</p>
</div>
<div class="paragraph">
<p><strong>Autoloading failures</strong>: Ruby&#8217;s autoloader may fail to properly load the class, leading to <code>NameError</code> or <code>LoadError</code> exceptions at runtime.</p>
</div>
<div class="paragraph">
<p><strong>Circular dependency errors</strong>: When a class in a subdirectory references other classes in the same namespace without proper module wrapping, it can create circular dependencies. The autoloader gets confused about which file to load first, resulting in <code>RuntimeError: Circular dependency detected while autoloading constant</code> errors.</p>
</div>
<div class="paragraph">
<p><strong>Inconsistent behavior</strong>: The application may work in some environments (like production with eager loading) but fail in development mode where autoloading is more active.</p>
</div>
<div class="paragraph">
<p><strong>Maintenance issues</strong>: Code becomes harder to understand and maintain when the file structure doesn&#8217;t match the namespace structure, violating the principle of least surprise.</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="paragraph">
<p>Applications may experience runtime errors, particularly <code>RuntimeError: Circular dependency detected while autoloading constant</code> exceptions. This can cause complete application failures in development environments and unpredictable behavior during deployment. The issue is especially problematic in Rails applications where autoloading is heavily used.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Wrap the class in the appropriate module that matches the directory structure. The module name should correspond to the subdirectory name.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-ruby" data-lang="ruby"># app/controllers/admin/users_controller.rb
class UsersController &lt; Admin::BaseController # Noncompliant
  def index
    # action code
  end
end</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-ruby" data-lang="ruby"># app/controllers/admin/users_controller.rb
module Admin
  class UsersController &lt; Admin::BaseController
    def index
      # action code
    end
  end
end</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it_in_rails">How to fix it in Rails</h2>
<div class="sectionbody">
<div class="paragraph">
<p>For Rails controllers in namespaced routes, ensure the controller class is properly wrapped in the namespace module to match the directory structure and route namespace.</p>
</div>
<div class="sect2">
<h3 id="_code_examples_2">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example_2">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-ruby" data-lang="ruby"># app/controllers/api/v1/users_controller.rb
class UsersController &lt; ApplicationController # Noncompliant
  def show
    render json: User.find(params[:id])
  end
end</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution_2">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-ruby" data-lang="ruby"># app/controllers/api/v1/users_controller.rb
module Api
  module V1
    class UsersController &lt; ApplicationController
      def show
        render json: User.find(params[:id])
      end
    end
  end
end</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Ruby Autoloading and Reloading Constants - <a href="https://guides.rubyonrails.org/autoloading_and_reloading_constants.html">Official Rails guide explaining autoloading mechanisms and best practices</a></p>
</li>
<li>
<p>Ruby Module and Class Naming Conventions - <a href="https://ruby-doc.org/core/doc/syntax/modules_and_classes_rdoc.html">Ruby documentation on module and class naming conventions</a></p>
</li>
</ul>
</div>
</div>
<div class="sect2">
<h3 id="_standards">Standards</h3>
<div class="ulist">
<ul>
<li>
<p>Ruby Style Guide - Modules - <a href="https://rubystyle.guide/#modules">Community Ruby style guide recommendations for module usage</a></p>
</li>
</ul>
</div>
</div>
<div class="sect2">
<h3 id="_related_rules">Related rules</h3>
<div class="ulist">
<ul>
<li>
<p><a data-rspec-id="S3903" class="rspec-auto-link">RSPEC-3903</a> - <a href="https://rules.sonarsource.com/vbnet/RSPEC-3903">VB.NET rule for proper namespace declaration</a></p>
</li>
<li>
<p><a data-rspec-id="S3259" class="rspec-auto-link">RSPEC-3259</a> - <a href="https://rules.sonarsource.com/csharp/RSPEC-3259">C# rule for proper namespace declaration</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>