<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Both the name and the field definitions of a parameter 2 data area used in an <code>IN</code> should be coded in the <code>D</code> specification. This ensures that all definition data is kept together, and the specific data is not manually extracted in the code, thus making the code easier to read and maintain.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-rpg" data-lang="rpg">D WSFD08          S             30A 0

C                   IN        WSFD08
C                   MOVEL     WSFD08        VAR9S</code></pre>
</div>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-rpg" data-lang="rpg">D WSFD08          S             30A 0

 /free
   in  WSFD08;
 /end-free</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-rpg" data-lang="rpg">D WSFD08          DS    30             DTAARA(HEPDTAR005)
D  W3SPNO               123    124B 0
D  W3LNCT               367    368B 0

C                   IN        WSFD08</code></pre>
</div>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-rpg" data-lang="rpg">D WSFD08          DS    30             DTAARA(HEPDTAR005)
D  W3SPNO               123    124B 0
D  W3LNCT               367    368B 0

 /free
   in  WSFD08;
 /end-free</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Define the data structure for this "IN" statement in its D spec</p>
</div>
</div>
</div>
</div>