<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This is an issue when code attempts to directly convert between <code>LocalDate</code> and <code>Instant</code> without specifying timezone information, which will cause a <code>DateTimeException</code> at runtime.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p><code>LocalDate</code> represents a date without time or timezone (like "2023-12-25"), while <code>Instant</code> represents a specific point in time on the UTC timeline.</p>
</div>
<div class="paragraph">
<p>Direct conversions fail because converting <code>LocalDate</code> to <code>Instant</code> requires time and timezone information, while converting <code>Instant</code> to <code>LocalDate</code> requires timezone context. Java throws a <code>DateTimeException</code> because it cannot assume missing temporal information.</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="paragraph">
<p>This causes runtime failures with <code>DateTimeException</code>, leading to application crashes. It&#8217;s particularly problematic in production where timezone bugs may only surface in different geographical regions.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Convert LocalDate to Instant via LocalDateTime and specify timezone.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">LocalDate date = LocalDate.of(2023, 12, 25);
Instant instant = Instant.from(date); // Noncompliant - throws DateTimeException</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">LocalDate date = LocalDate.of(2023, 12, 25);
Instant instant = date.atStartOfDay(ZoneId.systemDefault()).toInstant();
// or with explicit timezone
Instant instant2 = date.atStartOfDay().toInstant(ZoneOffset.UTC);</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>LocalDate JavaDoc - <a href="https://docs.oracle.com/javase/8/docs/api/java/time/LocalDate.html">Official documentation for LocalDate class</a></p>
</li>
<li>
<p>Instant JavaDoc - <a href="https://docs.oracle.com/javase/8/docs/api/java/time/Instant.html">Official documentation for Instant class</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>