<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>When you need to perform a complicated initialization of a <code>static</code> member, it should be done in a <code>static</code> initializer block. That&#8217;s because such blocks are only executed when the class is loaded into the JVM. That is, they run only once, and that happens before any instances are created. Non-<code>static</code> blocks, on the other hand, run once for each instance that&#8217;s created, so any <code>static</code> members "initialized" in such a block will be re-set for each new instance.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">public class MyClass {
  private static List&lt;String&gt; names = new ArrayList&lt;&gt;();

  {
    names.add("foo");  // Noncompliant
  }</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">public class MyClass {
  private static List&lt;String&gt; names = new ArrayList&lt;&gt;();

  static {
    names.add("foo");
  }</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_duplicates_s1171">duplicates: <a data-rspec-id="S1171" class="rspec-auto-link">S1171</a></h3>

</div>
<div class="sect2">
<h3 id="_on_14_dec_2015_104639_nicolas_peru_wrote">on 14 Dec 2015, 10:46:39 Nicolas Peru wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] The rule examples do not match the title and descritption :</p>
</div>
<div class="paragraph">
<p>a static initiliazer has the following form :</p>
</div>
<div class="listingblock">
<div class="content">
<pre>static {
   //some code
}</pre>
</div>
</div>
<div class="paragraph">
<p>if you use this on an instance variable, code does not compile.</p>
</div>
<div class="paragraph">
<p>However the other way around will compile and is probably a mistake (using a non static initializer on a static variable) :</p>
</div>
<div class="listingblock">
<div class="content">
<pre>class A {
public static List&lt;String&gt; foo = new ArrayList&lt;&gt;();

{
  foo.add("foo");
}
}</pre>
</div>
</div>
<div class="paragraph">
<p>As this code will be executed for every instanciation of A.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_14_dec_2015_200017_ann_campbell_wrote">on 14 Dec 2015, 20:00:17 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>Thanks [~nicolas.peru]. Look now.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_15_dec_2015_141041_nicolas_peru_wrote">on 15 Dec 2015, 14:10:41 Nicolas Peru wrote:</h3>
<div class="paragraph">
<p>LGTM. Moving issue to java top as this is easy to achieve.</p>
</div>
</div>
</div>
</div>