== How to fix it in PyJWT

=== Code examples

include::../../common/fix/code-rationale.adoc[]

==== Noncompliant code example

[source,python,diff-id=1,diff-type=noncompliant]
----
import jwt

jwt.decode(token, verify=False) # Noncompliant
----

==== Compliant solution

By default, verification is enabled for the method `decode`.

[source,python,diff-id=1,diff-type=compliant]
----
import jwt

jwt.decode(token, key, algorithm)
----

=== How does this work?

include::../../common/fix/verification.adoc[]

