== How to fix it in python-jwt

=== Code examples

include::../../common/fix/code-rationale.adoc[]

==== Noncompliant code example

[source,python,diff-id=1,diff-type=noncompliant]
----
import python_jwt as jwt

jwt.process_jwt(token) # Noncompliant
----

==== Compliant solution

A call to the method `process_jwt` should be followed by a call to `verify_jwt`
to verify the signature of the token.

[source,python,diff-id=1,diff-type=compliant]
----
import python_jwt as jwt

jwt.process_jwt(token)
jwt.verify_jwt(token, key, algo)
----

=== How does this work?

include::../../common/fix/signature.adoc[]

=== Going the extra mile

include::../../common/extra-mile/key-storage.adoc[]

include::../../common/extra-mile/key-rotation.adoc[]

