//// This rule is superseded by RSPEC-2583
//// Please consider implementing this latter instead.
include::../description.adoc[]

== Noncompliant Code Example

----
  if (true) {
    doSomething();
  }
  // ...
  if (false) {
    doSomethingElse();
  }
----

== Compliant Solution

----
  doSomething();
  // ...
----

include::../see.adoc[]

ifdef::env-github,rspecator-view[]
'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
