Deprecated features are those that have been retained temporarily for backward compatibility, but which will eventually be removed. In effect, deprecation announces a grace period to allow the smooth transition from the old features to the new ones. In that period, no use of the deprecated features should be added, and all existing uses should be gradually removed.


This rule raises an issue when ``++${pom.*}++`` properties are used in a pom.


== Noncompliant Code Example

----
  <build>
    <finalName>${pom.artifactId}-${pom.version}</finalName>  <!-- Noncompliant -->
----


== Compliant Solution

----
  <build>
    <finalName>${project.artifactId}-${project.version}</finalName>
----
or

----
  <build>
    <finalName>${artifactId}-${version}</finalName>
----



ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

include::highlighting.adoc[]

'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
