<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>In C&#43;&#43;, the literal 0 is both an integer type and the null-pointer-constant. To meet developer expectations, <code>NULL</code> should be used as the null-pointer-constant, and 0 for the integer zero.</p>
</div>
<div class="paragraph">
<p>Note: as a result of this rule, <code>NULL</code> is considered to have pointer type.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">#include &lt;cstddef&gt;

void f1 ( int32_t );
void f2 ( int32_t * );
void f3 ( )
{
  f1 ( NULL ); // Noncompliant, "NULL" used as an integer
  f2 ( NULL ); // Compliant, pointer expected
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">#include &lt;cstddef&gt;

void f1 ( int32_t );
void f2 ( int32_t * );
void f3 ( )
{
  f1 ( 0 ); // Compliant, integer expected
}</code></pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p>MISRA C&#43;&#43;:2008, 4-10-1</p>
</li>
</ul>
</div>
<hr>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Replace this use of "NULL" with the integer constant '0'.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_relates_to_s949">relates to: <a data-rspec-id="S949" class="rspec-auto-link">S949</a></h3>

</div>
<div class="sect2">
<h3 id="_on_16_oct_2014_130235_ann_campbell_wrote">on 16 Oct 2014, 13:02:35 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~samuel.mercier] please:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>fill in the appropriate reference field(s).</p>
</li>
<li>
<p>provide a See section.</p>
</li>
<li>
<p>use double curly braces around code in the description, e.g. NULL.</p>
</li>
<li>
<p>use the standard section titles: Noncompliant Code Example, Exception*s*</p>
</li>
</ul>
</div>
</div>
<div class="sect2">
<h3 id="_on_17_oct_2014_154200_ann_campbell_wrote">on 17 Oct 2014, 15:42:00 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~freddy.mallet] this rule an <a data-rspec-id="S949" class="rspec-auto-link">RSPEC-949</a> are 2 sides of the same coin</p>
</div>
</div>
</div>
</div>