In {cpp}, the literal 0 is both an integer type and the null-pointer-constant. To meet developer expectations, ``NULL`` should be used as the null-pointer-constant, and 0 for the integer zero.
Note: as a result of this rule, ``NULL`` is considered to have pointer type.


== Noncompliant Code Example

----
#include <cstddef>

void f1 ( int32_t );
void f2 ( int32_t * );
void f3 ( )
{
  f1 ( NULL ); // Noncompliant, "NULL" used as an integer
  f2 ( NULL ); // Compliant, pointer expected
}
----


== Compliant Solution

----
#include <cstddef>

void f1 ( int32_t );
void f2 ( int32_t * );
void f3 ( )
{
  f1 ( 0 ); // Compliant, integer expected
}
----


== See

* MISRA {cpp}:2008, 4-10-1

