== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,csharp]
----
int i = 0;
while (true)
{
  if (i == 10)
  {
    break;      // Non-Compliant
  }

  Console.WriteLine(i);
  i++;
}
----

=== Compliant solution

[source,csharp]
----
int i = 0;
while (i != 10) // Compliant
{
  Console.WriteLine(i);
  i++;
}
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
