<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>The <code>[[nodiscard]]</code> attribute can be placed on type declarations to indicate that any value of such type should not be discarded when returned from a function. Accompanying the attribute with the message, explaining why values should not be ignored, contributes to a better understanding of code. This is especially important in the case of types, as the reason for which values of the type should not be discarded is a defining property of that type (information about failure, handle owning a scarce resource).</p>
</div>
<div class="paragraph">
<p>Moreover, marking a type as <code>nodiscard</code>, causes a warning to be reported for invocation of every function that returns this type. As a consequence, the cause of the warning is not directly visible from the declaration of the function and requires further investigation from the user.</p>
</div>
<div class="paragraph">
<p>This rule raises an issue when <code>[[nodiscard]]</code> is used on a type without any explanation.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">struct [[nodiscard]] status_code {  // Noncompliant
   int code();
};

status_code open(std::string_view path);

int main()
{
   open("/home/user/list.txt");  // warning is raised here
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">struct [[nodiscard("Possible errors should be checked")]] status_code {
   int code();
};</code></pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p><a data-rspec-id="S6166" class="rspec-auto-link">S6166</a> - introducing messages for <code>[[nodiscard]]</code> on functions</p>
</li>
<li>
<p><a data-rspec-id="S6007" class="rspec-auto-link">S6007</a> - marking functions as <code>[[nodicard]]</code></p>
</li>
</ul>
</div>
<hr>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_relates_to_s6166">relates to: <a data-rspec-id="S6166" class="rspec-auto-link">S6166</a></h3>

</div>
</div>
</div>