== Why is this an issue?

A cache is a long-lived object that holds references to shorter-lived objects. To prevent a cache from growing indefinitely, old objects the cache should be removed when they're no longer used. This can be done with the use of soft references, which allow the garbage collector to remove unused cache entries when memory needs to be freed.


This rule raises an issue when a ``++static++`` collection does not use ``++SoftReference++``.


*Note* that this rule is automatically disabled when the project's ``++sonar.java.source++`` is lower than 7.


=== Noncompliant code example

[source,java]
----
public class MyClass {
  private static List<MyClass> cache = new ArrayList<>();  // Noncompliant
----


=== Compliant solution

[source,java]
----
public class MyClass {
  private static List<SoftReference<MyClass>> cache = new ArrayList<>();
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Add a maximum size to this "xxx" or use soft references.


=== Highlighting

cache declaration


endif::env-github,rspecator-view[]
