<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>In the past, it was required to load a JDBC driver before creating a <code>java.sql.Connection</code>. Nowadays, when using JDBC 4.0 drivers, this is no longer required and <code>Class.forName()</code> can be safely removed because JDBC 4.0 (JDK 6) drivers available in the classpath are automatically loaded.</p>
</div>
<div class="paragraph">
<p>This rule raises an issue when <code>Class.forName()</code> is used with one of the following values:</p>
</div>
<div class="ulist">
<ul>
<li>
<p><code>com.mysql.jdbc.Driver</code></p>
</li>
<li>
<p><code>oracle.jdbc.driver.OracleDriver</code></p>
</li>
<li>
<p><code>com.ibm.db2.jdbc.app.DB2Driver</code></p>
</li>
<li>
<p><code>com.ibm.db2.jdbc.net.DB2Driver</code></p>
</li>
<li>
<p><code>com.sybase.jdbc.SybDriver</code></p>
</li>
<li>
<p><code>com.sybase.jdbc2.jdbc.SybDriver</code></p>
</li>
<li>
<p><code>com.teradata.jdbc.TeraDriver</code></p>
</li>
<li>
<p><code>com.microsoft.sqlserver.jdbc.SQLServerDriver</code></p>
</li>
<li>
<p><code>org.postgresql.Driver</code></p>
</li>
<li>
<p><code>sun.jdbc.odbc.JdbcOdbcDriver</code></p>
</li>
<li>
<p><code>org.hsqldb.jdbc.JDBCDriver</code></p>
</li>
<li>
<p><code>org.h2.Driver</code></p>
</li>
<li>
<p><code>org.firebirdsql.jdbc.FBDriver</code></p>
</li>
<li>
<p><code>net.sourceforge.jtds.jdbc.Driver</code></p>
</li>
<li>
<p><code>com.ibm.db2.jcc.DB2Driver</code></p>
</li>
</ul>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">import java.sql.Connection;
import java.sql.DriverManager;
import java.sql.SQLException;

public class Demo {
  private static final String DRIVER_CLASS_NAME = "org.postgresql.Driver";
  private final Connection connection;

  public Demo(String serverURI) throws SQLException, ClassNotFoundException {
    Class.forName(DRIVER_CLASS_NAME); // Noncompliant; no longer required to load the JDBC Driver using Class.forName()
    connection = DriverManager.getConnection(serverURI);
  }
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">import java.sql.Connection;
import java.sql.DriverManager;
import java.sql.SQLException;

public class Demo {
    private final Connection connection;

    public Demo(String serverURI) throws SQLException {
        connection = DriverManager.getConnection(serverURI);
    }
}</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove this "Class.forName()", it is useless.</p>
</div>
</div>
</div>
</div>