<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when you use class-level update methods like <code>Model.update(id, attributes)</code> while you already have an instance of that model available.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>When you use class-level update methods like <code>User.update(user.id, attributes)</code>, Rails creates a new instance of the model to perform the update. This means your existing instance variable remains unchanged in memory, even though the database record is updated successfully.</p>
</div>
<div class="paragraph">
<p>This behavior can lead to subtle bugs where your code continues to work with stale data. For example, if you check the updated attributes on your original instance after the class-level update, you&#8217;ll see the old values, not the new ones from the database.</p>
</div>
<div class="paragraph">
<p>Instance-level methods like <code>user.update(attributes)</code> update both the database record and the in-memory object, ensuring consistency between what&#8217;s stored and what your code is working with.</p>
</div>
<div class="paragraph">
<p>Using instance methods when you already have an instance also makes your code more readable and expresses your intent more clearly - you want to update this specific object, not just any record with this ID.</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="paragraph">
<p>This issue can cause bugs where your application logic operates on outdated data, leading to incorrect behavior, inconsistent user interfaces, or failed validations that should have passed. While the database is updated correctly, any subsequent operations on the original instance will use stale data.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it_in_rails">How to fix it in Rails</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Replace the class-level update method with the instance-level update method. This ensures both the database and your in-memory object are updated consistently.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-ruby" data-lang="ruby">def update_profile
  @user = User.find(params[:id])
  User.update(@user.id, user_params) # Noncompliant
  # @user still has old values in memory
  redirect_to @user
end</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-ruby" data-lang="ruby">def update_profile
  @user = User.find(params[:id])
  @user.update(user_params)
  # @user now reflects the updated values
  redirect_to @user
end</code></pre>
</div>
</div>
<div class="paragraph">
<p>When working with multiple attributes or complex updates, the instance method still provides the same benefits of keeping your object in sync with the database.</p>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_code_examples_2">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example_2">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-ruby" data-lang="ruby">user = User.find(id)
User.update(user.id, { # Noncompliant
  name: new_name,
  email: new_email,
  updated_at: Time.current
})
user.name # Still has the old name</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution_2">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-ruby" data-lang="ruby">user = User.find(id)
user.update({
  name: new_name,
  email: new_email,
  updated_at: Time.current
})
user.name # Now has the new name</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Rails Active Record Persistence - <a href="https://api.rubyonrails.org/classes/ActiveRecord/Persistence.html#method-i-update">Official Rails documentation for instance-level update methods</a></p>
</li>
<li>
<p>Rails Active Record Persistence Class Methods - <a href="https://api.rubyonrails.org/classes/ActiveRecord/Persistence/ClassMethods.html#method-i-update">Official Rails documentation for class-level update methods</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>