It is more efficient to skip the function invocation and use the predefined string constants rather than calling ``++Chr$()++`` or ``++ChrW$()++`` for the following numbers:


* 0 - ``++vbNullChar++``
* 8 - ``++vbBack++``
* 9 - ``++vbTab++``
* 10 - ``++vbLf++``
* 11 - ``++vbVerticalTab++``
* 12 - ``++vbFormFeed++``
* 13 - ``++vbCr++``
* 13+10 - ``++vbCrLf++`` | ``++vbNewLine++``
* 34 - ``++""++``


== Noncompliant Code Example

----
Dim myChar As Char 
myChar = Chr$(0)
----


== Compliant Solution

----
Dim myChar As Char 
myChar = vbNullChar
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

endif::env-github,rspecator-view[]
