== Why is this an issue?

It is more efficient to skip the function invocation and use the predefined string constants rather than calling ``++Chr$()++`` or ``++ChrW$()++`` for the following numbers:


* 0 - ``++vbNullChar++``
* 8 - ``++vbBack++``
* 9 - ``++vbTab++``
* 10 - ``++vbLf++``
* 11 - ``++vbVerticalTab++``
* 12 - ``++vbFormFeed++``
* 13 - ``++vbCr++``
* 13+10 - ``++vbCrLf++`` | ``++vbNewLine++``
* 34 - ``++""++``


=== Noncompliant code example

[source,vb6]
----
Dim myChar As Char 
myChar = Chr$(0)
----


=== Compliant solution

[source,vb6]
----
Dim myChar As Char 
myChar = vbNullChar
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Use the predefined constant XXX instead


endif::env-github,rspecator-view[]
