<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when a map literal contains the same key more than once.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>When a map literal contains duplicate keys, only the last value associated with that key is retained. All previous values for the same key are silently discarded.</p>
</div>
<div class="paragraph">
<p>This behavior is rarely intentional and usually indicates a programming error such as:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>Copy-paste mistakes where the same key is accidentally duplicated</p>
</li>
<li>
<p>Typos in key names that result in unintended duplicates</p>
</li>
<li>
<p>Refactoring errors where old key-value pairs weren&#8217;t properly removed</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>The silent nature of this overwriting makes it particularly dangerous, as the code will run without errors but may not behave as expected. This can lead to configuration issues, data loss, or incorrect application behavior that is difficult to debug.</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="paragraph">
<p>The duplicate key issue can cause:</p>
</div>
<div class="ulist">
<ul>
<li>
<p><strong>Data loss</strong>: Earlier values are silently discarded, potentially losing important configuration or data</p>
</li>
<li>
<p><strong>Logic errors</strong>: The application may use unexpected values, leading to incorrect behavior</p>
</li>
<li>
<p><strong>Debugging difficulties</strong>: Since no error is thrown, the issue may go unnoticed until runtime problems occur</p>
</li>
<li>
<p><strong>Maintenance problems</strong>: Future developers may be confused about which value should be used</p>
</li>
</ul>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Remove the duplicate key and keep only the intended key-value pair. If both values are needed, use different key names.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-groovy" data-lang="groovy">def config = [
    'timeout': 30,
    'retries': 3,
    'timeout': 60  // Noncompliant
]</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-groovy" data-lang="groovy">def config = [
    'timeout': 60,
    'retries': 3
]</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Groovy Documentation - Maps - <a href="https://groovy-lang.org/syntax.html#_maps">Official Groovy documentation on map syntax and behavior</a></p>
</li>
<li>
<p>CodeNarc DuplicateMapKey Rule - <a href="https://codenarc.org/codenarc-rules-basic.html#duplicatemapkey-rule">CodeNarc rule documentation for detecting duplicate map keys</a></p>
</li>
</ul>
</div>
</div>
<div class="sect2">
<h3 id="_related_rules">Related rules</h3>
<div class="ulist">
<ul>
<li>
<p><a data-rspec-id="S1534" class="rspec-auto-link">S1534</a> - <a href="https://rules.sonarsource.com/javascript/RSPEC-1534/">JavaScript rule for duplicate property names in object literals</a></p>
</li>
<li>
<p><a data-rspec-id="S5780" class="rspec-auto-link">S5780</a> - <a href="https://rules.sonarsource.com/python/RSPEC-5780/">Python rule for duplicate keys in dictionary literals</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>