Testing frameworks Mocha and Jest provide a way to run a single test case for a file using the function `.only()`. This is convenient when debugging a specific use case. It makes no sense having it in production or development, but it appears there when cleanup is forgotten after such a debugging session.

It is therefore strongly recommended to not commit usages of `.only()` to version control.

== Noncompliant Code Example

[source,javascript]
----
describe("MyClass", function () {
    it.only("should run correclty", function () {
        /*...*/
    });
});;
----

== Compliant Solution

[source,javascript]
----
describe("MyClass", function () {
    it("should run correclty", function () {
        /*...*/
    });
});;
----

== See

- https://mochajs.org/#exclusive-tests[exclusive tests in Mocha]
- https://jestjs.io/docs/next/api#testonlyname-fn-timeout[test.only() in Jest]
- https://jestjs.io/docs/next/api#describeonlyname-fn[describe.only() in Jest]
