== Why is this an issue?

Testing frameworks like Mocha and Jest provide a way to run a single test case for a file by appending `.only()` to the test function. This is useful when debugging a specific use case. However, it should not be used in production or development, as it is likely a leftover from debugging and serves no purpose in those contexts. 

It is strongly recommended not to include `.only()` usages in version control.

=== Noncompliant code example

[source,javascript]
----
describe("MyClass", function () {
    it.only("should run correctly", function () {
        /*...*/
    });
});
----

=== Compliant solution

[source,javascript]
----
describe("MyClass", function () {
    it("should run correctly", function () {
        /*...*/
    });
});
----

== Resources

- https://mochajs.org/#exclusive-tests[exclusive tests in Mocha]
- https://jestjs.io/docs/next/api#testonlyname-fn-timeout[test.only() in Jest]
- https://jestjs.io/docs/next/api#describeonlyname-fn[describe.only() in Jest]
