<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>The <code>suspend</code> function modifier is used to mark a function which might take some time to execute and could suspend the caller coroutine. The location where such function is called is a "suspension point". Functions marked as <code>suspend</code> ("suspending functions") should themselves contain at least one suspension point, otherwise it makes no sense to mark it as <code>suspend</code></p>
</div>
<div class="paragraph">
<p>This rule reports an issue if a function with <code>suspend</code> modifier has no calls to other <code>suspend</code> functions inside its body.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-kotlin" data-lang="kotlin">suspend fun function() { // Noncompliant, redundant 'suspend' modifier
    println("Hello!")
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-kotlin" data-lang="kotlin">fun function() {
    println("Hello!")
}</code></pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p><a href="https://kotlinlang.org/docs/coroutines-guide.html">Coroutines guide</a></p>
</li>
</ul>
</div>
</div>
</div>