<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>The <code>font-family</code> (and the shorthand <code>font</code>) CSS property specifies a prioritized list of one or more font family names and/or generic family names for the selected element.</p>
</div>
<div class="paragraph">
<p>If none of the font names defined in a <code>font</code> or <code>font-family</code> declaration are available on the browser of the user, the browser will display the text using its default font. It&#8217;s recommended to always define a generic font family for each declaration of <code>font</code> or <code>font-family</code> to get a less degraded situation than relying on the default browser font. This lets the browser select an acceptable fallback font when necessary.</p>
</div>
<div class="paragraph">
<p>The list of generic font families is as follows:</p>
</div>
<div class="ulist">
<ul>
<li>
<p><code>serif</code>: Glyphs have finishing strokes, flared or tapering ends, or actual serifed endings.</p>
</li>
<li>
<p><code>sans-serif</code>: Glyphs have plain stroke endings.</p>
</li>
<li>
<p><code>cursive</code>: Glyphs in cursive fonts generally have either joining strokes or other cursive characteristics beyond those of italic typefaces.</p>
</li>
<li>
<p><code>fantasy</code>: Fantasy fonts are primarily decorative fonts that contain playful representations of characters.</p>
</li>
<li>
<p><code>monospace</code>: All glyphs have the same fixed width.</p>
</li>
<li>
<p><code>system-ui</code>: Glyphs are taken from the default user interface font on a given platform.</p>
</li>
<li>
<p><code>ui-serif</code>: The default user interface serif font.</p>
</li>
<li>
<p><code>ui-sans-serif</code>: The default user interface sans-serif font.</p>
</li>
<li>
<p><code>ui-monospace</code>: The default user interface monospace font.</p>
</li>
<li>
<p><code>ui-rounded</code>: The default user interface font that has rounded features.</p>
</li>
</ul>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>You should always include at least one generic family name in a <code>font-family</code> list, since there&#8217;s no guarantee that any given font is available.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-css" data-lang="css">a {
  font-family: Helvetica, Arial, Verdana, Tahoma; /* Noncompliant; there is no generic font family in the list */
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-css" data-lang="css">a {
  font-family: Helvetica, Arial, Verdana, Tahoma, sans-serif;
}</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>CSS Specification - <a href="https://www.w3.org/TR/CSS2/fonts.html#generic-font-families">Generic font families</a></p>
</li>
<li>
<p>MDN - <a href="https://developer.mozilla.org/en-US/docs/Web/CSS/font-family"><code>font-family</code></a></p>
</li>
<li>
<p>MDN - <a href="https://developer.mozilla.org/en-US/docs/Web/CSS/font"><code>font</code></a></p>
</li>
</ul>
</div>
</div>
</div>
</div>