The requirement for a final ``++ELSE++`` clause is defensive programming. The ``++CASE++`` expression should always provide a value.

== Noncompliant Code Example

----
CASE grade -- Noncompliant, can raise a CASE_NOT_FOUND exception.
  WHEN 'A' THEN DBMS_OUTPUT.PUT_LINE('Excellent');
  WHEN 'B' THEN DBMS_OUTPUT.PUT_LINE('Very Good');
END CASE;
----

== Compliant Solution

----
CASE grade
  WHEN 'A' THEN DBMS_OUTPUT.PUT_LINE('Excellent');
  WHEN 'B' THEN DBMS_OUTPUT.PUT_LINE('Very Good');
  ELSE DBMS_OUTPUT.PUT_LINE('No such grade');
END CASE;
----

include::../see.adoc[]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
