<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when inline SOQL queries are assigned to generic <code>Object</code> types instead of properly typed SObject collections like <code>List&lt;SObject&gt;</code> or <code>SObject[]</code>.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>When you execute SOQL queries in Apex, assigning the results to generic <code>Object</code> types defeats the purpose of Apex&#8217;s strong typing system.</p>
</div>
<div class="paragraph">
<p>SOQL queries return specific SObject types based on the queried object. When you assign these results to an <code>Object</code> variable, you lose several important benefits:</p>
</div>
<div class="ulist">
<ul>
<li>
<p><strong>Type safety</strong>: The compiler cannot verify that you&#8217;re accessing valid fields or methods, leading to potential runtime errors.</p>
</li>
<li>
<p><strong>IDE support</strong>: Code completion, syntax highlighting, and refactoring tools cannot provide accurate suggestions.</p>
</li>
<li>
<p><strong>Code readability</strong>: Other developers cannot easily understand what type of data the variable contains.</p>
</li>
<li>
<p><strong>Performance</strong>: Additional type casting may be required when working with the data.</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>Apex provides strong typing specifically to help you write more reliable and maintainable code. By using properly typed collections like <code>Account[]</code> or <code>List&lt;Account&gt;</code>, you enable the compiler to catch errors early and make your code more self-documenting.</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="paragraph">
<p>Using generic <code>Object</code> types for SOQL results can lead to runtime errors when accessing SObject fields or methods. It also reduces code maintainability and makes debugging more difficult, as type-related issues may only surface during execution rather than at compile time.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Replace the generic <code>Object</code> type with the specific SObject type that matches your SOQL query. Use either array notation (<code>SObject[]</code>) or List notation (<code>List&lt;SObject&gt;</code>).</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-apex" data-lang="apex">Object accts = [SELECT Name, Phone FROM Account]; // Noncompliant</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-apex" data-lang="apex">Account[] accts = [SELECT Name, Phone FROM Account];
// or alternatively:
List&lt;Account&gt; accts = [SELECT Name, Phone FROM Account];</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Salesforce SOQL and SOSL Reference - <a href="https://developer.salesforce.com/docs/atlas.en-us.soql_sosl.meta/soql_sosl/">Official Salesforce documentation for SOQL queries and best practices</a></p>
</li>
<li>
<p>Apex Developer Guide - Working with SOQL - <a href="https://developer.salesforce.com/docs/atlas.en-us.apexcode.meta/apexcode/langCon_apex_SOQL.htm">Comprehensive guide on using SOQL within Apex code</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>