== Why is this an issue?

This rule raises an issue when a configured Java package or class is used in a JSP file.


=== Noncompliant code example

Given a disallowed list of ``++java.sql,ArrayList++``:

[source,html]
----
<%@ page import="java.sql.*" %>     <!-- Noncompliant -->
<% java.util.ArrayList clients; %>  <!-- Noncompliant -->
<% java.lang.String name; %>        <!-- Compliant -->
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

include::parameters.adoc[]

endif::env-github,rspecator-view[]
