== Why is this an issue?

This rule raises an issue when a configured Java package or class is used in a JSP file.


=== Noncompliant code example

Given a disallowed list of ``++java.sql,ArrayList++``:

[source,html]
----
<%@ page import="java.sql.*" %>     <!-- Noncompliant -->
<% java.util.ArrayList clients; %>  <!-- Noncompliant -->
<% java.lang.String name; %>        <!-- Compliant -->
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Remove the use of "xxx".


=== Parameters

.libraries
****

Comma-separated list of Java packages or classes, such as java.sql or java.util.ArrayList
****
.message
****

----
Remove the usage of this library which is not allowed.
----

Issue message which is displayed in case of violation
****


endif::env-github,rspecator-view[]
