include::../description.adoc[]

== Noncompliant Code Example

----
def do_something()
   # TODO
end
----

include::../see.adoc[]

ifdef::env-github,rspecator-view[]
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
