<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Classes annotated as <code>@Controller</code> in Spring are responsible for handling incoming web requests.
When annotating methods or the entire controller with <code>@ResponseBody</code>, the return value of said methods will be serialized and set as the response body.
In other words, it tells the Spring framework that this method does not produce a view.
This mechanism is commonly used to create API endpoints.</p>
</div>
<div class="paragraph">
<p>Spring provides <code>@RestController</code> as a convenient annotation to replace the combination of <code>@Controller</code> and <code>@ResponseBody</code>.
The two are functionally identical, so the single annotation approach is preferred.</p>
</div>
<div class="paragraph">
<p>This rule will raise an issue on a class that is annotated with <code>@Controller</code> if:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>the class is also annotated with <code>@ResponseBody</code> or</p>
</li>
<li>
<p>all methods in said class are annotated with <code>@ResponseBody</code>.</p>
</li>
</ul>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Replace the <code>@Controller</code> annotation with the <code>@RestController</code> annotation and remove all <code>@ResponseBody</code> annotations from the class and its methods.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">@Controller
@ResponseBody
public class MyController {
    @GetMapping("/hello")
    public String hello() {
        return "Hello World!";
    }
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">@RestController
public class MyController {
    @GetMapping("/hello")
    public String hello() {
        return "Hello World!";
    }
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_noncompliant_code_example_2">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">@Controller
public class MyController {
    @ResponseBody
    @GetMapping("/hello")
    public String hello() {
        return "Hello World!";
    }

    @ResponseBody
    @GetMapping("/foo")
    public String foo() {
        return "Foo";
    }
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution_2">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">@RestController
public class MyController {
    @GetMapping("/hello")
    public String hello() {
        return "Hello World!";
    }

    @GetMapping("/foo")
    public String foo() {
        return "Foo";
    }
}</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_articles_blog_posts">Articles &amp; blog posts</h3>
<div class="ulist">
<ul>
<li>
<p>Spring Guides - <a href="https://spring.io/guides/gs/rest-service/">Building a RESTful Web Service</a></p>
</li>
<li>
<p>Baeldung - <a href="https://www.baeldung.com/spring-controller-vs-restcontroller">The Spring @Controller and @RestController Annotations</a></p>
</li>
<li>
<p>Baeldung - <a href="https://www.baeldung.com/spring-request-response-body">Spring&#8217;s RequestBody and ResponseBody Annotations</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>