<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>An inner class that extends another type can call methods from both the outer class and parent type directly, without prepending <code>super.</code> or <code>Outer.this.</code>.</p>
</div>
<div class="paragraph">
<p>When both the outer and parent classes contain a method with the same name, the compiler will resolve an unqualified call to the parent type&#8217;s implementation.
The maintainer or a future reader may confuse the method call as calling the outer class&#8217;s implementation, even though it really calls the super type&#8217;s.</p>
</div>
<div class="paragraph">
<p>To make matters worse, the maintainer sees the outer class&#8217;s implementation in the same file as the call in the inner class, while the parent type is often declared in another file.
The maintainer may not even be aware of the ambiguity present, as they do not see the parent&#8217;s implementation.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Explicitly call the super type&#8217;s method by prepending <code>super.</code> to the method call.
If the intention was to call the outer class&#8217;s implementation, prepend <code>Outer.this.</code> instead.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">public class Parent {
  public void foo() { ... }
}

public class Outer {
  public void foo() { ... }

  public class Inner extends Parent {
    public void doSomething() {
      foo();  // Noncompliant, it is not explicit if Outer#foo or Parent#foo is the intended implementation to be called.
      // ...
    }
  }
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">public class Parent {
  public void foo() { ... }
}

public class Outer {
  public void foo() { ... }

  public class Inner extends Parent {
    public void doSomething() {
      super.foo(); // Compliant, it is explicit that Parent#foo is the desired implementation to be called.
      // ...
    }
  }
}</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Prefix this call to "xxx" with "super.".</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_27_jan_2015_201519_freddy_mallet_wrote">on 27 Jan 2015, 20:15:19 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>+1 to activate this rule by default.</p>
</div>
</div>
</div>
</div>