include::../description.adoc[]

include::../how-to-fix-it.adoc[]

=== Code examples

==== Noncompliant code example

[source,vbnet]
----
Class Foo
  Sub DoSomething(Param As Integer)
    If Param > 100 Then             ' Noncompliant, 100 is a magic number
      ' Do something
    End If
  End Sub
End Class
----

==== Compliant solution

[source,vbnet]
----
Class Foo
  Private Const MaxOfSomething As Integer = 100

  Sub DoSomething(Param As Integer)
    If Param > MaxOfSomething Then  ' Compliant
      ' Do something
    End If
  End Sub
End Class
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

=== Parameters

.exceptions
****
_STRING_

----
-1,0,1,2
----
****


'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
