<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>The <code>LINKAGE</code> section describes data made available from another program through the CALL statement. The data structure defined in a <code>LINKAGE</code> section should be located in a COPYBOOK. Otherwise, at runtime multiple COBOL programs may try to share data structures which are not similar.</p>
</div>
<div class="paragraph">
<p>First level data items can also be defined in the main program as long as there are no structural pieces of information attached to this first level like the length, the format, and so on.</p>
</div>
<div class="paragraph">
<p>This rule raises an issue only on the first invalid data item of a <code>LINKAGE</code> section.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cobol" data-lang="cobol">LINKAGE SECTION.

01 LK-DATA.  *&gt; Noncompliant
  05 LK-LENGTH     PIC S9(04) COMP.
   05 LK-VARIABLE  PIC X(08).</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cobol" data-lang="cobol">LINKAGE SECTION.

COPY MY_COPYBOOK.</code></pre>
</div>
</div>
<div class="paragraph">
<p>or</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cobol" data-lang="cobol">LINKAGE SECTION.

01 LK-DATA.
  COPY MY_COPYBOOK.</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Move the definition of this data structure into a COPYBOOK.</p>
</div>
</div>
</div>
</div>