This is an issue when an ``++@InvocableMethod++`` annotation lacks a `label` and `description`, or when these attributes contain empty values.

== Why is this an issue?

Einstein Copilot is a conversational AI assistant that helps Salesforce users execute commands using natural language. When developers create custom actions using Apex methods annotated with ``++@InvocableMethod++``, Copilot relies on the `label` and `description` attributes to understand what each method does.

Without clear and descriptive labels and descriptions, Einstein Copilot cannot effectively match user conversation intents with the appropriate actions. This leads to poor user experience where:

* Users' natural language requests may not trigger the intended custom actions
* Copilot may suggest irrelevant or incorrect actions
* The AI assistant becomes less helpful and intuitive to use

The `label` should provide a concise, human-readable name for the action, while the `description` should explain what the method accomplishes. These descriptions are crucial for Copilot's intent matching algorithm to work effectively.

=== What is the potential impact?

Poor intent matching in Einstein Copilot leads to frustrated users who cannot effectively interact with custom actions through natural language. This reduces the value of the conversational AI assistant and may result in users avoiding the feature entirely.

== How to fix it

Add meaningful `label` and `description` attributes to the ``++@InvocableMethod++`` annotation. The label should be a clear, concise name for the action, and the description should explain what the method does.

=== Code examples

==== Noncompliant code example

[source,apex,diff-id=1,diff-type=noncompliant]
----
@InvocableMethod
public static List<String> sayHello(List<String> name) {
    return new List<String>{ 'Hello ' + name[0] };
} // Noncompliant
----

==== Compliant solution

[source,apex,diff-id=1,diff-type=compliant]
----
@InvocableMethod(label='Say Hello' description='Say Hello to the user')
public static List<String> sayHello(List<String> name) {
    return new List<String>{ 'Hello ' + name[0] };
}
----

== Resources

=== Documentation

 * InvocableMethod Annotation - https://developer.salesforce.com/docs/atlas.en-us.apexcode.meta/apexcode/apex_classes_annotation_InvocableMethod.htm[Official Salesforce documentation for the InvocableMethod annotation]

 * Einstein Copilot Custom Actions - https://help.salesforce.com/s/articleView?id=release-notes.rn_einstein_copilot_custom_actions.htm&release=248&type=5[Guide to creating custom actions for Einstein Copilot]
