include::../rule.adoc[]

Similarly, tagged templates allow for advanced forms of string interpolation by evaluating the tag as a function to call, passing the
template literal elements as arguments. Therefore the rule also verifies that template literals don't start on a separate line.

== Noncompliant Code Example

[source,text]
----
const foo = function() { /* ... */ }
`bar` // Noncompliant
----

== Compliant Solution

[source,text]
----
function foo() { /* ... */ }
const s = foo`bar`;
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../highlighting.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
