== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,sql]
----
label: -- Noncompliant
  PRINT 'hello world';
GO
----

=== Compliant solution

[source,sql]
----
PRINT 'hello world';
GO
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
