include::../description.adoc[]

== Noncompliant Code Example

----
label: -- Noncompliant
  PRINT 'hello world';
GO
----

== Compliant Solution

----
PRINT 'hello world';
GO
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
