== Why is this an issue?

Coding conventions allow teams to collaborate effectively. For maximum standardization and readability, PHP code should use the long ``++<?php ?>++`` tags or the short-echo ``++<?= ?>++`` tags; it should not use the other tag variations.


=== Noncompliant code example

[source,php]
----
<?
$foo = 1;
?>
----


=== Compliant solution

[source,php]
----
<?php
$foo = 1;
?>
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

include::highlighting.adoc[]

'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
