This is an issue when the HTML meta viewport tag prevents users from zooming by setting `user-scalable=no` or limiting `maximum-scale` to 1.0 or less.

== Why is this an issue?

The viewport meta tag controls how web pages are displayed on mobile devices. When it prevents zooming, it creates accessibility barriers for users who need to enlarge text to read content.

Many users rely on zoom functionality to make text readable:

* People with visual impairments who need larger text
* Users with reading difficulties like dyslexia
* Anyone using a device in bright sunlight or poor lighting conditions

The Web Content Accessibility Guidelines (WCAG) 2.2 require that users can resize text up to 200% without assistive technology. When the viewport meta tag blocks zooming with `user-scalable=no` or restricts it with `maximum-scale=1`, it violates this requirement.

This creates a poor user experience and excludes users who depend on zoom functionality to access your content effectively.

=== What is the potential impact?

Users with visual impairments or reading difficulties may be unable to access your content. This violates accessibility standards and can lead to legal compliance issues in jurisdictions that require web accessibility.

== How to fix it

Remove the user-scalable=no attribute to allow users to zoom in and out as needed.

=== Code examples

==== Noncompliant code example

[source,html,diff-id=1,diff-type=noncompliant]
----
<meta name="viewport" content="width=device-width, initial-scale=1, user-scalable=no"> <!-- Noncompliant -->
----

==== Compliant solution

[source,html,diff-id=1,diff-type=compliant]
----
<meta name="viewport" content="width=device-width, initial-scale=1">
----

== Resources

=== Documentation

 * MDN: Viewport meta tag - https://developer.mozilla.org/en-US/docs/Web/HTML/Viewport_meta_tag[Complete guide to the viewport meta tag and its attributes]

 * Deque University: Meta viewport accessibility - https://dequeuniversity.com/rules/axe/4.10/meta-viewport[Accessibility guidelines for viewport meta tags]

=== Standards

 * WCAG 2.2: Success Criterion 1.4.4 Resize text - https://www.w3.org/TR/WCAG22/#resize-text[Text must be resizable up to 200% without assistive technology]

 * ACT Rules: Meta viewport allows for zoom - https://act-rules.github.io/rules/b4f0c3[Accessibility conformance testing rule for viewport zoom restrictions]
