<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>When calling a CICS command other than <code>RETURN</code>, <code>ADDRESS</code>, or <code>ABEND</code>, either <code>RESP</code> should be used to specify where the response code should be written, or <code>NOHANDLE</code> should be used to specify that abnormal conditions should be ignored.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-text" data-lang="text">           EXEC CICS DELETEQ TS        *&gt; Noncompliant
             QNAME(WS-TS5FTARF-NAME)
           END-EXEC.</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-text" data-lang="text">           EXEC CICS DELETEQ TS
             QNAME(WS-TS5FTARF-NAME)
             RESP(WS-STATUS)
           END-EXEC.</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_21_mar_2017_133558_ann_campbell_wrote">on 21 Mar 2017, 13:35:58 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>targeting C/C&#43;&#43; based on https://www.ibm.com/support/knowledgecenter/en/SSGMGV_3.1.0/com.ibm.cics.ts31.doc/dfhp3/dfhp36i.htms</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_21_mar_2017_134904_ann_campbell_wrote">on 21 Mar 2017, 13:49:04 Ann Campbell wrote:</h3>
<div class="paragraph">
<p><a href="http://www.careerride.com/view/what-happens-when-a-cics-command-contains-the-nohandle-option-3753.aspx" class="bare">http://www.careerride.com/view/what-happens-when-a-cics-command-contains-the-nohandle-option-3753.aspx</a></p>
</div>
<div class="paragraph">
<p><a href="https://www.ibm.com/support/knowledgecenter/SSGMCP_5.1.0/com.ibm.cics.ts.applicationprogramming.doc/topics/dfhp4_nohandleintro.html" class="bare">https://www.ibm.com/support/knowledgecenter/SSGMCP_5.1.0/com.ibm.cics.ts.applicationprogramming.doc/topics/dfhp4_nohandleintro.html</a></p>
</div>
<div class="paragraph">
<p><a href="https://www.ibm.com/support/knowledgecenter/SSGMCP_5.2.0/com.ibm.cics.ts.applicationprogramming.doc/topics/dfhp4_resp2intro.html" class="bare">https://www.ibm.com/support/knowledgecenter/SSGMCP_5.2.0/com.ibm.cics.ts.applicationprogramming.doc/topics/dfhp4_resp2intro.html</a></p>
</div>
</div>
<div class="sect2">
<h3 id="_on_21_mar_2017_135002_ann_campbell_wrote">on 21 Mar 2017, 13:50:02 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~pierre-yves.nicolas] your changes look fine. I&#8217;ve further expanded. Would you double-check me, now?</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_21_mar_2017_135654_pierre_yves_nicolas_wrote">on 21 Mar 2017, 13:56:54 Pierre-Yves Nicolas wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] Looks good to me.</p>
</div>
</div>
</div>
</div>