When calling a CICS command other than ``++RETURN++``, ``++ADDRESS++``, or ``++ABEND++``, either ``++RESP++`` should be used to specify where the response code should be written, or ``++NOHANDLE++`` should be used to specify that abnormal conditions should be ignored.


== Noncompliant Code Example

[source,text]
----
           EXEC CICS DELETEQ TS        *> Noncompliant
             QNAME(WS-TS5FTARF-NAME)
           END-EXEC.
----


== Compliant Solution

[source,text]
----
           EXEC CICS DELETEQ TS        
             QNAME(WS-TS5FTARF-NAME)
             RESP(WS-STATUS)        
           END-EXEC.  
----

