<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>It is convention to name each class&#8217;s logger for the class itself. Doing so allows you to set up clear, communicative logger configuration. Naming loggers by some other convention confuses configuration, and using the same class name for multiple class loggers prevents the granular configuration of each class' logger. Some libraries, such as SLF4J warn about this, but not all do.</p>
</div>
<div class="paragraph">
<p>This rule raises an issue when a logger is not named for its enclosing class.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">public class MyClass {
  private final static Logger LOG = LoggerFactory.getLogger(WrongClass.class);  // Noncompliant; multiple classes using same logger
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">public class MyClass {
  private final static Logger LOG = LoggerFactory.getLogger(MyClass.class);
}</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Update this logger to use the current class.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>Xxx.class</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_25_nov_2015_092238_freddy_mallet_wrote">on 25 Nov 2015, 09:22:38 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>See my comment on relating Google Group thread [~ann.campbell.2]: https://groups.google.com/forum/?utm_medium=email&amp;utm_source=footer#!msg/sonarqube/7DGhtbqRsHk/j2rDIp6dAwAJ</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_7_jun_2018_141752_andrei_epure_wrote">on 7 Jun 2018, 14:17:52 Andrei Epure wrote:</h3>
<div class="paragraph">
<p>Also requested in https://groups.google.com/forum/?utm_source=digest&amp;utm_medium=email#!topic/sonarqube/qCK_pZJ7G_Q</p>
</div>
</div>
</div>
</div>