include::../description.adoc[]

== Noncompliant Code Example

[source,cs]
----
public bool CanVote(Person person) 
{
  return person.GetAge() > 18 ? true : true; // Noncompliant; is this what was intended?
}
----

== Compliant Solution

[source,cs]
----
public bool CanVote(Person person) 
{
  return person.GetAge() > 18 ? true : false; 
  // or even better: 
  // return person.GetAge() > 18;
}
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
