The W3C specifications define the valid pseudo-class selectors. Only the official and browser-specific pseudo-class selectors should be used to get the expected impact in the final rendering.


== Noncompliant Code Example

----
a:hoverr { /* Noncompliant; there is a typo on the word "hover" */
...
}
----


== Compliant Solution

----
a:hover {
...
}
----


