include::../description.adoc[]

== Noncompliant Code Example

----
public function setName($name) {
    $name = $name;
}
----

== Compliant Solution

----
public function setName($name) {
    $this->name = $name;
}
----

include::../see.adoc[]

ifdef::env-github,rspecator-view[]
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
