== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,php]
----
public function setName($name) {
    $name = $name;
}
----

=== Compliant solution

[source,php]
----
public function setName($name) {
    $this->name = $name;
}
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
