== Why is this an issue?

In PHP, keywords and constants are case-insensitive, meaning they can be written in either lower case or upper case without affecting their functionality.
This allows for more flexibility and ease of use when writing code.

However, it is generally recommended to follow a consistent casing convention for readability and maintainability purposes.
Relevant constants are `true`, `false` and `null`.


=== Noncompliant code example

[source,php,diff-id=1,diff-type=noncompliant]
----
<?php ECHO 'Hello World'; ?> 
----


=== Compliant solution

[source,php,diff-id=1,diff-type=compliant]
----
<?php echo 'Hello World'; ?> 
----

== Resources
=== Documentation
* https://www.php.net/manual/en/reserved.constants.php[PHP Manual - Predefined Constants]


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

endif::env-github,rspecator-view[]
