Using indifferently lower or upper case for PHP keywords and constants "true", "false" and "null" can impact the readability of PHP source code.


== Noncompliant Code Example

[source,php]
----
<?php ECHO 'Hello World'; ?> 
----


== Compliant Solution

[source,php]
----
<?php echo 'Hello World'; ?> 
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

endif::env-github,rspecator-view[]
