Using indifferently lower or upper case for PHP keywords and constants "true", "false" and "null" can impact the readability of PHP source code.


== Noncompliant Code Example

----
<?php ECHO 'Hello World'; ?> 
----


== Compliant Solution

----
<?php echo 'Hello World'; ?> 
----

