A string literal that is the first statement in a module, function, class, or method is a docstring. A docstring should document what a caller needs to know about the code. Information about what it does, what it returns, and what it requires are all valid candidates for documentation. Well written docstrings allow callers to use your code without having to first read it and understand its logic.


By convention, docstrings are enclosed in three sets of double-quotes.


== Noncompliant Code Example

----
def my_function(a,b):
----


== Compliant Solution

----
def my_function(a,b):
      """Do X"""
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

endif::env-github,rspecator-view[]
