== Why is this an issue?

JavaScript allows any kind of value in boolean expressions. Using numbers, strings, and even objects as a boolean expression is possible. The runtime coerces those to a boolean value. However, this automatic coercion can be misleading and lead to unexpected results.

The rule forbids the usage of strings and numbers in expressions where a boolean is expected:

- Argument to the logical negation operator (`!arg`).
- The condition in a conditional expression (`cond ? x : y`).
- Conditions for `if`, `for`, `while`, and `do-while` statements.
- Operands of logical binary operators (`||`, `&&`).

=== What is the potential impact?

Numbers get coerced to `true` except for the zero and `NaN` which are converted to `false`.

Strings get coerced to `true` as well, except for the empty string `''` which is converted to `false`.

These special cases can lead to bugs.

== How to fix it

Either check the string or number's value or use the function constructor `Boolean` to convert it to boolean.

=== Code examples

==== Noncompliant code example

[source,javascript,diff-id=1,diff-type=noncompliant]
----
const x = [1,2,3];
if (x.length) {
    /* ... */
}
----

==== Compliant solution

[source,javascript,diff-id=1,diff-type=compliant]
----
const x = [1,2,3];
if (x.length > 0) {
    /* ... */
}
----

==== Noncompliant code example

[source,javascript,diff-id=2,diff-type=noncompliant]
----
const s = 'foo';
if (s) {
    /* ... */
}
----

==== Compliant solution

[source,javascript,diff-id=2,diff-type=compliant]
----
const s = 'foo';
if (s !== '') {
    /* ... */
}
----

==== Noncompliant code example

[source,javascript,diff-id=3,diff-type=noncompliant]
----
const b = BigInt(42);
if (b) {
    /* ... */
}
----

==== Compliant solution

[source,javascript,diff-id=3,diff-type=compliant]
----
const b = BigInt(42);
if (Boolean(b)) {
    /* ... */
}
----

== Resources

* https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Boolean#boolean_coercion[MDN Boolean coercion]
