When the ``++FILE STATUS++`` is not specified on a file, any read operations on the file should handle the "AT END" or "INVALID KEY" conditions. 


== Noncompliant Code Example

----
SELECT MY_FILE
  ASSIGN TO 'foobar.txt'
  ORGANIZATION IS SEQUENTIAL.
...
READ MY_FILE
  NOT AT END PERFORM COMPUTE_LINE
END-READ.
----


== Compliant Solution

----
SELECT MY_FILE
  ASSIGN TO 'foobar.txt'
  ORGANIZATION IS SEQUENTIAL.
...
READ MY_FILE
  NOT AT END PERFORM COMPUTE_LINE
  AT END MOVE 'Y' TO EOF-FLAG
END-READ.
----

