This rule raises an issue when a custom event is created with a name that starts with the prefix 'on'.

== Why is this an issue?

In Lightning Web Components, event handler names conventionally start with 'on' (like `onclick`, `onchange`, etc.). When custom event names also start with 'on', it creates confusion and inconsistency.

For example, if you create a custom event named 'onselect', the corresponding event handler would need to be named 'ononselect', which is confusing and breaks the expected naming pattern.

This naming convention issue makes the code harder to read and understand, especially for developers who are familiar with standard web development practices where the 'on' prefix is reserved for event handlers, not event names themselves.

Following consistent naming conventions helps maintain code clarity and makes it easier for team members to understand the codebase.

=== What is the potential impact?

This issue affects code maintainability and readability. While it doesn't cause runtime errors, it can lead to confusion among developers and makes the code harder to understand and maintain. It also goes against established Lightning Web Components best practices.

== How to fix it

Remove the 'on' prefix from the custom event name. The event name should describe the action without the 'on' prefix.

=== Code examples

==== Noncompliant code example

[source,javascript,diff-id=1,diff-type=noncompliant]
----
sendData(){
    const customEvt = new CustomEvent('onselect'); // Noncompliant
    this.dispatchEvent(customEvt);
}
----

==== Compliant solution

[source,javascript,diff-id=1,diff-type=compliant]
----
sendData(){
    const customEvt = new CustomEvent('select');
    this.dispatchEvent(customEvt);
}
----

== Resources

=== Documentation

 * Lightning Web Components Best Practices - https://www.apexhours.com/20-tips-for-lightning-web-components[Comprehensive guide covering 20 tips for Lightning Web Components including naming conventions]

 * Lightning Web Components Developer Guide - Events - https://developer.salesforce.com/docs/component-library/documentation/en/lwc/lwc.events[Official Salesforce documentation on handling events in Lightning Web Components]
