<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>The C&#43;&#43; standards define some identifiers as having special meaning in specific contexts. These are:</p>
</div>
<div class="ulist">
<ul>
<li>
<p><code>final</code> and <code>override</code> since C&#43;&#43;11</p>
</li>
<li>
<p><code>module</code> and <code>import</code> since C&#43;&#43;20</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>While it is technically possible to use them as normal identifiers, it&#8217;s clearer for the reader of the code to consider them as if they were keywords and only use them with their special meaning.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">void module(int final); // Noncompliant</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">void precept(int finalValue); // Compliant</code></pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p>C&#43;&#43; reference - <a href="https://en.cppreference.com/w/cpp/language/final">final specifier</a></p>
</li>
<li>
<p>C&#43;&#43; reference - <a href="https://en.cppreference.com/w/cpp/language/override">override specifier</a></p>
</li>
<li>
<p>C&#43;&#43; reference - <a href="https://en.cppreference.com/w/cpp/identifier_with_special_meaning/module">C&#43;&#43; identifier with special meaning: module</a></p>
</li>
<li>
<p>C&#43;&#43; reference - <a href="https://en.cppreference.com/w/cpp/identifier_with_special_meaning/import">C&#43;&#43; identifier with special meaning: import</a></p>
</li>
</ul>
</div>
<hr>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Replace XXX with another name</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_is_related_to_s1190">is related to: <a data-rspec-id="S1190" class="rspec-auto-link">S1190</a></h3>

</div>
<div class="sect2">
<h3 id="_is_related_to_s2961">is related to: <a data-rspec-id="S2961" class="rspec-auto-link">S2961</a></h3>

</div>
<div class="sect2">
<h3 id="_on_11_dec_2015_173245_ann_campbell_wrote">on 11 Dec 2015, 17:32:45 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~evgeny.mandrikov], given</p>
</div>
<div class="quoteblock">
<blockquote>
<div class="paragraph">
<p>override is context-sensitive and has special meaning only when it&#8217;s used after a member function declaration;</p>
</div>
</blockquote>
</div>
<div class="paragraph">
<p>from https://msdn.microsoft.com/en-us/library/jj678987.aspx</p>
</div>
<div class="paragraph">
<p>I suggest extending this rule to C&#43;&#43;. WDYT?</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_16_feb_2017_170110_ann_campbell_wrote">on 16 Feb 2017, 17:01:10 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>For an understanding of why C&#43;&#43; is targeted, see the C# description</p>
</div>
</div>
</div>
</div>