<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>The ability to target the common language runtime from several languages means that clashes are possible when a word that is reserved in one language is used as the name of a namespace, type or member in another.</p>
</div>
<div class="paragraph">
<p>This rule raises an issue when a keyword from C&#43;&#43;/CLI, C# or Visual Basic is used as an identifier.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-csharp" data-lang="csharp">public string nameof(string s) { return s; }  // Noncompliant
...
public string Hello { get { return "World!"; } }
...
Console.WriteLine(nameof(Hello)); // prints "World!" instead of "Hello" as expected</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-csharp" data-lang="csharp">public string GetValue(string s) { return s; }
...
public string Hello { get { return "World!"; } }
...
Console.WriteLine(GetValue(Hello));</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Rename XXX</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_1_mar_2017_155046_valeri_hristov_wrote">on 1 Mar 2017, 15:50:46 Valeri Hristov wrote:</h3>
<div class="paragraph">
<p>It seems that not all C#/VB/C&#43;&#43; keywords are reported by this rule and there is no list of the keywords FxCop matches.</p>
</div>
<div class="paragraph">
<p>We are postponing the development of this rule because of its small target audience (it is best suited for library developers, who&#8217;s products will be used in different languages).</p>
</div>
</div>
<div class="sect2">
<h3 id="_is_related_to_s1190">is related to: <a data-rspec-id="S1190" class="rspec-auto-link">S1190</a></h3>

</div>
<div class="sect2">
<h3 id="_is_related_to_s2961">is related to: <a data-rspec-id="S2961" class="rspec-auto-link">S2961</a></h3>

</div>
<div class="sect2">
<h3 id="_on_11_dec_2015_173245_ann_campbell_wrote">on 11 Dec 2015, 17:32:45 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~evgeny.mandrikov], given</p>
</div>
<div class="quoteblock">
<blockquote>
<div class="paragraph">
<p>override is context-sensitive and has special meaning only when it&#8217;s used after a member function declaration;</p>
</div>
</blockquote>
</div>
<div class="paragraph">
<p>from https://msdn.microsoft.com/en-us/library/jj678987.aspx</p>
</div>
<div class="paragraph">
<p>I suggest extending this rule to C&#43;&#43;. WDYT?</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_16_feb_2017_170110_ann_campbell_wrote">on 16 Feb 2017, 17:01:10 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>For an understanding of why C&#43;&#43; is targeted, see the C# description</p>
</div>
</div>
</div>
</div>