This rule raises when a TaskGroup or Nursery body contains only a single call to `.start` or `.start_soon` without passing itself as a parameter.

== Why is this an issue?

TaskGroup and Nursery are powerful tools for structured concurrency that automatically manage the lifecycle of multiple concurrent tasks. However, when they are used to spawn only a single task that doesn't need the nursery/TaskGroup instance itself, they add unnecessary overhead and complexity.

The main issues with using TaskGroup and Nursery for single tasks are:

* **Unnecessary overhead**: Creating a nursery or TaskGroup involves additional setup and teardown that serves no purpose when only one task is spawned
* **Misleading code**: The presence of a nursery suggests that multiple tasks will be managed, which can be confusing
* **Reduced readability**: The structured concurrency syntax is more verbose than a simple function call

=== What is the potential impact?

* **Performance**: Minor performance overhead from unnecessary nursery/TaskGroup creation
* **Maintainability**: Code is more complex than necessary, making it harder to understand and maintain
* **Code clarity**: The intent of the code is obscured by unnecessary structured concurrency constructs


== How to fix it in Asyncio

Replace the TaskGroup with a direct function call when the TaskGroup body contains only a single `create_task()` call.

=== Code examples

==== Noncompliant code example

[source,python,diff-id=2,diff-type=noncompliant]
----
import asyncio

async def worker_task():
    await asyncio.sleep(1)

async def main():
    # Unnecessary TaskGroup for a single task
    async with asyncio.TaskGroup() as tg:
        tg.create_task(worker_task())
----

==== Compliant solution

[source,python,diff-id=2,diff-type=compliant]
----
import asyncio

async def worker_task():
    await asyncio.sleep(1)

async def main():
    # Direct function call is simpler and more efficient
    await worker_task()
----



== How to fix it in Trio

Replace the nursery with a direct function call when the nursery body contains only a single `start_soon()` or `start()` call.

=== Code examples

==== Noncompliant code example

[source,python,diff-id=1,diff-type=noncompliant]
----
import trio

async def worker_task():
    await trio.sleep(1)

async def main():
    # Unnecessary nursery for a single task
    async with trio.open_nursery() as nursery:
        nursery.start_soon(worker_task)
----

==== Compliant solution

[source,python,diff-id=1,diff-type=compliant]
----
import trio

async def worker_task():
    await trio.sleep(1)

async def main():
    # Direct function call is simpler and more efficient
    await worker_task()
----


== How to fix it in AnyIO

Replace the task group with a direct function call when the task group body contains only a single `start_soon()` or `start()` call.

=== Code examples

==== Noncompliant code example

[source,python,diff-id=3,diff-type=noncompliant]
----
import anyio

async def worker_task():
    await anyio.sleep(1)

async def main():
    # Unnecessary task group for a single task
    async with anyio.create_task_group() as tg:
        tg.start_soon(worker_task)
----

==== Compliant solution

[source,python,diff-id=3,diff-type=compliant]
----
import anyio

async def worker_task():
    await anyio.sleep(1)

async def main():
    # Direct function call is simpler and more efficient
    await worker_task()
----

== Resources
=== Documentation

* Asyncio documentation - https://docs.python.org/3/library/asyncio-task.html#task-groups[Task Groups]
* Trio documentation - https://trio.readthedocs.io/en/latest/reference-core.html#nurseries-and-spawning[Nurseries and spawning]
* AnyIO documentation - https://anyio.readthedocs.io/en/stable/tasks.html#creating-and-managing-tasks[Creating and managing tasks]


ifdef::env-github,rspecator-view[]

== Implementation Specification
(visible only on this page)

=== Message
Replace the TaskGroup or Nursery with a direct function call.

=== Highlighting
* Primary locations: The task group creation function in the `async with` statement

* Secondary locations: - The `.start` or `.start_soon` call
- the enclosing async function `async` keyword (message: "This function is async.")

endif::env-github,rspecator-view[]

