This rule raises an issue when the built-in `input()` function is called in an asynchronous function.

== Why is this an issue?

In Python's asynchronous programming (using `asyncio`, `Trio`, or `AnyIO`), an event loop manages concurrent tasks by having them yield control during time-consuming operations, enabling other tasks to run.

However, the synchronous `input()` function blocks the current thread until user input is received, and when called from a coroutine, it blocks the entire event loop, preventing other tasks from executing and making the application unresponsive - effectively defeating the purpose of asynchronous programming for applications requiring concurrent operations or user interaction.

== How to fix it in Asyncio

You can use `asyncio.to_thread()` to run the `input()` function in a separate thread.

=== Code examples

==== Noncompliant code example

[source,python,diff-id=1,diff-type=noncompliant]
----
import asyncio

async def get_name():
    print("Please enter your name:")
    name = input() # Noncompliant
    return name
----

==== Compliant solution

[source,python,diff-id=1,diff-type=compliant]
----
import asyncio

async def get_name():
    print("Please enter your name:")
    name = await asyncio.to_thread(input) # Compliant
    return name
----

== How to fix it in Trio

You can use `trio.to_thread.run_sync()` to run the `input()` function in a separate thread.

=== Code examples

==== Noncompliant code example

[source,python,diff-id=2,diff-type=noncompliant]
----
import trio

async def get_name():
    print("Please enter your name:")
    name = input() # Noncompliant
    return name
----

==== Compliant solution

[source,python,diff-id=2,diff-type=compliant]
----
import trio

async def get_name():
    print("Please enter your name:")
    name = await trio.to_thread.run_sync(input) # Compliant
    return name
----

== How to fix it in AnyIO

You can use `anyio.to_thread.run_sync()` to run the `input()` function in a separate thread.

=== Code examples

==== Noncompliant code example

[source,python,diff-id=3,diff-type=noncompliant]
----
import anyio

async def get_name():
    print("Please enter your name:")
    name = input() # Noncompliant
    return name
----

==== Compliant solution

[source,python,diff-id=3,diff-type=compliant]
----
import anyio

async def get_name():
    print("Please enter your name:")
    name = await anyio.to_thread.run_sync(input) # Compliant
    return name
----

== Resources
=== Documentation
* Python asyncio - https://docs.python.org/3/library/asyncio-task.html#asyncio.to_thread[Running in threads]
* Trio - https://trio.readthedocs.io/en/stable/reference-core.html#trio.to_thread.run_sync[Putting blocking I/O into worker threads]
* AnyIO - https://anyio.readthedocs.io/en/stable/threads.html#calling-synchronous-code-from-a-worker-thread[Calling synchronous code from a worker thread]

=== Articles & blog posts
* Python - https://realpython.com/python-concurrency/[Concurrency and Parallelism in Python]

ifdef::env-github,rspecator-view[]
== Implementation Specification
(visible only on this page)

=== Message
Wrap call to `input()` with `await asyncio.to_thread(input)` or `await trio.to_thread.run_sync(input)` or `await anyio.to_thread.run_sync(input)`.

If we can't detect the asynchronous library: Wrap call to `input()` with the appropriate function from the asynchronous library.

=== Highlighting
The `input()` function call.
Secondary: the `async` keyword of the function definition.

endif::env-github,rspecator-view[]
