This rule raises an issue when the same key is used multiple times when instantiating a dictionary literal.

== Why is this an issue?

By definition a dictionary cannot hold two identical values.
When instantiating a dictionary literal with the same key repeated multiple times,
only the last occurrence of the duplicated key will remain.
The duplicated key should be either modified or removed.

=== Noncompliant code example

[source,text,diff-id=1,diff-type=noncompliant]
----
{"one": 1, "two": 2, "one": 3}  # Noncompliant: Change or remove duplicates of this key.

def func(a1, a2, a3):
    {a1: 1, a2: 2, a1: 3}  # Noncompliant: Change or remove duplicates of this key.
----

=== Compliant solution

[source,text,diff-id=1,diff-type=compliant]
----
{"one": 1, "two": 2, "three": 3}

def func(a1, a2, a3):
    {a1: 1, a2: 2, a3: 3}
----


== Resources

=== Documentation

* Python Documentation - https://docs.python.org/3/reference/expressions.html#dictionary-displays[Dictionary displays]
