This rule raises an issue when the same key is used multiple times when instantiating a dictionary literal.

== Why is this an issue?

By definition a dictionary cannot hold multiple times the same key.
When instantiating a dictionary literal with the same key repeated multiple times,
only the last occurrence of the duplicated key will remain.

Creating a dictionary with redundant elements is prone to errors and confusion. A duplicated key in a dictionary literal should be either:

* modified, as it was mistakenly put in the dictionary instead an actual key which would lead to bugs and errors further in the program.
* removed, as it was a simple duplication, making the code confusing and difficult to maintain.

=== Code examples

==== Noncompliant code example

[source,python,diff-id=1,diff-type=noncompliant]
----
{"one": 1, "two": 2, "one": 3}  # Noncompliant: the key "one" has duplicate.

def func(a1, a2):
    {a1: 1, a2: 2, a1: 3}  # Noncompliant: the key a1 has duplicate.
----

==== Compliant solution

[source,python,diff-id=1,diff-type=compliant]
----
{"one": 1, "two": 2, "three": 3}

def func(a1, a2):
    {a1: 1, a2: 2}
----


== Resources

=== Documentation

* Python Documentation - https://docs.python.org/3/reference/expressions.html#dictionary-displays[Dictionary displays]
