<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>When a non-existent variable is referenced a <code>ReferenceError</code> is raised.</p>
</div>
<div class="paragraph">
<p>Due to the dynamic nature of JavaScript this can happen in a number of scenarios:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>When typo was made in a symbol&#8217;s name.</p>
</li>
<li>
<p>When using variable declared with <code>let</code> or <code>const</code> before declaration (unlike <code>var</code>-declarations, they are not hoisted to the top of the scope).</p>
</li>
<li>
<p>Due to confusion with scopes of <code>let</code>- and <code>const</code>-declarations (they have block scope, unlike <code>var</code>-declarations, having function scope).</p>
</li>
<li>
<p>When accessing a property in the wrong scope (e.g. forgetting to specify <code>this.</code>).</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>This rule does not raise issues on global variables which are defined with <code>sonar.javascript.globals</code> and <code>sonar.javascript.environments</code> properties.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-javascript" data-lang="javascript">var john = {
  firstName: "john",
  show: function() { console.log(firstName); } // Noncompliant: firstName is not defined
}
john.show();</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-javascript" data-lang="javascript">var john = {
  firstName: "john",
  show: function() { console.log(this.firstName); }
}
john.show();</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>XX does not exist. Change its name or declare it so that its usage doesn&#8217;t result in a "ReferenceError".</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>Primary location: Statement</p>
</div>
<div class="paragraph">
<p>Secondary location: the variable/class/function&#8217;s definition later in the same scope (if there is one)</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_is_related_to_s5807">is related to: <a data-rspec-id="S5807" class="rspec-auto-link">S5807</a></h3>

</div>
</div>
</div>