<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This is an issue when a Flask route decorator does not specify the methods parameter, especially if the route handler checks request.method for methods other than GET.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Flask routes only accept GET requests by default when no methods parameter is specified. This can cause unexpected "Method Not Allowed" errors at runtime.</p>
</div>
<div class="paragraph">
<p>When developers write route handlers that check request.method for POST, PUT, DELETE, or other HTTP methods, they often forget to explicitly allow these methods in the route decorator. This creates a mismatch between what the code expects and what Flask actually allows.</p>
</div>
<div class="paragraph">
<p>For example, if a route handler checks for POST requests but the decorator doesn&#8217;t include POST in the methods list, any POST request to that endpoint will result in a 405 Method Not Allowed error before the handler code even runs.</p>
</div>
<div class="paragraph">
<p>Explicitly specifying allowed methods makes the code more readable and prevents these runtime errors. It also serves as documentation, clearly showing which HTTP methods the endpoint supports.</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="paragraph">
<p>This issue can cause runtime errors that prevent users from successfully submitting forms or making API calls. The "Method Not Allowed" errors can be confusing to debug, especially for developers who are new to Flask. In production, this can lead to broken functionality and poor user experience.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it_in_flask">How to fix it in Flask</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Add the methods parameter to the route decorator and specify all HTTP methods that the route should accept. Include all methods that are checked within the handler function.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">@app.route('/api/users')
def handle_users():
    if request.method == 'POST':  # Noncompliant
        return create_user()
    return get_users()</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">@app.route('/api/users', methods=['GET', 'POST'])
def handle_users():
    if request.method == 'POST':
        return create_user()
    return get_users()</code></pre>
</div>
</div>
<div class="paragraph">
<p>For routes that only handle GET requests, explicitly specify methods=['GET'] to make the intent clear, even though GET is the default.</p>
</div>
</div>
<div class="sect3">
<h4 id="_noncompliant_code_example_2">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">@app.route('/dashboard')
def dashboard():  # Noncompliant
    return render_template('dashboard.html')</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution_2">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">@app.route('/dashboard', methods=['GET'])
def dashboard():
    return render_template('dashboard.html')</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="paragraph">
<p>Flask Routing Documentation - <a href="https://flask.palletsprojects.com/en/2.3.x/quickstart/#routing%5BOfficial" class="bare">https://flask.palletsprojects.com/en/2.3.x/quickstart/#routing%5BOfficial</a>  Flask documentation on URL routing and HTTP methods]</p>
</div>
<div class="paragraph">
<p>Flask HTTP Methods - <a href="https://flask.palletsprojects.com/en/2.3.x/quickstart/#http-methods%5BFlask" class="bare">https://flask.palletsprojects.com/en/2.3.x/quickstart/#http-methods%5BFlask</a>  documentation specifically about handling different HTTP methods]</p>
</div>
</div>
</div>
</div>