== Noncompliant Code Example

With default provided regular expression: XXXXX

----
public int DoSomething(){...}
----


== Compliant Solution

----
public int doSomething(){...}
----


== Exceptions

Overriding methods are excluded.


----
@Override
public int Do_Something(){...}
----

ifdef::env-github,rspecator-view[]

'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]

endif::env-github,rspecator-view[]
