== Why is this an issue?

=== Noncompliant code example

With default provided regular expression: XXXXX

[source,text]
----
public int DoSomething(){...}
----


=== Compliant solution

[source,text]
----
public int doSomething(){...}
----


=== Exceptions

Overriding methods are excluded.


----
@Override
public int Do_Something(){...}
----

ifdef::env-github,rspecator-view[]

'''
== Comments And Links
(visible only on this page)

=== on 19 Dec 2014, 19:34:51 Ann Campbell wrote:
The intent for this was that it be a sub-task, but there's no indication of what the parent should be.


endif::env-github,rspecator-view[]
