== Why is this an issue?

=== Noncompliant code example

With default provided regular expression: XXXXX

[source,text]
----
public int DoSomething(){...}
----


=== Compliant solution

[source,text]
----
public int doSomething(){...}
----


=== Exceptions

Overriding methods are excluded.


----
@Override
public int Do_Something(){...}
----

ifdef::env-github,rspecator-view[]

'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]

endif::env-github,rspecator-view[]
