== Noncompliant Code Example

With default provided regular expression: XXXXX

----
public int DoSomething(){...}
----


== Compliant Solution

----
public int doSomething(){...}
----


== Exceptions

Overriding methods are excluded.


----
@Override
public int Do_Something(){...}
----

