<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>When either the equality operator in a null test or the logical operator that follows it is reversed, the code has the appearance of safely null-testing the object before dereferencing it. Unfortunately the effect is just the opposite - the object is null-tested and then dereferenced <em>only</em> if it is null, leading to a guaranteed null pointer dereference.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-php" data-lang="php">if ($obj == null &amp;&amp; $obj-&gt;isOpen()) {
  echo "Object is open";
}

if ($obj != null || $obj-&gt;isOpen()) {
  echo "Object is not open";
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-php" data-lang="php">if ($obj == null || $obj-&gt;isOpen()) {
  echo "Object is open";
}

if ($obj != null &amp;&amp; !$obj-&gt;isOpen()) {
  echo "Object is not open";
}</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Either reverse the equality operator in the "XXX" null test, or reverse the logical operator that follows it.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_17_oct_2016_142105_yves_dubois_pèlerin_wrote">on 17 Oct 2016, 14:21:05 Yves Dubois-Pèlerin wrote:</h3>
<div class="paragraph">
<p>Activated the rule as part of SONARPHP-643 (Synchonize metadata of rules with RSPECs).</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_4_nov_2014_105532_freddy_mallet_wrote">on 4 Nov 2014, 10:55:32 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>@Ann, I think this rule should be activated by default</p>
</div>
</div>
</div>
</div>