include::../common/why.adoc[]

include::../common/how.adoc[]

=== Code examples

==== Noncompliant code example

[source,html,diff-id=1,diff-type=noncompliant]
----
<li role="button">Foo</li> <!--Noncompliant; "li" isn't interactive, but "button" is-->
----

==== Compliant solution

[source,html,diff-id=1,diff-type=compliant]
----
<li role="listitem">Foo</li>
----

include::../common/resources.adoc[]
