<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Text should provide enough contrast with its background so that it can be read easily by people with moderately low vision, including colorblind people.</p>
</div>
<div class="paragraph">
<p>This rule raises an issue when a CSS selector defines both the <code>background-color</code> and <code>color</code> properties with a contrast ratio of less than 4.5:1, which is the contrast required to meet WCAG AA level. It applies only when the selector is not on a heading (<code>h1</code>, <code>h2</code>,&#8230;&#8203;) or <code>title</code></p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-css" data-lang="css">body {
    font-size: 17pt;
}
.myclass {
    color: #777;  // NonCompliant
    background-color: #888;
}

.header h4 {
    font-size: 22pt;
}</code></pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p><a href="https://www.w3.org/TR/WCAG21/#contrast-minimum">WCAG2 - minimum contrast</a></p>
</li>
</ul>
</div>
<hr>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>change background-color or color to increase the contrast ratio</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>"color" property&#8217;s value</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_8_feb_2019_150005_nicolas_harraudeau_wrote">on 8 Feb 2019, 15:00:05 Nicolas Harraudeau wrote:</h3>
<div class="paragraph">
<p>Note: The contrast ratio function is defined here: https://www.w3.org/TR/WCAG20-TECHS/G18.html</p>
</div>
</div>
</div>
</div>