<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Creating a new <code>Exception</code> without actually raising it has no effect and is probably due to a mistake.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">def func(x):
    if not isinstance(x, int):
        TypeError("Wrong type for parameter 'x'. func expects an integer")  # Noncompliant
    if x &lt; 0:
        ValueError  # Noncompliant
    return x + 42</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">def func(x):
    if not isinstance(x, int):
        raise TypeError("Wrong type for parameter 'x'. func expects an integer")
    if x &lt; 0:
        raise ValueError
    return x + 42</code></pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p><a href="https://docs.python.org/3/tutorial/errors.html#raising-exceptions">Python documentation - Raising Exceptions</a></p>
</li>
</ul>
</div>
<hr>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Raise this exception or remove this useless statement</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>Primary: The statement creating an Exception.</p>
</div>
</div>
</div>
</div>