== Why is this an issue?

Creating a new ``++Throwable++`` without actually throwing it is useless and is probably due to a mistake.


=== Noncompliant code example

[source,text]
----
if (x < 0)
  new IllegalArgumentException("x must be nonnegative");
----


=== Compliant solution

[source,text]
----
if (x < 0)
  throw new IllegalArgumentException("x must be nonnegative");
----

