<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Creating a new <code>Exception</code> without actually throwing it is useless and is probably due to a mistake.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-vbnet" data-lang="vbnet">If x &lt; 0 Then
    Dim ex = New ArgumentException("x must be nonnegative")
End If</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-vbnet" data-lang="vbnet">If x &lt; 0 Then
    Throw New ArgumentException("x must be nonnegative")
End If</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Throw this exception or remove this useless statement</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>The useless statement</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_5_sep_2017_143539_duncan_pocklington_wrote">on 5 Sep 2017, 14:35:39 Duncan Pocklington wrote:</h3>
<div class="paragraph">
<p>The C# implementation of this rule catches one invalid case:</p>
</div>
<div class="paragraph">
<p>{{new Exception(&#8230;&#8203;)</p>
</div>
<div class="paragraph">
<p>}}</p>
</div>
<div class="paragraph">
<p>However, this isn&#8217;t valid syntax in VB.Net (i.e. it will be reported as a syntax error in the IDE).</p>
</div>
<div class="paragraph">
<p>For this rule to be useful in VB it would need to be more complex i.e. check for exceptions that are created and assigned to a variable/field/property that are not then returned, passed to another method, or returned, or thrown.</p>
</div>
<div class="paragraph">
<p>\[~jeanchristophe.collet] should we just park this issue for VB for now?</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_5_sep_2017_150219_jean_christophe_collet_wrote">on 5 Sep 2017, 15:02:19 Jean-Christophe Collet wrote:</h3>
<div class="paragraph">
<p>Agreed, work on this should be postponed for now.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_5_oct_2018_090808_loïc_joly_wrote">on 5 Oct 2018, 09:08:08 Loïc Joly wrote:</h3>
<div class="paragraph">
<p>The current implementation for C# is just checking that the exception object is used somehow. But in VB.NET, it&#8217;s not possible to create an object without using it, so the rule would never be triggered.</p>
</div>
<div class="paragraph">
<p>If the C# implementation tracked the value to see if it was really thrown (requiring inter-procedural symbolic execution), then it would make sense to reactivate this rule for VB.NET too.</p>
</div>
</div>
</div>
</div>