<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This raises an issue when <code>set()</code> is directly around the output of <code>sorted()</code>.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Calling <code>set(sorted(iterable))</code> is usually an indication of a misunderstanding of the desired outcome or an inefficient way to achieve it.
The <code>sorted()</code> function produces a <strong>list</strong> of items in sorted order.
Applying <code>set()</code> to this sorted list converts it into a set, which is an <strong>unordered</strong> collection of unique elements.
The effort spent on sorting is immediately negated if the final result is an unordered set, as the order established by <code>sorted()</code> is discarded.</p>
</div>
<div class="paragraph">
<p>If the intention is to obtain a sorted list of unique elements from an iterable, the pattern <code>set(sorted(iterable))</code> is inefficient.
It first sorts all elements, including duplicates (which can be computationally expensive for large lists with many duplicates),
and then removes these duplicates while also discarding the order established by <code>sorted()</code>.
The more efficient and standard idiom for getting unique, sorted items is to deduplicate <strong>first</strong> using <code>set()</code>, and then sort the unique items: <code>sorted(set(iterable))</code>.
This way, <code>sorted()</code> operates on a potentially smaller collection of unique items.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>To fix this issue remove the call to either <code>set()</code> or <code>sorted()</code>, or call <code>sorted()</code> on the output of <code>set()</code>.
If the goal is just to obtain unique items (and order is not important), then <code>set(iterable)</code> is sufficient; the <code>sorted()</code> call can be removed.
If the goal is to obtain sorted items with duplicates preserved, then <code>sorted(iterable)</code> is sufficient; the <code>set()</code> call can be removed.
If the goal is to obtain sorted items with duplicates removed, then <code>sorted(set(iterable))</code> is the correct way to proceed.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">data = [3, 4, 1, 2]
set(sorted(data)) # Noncompliant: set is called on a sorted list</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">data = [3, 4, 1, 2]
sorted(data) # Compliant</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Python Documentation - <a href="https://docs.python.org/3/library/functions.html#sorted">sorted</a></p>
</li>
<li>
<p>Python Documentation - <a href="https://docs.python.org/3/tutorial/datastructures.html#sets">sets</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>