include::../why-fields.adoc[]

=== Noncompliant code example

[source,flex]
----
class Foo {
  public var myField:int;

  public function doSomething():String {
    var myField:int = 0; // Noncompliant
    //...
  }
}
----

include::../rspecator.adoc[]