== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,flex]
----
class Foo {
  public var myField:int;

  public function doSomething():String {
    var myField:int = 0;
    ...
  }
}
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../highlighting.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
