<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when closures rely on implicit parameter declarations instead of using explicit parameter syntax.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Groovy closures can use implicit parameters, but this can lead to confusion and reduced code readability.</p>
</div>
<div class="paragraph">
<p>When a closure doesn&#8217;t explicitly declare parameters, Groovy automatically provides an implicit parameter named <code>it</code>. While this might seem convenient, it can create several problems:</p>
</div>
<div class="ulist">
<ul>
<li>
<p><strong>Unclear intent</strong>: It&#8217;s not immediately obvious whether the closure is intended to accept parameters or not</p>
</li>
<li>
<p><strong>Accidental parameter acceptance</strong>: A closure intended to accept no parameters will still accept one parameter through the implicit <code>it</code></p>
</li>
<li>
<p><strong>Poor readability</strong>: Using <code>it</code> doesn&#8217;t convey the meaning or purpose of the parameter</p>
</li>
<li>
<p><strong>Maintenance difficulties</strong>: Future developers may not understand what <code>it</code> represents in the context</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>Explicit parameter declarations make the code self-documenting and prevent misunderstandings about the closure&#8217;s interface.</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="paragraph">
<p>Using implicit parameters can lead to:</p>
</div>
<div class="ulist">
<ul>
<li>
<p><strong>Runtime errors</strong>: Closures may be called with unexpected arguments when they were intended to accept none</p>
</li>
<li>
<p><strong>Reduced code maintainability</strong>: Other developers may struggle to understand the purpose of parameters</p>
</li>
<li>
<p><strong>Debugging difficulties</strong>: Issues related to parameter usage become harder to trace</p>
</li>
<li>
<p><strong>Code review challenges</strong>: Reviewers cannot easily determine the intended closure interface</p>
</li>
</ul>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>For closures that should accept no parameters, use the explicit empty parameter list syntax <code>{&#8594;}</code>. This clearly indicates that the closure expects no arguments and will fail if called with arguments.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-groovy" data-lang="groovy">def magicNumber = { 42 }
// This closure implicitly accepts one parameter 'it'
// but is intended to accept none</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-groovy" data-lang="groovy">def magicNumber = {-&gt; 42 }
// Explicitly declares no parameters
// Will fail if called with arguments</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Groovy Closures Documentation - <a href="https://groovy-lang.org/closures.html">Official Groovy documentation covering closure syntax, parameters, and best practices</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>