== Why is this an issue?

If you use ``++std::unique_ptr<T> const &++`` for a function parameter type, it means that the function will not be able to alter the ownership of the pointed-to object by the ``++unique_ptr++``:

* It cannot acquire ownership of the pointed-to object (this would require a parameter of type ``++std::unique_ptr<T>++``)
* It cannot transfer the object ownership to someone else (this would require a ``++std::unique_ptr<T> &++``).

That means the function can only observe the pointed-to object, and in this case, passing a ``++T*++`` (if the ``++unique_ptr++`` can be null) or a ``++T&++`` (if it cannot) provides the same features, while also allowing the function to work with objects that are not handled by a ``++unique_ptr++`` (e.g., objects on the stack, in a ``++vector++``, or in another kind of smart pointer), thus making the function more general-purpose.


=== Noncompliant code example

[source,cpp,diff-id=1,diff-type=noncompliant]
----
void draw(std::unique_ptr<Shape> const &shape); // Noncompliant

void drawAll(std::vector<std::unique_ptr<Shape>> v)
{
  for (auto &shape : v) {
      if (shape) {
        draw(shape);
      }
  }
}
----


=== Compliant solution

[source,cpp,diff-id=1,diff-type=compliant]
----
void draw(Shape const &shape); // Compliant

void drawAll(std::vector<std::unique_ptr<Shape>> v)
{
  for (auto &shape : v) {
      if (shape) {
        draw(*shape);
      }
  }
}
----


== Resources

=== Documentation

* {cpp} reference - https://en.cppreference.com/w/cpp/memory/unique_ptr[std::unique_ptr]

=== External coding guidelines

* {cpp} Core Guidelines - https://github.com/isocpp/CppCoreGuidelines/blob/e49158a/CppCoreGuidelines.md#r32-take-a-unique_ptrwidget-parameter-to-express-that-a-function-assumes-ownership-of-a-widget[R.32: Take a `unique_ptr<widget>` parameter to express that a function assumes ownership of a `widget`]


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Replace this use of "unique_ptr" by a raw pointer or a reference (possibly const).


'''
== Comments And Links
(visible only on this page)

=== on 6 Nov 2018, 19:59:55 Ann Campbell wrote:
I've reworded slightly [~loic.joly]. Double-check me, please.


Also, it would be useful to give a "such as" here:


____
objects that are not handled by a ``++unique_ptr++``
____


Or to expand a little on the topic in general

=== on 7 Nov 2018, 09:27:50 Loïc Joly wrote:
\[~ann.campbell.2] I partially reverted your changes, could you review please (maybe we need to talk about it if you disagree)?

=== on 7 Nov 2018, 20:11:41 Ann Campbell wrote:
\[~loic.joly] it's not clear to me how "will not be able to impact the lifetime of the pointed-to object" is a summary/grouping/superset of the two bullet points that follow it, which is why in my edit I struggled, and then combined the three into a single list.

=== on 8 Nov 2018, 09:56:13 Loïc Joly wrote:
\[~ann.campbell.2] Lifetime and ownership are deeply linked subject: If A has ownership of B, it means that A is responsible for killing B at some time (at the very least, before its own lifetime ends).

Is the new version clearer?

endif::env-github,rspecator-view[]
