== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,abap]
----
 CHECK (SY-SUBRC NE 0). "compliant even if ignored by compiler
 IF ((SY-SUBRC EQ 0)). "Noncompliant
----

=== Compliant solution

[source,abap]
----
 CHECK (SY-SUBRC NE 0).
 IF (SY-SUBRC EQ 0). 
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../highlighting.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
