== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,abap]
----
 CHECK (SY-SUBRC NE 0). "compliant even if ignored by compiler
 IF ((SY-SUBRC EQ 0)). "Noncompliant
----

=== Compliant solution

[source,abap]
----
 CHECK (SY-SUBRC NE 0).
 IF (SY-SUBRC EQ 0). 
----

include::../rspecator.adoc[]
