include::../description.adoc[]

== Noncompliant Code Example

----
 CHECK (SY-SUBRC NE 0). "compliant even if ignored by compiler
 IF ((SY-SUBRC EQ 0)). "Noncompliant
----

== Compliant Solution

----
 CHECK (SY-SUBRC NE 0).
 IF (SY-SUBRC EQ 0). 
----

ifdef::rspecator-view[]
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::rspecator-view[]
