include::../description.adoc[]

== Noncompliant Code Example

[source,python]
----
r"[0-9]" # Noncompliant - same as r"\d"
r"[^0-9]" # Noncompliant - same as r"\D"
r"[A-Za-z0-9_]" # Noncompliant - same as r"\w"
r"[\w\W]" # Noncompliant - same as r"."
r"a{0,}" # Noncompliant - same as r"a*"
----

== Compliant Solution

[source,python]
----
r"\d"
r"\D"
r"\w"
r"."
r"a*"
----

include::../implemtation.adoc[]
