== Why is this an issue?

When the `RLSE` parameter is specified in a `SPACE` directive, the system releases unused space that has been allocated for the data set.

Omitting this parameter can cause a waste of unneeded space reserved for the data set.

== How to fix it

=== Code examples

==== Noncompliant code example

[source,jcl,diff-id=1,diff-type=noncompliant]
----
//STEP1 EXEC PGM=IEFBR14
//OUT1  DD  DSN=A.B.C,
//          UNIT=3390,VOL=SER=Z19SPL,
//          DISP=(NEW,CATLG),
//          SPACE=(CYL,(1,1)) <-- Noncompliant
----

==== Compliant solution

[source,jcl,diff-id=1,diff-type=compliant]
----
//STEP1 EXEC PGM=IEFBR14
//OUT1  DD  DSN=A.B.C,
//          UNIT=3390,VOL=SER=Z19SPL,
//          DISP=(NEW,CATLG),
//          SPACE=(CYL,(1,1),RLSE)
----

== Resources
=== Documentation

* IBM Documentation - https://www.ibm.com/docs/en/zos/3.1.0?topic=statement-space-parameter[SPACE parameter]
* IBM Documentation - https://www.ibm.com/docs/en/zos/3.1.0?topic=definition-system-assignment-space[System assignment of space]

