This is an issue when an import statement uses an alternative repository path instead of the canonical module path declared in the module's go.mod file.

== Why is this an issue?

Go modules require import paths to match the canonical module path declared in the module's `go.mod` file. When a module is available through multiple repository URLs, only one path is considered canonical.

Using non-canonical import paths can cause several problems:

* **Module resolution failures**: The Go toolchain may fail to resolve dependencies correctly when non-canonical paths are used.
* **Inconsistent dependency management**: Different parts of your codebase might reference the same module through different paths, leading to confusion and potential version conflicts.
* **Migration issues**: When migrating existing code to Go modules, non-canonical import paths must be updated to match the canonical module path.
* **Tooling incompatibility**: Go development tools expect canonical paths for proper module analysis and dependency tracking.

For example, both `github.com/golang/lint` and `golang.org/x/lint` resolve to the same repository, but only `golang.org/x/lint` is the canonical module path. Using the non-canonical GitHub path will cause issues in module-aware builds.

=== What is the potential impact?

Using non-canonical import paths can lead to build failures, dependency resolution errors, and inconsistent module management. This can prevent successful compilation and deployment of Go applications using modules.

== How to fix it

Replace the non-canonical import path with the canonical module path as declared in the module's go.mod file.

=== Code examples

==== Noncompliant code example

[source,go,diff-id=1,diff-type=noncompliant]
----
import "github.com/golang/lint" // Noncompliant
----

==== Compliant solution

[source,go,diff-id=1,diff-type=compliant]
----
import "golang.org/x/lint"
----

== Resources

=== Documentation

 * Go Modules: Migrating to Go Modules - https://go.dev/blog/migrating-to-go-modules[Official Go blog post explaining module migration and canonical import paths]

 * Go Command: Remote Import Paths - https://go.dev/cmd/go/#hdr-Remote_import_paths[Documentation on how Go resolves remote import paths]

 * Go Modules Reference - https://go.dev/ref/mod[Complete reference for Go modules including module paths and versioning]
