== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,sql]
----
SELECT HASHBYTES('SHA1', MyColumn) FROM dbo.MyTable;  
----

=== Compliant solution

[source,sql]
----
SELECT HASHBYTES('SHA2_256', MyColumn) FROM dbo.MyTable;  
----

include::../see.adoc[]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
