include::../description.adoc[]

== Noncompliant Code Example

----
string actual = "Hello World!";
actual.Should();  // Noncompliant
----

== Compliant Solution

----
string actual = "Hello World!";
actual.Should().Contain("Hello");
----

ifdef::rspecator-view[]
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::rspecator-view[]
