include::../description.adoc[]

== Noncompliant Code Example

----
string actual = "Hello World!";
actual.Should();  // Noncompliant
----

== Compliant Solution

----
string actual = "Hello World!";
actual.Should().Contain("Hello");
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
