<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>A function which does not return a value and which does not have external side effects will only consume time and will not contribute to the generation of any outputs, which may not meet developer expectations.</p>
</div>
<div class="paragraph">
<p>The following are examples of external side effects:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>Reading or writing to a file, stream, etc.;</p>
</li>
<li>
<p>Changing the value of a non local variable;</p>
</li>
<li>
<p>Changing the value of an argument having reference type;</p>
</li>
<li>
<p>Using a volatile object;</p>
</li>
<li>
<p>Raising an exception.</p>
</li>
</ul>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">void pointless ( void ) // Non-compliant â€“ no external side effects
{
 int16_t local;
 local = 0;
}</code></pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p>MISRA C&#43;&#43;:2008, 0-1-8</p>
</li>
</ul>
</div>
<hr>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Either refactor or remove this function without side effect(s).</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_16_oct_2014_142517_ann_campbell_wrote">on 16 Oct 2014, 14:25:17 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~samuel.mercier] FYI, I&#8217;ve switched this to Efficiency because the description says such functions "only consume time and not contribute"</p>
</div>
</div>
</div>
</div>