<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>C&#43;&#43; does not support the nesting of C-style comments even though some compilers support this as a non-portable language extension.</p>
</div>
<div class="paragraph">
<p>A comment beginning with <code>/*</code> continues until the first <code>*/</code> is encountered.</p>
</div>
<div class="paragraph">
<p>Any <code>/*</code> occurring inside a comment is a violation of this rule.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">/* some comment, end comment marker accidentally omitted
Perform_Critical_Safety_Function(X);
/* this "comment" is Noncompliant */</code></pre>
</div>
</div>
<div class="paragraph">
<p>In reviewing the code containing the call to the function, the assumption is that it is executed code.</p>
</div>
<div class="paragraph">
<p>Because of the accidental omission of the end comment marker, the call to Perform_Critical_ Safety_Function will not be executed.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p>MISRA C:2004, 2.3 - The character sequence /* shall not be used within a comment.</p>
</li>
<li>
<p>MISRA C&#43;&#43;:2008, 2-7-1 - The character sequence /* shall not be used within a C-style comment.</p>
</li>
<li>
<p>MISRA C:2012, 3.1 - The character sequences /* and // shall not be used within a comment</p>
</li>
<li>
<p><a href="https://wiki.sei.cmu.edu/confluence/x/k9YxBQ">CERT, MSC04-C.</a> - Use comments consistently and in a readable fashion</p>
</li>
</ul>
</div>
<hr>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove this nested comment.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_7_apr_2015_231359_evgeny_mandrikov_wrote">on 7 Apr 2015, 23:13:59 Evgeny Mandrikov wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] why this targets C and C&#43;&#43;, where we already have implementation of two rules - "NestedComment" and superseeding "<a data-rspec-id="S1103" class="rspec-auto-link">S1103</a>"?</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_8_apr_2015_180936_ann_campbell_wrote">on 8 Apr 2015, 18:09:36 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>In fact, [~evgeny.mandrikov], this rule <em>is</em> NestedComment (which the History tab tells me you already knew). Further, this rule is deprecated in favor of <a data-rspec-id="S1103" class="rspec-auto-link">RSPEC-1103</a>.</p>
</div>
<div class="paragraph">
<p>I&#8217;ve entered RULEAPI-208 to handle this cleanup automatically in the future. For now, I&#8217;ll remove the targeted languages from this ticket.</p>
</div>
</div>
</div>
</div>