== Noncompliant Code Example

[source,text]
----
/*
  This is a comment block.
  It may be difficult to figure out that the following line of code is actually commented


variable = function_call();
/* variable contains the result. Noncompliant; inner comment */
*/
----
