<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Nested comments are confusing and can lead maintainers to misunderstand which code is active.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-swift" data-lang="swift">/*
  This is a comment block.
  It may be difficult to figure out that the following line of code is actually commented


variable = function_call();
/* variable contains the result. Noncompliant; inner comment */
*/</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove this nested comment.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_7_apr_2015_231359_evgeny_mandrikov_wrote">on 7 Apr 2015, 23:13:59 Evgeny Mandrikov wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] why this targets C and C&#43;&#43;, where we already have implementation of two rules - "NestedComment" and superseeding "<a data-rspec-id="S1103" class="rspec-auto-link">S1103</a>"?</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_8_apr_2015_180936_ann_campbell_wrote">on 8 Apr 2015, 18:09:36 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>In fact, [~evgeny.mandrikov], this rule <em>is</em> NestedComment (which the History tab tells me you already knew). Further, this rule is deprecated in favor of <a data-rspec-id="S1103" class="rspec-auto-link">RSPEC-1103</a>.</p>
</div>
<div class="paragraph">
<p>I&#8217;ve entered RULEAPI-208 to handle this cleanup automatically in the future. For now, I&#8217;ll remove the targeted languages from this ticket.</p>
</div>
</div>
</div>
</div>