== Why is this an issue?

In the interests of keeping code clean, the simplest possible conditional syntax should be used. That means 
using the ``++??=++`` (conditional assignment) operator for a self-assign-if-not-null operation.

=== Noncompliant code example

[source, dart]
----
if (a == null) {  // Noncompliant
    a = new object();
}

----


=== Compliant solution

[source,csharp]
----
a ??= new object();
----

== Resources

* https://dart.dev/tools/linter-rules/prefer_conditional_assignment[Dart Lint rule]
