== Why is this an issue?

Casting a base-class pointer/reference to a derived-class pointer/reference is commonly referred to as downcasting which can only be done using an explicit cast.

However, the use of ``++static_cast++`` for such a cast is unsafe because it doesn't do any runtime check.  If the cast memory doesn't contain an object of the expected derived type, your program enters the undefined behavior territory.


If your object is polymorphic, you might prefer using ``++dynamic_cast++`` instead, as it allows safe downcasting by performing a run-time check:

* If the cast memory contains an object of the expected derived type, the check succeeds. The result of the ``++dynamic_cast++`` points/refers to the derived object. 
* If the cast memory doesn't contain an object of the expected derived type, the check fails. If the ``++dynamic_cast++`` is used on a pointer, ``++nullptr++`` is returned. If it was used on a reference, ``++std::bad_cast++`` is thrown.

This rule raises an issue when ``++static_cast++`` is used for downcasting.


=== Noncompliant code example

[source,cpp]
----
struct Shape {
  virtual ~Shape();
  // ...
};

struct Rectangle : public Shape {
  double width;
  double height;
};

struct Circle : public Shape {
  double radius;
};

double computeArea(const Shape* shape) {
  const auto* rectangle = static_cast<const Rectangle*>(shape); // Noncompliant
  return rectangle->width * rectangle->height;
}
----


=== Compliant solution

[source,cpp]
----
struct Shape {
  virtual ~Shape();
  // ...
};

struct Rectangle : public Shape {
  double width;
  double height;
};

struct Circle : public Shape {
  int radius;
};

double computeArea(const Shape* shape) {
  if(const auto* rectangle = dynamic_cast<const Rectangle*>(shape)) { // Compliant
    return rectangle->width * rectangle->height; 
  }
  return 0;
}
----


== Resources

* https://github.com/isocpp/CppCoreGuidelines/blob/c553535fb8dda2839d13ab5f807ffbc66b63d67b/CppCoreGuidelines.md#type2-dont-use-static_cast-downcasts-use-dynamic_cast-instead[{cpp} Core Guidelines - Type safety profile - Type.2]: Don’t use static_cast to downcast. Use dynamic_cast instead.


ifdef::env-github,rspecator-view[]
'''
== Comments And Links
(visible only on this page)

=== relates to: S5823

endif::env-github,rspecator-view[]
