== Why is this an issue?

Code is unreachable if there is no syntactic (control flow) path to it. If such code exists, it is unclear if this is intentional or simply that an appropriate path has been accidentally omitted.

Compilers may choose not to generate code for these constructs, meaning that, even if the unreachable code is intentional, it may not be present in the final executable code.

Missing statements, often caused by editing activities, are a common source of unreachable code.


=== Noncompliant code example

[source,cpp]
----
int16_t with_unreach ( int16_t para )
{
 int16_t local;
 local = 0;
 switch ( para )
 {
    local = para; // unreachable – Noncompliant
  case 1:
    break;
    para++; // unreachable – Noncompliant
  default:
    break;
  }
  return para;
  para++; // unreachable – Noncompliant
}
----


=== Compliant solution

[source,cpp]
----
int16_t with_unreach ( int16_t para )
{
 int16_t local;
 local = 0;
 switch ( para )
 {
  case 1:
    break;
  default:
    break;
  }
  return para;
}
----


== Resources

* MISRA C:2004, 14.1
* MISRA {cpp}:2008, 0-1-1
* MISRA C:2012, 2.1


ifdef::env-github,rspecator-view[]
'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
