include::../description.adoc[]

== Noncompliant Code Example

----
func compute(start int) { // Noncompliant; start is not used
	sum := 0
	for i := 0; i < 10; i++ {
		sum += i
	}
	fmt.Println("Result:", sum)
}
----

== Compliant Solution

----
func compute() {
	sum := 0
	for i := 0; i < 10; i++ {
		sum += i
	}
	fmt.Println("Result:", sum)
}
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
