Shared coding conventions allow teams to collaborate effectively. This rule checks that labels match a provided regular expression.


== Noncompliant Code Example

With default provided regular expression ``++^[A-Z][A-Z0-9_]*$++``:

[source,text]
----
exit:  // Noncompliant
  doCleanup();
----


== Compliant Solution

[source,text]
----
EXIT:  // Compliant
  doCleanup();
----

