== Why is this an issue?

Shared coding conventions allow teams to collaborate effectively. This rule checks that labels match a provided regular expression.


=== NonCompliant code example

With default provided regular expression ``++^[A-Z][A-Z0-9_]*$++``:

[source,text]
----
exit:  // Noncompliant
  doCleanup();
----


=== Compliant solution

[source,text]
----
EXIT:  // Compliant
  doCleanup();
----

