Shared coding conventions allow teams to collaborate effectively. This rule checks that labels match a provided regular expression.


== Noncompliant Code Example

With default provided regular expression ``^[A-Z][A-Z0-9_]*$``:

----
exit:  // Noncompliant
  doCleanup();
----


== Compliant Solution

----
EXIT:  // Compliant
  doCleanup();
----

