:identifier_capital_plural: Local variables and function parameters
:identifier: local variable and function parameter
:identifier_plural: local variables and function parameters
:identifier_or: local variable or function parameter
:regex: ^[a-z_][a-zA-Z0-9_]*$

include::../rule.adoc[]

=== Code examples

==== Noncompliant code example

With the default regular expression ``{regex}``:

[source,php,diff-id=1,diff-type=noncompliant]
----
function printSomething($My_Param){ //Noncompliant
  $LOCAL = ""; // Noncompliant
  echo $My_Param . $LOCAL;
}
----

==== Compliant solution

[source,php,diff-id=1,diff-type=compliant]
----
function dprintSomething($my_Param){
  $local = "";
  echo $my_Param . $local;
}
----

== Resources

=== Documentation

* Wikipedia - https://en.wikipedia.org/wiki/Naming_convention_(programming)[Naming Convention (programming)]

=== Related rules

* S100 - Function names should comply with a naming convention
* S101 - Class names should comply with a naming convention
* S114 - Interface names should comply with a naming convention
* S115 - Constant names should comply with a naming convention
* S116 - Field names should comply with a naming convention
* S1578 - File names should comply with a naming convention
* S3360 - Test class names should end with "Test"

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../parameters.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
