include::../description.adoc[]

== Noncompliant Code Example

With the default regular expression ``++^[a-z][a-zA-Z0-9]*$++``:

[source,php]
----
public function doSomething($my_param){  
  $LOCAL;
  ...
}
----

== Compliant Solution

[source,php]
----
public function doSomething($myParam){  
  $local;
  ...
}
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::parameters.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
