== Why is this an issue?

PHP lets you make static calls to non-static methods, ex: ``++A::f();++``. But the fact that you can doesn't mean you _should_. While such calls will work when there's no ``++$this++`` reference in the called method, a fatal error will occur when there is. Furthermore, such calls have been deprecated in PHP 7, and their support may be removed from future releases.


=== Noncompliant code example

[source,php]
----
class A {
  function f() {
    // ...
  }
}

A::f();      // Noncompliant, "f" is non-static
----


=== Compliant solution

[source,php]
----
class A {
  function f() {
    // ...
  }
}

$a = new A();
$a->f();
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

include::highlighting.adoc[]

endif::env-github,rspecator-view[]
