include::../why-dotnet.adoc[]

For example, a `String` should never be used for locking. When a `String` is https://en.wikipedia.org/wiki/Interning_(computer_science)[interned] by the runtime, it can be shared by multiple threads, breaking the locking mechanism.

Instead, a dedicated private `Object` instance should be used for each shared resource. This minimizes access to the lock instance, avoiding deadlocks and lock contention.

The following objects are considered as shared resources:

* a reference to https://learn.microsoft.com/en-us/dotnet/visual-basic/programming-guide/program-structure/me-my-mybase-and-myclass#me[Me]: if the instance is publicly accessible, the lock might be shared
* a https://learn.microsoft.com/en-us/dotnet/api/system.type[Type] object: if the type class is publicly accessible, the lock might be shared
* a https://learn.microsoft.com/en-us/dotnet/visual-basic/language-reference/data-types/string-data-type[String] literal or instance: if any other part of the program uses the same string, the lock is shared because of interning

== How to fix it

=== Code examples

==== Noncompliant code example

[source,vbnet,diff-id=1,diff-type=noncompliant]
----
Public Sub MyLockingMethod()
    SyncLock Me 'Noncompliant
        ' ...
    End SyncLock
End Sub
----

==== Compliant solution

[source,vbnet,diff-id=1,diff-type=compliant]
----
Private lockObj As New Object()
Public Sub MyLockingMethod()
    SyncLock lockObj
        ' ...
    End SyncLock
End Sub
----

include::../resources-dotnet.adoc[]

include::../rspecator.adoc[]
