include::../common/rationale.adoc[]

== Why is this an issue?

include::../common/description.adoc[]

=== What is the potential impact?

include::../common/impact.adoc[]

== How to fix it
 
=== Code examples

include::../common/fix/code-rationale.adoc[]

==== Noncompliant code example

[source,vbnet,diff-id=1,diff-type=noncompliant]
----
Imports System.IO

Sub Example()
    Dim TempPath = Path.GetTempFileName() 'Noncompliant

    Using Writer As New StreamWriter(TempPath)
        Writer.WriteLine("content")
    End Using
End Sub
----


==== Compliant solution

[source,vbnet,diff-id=1,diff-type=compliant]
----
Imports System.IO

Sub Example()
    Dim RandomPath = Path.Combine(Path.GetTempPath(), Path.GetRandomFileName())

    Using FileStream As New FileStream(RandomPath, FileMode.CreateNew, FileAccess.Write, FileShare.None, 4096, FileOptions.DeleteOnClose)
        Using Writer As New StreamWriter(FileStream)
            Writer.WriteLine("content")
        End Using
    End Using
End Sub
----

=== How does this work?

include::../common/fix/introduction.adoc[]

include::../common/fix/manual-setup.adoc[]

Here the example compliant code uses the `Path.GetTempPath` and
`Path.GetRandomFileName` functions to generate a unique random file name. The
file is then open with the `FileMode.CreateNew` option that will ensure the
creation fails if the file already exists. The `FileShare.None` option will
additionally prevent the file from being opened again by any process. To finish,
this code ensures the file will get destroyed once the application has finished
using it with the `FileOptions.DeleteOnClose` option.

//=== Pitfalls

//=== Going the extra mile


== Resources

include::../common/resources/documentation.adoc[]

//=== Articles & blog posts
//=== Conference presentations

include::../common/resources/standards.adoc[]

//=== Benchmarks

ifdef::env-github,rspecator-view[]
'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]