While a C-style ``++enum++`` will work in Objective-C, the newer ``++NS_ENUM++`` and ``++NS_OPTIONS++`` macros offer a simple way to define enumerations and options, specifying the type and size for each, and improving code completion in Xcode.

== Noncompliant Code Example

----
typedef enum Letter {  // Noncompliant
  A,
  B, 
  C
} Letter;
----

== Compliant Solution

----
typedef NS_ENUM (NSInteger, Letter) {
  A,
  B, 
  C
} Letter;
----
