== Why is this an issue?

While a C-style ``++enum++`` will work in Objective-C, the newer ``++NS_ENUM++`` and ``++NS_OPTIONS++`` macros offer a simple way to define enumerations and options, specifying the type and size for each, and improving code completion in Xcode.


=== Noncompliant code example

[source,cpp]
----
typedef enum Letter {  // Noncompliant
  A,
  B, 
  C
} Letter;
----


=== Compliant solution

[source,cpp]
----
typedef NS_ENUM (NSInteger, Letter) {
  A,
  B, 
  C
} Letter;
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

endif::env-github,rspecator-view[]
