<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Shared coding conventions allow teams to collaborate efficiently. This rule checks that object attribute names match the provided regular expression.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="paragraph">
<p>With the default regular expression <code>[a-zA-Z](_?+[a-zA-Z0-9])*+</code> :</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-sql" data-lang="sql">CREATE TYPE my_type AS OBJECT(
  foo__bar INTEGER             -- Non-Compliant
);
/

DROP TYPE my_type;</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-sql" data-lang="sql">CREATE TYPE my_type AS OBJECT(
  foo_bar INTEGER              -- Compliant
);
/

DROP TYPE my_type;</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Rename this attribute to match the provided regular expression: xxx</p>
</div>
</div>
<div class="sect2">
<h3 id="_parameters">Parameters</h3>
<div class="sidebarblock">
<div class="content">
<div class="title">format</div>
<div class="listingblock">
<div class="content">
<pre>[a-zA-Z](_?+[a-zA-Z0-9])*+</pre>
</div>
</div>
<div class="paragraph">
<p>The regular expression the name should match</p>
</div>
</div>
</div>
</div>
</div>
</div>