<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Inexperienced Java developers might expect the <code>Object.equals(Object obj)</code> method to correctly handle the case where the left hand side is null, but that is not the case.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-text" data-lang="text">if (variable.equals(null)) { /* ... */ }   // Noncompliant - "variable" is really null, a NullPointerException is thrown</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-text" data-lang="text">if (variable == null) { /* ... */ }        // Compliant</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_relates_to_s2220">relates to: <a data-rspec-id="S2220" class="rspec-auto-link">S2220</a></h3>

</div>
<div class="sect2">
<h3 id="_on_15_oct_2013_090554_freddy_mallet_wrote">on 15 Oct 2013, 09:05:54 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>Is implemented by http://jira.codehaus.org/browse/SONARJAVA-360</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_2_feb_2015_202835_sébastien_gioria_wrote">on 2 Feb 2015, 20:28:35 Sébastien Gioria wrote:</h3>
<div class="paragraph">
<p>Part of OWASP Top10 2013 A6</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_3_feb_2015_202152_ann_campbell_wrote">on 3 Feb 2015, 20:21:52 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>I don&#8217;t understand the relationship [~sebastien.gioria]</p>
</div>
</div>
</div>
</div>