== How to fix it in AWS Identity and Access Management

=== Code examples

In this example, the IAM policy allows an attacker to update the code of any Lambda function. An attacker can achieve privilege escalation by altering the code of a Lambda that executes with high privileges.

==== Noncompliant code example

[source,terraform]
----
resource "aws_iam_policy" "example" {
  name = "example"
  policy =<<EOF
{
    "Version": "2012-10-17",
    "Statement": [
        {
            "Effect": "Allow",
            "Action": [
                "lambda:UpdateFunctionCode"
            ],
            "Resource": "*"
        }
    ]
}
EOF
}
----

==== Compliant solution


The policy is narrowed such that only updates to the code of certain Lambda functions (without high privileges) are allowed.

[source,terraform]
----
resource "aws_iam_policy" "example" {
  name = "example"
  policy =<<EOF
{
    "Version": "2012-10-17",
    "Statement": [
        {
            "Effect": "Allow",
            "Action": [
                "lambda:UpdateFunctionCode"
            ],
            "Resource": "arn:aws:lambda:us-east-2:123456789012:function:my-function:1"
        }
    ]
}
EOF
}
----

=== How does this work?

include::../../common/fix/least-privilege.adoc[]