<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>There&#8217;s no valid reason to test <code>this</code> with <code>instanceof</code>. The only plausible explanation for such a test is that you&#8217;re executing code in a parent class conditionally based on the kind of child class <code>this</code> is. But code that&#8217;s specific to a child class should be <em>in</em> that child class, not in the parent.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">public class JunkFood{
  public void doSomething() {
    if (this instanceof Pizza) {  // Noncompliant
      // ...
    } else if (...
  }
}</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Offload the code that&#8217;s conditional on this "instanceof" test to the appropriate subclass and remove the test.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_12_jun_2015_175304_ann_campbell_wrote">on 12 Jun 2015, 17:53:04 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>CodePro: Use of "instanceof" with "this"</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_12_jun_2015_175551_ann_campbell_wrote">on 12 Jun 2015, 17:55:51 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>FYI [~tamas.vajk]</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_15_jun_2015_201833_nicolas_peru_wrote">on 15 Jun 2015, 20:18:33 Nicolas Peru wrote:</h3>
<div class="paragraph">
<p>Looks good</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_16_jun_2015_095906_tamas_vajk_wrote">on 16 Jun 2015, 09:59:06 Tamas Vajk wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] I&#8217;ve added the C# specific parts. (The operator is called <code>is</code> in C#.)</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_16_jun_2015_112455_ann_campbell_wrote">on 16 Jun 2015, 11:24:55 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>Thanks [~tamas.vajk]. FYI, I&#8217;ve moved the C#-specific stuff to a subtask. I know that it seems a shame with so few tiny differences, but brackes &amp; pipes is what we used before we came up with creating subtasks &amp; eventually everything will be converted from brackets to subtasks.</p>
</div>
</div>
</div>
</div>