include::../description.adoc[]

== Noncompliant Code Example

----
var i:Int
for i in 1...10 {
    print("i is \(i)")
    continue  // this is meaningless; the loop would continue anyway
}
----

== Compliant Solution

----
var i:Int
for i in 1...10 {
    print("i is \(i)")
}
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../highlighting.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
