<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>When specifying base classes for a class, the access level is private by default. While private inheritance has its uses, it is far less used than public inheritance. Therefore:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>Either you want public inheritance, and you have to specify it</p>
</li>
<li>
<p>Or you want private inheritance, and you should be explicit about this design decision, by specifying it too.</p>
</li>
</ul>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">class B {
};

class C : B {
};</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">class B {
};

class C :  private B { // Or public, if it was private by mistake
};</code></pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_related_rules">Related rules</h3>
<div class="ulist">
<ul>
<li>
<p><a data-rspec-id="S5966" class="rspec-auto-link">S5966</a> is a similar rule that deals with base visibility in structs</p>
</li>
<li>
<p><a data-rspec-id="S3540" class="rspec-auto-link">S3540</a> is another rule that favors a different coding style for base classes. It should not be activated at the same time as this rule.</p>
</li>
</ul>
</div>
</div>
</div>
</div>