<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when a struct method doesn&#8217;t access any fields of the struct, suggesting it could be a standalone function instead.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Go emphasizes simplicity and directness in code design. When methods are defined on structs but don&#8217;t actually use the struct&#8217;s fields or maintain any meaningful state, they create unnecessary complexity.</p>
</div>
<div class="paragraph">
<p>This pattern often emerges when developers coming from object-oriented languages try to group related functionality using struct methods, similar to how they would use classes. However, Go&#8217;s philosophy favors functions over methods when no state management is involved.</p>
</div>
<div class="paragraph">
<p>Stateless struct methods introduce several problems:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>They add cognitive overhead by suggesting there&#8217;s state management when there isn&#8217;t</p>
</li>
<li>
<p>They require unnecessary struct instantiation</p>
</li>
<li>
<p>They make the code less idiomatic and harder to understand for Go developers</p>
</li>
<li>
<p>They can mislead other developers about the purpose and design of the struct</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>Go&#8217;s design encourages using the simplest construct that solves the problem. If functionality doesn&#8217;t require access to struct fields, a function is the more appropriate choice.</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="paragraph">
<p>Using struct methods unnecessarily reduces code clarity and maintainability. It creates false complexity that makes the codebase harder to understand and maintain. The extra indirection through struct instantiation also adds minor performance overhead without any benefit.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Convert the struct method to a standalone function. Remove the receiver parameter and make the function independent.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-go" data-lang="go">type Calculator struct{}

func (c Calculator) Add(a, b int) int { // Noncompliant
    return a + b
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-go" data-lang="go">func Add(a, b int) int {
    return a + b
}</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Effective Go - Methods - <a href="https://golang.org/doc/effective_go#methods">Official Go documentation on when and how to use methods effectively</a></p>
</li>
<li>
<p>Go Code Review Comments - Receiver Type - <a href="https://github.com/golang/go/wiki/CodeReviewComments#receiver-type">Guidelines on choosing appropriate receiver types for methods</a></p>
</li>
</ul>
</div>
</div>
<div class="sect2">
<h3 id="_standards">Standards</h3>
<div class="ulist">
<ul>
<li>
<p>Go Best Practices - Simplicity - <a href="https://golang.org/doc/effective_go#simplicity">Go&#8217;s philosophy emphasizing simplicity and avoiding unnecessary abstractions</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>