include::../description.adoc[]

== Noncompliant Code Example

----
WRITE 'Hello World'. WRITE 'You can use multiple statements on a line'.
----

== Compliant Solution

----
WRITE 'Hello World'. 
WRITE 'You can use multiple statements on a line'.
WRITE 'But you shouldn't.'.
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../highlighting.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
