Enabling case-insensitive matching using `(?i)` within a regex will only apply to characters in the ASCII range. To include characters outside of this range, use the `RegexOption.IGNORE_CASE` flag. Alternatively, you can use `(?iu)` or `(?iU)` within the regex instead.

If not done, regular expressions involving non-ASCII letters will still handle those letters as being case sensitive.


== Noncompliant Code Example

----
Regex("(?i)söme pättern")
Regex("(?i:söme) pättern")

"(?i)söme pättern".toRegex()
"(?i:söme) pättern".toRegex()
----


== Compliant Solution

----
Regex("söme pättern", RegexOption.IGNORE_CASE)
Regex("(?iu)söme pättern")
Regex("(?iu:söme) pättern")

"söme pättern".toRegex(RegexOption.IGNORE_CASE)
"(?iu)söme pättern".toRegex()
"(?iu:söme) pättern".toRegex()
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../highlighting.adoc[]

endif::env-github,rspecator-view[]
