<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This is an issue when a single-method interface doesn&#8217;t use the method name plus an "-er" suffix or similar modification to construct an agent noun.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Go has established naming conventions for interfaces, particularly single-method interfaces. According to the official Go documentation "Effective Go", single-method interfaces should be named by taking the method name and adding an "-er" suffix or similar modification to create an agent noun.</p>
</div>
<div class="paragraph">
<p>This convention serves several important purposes:</p>
</div>
<div class="ulist">
<ul>
<li>
<p><strong>Predictability</strong>: Developers can quickly understand what an interface does just by looking at its name</p>
</li>
<li>
<p><strong>Consistency</strong>: Following established patterns makes codebases more uniform and easier to navigate</p>
</li>
<li>
<p><strong>Readability</strong>: Agent nouns like <code>Reader</code>, <code>Writer</code>, <code>Formatter</code> immediately convey the interface&#8217;s purpose</p>
</li>
<li>
<p><strong>Community standards</strong>: Adhering to Go conventions makes code more familiar to other Go developers</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>When interfaces don&#8217;t follow this pattern, they become less intuitive. For example, <code>FileRead</code> doesn&#8217;t immediately suggest it&#8217;s an interface, while <code>Reader</code> clearly indicates both the action and that it&#8217;s likely an interface following Go conventions.</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="paragraph">
<p>Not following Go naming conventions reduces code readability and makes it harder for developers to understand the codebase quickly. It can also make the code feel inconsistent with standard Go libraries and community practices, potentially confusing team members who expect conventional naming patterns.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Rename the interface to follow Go conventions by using the method name plus an "-er" suffix or similar agent noun construction.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-go" data-lang="go">type FileRead interface {
    Read([]byte) (int, error)
} // Noncompliant</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-go" data-lang="go">type Reader interface {
    Read([]byte) (int, error)
}</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Effective Go - Interface names - <a href="https://go.dev/doc/effective_go#interface-names">Official Go documentation explaining interface naming conventions</a></p>
</li>
<li>
<p>Go Code Review Comments - Interface Names - <a href="https://github.com/golang/go/wiki/CodeReviewComments#interface-names">Go team&#8217;s code review guidelines for interface naming</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>