This rule raises an issue when complex nested anonymous structs are used in field definitions, making code harder to read, reuse, and maintain.

== Why is this an issue?

Anonymous structs can make code harder to understand and maintain, especially when they contain multiple fields or are nested within other structs.

When you use anonymous structs for complex data structures, you create several problems:

* **Poor readability**: Nested anonymous structs create deeply indented code that is difficult to scan and understand quickly.
* **No reusability**: Anonymous structs cannot be reused across different parts of your codebase, leading to code duplication.
* **Testing difficulties**: You cannot easily create instances of anonymous struct types in test files, often forcing you to duplicate the struct definition.
* **Limited documentation**: Anonymous structs cannot have their own methods or be documented separately, reducing code clarity.

Named types solve these issues by providing clear, reusable definitions that can be documented, tested, and maintained independently. They also make your code more modular and easier to refactor.

=== What is the potential impact?

Using anonymous structs for complex nested structures reduces code maintainability and readability. It can lead to code duplication across files, make testing more difficult, and create confusion for developers working with the codebase.

== How to fix it

Extract the anonymous struct into a named type. Define the named type separately and use it in your struct field definition.

=== Code examples

==== Noncompliant code example

[source,go,diff-id=1,diff-type=noncompliant]
----
type Service struct {
	Readiness *struct {
		TimeoutSeconds   int
		PeriodSeconds    int
		HttpGet          *struct { // Noncompliant
			Path string
			Port int
		}
	}
}
----

==== Compliant solution

[source,go,diff-id=1,diff-type=compliant]
----
type HTTPGetProbe struct {
	Path string
	Port int
}

type ReadinessProbe struct {
	TimeoutSeconds int
	PeriodSeconds  int
	HttpGet        *HTTPGetProbe
}

type Service struct {
	Readiness *ReadinessProbe
}
----

== Resources

=== Documentation

 * Go Language Specification - Struct types - https://golang.org/ref/spec#Struct_types[Official Go specification for struct types and anonymous structs]

 * Effective Go - Names - https://golang.org/doc/effective_go#names[Guidelines for naming conventions in Go, including type names]
