Some statement are required to have a name/label defined.

== Why is this an issue?

In JCL, some statement require to have a label or name defined, otherwise it will trigger a JCL error.
The statements that require a name/label are:
* CNTL
* ENDCNTL
* JOB
* NOTIFY
* OUTPUT

PROC can also require a name in case it is an in-stream procedure, otherwise it is optional for cataloged procedure.

== How to fix it

=== Code examples

==== Noncompliant code example

[source,text,diff-id=1,diff-type=noncompliant]
----
// JOB MYJOB
----

==== Compliant solution

[source,text,diff-id=1,diff-type=compliant]
----
//JOBNAME JOB MYJOB
----

== Resources

=== Documentation

* https://www.ibm.com/docs/en/zos/3.1.0?topic=description-label-field[IBM Reference - CNTL Label field]
* https://www.ibm.com/docs/en/zos/3.1.0?topic=d-label-field[IBM Reference - ENDCNTL Label field]
* https://www.ibm.com/docs/en/zos/3.1.0?topic=d-name-field-5[IBM Reference - JOB Name field]
* https://www.ibm.com/docs/en/zos/3.1.0?topic=statement-label-field[IBM Reference - NOTIFY Label field]
* https://www.ibm.com/docs/en/zos/3.1.0?topic=d-name-field-6[IBM Reference - OUTPUT Name field]
