<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>If a collection is declared and populated but its values are never read anywhere in the code, it can be considered unused. This often arises from incomplete refactoring, copy-pasting errors, or typos. Unused collections can lead to wasted memory and degraded application performance. Additionally, their presence makes the code harder to read and understand.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Remove unused collection.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">int getLength(String a, String b, String c) {
  List&lt;String&gt; strings = new ArrayList&lt;&gt;();  // Noncompliant: List is declared and populated but never read.
  strings.add(a);
  strings.add(b);
  strings.add(c);

  return a.length() + b.length() + c.length();
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">int getLength(String a, String b, String c) {
  return a.length() + b.length() + c.length();
}</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>