=== Noncompliant code example

[source,text]
----
public int getLength(String a, String b, String c) {
  ArrayList<String> strings = new ArrayList<>();  // Noncompliant
  strings.add(a);
  strings.add(b);
  strings.add(c);

  return a.length() + b.length() + c.length();
}
----
