<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when the expression used in an <code>except</code> block is a boolean expression of exceptions.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>The only two possible types for an <code>except</code>'s expression are a class deriving from <code>BaseException</code>, or a tuple composed of such classes.</p>
</div>
<div class="paragraph">
<p>Trying to catch multiple exception in the same <code>except</code> with a boolean expression of exceptions may not work as intended.
The result of a boolean expression of exceptions is a single exception class, thus using a boolean expression in an <code>except</code> block will result in catching only one kind of exception.</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">error = ValueError or TypeError
error is ValueError # True
error is TypeError # False

error = ValueError and TypeError
error is ValueError # False
error is TypeError # True</code></pre>
</div>
</div>
<div class="paragraph">
<p><strong>Note</strong>: <em>In Python 2 it is possible to raise an exception from an old-style class that does not derive from <code>BaseException</code>.</em></p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Make sure to use a tuple of the exceptions that should be caught in the <code>except</code> block.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">try:
    raise TypeError()
except ValueError or TypeError:  # Noncompliant
    print("Catching only ValueError")
except ValueError and TypeError:  # Noncompliant
    print("Catching only TypeError")
except (ValueError or TypeError) as exception:  # Noncompliant
    print("Catching only ValueError")

foo = ValueError or TypeError  # foo == ValueError
foo = ValueError and TypeError  # foo == TypeError</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">try:
    raise TypeError()
except (ValueError, TypeError) as exception:
    print("Catching ValueError and TypeError")</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p><a href="https://docs.python.org/3/reference/compound_stmts.html#except">the <code>try</code> statement</a> - Python try statement</p>
</li>
</ul>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Rewrite this "except" expression as a tuple of exception classes</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>The "except"'s expression</p>
</div>
</div>
</div>
</div>