If a closure expression is provided as the function or method’s only argument and you provide that expression as a trailing closure, you do not need to write a pair of parentheses ``++()++`` after the function or method’s name when you call the function. This makes the code somewhat easier to read.


== Noncompliant Code Example

----
reversedNames = names.sorted() { $0 > $1 } // Noncompliant
----


== Compliant Solution

----
eversedNames = names.sorted { $0 > $1 }
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

include::highlighting.adoc[]

endif::env-github,rspecator-view[]
