This rule raises an issue when the deprecated `reflect.Value.InterfaceData()` method is called.

== Why is this an issue?

The `reflect.Value.InterfaceData()` method has been deprecated since Go 1.4 and should not be used in modern Go code.

In earlier versions of Go, this method returned the interface's value as a uintptr pair. However, as of Go 1.4, changes to the implementation of interface values mean that `InterfaceData()` has no defined use. The memory representation of interface values is no longer compatible with the data returned by this method.

Using deprecated methods can lead to several problems:

* *Unpredictable behavior*: The method may not work as expected due to internal implementation changes
* *Future compatibility issues*: Deprecated methods may be removed in future Go versions
* *Maintenance burden*: Code using deprecated APIs becomes harder to maintain and update

The Go documentation explicitly states that the memory representation of interface values is not compatible with `InterfaceData`, making any use of this method unreliable and potentially dangerous.

=== What is the potential impact?

Using the deprecated `InterfaceData()` method can lead to unpredictable behavior and runtime issues. Since the method has no defined use and the underlying memory representation has changed, code relying on it may produce incorrect results or fail unexpectedly. Additionally, future Go versions may remove this method entirely, breaking code that depends on it.

== How to fix it

Replace the deprecated `InterfaceData()` method with the `Interface()` method, which safely returns the interface value. Always check if the value can be interfaced using `CanInterface()` to avoid panics.

=== Code examples

==== Noncompliant code example

[source,go,diff-id=1,diff-type=noncompliant]
----
import "reflect"

func getInterfaceData(v reflect.Value) [2]uintptr {
    return v.InterfaceData() // Noncompliant
}
----

==== Compliant solution

[source,go,diff-id=1,diff-type=compliant]
----
import "reflect"

func getInterface(v reflect.Value) interface{} {
    if v.CanInterface() {
        return v.Interface()
    }
    return nil
}
----

== Resources

=== Documentation

 * Go reflect package documentation - https://pkg.go.dev/reflect#Value.InterfaceData[Official Go documentation for the deprecated InterfaceData method]

 * Go reflect package - https://pkg.go.dev/reflect[Complete documentation for Go's reflect package]
