This is an issue when pagination methods like `paginate` are called but their results are not assigned, or when paginated collections are subsequently overwritten with regular arrays or ActiveRecord relations.

== Why is this an issue?

Pagination libraries like will_paginate and kaminari return special collection objects that extend regular arrays with pagination-specific methods such as `total_pages`, `current_page`, `per_page`, and `offset`. These methods are essential for rendering pagination controls in views.

When you call pagination methods without assigning their results, or when you overwrite a paginated collection with a regular array or ActiveRecord relation, you lose access to these pagination methods. This leads to runtime errors when your views try to access pagination information.

For example, calling `@locations.paginate(page: 1, per_page: 2)` without assignment means the paginated result is discarded, and `@locations` remains a regular array. Similarly, operations like `map`, `select`, or additional ActiveRecord queries after pagination will return regular collections, replacing the paginated one.

This issue is particularly common when developers chain operations or perform additional filtering after pagination, not realizing that these operations destroy the pagination wrapper.

=== What is the potential impact?

When paginated collections are overwritten, views that depend on pagination methods will fail with `NoMethodError` exceptions. This breaks the user interface and can cause application crashes in production.

Additionally, losing pagination functionality means users cannot navigate through large datasets effectively, leading to poor user experience and potential performance issues if all records are loaded at once.

== How to fix it

Always assign the result of pagination methods to preserve the paginated collection object.

=== Code examples

==== Noncompliant code example

[source,ruby,diff-id=1,diff-type=noncompliant]
----
@locations = Location.find_all_by_city("springfield")
@locations.paginate(page: 1, per_page: 2)  # Noncompliant
----

==== Compliant solution

[source,ruby,diff-id=1,diff-type=compliant]
----
@locations = Location.where(city: "springfield")
@locations = @locations.paginate(page: params[:page], per_page: 10)
----

== Resources

=== Documentation

 * will_paginate documentation - https://github.com/mislav/will_paginate[Official documentation for the will_paginate gem]

 * Kaminari documentation - https://github.com/kaminari/kaminari[Official documentation for the Kaminari pagination gem]
