<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when the <code>reversed()</code> function is called with a <code>sorted()</code> as an argument.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>The <code>sorted()</code> function has a <code>reverse</code> parameter that provides the same functionality as the <code>reversed()</code> function.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Use the <code>reverse</code> parameter of the <code>sorted()</code> function to sort in descending order instead of using <code>reversed()</code>.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">data = [3, 1, 4, 1, 5, 9]
result = reversed(sorted(data)) # Noncompliant</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">data = [3, 1, 4, 1, 5, 9]
result = sorted(data, reverse=True)</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Python reference documentation - <a href="https://docs.python.org/3/library/functions.html#sorted">sorted</a></p>
</li>
<li>
<p>Python reference documentation - <a href="https://docs.python.org/3/library/functions.html#reversed">reversed</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>