<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule reports when a source file does not adhere to the architectural constraints defined for a project.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>There is an import from one source file to another that is prohibited by the architectural constraints defined for the project.
These constraints are established by the project maintainers in the architecture configuration file.</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="paragraph">
<p>Over time, codebases often drift from the intended project architecture,
gradually degrading the code structure and misaligning it with the original design.</p>
</div>
<div class="paragraph">
<p>This misalignment reduces visibility and control over daily decisions affecting the architecture.
As these small decisions accumulate, the codebase becomes harder to understand, and the architecture grows increasingly complex and unstructured.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Refactor your source code to adhere to the architectural constraints.
The specific approach will depend on your project architecture and the code not adhering to the constraints.
For example, you might replace an import with an alternative import from another module
or create a new function in the appropriate module to maintain a clean architecture.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="paragraph">
<p>Assuming that files in the directory <code>./src/panels</code> should not directly access files in the directory <code>./src/repos</code>:</p>
</div>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-javascript" data-lang="javascript">// ./src/panels/showCustomersPanel.js
import { CustomerRepo } from '../repos/customerRepo.js';

class ShowCustomersPanel {
    constructor() {
        this.customerRepo = new CustomerRepo();
    }

    getCustomers() {
        return this.customerRepo.findAll();
    }
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-javascript" data-lang="javascript">// ./src/panels/showCustomersPanel.js
import { CustomerService } from '../services/customerService.js';

class ShowCustomersPanel {
    constructor() {
        this.customerService = new CustomerService();
    }

    getCustomers() {
        return this.customerService.getAllCustomers();
    }
}</code></pre>
</div>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Defining architectural constraints for SonarQube</p>
</li>
</ul>
</div>
</div>
</div>
</div>