include::../description.adoc[]

== Noncompliant Code Example

----
from django.http import HttpResponse

def index(request):
    key = request.GET.get("key")
    value = request.GET.get("value")
    response = HttpResponse("")
    response[key] = value  # Noncompliant
    response.set_cookie(key, value)  # Noncompliant
    return response
----

== Compliant Solution

----
from django.http import HttpResponse

def index(request):
    value = request.GET.get("value")
    response = HttpResponse("")
    response["X-Data"] = value
    response.set_cookie("data", value)
    return response
----

include::../see.adoc[]
