<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when a <code>torch.utils.data.Dataloader</code> is instantiated without specifying the <code>num_workers</code> parameter.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>In the PyTorch library, the data loaders are used to provide an interface where common operations such as batching can be implemented.
It is also possible to parallelize the data loading process by using multiple worker processes.
This can improve performance by increasing the number of batches being fetched in parallel, at the cost of higher memory usage.
This performance increase can also be attributed to avoiding the Global Interpreter Lock (GIL) in the Python interpreter.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Specify the <code>num_workers</code> parameter when instantiating the <code>torch.utils.data.Dataloader</code> object.</p>
</div>
<div class="paragraph">
<p>The default value of <code>0</code> will use the main process to load the data, and might be faster for small datasets that can fit completely in memory.</p>
</div>
<div class="paragraph">
<p>For larger datasets, it is recommended to use a value of <code>1</code> or higher to parallelize the data loading process.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">from torch.utils.data import DataLoader
from torchvision import datasets
from torchvision.transforms import ToTensor

train_dataset = datasets.MNIST(root='data', train=True, transform=ToTensor())
train_data_loader = DataLoader(train_dataset, batch_size=32)# Noncompliant: the num_workers parameter is not specified</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">from torch.utils.data import DataLoader
from torchvision import datasets
from torchvision.transforms import ToTensor

train_dataset = datasets.MNIST(root='data', train=True, transform=ToTensor())
train_data_loader = DataLoader(train_dataset, batch_size=32, num_workers=4)</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>PyTorch documentation - <a href="https://pytorch.org/docs/stable/data.html#single-and-multi-process-data-loading">Single- and Multi-process Data Loading</a></p>
</li>
<li>
<p>PyTorch documentation - <a href="https://pytorch.org/tutorials/beginner/basics/data_tutorial.html">Datasets and DataLoaders</a></p>
</li>
</ul>
</div>
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation specification</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Primary : Specify the <code>num_workers</code> parameter.</p>
</div>
</div>
<div class="sect2">
<h3 id="_issue_location">Issue location</h3>
<div class="paragraph">
<p>Primary : Name of the instantiation</p>
</div>
</div>
<div class="sect2">
<h3 id="_quickfix">Quickfix</h3>
<div class="paragraph">
<p>Fill in with the default parameter</p>
</div>
</div>
</div>
</div>