<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>When links with different targets are given identical link text, this can produce confusion for users of assistive technologies, some of which provide users the ability to view a list of all links on the page.</p>
</div>
<div class="paragraph">
<p>When this list of links is presented to the user they may be left not knowing the links go to different destinations.</p>
</div>
<div class="paragraph">
<p>Even if they do realize the links go to different destinations, they may be left not knowing which link to follow to go to the destination they desire.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-html" data-lang="html">&lt;a href="a.html"&gt;qux&lt;/a&gt;
&lt;a href="b.html"&gt;qux&lt;/a&gt;          &lt;!-- Noncompliant; same text, different targets --&gt;

&lt;a href="c.html"&gt;foo&lt;/a&gt;
&lt;a href="d.html"&gt;foo&lt;/a&gt;          &lt;!-- Noncompliant; same text, different targets --&gt;</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-html" data-lang="html">&lt;a href="a.html"&gt;qux&lt;/a&gt;
&lt;a href="b.html"&gt;qaz&lt;/a&gt;          &lt;!-- Compliant; different text, different targets --&gt;

&lt;a href="c.html"&gt;foo&lt;/a&gt;
&lt;a href="c.html"&gt;foo&lt;/a&gt;          &lt;!-- Compliant; same text, same targets --&gt;</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Use distinct texts or point to the same target for this link and the one at line X.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_24_jun_2013_135159_dinesh_bolkensteyn_wrote">on 24 Jun 2013, 13:51:59 Dinesh Bolkensteyn wrote:</h3>
<div class="paragraph">
<p>SQALE to be added</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_8_jul_2013_181811_freddy_mallet_wrote">on 8 Jul 2013, 18:18:11 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>Is implemented by http://jira.codehaus.org/browse/SONARPLUGINS-3010</p>
</div>
</div>
</div>
</div>