== Why is this an issue?

Browsers best support the ``++window++`` mode for the ``++wmode++`` parameter, also in terms of accessibility.

As it is the default mode, it is acceptable to either not specify a ``++wmode++`` parameter altogether, or to set it explicitly to ``++window++``.


=== Noncompliant code example

[source,html]
----
<object classid="clsid:d27cdb6e-ae6d-11cf-96b8-444553540000" width="550" height="400">
  <param name="movie" value="movie_name.swf" />
  <param name="wmode" value="direct" />                              <!-- Non-Compliant -->
</object>

<embed src="movie_name.swf"
       width="550"
       height="400"
       wmode="direct"                                                                 <!-- Non-Compliant -->
       type="application/x-shockwave-flash"
       pluginspage="http://www.macromedia.com/go/getflashplayer" />
----


=== Compliant solution

[source,html]
----
<object classid="clsid:d27cdb6e-ae6d-11cf-96b8-444553540000" width="550" height="400">
  <param name="movie" value="movie_name.swf" />
</object>

<embed src="movie_name.swf"
       width="550"
       height="400"
       type="application/x-shockwave-flash"
       pluginspage="http://www.macromedia.com/go/getflashplayer" />
----

or


[source,html]
----
<object classid="clsid:d27cdb6e-ae6d-11cf-96b8-444553540000" width="550" height="400">
  <param name="movie" value="movie_name.swf" />
  <param name="wmode" value="window" />
</object>

<embed src="movie_name.swf"
       width="550"
       height="400"
       wmode="window"
       type="application/x-shockwave-flash"
       pluginspage="http://www.macromedia.com/go/getflashplayer" />
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Set the value of the 'wmode' parameter to 'window'.


'''
== Comments And Links
(visible only on this page)

=== on 24 Jun 2013, 08:52:07 Dinesh Bolkensteyn wrote:
http://helpx.adobe.com/flash/kb/flash-object-embed-tag-attributes.html#main_Browser_support_for_Window_Mode__wmode__values

=== on 8 Jul 2013, 18:24:05 Freddy Mallet wrote:
Is implemented by \http://jira.codehaus.org/browse/SONARPLUGINS-2988

endif::env-github,rspecator-view[]
