== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,csharp]
----
public void DoSomething(string[] strings)
{
  for (int i = 0; i < strings.Length; i--)  // Noncompliant
  {
    string s = strings[i];  // IndexOutOfRangeException  when i reaches -1
    // do stuff
  }
}
----

=== Compliant solution

[source,csharp]
----
public void DoSomething(string[] strings)
{
  for (int i = 0; i < strings.Length; i++)
  {
    string s = strings[i];
    // do stuff
  }
}
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
