The use of ``++Option Base++`` to change the lower bound of an array's index values results in confusing code.


== Noncompliant Code Example

----
Option Explicit
Option Base 1

'...
    Dim MyArray(1 To 3) As Integer
    For I = 1 To 3
        MsgBox MyArray(I)
    Next I
----


== Compliant Solution

----
Option Explicit

'...
    Dim MyArray(0 To 2) As Integer
    For I = 0 To 2
        MsgBox MyArray(I)
    Next I
----


