<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>The use of <code>Option Base</code> to change the lower bound of an array&#8217;s index values results in confusing code.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-vb6" data-lang="vb6">Option Explicit
Option Base 1

'...
    Dim MyArray(1 To 3) As Integer
    For I = 1 To 3
        MsgBox MyArray(I)
    Next I</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-vb6" data-lang="vb6">Option Explicit

'...
    Dim MyArray(0 To 2) As Integer
    For I = 0 To 2
        MsgBox MyArray(I)
    Next I</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove this statement</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_9_apr_2014_134538_pierre_yves_nicolas_wrote">on 9 Apr 2014, 13:45:38 Pierre-Yves Nicolas wrote:</h3>
<div class="paragraph">
<p>Would the following examples make more sense?</p>
</div>
<div class="paragraph">
<p>Noncompliant:</p>
</div>
<div class="listingblock">
<div class="content">
<pre>Option Explicit
Option Base 1

'...
    Dim MyArray(3) As Integer
    For I = 1 To 3
        MsgBox MyArray(I)
    Next I</pre>
</div>
</div>
<div class="paragraph">
<p>Compliant:</p>
</div>
<div class="listingblock">
<div class="content">
<pre>Option Explicit

'...
    Dim MyArray(2) As Integer
    For I = 0 To 2
        MsgBox MyArray(I)
    Next I</pre>
</div>
</div>
</div>
</div>
</div>