<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>The integrator function of stream gatherers can pass data downstream using the push method.
The down stream should return a result indicating whether it accepted the latest value.</p>
</div>
<div class="paragraph">
<p>Ignoring the value returned by <code>Downstream.push</code> might lead to inconsistent behavior of the integrator.
Indeed, if <code>downstream.push</code> returns <code>false</code>, and no longer accepts elements, the decision to continue processing the incoming stream should be deliberate.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Use or return the result of <code>Downstream.push</code>.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">Gatherer.ofSequential(
  () -&gt; new AtomicInteger(100),
  (state, element, downstream) -&gt; {
      int newState = state.addAndGet(element);
      downstream.push(newState); // Noncompliant: ignoring return value
      return true;
  });</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">Gatherer.ofSequential(
  () -&gt; new AtomicInteger(100),
  (state, element, downstream) -&gt; {
      int newState = state.addAndGet(element);
      return downstream.push(newState); // Compliant
  });</code></pre>
</div>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">Gatherer.ofSequential(
  () -&gt; new AtomicInteger(100),
  (state, element, downstream) -&gt; {
      int newState = state.addAndGet(element);
      if (!downstream.push(newState)) {
        // Since the downstream is no longer accepting elements, we stop processing incoming streams
        return false;
      }
      return true;
  });</code></pre>
</div>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">Gatherer.ofSequential(
  () -&gt; new AtomicInteger(100),
  (state, element, downstream) -&gt; {
      int newState = state.addAndGet(element);
      return handleResult(downstream.push(newState));
  });</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_articles_blog_posts">Articles &amp; blog posts</h3>
<div class="ulist">
<ul>
<li>
<p>Oracle Java 24 Core Libraries Documentation - <a href="https://docs.oracle.com/en/java/javase/24/core/stream-gatherers.html">Stream Gatherers</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>