This rule raises an issue when a NumPy function expecting a square matrix is called with a non-square matrix.

== Why is this an issue?

Some matrix operations only make sense on square matrices. For example, a determinant can be computed only if a matrix is square.

Therefore, NumPy functions performing these operations should be provided with a square matrix as an argument. If something other than a square matrix is provided, the computation does not make sense and an exception will be raised.

The following functions will raise an exception when provided with a non-square matrix:

* ``++np.linalg.det(a)++``: computes the determinant of an matrix.
* ``++np.linalg.eig(a)++``: computes the eigenvalues and right eigenvectors of a square matrix.
* ``++np.linalg.matrix_power(a, n)++``: raises a square matrix to the (integer) power n.
* ``++np.linalg.solve(a, b)++``: Solve a linear matrix equation, or system of linear scalar equations. ``++a++`` should be square.
* ``++np.linalg.inv(a)++``: computes the (multiplicative) inverse of a matrix.
** Also requires a non-singular matrix (a matrix is singular if its determinant is 0).

You can use ``++numpy.shape()++`` function to check the dimensions of the matrix before performing the operation.

=== Code examples

==== Noncompliant code example

[source,python,diff-id=1,diff-type=noncompliant]
----
import numpy as np

a = np.array([[1, 2], [3, 4], [5, 6]])
b = np.linalg.inv(a) # Noncompliant: LinAlgError: Last 2 dimensions of the array must be square
----


==== Compliant solution

[source,python,diff-id=1,diff-type=compliant]
----
import numpy as np

a = np.array([[1, 2], [3, 4]])
b = np.linalg.inv(a) # Compliant
----

== Resources
=== Documentation
* NumPy Documentation - https://numpy.org/doc/stable/reference/generated/numpy.linalg.det.html[numpy.linalg.det]
* NumPy Documentation - https://numpy.org/doc/stable/reference/generated/numpy.linalg.eig.html[numpy.linalg.eig]
* NumPy Documentation - https://numpy.org/doc/stable/reference/generated/numpy.linalg.inv.html[numpy.linalg.inv]
* NumPy Documentation - https://numpy.org/doc/stable/reference/generated/numpy.linalg.matrix_power.html[numpy.linalg.matrix_power]
* NumPy Documentation - https://numpy.org/doc/stable/reference/generated/numpy.linalg.solve.html[numpy.linalg.solve]
* NumPy Documentation - https://numpy.org/doc/stable/reference/generated/numpy.shape.html[numpy.shape]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)



'''
== Comments And Links
(visible only on this page)

endif::env-github,rspecator-view[]
