This rule raises an issue when the shape of an array doesn't meet the square shape requirements of a NumPy function it passed to.

== Why is this an issue?

Certain NumPy operations are sensitive to the shapes and dimensions of matrices, as these properties affect how the operations are performed.
There are some functions requires an argument matrix to be square, otherwise it will throw an exception:

* ``++np.linalg.det(a)++``: compute the determinant of an array.
* ``++np.linalg.eig(a)++``: compute the eigenvalues and right eigenvectors of a square array.
* ``++np.linalg.inv(a)++``: compute the (multiplicative) inverse of a matrix.
** Also requires a non-singular matrix (a matrix is singular if its determinant is 0)
* ``++np.linalg.matrix_power(a, n)++``: raise a square matrix to the (integer) power n.
* ``++np.linalg.solve(a, b)++``: Solve a linear matrix equation, or system of linear scalar equations. ``++a++`` should be square.

You can use ``++numpy.shape()++`` function to check the dimensions of the arrays before performing the operation.

=== Code examples

==== Noncompliant code example

[source,python,diff-id=1,diff-type=noncompliant]
----
import numpy as np

a = np.array([[1, 2], [3, 4], [5, 6]])
b = np.linalg.inv(a) # Noncompliant: LinAlgError: Last 2 dimensions of the array must be square
----


==== Compliant solution

[source,python,diff-id=1,diff-type=compliant]
----
import numpy as np

a = np.array([[1, 2], [3, 4]])
b = np.linalg.inv(a) # Compliant
----

== Resources
=== Documentation
* NumPy Documentation - https://numpy.org/doc/stable/reference/generated/numpy.linalg.det.html[numpy.linalg.det]
* NumPy Documentation - https://numpy.org/doc/stable/reference/generated/numpy.linalg.eig.html[numpy.linalg.eig]
* NumPy Documentation - https://numpy.org/doc/stable/reference/generated/numpy.linalg.inv.html[numpy.linalg.inv]
* NumPy Documentation - https://numpy.org/doc/stable/reference/generated/numpy.linalg.matrix_power.html[numpy.linalg.matrix_power]
* NumPy Documentation - https://numpy.org/doc/stable/reference/generated/numpy.linalg.solve.html[numpy.linalg.solve]
* NumPy Documentation - https://numpy.org/doc/stable/reference/generated/numpy.shape.html[numpy.shape]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)



'''
== Comments And Links
(visible only on this page)

endif::env-github,rspecator-view[]
