This rule raises an issue when the shape of an array doesn't meet the requirements of a NumPy function it passed to.

== Why is this an issue?

Certain NumPy operations are sensitive to the shapes and dimensions of matrices, as these properties affect how the operations are performed. Here's a list of NumPy operations that are sensitive to matrix shapes:

* Requires a square matrix:
** ``++np.trace(a)++``
** ``++np.linalg.det(a)++``
** ``++np.linalg.eig(a)++``
** ``++np.linalg.inv(a)++``: also requires a non-singular matrix
** ``++np.linalg.matrix_power(a, n)++``
** ``++np.linalg.solve(a, b)++``: also requires compatible dimensions between ``++a++`` and ``++b++``.

You can use ``++numpy.shape()++`` function to check the dimensions of the arrays before performing the operation.

=== Code examples

==== Noncompliant code example

[source,python,diff-id=1,diff-type=noncompliant]
----
import numpy as np

a = np.array([[1, 2], [3, 4], [5, 6]])
b = np.linalg.inv(a) # Noncompliant: LinAlgError: Last 2 dimensions of the array must be square
----


==== Compliant solution

[source,python,diff-id=1,diff-type=compliant]
----
import numpy as np

a = np.array([[1, 2], [3, 4]])
b = np.linalg.inv(a) # Compliant
----

== Resources
=== Documentation
* NumPy Documentation - https://numpy.org/doc/stable/reference/generated/numpy.trace.html[numpy.trace]
* NumPy Documentation - https://numpy.org/doc/stable/reference/generated/numpy.linalg.det.html[numpy.linalg.det]
* NumPy Documentation - https://numpy.org/doc/stable/reference/generated/numpy.linalg.eig.html[numpy.linalg.eig]
* NumPy Documentation - https://numpy.org/doc/stable/reference/generated/numpy.linalg.inv.html[numpy.linalg.inv]
* NumPy Documentation - https://numpy.org/doc/stable/reference/generated/numpy.linalg.matrix_power.html[numpy.linalg.matrix_power]
* NumPy Documentation - https://numpy.org/doc/stable/reference/generated/numpy.linalg.solve.html[numpy.linalg.solve]
* NumPy Documentation - https://numpy.org/doc/stable/reference/generated/numpy.shape.html[numpy.shape]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)



'''
== Comments And Links
(visible only on this page)

endif::env-github,rspecator-view[]
