include::../dotnet/introduction.adoc[]

== Why is this an issue?

include::../dotnet/why.adoc[]

== How to fix it

include::../dotnet/how.adoc[]

=== Code examples

==== Noncompliant code example

[source,vbnet,diff-id=1,diff-type=noncompliant]
----
Public Class Account
    Public Property Id As DateTime
	
    Public Property Name As String
    Public Property Surname As String
End Class
----

==== Compliant solution

[source,vbnet,diff-id=1,diff-type=compliant]
----
Public Class Account
    Public Property Id As Guid
	
    Public Property Name As String
    Public Property Surname As String
End Class
----

or

==== Noncompliant code example

[source,vbnet,diff-id=2,diff-type=noncompliant]
----
Public Class Person
    <Key>
    Public Property BirthDate As DateTime
	
    Public Property Name As String
    Public Property Surname As String
End Class
----

==== Compliant solution

[source,vbnet,diff-id=2,diff-type=compliant]
----
Public Class Person
    Public Property Id As Guid
	
    Public Property BirthDate As DateTime
    Public Property Name As String
    Public Property Surname As String
End Class
----

== Resources

include::../dotnet/resources.adoc[]