include::../rationale.adoc[]

== How to fix it in ARM Templates

Remove the unused variable, after ensuring it is not part of an incomplete implementation leading to bugs.

=== Code examples

==== Noncompliant code example

[source,json,diff-id=1,diff-type=noncompliant]
----
{
  "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#",
  "variables": {
      "virtualMachinesName": "[concat('virtualMachinesResource', uniqueString(resourceGroup().id))]",
      "unusedVariable": "unusedValue"
  },
  "resources": [
    {
      "type": "Microsoft.Compute/virtualMachines",
      "name": "[variables('virtualMachinesName')]",
      "apiVersion": "2023-09-01",
      "location": "[resourceGroup().location]"
    }
  ]
}
----

==== Compliant solution

[source,json,diff-id=1,diff-type=compliant]
----
{
  "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#",
  "variables": {
      "virtualMachinesName": "[concat('virtualMachinesResource', uniqueString(resourceGroup().id))]"
  },
  "resources": [
    {
      "type": "Microsoft.Compute/virtualMachines",
      "name": "[variables('virtualMachinesName')]",
      "apiVersion": "2023-09-01",
      "location": "[resourceGroup().location]"
    }
  ]
}
----

== How to fix it in Bicep

Remove the unused variable, after ensuring it is not part of an incomplete implementation leading to bugs.

=== Code examples

==== Noncompliant code example

[source,bicep,diff-id=2,diff-type=noncompliant]
----
var storageName = '${toLower(storageNamePrefix)}${uniqueString(resourceGroup().id)}'
var unusedVariable = 'unusedValue' # Noncompliant

resource demoAccount 'Microsoft.Compute/virtualMachines@2023-09-01' = {
  name: virtualMachinesName
  location: resourceGroup().location
}
----

==== Compliant solution

[source,bicep,diff-id=2,diff-type=compliant]
----
var storageName = '${toLower(storageNamePrefix)}${uniqueString(resourceGroup().id)}'

resource demoAccount 'Microsoft.Compute/virtualMachines@2023-09-01' = {
  name: virtualMachinesName
  location: resourceGroup().location
}
----



== Resources

=== Documentation

* Microsoft - https://learn.microsoft.com/en-us/azure/azure-resource-manager/templates/best-practices#variables[ARM template best practices]
* Microsoft - https://learn.microsoft.com/en-us/azure/azure-resource-manager/bicep/best-practices#names[Best practices for Bicep]
* Microsoft - https://learn.microsoft.com/en-us/azure/azure-resource-manager/templates/variables[Variables in ARM templates]
* Microsoft - https://learn.microsoft.com/en-us/azure/azure-resource-manager/bicep/variables[Variables in Bicep]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
