<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p><code>Exit Function</code>, <code>Exit Property</code>, and <code>Exit Sub</code> are all poor, less-readable substitutes for a simple <code>Return</code>, and if used with code that should return a value (<code>Exit Function</code> and in some cases <code>Exit Property</code>) they could result in a <code>NullReferenceException</code>.</p>
</div>
<div class="paragraph">
<p>This rule raises an issue for all their usages.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-vbnet" data-lang="vbnet">Public Class Sample

  Public Sub MySub(Condition As Boolean)
    If Condition Then Exit Sub                  ' Noncompliant
    ' ...
  End Sub

  Public Function MyFunction(Condition As Boolean) As Integer
    If Condition Then
        MyFunction = 42
        Exit Function              ' Noncompliant
    End If
    ' ...
  End Function

End Class</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-vbnet" data-lang="vbnet">Public Class Sample

  Public Sub MySub(Condition As Boolean)
    If Condition Then Return                  ' Noncompliant
    ' ...
  End Sub

  Public Function MyFunction(Condition As Boolean) As Integer
    If Condition Then Return 42
    ' ...
  End Function

End Class</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove this "Exit" statement.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>Exit [xxx]</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_relates_to_s1227">relates to: <a data-rspec-id="S1227" class="rspec-auto-link">S1227</a></h3>

</div>
</div>
</div>