``++SELECT++`` can contain an ``++OTHERWISE++`` clause for various reasons: to handle unexpected values, to show that all the cases were properly considered.

For readability purpose, to help a developer to quickly find the default behavior of a ``++SELECT++`` statement, it is recommended to put the ``++OTHERWISE++`` clause at the end of the ``++SELECT++`` statement. This rule raises an issue if the ``++OTHERWISE++`` clause is not the first or the last one of the ``++SELECT++``'s cases.

== Noncompliant Code Example

----
foo: proc options(main);
  declare i fixed decimal init (42);

  select (i);
    when (0) put list ('i = 0');
    otherwise put list ('i = 2'); /* Non-Compliant */
    when (1) put list ('i = 1');
  end;
end;
----

== Compliant Solution

----
foo: proc options(main);
  declare i fixed decimal init (42);

  select (i); /* Compliant */
    when (0) put list ('i = 0');
    when (1) put list ('i = 1');
    otherwise put list ('i = 2');
  end;
end;
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
