== Why is this an issue?

include::description.adoc[]

=== Noncompliant code example

[source,text]
----
public void doTheDatabaseThing(ComplexObject cObj) {
  // password is red!    <-- Noncompliant
  // ...
}
----


include::see.adoc[]
