<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Exception handling provides a way to help developers handle and recover from runtime errors and unexpected events during program execution.
When an error occurs, an exception is thrown and can be caught by an appropriate try-catch block, allowing the program to handle the exception and prevent the program from crashing.</p>
</div>
<div class="paragraph">
<p>A problem arises, when an exception or error class&#8201;&#8212;&#8201;an instance of <code>Throwable</code>&#8201;&#8212;&#8201;in a catch clause does not exist, as PHP won&#8217;t generate an error.</p>
</div>
<div class="paragraph">
<p>This typically occurs when being in a namespace trying to catch PHP built-in exception classes without escaping to the global namespace or importing the classes.</p>
</div>
<div class="paragraph">
<p>In summary, this rule raises an issue when, being in a namespace, an undefined class belonging to that namespace is caught.</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="paragraph">
<p>Initially, since the exception is not being caught, it will cause a runtime error in production.</p>
</div>
<div class="paragraph">
<p>Additionally, this can lead to complex debugging problems as the root cause of the issue may not be obvious.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it_in_core_php">How to fix it in Core PHP</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Escape the caught exception to the global namespace or import the class.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-php" data-lang="php">namespace Foo\Bar;

try {
    doSomething();
} catch (Exception $e) { // Noncompliant - Exception will never be caught because the class Exception does not exist in the namespace
    echo $e-&gt;message;
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-php" data-lang="php">namespace Foo\Bar;

try {
    doSomething();
} catch (\Exception $e) { // Compliant: Used by global namespace
    echo $e-&gt;message;
}</code></pre>
</div>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-php" data-lang="php">namespace Foo\Bar;

use Exception;

try {
    doSomething();
} catch (Exception $e) { // Compliant: Imported by use statement
    echo $e-&gt;message;
}</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p><a href="https://www.php.net/manual/en/language.exceptions.php">PHP Manual - Exceptions</a></p>
</li>
</ul>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Create class {Exception} or check correct import of class.</p>
</div>
</div>
</div>
</div>