== Why is this an issue?

This rule enforces the idea to use ES6 style to create React Component instead of the ES5 way.


=== Noncompliant code example

[source,javascript]
----
var Rating = createReactClass({ // Noncompliant
  [...]
});
----


=== Compliant solution

[source,javascript]
----
export default class Rating extends React.Component { // Compliant
[...]
}
----


ifdef::env-github,rspecator-view[]
'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
