== Why is this an issue?

A ``++NullPointerException++`` should indicate that a ``++null++`` value was unexpectedly encountered. Good programming practice dictates that code is structured to avoid NPE's. 


Explicitly throwing ``++NullPointerException++`` forces a method's callers to explicitly catch it, rather than coding to avoid it. Further, it makes it difficult to distinguish between the unexpectedly-encountered ``++null++`` value and the condition which causes the method to purposely throw an NPE.


If an NPE is being thrown to indicate that a parameter to the method should not have been null, use the ``++@NotNull++`` annotation instead.


=== Noncompliant code example

[source,text]
----
public void doSomething (String aString) throws NullPointerException {
     throw new NullPointerException();
}
----


=== Compliant solution

[source,text]
----
public void doSomething (@NotNull String aString) {
}
----


