== Why is this an issue?

include::../why-dotnet.adoc[]

== How to fix it

include::../how-dotnet.adoc[]

=== Code examples

==== Noncompliant code example

[source,vbnet]
----
Module Module1
    Sub WriteMatrix(ByVal matrix As Integer()()) ' Non-Compliant
        ' ...
    End Sub
End Module
----

==== Compliant solution

[source,vbnet]
----
Class Matrix
    ' ...
End Class

Module Module1
    Sub WriteMatrix(ByVal matrix As Matrix)      ' Compliant
        ' ...
    End Sub
End Module
----

include::../resources-dotnet.adoc[]

include::../rspecator.adoc[]
