<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when increment (<code>&#43;&#43;</code>) or decrement (<code>--</code>) operators are used with floating-point variables.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Increment and decrement operators (<code>&#43;&#43;</code> and <code>--</code>) shouldn&#8217;t be used with floating-point variables (like <code>float</code> or <code>double</code>).
While the language allows it, the usage is not idiomatic, and most developers intuitively expect <code>x&#43;&#43;</code> to apply to integer types.
Using it on a float violates this common expectation and can lead to misleading code.</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="paragraph">
<p>Floating-point arithmetic has some non-intuitive properties, which can lead to unexpected bugs.
For example, the following loop will not terminate. This happens because <code>float</code> has only 24 bits of precision (mantissa) and once a number gets large enough, adding <code>1.0</code> becomes insignificant and the increment operation does nothing:</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">for (float x = 16_000_000; x &lt; 17_000_000; x++) {
  // ...
}
// The loop does not terminate</code></pre>
</div>
</div>
<div class="paragraph">
<p>The problem would not occur if <code>int</code> is used instead of <code>float</code>, even though both types occupy 32 bits:</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">for (int x = 16_000_000; x &lt; 17_000_000; x++) {
  // ...
}
// This loop terminates.</code></pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Using the compound assignment operators (<code>&#43;=</code> and <code>-=</code>) makes the intent clearer and avoids the surprising use of <code>&#43;&#43;</code> and <code>--</code> on floating-point types.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">float x = 0f;
double y = 1.0;

x++;
y--;</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">float x = 0f;
double y = 1.0;

x += 1.0;
y -= 1.0;</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>