include::../why-dotnet.adoc[]

== How to fix it

Use the dedicated overload that takes the log level, event id, and/or exception as arguments.

=== Noncompliant code example

[source,csharp,diff-id=1,diff-type=noncompliant]
----
try { }
catch (Exception ex)
{
    logger.LogDebug("An exception occured {Exception} with {EventId}.", ex, eventId); // Noncompliant
}
----

=== Compliant solution

[source,csharp,diff-id=1,diff-type=compliant]
----
try { }
catch (Exception ex)
{
    logger.LogDebug(eventId, ex, "An exception occured.");
}
----

include::../exceptions-dotnet.adoc[]

[source,csharp]
---- 
try { }
catch (Exception ex)
{
    logger.LogDebug(ex, "An exception occured {Exception}.", ex); // Compliant
}
----

include::../resources-dotnet.adoc[]

include::../rspecator.adoc[]