<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Debug statements are always useful during development. But include them in production code - particularly in code that runs client-side - and you run the risk of inadvertently exposing sensitive information.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-csharp" data-lang="csharp">private void DoSomething()
{
    // ...
    Console.WriteLine("so far, so good..."); // Noncompliant
    // ...
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_exceptions">Exceptions</h3>
<div class="paragraph">
<p>The following are ignored by this rule:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>Console Applications</p>
</li>
<li>
<p>Calls in methods decorated with <code>[Conditional ("DEBUG")]</code></p>
</li>
<li>
<p>Calls included in DEBUG preprocessor branches (<code>#if DEBUG</code>)</p>
</li>
</ul>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p>OWASP - <a href="https://owasp.org/www-project-top-ten/2017/A3_2017-Sensitive_Data_Exposure">Top 10 2017 Category A3 - Sensitive Data Exposure</a></p>
</li>
</ul>
</div>
<hr>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove this logging statement.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_4_feb_2015_152644_ann_campbell_wrote">on 4 Feb 2015, 15:26:44 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>Gendarme DisableDebuggingCodeRule</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_12_dec_2014_211913_sébastien_gioria_wrote">on 12 Dec 2014, 21:19:13 Sébastien Gioria wrote:</h3>
<div class="paragraph">
<p>could be tag OWASP-TOP10-A6</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_15_dec_2014_073629_freddy_mallet_wrote">on 15 Dec 2014, 07:36:29 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>Done Sébastien, thanks !</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_1_dec_2015_154652_ann_campbell_wrote">on 1 Dec 2015, 15:46:52 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>Note that the Java implementation of this may want to check log4j configuration files.</p>
</div>
<div class="paragraph">
<p>cc [~michael.gumowski]</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_30_aug_2018_145754_nicolas_harraudeau_wrote">on 30 Aug 2018, 14:57:54 Nicolas Harraudeau wrote:</h3>
<div class="paragraph">
<p>This rule is superseded by <a data-rspec-id="S106" class="rspec-auto-link">RSPEC-106</a>.</p>
</div>
<div class="paragraph">
<p>Reason: Console is just one specific case of Standard Output. A hotspot will later be created for the security part. See <a data-rspec-id="S4805" class="rspec-auto-link">RSPEC-4805</a>.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_17_feb_2021_091226_eric_therond_wrote">on 17 Feb 2021, 09:12:26 Eric Therond wrote:</h3>
<div class="paragraph">
<p>Can be safely deprecated now,</p>
</div>
<div class="paragraph">
<p>the replacement RSPC-106 is well more elaborate and on the security part there are different rules about logging (<a data-rspec-id="S5145" class="rspec-auto-link">RSPEC-5145</a>, <a data-rspec-id="S5757" class="rspec-auto-link">RSPEC-5757</a>) and more to come.</p>
</div>
</div>
</div>
</div>