== Why is this an issue?

Given an expression `e` of type `T`, `sizeof(e)` returns the size in bytes of `T`, without evaluating `e`.
Furthermore, the expression `sizeof(e)` has type `size_t`.

This means that `sizeof(sizeof(e))` is equivalent to `sizeof(size_t)`.
In other words, its value does not depend on `e` nor has any side effects.

== How to fix it

This defect is usually the consequence of a typo, and the fix is to remove on level of `sizeof`.

=== Code examples

==== Noncompliant code example

[source,c,diff-id=1,diff-type=noncompliant]
----
#include <string.h>

int main(int argc, char* argv[])
{
  char buffer[42];
  char buffer2[sizeof(sizeof(buffer))]; /* Noncompliant - a single sizeof() was intended */

  memcpy(buffer, "Hello, world!", strlen("Hello, world!")+1);
  memcpy(buffer2, buffer, sizeof(buffer)); /* Buffer overflow */

  return 0;
}
----

==== Compliant solution

[source,c,diff-id=1,diff-type=compliant]
----
#include <string.h>

int main(int argc, char* argv[])
{
  char buffer[42];
  char buffer2[sizeof(buffer)]; /* Compliant */

  memcpy(buffer, "Hello, world!", strlen("Hello, world!")+1);
  memcpy(buffer2, buffer, sizeof(buffer));

  return 0;
}
----

== Resources

=== Documentation

* {cpp} reference - https://en.cppreference.com/w/cpp/language/sizeof[`sizeof` operator]


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Remove the inner "sizeof" call.


endif::env-github,rspecator-view[]
