In PHP, PCRE functions, such as `preg_match`, are often used to perform regular expression operations. It is necessary to enclose the actual pattern with delimiters. The delimiters are usually `/`. This leads to the fact that each character must be escaped identically to the delimiter in the pattern.

However, delimiters may also be any non-alphanumeric, non-backslash or non-whitespace character.
In addition to two identical delimiters, it is also allowed to use bracket style delimiters with the opening and closing brackets. This set of different delimiters allows to avoid unnecessary escaping of used characters.

== Noncompliant Code Example

----
"/https:\/\/w{3}/i"
----

== Compliant Solution

----
"#https://w{3}#i"
"<https://w{3}>i"
----

== See

* https://www.php.net/manual/en/regexp.reference.delimiters.php[PCRE regex syntax] - Delimiters

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Primary location: Poorly chosen delimiter.
Secondary location: Unnecessary escaping character.

=== Highlighting

Primary location on first delimiter.
Secondary location on each escaped character which is equal to delimiter.


'''

endif::env-github,rspecator-view[]
