In PHP, PCRE functions, such as `preg_match`, are often used to perform regular expression operations. It is necessary to enclose the actual pattern with delimiters. The delimiters are usually `/`. Every occurrence of the delimiter character in the pattern has to be escaped with a backslash.

However, delimiters may also be any non-alphanumeric, non-backslash or non-whitespace character.
In addition to two identical delimiters, it is also allowed to use bracket style delimiters with the opening and closing brackets. Using different delimiters out of these possibilities allows avoiding unnecessary escaping, and increases the readability.

== Noncompliant Code Example

----
"/https:\/\/w{3}/i"
----

== Compliant Solution

----
"#https://w{3}#i"
"<https://w{3}>i"
----

== See

* https://www.php.net/manual/en/regexp.reference.delimiters.php[PCRE regex syntax] - Delimiters

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Primary location: Poorly chosen delimiter.
Secondary location: Unnecessary escaping character.

=== Highlighting

Primary location on first delimiter.
Secondary location on each escaped character which is equal to delimiter.


'''

endif::env-github,rspecator-view[]
