include::../summary.adoc[]

== Why is this an issue?

include::../rationale.adoc[]

include::../impact.adoc[]

== How to fix it

=== Code examples

==== Noncompliant code example

[source,vbnet,diff-id=1,diff-type=noncompliant]
----
Public Class ExampleController
    Inherits Controller

    Public Function Compute(data As Integer) As IActionResult
        For i As Integer = 0 To data - 1 ' Noncompliant
            Console.WriteLine("Hello")
        Next

        Enumerable.Range(1, data). ' Noncompliant
            ToList().
            ForEach(Sub(i) Console.WriteLine("World"))

        Return Ok()
    End Function
End Class
----

==== Compliant solution

[source,vbnet,diff-id=1,diff-type=compliant]
----
Public Class ExampleController
    Inherits Controller

    Public Shared MAX_BOUNDARY As Integer = 1337
    Public Shared MIN_BOUNDARY As Integer = 1

    Public Function Compute(data As Integer) As IActionResult
        If MIN_BOUNDARY > data Then
            data = MIN_BOUNDARY
        ElseIf data > MAX_BOUNDARY Then
            data = MAX_BOUNDARY
        End If

        For i As Integer = 0 To data - 1
            Console.WriteLine("Hello")
        Next

        Enumerable.Range(1, data).
            ToList().
            ForEach(Sub(i) Console.WriteLine("World"))

        Return Ok()
    End Function
End Class
----

=== How does this work?

include::../common/fix/limits.adoc[]

== Resources

include::../common/resources/standards.adoc[]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../highlighting.adoc[]

'''
endif::env-github,rspecator-view[]
