include::../summary.adoc[]

== Why is this an issue?

include::../rationale.adoc[]

include::../impact.adoc[]

== How to fix it

=== Code examples

==== Noncompliant code example

[source,vbnet,diff-id=1,diff-type=noncompliant]
----
Public Class ExampleController
    Inherits Controller

    Public Function Compute(Data As Integer) As IActionResult
        Enumerable.Range(1, Data). ' Noncompliant
            ToList().
            ForEach(Sub(I) Console.WriteLine("World"))

        For I As Integer = 0 To Data - 1 ' Noncompliant
            Console.WriteLine("Hello")
        Next

        Return Ok()
    End Function
End Class
----

==== Compliant solution

[source,vbnet,diff-id=1,diff-type=compliant]
----
Public Class ExampleController
    Inherits Controller

    Public Shared MaxBoundary As Integer = 1337
    Public Shared MinBoundary As Integer = 1

    Public Function Compute(Data As Integer) As IActionResult
        If MinBoundary > Data Then
            Data = MinBoundary
        ElseIf Data > MaxBoundary Then
            Data = MaxBoundary
        End If

        Enumerable.Range(1, Data).
            ToList().
            ForEach(Sub(I) Console.WriteLine("World"))

        For I As Integer = 0 To Data - 1
            Console.WriteLine("Hello")
        Next

        Return Ok()
    End Function
End Class
----

=== How does this work?

include::../common/fix/limits.adoc[]

== Resources

include::../common/resources/standards.adoc[]

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../highlighting.adoc[]

'''
endif::env-github,rspecator-view[]
