<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Ideally, identifiers (names of classes, variables, methods, structures, macros, labeles, &amp;etc.) are all easily distinguished from one another. Unfortunately, some characters or combinations of characters look enough alike that that&#8217;s not always the case.</p>
</div>
<div class="paragraph">
<p>Therefore, identifiers that differ from each other only by one of these problematic characters (or character combinations) should be renamed.</p>
</div>
<div class="paragraph">
<p>The problem pairs are:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>capital 'O' and digit 0</p>
</li>
<li>
<p>capital 'I' (eye) and digit 1</p>
</li>
<li>
<p>capital 'I' (eye) and lowercase 'l' (el)</p>
</li>
<li>
<p>lowercase 'l' (el) and digit 1</p>
</li>
<li>
<p>capital 'S' and digit 5</p>
</li>
<li>
<p>capital 'Z' and digit 2</p>
</li>
<li>
<p>lowercase 'n' and lowercase 'h'</p>
</li>
<li>
<p>capital 'B' and digit 8</p>
</li>
<li>
<p>lowercase 'm' and the combination "rn" ('r' followed by 'n')</p>
</li>
<li>
<p>any lowercase letter and its uppercase variant</p>
</li>
<li>
<p>the presence or absence of an underscore</p>
</li>
</ul>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-text" data-lang="text">int myl;
int myI;
int my1;</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-text" data-lang="text">int my_el;
int my_upper_i;
int my_one;</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Rename "xxx"; it looks too much like "yyy"</p>
</div>
</div>
</div>
</div>