The use of ``++%TYPE++`` and ``++%ROWTYPE++`` is an easy way to make sure that a variable's type always matches the type of the relevant column in an existing table. If the column type changes, then the variable changes with it. 


== Noncompliant Code Example

----
CREATE FUNCTION func1(acc_no IN NUMBER) 
   RETURN NUMBER 
   IS total NUMBER(11,2); -- Noncompliant
   BEGIN 
      SELECT order_total 
      INTO total -- total is assigned here: it could be defined as orders.order_total%TYPE
      FROM orders 
      WHERE customer_id = acc_no; 
      RETURN(total); 
    END;
----


== Compliant Solution

----
CREATE FUNCTION func1(acc_no IN NUMBER) 
   RETURN NUMBER 
   IS total orders.order_total%TYPE;
   BEGIN 
      SELECT order_total 
      INTO total
      FROM orders 
      WHERE customer_id = acc_no; 
      RETURN(total); 
   END;
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
