== Why is this an issue?

The use of ``++%TYPE++`` and ``++%ROWTYPE++`` is an easy way to make sure that a variable's type always matches the type of the relevant column in an existing table. If the column type changes, then the variable changes with it. 


=== Noncompliant code example

[source,sql]
----
CREATE FUNCTION func1(acc_no IN NUMBER) 
   RETURN NUMBER 
   IS total NUMBER(11,2); -- Noncompliant
   BEGIN 
      SELECT order_total 
      INTO total -- total is assigned here: it could be defined as orders.order_total%TYPE
      FROM orders 
      WHERE customer_id = acc_no; 
      RETURN(total); 
    END;
----


=== Compliant solution

[source,sql]
----
CREATE FUNCTION func1(acc_no IN NUMBER) 
   RETURN NUMBER 
   IS total orders.order_total%TYPE;
   BEGIN 
      SELECT order_total 
      INTO total
      FROM orders 
      WHERE customer_id = acc_no; 
      RETURN(total); 
   END;
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Use "[%TYPE|%ROWTYPE]" here.


'''
== Comments And Links
(visible only on this page)

=== relates to: S1703

=== on 10 Jan 2017, 17:58:40 Ann Campbell wrote:
Note that this RSpec is an almost exact inverse of RSPEC-1703. The difference between them, and the reason to have both, is Oracle Forms. With Forms in play, RSPEC-1703 should be used. Without it/them, this is the rule that's preferred.

=== on 19 Jan 2017, 11:15:39 Pierre-Yves Nicolas wrote:
\[~ann.campbell.2] I think that this rule would trigger a lot of noise if it's defined like that.

I think that using "%TYPE" and "%ROWTYPE" is not always possible. For example, if a variable is used as a counter, it would not refer to any column.

I would restrict that rule to cases where we are sure that the variable can be defined with "%TYPE" or "%ROWTYPE" and that probably requires SQL queries.

=== on 19 Jan 2017, 12:31:28 Pierre-Yves Nicolas wrote:
\[~ann.campbell.2] Following my previous comment, I don't think we can raise issues on package specifications because we need to know how the variable is used. As a first step, maybe we should only target variables which are assigned in a ``++SELECT...INTO++``. Example:

----
CREATE FUNCTION func1(acc_no IN NUMBER) 
   RETURN NUMBER 
   IS total NUMBER(11,2); -- Noncompliant
   BEGIN 
      SELECT order_total 
      INTO total -- total is assigned here: it could be defined as orders.order_total%TYPE
      FROM orders 
      WHERE customer_id = acc_no; 
      RETURN(total); 
    END;
/
----

=== on 23 Jan 2017, 09:29:22 Ann Campbell wrote:
\[~pierre-yves.nicolas] I've updated the title and the code samples. Please double-check that I've converted the Compliant Solution correctly

=== on 23 Jan 2017, 09:34:38 Pierre-Yves Nicolas wrote:
\[~ann.campbell.2] Looks good to me.

endif::env-github,rspecator-view[]
