
== Why is this an issue?
When casting a raw pointer to a slice, the produced raw pointer does not correctly update its length metadata, leading to potential issues such as creating a slice with less data than intended or more data, risking undefined behavior upon dereferencing.

== How to fix it

Use `ptr::slice_from_raw_parts` to construct a slice from a data pointer and the correct length.

=== Code examples

==== Noncompliant code example
[source,rust,diff-id=1,diff-type=noncompliant]
----
let a = [1_u8, 2, 3, 4];
let p = &a as *const [u8] as *const [u32];
unsafe {
    println!("{:?}", &*p); // Noncompliant: Undefined behavior
}
----

==== Compliant solution

[source,rust,diff-id=1,diff-type=compliant]
----
let a = [1_i32, 2, 3, 4];
let old_ptr = &a as *const [i32];
// The data pointer is cast to a pointer to the target `u8` not `[u8]`
// The length comes from the known length of 4 i32s times the 4 bytes per i32
let new_ptr = core::ptr::slice_from_raw_parts(old_ptr as *const u8, 16);
unsafe {
    println!("{:?}", &*new_ptr);
}
----

== Resources
=== Documentation

* Clippy Lints - https://rust-lang.github.io/rust-clippy/master/index.html#cast_slice_different_sizes
