include::../description.adoc[]

== Why is this an issue?

include::../why.adoc[]

== How to fix it

include::../how-to-fix.adoc[]

=== Code examples

==== Noncompliant code example

This example demonstrates pathname expansion using the `echo` command:
[source,bash,diff-id=1,diff-type=noncompliant]
----
#!/bin/bash

test="command t*.sh" && echo $test
----
Suppose this code is executed in a directory that contains two files: `temp1.sh` and `temp2.sh`. This code will print `"command temp1.sh temp2.sh"`,
as `*` is substituted with matching files in the current folder.

This example demonstrates word splitting using the `echo` command:
[source,bash,diff-id=2,diff-type=noncompliant]
----
#!/bin/bash

test=" Hello World " && echo $test
----
This code will print `"Hello World"`, omitting the leading and trailing whitespaces.

==== Compliant solution

This example demonstrates pathname expansion using the `echo` command, which will print ``++"command t*.sh"++`` as intended:
[source,bash,diff-id=1,diff-type=compliant]
----
#!/bin/bash

test="command t*.sh" && echo "$test"
----

This example demonstrates word splitting using the `echo` command, which will print `" Hello World "` as intended:
[source,bash,diff-id=2,diff-type=compliant]
----
#!/bin/bash

test=" Hello World " && echo "$test"
----

== Resources

include::../resources.adoc[]

ifdef::env-github,rspecator-view[]
'''
== Implementation Specification
(visible only on this page)

=== Message

include::../message.adoc[]

=== Highlighting

include::../highlighting.adoc[]

'''
endif::env-github,rspecator-view[]
