<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Global variables can seem like a handy way to pass state information around in a program, but the use of global variables only works well in very small programs. As the code base grows, you&#8217;ll need to understand every subprocedure&#8217;s impact on the global state in order to understand how the program works. This is a task that quickly becomes impossible.</p>
</div>
<div class="paragraph">
<p>To control the situation, only the main procedure should be allowed access to global variables; it can then pass that state information to subprocedures as parameters.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-rpg" data-lang="rpg">     D FirstName       S             20A
     D LastName        S             20A

      /free
       FirstName = 'John';
       LastName = 'Smith';
       DSPLY FullName();
      /end-free

     P FullName        B
     D FullName        PI            41A
      /free
       return FirstName + ' ' + LastName;
      /end-Free
     P                 E</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-rpg" data-lang="rpg">      /free
       DSPLY FullName('John':'Smith');
      /end-free

     P FullName        B
     D FullName        PI            41A
     D  FirstName                    20A   Const
     D  LastName                     20A   Const
      /free
       return FirstName + ' ' + LastName;
      /end-Free
     P                 E</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Pass "XXX" to this subprocedure as a parameter.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_relates_to_s2536">relates to: <a data-rspec-id="S2536" class="rspec-auto-link">S2536</a></h3>

</div>
<div class="sect2">
<h3 id="_on_12_sep_2014_144835_ann_campbell_wrote">on 12 Sep 2014, 14:48:35 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~pierre-yves.nicolas] do you find this rule interesting enough to provide code samples for me? The initial requester lost interest before doing so.</p>
</div>
</div>
</div>
</div>