<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Providing a <code>serialVersionUID</code> field on <code>Serializable</code> classes is strongly recommended by the <code>Serializable</code> documentation but blindly following that recommendation can be harmful.</p>
</div>
<div class="paragraph">
<p><code>serialVersionUID</code> value is stored with the serialized data and this field is verified when deserializing the data to ensure that the code reading the data is compatible with the serialized data. In case of failure, it means the serialized data and the code are not in sync and this fine because you know what&#8217;s wrong.</p>
</div>
<div class="paragraph">
<p>When the <code>serialVersionUID</code> is generated by an IDE or blindly hard-coded, there is a high probability that one will forget to update the <code>serialVersionUID</code> value when the <code>Serializable</code> class is later enriched with additional fields. As a consequence, old serialized data will incorrectly be considered compatible with the newer version of the code creating situations which are hard to debug.</p>
</div>
<div class="paragraph">
<p>Therefore, defining <code>serialVersionUID</code> should be done with care. This rule raises an issue on each <code>serialVersionUID</code> field declared on classes implementing <code>Serializable</code> to be sure the presence and the value of the <code>serialVersionUID</code> field is challenged and validated by the team.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">public class Foo implements Serializable {
  private static final long serialVersionUID = 1;
}

public class BarException extends RuntimeException {
  private static final long serialVersionUID = 8582433437601788991L;
}</code></pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p>Vojtech Ruzicka&#8217;s Programming Blog: <a href="https://www.vojtechruzicka.com/explicitly-declare-serialversionuid/">Should I explicitly declare serialVersionUID?</a></p>
</li>
</ul>
</div>
<hr>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove this "serialVersionUID"</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>"serialVersionUID"</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_1_dec_2018_103749_jens_bannmann_wrote">on 1 Dec 2018, 10:37:49 Jens Bannmann wrote:</h3>
<div class="paragraph">
<p>The "serialization" tag <a href="https://jira.sonarsource.com/issues/?jql=project%20%3D%20RSPEC%20AND%20status%20%3D%20Active%20AND%20labels%20%3D%20serialization%20ORDER%20BY%20key">currently includes 14 rules</a>, but this one is surprisingly missing. I suggest adding it.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_3_dec_2018_151954_ann_campbell_wrote">on 3 Dec 2018, 15:19:54 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>Thanks [~bannmann]. Oversight fixed.</p>
</div>
</div>
</div>
</div>