<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>In PHP, references allow you to create multiple names for the same variable, enabling you to access and manipulate its value through different identifiers.
They are denoted by the ampersand symbol &amp; placed before the variable name during declaration or assignment.</p>
</div>
<div class="paragraph">
<p>When a reference is used in a <code>foreach</code> loop instead of using a simple variable, the reference remains assigned and keeps its "value" which is a reference, even after the <code>foreach</code> execution.</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="paragraph">
<p>Not unsetting the reference can lead to bugs later in the code, as most of the time, this behavior is different from what the developer is expecting.
For example, the reference may be used incorrectly with the previous value.</p>
</div>
<div class="paragraph">
<p>To avoid unexpected side effects, it is recommended to always <code>unset</code> a reference that is used in a <code>foreach</code> loop.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it_in_core_php">How to fix it in Core PHP</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Unset the reference that is used in the <code>foreach</code> loop.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-php" data-lang="php">$arr = array(1, 2, 3);
foreach ($arr as &amp;$value) { // Noncompliant; $value is still alive after the loop and references the last item of the array: $arr[2]
    $value = $value * 2;
}
$value = 'x';</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-php" data-lang="php">$arr = array(1, 2, 3);
foreach ($arr as &amp;$value) { // Compliant; there is no risk to use by mistake the content of $value pointing to $arr[2]
    $value = $value * 2;
}
unset($value);
$value = 'x';</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="paragraph">
<p>PHP Documentation:</p>
</div>
<div class="ulist">
<ul>
<li>
<p><a href="https://www.php.net/manual/en/language.references.php">References</a></p>
</li>
<li>
<p><a href="https://php.net/manual/en/control-structures.foreach.php">Foreach</a></p>
</li>
</ul>
</div>
</div>
<div class="sect2">
<h3 id="_articles_blog_posts">Articles &amp; blog posts</h3>
<div class="ulist">
<ul>
<li>
<p><a href="https://schlueters.de/blog/archives/141-References-and-foreach.html">References and Foreach</a></p>
</li>
</ul>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Unset XXX just after the end of the foreach loop to avoid side effects.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>The reference declaration in the foreach.</p>
</div>
</div>
</div>
</div>