== Why is this an issue?

Before Java 8, the only way to partially support closures in Java was by using anonymous inner classes. But the syntax of anonymous classes may seem unwieldy and unclear.


With Java 8, most uses of anonymous inner classes should be replaced by lambdas to highly increase the readability of the source code.


*Note* that this rule is automatically disabled when the project's ``++sonar.java.source++`` is lower than ``++8++``.


=== Noncompliant code example

[source,java]
----
myCollection.stream().map(new Mapper<String,String>() {
  public String map(String input) {
    return new StringBuilder(input).reverse().toString();
  }
});

Predicate<String> isEmpty = new Predicate<String> {
    boolean test(String myString) {
        return myString.isEmpty();
    }
}
----


=== Compliant solution

[source,java]
----
myCollection.stream().map(input -> new StringBuilder(input).reverse().toString());

Predicate<String> isEmpty = myString -> myString.isEmpty();
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Make this anonymous inner class a lambda. [(sonar.java.source not set. Assuming 8 or greater.)]


'''
== Comments And Links
(visible only on this page)

=== on 24 Feb 2014, 10:54:33 Freddy Mallet wrote:
Is implemented by \http://jira.codehaus.org/browse/SONARJAVA-455

endif::env-github,rspecator-view[]
