== How to fix it in Gorm

=== Code examples

include::../../common/fix/code-rationale.adoc[]

==== Noncompliant code example

[source,go,diff-id=1,diff-type=noncompliant]
----
func authenticate(w http.ResponseWriter, r *http.Request, db *gorm.DB) {
    username := r.FormValue("username")
    password := r.FormValue("password")
    var user User
    db.First(&user, "username='" + username + "' AND password = '" + password + "'").Error // Noncompliant
    if err != nil {
        w.WriteHeader(http.StatusUnauthorized)
        fmt.Fprintf(w, "Unauthorized")
        return
    }
    fmt.Fprintf(w, "Authentication successful")
}
----

==== Compliant solution

[source,go,diff-id=1,diff-type=compliant]
----
func authenticate(w http.ResponseWriter, r *http.Request, db *gorm.DB) {
    username := r.FormValue("username")
    password := r.FormValue("password")
    var user User
    db.First(&user, "username=? AND password=?", username, password).Error
    if err != nil {
        w.WriteHeader(http.StatusUnauthorized)
        fmt.Fprintf(w, "Unauthorized")
        return
    }
    fmt.Fprintf(w, "Authentication successful")
}
----

=== How does this work?

include::../../common/fix/prepared-statements.adoc[]

