== How to fix it in sqlx
=== Code examples

include::../../common/fix/code-rationale.adoc[]

==== Noncompliant code example

[source,go,diff-id=1,diff-type=noncompliant]
----
func authenticate(w http.ResponseWriter, r *http.Request, db *sqlx.DB) {
    username := r.FormValue("username")
    password := r.FormValue("password")
    var user User
    row := db.QueryRow("SELECT * FROM users WHERE username='" + username + "' AND password = '" + password + "'") // Noncompliant
    if err := row.Scan(&user); err != nil {
        http.Error(w, "Unauthorized", http.StatusUnauthorized)
        return
    }
    fmt.Fprintf(w, "Authentication successful")
}
----

==== Compliant solution

[source,go,diff-id=1,diff-type=compliant]
----
func authenticate(w http.ResponseWriter, r *http.Request, db *sqlx.DB) {
    username := r.FormValue("username")
    password := r.FormValue("password")
    var user User
    row := s.db.QueryRow("SELECT * FROM users WHERE username=? AND password=?", username, password) 
    if err := row.Scan(&user); err != nil {
        http.Error(w, "Unauthorized", http.StatusUnauthorized)
        return
    }
    fmt.Fprintf(w, "Authentication successful")
}
----

=== How does this work?

include::../../common/fix/prepared-statements.adoc[]

