== How to fix it in Go Standard Library

=== Code examples

include::../../common/fix/code-rationale.adoc[]

==== Noncompliant code example

[source,go,diff-id=1,diff-type=noncompliant]
----
func authenticate(w http.ResponseWriter, r *http.Request, db *sql.DB) {
    username := r.FormValue("username")
    password := r.FormValue("password")
    row := db.QueryRow("SELECT username FROM users WHERE username='" + username + "' AND password = '" + password + "'") // Noncompliant
    if err := row.Scan(&username); err != nil {
        w.WriteHeader(http.StatusUnauthorized)
        fmt.Fprintf(w, "Unauthorized")
        return
    }
    fmt.Fprintf(w, "Authentication successful")
}
----

==== Compliant solution

[source,go,diff-id=1,diff-type=compliant]
----
func authenticate(w http.ResponseWriter, r *http.Request, db *sql.DB) {
    username := r.FormValue("username")
    password := r.FormValue("password")
    row := s.db.QueryRow("SELECT username FROM users WHERE username=? AND password=?", username, password) 
    if err := row.Scan(&username); err != nil {
        w.WriteHeader(http.StatusUnauthorized)
        fmt.Fprintf(w, "Unauthorized")
        return
    }
    fmt.Fprintf(w, "Authentication successful")
}
----

=== How does this work?

include::../../common/fix/prepared-statements.adoc[]

