<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Using closures when a function pointer can be used instead can negatively impact readability and maintainability of the code. Closures can obscure the intent of the code, making it harder for other developers to understand what the code is doing at a glance. Function pointers, on the other hand, are more explicit and can make the code more concise and easier to read.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Replace closures with function pointers when the closure simply calls a function. This makes the code more readable and easier to understand.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-rust" data-lang="rust">let result = Some('a').map(|s| s.to_uppercase());</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-rust" data-lang="rust">let result = Some('a').map(char::to_uppercase);</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Clippy Lints - <a href="https://rust-lang.github.io/rust-clippy/master/index.html#redundant_closure_for_method_calls" class="bare">https://rust-lang.github.io/rust-clippy/master/index.html#redundant_closure_for_method_calls</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>