== Why is this an issue?

Shared coding conventions allow teams to collaborate efficiently. This rule checks that object attribute names match the provided regular expression.


=== Noncompliant code example

With the default regular expression ``++[a-zA-Z](_?+[a-zA-Z0-9])*+++`` :

[source,sql]
----
CREATE TYPE my_type AS OBJECT(
  foo__bar INTEGER             -- Non-Compliant
);
/

DROP TYPE my_type;
----


=== Compliant solution

[source,sql]
----
CREATE TYPE my_type AS OBJECT(
  foo_bar INTEGER              -- Compliant
);
/

DROP TYPE my_type;
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Rename this attribute to match the provided regular expression: xxx


=== Parameters

.format
****

----
[a-zA-Z](_?+[a-zA-Z0-9])*+
----

The regular expression the name should match
****


endif::env-github,rspecator-view[]
