== How to fix it in .NET

=== Code examples

include::../../common/fix/code-rationale.adoc[]

==== Noncompliant code example

[source,vbnet,diff-id=1,diff-type=noncompliant]
----
Public Class ExampleController
    Inherits Controller

    Public Sub Run(Binary As String)
        Dim P As New Process()
        P.StartInfo.FileName = Binary
        P.Start()
    End Sub
End Class
----

==== Compliant solution

[source,vbnet,diff-id=1,diff-type=compliant]
----
Public Class ExampleController
    Inherits Controller

    Public Sub Run(Binary As String)
        If Binary.Equals("/usr/bin/ls") OrElse Binary.Equals("/usr/bin/cat") Then
            ' only ls and cat commands are authorized
            Dim P As New Process()
            P.StartInfo.FileName = Binary
            P.Start()
        End If
    End Sub
End Class
----

=== How does this work?

include::../../common/fix/introduction.adoc[]

include::../../common/fix/pre-approved-list.adoc[]

:sanitizationLib: System.Diagnostics.ProcessStartInfo

include::../../common/fix/sanitize-meta-characters.adoc[]

Here, using the `ProcessStartInfo` structure helps escaping the passed
arguments and internally creates a single string given to the operating system
when `System.Diagnostics.Process.Start()` is called.
