== Why is this an issue?

``++SELECT INTO TABLE++`` is much more efficient than ``++SELECT ... ENDSELECT++``. ``++SELECT INTO TABLE++`` needs more memory to hold the result set, but in normal situations, this is not a concern. When memory is a concern, the result set can be divided into smaller sets.


=== Noncompliant code example

[source,abap]
----
SELECT * FROM T006 INTO X006_WA.
  ...
ENDSELECT.
----


=== Compliant solution

[source,abap]
----
SELECT * FROM T006 INTO TABLE X006.
LOOP AT X006 INTO X006_WA.
  ...
ENDLOOP.
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Refactor this select to use "SELECT INTO TABLE int1 + LOOP AT int1".


'''
== Comments And Links
(visible only on this page)

=== on 2 Dec 2014, 14:39:21 Ann Campbell wrote:
\[~freddy.mallet] I'm confused by this description. ``++SELECT INTO TABLE++`` is more efficient but it needs more memory...?

=== on 4 Dec 2014, 09:41:22 Freddy Mallet wrote:
Indeed @Ann:

* In one case, we we're going to consume less memory but the execution time might be really bigger
* In the other case, we start by quickly loading all information in memory to then process it

endif::env-github,rspecator-view[]
