<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Using a <code>match</code> expression with a boolean condition is considered a bad practice because it adds unnecessary complexity to the code. The <code>match</code> expression is designed for pattern matching and is more powerful and flexible than an <code>if-else</code> construct. However, when used with a simple boolean condition, it introduces extra verbosity and reduces code readability. Instead, it is generally better to use an <code>if-else</code> construct.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-rust" data-lang="rust">fn check_value(value: bool) -&gt; &amp;'static str {
    match value {
        true =&gt; "Value is true",
        false =&gt; "Value is false",
    }
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-rust" data-lang="rust">fn check_value(value: bool) -&gt; &amp;'static str {
    if value {
        "Value is true"
    } else {
        "Value is false"
    }
}</code></pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Clippy Lints - <a href="https://rust-lang.github.io/rust-clippy/master/index.html#match_bool" class="bare">https://rust-lang.github.io/rust-clippy/master/index.html#match_bool</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>