<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when <code>where(&#8230;&#8203;).take</code> is used to retrieve a single record by attributes.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Using <code>where(&#8230;&#8203;).take</code> for single record retrieval is less semantic and readable compared to <code>find_by</code>. The <code>find_by</code> method clearly expresses the intent to find a single record by specific attributes and return <code>nil</code> if not found.</p>
</div>
<div class="paragraph">
<p>The <code>where.take</code> pattern requires readers to understand that:</p>
</div>
<div class="ulist">
<ul>
<li>
<p><code>where</code> creates a query scope</p>
</li>
<li>
<p><code>take</code> retrieves the first record from that scope</p>
</li>
<li>
<p>The combination returns <code>nil</code> if no record matches</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>In contrast, <code>find_by</code> makes this intent explicit and follows Rails conventions. It&#8217;s the idiomatic way to retrieve a single record by attributes in ActiveRecord.</p>
</div>
<div class="paragraph">
<p>Using <code>find_by</code> also makes code more consistent with Rails style guidelines and community best practices, improving maintainability and reducing cognitive load for developers familiar with Rails conventions.</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="paragraph">
<p>This issue affects code readability and maintainability. While both approaches have similar performance characteristics, using non-idiomatic patterns can:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>Make code harder to understand for developers familiar with Rails conventions</p>
</li>
<li>
<p>Reduce consistency across the codebase</p>
</li>
<li>
<p>Increase cognitive load when reading and maintaining the code</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>The impact is primarily on code quality rather than functionality, as both patterns produce the same results.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Replace <code>where(&#8230;&#8203;).take</code> with <code>find_by(&#8230;&#8203;)</code> to make the intent clearer and follow Rails conventions.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-ruby" data-lang="ruby">User.where(email: email).take # Noncompliant</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-ruby" data-lang="ruby">User.find_by(email: email)</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Rails Guides - Active Record Query Interface - <a href="https://guides.rubyonrails.org/active_record_querying.html#find_by">Official Rails documentation on using find_by for single record retrieval</a></p>
</li>
<li>
<p>ActiveRecord::FinderMethods#find_by - <a href="https://api.rubyonrails.org/classes/ActiveRecord/FinderMethods.html#method-i-find_by">API documentation for the find_by method</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>