== Why is this an issue?

Variables, Classes and functions should not be undefined, otherwise the code will fail with a NameError.


=== Noncompliant code example

[source,python]
----
my_var # Noncompliant (variable is never defined)

def noncompliant():
    foo()  # Noncompliant
    MyClass()  # Noncompliant
----


=== Compliant solution

[source,python]
----
from mod import my_var

my_var

def compliant():
    foo = sum
    foo()

    class MyClass:
        pass
    MyClass()
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

XX is not defined. Change its name or define it before using it.


endif::env-github,rspecator-view[]
