== Why is this an issue?

include::description.adoc[]

=== Noncompliant code example

[source,text]
----
private final static String CODE = "bounteous";

public String calculateCode() {
  doTheThing();
  return CODE;
}

public String getName() {  // Noncompliant
  doTheThing();
  return CODE;
}
----


=== Compliant solution

[source,text]
----
private final static String CODE = "bounteous";

public String getCode() {
  doTheThing();
  return CODE;
}

public String getName() {
  return getCode();
}
----


include::exceptions.adoc[]
