== How to fix it in Android

=== Code examples

==== Noncompliant code example

[source,kotlin,diff-id=21,diff-type=noncompliant]
----
private val Context.dataStore: DataStore<Preferences> by preferencesDataStore(name = "user_prefs_datastore")

@Composable
fun PasswordInputScreen(modifier: Modifier = Modifier, onSavePassword: (String) -> Unit) {
    var password by remember { mutableStateOf("") }
    val context = LocalContext.current
    val coroutineScope = rememberCoroutineScope()  // Add this line

    Column() {
        TextField(
            value = password,
            onValueChange = { password = it },
            label = { Text("Password") },
            visualTransformation = PasswordVisualTransformation(),
        )
        Button(onClick = {
            coroutineScope.launch {  // Launch a coroutine
                context.dataStore.edit { preferences ->
                    preferences[stringPreferencesKey("saved_password")] = password
                }
            }
        }) {
            Text("Save Password")
        }
    }
}
----

==== Compliant solution

include::../../common/compliant.adoc[]

[source,kotlin,diff-id=21,diff-type=compliant]
----
import androidx.credentials.CredentialManager

@Composable
fun PasswordInputScreen(modifier: Modifier = Modifier, onSavePassword: (String) -> Unit) {
    var password by remember { mutableStateOf("") }
    val context = LocalContext.current
    val coroutineScope = rememberCoroutineScope()  // Add this line

    Column() {
        TextField(
            value = password,
            onValueChange = { password = it },
            label = { Text("Password") },
            visualTransformation = PasswordVisualTransformation(),
        )
        Button(onClick = {
            coroutineScope.launch {  // Launch a coroutine
                val credentialManager = CredentialManager.create(context)
                val request = CreatePasswordRequest(
                    id = "example",
                    password = password
                )
                credentialManager.createCredential(
                    request = request,
                    context = context
                )
            }
        }) {
            Text("Save Password")
        }
    }
}
----