<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule reports when a class is placed in a location which deviates from the intended architecture.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This project has an intended architecture defined, which specifies high-level components and their allowed relationships.
This might include that specific classes should be located in specific packages or modules, or that packages are renamed.
This issue is being raised because this class’s location deviates from the intended architecture.</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="paragraph">
<p>When the code location deviates from the intended architecture, it gradually erodes the project’s structure.
As these deviations accumulate, the code becomes more complex and less consistent with the intent, making future development increasingly challenging.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Move the relevant class to its intended location or rename the relevant package as intended.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="paragraph">
<p>If the intended architecture declares developers should use the package name <code>com.example.panel</code>, not <code>com.example.panels</code>, then the following code example is noncompliant:</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">package com.example.panels;

class ShowCustomersPanel extends Panel {
  // ...
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="paragraph">
<p>The solution is to move the class to its appropriate package, or respectively, to rename the package <code>panels</code> to <code>panel</code>.</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">package com.example.panel;

import com.example.services.CustomerService;

class ShowCustomersPanel extends Panel {
  // ...
}</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>