<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule reports when a file is placed in a location which deviates from the intended architecture.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This project has an intended architecture defined, which specifies high-level components and their allowed relationships.
This might include that specific files should be located in specific directories, or that directories are renamed.
This issue is being raised because this file’s location deviates from the intended architecture.</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="paragraph">
<p>When the code location deviates from the intended architecture, it gradually erodes the project’s structure.
As these deviations accumulate, the code becomes more complex and less consistent with the intent, making future development increasingly challenging.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Move the relevant file to its intended location or rename the relevant directory as intended.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="paragraph">
<p>If the intended architecture declares developers should use the directory name <code>./panel</code>, not <code>./panels</code>, then the following code example is noncompliant:</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-javascript" data-lang="javascript"># panels/ShowCustomersPanel.js
class ShowCustomersPanel ...</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="paragraph">
<p>The solution is to move the file to its appropriate directory, or respectively, to rename the directory <code>panels</code> to <code>panel</code>.</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-javascript" data-lang="javascript"># panel/ShowCustomersPanel.js
class ShowCustomersPanel ...</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>