== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,cpp]
----
printf("%s", (i > 10 ? "yes" : "no"));
----


=== Compliant solution

[source,cpp]
----
if (i > 10) {
  printf("yes");
} else {
  printf("no");
}
----


=== Exceptions

For {cpp}11 mode only, the issue is not raised for ternary operators used inside `constexpr` functions. 
In {cpp}11 such functions are limited to just a return statement, so the use of a ternary operator
is required in them. This restriction is lifted in later standards, and thus issues are raised.


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
