<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule reports when a class depends on another class, deviating from the intended architecture.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This project has an intended architecture defined, which includes high-level components and their allowed relationships.
This issue is being raised because a dependency in the code is not allowed since it constitutes an architecture deviation.</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="paragraph">
<p>When code deviates from the intended architecture, it gradually erodes the projectâ€™s structure.
As these deviations accumulate, the code becomes more complex and less consistent with the intent, making future development increasingly challenging.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Remove or refactor the deviating dependency.
This may involve replacing the dependency with an allowed one, moving code to appropriate classes or namespaces, or introducing new abstractions. The exact solution depends on your project&#8217;s specific situation and the code deviating the intended architecture.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="paragraph">
<p>If the intended architecture prohibits classes in namespace <code>Example.Panels</code> to depend on classes in <code>Example.Repos</code>, then the following code example is noncompliant:</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">using Example.Repos;

namespace Example.Panels;

public class ShowCustomersPanel
{
    CustomerRepo customerRepo = ...;
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="paragraph">
<p>A solution could be to call into the service layer instead.</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">using Example.Services;

namespace Example.Panels;

public class ShowCustomersPanel
{
    CustomerService customerService = ...;
}</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>