//// This rule is superseded by RSPEC-2201
//// Please consider implementing this latter instead.
Doing an operation on a string without using the result of the operation is useless and is certainly due to a misunderstanding.


== Noncompliant Code Example

----
var str = "..."
str.toUpperCase(); // Noncompliant
----


== Compliant Solution

----
var str = "..."
str = str.toUpperCase();
----

== See

* https://wiki.sei.cmu.edu/confluence/x/xzdGBQ[CERT, EXP00-J.] - Do not ignore values returned by methods

