Doing an operation on a string without using the result of the operation is useless and is certainly due to a misunderstanding. 


== Noncompliant Code Example

----
var str = "..."
str.toUpperCase(); // Noncompliant
----


== Compliant Solution

----
var str = "..."
str = str.toUpperCase();
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
