Doing an operation on a string without using the result of the operation is useless and is certainly due to a misunderstanding. 


== Noncompliant Code Example

----
var str = "..."
str.toUpperCase(); // Noncompliant
----


== Compliant Solution

----
var str = "..."
str = str.toUpperCase();
----

== See

* https://wiki.sei.cmu.edu/confluence/x/xzdGBQ[CERT, EXP00-J.] - Do not ignore values returned by methods


ifdef::env-github,rspecator-view[]
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
