== Why is this an issue?

This excludes the use of ``++//++`` C99 style comments and {cpp} style comments, since these are not permitted in C90. Many compilers support the ``++//++`` style of comments as an extension to C90. The use of ``++//++`` in preprocessor directives (e.g. ``++#define++``) can vary. Also the mixing of ``++/* ... */++`` and ``++//++`` is not consistent. This is more than a style issue, since different (pre C99) compilers may behave differently.


=== Noncompliant code example

[source,text]
----
int main(int argc, char* argv[])
{
  // Do nothing - Noncompliant
  return 0;
}
----


=== Compliant solution

[source,text]
----
int main(int argc, char* argv[])
{
  /* Do nothing - Compliant */
  return 0;
}
----


