== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,text]
----
Public Class Foo
    Public Const Version = 1.0           ' Noncompliant
End Class
----


=== Compliant solution

[source,text]
----
Public Class Foo
    Public Shared ReadOnly Property Version = 1.0 ' Compliant
End Class
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

endif::env-github,rspecator-view[]
