== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

With the default regular expression ``++^[a-z_]+(\.[a-z_][a-z0-9_]*)*$++``:

[source,java]
----
package org.exAmple; // Noncompliant
----

=== Compliant solution

[source,java]
----
package org.example; 
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

=== Parameters

.format
****

----
^[a-z_]+(\.[a-z_][a-z0-9_]*)*$
----

Regular expression used to check the package names against.
****


'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
