== Why is this an issue?

include::../rationale.adoc[]

include::../impact.adoc[]

== How to fix it

=== Code examples

==== Noncompliant code example

Setting the `sandbox` property of `webPreferences` to false or setting 
`nodeIntegration` to true will result in a webview that will not be sandboxed.


[source,html,diff-id=11,diff-type=noncompliant]
----
<webview 
    src="https://example.com/index.html"
    nodeintegration="true"
    >
</webview><!-- Noncompliant -->
----

[source,html,diff-id=12,diff-type=noncompliant]
----
<webview 
    src="https://example.com/index.html"
    webpreferences="sandbox=false"
    >
</webview><!-- Noncompliant -->
----

==== Compliant solution

[source,html,diff-id=11,diff-type=compliant]
----
<webview 
    src="https://example.com/index.html"
    nodeintegration="false"
    >
</webview>
----

[source,html,diff-id=12,diff-type=compliant]
----
<webview 
    src="https://example.com/index.html"
    webpreferences="sandbox=true"
    >
</webview>
----

=== How does this work?

In the compliant examples `sandbox` or `nodeIntegration` are explicitly 
set to their secure value. It is also sufficient to not set any of these 
properties since they will default to secure values.

//=== Pitfalls

//=== Going the extra mile


== Resources

include::../common/resources/docs.adoc[]

//=== Articles & blog posts
//=== Conference presentations
//=== Standards
//=== External coding guidelines
//=== Benchmarks

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

=== Highlighting

Highlight the `nodeIntegration` or `webPreferences` attribute of the <webview> tag (depending on which one caused the issue to be raised).

'''
== Comments And Links
(visible only on this page)

endif::env-github,rspecator-view[]