include::../description.adoc[]

== Noncompliant Code Example

----
for i := 0; i < 10; i++ { // noncompliant, loop only executes once
	fmt.Println(i)
	break
}
----

== Compliant Solution

----
for i := 0; i < 10; i++ {
	fmt.Println(i)
}
----

ifdef::env-github,rspecator-view[]
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
