<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p><code>%TYPE</code> and <code>%ROWTYPE</code> seem like an easy way to make sure that a variable&#8217;s type always matches the type of the relevant column in an existing table. If the column type changes, then the variable changes with it.</p>
</div>
<div class="paragraph">
<p>However, Oracle Forms compiled against a procedure using either of these two symbols won&#8217;t get the benefit of that flexibility. Instead, at compile time, the relevant type is looked up from the underlying database and used in the form. If the column type changes later or the form is running against a database with different length semantics, attempting to use the form results in an "ORA-04062: Signature of package has been changed" error on the package in question. And the form needs to be recompiled on exactly the same database environment where it will run to avoid the error.</p>
</div>
<div class="paragraph">
<p>Note that <code>%TYPE</code> and <code>%ROWTYPE</code> can be used in a package&#8217;s private procedures and functions, private package variables, and local variables without issue, but not in the package specification.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-sql" data-lang="sql">CREATE OR REPLACE PACKAGE PACK IS
  TYPE mytype IS RECORD (
    var1 mytable.mycolumn%TYPE -- Noncompliant
  );

  FUNCTION MY_FUNC(param1 IN mytable.mycolumn%TYPE) RETURN VARCHAR2; -- Noncompliant

  FUNCTION MY_FUNC2(param1 IN mytable%ROWTYPE) RETURN VARCHAR2; -- Noncompliant

END;</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-sql" data-lang="sql">CREATE OR REPLACE PACKAGE PACK IS
  TYPE mytype IS RECORD (
    var1 VARCHAR2(100) -- Compliant
  );

  FUNCTION MY_FUNC(param1 IN VARCHAR2) RETURN VARCHAR2; -- Compliant

  TYPE myrowtype IS RECORD (
    col1 NUMBER,
    col2 VARCHAR2(30)
  );
  FUNCTION MY_FUNC2(param1 IN myrowtype) RETURN VARCHAR2; -- Compliant

END;</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Specify "XXX"'s type explicitly</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_is_related_to_s3829">is related to: <a data-rspec-id="S3829" class="rspec-auto-link">S3829</a></h3>

</div>
<div class="sect2">
<h3 id="_on_12_feb_2019_142323_pierre_yves_nicolas_wrote">on 12 Feb 2019, 14:23:23 Pierre-Yves Nicolas wrote:</h3>
<div class="paragraph">
<p>\[~alexandre.gigleux] What about having a tag for rules which are related to Oracle Forms?</p>
</div>
</div>
</div>
</div>