<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>The Java Language Specification defines a set of rules called naming conventions that apply to Java programs.
These conventions provide recommendations for naming packages, classes, methods, and variables.</p>
</div>
<div class="paragraph">
<p>By following shared naming conventions, teams can collaborate more efficiently.</p>
</div>
<div class="paragraph">
<p>This rule checks that static non-final field names match a provided regular expression.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="paragraph">
<p>The default regular expression applied by the rule is <code>^[a-z][a-zA-Z0-9]*$</code>:</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">public class MyClass {
   private static String foo_bar; // Noncompliant
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">public class MyClass {
   private static String fooBar;
}</code></pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p><a href="https://www.oreilly.com/library/view/java-8-pocket/9781491901083/ch01.html">O&#8217;Reilly - Java 8 in pocket - Naming Conventions</a></p>
</li>
<li>
<p><a href="https://www.educative.io/answers/what-are-the-naming-conventions-in-java">Educative - Naming conventions in Java</a></p>
</li>
</ul>
</div>
<hr>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Rename this field "XXX" to match the regular expression ${format}.</p>
</div>
</div>
<div class="sect2">
<h3 id="_parameters">Parameters</h3>
<div class="sidebarblock">
<div class="content">
<div class="title">format</div>
<div class="listingblock">
<div class="content">
<pre>^[a-z][a-zA-Z0-9]*$</pre>
</div>
</div>
<div class="paragraph">
<p>Regular expression used to check the field names against.</p>
</div>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_9_jun_2015_134521_ann_campbell_wrote">on 9 Jun 2015, 13:45:21 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>FYI [~alexandre.gigleux] I&#8217;ve made a few updates.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_9_jun_2015_134813_ann_campbell_wrote">on 9 Jun 2015, 13:48:13 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>Note for Java: same as <a data-rspec-id="S116" class="rspec-auto-link">RSPEC-116</a> but for static non-final fields</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_12_jun_2015_160459_ann_campbell_wrote">on 12 Jun 2015, 16:04:59 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>CodePro: Static Field Naming Convention</p>
</div>
</div>
</div>
</div>