== Why is this an issue?

HTML, SVG, and MathML define the selectors which can be used in a CSS. A selector that is not part of them is likely to be a typo or a misunderstanding of the CSS syntax.


=== Noncompliant code example

[source,css]
----
field {}

ul list {}
----


=== Compliant solution

[source,css]
----
input {}

ul li {}
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Replace this selector with a known one.


endif::env-github,rspecator-view[]
