== Why is this an issue?

When an anonymous type's properties are copied from properties or variables with the same names, it yields cleaner code to omit the new type's property name and the assignment operator.


=== Noncompliant code example

[source,csharp]
----
var X = 5;

var anon = new 
{
  X = X, //Noncompliant, the new object would have the same property without the "X =" part.
  Y = "my string"
};
----


=== Compliant solution

[source,csharp]
----
var X = 5;

var anon = new 
{
  X,
  Y = "my string"
};
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Remove the redundant "xxx=".


=== Highlighting

xxx=


'''
== Comments And Links
(visible only on this page)

=== on 8 Dec 2015, 08:42:49 Tamas Vajk wrote:
\[~ann.campbell.2] LGTM

endif::env-github,rspecator-view[]
