include::../description.adoc[]

== Noncompliant Code Example

[source,php]
----
if (condition1) {
  if (condition2) {        
    ...
  }
}
----

== Compliant Solution

[source,php]
----
if (condition1 && condition2) {        
  ...
}
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../highlighting.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
