Methods that override parent class implementations should end with calls to the ``++super++`` implementations. Otherwise, you risk leaving the job half done.


This rule raises an issue when the following ``++Activity++`` methods do not end with a call to ``++super++``:

* ``++onDestroy++``
* ``++onPause++``
* ``++onSaveInstanceState++``
* ``++onStop++``
* ``++onTerminate++``


== Noncompliant Code Example

[source,text]
----
//TODO
----


== Compliant Solution

[source,text]
----
//TODO
----

