== Why is this an issue?

:link-with-uscores1: https://docs.python.org/3/reference/datamodel.html#object.__call__

In order to be callable, a python class should implement the {link-with-uscores1}[``++__call__++``] method.


This rule raises an issue when a non-callable object is called.


=== Noncompliant code example

[source,python]
----
class MyClass:
    pass

myvar = MyClass()
myvar()  # Noncompliant

none_var = None
none_var()  # Noncompliant
----


=== Compliant solution

[source,python]
----
class MyClass:
    def __call__(self):
        print("called")

myvar = MyClass()
myvar()
----


:link-with-uscores1: https://docs.python.org/3/reference/datamodel.html#object.__call__

== Resources

* {link-with-uscores1}[Python documentation - ++__call__++ method]


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

include::highlighting.adoc[]

'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
