This rule raises an issue when `logging.getLogger()` is called inside a function or method definition.

== Why is this an issue?

Declaring loggers inside functions or methods creates unnecessary overhead and goes against Django's recommended conventions.

When you call `logging.getLogger()` inside a function, Python creates a new logger instance every time the function runs. This is inefficient because the same logger could be reused across multiple function calls.

The Django documentation specifically recommends declaring loggers at module level as a good convention. This approach:

* Improves performance by creating the logger only once per module
* Makes the code more readable by clearly showing which loggers the module uses
* Follows the standard Python logging pattern used throughout the Django framework
* Ensures consistent logger naming when using `\__name__`

Module-level logger declarations are the standard practice in Python applications and are explicitly recommended in Django's logging documentation.

=== What is the potential impact?

Declaring loggers inside functions creates minor performance overhead due to repeated logger instantiation. While not critical, it violates established conventions and can make code less maintainable.

== How to fix it

Move the logger declaration to module level, outside of any function or method definition.

=== Code examples

==== Noncompliant code example

[source,python,diff-id=1,diff-type=noncompliant]
----
import logging

def some_view(request):
    logger = logging.getLogger(__name__)  # Noncompliant
    logger.warning('Something happened')
    return HttpResponse('OK')
----

==== Compliant solution

[source,python,diff-id=1,diff-type=compliant]
----
import logging

logger = logging.getLogger(__name__)  # Declared at module level

def some_view(request):
    logger.warning('Something happened')
    return HttpResponse('OK')
----

== Resources

=== Documentation

 * Django Logging Documentation - https://docs.djangoproject.com/en/stable/howto/logging/[Official Django guide on configuring and using logging]

 * Python Logging HOWTO - https://docs.python.org/3/howto/logging.html[Python's official logging documentation and best practices]
