==== Noncompliant code example

[source,java,diff-id=1,diff-type=noncompliant,subs="attributes"]
----
ifdef::context_key+context_value[]
props.set("{context_key}", "{context_value}")
endif::[]

props.set("{example_name}", "{example_secret}") // Noncompliant
----

==== Compliant solution

[source,java,diff-id=1,diff-type=compliant,subs="attributes"]
----
ifdef::context_key+context_value[]
props.set("{context_key}", "{context_value}")
endif::[]

props.set("{example_name}", System.getenv("{example_env}"))
----
