<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Using several "--" or "&#43;&#43;" unary operators in the same arithmetic expression can quickly make the expression unreadable.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-flex" data-lang="flex">var j:int = foo++ - --bar;</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-flex" data-lang="flex">bar--;
var j:int = foo++ - bar;</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Split this expression into multiple expressions so that each one contains no more than a single "&#43;&#43;" or "--" unary operator</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_22_nov_2013_135146_freddy_mallet_wrote">on 22 Nov 2013, 13:51:46 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>Is implemented by https://jira.codehaus.org/browse/SONARPLUGINS-3284 for Flex</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_10_dec_2013_115306_freddy_mallet_wrote">on 10 Dec 2013, 11:53:06 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>Is implemented by http://jira.codehaus.org/browse/SONARJAVA-393 for Java</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_11_dec_2013_144546_freddy_mallet_wrote">on 11 Dec 2013, 14:45:46 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>I&#8217;m closing this spec as <a data-rspec-id="S881" class="rspec-auto-link">RSPEC-881</a> is definitely better.</p>
</div>
</div>
</div>
</div>