== Why is this an issue?

Using a type parameter when you don't have to simply obfuscates the code. Qualifying an inner type with a type parameter will compile, but confuse maintainers.


=== Noncompliant code example

[source,java]
----
<T extends Map> T doTheThing(T.Entry type) {  // Noncompliant
  //...
}
----


=== Compliant solution

[source,java]
----
<T extends Map> T doTheThing(Map.Entry type) {
  //...
}
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Use "xxx" here instead of "y".


=== Highlighting

Unnecessary type param


endif::env-github,rspecator-view[]
