== How to fix it in Dom4j

=== Code examples

==== Noncompliant code example

[source,java,diff-id=2,diff-type=noncompliant]
----
import org.dom4j.io.SAXReader;

SAXReader xmlReader = new SAXReader();
xmlReader.setFeature(XMLConstants.FEATURE_SECURE_PROCESSING, false); // Noncompliant
----

==== Compliant solution

[source,java,diff-id=2,diff-type=compliant]
----
import org.dom4j.io.SAXReader;

SAXReader xmlReader = new SAXReader();
xmlReader.setFeature(XMLConstants.FEATURE_SECURE_PROCESSING, true);
----
