== How to fix it in Jdom2

=== Code examples

==== Noncompliant code example

[source,java,diff-id=3,diff-type=noncompliant]
----
import org.jdom2.input.SAXBuilder;

SAXBuilder builder = new SAXBuilder();
builder.setFeature(XMLConstants.FEATURE_SECURE_PROCESSING, false);  // Noncompliant
----

==== Compliant solution

[source,java,diff-id=3,diff-type=compliant]
----
import org.jdom2.input.SAXBuilder;

SAXBuilder builder = new SAXBuilder();
builder.setFeature(XMLConstants.FEATURE_SECURE_PROCESSING, true);
----
