<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Earlier versions of z/OS and certain installations do not support in-stream data in procedures.
This can lead to inconsistencies within the codebase and compatibility issues.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Remove the uses of <code>DD *</code> and <code>DD DATA</code> from procedures.
You can add in-stream data to a procedure step by using <code>DD DUMMY</code> inside the procedure and overriding that by a <code>DD *</code> or <code>DD DATA</code> statement in the calling step.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-jcl" data-lang="jcl">//MYPROC   PROC
//STEP1    EXEC PGM=DOTHING
//SYSIN    DD *
  Input data
/*
//         PEND
//* Call procedure
//CALLPRC  EXEC PROC=MYPROC</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-jcl" data-lang="jcl">//MYPROC   PROC
//STEP1    EXEC PGM=DOTHING
//SYSIN    DD DUMMY
//         PEND
//* Call procedure
//CALLPRC  EXEC PROC=MYPROC
//STEP1.SYSIN DD *
  Input data
/*</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>