<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when a <code>TypeVar</code> is used in a function instead of the generic function type syntax.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Prior to Python 3.12 functions using generic types were created as follows:</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">from typing import TypeVar

_T = TypeVar("_T")

def func(a: _T, b: _T) -&gt; _T:
    ...</code></pre>
</div>
</div>
<div class="paragraph">
<p>This snippet of code can be confusing and difficult to read. This is why is it a good idea to
use the new type parameter syntax of Python 3.12 that allows for a more concise and more readable code by removing the
<code>TypeVar</code> and its import statement:</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">def func[T](a: T, b: T) -&gt; T:
    ...</code></pre>
</div>
</div>
<div class="sect2">
<h3 id="_exceptions">Exceptions</h3>
<div class="paragraph">
<p>This rule will only raise an issue when the Python version of the analyzed project is set to 3.12 or higher.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>To fix this issue, replace the usage of a <code>TypeVar</code> as a parameter type with the generic type parameter syntax.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">from typing import TypeVar

_T = TypeVar("_T", bound=str)

def func(a: _T, b: int) -&gt; _T: # Noncompliant: the usage of a TypeVar could be simplified with a generic type parameter.
    ...</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">def func[T: str](a: T, b: int) -&gt; T: # Compliant: the code is clear and concise.
    ...</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Python Documentation - <a href="https://docs.python.org/3.12/reference/simple_stmts.html#type">The type statement</a></p>
</li>
<li>
<p>Python 3.12 Release Notes - <a href="https://docs.python.org/3.12/whatsnew/3.12.html#pep-695-type-parameter-syntax">PEP 695: Type Parameter Syntax</a></p>
</li>
<li>
<p>PEP 695 - <a href="https://peps.python.org/pep-0695/">Type Parameter Syntax</a></p>
</li>
</ul>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="paragraph">
<p>Use a generic type parameter for this function instead of a <code>TypeVar</code>.</p>
</div>
<hr>
</div>
</div>
</div>