== Why is this an issue?

All reserved words should be written using the same case to ensure consistency in the code.

This rule checks that reserved words are all in lower case.


=== Noncompliant code example

[source,sql]
----
begin
  null;
  NULL; -- Noncompliant
end;
/
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Lowercase the reserved word "{0}".


endif::env-github,rspecator-view[]
