All reserved words should be written using the same case to ensure consistency in the code.

This rule checks that reserved words are all in lower case.


== Noncompliant Code Example

----
begin
  null;
  NULL; -- Noncompliant
end;
/
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

endif::env-github,rspecator-view[]
