<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when a method annotated with <code>@future</code> is not declared as <code>static</code> or does not have a <code>void</code> return type.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Methods annotated with <code>@future</code> in Apex execute asynchronously in a separate thread when system resources become available. This asynchronous execution model has two fundamental requirements:</p>
</div>
<div class="paragraph">
<p>First, future methods must be <code>static</code> because they cannot rely on instance state. When a future method executes, it runs independently of the original calling context, potentially long after the original object instance has been destroyed or modified. Static methods ensure the code can execute without requiring access to instance variables or methods.</p>
</div>
<div class="paragraph">
<p>Second, future methods must return <code>void</code> because they cannot return values to the calling code. The calling thread continues execution immediately after invoking the future method, without waiting for its completion. By the time the future method actually executes, the original caller has already moved on, making it impossible to receive any return value.</p>
</div>
<div class="paragraph">
<p>These are not just recommendations but strict compilation requirements enforced by the Apex compiler. Any attempt to declare a future method as non-static or with a non-void return type will result in compilation errors, preventing the code from being saved or deployed.</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="paragraph">
<p>Violating this rule will cause compilation errors, preventing the code from being saved, deployed, or executed. This breaks the build process and blocks development progress until the method signature is corrected.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Declare the method as static and change the return type to void. If you need to return data, consider using alternative approaches like updating database records or sending platform events.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-apex" data-lang="apex">@future
public String processData(String data) { // Noncompliant
    // Process data
    return "processed";
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-apex" data-lang="apex">@future
public static void processData(String data) {
    // Process data
    // Store results in database or send platform events if needed
    System.debug('Data processed: ' + data);
}</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Future Methods - <a href="https://developer.salesforce.com/docs/atlas.en-us.apexcode.meta/apexcode/apex_invoking_future_methods.htm">Official Salesforce documentation on future methods and their requirements</a></p>
</li>
<li>
<p>Asynchronous Apex - <a href="https://developer.salesforce.com/docs/atlas.en-us.apexcode.meta/apexcode/apex_async_overview.htm">Overview of asynchronous processing options in Apex</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>