Trailing closure syntax can only be used with the last argument to a function call. Place a function type parameter anywhere else in the list and you limit the options of the caller.


== Noncompliant Code Example

----
func foo(p1: Int->Int, p2: Int){  // Noncompliant; p1 should come at the end
  print(p1(p2))
}

foo({a in a * 2}, 42) // Trailing closure syntax can't be used here
----


== Compliant Solution

----
func foo(p2: Int, p1: Int->Int){
  print(p1(p2))
}

foo(42) {a in a * 2}
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
