This rule raises an issue when the arguments passed to a function call match the names of its parameters but are passed in a different order.

== Why is this an issue?
When the names of parameters in a function/method call match the names of the method arguments, it contributes to clearer, more readable code. However, when the names match, but are passed in a different order than the function/method arguments, it indicates a mistake in the parameter order which will likely lead to unexpected results.

Python functions typically expect arguments to be provided in a specific order to ensure the correct execution of the function. When arguments are passed in a different order, the function may receive incorrect values for its parameters, resulting in semantic discrepancies. This can lead to incorrect computations, unexpected behavior, or even errors.

In case of when parameters with matching names set in different order are correct from a code execution perspective, it could be confusing from code understanding and should be avoided.

== How to fix it

Make sure that function call arguments are in the correct order.

=== Code examples

==== Noncompliant code example

[source,python,diff-id=1,diff-type=noncompliant]
----
def move_point(coord, speed):
    new_x = coord[0] + speed[0]
    new_y = coord[1] + speed[1]
    return (new_x, new_y)

coord = (3, 4)
speed = (1, 2)

move_point(speed, coord)  # Noncompliant
----

==== Compliant solution 

[source,python,diff-id=1,diff-type=compliant]
----
def move_point(coord, speed):
    new_x = coord[0] + speed[0]
    new_y = coord[1] + speed[1]
    return (new_x, new_y)

coord = (3, 4)
speed = (1, 2)

move_point(coord, speed)
----

== Resources

=== Documentation

* * Python Documentation - https://docs.python.org/3/tutorial/controlflow.html#defining-functions[Defining functions]


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

* Make sure this method arguments set in correct order.


=== Highlighting

Primary: method name

'''
== Comments And Links
(visible only on this page)


endif::env-github,rspecator-view[]
