This rule raises an issue when the argument to or the return type from the functions `list()`, `tuple()`, `set()`, `sorted()`, or `reversed()` are unnecessarily casted.

== Why is this an issue?

Python's built-in functions for processing iterables such as `list()`, `tuple()`, `set()`, `sorted()`, and `reversed()` are designed to accept any iterable as input. When these functions are unnecessarily nested within each other, it creates redundant operations that add unnecessary computational overhead by creating intermediate data structures, decrease code readability and make the intention less clear, and waste memory by duplicating data structures temporarily.


== How to fix it

When the outer function is given a collection but could be given an iterable, the unnecessary cast to the collection should be removed. For example, in `sorted(list(iterable))`, the outer `sorted()` function can accept an iterable directly, so the inner `list()` call is redundant and should be removed.

When the function `sorted()` is casted to `list()`, remove the cast, since `sorted()` already returns a list.

=== Code examples

==== Noncompliant code example

[source,python,diff-id=1,diff-type=noncompliant]
----
iterable = (3, 1, 4, 1)

list_of_list = list(list(iterable)) # Noncompliant
set_of_list = set(list(iterable)) # Noncompliant
sorted_of_list = list(sorted(iterable)) # Noncompliant
----

==== Compliant solution

[source,python,diff-id=1,diff-type=compliant]
----
iterable = (3, 1, 4, 1)

list_of_list = list(iterable) # Noncompliant
set_of_list = set(iterable) # Noncompliant
sorted_of_list = sorted(iterable) # Noncompliant
----

== Resources
=== Documentation
* Python Documentation - https://docs.python.org/3/library/functions.html#list[list]
* Python Documentation - https://docs.python.org/3/library/functions.html#tuple[tuple]
* Python Documentation - https://docs.python.org/3/library/functions.html#set[set]
* Python Documentation - https://docs.python.org/3/library/functions.html#sorted[sorted]
* Python Documentation - https://docs.python.org/3/library/functions.html#reversed[reversed]

ifdef::env-github,rspecator-view[]

== Implementation details

The rule specified by SONARPY-2877 specifies the special case of `reverse(sorted(...))` and `sorted(reverse(...))`. Because of this, this rule cannot raise in these cases to avoid raising twice for the same issue.


=== Highlight

The function call which should be removed should be highlighted.

=== Message

Remove this redundant cast.

=== Quickfix


endif::env-github,rspecator-view[]