include::../description.adoc[]

== Noncompliant Code Example

----
      /free
        name = name;
      /end-free
----

== Compliant Solution

----
      /free
        customer.name = name;
      /end-free
----

include::../see.adoc[]

ifdef::env-github,rspecator-view[]
'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
