== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,rpg]
----
      /free
        name = name;
      /end-free
----

=== Compliant solution

[source,rpg]
----
      /free
        customer.name = name;
      /end-free
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
