== Why is this an issue?

Surrounding your operators with whitespace in operator declarations will help maintainers derive meaning from what might otherwise look like a meaningless jumble of punctuation.


=== Noncompliant code example

[source,swift]
----
func <*>(a: MyClass, b: MyClass) -> Boolean { // Noncompliant
----


=== Compliant solution

[source,swift]
----
func <*> (a: MyClass, b: MyClass) -> Boolean {
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
