== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

With the default value: ``++^((Test|IT)[a-zA-Z0-9]+|[A-Z][a-zA-Z0-9]*(Test|IT|TestCase|ITCase))$++``

[source,csharp]
----
[TestClass]
public class Foo // Noncompliant
{
----

=== Compliant solution

[source,csharp]
----
[TestClass]
public class FooTest
{
----
ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../parameters.adoc[]

endif::env-github,rspecator-view[]
