<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p><code>std::format</code> takes as an argument a format string that contains replacement fields (surrounded with <code>{}</code>)
and a set of extra arguments that will be formatted inside the replacement fields.
Even if the format string is checked at compile-time, it is possible to have a mismatch between the format string and the arguments. For example:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>The format string contains fewer replacement fields than the number of extra arguments:
<code>std::format("{} {}", 1, 2, 3);</code></p>
</li>
<li>
<p>The format string uses indexes for the replacement fields, but one index is missing:
<code>std::format("{0} {0} {2}", 1, 2, 3);</code></p>
</li>
</ul>
</div>
<div class="paragraph">
<p>In these cases, the extra arguments are silently ignored. In the best-case scenario, it leads to dead code.
Otherwise, it is a typo, and the output will not be intended.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>An unused argument comes from a complex expression, such as a function call.</p>
</li>
<li>
<p>A combination of numbered replacement fields, a missing index, and a repeated index exists.</p>
</li>
</ul>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">std::cout &lt;&lt; std::format("{} {}", 1, 2, sqrt(2));
std::cout &lt;&lt; std::format("{0} {0} {2}", 1, 2, 3);</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">std::cout &lt;&lt; std::format("{} {} {}", 1, 2, sqrt(2));
std::cout &lt;&lt; std::format("{0} {0} {2} {1}", 1, 2, 3);</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p><a data-rspec-id="S6488" class="rspec-auto-link">S6488</a> - To detect all unused arguments in <code>std::format</code>.</p>
</li>
</ul>
</div>
</div>
</div>