== Why is this an issue?

The use of ``++Option Base++`` to change the lower bound of an array's index values results in confusing code.


=== Noncompliant code example

[source,vb6]
----
Option Explicit
Option Base 1

'...
    Dim MyArray(1 To 3) As Integer
    For I = 1 To 3
        MsgBox MyArray(I)
    Next I
----


=== Compliant solution

[source,vb6]
----
Option Explicit

'...
    Dim MyArray(0 To 2) As Integer
    For I = 0 To 2
        MsgBox MyArray(I)
    Next I
----



ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
