<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Having all the pieces in front of you makes it easier to understand and debug or maintain a piece of code. Unfortunately, that&#8217;s often not the case with compile-time arrays, since they are often separated from the code that uses them by many, many lines. Instead, data structures are often a better option.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-rpg" data-lang="rpg">     D directions      S              5  0 dim(4) ctdata perrcd(1)</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-rpg" data-lang="rpg">     D dirData         DS
     D                                5    Inz('North')
     D                                5    Inz('South')
     D                                5    Inz('East')
     D                                5    Inz('West')
     D directions                     5    Dim(4) Overlay(dirData)</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Move this array into a data structure</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_2_apr_2015_190330_ann_campbell_wrote">on 2 Apr 2015, 19:03:30 Ann Campbell wrote:</h3>
<div class="paragraph">
<p><a href="http://www.bmeyers.net/faqs/14-tips/32-rpg-iv-style?start=3" class="bare">http://www.bmeyers.net/faqs/14-tips/32-rpg-iv-style?start=3</a></p>
</div>
</div>
</div>
</div>