Certain functions exist in PHP only as aliases of other functions. These aliases have been made available for backward compatibility, but should really be removed from code. 

This rule looks for uses of the following aliases:
||Alias||Replacement||
|``chop``|``rtrim``|
|``close``|``closedir``|
|``doubleval``|``floatval``|
|``fputs``|``fwrite``|
|``ini_alter``|``ini_set``|
|``is_double``|``is_float``|
|``is_integer``|``is_int``|
|``is_long``|``is_int``|
|``is_real``|``is_float``|
|``is_writeable``|``is_writable``|
|``join``|``implode``|
|``key_exists``|``array_key_exists``|
|``magic_quotes_runtime``|``set_magic_quotes_runtime``|
|``pos``|``current``|
|``show_source``|``highlight_file``|
|``sizeof``|``count``|
|``strchr``|``strstr``|


== Noncompliant Code Example

----
$arr=array("apple", "pear","banana");
echo sizeof($arr);  // Noncompliant
----


== Compliant Solution

----
$arr=array("apple", "pear","banana");
echo count($arr);
----

