== Why is this an issue?

Shared coding conventions allow teams to collaborate efficiently. This rule checks that reserved words are written in upper case.


=== Noncompliant code example

[source,sql]
----
BEGIN
  null; -- Noncompliant
END;
/
----


=== Compliant solution

[source,sql]
----
BEGIN
  NULL;
END;
/
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Use "XXX" instead.


endif::env-github,rspecator-view[]
