<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Redundant comparisons against constants results in always-true or always-false conditions. They add unnecessary complexity to the code and often indicate a logical error where the developer might have intended a different comparison operator or a comparison with another value.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-rust" data-lang="rust">let status_code = 200;
if status_code &lt;= 400 &amp;&amp; status_code &lt; 500 {} // Noncompliant: Redundant double comparison.</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-rust" data-lang="rust">let status_code = 200;
if status_code &lt; 500 {} // Compliant: Corrected the comparison logic.</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Clippy Lints - <a href="https://rust-lang.github.io/rust-clippy/master/index.html#redundant_comparisons" class="bare">https://rust-lang.github.io/rust-clippy/master/index.html#redundant_comparisons</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>