``++Eval++`` and ``++Callp++`` are the only two exceptions to the rule that each free-format line must start with an operation code. Since you can leave these two opcodes out, you should for cleaner, more readable code.


== Noncompliant Code Example

----
Eval Regpay = hours * perHour;
Callp calcTax();
----


== Compliant Solution

----
Regpay = hours * perHour;
calcTax();
----



ifdef::env-github,rspecator-view[]
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
