== Why is this an issue?

``++Eval++`` and ``++Callp++`` are the only two exceptions to the rule that each free-format line must start with an operation code. Since you can leave these two opcodes out, you should for cleaner, more readable code.


=== Noncompliant code example

[source,rpg]
----
Eval Regpay = hours * perHour;
Callp calcTax();
----


=== Compliant solution

[source,rpg]
----
Regpay = hours * perHour;
calcTax();
----



ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

"X" is optional here; leave it out.


'''
== Comments And Links
(visible only on this page)

=== on 2 Apr 2015, 18:00:30 Ann Campbell wrote:
http://www.bmeyers.net/faqs/14-tips/32-rpg-iv-style?start=1

endif::env-github,rspecator-view[]
