== Why is this an issue?

Declaring a variable with the NOT NULL constraint incurs a small performance cost - while this constraint may not really be required. Using such a constraint should be avoided.


=== Noncompliant code example

[source,sql]
----
DECLARE
  counter PLS_INTEGER NOT NULL := 0; -- Noncompliant
BEGIN
  NULL;
END;
/
----


=== Compliant solution

[source,sql]
----
DECLARE
  counter PLS_INTEGER := 0; -- Compliant
BEGIN
  NULL;
END;
/
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

endif::env-github,rspecator-view[]
