If a type is declared but not used, then it is unclear to a reviewer if the type is redundant or it has been left unused by mistake.


== Noncompliant Code Example

----
void unusedtype()
{
  typedef int local_Type; // Noncompliant, unused
}
----


== See

* MISRA {cpp}:2008, 0-1-5 - A project shall not contain unused type declarations.


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
