== Why is this an issue?

If a type is declared but not used, then it is unclear to a reviewer if the type is redundant or it has been left unused by mistake.


=== Noncompliant code example

[source,cpp]
----
void unusedtype()
{
  typedef int local_Type; // Noncompliant, unused
}
----


== Resources

* MISRA {cpp}:2008, 0-1-5 - A project shall not contain unused type declarations.


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Remove the unused declaration of  type "xxx".


'''
== Comments And Links
(visible only on this page)

=== is related to: S1763

=== relates to: S901

=== is related to: S1029

=== on 22 Oct 2014, 16:47:43 Samuel Mercier wrote:
Should not be enabled while analysing a library.

=== on 23 Oct 2014, 14:46:58 Ann Campbell wrote:
\[~samuel.mercier] please consider other relevant tags. E.g. 'unused' for this rule

endif::env-github,rspecator-view[]
