include::../description.adoc[]

== Noncompliant Code Example

----
public function divide($divisor, $dividend) {
  return $divisor/$dividend;
}

public function doTheThing() {
  $divisor = 15;
  $dividend = 5;

  $result = $this->divide($dividend, $divisor);  // Noncompliant; operation succeeds, but result is unexpected
}
----

== Compliant Solution

----
public function divide($divisor, $dividend) { 
  return $divisor/$dividend;
}

public function doTheThing() {
  $divisor = 15;
  $dividend = 5;

  $result = $this->divide($divisor, $dividend); // Compliant
}{code}
h4. Â 
----

ifdef::env-github,rspecator-view[]
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
