== Why is this an issue?

include::../description.adoc[]

=== Noncompliant code example

[source,php]
----
public function divide($divisor, $dividend) {
  return $divisor/$dividend;
}

public function doTheThing() {
  $divisor = 15;
  $dividend = 5;

  $result = $this->divide($dividend, $divisor);  // Noncompliant; operation succeeds, but result is unexpected
}
----

=== Compliant solution

[source,php]
----
public function divide($divisor, $dividend) { 
  return $divisor/$dividend;
}

public function doTheThing() {
  $divisor = 15;
  $dividend = 5;

  $result = $this->divide($divisor, $dividend); // Compliant
}{code}
h4. Â 
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../highlighting.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]

endif::env-github,rspecator-view[]
