<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Calling <code>Class.newInstance</code> invokes the class' default constructor. Unfortunately, since it&#8217;s not a direct call to the constructor, compile-time checking will be unable to detect the possibility. This means that your code will compile even if you haven&#8217;t put the invocation in a <code>try</code> block.</p>
</div>
<div class="paragraph">
<p>On the other hand, <code>Construtor.newInstance</code> handles exceptions by wrapping them in an <code>InvocationTargetException</code> and explicitly throwing them.</p>
</div>
<div class="paragraph">
<p>This rule raises an issue when <code>Class.newInstance</code> is used to invoke a constructor that throws checked exceptions.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">Foo f = Foo.class.newInstance();  // Noncompliant</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">Foo f = Foo.class.getConstructor().newInstance();</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="ulist">
<ul>
<li>
<p>Use "xxx.getConstructor().newInstance()" instead.</p>
</li>
</ul>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p><code>xxx.newInstance()</code></p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_14_jul_2016_180106_ann_campbell_wrote">on 14 Jul 2016, 18:01:06 Ann Campbell wrote:</h3>
<div class="paragraph">
<p><a href="https://github.com/google/error-prone/blob/master/docs/bugpattern/ClassNewInstance.md" class="bare">https://github.com/google/error-prone/blob/master/docs/bugpattern/ClassNewInstance.md</a></p>
</div>
</div>
<div class="sect2">
<h3 id="_on_15_jul_2016_093816_freddy_mallet_wrote">on 15 Jul 2016, 09:38:16 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>Hi [~ann.campbell.2], I would convert this "Code Smells Detection Rule" into a "Bugs Detection Rule" by making it raising an issue : IF AND ONLY IF the "class.newInstance()" leads to call a constructor having some checked exceptions. This is another example of the following pattern : instead of enforcing a coding practice allowing to avoid facing some bugs from time to time, let&#8217;s provide a rule able to precisely detect those bugs.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_15_jul_2016_144014_freddy_mallet_wrote">on 15 Jul 2016, 14:40:14 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>And in that case [~ann.campbell.2], the title and description of the rule should also evolve. For the title, that might become: "'Class.newIntance(&#8230;&#8203;)' should be not be called when the relating nullary constructor throws checked exceptions".</p>
</div>
</div>
</div>
</div>