Calling ``++Class.newInstance++`` invokes the class' default constructor. Unfortunately, since it's not a direct call to the constructor, compile-time checking will be unable to detect the possibility. This means that your code will compile even if you haven't put the invocation in a ``++try++`` block.

On the other hand, ``++Construtor.newInstance++`` handles exceptions by wrapping them in an ``++InvocationTargetException++`` and explicitly throwing them. 

This rule raises an issue when ``++Class.newInstance++`` is used to invoke a constructor that throws checked exceptions.


== Noncompliant Code Example

----
Foo f = Foo.class.newInstance();  // Noncompliant
----


== Compliant Solution

----
Foo f = Foo.class.getConstructor().newInstance();
----

