== How to fix it in archive/zip

=== Code examples

:canonicalization_function1: filepath.Clean
:canonicalization_function2: filepath.Abs

include::../../common/fix/code-rationale.adoc[]

==== Noncompliant code example

[source,go,diff-id=1,diff-type=noncompliant]
----
import (
	"archive/zip"
	"io"
	"os"
	"path/filepath"
)

func Extract(zipFile *zip.Reader, targetDirectory string) {
	for _, file := range zipFile.File {
		srcFile, _ := file.Open()
		outFilename := filepath.Join(targetDirectory, file.Name)

		outFile, _ := os.Create(outFilename)
		io.Copy(outFile, srcFile)
	}
}

func Example() {
	zipFile, _ := zip.OpenReader("testdata/example.zip")
	Extract(&zipFile.Reader, "/tmp/uploads")
}
----

==== Compliant solution

[source,go,diff-id=1,diff-type=compliant]
----
import (
	"archive/zip"
	"io"
	"os"
	"path/filepath"
	"strings"
)

func Extract(zipFile *zip.Reader, targetDirectory string) {
	for _, file := range zipFile.File {
		srcFile, _ := file.Open()
		outFilename := filepath.Join(targetDirectory, file.Name)

		if !strings.HasPrefix(filepath.Clean(outFilename), targetDirectory) {
			continue // zip slip detected
		}

		outFile, _ := os.Create(outFilename)
		io.Copy(outFile, srcFile)
	}
}

func Example() {
	zipFile, _ := zip.OpenReader("testdata/example.zip")
	Extract(&zipFile.Reader, "/tmp/uploads/") // Note the trailing slash
}
----

=== How does this work?

include::../../common/fix/how-does-this-work.adoc[]

=== Pitfalls

include::../../common/pitfalls/partial-path-traversal.adoc[]

include::go-pitfall-example.adoc[]
