<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>A <code>for</code> loop stop condition should test the loop counter against an invariant value (i.e. one that is true at both the beginning and ending of every loop iteration). Ideally, this means that the stop condition is set to a local variable just before the loop begins.</p>
</div>
<div class="paragraph">
<p>Stop conditions that are not invariant are slightly less efficient, as well as being difficult to understand and maintain, and likely lead to the introduction of errors in the future.</p>
</div>
<div class="paragraph">
<p>This rule tracks three types of non-invariant stop conditions:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>When the loop counters are updated in the body of the <code>for</code> loop</p>
</li>
<li>
<p>When the stop condition depend upon a method call</p>
</li>
<li>
<p>When the stop condition depends on an object property, since such properties could change during the execution of the loop.</p>
</li>
</ul>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-php" data-lang="php">for ($i = 0; $i &lt; 10; $i++) {
  echo $i;
  if(condition) {
    $i = 20;
  }
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-php" data-lang="php">for ($i = 0; $i &lt; 10; $i++) {
  echo $i;
}</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="ulist">
<ul>
<li>
<p>Do not update the loop counter "X" within the loop body.</p>
</li>
<li>
<p>Calculate the stop condition value outside the loop and set it to a variable.</p>
</li>
</ul>
</div>
</div>
<div class="sect2">
<h3 id="_parameters">Parameters</h3>
<div class="sidebarblock">
<div class="content">
<div class="title">check_attributes</div>
<div class="listingblock">
<div class="content">
<pre>false</pre>
</div>
</div>
<div class="paragraph">
<p>Whether the rule should flag object attributes</p>
</div>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_deprecates_s1653">deprecates: <a data-rspec-id="S1653" class="rspec-auto-link">S1653</a></h3>

</div>
<div class="sect2">
<h3 id="_deprecates_s2310">deprecates: <a data-rspec-id="S2310" class="rspec-auto-link">S2310</a></h3>

</div>
<div class="sect2">
<h3 id="_is_related_to_s5313">is related to: <a data-rspec-id="S5313" class="rspec-auto-link">S5313</a></h3>

</div>
<div class="sect2">
<h3 id="_is_related_to_s1247">is related to: <a data-rspec-id="S1247" class="rspec-auto-link">S1247</a></h3>

</div>
<div class="sect2">
<h3 id="_on_23_may_2014_143710_ann_campbell_wrote">on 23 May 2014, 14:37:10 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>Not valid for Python</p>
</div>
</div>
</div>
</div>