== How to fix it in Dom4j

=== Code examples

include::../common/code-rationale.adoc[]

==== Noncompliant code example

[source,java,diff-id=2,diff-type=noncompliant]
----
import org.dom4j.io.SAXReader;

SAXReader xmlReader = new SAXReader();
xmlReader.setFeature("http://apache.org/xml/features/xinclude", true); // Noncompliant
----

==== Compliant solution

[source,java,diff-id=2,diff-type=compliant]
----
import org.dom4j.io.SAXReader;

SAXReader xmlReader = new SAXReader(); 
xmlReader.setFeature("http://apache.org/xml/features/xinclude", false);
----