== How to fix it in Jdom2

=== Code examples

include::../common/code-rationale.adoc[]

==== Noncompliant code example

[source,java,diff-id=3,diff-type=noncompliant]
----
import org.jdom2.input.SAXBuilder;

SAXBuilder builder = new SAXBuilder();
builder.setFeature("http://apache.org/xml/features/xinclude", true); // Noncompliant
----

==== Compliant solution

[source,java,diff-id=3,diff-type=compliant]
----
import org.jdom2.input.SAXBuilder;

SAXBuilder builder = new SAXBuilder();
builder.setFeature("http://apache.org/xml/features/xinclude", false);
----