include::../description.adoc[]

== Noncompliant Code Example

[source,javascript]
----
if (!(a === 2)) { ... }  // Noncompliant
----

== Compliant Solution

[source,javascript]
----
if (a !== 2) { ... } 
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
