<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>The type of an integer literal depends on three parameters: the value of the literal, whether it is written in decimal or not, and its suffix.</p>
</div>
<div class="paragraph">
<p>A literal with a single <code>L</code> (or <code>l</code>) will usually be of a (possibly <code>unsigned</code>) <code>long</code> type. But if its value cannot be represented in this type, it will be of a (possibly <code>unsigned</code>) <code>long long</code> type instead.</p>
</div>
<div class="paragraph">
<p>In such a case, it is more straightforward to use <code>LL</code> in the literal suffix, which unambiguously specifies a <code>long long</code> type.</p>
</div>
<div class="paragraph">
<p>Note: This rule targets classical integer literals, not user-defined literals whose type is entirely determined by their suffix.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="paragraph">
<p>This example assumes that <code>long</code> is 32 bits and <code>long long</code> is 64 bits. In this case, <code>12345</code> fits into a <code>long</code>, but <code>12345678901</code> does not. Similarly, <code>0xdeadbeef</code> fits into an <code>unsigned long</code>, but <code>0xfeeddeadbeef</code> does not.</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">auto i1 = 12345678901L; // Non-compliant, the literal is of type long long
auto i2 = 12345678901UL; // Non-compliant, the literal is of type unsigned long long
auto i3 = 0xfeeddeadbeefL; // Non-compliant, the literal is of type unsigned long long

auto i4 = 12345L; // Compliant, the literal is of type long
auto i5 = 12345UL; // Compliant, the literal is of type unsigned long
auto i6 = 0xdeadbeef; // Compliant, the literal is of type unsigned long
auto i7 = 12345678901; // Compliant, no suffix is present: the compiler is free to decide</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">auto i1 = 12345678901LL; // Compliant, the literal is of type long long
auto i2 = 12345678901ULL; // Compliant, the literal is of type unsigned long long
auto i3 = 0xfeeddeadbeefLL; // Compliant, the literal is of type unsigned long long

auto i4 = 12345L; // Compliant, the literal is of type long
auto i5 = 12345UL; // Compliant, the literal is of type unsigned long
auto i6 = 0xdeadbeef; // Compliant, the literal is of type unsigned long
auto i7 = 12345678901; // Compliant, no suffix is present: the compiler is free to decide</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_standards">Standards</h3>

</div>
<div class="sect2">
<h3 id="_external_coding_guidelines">External coding guidelines</h3>
<div class="ulist">
<ul>
<li>
<p>MISRA C&#43;&#43;:2023, 5.13.6 - An integer-literal of type long long shall not use a single L or l in any suffix</p>
</li>
</ul>
</div>
</div>
</div>
</div>