<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Shared naming conventions allow teams to collaborate efficiently. This rule checks that all <code>enum</code> names match a provided regular expression.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="paragraph">
<p>With default provided regular expression: <code>^[A-Z][a-zA-Z0-9]*</code></p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-swift" data-lang="swift">enum someEnumeration { // Non-Compliant
    case Bar
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-swift" data-lang="swift">enum SomeEnumeration {
    case Bar
}</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>key: format</p>
</div>
<div class="paragraph">
<p>description: Regular expression used to check the enumeration type names against</p>
</div>
<div class="ulist">
<ul>
<li>
<p>default: ^[A-Z][a-zA-Z0-9]*$</p>
</li>
</ul>
</div>
</div>
<div class="sect2">
<h3 id="_parameters">Parameters</h3>
<div class="sidebarblock">
<div class="content">
<div class="title">format</div>
<div class="listingblock">
<div class="content">
<pre>^([A-Z]{1,3}[a-z0-9]+)*([A-Z]{2})?$</pre>
</div>
</div>
<div class="paragraph">
<p>Regular expression used to check the enumeration type names against</p>
</div>
</div>
</div>
</div>
</div>
</div>