<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Shared naming conventions allow teams to collaborate efficiently. This rule checks that all enum names match a provided regular expression.</p>
</div>
<div class="paragraph">
<p>The default configuration is the one recommended by Microsoft:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>Pascal casing, starting with an upper case character, e.g. BackColor</p>
</li>
<li>
<p>Short abbreviations of 2 letters can be capitalized, e.g. GetID</p>
</li>
<li>
<p>Longer abbreviations need to be lower case, e.g. GetHtml</p>
</li>
<li>
<p>If the enum is marked as [Flags] then its name should be plural (e.g. MyOptions), otherwise, names should be singular (e.g. MyOption)</p>
</li>
</ul>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="paragraph">
<p>With the default regular expression for non-flags enums: <code>^([A-Z]{1,3}[a-z0-9]+)*([A-Z]{2})?$</code></p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-vbnet" data-lang="vbnet">Public Enum foo ' Noncompliant
    FooValue = 0
End Enum</code></pre>
</div>
</div>
<div class="paragraph">
<p>With the default regular expression for flags enums: <code>^([A-Z]{1,3}[a-z0-9]+)*([A-Z]{2})?s$</code></p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-vbnet" data-lang="vbnet">&lt;Flags()&gt;
Public Enum Option ' Noncompliant
    None = 0,
    Option1 = 1,
    Option2 = 2
End Enum</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-vbnet" data-lang="vbnet">Public Enum Foo
    FooValue = 0
End Enum</code></pre>
</div>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-vbnet" data-lang="vbnet">&lt;Flags()&gt;
Public Enum Options
    None = 0,
    Option1 = 1,
    Option2 = 2
End Enum</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Rename this enumeration to match the regular expression "xxx".</p>
</div>
</div>
<div class="sect2">
<h3 id="_parameters">Parameters</h3>
<div class="sidebarblock">
<div class="content">
<div class="title">format</div>
<div class="listingblock">
<div class="content">
<pre>^([A-Z]{1,3}[a-z0-9]+)*([A-Z]{2})?$</pre>
</div>
</div>
<div class="paragraph">
<p>Regular expression used to check the enumeration type names against</p>
</div>
</div>
</div>
<div class="sidebarblock">
<div class="content">
<div class="title">flagsAttributeFormat</div>
<div class="listingblock">
<div class="content">
<pre>^([A-Z]{1,3}[a-z0-9]+)*([A-Z]{2})?s$</pre>
</div>
</div>
<div class="paragraph">
<p>Regular expression used to check FlagsAttribute enumeration type names against</p>
</div>
</div>
</div>
</div>
</div>
</div>