<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>An <code>INSERT</code> statement that does not explicitly list the columns being inserted into, as well as the values being inserted, is dependent for correct functioning on the structure of the table not changing. Additionally, not having the explicit column list degrades the readability and understandability of the code. Therefore, <code>INSERT</code> statements should always contain an explicit column list.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-text" data-lang="text">INSERT INTO PERSONS VALUES (1, 'DUPONT', 'Marcel')</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-text" data-lang="text">INSERT INTO PERSONS (ID, LAST_NAME, FIRST_NAME)
VALUES (1, 'DUPONT', 'Marcel')</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>List the columns to be set in this "INSERT" statement.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_6_jul_2017_135748_pierre_yves_nicolas_wrote">on 6 Jul 2017, 13:57:48 Pierre-Yves Nicolas wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] To me, this rule should not be a "bug detection" rule.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_6_jul_2017_145310_ann_campbell_wrote">on 6 Jul 2017, 14:53:10 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>updated [~pierre-yves.nicolas]</p>
</div>
</div>
</div>
</div>