include::../introduction.adoc[]

include::../why.adoc[]

include::../howToFixIt.adoc[]

=== Code examples

==== Noncompliant code example

[source,csharp]
----
Dim myDate As DateTime = New DateTime(2008, 6, 19, 7, 0, 0, DateTimeKind.Local) ' Noncompliant

Dim now = DateTime.Now ' Noncompliant
----

==== Compliant solution

[source,csharp]
----
Dim myDate As DateTimeOffset = New DateTimeOffset(2008, 6, 19, 7, 0, 0, TimeSpan.FromHours(-7)) ' Compliant

Dim now = DateTimeOffset.Now ' Compliant
----

include::../pitfalls.adoc[]

include::../resources.adoc[]