<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when Java <code>Instant</code> methods are called with unsupported temporal units that will cause runtime exceptions.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Java&#8217;s <code>Instant</code> class represents a specific point in time on the timeline. It only supports temporal units that represent exact durations, not calendar-based units that can vary in length.</p>
</div>
<div class="paragraph">
<p>The supported temporal units are:</p>
</div>
<div class="ulist">
<ul>
<li>
<p><code>NANOS</code> - nanoseconds</p>
</li>
<li>
<p><code>MICROS</code> - microseconds</p>
</li>
<li>
<p><code>MILLIS</code> - milliseconds</p>
</li>
<li>
<p><code>SECONDS</code> - seconds</p>
</li>
<li>
<p><code>MINUTES</code> - minutes</p>
</li>
<li>
<p><code>HOURS</code> - hours</p>
</li>
<li>
<p><code>HALF_DAYS</code> - 12-hour periods</p>
</li>
<li>
<p><code>DAYS</code> - 24-hour periods</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>Using unsupported units like <code>WEEKS</code>, <code>MONTHS</code>, or <code>YEARS</code> will throw an <code>UnsupportedTemporalTypeException</code> at runtime. These units are not supported because their actual duration can vary - for example, months can have 28, 29, 30, or 31 days.</p>
</div>
<div class="paragraph">
<p>Since <code>Instant</code> represents a precise moment in time, it cannot work with units that don&#8217;t have a fixed duration. This design ensures that time calculations remain predictable and accurate.</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="paragraph">
<p>Using unsupported temporal units with <code>Instant</code> methods will cause runtime crashes with <code>UnsupportedTemporalTypeException</code>, leading to service disruptions.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Replace unsupported temporal units with supported ones, or use <code>LocalDateTime</code> for calendar operations.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">Instant instant = Instant.now();
Instant result = instant.plus(1, ChronoUnit.MONTHS); // Noncompliant</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">Instant instant = Instant.now();
Instant result = instant.plus(30, ChronoUnit.DAYS);

// Or use LocalDateTime for calendar operations
LocalDateTime dateTime = LocalDateTime.now();
LocalDateTime result2 = dateTime.plus(1, ChronoUnit.MONTHS);</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Java Instant Documentation - <a href="https://docs.oracle.com/en/java/javase/25/docs/api/java.base/java/time/Instant.html">Official Java documentation for the Instant class</a></p>
</li>
<li>
<p>Java ChronoUnit Documentation - <a href="https://docs.oracle.com/en/java/javase/25/docs/api/java.base/java/time/temporal/ChronoUnit.html">Official documentation for ChronoUnit enum values</a></p>
</li>
<li>
<p>Java Time API Guide - <a href="https://docs.oracle.com/javase/tutorial/datetime/">Oracle&#8217;s tutorial on the Java Date and Time API</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>