== Why is this an issue?

When a closable statement contains nested statements, it can quickly become difficult to see which statements are nested and which are not. That's why ending a list of nested statements with ``++END-${STATEMENT-NAME}++`` is advised.


=== Noncompliant code example

[source,cobol]
----
READ DF-PARAM-SPILOTE AT END
  GO TO F-LECT-SPILOTE.
----


=== Compliant solution

[source,cobol]
----
READ DF-PARAM-SPILOTE AT END
  GO TO F-LECT-SPILOTE
END-READ.
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::parameters.adoc[]

endif::env-github,rspecator-view[]
