== Why is this an issue?

The output of ``++DBMS_OUTPUT.PUT_LINE++`` is not always visible, for example when ``++SERVEROUTPUT++`` is set to ``++OFF++``. Moreover, there is no standardized way to specify the importance of the message. It is better to use a logging mechanism instead.


=== Noncompliant code example

[source,sql]
----
SET SERVEROUTPUT ON

BEGIN
  DBMS_OUTPUT.PUT_LINE('An error occured'); -- Noncompliant
END;
/
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

endif::env-github,rspecator-view[]
