include::../description.adoc[]

== Noncompliant Code Example

With a maximum number of 4 parameters:

----
function doSomething(param1, param2, param3, param4, param5) { 
...
}
----

== Compliant Solution

----
function doSomething(param1, param2, param3, param4) {
...
}
----

ifdef::rspecator-view[]
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::rspecator-view[]
