``++StringBuilder++`` instances that are never ``++ToString++``ed needlessly clutter the code, and worse are a drag on performance. Either they should be removed, or the missing ``++ToString++`` call added.

== Noncompliant Code Example

[source,vbnet]
----
Public Sub doSomething(ByVal strings As List(Of String))
    Dim sb As StringBuilder = New StringBuilder()
    sb.Append("Got: ")

    For Each str As String In strings
        sb.Append(str).Append(", ")
    Next
End Sub
----

== Compliant Solution

[source,vbnet]
----
Public Sub doSomething(ByVal strings As List(Of String))
    For Each str As String In strings
    Next
End Sub
----
or
[source,vbnet]
----
Public Sub doSomething(ByVal strings As List(Of String))
    Dim sb As StringBuilder = New StringBuilder()
    sb.Append("Got: ")

    For Each str As String In strings
        sb.Append(str).Append(", ")
    Next

    My.Application.Log.WriteEntry(sb.toString)
End Sub
----

== Exceptions

No issue is reported when ``++StringBuilder++`` is:

* passed as method arguments, on the grounds that it will likely ``++ToString++``ed there.
[source,vbnet]
----
Dim sb = New StringBuilder() ' Compliant
ExternalMethod(sb)
----
* an argument of the current method
[source,vbnet]
----
Public Sub MyMethod(ByVal builder As StringBuilder) ' Compliant  
End Sub
----
* retrieved by a custom function
[source,vbnet]
----
Dim sb As StringBuilder = GetStringBuilder() ' Compliant
----
* returned by the method
[source,vbnet]
----
Dim sb = New StringBuilder() ' Compliant
Return sb
----
* a local variable but the .ToString() call is unreachable
[source,vbnet]
----
Dim sb As StringBuilder = New StringBuilder() ' Compliant
If False Then
    sb.ToString()
End If
----