`StringBuilder` instances that are not accessed through a `ToString()` invocation needlessly clutter the code, and worse are a drag on performance. Either they should be removed, or the missing `ToString()` call should be added.

== Noncompliant Code Example

[source,vbnet]
----
Public Sub DoSomething(ByVal strings As List(Of String))
    Dim sb As StringBuilder = New StringBuilder() ' Noncompliant
    sb.Append("Got: ")

    For Each str As String In strings
        sb.Append(str).Append(", ")
    Next
End Sub
----

== Compliant Solution

[source,vbnet]
----
Public Sub DoSomething(ByVal strings As List(Of String))
    For Each str As String In strings
    Next
End Sub
----
or
[source,vbnet]
----
Public Sub DoSomething(ByVal strings As List(Of String))
    Dim sb As StringBuilder = New StringBuilder()
    sb.Append("Got: ")

    For Each str As String In strings
        sb.Append(str).Append(", ")
    Next

    My.Application.Log.WriteEntry(sb.toString)
End Sub
----

== Exceptions

No issue is reported when `StringBuilder` is:

* Accessed through a `CopyTo()` or `GetChunks()` invocation.
* Passed as method arguments, on the grounds that it will likely accessed through a `ToString()` invocation there.
* A parameter of the current method.
* Retrieved by a custom function (`Dim sb As StringBuilder = GetStringBuilder()`).
* Returned by the method.