`StringBuilder` instances that are never `ToString`ed needlessly clutter the code, and worse are a drag on performance. Either they should be removed, or the missing `ToString` call added.

== Noncompliant Code Example

[source,vbnet]
----
Public Sub DoSomething(ByVal strings As List(Of String))
    Dim sb As StringBuilder = New StringBuilder() ' Noncompliant
    sb.Append("Got: ")

    For Each str As String In strings
        sb.Append(str).Append(", ")
    Next
End Sub
----

== Compliant Solution

[source,vbnet]
----
Public Sub DoSomething(ByVal strings As List(Of String))
    For Each str As String In strings
    Next
End Sub
----
or
[source,vbnet]
----
Public Sub DoSomething(ByVal strings As List(Of String))
    Dim sb As StringBuilder = New StringBuilder()
    sb.Append("Got: ")

    For Each str As String In strings
        sb.Append(str).Append(", ")
    Next

    My.Application.Log.WriteEntry(sb.toString)
End Sub
----

== Exceptions

No issue is reported when `StringBuilder` is:

* accessed through a ``CopyTo()`` or ``GetChunks()`` invocation.
[source,vbnet]
----
var sb = new StringBuilder(); // Compliant
sb.GetChunks()
----
* passed as method arguments, on the grounds that it will likely `ToString`ed there.
* a parameter of the current method
* retrieved by a custom function
[source,vbnet]
----
Dim sb As StringBuilder = GetStringBuilder() ' Compliant
----
* returned by the method