include::../description.adoc[]

== Noncompliant Code Example

----
System.out.println("My Message");  // Noncompliant
----

== Compliant Solution

----
logger.log("My Message");
----

== See

* https://wiki.sei.cmu.edu/confluence/x/nzdGBQ[CERT, ERR02-J.] - Prevent exceptions while logging data

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
