Availability of rare resources, such as mutexes, often has a direct impact on the performance of an application. Holding onto rare resources for too long might lead to a significant slowdown or a deadlock.


``++co_yield++`` and ``++co_await++`` suspend the execution of a coroutine for an indefinite time, and ``++co_return++`` terminates it. At this point, all the rare resources acquired by the coroutine must be released to enable other potential consumers to use them while the coroutine is inactive.


This rule reports execution scenarios that reach ``++co_yield++``, ``++co_await++``, or ``++co_return++`` without releasing a rare resource, such as a mutex.


== Noncompliant Code Example

[source,cpp]
----
std::mutex m;
int shared = 0;
cppcoro::generator<int> coroutine()
{
  while(true) {
    m.lock();
    co_yield ++shared; // Noncompliant: m is locked
    m.unlock();
  }
}
----


== Compliant Solution

[source,cpp]
----
std::mutex m;
int shared = 0;
cppcoro::generator<int> coroutine()
{
  while(true) {
    m.lock();
    ++shared;
    m.unlock();
    co_yield shared; // Compliant: m is available
  }
}
----

