== Why is this an issue?

According to http://docs.oracle.com/javase/specs/jls/se7/html/jls-8.html#jls-8.9[the Java Language Specification-8.9]:

____
Nested enum types are implicitly ``++static++``.
____


So there's no need to declare them ``++static++`` explicitly.


=== Noncompliant code example

[source,java]
----
public class Flower {
  static enum Color { // Noncompliant; static is redundant here
    RED, YELLOW, BLUE, ORANGE
  }

  // ...
}
----


=== Compliant solution

[source,java]
----
public class Flower {
  enum Color { // Compliant
    RED, YELLOW, BLUE, ORANGE
  }

  // ...
}
----



ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::message.adoc[]

include::highlighting.adoc[]

'''
== Comments And Links
(visible only on this page)

include::comments-and-links.adoc[]
endif::env-github,rspecator-view[]
