== Why is this an issue?

Each function and procedure parameter name must match a given regular expression.

=== Noncompliant code example

[source,sql]
----
DECLARE
  FUNCTION myfunction2(parameter_ PLS_INTEGER) RETURN PLS_INTEGER; -- Noncompliant

  PROCEDURE myprocedure2(parameter_ PLS_INTEGER); -- Noncompliant

  FUNCTION myfunction2(parameter_ PLS_INTEGER) RETURN PLS_INTEGER AS -- Noncompliant
  BEGIN
    RETURN 42;
  END;

  PROCEDURE myprocedure2(parameter_ PLS_INTEGER) AS -- Noncompliant
  BEGIN
    NULL;
  END;
BEGIN
  NULL;
END;
/
----

=== Compliant solution

[source,sql]
----
DECLARE
  FUNCTION myfunction1(parameter PLS_INTEGER) RETURN PLS_INTEGER; -- Compliant

  PROCEDURE myprocedure1(parameter PLS_INTEGER); -- Compliant

  FUNCTION myfunction1(parameter PLS_INTEGER) RETURN PLS_INTEGER AS -- Compliant
  BEGIN
    RETURN 42;
  END;

  PROCEDURE myprocedure1(parameter PLS_INTEGER) AS -- Compliant
  BEGIN
    NULL;
  END;

BEGIN
  NULL;
END;
/
----
ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

include::../parameters.adoc[]

endif::env-github,rspecator-view[]
