== Why is this an issue?

Using the same macro name in the include guards of two different files means that the contents of one of the files will not be included in the compilation unit. The pernicious bugs caused by this accidental exclusion can be particularly difficult to track down.


=== Noncompliant code example

[source,cpp]
----
//my_header.h
#ifndef MY_HEADER
#define MY_HEADER
...
#endif

//my_other_header.h
#ifndef MY_HEADER    // a copy-paste error?
# define MY_HEADER
...
#endif
----


=== Compliant solution

[source,cpp]
----
//my_header.h
#ifndef MY_HEADER
#define MY_HEADER
...
#endif

//my_other_header.h
#ifndef MY_OTHER_HEADER
# define MY_OTHER_HEADER
...
#endif
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Rename the "XXX" macro; it is also used in the include guard for FILE_NAME.


endif::env-github,rspecator-view[]
