<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Non-interactive DOM elements are HTML elements that are not designed to be interacted with by the user, for instance <code>&lt;div&gt;</code>, <code>&lt;span&gt;</code>, and <code>&lt;footer&gt;</code>, etc. They are typically used to structure content and do not have built-in interactivity or keyboard accessibility.</p>
</div>
<div class="paragraph">
<p>Interactive handlers, on the other hand, are event handlers that respond to user interactions. These include handlers like <code>onClick</code>, <code>onKeyDown</code>, <code>onMouseUp</code>, and more. When these handlers are added to an HTML element, they make the element respond to the specified user interaction.</p>
</div>
<div class="paragraph">
<p>When non-interactive elements have interactive handlers, it can lead to several problems:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>Unexpected behavior: Non-interactive elements are not designed to be interactive, so adding interactive handlers can cause unexpected behavior. For example, non-interactive elements do not naturally receive keyboard focus, so keyboard users might not be able to activate the handler.</p>
</li>
<li>
<p>Confusing for assistive technologies: Assistive technologies might not announce non-interactive elements with interactive handlers correctly. This can make it difficult for users to understand how to interact with the content.</p>
</li>
<li>
<p>Violation of HTML standards: Using interactive handlers on non-interactive elements can be seen as a misuse of HTML, as it goes against the intended use of these elements.</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>By enforcing that interactive handlers are only used on interactive elements, this rule helps create a more predictable and user-friendly experience for all users, and ensures that web content adheres to accessibility standards and best practices.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>The simplest and most recommended way is to replace the non-interactive elements with interactive ones. If for some reason you can&#8217;t replace the non-interactive element, you can add an interactive <code>role</code> attribute to it and manually manage its keyboard event handlers and focus. Common interactive roles include
<code>button</code>, <code>link</code>, <code>checkbox</code>, <code>menuitem</code>, <code>menuitemcheckbox</code>, <code>menuitemradio</code>, <code>option</code>, <code>radio</code>, <code>searchbox</code>, <code>switch</code>, and <code>textbox</code>.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-html" data-lang="html">&lt;div onClick="showMessage('Hello')" /&gt; &lt;!-- Noncompliant --&gt;</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-html" data-lang="html">&lt;div onClick="showMessage('Hello')" role="button" /&gt;</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>WCAG - <a href="https://www.w3.org/WAI/WCAG21/Understanding/name-role-value">Name, Role, Value</a></p>
</li>
<li>
<p>MDN web docs - <a href="https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles">WAI-ARIA Roles</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>