== Why is this an issue?

Using the standard ``++getClassLoader()++`` may not return the _right_ class loader in a JEE context. Instead, go through the ``++currentThread++``.


=== Noncompliant code example

[source,java]
----
ClassLoader cl = this.getClass().getClassLoader();  // Noncompliant
----


=== Compliant solution

[source,java]
----
ClassLoader cl = Thread.currentThread().getContextClassLoader();
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Use "Thread.currentThread().getContextClassLoader()" instead.


'''
== Comments And Links
(visible only on this page)

=== on 15 Oct 2018, 11:19:25 Massimo PALADIN wrote:
Rule removed from "Sonar Way", we cannot detect a JEE context and would generate too much noise to raise an issue for each ``++getClassLoader()++`` method invocation.

endif::env-github,rspecator-view[]
