== How to fix it

=== Code examples

include::../../common/fix/code-rationale.adoc[]

==== Noncompliant code example

[source,bash,diff-id=1,diff-type=noncompliant]
----
# Noncompliant
curl --insecure -O https://expired.example.com/downloads/install.sh
----

==== Compliant solution

[source,bash,diff-id=1,diff-type=compliant]
----
curl -O https://new.example.com/downloads/install.sh
----

==== Noncompliant code example

[source,bash,diff-id=2,diff-type=noncompliant]
----
# Noncompliant
wget --no-check-certificate https://expired.example.com/downloads/install.sh
----

==== Compliant solution

[source,bash,diff-id=2,diff-type=compliant]
----
wget https://new.example.com/downloads/install.sh
----

=== How does this work?

include::../../common/fix/validation.adoc[]
