Shared coding conventions allow teams to collaborate effectively. This rule checks that all {cpp} concept names match a provided regular expression.


== Noncompliant Code Example

With default provided regular expression ``++^[A-Z][a-zA-Z0-9]*$++``:

----
template <class T>
concept integral = std::is_integral_v<T>; // Noncompliant
----


== Compliant Solution

----
template <class T>
concept Integral = std::is_integral_v<T>; // Compliant
----


