== Why is this an issue?

Shared coding conventions allow teams to collaborate effectively. This rule checks that all {cpp} concept names match a provided regular expression.


=== Noncompliant code example

With default provided regular expression ``++^[A-Z][a-zA-Z0-9]*$++``:

[source,cpp]
----
template <class T>
concept integral = std::is_integral_v<T>; // Noncompliant
----


=== Compliant solution

[source,cpp]
----
template <class T>
concept Integral = std::is_integral_v<T>; // Compliant
----


ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Parameters

.format
****
_STRING_

----
^[A-Z][a-zA-Z0-9]*$
----

Regular expression used to check the class names against.
****


endif::env-github,rspecator-view[]
