<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule raises an issue when a method is annotated with <code>@AuraEnabled</code> but has <code>private</code> or <code>protected</code> visibility.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Lightning Web Components can only import and call Apex methods that are accessible from outside the class. The <code>@AuraEnabled</code> annotation marks a method as available for Lightning components, but the method must also have the proper visibility to be actually accessible.</p>
</div>
<div class="paragraph">
<p>When a method is marked as <code>private</code> or <code>protected</code>, it cannot be imported by Lightning Web Components, even if it has the <code>@AuraEnabled</code> annotation. This creates a mismatch between the developer&#8217;s intent (making the method available to Lightning components) and the actual accessibility of the method.</p>
</div>
<div class="paragraph">
<p>The Salesforce platform requires <code>@AuraEnabled</code> methods to be either <code>public</code> or <code>global</code> to ensure they can be properly imported and called from Lightning Web Components using the ES6 import syntax.</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="paragraph">
<p>The method will not be accessible from Lightning Web Components, causing import errors or runtime failures when the component tries to call the method. This can break the functionality of Lightning components that depend on these Apex methods.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Change the method visibility from <code>private</code> to <code>public</code> to make it accessible from Lightning Web Components.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-apex" data-lang="apex">@AuraEnabled
private static List&lt;Account&gt; getAccounts(String accountName) { // Noncompliant
    return [SELECT Id, Name FROM Account WHERE Name LIKE :accountName];
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-apex" data-lang="apex">@AuraEnabled
public static List&lt;Account&gt; getAccounts(String accountName) {
    return [SELECT Id, Name FROM Account WHERE Name LIKE :accountName];
}</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p>Lightning Web Components Developer Guide - Call Apex Methods - <a href="https://developer.salesforce.com/docs/platform/lwc/guide/apex.html">Official Salesforce documentation on calling Apex methods from Lightning Web Components</a></p>
</li>
<li>
<p>Apex Developer Guide - @AuraEnabled Annotation - <a href="https://developer.salesforce.com/docs/atlas.en-us.apexcode.meta/apexcode/apex_classes_annotation_AuraEnabled.htm">Documentation on the @AuraEnabled annotation and its requirements</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>