== How to fix it in WordPress

=== Code examples

include::../../common/fix/code-rationale.adoc[]

==== Noncompliant code example

[source,php,diff-id=11,diff-type=noncompliant]
----
define( 'WP_DEBUG', true ); // Noncompliant
----

==== Compliant solution

[source,php,diff-id=11,diff-type=compliant]
----
define( 'WP_DEBUG', false );
----

=== How does this work?

include::../../common/fix/how.adoc[]
