== How to fix it in WordPress

Do not enable debugging features on production servers or applications distributed to end users.

=== Code examples

==== Noncompliant code example

[source,php,diff-id=3,diff-type=noncompliant]
----
define( 'WP_DEBUG', true ); // Noncompliant
----

==== Compliant Solution

[source,php,diff-id=3,diff-type=compliant]
----
define( 'WP_DEBUG', false ); // Compliant
----
