include::../description.adoc[]

== Noncompliant Code Example

[source,apex]
----
public void doSomething() {
}
----

== Compliant Solution

[source,apex]
----
public void doSomething() {
  // Do nothing because of X and Y.
}
----

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

include::../message.adoc[]

'''
== Comments And Links
(visible only on this page)

include::../comments-and-links.adoc[]
endif::env-github,rspecator-view[]
