== Why is this an issue?

In the interests of readability, code that can be simplified should be simplified. When you would like to filter elements of iterable by some type, it's more concise to use method `whereType()`.
This approach improves readability band is less error-prone.

== How to fix it

Use `whereType` instead of `where((x) => x is Type)`.

=== Code examples

==== Noncompliant code example

[source,dart,diff-id=1,diff-type=noncompliant]
----
void foo(Iterable<Vehicle> iterable)
{
    iterable.where((x) => x is Car)              // Noncompliant
      .forEach((car) => print(car.model));

}
----

==== Compliant solution

[source,dart,diff-id=1,diff-type=compliant]
----
void foo(Iterable<Vehicle> iterable)
{
    iterable.whereType<Car>()
      .forEach((car) => print(car.model));

}
----

== Resources

* https://dart.dev/tools/linter-rules/prefer_iterable_whereType[Dart Lint rule]
