<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>The <code>ISerializable</code> interface gives you control over <em>how</em> your class is serialized, but does not itself make the class serializable. Such classes must also have the <code>[Serializable]</code> attribute.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-csharp" data-lang="csharp">public class Person : ISerializable {  // Noncompliant; [Serializable] attribute missing
  // ...
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-csharp" data-lang="csharp">[Serializable]
public class Person : ISerializable {
  // ...
}</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_4_feb_2015_141717_ann_campbell_wrote">on 4 Feb 2015, 14:17:17 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>Gendarme MissingSerializableAttributeOnISerializableTypeRule</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_13_apr_2015_104740_freddy_mallet_wrote">on 13 Apr 2015, 10:47:40 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>@Tamas, does this rule makes sense to you ? Thanks</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_22_may_2015_092353_tamas_vajk_wrote">on 22 May 2015, 09:23:53 Tamas Vajk wrote:</h3>
<div class="paragraph">
<p>\[~freddy.mallet] Yes, it makes sense. BUT it is already implemented by the Roslyn team:</p>
</div>
<div class="paragraph">
<p><a href="https://msdn.microsoft.com/en-us/library/ms182350.aspx" class="bare">https://msdn.microsoft.com/en-us/library/ms182350.aspx</a></p>
</div>
<div class="paragraph">
<p><a href="https://github.com/dotnet/roslyn/blob/master/src/Diagnostics/FxCop/Core/Usage/SerializationRulesDiagnosticAnalyzer.cs" class="bare">https://github.com/dotnet/roslyn/blob/master/src/Diagnostics/FxCop/Core/Usage/SerializationRulesDiagnosticAnalyzer.cs</a></p>
</div>
</div>
<div class="sect2">
<h3 id="_on_28_may_2015_124455_tamas_vajk_wrote">on 28 May 2015, 12:44:55 Tamas Vajk wrote:</h3>
<div class="paragraph">
<p>This rule duplicates CA2237 (https://msdn.microsoft.com/en-us/library/ms182350.aspx), which is already implemented by the Roslyn team.</p>
</div>
</div>
</div>
</div>