== How to fix it in Java Lang Package

=== Code examples

include::../../common/fix/code-rationale.adoc[]

==== Noncompliant code example

[source,kotlin,diff-id=1,diff-type=noncompliant]
----
import java.lang.Class

@RestController
class EffectController {
    @GetMapping("/filter/apply")
    @ResponseBody
    fun apply(@RequestParam("effect") effectName: String): ResponseEntity<String> {
        val effectClass = Class.forName(effectName) // Noncompliant
        val effectConstructor = effectClass.getConstructor()
        val effectObject = effectConstructor.newInstance()
        val applyMethod = effectClass.getMethod("applyFilter")
        val result =  applyMethod.invoke(effectObject) as Boolean
        if (result) {
            return ResponseEntity("Filter Applied", HttpStatus.OK)
        }
        return ResponseEntity("Filter Failure", HttpStatus.FORBIDDEN)
    }
}
----

==== Compliant solution

[source,kotlin,diff-id=1,diff-type=compliant]
----
import java.lang.Class

@RestController
class EffectController {
    companion object {
        private val EFFECT_ALLOW_LIST = setOf("SepiaEffect", "BlackAndWhiteEffect", "WaterColorEffect")
    }

    @GetMapping("/filter/apply")
    @ResponseBody
    fun apply(@RequestParam("effect") effectName: String): ResponseEntity<String> {
        if (!EFFECT_ALLOW_LIST.contains(effectName)) {
            return ResponseEntity("Filter Failure", HttpStatus.FORBIDDEN)
        }

        val effectClass = Class.forName(effectName)
        val effectConstructor = effectClass.getConstructor()
        val effectObject = effectConstructor.newInstance()
        val applyMethod = effectClass.getMethod("applyFilter")
        val result = applyMethod.invoke(effectObject) as Boolean
        if (result) {
            return ResponseEntity("Filter Applied", HttpStatus.OK)
        }
        return ResponseEntity("Filter Failure", HttpStatus.FORBIDDEN)
    }
}
----

=== How does this work?

include::../../common/fix/pre-approved-list.adoc[]

