= Invalid header level

Shared naming conventions allow teams to collaborate efficiently. This rule checks that all function names match a provided regular expression.

== Invalid header

== Noncompliant Code Example

With default provided regular expression: ``++^[a-z][a-zA-Z0-9]*$++``:

[source, unknown]
----
void DoSomething (void);
----

== See

=== Parameters

****
*format*
_STRING_
----
^([A-Z0-9_]*|[a-z0-9_]*)$
----

Regular expression used to check the [method|function|subroutine] names against
****
