This is an issue when `partial` is used as a return type without the `@` escape prefix in C# 14 or later.

== Why is this an issue?

Starting with C# 14, the `partial` keyword becomes a contextual keyword in more places within the language. This expansion means that `partial` can no longer be used directly as a return type identifier without escaping.

When the compiler encounters `partial` in a return type position, it now interprets it as a modifier rather than a type name. This causes compilation errors in code that previously worked in earlier C# versions.

The issue specifically affects method declarations where a custom type named `partial` is used as the return type. Other uses of the `partial` type (such as in variable declarations or constructor calls) are not affected by this change.

=== What is the potential impact?

This breaking change will cause compilation failures when upgrading existing codebases to C# 14. The compilation error prevents the application from building, requiring immediate attention to fix the syntax.

While this is not a runtime issue, it can block development and deployment processes until resolved. The fix is straightforward but must be applied to all affected method declarations.

== How to fix it

Add the `@` prefix before `partial` when used as a return type. This tells the compiler to treat `partial` as an identifier rather than a keyword.

=== Code examples

==== Noncompliant code example

[source,csharp,diff-id=1,diff-type=noncompliant]
----
class C
{
    partial F() => new partial(); // Noncompliant
}

class partial { }
----

==== Compliant solution

[source,csharp,diff-id=1,diff-type=compliant]
----
class C
{
    @partial F() => new partial();
}

class partial { }
----

== Resources

=== Documentation

 * C# Breaking Changes in .NET 10 - https://learn.microsoft.com/en-us/dotnet/csharp/whats-new/breaking-changes/compiler%20breaking%20changes%20-%20dotnet%2010[Official Microsoft documentation about breaking changes in C# 14, including the partial keyword restrictions]

 * C# Identifiers - https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/language-specification/lexical-structure#643-identifiers[C# language specification explaining identifier rules and escaping with the @ prefix]
