== Why is this an issue?

This rule leads to greater precision in the definition of local variables by making the developer intention about modifying the variable explicit. The ``++const++`` qualification shall be applied to the object pointed to, not to the pointer, since it is the object itself that is being protected.


=== Noncompliant code example

[source,cpp]
----
std::string& getString();
void myfunc()
{
  std::string& s = getString(); // Noncompliant
  if (s.size()) {
   std::cout << s;
  }
}
----


=== Compliant solution

[source,cpp]
----
std::string& getString();
void myfunc () { 
  const std::string& x = getString(); 
  if (s.size()) {
    std::cout << s;
  }
} 
----


== Resources

* MISRA C:2012, 8.13 - A pointer should point to a const-qualified type whenever possible

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

Mark "XXX" as const at every possible pointer level.


endif::env-github,rspecator-view[]
