<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Functions with <code>OUT</code> parameters are complex to understand. Indeed, it is impossible to tell, just by looking at the function call, whether an argument is a input or output. Moreover, functions with <code>OUT</code> parameters cannot be called from SQL. It is better to either break such functions up into smaller ones, which each return a single value, or to return several values at once, by combining them in a collection, record, type, or table row.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-sql" data-lang="sql">SET SERVEROUTPUT ON

CREATE TABLE employee(
  firstName VARCHAR2(42),
  name VARCHAR2(42),
  phone VARCHAR2(42)
);

INSERT INTO employee VALUES ('John', 'Smith', '+1');

DECLARE
  firstName VARCHAR2(42);
  name VARCHAR2(42);
  phone VARCHAR2(42);

  -- This DOES NOT return the employee name
  FUNCTION getEmployeeInfos(firstName OUT VARCHAR2, phone OUT VARCHAR2) RETURN VARCHAR2 AS -- Non-Compliant, confusing
    name VARCHAR2(42);
  BEGIN
    SELECT firstName, name, phone INTO firstName, name, phone FROM employee;
    RETURN name;
  END;
BEGIN
  name := getEmployeeInfos(firstName, phone);

  DBMS_OUTPUT.PUT_LINE('firstName: ' || firstName);
  DBMS_OUTPUT.PUT_LINE('name: ' || name);
  DBMS_OUTPUT.PUT_LINE('phone: ' || phone);
END;
/</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-sql" data-lang="sql">SET SERVEROUTPUT ON

CREATE TABLE employee(
  firstName VARCHAR2(42),
  name VARCHAR2(42),
  phone VARCHAR2(42)
);

INSERT INTO employee VALUES ('John', 'Smith', '+1');

DECLARE
  emp employee%ROWTYPE;

  FUNCTION getEmployeeInfos RETURN employee%ROWTYPE AS -- Compliant
    emp employee%ROWTYPE;
  BEGIN
    SELECT * INTO emp FROM employee;
    RETURN emp;
  END;
BEGIN
  emp := getEmployeeInfos;

  DBMS_OUTPUT.PUT_LINE('firstName: ' || emp.firstName);
  DBMS_OUTPUT.PUT_LINE('name: ' || emp.name);
  DBMS_OUTPUT.PUT_LINE('phone: ' || emp.phone);
END;
/

DROP TABLE employee;</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove the "OUT" parameter(s) from this function.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p><code>out|ref</code></p>
</div>
</div>
</div>
</div>