This rule raises an issue when a Django template uses the `{% extends %}` tag to reference itself, creating a circular dependency.

== Why is this an issue?

When overriding Django templates, developers sometimes accidentally create circular references by having a template extend itself instead of the original template.

Django's template system allows you to override templates by placing them in specific directories. For example, you can override `admin/base_site.html` by creating your own version at `templates/admin/base_site.html`. However, when extending the original template, you must reference it correctly.

If your override template extends `templates/admin/base_site.html`, it's actually extending itself, creating an infinite loop. Django's template loader will keep trying to load the same template repeatedly until it hits a recursion limit.

The correct approach is to extend `admin/base_site.html` (without the `templates/` prefix). Django's template loader will then resolve this to the original template from the application or Django's built-in templates, not your override.

=== What is the potential impact?

This issue will cause your Django application to crash with a `RecursionError` or `TemplateDoesNotExist` error when the problematic template is rendered. This can make entire pages or admin sections completely inaccessible, severely impacting the application's functionality.

== How to fix it

Use the relative template path in the extends tag, not the full path that includes your override directory. This allows Django's template loader to find the original template you want to extend.

=== Code examples

==== Noncompliant code example

[source,django,diff-id=1,diff-type=noncompliant]
----
{% comment %}
File: templates/admin/base_site.html
{% endcomment %}
{% extends "templates/admin/base_site.html" %} {# Noncompliant #}
{% block branding %}
  <h1>My Custom Admin</h1>
{% endblock %}
----

==== Compliant solution

[source,django,diff-id=1,diff-type=compliant]
----
{% comment %}
File: templates/admin/base_site.html
{% endcomment %}
{% extends "admin/base_site.html" %}
{% block branding %}
  <h1>My Custom Admin</h1>
{% endblock %}
----

== Resources

=== Documentation

 * Django Template Overriding - https://docs.djangoproject.com/en/stable/howto/overriding-templates/[Official Django documentation on how to properly override templates]

 * Django Template Extends Tag - https://docs.djangoproject.com/en/stable/ref/templates/builtins/#extends[Documentation for the extends template tag]
